diff --git src/kstatusnotifieritem.cpp src/kstatusnotifieritem.cpp
index f9bf460..029d5dc 100644
--- src/kstatusnotifieritem.cpp
+++ src/kstatusnotifieritem.cpp
@@ -701,8 +701,21 @@ void KStatusNotifierItemPrivate::init(const QString &extraId)
     if (title.isEmpty()) {
         title = QCoreApplication::applicationName();
     }
+#ifdef Q_OS_OSX
+    // OS X doesn't have texted separators so we emulate QAction::addSection():
+    // we first add an action with the desired text (title) and icon
+    titleAction = m->addAction(qApp->windowIcon(), title);
+    // this action should be disabled
+    titleAction->setEnabled(false);
+    // it seems reasonable to display the icon for this title item, despite general platform guidelines:
+    titleAction->setIconVisibleInMenu(true);
+    m->addAction(titleAction);
+    // now add a regular separator
+    m->addSeparator();
+#else
     titleAction = m->addSection(qApp->windowIcon(), title);
     m->setTitle(title);
+#endif
     q->setContextMenu(m);
 
     QAction *action = new QAction(q);
