diff --git src/notifybypopup.cpp src/notifybypopup.cpp
index c0050b2..f28ca16 100644
--- src/notifybypopup.cpp
+++ src/notifybypopup.cpp
@@ -183,7 +183,11 @@ NotifyByPopup::NotifyByPopup(QObject *parent)
 #ifdef Q_WS_WIN
         startfdo = true;
 #else
-        if (qEnvironmentVariableIsEmpty("KDE_FULL_SESSION")) {
+#ifndef Q_OS_OSX
+        // ignore KDE_FULL_SESSION on OS X, where it might be set only to enable full theming
+        if (qEnvironmentVariableIsEmpty("KDE_FULL_SESSION"))
+#endif
+        {
             QDBusMessage message = QDBusMessage::createMethodCall(QStringLiteral("org.freedesktop.DBus"),
                                                                   QStringLiteral("/org/freedesktop/DBus"),
                                                                   QStringLiteral("org.freedesktop.DBus"),
