diff --git a/src/notifybyaudio.cpp b/src/notifybyaudio.cpp
index 903e996..e4d75c0 100644
--- a/src/notifybyaudio.cpp
+++ b/src/notifybyaudio.cpp
@@ -27,6 +27,7 @@
 #include <QTextStream>
 #include <QUrl>
 #include <QString>
+#include <QApplication>
 
 #include "knotifyconfig.h"
 #include "knotification.h"
@@ -49,23 +50,30 @@ NotifyByAudio::~NotifyByAudio()
 
 void NotifyByAudio::notify(KNotification *notification, KNotifyConfig *config)
 {
-    if (!m_audioOutput) {
-        m_audioOutput = new Phonon::AudioOutput(Phonon::NotificationCategory, this);
-    }
     QString soundFilename = config->readEntry(QStringLiteral("Sound"));
     if (soundFilename.isEmpty()) {
-        qCWarning(LOG_KNOTIFICATIONS) << "Audio notification requested, but no sound file provided in notifyrc file, aborting audio notification";
-
+        qCWarning(LOG_KNOTIFICATIONS) << "Audio notification requested because" << notification->text()
+            << "/" << config->eventid
+            << ", but no sound file provided in file" << config->configfile->name() << ", aborting audio notification";
+        QApplication::beep();
         finish(notification);
         return;
     }
 
+    if (soundFilename.compare(QStringLiteral("beep"), Qt:CaseInsensitive) == 0) {
+        QApplication::beep();
+        finish(notification);
+        return;
+    }
+    if (!m_audioOutput) {
+        m_audioOutput = new Phonon::AudioOutput(Phonon::NotificationCategory, this);
+    }
     QUrl soundURL;
     const auto dataLocations = QStandardPaths::standardLocations(QStandardPaths::GenericDataLocation);
     foreach (const QString &dataLocation, dataLocations) {
         soundURL = QUrl::fromUserInput(soundFilename,
-                                       dataLocation + "/sounds",
-                                       QUrl::AssumeLocalFile);
+                                    dataLocation + "/sounds",
+                                    QUrl::AssumeLocalFile);
         if (soundURL.isLocalFile() && QFile::exists(soundURL.toLocalFile())) {
             break;
         } else if (!soundURL.isLocalFile() && soundURL.isValid()) {
@@ -75,6 +83,7 @@ void NotifyByAudio::notify(KNotification *notification, KNotifyConfig *config)
     }
     if (soundURL.isEmpty()) {
         qCWarning(LOG_KNOTIFICATIONS) << "Audio notification requested, but sound file from notifyrc file was not found, aborting audio notification";
+        QApplication::beep();
         finish(notification);
         return;
     }
