--- src/xz/Makefile.am.orig	2015-09-29 12:57:36.000000000 +0200
+++ src/xz/Makefile.am	2016-11-02 21:51:36.000000000 +0100
@@ -5,7 +5,7 @@
 ## You can do whatever you want with this file.
 ##
 
-bin_PROGRAMS = xz
+bin_PROGRAMS = xz xz_r
 
 xz_SOURCES = \
 	args.c \
@@ -50,14 +50,21 @@
 	-I$(top_builddir)/lib
 
 xz_LDADD = $(top_builddir)/src/liblzma/liblzma.la
+xz_r_LDADD = $(top_builddir)/src/liblzma/liblzma.la
 
 if COND_GNULIB
 xz_LDADD += $(top_builddir)/lib/libgnu.a
+xz_r_LDADD += $(top_builddir)/lib/libgnu.a
 endif
 
 # libgnu.a may need these libs, so this must be after libgnu.a.
 xz_LDADD += $(LTLIBINTL)
 
+xz_r_SOURCES = $(xz_SOURCES)
+xz_r_CPPFLAGS = $(xz_CPPFLAGS)
+xz_r_LDADD += @PREFIX@/lib/libintl.a @PREFIX@/lib/libiconv.a
+xz_r_LDFLAGS = -static-libtool-libs -Wl,-framework -Wl,CoreFoundation
+
 
 # Windows resource compiler support
 .rc.o:
