--- src/xz/orig.Makefile.am	2016-12-30 12:08:20.000000000 +0100
+++ src/xz/Makefile.am	2017-02-12 12:46:40.341599925 +0100
@@ -5,7 +5,7 @@
 ## You can do whatever you want with this file.
 ##
 
-bin_PROGRAMS = xz
+bin_PROGRAMS = xz xz_r
 
 xz_SOURCES = \
 	args.c \
@@ -54,14 +54,20 @@
 	-I$(top_builddir)/lib
 
 xz_LDADD = $(top_builddir)/src/liblzma/liblzma.la $(CAPSICUM_LIB)
+xz_r_LDADD = $(top_builddir)/src/liblzma/liblzma.la $(CAPSICUM_LIB)
 
 if COND_GNULIB
 xz_LDADD += $(top_builddir)/lib/libgnu.a
+xz_r_LDADD += $(top_builddir)/lib/libgnu.a
 endif
 
 # libgnu.a may need these libs, so this must be after libgnu.a.
 xz_LDADD += $(LTLIBINTL)
 
+xz_r_SOURCES = $(xz_SOURCES)
+xz_r_CPPFLAGS = $(xz_CPPFLAGS)
+xz_r_LDADD += @PREFIX@/lib/libintl.a @PREFIX@/lib/libiconv.a
+xz_r_LDFLAGS = -static-libtool-libs -Wl,-framework -Wl,CoreFoundation
 
 # Windows resource compiler support
 .rc.o:
