diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index c0d6c18..581b7bf 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -42,6 +42,7 @@ add_subdirectory(kbookmarkmodel)
 set(kbookmarkmerger_SRCS kbookmarkmerger.cpp )
 
 add_executable(kbookmarkmerger ${kbookmarkmerger_SRCS})
+ecm_mark_nongui_executable(kbookmarkmerger)
 
 target_link_libraries(kbookmarkmerger
    KF5::KIOCore
@@ -78,6 +79,7 @@ qt5_add_dbus_interface( keditbookmarks_SRCS ${toplevel_xml} toplevel_interface )
 kconfig_add_kcfg_files(keditbookmarks_SRCS settings.kcfgc )
 
 add_executable( keditbookmarks ${keditbookmarks_SRCS})
+ecm_mark_nongui_executable( keditbookmarks )
 
 target_link_libraries(keditbookmarks kbookmarkmodel_private5
    KF5::Parts

