diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index c0d6c18..f88e826 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -79,7 +79,7 @@ kconfig_add_kcfg_files(keditbookmarks_SRCS settings.kcfgc )
 
 add_executable( keditbookmarks ${keditbookmarks_SRCS})
 
-target_link_libraries(keditbookmarks kbookmarkmodel_private
+target_link_libraries(keditbookmarks kbookmarkmodel_private5
    KF5::Parts
    KF5::Bookmarks
    KF5::WindowSystem
diff --git a/src/kbookmarkmodel/CMakeLists.txt b/src/kbookmarkmodel/CMakeLists.txt
index d013328..0f3d5d6 100644
--- a/src/kbookmarkmodel/CMakeLists.txt
+++ b/src/kbookmarkmodel/CMakeLists.txt
@@ -8,8 +8,8 @@ set(kbookmarkmodel_SRCS
   view.cpp
 )
 
-add_library(kbookmarkmodel_private SHARED ${kbookmarkmodel_SRCS})
-target_link_libraries(kbookmarkmodel_private
+add_library(kbookmarkmodel_private5 SHARED ${kbookmarkmodel_SRCS})
+target_link_libraries(kbookmarkmodel_private5
 PUBLIC
    Qt5::Widgets
    KF5::XmlGui # KActionCollection
@@ -19,12 +19,12 @@ PRIVATE
 )
 
 include(GenerateExportHeader)
-generate_export_header(kbookmarkmodel_private BASE_NAME kbookmarkmodel EXPORT_FILE_NAME kbookmarkmodel_export.h)
+generate_export_header(kbookmarkmodel_private5 BASE_NAME kbookmarkmodel EXPORT_FILE_NAME kbookmarkmodel_export.h)
 
-set_target_properties(kbookmarkmodel_private PROPERTIES
+set_target_properties(kbookmarkmodel_private5 PROPERTIES
      VERSION ${KEDITBOOKMARKS_VERSION}
      SOVERSION "6"
 )
 
 
-install(TARGETS kbookmarkmodel_private ${KDE_INSTALL_TARGETS_DEFAULT_ARGS})
+install(TARGETS kbookmarkmodel_private5 ${KDE_INSTALL_TARGETS_DEFAULT_ARGS})
diff --git a/src/kbookmarkmodel/tests/CMakeLists.txt b/src/kbookmarkmodel/tests/CMakeLists.txt
index 6175eaa..3977e25 100644
--- a/src/kbookmarkmodel/tests/CMakeLists.txt
+++ b/src/kbookmarkmodel/tests/CMakeLists.txt
@@ -7,5 +7,5 @@
 set(kbookmarkmodeltest_SRCS kbookmarkmodeltest.cpp)
 ecm_add_test(${kbookmarkmodeltest_SRCS}
     TEST_NAME kbookmarkmodeltest
-    LINK_LIBRARIES kbookmarkmodel_private KF5::Bookmarks Qt5::Test
+    LINK_LIBRARIES kbookmarkmodel_private5 KF5::Bookmarks Qt5::Test
 )
