diff --git a/src/main.cpp b/src/main.cpp
index 6125393..275f19a 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -115,7 +115,7 @@ extern "C" Q_DECL_EXPORT int kdemain(int argc, char **argv)
     aboutData.addAuthor(i18n("Alexander Kellett"), i18n("Author"), "lypanov@kde.org");
     KAboutData::setApplicationData(aboutData);
 
-    app.setWindowIcon(QIcon::fromTheme(QStringLiteral("bookmarks-organize")));
+    app.setWindowIcon(QIcon::fromTheme(QStringLiteral("bookmarks-organize"), app.windowIcon()));
 
     QCommandLineParser parser;
     parser.setApplicationDescription(aboutData.shortDescription());

