--- libkcddb/musicbrainz/orig.musicbrainzlookup.cpp	2017-08-21 17:58:30.000000000 +0200
+++ libkcddb/musicbrainz/musicbrainzlookup.cpp	2017-08-21 17:58:43.000000000 +0200
@@ -65,6 +65,7 @@
     // Code adapted from libmusicbrainz/examples/cdlookup.cc
 
     try {
+        qDebug() << Q_FUNC_INFO << "query discid" << discId;
       MusicBrainz5::CMetadata Metadata=Query.Query("discid",discId.toLatin1().constData());
 
       if (Metadata.Disc() && Metadata.Disc()->ReleaseList())
@@ -179,60 +180,70 @@
 
     catch (MusicBrainz5::CConnectionError& Error)
     {
-      qDebug() << "Connection Exception: '" << Error.what() << "'";
-      qDebug() << "LastResult: " << Query.LastResult();
-      qDebug() << "LastHTTPCode: " << Query.LastHTTPCode();
-      qDebug() << "LastErrorMessage: " << QString::fromUtf8(Query.LastErrorMessage().c_str());
+      qWarning() << "Connection Exception: '" << Error.what() << "'";
+      qWarning() << "LastResult: " << Query.LastResult();
+      qWarning() << "LastHTTPCode: " << Query.LastHTTPCode();
+      qWarning() << "LastErrorMessage: " << QString::fromUtf8(Query.LastErrorMessage().c_str());
 
       return ServerError;
     }
 
     catch (MusicBrainz5::CTimeoutError& Error)
     {
-      qDebug() << "Timeout Exception: '" << Error.what() << "'";
-      qDebug() << "LastResult: " << Query.LastResult();
-      qDebug() << "LastHTTPCode: " << Query.LastHTTPCode();
-      qDebug() << "LastErrorMessage: " << QString::fromUtf8(Query.LastErrorMessage().c_str());
+      qWarning() << "Timeout Exception: '" << Error.what() << "'";
+      qWarning() << "LastResult: " << Query.LastResult();
+      qWarning() << "LastHTTPCode: " << Query.LastHTTPCode();
+      qWarning() << "LastErrorMessage: " << QString::fromUtf8(Query.LastErrorMessage().c_str());
 
       return ServerError;
     }
 
     catch (MusicBrainz5::CAuthenticationError& Error)
     {
-      qDebug() << "Authentication Exception: '" << Error.what() << "'";
-      qDebug() << "LastResult: " << Query.LastResult();
-      qDebug() << "LastHTTPCode: " << Query.LastHTTPCode();
-      qDebug() << "LastErrorMessage: " << QString::fromUtf8(Query.LastErrorMessage().c_str());
+      qWarning() << "Authentication Exception: '" << Error.what() << "'";
+      qWarning() << "LastResult: " << Query.LastResult();
+      qWarning() << "LastHTTPCode: " << Query.LastHTTPCode();
+      qWarning() << "LastErrorMessage: " << QString::fromUtf8(Query.LastErrorMessage().c_str());
 
       return ServerError;
     }
 
     catch (MusicBrainz5::CFetchError& Error)
     {
-      qDebug() << "Fetch Exception: '" << Error.what() << "'";
-      qDebug() << "LastResult: " << Query.LastResult();
-      qDebug() << "LastHTTPCode: " << Query.LastHTTPCode();
-      qDebug() << "LastErrorMessage: " << QString::fromUtf8(Query.LastErrorMessage().c_str());
+      qWarning() << "Fetch Exception: '" << Error.what() << "'";
+      qWarning() << "LastResult: " << Query.LastResult();
+      qWarning() << "LastHTTPCode: " << Query.LastHTTPCode();
+      qWarning() << "LastErrorMessage: " << QString::fromUtf8(Query.LastErrorMessage().c_str());
 
       return ServerError;
     }
 
     catch (MusicBrainz5::CRequestError& Error)
     {
-      qDebug() << "Request Exception: '" << Error.what() << "'";
-      qDebug() << "LastResult: " << Query.LastResult();
-      qDebug() << "LastHTTPCode: " << Query.LastHTTPCode();
-      qDebug() << "LastErrorMessage: " << QString::fromUtf8(Query.LastErrorMessage().c_str());
+      qWarning() << "Request Exception: '" << Error.what() << "'";
+      qWarning() << "LastResult: " << Query.LastResult();
+      qWarning() << "LastHTTPCode: " << Query.LastHTTPCode();
+      qWarning() << "LastErrorMessage: " << QString::fromUtf8(Query.LastErrorMessage().c_str());
 
       return ServerError;
     }
 
     catch (MusicBrainz5::CResourceNotFoundError& Error)
     {
-      qDebug() << "ResourceNotFound Exception: '" << Error.what() << "'";
-      qDebug() << "LastResult: " << Query.LastResult();
-      qDebug() << "LastHTTPCode: " << Query.LastHTTPCode();
-      qDebug() << "LastErrorMessage: " << QString::fromUtf8(Query.LastErrorMessage().c_str());
+      qWarning() << "ResourceNotFound Exception: '" << Error.what() << "'";
+      qWarning() << "LastResult: " << Query.LastResult();
+      qWarning() << "LastHTTPCode: " << Query.LastHTTPCode();
+      qWarning() << "LastErrorMessage: " << QString::fromUtf8(Query.LastErrorMessage().c_str());
+
+      return ServerError;
+    }
+
+    catch (const std::exception& Error)
+    {
+      qWarning() << "Exception: '" << Error.what() << "'";
+      qWarning() << "LastResult: " << Query.LastResult();
+      qWarning() << "LastHTTPCode: " << Query.LastHTTPCode();
+      qWarning() << "LastErrorMessage: " << QString::fromUtf8(Query.LastErrorMessage().c_str());
 
       return ServerError;
     }
