--- plugins/wav/orig.encodercda.cpp	2017-06-20 14:55:21.000000000 +0200
+++ plugins/wav/encodercda.cpp	2017-08-12 15:55:10.000000000 +0200
@@ -43,9 +43,7 @@
 extern "C"
 {
   //cdda_interface.h in cdparanoia 10.2 has a member called 'private' which the C++ compiler doesn't like
-  #define private _private
-  #include <cdda_interface.h>
-  #undef private
+  #include "cdda_interface_local.h"
 }
 
 inline int16_t swap16 (int16_t i)
--- orig.audiocd.cpp	2017-06-20 14:55:21.000000000 +0200
+++ audiocd.cpp	2017-08-12 16:54:12.000000000 +0200
@@ -28,10 +28,8 @@
 extern "C"
 {
 	//cdda_interface.h in cdparanoia 10.2 has a member called 'private' which the C++ compiler doesn't like
-	#define private _private
-	#include <cdda_interface.h>
+	#include "cdda_interface_local.h"
 	#include <cdda_paranoia.h>
-	#undef private
 	void paranoiaCallback(long, int);
 
 	KDE_EXPORT int kdemain(int argc, char ** argv);
@@ -806,12 +804,16 @@
 {
 	const QByteArray device(QFile::encodeName(d->device));
 
+#ifndef Q_OS_MACOS
 	if (device.isEmpty())
 		return 0;
 
 	struct cdrom_drive * drive = 0;
 
 	drive = cdda_identify(device, CDDA_MESSAGE_FORGETIT, 0);
+#else
+	struct cdrom_drive * drive = cdda_find_a_cdrom(CDDA_MESSAGE_PRINTIT, 0);
+#endif
 
 	if (0 == drive) {
 		kDebug(7117) << "Can't find an audio CD on: \"" << d->device << "\"";
