diff --git src/lib/tools/orig.disablewindowtabbbing.mm src/lib/tools/disablewindowtabbbing.mm
index e24443a..96aee8e 100644
--- src/lib/tools/orig.disablewindowtabbbing.mm
+++ src/lib/tools/disablewindowtabbbing.mm
@@ -19,6 +19,8 @@
 #import <AppKit/AppKit.h>
 #import <Availability.h>
 
+#include <QtCore/private/qglobal_p.h>
+
 // code taken from: https://www.mail-archive.com/interest@qt-project.org/msg23593.html
 // Disables auto window tabbing where supported, otherwise a no-op.
 void disableWindowTabbing()
