diff --git src/lib/lib.pro.orig src/lib/lib.pro
index 880ea70..b8aebbe 100644
--- src/lib/lib.pro.orig
+++ src/lib/lib.pro
@@ -468,7 +468,7 @@ RESOURCES += \
         QT *= x11extras
     }
 
-    LIBS += -lcrypto
+    LIBS += @SSLPREFIX@/lib/libcrypto.so
 }
 
 win32 {
@@ -484,12 +484,12 @@ os2 {
 }
 
 mac {
-    # homebrew openssl
-    BREW_OPENSSL = /usr/local$$system("readlink `brew --prefix openssl` | sed 's/..//'")
-    INCLUDEPATH += $$BREW_OPENSSL/include
-    LIBS += -L$$BREW_OPENSSL/lib
+#     # homebrew openssl
+#     BREW_OPENSSL = /usr/local$$system("readlink `brew --prefix openssl` | sed 's/..//'")
+#     INCLUDEPATH += $$BREW_OPENSSL/include
+#     LIBS += -L$$BREW_OPENSSL/lib
 
-    LIBS += -lcrypto -framework CoreServices
+    LIBS += @SSLPREFIX@/lib/libcrypto.dylib -framework CoreServices
 
     OBJECTIVE_SOURCES += tools/disablewindowtabbbing.mm
     LIBS += -framework AppKit
