--- src/lib/tools/orig.disablewindowtabbbing.mm	2017-11-01 17:15:12.000000000 +0100
+++ src/lib/tools/disablewindowtabbbing.mm	2017-12-02 18:52:21.000000000 +0100
@@ -23,6 +23,7 @@
 void disableWindowTabbing()
 {
     if ([NSWindow respondsToSelector:@selector(allowsAutomaticWindowTabbing)]) {
-        NSWindow.allowsAutomaticWindowTabbing = NO;
+        [NSWindow setAllowsAutomaticWindowTabbing:NO];
     }
 }
+
