--- src/lib/lib.pro.orig	2017-11-01 17:15:12.000000000 +0100
+++ src/lib/lib.pro	2017-12-02 17:49:23.000000000 +0100
@@ -487,10 +487,10 @@
 }
 
 mac {
-    # homebrew openssl
-    BREW_OPENSSL = /usr/local$$system("readlink `brew --prefix openssl` | sed 's/..//'")
-    INCLUDEPATH += $$BREW_OPENSSL/include
-    LIBS += -L$$BREW_OPENSSL/lib
+#     # homebrew openssl
+#     BREW_OPENSSL = /usr/local$$system("readlink `brew --prefix openssl` | sed 's/..//'")
+#     INCLUDEPATH += $$BREW_OPENSSL/include
+#     LIBS += -L$$BREW_OPENSSL/lib
 
     LIBS += -lcrypto -framework CoreServices
 
