diff --git src/lib/webengine/orig.loadrequest.h src/lib/webengine/loadrequest.h
index a486a6d..2536d2d 100644
--- src/lib/webengine/orig.loadrequest.h
+++ src/lib/webengine/loadrequest.h
@@ -19,7 +19,9 @@
 #define LOADREQUEST_H
 
 #include <QUrl>
+#if QTWEBENGINECORE_VERSION >= QT_VERSION_CHECK(5, 9, 0)
 #include <QWebEngineHttpRequest>
+#endif
 
 #include "qzcommon.h"
 
@@ -50,7 +52,9 @@ public:
     QByteArray data() const;
     void setData(const QByteArray &data);
 
+#if QTWEBENGINECORE_VERSION >= QT_VERSION_CHECK(5, 9, 0)
     QWebEngineHttpRequest webRequest() const;
+#endif
 
 private:
     QUrl m_url;
diff --git src/lib/webengine/orig.loadrequest.cpp src/lib/webengine/loadrequest.cpp
index 8342f9f..85c76b8 100644
--- src/lib/webengine/orig.loadrequest.cpp
+++ src/lib/webengine/loadrequest.cpp
@@ -84,9 +84,11 @@ void LoadRequest::setData(const QByteArray &data)
     m_data = data;
 }
 
+#if QTWEBENGINECORE_VERSION >= QT_VERSION_CHECK(5, 9, 0)
 QWebEngineHttpRequest LoadRequest::webRequest() const
 {
     QWebEngineHttpRequest req(m_url, m_operation == GetOperation ? QWebEngineHttpRequest::Get : QWebEngineHttpRequest::Post);
     req.setPostData(m_data);
     return req;
 }
+#endif
diff --git src/lib/webengine/orig.webview.cpp src/lib/webengine/webview.cpp
index 9450c10..7d37405 100644
--- src/lib/webengine/orig.webview.cpp
+++ src/lib/webengine/webview.cpp
@@ -1281,7 +1281,14 @@ bool WebView::focusNextPrevChild(bool next)
 
 void WebView::loadRequest(const LoadRequest &req)
 {
+#if QTWEBENGINECORE_VERSION >= QT_VERSION_CHECK(5, 9, 0)
     QWebEngineView::load(req.webRequest());
+#else
+    if (req.operation() == LoadRequest::GetOperation)
+        load(req.url());
+    else
+        page()->runJavaScript(Scripts::sendPostData(req.url(), req.data()), WebPage::SafeJsWorld);
+#endif
 }
 
 bool WebView::eventFilter(QObject *obj, QEvent *event)
