--- src/orig.defines.pri	2016-06-08 16:24:44.000000000 +0200
+++ src/defines.pri	2016-07-02 01:23:29.000000000 +0200
@@ -47,7 +47,7 @@
 }
 
 # QtDBus not available on Mac
-mac: DEFINES *= DISABLE_DBUS
+# mac: DEFINES *= DISABLE_DBUS
 
 haiku-* {
     DEFINES *= DISABLE_DBUS
--- src/lib/orig.lib.pro	2016-06-08 16:24:44.000000000 +0200
+++ src/lib/lib.pro	2016-07-02 01:23:54.000000000 +0200
@@ -479,10 +479,10 @@
 }
 
 mac {
-    # homebrew openssl
-    BREW_OPENSSL = $$system("brew --prefix openssl")
-    INCLUDEPATH += $$BREW_OPENSSL/include
-    LIBS += -L$$BREW_OPENSSL/lib
+#    # homebrew openssl
+#    BREW_OPENSSL = $$system("brew --prefix openssl")
+#    INCLUDEPATH += $$BREW_OPENSSL/include
+#    LIBS += -L$$BREW_OPENSSL/lib
 
     LIBS += -lcrypto -framework CoreServices
 }
