--- mac/orig.macdeploy.sh	2016-07-02 00:55:12.000000000 +0200
+++ mac/macdeploy.sh	2016-07-02 02:20:56.000000000 +0200
@@ -29,13 +29,13 @@
 # fix plugins
 for plugin in QupZilla.app/Contents/Resources/plugins/*.dylib
 do
- install_name_tool -change $LIBRARY_NAME @executable_path/$LIBRARY_NAME $plugin
+ install_name_tool -change $LIBRARY_NAME @loader_path/../../MacOS/$LIBRARY_NAME $plugin
 done
 
 # run macdeployqt
 $MACDEPLOYQT QupZilla.app
 
 # create final dmg image
-cd ../mac
-./create_dmg.sh
+# cd ../mac
+# ./create_dmg.sh
 
