--- src/lib/tools/orig.disablewindowtabbbing.mm	2017-11-01 17:15:12.000000000 +0100
+++ src/lib/tools/disablewindowtabbbing.mm	2017-12-02 22:56:10.000000000 +0100
@@ -18,11 +18,15 @@
 * ============================================================ */
 #import <AppKit/AppKit.h>
 
+#include <QtCore/private/qglobal_p.h>
+
 // code taken from: https://www.mail-archive.com/interest@qt-project.org/msg23593.html
 // Disables auto window tabbing where supported, otherwise a no-op.
 void disableWindowTabbing()
 {
-    if ([NSWindow respondsToSelector:@selector(allowsAutomaticWindowTabbing)]) {
+#if __MAC_OS_X_VERSION_MAX_ALLOWED >= 101200
+    if (__builtin_available(macOS 10.12, *)) {
         NSWindow.allowsAutomaticWindowTabbing = NO;
     }
+#endif
 }
