--- mtp/kiod_module/orig.mtpstorage.cpp	2019-07-06 11:31:14.000000000 +0200
+++ mtp/kiod_module/mtpstorage.cpp	2019-07-30 00:23:40.000000000 +0200
@@ -545,7 +545,11 @@
         int result = 1;
         QT_STATBUF srcBuf;
         if (QT_FSTAT(descriptor.fileDescriptor(), &srcBuf) != -1) {
+#ifdef Q_OS_MACOS
+            const QDateTime lastModified = QDateTime::fromSecsSinceEpoch(srcBuf.st_mtimespec.tv_sec);
+#else
             const QDateTime lastModified = QDateTime::fromSecsSinceEpoch(srcBuf.st_mtim.tv_sec);
+#endif
 
             LIBMTP_file_t *file = LIBMTP_new_file_t();
             file->parent_id = parentId;
