--- Source/WebCore/platform/graphics/opengl/orig.GraphicsContextGLOpenGLCommon.cpp	2021-05-03 12:41:16.991827472 +0200
+++ Source/WebCore/platform/graphics/opengl/GraphicsContextGLOpenGLCommon.cpp	2021-05-03 12:41:28.242071936 +0200
@@ -2029,11 +2029,13 @@
     getExtensions().vertexAttribDivisor(index, divisor);
 }
 
-#if HAVE(OPENGL_4) && ENABLE(WEBGL2)
+#if USE(OPENGL) && ENABLE(WEBGL2)
 void GraphicsContextGLOpenGL::primitiveRestartIndex(GCGLuint index)
 {
     makeContextCurrent();
+#if HAVE(OPENGL_4)
     ::glPrimitiveRestartIndex(index);
+#endif
 }
 #endif
 
