diff --git a/src/pextlib1.0/Pextlib.c b/src/pextlib1.0/Pextlib.c
index f9c88e8..77b3116 100644
--- a/src/pextlib1.0/Pextlib.c
+++ b/src/pextlib1.0/Pextlib.c
@@ -1076,6 +1076,7 @@ int Pextlib_Init(Tcl_Interp *interp)
 	Tcl_CreateObjCommand(interp, "adv-flock", AdvFlockCmd, NULL, NULL);
 	Tcl_CreateObjCommand(interp, "readdir", ReaddirCmd, NULL, NULL);
 	Tcl_CreateObjCommand(interp, "strsed", StrsedCmd, NULL, NULL);
+	Tcl_CreateObjCommand(interp, "mkstemp", MkstempCmd, NULL, NULL);
 	Tcl_CreateObjCommand(interp, "mktemp", MktempCmd, NULL, NULL);
 	Tcl_CreateObjCommand(interp, "mkdtemp", MkdtempCmd, NULL, NULL);
 	Tcl_CreateObjCommand(interp, "existsuser", ExistsuserCmd, NULL, NULL);
diff --git a/src/pextlib1.0/mktemp.h b/src/pextlib1.0/mktemp.h
index 85ffda0..4314211 100644
--- a/src/pextlib1.0/mktemp.h
+++ b/src/pextlib1.0/mktemp.h
@@ -31,3 +31,4 @@
 
 int MkdtempCmd(ClientData, Tcl_Interp *, int, Tcl_Obj *const objv[]);
 int MktempCmd(ClientData, Tcl_Interp *, int, Tcl_Obj *const objv[]);
+int MkstempCmd(ClientData, Tcl_Interp *, int, Tcl_Obj *const objv[]);
