diff --git a/configure.ac b/configure.ac
index ab77f37..cd4dbe9 100644
--- a/configure.ac
+++ b/configure.ac
diff --git a/configure.ac b/configure.ac
index ab77f37..78ce736 100644
--- a/configure.ac
+++ b/configure.ac
@@ -399,8 +399,10 @@ INCLUDES="-I.. -I. $INCLUDES"
 AC_SUBST(INCLUDES)
 
 ## sqlite3
-SQLITE3_CPPFLAGS=-I`echo ${TCL_SRC_DIR}/pkgs/sqlite3*/compat/sqlite3`
-SQLITE3_LIBNAME=`basename ${TCL_SRC_DIR}/pkgs/sqlite3*`
+# headers under $prefix/include as usual
+SQLITE3_CPPFLAGS=""
+# libsqlite3 will be in $prefix/lib; also business as usual
+SQLITE3_LIBNAME="-lsqlite3"
 AC_SUBST(SQLITE3_CPPFLAGS)
 AC_SUBST(SQLITE3_LIBNAME)
 
diff --git a/Mk/macports.autoconf.mk.in b/Mk/macports.autoconf.mk.in
index bbf3ef9..beb9097 100644
--- a/Mk/macports.autoconf.mk.in
+++ b/Mk/macports.autoconf.mk.in
@@ -27,7 +27,7 @@ TCL_STUB_LIB_SPEC	= @TCL_STUB_LIB_SPEC@
 LIBS			= @LIBS@
 READLINE_LIBS		= @READLINE_LIBS@
 MD5_LIBS		= @MD5_LIBS@
-SQLITE3_LIBS		= @TCL_BIN_DIR@/pkgs/@SQLITE3_LIBNAME@/lib@SQLITE3_LIBNAME@$(SHLIB_SUFFIX)
+SQLITE3_LIBS		= @SQLITE3_LIBNAME@
 CURL_LIBS		= @LDFLAGS_LIBCURL@
 INSTALL			= @INSTALL@
 MTREE			= @MTREE@
