diff --git a/configure.ac b/configure.ac
index ab77f3746..a5a604ade 100644
--- a/configure.ac
+++ b/configure.ac
@@ -14,8 +14,8 @@ AC_MSG_RESULT([$MACPORTS_VERSION])
 AC_SUBST(MACPORTS_VERSION)
 
 # Check user PATH for MP paths, and remove as necessary.
-MP_PATH_SCAN
-MP_FLAGS_SCAN
+# MP_PATH_SCAN
+# MP_FLAGS_SCAN
 
 AC_PATH_PROG(SW_VERS, sw_vers)
 AC_PATH_PROG(DEFAULTS, defaults)
diff --git a/Makefile.in b/Makefile.in
index c6b8391..c72c466 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -2,7 +2,6 @@
 srcdir      = @srcdir@
 VPATH       = @srcdir@
 
-PATH		= @PATH_CLEANED@
 SUBDIR		= vendor src tests doc
 DISTDIR		= dist
 DISTVER		=
