diff --git a/qtbase/src/platformsupport/fontdatabases/mac/qcoretextfontdatabase.mm b/qtbase/src/platformsupport/fontdatabases/mac/qcoretextfontdatabase.mm
index 237e8a8..ce3c50b 100644
--- a/qtbase/src/platformsupport/fontdatabases/mac/qcoretextfontdatabase.mm
+++ b/qtbase/src/platformsupport/fontdatabases/mac/qcoretextfontdatabase.mm
@@ -220,6 +220,7 @@ bool QCoreTextFontDatabase::populateFamilyAliases()
     if (m_hasPopulatedAliases)
         return false;
 
+    QMacAutoReleasePool pool;
     QCFType<CFArrayRef> familyNames = availableFamilyNames();
     for (NSString *familyName in familyNames.as<const NSArray *>()) {
         NSFontManager *fontManager = [NSFontManager sharedFontManager];
diff --git a/qtbase/src/plugins/platforms/cocoa/qcocoanativeinterface.mm b/qtbase/src/plugins/platforms/cocoa/qcocoanativeinterface.mm
index 26ab07f..ee762f6 100644
--- a/qtbase/src/plugins/platforms/cocoa/qcocoanativeinterface.mm
+++ b/qtbase/src/plugins/platforms/cocoa/qcocoanativeinterface.mm
@@ -261,6 +261,7 @@ void QCocoaNativeInterface::setDockMenu(QPlatformMenu *platformMenu)
 
 void *QCocoaNativeInterface::qMenuToNSMenu(QPlatformMenu *platformMenu)
 {
+    QMacAutoReleasePool pool;
     QCocoaMenu *cocoaPlatformMenu = static_cast<QCocoaMenu *>(platformMenu);
     NSMenu *menu = cocoaPlatformMenu->nsMenu();
     return reinterpret_cast<void *>(menu);
diff --git a/qtbase/src/plugins/platforms/cocoa/qcocoawindow.mm b/qtbase/src/plugins/platforms/cocoa/qcocoawindow.mm
index 5cd4beb..4dbb8ee 100644
--- a/qtbase/src/plugins/platforms/cocoa/qcocoawindow.mm
+++ b/qtbase/src/plugins/platforms/cocoa/qcocoawindow.mm
@@ -2173,6 +2173,8 @@ void QCocoaWindow::applyContentBorderThickness(NSWindow *window)
     if (!window)
         return;
 
+    QMacAutoReleasePool pool;
+
     if (!m_drawContentBorderGradient) {
         [window setStyleMask:[window styleMask] & ~NSTexturedBackgroundWindowMask];
         [[[window contentView] superview] setNeedsDisplay:YES];
