diff --git a/qtbase/src/plugins/platforms/xcb/qxcbintegration.cpp b/qtbase/src/plugins/platforms/xcb/qxcbintegration.cpp
index c9ecdce..2f540e6 100644
--- a/qtbase/src/plugins/platforms/xcb/qxcbintegration.cpp
+++ b/qtbase/src/plugins/platforms/xcb/qxcbintegration.cpp
@@ -423,6 +423,11 @@ QVariant QXcbIntegration::styleHint(QPlatformIntegration::StyleHint hint) const
         return false;
     case QPlatformIntegration::ReplayMousePressOutsidePopup:
         return false;
+
+    case QPlatformIntegration::FontSmoothingGamma:
+        return qreal(0.95);
+        break;
+
     default:
         break;
     }
