diff --git a/qtbase/src/plugins/platforms/xcb/qxcbintegration.cpp b/qtbase/src/plugins/platforms/xcb/qxcbintegration.cpp
index c9ecdceb0de0f6071519086ed073f5ea0e46b303..7f611b1240dfb8f8a6eba4ce83e94a3003fda757 100644
--- a/qtbase/src/plugins/platforms/xcb/qxcbintegration.cpp
+++ b/qtbase/src/plugins/platforms/xcb/qxcbintegration.cpp
@@ -118,6 +118,8 @@ QXcbIntegration::QXcbIntegration(const QStringList &parameters, int &argc, char
     , m_instanceName(0)
     , m_canGrab(true)
     , m_defaultVisualId(UINT_MAX)
+//     , m_fontSmoothingGamma(QPlatformIntegration::styleHint(QPlatformIntegration::FontSmoothingGamma).toReal())
+    , m_fontSmoothingGamma(0.93)
 {
     m_instance = this;
     qApp->setAttribute(Qt::AA_CompressHighFrequencyEvents, true);
@@ -128,6 +130,14 @@ QXcbIntegration::QXcbIntegration(const QStringList &parameters, int &argc, char
 #endif
     m_nativeInterface.reset(new QXcbNativeInterface);
 
+    if (qEnvironmentVariableIsSet("QT_XCB_FONT_GAMMA")) {
+        bool ok = false;
+        qreal fontgamma = qgetenv("QT_XCB_FONT_GAMMA").toDouble(&ok);
+        if (ok) {
+            m_fontSmoothingGamma = fontgamma;
+        }
+    }
+
     // Parse arguments
     const char *displayName = 0;
     bool noGrabArg = false;
@@ -152,6 +162,13 @@ QXcbIntegration::QXcbIntegration(const QStringList &parameters, int &argc, char
                 if (!ok)
                     m_defaultVisualId = UINT_MAX;
             }
+            else if (arg == "-fontgamma" && i < argc - 1) {
+                bool ok = false;
+                qreal fontgamma = QByteArray(argv[++i]).toDouble(&ok);
+                if (ok) {
+                    m_fontSmoothingGamma = fontgamma;
+                }
+            }
             else
                 argv[j++] = argv[i];
         }
@@ -423,6 +440,11 @@ QVariant QXcbIntegration::styleHint(QPlatformIntegration::StyleHint hint) const
         return false;
     case QPlatformIntegration::ReplayMousePressOutsidePopup:
         return false;
+
+    case QPlatformIntegration::FontSmoothingGamma:
+        return m_fontSmoothingGamma;
+        break;
+
     default:
         break;
     }
diff --git a/qtbase/src/plugins/platforms/xcb/qxcbintegration.h b/qtbase/src/plugins/platforms/xcb/qxcbintegration.h
index b3d72c19d045f2ce235e536ca3e262ad98cfdc3a..5b8146ac9c55676967282ae34bbdf87663dccc6e 100644
--- a/qtbase/src/plugins/platforms/xcb/qxcbintegration.h
+++ b/qtbase/src/plugins/platforms/xcb/qxcbintegration.h
@@ -136,6 +136,7 @@ private:
     const char *m_instanceName;
     bool m_canGrab;
     xcb_visualid_t m_defaultVisualId;
+    qreal m_fontSmoothingGamma;
 
     static QXcbIntegration *m_instance;
 };
