diff --git a/qtbase/src/plugins/platforms/cocoa/qcocoacolordialoghelper.mm b/qtbase/src/plugins/platforms/cocoa/qcocoacolordialoghelper.mm
index 5d331c0..f1eca59 100644
--- a/qtbase/src/plugins/platforms/cocoa/qcocoacolordialoghelper.mm
+++ b/qtbase/src/plugins/platforms/cocoa/qcocoacolordialoghelper.mm
@@ -123,6 +123,7 @@ QT_NAMESPACE_ALIAS_OBJC_CLASS(QNSColorPanelDelegate);
         // create a new content view and add the stolen one as a subview
         mPanelButtons = [[QNSPanelContentsWrapper alloc] initWithPanelDelegate:self];
         [mPanelButtons addSubview:mStolenContentView];
+        [mPanelButtons setFrameSize:[mPanelButtons fittingSize]];
         mColorPanel.contentView = mPanelButtons;
         mColorPanel.defaultButtonCell = mPanelButtons.okButton.cell;
     }
diff --git a/qtbase/src/plugins/platforms/cocoa/qcocoafontdialoghelper.mm b/qtbase/src/plugins/platforms/cocoa/qcocoafontdialoghelper.mm
index 19baf93..8858bb0 100644
--- a/qtbase/src/plugins/platforms/cocoa/qcocoafontdialoghelper.mm
+++ b/qtbase/src/plugins/platforms/cocoa/qcocoafontdialoghelper.mm
@@ -144,6 +144,7 @@ QT_NAMESPACE_ALIAS_OBJC_CLASS(QNSFontPanelDelegate);
         mPanelButtons = [[QNSPanelContentsWrapper alloc] initWithPanelDelegate:self];
         [mPanelButtons addSubview:mStolenContentView];
         mPanelButtons.panelContentsMargins = NSEdgeInsetsMake(0, 0, 7, 0);
+        [mPanelButtons setFrameSize:[mPanelButtons fittingSize]];
         mFontPanel.contentView = mPanelButtons;
         mFontPanel.defaultButtonCell = mPanelButtons.okButton.cell;
     }
diff --git a/qtbase/src/plugins/platforms/cocoa/qcocoahelpers.mm b/qtbase/src/plugins/platforms/cocoa/qcocoahelpers.mm
index 5d0f13c..c9ec5fd 100644
--- a/qtbase/src/plugins/platforms/cocoa/qcocoahelpers.mm
+++ b/qtbase/src/plugins/platforms/cocoa/qcocoahelpers.mm
@@ -407,6 +407,7 @@ QT_END_NAMESPACE
     self.panelContents.needsDisplay = YES;
 
     self.needsDisplay = YES;
+    [super layout];
 }
 
 @end
