diff --git a/qtbase/src/platformsupport/fontdatabases/mac/coretext.pri b/qtbase/src/platformsupport/fontdatabases/mac/coretext.pri
index 2a9b32f..31a4ab0 100644
--- a/qtbase/src/platformsupport/fontdatabases/mac/coretext.pri
+++ b/qtbase/src/platformsupport/fontdatabases/mac/coretext.pri
@@ -1,7 +1,7 @@
 HEADERS += $$PWD/qcoretextfontdatabase_p.h $$PWD/qfontengine_coretext_p.h
 OBJECTIVE_SOURCES += $$PWD/qfontengine_coretext.mm $$PWD/qcoretextfontdatabase.mm
 
-qtConfig(freetype) {
+qtConfig(freetype)|qtConfig(system-freetype) {
     QMAKE_USE_PRIVATE += freetype
     HEADERS += freetype/qfontengine_ft_p.h
     SOURCES += freetype/qfontengine_ft.cpp
