--- qt-everywhere-opensource-src-5.9.3/qtwebengine/src/3rdparty/chromium/third_party/WebKit/Source/wtf/Compiler.h.orig	2017-11-08 20:13:31.000000000 +0100
+++ qt-everywhere-opensource-src-5.9.3/qtwebengine/src/3rdparty/chromium/third_party/WebKit/Source/wtf/Compiler.h	2017-12-20 14:54:53.000000000 +0100
@@ -117,8 +117,9 @@
  * unrelated objects (-fsanitize=cfi-unrelated-cast or -fsanitize=vptr). */
 
 #if COMPILER(CLANG)
-#define NO_SANITIZE_UNRELATED_CAST \
-  __attribute__((no_sanitize("cfi-unrelated-cast", "vptr")))
+// #define NO_SANITIZE_UNRELATED_CAST \
+//   __attribute__((no_sanitize("cfi-unrelated-cast", "vptr")))
+#define NO_SANITIZE_UNRELATED_CAST
 #else
 #define NO_SANITIZE_UNRELATED_CAST
 #endif
