diff --git a/qtbase/src/corelib/kernel/qobject.cpp b/qtbase/src/corelib/kernel/qobject.cpp
index 086b8a51ba984cd17c26c30cd098fee8b73aa71f..5cc4d1a4b603c22ba58adac1ebe6edf1c81c7b0b 100644
--- a/qtbase/src/corelib/kernel/qobject.cpp
+++ b/qtbase/src/corelib/kernel/qobject.cpp
@@ -211,9 +211,14 @@ QObjectPrivate::QObjectPrivate(int version)
     // This allows incompatible versions to be loaded, possibly for testing.
     Q_UNUSED(version);
 #else
-    if (Q_UNLIKELY(version != QObjectPrivateVersion))
-        qFatal("Cannot mix incompatible Qt library (version 0x%x) with this library (version 0x%x)",
-                version, QObjectPrivateVersion);
+    if (Q_UNLIKELY(version != QObjectPrivateVersion)) {
+        if (version < QObjectPrivateVersion && (QObjectPrivateVersion - version) < 0x100)
+            qCritical("!! Warning: mixing incompatible Qt library (version 0x%x) with the currently running version 0x%x",
+                    version, QObjectPrivateVersion);
+        else
+            qFatal("Cannot mix incompatible Qt library (version 0x%x) with this library (version 0x%x)",
+                    version, QObjectPrivateVersion);
+    }
 #endif
 
     // QObjectData initialization
