diff --git a/qtbase/src/plugins/plugins.pro b/qtbase/src/plugins/plugins.pro
index 6d62420..619e3d0 100644
--- a/qtbase/src/plugins/plugins.pro
+++ b/qtbase/src/plugins/plugins.pro
@@ -2,7 +2,7 @@ TEMPLATE = subdirs
 QT_FOR_CONFIG += network
 
 qtHaveModule(sql): SUBDIRS += sqldrivers
-qtHaveModule(network):qtConfig(bearermanagement): SUBDIRS += bearer
+# qtHaveModule(network):qtConfig(bearermanagement): SUBDIRS += bearer
 qtHaveModule(gui) {
     SUBDIRS *= platforms platforminputcontexts platformthemes
     qtConfig(imageformatplugin): SUBDIRS *= imageformats
