diff --git a/qtbase/mkspecs/common/clang-mac.conf b/qtbase/mkspecs/common/clang-mac.conf
index cbae2e6..e751b9b 100644
--- a/qtbase/mkspecs/common/clang-mac.conf
+++ b/qtbase/mkspecs/common/clang-mac.conf
@@ -6,8 +6,8 @@ QMAKE_OBJCXXFLAGS_USE_PRECOMPILE = $$QMAKE_CFLAGS_USE_PRECOMPILE
 
 QMAKE_XCODE_GCC_VERSION = com.apple.compilers.llvm.clang.1_0
 
-QMAKE_CXXFLAGS += -stdlib=libc++
-QMAKE_LFLAGS   += -stdlib=libc++
+QMAKE_CXXFLAGS += -stdlib=__MACPORTS_CXX_STDLIB__
+QMAKE_LFLAGS   += -stdlib=__MACPORTS_CXX_STDLIB__
 QMAKE_AR_LTCG  = libtool -static -o
 
 QMAKE_CFLAGS_APPLICATION_EXTENSION  = -fapplication-extension
diff --git a/qtbase/mkspecs/macx-clang/qmake.conf b/qtbase/mkspecs/macx-clang/qmake.conf
index e34c089..8f2df89 100644
--- a/qtbase/mkspecs/macx-clang/qmake.conf
+++ b/qtbase/mkspecs/macx-clang/qmake.conf
@@ -2,9 +2,9 @@
 # qmake configuration for Clang on OS X
 #
 
-QMAKE_MACOSX_DEPLOYMENT_TARGET = 10.10
+QMAKE_MACOSX_DEPLOYMENT_TARGET = __MACPORTS_DEPLOYMENT_TARGET__
 
-QMAKE_APPLE_DEVICE_ARCHS = x86_64
+QMAKE_APPLE_DEVICE_ARCHS = __MACPORTS_DEVICE_ARCHS__
 
 include(../common/macx.conf)
 include(../common/gcc-base-mac.conf)
diff --git a/qtbase/mkspecs/macx-g++/qmake.conf b/qtbase/mkspecs/macx-g++/qmake.conf
index 4209c9a..3c82c92 100644
--- a/qtbase/mkspecs/macx-g++/qmake.conf
+++ b/qtbase/mkspecs/macx-g++/qmake.conf
@@ -10,9 +10,9 @@ MAKEFILE_GENERATOR      = UNIX
 CONFIG                 += app_bundle incremental global_init_link_order lib_version_first
 QMAKE_INCREMENTAL_STYLE = sublib
 
-QMAKE_MACOSX_DEPLOYMENT_TARGET = 10.10
+QMAKE_MACOSX_DEPLOYMENT_TARGET = __MACPORTS_DEPLOYMENT_TARGET__
 
-QMAKE_APPLE_DEVICE_ARCHS = x86_64
+QMAKE_APPLE_DEVICE_ARCHS = __MACPORTS_DEVICE_ARCHS__
 
 include(../common/macx.conf)
 include(../common/gcc-base-mac.conf)
diff --git a/qtbase/mkspecs/macx-icc/qmake.conf b/qtbase/mkspecs/macx-icc/qmake.conf
index e999aa3..7b0070b 100644
--- a/qtbase/mkspecs/macx-icc/qmake.conf
+++ b/qtbase/mkspecs/macx-icc/qmake.conf
@@ -95,9 +95,9 @@ QMAKE_CXXFLAGS_PRECOMPILE = -c -pch-create ${QMAKE_PCH_OUTPUT} -include ${QMAKE_
 QMAKE_CFLAGS_HIDESYMS   += -fvisibility=hidden
 QMAKE_CXXFLAGS_HIDESYMS += $$QMAKE_CFLAGS_HIDESYMS -fvisibility-inlines-hidden
 
-QMAKE_MACOSX_DEPLOYMENT_TARGET = 10.10
+QMAKE_MACOSX_DEPLOYMENT_TARGET = __MACPORTS_DEPLOYMENT_TARGET__
 
-QMAKE_APPLE_DEVICE_ARCHS = x86_64
+QMAKE_APPLE_DEVICE_ARCHS = __MACPORTS_DEVICE_ARCHS__
 
 include(../common/macx.conf)
 
