--- qt5kdweb-mp9-work/qtwebengine/src/3rdparty/chromium/base/process/orig.memory_mac.mm	2017-02-07 18:35:45.000000000 +0100
+++ qt5kdweb-mp9-work/qtwebengine/src/3rdparty/chromium/base/process/memory_mac.mm	2017-02-07 18:38:17.000000000 +0100
@@ -248,13 +248,15 @@
 // === Core Foundation CFAllocators ===
 
 bool CanGetContextForCFAllocator() {
-  return !base::mac::IsOSLaterThanSierra_DontCallThis();
+//   return !base::mac::IsOSLaterThanSierra_DontCallThis();
+    return false;
 }
 
 CFAllocatorContext* ContextForCFAllocator(CFAllocatorRef allocator) {
   ChromeCFAllocatorLions* our_allocator =
       const_cast<ChromeCFAllocatorLions*>(
           reinterpret_cast<const ChromeCFAllocatorLions*>(allocator));
+  NSLog(@"ContextForCFAllocator:%d(%@)=0x%lx", __LINE__, allocator, &our_allocator->_context);
   return &our_allocator->_context;
 }
 
