--- qt-everywhere-opensource-src-5.9.3/qtwebview/src/webview/orig.qwebview_darwin.mm	2017-11-10 10:11:26.000000000 +0100
+++ qt-everywhere-opensource-src-5.9.3/qtwebview/src/webview/qwebview_darwin.mm	2018-01-10 19:47:33.000000000 +0100
@@ -253,7 +280,7 @@
 {
     if (url.isValid()) {
         requestFrameCount = 0;
-        if (!url.isLocalFile()) {
+        if (!url.isLocalFile() || ![wkWebView respondsToSelector:@selector(loadFileURL:allowingReadAccessToURL:)]) {
             [wkWebView loadRequest:[NSURLRequest requestWithURL:url.toNSURL()]];
         } else {
             // We need to pass local files via loadFileURL and the read access should cover
