diff --git a/qttools/src/assistant/assistant/assistant.pro b/qttools/src/assistant/assistant/assistant.pro
index 82f5509..0250451 100644
--- a/qttools/src/assistant/assistant/assistant.pro
+++ b/qttools/src/assistant/assistant/assistant.pro
@@ -1,11 +1,11 @@
-qtHaveModule(webkitwidgets):!contains(QT_CONFIG, static) {
+# qtHaveModule(webkitwidgets):!contains(QT_CONFIG, static) {
     BROWSER = qtwebkit
-} else {
-    BROWSER = qtextbrowser
-}
+# } else {
+#     BROWSER = qtextbrowser
+# }
 
 QT += widgets network help sql help
-qtHaveModule(printsupport): QT += printsupport
+QT += printsupport
 PROJECTNAME = Assistant
 
 include(../../shared/fontpanel/fontpanel.pri)
@@ -119,4 +119,4 @@ mac {
     QMAKE_INFO_PLIST = Info_mac.plist
 }
 
-load(qt_app)
+# load(qt_app)
