diff --git a/qtbase/src/widgets/styles/qmacstyle_mac.mm b/qtbase/src/widgets/styles/qmacstyle_mac.mm
index 5913f3f..58aba3a 100644
--- a/qtbase/src/widgets/styles/qmacstyle_mac.mm
+++ b/qtbase/src/widgets/styles/qmacstyle_mac.mm
@@ -566,25 +566,24 @@ QRegion qt_mac_fromHIShapeRef(HIShapeRef shape)
     return returnRegion;
 }
 
+NSWindow *qt_mac_toNSWindow(const QWidget *window)
+{
+    if (QWindow *w = window->windowHandle()) {
+        auto *iFace = QGuiApplication::platformNativeInterface();
+        if (Q_LIKELY(iFace) && w->handle())
+            return static_cast<NSWindow*>(iFace->nativeResourceForWindow(QByteArrayLiteral("nswindow"), w));
+    }
+    return nil;
+}
+
 bool qt_macWindowIsTextured(const QWidget *window)
 {
-    if (QWindow *w = window->windowHandle())
-        if (w->handle())
-            if (NSWindow *nswindow = static_cast<NSWindow*>(QGuiApplication::platformNativeInterface()->nativeResourceForWindow(QByteArrayLiteral("NSWindow"), w)))
-                return ([nswindow styleMask] & NSTexturedBackgroundWindowMask) ? true : false;
-    return false;
+    return ([qt_mac_toNSWindow(window) styleMask] & NSTexturedBackgroundWindowMask) ? true : false;
 }
 
 static bool qt_macWindowMainWindow(const QWidget *window)
 {
-    if (QWindow *w = window->windowHandle()) {
-        if (w->handle()) {
-            if (NSWindow *nswindow = static_cast<NSWindow*>(QGuiApplication::platformNativeInterface()->nativeResourceForWindow(QByteArrayLiteral("nswindow"), w))) {
-                return [nswindow isMainWindow];
-            }
-        }
-    }
-    return false;
+    return [qt_mac_toNSWindow(window) isMainWindow];
 }
 
 /*****************************************************************************
