diff --git a/qttools/src/qtpaths/qtpaths.cpp b/qttools/src/qtpaths/qtpaths.cpp
index 35d6dbf..1a40035 100644
--- a/qttools/src/qtpaths/qtpaths.cpp
+++ b/qttools/src/qtpaths/qtpaths.cpp
@@ -216,8 +216,13 @@ int main(int argc, char **argv)
                                  QCoreApplication::translate("qtpaths", "Location of Qt plugins."));
     parser.addOption(plugindir);
 
+    QCommandLineOption qspalt(QStringList() << QStringLiteral("altmode") << QStringLiteral("alt-mode"),
+                                 QCoreApplication::translate("qtpaths", "Activate ALT mode."));
+    parser.addOption(qspalt);
+
     parser.process(app);
 
+    QStandardPaths::setALTLocationsEnabled(parser.isSet(qspalt));
     QStandardPaths::enableTestMode(parser.isSet(testmode));
 
     QStringList results;
