diff --git a/qtbase/src/plugins/platforms/minimal/qminimalintegration.cpp b/qtbase/src/plugins/platforms/minimal/qminimalintegration.cpp
index ca33689..0df622a 100644
--- a/qtbase/src/plugins/platforms/minimal/qminimalintegration.cpp
+++ b/qtbase/src/plugins/platforms/minimal/qminimalintegration.cpp
@@ -52,8 +52,8 @@
 #  if QT_CONFIG(freetype)
 #    include <QtFontDatabaseSupport/private/qwindowsfontdatabase_ft_p.h>
 #  endif
-#elif defined(Q_OS_DARWIN)
-#  include <QtFontDatabaseSupport/private/qcoretextfontdatabase_p.h>
+// #elif defined(Q_OS_DARWIN)
+// #  include <QtFontDatabaseSupport/private/qcoretextfontdatabase_p.h>
 #elif QT_CONFIG(fontconfig)
 #  include <QtFontDatabaseSupport/private/qgenericunixfontdatabase_p.h>
 #  include <qpa/qplatformfontdatabase.h>
