diff --git libavcodec/orig.sbrdsp_template.c libavcodec/sbrdsp_template.c
index 89e389d..0e9073f 100644
--- libavcodec/orig.sbrdsp_template.c
+++ libavcodec/sbrdsp_template.c
@@ -106,8 +106,9 @@ av_cold void AAC_RENAME(ff_sbrdsp_init)(SBRDSPContext *s)
 #endif /* !USE_FIXED */
 }
 
+#include "sbr_noise_table_visibility.h"
 /* First eight entries repeated at end to simplify SIMD implementations. */
-const attribute_visibility_hidden DECLARE_ALIGNED(16, INTFLOAT, AAC_RENAME(ff_sbr_noise_table))[][2] = {
+const NOISE_TABLE_HIDDEN  DECLARE_ALIGNED(16, INTFLOAT, AAC_RENAME(ff_sbr_noise_table))[][2] = {
 {Q31(-0.99948153278296f), Q31(-0.59483417516607f)}, {Q31( 0.97113454393991f), Q31(-0.67528515225647f)},
 {Q31( 0.14130051758487f), Q31(-0.95090983575689f)}, {Q31(-0.47005496701697f), Q31(-0.37340549728647f)},
 {Q31( 0.80705063769351f), Q31( 0.29653668284408f)}, {Q31(-0.38981478896926f), Q31( 0.89572605717087f)},
