diff --git libavfilter/orig.af_ladspa.c libavfilter/af_ladspa.c
index 35ff90b..be05c43 100644
--- libavfilter/orig.af_ladspa.c
+++ libavfilter/af_ladspa.c
@@ -527,7 +527,7 @@ static av_cold int init(AVFilterContext *ctx)
         }
 
         if (!s->dl_handle)
-            s->dl_handle = try_load("/usr/local/lib/ladspa", s->dl_name);
+            s->dl_handle = try_load("@PREFIX@/lib/ladspa", s->dl_name);
 
         if (!s->dl_handle)
             s->dl_handle = try_load("/usr/lib/ladspa", s->dl_name);
