--- orig.main.c	2022-08-23 12:28:44.000000000 +0200
+++ main.c	2022-08-23 12:32:25.838153085 +0200
@@ -367,7 +367,9 @@
 
 rearg:
 	optind = 1;	/* since we're called more than once */
+#ifndef __linux__
 	optreset = 1;
+#endif
 #define OPTFLAGS "ABC:D:E:I:PSV:Xd:ef:ij:km:nqrstvx:"
 	for (;;) {
 		if ((optind < argc) && strcmp(argv[optind], "--") == 0) {
--- orig.main.c	2022-08-23 12:51:41.000000000 +0200
+++ main.c	2022-08-23 12:56:23.201436992 +0200
@@ -765,6 +765,7 @@
 	 * make binary on old FreeBSD/pc98 systems, and have the
 	 * MACHINE variable set properly.
 	 */
+#ifndef __linux__
 	if ((machine = getenv("MACHINE")) == NULL) {
 		int	ispc98;
 		size_t	len;
@@ -775,6 +776,7 @@
 				machine = "pc98";
 		}
 	}
+#endif
 
 	/*
 	 * Get the name of this type of MACHINE from utsname
