--- orig.CMakeLists.txt	2018-11-19 22:49:09.000000000 +0100
+++ CMakeLists.txt	2018-11-19 23:04:41.000000000 +0100
@@ -26,6 +26,7 @@
 #include(ECMAddTests)
 #include(ECMMarkNonGuiExecutable)
 #include(ECMInstallIcons)
+include(ECMAddAppIcon)
 include(ECMUninstallTarget)
 include(FeatureSummary)
 
--- src/orig.CMakeLists.txt	2018-09-03 02:35:39.000000000 +0200
+++ src/CMakeLists.txt	2018-11-19 23:07:01.000000000 +0100
@@ -70,6 +70,8 @@
    ../icons/icons.qrc
 )
 
+ecm_add_app_icon(tellico_SRCS ICONS ${CMAKE_CURRENT_SOURCE_DIR}/../icons/tellico.svg)
+
 add_executable(tellico ${tellico_SRCS})
 
 TARGET_LINK_LIBRARIES(tellico
