diff --git cmake/modules/orig.FindQImageBlitz.cmake cmake/modules/FindQImageBlitz.cmake
index 75cd828..772cbab 100644
--- cmake/modules/orig.FindQImageBlitz.cmake
+++ cmake/modules/FindQImageBlitz.cmake
@@ -20,7 +20,7 @@ if (NOT WIN32)
     # use pkg-config to get the directories and then use these values
     # in the FIND_PATH() and FIND_LIBRARY() calls
     find_package(PkgConfig)
-    pkg_check_modules(PC_QIMAGEBLITZ5 QUIET qimageblitz5)
+    pkg_check_modules(PC_QIMAGEBLITZ5 QUIET q5imageblitz)
     if(NOT PC_QIMAGEBLITZ5_FOUND)
         pkg_check_modules(PC_QIMAGEBLITZ QUIET qimageblitz>=5.0)
     endif()
@@ -28,7 +28,7 @@ endif (NOT WIN32)
 
 find_path(QIMAGEBLITZ_INCLUDES
   NAMES qimageblitz.h
-  PATH_SUFFIXES qimageblitz5 qimageblitz
+  PATH_SUFFIXES q5imageblitz
   HINTS
   $ENV{QIMAGEBLITZDIR}/include
   ${PC_QIMAGEBLITZ5_INCLUDEDIR}
@@ -38,7 +38,7 @@ find_path(QIMAGEBLITZ_INCLUDES
 
 find_library_with_debug(QIMAGEBLITZ_LIBRARIES
   WIN32_DEBUG_POSTFIX d
-  NAMES qimageblitz5 qimageblitz
+  NAMES q5imageblitz
   HINTS
   $ENV{QIMAGEBLITZDIR}/lib
   ${PC_QIMAGEBLITZ5_LIBDIR}
