diff --git src/translators/orig.audiofileimporter.cpp src/translators/audiofileimporter.cpp
index 2f7c208..12aec19 100644
--- src/translators/orig.audiofileimporter.cpp
+++ src/translators/audiofileimporter.cpp
@@ -37,13 +37,13 @@
 #include "../tellico_debug.h"
 
 #ifdef HAVE_TAGLIB
-#include <fileref.h>
-#include <tag.h>
-#include <id3v2tag.h>
-#include <mpegfile.h>
-#include <vorbisfile.h>
-#include <flacfile.h>
-#include <audioproperties.h>
+#include <taglib/fileref.h>
+#include <taglib/tag.h>
+#include <taglib/id3v2tag.h>
+#include <taglib/mpegfile.h>
+#include <taglib/vorbisfile.h>
+#include <taglib/flacfile.h>
+#include <taglib/audioproperties.h>
 #endif
 
 #include <KLocalizedString>
diff --git src/translators/orig.CMakeLists.txt src/translators/CMakeLists.txt
index 2639367..1689cf2 100644
--- src/translators/orig.CMakeLists.txt
+++ src/translators/CMakeLists.txt
@@ -75,7 +75,9 @@ ELSEIF( Libkcddb_FOUND )
 ENDIF( KF5Cddb_FOUND)
 
 IF( Poppler_Qt5_FOUND )
-  TARGET_LINK_LIBRARIES(translators Poppler::Qt5)
+  set_target_properties(translators PROPERTIES CXX_STANDARD 17)
+  TARGET_INCLUDE_DIRECTORIES(translators PRIVATE ${Poppler_Core_INCLUDE_DIR} ${Poppler_Qt5_INCLUDE_DIR})
+  TARGET_LINK_LIBRARIES(translators ${Poppler_Qt5_LIBRARY})
 ENDIF( Poppler_Qt5_FOUND )
 
 IF( KF5FileMetaData_FOUND )
