diff --git mesonbuild/backend/ninjabackend.py.orig mesonbuild/backend/ninjabackend.py
index cd35d2f..8356add 100644
--- mesonbuild/backend/ninjabackend.py.orig
+++ mesonbuild/backend/ninjabackend.py
@@ -659,7 +659,9 @@ class NinjaBackend(backends.Backend):
                     if isinstance(target, build.BuildTarget):
                         captured_compile_args_per_target[target.get_id()] = self.generate_common_compile_args_per_src_type(target)
 
-            for t in ProgressBar(self.build.get_targets().values(), desc='Generating targets'):
+            mlog.log('Generating targets')
+            #for t in ProgressBar(self.build.get_targets().values(), desc='Generating targets'):
+            for t in self.build.get_targets().values():
                 self.generate_target(t)
             mlog.log_timestamp("Targets generated")
             self.add_build_comment(NinjaComment('Test rules'))
@@ -1423,7 +1425,8 @@ class NinjaBackend(backends.Backend):
             r.write(outfile)
 
     def write_builds(self, outfile: T.TextIO) -> None:
-        for b in ProgressBar(self.build_elements, desc='Writing build.ninja'):
+        mlog.log('Writing build.ninja')
+        for b in self.build_elements:
             b.write(outfile)
         mlog.log_timestamp("build.ninja generated")
 
