diff --git qtbase/src/corelib/io/qstandardpaths_mac.cpp qtbase/src/corelib/io/qstandardpaths_mac.cpp
index 13b8646..cf2f299 100644
--- qtbase/src/corelib/io/qstandardpaths_mac.cpp
+++ qtbase/src/corelib/io/qstandardpaths_mac.cpp
@@ -169,6 +169,8 @@ QString QStandardPaths::writableLocation(StandardLocation type)
     case GenericCacheLocation:
     case CacheLocation:
     case RuntimeLocation:
+    case FontsLocation:
+        // the font location that is writable for all users is ~/Library/Fonts
         return macLocation(type, kUserDomain);
     default:
         return macLocation(type, kOnAppropriateDisk);
@@ -209,6 +211,12 @@ QStringList QStandardPaths::standardLocations(StandardLocation type)
                 dirs.append(bundlePath + resourcesPath);
         }
     }
+    if (type == FontsLocation) {
+        // /Library/Fonts
+        dirs.append(macLocation(type,kLocalDomain));
+        // /System/Library/Fonts
+        dirs.append(macLocation(type,kSystemDomain));
+    }
     const QString localDir = writableLocation(type);
     dirs.prepend(localDir);
     return dirs;
