diff --git src/kstyle/CMakeLists.txt src/kstyle/CMakeLists.txt
index bc26667..8e2031e 100644
--- src/kstyle/CMakeLists.txt
+++ src/kstyle/CMakeLists.txt
@@ -7,7 +7,7 @@ set_package_properties(XCB PROPERTIES
     PURPOSE "Required to pass style properties to native Windows on X11 Platform"
 )
 
-if(NOT APPLE)
+if(NOT APPLE OR APPLE_FORCE_X11)
   set(HAVE_X11 ${XCB_XCB_FOUND})
   if (XCB_XCB_FOUND)
     find_package(Qt5 ${REQUIRED_QT_VERSION} CONFIG REQUIRED X11Extras)
