diff --git src/kstyle/kstyle.cpp src/kstyle/kstyle.cpp
index 1baf6c6..1089f07 100644
--- src/kstyle/kstyle.cpp
+++ src/kstyle/kstyle.cpp
@@ -143,6 +143,9 @@ public:
     QHash<QString, int> styleElements;
     int hintCounter, controlCounter, subElementCounter;
     QScopedPointer<ColorSchemeFilter> colorSchemeFilter;
+#ifdef Q_OS_MACOS
+    bool m_isCocoa;
+#endif
 };
 
 KStylePrivate::KStylePrivate()
@@ -187,6 +190,16 @@ QStyle::SubElement KStyle::customSubElement(const QString &element, const QWidge
 
 KStyle::KStyle() : d(new KStylePrivate)
 {
+#ifdef Q_OS_MACOS
+    const auto platformName = QGuiApplication::platformName();
+    if (!platformName.contains(QLatin1String("cocoa"))) {
+        QCoreApplication::setAttribute(Qt::AA_DontUseNativeMenuBar, true);
+        QCoreApplication::setAttribute(Qt::AA_MacDontSwapCtrlAndMeta, true);
+        d->m_isCocoa = false;
+    } else {
+        d->m_isCocoa = true;
+    }
+#endif
 }
 
 KStyle::~KStyle()
@@ -482,6 +495,15 @@ int KStyle::styleHint(StyleHint hint, const QStyleOption *option, const QWidget
 
         return d->styleElements.value(widget->objectName(), 0);
 
+#ifdef Q_OS_MACOS
+    case SH_TitleBar_ModifyNotification: {
+        // when on Mac and using the Cocoa QPA, SH_TitleBar_ModifyNotification should return false so
+        // windowModified notification is done only via the native mechanism (in the window close button).
+        const auto ret = d->m_isCocoa ? false : true;
+        return ret;
+    }
+#endif
+
     default:
         break;
     };
