diff --git a/src/samplewidget.cpp b/src/samplewidget.cpp
index fe2cd988eb0cbb263b0c9e457146b340e43234ab..dd34513a3b28e279196a6d0505baa245713acc0e 100644
--- a/src/samplewidget.cpp
+++ b/src/samplewidget.cpp
@@ -461,11 +461,15 @@ void SampleWidget::drawPixmap(int index, double fontsize, double x, double y)
 	FontItem * f( FMFontDb::DB()->Font( fontIdentifier ) );
 	if(!f)
 		return;
-	++pixmapDrawn;
 	QGraphicsPixmapItem *glyph = f->itemFromGindexPix ( index , fontsize );
+    if (!glyph) {
+        return;
+    }
+	++pixmapDrawn;
 //	qDebug()<<"SampleWidget::drawPixmap index:"<<index<< y << glyph->data(GLYPH_DATA_BITMAPTOP).toDouble();
 	ftScene->addItem ( glyph );
-	glyph->setZValue ( 100.0 );
+
+    glyph->setZValue ( 100.0 );
 	glyph->setPos ( x,y );
 //	QGraphicsLineItem * l = ftScene->addLine(x,y,x,y + glyph->data(GLYPH_DATA_BITMAPTOP).toDouble());
 //	l->setData(GLYPH_DATA_GLYPH, 1);
