diff --git a/src/fmpaths.cpp b/src/fmpaths.cpp
index 4ccc03c5decd9e3345582abd318e206385b3ea1a..809dcdea0d9f48aaa106ff67c63e78cc93a65c12 100644
--- a/src/fmpaths.cpp
+++ b/src/fmpaths.cpp
@@ -28,7 +28,7 @@ QString FMPaths::TranslationsDir()
 	if(getThis()->FMPathsDB.contains("TranslationsDir"))
 		return getThis()->FMPathsDB["TranslationsDir"];
 	QString dirsep(QDir::separator());
-#ifdef PLATFORM_APPLE
+#if defined(PLATFORM_APPLE) && defined(APPLE_STANDALONE_BUNDLE)
 	QString QMDirPath = QApplication::applicationDirPath();
 	QMDirPath +=  dirsep + ".." + dirsep + "Resources" + dirsep + "Locales" + dirsep;	
 #elif _WIN32
@@ -49,7 +49,7 @@ QString FMPaths::HelpDir()
 		return getThis()->FMPathsDB["HelpDir"];
 	QString hf;
 	QString dirsep(QDir::separator());
-#ifdef PLATFORM_APPLE
+#if defined(PLATFORM_APPLE) && defined(APPLE_STANDALONE_BUNDLE)
 	hf = LocalizedDirPath( QApplication::applicationDirPath() + dirsep + "help" + dirsep );
 #elif _WIN32
 	hf = LocalizedDirPath(QApplication::applicationDirPath() + dirsep + "help" + dirsep );
@@ -65,7 +65,7 @@ QString FMPaths::ResourcesDir()
 	if(getThis()->FMPathsDB.contains("ResourcesDir"))
 		return getThis()->FMPathsDB["ResourcesDir"];
 	QString dirsep(QDir::separator());
-#ifdef PLATFORM_APPLE
+#if defined(PLATFORM_APPLE) && defined(APPLE_STANDALONE_BUNDLE)
 	QString QMDirPath = QApplication::applicationDirPath();
 	QMDirPath +=  dirsep + ".." + dirsep + "Resources" + dirsep ;	
 #elif _WIN32
diff --git a/src/typotek.cpp b/src/typotek.cpp
index b72ff884993a7dfd39e2fbe6fbb9397f67350e5b..0d0af5708309cdd2d323a1bd7a1bf4371df54a6d 100644
--- a/src/typotek.cpp
+++ b/src/typotek.cpp
@@ -1178,6 +1178,7 @@ QStringList typotek::getSystemFontDirs()
 
 #endif //HAVE_FONTCONFIG
 #ifdef PLATFORM_APPLE
+	retList << QDir::homePath() + "/Library/Fonts";
 	retList << "/Library/Fonts";
 	retList << "/System/Library/Fonts";
 #endif // PLATFORM_APPLE
