diff --git a/src/icushaper.cpp b/src/icushaper.cpp
index c503e80b83b063ff6aa25fdf8edb44be7ce3e680..d88b0d10a00bc1893de852ccebafeccf8a83197a 100644
--- a/src/icushaper.cpp
+++ b/src/icushaper.cpp
@@ -248,6 +248,12 @@ IcuFontImpl::~ IcuFontImpl()
 
 }
 
+const void * IcuFontImpl::getFontTable ( LETag tableTag ) const
+{
+    size_t length_sz;
+    return getFontTable(tableTag, length_sz);
+}
+
 const void * IcuFontImpl::getFontTable ( LETag tableTag, size_t &length_sz ) const
 {
 // 	qDebug()<< "IcuFontImpl::getFontTable" << OTF_tag_name( tableTag );
diff --git a/src/icushaper.h b/src/icushaper.h
index 6a8da5ef96f11c1074cef8de17ccbd0d87bdd50a..05902b5b2d7fc714d5e87416a58c168a45708dbd 100644
--- a/src/icushaper.h
+++ b/src/icushaper.h
@@ -27,6 +27,8 @@ class IcuFontImpl : public LEFontInstance
 		~IcuFontImpl();
 		// implements pure virtual methods of LEFontInstance
 
+        // for backwards ICU compatibility
+        virtual const void *getFontTable(LETag tableTag) const;
 		const void* getFontTable(LETag   tableTag, size_t &length_sz ) const;
 		le_bool 	canDisplay (LEUnicode32 ch) const {return true;}
 		le_int32 	getUnitsPerEM () const;
