diff --git a/appmenu/appmenu.cpp b/appmenu/appmenu.cpp
index 7c956c6e9709c2883a04721306579a54e7b7dcef..4ab71067738e9a413dfe031061a5aa9c171c74ab 100644
--- a/appmenu/appmenu.cpp
+++ b/appmenu/appmenu.cpp
@@ -106,7 +106,7 @@ AppMenuModule::~AppMenuModule() = default;
 
 void AppMenuModule::slotWindowRegistered(WId id, const QString &serviceName, const QDBusObjectPath &menuObjectPath)
 {
-#ifdef HAVE_X11
+#if HAVE_X11
     if (KWindowSystem::isPlatformX11()) {
         auto *c = QX11Info::connection();
 
diff --git a/shell/scripting/appinterface.cpp b/shell/scripting/appinterface.cpp
index 0393ed1c142e05b745b50ace550a84ca97fe5b96..cfecc034a5a908b50e0dacc90012b900980eb289 100644
--- a/shell/scripting/appinterface.cpp
+++ b/shell/scripting/appinterface.cpp
@@ -37,7 +37,7 @@
 #include <windows.h>
 #endif
 
-#ifdef HAVE_X11
+#if HAVE_X11
 #include <X11/Xlib.h>
 #include <fixx11h.h>
 #endif
