diff --git drkonqi/CMakeLists.txt drkonqi/CMakeLists.txt
index 4448e8b7..c8be86e6 100644
--- drkonqi/CMakeLists.txt
+++ drkonqi/CMakeLists.txt
@@ -111,6 +111,11 @@ if (HAVE_X11)
         Qt5::X11Extras
     )
 endif()
+if (APPLE)
+    target_link_libraries(drkonqi
+        KF5::WindowSystem
+    )
+endif()
 
 install(TARGETS drkonqi DESTINATION ${KDE_INSTALL_LIBEXECDIR})
 
diff --git drkonqi/aboutbugreportingdialog.cpp drkonqi/aboutbugreportingdialog.cpp
index 625f512b..baea8ab8 100644
--- drkonqi/aboutbugreportingdialog.cpp
+++ drkonqi/aboutbugreportingdialog.cpp
@@ -36,7 +36,7 @@ AboutBugReportingDialog::AboutBugReportingDialog(QWidget * parent):
 {
     setAttribute(Qt::WA_DeleteOnClose, true);
 
-    setWindowIcon(QIcon::fromTheme(QStringLiteral("help-hint")));
+    setWindowIcon(QIcon::fromTheme(QStringLiteral("help-hint"), windowIcon()));
     setWindowTitle(i18nc("@title title of the dialog", "About Bug Reporting - Help"));
 
     QVBoxLayout* layout = new QVBoxLayout(this);
diff --git drkonqi/bugzillaintegration/reportassistantdialog.cpp drkonqi/bugzillaintegration/reportassistantdialog.cpp
index a5d7eeca..8d6f53f9 100644
--- drkonqi/bugzillaintegration/reportassistantdialog.cpp
+++ drkonqi/bugzillaintegration/reportassistantdialog.cpp
@@ -50,7 +50,7 @@ ReportAssistantDialog::ReportAssistantDialog(QWidget * parent) :
 
     //Set window properties
     setWindowTitle(i18nc("@title:window","Crash Reporting Assistant"));
-    setWindowIcon(QIcon::fromTheme(QStringLiteral("tools-report-bug")));
+    setWindowIcon(QIcon::fromTheme(QStringLiteral("tools-report-bug"), windowIcon()));
 
     connect(this, &ReportAssistantDialog::currentPageChanged, this, &ReportAssistantDialog::currentPageChanged_slot);
     connect(button(QDialogButtonBox::Help), &QPushButton::clicked, this, &ReportAssistantDialog::showHelp);
diff --git drkonqi/drkonqidialog.cpp drkonqi/drkonqidialog.cpp
index 46b404fc..1f77c805 100644
--- drkonqi/drkonqidialog.cpp
+++ drkonqi/drkonqidialog.cpp
@@ -53,7 +53,7 @@ DrKonqiDialog::DrKonqiDialog(QWidget * parent) :
 
     //Setting dialog title and icon
     setWindowTitle(DrKonqi::crashedApplication()->name());
-    setWindowIcon(QIcon::fromTheme(QStringLiteral("tools-report-bug")));
+    setWindowIcon(QIcon::fromTheme(QStringLiteral("tools-report-bug"), windowIcon()));
 
     QVBoxLayout* l = new QVBoxLayout(this);
     m_tabWidget = new QTabWidget(this);
diff --git drkonqi/main.cpp drkonqi/main.cpp
index e88a0646..a6b7e3de 100644
--- drkonqi/main.cpp
+++ drkonqi/main.cpp
@@ -36,6 +36,9 @@
 #if HAVE_X11
 #include <QX11Info>
 #endif
+#ifdef Q_OS_MACOS
+#include <KWindowSystem>
+#endif
 
 #include "drkonqi.h"
 #include "drkonqidialog.h"
@@ -79,7 +82,7 @@ int main(int argc, char* argv[])
                          QStringLiteral("gkiagia@users.sourceforge.net"));
     aboutData.addAuthor(i18nc("@info:credit","A. L. Spehr"), QString(),
                          QStringLiteral("spehr@kde.org"));
-    qa.setWindowIcon(QIcon::fromTheme(QStringLiteral("tools-report-bug")));
+    qa.setWindowIcon(QIcon::fromTheme(QStringLiteral("tools-report-bug"), qa.windowIcon()));
 
     QCommandLineParser parser;
     parser.setApplicationDescription(description);
@@ -152,6 +155,9 @@ int main(int argc, char* argv[])
         DrKonqiDialog *w = new DrKonqiDialog();
         QObject::connect(w, &DrKonqiDialog::rejected, &qa, &QApplication::quit);
         w->show();
+#ifdef Q_OS_MACOS
+        KWindowSystem::forceActiveWindow(w->winId());
+#endif
     };
 
     bool restarted = parser.isSet(restartedOption);
