diff --git kioslave/desktop/kio_desktop.cpp kioslave/desktop/kio_desktop.cpp
index 06616d1ae1d55a3032865de6e90c9b704827a3e0..d70b89601897f5512367c4a89c5a85af3254000f 100644
--- kioslave/desktop/kio_desktop.cpp
+++ kioslave/desktop/kio_desktop.cpp
@@ -30,7 +30,6 @@
 #include <QCoreApplication>
 #include <QFile>
 #include <QDBusInterface>
-#include <QDesktopServices>
 #include <QDir>
 #include <QStandardPaths>
 
@@ -65,10 +64,10 @@ DesktopProtocol::~DesktopProtocol()
 void DesktopProtocol::checkLocalInstall()
 {
 #ifndef Q_WS_WIN
-    // We can't use QStandardPaths::writableLocation(QStandardPaths::DesktopLocation) here, since it returns the home dir
-    // if the desktop folder doesn't exist.
-    QString desktopPath = QDesktopServices::storageLocation(QDesktopServices::DesktopLocation);
-    if (desktopPath.isEmpty())
+    // QStandardPaths::writableLocation(QStandardPaths::DesktopLocation) returns the home dir
+    // if the desktop folder doesn't exist, so verify its result
+    QString desktopPath = QStandardPaths::writableLocation(QStandardPaths::DesktopLocation);
+    if (desktopPath.isEmpty() || desktopPath == QDir::homePath())
         desktopPath = QDir::homePath() + "/Desktop";
 
     const QDir desktopDir(desktopPath);
