--- project/release/orig.fetch_l10n.rb	2014-09-18 12:29:49.000000000 +0200
+++ project/release/fetch_l10n.rb	2014-09-18 12:30:36.000000000 +0200
@@ -47,35 +47,35 @@
     Dir.mkdir( "doc-translated" )
 end
 
-Dir.chdir( "doc-translated" )
-docmakefile = File.new( "CMakeLists.txt", File::CREAT | File::RDWR | File::TRUNC )
-i18nlangs.each_line do |lang|
-    lang.chomp!()
-    if (lang != nil && lang != "")
-        if !(File.exists?(lang) && File.directory?(lang))
-            Dir.mkdir(lang)
-        end
-        Dir.chdir(lang)
-        for part in ['color-management', 'credits-annex', 'editor-color', 'editor-decorate', 'editor-enhance', 'editor-filters', 'editor-transform', 'file-formats', 'ie-menu', 'index', 'menu-descriptions', 'photo-editing', 'sidebar']
-            puts "Copying #{lang}'s #{part}.docbook over..  "
-            if isWindows
-                `svn cat svn://anonsvn.kde.org/home/kde/#{branch}/l10n-kde4/#{lang}/docs/extragear-graphics/digikam/#{part}.docbook > #{part}.docbook`
-            else
-                `svn cat svn://anonsvn.kde.org/home/kde/#{branch}/l10n-kde4/#{lang}/docs/extragear-graphics/digikam/#{part}.docbook 2> /dev/null | tee #{part}.docbook`
-            end
-            if File.exists?("#{part}.docbook") and FileTest.size( "#{part}.docbook" ) == 0
-                File.delete( "#{part}.docbook" )
-                puts "Delete File #{part}.docbook"
-            end
-            makefile = File.new( "CMakeLists.txt", File::CREAT | File::RDWR | File::TRUNC )
-            makefile << "kde4_create_handbook( index.docbook INSTALL_DESTINATION ${HTML_INSTALL_DIR}/#{lang}/ SUBDIR digikam )"
-            makefile.close()
-	    puts( "done.\n" )
-        end
-        Dir.chdir("..")
-    end
-end
-
+#Dir.chdir( "doc-translated" )
+#docmakefile = File.new( "CMakeLists.txt", File::CREAT | File::RDWR | File::TRUNC )
+#i18nlangs.each_line do |lang|
+#    lang.chomp!()
+#    if (lang != nil && lang != "")
+#        if !(File.exists?(lang) && File.directory?(lang))
+#            Dir.mkdir(lang)
+#        end
+#        Dir.chdir(lang)
+#        for part in ['color-management', 'credits-annex', 'editor-color', 'editor-decorate', 'editor-enhance', 'editor-filters', 'editor-transform', 'file-formats', 'ie-menu', 'index', 'menu-descriptions', 'photo-editing', 'sidebar']
+#            puts "Copying #{lang}'s #{part}.docbook over..  "
+#            if isWindows
+#                `svn cat svn://anonsvn.kde.org/home/kde/#{branch}/l10n-kde4/#{lang}/docs/extragear-graphics/digikam/#{part}.docbook > #{part}.docbook`
+#            else
+#                `svn cat svn://anonsvn.kde.org/home/kde/#{branch}/l10n-kde4/#{lang}/docs/extragear-graphics/digikam/#{part}.docbook 2> /dev/null | tee #{part}.docbook`
+#            end
+#            if File.exists?("#{part}.docbook") and FileTest.size( "#{part}.docbook" ) == 0
+#                File.delete( "#{part}.docbook" )
+#                puts "Delete File #{part}.docbook"
+#            end
+#            makefile = File.new( "CMakeLists.txt", File::CREAT | File::RDWR | File::TRUNC )
+#            makefile << "kde4_create_handbook( index.docbook INSTALL_DESTINATION ${HTML_INSTALL_DIR}/#{lang}/ SUBDIR digikam )"
+#            makefile.close()
+#	    puts( "done.\n" )
+#        end
+#        Dir.chdir("..")
+#    end
+#end
+#
 #################
 
 if !(File.exists?("po") && File.directory?("po"))
