commit 9c3f989e33eff7734a3da2a1f0d67495f6f29616
Author: René Bertin <rjvbertin@gmail.com>
Date:   Thu May 30 13:59:42 2024 +0200

    lamarck/patch-configure-libcurl.diff

diff --git a/configure b/configure
index e71cddeaf..2c9c18aff 100755
--- a/configure
+++ b/configure
@@ -7961,15 +7961,16 @@ fi
 	fi
 
 	CFLAGS_LIBCURL=$($CURL_CONFIG --cflags)
-	if test "x$curlprefix" = "x"; then
-		# System curl-config emits absurd output for --libs
-		# See rdar://7244457
-		LDFLAGS_LIBCURL="-lcurl"
-	else
-		# Due to a bug in dist, --arch flags are improperly supplied by curl-config.
-		# Get rid of them.
-		LDFLAGS_LIBCURL=$($CURL_CONFIG --libs | sed 's/-arch [A-Za-z0-9_]* //g')
-	fi
+# 	if test "x$curlprefix" = "x"; then
+# 		# System curl-config emits absurd output for --libs
+# 		# See rdar://7244457
+# 		LDFLAGS_LIBCURL="-lcurl"
+# 	else
+# 		# Due to a bug in dist, --arch flags are improperly supplied by curl-config.
+# 		# Get rid of them.
+# 		LDFLAGS_LIBCURL=$($CURL_CONFIG --libs | sed 's/-arch [A-Za-z0-9_]* //g')
+# 	fi
+    LDFLAGS_LIBCURL=$($CURL_CONFIG --libs)
 
 
 
