commit 412d6d2a9f0d34ecfc6fc5848934229c314245ab
Author: René Bertin <rjvbertin@gmail.com>
Date:   Thu May 30 13:59:45 2024 +0200

    lamarck/patch-machistawrap.diff

diff --git a/src/machista1.0/machista_wrap.c b/src/machista1.0/machista_wrap.c
index eb04b779b..6a4c57e67 100644
--- a/src/machista1.0/machista_wrap.c
+++ b/src/machista1.0/machista_wrap.c
@@ -1866,9 +1866,11 @@ _wrap_macho_loadcmd_mlt_install_name_get(ClientData clientData SWIGUNUSED, Tcl_I
     SWIG_exception_fail(SWIG_ArgError(res1), "in method '" "macho_loadcmd_mlt_install_name_get" "', argument " "1"" of type '" "struct macho_loadcmd *""'"); 
   }
   arg1 = (struct macho_loadcmd *)(argp1);
-  result = (char *) ((arg1)->mlt_install_name);
-  Tcl_SetObjResult(interp,SWIG_FromCharPtr((const char *)result));
-  return TCL_OK;
+  if (arg1) {
+    result = (char *) ((arg1)->mlt_install_name);
+    Tcl_SetObjResult(interp,SWIG_FromCharPtr((const char *)result));
+    return TCL_OK;
+  }
 fail:
   return TCL_ERROR;
 }
