commit b9e1dc1b8e6c690d773c3c18a5f1be42b4055a50
Author: René Bertin <rjvbertin@gmail.com>
Date:   Thu May 30 13:59:45 2024 +0200

    lamarck/patch-pextlib-no-sip.diff

diff --git a/src/pextlib1.0/sip_copy_proc.c b/src/pextlib1.0/sip_copy_proc.c
index b329ee78c..954cd4226 100644
--- a/src/pextlib1.0/sip_copy_proc.c
+++ b/src/pextlib1.0/sip_copy_proc.c
@@ -274,6 +274,11 @@ static copy_needed_return_t copy_needed(const char *path, char *const argv[],
 }
 
 static char *lazy_copy(const char *path, struct stat *in_st) {
+#ifndef SF_RESTRICTED
+    (void) path;
+    (void) in_st;
+    return NULL;
+#else
     char *retval = NULL;
     uid_t euid = geteuid();
     int outfd = -1;
@@ -442,6 +447,7 @@ lazy_copy_out:
         free(target_path);
     }
     return retval;
+#endif // SF_RESTRICTED
 }
 
 /**
