commit fee8ce9f26e9aa6120dd37469c80d11179d1e6f2
Author: René Bertin <rjvbertin@gmail.com>
Date:   Thu May 30 13:59:48 2024 +0200

    lamarck/patch-xinstall-have_setmode.diff

diff --git a/src/pextlib1.0/Pextlib.c b/src/pextlib1.0/Pextlib.c
index 7406124f5..5b1c0f332 100644
--- a/src/pextlib1.0/Pextlib.c
+++ b/src/pextlib1.0/Pextlib.c
@@ -111,6 +111,8 @@ extern char **environ;
 
 #ifndef HAVE_SETMODE
 #include "setmode.h"
+#else
+#include <bsd/unistd.h>
 #endif
 
 __attribute__((format(printf, 3, 0)))
diff --git a/src/pextlib1.0/xinstall.c b/src/pextlib1.0/xinstall.c
index 7dcdf9f87..a4da7655e 100644
--- a/src/pextlib1.0/xinstall.c
+++ b/src/pextlib1.0/xinstall.c
@@ -108,6 +108,8 @@ extern int copyfile(const char *from, const char *to, void *state,
 
 #ifndef HAVE_SETMODE
 #include "setmode.h"
+#else
+#include <bsd/unistd.h>
 #endif
 
 /* Bootstrap aid - this doesn't exist in most older releases */
