commit 6b4656e91feb4f3a44fa4df85b694b6d90b53379
Author: René Bertin <rjvbertin@gmail.com>
Date:   Thu May 30 13:59:47 2024 +0200

    lamarck/patch-no-arch-flag.diff

diff --git a/src/port1.0/portconfigure.tcl b/src/port1.0/portconfigure.tcl
index 03162f36d..d692fdf6e 100644
--- a/src/port1.0/portconfigure.tcl
+++ b/src/port1.0/portconfigure.tcl
@@ -702,15 +702,20 @@ proc portconfigure::configure_get_universal_archflags {} {
 
 # internal proc to determine if the compiler supports -arch
 proc portconfigure::arch_flag_supported {compiler {multiple_arch_flags no}} {
-    if {${multiple_arch_flags}} {
-        return [regexp {^gcc-4|llvm|apple|clang} ${compiler}]
-    } else {
-        # GCC prior to 4.7 does not accept -arch flag
-        if {[regexp {^macports(?:-[^-]+)?-gcc-4\.[0-6]} ${compiler}]} {
-            return no
+    global os.platform
+    if {${os.platform} eq "darwin"} {
+        if {${multiple_arch_flags}} {
+            return [regexp {^gcc-4|llvm|apple|clang} ${compiler}]
         } else {
-            return yes
+            # GCC prior to 4.7 does not accept -arch flag
+            if {[regexp {^macports(?:-[^-]+)?-gcc-4\.[0-6]} ${compiler}]} {
+                return no
+            } else {
+                return yes
+            }
         }
+    } else {
+        return no
     }
 }
 
