commit 62cca1deb972dd61194de78b7d4efcdd397ec421
Author: René Bertin <rjvbertin@gmail.com>
Date:   Thu May 30 13:59:48 2024 +0200

    lamarck/patch_sysctlbyname_emu.diff

diff --git a/src/macports1.0/macports.tcl b/src/macports1.0/macports.tcl
index b064f174a..dbf055d1e 100644
--- a/src/macports1.0/macports.tcl
+++ b/src/macports1.0/macports.tcl
@@ -1220,6 +1220,19 @@ match macports.conf.default."
                     set macports::build_arch i386
                 }
             }
+        } elseif {$os_platform eq "linux"} {
+            try {
+                if {[string match *64-bit* [exec lscpu]]} {
+                    ui_msg "build_arch x86_64"
+                    set macports::build_arch x86_64
+                } else {
+                    ui_msg "build_arch i386"
+                    set macports::build_arch i386
+                }
+            } catch {{*} eCode eMessage} {
+                ui_debug "Error setting macports::build_arch: $eMessage"
+                set macports::build_arch {}
+            }
         } else {
             set macports::build_arch {}
         }
diff --git a/src/port1.0/portbuild.tcl b/src/port1.0/portbuild.tcl
index 570be435d..1e996519a 100644
--- a/src/port1.0/portbuild.tcl
+++ b/src/port1.0/portbuild.tcl
@@ -140,17 +140,25 @@ proc portbuild::build_getjobs {args} {
     # if set to '0', use the number of cores for the number of jobs
     if {$jobs == 0} {
         try -pass_signal {
-            set jobs [sysctl hw.activecpu]
+            if {[option os.platform] eq "linux"} {
+                set jobs [exec nproc]
+            } else {
+                set jobs [sysctl hw.activecpu]
+            }
         } catch {{*} eCode eMessage} {
             set jobs 2
-            ui_warn "failed to determine the number of available CPUs (probably not supported on this platform)"
+            ui_warn "failed to determine the number of available CPUs (probably not supported on this platform: $eMessage)"
             ui_warn "defaulting to $jobs jobs, consider setting buildmakejobs to a nonzero value in macports.conf"
         }
 
         try -pass_signal {
-            set memsize [sysctl hw.memsize]
-            if {$jobs > $memsize / (1024 * 1024 * 1024) + 1} {
-                set jobs [expr {$memsize / (1024 * 1024 * 1024) + 1}]
+            if {[option os.platform] eq "linux"} {
+                set memsize [expr [exec sed -e "s|MemTotal:\[ \]*||g" -e "s| kB.*||g" -e 1q /proc/meminfo] / (1024 * 1024)]
+            } else {
+                set memsize [expr [sysctl hw.memsize] / (1024 * 1024 * 1024)]
+            }
+            if {$jobs > $memsize + 1} {
+                set jobs [expr {$memsize + 1}]
             }
         } catch {*} {}
     }
