commit e03ec6d300b862adffb46ea37eb7743643b71b3b
Author: René Bertin <rjvbertin@gmail.com>
Date:   Thu May 30 13:59:47 2024 +0200

    lamarck/patch-vcs-from-macports.diff

diff --git a/src/port1.0/portfetch.tcl b/src/port1.0/portfetch.tcl
index 8cd93ea35..c2566bb98 100644
--- a/src/port1.0/portfetch.tcl
+++ b/src/port1.0/portfetch.tcl
@@ -66,7 +66,10 @@ commands svn
 default extract.suffix .tar.gz
 default fetch.type standard
 
-default bzr.cmd {[findBinary bzr $portutil::autoconf::bzr_path]}
+#default bzr.cmd {[findBinary bzr $portutil::autoconf::bzr_path]}
+# accessing ${prefix_frozen} directly here used to work, now we need a wrapper
+# (At least for building MacPorts?)
+default bzr.cmd {[portfetch::find_bzr_path]}
 default bzr.dir {${workpath}}
 default bzr.revision -1
 default bzr.pre_args "--builtin --no-aliases checkout --lightweight --verbose"
@@ -207,11 +210,16 @@ proc portfetch::set_fetch_type {option action args} {
 proc portfetch::find_git_path {args} {
     global prefix_frozen os.platform os.major
     # Oldest macOS version whose git can validate GitHub's SSL certificate.
-    if {${os.major} >= 14 || ${os.platform} ne "darwin"} {
-        return [findBinary git $portutil::autoconf::git_path]
-    } else {
+#     if {${os.major} >= 14 || ${os.platform} ne "darwin"} {
+#         return [findBinary git $portutil::autoconf::git_path]
+#     } else {
         return ${prefix_frozen}/bin/git
-    }
+#     }
+}
+
+proc portfetch::find_bzr_path {args} {
+    global prefix_frozen
+    return ${prefix_frozen}/bin/bzr
 }
 
 set_ui_prefix
