diff --git b/src/daemon/orig.main.c a/src/daemon/main.c
index 924a4d4..690e73b 100644
--- src/daemon/orig.main.c
+++ src/daemon/main.c
@@ -589,7 +589,7 @@ int main(int argc, char *argv[]) {
             pa_xfree(canonical_rp);
 
         } else
-            pa_log_warn("Couldn't canonicalize binary path, cannot self execute.");
+            pa_log_warn("Couldn't canonicalize binary path \"%s\", cannot self execute (%s).", PA_BINARY, pa_cstrerror(errno));
     }
 #endif
 
diff --git b/orig.meson.build a/meson.build
index d4cdbd6..181e825 100644
--- orig.meson.build
+++ meson.build
@@ -151,7 +151,7 @@ cdata.set_quoted('PA_BUILDDIR', meson.current_build_dir())
 cdata.set_quoted('PA_SOEXT', '.so')
 cdata.set_quoted('PA_DEFAULT_CONFIG_DIR', pulsesysconfdir)
 cdata.set('PA_DEFAULT_CONFIG_DIR_UNQUOTED', pulsesysconfdir)
-cdata.set_quoted('PA_BINARY', join_paths(bindir, 'pulseaudio'))
+cdata.set_quoted('PA_BINARY', join_paths(bindir, 'pulseaudio.daemon'))
 cdata.set_quoted('PA_SYSTEM_RUNTIME_PATH', join_paths(localstatedir, 'run', 'pulse'))
 cdata.set_quoted('PA_SYSTEM_CONFIG_PATH', join_paths(localstatedir, 'lib', 'pulse'))
 cdata.set_quoted('PA_SYSTEM_STATE_PATH', join_paths(localstatedir, 'lib', 'pulse'))
