diff --git a/CMakeLists.txt b/CMakeLists.txt
index 56654bf0..318c810a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1143,11 +1143,16 @@ else()
 
     if(BUILD_SHARED_LIBS)
         target_sources(zlib PRIVATE ${ZLIB_DLL_SRCS})
+        if (DEFINED BUILD_STATIC_LIBS AND BUILD_STATIC_LIBS)
+            add_library(zlibstatic STATIC ${ZLIB_ALL_SRCS})
+            set_target_properties(zlibstatic PROPERTIES OUTPUT_NAME z)
+            set(ZLIB_INSTALL_LIBRARIES zlibstatic)
+        endif()
     else()
         add_library(zlibstatic ALIAS zlib)
     endif()
 
-    set(ZLIB_INSTALL_LIBRARIES zlib)
+    set(ZLIB_INSTALL_LIBRARIES zlib ${ZLIB_INSTALL_LIBRARIES})
 endif()
 
 # INFO: Mimics official zlib CMake target
