diff --git src/3rdparty/masm/yarr/Yarr.h src/3rdparty/masm/yarr/Yarr.h
index ccf78f98..cbb42c60 100644
--- src/3rdparty/masm/yarr/Yarr.h
+++ src/3rdparty/masm/yarr/Yarr.h
@@ -27,7 +27,7 @@
 
 #pragma once
 
-#include <limits.h>
+#include <limits>
 #include "YarrErrorCode.h"
 
 namespace JSC { namespace Yarr {
diff --git src/qml/jsruntime/qv4propertykey_p.h src/qml/jsruntime/qv4propertykey_p.h
index 47867765..a8cc663a 100644
--- src/qml/jsruntime/qv4propertykey_p.h
+++ src/qml/jsruntime/qv4propertykey_p.h
@@ -50,6 +50,8 @@
 // We mean it.
 //
 
+#include <limits>
+
 #include <private/qv4global_p.h>
 
 QT_BEGIN_NAMESPACE
diff --git src/qmldebug/qqmlprofilerevent_p.h src/qmldebug/qqmlprofilerevent_p.h
index 1e205d8d..1f499792 100644
--- src/qmldebug/qqmlprofilerevent_p.h
+++ src/qmldebug/qqmlprofilerevent_p.h
@@ -49,6 +49,7 @@
 
 #include <initializer_list>
 #include <type_traits>
+#include <limits>
 
 //
 //  W A R N I N G
