diff --git a/src/main.cpp b/src/main.cpp
index 8c5e1b3..5412c0b 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -102,7 +102,7 @@ int main(int argc, char* argv[])
     app.setOrganizationDomain(aboutData.organizationDomain());
     app.setApplicationVersion(aboutData.version());
     app.setApplicationDisplayName(aboutData.displayName());
-    app.setWindowIcon(QIcon::fromTheme(QLatin1String("konversation")));
+    app.setWindowIcon(QIcon::fromTheme(QLatin1String("konversation"), app.windowIcon()));
     app.setAttribute(Qt::AA_UseHighDpiPixmaps, true);
 
     QCommandLineParser cmdLineParser;
