diff --git a/data/konversationui.rc b/data/konversationui.rc
index 171d32a..4fb8091 100644
--- a/data/konversationui.rc
+++ b/data/konversationui.rc
@@ -66,7 +66,7 @@
 
   <Menu name="tabContextMenu">
     <Action name="tab_notifications" />
-    <Action name="tab_encoding" />
+<!--     <Action name="tab_encoding" /> -->
     <Separator />
     <ActionList name="server_actions" />
     <Separator />

