diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index f808bb9..f939548 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1,3 +1,5 @@
+include(ECMAddAppIcon)
+
 include_directories(${CMAKE_CURRENT_SOURCE_DIR}/config)
 include_directories(config dcc irc viewer upnp)
 
@@ -208,6 +210,10 @@ ki18n_wrap_ui(konversation_SRCS
 
 kconfig_add_kcfg_files(konversation_SRCS config/preferences_base.kcfgc)
 
+# Sets the icon on Windows and OSX
+file(GLOB ICONS_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/../data/images/icons/*apps-konversation.png")
+ecm_add_app_icon(konversation_SRCS ICONS ${ICONS_SRCS})
+
 add_executable(konversation ${konversation_SRCS})
 
 target_link_libraries(konversation
