diff --git a/kcms/keyboard/iso_codes.h b/kcms/keyboard/iso_codes.h
index 126dd67..a3d7978 100644
--- a/kcms/keyboard/iso_codes.h
+++ b/kcms/keyboard/iso_codes.h
@@ -52,7 +52,7 @@ public:
 	/**
 	 * @param isoCode Code for iso standard, i.e. "639", for convenience there's iso_* constants defined
 	 */
-	explicit IsoCodes(const QString& isoCode, const QString& isoCodesXmlDir=QStringLiteral("/usr/share/xml/iso-codes"));
+	explicit IsoCodes(const QString& isoCode, const QString& isoCodesXmlDir=QStringLiteral("@PREFIX@/share/xml/iso-codes"));
 	~IsoCodes();
 
 	/**
diff --git a/kcms/lookandfeel/kcm.cpp b/kcms/lookandfeel/kcm.cpp
index bfdcc3f..4c14fd4 100644
--- a/kcms/lookandfeel/kcm.cpp
+++ b/kcms/lookandfeel/kcm.cpp
@@ -493,7 +493,7 @@ const QStringList KCMLookandFeel::cursorSearchPaths()
 
 #if XCURSOR_LIB_MAJOR == 1 && XCURSOR_LIB_MINOR < 1
     // These are the default paths Xcursor will scan for cursor themes
-    QString path("~/.icons:/usr/share/icons:/usr/share/pixmaps:/usr/X11R6/lib/X11/icons");
+    QString path("~/.icons:@PREFIX@/share/icons:@PREFIX@/share/pixmaps:/usr/X11R6/lib/X11/icons");
 
     // If XCURSOR_PATH is set, use that instead of the default path
     char *xcursorPath = std::getenv("XCURSOR_PATH");
