diff --git a/CMakeLists.txt b/CMakeLists.txt
index cf2e79a..441aef6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -5,6 +5,12 @@ set(PROJECT_VERSION_MAJOR 5)
 cmake_minimum_required(VERSION 2.8.12 FATAL_ERROR)
 set(QT_MIN_VERSION "5.5.0")
 
+if(APPLE)
+    option(PLASMA_DESKTOP_BUILD_ALL "Build the full set of desktop components" OFF)
+else()
+    option(PLASMA_DESKTOP_BUILD_ALL "Build the full set of desktop components" ON)
+endif()
+
 find_package(ECM 0.0.11 REQUIRED NO_MODULE)
 set(CMAKE_MODULE_PATH ${ECM_MODULE_PATH} ${ECM_KDE_MODULE_DIR} ${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules)
 
@@ -49,23 +55,26 @@ find_package(KF5 ${KF5_MIN_VERSION} REQUIRED COMPONENTS
     People
     DBusAddons
     Activities
-    ActivitiesStats
     Config
 )
 
-find_package(LibKWorkspace CONFIG REQUIRED)
-find_package(LibTaskManager CONFIG REQUIRED)
-find_package(KWinDBusInterface CONFIG REQUIRED)
-find_package(ScreenSaverDBusInterface CONFIG REQUIRED)
-find_package(KRunnerAppDBusInterface CONFIG REQUIRED)
-find_package(KSMServerDBusInterface CONFIG REQUIRED)
+if(PLASMA_DESKTOP_BUILD_ALL)
+    find_package(KF5 ${KF5_MIN_VERSION} REQUIRED COMPONENTS ActivitiesStats)
+    find_package(LibKWorkspace CONFIG REQUIRED)
+    find_package(LibTaskManager CONFIG REQUIRED)
+    find_package(KWinDBusInterface CONFIG REQUIRED)
+    find_package(ScreenSaverDBusInterface CONFIG REQUIRED)
+    find_package(KRunnerAppDBusInterface CONFIG REQUIRED)
+    find_package(KSMServerDBusInterface CONFIG REQUIRED)
+    find_package(PackageKitQt5 0.9)
+    set_package_properties(PackageKitQt5 PROPERTIES DESCRIPTION "Software Manager integration"
+                        TYPE OPTIONAL
+                        PURPOSE "Provides package management integration to the application launcher."
+                        )
+endif()
+
 find_package(KF5ItemModels CONFIG REQUIRED)
 find_package(KF5Emoticons CONFIG REQUIRED)
-find_package(PackageKitQt5 0.9)
-set_package_properties(PackageKitQt5 PROPERTIES DESCRIPTION "Software Manager integration"
-                       TYPE OPTIONAL
-                       PURPOSE "Provides package management integration to the application launcher."
-                      )
 
 # Baloo is currently part of the KF5 namespace, but it is not
 # a proper framework. Hence the strange versioning.
@@ -145,16 +154,20 @@ add_definitions(-DQT_NO_URL_CAST_FROM_STRING)
 add_subdirectory(layout-templates)
 
 add_subdirectory(doc)
-add_subdirectory(runners)
-add_subdirectory(containments)
-add_subdirectory(toolboxes)
-add_subdirectory(applets)
-add_subdirectory(dataengines)
+if(PLASMA_DESKTOP_BUILD_ALL)
+    add_subdirectory(runners)
+    add_subdirectory(containments)
+    add_subdirectory(toolboxes)
+    add_subdirectory(applets)
+    add_subdirectory(dataengines)
+endif()
 add_subdirectory(kcms)
 add_subdirectory(knetattach)
 add_subdirectory(attica-kde)
-add_subdirectory(imports/activitymanager/)
-add_subdirectory(solid-device-automounter)
+if(PLASMA_DESKTOP_BUILD_ALL)
+    add_subdirectory(imports/activitymanager/)
+    add_subdirectory(solid-device-automounter)
+endif()
 if(X11_Xkb_FOUND AND XCB_XKB_FOUND)
     add_subdirectory(kaccess)
 endif()
diff --git a/kcms/CMakeLists.txt b/kcms/CMakeLists.txt
index 53ca49a..49a544f 100644
--- a/kcms/CMakeLists.txt
+++ b/kcms/CMakeLists.txt
@@ -20,18 +20,22 @@ endif()
 
 add_subdirectory( access )
 add_subdirectory( dateandtime )
-add_subdirectory( autostart )
-add_subdirectory( ksplash )
+if(PLASMA_DESKTOP_BUILD_ALL)
+    add_subdirectory( autostart )
+    add_subdirectory( ksplash )
 
-add_subdirectory( launch )
+    add_subdirectory( launch )
+endif()
 add_subdirectory( colors )
 add_subdirectory( krdb )
 add_subdirectory( style )
 add_subdirectory( desktoptheme )
 add_subdirectory( standard_actions )
 add_subdirectory( keys )
-add_subdirectory( ksmserver )
-add_subdirectory( lookandfeel )
+if(PLASMA_DESKTOP_BUILD_ALL)
+    add_subdirectory( ksmserver )
+    add_subdirectory( lookandfeel )
+endif()
 
 add_subdirectory( hardware )
 add_subdirectory( desktoppaths )
