diff --git a/CMakeLists.txt b/CMakeLists.txt
index a7016bf..985b1d4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -50,19 +50,19 @@ find_package(KF5 ${KF5_MIN_VERSION} REQUIRED COMPONENTS
     DBusAddons
 )
 
-find_package(LibKWorkspace CONFIG REQUIRED)
-find_package(LibTaskManager CONFIG REQUIRED)
-find_package(KWinDBusInterface CONFIG REQUIRED)
-find_package(ScreenSaverDBusInterface CONFIG REQUIRED)
-find_package(KRunnerAppDBusInterface CONFIG REQUIRED)
-find_package(KSMServerDBusInterface CONFIG REQUIRED)
+# find_package(LibKWorkspace CONFIG REQUIRED)
+# find_package(LibTaskManager CONFIG REQUIRED)
+# find_package(KWinDBusInterface CONFIG REQUIRED)
+# find_package(ScreenSaverDBusInterface CONFIG REQUIRED)
+# find_package(KRunnerAppDBusInterface CONFIG REQUIRED)
+# find_package(KSMServerDBusInterface CONFIG REQUIRED)
 find_package(KF5ItemModels CONFIG REQUIRED)
 find_package(KF5Emoticons CONFIG REQUIRED)
-find_package(PackageKitQt5 0.9)
-set_package_properties(PackageKitQt5 PROPERTIES DESCRIPTION "Software Manager integration"
-                       TYPE OPTIONAL
-                       PURPOSE "Provides package management integration to the application launcher."
-                      )
+# find_package(PackageKitQt5 0.9)
+# set_package_properties(PackageKitQt5 PROPERTIES DESCRIPTION "Software Manager integration"
+#                        TYPE OPTIONAL
+#                        PURPOSE "Provides package management integration to the application launcher."
+#                       )
 
 find_package(KF5Activities ${KF5_MIN_VERSION})
 set_package_properties(KF5Activities PROPERTIES DESCRIPTION "management of Plasma activities"
@@ -148,15 +148,15 @@ plasma_install_package(layout-templates/org.kde.plasma.desktop.defaultPanel org.
 add_subdirectory(doc)
 add_subdirectory(libs)
 # add_subdirectory(runners)
-add_subdirectory(containments)
-add_subdirectory(toolboxes)
-add_subdirectory(applets)
-add_subdirectory(dataengines)
+# add_subdirectory(containments)
+# add_subdirectory(toolboxes)
+# add_subdirectory(applets)
+# add_subdirectory(dataengines)
 add_subdirectory(kcms)
 add_subdirectory(knetattach)
 add_subdirectory(attica-kde)
-add_subdirectory(imports/activitymanager/)
-add_subdirectory(solid-device-automounter)
+# add_subdirectory(imports/activitymanager/)
+# add_subdirectory(solid-device-automounter)
 if(X11_Xkb_FOUND AND XCB_XKB_FOUND)
     add_subdirectory(kaccess)
 endif()
diff --git a/kcms/CMakeLists.txt b/kcms/CMakeLists.txt
index 53ca49a..af63344 100644
--- a/kcms/CMakeLists.txt
+++ b/kcms/CMakeLists.txt
@@ -20,17 +20,17 @@ endif()
 
 add_subdirectory( access )
 add_subdirectory( dateandtime )
-add_subdirectory( autostart )
-add_subdirectory( ksplash )
+# add_subdirectory( autostart )
+# add_subdirectory( ksplash )
 
-add_subdirectory( launch )
+# add_subdirectory( launch )
 add_subdirectory( colors )
 add_subdirectory( krdb )
 add_subdirectory( style )
 add_subdirectory( desktoptheme )
 add_subdirectory( standard_actions )
 add_subdirectory( keys )
-add_subdirectory( ksmserver )
+# add_subdirectory( ksmserver )
 add_subdirectory( lookandfeel )
 
 add_subdirectory( hardware )
