diff --git a/kcms/keyboard/CMakeLists.txt b/kcms/keyboard/CMakeLists.txt
index bdd4219d..cce03663 100644
--- a/kcms/keyboard/CMakeLists.txt
+++ b/kcms/keyboard/CMakeLists.txt
@@ -136,6 +136,8 @@ ki18n_wrap_ui(kcm_keyboard_PART_SRCS kcm_keyboard.ui kcm_add_layout_dialog.ui kc
 
 add_library(kcm_keyboard MODULE ${kcm_keyboard_PART_SRCS})
 
+kde_target_enable_exceptions(kcm_keyboard PRIVATE)
+
 target_link_libraries(kcm_keyboard
     Qt5::Concurrent
     Qt5::Widgets
