diff --git a/kcms/fonts/fonts.cpp b/kcms/fonts/fonts.cpp
index b4de312179dc530470bd8182dc971b90dc38d61d..7d900125105833459137e5e1b327194650e95c25 100644
--- a/kcms/fonts/fonts.cpp
+++ b/kcms/fonts/fonts.cpp
@@ -681,6 +681,11 @@ KFonts::KFonts(QWidget *parent, const QVariantList &args)
                             " properly with real DPI value better fonts should be used or configuration"
                             " of font hinting should be checked.</p>");
     spinboxDpi->setWhatsThis(whatsthis);
+#if !defined(HAVE_X11) || !HAVE_X11
+    QString nothere = i18n("This property has no effect on this platform");
+    spinboxDpi->setToolTip(nothere);
+    checkboxForceDpi->setToolTip(nothere);
+#endif
     checkboxForceDpi->setChecked(false);
     spinboxDpi->setEnabled(false);
     connect(spinboxDpi, SIGNAL(valueChanged(int)), SLOT(changed()));
diff --git a/kcms/fonts/fonts.h b/kcms/fonts/fonts.h
index 3c7a180d4033c648de61572ee2a3a7ebe4e22658..2bff41291b9131c20e6116bf8477e2beb1f10aa2 100644
--- a/kcms/fonts/fonts.h
+++ b/kcms/fonts/fonts.h
@@ -142,11 +142,9 @@ private:
     QPushButton *aaSettingsButton;
     FontAASettings *aaSettings;
 #endif
-#if HAVE_X11
     int dpi_original;
     QCheckBox *checkboxForceDpi;
     QSpinBox *spinboxDpi;
-#endif
     QList<FontUseItem *> fontUseList;
 };
 
