diff --git a/kcms/krdb/krdb.cpp b/kcms/krdb/krdb.cpp
index 7ee0e90..b5e1025 100644
--- a/kcms/krdb/krdb.cpp
+++ b/kcms/krdb/krdb.cpp
@@ -62,14 +62,14 @@ inline const char * sysGtkrc(int version)
 	if(access("/etc/opt/gnome/gtk-2.0", F_OK) == 0)
 	    return "/etc/opt/gnome/gtk-2.0/gtkrc";
 	else
-	    return "/etc/gtk-2.0/gtkrc";
+	    return "@PREFIX@/etc/gtk-2.0/gtkrc";
     }
     else
     {
 	if(access("/etc/opt/gnome/gtk", F_OK) == 0)
 	    return "/etc/opt/gnome/gtk/gtkrc";
 	else
-	    return "/etc/gtk/gtkrc";
+	    return "@PREFIX@/etc/gtk/gtkrc";
     }
 }
 
