diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2a0161f..06c9fde 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -30,10 +30,12 @@ find_package(Qt5 ${QT_MIN_VERSION} CONFIG REQUIRED COMPONENTS
     QuickWidgets
     DBus
     Widgets
-    X11Extras
     Svg
     Concurrent
 )
+if(NOT APPLE)
+    find_package(Qt5 ${QT_MIN_VERSION} CONFIG REQUIRED COMPONENTS X11Extras)
+endif()
 
 set(KF5_MIN_VERSION "5.20.0")
 find_package(KF5 ${KF5_MIN_VERSION} REQUIRED COMPONENTS
@@ -75,15 +77,17 @@ endif()
 find_package(KF5ItemModels CONFIG REQUIRED)
 find_package(KF5Emoticons CONFIG REQUIRED)
 
-# Baloo is currently part of the KF5 namespace, but it is not
-# a proper framework. Hence the strange versioning.
-# This will be fixed with Plasma 5.2, as Baloo should have
-# become a framework by then
-find_package(KF5Baloo "5.1.90")
-set_package_properties(KF5Baloo PROPERTIES DESCRIPTION "File Searching"
-                       TYPE REQUIRED
-                       PURPOSE "Needed to build to File Search KCM"
-                      )
+if(NOT APPLE)
+    # Baloo is currently part of the KF5 namespace, but it is not
+    # a proper framework. Hence the strange versioning.
+    # This will be fixed with Plasma 5.2, as Baloo should have
+    # become a framework by then
+    find_package(KF5Baloo "5.1.90")
+    set_package_properties(KF5Baloo PROPERTIES DESCRIPTION "File Searching"
+                        TYPE REQUIRED
+                        PURPOSE "Needed to build to File Search KCM"
+                        )
+endif()
 
 find_package(Fontconfig)
 set_package_properties(Fontconfig PROPERTIES DESCRIPTION "Font access configuration library"
@@ -92,38 +96,42 @@ set_package_properties(Fontconfig PROPERTIES DESCRIPTION "Font access configurat
                        PURPOSE "Needed to build font configuration and installation tools"
                       )
 
-find_package(X11)
-set_package_properties(X11 PROPERTIES DESCRIPTION "X11 libraries"
-                        URL "http://www.x.org"
-                        TYPE REQUIRED
-                        PURPOSE "Required for building the X11 based workspace"
-                      )
-
-find_package(UDev)
-set_package_properties(UDev PROPERTIES DESCRIPTION "UDev library"
-                        URL "http://www.kernel.org/pub/linux/utils/kernel/hotplug/udev.html"
-                        TYPE OPTIONAL
-                        PURPOSE "Required for device discovery in keyboard daemon"
-                      )
-
-find_package(XCB
-    REQUIRED COMPONENTS
-        XCB SHM IMAGE
-    OPTIONAL_COMPONENTS
-        XKB XINPUT
-)
-set_package_properties(XCB PROPERTIES TYPE REQUIRED)
-add_feature_info("XCB-XKB" XCB_XKB_FOUND "Required for building kcm/keyboard")
-add_feature_info("libxft" X11_Xft_FOUND "X FreeType interface library required for font installation")
+if(NOT APPLE)
+    find_package(X11)
+    set_package_properties(X11 PROPERTIES DESCRIPTION "X11 libraries"
+                            URL "http://www.x.org"
+                            TYPE REQUIRED
+                            PURPOSE "Required for building the X11 based workspace"
+                        )
 
-find_package(Evdev)
-set_package_properties(Evdev PROPERTIES TYPE OPTIONAL)
-add_feature_info("Evdev" EVDEV_FOUND "Evdev driver headers needed for input KCM")
+    find_package(UDev)
+    set_package_properties(UDev PROPERTIES DESCRIPTION "UDev library"
+                            URL "http://www.kernel.org/pub/linux/utils/kernel/hotplug/udev.html"
+                            TYPE OPTIONAL
+                            PURPOSE "Required for device discovery in keyboard daemon"
+                        )
 
-find_package(Synaptics)
-set_package_properties(Synaptics PROPERTIES TYPE OPTIONAL)
-add_feature_info("Synaptics" SYNAPTICS_FOUND "Synaptics libraries needed for touchpad KCM")
-include(ConfigureChecks.cmake)
+    find_package(XCB
+        REQUIRED COMPONENTS
+            XCB SHM IMAGE
+        OPTIONAL_COMPONENTS
+            XKB XINPUT
+    )
+    set_package_properties(XCB PROPERTIES TYPE REQUIRED)
+    add_feature_info("XCB-XKB" XCB_XKB_FOUND "Required for building kcm/keyboard")
+    add_feature_info("libxft" X11_Xft_FOUND "X FreeType interface library required for font installation")
+
+    find_package(Evdev)
+    set_package_properties(Evdev PROPERTIES TYPE OPTIONAL)
+    add_feature_info("Evdev" EVDEV_FOUND "Evdev driver headers needed for input KCM")
+
+    find_package(Synaptics)
+    set_package_properties(Synaptics PROPERTIES TYPE OPTIONAL)
+    add_feature_info("Synaptics" SYNAPTICS_FOUND "Synaptics libraries needed for touchpad KCM")
+    include(ConfigureChecks.cmake)
+else()
+    set(X11_FOUND 0)
+endif()
 
 if(${Qt5Gui_OPENGL_IMPLEMENTATION} STREQUAL "GL")
     find_package(OpenGL)
@@ -159,7 +167,9 @@ if(PLASMA_DESKTOP_BUILD_ALL)
     add_subdirectory(dataengines)
 endif()
 add_subdirectory(kcms)
-add_subdirectory(knetattach)
+if(NOT APPLE)
+    add_subdirectory(knetattach)
+endif()
 add_subdirectory(attica-kde)
 if(PLASMA_DESKTOP_BUILD_ALL)
     add_subdirectory(imports/activitymanager/)
diff --git a/kcms/CMakeLists.txt b/kcms/CMakeLists.txt
index 49a544f..5d407aa 100644
--- a/kcms/CMakeLists.txt
+++ b/kcms/CMakeLists.txt
@@ -18,8 +18,12 @@ if (EVDEV_FOUND AND X11_Xinput_FOUND)
     add_subdirectory( input )
 endif()
 
-add_subdirectory( access )
-add_subdirectory( dateandtime )
+if(HAVE_X11)
+    add_subdirectory( access )
+endif()
+if(NOT APPLE)
+    add_subdirectory( dateandtime )
+endif()
 if(PLASMA_DESKTOP_BUILD_ALL)
     add_subdirectory( autostart )
     add_subdirectory( ksplash )
@@ -29,7 +33,9 @@ endif()
 add_subdirectory( colors )
 add_subdirectory( krdb )
 add_subdirectory( style )
-add_subdirectory( desktoptheme )
+if(NOT APPLE)
+    add_subdirectory( desktoptheme )
+endif()
 add_subdirectory( standard_actions )
 add_subdirectory( keys )
 if(PLASMA_DESKTOP_BUILD_ALL)
@@ -37,8 +43,10 @@ if(PLASMA_DESKTOP_BUILD_ALL)
     add_subdirectory( lookandfeel )
 endif()
 
-add_subdirectory( hardware )
-add_subdirectory( desktoppaths )
+if(NOT APPLE)
+    add_subdirectory( hardware )
+    add_subdirectory( desktoppaths )
+endif()
 
 add_subdirectory(activities)
 add_subdirectory(componentchooser)
@@ -50,18 +58,20 @@ add_subdirectory(formats)
 add_subdirectory(translations)
 add_subdirectory(spellchecking)
 add_subdirectory(phonon)
-add_subdirectory(runners)
-# useraccount replaced by user-manager for now
-#add_subdirectory(useraccount)
+if(NOT APPLE)
+    add_subdirectory(runners)
+    # useraccount replaced by user-manager for now
+    #add_subdirectory(useraccount)
 
-#yet the icons still need installing
-add_subdirectory(useraccount/pics)
+    #yet the icons still need installing
+    add_subdirectory(useraccount/pics)
 
 
-add_subdirectory(workspaceoptions)
-add_subdirectory(baloo)
-add_subdirectory(solid_actions)
-add_subdirectory(cursortheme)
+    add_subdirectory(workspaceoptions)
+    add_subdirectory(baloo)
+    add_subdirectory(solid_actions)
+    add_subdirectory(cursortheme)
+endif()
 
 if (SYNAPTICS_FOUND AND X11_Xinput_FOUND)
     add_subdirectory(touchpad)
diff --git a/kcms/componentchooser/CMakeLists.txt b/kcms/componentchooser/CMakeLists.txt
index 5820689..079a194 100644
--- a/kcms/componentchooser/CMakeLists.txt
+++ b/kcms/componentchooser/CMakeLists.txt
@@ -32,8 +32,10 @@ add_library(kcm_componentchooser MODULE ${kcm_componentchooser_SRCS})
 
 target_link_libraries(kcm_componentchooser
                       KF5::KDELibs4Support
-                      Qt5::X11Extras
                       )
+if(X11_FOUND)
+    target_link_libraries(kcm_componentchooser Qt5::X11Extras)
+endif()
 
 install(TARGETS kcm_componentchooser  DESTINATION ${PLUGIN_INSTALL_DIR} )
 
diff --git a/kcms/krdb/CMakeLists.txt b/kcms/krdb/CMakeLists.txt
index 3c89539..e60f650 100644
--- a/kcms/krdb/CMakeLists.txt
+++ b/kcms/krdb/CMakeLists.txt
@@ -1,11 +1,13 @@
-# KI18N Translation Domain for this library
-add_definitions(-DTRANSLATION_DOMAIN=\"krdb\")
+if(X11_FOUND)
+    # KI18N Translation Domain for this library
+    add_definitions(-DTRANSLATION_DOMAIN=\"krdb\")
 
-add_subdirectory( ad ) 
-add_subdirectory( pics ) 
+    add_subdirectory( ad ) 
+    add_subdirectory( pics ) 
 
-install(FILES krdb_libpathwipe.upd DESTINATION ${DATA_INSTALL_DIR}/kconf_update)
+    install(FILES krdb_libpathwipe.upd DESTINATION ${DATA_INSTALL_DIR}/kconf_update)
 
-add_executable(krdb_clearlibrarypath krdb_clearlibrarypath.cpp)
-target_link_libraries(krdb_clearlibrarypath Qt5::Core KF5::KDELibs4Support)
-install(TARGETS krdb_clearlibrarypath DESTINATION ${LIB_INSTALL_DIR}/kconf_update_bin)
+    add_executable(krdb_clearlibrarypath krdb_clearlibrarypath.cpp)
+    target_link_libraries(krdb_clearlibrarypath Qt5::Core KF5::KDELibs4Support)
+    install(TARGETS krdb_clearlibrarypath DESTINATION ${LIB_INSTALL_DIR}/kconf_update_bin)
+endif()
diff --git a/kcms/krdb/krdb.cpp b/kcms/krdb/krdb.cpp
index b5e1025..a564e2b 100644
--- a/kcms/krdb/krdb.cpp
+++ b/kcms/krdb/krdb.cpp
@@ -423,6 +423,7 @@ void runRdb( uint flags )
   createGtkrc( exportColors, newPal, exportGtkTheme, gtkTheme, 1 );
   createGtkrc( exportColors, newPal, exportGtkTheme, gtkTheme, 2 );
 
+#if HAVE_X11
   // Export colors to non-(KDE/Qt) apps (e.g. Motif, GTK+ apps)
   if (exportColors)
   {
@@ -551,6 +552,7 @@ void runRdb( uint flags )
   proc << "xrdb" << "-quiet" << "-merge" << tmpFile.fileName();
 #endif
   proc.execute();
+#endif
 
   applyGtkStyles(exportColors, 1);
   applyGtkStyles(exportColors, 2);
diff --git a/kcms/style/CMakeLists.txt b/kcms/style/CMakeLists.txt
index afd603b..b5f88d3 100644
--- a/kcms/style/CMakeLists.txt
+++ b/kcms/style/CMakeLists.txt
@@ -13,7 +13,6 @@ ki18n_wrap_ui(kcm_style_PART_SRCS stylepreview.ui finetuning.ui)
 add_library(kcm_style MODULE ${kcm_style_PART_SRCS})
 
 target_link_libraries(kcm_style
-    Qt5::X11Extras
     Qt5::DBus
     KF5::KCMUtils
     KF5::Completion
@@ -24,6 +23,9 @@ target_link_libraries(kcm_style
     KF5::KDELibs4Support
     KF5::NewStuff
 )
+if(NOT APPLE)
+    target_link_libraries(kcm_style Qt5::X11Extras)
+endif()
 
 install(TARGETS kcm_style  DESTINATION ${PLUGIN_INSTALL_DIR})
 
diff --git a/knetattach/CMakeLists.txt b/knetattach/CMakeLists.txt
index ab56962..087ad7e 100644
--- a/knetattach/CMakeLists.txt
+++ b/knetattach/CMakeLists.txt
@@ -6,6 +6,7 @@ ki18n_wrap_ui(knetattach_SRCS knetattach.ui )
 # kde4_add_app_icon(knetattach_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/hi*-app-knetattach.png")
 
 add_executable(knetattach ${knetattach_SRCS})
+ecm_mark_nongui_executable(knetattach)
 
 target_compile_definitions(knetattach PRIVATE -DPROJECT_VERSION="${PROJECT_VERSION}")
 
