--- src/orig.scanner.c	2017-01-25 00:27:07.000000000 +0100
+++ src/scanner.c	2017-05-01 22:35:32.000000000 +0200
@@ -43,9 +43,39 @@
 #if HAVE_LIBXML
 #include <libxml/parser.h>
 
-/* Embedded wayland.dtd file, see dtddata.S */
-extern char DTD_DATA_begin;
-extern int DTD_DATA_len;
+char DTD_DATA_begin[] =
+"<!ELEMENT protocol (copyright?, description?, interface+)>\n\
+  <!ATTLIST protocol name CDATA #REQUIRED>\n\
+<!ELEMENT copyright (#PCDATA)>\n\
+<!ELEMENT interface (description?,(request|event|enum)+)>\n\
+  <!ATTLIST interface name CDATA #REQUIRED>\n\
+  <!ATTLIST interface version CDATA #REQUIRED>\n\
+<!ELEMENT request (description?,arg*)>\n\
+  <!ATTLIST request name CDATA #REQUIRED>\n\
+  <!ATTLIST request type CDATA #IMPLIED>\n\
+  <!ATTLIST request since CDATA #IMPLIED>\n\
+<!ELEMENT event (description?,arg*)>\n\
+  <!ATTLIST event name CDATA #REQUIRED>\n\
+  <!ATTLIST event since CDATA #IMPLIED>\n\
+<!ELEMENT enum (description?,entry*)>\n\
+  <!ATTLIST enum name CDATA #REQUIRED>\n\
+  <!ATTLIST enum since CDATA #IMPLIED>\n\
+  <!ATTLIST enum bitfield CDATA #IMPLIED>\n\
+<!ELEMENT entry (description?)>\n\
+  <!ATTLIST entry name CDATA #REQUIRED>\n\
+  <!ATTLIST entry value CDATA #REQUIRED>\n\
+  <!ATTLIST entry summary CDATA #IMPLIED>\n\
+  <!ATTLIST entry since CDATA #IMPLIED>\n\
+<!ELEMENT arg (description?)>\n\
+  <!ATTLIST arg name CDATA #REQUIRED>\n\
+  <!ATTLIST arg type CDATA #REQUIRED>\n\
+  <!ATTLIST arg summary CDATA #IMPLIED>\n\
+  <!ATTLIST arg interface CDATA #IMPLIED>\n\
+  <!ATTLIST arg allow-null CDATA #IMPLIED>\n\
+  <!ATTLIST arg enum CDATA #IMPLIED>\n\
+<!ELEMENT description (#PCDATA)>\n\
+  <!ATTLIST description summary CDATA #REQUIRED>\n";
+int DTD_DATA_len = sizeof(DTD_DATA_begin)/sizeof(char);
 #endif
 
 #include "wayland-util.h"
@@ -100,7 +130,7 @@
 	if (!ctx || !dtdctx)
 		abort();
 
-	buffer = xmlParserInputBufferCreateMem(&DTD_DATA_begin,
+	buffer = xmlParserInputBufferCreateMem(DTD_DATA_begin,
 					       DTD_DATA_len,
 					       XML_CHAR_ENCODING_UTF8);
 	if (!buffer) {
--- src/orig.dtddata.S	2017-02-07 23:59:06.000000000 +0100
+++ src/dtddata.S	2017-05-01 22:32:25.000000000 +0200
@@ -46,5 +46,7 @@
 	.int (\name\()_end - \name\()_begin)
 .endm
 
+#if defined(__linux__) && defined(__ELF__)
 .section .rodata
 binfile DTD_DATA src/wayland.dtd.embed
+#endif
