diff --git kde-modules/KDECompilerSettings.cmake kde-modules/KDECompilerSettings.cmake
index 8f87ecc1970b3523890e0c91d7627dd7ec5fb542..ae2eaffc79273185a196ee00986a017245f3c09e 100644
--- kde-modules/KDECompilerSettings.cmake
+++ kde-modules/KDECompilerSettings.cmake
@@ -219,6 +219,27 @@ if (CMAKE_C_COMPILER_ID STREQUAL "GNU" OR
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-operator-names")
 endif()
 
+# allow projects that do require named operators to (re)activate them
+# (this includes projects using boost)
+function(KDE_ENABLE_NAMED_OPERATORS)
+    if (MSVC)
+        if (${MSVC_VERSION} GREATER 1900)
+            # /permissive- became the preferred switch to obtain standards-compliant
+            # behaviour somewhere after MSVC_VERSION 1900 (MSVC 2015 Update 3 to be exact)
+            # Note the trailing dash, that's not a typo! See:
+            # https://blogs.msdn.microsoft.com/vcblog/2016/11/16/permissive-switch/
+            # and
+            # https://blogs.msdn.microsoft.com/vcblog/2016/06/07/standards-version-switches-in-the-compiler/
+            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /permissive-")
+        elseif()
+            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /Za")
+        endif()
+    else()
+        string(REPLACE "-fno-operator-names" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
+    endif()
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}" PARENT_SCOPE)
+endfunction()
+
 # Default to hidden visibility for symbols
 set(CMAKE_C_VISIBILITY_PRESET hidden)
 set(CMAKE_CXX_VISIBILITY_PRESET hidden)
