diff --git a/kde-modules/KDEInstallDirsCommon.cmake b/kde-modules/KDEInstallDirsCommon.cmake
index a502700851aa174e6a8ad0193096d9756b558b59..1a24154a87b68fd42e726ae7ddc031f1ddf4c06c 100644
--- a/kde-modules/KDEInstallDirsCommon.cmake
+++ b/kde-modules/KDEInstallDirsCommon.cmake
@@ -43,6 +43,10 @@ if((CMAKE_SYSTEM_NAME MATCHES "Linux|kFreeBSD" OR CMAKE_SYSTEM_NAME STREQUAL "GN
       endif()
     endif()
   endif()
+elseif (APPLE)
+  # for now we don't change the current behaviour by default
+  option(APPLE_FORCE_UNIX_DIRS "Force enable installation to standard Unix locations on Mac" ON)
+  option(APPLE_SUPPRESS_INSTALLDIRS_WARNING "Suppress warnings about the chosen install locations on Mac" OFF)
 endif()
 
 set(_gnu_install_dirs_vars
@@ -268,10 +272,18 @@ if (WIN32)
     _define_relative(DATAROOTDIR BINDIR "data"
         "read-only architecture-independent data root"
         SHARE_INSTALL_PREFIX)
+    set(KDE_INSTALL_CROSSPLATFORM_DATAROOTDIR ${KDE_INSTALL_DATAROOTDIR})
+elseif(APPLE AND NOT APPLE_FORCE_UNIX_DIRS)
+    _define_absolute(DATAROOTDIR "/Library/Application Support"
+        "read-only architecture-independent data root")
+    _define_absolute(CROSSPLATFORM_DATAROOTDIR "share"
+        "read-only architecture-independent data root"
+        SHARE_INSTALL_PREFIX)
 else()
     _define_absolute(DATAROOTDIR "share"
         "read-only architecture-independent data root"
         SHARE_INSTALL_PREFIX)
+    set(KDE_INSTALL_CROSSPLATFORM_DATAROOTDIR ${KDE_INSTALL_DATAROOTDIR})
 endif()
 
 _define_relative(DATADIR DATAROOTDIR ""
@@ -299,40 +311,40 @@ _define_relative(JARDIR "" "jar"
 
 
 # Cross-desktop or other system things
-_define_relative(ICONDIR DATAROOTDIR "icons"
+_define_relative(ICONDIR CROSSPLATFORM_DATAROOTDIR "icons"
     "icons"
     ICON_INSTALL_DIR)
-_define_relative(LOCALEDIR DATAROOTDIR "locale"
+_define_relative(LOCALEDIR CROSSPLATFORM_DATAROOTDIR "locale"
     "knotify description files"
     LOCALE_INSTALL_DIR)
-_define_relative(SOUNDDIR DATAROOTDIR "sounds"
+_define_relative(SOUNDDIR CROSSPLATFORM_DATAROOTDIR "sounds"
     "sound files"
     SOUND_INSTALL_DIR)
-_define_relative(TEMPLATEDIR DATAROOTDIR "templates"
+_define_relative(TEMPLATEDIR CROSSPLATFORM_DATAROOTDIR "templates"
     "templates"
     TEMPLATES_INSTALL_DIR)
-_define_relative(WALLPAPERDIR DATAROOTDIR "wallpapers"
+_define_relative(WALLPAPERDIR CROSSPLATFORM_DATAROOTDIR "wallpapers"
     "desktop wallpaper images"
     WALLPAPER_INSTALL_DIR)
-_define_relative(APPDIR DATAROOTDIR "applications"
+_define_relative(APPDIR CROSSPLATFORM_DATAROOTDIR "applications"
     "application desktop files"
     XDG_APPS_INSTALL_DIR)
-_define_relative(DESKTOPDIR DATAROOTDIR "desktop-directories"
+_define_relative(DESKTOPDIR CROSSPLATFORM_DATAROOTDIR "desktop-directories"
     "desktop directories"
     XDG_DIRECTORY_INSTALL_DIR)
-_define_relative(MIMEDIR DATAROOTDIR "mime/packages"
+_define_relative(MIMEDIR CROSSPLATFORM_DATAROOTDIR "mime/packages"
     "mime description files"
     XDG_MIME_INSTALL_DIR)
-_define_relative(METAINFODIR DATAROOTDIR "metainfo"
+_define_relative(METAINFODIR CROSSPLATFORM_DATAROOTDIR "metainfo"
     "AppStream component metadata")
-_define_relative(QCHDIR DATAROOTDIR "doc/qch"
+_define_relative(QCHDIR CROSSPLATFORM_DATAROOTDIR "doc/qch"
     "documentation bundles in QCH format")
-_define_relative(MANDIR DATAROOTDIR "man"
+_define_relative(MANDIR UNIXCROSSPLATFORM_DATAROOTDIR "man"
     "man documentation"
     MAN_INSTALL_DIR)
-_define_relative(INFODIR DATAROOTDIR "info"
+_define_relative(INFODIR CROSSPLATFORM_DATAROOTDIR "info"
     "info documentation")
-_define_relative(DBUSDIR DATAROOTDIR "dbus-1"
+_define_relative(DBUSDIR CROSSPLATFORM_DATAROOTDIR "dbus-1"
     "D-Bus")
 _define_relative(DBUSINTERFACEDIR DBUSDIR "interfaces"
     "D-Bus interfaces"
diff --git a/kde-modules/KDEInstallDirs5.cmake b/kde-modules/KDEInstallDirs5.cmake
index a7e12b797a6a54dc2afd9695883e830fdb87b943..a3c26cbe08d1aad42409d106fd3c828252b94d30 100644
--- a/kde-modules/KDEInstallDirs5.cmake
+++ b/kde-modules/KDEInstallDirs5.cmake
@@ -239,6 +239,9 @@ if(NOT DEFINED KDE_INSTALL_USE_QT_SYS_PATHS)
     if(qt_install_prefix_dir STREQUAL "${CMAKE_INSTALL_PREFIX}")
         message(STATUS "Installing in the same prefix as Qt, adopting their path scheme.")
         set(_default_KDE_INSTALL_USE_QT_SYS_PATHS ON)
+    elseif(APPLE AND NOT APPLE_FORCE_UNIX_DIRS)
+        message(STATUS "Using native Mac installation locations, adopting Qt's path scheme.")
+        set(_default_KDE_INSTALL_USE_QT_SYS_PATHS ON)
     endif()
 endif()
 
@@ -301,11 +304,24 @@ if(NOT KDE_INSTALL_DIRS_NO_DEPRECATED)
 endif()
 
 
-_define_non_cache(DATADIR_KF5 "${CMAKE_INSTALL_DATADIR}/kf5")
+if(APPLE AND NOT APPLE_FORCE_UNIX_DIRS)
+    _define_non_cache(DATADIR_KF5 "/Library/Application Support/kf5")
+else()
+    _define_non_cache(DATADIR_KF5 "${CMAKE_INSTALL_DATADIR}/kf5")
+endif()
 if(NOT KDE_INSTALL_DIRS_NO_DEPRECATED)
     set(KF5_DATA_INSTALL_DIR "${CMAKE_INSTALL_DATADIR_KF5}")
 endif()
 
+if (APPLE AND NOT APPLE_SUPPRESS_INSTALLDIRS_WARNING)
+  if(APPLE_FORCE_UNIX_DIRS)
+    # for now we print a regular status message for the traditional default
+    message(STATUS "Using standard Unix installation directories under ${CMAKE_INSTALL_PREFIX}/${KDE_INSTALL_DATAROOTDIR}")
+  else()
+    message(WARNING "Using Mac-specific installation directories under ${KDE_INSTALL_DATAROOTDIR}")
+  endif()
+endif()
+
 # Qt-specific data vars
 if(KDE_INSTALL_USE_QT_SYS_PATHS)
     query_qmake(qt_docs_dir QT_INSTALL_DOCS)
@@ -363,7 +379,11 @@ endif()
 # version-less forward compatibility variants, see also KDEInstallDirs6.cmake
 set(KF_INSTALL_TARGETS_DEFAULT_ARGS ${KF5_INSTALL_TARGETS_DEFAULT_ARGS})
 _define_non_cache(INCLUDEDIR_KF "${KDE_INSTALL_INCLUDEDIR_KF5}")
-_define_non_cache(DATADIR_KF "${KDE_INSTALL_DATADIR_KF5}")
+if(APPLE AND NOT APPLE_FORCE_UNIX_DIRS)
+    _define_non_cache(DATADIR_KF "/Library/Application Support/kf5")
+else()
+    _define_non_cache(DATADIR_KF "${KDE_INSTALL_DATADIR_KF5}/kf5")
+endif()
 _define_non_cache(LIBEXECDIR_KF "${KDE_INSTALL_LIBEXECDIR_KF5}")
 
 include(${CMAKE_CURRENT_LIST_DIR}/KDESetupPrefixScript.cmake)
diff --git a/kde-modules/KDEInstallDirs6.cmake b/kde-modules/KDEInstallDirs6.cmake
index fecd1b10b5202f1a2b2f81d46cb10243cfdfad6e..8321e78e9ed368ff79c0d176a59e54a63856ad7e 100644
--- a/kde-modules/KDEInstallDirs6.cmake
+++ b/kde-modules/KDEInstallDirs6.cmake
@@ -207,6 +207,9 @@ if(NOT DEFINED KDE_INSTALL_USE_QT_SYS_PATHS)
     #if(qt_install_prefix_dir STREQUAL "${CMAKE_INSTALL_PREFIX}")
     #    message(STATUS "Installing in the same prefix as Qt, adopting their path scheme.")
     #    set(_default_KDE_INSTALL_USE_QT_SYS_PATHS ON)
+    #elseif(APPLE AND NOT APPLE_FORCE_UNIX_DIRS)
+    #    message(STATUS "Using native Mac installation locations, adopting Qt's path scheme.")
+    #    set(_default_KDE_INSTALL_USE_QT_SYS_PATHS ON)
     #endif()
 endif()
 
@@ -248,7 +251,20 @@ _define_relative(PLUGINDIR QTPLUGINDIR ""
 
 _define_non_cache(INCLUDEDIR_KF "${CMAKE_INSTALL_INCLUDEDIR}/KF6")
 
-_define_non_cache(DATADIR_KF "${CMAKE_INSTALL_DATADIR}/kf6")
+if(APPLE AND NOT APPLE_FORCE_UNIX_DIRS)
+    _define_non_cache(DATADIR_KF "/Library/Application Support/kf6")
+else()
+    _define_non_cache(DATADIR_KF "${CMAKE_INSTALL_DATADIR}/kf6")
+endif()
+
+if (APPLE AND NOT APPLE_SUPPRESS_INSTALLDIRS_WARNING)
+  if(APPLE_FORCE_UNIX_DIRS)
+    # for now we print a regular status message for the traditional default
+    message(STATUS "Using standard Unix installation directories under ${CMAKE_INSTALL_PREFIX}/${KDE_INSTALL_DATAROOTDIR}")
+  else()
+    message(WARNING "Using Mac-specific installation directories under ${KDE_INSTALL_DATAROOTDIR}")
+  endif()
+endif()
 
 # TODO TODO TODO
 # Qt-specific data vars
