diff --git a/find-modules/FindWaylandScanner.cmake b/find-modules/FindWaylandScanner.cmake
index 7b493d810ea4c493469610c59c8aa20583e9adcb..850b1f690f2bd8d25503209228f8b416e4de2c06 100644
--- a/find-modules/FindWaylandScanner.cmake
+++ b/find-modules/FindWaylandScanner.cmake
@@ -110,7 +110,7 @@ function(ecm_add_wayland_client_protocol out_var)
         DEPENDS ${_infile} VERBATIM)
 
     add_custom_command(OUTPUT "${_code}"
-        COMMAND ${WaylandScanner_EXECUTABLE} public-code ${_infile} ${_code}
+        COMMAND ${WaylandScanner_EXECUTABLE} code ${_infile} ${_code}
         DEPENDS ${_infile} ${_client_header} VERBATIM)
 
     list(APPEND ${out_var} "${_client_header}" "${_code}")
