diff --git kde-modules/KDECMakeSettings.cmake kde-modules/KDECMakeSettings.cmake
index 2dd54ed509fbf55a50a27320c503b0da8265cea5..3f7f5a82ead1b4cef8ad00657542c3e2c914ee3b 100644
--- kde-modules/KDECMakeSettings.cmake
+++ kde-modules/KDECMakeSettings.cmake
@@ -289,7 +289,7 @@ if(NOT KDE_SKIP_BUILD_SETTINGS)
 
 endif()
 
-if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
     option(ENABLE_CLAZY "Enable Clazy warnings" OFF)
 
     if(ENABLE_CLAZY)
diff --git kde-modules/KDECompilerSettings.cmake kde-modules/KDECompilerSettings.cmake
index 44358fb4d38bf635f9a20f9c5af88e7eb2003010..c713cc2f73d6fc07fb6cef11ac712dee371fb695 100644
--- kde-modules/KDECompilerSettings.cmake
+++ kde-modules/KDECompilerSettings.cmake
@@ -341,7 +341,7 @@ endfunction()
 ############################################################
 
 if ((CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND NOT APPLE) OR
-        (CMAKE_CXX_COMPILER_ID STREQUAL "Clang" AND NOT APPLE) OR
+        (CMAKE_CXX_COMPILER_ID MATCHES "Clang" AND NOT APPLE) OR
         (CMAKE_CXX_COMPILER_ID STREQUAL "Intel" AND NOT WIN32))
     # Linker warnings should be treated as errors
     set(CMAKE_SHARED_LINKER_FLAGS "-Wl,--fatal-warnings ${CMAKE_SHARED_LINKER_FLAGS}")
@@ -488,7 +488,7 @@ endif()
 
 if (CMAKE_GENERATOR STREQUAL "Ninja" AND
     ((CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS 4.9) OR
-     (CMAKE_CXX_COMPILER_ID STREQUAL "Clang" AND NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS 3.5)))
+     (CMAKE_CXX_COMPILER_ID MATCHES "Clang" AND NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS 3.5)))
     # Force colored warnings in Ninja's output, if the compiler has -fdiagnostics-color support.
     # Rationale in https://github.com/ninja-build/ninja/issues/814
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fdiagnostics-color=always")
diff --git kde-modules/KDEFrameworkCompilerSettings.cmake kde-modules/KDEFrameworkCompilerSettings.cmake
index 1d3e1f7c1b2819ed82b9096d736f8e1914751d90..b191e550d02f76cd06a96883f27431c000fe0607 100644
--- kde-modules/KDEFrameworkCompilerSettings.cmake
+++ kde-modules/KDEFrameworkCompilerSettings.cmake
@@ -71,7 +71,7 @@ if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
    endif()
 endif()
 
-if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+if (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
    if (NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS "5.0.0")
       set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wzero-as-null-pointer-constant" )
    endif()
diff --git modules/ECMEnableSanitizers.cmake modules/ECMEnableSanitizers.cmake
index 9a77c6f1fb4b15edf8dbaa6dd1f4f50e15b91628..10b058c92ee359b7746944c1af22f1a9481519f4 100644
--- modules/ECMEnableSanitizers.cmake
+++ modules/ECMEnableSanitizers.cmake
@@ -165,7 +165,7 @@ if (ECM_ENABLE_SANITIZERS)
             if(CMAKE_CXX_COMPILER_ID MATCHES "GNU")
               link_libraries(${XSAN_LINKER_FLAGS})
             endif()
-            if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+            if (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
                 string(REPLACE "-Wl,--no-undefined" "" CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS}")
                 string(REPLACE "-Wl,--no-undefined" "" CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS}")
             endif ()
