diff --git a/src/plugins/CMakeLists.txt b/src/plugins/CMakeLists.txt
index f3f7917..9f6de10 100644
--- a/src/plugins/CMakeLists.txt
+++ b/src/plugins/CMakeLists.txt
@@ -23,7 +23,7 @@ include(QMLModules)
 find_qmlmodule(Ubuntu.OnlineAccounts 0.1)
 find_qmlmodule(org.kde.kdeconnect 1.0)

-find_package(KAccounts REQUIRED)
+# find_package(KAccounts REQUIRED)
 find_package(KF5KIO ${KF5_DEP_VERSION} REQUIRED)

 set_package_properties(KAccounts PROPERTIES
