diff --git a/src/plugins/CMakeLists.txt b/src/plugins/CMakeLists.txt
index 9311290..3deacfd 100644
--- a/src/plugins/CMakeLists.txt
+++ b/src/plugins/CMakeLists.txt
@@ -19,7 +19,7 @@ function(kpackage_install_package dir component root)
 endfunction()
 
 set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake" ${CMAKE_MODULE_PATH})
-find_package(KAccounts REQUIRED)
+# find_package(KAccounts REQUIRED)
 find_package(QMLModule 0.1 COMPONENTS Ubuntu.OnlineAccounts)
 find_package(QMLModule 1.0 COMPONENTS org.kde.kdeconnect)
 find_package(KF5KIO ${KF5_DEP_VERSION} REQUIRED)
