diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6d0510e..ae94b71 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -19,6 +19,7 @@ include(ECMSetupVersion)
 include(ECMAddTests)
 include(ECMInstallIcons)
 include(ECMGenerateHeaders)
+include(ECMMarkNonGuiExecutable)
 
 set(KF5_VERSION "5.7.0") # handled by release scripts
 set(KF5_DEP_VERSION "5.7.0") # handled by release scripts
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index bb7f689..ff4cff4 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -48,6 +48,7 @@ install(FILES
 )
 
 add_executable(purposeprocess externalprocess/purposeprocess_main.cpp helper.cpp)
+ecm_mark_nongui_executable(purposeprocess)
 target_link_libraries(purposeprocess KF5::Purpose Qt5::Network Qt5::Widgets)
 install(TARGETS purposeprocess DESTINATION ${KDE_INSTALL_LIBEXECDIR_KF5})
 
