diff --git a/src/plugins/cmake/FindQMLModule.cmake b/src/plugins/cmake/FindQMLModule.cmake
index 1f1d5c8..5c27bbd 100644
--- a/src/plugins/cmake/FindQMLModule.cmake
+++ b/src/plugins/cmake/FindQMLModule.cmake
@@ -50,7 +50,7 @@ string (REPLACE ";" ", " THE_COMPONENTS "${QMLModule_FIND_COMPONENTS}")
 
 set(QMLModule_VERSION "${QMLModule_FIND_VERSION}")
 foreach(_module ${QMLModule_FIND_COMPONENTS})
-    execute_process(COMMAND qmlplugindump "${_module}" "${QMLModule_FIND_VERSION}" OUTPUT_VARIABLE FUUUUU RESULT_VARIABLE ExitCode)
+    execute_process(COMMAND @QTBINDIR@/qmlplugindump "${_module}" "${QMLModule_FIND_VERSION}" OUTPUT_VARIABLE FUUUUU RESULT_VARIABLE ExitCode)
 
     if(ExitCode EQUAL 0)
         set(QMLModule_${_module}_FOUND TRUE)
