diff --git a/src/quick/RunningJob.qml b/src/quick/RunningJob.qml
index c5c040f..382be27 100644
--- a/src/quick/RunningJob.qml
+++ b/src/quick/RunningJob.qml
@@ -43,7 +43,7 @@ ColumnLayout {
     }
     ProgressBar {
         //FIXME: this is not really working yet, as QML doesn't understand ulong
-        value: root.job.percent
+        value: root.job.percent * 1.0
         maximumValue: 100
         Layout.fillWidth: true
     }
