--- orig.CMakeLists.txt	2015-10-17 13:56:29.000000000 +0200
+++ CMakeLists.txt	2015-12-19 22:22:58.000000000 +0100
@@ -11,6 +11,7 @@
 include(KDEInstallDirs)
 include(KDECMakeSettings)
 include(KDECompilerSettings)
+include(ECMAddAppIcon)
 include(ECMInstallIcons)
 include(GenerateExportHeader)
 
@@ -55,7 +56,8 @@
 kompareurldialog.cpp
 )
 
-# kde4_add_app_icon(kompare_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/pics/hi*-app-kompare.png")
+file(GLOB ICONS_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/pics/*-apps-kompare.png")
+ecm_add_app_icon(kompare_SRCS ICONS ${ICONS_SRCS})
 
 add_executable(kompare ${kompare_SRCS})
 
