diff --git a/migration/kmail/CMakeLists.txt b/migration/kmail/CMakeLists.txt
index 001a80a..c59c5f4 100644
--- a/migration/kmail/CMakeLists.txt
+++ b/migration/kmail/CMakeLists.txt
@@ -41,7 +41,7 @@ add_resource_iface( maildir/maildirresource.kcfg
 add_resource_iface( mixedmaildir/mixedmaildirresource.kcfg
                     org.kde.Akonadi.MixedMaildir.Settings MixedMaildirSettings )
 
-kde4_add_app_icon(kmail-migrator_srcs "${KDE4_ICON_DIR}/oxygen/*/apps/akonadi.png")
+kde4_add_app_icon(kmail-migrator_srcs "${KDE4_ICON_INSTALL_DIR}/oxygen/*/apps/akonadi.png")
 kde4_add_executable(kmail-migrator ${kmail-migrator_srcs} )
 target_link_libraries(kmail-migrator
   ${KDEPIMLIBS_KMBOX_LIBS}
diff --git a/migration/kres/CMakeLists.txt b/migration/kres/CMakeLists.txt
index 7ba117e..4175e19 100644
--- a/migration/kres/CMakeLists.txt
+++ b/migration/kres/CMakeLists.txt
@@ -20,7 +20,7 @@ add_resource_iface( birthdays/birthdaysresource.kcfg org.kde.Akonadi.Birthdays.S
 add_resource_iface( maildir/maildirresource.kcfg org.kde.Akonadi.Maildir.Settings MaildirSettings )
 add_resource_iface( dav/resource/davgroupwareresource.kcfg org.kde.Akonadi.davGroupware.Settings DavSettings )
 
-kde4_add_app_icon(kres-migrator_srcs "${KDE4_ICON_DIR}/oxygen/*/apps/akonadi.png")
+kde4_add_app_icon(kres-migrator_srcs "${KDE4_ICON_INSTALL_DIR}/oxygen/*/apps/akonadi.png")
 kde4_add_executable(kres-migrator ${kres-migrator_srcs})
 
 target_link_libraries(kres-migrator
