diff --git src/lib/corelib/jsextensions/utilitiesextension.cpp src/lib/corelib/jsextensions/utilitiesextension.cpp
index a10251c546f1850d958906d0c4d97e4c9b77bcae..b636e8862defe2a375686a2158e7c5f707ac5d3d 100644
--- src/lib/corelib/jsextensions/utilitiesextension.cpp
+++ src/lib/corelib/jsextensions/utilitiesextension.cpp
@@ -332,7 +332,7 @@ QScriptValue UtilitiesExtension::js_getNativeSetting(QScriptContext *context, QS
 
     QVariant defaultValue = context->argumentCount() > 2 ? context->argument(2).toVariant() : QVariant();
 
-    QSettings settings(context->argument(0).toString(), QSettings::NativeFormat);
+    QSettings settings(context->argument(0).toString(), QSettings::IniFormat);
     QVariant value = settings.value(key, defaultValue);
     return value.isNull() ? engine->undefinedValue() : engine->toScriptValue(value);
 }
@@ -351,7 +351,7 @@ QScriptValue UtilitiesExtension::js_nativeSettingGroups(QScriptContext *context,
                                    QLatin1String("nativeSettingGroups expects 1 argument"));
     }
 
-    QSettings settings(context->argument(0).toString(), QSettings::NativeFormat);
+    QSettings settings(context->argument(0).toString(), QSettings::IniFormat);
     return engine->toScriptValue(settings.childGroups());
 }
 
diff --git src/lib/corelib/tools/settingscreator.cpp src/lib/corelib/tools/settingscreator.cpp
index 51c0207788bcda7782998e560ed9587cec13f453..df88ea33395c4bc2d985bc6a8fd0252012c45959 100644
--- src/lib/corelib/tools/settingscreator.cpp
+++ src/lib/corelib/tools/settingscreator.cpp
@@ -52,7 +52,8 @@ namespace Internal {
 
 static QSettings::Format format()
 {
-    return HostOsInfo::isWindowsHost() ? QSettings::IniFormat : QSettings::NativeFormat;
+//     return HostOsInfo::isWindowsHost() ? QSettings::IniFormat : QSettings::NativeFormat;
+    return QSettings::IniFormat;
 }
 
 
diff --git tests/auto/tools/tst_tools.cpp tests/auto/tools/tst_tools.cpp
index 0073f9e6830f1fd22f090485f6691f68de7127fb..e65555fc8a6d362bc6b60c52f4827ef8b3b05f71 100644
--- tests/auto/tools/tst_tools.cpp
+++ tests/auto/tools/tst_tools.cpp
@@ -348,7 +348,7 @@ QString TestTools::setupSettingsDir1()
     for (const Version &v : versions) {
         const QString settingsDir = baseDir->path() + "/qbs/" + v.toString();
         QSettings s(settingsDir + "/qbs.conf",
-                    HostOsInfo::isWindowsHost() ? QSettings::IniFormat : QSettings::NativeFormat);
+                    /*HostOsInfo::isWindowsHost() ? QSettings::IniFormat : QSettings::NativeFormat*/ QSettings::IniFormat);
         const QString profilesDir = settingsDir + "/profiles";
         QDir::root().mkpath(profilesDir);
         const QString magicString = v == predecessor ? "right" : "wrong";
@@ -366,7 +366,7 @@ QString TestTools::setupSettingsDir2()
     m_tmpDirs.push_back(baseDir);
     const QString settingsDir = baseDir->path();
     QSettings s(settingsDir + QLatin1String("/qbs.conf"),
-                HostOsInfo::isWindowsHost() ? QSettings::IniFormat : QSettings::NativeFormat);
+                /*HostOsInfo::isWindowsHost() ? QSettings::IniFormat : QSettings::NativeFormat*/ QSettings::IniFormat);
     const QString profilesDir = settingsDir + QLatin1String("/qbs/profiles");
     QDir::root().mkpath(profilesDir);
     QFile f(profilesDir + "/right.txt");
