diff --git src/lib/qtprofilesetup/qtprofilesetup.cpp src/lib/qtprofilesetup/qtprofilesetup.cpp
index adeeef7bcae6f33b6ece69b62e67fb981c72b354..1f69ea50490aaacbd168257c4ae3f1b4104dc4c5 100644
--- src/lib/qtprofilesetup/qtprofilesetup.cpp
+++ src/lib/qtprofilesetup/qtprofilesetup.cpp
@@ -68,7 +68,7 @@
 #include <mach/machine.h>
 #include <mach-o/fat.h>
 #include <mach-o/loader.h>
-#if __MAC_OS_X_VERSION_MAX_ALLOWED < 101200
+#ifndef FAT_MAGIC_64
 #define FAT_MAGIC_64 0xcafebabf
 #define FAT_CIGAM_64 0xbfbafeca
 struct fat_arch_64 {
