diff --git a/cmake/modules/FindPOPPLER.cmake b/cmake/modules/FindPOPPLER.cmake
index 8dc7cbd..fa9e562 100644
--- a/cmake/modules/FindPOPPLER.cmake
+++ b/cmake/modules/FindPOPPLER.cmake
@@ -5,9 +5,9 @@ if (${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION} LESS 2.5)
 	set(PKG_POPPLER_LIBRARIES ${_libPopplerLinkDir})
 else (${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION} LESS 2.5)
 	include(FindPkgConfig)
-	pkg_search_module(POPPLER libpoppler>=0.58.0 poppler>=0.58.0)
+	pkg_search_module(POPPLER libpoppler>=0.58.0 poppler>=0.58.0 libpoppler<=0.73.0 poppler<=0.73.0)
 	if (POPPLER_FOUND)
-		pkg_search_module(POPPLER_CPP REQUIRED libpoppler-cpp>=0.58.0 poppler-cpp>=0.58.0)
+		pkg_search_module(POPPLER_CPP REQUIRED libpoppler-cpp>=0.58.0 poppler-cpp>=0.58.0 libpoppler-cpp<=0.73.0 poppler-cpp<=0.73.0)
 	endif(POPPLER_FOUND)
 endif (${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION} LESS 2.5)
  
diff --git a/scribus/plugins/import/pdf/importpdf.cpp b/scribus/plugins/import/pdf/importpdf.cpp
index 79630dc..ec6bcbb 100644
--- a/scribus/plugins/import/pdf/importpdf.cpp
+++ b/scribus/plugins/import/pdf/importpdf.cpp
@@ -18,15 +18,15 @@ for which a new license (GPL+exception) is in place.
 #include <QRegExp>
 #include <QStack>
 
-#include <poppler/ErrorCodes.h>
-#include <poppler/GlobalParams.h>
-#include <poppler/OptionalContent.h>
-#include <poppler/PageTransition.h>
-#include <poppler/ViewerPreferences.h>
-#include <poppler/poppler-config.h>
-#include <poppler/cpp/poppler-version.h>
-#include <poppler/SplashOutputDev.h>
-#include <poppler/splash/SplashBitmap.h>
+#include <ErrorCodes.h>
+#include <GlobalParams.h>
+#include <OptionalContent.h>
+#include <PageTransition.h>
+#include <ViewerPreferences.h>
+#include <poppler-config.h>
+#include <cpp/poppler-version.h>
+#include <SplashOutputDev.h>
+#include <splash/SplashBitmap.h>
 
 #include "importpdf.h"
 #include "importpdfconfig.h"
diff --git a/scribus/plugins/import/pdf/importpdfconfig.h b/scribus/plugins/import/pdf/importpdfconfig.h
index 80ac796..61ac53d 100644
--- a/scribus/plugins/import/pdf/importpdfconfig.h
+++ b/scribus/plugins/import/pdf/importpdfconfig.h
@@ -7,7 +7,7 @@ for which a new license (GPL+exception) is in place.
 #ifndef IMPORTPDFCONFIG_H
 #define IMPORTPDFCONFIG_H
 
-#include <poppler/cpp/poppler-version.h>
+#include <cpp/poppler-version.h>
 
 #define POPPLER_VERSION_ENCODE(major, minor, micro) (	\
 	  ((major) * 10000)				\
diff --git a/scribus/plugins/import/pdf/slaoutput.cpp b/scribus/plugins/import/pdf/slaoutput.cpp
index cfc438b..43b7e2c 100644
--- a/scribus/plugins/import/pdf/slaoutput.cpp
+++ b/scribus/plugins/import/pdf/slaoutput.cpp
@@ -7,10 +7,10 @@ for which a new license (GPL+exception) is in place.
 
 #include "slaoutput.h"
 
-#include <poppler/GlobalParams.h>
-#include <poppler/poppler-config.h>
-#include <poppler/FileSpec.h>
-#include <poppler/fofi/FoFiTrueType.h>
+#include <GlobalParams.h>
+#include <poppler-config.h>
+#include <FileSpec.h>
+#include <fofi/FoFiTrueType.h>
 #include <QApplication>
 #include <QFile>
 #include "commonstrings.h"
diff --git a/scribus/plugins/import/pdf/slaoutput.h b/scribus/plugins/import/pdf/slaoutput.h
index 4dac7c3..a8e5dcc 100644
--- a/scribus/plugins/import/pdf/slaoutput.h
+++ b/scribus/plugins/import/pdf/slaoutput.h
@@ -29,29 +29,29 @@ for which a new license (GPL+exception) is in place.
 #include "vgradient.h"
 
 #if POPPLER_ENCODED_VERSION < POPPLER_VERSION_ENCODE(0, 73, 0)
-#include <poppler/goo/gtypes.h>
+#include <goo/gtypes.h>
 #endif
-#include <poppler/Object.h>
-#include <poppler/OutputDev.h>
-#include <poppler/Gfx.h>
-#include <poppler/GfxState.h>
-#include <poppler/Stream.h>
-#include <poppler/GfxFont.h>
-#include <poppler/Link.h>
-#include <poppler/PDFDoc.h>
-#include <poppler/Error.h>
-#include <poppler/Form.h>
-#include <poppler/Page.h>
-#include <poppler/Catalog.h>
-#include <poppler/CharCodeToUnicode.h>
-#include <poppler/FontEncodingTables.h>
-#include <poppler/splash/SplashFontFileID.h>
-#include <poppler/splash/SplashFontFile.h>
-#include <poppler/splash/SplashFontEngine.h>
-#include <poppler/splash/SplashFont.h>
-#include <poppler/splash/SplashMath.h>
-#include <poppler/splash/SplashPath.h>
-#include <poppler/splash/SplashGlyphBitmap.h>
+#include <Object.h>
+#include <OutputDev.h>
+#include <Gfx.h>
+#include <GfxState.h>
+#include <Stream.h>
+#include <GfxFont.h>
+#include <Link.h>
+#include <PDFDoc.h>
+#include <Error.h>
+#include <Form.h>
+#include <Page.h>
+#include <Catalog.h>
+#include <CharCodeToUnicode.h>
+#include <FontEncodingTables.h>
+#include <splash/SplashFontFileID.h>
+#include <splash/SplashFontFile.h>
+#include <splash/SplashFontEngine.h>
+#include <splash/SplashFont.h>
+#include <splash/SplashMath.h>
+#include <splash/SplashPath.h>
+#include <splash/SplashGlyphBitmap.h>
 
 //------------------------------------------------------------------------
 // LinkSubmitData
