--- qt-everywhere-opensource-src-5.9.3/qtwebengine/examples/webenginewidgets/webenginewidgets.pro	2017-12-21 16:55:05.000000000 +0100
+++ qt-everywhere-opensource-src-5.9.3/qtwebengine/examples/webenginewidgets/orig.webenginewidgets.pro	2017-11-12 17:42:34.000000000 +0100
@@ -5,6 +5,7 @@
     contentmanipulation \
     cookiebrowser \
     demobrowser \
+    html2pdf \
     markdowneditor \
     simplebrowser \
     videoplayer
--- qt-everywhere-opensource-src-5.9.3/qtwebengine/examples/webenginewidgets/simplebrowser/orig.downloadwidget.cpp	2017-11-12 17:42:34.000000000 +0100
+++ qt-everywhere-opensource-src-5.9.3/qtwebengine/examples/webenginewidgets/simplebrowser/downloadwidget.cpp	2017-12-21 17:00:53.000000000 +0100
@@ -141,8 +141,7 @@
         m_progressBar->setValue(0);
         m_progressBar->setDisabled(true);
         m_progressBar->setFormat(
-            tr("interrupted: %1")
-            .arg(m_download->interruptReasonString()));
+            tr("interrupted"));
         break;
     }
 
