--- llvm_release_39/projects/libcxx/lib/buildit	2017-06-05 22:31:24.000000000 +0200
+++ llvm_release_39/projects/libcxx/lib/buildit	2017-06-05 22:31:32.000000000 +0200
@@ -40,7 +40,7 @@
   RC_ProjectSourceVersion=1
 fi
 
-EXTRA_FLAGS="-nostdinc++ -std=${CXX_LANG} -fstrict-aliasing -Wall -Wextra -Wshadow -Wconversion \
+EXTRA_FLAGS="${CXXFLAGS} -nostdinc++ -std=${CXX_LANG} -fstrict-aliasing -Wall -Wextra -Wshadow -Wconversion \
              -Wstrict-aliasing=2 -Wstrict-overflow=4 -D_LIBCPP_BUILDING_LIBRARY "
 
 case $TRIPLE in
