--- llvm-8.0.0.src/orig.CMakeLists.txt	2019-02-08 12:06:27.000000000 +0100
+++ llvm-8.0.0.src/CMakeLists.txt	2019-07-04 14:58:10.000000000 +0200
@@ -29,6 +29,8 @@
     "${LLVM_VERSION_MAJOR}.${LLVM_VERSION_MINOR}.${LLVM_VERSION_PATCH}${LLVM_VERSION_SUFFIX}")
 endif()
 
+include(FeatureSummary)
+
 if ((CMAKE_GENERATOR MATCHES "Visual Studio") AND (CMAKE_GENERATOR_TOOLSET STREQUAL ""))
   message(WARNING "Visual Studio generators use the x86 host compiler by "
                   "default, even for 64-bit targets. This can result in linker "
@@ -248,7 +250,7 @@
 
 if (CMAKE_BUILD_TYPE AND
     NOT uppercase_CMAKE_BUILD_TYPE MATCHES "^(DEBUG|RELEASE|RELWITHDEBINFO|MINSIZEREL)$")
-  message(FATAL_ERROR "Invalid value for CMAKE_BUILD_TYPE: ${CMAKE_BUILD_TYPE}")
+  message(STATUS "Custom value for CMAKE_BUILD_TYPE: ${CMAKE_BUILD_TYPE}")
 endif()
 
 set(LLVM_LIBDIR_SUFFIX "" CACHE STRING "Define suffix of library directory name (32/64)" )
@@ -1080,3 +1082,12 @@
   add_subdirectory(utils/benchmark)
   add_subdirectory(benchmarks)
 endif()
+
+feature_summary(WHAT ALL   FATAL_ON_MISSING_REQUIRED_PACKAGES)
+if(CMAKE_OSX_DEPLOYMENT_TARGET)
+    message(STATUS "OS X Deployment target: ${CMAKE_OSX_DEPLOYMENT_TARGET}")
+endif()
+if(CMAKE_OSX_SYSROOT)
+    message(STATUS "OS X SDK sysroot: ${CMAKE_OSX_SYSROOT}")
+endif()
+
