diff --git b/projects/libcxx/utils/libcxx/test/orig.target_info.py a/projects/libcxx/utils/libcxx/test/target_info.py
index 6da30e6..089d795 100644
--- b/projects/libcxx/utils/libcxx/test/orig.target_info.py
+++ a/projects/libcxx/utils/libcxx/test/target_info.py
@@ -133,6 +133,9 @@ class DarwinLocalTI(DefaultTargetInfo):
             self.full_config.lit_config.warning("Could not determine macOS SDK path! stderr was " + err)
         if exit_code == 0 and out:
             sdk_path = out.strip()
+            if sdk_path == '':
+                sdk_path = os.environ['SDKPATH']
+                self.full_config.lit_config.note('using SDKPATH: %r' % sdk_path)
             self.full_config.lit_config.note('using SDKROOT: %r' % sdk_path)
             assert isinstance(sdk_path, str)
             flags += ["-isysroot", sdk_path]
