diff --git a/libcxx/lib/libc++abi.v1.exp b/libcxx/lib/libc++abi.v1.exp
index c245b17..879b4dd 100644
--- a/libcxx/lib/libc++abi.v1.exp
+++ b/libcxx/lib/libc++abi.v1.exp
@@ -1,11 +1,20 @@
+___cxa_allocate_exception
+___cxa_end_catch
 ___cxa_demangle
+___cxa_current_exception_type
+___cxa_call_unexpected
+___cxa_free_exception
+___cxa_get_exception_ptr
 ___cxa_get_globals
 ___cxa_get_globals_fast
 ___cxa_guard_abort
 ___cxa_guard_acquire
 ___cxa_guard_release
+___cxa_rethrow
 ___cxa_pure_virtual
 ___cxa_deleted_virtual
+___cxa_begin_catch
+___cxa_throw
 ___cxa_vec_cctor
 ___cxa_vec_cleanup
 ___cxa_vec_ctor
@@ -17,6 +26,7 @@ ___cxa_vec_new
 ___cxa_vec_new2
 ___cxa_vec_new3
 ___dynamic_cast
+___gxx_personality_v0
 __ZTIDi
 __ZTIDn
 __ZTIDs
diff --git a/libcxx/lib/libc++abi.v2.exp b/libcxx/lib/libc++abi.v2.exp
index 6a3e6b9..dfc308f 100644
--- a/libcxx/lib/libc++abi.v2.exp
+++ b/libcxx/lib/libc++abi.v2.exp
@@ -1,11 +1,20 @@
+___cxa_allocate_exception
+___cxa_end_catch
 ___cxa_demangle
+___cxa_current_exception_type
+___cxa_call_unexpected
+___cxa_free_exception
+___cxa_get_exception_ptr
 ___cxa_get_globals
 ___cxa_get_globals_fast
 ___cxa_guard_abort
 ___cxa_guard_acquire
 ___cxa_guard_release
+___cxa_rethrow
 ___cxa_pure_virtual
 ___cxa_deleted_virtual
+___cxa_begin_catch
+___cxa_throw
 ___cxa_throw_bad_array_new_length
 ___cxa_uncaught_exceptions
 ___cxa_vec_cctor
@@ -19,6 +28,7 @@ ___cxa_vec_new
 ___cxa_vec_new2
 ___cxa_vec_new3
 ___dynamic_cast
+___gxx_personality_v0
 __ZTIDi
 __ZTIDn
 __ZTIDs
