diff --git a/libcxx/src/CMakeLists.txt b/libcxx/src/CMakeLists.txt
index 496429f..d599d37 100644
--- a/libcxx/src/CMakeLists.txt
+++ b/libcxx/src/CMakeLists.txt
@@ -337,6 +337,7 @@ if (LIBCXX_ENABLE_EXPERIMENTAL_LIBRARY)
     experimental/memory_resource.cpp
     )
   add_library(cxx_experimental STATIC ${LIBCXX_EXPERIMENTAL_SOURCES})
+  target_compile_options(cxx_experimental PRIVATE "-fno-lto")
   if (LIBCXX_ENABLE_SHARED)
     target_link_libraries(cxx_experimental PRIVATE cxx_shared)
   else()
