diff --git CMakeLists.txt CMakeLists.txt
index c8fb13d..871ecce 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -378,25 +378,26 @@ if(GRAPHVIZ_FOUND)
 endif()
 
 #VTK discovery works a lot better if you give CMake a hint using the VTK_DIR variable
-find_path(VTK_DIR VTKConfig.cmake
-  /usr/lib64/vtk /usr/lib/vtk /usr/local/lib64/vtk /usr/local/lib/vtk
-)
-find_package(VTK)
-set_package_properties(VTK PROPERTIES
-  TYPE OPTIONAL
-  DESCRIPTION "Visualization Toolkit"
-  PURPOSE "Needed for the object visualizer plugin."
-  URL "http://www.vtk.org"
-)
-if(VTK_FOUND)
-  find_path(VTK_QT_INCLUDE_DIR NAMES QVTKWidget.h HINTS ${VTK_INCLUDE_DIRS})
-  if(NOT VTK_QT_INCLUDE_DIR)
-    message(STATUS "Cannot locate QVTKWidget.h in ${VTK_INCLUDE_DIRS}")
-    set(VTK_FOUND FALSE)
-    add_feature_info("QVTKWidget header" VTK_FOUND "Object visualizer plugin requires QVTKWidget.h header.")
-  endif()
-endif()
-set(HAVE_VTK ${VTK_FOUND})
+#find_path(VTK_DIR VTKConfig.cmake
+#  /usr/lib64/vtk /usr/lib/vtk /usr/local/lib64/vtk /usr/local/lib/vtk
+#)
+#find_package(VTK)
+#set_package_properties(VTK PROPERTIES
+#  TYPE OPTIONAL
+#  DESCRIPTION "Visualization Toolkit"
+#  PURPOSE "Needed for the object visualizer plugin."
+#  URL "http://www.vtk.org"
+#)
+#if(VTK_FOUND)
+#  find_path(VTK_QT_INCLUDE_DIR NAMES QVTKWidget.h HINTS ${VTK_INCLUDE_DIRS})
+#  if(NOT VTK_QT_INCLUDE_DIR)
+#    message(STATUS "Cannot locate QVTKWidget.h in ${VTK_INCLUDE_DIRS}")
+#    set(VTK_FOUND FALSE)
+#    add_feature_info("QVTKWidget header" VTK_FOUND "Object visualizer plugin requires QVTKWidget.h header.")
+#  endif()
+#endif()
+#set(HAVE_VTK ${VTK_FOUND})
+set(HAVE_VTK FALSE)
 
 # ELF header for ABI detection
 find_file(ELF_H elf.h)
