--- src/orig.xxdiff.pro	2013-08-08 04:37:03.000000000 +0200
+++ src/xxdiff.pro	2014-08-06 15:39:41.000000000 +0200
@@ -40,9 +40,9 @@
 
 
 # To enable KDE support, uncomment the following lines:
-#QMAKE_CFLAGS += -DXX_KDE -I/usr/include/kde
-#QMAKE_CXXFLAGS += -DXX_KDE -I/usr/include/kde
-#QMAKE_LIBS += -lkdeui -lkio -lqt-mt
+QMAKE_CFLAGS += -DXX_KDE
+QMAKE_CXXFLAGS += -DXX_KDE
+QMAKE_LIBS += -lkdeui -lkio
 
 
 # Enable debugging modes within xxdiff.
