diff -r b193168bad98 src/Makefile.bootstrap
--- src/Makefile.bootstrap	Wed Apr 15 14:02:15 2015 -0400
+++ src/Makefile.bootstrap	Tue Jan 26 10:52:37 2016 +0100
@@ -6,6 +6,8 @@
 
 # Rule to generate the qmake Makefile for building xxdiff.
 QMAKE ?= /usr/bin/qmake-qt4
+MAKEDIR ?= .
+QMAKEOPTS =
 
 all: Makefile
 
@@ -15,9 +17,10 @@
     export QMAKESPEC=macx-g++
 endif
 
-Makefile.qmake: xxdiff.pro
-	$(QMAKE) -o Makefile.qmake $<
+Makefile.qmake: $(MAKEDIR)/xxdiff.pro
+	$(QMAKE) $(QMAKEOPTS) -o Makefile.qmake $<
 
-Makefile: Makefile.qmake Makefile.extra
-	cat $^ > Makefile
+Makefile: Makefile.qmake $(MAKEDIR)/Makefile.extra
+	echo "MAKEDIR = $(MAKEDIR)" > Makefile
+	cat $^ >> Makefile
 
diff -r b193168bad98 src/Makefile.extra
--- src/Makefile.extra	Wed Apr 15 14:02:15 2015 -0400
+++ src/Makefile.extra	Tue Jan 26 10:52:37 2016 +0100
@@ -4,7 +4,7 @@
 #
 
 # Extra targets.
-EXTRA_FILES = version.h doc.h
+EXTRA_FILES = version.h $(MAKEDIR)/doc.h
 
 .SUFFIXES: .html .txt .h
 
@@ -23,8 +23,8 @@
 #
 
 # Automatically generate a simple include file with version number in it.
-version.h: ../VERSION
-	echo "#define XX_VERSION \"`cat ../VERSION`\"" > $@
+version.h: $(MAKEDIR)/../VERSION
+	echo "#define XX_VERSION \"`cat $(MAKEDIR)/../VERSION`\"" > $@
 
 # Dependencies for the generated version file.
 proginfo.o: version.h
@@ -41,13 +41,13 @@
 	sed -e 's/\"/\\\"/g;s/$$/\\n\\/;1s/^/char text[]=\"/;$$s/\\$$/\"\;/' $< > $@
 
 # Dependencies for the generated documentation file.
-help.o: doc.h
-help.obj: doc.h
+help.o: $(MAKEDIR)/doc.h
+help.obj: $(MAKEDIR)/doc.h
 
 # Convert the reStructuredText documentation to html.  (this is only used by the
 # author, directly, manually, not that important for xxdiff packagers.)
-.txt.html:
-	rst2html.py --output-encoding=iso-8859-1 $< $@
+#.txt.html:
+#	rst2html.py --output-encoding=iso-8859-1 $< $@
 
 #
 # clean_extra
@@ -60,13 +60,10 @@
 	rm -f $(EXTRA_FILES)
 
 
-# Override the qmake we use because the one in the path might be different.
-QMAKE = $(QTDIR)/bin/qmake
-
 
 # Note: we would need to add this to be correct.  Danger!  If the Makefile gets
 # remade, this file does not get appended again to the new makefile!
 all: Makefile
 
-Makefile: xxdiff.pro
-	$(MAKE) -f Makefile.bootstrap Makefile
+Makefile: $(MAKEDIR)/xxdiff.pro
+	$(MAKE) -f $(MAKEDIR)/Makefile.bootstrap Makefile
