diff -r b193168bad98 src/xxdiff.pro
--- src/xxdiff.pro	Wed Apr 15 14:02:15 2015 -0400
+++ src/xxdiff.pro	Thu May 14 20:18:26 2015 +0200
@@ -30,7 +30,7 @@
 CONFIG -= debug
 CONFIG += qt warn_on thread
 
-DESTDIR=../bin
+DESTDIR=./bin
 TARGET = xxdiff
 
 
@@ -140,7 +140,8 @@
 
    QMAKE_EXTRA_TARGETS += macdeployqt dmg deploy
    QMAKE_CXXFLAGS -= -O2
-   QMAKE_CXXFLAGS += -mdynamic-no-pic -O3 -ftracer -msse2 -msse3 -mssse3 -ftree-vectorize
+   QMAKE_CXXFLAGS += -mdynamic-no-pic -O3 -g
+   INCLUDEPATH += -I@@QT_INCLUDES_DIR@@
 }
 
 #----------------------------------------
