diff -r 815d97a7a0a1 src/Makefile.extra
--- src/Makefile.extra	Sat Nov 12 16:44:13 2016 -0500
+++ src/Makefile.extra	Sun Nov 13 22:02:04 2016 +0100
@@ -47,7 +47,7 @@
 # Convert the reStructuredText documentation to html.  (this is only used by the
 # author, directly, manually, not that important for xxdiff packagers.)
 .txt.html:
-	rst2html.py --output-encoding=iso-8859-1 $< $@
+	rst2html --output-encoding=iso-8859-1 $< $@
 
 #
 # clean_extra
