diff -r 815d97a7a0a1 src/markersWidgetBase.ui
--- src/markersWidgetBase.ui	Sat Nov 12 16:44:13 2016 -0500
+++ src/markersWidgetBase.ui	Tue Jan 02 15:19:46 2018 +0100
@@ -27,7 +27,6 @@
         <widget class="QLabel" name="TextLabel1_2">
           <property name="font">
             <font>
-              <family>helvetica</family>
               <pointsize>14</pointsize>
             </font>
           </property>
