diff --git examples/android/CMakeLists.txt examples/android/CMakeLists.txt
index 861bba117d85eca81eb4a139e5229e6c82316235..216a9afe1ac0c649da92814e5de81df9cd1b334a 100644
--- examples/android/CMakeLists.txt
+++ examples/android/CMakeLists.txt
@@ -1,9 +1,11 @@
+include(ECMAddAppIcon)
 
 set(kirigami2gallery_SRCS
     main.cpp
     )
 
 qt5_add_resources(RESOURCES resources.qrc)
+ecm_add_app_icon(kirigami2gallery_SRCS ICONS "${CMAKE_CURRENT_SOURCE_DIR}/kirigami.svgz")
 
 add_executable(kirigami2gallery ${kirigami2gallery_SRCS} ${RESOURCES})
 target_link_libraries(kirigami2gallery Qt5::Core  Qt5::Qml Qt5::Quick Qt5::Svg Qt5::QuickControls2)
