--- src/widgets/orig.ktextedit.cpp	2015-12-06 22:39:32.000000000 +0100
+++ src/widgets/ktextedit.cpp	2015-12-20 20:07:00.000000000 +0100
@@ -366,7 +366,7 @@
         dialog.setLanguage(d->spellCheckingLanguage);
     }
     if (!windowIcon.isEmpty()) {
-        dialog.setWindowIcon(QIcon::fromTheme(windowIcon));
+        dialog.setWindowIcon(QIcon::fromTheme(windowIcon, dialog.windowIcon()));
     }
     if (dialog.exec()) {
         setSpellCheckingLanguage(dialog.language());
