--- cmake/FindUDev.cmake.orig	2015-11-08 13:41:10.000000000 +0100
+++ cmake/FindUDev.cmake	2015-11-18 15:19:44.924619651 +0100
@@ -31,8 +31,14 @@
 # OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
 # SUCH DAMAGE.
 
-find_path(UDEV_INCLUDE_DIR libudev.h)
-find_library(UDEV_LIBS udev)
+include(FindPkgConfig)
+pkg_check_modules(PC_UDEV libudev)
+find_path(UDEV_INCLUDE_DIR
+	NAMES libudev.h
+	HINTS ${PC_UDEV_INCLUDEDIR} ${PC_UDEV_INCLUDE_DIRS})
+find_library(UDEV_LIBS
+	NAMES udev libudev
+	HINTS ${PC_UDEV_LIBDIR} ${PC_UDEV_LIBRARY_DIRS})
 
 if(UDEV_INCLUDE_DIR AND UDEV_LIBS)
    include(CheckFunctionExists)
