--- src/plasmaquick/orig.configview.cpp	2015-12-12 10:50:48.000000000 +0100
+++ src/plasmaquick/configview.cpp	2016-02-05 16:26:21.149559503 +0100
@@ -116,7 +116,7 @@
         corona = applet.data()->containment()->corona();
     }
 
-    if (corona->package().isValid()) {
+    if (corona && corona->package().isValid()) {
         PackageUrlInterceptor *interceptor = new PackageUrlInterceptor(q->engine(), corona->package());
         interceptor->addAllowedPath(applet.data()->package().path());
         q->engine()->setUrlInterceptor(interceptor);
@@ -281,7 +281,9 @@
 
 void ConfigView::init()
 {
-    setSource(QUrl::fromLocalFile(d->corona->package().filePath("appletconfigurationui")));
+    if (d && d->corona) {
+        setSource(QUrl::fromLocalFile(d->corona->package().filePath("appletconfigurationui")));
+    }
 }
 
 Plasma::Applet *ConfigView::applet()
