--- src/orig.kwindowsystem.cpp	2015-12-08 17:39:13.000000000 +0100
+++ src/kwindowsystem.cpp	2015-12-08 17:40:03.000000000 +0100
@@ -460,6 +460,7 @@
 
 void KWindowSystem::setMainWindow(QWidget *subWidget, WId mainWindowId)
 {
+#ifndef Q_OS_OSX
     // Set the WA_NativeWindow attribute to force the creation of the QWindow.
     // Without this QWidget::windowHandle() returns 0.
     subWidget->setAttribute(Qt::WA_NativeWindow, true);
@@ -470,6 +471,7 @@
     // mainWindow is not the child of any object, so make sure it gets deleted at some point
     connect(subWidget, &QObject::destroyed, mainWindow, &QObject::deleteLater);
     subWindow->setTransientParent(mainWindow);
+#endif
 }
 
 #ifndef KWINDOWSYSTEM_NO_DEPRECATED
