--- orig.CMakeLists.txt	2015-12-15 18:12:29.000000000 +0100
+++ CMakeLists.txt	2015-12-15 19:10:44.000000000 +0100
@@ -23,7 +23,7 @@
 ecm_setup_version(${KF5_VERSION}
                   VARIABLE_PREFIX KFILEMETADATA
                   PACKAGE_VERSION_FILE "${CMAKE_CURRENT_BINARY_DIR}/KF5FileMetaDataConfigVersion.cmake"
-                  SOVERSION 3)
+                  SOVERSION 5)
 
 # Dependencies
 set(REQUIRED_QT_VERSION 5.3.0)
