diff --git src/codecs/CMakeLists.txt src/codecs/CMakeLists.txt
index 9df7796..0b1fdb5 100644
--- src/codecs/CMakeLists.txt
+++ src/codecs/CMakeLists.txt
@@ -10,6 +10,9 @@ add_library(KF5BalooCodecs STATIC ${BALOO_CODECS_SRCS})
 add_library(KF5::BalooCodecs ALIAS KF5BalooCodecs)
 
 target_link_libraries(KF5BalooCodecs
+PUBLIC
     Qt5::Core
     KF5::CoreAddons
+PRIVATE
+    Qt5::QspXDG
 )
diff --git src/engine/CMakeLists.txt src/engine/CMakeLists.txt
index 96d63f4..2a33562 100644
--- src/engine/CMakeLists.txt
+++ src/engine/CMakeLists.txt
@@ -35,6 +35,7 @@ target_link_libraries(KF5BalooEngine
     ${LMDB_LIBRARIES}
     PRIVATE
     KF5BalooCodecs
+    Qt5::QspXDG
 )
 
 set_target_properties(KF5BalooEngine PROPERTIES
diff --git src/lib/CMakeLists.txt src/lib/CMakeLists.txt
index 87f5b7c..47b3136 100644
--- src/lib/CMakeLists.txt
+++ src/lib/CMakeLists.txt
@@ -43,6 +43,7 @@ target_link_libraries(KF5Baloo
     Qt5::DBus
     KF5::Solid
     KF5::BalooEngine
+    Qt5::QspXDG
 )
 
 set_target_properties(KF5Baloo PROPERTIES
