diff --git xar/src/orig.xar.c xar/src/xar.c
index 6332611..e5393ad 100644
--- xar/src/orig.xar.c
+++ xar/src/xar.c
@@ -52,7 +52,7 @@
 #include "xar_internal.h"
 #include "config.h"
 #include "../lib/filetree.h"
-#include "util.h"
+#include "../lib/util.h"
 #include "xar.h"
 #define SYMBOLIC 1
 #define NUMERIC  2
diff --git xar/lib/util.c.orig xar/lib/util.c
index 0ea661a..134ced4 100644
--- xar/lib/util.c.orig
+++ xar/lib/util.c
@@ -50,6 +50,9 @@
 #ifndef HAVE_ASPRINTF
 #include "asprintf.h"
 #endif
+#ifdef __linux__
+#include <sys/vfs.h>
+#endif
 #include "xar.h"
 #include "archive.h"
 #include "filetree.h"
@@ -588,6 +591,7 @@ size_t xar_optimal_io_size_at_path(const char *path)
 	// Start at 1MiB
 	size_t optimal_rsize = 1024 * 1024;
 	
+#ifdef __APPLE__
 	// Stat the destination of the archive to determine the optimal fs operation size
 	struct statfs target_mount_stat_fs;
 	if ( statfs(path, &target_mount_stat_fs) == 0 )
@@ -615,6 +619,6 @@ size_t xar_optimal_io_size_at_path(const char *path)
 			optimal_rsize = fs_iosize;
 		}
 	}
-	
+#endif
 	return optimal_rsize;
 }
