diff --git src/kded.cpp src/kded.cpp
index 6929d7d..fc644b4 100644
--- src/kded.cpp
+++ src/kded.cpp
@@ -196,7 +196,14 @@ static KPluginMetaData findModule(const QString &id)
 void Kded::initModules()
 {
     m_dontLoad.clear();
+#ifdef Q_OS_OSX
+    // it seems there is no reason to honour KDE_FULL_SESSION on OS X if it's set as
+    // it means kdeinit5 is not auto-started if required (and the kded is a good candidate
+    // for starting this service).
+    bool kde_running = false;
+#else
     bool kde_running = !qEnvironmentVariableIsEmpty("KDE_FULL_SESSION");
+#endif
     if (kde_running) {
 #ifndef Q_OS_WIN
         // not the same user like the one running the session (most likely we're run via sudo or something)
