diff --git org.kde.kded5.desktop org.kde.kded5.desktop
index a44440c7e097ba389524a09ebbf44df4f92371bc..b51ec4e4c97b9110e930c0ef8a63c77e5e8f7234 100644
--- org.kde.kded5.desktop
+++ org.kde.kded5.desktop
@@ -1,7 +1,7 @@
 [Desktop Entry]
 #This .desktop file is never shown in a menu, hence no real Name or Exec
 #it's here mostly so wayland can get an icon on popups
-Type=SystemService
+Type=Service
 Icon=kde-frameworks
 NoDisplay=true
 
