diff --git a/release/CMakeLists.txt b/release/CMakeLists.txt
index 9bf0816e..09089430 100644
--- a/release/CMakeLists.txt
+++ b/release/CMakeLists.txt
@@ -49,9 +49,9 @@ add_custom_target(dmg sh makemacapp
   DEPENDS vncviewer
   SOURCES makemacapp)
 
-add_custom_target(udmg sh makemacapp universal
-  DEPENDS vncviewer
-  SOURCES makemacapp)
+#add_custom_target(udmg sh makemacapp universal
+#  DEPENDS vncviewer
+#  SOURCES makemacapp)
 
 endif() # APPLE
 
