--- orig.configure.ac	2013-05-18 07:29:13.000000000 +0200
+++ configure.ac	2021-05-08 12:57:05.560934369 +0200
@@ -1,4 +1,6 @@
-AC_INIT
+AC_INIT(darksnow, 0.7.1)
+AM_INIT_AUTOMAKE
+
 AC_CONFIG_SRCDIR([darksnow.c])
 AC_CONFIG_HEADERS([config.h])
 
@@ -8,8 +10,8 @@
 dnl tests C compiler
 AC_LANG([C])
 
-VERSION="0.7.1"
-AC_SUBST(VERSION)
+# VERSION="0.7.1"
+# AC_SUBST(VERSION)
 
 AC_CHECK_FUNC(mmap,,AC_MSG_ERROR(Sorry, you need mmap. If you are reading this, email-me and report this problem and your system description - rafael@riseup.net))
 AC_CHECK_FUNC(gettext,,AC_MSG_ERROR(Sorry, you need gettext. If you are reading this, email-me and report this problem and your system description - rafael@riseup.net))
--- orig.Makefile.in	2013-05-14 19:20:23.000000000 +0200
+++ Makefile.in	2021-05-08 13:13:22.832323570 +0200
@@ -26,6 +26,7 @@
 CFLAGS+=-Wall
 INSTALL=install
 ARCH=`uname -m`
+LDFLAGS=@LDFLAGS@
 
 #set debugging symbols:
 #CFLAGS+= -g 
@@ -39,7 +40,7 @@
 all: darksnow
 
 darksnow: darksnow.o interface.o config_files.o tooltips.o man_gzip
-	$(CC) darksnow.o interface.o config_files.o tooltips.o -o darksnow  @GTK_LIBS@ @GTK_CFLAGS@ -DVERSION="\"$(VERSION)\""
+	$(CC) $(LDFLAGS) darksnow.o interface.o config_files.o tooltips.o -o darksnow  @GTK_LIBS@ @GTK_CFLAGS@ -DVERSION="\"$(VERSION)\""
 
 darksnow.o: darksnow.c
 	$(CC) $(FUNNY) @GTK_CFLAGS@ -DINTLPREFIX="\"$(INTLPREFIX)\"" -DVERSION="\"$(VERSION)\"" -c darksnow.c -o darksnow.o
@@ -61,11 +62,11 @@
 	rm -rf slack darksnow.tgz
 
 install: darksnow
-	$(INSTALL) -D --mode=0755 darksnow $(PREFIX)/bin/darksnow
-	$(INSTALL) -D --mode=0644 documentation/CHANGES $(PREFIX)/share/darksnow/changelog
-	$(INSTALL) -D --mode=0644 po/pt_BR.mo $(PREFIX)/share/locale/pt_BR/LC_MESSAGES/darksnow.mo
-	$(INSTALL) -D --mode=0644 icon/darksnow_icon.png $(PREFIX)/share/icons/hicolor/32x32/apps/darksnow.png
-	$(INSTALL) -D --mode=0644 man/darksnow.1.gz $(PREFIX)/share/man/man1/darksnow.1.gz
+	$(INSTALL) -D --mode=0755 darksnow $(DESTDIR)$(PREFIX)/bin/darksnow
+	$(INSTALL) -D --mode=0644 documentation/CHANGES $(DESTDIR)$(PREFIX)/share/darksnow/changelog
+	$(INSTALL) -D --mode=0644 po/pt_BR.mo $(DESTDIR)$(PREFIX)/share/locale/pt_BR/LC_MESSAGES/darksnow.mo
+	$(INSTALL) -D --mode=0644 icon/darksnow_icon.png $(DESTDIR)$(PREFIX)/share/icons/hicolor/32x32/apps/darksnow.png
+	$(INSTALL) -D --mode=0644 man/darksnow.1.gz $(DESTDIR)$(PREFIX)/share/man/man1/darksnow.1.gz
 
 make_pkg: darksnow
 	echo "Remember to run this option as root!"
