--- orig.configure.ac	2013-05-18 07:29:13.000000000 +0200
+++ configure.ac	2021-05-08 12:57:05.560934369 +0200
@@ -1,4 +1,6 @@
-AC_INIT
+AC_INIT(darksnow, 0.7.1)
+AM_INIT_AUTOMAKE
+
 AC_CONFIG_SRCDIR([darksnow.c])
 AC_CONFIG_HEADERS([config.h])
 
@@ -8,8 +10,8 @@
 dnl tests C compiler
 AC_LANG([C])
 
-VERSION="0.7.1"
-AC_SUBST(VERSION)
+# VERSION="0.7.1"
+# AC_SUBST(VERSION)
 
 AC_CHECK_FUNC(mmap,,AC_MSG_ERROR(Sorry, you need mmap. If you are reading this, email-me and report this problem and your system description - rafael@riseup.net))
 AC_CHECK_FUNC(gettext,,AC_MSG_ERROR(Sorry, you need gettext. If you are reading this, email-me and report this problem and your system description - rafael@riseup.net))
@@ -15,6 +15,7 @@
 
 AC_CHECK_FUNC(mmap,,AC_MSG_ERROR(Sorry, you need mmap. If you are reading this, email-me and report this problem and your system description - rafael@riseup.net))
 AC_CHECK_FUNC(gettext,,AC_MSG_ERROR(Sorry, you need gettext. If you are reading this, email-me and report this problem and your system description - rafael@riseup.net))
+AC_CHECK_FUNC(asprintf)
 
 AM_PATH_GTK_2_0(2.14.0)
 
--- orig.autogen.sh	2013-05-07 20:58:38.000000000 +0200
+++ autogen.sh	2021-05-09 18:30:41.000000000 +0200
@@ -35,9 +35,9 @@
     DIE=1
 }
 
-(libtool --help) < /dev/null > /dev/null 2>&1 || {
+(glibtool --help) < /dev/null > /dev/null 2>&1 || {
     echo
-    echo "You must have libtool installed to compile $package."
+    echo "You must have glibtool installed to compile $package."
     echo "Download the appropriate package for your system,"
     echo "or get the source from one of the GNU ftp sites"
     echo "listed in http://www.gnu.org/order/ftp.html"
@@ -59,8 +59,8 @@
 aclocal $ACLOCAL_FLAGS
 echo "  autoheader"
 autoheader
-echo "  libtoolize --automake"
-libtoolize --automake
+echo "  glibtoolize --automake"
+glibtoolize --automake
 echo "  automake --add-missing $AUTOMAKE_FLAGS"
 automake --add-missing $AUTOMAKE_FLAGS 
 echo "  autoconf"
--- orig.Makefile.in	2013-05-14 19:20:23.000000000 +0200
+++ Makefile.in	2021-05-09 18:57:52.000000000 +0200
@@ -23,9 +23,10 @@
 CC=@CC@
 VERSION=@VERSION@
 #CFLAGS+=-Wall -DGTK_DISABLE_DEPRECATED=1
-CFLAGS+=-Wall
+CFLAGS=@CFLAGS@ -Wall
 INSTALL=install
 ARCH=`uname -m`
+LDFLAGS=@LDFLAGS@
 
 #set debugging symbols:
 #CFLAGS+= -g 
@@ -38,20 +39,11 @@
 
 all: darksnow
 
-darksnow: darksnow.o interface.o config_files.o tooltips.o man_gzip
-	$(CC) darksnow.o interface.o config_files.o tooltips.o -o darksnow  @GTK_LIBS@ @GTK_CFLAGS@ -DVERSION="\"$(VERSION)\""
-
-darksnow.o: darksnow.c
-	$(CC) $(FUNNY) @GTK_CFLAGS@ -DINTLPREFIX="\"$(INTLPREFIX)\"" -DVERSION="\"$(VERSION)\"" -c darksnow.c -o darksnow.o
-
-interface.o: interface.c
-	$(CC) $(CFLAGS) $(FUNNY) @GTK_CFLAGS@ -DINTLPREFIX="\"$(INTLPREFIX)\"" -DVERSION="\"$(VERSION)\"" -c interface.c -o interface.o
+.c.o:
+	$(CC) $(CFLAGS) $(FUNNY) @GTK_CFLAGS@ -DINTLPREFIX="\"$(INTLPREFIX)\"" -DVERSION="\"$(VERSION)\"" -c $< -o $@
 
-config_files.o: config_files.c
-	$(CC) $(CFLAGS) $(FUNNY) @GTK_CFLAGS@ -DINTLPREFIX="\"$(INTLPREFIX)\"" -DVERSION="\"$(VERSION)\"" -c config_files.c -o config_files.o
-
-tooltips.o: tooltips.c
-	$(CC) $(CFLAGS) $(FUNNY) @GTK_CFLAGS@ -DINTLPREFIX="\"$(INTLPREFIX)\"" -DVERSION="\"$(VERSION)\"" -c tooltips.c -o tooltips.o
+darksnow: darksnow.o interface.o config_files.o tooltips.o man_gzip
+	$(CC) $(LDFLAGS) darksnow.o interface.o config_files.o tooltips.o -o darksnow  @GTK_LIBS@ @GTK_CFLAGS@ -DVERSION="\"$(VERSION)\""
 
 man_gzip:
 	gzip -c man/darksnow.1 > man/darksnow.1.gz 
@@ -61,11 +53,11 @@
 	rm -rf slack darksnow.tgz
 
 install: darksnow
-	$(INSTALL) -D --mode=0755 darksnow $(PREFIX)/bin/darksnow
-	$(INSTALL) -D --mode=0644 documentation/CHANGES $(PREFIX)/share/darksnow/changelog
-	$(INSTALL) -D --mode=0644 po/pt_BR.mo $(PREFIX)/share/locale/pt_BR/LC_MESSAGES/darksnow.mo
-	$(INSTALL) -D --mode=0644 icon/darksnow_icon.png $(PREFIX)/share/icons/hicolor/32x32/apps/darksnow.png
-	$(INSTALL) -D --mode=0644 man/darksnow.1.gz $(PREFIX)/share/man/man1/darksnow.1.gz
+	$(INSTALL) -D --mode=0755 darksnow $(DESTDIR)$(PREFIX)/bin/darksnow
+	$(INSTALL) -D --mode=0644 documentation/CHANGES $(DESTDIR)$(PREFIX)/share/darksnow/changelog
+	$(INSTALL) -D --mode=0644 po/pt_BR.mo $(DESTDIR)$(PREFIX)/share/locale/pt_BR/LC_MESSAGES/darksnow.mo
+	$(INSTALL) -D --mode=0644 icon/darksnow_icon.png $(DESTDIR)$(PREFIX)/share/icons/hicolor/32x32/apps/darksnow.png
+	$(INSTALL) -D --mode=0644 man/darksnow.1.gz $(DESTDIR)$(PREFIX)/share/man/man1/darksnow.1.gz
 
 make_pkg: darksnow
 	echo "Remember to run this option as root!"
--- orig.darksnow.c	2013-05-18 07:29:13.000000000 +0200
+++ darksnow.c	2021-05-09 19:05:22.000000000 +0200
@@ -52,7 +52,7 @@
   DIR *directory;
   char *foo;
   char bar[256];
-  char locale[16];
+  char *locale = NULL;
   int i = 0, j = 0, fd;
 
   if (!gtk_init_check (&argc, &argv)) {
@@ -105,8 +105,7 @@
   /* internacionalization stuff */
   setlocale (LC_ALL, "");
 
-  strcpy(locale, INTLPREFIX);
-  strcat(locale, "/share/locale");
+  asprintf(&locale, "%s/share/locale", INTLPREFIX);
 
   bindtextdomain ("darksnow", locale);
   textdomain ("darksnow");
