Index: src/quicktimevideo.cpp
===================================================================
--- src/quicktimevideo.cpp	(revision 3839)
+++ src/quicktimevideo.cpp	(working copy)
@@ -886,6 +886,7 @@
     void QuickTimeVideo::userDataDecoder(unsigned long size_external)
     {
         uint64_t cur_pos = io_->tell();
+#if 0
         const TagVocabulary* td;
         const TagVocabulary* tv, *tv_internal;
 
@@ -949,6 +950,7 @@
             else if(td)
                 tagDecoder(buf,size-8);
         }
+#endif
 
         io_->seek(cur_pos + size_external, BasicIo::beg);
     } // QuickTimeVideo::userDataDecoder
