--- orig.CMakeLists.txt	2017-04-26 18:26:50.000000000 +0200
+++ CMakeLists.txt	2017-07-13 10:00:05.308219406 +0200
@@ -83,9 +83,8 @@
     execute_process(COMMAND ${CMAKE_CXX_COMPILER} -dumpversion OUTPUT_VARIABLE GCC_VERSION)
     string(REGEX MATCHALL "[0-9]+" GCC_VERSION_COMPONENTS ${GCC_VERSION})
     list(GET GCC_VERSION_COMPONENTS 0 GCC_MAJOR)
-    list(GET GCC_VERSION_COMPONENTS 1 GCC_MINOR)
 
-    message(STATUS Compiler: ${COMPILER} " Major:" ${GCC_MAJOR} " Minor:" ${GCC_MINOR})
+    message(STATUS Compiler: ${COMPILER} " Major:" ${GCC_MAJOR} " VERSION:" ${GCC_VERSION})
 
     IF ( CYGWIN OR ( ${GCC_MAJOR} GREATER 5 ))
         ADD_DEFINITIONS( -std=gnu++98 ) # to support snprintf
