--- a/qtbase/mkspecs/common/macx.conf
+++ b/qtbase/mkspecs/common/macx.conf
@@ -5,4 +5,9 @@
 QMAKE_PLATFORM         += osx macx
 QMAKE_MAC_SDK           = macosx
 
+# X11 is not inexistant on OS X; provide the library requirements but leave it to the
+# user to make them accessible during the Qt build.
+QMAKE_LIBS_X11          = -lXext -lX11
+QMAKE_LIBS_OPENVG       = -lOpenVG
+
 include(mac.conf)
--- a/qtbase/src/platformsupport/fontdatabases/fontdatabases.pri
+++ b/qtbase/src/platformsupport/fontdatabases/fontdatabases.pri
@@ -1,8 +1,8 @@
-!win32|contains(QT_CONFIG, freetype):!mac {
+!win32|contains(QT_CONFIG, freetype) {
     include($$PWD/basic/basic.pri)
 }
 
-unix:!mac {
+unix {
     CONFIG += qpa/genericunixfontdatabase
     include($$PWD/genericunix/genericunix.pri)
     contains(QT_CONFIG,fontconfig) {
--- a/qtbase/src/platformsupport/services/services.pri
+++ b/qtbase/src/platformsupport/services/services.pri
@@ -1,3 +1,3 @@
-unix:!mac {
+unix {
     include($$PWD/genericunix/genericunix.pri)
 }
--- a/qtbase/src/plugins/platforms/offscreen/offscreen.pro
+++ b/qtbase/src/plugins/platforms/offscreen/offscreen.pro
@@ -18,7 +18,7 @@
 
 OTHER_FILES += offscreen.json
 
-contains(QT_CONFIG, xlib):contains(QT_CONFIG, opengl):!contains(QT_CONFIG, opengles2) {
+contains(QT_CONFIG, xlib):contains(QT_CONFIG, opengl):!contains(QT_CONFIG, opengles2):!mac {
     SOURCES += qoffscreenintegration_x11.cpp
     HEADERS += qoffscreenintegration_x11.h
     system(echo "Using X11 offscreen integration with GLX")
--- a/qtbase/src/plugins/platforms/xcb/xcb_qpa_lib.pro
+++ b/qtbase/src/plugins/platforms/xcb/xcb_qpa_lib.pro
@@ -65,6 +65,9 @@
                        LIBXI_PATCH=$$QMAKE_LIBXI_VERSION_PATCH
         }
     }
+    mac {
+	    LIBS += -lGL -framework OpenGL -framework AGL
+    }
 }
 
 # to support custom cursors with depth > 1
