--- qtbase/tests/auto/corelib/tools/qarraydata/tst_qarraydata.cpp
+++ qtbase/tests/auto/corelib/tools/qarraydata/tst_qarraydata.cpp
@@ -789,7 +789,7 @@
                     + minAlignment - Q_ALIGNOF(QArrayData)));
 
         // Data is aligned
-        QCOMPARE(quintptr(data->data()) % alignment, quintptr(0u));
+        QCOMPARE(quintptr(quintptr(data->data()) % alignment), quintptr(0));
 
         // Check that the allocated array can be used. Best tested with a
         // memory checker, such as valgrind, running.
