--- a/qtbase/src/platformsupport/fontdatabases/mac/qcoretextfontdatabase.mm
+++ b/qtbase/src/platformsupport/fontdatabases/mac/qcoretextfontdatabase.mm
@@ -209,6 +209,7 @@
         QPlatformFontDatabase::registerFontFamily(familyName);
 
 #if defined(Q_OS_OSX)
+        QMacAutoReleasePool pool;
         QString localizedFamilyName = QString::fromNSString([[NSFontManager sharedFontManager] localizedNameForFamily:(NSString*)familyNameRef face:nil]);
         if (familyName != localizedFamilyName)
             QPlatformFontDatabase::registerAliasToFontFamily(familyName, localizedFamilyName);
--- a/qtbase/src/plugins/platforms/cocoa/qcocoanativeinterface.mm
+++ b/qtbase/src/plugins/platforms/cocoa/qcocoanativeinterface.mm
@@ -253,6 +253,7 @@
 
 void *QCocoaNativeInterface::qMenuToNSMenu(QPlatformMenu *platformMenu)
 {
+    QMacAutoReleasePool pool;
     QCocoaMenu *cocoaPlatformMenu = static_cast<QCocoaMenu *>(platformMenu);
     NSMenu *menu = cocoaPlatformMenu->nsMenu();
     return reinterpret_cast<void *>(menu);
--- a/qtbase/src/plugins/platforms/cocoa/qcocoawindow.mm
+++ b/qtbase/src/plugins/platforms/cocoa/qcocoawindow.mm
@@ -1688,6 +1688,8 @@
     if (!window)
         return;
 
+    QMacAutoReleasePool pool;
+
     if (!m_drawContentBorderGradient) {
         [window setStyleMask:[window styleMask] & ~NSTexturedBackgroundWindowMask];
         [[[window contentView] superview] setNeedsDisplay:YES];
