--- a/qtbase/src/widgets/widgets/qmenu_mac.mm
+++ b/qtbase/src/widgets/widgets/qmenu_mac.mm
@@ -92,7 +92,9 @@
     QPlatformNativeInterface::NativeResourceForIntegrationFunction function = resolvePlatformFunction("setdockmenu");
     if (function) {
         typedef void (*SetDockMenuFunction)(QPlatformMenu *platformMenu);
-        reinterpret_cast<SetDockMenuFunction>(function)(platformMenu());
+        QPlatformMenu *pMenu = platformMenu();
+        if (pMenu)
+            reinterpret_cast<SetDockMenuFunction>(function)(pMenu);
     }
 }
 
