diff --git a/CMakeLists.txt b/CMakeLists.txt
index b0b638e..4c4ad39 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -18,7 +18,7 @@ set(QTXDG_PATCH_VERSION 0)
 set(QTXDG_VERSION_STRING ${QTXDG_MAJOR_VERSION}.${QTXDG_MINOR_VERSION}.${QTXDG_PATCH_VERSION})
 
 set(LXQTBT_MINIMUM_VERSION "0.7.0")
-set(QT_MINIMUM_VERSION "5.10.0")
+set(QT_MINIMUM_VERSION "5.9.8")
 set(GLIB_MINIMUM_VERSION "2.41.0") # Mime Apps new implementation
 
 find_package(lxqt-build-tools ${LXQTBT_MINIMUM_VERSION} REQUIRED)
