diff --git phonon/CMakeLists.txt phonon/CMakeLists.txt
index d93124b..fc37fe4 100644
--- phonon/CMakeLists.txt
+++ phonon/CMakeLists.txt
@@ -47,10 +47,12 @@ set_directory_properties(PROPERTIES ADDITIONAL_MAKE_CLEAN_FILES ${CMAKE_CURRENT_
 
 # ---------------------------------------------------------------------------- #
 
-macro_optional_find_package(PulseAudio 0.9.15)
-macro_log_feature(PULSEAUDIO_FOUND "PulseAudio" "A cross-platform, networked sound server." "http://www.pulseaudio.org" FALSE "" "Allows audio playback via the PulseAudio soundserver when it is running")
-macro_optional_find_package(GLIB2)
-macro_log_feature(GLIB2_FOUND "GLib2" "GLib 2 is required to compile the pulseaudio for Phonon" "http://www.gtk.org/download/" FALSE)
+if (NOT APPLE)
+   macro_optional_find_package(PulseAudio 0.9.15)
+   macro_log_feature(PULSEAUDIO_FOUND "PulseAudio" "A cross-platform, networked sound server." "http://www.pulseaudio.org" FALSE "" "Allows audio playback via the PulseAudio soundserver when it is running")
+   macro_optional_find_package(GLIB2)
+   macro_log_feature(GLIB2_FOUND "GLib2" "GLib 2 is required to compile the pulseaudio for Phonon" "http://www.gtk.org/download/" FALSE)
+endif(NOT APPLE)
 
 if (GLIB2_FOUND AND PULSEAUDIO_FOUND)
    set(PHONON_PULSESUPPORT TRUE CACHE BOOL "Has Phonon pulseaudio support ?")
