diff --git vncviewer/CMakeLists.txt vncviewer/CMakeLists.txt
index 51debc5e..bacfcd19 100644
--- vncviewer/CMakeLists.txt
+++ vncviewer/CMakeLists.txt
@@ -85,6 +85,8 @@ endif()
 install(TARGETS vncviewer DESTINATION ${CMAKE_INSTALL_FULL_BINDIR})
 if(UNIX)
   install(FILES vncviewer.man DESTINATION ${CMAKE_INSTALL_FULL_MANDIR}/man1 RENAME vncviewer.1)
+endif()
+if(FALSE)
 
   file(STRINGS ${CMAKE_SOURCE_DIR}/po/LINGUAS LINGUAS)
   set(po_FILES "")
