diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1cd1f696116c389e4c37e7896a97df67b04bd430..4978b9c4c6e9d88abe72e4402d03bd527aaa0c29 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -234,6 +234,12 @@ set(FLTK_SKIP_FLUID TRUE)
 set(FLTK_SKIP_OPENGL TRUE)
 set(FLTK_SKIP_FORMS TRUE)
 find_package(FLTK)
+find_program(FLTK_CONFIG_SCRIPT fltk-config PATHS ${FLTK_BIN_DIR})
+if(FLTK_CONFIG_SCRIPT)
+  exec_program(${FLTK_CONFIG_SCRIPT} ARGS --cxxflags OUTPUT_VARIABLE FLTK_CXXFLAGS)
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${FLTK_CXXFLAGS}")
+endif()
+
 
 if(UNIX AND NOT APPLE)
   # No proper handling for extra X11 libs that FLTK might need...
