diff --git unix/w0vncserver/wayland/objects/Shm.h unix/w0vncserver/wayland/objects/Shm.h
index 7f25fd0c..25f9eef4 100644
--- unix/w0vncserver/wayland/objects/Shm.h
+++ unix/w0vncserver/wayland/objects/Shm.h
@@ -19,6 +19,24 @@
 #ifndef __WAYLAND_SHM_H__
 #define __WAYLAND_SHM_H__
 
+/* memfd_create and mlock2 were introduced in glibc 2.27 */
+#       if !defined(__GLIBC__) || (__GLIBC__ >= 2 && __GLIBC_MINOR__ >= 27)
+// all good
+#       elif defined(__USE_GNU)
+
+   #include <unistd.h>
+   #include <sys/syscall.h>
+
+   /* Flags for memfd_create.  */
+   #ifndef MFD_CLOEXEC
+   # define MFD_CLOEXEC 1U
+   # define MFD_ALLOW_SEALING 2U
+   # define MFD_HUGETLB 4U
+   #endif
+
+   #define memfd_create(__name,__flags) syscall(__NR_memfd_create,(__name),(__flags))
+#endif
+
 #include "Object.h"
 
 struct wl_shm_listener;
