diff --git a/release/makemacapp.in b/release/makemacapp.in
index 2bfd869e..1cdbdd0b 100644
--- a/release/makemacapp.in
+++ b/release/makemacapp.in
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/sh -x
 
 set -u
 set -e
@@ -10,9 +10,7 @@ TMPDIR=
 
 onexit()
 {
-	if [ ! "$TMPDIR" = "" ]; then
-		rm -rf $TMPDIR
-	fi
+	echo "onexit $0 $@" 1>&2
 }
 
 usage()
@@ -42,8 +40,7 @@ if [ -f $PACKAGE_NAME.dmg ]; then
 fi
 
 umask 022
-TMPDIR=`mktemp -d /tmp/$PACKAGE_NAME-build.XXXXXX`
-APPROOT="$TMPDIR/dmg/TigerVNC Viewer $VERSION.app"
+APPROOT="./TigerVNC Viewer.app"
 mkdir -p "$APPROOT/Contents/MacOS"
 mkdir -p "$APPROOT/Contents/Resources"
 
@@ -67,12 +64,5 @@ fi
 install -m 644 $SRCDIR/release/tigervnc.icns "$APPROOT/Contents/Resources/"
 install -m 644 release/Info.plist "$APPROOT/Contents/"
 
-install -m 644 $SRCDIR/LICENCE.TXT $TMPDIR/dmg/
-install -m 644 $SRCDIR/README.rst $TMPDIR/dmg/
-
-hdiutil create -fs HFS+ -volname $PACKAGE_NAME-$VERSION \
-	-srcfolder "$TMPDIR/dmg" \
-	$TMPDIR/$PACKAGE_NAME-$VERSION.dmg 
-cp $TMPDIR/$PACKAGE_NAME-$VERSION.dmg . 
-
 exit
+
