diff --git vncviewer/parameters.cxx vncviewer/parameters.cxx
index 75d46dca..def60bca 100644
--- vncviewer/parameters.cxx
+++ vncviewer/parameters.cxx
@@ -126,7 +126,7 @@ BoolParameter listenMode("listen", "Listen for connections from VNC servers", fa
 BoolParameter remoteResize("RemoteResize",
                            "Dynamically resize the remote desktop size as "
                            "the size of the local client window changes. "
-                           "(Does not work with all servers)", true);
+                           "(Does not work with all servers)", false);
 
 BoolParameter viewOnly("ViewOnly",
                        "Don't send any mouse or keyboard events to the server",
