diff --git vncviewer/fltk/theme.cxx vncviewer/fltk/theme.cxx
index 22e04dc6..3a394400 100644
--- vncviewer/fltk/theme.cxx
+++ vncviewer/fltk/theme.cxx
@@ -253,10 +253,10 @@ void init_theme()
   // Windows 11 default accent color
   Fl::set_color(FL_SELECTION_COLOR, 0, 103, 192);
 #elif defined(__APPLE__)
-  Fl::set_color(FL_SELECTION_COLOR, 0, 122, 255);
+  Fl::set_color(FL_SELECTION_COLOR, 184, 204, 126);
 #else
   // GNOME
-  Fl::set_color(FL_SELECTION_COLOR, 53, 132, 228);
+  Fl::set_color(FL_SELECTION_COLOR, 184, 204, 126);
 #endif
 
   // The arrow on Fl_Return_Button gets a invisible, so let's adjust it
