diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1cd1f696..5fb86447 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -147,6 +147,7 @@ find_package(Pixman REQUIRED)
 
 # Check for gettext
 option(ENABLE_NLS "Enable translation of program messages" ON)
+option(USE_GETTEXT_FROM_LIBINTL "Force the use of gettext functions from libintl instead of libc" OFF)
 if(ENABLE_NLS)
   # Tools
   find_package(Gettext)
@@ -163,7 +164,7 @@ if(ENABLE_NLS)
       set(CMAKE_REQUIRED_LIBRARIES ${ICONV_LIBRARIES})
       set(CMAKE_REQUIRED_FLAGS -fno-builtin-dgettext)
       check_function_exists(dgettext LIBC_HAS_DGETTEXT)
-      if(LIBC_HAS_DGETTEXT)
+      if(LIBC_HAS_DGETTEXT AND NOT USE_GETTEXT_FROM_LIBINTL)
         set(GETTEXT_FOUND TRUE)
       else()
         find_library(LIBINTL_LIBRARY NAMES intl libintl)
