--- plasmate/app/orig.main.cpp	2016-01-26 20:44:27.000000000 +0100
+++ plasmate/app/main.cpp	2016-02-03 18:47:15.000000000 +0100
@@ -51,7 +51,7 @@
     aboutData.addAuthor(i18n("Giorgos Tsiapaliokas"), i18n("Author"), "terietor@gmail.com");
     aboutData.addAuthor(i18n("Antonis Tsiapaliokas"), i18n("Author"), "kok3rs@gmail.com");
 
-    app.setWindowIcon(QIcon::fromTheme("plasmagik"));
+    app.setWindowIcon(QIcon::fromTheme("plasmagik", app.windowIcon()));
 
     QCommandLineParser parser;
     parser.addVersionOption();
