--- src/orig.kuitmarkup.cpp	2019-07-07 20:34:10.000000000 +0200
+++ src/kuitmarkup.cpp	2021-03-14 18:44:56.000000000 +0100
@@ -496,11 +496,11 @@
             formattedText = modText;
         }
     } else if (patterns.contains(attribKey)) {
-        qWarning() << QStringLiteral(
+        qDebug() << QStringLiteral(
             "Undefined visual format for tag <%1> and attribute combination %2: %3.")
                    .arg(name, attribKey, s->namesByFormat.value(format));
     } else {
-        qWarning() << QStringLiteral(
+        qDebug() << QStringLiteral(
                        "Undefined attribute combination for tag <%1>: %2.")
                    .arg(name, attribKey);
     }
@@ -1254,7 +1254,7 @@
     Kuit::Role role = s->rolesByName.value(roleName, Kuit::UndefinedRole);
     if (role == Kuit::UndefinedRole) { // unknown role
         if (!roleName.isEmpty()) {
-            qWarning() << QStringLiteral(
+            qDebug() << QStringLiteral(
                            "Unknown role '@%1' in UI marker in context {%2}.")
                        .arg(roleName, shorten(context));
         }
@@ -1267,13 +1267,13 @@
         if (cue != Kuit::UndefinedCue) { // known subcue
             if (!s->knownRoleCues.value(role).contains(cue)) {
                 cue = Kuit::UndefinedCue;
-                qWarning() << QStringLiteral(
+                qDebug() << QStringLiteral(
                                "Subcue ':%1' does not belong to role '@%2' in UI marker in context {%3}.")
                            .arg(cueName, roleName, shorten(context));
             }
         } else { // unknown or not given subcue
             if (!cueName.isEmpty()) {
-                qWarning() << QStringLiteral(
+                qDebug() << QStringLiteral(
                                "Unknown subcue ':%1' in UI marker in context {%2}.")
                            .arg(cueName, shorten(context));
             }
@@ -1296,7 +1296,7 @@
             }
         }
         if (!formatName.isEmpty()) {
-            qWarning() << QStringLiteral(
+            qDebug() << QStringLiteral(
                            "Unknown format '/%1' in UI marker for message {%2}.")
                        .arg(formatName, shorten(context));
         }
@@ -1464,7 +1464,7 @@
             oel.handling = OpenEl::Proper;
         } else {
             oel.handling = OpenEl::Dropout;
-            qWarning() << QStringLiteral(
+            qDebug() << QStringLiteral(
                            "Structuring tag ('%1') cannot be subtag of phrase tag ('%2') in message {%3}.")
                        .arg(tag.name, etag.name, shorten(text));
         }
@@ -1477,7 +1477,7 @@
                 attset << att;
                 oel.attributes[att] = attribValues[i];
             } else {
-                qWarning() << QStringLiteral(
+                qDebug() << QStringLiteral(
                                "Attribute '%1' not defined for tag '%2' in message {%3}.")
                            .arg(att, tag.name, shorten(text));
             }
@@ -1489,7 +1489,7 @@
 
     } else { // unknown element, leave it in verbatim
         oel.handling = OpenEl::Ignored;
-        qWarning() << QStringLiteral(
+        qDebug() << QStringLiteral(
                        "Tag '%1' is not defined in message {%2}.")
                    .arg(oel.name, shorten(text));
     }
