diff --git a/CMakeLists.txt b/CMakeLists.txt
index 00e1413..b907143 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,6 +9,7 @@ include(KDEInstallDirs)
 include(KDECMakeSettings)
 include(KDECompilerSettings NO_POLICY_SCOPE)
 include(ECMQtDeclareLoggingCategory)
+include(ECMMarkNonGuiExecutable)
 
 # Required for ki18n_install(po) added on tarball generation
 find_package(KF5 REQUIRED COMPONENTS I18n)
diff --git a/kpartloader/CMakeLists.txt b/kpartloader/CMakeLists.txt
index 1fe4136..8c88177 100644
--- a/kpartloader/CMakeLists.txt
+++ b/kpartloader/CMakeLists.txt
@@ -20,6 +20,7 @@ set(kpartloader_SRCS
 )
 
 add_executable(kpartloader ${kpartloader_SRCS})
+ecm_mark_nongui_executable(kpartloader)
 target_link_libraries(kpartloader KF5::Parts)
 install(TARGETS kpartloader ${KDE_INSTALL_TARGETS_DEFAULT_ARGS})
 install(FILES kpartloaderui.rc DESTINATION ${KDE_INSTALL_KXMLGUI5DIR}/kpartloader)
diff --git a/kuiviewer/CMakeLists.txt b/kuiviewer/CMakeLists.txt
index 3fe987e..2b350bd 100644
--- a/kuiviewer/CMakeLists.txt
+++ b/kuiviewer/CMakeLists.txt
@@ -38,6 +38,7 @@ file(GLOB ICONS_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/*-apps-kuiviewer.png")
 ecm_add_app_icon(kuiviewer_SRCS ICONS ${ICONS_SRCS})
 
 add_executable(kuiviewer ${kuiviewer_SRCS})
+ecm_mark_nongui_executable(kuiviewer)
 
 target_link_libraries(kuiviewer  KF5::Parts  )
 
