diff --git formatters/customscript/customscript_plugin.cpp formatters/customscript/customscript_plugin.cpp
index 67d7537..49c7616 100644
--- formatters/customscript/customscript_plugin.cpp
+++ formatters/customscript/customscript_plugin.cpp
@@ -244,12 +244,12 @@ KDevelop::SourceFormatterStyle CustomScriptPlugin::predefinedStyle(const QString
     } else if (name == "GNU_indent_orig") {
         result.setCaption(i18n("Gnu Indent: Original Berkeley indent style"));
         result.setContent("indent -orig");
-    } else if (name == "kdev_format_source") {
-        result.setCaption("KDevelop: kdev_format_source");
-        result.setContent("kdev_format_source $FILE $TMPFILE");
+    } else if (name == "kdev_format_source5") {
+        result.setCaption("KDevelop: kdev_format_source5");
+        result.setContent("kdev_format_source5 $FILE $TMPFILE");
         result.setUsePreview(false);
         result.setDescription(i18n("Description:<br />"
-                                   "<b>kdev_format_source</b> is a script bundled with KDevelop "
+                                   "<b>kdev_format_source5</b> is a script bundled with KDevelop "
                                    "which allows using fine-grained formatting rules by placing "
                                    "meta-files called <b>format_sources</b> into the file-system.<br /><br />"
                                    "Each line of the <b>format_sources</b> files defines a list of wildcards "
@@ -276,7 +276,7 @@ KDevelop::SourceFormatterStyle CustomScriptPlugin::predefinedStyle(const QString
 QList<KDevelop::SourceFormatterStyle> CustomScriptPlugin::predefinedStyles()
 {
     QList<KDevelop::SourceFormatterStyle> styles = stylesFromLanguagePlugins();
-    styles << predefinedStyle("kdev_format_source");
+    styles << predefinedStyle("kdev_format_source5");
     styles << predefinedStyle("GNU_indent_GNU");
     styles << predefinedStyle("GNU_indent_KR");
     styles << predefinedStyle("GNU_indent_orig");
