diff --git formatters/astyle/astyle_plugin.cpp formatters/astyle/astyle_plugin.cpp
index 7a6388d..0a17a92 100644
--- formatters/astyle/astyle_plugin.cpp
+++ formatters/astyle/astyle_plugin.cpp
@@ -103,6 +103,9 @@ static SourceFormatterStyle::MimeList supportedMimeTypes()
         {QStringLiteral("text/x-csrc"), QStringLiteral("C")},
         {QStringLiteral("text/x-java"), QStringLiteral("Java")},
         {QStringLiteral("text/x-csharp"), QStringLiteral("C#")},
+        {QStringLiteral("text/x-objchdr"), QStringLiteral("ObjC")},
+        {QStringLiteral("text/x-objcsrc"), QStringLiteral("ObjC")},
+        {QStringLiteral("text/x-objc++src"), QStringLiteral("ObjC++")},
     };
 }
 
diff --git languages/clang/duchain/documentfinderhelpers.cpp languages/clang/duchain/documentfinderhelpers.cpp
index 0873bea..8a7e67d 100644
--- languages/clang/duchain/documentfinderhelpers.cpp
+++ languages/clang/duchain/documentfinderhelpers.cpp
@@ -185,9 +185,11 @@ QStringList mimeTypesList()
     static const QStringList mimeTypes = {
         QStringLiteral("text/x-chdr"),
         QStringLiteral("text/x-c++hdr"),
+        QStringLiteral("text/x-objchdr"),
         QStringLiteral("text/x-csrc"),
         QStringLiteral("text/x-c++src"),
-        QStringLiteral("text/x-objcsrc")
+        QStringLiteral("text/x-objcsrc"),
+        QStringLiteral("text/x-objc++src")
     };
     return mimeTypes;
 }
diff --git languages/clang/duchain/parsesession.cpp languages/clang/duchain/parsesession.cpp
index 6527514..fbb0dd5 100644
--- languages/clang/duchain/parsesession.cpp
+++ languages/clang/duchain/parsesession.cpp
@@ -68,7 +68,8 @@ QVector<QByteArray> extraArgs()
 QVector<QByteArray> argsForSession(const QString& path, ParseSessionData::Options options, const ParserSettings& parserSettings)
 {
     QMimeDatabase db;
-    if(db.mimeTypeForFile(path).name() == QStringLiteral("text/x-objcsrc")) {
+    QString mimeType = db.mimeTypeForFile(path).name();
+    if( mimeType == QStringLiteral("text/x-objcsrc") || mimeType == QStringLiteral("text/x-objc++src")) {
         return {QByteArrayLiteral("-xobjective-c++")};
     }
 
diff --git languages/clang/kdevclangsupport.json languages/clang/kdevclangsupport.json
index 1025366..c90e2c5 100644
--- languages/clang/kdevclangsupport.json
+++ languages/clang/kdevclangsupport.json
@@ -57,6 +57,10 @@
         "text/x-c++hdr", 
         "text/x-csrc", 
         "text/x-c++src", 
-        "text/x-objcsrc"
+        "text/x-objcsrc",
+        "text/x-objchdr",
+        "text/x-objcsrc",
+        "text/x-objc++src"
+
     ]
 }
\ No newline at end of file
