diff --git projectmanagers/cmake/settings/cmakebuildsettings.ui projectmanagers/cmake/settings/cmakebuildsettings.ui
index 96362dd..0d5abbf 100644
--- projectmanagers/cmake/settings/cmakebuildsettings.ui
+++ projectmanagers/cmake/settings/cmakebuildsettings.ui
@@ -7,7 +7,7 @@
     <x>0</x>
     <y>0</y>
     <width>605</width>
-    <height>537</height>
+    <height>547</height>
    </rect>
   </property>
   <property name="sizePolicy">
@@ -110,14 +110,78 @@
       <property name="fieldGrowthPolicy">
        <enum>QFormLayout::ExpandingFieldsGrow</enum>
       </property>
+      <item row="0" column="0">
+       <widget class="QLabel" name="installationPrefixLabel">
+        <property name="text">
+         <string>Installation Prefix</string>
+        </property>
+       </widget>
+      </item>
+      <item row="0" column="1">
+       <widget class="KUrlRequester" name="installationPrefix"/>
+      </item>
       <item row="1" column="0">
+       <widget class="QLabel" name="buildTypeLabel">
+        <property name="text">
+         <string>Build Type</string>
+        </property>
+       </widget>
+      </item>
+      <item row="1" column="1">
+       <widget class="QComboBox" name="buildType">
+        <property name="editable">
+         <bool>true</bool>
+        </property>
+        <item>
+         <property name="text">
+          <string notr="true">Debug</string>
+         </property>
+        </item>
+        <item>
+         <property name="text">
+          <string notr="true">Release</string>
+         </property>
+        </item>
+        <item>
+         <property name="text">
+          <string notr="true">RelWithDebInfo</string>
+         </property>
+        </item>
+        <item>
+         <property name="text">
+          <string notr="true">MinSizeRel</string>
+         </property>
+        </item>
+       </widget>
+      </item>
+      <item row="2" column="0">
+       <widget class="QLabel" name="extraArgumentsLabel">
+        <property name="text">
+         <string>Extra Arguments</string>
+        </property>
+       </widget>
+      </item>
+      <item row="2" column="1">
+       <widget class="KComboBox" name="extraArguments">
+        <property name="sizePolicy">
+         <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
+          <horstretch>0</horstretch>
+          <verstretch>0</verstretch>
+         </sizepolicy>
+        </property>
+        <property name="editable">
+         <bool>true</bool>
+        </property>
+       </widget>
+      </item>
+      <item row="3" column="0">
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Configure Environment</string>
         </property>
        </widget>
       </item>
-      <item row="1" column="1">
+      <item row="3" column="1">
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="KDevelop::EnvironmentSelectionWidget" name="environment">
@@ -137,13 +201,23 @@
         </item>
        </layout>
       </item>
-      <item row="2" column="1">
+      <item row="5" column="1">
        <widget class="QCheckBox" name="showInternal">
         <property name="text">
          <string>Show Advanced Values</string>
         </property>
        </widget>
       </item>
+      <item row="4" column="1">
+       <widget class="KUrlRequester" name="cMakeBinary"/>
+      </item>
+      <item row="4" column="0">
+       <widget class="QLabel" name="cMakeBinaryLabel">
+        <property name="text">
+         <string>CMake Binary</string>
+        </property>
+       </widget>
+      </item>
      </layout>
     </widget>
    </item>
@@ -156,6 +230,11 @@
    <header>kcombobox.h</header>
   </customwidget>
   <customwidget>
+   <class>KUrlRequester</class>
+   <extends>QWidget</extends>
+   <header>kurlrequester.h</header>
+  </customwidget>
+  <customwidget>
    <class>KDevelop::EnvironmentSelectionWidget</class>
    <extends>KComboBox</extends>
    <header>util/environmentselectionwidget.h</header>
diff --git projectmanagers/cmake/settings/cmakepreferences.cpp projectmanagers/cmake/settings/cmakepreferences.cpp
index e7cfbbb..1df33b9 100644
--- projectmanagers/cmake/settings/cmakepreferences.cpp
+++ projectmanagers/cmake/settings/cmakepreferences.cpp
@@ -30,6 +30,7 @@
 #include <QFile>
 #include <QDir>
 #include <QHeaderView>
+#include <QComboBox>
 
 #include "ui_cmakebuildsettings.h"
 #include "cmakecachedelegate.h"
@@ -79,6 +80,19 @@ CMakePreferences::CMakePreferences(IPlugin* plugin, const ProjectConfigOptions&
     connect(m_prefsUi->configureEnvironment, &EnvironmentConfigureButton::environmentConfigured,
             this, &CMakePreferences::changed);
 
+    connect(m_prefsUi->installationPrefix, &KUrlRequester::textChanged,
+            this, &CMakePreferences::changed);
+    connect(m_prefsUi->buildType, static_cast<void(QComboBox::*)(const QString&)>(&QComboBox::currentIndexChanged),
+            this, &CMakePreferences::changed);
+    connect(m_prefsUi->buildType, &QComboBox::currentTextChanged,
+            this, &CMakePreferences::changed);
+    connect(m_prefsUi->extraArguments, &KComboBox::currentTextChanged,
+            this, &CMakePreferences::changed);
+    connect(m_prefsUi->extraArguments, &KComboBox::editTextChanged,
+            this, &CMakePreferences::changed);
+    connect(m_prefsUi->cMakeBinary, &KUrlRequester::textChanged,
+            this, &CMakePreferences::changed);
+
     showInternal(m_prefsUi->showInternal->checkState());
     m_subprojFolder = Path(options.projectTempFile).parent();
 
@@ -101,6 +115,19 @@ CMakePreferences::~CMakePreferences()
     delete m_prefsUi;
 }
 
+void CMakePreferences::initAdvanced()
+{
+    m_prefsUi->environment->setCurrentProfile( CMake::currentEnvironment(m_project) );
+    m_prefsUi->installationPrefix->setText(CMake::currentInstallDir(m_project).toLocalFile());
+    const QString buildType = CMake::currentBuildType(m_project);
+    if (m_prefsUi->buildType->findText(buildType) == -1) {
+        m_prefsUi->buildType->addItem(buildType);
+    }
+    m_prefsUi->buildType->setCurrentIndex(m_prefsUi->buildType->findText(buildType));
+    m_prefsUi->extraArguments->setEditText(CMake::currentExtraArguments(m_project));
+    m_prefsUi->cMakeBinary->setText(CMake::currentCMakeBinary(m_project).toLocalFile());
+}
+
 void CMakePreferences::reset()
 {
     qCDebug(CMAKE) << "********loading";
@@ -108,7 +135,8 @@ void CMakePreferences::reset()
     m_prefsUi->buildDirs->addItems( CMake::allBuildDirs(m_project) );
     CMake::removeOverrideBuildDirIndex(m_project); // addItems() triggers buildDirChanged(), compensate for it
     m_prefsUi->buildDirs->setCurrentIndex( CMake::currentBuildDirIndex(m_project) );
-    m_prefsUi->environment->setCurrentProfile( CMake::currentEnvironment(m_project) );
+
+    initAdvanced();
 
     m_srcFolder = m_project->path();
 
@@ -133,6 +161,15 @@ void CMakePreferences::apply()
 
     CMake::setCurrentEnvironment( m_project, m_prefsUi->environment->currentProfile() );
 
+    CMake::setCurrentInstallDir( m_project, Path(m_prefsUi->installationPrefix->text()) );
+    const QString buildType = m_prefsUi->buildType->currentText();
+    if (m_prefsUi->buildType->findText(buildType) == -1) {
+        m_prefsUi->buildType->addItem(buildType);
+    }
+    CMake::setCurrentBuildType( m_project, buildType );
+    CMake::setCurrentExtraArguments( m_project, m_prefsUi->extraArguments->currentText() );
+    CMake::setCurrentCMakeBinary( m_project, Path(m_prefsUi->cMakeBinary->text()) );
+
     qCDebug(CMAKE) << "writing to cmake config: using builddir " << CMake::currentBuildDirIndex(m_project);
     qCDebug(CMAKE) << "writing to cmake config: builddir path " << CMake::currentBuildDir(m_project);
     qCDebug(CMAKE) << "writing to cmake config: installdir " << CMake::currentInstallDir(m_project);
@@ -232,7 +269,7 @@ void CMakePreferences::buildDirChanged(int index)
 {
     CMake::setOverrideBuildDirIndex( m_project, index );
     const Path buildDir = CMake::currentBuildDir(m_project);
-    m_prefsUi->environment->setCurrentProfile( CMake::currentEnvironment( m_project ) );
+    initAdvanced();
     updateCache(buildDir);
     qCDebug(CMAKE) << "builddir Changed" << buildDir;
     emit changed();
diff --git projectmanagers/cmake/settings/cmakepreferences.h projectmanagers/cmake/settings/cmakepreferences.h
index d653927..7bca25e 100644
--- projectmanagers/cmake/settings/cmakepreferences.h
+++ projectmanagers/cmake/settings/cmakepreferences.h
@@ -63,6 +63,7 @@ class CMakePreferences : public KDevelop::ConfigPage
         void configureCacheView();
     private:
         void configure();
+        void initAdvanced();
         
         KDevelop::IProject* m_project;
         KDevelop::Path m_srcFolder;
