diff --git application.cpp application.cpp
index 83e050dc..77a954c5 100644
--- application.cpp
+++ application.cpp
@@ -100,7 +100,7 @@ extern "C" int Q_DECL_EXPORT kdemain( int argc, char **argv )
   aboutData.addAuthor( QStringLiteral("Frerich Raabe"), QString(), QStringLiteral("raabe@kde.org") );
   aboutData.addAuthor( QStringLiteral("Matthias Elter"), i18n("Original Author"), QStringLiteral("me@kde.org") );
   aboutData.addAuthor( QStringLiteral("Wojciech Smigaj"), i18n("Info page support"), QStringLiteral("achu@klub.chip.pl") );
-  app.setWindowIcon( QIcon::fromTheme(QStringLiteral("help-browser")) );
+  app.setWindowIcon( QIcon::fromTheme(QStringLiteral("help-browser"), app.windowIcon()) );
   aboutData.setDesktopFileName ( QStringLiteral("org.kde.Help") );
   KAboutData::setApplicationData(aboutData);
 
