--- orig.CMakeLists.txt	2016-04-19 13:08:48.000000000 +0200
+++ CMakeLists.txt	2016-04-21 15:36:00.000000000 +0200
@@ -11,6 +11,7 @@
 include(KDECMakeSettings)
 include(KDECompilerSettings NO_POLICY_SCOPE)
 include(ECMInstallIcons)
+include(ECMAddAppIcon)
 include(ECMMarkAsTest)
 include(ECMMarkNonGuiExecutable)
 include(ECMOptionalAddSubdirectory)
@@ -87,6 +88,9 @@
 
 kconfig_add_kcfg_files(khelpcenter_KDEINIT_SRCS prefs.kcfgc )
 # kf5_add_app_icon(khelpcenter_KDEINIT_SRCS "${KDE4_INSTALL_DIR}/share/icons/oxygen/*/apps/help-browser.png")
+# Sets the icon on Windows and OSX
+file(GLOB ICONS_SRCS "${CMAKE_CURRENT_BINARY_DIR}/icons/*help-browser.png")
+ecm_add_app_icon(khelpcenter_KDEINIT_SRCS ICONS ${ICONS_SRCS})
 
 kf5_add_kdeinit_executable( khelpcenter ${khelpcenter_KDEINIT_SRCS})
 target_compile_definitions(kdeinit_khelpcenter PRIVATE -DPROJECT_VERSION="${PROJECT_VERSION}")
