--- orig.application.cpp	2016-04-19 13:08:15.000000000 +0200
+++ application.cpp	2016-04-19 21:42:22.674363096 +0200
@@ -90,7 +90,7 @@
   aboutData.addAuthor( "Frerich Raabe", QString(), "raabe@kde.org" );
   aboutData.addAuthor( "Matthias Elter", i18n("Original Author"), "me@kde.org" );
   aboutData.addAuthor( "Wojciech Smigaj", i18n("Info page support"), "achu@klub.chip.pl" );
-  app.setWindowIcon( QIcon::fromTheme("help-browser") );
+  app.setWindowIcon( QIcon::fromTheme("help-browser", app.windowIcon()) );
   KAboutData::setApplicationData(aboutData);
 
   QCommandLineParser *cmd = app.cmdParser();
