--- orig.khelpcenter.kcfg	2016-04-28 15:01:24.000000000 +0200
+++ khelpcenter.kcfg	2016-06-01 21:32:58.000000000 +0200
@@ -71,7 +71,7 @@
   <group name="Info pages">
     <entry type="PathList" key="Search paths">
       <!-- Default paths taken from kio-extras/info/kde-info2html.conf -->
-      <default>/usr/share/info,/usr/info,/usr/lib/info,/usr/local/info,/usr/local/lib/info,/usr/X11R6/info,/usr/X11R6/lib/info,/usr/X11R6/lib/xemacs/info</default>
+      <default>/usr/share/info,/usr/info,/usr/lib/info,/opt/local/share/info,/usr/local/info,/usr/local/lib/info,/usr/X11R6/info,/usr/X11R6/lib/info,/usr/X11R6/lib/xemacs/info</default>
     </entry>
   </group>
 
