--- orig.infotree.cpp	2016-04-19 13:08:15.000000000 +0200
+++ infotree.cpp	2016-04-19 21:44:39.757368810 +0200
@@ -106,8 +106,7 @@
     infoDirFiles << "/usr/share/info";
     infoDirFiles << "/usr/info";
     infoDirFiles << "/usr/lib/info";
-    infoDirFiles << "/usr/local/info";
-    infoDirFiles << "/usr/local/lib/info";
+    infoDirFiles << "/opt/local/share/info";
     infoDirFiles << "/usr/X11R6/info";
     infoDirFiles << "/usr/X11R6/lib/info";
     infoDirFiles << "/usr/X11R6/lib/xemacs/info";
