diff --git application.cpp application.cpp
index 6944f47..74c01a1 100644
--- application.cpp
+++ application.cpp
@@ -101,7 +101,7 @@ extern "C" int Q_DECL_EXPORT kdemain( int argc, char **argv )
   aboutData.addAuthor( "Frerich Raabe", QString(), "raabe@kde.org" );
   aboutData.addAuthor( "Matthias Elter", i18n("Original Author"), "me@kde.org" );
   aboutData.addAuthor( "Wojciech Smigaj", i18n("Info page support"), "achu@klub.chip.pl" );
-  app.setWindowIcon( QIcon::fromTheme("help-browser") );
+  app.setWindowIcon( QIcon::fromTheme("help-browser", app.windowIcon()) );
   aboutData.setDesktopFileName ( "org.kde.Help" );
   KAboutData::setApplicationData(aboutData);
 
