diff --git a/thumbnail/textcreator.cpp b/thumbnail/textcreator.cpp
index a738557477d2a7f08dd7ca6dc0e2f2b8da0c64f6..ae526c4268913b187cb4523ecf2df08a396d828e 100644
--- a/thumbnail/textcreator.cpp
+++ b/thumbnail/textcreator.cpp
@@ -162,11 +162,11 @@ bool TextCreator::create(const QString &path, int width, int height, QImage &img
             textOption.setWrapMode( QTextOption::WrapAtWordBoundaryOrAnywhere );
             textDocument.setDefaultTextOption(textOption);
 
-            KSyntaxHighlighting::SyntaxHighlighter syntaxHighlighter;
+            KSyntaxHighlighting::SyntaxHighlighter syntaxHighlighter(&textDocument);
             syntaxHighlighter.setDefinition(m_highlightingRepository.definitionForFileName(path));
             const auto highlightingTheme = m_highlightingRepository.defaultTheme(KSyntaxHighlighting::Repository::LightTheme);
             syntaxHighlighter.setTheme(highlightingTheme);
-            syntaxHighlighter.setDocument(&textDocument);
+//             syntaxHighlighter.setDocument(&textDocument);
             syntaxHighlighter.rehighlight();
 
             // draw page-in-page, with clipping as needed
