diff --git a/sftp/kio_sftp.cpp b/sftp/kio_sftp.cpp
index 2beff9d77afb2b7f4bc5bf589e253928b71c28b9..f14577389bccd8d91461e6b1394ab7d97915a3f1 100644
--- a/sftp/kio_sftp.cpp
+++ b/sftp/kio_sftp.cpp
@@ -2033,6 +2033,7 @@ sftpProtocol::StatusCode sftpProtocol::sftpCopyGet(const QUrl& url, const QStrin
     if (!mtimeStr.isEmpty()) {
         QDateTime dt = QDateTime::fromString(mtimeStr, Qt::ISODate);
         if (dt.isValid()) {
+#if QT_VERSION >= QT_VERSION_CHECK(5, 10, 0)
             QFile receivedFile(sCopyFile);
             if (receivedFile.exists()) {
                 if (!receivedFile.open(QIODevice::ReadWrite | QIODevice::Text)) {
@@ -2042,6 +2043,14 @@ sftpProtocol::StatusCode sftpProtocol::sftpCopyGet(const QUrl& url, const QStrin
                     receivedFile.setFileTime(dt, QFileDevice::FileModificationTime);
                 }
             }
+#else
+            struct utimbuf utbuf;
+            QT_STATBUF buff;
+            QT_STAT(QFile::encodeName(sCopyFile), &buff);
+            utbuf.actime = buff.st_atime; // access time, unchanged
+            utbuf.modtime = dt.toTime_t(); // modification time
+            utime(QFile::encodeName(sCopyFile), &utbuf);
+#endif
         }
     }
 
