diff --git a/thumbnail/CMakeLists.txt b/thumbnail/CMakeLists.txt
index 966039d0a5890aa2ef80d967372432d198430154..6155dd93ae7bbe81e466cbbc5e8732305d55a465 100644
--- a/thumbnail/CMakeLists.txt
+++ b/thumbnail/CMakeLists.txt
@@ -204,7 +204,7 @@ install(TARGETS kritathumbnail  DESTINATION ${KDE_INSTALL_PLUGINDIR})
 
 ########### next target #################
 
-find_package(Taglib 1.11)
+find_package(TaglibX 1.11)
 
 if(TAGLIB_FOUND)
     set(audiothumbnail_SRCS audiocreator.cpp)
