diff --git a/qtbase/src/plugins/platforms/xcb/qxcbclipboard.cpp b/qtbase/src/plugins/platforms/xcb/qxcbclipboard.cpp
index 01b3bca0d2b75752d4d2fdc1019e9b196778b870..cc8c85206a2c8fac99f1eac08712a6e2a4578b98 100644
--- a/qtbase/src/plugins/platforms/xcb/qxcbclipboard.cpp
+++ b/qtbase/src/plugins/platforms/xcb/qxcbclipboard.cpp
@@ -419,10 +419,20 @@ void QXcbClipboard::setMimeData(QMimeData *data, QClipboard::Mode mode)
         m_timestamp[mode] = connection()->time();
     }
 
-    xcb_set_selection_owner(xcb_connection(), newOwner, modeAtom, connection()->time());
+    xcb_void_cookie_t cookie = xcb_set_selection_owner(xcb_connection(), newOwner, modeAtom, connection()->time());
+    xcb_generic_error_t *error = xcb_request_check(xcb_connection(), cookie);
 
-    if (getSelectionOwner(modeAtom) != newOwner) {
+    if (getSelectionOwner(modeAtom) != newOwner || error) {
         qWarning("QXcbClipboard::setMimeData: Cannot set X11 selection owner");
+        qWarning() << "\tclipboard mode=" << mode << "connection=" << xcb_connection() << "time=" << connection()->time()
+            << "newOwner=" << newOwner << "effective owner:" << getSelectionOwner(modeAtom);
+        if (error) {
+            qWarning() << "\tX11 error" << error->error_code << "opcode" << error->major_code << ":" << error->minor_code;
+            connection()->handleXcbError(error);
+        }
+    }
+    if (error) {
+        free(error);
     }
 
     emitChanged(mode);
