--- builds/unix/orig.ltmain.sh	2017-09-16 19:09:12.000000000 +0200
+++ builds/unix/ltmain.sh	2017-12-31 17:43:13.000000000 +0100
@@ -8879,14 +8879,14 @@
 	  func_arith $current + 1
 	  minor_current=$func_arith_result
 	  xlcverstring="$wl-compatibility_version $wl$minor_current $wl-current_version $wl$minor_current.$revision"
-	  verstring="-compatibility_version $minor_current -current_version $minor_current.$revision"
+	  verstring="-compatibility_version 20 -current_version $minor_current.$revision"
           # On Darwin other compilers
           case $CC in
               nagfor*)
-                  verstring="$wl-compatibility_version $wl$minor_current $wl-current_version $wl$minor_current.$revision"
+                  verstring="$wl-compatibility_version $wl20 $wl-current_version $wl$minor_current.$revision"
                   ;;
               *)
-                  verstring="-compatibility_version $minor_current -current_version $minor_current.$revision"
+                  verstring="-compatibility_version 20 -current_version $minor_current.$revision"
                   ;;
           esac
 	  ;;
