diff --git src/gui/osutils/macutils/AppKitImpl.mm src/gui/osutils/macutils/AppKitImpl.mm
index fbf4a9d0..3a5f1afd 100644
--- src/gui/osutils/macutils/AppKitImpl.mm
+++ src/gui/osutils/macutils/AppKitImpl.mm
@@ -66,7 +66,7 @@
 
 - (void) observeValueForKeyPath:(NSString *)keyPath
                       ofObject:(id)object
-                        change:(NSDictionary<NSKeyValueChangeKey,id> *)change
+                        change:(NSDictionary *)change
                        context:(void *)context
 {
     Q_UNUSED(object)
@@ -79,9 +79,17 @@
                 emit m_appkit->interfaceThemeChanged();
             };
 
-            if(@available(macOS 11.0, *)) {
+            if(
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= 110000
+			  @available(macOS 11.0, *)
+#else
+			  false
+#endif
+		  ) {
+#if __clang_major__ >= 9 && MAC_OS_X_VERSION_MIN_REQUIRED >= 101400
                 // Not sure why exactly this call is needed, but Apple sample code uses it so it's best to use it here too
                 [NSApp.effectiveAppearance performAsCurrentDrawingAppearance:emitBlock];
+#endif
             }
             else {
                 emitBlock();
@@ -139,7 +147,11 @@
 //
 - (bool) isDarkMode
 {
+#if __clang_major__ >= 9 && MAC_OS_X_VERSION_MIN_REQUIRED >= 101400
     return [NSApp.effectiveAppearance.name isEqualToString:NSAppearanceNameDarkAqua];
+#else
+    return false;
+#endif
 }
 
 
@@ -148,7 +160,7 @@
 //
 - (bool) isStatusBarDark
 {
-#if __clang_major__ >= 9 && MAC_OS_X_VERSION_MIN_REQUIRED >= 101000
+#if __clang_major__ >= 9 && MAC_OS_X_VERSION_MIN_REQUIRED >= 101700
     if (@available(macOS 10.17, *)) {
         // This is an ugly hack, but I couldn't find a way to access QTrayIcon's NSStatusItem.
         NSStatusItem* dummy = [[NSStatusBar systemStatusBar] statusItemWithLength:0];
@@ -191,7 +203,7 @@
 //
 - (bool) enableScreenRecording
 {
-#if __clang_major__ >= 9 && MAC_OS_X_VERSION_MIN_REQUIRED >= 1080
+#if __clang_major__ >= 9 && MAC_OS_X_VERSION_MIN_REQUIRED >= 101500
     if (@available(macOS 10.15, *)) {
         // Request screen recording permission on macOS 10.15+
         // This is necessary to get the current window title
