diff --git CMakeLists.txt CMakeLists.txt
index 074c7093..9814ce42 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -57,7 +57,7 @@ option(WITH_XC_YUBIKEY "Include YubiKey support." OFF)
 option(WITH_XC_SSHAGENT "Include SSH agent support." OFF)
 option(WITH_XC_KEESHARE "Sharing integration with KeeShare" OFF)
 option(WITH_XC_UPDATECHECK "Include automatic update checks; disable for controlled distributions" ON)
-if(UNIX AND NOT APPLE)
+if(UNIX)
     option(WITH_XC_FDOSECRETS "Implement freedesktop.org Secret Storage Spec server side API." OFF)
 endif()
 option(WITH_XC_DOCS "Enable building of documentation" ON)
@@ -101,7 +137,7 @@ if(WITH_XC_ALL)
     set(WITH_XC_YUBIKEY ON)
     set(WITH_XC_SSHAGENT ON)
     set(WITH_XC_KEESHARE ON)
-    if(UNIX AND NOT APPLE)
+    if(UNIX)
         set(WITH_XC_FDOSECRETS ON)
     endif()
 endif()
diff --git src/CMakeLists.txt src/CMakeLists.txt
index b8099eed..25e1864b 100644
--- src/CMakeLists.txt
+++ src/CMakeLists.txt
@@ -240,7 +240,7 @@ add_feature_info(SSHAgent WITH_XC_SSHAGENT "SSH agent integration compatible wit
 add_feature_info(KeeShare WITH_XC_KEESHARE "Sharing integration with KeeShare")
 add_feature_info(YubiKey WITH_XC_YUBIKEY "YubiKey HMAC-SHA1 challenge-response")
 add_feature_info(UpdateCheck WITH_XC_UPDATECHECK "Automatic update checking")
-if(UNIX AND NOT APPLE)
+if(UNIX)
     add_feature_info(FdoSecrets WITH_XC_FDOSECRETS "Implement freedesktop.org Secret Storage Spec server side API.")
 endif()
 
