diff --git src/gui/MessageBox.h src/gui/MessageBox.h
index 46939a535a459723414e62aae87e236818917535..9755d4db6808aefd90cc484e36d01255f1045d46 100644
--- src/gui/MessageBox.h
+++ src/gui/MessageBox.h
@@ -22,7 +22,7 @@
 #include <QMap>
 #include <QMessageBox>
 
-class MessageBox
+class Q_DECL_EXPORT MessageBox
 {
 public:
     enum Button : uint64_t
diff --git src/core/Tools.h src/core/Tools.h
index a8094d0a30e5c70db4258fb08fabfbd576f8537d..1bdfca9d6669a7719c7eb6f5ec7a6fcee55c41e4 100644
--- src/core/Tools.h
+++ src/core/Tools.h
@@ -36,7 +36,7 @@ namespace Tools
     bool readAllFromDevice(QIODevice* device, QByteArray& data);
     bool isHex(const QByteArray& ba);
     bool isBase64(const QByteArray& ba);
-    void sleep(int ms);
+    void sleep(int ms) Q_DECL_EXPORT;
     void wait(int ms);
     bool checkUrlValid(const QString& urlField);
     QString uuidToHex(const QUuid& uuid);
diff --git src/gui/osutils/macutils/MacUtils.h src/gui/osutils/macutils/MacUtils.h
index 1281aa07207d115b2f0c5e9f421564edf2927450..26b940a81e6b05299ff860dfa255a373fe64828a 100644
--- src/gui/osutils/macutils/MacUtils.h
+++ src/gui/osutils/macutils/MacUtils.h
@@ -28,7 +28,7 @@
 #include <QScopedPointer>
 #include <qwindowdefs.h>
 
-class MacUtils : public OSUtilsBase
+class Q_DECL_EXPORT MacUtils : public OSUtilsBase
 {
     Q_OBJECT
 
diff --git src/autotype/mac/CMakeLists.txt src/autotype/mac/CMakeLists.txt
index e0df901fdd10232a52efa1e7f047ad933bb951a9..b8248b3a90ea62497ed50aa106d2182fd2cec9fd 100644
--- src/autotype/mac/CMakeLists.txt
+++ src/autotype/mac/CMakeLists.txt
@@ -1,8 +1,8 @@
 set(autotype_mac_SOURCES AutoTypeMac.cpp)
 
 add_library(keepassxc-autotype-cocoa MODULE ${autotype_mac_SOURCES})
-set_target_properties(keepassxc-autotype-cocoa PROPERTIES LINK_FLAGS "-framework Foundation -framework AppKit -framework Carbon")
-target_link_libraries(keepassxc-autotype-cocoa ${PROGNAME} Qt5::Core Qt5::Widgets)
+set_target_properties(keepassxc-autotype-cocoa PROPERTIES LINK_FLAGS "-undefined dynamic_lookup -framework Foundation -framework AppKit -framework Carbon")
+target_link_libraries(keepassxc-autotype-cocoa Qt5::Core Qt5::Widgets)
 
 install(TARGETS keepassxc-autotype-cocoa
         LIBRARY DESTINATION ${PLUGIN_INSTALL_DIR} COMPONENT Runtime)
