diff --git src/orig.meson.build src/meson.build
index 3f89f49..662f9cf 100644
--- src/orig.meson.build
+++ src/meson.build
@@ -303,6 +303,12 @@ if get_option('gtk3')
     libvte_gtk3_sources += a11y_sources
   endif
 
+  compat_version = '@0@.@1@.@2@'.format(
+    vte_minor_version * 100,
+    0, 0)
+  current_version = '@0@.@1@.@2@'.format(
+    vte_minor_version * 100 + vte_micro_version,
+    vte_micro_version, 0)
   libvte_gtk3 = shared_library(
     vte_gtk3_api_name,
     sources: libvte_gtk3_sources,
@@ -311,6 +317,7 @@ if get_option('gtk3')
     dependencies: libvte_gtk3_deps,
     cpp_args: libvte_gtk3_cppflags,
     install: true,
+    darwin_versions: [compat_version, current_version],
   )
 
   libvte_gtk3_dep = declare_dependency(
