--- lang/qt/src/orig.dn.cpp	2016-11-03 08:55:36.000000000 +0100
+++ lang/qt/src/dn.cpp	2016-12-20 21:21:17.000000000 +0100
@@ -37,6 +37,8 @@
 
 #include "dn.h"
 
+#include <strings.h>
+
 static const struct {
     const char *name;
     const char *oid;
@@ -165,7 +167,7 @@
     for (unsigned int i = 0; i < numOidMaps; ++i)
         if (!strcasecmp((char *)p, oidmap[i].oid)) {
             free(p);
-            p = strdup(oidmap[i].name);
+            p = qstrdup(oidmap[i].name);
             break;
         }
     array->key = p;
--- lang/qt/tests/orig.t-support.h	2016-10-18 19:22:02.000000000 +0200
+++ lang/qt/tests/t-support.h	2016-12-20 21:20:47.000000000 +0100
@@ -42,7 +42,7 @@
     char *getPassphrase(const char * /*useridHint*/, const char * /*description*/,
                         bool /*previousWasBad*/, bool &/*canceled*/) Q_DECL_OVERRIDE
     {
-        return strdup("abc");
+        return qstrdup("abc");
     }
 };
 } // namespace GpgME
