diff --git configure.ac configure.ac
index 4495093..7109194 100644
--- configure.ac
+++ configure.ac
@@ -182,6 +182,7 @@ have_dosish_system=no
 have_android_system=no
 have_w32_system=no
 have_w64_system=no
+have_macos_system=no
 build_w32_glib=no
 build_w32_qt=no
 available_languages="cl cpp python python2 python3 qt"
@@ -196,6 +197,9 @@ case "${host}" in
     *-linux-androideabi)
         have_android_system=yes
         ;;
+    *-apple-darwin*)
+        have_macos_system=yes
+        ;;
 esac
 case "${host}" in
     *-mingw32ce*|*-mingw32*)
@@ -253,6 +257,12 @@ if test "$have_android_system" = yes; then
 fi
 AM_CONDITIONAL(HAVE_ANDROID_SYSTEM, test "$have_android_system" = yes)
 
+if test "$have_macos_system" = yes; then
+   AC_DEFINE(HAVE_MACOS_SYSTEM,1,
+             [Defined if we build for an MacOS system])
+fi
+AM_CONDITIONAL(HAVE_MACOS_SYSTEM, test "$have_macos_system" = yes)
+
 AM_CONDITIONAL(BUILD_W32_GLIB, test "$build_w32_glib" = yes)
 
 
diff --git lang/cpp/src/Makefile.am lang/cpp/src/Makefile.am
index 92ed784..c62bd6c 100644
--- lang/cpp/src/Makefile.am
+++ lang/cpp/src/Makefile.am
@@ -69,6 +69,12 @@ libgpgmepp_la_LIBADD = ../../../src/libgpgme.la @LIBASSUAN_LIBS@
 libgpgmepp_la_LDFLAGS = -no-undefined -version-info \
     @LIBGPGMEPP_LT_CURRENT@:@LIBGPGMEPP_LT_REVISION@:@LIBGPGMEPP_LT_AGE@
 
+if HAVE_MACOS_SYSTEM
+libsuffix=.dylib
+else
+libsuffix=.so
+endif
+
 if HAVE_W32_SYSTEM
 GpgmeppConfig.cmake: GpgmeppConfig-w32.cmake.in
 	sed -e 's|[@]resolved_bindir@|$(bindir)|g' < "$<" | \
@@ -77,6 +83,7 @@ GpgmeppConfig.cmake: GpgmeppConfig-w32.cmake.in
 else
 GpgmeppConfig.cmake: GpgmeppConfig.cmake.in
 	sed -e 's|[@]resolved_libdir@|$(libdir)|g' < "$<" | \
+	sed -e 's|[@]libsuffix@|$(libsuffix)|g' | \
 	sed -e 's|[@]resolved_includedir@|$(includedir)|g' > $@
 endif
 install-cmake-files: GpgmeppConfig.cmake GpgmeppConfigVersion.cmake
diff --git lang/qt/src/Makefile.am lang/qt/src/Makefile.am
index 87e2ec2..55af1cf 100644
--- lang/qt/src/Makefile.am
+++ lang/qt/src/Makefile.am
@@ -220,6 +220,12 @@ libqgpgme_la_LIBADD = ../../cpp/src/libgpgmepp.la ../../../src/libgpgme.la \
 libqgpgme_la_LDFLAGS = -no-undefined -version-info \
     @LIBQGPGME_LT_CURRENT@:@LIBQGPGME_LT_REVISION@:@LIBQGPGME_LT_AGE@
 
+if HAVE_MACOS_SYSTEM
+libsuffix=.dylib
+else
+libsuffix=.so
+endif
+
 if HAVE_W32_SYSTEM
 QGpgmeConfig.cmake: QGpgmeConfig-w32.cmake.in
 	sed -e 's|[@]resolved_bindir@|$(bindir)|g' < "$<" | \
@@ -228,6 +234,7 @@ QGpgmeConfig.cmake: QGpgmeConfig-w32.cmake.in
 else
 QGpgmeConfig.cmake: QGpgmeConfig.cmake.in
 	sed -e 's|[@]resolved_libdir@|$(libdir)|g' < "$<" | \
+	sed -e 's|[@]libsuffix@|$(libsuffix)|g' | \
 	sed -e 's|[@]resolved_includedir@|$(includedir)|g' > $@
 endif
 
diff --git lang/qt/src/QGpgmeConfig.cmake.in.in lang/qt/src/QGpgmeConfig.cmake.in.in
index 88ed242..a17a19f 100644
--- lang/qt/src/QGpgmeConfig.cmake.in.in
+++ lang/qt/src/QGpgmeConfig.cmake.in.in
@@ -64,7 +64,7 @@ add_library(QGpgme SHARED IMPORTED)
 set_target_properties(QGpgme PROPERTIES
   INTERFACE_INCLUDE_DIRECTORIES "@resolved_includedir@/qgpgme;@resolved_includedir@"
   INTERFACE_LINK_LIBRARIES "Gpgmepp;Qt5::Core"
-  IMPORTED_LOCATION "@resolved_libdir@/libqgpgme.so"
+  IMPORTED_LOCATION "@resolved_libdir@/libqgpgme@libsuffix@"
 )
 
 if(CMAKE_VERSION VERSION_LESS 2.8.12)
diff --git lang/cpp/src/GpgmeppConfig.cmake.in.in lang/cpp/src/GpgmeppConfig.cmake.in.in
index cbe9713..7f42f31 100644
--- lang/cpp/src/GpgmeppConfig.cmake.in.in
+++ lang/cpp/src/GpgmeppConfig.cmake.in.in
--- lang/cpp/src/orig.GpgmeppConfig.cmake.in.in	2016-11-16 10:17:20.000000000 +0100
+++ lang/cpp/src/GpgmeppConfig.cmake.in.in	2017-01-02 16:26:40.000000000 +0100
@@ -64,7 +64,7 @@
 set_target_properties(Gpgmepp PROPERTIES
   INTERFACE_INCLUDE_DIRECTORIES "@resolved_includedir@/gpgme++;@resolved_includedir@"
   INTERFACE_LINK_LIBRARIES "pthread;@resolved_libdir@/libgpgme@libsuffix@;@LIBASSUAN_LIBS@"
-  IMPORTED_LOCATION "@resolved_libdir@/libgpgmepp.so"
+  IMPORTED_LOCATION "@resolved_libdir@/libgpgmepp@libsuffix@"
 )
 
 if(CMAKE_VERSION VERSION_LESS 2.8.12)
