--- orig.configure	2018-03-30 13:24:14.000000000 +0200
+++ configure	2018-03-30 13:35:22.000000000 +0200
@@ -22958,7 +22958,6 @@
 
 ac_config_files="$ac_config_files lang/qt/src/QGpgmeConfigVersion.cmake"
 
-ac_config_files="$ac_config_files lang/qt/tests/Makefile"
 
 ac_config_files="$ac_config_files lang/qt/src/qgpgme_version.h"
 
@@ -24264,7 +24263,6 @@
     "lang/qt/src/QGpgmeConfig-w32.cmake.in") CONFIG_FILES="$CONFIG_FILES lang/qt/src/QGpgmeConfig-w32.cmake.in" ;;
     "lang/qt/src/QGpgmeConfig.cmake.in") CONFIG_FILES="$CONFIG_FILES lang/qt/src/QGpgmeConfig.cmake.in" ;;
     "lang/qt/src/QGpgmeConfigVersion.cmake") CONFIG_FILES="$CONFIG_FILES lang/qt/src/QGpgmeConfigVersion.cmake" ;;
-    "lang/qt/tests/Makefile") CONFIG_FILES="$CONFIG_FILES lang/qt/tests/Makefile" ;;
     "lang/qt/src/qgpgme_version.h") CONFIG_FILES="$CONFIG_FILES lang/qt/src/qgpgme_version.h" ;;
     "lang/Makefile") CONFIG_FILES="$CONFIG_FILES lang/Makefile" ;;
     "lang/cl/Makefile") CONFIG_FILES="$CONFIG_FILES lang/cl/Makefile" ;;
--- lang/qt/orig.Makefile.am	2017-07-12 14:35:02.000000000 +0200
+++ lang/qt/Makefile.am	2018-03-30 13:37:44.000000000 +0200
@@ -19,6 +19,6 @@
 # Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
 # 02111-1307, USA
 
-SUBDIRS = src tests doc
+SUBDIRS = src doc
 
 EXTRA_DIST = README
--- lang/qt/orig.Makefile.in	2018-03-30 13:24:17.000000000 +0200
+++ lang/qt/Makefile.in	2018-03-30 13:37:53.000000000 +0200
@@ -406,7 +406,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = src tests doc
+SUBDIRS = src doc
 EXTRA_DIST = README
 all: all-recursive
 
