diff --git lang/qt/src/Makefile.am lang/qt/src/Makefile.am
index c15da18..d3791cd 100644
--- lang/qt/src/Makefile.am
+++ lang/qt/src/Makefile.am
@@ -17,7 +17,7 @@
 # along with this program; if not, write to the Free Software
 # Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
 # 02111-1307, USA
-lib_LTLIBRARIES = libqgpgme.la
+lib_LTLIBRARIES = libqt5gpgme.la
 EXTRA_DIST = QGpgmeConfig.cmake.in.in QGpgmeConfigVersion.cmake.in \
              gpgme_backend_debug.h qgpgme_version.h.in
 
@@ -203,15 +203,15 @@ camelcaseincludedir = $(includedir)/QGpgME
 camelcaseinclude_HEADERS = $(camelcase_headers)
 nodist_include_HEADERS = qgpgme_version.h
 
-libqgpgme_la_SOURCES = $(qgpgme_sources) $(qgpgme_headers) $(private_qgpgme_headers)
+libqt5gpgme_la_SOURCES = $(qgpgme_sources) $(qgpgme_headers) $(private_qgpgme_headers)
 
 AM_CPPFLAGS = -I$(top_srcdir)/lang/cpp/src -I$(top_builddir)/src \
               @GPGME_QT_CFLAGS@ @GPG_ERROR_CFLAGS@ @LIBASSUAN_CFLAGS@ \
               -DBUILDING_QGPGME
 
-libqgpgme_la_LIBADD = ../../cpp/src/libgpgmepp.la ../../../src/libgpgme.la \
+libqt5gpgme_la_LIBADD = ../../cpp/src/libgpgmepp.la ../../../src/libgpgme.la \
                        @LIBASSUAN_LIBS@ @GPGME_QT_LIBS@
-libqgpgme_la_LDFLAGS = -version-info \
+libqt5gpgme_la_LDFLAGS = -version-info \
     @LIBQGPGME_LT_CURRENT@:@LIBQGPGME_LT_REVISION@:@LIBQGPGME_LT_AGE@
 
 if HAVE_W32_SYSTEM
@@ -250,7 +250,7 @@ BUILT_SOURCES = $(qgpgme_moc_sources) $(camelcase_headers)
 
 CLEANFILES = $(qgpgme_moc_sources) $(camelcase_headers) QGpgmeConfig.cmake
 
-nodist_libqgpgme_la_SOURCES = $(qgpgme_moc_sources)
+nodist_libqt5gpgme_la_SOURCES = $(qgpgme_moc_sources)
 
 .h.moc:
 	$(MOC) `test -f '$<' || echo '$(srcdir)/'`$< -o $@
diff --git lang/qt/src/Makefile.in lang/qt/src/Makefile.in
index d10eca3..1183b03 100644
--- lang/qt/src/Makefile.in
+++ lang/qt/src/Makefile.in
@@ -136,7 +136,7 @@ am__installdirs = "$(DESTDIR)$(libdir)" \
 	"$(DESTDIR)$(camelcaseincludedir)" "$(DESTDIR)$(includedir)" \
 	"$(DESTDIR)$(qgpgmeincludedir)"
 LTLIBRARIES = $(lib_LTLIBRARIES)
-libqgpgme_la_DEPENDENCIES = ../../cpp/src/libgpgmepp.la \
+libqt5gpgme_la_DEPENDENCIES = ../../cpp/src/libgpgmepp.la \
 	../../../src/libgpgme.la
 am__objects_1 = dataprovider.lo job.lo qgpgmeadduseridjob.lo \
 	qgpgmebackend.lo qgpgmechangeexpiryjob.lo \
@@ -154,18 +154,18 @@ am__objects_1 = dataprovider.lo job.lo qgpgmeadduseridjob.lo \
 	gpgme_backend_debug.lo qgpgmetofupolicyjob.lo \
 	defaultkeygenerationjob.lo qgpgmewkspublishjob.lo
 am__objects_2 =
-am_libqgpgme_la_OBJECTS = $(am__objects_1) $(am__objects_2) \
+am_libqt5gpgme_la_OBJECTS = $(am__objects_1) $(am__objects_2) \
 	$(am__objects_2)
-nodist_libqgpgme_la_OBJECTS = $(am__objects_2)
-libqgpgme_la_OBJECTS = $(am_libqgpgme_la_OBJECTS) \
-	$(nodist_libqgpgme_la_OBJECTS)
+nodist_libqt5gpgme_la_OBJECTS = $(am__objects_2)
+libqt5gpgme_la_OBJECTS = $(am_libqt5gpgme_la_OBJECTS) \
+	$(nodist_libqt5gpgme_la_OBJECTS)
 AM_V_lt = $(am__v_lt_@AM_V@)
 am__v_lt_ = $(am__v_lt_@AM_DEFAULT_V@)
 am__v_lt_0 = --silent
 am__v_lt_1 = 
-libqgpgme_la_LINK = $(LIBTOOL) $(AM_V_lt) --tag=CXX $(AM_LIBTOOLFLAGS) \
+libqt5gpgme_la_LINK = $(LIBTOOL) $(AM_V_lt) --tag=CXX $(AM_LIBTOOLFLAGS) \
 	$(LIBTOOLFLAGS) --mode=link $(CXXLD) $(AM_CXXFLAGS) \
-	$(CXXFLAGS) $(libqgpgme_la_LDFLAGS) $(LDFLAGS) -o $@
+	$(CXXFLAGS) $(libqt5gpgme_la_LDFLAGS) $(LDFLAGS) -o $@
 AM_V_P = $(am__v_P_@AM_V@)
 am__v_P_ = $(am__v_P_@AM_DEFAULT_V@)
 am__v_P_0 = false
@@ -218,8 +218,8 @@ AM_V_CCLD = $(am__v_CCLD_@AM_V@)
 am__v_CCLD_ = $(am__v_CCLD_@AM_DEFAULT_V@)
 am__v_CCLD_0 = @echo "  CCLD    " $@;
 am__v_CCLD_1 = 
-SOURCES = $(libqgpgme_la_SOURCES) $(nodist_libqgpgme_la_SOURCES)
-DIST_SOURCES = $(libqgpgme_la_SOURCES)
+SOURCES = $(libqt5gpgme_la_SOURCES) $(nodist_libqt5gpgme_la_SOURCES)
+DIST_SOURCES = $(libqt5gpgme_la_SOURCES)
 am__can_run_installinfo = \
   case $$AM_UPDATE_INFO_DIR in \
     n|no|NO) false;; \
@@ -459,7 +459,7 @@ top_srcdir = @top_srcdir@
 # along with this program; if not, write to the Free Software
 # Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
 # 02111-1307, USA
-lib_LTLIBRARIES = libqgpgme.la
+lib_LTLIBRARIES = libqt5gpgme.la
 EXTRA_DIST = QGpgmeConfig.cmake.in.in QGpgmeConfigVersion.cmake.in \
              gpgme_backend_debug.h qgpgme_version.h.in
 
@@ -645,22 +645,22 @@ qgpgmeinclude_HEADERS = $(qgpgme_headers)
 camelcaseincludedir = $(includedir)/QGpgME
 camelcaseinclude_HEADERS = $(camelcase_headers)
 nodist_include_HEADERS = qgpgme_version.h
-libqgpgme_la_SOURCES = $(qgpgme_sources) $(qgpgme_headers) $(private_qgpgme_headers)
+libqt5gpgme_la_SOURCES = $(qgpgme_sources) $(qgpgme_headers) $(private_qgpgme_headers)
 AM_CPPFLAGS = -I$(top_srcdir)/lang/cpp/src -I$(top_builddir)/src \
               @GPGME_QT_CFLAGS@ @GPG_ERROR_CFLAGS@ @LIBASSUAN_CFLAGS@ \
               -DBUILDING_QGPGME
 
-libqgpgme_la_LIBADD = ../../cpp/src/libgpgmepp.la ../../../src/libgpgme.la \
+libqt5gpgme_la_LIBADD = ../../cpp/src/libgpgmepp.la ../../../src/libgpgme.la \
                        @LIBASSUAN_LIBS@ @GPGME_QT_LIBS@
 
-libqgpgme_la_LDFLAGS = -version-info \
+libqt5gpgme_la_LDFLAGS = -version-info \
     @LIBQGPGME_LT_CURRENT@:@LIBQGPGME_LT_REVISION@:@LIBQGPGME_LT_AGE@
 
 @HAVE_W32_SYSTEM_FALSE@libsuffix = .so
 @HAVE_W32_SYSTEM_TRUE@libsuffix = .dll.a
 BUILT_SOURCES = $(qgpgme_moc_sources) $(camelcase_headers)
 CLEANFILES = $(qgpgme_moc_sources) $(camelcase_headers) QGpgmeConfig.cmake
-nodist_libqgpgme_la_SOURCES = $(qgpgme_moc_sources)
+nodist_libqt5gpgme_la_SOURCES = $(qgpgme_moc_sources)
 all: $(BUILT_SOURCES)
 	$(MAKE) $(AM_MAKEFLAGS) all-am
 
@@ -738,8 +738,8 @@ clean-libLTLIBRARIES:
 	  rm -f $${locs}; \
 	}
 
-libqgpgme.la: $(libqgpgme_la_OBJECTS) $(libqgpgme_la_DEPENDENCIES) $(EXTRA_libqgpgme_la_DEPENDENCIES) 
-	$(AM_V_CXXLD)$(libqgpgme_la_LINK) -rpath $(libdir) $(libqgpgme_la_OBJECTS) $(libqgpgme_la_LIBADD) $(LIBS)
+libqt5gpgme.la: $(libqt5gpgme_la_OBJECTS) $(libqt5gpgme_la_DEPENDENCIES) $(EXTRA_libqt5gpgme_la_DEPENDENCIES) 
+	$(AM_V_CXXLD)$(libqt5gpgme_la_LINK) -rpath $(libdir) $(libqt5gpgme_la_OBJECTS) $(libqt5gpgme_la_LIBADD) $(LIBS)
 
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT)
diff --git lang/qt/src/QGpgmeConfig.cmake.in.in lang/qt/src/QGpgmeConfig.cmake.in.in
index 3d19696..33cccb9 100644
--- lang/qt/src/QGpgmeConfig.cmake.in.in
+++ lang/qt/src/QGpgmeConfig.cmake.in.in
@@ -70,7 +70,7 @@ add_library(QGpgme SHARED IMPORTED)
 set_target_properties(QGpgme PROPERTIES
   INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include;${_IMPORT_PREFIX}/include/qgpgme"
   INTERFACE_LINK_LIBRARIES "Gpgmepp;Qt5::Core"
-  IMPORTED_LOCATION "@resolved_libdir@/libqgpgme@libsuffix@"
+  IMPORTED_LOCATION "@resolved_libdir@/libqt5gpgme@libsuffix@"
 )
 
 if(CMAKE_VERSION VERSION_LESS 2.8.12)
diff --git lang/qt/tests/Makefile.am lang/qt/tests/Makefile.am
index f243a55..81ca9ce 100644
--- lang/qt/tests/Makefile.am
+++ lang/qt/tests/Makefile.am
@@ -32,7 +32,7 @@ moc_files = t-keylist.moc t-keylocate.moc t-ownertrust.moc t-tofuinfo.moc \
 
 AM_LDFLAGS = -no-install
 
-LDADD = ../../cpp/src/libgpgmepp.la ../src/libqgpgme.la \
+LDADD = ../../cpp/src/libgpgmepp.la ../src/libqt5gpgme.la \
         ../../../src/libgpgme.la @GPGME_QT_LIBS@ @GPG_ERROR_LIBS@ \
         @GPGME_QTTEST_LIBS@ -lstdc++
 
diff --git lang/qt/tests/Makefile.in lang/qt/tests/Makefile.in
index 45d4c76..01f9de6 100644
--- lang/qt/tests/Makefile.in
+++ lang/qt/tests/Makefile.in
@@ -129,7 +129,7 @@ am_run_keyformailboxjob_OBJECTS = run-keyformailboxjob.$(OBJEXT)
 run_keyformailboxjob_OBJECTS = $(am_run_keyformailboxjob_OBJECTS)
 run_keyformailboxjob_LDADD = $(LDADD)
 run_keyformailboxjob_DEPENDENCIES = ../../cpp/src/libgpgmepp.la \
-	../src/libqgpgme.la ../../../src/libgpgme.la
+	../src/libqt5gpgme.la ../../../src/libgpgme.la
 AM_V_lt = $(am__v_lt_@AM_V@)
 am__v_lt_ = $(am__v_lt_@AM_DEFAULT_V@)
 am__v_lt_0 = --silent
@@ -139,7 +139,7 @@ am_t_encrypt_OBJECTS = t-encrypt.$(OBJEXT) $(am__objects_1)
 t_encrypt_OBJECTS = $(am_t_encrypt_OBJECTS)
 t_encrypt_LDADD = $(LDADD)
 t_encrypt_DEPENDENCIES = ../../cpp/src/libgpgmepp.la \
-	../src/libqgpgme.la ../../../src/libgpgme.la
+	../src/libqt5gpgme.la ../../../src/libgpgme.la
 am_t_keylist_OBJECTS = t-keylist.$(OBJEXT) $(am__objects_1)
 am__objects_2 =
 nodist_t_keylist_OBJECTS = $(am__objects_2)
@@ -147,27 +147,27 @@ t_keylist_OBJECTS = $(am_t_keylist_OBJECTS) \
 	$(nodist_t_keylist_OBJECTS)
 t_keylist_LDADD = $(LDADD)
 t_keylist_DEPENDENCIES = ../../cpp/src/libgpgmepp.la \
-	../src/libqgpgme.la ../../../src/libgpgme.la
+	../src/libqt5gpgme.la ../../../src/libgpgme.la
 am_t_keylocate_OBJECTS = t-keylocate.$(OBJEXT) $(am__objects_1)
 t_keylocate_OBJECTS = $(am_t_keylocate_OBJECTS)
 t_keylocate_LDADD = $(LDADD)
 t_keylocate_DEPENDENCIES = ../../cpp/src/libgpgmepp.la \
-	../src/libqgpgme.la ../../../src/libgpgme.la
+	../src/libqt5gpgme.la ../../../src/libgpgme.la
 am_t_ownertrust_OBJECTS = t-ownertrust.$(OBJEXT) $(am__objects_1)
 t_ownertrust_OBJECTS = $(am_t_ownertrust_OBJECTS)
 t_ownertrust_LDADD = $(LDADD)
 t_ownertrust_DEPENDENCIES = ../../cpp/src/libgpgmepp.la \
-	../src/libqgpgme.la ../../../src/libgpgme.la
+	../src/libqt5gpgme.la ../../../src/libgpgme.la
 am_t_tofuinfo_OBJECTS = t-tofuinfo.$(OBJEXT) $(am__objects_1)
 t_tofuinfo_OBJECTS = $(am_t_tofuinfo_OBJECTS)
 t_tofuinfo_LDADD = $(LDADD)
 t_tofuinfo_DEPENDENCIES = ../../cpp/src/libgpgmepp.la \
-	../src/libqgpgme.la ../../../src/libgpgme.la
+	../src/libqt5gpgme.la ../../../src/libgpgme.la
 am_t_wkspublish_OBJECTS = t-wkspublish.$(OBJEXT) $(am__objects_1)
 t_wkspublish_OBJECTS = $(am_t_wkspublish_OBJECTS)
 t_wkspublish_LDADD = $(LDADD)
 t_wkspublish_DEPENDENCIES = ../../cpp/src/libgpgmepp.la \
-	../src/libqgpgme.la ../../../src/libgpgme.la
+	../src/libqt5gpgme.la ../../../src/libgpgme.la
 AM_V_P = $(am__v_P_@AM_V@)
 am__v_P_ = $(am__v_P_@AM_DEFAULT_V@)
 am__v_P_0 = false
@@ -474,7 +474,7 @@ moc_files = t-keylist.moc t-keylocate.moc t-ownertrust.moc t-tofuinfo.moc \
             t-encrypt.moc t-support.hmoc t-wkspublish.moc
 
 AM_LDFLAGS = -no-install
-LDADD = ../../cpp/src/libgpgmepp.la ../src/libqgpgme.la \
+LDADD = ../../cpp/src/libgpgmepp.la ../src/libqt5gpgme.la \
         ../../../src/libgpgme.la @GPGME_QT_LIBS@ @GPG_ERROR_LIBS@ \
         @GPGME_QTTEST_LIBS@ -lstdc++
 
