diff --git plasma/CMakeLists.txt plasma/CMakeLists.txt
index 199dbb0..2175e39 100644
--- plasma/CMakeLists.txt
+++ plasma/CMakeLists.txt
@@ -12,7 +12,9 @@ ENDMACRO(installPackage)
 if(${KDE_PLATFORM_PROFILE} STREQUAL "Desktop")
    add_subdirectory(desktop)
    add_subdirectory(netbook)
-   add_subdirectory(screensaver)
+    if(NOT APPLE)
+       add_subdirectory(screensaver)
+    endif(NOT APPLE)
 endif(${KDE_PLATFORM_PROFILE} STREQUAL "Desktop")
 
 add_subdirectory(generic)
diff --git plasma/desktop/CMakeLists.txt plasma/desktop/CMakeLists.txt
index 2de78dd..fb612e8 100644
--- plasma/desktop/CMakeLists.txt
+++ plasma/desktop/CMakeLists.txt
@@ -1,5 +1,7 @@
 add_subdirectory(applets)
 add_subdirectory(containments)
 add_subdirectory(runners)
-add_subdirectory(shell)
+#if(NOT APPLE)
+    add_subdirectory(shell)
+#endif(NOT APPLE)
 add_subdirectory(toolboxes)
diff --git plasma/desktop/applets/CMakeLists.txt plasma/desktop/applets/CMakeLists.txt
index 6f80cec..5202f83 100644
--- plasma/desktop/applets/CMakeLists.txt
+++ plasma/desktop/applets/CMakeLists.txt
@@ -2,10 +2,10 @@ add_subdirectory(kickoff)
 add_subdirectory(showActivityManager)
 add_subdirectory(trash)
 
-if(NOT WIN32)
+if(NOT WIN32 AND NOT APPLE)
     #task and windowlist depend on libs/taskmanager
    add_subdirectory(tasks)
    add_subdirectory(windowlist)
 
    add_subdirectory(pager)
-endif(NOT WIN32)
+endif(NOT WIN32 AND NOT APPLE)
diff --git plasma/desktop/shell/desktopview.cpp plasma/desktop/shell/desktopview.cpp
index f0f0371..cfce13c 100644
--- plasma/desktop/shell/desktopview.cpp
+++ plasma/desktop/shell/desktopview.cpp
@@ -314,7 +314,7 @@ void DesktopView::toolBoxOpened(bool open)
         return;
     }
 
-#ifndef Q_WS_WIN
+#ifdef Q_WS_X11
     NETRootInfo info(QX11Info::display(), NET::Supported);
     if (!info.isSupported(NET::WM2ShowingDesktop)) {
         return;
diff --git plasma/desktop/shell/plasmaapp.cpp plasma/desktop/shell/plasmaapp.cpp
index c1294a1..49846fb 100644
--- plasma/desktop/shell/plasmaapp.cpp
+++ plasma/desktop/shell/plasmaapp.cpp
@@ -31,7 +31,7 @@
 #include <unistd.h>
 
 #ifndef _SC_PHYS_PAGES
-    #ifdef Q_OS_FREEBSD
+    #if defined(Q_OS_FREEBSD) || defined(Q_OS_MAC)
     #include <sys/types.h>
     #include <sys/sysctl.h>
     #endif
@@ -216,7 +216,7 @@ PlasmaApp::PlasmaApp()
     int memorySize = sysconf(_SC_PHYS_PAGES);
     memorySize *= sysconf(_SC_PAGESIZE) / 1024;
 #else
-#ifdef Q_OS_FREEBSD
+#if defined(Q_OS_FREEBSD) || defined(Q_OS_MAC)
     int sysctlbuf[2];
     size_t size = sizeof(sysctlbuf);
     int memorySize;
diff --git plasma/generic/CMakeLists.txt plasma/generic/CMakeLists.txt
index cfaf14f..294faaf 100644
--- plasma/generic/CMakeLists.txt
+++ plasma/generic/CMakeLists.txt
@@ -7,9 +7,11 @@ if(${KDE_PLATFORM_PROFILE} STREQUAL "Desktop")
 endif(${KDE_PLATFORM_PROFILE} STREQUAL "Desktop")
 
 add_subdirectory(applets)
-add_subdirectory(dataengines)
+# add_subdirectory(dataengines)
 add_subdirectory(runners)
 add_subdirectory(scriptengines)
 add_subdirectory(shells)
-add_subdirectory(wallpapers)
+if(NOT APPLE)
+    add_subdirectory(wallpapers)
+endif(NOT APPLE)
 
diff --git plasma/generic/applets/CMakeLists.txt plasma/generic/applets/CMakeLists.txt
index 2b888ee..0e1131d 100644
--- plasma/generic/applets/CMakeLists.txt
+++ plasma/generic/applets/CMakeLists.txt
@@ -16,7 +16,7 @@ if(QT_QTWEBKIT_FOUND)
    add_subdirectory(webbrowser)
 endif(QT_QTWEBKIT_FOUND)
 
-if(NOT WIN32)
+if(NOT WIN32 AND NOT APPLE)
    add_subdirectory(system-monitor)
    #notifications
    #should compile also on windows? (even if doesn't really make sense)
@@ -24,4 +24,4 @@ if(NOT WIN32)
    if(${KDE_PLATFORM_PROFILE} STREQUAL "Desktop")
       add_subdirectory(systemtray)
    endif(${KDE_PLATFORM_PROFILE} STREQUAL "Desktop")
-endif(NOT WIN32)
+endif(NOT WIN32 AND NOT APPLE)
diff --git plasma/generic/dataengines/CMakeLists.txt plasma/generic/dataengines/CMakeLists.txt
index d240683..35a7a10 100644
--- plasma/generic/dataengines/CMakeLists.txt
+++ plasma/generic/dataengines/CMakeLists.txt
@@ -34,11 +34,11 @@ if(Boost_FOUND AND KdepimLibs_FOUND)
    add_subdirectory(rss)
 endif(Boost_FOUND AND KdepimLibs_FOUND)
 
-if(NOT WIN32)
+if(NOT WIN32 AND NOT APPLE)
    add_subdirectory(mouse)
    add_subdirectory(systemmonitor)
    if(${KDE_PLATFORM_PROFILE} STREQUAL "Desktop")
       add_subdirectory(tasks)
    endif(${KDE_PLATFORM_PROFILE} STREQUAL "Desktop")
-endif(NOT WIN32)
+endif(NOT WIN32 AND NOT APPLE)
 
diff --git plasma/generic/runners/CMakeLists.txt plasma/generic/runners/CMakeLists.txt
index 6831ac0..963fe59 100644
--- plasma/generic/runners/CMakeLists.txt
+++ plasma/generic/runners/CMakeLists.txt
@@ -14,10 +14,10 @@ if(NepomukCore_FOUND)
   add_subdirectory(nepomuksearch)
 endif(NepomukCore_FOUND)
 
-if(NOT WIN32)
+if(NOT WIN32 AND NOT APPLE)
 add_subdirectory(powerdevil)
 add_subdirectory(sessions)
 add_subdirectory(windows)
 add_subdirectory(kill)
-endif(NOT WIN32)
+endif(NOT WIN32 AND NOT APPLE)
 
diff --git plasma/generic/shells/plasma-windowed/CMakeLists.txt plasma/generic/shells/plasma-windowed/CMakeLists.txt
index 86b7770..74b9641 100644
--- plasma/generic/shells/plasma-windowed/CMakeLists.txt
+++ plasma/generic/shells/plasma-windowed/CMakeLists.txt
@@ -10,6 +10,11 @@ set(plasma-windowed_SRCS
 
 
 kde4_add_kdeinit_executable(plasma-windowed ${plasma-windowed_SRCS})
+if (Q_WS_MAC)
+    set_target_properties(plasma-windowed PROPERTIES MACOSX_BUNDLE_INFO_PLIST ${CMAKE_CURRENT_SOURCE_DIR}/Info.plist.template)
+    set_target_properties(plasma-windowed PROPERTIES MACOSX_BUNDLE_GUI_IDENTIFIER "org.kde.plasma.windowed")
+    set_target_properties(plasma-windowed PROPERTIES MACOSX_BUNDLE_BUNDLE_NAME "Runs KDE Plasma widget (Plasmoid) in a window")
+endif (Q_WS_MAC)
 
 target_link_libraries(kdeinit_plasma-windowed ${KDE4_PLASMA_LIBS} kworkspace ${KDE4_KIO_LIBS} ${X11_LIBRARIES} ${KDE4_KFILE_LIBS} )
 
diff --git plasma/generic/shells/plasma-windowed/plasmaapp.cpp plasma/generic/shells/plasma-windowed/plasmaapp.cpp
index dbdff47..5702001 100644
--- plasma/generic/shells/plasma-windowed/plasmaapp.cpp
+++ plasma/generic/shells/plasma-windowed/plasmaapp.cpp
@@ -144,7 +144,11 @@ int  PlasmaApp::newInstance()
     SingleView *view = new SingleView(m_corona, containment, pluginName, appletId, appletArgs);
 
     if (!view->applet()) {
+#ifdef Q_OS_MAC
+        view->deleteLater();
+#else
         delete view;
+#endif
         return 0;
     }
 
@@ -180,6 +184,22 @@ int  PlasmaApp::newInstance()
     return 0;
 }
 
+#ifdef Q_OS_MAC
+template <typename ForwardIterator>
+Q_OUTOFLINE_TEMPLATE void qDeleteLaterAll(ForwardIterator begin, ForwardIterator end)
+{
+    while (begin != end) {
+        (*begin)->deleteLater();
+        ++begin;
+    }
+}
+
+template <typename Container>
+inline void qDeleteLaterAll(const Container &c)
+{
+    qDeleteLaterAll(c.begin(), c.end());
+}
+#endif
 
 void PlasmaApp::cleanup()
 {
@@ -187,9 +207,14 @@ void PlasmaApp::cleanup()
         m_corona->saveLayout();
     }
 
+#ifdef Q_OS_MAC
+    qDeleteLaterAll(m_viewForPlugin);
+    m_corona->deleteLater();
+#else
     qDeleteAll(m_viewForPlugin);
 
     delete m_corona;
+#endif
     m_corona = 0;
 
     //TODO: This manual sync() should not be necessary?
@@ -236,7 +261,11 @@ void PlasmaApp::storeApplet(Plasma::Applet *applet)
     KConfigGroup cg(applet->containment()->config());
     cg = KConfigGroup(&cg, "Applets");
     cg = KConfigGroup(&cg, QString::number(applet->id()));
+#ifdef Q_OS_MAC
+    applet->deleteLater();
+#else
     delete applet;
+#endif
 //    kDebug() << "storing" << applet->name() << applet->id() << "to" << storage.name() << ", applet config is" << cg.name();
     cg.reparent(&storage);
 }
diff --git plasma/netbook/containments/CMakeLists.txt plasma/netbook/containments/CMakeLists.txt
index c96a688..de6fe3b 100644
--- plasma/netbook/containments/CMakeLists.txt
+++ plasma/netbook/containments/CMakeLists.txt
@@ -1,4 +1,4 @@
 add_subdirectory(netpanel)
-if(NOT WIN32)
+if(NOT WIN32 AND NOT APPLE)
 add_subdirectory(sal)
-endif(NOT WIN32)
+endif(NOT WIN32 AND NOT APPLE)
diff --git plasma/netbook/shell/CMakeLists.txt plasma/netbook/shell/CMakeLists.txt
index e2a83fd..34a7aa6 100644
--- plasma/netbook/shell/CMakeLists.txt
+++ plasma/netbook/shell/CMakeLists.txt
@@ -19,10 +19,14 @@ qt4_add_dbus_interface(plasma-netbook_SRCS ${ksmserver_xml} ksmserver_interface)
 
 kde4_add_kdeinit_executable(plasma-netbook ${plasma-netbook_SRCS})
 
-target_link_libraries(kdeinit_plasma-netbook ${KDE4_PLASMA_LIBS} kworkspace ${KDE4_KIO_LIBS} ${X11_LIBRARIES} ${KDE4_KFILE_LIBS} kephal plasmagenericshell ${QT_QTOPENGL_LIBRARY} ${QT_QTSCRIPT_LIBRARY})
-if(X11_Xrender_FOUND)
-   target_link_libraries(kdeinit_plasma-netbook ${X11_Xrender_LIB})
-endif(X11_Xrender_FOUND)
+if(APPLE)
+    target_link_libraries(kdeinit_plasma-netbook ${KDE4_PLASMA_LIBS} kworkspace ${KDE4_KIO_LIBS} ${KDE4_KFILE_LIBS} kephal plasmagenericshell ${QT_QTOPENGL_LIBRARY} ${QT_QTSCRIPT_LIBRARY})
+else()
+    target_link_libraries(kdeinit_plasma-netbook ${KDE4_PLASMA_LIBS} kworkspace ${KDE4_KIO_LIBS} ${X11_LIBRARIES} ${KDE4_KFILE_LIBS} kephal plasmagenericshell ${QT_QTOPENGL_LIBRARY} ${QT_QTSCRIPT_LIBRARY})
+    if(X11_Xrender_FOUND)
+       target_link_libraries(kdeinit_plasma-netbook ${X11_Xrender_LIB})
+    endif(X11_Xrender_FOUND)
+endif(APPLE)
 
 install(TARGETS kdeinit_plasma-netbook DESTINATION ${LIB_INSTALL_DIR})
 install(TARGETS plasma-netbook ${INSTALL_TARGETS_DEFAULT_ARGS})
diff --git plasma/netbook/shell/plasmaapp.cpp plasma/netbook/shell/plasmaapp.cpp
index c364238..f19329e 100644
--- plasma/netbook/shell/plasmaapp.cpp
+++ plasma/netbook/shell/plasmaapp.cpp
@@ -1027,6 +1027,7 @@ bool PlasmaApp::eventFilter(QObject * watched, QEvent *event)
     return false;
 }
 
+#ifdef Q_WS_X11
 bool PlasmaApp::x11EventFilter(XEvent *event)
 {
 
@@ -1061,6 +1062,13 @@ bool PlasmaApp::x11EventFilter(XEvent *event)
 
     return KUniqueApplication::x11EventFilter(event);
 }
+#else
+bool PlasmaApp::x11EventFilter(void *event)
+{
+    Q_UNUSED(event);
+    return false;
+}
+#endif
 
 bool PlasmaApp::hasForegroundWindows() const
 {
@@ -1113,7 +1121,12 @@ void PlasmaApp::controlBarVisibilityUpdate()
     //would be nice to avoid this
     QPoint cursorPos = QCursor::pos();
 
-    if (m_triggerZone.adjusted(-1, -1, 1, 1).contains(cursorPos) || hasForegroundWindows()) {
+    if (
+#ifdef Q_WS_X11
+        m_triggerZone.adjusted(-1, -1, 1, 1).contains(cursorPos) ||
+#endif
+        hasForegroundWindows())
+    {
         if (!m_controlBar->isVisible()) {
             destroyUnHideTrigger();
             Plasma::WindowEffects::slideWindow(m_controlBar, m_controlBar->location());
diff --git plasma/netbook/shell/plasmaapp.h plasma/netbook/shell/plasmaapp.h
index 8374e53..002a7da 100644
--- plasma/netbook/shell/plasmaapp.h
+++ plasma/netbook/shell/plasmaapp.h
@@ -94,7 +94,11 @@ public Q_SLOTS:
 
 protected:
     bool eventFilter(QObject *watched, QEvent *event);
+#ifdef Q_WS_X11
     bool x11EventFilter(XEvent *event);
+#else
+    bool x11EventFilter(void *event);
+#endif
     //true if the application has an active window
     bool hasForegroundWindows() const;
     //more strict than hasForegroundWindows, the mainview has to be not only active, but also on top of other windows
--- /dev/null	2015-05-05 02:27:21.000000000 -0700
+++ plasma/generic/shells/plasma-windowed/Info.plist.template	2015-03-24 15:06:27.000000000 -0700
@@ -0,0 +1,36 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple Computer//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+    <key>CFBundleDevelopmentRegion</key>
+    <string>English</string>
+    <key>CFBundleExecutable</key>
+    <string>${MACOSX_BUNDLE_EXECUTABLE_NAME}</string>
+    <key>CFBundleGetInfoString</key>
+    <string>${MACOSX_BUNDLE_INFO_STRING}</string>
+    <key>CFBundleIconFile</key>
+    <string>${MACOSX_BUNDLE_ICON_FILE}</string>
+    <key>CFBundleIdentifier</key>
+    <string>${MACOSX_BUNDLE_GUI_IDENTIFIER}</string>
+    <key>CFBundleInfoDictionaryVersion</key>
+    <string>6.0</string>
+    <key>CFBundleLongVersionString</key>
+    <string>${MACOSX_BUNDLE_LONG_VERSION_STRING}</string>
+    <key>CFBundleName</key>
+    <string>${MACOSX_BUNDLE_BUNDLE_NAME}</string>
+    <key>CFBundlePackageType</key>
+    <string>APPL</string>
+    <key>CFBundleShortVersionString</key>
+    <string>${MACOSX_BUNDLE_SHORT_VERSION_STRING}</string>
+    <key>CFBundleVersion</key>
+    <string>${MACOSX_BUNDLE_BUNDLE_VERSION}</string>
+    <key>CSResourcesFileMapped</key>
+    <true/>
+    <key>LSRequiresCarbon</key>
+    <true/>
+    <key>LSUIElement</key>
+    <string>1</string>
+    <key>NSHumanReadableCopyright</key>
+    <string>${MACOSX_BUNDLE_COPYRIGHT}</string>
+</dict>
+</plist>
