diff --git plasma/CMakeLists.txt plasma/CMakeLists.txt
index 199dbb0..2175e39 100644
--- plasma/CMakeLists.txt
+++ plasma/CMakeLists.txt
@@ -12,7 +12,9 @@ ENDMACRO(installPackage)
 if(${KDE_PLATFORM_PROFILE} STREQUAL "Desktop")
    add_subdirectory(desktop)
    add_subdirectory(netbook)
-   add_subdirectory(screensaver)
+    if(NOT APPLE)
+       add_subdirectory(screensaver)
+    endif(NOT APPLE)
 endif(${KDE_PLATFORM_PROFILE} STREQUAL "Desktop")
 
 add_subdirectory(generic)
diff --git plasma/desktop/CMakeLists.txt plasma/desktop/CMakeLists.txt
index 2de78dd..fb612e8 100644
--- plasma/desktop/CMakeLists.txt
+++ plasma/desktop/CMakeLists.txt
@@ -1,5 +1,7 @@
 add_subdirectory(applets)
 add_subdirectory(containments)
 add_subdirectory(runners)
-add_subdirectory(shell)
+#if(NOT APPLE)
+    add_subdirectory(shell)
+#endif(NOT APPLE)
 add_subdirectory(toolboxes)
diff --git plasma/desktop/applets/CMakeLists.txt plasma/desktop/applets/CMakeLists.txt
index 6f80cec..5202f83 100644
--- plasma/desktop/applets/CMakeLists.txt
+++ plasma/desktop/applets/CMakeLists.txt
@@ -2,10 +2,10 @@ add_subdirectory(kickoff)
 add_subdirectory(showActivityManager)
 add_subdirectory(trash)
 
-if(NOT WIN32)
+if(NOT WIN32 AND NOT APPLE)
     #task and windowlist depend on libs/taskmanager
    add_subdirectory(tasks)
    add_subdirectory(windowlist)
 
    add_subdirectory(pager)
-endif(NOT WIN32)
+endif(NOT WIN32 AND NOT APPLE)
diff --git plasma/desktop/applets/pager/plasma-pager-default.desktop plasma/desktop/applets/pager/plasma-pager-default.desktop
index 96cf864..123361b 100644
--- plasma/desktop/applets/pager/plasma-pager-default.desktop
+++ plasma/desktop/applets/pager/plasma-pager-default.desktop
@@ -56,7 +56,7 @@ Name[nn]=Skrivebordsbytar
 Name[oc]=Recebedor d'apèl
 Name[or]=ପେଜର
 Name[pa]=ਪੇਜ਼ਰ
-Name[pl]=Pager
+Name[pl]=Podgląd pulpitów
 Name[pt]=Paginador
 Name[pt_BR]=Paginador
 Name[ro]=Paginator
diff --git plasma/desktop/applets/showActivityManager/package/metadata.desktop plasma/desktop/applets/showActivityManager/package/metadata.desktop
index bfd1bad..3f5dd76 100644
--- plasma/desktop/applets/showActivityManager/package/metadata.desktop
+++ plasma/desktop/applets/showActivityManager/package/metadata.desktop
@@ -88,7 +88,7 @@ Comment[nb]=Viser aktivitetshåndtereren
 Comment[nds]=Wiest den Aktivitetenpleger
 Comment[nl]=Toont de activiteitenbeheerder
 Comment[pa]=ਸਰਗਰਮੀ ਮੈਨੇਜਰ ਵੇਖਾਉਂਦਾ ਹੈ
-Comment[pl]=Pokazuje menadżera działań
+Comment[pl]=Pokazuje zarządzanie działaniami
 Comment[pt]=Mostra o gestor de actividades
 Comment[pt_BR]=Mostra o gerenciador de atividades
 Comment[ro]=Arată Gestionarul de activități
diff --git plasma/desktop/shell/desktopview.cpp plasma/desktop/shell/desktopview.cpp
index f0f0371..cfce13c 100644
--- plasma/desktop/shell/desktopview.cpp
+++ plasma/desktop/shell/desktopview.cpp
@@ -314,7 +314,7 @@ void DesktopView::toolBoxOpened(bool open)
         return;
     }
 
-#ifndef Q_WS_WIN
+#ifdef Q_WS_X11
     NETRootInfo info(QX11Info::display(), NET::Supported);
     if (!info.isSupported(NET::WM2ShowingDesktop)) {
         return;
diff --git plasma/desktop/shell/plasmaapp.cpp plasma/desktop/shell/plasmaapp.cpp
index c1294a1..49846fb 100644
--- plasma/desktop/shell/plasmaapp.cpp
+++ plasma/desktop/shell/plasmaapp.cpp
@@ -31,7 +31,7 @@
 #include <unistd.h>
 
 #ifndef _SC_PHYS_PAGES
-    #ifdef Q_OS_FREEBSD
+    #if defined(Q_OS_FREEBSD) || defined(Q_OS_MAC)
     #include <sys/types.h>
     #include <sys/sysctl.h>
     #endif
@@ -216,7 +216,7 @@ PlasmaApp::PlasmaApp()
     int memorySize = sysconf(_SC_PHYS_PAGES);
     memorySize *= sysconf(_SC_PAGESIZE) / 1024;
 #else
-#ifdef Q_OS_FREEBSD
+#if defined(Q_OS_FREEBSD) || defined(Q_OS_MAC)
     int sysctlbuf[2];
     size_t size = sizeof(sysctlbuf);
     int memorySize;
diff --git plasma/generic/CMakeLists.txt plasma/generic/CMakeLists.txt
index cfaf14f..294faaf 100644
--- plasma/generic/CMakeLists.txt
+++ plasma/generic/CMakeLists.txt
@@ -11,5 +11,7 @@ add_subdirectory(dataengines)
 add_subdirectory(runners)
 add_subdirectory(scriptengines)
 add_subdirectory(shells)
-add_subdirectory(wallpapers)
+if(NOT APPLE)
+    add_subdirectory(wallpapers)
+endif(NOT APPLE)
 
diff --git plasma/generic/applets/CMakeLists.txt plasma/generic/applets/CMakeLists.txt
index 2b888ee..0e1131d 100644
--- plasma/generic/applets/CMakeLists.txt
+++ plasma/generic/applets/CMakeLists.txt
@@ -16,7 +16,7 @@ if(QT_QTWEBKIT_FOUND)
    add_subdirectory(webbrowser)
 endif(QT_QTWEBKIT_FOUND)
 
-if(NOT WIN32)
+if(NOT WIN32 AND NOT APPLE)
    add_subdirectory(system-monitor)
    #notifications
    #should compile also on windows? (even if doesn't really make sense)
@@ -24,4 +24,4 @@ if(NOT WIN32)
    if(${KDE_PLATFORM_PROFILE} STREQUAL "Desktop")
       add_subdirectory(systemtray)
    endif(${KDE_PLATFORM_PROFILE} STREQUAL "Desktop")
-endif(NOT WIN32)
+endif(NOT WIN32 AND NOT APPLE)
diff --git plasma/generic/applets/batterymonitor/metadata.desktop plasma/generic/applets/batterymonitor/metadata.desktop
index e0d2486..5ee08b4 100644
--- plasma/generic/applets/batterymonitor/metadata.desktop
+++ plasma/generic/applets/batterymonitor/metadata.desktop
@@ -159,7 +159,7 @@ Comment[zh_CN]=查看电池电源的状态
 Comment[zh_TW]=查看您的電池的電力狀態
 Encoding=UTF-8
 Keywords=Power Management;Battery;System;Energy;
-Keywords[bs]=Upravljanje napajenjem, Baterija, Sistem, Energija
+Keywords[bs]=Upravljanje napajenjem;Baterija;Sistem;Energija;
 Keywords[ca]=Sistema de gestió d'energia;Bateria;Sistema;Energia;
 Keywords[ca@valencia]=Sistema de gestió d'energia;Bateria;Sistema;Energia;
 Keywords[cs]=Správa napájení;Baterie;Systém;Energie;
@@ -188,17 +188,17 @@ Keywords[pl]=Zarządzanie Energią;Bateria;System;Energia
 Keywords[pt]=Gestão de Energia;Bateria;Sistema;Energia;
 Keywords[pt_BR]=Gerenciamento de energia;Bateria;Sistema;Energia;
 Keywords[ro]=gestiunea alimentării;acumulator;sistem;energie;
-Keywords[ru]=Power Management;Battery;System;Energy;управление питанием;батарея;система;энергия;энергопотребление
+Keywords[ru]=Power Management;Battery;System;Energy;управление питанием;батарея;система;энергия;энергопотребление;
 Keywords[sk]=Správa napájania;Batéria;Systém;Energia;
 Keywords[sl]=upravljanje z energijo;baterija;sistem;energija;
-Keywords[sr]=Power Management;Battery;System;Energy;управљање напајањем;батерија; систем;енергија
-Keywords[sr@ijekavian]=Power Management;Battery;System;Energy;управљање напајањем;батерија; систем;енергија
-Keywords[sr@ijekavianlatin]=Power Management;Battery;System;Energy;upravljanje napajanjem;baterija; sistem;energija
-Keywords[sr@latin]=Power Management;Battery;System;Energy;upravljanje napajanjem;baterija; sistem;energija
+Keywords[sr]=Power Management;Battery;System;Energy;управљање напајањем;батерија; систем;енергија;
+Keywords[sr@ijekavian]=Power Management;Battery;System;Energy;управљање напајањем;батерија; систем;енергија;
+Keywords[sr@ijekavianlatin]=Power Management;Battery;System;Energy;upravljanje napajanjem;baterija; sistem;energija;
+Keywords[sr@latin]=Power Management;Battery;System;Energy;upravljanje napajanjem;baterija; sistem;energija;
 Keywords[sv]=Strömhantering;Batteri;System;Energi;
 Keywords[tr]=Güç Yönetimi;Pil;Sistem;Enerji;
 Keywords[uk]=Power Management;Battery;System;Energy;керування живленням;акумулятор;система;енергія;батарея;батарейка;живлення;
-Keywords[x-test]=xxPower Management;Battery;System;Energy;xx
+Keywords[x-test]=xxPower Managementxx;xxBatteryxx;xxSystemxx;xxEnergyxx;
 Keywords[zh_CN]=Power Management;Battery;System;Energy;电源管理;电池;系统;能源;
 Keywords[zh_TW]=Power Management;Battery;System;Energy;
 
diff --git plasma/generic/applets/lock_logout/metadata.desktop plasma/generic/applets/lock_logout/metadata.desktop
index b07df34..df225d6 100644
--- plasma/generic/applets/lock_logout/metadata.desktop
+++ plasma/generic/applets/lock_logout/metadata.desktop
@@ -159,10 +159,10 @@ Comment[zh_CN]=锁定屏幕或注销
 Comment[zh_TW]=鎖定螢幕或登出
 Encoding=UTF-8
 Keywords=Lock;Logout;Sleep;Hibernate;Switch User;
-Keywords[bs]=Zaključavanje, Odjava, Spavanje, Hibernacija, Promijena korisnika
+Keywords[bs]=Zaključavanje;Odjava;Spavanje;Hibernacija;Promjena korisnika;
 Keywords[ca]=Bloqueig;Sortida;Suspensió;Hibernació;Canvi d'usuari;
 Keywords[ca@valencia]=Bloqueig;Eixida;Suspensió;Hibernació;Canvi d'usuari;
-Keywords[da]=Lås;Log ud;Slumre;Dvale;Suspend;Skift bruger;suspender
+Keywords[da]=Lås;Log ud;Slumre;Dvale;Suspend;Skift bruger;suspender;
 Keywords[de]=Sperren;Abmelden;Ruhezustand;Tiefschlaf;Benutzer wechseln;
 Keywords[el]=Κλείδωμα·έξοδος·ύπνωση·νάρκη·εναλλαγήχρήστη·
 Keywords[en_GB]=Lock;Logout;Sleep;Hibernate;Switch User;
@@ -180,7 +180,7 @@ Keywords[ko]=Lock;Logout;Sleep;Hibernate;Switch User;잠금;로그아웃;대기;
 Keywords[mr]=कुलूप; बाहेर पडा; झोप; हायबरनेट; वापरकर्ता बदला;
 Keywords[nb]=Lås; Logg ut; Hvile; Dvale; Bytt bruker;
 Keywords[nds]=Slott,afsluten,afmellen,slapen,infreren,Bruker wesseln;Brukerwessel
-Keywords[nl]=Vergrendelen;Afmelden;Slapen;Slaapstand naar schijf;Gebruiker wisselen
+Keywords[nl]=Vergrendelen;Afmelden;Slapen;Slaapstand naar schijf;Gebruiker wisselen;
 Keywords[pa]=ਲਾਕ;ਲਾਗਆਉਟ;ਸਲੀਪ;ਹਾਈਬਰਨੇਟ;ਯੂਜ਼ਰ ਬਦਲੋ;
 Keywords[pl]=Zablokuj;Wyloguj;Uśpij;Hibernuj;Przełącz użytkownika;
 Keywords[pt]=Bloquear;Encerrar;Suspender;Hibernar;Mudar de Utilizador;
@@ -189,14 +189,14 @@ Keywords[ro]=blochează;ieși;adormire;dormi;hibernare;schimbă utilizatorul;com
 Keywords[ru]=Lock;Logout;Sleep;Hibernate;Switch User;блокировать;блокировка;выход;выход из системы;спящий режим;сон;режим гибернации;гибернация;сменить пользователя;смена пользователя;
 Keywords[sk]=Zamknúť;Odhlásiť;Uspať;Hibernovať;Prepnúť používateľa;
 Keywords[sl]=zaklep;odjava;pripravljenost;mirovanje;zamenjava uporabnika;
-Keywords[sr]=Lock;Logout;Sleep;Hibernate;Switch User;закључавање;одјављивање;спавање;хибернација;пребацивање
-Keywords[sr@ijekavian]=Lock;Logout;Sleep;Hibernate;Switch User;закључавање;одјављивање;спавање;хибернација;пребацивање
-Keywords[sr@ijekavianlatin]=Lock;Logout;Sleep;Hibernate;Switch User;zaključavanje;odjavljivanje;spavanje;hibernacija;prebacivanje
-Keywords[sr@latin]=Lock;Logout;Sleep;Hibernate;Switch User;zaključavanje;odjavljivanje;spavanje;hibernacija;prebacivanje
+Keywords[sr]=Lock;Logout;Sleep;Hibernate;Switch User;закључавање;одјављивање;спавање;хибернација;пребацивање;
+Keywords[sr@ijekavian]=Lock;Logout;Sleep;Hibernate;Switch User;закључавање;одјављивање;спавање;хибернација;пребацивање;
+Keywords[sr@ijekavianlatin]=Lock;Logout;Sleep;Hibernate;Switch User;zaključavanje;odjavljivanje;spavanje;hibernacija;prebacivanje;
+Keywords[sr@latin]=Lock;Logout;Sleep;Hibernate;Switch User;zaključavanje;odjavljivanje;spavanje;hibernacija;prebacivanje;
 Keywords[sv]=Lås;Utloggning;Viloläge;Dvala;Byt användare;
 Keywords[tr]=Kilitle;Oturumu Kapat;Uyku;Askıya Al;Kullanıcı Değiştir;
-Keywords[uk]=Lock;Logout;Sleep;Hibernate;Switch User;блокування;вихід;присипляння;сон;перемикання;користувач
-Keywords[x-test]=xxLock;Logout;Sleep;Hibernate;Switch User;xx
+Keywords[uk]=Lock;Logout;Sleep;Hibernate;Switch User;блокування;вихід;присипляння;сон;перемикання;користувач;
+Keywords[x-test]=xxLockxx;xxLogoutxx;xxSleepxx;xxHibernatexx;xxSwitch Userxx;
 Keywords[zh_CN]=Lock;Logout;Sleep;Hibernate;Switch User;锁定;注销;待机;休眠;切换用户;
 Keywords[zh_TW]=Lock;Logout;Sleep;Hibernate;Switch User;
 
diff --git plasma/generic/applets/webbrowser/plasma-applet-webbrowser.desktop plasma/generic/applets/webbrowser/plasma-applet-webbrowser.desktop
index 5d83d24..fd40a83 100644
--- plasma/generic/applets/webbrowser/plasma-applet-webbrowser.desktop
+++ plasma/generic/applets/webbrowser/plasma-applet-webbrowser.desktop
@@ -13,7 +13,6 @@ Name[br]=Furcher an Internet
 Name[bs]=veb pregledač
 Name[ca]=Navegació Web
 Name[ca@valencia]=Navegació Web
-Name[cs]=Prohlížeč Webu
 Name[csb]=Przezérnik WWW
 Name[cy]=Porydd Gwê
 Name[da]=Browser
@@ -26,7 +25,6 @@ Name[et]=Veebibrauser
 Name[eu]=Web-arakatzailea
 Name[fa]=مرورگر وب‌‌
 Name[fi]=Verkkoselain
-Name[fr]=Navigateur Web
 Name[fy]=Webblêder
 Name[ga]=Brabhsálaí Gréasáin
 Name[gl]=Navegador web
@@ -41,12 +39,10 @@ Name[ia]=Navigator Web
 Name[id]=Peramban Web
 Name[is]=Vafri
 Name[it]=Browser Web
-Name[ja]=ウェブブラウザ
 Name[ka]=ვებ ბრაუზერი
 Name[kk]=Веб шолғышы
 Name[km]=កម្មវិធី​រុករក​បណ្ដាញ
 Name[kn]=ಚಾಲ ವೀಕ್ಷಕ
-Name[ko]=웹 브라우저
 Name[ku]=Geroka Torê
 Name[lt]=Žiniatinklio naršyklė
 Name[lv]=Tīmekļa pārlūks
@@ -102,7 +98,6 @@ Comment[bn_IN]=একটি সাধারণ ওয়েব ব্রাউজ
 Comment[bs]=Jednostavan veb pregledač
 Comment[ca]=Un navegador web senzill
 Comment[ca@valencia]=Un navegador web senzill
-Comment[cs]=Jednoduchý webový prohlížeč
 Comment[csb]=Prosti przezérnik WWW
 Comment[da]=En simpel webbrowser
 Comment[de]=Einfacher Webbrowser
@@ -113,7 +108,6 @@ Comment[es]=Navegador sencillo
 Comment[et]=Lihtne veebibrauser
 Comment[eu]=Web-arakatzaile soila
 Comment[fi]=Yksinkertainen selain
-Comment[fr]=Un navigateur Web simple
 Comment[fy]=In ienfâldige webblêder
 Comment[ga]=Brabhsálaí simplí
 Comment[gl]=Un navegador web simples
@@ -128,18 +122,15 @@ Comment[ia]=Un simple navigator web
 Comment[id]=Peramban web sederhana
 Comment[is]=Einfaldur vafri
 Comment[it]=Un semplice browser Web
-Comment[ja]=シンプルなウェブブラウザ
 Comment[kk]=Қарапайым веб шолғышы
 Comment[km]=កម្មវិធី​រុករក​បណ្ដាញ​ធម្មតា
 Comment[kn]=ಒಂದು ಸರಳ ಜಾಲ ವೀಕ್ಷಕ
-Comment[ko]=간단한 웹 브라우저
 Comment[ku]=Geroka torê yê hêsan
 Comment[lt]=Paprasta žiniatinklio naršyklė
 Comment[lv]=Vienkāršs tīmekļa pārlūks
 Comment[mk]=Едноставен веб-прелистувач
 Comment[ml]=ലളിതമായ വെബ് ബ്രൌസര്‍
 Comment[mr]=सोपा वेब ब्राऊजर
-Comment[nb]=Enkel nettleser
 Comment[nds]=En eenfach Nettkieker
 Comment[nl]=Eenvoudige webbrowser
 Comment[nn]=Enkel nettlesar
diff --git plasma/generic/dataengines/CMakeLists.txt plasma/generic/dataengines/CMakeLists.txt
index d240683..35a7a10 100644
--- plasma/generic/dataengines/CMakeLists.txt
+++ plasma/generic/dataengines/CMakeLists.txt
@@ -34,11 +34,11 @@ if(Boost_FOUND AND KdepimLibs_FOUND)
    add_subdirectory(rss)
 endif(Boost_FOUND AND KdepimLibs_FOUND)
 
-if(NOT WIN32)
+if(NOT WIN32 AND NOT APPLE)
    add_subdirectory(mouse)
    add_subdirectory(systemmonitor)
    if(${KDE_PLATFORM_PROFILE} STREQUAL "Desktop")
       add_subdirectory(tasks)
    endif(${KDE_PLATFORM_PROFILE} STREQUAL "Desktop")
-endif(NOT WIN32)
+endif(NOT WIN32 AND NOT APPLE)
 
diff --git plasma/generic/dataengines/filebrowser/plasma-dataengine-filebrowser.desktop plasma/generic/dataengines/filebrowser/plasma-dataengine-filebrowser.desktop
index 04ea347..b8f252a 100644
--- plasma/generic/dataengines/filebrowser/plasma-dataengine-filebrowser.desktop
+++ plasma/generic/dataengines/filebrowser/plasma-dataengine-filebrowser.desktop
@@ -93,7 +93,7 @@ Comment[eo]=Informoj pri dosieroj kaj dosierujoj
 Comment[es]=Información sobre los archivos y directorios.
 Comment[et]=Teave failide ja kataloogide kohta.
 Comment[eu]=Fitxategi eta direktorioei buruzko informazioa
-Comment[fi]=Tietoja tiedostosta ja hakemistoista.
+Comment[fi]=Tietoa tiedostosta ja hakemistoista.
 Comment[fr]=Informations sur les fichiers et les dossiers.
 Comment[fy]=Ynformaasje oer triemmen en triemtafels.
 Comment[ga]=Eolas faoi chomhaid agus faoi chomhadlanna.
diff --git plasma/generic/dataengines/powermanagement/plasma-dataengine-powermanagement.desktop plasma/generic/dataengines/powermanagement/plasma-dataengine-powermanagement.desktop
index 0e979a9..162dc2a 100644
--- plasma/generic/dataengines/powermanagement/plasma-dataengine-powermanagement.desktop
+++ plasma/generic/dataengines/powermanagement/plasma-dataengine-powermanagement.desktop
@@ -100,7 +100,7 @@ Comment[ja]=バッテリ、商用電源、スリープ、PowerDevil の情報。
 Comment[kk]=Батарея, тоқ, ұйықтау күйі және PowerDevil мәліметі.
 Comment[km]=ព័ត៌មាន ថ្ម AC ដេក និង​ PowerDevil ។
 Comment[kn]=ವಿದ್ಯುತ್ಕೋಶ, AC, ಜಡ ಹಾಗು PowerDevil ಮಾಹಿತಿ.
-Comment[ko]=배터리, AC, 절전 모드 및 PowerDevil 정보입니다.
+Comment[ko]=배터리; AC; 절전 모드 및 PowerDevil 정보입니다.
 Comment[lt]=Akumuliatoriaus, AC, miego ir PowerDevil informacija.
 Comment[lv]=Baterija, barošana, gulēšanas un PowerDevil informācija.
 Comment[ml]=ബാറ്ററി, എസി, നിദ്ര പിന്നെ പവര്‍ഡെവിള്‍ എന്നീ വിവരങ്ങള്‍.
diff --git plasma/generic/runners/CMakeLists.txt plasma/generic/runners/CMakeLists.txt
index 6831ac0..963fe59 100644
--- plasma/generic/runners/CMakeLists.txt
+++ plasma/generic/runners/CMakeLists.txt
@@ -14,10 +14,10 @@ if(NepomukCore_FOUND)
   add_subdirectory(nepomuksearch)
 endif(NepomukCore_FOUND)
 
-if(NOT WIN32)
+if(NOT WIN32 AND NOT APPLE)
 add_subdirectory(powerdevil)
 add_subdirectory(sessions)
 add_subdirectory(windows)
 add_subdirectory(kill)
-endif(NOT WIN32)
+endif(NOT WIN32 AND NOT APPLE)
 
diff --git plasma/generic/runners/kill/plasma-runner-kill_config.desktop plasma/generic/runners/kill/plasma-runner-kill_config.desktop
index b78523a..9d65503 100644
--- plasma/generic/runners/kill/plasma-runner-kill_config.desktop
+++ plasma/generic/runners/kill/plasma-runner-kill_config.desktop
@@ -61,10 +61,10 @@ Name[ru]=Завершение приложений
 Name[si]=KDE යෙදුම්
 Name[sk]=Zabiť aplikácie
 Name[sl]=Uniči programe
-Name[sr]=Обустављање програма
-Name[sr@ijekavian]=Обустављање програма
-Name[sr@ijekavianlatin]=Obustavljanje programa
-Name[sr@latin]=Obustavljanje programa
+Name[sr]=обустављање програма
+Name[sr@ijekavian]=обустављање програма
+Name[sr@ijekavianlatin]=obustavljanje programa
+Name[sr@latin]=obustavljanje programa
 Name[sv]=Döda program
 Name[tg]=Барномаҳои KDE
 Name[th]=ฆ่าโพรเซสของโปรแกรม
diff --git plasma/generic/runners/locations/plasma-runner-locations.desktop plasma/generic/runners/locations/plasma-runner-locations.desktop
index 88afd22..ae68a6a 100644
--- plasma/generic/runners/locations/plasma-runner-locations.desktop
+++ plasma/generic/runners/locations/plasma-runner-locations.desktop
@@ -20,7 +20,6 @@ Name[es]=Ubicaciones
 Name[et]=Asukohad
 Name[eu]=Kokalekuak
 Name[fi]=Sijainnit
-Name[fr]=Emplacements
 Name[fy]=Lokaasjes
 Name[ga]=Áiteanna
 Name[gl]=Lugares
@@ -34,7 +33,6 @@ Name[ia]=Locationes
 Name[id]=Lokasi
 Name[is]=Staðsetningar
 Name[it]=Posizioni
-Name[ja]=場所
 Name[kk]=Адрестер
 Name[km]=ទីតាំង
 Name[kn]=ಸ್ಥಳಗಳು
@@ -45,7 +43,6 @@ Name[lv]=Vietas
 Name[mk]=Локации
 Name[ml]=സ്ഥാനങ്ങള്‍
 Name[mr]=स्थान
-Name[nb]=Steder
 Name[nds]=Steden
 Name[nl]=Locaties
 Name[nn]=Adresser
@@ -59,10 +56,10 @@ Name[ru]=Расположения
 Name[si]=පිහිටුම
 Name[sk]=Miesta
 Name[sl]=Mesta
-Name[sr]=локације
-Name[sr@ijekavian]=локације
-Name[sr@ijekavianlatin]=lokacije
-Name[sr@latin]=lokacije
+Name[sr]=Локације
+Name[sr@ijekavian]=Локације
+Name[sr@ijekavianlatin]=Lokacije
+Name[sr@latin]=Lokacije
 Name[sv]=Platser
 Name[ta]=Locations
 Name[te]=స్థానములు
diff --git plasma/generic/runners/services/plasma-runner-services.desktop plasma/generic/runners/services/plasma-runner-services.desktop
index 4bdd62d..e7dce9d 100644
--- plasma/generic/runners/services/plasma-runner-services.desktop
+++ plasma/generic/runners/services/plasma-runner-services.desktop
@@ -127,7 +127,7 @@ Comment[ja]=アプリケーション、コントロールパネル、サービ
 Comment[kk]=Қолданбаларды табу, панельдер мен қызметтерді басқару
 Comment[km]=រក​កម្មវិធី បន្ទះ​ត្រួតពិនិត្យ និង​សេវា
 Comment[kn]=ಅನ್ವಯಗಳು, ನಿಯಂತ್ರಣಾ ಪುಟೀಪುಗಳು (ಕಂಟ್ರೋಲ್ ಪಾನಲ್ಸ್) ಹಾಗೂ ಸೇವೆಗಳನ್ನು ಹುಡುಕು
-Comment[ko]=프로그램, 제어판, 서비스 찾기
+Comment[ko]=프로그램; 제어판; 서비스 찾기
 Comment[ku]=Sepanan, panelên kontrolê û servîsan bibîne
 Comment[lt]=Rasti programas, valdymo pultus ir tarnybas
 Comment[lv]=Meklēt lietotnes, kontroles paneļus un servisus
diff --git plasma/generic/runners/webshortcuts/plasma-runner-webshortcuts.desktop plasma/generic/runners/webshortcuts/plasma-runner-webshortcuts.desktop
index 5e92780..89db96c 100644
--- plasma/generic/runners/webshortcuts/plasma-runner-webshortcuts.desktop
+++ plasma/generic/runners/webshortcuts/plasma-runner-webshortcuts.desktop
@@ -76,7 +76,7 @@ Comment[ca]=Permet que l'usuari utilitzi les dreceres web del Konqueror
 Comment[ca@valencia]=Permet que l'usuari utilitze les dreceres web del Konqueror
 Comment[cs]=Umožňuje používat webové zkratky Konqueroru
 Comment[da]=Lader brugeren anvende Konquerors webgenveje
-Comment[de]=rmöglicht die Verwendung von Web-Kürzeln in Konqueror
+Comment[de]=Ermöglicht die Verwendung von Web-Kürzeln in Konqueror
 Comment[el]=Επιτρέπει τη χρήση συντομεύσεων ιστού του Konqueror
 Comment[en_GB]=Allows user to use Konqueror's web shortcuts
 Comment[es]=Permite al usuario usar accesos rápidos web de Konqueror
diff --git plasma/generic/shells/plasma-windowed/CMakeLists.txt plasma/generic/shells/plasma-windowed/CMakeLists.txt
index 86b7770..74b9641 100644
--- plasma/generic/shells/plasma-windowed/CMakeLists.txt
+++ plasma/generic/shells/plasma-windowed/CMakeLists.txt
@@ -10,6 +10,11 @@ set(plasma-windowed_SRCS
 
 
 kde4_add_kdeinit_executable(plasma-windowed ${plasma-windowed_SRCS})
+if (Q_WS_MAC)
+    set_target_properties(plasma-windowed PROPERTIES MACOSX_BUNDLE_INFO_PLIST ${CMAKE_CURRENT_SOURCE_DIR}/Info.plist.template)
+    set_target_properties(plasma-windowed PROPERTIES MACOSX_BUNDLE_GUI_IDENTIFIER "org.kde.plasma.windowed")
+    set_target_properties(plasma-windowed PROPERTIES MACOSX_BUNDLE_BUNDLE_NAME "Runs KDE Plasma widget (Plasmoid) in a window")
+endif (Q_WS_MAC)
 
 target_link_libraries(kdeinit_plasma-windowed ${KDE4_PLASMA_LIBS} kworkspace ${KDE4_KIO_LIBS} ${X11_LIBRARIES} ${KDE4_KFILE_LIBS} )
 
diff --git plasma/generic/shells/plasma-windowed/plasmaapp.cpp plasma/generic/shells/plasma-windowed/plasmaapp.cpp
index dbdff47..5702001 100644
--- plasma/generic/shells/plasma-windowed/plasmaapp.cpp
+++ plasma/generic/shells/plasma-windowed/plasmaapp.cpp
@@ -144,7 +144,11 @@ int  PlasmaApp::newInstance()
     SingleView *view = new SingleView(m_corona, containment, pluginName, appletId, appletArgs);
 
     if (!view->applet()) {
+#ifdef Q_OS_MAC
+        view->deleteLater();
+#else
         delete view;
+#endif
         return 0;
     }
 
@@ -180,6 +184,22 @@ int  PlasmaApp::newInstance()
     return 0;
 }
 
+#ifdef Q_OS_MAC
+template <typename ForwardIterator>
+Q_OUTOFLINE_TEMPLATE void qDeleteLaterAll(ForwardIterator begin, ForwardIterator end)
+{
+    while (begin != end) {
+        (*begin)->deleteLater();
+        ++begin;
+    }
+}
+
+template <typename Container>
+inline void qDeleteLaterAll(const Container &c)
+{
+    qDeleteLaterAll(c.begin(), c.end());
+}
+#endif
 
 void PlasmaApp::cleanup()
 {
@@ -187,9 +207,14 @@ void PlasmaApp::cleanup()
         m_corona->saveLayout();
     }
 
+#ifdef Q_OS_MAC
+    qDeleteLaterAll(m_viewForPlugin);
+    m_corona->deleteLater();
+#else
     qDeleteAll(m_viewForPlugin);
 
     delete m_corona;
+#endif
     m_corona = 0;
 
     //TODO: This manual sync() should not be necessary?
@@ -236,7 +261,11 @@ void PlasmaApp::storeApplet(Plasma::Applet *applet)
     KConfigGroup cg(applet->containment()->config());
     cg = KConfigGroup(&cg, "Applets");
     cg = KConfigGroup(&cg, QString::number(applet->id()));
+#ifdef Q_OS_MAC
+    applet->deleteLater();
+#else
     delete applet;
+#endif
 //    kDebug() << "storing" << applet->name() << applet->id() << "to" << storage.name() << ", applet config is" << cg.name();
     cg.reparent(&storage);
 }
diff --git plasma/netbook/containments/CMakeLists.txt plasma/netbook/containments/CMakeLists.txt
index c96a688..de6fe3b 100644
--- plasma/netbook/containments/CMakeLists.txt
+++ plasma/netbook/containments/CMakeLists.txt
@@ -1,4 +1,4 @@
 add_subdirectory(netpanel)
-if(NOT WIN32)
+if(NOT WIN32 AND NOT APPLE)
 add_subdirectory(sal)
-endif(NOT WIN32)
+endif(NOT WIN32 AND NOT APPLE)
diff --git plasma/netbook/containments/sal/services/plasma-sal-utility.desktop plasma/netbook/containments/sal/services/plasma-sal-utility.desktop
index 9bcfa5e..b8c455f 100644
--- plasma/netbook/containments/sal/services/plasma-sal-utility.desktop
+++ plasma/netbook/containments/sal/services/plasma-sal-utility.desktop
@@ -98,7 +98,7 @@ Comment[bg]=Разни инструменти
 Comment[bs]=Manje priručne alatke
 Comment[ca]=Petites utilitats i accessoris
 Comment[ca@valencia]=Petites utilitats i accessoris
-Comment[cs]=Malé utility a příslušenství
+Comment[cs]=Malé nástroje a příslušenství
 Comment[csb]=Môłé nôrzãdza ë pòmòcné aplikacëje
 Comment[da]=Små værktøjer og tilbehør
 Comment[de]=Kleine Werkzeuge und Zubehörprogramme
diff --git plasma/netbook/shell/CMakeLists.txt plasma/netbook/shell/CMakeLists.txt
index e2a83fd..34a7aa6 100644
--- plasma/netbook/shell/CMakeLists.txt
+++ plasma/netbook/shell/CMakeLists.txt
@@ -19,10 +19,14 @@ qt4_add_dbus_interface(plasma-netbook_SRCS ${ksmserver_xml} ksmserver_interface)
 
 kde4_add_kdeinit_executable(plasma-netbook ${plasma-netbook_SRCS})
 
-target_link_libraries(kdeinit_plasma-netbook ${KDE4_PLASMA_LIBS} kworkspace ${KDE4_KIO_LIBS} ${X11_LIBRARIES} ${KDE4_KFILE_LIBS} kephal plasmagenericshell ${QT_QTOPENGL_LIBRARY} ${QT_QTSCRIPT_LIBRARY})
-if(X11_Xrender_FOUND)
-   target_link_libraries(kdeinit_plasma-netbook ${X11_Xrender_LIB})
-endif(X11_Xrender_FOUND)
+if(APPLE)
+    target_link_libraries(kdeinit_plasma-netbook ${KDE4_PLASMA_LIBS} kworkspace ${KDE4_KIO_LIBS} ${KDE4_KFILE_LIBS} kephal plasmagenericshell ${QT_QTOPENGL_LIBRARY} ${QT_QTSCRIPT_LIBRARY})
+else()
+    target_link_libraries(kdeinit_plasma-netbook ${KDE4_PLASMA_LIBS} kworkspace ${KDE4_KIO_LIBS} ${X11_LIBRARIES} ${KDE4_KFILE_LIBS} kephal plasmagenericshell ${QT_QTOPENGL_LIBRARY} ${QT_QTSCRIPT_LIBRARY})
+    if(X11_Xrender_FOUND)
+       target_link_libraries(kdeinit_plasma-netbook ${X11_Xrender_LIB})
+    endif(X11_Xrender_FOUND)
+endif(APPLE)
 
 install(TARGETS kdeinit_plasma-netbook DESTINATION ${LIB_INSTALL_DIR})
 install(TARGETS plasma-netbook ${INSTALL_TARGETS_DEFAULT_ARGS})
diff --git plasma/netbook/shell/plasmaapp.cpp plasma/netbook/shell/plasmaapp.cpp
index c364238..f19329e 100644
--- plasma/netbook/shell/plasmaapp.cpp
+++ plasma/netbook/shell/plasmaapp.cpp
@@ -1027,6 +1027,7 @@ bool PlasmaApp::eventFilter(QObject * watched, QEvent *event)
     return false;
 }
 
+#ifdef Q_WS_X11
 bool PlasmaApp::x11EventFilter(XEvent *event)
 {
 
@@ -1061,6 +1062,13 @@ bool PlasmaApp::x11EventFilter(XEvent *event)
 
     return KUniqueApplication::x11EventFilter(event);
 }
+#else
+bool PlasmaApp::x11EventFilter(void *event)
+{
+    Q_UNUSED(event);
+    return false;
+}
+#endif
 
 bool PlasmaApp::hasForegroundWindows() const
 {
@@ -1113,7 +1121,12 @@ void PlasmaApp::controlBarVisibilityUpdate()
     //would be nice to avoid this
     QPoint cursorPos = QCursor::pos();
 
-    if (m_triggerZone.adjusted(-1, -1, 1, 1).contains(cursorPos) || hasForegroundWindows()) {
+    if (
+#ifdef Q_WS_X11
+        m_triggerZone.adjusted(-1, -1, 1, 1).contains(cursorPos) ||
+#endif
+        hasForegroundWindows())
+    {
         if (!m_controlBar->isVisible()) {
             destroyUnHideTrigger();
             Plasma::WindowEffects::slideWindow(m_controlBar, m_controlBar->location());
diff --git plasma/netbook/shell/plasmaapp.h plasma/netbook/shell/plasmaapp.h
index 8374e53..002a7da 100644
--- plasma/netbook/shell/plasmaapp.h
+++ plasma/netbook/shell/plasmaapp.h
@@ -94,7 +94,11 @@ public Q_SLOTS:
 
 protected:
     bool eventFilter(QObject *watched, QEvent *event);
+#ifdef Q_WS_X11
     bool x11EventFilter(XEvent *event);
+#else
+    bool x11EventFilter(void *event);
+#endif
     //true if the application has an active window
     bool hasForegroundWindows() const;
     //more strict than hasForegroundWindows, the mainview has to be not only active, but also on top of other windows
