diff --git CMakeLists.txt CMakeLists.txt
index 3d4287c..273dcdd 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -19,6 +19,7 @@ find_package(KDE4 4.9.4 REQUIRED)
 include(KDE4Defaults)
 
 option(WITH_XINERAMA "Xinerama support for multi-headed X displays" ON)
+option(BUILD_OXYGEN "Build the Oxygen style for Qt4/KDE4" ON)
 
 find_package(Strigi)
 
@@ -194,7 +195,7 @@ set_package_properties(QJSON PROPERTIES DESCRIPTION "Library to manage JSON obje
 
 
 include(ConfigureChecks.cmake)
-if(NOT WIN32)
+if(NOT WIN32 AND NOT APPLE)
   configure_file(config-unix.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/config-unix.h )
   configure_file(config-X11.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/config-X11.h )
   if(NOT CMAKE_INSTALL_PREFIX STREQUAL "/usr")
@@ -221,15 +222,17 @@ endif(${KDE_PLATFORM_PROFILE} STREQUAL "Desktop")
 
 # core apps
 if(NOT WIN32)
-   macro_optional_add_subdirectory( kcheckpass )
-   macro_optional_add_subdirectory( kwin )
-   macro_optional_add_subdirectory( ksmserver )
+   if(NOT APPLE)
+      macro_optional_add_subdirectory( kcheckpass )
+      macro_optional_add_subdirectory( kwin )
+      macro_optional_add_subdirectory( ksmserver )
 
-   find_package(JPEG REQUIRED)
-   find_package(PNG REQUIRED)
-   macro_optional_add_subdirectory( ksplash )
+      find_package(JPEG REQUIRED)
+      find_package(PNG REQUIRED)
+      macro_optional_add_subdirectory( ksplash )
 
-   macro_optional_add_subdirectory( powerdevil )
+      macro_optional_add_subdirectory( powerdevil )
+   endif(NOT APPLE)
    macro_optional_add_subdirectory( qguiplatformplugin_kde )
    if (NOT CMAKE_SYSTEM_NAME MATCHES Darwin)
       macro_optional_add_subdirectory( ksysguard )
@@ -238,14 +241,16 @@ endif(NOT WIN32)
 
 if(${KDE_PLATFORM_PROFILE} STREQUAL "Desktop")
    macro_optional_add_subdirectory(kcontrol)
-   macro_optional_add_subdirectory(klipper)
-   macro_optional_add_subdirectory(kmenuedit)
-   macro_optional_add_subdirectory(krunner)
-   macro_optional_add_subdirectory(solid-actions-kcm)
-   macro_optional_add_subdirectory(kstartupconfig)
-   macro_optional_add_subdirectory(freespacenotifier)
-   macro_optional_add_subdirectory(kscreensaver)
-   macro_optional_add_subdirectory(kinfocenter)
+   if(NOT APPLE)
+      macro_optional_add_subdirectory(klipper)
+      macro_optional_add_subdirectory(kmenuedit)
+      macro_optional_add_subdirectory(krunner)
+      macro_optional_add_subdirectory(solid-actions-kcm)
+      macro_optional_add_subdirectory(kstartupconfig)
+      macro_optional_add_subdirectory(freespacenotifier)
+      macro_optional_add_subdirectory(kscreensaver)
+      macro_optional_add_subdirectory(kinfocenter)
+   endif(NOT APPLE)
 
    # data
    macro_optional_add_subdirectory(doc)
@@ -254,7 +259,7 @@ if(${KDE_PLATFORM_PROFILE} STREQUAL "Desktop")
       macro_optional_add_subdirectory(ktouchpadenabler)
    endif (Q_WS_X11)
 
-   if(NOT WIN32)
+   if(NOT WIN32 AND NOT APPLE)
       macro_optional_add_subdirectory(kcminit)
       macro_optional_add_subdirectory(khotkeys)
       macro_optional_add_subdirectory(kwrited)
@@ -269,17 +274,19 @@ if(${KDE_PLATFORM_PROFILE} STREQUAL "Desktop")
    endif(X11_Xau_FOUND AND X11_Xdmcp_FOUND)
 endif(${KDE_PLATFORM_PROFILE} STREQUAL "Desktop")
 
-if(NOT WIN32)
+if(NOT WIN32 AND NOT APPLE)
     # data
     macro_optional_add_subdirectory(cursors)
 endif(NOT WIN32)
 
-macro_optional_add_subdirectory(plasma)
+if(NOT APPLE)
+    macro_optional_add_subdirectory(plasma)
+endif()
 macro_optional_add_subdirectory(statusnotifierwatcher)
 macro_optional_add_subdirectory(kstyles)
 
 ########### install files ###############
-if(NOT WIN32)
+if(NOT WIN32 AND NOT APPLE)
   install(PROGRAMS ${CMAKE_CURRENT_BINARY_DIR}/startkde DESTINATION ${BIN_INSTALL_DIR})
 endif(NOT WIN32)
 
@@ -352,9 +359,11 @@ install(FILES ${CMAKE_CURRENT_BINARY_DIR}/KDE4WorkspaceConfig.cmake
               ${CMAKE_CURRENT_BINARY_DIR}/KDE4WorkspaceConfigVersion.cmake
         DESTINATION ${_KDE4WorkspaceConfig_INSTALL_DIR})
 
-install(EXPORT kdeworkspaceLibraryTargets 
+if(NOT APPLE)
+    install(EXPORT kdeworkspaceLibraryTargets 
         NAMESPACE ${KDE4WORKSPACE_TARGET_PREFIX}
         DESTINATION ${_KDE4WorkspaceConfig_INSTALL_DIR}
         FILE KDE4WorkspaceLibraryTargets.cmake )
+endif()
 
 
diff --git libs/CMakeLists.txt libs/CMakeLists.txt
index c0576fe6e77eedd71ec548c8cf38dda819dfaa99..0f522ab59b1928d1e63f561b3b98e06d4d703910 100644
--- libs/CMakeLists.txt
+++ libs/CMakeLists.txt
@@ -1,19 +1,20 @@
-add_subdirectory(kephal)
-add_subdirectory(kworkspace)
-add_subdirectory(oxygen)
-add_subdirectory(plasmaclock)
+if(NOT APPLE)
+    add_subdirectory(kephal)
+    add_subdirectory(kworkspace)
+    add_subdirectory(plasmaclock)
 
-if(${KDE_PLATFORM_PROFILE} STREQUAL "Desktop")
-   add_subdirectory(kdm)
-   add_subdirectory(plasmagenericshell)
-   if(NOT WIN32)
-        add_subdirectory(taskmanager)
-   endif(NOT WIN32)
-endif(${KDE_PLATFORM_PROFILE} STREQUAL "Desktop")
+    if(${KDE_PLATFORM_PROFILE} STREQUAL "Desktop")
+       add_subdirectory(kdm)
+       add_subdirectory(plasmagenericshell)
+       if(NOT WIN32 AND NOT CMAKE_SYSTEM_NAME MATCHES Darwin)
+            add_subdirectory(taskmanager)
+       endif(NOT WIN32 AND NOT CMAKE_SYSTEM_NAME MATCHES Darwin)
+    endif(${KDE_PLATFORM_PROFILE} STREQUAL "Desktop")
 
-if(NOT WIN32)
-    if (NOT CMAKE_SYSTEM_NAME MATCHES Darwin)
+    if(NOT WIN32 AND NOT CMAKE_SYSTEM_NAME MATCHES Darwin)
         add_subdirectory(ksysguard)
-    endif (NOT CMAKE_SYSTEM_NAME MATCHES Darwin)
-endif(NOT WIN32)
-
+    endif(NOT WIN32 AND NOT CMAKE_SYSTEM_NAME MATCHES Darwin)
+endif()
+if(BUILD_OXYGEN)
+    add_subdirectory(oxygen)
+endif()
diff --git kstyles/CMakeLists.txt kstyles/CMakeLists.txt
index 260f65b784121c3455e1a3e129d92fe37f30d12b..34585c6e46be2fb465f1cce9a41f63e42f2d05fb 100644
--- kstyles/CMakeLists.txt
+++ kstyles/CMakeLists.txt
@@ -1,2 +1,4 @@
-add_subdirectory( oxygen )
+if(BUILD_OXYGEN)
+    add_subdirectory( oxygen )
+endif()
 add_subdirectory( themes )
