diff --git kcontrol/input/kmousedlg.ui kcontrol/input/kmousedlg.ui
index b48a606..c2dde85 100644
--- kcontrol/input/kmousedlg.ui
+++ kcontrol/input/kmousedlg.ui
@@ -76,6 +76,16 @@
          </property>
         </widget>
        </item>
+       <item>
+        <widget class="QCheckBox" name="cbWheelZooms">
+         <property name="whatsThis">
+          <string>Scrolling with the mouse wheel with the Control (or Command) key may zoom text, or else may accelerate the scrolling.</string>
+         </property>
+         <property name="text">
+          <string>Mo&amp;use wheel may zoom text</string>
+         </property>
+        </widget>
+       </item>
       </layout>
      </item>
      <item>
@@ -249,6 +259,7 @@
   <tabstop>rightHanded</tabstop>
   <tabstop>leftHanded</tabstop>
   <tabstop>cbScrollPolarity</tabstop>
+  <tabstop>cbWheelZooms</tabstop>
   <tabstop>doubleClick</tabstop>
   <tabstop>singleClick</tabstop>
   <tabstop>cb_pointershape</tabstop>
diff --git kcontrol/input/mouse.cpp kcontrol/input/mouse.cpp
index cebb174..d65db1c 100644
--- kcontrol/input/mouse.cpp
+++ kcontrol/input/mouse.cpp
@@ -180,6 +180,9 @@ MouseConfig::MouseConfig(QWidget *parent, const QVariantList &args)
     connect(generalTab->cbScrollPolarity, SIGNAL(clicked()), this, SLOT(changed()));
     connect(generalTab->cbScrollPolarity, SIGNAL(clicked()), this, SLOT(slotScrollPolarityChanged()));
 
+    connect(generalTab->cbWheelZooms, SIGNAL(clicked()), this, SLOT(changed()));
+    connect(generalTab->cbWheelZooms, SIGNAL(clicked()), this, SLOT(slotWheelZoomsChanged()));
+
     // Advanced tab
     advancedTab = new QWidget(0);
     advancedTab->setObjectName("Advanced Tab");
@@ -457,6 +460,7 @@ void MouseConfig::load()
   if ( generalTab->cbScrollPolarity->isEnabled() )
     generalTab->cbScrollPolarity->setEnabled(settings->handedEnabled);
   generalTab->cbScrollPolarity->setChecked( settings->reverseScrollPolarity );
+  generalTab->cbWheelZooms->setChecked( settings->wheelZooms );
 
   setAccel(settings->accelRate);
   setThreshold(settings->thresholdMove);
@@ -524,6 +528,7 @@ void MouseConfig::save()
 //  settings->changeCursor = generalTab->singleClick->isChecked();
   settings->changeCursor = generalTab->cb_pointershape->isChecked();
   settings->reverseScrollPolarity = generalTab->cbScrollPolarity->isChecked();
+  settings->wheelZooms = generalTab->cbWheelZooms->isChecked();
 
   settings->apply();
   KConfig config( "kcminputrc" );
@@ -568,6 +573,7 @@ void MouseConfig::defaults()
     generalTab->slAutoSelect->setValue( KDE_DEFAULT_AUTOSELECTDELAY == -1 ? 50 : KDE_DEFAULT_AUTOSELECTDELAY );
     generalTab->singleClick->setChecked( KDE_DEFAULT_SINGLECLICK );
     generalTab->cb_pointershape->setChecked(KDE_DEFAULT_CHANGECURSOR);
+    generalTab->cbWheelZooms->setChecked(KDE_DEFAULT_WHEEL_ZOOM);
     slotClick();
 
   mouseKeys->setChecked(false);
@@ -681,6 +687,7 @@ void MouseSettings::load(KConfig *config)
   singleClick = group.readEntry("SingleClick", KDE_DEFAULT_SINGLECLICK);
   autoSelectDelay = group.readEntry("AutoSelectDelay", KDE_DEFAULT_AUTOSELECTDELAY);
   changeCursor = group.readEntry("ChangeCursor", KDE_DEFAULT_CHANGECURSOR);
+  wheelZooms = group.readEntry("WheelMouseZooms", KDE_DEFAULT_WHEEL_ZOOM );
 }
 
 void MouseConfig::slotThreshChanged(int value)
@@ -796,6 +803,7 @@ void MouseSettings::save(KConfig *config)
   group.writeEntry("SingleClick", singleClick, KConfig::Persistent|KConfig::Global);
   group.writeEntry("AutoSelectDelay", autoSelectDelay, KConfig::Persistent|KConfig::Global);
   group.writeEntry("ChangeCursor", changeCursor,KConfig::Persistent|KConfig::Global);
+  group.writeEntry("WheelMouseZooms", wheelZooms, KConfig::Persistent|KConfig::Global);
   // This iterates through the various Logitech mice, if we have support.
 #ifdef HAVE_LIBUSB
   LogitechMouse *logitechMouse;
@@ -812,6 +820,11 @@ void MouseConfig::slotScrollPolarityChanged()
   settings->m_handedNeedsApply = true;
 }
 
+void MouseConfig::slotWheelZoomsChanged()
+{
+  settings->wheelZooms = generalTab->cbWheelZooms->isChecked();
+}
+
 void MouseConfig::slotSmartSliderEnabling()
 {
   bool enabled = generalTab->singleClick->isChecked() ? generalTab->cbAutoSelect->isChecked() : false;
diff --git kcontrol/input/mouse.h kcontrol/input/mouse.h
index d926a99..dcc160f 100644
--- kcontrol/input/mouse.h
+++ kcontrol/input/mouse.h
@@ -88,6 +88,7 @@ public:
  bool changeCursor;
  int wheelScrollLines;
  bool reverseScrollPolarity;
+ bool wheelZooms;
 
  #ifdef HAVE_LIBUSB
  // TODO: In Qt4, replace with a better container.
@@ -112,6 +113,7 @@ private Q_SLOTS:
   /** No descriptions */
   void slotHandedChanged(int val);
   void slotScrollPolarityChanged();
+  void slotWheelZoomsChanged();
   void checkAccess();
   void slotThreshChanged(int value);
   void slotDragStartDistChanged(int value);
