diff --git a/CMakeLists.txt b/CMakeLists.txt
index a2475153b97c40b25a4be6be2af3b58dca907601..8f434b30d042d4a6af324a43af0a1dedf6307c59 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -4,8 +4,8 @@ project(PlasmaIntegration)
 set(PROJECT_VERSION "5.15.80")
 set(PROJECT_VERSION_MAJOR 5)
 
-set(QT_MIN_VERSION "5.12.0")
-set(KF5_MIN_VERSION "5.66.0")
+set(QT_MIN_VERSION "5.9.8")
+set(KF5_MIN_VERSION "5.60.0")
 
 find_package(ECM ${KF5_MIN_VERSION}  REQUIRED NO_MODULE)
 set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules ${ECM_MODULE_PATH} ${ECM_KDE_MODULE_DIR})
diff --git a/src/platformtheme/khintssettings.cpp b/src/platformtheme/khintssettings.cpp
index d33a058955e53ad27c5d4dfc198d9f0b8803cf21..128e5c5e8615e74b80f6a540f92bccf54a6c2e0d 100644
--- a/src/platformtheme/khintssettings.cpp
+++ b/src/platformtheme/khintssettings.cpp
@@ -153,7 +153,9 @@ KHintsSettings::KHintsSettings(KSharedConfig::Ptr kdeglobals)
 
     updateShowIconsInMenuItems(cg);
 
+#if QT_VERSION >= QT_VERSION_CHECK(5, 11, 0)
     m_hints[QPlatformTheme::ShowShortcutsInContextMenus] = true;
+#endif
 
     QMetaObject::invokeMethod(this, "delayedDBusConnects", Qt::QueuedConnection);
     QMetaObject::invokeMethod(this, "setupIconLoader", Qt::QueuedConnection);
diff --git a/src/platformtheme/kdirselectdialog.cpp b/src/platformtheme/kdirselectdialog.cpp
index b830961b775e8aa0f50e43011624d70747282345..b829fa892024fcd327f3ce7a4191d002a2ebcadd 100644
--- a/src/platformtheme/kdirselectdialog.cpp
+++ b/src/platformtheme/kdirselectdialog.cpp
@@ -54,7 +54,11 @@
 #include <kurlcompletion.h>
 #include <kurlpixmapprovider.h>
 #include <kfilewidget.h>
+#include <kcompletion_version.h>
+#include <kio_version.h>
+#if KIO_VERSION >= QT_VERSION_CHECK(5, 66, 0)
 #include <KFileUtils>
+#endif
 
 #include "kfiletreeview_p.h"
 #include <kfileplacesview.h>
@@ -132,7 +136,11 @@ void KDirSelectDialog::Private::slotMkdir()
     QString where = m_parent->url().toDisplayString(QUrl::PreferLocalFile);
     QString name = i18nc("folder name", "New Folder");
     if (m_parent->url().isLocalFile() && QFileInfo::exists(m_parent->url().toLocalFile() + QLatin1Char('/') + name)) {
+#if KIO_VERSION >= QT_VERSION_CHECK(5, 66, 0)
         name = KFileUtils::suggestName(m_parent->url(), name);
+#else
+        name = KIO::suggestName(m_parent->url(), name);
+#endif
     }
 
     QString directory = KIO::encodeFileName(QInputDialog::getText(m_parent, i18nc("@title:window", "New Folder"),
@@ -328,9 +336,13 @@ KDirSelectDialog::KDirSelectDialog(const QUrl &startDir, bool localOnly, QWidget
     d->m_urlCombo->setLayoutDirection(Qt::LeftToRight);
     d->m_urlCombo->setSizeAdjustPolicy(QComboBox::AdjustToMinimumContentsLength);
     d->m_urlCombo->setTrapReturnKey(true);
+#if KCOMPLETION_VERSION >= QT_VERSION_CHECK(5, 66, 0)
     d->m_urlCombo->setIconProvider([](const QString &name) {
         return QIcon::fromTheme(KIO::iconNameForUrl(QUrl::fromUserInput(name)));
     });
+#else
+    d->m_urlCombo->setPixmapProvider(new KUrlPixmapProvider());
+#endif
     KUrlCompletion *comp = new KUrlCompletion();
     comp->setMode(KUrlCompletion::DirCompletion);
     d->m_urlCombo->setCompletionObject(comp, true);
