diff --git a/src/platformtheme/kfontsettingsdata.cpp b/src/platformtheme/kfontsettingsdata.cpp
index c8e3e81457b15f68ce903d3c552e4ef4fc6db740..6cc0f7c42933456ead5d9c2f88158e312d9a2ba1 100644
--- a/src/platformtheme/kfontsettingsdata.cpp
+++ b/src/platformtheme/kfontsettingsdata.cpp
@@ -27,6 +27,7 @@
 #include <QDBusConnection>
 #include <QDBusReply>
 #include <qpa/qwindowsysteminterface.h>
+#include <QDebug>
 
 #include <ksharedconfig.h>
 #include <kconfiggroup.h>
@@ -39,8 +39,20 @@ static inline bool checkUsePortalSupport()
 KFontSettingsData::KFontSettingsData()
     : QObject(nullptr)
     , mUsePortal(checkUsePortalSupport())
-    , mKdeGlobals(KSharedConfig::openConfig())
+    , mKdeGlobals(nullptr)
 {
+    if (qEnvironmentVariableIsSet("QT_QPA_PLATFORMTHEME_CONFIG_FILE")) {
+        const auto fname = QString::fromUtf8(qgetenv("QT_QPA_PLATFORMTHEME_CONFIG_FILE"));
+        mKdeGlobals = KSharedConfig::openConfig(fname, KConfig::NoGlobals);
+        const auto foundFile = QStandardPaths::locate(mKdeGlobals->locationType(), mKdeGlobals->name());
+        if (!mKdeGlobals || foundFile.isEmpty()) {
+            qWarning() << "WARNING: could not open config file" << fname
+                << "in" << QStandardPaths::standardLocations(mKdeGlobals->locationType());
+        }
+    } 
+    if (!mKdeGlobals) {
+        mKdeGlobals = KSharedConfig::openConfig();
+    }
     QMetaObject::invokeMethod(this, "delayedDBusConnects", Qt::QueuedConnection);
 
     for (int i = 0; i < FontTypesCount; ++i) {
diff --git a/src/platformtheme/khintssettings.cpp b/src/platformtheme/khintssettings.cpp
index d33a058955e53ad27c5d4dfc198d9f0b8803cf21..d52e8dd5f4af3fcc8af6523499edf59368e4288b 100644
--- a/src/platformtheme/khintssettings.cpp
+++ b/src/platformtheme/khintssettings.cpp
@@ -87,7 +87,18 @@ KHintsSettings::KHintsSettings(KSharedConfig::Ptr kdeglobals)
     , mUsePortal(checkUsePortalSupport())
 {
     if (!mKdeGlobals) {
-        mKdeGlobals = KSharedConfig::openConfig();
+        if (qEnvironmentVariableIsSet("QT_QPA_PLATFORMTHEME_CONFIG_FILE")) {
+            const auto fname = QString::fromUtf8(qgetenv("QT_QPA_PLATFORMTHEME_CONFIG_FILE"));
+            mKdeGlobals = KSharedConfig::openConfig(fname, KConfig::NoGlobals);
+            const auto foundFile = QStandardPaths::locate(mKdeGlobals->locationType(), mKdeGlobals->name());
+            if (!mKdeGlobals || foundFile.isEmpty()) {
+                qWarning() << "WARNING: could not open config file" << fname
+                    << "in" << QStandardPaths::standardLocations(mKdeGlobals->locationType());
+            }
+        } 
+        if (!mKdeGlobals) {
+            mKdeGlobals = KSharedConfig::openConfig();
+        }
     }
     KConfigGroup cg(mKdeGlobals, "KDE");
 
