diff --git a/src/platformtheme/kfontsettingsdata.cpp b/src/platformtheme/kfontsettingsdata.cpp
index ea959162efb50cbd0c928f520f5b56f74db0e953..4688e226f208d46b2d9f6b5e282c2207a70f583e 100644
--- a/src/platformtheme/kfontsettingsdata.cpp
+++ b/src/platformtheme/kfontsettingsdata.cpp
@@ -26,14 +26,27 @@
 #include <QDBusMessage>
 #include <QDBusConnection>
 #include <qpa/qwindowsysteminterface.h>
+#include <QDebug>
 
 #include <ksharedconfig.h>
 #include <kconfiggroup.h>
 
 KFontSettingsData::KFontSettingsData()
     : QObject(nullptr),
-    mKdeGlobals(KSharedConfig::openConfig())
+    mKdeGlobals(nullptr)
 {
+    if (qEnvironmentVariableIsSet("QT_QPA_PLATFORMTHEME_CONFIG_FILE")) {
+        const auto fname = QString::fromUtf8(qgetenv("QT_QPA_PLATFORMTHEME_CONFIG_FILE"));
+        mKdeGlobals = KSharedConfig::openConfig(fname, KConfig::NoGlobals);
+        const auto foundFile = QStandardPaths::locate(mKdeGlobals->locationType(), mKdeGlobals->name());
+        if (!mKdeGlobals || foundFile.isEmpty()) {
+            qWarning() << "WARNING: could not open config file" << fname
+                << "in" << QStandardPaths::standardLocations(mKdeGlobals->locationType());
+        }
+    } 
+    if (!mKdeGlobals) {
+        mKdeGlobals = KSharedConfig::openConfig();
+    }
     QMetaObject::invokeMethod(this, "delayedDBusConnects", Qt::QueuedConnection);
 
     for (int i = 0; i < FontTypesCount; ++i) {
diff --git a/src/platformtheme/khintssettings.cpp b/src/platformtheme/khintssettings.cpp
index bb4b412485ad8e639567c0fd1e6465a1584486d7..f7be8aca69fce7104d79ee3c98398d6c4a4e719e 100644
--- a/src/platformtheme/khintssettings.cpp
+++ b/src/platformtheme/khintssettings.cpp
@@ -61,7 +61,18 @@ KHintsSettings::KHintsSettings(KSharedConfig::Ptr kdeglobals)
     , mKdeGlobals(kdeglobals)
 {
     if (!mKdeGlobals) {
-        mKdeGlobals = KSharedConfig::openConfig();
+        if (qEnvironmentVariableIsSet("QT_QPA_PLATFORMTHEME_CONFIG_FILE")) {
+            const auto fname = QString::fromUtf8(qgetenv("QT_QPA_PLATFORMTHEME_CONFIG_FILE"));
+            mKdeGlobals = KSharedConfig::openConfig(fname, KConfig::NoGlobals);
+            const auto foundFile = QStandardPaths::locate(mKdeGlobals->locationType(), mKdeGlobals->name());
+            if (!mKdeGlobals || foundFile.isEmpty()) {
+                qWarning() << "WARNING: could not open config file" << fname
+                    << "in" << QStandardPaths::standardLocations(mKdeGlobals->locationType());
+            }
+        } 
+        if (!mKdeGlobals) {
+            mKdeGlobals = KSharedConfig::openConfig();
+        }
     }
     KConfigGroup cg(mKdeGlobals, "KDE");
 
