diff --git a/CMakeLists.txt b/CMakeLists.txt
index a2475153b97c40b25a4be6be2af3b58dca907601..8f434b30d042d4a6af324a43af0a1dedf6307c59 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -4,8 +4,8 @@ project(PlasmaIntegration)
 set(PROJECT_VERSION "5.15.80")
 set(PROJECT_VERSION_MAJOR 5)
 
-set(QT_MIN_VERSION "5.11.0")
-set(KF5_MIN_VERSION "5.54.0")
+set(QT_MIN_VERSION "5.9.0")
+set(KF5_MIN_VERSION "5.52.0")
 
 find_package(ECM ${KF5_MIN_VERSION}  REQUIRED NO_MODULE)
 set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules ${ECM_MODULE_PATH} ${ECM_KDE_MODULE_DIR})
diff --git a/src/platformtheme/khintssettings.cpp b/src/platformtheme/khintssettings.cpp
index d33a058955e53ad27c5d4dfc198d9f0b8803cf21..128e5c5e8615e74b80f6a540f92bccf54a6c2e0d 100644
--- a/src/platformtheme/khintssettings.cpp
+++ b/src/platformtheme/khintssettings.cpp
@@ -153,7 +153,9 @@ KHintsSettings::KHintsSettings(KSharedConfig::Ptr kdeglobals)
 
     updateShowIconsInMenuItems(cg);
 
+#if QT_VERSION >= QT_VERSION_CHECK(5, 11, 0)
     m_hints[QPlatformTheme::ShowShortcutsInContextMenus] = true;
+#endif
 
     QMetaObject::invokeMethod(this, "delayedDBusConnects", Qt::QueuedConnection);
     QMetaObject::invokeMethod(this, "setupIconLoader", Qt::QueuedConnection);
