diff --git a/CMakeLists.txt b/CMakeLists.txt
index 49414bfa9b16219897208a43bad435bed9bbba5f..bbe7dc71c8fe804c8b5e5411f31160a21f050777 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -39,7 +39,7 @@ set_package_properties(XCB PROPERTIES
     PURPOSE "Required to pass style properties to native Windows on X11 Platform"
 )
 
-find_package(Breeze ${PROJECT_VERSION} CONFIG)
+find_package(Breeze CONFIG)
 set_package_properties(Breeze PROPERTIES
                        TYPE REQUIRED
                        PURPOSE "For setting the default QStyle name")

