diff --git a/libcxx/include/__config b/libcxx/include/__config
index 97e33f3..c4a71bf 100644
--- a/libcxx/include/__config
+++ b/libcxx/include/__config
@@ -22,6 +22,12 @@
 #pragma GCC system_header
 #endif
 
+// we are not going to use Apple Availability testing in these headers
+#define _LIBCPP_DISABLE_AVAILABILITY
+// make per TU ABI insulation enabled by default; it can be overridden by
+// -D_LIBCPP_HIDE_FROM_ABI_PER_TU=0
+#define _LIBCPP_HIDE_FROM_ABI_PER_TU_BY_DEFAULT
+
 #ifdef __cplusplus
 
 #ifdef __GNUC__
