diff --git a/qtbase/src/corelib/Qt5CoreConfigExtras.cmake.in b/qtbase/src/corelib/Qt5CoreConfigExtras.cmake.in
index a5ed8b2..99e697c 100644
--- a/qtbase/src/corelib/Qt5CoreConfigExtras.cmake.in
+++ b/qtbase/src/corelib/Qt5CoreConfigExtras.cmake.in
@@ -114,7 +114,9 @@ set(Qt5_DISABLED_FEATURES
 
 set_property(TARGET Qt5::Core APPEND PROPERTY INTERFACE_COMPILE_DEFINITIONS $<$<NOT:$<CONFIG:Debug>>:QT_NO_DEBUG>)
 
-set_property(TARGET Qt5::Core PROPERTY INTERFACE_COMPILE_FEATURES cxx_decltype)
+if (NOT APPLE)
+    set_property(TARGET Qt5::Core PROPERTY INTERFACE_COMPILE_FEATURES cxx_decltype)
+endif ()
 
 !!IF contains(QT_CONFIG, reduce_exports)
 set(QT_VISIBILITY_AVAILABLE \"True\")
