diff --git src/kiod/CMakeLists.txt src/kiod/CMakeLists.txt
index 51d7f23eb00b8428b7dc30b05e3c8e6d681e2217..a65da0c1a04bfaff683558952962cd4ac8c78c47 100644
--- src/kiod/CMakeLists.txt
+++ src/kiod/CMakeLists.txt
@@ -1,4 +1,9 @@
+include(ECMMarkNonGuiExecutable)
+
 set(kiod_SRCS kiod_main.cpp)
+if (APPLE)
+    set(kiod_SRCS ${kiod_SRCS} kiod_agent.mm)
+endif()
 
 add_executable(kiod5 ${kiod_SRCS})
 
@@ -11,6 +16,13 @@ target_link_libraries(kiod5
    Qt5::Widgets     # QApplication
 )
 
+if (APPLE)
+    target_link_libraries(kiod5 "-framework AppKit -framework CoreFoundation")
+
+    # Mark it as non-gui so we won't create an app bundle on Mac OS X
+    ecm_mark_nongui_executable(kiod5)
+endif ()
+
 install(TARGETS kiod5 DESTINATION ${KDE_INSTALL_LIBEXECDIR_KF5})
 
 configure_file(org.kde.kiod5.service.in
diff --git src/kiod/kiod_main.cpp src/kiod/kiod_main.cpp
index 79d4cd2dac6be34b24b59c2b8ec2087428f8629a..771043cdb20c807ee32163e9ed358bcff4cf0f47 100644
--- src/kiod/kiod_main.cpp
+++ src/kiod/kiod_main.cpp
@@ -94,6 +94,12 @@ extern Q_DBUS_EXPORT void qDBusAddSpyHook(void (*)(const QDBusMessage&));
 
 int main(int argc, char *argv[])
 {
+#ifdef Q_OS_MACOS
+    // do the "early" step to make this an "agent" application:
+    // set the LSUIElement InfoDict key programmatically.
+    extern void makeAgentApplication();
+    makeAgentApplication();
+#endif
     qunsetenv("SESSION_MANAGER"); // disable session management
 
     QApplication app(argc, argv); // GUI needed for kpasswdserver's dialogs
@@ -120,6 +126,14 @@ int main(int argc, char *argv[])
     self(); // create it in this thread
     qDBusAddSpyHook(messageFilter);
 
+#ifdef Q_OS_MACOS
+    // In the case of kiod5 we need to confirm the agent nature,
+    // possibly because of how things have been set up after creating
+    // the QApplication instance. Failure to do this will make disable
+    // text input into dialogs we may post.
+    extern void setAgentActivationPolicy();
+    setAgentActivationPolicy();
+#endif
     return app.exec();
 }
 
diff --git src/kpasswdserver/kpasswdserver.cpp src/kpasswdserver/kpasswdserver.cpp
index c108db92775a0033c832d503ada5c195a2e2f8cc..7032ad1b3a7ec3235cf1c025529f2a2ad1b7698e 100644
--- src/kpasswdserver/kpasswdserver.cpp
+++ src/kpasswdserver/kpasswdserver.cpp
@@ -894,11 +894,15 @@ void KPasswdServer::showPasswordDialog (KPasswdServer::Request* request)
     if (info.getExtraField(AUTHINFO_EXTRAFIELD_ANONYMOUS).isValid () && password.isEmpty() && username.isEmpty())
         dlg->setAnonymousMode(info.getExtraField(AUTHINFO_EXTRAFIELD_ANONYMOUS).toBool());
 
+#ifndef Q_OS_MACOS
 #ifndef Q_WS_WIN
     KWindowSystem::setMainWindow(dlg, request->windowId);
 #else
     KWindowSystem::setMainWindow(dlg, (HWND)request->windowId);
 #endif
+#else
+    KWindowSystem::forceActiveWindow(dlg->winId(), 0);
+#endif
 
     qCDebug(category) << "Showing password dialog" << dlg << ", window-id=" << request->windowId;
     m_authInProgress.insert(dlg, request);
