--- src/filewidgets/orig.kurlnavigatorbuttonbase.cpp	2016-05-24 14:39:23.000000000 +0200
+++ src/filewidgets/kurlnavigatorbuttonbase.cpp	2016-05-24 14:39:46.000000000 +0200
@@ -39,6 +39,7 @@
     setFocusPolicy(Qt::TabFocus);
     setSizePolicy(QSizePolicy::Maximum, QSizePolicy::Fixed);
     setMinimumHeight(parent->minimumHeight());
+    setAttribute(Qt::WA_LayoutUsesWidgetRect);
 
     connect(this, SIGNAL(pressed()), parent, SLOT(requestActivation()));
 }
