--- src/ioslaves/http/orig.CMakeLists.txt	2015-10-05 10:11:41.000000000 +0200
+++ src/ioslaves/http/CMakeLists.txt	2015-11-14 23:13:06.000000000 +0100
@@ -39,6 +39,11 @@
 
 
 add_executable(kio_http_cache_cleaner ${kio_http_cache_cleaner_SRCS})
+if (APPLE)
+    include(ECMMarkNonGuiExecutable)
+    # Mark it as non-gui so we won't create an app bundle on Mac OS X
+    ecm_mark_nongui_executable(kio_http_cache_cleaner)
+endif ()
 
 target_link_libraries(kio_http_cache_cleaner
    Qt5::DBus
