diff --git src/core/kmountpoint.cpp src/core/kmountpoint.cpp
index 94de46854224df7dec7afe0a7a59955cc6bf329d..6e179f42bf92c551d371d30b88f376e35fcebf52 100644
--- src/core/kmountpoint.cpp
+++ src/core/kmountpoint.cpp
@@ -416,7 +416,7 @@ KMountPoint::Ptr KMountPoint::List::findByPath(const QString &path) const
 {
 #ifndef Q_OS_WIN
     /* If the path contains symlinks, get the real name */
-    QFileInfo fileinfo(path);
+    QFileInfo fileinfo = Q_UNLIKELY(path.isEmpty()) ? QFileInfo() : QFileInfo(path);
     const QString realname = fileinfo.exists()
                              ? fileinfo.canonicalFilePath()
                              : fileinfo.absolutePath(); //canonicalFilePath won't work unless file exists
@@ -424,8 +424,11 @@ KMountPoint::Ptr KMountPoint::List::findByPath(const QString &path) const
     const QString realname = QDir::fromNativeSeparators(QDir(path).absolutePath());
 #endif
 
-    int max = 0;
     KMountPoint::Ptr result;
+    if (Q_UNLIKELY(realname.isEmpty())) {
+        return result;
+    }
+    int max = 0;
     for (const_iterator it = begin(); it != end(); ++it) {
         const QString mountpoint = (*it)->d->mountPoint;
         const int length = mountpoint.length();
