diff --git src/filewidgets/kfilewidget.cpp src/filewidgets/kfilewidget.cpp
index 41ac5a7..4c08f0c 100644
--- src/filewidgets/kfilewidget.cpp
+++ src/filewidgets/kfilewidget.cpp
@@ -70,6 +70,7 @@
 #include <QPushButton>
 #include <QStandardPaths>
 #include <qmimedatabase.h>
+#include <QStyleFactory>
 
 #include <kshell.h>
 #include <kmessagebox.h>
@@ -502,6 +503,19 @@ KFileWidget::KFileWidget(const QUrl &_startDir, QWidget *parent)
     menu->addAction(coll->action(QStringLiteral("preview")));
 
     menu->setDelayed(false);
+    if (!qEnvironmentVariableIsSet("KIOFILEWIDGET_FORCE_QTCURVE") && QApplication::style()->objectName() == QLatin1String("qtcurve")) {
+#ifdef Q_OS_MACOS
+        static QPointer<QStyle> fusionStyle = QStyleFactory::create("macintosh");
+        if (!fusionStyle) {
+            fusionStyle = QStyleFactory::create("fusion");
+        }
+#else
+        static QPointer<QStyle> fusionStyle = QStyleFactory::create("fusion");
+#endif
+        if (fusionStyle) {
+            menu->menu()->setStyle(fusionStyle);
+        }
+    }
     connect(menu->menu(), SIGNAL(aboutToShow()),
             d->ops, SLOT(updateSelectionDependentActions()));
 
