--- src/ioslaves/http/kcookiejar/orig.kcookiewin.cpp	2016-05-30 17:35:52.000000000 +0200
+++ src/ioslaves/http/kcookiejar/kcookiewin.cpp	2016-05-30 17:35:38.000000000 +0200
@@ -65,6 +65,10 @@
     setObjectName(QStringLiteral("cookiealert"));
     setWindowTitle(i18n("Cookie Alert"));
     setWindowIcon(QIcon::fromTheme(QStringLiteral("preferences-web-browser-cookies"), windowIcon()));
+#ifdef Q_OS_OSX
+    // make sure the user notices our dialog; windowIds() is useless on this platform
+    KWindowSystem::forceActiveWindow(winId(), 0);
+#else
     // all cookies in the list should have the same window at this time, so let's take the first
     if (cookieList.first().windowIds().count() > 0) {
         KWindowSystem::setMainWindow(this, cookieList.first().windowIds().first());
@@ -73,6 +77,7 @@
         KWindowSystem::setState(winId(), NET::KeepAbove);
         KUserTimestamp::updateUserTimestamp();
     }
+#endif
 
     const int count = cookieList.count();
     const KHttpCookie &cookie = cookieList.first();
