--- orig.CMakeLists.txt	2017-05-01 23:21:07.000000000 +0200
+++ CMakeLists.txt	2017-05-01 23:35:23.000000000 +0200
@@ -56,8 +56,8 @@
     ecm_install_po_files_as_qm(po)
 endif()
 add_subdirectory(src)
-add_subdirectory(autotests)
-add_subdirectory(tests)
+# add_subdirectory(autotests)
+# add_subdirectory(tests)
 
 # create a Config.cmake and a ConfigVersion.cmake file and install them
 set(CMAKECONFIG_INSTALL_DIR "${CMAKECONFIG_INSTALL_PREFIX}/KF5Wayland")
--- src/tools/orig.CMakeLists.txt	2017-03-03 14:09:38.000000000 +0100
+++ src/tools/CMakeLists.txt	2017-05-01 23:36:52.000000000 +0200
@@ -1,4 +1,4 @@
-add_subdirectory(testserver)
+# add_subdirectory(testserver)
 
 include(ECMMarkAsTest)
 
