diff --git autotests/kbookmarkmenutest.cpp autotests/kbookmarkmenutest.cpp
index 311a5a6924aa884dfd233f5c31ee7cdb6a0f0370..8f31339c3e668967b980c532065ca6a6d66379c6 100644
--- autotests/kbookmarkmenutest.cpp
+++ autotests/kbookmarkmenutest.cpp
@@ -25,7 +25,7 @@
 #include <QDebug>
 #include <QMenu>
 #include <QStandardPaths>
-#include <QStringLiteral>
+#include <QString>
 #include <QTest>
 
 class TestKBookmarkOwner;
