diff --git b/projects/libcxxabi/src/orig.CMakeLists.txt a/projects/libcxxabi/src/CMakeLists.txt
index 948797b..3999ae8 100644
--- b/projects/libcxxabi/src/orig.CMakeLists.txt
+++ a/projects/libcxxabi/src/CMakeLists.txt
@@ -190,19 +190,21 @@ if (LIBCXXABI_ENABLE_SHARED)
   endif()
 
   # -exported_symbols_list is only available on Apple platforms
-  if (APPLE)
-    target_link_libraries(cxxabi_shared PRIVATE "-Wl,-exported_symbols_list,${CMAKE_CURRENT_SOURCE_DIR}/../lib/itanium-base.exp")
-
-    if (LIBCXXABI_ENABLE_NEW_DELETE_DEFINITIONS)
-      target_link_libraries(cxxabi_shared PRIVATE "-Wl,-exported_symbols_list,${CMAKE_CURRENT_SOURCE_DIR}/../lib/new-delete.exp")
-    endif()
-
-    if (LIBCXXABI_USE_SJLJ_EXCEPTIONS)
-      target_link_libraries(cxxabi_shared PRIVATE "-Wl,-exported_symbols_list,${CMAKE_CURRENT_SOURCE_DIR}/../lib/personality-sjlj.exp")
-    else()
-      target_link_libraries(cxxabi_shared PRIVATE "-Wl,-exported_symbols_list,${CMAKE_CURRENT_SOURCE_DIR}/../lib/personality-v0.exp")
-    endif()
-  endif()
+# Don't hide any symbols, like in libcxx 8.0.0; not all to-be-exported symbols from itanium-base.exp are
+# available on all OS X versions.
+#   if (APPLE)
+#     target_link_libraries(cxxabi_shared PRIVATE "-Wl,-exported_symbols_list,${CMAKE_CURRENT_SOURCE_DIR}/../lib/itanium-base.exp")
+# 
+#     if (LIBCXXABI_ENABLE_NEW_DELETE_DEFINITIONS)
+#       target_link_libraries(cxxabi_shared PRIVATE "-Wl,-exported_symbols_list,${CMAKE_CURRENT_SOURCE_DIR}/../lib/new-delete.exp")
+#     endif()
+# 
+#     if (LIBCXXABI_USE_SJLJ_EXCEPTIONS)
+#       target_link_libraries(cxxabi_shared PRIVATE "-Wl,-exported_symbols_list,${CMAKE_CURRENT_SOURCE_DIR}/../lib/personality-sjlj.exp")
+#     else()
+#       target_link_libraries(cxxabi_shared PRIVATE "-Wl,-exported_symbols_list,${CMAKE_CURRENT_SOURCE_DIR}/../lib/personality-v0.exp")
+#     endif()
+#   endif()
 endif()
 
 # Build the static library.
