diff --git b/projects/libcxx/include/__config.orig a/projects/libcxx/include/__config
index 1ecced9..2217a96 100644
--- b/projects/libcxx/include/__config.orig
+++ a/projects/libcxx/include/__config
@@ -20,6 +20,15 @@
 #pragma GCC system_header
 #endif
 
+// we are not going to use Apple Availability testing in these headers
+#ifndef _LIBCPP_DISABLE_AVAILABILITY
+#define _LIBCPP_DISABLE_AVAILABILITY
+#endif
+// make per TU ABI insulation enabled by default; it can be overridden by
+// -D_LIBCPP_HIDE_FROM_ABI_PER_TU=0
+#define _LIBCPP_HIDE_FROM_ABI_PER_TU_BY_DEFAULT
+
+
 #ifdef __cplusplus
 
 #ifdef __GNUC__
