diff --git b/projects/libcxxabi/src/../lib/orig.itanium-base.exp a/projects/libcxxabi/src/../lib/itanium-base.exp
index ff0f812..fd9ac82 100644
--- b/projects/libcxxabi/src/../lib/orig.itanium-base.exp
+++ a/projects/libcxxabi/src/../lib/itanium-base.exp
@@ -7,7 +7,6 @@ __ZTIDh
 __ZTIDi
 __ZTIDn
 __ZTIDs
-__ZTIDu
 __ZTIe
 __ZTIf
 __ZTIg
@@ -26,7 +25,6 @@ __ZTIPDh
 __ZTIPDi
 __ZTIPDn
 __ZTIPDs
-__ZTIPDu
 __ZTIPe
 __ZTIPf
 __ZTIPg
@@ -41,7 +39,6 @@ __ZTIPKDh
 __ZTIPKDi
 __ZTIPKDn
 __ZTIPKDs
-__ZTIPKDu
 __ZTIPKe
 __ZTIPKf
 __ZTIPKg
@@ -84,7 +81,6 @@ __ZTSDh
 __ZTSDi
 __ZTSDn
 __ZTSDs
-__ZTSDu
 __ZTSe
 __ZTSf
 __ZTSg
@@ -103,7 +99,6 @@ __ZTSPDh
 __ZTSPDi
 __ZTSPDn
 __ZTSPDs
-__ZTSPDu
 __ZTSPe
 __ZTSPf
 __ZTSPg
@@ -118,7 +113,6 @@ __ZTSPKDh
 __ZTSPKDi
 __ZTSPKDn
 __ZTSPKDs
-__ZTSPKDu
 __ZTSPKe
 __ZTSPKf
 __ZTSPKg
