diff --git b/orig.CMakeLists.txt a/CMakeLists.txt
index 8635159..1c83731 100644
--- b/orig.CMakeLists.txt
+++ a/CMakeLists.txt
@@ -48,6 +48,11 @@ if (CMAKE_GENERATOR STREQUAL "Xcode" AND NOT CMAKE_OSX_ARCHITECTURES)
   set(CMAKE_OSX_ARCHITECTURES "x86_64")
 endif()
 
+if (NOT DEFINED LIBCXX_INSTALL_PREFIX)
+	set(LIBCXX_INSTALL_PREFIX "/usr")
+endif()
+set(LIBCXX_INSTALL_PREFIX ${LIBCXX_INSTALL_PREFIX} CACHE PATH "Location where libc++ and libc++abi will be installed")
+
 project(LLVM
   VERSION ${LLVM_VERSION_MAJOR}.${LLVM_VERSION_MINOR}.${LLVM_VERSION_PATCH}
   LANGUAGES C CXX ASM)
diff --git b/projects/libcxx/src/orig.CMakeLists.txt a/projects/libcxx/src/CMakeLists.txt
index 31cd243..26f55b9 100644
--- b/projects/libcxx/src/orig.CMakeLists.txt
+++ a/projects/libcxx/src/CMakeLists.txt
@@ -1,4 +1,4 @@
-set(LIBCXX_LIB_CMAKEFILES_DIR "${CMAKE_CURRENT_BINARY_DIR}${CMAKE_FILES_DIRECTORY}"  PARENT_SCOPE)
+set(LIBCXX_LIB_CMAKEFILES_DIR "${CMAKE_BINARY_DIR}/lib${CMAKE_FILES_DIRECTORY}"  PARENT_SCOPE)
 
 # Get sources
 set(LIBCXX_SOURCES
@@ -240,7 +240,7 @@ if (LIBCXX_ENABLE_SHARED)
       COMPILE_FLAGS "${LIBCXX_COMPILE_FLAGS}"
       LINK_FLAGS    "${LIBCXX_LINK_FLAGS}"
       OUTPUT_NAME   "c++"
-      VERSION       "${LIBCXX_ABI_VERSION}.0"
+      VERSION       "${VERSION_ACCORDING_TO_MACPORTS}"
       SOVERSION     "${LIBCXX_ABI_VERSION}"
       DEFINE_SYMBOL ""
   )
@@ -285,6 +285,8 @@ if (LIBCXX_ENABLE_SHARED)
       set(RE_EXPORT_LIST "${CMAKE_CURRENT_SOURCE_DIR}/../lib/libc++abi.v${LIBCXX_LIBCPPABI_VERSION}.exp")
     endif()
     target_link_libraries(cxx_shared PRIVATE
+      "-current_version ${VERSION_ACCORDING_TO_MACPORTS}"
+      "-install_name ${LIBCXX_INSTALL_PREFIX}/lib/libc++.1.dylib"
       "-Wl,-unexported_symbols_list,${CMAKE_CURRENT_SOURCE_DIR}/../lib/libc++unexp.exp"
       "-Wl,-reexported_symbols_list,${RE_EXPORT_LIST}"
       "-Wl,-force_symbols_not_weak_list,${CMAKE_CURRENT_SOURCE_DIR}/../lib/notweak.exp"
@@ -435,23 +437,23 @@ endif()
 if (LIBCXX_INSTALL_LIBRARY)
   if (LIBCXX_INSTALL_SHARED_LIBRARY)
     install(TARGETS cxx_shared
-      ARCHIVE DESTINATION ${LIBCXX_INSTALL_PREFIX}${LIBCXX_INSTALL_LIBRARY_DIR} COMPONENT cxx
-      LIBRARY DESTINATION ${LIBCXX_INSTALL_PREFIX}${LIBCXX_INSTALL_LIBRARY_DIR} COMPONENT cxx
+      ARCHIVE DESTINATION ${LIBCXX_INSTALL_PREFIX}/${LIBCXX_INSTALL_LIBRARY_DIR} COMPONENT cxx
+      LIBRARY DESTINATION ${LIBCXX_INSTALL_PREFIX}/${LIBCXX_INSTALL_LIBRARY_DIR} COMPONENT cxx
       RUNTIME DESTINATION ${LIBCXX_INSTALL_PREFIX}bin COMPONENT cxx)
   endif()
 
   if (LIBCXX_INSTALL_STATIC_LIBRARY)
     install(TARGETS cxx_static
-      ARCHIVE DESTINATION ${LIBCXX_INSTALL_PREFIX}${LIBCXX_INSTALL_LIBRARY_DIR} COMPONENT cxx
-      LIBRARY DESTINATION ${LIBCXX_INSTALL_PREFIX}${LIBCXX_INSTALL_LIBRARY_DIR} COMPONENT cxx
-      RUNTIME DESTINATION ${LIBCXX_INSTALL_PREFIX}bin COMPONENT cxx)
+      ARCHIVE DESTINATION ${LIBCXX_INSTALL_PREFIX}/${LIBCXX_INSTALL_LIBRARY_DIR} COMPONENT cxx
+      LIBRARY DESTINATION ${LIBCXX_INSTALL_PREFIX}/${LIBCXX_INSTALL_LIBRARY_DIR} COMPONENT cxx
+      RUNTIME DESTINATION ${LIBCXX_INSTALL_PREFIX}/bin COMPONENT cxx)
   endif()
 
   if(LIBCXX_INSTALL_EXPERIMENTAL_LIBRARY)
     install(TARGETS ${LIBCXX_INSTALL_TARGETS} ${experimental_lib}
-      LIBRARY DESTINATION ${LIBCXX_INSTALL_PREFIX}${LIBCXX_INSTALL_LIBRARY_DIR} COMPONENT cxx
-      ARCHIVE DESTINATION ${LIBCXX_INSTALL_PREFIX}${LIBCXX_INSTALL_LIBRARY_DIR} COMPONENT cxx
-      RUNTIME DESTINATION ${LIBCXX_INSTALL_PREFIX}bin COMPONENT cxx)
+      LIBRARY DESTINATION ${LIBCXX_INSTALL_PREFIX}/${LIBCXX_INSTALL_LIBRARY_DIR} COMPONENT cxx
+      ARCHIVE DESTINATION ${LIBCXX_INSTALL_PREFIX}/${LIBCXX_INSTALL_LIBRARY_DIR} COMPONENT cxx
+      RUNTIME DESTINATION ${LIBCXX_INSTALL_PREFIX}/bin COMPONENT cxx)
   endif()
 
   # NOTE: This install command must go after the cxx install command otherwise
@@ -460,7 +462,7 @@ if (LIBCXX_INSTALL_LIBRARY)
     # Replace the libc++ filename with $<TARGET_LINKER_FILE:cxx>
     # after we required CMake 3.0.
     install(FILES "${LIBCXX_LIBRARY_DIR}/libc++${CMAKE_SHARED_LIBRARY_SUFFIX}"
-      DESTINATION ${LIBCXX_INSTALL_PREFIX}${LIBCXX_INSTALL_LIBRARY_DIR}
+      DESTINATION ${LIBCXX_INSTALL_PREFIX}/${LIBCXX_INSTALL_LIBRARY_DIR}
       COMPONENT libcxx)
   endif()
 endif()
diff --git b/projects/libcxxabi/src/orig.CMakeLists.txt a/projects/libcxxabi/src/CMakeLists.txt
index 45d4d02..948797b 100644
--- b/projects/libcxxabi/src/orig.CMakeLists.txt
+++ a/projects/libcxxabi/src/CMakeLists.txt
@@ -101,14 +101,15 @@ if ( APPLE )
     list(APPEND LIBCXXABI_COMPILE_FLAGS "-U__STRICT_ANSI__")
     list(APPEND LIBCXXABI_SHARED_LINK_FLAGS
       "-compatibility_version 1"
-      "-current_version 1"
-      "-install_name /usr/lib/libc++abi.1.dylib")
+      "-current_version ${VERSION_ACCORDING_TO_MACPORTS}"
+      "-install_name ${LIBCXX_INSTALL_PREFIX}/lib/libc++abi.1.dylib")
     list(APPEND LIBCXXABI_LINK_FLAGS
         "/usr/lib/libSystem.B.dylib")
   else()
     list(APPEND LIBCXXABI_SHARED_LINK_FLAGS
       "-compatibility_version 1"
-      "-install_name /usr/lib/libc++abi.1.dylib")
+      "-current_version ${VERSION_ACCORDING_TO_MACPORTS}"
+      "-install_name ${LIBCXX_INSTALL_PREFIX}/lib/libc++abi.1.dylib")
   endif()
 
   if (LLVM_USE_SANITIZER)
@@ -177,7 +178,7 @@ if (LIBCXXABI_ENABLE_SHARED)
                           SOVERSION
                             "1"
                           VERSION
-                            "1.0")
+                            "${VERSION_ACCORDING_TO_MACPORTS}")
 
   if(LIBCXXABI_ENABLE_PIC)
     set_target_properties(cxxabi_shared PROPERTIES POSITION_INDEPENDENT_CODE ON)
