diff --git a/CMakeLists.txt b/CMakeLists.txt
index 120c5582..0a49b207 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1761,7 +1761,7 @@ set(MAX_EXTENDED_ALIGNMENT "${ALIGNOF_DOUBLE16}")
 
 
 string(TOUPPER "${CMAKE_BUILD_TYPE}" BTYPE)
-if("${CMAKE_C_FLAGS_${BTYPE}}" MATCHES "DNDEBUG")
+if("${CMAKE_C_FLAGS_${BTYPE}}" MATCHES "DNDEBUG" OR "${CMAKE_C_FLAGS}" MATCHES "DNDEBUG")
   set(POCL_ASSERTS_BUILD 0)
 else()
   set(POCL_ASSERTS_BUILD 1)
