diff --git a/modules/ECMQmLoader.cpp.in b/modules/ECMQmLoader.cpp.in
index 8db55833..d50b4061 100644
--- a/modules/ECMQmLoader.cpp.in
+++ b/modules/ECMQmLoader.cpp.in
@@ -26,7 +26,11 @@ namespace {
         // language was set by kxmlgui because Qt ignores LANGUAGE on Windows and Apple OSs.
         // The following code is a simplified variant of QSystemLocale::fallbackUiLocale()
         // (in qlocale_unix.cpp) ignoring LC_ALL, LC_MESSAGES, and LANG.
+#if QT_VERSION >= QT_VERSION_CHECK(5,10,0)
         QString language = qEnvironmentVariable("LANGUAGE");
+#else
+        QString language = qgetenv("LANGUAGE");
+#endif
         if (!language.isEmpty()) {
             language = language.split(QLatin1Char{':'}).constFirst();
             if (!language.isEmpty()) {
