diff --git a/kde-modules/KDEInstallDirs5.cmake b/kde-modules/KDEInstallDirs5.cmake
index a7e12b797a6a54dc2afd9695883e830fdb87b943..c5aeeac6080e3b4707c33952db4400a10cd4bdbf 100644
--- a/kde-modules/KDEInstallDirs5.cmake
+++ b/kde-modules/KDEInstallDirs5.cmake
@@ -31,7 +31,7 @@ where ``<dir>`` is one of (default values in parentheses and alternative,
 deprecated variable name in square brackets):
 
 ``BUNDLEDIR``
-    application bundles (``/Applications/KDE``) [``BUNDLE_INSTALL_DIR``]
+    application bundles (``/Applications/MacPorts/KF5``) [``BUNDLE_INSTALL_DIR``]
 ``EXECROOTDIR``
     executables and libraries (``<empty>``) [``EXEC_INSTALL_PREFIX``]
 ``BINDIR``
diff --git a/kde-modules/KDEInstallDirs6.cmake b/kde-modules/KDEInstallDirs6.cmake
index fecd1b10b5202f1a2b2f81d46cb10243cfdfad6e..e0ee8e6cc34f4bec1ea312c79f42ea8af4fbe78c 100644
--- a/kde-modules/KDEInstallDirs6.cmake
+++ b/kde-modules/KDEInstallDirs6.cmake
@@ -33,7 +33,7 @@ Inclusion of this module defines the following variables:
 where ``<dir>`` is one of (default values in parentheses):
 
 ``BUNDLEDIR``
-    application bundles (``/Applications/KDE``)
+    application bundles (``/Applications/MacPorts/KF6``)
 ``EXECROOTDIR``
     executables and libraries (``<empty>``)
 ``BINDIR``
