diff --git a/kdbg/exprwnd.cpp b/kdbg/exprwnd.cpp
index ca68031..c3e5a3c 100644
--- a/kdbg/exprwnd.cpp
+++ b/kdbg/exprwnd.cpp
@@ -728,7 +728,11 @@ void ExprWnd::editValue(VarTree* item, const QString& text)
      * less than half of it if the text is shorter).
      */
     QFontMetrics metr(m_edit->font());
+#if QT_VERSION >= QT_VERSION_CHECK(5, 11, 0)
     int wMin = metr.horizontalAdvance("88888");
+#else
+    int wMin = metr.width("88888");
+#endif
     if (w < wMin)
 	w = wMin;
     int wThis = viewport()->width();
diff --git a/kdbg/memwindow.cpp b/kdbg/memwindow.cpp
index 4f5c1fc..7c784e2 100644
--- a/kdbg/memwindow.cpp
+++ b/kdbg/memwindow.cpp
@@ -282,6 +282,10 @@ void MemoryWindow::slotNewMemoryDump(const QString& msg, const std::list<MemoryD
 {
     QFontMetrics fm(font());
 
+#if QT_VERSION < QT_VERSION_CHECK(5, 11, 0)
+#define horizontalAdvance(t) width((t))
+#endif
+
     if (!msg.isEmpty()) {
 	for (int i = MAX_COL; i > 0; i--) {
 	    m_memory.setColumnHidden(i, true);
diff --git a/kdbg/sourcewnd.cpp b/kdbg/sourcewnd.cpp
index 201b189..848d321 100644
--- a/kdbg/sourcewnd.cpp
+++ b/kdbg/sourcewnd.cpp
@@ -781,8 +781,16 @@ void SourceWindow::setTabWidth(int numChars)
 	numChars = 8;
     QFontMetrics fm(document()->defaultFont());
     QString s;
+#if QT_VERSION >= QT_VERSION_CHECK(5, 11, 0)
     int w = fm.horizontalAdvance(s.fill('x', numChars));
+#else
+    int w = fm.width(s.fill('x', numChars));
+#endif
+#if QT_VERSION >= QT_VERSION_CHECK(5, 10, 0)
     setTabStopDistance(w);
+#else
+    setTabStopWidth(w);
+#endif
 }
 
 void SourceWindow::registerExpandedLine(int line)
