diff --git a/karbon/CMakeLists.txt b/karbon/CMakeLists.txt
index a910655074074ce0e385d61f4dfafe4e1a7c9751..542c587eb8c713d2c66652ed7acc719f8c66320a 100644
--- a/karbon/CMakeLists.txt
+++ b/karbon/CMakeLists.txt
@@ -51,7 +51,7 @@ file(GLOB karbon_icons "${CMAKE_CURRENT_SOURCE_DIR}/pics/action/*-actions-*.png"
 ecm_install_icons(ICONS ${karbon_icons} DESTINATION ${DATA_INSTALL_DIR}/karbon/icons)
 
 file(GLOB karbon_app_icons "${CMAKE_CURRENT_SOURCE_DIR}/pics/app/*-apps-calligrakarbon.png")
-ecm_add_app_icon(kdeinit_app_ICONS_SRCS ICONS ${karbon_app_icons})
+ecm_add_app_icon(kdeinit_app_ICONS_SRCS ICONS ${CMAKE_CURRENT_SOURCE_DIR}/pics/app/sc-apps-calligrakarbon.svgz ${karbon_app_icons})
 if(WIN32)
     set(_resourcefile "${CMAKE_CURRENT_BINARY_DIR}/kdeinit_app_ICONS_SRCS.rc")
 endif()
@@ -67,7 +67,6 @@ if (APPLE)
    set_target_properties(karbon PROPERTIES MACOSX_BUNDLE_INFO_PLIST ${CMAKE_CURRENT_SOURCE_DIR}/Info.plist.template)
    set_target_properties(karbon PROPERTIES MACOSX_BUNDLE_GUI_IDENTIFIER "org.calligra.karbon")
    set_target_properties(karbon PROPERTIES MACOSX_BUNDLE_BUNDLE_NAME "Karbon")
-   install( FILES ${CMAKE_CURRENT_BINARY_DIR}/karbon_KDEINIT_SRCS.icns DESTINATION ${BUNDLE_INSTALL_DIR}/karbon.app/Contents/Resources)
 endif()
 
 target_link_libraries(kdeinit_karbon karbonui)
diff --git a/words/app/CMakeLists.txt b/words/app/CMakeLists.txt
index 1ac94efe5c0d7bf2c55fd244b6de6dd7ee9657fb..1e4edcff159ecd4f2111b36c537a4f4336121a18 100644
--- a/words/app/CMakeLists.txt
+++ b/words/app/CMakeLists.txt
@@ -8,7 +8,7 @@ include_directories(
 set(words_KDEINIT_SRCS main.cpp )
 
 file(GLOB ICONS_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/../pics/*-apps-calligrawords.png")
-ecm_add_app_icon(kdeinit_app_ICONS_SRCS ICONS ${ICONS_SRCS})
+ecm_add_app_icon(kdeinit_app_ICONS_SRCS ICONS ${CMAKE_CURRENT_SOURCE_DIR}/../pics/sc-apps-calligrawords.svgz ${ICONS_SRCS})
 if(WIN32)
     set(_resourcefile "${CMAKE_CURRENT_BINARY_DIR}/kdeinit_app_ICONS_SRCS.rc")
 endif()
@@ -19,7 +19,6 @@ if (APPLE)
    set_target_properties(calligrawords PROPERTIES MACOSX_BUNDLE_INFO_PLIST ${CMAKE_CURRENT_SOURCE_DIR}/Info.plist.template)
    set_target_properties(calligrawords PROPERTIES MACOSX_BUNDLE_GUI_IDENTIFIER "org.kde.calligrawords")
    set_target_properties(calligrawords PROPERTIES MACOSX_BUNDLE_BUNDLE_NAME "Words")
-   install( FILES ${CMAKE_CURRENT_BINARY_DIR}/words_KDEINIT_SRCS.icns DESTINATION ${BUNDLE_INSTALL_DIR}/calligrawords.app/Contents/Resources)
 endif ()
 
 target_link_libraries(kdeinit_calligrawords   komain )
diff --git a/braindump/src/CMakeLists.txt b/braindump/src/CMakeLists.txt
index 95e077288f880a67b6eceec74c3ae2e7890031dc..3bae7d5c7a1a06b8f9a2d1c4d3b7c2e2a156fa5c 100644
--- a/braindump/src/CMakeLists.txt
+++ b/braindump/src/CMakeLists.txt
@@ -43,7 +43,7 @@ ki18n_wrap_ui(braindump_SRCS
 )
 
 file(GLOB ICONS_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/../data/icons/*-apps-braindump.png")
-ecm_add_app_icon( braindump_SRCS ICONS ${ICONS_SRCS})
+ecm_add_app_icon( braindump_SRCS ICONS ${CMAKE_CURRENT_SOURCE_DIR}/../data/icons/sc-apps-braindump.svgz ${ICONS_SRCS})
 
 add_executable(braindump ${braindump_SRCS})
 
diff --git a/flow/part/CMakeLists.txt b/flow/part/CMakeLists.txt
index f5d0893e84eb6e60fbe21ad2ba86e86c90d2e9f6..8fe8f62ff553d9d1091807fba529ea8ceaa7be8a 100644
--- a/flow/part/CMakeLists.txt
+++ b/flow/part/CMakeLists.txt
@@ -36,7 +36,7 @@ install(TARGETS flowpart DESTINATION ${PLUGIN_INSTALL_DIR}/calligra/parts)
 set(flow_KDEINIT_SRCS main.cpp )
 
 file(GLOB ICONS_SRCS "../pics/hi*-app-calligraflow.png")
-ecm_add_app_icon(flow_KDEINIT_SRCS ICONS ${ICONS_SRCS})
+ecm_add_app_icon(flow_KDEINIT_SRCS ICONS ../pics/hisc-app-calligraflow.svgz ${ICONS_SRCS})
 
 kf5_add_kdeinit_executable(calligraflow ${flow_KDEINIT_SRCS})
 
@@ -55,6 +55,3 @@ install( FILES org.kde.calligraflow.appdata.xml DESTINATION ${KDE_INSTALL_METAIN
 ### GUI files ###
 install( FILES flow.rc DESTINATION ${KXMLGUI_INSTALL_DIR}/flow)
 install( FILES flowrc DESTINATION ${CONFIG_INSTALL_DIR} )
-if(APPLE)
-   install( FILES ${CMAKE_CURRENT_BINARY_DIR}/flow_KDEINIT_SRCS.icns DESTINATION ${BUNDLE_INSTALL_DIR}/calligraflow.app/Contents/Resources)
-endif()
diff --git a/gemini/CMakeLists.txt b/gemini/CMakeLists.txt
index a9b2ccf07f23eb7d78195c98aedd2f21c8abd4f0..f552e10de0098f4801f3a7cdc852c97546fa02b9 100644
--- a/gemini/CMakeLists.txt
+++ b/gemini/CMakeLists.txt
@@ -56,7 +56,7 @@ set(thumbnailhelper_SRCS
 )
 
 file(GLOB ICONS_SRCS "pics/*-apps-calligragemini.png")
-ecm_add_app_icon(calligragemini_SRCS ICONS ${ICONS_SRCS})
+ecm_add_app_icon(calligragemini_SRCS ICONS pics/sc-apps-calligragemini.svgz ${ICONS_SRCS})
 
 # Thumbnail helper application
 # =============================================================================
@@ -93,9 +93,6 @@ install(FILES     calligragemini.qml                   DESTINATION ${DATA_INSTAL
 #install(FILES     calligragemini.rc                    DESTINATION ${DATA_INSTALL_DIR}/calligragemini)
 install(DIRECTORY qml                                       DESTINATION ${DATA_INSTALL_DIR}/calligragemini)
 install(DIRECTORY themes                                    DESTINATION ${DATA_INSTALL_DIR}/calligragemini)
-if(APPLE)
-   install( FILES ${CMAKE_CURRENT_BINARY_DIR}/calligragemini_SRCS.icns DESTINATION ${BUNDLE_INSTALL_DIR}/calligragemini.app/Contents/Resources)
-endif()
 
 add_subdirectory(pics)
 add_subdirectory(cloud)
diff --git a/sheets/CMakeLists.txt b/sheets/CMakeLists.txt
index d5b6241c0a900671f017a4695b1e4ec544c854d2..3fa0c9b613727013c5e5eee45990888bd581385e 100644
--- a/sheets/CMakeLists.txt
+++ b/sheets/CMakeLists.txt
@@ -462,7 +462,7 @@ if (SHOULD_BUILD_APP_SHEETS)
 set (calligrasheets_KDEINIT_SRCS part/Main.cpp )
 
 file(GLOB ICONS_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/data/pics/*-apps-calligrasheets.png")
-ecm_add_app_icon(kdeinit_app_ICONS_SRCS ICONS ${ICONS_SRCS})
+ecm_add_app_icon(kdeinit_app_ICONS_SRCS ICONS ${CMAKE_CURRENT_SOURCE_DIR}/data/pics/sc-apps-calligrasheets.svgz ${ICONS_SRCS})
 if(WIN32)
     set(_resourcefile "${CMAKE_CURRENT_BINARY_DIR}/kdeinit_app_ICONS_SRCS.rc")
 endif()
@@ -473,7 +473,6 @@ if (APPLE)
 set_target_properties(calligrasheets PROPERTIES MACOSX_BUNDLE_INFO_PLIST ${CMAKE_CURRENT_SOURCE_DIR}/Info.plist.template)
 set_target_properties(calligrasheets PROPERTIES MACOSX_BUNDLE_GUI_IDENTIFIER "org.calligra.sheets")
 set_target_properties(calligrasheets PROPERTIES MACOSX_BUNDLE_BUNDLE_NAME "Calligra Sheets 2")
-install( FILES ${CMAKE_CURRENT_BINARY_DIR}/calligrasheets_KDEINIT_SRCS.icns DESTINATION ${BUNDLE_INSTALL_DIR}/calligrasheets.app/Contents/Resources)
 endif ()
 
 target_link_libraries(kdeinit_calligrasheets komain KF5::KDELibs4Support)
diff --git a/stage/app/CMakeLists.txt b/stage/app/CMakeLists.txt
index b5b4464d3e997de609d8b2c42427a4b071a22386..bb74ef28b329d13e7b4bf963f9d33e713a0162ab 100644
--- a/stage/app/CMakeLists.txt
+++ b/stage/app/CMakeLists.txt
@@ -10,7 +10,7 @@ include_directories(
 set(stage_KDEINIT_SRCS main.cpp )
 
 file(GLOB ICONS_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/../pics/*-apps-calligrastage.png")
-ecm_add_app_icon(stage_KDEINIT_SRCS ICONS ${ICONS_SRCS})
+ecm_add_app_icon(stage_KDEINIT_SRCS ICONS ${CMAKE_CURRENT_SOURCE_DIR}/../pics/sc-apps-calligrastage.svgz ${ICONS_SRCS})
 
 kf5_add_kdeinit_executable(calligrastage ${stage_KDEINIT_SRCS})
 
@@ -18,7 +18,6 @@ if (APPLE)
    set_target_properties(calligrastage PROPERTIES MACOSX_BUNDLE_INFO_PLIST ${CMAKE_CURRENT_SOURCE_DIR}/Info.plist.template)
    set_target_properties(calligrastage PROPERTIES MACOSX_BUNDLE_GUI_IDENTIFIER "org.kde.calligrastage")
    set_target_properties(calligrastage PROPERTIES MACOSX_BUNDLE_BUNDLE_NAME "Calligra Stage 2")
-   install( FILES ${CMAKE_CURRENT_BINARY_DIR}/stage_KDEINIT_SRCS.icns DESTINATION ${BUNDLE_INSTALL_DIR}/calligrastage.app/Contents/Resources)
 endif ()
 
 target_link_libraries(kdeinit_calligrastage komain)
