diff --git a/CMakeLists.txt b/CMakeLists.txt
index b237f68fc..1be0df4b1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -28,10 +28,10 @@ endif()
 
 # define common versions of Calligra applications, used to generate calligraversion.h
 # update these version for every release:
-set(CALLIGRA_VERSION_STRING "3.2.89")
+set(CALLIGRA_VERSION_STRING "3.2.@MPVERSION3d@")
 set(CALLIGRA_STABLE_VERSION_MAJOR 3) # 3 for 3.x, 4 for 4.x, etc.
 set(CALLIGRA_STABLE_VERSION_MINOR 2) # 0 for 3.0, 1 for 3.1, etc.
-set(CALLIGRA_VERSION_RELEASE 89)     # 89 for Alpha, increase for next test releases, set 0 for first Stable, etc.
+set(CALLIGRA_VERSION_RELEASE 89)     # 89 for Alpha, increase for next test releases, set 0 for first Stable, etc.
 set(CALLIGRA_ALPHA 1) # uncomment only for Alpha
 # set(CALLIGRA_BETA 1) # uncomment only for Beta
 #set(CALLIGRA_RC 1) # uncomment only for RC
@@ -226,7 +226,7 @@ endif()
 
 set(REQUIRED_QT_VERSION "5.6.0")
 
-find_package(Qt5 ${REQUIRED_QT_VERSION} REQUIRED
+find_package(Qt5 ${REQUIRED_QT_VERSION} 
     COMPONENTS
     Core
     Gui
@@ -244,6 +244,8 @@ find_package(Qt5 ${REQUIRED_QT_VERSION} QUIET
     Quick
     QuickWidgets
     Sql
+    WebKit
+    WebKitWidgets
 )
 
 set_package_properties(Qt5DBus PROPERTIES
@@ -387,27 +389,27 @@ set_package_properties(KF5CalendarCore PROPERTIES
     TYPE OPTIONAL
     )
 
-##
-## Test for KF5Contacts
-##
-find_package(KF5Contacts CONFIG QUIET)
-set_package_properties(KF5Contacts PROPERTIES
-    DESCRIPTION "KDE Address book Library"
-    URL "https://www.kde.org/"
-    PURPOSE "Optionally used by semantic item Contact"
-    TYPE OPTIONAL
-    )
-
-##
-## Test for KF5AkonadiCore
-##
-find_package(KF5Akonadi CONFIG QUIET)
-set_package_properties(KF5Akonadi PROPERTIES
-    DESCRIPTION "Library for general Access to Akonadi"
-    URL "https://www.kde.org/"
-    PURPOSE "Optionally used by semantic items Event and Contact"
-    TYPE OPTIONAL
-    )
+# ##
+# ## Test for KF5Contacts
+# ##
+# find_package(KF5Contacts CONFIG QUIET)
+# set_package_properties(KF5Contacts PROPERTIES
+#     DESCRIPTION "KDE Address book Library"
+#     URL "https://www.kde.org/"
+#     PURPOSE "Optionally used by semantic item Contact"
+#     TYPE OPTIONAL
+#     )
+# 
+# ##
+# ## Test for KF5AkonadiCore
+# ##
+# find_package(KF5Akonadi CONFIG QUIET)
+# set_package_properties(KF5Akonadi PROPERTIES
+#     DESCRIPTION "Library for general Access to Akonadi"
+#     URL "https://www.kde.org/"
+#     PURPOSE "Optionally used by semantic items Event and Contact"
+#     TYPE OPTIONAL
+#     )
 
 ##
 ## Test for KChart
@@ -575,7 +577,7 @@ endif()
 ## Test for Fontconfig
 ##
 ## Only test if on non-Windows system
-if(NOT WIN32 AND NOT APPLE)
+if(NOT WIN32)
     macro_optional_find_package(Fontconfig)
     set_package_properties(Fontconfig PROPERTIES
 	DESCRIPTION "Library for configuring and customizing font access"
@@ -589,7 +591,7 @@ endif()
 ## Test for Freetype
 ##
 ## Only test if on non-Windows system
-if(NOT WIN32 AND NOT APPLE)
+if(NOT WIN32)
     macro_optional_find_package(Freetype)
     set_package_properties(Freetype PROPERTIES
 	DESCRIPTION "A Free, High-Quality, and Portable Font Engine"
@@ -982,10 +984,10 @@ calligra_drop_product_on_bad_condition( FILTER_KSPREAD_TO_LATEX
     NOT_COCOA "not supported with Qt Cocoa"
     )
 
-calligra_drop_product_on_bad_condition( APP_BRAINDUMP
-    NOT_WIN "unmaintained on Windows"
-    Qt5WebKitWidgets_FOUND "QWebPage needed for webpage plugin"
-    )
+# calligra_drop_product_on_bad_condition( APP_BRAINDUMP
+#     NOT_WIN "unmaintained on Windows"
+#     Qt5WebKitWidgets_FOUND "QWebPage needed for webpage plugin"
+#     )
 
 calligra_drop_product_on_bad_condition( PLUGIN_CALLIGRAGEMINI_GIT
     LIBGIT2_FOUND  "libgit2 devel not found"
@@ -1030,7 +1032,7 @@ calligra_drop_products_on_old_flag(words  PART_WORDS APP_WORDS)
 # e.g.:
 # calligra_disable_product(APP_FOO "isn't buildable at the moment")
 
-calligra_disable_product(APP_BRAINDUMP "Disabled, will (probably) be removed from Calligra")
+# calligra_disable_product(APP_BRAINDUMP "Disabled, will (probably) be removed from Calligra")
 
 #############################################
 ####     Calculate buildable products    ####
