diff --git a/CMakeLists.txt b/CMakeLists.txt
index 64acbd9857dd31f0d3cf36a4fb7015b1a20b5d53..3cc95194ec740bb432d7626059677cdad88b0076 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -254,7 +254,7 @@ endif()
 
 set(REQUIRED_QT_VERSION "5.3.0")
 
-find_package(Qt5 ${REQUIRED_QT_VERSION} REQUIRED
+find_package(Qt5 ${REQUIRED_QT_VERSION} 
     COMPONENTS
         Core
         Gui
@@ -465,24 +465,24 @@ set_package_properties(KF5Contacts PROPERTIES
 ##
 ## Test for KF5AkonadiContact
 ##
-find_package(KF5AkonadiContact CONFIG QUIET)
-set_package_properties(KF5AkonadiContact PROPERTIES
-    DESCRIPTION "Library for Accessing Contacts stored in Akonadi"
-    URL "https://www.kde.org/"
-    PURPOSE "Optionally used by Plan"
-    TYPE OPTIONAL
-)
+# find_package(KF5AkonadiContact CONFIG QUIET)
+# set_package_properties(KF5AkonadiContact PROPERTIES
+#     DESCRIPTION "Library for Accessing Contacts stored in Akonadi"
+#     URL "https://www.kde.org/"
+#     PURPOSE "Optionally used by Plan"
+#     TYPE OPTIONAL
+# )
 
 ##
 ## Test for KF5AkonadiCore
 ##
-find_package(KF5Akonadi CONFIG QUIET)
-set_package_properties(KF5Akonadi PROPERTIES
-    DESCRIPTION "Library for general Access to Akonadi"
-    URL "https://www.kde.org/"
-    PURPOSE "Optionally used by semantic items Event and Contact"
-    TYPE OPTIONAL
-)
+# find_package(KF5Akonadi CONFIG QUIET)
+# set_package_properties(KF5Akonadi PROPERTIES
+#     DESCRIPTION "Library for general Access to Akonadi"
+#     URL "https://www.kde.org/"
+#     PURPOSE "Optionally used by semantic items Event and Contact"
+#     TYPE OPTIONAL
+# )
 
 ##
 ## Test for KGantt
@@ -654,7 +654,7 @@ endif()
 ## Test for Fontconfig
 ##
 ## Only test if on non-Windows system
-if(NOT WIN32 AND NOT APPLE)
+if(NOT WIN32)
     macro_optional_find_package(Fontconfig)
     set_package_properties(Fontconfig PROPERTIES
         DESCRIPTION "Library for configuring and customizing font access"
@@ -668,7 +668,7 @@ endif()
 ## Test for Freetype
 ##
 ## Only test if on non-Windows system
-if(NOT WIN32 AND NOT APPLE)
+if(NOT WIN32)
     macro_optional_find_package(Freetype)
     set_package_properties(Freetype PROPERTIES
         DESCRIPTION "A Free, High-Quality, and Portable Font Engine"
