diff --git a/CalligraProducts.cmake b/CalligraProducts.cmake
index ad8251434..3a0de22ce 100644
--- a/CalligraProducts.cmake
+++ b/CalligraProducts.cmake
@@ -75,7 +75,7 @@ calligra_define_product(LIB_MSO "libmso"  REQUIRES LIB_CALLIGRA)
 calligra_define_product(LIB_KOMSOOXML "libkomsooxml"  REQUIRES LIB_CALLIGRA LIB_KOODF2 LIB_KOMAIN)
 
 # features
-calligra_define_feature(FEATURE_SCRIPTING  UNMAINTAINED  "Scripting feature")
+calligra_define_feature(FEATURE_SCRIPTING   "Scripting feature")
 calligra_define_feature(FEATURE_RDF  UNMAINTAINED "RDF feature")
 
 # plugins
@@ -94,7 +94,7 @@ calligra_define_product(APP_WORDS "Words app (for Desktop)"  REQUIRES PART_WORDS
 calligra_define_product(APP_STAGE "Stage app (for Desktop)"   REQUIRES PART_STAGE)
 calligra_define_product(APP_SHEETS "Sheets app (for Desktop)" REQUIRES PART_SHEETS)
 calligra_define_product(APP_KARBON "Karbon app (for Desktop)"  REQUIRES LIB_CALLIGRA LIB_KOMAIN LIB_KOPAGEAPP)
-calligra_define_product(APP_BRAINDUMP "Braindump app (for Desktop)"  UNMAINTAINED  REQUIRES LIB_CALLIGRA LIB_KOMAIN)
+calligra_define_product(APP_BRAINDUMP "Braindump app (for Desktop)"  REQUIRES LIB_CALLIGRA LIB_KOMAIN)
 calligra_define_product(DOC "Calligra Documentations")
 
 # staging apps
