diff --git a/libs/main/KoPart.cpp b/libs/main/KoPart.cpp
index 27dc8e24aa393e6c18eb8db3dc03e94161c7b655..4c3400aaea1c554782a2f18160a9335f48a2e2ca 100644
--- a/libs/main/KoPart.cpp
+++ b/libs/main/KoPart.cpp
@@ -51,11 +51,16 @@
 #include "KoPartAdaptor.h"
 #endif
 
-class Q_DECL_HIDDEN KoPart::Private
+#ifdef Q_OS_MACOS
+#include <QFileOpenEvent>
+#endif
+
+class Q_DECL_HIDDEN KoPart::Private : public QObject
 {
 public:
     Private(const KoComponentData &componentData_, KoPart *_parent)
-        : parent(_parent)
+        : QObject(_parent)
+        , parent(_parent)
         , document(0)
         , canvasItem(0)
         , startUpWidget(0)
@@ -72,6 +77,23 @@ public:
         //delete canvasItem;
     }
 
+    bool eventFilter(QObject *obj, QEvent *event) override
+    {
+        if (event->type() == QEvent::FileOpen) {
+            // handle file open events on Mac:
+            // try to inform KoApplication::start() that we are not
+            // started without file(s) to open:
+            QFileOpenEvent *fileOpenEvent = static_cast<QFileOpenEvent*>(event);
+//             debugMain << "FileOpen event for" << fileOpenEvent->url();
+            warnMain << "Direct FileOpen event for" << fileOpenEvent->url();
+            parent->openExistingFile(fileOpenEvent->url());
+            // don't eat the event
+            return false;
+        }
+
+        return QObject::eventFilter(obj, event);
+    }
+
     KoPart *parent;
 
     QList<KoView*> views;
@@ -249,6 +271,9 @@ void KoPart::openExistingFile(const QUrl &url)
     QApplication::setOverrideCursor(Qt::BusyCursor);
     d->document->openUrl(url);
     d->document->setModified(false);
+    // set the window file path. On Mac this improves integration by
+    // showing the document icon in the titlebar.
+    currentMainwindow()->setWindowFilePath(url.path());
     QApplication::restoreOverrideCursor();
 }
 
@@ -327,6 +352,9 @@ void KoPart::showStartUpWidget(KoMainWindow *mainWindow, bool alwaysShow)
     }
 
     mainWindow->setPartToOpen(this);
+#ifdef Q_OS_MACOS
+    KoApplication::koApplication()->installEventFilter(d);
+#endif
 }
 
 void KoPart::deleteOpenPane(bool closing)
@@ -341,6 +369,9 @@ void KoPart::deleteOpenPane(bool closing)
                                                                   mainWindows().first())->show();
         }
     }
+#ifdef Q_OS_MACOS
+    KoApplication::koApplication()->removeEventFilter(d);
+#endif
 }
 
 QList<KoPart::CustomDocumentWidgetItem> KoPart::createCustomDocumentWidgets(QWidget * /*parent*/)

diff --git a/karbon/Info.plist.template b/karbon/Info.plist.template
index 39e70e7d85b4bfb0e3583cd9a1f88e63da8550ca..c574fa9a3364566ec3b1e676957b9ad1235bd77d 100644
--- a/karbon/Info.plist.template
+++ b/karbon/Info.plist.template
@@ -124,6 +124,24 @@
             <key>CFBundleTypeRole</key>
             <string>Editor</string>
         </dict>
+        <dict>
+            <key>CFBundleTypeExtensions</key>
+            <array>
+                <string>pdf</string>
+            </array>
+            <key>CFBundleTypeMIMETypes</key>
+            <array>
+                <string>application/pdf</string>
+            </array>
+            <key>CFBundleTypeName</key>
+            <string>PDF Document</string>
+            <key>CFBundleTypeOSTypes</key>
+            <array>
+                <string>PDF</string>
+            </array>
+            <key>CFBundleTypeRole</key>
+            <string>Editor</string>
+        </dict>
     </array>
 </dict>
 </plist>
diff --git a/karbon/data/karbonpart.desktop b/karbon/data/karbonpart.desktop
index e4ef5b6832e6fbdabd4591de9ceee07e34b103b1..e2d12891c244f04e5d59939b3b18a82ce05e6d57 100644
--- a/karbon/data/karbonpart.desktop
+++ b/karbon/data/karbonpart.desktop
@@ -36,7 +36,7 @@ Name[x-test]=xxCalligra Scalable Graphics Componentxx
 Name[zh_CN]=Calligra 可缩放图像组件
 Name[zh_TW]=Calligra 可縮放圖片元件
 X-KDE-Library=karbonpart
-MimeType=application/x-karbon;image/x-eps;application/postscript;application/vnd.oasis.opendocument.graphics;application/vnd.oasis.opendocument.graphics-template;image/x-xfig;
+MimeType=application/x-karbon;image/x-eps;application/postscript;application/pdf;application/vnd.oasis.opendocument.graphics;application/vnd.oasis.opendocument.graphics-template;image/x-xfig;
 Type=Service
 X-KDE-ServiceTypes=Calligra/Part
 X-KDE-NativeMimeType=application/vnd.oasis.opendocument.graphics
