diff --git a/stage/app/CMakeLists.txt b/stage/app/CMakeLists.txt
index b5b4464..588c630 100644
--- a/stage/app/CMakeLists.txt
+++ b/stage/app/CMakeLists.txt
@@ -10,7 +10,7 @@ include_directories(
 set(stage_KDEINIT_SRCS main.cpp )
 
 file(GLOB ICONS_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/../pics/*-apps-calligrastage.png")
-ecm_add_app_icon(stage_KDEINIT_SRCS ICONS ${ICONS_SRCS})
+ecm_add_app_icon(stage_KDEINIT_SRCS ICONS ${CMAKE_CURRENT_SOURCE_DIR}/../pics/sc-apps-calligrastage.svgz ${ICONS_SRCS})
 
 kf5_add_kdeinit_executable(calligrastage ${stage_KDEINIT_SRCS})
 
