diff --git a/karbon/Info.plist.template b/karbon/Info.plist.template
index 39e70e7d8..c574fa9a3 100644
--- a/karbon/Info.plist.template
+++ b/karbon/Info.plist.template
@@ -124,6 +124,24 @@
             <key>CFBundleTypeRole</key>
             <string>Editor</string>
         </dict>
+        <dict>
+            <key>CFBundleTypeExtensions</key>
+            <array>
+                <string>pdf</string>
+            </array>
+            <key>CFBundleTypeMIMETypes</key>
+            <array>
+                <string>application/pdf</string>
+            </array>
+            <key>CFBundleTypeName</key>
+            <string>PDF Document</string>
+            <key>CFBundleTypeOSTypes</key>
+            <array>
+                <string>PDF</string>
+            </array>
+            <key>CFBundleTypeRole</key>
+            <string>Editor</string>
+        </dict>
     </array>
 </dict>
 </plist>
diff --git a/karbon/data/karbonpart.desktop b/karbon/data/karbonpart.desktop
index fa138caa4..c442ddd5f 100644
--- a/karbon/data/karbonpart.desktop
+++ b/karbon/data/karbonpart.desktop
@@ -36,7 +36,7 @@ Name[x-test]=xxCalligra Scalable Graphics Componentxx
 Name[zh_CN]=Calligra 矢量图形处理组件
 Name[zh_TW]=Calligra 可縮放圖片元件
 X-KDE-Library=karbonpart
-MimeType=application/x-karbon;image/x-eps;application/postscript;application/vnd.oasis.opendocument.graphics;application/vnd.oasis.opendocument.graphics-template;image/x-xfig;
+MimeType=application/x-karbon;image/x-eps;application/postscript;application/pdf;application/vnd.oasis.opendocument.graphics;application/vnd.oasis.opendocument.graphics-template;image/x-xfig;
 Type=Service
 X-KDE-ServiceTypes=Calligra/Part
 X-KDE-NativeMimeType=application/vnd.oasis.opendocument.graphics
diff --git a/libs/main/KoApplication.cpp b/libs/main/KoApplication.cpp
index f362ede7e..707b7d3a3 100644
--- a/libs/main/KoApplication.cpp
+++ b/libs/main/KoApplication.cpp
@@ -28,6 +28,8 @@
 #include <KoConfig.h>
 #include <KoResourcePaths.h>
 #include <KoComponentData.h>
+#include "KoDocument.h"
+#include "KoView.h"
 
 #include <klocalizedstring.h>
 #include <kdesktopfile.h>
@@ -60,6 +62,10 @@
 #include <tchar.h>
 #endif
 
+#ifdef Q_OS_MACOS
+#include <QFileOpenEvent>
+#endif
+
 
 #include "MainDebug.h"
 
@@ -691,3 +697,37 @@ KoApplication *KoApplication::koApplication()
 {
     return KoApp;
 }
+
+#ifdef Q_OS_MACOS
+bool KoApplication::event(QEvent *event)
+{
+    if (event->type() == QEvent::FileOpen) {
+        // handle file open events on Mac:
+        // try to inform KoApplication::start() that we are not
+        // started without file(s) to open:
+        const auto fileOpenEvent = static_cast<QFileOpenEvent*>(event);
+        const auto url = fileOpenEvent->url();
+        debugMain << "FileOpen event for" << fileOpenEvent->url();
+        // check if we already have the document open; if so,
+        // activate it as per Apple's HIG.
+        for (auto kw : KoMainWindow::memberList()) {
+            if (auto mw = qobject_cast<KoMainWindow*>(kw)) {
+                auto doc = mw->rootDocument();
+                if (doc && doc->url() == url) {
+                    debugMain << "\tActivating doc" << doc << "in" << mw;
+                    mw->show();
+                    mw->activateWindow();
+                    mw->raise();
+                    return true;
+                }
+            }
+        }
+        emit openFileEvent(url);
+        return true;
+    }
+
+    return QApplication::event(event);
+}
+#endif
+
+
diff --git a/libs/main/KoApplication.h b/libs/main/KoApplication.h
index e67e14355..19cfc712f 100644
--- a/libs/main/KoApplication.h
+++ b/libs/main/KoApplication.h
@@ -106,6 +106,10 @@ public:
      */
     static KoApplication* koApplication();
 
+#ifdef Q_OS_MACOS
+    virtual bool event(QEvent *e) override;
+#endif
+
 Q_SIGNALS:
 
     /// KoPart needs to be able to emit document signals from here. These
@@ -123,6 +127,11 @@ Q_SIGNALS:
      */
     void documentClosed(const QString &ref);
 
+    /**
+     * emitted on Mac if a QFileOpenEvent is received
+     */
+    void openFileEvent(const QUrl &url);
+
 private Q_SLOTS:
     void benchmarkLoadingFinished();
     void slotFilePrint(KoMainWindow *view);
diff --git a/libs/main/KoMainWindow.cpp b/libs/main/KoMainWindow.cpp
index c1c3b34d2..e3f08b875 100644
--- a/libs/main/KoMainWindow.cpp
+++ b/libs/main/KoMainWindow.cpp
@@ -407,6 +407,10 @@ void KoMainWindow::setNoCleanup(bool noCleanup)
 
 KoMainWindow::~KoMainWindow()
 {
+#ifdef Q_OS_MACOS
+    enableFileOpenEventHandling(false);
+#endif
+
     KConfigGroup cfg( KSharedConfig::openConfig(), "MainWindow");
     cfg.writeEntry("ko_geometry", saveGeometry().toBase64());
     cfg.writeEntry("ko_windowstate", saveState().toBase64());
@@ -2141,3 +2145,18 @@ void KoMainWindow::showDockerTitleBars(bool show)
     KConfigGroup configGroupInterface =  KSharedConfig::openConfig()->group("Interface");
     configGroupInterface.writeEntry("ShowDockerTitleBars", show);
 }
+
+#ifdef Q_OS_MACOS
+void KoMainWindow::enableFileOpenEventHandling(bool enabled)
+{
+    if (enabled) {
+        // disconnect all other listeners
+        disconnect(KoApplication::koApplication(), &KoApplication::openFileEvent, nullptr, nullptr);
+        // connect us
+        connect(KoApplication::koApplication(), &KoApplication::openFileEvent,
+                this, [&](const QUrl &url) { openDocument(url); }, Qt::UniqueConnection);
+    } else {
+        disconnect(KoApplication::koApplication(), nullptr, this, nullptr);
+    }
+}
+#endif
diff --git a/libs/main/KoMainWindow.h b/libs/main/KoMainWindow.h
index 594dca14d..edb97ca20 100644
--- a/libs/main/KoMainWindow.h
+++ b/libs/main/KoMainWindow.h
@@ -425,6 +425,10 @@ private:
      */
     virtual void setActivePart(KoPart *part, QWidget *widget);
 
+#ifdef Q_OS_MACOS
+    void enableFileOpenEventHandling(bool enable);
+#endif
+
 private Q_SLOTS:
 
     /**
diff --git a/libs/main/KoPart.cpp b/libs/main/KoPart.cpp
index 10b87549f..d5e4cd803 100644
--- a/libs/main/KoPart.cpp
+++ b/libs/main/KoPart.cpp
@@ -38,6 +38,10 @@
 #include "KoPartAdaptor.h"
 #endif
 
+#ifdef Q_OS_MACOS
+#include <QFileOpenEvent>
+#endif
+
 class Q_DECL_HIDDEN KoPart::Private
 {
 public:
@@ -52,9 +56,28 @@ public:
 
     ~Private()
     {
+        handleOpenFileEvents(false);
+        listenerPartList.removeAll(parent);
         delete proxyWidget;
     }
 
+    void handleOpenFileEvents(bool enabled)
+    {
+        if (enabled) {
+            // there should always be only a single handler for QFileOpenEvents
+            // so disconnect all other listeners
+            disconnect(KoApplication::koApplication(), &KoApplication::openFileEvent, nullptr, nullptr);
+            connect(KoApplication::koApplication(), &KoApplication::openFileEvent,
+                    parent, &KoPart::openExistingFile, Qt::UniqueConnection);
+            if (!listenerPartList.contains(parent)) {
+                listenerPartList.append(parent);
+            }
+        } else {
+            disconnect(KoApplication::koApplication(), &KoApplication::openFileEvent, parent, &KoPart::openExistingFile);
+            listenerPartList.removeAll(parent);
+        }
+    }
+
     KoPart *parent;
 
     QList<KoView*> views;
@@ -66,8 +89,9 @@ public:
     QString templatesResourcePath;
 
     KoComponentData componentData;
+    static QList<KoPart*> listenerPartList;
 };
-
+QList<KoPart*> KoPart::Private::listenerPartList;
 
 KoPart::KoPart(const KoComponentData &componentData, QObject *parent)
         : QObject(parent)
@@ -232,6 +256,9 @@ void KoPart::openExistingFile(const QUrl &url)
     QApplication::setOverrideCursor(Qt::BusyCursor);
     d->document->openUrl(url);
     d->document->setModified(false);
+    // set the window file path. On Mac this improves integration by
+    // showing the document icon in the titlebar.
+    currentMainwindow()->setWindowFilePath(url.path());
     QApplication::restoreOverrideCursor();
 }
 
@@ -310,6 +337,9 @@ void KoPart::showStartUpWidget(KoMainWindow *mainWindow, bool alwaysShow)
     }
 
     mainWindow->setPartToOpen(this);
+#ifdef Q_OS_MACOS
+    d->handleOpenFileEvents(true);
+#endif
 }
 
 void KoPart::deleteOpenPane(bool closing)
@@ -324,6 +354,22 @@ void KoPart::deleteOpenPane(bool closing)
                                                                   mainWindows().first())->show();
         }
     }
+#ifdef Q_OS_MACOS
+    // there should always be only a single handler for QFileOpenEvents
+    // (it doesn't matter who).
+    d->handleOpenFileEvents(false);
+    if (!closing) {
+        // multiple KoOpenPane startUpWidgets can be open in as many KoParts
+        // so check if there are others that have not yet disabled their
+        // OpenFileEvent handler.
+        if (d->listenerPartList.isEmpty()) {
+            // no other KoOpenPanes left
+            mainWindows().first()->enableFileOpenEventHandling(true);
+        } else {
+            d->listenerPartList.first()->d->handleOpenFileEvents(true);
+        }
+    }
+#endif
 }
 
 QList<KoPart::CustomDocumentWidgetItem> KoPart::createCustomDocumentWidgets(QWidget * /*parent*/)
