diff --git a/libs/text/CMakeLists.txt b/libs/text/CMakeLists.txt
index 429afe79..81c13fba 100644
--- a/libs/text/CMakeLists.txt
+++ b/libs/text/CMakeLists.txt
@@ -152,7 +152,7 @@ if( SHOULD_BUILD_FEATURE_RDF )
 endif()
 
 if( FONTCONFIG_FOUND )
-    target_link_libraries(kotext PRIVATE Fontconfig::Fontconfig)
+    target_link_libraries(kotext PRIVATE ${PKG_FONTCONFIG_LIBRARIES})
 endif()
 
 if( FREETYPE_FOUND )
