diff --git a/CalligraProducts.cmake b/CalligraProducts.cmake
index 0e94ede94d78dc94ee83a7cc7b97612eca0c36ca..f3461272fbc921cd9236e56a14848d25480f3d43 100644
--- a/CalligraProducts.cmake
+++ b/CalligraProducts.cmake
@@ -90,12 +90,12 @@ calligra_define_product(PART_COMPONENTS "QtQuick2 Plugin that provides Calligra
 
 # apps
 calligra_define_product(APP_WORDS "Words app (for Desktop)"  REQUIRES PART_WORDS)
-calligra_define_product(APP_STAGE "Stage app (for Desktop)"  UNMAINTAINED   REQUIRES PART_STAGE)
+calligra_define_product(APP_STAGE "Stage app (for Desktop)"  REQUIRES PART_STAGE)
 calligra_define_product(APP_SHEETS "Sheets app (for Desktop)" REQUIRES PART_SHEETS)
 calligra_define_product(APP_KARBON "Karbon app (for Desktop)"  REQUIRES LIB_CALLIGRA LIB_KOMAIN)
 calligra_define_product(APP_FLOW "Flow app (for Desktop)" UNPORTED  REQUIRES LIB_CALLIGRA LIB_KOMAIN LIB_KOPAGEAPP)
 calligra_define_product(APP_PLAN "Plan app (for Desktop)")
-calligra_define_product(APP_BRAINDUMP "Braindump app (for Desktop)"  UNMAINTAINED  REQUIRES LIB_CALLIGRA LIB_KOMAIN)
+calligra_define_product(APP_BRAINDUMP "Braindump app (for Desktop)"  REQUIRES LIB_CALLIGRA LIB_KOMAIN)
 calligra_define_product(DOC "Calligra Documentations" STAGING)
 
 # staging apps
