--- calligra-3.0.0.1/karbon/orig.CMakeLists.txt	2016-12-20 10:48:40.000000000 +0100
+++ calligra-3.0.0.1/karbon/CMakeLists.txt	2017-01-09 22:31:21.000000000 +0100
@@ -51,7 +51,11 @@
 ecm_install_icons(ICONS ${karbon_icons} DESTINATION ${DATA_INSTALL_DIR}/karbon/icons)
 
 file(GLOB karbon_app_icons "${CMAKE_CURRENT_SOURCE_DIR}/pics/app/*-apps-calligrakarbon.png")
-ecm_add_app_icon(karbon_KDEINIT_SRCS ICONS ${karbon_app_icons})
+if(APPLE)
+    ecm_add_app_icon(karbon_KDEINIT_SRCS ICONS "${CMAKE_CURRENT_SOURCE_DIR}/pics/app/sc-apps-calligrakarbon.svgz")
+else()
+    ecm_add_app_icon(karbon_KDEINIT_SRCS ICONS ${karbon_app_icons})
+endif()
 ecm_install_icons(ICONS
     ${karbon_app_icons}
     "${CMAKE_CURRENT_SOURCE_DIR}/pics/app/sc-apps-calligrakarbon.svgz"
@@ -64,7 +68,7 @@
    set_target_properties(karbon PROPERTIES MACOSX_BUNDLE_INFO_PLIST ${CMAKE_CURRENT_SOURCE_DIR}/Info.plist.template)
    set_target_properties(karbon PROPERTIES MACOSX_BUNDLE_GUI_IDENTIFIER "org.calligra.karbon")
    set_target_properties(karbon PROPERTIES MACOSX_BUNDLE_BUNDLE_NAME "Karbon")
-   install( FILES ${CMAKE_CURRENT_BINARY_DIR}/karbon_KDEINIT_SRCS.icns DESTINATION ${BUNDLE_INSTALL_DIR}/karbon.app/Contents/Resources)
+#    install( FILES ${CMAKE_CURRENT_BINARY_DIR}/karbon_KDEINIT_SRCS.icns DESTINATION ${BUNDLE_INSTALL_DIR}/karbon.app/Contents/Resources)
 endif()
 
 target_link_libraries(kdeinit_karbon karbonui)
