diff --git a/filters/words/msword-odf/wv2/src/parser9x.cpp b/filters/words/msword-odf/wv2/src/parser9x.cpp
index d0efe9b..9a6d160 100644
--- a/filters/words/msword-odf/wv2/src/parser9x.cpp
+++ b/filters/words/msword-odf/wv2/src/parser9x.cpp
@@ -301,7 +301,7 @@ void Parser9x::parseTableRow( const TableRowData& data )
 #endif
 }
 
-void Parser9x::parseTextBox(uint index, bool stylesxml)
+void Parser9x::parseTextBox(unsigned int index, bool stylesxml)
 {
     const PLCF<Word97::FTXBXS>* plcftxbxTxt = 0;
     if (stylesxml) {
diff --git a/filters/words/msword-odf/wv2/src/parser9x.h b/filters/words/msword-odf/wv2/src/parser9x.h
index c29847b..a5cdfcb 100644
--- a/filters/words/msword-odf/wv2/src/parser9x.h
+++ b/filters/words/msword-odf/wv2/src/parser9x.h
@@ -130,7 +130,7 @@ namespace wvWare
         void parsePicture( const PictureData& data );
 
         //Can't create Functor for textbox in advance.  Index into plcfTxbxTxt unknown.
-        virtual void parseTextBox(uint index, bool stylesxml);
+        virtual void parseTextBox(unsigned int index, bool stylesxml);
 
     protected:
         // First all variables which don't change their state during
