--- calligra-3.0.0.1/extras/converter/orig.CMakeLists.txt	2017-01-09 22:17:30.000000000 +0100
+++ calligra-3.0.0.1/extras/converter/CMakeLists.txt	2017-01-09 22:18:04.000000000 +0100
@@ -3,6 +3,7 @@
 set(calligraconverter_SRCS calligraconverter.cpp)
 
 add_executable(calligraconverter ${calligraconverter_SRCS})
+ecm_mark_nongui_executable(calligraconverter)
 
 target_link_libraries(calligraconverter komain)
 
