diff --git src/plugins/platforms/cocoa/messages.cpp src/plugins/platforms/cocoa/messages.cpp
index 06e3dd454ee187a79df685d5c28876917047b8fc..0db84ab01af6ca460be429b6f9c24fc87459c9c3 100644
--- src/plugins/platforms/cocoa/messages.cpp
+++ src/plugins/platforms/cocoa/messages.cpp
@@ -77,21 +77,21 @@ QPlatformMenuItem::MenuRole detectMenuRole(const QString &caption)
 {
     QString captionNoAmpersand(caption);
     captionNoAmpersand.remove(QLatin1Char('&'));
-    const QString aboutString = QCoreApplication::translate("QCocoaMenuItem", "About");
-    if (captionNoAmpersand.startsWith(aboutString, Qt::CaseInsensitive)
-        || captionNoAmpersand.endsWith(aboutString, Qt::CaseInsensitive)) {
-        static const QRegularExpression qtRegExp(QLatin1String("qt$"), QRegularExpression::CaseInsensitiveOption);
-        if (captionNoAmpersand.contains(qtRegExp))
-            return QPlatformMenuItem::AboutQtRole;
-        return QPlatformMenuItem::AboutRole;
-    }
-    if (captionNoAmpersand.startsWith(QCoreApplication::translate("QCocoaMenuItem", "Config"), Qt::CaseInsensitive)
-        || captionNoAmpersand.startsWith(QCoreApplication::translate("QCocoaMenuItem", "Preference"), Qt::CaseInsensitive)
-        || captionNoAmpersand.startsWith(QCoreApplication::translate("QCocoaMenuItem", "Options"), Qt::CaseInsensitive)
-        || captionNoAmpersand.startsWith(QCoreApplication::translate("QCocoaMenuItem", "Setting"), Qt::CaseInsensitive)
-        || captionNoAmpersand.startsWith(QCoreApplication::translate("QCocoaMenuItem", "Setup"), Qt::CaseInsensitive)) {
-        return QPlatformMenuItem::PreferencesRole;
-    }
+//     const QString aboutString = QCoreApplication::translate("QCocoaMenuItem", "About");
+//     if (captionNoAmpersand.startsWith(aboutString, Qt::CaseInsensitive)
+//         || captionNoAmpersand.endsWith(aboutString, Qt::CaseInsensitive)) {
+//         static const QRegularExpression qtRegExp(QLatin1String("qt$"), QRegularExpression::CaseInsensitiveOption);
+//         if (captionNoAmpersand.contains(qtRegExp))
+//             return QPlatformMenuItem::AboutQtRole;
+//         return QPlatformMenuItem::AboutRole;
+//     }
+//     if (captionNoAmpersand.startsWith(QCoreApplication::translate("QCocoaMenuItem", "Config"), Qt::CaseInsensitive)
+//         || captionNoAmpersand.startsWith(QCoreApplication::translate("QCocoaMenuItem", "Preference"), Qt::CaseInsensitive)
+//         || captionNoAmpersand.startsWith(QCoreApplication::translate("QCocoaMenuItem", "Options"), Qt::CaseInsensitive)
+//         || captionNoAmpersand.startsWith(QCoreApplication::translate("QCocoaMenuItem", "Setting"), Qt::CaseInsensitive)
+//         || captionNoAmpersand.startsWith(QCoreApplication::translate("QCocoaMenuItem", "Setup"), Qt::CaseInsensitive)) {
+//         return QPlatformMenuItem::PreferencesRole;
+//     }
     if (captionNoAmpersand.startsWith(QCoreApplication::translate("QCocoaMenuItem", "Quit"), Qt::CaseInsensitive)
         || captionNoAmpersand.startsWith(QCoreApplication::translate("QCocoaMenuItem", "Exit"), Qt::CaseInsensitive)) {
         return QPlatformMenuItem::QuitRole;
