diff --git src/widgets/widgets/qfontcombobox.cpp src/widgets/widgets/qfontcombobox.cpp
index 957a464b7168c5e1381c0ecdfa0b1daa57138510..d019b779fcdef9cbf8cf360eb8022745953d525d 100644
--- src/widgets/widgets/qfontcombobox.cpp
+++ src/widgets/widgets/qfontcombobox.cpp
@@ -213,9 +213,15 @@ void QFontFamilyDelegate::paint(QPainter *painter,
 {
     QString text = index.data(Qt::DisplayRole).toString();
     QFont font(option.font);
-    font.setPointSize(QFontInfo(font).pointSize() * 3 / 2);
-    QFont font2 = font;
-    font2.setFamily(text);
+    const bool usePreview = !qEnvironmentVariableIsSet("QT_FONTCOMBOBOX_NO_PREVIEW");
+    QFont font2;
+    if (usePreview) {
+        font.setPointSize(QFontInfo(font).pointSize() * 3 / 2);
+        font2 = font;
+        font2.setFamily(text);
+    } else {
+        font2 = font;
+    }
 
     bool hasLatin;
     QFontDatabase::WritingSystem system = writingSystemForFont(font2, &hasLatin);
