diff --git src/plugins/platforms/cocoa/qcocoaintegration.mm src/plugins/platforms/cocoa/qcocoaintegration.mm
index cfaf56a03e08b8c2744bb2e4297f55a13b56233e..a5611f2280d42049ccc0a2ea6ace22df47cff125 100644
--- src/plugins/platforms/cocoa/qcocoaintegration.mm
+++ src/plugins/platforms/cocoa/qcocoaintegration.mm
@@ -511,7 +511,7 @@ QVariant QCocoaIntegration::styleHint(StyleHint hint) const
 {
     switch (hint) {
     case FontSmoothingGamma:
-        return QCoreTextFontEngine::fontSmoothingGamma();
+        return mOptions.testFlag(UseFreeTypeFontEngine)? 1.5 : QCoreTextFontEngine::fontSmoothingGamma();
     case ShowShortcutsInContextMenus:
         return QVariant(false);
     default: break;
diff --git src/plugins/platforms/xcb/qxcbintegration.cpp src/plugins/platforms/xcb/qxcbintegration.cpp
index ed9e87a03698947678650b8928726438c48c0d2f..5cb1dc6583258aaeb8f05a46c1aa5b892b1a8003 100644
--- src/plugins/platforms/xcb/qxcbintegration.cpp
+++ src/plugins/platforms/xcb/qxcbintegration.cpp
@@ -133,6 +133,8 @@ QXcbIntegration::QXcbIntegration(const QStringList &parameters, int &argc, char
     , m_instanceName(0)
     , m_canGrab(true)
     , m_defaultVisualId(UINT_MAX)
+//     , m_fontSmoothingGamma(QPlatformIntegration::styleHint(QPlatformIntegration::FontSmoothingGamma).toReal())
+    , m_fontSmoothingGamma(0.93)
 {
     m_instance = this;
     qApp->setAttribute(Qt::AA_CompressHighFrequencyEvents, true);
@@ -145,6 +147,14 @@ QXcbIntegration::QXcbIntegration(const QStringList &parameters, int &argc, char
 #endif
     m_nativeInterface.reset(new QXcbNativeInterface);
 
+    if (qEnvironmentVariableIsSet("QT_XCB_FONT_GAMMA")) {
+        bool ok = false;
+        qreal fontgamma = qgetenv("QT_XCB_FONT_GAMMA").toDouble(&ok);
+        if (ok) {
+            m_fontSmoothingGamma = fontgamma;
+        }
+    }
+
     // Parse arguments
     const char *displayName = 0;
     bool noGrabArg = false;
@@ -169,6 +179,13 @@ QXcbIntegration::QXcbIntegration(const QStringList &parameters, int &argc, char
                 if (!ok)
                     m_defaultVisualId = UINT_MAX;
             }
+            else if (arg == "-fontgamma" && i < argc - 1) {
+                bool ok = false;
+                qreal fontgamma = QByteArray(argv[++i]).toDouble(&ok);
+                if (ok) {
+                    m_fontSmoothingGamma = fontgamma;
+                }
+            }
             else
                 argv[j++] = argv[i];
         }
@@ -475,6 +492,11 @@ QVariant QXcbIntegration::styleHint(QPlatformIntegration::StyleHint hint) const
         return false;
     case QPlatformIntegration::ReplayMousePressOutsidePopup:
         return false;
+
+    case QPlatformIntegration::FontSmoothingGamma:
+        return m_fontSmoothingGamma;
+        break;
+
     default:
         break;
     }
diff --git src/plugins/platforms/xcb/qxcbintegration.h src/plugins/platforms/xcb/qxcbintegration.h
index 571726c354e1149b3e37d8a937d445e694b8e9d0..a630bcc447172108a613685a2b77f513951c8405 100644
--- src/plugins/platforms/xcb/qxcbintegration.h
+++ src/plugins/platforms/xcb/qxcbintegration.h
@@ -141,6 +141,7 @@ private:
     const char *m_instanceName;
     bool m_canGrab;
     xcb_visualid_t m_defaultVisualId;
+    qreal m_fontSmoothingGamma;
 
     static QXcbIntegration *m_instance;
 };
