diff --git src/platformsupport/services/genericunix/qgenericunixservices.cpp src/platformsupport/services/genericunix/qgenericunixservices.cpp
index 734bdcaf7572c848a6ffd0e1b5a45aabcf07ad69..219c58e0c06ea5af270d38ffe1292f25fecd3835 100644
--- src/platformsupport/services/genericunix/qgenericunixservices.cpp
+++ src/platformsupport/services/genericunix/qgenericunixservices.cpp
@@ -114,7 +114,12 @@ static inline QByteArray detectDesktopEnvironment()
     else if (desktopSession == "kde")
         return QByteArrayLiteral("KDE");
 
+#ifdef Q_OS_MACOS
+    // default to the KDE DE. This still requires a KDE platform theme plugin to have any effect!
+    return QByteArrayLiteral("KDE");
+#else
     return QByteArrayLiteral("UNKNOWN");
+#endif
 }
 
 static inline bool checkExecutable(const QString &candidate, QString *result)
@@ -130,6 +135,10 @@ static inline bool detectWebBrowser(const QByteArray &desktop,
     const char *browsers[] = {"google-chrome", "firefox", "mozilla", "opera"};
 
     browser->clear();
+#ifdef Q_OS_MACOS
+    if (checkExecutable(QStringLiteral("open"), browser))
+        return true;
+#endif
     if (checkExecutable(QStringLiteral("xdg-open"), browser))
         return true;
 
