diff --git src/qtdiag/main.cpp src/qtdiag/main.cpp
index de56d7201acbcf7e1c0604f3ffd999b6ad846543..f1de734a1f3bf6da045d0222188d740d98a45697 100644
--- src/qtdiag/main.cpp
+++ src/qtdiag/main.cpp
@@ -30,6 +30,7 @@
 
 #include <QtGui/QGuiApplication>
 #include <QtCore/QCommandLineParser>
+#include <QtCore/QStandardPaths>
 
 #include <iostream>
 #include <string>
@@ -50,11 +51,19 @@ int main(int argc, char **argv)
     const QCommandLineOption glExtensionOption(QStringLiteral("gl-extensions"), QStringLiteral("List GL extensions"));
     const QCommandLineOption fontOption(QStringLiteral("fonts"), QStringLiteral("Output list of fonts"));
     const QCommandLineOption noVkOption(QStringLiteral("no-vulkan"), QStringLiteral("Do not output Vulkan information"));
+    const QCommandLineOption qspTestModeOption(QStringLiteral("qsp-testmode"), QStringLiteral("output the QStandardPaths locations in testing mode"));
+#ifdef Q_OS_MACOS
+    const QCommandLineOption qspXDGModeOption(QStringLiteral("qsp-xdgmode"), QStringLiteral("output the QStandardPaths locations in XDG/Freedesktop compliant mode"));
+#endif
     commandLineParser.setApplicationDescription(QStringLiteral("Prints diagnostic output about the Qt library."));
     commandLineParser.addOption(noGlOption);
     commandLineParser.addOption(glExtensionOption);
     commandLineParser.addOption(fontOption);
     commandLineParser.addOption(noVkOption);
+    commandLineParser.addOption(qspTestModeOption);
+#ifdef Q_OS_MACOS
+    commandLineParser.addOption(qspXDGModeOption);
+#endif
     commandLineParser.addHelpOption();
     commandLineParser.process(app);
     unsigned flags = commandLineParser.isSet(noGlOption) ? 0u : unsigned(QtDiagGl);
@@ -64,6 +73,14 @@ int main(int argc, char **argv)
         flags |= QtDiagFonts;
     if (!commandLineParser.isSet(noVkOption))
         flags |= QtDiagVk;
+    if (commandLineParser.isSet(qspTestModeOption))
+        flags |= QtDiagQSPTestMode;
+#ifdef Q_OS_MACOS
+    if (commandLineParser.isSet(qspXDGModeOption)) {
+        flags |= QtDiagQSPXDGMode;
+        QStandardPaths::setALTLocationsEnabled(true);
+    }
+#endif
 
     std::wcout << qtDiag(flags).toStdWString();
     std::wcout.flush();
diff --git src/qtdiag/qtdiag.cpp src/qtdiag/qtdiag.cpp
index c7de04de9f7b537ca5a667148f42f1c2361ae545..eb292de7c59850bf6dd1fcb9a2679b55b05735fb 100644
--- src/qtdiag/qtdiag.cpp
+++ src/qtdiag/qtdiag.cpp
@@ -265,7 +265,7 @@ void dumpVkInfo(QTextStream &str)
 // Dump values of QStandardPaths, indicate writable locations by asterisk.
 static void dumpStandardLocation(QTextStream &str, QStandardPaths::StandardLocation location)
 {
-    str << '"' << QStandardPaths::displayName(location) << '"';
+    str << '"' << QStandardPaths::displayName(location) << "\" =";
     const QStringList directories = QStandardPaths::standardLocations(location);
     const QString writableDirectory = QStandardPaths::writableLocation(location);
     const int writableIndex = writableDirectory.isEmpty() ? -1 : directories.indexOf(writableDirectory);
@@ -449,7 +449,17 @@ QString qtDiag(unsigned flags)
     DUMP_LIBRARYPATH(str, TestsPath)
     DUMP_LIBRARYPATH(str, SettingsPath)
 
-    str << "\nStandard paths [*...* denote writable entry]:\n";
+    if (flags & QtDiagQSPTestMode) {
+        QStandardPaths::enableTestMode(true);
+        str << "\nStandard paths, testing mode [*...* denote writable entry]:\n";
+    } else {
+        str << "\nStandard paths [*...* denote writable entry]:\n";
+    }
+#ifdef Q_OS_MACOS
+    if ((flags & QtDiagQSPXDGMode) || QStandardPaths::isALTLocationsEnabled()) {
+        str << "\t- QStandardPaths is in XDG/Freedesktop compliant mode -\n";
+    }
+#endif
     DUMP_STANDARDPATH(str, DesktopLocation)
     DUMP_STANDARDPATH(str, DocumentsLocation)
     DUMP_STANDARDPATH(str, FontsLocation)
@@ -459,6 +469,7 @@ QString qtDiag(unsigned flags)
     DUMP_STANDARDPATH(str, PicturesLocation)
     DUMP_STANDARDPATH(str, TempLocation)
     DUMP_STANDARDPATH(str, HomeLocation)
+    DUMP_STANDARDPATH(str, DataLocation)
     DUMP_STANDARDPATH(str, AppLocalDataLocation)
     DUMP_STANDARDPATH(str, CacheLocation)
     DUMP_STANDARDPATH(str, GenericDataLocation)
@@ -666,8 +677,6 @@ QString qtDiag(unsigned flags)
         dumpGlInfo(str, flags & QtDiagGlExtensions);
         str << "\n\n";
     }
-#else
-    Q_UNUSED(flags)
 #endif // !QT_NO_OPENGL
 
 #if QT_CONFIG(vulkan)
diff --git src/qtdiag/qtdiag.h src/qtdiag/qtdiag.h
index 359840647f3eb13a9673144c2477ecedd59d10bc..fd32bd825de3631c74b2465186aedd61d9b03f37 100644
--- src/qtdiag/qtdiag.h
+++ src/qtdiag/qtdiag.h
@@ -37,7 +37,9 @@ enum QtDiagFlags {
     QtDiagGl = 0x1,
     QtDiagGlExtensions = 0x2,
     QtDiagFonts = 0x4,
-    QtDiagVk = 0x8
+    QtDiagVk = 0x8,
+    QtDiagQSPTestMode = (1<<4),
+    QtDiagQSPXDGMode = (1<<5),
 };
 
 QString qtDiag(unsigned flags = 0);
