diff --git src/widgets/widgets/qcombobox.cpp src/widgets/widgets/qcombobox.cpp
index bdd2462c926813169c17f80e424194eb5010fb95..c87b7d44ac2e9345b3a5682d63b728eef43edef6 100644
--- src/widgets/widgets/qcombobox.cpp
+++ src/widgets/widgets/qcombobox.cpp
@@ -2793,7 +2793,7 @@ void QComboBox::showPopup()
 #if QT_CONFIG(effects)
     bool scrollDown = (listRect.topLeft() == below);
     if (QApplication::isEffectEnabled(Qt::UI_AnimateCombo)
-        && !style->styleHint(QStyle::SH_ComboBox_Popup, &opt, this) && !window()->testAttribute(Qt::WA_DontShowOnScreen))
+        && !usePopup && !window()->testAttribute(Qt::WA_DontShowOnScreen))
         qScrollEffect(container, scrollDown ? QEffects::DownScroll : QEffects::UpScroll, 150);
 #endif
 
@@ -2829,7 +2829,7 @@ void QComboBox::showPopup()
     view()->setFocus();
 
     view()->scrollTo(view()->currentIndex(),
-                     style->styleHint(QStyle::SH_ComboBox_Popup, &opt, this)
+                     usePopup
                              ? QAbstractItemView::PositionAtCenter
                              : QAbstractItemView::EnsureVisible);
 
