diff --git src/plugins/platforms/cocoa/qcocoatheme.mm src/plugins/platforms/cocoa/qcocoatheme.mm
index 7c10456824ce787cde6b704ebcf126223a0a7244..dd3816ad78f8c40ebd9b873ff10a090c1b2e2523 100644
--- src/plugins/platforms/cocoa/qcocoatheme.mm
+++ src/plugins/platforms/cocoa/qcocoatheme.mm
@@ -327,7 +327,7 @@ QVariant QCocoaTheme::themeHint(ThemeHint hint) const
 {
     switch (hint) {
     case QPlatformTheme::StyleNames:
-        return QStringList(QStringLiteral("macintosh"));
+        return QStringList({QStringLiteral("aqua"), QStringLiteral("macintosh")});
     case QPlatformTheme::DialogButtonBoxLayout:
         return QVariant(QPlatformDialogHelper::MacLayout);
     case KeyboardScheme:
