diff --git src/platformsupport/linuxaccessibility/constant_mappings.cpp src/platformsupport/linuxaccessibility/constant_mappings.cpp
index fce2919e73b3ffc3ddff22fc1bd645e049d6ff53..668ed0c28580415cb9f4e04d31cd4ac5e7783926 100644
--- src/platformsupport/linuxaccessibility/constant_mappings.cpp
+++ src/platformsupport/linuxaccessibility/constant_mappings.cpp
@@ -79,7 +79,11 @@ quint64 spiStatesFromQState(QAccessible::State state)
     if (state.checkStateMixed)
         setSpiStateBit(&spiState, ATSPI_STATE_INDETERMINATE);
     if (state.readOnly)
-        setSpiStateBit(&spiState, ATSPI_STATE_READ_ONLY);
+#ifdef ATSPI_STATE_INDETERMINATE
+        setSpiStateBit(&spiState, ATSPI_STATE_INDETERMINATE);
+#else
+        unsetSpiStateBit(&spiState, ATSPI_STATE_EDITABLE);
+#endif
     //        if (state.HotTracked)
     if (state.defaultButton)
         setSpiStateBit(&spiState, ATSPI_STATE_IS_DEFAULT);
