diff --git src/corelib/kernel/qobject.cpp src/corelib/kernel/qobject.cpp
index 644f1916..13e84244 100644
--- src/corelib/kernel/qobject.cpp
+++ src/corelib/kernel/qobject.cpp
@@ -216,9 +216,15 @@ QObjectPrivate::QObjectPrivate(int version)
     // This allows incompatible versions to be loaded, possibly for testing.
     Q_UNUSED(version);
 #else
-    if (Q_UNLIKELY(version != QObjectPrivateVersion))
-        qFatal("Cannot mix incompatible Qt library (version 0x%x) with this library (version 0x%x)",
-                version, QObjectPrivateVersion);
+    if (Q_UNLIKELY(version != QObjectPrivateVersion)) {
+        if (version < QObjectPrivateVersion && (QObjectPrivateVersion - version) < 0x100
+                && !qEnvironmentVariableIsSet("QT_VERSION_MISMATCH_FATAL"))
+            qCritical("!! Warning: mixing older Qt library (version 0x%x) with the currently running version 0x%x (QT_VERSION_MISMATCH_FATAL not set)",
+                    version, QObjectPrivateVersion);
+        else
+            qFatal("Cannot mix incompatible Qt library (version 0x%x) with this library (version 0x%x)",
+                    version, QObjectPrivateVersion);
+    }
 #endif
 
     // QObjectData initialization
