diff --git configure configure
index ef7bad1bfce5bb2cfb70ee58a3cea5854d1df4c8..223c4d6b354c31b276d9f8273e97bd86a34e8b5e 100755
--- configure
+++ configure
@@ -305,9 +305,9 @@ UNAME_RELEASE=`(uname -r) 2>/dev/null` || UNAME_RELEASE=unknown
 UNAME_SYSTEM=`(uname -s) 2>/dev/null`  || UNAME_SYSTEM=unknown
 
 BUILD_ON_MAC=no
-if [ -d /System/Library/Frameworks/Cocoa.framework ]; then
-    BUILD_ON_MAC=yes
-fi
+# if [ -d /System/Library/Frameworks/Cocoa.framework ]; then
+#     BUILD_ON_MAC=yes
+# fi
 if [ "$OSTYPE" = "msys" ]; then
     relpath=`(cd "$relpath"; pwd -W)`
     outpath=`pwd -W`
