diff --git src/plugins/platforms/cocoa/qcocoacolordialoghelper.mm src/plugins/platforms/cocoa/qcocoacolordialoghelper.mm
index d7850b148108c9de6e9171784f41577f1f129db5..b97f86e6ef1294311cf54844b935167d5cb112a0 100644
--- src/plugins/platforms/cocoa/qcocoacolordialoghelper.mm
+++ src/plugins/platforms/cocoa/qcocoacolordialoghelper.mm
@@ -122,6 +122,7 @@ QT_NAMESPACE_ALIAS_OBJC_CLASS(QNSColorPanelDelegate);
         // create a new content view and add the stolen one as a subview
         mPanelButtons = [[QNSPanelContentsWrapper alloc] initWithPanelDelegate:self];
         [mPanelButtons addSubview:mStolenContentView];
+        [mPanelButtons setFrameSize:[mPanelButtons fittingSize]];
         mColorPanel.contentView = mPanelButtons;
         mColorPanel.defaultButtonCell = mPanelButtons.okButton.cell;
     }
diff --git src/plugins/platforms/cocoa/qcocoafontdialoghelper.mm src/plugins/platforms/cocoa/qcocoafontdialoghelper.mm
index 8c0af97a6852d3ea834e767ba7b951e654bde029..1b3d29c87f6421dbe1212df2be20db1c2ec0b5bb 100644
--- src/plugins/platforms/cocoa/qcocoafontdialoghelper.mm
+++ src/plugins/platforms/cocoa/qcocoafontdialoghelper.mm
@@ -140,6 +140,7 @@ QT_NAMESPACE_ALIAS_OBJC_CLASS(QNSFontPanelDelegate);
         mPanelButtons = [[QNSPanelContentsWrapper alloc] initWithPanelDelegate:self];
         [mPanelButtons addSubview:mStolenContentView];
         mPanelButtons.panelContentsMargins = NSEdgeInsetsMake(0, 0, 7, 0);
+        [mPanelButtons setFrameSize:[mPanelButtons fittingSize]];
         mFontPanel.contentView = mPanelButtons;
         mFontPanel.defaultButtonCell = mPanelButtons.okButton.cell;
     }
