diff --git src/corelib/Qt5CoreConfigExtrasMkspecDir.cmake.in src/corelib/Qt5CoreConfigExtrasMkspecDir.cmake.in
index c357237d0e3d6e02d10325f0b9738b451ef34c2c..d5467879a05b9a5cc39056e52b06aa7256ce1ed8 100644
--- src/corelib/Qt5CoreConfigExtrasMkspecDir.cmake.in
+++ src/corelib/Qt5CoreConfigExtrasMkspecDir.cmake.in
@@ -1,6 +1,10 @@
 
+if(NOT CMAKE_MKSPEC)
+	set(CMAKE_MKSPEC "$${CMAKE_MKSPEC}")
+endif()
+
 !!IF isEmpty(CMAKE_HOST_DATA_DIR_IS_ABSOLUTE)
-set(_qt5_corelib_extra_includes \"${_qt5Core_install_prefix}/$${CMAKE_HOST_DATA_DIR}/mkspecs/$${CMAKE_MKSPEC}\")
+set(_qt5_corelib_extra_includes \"${_qt5Core_install_prefix}/$${CMAKE_HOST_DATA_DIR}/mkspecs/${CMAKE_MKSPEC}\")
 !!ELSE
-set(_qt5_corelib_extra_includes \"$${CMAKE_HOST_DATA_DIR}mkspecs/$${CMAKE_MKSPEC}\")
+set(_qt5_corelib_extra_includes \"$${CMAKE_HOST_DATA_DIR}mkspecs/${CMAKE_MKSPEC}\")
 !!ENDIF
diff --git src/corelib/Qt5CoreConfigExtrasMkspecDirForInstall.cmake.in src/corelib/Qt5CoreConfigExtrasMkspecDirForInstall.cmake.in
index 706304cf34e73b76fd26d82a2c59aa43942bf1e6..b2fff54289c3a1a6539c76aa8a66c8ce0523915b 100644
--- src/corelib/Qt5CoreConfigExtrasMkspecDirForInstall.cmake.in
+++ src/corelib/Qt5CoreConfigExtrasMkspecDirForInstall.cmake.in
@@ -1,4 +1,8 @@
 
+if(NOT CMAKE_MKSPEC)
+	set(CMAKE_MKSPEC "$${CMAKE_MKSPEC}")
+endif()
+
 !!IF isEmpty(CMAKE_INSTALL_DATA_DIR_IS_ABSOLUTE)
 set(_qt5_corelib_extra_includes \"${_qt5Core_install_prefix}/$${CMAKE_INSTALL_DATA_DIR}/mkspecs/$${CMAKE_MKSPEC}\")
 !!ELSE
