diff --git mkspecs/features/toolchain.prf mkspecs/features/toolchain.prf
index 03612e56896e4d404f1a0c2de82d905ecbebc615..fc6ebb601a5bd0e1b824e9b100c90e2b0c8473e8 100644
--- mkspecs/features/toolchain.prf
+++ mkspecs/features/toolchain.prf
@@ -173,7 +173,7 @@ isEmpty($${target_prefix}.INCDIRS) {
         else: \
             cmd_prefix = "LC_ALL=C"
 
-        cxx_flags = $$QMAKE_CXXFLAGS
+        !darwin: cxx_flags = $$QMAKE_CXXFLAGS
 
         # Manually inject the sysroot for Apple Platforms because its resolution
         # normally does not happen until default_post.prf. This is especially
@@ -182,6 +182,9 @@ isEmpty($${target_prefix}.INCDIRS) {
         # UIKit simulator platforms will see the device SDK's sysroot in
         # QMAKE_DEFAULT_*DIRS, because they're handled in a single build pass.
         darwin {
+            opts = $$split(QMAKE_CXXFLAGS, " ")
+            for (opt, opts): \
+                !equals(opt, "-g"): cxx_flags += $$opt
             uikit:macx-xcode: \
                 cxx_flags += -isysroot $$sdk_path_device.value
             else: \
@@ -396,7 +399,7 @@ isEmpty($${target_prefix}.INCDIRS) {
     }
 
     unix:if(!cross_compile|host_build) {
-        isEmpty(QMAKE_DEFAULT_INCDIRS): QMAKE_DEFAULT_INCDIRS = /usr/include /usr/local/include
+        isEmpty(QMAKE_DEFAULT_INCDIRS): QMAKE_DEFAULT_INCDIRS = /usr/include @PREFIX@/include
         isEmpty(QMAKE_DEFAULT_LIBDIRS): QMAKE_DEFAULT_LIBDIRS = /lib /usr/lib
     }
 
