diff --git Makefile.in Makefile.in
index 6a7c8f6d4d19ea23522fa69023e6495a71c04ace..d7d3296b6ccf23cdda5e1c0858b9f7bf06792cc0 100644
--- Makefile.in
+++ Makefile.in
@@ -27,6 +27,9 @@ LD=@LD@
 BUILDINFO_ARCH_NAME=$(shell $(GCC) -dumpmachine)
 BUILDINFO_BUILD_DATE=$(shell date)
 
+SHLIBEXT=so
+MMBD_LDFLAGS= -shared -Wl,-z,defs -fPIC -Xlinker -dy -Xlinker --version-script=libmmbd/src/libmmbd.vers -Xlinker -soname=libmmbd.$(SHLIBEXT) -lrt -pthread
+
 top_srcdir ?= .
 INCF=-I$(top_srcdir)/
 
@@ -38,37 +41,23 @@ ifeq ($(ENABLE_DEBUG),no)
 STRIP_FLAGS=--strip-all --strip-debug --strip-unneeded --discard-all
 endif
 
-all: out/libdriveio.so.0 out/libmakemkv.so.1 $(OUT_GUI) out/libmmbd.so.0
+all: out/libmmbd.$(SHLIBEXT)
 	@echo "type \"sudo make install\" to install"
 
 clean:
 	-rm -rf out tmp
 
-install: out/libdriveio.so.0 out/libmakemkv.so.1 $(OUT_GUI) out/libmmbd.so.0
-	$(INSTALL) -D -m 644 out/libdriveio.so.0 $(DESTDIR)$(libdir)/libdriveio.so.0
-	$(INSTALL) -D -m 644 out/libmakemkv.so.1 $(DESTDIR)$(libdir)/libmakemkv.so.1
-	$(INSTALL) -D -m 644 out/libmmbd.so.0 $(DESTDIR)$(libdir)/libmmbd.so.0
-ifeq ($(DESTDIR),)
-	ldconfig
-endif
-ifeq ($(ENABLE_GUI),yes)
-	$(INSTALL) -D -m 755 out/makemkv $(DESTDIR)$(bindir)/makemkv
-	$(INSTALL) -D -m 644 makemkvgui/share/makemkv.desktop $(DESTDIR)$(datadir)/applications/makemkv.desktop
-	$(INSTALL) -D -m 644 makemkvgui/share/icons/16x16/makemkv.png $(DESTDIR)$(datadir)/icons/hicolor/16x16/apps/makemkv.png
-	$(INSTALL) -D -m 644 makemkvgui/share/icons/22x22/makemkv.png $(DESTDIR)$(datadir)/icons/hicolor/22x22/apps/makemkv.png
-	$(INSTALL) -D -m 644 makemkvgui/share/icons/32x32/makemkv.png $(DESTDIR)$(datadir)/icons/hicolor/32x32/apps/makemkv.png
-	$(INSTALL) -D -m 644 makemkvgui/share/icons/64x64/makemkv.png $(DESTDIR)$(datadir)/icons/hicolor/64x64/apps/makemkv.png
-	$(INSTALL) -D -m 644 makemkvgui/share/icons/128x128/makemkv.png $(DESTDIR)$(datadir)/icons/hicolor/128x128/apps/makemkv.png
-endif
+install: out/libmmbd.$(SHLIBEXT)
+	$(INSTALL) -D -m 644 out/libmmbd.$(SHLIBEXT) $(DESTDIR)$(libdir)/libmmbd.$(SHLIBEXT)
 
 out/%: out/%.full
 	$(OBJCOPY) $(STRIP_FLAGS) $< $@
 
-out/libdriveio.so.0.full:
+out/libdriveio.$(SHLIBEXT).full:
 	mkdir -p out
 	$(GCC) $(CFLAGS) $(LDFLAGS) -D_REENTRANT -shared -Wl,-z,defs -o$@ $(INCF)$(LIBDRIVEIO_INC) $(LIBDRIVEIO_SRC) \
 	-fPIC -Xlinker -dy -Xlinker --version-script=libdriveio/src/libdriveio.vers \
-	-Xlinker -soname=libdriveio.so.0 -lc -lstdc++
+	-Xlinker -soname=libdriveio.$(SHLIBEXT) -lc -lstdc++
 
 out/libmakemkv.so.1.full: tmp/gen_buildinfo.h
 	mkdir -p out
@@ -80,12 +69,11 @@ out/libmakemkv.so.1.full: tmp/gen_buildinfo.h
 	-fPIC -Xlinker -dy -Xlinker --version-script=libmakemkv/src/libmakemkv.vers \
 	-Xlinker -soname=libmakemkv.so.1 -lc -lstdc++ -lcrypto -lz -lexpat $(FFMPEG_LIBS) -lm -lrt
 
-out/libmmbd.so.0.full:
+out/libmmbd.$(SHLIBEXT).full:
 	mkdir -p out
-	$(GCC) $(CFLAGS) $(LDFLAGS) -D_REENTRANT -shared -Wl,-z,defs -o$@ $(INCF)$(MAKEMKVGUI_INC) $(INCF)$(LIBMMBD_INC) \
+	$(CXX) $(CFLAGS) $(LDFLAGS) -D_REENTRANT -o$@ $(INCF)$(MAKEMKVGUI_INC) $(INCF)$(LIBMMBD_INC) \
 	$(INCF)$(LIBMAKEMKV_INC) $(INCF)$(SSTRING_INC) $(INCF)$(LIBABI_INC) $(LIBMMBD_SRC) $(LIBMMBD_SRC_LINUX) $(SSTRING_SRC) \
-	-fPIC -Xlinker -dy -Xlinker --version-script=libmmbd/src/libmmbd.vers \
-	-Xlinker -soname=libmmbd.so.0 -lc -lstdc++ -lrt -lpthread -lcrypto
+	$(MMBD_LDFLAGS) -lcrypto
 
 out/makemkv.full: $(MAKEMKVGUI_GEN) $(MAKEMKVGUI_SRC_LINUX) tmp/gen_buildinfo.h
 	mkdir -p out
diff --git configure configure
index 92d8811d94e0e7a1edc9e1a1405ba5d3ef94484c..d35b61179ea016e08552afe4a6f91ffe104f46b8 100755
--- configure
+++ configure
@@ -1395,9 +1395,9 @@ Optional Features:
   --enable-fast-install[=PKGS]
                           optimize for fast installation [default=yes]
   --disable-libtool-lock  avoid locking (might break parallel builds)
-  --disable-gui           do not build GUI (default is build GUI)
-  --disable-qt5           do not use Qt5
-  --disable-qt4           do not use Qt4
+  --enable-gui            build GUI (default is don't build GUI)
+  --enable-qt5            use Qt5
+  --enable-qt4            use Qt4
   --enable-noec           use private ECDSA code (default - use openssl ECDSA
                           if available)
   --enable-debug          keep debug symbols in binarfies (default - strip
@@ -2819,7 +2819,7 @@ test -n "$target_alias" &&
   test "$program_prefix$program_suffix$program_transform_name" = \
     NONENONEs,x,x, &&
   program_prefix=${target_alias}-
-ac_config_headers="$ac_config_headers libffabi/src/ffabicfg.h"
+# ac_config_headers="$ac_config_headers libffabi/src/ffabicfg.h"
 
 
 
@@ -11185,7 +11185,7 @@ CC="$lt_save_CC"
 if test "${enable_gui+set}" = set; then :
   enableval=$enable_gui;
 else
-  enable_gui=yes
+  enable_gui=no
 fi
 
 ENABLE_GUI=$enable_gui
@@ -11196,7 +11196,7 @@ ENABLE_GUI=$enable_gui
 if test "${enable_qt5+set}" = set; then :
   enableval=$enable_qt5;
 else
-  enable_qt5=yes
+  enable_qt5=no
 fi
 
 
@@ -11204,7 +11204,7 @@ fi
 if test "${enable_qt4+set}" = set; then :
   enableval=$enable_qt4;
 else
-  enable_qt4=yes
+  enable_qt4=no
 fi
 
 
@@ -15629,72 +15629,7 @@ See \`config.log' for more details" "$LINENO" 5; }
 fi
 
 
-# expat
-for ac_header in expat.h
-do :
-  ac_fn_c_check_header_mongrel "$LINENO" "expat.h" "ac_cv_header_expat_h" "$ac_includes_default"
-if test "x$ac_cv_header_expat_h" = xyes; then :
-  cat >>confdefs.h <<_ACEOF
-#define HAVE_EXPAT_H 1
-_ACEOF
-
-else
-
-  { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
-$as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
-as_fn_error $? "expat library header files not found
-See \`config.log' for more details" "$LINENO" 5; }
-
-fi
-
-done
-
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for XML_ParserCreate in -lexpat" >&5
-$as_echo_n "checking for XML_ParserCreate in -lexpat... " >&6; }
-if ${ac_cv_lib_expat_XML_ParserCreate+:} false; then :
-  $as_echo_n "(cached) " >&6
-else
-  ac_check_lib_save_LIBS=$LIBS
-LIBS="-lexpat  $LIBS"
-cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
-
-/* Override any GCC internal prototype to avoid an error.
-   Use char because int might match the return type of a GCC
-   builtin and then its argument prototype would still apply.  */
-#ifdef __cplusplus
-extern "C"
-#endif
-char XML_ParserCreate ();
-int
-main ()
-{
-return XML_ParserCreate ();
-  ;
-  return 0;
-}
-_ACEOF
-if ac_fn_c_try_link "$LINENO"; then :
-  ac_cv_lib_expat_XML_ParserCreate=yes
-else
-  ac_cv_lib_expat_XML_ParserCreate=no
-fi
-rm -f core conftest.err conftest.$ac_objext \
-    conftest$ac_exeext conftest.$ac_ext
-LIBS=$ac_check_lib_save_LIBS
-fi
-{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_expat_XML_ParserCreate" >&5
-$as_echo "$ac_cv_lib_expat_XML_ParserCreate" >&6; }
-if test "x$ac_cv_lib_expat_XML_ParserCreate" = xyes; then :
-  :
-else
-
-  { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
-$as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
-as_fn_error $? "expat library not found
-See \`config.log' for more details" "$LINENO" 5; }
-
-fi
+# expat checking removed
 
 
 # ffmpeg
@@ -15819,307 +15754,7 @@ $as_echo "no" >&6; }
 	fi
 fi
 
-pkg_failed=no
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for ffmpeg" >&5
-$as_echo_n "checking for ffmpeg... " >&6; }
-
-if test -n "$ffmpeg_CFLAGS"; then
-    pkg_cv_ffmpeg_CFLAGS="$ffmpeg_CFLAGS"
- elif test -n "$PKG_CONFIG"; then
-    if test -n "$PKG_CONFIG" && \
-    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"libavcodec libavutil\""; } >&5
-  ($PKG_CONFIG --exists --print-errors "libavcodec libavutil") 2>&5
-  ac_status=$?
-  $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
-  test $ac_status = 0; }; then
-  pkg_cv_ffmpeg_CFLAGS=`$PKG_CONFIG --cflags "libavcodec libavutil" 2>/dev/null`
-		      test "x$?" != "x0" && pkg_failed=yes
-else
-  pkg_failed=yes
-fi
- else
-    pkg_failed=untried
-fi
-if test -n "$ffmpeg_LIBS"; then
-    pkg_cv_ffmpeg_LIBS="$ffmpeg_LIBS"
- elif test -n "$PKG_CONFIG"; then
-    if test -n "$PKG_CONFIG" && \
-    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"libavcodec libavutil\""; } >&5
-  ($PKG_CONFIG --exists --print-errors "libavcodec libavutil") 2>&5
-  ac_status=$?
-  $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
-  test $ac_status = 0; }; then
-  pkg_cv_ffmpeg_LIBS=`$PKG_CONFIG --libs "libavcodec libavutil" 2>/dev/null`
-		      test "x$?" != "x0" && pkg_failed=yes
-else
-  pkg_failed=yes
-fi
- else
-    pkg_failed=untried
-fi
-
-
-
-if test $pkg_failed = yes; then
-   	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
-$as_echo "no" >&6; }
-
-if $PKG_CONFIG --atleast-pkgconfig-version 0.20; then
-        _pkg_short_errors_supported=yes
-else
-        _pkg_short_errors_supported=no
-fi
-        if test $_pkg_short_errors_supported = yes; then
-	        ffmpeg_PKG_ERRORS=`$PKG_CONFIG --short-errors --print-errors --cflags --libs "libavcodec libavutil" 2>&1`
-        else
-	        ffmpeg_PKG_ERRORS=`$PKG_CONFIG --print-errors --cflags --libs "libavcodec libavutil" 2>&1`
-        fi
-	# Put the nasty error message in config.log where it belongs
-	echo "$ffmpeg_PKG_ERRORS" >&5
-
-	 as_fn_error $? "${ffmpeg_PKG_ERRORS}." "$LINENO" 5
-elif test $pkg_failed = untried; then
-     	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
-$as_echo "no" >&6; }
-	 as_fn_error $? "${ffmpeg_PKG_ERRORS}." "$LINENO" 5
-else
-	ffmpeg_CFLAGS=$pkg_cv_ffmpeg_CFLAGS
-	ffmpeg_LIBS=$pkg_cv_ffmpeg_LIBS
-        { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
-$as_echo "yes" >&6; }
-
-fi
-
-OLD_CPPFLAGS="$CPPFLAGS"
-OLD_LIBS="$LIBS"
-
-CPPFLAGS="$ffmpeg_CFLAGS"
-LIBS="$ffmpeg_LIBS -lrt"
-
-# check for a non-ancient libavcodec
-ac_fn_c_check_decl "$LINENO" "LIBAVCODEC_VERSION_MAJOR" "ac_cv_have_decl_LIBAVCODEC_VERSION_MAJOR" "#include <libavcodec/avcodec.h>
-"
-if test "x$ac_cv_have_decl_LIBAVCODEC_VERSION_MAJOR" = xyes; then :
-
-else
-
-  { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
-$as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
-as_fn_error $? "LIBAVCODEC_VERSION_MAJOR is not declared in libavcodec.h
-See \`config.log' for more details" "$LINENO" 5; }
-
-fi
-
-
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking LIBAVCODEC_VERSION_MAJOR" >&5
-$as_echo_n "checking LIBAVCODEC_VERSION_MAJOR... " >&6; }
-if ac_fn_c_compute_int "$LINENO" "LIBAVCODEC_VERSION_MAJOR" "LIBAVCODEC_VERSION_MAJOR"        "
-  #include <libavcodec/avcodec.h>
-  "; then :
-
-else
-
-  { $as_echo "$as_me:${as_lineno-$LINENO}: result: failed" >&5
-$as_echo "failed" >&6; }
-  { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
-$as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
-as_fn_error $? "LIBAVCODEC_VERSION_MAJOR is not known at compile time in libavcodec.h
-See \`config.log' for more details" "$LINENO" 5; }
-
-fi
-
-{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $LIBAVCODEC_VERSION_MAJOR" >&5
-$as_echo "$LIBAVCODEC_VERSION_MAJOR" >&6; }
-
-# check for ancient frame format
-ac_fn_c_check_member "$LINENO" "AVFrame" "nb_samples" "ac_cv_member_AVFrame_nb_samples" "#include <libavcodec/avcodec.h>
-"
-if test "x$ac_cv_member_AVFrame_nb_samples" = xyes; then :
-
-else
-
-  LIBAVCODEC_VERSION_MAJOR=0;
-
-fi
-
-
-# we need planar audio
-ac_fn_c_check_decl "$LINENO" "AV_SAMPLE_FMT_U8P" "ac_cv_have_decl_AV_SAMPLE_FMT_U8P" "#include <libavutil/samplefmt.h>
-"
-if test "x$ac_cv_have_decl_AV_SAMPLE_FMT_U8P" = xyes; then :
-
-else
-
-  LIBAVCODEC_VERSION_MAJOR=0;
-
-fi
-
-
-# avcodec_encode_audio2
-ac_fn_c_check_func "$LINENO" "avcodec_encode_audio2" "ac_cv_func_avcodec_encode_audio2"
-if test "x$ac_cv_func_avcodec_encode_audio2" = xyes; then :
-
-else
-
-  LIBAVCODEC_VERSION_MAJOR=0;
-
-fi
-
-
-if test $LIBAVCODEC_VERSION_MAJOR -lt 53; then
-  as_fn_error $? "The libavcodec library is too old. Please get a recent one from http://www.ffmpeg.org" "$LINENO" 5
-fi
-
-# check various ffmpeg features
-ac_fn_c_check_member "$LINENO" "AVCodecContext" "refcounted_frames" "ac_cv_member_AVCodecContext_refcounted_frames" "#include <libavcodec/avcodec.h>
-"
-if test "x$ac_cv_member_AVCodecContext_refcounted_frames" = xyes; then :
-
-
-$as_echo "#define FFABI_HAVE_REFCOUNTED_FRAMES 1" >>confdefs.h
-
-
-fi
-
-
-ac_fn_c_check_decl "$LINENO" "avcodec_free_frame" "ac_cv_have_decl_avcodec_free_frame" "#include <libavcodec/avcodec.h>
-"
-if test "x$ac_cv_have_decl_avcodec_free_frame" = xyes; then :
-
-
-$as_echo "#define FFABI_HAVE_AVCODEC_FREE_FRAME 1" >>confdefs.h
-
-
-fi
-
-
-ac_fn_c_check_decl "$LINENO" "av_frame_free" "ac_cv_have_decl_av_frame_free" "#include <libavutil/frame.h>
-"
-if test "x$ac_cv_have_decl_av_frame_free" = xyes; then :
-
-
-$as_echo "#define FFABI_HAVE_AV_FRAME_FREE 1" >>confdefs.h
-
-
-fi
-
-
-ac_fn_c_check_func "$LINENO" "av_log_format_line" "ac_cv_func_av_log_format_line"
-if test "x$ac_cv_func_av_log_format_line" = xyes; then :
-
-
-$as_echo "#define FFABI_HAVE_AV_LOG_FORMAT_LINE 1" >>confdefs.h
-
-
-fi
-
-
-ac_fn_c_check_type "$LINENO" "enum AVCodecID" "ac_cv_type_enum_AVCodecID" "#include <libavcodec/avcodec.h>
-"
-if test "x$ac_cv_type_enum_AVCodecID" = xyes; then :
-
-
-$as_echo "#define FFABI_HAVE_AVCODECID 1" >>confdefs.h
-
-
-else
-
-  ac_fn_c_check_type "$LINENO" "enum CodecID" "ac_cv_type_enum_CodecID" "#include <libavcodec/avcodec.h>
-"
-if test "x$ac_cv_type_enum_CodecID" = xyes; then :
-
-else
-
-    as_fn_error $? "Neither AVCodecID nor CodecID defined in libavcodec.h" "$LINENO" 5
-
-fi
-
-
-fi
-
-
-ac_fn_c_check_decl "$LINENO" "AV_CODEC_ID_NONE" "ac_cv_have_decl_AV_CODEC_ID_NONE" "#include <libavcodec/avcodec.h>
-"
-if test "x$ac_cv_have_decl_AV_CODEC_ID_NONE" = xyes; then :
-
-
-$as_echo "#define FFABI_HAVE_AV_CODEC_ID_NAMES 1" >>confdefs.h
-
-
-else
-
-  ac_fn_c_check_decl "$LINENO" "CODEC_ID_NONE" "ac_cv_have_decl_CODEC_ID_NONE" "#include <libavcodec/avcodec.h>
-"
-if test "x$ac_cv_have_decl_CODEC_ID_NONE" = xyes; then :
-
-else
-
-    as_fn_error $? "Neither AV_CODEC_ID_XXX nor CODEC_ID_XXX defined in libavcodec.h" "$LINENO" 5
-
-fi
-
-
-fi
-
-
-ac_fn_c_check_decl "$LINENO" "av_frame_get_channels" "ac_cv_have_decl_av_frame_get_channels" "#include <libavcodec/avcodec.h>
-"
-if test "x$ac_cv_have_decl_av_frame_get_channels" = xyes; then :
-
-
-$as_echo "#define FFABI_HAVE_AV_FRAME_CHANNELS 1" >>confdefs.h
-
-
-fi
-
-
-ac_fn_c_check_decl "$LINENO" "av_frame_get_sample_rate" "ac_cv_have_decl_av_frame_get_sample_rate" "#include <libavcodec/avcodec.h>
-"
-if test "x$ac_cv_have_decl_av_frame_get_sample_rate" = xyes; then :
-
-
-$as_echo "#define FFABI_HAVE_AV_FRAME_SAMPLE_RATE 1" >>confdefs.h
-
-
-fi
-
-
-ac_fn_c_check_decl "$LINENO" "av_frame_set_channel_layout" "ac_cv_have_decl_av_frame_set_channel_layout" "#include <libavcodec/avcodec.h>
-"
-if test "x$ac_cv_have_decl_av_frame_set_channel_layout" = xyes; then :
-
-
-$as_echo "#define FFABI_HAVE_AV_FRAME_CHANNEL_LAYOUT 1" >>confdefs.h
-
-
-fi
-
-
-ac_fn_c_check_member "$LINENO" "AVCodecParserContext" "duration" "ac_cv_member_AVCodecParserContext_duration" "#include <libavcodec/avcodec.h>
-"
-if test "x$ac_cv_member_AVCodecParserContext_duration" = xyes; then :
-
-
-$as_echo "#define FFABI_HAVE_PARSER_DURATION 1" >>confdefs.h
-
-
-fi
-
-
-ac_fn_c_check_decl "$LINENO" "AV_CH_TOP_BACK_CENTER" "ac_cv_have_decl_AV_CH_TOP_BACK_CENTER" "#include <libavutil/channel_layout.h>
-"
-if test "x$ac_cv_have_decl_AV_CH_TOP_BACK_CENTER" = xyes; then :
-
-
-$as_echo "#define FFABI_HAVE_AVUTIL_CHANNEL_LAYOUT_H 1" >>confdefs.h
-
-
-fi
-
-
-
-CPPFLAGS="$OLD_CPPFLAGS"
-LIBS="$OLD_LIBS"
+# ffmpeg checking removed
 
 qt_ok=no
 if test "x$enable_gui" != "xno"; then :
