diff --git Makefile.in Makefile.in
index d7d3296b6ccf23cdda5e1c0858b9f7bf06792cc0..618dca43d0a16815f0b1f2330fa9e0a53b9ed11a 100644
--- Makefile.in
+++ Makefile.in
@@ -72,7 +72,7 @@ out/libmakemkv.so.1.full: tmp/gen_buildinfo.h
 out/libmmbd.$(SHLIBEXT).full:
 	mkdir -p out
 	$(CXX) $(CFLAGS) $(LDFLAGS) -D_REENTRANT -o$@ $(INCF)$(MAKEMKVGUI_INC) $(INCF)$(LIBMMBD_INC) \
-	$(INCF)$(LIBMAKEMKV_INC) $(INCF)$(SSTRING_INC) $(INCF)$(LIBABI_INC) $(LIBMMBD_SRC) $(LIBMMBD_SRC_LINUX) $(SSTRING_SRC) \
+	$(INCF)$(LIBMAKEMKV_INC) $(INCF)$(SSTRING_INC) $(INCF)$(LIBABI_INC) $(LIBMMBD_SRC) $(LIBMMBD_SRC_DARWIN) $(SSTRING_SRC) \
 	$(MMBD_LDFLAGS) -lcrypto
 
 out/makemkv.full: $(MAKEMKVGUI_GEN) $(MAKEMKVGUI_SRC_LINUX) tmp/gen_buildinfo.h
diff --git makefile.common makefile.common
index 97663626f4c321bbf35ea47fe63dc59f0f1962ad..bcb166ce7da7d3681dc9157c9131e4d8edb8df12 100644
--- makefile.common
+++ makefile.common
@@ -80,3 +80,6 @@ LIBMMBD_SRC=makemkvgui/src/client.cpp makemkvgui/src/marshall.cpp libmmbd/src/ma
 LIBMMBD_SRC_LINUX=makemkvgui/src/api_posix.cpp makemkvgui/src/api_linux.cpp makemkvgui/src/sem_posix.cpp \
   makemkvgui/src/spawn_posix.cpp
 
+LIBMMBD_SRC_DARWIN=makemkvgui/src/api_posix.cpp makemkvgui/src/api_darwin.cpp makemkvgui/src/sem_posix.cpp \
+  makemkvgui/src/spawn_posix.cpp
+
