diff --git makemkvgui/src/sem_posix.cpp makemkvgui/src/sem_posix.cpp
index 037c11045628b3d8edaaf938ab4138dd901f9074..95eaf110eec54456893eebac786ac34f19c2ca54 100644
--- makemkvgui/src/sem_posix.cpp
+++ makemkvgui/src/sem_posix.cpp
@@ -24,6 +24,12 @@
 #include <fcntl.h>
 #include <stdio.h>
 
+#ifdef __APPLE__
+extern "C" {
+int sem_timedwait(sem_t *, const struct timespec *);
+}
+#endif
+
 bool ApOpenSgrp(AP_SGRP *sgrp,const uint64_t* data)
 {
     sem_t* psem[2];
diff --git sstring/src/sstring.cpp sstring/src/sstring.cpp
index cd808a392afc237303d645b05d9bb31de50e6184..fa7c0b162395708706b9b0cfca3a7a99e26797c6 100644
--- sstring/src/sstring.cpp
+++ sstring/src/sstring.cpp
@@ -13,7 +13,7 @@
 
 #ifndef ALREADY_HAVE_SSTRING_API
 
-#if defined(__GLIBC__) || defined(_darwin_)
+#if defined(__GLIBC__) || defined(__APPLE__)
 #define _vsnprintf vsnprintf
 #define _vsnwprintf vswprintf
 #endif
