--- a/src/orig.main.cpp	2016-08-10 01:01:34.000000000 +0200
+++ b/src/main.cpp	2016-09-06 16:40:44.000000000 +0200
@@ -86,10 +86,10 @@
     // will work correctly in the terminal
     app->setAttribute(Qt::AA_MacDontSwapCtrlAndMeta);
 
-    // KDE's menuBar()->isTopLevel() hasn't worked in a while.
-    // For now, put menus inside Konsole window; this also make
-    // the keyboard shortcut to show menus look reasonable.
-    app->setAttribute(Qt::AA_DontUseNativeMenuBar);
+    // KDE's menuBar()->isTopLevel() hasn't worked in a while.
+    // For now, put menus inside Konsole window; this also make
+    // the keyboard shortcut to show menus look reasonable.
+//     app->setAttribute(Qt::AA_DontUseNativeMenuBar);
 #endif
 
     KLocalizedString::setApplicationDomain("konsole");
