diff --git runtime/tools/orig.check-depends.py runtime/tools/check-depends.py
index f185900..99e063e 100644
--- runtime/tools/orig.check-depends.py
+++ runtime/tools/check-depends.py
@@ -152,10 +152,9 @@ def main():
     for dep in deps:
         print_info_line("    {}".format(dep))
     if unexpected:
-        print_error_line("Unexpected Dependencies:")
+        print_error_line("Unexpected Dependencies (ignoring):")
         for dep in unexpected:
             print_error_line("    {}".format(dep))
-        error("found unexpected dependencies")
 
 
 if __name__ == "__main__":
