diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9ea1237bc5708f3ad3bd76e54d8a388e9c7fbdfa..38d9033d334ff9d7f8f06bdfbbe0a4d62a0075d2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -8,7 +8,7 @@ project(kgapi VERSION ${KGAPI_LIB_VERSION})
 set(CMAKE_CXX_STANDARD 14)
 
 # ECM setup
-set(KF5_MIN_VERSION "5.64.0")
+set(KF5_MIN_VERSION "5.60.0")
 
 find_package(ECM ${KF5_MIN_VERSION} CONFIG REQUIRED)
 set(CMAKE_MODULE_PATH ${ECM_MODULE_PATH})
@@ -34,24 +34,38 @@ ecm_setup_version(PROJECT
     SOVERSION 5
 )
 
+option(ENABLE_QTWEBKIT "Enable QtWebKit backend (requires QtWebKit 5.212)" OFF)
+
 ############## Find Packages ##############
-set(QT_REQUIRED_VERSION "5.12.0")
+set(QT_REQUIRED_VERSION "5.9.0")
 find_package(Qt5 ${QT_REQUIRED_VERSION} REQUIRED COMPONENTS
     Core
     Network
     Widgets
-    WebEngineWidgets
     Xml
 )
 
+if(ENABLE_QTWEBKIT)
+    find_package(Qt5WebKitWidgets 5.212.0 REQUIRED)
+    set_package_properties(Qt5WebKit PROPERTIES URL "https://qtwebkit.github.io/" DESCRIPTION "QtWebKit based backend (core)" TYPE REQUIRED)
+    set_package_properties(Qt5WebKitWidgets PROPERTIES URL "https://qtwebkit.github.io/" DESCRIPTION "QtWebKit based backend (widgets)" TYPE REQUIRED)
+else()
+    find_package(Qt5 ${QT_REQUIRED_VERSION} REQUIRED COMPONENTS
+        WebEngineWidgets
+    )
+endif()
+
 find_package(KF5 ${KF5_MIN_VERSION} REQUIRED COMPONENTS
     KIO
     WindowSystem
     Wallet
 )
 
-find_package(KF5CalendarCore ${KF5_MIN_VERSION} CONFIG REQUIRED)
-find_package(KF5Contacts ${KF5_MIN_VERSION} CONFIG REQUIRED)
+set(KCALENDARCORE_LIB_VERSION "5.12.2")
+set(KCONTACTS_LIB_VERSION "5.12.2")
+
+find_package(KF5CalendarCore ${KCALENDARCORE_LIB_VERSION} CONFIG REQUIRED)
+find_package(KF5Contacts ${KCONTACTS_LIB_VERSION} CONFIG REQUIRED)
 
 find_package(Sasl2)
 set_package_properties(Sasl2 PROPERTIES TYPE REQUIRED)
diff --git a/KPimGAPIConfig.cmake.in b/KPimGAPIConfig.cmake.in
index 366c2960cc2e98bcb1fe7dd1ef7dba0d1a445614..3d0db46b808754ceef4a0dfc0494379bcab73c6a 100644
--- a/KPimGAPIConfig.cmake.in
+++ b/KPimGAPIConfig.cmake.in
@@ -2,8 +2,8 @@
 
 include(CMakeFindDependencyMacro)
 
-find_dependency(KF5CalendarCore "@KF5_MIN_VERSION@")
-find_dependency(KF5Contacts "@KF5_MIN_VERSION@")
+find_dependency(KF5CalendarCore "@KCALENDARCORE_LIB_VERSION@")
+find_dependency(KF5Contacts "@KCONTACTS_LIB_VERSION@")
 
 if (@KGAPI_KF5_COMPAT@)
     message(WARNING "\n"
diff --git a/src/calendar/calendar.cpp b/src/calendar/calendar.cpp
index 92d5c6c0ec3e0a11dfc6a453ea40b71d568d5f68..a0b385812ecfd1aecdb63a650d2681d18db389ee 100644
--- a/src/calendar/calendar.cpp
+++ b/src/calendar/calendar.cpp
@@ -52,7 +52,7 @@ Calendar::Calendar() :
 
 Calendar::Calendar(const Calendar &other) :
     Object(other),
-    d(new Private(*(other.d.get())))
+    d(new Private(*(other.d)))
 {
 }
 
diff --git a/src/calendar/calendarservice.cpp b/src/calendar/calendarservice.cpp
index cfb94236cdb32d768cc7d31a1fa6b0d6aa5ffcbf..4bb6a6bdcbe5ec4aa5e20344e2abcd4dcde6aad8 100644
--- a/src/calendar/calendarservice.cpp
+++ b/src/calendar/calendarservice.cpp
@@ -1087,7 +1087,7 @@ QString Private::checkAndConverCDOTZID(const QString &tzid, const EventPtr& even
     const QStringList properties = vcard.split(QLatin1Char('\n'));
     int CDOId = -1;
     for (const QString &property : properties) {
-        if (property.startsWith(u"X-MICROSOFT-CDO-TZID")) {
+        if (property.startsWith(QStringLiteral(u"X-MICROSOFT-CDO-TZID"))) {
             QStringList parsed = property.split(QLatin1Char(':'));
             if (parsed.length() != 2) {
                 break;
diff --git a/src/core/authjob.cpp b/src/core/authjob.cpp
index 7c2e7e71b8e4063d3a1e99526f1d7ea5131b639c..8c0dbf5f15de1e384d0f33dbd3fc2f079c496216 100644
--- a/src/core/authjob.cpp
+++ b/src/core/authjob.cpp
@@ -239,7 +239,7 @@ void AuthJob::start()
         d->dialog->setModal(true);
         d->dialog->resize(840, 760);
         d->dialog->setAttribute(Qt::WA_NativeWindow, true);
-        KWindowSystem::setMainWindow(d->dialog->windowHandle(), KWindowSystem::activeWindow());
+        KWindowSystem::setMainWindow(d->dialog, KWindowSystem::activeWindow());
 
         QVBoxLayout *layout = new QVBoxLayout(d->dialog);
         layout->addWidget(widget, 2);
