diff --git a/CMakeLists.txt b/CMakeLists.txt
index b9a431b..5cb472a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,7 +6,7 @@ set(KGAPI_LIB_VERSION ${PIM_VERSION})
 project(kgapi VERSION ${KGAPI_LIB_VERSION})
 
 # ECM setup
-set(KF5_MIN_VERSION "5.56.0")
+set(KF5_MIN_VERSION "5.52.0")
 
 find_package(ECM ${KF5_MIN_VERSION} CONFIG REQUIRED)
 set(CMAKE_MODULE_PATH ${ECM_MODULE_PATH})
@@ -24,8 +24,8 @@ include(KDEFrameworkCompilerSettings NO_POLICY_SCOPE)
 
 
 
-set(KCALENDARCORE_LIB_VERSION "5.10.80")
-set(KCONTACTS_LIB_VERSION "5.10.80")
+set(KCALENDARCORE_LIB_VERSION "5.10.3")
+set(KCONTACTS_LIB_VERSION "5.10.3")
 
 ecm_setup_version(PROJECT
     VARIABLE_PREFIX KGAPI
@@ -35,7 +35,7 @@ ecm_setup_version(PROJECT
 )
 
 ############## Find Packages ##############
-set(QT_REQUIRED_VERSION "5.10.0")
+set(QT_REQUIRED_VERSION "5.9.0")
 find_package(Qt5 ${QT_REQUIRED_VERSION} REQUIRED COMPONENTS
     Core
     Network
