diff --git a/src/saslplugin/plugin_common.c b/src/saslplugin/plugin_common.c
index 73ad7a9..862630c 100644
--- a/src/saslplugin/plugin_common.c
+++ b/src/saslplugin/plugin_common.c
@@ -43,7 +43,7 @@
  */
 
 #include <config.h>
-#ifndef macintosh
+// #ifndef macintosh
 #ifdef WIN32
 # include <winsock2.h>
 #else
@@ -54,7 +54,7 @@
 # include <netdb.h>
 # include <sys/utsname.h>
 #endif /* WIN32 */
-#endif /* macintosh */
+// #endif /* macintosh */
 #ifdef HAVE_UNISTD_H
 #include <unistd.h>
 #endif
