--- qtwebkit-5.212.0-alpha2/Source/WTF/wtf/spi/darwin/orig.XPCSPI.h	2019-01-02 18:54:41.000000000 +0100
+++ qtwebkit-5.212.0-alpha2/Source/WTF/wtf/spi/darwin/XPCSPI.h	2019-01-02 18:57:55.000000000 +0100
@@ -90,8 +90,9 @@
 
 EXTERN_C xpc_object_t xpc_array_create(const xpc_object_t*, size_t count);
 #if COMPILER_SUPPORTS(BLOCKS)
-EXTERN_C bool xpc_array_apply(xpc_object_t, xpc_array_applier_t);
-EXTERN_C bool xpc_dictionary_apply(xpc_object_t xdict, xpc_dictionary_applier_t applier);
+// these are defined in xpc/xpc.h on 10.7 and newer.
+// EXTERN_C bool xpc_array_apply(xpc_object_t, xpc_array_applier_t);
+// EXTERN_C bool xpc_dictionary_apply(xpc_object_t xdict, xpc_dictionary_applier_t applier);
 #endif
 EXTERN_C size_t xpc_array_get_count(xpc_object_t);
 EXTERN_C const char* xpc_array_get_string(xpc_object_t, size_t index);
