diff --git a/Source/WebKit/CMakeLists.txt b/Source/WebKit/CMakeLists.txt
index c845856..4b0d4ec 100644
--- a/Source/WebKit/CMakeLists.txt
+++ b/Source/WebKit/CMakeLists.txt
@@ -99,3 +99,7 @@ add_dependencies(WebKit WebCore)
 if (TARGET WebKitGUID)
     add_dependencies(WebKit WebKitGUID)
 endif ()
+
+if (ENABLE_NETSCAPE_PLUGIN_API AND APPLE)
+    target_link_libraries(WebKit PRIVATE "-framework CoreGraphics -framework AppKit")
+endif ()
diff --git a/Source/WebKit/PlatformQt.cmake b/Source/WebKit/PlatformQt.cmake
index 9c44d43..20dc29d 100644
--- a/Source/WebKit/PlatformQt.cmake
+++ b/Source/WebKit/PlatformQt.cmake
@@ -320,6 +320,13 @@ if (ENABLE_NETSCAPE_PLUGIN_API)
             qt/Plugins/PluginViewQt.cpp
         )
     endif ()
+    if (APPLE)
+        list(APPEND WebKit_SOURCES
+            mac/Plugins/PluginPackageMac.cpp
+            mac/Plugins/PluginViewMac.mm
+            mac/Plugins/WebCoreNSStringExtras.mm
+        )
+    endif ()
 
     if (WIN32)
         list(APPEND WebKit_INCLUDE_DIRECTORIES
diff --git a/Source/WebKit/win/Plugins/PluginDatabase.cpp b/Source/WebKit/win/Plugins/PluginDatabase.cpp
index eefcb30..1304ddf 100644
--- a/Source/WebKit/win/Plugins/PluginDatabase.cpp
+++ b/Source/WebKit/win/Plugins/PluginDatabase.cpp
@@ -26,6 +26,7 @@
 
 #include "config.h"
 #include "PluginDatabase.h"
+#include "npapi.h"
 
 #include "Frame.h"
 #include "URL.h"
