--- qtwebkit-5.212.0-alpha2/Source/cmake/orig.OptionsQt.cmake	2017-06-16 01:36:48.000000000 +0200
+++ qtwebkit-5.212.0-alpha2/Source/cmake/OptionsQt.cmake	2017-11-12 02:37:35.000000000 +0100
@@ -284,9 +284,9 @@
     message(FATAL_ERROR "MSVC 2015 is required to build QtWebKit, use a newer MSVC version")
 endif ()
 
-if (APPLE AND CMAKE_SYSTEM_VERSION VERSION_LESS 14.0.0)
-    message(FATAL_ERROR "macOS 10.10 or higher is required to build and run QtWebKit")
-endif ()
+# if (APPLE AND CMAKE_SYSTEM_VERSION VERSION_LESS 14.0.0)
+#     message(FATAL_ERROR "macOS 10.10 or higher is required to build and run QtWebKit")
+# endif ()
 
 set(ENABLE_WEBKIT ON)
 set(WTF_USE_UDIS86 1)
@@ -553,7 +552,7 @@
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-keep-inline-dllexport")
 endif ()
 
-if (APPLE)
+if (APPLE AND CMAKE_SYSTEM_VERSION VERSION_GREATER_EQUAL 14.0.0)
     SET_AND_EXPOSE_TO_BUILD(HAVE_QOS_CLASSES 1)
 endif ()
 
--- qtwebkit-5.212.0-alpha2/Source/WTF/wtf/orig.Platform.h	2017-06-04 22:16:06.000000000 +0200
+++ qtwebkit-5.212.0-alpha2/Source/WTF/wtf/Platform.h	2017-11-12 02:15:36.000000000 +0100
@@ -1054,6 +1054,8 @@
 #define HAVE_QOS_CLASSES 1
 #endif
 #endif
+// not on 10.9
+#undef HAVE_QOS_CLASSES
 
 #if PLATFORM(QT)
 #ifdef __cplusplus
