--- qtwebkit-5.212.0-alpha2/Source/WebKit/orig.PlatformQt.cmake	2017-06-09 16:11:36.000000000 +0200
+++ qtwebkit-5.212.0-alpha2/Source/WebKit/PlatformQt.cmake	2017-11-12 03:05:58.000000000 +0100
@@ -476,7 +476,7 @@
     endforeach ()
 endif ()
 
-if (NOT MACOS_BUILD_FRAMEWORKS)
+# if (NOT MACOS_BUILD_FRAMEWORKS)
     ecm_generate_pkgconfig_file(
         BASE_NAME Qt5WebKit
         DEPS "${WEBKIT_PKGCONGIG_DEPS}"
@@ -484,7 +484,7 @@
     )
     set(ECM_PKGCONFIG_INSTALL_DIR "${LIB_INSTALL_DIR}/pkgconfig" CACHE PATH "The directory where pkgconfig will be installed to.")
     install(FILES ${WebKit_PKGCONFIG_FILENAME} DESTINATION ${ECM_PKGCONFIG_INSTALL_DIR} COMPONENT Data)
-endif ()
+# endif ()
 
 if (KDE_INSTALL_USE_QT_SYS_PATHS)
     set(WebKit_PRI_ARGUMENTS
@@ -666,14 +666,14 @@
     COMPONENT Data
 )
 
-if (NOT MACOS_BUILD_FRAMEWORKS)
+# if (NOT MACOS_BUILD_FRAMEWORKS)
     ecm_generate_pkgconfig_file(
         BASE_NAME Qt5WebKitWidgets
         DEPS "${WEBKITWIDGETS_PKGCONFIG_DEPS}"
         FILENAME_VAR WebKitWidgets_PKGCONFIG_FILENAME
     )
     install(FILES ${WebKitWidgets_PKGCONFIG_FILENAME} DESTINATION ${ECM_PKGCONFIG_INSTALL_DIR} COMPONENT Data)
-endif ()
+# endif ()
 
 if (KDE_INSTALL_USE_QT_SYS_PATHS)
     set(WebKitWidgets_PRI_ARGUMENTS
