diff --git a/Source/cmake/OptionsQt.cmake b/Source/cmake/OptionsQt.cmake
index 1ee60b777106d8f66b1f938e78db870c526ad246..f11698972428a39a077c3b0fc7600d093173a884 100644
--- a/Source/cmake/OptionsQt.cmake
+++ b/Source/cmake/OptionsQt.cmake
@@ -1,6 +1,6 @@
 include(CheckCXXSourceCompiles)
 include(FeatureSummary)
-include(ECMEnableSanitizers)
+# include(ECMEnableSanitizers)
 include(ECMPackageConfigHelpers)
 
 set(ECM_MODULE_DIR ${CMAKE_MODULE_PATH})
