diff --git a/Source/WebKit/PlatformQt.cmake b/Source/WebKit/PlatformQt.cmake
index 9c44d43..a0d5768 100644
--- a/Source/WebKit/PlatformQt.cmake
+++ b/Source/WebKit/PlatformQt.cmake
@@ -499,7 +499,7 @@ if (QT_STATIC_BUILD)
     endforeach ()
 endif ()
 
-if (NOT MACOS_BUILD_FRAMEWORKS)
+# if (NOT MACOS_BUILD_FRAMEWORKS)
     ecm_generate_pkgconfig_file(
         BASE_NAME Qt5WebKit
         DESCRIPTION "Qt WebKit module"
@@ -509,7 +509,7 @@ if (NOT MACOS_BUILD_FRAMEWORKS)
     )
     set(ECM_PKGCONFIG_INSTALL_DIR "${LIB_INSTALL_DIR}/pkgconfig" CACHE PATH "The directory where pkgconfig will be installed to.")
     install(FILES ${WebKit_PKGCONFIG_FILENAME} DESTINATION ${ECM_PKGCONFIG_INSTALL_DIR} COMPONENT Data)
-endif ()
+# endif ()
 
 if (KDE_INSTALL_USE_QT_SYS_PATHS)
     set(WebKit_PRI_ARGUMENTS
@@ -725,7 +725,7 @@ install(
     COMPONENT Data
 )
 
-if (NOT MACOS_BUILD_FRAMEWORKS)
+# if (NOT MACOS_BUILD_FRAMEWORKS)
     ecm_generate_pkgconfig_file(
         BASE_NAME Qt5WebKitWidgets
         DESCRIPTION "Qt WebKitWidgets module"
@@ -734,7 +734,7 @@ if (NOT MACOS_BUILD_FRAMEWORKS)
         FILENAME_VAR WebKitWidgets_PKGCONFIG_FILENAME
     )
     install(FILES ${WebKitWidgets_PKGCONFIG_FILENAME} DESTINATION ${ECM_PKGCONFIG_INSTALL_DIR} COMPONENT Data)
-endif ()
+# endif ()
 
 if (KDE_INSTALL_USE_QT_SYS_PATHS)
     set(WebKitWidgets_PRI_ARGUMENTS
