--- qtwebkit-5.212.0-alpha2/Source/WebKit/orig.PlatformQt.cmake	2017-06-09 16:11:36.000000000 +0200
+++ qtwebkit-5.212.0-alpha2/Source/WebKit/PlatformQt.cmake	2017-11-12 03:14:45.000000000 +0100
@@ -753,7 +753,7 @@
 
 WEBKIT_FRAMEWORK(WebKitWidgets)
 add_dependencies(WebKitWidgets WebKit)
-set_target_properties(WebKitWidgets PROPERTIES VERSION ${PROJECT_VERSION} SOVERSION ${PROJECT_VERSION_MAJOR})
+set_target_properties(WebKitWidgets PROPERTIES VERSION ${PROJECT_VERSION} SOVERSION ${COMPATIBILITY_VERSION})
 install(TARGETS WebKitWidgets EXPORT Qt5WebKitWidgetsTargets
         DESTINATION "${LIB_INSTALL_DIR}"
         RUNTIME DESTINATION "${BIN_INSTALL_DIR}"
--- qtwebkit-5.212.0-alpha2/Source/WebKit/orig.CMakeLists.txt	2017-06-13 10:26:18.000000000 +0200
+++ qtwebkit-5.212.0-alpha2/Source/WebKit/CMakeLists.txt	2017-11-12 03:14:26.000000000 +0100
@@ -55,7 +55,7 @@
 WEBKIT_FRAMEWORK(WebKit)
 
 if (${PORT} STREQUAL "Qt")
-    set_target_properties(WebKit PROPERTIES VERSION ${PROJECT_VERSION} SOVERSION ${PROJECT_VERSION_MAJOR})
+    set_target_properties(WebKit PROPERTIES VERSION ${PROJECT_VERSION} SOVERSION ${COMPATIBILITY_VERSION})
     install(TARGETS WebKit EXPORT WebKitTargets
             DESTINATION "${LIB_INSTALL_DIR}"
             RUNTIME DESTINATION "${BIN_INSTALL_DIR}"
