diff --git a/configure b/configure
index dc5e79e..3f987cf 100755
--- a/configure
+++ b/configure
@@ -23424,21 +23424,21 @@ case "${host_os}" in
       if [ "$use_quicktime" = "auto" ] ; then
          use_quicktime="yes"
       fi
-      ;;
-
-   cygwin*)
-            LIBS="$LIBS -lkernel32 -luser32 -lgdi32 -lwinspool -lcomdlg32 -ladvapi32 -lshell32 -lole32 -loleaut32 -luuid -lodbc32 -lodbccp32 -lwsock32 -lwinmm"
-      CONFIGHEADER="configwin.h"
-
-
-$as_echo "#define __CYGWIN__ 1" >>confdefs.h
-
-
-$as_echo "#define _FILE_OFFSET_BITS 32" >>confdefs.h
-
-      ;;
-
-   *)
+#      ;;
+#
+#   cygwin*)
+#            LIBS="$LIBS -lkernel32 -luser32 -lgdi32 -lwinspool -lcomdlg32 -ladvapi32 -lshell32 -lole32 -loleaut32 -luuid -lodbc32 -lodbccp32 -lwsock32 -lwinmm"
+#      CONFIGHEADER="configwin.h"
+#
+#
+#$as_echo "#define __CYGWIN__ 1" >>confdefs.h
+#
+#
+#$as_echo "#define _FILE_OFFSET_BITS 32" >>confdefs.h
+#
+#      ;;
+#
+#   *)
             CONFIGHEADER="configunix.h"
 
 
diff --git a/configure.ac b/configure.ac
index 4ca3210..22bf6e8 100644
--- a/configure.ac
+++ b/configure.ac
@@ -590,19 +590,19 @@ case "${host_os}" in
       if [[ "$use_quicktime" = "auto" ]] ; then
          use_quicktime="yes"
       fi
-      ;;
-
-   cygwin*)
-      dnl Windows/CygWin configuration
-      LIBS="$LIBS -lkernel32 -luser32 -lgdi32 -lwinspool -lcomdlg32 -ladvapi32 -lshell32 -lole32 -loleaut32 -luuid -lodbc32 -lodbccp32 -lwsock32 -lwinmm"
-      CONFIGHEADER="configwin.h"
-
-      AC_DEFINE(__CYGWIN__,1,[We're using cygwin])
-      dnl ' (end endless string in vim)
-      AC_DEFINE(_FILE_OFFSET_BITS,32,[Placeholder for large file support])
-      ;;
-
-   *)
+#      ;;
+#
+#   cygwin*)
+#      dnl Windows/CygWin configuration
+#      LIBS="$LIBS -lkernel32 -luser32 -lgdi32 -lwinspool -lcomdlg32 -ladvapi32 -lshell32 -lole32 -loleaut32 -luuid -lodbc32 -lodbccp32 -lwsock32 -lwinmm"
+#      CONFIGHEADER="configwin.h"
+#
+#      AC_DEFINE(__CYGWIN__,1,[We're using cygwin])
+#      dnl ' (end endless string in vim)
+#      AC_DEFINE(_FILE_OFFSET_BITS,32,[Placeholder for large file support])
+#      ;;
+#
+#   *)
       dnl Unix configuration
       CONFIGHEADER="configunix.h"
 
