--- a/src/FileIO.cpp
+++ b/src/FileIO.cpp
@@ -85,10 +85,10 @@
    wxFileName fn(mName);
    wxString ext = fn.GetExt().MakeUpper() + wxT("    ");
 
-   type = (ext[0] & 0xff) << 24 |
-          (ext[1] & 0xff) << 16 |
-          (ext[2] & 0xff) << 8  |
-          (ext[3] & 0xff);
+   type = (int(ext[0]) & 0xff) << 24 |
+          (int(ext[1]) & 0xff) << 16 |
+          (int(ext[2]) & 0xff) << 8  |
+          (int(ext[3]) & 0xff);
 
    SetCatalogInfo(type);
 #endif
