diff --git a/lib-src/FileDialog/mac/FileDialogPrivate.h b/lib-src/FileDialog/mac/FileDialogPrivate.h
index 36caf694ecb348d9e31fcd50bf19b9e591e2ace8..e13b260e1f38f9cbf6620bf0d38be21000081418 100644
--- a/lib-src/FileDialog/mac/FileDialogPrivate.h
+++ b/lib-src/FileDialog/mac/FileDialogPrivate.h
@@ -97,8 +97,10 @@ protected:
     int m_firstFileTypeFilter;
     wxArrayString m_currentExtensions;
     WX_NSObject m_delegate;
+#if !wxCHECK_VERSION(3, 0, 3)
     WX_NSObject m_sheetDelegate;
 #endif
+#endif
 
 private:
     // Common part of all ctors.
diff --git a/lib-src/FileDialog/mac/FileDialogPrivate.mm b/lib-src/FileDialog/mac/FileDialogPrivate.mm
index ca599098502bd9d622d762c91e18d627add0b809..7736c6c2abe3418416e7d1ccfc827787abf4dde9 100644
--- a/lib-src/FileDialog/mac/FileDialogPrivate.mm
+++ b/lib-src/FileDialog/mac/FileDialogPrivate.mm
@@ -147,7 +147,9 @@ void FileDialog::Init()
 {
     m_filterIndex = -1;
     m_delegate = nil;
+#if !wxCHECK_VERSION(3, 0, 3)
     m_sheetDelegate = nil;
+#endif
     m_filterPanel = NULL;
     m_filterChoice = NULL;
 }
@@ -159,13 +161,17 @@ void FileDialog::Create(
 {
     FileDialogBase::Create(parent, message, defaultDir, defaultFileName, wildCard, style, pos, sz, name);
 
+// #if !wxCHECK_VERSION(3, 0, 3)
 //    m_sheetDelegate = [[ModalDialogDelegate alloc] init];
 //    [(ModalDialogDelegate*)m_sheetDelegate setImplementation: this];
+// #endif
 }
 
 FileDialog::~FileDialog()
 {
+#if !wxCHECK_VERSION(3, 0, 3)
     [m_sheetDelegate release];
+#endif
 }
 
 bool FileDialog::SupportsExtraControl() const
