diff --git a/src/Makefile.am b/src/Makefile.am
index aab9208..6e7cf1a 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -258,7 +259,7 @@ audacity_SOURCES = \
 	commands/CommandDirectory.h \
 	commands/CommandHandler.cpp \
 	commands/CommandHandler.h \
-	commands/CommandManager.cpp \
+	commands/CommandManager.mm \
 	commands/CommandManager.h \
 	commands/CommandMisc.h \
 	commands/CommandSignature.cpp \
diff --git a/src/Makefile.in b/src/Makefile.in
index e5becb0..5289023 100644
--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -326,7 +326,7 @@ am__audacity_SOURCES_DIST = BlockFile.cpp BlockFile.h DirManager.cpp \
 	commands/Command.h commands/CommandBuilder.cpp \
 	commands/CommandBuilder.h commands/CommandDirectory.cpp \
 	commands/CommandDirectory.h commands/CommandHandler.cpp \
-	commands/CommandHandler.h commands/CommandManager.cpp \
+	commands/CommandHandler.h commands/CommandManager.mm \
 	commands/CommandManager.h commands/CommandMisc.h \
 	commands/CommandSignature.cpp commands/CommandSignature.h \
 	commands/CommandTargets.h commands/CommandType.cpp \
@@ -1196,7 +1197,7 @@ audacity_SOURCES = $(libaudacity_la_SOURCES) AboutDialog.cpp \
 	commands/Command.h commands/CommandBuilder.cpp \
 	commands/CommandBuilder.h commands/CommandDirectory.cpp \
 	commands/CommandDirectory.h commands/CommandHandler.cpp \
-	commands/CommandHandler.h commands/CommandManager.cpp \
+	commands/CommandHandler.h commands/CommandManager.mm \
 	commands/CommandManager.h commands/CommandMisc.h \
 	commands/CommandSignature.cpp commands/CommandSignature.h \
 	commands/CommandTargets.h commands/CommandType.cpp \
@@ -3699,19 +3700,19 @@ commands/audacity-CommandHandler.obj: commands/CommandHandler.cpp
 @AMDEP_TRUE@@am__fastdepCXX_FALSE@	DEPDIR=$(DEPDIR) $(CXXDEPMODE) $(depcomp) @AMDEPBACKSLASH@
 @am__fastdepCXX_FALSE@	$(AM_V_CXX@am__nodep@)$(CXX) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(audacity_CPPFLAGS) $(CPPFLAGS) $(audacity_CXXFLAGS) $(CXXFLAGS) -c -o commands/audacity-CommandHandler.obj `if test -f 'commands/CommandHandler.cpp'; then $(CYGPATH_W) 'commands/CommandHandler.cpp'; else $(CYGPATH_W) '$(srcdir)/commands/CommandHandler.cpp'; fi`
 
-commands/audacity-CommandManager.o: commands/CommandManager.cpp
-@am__fastdepCXX_TRUE@	$(AM_V_CXX)$(CXX) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(audacity_CPPFLAGS) $(CPPFLAGS) $(audacity_CXXFLAGS) $(CXXFLAGS) -MT commands/audacity-CommandManager.o -MD -MP -MF commands/$(DEPDIR)/audacity-CommandManager.Tpo -c -o commands/audacity-CommandManager.o `test -f 'commands/CommandManager.cpp' || echo '$(srcdir)/'`commands/CommandManager.cpp
+commands/audacity-CommandManager.o: commands/CommandManager.mm
+@am__fastdepCXX_TRUE@	$(AM_V_CXX)$(CXX) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(audacity_CPPFLAGS) $(CPPFLAGS) $(audacity_CXXFLAGS) $(CXXFLAGS) -MT commands/audacity-CommandManager.o -MD -MP -MF commands/$(DEPDIR)/audacity-CommandManager.Tpo -c -o commands/audacity-CommandManager.o `test -f 'commands/CommandManager.mm' || echo '$(srcdir)/'`commands/CommandManager.mm
 @am__fastdepCXX_TRUE@	$(AM_V_at)$(am__mv) commands/$(DEPDIR)/audacity-CommandManager.Tpo commands/$(DEPDIR)/audacity-CommandManager.Po
-@AMDEP_TRUE@@am__fastdepCXX_FALSE@	$(AM_V_CXX)source='commands/CommandManager.cpp' object='commands/audacity-CommandManager.o' libtool=no @AMDEPBACKSLASH@
+@AMDEP_TRUE@@am__fastdepCXX_FALSE@	$(AM_V_CXX)source='commands/CommandManager.mm' object='commands/audacity-CommandManager.o' libtool=no @AMDEPBACKSLASH@
 @AMDEP_TRUE@@am__fastdepCXX_FALSE@	DEPDIR=$(DEPDIR) $(CXXDEPMODE) $(depcomp) @AMDEPBACKSLASH@
-@am__fastdepCXX_FALSE@	$(AM_V_CXX@am__nodep@)$(CXX) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(audacity_CPPFLAGS) $(CPPFLAGS) $(audacity_CXXFLAGS) $(CXXFLAGS) -c -o commands/audacity-CommandManager.o `test -f 'commands/CommandManager.cpp' || echo '$(srcdir)/'`commands/CommandManager.cpp
+@am__fastdepCXX_FALSE@	$(AM_V_CXX@am__nodep@)$(CXX) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(audacity_CPPFLAGS) $(CPPFLAGS) $(audacity_CXXFLAGS) $(CXXFLAGS) -c -o commands/audacity-CommandManager.o `test -f 'commands/CommandManager.mm' || echo '$(srcdir)/'`commands/CommandManager.mm
 
-commands/audacity-CommandManager.obj: commands/CommandManager.cpp
-@am__fastdepCXX_TRUE@	$(AM_V_CXX)$(CXX) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(audacity_CPPFLAGS) $(CPPFLAGS) $(audacity_CXXFLAGS) $(CXXFLAGS) -MT commands/audacity-CommandManager.obj -MD -MP -MF commands/$(DEPDIR)/audacity-CommandManager.Tpo -c -o commands/audacity-CommandManager.obj `if test -f 'commands/CommandManager.cpp'; then $(CYGPATH_W) 'commands/CommandManager.cpp'; else $(CYGPATH_W) '$(srcdir)/commands/CommandManager.cpp'; fi`
+commands/audacity-CommandManager.obj: commands/CommandManager.mm
+@am__fastdepCXX_TRUE@	$(AM_V_CXX)$(CXX) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(audacity_CPPFLAGS) $(CPPFLAGS) $(audacity_CXXFLAGS) $(CXXFLAGS) -MT commands/audacity-CommandManager.obj -MD -MP -MF commands/$(DEPDIR)/audacity-CommandManager.Tpo -c -o commands/audacity-CommandManager.obj `if test -f 'commands/CommandManager.mm'; then $(CYGPATH_W) 'commands/CommandManager.mm'; else $(CYGPATH_W) '$(srcdir)/commands/CommandManager.mm'; fi`
 @am__fastdepCXX_TRUE@	$(AM_V_at)$(am__mv) commands/$(DEPDIR)/audacity-CommandManager.Tpo commands/$(DEPDIR)/audacity-CommandManager.Po
-@AMDEP_TRUE@@am__fastdepCXX_FALSE@	$(AM_V_CXX)source='commands/CommandManager.cpp' object='commands/audacity-CommandManager.obj' libtool=no @AMDEPBACKSLASH@
+@AMDEP_TRUE@@am__fastdepCXX_FALSE@	$(AM_V_CXX)source='commands/CommandManager.mm' object='commands/audacity-CommandManager.obj' libtool=no @AMDEPBACKSLASH@
 @AMDEP_TRUE@@am__fastdepCXX_FALSE@	DEPDIR=$(DEPDIR) $(CXXDEPMODE) $(depcomp) @AMDEPBACKSLASH@
-@am__fastdepCXX_FALSE@	$(AM_V_CXX@am__nodep@)$(CXX) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(audacity_CPPFLAGS) $(CPPFLAGS) $(audacity_CXXFLAGS) $(CXXFLAGS) -c -o commands/audacity-CommandManager.obj `if test -f 'commands/CommandManager.cpp'; then $(CYGPATH_W) 'commands/CommandManager.cpp'; else $(CYGPATH_W) '$(srcdir)/commands/CommandManager.cpp'; fi`
+@am__fastdepCXX_FALSE@	$(AM_V_CXX@am__nodep@)$(CXX) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(audacity_CPPFLAGS) $(CPPFLAGS) $(audacity_CXXFLAGS) $(CXXFLAGS) -c -o commands/audacity-CommandManager.obj `if test -f 'commands/CommandManager.mm'; then $(CYGPATH_W) 'commands/CommandManager.mm'; else $(CYGPATH_W) '$(srcdir)/commands/CommandManager.mm'; fi`
 
 commands/audacity-CommandSignature.o: commands/CommandSignature.cpp
 @am__fastdepCXX_TRUE@	$(AM_V_CXX)$(CXX) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(audacity_CPPFLAGS) $(CPPFLAGS) $(audacity_CXXFLAGS) $(CXXFLAGS) -MT commands/audacity-CommandSignature.o -MD -MP -MF commands/$(DEPDIR)/audacity-CommandSignature.Tpo -c -o commands/audacity-CommandSignature.o `test -f 'commands/CommandSignature.cpp' || echo '$(srcdir)/'`commands/CommandSignature.cpp
