--- audacity-audacity-660b8c0/src/prefs/orig.PrefsDialog.cpp	2019-06-21 10:29:32.000000000 +0200
+++ audacity-audacity-660b8c0/src/prefs/PrefsDialog.cpp	2019-12-04 12:52:18.720278628 +0100
@@ -19,6 +19,7 @@
 
 #include "../Experimental.h"
 
+#include <wx/app.h>
 #include <wx/setup.h> // for wxUSE_* macros
 #include <wx/defs.h>
 #include <wx/button.h>
@@ -33,6 +34,7 @@
 #include <wx/listbook.h>
 
 #include <wx/treebook.h>
+#include <wx/treectrl.h>
 
 #include "../AudioIOBase.h"
 #include "../Prefs.h"
