diff --git a/src/effects/VST/VSTControlOSX.mm b/src/effects/VST/VSTControlOSX.mm
index d5c2b53366b85e773dbbdef3d6e2bb7aca92d0a6..98e5379984a37096c6ffd216133ab4db9e042222 100644
--- a/src/effects/VST/VSTControlOSX.mm
+++ b/src/effects/VST/VSTControlOSX.mm
@@ -45,7 +45,7 @@
 @end
 
 VSTControlImpl::VSTControlImpl(wxWindowMac *peer, NSView *view)
-:  wxWidgetCocoaImpl(peer, view, false, false)
+:  wxWidgetCocoaImpl(peer, view)
 {
 }
 
diff --git a/src/effects/audiounits/AUControl.mm b/src/effects/audiounits/AUControl.mm
index fdb6ae7ef1fa1838a2d28eb61150198eda6a1435..07d45f89e722abb4faec9c731bc10230a00df498 100644
--- a/src/effects/audiounits/AUControl.mm
+++ b/src/effects/audiounits/AUControl.mm
@@ -77,7 +77,7 @@
 @end
 
 AUControlImpl::AUControlImpl(wxWindowMac *peer, NSView *view)
-:  wxWidgetCocoaImpl(peer, view, false, false)
+:  wxWidgetCocoaImpl(peer, view)
 {
 }
 
