diff --git a/Makefile b/Makefile
index 4be0cd6..4e4e036 100644
--- a/Makefile
+++ b/Makefile
@@ -46,7 +46,7 @@ BUILDSYSLIBFLAGS = -dynamiclib -headerpad_max_install_names \
                    -current_version $(SOCURVERSION) \
                    -compatibility_version $(SOCOMPATVERSION)
 SYSREEXPORTFLAG  = -Wl,-reexport_library,/usr/lib/libSystem.B.dylib
-BUILDSLIBFLAGS   = -qs
+BUILDSLIBFLAGS   = qs
 POSTINSTALL      = install_name_tool
 
 # The defaults for C[XX]FLAGS are defined as XC[XX]FLAGS, so that supplied
