diff --git a/Makefile b/Makefile
index 813fc0e..1b18a12 100644
--- a/Makefile
+++ b/Makefile
@@ -53,7 +53,7 @@ OSLIBLINK        = System
 XLIBDIR          = xlib
 XLIBPATH         = $(XLIBDIR)/lib$(OSLIBLINK)$(SOEXT)
 SYSREEXPORTFLAG  = -Wl,-reexport_library,$(OSLIBDIR)/lib$(OSLIBNAME)$(SOEXT)
-BUILDSLIBFLAGS   = -qs
+BUILDSLIBFLAGS   = qs
 POSTINSTALL     ?= /usr/bin/install_name_tool
 
 # The defaults for C[XX]FLAGS are defined as XC[XX]FLAGS, so that supplied
