diff --git b/orig.Makefile a/Makefile
index 952e9c3..611a3e7 100644
--- b/orig.Makefile
+++ a/Makefile
@@ -45,7 +45,7 @@ BUILDSYSLIBFLAGS = -dynamiclib -headerpad_max_install_names \
                    -current_version $(SOCURVERSION) \
                    -compatibility_version $(SOCOMPATVERSION)
 SYSREEXPORTFLAG  = -Wl,-reexport_library /usr/lib/libSystem.B.dylib
-BUILDSLIBFLAGS   = -qs
+BUILDSLIBFLAGS   = qs
 POSTINSTALL      = install_name_tool
 
 MAX_DARWIN_REEXPORT ?= 19
