diff --git a/Makefile b/Makefile
index d26e79e..84e8219 100644
--- a/Makefile
+++ b/Makefile
@@ -50,7 +50,7 @@ OSLIBLINK        = System
 XLIBDIR          = xlib
 XLIBPATH         = $(XLIBDIR)/lib$(OSLIBLINK)$(SOEXT)
 SYSREEXPORTFLAG  = -Wl,-reexport_library,$(OSLIBDIR)/lib$(OSLIBNAME)$(SOEXT)
-BUILDSLIBFLAGS   = -qs
+BUILDSLIBFLAGS   = qs
 POSTINSTALL      = install_name_tool
 
 # The defaults for C[XX]FLAGS are defined as XC[XX]FLAGS, so that supplied
