diff --git a/src/clonefile.c b/src/clonefile.c
index ef03f02..8a30e4b 100644
--- a/src/clonefile.c
+++ b/src/clonefile.c
@@ -1,61 +1,71 @@
-/*
- * Copyright (c) 2025 Frederick H. G. Wright II <fw@fwright.net>
- *
- * Permission to use, copy, modify, and distribute this software for any
- * purpose with or without fee is hereby granted, provided that the above
- * copyright notice and this permission notice appear in all copies.
- *
- * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
- * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
- * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
- * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
- * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
- * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
- * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
- */
-
 /* MP support header */
 #include "MacportsLegacySupport.h"
 
 #if __MPLS_LIB_SUPPORT_CLONEFILE__
+#include <stdlib.h>
+#include <syslog.h>
+#include <sys/fcntl.h>
+#include <errno.h>
 
-/*
- * This provides degenerate implementations of *clonefile*() which always
- * fail, since the real versions only work on APFS, and no OS version that
- * uses this code supports APFS.
- *
- * At present, these functions always fail immediately with ENOTSUP, though
- * real implementations might fail in other ways when given bad arguments.
- */
+// runtime shims for the clonefile(2) function family, on systems < 10.12
+//
+// These do relevant minimal argument checking, and then return ENOTSUP
+// to indicate that the underlying filesystem doesn't support the call.
+// (Only APFS does, and APFS can only be used from 10.12 onwards.)
 
-#include <errno.h>
+/* Options for clonefile calls */
+#define CLONE_NOFOLLOW      0x0001     /* Don't follow symbolic links */
+#define CLONE_NOOWNERCOPY   0x0002     /* Don't copy ownership information from source */
 
-#include <sys/clonefile.h>
-#include <sys/fcntl.h>
+// static int debug = -1;
+// 
+// static int check_debug()
+// {
+// 	if (debug < 0) {
+// 		debug = getenv("CLONEFILE_DEBUG")? 1 : 0;
+// 	}
+// 	return debug;
+// }
 
-int
-clonefile(const char *src, const char *dst, uint32_t flags)
+static int implementation(int flags)
 {
-  return clonefileat(AT_FDCWD, src, AT_FDCWD, dst, flags);
+	// Check that the flags are valid.
+	// Upstream does this first too.
+	if (flags & ~(CLONE_NOFOLLOW | CLONE_NOOWNERCOPY)) {
+		errno = EINVAL;
+	} else {
+		// flags are valid but we're on an OS that doesn't support APFS
+		// so also cannot support clonefile & family. Set ENOTSUP .
+		errno = ENOTSUP;
+	}
+	return -1;
 }
 
-int
-clonefileat(int src_dirfd, const char *src,
-            int dst_dirfd, const char *dst, uint32_t flags)
+int clonefileat(int src_dirfd, const char * src, int dst_dirfd, const char * dst, int flags)
 {
-  (void) src_dirfd; (void) src; (void) dst_dirfd; (void) dst; (void) flags;
-
-  errno = ENOTSUP;
-  return -1;
+// 	if (check_debug()) {
+// 		syslog(LOG_NOTICE, "%s:%d::%s:\n\tsrc_dirfd=%d src=\"%s\" dst_dirfd=%d dst=\"%s\" flags=%x",
+// 			__FILE__, __LINE__, __FUNCTION__, src_dirfd, src, dst_dirfd, dst, flags);
+// 	}
+	return implementation(flags);
 }
 
-int
-fclonefileat(int srcfd, int dst_dirfd, const char *dst, uint32_t flags)
+int fclonefileat(int srcfd, int dst_dirfd, const char * dst, int flags)
 {
-  (void) srcfd; (void) dst_dirfd; (void) dst; (void) flags;
+// 	if (check_debug()) {
+// 		syslog(LOG_NOTICE, "%s:%d::%s:\n\tsrcfd=%d dst_dirfd=%d dst=\"%s\" flags=%x",
+// 			__FILE__, __LINE__, __FUNCTION__, srcfd, dst_dirfd, dst, flags);
+// 	}
+	return implementation(flags);
+}
 
-  errno = ENOTSUP;
-  return -1;
+int clonefile(const char * src, const char * dst, int flags)
+{
+// 	if (check_debug()) {
+// 		syslog(LOG_NOTICE, "%s:%d::%s:\n\tsrc=\"%s\" dst=\"%s\" flags=%x",
+// 			__FILE__, __LINE__, __FUNCTION__, src, dst, flags);
+// 	}
+	return implementation(flags);
 }
 
-#endif /* __MPLS_LIB_SUPPORT_CLONEFILE__ */
+#endif // __MPLS_LIB_SUPPORT_CLONEFILE__
