diff --git a/qtbase/src/corelib/kernel/qcoreapplication.cpp b/qtbase/src/corelib/kernel/qcoreapplication.cpp
index 2179a0d..ab03be2 100644
--- a/qtbase/src/corelib/kernel/qcoreapplication.cpp
+++ b/qtbase/src/corelib/kernel/qcoreapplication.cpp
@@ -325,9 +325,16 @@ Qt::HANDLE qt_application_thread_id = 0;
 #endif // QT_NO_QOBJECT
 
 QCoreApplication *QCoreApplication::self = 0;
+#ifdef Q_OS_MACOS
+uint QCoreApplicationPrivate::attribs =
+    (1 << Qt::AA_SynthesizeMouseForUnhandledTouchEvents) |
+    (1 << Qt::AA_SynthesizeMouseForUnhandledTabletEvents) |
+    (1 << Qt::AA_DontShowIconsInMenus);
+#else
 uint QCoreApplicationPrivate::attribs =
     (1 << Qt::AA_SynthesizeMouseForUnhandledTouchEvents) |
     (1 << Qt::AA_SynthesizeMouseForUnhandledTabletEvents);
+#endif
 
 struct QCoreApplicationData {
     QCoreApplicationData() Q_DECL_NOTHROW {
