diff --git a/src/assistant/help/qhelpengine.cpp b/src/assistant/help/qhelpengine.cpp
index 19e44cd915f6d0b6b93a0d971450fc3e1499237e..b78da80d913c2ef5070f1d0acec843bf3b2759a3 100644
--- a/src/assistant/help/qhelpengine.cpp
+++ b/src/assistant/help/qhelpengine.cpp
@@ -37,6 +37,7 @@
 **
 ****************************************************************************/
 
+#include "qhelpenginecore.h"
 #include "qhelpengine.h"
 #include "qhelpengine_p.h"
 #include "qhelpdbreader_p.h"
diff --git a/src/assistant/assistant.pro b/src/assistant/assistant.pro
index 2d3ca56902667ae9ab43c85243a62aa31bc429a6..35518d1ebf018ab4c562fc0c4f0036affb522ff9 100644
--- a/src/assistant/assistant.pro
+++ b/src/assistant/assistant.pro
@@ -1,15 +1,15 @@
 requires(qtHaveModule(sql))
 TEMPLATE = subdirs
-
+CONFIG += ordered
 SUBDIRS += \
-           help \
+           helpng \
            assistant \
            qhelpgenerator
 
-assistant.depends = help
-qhelpgenerator.depends = help
+assistant.depends = helpng
+qhelpgenerator.depends = helpng
 
-qtNomakeTools( \
-    assistant \
-    qhelpgenerator \
-)
+# qtNomakeTools( \
+#     assistant \
+#     qhelpgenerator \
+# )
diff --git a/src/assistant/assistant/assistant.pro b/src/assistant/assistant/assistant.pro
index 080aba03f31370a53d5740970679b0b289d3723f..2c3db23db31d8916032f105225b891fd0baebb38 100644
--- a/src/assistant/assistant/assistant.pro
+++ b/src/assistant/assistant/assistant.pro
@@ -1,12 +1,18 @@
-qtHaveModule(webkitwidgets):!contains(QT_CONFIG, static) {
+# qtHaveModule(webkitwidgets):!contains(QT_CONFIG, static) {
     BROWSER = qtwebkit
+# } else {
+#     BROWSER = qtextbrowser
+# }
+
+QT += widgets network sql
+mac {
+    # unorthodox framework linking...
+    LIBS += ../helpng/QtHelpNG.framework/QtHelpNG
 } else {
-    BROWSER = qtextbrowser
+    LIBS += -L../helpng -lQtHelpNG
 }
-
-QT += widgets network help sql
-qtHaveModule(printsupport): QT += printsupport
-PROJECTNAME = Assistant
+QT += printsupport
+PROJECTNAME = QtAssistant
 
 include(../../shared/fontpanel/fontpanel.pri)
 
@@ -116,8 +122,10 @@ win32 {
 
 mac {
     ICON = assistant.icns
-    TARGET = Assistant
+    TARGET = QtAssistant
     QMAKE_INFO_PLIST = Info_mac.plist
 }
 
-load(qt_app)
+# load(qt_app)
+target.path = $$[QT_INSTALL_BINS]
+INSTALLS += target
diff --git a/src/assistant/assistant/centralwidget.cpp b/src/assistant/assistant/centralwidget.cpp
index 43a8b9a3d9eaabd1c85d62b2786eb450d07595df..928ca80019bc7149fc7457ed911d7148d43dd7b7 100644
--- a/src/assistant/assistant/centralwidget.cpp
+++ b/src/assistant/assistant/centralwidget.cpp
@@ -26,6 +26,9 @@
 **
 ****************************************************************************/
 
+#include "../help/qhelpsearchengine.h"
+#include "../help/qhelpsearchquerywidget.h"
+
 #include "centralwidget.h"
 
 #include "findwidget.h"
diff --git a/src/assistant/assistant/helpenginewrapper.cpp b/src/assistant/assistant/helpenginewrapper.cpp
index 17e4bde8c9639023dc8cc879f4c06955cb588c5b..aa9fe17ff4c9eef49cef9fce5bb5161d033a134c 100644
--- a/src/assistant/assistant/helpenginewrapper.cpp
+++ b/src/assistant/assistant/helpenginewrapper.cpp
@@ -27,6 +27,9 @@
 ****************************************************************************/
 #include "tracer.h"
 
+#include "../help/qhelpsearchengine.h"
+#include "../help/qhelpsearchquerywidget.h"
+
 #include "helpenginewrapper.h"
 #include "../shared/collectionconfiguration.h"
 #include "../help/qhelpengine_p.h"
@@ -38,8 +41,9 @@
 #include <QtCore/QSharedPointer>
 #include <QtCore/QTimer>
 #include <QtHelp/QHelpContentModel>
-#include <QtHelp/QHelpEngine>
-#include <QtHelp/QHelpFilterEngine>
+#include "../help/qhelpenginecore.h"
+#include "../help/qhelpengine.h"
+#include "../help/qhelpfilterengine.h"
 #include <QtHelp/QHelpIndexModel>
 #include <QtHelp/QHelpSearchEngine>
 
diff --git a/src/assistant/assistant/preferencesdialog.cpp b/src/assistant/assistant/preferencesdialog.cpp
index bcc6616904ac7d37ca14c68af88809f16724db39..48e3447fbfbdfe10b030437a367fce0da16b74d1 100644
--- a/src/assistant/assistant/preferencesdialog.cpp
+++ b/src/assistant/assistant/preferencesdialog.cpp
@@ -37,9 +37,9 @@
 #include <QtWidgets/QMessageBox>
 
 #include <QtHelp/QHelpEngineCore>
-#include <QtHelp/QHelpFilterData>
-#include <QtHelp/QHelpFilterEngine>
-#include <QtHelp/QHelpCollectionDetails>
+#include "../help/qhelpfilterdata.h"
+#include "../help/qhelpfilterengine.h"
+#include "../help/qhelpcollectiondetails.h"
 
 #include <QtWidgets/QFileDialog>
 
diff --git a/src/assistant/assistant/remotecontrol.cpp b/src/assistant/assistant/remotecontrol.cpp
index 1c16ac104b20337473f608d794dd7c33937c5e9f..7f5d499c4f88c49cd3e037154a50e37c44e4e9bb 100644
--- a/src/assistant/assistant/remotecontrol.cpp
+++ b/src/assistant/assistant/remotecontrol.cpp
@@ -27,6 +27,9 @@
 ****************************************************************************/
 #include "remotecontrol.h"
 
+#include "../help/qhelpsearchengine.h"
+#include "../help/qhelpsearchquerywidget.h"
+
 #include "centralwidget.h"
 #include "helpenginewrapper.h"
 #include "mainwindow.h"
@@ -42,7 +45,7 @@
 #include <QtWidgets/QApplication>
 
 #include <QtHelp/QHelpEngine>
-#include <QtHelp/QHelpFilterEngine>
+#include "../help/qhelpfilterengine.h"
 #include <QtHelp/QHelpIndexWidget>
 #include <QtHelp/QHelpSearchQueryWidget>
 
diff --git a/src/assistant/assistant/searchwidget.cpp b/src/assistant/assistant/searchwidget.cpp
index 136403d6f8307dda44bb3cd3e452cc056226500f..970ecc8f405c77edee830f0f9d4819dd31279aba 100644
--- a/src/assistant/assistant/searchwidget.cpp
+++ b/src/assistant/assistant/searchwidget.cpp
@@ -27,6 +27,9 @@
 ****************************************************************************/
 #include "tracer.h"
 
+#include "../help/qhelpsearchengine.h"
+#include "../help/qhelpsearchquerywidget.h"
+
 #include "mainwindow.h"
 #include "searchwidget.h"
 
diff --git a/src/assistant/assistant/mainwindow.cpp b/src/assistant/assistant/mainwindow.cpp
index c9c1209501b7bcb2ca13135f6b250fc66b8c2d92..2220ab53ddc6d9319f95f2572b35413e41f03077 100644
--- a/src/assistant/assistant/mainwindow.cpp
+++ b/src/assistant/assistant/mainwindow.cpp
@@ -80,7 +80,7 @@
 #include <QtHelp/QHelpEngineCore>
 #include <QtHelp/QHelpIndexModel>
 #include <QtHelp/QHelpSearchEngine>
-#include <QtHelp/QHelpFilterEngine>
+#include "../help/qhelpfilterengine.h"
 
 #include <cstdlib>
 
diff --git a/src/assistant/help/help.pro b/src/assistant/help/help.pro
index 7ffa008c5cad120c0ed702614f4fb6b90cad5c10..53d1772a34b4d1ed092fd09008353fd29ed18684 100644
--- a/src/assistant/help/help.pro
+++ b/src/assistant/help/help.pro
@@ -1,4 +1,4 @@
-TARGET = QtHelp
+TARGET = QtHelpNG
 
 QT = core-private gui widgets sql
 QT_PRIVATE = network
@@ -39,4 +39,18 @@ HEADERS += qhelpenginecore.h \
     qhelpsearchindexreader_default_p.h \
     qhelpsearchindexreader_p.h
 
-load(qt_module)
+# load(qt_module)
+TEMPLATE = lib
+VERSION = $$[QT_VERSION]
+mac {
+    CONFIG += lib_bundle
+    FRAMEWORK_HEADERS.version = Versions
+    FRAMEWORK_HEADERS.files = $$HEADERS
+    FRAMEWORK_HEADERS.path = Headers
+    QMAKE_BUNDLE_DATA += FRAMEWORK_HEADERS
+}
+# CONFIG += staticlib
+# CONFIG += shared_and_static build_all
+target.path = $$[QT_INSTALL_LIBS]
+QMAKE_SONAME_PREFIX = $$target.path
+INSTALLS += target
diff --git a/src/assistant/help/qhelpsearchresultwidget.cpp b/src/assistant/help/qhelpsearchresultwidget.cpp
index e0ca3cc087eb53072188a7dbfe16bb49f81bace6..dda639b6fb626166e1f79e0ba71b30e40ee9cfd1 100644
--- a/src/assistant/help/qhelpsearchresultwidget.cpp
+++ b/src/assistant/help/qhelpsearchresultwidget.cpp
@@ -37,6 +37,8 @@
 **
 ****************************************************************************/
 
+#include "qhelpsearchengine.h"
+
 #include "qhelpsearchresultwidget.h"
 
 #include <QtCore/QList>
diff --git a/src/assistant/qhelpgenerator/qhelpgenerator.pro b/src/assistant/qhelpgenerator/qhelpgenerator.pro
index dabd87ccce4e54211c52e5c2b12aace451f24bd3..34fabf065aac78e7118d42121d467ab31784e990 100644
--- a/src/assistant/qhelpgenerator/qhelpgenerator.pro
+++ b/src/assistant/qhelpgenerator/qhelpgenerator.pro
@@ -1,4 +1,5 @@
-QT += network help-private
+QT += network sql
+INCLUDEPATH += ../helpng
 
 QTPLUGIN.platforms = qminimal
 
@@ -16,4 +17,14 @@ HEADERS += ../shared/collectionconfiguration.h \
            qhelpdatainterface_p.h
 
 QMAKE_TARGET_DESCRIPTION = "Qt Compressed Help File Generator"
-load(qt_tool)
+# load(qt_tool)
+mac {
+    CONFIG -= app_bundle
+    LIBS += -F../helpng -framework QtHelpNG
+} else {
+    LIBS += -L../helpng -lQtHelpNG
+}
+
+TARGET=qhelpgeneratorng
+target.path = $$[QT_INSTALL_BINS]
+INSTALLS += target
diff --git a/src/assistant/qhelpgenerator/qhelpprojectdata.cpp b/src/assistant/qhelpgenerator/qhelpprojectdata.cpp
index 23419bfd9aa18fd90f45297d5135669d113e3d9a..66abb886da6ffe5449e715c06ec168676ca3be9b 100644
--- a/src/assistant/qhelpgenerator/qhelpprojectdata.cpp
+++ b/src/assistant/qhelpgenerator/qhelpprojectdata.cpp
@@ -86,7 +86,7 @@ void QHelpProjectDataPrivate::skipUnknownToken()
 {
     const QString message = QCoreApplication::translate("QHelpProject",
                   "Skipping unknown token <%1> in file \"%2\".")
-                  .arg(name()).arg(fileName) + QLatin1Char('\n');
+                  .arg(*name().string()).arg(fileName) + QLatin1Char('\n');
     fputs(qPrintable(message), stdout);
 
     skipCurrentElement();
