diff --git a/examples/examples.pro b/examples/examples.pro
index 2003bc51..a6675496 100644
--- a/examples/examples.pro
+++ b/examples/examples.pro
@@ -1,8 +1,8 @@
-TEMPLATE = subdirs
-qtHaveModule(widgets): SUBDIRS += help designer linguist uitools assistant
-
-!qtConfig(process): SUBDIRS -= assistant designer
-
-qtNomakeTools( \
-    assistant \
-)
+# TEMPLATE = subdirs
+# qtHaveModule(widgets): SUBDIRS += help designer linguist uitools assistant
+# 
+# !qtConfig(process): SUBDIRS -= assistant designer
+# 
+# qtNomakeTools( \
+#     assistant \
+# )
diff --git a/src/assistant/help/help.pro b/src/assistant/help/help.pro
index a1616b23..67f3797b 100644
--- a/src/assistant/help/help.pro
+++ b/src/assistant/help/help.pro
@@ -46,3 +46,4 @@ HEADERS += qhelpenginecore.h \
     qhelpsearchindexreader_p.h
 
 load(qt_module)
+VERSION = $$[QT_VERSION]
diff --git a/src/src.pro b/src/src.pro
index 7a1af007..20b4df24 100644
--- a/src/src.pro
+++ b/src/src.pro
@@ -6,47 +6,47 @@ qtHaveModule(widgets) {
     } else {
         QT_FOR_CONFIG += widgets
         qtConfig(pushbutton):qtConfig(toolbutton) {
-            SUBDIRS = assistant \
-                      designer \
-                      pixeltool
-
-            linguist.depends = designer
+            SUBDIRS = assistant
+#                       designer \
+#                       pixeltool
+# 
+#             linguist.depends = designer
         }
-        qtHaveModule(quick):qtConfig(thread):qtConfig(toolbutton): SUBDIRS += distancefieldgenerator
+#         qtHaveModule(quick):qtConfig(thread):qtConfig(toolbutton): SUBDIRS += distancefieldgenerator
     }
 }
 
-SUBDIRS += linguist \
-    qtattributionsscanner
+# SUBDIRS += linguist \
+#     qtattributionsscanner
 
-qtConfig(library) {
-    !android|android_app: SUBDIRS += qtplugininfo
-}
+# qtConfig(library) {
+#     !android|android_app: SUBDIRS += qtplugininfo
+# }
 
-config_clang: qtConfig(thread): SUBDIRS += qdoc
+# config_clang: qtConfig(thread): SUBDIRS += qdoc
 
-!android|android_app: SUBDIRS += qtpaths
+# !android|android_app: SUBDIRS += qtpaths
 
-macos {
-    SUBDIRS += macdeployqt
-}
+# macos {
+#     SUBDIRS += macdeployqt
+# }
 
-qtHaveModule(dbus): SUBDIRS += qdbus
+# qtHaveModule(dbus): SUBDIRS += qdbus
 
-win32|winrt:SUBDIRS += windeployqt
-winrt:SUBDIRS += winrtrunner
-qtHaveModule(gui):!android:!uikit:!qnx:!winrt: SUBDIRS += qtdiag
+# win32|winrt:SUBDIRS += windeployqt
+# winrt:SUBDIRS += winrtrunner
+# qtHaveModule(gui):!android:!uikit:!qnx:!winrt: SUBDIRS += qtdiag
 
-qtNomakeTools( \
-    distancefieldgenerator \
-    pixeltool \
-)
-
-# This is necessary to avoid a race condition between toolchain.prf
-# invocations in a module-by-module cross-build.
-cross_compile:isEmpty(QMAKE_HOST_CXX.INCDIRS) {
-    qdoc.depends += qtattributionsscanner
-    windeployqt.depends += qtattributionsscanner
-    winrtrunner.depends += qtattributionsscanner
-    linguist.depends += qtattributionsscanner
-}
+# qtNomakeTools( \
+#     distancefieldgenerator \
+#     pixeltool \
+# )
+# 
+# # This is necessary to avoid a race condition between toolchain.prf
+# # invocations in a module-by-module cross-build.
+# cross_compile:isEmpty(QMAKE_HOST_CXX.INCDIRS) {
+#     qdoc.depends += qtattributionsscanner
+#     windeployqt.depends += qtattributionsscanner
+#     winrtrunner.depends += qtattributionsscanner
+#     linguist.depends += qtattributionsscanner
+# }
diff --git a/tests/tests.pro b/tests/tests.pro
index 85e4f3a5..8e7a68a0 100644
--- a/tests/tests.pro
+++ b/tests/tests.pro
@@ -1,2 +1,2 @@
-TEMPLATE = subdirs
-SUBDIRS +=  auto
+# TEMPLATE = subdirs
+# SUBDIRS +=  auto

diff --git a/src/assistant/qhelpgenerator/qhelpprojectdata.cpp b/src/assistant/qhelpgenerator/qhelpprojectdata.cpp
index 23419bfd..66abb886 100644
--- a/src/assistant/qhelpgenerator/qhelpprojectdata.cpp
+++ b/src/assistant/qhelpgenerator/qhelpprojectdata.cpp
@@ -86,7 +86,7 @@ void QHelpProjectDataPrivate::skipUnknownToken()
 {
     const QString message = QCoreApplication::translate("QHelpProject",
                   "Skipping unknown token <%1> in file \"%2\".")
-                  .arg(name()).arg(fileName) + QLatin1Char('\n');
+                  .arg(*name().string()).arg(fileName) + QLatin1Char('\n');
     fputs(qPrintable(message), stdout);
 
     skipCurrentElement();
