diff --git a/abstract_client.cpp b/abstract_client.cpp
index a12cc5e..dddd5d0 100644
--- a/abstract_client.cpp
+++ b/abstract_client.cpp
@@ -1068,9 +1068,12 @@ Options::MouseCommand AbstractClient::getWheelCommand(Qt::Orientation orientatio
     if (orientation != Qt::Vertical) {
         return Options::MouseNothing;
     }
-    if (!isActive()) {
+    const auto cmd = options->commandWindowWheel();
+    if (!isActive()
+        || (cmd == Options::MouseActivateRaiseAndPassClick
+            && this != workspace()->topClientOnDesktop(desktop(), screen()))) {
         *handled = true;
-        return options->commandWindowWheel();
+        return cmd;
     }
     return Options::MouseNothing;
 }
