diff --git a/useractions.cpp b/useractions.cpp
index c5d7ec0..f0cdacc 100644
--- a/useractions.cpp
+++ b/useractions.cpp
@@ -309,6 +309,18 @@ void UserActionsMenu::init()
     m_noBorderOperation->setCheckable(true);
     m_noBorderOperation->setData(Options::NoBorderOp);
 
+    advancedMenu->addSeparator();
+    m_toggleCompositor = advancedMenu->addAction(i18n("Toggle Compositing"));
+    setShortcut(m_toggleCompositor, QStringLiteral("Suspend Compositing"));
+    m_toggleCompositor->setCheckable(true);
+    m_toggleCompositor->setChecked(Compositor::compositing());
+    connect(m_toggleCompositor, &QAction::triggered, this,
+                [this]() {
+                    Compositor::self()->slotToggleCompositing();
+                    m_toggleCompositor->setChecked(Compositor::compositing());
+                }
+    );
+
     advancedMenu->addSeparator();
 
     m_shortcutOperation = advancedMenu->addAction(i18n("Window Short&cut..."));
diff --git a/useractions.h b/useractions.h
index e032fbf..41e68d4 100644
--- a/useractions.h
+++ b/useractions.h
@@ -277,6 +277,8 @@ private:
     QWeakPointer<AbstractClient> m_client;
     QAction *m_rulesOperation = nullptr;
     QAction *m_applicationRulesOperation = nullptr;
+
+    QAction *m_toggleCompositor = nullptr;
 };
 
 class ShortcutDialog
