--- orig.main.cpp	2015-10-17 13:56:29.000000000 +0200
+++ main.cpp	2015-12-19 23:03:52.000000000 +0100
@@ -83,7 +83,7 @@
 	aboutData.addCredit( i18n("Bernd Gehrmann"), i18n("Cervisia diff viewer"), "bernd@physik.hu-berlin.de" );
 
 	QApplication app(argc, argv);
-	app.setWindowIcon(QIcon::fromTheme(QStringLiteral("kompare")));
+	app.setWindowIcon(QIcon::fromTheme(QStringLiteral("kompare"), app.windowIcon()));
 	QCommandLineParser parser;
 	KAboutData::setApplicationData(aboutData);
 	parser.addVersionOption();
