diff --git ext/soup/orig.gstsouploader.c ext/soup/gstsouploader.c
index 9192e4d..da7c4d0 100644
--- ext/soup/orig.gstsouploader.c
+++ ext/soup/gstsouploader.c
@@ -34,12 +34,17 @@ GST_DEBUG_CATEGORY (gst_soup_debug);
 
 #ifndef LINK_SOUP
 
-#if defined(__APPLE__) || defined(G_OS_WIN32)
-#error "dlopen of libsoup is only supported on Linux"
+#if  defined(G_OS_WIN32)
+#error "dlopen of libsoup is only supported on Mac & Linux"
 #endif
 
+#ifdef __APPLE__
+#define LIBSOUP_3_SONAME "libsoup-3.0.0.dylib"
+#define LIBSOUP_2_SONAME "libsoup-2.4.1.dylib"
+#else
 #define LIBSOUP_3_SONAME "libsoup-3.0.so.0"
 #define LIBSOUP_2_SONAME "libsoup-2.4.so.1"
+#endif
 
 #define LOAD_SYMBOL(name) G_STMT_START {                                \
     if (!g_module_symbol (module, G_STRINGIFY (name), (gpointer *) &G_PASTE (vtable->_, name))) { \
diff --git ext/soup/orig.meson.build ext/soup/meson.build
index aaa01db..6b994e3 100644
--- ext/soup/orig.meson.build
+++ ext/soup/meson.build
@@ -20,8 +20,8 @@ soup_link_deps = []
 libsoup2_dep = disabler()
 libsoup3_dep = disabler()
 default_library = get_option('default_library')
-soup_lookup_dep = get_option('soup-lookup-dep') and host_system == 'linux'
-if host_system != 'linux' or default_library in ['static', 'both'] or soup_lookup_dep
+soup_lookup_dep = get_option('soup-lookup-dep')
+if default_library in ['static', 'both'] or soup_lookup_dep
   if soup_ver_opt in ['auto', '3']
     libsoup3_dep = dependency('libsoup-3.0', allow_fallback: true,
                               required: soup_ver_opt == '3' and soup_opt.enabled())
@@ -33,7 +33,7 @@ if host_system != 'linux' or default_library in ['static', 'both'] or soup_looku
   endif
 endif
 
-if host_system != 'linux' or default_library in ['static', 'both']
+if default_library in ['static', 'both']
   if libsoup3_dep.found()
     soup_link_deps += libsoup3_dep
     soup_link_args += '-DLINK_SOUP=3'
@@ -60,15 +60,15 @@ soup_library_kwargs = {
 soup_library_deps = [gst_dep, gstbase_dep, gsttag_dep, gmodule_dep, gio_dep, libdl_dep]
 soup_library_c_args = gst_plugins_good_args
 
-if host_system != 'linux'
-  gstsouphttpsrc = library('gstsoup',
-    c_args : soup_library_c_args + soup_link_args,
-    dependencies : soup_library_deps + soup_link_deps,
-    kwargs: soup_library_kwargs,
-  )
-  gstsouphttpsrc_shared = gstsouphttpsrc
-  gstsouphttpsrc_static = gstsouphttpsrc
-else
+# if host_system != 'linux'
+#   gstsouphttpsrc = library('gstsoup',
+#     c_args : soup_library_c_args + soup_link_args,
+#     dependencies : soup_library_deps + soup_link_deps,
+#     kwargs: soup_library_kwargs,
+#   )
+#   gstsouphttpsrc_shared = gstsouphttpsrc
+#   gstsouphttpsrc_static = gstsouphttpsrc
+# else
   if default_library in ['static', 'both']
     gstsouphttpsrc_static = static_library('gstsoup',
       c_args : soup_library_c_args + soup_link_args,
@@ -83,7 +83,7 @@ else
       kwargs: soup_library_kwargs,
     )
   endif
-endif
+# endif
 
 # Use the static library to generate the .pc file on Linux if it's available.
 # In that case, the shared library .pc file does not have a Requires: on
diff --git ext/adaptivedemux2/orig.meson.build ext/adaptivedemux2/meson.build
index 711b38a..b4a4d2e 100644
--- ext/adaptivedemux2/orig.meson.build
+++ ext/adaptivedemux2/meson.build
@@ -82,7 +82,7 @@ soup_link_args = []
 soup_link_deps = []
 
 default_library = get_option('default_library')
-if host_system != 'linux' or default_library in ['static', 'both']
+if default_library in ['static', 'both']
   if soup_ver_opt in ['auto', '3']
     libsoup3_dep = dependency('libsoup-3.0', allow_fallback: true,
                               required: soup_ver_opt == '3' and soup_opt.enabled())
@@ -120,14 +120,14 @@ adaptive_deps = [gmodule_dep, gst_dep, gsttag_dep, gstnet_dep, gstbase_dep, gstp
 adaptive_args = [gst_plugins_good_args, soup_loader_args, hls_cargs,
     '-DGST_ISOFF_API=G_GNUC_INTERNAL']
 
-if host_system != 'linux'
-    adaptivedemux2 = library('gstadaptivedemux2',
-      c_args: [adaptive_args, soup_link_args],
-      dependencies: [adaptive_deps, soup_link_deps],
-      kwargs: adaptive_kwargs)
-    adaptivedemux2_static = adaptivedemux2
-    adaptivedemux2_shared = adaptivedemux2
-else
+# if host_system != 'linux'
+#     adaptivedemux2 = library('gstadaptivedemux2',
+#       c_args: [adaptive_args, soup_link_args],
+#       dependencies: [adaptive_deps, soup_link_deps],
+#       kwargs: adaptive_kwargs)
+#     adaptivedemux2_static = adaptivedemux2
+#     adaptivedemux2_shared = adaptivedemux2
+# else
   if default_library in ['static', 'both']
     # Static plugin links to libsoup directly at build time
     adaptivedemux2_static = static_library('gstadaptivedemux2',
@@ -141,7 +141,7 @@ else
       dependencies: adaptive_deps,
       kwargs: adaptive_kwargs)
   endif
-endif
+# endif
 
 # Use the static library to generate the .pc file if it's available. The shared
 # library .pc file does not have a Requires: on libsoup, and we use plugin
