diff --git ext/qt/orig.meson.build ext/qt/meson.build
index 192a644..5d095aa 100644
--- ext/qt/orig.meson.build
+++ ext/qt/meson.build
@@ -154,16 +154,18 @@ if host_system == 'windows'
 endif
 
 # macOS windowing
-if host_system == 'darwin'
-  qt5_macos = qt5_option \
-      .require(gst_gl_have_window_cocoa, error_message: 'cocoa windowing support in gstreamer-gl is required') \
-      .require(gst_gl_have_platform_cgl, error_message: 'cgl platform support in gstreamer-gl is required')
-  if qt5_macos.allowed()
-    qt5macextras = dependency('qt5', modules : ['MacExtras'], method: qt5_method, required : qt5_macos)
-    if qt5macextras.found()
-      qt_defines += ['-DHAVE_QT_MAC']
-      optional_deps += qt5macextras
-      have_qt_windowing = true
+if not qt5_x11.allowed()
+  if host_system == 'darwin'
+    qt5_macos = qt5_option \
+        .require(gst_gl_have_window_cocoa, error_message: 'cocoa windowing support in gstreamer-gl is required') \
+        .require(gst_gl_have_platform_cgl, error_message: 'cgl platform support in gstreamer-gl is required')
+    if qt5_macos.allowed()
+      qt5macextras = dependency('qt5', modules : ['MacExtras'], method: qt5_method, required : qt5_macos)
+      if qt5macextras.found()
+        qt_defines += ['-DHAVE_QT_MAC']
+        optional_deps += qt5macextras
+        have_qt_windowing = true
+      endif
     endif
   endif
 endif
