diff --git a/libportal/notification.c b/libportal/notification.c
index 1cd67b2..045dd09 100644
--- a/libportal/notification.c
+++ b/libportal/notification.c
@@ -35,6 +35,21 @@
 #include "notification.h"
 #include "portal-private.h"
 
+/* memfd_create was introduced in glibc 2.27 */
+#if !defined(__GLIBC__) || (__GLIBC__ >= 2 && __GLIBC_MINOR__ >= 27)
+// all good
+#else
+#include <sys/syscall.h>
+
+// we wrap memfd_create(2) but only if it is not provided by the local glibc library!
+int memfd_create(const char *__name, unsigned int __flags)
+{
+    return syscall(__NR_memfd_create, __name, __flags);
+}
+
+#endif
+
+
 static void
 action_invoked (GDBusConnection *bus,
                 const char *sender_name,
