--- qt-everywhere-opensource-src-5.9.6/qtlocation/src/3rdparty/mapbox-gl-native/orig.mapbox-gl-native.pro	2017-11-29 19:38:56.000000000 +0100
+++ qt-everywhere-opensource-src-5.9.6/qtlocation/src/3rdparty/mapbox-gl-native/mapbox-gl-native.pro	2018-06-20 22:56:33.061494312 +0200
@@ -28,6 +28,8 @@
 }
 
 darwin {
+    QMAKE_CXXFLAGS += \
+         -D_LIBCPP_ENABLE_CXX17_REMOVED_AUTO_PTR
     LIBS += \
         -framework Foundation
 }
