--- qt-everywhere-opensource-src-5.9.6/qtmultimedia/src/plugins/pulseaudio/orig.qaudiooutput_pulse.cpp	2018-08-03 14:54:34.447207000 +0200
+++ qt-everywhere-opensource-src-5.9.6/qtmultimedia/src/plugins/pulseaudio/qaudiooutput_pulse.cpp	2018-08-03 14:54:40.717207000 +0200
@@ -473,6 +473,11 @@
 
 qint64 QPulseAudioOutput::write(const char *data, qint64 len)
 {
+    if (Q_UNLIKELY(!m_stream)) {
+        qWarning("QAudioOutput(pulseaudio): no stream");
+        return 0;
+    }
+
     QPulseAudioEngine *pulseEngine = QPulseAudioEngine::instance();
 
     pulseEngine->lock();
