commit f739fd13ca44f0c4d5a86b69031c1f85c6f326d9
Author: Stefan Csomor <csomor@advancedconcepts.ch>
Date:   Mon Jun 29 22:37:01 2020 +0200

    switching to 10.10+ API

diff --git b/src/osx/cocoa/dataview.mm a/src/osx/cocoa/dataview.mm
index b093ca6a6d..48622392ce 100644
--- b/src/osx/cocoa/dataview.mm
+++ a/src/osx/cocoa/dataview.mm
@@ -1180,7 +1180,7 @@ outlineView:(NSOutlineView*)outlineView
     wxDataViewCustomRenderer * const renderer = obj->customRenderer;
 
     // if this method is called everything is already setup correctly,
-    CGContextRef context = [[NSGraphicsContext currentContext] CGContext];
+    CGContextRef context = (CGContextRef) [[NSGraphicsContext currentContext] graphicsPort];
     CGContextSaveGState( context );
 
     if ( ![controlView isFlipped] )
diff --git b/src/osx/cocoa/overlay.mm a/src/osx/cocoa/overlay.mm
index 5e7eac8048..b4c8f0368c 100644
--- b/src/osx/cocoa/overlay.mm
+++ a/src/osx/cocoa/overlay.mm
@@ -112,7 +112,7 @@ void wxOverlayImpl::Init( wxDC* dc, int x , int y , int width , int height )
 
     CreateOverlayWindow(dc);
     wxASSERT_MSG(m_overlayWindow != NULL, _("Couldn't create the overlay window"));
-    m_overlayContext = [[m_overlayWindow graphicsContext] CGContext];
+    m_overlayContext = (CGContextRef) [[m_overlayWindow graphicsContext] graphicsPort];
     wxASSERT_MSG(  m_overlayContext != NULL , _("Couldn't init the context on the overlay window") );
     [(id)m_overlayContext retain];
 }
diff --git b/src/osx/cocoa/window.mm a/src/osx/cocoa/window.mm
index 640287c90a..67e80fb79b 100644
--- b/src/osx/cocoa/window.mm
+++ a/src/osx/cocoa/window.mm
@@ -1255,7 +1255,7 @@ void wxOSX_drawRect(NSView* self, SEL _cmd, NSRect rect)
             if ( win->UseBgCol() )
             {
                 
-                CGContextRef context = [[NSGraphicsContext currentContext] CGContext];
+                CGContextRef context = (CGContextRef) [[NSGraphicsContext currentContext] graphicsPort];
                 CGContextSaveGState( context );
 
                 CGContextSetFillColorWithColor( context, win->GetBackgroundColour().GetCGColor());
@@ -2298,7 +2298,7 @@ void wxWidgetCocoaImpl::drawRect(void* rect, WXWidget slf, void *WXUNUSED(_cmd))
 
     // setting up the drawing context
     // note that starting from 10.14 this may be NULL in certain views
-    CGContextRef context = [[NSGraphicsContext currentContext] CGContext];
+    CGContextRef context = (CGContextRef) [[NSGraphicsContext currentContext] graphicsPort];
     wxpeer->MacSetCGContextRef( context );
     if ( context != NULL )
     {
