commit 360240a70bae3fef6ccc44e62f0f32c7845b576b
Author: dvertx <dvertx@gmail.com>
Date:   Tue Jun 30 03:27:25 2020 +0700

    Replace deprecated methods from utilscocoa.mm (#1916)

diff --git a/src/osx/carbon/utilscocoa.mm b/src/osx/carbon/utilscocoa.mm
index a3062ba261..081affa5b6 100644
--- a/src/osx/carbon/utilscocoa.mm
+++ b/src/osx/carbon/utilscocoa.mm
@@ -46,7 +46,7 @@ wxMacAutoreleasePool::~wxMacAutoreleasePool()
 
 CGContextRef wxOSXGetContextFromCurrentContext()
 {
-    CGContextRef context = [[NSGraphicsContext currentContext] CGContext];
+    CGContextRef context = (CGContextRef)[[NSGraphicsContext currentContext] graphicsPort];
     return context;
 }
 
@@ -208,7 +208,7 @@ WXImage  wxOSXGetImageFromCGImage( CGImageRef image, double scaleFactor, bool is
     sz.height = CGImageGetHeight(image)/scaleFactor;
     sz.width = CGImageGetWidth(image)/scaleFactor;
     NSImage* newImage = [[NSImage alloc] initWithCGImage:image size:sz];
-
+    
     [newImage setTemplate:isTemplate];
 
     [newImage autorelease];
@@ -239,7 +239,7 @@ CGImageRef WXDLLIMPEXP_CORE wxOSXGetCGImageFromImage( WXImage nsimage, CGRect* r
 #if wxOSX_USE_COCOA
     NSRect nsRect = NSRectFromCGRect(*r);
     return [nsimage CGImageForProposedRect:&nsRect
-                               context:[NSGraphicsContext graphicsContextWithCGContext:cg flipped:YES]
+                               context:[NSGraphicsContext graphicsContextWithGraphicsPort:cg flipped:YES]
                                         hints:nil];
 #else
     return [nsimage CGImage];
@@ -249,21 +249,21 @@ CGImageRef WXDLLIMPEXP_CORE wxOSXGetCGImageFromImage( WXImage nsimage, CGRect* r
 CGContextRef WXDLLIMPEXP_CORE wxOSXCreateBitmapContextFromImage( WXImage nsimage, bool *isTemplate)
 {
     // based on http://www.mail-archive.com/cocoa-dev@lists.apple.com/msg18065.html
-
+    
     CGContextRef hbitmap = NULL;
     if (nsimage != nil)
     {
         double scale = wxOSXGetMainScreenContentScaleFactor();
 
         CGSize imageSize = wxOSXGetImageSize(nsimage);
-
+        
         hbitmap = CGBitmapContextCreate(NULL, imageSize.width*scale, imageSize.height*scale, 8, 0, wxMacGetGenericRGBColorSpace(), kCGImageAlphaPremultipliedFirst);
         CGContextScaleCTM( hbitmap, scale, scale );
         CGContextClearRect(hbitmap,CGRectMake(0, 0, imageSize.width, imageSize.height));
 
 #if wxOSX_USE_COCOA
         NSGraphicsContext *previousContext = [NSGraphicsContext currentContext];
-        NSGraphicsContext *nsGraphicsContext = [NSGraphicsContext graphicsContextWithCGContext:hbitmap flipped:NO];
+        NSGraphicsContext *nsGraphicsContext = [NSGraphicsContext graphicsContextWithGraphicsPort:hbitmap flipped:NO];
         [NSGraphicsContext setCurrentContext:nsGraphicsContext];
         [nsimage drawAtPoint:NSZeroPoint fromRect:NSZeroRect operation:NSCompositeCopy fraction:1.0];
         [NSGraphicsContext setCurrentContext:previousContext];
@@ -293,7 +293,7 @@ void WXDLLIMPEXP_CORE wxOSXDrawNSImage(
 
 #if wxOSX_USE_COCOA
        NSGraphicsContext *previousContext = [NSGraphicsContext currentContext];
-        NSGraphicsContext *nsGraphicsContext = [NSGraphicsContext graphicsContextWithCGContext:inContext flipped:NO];
+        NSGraphicsContext *nsGraphicsContext = [NSGraphicsContext graphicsContextWithGraphicsPort:inContext flipped:NO];
         [NSGraphicsContext setCurrentContext:nsGraphicsContext];
         [inImage drawInRect:NSRectFromCGRect(r) fromRect:NSZeroRect operation:NSCompositeSourceOver fraction:1.0];
         [NSGraphicsContext setCurrentContext:previousContext];
@@ -560,10 +560,10 @@ WX_NSCursor wxMacCocoaCreateStockCursor( int cursor_type )
     default:
         break;
     }
-
+    
     if ( cursor == nil )
         cursor = [[NSCursor arrowCursor] retain];
-
+    
     return cursor;
 }
 
@@ -652,3 +652,4 @@ wxOSXEffectiveAppearanceSetter::~wxOSXEffectiveAppearanceSetter()
 }
 
 #endif
+
