diff --git a/lib-src/FileDialog/Makefile.am b/lib-src/FileDialog/Makefile.am
index a07b4ea..3be796e 100644
--- a/lib-src/FileDialog/Makefile.am
+++ b/lib-src/FileDialog/Makefile.am
@@ -25,7 +25,7 @@ libFileDialog_la_SOURCES += \
 	$(NULL)
 endif
 
-if MAC
+if MACNATIVE
 libFileDialog_la_SOURCES += mac/FileDialogPrivate.cpp mac/FileDialogPrivate.h
 endif
 
diff --git a/lib-src/FileDialog/Makefile.in b/lib-src/FileDialog/Makefile.in
index 61c60c7..7b1ce94 100644
--- a/lib-src/FileDialog/Makefile.in
+++ b/lib-src/FileDialog/Makefile.in
@@ -64,7 +64,7 @@ host_triplet = @host@
 @GTK_TRUE@ gtk/private.h \
 @GTK_TRUE@ $(NULL)
 
-@MAC_TRUE@am__append_5 = mac/FileDialogPrivate.cpp mac/FileDialogPrivate.h
+@MACNATIVE_TRUE@am__append_5 = mac/FileDialogPrivate.cpp mac/FileDialogPrivate.h
 @WINDOWS_TRUE@am__append_6 = win/FileDialogPrivate.cpp win/FileDialogPrivate.h
 subdir = .
 DIST_COMMON = $(am__configure_deps) $(srcdir)/Makefile.am \
@@ -127,7 +127,7 @@ am__dirstamp = $(am__leading_dot)dirstamp
 @GENERIC_TRUE@am__objects_1 =  \
 @GENERIC_TRUE@ generic/libFileDialog_la-FileDialogPrivate.lo
 @GTK_TRUE@am__objects_2 = gtk/libFileDialog_la-FileDialogPrivate.lo
-@MAC_TRUE@am__objects_3 = mac/libFileDialog_la-FileDialogPrivate.lo
+@MACNATIVE_TRUE@am__objects_3 = mac/libFileDialog_la-FileDialogPrivate.lo
 @WINDOWS_TRUE@am__objects_4 =  \
 @WINDOWS_TRUE@ win/libFileDialog_la-FileDialogPrivate.lo
 am_libFileDialog_la_OBJECTS = libFileDialog_la-FileDialog.lo \
diff --git a/lib-src/FileDialog/configure b/lib-src/FileDialog/configure
index b449cad..8887b8d 100755
--- a/lib-src/FileDialog/configure
+++ b/lib-src/FileDialog/configure
@@ -15562,9 +15562,9 @@ fi
 
 IMPLEMENTATION="generic"
 case "${host_os}" in
-   darwin*)
-            IMPLEMENTATION="mac"
-   ;;
+#    darwin*)
+#             IMPLEMENTATION="mac"
+#    ;;
 
    cygwin*)
             IMPLEMENTATION="win"
diff --git a/lib-src/FileDialog/configure.ac b/lib-src/FileDialog/configure.ac
index d1019f5..94dcbb1 100644
--- a/lib-src/FileDialog/configure.ac
+++ b/lib-src/FileDialog/configure.ac
@@ -84,10 +84,10 @@ AC_CANONICAL_HOST
 
 IMPLEMENTATION="generic"
 case "${host_os}" in
-   darwin*)
-      dnl Mac OS X configuration
-      IMPLEMENTATION="mac"
-   ;;
+dnl    darwin*)
+dnl       dnl Mac OS X configuration
+dnl       IMPLEMENTATION="mac"
+dnl    ;;
 
    cygwin*)
       dnl Windows/CygWin configuration
