--- src/kdecore/orig.kmimetyperepository.cpp	2015-11-08 13:16:09.000000000 +0100
+++ src/kdecore/kmimetyperepository.cpp	2015-12-05 10:33:44.179915627 +0100
@@ -148,6 +148,7 @@
 #elif defined(Q_OS_OPENBSD) || defined(Q_OS_NETBSD) || defined(Q_OS_SOLARIS)
     paths << QLatin1String("/usr/local/lib/pkgconfig"); // {Net,Open}BSD/OpenSolaris
 #elif defined (Q_OS_UNIX)
+    paths << QLatin1String("@PREFIX@/share/pkgconfig"); // MacPorts
     paths << QLatin1String("/usr/share/pkgconfig"); // Linux and all other unix
 #endif
 }
