diff --git src/kiod/CMakeLists.txt src/kiod/CMakeLists.txt
index 51d7f23..68e7c01 100644
--- src/kiod/CMakeLists.txt
+++ src/kiod/CMakeLists.txt
@@ -1,3 +1,5 @@
+include(ECMMarkNonGuiExecutable)
+
 set(kiod_SRCS kiod_main.cpp)
 
 add_executable(kiod5 ${kiod_SRCS})
@@ -11,6 +13,13 @@ target_link_libraries(kiod5
    Qt5::Widgets     # QApplication
 )
 
+if (APPLE)
+    target_link_libraries(kiod5 "-framework CoreFoundation")
+
+    # Mark it as non-gui so we won't create an app bundle on Mac OS X
+    ecm_mark_nongui_executable(kiod5)
+endif ()
+
 install(TARGETS kiod5 DESTINATION ${KDE_INSTALL_LIBEXECDIR_KF5})
 
 configure_file(org.kde.kiod5.service.in
diff --git src/kiod/kiod_main.cpp src/kiod/kiod_main.cpp
index ae9a187..cd650d3 100644
--- src/kiod/kiod_main.cpp
+++ src/kiod/kiod_main.cpp
@@ -33,6 +33,10 @@
 Q_DECLARE_LOGGING_CATEGORY(KIOD_CATEGORY)
 Q_LOGGING_CATEGORY(KIOD_CATEGORY, "kf5.kiod")
 
+#ifdef Q_OS_OSX
+#include <CoreFoundation/CoreFoundation.h>
+#endif
+
 class KIOD
 {
 public:
@@ -90,6 +94,20 @@ extern Q_DBUS_EXPORT void qDBusAddSpyHook(void (*)(const QDBusMessage&));
 
 int main(int argc, char *argv[])
 {
+#ifdef Q_OS_OSX
+    CFBundleRef mainBundle = CFBundleGetMainBundle();
+    if (mainBundle) {
+        // get the application's Info Dictionary. For app bundles this would live in the bundle's Info.plist,
+        // for regular executables it is obtained in another way.
+        CFMutableDictionaryRef infoDict = (CFMutableDictionaryRef) CFBundleGetInfoDictionary(mainBundle);
+        if (infoDict) {
+            // Add or set the "LSUIElement" key with/to value "1". This can simply be a CFString.
+            CFDictionarySetValue(infoDict, CFSTR("LSUIElement"), CFSTR("1"));
+            // That's it. We're now considered as an "agent" by the window server, and thus will have
+            // neither menubar nor presence in the Dock or App Switcher.
+        }
+    }
+#endif
     QApplication app(argc, argv); // GUI needed for kpasswdserver's dialogs
     app.setApplicationName(QStringLiteral("kiod5"));
     app.setOrganizationDomain(QStringLiteral("kde.org"));
