--- orig.CMakeLists.txt	2015-11-06 22:00:55.000000000 +0100
+++ CMakeLists.txt	2015-11-06 22:03:23.000000000 +0100
@@ -19,6 +19,7 @@
 
 set(REQUIRED_QT_VERSION 5.3.0)
 find_package(Qt5Gui ${REQUIRED_QT_VERSION} REQUIRED NO_MODULE)
+find_package(Qt5QspXDG ${REQUIRED_QT_VERSION} REQUIRED NO_MODULE)
 
 # EPS support depends on the gs utility; non-UNIX systems are unlikely to have
 # this available in PATH
--- src/imageformats/orig.CMakeLists.txt	2015-10-03 12:22:35.000000000 +0200
+++ src/imageformats/CMakeLists.txt	2015-11-06 22:03:54.000000000 +0100
@@ -9,7 +9,7 @@
 if (BUILD_EPS_PLUGIN)
     if (Qt5PrintSupport_FOUND)
         add_library(kimg_eps MODULE eps.cpp)
-        target_link_libraries(kimg_eps  Qt5::Gui Qt5::PrintSupport)
+        target_link_libraries(kimg_eps  PUBLIC Qt5::Gui Qt5::PrintSupport PRIVATE Qt5::QspXDG)
 
         install(TARGETS kimg_eps  DESTINATION ${KDE_INSTALL_QTPLUGINDIR}/imageformats/)
         install(FILES eps.desktop DESTINATION ${KDE_INSTALL_KSERVICES5DIR}/qimageioplugins/)
@@ -26,7 +26,7 @@
 
 if(OpenEXR_FOUND)
     add_library(kimg_exr MODULE exr.cpp)
-    target_link_libraries(kimg_exr Qt5::Gui OpenEXR::IlmImf)
+    target_link_libraries(kimg_exr PUBLIC Qt5::Gui OpenEXR::IlmImf PRIVATE Qt5::QspXDG)
     kde_target_enable_exceptions(kimg_exr PRIVATE)
 
     install(TARGETS kimg_exr  DESTINATION ${KDE_INSTALL_QTPLUGINDIR}/imageformats/)
@@ -36,7 +36,7 @@
 ##################################
 
 add_library(kimg_pcx MODULE pcx.cpp)
-target_link_libraries(kimg_pcx  Qt5::Gui)
+target_link_libraries(kimg_pcx  PUBLIC Qt5::Gui PRIVATE Qt5::QspXDG)
 
 install(TARGETS kimg_pcx  DESTINATION ${KDE_INSTALL_QTPLUGINDIR}/imageformats/)
 install(FILES pcx.desktop DESTINATION ${KDE_INSTALL_KSERVICES5DIR}/qimageioplugins/)
@@ -44,7 +44,7 @@
 ##################################
 
 add_library(kimg_pic MODULE pic.cpp)
-target_link_libraries(kimg_pic Qt5::Gui)
+target_link_libraries(kimg_pic PUBLIC Qt5::Gui PRIVATE Qt5::QspXDG)
 
 install(TARGETS kimg_pic  DESTINATION ${KDE_INSTALL_QTPLUGINDIR}/imageformats/)
 install(FILES pic.desktop DESTINATION ${KDE_INSTALL_KSERVICES5DIR}/qimageioplugins/)
@@ -52,7 +52,7 @@
 ##################################
 
 add_library(kimg_psd MODULE psd.cpp)
-target_link_libraries(kimg_psd Qt5::Gui)
+target_link_libraries(kimg_psd PUBLIC Qt5::Gui PRIVATE Qt5::QspXDG)
 
 install(TARGETS kimg_psd  DESTINATION ${KDE_INSTALL_QTPLUGINDIR}/imageformats/)
 install(FILES psd.desktop DESTINATION ${KDE_INSTALL_KSERVICES5DIR}/qimageioplugins/)
@@ -60,7 +60,7 @@
 ##################################
 
 add_library(kimg_ras MODULE ras.cpp)
-target_link_libraries(kimg_ras  Qt5::Gui)
+target_link_libraries(kimg_ras  PUBLIC Qt5::Gui PRIVATE Qt5::QspXDG)
 
 install(TARGETS kimg_ras  DESTINATION ${KDE_INSTALL_QTPLUGINDIR}/imageformats/)
 install(FILES ras.desktop DESTINATION ${KDE_INSTALL_KSERVICES5DIR}/qimageioplugins/)
@@ -68,7 +68,7 @@
 ##################################
 
 add_library(kimg_rgb MODULE rgb.cpp)
-target_link_libraries(kimg_rgb  Qt5::Gui)
+target_link_libraries(kimg_rgb  PUBLIC Qt5::Gui PRIVATE Qt5::QspXDG)
 
 install(TARGETS kimg_rgb  DESTINATION ${KDE_INSTALL_QTPLUGINDIR}/imageformats/)
 install(FILES rgb.desktop DESTINATION ${KDE_INSTALL_KSERVICES5DIR}/qimageioplugins/)
@@ -76,7 +76,7 @@
 ##################################
 
 add_library(kimg_tga MODULE tga.cpp)
-target_link_libraries(kimg_tga  Qt5::Gui)
+target_link_libraries(kimg_tga  PUBLIC Qt5::Gui PRIVATE Qt5::QspXDG)
 
 install(TARGETS kimg_tga  DESTINATION ${KDE_INSTALL_QTPLUGINDIR}/imageformats/)
 install(FILES tga.desktop DESTINATION ${KDE_INSTALL_KSERVICES5DIR}/qimageioplugins/)
@@ -84,7 +84,7 @@
 ##################################
 
 add_library(kimg_xcf MODULE xcf.cpp)
-target_link_libraries(kimg_xcf Qt5::Gui)
+target_link_libraries(kimg_xcf PUBLIC Qt5::Gui PRIVATE Qt5::QspXDG)
 
 install(TARGETS kimg_xcf  DESTINATION ${KDE_INSTALL_QTPLUGINDIR}/imageformats/)
 install(FILES xcf.desktop DESTINATION ${KDE_INSTALL_KSERVICES5DIR}/qimageioplugins/)
