--- orig.CMakeLists.txt	2015-12-15 18:14:59.000000000 +0100
+++ CMakeLists.txt	2015-12-15 18:33:47.000000000 +0100
@@ -46,7 +46,7 @@
 find_package(KF5CoreAddons ${KF5_DEP_VERSION} REQUIRED)
 find_package(KF5Config ${KF5_DEP_VERSION} REQUIRED)
 
-find_package(KF5DocTools ${KF5_DEP_VERSION})
+find_package(KF5DocTools)
 set_package_properties(KF5DocTools PROPERTIES DESCRIPTION "Tools to generate documentation"
                        TYPE OPTIONAL
                       )
