diff --git src/kselectaction.cpp src/kselectaction.cpp
index 1381099..030028c 100644
--- src/kselectaction.cpp
+++ src/kselectaction.cpp
@@ -97,8 +97,12 @@ KSelectAction::KSelectAction(KSelectActionPrivate &dd, QObject *parent)
 
 KSelectAction::~KSelectAction()
 {
+    QMenu *m = menu();
+    setMenu(Q_NULLPTR);
+    if (m) {
+        m->deleteLater();
+    }
     delete d_ptr;
-    delete menu();
 }
 
 void KSelectActionPrivate::init(KSelectAction *q)
