--- src/data/orig.katepart5ui.rc	2015-12-06 22:38:46.000000000 +0100
+++ src/data/katepart5ui.rc	2016-01-02 22:48:01.000000000 +0100
@@ -150,15 +150,12 @@
   <Action name="edit_cut" group="popup_operations" />
   <Action name="edit_copy" group="popup_operations" />
   <Action name="edit_paste" group="popup_operations" />
-  <Action name="edit_paste_menu" group="popup_operations" />
   <Separator group="popup_operations" />
   <Action name="edit_select_all" group="popup_operations" />
   <Action name="edit_deselect" group="popup_operations" />
   <Separator group="popup_operations" />
   <Action name="spelling_suggestions" group="popup_operations" />
   <Separator group="popup_operations" />
-  <Action name="bookmarks" group="popup_operations" />
-  <Separator group="popup_operations" />
   <Action name="tools_create_snippet" group="popup_operations" />
   <Separator group="popup_operations" />
 </Menu>
