diff --git soprano/sopranodirs.cpp soprano/sopranodirs.cpp
index 4518f36..b9eec3c 100644
--- soprano/sopranodirs.cpp
+++ soprano/sopranodirs.cpp
@@ -131,8 +131,8 @@ QStringList Soprano::libDirs()
     paths << getWinPrefix() + QLatin1String( SOPRANO_FULL_LIB_DIR );
 #else
     paths << QLatin1String( "/usr/" SOPRANO_LIB_DIR );
-    paths << QLatin1String( "/usr/local/" SOPRANO_LIB_DIR );
-    paths += Soprano::envDirList( "LD_LIBRARY_PATH" );
+    paths << QLatin1String( "@PREFIX@/" SOPRANO_LIB_DIR );
+    paths += Soprano::envDirList( "DYLD_LIBRARY_PATH" );
 #endif
     return paths;
 }

