diff --git debuggers/CMakeLists.txt debuggers/CMakeLists.txt
index 8fe222c..592c1fc 100644
--- debuggers/CMakeLists.txt
+++ debuggers/CMakeLists.txt
@@ -1,4 +1,4 @@
-if(NOT WIN32 AND NOT APPLE)
+if(NOT WIN32)
   add_subdirectory(gdb)
-endif(NOT WIN32 AND NOT APPLE )
+endif(NOT WIN32)
 
diff --git debuggers/gdb/CMakeLists.txt debuggers/gdb/CMakeLists.txt
index 3d1125c..ef6100f 100644
--- debuggers/gdb/CMakeLists.txt
+++ debuggers/gdb/CMakeLists.txt
@@ -1,11 +1,13 @@
 project(gdb)
 
+if(NOT APPLE)
 macro_optional_find_package( KDE4Workspace 4.2.0 )
 
 macro_log_feature( KDE4WORKSPACE_FOUND
 "KDE 4 Workspace" "KDE 4 Workspace libraries" "kdebase/workspace" FALSE
 "" "Required for building the process selection dialog for attaching GDB to a running process."
 )
+endif(NOT APPLE)
 
 if(KDE4WORKSPACE_FOUND)
     include_directories(
@@ -80,7 +82,7 @@ set(kdevgdb_PART_UI
 
 kde4_add_ui_files(kdevgdb_PART_SRCS ${kdevgdb_PART_UI})
 kde4_add_plugin(kdevgdb ${kdevgdb_PART_SRCS})
-target_link_libraries(kdevgdb sublime
+target_link_libraries(kdevgdb ${KDEVPLATFORM_SUBLIME_LIBRARIES}
     ${KDEVPLATFORM_INTERFACES_LIBRARIES}
     ${KDEVPLATFORM_LANGUAGE_LIBRARIES}
     ${KDEVPLATFORM_DEBUGGER_LIBRARIES}
