diff --git CMakeLists.txt CMakeLists.txt
index 8edac05..9c5373a 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -50,7 +50,8 @@ endif()
 
 if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
     # in Qt 4, clang support is lacking...
-    add_definitions("-DQ_COMPILER_INITIALIZER_LISTS=1")
+    # but not in 4.8.7
+    # add_definitions("-DQ_COMPILER_INITIALIZER_LISTS=1")
 endif()
 
 # Turn off missing-field-initializers warning to avoid noise from false positives with empty {}
