diff --git projectmanagers/cmake/cmakecommitchangesjob.cpp projectmanagers/cmake/cmakecommitchangesjob.cpp
index 7ce24fb..ab71b55 100644
--- projectmanagers/cmake/cmakecommitchangesjob.cpp
+++ projectmanagers/cmake/cmakecommitchangesjob.cpp
@@ -31,6 +31,7 @@
 #include <KUrl>
 
 #include <QThread>
+#include <QDebug>
 
 using namespace KDevelop;
 
@@ -219,7 +220,14 @@ void CMakeCommitChangesJob::start()
 
 void CMakeCommitChangesJob::makeChanges()
 {
+#ifndef QT_NO_DEBUG
     Q_ASSERT(m_project->thread() == QThread::currentThread());
+#else
+    if(m_project->thread() != QThread::currentThread()){
+        qFatal( "In the wrong thread, %s:%d", __FILE__, __LINE__ );
+        abort();
+    }
+#endif
     ProjectFolderItem* f = m_parentItem;
     m_manager->addWatcher(m_project, m_path.toLocalFile());
 
@@ -229,7 +237,14 @@ void CMakeCommitChangesJob::makeChanges()
     }
 
     CMakeFolderItem* folder = dynamic_cast<CMakeFolderItem*>(f);
+#ifndef QT_NO_DEBUG
     Q_ASSERT(folder);
+#else
+    if(!folder){
+        qFatal( "\nNULL CMakeFolderItem, %s:%d (this=%p, m_parentItem=%p, f=%p)", __FILE__, __LINE__, this, m_parentItem, f );
+        abort();
+    }
+#endif
     qDeleteAll(cleanupBuildFolders(folder, m_subdirectories));
     foreach(const Subdirectory& subf, m_subdirectories)
     {
@@ -422,8 +437,15 @@ void CMakeCommitChangesJob::folderAvailable(ProjectFolderItem* item)
 }
 void CMakeCommitChangesJob::reloadFiles()
 {
+#ifndef QT_NO_DEBUG
     Q_ASSERT(m_project->thread() == QThread::currentThread());
     Q_ASSERT(m_parentItem);
+#else
+    if(!m_parentItem || m_project->thread() != QThread::currentThread()){
+        qFatal( "In the wrong thread and/or NULL CMakeFolderItem, %s:%d", __FILE__, __LINE__ );
+        abort();
+    }
+#endif
     reloadFiles(m_parentItem);
     emitResult();
 }
