diff --git languages/cpp/kdevcppsupport.desktop.cmake languages/cpp/kdevcppsupport.desktop.cmake
index 59eaf5a..66b9318 100644
--- languages/cpp/kdevcppsupport.desktop.cmake
+++ languages/cpp/kdevcppsupport.desktop.cmake
@@ -125,7 +125,7 @@ X-KDevelop-Version=@KDEV_PLUGIN_VERSION@
 X-KDevelop-Language=C++
 X-KDevelop-Args=CPP
 X-KDevelop-Interfaces=ILanguageSupport
-X-KDevelop-SupportedMimeTypes=text/x-chdr,text/x-c++hdr,text/x-csrc,text/x-c++src
+X-KDevelop-SupportedMimeTypes=text/x-chdr,text/x-c++hdr,text/x-objchdr,text/x-csrc,text/x-c++src,text/x-objcsrc,text/x-objc++src
 X-KDevelop-IRequired=org.kdevelop.IDefinesAndIncludesManager
 X-KDevelop-Mode=NoGUI
 X-KDevelop-LoadMode=AlwaysOn
