commit d12037d0e0036255ebd7d5b6f95589b3f2830104
Author: René Bertin <rjvbertin@gmail.com>
Date:   Wed May 29 13:45:32 2024 +0200

    lamarck/patch-pextlib-no-sip.diff

diff --git a/src/pextlib1.0/sip_copy_proc.c b/src/pextlib1.0/sip_copy_proc.c
index 10aa5df86..994c625ac 100644
--- a/src/pextlib1.0/sip_copy_proc.c
+++ b/src/pextlib1.0/sip_copy_proc.c
@@ -335,6 +335,11 @@ resign_out:
 }
 
 static char *lazy_copy(const char *path, struct stat *in_st) {
+#ifndef SF_RESTRICTED
+    (void) path;
+    (void) in_st;
+    return NULL;
+#else
     char *retval = NULL;
     uid_t euid = geteuid();
     int outfd = -1;
@@ -529,6 +534,7 @@ lazy_copy_out:
         free(target_path);
     }
     return retval;
+#endif // SF_RESTRICTED
 }
 
 /**
