diff --git a/configure.ac b/configure.ac
index ab77f3746..3b8893a7f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -162,7 +162,7 @@ AC_PATH_PROG(MAN, [man])
 AC_PATH_PROG(MDFIND, [mdfind], [])
 AC_PATH_PROG(MDLS, [mdls], [])
 AC_PATH_PROG(MKBOM, [mkbom], [])
-AC_PATH_PROGS(MTREE, [mtree fmtree], [], [$PATH:/usr/sbin])
+AC_PATH_PROGS(MTREE, [mtree fmtree], [], [$PATH:/usr/sbin:/usr/local/bin])
 AC_PATH_PROG(OPEN, [open], [])
 AC_PATH_PROG(OPENSSL, [openssl], [])
 AC_PATH_PROG(PATCH, [patch], [])
@@ -181,6 +181,9 @@ AC_PATH_PROG(XZ, [xz], [])
 AC_PATH_PROG(ZIP, [zip])
 
 if test "x$MTREE" = "x"; then
+    AC_MSG_RESULT([The mtree or fmtree command is required. Use --with-mtree=/path/to/mtree \
+to indicate a non-standard location. Or install it, on Debian/Ubuntu via package freebsd-buildutils \
+or else from e.g. https://github.com/archiecobbs/mtree-port?tab=readme-ov-file])
     AC_MSG_ERROR([mtree not found])
 fi
 
