--- orig.CMakeLists.txt	2016-04-10 14:24:18.000000000 +0200
+++ CMakeLists.txt	2016-04-10 14:26:28.000000000 +0200
@@ -22,7 +22,7 @@
 find_package(SharedDesktopOntologies 0.10.51)
 set_package_properties(SharedDesktopOntologies PROPERTIES DESCRIPTION "Shared Desktop Ontologies" URL "http://oscaf.sourceforge.net" TYPE REQUIRED PURPOSE "Ontologies necessary for the Nepomuk semantic desktop")
 
-find_package(Soprano 2.9.3 COMPONENTS PLUGIN_RAPTORPARSER PLUGIN_REDLANDBACKEND PLUGIN_VIRTUOSOBACKEND)
+find_package(Soprano 2.9.3 COMPONENTS PLUGIN_RAPTORPARSER PLUGIN_REDLANDBACKEND)
 set_package_properties(Soprano PROPERTIES DESCRIPTION "Qt-based RDF storage and parsing solution" URL "http://soprano.sourceforge.net" TYPE REQUIRED PURPOSE "Required for everything (storage and general data management)")
 
 # Poppler
