diff --git a/src/Profile.cpp b/src/Profile.cpp
index 4d127f6a470ee4daaaffd73ef3930dfe675d2d5e..ed25367088d11d813a0a4379607eeb3c0b5d3962 100644
--- a/src/Profile.cpp
+++ b/src/Profile.cpp
@@ -189,7 +189,7 @@ void Profile::useFallback()
     setProperty(HistorySize, 1000);
     setProperty(ScrollBarPosition, Enum::ScrollBarRight);
     setProperty(ScrollFullPage, false);
-    setProperty(HighlightScrolledLines, true);
+    setProperty(HighlightScrolledLines, false);
 
     setProperty(FlowControlEnabled, true);
     setProperty(UrlHintsModifiers, 0);
