diff --git desktop/sessionui.rc desktop/sessionui.rc
index 287e1dea..5d78c390 100644
--- desktop/sessionui.rc
+++ desktop/sessionui.rc
@@ -33,14 +33,12 @@
             <Separator group="session-view-operations"/>
             <Action name="enlarge-font" group="session-view-operations"/>
             <Action name="shrink-font" group="session-view-operations"/>
-            <Action name="set-encoding" group="session-view-operations"/>
             <Separator group="session-view-operations"/>
             <Action name="clear-history" group="session-view-operations"/>
             <Action name="clear-history-and-reset" group="session-view-operations"/>
         </Menu>
         <Menu name="settings">
             <Action name="edit-current-profile" group="session-settings"/>
-            <Action name="switch-profile" group="session-settings"/>
         </Menu>
     </MenuBar>
     <Menu name="session-popup-menu">
