diff --git src/TerminalDisplay.cpp src/TerminalDisplay.cpp
index af389664..aae29814 100644
--- src/TerminalDisplay.cpp
+++ src/TerminalDisplay.cpp
@@ -1535,6 +1535,10 @@ void TerminalDisplay::paintFilters(QPainter& painter)
     if (_filterUpdateRequired) {
         return;
     }
+    QList<Filter::HotSpot*> spots = _filterChain->hotSpots();
+    if (spots.isEmpty()) {
+        return;
+    }
 
     // get color of character under mouse and use it to draw
     // lines for filters
@@ -1550,7 +1554,6 @@ void TerminalDisplay::paintFilters(QPainter& painter)
     // iterate over hotspots identified by the display's currently active filters
     // and draw appropriate visuals to indicate the presence of the hotspot
 
-    QList<Filter::HotSpot*> spots = _filterChain->hotSpots();
     int urlNumber, urlNumInc;
     if (_reverseUrlHints) {
         urlNumber = spots.size() + 1;
@@ -1654,7 +1657,7 @@ void TerminalDisplay::paintFilters(QPainter& painter)
                 const int baseline = r.bottom() - metrics.descent();
                 // find the position of the underline below that
                 const int underlinePos = baseline + metrics.underlinePos();
-                if (_showUrlHint || region.contains(mapFromGlobal(QCursor::pos()))) {
+                if (_showUrlHint || region.contains(cursorPos)) {
                     painter.drawLine(r.left() , underlinePos ,
                                      r.right() , underlinePos);
                 }
