diff --git a/src/widgets/TerminalDisplay.cpp b/src/widgets/TerminalDisplay.cpp
index af389664..aae29814 100644
--- a/src/widgets/TerminalDisplay.cpp
+++ b/src/widgets/TerminalDisplay.cpp
@@ -1408,6 +1408,10 @@ void TerminalDisplay::paintFilters(QPainter& painter)
     if (_filterUpdateRequired) {
         return;
     }
+    const auto spots = _filterChain->hotSpots();
+    if (spots.isEmpty()) {
+        return;
+    }
 
     // get color of character under mouse and use it to draw
     // lines for filters
@@ -1449,7 +1453,6 @@ void TerminalDisplay::paintFilters(QPainter& painter)
     // iterate over hotspots identified by the display's currently active filters
     // and draw appropriate visuals to indicate the presence of the hotspot
 
-    const auto spots = _filterChain->hotSpots();
     int urlNumber = 0;
     int urlNumInc;
     if (_reverseUrlHints) {
@@ -1551,7 +1554,7 @@ void TerminalDisplay::paintFilters(QPainter& painter)
                         (line + 1)*_fontHeight + _contentRect.top() - 1);
 
             // Underline link hotspots
-            const bool hasMouse = region.contains(mapFromGlobal(QCursor::pos()));
+            const bool hasMouse = region.contains(cursorPos);
             if ((spot->type() == HotSpot::Link && _showUrlHint) || hasMouse) {
                 QFontMetrics metrics(font());
 
