diff --git src/ViewSplitter.cpp src/ViewSplitter.cpp
index 43cbe923133a6d9728e9d4be42b8246e48378508..5329e9ffeb47a9a26597761090fa19816f94e30e 100644
--- src/ViewSplitter.cpp
+++ src/ViewSplitter.cpp
@@ -38,6 +38,17 @@ ViewSplitter::ViewSplitter(QWidget *parent) :
 {
 }
 
+ViewSplitter::~ViewSplitter()
+{
+    // unregister all containers now, to prevent crashing when they
+    // are deleted later on (being our children) and then send us the
+    // destroyed signal to be unregistered. By then we can no longer
+    // be certain if _containers is still valid.
+    for (const auto container : _containers) {
+        unregisterContainer(container);
+    }
+}
+
 void ViewSplitter::childEmpty(ViewSplitter *splitter)
 {
     delete splitter;
@@ -89,7 +100,7 @@ ViewSplitter *ViewSplitter::activeSplitter()
 void ViewSplitter::registerContainer(TabbedViewContainer *container)
 {
     _containers << container;
-    connect(container, SIGNAL(empty(TabbedViewContainer*)), this, SLOT(containerEmpty(TabbedViewContainer*)));
+    connect(container , &Konsole::TabbedViewContainer::empty , this , &Konsole::ViewSplitter::containerEmpty);
 }
 
 void ViewSplitter::unregisterContainer(TabbedViewContainer *container)
diff --git src/ViewSplitter.h src/ViewSplitter.h
index 70ca0040e39feaf97b1da51d887b1774c596290b..0968b3bc9b91e35ba6ef41cd172eee83c929dab4 100644
--- src/ViewSplitter.h
+++ src/ViewSplitter.h
@@ -49,6 +49,7 @@ class ViewSplitter : public QSplitter
 
 public:
     explicit ViewSplitter(QWidget *parent = nullptr);
+    ~ViewSplitter() Q_DECL_OVERRIDE;
 
     /**
      * Locates the child ViewSplitter widget which currently has the focus
