diff --git a/src/profile/Profile.cpp b/src/profile/Profile.cpp
index 0ea2833daa9260206c615a220912c15e08be7a14..af10f74cccaa48bab354e81469a8e2db3d7a2ff3 100644
--- a/src/profile/Profile.cpp
+++ b/src/profile/Profile.cpp
@@ -195,7 +195,7 @@ void Profile::useFallback()
     setProperty(HistorySize, 1000);
     setProperty(ScrollBarPosition, Enum::ScrollBarRight);
     setProperty(ScrollFullPage, false);
-    setProperty(HighlightScrolledLines, true);
+    setProperty(HighlightScrolledLines, false);
 
     setProperty(FlowControlEnabled, true);
     setProperty(UrlHintsModifiers, 0);
