diff --git a/src/main.cpp b/src/main.cpp
index b84ff736..e771ccb4 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -97,6 +97,7 @@ extern "C" int Q_DECL_EXPORT kdemain(int argc, char *argv[])
     }
 
 #if defined(Q_OS_LINUX) && (QT_VERSION < QT_VERSION_CHECK(5, 11, 2))
+    // RJVB - keep this until we update port:qt5-kde
     // Workaround for https://bugreports.qt.io/browse/QTBUG-48344
     // See also https://bugs.kde.org/show_bug.cgi?id=230184
     // The Qt glib event loop doesn't let timers deliver events if there are a
@@ -106,7 +107,13 @@ extern "C" int Q_DECL_EXPORT kdemain(int argc, char *argv[])
 #endif
 
     auto app = new QApplication(argc, argv);
-    app->setStyle(new MenuStyle());
+    const QByteArray xdgCurrentDesktop = qgetenv("XDG_CURRENT_DESKTOP");
+    if (xdgCurrentDesktop.isNull() || xdgCurrentDesktop.toLower() != "kde") {
+        // Using the QProxyStyle can cause issues with QtCurve so don't do
+        // it if it's not necessary (see #0107e68e2)
+        qDebug() << "XDG_CURRENT_DESKTOP=" << xdgCurrentDesktop << "; setting QProxyStyle";
+        app->setStyle(new MenuStyle());
+    }
 
 #if defined(Q_OS_LINUX) && (QT_VERSION < QT_VERSION_CHECK(5, 11, 2))
     if (qtUseGLibOld.isNull()) {
