diff --git src/kcolorscheme.h.timestap src/kcolorscheme.h
index 339cb75..3bf6f4a 100644
--- src/kcolorscheme.h.timestap
+++ src/kcolorscheme.h
@@ -324,7 +324,7 @@ public:
      * of states (e.g. windows that cannot be inactive), consider using a
      * ::KStatefulBrush instead.
      */
-    explicit KColorScheme(QPalette::ColorGroup, ColorSet = View, KSharedConfigPtr = KSharedConfigPtr());
+    explicit KColorScheme(QPalette::ColorGroup = QPalette::Normal, ColorSet = View, KSharedConfigPtr = KSharedConfigPtr());
 
     /**
      * Retrieve the requested background brush.
