diff --git a/src/integration/propertiesplugin/gdrivepropertiesplugin.cpp b/src/integration/propertiesplugin/gdrivepropertiesplugin.cpp
index 75bbc20241e49784004d5001d44aef7bc1146df3..efc9237c8002f62345b8f10f30e209df449cc6cd 100644
--- a/src/integration/propertiesplugin/gdrivepropertiesplugin.cpp
+++ b/src/integration/propertiesplugin/gdrivepropertiesplugin.cpp
@@ -86,7 +86,7 @@ void GDrivePropertiesPlugin::statJobFinished(KJob *job)
 {
     KIO::StatJob *statJob = qobject_cast<KIO::StatJob*>(job);
     if (!statJob || statJob->error()) {
-        qCDebug(GDRIVE) << "Failed stat()ing" << statJob->url() << statJob->errorString();
+        qCWarning(GDRIVE) << "Failed stat()ing" << statJob->url() << statJob->errorString();
         qCDebug(GDRIVE) << "Not showing Google Drive properties tab";
         return;
     }
