diff --git a/CMakeLists.txt b/CMakeLists.txt
index 51234128d1865f61a16471e9c7d1400e264f20fa..a019cc30f0b234704eaa522ed052efd395cff18c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -8,7 +8,7 @@ set (RELEASE_SERVICE_VERSION "${RELEASE_SERVICE_VERSION_MAJOR}.${RELEASE_SERVICE
 project(kio-gdrive VERSION ${RELEASE_SERVICE_VERSION})
 
 set(QT_MIN_VERSION 5.2.0)
-set(KF5_MIN_VERSION 5.71.0)
+set(KF5_MIN_VERSION 5.60.0)
 set(KGAPI_MIN_VERSION 5.11.41)
 set(KACCOUNTS_MIN_VERSION 20.03.80)
 set(QTKEYCHAIN_MIN_VERSION 0.6.0)
diff --git a/src/kio_gdrive.cpp b/src/kio_gdrive.cpp
index 514d7470f95e6fdd17606bc9e82328b84b5640da..0725f75dcde1b65e88eaf42cb6032bdcb481da4a 100644
--- a/src/kio_gdrive.cpp
+++ b/src/kio_gdrive.cpp
@@ -47,6 +47,8 @@
 #include <KIO/Job>
 #include <KLocalizedString>
 
+#include <KPim/kgapi_version.h>
+
 using namespace KGAPI2;
 using namespace Drive;
 
@@ -226,7 +228,9 @@ KIO::UDSEntry KIOGDrive::fileToUDSEntry(const FilePtr &origFile, const QString &
 
     entry.fastInsert(GDriveUDSEntryExtras::Id, file->id());
     entry.fastInsert(GDriveUDSEntryExtras::Url, file->alternateLink().toString());
+#if KGAPI_VERSION >= QT_VERSION_CHECK(5,15,80)
     entry.fastInsert(GDriveUDSEntryExtras::Version, QString::number(file->version()));
+#endif
     entry.fastInsert(GDriveUDSEntryExtras::Md5, file->md5Checksum());
     entry.fastInsert(GDriveUDSEntryExtras::LastModifyingUser, file->lastModifyingUserName());
     entry.fastInsert(GDriveUDSEntryExtras::Owners, file->ownerNames().join(QStringLiteral(", ")));
