--- cmake/orig.ranges_env.cmake	2020-08-07 01:05:10.000000000 +0200
+++ cmake/ranges_env.cmake	2020-10-18 21:40:58.000000000 +0200
@@ -74,8 +74,8 @@
     message(STATUS "[range-v3]: build type is release.")
   endif()
 else()
-  message(WARNING "[range-v3 warning]: unknown build type, defaults to release!")
-  set(CMAKE_BUILD_TYPE "Release")
+  message(WARNING "[range-v3 warning]: unknown build type, assuming a release build!")
+  #set(CMAKE_BUILD_TYPE "Release")
   set(RANGES_RELEASE_BUILD TRUE)
 endif()
 
