--- src/orig.kcolorbutton.cpp	2017-06-03 12:22:27.000000000 +0200
+++ src/kcolorbutton.cpp	2017-06-20 16:35:07.000000000 +0200
@@ -310,11 +310,12 @@
     }
 
     dialog = new QColorDialog(q);
-    dialog->setCurrentColor(q->color());
+    dialog->setOption(QColorDialog::DontUseNativeDialog, true);
     dialog->setOption(QColorDialog::ShowAlphaChannel, m_alphaChannel);
     dialog->setAttribute(Qt::WA_DeleteOnClose);
     connect(dialog, SIGNAL(accepted()), q, SLOT(_k_colorChosen()));
     dialogPtr = dialog;
+    dialog->setCurrentColor(q->color());
     dialog->show();
 }
 
