diff --git autotests/kcharselect_unittest.cpp autotests/kcharselect_unittest.cpp
index c74fcba..427d01c 100644
--- autotests/kcharselect_unittest.cpp
+++ autotests/kcharselect_unittest.cpp
@@ -34,8 +34,10 @@ private Q_SLOTS:
     void createInstance()
     {
         KCharSelect selector(nullptr, nullptr);
+#if QT_VERSION >= QT_VERSION_CHECK(5, 11, 0)
         QCOMPARE(selector.displayedCodePoints().count(), 128);
         QCOMPARE(selector.currentCodePoint(), 0);
+#endif
     }
 
     void changeBlock()
