=== modified file 'gui/mainui.py'
--- gui/mainui.pyold	2016-10-26 07:38:00 +0000
+++ gui/mainui.py	2017-02-22 15:44:58 +0000
@@ -41,6 +41,10 @@
         self.core = core
         self.resize(size[0], size[1])
         self.setWindowTitle("Qarte-%s" % VERSION_STR)
+        appIcon = QIcon()
+        appIcon.addPixmap(QPixmap("medias/qarte_logo.png"))
+        self.setWindowIcon(appIcon)
+        QCoreApplication.instance().setWindowIcon(appIcon)
         self.centralwidget = QWidget(self)
         self.gridLayout = QGridLayout(self.centralwidget)
         self.stackedWidget = QStackedWidget(self.centralwidget)

