diff --git a/CMakeLists.txt b/CMakeLists.txt
index e8dd723..7b1964a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -44,6 +44,10 @@ option(QTC_QT4_STYLE_SUPPORT
 option(QTC_GTK2_MODIFY_MOZILLA "Modify Mozilla Gtk2 with js and css." Off)
 
 ## Common settings.
+# support an infix argument to have different qtcurve-utils libraries, e.g. for
+# GTk2, Qt4 and Qt5
+qtc_option(QTC_UTILSLIB_INFIX "an infix argument to have different qtcurve-utils libraries" "")
+
 # Only try to figure out what kde prefix is if not specified at command line.
 if(NOT DEFINED QTC_KDE4_PREFIX)
   find_package(KDE4)
diff --git a/lib/utils/CMakeLists.txt b/lib/utils/CMakeLists.txt
index 0dd71c1..1f41724 100644
--- a/lib/utils/CMakeLists.txt
+++ b/lib/utils/CMakeLists.txt
@@ -42,6 +42,7 @@ set_target_properties(qtcurve-utils PROPERTIES
   VERSION 1.0
   SOVERSION 1
   COMPILE_FLAGS "-fvisibility=hidden"
-  LIBRARY_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}")
+  LIBRARY_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}"
+  OUTPUT_NAME "qtcurve-utils${QTC_UTILSLIB_INFIX}")
 install(TARGETS qtcurve-utils LIBRARY
   DESTINATION "${LIB_INSTALL_DIR}")
