diff --git a/qt5/style/qtcurve.h b/qt5/style/qtcurve.h
index 56960a59a300ef9a69037b3dd5d1406c1ddd457c..f5c3f6d1376a88d075f4dd1b53300ffa1caa008a 100644
--- a/qt5/style/qtcurve.h
+++ b/qt5/style/qtcurve.h
@@ -564,18 +564,18 @@ private:
     mutable QList<int> m_mdiButtons[2]; // 0=left, 1=right
     mutable int m_titlebarHeight;
 
-    QDBusInterface *m_dBus;
     ShadowHelper *m_shadowHelper;
     mutable QScrollBar *m_sViewSBar;
     mutable QMap<QWidget*, QSet<QWidget*> > m_sViewContainers;
     WindowManager *m_windowManager;
     BlurHelper *m_blurHelper;
     ShortcutHandler *m_shortcutHandler;
-    bool m_dbusConnected;
 #ifdef QTC_QT5_ENABLE_KDE
     KSharedConfigPtr m_configFile;
     KSharedConfigPtr m_kdeGlobals;
 #endif
+    class Private;
+    Private *d;
 protected:
     StylePlugin *m_plugin;
     friend class StylePlugin;
