diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8f308d177b14829acd05311dffb5385a374a41e9..60145549d35b0a4497294f330a1bb16cf4403190 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -228,7 +228,10 @@ if(NOT DEFINED LIB_INSTALL_DIR)
 endif()
 
 cmake_c_include_fix_path()
-add_subdirectory(lib)
+if(ENABLE_QT4 OR ENABLE_QT5 OR ENABLE_GTK2)
+    add_subdirectory(lib)
+    add_subdirectory(tools)
+endif()
 if(ENABLE_QT4)
     add_subdirectory(qt4)
 endif()
@@ -238,10 +241,17 @@ endif()
 if(ENABLE_GTK2)
     add_subdirectory(gtk2)
 endif()
-add_subdirectory(tools)
 if(BUILD_TESTING)
   enable_testing()
   add_subdirectory(test)
 endif()
 
+if(QTC_INSTALL_PO)
+    find_package(ECM 1.2.0 REQUIRED NO_MODULE)
+    set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${ECM_MODULE_PATH})
+
+    find_package(KF5 REQUIRED COMPONENTS
+      I18n)
+    ki18n_install(po)
+endif()
 feature_summary(WHAT ALL FATAL_ON_MISSING_REQUIRED_PACKAGES)
