diff --git gtk2/common/config_file.c gtk2/common/config_file.c
index 7c19596..00b07dc 100644
--- gtk2/common/config_file.c
+++ gtk2/common/config_file.c
@@ -1692,7 +1692,11 @@ bool qtcReadConfig(const char *file, Options *opts, Options *defOpts)
 
 static const char * getSystemConfigFile()
 {
-    static const char * constFiles[]={ /*"/etc/qt4/"OLD_CONFIG_FILE, "/etc/qt3/"OLD_CONFIG_FILE, "/etc/qt/"OLD_CONFIG_FILE,*/ "/etc/"OLD_CONFIG_FILE, NULL };
+    static const char * constFiles[]={
+        "/etc/" OLD_CONFIG_FILE,
+        "@@PREFIX@@/etc/xdg/qtcurve/" CONFIG_FILE,
+        NULL
+    };
 
     for (int i = 0;constFiles[i];++i) {
         if (qtcIsRegFile(constFiles[i])) {

