diff --git a/hw/xquartz/GL/visualConfigs.c b/hw/xquartz/GL/visualConfigs.c
index d810e2dfee800e1013fef8b5cc77229da9ac2107..224de8b023d2136938aabd1e24021f7fbfca545b 100644
--- a/hw/xquartz/GL/visualConfigs.c
+++ b/hw/xquartz/GL/visualConfigs.c
@@ -121,6 +121,7 @@ __GLXconfig *__glXAquaCreateVisualConfigs(int *numConfigsPtr, int screenNumber)
 
     l = NULL;
     c = visualConfigs; /* current buffer */
+    int confCount = 0;
     for(conf = caps.configurations; conf; conf = conf->next) {
         for(stereo = 0; stereo < (conf->stereo ? 2 : 1); ++stereo) {
             for(aux = 0; aux < (conf->aux_buffers ? 2 : 1); ++aux) {
@@ -133,6 +134,7 @@ __GLXconfig *__glXAquaCreateVisualConfigs(int *numConfigsPtr, int screenNumber)
                                 for(depth = 0; depth < conf->total_depth_buffer_depths; ++depth) {
                                     for(msample = 0; msample < (conf->multisample_buffers + 1); ++msample) {
 
+                                        ++confCount;
                                         // Global
                                         c->visualID = -1;
                                         c->visualType = GLX_TRUE_COLOR;
@@ -272,9 +274,13 @@ __GLXconfig *__glXAquaCreateVisualConfigs(int *numConfigsPtr, int screenNumber)
         }
     }
 
-    free(c);
+    free(l->next);
     l->next = NULL;
 
+    if (confCount != numConfigs) {
+        FatalError("numConfigs calculation error in setVisualConfigs!  numConfigs is %d count is %d\n", numConfigs, confCount);
+    }
+
     freeGlCapabilities(&caps);
     return visualConfigs;
 }
