diff --git a/hw/xquartz/NSUserDefaults+XQuartzDefaults.m b/hw/xquartz/NSUserDefaults+XQuartzDefaults.m
index bd5dd2305..4c7f3c441 100644
--- a/hw/xquartz/NSUserDefaults+XQuartzDefaults.m
+++ b/hw/xquartz/NSUserDefaults+XQuartzDefaults.m
@@ -9,6 +9,10 @@
 #import "NSUserDefaults+XQuartzDefaults.h"
 #import <dispatch/dispatch.h>
 
+#ifndef BUNDLE_ID_PREFIX
+#define BUNDLE_ID_PREFIX "org.macports"
+#endif
+
 NSString * const XQuartzPrefKeyAppsMenu = @"apps_menu";
 NSString * const XQuartzPrefKeyFakeButtons = @"enable_fake_buttons";
 NSString * const XQuartzPrefKeyFakeButton2 = @"fake_button2";
