diff --git a/hw/xquartz/quartz.c b/hw/xquartz/quartz.c
index 980aa4a36..ca49258e7 100644
--- a/hw/xquartz/quartz.c
+++ b/hw/xquartz/quartz.c
@@ -79,7 +79,8 @@ extern OSErr UpdateSystemActivity(UInt8 activity);
 #endif
 
 DevPrivateKeyRec quartzScreenKeyRec;
-int aquaMenuBarHeight = 0;
+// RJVB: initialise at a common value
+int aquaMenuBarHeight = 24;
 QuartzModeProcsPtr quartzProcs = NULL;
 const char             *quartzOpenGLBundle = NULL;
 
diff --git a/hw/xquartz/X11Application.m b/hw/xquartz/X11Application.m
index 821e1c5a1..0d5376c80 100644
--- a/hw/xquartz/X11Application.m
+++ b/hw/xquartz/X11Application.m
@@ -1219,6 +1219,7 @@ X11ApplicationMain(int argc, char **argv, char **envp)
 
     /* Calculate the height of the menubar so we can avoid it. */
     aquaMenuBarHeight = [[NSApp mainMenu] menuBarHeight];
+    NSLog(@"aquaMenuBarHeight according to [NSApp mainMenu]: %d", aquaMenuBarHeight);
 #if ! __LP64__
     if (!aquaMenuBarHeight) {
         aquaMenuBarHeight = [NSMenuView menuBarHeight];
@@ -1227,6 +1228,8 @@ X11ApplicationMain(int argc, char **argv, char **envp)
     if (!aquaMenuBarHeight) {
         NSScreen* primaryScreen = [[NSScreen screens] objectAtIndex:0];
         aquaMenuBarHeight = NSHeight([primaryScreen frame]) - NSMaxY([primaryScreen visibleFrame]);
+        NSLog(@"aquaMenuBarHeight calculated for primaryScreen %@ with height=%g and visibleFrameHeight=%g : %d",
+            primaryScreen, NSHeight([primaryScreen frame]), NSMaxY([primaryScreen visibleFrame]), aquaMenuBarHeight);
     }
 
 #ifdef HAVE_LIBDISPATCH
