diff --git a/hw/kdrive/ephyr/ephyr.c b/hw/kdrive/ephyr/ephyr.c
index 4f1f19860f77569e4da51cf248362ead5e02f713..4f651fd9d9d5427a27f4557f130151b3fb20dc78 100644
--- a/hw/kdrive/ephyr/ephyr.c
+++ b/hw/kdrive/ephyr/ephyr.c
@@ -61,6 +61,13 @@ Bool EphyrWantGrayScale = 0;
 Bool EphyrWantResize = 0;
 Bool EphyrWantNoHostGrab = 0;
 
+#ifdef __APPLE__
+// used in librootless
+int darwinMainScreenX = 0;
+int darwinMainScreenY = 0;
+Bool no_configure_window = FALSE;
+#endif
+
 Bool
 ephyrInitialize(KdCardInfo * card, EphyrPriv * priv)
 {
