diff --git a/glx/glxscreens.c b/glx/glxscreens.c
index 596d972e06a302562bb60563a3801d51ef26a13d..1db7650de67d924ddd550bf312f4facfeda10faa 100644
--- a/glx/glxscreens.c
+++ b/glx/glxscreens.c
@@ -437,6 +437,6 @@ __glXScreenDestroy(__GLXscreen * screen)
 
     for (config = screen->fbconfigs; config != NULL; config = next) {
         next = config->next;
-        free(config);
+//         free(config);
     }
 }
