diff --git a/meson.build b/meson.build
index ea45ca3d3..71e416399 100644
--- a/meson.build
+++ b/meson.build
@@ -373,9 +373,9 @@ else
 endif
 
 build_dpms = get_option('dpms')
-if build_xquartz
-    build_dpms = false
-endif
+# if build_xquartz
+#     build_dpms = false
+# endif
 
 build_xf86bigfont = get_option('xf86bigfont')
 build_screensaver = get_option('screensaver')
diff --git a/miext/sync/meson.build b/miext/sync/meson.build
index 8b7ee4dba..bb3eef0f1 100644
--- a/miext/sync/meson.build
+++ b/miext/sync/meson.build
@@ -10,7 +10,7 @@ hdrs_miext_sync = [
     'misyncstr.h',
 ]
 
-if build_dri3
+if build_dri3 or build_glamor
     srcs_miext_sync += 'misyncshm.c'
 endif
 
