diff --git a/hw/xquartz/quartz.c b/hw/xquartz/quartz.c
index 980aa4a36..ca49258e7 100644
--- a/hw/xquartz/quartz.c
+++ b/hw/xquartz/quartz.c
@@ -79,7 +79,8 @@ extern OSErr UpdateSystemActivity(UInt8 activity);
 #endif
 
 DevPrivateKeyRec quartzScreenKeyRec;
-int aquaMenuBarHeight = 0;
+// RJVB: initialise at a common value
+int aquaMenuBarHeight = 24;
 QuartzModeProcsPtr quartzProcs = NULL;
 const char             *quartzOpenGLBundle = NULL;
 
diff --git a/hw/xquartz/X11Application.m b/hw/xquartz/X11Application.m
index ab126f16416deb05ec0b746e1e951f184d77a758..bd340f73ac22c32fc3ce72545dcf64217ab023cf 100644
--- a/hw/xquartz/X11Application.m
+++ b/hw/xquartz/X11Application.m
@@ -775,9 +775,12 @@ X11ApplicationMain(int argc, char **argv, char **envp)
 
         /* Calculate the height of the menubar so we can avoid it. */
         aquaMenuBarHeight = NSApp.mainMenu.menuBarHeight;
+        NSLog(@"aquaMenuBarHeight according to [NSApp mainMenu]: %d", aquaMenuBarHeight);
         if (!aquaMenuBarHeight) {
             NSScreen* primaryScreen = NSScreen.screens[0];
             aquaMenuBarHeight = NSHeight(primaryScreen.frame) - NSMaxY(primaryScreen.visibleFrame);
+            NSLog(@"aquaMenuBarHeight calculated for primaryScreen %@ with height=%g and visibleFrameHeight=%g : %d",
+                primaryScreen, NSHeight([primaryScreen frame]), NSMaxY([primaryScreen visibleFrame]), aquaMenuBarHeight);
         }
 
         eventTranslationQueue = dispatch_queue_create(BUNDLE_ID_PREFIX ".X11.NSEventsToX11EventsQueue", NULL);
