diff --git a/hw/xquartz/xpr/xpr.h b/hw/xquartz/xpr/xpr.h
index fc614ebc3..b66a1c248 100644
--- a/hw/xquartz/xpr/xpr.h
+++ b/hw/xquartz/xpr/xpr.h
@@ -38,6 +38,7 @@
 
 #undef DEBUG_LOG
 #define DEBUG_LOG(msg, args ...) ASL_LOG(ASL_LEVEL_DEBUG, "xpr", msg, ## args)
+#define  WARN_LOG(msg, args ...) ASL_LOG(ASL_LEVEL_WARNING, "xpr", msg, ## args)
 
 Bool
 QuartzModeBundleInit(void);
diff --git a/hw/xquartz/xpr/xprScreen.c b/hw/xquartz/xpr/xprScreen.c
index 30f2218b6..dbfe55fb7 100644
--- a/hw/xquartz/xpr/xprScreen.c
+++ b/hw/xquartz/xpr/xprScreen.c
@@ -167,7 +167,7 @@ displayScreenBounds(CGDirectDisplayID id)
 
     frame = CGDisplayBounds(id);
 
-    DEBUG_LOG("    %dx%d @ (%d,%d).\n",
+    WARN_LOG("    %dx%d @ (%d,%d).\n",
               (int)frame.size.width, (int)frame.size.height,
               (int)frame.origin.x, (int)frame.origin.y);
 
@@ -192,9 +192,10 @@ displayScreenBounds(CGDirectDisplayID id)
         (spacePerDisplay || (frame.origin.x == 0 && frame.origin.y == 0))) {
         frame.origin.y += aquaMenuBarHeight;
         frame.size.height -= aquaMenuBarHeight;
+	   WARN_LOG("    aquaMenuBarHeight compensation\n");
     }
 
-    DEBUG_LOG("    %dx%d @ (%d,%d).\n",
+    WARN_LOG("    %dx%d @ (%d,%d).\n",
               (int)frame.size.width, (int)frame.size.height,
               (int)frame.origin.x, (int)frame.origin.y);
 

