--- Source/bmalloc/bmalloc/ProcessCheck.h.orig	2023-02-20 09:22:22
+++ Source/bmalloc/bmalloc/ProcessCheck.h	2024-12-09 00:25:19
@@ -40,7 +40,11 @@
 #endif
 
 #if BOS(DARWIN)
-bool shouldAllowMiniMode();
+// for some reason this function isnt available for arm64
+// - but all it does is disable MiniMode for webkit processes
+// - so just directly disable here
+//bool shouldAllowMiniMode();
+inline bool shouldAllowMiniMode() { return false; }
 #else
 inline bool shouldAllowMiniMode() { return true; }
 #endif
