--- Source/WebKit/Platform/SharedMemory.h.orig	2023-02-20 09:22:20
+++ Source/WebKit/Platform/SharedMemory.h	2024-12-23 14:33:13
@@ -37,7 +37,7 @@
 #include <wtf/win/Win32Handle.h>
 #endif
 
-#if OS(DARWIN)
+#if PLATFORM(COCOA)
 #include <wtf/MachSendRight.h>
 #endif
 
@@ -85,7 +85,7 @@
     private:
 #if USE(UNIX_DOMAIN_SOCKETS)
         mutable UnixFileDescriptor m_handle;
-#elif OS(DARWIN)
+#elif PLATFORM(COCOA)
         mutable MachSendRight m_handle;
 #elif OS(WINDOWS)
         mutable Win32Handle m_handle;
@@ -103,7 +103,7 @@
     static RefPtr<SharedMemory> map(const Handle&, Protection);
 #if USE(UNIX_DOMAIN_SOCKETS)
     static RefPtr<SharedMemory> wrapMap(void*, size_t, int fileDescriptor);
-#elif OS(DARWIN)
+#elif PLATFORM(COCOA)
     static RefPtr<SharedMemory> wrapMap(void*, size_t, Protection);
 #endif
 #if OS(WINDOWS)
@@ -132,7 +132,7 @@
     Ref<WebCore::SharedBuffer> createSharedBuffer(size_t) const;
 
 private:
-#if OS(DARWIN)
+#if PLATFORM(COCOA)
     WTF::MachSendRight createSendRight(Protection) const;
 #endif
 
@@ -145,7 +145,7 @@
 #if USE(UNIX_DOMAIN_SOCKETS)
     UnixFileDescriptor m_fileDescriptor;
     bool m_isWrappingMap { false };
-#elif OS(DARWIN)
+#elif PLATFORM(COCOA)
     MachSendRight m_sendRight;
 #elif OS(WINDOWS)
     Win32Handle m_handle;
