diff --git Tools/MiniBrowser/gtk/orig.CMakeLists.txt Tools/MiniBrowser/gtk/CMakeLists.txt
index 2f4f61f..9498326 100644
--- Tools/MiniBrowser/gtk/orig.CMakeLists.txt
+++ Tools/MiniBrowser/gtk/CMakeLists.txt
@@ -26,6 +26,7 @@ set(MiniBrowser_PRIVATE_INCLUDE_DIRECTORIES
     ${WebKitGtk_FRAMEWORK_HEADERS_DIR}
     ${WebKitGtk_FRAMEWORK_HEADERS_DIR}/webkitgtk-${WEBKITGTK_API_VERSION}
     ${CMAKE_SOURCE_DIR}/Source
+    ${JavaScriptCoreGLib_FRAMEWORK_HEADERS_DIR}
 )
 
 set(MiniBrowser_SYSTEM_INCLUDE_DIRECTORIES
@@ -33,6 +34,10 @@ set(MiniBrowser_SYSTEM_INCLUDE_DIRECTORIES
     ${LIBSOUP_INCLUDE_DIRS}
 )
 
+if ((NOT JavaScriptCore_LIBRARY_NAME) OR (JavaScriptCore_LIBRARY_NAME STREQUAL ""))
+	set(JavaScriptCore_LIBRARY_NAME "${CMAKE_BINARY_DIR}/lib/libjavascriptcoregtk-@API@.@SHEXT@")
+	message(WARNING "JavaScriptCore_LIBRARY_NAME not set")
+endif()
 set(MiniBrowser_PRIVATE_LIBRARIES
     ${GLIB_LIBRARIES}
     ${JavaScriptCore_LIBRARY_NAME}
