--- Source/WTF/wtf/PlatformUse.h.orig   2024-12-29 19:26:29
+++ Source/WTF/wtf/PlatformUse.h        2024-12-30 09:25:05
@@ -326,7 +326,7 @@
 #define USE_CTFONTHASTABLE 1
 #endif

-#if (PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 110000) \
+#if (OS(MACOS) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 110000) \
     || (PLATFORM(MACCATALYST) && __IPHONE_OS_VERSION_MIN_REQUIRED >= 140000) \
     || (PLATFORM(IOS) && PLATFORM(IOS_FAMILY_SIMULATOR) && __IPHONE_OS_VERSION_MIN_REQUIRED >= 140000) \
     || (PLATFORM(WATCHOS) && PLATFORM(IOS_FAMILY_SIMULATOR) && __WATCH_OS_VERSION_MIN_REQUIRED >= 70000) \
--- Source/JavaScriptCore/assembler/CPU.h.orig	2023-01-10 06:58:45
+++ Source/JavaScriptCore/assembler/CPU.h	2024-12-30 08:31:35
@@ -76,7 +76,7 @@
 constexpr bool isARM64E_FPAC() { return false; }
 #endif
 
-#if CPU(ARM64E) || OS(MAC_OS_X)
+#if CPU(ARM64E) || OS(MACOS)
 // ARM64E or all macOS ARM64 CPUs have LSE.
 constexpr bool isARM64_LSE() { return true; }
 #else
--- Source/JavaScriptCore/assembler/CPU.cpp.orig	2023-01-10 06:58:45
+++ Source/JavaScriptCore/assembler/CPU.cpp	2024-12-30 08:31:50
@@ -96,7 +96,7 @@
 
 #endif // #if (CPU(X86) || CPU(X86_64)) && OS(DARWIN)
 
-#if CPU(ARM64) && !(CPU(ARM64E) || OS(MAC_OS_X))
+#if CPU(ARM64) && !(CPU(ARM64E) || OS(MACOS))
 bool isARM64_LSE()
 {
 #if ENABLE(ASSEMBLER)
