--- Source/WebKit/UIProcess/glib/WebProcessPoolGLib.cpp.orig	2023-02-20 09:22:21
+++ Source/WebKit/UIProcess/glib/WebProcessPoolGLib.cpp	2025-01-07 15:54:35
@@ -120,7 +120,15 @@
     parameters.inspectorServerAddress = Inspector::RemoteInspector::inspectorServerAddress();
 #endif
 
+// so sharedDisplay calls createPlatformDisplay which only uses the following defines
+// PLATFORM(X11), PLATFORM(WAYLAND), USE(WPE_RENDERER) and PLATFORM(WIN)
+// none of which are defined if using quartz - so createPlatformDisplay crashes
+// as the ending RELEASE_ASSERT_NOT_REACHED is reached because no prior return hit
+// so going to bracket the following with these defines (not clear if that means ATSPI works or not)
+// could just disable ATSPI
+// (note that previous webkit versions did not have this code)
 #if USE(ATSPI)
+#if PLATFORM(X11) || PLATFORM(WAYLAND) || USE(WPE_RENDERER) || PLATFORM(WIN)
     parameters.accessibilityBusAddress = [this] {
         if (auto* address = getenv("WEBKIT_A11Y_BUS_ADDRESS"))
             return String::fromUTF8(address);
@@ -134,6 +142,7 @@
         return WebCore::PlatformDisplay::sharedDisplay().accessibilityBusAddress();
     }();
 #endif
+#endif
 
 #if PLATFORM(GTK)
     parameters.gtkSettings = GtkSettingsManager::singleton().settingsState();
