diff --git a/data/meson.build b/data/meson.build
index 026bd80..07d7071 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -12,26 +12,26 @@ desktop_in = configure_file(
   configuration: desktop_conf,
 )
 
-i18n.merge_file(
-  desktop,
-  type: 'desktop',
-  input: desktop_in,
-  output: '@BASENAME@',
-  po_dir: po_dir,
-  install: true,
-  install_dir: df_datadir / 'applications',
-)
+# i18n.merge_file(
+#   desktop,
+#   type: 'desktop',
+#   input: desktop_in,
+#   output: '@BASENAME@',
+#   po_dir: po_dir,
+#   install: true,
+#   install_dir: df_datadir / 'applications',
+# )
 
 appdata = df_namespace + '.appdata.xml'
 
-i18n.merge_file(
-  appdata,
-  input: appdata + '.in',
-  output: '@BASENAME@',
-  po_dir: po_dir,
-  install: true,
-  install_dir: df_datadir / 'metainfo',
-)
+# i18n.merge_file(
+#   appdata,
+#   input: appdata + '.in',
+#   output: '@BASENAME@',
+#   po_dir: po_dir,
+#   install: true,
+#   install_dir: df_datadir / 'metainfo',
+# )
 
 install_data(
   df_namespace + '.gschema.xml',
diff --git a/meson.build b/meson.build
index 4d3ee71..a44f869 100644
--- a/meson.build
+++ b/meson.build
@@ -51,7 +51,7 @@ foreach value: values
 endforeach
 
 # just check that they exist
-dependency('gtk+-3.0', version: '>= 3.9.4')
+dependency('gtk+-3.0', version: '>= 3.24.0')
 dependency('gobject-introspection-1.0', version: '>= 0.9.6')
 
 gio_schemasdir = dependency('gio-2.0').get_pkgconfig_variable(
