diff --git CMakeLists.txt CMakeLists.txt
index 1950fc4..456a75e 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -30,7 +30,7 @@ if (QT_VERSION_MAJOR EQUAL 6)
     set(QT_MODULES Core Gui Widgets Core5Compat)
     set(QT_LIBRARIES Qt6::Widgets Qt6::Core5Compat)
 elseif (QT_VERSION_MAJOR EQUAL 5)
-    set(QT_MIN_VERSION 5.11.0)
+    set(QT_MIN_VERSION 5.9.8)
     set(QT_MODULES Core Gui Widgets)
     set(QT_LIBRARIES Qt5::Widgets)
 elseif (QT_MAJOR_VERSION EQUAL 4)
