diff --git CMakeLists.txt CMakeLists.txt
index ddccd04..024d6df 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -98,7 +98,20 @@ SET(RESOURCE_FILES
 # in result SAMPLE_RC_SRCS variable will contain paths to files produced by rcc
 QT_ADD_RESOURCES(RC_SRCS ${RESOURCE_FILES})
 
-add_executable(qgit ${CPP_SOURCES} ${UIS_HDRS} ${RC_SRCS})
+if(APPLE)
+    set(MACOSX_BUNDLE_ICON_FILE qgit.icns)
+    set(ICON ${CMAKE_CURRENT_SOURCE_DIR}/src/resources/qgit.icns)
+    set_source_files_properties(${ICON} PROPERTIES MACOSX_PACKAGE_LOCATION "Resources")
+    add_executable(qgit MACOSX_BUNDLE ${CPP_SOURCES} ${UIS_HDRS} ${RC_SRCS} ${ICON})
+    set_target_properties(qgit PROPERTIES WIN32_EXECUTABLE TRUE MACOSX_BUNDLE TRUE)
+    set_target_properties(qgit PROPERTIES MACOSX_BUNDLE_ICON_FILE qgit.icns )
+    set_target_properties(qgit PROPERTIES MACOSX_BUNDLE_LONG_VERSION_STRING 2.6.120 )
+    set_target_properties(qgit PROPERTIES MACOSX_BUNDLE_SHORT_VERSION_STRING 2.6.120 )
+    set_target_properties(qgit PROPERTIES MACOSX_BUNDLE_BUNDLE_VERSION 2.6.120 )
+    set_target_properties(qgit PROPERTIES MACOSX_BUNDLE_BUNDLE_NAME "QGit")
+elseif()
+    add_executable(qgit ${CPP_SOURCES} ${UIS_HDRS} ${RC_SRCS})
+endif()
 target_link_libraries(qgit ${QT_LIBRARIES})
 
 
