diff --git CMakeLists.txt CMakeLists.txt
index 1950fc4..4996136 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -9,6 +9,7 @@ cmake_minimum_required(VERSION 3.10.2)
 
 project(qgit)
 include(GNUInstallDirs)
+include(FeatureSummary)
 
 # As moc files are generated in the binary dir, tell CMake
 # to always look for includes there:
@@ -30,7 +31,7 @@ if (QT_VERSION_MAJOR EQUAL 6)
     set(QT_MODULES Core Gui Widgets Core5Compat)
     set(QT_LIBRARIES Qt6::Widgets Qt6::Core5Compat)
 elseif (QT_VERSION_MAJOR EQUAL 5)
-    set(QT_MIN_VERSION 5.11.0)
+    set(QT_MIN_VERSION 5.6.0)
     set(QT_MODULES Core Gui Widgets)
     set(QT_LIBRARIES Qt5::Widgets)
 elseif (QT_MAJOR_VERSION EQUAL 4)
@@ -39,6 +40,7 @@ elseif (QT_MAJOR_VERSION EQUAL 4)
 endif()
 
 find_package(${QT} ${QT_MIN_VERSION} REQUIRED COMPONENTS ${QT_MODULES})
+message(STATUS "Qt version ${QT_VERSION}")
 
 if (QT_VERSION VERSION_LESS 5.15)
     if (QT_VERSION_MAJOR EQUAL 4)
@@ -131,6 +133,8 @@ if (UNIX)
     install(FILES qgit.appdata.xml DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/metainfo)
 endif()
 
+feature_summary(WHAT ALL   FATAL_ON_MISSING_REQUIRED_PACKAGES)
+
 # kate: indent-width 4; replace-tabs on;
 
 # notes:
diff --git src/git.cpp src/git.cpp
index f125bb8..93cd67a 100644
--- src/git.cpp
+++ src/git.cpp
@@ -1673,7 +1673,11 @@ const QString Git::getLocalDate(SCRef gitDate) {
         // cache miss
         if (localDate.isEmpty()) {
                 static QDateTime d;
+#if QT_VERSION >= QT_VERSION_CHECK(5, 8, 0)
                 d.setSecsSinceEpoch(gitDate.toUInt());
+#else
+                d.setMSecsSinceEpoch(gitDate.toUInt() * 1000);
+#endif
                 localDate = QLocale::system().toString(d, QLocale::ShortFormat);
 
                 // save to cache
@@ -1981,7 +1985,11 @@ const Rev* Git::fakeWorkDirRev(SCRef parent, SCRef log, SCRef longLog, int idx,
         if (!isMainHistory(fh))
                 patch = getWorkDirDiff(fh->fileNames().first());
 
+#if QT_VERSION >= QT_VERSION_CHECK(5, 8, 0)
         QString date(QString::number(QDateTime::currentDateTime().toSecsSinceEpoch()));
+#else
+        QString date(QString::number(qint64(QDateTime::currentDateTime().toMSecsSinceEpoch() / 1000 + 0.5)));
+#endif
         QString author("-");
         QStringList parents(parent);
         Rev* c = fakeRevData(ZERO_SHA, parents, author, date, log, longLog, patch, idx, fh);
diff --git src/FileHistory.cc src/FileHistory.cc
index d479116..72429a0 100644
--- src/FileHistory.cc
+++ src/FileHistory.cc
@@ -122,7 +122,7 @@ void FileHistory::clear(bool complete) {
   rowData.clear();
 
   if (testFlag(REL_DATE_F)) {
-#if QT_VERSION >= 0x06000
+#if QT_VERSION >= QT_VERSION_CHECK(5, 8, 0)
     secs = QDateTime::currentDateTime().toSecsSinceEpoch();
 #else
     secs = QDateTime::currentDateTime().toTime_t();
