diff --git phonon/backendinterface.h b/phonon/backendinterface.h
index 1f511eb3..50fcb669 100644
--- phonon/backendinterface.h
+++ phonon/backendinterface.h
@@ -176,7 +176,7 @@ class BackendInterface
         /**
          * Classes that the createObject function has to handle.
          */
-        enum Class {
+        enum Class : unsigned {
             /**
              * Request to return a %MediaObject object.
              */
diff --git phonon/objectdescription.h b/phonon/objectdescription.h
index f1d6c009..68072617 100644
--- phonon/objectdescription.h
+++ phonon/objectdescription.h
@@ -43,7 +43,7 @@ namespace Phonon
      *
      * \ingroup Backend
      */
-    enum ObjectDescriptionType
+    enum ObjectDescriptionType : unsigned
     {
         /**
          * Audio output devices. This can be soundcards (with different drivers), soundservers or
