diff --git gio/orig.gtestdbus.c gio/gtestdbus.c
index ee1d972..0c743c3 100644
--- gio/orig.gtestdbus.c
+++ gio/gtestdbus.c
@@ -53,6 +53,13 @@
 #include "glib-unixprivate.h"
 #endif
 
+#ifndef TARGET_OS_WATCH
+#define TARGET_OS_WATCH 0
+#endif
+#ifndef TARGET_OS_TV
+#define TARGET_OS_TV 0
+#endif
+
 /* -------------------------------------------------------------------------- */
 /* Utility: Wait until object has a single ref  */
 
diff --git glib/orig.gtestutils.c glib/gtestutils.c
index 867604d..534693e 100644
--- glib/orig.gtestutils.c
+++ glib/gtestutils.c
@@ -58,6 +58,12 @@
 #ifdef __APPLE__
 #include <TargetConditionals.h>
 #endif
+#ifndef TARGET_OS_WATCH
+#define TARGET_OS_WATCH 0
+#endif
+#ifndef TARGET_OS_TV
+#define TARGET_OS_TV 0
+#endif
 
 #include "gmain.h"
 #include "gpattern.h"
