diff --git gstreamer/CMakeLists.txt gstreamer/CMakeLists.txt
index cb01285..b7f542f 100644
--- gstreamer/CMakeLists.txt
+++ gstreamer/CMakeLists.txt
@@ -88,12 +88,12 @@ if (OPENGL_FOUND)
     list(APPEND phonon_gstreamer_SRCS glrenderer.cpp)
 endif ()
 
-set(BUILD_X11RENDERER TRUE)
+set(BUILD_X11RENDERER FALSE)
 if (PHONON_BUILD_PHONON4QT5)
     if (Qt5X11Extras_FOUND)
         set(BUILD_X11RENDERER TRUE)
     endif()
-elseif (NOT WIN32)
+elseif (NOT WIN32 AND NOT APPLE)
     set(BUILD_X11RENDERER TRUE)
 endif()
 
diff --git gstreamer/videowidget.cpp gstreamer/videowidget.cpp
index bd27b21..06e25a1 100644
--- gstreamer/videowidget.cpp
+++ gstreamer/videowidget.cpp
@@ -80,10 +80,12 @@ VideoWidget::~VideoWidget()
 
 void VideoWidget::updateWindowID()
 {
+#ifdef BUILD_X11RENDERER
     X11Renderer *render = dynamic_cast<X11Renderer*>(m_renderer);
     if (render) {
         render->setOverlay();
     }
+#endif
 }
 
 void Gstreamer::VideoWidget::syncX()
