--- gstreamer/orig.CMakeLists.txt	2016-04-18 06:25:19.000000000 +0200
+++ gstreamer/CMakeLists.txt	2016-09-18 15:34:04.000000000 +0200
@@ -88,12 +88,12 @@
     list(APPEND phonon_gstreamer_SRCS glrenderer.cpp)
 endif ()
 
-set(BUILD_X11RENDERER TRUE)
+set(BUILD_X11RENDERER FALSE)
 if (PHONON_BUILD_PHONON4QT5)
     if (Qt5X11Extras_FOUND)
         set(BUILD_X11RENDERER TRUE)
     endif()
-elseif (NOT WIN32)
+elseif (NOT WIN32 AND NOT APPLE)
     set(BUILD_X11RENDERER TRUE)
 endif()
 
