diff --git src/plugins/xsync/xsyncbasedpoller.cpp src/plugins/xsync/xsyncbasedpoller.cpp
index 48081c4..c0c9abc 100644
--- src/plugins/xsync/xsyncbasedpoller.cpp.orig	2016-10-11 17:44:57.000000000 +0200
+++ src/plugins/xsync/xsyncbasedpoller.cpp	2016-10-11 17:59:52.000000000 +0200
@@ -151,9 +151,9 @@ XSyncBasedPoller::XSyncBasedPoller(QObject *parent)
     }
 
     if (m_available) {
-        qCDebug(KIDLETIME_XSYNC_PLUGIN) << "XSync seems available and ready";
+//         qCDebug(KIDLETIME_XSYNC_PLUGIN) << "XSync seems available and ready";
     } else {
-        qCDebug(KIDLETIME_XSYNC_PLUGIN) << "XSync seems not available";
+        qCWarning(KIDLETIME_XSYNC_PLUGIN) << "XSync seems not available";
     }
 }
 
@@ -172,11 +172,11 @@ bool XSyncBasedPoller::setUpPoller()
         return false;
     }
 
-    qCDebug(KIDLETIME_XSYNC_PLUGIN) << "XSync Inited";
+//     qCDebug(KIDLETIME_XSYNC_PLUGIN) << "XSync Inited";
 
     s_globalXSyncBasedPoller()->isActive = true;
 
-    qCDebug(KIDLETIME_XSYNC_PLUGIN) << "Supported, init completed";
+//     qCDebug(KIDLETIME_XSYNC_PLUGIN) << "Supported, init completed";
 
     return true;
 }
@@ -332,7 +332,7 @@ void XSyncBasedPoller::setAlarm(Display *dpy, XSyncAlarm *alarm, XSyncCounter co
         XSyncChangeAlarm(dpy, *alarm, flags, &attr);
     } else {
         *alarm = XSyncCreateAlarm(dpy, flags, &attr);
-        qCDebug(KIDLETIME_XSYNC_PLUGIN) << "Created alarm" << *alarm;
+//         qCDebug(KIDLETIME_XSYNC_PLUGIN) << "Created alarm" << *alarm;
     }
 
     XFlush(m_display);
