diff --git src/kdeclarative/CMakeLists.txt src/kdeclarative/CMakeLists.txt
index ce12546..ecbb99e 100644
--- src/kdeclarative/CMakeLists.txt
+++ src/kdeclarative/CMakeLists.txt
@@ -27,6 +27,7 @@ target_link_libraries(KF5Declarative
     KF5::I18n       #i18n bindings
     KF5::KIOWidgets     #KIO::AccessManager
     KF5::IconThemes #KIconProvider
+    Qt5::QspXDG
 )
 target_include_directories(KF5Declarative PUBLIC "$<BUILD_INTERFACE:${KDeclarative_BINARY_DIR};${CMAKE_CURRENT_BINARY_DIR}>")
 
diff --git src/qmlcontrols/draganddrop/CMakeLists.txt src/qmlcontrols/draganddrop/CMakeLists.txt
index e8127e4..5bd3359 100644
--- src/qmlcontrols/draganddrop/CMakeLists.txt
+++ src/qmlcontrols/draganddrop/CMakeLists.txt
@@ -11,10 +11,13 @@ set(declarativedragdrop_SRCS
 
 add_library(draganddropplugin SHARED ${declarativedragdrop_SRCS})
 target_link_libraries(draganddropplugin
+PUBLIC
         Qt5::Core
         Qt5::Quick
         Qt5::Qml
         Qt5::Gui
+PRIVATE
+        Qt5::QspXDG
 )
 
 install(TARGETS draganddropplugin DESTINATION ${KDE_INSTALL_QMLDIR}/org/kde/draganddrop)
diff --git src/qmlcontrols/kcoreaddons/CMakeLists.txt src/qmlcontrols/kcoreaddons/CMakeLists.txt
index 3f77f2d..7a040cf 100644
--- src/qmlcontrols/kcoreaddons/CMakeLists.txt
+++ src/qmlcontrols/kcoreaddons/CMakeLists.txt
@@ -8,10 +8,13 @@ set(kcoreaddonsplugin_SRCS
 
 add_library(kcoreaddonsplugin SHARED ${kcoreaddonsplugin_SRCS})
 target_link_libraries(kcoreaddonsplugin
+PUBLIC
         Qt5::Core
         Qt5::Quick
         Qt5::Qml
         KF5::CoreAddons
+PRIVATE
+        Qt5::QspXDG
 )
 
 install(TARGETS kcoreaddonsplugin DESTINATION ${KDE_INSTALL_QMLDIR}/org/kde/kcoreaddons)
diff --git src/qmlcontrols/kioplugin/CMakeLists.txt src/qmlcontrols/kioplugin/CMakeLists.txt
index 7b258e0..29cfbb4 100644
--- src/qmlcontrols/kioplugin/CMakeLists.txt
+++ src/qmlcontrols/kioplugin/CMakeLists.txt
@@ -7,9 +7,12 @@ set(kioplugin_SRCS
 
 add_library(kio SHARED ${kioplugin_SRCS})
 target_link_libraries(kio
+PUBLIC
         Qt5::Core
         Qt5::Qml
         KF5::KIOWidgets
+PRIVATE
+        Qt5::QspXDG
 )
 
 install(TARGETS kio DESTINATION ${KDE_INSTALL_QMLDIR}/org/kde/kio)
diff --git src/qmlcontrols/kquickcontrols/private/CMakeLists.txt src/qmlcontrols/kquickcontrols/private/CMakeLists.txt
index da355c1..b71112a 100644
--- src/qmlcontrols/kquickcontrols/private/CMakeLists.txt
+++ src/qmlcontrols/kquickcontrols/private/CMakeLists.txt
@@ -9,6 +9,7 @@ set(kquickcontrolsprivate_SRCS
 add_library(kquickcontrolsprivateplugin SHARED ${kquickcontrolsprivate_SRCS})
 
 target_link_libraries(kquickcontrolsprivateplugin
+PUBLIC
         Qt5::Core
         Qt5::Quick
         Qt5::Qml
@@ -17,6 +18,8 @@ target_link_libraries(kquickcontrolsprivateplugin
         KF5::WidgetsAddons
         KF5::WindowSystem
         KF5::GlobalAccel
+PRIVATE
+        Qt5::QspXDG
         )
 
 install(TARGETS kquickcontrolsprivateplugin DESTINATION ${KDE_INSTALL_QMLDIR}/org/kde/private/kquickcontrols)
diff --git src/qmlcontrols/kwindowsystemplugin/CMakeLists.txt src/qmlcontrols/kwindowsystemplugin/CMakeLists.txt
index ce0ea74..a00d296 100644
--- src/qmlcontrols/kwindowsystemplugin/CMakeLists.txt
+++ src/qmlcontrols/kwindowsystemplugin/CMakeLists.txt
@@ -7,9 +7,12 @@ set(kwindowsystemplugin_SRCS
 
 add_library(kwindowsystem SHARED ${kwindowsystemplugin_SRCS})
 target_link_libraries(kwindowsystem
+PUBLIC
         Qt5::Core
         Qt5::Qml
         KF5::WindowSystem
+PRIVATE
+        Qt5::QspXDG
 )
 
 install(TARGETS kwindowsystem DESTINATION ${QML_INSTALL_DIR}/org/kde/kwindowsystem)
diff --git src/quickaddons/CMakeLists.txt src/quickaddons/CMakeLists.txt
index 8dc431f..7548abb 100644
--- src/quickaddons/CMakeLists.txt
+++ src/quickaddons/CMakeLists.txt
@@ -15,6 +15,7 @@
   PRIVATE
     KF5::I18n
     KF5::Declarative
+    Qt5::QspXDG
 )
 
 set_target_properties(KF5QuickAddons
diff --git src/calendarevents/orig.CMakeLists.txt	src/calendarevents/CMakeLists.txt	
--- src/calendarevents/orig.CMakeLists.txt	2015-12-06 22:21:30.000000000 +0100
+++ src/calendarevents/CMakeLists.txt	2015-12-16 13:23:04.000000000 +0100
@@ -7,7 +7,7 @@
 generate_export_header(KF5CalendarEvents BASE_NAME CalendarEvents)
 add_library(KF5::CalendarEvents ALIAS KF5CalendarEvents)
 
-target_link_libraries(KF5CalendarEvents PUBLIC Qt5::Core)
+target_link_libraries(KF5CalendarEvents PUBLIC Qt5::Core PRIVATE Qt5::QspXDG)
 
 target_include_directories(KF5CalendarEvents INTERFACE "$<INSTALL_INTERFACE:${KDE_INSTALL_INCLUDEDIR_KF5}/KDeclarative>")
 set_target_properties(KF5CalendarEvents PROPERTIES VERSION ${KDECLARATIVE_VERSION_STRING}
