--- src/orig.CMakeLists.txt	2016-03-23 22:55:11.000000000 +0100
+++ src/CMakeLists.txt	2016-04-21 16:09:24.000000000 +0200
@@ -1,5 +1,6 @@
 #add_subdirectory(naturalqueryparser)
 add_definitions(-DTRANSLATION_DOMAIN=\"baloowidgets5\")
+include(ECMMarkNonGuiExecutable)
 
 set(widgets_SRCS
   kblocklayout.cpp
@@ -82,6 +83,7 @@
 # Extractor Process
 #
 add_executable(baloo_filemetadata_temp_extractor extractor.cpp)
+ecm_mark_nongui_executable(baloo_filemetadata_temp_extractor)
 target_link_libraries(baloo_filemetadata_temp_extractor
   Qt5::Core
   KF5::I18n
