--- src/tagsfileitemactionplugin/orig.tagsfileitemaction.cpp	2019-11-01 22:58:26.000000000 +0100
+++ src/tagsfileitemactionplugin/tagsfileitemaction.cpp	2020-04-24 17:32:56.000000000 +0200
@@ -34,6 +34,7 @@
 #include <KPluginFactory>
 #include <KPluginLoader>
 #include <KLocalizedString>
+#include <KProtocolManager>
 
 K_PLUGIN_FACTORY_WITH_JSON(TagsFileItemActionFactory, "tagsfileitemaction.json", registerPlugin<TagsFileItemAction>();)
 
@@ -116,7 +117,10 @@
         return {};
     }
 
-    m_tagsLister.openUrl(QUrl(QStringLiteral("tags:/")), KCoreDirLister::OpenUrlFlag::Reload);
+    const auto url = QUrl(QStringLiteral("tags:/"));
+    if (KProtocolManager::supportsListing(url)) {
+        m_tagsLister.openUrl(url, KCoreDirLister::OpenUrlFlag::Reload);
+    }
 
     return {m_menu->menuAction()};
 }
--- src/orig.CMakeLists.txt	2020-04-24 17:29:13.000000000 +0200
+++ src/CMakeLists.txt	2020-04-24 17:46:22.000000000 +0200
@@ -2,7 +2,9 @@
 include(ECMMarkNonGuiExecutable)
 
 add_subdirectory(filepropertiesplugin)
-add_subdirectory(tagsfileitemactionplugin)
+if (NOT APPLE)
+  add_subdirectory(tagsfileitemactionplugin)
+endif()
 
 set(widgets_SRCS
   kblocklayout.cpp
