--- src/CMakeLists.txt.orig	2019-11-01 22:58:26.000000000 +0100
+++ src/CMakeLists.txt	2020-04-21 10:13:57.720497628 +0200
@@ -1,4 +1,5 @@
 add_definitions(-DTRANSLATION_DOMAIN=\"baloowidgets5\")
+include(ECMMarkNonGuiExecutable)
 
 add_subdirectory(filepropertiesplugin)
 add_subdirectory(tagsfileitemactionplugin)
@@ -79,6 +80,7 @@
 # Extractor Process
 #
 add_executable(baloo_filemetadata_temp_extractor extractor.cpp filemetadatautil.cpp)
+ecm_mark_nongui_executable(baloo_filemetadata_temp_extractor)
 target_link_libraries(baloo_filemetadata_temp_extractor
   Qt5::Core
   KF5::I18n
