diff --git projectmanagers/cmake/cmakemanager.cpp projectmanagers/cmake/cmakemanager.cpp
index 04e26f64789613a077b908c511e709b4e3804912..25b5cfd574a0c1988ce4d85bd8223a7842137e6b 100644
--- projectmanagers/cmake/cmakemanager.cpp
+++ projectmanagers/cmake/cmakemanager.cpp
@@ -206,7 +206,7 @@ KJob* CMakeManager::createImportJob(ProjectFolderItem* item)
     auto job = new ChooseCMakeInterfaceJob(project, this);
     connect(job, &CMakeImportJsonJob::result, this, [this, job, project](){
         if (job->error() != 0) {
-            qCWarning(CMAKE) << "couldn't load json successfully" << project->name();
+            qCDebug(CMAKE) << "couldn't load json successfully" << project->name();
             m_projects.remove(project);
         }
     });

