diff --git src/core/orig.flameshot.cpp src/core/flameshot.cpp
index 6ab5bb8..61d759a 100644
--- src/core/orig.flameshot.cpp
+++ src/core/flameshot.cpp
@@ -133,6 +133,19 @@ CaptureWidget* Flameshot::gui(const CaptureRequest& req)
     }
 }
 
+static QScreen* screenAt(const QPoint& pos)
+{
+#if (QT_VERSION >= QT_VERSION_CHECK(5, 10, 0))
+    return qApp->screenAt(pos);
+#else
+    for (QScreen* const screen : QGuiApplication::screens()) {
+        if (screen->geometry().contains(pos)) {
+            return screen;
+        }
+    }
+#endif
+}
+
 void Flameshot::screen(CaptureRequest req, const int screenNumber)
 {
     if (!resolveAnyConfigErrors()) {
@@ -144,7 +157,7 @@ void Flameshot::screen(CaptureRequest req, const int screenNumber)
 
     if (screenNumber < 0) {
         QPoint globalCursorPos = QCursor::pos();
-        screen = qApp->screenAt(globalCursorPos);
+        screen = screenAt(globalCursorPos);
     } else if (screenNumber >= qApp->screens().count()) {
         AbstractLogger() << QObject::tr(
           "Requested screen exceeds screen count");
diff --git src/widgets/orig.trayicon.cpp src/widgets/trayicon.cpp
index 3221118..b4c48a0 100644
--- src/widgets/orig.trayicon.cpp
+++ src/widgets/trayicon.cpp
@@ -21,7 +21,7 @@ TrayIcon::TrayIcon(QObject* parent)
     initMenu();
 
     setToolTip(QStringLiteral("Flameshot"));
-#if defined(Q_OS_MACOS)
+#if defined(Q_OS_MACOS) && (QT_VERSION >= QT_VERSION_CHECK(6, 0, 0))
     // Because of the following issues on MacOS "Catalina":
     // https://bugreports.qt.io/browse/QTBUG-86393
     // https://developer.apple.com/forums/thread/126072
@@ -36,7 +36,7 @@ TrayIcon::TrayIcon(QObject* parent)
       QIcon::fromTheme("flameshot-tray", QIcon(GlobalValues::iconPathPNG()));
     setIcon(icon);
 
-#if defined(Q_OS_MACOS)
+#if defined(Q_OS_MACOS) && (QT_VERSION >= QT_VERSION_CHECK(6, 0, 0))
     if (currentMacOsVersion < currentMacOsVersion.MacOSBigSur) {
         // Because of the following issues on MacOS "Catalina":
         // https://bugreports.qt.io/browse/QTBUG-86393
@@ -97,7 +97,7 @@ void TrayIcon::initMenu()
 
     auto* captureAction = new QAction(tr("&Take Screenshot"), this);
     connect(captureAction, &QAction::triggered, this, [this]() {
-#if defined(Q_OS_MACOS)
+#if defined(Q_OS_MACOS) && (QT_VERSION >= QT_VERSION_CHECK(6, 0, 0))
         auto currentMacOsVersion = QOperatingSystemVersion::current();
         if (currentMacOsVersion >= currentMacOsVersion.MacOSBigSur) {
             startGuiCapture();
