--- coregrind/orig.link_tool_exe_darwin.in	2018-05-05 09:42:22.000000000 +0200
+++ coregrind/link_tool_exe_darwin.in	2019-08-24 17:51:58.000000000 +0200
@@ -138,7 +138,7 @@
 
 
 # build the command line
-my $cmd = "/usr/bin/ld";
+my $cmd = "@PREFIX@/bin/ld";
 
 $cmd = "$cmd -static";
 
diff --git configure.orig configure
index 5f7e236..0b77812 100755
--- configure.orig
+++ configure
@@ -11565,7 +11565,7 @@ if test "x${vg_cv_lto}" != "xno" -a "x${LTO_AR}" != "x" -a "x${LTO_RANLIB}" != "
 { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking if toolchain accepts lto" >&5
 printf %s "checking if toolchain accepts lto... " >&6; }
 safe_CFLAGS=$CFLAGS
-TEST_LTO_CFLAGS="-flto -fuse-linker-plugin"
+TEST_LTO_CFLAGS="-flto"
 CFLAGS="$TEST_LTO_CFLAGS -Werror"
 
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
