--- coregrind/orig.link_tool_exe_darwin.in	2018-05-05 09:42:22.000000000 +0200
+++ coregrind/link_tool_exe_darwin.in	2019-08-24 17:51:58.000000000 +0200
@@ -138,7 +138,7 @@
 
 
 # build the command line
-my $cmd = "/usr/bin/ld";
+my $cmd = "@PREFIX@/bin/ld";
 
 $cmd = "$cmd -static";
 
--- orig.configure	2018-10-09 09:29:07.000000000 +0200
+++ configure	2019-08-24 17:08:02.000000000 +0200
@@ -9296,7 +9296,7 @@
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking if toolchain accepts lto" >&5
 $as_echo_n "checking if toolchain accepts lto... " >&6; }
 safe_CFLAGS=$CFLAGS
-TEST_LTO_CFLAGS="-flto -flto-partition=one -fuse-linker-plugin"
+TEST_LTO_CFLAGS="-flto"
 # Note : using 'one' partition is giving a slightly smaller/faster memcheck
 # and ld/lto-trans1 still needs a reasonable memory (about 0.5GB) when linking.
 CFLAGS="$TEST_LTO_CFLAGS -Werror"
