--- coregrind/m_syswrap/orig.syswrap-darwin.c	2018-08-20 11:34:59.000000000 +0200
+++ coregrind/m_syswrap/syswrap-darwin.c	2019-08-24 17:40:25.000000000 +0200
@@ -68,6 +68,9 @@
 #include <mach/mach.h>
 #include <mach/mach_vm.h>
 #include <semaphore.h>
+#define __APPLE_API_EVOLVING
+#define KERNEL
+#include <sys/kauth.h>
 /* --- !!! --- EXTERNAL HEADERS end --- !!! --- */
 
 #define msgh_request_port      msgh_remote_port
