diff --git a/rkward/rbackend/rkfrontendtransmitter.cpp b/rkward/rbackend/rkfrontendtransmitter.cpp
index ca8c0e2e..356d91a1 100644
--- a/rkward/rbackend/rkfrontendtransmitter.cpp
+++ b/rkward/rbackend/rkfrontendtransmitter.cpp
@@ -80,7 +80,10 @@ void RKFrontendTransmitter::run () {
 	server = new QLocalServer (this);
 	// we add a bit of randomness to the servername, as in general the servername must be unique
 	// there could be conflicts with concurrent or with previous crashed rkward sessions.
-	if (!server->listen ("rkward" + KRandom::randomString (8))) handleTransmissionError ("Failure to start frontend server: " + server->errorString ());
+    QString lAddress = "rkward" + KRandom::randomString (8);
+	if (!server->listen (lAddress)) {
+        handleTransmissionError ("Failure to start frontend server \"" + lAddress + "\": " + server->errorString ());
+    }
 	connect (server, &QLocalServer::newConnection, this, &RKFrontendTransmitter::connectAndEnterLoop, Qt::QueuedConnection);
 
 	// start backend
