diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2bcd1ef..389b2d0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -309,11 +309,13 @@ else()
     set(pc_libraries "${pc_libraries} -pthread")
   endif()
   find_library(MI_LIBRT rt)
-  if(MI_LIBRT)
-    list(APPEND mi_libraries ${MI_LIBRT})
-    set(pc_libraries "${pc_libraries} -lrt")
+  if (NOT APPLE)
+    if(MI_LIBRT)
+      list(APPEND mi_libraries ${MI_LIBRT})
+      set(pc_libraries "${pc_libraries} -lrt")
+    endif()
+    find_library(MI_LIBATOMIC atomic)
   endif()
-  find_library(MI_LIBATOMIC atomic)
   if (NOT MI_LIBATOMIC AND MI_USE_LIBATOMIC)
     set(MI_LIBATOMIC atomic)
   endif()
@@ -353,9 +355,6 @@ if(MI_TRACK_ASAN)
   set(mi_basename "${mi_basename}-asan")
 endif()
 string(TOLOWER "${CMAKE_BUILD_TYPE}" CMAKE_BUILD_TYPE_LC)
-if(NOT(CMAKE_BUILD_TYPE_LC MATCHES "^(release|relwithdebinfo|minsizerel|none)$"))
-  set(mi_basename "${mi_basename}-${CMAKE_BUILD_TYPE_LC}") #append build type (e.g. -debug) if not a release version
-endif()
 
 if(MI_BUILD_SHARED)
   list(APPEND mi_build_targets "shared")
