diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5ce084f6..6a833b11 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -491,13 +491,15 @@ else()
   if(MI_LIB_PTHREAD)
     list(APPEND mi_libraries "${MI_LIB_PTHREAD}")
   endif()
-  find_link_library("rt" MI_LIB_RT)
-  if(MI_LIB_RT)
-    list(APPEND mi_libraries "${MI_LIB_RT}")
-  endif()
-  find_link_library("atomic" MI_LIB_ATOMIC)
-  if(MI_LIB_ATOMIC)
-    list(APPEND mi_libraries "${MI_LIB_ATOMIC}")
+  if (NOT APPLE)
+    find_link_library("rt" MI_LIB_RT)
+    if(MI_LIB_RT)
+      list(APPEND mi_libraries "${MI_LIB_RT}")
+    endif()
+    find_link_library("atomic" MI_LIB_ATOMIC)
+    if(MI_LIB_ATOMIC)
+      list(APPEND mi_libraries "${MI_LIB_ATOMIC}")
+    endif()
   endif()
 endif()
 
@@ -533,10 +535,8 @@ if(MI_TRACK_ASAN)
 endif()
 string(TOLOWER "${CMAKE_BUILD_TYPE}" CMAKE_BUILD_TYPE_LC)
 list(APPEND mi_defines "MI_CMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE_LC}")  #todo: multi-config project needs $<CONFIG> ?
-if(CMAKE_BUILD_TYPE_LC MATCHES "^(release|relwithdebinfo|minsizerel|none)$")
+if(CMAKE_BUILD_TYPE_LC MATCHES "^(release|relwithdebinfo|minsizerel|none|macports)$")
   list(APPEND mi_defines MI_BUILD_RELEASE)
-else()
-  set(mi_libname "${mi_libname}-${CMAKE_BUILD_TYPE_LC}") #append build type (e.g. -debug) if not a release version
 endif()
 
 if(MI_BUILD_SHARED)
