diff --git a/CMakeLists.txt b/CMakeLists.txt
index 283b0f57..6e3f1693 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -454,13 +454,15 @@ else()
   if(MI_LIB_PTHREAD)
     list(APPEND mi_libraries "${MI_LIB_PTHREAD}")
   endif()
-  find_link_library("rt" MI_LIB_RT)
-  if(MI_LIB_RT)
-    list(APPEND mi_libraries "${MI_LIB_RT}")
-  endif()
-  find_link_library("atomic" MI_LIB_ATOMIC)
-  if(MI_LIB_ATOMIC)
-    list(APPEND mi_libraries "${MI_LIB_ATOMIC}")
+  if (NOT APPLE)
+    find_link_library("rt" MI_LIB_RT)
+    if(MI_LIB_RT)
+      list(APPEND mi_libraries "${MI_LIB_RT}")
+    endif()
+    find_link_library("atomic" MI_LIB_ATOMIC)
+    if(MI_LIB_ATOMIC)
+      list(APPEND mi_libraries "${MI_LIB_ATOMIC}")
+    endif()
   endif()
 endif()
 
@@ -495,9 +497,6 @@ if(MI_TRACK_ASAN)
   set(mi_basename "${mi_basename}-asan")
 endif()
 string(TOLOWER "${CMAKE_BUILD_TYPE}" CMAKE_BUILD_TYPE_LC)
-if(NOT(CMAKE_BUILD_TYPE_LC MATCHES "^(release|relwithdebinfo|minsizerel|none)$"))
-  set(mi_basename "${mi_basename}-${CMAKE_BUILD_TYPE_LC}") #append build type (e.g. -debug) if not a release version
-endif()
 
 if(MI_BUILD_SHARED)
   list(APPEND mi_build_targets "shared")
