--- src/file/orig.dl_posix.c	2017-03-02 14:33:49.000000000 +0100
+++ src/file/dl_posix.c	2017-08-22 14:59:52.000000000 +0200
@@ -65,7 +65,7 @@
        @executable_path - location of running binary (ex. /Applications/Some.app/Contents/MacOS)
        @rpath           - search rpaths of running binary (man install_name_path)
     */
-    static const char *search_paths[] = {"", "@loader_path/lib/", "@loader_path/", "@executable_path/",
+    static const char *search_paths[] = {"", "@PREFIX@/lib/", "@loader_path/lib/", "@loader_path/", "@executable_path/",
                                          "@executable_path/lib/", "@executable_path/../lib/",
                                          "@executable_path/../Resources/", "@rpath/", NULL};
     version = NULL;
