diff --git autotests/kcompressiondevicetest.cpp autotests/kcompressiondevicetest.cpp
index e161264..5fa1a1b 100644
--- autotests/kcompressiondevicetest.cpp
+++ autotests/kcompressiondevicetest.cpp
@@ -194,14 +194,17 @@ void KCompressionDeviceTest::testWriteErrorOnClose()
 
 void KCompressionDeviceTest::testSeekReadUncompressedBuffer_data()
 {
+#if QT_VERSION >= QT_VERSION_CHECK(5, 11, 0)
     QTest::addColumn<int>("dataSize");
     QTest::addColumn<int>("realDataPos");
     QTest::newRow("1.5buffer") << BUFFER_SIZE + BUFFER_SIZE / 2 << BUFFER_SIZE;
     QTest::newRow("5seekbuffer") << 5 * SEEK_BUFFER_SIZE << 4 * SEEK_BUFFER_SIZE;
+#endif
 }
 
 void KCompressionDeviceTest::testSeekReadUncompressedBuffer()
 {
+#if QT_VERSION >= QT_VERSION_CHECK(5, 11, 0)
     QFETCH(int, dataSize);
     QFETCH(int, realDataPos);
 
@@ -226,4 +229,5 @@ void KCompressionDeviceTest::testSeekReadUncompressedBuffer()
     for (int i = 0; i < kcdData.size(); ++i) {
         QCOMPARE(kcdData[i], i);
     }
+#endif
 }
