diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index db84dbf894a7fe6d25d90ce5eb0f4fe6c5c683e3..faefd331812e5acfaffdf9224fcb6768b3410895 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -162,7 +162,7 @@ if (APPLE)
         )
     else()
       list(APPEND SOURCES 
-        codec/video/SurfaceInteropIOSurface.cpp
+        codec/video/SurfaceInteropIOSurface.mm
         )
     endif()
   endif()
