=== modified file 'libdbusmenu-glib/Makefile.am'
--- libdbusmenu-glib/Makefile.am	2014-10-30 13:56:39 +0000
+++ libdbusmenu-glib/Makefile.am	2018-03-05 22:44:15 +0000
@@ -213,10 +213,10 @@
 if HAVE_INTROSPECTION
 
 vapidir = $(datadir)/vala/vapi
-vapi_DATA = Dbusmenu-0.4.vapi
+vapi_DATA = dbusmenu-glib-0.4.vapi
 
-Dbusmenu-0.4.vapi: Dbusmenu-0.4.gir
-	$(VALA_API_GEN) --library=Dbusmenu-0.4 $<
+dbusmenu-glib-0.4.vapi: Dbusmenu-0.4.gir
+	$(VALA_API_GEN) --library=dbusmenu-glib-0.4 $<
 
 CLEANFILES += $(vapi_DATA)
 

=== modified file 'libdbusmenu-gtk/Makefile.am'
--- libdbusmenu-gtk/Makefile.am	2014-10-30 13:56:39 +0000
+++ libdbusmenu-gtk/Makefile.am	2018-03-05 22:44:15 +0000
@@ -157,14 +157,14 @@
 if HAVE_INTROSPECTION
 
 vapidir = $(datadir)/vala/vapi
-vapi_DATA = DbusmenuGtk$(VER)-0.4.vapi
+vapi_DATA = dbusmenu-gtk$(VER)-0.4.vapi
 
-DbusmenuGtk$(VER)-0.4.vapi: DbusmenuGtk$(VER)-0.4.tmp.gir Makefile.am
-	$(VALA_API_GEN) --library=DbusmenuGtk$(VER)-0.4 \
+dbusmenu-gtk$(VER)-0.4.vapi: DbusmenuGtk$(VER)-0.4.tmp.gir Makefile.am
+	$(VALA_API_GEN) --library=dbusmenu-gtk$(VER)-0.4 \
 		--pkg gdk-pixbuf-2.0 \
 		--pkg $(GTKVALA) \
 		--pkg atk \
-		--pkg Dbusmenu-0.4 \
+		--pkg dbusmenu-glib-0.4 \
 		--vapidir=$(top_builddir)/libdbusmenu-glib \
 		$<
 

