--- avidemux/common/ADM_videoCodec/src/orig.ADM_ffmpeg_vdpau.cpp	2020-07-05 19:22:21.000000000 +0200
+++ avidemux/common/ADM_videoCodec/src/ADM_ffmpeg_vdpau.cpp	2020-08-25 16:57:17.990374818 +0200
@@ -227,7 +227,9 @@
             switch(avctx->codec_id)
             {
                 FMT_V_CHECK(H264,      VDP_DECODER_PROFILE_H264_HIGH)
+#ifdef VDP_DECODER_PROFILE_HEVC_MAIN
                 FMT_V_CHECK(HEVC,      VDP_DECODER_PROFILE_HEVC_MAIN) 
+#endif
                 FMT_V_CHECK(MPEG1VIDEO,VDP_DECODER_PROFILE_MPEG1)
                 FMT_V_CHECK(MPEG2VIDEO,VDP_DECODER_PROFILE_MPEG2_MAIN)
                 FMT_V_CHECK(WMV3,      VDP_DECODER_PROFILE_VC1_MAIN)
