--- qt4/ADM_userInterfaces/ADM_gui/orig.Q_gui2.cpp	2020-07-05 19:22:21.000000000 +0200
+++ qt4/ADM_userInterfaces/ADM_gui/Q_gui2.cpp	2020-08-27 21:47:56.000000000 +0200
@@ -485,7 +485,9 @@
 #ifndef __APPLE__
     setWindowIcon(QIcon(MKICON(avidemux-icon)));
 #else
-    setWindowIcon(QIcon(MKOSXICON(avidemux)));
+    if (windowIcon().isNull()) {
+        setWindowIcon(QIcon(MKOSXICON(avidemux)));
+    }
 #endif
 
     // Hook also the toolbar
