--- qt4/ADM_userInterfaces/ADM_gui/orig.Q_gui2.cpp	2020-07-05 19:22:21.000000000 +0200
+++ qt4/ADM_userInterfaces/ADM_gui/Q_gui2.cpp	2020-08-27 21:47:56.000000000 +0200
@@ -485,7 +485,9 @@
 #ifndef __APPLE__
     setWindowIcon(QIcon(MKICON(avidemux-icon)));
 #else
-    setWindowIcon(QIcon(MKOSXICON(avidemux)));
+    if (windowIcon().isNull()) {
+        setWindowIcon(QIcon(MKOSXICON(avidemux)));
+    }
 #endif
 
     // Hook also the toolbar
--- avidemux/qt4/orig.CMakeLists.txt	2020-07-05 19:22:21.000000000 +0200
+++ avidemux/qt4/CMakeLists.txt	2020-08-28 12:55:26.000000000 +0200
@@ -79,6 +79,14 @@
 ADD_DEFINITIONS(${QT_DEFINITIONS})
 #--
 
+INCLUDE(admCheckFreeType)
+INCLUDE(admCheckFontConfig)
+checkFreeType()
+checkFontConfig()
+IF (FONTCONFIG_FOUND)
+  add_definitions(-DUSE_FONTCONFIG)
+endif()
+
 #
 ##########################################
 # Config
