--- cmake/orig.admFFmpegBuild_native.cmake	2020-07-05 19:22:21.000000000 +0200
+++ cmake/admFFmpegBuild_native.cmake	2020-08-26 16:30:59.000000000 +0200
@@ -46,7 +46,11 @@
 #@@
 ADM_FF_ADD_OPTIONS()
 
-xadd(--cc "${CMAKE_C_COMPILER}")
+if(CMAKE_C_COMPILER_LAUNCHER)
+    xadd(--cc "${CMAKE_C_COMPILER_LAUNCHER} ${CMAKE_C_COMPILER}")
+else()
+    xadd(--cc "${CMAKE_C_COMPILER}")
+endif()
 xadd(--ld "${CMAKE_C_COMPILER}")
 xadd(--ar "${CMAKE_AR}")
 
