diff --git mesonbuild/compilers/mixins/clike.py mesonbuild/compilers/mixins/clike.py
index 24f4796fdf..124c49c9b8 100644
--- mesonbuild/compilers/mixins/clike.py
+++ mesonbuild/compilers/mixins/clike.py
@@ -919,7 +919,7 @@ def _get_file_from_list(env, files: T.List[str]) -> Path:
         for f in files:
             if not f.is_file():
                 continue
-            archs = mesonlib.darwin_get_object_archs(f)
+            archs = mesonlib.darwin_get_object_archs(str(f))
             if archs and env.machines.host.cpu_family in archs:
                 return f
             else:
