--- src/orig.main.cpp	2016-04-13 18:49:42.000000000 +0200
+++ src/main.cpp	2016-04-21 16:35:06.000000000 +0200
@@ -39,7 +39,7 @@
 {
     QApplication app(argc, argv);
     app.setAttribute(Qt::AA_UseHighDpiPixmaps, true);
-    app.setWindowIcon(QIcon::fromTheme(QStringLiteral("system-file-manager")));
+    app.setWindowIcon(QIcon::fromTheme(QStringLiteral("system-file-manager"), app.windowIcon()));
 
     KCrash::initialize();
 
