--- src/orig.CMakeLists.txt	2016-04-13 18:49:42.000000000 +0200
+++ src/CMakeLists.txt	2016-04-21 17:08:56.000000000 +0200
@@ -1,3 +1,4 @@
+include(ECMAddAppIcon)
 
 configure_file(config-baloo.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/config-baloo.h)
 
@@ -270,6 +271,10 @@
 # TODO Does anything replace kde4_add_app_icon ?
 #kde4_add_app_icon(dolphin_SRCS "${KDE4_ICON_INSTALL_DIR}/oxygen/*/apps/system-file-manager.png")
 
+# Sets the icon on Windows and OSX
+file(GLOB ICONS_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/icons/*system-file-manager.png")
+ecm_add_app_icon(dolphin_SRCS ICONS ${ICONS_SRCS})
+
 kf5_add_kdeinit_executable(dolphin ${dolphin_SRCS})
 
 target_include_directories(kdeinit_dolphin PRIVATE ${PHONON_INCLUDES})
