diff --git osdep/orig.macosx_menubar.m osdep/macosx_menubar.m
index 92bd3fa..11e292b 100644
--- osdep/orig.macosx_menubar.m
+++ osdep/macosx_menubar.m
@@ -602,7 +602,7 @@
 - (NSMenu *)mainMenu
 {
     NSMenu *mainMenu = [[NSMenu alloc] initWithTitle:@"MainMenu"];
-    NSApp.servicesMenu = [NSMenu alloc];
+    [NSApp setServicesMenu:[NSMenu alloc]];
 
     for(id mMenu in menuTree) {
         NSMenu *menu = [[NSMenu alloc] initWithTitle:mMenu[@"name"]];
@@ -633,7 +633,7 @@
                 }
 
                 if ([subMenu[@"name"] isEqual:@"Services"]) {
-                    iItem.submenu = NSApp.servicesMenu;
+                    iItem.submenu = [NSApp servicesMenu];
                 }
             }
         }
@@ -712,7 +712,9 @@
     [alert setIcon:[self getMPVIcon]];
 
     NSTextField *input = [[NSTextField alloc] initWithFrame:NSMakeRect(0, 0, 300, 24)];
-    [input setPlaceholderString:@"URL"];
+    if ([input respondsToSelector:@selector(setPlaceholderString:)]) {
+        [input setPlaceholderString:@"URL"];
+    }
     [alert setAccessoryView:input];
 
     dispatch_after(dispatch_time(DISPATCH_TIME_NOW, 0.1 * NSEC_PER_SEC), dispatch_get_main_queue(), ^{
