diff --git common/orig.encode_lavc.c common/encode_lavc.c
index 7473238..d7524ff 100644
--- common/orig.encode_lavc.c
+++ common/encode_lavc.c
@@ -422,7 +422,7 @@ void encode_lavc_finish(struct encode_lavc_context *ctx)
         }
 
         for (i = 0; i < ctx->avc->nb_streams; i++) {
-            av_free(ctx->avc->streams[i]->info);
+//             av_free(ctx->avc->streams[i]->info);
             av_free(ctx->avc->streams[i]);
         }
         ctx->vst = NULL;
