diff --git waftools/checks/custom.py.orig waftools/checks/custom.py
index 3af9bdb..68cc921 100644
--- waftools/checks/custom.py.orig
+++ waftools/checks/custom.py
@@ -115,9 +115,10 @@ def check_cocoa(ctx, dependency_identifier):
     res = fn(ctx, dependency_identifier)
     # on macOS we explicitly need to set the SDK path, otherwise it can lead to
     # linking warnings or errors
-    if res:
+    sysroot = _run('xcrun --sdk macosx --show-sdk-path')
+    if res and len(sysroot) > 0:
         ctx.env.append_value('LINKFLAGS', [
-            '-isysroot', '%s' % _run('xcrun --sdk macosx --show-sdk-path')
+            '-isysroot', '%s' % sysroot
         ])
 
     return res
