diff --git a/lib-src/portaudio-v19/test/patest_mono.c b/lib-src/portaudio-v19/test/patest_mono.c
index f55d4f4..4ae0161 100644
--- a/lib-src/portaudio-v19/test/patest_mono.c
+++ b/lib-src/portaudio-v19/test/patest_mono.c
@@ -152,4 +152,4 @@ error:
     fprintf( stderr, "Error number: %d\n", err );
     fprintf( stderr, "Error message: %s\n", Pa_GetErrorText( err ) );
     return err;
-}
\ No newline at end of file
+}
