diff --git a/lib-src/FileDialog/configure b/lib-src/FileDialog/configure
index 91b644a..4b44ddf 100755
--- a/lib-src/FileDialog/configure
+++ b/lib-src/FileDialog/configure
@@ -15615,9 +15615,9 @@ fi
 
 IMPLEMENTATION="generic"
 case "${host_os}" in
-   darwin*)
-            IMPLEMENTATION="mac"
-   ;;
+#    darwin*)
+#             IMPLEMENTATION="mac"
+#    ;;
 
    cygwin*)
             IMPLEMENTATION="win"
diff --git a/lib-src/FileDialog/configure.ac b/lib-src/FileDialog/configure.ac
index f5a9365..f84ce57 100644
--- a/lib-src/FileDialog/configure.ac
+++ b/lib-src/FileDialog/configure.ac
@@ -73,10 +73,10 @@ AC_CANONICAL_HOST
 
 IMPLEMENTATION="generic"
 case "${host_os}" in
-   darwin*)
-      dnl Mac OS X configuration
-      IMPLEMENTATION="mac"
-   ;;
+#   darwin*)
+#      dnl Mac OS X configuration
+#      IMPLEMENTATION="mac"
+#   ;;
 
    cygwin*)
       dnl Windows/CygWin configuration
