diff --git a/src/widgets/ProgressDialog.cpp b/src/widgets/ProgressDialog.cpp
index e130efc69d0b7a30bf68bdf6dc9a5d20af02cd18..1d7207dbc25e4da0c7118efbe5acb941693c7c3c 100644
--- a/src/widgets/ProgressDialog.cpp
+++ b/src/widgets/ProgressDialog.cpp
@@ -53,6 +53,10 @@
 #include "../Prefs.h"
 #include "../Internat.h"
 
+#ifdef __WXOSX_COCOA__
+extern "C" void NSBeep(void);
+#endif
+
 // This really should be a Preferences setting
 static const unsigned char beep[] =
 {
@@ -1593,7 +1597,13 @@ void ProgressDialog::Beep() const
 
       if (name.IsEmpty())
       {
+#ifdef __WXOSX_COCOA__
+         // wxSound::Create(size_t,const void*) isn't implemented;
+         // use the system beep function.
+         NSBeep();
+#else
          s.Create(sizeof(beep), beep);
+#endif
       }
       else
       {
