diff --git a/src/ShuttleGui.h b/src/ShuttleGui.h
index 12390c957590bec555f37170b0fa252582e5b36a..2cd7b101f09db964e99c22d3fb655d12fd89025e 100644
--- a/src/ShuttleGui.h
+++ b/src/ShuttleGui.h
@@ -86,7 +86,9 @@ public:
    wxStaticBoxWrapper( Args &&...args )
       : wxStaticBox( std::forward<Args>(args)... )
    {
-      m_container.EnableSelfFocus();
+#if wxCHECK_VERSION(3, 1, 1)
+       m_container.EnableSelfFocus();
+#endif
    }
 };
 
diff --git a/lib-src/FileDialog/mac/FileDialogPrivate.h b/lib-src/FileDialog/mac/FileDialogPrivate.h
index 697969fd2a31d24501361b71bfc076f37671f813..a2db97e9f21661e874984c01d8375b44d93ecd16 100644
--- a/lib-src/FileDialog/mac/FileDialogPrivate.h
+++ b/lib-src/FileDialog/mac/FileDialogPrivate.h
@@ -96,7 +96,9 @@ protected:
     int m_firstFileTypeFilter;
     wxArrayString m_currentExtensions;
     WX_NSObject m_delegate;
+#if !wxCHECK_VERSION(3, 0, 3)
     WX_NSObject m_sheetDelegate;
+#endif
     wxString m_noOverwritePromptFilename;
 #endif
 
diff --git a/lib-src/FileDialog/mac/FileDialogPrivate.mm b/lib-src/FileDialog/mac/FileDialogPrivate.mm
index ca599098502bd9d622d762c91e18d627add0b809..7736c6c2abe3418416e7d1ccfc827787abf4dde9 100644
--- a/lib-src/FileDialog/mac/FileDialogPrivate.mm
+++ b/lib-src/FileDialog/mac/FileDialogPrivate.mm
@@ -147,7 +147,9 @@ void FileDialog::Init()
 {
     m_filterIndex = -1;
     m_delegate = nil;
+#if !wxCHECK_VERSION(3, 0, 3)
     m_sheetDelegate = nil;
+#endif
     m_filterPanel = NULL;
     m_filterChoice = NULL;
 }
@@ -159,13 +161,17 @@ void FileDialog::Create(
 {
     FileDialogBase::Create(parent, message, defaultDir, defaultFileName, wildCard, style, pos, sz, name);
 
+// #if !wxCHECK_VERSION(3, 0, 3)
 //    m_sheetDelegate = [[ModalDialogDelegate alloc] init];
 //    [(ModalDialogDelegate*)m_sheetDelegate setImplementation: this];
+// #endif
 }
 
 FileDialog::~FileDialog()
 {
+#if !wxCHECK_VERSION(3, 0, 3)
     [m_sheetDelegate release];
+#endif
 }
 
 bool FileDialog::SupportsExtraControl() const
@@ -315,10 +321,23 @@ void FileDialog::ShowWindowModal()
         [sPanel setAllowsOtherFileTypes:NO];
         
         NSWindow* nativeParent = parentWindow->GetWXWindow();
+#if wxCHECK_VERSION(3, 0, 3)
+        // the ModalDialogDelegate class is no longer exported, but the
+        // beginSheetForDirectory: message has been deprecated since 10.6 
+        // Use the modern equivalent:
+        [sPanel setDirectoryURL:[NSURL fileURLWithPath:dir.AsNSString() isDirectory:YES]];
+        [sPanel setNameFieldStringValue:file.AsNSString()];
+        [sPanel beginSheetModalForWindow:nativeParent completionHandler:^(NSInteger result) {
+            if (result == NSFileHandlingPanelOKButton) {
+                NSLog(@"Saved file %@", [sPanel URL]);
+            }
+        }];
+#else
         [sPanel beginSheetForDirectory:dir.AsNSString() file:file.AsNSString()
             modalForWindow: nativeParent modalDelegate: nil // m_sheetDelegate
             didEndSelector: @selector(sheetDidEnd:returnCode:contextInfo:)
             contextInfo: nil];
+#endif
     }
     else 
     {
@@ -334,11 +353,22 @@ void FileDialog::ShowWindowModal()
         [oPanel setAllowsMultipleSelection: (HasFlag(wxFD_MULTIPLE) ? YES : NO )];
         
         NSWindow* nativeParent = parentWindow->GetWXWindow();
+#if wxCHECK_VERSION(3, 0, 3)
+        [oPanel setDirectoryURL:[NSURL fileURLWithPath:dir.AsNSString() isDirectory:YES]];
+        [oPanel setNameFieldStringValue:file.AsNSString()];
+        [oPanel setAllowedFileTypes:types];
+        [oPanel beginSheetModalForWindow:nativeParent completionHandler:^(NSInteger result) {
+            if (result == NSFileHandlingPanelOKButton) {
+                NSLog(@"Opened file(s) %@", [oPanel URLs]);
+            }
+        }];
+#else
         [oPanel beginSheetForDirectory:dir.AsNSString() file:file.AsNSString()
             types: types modalForWindow: nativeParent
             modalDelegate: nil // m_sheetDelegate
             didEndSelector: @selector(sheetDidEnd:returnCode:contextInfo:)
             contextInfo: nil];
+#endif
     }
 }
 
