diff --git a/CMakeLists.txt b/CMakeLists.txt
index eebcc2cdc..de2e76803 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -31,8 +31,8 @@ endif()
 # Increment as appropriate after release of a new version, and set back
 # AUDACITY_BUILD_LEVEL to 0
 set( AUDACITY_VERSION 3 )
-set( AUDACITY_RELEASE 6 )
-set( AUDACITY_REVISION 0 )
+set( AUDACITY_RELEASE @VERSION_MINOR@ )
+set( AUDACITY_REVISION @VERSION_PATCH@ )
 set( AUDACITY_MODLEVEL 0 )
 
 string( TIMESTAMP __TDATE__ "%Y%m%d" )
