diff --git a/cmake-proxies/cmake-modules/Findvst3sdk.cmake b/cmake-proxies/cmake-modules/Findvst3sdk.cmake
index dea6e1102..8d053faf1 100644
--- a/cmake-proxies/cmake-modules/Findvst3sdk.cmake
+++ b/cmake-proxies/cmake-modules/Findvst3sdk.cmake
@@ -47,7 +47,7 @@ if( NOT vst3sdk_FOUND )
       ${CMAKE_COMMAND}
          --build ${vst3sdk_BUILD}
          --config ${CMAKE_BUILD_TYPE}
-         --parallel
+         --parallel $ENV{BUILD_JOBS}
 
       RESULT_VARIABLE vst3sdk_build_result
    )
