diff --git a/src/effects/VST/VSTControlOSX.mm b/src/effects/VST/VSTControlOSX.mm
index d5c2b53366b85e773dbbdef3d6e2bb7aca92d0a6..98e5379984a37096c6ffd216133ab4db9e042222 100644
--- a/src/effects/VST/VSTControlOSX.mm
+++ b/src/effects/VST/VSTControlOSX.mm
@@ -45,7 +45,7 @@
 @end
 
 VSTControlImpl::VSTControlImpl(wxWindowMac *peer, NSView *view)
-:  wxWidgetCocoaImpl(peer, view, false, false)
+:  wxWidgetCocoaImpl(peer, view)
 {
 }
 
diff --git a/src/effects/audiounits/AUControl.mm b/src/effects/audiounits/AUControl.mm
index fdb6ae7ef1fa1838a2d28eb61150198eda6a1435..07d45f89e722abb4faec9c731bc10230a00df498 100644
--- a/src/effects/audiounits/AUControl.mm
+++ b/src/effects/audiounits/AUControl.mm
@@ -77,7 +77,7 @@
 @end
 
 AUControlImpl::AUControlImpl(wxWindowMac *peer, NSView *view)
-:  wxWidgetCocoaImpl(peer, view, false, false)
+:  wxWidgetCocoaImpl(peer, view)
 {
 }
 
diff --git a/src/prefs/PrefsDialog.cpp b/src/prefs/PrefsDialog.cpp
index 8eb749af78c9ef38fd5aee0429c1060ac067f72f..462189660d7487048b620678b838f605e81572dd 100644
--- a/src/prefs/PrefsDialog.cpp
+++ b/src/prefs/PrefsDialog.cpp
@@ -33,6 +33,7 @@
 #include <wx/listbook.h>
 
 #include <wx/treebook.h>
+#include <wx/treectrl.h>
 
 #include "../AudioIOBase.h"
 #include "../Prefs.h"
diff --git a/lib-src/FileDialog/mac/FileDialogPrivate.mm b/lib-src/FileDialog/mac/FileDialogPrivate.mm
index 508d8b8002dc18b17c8700f9eb7061c4178e0e6b..fca75e17da856475cb602d8d5040bcffce2fa7d5 100644
--- a/lib-src/FileDialog/mac/FileDialogPrivate.mm
+++ b/lib-src/FileDialog/mac/FileDialogPrivate.mm
@@ -47,7 +47,7 @@
 // implementation
 // ============================================================================
 
-@interface OSPanelDelegate : NSObject wxOSX_10_6_AND_LATER(<NSOpenSavePanelDelegate>)
+@interface OSPanelDelegate : NSObject <NSOpenSavePanelDelegate>
 {
     FileDialog* _dialog;
 }
