diff --git a/conan/helpers/profiles.py b/conan/helpers/profiles.py
index 1b5d8010c..daf6da5af 100644
--- a/conan/helpers/profiles.py
+++ b/conan/helpers/profiles.py
@@ -169,10 +169,12 @@ class Profiles:
             profile['settings']['compiler.runtime'] = "dynamic"
             if not is_host_profile:
                 profile['settings']['compiler.runtime_type'] = "Release"
-        elif args.compiler == 'GNU':
-            profile['settings']['compiler.libcxx'] = 'libstdc++11'
-        elif args.compiler == 'Clang' or args.compiler == 'AppleClang':
-            profile['settings']['compiler.libcxx'] = 'libc++'
+        else:
+            profile['settings']['compiler.libcxx'] = '@CXX_STDLIB@'
+#         elif args.compiler == 'GNU':
+#             profile['settings']['compiler.libcxx'] = 'libstdc++11'
+#         elif args.compiler == 'Clang' or args.compiler == 'AppleClang':
+#             profile['settings']['compiler.libcxx'] = 'libc++'
 
         if args.min_os_version and is_host_profile:
             # Only set the minimum OS version for host profiles
