diff --git a/cmake-proxies/cmake-modules/Findvst3sdk.cmake b/cmake-proxies/cmake-modules/Findvst3sdk.cmake
index dea6e110..917c96d1 100644
--- a/cmake-proxies/cmake-modules/Findvst3sdk.cmake
+++ b/cmake-proxies/cmake-modules/Findvst3sdk.cmake
@@ -28,6 +28,10 @@ if( NOT vst3sdk_FOUND )
          -G ${CMAKE_GENERATOR}
          -S ${vst3sdk_DIR}
          -B ${vst3sdk_BUILD}
+	    -D CMAKE_CXX_COMPILER_LAUNCHER=${CMAKE_CXX_COMPILER_LAUNCHER}
+	    -D CMAKE_OBJCXX_COMPILER_LAUNCHER=${CMAKE_OBJCXX_COMPILER_LAUNCHER}
+	    -D CMAKE_COLOR_DIAGNOSTICS=${CMAKE_COLOR_DIAGNOSTICS}
+	    -D CMAKE_COLOR_MAKEFILE=${CMAKE_COLOR_MAKEFILE}
          -D CMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
          -D SMTG_ENABLE_VST3_HOSTING_EXAMPLES=OFF
          -D SMTG_ENABLE_VST3_PLUGIN_EXAMPLES=OFF
@@ -47,7 +51,7 @@ if( NOT vst3sdk_FOUND )
       ${CMAKE_COMMAND}
          --build ${vst3sdk_BUILD}
          --config ${CMAKE_BUILD_TYPE}
-         --parallel
+         --parallel $ENV{BUILD_JOBS}
 
       RESULT_VARIABLE vst3sdk_build_result
    )
