diff --git a/src/Theme.cpp b/src/Theme.cpp
index 386d4b0..7af4cab 100644
--- a/src/Theme.cpp
+++ b/src/Theme.cpp
@@ -382,7 +382,7 @@ void ThemeBase::RegisterImage( int &iIndex, const wxImage &Image, const wxString
    wxASSERT( iIndex == -1 ); // Don't initialise same bitmap twice!
    mImages.Add( Image );
 
-#ifdef __WXMAC__
+#ifdef __APPLE__
    // On Mac, bitmaps with alpha don't work.
    // So we convert to a mask and use that.
    // It isn't quite as good, as alpha gives smoother edges.
diff --git a/src/widgets/Ruler.cpp b/src/widgets/Ruler.cpp
index d5cd00b..82aceaf 100644
--- a/src/widgets/Ruler.cpp
+++ b/src/widgets/Ruler.cpp
@@ -1692,7 +1692,7 @@ void AdornedRulerPanel::OnErase(wxEraseEvent & WXUNUSED(evt))
 
 void AdornedRulerPanel::OnPaint(wxPaintEvent & WXUNUSED(evt))
 {
-#if defined(__WXMAC__)
+#if defined(__APPLE__)
    wxPaintDC dc(this);
 #else
    wxBufferedPaintDC dc(this);
diff --git a/src/widgets/Meter.cpp b/src/widgets/Meter.cpp
index 2a406ed..3156f11 100644
--- a/src/widgets/Meter.cpp
+++ b/src/widgets/Meter.cpp
@@ -39,6 +39,7 @@
 *//******************************************************************/
 
 #include "../Audacity.h"
+#undef USE_AQUA_THEME
 #include "../AudacityApp.h"
 
 #include <wx/defs.h>
