diff --git a/src/ShuttleGui.h b/src/ShuttleGui.h
index 12390c957590bec555f37170b0fa252582e5b36a..2cd7b101f09db964e99c22d3fb655d12fd89025e 100644
--- a/src/ShuttleGui.h
+++ b/src/ShuttleGui.h
@@ -86,7 +86,9 @@ public:
    wxStaticBoxWrapper( Args &&...args )
       : wxStaticBox( std::forward<Args>(args)... )
    {
-      m_container.EnableSelfFocus();
+#if wxCHECK_VERSION(3, 1, 1)
+       m_container.EnableSelfFocus();
+#endif
    }
 };
 
diff --git a/src/widgets/FileDialog/mac/FileDialogPrivate.h b/src/widgets/FileDialog/mac/FileDialogPrivate.h
index 7fe79111c3c4bc30bae9470f09b1a91ec04cacb4..7453bedbb3be285feeda8eca85e0e3c5a0d0e85a 100644
--- a/src/widgets/FileDialog/mac/FileDialogPrivate.h
+++ b/src/widgets/FileDialog/mac/FileDialogPrivate.h
@@ -96,8 +96,10 @@ protected:
     int m_firstFileTypeFilter;
     wxArrayString m_currentExtensions;
     WX_NSObject m_delegate;
+#if !wxCHECK_VERSION(3, 0, 3)
     WX_NSObject m_sheetDelegate;
 #endif
+#endif
 
 private:
     // Common part of all ctors.
