diff --git a/cmake-proxies/cmake-modules/Findvst3sdk.cmake b/cmake-proxies/cmake-modules/Findvst3sdk.cmake
index dea6e110..6a2884df 100644
--- a/cmake-proxies/cmake-modules/Findvst3sdk.cmake
+++ b/cmake-proxies/cmake-modules/Findvst3sdk.cmake
@@ -58,8 +58,9 @@ if( NOT vst3sdk_FOUND )
 
    add_library(vst3sdk::base STATIC IMPORTED GLOBAL)
    add_library(vst3sdk::pluginterfaces STATIC IMPORTED GLOBAL)
-   add_library(vst3sdk::sdk_hosting STATIC IMPORTED GLOBAL)
-   add_library(vst3sdk::sdk_common STATIC IMPORTED GLOBAL)
+   add_library(vst3sdk::sdk STATIC IMPORTED GLOBAL)
+#    add_library(vst3sdk::sdk_hosting STATIC IMPORTED GLOBAL)
+#    add_library(vst3sdk::sdk_common STATIC IMPORTED GLOBAL)
 
    set_target_properties(vst3sdk::base PROPERTIES
       IMPORTED_LOCATION ${vst3sdk_BUILD}/lib/${CMAKE_BUILD_TYPE}/${CMAKE_STATIC_LIBRARY_PREFIX}base${CMAKE_STATIC_LIBRARY_SUFFIX}
@@ -71,15 +72,20 @@ if( NOT vst3sdk_FOUND )
       INTERFACE_INCLUDE_DIRECTORIES ${vst3sdk_DIR}
    )
 
-   set_target_properties(vst3sdk::sdk_hosting PROPERTIES
-      IMPORTED_LOCATION ${vst3sdk_BUILD}/lib/${CMAKE_BUILD_TYPE}/${CMAKE_STATIC_LIBRARY_PREFIX}sdk_hosting${CMAKE_STATIC_LIBRARY_SUFFIX}
+   set_target_properties(vst3sdk::sdk PROPERTIES
+      IMPORTED_LOCATION ${vst3sdk_BUILD}/lib/${CMAKE_BUILD_TYPE}/${CMAKE_STATIC_LIBRARY_PREFIX}sdk${CMAKE_STATIC_LIBRARY_SUFFIX}
       INTERFACE_INCLUDE_DIRECTORIES ${vst3sdk_DIR}
    )
 
-   set_target_properties(vst3sdk::sdk_common PROPERTIES
-      IMPORTED_LOCATION ${vst3sdk_BUILD}/lib/${CMAKE_BUILD_TYPE}/${CMAKE_STATIC_LIBRARY_PREFIX}sdk_common${CMAKE_STATIC_LIBRARY_SUFFIX}
-      INTERFACE_INCLUDE_DIRECTORIES ${vst3sdk_DIR}
-   )
+#    set_target_properties(vst3sdk::sdk_hosting PROPERTIES
+#       IMPORTED_LOCATION ${vst3sdk_BUILD}/lib/${CMAKE_BUILD_TYPE}/${CMAKE_STATIC_LIBRARY_PREFIX}sdk_hosting${CMAKE_STATIC_LIBRARY_SUFFIX}
+#       INTERFACE_INCLUDE_DIRECTORIES ${vst3sdk_DIR}
+#    )
+# 
+#    set_target_properties(vst3sdk::sdk_common PROPERTIES
+#       IMPORTED_LOCATION ${vst3sdk_BUILD}/lib/${CMAKE_BUILD_TYPE}/${CMAKE_STATIC_LIBRARY_PREFIX}sdk_common${CMAKE_STATIC_LIBRARY_SUFFIX}
+#       INTERFACE_INCLUDE_DIRECTORIES ${vst3sdk_DIR}
+#    )
 
    find_package(X11 REQUIRED)
 
diff --git a/libraries/lib-vst3/CMakeLists.txt b/libraries/lib-vst3/CMakeLists.txt
index 9a2c45b6..104358c4 100644
--- a/libraries/lib-vst3/CMakeLists.txt
+++ b/libraries/lib-vst3/CMakeLists.txt
@@ -34,8 +34,7 @@ set( SOURCES
 set( LIBRARIES
    vst3sdk::base
    vst3sdk::pluginterfaces
-   vst3sdk::sdk_hosting
-   vst3sdk::sdk_common
+   vst3sdk::sdk
    lib-effects-interface
 )
 audacity_library( lib-vst3 "${SOURCES}" "${LIBRARIES}"
