diff --git a/src/Makefile.am b/src/Makefile.am
index 75c7691b7bd7871b5cba2e76039387ce9d7a6a1b..ca68c32fd746894ed16caf6d0866ad6fb03d3886 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -79,8 +79,9 @@ audacity_CPPFLAGS = \
 #  /usr/local/include/wx-2.8/wx/buffer.h:289:9: note: in expansion of macro 'wxCHECK_RET'
 #           wxCHECK_RET( m_bufdata->m_data, wxT("invalid wxMemoryBuffer") );
 #           ^
-audacity_CXXFLAGS = \
-   -Wno-unused-local-typedefs
+#audacity_CXXFLAGS = \
+#   -Wno-unused-local-typedefs
+audacity_CXXFLAGS =
 
 audacity_LDFLAGS = -rdynamic
 audacity_LDADD = \
diff --git a/src/Makefile.in b/src/Makefile.in
index 26bb40df3971bd9e412e17d379d68981585ea3ab..75db030595910371a7c449acf5a9128d4cb5a73a 100644
--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -1324,8 +1324,9 @@ audacity_CPPFLAGS = -std=c++11 -Wno-deprecated-declarations \
 #  /usr/local/include/wx-2.8/wx/buffer.h:289:9: note: in expansion of macro 'wxCHECK_RET'
 #           wxCHECK_RET( m_bufdata->m_data, wxT("invalid wxMemoryBuffer") );
 #           ^
-audacity_CXXFLAGS = \
-   -Wno-unused-local-typedefs
+#audacity_CXXFLAGS = \
+#   -Wno-unused-local-typedefs
+audacity_CXXFLAGS =
 
 audacity_LDFLAGS = -rdynamic
 audacity_LDADD = $(EXPAT_LIBS) $(FILEDIALOG_LIBS) $(PORTAUDIO_LIBS) \
