diff --git a/src/previewclient.cpp b/src/previewclient.cpp
index 696456b..aea2fbf 100644
--- a/src/previewclient.cpp
+++ b/src/previewclient.cpp
@@ -135,6 +135,12 @@ int PreviewClient::height() const
     return m_height;
 }
 
+QSize PreviewClient::size() const
+{
+    QSize size(m_width, m_height);
+    return size;
+}
+
 QString PreviewClient::caption() const
 {
     return m_caption;
diff --git a/src/previewclient.h b/src/previewclient.h
index 7b12701..526b60e 100644
--- a/src/previewclient.h
+++ b/src/previewclient.h
@@ -93,6 +93,7 @@ public:
 
     int width() const override;
     int height() const override;
+    QSize size() const override;
     QPalette palette() const override;
     Qt::Edges adjacentScreenEdges() const override;
 
