diff --git a/xvisbell.cpp b/xvisbell.cpp
index 04bc767..486906d 100644
--- a/xvisbell.cpp
+++ b/xvisbell.cpp
@@ -37,7 +37,7 @@ const struct timeval window_timeout = {0, 100000};
 struct {
   int x, y;
   int w, h;
-} geometry = {0, 0, -1, 16};
+} geometry = {0, 0, -1, -1};
 
 bool operator<(const struct timeval & a,
                const struct timeval & b) {
@@ -61,6 +61,7 @@ int main() {
   if (!dpy) {
     throw std::runtime_error("XOpenDisplay() error");
   }
+  XSynchronize(dpy, true);
 
   auto scr = XDefaultScreen(dpy);
   auto root = XRootWindow(dpy, scr);
