diff --git a/module/os/macos/spl/spl-thread.c b/module/os/macos/spl/spl-thread.c
index c87033087..1aec28afa 100644
--- a/module/os/macos/spl/spl-thread.c
+++ b/module/os/macos/spl/spl-thread.c
@@ -195,6 +195,7 @@ void
 set_thread_throughput_named(thread_t thread,
     thread_throughput_qos_t throughput, const char *name)
 {
+#if defined(MAC_OS_X_VERSION_10_10) && (MAC_OS_X_VERSION_MIN_REQUIRED >= MAC_OS_X_VERSION_10_10)
 	/*
 	 * TIERs: 0 is USER_INTERACTIVE, 1 is USER_INITIATED, 1 is LEGACY,
 	 *        2 is UTILITY, 5 is BACKGROUND, 5 is MAINTENANCE
@@ -216,6 +217,7 @@ set_thread_throughput_named(thread_t thread,
 		    __func__, __LINE__, qoskret,
 		    qosp.thread_throughput_qos_tier, name);
 	}
+#endif
 }
 
 void
@@ -230,6 +232,7 @@ void
 set_thread_latency_named(thread_t thread,
     thread_latency_qos_t latency, const char *name)
 {
+#if defined(MAC_OS_X_VERSION_10_10) && (MAC_OS_X_VERSION_MIN_REQUIRED >= MAC_OS_X_VERSION_10_10)
 	/*
 	 * TIERs: 0 is USER_INTERACTIVE, 1 is USER_INITIATED, 1 is LEGACY,
 	 *        3 is UTILITY, 3 is BACKGROUND, 5 is MAINTENANCE
@@ -252,6 +255,7 @@ set_thread_latency_named(thread_t thread,
 		    qoskret, qosp.thread_latency_qos_tier,
 		    name);
 	}
+#endif
 }
 
 void
