diff --git a/lib/libzfs/os/macos/libzfs_util_os.c b/lib/libzfs/os/macos/libzfs_util_os.c
index a0b43466639c081f94d5b9658240f3c961b5e100..d47a1a546cf2f168b5df3a72c61b1de6dba4360b 100644
--- a/lib/libzfs/os/macos/libzfs_util_os.c
+++ b/lib/libzfs/os/macos/libzfs_util_os.c
@@ -101,13 +101,16 @@ libzfs_module_loaded(const char *module)
  * - ZFS_MODULE_TIMEOUT="<seconds>"     - Seconds to wait for ZFS_DEV
  */
 static int
-libzfs_load_module_impl(const char *module)
+libzfs_load_module_impl()
 {
-	char *argv[4] = {"/sbin/kextload", (char *)module, (char *)0};
+	const char *module = ZFS_DRIVER;
+	char kext[(sizeof(KERNEL_MODPREFIX) + sizeof(ZFS_DRIVER)) / sizeof(char) + 7] =
+		KERNEL_MODPREFIX "/" ZFS_DRIVER ".kext";
+	char *argv[4] = {"/sbin/kextload", "-v", kext, (char *)0};
 	char *load_str, *timeout_str;
 	long timeout = 10; /* seconds */
 	long busy_timeout = 10; /* milliseconds */
-	int load = 0, fd;
+	int load = 1, fd;
 	hrtime_t start;
 
 	/* Optionally request module loading */
@@ -122,7 +125,9 @@ libzfs_load_module_impl(const char *module)
 		}
 
 		if (load) {
-			if (libzfs_run_process("/sbin/kextload", argv, 0))
+// 			fprintf( stderr, dgettext(TEXT_DOMAIN, "Executing `%s %s %s`\n"),
+// 				 argv[0], argv[1], argv[2] );
+			if (libzfs_run_process("/sbin/kextload", argv, STDERR_VERBOSE|STDOUT_VERBOSE))
 				return (ENOEXEC);
 		}
 
@@ -169,7 +174,7 @@ libzfs_load_module(void)
 	// Using this as a libzfs_init_os() - we should probably do it properly
 	libdiskmgt_init();
 
-	return (libzfs_load_module_impl(ZFS_DRIVER));
+	return (libzfs_load_module_impl());
 }
 
 int
@@ -187,6 +192,10 @@ zfs_version_kernel(char *version, int len)
 {
 	size_t rlen = len;
 
+	if (!libzfs_module_loaded(ZFS_DRIVER)) {
+		libzfs_load_module();
+		sleep(1);
+	}
 	if (sysctlbyname("zfs.kext_version",
 	    version, &rlen, NULL, 0) == -1)
 		return (-1);
