diff --git a/Makefile.am b/Makefile.am
index f3317308addc7f437f21387ad6670761bdd402d7..a484dec86bb8e341a0010cb27db4538379516999 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -74,20 +74,20 @@ pkg-utils: @DEFAULT_PACKAGE@-utils
 
 if CONFIG_USER
 install-data-hook::
-	@echo
-	if [ -z "$(DESTDIR)" ] || [ x"$(DESTDIR)" = x"/" ]; then \
-		for plistfile in $(launchddaemon_DATA); do \
-			bn=`basename "$$plistfile"`; \
-			echo "Attempting to unload $$bn ..."; \
-			$(DESTDIR)/bin/launchctl unload -F \
-			$(DESTDIR)$(launchddaemondir)/"$$bn" \
-			    || echo "Ignoring errors"; \
-			echo "Attempting to load $$bn ..."; \
-			$(DESTDIR)/bin/launchctl load -wF \
-			$(DESTDIR)$(launchddaemondir)/"$$bn" \
-			    || echo "Ignoring errors"; \
-		done \
-	fi
+# 	@echo
+# 	if [ -z "$(DESTDIR)" ] || [ x"$(DESTDIR)" = x"/" ]; then \
+# 		for plistfile in $(launchddaemon_DATA); do \
+# 			bn=`basename "$$plistfile"`; \
+# 			echo "Attempting to unload $$bn ..."; \
+# 			$(DESTDIR)/bin/launchctl unload -F \
+# 			$(DESTDIR)$(launchddaemondir)/"$$bn" \
+# 			    || echo "Ignoring errors"; \
+# 			echo "Attempting to load $$bn ..."; \
+# 			$(DESTDIR)/bin/launchctl load -wF \
+# 			$(DESTDIR)$(launchddaemondir)/"$$bn" \
+# 			    || echo "Ignoring errors"; \
+# 		done \
+# 	fi
 	echo "Install done."
 
 uninstall-hook::
