diff --git a/config/user-libblkid.m4 b/config/user-libblkid.m4
index 2dd26238c..ecfd3972e 100644
--- a/config/user-libblkid.m4
+++ b/config/user-libblkid.m4
@@ -35,8 +35,8 @@ AC_DEFUN([ZFS_AC_CONFIG_USER_LIBBLKID], [
 		[
 			AC_MSG_CHECKING([for blkid zfs support])
 
-			ZFS_DEV=`mktemp`
-			truncate -s 64M $ZFS_DEV
+			ZFS_DEV=`gmktemp`
+			gtruncate -s 64M $ZFS_DEV
 			echo -en "\x0c\xb1\xba\0\0\0\0\0" | \
 				dd of=$ZFS_DEV bs=1k count=8 \
 				seek=128 conv=notrunc &>/dev/null \
