diff --git libs/kopageapp/KoPAView.cpp libs/kopageapp/KoPAView.cpp
index fa0e381..444f0b4 100644
--- libs/kopageapp/KoPAView.cpp
+++ libs/kopageapp/KoPAView.cpp
@@ -427,6 +427,10 @@ void KoPAView::initActions()
     d->actionConfigure = new KAction(koIcon("configure"), i18n("Configure..."), this);
     actionCollection()->addAction("configure", d->actionConfigure);
     connect(d->actionConfigure, SIGNAL(triggered()), this, SLOT(configure()));
+    // not sure why this isn't done through KStandardAction, but since it isn't
+    // we ought to set the MenuRole manually so the item ends up in the appropriate
+    // menu on OS X:
+    d->actionConfigure->setMenuRole(QAction::PreferencesRole);
 
     d->find = new KoFind( this, d->canvas->resourceManager(), actionCollection() );
     connect( d->find, SIGNAL( findDocumentSetNext( QTextDocument * ) ),
