diff --git a/qtbase/mkspecs/features/qt_common.prf b/qtbase/mkspecs/features/qt_common.prf
index d30983f..2fbcf89 100644
--- a/qtbase/mkspecs/features/qt_common.prf
+++ b/qtbase/mkspecs/features/qt_common.prf
@@ -52,9 +52,9 @@ host_build:cross_compile: return()
 # -Wdate-time: warn if we use __DATE__ or __TIME__ (we want to be able to reproduce the exact same binary)
 # -Wvla: use of variable-length arrays (an extension to C++)
 clang {
-    # Clang 3.5 introduced -Wdate-time
+    # Clang 3.5 introduced -Wdate-time but Apple's clang on OS X 10.9 doesn't know it yet
     # The conditional assumes we aren't compiling against Clang 2.x anymore
-    greaterThan(QT_CLANG_MAJOR_VERSION, 3)|greaterThan(QT_CLANG_MINOR_VERSION, 4): \
+    greaterThan(QT_CLANG_MAJOR_VERSION, 3)|greaterThan(QT_CLANG_MINOR_VERSION, 5): \
         QMAKE_CXXFLAGS_WARN_ON += -Wdate-time
 } else: gcc:!intel_icc {
     QMAKE_CXXFLAGS_WARN_ON += -Wvla
