diff --git a/qtbase/src/platformsupport/themes/genericunix/qgenericunixthemes.cpp b/qtbase/src/platformsupport/themes/genericunix/qgenericunixthemes.cpp
index 1e1b1af..e7647e4 100644
--- a/qtbase/src/platformsupport/themes/genericunix/qgenericunixthemes.cpp
+++ b/qtbase/src/platformsupport/themes/genericunix/qgenericunixthemes.cpp
@@ -48,6 +48,8 @@
 #include <QtCore/QFileInfo>
 #include <QtCore/QFile>
 #include <QtCore/QDebug>
+#undef QT_NO_DEBUG
+#undef QT_NO_DEBUG_OUTPUT
 #include <QtCore/QHash>
 #include <QtCore/QMimeDatabase>
 #include <QtCore/QLoggingCategory>
@@ -59,6 +61,14 @@
 #include <qpa/qplatformintegration.h>
 #include <qpa/qplatformservices.h>
 #include <qpa/qplatformdialoghelper.h>
+
+// FIXME: find a more elegant solution?
+#ifdef Q_OS_MACOS
+// platformsupport plugins shouldn't use DBus on OS X
+// and systray functionality doesn't depend on it
+#define QT_NO_DBUS
+#endif
+
 #ifndef QT_NO_DBUS
 #include "qdbusplatformmenu_p.h"
 #include "qdbusmenubar_p.h"
@@ -186,7 +186,7 @@ QStringList QGenericUnixTheme::xdgIconThemePaths()
 
     QString xdgDirString = QFile::decodeName(qgetenv("XDG_DATA_DIRS"));
     if (xdgDirString.isEmpty())
-        xdgDirString = QLatin1String("/usr/local/share/:/usr/share/");
+        xdgDirString = QLatin1String("@PREFIX@/share/:/usr/share/");
     const auto xdgDirs = xdgDirString.splitRef(QLatin1Char(':'));
     for (const QStringRef &xdgDir : xdgDirs) {
         const QFileInfo xdgIconsDir(xdgDir + QLatin1String("/icons"));
@@ -194,7 +194,7 @@ QStringList QGenericUnixTheme::xdgIconThemePaths()
             paths.append(xdgIconsDir.absoluteFilePath());
     }
 
-    const QFileInfo pixmapsIconsDir(QStringLiteral("/usr/share/pixmaps"));
+    const QFileInfo pixmapsIconsDir(QStringLiteral("@PREFIX@/share/pixmaps"));
     if (pixmapsIconsDir.isDir())
         paths.append(pixmapsIconsDir.absoluteFilePath());
 
@@ -276,7 +276,9 @@ public:
         , toolBarIconSize(0)
         , singleClick(true)
         , wheelScrollLines(3)
-    { }
+    {
+        qWarning() << Q_FUNC_INFO << "kdeVersion=" << kdeVersion << "kdeDirs=" << kdeDirs;
+    }
 
     static QString kdeGlobals(const QString &kdeDir, int kdeVersion)
     {
@@ -792,7 +802,9 @@ QPlatformTheme *QGenericUnixTheme::createUnixTheme(const QString &name)
 #endif
     if (name == QLatin1String(QGnomeTheme::name))
         return new QGnomeTheme;
-    return Q_NULLPTR;
+//     return Q_NULLPTR;
+    // On OS X prefer to return something valid always?
+    return new QGenericUnixTheme;
 }
 
 QStringList QGenericUnixTheme::themeNames()
diff --git a/qtbase/src/platformsupport/themes/genericunix/qgenericunixthemes_p.h b/qtbase/src/platformsupport/themes/genericunix/qgenericunixthemes_p.h
index da13390..f0ca37a 100644
--- a/qtbase/src/platformsupport/themes/genericunix/qgenericunixthemes_p.h
+++ b/qtbase/src/platformsupport/themes/genericunix/qgenericunixthemes_p.h
@@ -85,10 +85,10 @@ public:
     QVariant themeHint(ThemeHint hint) const Q_DECL_OVERRIDE;
 
     static QStringList xdgIconThemePaths();
-#ifndef QT_NO_DBUS
+#if !defined(QT_NO_DBUS) && !defined(Q_OS_MACOS)
     QPlatformMenuBar *createPlatformMenuBar() const Q_DECL_OVERRIDE;
 #endif
-#if !defined(QT_NO_DBUS) && !defined(QT_NO_SYSTEMTRAYICON)
+#if !defined(QT_NO_DBUS) && !defined(QT_NO_SYSTEMTRAYICON) && !defined(Q_OS_MACOS)
     QPlatformSystemTrayIcon *createPlatformSystemTrayIcon() const Q_DECL_OVERRIDE;
 #endif
 
@@ -113,10 +113,10 @@ public:
     const QPalette *palette(Palette type = SystemPalette) const Q_DECL_OVERRIDE;
 
     const QFont *font(Font type) const Q_DECL_OVERRIDE;
-#ifndef QT_NO_DBUS
+#if !defined(QT_NO_DBUS) && !defined(Q_OS_MACOS)
     QPlatformMenuBar *createPlatformMenuBar() const Q_DECL_OVERRIDE;
 #endif
-#if !defined(QT_NO_DBUS) && !defined(QT_NO_SYSTEMTRAYICON)
+#if !defined(QT_NO_DBUS) && !defined(QT_NO_SYSTEMTRAYICON) && !defined(Q_OS_MACOS)
     QPlatformSystemTrayIcon *createPlatformSystemTrayIcon() const Q_DECL_OVERRIDE;
 #endif
 
@@ -138,10 +138,10 @@ public:
     QString standardButtonText(int button) const Q_DECL_OVERRIDE;
 
     virtual QString gtkFontName() const;
-#ifndef QT_NO_DBUS
+#if !defined(QT_NO_DBUS) && !defined(Q_OS_MACOS)
     QPlatformMenuBar *createPlatformMenuBar() const Q_DECL_OVERRIDE;
 #endif
-#if !defined(QT_NO_DBUS) && !defined(QT_NO_SYSTEMTRAYICON)
+#if !defined(QT_NO_DBUS) && !defined(QT_NO_SYSTEMTRAYICON) && !defined(Q_OS_MACOS)
     QPlatformSystemTrayIcon *createPlatformSystemTrayIcon() const Q_DECL_OVERRIDE;
 #endif
 
diff --git a/qtbase/src/platformsupport/themes/themes.pro b/qtbase/src/platformsupport/themes/themes.pro
index 2aeb1f8..4b51b5b 100644
--- a/qtbase/src/platformsupport/themes/themes.pro
+++ b/qtbase/src/platformsupport/themes/themes.pro
@@ -7,7 +7,7 @@ CONFIG += static internal_module
 DEFINES += QT_NO_CAST_FROM_ASCII
 PRECOMPILED_HEADER = ../../corelib/global/qt_pch.h
 
-unix:!darwin: \
+unix: \
     include($$PWD/genericunix/genericunix.pri)
 
 HEADERS += \
diff --git a/qtbase/src/plugins/platforms/cocoa/qcocoaintegration.mm b/qtbase/src/plugins/platforms/cocoa/qcocoaintegration.mm
index 92fffb4..91fcd1c 100644
--- a/qtbase/src/plugins/platforms/cocoa/qcocoaintegration.mm
+++ b/qtbase/src/plugins/platforms/cocoa/qcocoaintegration.mm
@@ -62,6 +62,8 @@
 
 #include <IOKit/graphics/IOGraphicsLib.h>
 
+#include <private/qgenericunixthemes_p.h>
+
 static void initResources()
 {
     Q_INIT_RESOURCE(qcocoaresources);
@@ -546,14 +548,33 @@ QCocoaDrag *QCocoaIntegration::drag() const
 
 QStringList QCocoaIntegration::themeNames() const
 {
-    return QStringList(QLatin1String(QCocoaTheme::name));
+    QStringList themes;
+    themes.push_back(QLatin1String(QCocoaTheme::name));
+    const QByteArray kdeSessionVersion = qgetenv("KDE_SESSION_VERSION");
+    const int kdeVersion = kdeSessionVersion.toInt();
+    if (kdeVersion >= 4)
+        themes.push_back(QLatin1String("kde"));
+    return themes;
 }
 
 QPlatformTheme *QCocoaIntegration::createPlatformTheme(const QString &name) const
 {
-    if (name == QLatin1String(QCocoaTheme::name))
+    // qWarning() << "QCocoaIntegration::createPlatformTheme(" << name << ")";
+    if (name == QLatin1String(QCocoaTheme::name)) {
+        // qWarning() << "returning the QCocoaTheme";
         return new QCocoaTheme;
-    return QPlatformIntegration::createPlatformTheme(name);
+    }
+    QPlatformTheme *unixTheme = QGenericUnixTheme::createUnixTheme(name);
+    if (unixTheme) {
+        qDebug() << "returning QGenericUnixTheme" << unixTheme;
+        return unixTheme;
+    }
+    else {
+        unixTheme = QPlatformIntegration::createPlatformTheme(name);
+        qDebug() << "returning QPlatformIntegration::createPlatformTheme(" << name << ")=" << unixTheme;
+        return unixTheme;
+//         return QPlatformIntegration::createPlatformTheme(name);
+    }
 }
 
 QCocoaServices *QCocoaIntegration::services() const
diff --git a/qtbase/src/widgets/styles/qmacstyle_mac.mm b/qtbase/src/widgets/styles/qmacstyle_mac.mm
index 007ce20..dedd142 100644
--- a/qtbase/src/widgets/styles/qmacstyle_mac.mm
+++ b/qtbase/src/widgets/styles/qmacstyle_mac.mm
@@ -4218,7 +4218,10 @@ void QMacStyle::drawControl(ControlElement ce, const QStyleOption *opt, QPainter
             // (QComboMenuItem is internal and should never be seen by the
             // outside world, unless they read the source, in which case, it's
             // their own fault).
-            bool nonDefaultFont = p->font() != qt_app_fonts_hash()->value("QComboMenuItem");
+//             bool nonDefaultFont = p->font() != qt_app_fonts_hash()->value("QComboMenuItem");
+            // We support non-native themes in this patched Qt version, so we cannot rely on the result
+            // of the test above. It's best to presume it always returns true.
+            bool nonDefaultFont = true;
             bool isSelectedAndNeedsShadow = selected && !usingYosemiteOrLater;
             if (isSelectedAndNeedsShadow || verticalTabs || nonDefaultFont || !tab->icon.isNull()
                 || !myTab.leftButtonSize.isEmpty() || !myTab.rightButtonSize.isEmpty()) {
