diff --git a/qttools/src/assistant/help/qhelpdbreader.cpp b/qttools/src/assistant/help/qhelpdbreader.cpp
index 8cbf88f1cf8d6e1395dfafb85cebb0fd6bf1cd99..77777c2598392bfc79215aae8784aa6945583a70 100644
--- a/qttools/src/assistant/help/qhelpdbreader.cpp
+++ b/qttools/src/assistant/help/qhelpdbreader.cpp
@@ -87,6 +87,19 @@ bool QHelpDBReader::init()
     if (!QFile::exists(m_dbName))
         return false;
 
+    if (!initDB()) {
+        QSqlDatabase::removeDatabase(m_uniqueId);
+        return false;
+    }
+
+    m_initDone = true;
+    m_query = new QSqlQuery(QSqlDatabase::database(m_uniqueId));
+
+    return true;
+}
+
+bool QHelpDBReader::initDB()
+{
     QSqlDatabase db = QSqlDatabase::addDatabase(QLatin1String("QSQLITE"), m_uniqueId);
     db.setConnectOptions(QLatin1String("QSQLITE_OPEN_READONLY"));
     db.setDatabaseName(m_dbName);
@@ -95,13 +108,8 @@ bool QHelpDBReader::init()
                                   %2 - The unique id for the connection
                                   %3 - The actual error string */
         m_error = tr("Cannot open database '%1' '%2': %3").arg(m_dbName, m_uniqueId, db.lastError().text());
-        QSqlDatabase::removeDatabase(m_uniqueId);
         return false;
     }
-
-    m_initDone = true;
-    m_query = new QSqlQuery(db);
-
     return true;
 }
 
