diff --git a/qtbase/src/corelib/global/qglobal.h b/qtbase/src/corelib/global/qglobal.h
index 3217691..36f1cae 100644
--- a/qtbase/src/corelib/global/qglobal.h
+++ b/qtbase/src/corelib/global/qglobal.h
@@ -621,6 +621,13 @@ class QDataStream;
 
 inline void qt_noop(void) {}
 
+/* 
+   Make sure we can test for exceptions reliably and easily
+*/
+#if !defined(__clang__) && !defined(__has_feature)
+#  define __has_feature(x) 0
+#endif
+
 /* These wrap try/catch so we can switch off exceptions later.
 
    Beware - do not use more than one QT_CATCH per QT_TRY, and do not use
@@ -629,10 +636,13 @@ inline void qt_noop(void) {}
    Use the QT_NO_EXCEPTIONS macro to protect your code instead.
 */
 
-#if !defined(QT_NO_EXCEPTIONS)
-#  if defined(QT_BOOTSTRAPPED) || (defined(Q_CC_GNU) && !defined (__EXCEPTIONS) && !defined(Q_MOC_RUN))
+#if !defined(QT_NO_EXCEPTIONS) && !defined(Q_MOC_RUN)
+#  if (defined(Q_CC_CLANG) && !defined(Q_CC_INTEL) && !__has_feature(cxx_exceptions)) || \
+      (defined(Q_CC_GNU) && !defined(__EXCEPTIONS))
 #    define QT_NO_EXCEPTIONS
 #  endif
+#elseif defined(QT_BOOTSTRAPPED)
+#  define QT_NO_EXCEPTIONS
 #endif
 
 #ifdef QT_NO_EXCEPTIONS
