--- qt5kdweb-mp9-work/qtwebengine/src/3rdparty/chromium/base/process/memory_mac.mm.orig	2016-12-28 18:23:48.000000000 +0100
+++ qt5kdweb-mp9-work/qtwebengine/src/3rdparty/chromium/base/process/memory_mac.mm	2016-12-28 18:21:18.000000000 +0100
@@ -248,7 +248,8 @@
 // === Core Foundation CFAllocators ===
 
 bool CanGetContextForCFAllocator() {
-  return !base::mac::IsOSLaterThanSierra_DontCallThis();
+//   return !base::mac::IsOSLaterThanSierra_DontCallThis();
+    return false;
 }
 
 CFAllocatorContext* ContextForCFAllocator(CFAllocatorRef allocator) {
@@ -256,6 +257,7 @@
     ChromeCFAllocatorLeopards* our_allocator =
         const_cast<ChromeCFAllocatorLeopards*>(
             reinterpret_cast<const ChromeCFAllocatorLeopards*>(allocator));
+    NSLog(@"ContextForCFAllocator:%d(%@)=0x%lx", __LINE__, allocator, &our_allocator->_context);
     return &our_allocator->_context;
   } else if (base::mac::IsOSLion() ||
              base::mac::IsOSMountainLion() ||
@@ -266,6 +268,7 @@
     ChromeCFAllocatorLions* our_allocator =
         const_cast<ChromeCFAllocatorLions*>(
             reinterpret_cast<const ChromeCFAllocatorLions*>(allocator));
+    NSLog(@"ContextForCFAllocator:%d(%@)=0x%lx", __LINE__, allocator, &our_allocator->_context);
     return &our_allocator->_context;
   } else {
     return NULL;
