diff --git app/kdevelop! app/kdevelop!
index 88fb659..ce9c507 100755
--- app/kdevelop!
+++ app/kdevelop!
@@ -28,7 +28,7 @@ if ! [ "$KDEV_ATTACHED_PID" ]; then
     else
         ARGS="$@"
     fi
-    export KDEV_ATTACHED_PID=$($KDEV_BASEDIR/kdevelop $ARGS --pid)
+    export KDEV_ATTACHED_PID=$($KDEV_BASEDIR/kdevelop5 $ARGS --pid)
 fi
 
 if ! [ "$KDEV_ATTACHED_PID" ]; then
@@ -52,5 +52,5 @@ export KDEV_DBUS_ID="org.kdevelop.kdevelop-$KDEV_ATTACHED_PID"
 
 case $(basename $SHELL) in
     zsh) OLD_ZDOTDIR="${ZDOTDIR-$HOME}" ZDOTDIR="$KDEV_BASEDIR/../share/kdevplatform/shellutils/" zsh -i ;;
-    *) bash --init-file "$KDEV_BASEDIR/kdevplatform_shell_environment.sh" -i ;;
+    *) bash --init-file "$KDEV_BASEDIR/kdevplatform_shell_environment5.sh" -i ;;
 esac
