diff --git documentation/manpage/manpageplugin.cpp documentation/manpage/manpageplugin.cpp
index 0af9ec4..27dadda 100644
--- documentation/manpage/manpageplugin.cpp
+++ documentation/manpage/manpageplugin.cpp
@@ -94,7 +94,7 @@ IDocumentation::Ptr ManPagePlugin::documentationForDeclaration( Declaration* dec
 
     // Don't show man-page documentation for files that are not in /usr/include, because then we
     // most probably will be confusing the global function-name with a local one
-    if (!dec->topContext()->url().str().startsWith("/usr/"))
+    if (!dec->topContext()->url().str().startsWith("/usr/") && !dec->topContext()->url().str().startsWith("/opt/local/"))
         return {};
 
     ///@todo Do more verification to make sure that we're showing the correct documentation for the declaration
