diff --git app/Info.plist.in app/Info.plist.in
index be256f9..ffe65a6 100644
--- app/Info.plist.in
+++ app/Info.plist.in
@@ -40,5 +40,165 @@
     <string>True</string>
     <key>NSHumanReadableCopyright</key>
     <string>${MACOSX_BUNDLE_COPYRIGHT}</string>
+    <key>LSMultipleInstancesProhibited</key>
+    <false/>
+    <key>CFBundleDocumentTypes</key>
+    <array>
+        <dict>
+            <key>CFBundleTypeExtensions</key>
+            <array>
+                <string>h</string>
+            </array>
+            <key>CFBundleTypeName</key>
+            <string>C Header Source</string>
+            <key>CFBundleTypeRole</key>
+            <string>Editor</string>
+            <key>LSItemContentTypes</key>
+            <array>
+                <string>public.c-header</string>
+            </array>
+        </dict>
+        <dict>
+            <key>CFBundleTypeExtensions</key>
+            <array>
+                <string>hh</string>
+                <string>hp</string>
+                <string>hpp</string>
+                <string>hxx</string>
+                <string>h++</string>
+            </array>
+            <key>CFBundleTypeName</key>
+            <string>C++ Header Source</string>
+            <key>CFBundleTypeRole</key>
+            <string>Editor</string>
+            <key>LSItemContentTypes</key>
+            <array>
+                <string>public.c-plus-plus-header</string>
+            </array>
+        </dict>
+        <dict>
+            <key>CFBundleTypeExtensions</key>
+            <array>
+                <string>c</string>
+            </array>
+            <key>CFBundleTypeName</key>
+            <string>C Source</string>
+            <key>CFBundleTypeRole</key>
+            <string>Editor</string>
+            <key>LSItemContentTypes</key>
+            <array>
+                <string>public.c-source</string>
+            </array>
+        </dict>
+        <dict>
+            <key>CFBundleTypeExtensions</key>
+            <array>
+                <string>cc</string>
+                <string>cp</string>
+                <string>cpp</string>
+                <string>cxx</string>
+                <string>c++</string>
+            </array>
+            <key>CFBundleTypeName</key>
+            <string>C++ Source</string>
+            <key>CFBundleTypeRole</key>
+            <string>Editor</string>
+            <key>LSItemContentTypes</key>
+            <array>
+                <string>public.c-plus-plus-source</string>
+            </array>
+        </dict>
+        <dict>
+            <key>CFBundleTypeExtensions</key>
+            <array>
+                <string>m</string>
+            </array>
+            <key>CFBundleTypeName</key>
+            <string>Objective-C Source</string>
+            <key>CFBundleTypeRole</key>
+            <string>Editor</string>
+            <key>LSItemContentTypes</key>
+            <array>
+                <string>public.objective-c-source</string>
+            </array>
+        </dict>
+        <dict>
+            <key>CFBundleTypeExtensions</key>
+            <array>
+                <string>mm</string>
+                <string>M</string>
+            </array>
+            <key>CFBundleTypeName</key>
+            <string>Objective-C++ Source</string>
+            <key>CFBundleTypeRole</key>
+            <string>Editor</string>
+            <key>LSItemContentTypes</key>
+            <array>
+                <string>public.objective-c-plus-plus-source</string>
+            </array>
+        </dict>
+        <dict>
+            <key>CFBundleTypeExtensions</key>
+            <array>
+                <string>make</string>
+            </array>
+            <key>CFBundleTypeName</key>
+            <string>Makefile</string>
+            <key>CFBundleTypeRole</key>
+            <string>Editor</string>
+            <key>LSItemContentTypes</key>
+            <array>
+                <string>com.apple.xcode.make-script</string>
+            </array>
+        </dict>
+        <dict>
+            <key>CFBundleTypeExtensions</key>
+            <array>
+                <string>plist</string>
+            </array>
+            <key>CFBundleTypeName</key>
+            <string>XML Property List</string>
+            <key>CFBundleTypeRole</key>
+            <string>Editor</string>
+            <key>LSItemContentTypes</key>
+            <array>
+                <string>com.apple.xml-property-list</string>
+            </array>
+        </dict>
+        <dict>
+            <key>CFBundleTypeName</key>
+            <string>Source Code</string>
+            <key>CFBundleTypeRole</key>
+            <string>Editor</string>
+            <key>LSItemContentTypes</key>
+            <array>
+                <string>public.source-code</string>
+            </array>
+        </dict>
+        <dict>
+            <key>CFBundleTypeName</key>
+            <string>Text Document</string>
+            <key>CFBundleTypeRole</key>
+            <string>Editor</string>
+            <key>LSItemContentTypes</key>
+            <array>
+                <string>public.text</string>
+            </array>
+        </dict>
+        <dict>
+            <key>CFBundleTypeExtensions</key>
+            <array>
+                <string>*</string>
+            </array>
+            <key>CFBundleTypeName</key>
+            <string>Anything</string>
+            <key>CFBundleTypeOSTypes</key>
+            <array>
+                <string>****</string>
+            </array>
+            <key>CFBundleTypeRole</key>
+            <string>Viewer</string>
+        </dict>
+    </array>
 </dict>
 </plist>
diff --git app/main.cpp app/main.cpp
index 1129e65..f1273d0 100644
--- app/main.cpp
+++ app/main.cpp
@@ -75,11 +75,15 @@ Q_LOGGING_CATEGORY(APP, "kdevelop.app")
 
 #include <iostream>
 
-#ifdef Q_OS_MAC
+#include <QFileOpenEvent>
+#ifdef Q_OS_OSX
 #include <CoreFoundation/CoreFoundation.h>
+#include <unistd.h>
+#include <stdio.h>
 #endif
 
 using namespace KDevelop;
+struct UrlInfo;
 
 namespace {
 
@@ -112,12 +116,16 @@ class KDevelopApplication:
     public QApplication
 #endif
 {
+    // Q_OBJECT wouldn't be required in KDEVELOP_SINGLE_APP if moc respected CPP tokens.
+    Q_OBJECT
 public:
     explicit KDevelopApplication(int &argc, char **argv, bool GUIenabled = true)
 #if KDEVELOP_SINGLE_APP
         : SharedTools::QtSingleApplication(QStringLiteral("KDevelop"), argc, argv)
 #else
         : QApplication(argc, argv, GUIenabled)
+        , m_session(0)
+        , queueFileOpenEvents(false)
 #endif
         {
 #if KDEVELOP_SINGLE_APP
@@ -153,6 +161,16 @@ public Q_SLOTS:
     {
         openFiles({UrlInfo(file)});
     }
+#else
+    void startHandlingFileOpenEvents(QString *session);
+    void handleQueuedFileOpenRequests();
+
+protected:
+    /**
+     * Event filter for QApplication to handle Mac OS X style file open requests
+     * which Qt translates into FileOpen application events.
+     */
+    bool eventFilter(QObject *obj, QEvent *event) Q_DECL_OVERRIDE;
 #endif
 
 private Q_SLOTS:
@@ -169,6 +187,12 @@ private Q_SLOTS:
             sm.setRestartCommand(QStringList() << QCoreApplication::applicationFilePath() << "-session" << x11SessionId << "-s" << kdevelopSessionId);
         }
     }
+#if !KDEVELOP_SINGLE_APP
+private:
+    QString *m_session;
+    bool queueFileOpenEvents;
+    QVector<UrlInfo> fileOpenRequestList;
+#endif
 };
 
 /// Tries to find a session identified by @p data in @p sessions.
@@ -216,7 +240,7 @@ static int openFilesInRunningInstance(const QVector<UrlInfo>& files, qint64 pid)
 
 /// Gets the PID of a running KDevelop instance, eventually asking the user if there is more than one.
 /// Returns -1 in case there are no running sessions.
-static qint64 getRunningSessionPid()
+static qint64 getRunningSessionPid(QString *sessionNameReturn=NULL)
 {
     SessionInfos candidates;
     foreach( const KDevelop::SessionInfo& si, KDevelop::SessionController::availableSessionInfos() ) {
@@ -236,6 +260,9 @@ static qint64 getRunningSessionPid()
         const QString title = i18n("Select the session to open the document in");
         sessionUuid = KDevelop::SessionController::showSessionChooserDialog(title, true);
     }
+    if (sessionNameReturn) {
+        *sessionNameReturn = sessionUuid;
+    }
     return KDevelop::SessionController::sessionRunInfo(sessionUuid).holderPid;
 }
 
@@ -273,13 +300,81 @@ static qint64 findSessionPid(const QString &sessionId)
     return sessionInfo.holderPid;
 }
 
+#if !KDEVELOP_SINGLE_APP
+void KDevelopApplication::startHandlingFileOpenEvents(QString *session)
+{
+    queueFileOpenEvents = true;
+    m_session = session;
+    installEventFilter(this);
+    // let the eventFilter construct a list of all files to open, from any pending
+    // FileOpen events:
+    processEvents();
+    queueFileOpenEvents = false;
+    qCDebug(APP) << "initial FileOpen events processed:" << fileOpenRequestList.count();
+    // process the list of files to open:
+    handleQueuedFileOpenRequests();
+}
+
+void KDevelopApplication::handleQueuedFileOpenRequests()
+{
+    if (fileOpenRequestList.isEmpty()) {
+        return;
+    }
+    int pid = getRunningSessionPid(m_session);
+    if (pid > 0) {
+        if (m_session) {
+            qCDebug(APP) << "opening" << fileOpenRequestList.count() << "file(s) in session" << *m_session << "pid=" << pid;
+        }
+        openFilesInRunningInstance(fileOpenRequestList, pid);
+        // in a more subtle implementation, openFilesInRunningInstance() would return a list
+        // containing the documents that failed to open, and we could reininitialise fileOpenRequestList
+        // from that list.
+        fileOpenRequestList.clear();
+    }
+}
+
+bool KDevelopApplication::eventFilter(QObject *obj, QEvent *event)
+{
+    if (event->type() == QEvent::FileOpen) {
+        QFileOpenEvent *fileOpenEvent = static_cast<QFileOpenEvent*>(event);
+        qCDebug(APP) << "FileOpen event for" << fileOpenEvent->url();
+        fileOpenRequestList.append(UrlInfo(fileOpenEvent->file()));
+        if (!queueFileOpenEvents){
+            handleQueuedFileOpenRequests();
+        }
+        // don't eat the event
+        return false;
+    }
+
+    return QObject::eventFilter(obj, event);
+}
+
+static bool shouldSimulatePSOption(int argc, char *argv[])
+{
+#ifdef Q_OS_OSX
+    if ((argc == 2 && qstrncmp(argv[1], "-psn_", 5) == 0)
+        || (argc== 1 && !isatty(fileno(stdin)))){
+        // this is a pure start from the Finder or Dock, without any arguments to open.
+        // Present the session selection dialog (= simulate the --ps option) if the user
+        // holds the Command (sic) key while we start. Without that modified we'd also
+        // be doing this instead of handling the FileOpen event associate with a file-open
+        // request from the Finder or the Dock.
+        if (QApplication::queryKeyboardModifiers() & Qt::ControlModifier) {
+            return true;
+        }
+    }
+#endif
+    return false;
+}
+#endif
+
 int main( int argc, char *argv[] )
 {
     QElapsedTimer timer;
     timer.start();
 
     // TODO: Maybe generalize, add KDEVELOP_STANDALONE build option
-#if defined(Q_OS_WIN) || defined(Q_OS_MAC)
+#if defined(Q_OS_WIN) || defined(Q_OS_OSX)
     qputenv("KDE_FORK_SLAVES", "1"); // KIO slaves will be forked off instead of being started via DBus
 #endif
 
@@ -297,7 +392,7 @@ int main( int argc, char *argv[] )
     KLocalizedString::setApplicationDomain("kdevelop");
     QCoreApplication::setAttribute(Qt::AA_DontCreateNativeWidgetSiblings);
 
-#ifdef Q_OS_MAC
+#ifdef Q_OS_OSX
     CFBundleRef mainBundle = CFBundleGetMainBundle();
     if (mainBundle) {
         // get the application's Info Dictionary. For app bundles this would live in the bundle's Info.plist,
@@ -475,6 +570,8 @@ int main( int argc, char *argv[] )
         return 0;
     }
 
+    QString session;
+
     // Handle extra arguments, which stand for files to open
     QVector<UrlInfo> initialFiles;
     foreach (const QString &file, parser.positionalArguments()) {
@@ -494,7 +591,7 @@ int main( int argc, char *argv[] )
 #else
         qint64 pid = -1;
         if (parser.isSet("open-session")) {
-            const QString session = findSessionId(availableSessionInfos, parser.value("open-session"));
+            session = findSessionId(availableSessionInfos, parser.value("open-session"));
             if (session.isEmpty()) {
                 return 1;
             } else if (KDevelop::SessionController::isSessionRunning(session)) {
@@ -511,7 +608,6 @@ int main( int argc, char *argv[] )
     }
 
     // if empty, restart kdevelop with last active session, see SessionController::defaultSessionId
-    QString session;
 
     uint nRunningSessions = 0;
     foreach(const KDevelop::SessionInfo& si, availableSessionInfos)
@@ -554,12 +650,18 @@ int main( int argc, char *argv[] )
         }
     }
 
+#if KDEVELOP_SINGLE_APP
+    const bool simulatePS = false;
     if(parser.isSet("ps"))
+#else
+    const bool simulatePS = shouldSimulatePSOption(argc, argv);
+    if(parser.isSet("ps") || simulatePS)
+#endif
     {
         bool onlyRunning = parser.isSet("pid");
         session = KDevelop::SessionController::showSessionChooserDialog(i18n("Select the session you would like to use"), onlyRunning);
         if(session.isEmpty())
-            return 1;
+            return simulatePS ? 0 : 1;
     }
 
     if ( parser.isSet("debug") ) {
@@ -648,12 +750,37 @@ int main( int argc, char *argv[] )
 
     KDevIDEExtension::init();
 
+#if !KDEVELOP_SINGLE_APP
+    // start filtering events, so that on OS X we can handle file open requests
+    // sent through LaunchServices (i.e. the Finder).
+    // startHandlingFileOpenEvents() will call processEvents(), and if a FileOpen event
+    // was already queued it may present a session selection dialog in which case
+    // session has to be changed. This happens on OS X when we were started because the user asked
+    // the Finder or Dock to open a file with KDevelop, and is the reason a pointer to <session>
+    // is handed to startHandlingFileOpenEvents().
+    // startHandlingFileOpenEvents() will cause all pending FileOpen events to be appended
+    // to a list, and will then attempt to open the documents in that list. If that attempt
+    // succeeds (because at least 1 session is already open and running), session will
+    // have been set to the session where the document(s) was/were opened.
+    app.startHandlingFileOpenEvents(&session);
+#endif
+
     if(!Core::initialize(nullptr, Core::Default, session))
         return 5;
 
     // register a DBUS service for this process, so that we can open files in it from other invocations
     QDBusConnection::sessionBus().registerService(QString("org.kdevelop.kdevelop-%1").arg(app.applicationPid()));
 
+#if !KDEVELOP_SINGLE_APP
+    // When the user requests to open a file (or multiple files) in KDevelop on OS X, and no KDevelop instance is
+    // already running, the initial attempt to open these files may fail because getRunningSessionPid() fails.
+    // In that case the FileOpen requests are still queued, and we should be able to process them now.
+    // NB: it is not impossible that another session was opened between this point and the initial call to
+    // startHandlingFileOpenEvents(). If that happens, the user will probably be asked to pick a session to open
+    // the queued document(s) in, meaning session could change.
+    app.handleQueuedFileOpenRequests();
+#endif
+
     Core* core = Core::self();
     if (!QProcessEnvironment::systemEnvironment().contains("KDEV_DISABLE_WELCOMEPAGE")) {
         core->pluginController()->loadPlugin("KDevWelcomePage");
@@ -762,3 +889,5 @@ int main( int argc, char *argv[] )
 
     return app.exec();
 }
+
+#include "main.moc"
