diff --git app/main.cpp app/main.cpp
index 1129e65..f1273d0 100644
--- app/main.cpp
+++ app/main.cpp
@@ -293,6 +390,9 @@ int main( int argc, char *argv[] )
             CFDictionarySetValue(infoDict, CFSTR("NSSupportsAutomaticTermination"), kCFBooleanFalse);
         }
     }
+    if (!QGuiApplication::platformName().contains(QLatin1String("cocoa"))) {
+        QCoreApplication::setAttribute(Qt::AA_DontUseNativeMenuBar);
+    }
 #endif
 
     static const char description[] = I18N_NOOP( "The KDevelop Integrated Development Environment" );
