diff --git projectmanagers/cmake/cmakeutils.cpp projectmanagers/cmake/cmakeutils.cpp
index 5ee5857..5eb3b54 100644
--- projectmanagers/cmake/cmakeutils.cpp
+++ projectmanagers/cmake/cmakeutils.cpp
@@ -296,7 +296,10 @@ QString currentBuildType( KDevelop::IProject* project )
 
 QString findExecutable()
 {
-    auto cmake = QStandardPaths::findExecutable("cmake");
+    auto cmake = QStandardPaths::findExecutable("cmake", QStringList(QStringLiteral("@PREFIX@/bin")));
+    if (cmake.isEmpty()) {
+        cmake = QStandardPaths::findExecutable("cmake");
+    }
 #ifdef Q_OS_WIN
     if (cmake.isEmpty())
         cmake = QStandardPaths::findExecutable("cmake",{
@@ -323,7 +326,7 @@ KDevelop::Path currentCMakeBinary( KDevelop::IProject* project )
 
 KDevelop::Path currentInstallDir( KDevelop::IProject* project )
 {
-    return KDevelop::Path(readProjectParameter( project, Config::Specific::cmakeInstallDirKey, "/usr/local" ));
+    return KDevelop::Path(readProjectParameter( project, Config::Specific::cmakeInstallDirKey, "/opt/local" ));
 }
 
 QString projectRootRelative( KDevelop::IProject* project )
