diff --git languages/clang/duchain/clangproblem.cpp languages/clang/duchain/clangproblem.cpp
index 7afbb50..a60a069 100644
--- languages/clang/duchain/clangproblem.cpp
+++ languages/clang/duchain/clangproblem.cpp
@@ -186,8 +186,11 @@ ClangFixits ClangProblem::allFixits() const
 
     for (const IProblem::Ptr& diagnostic : diagnostics()) {
         const Ptr problem(dynamic_cast<ClangProblem*>(diagnostic.data()));
-        Q_ASSERT(problem);
-        result << problem->allFixits();
+        if (problem) {
+            result << problem->allFixits();
+        } else {
+            qWarning() << Q_FUNC_INFO << "dynamic cast to ClangProblem failed of" << diagnostic.data();
+        }
     }
     return result;
 }

