diff --git debuggers/gdb/disassemblewidget.cpp debuggers/gdb/disassemblewidget.cpp
index f75ef83..730aa98 100644
--- debuggers/gdb/disassemblewidget.cpp
+++ debuggers/gdb/disassemblewidget.cpp
@@ -183,7 +183,7 @@ DisassembleWidget::DisassembleWidget(CppDebuggerPlugin* plugin, QWidget *parent)
 
     setLayout(topLayout);
 
-    setWindowIcon( QIcon::fromTheme("system-run") );
+    setWindowIcon( QIcon::fromTheme("system-run", windowIcon()) );
     setWindowTitle(i18n("Disassemble/Registers View"));
 
     KDevelop::IDebugController* pDC=KDevelop::ICore::self()->debugController();
diff --git debuggers/gdb/gdboutputwidget.cpp debuggers/gdb/gdboutputwidget.cpp
index e7a7039..1078a49 100644
--- debuggers/gdb/gdboutputwidget.cpp
+++ debuggers/gdb/gdboutputwidget.cpp
@@ -60,7 +60,7 @@ GDBOutputWidget::GDBOutputWidget(CppDebuggerPlugin* plugin, QWidget *parent) :
     showInternalCommands_(false),
     maxLines_(5000)
 {
-    setWindowIcon(QIcon::fromTheme("dialog-scripts"));
+    setWindowIcon(QIcon::fromTheme("dialog-scripts", windowIcon()));
     setWindowTitle(i18n("GDB Output"));
     setWhatsThis(i18n("<b>GDB output</b><p>"
                     "Shows all gdb commands being executed. "
diff --git debuggers/gdb/memviewdlg.cpp debuggers/gdb/memviewdlg.cpp
index b9ff3c1..7153cd8 100644
--- debuggers/gdb/memviewdlg.cpp
+++ debuggers/gdb/memviewdlg.cpp
@@ -469,7 +469,7 @@ namespace GDBDebugger
     MemoryViewerWidget::MemoryViewerWidget(CppDebuggerPlugin* /*plugin*/, QWidget* parent)
     : QWidget(parent)
     {
-        setWindowIcon(QIcon::fromTheme("server-database"));
+        setWindowIcon(QIcon::fromTheme("server-database", windowIcon()));
         setWindowTitle(i18n("Memory viewer"));
 
         QAction * newMemoryViewerAction = new QAction(this);
diff --git utils/okteta/oktetatoolviewfactory.cpp utils/okteta/oktetatoolviewfactory.cpp
index 702deda..39f998e 100644
--- utils/okteta/oktetatoolviewfactory.cpp
+++ utils/okteta/oktetatoolviewfactory.cpp
@@ -67,7 +67,7 @@ QWidget* OktetaToolViewFactory::create( QWidget* parent )
     Kasten::AbstractToolView* toolView = mToolViewFactory->create( tool );
 
     QWidget* widget = new KastenToolViewWidget( toolView, parent );
-    widget->setWindowIcon( QIcon::fromTheme(mToolViewFactory->iconName()) );
+    widget->setWindowIcon( QIcon::fromTheme(mToolViewFactory->iconName(), windowIcon()) );
     return widget;
 }
 
