diff --git app/plasma/CMakeLists.txt app/plasma/CMakeLists.txt
index 243f739..335b0c1 100644
--- app/plasma/CMakeLists.txt
+++ app/plasma/CMakeLists.txt
@@ -1,11 +1,11 @@
 add_subdirectory(dataengine)
 add_subdirectory(applet)
 
-find_package(KF5Runner)
-set_package_properties(KF5Runner PROPERTIES
-    PURPOSE "Framework for Plasma runners. For enabling the KDevelop runner"
-    URL "http://www.kde.org"
-    TYPE OPTIONAL)
-if(KF5Runner_FOUND)
-    add_subdirectory(runner)
-endif()
+#find_package(KF5Runner)
+#set_package_properties(KF5Runner PROPERTIES
+#    PURPOSE "Framework for Plasma runners. For enabling the KDevelop runner"
+#    URL "http://www.kde.org"
+#    TYPE OPTIONAL)
+#if(KF5Runner_FOUND)
+#    add_subdirectory(runner)
+#endif()
