Index: app/main.cpp
===================================================================
--- app/main.cpp
+++ app/main.cpp
@@ -448,7 +448,7 @@
     // which only shows the running sessions, and the user can pick one.
     parser.addOption(QCommandLineOption{QStringList{"pid"}});
 
-    parser.addPositionalArgument("files", i18n( "Files to load" ), "[FILE...]");
+    parser.addPositionalArgument("files", i18n( "Files to load, with an optional :line or :line:column initial cursor position appended" ), "[FILE...]");
 
     // The session-controller needs to arguments to eventually pass them to newly opened sessions
     KDevelop::SessionController::setArguments(argc, argv);
