diff --git languages/clang/duchain/unknowndeclarationproblem.cpp languages/clang/duchain/unknowndeclarationproblem.cpp
index 5da0350..eb320f2 100644
--- languages/clang/duchain/unknowndeclarationproblem.cpp
+++ languages/clang/duchain/unknowndeclarationproblem.cpp
@@ -46,6 +46,7 @@
 
 #include <QDir>
 #include <QProcess>
+#include <QCoreApplication>
 
 #include <algorithm>
 
@@ -84,6 +85,9 @@ QStringList scanIncludePaths( const QString& identifier, const QDir& dir, int ma
         return {};
     }
 
+    QCoreApplication::sendPostedEvents();
+    QCoreApplication::processEvents();
+
     QStringList candidates;
     const auto path = dir.absolutePath();
 
