diff --git languages/clang/duchain/unknowndeclarationproblem.cpp languages/clang/duchain/unknowndeclarationproblem.cpp
index 5da0350..eb320f2 100644
--- languages/clang/duchain/unknowndeclarationproblem.cpp
+++ languages/clang/duchain/unknowndeclarationproblem.cpp
@@ -47,9 +47,13 @@
 
 #include <QDir>
 #include <QProcess>
+#include <QCoreApplication>
+#include <QAbstractEventDispatcher>
 
 #include <algorithm>
 
+#include <QDebug>
+
 using namespace KDevelop;
 
 namespace {
@@ -85,6 +89,18 @@ QStringList scanIncludePaths( const QString& identifier, const QDir& dir, int ma
         return {};
     }
 
+    QCoreApplication::sendPostedEvents();
+    QCoreApplication::flush();
+    QAbstractEventDispatcher *eD = QCoreApplication::eventDispatcher();
+    if (eD) {
+        if (eD->processEvents(QEventLoop::AllEvents)) {
+            qWarning() << Q_FUNC_INFO << "processed events, this run";
+            return {};
+        }
+    } else {
+        QCoreApplication::processEvents();
+    }
+
     QStringList candidates;
     const auto path = dir.absolutePath();
 
