diff --git app/Info.plist.in app/Info.plist.in
index be256f9..ffe65a6 100644
--- app/Info.plist.in
+++ app/Info.plist.in
@@ -40,5 +40,165 @@
     <string>True</string>
     <key>NSHumanReadableCopyright</key>
     <string>${MACOSX_BUNDLE_COPYRIGHT}</string>
+    <key>LSMultipleInstancesProhibited</key>
+    <false/>
+    <key>CFBundleDocumentTypes</key>
+    <array>
+        <dict>
+            <key>CFBundleTypeExtensions</key>
+            <array>
+                <string>h</string>
+            </array>
+            <key>CFBundleTypeName</key>
+            <string>C Header Source</string>
+            <key>CFBundleTypeRole</key>
+            <string>Editor</string>
+            <key>LSItemContentTypes</key>
+            <array>
+                <string>public.c-header</string>
+            </array>
+        </dict>
+        <dict>
+            <key>CFBundleTypeExtensions</key>
+            <array>
+                <string>hh</string>
+                <string>hp</string>
+                <string>hpp</string>
+                <string>hxx</string>
+                <string>h++</string>
+            </array>
+            <key>CFBundleTypeName</key>
+            <string>C++ Header Source</string>
+            <key>CFBundleTypeRole</key>
+            <string>Editor</string>
+            <key>LSItemContentTypes</key>
+            <array>
+                <string>public.c-plus-plus-header</string>
+            </array>
+        </dict>
+        <dict>
+            <key>CFBundleTypeExtensions</key>
+            <array>
+                <string>c</string>
+            </array>
+            <key>CFBundleTypeName</key>
+            <string>C Source</string>
+            <key>CFBundleTypeRole</key>
+            <string>Editor</string>
+            <key>LSItemContentTypes</key>
+            <array>
+                <string>public.c-source</string>
+            </array>
+        </dict>
+        <dict>
+            <key>CFBundleTypeExtensions</key>
+            <array>
+                <string>cc</string>
+                <string>cp</string>
+                <string>cpp</string>
+                <string>cxx</string>
+                <string>c++</string>
+            </array>
+            <key>CFBundleTypeName</key>
+            <string>C++ Source</string>
+            <key>CFBundleTypeRole</key>
+            <string>Editor</string>
+            <key>LSItemContentTypes</key>
+            <array>
+                <string>public.c-plus-plus-source</string>
+            </array>
+        </dict>
+        <dict>
+            <key>CFBundleTypeExtensions</key>
+            <array>
+                <string>m</string>
+            </array>
+            <key>CFBundleTypeName</key>
+            <string>Objective-C Source</string>
+            <key>CFBundleTypeRole</key>
+            <string>Editor</string>
+            <key>LSItemContentTypes</key>
+            <array>
+                <string>public.objective-c-source</string>
+            </array>
+        </dict>
+        <dict>
+            <key>CFBundleTypeExtensions</key>
+            <array>
+                <string>mm</string>
+                <string>M</string>
+            </array>
+            <key>CFBundleTypeName</key>
+            <string>Objective-C++ Source</string>
+            <key>CFBundleTypeRole</key>
+            <string>Editor</string>
+            <key>LSItemContentTypes</key>
+            <array>
+                <string>public.objective-c-plus-plus-source</string>
+            </array>
+        </dict>
+        <dict>
+            <key>CFBundleTypeExtensions</key>
+            <array>
+                <string>make</string>
+            </array>
+            <key>CFBundleTypeName</key>
+            <string>Makefile</string>
+            <key>CFBundleTypeRole</key>
+            <string>Editor</string>
+            <key>LSItemContentTypes</key>
+            <array>
+                <string>com.apple.xcode.make-script</string>
+            </array>
+        </dict>
+        <dict>
+            <key>CFBundleTypeExtensions</key>
+            <array>
+                <string>plist</string>
+            </array>
+            <key>CFBundleTypeName</key>
+            <string>XML Property List</string>
+            <key>CFBundleTypeRole</key>
+            <string>Editor</string>
+            <key>LSItemContentTypes</key>
+            <array>
+                <string>com.apple.xml-property-list</string>
+            </array>
+        </dict>
+        <dict>
+            <key>CFBundleTypeName</key>
+            <string>Source Code</string>
+            <key>CFBundleTypeRole</key>
+            <string>Editor</string>
+            <key>LSItemContentTypes</key>
+            <array>
+                <string>public.source-code</string>
+            </array>
+        </dict>
+        <dict>
+            <key>CFBundleTypeName</key>
+            <string>Text Document</string>
+            <key>CFBundleTypeRole</key>
+            <string>Editor</string>
+            <key>LSItemContentTypes</key>
+            <array>
+                <string>public.text</string>
+            </array>
+        </dict>
+        <dict>
+            <key>CFBundleTypeExtensions</key>
+            <array>
+                <string>*</string>
+            </array>
+            <key>CFBundleTypeName</key>
+            <string>Anything</string>
+            <key>CFBundleTypeOSTypes</key>
+            <array>
+                <string>****</string>
+            </array>
+            <key>CFBundleTypeRole</key>
+            <string>Viewer</string>
+        </dict>
+    </array>
 </dict>
 </plist>
diff --git app/main.cpp app/main.cpp
index b5f7bb9..c2b92fa 100644
--- app/main.cpp
+++ app/main.cpp
@@ -71,7 +71,8 @@ Q_LOGGING_CATEGORY(APP, "kdevelop.app")
 
 #include "splash.h"
 
-#ifdef Q_OS_MAC
+#ifdef Q_OS_OSX
+#include <QFileOpenEvent>
 #include <CoreFoundation/CoreFoundation.h>
 #endif
 
@@ -79,6 +80,7 @@ using namespace KDevelop;
 
 class KDevelopApplication: public QApplication
 {
+    Q_OBJECT
 public:
     explicit KDevelopApplication(int &argc, char **argv, bool GUIenabled = true)
         : QApplication(argc, argv, GUIenabled)
@@ -86,6 +88,16 @@ public:
             connect(this, &QGuiApplication::saveStateRequest, this, &KDevelopApplication::saveState);
         }
 
+    void startFilteringEvents() {
+        installEventFilter(this);
+    }
+
+protected:
+    /**
+     * Event filter for QApplication to handle Mac OS X like file open
+     */
+    bool eventFilter(QObject *obj, QEvent *event);
+
 private Q_SLOTS:
     void saveState( QSessionManager& sm ) {
         if (KDevelop::Core::self() && KDevelop::Core::self()->sessionController()) {
@@ -232,10 +241,34 @@ static qint64 findSessionPid(const QString &sessionId)
     return sessionInfo.holderPid;
 }
 
+bool KDevelopApplication::eventFilter(QObject *obj, QEvent *event)
+{
+    if (event->type() == QEvent::FileOpen) {
+        QFileOpenEvent *foe = static_cast<QFileOpenEvent*>(event);
+        qDebug() << Q_FUNC_INFO << "FileOpen event for" << foe->url();
+        QVector<UrlInfo> fileList;
+        fileList.append(UrlInfo(foe->file()));
+        if ( ! fileList.isEmpty() )
+        {
+            int pid = getRunningSessionPid();
+            if ( pid > 0 ) {
+                openFilesInRunningInstance(fileList, pid);
+            }
+        }
+        // don't eat the event
+        return false;
+    }
+
+    /**
+     * else: pass over to default implementation
+     */
+    return QObject::eventFilter(obj, event);
+}
+
 int main( int argc, char *argv[] )
 {
     // TODO: Maybe generalize, add KDEVELOP_STANDALONE build option
-#if defined(Q_OS_WIN) || defined(Q_OS_MAC)
+#if defined(Q_OS_WIN) || defined(Q_OS_OSX)
     qputenv("KDE_FORK_SLAVES", "1"); // KIO slaves will be forked off instead of being started via DBus
 #endif
 
@@ -254,7 +287,7 @@ int main( int argc, char *argv[] )
     KLocalizedString::setApplicationDomain("kdevelop");
     QCoreApplication::setAttribute(Qt::AA_DontCreateNativeWidgetSiblings);
 
-#ifdef Q_OS_MAC
+#ifdef Q_OS_OSX
     CFBundleRef mainBundle = CFBundleGetMainBundle();
     if (mainBundle) {
         // get the application's Info Dictionary. For app bundles this would live in the bundle's Info.plist,
@@ -649,5 +685,11 @@ int main( int argc, char *argv[] )
 #endif
 
     qCDebug(APP) << "Done startup";
+
+    // now install the event filter that will allow us to handle
+    // file open requests from the Finder on OS X.
+    app.startFilteringEvents();
     return app.exec();
 }
+
+#include "main.moc"
\ No newline at end of file
