diff --git app/kdevelopui.rc app/kdevelopui.rc
index 2fe1594..0273900 100644
--- app/kdevelopui.rc
+++ app/kdevelopui.rc
@@ -1,5 +1,5 @@
 <!DOCTYPE kpartgui SYSTEM "kpartgui.dtd">
-<kpartgui version="140" name="kdevelop" translationDomain="kdevelop">
+<kpartgui version="141" name="kdevelop" translationDomain="kdevelop">
 <MenuBar>
 
   <Menu name="session" append="first_menu">
@@ -149,6 +149,8 @@
     <DefineGroup append="show_toolbar_merge" name="show_toolbar_merge"/>
     <DefineGroup append="show_merge" name="show_merge" />
     <DefineGroup append="configure_merge" name="configure_merge" />
+    <Separator/>
+    <Action name="widgetstyle_submenu"/>
   </Menu>
 
   <Menu name="help">
