diff --git src/orig.CMakeLists.txt src/CMakeLists.txt
index c774daa..3a84ceb 100644
--- src/orig.CMakeLists.txt
+++ src/CMakeLists.txt
@@ -183,9 +183,10 @@ if (CONFIG_PIPEWIRE)
 endif ()
 
 
-if (UNIX AND NOT APPLE)
+if (UNIX)
   install (TARGETS ${PROJECT_NAME} RUNTIME
-    DESTINATION ${CMAKE_INSTALL_BINDIR})
+    DESTINATION ${CMAKE_INSTALL_BINDIR}
+    BUNDLE DESTINATION ${BUNDLE_INSTALL_DIR})
   install (FILES ${QM_FILES}
     DESTINATION ${CMAKE_INSTALL_DATADIR}/${PROJECT_NAME}/translations)
   install (FILES images/${PROJECT_NAME}.png
