diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5131e51..cdf860a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,7 +2,7 @@ project(QSYNTH)
 
 cmake_minimum_required(VERSION 2.6)
 
-set (VERSION "0.4.2")
+set (VERSION "0.4.3")
 
 set (PACKAGE_NAME "Qsynth")
 set (PACKAGE_VERSION "${VERSION}")
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 3843f0d..8465413 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -10,6 +10,8 @@ link_directories (
     ${QT_LIBRARY_DIR}
 )
 
+set(CONFIG_BUILD_VERSION ${VERSION})
+string(TIMESTAMP CONFIG_BUILD_DATE)
 configure_file ( cmakeconfig.cmake ${CMAKE_CURRENT_BINARY_DIR}/config.h )
 
 set ( ac_prefix ${CMAKE_INSTALL_PREFIX} )
@@ -103,7 +105,16 @@ if (WIN32)
 endif (WIN32)
 
 if (APPLE)
-    set_target_properties ( qsynth PROPERTIES MACOSX_BUNDLE true )
+    set_target_properties ( qsynth PROPERTIES
+        MACOSX_BUNDLE true
+        MACOSX_BUNDLE_GUI_IDENTIFIER "org.rncbc.qsynth"
+        MACOSX_BUNDLE_BUNDLE_NAME "Qsynth"
+        MACOSX_BUNDLE_DISPLAY_NAME "Qsynth"
+        MACOSX_BUNDLE_INFO_STRING "Qsynth, a FluidSynth Qt GUI Interface"
+        MACOSX_BUNDLE_LONG_VERSION_STRING "Qsynth ${VERSION}"
+        MACOSX_BUNDLE_SHORT_VERSION_STRING "${VERSION}"
+        MACOSX_BUNDLE_BUNDLE_VERSION "${VERSION}"
+        MACOSX_BUNDLE_COPYRIGHT "Copyright (C) 2003-2016, rncbc aka Rui Nuno Capela. All rights reserved.")
 endif (APPLE)
 
 target_link_libraries ( qsynth
@@ -124,16 +135,17 @@ set ( TRANSLATIONS
 qt5_add_translation ( QM_FILES ${TRANSLATIONS} )
 add_custom_target( translations ALL DEPENDS ${QM_FILES} )
 
-if (UNIX AND NOT APPLE)
+if (UNIX)
     install ( TARGETS qsynth
-              RUNTIME DESTINATION bin )
+              RUNTIME DESTINATION bin
+              BUNDLE DESTINATION ${BUNDLE_INSTALL_DIR} )
     install ( FILES ${QM_FILES}
               DESTINATION share/locale )
     install ( FILES ${CMAKE_CURRENT_BINARY_DIR}/qsynth.desktop
               DESTINATION share/applications )
     install ( FILES images/qsynth.png
               DESTINATION share/pixmaps )
-endif (UNIX AND NOT APPLE)
+endif (UNIX)
 
 if (WIN32)
     install ( TARGETS qsynth
diff --git a/src/cmakeconfig.cmake b/src/cmakeconfig.cmake
index d0fd2be..7908c6c 100644
--- a/src/cmakeconfig.cmake
+++ b/src/cmakeconfig.cmake
@@ -1,6 +1,12 @@
 #ifndef CONFIG_H
 #define CONFIG_H
 
+/* Build date and time. */
+#cmakedefine CONFIG_BUILD_DATE "@CONFIG_BUILD_DATE@"
+
+/* Build version string. */
+#cmakedefine CONFIG_BUILD_VERSION "@CONFIG_BUILD_VERSION@"
+
 /* Define if debugging is enabled. */
 #cmakedefine CONFIG_DEBUG @CONFIG_DEBUG@
 
