diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
--- qsynth-0.5.4/src/CMakeLists.txt.orig	2018-12-05 17:10:36.345638996 +0100
+++ qsynth-0.5.4/src/CMakeLists.txt	2019-01-29 11:41:19.285405735 +0100
@@ -11,6 +11,8 @@
   ${QT_LIBRARY_DIR}
 )
 
+set(CONFIG_BUILD_VERSION ${VERSION})
+string(TIMESTAMP CONFIG_BUILD_DATE)
 configure_file ( cmakeconfig.cmake ${CMAKE_CURRENT_BINARY_DIR}/config.h )
 
 set ( ac_prefix ${CMAKE_INSTALL_PREFIX} )
@@ -103,7 +105,16 @@
 endif ()
 
 if (APPLE)
-  set_target_properties ( qsynth PROPERTIES MACOSX_BUNDLE true )
+    set_target_properties ( qsynth PROPERTIES
+        MACOSX_BUNDLE true
+        MACOSX_BUNDLE_GUI_IDENTIFIER "org.rncbc.qsynth"
+        MACOSX_BUNDLE_BUNDLE_NAME "Qsynth"
+        MACOSX_BUNDLE_DISPLAY_NAME "Qsynth"
+        MACOSX_BUNDLE_INFO_STRING "Qsynth, a FluidSynth Qt GUI Interface"
+        MACOSX_BUNDLE_LONG_VERSION_STRING "Qsynth ${VERSION}"
+        MACOSX_BUNDLE_SHORT_VERSION_STRING "${VERSION}"
+        MACOSX_BUNDLE_BUNDLE_VERSION "${VERSION}"
+        MACOSX_BUNDLE_COPYRIGHT "Copyright (C) 2003-2016, rncbc aka Rui Nuno Capela. All rights reserved.")
 endif ()
 
 if (UNIX AND NOT APPLE)
@@ -135,12 +146,13 @@
 qt5_add_translation ( QM_FILES ${TRANSLATIONS} )
 add_custom_target( translations ALL DEPENDS ${QM_FILES} )
 
-if (UNIX AND NOT APPLE)
+if (UNIX)
   install ( TARGETS qsynth
-     RUNTIME DESTINATION bin )
+     RUNTIME DESTINATION bin
+     BUNDLE DESTINATION ${BUNDLE_INSTALL_DIR} )
   install ( FILES ${QM_FILES}
      DESTINATION share/qsynth/translations )
-  install ( FILES ${CMAKE_CURRENT_BINARY_DIR}/qsynth.desktop
+  install ( FILES ${CMAKE_CURRENT_SOURCE_DIR}/qsynth.desktop
      DESTINATION share/applications )
   install ( FILES images/qsynth.png
      DESTINATION share/pixmaps )
diff --git a/src/cmakeconfig.cmake b/src/cmakeconfig.cmake
--- qsynth-0.5.4/src/cmakeconfig.cmake.orig	2019-01-29 11:09:45.515405735 +0100
+++ qsynth-0.5.4/src/cmakeconfig.cmake	2019-01-29 11:12:03.945405735 +0100
@@ -13,6 +13,12 @@
 /* Build version string. */
 #cmakedefine CONFIG_BUILD_VERSION "@CONFIG_BUILD_VERSION@"
 
+/* Build date and time. */
+#cmakedefine CONFIG_BUILD_DATE "@CONFIG_BUILD_DATE@"
+
+/* Build version string. */
+#cmakedefine CONFIG_BUILD_VERSION "@CONFIG_BUILD_VERSION@"
+
 /* Define if debugging is enabled. */
 #cmakedefine CONFIG_DEBUG @CONFIG_DEBUG@
 
