diff --git cmake/orig.DeploymentUtils.cmake cmake/DeploymentUtils.cmake
index e4a466d..4447d71 100644
--- cmake/orig.DeploymentUtils.cmake
+++ cmake/DeploymentUtils.cmake
@@ -43,7 +43,7 @@ if (WIN32 AND WINDEPLOYQT_EXECUTABLE)
     )
 endif()
 
-if (APPLE AND MACDEPLOYQT_EXECUTABLE)
+if (APPLE_STANDALONE AND MACDEPLOYQT_EXECUTABLE)
     add_custom_command(TARGET ${PROJECT_NAME} POST_BUILD
         COMMAND "${MACDEPLOYQT_EXECUTABLE}"
                 "$<TARGET_BUNDLE_DIR:${PROJECT_NAME}>"
