diff --git src/orig.qsynthSetupForm.cpp src/qsynthSetupForm.cpp
index 9fb77c7..28d2758 100644
--- src/orig.qsynthSetupForm.cpp
+++ src/qsynthSetupForm.cpp
@@ -1172,8 +1172,10 @@ qsynthSettingsItemEditor::qsynthSettingsItemEditor (
 		m_u.pSpinBox->setMinimum(iRangeMin);
 		m_u.pSpinBox->setMaximum(iRangeMax);
 		m_u.pSpinBox->setAccelerated(true);
+#if QT_VERSION >= QT_VERSION_CHECK(5, 12, 0)
 		m_u.pSpinBox->setStepType(
 			QAbstractSpinBox::AdaptiveDecimalStepType);
+#endif
 		QObject::connect(m_u.pSpinBox,
 			SIGNAL(valueChanged(int)),
 			SLOT(committed())
@@ -1193,8 +1195,10 @@ qsynthSettingsItemEditor::qsynthSettingsItemEditor (
 		m_u.pDoubleSpinBox->setMinimum(fRangeMin);
 		m_u.pDoubleSpinBox->setMaximum(fRangeMax);
 		m_u.pDoubleSpinBox->setAccelerated(true);
+#if QT_VERSION >= QT_VERSION_CHECK(5, 12, 0)
 		m_u.pDoubleSpinBox->setStepType(
 			QAbstractSpinBox::AdaptiveDecimalStepType);
+#endif
 		QObject::connect(m_u.pDoubleSpinBox,
 			SIGNAL(valueChanged(double)),
 			SLOT(committed())
