

Index: /Volumes/Debian/Users/bertin/work/src/new/KDE/kde-workspace/bespin-1725/CMakeLists.txt
===================================================================
--- /Volumes/Debian/Users/bertin/work/src/new/KDE/kde-workspace/bespin-1725/CMakeLists.txt	(revision 1725)
+++ /Volumes/Debian/Users/bertin/work/src/new/KDE/kde-workspace/bespin-1725/CMakeLists.txt	(working copy)
@@ -9,11 +9,17 @@
 #     add_definitions("-D_CRT_SECURE_NO_WARNINGS")
 endif(WIN32)
 
-option(ENABLE_ARGB "Enable translucent windows" ON)
 option(ENABLE_KDE "Enable Bespin support for KDE4" ON)
 option(ENABLE_KWIN "Build Bespin KWin decoration" ON)
+if(NOT APPLE)
+option(ENABLE_ARGB "Enable translucent windows" ON)
 option(ENABLE_XBAR "Enable the XBar plasmoid (Mac-Alike-Menubar)" ON)
 option(ENABLE_474_SHADOWS "Enable shadows with variable pixmap size (requires KWin >= 4.7.4)" ON)
+else()
+option(ENABLE_ARGB "Enable translucent windows" OFF)
+option(ENABLE_XBAR "Enable the XBar plasmoid (Mac-Alike-Menubar)" OFF)
+option(ENABLE_474_SHADOWS "Enable shadows with variable pixmap size (requires KWin >= 4.7.4)" OFF)
+endif(NOT APPLE)
 
 if (ENABLE_KDE)
     find_package(KDE4)
@@ -33,7 +39,9 @@
     add_definitions ( -DVARYING_SHADOWS=0 )
 endif (ENABLE_474_SHADOWS)
 
+if(NOT APPLE)
 find_package(X11)
+endif(NOT APPLE)
 
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
 add_custom_target(revision.h ALL COMMAND ${CMAKE_COMMAND} ARGS -P ${CMAKE_CURRENT_SOURCE_DIR}/revision.cmake)
@@ -116,20 +124,24 @@
 add_subdirectory (config)
 
 if (KDE4_FOUND)
-    add_subdirectory (be.shadowed)
+    if(X11_FOUND)
+        add_subdirectory (be.shadowed)
+    endif(X11_FOUND)
     install (FILES bespin.themerc DESTINATION ${DATA_INSTALL_DIR}/kstyle/themes)
-    if (X11_Xrender_FOUND)
-        if(ENABLE_KWIN)
-            add_subdirectory (kwin)
-        else(ENABLE_KWIN)
-            message (STATUS "WARNING: *** KWin decoration will not be built ***")
-        endif(ENABLE_KWIN)
-    endif (X11_Xrender_FOUND)
-    if ( X11_FOUND )
-        if(ENABLE_XBAR)
-            add_subdirectory (XBar)
-        else(ENABLE_XBAR)
-            message (STATUS "WARNING: *** XBar plasmoid will not be built ***")
-        endif(ENABLE_XBAR)
-    endif ( X11_FOUND )
+    if(NOT APPLE)
+        if (X11_Xrender_FOUND)
+            if(ENABLE_KWIN)
+                add_subdirectory (kwin)
+            else(ENABLE_KWIN)
+                message (STATUS "WARNING: *** KWin decoration will not be built ***")
+            endif(ENABLE_KWIN)
+        endif (X11_Xrender_FOUND)
+        if ( X11_FOUND )
+            if(ENABLE_XBAR)
+                add_subdirectory (XBar)
+            else(ENABLE_XBAR)
+                message (STATUS "WARNING: *** XBar plasmoid will not be built ***")
+            endif(ENABLE_XBAR)
+        endif ( X11_FOUND )
+    endif(NOT APPLE)
 endif (KDE4_FOUND)


Index: /Volumes/Debian/Users/bertin/work/src/new/KDE/kde-workspace/bespin-1725/bespin.cpp
===================================================================
--- /Volumes/Debian/Users/bertin/work/src/new/KDE/kde-workspace/bespin-1725/bespin.cpp	(revision 1725)
+++ /Volumes/Debian/Users/bertin/work/src/new/KDE/kde-workspace/bespin-1725/bespin.cpp	(working copy)
@@ -290,7 +290,9 @@
 Style::Style() : QCommonStyle()
 {
     setObjectName(QLatin1String("Bespin"));
+#ifdef Q_WS_X11
     XProperty::init();
+#endif
     FX::init();
 #if BESPIN_STANDARD_PALETTE_HACK
     if (usingStandardPalette)


Index: /Volumes/Debian/Users/bertin/work/src/new/KDE/kde-workspace/bespin-1725/blib/CMakeLists.txt
===================================================================
--- /Volumes/Debian/Users/bertin/work/src/new/KDE/kde-workspace/bespin-1725/blib/CMakeLists.txt	(revision 1725)
+++ /Volumes/Debian/Users/bertin/work/src/new/KDE/kde-workspace/bespin-1725/blib/CMakeLists.txt	(working copy)
@@ -4,7 +4,9 @@
 
 set (CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules )
 
-find_package(X11)
+if(NOT APPLE)
+    find_package(X11)
+endif(NOT APPLE)
 
 # add_definitions (-DQT_NO_CAST_FROM_ASCII -DQT_NO_CAST_TO_ASCII)
 
