diff --git CMakeLists.txt CMakeLists.txt
index 7d1db3a..aed70c6 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -17,6 +17,7 @@ find_package (ECM 1.1.0 REQUIRED NO_MODULE)
 set (CMAKE_MODULE_PATH ${ECM_MODULE_PATH})
 include(ECMOptionalAddSubdirectory)
 include(ECMInstallIcons)
+include(ECMAddAppIcon)
 include(ECMSetupVersion)
 include(ECMMarkNonGuiExecutable)
 include(ECMGenerateHeaders)
diff --git kate/src/CMakeLists.txt kate/src/CMakeLists.txt
index ae167c8..b8983c3 100644
--- kate/src/CMakeLists.txt
+++ kate/src/CMakeLists.txt
@@ -40,6 +40,10 @@ ki18n_wrap_ui(KATE_APPLICATION_SRCS
     ui/sessionconfigwidget.ui
 )
 
+# Sets the icon on Windows and OSX
+file(GLOB ICONS_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/../icons/*-apps-kate.png")
+ecm_add_app_icon(KATE_APPLICATION_SRCS ICONS ${ICONS_SRCS})
+
 qt5_add_resources( KATE_APPLICATION_SRCS data/kate.qrc )
 
 # create kde init executable
diff --git kwrite/CMakeLists.txt kwrite/CMakeLists.txt
index c63ea9a..d064972 100644
--- kwrite/CMakeLists.txt
+++ kwrite/CMakeLists.txt
@@ -6,6 +6,10 @@ find_package(KF5 REQUIRED COMPONENTS DBusAddons)
 set(kwrite_KDEINIT_SRCS main.cpp kwrite.cpp)
 
 
+# Sets the icon on Windows and OSX
+file(GLOB ICONS_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/../kate/icons/*-apps-kate.png")
+ecm_add_app_icon(kwrite_KDEINIT_SRCS ICONS ${ICONS_SRCS})
+
 
 kf5_add_kdeinit_executable(kwrite ${kwrite_KDEINIT_SRCS})
 target_link_libraries(kdeinit_kwrite
