diff --git addons/snippets/snippetview.cpp addons/snippets/snippetview.cpp
index 7a10254..a23d569 100644
--- addons/snippets/snippetview.cpp
+++ addons/snippets/snippetview.cpp
@@ -86,7 +86,7 @@ SnippetView::SnippetView(KateSnippetGlobal* plugin, QWidget* parent)
     Ui::SnippetViewBase::setupUi(this);
 
     setWindowTitle(i18n("Snippets"));
-    setWindowIcon(QIcon::fromTheme(QLatin1String("document-new")));
+    setWindowIcon(QIcon::fromTheme(QLatin1String("document-new"), windowIcon()));
 
     snippetTree->setContextMenuPolicy( Qt::CustomContextMenu );
     snippetTree->viewport()->installEventFilter( this );
diff --git kate/main.cpp kate/main.cpp
index 554de7a..beff088 100644
--- kate/main.cpp
+++ kate/main.cpp
@@ -137,7 +142,7 @@ int main(int argc, char **argv)
     /**
      * set the program icon
      */
-    QApplication::setWindowIcon(QIcon::fromTheme(QLatin1String("kate")));
+    QApplication::setWindowIcon(QIcon::fromTheme(QLatin1String("kate"), app.windowIcon()));
 
     /**
      * Create command line parser and feed it with known options
diff --git kwrite/main.cpp kwrite/main.cpp
index 4a26276..f73e2c9 100644
--- kwrite/main.cpp
+++ kwrite/main.cpp
@@ -128,7 +133,7 @@ extern "C" Q_DECL_EXPORT int main(int argc, char **argv)
     /**
      * set the program icon
      */
-    QApplication::setWindowIcon(QIcon::fromTheme(QLatin1String("accessories-text-editor")));
+    QApplication::setWindowIcon(QIcon::fromTheme(QLatin1String("accessories-text-editor"), app.windowIcon()));
 
     /**
      * Create command line parser and feed it with known options
