--- xfce4-session/orig.xfsm-startup.c	2014-09-28 16:51:01.000000000 +0200
+++ xfce4-session/xfsm-startup.c	2016-12-04 18:20:40.000000000 +0100
@@ -106,7 +106,10 @@
   if (fields != NULL)
     {
       /* second field of GPG_AGENT_INFO is a PID */
-      pid = atoi (fields[1]);
+      if (fields[1])
+        pid = atoi (fields[1]);
+      else
+        pid = -1;
       g_strfreev (fields);
     }
 
