--- src/konfigurator/orig.konfigurator.cpp	2015-12-22 01:46:08.000000000 +0100
+++ src/konfigurator/konfigurator.cpp	2015-12-22 01:46:57.000000000 +0100
@@ -60,7 +60,7 @@
                      QStringLiteral(),
                      QStringLiteral("staikos@kde.org"));
     setAboutData(about);
-    setNeedsAuthorization(true);
+    setNeedsAuthorization(false);
 
     QVBoxLayout *vbox = new QVBoxLayout(this);
     //PORT QT5 vbox->setSpacing(KDialog::spacingHint());
@@ -267,18 +267,19 @@
     QVariantMap args;
     KAuth::Action action = authAction();
     if (!action.isValid()) {
-        qDebug() << "There's no authAction, not saving settings";
-        return;
-    }
-    action.setArguments(args);
+//         qDebug() << "There's no authAction, not saving settings";
+//         return;
+    } else {
+        action.setArguments(args);
 
-    KAuth::ExecuteJob *j = action.execute();
+        KAuth::ExecuteJob *j = action.execute();
 
-    if (!j->exec()) {
-        qDebug() << j->errorText();
-        KMessageBox::error(this, j->errorString(), i18n("KDE Wallet Control Module"));
-        load();
-        return;
+        if (!j->exec()) {
+            qDebug() << j->errorText();
+            KMessageBox::error(this, j->errorString(), i18n("KDE Wallet Control Module"));
+            load();
+            return;
+        }
     }
 
     KConfigGroup config(_cfg, "Wallet");
