diff --git src/kdeui/kglobalsettings.cpp src/kdeui/kglobalsettings.cpp
index e58daed..21c8cc2 100644
--- src/kdeui/kglobalsettings.cpp
+++ src/kdeui/kglobalsettings.cpp
@@ -81,7 +81,12 @@ public:
         , isX11(QX11Info::isPlatformX11())
 #endif
     {
+#ifdef Q_OS_MACOS
+        // provide support for all features unlocked when KDE_FULL_SESSION is set
+        kdeFullSession = true;
+#else
         kdeFullSession = !qgetenv("KDE_FULL_SESSION").isEmpty();
+#endif
     }
 
     QPalette createApplicationPalette(const KSharedConfigPtr &config);
