From d17669655dc443398594ea891d27bd0388d9f6c5 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Wed, 27 Nov 2024 22:29:40 +0800
Subject: [PATCH] OwlShmBuffer.m: fix type

---
 Sources/Compositor/OwlShmBuffer.m | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git Sources/Compositor/OwlShmBuffer.m Sources/Compositor/OwlShmBuffer.m
index e984ef1..4368c2b 100644
--- Sources/Compositor/OwlShmBuffer.m
+++ Sources/Compositor/OwlShmBuffer.m
@@ -172,7 +172,8 @@ static void dataReleaseCallback(void *info, const void *data, size_t size) {
 #ifdef OWL_PLATFORM_APPLE
     CGContextRef context = [[NSGraphicsContext currentContext] graphicsPort];
     CGContextSetBlendMode(context, kCGBlendModeCopy);
-    CGContextDrawImage(context, rect, _image);
+    CGRect cgRect = CGRectMake(rect.origin.x, rect.origin.y, rect.size.width, rect.size.height);
+    CGContextDrawImage(context, cgRect, _image);
 #else
     [_rep drawInRect: rect
             fromRect: rect
