From 5076bf90bd13065a5b7e647d9bd53c750216e585 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Wed, 27 Nov 2024 22:48:03 +0800
Subject: [PATCH] OwlIOSurfaceBuffer.m: IOSurfaceDecrementUseCount may not
 exist

---
 Sources/Compositor/OwlIOSurfaceBuffer.m | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/Sources/Compositor/OwlIOSurfaceBuffer.m b/Sources/Compositor/OwlIOSurfaceBuffer.m
index eb25021..9b24fbe 100644
--- Sources/Compositor/OwlIOSurfaceBuffer.m
+++ Sources/Compositor/OwlIOSurfaceBuffer.m
@@ -19,11 +19,12 @@
 #import "OwlFeatures.h"
 #ifdef OWL_PLATFORM_APPLE
 
+#include <AvailabilityMacros.h>
+
 #import "OwlIOSurfaceBuffer.h"
 #import <OpenGL/gl.h>
 #import <OpenGL/CGLIOSurface.h>
 
-
 @implementation OwlIOSurfaceBuffer
 
 - (id) initWithResource: (struct wl_resource *) resource
@@ -35,7 +36,9 @@
 }
 
 - (void) dealloc {
+#if MAC_OS_X_VERSION_MIN_REQUIRED > 1060
     IOSurfaceDecrementUseCount(_surface);
+#endif
     [super dealloc];
 }
 
