diff --git Sources/Compositor/OwlSurface.m Sources/Compositor/OwlSurface.m
index 0774a05..7ce425f 100644
--- Sources/Compositor/OwlSurface.m
+++ Sources/Compositor/OwlSurface.m
@@ -249,6 +249,19 @@ static void surface_set_buffer_transform_handler(
     // TODO
 }
 
+static void surface_damage_buffer_handler(
+    struct wl_client *client,
+    struct wl_resource *resource,
+    int x,
+    int y,
+    int width,
+    int height
+) {
+    OwlSurface *self = wl_resource_get_user_data(resource);
+    NSRect rect = NSMakeRect(x, y, width, height);
+    [self->_pendingState addDamage: rect];
+}
+
 static void surface_frame_handler(
     struct wl_client *client,
     struct wl_resource *resource,
@@ -275,7 +288,8 @@ static const struct wl_surface_interface surface_interface = {
     .set_opaque_region = surface_set_opaque_region_handler,
     .set_input_region = surface_set_input_region_handler,
     .set_buffer_scale = surface_set_buffer_scale_handler,
-    .set_buffer_transform = surface_set_buffer_transform_handler
+    .set_buffer_transform = surface_set_buffer_transform_handler,
+    .damage_buffer = surface_damage_buffer_handler
     // TODO
 };
 
diff --git Sources/Shell/OwlXdgToplevel.m Sources/Shell/OwlXdgToplevel.m
index ecd2f43..aa75912 100644
--- Sources/Shell/OwlXdgToplevel.m
+++ Sources/Shell/OwlXdgToplevel.m
@@ -118,6 +118,36 @@ static void xdg_toplevel_unset_maximized_handler(
     [self->_window unmaximize];
 }
 
+static void xdg_toplevel_set_fullscreen_handler(
+    struct wl_client *client,
+    struct wl_resource *resource,
+    struct wl_resource *output_resource
+) {
+    OwlXdgToplevel *self = wl_resource_get_user_data(resource);
+    // TODO: output_resource
+    NSWindow *window = [self->_window window];
+    if (self->_fullscreen) {
+        return;
+    }
+    if ([window respondsToSelector: @selector(toggleFullScreen:)]) {
+        [window toggleFullScreen: self];
+    }
+}
+
+static void xdg_toplevel_unset_fullscreen_handler(
+    struct wl_client *client,
+    struct wl_resource *resource
+) {
+    OwlXdgToplevel *self = wl_resource_get_user_data(resource);
+    NSWindow *window = [self->_window window];
+    if (!self->_fullscreen) {
+        return;
+    }
+    if ([window respondsToSelector: @selector(toggleFullScreen:)]) {
+        [window toggleFullScreen: self];
+    }
+}
+
 static const struct xdg_toplevel_interface xdg_toplevel_impl = {
     .destroy = xdg_toplevel_destroy_handler,
     .set_parent = xdg_toplevel_set_parent_handler,
@@ -128,7 +158,9 @@ static const struct xdg_toplevel_interface xdg_toplevel_impl = {
     .set_min_size = xdg_toplevel_set_min_size_handler,
     .set_minimized = xdg_toplevel_set_minimized_handler,
     .set_maximized = xdg_toplevel_set_maximized_handler,
-    .unset_maximized = xdg_toplevel_unset_maximized_handler
+    .unset_maximized = xdg_toplevel_unset_maximized_handler,
+    .set_fullscreen = xdg_toplevel_set_fullscreen_handler,
+    .unset_fullscreen = xdg_toplevel_unset_fullscreen_handler
     // TODO
 };
 
@@ -240,7 +272,7 @@ static const struct xdg_toplevel_interface xdg_toplevel_impl = {
     [self sendConfigureWithSize: s];
 }
 
-- (BOOL) windowShouldClose: (NSWindow *) w {
+- (BOOL) windowShouldClose: (id) sender {
     xdg_toplevel_send_close(_resource);
     [[OwlServer sharedServer] flushClientsLater];
     return NO;
@@ -261,6 +293,18 @@ static const struct xdg_toplevel_interface xdg_toplevel_impl = {
     [self sendConfigureWithSize: NSZeroSize];
 }
 
+- (void) windowDidEnterFullScreen: (NSNotification *)notification {
+    _fullscreen = YES;
+    // TODO: This should probably send the new size,
+    // unless windowDidResize: already does.
+    [self sendConfigureWithSize: NSZeroSize];
+}
+
+- (void) windowDidExitFullScreen: (NSNotification *)notification {
+    _fullscreen = YES;
+    [self sendConfigureWithSize: NSZeroSize];
+}
+
 - (void) windowDidBecomeKey: (NSNotification *) notification {
     [[self keyboard] sendEnterSurface: _surface];
     [[self dataDevice] focused];
diff --git Sources/Shell/OwlZxdgToplevelV6.m Sources/Shell/OwlZxdgToplevelV6.m
index 359b675..708d27c 100644
--- Sources/Shell/OwlZxdgToplevelV6.m
+++ Sources/Shell/OwlZxdgToplevelV6.m
@@ -240,7 +240,7 @@ static const struct zxdg_toplevel_v6_interface xdg_toplevel_v6_impl = {
     [self sendConfigureWithSize: s];
 }
 
-- (BOOL) windowShouldClose: (NSWindow *) w {
+- (BOOL) windowShouldClose: (id) sender {
     zxdg_toplevel_v6_send_close(_resource);
     [[OwlServer sharedServer] flushClientsLater];
     return NO;
