diff --git Sources/Server/OwlServer.m Sources/Server/OwlServer.m
index 4ebbcb8..4b917b9 100644
--- Sources/Server/OwlServer.m
+++ Sources/Server/OwlServer.m
@@ -55,11 +55,16 @@
 
         // See if XDG_RUNTIME_DIR is even set.
         NSDictionary *environment = [[NSProcessInfo processInfo] environment];
+        char *envvar = "XDG_RUNTIME_DIR";
         NSString *runtimeDir = [environment objectForKey: @"XDG_RUNTIME_DIR"];
         if (runtimeDir == nil) {
-            [alert setInformativeText: @"XDG_RUNTIME_DIR is unset"];
+            runtimeDir = [environment objectForKey: @"TMPDIR"];
+            envvar = "TMPDIR";
+        }
+        if (runtimeDir == nil) {
+            [alert setInformativeText: @"XDG_RUNTIME_DIR nor TMPDIR are set"];
         } else if (![[NSFileManager defaultManager] fileExistsAtPath: runtimeDir]) {
-            NSString *s = [NSString stringWithFormat: @"XDG_RUNTIME_DIR (%@) does not exist", runtimeDir];
+            NSString *s = [NSString stringWithFormat: @"%s (%@) does not exist", envvar, runtimeDir];
             [alert setInformativeText: s];
         }
         [alert runModal];
