--- setup.py.orig	2016-05-02 22:54:33.000000000 +0200
+++ setup.py	2016-11-18 12:52:15.000000000 +0100
@@ -5,22 +5,20 @@
 import os
 
 setup(
-        name='kapidox',
+        name='kgenapidox',
         version='5.22.0',
         description='KDE API documentation generation tools',
         maintainer = 'Alex Merry',
         maintainer_email = 'alex.merry@kde.org',
         url='https://projects.kde.org/projects/frameworks/kapidox',
         packages = [
-            'kapidox',
-            'kapidox.depdiagram'
+            'kgenapidox'
         ],
         package_dir = {
-            'kapidox': 'src/kapidox',
-            'kapidox.depdiagram': 'src/kapidox/depdiagram',
+            'kgenapidox': 'src/kapidox'
         },
         package_data = {
-            'kapidox': [
+            'kgenapidox': [
                 'data/*.*',
                 'data/htmlresource/*.*',
                 'data/htmlresource/icons/*.*',
@@ -29,16 +27,11 @@
         },
         scripts = [
             'src/kgenapidox',
-            'src/kgenframeworksapidox',
-            'src/depdiagram-prepare',
-            'src/depdiagram-generate',
-            'src/depdiagram-generate-all',
+            'src/kgenframeworksapidox'
         ],
         data_files= [
             (os.path.join('share', 'man', 'man1'), ['docs/kgenapidox.1',
-                'docs/kgenframeworksapidox.1', 'docs/depdiagram-prepare.1',
-                'docs/depdiagram-generate.1',
-                'docs/depdiagram-generate-all.1'])],
+                'docs/kgenframeworksapidox.1'])],
         classifiers = [
             'Environment :: Console',
             'Intended Audience :: Developers',
--- src/kgenapidox.orig	2016-05-02 22:54:33.000000000 +0200
+++ src/kgenapidox	2016-11-18 12:55:39.000000000 +0100
@@ -35,9 +35,9 @@
 import shutil
 import tempfile
 
-from kapidox import argparserutils
-from kapidox import utils
-from kapidox.generator import *
+from kgenapidox import argparserutils
+from kgenapidox import utils
+from kgenapidox.generator import *
 
 DEFAULT_OUTPUT_DIR = 'apidocs'
 
--- src/kgenframeworksapidox.orig	2016-05-02 22:54:33.000000000 +0200
+++ src/kgenframeworksapidox	2016-11-18 12:55:52.000000000 +0100
@@ -41,11 +41,11 @@
 import jinja2
 import yaml
 
-from kapidox import argparserutils
-from kapidox import utils
-from kapidox.generator import *
+from kgenapidox import argparserutils
+from kgenapidox import utils
+from kgenapidox.generator import *
 try:
-    from kapidox import depdiagram
+    from kgenapidox import depdiagram
     DEPDIAGRAM_AVAILABLE = True
 except ImportError:
     DEPDIAGRAM_AVAILABLE = False
