diff --git a/CMakeLists.txt b/CMakeLists.txt
index c8899a13d6754413305f35c5e7b646208082e892..6cd5b2b8b77a7842f078ad8e6a48caea386da657 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -22,6 +22,7 @@ include(ECMSetupVersion)
 include(FeatureSummary)
 include(GenerateExportHeader)
 include(KDECompilerSettings)
+include(ECMMarkNonGuiExecutable)
 
 enable_testing()
 
diff --git a/src/cli/CMakeLists.txt b/src/cli/CMakeLists.txt
index 67a616ec1f88932029b5692f1f060dce91560fd2..104f58a9bddccf5db72ef137afa40b5507007100 100644
--- a/src/cli/CMakeLists.txt
+++ b/src/cli/CMakeLists.txt
@@ -3,6 +3,7 @@ set(uf_cli_srcs
 )
 
 add_executable(userfeedbackctl ${uf_cli_srcs})
+ecm_mark_nongui_executable(userfeedbackctl)
 target_compile_features(userfeedbackctl PRIVATE cxx_generic_lambdas)
 target_link_libraries(userfeedbackctl KUserFeedbackConsole)
 install(TARGETS userfeedbackctl ${KDE_INSTALL_TARGETS_DEFAULT_ARGS})
