--- kioclient/orig.CMakeLists.txt	2015-12-15 11:19:17.000000000 +0100
+++ kioclient/CMakeLists.txt	2015-12-21 10:44:09.486336289 +0100
@@ -9,6 +9,7 @@
 
 function(add_kioclient_interface TARGET_NAME)
     add_executable(${TARGET_NAME} kioclient.cpp)
+    ecm_mark_nongui_executable(${TARGET_NAME})
     string(TOUPPER "${TARGET_NAME}" UPPER_TARGET_NAME)
     target_compile_definitions(${TARGET_NAME} PRIVATE "-DKIOCLIENT_AS_${UPPER_TARGET_NAME}")
     target_compile_definitions(${TARGET_NAME} PRIVATE -DPROJECT_VERSION="${PROJECT_VERSION}")
@@ -27,6 +28,7 @@
 
 if (NOT KIOCORE_ONLY)
 add_executable(kde-open5 kioclient.cpp)
+ecm_mark_nongui_executable(kde-open5)
 target_compile_definitions(kde-open5 PRIVATE "-DKIOCLIENT_AS_KDEOPEN")
 target_compile_definitions(kde-open5 PRIVATE -DPROJECT_VERSION="${PROJECT_VERSION}")
 target_link_libraries(kde-open5 Qt5::DBus KF5::CoreAddons KF5::KIOWidgets KF5::I18n)
--- keditfiletype/orig.CMakeLists.txt	2015-12-15 11:19:17.000000000 +0100
+++ keditfiletype/CMakeLists.txt	2015-12-21 10:44:56.620338253 +0100
@@ -35,6 +35,7 @@
 set(keditfiletype_SRCS keditfiletype.cpp ${libfiletypes_SRCS})
 
 add_executable(keditfiletype5 ${keditfiletype_SRCS})
+ecm_mark_nongui_executable(keditfiletype5)
 target_compile_definitions(keditfiletype5 PRIVATE -DPROJECT_VERSION="${PROJECT_VERSION}")
 
 target_link_libraries(keditfiletype5
--- kdesu/orig.CMakeLists.txt	2015-12-15 11:19:17.000000000 +0100
+++ kdesu/CMakeLists.txt	2015-12-21 10:45:56.412340745 +0100
@@ -1,4 +1,5 @@
 add_executable(kdesu_executable kdesu.cpp sudlg.cpp)
+ecm_mark_nongui_executable(kdesu_executable)
 target_compile_definitions(kdesu_executable PRIVATE -DPROJECT_VERSION="${PROJECT_VERSION}")
 
 # in KDELibsDependencies.cmake installed by kdelibs there is a dependency to "kdesu"
--- kmimetypefinder/orig.CMakeLists.txt	2015-12-15 11:19:17.000000000 +0100
+++ kmimetypefinder/CMakeLists.txt	2015-12-21 10:46:28.089342065 +0100
@@ -1,4 +1,5 @@
 add_executable(kmimetypefinder5 kmimetypefinder.cpp)
+ecm_mark_nongui_executable(kmimetypefinder5)
 target_compile_definitions(kmimetypefinder5 PRIVATE -DPROJECT_VERSION="${PROJECT_VERSION}")
 ecm_mark_nongui_executable(kmimetypefinder5)
 
--- kcmshell/orig.CMakeLists.txt  2015-12-15 11:19:17.000000000 +0100
+++ kcmshell/CMakeLists.txt     2015-12-22 14:09:38.000000000 +0100
@@ -3,7 +3,7 @@
 
 kf5_add_kdeinit_executable( kcmshell5 ${kcmshell_KDEINIT_SRCS})
 target_compile_definitions(kdeinit_kcmshell5 PRIVATE -DPROJECT_VERSION="${PROJECT_VERSION}")
-ecm_mark_nongui_executable(kcmshell5)
+#ecm_mark_nongui_executable(kcmshell5)
 
 target_link_libraries(kdeinit_kcmshell5
    Qt5::DBus
