--- orig.CMakeLists.txt	2015-12-15 11:19:49.000000000 +0100
+++ CMakeLists.txt	2015-12-21 10:34:45.000000000 +0100
@@ -49,11 +49,13 @@
 # Disables automatic conversions from QString (or char *) to QUrl.
 add_definitions(-DQT_NO_URL_CAST_FROM_STRING)
 
-find_package(X11)
-find_package(XCB COMPONENTS XCB)
-if(X11_FOUND AND XCB_XCB_FOUND)
-  set(HAVE_X11 1)
-  find_package(Qt5X11Extras ${QT_MIN_VERSION} CONFIG)
+if(NOT APPLE)
+  find_package(X11)
+  find_package(XCB COMPONENTS XCB)
+  if(X11_FOUND AND XCB_XCB_FOUND)
+    set(HAVE_X11 1)
+    find_package(Qt5X11Extras ${QT_MIN_VERSION} CONFIG)
+  endif()
 endif()
 
 include(KDEInstallDirs)
