--- pycontrib/gdraw/orig._gdraw.py	2017-07-22 07:00:39.000000000 +0200
+++ pycontrib/gdraw/_gdraw.py	2019-03-03 18:05:41.000000000 +0100
@@ -1,7 +1,7 @@
 from ctypes import *
 
 _libraries = {}
-_libraries['libgdraw.so.5'] = CDLL('libgdraw.so.5')
+_libraries['libgdraw.5.dylib'] = CDLL('@PREFIX@/lib/libgdraw.5.dylib')
 STRING = c_char_p
 
 
@@ -20,13 +20,13 @@
 class gwindow_attrs(Structure):
     pass
 GWindowAttrs = gwindow_attrs
-GDrawCreateTopWindow = _libraries['libgdraw.so.5'].GDrawCreateTopWindow
+GDrawCreateTopWindow = _libraries['libgdraw.5.dylib'].GDrawCreateTopWindow
 GDrawCreateTopWindow.restype = GWindow
 GDrawCreateTopWindow.argtypes = [POINTER(GDisplay), POINTER(GRect), CFUNCTYPE(c_int, GWindow, POINTER(GEvent)), c_void_p, POINTER(GWindowAttrs)]
-GDrawDestroyWindow = _libraries['libgdraw.so.5'].GDrawDestroyWindow
+GDrawDestroyWindow = _libraries['libgdraw.5.dylib'].GDrawDestroyWindow
 GDrawDestroyWindow.restype = None
 GDrawDestroyWindow.argtypes = [GWindow]
-GDrawSetVisible = _libraries['libgdraw.so.5'].GDrawSetVisible
+GDrawSetVisible = _libraries['libgdraw.5.dylib'].GDrawSetVisible
 GDrawSetVisible.restype = None
 GDrawSetVisible.argtypes = [GWindow, c_int]
 class gtimer(Structure):
@@ -34,10 +34,10 @@
 GTimer = gtimer
 int32_t = c_int32
 int32 = int32_t
-GDrawRequestTimer = _libraries['libgdraw.so.5'].GDrawRequestTimer
+GDrawRequestTimer = _libraries['libgdraw.5.dylib'].GDrawRequestTimer
 GDrawRequestTimer.restype = POINTER(GTimer)
 GDrawRequestTimer.argtypes = [GWindow, int32, int32, c_void_p]
-GDrawCancelTimer = _libraries['libgdraw.so.5'].GDrawCancelTimer
+GDrawCancelTimer = _libraries['libgdraw.5.dylib'].GDrawCancelTimer
 GDrawCancelTimer.restype = None
 GDrawCancelTimer.argtypes = [POINTER(GTimer)]
 class ggc(Structure):
