diff --git a/libcxx/CMakeLists.txt b/libcxx/CMakeLists.txt
index 5ef0a77..11d637f 100644
--- a/libcxx/CMakeLists.txt
+++ b/libcxx/CMakeLists.txt
@@ -82,6 +82,11 @@ endif()
 include(CMakeDependentOption)
 include(HandleCompilerRT)
 
+if (NOT DEFINED LIBCXX_INSTALL_PREFIX)
+    set(LIBCXX_INSTALL_PREFIX "/usr")
+endif()
+set(LIBCXX_INSTALL_PREFIX ${LIBCXX_INSTALL_PREFIX} CACHE PATH "Location where libc++ and libc++abi will be installed")
+
 # Basic options ---------------------------------------------------------------
 option(LIBCXX_ENABLE_ASSERTIONS "Enable assertions independent of build mode." OFF)
 option(LIBCXX_ENABLE_SHARED "Build libc++ as a shared library." ON)
diff --git a/libcxx/src/CMakeLists.txt b/libcxx/src/CMakeLists.txt
index 2a147b0..bccc758 100644
--- a/libcxx/src/CMakeLists.txt
+++ b/libcxx/src/CMakeLists.txt
@@ -1,4 +1,4 @@
-set(LIBCXX_LIB_CMAKEFILES_DIR "${CMAKE_CURRENT_BINARY_DIR}${CMAKE_FILES_DIRECTORY}"  PARENT_SCOPE)
+set(LIBCXX_LIB_CMAKEFILES_DIR "${CMAKE_BINARY_DIR}/lib${CMAKE_FILES_DIRECTORY}"  PARENT_SCOPE)
 
 # Get sources
 set(LIBCXX_SOURCES
@@ -200,7 +200,7 @@ if (LIBCXX_ENABLE_SHARED)
       COMPILE_FLAGS "${LIBCXX_COMPILE_FLAGS}"
       LINK_FLAGS    "${LIBCXX_LINK_FLAGS}"
       OUTPUT_NAME   "c++"
-      VERSION       "${LIBCXX_ABI_VERSION}.0"
+      VERSION       "${VERSION_ACCORDING_TO_MACPORTS}"
       SOVERSION     "${LIBCXX_ABI_VERSION}"
       DEFINE_SYMBOL ""
       POSITION_INDEPENDENT_CODE ON
@@ -244,6 +244,8 @@ if (LIBCXX_ENABLE_SHARED)
 if (CMAKE_OSX_DEPLOYMENT_TARGET GREATER_EQUAL 10.7)
   if (LIBCXX_OSX_REEXPORT_LIBCXXABI_SYMBOLS)
     target_link_libraries(cxx_shared PRIVATE
+      "-current_version ${VERSION_ACCORDING_TO_MACPORTS}"
+      "-install_name ${LIBCXX_INSTALL_PREFIX}/lib/libc++.1.dylib"
       "-Wl,-unexported_symbols_list,${CMAKE_CURRENT_SOURCE_DIR}/../lib/libc++unexp.exp"
       "-Wl,-reexported_symbols_list,${CMAKE_CURRENT_SOURCE_DIR}/../lib/libc++abi.v${LIBCXX_LIBCPPABI_VERSION}.exp"
       "-Wl,-force_symbols_not_weak_list,${CMAKE_CURRENT_SOURCE_DIR}/../lib/notweak.exp"
@@ -388,23 +390,23 @@ endif()
 
 if (LIBCXX_INSTALL_SHARED_LIBRARY)
   install(TARGETS cxx_shared
-    ARCHIVE DESTINATION ${LIBCXX_INSTALL_LIBRARY_DIR} COMPONENT cxx
-    LIBRARY DESTINATION ${LIBCXX_INSTALL_LIBRARY_DIR} COMPONENT cxx
+     ARCHIVE DESTINATION ${LIBCXX_INSTALL_PREFIX}/${LIBCXX_INSTALL_LIBRARY_DIR} COMPONENT cxx
+     LIBRARY DESTINATION ${LIBCXX_INSTALL_PREFIX}/${LIBCXX_INSTALL_LIBRARY_DIR} COMPONENT cxx
     RUNTIME DESTINATION bin COMPONENT cxx)
 endif()
 
 if (LIBCXX_INSTALL_STATIC_LIBRARY)
   install(TARGETS cxx_static
-    ARCHIVE DESTINATION ${LIBCXX_INSTALL_LIBRARY_DIR} COMPONENT cxx
-    LIBRARY DESTINATION ${LIBCXX_INSTALL_LIBRARY_DIR} COMPONENT cxx
+     ARCHIVE DESTINATION ${LIBCXX_INSTALL_PREFIX}/${LIBCXX_INSTALL_LIBRARY_DIR} COMPONENT cxx
+     LIBRARY DESTINATION ${LIBCXX_INSTALL_PREFIX}/${LIBCXX_INSTALL_LIBRARY_DIR} COMPONENT cxx
     RUNTIME DESTINATION bin COMPONENT cxx)
 endif()
 
 if(LIBCXX_INSTALL_EXPERIMENTAL_LIBRARY)
   install(TARGETS cxx_experimental
-    LIBRARY DESTINATION ${LIBCXX_INSTALL_LIBRARY_DIR} COMPONENT cxx
-    ARCHIVE DESTINATION ${LIBCXX_INSTALL_LIBRARY_DIR} COMPONENT cxx
-    RUNTIME DESTINATION bin COMPONENT cxx)
+     LIBRARY DESTINATION ${LIBCXX_INSTALL_PREFIX}/${LIBCXX_INSTALL_LIBRARY_DIR} COMPONENT cxx
+     ARCHIVE DESTINATION ${LIBCXX_INSTALL_PREFIX}/${LIBCXX_INSTALL_LIBRARY_DIR} COMPONENT cxx
+     RUNTIME DESTINATION ${LIBCXX_INSTALL_PREFIX}/bin COMPONENT cxx)
 endif()
 
 # NOTE: This install command must go after the cxx install command otherwise
diff --git a/libcxxabi/CMakeLists.txt b/libcxxabi/CMakeLists.txt
index dac1aab..d30a7f3 100644
--- a/libcxxabi/CMakeLists.txt
+++ b/libcxxabi/CMakeLists.txt
@@ -507,6 +507,23 @@ string(REPLACE ";" " " LIBCXXABI_CXX_FLAGS "${LIBCXXABI_CXX_FLAGS}")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${LIBCXXABI_CXX_FLAGS}")
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${LIBCXXABI_C_FLAGS}")
 
+if ( APPLE )
+  if ( CMAKE_OSX_DEPLOYMENT_TARGET STREQUAL "10.6" )
+    list(APPEND LIBCXXABI_COMPILE_FLAGS "-U__STRICT_ANSI__")
+    list(APPEND LIBCXXABI_LINK_FLAGS
+      "-compatibility_version 1"
+      "-current_version ${VERSION_ACCORDING_TO_MACPORTS}"
+      "-install_name ${LIBCXX_INSTALL_PREFIX}/lib/libc++abi.1.dylib")
+    list(APPEND LIBCXXABI_LINK_FLAGS
+        "/usr/lib/libSystem.B.dylib")
+  else()
+    list(APPEND LIBCXXABI_LINK_FLAGS
+      "-compatibility_version 1"
+      "-current_version ${VERSION_ACCORDING_TO_MACPORTS}"
+      "-install_name ${LIBCXX_INSTALL_PREFIX}/lib/libc++abi.1.dylib")
+  endif()
+endif()
+
 #===============================================================================
 # Setup Source Code
 #===============================================================================
diff --git a/libcxxabi/src/CMakeLists.txt b/libcxxabi/src/CMakeLists.txt
index de6fe91..cb808b1 100644
--- a/libcxxabi/src/CMakeLists.txt
+++ b/libcxxabi/src/CMakeLists.txt
@@ -189,7 +189,7 @@ if (LIBCXXABI_ENABLE_SHARED)
       LINK_FLAGS "${LIBCXXABI_LINK_FLAGS}"
       OUTPUT_NAME "c++abi"
       SOVERSION "1"
-      VERSION "${LIBCXXABI_LIBRARY_VERSION}"
+      VERSION "${VERSION_ACCORDING_TO_MACPORTS}"
       DEFINE_SYMBOL ""
       POSITION_INDEPENDENT_CODE ON
   )
