diff --git deps/v8/src/wasm/orig.wasm-objects.cc deps/v8/src/wasm/wasm-objects.cc
index 91ebf45..7222b90 100644
--- deps/v8/src/wasm/orig.wasm-objects.cc
+++ deps/v8/src/wasm/wasm-objects.cc
@@ -5,10 +5,56 @@
 #if V8_TARGET_OS_LINUX
 #include <sys/mman.h>
 #include <sys/stat.h>
+#include <linux/memfd.h>
 // `sys/mman.h defines `MAP_TYPE`, but `MAP_TYPE` also gets defined within V8.
 // Since we don't need `sys/mman.h`'s `MAP_TYPE`, we undefine it immediately
 // after the `#include`.
 #undef MAP_TYPE
+
+/* memfd_create was introduced in glibc 2.27 */
+#if !defined(__GLIBC__) || (__GLIBC__ >= 2 && __GLIBC_MINOR__ >= 27)
+// all good
+#else
+#include <unistd.h>
+#include <sys/syscall.h>
+
+#ifndef SYS_memfd_create
+#  ifndef __NR_memfd_create
+#    if defined __x86_64__
+#      define __NR_memfd_create 319
+#    elif defined __arm__
+#      define __NR_memfd_create 385
+#    elif defined __aarch64__
+#      define __NR_memfd_create 279
+#    elif defined __s390__
+#      define __NR_memfd_create 350
+#    elif defined _MIPS_SIM
+#      if _MIPS_SIM == _MIPS_SIM_ABI32
+#        define __NR_memfd_create 4354
+#      endif
+#      if _MIPS_SIM == _MIPS_SIM_NABI32
+#        define __NR_memfd_create 6318
+#      endif
+#      if _MIPS_SIM == _MIPS_SIM_ABI64
+#        define __NR_memfd_create 5314
+#      endif
+#    elif defined __i386__
+#      define __NR_memfd_create 356
+#    elif defined __arc__
+#      define __NR_memfd_create 279
+#    else
+#      warning "__NR_memfd_create unknown for your architecture"
+#    endif
+#  endif
+#define SYS_memfd_create __NR_memfd_create
+#endif
+
+static inline int memfd_create(const char *name, unsigned int flags)
+{
+	return syscall(SYS_memfd_create, name, flags);
+}
+#endif
+
 #endif  // V8_TARGET_OS_LINUX
 
 #include "src/wasm/wasm-objects.h"
