diff --git src/crypto/orig.crypto_context.cc src/crypto/crypto_context.cc
index d1430cd..483d1f3 100644
--- src/crypto/orig.crypto_context.cc
+++ src/crypto/crypto_context.cc
@@ -20,6 +20,7 @@
 #endif  // !OPENSSL_NO_ENGINE
 #ifdef __APPLE__
 #include <Security/Security.h>
+# include <AvailabilityMacros.h>
 #endif
 
 #ifdef _WIN32
@@ -412,6 +413,7 @@ TrustStatus IsTrustSettingsTrustedForPolicy(CFArrayRef trust_settings,
   return TrustStatus::UNSPECIFIED;
 }
 
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= 101400
 bool IsCertificateTrustValid(SecCertificateRef ref) {
   SecTrustRef sec_trust = nullptr;
   CFMutableArrayRef subj_certs =
@@ -439,6 +441,7 @@ bool IsCertificateTrustValid(SecCertificateRef ref) {
   }
   return result;
 }
+#endif
 
 bool IsCertificateTrustedForPolicy(X509* cert, SecCertificateRef ref) {
   OSStatus err;
@@ -474,10 +477,12 @@ bool IsCertificateTrustedForPolicy(X509* cert, SecCertificateRef ref) {
     // No trust-settings array means
     // “this certificate must be verifiable using a known trusted certificate”.
     if (trust_settings == nullptr && !trust_evaluated) {
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= 101400
       bool result = IsCertificateTrustValid(ref);
       if (result) {
         return true;
       }
+#endif
       // no point re-evaluating this in the admin domain
       trust_evaluated = true;
     } else if (trust_settings) {
