diff --git orig.node.gypi node.gypi
index e9b3665..e4144df 100644
--- orig.node.gypi
+++ node.gypi
@@ -26,13 +26,6 @@
   },
 
   'conditions': [
-    [ 'clang==1', {
-      'cflags': [
-        '-Werror=undefined-inline',
-        '-Werror=extra-semi',
-        '-Werror=ctad-maybe-unsupported',
-      ],
-    }],
     [ '"<(_type)"=="executable"', {
       'msvs_settings': {
         'VCManifestTool': {
