diff --git CMakeLists.txt CMakeLists.txt
index f0820a8..c4fc04f 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -13,6 +13,7 @@ include(ECMSetupVersion)
 include(CMakePackageConfigHelpers)
 include(FeatureSummary)
 include(WriteBasicConfigVersionFile)
+include(ECMMarkNonGuiExecutable)
 
 include(KDEInstallDirs)
 include(KDECMakeSettings)
@@ -70,6 +71,7 @@ set(policykit_SRCS
 )
 
 add_executable(polkit-kde-authentication-agent-1 ${policykit_SRCS})
+ecm_mark_nongui_executable(polkit-kde-authentication-agent-1)
 
 target_link_libraries(polkit-kde-authentication-agent-1
     KF5::DBusAddons
