diff --git akonadi/resourcebase.cpp akonadi/resourcebase.cpp
index 8a5bb60..06e4168 100644
--- akonadi/resourcebase.cpp
+++ akonadi/resourcebase.cpp
@@ -954,7 +954,10 @@ void ResourceBasePrivate::slotSynchronizeCollectionAttributes(const Collection &
     Akonadi::CollectionFetchJob *fetchJob = new Akonadi::CollectionFetchJob(col, CollectionFetchJob::Base, this);
     fetchJob->setFetchScope(q->changeRecorder()->collectionFetchScope());
     connect(fetchJob, SIGNAL(result(KJob*)), q, SLOT(slotAttributeRetrievalCollectionFetchDone(KJob*)));
-    Q_ASSERT(!mCurrentCollectionFetchJob);
+//     Q_ASSERT(!mCurrentCollectionFetchJob);
+    if (mCurrentCollectionFetchJob) {
+        qWarning() << Q_FUNC_INFO << "mCurrentCollectionFetchJob != NULL" << mCurrentCollectionFetchJob;
+    }
     mCurrentCollectionFetchJob = fetchJob;
 }
 
