diff --git akonadi/attributefactory.cpp akonadi/attributefactory.cpp
index 47a8839..184c224 100644
--- akonadi/attributefactory.cpp
+++ akonadi/attributefactory.cpp
@@ -138,7 +138,7 @@ void AttributeFactory::registerAttribute(Attribute *attr)
     Q_ASSERT(!attr->type().contains(' ') && !attr->type().contains('\'') && !attr->type().contains('"'));
     QHash<QByteArray, Attribute *>::Iterator it = d->attributes.find(attr->type());
     if (it != d->attributes.end()) {
-        delete *it;
+//         delete *it;
         d->attributes.erase(it);
     }
     d->attributes.insert(attr->type(), attr);
