diff --git kdeui/notifications/kstatusnotifieritem.cpp kdeui/notifications/kstatusnotifieritem.cpp
index 1b15d40..1ed7de3 100644
--- kdeui/notifications/kstatusnotifieritem.cpp
+++ kdeui/notifications/kstatusnotifieritem.cpp
@@ -683,6 +683,9 @@ bool KStatusNotifierItemPrivate::checkVisibility(QPoint pos, bool perform)
 
         return false;
     }
+#else
+    Q_UNUSED(pos);
+    Q_UNUSED(perform);
 #endif
 
     return true;
@@ -745,8 +748,20 @@ void KStatusNotifierItemPrivate::init(const QString &extraId)
 
     //create a default menu, just like in KSystemtrayIcon
     KMenu *m = new KMenu(associatedWidget);
+#ifdef Q_OS_MAC
+    // emulate addTitle/setTitle by adding an inactive menu item.
+    titleAction = m->addAction( qApp->windowIcon(), KGlobal::mainComponent().aboutData()->programName() );
+    titleAction->setEnabled(false);
+    titleAction->setIconVisibleInMenu(true);
+    m->addAction( titleAction );
+    m->addSeparator();
+    kDebug() << "### Added SystemTray titleAction=" << titleAction;
+    kDebug() << "### SystemTray for app" << qApp << "=" << KGlobal::mainComponent().aboutData()->programName()
+        << "/" << KGlobal::caption() << "with icon" << qApp->windowIcon().name();
+#else
     titleAction = m->addTitle(qApp->windowIcon(), KGlobal::caption());
     m->setTitle(KGlobal::mainComponent().aboutData()->programName());
+#endif
     q->setContextMenu(m);
 
     KStandardAction::quit(q, SLOT(maybeQuit()), actionCollection);
diff --git kdeui/actions/kaction.cpp kdeui/actions/kaction.cpp
index 9e8f7fb..54088d2 100644
--- kdeui/actions/kaction.cpp
+++ kdeui/actions/kaction.cpp
@@ -140,6 +140,7 @@ KAction::KAction(QObject *parent)
   : QWidgetAction(parent), d(new KActionPrivate)
 {
   d->init(this);
+  setMenuRole(QAction::NoRole);
 }
 
 KAction::KAction(const QString &text, QObject *parent)
@@ -147,6 +148,7 @@ KAction::KAction(const QString &text, QObject *parent)
 {
   d->init(this);
   setText(text);
+  setMenuRole(QAction::NoRole);
 }
 
 KAction::KAction(const KIcon &icon, const QString &text, QObject *parent)
@@ -155,6 +157,7 @@ KAction::KAction(const KIcon &icon, const QString &text, QObject *parent)
   d->init(this);
   setIcon(icon);
   setText(text);
+  setMenuRole(QAction::NoRole);
 }
 
 KAction::~KAction()
