diff --git a/kdeui/actions/kaction.cpp b/kdeui/actions/kaction.cpp
index 9e8f7fb..54088d2 100644
--- a/kdeui/actions/kaction.cpp
+++ b/kdeui/actions/kaction.cpp
@@ -140,6 +140,7 @@ KAction::KAction(QObject *parent)
   : QWidgetAction(parent), d(new KActionPrivate)
 {
   d->init(this);
+  setMenuRole(QAction::NoRole);
 }
 
 KAction::KAction(const QString &text, QObject *parent)
@@ -147,6 +148,7 @@ KAction::KAction(const QString &text, QObject *parent)
 {
   d->init(this);
   setText(text);
+  setMenuRole(QAction::NoRole);
 }
 
 KAction::KAction(const KIcon &icon, const QString &text, QObject *parent)
@@ -155,6 +157,7 @@ KAction::KAction(const KIcon &icon, const QString &text, QObject *parent)
   d->init(this);
   setIcon(icon);
   setText(text);
+  setMenuRole(QAction::NoRole);
 }
 
 KAction::~KAction()
