--- kdelibs-git/kdecore/kernel/orig.kkernel_mac.cpp	2014-09-02 13:13:06.000000000 +0200
+++ kdelibs-git/kdecore/kernel/kkernel_mac.cpp	2014-09-06 12:45:38.000000000 +0200
@@ -52,14 +52,19 @@
 */
 
 QString convert_CFString_to_QString(CFStringRef str) {
-	CFIndex length = CFStringGetLength(str);
-	const UniChar *chars = CFStringGetCharactersPtr(str);
-	if (chars)
-		return QString(reinterpret_cast<const QChar *>(chars), length);
+	if( str ){
+		CFIndex length = CFStringGetLength(str);
+		const UniChar *chars = CFStringGetCharactersPtr(str);
+		if (chars)
+			return QString(reinterpret_cast<const QChar *>(chars), length);
 
-	QVarLengthArray<UniChar> buffer(length);
-	CFStringGetCharacters(str, CFRangeMake(0, length), buffer.data());
-	return QString(reinterpret_cast<const QChar *>(buffer.constData()), length);
+		QVarLengthArray<UniChar> buffer(length);
+		CFStringGetCharacters(str, CFRangeMake(0, length), buffer.data());
+		return QString(reinterpret_cast<const QChar *>(buffer.constData()), length);
+	}
+	else{
+		return QString();
+	}
 }
 
 /**
