diff --git a/kdecore/services/kmimetyperepository.cpp b/kdecore/services/kmimetyperepository.cpp
index a8fa4c1..7a4fd42 100644
--- a/kdecore/services/kmimetyperepository.cpp
+++ b/kdecore/services/kmimetyperepository.cpp
@@ -699,6 +699,7 @@ static void addPlatformSpecificPkgConfigPath(QStringList& paths)
 #elif defined(Q_OS_OPENBSD) || defined(Q_OS_NETBSD) || defined(Q_OS_SOLARIS)
     paths << QLatin1String("/usr/local/lib/pkgconfig"); // {Net,Open}BSD/OpenSolaris
 #elif defined (Q_OS_UNIX)
+    paths << QLatin1String("@PREFIX@/share/pkgconfig"); // MacPorts
     paths << QLatin1String("/usr/share/pkgconfig"); // Linux and all other unix
 #endif
 }
