diff --git a/kparts/part.cpp b/kparts/part.cpp
index 2cfee81..20089d4 100644
--- a/kparts/part.cpp
+++ b/kparts/part.cpp
@@ -350,7 +350,7 @@ void Part::slotWidgetDestroyed()
     d->m_widget = 0;
     if (d->m_autoDeletePart) {
         kDebug(1000) << "deleting part" << objectName();
-        delete this; // ouch, this should probably be deleteLater()
+        this->deleteLater();
     }
 }
 
diff --git a/kparts/tests/parttest.cpp b/kparts/tests/parttest.cpp
index e48b578..232aa07 100644
--- a/kparts/tests/parttest.cpp
+++ b/kparts/tests/parttest.cpp
@@ -48,6 +48,7 @@ void PartTest::testAutoDeletePart()
     KParts::Part* part = new TestPart(0, 0);
     QPointer<KParts::Part> partPointer(part);
     delete part->widget();
+    QCoreApplication::sendPostedEvents(0, QEvent::DeferredDelete);
     QVERIFY(partPointer.isNull());
 }
 
@@ -57,6 +58,7 @@ void PartTest::testAutoDeleteWidget()
     QPointer<KParts::Part> partPointer(part);
     QPointer<QWidget> widgetPointer(part->widget());
     delete part;
+    QCoreApplication::sendPostedEvents(0, QEvent::DeferredDelete);
     QVERIFY(widgetPointer.isNull());
 }
 
