# backport from https://git.reviewboard.kde.org/r/128005

diff --git a/kdeui/widgets/kmultitabbar.cpp b/kdeui/widgets/kmultitabbar.cpp
index d090f5c04c682d9474d0becd4de64cebcfc8f869..62112aed74796ae7db3ba5816b6d0a6b5aa507fd 100644
--- a/kdeui/widgets/kmultitabbar.cpp
+++ b/kdeui/widgets/kmultitabbar.cpp
@@ -151,6 +151,7 @@ KMultiTabBarButton::KMultiTabBarButton(const QPixmap& pic, const QString& text,
 	// If keyboard navigation is wanted, then only the bar should take focus,
 	// and arrows could change the focused button; but generally, tabbars don't take focus anyway.
 	setFocusPolicy(Qt::NoFocus);
+	setAttribute(Qt::WA_LayoutUsesWidgetRect);
 }
 
 KMultiTabBarButton::~KMultiTabBarButton()
diff --git a/kfile/kurlnavigatorbutton.cpp b/kfile/kurlnavigatorbutton.cpp
index a37f2e63e93ee2dc71a8afd2a25ddce8bee7bf66..e11545a413f431bf567a23ad229d40fca015ed56 100644
--- a/kfile/kurlnavigatorbutton.cpp
+++ b/kfile/kurlnavigatorbutton.cpp
@@ -55,6 +55,7 @@ KUrlNavigatorButton::KUrlNavigatorButton(const KUrl& url, QWidget* parent) :
     setAcceptDrops(true);
     setUrl(url);
     setMouseTracking(true);
+    setAttribute(Qt::WA_LayoutUsesWidgetRect);
 
     m_openSubDirsTimer = new QTimer(this);
     m_openSubDirsTimer->setSingleShot(true);
