diff --git a/cmake/modules/FindSoprano.cmake b/cmake/modules/FindSoprano.cmake
index 99daa4f..b4f48e2 100644
--- a/cmake/modules/FindSoprano.cmake
+++ b/cmake/modules/FindSoprano.cmake
@@ -142,7 +142,11 @@ endif( Soprano_FIND_COMPONENTS )
 
 #look for parser plugins
 if(SOPRANO_INCLUDE_DIR)
-  get_filename_component(_SOPRANO_PREFIX ${SOPRANO_INCLUDE_DIR} PATH)
+  if(SOPRANO_PREFIX)
+    set(_SOPRANO_PREFIX ${SOPRANO_PREFIX})
+  else()
+    get_filename_component(_SOPRANO_PREFIX ${SOPRANO_INCLUDE_DIR} PATH)
+  endif()
 
   find_path(SOPRANO_PLUGIN_ROOT_DIR 
     NAMES

