diff --git a/kinit/klauncher.cpp b/kinit/klauncher.cpp
index 1a047b7..6f6d52c 100644
--- a/kinit/klauncher.cpp
+++ b/kinit/klauncher.cpp
@@ -57,6 +57,10 @@
 #include <kio/connection.h>
 #include <kio/slaveinterface.h>
 
+#ifdef Q_OS_MAC
+#include <AvailabilityMacros.h>
+#endif
+
 // Dispose slaves after being idle for SLAVE_MAX_IDLE seconds
 #define SLAVE_MAX_IDLE	30
 
@@ -1189,7 +1193,11 @@ KLauncher::requestSlave(const QString &protocol,
        request_header.arg_length = 0;
        kde_safe_write(kdeinitSocket, &request_header, sizeof(request_header));
 #else
+#if !defined(__MAC_OS_X_VERSION_MAX_ALLOWED) || __MAC_OS_X_VERSION_MAX_ALLOWED <= 1070
       name = QString::fromLatin1("gdb");
+#else // Mac OS X >= 10.8
+      name = QString::fromLatin1("lldb");
+#endif
 #endif
     }
     if (mSlaveValgrind == protocol) {
@@ -1218,6 +1226,9 @@ KLauncher::requestSlave(const QString &protocol,
     request->status = KLaunchRequest::Launching;
     requestStart(request);
     pid_t pid = request->pid;
+#ifdef Q_OS_MAC
+    kDebug(7016) << "Slave" << name << "launched, args=" << arg_list << "pid=" << pid << "for protocol" << protocol;
+#endif
 
 //    kDebug(7016) << "Slave launched, pid = " << pid;
 
