diff --git a/kutils/CMakeLists.txt b/kutils/CMakeLists.txt
index b3eb3d6..a56b4aa 100644
--- a/kutils/CMakeLists.txt
+++ b/kutils/CMakeLists.txt
@@ -58,6 +58,9 @@ endif (Q_WS_X11)
 
 kde4_add_library(kidletime ${LIBRARY_TYPE} ${kidletime_LIB_SRCS})
 target_link_libraries(kidletime LINK_PUBLIC kdecore kdeui ${QT_QTGUI_LIBRARY})
+if (Q_WS_MAC)
+    target_link_libraries(kidletime "-framework IOKit")
+endif (Q_WS_MAC)
 target_link_libraries(kidletime LINK_PRIVATE ${KDE4_KDEUI_LIBS})
 
 if (Q_WS_X11)
