diff --git a/kdeui/notifications/knotification.cpp b/kdeui/notifications/knotification.cpp
index d497c69..614a188 100644
--- a/kdeui/notifications/knotification.cpp
+++ b/kdeui/notifications/knotification.cpp
@@ -52,6 +52,10 @@
 #include <QDateTime>
 #include <QDBusError>
 
+#if defined(Q_WS_MAC)
+#include <Carbon/Carbon.h>
+#endif
+
 struct KNotification::Private
 {
     QString eventId;
@@ -69,8 +73,18 @@ struct KNotification::Private
 
     QTimer updateTimer;
     bool needUpdate;
-
+#if defined(Q_WS_MAC)
+    ProcessSerialNumber previouslyActivePSN;
+    bool hasPrevPSN;
+    Private()
+        : id(0), ref(1), widget(0l), needUpdate(false), hasPrevPSN(false)
+    {
+        memset(&previouslyActivePSN, 0, sizeof(ProcessSerialNumber));
+    }
+#else
     Private() : id(0), ref(1), widget(0l), needUpdate(false) {}
+#endif
+
     /**
      * recursive function that raise the widget. @p w
      *
@@ -109,6 +123,11 @@ KNotification::~KNotification()
 {
    if(d ->id > 0)
        KNotificationManager::self()->close( d->id );
+#ifdef Q_WS_MAC
+    if (d->hasPrevPSN) {
+        SetFrontProcess(&d->previouslyActivePSN);
+    }
+#endif
    delete d;
 }
 
@@ -260,6 +279,9 @@ void KNotification::raiseWidget()
         return;
     }
 
+#ifdef Q_WS_MAC
+    d->hasPrevPSN = (GetFrontProcess(&d->previouslyActivePSN) == noErr);
+#endif
     Private::raiseWidget( d->widget );
 }
 
@@ -271,7 +293,8 @@ void KNotification::Private::raiseWidget(QWidget *w)
    {
        w->raise();
 #if defined(Q_WS_MAC)
-       w->activateWindow();
+        KWindowSystem::activateWindow( w->winId() );
+        w->activateWindow();
 #else
        KWindowSystem::activateWindow( w->winId() );
 #endif
@@ -371,7 +394,10 @@ void KNotification::sendEvent()
 
        if (KNotificationManager::self()->notify( this , d->pixmap , d->actions , d->contexts , appname ))
            d->id = -1;
-   }
+#ifdef Q_WS_MAC
+        d->hasPrevPSN = (GetFrontProcess(&d->previouslyActivePSN) == noErr);
+#endif
+    }
    else if(d->id > 0)
    {
        KNotificationManager::self()->reemit(this , d->id );
