diff --git a/kinit/kioslave.cpp b/kinit/kioslave.cpp
index 7062564..97bb6bc 100644
--- a/kinit/kioslave.cpp
+++ b/kinit/kioslave.cpp
@@ -31,6 +31,10 @@
 #include <QtCore/QLibrary>
 #include <QtCore/QFile>
 
+#ifdef Q_OS_MAC
+#include <CoreFoundation/CoreFoundation.h>
+#endif
+
 #if defined(Q_WS_WIN) || defined(Q_WS_MAC)
 #define USE_KPROCESS_FOR_KIOSLAVES
 #endif
@@ -62,6 +66,20 @@ int main(int argc, char **argv)
 #ifndef _WIN32_WCE
      setlocale(LC_ALL, "");
 #endif
+#ifdef Q_OS_MAC
+    CFBundleRef mainBundle = CFBundleGetMainBundle();
+    if (mainBundle) {
+        // get the application's Info Dictionary. For app bundles this would live in the bundle's Info.plist,
+        // for regular executables it is obtained in another way.
+        CFMutableDictionaryRef infoDict = (CFMutableDictionaryRef) CFBundleGetInfoDictionary(mainBundle);
+        if (infoDict) {
+            // Add or set the "LSUIElement" key with/to value "1". This can simply be a CFString.
+            CFDictionarySetValue(infoDict, CFSTR("LSUIElement"), CFSTR("1"));
+            // That's it. We're now considered as an "agent" by the window server, and thus will have
+            // neither menubar nor presence in the Dock or App Switcher.
+        }
+    }
+#endif
      QString libpath = QFile::decodeName(argv[1]);
 
      if (libpath.isEmpty())
