diff --git a/kdeui/kernel/kapplication.h b/kdeui/kernel/kapplication.h
index fa2ab26..5b5ba71 100644
--- a/kdeui/kernel/kapplication.h
+++ b/kdeui/kernel/kapplication.h
@@ -39,7 +39,10 @@ class KConfig;
 
 typedef unsigned long Atom;
 #if !defined(Q_WS_X11)
-typedef void Display;
+// typedef void Display;
+// use the definition from the XOrg X server, so a redefinition
+// will be accepted under C++11 at least
+typedef struct _XDisplay Display;
 #endif
 
 #include <QtGui/QApplication>
