diff --git kdeui/kernel/kglobalsettings.cpp kdeui/kernel/kglobalsettings.cpp
index 2e3a7eb..5e9196b 100644
--- kdeui/kernel/kglobalsettings.cpp
+++ kdeui/kernel/kglobalsettings.cpp
@@ -150,7 +150,11 @@ class KGlobalSettings::Private
         Private(KGlobalSettings *q)
             : q(q), activated(false), paletteCreated(false)
         {
+#ifdef Q_OS_MAC
+            kdeFullSession = true;
+#else
             kdeFullSession = !qgetenv("KDE_FULL_SESSION").isEmpty();
+#endif
         }
 
         QPalette createApplicationPalette(const KSharedConfigPtr &config);
