diff --git a/kdeui/widgets/kmainwindow.cpp b/kdeui/widgets/kmainwindow.cpp
index 85beacc..46dfd8e 100644
--- a/kdeui/widgets/kmainwindow.cpp
+++ b/kdeui/widgets/kmainwindow.cpp
@@ -854,7 +854,49 @@ void KMainWindow::saveWindowSize( const KConfigGroup & _cg ) const
     KConfigGroup cg(_cg);
     cg.writeEntry( geometryKey, geometry.toBase64() );
 }
-#else
+
+#elif defined(Q_OS_MAC)
+
+void KMainWindow::restoreWindowSize( const KConfigGroup & _cg )
+{
+
+    QString geometryKey = QString::fromLatin1("geometry");
+    QByteArray geometry = _cg.readEntry( geometryKey, QByteArray() );
+    kDebug(200) << "restoreWindowSize():" << geometryKey << "=" << geometry;
+    if (!restoreGeometry( QByteArray::fromBase64(geometry) )) {
+        // try first if there data has been stored using the X11-like approach from earlier kdelibs versions.
+        // This code is copied from restoreWindowSize() below.
+        int scnum = QApplication::desktop()->screenNumber(window());
+        QRect desk = QApplication::desktop()->screenGeometry(scnum);
+        const QSize size( _cg.readEntry( QString::fromLatin1("Width %1").arg(desk.width()), 0 ),
+                    _cg.readEntry( QString::fromLatin1("Height %1").arg(desk.height()), 0 ) );
+        if ( !size.isEmpty() ) {
+            kDebug(200) << "restoreWindowSize(): falling back to legacy saved state" << size;
+            if (size.width() > desk.width() || size.height() > desk.height()) {
+                setWindowState( Qt::WindowMaximized );
+            }
+            else {
+                resize( size );
+            }
+        }
+        else {
+            // if first time run, centre window
+            kDebug(200) << "restoreWindowSize(): moving the window to the screen centre of" << desk;
+            move( (desk.width()-width())/2, (desk.height()-height())/2 );
+        }
+    }
+}
+
+void KMainWindow::saveWindowSize( const KConfigGroup & _cg ) const
+{
+    QString geometryKey = QString::fromLatin1("geometry");
+    QByteArray geometry = saveGeometry();
+    kDebug(200) << "saveWindowSize():" << geometryKey << "=" << geometry;
+    KConfigGroup cg(_cg);
+    cg.writeEntry( geometryKey, geometry.toBase64() );
+}
+
+#else // Q_WS_X11 and select others.
 void KMainWindow::saveWindowSize( const KConfigGroup & _cg ) const
 {
     K_D(const KMainWindow);
@@ -1020,7 +1062,7 @@ bool KMainWindow::event( QEvent* ev )
 {
     K_D(KMainWindow);
     switch( ev->type() ) {
-#ifdef Q_WS_WIN
+#if defined(Q_WS_WIN) || defined(Q_OS_MAC)
     case QEvent::Move:
 #endif
     case QEvent::Resize:
