diff --git a/data/data-en_US.sql b/data/data-en_US.sql
index 04cc8c5745fc790be14e11c65f850d7e8269709a..86b310b498dfa8ce991828231ea27806e5b9cdc7 100644
--- a/data/data-en_US.sql
+++ b/data/data-en_US.sql
@@ -33,6 +33,8 @@ INSERT INTO units VALUES (22,'ounce','oz','ounces','oz',1);
 INSERT INTO units VALUES (23,'square','sq','squares','sq',0);
 INSERT INTO units VALUES (24,'',NULL,'',NULL,0);
 INSERT INTO units VALUES (25,'milligram','mg','milligrams','mg',1);
+INSERT INTO units VALUES (26,'Celsius','C',NULL,'C',0);
+INSERT INTO units VALUES (27,'Fahrenheit','F',NULL,'F',0);
 INSERT INTO rating_criteria VALUES (1,'Overall');
 INSERT INTO rating_criteria VALUES (2,'Taste');
 INSERT INTO rating_criteria VALUES (3,'Appearance');
diff --git a/data/unit-data-en_US.txt b/data/unit-data-en_US.txt
index 119212ad4706a7e3babc425b56ac11bcbba495f8..1acf76880e3c6914e34e7a37da1802a34d570a82 100644
--- a/data/unit-data-en_US.txt
+++ b/data/unit-data-en_US.txt
@@ -40,4 +40,6 @@ tbsp:tbsp::
 teaspoon:teaspoons::
 tsp:tsp::
 tube:tubes::
-unit:units::
\ No newline at end of file
+unit:units::
+Celsius:::
+Fahrenheit:::
diff --git a/src/backends/recipedb.cpp b/src/backends/recipedb.cpp
index 5afd91db96689006a0c1c6c2efc366a7e930138a..66358407525540f520cf5303fa0c5cf9c4ff60cd 100644
--- a/src/backends/recipedb.cpp
+++ b/src/backends/recipedb.cpp
@@ -192,6 +192,20 @@ RecipeDB::ConversionStatus RecipeDB::convertIngredientUnits( const Ingredient &f
 	if ( from.units.id() == to.id() )
 		return Success;
 
+    qWarning() << to.name().compare("celsius") << from.units.name().compare("fahrenheit")
+        << to.name().compare("fahrenheit") << from.units.name().compare("celsius");
+	if ( from.units.type() == to.type() && to.type() == Unit::Other 
+        && ((to.name().compare("celsius", Qt::CaseInsensitive) == 0 && from.units.name().compare("fahrenheit", Qt::CaseInsensitive) == 0)
+            || (to.name().compare("fahrenheit", Qt::CaseInsensitive) == 0 && from.units.name().compare("celsius", Qt::CaseInsensitive) == 0)) ){
+        if (to.name().compare("celsius", Qt::CaseInsensitive) == 0) {
+            result.amount = (from.amount - 32.0) * 5.0 / 9.0;
+        } else {
+            result.amount = from.amount * 9.0 / 5.0 + 32.0;
+        }
+        result.units = to;
+        return Success;
+    }
+
 	if ( from.units.type() == to.type() && to.type() != Unit::Other ) {
 		double ratio = unitRatio( from.units.id(), to.id() );
 		if ( ratio > 0 ) {

