From: Christer Stenbrenden <christer@uggwar.net>
Date: Tue, 25 Feb 2014 22:14:13 +0100
Subject: Don't crash if pa_stream_peek returns NULL

---
 src/mainwindow.cc | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/src/mainwindow.cc
+++ b/src/mainwindow.cc
@@ -488,6 +488,11 @@
         return;
     }
 
+    if (!data) {
+        pa_stream_drop(s);
+        return;
+    }
+
     assert(length > 0);
     assert(length % sizeof(float) == 0);
 
