diff --git a/src/devicewidget.cc b/src/devicewidget.cc
index 1c34ea6..165a4b8 100644
--- a/src/devicewidget.cc
+++ b/src/devicewidget.cc
@@ -63,8 +63,10 @@ DeviceWidget::DeviceWidget(BaseObjectType* cobject, const Glib::RefPtr<Gtk::Buil
     for (unsigned i = 0; i < PA_CHANNELS_MAX; i++)
         channelWidgets[i] = NULL;
 
+#ifdef HAVE_GTK3
     offsetAdjustment = Gtk::Adjustment::create(0.0, -2000.0, 2000.0, 10.0, 50.0, 0.0);
     offsetButton->configure(offsetAdjustment, 0, 2);
+#endif
 }
 
 void DeviceWidget::init(MainWindow* mainWindow, Glib::ustring deviceType) {
