--- src/orig.main.cpp	2015-02-22 19:22:19.000000000 +0100
+++ src/main.cpp	2016-12-15 23:25:21.000000000 +0100
@@ -31,7 +31,7 @@
 	#include <kaboutdata.h>
 #endif
 
-#if defined (Q_WS_MAC)
+#if defined (Q_OS_OSX)
         #include "kchmviewerapp.h"
 #else
         typedef QApplication  KchmviewerApp;
@@ -70,7 +70,7 @@
 	// Configuration
 	pConfig = new Config();
 
-#if !defined (WIN32) && !defined(Q_WS_MAC)
+#if !defined (WIN32)
 	if ( QDBusConnection::sessionBus().isConnected() )
 	{
 		if ( QDBusConnection::sessionBus().registerService(SERVICE_NAME) )
