diff --git a/src/kdesvn_part.cpp b/src/kdesvn_part.cpp
index a84fc53..139bb44 100644
--- a/src/kdesvn_part.cpp
+++ b/src/kdesvn_part.cpp
@@ -292,7 +292,7 @@ void kdesvnpart::showAboutApplication()
         about.addAuthor(QLatin1String("Rajko Albrecht"), i18n("Original author and maintainer"), QLatin1String("ral@alwins-world.de"));
         about.addAuthor(QLatin1String("Christian Ehrlicher"), i18n("Developer"), QLatin1String("ch.ehrlicher@gmx.de"));
         about.setHomepage("https://projects.kde.org/kdesvn");
-        qApp->setWindowIcon(QIcon::fromTheme(QLatin1String("kdesvn")));
+        qApp->setWindowIcon(QIcon::fromTheme(QLatin1String("kdesvn"), qApp->windowIcon()));
         m_aboutDlg = new KAboutApplicationDialog(about);
     }
     if (m_aboutDlg == 0) {
diff --git a/src/main.cpp b/src/main.cpp
index 8577e9c..739f252 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -38,7 +38,7 @@ int main(int argc, char **argv)
     app.setApplicationDisplayName("kdesvn");
     app.setOrganizationDomain("kde.org");
     app.setApplicationVersion(KDESVN_VERSION);
-    app.setWindowIcon(QIcon::fromTheme(QLatin1String("kdesvn")));
+    app.setWindowIcon(QIcon::fromTheme(QLatin1String("kdesvn"), app.windowIcon()));
 
     KAboutData aboutData(QLatin1String("kdesvn"), i18n("kdesvn"), QString(KDESVN_VERSION), i18n(description),
                          KAboutLicense::GPL, i18n("(C) 2005-2009 Rajko Albrecht"));
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 008fe95..b33ee37 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -18,7 +18,11 @@
  #   51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.         #
  ####
 
+include(ECMMarkNonGuiExecutable)
+include(ECMAddAppIcon)
+
 add_executable(kdesvnaskpass askpass/kdesvn-askpass.cpp)
+ecm_mark_nongui_executable(kdesvnaskpass)
 
 ki18n_wrap_ui(kdesvn_ui
     urldlg.ui
@@ -30,6 +34,10 @@ set(kdesvnsrc
     urldlg.cpp
 )
 
+# Sets the icon on Windows and OSX
+file(GLOB ICONS_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/icons/*-apps-kdesvn.png")
+ecm_add_app_icon(kdesvnsrc ICONS ${ICONS_SRCS})
+
 set(partsrc
     commandline_part.cpp
     kdesvn_part.cpp
