diff --git a/src/svnqt/cmake/FindSubversion.cmake b/src/svnqt/cmake/FindSubversion.cmake
index 879c301..0415668 100644
--- a/src/svnqt/cmake/FindSubversion.cmake
+++ b/src/svnqt/cmake/FindSubversion.cmake
@@ -37,7 +37,7 @@ if(UNIX)
     find_library(${targetvar} ${libname}
         PATHS
         /usr/lib
-        /usr/local/lib
+        @PREFIX@/lib
         )
   endmacro()
 
@@ -50,7 +50,7 @@ if(UNIX)
   endif(SUBVERSION_INSTALL_PATH)
   find_path(SUBVERSION_INCLUDE_DIR svn_client.h
         /usr/include/subversion-1
-        /usr/local/include/subversion-1)
+        @PREFIX@/include/subversion-1)
 
   find_sub_lib(SUBVERSION_CLIENTLIB svn_client-1)
   find_sub_lib(SUBVERSION_REPOSITORYLIB svn_repos-1)
@@ -62,12 +62,12 @@ if(UNIX)
 
   find_program(APR_CONFIG NAMES apr-config apr-1-config
     PATHS
-    /usr/local/apr/bin
+    @PREFIX@/bin
   )
 
   find_program(APU_CONFIG NAMES apu-config apu-1-config
     PATHS
-    /usr/local/apr/bin
+    @PREFIX@/bin
   )
 
   if(NOT APR_CONFIG)
@@ -100,11 +100,11 @@ if(UNIX)
   execute_process(COMMAND ${APU_CONFIG} "--link-ld" "--libs" OUTPUT_VARIABLE APU_LIBS OUTPUT_STRIP_TRAILING_WHITESPACE)
   string(REGEX REPLACE "^ +" "" APU_LIBS "${APU_LIBS}")
   message(STATUS "Found apu libs: ${APU_LIBS}")
-  set(SUBVERSION_ALL_LIBS ${SUBVERSION_ALL_LIBS} ${APU_LIBS})
 
   execute_process(COMMAND ${APU_CONFIG} "--ldflags" OUTPUT_VARIABLE APU_EXTRA_LDFLAGS OUTPUT_STRIP_TRAILING_WHITESPACE)
   string(REGEX REPLACE "^ +" "" APU_EXTRA_LDFLAGS "${APU_EXTRA_LDFLAGS}")
   message(STATUS "Found apu extra ldflags: ${APU_EXTRA_LDFLAGS}")
+  set(SUBVERSION_ALL_LIBS ${SUBVERSION_ALL_LIBS} ${APU_EXTRA_LDFLAGS} ${APU_LIBS})
 
   check_include_files(execinfo.h HAS_BACKTRACE_H)
   if(HAS_BACKTRACE_H)
