diff --git a/contrib/amd64/crc32-pclmul_asm.S b/contrib/amd64/crc32-pclmul_asm.S
index ed23699..74d98fc 100644
--- a/contrib/amd64/crc32-pclmul_asm.S
+++ b/contrib/amd64/crc32-pclmul_asm.S
@@ -44,6 +44,8 @@
  *   - prepend '$' to some immediate operands to make assembler happy.
  */
 
+#if !defined(__APPLE__)
+
 #define ENTRY(name) \
 .globl name; \
 .hidden name; \
@@ -53,6 +55,17 @@ name:
 #define ENDPROC(name) \
 .size name, .-name
 
+#else // __APPLE__
+
+#define ENTRY(name) \
+.globl _ ## name; \
+.private_extern _ ## name; \
+_ ## name:
+
+#define ENDPROC(name) /**/
+
+#endif
+
 .align 16
 /*
  * [x4*128+32 mod P(x) << 32)]'  << 1   = 0x154442bd4
