diff -r b193168bad98 src/xxdiff.pro
--- src/xxdiff.pro	Wed Apr 15 14:02:15 2015 -0400
+++ src/xxdiff.pro	Thu May 14 20:19:45 2015 +0200
@@ -29,8 +29,9 @@
 TEMPLATE = app
 CONFIG -= debug
 CONFIG += qt warn_on thread
+QT += widgets help
 
-DESTDIR=../bin
+DESTDIR=./bin
 TARGET = xxdiff
 
 
@@ -140,7 +141,8 @@
 
    QMAKE_EXTRA_TARGETS += macdeployqt dmg deploy
    QMAKE_CXXFLAGS -= -O2
-   QMAKE_CXXFLAGS += -mdynamic-no-pic -O3 -ftracer -msse2 -msse3 -mssse3 -ftree-vectorize
+   QMAKE_CXXFLAGS += -mdynamic-no-pic -O3 -g
+   INCLUDEPATH += -I@@QT_INCLUDES_DIR@@
 }
 
 #----------------------------------------
@@ -186,6 +188,7 @@
 	main.h \
 	overview.h \
 	resParser.h \
+	resParser_lex.h \
 	resources.h \
 	resources.inline.h \
 	accelUtil.h \
