diff -r b193168bad98 src/cmdline.cpp
--- src/cmdline.cpp	Wed Apr 15 14:02:15 2015 -0400
+++ src/cmdline.cpp	Thu May 14 22:37:18 2015 +0200
@@ -289,9 +289,13 @@
      "Sets the X display (default is $DISPLAY)."
    }, 
    { "style", 0, true, 's',
-     "Sets the application GUI style. Possible values are motif, windows, "
-     "and platinum. "
-   }, 
+     "Sets the application GUI style. Possible values are fusion, windows, "
+     "or QtCurve when that style has been installed. "
+   },
+   { "platform", 0, true, 'x',
+      "Sets the platform plugin to be used. For instance, on OS X one can use "
+      "cocoa or xcb as platform plugins."
+   },
    { "geometry", 0, true, 'g',
      "Sets the client geometry of the main widget."
    }, 
@@ -339,6 +343,7 @@
 //------------------------------------------------------------------------------
 //
 XxCmdline::XxCmdline() :
+   _forcePlatform( false ),
    _forceStyle( false ),
    _forceGeometry( false ),
    _forceFont( false ),
@@ -642,7 +647,8 @@
          } break;
 
          case 'd':
-         case 's':
+         case 's':  // --style
+         case 'x':  // --platform
          case 'G':
          case 'g':
          case 'F':
@@ -771,6 +777,9 @@
       if ( strncmp( _qtOptions[ ii ], "-style", 6 ) == 0 ) {
          _forceStyle = true;
       }
+      else if ( strncmp( _qtOptions[ ii ], "-platform", 9 ) == 0 ) {
+         _forcePlatform = true;
+      }
       else if ( strncmp( _qtOptions[ ii ], "-geometry", 9 ) == 0 ) {
          _forceGeometry = true;
       }
diff -r b193168bad98 src/cmdline.h
--- src/cmdline.h	Wed Apr 15 14:02:15 2015 -0400
+++ src/cmdline.h	Thu May 14 22:37:18 2015 +0200
@@ -117,6 +117,7 @@
    /*----- data members -----*/
 
    // Cmdline-related variables.
+   bool          _forcePlatform;
    bool          _forceStyle;
    bool          _forceGeometry;
    bool          _forceFont;
