diff -r b193168bad98 src/xxdiff.pro
--- src/xxdiff.pro	Wed Apr 15 14:02:15 2015 -0400
+++ src/xxdiff.pro	Tue Jan 26 11:02:25 2016 +0100
@@ -29,8 +29,9 @@
 TEMPLATE = app
 CONFIG -= debug
 CONFIG += qt warn_on thread
+QT += widgets help
 
-DESTDIR=../bin
+DESTDIR=./bin
 TARGET = xxdiff
 
 
@@ -110,6 +111,11 @@
 ## macx-g++:QMAKE_CXXFLAGS += -D__GNU_LIBRARY__
 ## macx-g++:QMAKE_CXXFLAGS -= -fno-exceptions
 
+QMAKE_CFLAGS -= -O2
+QMAKE_CXXFLAGS -= -O2
+QMAKE_CFLAGS += @@CFLAGS@@
+QMAKE_CXXFLAGS += @@CXXFLAGS@@
+QMAKE_LFLAGS += @@LFLAGS@@
 
 #----------------------------------------
 # Max OS X (macx-g++ for command line build)
@@ -139,8 +145,8 @@
    deploy.depends = $$dmg.target
 
    QMAKE_EXTRA_TARGETS += macdeployqt dmg deploy
-   QMAKE_CXXFLAGS -= -O2
-   QMAKE_CXXFLAGS += -mdynamic-no-pic -O3 -ftracer -msse2 -msse3 -mssse3 -ftree-vectorize
+   QMAKE_CXXFLAGS += -mdynamic-no-pic
+   INCLUDEPATH += -I@@QT_INCLUDES_DIR@@
 }
 
 #----------------------------------------
@@ -186,6 +192,7 @@
 	main.h \
 	overview.h \
 	resParser.h \
+	resParser_lex.h \
 	resources.h \
 	resources.inline.h \
 	accelUtil.h \
