diff --git bootstrap bootstrap
index 01fcfae8dc..223315f0fc 100755
--- bootstrap
+++ bootstrap
@@ -2047,20 +2047,29 @@ fi
 
 echo "---------------------------------------------"
 
-# Run make to build bootstrap cmake
-if test "${cmake_bootstrap_generator}" = "Ninja"; then
-  ninja_v=-v
-else
-  ninja_v=
-fi
-if test "x${cmake_parallel_make}" != "x"; then
-  ${cmake_make_processor} ${cmake_make_flags} ${ninja_v}
+if [ -x ${cmake_prefix_dir}/bin/cmake ] ;then
+  # use the existing cmake that we'll be replacing.
+  ln -s ${cmake_prefix_dir}/bin/cmake ${cmake_bootstrap_dir}/cmake
+  if test -n "${cmake_verbose}"; then
+    echo "Bootstrap cmake: ${cmake_prefix_dir}/bin/cmake"
+    echo "---------------------------------------------"
+  fi
 else
-  ${cmake_make_processor} ${ninja_v}
-fi
-RES=$?
-if test "${RES}" -ne "0"; then
-  cmake_error 9 "Problem while running ${cmake_make_processor}"
+    # Run make to build bootstrap cmake
+    if test "${cmake_bootstrap_generator}" = "Ninja"; then
+      ninja_v=-v
+    else
+      ninja_v=
+    fi
+    if test "x${cmake_parallel_make}" != "x"; then
+      ${cmake_make_processor} ${cmake_make_flags} ${ninja_v}
+    else
+      ${cmake_make_processor} ${ninja_v}
+    fi
+    RES=$?
+    if test "${RES}" -ne "0"; then
+      cmake_error 9 "Problem while running ${cmake_make_processor}"
+    fi
 fi
 cd "${cmake_binary_dir}"
 
@@ -2077,7 +2086,7 @@ export CXXFLAGS
 export LDFLAGS
 
 # Run bootstrap CMake to configure real CMake
-cmake_options="-DCMAKE_BOOTSTRAP=1"
+cmake_options="-DCMAKE_BOOTSTRAP=1 -DCMAKE_EXPORT_COMPILE_COMMANDS=ON"
 if test -n "${cmake_verbose}"; then
   cmake_options="${cmake_options} -DCMAKE_VERBOSE_MAKEFILE=1"
 fi
