diff --git Help/prop_tgt/MACOSX_BUNDLE_INFO_PLIST.rst Help/prop_tgt/MACOSX_BUNDLE_INFO_PLIST.rst
index 8515acc..342fc5f 100644
--- Help/prop_tgt/MACOSX_BUNDLE_INFO_PLIST.rst
+++ Help/prop_tgt/MACOSX_BUNDLE_INFO_PLIST.rst
@@ -28,6 +28,8 @@ configured into the file:
   Sets ``CFBundleLongVersionString``.
 ``MACOSX_BUNDLE_SHORT_VERSION_STRING``
   Sets ``CFBundleShortVersionString``.
+``MACOSX_BUNDLE_LSUILEMENT
+  Sets ``LSUIElement``.
 
 CMake variables of the same name may be set to affect all targets in a
 directory that do not have each specific property set.  If a custom
diff --git Modules/MacOSXBundleInfo.plist.in Modules/MacOSXBundleInfo.plist.in
index e06b17e..57c004b 100644
--- Modules/MacOSXBundleInfo.plist.in
+++ Modules/MacOSXBundleInfo.plist.in
@@ -2,6 +2,8 @@
 <!DOCTYPE plist PUBLIC "-//Apple Computer//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
 <plist version="1.0">
 <dict>
+	<key>NSPrincipalClass</key>
+	<string>NSApplication</string>
 	<key>CFBundleDevelopmentRegion</key>
 	<string>English</string>
 	<key>CFBundleExecutable</key>
@@ -28,6 +30,8 @@
 	<string>${MACOSX_BUNDLE_BUNDLE_VERSION}</string>
 	<key>CSResourcesFileMapped</key>
 	<true/>
+	<key>LSUIElement</key>
+	<${MACOSX_BUNDLE_LSUIELEMENT}/>
 	<key>NSHumanReadableCopyright</key>
 	<string>${MACOSX_BUNDLE_COPYRIGHT}</string>
 </dict>
diff --git Source/cmLocalGenerator.cxx Source/cmLocalGenerator.cxx
index 2c5db10..7c74fe8 100644
--- Source/cmLocalGenerator.cxx
+++ Source/cmLocalGenerator.cxx
@@ -2601,6 +2601,7 @@ void cmLocalGenerator::GenerateAppleInfoPList(cmGeneratorTarget* target,
   cmMakefile* mf = this->Makefile;
   cmMakefile::ScopePushPop varScope(mf);
   mf->AddDefinition("MACOSX_BUNDLE_EXECUTABLE_NAME", targetName.c_str());
+  mf->AddDefinition("MACOSX_BUNDLE_LSUIELEMENT", "false");
   cmLGInfoProp(mf, target, "MACOSX_BUNDLE_INFO_STRING");
   cmLGInfoProp(mf, target, "MACOSX_BUNDLE_ICON_FILE");
   cmLGInfoProp(mf, target, "MACOSX_BUNDLE_GUI_IDENTIFIER");
@@ -2609,6 +2610,7 @@ void cmLocalGenerator::GenerateAppleInfoPList(cmGeneratorTarget* target,
   cmLGInfoProp(mf, target, "MACOSX_BUNDLE_SHORT_VERSION_STRING");
   cmLGInfoProp(mf, target, "MACOSX_BUNDLE_BUNDLE_VERSION");
   cmLGInfoProp(mf, target, "MACOSX_BUNDLE_COPYRIGHT");
+  cmLGInfoProp(mf, target, "MACOSX_BUNDLE_LSUIELEMENT");
   mf->ConfigureFile(inFile.c_str(), fname, false, false, false);
 }
 
