--- Modules/FindQt4.cmake.orig	2018-10-09 12:15:21.719598141 +0200
+++ Modules/FindQt4.cmake	2018-10-09 12:19:26.799598141 +0200
@@ -729,7 +729,8 @@
       find_path(QT_IMPORTS_DIR NAMES Qt
         HINTS ${qt_imports_dir}
         PATH_SUFFIXES imports lib/qt4/imports
-        DOC "The location of the Qt imports")
+        DOC "The location of the Qt imports"
+        NO_DEFAULT_PATH)
     else()
       set(QT_IMPORTS_DIR ${qt_imports_dir} CACHE PATH "The location of the Qt imports")
     endif()
