diff --git Utilities/Sphinx/apply_qthelp_css_workaround.cmake Utilities/Sphinx/apply_qthelp_css_workaround.cmake
index 8b74d12..c84afc1 100644
--- Utilities/Sphinx/apply_qthelp_css_workaround.cmake
+++ Utilities/Sphinx/apply_qthelp_css_workaround.cmake
@@ -1,7 +1,7 @@
 
 file(READ "${CSS_DIR}/basic.css" BasicCssContent)
 
-file(READ "${CSS_DIR}/default.css" DefaultCssContent)
+file(READ "${CSS_DIR}/classic.css" DefaultCssContent)
 string(REPLACE
   "@import url(\"basic.css\")" "${BasicCssContent}"
   DefaultCssContent "${DefaultCssContent}"
@@ -9,7 +9,7 @@ string(REPLACE
 
 file(READ "${CSS_DIR}/cmake.css" CMakeCssContent)
 string(REPLACE
-  "@import url(\"default.css\")" "${DefaultCssContent}"
+  "@import url(\"classic.css\")" "${DefaultCssContent}"
   CMakeCssContent "${CMakeCssContent}"
 )
 file(WRITE "${CSS_DIR}/cmake.css" "${CMakeCssContent}")
diff --git Utilities/Sphinx/conf.py.in Utilities/Sphinx/conf.py.in
index eb24a6e..046c720 100644
--- Utilities/Sphinx/conf.py.in
+++ Utilities/Sphinx/conf.py.in
@@ -59,7 +59,7 @@ man_show_urls = False
 html_show_sourcelink = True
 html_static_path = ['@conf_path@/static']
 html_style = 'cmake.css'
-html_theme = 'default'
+html_theme = 'classic'
 html_title = 'CMake %s Documentation' % release
 html_short_title = '%s Documentation' % release
 html_favicon = '@conf_path@/static/cmake-favicon.ico'
diff --git Utilities/Sphinx/static/cmake.css Utilities/Sphinx/static/cmake.css
index 2a326d4..0d95b29 100644
--- Utilities/Sphinx/static/cmake.css
+++ Utilities/Sphinx/static/cmake.css
@@ -1,5 +1,5 @@
 /* Import the Sphinx theme style.  */
-@import url("default.css");
+@import url("classic.css");
 
 /* Wrap sidebar content even within words so that long
    document names do not escape sidebar borders.  */
