diff --git bootstrap bootstrap
index 1f8eaa50394a6f32a0121ef911ca8b7e6f2f2f35..69687faa7b26d9e9ea8c8d623cd28e23ef0224d6 100755
--- bootstrap
+++ bootstrap
@@ -1644,15 +1644,20 @@ fi
 
 echo "---------------------------------------------"
 
-# Run make to build bootstrap cmake
-if [ "x${cmake_parallel_make}" != "x" ]; then
-  ${cmake_make_processor} ${cmake_make_flags}
+if [ -x ${cmake_prefix_dir}/bin/cmake ] ;then
+  # use the existing cmake that we'll be replacing.
+  ln -s ${cmake_prefix_dir}/bin/cmake ${cmake_bootstrap_dir}/cmake
 else
-  ${cmake_make_processor}
-fi
-RES=$?
-if [ "${RES}" -ne "0" ]; then
-  cmake_error 9 "Problem while running ${cmake_make_processor}"
+  # Run make to build bootstrap cmake
+  if [ "x${cmake_parallel_make}" != "x" ]; then
+    ${cmake_make_processor} ${cmake_make_flags}
+  else
+    ${cmake_make_processor}
+  fi
+  RES=$?
+  if [ "${RES}" -ne "0" ]; then
+    cmake_error 9 "Problem while running ${cmake_make_processor}"
+  fi
 fi
 cd "${cmake_binary_dir}"
 
