--- CMakeLists.txt.orig	2015-09-26 22:19:05.118186833 +0200
+++ CMakeLists.txt	2015-09-26 22:19:21.209186414 +0200
@@ -488,7 +488,7 @@
 
   # The CMake executables usually do not need any rpath to run in the build or
   # install tree.
-  set(CMAKE_SKIP_RPATH ON CACHE INTERNAL "CMake does not need RPATHs.")
+  set(CMAKE_SKIP_RPATH OFF CACHE INTERNAL "CMake does not need RPATHs.")
 
   # Load install destinations.
   include(Source/CMakeInstallDestinations.cmake)
