--- Modules/Compiler/orig.Clang-CXX.cmake	2016-11-30 16:14:32.000000000 +0100
+++ Modules/Compiler/Clang-CXX.cmake	2017-02-20 19:57:02.000000000 +0100
@@ -6,9 +6,9 @@
 endif()
 
 cmake_policy(GET CMP0025 appleClangPolicy)
-if(APPLE AND NOT appleClangPolicy STREQUAL NEW)
-  return()
-endif()
+#if(APPLE AND NOT appleClangPolicy STREQUAL NEW)
+#  return()
+#endif()
 
 if(NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS 2.1)
   set(CMAKE_CXX98_STANDARD_COMPILE_OPTION "-std=c++98")
--- Modules/Compiler/Clang-C.cmake.orig	2017-12-09 11:50:21.922111342 +0100
+++ Modules/Compiler/Clang-C.cmake	2017-12-09 11:52:49.634884876 +0100
@@ -2,9 +2,9 @@
 __compiler_clang(C)
 
 cmake_policy(GET CMP0025 appleClangPolicy)
-if(APPLE AND NOT appleClangPolicy STREQUAL NEW)
-  return()
-endif()
+# if(APPLE AND NOT appleClangPolicy STREQUAL NEW)
+#   return()
+# endif()
 
 if(NOT CMAKE_C_COMPILER_VERSION VERSION_LESS 3.4)
   if(NOT "x${CMAKE_C_SIMULATE_ID}" STREQUAL "xMSVC")
