diff --git src/widgets/ktextedit.cpp src/widgets/ktextedit.cpp
index d0860b3..3755fe6 100644
--- src/widgets/ktextedit.cpp
+++ src/widgets/ktextedit.cpp
@@ -80,7 +80,7 @@ public:
           lastReplacedPosition(-1)
     {
         //Check the default sonnet settings to see if spellchecking should be enabled.
-        QSettings settings(QStringLiteral("KDE"), QStringLiteral("Sonnet"));
+        QSettings settings(QSettings::IniFormat, QSettings::UserScope, QStringLiteral("KDE"), QStringLiteral("Sonnet"));
         spellCheckingEnabled = settings.value(QStringLiteral("checkerEnabledByDefault"), false).toBool();
 #ifdef HAVE_SPEECH
         textToSpeech = new QTextToSpeech(parent);
