diff --git a/Makefile.in b/Makefile.in
index 1efe11f6f27b51fcf984da10c5aab9c3156d876d..097a323ad962385a4e075ae56a81178b0688f99d 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -145,7 +148,7 @@ SSHKEYGEN_OBJS=	ssh-keygen.o sshsig.o $(SKOBJS)
 
 SSHKEYSIGN_OBJS=ssh-keysign.o readconf.o uidswap.o $(SKOBJS)
 
-P11HELPER_OBJS=	ssh-pkcs11-helper.o ssh-pkcs11.o $(SKOBJS)
+P11HELPER_OBJS=	ssh-pkcs11-helper.o $(SKOBJS)
 
 SKHELPER_OBJS=	ssh-sk-helper.o ssh-sk.o sk-usbhid.o
 
