--- samba-4.8.9/release-scripts/orig.build-docs	2018-01-14 21:41:58.000000000 +0100
+++ samba-4.8.9/release-scripts/build-docs	2019-07-27 01:15:47.000000000 +0200
@@ -11,7 +11,7 @@
 pushd $DOCSRCDIR || exit 1
 
 git clean -d -x -f
-export XML_CATALOG_FILES="file:///etc/xml/catalog file://$(pwd)/build/catalog.xml"
+export XML_CATALOG_FILES="file://@PREFIX@/etc/xml/catalog file://$(pwd)/build/catalog.xml"
 autoconf && \
 	./configure --with-papersize=letter && \
 	make smbdotconf/parameters.all.xml && \
--- samba-4.8.9/release-scripts/orig.build-htmlman-git	2018-01-14 21:41:58.000000000 +0100
+++ samba-4.8.9/release-scripts/build-htmlman-git	2019-07-27 01:16:12.000000000 +0200
@@ -34,7 +34,7 @@
 
 ./configure || _exit "Error running configure."
 
-export XML_CATALOG_FILES="file:///etc/xml/catalog file://$(pwd)/build/catalog.xml"
+export XML_CATALOG_FILES="file://@PREFIX@/etc/xml/catalog file://$(pwd)/build/catalog.xml"
 
 make htmlman || _exit "Error running make htmlman"
 
--- samba-4.8.9/release-scripts/orig.build-htmlman-nogit	2018-01-14 21:41:58.000000000 +0100
+++ samba-4.8.9/release-scripts/build-htmlman-nogit	2019-07-27 01:16:30.000000000 +0200
@@ -32,7 +32,7 @@
 
 ./configure || _exit "Error running configure."
 
-export XML_CATALOG_FILES="file:///etc/xml/catalog file://$(pwd)/build/catalog.xml"
+export XML_CATALOG_FILES="file://@PREFIX@/etc/xml/catalog file://$(pwd)/build/catalog.xml"
 
 make clean || _exit "Error running make clean"
 
--- samba-4.8.9/release-scripts/orig.build-manpages-git	2018-01-14 21:41:58.000000000 +0100
+++ samba-4.8.9/release-scripts/build-manpages-git	2019-07-27 01:16:47.000000000 +0200
@@ -34,7 +34,7 @@
 
 ./configure || _exit "Error running configure."
 
-export XML_CATALOG_FILES="file:///etc/xml/catalog file://$(pwd)/build/catalog.xml"
+export XML_CATALOG_FILES="file://@PREFIX@/etc/xml/catalog file://$(pwd)/build/catalog.xml"
 
 make manpages || _exit "Error running make manpages"
 
--- samba-4.8.9/release-scripts/orig.build-manpages-nogit	2018-01-14 21:41:58.000000000 +0100
+++ samba-4.8.9/release-scripts/build-manpages-nogit	2019-07-27 01:17:03.000000000 +0200
@@ -32,7 +32,7 @@
 
 ./configure || _exit "Error running configure."
 
-export XML_CATALOG_FILES="file:///etc/xml/catalog file://$(pwd)/build/catalog.xml"
+export XML_CATALOG_FILES="file://@PREFIX@/etc/xml/catalog file://$(pwd)/build/catalog.xml"
 
 make clean || _exit "Error running make clean"
 
--- samba-4.8.9/buildtools/wafsamba/orig.wafsamba.py	2018-01-17 09:08:39.000000000 +0100
+++ samba-4.8.9/buildtools/wafsamba/wafsamba.py	2019-07-27 01:13:27.000000000 +0200
@@ -934,7 +934,7 @@
     bld.env.SAMBA_EXPAND_XSL = bld.srcnode.abspath() + '/docs-xml/xslt/expand-sambadoc.xsl'
     bld.env.SAMBA_MAN_XSL = bld.srcnode.abspath() + '/docs-xml/xslt/man.xsl'
     bld.env.SAMBA_CATALOG = bld.srcnode.abspath() + '/bin/default/docs-xml/build/catalog.xml'
-    bld.env.SAMBA_CATALOGS = 'file:///etc/xml/catalog file:///usr/local/share/xml/catalog file://' + bld.env.SAMBA_CATALOG
+    bld.env.SAMBA_CATALOGS = 'file://@PREFIX@/etc/xml/catalog file:///usr/local/share/xml/catalog file://' + bld.env.SAMBA_CATALOG
 
     for m in manpages.split():
         source = m + '.xml'
@@ -947,7 +947,7 @@
                             dep_vars=['SAMBA_MAN_XSL', 'SAMBA_EXPAND_XSL', 'SAMBA_CATALOG'],
                             rule='''XML_CATALOG_FILES="${SAMBA_CATALOGS}"
                                     export XML_CATALOG_FILES
-                                    ${XSLTPROC} --xinclude --stringparam noreference 0 -o ${TGT}.xml --nonet ${SAMBA_EXPAND_XSL} ${SRC[0].abspath(env)}
+                                    ${XSLTPROC} --xinclude --stringparam noreference 0 -o ${TGT}.xml ${SAMBA_EXPAND_XSL} ${SRC[0].abspath(env)}
                                     ${XSLTPROC} --nonet -o ${TGT} ${SAMBA_MAN_XSL} ${TGT}.xml'''
                             )
         bld.INSTALL_FILES('${MANDIR}/man%s' % m[-1], m, flat=True)
--- samba-4.8.9/docs-xml/orig.Makefile	2018-01-17 09:08:39.000000000 +0100
+++ samba-4.8.9/docs-xml/Makefile	2019-07-27 01:15:06.000000000 +0200
@@ -6,7 +6,7 @@
 # 		Jelmer Vernooij <jelmer@samba.org>
 include Makefile.settings
 
-export XML_CATALOG_FILES := $(XML_CATALOG_FILES) /etc/xml/catalog  $(shell pwd)/build/catalog.xml
+export XML_CATALOG_FILES := $(XML_CATALOG_FILES) @PREFIX@/etc/xml/catalog  $(shell pwd)/build/catalog.xml
 
 # Docs to build
 MAIN_DOCS = $(patsubst %/index.xml,%,$(wildcard */index.xml))
