--- kwin-5.13.3/orig.CMakeLists.txt	2018-07-10 12:08:42.000000000 +0200
+++ kwin-5.13.3/CMakeLists.txt	2020-07-03 15:06:26.408166544 +0200
@@ -268,12 +268,12 @@
     PURPOSE "Needed for running kwin_wayland"
     )
 
-find_package(Libcap)
-set_package_properties(Libcap PROPERTIES
-    TYPE OPTIONAL
-    PURPOSE "Needed for running kwin_wayland with real-time scheduling policy"
-)
-set(HAVE_LIBCAP ${Libcap_FOUND})
+# find_package(Libcap)
+# set_package_properties(Libcap PROPERTIES
+#     TYPE OPTIONAL
+#     PURPOSE "Needed for running kwin_wayland with real-time scheduling policy"
+# )
+# set(HAVE_LIBCAP ${Libcap_FOUND})
 
 include(ECMQMLModules)
 ecm_find_qmlmodule(QtQuick 2.3)
@@ -645,15 +645,15 @@
 endif()
 
 install(TARGETS kwin_wayland         ${INSTALL_TARGETS_DEFAULT_ARGS} )
-if (HAVE_LIBCAP)
-    install(
-    CODE "execute_process(
-            COMMAND
-                ${SETCAP_EXECUTABLE}
-                CAP_SYS_NICE=+ep
-                \$ENV{DESTDIR}${CMAKE_INSTALL_FULL_BINDIR}/kwin_wayland)"
-    )
-endif()
+# if (HAVE_LIBCAP)
+#     install(
+#     CODE "execute_process(
+#             COMMAND
+#                 ${SETCAP_EXECUTABLE}
+#                 CAP_SYS_NICE=+ep
+#                 \$ENV{DESTDIR}${CMAKE_INSTALL_FULL_BINDIR}/kwin_wayland)"
+#     )
+# endif()
 
 add_subdirectory(platformsupport)
 add_subdirectory(plugins)
