diff --git a/decorations/decoratedclient.cpp b/decorations/decoratedclient.cpp
index 1d3785b..ae9c7b8 100644
--- a/decorations/decoratedclient.cpp
+++ b/decorations/decoratedclient.cpp
@@ -66,6 +66,11 @@ DecoratedClientImpl::DecoratedClientImpl(AbstractClient *client, KDecoration2::D
             if (oldSize.height() != m_clientSize.height()) {
                 emit decoratedClient->heightChanged(m_clientSize.height());
             }
+#if KDECORATION2_VERSION >= QT_VERSION_CHECK(5, 18, 0)
+            emit decoratedClient->sizeChanged(m_clientSize);
+#else
+            emit sizeChanged(m_clientSize);
+#endif
         }
     );
     connect(client, &AbstractClient::desktopChanged, this,
@@ -268,6 +273,11 @@ int DecoratedClientImpl::height() const
     return m_clientSize.height();
 }
 
+QSize DecoratedClientImpl::size() const
+{
+    return m_clientSize;
+}
+
 bool DecoratedClientImpl::isMaximizedVertically() const
 {
     return m_client->requestedMaximizeMode() & MaximizeVertical;
diff --git a/decorations/decoratedclient.h b/decorations/decoratedclient.h
index e9c1b3b..cc2c44d 100644
--- a/decorations/decoratedclient.h
+++ b/decorations/decoratedclient.h
@@ -27,6 +27,8 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #include <QObject>
 #include <QTimer>
 
+#include <kdecoration2_version.h>
+
 namespace KWin
 {
 
@@ -66,6 +68,11 @@ public:
     QPalette palette() const override;
     QColor color(KDecoration2::ColorGroup group, KDecoration2::ColorRole role) const override;
     bool providesContextHelp() const override;
+#if KDECORATION2_VERSION >= QT_VERSION_CHECK(5, 18, 0)
+    QSize size() const override;
+#else
+    QSize size() const;
+#endif
     int width() const override;
     WId windowId() const override;
 
@@ -87,7 +94,11 @@ public:
     void requestToggleOnAllDesktops() override;
     void requestToggleShade() override;
 
+#if KDECORATION2_VERSION >= QT_VERSION_CHECK(5, 18, 0)
+    void showApplicationMenu(int actionId) override;
+#else
     void showApplicationMenu(int actionId);
+#endif
 
     AbstractClient *client() {
         return m_client;
@@ -101,6 +112,11 @@ public:
 
     void signalShadeChange();
 
+#if KDECORATION2_VERSION < QT_VERSION_CHECK(5, 18, 0)
+Q_SIGNALS:
+    void sizeChanged(QSize&);
+#endif
+
 private Q_SLOTS:
     void delayedRequestToggleMaximization(Options::WindowOperation operation);
 
diff --git a/decorations/decorationrenderer.cpp b/decorations/decorationrenderer.cpp
index 3efbceb..bad4385 100644
--- a/decorations/decorationrenderer.cpp
+++ b/decorations/decorationrenderer.cpp
@@ -46,6 +46,9 @@ Renderer::Renderer(DecoratedClientImpl *client)
     connect(client->decoration(), &KDecoration2::Decoration::bordersChanged, this, markImageSizesDirty);
     connect(client->decoratedClient(), &KDecoration2::DecoratedClient::widthChanged, this, markImageSizesDirty);
     connect(client->decoratedClient(), &KDecoration2::DecoratedClient::heightChanged, this, markImageSizesDirty);
+#if KDECORATION2_VERSION >= QT_VERSION_CHECK(5, 18, 0)
+    connect(client->decoratedClient(), &KDecoration2::DecoratedClient::sizeChanged, this, markImageSizesDirty);
+#endif
 }
 
 Renderer::~Renderer() = default;
diff --git a/kcmkwin/kwindecoration/declarative-plugin/previewclient.cpp b/kcmkwin/kwindecoration/declarative-plugin/previewclient.cpp
index c5856eb..4caa524 100644
--- a/kcmkwin/kwindecoration/declarative-plugin/previewclient.cpp
+++ b/kcmkwin/kwindecoration/declarative-plugin/previewclient.cpp
@@ -79,6 +79,9 @@ PreviewClient::PreviewClient(DecoratedClient *c, Decoration *decoration)
     connect(this, &PreviewClient::onAllDesktopsChanged,         c, &DecoratedClient::onAllDesktopsChanged);
     connect(this, &PreviewClient::widthChanged,                 c, &DecoratedClient::widthChanged);
     connect(this, &PreviewClient::heightChanged,                c, &DecoratedClient::heightChanged);
+#if KDECORATION2_VERSION >= QT_VERSION_CHECK(5, 18, 0)
+    connect(this, &PreviewClient::sizeChanged,                  c, &DecoratedClient::sizeChanged);
+#endif
     connect(this, &PreviewClient::iconChanged,                  c, &DecoratedClient::iconChanged);
     connect(this, &PreviewClient::paletteChanged,               c, &DecoratedClient::paletteChanged);
 //         connect(this, &PreviewClient::, c, &DecoratedClient::);
@@ -135,6 +138,12 @@ int PreviewClient::height() const
     return m_height;
 }
 
+QSize PreviewClient::size() const
+{
+    QSize size(m_width, m_height);
+    return size;
+}
+
 QString PreviewClient::caption() const
 {
     return m_caption;
diff --git a/kcmkwin/kwindecoration/declarative-plugin/previewclient.h b/kcmkwin/kwindecoration/declarative-plugin/previewclient.h
index df36083..1beec59 100644
--- a/kcmkwin/kwindecoration/declarative-plugin/previewclient.h
+++ b/kcmkwin/kwindecoration/declarative-plugin/previewclient.h
@@ -26,6 +26,8 @@
 #include <QObject>
 #include <QPalette>
 
+#include <kdecoration2_version.h>
+
 class QAbstractItemModel;
 
 namespace KDecoration2
@@ -90,6 +92,11 @@ public:
 
     int width() const override;
     int height() const override;
+#if KDECORATION2_VERSION >= QT_VERSION_CHECK(5, 18, 0)
+    QSize size() const override;
+#else
+    QSize size() const;
+#endif
     QPalette palette() const override;
     QColor color(ColorGroup group, ColorRole role) const override;
     Qt::Edges adjacentScreenEdges() const override;
@@ -110,7 +117,11 @@ public:
     void requestShowApplicationMenu(const QRect &rect, int actionId) override;
     void requestToggleOnAllDesktops() override;
 
+#if KDECORATION2_VERSION >= QT_VERSION_CHECK(5, 18, 0)
+    void showApplicationMenu(int actionId) override;
+#else
     void showApplicationMenu(int actionId);
+#endif
 
     void setCaption(const QString &caption);
     void setActive(bool active);
@@ -169,6 +180,7 @@ Q_SIGNALS:
     void desktopChanged(int);
     void widthChanged(int);
     void heightChanged(int);
+    void sizeChanged(QSize&);
     void paletteChanged(const QPalette&);
     void bordersTopEdgeChanged(bool);
     void bordersLeftEdgeChanged(bool);
