diff --git a/src/modules/CUVID/CMakeLists.txt b/src/modules/CUVID/CMakeLists.txt
index a9fca1e1..03d57dc8 100644
--- a/src/modules/CUVID/CMakeLists.txt
+++ b/src/modules/CUVID/CMakeLists.txt
@@ -48,4 +48,8 @@ if(WIN32)
     target_link_libraries(${PROJECT_NAME} opengl32)
 endif()
 
+if(APPLE)
+    target_link_libraries(${PROJECT_NAME} ${OPENGL_LIBRARY})
+endif()
+
 install(TARGETS ${PROJECT_NAME} LIBRARY DESTINATION ${MODULES_INSTALL_PATH})
diff --git a/src/modules/FFmpeg/VAAPI.cpp b/src/modules/FFmpeg/VAAPI.cpp
index 3c70c592b1290799c67614f14f3faef8535ec6b0..81a4f9f0ff41d00e80f48600331aa3dc2844a4eb 100644
--- a/src/modules/FFmpeg/VAAPI.cpp
+++ b/src/modules/FFmpeg/VAAPI.cpp
@@ -502,7 +502,7 @@ bool VAAPI::checkCodec(const char *codecName) const
         ;
     }
 #endif
-#if VA_VERSION_HEX >= 0x270000 // 1.7.0
+#if VA_CHECK_VERSION(1, 7, 0) // 1.7.0
     else if (qstrcmp(codecName, "av1") == 0)
     {
         return vaProfiles.contains(VAProfileAV1Profile0) || vaProfiles.contains(VAProfileAV1Profile1);
