diff --git a/src/gui/macOS/QMPlay2MacExtensions.mm b/src/gui/macOS/QMPlay2MacExtensions.mm
index c67d93f..d306ae9 100644
--- a/src/gui/macOS/QMPlay2MacExtensions.mm
+++ b/src/gui/macOS/QMPlay2MacExtensions.mm
@@ -21,7 +21,12 @@ private:
 		if (eventType == "mac_generic_NSEvent")
 		{
 			NSEvent *event = static_cast<NSEvent *>(message);
-			if ([event type] == NSEventTypeSystemDefined)
+			if (
+#if defined( MAC_OS_X_VERSION_10_12 ) && (MAC_OS_X_VERSION_MIN_REQUIRED >= MAC_OS_X_VERSION_10_12)
+				[event type] == NSEventTypeSystemDefined)
+#else
+				[event type] == NSSystemDefined)
+#endif
 			{
 				const int  keyCode   = ([event data1] & 0xFFFF0000) >> 16;
 				const int  keyFlags  = ([event data1] & 0x0000FFFF);
