diff --git a/src/modules/FFmpeg/FormatContext.cpp b/src/modules/FFmpeg/FormatContext.cpp
index a5c45496..8e14b5d8 100644
--- a/src/modules/FFmpeg/FormatContext.cpp
+++ b/src/modules/FFmpeg/FormatContext.cpp
@@ -363,7 +363,11 @@ QList<QMPlay2Tag> FormatContext::tags() const
         else
         {
             // check for all fields starting with "lyrics-" because libavformat's ID3v2 code adds lyric fields like that
+#if LIBAVUTIL_VERSION_MAJOR >= 58
             for (const AVDictionaryEntry *avTag = av_dict_iterate(dict, nullptr); avTag; avTag = av_dict_iterate(dict, avTag))
+#else
+            for (const AVDictionaryEntry *avTag = av_dict_get(dict, "", nullptr, AV_DICT_IGNORE_SUFFIX); avTag; avTag = av_dict_get(dict, "", avTag, AV_DICT_IGNORE_SUFFIX))
+#endif
             {
                 if (!std::strncmp(avTag->key, "lyrics-", sizeof("lyrics-")) && !(value = getTag(avTag)).isEmpty())
                 {
