diff --git a/src/gui/MainWidget.cpp b/src/gui/MainWidget.cpp
index b61316b7d19d5ef5d231a2b2b4a85a0e61716889..f2e6983d77aef6cf5542cd95b85a2c6a73c745c3 100644
--- a/src/gui/MainWidget.cpp
+++ b/src/gui/MainWidget.cpp
@@ -613,8 +613,10 @@ void MainWidget::videoStarted(bool noVideo)
     if (autoRestoreMainWindowOnVideo || (noVideo && autoOpenVideoWindow))
     {
         if (!videoDock->isVisible())
+        {
             videoDock->show();
-        videoDock->raise();
+            videoDock->raise();
+        }
     }
     if (autoRestoreMainWindowOnVideo)
     {
@@ -622,11 +624,11 @@ void MainWidget::videoStarted(bool noVideo)
         {
             toggleVisibility();
         }
-        else
-        {
-            activateWindow();
-            raise();
-        }
+//         else
+//         {
+//             activateWindow();
+//             raise();
+//         }
     }
     m_restoreWindowOnVideo = false;
 }
