diff --git a/src/gui/Main.cpp b/src/gui/Main.cpp
index e21dc2c74864dbdc9cbfa2cdad311774979e3ecb..a53665eee3a9b79b873fc07c30bb97674bb2ee43 100644
--- a/src/gui/Main.cpp
+++ b/src/gui/Main.cpp
@@ -43,7 +43,10 @@
 #include <QFile>
 #include <QDir>
 #ifdef Q_OS_MACOS
+	#include <QMap>
 	#include <QProcess>
+	#include <syslog.h>
+	#include <stdarg.h>
 #endif
 
 #include <csignal>
@@ -459,8 +462,19 @@ static LRESULT CALLBACK MMKeysHookProc(int code, WPARAM wparam, LPARAM lparam)
 #ifndef Q_OS_ANDROID
 static void messageHandler(QtMsgType type, const QMessageLogContext &context, const QString &message)
 {
-	fprintf(stderr, "%s\n", qFormatLogMessage(type, context, message).toLocal8Bit().constData());
+	const char *messageText = qFormatLogMessage(type, context, message).toLocal8Bit().constData();
+	fprintf(stderr, "%s\n", messageText);
 	fflush(stderr);
+#if defined Q_OS_MACOS
+	static const QMap<QtMsgType,int> type2Level = {
+		{QtDebugMsg, LOG_DEBUG},
+		{QtWarningMsg, LOG_WARNING},
+		{QtCriticalMsg, LOG_CRIT},
+		{QtFatalMsg, LOG_CRIT},
+		{QtInfoMsg, LOG_INFO}
+	};
+	syslog(LOG_NDELAY|LOG_PID|type2Level[type], "%s", messageText);
+#endif
 }
 #endif
 
