diff --git a/src/modules/Extensions/YouTube.cpp b/src/modules/Extensions/YouTube.cpp
index 35d09fbb3aafd184b323733baceeef08b4507263..e3000b4f5f98780767fe58fac5ed139c78a3b096 100644
--- a/src/modules/Extensions/YouTube.cpp
+++ b/src/modules/Extensions/YouTube.cpp
@@ -173,7 +173,12 @@ void ResultsYoutube::contextMenu(const QPoint &point)
 
         if (i == 0)
         {
+#if !(QT_VERSION_MAJOR == 5 && QT_VERSION_MINOR == 12)
+            // in Qt 5.12.6 calling QDesktopServices::openUrl() can cause a deadlock when a video
+            // is playing. I'm not seeing a simple way to test for `isPlaying` so we just
+            // remove the functionality for now.
             subMenu->addAction(tr("Open the page in the browser"), this, SLOT(openPage()));
+#endif
             subMenu->addAction(tr("Copy page address"), this, SLOT(copyPageURL()));
             subMenu->addSeparator();
         }
