--- a/src/qmplay2/orig.YouTubeDL.cpp	2025-11-18 01:17:46.841576530 +0100
+++ b/src/qmplay2/YouTubeDL.cpp	2025-11-18 01:29:19.278532308 +0100
@@ -225,6 +225,7 @@
     QStringList processArgs, process2Args;
     if (!rawOutput)
         processArgs += "-g";
+    processArgs += {"--js-runtimes", "node", "--js-runtimes", "quickjs"};
     processArgs += args;
     process2Args = processArgs;
     processArgs += m_commonArgs;
