diff --git a/src/modules/CUVID/CMakeLists.txt b/src/modules/CUVID/CMakeLists.txt
index a9fca1e1..03d57dc8 100644
--- a/src/modules/CUVID/CMakeLists.txt
+++ b/src/modules/CUVID/CMakeLists.txt
@@ -48,4 +48,8 @@ if(WIN32)
     target_link_libraries(${PROJECT_NAME} opengl32)
 endif()
 
+if(APPLE)
+    target_link_libraries(${PROJECT_NAME} ${OPENGL_LIBRARY})
+endif()
+
 install(TARGETS ${PROJECT_NAME} LIBRARY DESTINATION ${MODULES_INSTALL_PATH})
diff --git a/lang/zh_CN.ts b/lang/zh_CN.ts
index 99e40f3c..15077db3 100644
--- a/lang/zh_CN.ts
+++ b/lang/zh_CN.ts
@@ -1794,10 +1794,6 @@
         <source>Exclusive mode</source>
         <translation>独占模式</translation>
     </message>
-    <message>
-        <source>Preferred video codec</source>
-        <translation type>首选视频编解码器</translation>
-    </message>
 </context>
 <context>
     <name>ModulesList</name>
