diff --git a/src/modules/CUVID/CMakeLists.txt b/src/modules/CUVID/CMakeLists.txt
index a9fca1e1..03d57dc8 100644
--- a/src/modules/CUVID/CMakeLists.txt
+++ b/src/modules/CUVID/CMakeLists.txt
@@ -48,4 +48,8 @@ if(WIN32)
     target_link_libraries(${PROJECT_NAME} opengl32)
 endif()
 
+if(APPLE)
+    target_link_libraries(${PROJECT_NAME} ${OPENGL_LIBRARY})
+endif()
+
 install(TARGETS ${PROJECT_NAME} LIBRARY DESTINATION ${MODULES_INSTALL_PATH})
