diff --git a/CMakeLists.txt b/CMakeLists.txt
index a75ed2c692adec5bf4529867e563570f526acf3c..3cf57583a46fc364c1651a677c1636b6807dddfa 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -226,6 +226,7 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" OR CMAKE_CXX_COMPILER_ID MATCHES "Clang"
         set(CMAKE_CXX_FLAGS "-fsanitize=undefined ${CMAKE_CXX_FLAGS}")
     endif()
 
+    set(CMAKE_CXX_FLAGS            "-fno-exceptions ${CMAKE_CXX_FLAGS}")
     # Disable exceptions, hide symbols, and also make sure that debugging symbols are disabled
     set(CMAKE_CXX_FLAGS_RELEASE    "-fno-exceptions ${RELEASE_DEBUG_FLAG} -fvisibility=hidden -fvisibility-inlines-hidden ${CMAKE_CXX_FLAGS_RELEASE}")
     set(CMAKE_CXX_FLAGS_MINSIZEREL "-fno-exceptions ${RELEASE_DEBUG_FLAG} -fvisibility=hidden -fvisibility-inlines-hidden ${CMAKE_CXX_FLAGS_MINSIZEREL}")
@@ -241,8 +242,9 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" OR CMAKE_CXX_COMPILER_ID MATCHES "Clang"
     set(CMAKE_CXX_FLAGS "-Wall ${CMAKE_CXX_FLAGS}")
 
     if(USE_LINK_TIME_OPTIMIZATION)
+        set(CMAKE_CXX_FLAGS            "-flto ${CMAKE_CXX_FLAGS}")
         set(CMAKE_CXX_FLAGS_RELEASE    "-flto ${CMAKE_CXX_FLAGS_RELEASE}")
-        set(CMAKE_CXX_FLAGS_MINSIZEREL "-flto ${CMAKE_CXX_FLAGS_RELEASE}")
+        set(CMAKE_CXX_FLAGS_MINSIZEREL "-flto ${CMAKE_CXX_FLAGS_MINSIZEREL}")
     endif()
 
     if(CMAKE_MODULE_LINKER_FLAGS)
@@ -265,19 +267,22 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" OR CMAKE_CXX_COMPILER_ID MATCHES "Clang"
 endif()
 
 if(APPLE)
-    set(MAC_BUNDLE_PATH "${CMAKE_INSTALL_PREFIX}/QMPlay2.app")
+    set(MAC_BUNDLE_PATH "${BUNDLE_INSTALL_DIR}/QMPlay2.app")
 
     # set paths to Bundle
     set(CMAKE_INSTALL_DATAROOTDIR "${MAC_BUNDLE_PATH}/Contents/share")
     set(CMAKE_INSTALL_LIBDIR      "${MAC_BUNDLE_PATH}/Contents/MacOS")
+    if(NOT APPLE_STANDALONE_BUNDLE)
+        include(GNUInstallDirs)
+    endif()
 elseif(NOT WIN32)
     include(GNUInstallDirs)
 endif()
 
-if(NOT APPLE)
+# if(NOT APPLE)
     set(INSTALL_PATH_ICONS "${CMAKE_INSTALL_DATAROOTDIR}/icons")
     set(INSTALL_PATH_MIME "${CMAKE_INSTALL_DATAROOTDIR}/mime")
-endif()
+# endif()
 
 if(CMAKE_SYSTEM_NAME STREQUAL "FreeBSD" AND CMAKE_INSTALL_PREFIX STREQUAL "/usr/local")
     set(INSTALL_PATH_MAN "man") # exception for FreeBSD
@@ -322,7 +327,7 @@ else()
     install(FILES AUTHORS ChangeLog LICENSE TODO README.md DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/qmplay2")
 endif()
 
-if(NOT APPLE AND NOT WIN32)
+# if(NOT APPLE AND NOT WIN32)
     # uninstall target
     configure_file(
         "${CMAKE_CURRENT_SOURCE_DIR}/src/cmake/Uninstall.cmake.in"
@@ -331,7 +336,7 @@ if(NOT APPLE AND NOT WIN32)
 
     add_custom_target(uninstall
         COMMAND ${CMAKE_COMMAND} -P "${CMAKE_CURRENT_BINARY_DIR}/cmake_uninstall.cmake")
-endif()
+# endif()
 
 # Detect QtSvg and show warning if it doesn't exist
 if(NOT SVG_MODULE_FOUND)
diff --git a/src/gui/CMakeLists.txt b/src/gui/CMakeLists.txt
index 7f8c22b59971aa20919831d84aeffa8a58ba46c7..a664af9f675f24471e291a0c2738457aee26e496 100644
--- a/src/gui/CMakeLists.txt
+++ b/src/gui/CMakeLists.txt
@@ -214,47 +214,49 @@ endif()
 if(WIN32)
     install(TARGETS ${PROJECT_NAME} RUNTIME DESTINATION ${CMAKE_INSTALL_PREFIX}/)
 elseif(APPLE)
-    install(TARGETS ${PROJECT_NAME} BUNDLE DESTINATION ${CMAKE_INSTALL_PREFIX})
+    install(TARGETS ${PROJECT_NAME} BUNDLE DESTINATION ${BUNDLE_INSTALL_DIR})
 
     set(QT_LIBS_DIR "${Qt5Widgets_DIR}/../..")
     set(QT_PLUGINS_DIR "${QT_LIBS_DIR}/../plugins")
-    install(FILES
-        "${QT_PLUGINS_DIR}/platforms/libqcocoa.dylib"
-        DESTINATION "${MAC_BUNDLE_PATH}/Contents/plugins/platforms")
-    install(FILES
-        "${QT_PLUGINS_DIR}/iconengines/libqsvgicon.dylib"
-        DESTINATION "${MAC_BUNDLE_PATH}/Contents/plugins/iconengines")
-    install(FILES
-        "${QT_PLUGINS_DIR}/styles/libqmacstyle.dylib"
-        DESTINATION "${MAC_BUNDLE_PATH}/Contents/plugins/styles"
-        OPTIONAL)
-    install(DIRECTORY
-        "${QT_PLUGINS_DIR}/imageformats"
-        DESTINATION "${MAC_BUNDLE_PATH}/Contents/plugins"
-        PATTERN "*_debug.dylib" EXCLUDE
-        PATTERN "*.dSYM" EXCLUDE)
-    install(DIRECTORY
-        "${QT_LIBS_DIR}/../translations"
-        DESTINATION "${MAC_BUNDLE_PATH}/Contents"
-        FILES_MATCHING
-        PATTERN "qtbase_*.qm")
-    if(EXISTS "/usr/local/bin/ffmpeg")
-        install(PROGRAMS
-            "/usr/local/bin/ffmpeg"
-            DESTINATION "${MAC_BUNDLE_PATH}/Contents/MacOS")
-    else()
-        message(WARNING "FFmpeg executable not copied!")
+    if(APPLE_STANDALONE_BUNDLE)
+        install(FILES
+            "${QT_PLUGINS_DIR}/platforms/libqcocoa.dylib"
+            DESTINATION "${MAC_BUNDLE_PATH}/Contents/plugins/platforms")
+        install(FILES
+            "${QT_PLUGINS_DIR}/iconengines/libqsvgicon.dylib"
+            DESTINATION "${MAC_BUNDLE_PATH}/Contents/plugins/iconengines")
+        install(FILES
+            "${QT_PLUGINS_DIR}/styles/libqmacstyle.dylib"
+            DESTINATION "${MAC_BUNDLE_PATH}/Contents/plugins/styles"
+            OPTIONAL)
+        install(DIRECTORY
+            "${QT_PLUGINS_DIR}/imageformats"
+            DESTINATION "${MAC_BUNDLE_PATH}/Contents/plugins"
+            PATTERN "*_debug.dylib" EXCLUDE
+            PATTERN "*.dSYM" EXCLUDE)
+        install(DIRECTORY
+            "${QT_LIBS_DIR}/../translations"
+            DESTINATION "${MAC_BUNDLE_PATH}/Contents"
+            FILES_MATCHING
+            PATTERN "qtbase_*.qm")
+        if(EXISTS "@PREFIX@/bin/ffmpeg")
+            install(PROGRAMS
+                "@PREFIX@/bin/ffmpeg"
+                DESTINATION "${MAC_BUNDLE_PATH}/Contents/MacOS")
+        else()
+            message(WARNING "FFmpeg executable not copied!")
+        endif()
+        install(CODE "
+            include(BundleUtilities)
+            set(BU_CHMOD_BUNDLE_ITEMS ON)
+            list(APPEND DIRS @PREFIX@lib ${QT_LIBS_DIR})
+            file(GLOB_RECURSE QMPLAY2_MODULES_AND_QT_PLUGINS
+                \"${MAC_BUNDLE_PATH}/Contents/MacOS/modules/*\"
+                \"${MAC_BUNDLE_PATH}/Contents/plugins/*.dylib\")
+            file(WRITE \"${MAC_BUNDLE_PATH}/Contents/Resources/qt.conf\")
+            fixup_bundle(${MAC_BUNDLE_PATH} \"\${QMPLAY2_MODULES_AND_QT_PLUGINS}\" \"\${DIRS}\")
+        ")
     endif()
-    install(CODE "
-        include(BundleUtilities)
-        set(BU_CHMOD_BUNDLE_ITEMS ON)
-        list(APPEND DIRS /usr/local/lib ${QT_LIBS_DIR})
-        file(GLOB_RECURSE QMPLAY2_MODULES_AND_QT_PLUGINS
-            \"${MAC_BUNDLE_PATH}/Contents/MacOS/modules/*\"
-            \"${MAC_BUNDLE_PATH}/Contents/plugins/*.dylib\")
-        file(WRITE \"${MAC_BUNDLE_PATH}/Contents/Resources/qt.conf\")
-        fixup_bundle(${MAC_BUNDLE_PATH} \"\${QMPLAY2_MODULES_AND_QT_PLUGINS}\" \"\${DIRS}\")
-    ")
 else()
     # executable
     install(TARGETS ${PROJECT_NAME} RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
diff --git a/src/gui/MainWidget.cpp b/src/gui/MainWidget.cpp
index 016d3bc0b444f0a85727b5a8f0276c1d5a3d1ec4..eeeddf730d9ea1840a528d94787f43ba1ad96f8e 100644
--- a/src/gui/MainWidget.cpp
+++ b/src/gui/MainWidget.cpp
@@ -94,7 +94,7 @@ public:
     }
 };
 
-#ifndef Q_OS_MACOS
+// #ifndef Q_OS_MACOS
 static void copyMenu(QMenu *dest, QMenu *src, QMenu *dontCopy = nullptr)
 {
     QMenu *newMenu = new QMenu(src->title(), dest);
@@ -110,7 +110,7 @@ static void copyMenu(QMenu *dest, QMenu *src, QMenu *dontCopy = nullptr)
     }
     dest->addMenu(newMenu);
 }
-#endif
+// #endif
 
 /* MainWidget */
 MainWidget::MainWidget(QList<QPair<QString, QString>> &arguments)
@@ -158,7 +158,7 @@ MainWidget::MainWidget(QList<QPair<QString, QString>> &arguments)
 
     QMPlay2GUI.menuBar = new MenuBar;
 
-#if !defined Q_OS_MACOS && !defined Q_OS_ANDROID
+#if /*!defined Q_OS_MACOS &&*/ !defined Q_OS_ANDROID
     tray = new QSystemTrayIcon(this);
     tray->setIcon(QMPlay2Core.getIconFromTheme("QMPlay2-panel", QMPlay2Core.getQMPlay2Icon()));
     tray->setVisible(settings.getBool("TrayVisible", true));
@@ -341,7 +341,7 @@ MainWidget::MainWidget(QList<QPair<QString, QString>> &arguments)
     if (settings.getBool("MainWidget/TabPositionNorth"))
         setTabPosition(Qt::AllDockWidgetAreas, QTabWidget::North);
 
-#if !defined Q_OS_MACOS && !defined Q_OS_ANDROID
+#if /*!defined Q_OS_MACOS &&*/ !defined Q_OS_ANDROID
     const bool menuHidden = settings.getBool("MainWidget/MenuHidden", false);
     menuBar->setVisible(!menuHidden);
     hideMenuAct = new QAction(tr("&Hide menu bar"), menuBar);
@@ -397,7 +397,7 @@ MainWidget::MainWidget(QList<QPair<QString, QString>> &arguments)
     fullScreenDockWidgetState = settings.getByteArray("MainWidget/FullScreenDockWidgetState");
     if (settings.getBool("MainWidget/AlwaysOnTop"))
         menuBar->window->alwaysOnTop->trigger();
-#if defined Q_OS_MACOS || defined Q_OS_ANDROID
+#if /*defined Q_OS_MACOS ||*/ defined Q_OS_ANDROID
     show();
 #else
     setVisible(settings.getBool("MainWidget/isVisible", true) ? true : !isTrayVisible());
@@ -940,7 +940,7 @@ void MainWidget::createMenuBar()
     setMenuBar(menuBar);
 
     QMenu *secondMenu = new QMenu(this);
-#ifndef Q_OS_MACOS
+// #ifndef Q_OS_MACOS
     copyMenu(secondMenu, menuBar->window);
     secondMenu->addMenu(menuBar->widgets);
     copyMenu(secondMenu, menuBar->playlist, menuBar->playlist->extensions);
@@ -950,32 +950,32 @@ void MainWidget::createMenuBar()
     copyMenu(secondMenu, menuBar->help);
     if (tray)
         tray->setContextMenu(secondMenu);
-#else //On OS X add only the most important menu actions to dock menu
-    secondMenu->addAction(menuBar->player->togglePlay);
-    secondMenu->addAction(menuBar->player->stop);
-    secondMenu->addAction(menuBar->player->next);
-    secondMenu->addAction(menuBar->player->prev);
-    secondMenu->addSeparator();
-    secondMenu->addAction(menuBar->player->toggleMute);
-    secondMenu->addSeparator();
-    // Copy action, because PreferencesRole doesn't show in dock menu.
-    QAction *settings = new QAction(menuBar->options->settings->icon(), menuBar->options->settings->text(), menuBar->options->settings->parent());
-    connect(settings, &QAction::triggered, menuBar->options->settings, &QAction::trigger);
-    secondMenu->addAction(settings);
-
-    QAction *newInstanceAct = new QAction(tr("New window"), secondMenu);
-    connect(newInstanceAct, &QAction::triggered, [] {
-        QProcess::startDetached(QCoreApplication::applicationFilePath(), {"-noplay"}, QCoreApplication::applicationDirPath());
-    });
-    secondMenu->addSeparator();
-    secondMenu->addAction(newInstanceAct);
-
-    qt_mac_set_dock_menu(secondMenu);
-#endif
+// #else //On OS X add only the most important menu actions to dock menu
+//     secondMenu->addAction(menuBar->player->togglePlay);
+//     secondMenu->addAction(menuBar->player->stop);
+//     secondMenu->addAction(menuBar->player->next);
+//     secondMenu->addAction(menuBar->player->prev);
+//     secondMenu->addSeparator();
+//     secondMenu->addAction(menuBar->player->toggleMute);
+//     secondMenu->addSeparator();
+//     // Copy action, because PreferencesRole doesn't show in dock menu.
+//     QAction *settings = new QAction(menuBar->options->settings->icon(), menuBar->options->settings->text(), menuBar->options->settings->parent());
+//     connect(settings, &QAction::triggered, menuBar->options->settings, &QAction::trigger);
+//     secondMenu->addAction(settings);
+// 
+//     QAction *newInstanceAct = new QAction(tr("New window"), secondMenu);
+//     connect(newInstanceAct, &QAction::triggered, [] {
+//         QProcess::startDetached(QCoreApplication::applicationFilePath(), {"-noplay"}, QCoreApplication::applicationDirPath());
+//     });
+//     secondMenu->addSeparator();
+//     secondMenu->addAction(newInstanceAct);
+// 
+//     qt_mac_set_dock_menu(secondMenu);
+// #endif
 }
 void MainWidget::trayIconClicked(QSystemTrayIcon::ActivationReason reason)
 {
-#if !defined Q_OS_MACOS && !defined Q_OS_ANDROID
+#if /*!defined Q_OS_MACOS &&*/ !defined Q_OS_ANDROID
     switch (reason)
     {
         case QSystemTrayIcon::Trigger:
@@ -1003,7 +1003,7 @@ void MainWidget::toggleCompactView()
 
         hideAllExtensions();
 
-#if !defined Q_OS_MACOS && !defined Q_OS_ANDROID
+#if /*!defined Q_OS_MACOS &&*/ !defined Q_OS_ANDROID
         menuBar->hide();
 #endif
         mainTB->hide();
@@ -1026,7 +1026,7 @@ void MainWidget::toggleCompactView()
 
         videoDock->fullScreen(false);
 
-#if !defined Q_OS_MACOS && !defined Q_OS_ANDROID
+#if /*!defined Q_OS_MACOS &&*/ !defined Q_OS_ANDROID
         menuBar->setVisible(!hideMenuAct->isChecked());
 #endif
 
@@ -1082,7 +1082,7 @@ void MainWidget::toggleFullScreen()
         dockWidgetState = saveState();
 #endif // Q_OS_ANDROID
 
-#if !defined Q_OS_MACOS && !defined Q_OS_ANDROID
+#if /*!defined Q_OS_MACOS &&*/ !defined Q_OS_ANDROID
         menuBar->hide();
 #endif
         statusBar->hide();
@@ -1111,9 +1111,9 @@ void MainWidget::toggleFullScreen()
         videoDock->fullScreen(true);
         videoDock->show();
 
-#ifdef Q_OS_MACOS
+// #ifdef Q_OS_MACOS
         menuBar->window->toggleVisibility->setEnabled(false);
-#endif
+// #endif
         menuBar->window->toggleCompactView->setEnabled(false);
         menuBar->window->toggleFullScreen->setShortcuts(QList<QKeySequence>() << menuBar->window->toggleFullScreen->shortcut() << QKeySequence("ESC"));
         fullScreen = true;
@@ -1132,9 +1132,9 @@ void MainWidget::toggleFullScreen()
     }
     else
     {
-#ifdef Q_OS_MACOS
+// #ifdef Q_OS_MACOS
         menuBar->window->toggleVisibility->setEnabled(true);
-#endif
+// #endif
         menuBar->window->toggleCompactView->setEnabled(true);
         menuBar->window->toggleFullScreen->setShortcuts(QList<QKeySequence>() << menuBar->window->toggleFullScreen->shortcut());
 
@@ -1177,7 +1177,7 @@ void MainWidget::toggleFullScreen()
             if (QDockWidget *dw = QMPlay2Ext->getDockWidget())
                 dw->setFeatures(QDockWidget::AllDockWidgetFeatures);
 
-#if !defined Q_OS_MACOS && !defined Q_OS_ANDROID
+#if /*!defined Q_OS_MACOS &&*/ !defined Q_OS_ANDROID
         menuBar->setVisible(!hideMenuAct->isChecked());
 #endif
         statusBar->show();
@@ -1403,7 +1403,7 @@ void MainWidget::about()
     }
 }
 
-#if !defined Q_OS_MACOS && !defined Q_OS_ANDROID
+// #if !defined Q_OS_MACOS && !defined Q_OS_ANDROID
 void MainWidget::hideMenu(bool h)
 {
     if (fullScreen || isCompactView)
@@ -1414,7 +1414,7 @@ void MainWidget::hideMenu(bool h)
         QMPlay2Core.getSettings().set("MainWidget/MenuHidden", h);
     }
 }
-#endif
+// #endif
 void MainWidget::lockWidgets(bool l)
 {
     if (fullScreen || isCompactView)
@@ -1498,7 +1498,7 @@ QMenu *MainWidget::createPopupMenu()
     QMenu *popupMenu = QMainWindow::createPopupMenu();
     if (!fullScreen && !isCompactView)
     {
-#if !defined Q_OS_MACOS && !defined Q_OS_ANDROID
+#if /*!defined Q_OS_MACOS &&*/ !defined Q_OS_ANDROID
         popupMenu->insertAction(popupMenu->actions().value(0), hideMenuAct);
         popupMenu->insertSeparator(popupMenu->actions().value(1));
         popupMenu->addSeparator();
@@ -1690,6 +1690,13 @@ void MainWidget::mouseMoveEvent(QMouseEvent *e)
         if (canDisplayLeftPanel && !playlistDock->isVisible() && mPosX <= trigger1)
         {
             showToolBar(true); //Before restoring dock widgets - show toolbar and status bar
+            // FullScreenDockWidgetState has been known tp become corrupted, which can cause
+            // Qt to get stuck in an infinite loop trying to layout the videoDock. This can
+            // lead to a crash or abort. Prevent this by clearing the setting here. If all
+            // goes well, the value we will be using now will be saved again.
+            auto &settings = QMPlay2Core.getSettings();
+            settings.remove("MainWidget/FullScreenDockWidgetState");
+            settings.flush();
             restoreState(fullScreenDockWidgetState);
 
             const QList<QDockWidget *> tDW = tabifiedDockWidgets(infoDock);
@@ -1772,9 +1779,9 @@ void MainWidget::closeEvent(QCloseEvent *e)
         settings.set("MainWidget/DockWidgetState", dockWidgetState);
     settings.set("MainWidget/FullScreenDockWidgetState", fullScreenDockWidgetState);
     settings.set("MainWidget/AlwaysOnTop", !!(windowFlags() & Qt::WindowStaysOnTopHint));
-#ifndef Q_OS_MACOS
+// #ifndef Q_OS_MACOS
     settings.set("MainWidget/isVisible", isVisible());
-#endif
+// #endif
     if (tray)
         settings.set("TrayVisible", tray->isVisible());
     settings.set("VolumeL", volW->volumeL());
diff --git a/src/gui/MainWidget.hpp b/src/gui/MainWidget.hpp
index 2fe04bc5650a643a232ed691d34b7b1a30d8c401..33443233f83653039932634d50aaebaccb7d412a 100644
--- a/src/gui/MainWidget.hpp
+++ b/src/gui/MainWidget.hpp
@@ -108,7 +108,7 @@ private slots:
 
     void about();
 
-#if !defined Q_OS_MACOS && !defined Q_OS_ANDROID
+#if /*!defined Q_OS_MACOS &&*/ !defined Q_OS_ANDROID
     void hideMenu(bool);
 #else
     void hideMenu(bool) {} //To avoid incorrect CPP file generated by MOC on Android
@@ -180,7 +180,7 @@ private:
     AboutWidget *aboutW;
     bool isCompactView, wasShow, fullScreen, seekSFocus, m_restoreWindowOnVideo = false;
 
-#if !defined Q_OS_MACOS && !defined Q_OS_ANDROID
+#if /*!defined Q_OS_MACOS &&*/ !defined Q_OS_ANDROID
     QAction *hideMenuAct;
 #endif
     QAction *lockWidgetsAct;
diff --git a/src/gui/SettingsWidget.cpp b/src/gui/SettingsWidget.cpp
index 93960fa78518ef0e350b468772a6146839c3d1e5..3b78c6e9047b524b053bb4edb697c2b0dede7e5b 100644
--- a/src/gui/SettingsWidget.cpp
+++ b/src/gui/SettingsWidget.cpp
@@ -59,7 +59,7 @@
 #include "ui_SettingsPlayback.h"
 #include "ui_SettingsPlaybackModulesList.h"
 
-#if !defined(Q_OS_WIN) && !defined(Q_OS_MACOS) && !defined(Q_OS_ANDROID)
+#if !defined(Q_OS_WIN) /*&& !defined(Q_OS_MACOS)*/ && !defined(Q_OS_ANDROID)
     #define ICONS_FROM_THEME
 #endif
 
diff --git a/src/qmplay2/CMakeLists.txt b/src/qmplay2/CMakeLists.txt
index 108f2b39a153e4f64e56a3755c975ae2b462424a..8f9701e52b5e869d8dfe4a9ee9a1bbcdbcb1e606 100644
--- a/src/qmplay2/CMakeLists.txt
+++ b/src/qmplay2/CMakeLists.txt
@@ -220,7 +220,7 @@ if(WIN32)
     install(TARGETS ${PROJECT_NAME} RUNTIME DESTINATION ${CMAKE_INSTALL_PREFIX})
 else()
     install(TARGETS ${PROJECT_NAME} LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
-    if(NOT APPLE)
+    if(NOT APPLE_STANDALONE_BUNDLE)
         install(FILES ${QMPLAY2_HDR} DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/QMPlay2")
     endif()
 
