diff --git a/src/gui/DemuxerThr.cpp b/src/gui/DemuxerThr.cpp
index 59845226329c1954cd0eb09c17d8fa3c1e2101f0..2bb8f70a6aa6ab6df69fb283d4fcd4e6a199ccab 100644
--- a/src/gui/DemuxerThr.cpp
+++ b/src/gui/DemuxerThr.cpp
@@ -696,7 +696,7 @@ void DemuxerThr::updateCoverAndPlaying(bool doCompare)
     const QString prevTitle  = title;
     const QString prevArtist = artist;
     const QString prevAlbum  = album;
-    QString lyrics = QMPlay2Core.getDescriptionForUrl(url);
+    QString lyrics = description = QMPlay2Core.getDescriptionForUrl(url);
     title.clear();
     artist.clear();
     album.clear();
@@ -858,6 +858,19 @@ void DemuxerThr::emitInfo()
     if (demuxer->bitrate() > 0)
         info += "<b>" + tr("Bitrate") + ":</b> " + QString::number(demuxer->bitrate()) + "kbps<br/>";
     info += "<b>" + tr("Format") + ":</b> " + demuxer->name();
+    //description
+    auto desc = description.split("\n");
+    description.clear();
+    if (!desc.isEmpty()) {
+        if (!desc.isEmpty()) {
+            auto line = desc.takeFirst();
+            info += "<br><b>" + tr("Description") + ":</b> " + line;
+            while (!desc.isEmpty()) {
+                line = desc.takeFirst();
+                info += "<br>&nbsp;&nbsp;" + line;
+            }
+        }
+    }
 
     if (!demuxer->image().isNull())
         info += "<br/><br/><a href='save_cover'>" + tr("Save cover picture") + "</a>";
diff --git a/src/gui/DemuxerThr.hpp b/src/gui/DemuxerThr.hpp
index b88aa1d74322b6bda0d7190b7aa9fca6db3f867a..af27ab9370f8eafae1d488e1e9462864120c49b6 100644
--- a/src/gui/DemuxerThr.hpp
+++ b/src/gui/DemuxerThr.hpp
@@ -97,7 +97,7 @@ private:
     QMutex stopVAMutex, endMutex, seekMutex;
     IOController<> ioCtrl;
     IOController<Demuxer> demuxer;
-    QString title, artist, album;
+    QString title, artist, album, description;
     double playIfBuffered, time, updateBufferedTime;
 private slots:
     void stopVADec();
