diff --git a/src/modules/PortAudio/PortAudioWriter.cpp b/src/modules/PortAudio/PortAudioWriter.cpp
index 38dc27d9592433ca22d175779e93416fbe756276..8856be605b2b99ba67bf412e51d20c19be81a223 100644
--- a/src/modules/PortAudio/PortAudioWriter.cpp
+++ b/src/modules/PortAudio/PortAudioWriter.cpp
@@ -191,7 +191,24 @@ void PortAudioWriter::pause()
 
 QString PortAudioWriter::name() const
 {
-	return PortAudioWriterName;
+	QString name = QStringLiteral(PortAudioWriterName);
+	if (stream)
+	{
+		if (const PaDeviceInfo *dInfo = Pa_GetDeviceInfo(outputParameters.device))
+		{
+			name += QStringLiteral(" (%1").arg(dInfo->name);
+			if (const PaHostApiInfo *hInfo = Pa_GetHostApiInfo(dInfo->hostApi))
+			{
+				name += QStringLiteral("; %1").arg(hInfo->name);
+			}
+			name += QStringLiteral(")");
+		}
+		if (const PaStreamInfo *strInfo = Pa_GetStreamInfo(stream))
+		{
+			name += QStringLiteral(", %1Hz").arg(strInfo->sampleRate); 
+		}
+	}
+	return name;
 }
 
 bool PortAudioWriter::open()
@@ -282,3 +299,5 @@ void PortAudioWriter::close()
 	}
 	err = false;
 }
+
+// kate: replace-tabs false;
