diff --git a/src/modules/Extensions/YouTube.cpp b/src/modules/Extensions/YouTube.cpp
index d24da8c5..023b2154 100644
--- a/src/modules/Extensions/YouTube.cpp
+++ b/src/modules/Extensions/YouTube.cpp
@@ -20,6 +20,7 @@
 
 #include <YouTubeDL.hpp>
 #include <LineEdit.hpp>
+#include <Functions.hpp>
 
 #include <QLoggingCategory>
 #include <QStringListModel>
@@ -46,7 +47,6 @@
 
 #include <QDateTime>
 #include <QDeadlineTimer>
-#include <unistd.h>
 
 Q_LOGGING_CATEGORY(youtube, "Extensions/YouTube")
 
@@ -1304,9 +1304,9 @@ QStringList YouTube::getYouTubeVideo(const QString &param, const QString &url, I
                 urlLanguages[url] = format[QStringLiteral("language")].toString();
                 urlNotes[url] = note;
             }
-            if (format.contains("available_at"))
+            if (format.contains(QStringLiteral("available_at")))
             {
-                auto available_at = format["available_at"].toDouble() * 1000.0;
+                auto available_at = format[QStringLiteral("available_at")].toDouble() * 1000.0;
                 qCDebug(youtube) << "url for format" << format["format_id"].toString() << "will be available at"
                     << QDateTime::fromMSecsSinceEpoch(available_at) << "so in" << (available_at - QDateTime::currentMSecsSinceEpoch()) / 1000.0 << "s";
                 if (available_at > latest_available_at)
@@ -1454,22 +1454,24 @@ QStringList YouTube::getYouTubeVideo(const QString &param, const QString &url, I
 
     if (latest_available_at > 0)
     {
-        const auto dt = int( (latest_available_at - QDateTime::currentMSecsSinceEpoch()) / 1000.0 + 0.5);
+        // The number of whole seconds until the "slowest" stream is promised to be available, plus 1
+        // This interval can be <=0 if the stream is already available!
+        const auto dt = int( (latest_available_at - QDateTime::currentMSecsSinceEpoch()) / 1000.0) + 1;
         if (dt > 0) {
             QMPlay2Core.logInfo(QString::asprintf("Waiting for %ds as required by YouTube", dt));
             // do an active wait; a less active clone of QTest::qWait():
-            auto remaining = dt * 1000;
+            auto remaining = dt * 1000.0;
             QDeadlineTimer timer(remaining, Qt::PreciseTimer);
             do
             {
                 QCoreApplication::processEvents(QEventLoop::AllEvents, remaining);
-                QCoreApplication::sendPostedEvents(Q_NULLPTR, QEvent::DeferredDelete);
+                QCoreApplication::sendPostedEvents(nullptr, QEvent::DeferredDelete);
                 remaining = timer.remainingTime();
                 if (remaining <= 0)
                 {
                     break;
                 }
-                usleep(qMin(100, remaining) * 1000);
+                Functions::s_wait(qMin(100.0, remaining) / 1000.0);
                 remaining = timer.remainingTime();
             } while (remaining > 0);
         }
