diff --git b/src/qmplay2/Functions.cpp a/src/qmplay2/Functions.cpp
index 57a653a8..4cf5c202 100644
--- b/src/qmplay2/Functions.cpp
+++ a/src/qmplay2/Functions.cpp
@@ -1110,7 +1110,7 @@ QString Functions::getSeqFile(const QString &dir, const QString &ext, const QStr
     quint16 num = 0;
     for (const QString &f : QDir(dir).entryList({QString("QMPlay2_%1_?????%2").arg(frag, ext)}, QDir::Files, QDir::Name))
     {
-        const quint16 n = QStringView(f).mid(8 + frag.size() + 1, 5)MAYBE_TO_STRING.toUShort();
+        const quint16 n = QStringView(f).mid(8 + frag.size() + 1, 5).toUShort();
         if (n > num)
             num = n;
     }
