diff --git a/src/qmplay2/NetworkAccess.cpp b/src/qmplay2/NetworkAccess.cpp
index 4cf405e73af5e6b57e480d3e6033d745aaedbe3e..4b43ce0d6138181737deaaa30d9f122235403eaa 100644
--- a/src/qmplay2/NetworkAccess.cpp
+++ b/src/qmplay2/NetworkAccess.cpp
@@ -65,7 +65,7 @@ public:
         m_networkReply(networkReply),
         m_url(url),
         m_postData(postData),
-        m_rawHeaders(rawHeaders),
+        m_rawHeaders(rawHeaders + "Accept-Language: " + QMPlay2Core.getLanguage().toLatin1() + "\r\n" ),
         m_customUserAgent(params.customUserAgent),
         m_maxSize(params.maxSize),
         m_retries(params.retries),
