diff --git a/CMakeLists.txt b/CMakeLists.txt
index d8561fd5..d21b02b4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -42,14 +42,27 @@ if(BUILD_WITH_QT6)
     find_package(Qt6Core5Compat 6.1 REQUIRED)
 else()
     set(QT_PREFIX Qt5)
-    find_package(Qt5Widgets 5.10 REQUIRED)
+    if (NOT QT5_MIN_VERSION)
+        set(QT5_MIN_VERSION 5.10)
+    endif()
+    find_package(Qt5Widgets ${QT5_MIN_VERSION} REQUIRED)
 endif()
 
-pkg_check_modules(LIBAVFORMAT REQUIRED libavformat>=58.12.100)
-pkg_check_modules(LIBAVCODEC REQUIRED libavcodec>=58.18.100)
-pkg_check_modules(LIBSWSCALE REQUIRED libswscale>=5.1.100)
-pkg_check_modules(LIBAVUTIL REQUIRED libavutil>=56.14.100)
-pkg_check_modules(LIBSWRESAMPLE REQUIRED libswresample>=3.1.100)
+option(BUILD_WITH_FFMPEG6 "Build with FFmpeg 6+" OFF)
+if(BUILD_WITH_FFMPEG6)
+	pkg_check_modules(LIBAVFORMAT REQUIRED libavformat>=60.00.000)
+	pkg_check_modules(LIBAVCODEC REQUIRED libavcodec>=60.00.000)
+	pkg_check_modules(LIBSWSCALE REQUIRED libswscale>=7.5.000)
+	pkg_check_modules(LIBAVUTIL REQUIRED libavutil>=58.00.000)
+	pkg_check_modules(LIBSWRESAMPLE REQUIRED libswresample>=4.0.000)
+else()
+	# forcing a build against FFmpeg 4 may require setting a ceiling in the future?
+	pkg_check_modules(LIBAVFORMAT REQUIRED libavformat>=58.12.100)
+	pkg_check_modules(LIBAVCODEC REQUIRED libavcodec>=58.18.100)
+	pkg_check_modules(LIBSWSCALE REQUIRED libswscale>=5.1.100)
+	pkg_check_modules(LIBAVUTIL REQUIRED libavutil>=56.14.100)
+	pkg_check_modules(LIBSWRESAMPLE REQUIRED libswresample>=3.1.100)
+endif()
 
 set(CMAKE_LINK_DEPENDS_NO_SHARED ON)
 set(CMAKE_INCLUDE_CURRENT_DIR ON)
@@ -299,7 +309,7 @@ option(USE_PIPEWIRE "Build with PipeWire module" ${LIBPIPEWIRE_FOUND})
 add_feature_info(PipeWire USE_PIPEWIRE "Build with PipeWire module")
 
 if(NOT DEFINED USE_XVIDEO)
-    pkg_check_modules(LIB_X11_XV x11 xv)
+    pkg_check_modules(LIB_X11_XV x11 xv xext)
 endif()
 option(USE_XVIDEO "Build with XVideo module" ${LIB_X11_XV_FOUND})
 add_feature_info(XVideo USE_XVIDEO "Build with XVideo module")
@@ -336,6 +346,9 @@ if(USE_GIT_VERSION)
 endif()
 add_feature_info("Git version" QMPLAY2_GIT_HEAD "Append Git HEAD to QMPlay2 version")
 
+option(USE_HIDDEN_VISIBILITY "Use selective export control in shared libraries" ON)
+add_feature_info(VISIBILITY USE_HIDDEN_VISIBILITY "Use selective export control in shared libraries")
+
 if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" OR CMAKE_CXX_COMPILER_ID MATCHES "Clang") # GCC or Clang
     option(USE_LINK_TIME_OPTIMIZATION "Enable link time optimization for release builds" OFF)
     add_feature_info("Link Time Optimization" USE_LINK_TIME_OPTIMIZATION "Enable link time optimization for release builds")
@@ -358,9 +371,19 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" OR CMAKE_CXX_COMPILER_ID MATCHES "Clang"
     endif()
 
     # Hide symbols, and also make sure that debugging symbols are disabled
-    set(CMAKE_CXX_FLAGS_RELEASE    "${RELEASE_DEBUG_FLAG} -fvisibility=hidden -fvisibility-inlines-hidden ${CMAKE_CXX_FLAGS_RELEASE}")
-    set(CMAKE_CXX_FLAGS_MINSIZEREL "${RELEASE_DEBUG_FLAG} -fvisibility=hidden -fvisibility-inlines-hidden ${CMAKE_CXX_FLAGS_MINSIZEREL}")
+    if(USE_HIDDEN_VISIBILITY)
+        set(CMAKE_CXX_VISIBILITY_PRESET hidden)
+    else()
+        # be sure to reset these so code can be rebuilt without the option
+        set(CMAKE_CXX_VISIBILITY_PRESET default)
+    endif()
+    # this one can always be enabled:
+    set(CMAKE_VISIBILITY_INLINES_HIDDEN 1)
+    set(CMAKE_CXX_FLAGS            "${CMAKE_CXX_FLAGS}")
+    set(CMAKE_CXX_FLAGS_RELEASE    "${RELEASE_DEBUG_FLAG} ${CMAKE_CXX_FLAGS_RELEASE}")
+    set(CMAKE_CXX_FLAGS_MINSIZEREL "${RELEASE_DEBUG_FLAG} ${CMAKE_CXX_FLAGS_MINSIZEREL}")
     if(NOT USE_VULKAN)
+        set(CMAKE_CXX_FLAGS            "-fno-exceptions ${CMAKE_CXX_FLAGS}")
         set(CMAKE_CXX_FLAGS_RELEASE    "-fno-exceptions ${CMAKE_CXX_FLAGS_RELEASE}")
         set(CMAKE_CXX_FLAGS_MINSIZEREL "-fno-exceptions ${CMAKE_CXX_FLAGS_MINSIZEREL}")
     endif()
@@ -380,8 +403,9 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" OR CMAKE_CXX_COMPILER_ID MATCHES "Clang"
     endif()
 
     if(USE_LINK_TIME_OPTIMIZATION)
+        set(CMAKE_CXX_FLAGS            "-flto ${CMAKE_CXX_FLAGS}")
         set(CMAKE_CXX_FLAGS_RELEASE    "-flto ${CMAKE_CXX_FLAGS_RELEASE}")
-        set(CMAKE_CXX_FLAGS_MINSIZEREL "-flto ${CMAKE_CXX_FLAGS_RELEASE}")
+        set(CMAKE_CXX_FLAGS_MINSIZEREL "-flto ${CMAKE_CXX_FLAGS_MINSIZEREL}")
     endif()
 
     if(CMAKE_MODULE_LINKER_FLAGS)
@@ -395,19 +419,22 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" OR CMAKE_CXX_COMPILER_ID MATCHES "Clang"
 endif()
 
 if(APPLE)
-    set(MAC_BUNDLE_PATH "${CMAKE_INSTALL_PREFIX}/QMPlay2.app")
+    set(MAC_BUNDLE_PATH "${BUNDLE_INSTALL_DIR}/QMPlay2.app")
 
     # set paths to Bundle
     set(CMAKE_INSTALL_DATAROOTDIR "${MAC_BUNDLE_PATH}/Contents/share")
     set(CMAKE_INSTALL_LIBDIR      "${MAC_BUNDLE_PATH}/Contents/MacOS")
+    if(NOT APPLE_STANDALONE_BUNDLE)
+        include(GNUInstallDirs)
+    endif()
 elseif(NOT WIN32)
     include(GNUInstallDirs)
 endif()
 
-if(NOT APPLE)
+# if(NOT APPLE)
     set(INSTALL_PATH_ICONS "${CMAKE_INSTALL_DATAROOTDIR}/icons")
     set(INSTALL_PATH_MIME "${CMAKE_INSTALL_DATAROOTDIR}/mime")
-endif()
+# endif()
 
 if(CMAKE_SYSTEM_NAME STREQUAL "FreeBSD" AND CMAKE_INSTALL_PREFIX STREQUAL "/usr/local")
     set(INSTALL_PATH_MAN "man") # exception for FreeBSD
@@ -463,7 +490,7 @@ else()
     install(FILES AUTHORS ChangeLog LICENSE README.md DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/qmplay2")
 endif()
 
-if(NOT APPLE AND NOT WIN32)
+# if(NOT APPLE AND NOT WIN32)
     # uninstall target
     configure_file(
         "${CMAKE_CURRENT_SOURCE_DIR}/src/cmake/Uninstall.cmake.in"
@@ -472,7 +499,7 @@ if(NOT APPLE AND NOT WIN32)
 
     add_custom_target(uninstall
         COMMAND ${CMAKE_COMMAND} -P "${CMAKE_CURRENT_BINARY_DIR}/cmake_uninstall.cmake")
-endif()
+# endif()
 
 feature_summary(WHAT ENABLED_FEATURES DESCRIPTION "Enabled features:")
 feature_summary(WHAT DISABLED_FEATURES DESCRIPTION "Disabled features:")
diff --git a/src/gui/AboutWidget.cpp b/src/gui/AboutWidget.cpp
index 1251da01..2848f34e 100644
--- a/src/gui/AboutWidget.cpp
+++ b/src/gui/AboutWidget.cpp
@@ -45,7 +45,7 @@ AboutWidget::AboutWidget()
     labelText += "<b>QMPlay2:</b> " + tr("video and audio player");
     labelText += "<br/><b>" + tr("Programmer") + ":</b> <a href='mailto:mumei6102@gmail.com'>Błażej Szczygieł</a>";
     labelText += "<br/><b>" + tr("Version") + ":</b> " + Version::get() + " (" + QSysInfo::buildCpuArchitecture();
-#if !defined(Q_OS_MACOS) && !defined(Q_OS_WIN)
+#if !defined(Q_OS_WIN)
     labelText += ", " + QGuiApplication::platformName();
 #endif
     labelText += ")";
diff --git a/src/gui/CMakeLists.txt b/src/gui/CMakeLists.txt
index c5abe55c6c3ca0f4fa6cca121a16e6f9ccbe2b48..82d716da80f466ce215281f5a011660d04ce122c 100644
--- a/src/gui/CMakeLists.txt
+++ b/src/gui/CMakeLists.txt
@@ -4,6 +4,9 @@ project(QMPlay2)
 if(WIN32 AND NOT BUILD_WITH_QT6)
     find_package(Qt5WinExtras 5.10 REQUIRED)
 endif()
+if(USE_XVIDEO)
+    add_definitions(-DHAVE_X11)
+endif()
 
 set(GUI_HDR
     Main.hpp
@@ -263,51 +266,54 @@ endif()
 if(WIN32)
     install(TARGETS ${PROJECT_NAME} RUNTIME DESTINATION ${CMAKE_INSTALL_PREFIX}/)
 elseif(APPLE)
-    install(TARGETS ${PROJECT_NAME} BUNDLE DESTINATION ${CMAKE_INSTALL_PREFIX})
+    install(TARGETS ${PROJECT_NAME} BUNDLE DESTINATION ${BUNDLE_INSTALL_DIR})
 
     set(QT_LIBS_DIR "${${QT_PREFIX}Widgets_DIR}/../..")
     set(QT_PLUGINS_DIR "${QT_LIBS_DIR}/../plugins")
-    install(FILES
-        "${QT_PLUGINS_DIR}/platforms/libqcocoa.dylib"
-        DESTINATION "${MAC_BUNDLE_PATH}/Contents/plugins/platforms")
-    install(FILES
-        "${QT_PLUGINS_DIR}/iconengines/libqsvgicon.dylib"
-        DESTINATION "${MAC_BUNDLE_PATH}/Contents/plugins/iconengines")
-    install(FILES
-        "${QT_PLUGINS_DIR}/styles/libqmacstyle.dylib"
-        DESTINATION "${MAC_BUNDLE_PATH}/Contents/plugins/styles"
-        OPTIONAL)
-    install(DIRECTORY
-        "${QT_PLUGINS_DIR}/imageformats"
-        DESTINATION "${MAC_BUNDLE_PATH}/Contents/plugins"
-        PATTERN "*_debug.dylib" EXCLUDE
-        PATTERN "*.dSYM" EXCLUDE)
-    install(DIRECTORY
-        "${QT_LIBS_DIR}/../translations"
-        DESTINATION "${MAC_BUNDLE_PATH}/Contents"
-        FILES_MATCHING
-        PATTERN "qtbase_*.qm")
-    if(EXISTS "/usr/local/bin/ffmpeg")
-        install(PROGRAMS
-            "/usr/local/bin/ffmpeg"
-            DESTINATION "${MAC_BUNDLE_PATH}/Contents/MacOS")
-    else()
-        message(WARNING "FFmpeg executable not copied!")
+    if(APPLE_STANDALONE_BUNDLE)
+        install(FILES
+            "${QT_PLUGINS_DIR}/platforms/libqcocoa.dylib"
+            DESTINATION "${MAC_BUNDLE_PATH}/Contents/plugins/platforms")
+        install(FILES
+            "${QT_PLUGINS_DIR}/iconengines/libqsvgicon.dylib"
+            DESTINATION "${MAC_BUNDLE_PATH}/Contents/plugins/iconengines")
+        install(FILES
+            "${QT_PLUGINS_DIR}/styles/libqmacstyle.dylib"
+            DESTINATION "${MAC_BUNDLE_PATH}/Contents/plugins/styles"
+            OPTIONAL)
+        install(DIRECTORY
+            "${QT_PLUGINS_DIR}/imageformats"
+            DESTINATION "${MAC_BUNDLE_PATH}/Contents/plugins"
+            PATTERN "*_debug.dylib" EXCLUDE
+            PATTERN "*.dSYM" EXCLUDE)
+        install(DIRECTORY
+            "${QT_LIBS_DIR}/../translations"
+            DESTINATION "${MAC_BUNDLE_PATH}/Contents"
+            FILES_MATCHING
+            PATTERN "qtbase_*.qm")
+        if(EXISTS "/usr/local/bin/ffmpeg")
+            install(PROGRAMS
+                "/usr/local/bin/ffmpeg"
+                DESTINATION "${MAC_BUNDLE_PATH}/Contents/MacOS")
+        else()
+            message(WARNING "FFmpeg executable not copied!")
+        endif()
+        install(CODE "
+            include(BundleUtilities)
+            set(BU_CHMOD_BUNDLE_ITEMS ON)
+            list(APPEND DIRS /usr/local/lib ${QT_LIBS_DIR})
+            file(GLOB_RECURSE QMPLAY2_MODULES_AND_QT_PLUGINS
+                \"${MAC_BUNDLE_PATH}/Contents/MacOS/modules/*\"
+                \"${MAC_BUNDLE_PATH}/Contents/plugins/*.dylib\")
+            file(WRITE \"${MAC_BUNDLE_PATH}/Contents/Resources/qt.conf\")
+            fixup_bundle(${MAC_BUNDLE_PATH} \"\${QMPLAY2_MODULES_AND_QT_PLUGINS}\" \"\${DIRS}\")
+        ")
     endif()
-    install(CODE "
-        include(BundleUtilities)
-        set(BU_CHMOD_BUNDLE_ITEMS ON)
-        list(APPEND DIRS /usr/local/lib ${QT_LIBS_DIR})
-        file(GLOB_RECURSE QMPLAY2_MODULES_AND_QT_PLUGINS
-            \"${MAC_BUNDLE_PATH}/Contents/MacOS/modules/*\"
-            \"${MAC_BUNDLE_PATH}/Contents/plugins/*.dylib\")
-        file(WRITE \"${MAC_BUNDLE_PATH}/Contents/Resources/qt.conf\")
-        fixup_bundle(${MAC_BUNDLE_PATH} \"\${QMPLAY2_MODULES_AND_QT_PLUGINS}\" \"\${DIRS}\")
-    ")
 elseif(NOT ANDROID)
     # executable
     install(TARGETS ${PROJECT_NAME} RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
-
+endif()
+if ((APPLE OR UNIX) AND NOT ANDROID)
     # desktop files
     file(GLOB DESKTOP_FILES Unix/QMPlay2*.desktop)
     install(FILES ${DESKTOP_FILES} DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/applications")
diff --git a/src/gui/MainWidget.cpp b/src/gui/MainWidget.cpp
index 7408590c58ae7b30136f6fd8ea9d214bd9a3d73e..08090e1b3a40f1e886e1b1a0fd3562cd3da9227f 100644
--- a/src/gui/MainWidget.cpp
+++ b/src/gui/MainWidget.cpp
@@ -96,7 +96,7 @@ public:
     }
 };
 
-#ifndef Q_OS_MACOS
+// #ifndef Q_OS_MACOS
 static void copyMenu(QMenu *dest, QMenu *src, const QSet<QMenu *> &dontCopy = {}, const QSet<QMenu *> &dontAdd = {})
 {
     QMenu *newMenu = new QMenu(src->title(), dest);
@@ -131,7 +131,7 @@ static void copyMenu(QMenu *dest, QMenu *src, const QSet<QMenu *> &dontCopy = {}
     }
     dest->addMenu(newMenu);
 }
-#endif
+// #endif
 
 /* MainWidget */
 MainWidget::MainWidget(QList<QPair<QString, QString>> &arguments)
@@ -174,7 +174,7 @@ MainWidget::MainWidget(QList<QPair<QString, QString>> &arguments)
 
     QMPlay2GUI.menuBar = new MenuBar;
 
-#if !defined Q_OS_MACOS && !defined Q_OS_ANDROID
+#if /*!defined Q_OS_MACOS &&*/ !defined Q_OS_ANDROID
     tray = new QSystemTrayIcon(this);
     tray->setIcon(QMPlay2Core.getIconFromTheme("QMPlay2-panel", QMPlay2Core.getQMPlay2Icon()));
     tray->setVisible(settings.getBool("TrayVisible", true));
@@ -367,7 +367,7 @@ MainWidget::MainWidget(QList<QPair<QString, QString>> &arguments)
     if (settings.getBool("MainWidget/TabPositionNorth"))
         setTabPosition(Qt::AllDockWidgetAreas, QTabWidget::North);
 
-#if !defined Q_OS_MACOS && !defined Q_OS_ANDROID
+#if /*!defined Q_OS_MACOS &&*/ !defined Q_OS_ANDROID
     const bool menuHidden = settings.getBool("MainWidget/MenuHidden", false);
     menuBar->setVisible(!menuHidden);
     hideMenuAct = new QAction(tr("&Hide menu bar"), menuBar);
@@ -432,7 +432,7 @@ MainWidget::MainWidget(QList<QPair<QString, QString>> &arguments)
         setAttribute(Qt::WA_NoSystemBackground, false);
     }
 
-#if defined Q_OS_MACOS || defined Q_OS_ANDROID
+#if /*defined Q_OS_MACOS ||*/ defined Q_OS_ANDROID
     show();
 #else
     setVisible(settings.getBool("MainWidget/isVisible", true) ? true : !isTrayVisible());
@@ -1001,7 +1001,7 @@ void MainWidget::createMenuBar()
 
     setMenuBar(menuBar);
 
-#ifndef Q_OS_MACOS
+#ifndef Q_OS_MACOS_FOR_REAL
     if (tray)
     {
         auto secondMenu = new QMenu(this);
@@ -1098,7 +1098,7 @@ void MainWidget::toggleCompactView()
 
         hideAllExtensions();
 
-#if !defined Q_OS_MACOS && !defined Q_OS_ANDROID
+#if /*!defined Q_OS_MACOS &&*/ !defined Q_OS_ANDROID
         menuBar->hide();
 #endif
         mainTB->hide();
@@ -1121,7 +1121,7 @@ void MainWidget::toggleCompactView()
         restoreState(dockWidgetState);
         dockWidgetState.clear();
 
-#if !defined Q_OS_MACOS && !defined Q_OS_ANDROID
+#if /*!defined Q_OS_MACOS &&*/ !defined Q_OS_ANDROID
         menuBar->setVisible(!hideMenuAct->isChecked());
 #endif
 
@@ -1147,7 +1147,7 @@ void MainWidget::toggleFullScreen()
 #ifndef Q_OS_ANDROID
     static bool maximized;
 #endif
-#ifdef Q_OS_MACOS
+#if defined(Q_OS_MACOS) && !defined(HAVE_X11)
     if (isFullScreen())
     {
         showNormal();
@@ -1164,7 +1164,7 @@ void MainWidget::toggleFullScreen()
 #ifndef Q_OS_ANDROID
         maximized = isMaximized();
 
-#ifndef Q_OS_MACOS
+#if !defined(Q_OS_MACOS) || defined(HAVE_X11)
 #ifndef Q_OS_WIN
         if (isFullScreen())
 #endif
@@ -1178,9 +1178,9 @@ void MainWidget::toggleFullScreen()
         dockWidgetState = saveState();
 #endif // Q_OS_ANDROID
 
-#if !defined Q_OS_MACOS
+// #if !defined Q_OS_MACOS
         menuBar->hide();
-#endif
+// #endif
         statusBar->hide();
 
         mainTB->hide();
@@ -1207,14 +1207,14 @@ void MainWidget::toggleFullScreen()
         videoDock->fullScreen(true);
         videoDock->show();
 
-#ifdef Q_OS_MACOS
+// #if defined(Q_OS_MACOS)
         menuBar->window->toggleVisibility->setEnabled(false);
-#endif
+// #endif
         menuBar->window->toggleCompactView->setEnabled(false);
         menuBar->window->toggleFullScreen->setShortcuts(QList<QKeySequence>() << menuBar->window->toggleFullScreen->shortcut() << QKeySequence("ESC"));
         fullScreen = true;
 
-#ifndef Q_OS_MACOS
+#if !defined(Q_OS_MACOS) || defined(HAVE_X11)
         showFullScreen();
 #else
         const auto geo = window()->windowHandle()->screen()->geometry();
@@ -1230,9 +1230,9 @@ void MainWidget::toggleFullScreen()
     }
     else
     {
-#ifdef Q_OS_MACOS
+// #if defined(Q_OS_MACOS)
         menuBar->window->toggleVisibility->setEnabled(true);
-#endif
+// #endif
         menuBar->window->toggleCompactView->setEnabled(true);
         menuBar->window->toggleFullScreen->setShortcuts(QList<QKeySequence>() << menuBar->window->toggleFullScreen->shortcut());
 
@@ -1251,7 +1251,7 @@ void MainWidget::toggleFullScreen()
             showMaximized();
         else
         {
-#ifdef Q_OS_MACOS
+#if defined(Q_OS_MACOS) && !defined(HAVE_X11)
             showNormal();
 #endif
             setGeometry(savedGeo);
@@ -1279,7 +1279,7 @@ void MainWidget::toggleFullScreen()
 
 #if defined(Q_OS_ANDROID)
         menuBar->setVisible(true);
-#elif !defined Q_OS_MACOS
+#else //if !defined Q_OS_MACOS
         menuBar->setVisible(!hideMenuAct->isChecked());
 #endif
         statusBar->show();
@@ -1512,7 +1512,7 @@ void MainWidget::about()
     }
 }
 
-#if !defined Q_OS_MACOS && !defined Q_OS_ANDROID
+// #if !defined Q_OS_MACOS && !defined Q_OS_ANDROID
 void MainWidget::hideMenu(bool h)
 {
     if (fullScreen || isCompactView)
@@ -1523,7 +1523,7 @@ void MainWidget::hideMenu(bool h)
         QMPlay2Core.getSettings().set("MainWidget/MenuHidden", h);
     }
 }
-#endif
+// #endif
 void MainWidget::lockWidgets(bool l)
 {
     if (fullScreen || isCompactView)
@@ -1656,7 +1656,7 @@ QMenu *MainWidget::createPopupMenu()
     QMenu *popupMenu = QMainWindow::createPopupMenu();
     if (!fullScreen && !isCompactView)
     {
-#if !defined Q_OS_MACOS && !defined Q_OS_ANDROID
+#if /*!defined Q_OS_MACOS &&*/ !defined Q_OS_ANDROID
         popupMenu->insertAction(popupMenu->actions().value(0), hideMenuAct);
         popupMenu->insertSeparator(popupMenu->actions().value(1));
         popupMenu->addSeparator();
@@ -1901,6 +1901,16 @@ void MainWidget::mouseMoveEvent(QMouseEvent *e)
         {
             showToolBar(true); //Before restoring dock widgets - show toolbar and status bar
 
+            // FullScreenDockWidgetState has been known to become corrupted, which can cause
+            // Qt to get stuck in an infinite loop trying to restore layout of the videoDock. 
+            // This can lead to a crash or abort. Prevent this in 2 ways:
+            // 1) don't do our mouseMove event processing during the call to restoreState()
+            //    There's only ever a single MainWidget instance so we just use a local static.
+            // 2) by clearing the setting here. If all goes well, the value we will be using
+            //    now will be saved again. If not, we'll have gotten rid of the corrupt value.
+            auto &settings = QMPlay2Core.getSettings();
+            settings.remove("MainWidget/FullScreenDockWidgetState");
+            settings.flush();
             inRestoreState = true;
             restoreState(fullScreenDockWidgetState);
             inRestoreState = false;
diff --git a/src/gui/MainWidget.hpp b/src/gui/MainWidget.hpp
index f7fa92f4..357ba97a 100644
--- a/src/gui/MainWidget.hpp
+++ b/src/gui/MainWidget.hpp
@@ -110,7 +110,7 @@ private slots:
 
     void about();
 
-#if !defined Q_OS_MACOS && !defined Q_OS_ANDROID
+#if /*!defined Q_OS_MACOS &&*/ !defined Q_OS_ANDROID
     void hideMenu(bool);
 #else
     void hideMenu(bool) {} //To avoid incorrect CPP file generated by MOC on Android
@@ -190,7 +190,7 @@ private:
     bool m_compactViewBeforeFullScreen = false;
     bool m_loaded = false;
 
-#if !defined Q_OS_MACOS && !defined Q_OS_ANDROID
+#if /*!defined Q_OS_MACOS &&*/ !defined Q_OS_ANDROID
     QAction *hideMenuAct;
 #endif
     QAction *lockWidgetsAct;
diff --git a/src/gui/SettingsWidget.cpp b/src/gui/SettingsWidget.cpp
index f5819896..098c886b 100644
--- a/src/gui/SettingsWidget.cpp
+++ b/src/gui/SettingsWidget.cpp
@@ -68,7 +68,7 @@
 #include "ui_SettingsPlayback.h"
 #include "ui_SettingsPlaybackModulesList.h"
 
-#if !defined(Q_OS_WIN) && !defined(Q_OS_MACOS) && !defined(Q_OS_ANDROID)
+#if !defined(Q_OS_WIN) /*&& !defined(Q_OS_MACOS)*/ && !defined(Q_OS_ANDROID)
     #define ICONS_FROM_THEME
 #endif
 
diff --git a/src/modules/FFmpeg/CMakeLists.txt b/src/modules/FFmpeg/CMakeLists.txt
index 1a4754eb..0ec10e0b 100644
--- a/src/modules/FFmpeg/CMakeLists.txt
+++ b/src/modules/FFmpeg/CMakeLists.txt
@@ -1,6 +1,9 @@
 cmake_minimum_required(VERSION 3.12)
 project(FFmpeg)
 
+# string(REPLACE "-fvisibility=hidden" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
+# string(REPLACE "-fvisibility-inlines-hidden" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
+
 set(FFmpeg_HDR
     FFmpeg.hpp
     FFDemux.hpp
diff --git a/src/modules/FFmpeg/FFDecVTB.cpp b/src/modules/FFmpeg/FFDecVTB.cpp
index 3c764475..eb29485c 100644
--- a/src/modules/FFmpeg/FFDecVTB.cpp
+++ b/src/modules/FFmpeg/FFDecVTB.cpp
@@ -26,6 +26,7 @@
 #   include <VTBOpenGL.hpp>
 #endif
 
+#include <QGuiApplication>
 #include <QOperatingSystemVersion>
 #include <QDebug>
 
@@ -100,6 +101,11 @@ bool FFDecVTB::open(StreamInfo &streamInfo)
 {
     if (streamInfo.params->codec_type != AVMEDIA_TYPE_VIDEO || !hasHWAccel("videotoolbox"))
         return false;
+    if (QGuiApplication::platformName() == "xcb" && qgetenv("QT_XCB_GL_INTEGRATION") == "xcb_egl")
+    {
+        qWarning() << Q_FUNC_INFO << "VTB cannot be used with EGL under X11";
+        return false;
+    }
 
     const AVPixelFormat pix_fmt = streamInfo.pixelFormat();
     if (pix_fmt == AV_PIX_FMT_YUV420P10)
diff --git a/src/modules/FFmpeg/VTBOpenGL.cpp b/src/modules/FFmpeg/VTBOpenGL.cpp
index 97ad0fe0..a53a5247 100644
--- a/src/modules/FFmpeg/VTBOpenGL.cpp
+++ b/src/modules/FFmpeg/VTBOpenGL.cpp
@@ -22,6 +22,7 @@
 
 #include <QOpenGLContext>
 #include <QImage>
+#include <QLoggingCategory>
 
 extern "C"
 {
@@ -105,12 +106,17 @@ bool VTBOpenGL::mapFrame(Frame &videoFrame)
             internalFormat1 = GL_RG8;
             type = GL_UNSIGNED_BYTE;
             break;
+#if defined(__MAC_10_13) && defined(__MAC_OS_X_VERSION_MAX_ALLOWED) && __MAC_OS_X_VERSION_MAX_ALLOWED >= __MAC_10_13
         case kCVPixelFormatType_420YpCbCr10BiPlanarVideoRange:
+#else
+        case 'x420':
+#endif
             internalFormat0 = GL_R16;
             internalFormat1 = GL_RG16;
             type = GL_UNSIGNED_SHORT;
             break;
         default:
+            qWarning() << Q_FUNC_INFO << "Unsupported pixelFormat" << pixelFormat;
             m_error = true;
             return false;
     }
diff --git a/src/modules/XVideo/CMakeLists.txt b/src/modules/XVideo/CMakeLists.txt
index 32da2f99..68560102 100644
--- a/src/modules/XVideo/CMakeLists.txt
+++ b/src/modules/XVideo/CMakeLists.txt
@@ -17,7 +17,7 @@ set(XVideo_RESOURCES
     icon.qrc
 )
 
-pkg_check_modules(LIB_X11_XV REQUIRED x11 xv)
+pkg_check_modules(LIB_X11_XV REQUIRED x11 xv xext)
 link_directories(${LIB_X11_XV_LIBRARY_DIRS})
 
 include_directories(${LIB_X11_XV_INCLUDE_DIRS})
diff --git a/src/qmplay2/CMakeLists.txt b/src/qmplay2/CMakeLists.txt
index babb2d03..e1c29a1b 100644
--- a/src/qmplay2/CMakeLists.txt
+++ b/src/qmplay2/CMakeLists.txt
@@ -347,7 +347,7 @@ if(WIN32)
     install(TARGETS ${PROJECT_NAME} RUNTIME DESTINATION ${CMAKE_INSTALL_PREFIX})
 else()
     install(TARGETS ${PROJECT_NAME} LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
-    if(NOT APPLE)
+    if(NOT APPLE_STANDALONE_BUNDLE)
         install(FILES ${QMPLAY2_HDR} DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/QMPlay2")
     endif()
 endif()
