diff --git a/src/modules/Modplug/libmodplug/load_j2b.cpp b/src/modules/Modplug/libmodplug/load_j2b.cpp
index 8de88b5f..a1375e01 100644
--- a/src/modules/Modplug/libmodplug/load_j2b.cpp
+++ b/src/modules/Modplug/libmodplug/load_j2b.cpp
@@ -14,6 +14,7 @@
 #include "stdafx.hpp"
 #include "sndfile.hpp"
 
+#include <limits>
 #include <QByteArray>
 #include <QtEndian>
 
diff --git a/src/gui/macOS/QMPlay2MacExtensions.mm b/src/gui/macOS/QMPlay2MacExtensions.mm
index 02183c73..2f20222a 100644
--- a/src/gui/macOS/QMPlay2MacExtensions.mm
+++ b/src/gui/macOS/QMPlay2MacExtensions.mm
@@ -1,14 +1,17 @@
 #include "QMPlay2MacExtensions.hpp"
 
+#ifdef __clang__
 #include <QAbstractNativeEventFilter>
 #include <QGuiApplication>
 #include <QWindow>
+#endif
 
 #include <IOKit/hidsystem/ev_keymap.h>
 #include <AppKit/NSApplication.h>
 #include <AppKit/NSScreen.h>
 #include <AppKit/NSEvent.h>
 
+#ifdef __clang__
 class MediaKeysFilter : public QAbstractNativeEventFilter
 {
 public:
@@ -63,6 +66,7 @@ private:
 
     QMPlay2MacExtensions::MediaKeysCallback m_mediaKeysCallback;
 } static *g_mediaKeysFilter;
+#endif
 
 /**/
 
@@ -77,25 +81,29 @@ void QMPlay2MacExtensions::setApplicationVisible(bool visible)
 
 void QMPlay2MacExtensions::registerMacOSMediaKeys(const MediaKeysCallback &cb)
 {
+#ifdef __clang__
     if (!g_mediaKeysFilter)
     {
         g_mediaKeysFilter = new MediaKeysFilter(cb);
         QCoreApplication::instance()->installNativeEventFilter(g_mediaKeysFilter);
     }
+#endif
 }
 void QMPlay2MacExtensions::unregisterMacOSMediaKeys()
 {
+#ifdef __clang__
     if (g_mediaKeysFilter)
     {
         QCoreApplication::instance()->removeNativeEventFilter(g_mediaKeysFilter);
         delete g_mediaKeysFilter;
         g_mediaKeysFilter = nullptr;
     }
+#endif
 }
 
 void QMPlay2MacExtensions::showSystemUi(QWindow *mainWindow, bool visible)
 {
-#if defined(MAC_OS_X_VERSION_10_9) && (MAC_OS_X_VERSION_MIN_REQUIRED >= MAC_OS_X_VERSION_10_9)
+#if defined(MAC_OS_X_VERSION_10_9) && (MAC_OS_X_VERSION_MIN_REQUIRED >= MAC_OS_X_VERSION_10_9) && defined(__clang__)
     if (mainWindow && mainWindow->screen() != QGuiApplication::primaryScreen() && ![NSScreen screensHaveSeparateSpaces])
         return;
 #endif
diff --git a/CMakeLists.txt b/CMakeLists.txt
index bd417b1e..cf4c0e6d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,9 @@
 cmake_minimum_required(VERSION 3.13)
-project(QMPlay2)
+if(APPLE)
+    project(QMPlay2 LANGUAGES C CXX OBJCXX)
+else()
+    project(QMPlay2)
+endif()
 
 if(WIN32)
     set(CUSTOM_INCLUDE_DIR "" CACHE STRING "Custom include directory")
