diff --git a/src/modules/FFmpeg/FFDecVTB.hpp b/src/modules/FFmpeg/FFDecVTB.hpp
index d47eede2..1d542e24 100644
--- a/src/modules/FFmpeg/FFDecVTB.hpp
+++ b/src/modules/FFmpeg/FFDecVTB.hpp
@@ -20,7 +20,7 @@
 
 #include <FFDecHWAccel.hpp>
 
-class FFMPEGVTB_EXPORT FFDecVTB final : public FFDecHWAccel
+class FFMPEG_EXPORT FFDecVTB final : public FFDecHWAccel
 {
 public:
     FFDecVTB(Module &module);
