diff --git a/src/gui/PlayClass.cpp b/src/gui/PlayClass.cpp
index ef40ca31..0930b67d 100644
--- a/src/gui/PlayClass.cpp
+++ b/src/gui/PlayClass.cpp
@@ -275,14 +275,14 @@ void PlayClass::seek(double pos, bool allowAccurate)
 void PlayClass::chStream(const QString &s)
 {
     if (s.startsWith("audio"))
-        chosenAudioStream = QStringView(s).right(s.length() - 5)MAYBE_TO_STRING.toInt();
+        chosenAudioStream = QStringView(s).right(s.length() - 5).toInt();
     else if (s.startsWith("video"))
-        chosenVideoStream = QStringView(s).right(s.length() - 5)MAYBE_TO_STRING.toInt();
+        chosenVideoStream = QStringView(s).right(s.length() - 5).toInt();
     else if (s.startsWith("subtitles"))
-        chosenSubtitlesStream = QStringView(s).right(s.length() - 9)MAYBE_TO_STRING.toInt();
+        chosenSubtitlesStream = QStringView(s).right(s.length() - 9).toInt();
     else if (s.startsWith("fileSubs"))
     {
-        int idx = QStringView(s).right(s.length() - 8)MAYBE_TO_STRING.toInt();
+        int idx = QStringView(s).right(s.length() - 8).toInt();
         if (fileSubsList.count() > idx)
             loadSubsFile(fileSubsList[idx]);
     }
diff --git b/src/qmplay2/Functions.cpp a/src/qmplay2/Functions.cpp
index 57a653a8..4cf5c202 100644
--- b/src/qmplay2/Functions.cpp
+++ a/src/qmplay2/Functions.cpp
@@ -1110,7 +1110,7 @@ QString Functions::getSeqFile(const QString &dir, const QString &ext, const QStr
     quint16 num = 0;
     for (const QString &f : QDir(dir).entryList({QString("QMPlay2_%1_?????%2").arg(frag, ext)}, QDir::Files, QDir::Name))
     {
-        const quint16 n = QStringView(f).mid(8 + frag.size() + 1, 5)MAYBE_TO_STRING.toUShort();
+        const quint16 n = QStringView(f).mid(8 + frag.size() + 1, 5).toUShort();
         if (n > num)
             num = n;
     }
