diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3f0ea184..ab7d3a3d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -441,7 +441,7 @@ find_package(${QT_PREFIX}Svg REQUIRED)
 if(USE_VULKAN)
     set(Vulkan_INCLUDE_DIRS "${CMAKE_SOURCE_DIR}/src/qmplay2/vulkan/headers/include")
     set(QMVK_USE_PCH ${USE_PCH})
-    set(QMVK_WAIT_TIMEOUT_MS 2500)
+    set(QMVK_WAIT_TIMEOUT_MS 5000)
     add_subdirectory(src/qmvk)
 endif()
 add_subdirectory(src/qmplay2)
