diff --git a/src/modules/CUVID/CMakeLists.txt b/src/modules/CUVID/CMakeLists.txt
index a9fca1e1..03d57dc8 100644
--- a/src/modules/CUVID/CMakeLists.txt
+++ b/src/modules/CUVID/CMakeLists.txt
@@ -48,4 +48,8 @@ if(WIN32)
     target_link_libraries(${PROJECT_NAME} opengl32)
 endif()
 
+if(APPLE)
+    target_link_libraries(${PROJECT_NAME} ${OPENGL_LIBRARY})
+endif()
+
 install(TARGETS ${PROJECT_NAME} LIBRARY DESTINATION ${MODULES_INSTALL_PATH})
diff --git a/src/modules/Extensions/YouTube.cpp b/src/modules/Extensions/YouTube.cpp
index 0e859608..31f5fbe5 100644
--- a/src/modules/Extensions/YouTube.cpp
+++ b/src/modules/Extensions/YouTube.cpp
@@ -1080,7 +1080,7 @@ QJsonDocument YouTube::getYtInitialData(const QByteArray &data)
     if (idx2 < 0)
         return QJsonDocument();
 
-    auto jsonData = data.mid(idx, idx2 - idx);
+    auto jsonData = data.mid(idx, idx2 - idx).trimmed();
     if (jsonData.endsWith(';'))
         jsonData.chop(1);
 
