diff --git a/src/gui/Main.cpp b/src/gui/Main.cpp
index a394273ce447f8ebbf5639a84d36ca5dfee16d36..60672338a89eeb0d235f3c350ac7750153669a0d 100644
--- a/src/gui/Main.cpp
+++ b/src/gui/Main.cpp
@@ -43,7 +43,10 @@
 #include <QFile>
 #include <QDir>
 #ifdef Q_OS_MACOS
+    #include <QMap>
     #include <QProcess>
+    #include <syslog.h>
+    #include <stdarg.h>
 #endif
 
 #include <csignal>
@@ -497,6 +500,23 @@ static void messageHandler(QtMsgType type, const QMessageLogContext &context, co
         g_messageHandlerMutex.unlock();
 #endif
     }
+#ifdef Q_OS_MACOS
+    else
+    {
+        // send a copy of terminal output to the syslog.
+        static const QMap<QtMsgType,int> type2Level = {
+            {QtDebugMsg, LOG_DEBUG},
+            {QtWarningMsg, LOG_WARNING},
+            {QtCriticalMsg, LOG_CRIT},
+            {QtFatalMsg, LOG_CRIT},
+            {QtInfoMsg, LOG_INFO}
+        };
+        g_messageHandlerMutex.lock();
+        syslog(LOG_NDELAY|LOG_PID|type2Level[type], "%s",
+               qFormatLogMessage(type, context, message).toLocal8Bit().constData());
+        g_messageHandlerMutex.unlock();
+    }
+#endif
 }
 
 int main(int argc, char *argv[])
