--- a/src/glsl/meson.build	2026-02-09 21:40:06.000000000 +0800
+++ b/src/glsl/meson.build	2026-03-03 23:26:30.000000000 +0800
@@ -30,6 +30,7 @@
 elif not glslang_req.disabled()
 
   glslang_deps = [
+    cxx.find_library('glslang', required: false),
     cxx.find_library('glslang-default-resource-limits', required: false)
   ]
 
@@ -49,7 +50,7 @@
     static   = arg[0]
     required = arg[1]
 
-    spirv = cxx.find_library('SPIRV', required: required, static: static, dirs: vulkan_lib_dirs)
+    spirv = cxx.find_library('SPIRV', required: required)
 
     if not spirv.found()
       continue
