diff --git cmake/modules/FindMeson.cmake cmake/modules/FindMeson.cmake
index f969cdf804231442e239949af6d97b5ddac9b90c..d3f04ed634053c8a95e3e8c2dbc65d3e0a7662f8 100644
--- cmake/modules/FindMeson.cmake
+++ cmake/modules/FindMeson.cmake
@@ -26,6 +26,9 @@ if(Meson_EXECUTABLE)
         OUTPUT_VARIABLE Meson_VERSION
         OUTPUT_STRIP_TRAILING_WHITESPACE
     )
+    message(STATUS "Found ${Meson_EXECUTABLE}, version ${Meson_VERSION}")
+else()
+    message(WARNING "Meson executable not found on path")
 endif()
 
 include(FindPackageHandleStandardArgs)
