diff --git plugins/contextbrowser/contextbrowser.cpp plugins/contextbrowser/contextbrowser.cpp
index e3169e92f6caad1d21fcd7d3eb31006cea04bfa4..0ad85faa96fc22045993fb545791b1675880c633 100644
--- plugins/contextbrowser/contextbrowser.cpp
+++ plugins/contextbrowser/contextbrowser.cpp
@@ -34,6 +34,8 @@
 #include <QTimer>
 #include <QToolButton>
 #include <QWidgetAction>
+#include <QApplication>
+#include <QDesktopWidget>
 
 #include <KActionCollection>
 #include <KLocalizedString>
@@ -630,12 +632,19 @@ QWidget* ContextBrowserPlugin::navigationWidgetForPosition(KTextEditor::View* vi
         m_currentToolTipDeclaration = IndexedDeclaration(decl);
 
         if (problemWidget && declWidget) {
-            auto* combinedWidget = new QuickOpenEmbeddedWidgetCombiner;
-            combinedWidget->layout()->addWidget(problemWidget);
-            combinedWidget->layout()->addWidget(declWidget);
-            return combinedWidget;
+            int maxHeight;
+            auto geom = QApplication::desktop()->availableGeometry(QCursor::pos());
+            maxHeight = geom.height() / 3;
+            if (problemWidget->sizeHint().height() + declWidget->sizeHint().height() <= maxHeight) {
+                auto* combinedWidget = new QuickOpenEmbeddedWidgetCombiner;
+                combinedWidget->layout()->addWidget(problemWidget);
+                combinedWidget->layout()->addWidget(declWidget);
+                return combinedWidget;
+            }
         }
-        if (problemWidget) {
+        if (declWidget) {
+            return declWidget;
+        } else if (problemWidget) {
             return problemWidget;
         }
         return declWidget;
