diff --git a/plugins/custom-definesandincludes/kcm_widget/parserwidget.ui b/plugins/custom-definesandincludes/kcm_widget/parserwidget.ui
index a30bd26572..84cd475145 100644
--- a/plugins/custom-definesandincludes/kcm_widget/parserwidget.ui
+++ b/plugins/custom-definesandincludes/kcm_widget/parserwidget.ui
@@ -71,11 +71,21 @@
           <string notr="true">c++11</string>
          </property>
         </item>
+        <item>
+         <property name="text">
+          <string notr="true">gnu++11</string>
+         </property>
+        </item>
         <item>
          <property name="text">
           <string notr="true">c++14</string>
          </property>
         </item>
+        <item>
+         <property name="text">
+          <string notr="true">gnu++14</string>
+         </property>
+        </item>
         <item>
          <property name="text">
           <string notr="true">c++17</string>
@@ -83,7 +93,17 @@
         </item>
         <item>
          <property name="text">
-          <string notr="true">c++2a</string>
+          <string notr="true">gnu++17</string>
+         </property>
+        </item>
+        <item>
+         <property name="text">
+          <string notr="true">c++20</string>
+         </property>
+        </item>
+        <item>
+         <property name="text">
+          <string notr="true">gnu++20</string>
          </property>
         </item>
        </widget>
