--- fontconfig_patches/orig.fc-presets	2015-01-01 03:33:26.000000000 +0100
+++ fontconfig_patches/fc-presets	2015-01-21 23:18:33.000000000 +0100
@@ -8,10 +8,10 @@
 ### The MIT License (MIT) http://opensource.org/licenses/MIT ###
 ###      part of infinality-bundle  http://bohoomil.com      ###
 
-RED='\E[01;31m'
-GRE='\E[01;32m'
-YEL='\E[01;33m'
-RES='\E[0m'
+RED=""
+GRE=""
+YEL=""
+RES=""
 
 presets_dir="/etc/fonts/conf.avail.infinality"
 config_dir="/etc/fonts/conf.d"
