--- fontconfig-2.12.4/fc-blanks/orig.Makefile.in	2017-07-05 10:38:16.000000000 +0200
+++ fontconfig-2.12.4/fc-blanks/Makefile.in	2017-12-04 18:00:29.000000000 +0100
@@ -592,10 +592,11 @@
 
 
 $(BLANKS_H): $(TMPL) $(BLANKSPY)
-@HAVE_PYTHON_TRUE@	$(AM_V_GEN) $(PYTHON) $(srcdir)/$(BLANKSPY) < $< > $(BLANKS_H).tmp && \
-@HAVE_PYTHON_TRUE@	mv $(BLANKS_H).tmp $(BLANKS_H) || ($(RM) $(BLANKS_H).tmp && false)
-@HAVE_PYTHON_FALSE@	@echo "No python installed. please install python to build $(BLANKS_H)."
-@HAVE_PYTHON_FALSE@	@false
+	@true
+#@HAVE_PYTHON_TRUE@	$(AM_V_GEN) $(PYTHON) $(srcdir)/$(BLANKSPY) < $< > $(BLANKS_H).tmp && \
+#@HAVE_PYTHON_TRUE@	mv $(BLANKS_H).tmp $(BLANKS_H) || ($(RM) $(BLANKS_H).tmp && false)
+#@HAVE_PYTHON_FALSE@	@echo "No python installed. please install python to build $(BLANKS_H)."
+#@HAVE_PYTHON_FALSE@	@false
 
 -include $(top_srcdir)/git.mk
 
--- fontconfig-2.12.4/fc-blanks/orig.Makefile.am	2016-12-02 04:22:19.000000000 +0100
+++ fontconfig-2.12.4/fc-blanks/Makefile.am	2017-12-04 17:59:49.000000000 +0100
@@ -28,13 +28,14 @@
 noinst_HEADERS = $(BLANKS_H)
 
 $(BLANKS_H): $(TMPL) $(BLANKSPY)
-if HAVE_PYTHON
-	$(AM_V_GEN) $(PYTHON) $(srcdir)/$(BLANKSPY) < $< > $(BLANKS_H).tmp && \
-	mv $(BLANKS_H).tmp $(BLANKS_H) || ($(RM) $(BLANKS_H).tmp && false)
-else
-	@echo "No python installed. please install python to build $(BLANKS_H)."
-	@false
-endif
+	@true
+#if HAVE_PYTHON
+#	$(AM_V_GEN) $(PYTHON) $(srcdir)/$(BLANKSPY) < $< > $(BLANKS_H).tmp && \
+#	mv $(BLANKS_H).tmp $(BLANKS_H) || ($(RM) $(BLANKS_H).tmp && false)
+#else
+#	@echo "No python installed. please install python to build $(BLANKS_H)."
+#	@false
+#endif
 
 EXTRA_DIST = 		\
 	$(BLANKSPY)	\
