diff --git b/orig.CMakeLists.txt a/CMakeLists.txt
index e1c2d25..d8ad2a2 100644
--- b/orig.CMakeLists.txt
+++ a/CMakeLists.txt
@@ -3,8 +3,8 @@ cmake_minimum_required(VERSION 3.0)
 project(decoration-api)
 set(PROJECT_VERSION "5.18.5")
 
-set(QT_MIN_VERSION "5.12.0")
-set(KF5_MIN_VERSION "5.66.0")
+set(QT_MIN_VERSION "5.9.8")
+set(KF5_MIN_VERSION "5.60.0")
 
 find_package(ECM ${KF5_MIN_VERSION} REQUIRED NO_MODULE)
 # where to look first for cmake modules, before ${CMAKE_ROOT}/Modules/ is checked
