--- kdelibs-4.12.5/kio/kfile/orig.kfiledialog.h	2014-04-28 06:37:51.000000000 +0200
+++ kdelibs-4.12.5/kio/kfile/kfiledialog.h	2014-07-11 00:09:28.000000000 +0200
@@ -749,7 +749,8 @@
      */
     static void setStartDir( const KUrl& directory );
 
-#ifdef Q_WS_WIN
+#if (defined Q_WS_WIN || defined Q_WS_MAC )
+    /** File dialogs are native by default on Windows and MAC. */
 public Q_SLOTS:
     int exec();
 #endif
--- kdelibs-4.12.5/kio/kfile/orig.kfiledialog.cpp	2014-04-28 06:37:51.000000000 +0200
+++ kdelibs-4.12.5/kio/kfile/kfiledialog.cpp	2014-07-11 00:13:19.000000000 +0200
@@ -44,8 +44,8 @@
 #include "krecentdirs.h"
 #include "kservice.h"
 
-/** File dialogs are native by default on Windows. */
-#if defined(Q_WS_WIN) || defined(Q_WS_MAEMO_5)
+/** File dialogs are native by default on Windows and Mac. */
+#if defined(Q_WS_WIN) || defined(Q_WS_MAEMO_5) || defined(Q_WS_MAC)
 const bool NATIVE_FILEDIALOGS_BY_DEFAULT = true;
 #else
 const bool NATIVE_FILEDIALOGS_BY_DEFAULT = false;
@@ -252,7 +252,7 @@
 
 KFileDialog::KFileDialog( const KUrl& startDir, const QString& filter,
                           QWidget *parent, QWidget* customWidget)
-#ifdef Q_WS_WIN
+#if (defined Q_WS_WIN || defined Q_WS_MAC )
     : KDialog( parent , Qt::WindowMinMaxButtonsHint),
 #else
     : KDialog( parent ),
@@ -312,8 +312,8 @@
     connect(fileQWidget, SIGNAL(accepted()), SLOT(accept()));
     //connect(fileQWidget, SIGNAL(canceled()), SLOT(slotCancel()));
 
-    if (customWidget)
-     d->w->setCustomWidget(QString(), customWidget);
+    if (customWidget){}
+     //d->w->setCustomWidget(QString(), customWidget);
 }
 
 
@@ -961,7 +961,7 @@
     return d->w;
 }
 
-#ifdef Q_WS_WIN
+#if (defined Q_WS_WIN || defined Q_WS_MAC)
 int KFileDialog::exec()
 {
     if (!d->native || !KFileDialogPrivate::Native::s_allowNative) {
@@ -1051,7 +1051,7 @@
 
     return res;
 }
-#endif // Q_WS_WIN
+#endif // Q_WS_WIN || Q_WS_MAC
 
 #ifdef Q_WS_WIN
 #define KF_EXTERN extern __declspec(dllimport)
