--- kdelibs-4.12.5/kdeui/widgets/orig.kmainwindow.cpp	2014-04-28 06:37:51.000000000 +0200
+++ kdelibs-4.12.5/kdeui/widgets/kmainwindow.cpp	2014-07-11 02:11:38.000000000 +0200
@@ -241,6 +241,8 @@
 
     q->setAttribute( Qt::WA_DeleteOnClose );
 
+    q->setUnifiedTitleAndToolBarOnMac(TRUE);
+
     // We handle this functionality (quitting the app) ourselves, with KGlobal::ref/deref.
     // This makes apps stay alive even if they only have a systray icon visible, or
     // a progress widget with "keep open" checked, for instance.
