diff --git src/service/CMakeLists.txt src/service/CMakeLists.txt
index 2557583..289a076 100644
--- src/service/CMakeLists.txt
+++ src/service/CMakeLists.txt
@@ -15,6 +15,7 @@ find_package (KF5WindowSystem ${KF5_DEP_VERSION} CONFIG REQUIRED)
 find_package (KF5GlobalAccel ${KF5_DEP_VERSION} CONFIG REQUIRED)
 find_package (KF5XmlGui ${KF5_DEP_VERSION} CONFIG REQUIRED)
 find_package (KF5KIO ${KF5_DEP_VERSION} CONFIG REQUIRED)
+include (ECMMarkNonGuiExecutable)
 
 # Standard stuff
 
@@ -58,6 +59,10 @@ qt5_add_dbus_adaptor (
    )
 
 add_executable (kactivitymanagerd ${kactivitymanager_SRCS})
+ecm_mark_nongui_executable (kactivitymanagerd)
+if (APPLE)
+    target_link_libraries(kactivitymanagerd "-framework CoreFoundation")
+endif()
 
 target_link_libraries (kactivitymanagerd
    Qt5::Core
diff --git src/service/Application.cpp src/service/Application.cpp
index 4cdd194..ad1a935 100644
--- src/service/orig.Application.cpp	2016-07-19 12:53:37.000000000 +0200
+++ src/service/Application.cpp	2016-07-20 11:42:44.529606644 +0200
@@ -60,6 +60,9 @@
 #include "Debug.h"
 #include "common/dbus/common.h"
 
+#ifdef Q_OS_OSX
+#include <CoreFoundation/CoreFoundation.h>
+#endif
 
 namespace {
     QList<QThread *> s_moduleThreads;
@@ -310,6 +313,20 @@
 
 int main(int argc, char **argv)
 {
+#ifdef Q_OS_OSX
+    CFBundleRef mainBundle = CFBundleGetMainBundle();
+    if (mainBundle) {
+        // get the application's Info Dictionary. For app bundles this would live in the bundle's Info.plist,
+        // for regular executables it is obtained in another way.
+        CFMutableDictionaryRef infoDict = (CFMutableDictionaryRef) CFBundleGetInfoDictionary(mainBundle);
+        if (infoDict) {
+            // Add or set the "LSUIElement" key with/to value "1". This can simply be a CFString.
+            CFDictionarySetValue(infoDict, CFSTR("LSUIElement"), CFSTR("1"));
+            // That's it. We're now considered as an "agent" by the window server, and thus will have
+            // neither menubar nor presence in the Dock or App Switcher.
+        }
+    }
+#endif
     // Disable session management for this process
     qunsetenv("SESSION_MANAGER");
 
