--- addons/search/plugin_search.cpp
+++ addons/search/plugin_search.cpp
@@ -362,6 +362,8 @@
     m_ui.replaceCombo->setInsertPolicy(QComboBox::NoInsert);
     m_ui.replaceCombo->lineEdit()->setClearButtonEnabled(true);
     m_ui.replaceCombo->setMaxCount(25);
+
+    m_toolView->setMinimumHeight(container->sizeHint().height());
 
     connect(m_mainWindow, SIGNAL(unhandledShortcutOverride(QEvent*)),
             this, SLOT(handleEsc(QEvent*)));

