--- addons/replicode/orig.replicodesettings.cpp	2015-11-04 23:25:33.000000000 +0100
+++ addons/replicode/replicodesettings.cpp	2015-12-11 19:24:18.000000000 +0100
@@ -31,7 +31,7 @@
 
 void ReplicodeSettings::load()
 {
-    QSettings settings(QStringLiteral("replicode"), QStringLiteral("replicode"));
+    QSettings settings(QSettings::IniFormat, QSettings::UserScope, QStringLiteral("replicode"), QStringLiteral("replicode"));
 
     settings.beginGroup(QStringLiteral("Load"));
     userOperatorPath = settings.value(QStringLiteral("User Operator Module Path"), QString()).toString();
@@ -95,7 +95,7 @@
 
 void ReplicodeSettings::save()
 {
-    QSettings settings(QStringLiteral("replicode"), QStringLiteral("replicode"));
+    QSettings settings(QSettings::IniFormat, QSettings::UserScope, QStringLiteral("replicode"), QStringLiteral("replicode"));
 
     settings.beginGroup(QStringLiteral("Load"));
     settings.setValue(QStringLiteral("User Operator Module Path"), userOperatorPath);
