diff --git kate/main.cpp kate/main.cpp
index bf4be52..51c46fc 100644
--- kate/main.cpp
+++ kate/main.cpp
@@ -50,6 +50,11 @@ extern "C" Q_DECL_EXPORT int kdemain(int argc, char **argv)
      * enable high dpi support
      */
     app.setAttribute(Qt::AA_UseHighDpiPixmaps, true);
+#ifdef Q_OS_OSX
+    if (!QGuiApplication::platformName().contains(QLatin1String("cocoa"))) {
+        QCoreApplication::setAttribute(Qt::AA_DontUseNativeMenuBar);
+    }
+#endif
 
     /**
      * Connect application with translation catalogs
diff --git kwrite/main.cpp kwrite/main.cpp
index a443c74..52e1099 100644
--- kwrite/main.cpp
+++ kwrite/main.cpp
@@ -46,6 +46,11 @@ extern "C" Q_DECL_EXPORT int kdemain(int argc, char **argv)
      * enable high dpi support
      */
     app.setAttribute(Qt::AA_UseHighDpiPixmaps, true);
+#ifdef Q_OS_OSX
+    if (!QGuiApplication::platformName().contains(QLatin1String("cocoa"))) {
+        QCoreApplication::setAttribute(Qt::AA_DontUseNativeMenuBar);
+    }
+#endif
 
     /**
      * Connect application with translation catalogs
