diff --git a/qtbase/src/network/bearer/qnetworkconfigmanager_p.cpp b/qtbase/src/network/bearer/qnetworkconfigmanager_p.cpp
index 2da073f..0629ae5 100644
--- a/qtbase/src/network/bearer/qnetworkconfigmanager_p.cpp
+++ b/qtbase/src/network/bearer/qnetworkconfigmanager_p.cpp
@@ -399,7 +399,7 @@ void QNetworkConfigurationManagerPrivate::updateConfigurations()
             }
         }
 
-        if (generic) {
+        if (generic && sessionEngines.count() == 0) {
             if (!envOK || skipGeneric <= 0)
                 sessionEngines.append(generic);
             else
