--- qt5kdweb-mp9-work/qtwebengine-opensource-src-5.7.0/src/3rdparty/chromium/base/process/orig.memory_mac.mm	2016-05-26 15:53:47.000000000 +0200
+++ qt5kdweb-mp9-work/qtwebengine-opensource-src-5.7.0/src/3rdparty/chromium/base/process/memory_mac.mm	2016-10-24 23:34:43.530498000 +0200
@@ -248,7 +248,8 @@
 // === Core Foundation CFAllocators ===
 
 bool CanGetContextForCFAllocator() {
-  return !base::mac::IsOSLaterThanElCapitan_DontCallThis();
+//   return !base::mac::IsOSLaterThanElCapitan_DontCallThis();
+    return false;
 }
 
 CFAllocatorContext* ContextForCFAllocator(CFAllocatorRef allocator) {
@@ -256,6 +257,7 @@
     ChromeCFAllocatorLeopards* our_allocator =
         const_cast<ChromeCFAllocatorLeopards*>(
             reinterpret_cast<const ChromeCFAllocatorLeopards*>(allocator));
+    NSLog(@"ContextForCFAllocator:%d(%@)=0x%lx", __LINE__, allocator, &our_allocator->_context);
     return &our_allocator->_context;
   } else if (base::mac::IsOSLion() ||
              base::mac::IsOSMountainLion() ||
@@ -265,6 +267,7 @@
     ChromeCFAllocatorLions* our_allocator =
         const_cast<ChromeCFAllocatorLions*>(
             reinterpret_cast<const ChromeCFAllocatorLions*>(allocator));
+    NSLog(@"ContextForCFAllocator:%d(%@)=0x%lx", __LINE__, allocator, &our_allocator->_context);
     return &our_allocator->_context;
   } else {
     return NULL;
