diff --git a/qtbase/src/gui/text/qtexthtmlparser.cpp b/qtbase/src/gui/text/qtexthtmlparser.cpp
index 9c758be..c438da0 100644
--- a/qtbase/src/gui/text/qtexthtmlparser.cpp
+++ b/qtbase/src/gui/text/qtexthtmlparser.cpp
@@ -1929,6 +1929,36 @@ QVector<QCss::Declaration> standardDeclarationForNode(const QTextHtmlParserNode
         val.type = QCss::Value::String;
         val.variant = QFontDatabase::systemFont(QFontDatabase::FixedFont).family();
         values << val;
+        val.type = QCss::Value::TermOperatorComma;
+        val.variant = QVariant();
+        values << val;
+        val.type = QCss::Value::String;
+        val.variant = QLatin1String("Monaco");
+        values << val;
+        val.type = QCss::Value::TermOperatorComma;
+        val.variant = QVariant();
+        values << val;
+        val.type = QCss::Value::String;
+        val.variant = QLatin1String("Lucida Console");
+        values << val;
+        val.type = QCss::Value::TermOperatorComma;
+        val.variant = QVariant();
+        values << val;
+        val.type = QCss::Value::String;
+        val.variant = QLatin1String("Consolas");
+        values << val;
+        val.type = QCss::Value::TermOperatorComma;
+        val.variant = QVariant();
+        values << val;
+        val.type = QCss::Value::String;
+        val.variant = QLatin1String("Courier New");
+        values << val;
+        val.type = QCss::Value::TermOperatorComma;
+        val.variant = QVariant();
+        values << val;
+        val.type = QCss::Value::String;
+        val.variant = QLatin1String("courier");
+        values << val;
         decl.d->values = values;
         decl.d->inheritable = true;
         decls << decl;
