--- qtbase/src/plugins/platforms/cocoa/orig.qcocoamenu.mm	2015-01-15 20:43:37.000000000 +0100
+++ qtbase/src/plugins/platforms/cocoa/qcocoamenu.mm	2015-01-15 20:52:42.000000000 +0100
@@ -303,8 +303,17 @@
     if (item->isMerged())
         return;
 
-    if ([item->nsItem() menu]) {
-        qWarning() << Q_FUNC_INFO << "Menu item is already in a menu, remove it from the other menu first before inserting";
+    NSMenu *m = [item->nsItem() menu];
+    if (m) {
+        QString mTitle = QCFString::toQString([m title]);
+        if (beforeItem) {
+            qWarning() << Q_FUNC_INFO << "Menu item" << item->text() << "is already in menu" << mTitle
+                       << "after item" << beforeItem->text()
+                       << ", remove it from the other menu first before inserting";
+        } else {
+            qWarning() << Q_FUNC_INFO << "Menu item" << item->text() << "is already in menu" << mTitle
+                       << ", remove it from the other menu first before inserting";
+        }
         return;
     }
     // if the item we're inserting before is merged, skip along until
