diff --git src/CMakeLists.txt src/CMakeLists.txt
index 02137ce0..93c16f3b 100644
--- src/CMakeLists.txt
+++ src/CMakeLists.txt
@@ -298,7 +298,7 @@ target_link_libraries(kleopatra_bin
   Qt5::PrintSupport # Printing secret keys
   ${_kleopatra_uiserver_extra_libs}
   ${_kleopatra_dbusaddons_libs}
-  kleopatraclientcore
+  KF5kleopatraclientcore
 )
 
 install(TARGETS kleopatra_bin ${KDE_INSTALL_TARGETS_DEFAULT_ARGS})
diff --git src/conf/CMakeLists.txt src/conf/CMakeLists.txt
index 5e2092c0..22de624e 100644
--- src/conf/CMakeLists.txt
+++ src/conf/CMakeLists.txt
@@ -17,7 +17,7 @@ if(BUILD_libkleopatraclient)
     ${kleopatra_SOURCE_DIR}/src/kcfg/smimevalidationpreferences.kcfgc
   )
 
-  set(_kcm_kleopatra_libkleopatraclient_extra_LIBS kleopatraclientgui)
+  set(_kcm_kleopatra_libkleopatraclient_extra_LIBS KF5kleopatraclientgui)
 
   set(_kcm_kleopatra_libkleopatraclient_extra_install_FILES
     kleopatra_config_smimevalidation.desktop
diff --git src/libkleopatraclient/core/CMakeLists.txt src/libkleopatraclient/core/CMakeLists.txt
index ff97d1a9..2beb8f54 100644
--- src/libkleopatraclient/core/CMakeLists.txt
+++ src/libkleopatraclient/core/CMakeLists.txt
@@ -11,7 +11,7 @@ endif()
 
 ecm_qt_declare_logging_category(_kleopatraclientcore_common_SRCS HEADER libkleopatraclientcore_debug.h IDENTIFIER LIBKLEOPATRACLIENTCORE_LOG CATEGORY_NAME org.kde.pim.libkleopatraclientcore)
 
-add_library(kleopatraclientcore
+add_library(KF5kleopatraclientcore
   ${_kleopatraclientcore_extra_SRCS}
   initialization.cpp
   command.cpp
@@ -21,22 +21,22 @@ add_library(kleopatraclientcore
   libkleopatraclientcore_debug.cpp
   ${_kleopatraclientcore_common_SRCS}
 )
-generate_export_header(kleopatraclientcore BASE_NAME kleopatraclientcore)
+generate_export_header(KF5kleopatraclientcore BASE_NAME kleopatraclientcore)
 
-set_target_properties(kleopatraclientcore PROPERTIES
+set_target_properties(KF5kleopatraclientcore PROPERTIES
   VERSION ${libkleopatraclient_version}
   SOVERSION ${libkleopatraclient_soversion}
 )
 
 if(WIN32)
   if(ASSUAN2_FOUND)
-    target_link_libraries(kleopatraclientcore
+    target_link_libraries(KF5kleopatraclientcore
       
       ${ASSUAN2_LIBRARIES}
       ws2_32
     )
   else()
-    target_link_libraries(kleopatraclientcore
+    target_link_libraries(KF5kleopatraclientcore
       
       ${ASSUAN_VANILLA_LIBRARIES}
       ws2_32
@@ -44,18 +44,18 @@ if(WIN32)
   endif()
 else()
   if(ASSUAN2_FOUND)
-    target_link_libraries(kleopatraclientcore
+    target_link_libraries(KF5kleopatraclientcore
       
       ${ASSUAN2_LIBRARIES}
     )
   else()
-    target_link_libraries(kleopatraclientcore
+    target_link_libraries(KF5kleopatraclientcore
       
       ${ASSUAN_PTHREAD_LIBRARIES}
     )
   endif()
 endif()
 
-target_link_libraries(kleopatraclientcore Qt5::Widgets KF5::I18n)
+target_link_libraries(KF5kleopatraclientcore Qt5::Widgets KF5::I18n)
 
-install(TARGETS kleopatraclientcore ${KDE_INSTALL_TARGETS_DEFAULT_ARGS})
+install(TARGETS KF5kleopatraclientcore ${KDE_INSTALL_TARGETS_DEFAULT_ARGS})
diff --git src/libkleopatraclient/gui/CMakeLists.txt src/libkleopatraclient/gui/CMakeLists.txt
index 114e3135..9efd13f0 100644
--- src/libkleopatraclient/gui/CMakeLists.txt
+++ src/libkleopatraclient/gui/CMakeLists.txt
@@ -1,16 +1,16 @@
 add_definitions(-DQT_NO_CAST_FROM_ASCII -DQT_NO_KEYWORDS -DQT_NO_CAST_TO_ASCII)
 
-add_library(kleopatraclientgui certificaterequester.cpp)
-generate_export_header(kleopatraclientgui BASE_NAME kleopatraclientgui)
+add_library(KF5kleopatraclientgui certificaterequester.cpp)
+generate_export_header(KF5kleopatraclientgui BASE_NAME kleopatraclientgui)
 
-target_link_libraries(kleopatraclientgui
-  kleopatraclientcore
+target_link_libraries(KF5kleopatraclientgui
+  KF5kleopatraclientcore
   KF5::I18n
 )
 
-set_target_properties(kleopatraclientgui PROPERTIES
+set_target_properties(KF5kleopatraclientgui PROPERTIES
   VERSION ${libkleopatraclient_version}
   SOVERSION ${libkleopatraclient_soversion}
 )
 
-install(TARGETS kleopatraclientgui ${KDE_INSTALL_TARGETS_DEFAULT_ARGS})
+install(TARGETS KF5kleopatraclientgui ${KDE_INSTALL_TARGETS_DEFAULT_ARGS})
diff --git src/libkleopatraclient/tests/CMakeLists.txt src/libkleopatraclient/tests/CMakeLists.txt
index d2a6b9cb..1a4c8abf 100644
--- src/libkleopatraclient/tests/CMakeLists.txt
+++ src/libkleopatraclient/tests/CMakeLists.txt
@@ -10,7 +10,7 @@ foreach(_kleoclient_test ${kleoclient_TESTS})
   ecm_mark_as_test(${_kleoclient_test})
 
   target_link_libraries("${_kleoclient_test}"
-    kleopatraclientcore
+    KF5kleopatraclientcore
     Qt5::Gui
     Qt5::Core
 )
