diff --git src/mainwindow.cpp src/mainwindow.cpp
index 5697107f..342a9ba4 100644
--- src/mainwindow.cpp
+++ src/mainwindow.cpp
@@ -211,8 +211,8 @@ public:
 
     void gnupgLogViewer()
     {
-        if (!QProcess::startDetached(QStringLiteral("kwatchgnupg")))
-            KMessageBox::error(q, i18n("Could not start the GnuPG Log Viewer (kwatchgnupg). "
+        if (!QProcess::startDetached(QStringLiteral("kwatchgnupg5")))
+            KMessageBox::error(q, i18n("Could not start the GnuPG Log Viewer (kwatchgnupg5). "
                                        "Please check your installation."),
                                i18n("Error Starting KWatchGnuPG"));
     }
@@ -377,7 +377,7 @@ void MainWindow::Private::setupActions()
 #ifndef Q_OS_WIN
         {
             "tools_start_kwatchgnupg", i18n("GnuPG Log Viewer"), QString(),
-            "kwatchgnupg", q, SLOT(gnupgLogViewer()), QString(), false, true
+            "kwatchgnupg5", q, SLOT(gnupgLogViewer()), QString(), false, true
         },
 #endif
 #ifdef Q_OS_WIN
diff --git src/data/org.kde.kleopatra.desktop src/data/org.kde.kleopatra.desktop
index 37807aa9..3ef391bc 100644
--- src/data/org.kde.kleopatra.desktop
+++ src/data/org.kde.kleopatra.desktop
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Type=Application
 Categories=Qt;KDE;Utility;
-Exec=kleopatra
+Exec=kleopatra5
 Icon=kleopatra
 Name=Kleopatra
 Name[af]=Kleopatra
