diff --git a/kdepasswd/CMakeLists.txt b/kdepasswd/CMakeLists.txt
index 509ada5..7bb2975 100644
--- a/kdepasswd/CMakeLists.txt
+++ b/kdepasswd/CMakeLists.txt
@@ -9,6 +9,7 @@ set(CMAKE_MODULE_PATH ${ECM_MODULE_PATH})
 include(ECMSetupVersion)
 include(FeatureSummary)
 include(ECMAddAppIcon)
+include(ECMMarkNonGuiExecutable)
 include(KDEInstallDirs)
 include(KDECMakeSettings)
 include(KDEFrameworkCompilerSettings NO_POLICY_SCOPE)
@@ -32,6 +33,7 @@ set(kdepasswd_SRCS
 )
 
 add_executable(kdepasswd ${kdepasswd_SRCS})
+ecm_mark_nongui_executable(kdepasswd)
 
 target_link_libraries(kdepasswd
     KF5::KIOWidgets
diff --git a/kdialog/CMakeLists.txt b/kdialog/CMakeLists.txt
index 81a4228..3c0f939 100644
--- a/kdialog/CMakeLists.txt
+++ b/kdialog/CMakeLists.txt
@@ -7,6 +7,7 @@ find_package(ECM 1.7.0 CONFIG REQUIRED)
 set(CMAKE_MODULE_PATH ${ECM_MODULE_PATH})
 
 include(FeatureSummary)
+include(ECMMarkNonGuiExecutable)
 include(KDEInstallDirs)
 include(KDECMakeSettings)
 include(KDEFrameworkCompilerSettings NO_POLICY_SCOPE)
@@ -33,6 +34,7 @@ set(kdialog_SRCS
 qt5_add_dbus_adaptor( kdialog_SRCS org.kde.kdialog.ProgressDialog.xml progressdialog.h KProgressDialog )
 
 add_executable(kdialog ${kdialog_SRCS})
+ecm_mark_nongui_executable(kdialog)
 
 # Need libkfile due to the code that adjusts the geometry of the KDirSelectDialog
 target_link_libraries(kdialog
diff --git a/keditbookmarks/CMakeLists.txt b/keditbookmarks/CMakeLists.txt
index c0d6c18..581b7bf 100644
--- a/keditbookmarks/CMakeLists.txt
+++ b/keditbookmarks/CMakeLists.txt
@@ -42,6 +42,7 @@ add_subdirectory(kbookmarkmodel)
 set(kbookmarkmerger_SRCS kbookmarkmerger.cpp )
 
 add_executable(kbookmarkmerger ${kbookmarkmerger_SRCS})
+ecm_mark_nongui_executable(kbookmarkmerger)
 
 target_link_libraries(kbookmarkmerger
    KF5::KIOCore
@@ -78,6 +79,7 @@ qt5_add_dbus_interface( keditbookmarks_SRCS ${toplevel_xml} toplevel_interface )
 kconfig_add_kcfg_files(keditbookmarks_SRCS settings.kcfgc )
 
 add_executable( keditbookmarks ${keditbookmarks_SRCS})
+ecm_mark_nongui_executable( keditbookmarks )
 
 target_link_libraries(keditbookmarks kbookmarkmodel_private5
    KF5::Parts
diff --git a/konqueror/CMakeLists.txt b/konqueror/CMakeLists.txt
index 77cf38a..da5d36d 100644
--- a/konqueror/CMakeLists.txt
+++ b/konqueror/CMakeLists.txt
@@ -15,6 +15,7 @@ include(KDEInstallDirs)
 include(KDECMakeSettings)
 include(KDECompilerSettings NO_POLICY_SCOPE)
 include(ECMInstallIcons)
+include(ECMMarkNonGuiExecutable)
 include(ECMSetupVersion)
 
 set(KONQUEROR_LIB_VERSION "5.0.97")
diff --git a/konqueror/plugins/fsview/CMakeLists.txt b/konqueror/plugins/fsview/CMakeLists.txt
index 31ca04c..24c6dcb 100644
--- a/konqueror/plugins/fsview/CMakeLists.txt
+++ b/konqueror/plugins/fsview/CMakeLists.txt
@@ -16,6 +16,7 @@ set(libfsview_SRCS
 set(fsview_SRCS main.cpp ${libfsview_SRCS} )
 
 add_executable(fsview ${fsview_SRCS})
+ecm_mark_nongui_executable(fsview)
 
 target_link_libraries(fsview KF5::KIOWidgets KF5::KDELibs4Support)
 
diff --git a/konqueror/client/CMakeLists.txt b/konqueror/client/CMakeLists.txt
index 745b0f3..5f49c48 100644
--- a/konqueror/client/CMakeLists.txt
+++ b/konqueror/client/CMakeLists.txt
@@ -24,6 +25,7 @@ else (WIN32)
     endif(X11_FOUND)
     install(TARGETS kdeinit_kfmclient  ${KDE_INSTALL_TARGETS_DEFAULT_ARGS} )
 endif (WIN32)
+ecm_mark_nongui_executable(kfmclient)
 
 install(TARGETS kfmclient ${KDE_INSTALL_TARGETS_DEFAULT_ARGS})
 
