diff --git a/kdepasswd/kdepasswd.cpp b/kdepasswd/kdepasswd.cpp
index 7e29ef9..036edc5 100644
--- a/kdepasswd/kdepasswd.cpp
+++ b/kdepasswd/kdepasswd.cpp
@@ -51,7 +51,7 @@ int main(int argc, char **argv)
     // enable high dpi support
     app.setAttribute(Qt::AA_UseHighDpiPixmaps, true);
 
-    app.setWindowIcon(QIcon::fromTheme(QLatin1String("preferences-desktop-user-password")));
+    app.setWindowIcon(QIcon::fromTheme(QLatin1String("preferences-desktop-user-password"), app.windowIcon()));
     QCommandLineParser parser;
     KAboutData::setApplicationData(aboutData);
     parser.addVersionOption();
diff --git a/keditbookmarks/main.cpp b/keditbookmarks/main.cpp
index 6125393..275f19a 100644
--- a/keditbookmarks/main.cpp
+++ b/keditbookmarks/main.cpp
@@ -115,7 +115,7 @@ extern "C" Q_DECL_EXPORT int kdemain(int argc, char **argv)
     aboutData.addAuthor(i18n("Alexander Kellett"), i18n("Author"), "lypanov@kde.org");
     KAboutData::setApplicationData(aboutData);
 
-    app.setWindowIcon(QIcon::fromTheme(QStringLiteral("bookmarks-organize")));
+    app.setWindowIcon(QIcon::fromTheme(QStringLiteral("bookmarks-organize"), app.windowIcon()));
 
     QCommandLineParser parser;
     parser.setApplicationDescription(aboutData.shortDescription());
diff --git a/kdialog/kdialog.cpp b/kdialog/kdialog.cpp
index 7f8acdb..eb7d936 100644
--- a/kdialog/kdialog.cpp
+++ b/kdialog/kdialog.cpp
@@ -229,7 +229,7 @@ int main(int argc, char *argv[])
     aboutData.addAuthor(i18n("Richard Moore"),QString(),"rich@kde.org");
     aboutData.addAuthor(i18n("Dawit Alemayehu"),QString(),"adawit@kde.org");
     aboutData.addAuthor(i18n("Kai Uwe Broulik"),QString(),"kde@privat.broulik.de");
-    QApplication::setWindowIcon(QIcon::fromTheme("system-run"));
+    QApplication::setWindowIcon(QIcon::fromTheme("system-run", app.windowIcon()));
 
     QCommandLineParser parser;
     KAboutData::setApplicationData(aboutData);
