diff --git a/src/plugins/plugins.pro b/src/plugins/plugins.pro
index 8afe610b4acf214518347930de5c819d60000142..947b64e22453d0abe2fa532b804af76bce481455 100644
--- a/src/plugins/plugins.pro
+++ b/src/plugins/plugins.pro
@@ -1,2 +1,3 @@
 TEMPLATE = subdirs
-SUBDIRS = platformthemes styles
+# SUBDIRS = platformthemes styles
+SUBDIRS = styles
diff --git a/src/plugins/styles/styles.pro b/src/plugins/styles/styles.pro
index fa9603374ed960b38a261f524db9e5f0abae26b9..ac1ad6422ddfbcfe22749b68f499c4b35e602ec4 100644
--- a/src/plugins/styles/styles.pro
+++ b/src/plugins/styles/styles.pro
@@ -4,11 +4,11 @@ SUBDIRS = \
     motif \
     plastique
 
-greaterThan(QT_MAJOR_VERSION, 5) | greaterThan(QT_MINOR_VERSION, 6) {
-    # only 5.7 or later
-    SUBDIRS += bb10style
-}
+# greaterThan(QT_MAJOR_VERSION, 5) | greaterThan(QT_MINOR_VERSION, 6) {
+#     # only 5.7 or later
+#     SUBDIRS += bb10style
+# }
 
-packagesExist(gtk+-2.0 x11) {
-    SUBDIRS += gtk2
-}
+# packagesExist(gtk+-2.0 x11) {
+#     SUBDIRS += gtk2
+# }
