diff --git a/src/plugins/plugins.pro b/src/plugins/plugins.pro
index 8afe610b4acf214518347930de5c819d60000142..947b64e22453d0abe2fa532b804af76bce481455 100644
--- a/src/plugins/plugins.pro
+++ b/src/plugins/plugins.pro
@@ -1,2 +1,3 @@
 TEMPLATE = subdirs
-SUBDIRS = platformthemes styles
+# SUBDIRS = platformthemes styles
+SUBDIRS = styles
diff --git a/src/plugins/styles/styles.pro b/src/plugins/styles/styles.pro
index 853e705..b1ff3dc 100644
--- a/src/plugins/styles/styles.pro
+++ b/src/plugins/styles/styles.pro
@@ -9,6 +9,6 @@ SUBDIRS = \
 #    SUBDIRS += bb10style
 #}
 
-packagesExist(gtk+-2.0 x11) {
-    SUBDIRS += gtk2
-}
+#packagesExist(gtk+-2.0 x11) {
+#    SUBDIRS += gtk2
+#}
