diff --git a/src/plugins/styles/styles.pro b/src/plugins/styles/styles.pro
index fa96033..853e705 100644
--- a/src/plugins/styles/styles.pro
+++ b/src/plugins/styles/styles.pro
@@ -4,10 +4,10 @@ SUBDIRS = \
     motif \
     plastique
 
-greaterThan(QT_MAJOR_VERSION, 5) | greaterThan(QT_MINOR_VERSION, 6) {
-    # only 5.7 or later
-    SUBDIRS += bb10style
-}
+#greaterThan(QT_MAJOR_VERSION, 5) | greaterThan(QT_MINOR_VERSION, 6) {
+#    # only 5.7 or later
+#    SUBDIRS += bb10style
+#}
 
 packagesExist(gtk+-2.0 x11) {
     SUBDIRS += gtk2
