diff --git a/libs/ui/input/wintab/kis_tablet_support.h b/libs/ui/input/wintab/kis_tablet_support.h
index 042885f..e8f501d 100644
--- a/libs/ui/input/wintab/kis_tablet_support.h
+++ b/libs/ui/input/wintab/kis_tablet_support.h
@@ -39,7 +39,7 @@
 
 struct QTabletDeviceData
 {
-#ifndef Q_WS_MAC
+#ifndef Q_OS_OSX
     int minPressure;
     int maxPressure;
     int minTanPressure;
@@ -54,7 +54,7 @@ struct QTabletDeviceData
                               int outOriginY, int outExtentY) const;
 #endif
 
-#if defined(HAVE_X11) || (defined(Q_WS_MAC) && !defined(QT_MAC_USE_COCOA))
+#if defined(HAVE_X11) || (defined(Q_OS_OSX) && !defined(QT_MAC_USE_COCOA))
     QPointer<QWidget> widgetToGetPress;
 #endif
 
@@ -74,7 +74,7 @@ struct QTabletDeviceData
     int xinput_button_release;
     int xinput_proximity_in;
     int xinput_proximity_out;
-#elif defined(Q_WS_MAC)
+#elif defined(Q_OS_OSX)
     quint64 tabletUniqueID;
     int tabletDeviceType;
     int tabletPointerType;
@@ -195,7 +195,7 @@ static inline int sign(int x)
     return x >= 0 ? 1 : -1;
 }
 
-#ifndef Q_WS_MAC
+#ifndef Q_OS_OSX
 inline QPointF QTabletDeviceData::scaleCoord(int coordX, int coordY,
                                             int outOriginX, int outExtentX,
                                             int outOriginY, int outExtentY) const

diff --git a/plugins/flake/textshape/TextTool.cpp b/plugins/flake/textshape/TextTool.cpp
index 34bdf6b..5b749fb 100644
--- a/plugins/flake/textshape/TextTool.cpp
+++ b/plugins/flake/textshape/TextTool.cpp
@@ -1658,7 +1658,7 @@ void TextTool::keyPressEvent(QKeyEvent *event)
         } else if (hit(item, KStandardShortcut::ForwardWord)) {
             moveOperation = QTextCursor::WordRight;
         }
-#ifdef Q_WS_MAC
+#ifdef Q_OS_OSX
         // Don't reject "alt" key, it may be used for typing text on Mac OS
         else if ((event->modifiers() & Qt::ControlModifier) || event->text().length() == 0) {
 #else
