diff --git src/runtime/kwalletd/CMakeLists.txt src/runtime/kwalletd/CMakeLists.txt
index ba9e7ba..e87ec15 100644
--- src/runtime/kwalletd/CMakeLists.txt
+++ src/runtime/kwalletd/CMakeLists.txt
@@ -118,6 +122,9 @@ if (KF5Gpgmepp_FOUND)
     target_link_libraries(kwalletd5 KF5::Gpgmepp )
     kde_target_enable_exceptions(kwalletd5 PRIVATE)
 endif(KF5Gpgmepp_FOUND)
+if (APPLE)
+    target_link_libraries(kwalletd5 "-framework CoreFoundation -framework AppKit" )
+endif ()
 
 install(TARGETS kwalletd5  ${KF5_INSTALL_TARGETS_DEFAULT_ARGS})
 
diff --git src/runtime/kwalletd/main.cpp src/runtime/kwalletd/main.cpp
index dd959b6..d038a07 100644
--- src/runtime/kwalletd/main.cpp
+++ src/runtime/kwalletd/main.cpp
@@ -40,6 +40,10 @@
 #include <sys/un.h>
 #include <unistd.h>
 
+#ifdef Q_OS_OSX
+#include <CoreFoundation/CoreFoundation.h>
+#endif
+
 #define BSIZE 1000
 static int pipefd = 0;
 static int socketfd = 0;
@@ -160,6 +164,20 @@ int main(int argc, char **argv)
 {
     char *hash = NULL;
 #ifndef Q_OS_WIN
+#ifdef Q_OS_OSX
+    CFBundleRef mainBundle = CFBundleGetMainBundle();
+    if (mainBundle) {
+        // get the application's Info Dictionary. For app bundles this would live in the bundle's Info.plist,
+        // for regular executables it is obtained in another way.
+        CFMutableDictionaryRef infoDict = (CFMutableDictionaryRef) CFBundleGetInfoDictionary(mainBundle);
+        if (infoDict) {
+            // Add or set the "LSUIElement" key with/to value "1". This can simply be a CFString.
+            CFDictionarySetValue(infoDict, CFSTR("LSUIElement"), CFSTR("1"));
+            // That's it. We're now considered as an "agent" by the window server, and thus will have
+            // neither menubar nor presence in the Dock or App Switcher.
+        }
+    }
+#endif // Q_OS_OSX
     if (getenv("PAM_KWALLET5_LOGIN")) {
         hash = checkPamModule(argc, argv);
     }
