diff --git addons/snippets/katesnippets.cpp addons/snippets/katesnippets.cpp
index d910cce..ee5d94d 100644
--- addons/snippets/katesnippets.cpp
+++ addons/snippets/katesnippets.cpp
@@ -101,7 +101,9 @@ KateSnippetsPluginView::~KateSnippetsPluginView()
         iface->unregisterCompletionModel(KateSnippetGlobal::self()->completionModel());
     }
 
-    m_mainWindow->guiFactory()->removeClient(this);
+    if (m_mainWindow && m_mainWindow->guiFactory()) {
+        m_mainWindow->guiFactory()->removeClient(this);
+    }
 
     if (m_toolView) {
         delete m_toolView;
