diff --git src/Application.cpp src/Application.cpp
index 06f9321..af1d141 100644
--- src/Application.cpp
+++ src/Application.cpp
@@ -58,7 +58,7 @@ void Application::init()
     // KDE's menuBar()->isTopLevel() hasn't worked in a while.
     // For now, put menus inside Konsole window; this also make
     // the keyboard shortcut to show menus look reasonable.
-    setAttribute(Qt::AA_DontUseNativeMenuBar);
+//     setAttribute(Qt::AA_DontUseNativeMenuBar);
 #endif
 }
 
