diff --git a/qtbase/src/widgets/widgets/qmenu.cpp b/qtbase/src/widgets/widgets/qmenu.cpp
index c194547..aaf357f 100644
--- a/qtbase/src/widgets/widgets/qmenu.cpp
+++ b/qtbase/src/widgets/widgets/qmenu.cpp
@@ -1762,7 +1762,8 @@ QAction *QMenu::addSeparator()
 
     The rendering of the hint is style and platform dependent. Widget
     styles can use the text information in the rendering for sections,
-    or can choose to ignore it and render sections like simple separators.
+    or can choose to ignore it and render sections like simple separators,
+    depending on whether the platform supports texted separators.
 
     QMenu takes ownership of the returned QAction.
 
@@ -1771,7 +1772,20 @@ QAction *QMenu::addSeparator()
 QAction *QMenu::addSection(const QString &text)
 {
     QAction *action = new QAction(text, this);
+#ifdef Q_OS_OSX
+    if (QGuiApplication::platformName().contains(QLatin1String("cocoa"))) {
+        // add the text item without turning it into a separator first
+        // but deactivated, also to make it visually distinct
+        action->setEnabled(false);
+        action->setMenuRole(QAction::NoRole);
+        addAction(action);
+        // create a 2nd action that will be returned and have the expected properties
+        // but show up as a simple separator on this platform
+        action = new QAction(text, this);
+    }
+#endif
     action->setSeparator(true);
+    action->setMenuRole(QAction::NoRole);
     addAction(action);
     return action;
 }
@@ -1786,7 +1800,8 @@ QAction *QMenu::addSection(const QString &text)
 
     The rendering of the hints is style and platform dependent. Widget
     styles can use the text and icon information in the rendering for sections,
-    or can choose to ignore them and render sections like simple separators.
+    or can choose to ignore them and render sections like simple separators,
+    depending on whether the platform supports texted separators.
 
     QMenu takes ownership of the returned QAction.
 
@@ -1795,7 +1810,20 @@ QAction *QMenu::addSection(const QString &text)
 QAction *QMenu::addSection(const QIcon &icon, const QString &text)
 {
     QAction *action = new QAction(icon, text, this);
+#ifdef Q_OS_OSX
+    if (QGuiApplication::platformName().contains(QLatin1String("cocoa"))) {
+        // add the text item without turning it into a separator first
+        // but deactivated, also to make it visually distinct
+        action->setEnabled(false);
+        action->setMenuRole(QAction::NoRole);
+        addAction(action);
+        // create a 2nd action that will be returned and have the expected properties
+        // but show up as a simple separator on this platform
+        action = new QAction(icon, text, this);
+    }
+#endif
     action->setSeparator(true);
+    action->setMenuRole(QAction::NoRole);
     addAction(action);
     return action;
 }
