diff --git a/qtbase/configure b/qtbase/configure
index 0b8b417..91c6bc1 100755
--- a/qtbase/configure
+++ b/qtbase/configure
@@ -5310,8 +5310,19 @@ if [ "$XPLATFORM_MAC" = "yes" ]; then
 fi
 
 # auto-detect OpenGL support (es2 = OpenGL ES 2.0 or higher)
+doOpenGLTest() {
+    if [ "$QT_QPA_DEFAULT_PLATFORM" = "cocoa" ] || [ "$QT_QPA_DEFAULT_PLATFORM" = "" -a "$XPLATFORM_MAC" = "yes" ]; then
+        # QT_QPA_DEFAULT_PLATFORM is either empty or set by the user, so we have to check 
+        # both manually and automatically selected cocoa. It qpa=cocoa, we do NOT want to
+        # use pkg-config to find the OpenGL libraries because it's too likely to find
+        # OpenGL libraries that are not the system frameworks appropriate for cocoa mode.
+        compileTest unix/opengldesktop "OpenGL"
+    else
+        compileTestWithPkgConfig gl unix/opengldesktop "OpenGL" OPENGL
+    fi
+}
 if [ "$CFG_OPENGL" = "auto" ] || [ "$CFG_OPENGL" = "yes" ]; then
-    if compileTestWithPkgConfig gl unix/opengldesktop "OpenGL" OPENGL; then
+    if doOpenGLTest; then
         CFG_OPENGL=desktop
     elif compileTestWithPkgConfig glesv2 unix/opengles2 "OpenGL ES 2.0" OPENGL_ES2; then
         CFG_OPENGL=es2
@@ -5339,7 +5350,7 @@ elif [ "$CFG_OPENGL" = "es2" ]; then
     fi
 elif [ "$CFG_OPENGL" = "desktop" ]; then
     # Desktop OpenGL support
-    compileTestWithPkgConfig gl unix/opengldesktop "OpenGL" OPENGL
+    doOpenGLTest
     if [ $? != "0" ]; then
         echo "The OpenGL functionality test failed!"
         echo " You might need to modify the include and library search paths by editing"
@@ -6484,6 +6495,9 @@ else
 fi
 if [ "$CFG_RPATH" = "yes" ]; then
     QT_CONFIG="$QT_CONFIG rpath"
+elif [ "$XPLATFORM_MAC" = "yes" ]; then
+    # store the absolute (= full) path in shared library install_names
+    QMAKE_CONFIG="$QMAKE_CONFIG absolute_library_soname"
 fi
 
 if [ '!' -z "$W_FLAGS" ]; then
