--- a/qtbase/src/widgets/widgets/qpushbutton.cpp
+++ b/qtbase/src/widgets/widgets/qpushbutton.cpp
@@ -332,8 +332,13 @@
     if (!d->flat && !d->down)
         option->state |= QStyle::State_Raised;
     option->text = d->text;
-    option->icon = d->icon;
-    option->iconSize = iconSize();
+    //if (option->text.isEmpty() || style()->styleHint(QStyle::SH_DialogButtonBox_ButtonsHaveIcons)) {
+        option->icon = d->icon;
+        option->iconSize = iconSize();
+    //} else {
+    //    option->icon = QIcon();
+    //    option->iconSize = QSize(0,0);
+    //}
 }
 
 void QPushButton::setAutoDefault(bool enable)
@@ -398,20 +403,20 @@
     initStyleOption(&opt);
 
     // calculate contents size...
+    QString s(text());
+    bool empty = s.isEmpty();
 #ifndef QT_NO_ICON
 
     bool showButtonBoxIcons = qobject_cast<QDialogButtonBox*>(parentWidget())
                           && style()->styleHint(QStyle::SH_DialogButtonBox_ButtonsHaveIcons);
 
-    if (!icon().isNull() || showButtonBoxIcons) {
+    if (!icon().isNull() && (showButtonBoxIcons || empty)) {
         int ih = opt.iconSize.height();
         int iw = opt.iconSize.width() + 4;
         w += iw;
         h = qMax(h, ih);
     }
 #endif
-    QString s(text());
-    bool empty = s.isEmpty();
     if (empty)
         s = QString::fromLatin1("XXXX");
     QFontMetrics fm = fontMetrics();
