diff --git a/src/src.pro b/src/src.pro
index 2524b5e..f9e7098 100644
--- a/src/src.pro
+++ b/src/src.pro
@@ -81,7 +81,7 @@ macx: {
     HEADERS += kchmviewerapp.h
     SOURCES += kchmviewerapp.cpp
     QMAKE_INFO_PLIST=resources/Info.plist
-    QMAKE_POST_LINK += cp resources/*.icns ${DESTDIR}/kchmviewer.app/Contents/Resources;
+    ICON = resources/kchmviewer.icns
     LIBS += ../lib/libebook/libebook.a
     POST_TARGETDEPS += ../lib/libebook/libebook.a
 }
diff --git a/src/kchmviewerapp.cpp b/src/kchmviewerapp.cpp
index 62dfb62..7a73570 100644
--- a/src/kchmviewerapp.cpp
+++ b/src/kchmviewerapp.cpp
@@ -63,5 +63,5 @@ void KchmviewerApp::onTimer()
         ++m_nResend;
         return;
     }
-    main->loadFile(m_filePath);
+    main->actionOpenRecentFile(m_filePath);
 }
diff --git a/src/main.cpp b/src/main.cpp
index b17d203..ab8f089 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -31,7 +31,7 @@
 	#include <kaboutdata.h>
 #endif
 
-#if defined (Q_WS_MAC)
+#if defined (Q_OS_OSX)
         #include "kchmviewerapp.h"
 #else
         typedef QApplication  KchmviewerApp;
diff --git a/lib/libebook/ebook_epub.cpp b/lib/libebook/ebook_epub.cpp
index 83a7773..28c384d 100644
--- a/lib/libebook/ebook_epub.cpp
+++ b/lib/libebook/ebook_epub.cpp
@@ -166,7 +166,7 @@ QString EBook_EPUB::currentEncoding() const
 
 bool EBook_EPUB::setCurrentEncoding(const char *)
 {
-	abort();
+	return false;
 }
 
 bool EBook_EPUB::isSupportedUrl(const QUrl &url)
