--- ltmain.sh.orig	2015-01-07 05:56:48.000000000 +0100
+++ ltmain.sh	2015-05-20 20:19:02.000000000 +0200
@@ -2991,7 +2991,8 @@
 	  versuffix="$major.$age.$revision"
 	  # Darwin ld doesn't like 0 for these options...
 	  minor_current=`expr $current + 1`
-	  verstring="-compatibility_version $minor_current -current_version $minor_current.$revision"
+	  #verstring="-compatibility_version $minor_current -current_version $minor_current.$revision"
+	  verstring="-compatibility_version 11 -current_version $minor_current.$revision"
 	  ;;
 
 	freebsd-aout)
