diff --git a/src/src.pri b/src/src.pri
index 1a46c39..00ac1c0 100644
--- a/src/src.pri
+++ b/src/src.pri
@@ -8,10 +8,10 @@ DEPENDPATH += $$PWD
 QT += webkit network webkitwidgets printsupport uitools
 
 # Share object files for faster compiling
-RCC_DIR     = $$PWD/.rcc
-UI_DIR      = $$PWD/.ui
-MOC_DIR     = $$PWD/.moc
-OBJECTS_DIR = $$PWD/.obj
+# RCC_DIR     = $$PWD/.rcc
+# UI_DIR      = $$PWD/.ui
+# MOC_DIR     = $$PWD/.moc
+# OBJECTS_DIR = $$PWD/.obj
 
 exists(../.git/HEAD) {
     GITVERSION=$$system(git log -n1 --pretty=format:%h)
diff --git a/src/src.pro b/src/src.pro
index eabac41..c89e439 100644
--- a/src/src.pro
+++ b/src/src.pro
@@ -15,7 +15,7 @@ include(src.pri)
 
 SOURCES += main.cpp
 
-DESTDIR = ../
+# DESTDIR = ../
 
 include(locale/locale.pri)
 
