diff --git a/src/locale/en_GB.ts b/src/locale/en_GB.ts
new file mode 100644
index 0000000..b8a5fa6
--- /dev/null
+++ b/src/locale/en_GB.ts
@@ -0,0 +1,2544 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!DOCTYPE TS>
+<TS version="2.1" language="en_GB">
+<context>
+    <name>AboutDialog</name>
+    <message>
+        <location filename="../aboutdialog.ui" line="117"/>
+        <source>Lightweight WebKit-based web browser</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../aboutdialog.ui" line="136"/>
+        <source>&lt;!doctype html&gt;
+&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
+p, li { white-space: pre-wrap; }
+&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot;font-size:9pt;&quot;&gt;
+&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Copyright © 2007-2010 Benjamin C. Meyer &amp;lt;&lt;a href=&quot;mailto:ben@meyerhome.net&quot;&gt;&lt;span style=&quot; text-decoration: underline; color:#0057ae;&quot;&gt;ben@meyerhome.net&lt;/span&gt;&lt;/a&gt;&amp;gt;&lt;/p&gt;
+&lt;br&gt;NOTE: This is an inofficial version of arora and not made by Benjamin C. Meyer.&lt;/body&gt;&lt;/html&gt;</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../aboutdialog.ui" line="163"/>
+        <source>&lt;a href=&quot;https://github.com/RaspberryPiFan/arora&quot;&gt;GitHub Repository&lt;/a&gt;</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../aboutdialog.ui" line="194"/>
+        <location filename="../aboutdialog.cpp" line="83"/>
+        <source>Authors</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../aboutdialog.ui" line="201"/>
+        <location filename="../aboutdialog.cpp" line="88"/>
+        <source>License</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../aboutdialog.ui" line="208"/>
+        <source>Close</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../aboutdialog.cpp" line="37"/>
+        <source>About %1</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../aboutdialog.cpp" line="41"/>
+        <source>WebKit version: %1</source>
+        <translation></translation>
+    </message>
+</context>
+<context>
+    <name>AcceptLanguage</name>
+    <message>
+        <location filename="../acceptlanguagedialog.ui" line="14"/>
+        <source>Languages</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../acceptlanguagedialog.ui" line="20"/>
+        <source>Languages: in order of preference:</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../acceptlanguagedialog.ui" line="33"/>
+        <source>Move &amp;Up</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../acceptlanguagedialog.ui" line="43"/>
+        <source>Move &amp;Down</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../acceptlanguagedialog.ui" line="53"/>
+        <source>&amp;Remove</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../acceptlanguagedialog.ui" line="76"/>
+        <source>Add...</source>
+        <translation></translation>
+    </message>
+</context>
+<context>
+    <name>AdBlockBlockedNetworkReply</name>
+    <message>
+        <location filename="../adblock/adblockblockednetworkreply.cpp" line="42"/>
+        <source>Blocked by AdBlockRule: %1</source>
+        <translation></translation>
+    </message>
+</context>
+<context>
+    <name>AdBlockDialog</name>
+    <message>
+        <location filename="../adblock/adblockdialog.ui" line="14"/>
+        <source>AdBlock Configuration</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../adblock/adblockdialog.ui" line="20"/>
+        <source>Enable AdBlock</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../adblock/adblockdialog.ui" line="68"/>
+        <source>Action</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../adblock/adblockdialog.cpp" line="75"/>
+        <source>Add Custom Rule</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../adblock/adblockdialog.cpp" line="78"/>
+        <source>Learn more about writing rules...</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../adblock/adblockdialog.cpp" line="85"/>
+        <source>Update Subscription</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../adblock/adblockdialog.cpp" line="90"/>
+        <source>Browse Subscriptions...</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../adblock/adblockdialog.cpp" line="95"/>
+        <source>Remove Subscription</source>
+        <translation></translation>
+    </message>
+</context>
+<context>
+    <name>AdBlockManager</name>
+    <message>
+        <location filename="../adblock/adblockmanager.cpp" line="119"/>
+        <source>Custom Rules</source>
+        <translation></translation>
+    </message>
+</context>
+<context>
+    <name>AdBlockModel</name>
+    <message>
+        <location filename="../adblock/adblockmodel.cpp" line="83"/>
+        <source>Rule</source>
+        <translation></translation>
+    </message>
+</context>
+<context>
+    <name>AdBlockSchemeAccessHandler</name>
+    <message>
+        <location filename="../adblock/adblockschemeaccesshandler.cpp" line="55"/>
+        <source>Subscribe?</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../adblock/adblockschemeaccesshandler.cpp" line="56"/>
+        <source>Subscribe to this AdBlock subscription?
+%1</source>
+        <translation></translation>
+    </message>
+</context>
+<context>
+    <name>AddBookmarkDialog</name>
+    <message>
+        <location filename="../bookmarks/addbookmarkdialog.ui" line="14"/>
+        <location filename="../bookmarks/addbookmarkdialog.cpp" line="173"/>
+        <source>Add Bookmark</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../bookmarks/addbookmarkdialog.ui" line="20"/>
+        <source>Type a name for the bookmark, and choose where to keep it.</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../bookmarks/addbookmarkdialog.cpp" line="120"/>
+        <source>Url</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../bookmarks/addbookmarkdialog.cpp" line="121"/>
+        <source>Title</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../bookmarks/addbookmarkdialog.cpp" line="170"/>
+        <source>Add Folder</source>
+        <translation></translation>
+    </message>
+</context>
+<context>
+    <name>AutoFillDialog</name>
+    <message>
+        <location filename="../autofilldialog.ui" line="14"/>
+        <source>Form Passwords</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../autofilldialog.ui" line="43"/>
+        <source>Remove</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../autofilldialog.ui" line="50"/>
+        <source>Remove All</source>
+        <translation></translation>
+    </message>
+</context>
+<context>
+    <name>AutoFillManager</name>
+    <message>
+        <location filename="../autofillmanager.cpp" line="199"/>
+        <source>&lt;b&gt;Would you like to save this password?&lt;/b&gt;&lt;br&gt;         To review passwords you have saved and remove them, open the AutoFill panel of preferences.</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../autofillmanager.cpp" line="201"/>
+        <source>Never for this site</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../autofillmanager.cpp" line="202"/>
+        <source>Not now</source>
+        <translation></translation>
+    </message>
+</context>
+<context>
+    <name>AutoFillModel</name>
+    <message>
+        <location filename="../autofilldialog.cpp" line="58"/>
+        <source>WebSite</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../autofilldialog.cpp" line="59"/>
+        <source>User Name</source>
+        <translation></translation>
+    </message>
+</context>
+<context>
+    <name>BookmarksDialog</name>
+    <message>
+        <location filename="../bookmarks/bookmarksdialog.ui" line="13"/>
+        <source>Bookmarks</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../bookmarks/bookmarksdialog.ui" line="40"/>
+        <source>&amp;Remove</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../bookmarks/bookmarksdialog.ui" line="47"/>
+        <source>Add Folder</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../bookmarks/bookmarksdialog.cpp" line="160"/>
+        <source>Open</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../bookmarks/bookmarksdialog.cpp" line="161"/>
+        <source>Open in New Tab</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../bookmarks/bookmarksdialog.cpp" line="165"/>
+        <source>Edit Name</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../bookmarks/bookmarksdialog.cpp" line="168"/>
+        <source>Edit Address</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../bookmarks/bookmarksdialog.cpp" line="171"/>
+        <source>Delete</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../bookmarks/bookmarksdialog.cpp" line="238"/>
+        <source>New Folder</source>
+        <translation></translation>
+    </message>
+</context>
+<context>
+    <name>BookmarksManager</name>
+    <message>
+        <location filename="../bookmarks/bookmarksmanager.cpp" line="89"/>
+        <source>Bookmarks Bar</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../bookmarks/bookmarksmanager.cpp" line="90"/>
+        <source>Bookmarks Menu</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../bookmarks/bookmarksmanager.cpp" line="135"/>
+        <location filename="../bookmarks/bookmarksmanager.cpp" line="316"/>
+        <source>Error when loading bookmarks on line %1, column %2:
+%3</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../bookmarks/bookmarksmanager.cpp" line="144"/>
+        <source>Toolbar Bookmarks</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../bookmarks/bookmarksmanager.cpp" line="152"/>
+        <source>Menu</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../bookmarks/bookmarksmanager.cpp" line="282"/>
+        <location filename="../bookmarks/bookmarksmanager.cpp" line="331"/>
+        <source>XBEL bookmarks</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../bookmarks/bookmarksmanager.cpp" line="283"/>
+        <source>HTML Netscape bookmarks</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../bookmarks/bookmarksmanager.cpp" line="285"/>
+        <source>Open File</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../bookmarks/bookmarksmanager.cpp" line="301"/>
+        <source>htmlToXBel tool required</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../bookmarks/bookmarksmanager.cpp" line="302"/>
+        <source>htmlToXBel tool, which is shipped with Arora and is needed to import HTML bookmarks, is not installed or not available in the search paths.</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../bookmarks/bookmarksmanager.cpp" line="305"/>
+        <source>Loading Bookmark</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../bookmarks/bookmarksmanager.cpp" line="306"/>
+        <source>Error when loading HTML bookmarks: %1
+</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../bookmarks/bookmarksmanager.cpp" line="323"/>
+        <source>Imported %1</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../bookmarks/bookmarksmanager.cpp" line="329"/>
+        <source>Save File</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../bookmarks/bookmarksmanager.cpp" line="330"/>
+        <source>%1 Bookmarks.xbel</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../bookmarks/bookmarksmanager.cpp" line="337"/>
+        <source>Export error</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../bookmarks/bookmarksmanager.cpp" line="337"/>
+        <source>error saving bookmarks</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../bookmarks/bookmarksmanager.cpp" line="341"/>
+        <source>Remove Bookmark</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../bookmarks/bookmarksmanager.cpp" line="375"/>
+        <source>Insert Bookmark</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../bookmarks/bookmarksmanager.cpp" line="389"/>
+        <source>Name Change</source>
+        <comment>Undo bookmark title change</comment>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../bookmarks/bookmarksmanager.cpp" line="392"/>
+        <source>Address Change</source>
+        <comment>Undo bookmark url change</comment>
+        <translation></translation>
+    </message>
+</context>
+<context>
+    <name>BookmarksMenu</name>
+    <message>
+        <location filename="../bookmarks/bookmarksmenu.cpp" line="115"/>
+        <source>Open in Tabs</source>
+        <translation></translation>
+    </message>
+</context>
+<context>
+    <name>BookmarksModel</name>
+    <message>
+        <location filename="../bookmarks/bookmarksmodel.cpp" line="150"/>
+        <source>Title</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../bookmarks/bookmarksmodel.cpp" line="151"/>
+        <source>Address</source>
+        <translation></translation>
+    </message>
+</context>
+<context>
+    <name>BookmarksToolBar</name>
+    <message>
+        <location filename="../bookmarks/bookmarkstoolbar.cpp" line="63"/>
+        <source>Open</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../bookmarks/bookmarkstoolbar.cpp" line="66"/>
+        <source>Open in New &amp;Tab</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../bookmarks/bookmarkstoolbar.cpp" line="72"/>
+        <source>Remove</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../bookmarks/bookmarkstoolbar.cpp" line="78"/>
+        <source>Add Bookmark...</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../bookmarks/bookmarkstoolbar.cpp" line="79"/>
+        <source>Add Folder...</source>
+        <translation></translation>
+    </message>
+</context>
+<context>
+    <name>BrowserApplication</name>
+    <message>
+        <location filename="../browserapplication.cpp" line="287"/>
+        <source>There are %1 windows and %2 tabs open
+Do you want to quit anyway?</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browserapplication.cpp" line="463"/>
+        <source>Restore failed</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browserapplication.cpp" line="464"/>
+        <source>Arora crashed while trying to restore this session.  Should I try again?</source>
+        <translation></translation>
+    </message>
+</context>
+<context>
+    <name>BrowserMainWindow</name>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="879"/>
+        <source>Hide Toolbar</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="879"/>
+        <source>Show Toolbar</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="880"/>
+        <source>Hide Bookmarks Bar</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="880"/>
+        <source>Show Bookmarks Bar</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="881"/>
+        <source>Hide Status Bar</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="881"/>
+        <source>Show Status Bar</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="900"/>
+        <source>Default</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="930"/>
+        <source>&amp;File</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="931"/>
+        <source>&amp;New Window</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="932"/>
+        <source>&amp;Open File...</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="933"/>
+        <source>Open &amp;Location...</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="934"/>
+        <source>&amp;Save As...</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="935"/>
+        <source>&amp;Import Bookmarks...</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="936"/>
+        <source>&amp;Export Bookmarks...</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="937"/>
+        <source>P&amp;rint Preview...</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="938"/>
+        <source>&amp;Print...</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="939"/>
+        <source>Private &amp;Browsing...</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="940"/>
+        <source>Close Window</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="941"/>
+        <source>&amp;Quit</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="943"/>
+        <source>&amp;Edit</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="944"/>
+        <source>&amp;Undo</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="945"/>
+        <source>&amp;Redo</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="946"/>
+        <source>Cu&amp;t</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="947"/>
+        <source>&amp;Copy</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="948"/>
+        <source>&amp;Paste</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="949"/>
+        <source>Select &amp;All</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="950"/>
+        <source>&amp;Find</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="951"/>
+        <source>Find Nex&amp;t</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="952"/>
+        <source>Find P&amp;revious</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="954"/>
+        <source>&amp;View</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="955"/>
+        <source>Ctrl+|</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="956"/>
+        <source>Alt+Ctrl+B</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="957"/>
+        <source>Ctrl+/</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="958"/>
+        <source>Show Menu Bar</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="959"/>
+        <source>&amp;Reload Page</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="960"/>
+        <source>&amp;Stop</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="961"/>
+        <source>Zoom &amp;In</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="962"/>
+        <source>Zoom &amp;Normal</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="963"/>
+        <source>Zoom &amp;Out</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="964"/>
+        <source>Zoom &amp;Text Only</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="965"/>
+        <source>Page S&amp;ource</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="966"/>
+        <source>Ctrl+Alt+U</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="967"/>
+        <source>&amp;Full Screen</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="968"/>
+        <source>Text Encoding</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="970"/>
+        <source>Hi&amp;story</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="971"/>
+        <source>Back</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="972"/>
+        <source>Forward</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="973"/>
+        <source>Home</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="974"/>
+        <source>Restore Last Session</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="976"/>
+        <location filename="../browsermainwindow.cpp" line="1002"/>
+        <source>&amp;Bookmarks</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="977"/>
+        <source>Show All Bookmarks...</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="978"/>
+        <source>Add Bookmark...</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="979"/>
+        <source>Add Folder...</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="981"/>
+        <source>&amp;Window</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="983"/>
+        <source>&amp;Tools</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="984"/>
+        <source>Web &amp;Search</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="985"/>
+        <source>Ctrl+K</source>
+        <comment>Web Search</comment>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="986"/>
+        <source>&amp;Clear Private Data</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="987"/>
+        <source>Ctrl+Shift+Delete</source>
+        <comment>Clear Private Data</comment>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="988"/>
+        <source>Enable Web &amp;Inspector</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="989"/>
+        <source>Options...</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="990"/>
+        <source>Ctrl+,</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="991"/>
+        <source>Configure Search Engines...</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="992"/>
+        <source>User Agent</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="993"/>
+        <source>&amp;Ad Block...</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="995"/>
+        <source>&amp;Help</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="996"/>
+        <source>Switch application language </source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="997"/>
+        <source>About &amp;Qt</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="998"/>
+        <source>About &amp;%1</source>
+        <comment>About Browser</comment>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="1001"/>
+        <source>Navigation</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="1004"/>
+        <source>Reload / Stop</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="1174"/>
+        <source>%1 - Arora</source>
+        <comment>Page title and Browser name</comment>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="1200"/>
+        <source>Open Web Resource</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="1201"/>
+        <source>Web Resources (*.html *.htm *.svg *.png *.gif *.svgz);;All files (*.*)</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="1230"/>
+        <source>Print Document</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="1239"/>
+        <source>Are you sure you want to turn on private browsing?</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="1240"/>
+        <source>When private browsing is turned on, some actions concerning your privacy will be disabled:</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="1243"/>
+        <source>Webpages are not added to the history.</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="1244"/>
+        <source>Items are automatically removed from the Downloads window.</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="1245"/>
+        <source>New cookies are not stored, current cookies can&apos;t be accessed.</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="1246"/>
+        <source>Site icons won&apos;t be stored.</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="1247"/>
+        <source>Session won&apos;t be saved.</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="1248"/>
+        <source>Searches are not added to the pop-up menu in the search box.</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="1249"/>
+        <source>No new network cache is written to disk.</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="1251"/>
+        <source>Until you close the window, you can still click the Back and Forward buttons to return to the webpages you have opened.</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="1257"/>
+        <source>Private Browsing</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="1297"/>
+        <source>Are you sure you want to close the window?  There are %1 tabs open</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="1423"/>
+        <source>Web Inspector</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="1424"/>
+        <source>The web inspector will only work correctly for pages that were loaded after enabling.
+Do you want to reload all pages?</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="1464"/>
+        <source>Stop loading the current page</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="1465"/>
+        <source>Stop</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="1467"/>
+        <source>Reload the current page</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="1468"/>
+        <source>Reload</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="1535"/>
+        <source>Downloads</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../browsermainwindow.cpp" line="1535"/>
+        <source>Ctrl+Y</source>
+        <comment>Download Manager</comment>
+        <translation></translation>
+    </message>
+</context>
+<context>
+    <name>ClearButton</name>
+    <message>
+        <location filename="../clearbutton.cpp" line="38"/>
+        <source>Clear</source>
+        <translation></translation>
+    </message>
+</context>
+<context>
+    <name>ClearPrivateData</name>
+    <message>
+        <location filename="../clearprivatedata.cpp" line="43"/>
+        <source>Clear Private Data</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../clearprivatedata.cpp" line="46"/>
+        <source>Clear the following items:</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../clearprivatedata.cpp" line="51"/>
+        <source>&amp;Browsing History</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../clearprivatedata.cpp" line="55"/>
+        <source>&amp;Download History</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../clearprivatedata.cpp" line="59"/>
+        <source>&amp;Search History</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../clearprivatedata.cpp" line="63"/>
+        <source>&amp;Cookies</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../clearprivatedata.cpp" line="67"/>
+        <source>C&amp;ached Web Pages</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../clearprivatedata.cpp" line="75"/>
+        <source>Website &amp;Icons</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../clearprivatedata.cpp" line="81"/>
+        <source>Clear &amp;Private Data</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../clearprivatedata.cpp" line="83"/>
+        <source>&amp;Cancel</source>
+        <translation></translation>
+    </message>
+</context>
+<context>
+    <name>CookieExceptionsModel</name>
+    <message>
+        <location filename="../network/cookiejar/cookieexceptionsmodel.cpp" line="93"/>
+        <source>Website</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../network/cookiejar/cookieexceptionsmodel.cpp" line="95"/>
+        <source>Rule</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../network/cookiejar/cookieexceptionsmodel.cpp" line="115"/>
+        <source>Allow</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../network/cookiejar/cookieexceptionsmodel.cpp" line="124"/>
+        <source>Block</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../network/cookiejar/cookieexceptionsmodel.cpp" line="133"/>
+        <source>Allow For Session</source>
+        <translation></translation>
+    </message>
+</context>
+<context>
+    <name>CookieModel</name>
+    <message>
+        <location filename="../network/cookiejar/cookiemodel.cpp" line="96"/>
+        <source>Website</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../network/cookiejar/cookiemodel.cpp" line="98"/>
+        <source>Name</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../network/cookiejar/cookiemodel.cpp" line="100"/>
+        <source>Path</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../network/cookiejar/cookiemodel.cpp" line="102"/>
+        <source>Secure</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../network/cookiejar/cookiemodel.cpp" line="104"/>
+        <source>Expires</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../network/cookiejar/cookiemodel.cpp" line="106"/>
+        <source>Contents</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../network/cookiejar/cookiemodel.cpp" line="149"/>
+        <source>true</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../network/cookiejar/cookiemodel.cpp" line="149"/>
+        <source>false</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../network/cookiejar/cookiemodel.cpp" line="151"/>
+        <source>Session cookie</source>
+        <translation></translation>
+    </message>
+</context>
+<context>
+    <name>CookiesDialog</name>
+    <message>
+        <location filename="../network/cookiejar/cookies.ui" line="14"/>
+        <source>Cookies</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../network/cookiejar/cookies.ui" line="41"/>
+        <source>&amp;Remove</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../network/cookiejar/cookies.ui" line="48"/>
+        <source>Remove &amp;All Cookies</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../network/cookiejar/cookies.ui" line="55"/>
+        <source>Add &amp;Rule</source>
+        <translation></translation>
+    </message>
+</context>
+<context>
+    <name>CookiesExceptionsDialog</name>
+    <message>
+        <location filename="../network/cookiejar/cookiesexceptions.ui" line="13"/>
+        <source>Cookie Exceptions</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../network/cookiejar/cookiesexceptions.ui" line="19"/>
+        <source>New Exception</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../network/cookiejar/cookiesexceptions.ui" line="27"/>
+        <source>Domain:</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../network/cookiejar/cookiesexceptions.ui" line="57"/>
+        <source>Block</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../network/cookiejar/cookiesexceptions.ui" line="67"/>
+        <source>Allow For Session</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../network/cookiejar/cookiesexceptions.ui" line="77"/>
+        <source>Allow</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../network/cookiejar/cookiesexceptions.ui" line="89"/>
+        <source>Exceptions</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../network/cookiejar/cookiesexceptions.ui" line="114"/>
+        <source>&amp;Remove</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../network/cookiejar/cookiesexceptions.ui" line="121"/>
+        <source>Remove &amp;All</source>
+        <translation></translation>
+    </message>
+</context>
+<context>
+    <name>DownloadDialog</name>
+    <message>
+        <location filename="../downloads.ui" line="14"/>
+        <location filename="../downloadmanager.cpp" line="606"/>
+        <source>Downloads</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../downloads.ui" line="44"/>
+        <source>Clean up</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../downloads.ui" line="66"/>
+        <source>0 Items</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../downloadmanager.cpp" line="604"/>
+        <source>Downloading %1</source>
+        <translation></translation>
+    </message>
+</context>
+<context>
+    <name>DownloadItem</name>
+    <message>
+        <location filename="../downloaditem.ui" line="25"/>
+        <source>Ico</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../downloaditem.ui" line="40"/>
+        <source>Filename</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../downloaditem.ui" line="87"/>
+        <source>Try Again</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../downloaditem.ui" line="94"/>
+        <source>Stop</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../downloaditem.ui" line="101"/>
+        <source>Open</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../downloadmanager.cpp" line="170"/>
+        <source>Save File</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../downloadmanager.cpp" line="175"/>
+        <source>Download cancelled: %1</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../downloadmanager.cpp" line="191"/>
+        <source>Download directory (%1) couldn&apos;t be created.</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../downloadmanager.cpp" line="293"/>
+        <source>Error opening output file: %1</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../downloadmanager.cpp" line="302"/>
+        <source>Error saving: %1</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../downloadmanager.cpp" line="318"/>
+        <source>Network Error: %1</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../downloadmanager.cpp" line="414"/>
+        <source>%1 of %2 (%3/sec) - %4</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../downloadmanager.cpp" line="416"/>
+        <source>?</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../downloadmanager.cpp" line="423"/>
+        <source>%1 of %2 - Download Complete</source>
+        <translation></translation>
+    </message>
+</context>
+<context>
+    <name>DownloadManager</name>
+    <message>
+        <location filename="../downloadmanager.cpp" line="510"/>
+        <source>There are %1 downloads in progress
+Do you want to quit anyway?</source>
+        <translation></translation>
+    </message>
+    <message numerus="yes">
+        <location filename="../downloadmanager.cpp" line="749"/>
+        <source>%n Download(s)</source>
+        <translation>
+            <numerusform></numerusform>
+            <numerusform></numerusform>
+        </translation>
+    </message>
+    <message numerus="yes">
+        <location filename="../downloadmanager.cpp" line="771"/>
+        <source>%n minutes remaining</source>
+        <translation>
+            <numerusform></numerusform>
+            <numerusform></numerusform>
+        </translation>
+    </message>
+    <message numerus="yes">
+        <location filename="../downloadmanager.cpp" line="775"/>
+        <source>%n seconds remaining</source>
+        <translation>
+            <numerusform></numerusform>
+            <numerusform></numerusform>
+        </translation>
+    </message>
+    <message>
+        <location filename="../downloadmanager.cpp" line="788"/>
+        <source>bytes</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../downloadmanager.cpp" line="791"/>
+        <source>kB</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../downloadmanager.cpp" line="794"/>
+        <source>MB</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../downloadmanager.cpp" line="797"/>
+        <source>GB</source>
+        <translation></translation>
+    </message>
+</context>
+<context>
+    <name>FileAccessReply</name>
+    <message>
+        <location filename="../network/fileaccesshandler.cpp" line="69"/>
+        <source>No Error</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../network/fileaccesshandler.cpp" line="119"/>
+        <source>Error opening: %1: No such file or directory</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../network/fileaccesshandler.cpp" line="125"/>
+        <source>Unable to read %1</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../network/fileaccesshandler.cpp" line="136"/>
+        <source>Contents of %1</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../network/fileaccesshandler.cpp" line="178"/>
+        <source>%1 KB</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../network/fileaccesshandler.cpp" line="188"/>
+        <source>Show Hidden Files</source>
+        <translation></translation>
+    </message>
+</context>
+<context>
+    <name>HistoryDialog</name>
+    <message>
+        <location filename="../history/history.ui" line="13"/>
+        <source>History</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../history/history.ui" line="40"/>
+        <source>&amp;Remove</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../history/history.ui" line="47"/>
+        <source>Remove &amp;All</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../history/history.cpp" line="450"/>
+        <source>Open</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../history/history.cpp" line="452"/>
+        <source>Copy</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../history/history.cpp" line="454"/>
+        <source>Delete</source>
+        <translation></translation>
+    </message>
+</context>
+<context>
+    <name>HistoryMenu</name>
+    <message>
+        <location filename="../history/history.cpp" line="375"/>
+        <source>Show All History</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../history/history.cpp" line="382"/>
+        <source>Clear History...</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../history/history.cpp" line="398"/>
+        <source>Clear History</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../history/history.cpp" line="398"/>
+        <source>Do you want to clear the history?</source>
+        <translation></translation>
+    </message>
+</context>
+<context>
+    <name>HistoryModel</name>
+    <message>
+        <location filename="../history/history.cpp" line="133"/>
+        <source>Title</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../history/history.cpp" line="134"/>
+        <source>Address</source>
+        <translation></translation>
+    </message>
+</context>
+<context>
+    <name>HistoryTreeModel</name>
+    <message>
+        <location filename="../history/history.cpp" line="751"/>
+        <source>Earlier Today</source>
+        <translation></translation>
+    </message>
+    <message numerus="yes">
+        <location filename="../history/history.cpp" line="755"/>
+        <source>%n item(s)</source>
+        <translation>
+            <numerusform></numerusform>
+            <numerusform></numerusform>
+        </translation>
+    </message>
+</context>
+<context>
+    <name>JavaScriptAroraObject</name>
+    <message>
+        <location filename="../webpage.cpp" line="63"/>
+        <source>Welcome to Arora!</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../webpage.cpp" line="64"/>
+        <source>Arora Start</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../webpage.cpp" line="65"/>
+        <source>Search!</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../webpage.cpp" line="66"/>
+        <source>Search results provided by</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../webpage.cpp" line="67"/>
+        <source>About Arora</source>
+        <translation></translation>
+    </message>
+</context>
+<context>
+    <name>LanguageManager</name>
+    <message>
+        <location filename="../utils/languagemanager.cpp" line="240"/>
+        <source>No translation files are installed at %1.</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../utils/languagemanager.cpp" line="268"/>
+        <source>Choose language</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../utils/languagemanager.cpp" line="269"/>
+        <source>&lt;p&gt;You can run with a different language than&lt;br&gt;the operating system default.&lt;/p&gt;&lt;p&gt;Please choose the language which should be used&lt;/p&gt;</source>
+        <translation></translation>
+    </message>
+</context>
+<context>
+    <name>NetworkAccessManager</name>
+    <message>
+        <location filename="../network/networkaccessmanager.cpp" line="215"/>
+        <source>&lt;qt&gt;Enter username and password for &quot;%1&quot; at %2&lt;/qt&gt;</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../network/networkaccessmanager.cpp" line="242"/>
+        <source>&lt;qt&gt;Connect to proxy &quot;%1&quot; using:&lt;/qt&gt;</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../network/networkaccessmanager.cpp" line="259"/>
+        <source>Issuer: %1</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../network/networkaccessmanager.cpp" line="261"/>
+        <source>Not valid before: %1</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../network/networkaccessmanager.cpp" line="262"/>
+        <source>Valid until: %1</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../network/networkaccessmanager.cpp" line="268"/>
+        <source>Alternate Names:</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../network/networkaccessmanager.cpp" line="307"/>
+        <source> - SSL Errors</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../network/networkaccessmanager.cpp" line="308"/>
+        <source>&lt;qt&gt;SSL Errors:&lt;br/&gt;&lt;br/&gt;for: &lt;tt&gt;%1&lt;/tt&gt;&lt;ul&gt;&lt;li&gt;%2&lt;/li&gt;&lt;/ul&gt;
+
+Do you want to ignore these errors?&lt;/qt&gt;</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../network/networkaccessmanager.cpp" line="321"/>
+        <source>&lt;qt&gt;Certificates:&lt;br/&gt;%1&lt;br/&gt;Do you want to accept all these certificates?&lt;/qt&gt;</source>
+        <translation></translation>
+    </message>
+</context>
+<context>
+    <name>OpenSearchDialog</name>
+    <message>
+        <location filename="../opensearch/opensearchdialog.ui" line="14"/>
+        <source>OpenSearch Manager</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../opensearch/opensearchdialog.ui" line="31"/>
+        <source>&amp;Add</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../opensearch/opensearchdialog.ui" line="38"/>
+        <source>&amp;Delete</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../opensearch/opensearchdialog.ui" line="45"/>
+        <source>&amp;Restore Defaults</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../opensearch/opensearchdialog.ui" line="64"/>
+        <source>&amp;Close</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../opensearch/opensearchdialog.cpp" line="61"/>
+        <source>Open File</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../opensearch/opensearchdialog.cpp" line="63"/>
+        <source>OpenSearch</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../opensearch/opensearchdialog.cpp" line="67"/>
+        <location filename="../opensearch/opensearchdialog.cpp" line="76"/>
+        <source>Error</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../opensearch/opensearchdialog.cpp" line="68"/>
+        <source>%1 is not a valid OpenSearch 1.1 description or is already on your list.</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../opensearch/opensearchdialog.cpp" line="77"/>
+        <source>You must have at least one search engine in here.</source>
+        <translation></translation>
+    </message>
+</context>
+<context>
+    <name>OpenSearchEngineModel</name>
+    <message>
+        <location filename="../opensearch/opensearchenginemodel.cpp" line="109"/>
+        <source>&lt;strong&gt;Description:&lt;/strong&gt; %1</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../opensearch/opensearchenginemodel.cpp" line="113"/>
+        <source>&lt;strong&gt;Provides contextual suggestions&lt;/strong&gt;</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../opensearch/opensearchenginemodel.cpp" line="127"/>
+        <source>Comma-separated list of keywords that may be entered in the location barfollowed by search terms to search with this engine</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../opensearch/opensearchenginemodel.cpp" line="165"/>
+        <source>Name</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../opensearch/opensearchenginemodel.cpp" line="167"/>
+        <source>Keywords</source>
+        <translation></translation>
+    </message>
+</context>
+<context>
+    <name>OpenSearchManager</name>
+    <message>
+        <location filename="../opensearch/opensearchmanager.cpp" line="312"/>
+        <source>Do you want to add the following engine to your list of search engines?&lt;br /&gt;&lt;br /&gt;Name: %1&lt;br /&gt;Searches on: %2</source>
+        <translation></translation>
+    </message>
+</context>
+<context>
+    <name>PasswordDialog</name>
+    <message>
+        <location filename="../network/passworddialog.ui" line="13"/>
+        <source>Authentication Required</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../network/passworddialog.ui" line="21"/>
+        <source>DUMMY ICON</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../network/passworddialog.ui" line="34"/>
+        <source>INTRO TEXT DUMMY</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../network/passworddialog.ui" line="43"/>
+        <source>Username:</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../network/passworddialog.ui" line="53"/>
+        <source>Password:</source>
+        <translation></translation>
+    </message>
+</context>
+<context>
+    <name>PlainTextEditSearch</name>
+    <message>
+        <location filename="../plaintexteditsearch.cpp" line="62"/>
+        <source>Not Found</source>
+        <translation></translation>
+    </message>
+</context>
+<context>
+    <name>ProxyDialog</name>
+    <message>
+        <location filename="../network/proxy.ui" line="14"/>
+        <source>Proxy Authentication</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../network/proxy.ui" line="27"/>
+        <source>Connect to proxy</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../network/proxy.ui" line="37"/>
+        <source>Username:</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../network/proxy.ui" line="47"/>
+        <source>Password:</source>
+        <translation></translation>
+    </message>
+</context>
+<context>
+    <name>QObject</name>
+    <message>
+        <location filename="../bookmarks/xbel/xbelreader.cpp" line="111"/>
+        <source>The file is not an XBEL version 1.0 file.</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../bookmarks/xbel/xbelreader.cpp" line="209"/>
+        <source>Unknown title</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../opensearch/opensearchreader.cpp" line="89"/>
+        <source>The file is not an OpenSearch 1.1 file.</source>
+        <translation></translation>
+    </message>
+</context>
+<context>
+    <name>SearchBanner</name>
+    <message>
+        <location filename="../searchbanner.ui" line="51"/>
+        <source>Highlight All</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../searchbanner.ui" line="81"/>
+        <source>Done</source>
+        <translation></translation>
+    </message>
+</context>
+<context>
+    <name>SearchLineEdit</name>
+    <message>
+        <location filename="../searchlineedit.cpp" line="56"/>
+        <source>Search</source>
+        <translation></translation>
+    </message>
+</context>
+<context>
+    <name>Settings</name>
+    <message>
+        <location filename="../settings.ui" line="14"/>
+        <source>Preferences</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="34"/>
+        <source>General</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="40"/>
+        <source>On startup:</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="51"/>
+        <source>Show my home page</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="56"/>
+        <source>Show a blank page</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="61"/>
+        <source>Restore windows and tabs from last time</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="69"/>
+        <source>Home Page:</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="86"/>
+        <source>Set to current page</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="106"/>
+        <source>Remove history items:</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="117"/>
+        <source>After one day</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="122"/>
+        <source>After one week</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="127"/>
+        <source>After two weeks</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="132"/>
+        <source>After one month</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="137"/>
+        <source>After one year</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="142"/>
+        <source>Manually</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="147"/>
+        <source>On application exit</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="155"/>
+        <source>Use the default search engine as fallback when the URL given by the user is invalid</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="165"/>
+        <source>Enable access keys</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="172"/>
+        <source>Downloads</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="178"/>
+        <source>Ask for a destination each time</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="187"/>
+        <source>Use this destination:</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="204"/>
+        <source>Choose Directory...</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="215"/>
+        <source>Use this external download program:</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="225"/>
+        <source>Choose Program...</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="251"/>
+        <source>Appearance</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="257"/>
+        <source>Standard font:</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="286"/>
+        <location filename="../settings.ui" line="316"/>
+        <location filename="../settings.ui" line="330"/>
+        <source>Select...</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="293"/>
+        <source>Fixed-width font:</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="323"/>
+        <source>Preferred languages for viewing webpages in:</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="337"/>
+        <source>Minimum Font size:</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="371"/>
+        <source>Privacy</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="377"/>
+        <source>Web Content</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="383"/>
+        <source>Block Popup Windows</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="393"/>
+        <source>Enable Plugins</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="419"/>
+        <source>Enable Javascript</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="429"/>
+        <source>View Images</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="439"/>
+        <source>Persistent Data Storage</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="452"/>
+        <source>Cookies</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="458"/>
+        <source>Accept Cookies:</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="469"/>
+        <source>Always</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="474"/>
+        <source>Never</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="479"/>
+        <source>Only from sites you navigate to</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="487"/>
+        <source>Exceptions...</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="494"/>
+        <source>Keep Cookies Until:</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="505"/>
+        <source>They expire</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="510"/>
+        <source>I exit the application</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="515"/>
+        <source>At most 90 days</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="523"/>
+        <source>Cookies...</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="543"/>
+        <source>Filter Tracking Cookies</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="550"/>
+        <source>A cookie session ends:</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="561"/>
+        <source>When I exit the application</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="566"/>
+        <source>1 day</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="571"/>
+        <source>2 days</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="576"/>
+        <source>3 days</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="581"/>
+        <source>7 days</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="586"/>
+        <source>30 days</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="611"/>
+        <source>Tabs</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="617"/>
+        <source>Select tabs and windows as they are created</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="624"/>
+        <source>Confirm when closing multiple tabs or windows</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="634"/>
+        <source>Show only one close button instead of one for each tab</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="644"/>
+        <source>Quit the application when last tab is closed</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="654"/>
+        <source>Opening links</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="660"/>
+        <source>Links that want to open in a new window:</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="671"/>
+        <location filename="../settings.ui" line="705"/>
+        <source>In a new window</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="676"/>
+        <location filename="../settings.ui" line="710"/>
+        <source>In a new selected tab in the current window</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="681"/>
+        <location filename="../settings.ui" line="715"/>
+        <source>In a new tab in the current window</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="686"/>
+        <location filename="../settings.ui" line="720"/>
+        <source>In the current tab</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="694"/>
+        <source>Open links from applications:</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="745"/>
+        <source>Proxy</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="751"/>
+        <source>Use proxy server</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="760"/>
+        <source>Type:</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="771"/>
+        <source>Socks5</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="776"/>
+        <source>Http (Secure)</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="781"/>
+        <source>Http (Transparent)</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="789"/>
+        <source>Host name:</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="806"/>
+        <source>Port:</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="839"/>
+        <source>User Name:</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="856"/>
+        <source>Password:</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="893"/>
+        <source>AutoFill</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="899"/>
+        <source>AutoFill web forms:</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="912"/>
+        <source>User names and passwords</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="919"/>
+        <source>Edit...</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="953"/>
+        <source>Advanced</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="959"/>
+        <source>Style Sheet:</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="973"/>
+        <source>Browse...</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="980"/>
+        <source>Enable network cache</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="992"/>
+        <source>Maximum Size:</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.ui" line="999"/>
+        <source> MB</source>
+        <translation></translation>
+    </message>
+</context>
+<context>
+    <name>SettingsDialog</name>
+    <message>
+        <location filename="../settings.cpp" line="450"/>
+        <source>Restart required</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.cpp" line="451"/>
+        <source>The network cache configuration has changed. So that it can be taken into account, the browser has to be restarted.</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.cpp" line="471"/>
+        <source>Choose Directory</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.cpp" line="477"/>
+        <source>Choose Program</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../settings.cpp" line="520"/>
+        <source>Choose CSS File</source>
+        <translation></translation>
+    </message>
+</context>
+<context>
+    <name>SourceViewer</name>
+    <message>
+        <location filename="../sourceviewer.cpp" line="42"/>
+        <source>Loading...</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../sourceviewer.cpp" line="47"/>
+        <source>&amp;Edit</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../sourceviewer.cpp" line="48"/>
+        <source>&amp;Find</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../sourceviewer.cpp" line="51"/>
+        <source>Source of Page %1</source>
+        <translation></translation>
+    </message>
+</context>
+<context>
+    <name>TabBar</name>
+    <message>
+        <location filename="../tabbar.cpp" line="145"/>
+        <source>Show Tab Bar</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../tabbar.cpp" line="145"/>
+        <source>Hide Tab Bar</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../tabbar.cpp" line="170"/>
+        <source>Duplicate Tab</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../tabbar.cpp" line="176"/>
+        <source>&amp;Close Tab</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../tabbar.cpp" line="180"/>
+        <source>Close &amp;Other Tabs</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../tabbar.cpp" line="186"/>
+        <source>Reload Tab</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../tabbar.cpp" line="192"/>
+        <source>Reload All Tabs</source>
+        <translation></translation>
+    </message>
+</context>
+<context>
+    <name>TabWidget</name>
+    <message>
+        <location filename="../tabwidget.cpp" line="426"/>
+        <source>Untitled</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../tabwidget.cpp" line="479"/>
+        <source>Saved Tabs</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../tabwidget.cpp" line="565"/>
+        <source>Do you really want to close this page?</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../tabwidget.cpp" line="566"/>
+        <source>You have modified this page and when closing it you would lose the modification.
+Do you really want to close this page?
+</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../tabwidget.cpp" line="635"/>
+        <source>Loading...</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../tabwidget.cpp" line="649"/>
+        <source>Loading %1% (%2 %3)...</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../tabwidget.cpp" line="674"/>
+        <source>Finished loading</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../tabwidget.cpp" line="676"/>
+        <source>Failed to load</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../tabwidget.cpp" line="756"/>
+        <source>Show Next Tab</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../tabwidget.cpp" line="760"/>
+        <source>Ctrl-]</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../tabwidget.cpp" line="764"/>
+        <source>Show Previous Tab</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../tabwidget.cpp" line="768"/>
+        <source>Ctrl-[</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../tabwidget.cpp" line="772"/>
+        <source>Recently Closed Tabs</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../tabwidget.cpp" line="773"/>
+        <source>New &amp;Tab</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../tabwidget.cpp" line="774"/>
+        <source>&amp;Close Tab</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../tabwidget.cpp" line="775"/>
+        <source>Bookmark All Tabs</source>
+        <translation></translation>
+    </message>
+</context>
+<context>
+    <name>ToolbarSearch</name>
+    <message>
+        <location filename="../toolbarsearch.cpp" line="292"/>
+        <source>Suggestions</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../toolbarsearch.cpp" line="341"/>
+        <source>Add &apos;%1&apos;</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../toolbarsearch.cpp" line="351"/>
+        <source>Clear Recent Searches</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../toolbarsearch.cpp" line="406"/>
+        <source>No Recent Searches</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../toolbarsearch.cpp" line="410"/>
+        <source>Recent Searches</source>
+        <translation></translation>
+    </message>
+</context>
+<context>
+    <name>UserAgentMenu</name>
+    <message>
+        <location filename="../useragent/useragentmenu.cpp" line="53"/>
+        <source>Default</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../useragent/useragentmenu.cpp" line="67"/>
+        <source>Other...</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../useragent/useragentmenu.cpp" line="132"/>
+        <source>Custom user agent</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../useragent/useragentmenu.cpp" line="133"/>
+        <source>User agent:</source>
+        <translation></translation>
+    </message>
+</context>
+<context>
+    <name>WebPage</name>
+    <message>
+        <location filename="../webpage.cpp" line="231"/>
+        <source>Resending POST request</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../webpage.cpp" line="232"/>
+        <source>In order to display the site, the request along with all the data must be sent once again, which may lead to some unexpected behaviour of the site e.g. the same action might be performed once again. Do you want to continue anyway?</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../webpage.cpp" line="387"/>
+        <source>Error loading page: %1</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../webpage.cpp" line="398"/>
+        <source>When connecting to: %1.</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../webpage.cpp" line="399"/>
+        <source>Check the address for errors such as &lt;b&gt;ww&lt;/b&gt;.arora-browser.org instead of &lt;b&gt;www&lt;/b&gt;.arora-browser.org</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../webpage.cpp" line="400"/>
+        <source>If the address is correct, try checking the network connection.</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../webpage.cpp" line="401"/>
+        <source>If your computer or network is protected by a firewall or proxy, make sure that the browser is permitted to access the network.</source>
+        <translation></translation>
+    </message>
+</context>
+<context>
+    <name>WebView</name>
+    <message>
+        <location filename="../webview.cpp" line="165"/>
+        <source>Open in New &amp;Window</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../webview.cpp" line="167"/>
+        <source>Open in New &amp;Tab</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../webview.cpp" line="170"/>
+        <source>Save Lin&amp;k</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../webview.cpp" line="171"/>
+        <source>&amp;Bookmark This Link</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../webview.cpp" line="175"/>
+        <source>&amp;Copy Link Location</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../webview.cpp" line="181"/>
+        <source>Open Image in New &amp;Window</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../webview.cpp" line="183"/>
+        <source>Open Image in New &amp;Tab</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../webview.cpp" line="186"/>
+        <source>&amp;Save Image</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../webview.cpp" line="187"/>
+        <source>&amp;Copy Image</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../webview.cpp" line="188"/>
+        <source>C&amp;opy Image Location</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../webview.cpp" line="190"/>
+        <source>Block Image</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../webview.cpp" line="199"/>
+        <source>Search with...</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../webview.cpp" line="225"/>
+        <source>Add to the toolbar search</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../webview.cpp" line="380"/>
+        <source>Method not supported</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../webview.cpp" line="381"/>
+        <source>%1 method is not supported.</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../webview.cpp" line="425"/>
+        <source>Search engine</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../webview.cpp" line="426"/>
+        <source>Choose the desired search engine</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../webview.cpp" line="440"/>
+        <source>Engine name</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../webview.cpp" line="440"/>
+        <source>Type in a name for the engine</source>
+        <translation></translation>
+    </message>
+    <message>
+        <location filename="../webview.cpp" line="531"/>
+        <source>Loading...</source>
+        <translation></translation>
+    </message>
+</context>
+<context>
+    <name>WebViewSearch</name>
+    <message>
+        <location filename="../webviewsearch.cpp" line="67"/>
+        <source>Not Found</source>
+        <translation></translation>
+    </message>
+</context>
+</TS>
diff --git a/src/locale/locale.pri b/src/locale/locale.pri
index e262031..50a6218 100644
--- a/src/locale/locale.pri
+++ b/src/locale/locale.pri
@@ -10,6 +10,7 @@ TRANSLATIONS += \
     da_DK.ts \
     de_DE.ts \
     el_GR.ts \
+    en_GB.ts \
     es.ts \
     es_CR.ts \
     et_EE.ts \
