diff --git a/src/browserapplication.cpp b/src/browserapplication.cpp
index 405f176c709d68221bf0794f85ec1ff71db375c7..3866734f3d095a4d106cf603134db16476525d12 100644
--- a/src/browserapplication.cpp
+++ b/src/browserapplication.cpp
@@ -608,6 +608,14 @@ LanguageManager *BrowserApplication::languageManager()
 {
     if (!s_languageManager) {
         s_languageManager = new LanguageManager();
+        // add the standard locations. One of them should contain the $prefix/share/arora/locale
+        // directory where `make install` puts the translation files.
+        const auto localeDirs =
+            QStandardPaths::locateAll(QStandardPaths::DataLocation, "locale", QStandardPaths::LocateDirectory) +
+            QStandardPaths::locateAll(QStandardPaths::GenericDataLocation, "arora/locale", QStandardPaths::LocateDirectory);
+        for (const auto localeDir : localeDirs) {
+            s_languageManager->addLocaleDirectory(localeDir);
+        }
         s_languageManager->addLocaleDirectory(dataFilePath(QLatin1String("locale")));
         s_languageManager->addLocaleDirectory(qApp->applicationDirPath() + QLatin1String("/src/.qm/locale"));
         s_languageManager->addLocaleDirectory(installedDataDirectory() + QLatin1String("/locale"));
