diff --git CMakeLists.txt CMakeLists.txt
index b2bf3c3..6457448 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -19,6 +19,7 @@ include(ECMQtDeclareLoggingCategory)
 include(KDEInstallDirs)
 include(KDEFrameworkCompilerSettings)
 include(KDECMakeSettings)
+include(ECMMarkNonGuiExecutable)
 
 set(KF5_VERSION "5.29.0")
 
diff --git src/cli/CMakeLists.txt src/cli/CMakeLists.txt
index d009bfd..1131153 100644
--- src/cli/CMakeLists.txt
+++ src/cli/CMakeLists.txt
@@ -1,4 +1,5 @@
 add_executable(kate-syntax-highlighter kate-syntax-highlighter.cpp)
+ecm_mark_nongui_executable(kate-syntax-highlighter)
 target_link_libraries(kate-syntax-highlighter KF5SyntaxHighlighting)
 
 install(TARGETS kate-syntax-highlighter ${INSTALL_TARGETS_DEFAULT_ARGS})
