diff --git a/CMakeLists.txt b/CMakeLists.txt
index 23bf5d6..0de1a58 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -30,7 +30,7 @@ ecm_setup_version(PROJECT VARIABLE_PREFIX PULSEAUDIOQT
                         PACKAGE_VERSION_FILE "${CMAKE_CURRENT_BINARY_DIR}/KF5PulseAudioQtConfigVersion.cmake"
                         SOVERSION 1)
 
-set(REQUIRED_QT_VERSION 5.10.0)
+set(REQUIRED_QT_VERSION 5.9.8)
 find_package(Qt5 ${REQUIRED_QT_VERSION} CONFIG REQUIRED Core Gui DBus)
 
 find_package(Qt5Test ${REQUIRED_QT_VERSION} CONFIG)
