--- kstyle/config/orig.main.cpp	2015-12-15 11:17:39.000000000 +0100
+++ kstyle/config/main.cpp	2015-12-21 21:21:45.000000000 +0100
@@ -38,7 +38,7 @@
 
     QApplication app( argc, argv );
     app.setApplicationName( i18n( "Breeze Settings" ) );
-    app.setWindowIcon( QIcon::fromTheme( QStringLiteral( "breeze-settings" ) ) );
+    app.setWindowIcon( QIcon::fromTheme( QStringLiteral( "breeze-settings" ), app.windowIcon() ) );
 
     KCMultiDialog dialog;
     dialog.setWindowTitle( i18n( "Breeze Settings" ) );
--- kstyle/config/orig.CMakeLists.txt	2015-12-15 11:17:39.000000000 +0100
+++ kstyle/config/CMakeLists.txt	2015-12-21 21:57:13.000000000 +0100
@@ -44,6 +44,12 @@
   if( KF5KCMUtils_FOUND )
 
       set(breeze_settings_SOURCES main.cpp)
+
+	 set(ICON_SRC "${CMAKE_CURRENT_BINARY_DIR}/icons/sc-apps-breeze-settings.icns")
+	 set(MACOSX_BUNDLE_ICON_FILE "sc-apps-breeze-settings.icns")
+	 set(breeze_settings_SOURCES "${breeze_settings_SOURCES}" "${ICON_SRC}")
+	 set_source_files_properties(${ICON_SRC} PROPERTIES MACOSX_PACKAGE_LOCATION Resources)
+
       add_executable(breeze-settings5 ${breeze_settings_SOURCES} )
       target_link_libraries(breeze-settings5 Qt5::Core Qt5::Gui Qt5::Widgets )
       target_link_libraries(breeze-settings5 KF5::I18n KF5::KCMUtils)
