diff --git a/src/kdialog.cpp b/src/kdialog.cpp
index 7f8acdb..eb7d936 100644
--- a/src/kdialog.cpp
+++ b/src/kdialog.cpp
@@ -229,7 +229,7 @@ int main(int argc, char *argv[])
     aboutData.addAuthor(i18n("Richard Moore"),QString(),"rich@kde.org");
     aboutData.addAuthor(i18n("Dawit Alemayehu"),QString(),"adawit@kde.org");
     aboutData.addAuthor(i18n("Kai Uwe Broulik"),QString(),"kde@privat.broulik.de");
-    QApplication::setWindowIcon(QIcon::fromTheme("system-run"));
+    QApplication::setWindowIcon(QIcon::fromTheme("system-run", app.windowIcon()));
 
     QCommandLineParser parser;
     KAboutData::setApplicationData(aboutData);
