--- kdialog-19.08.2/src/CMakeLists.txt.orig	2019-10-06 05:05:15.000000000 +0200
+++ kdialog-19.08.2/src/CMakeLists.txt	2019-12-09 11:02:40.902317252 +0100
@@ -1,5 +1,7 @@
 ecm_setup_version(${KDE_APPLICATIONS_VERSION} VARIABLE_PREFIX KDIALOG VERSION_HEADER kdialog_version.h)
 
+include(ECMMarkNonGuiExecutable)
+
 configure_file(config-kdialog.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/config-kdialog.h)
 
 set(kdialog_SRCS
@@ -10,6 +12,7 @@
 )
 
 add_executable(kdialog ${kdialog_SRCS})
+ecm_mark_nongui_executable(kdialog)
 
 target_link_libraries(kdialog
     KF5::KIOWidgets
@@ -42,6 +45,7 @@
 qt5_add_dbus_adaptor(kdialog_progress_helper_SRCS org.kde.kdialog.ProgressDialog.xml progressdialog.h ProgressDialog)
 
 add_executable(kdialog_progress_helper ${kdialog_progress_helper_SRCS})
+ecm_mark_nongui_executable(kdialog_progress_helper)
 
 target_link_libraries(kdialog_progress_helper
     KF5::DBusAddons
