diff --git a/src/kdialog.cpp b/src/kdialog.cpp
index 7f8acdb..eb7d936 100644
--- a/src/kdialog.cpp
+++ b/src/kdialog.cpp
@@ -277,7 +277,7 @@
     aboutData.addAuthor(i18n("Dawit Alemayehu"),QString(),"adawit@kde.org");
     aboutData.addAuthor(i18n("Kai Uwe Broulik"),QString(),"kde@privat.broulik.de");
     KAboutData::setApplicationData(aboutData);
-    QApplication::setWindowIcon(QIcon::fromTheme("system-run"));
+    QApplication::setWindowIcon(QIcon::fromTheme("system-run", app.windowIcon()));
 
     QCommandLineParser parser;
     parser.addVersionOption();
