--- kdialog-19.08.2/src/kdialog.cpp.orig	2019-12-09 10:58:31.247179674 +0100
+++ kdialog-19.08.2/src/kdialog.cpp	2019-12-09 11:00:26.959560651 +0100
@@ -337,7 +337,9 @@
 
     // execute kdialog command
 
-    QApplication::setWindowIcon(QIcon::fromTheme(parser.value(QStringLiteral("icon")), QIcon::fromTheme(QStringLiteral("system-run"))));
+    if (app.windowIcon().isNull()) {
+        QApplication::setWindowIcon(QIcon::fromTheme(parser.value(QStringLiteral("icon")), QIcon::fromTheme(QStringLiteral("system-run"))));
+    }
 
     const QStringList args = parser.positionalArguments();
     const QString title = parser.value(QStringLiteral("title"));
