--- qtbase/mkspecs/features/mac/orig.sdk.prf	2014-09-11 12:48:07.000000000 +0200
+++ qtbase/mkspecs/features/mac/sdk.prf	2015-03-27 15:31:15.000000000 +0100
@@ -27,22 +27,22 @@
 QMAKESPEC_NAME = $$basename(QMAKESPEC)
 
 # Resolve SDK version of various tools
-for(tool, $$list(QMAKE_CC QMAKE_CXX QMAKE_FIX_RPATH QMAKE_AR QMAKE_RANLIB QMAKE_LINK QMAKE_LINK_SHLIB)) {
-    tool_variable = QMAKE_MAC_SDK.$${QMAKESPEC_NAME}.$${QMAKE_MAC_SDK}.$${tool}
-    !isEmpty($$tool_variable) {
-        $$tool = $$eval($$tool_variable)
-        next()
-    }
-
-    value = $$eval($$tool)
-    isEmpty(value): next()
-
-    sysrooted = $$system("/usr/bin/xcrun -sdk $$QMAKE_MAC_SDK -find $$first(value) 2>/dev/null")
-    isEmpty(sysrooted): next()
-
-    $$tool = $$sysrooted $$member(value, 1, -1)
-    cache($$tool_variable, set stash, $$tool)
-}
+#for(tool, $$list(QMAKE_CC QMAKE_CXX QMAKE_FIX_RPATH QMAKE_AR QMAKE_RANLIB QMAKE_LINK QMAKE_LINK_SHLIB)) {
+#    tool_variable = QMAKE_MAC_SDK.$${QMAKESPEC_NAME}.$${QMAKE_MAC_SDK}.$${tool}
+#    !isEmpty($$tool_variable) {
+#        $$tool = $$eval($$tool_variable)
+#        next()
+#    }
+#
+#    value = $$eval($$tool)
+#    isEmpty(value): next()
+#
+#    sysrooted = $$system("/usr/bin/xcrun -sdk $$QMAKE_MAC_SDK -find $$first(value) 2>/dev/null")
+#    isEmpty(sysrooted): next()
+#
+#    $$tool = $$sysrooted $$member(value, 1, -1)
+#    cache($$tool_variable, set stash, $$tool)
+#}
 
 isEmpty(QMAKE_MAC_SDK.$${QMAKE_MAC_SDK}.platform_name) {
     QMAKE_MAC_PLATFORM_NAME = $$system("/usr/libexec/PlistBuddy -c 'print DefaultProperties:PLATFORM_NAME' $$QMAKE_MAC_SDK_PATH/SDKSettings.plist 2>/dev/null")
