diff --git a/src/lib/marble/ViewportParams.cpp b/src/lib/marble/ViewportParams.cpp
index 860efcc1deb1c1d8f28ecf1b9d6b1fe6f6bfaf5f..4aeee83b677767b46187e3b0aae92856fb5b4ffd 100644
--- a/src/lib/marble/ViewportParams.cpp
+++ b/src/lib/marble/ViewportParams.cpp
@@ -94,7 +94,7 @@ ViewportParamsPrivate::ViewportParamsPrivate( Projection projection,
       m_planetAxis(),
       m_planetAxisMatrix(),
       m_radius( radius ),
-      m_angularResolution(4.0 / std::abs(m_radius)),
+      m_angularResolution(4.0 / abs(m_radius)),
       m_size( size ),
       m_dirtyBox( true ),
       m_viewLatLonAltBox()
