diff --git a/src/apps/marble-kde/CMakeLists.txt b/src/apps/marble-kde/CMakeLists.txt
index 2841052..f94fddc 100644
--- a/src/apps/marble-kde/CMakeLists.txt
+++ b/src/apps/marble-kde/CMakeLists.txt
@@ -5,6 +5,7 @@ if(ECM_FOUND)
   include(KDEInstallDirs)
   include(KDECMakeSettings)
   include(ECMInstallIcons)
+  include(ECMAddAppIcon)
 endif()
 macro_optional_find_package(KF5 QUIET COMPONENTS Crash Wallet NewStuff Parts CoreAddons I18n)
 
@@ -48,6 +49,9 @@ ecm_install_icons(ICONS
   "${CMAKE_CURRENT_SOURCE_DIR}/../../../data/icons/48-apps-marble.png"
   "${CMAKE_CURRENT_SOURCE_DIR}/../../../data/icons/64-apps-marble.png"
   DESTINATION ${ICON_INSTALL_DIR})
+file(GLOB ICONS_SRCS
+  "${CMAKE_CURRENT_SOURCE_DIR}/../../../data/icons/*-apps-marble.png")
+ecm_add_app_icon(marble_kde_SRCS ICONS ${ICONS_SRCS})
 add_executable (marble ${marble_kde_SRCS})
 
 target_link_libraries (
