--- cmake/orig.FindMediaPlayerInfo.cmake	2015-11-08 13:41:10.000000000 +0100
+++ cmake/FindMediaPlayerInfo.cmake	2015-11-18 22:15:31.000000000 +0100
@@ -10,7 +10,7 @@
              PATH_SUFFIXES "media-player-info" NO_DEFAULT_PATH
             )
 else()
-   set(XDG_DATA_DIRS "/usr/share")
+   set(XDG_DATA_DIRS "@PREFIX@/share:/usr/share")
    message(STATUS "Warning: environment variable XDG_DATA_DIRS not set, falling back to ${XDG_DATA_DIRS}")
    find_path(MEDIAPLAYERINFO_PATH sony_psp.mpi
              PATHS "${XDG_DATA_DIRS}"
