diff --git src/solid/devices/backends/shared/udevqtclient.h src/solid/devices/backends/shared/udevqtclient.h
--- src/solid/devices/backends/shared/udevqtclient.h
+++ src/solid/devices/backends/shared/udevqtclient.h
@@ -65,6 +65,8 @@
     void deviceChanged(const UdevQt::Device &dev);
     void deviceOnlined(const UdevQt::Device &dev);
     void deviceOfflined(const UdevQt::Device &dev);
+    void deviceBound(const UdevQt::Device &dev);
+    void deviceUnbound(const UdevQt::Device &dev);
 
 private:
     friend class ClientPrivate;
diff --git src/solid/devices/backends/shared/udevqtclient.cpp src/solid/devices/backends/shared/udevqtclient.cpp
--- src/solid/devices/backends/shared/udevqtclient.cpp
+++ src/solid/devices/backends/shared/udevqtclient.cpp
@@ -115,6 +115,10 @@
         emit q->deviceOnlined(device);
     } else  if (action == "offline") {
         emit q->deviceOfflined(device);
+    } else  if (action == "bind") {
+        emit q->deviceBound(device);
+    } else  if (action == "unbind") {
+        emit q->deviceUnbound(device);
     } else {
         qWarning("UdevQt: unhandled device action \"%s\"", action.constData());
     }

