--- cmake/orig.FindFFmpeg.cmake	2016-10-11 21:21:21.000000000 +0200
+++ cmake/FindFFmpeg.cmake	2016-10-11 21:49:52.000000000 +0200
@@ -93,6 +93,8 @@
       ${PC_LIB${_component}_INCLUDE_DIRS}
     PATH_SUFFIXES
       ffmpeg
+			ffmpeg-2
+			ffmpeg-30
   )
 
   find_library(${_component}_LIBRARIES NAMES ${_library}
@@ -124,7 +126,7 @@
   find_component(AVDEVICE libavdevice avdevice libavdevice/avdevice.h)
   find_component(AVUTIL   libavutil   avutil   libavutil/avutil.h)
   find_component(SWSCALE  libswscale  swscale  libswscale/swscale.h)
-  find_component(POSTPROC libpostproc postproc libpostproc/postprocess.h)
+  find_component(POSTPROCESS libpostproc postproc libpostproc/postprocess.h)
 
   # Check if the required components were found and add their stuff to the FFMPEG_* vars.
   foreach (_component ${FFmpeg_FIND_COMPONENTS})
