diff --git src/extractors/orig.CMakeLists.txt src/extractors/CMakeLists.txt
index d556fcd..b361a1e 100644
--- src/extractors/orig.CMakeLists.txt
+++ src/extractors/CMakeLists.txt
@@ -8,7 +8,9 @@ if(Poppler_Qt5_FOUND)
         Poppler::Qt5
     )
 
-    set_target_properties(kfilemetadata_popplerextractor PROPERTIES LIBRARY_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/bin/kf5/kfilemetadata")
+    set_target_properties(kfilemetadata_popplerextractor PROPERTIES 
+        LIBRARY_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/bin/kf5/kfilemetadata"
+	   CXX_STANDARD 17)
     install(
     TARGETS kfilemetadata_popplerextractor
     DESTINATION ${PLUGIN_INSTALL_DIR}/kf5/kfilemetadata)
