diff --git src/programs/orig.pkexec.c src/programs/pkexec.c
index 98aa151..efb2842 100644
--- src/programs/orig.pkexec.c
+++ src/programs/pkexec.c
@@ -488,6 +488,7 @@ main (int argc, char *argv[])
   gchar *opt_user;
   pid_t pid_of_caller;
   gpointer local_agent_handle;
+  gchar *cmdline_short = NULL;
 
 
   /*
@@ -802,7 +803,6 @@ main (int argc, char *argv[])
   polkit_details_insert (details, "program", path);
   polkit_details_insert (details, "command_line", command_line);
 
-  gchar *cmdline_short = NULL;
   cmdline_short = g_strdup(command_line);
   if (strlen(command_line) > 80)
       g_stpcpy(g_stpcpy( cmdline_short + 38, " ... " ),
