diff --git orig.meson.build meson.build
index 3b96562..a974c36 100644
--- orig.meson.build
+++ meson.build
@@ -34,10 +34,10 @@ pk_pkgsysconfdir = pk_sysconfdir / pk_api_name
 pk_actiondir = pk_api_name / 'actions'
 pk_pkgactiondir = pk_datadir / pk_actiondir
 
-soversion = 0
+soversion = 1
 current = 0
 revision = 0
-libversion = '@0@.@1@.@2@'.format(soversion, current, revision)
+libversion = '@0@.@1@.@2@'.format(0, current, revision)
 
 gnome = import('gnome')
 i18n = import('i18n')
diff --git src/polkit/orig.meson.build src/polkit/meson.build
index 9b1b3bf..2210489 100644
--- src/polkit/orig.meson.build
+++ src/polkit/meson.build
@@ -87,6 +87,7 @@ endif
 libpolkit_gobject = shared_library(
   name,
   sources: sources,
+  soversion: soversion,
   version: libversion,
   include_directories: incs,
   dependencies: deps,
diff --git src/polkitagent/orig.meson.build src/polkitagent/meson.build
index bee3820..50e4351 100644
--- src/polkitagent/orig.meson.build
+++ src/polkitagent/meson.build
@@ -50,6 +50,7 @@ c_flags = [
 libpolkit_agent = shared_library(
   name,
   sources: sources,
+  soversion: soversion,
   version: libversion,
   include_directories: top_inc,
   dependencies: deps,
