diff --git src/polkit/orig.polkitunixprocess.c src/polkit/polkitunixprocess.c
index 289a82e..0ddab2b 100644
--- src/polkit/orig.polkitunixprocess.c
+++ src/polkit/polkitunixprocess.c
@@ -36,6 +36,10 @@
 #ifdef HAVE_OPENBSD
 #include <sys/sysctl.h>
 #endif
+#if defined(__APPLE__) && defined(__MACH__)
+#define HAVE_APPLEBSD
+#include <sys/sysctl.h>
+#endif
 #include <stdlib.h>
 #include <string.h>
 #include <errno.h>
@@ -167,7 +171,7 @@ static void subject_iface_init (PolkitSubjectIface *subject_iface);
 static guint64 get_start_time_for_pid (gint    pid,
                                        GError **error);
 
-#if defined(HAVE_FREEBSD) || defined(HAVE_NETBSD) || defined(HAVE_OPENBSD)
+#if defined(HAVE_FREEBSD) || defined(HAVE_NETBSD) || defined(HAVE_OPENBSD) || defined(HAVE_APPLEBSD)
 static gboolean get_kinfo_proc (gint pid,
 #if defined(HAVE_NETBSD)
                                 struct kinfo_proc2 *p);
@@ -645,7 +649,7 @@ get_kinfo_proc (pid_t pid, struct kinfo_proc *p)
 }
 #endif
 
-#if defined(HAVE_NETBSD) || defined(HAVE_OPENBSD)
+#if defined(HAVE_NETBSD) || defined(HAVE_OPENBSD) || defined(HAVE_APPLEBSD)
 static gboolean
 get_kinfo_proc (gint pid,
 #ifdef HAVE_NETBSD
@@ -668,8 +672,12 @@ get_kinfo_proc (gint pid,
 #endif
   name[namelen++] = KERN_PROC_PID;
   name[namelen++] = pid;
+#ifndef HAVE_APPLEBSD
   name[namelen++] = sz;
   name[namelen++] = 1;
+#else
+  memset(p, 0, sz);
+#endif
 
   if (sysctl (name, namelen, p, &sz, NULL, 0) == -1)
     return FALSE;
@@ -683,7 +691,7 @@ get_start_time_for_pid (pid_t    pid,
                         GError **error)
 {
   guint64 start_time;
-#if !defined(HAVE_FREEBSD) && !defined(HAVE_NETBSD) && !defined(HAVE_OPENBSD)
+#if !defined(HAVE_FREEBSD) && !defined(HAVE_NETBSD) && !defined(HAVE_OPENBSD) && !defined(HAVE_APPLEBSD)
   gchar *filename;
   gchar *contents;
   size_t length;
@@ -775,7 +783,9 @@ get_start_time_for_pid (pid_t    pid,
       goto out;
     }
 
-#ifdef HAVE_FREEBSD
+#ifdef HAVE_APPLEBSD
+  start_time = (guint64) p.kp_proc.p_starttime.tv_sec;
+#elif defined(HAVE_FREEBSD)
   start_time = (guint64) p.ki_start.tv_sec;
 #else
   start_time = (guint64) p.p_ustart_sec;
@@ -801,7 +811,7 @@ polkit_unix_process_get_racy_uid__ (PolkitUnixProcess  *process,
   gchar *contents;
   gchar **lines;
   guint64 start_time;
-#if defined(HAVE_FREEBSD) || defined(HAVE_OPENBSD)
+#if defined(HAVE_FREEBSD) || defined(HAVE_OPENBSD) || defined(HAVE_APPLEBSD)
   struct kinfo_proc p;
 #elif defined(HAVE_NETBSD)
   struct kinfo_proc2 p;
@@ -818,7 +828,7 @@ polkit_unix_process_get_racy_uid__ (PolkitUnixProcess  *process,
   lines = NULL;
   contents = NULL;
 
-#if defined(HAVE_FREEBSD) || defined(HAVE_NETBSD) || defined(HAVE_OPENBSD)
+#if defined(HAVE_FREEBSD) || defined(HAVE_NETBSD) || defined(HAVE_OPENBSD) || defined(HAVE_APPLEBSD)
   if (get_kinfo_proc (process->pid, &p) == 0)
     {
       g_set_error (error,
@@ -830,7 +840,10 @@ polkit_unix_process_get_racy_uid__ (PolkitUnixProcess  *process,
       goto out;
     }
 
-#if defined(HAVE_FREEBSD)
+#ifdef HAVE_APPLEBSD
+  result = p.kp_proc.p_pid;
+  start_time = (guint64) p.kp_proc.p_starttime.tv_sec;
+#elif defined(HAVE_FREEBSD)
   result = p.ki_uid;
   start_time = (guint64) p.ki_start.tv_sec;
 #else
