diff --git a/meson.build b/meson.build
index c4c3c998..69c4e1e4 100644
--- a/meson.build
+++ b/meson.build
@@ -31,7 +31,7 @@ xml = dependency('libxml-2.0', version: '>= 2.9')
 
 if get_option('webkit')
     soup = dependency('libsoup-3.0', version: '>= 3.0')
-    webkit = dependency('webkit2gtk-4.1', version: '>= 2.42')
+    webkit = dependency('webkit2gtk-4.1', version: '>= 2.40.5')
     add_global_arguments('-D', 'HAVE_WEBKIT', language: 'vala')
 endif
 
