diff --git a/mscore/scoreview.cpp b/mscore/scoreview.cpp
index a3d7ea148b3dd6e452094fb01db6d2df1c3c9c01..cbd8bbdeb16234f87d83015f7136d567c069ebd1 100644
--- a/mscore/scoreview.cpp
+++ b/mscore/scoreview.cpp
@@ -2534,7 +2534,7 @@ void ScoreView::cmd(const char* s)
                   cv->updateGrips();
                   }},
       #ifdef OMR
-            {{"show-omr"}, [](ScoreView* cv, const QByteArray&) {
+            {{"show-omr"}, [this](ScoreView* cv, const QByteArray&) {
                   if (cv->score()->masterScore()->omr())
                         showOmr(!_score->masterScore()->showOmr());
                   }},
diff --git a/omr/pattern.h b/omr/pattern.h
index b0404972c1be0a0179a6efb25a603ee51e428143..f8b867eb527b374c94580db96195052b3f7db02f 100644
--- a/omr/pattern.h
+++ b/omr/pattern.h
@@ -24,7 +24,7 @@
 
 namespace Ms {
 
-enum class SymId;
+// enum class SymId;
 class Sym;
 
 //---------------------------------------------------------
