diff --git a/libmscore/mscore.cpp b/libmscore/mscore.cpp
index e14c809019b60daf5a0920a6fdc4cd329d8ea98e..cffeeca384016289c7c9e4b4bf95c563e83c0686 100644
--- a/libmscore/mscore.cpp
+++ b/libmscore/mscore.cpp
@@ -375,6 +375,34 @@ void MScore::init()
                         exit(-1);
                   }
             }
+#else
+    // load the fonts from Contents/Resources/fonts
+    // so that we get normal function with all font engines as well as with the XCB QPA
+      static const char* fonts[] = {
+            "MuseJazzText.otf",
+            "FreeSans.ttf",
+            "FreeSerif.ttf",
+            "FreeSerifBold.ttf",
+            "FreeSerifItalic.ttf",
+            "FreeSerifBoldItalic.ttf",
+            "mscoreTab.ttf",
+            "mscore-BC.ttf",
+            "BravuraText.otf",
+            "GootvilleText.otf",
+            "MScoreText.ttf",
+            };
+
+      extern QString getSharePath();
+      const auto sharePath = getSharePath();
+      for (unsigned i = 0; i < sizeof(fonts)/sizeof(*fonts); ++i) {
+            const auto str = sharePath + QStringLiteral("/fonts/") + QString::fromLatin1(fonts[i]);
+            if (-1 == QFontDatabase::addApplicationFont(str)) {
+                  if (!MScore::testMode)
+                        qDebug("Mscore: fatal error: cannot load internal font <%s>", qPrintable(str));
+                  if (!MScore::debugMode && !MScore::testMode)
+                        exit(-1);
+                  }
+            }
 #endif
       initScoreFonts();
       StaffType::initStaffTypes();
diff --git a/libmscore/sym.cpp b/libmscore/sym.cpp
index d3a0a3c99a860e5ef6fd33d0934ff6e3a42ecc1c..4d56fabe49acbe79499e8d4858003bcb33de5c60 100644
--- a/libmscore/sym.cpp
+++ b/libmscore/sym.cpp
@@ -5709,7 +5709,14 @@ void ScoreFont::draw(SymId id, QPainter* painter, const QSizeF& mag, const QPoin
 
       if (MScore::pdfPrinting) {
             if (font == 0) {
+#ifdef Q_OS_MAC
+                  // On Mac we add the application font from the Resources/fonts directory
+                  // (addApplicationFont() can crash when loading a font from the embedded Qt resource)
+                  extern QString getSharePath();
+                  QString s(getSharePath() + QStringLiteral("/fonts/") + _filename);
+#else
                   QString s(_fontPath+_filename);
+#endif
                   if (-1 == QFontDatabase::addApplicationFont(s)) {
                         qDebug("Mscore: fatal error: cannot load internal font <%s>", qPrintable(s));
                         return;
diff --git a/mscore/CMakeLists.txt b/mscore/CMakeLists.txt
index d484a7cce52bb8bbf479765346d14b2442365d76..08211aa372d6a5a05f9902b643f10336423ff1ad 100644
--- a/mscore/CMakeLists.txt
+++ b/mscore/CMakeLists.txt
@@ -99,7 +99,15 @@ QT5_WRAP_UI (ui_headers
 if (APPLE)
       QT5_ADD_RESOURCES (qrc_files musescore.qrc
             qml.qrc # TODO: replace with qtquick_compiler_add_resources on Qt >= 5.11
-            musescorefonts-Mac.qrc shortcut-Mac.qrc
+            #musescorefonts-Mac.qrc
+            musescorefonts-MScore.qrc
+            musescorefonts-Gootville.qrc
+            musescorefonts-Bravura.qrc
+            musescorefonts-MuseJazz.qrc
+            musescorefonts-Campania.qrc
+            musescorefonts-FreeSerif.qrc
+            musescorefonts-Free.qrc
+            shortcut-Mac.qrc
             )
 else (APPLE)
       QT5_ADD_RESOURCES (qrc_files musescore.qrc
diff --git a/main/main.cpp b/main/main.cpp
index 1a5b79f6a578791b9d47b37dc4ea43b70c950baa..d837b0eb86da1766cd240fa1ab5d02b2579e625c 100644
--- a/main/main.cpp
+++ b/main/main.cpp
@@ -32,7 +32,14 @@ static void initResources()
 #ifdef Q_OS_MAC
       Q_INIT_RESOURCE(musescore);
       Q_INIT_RESOURCE(qml);
-      Q_INIT_RESOURCE(musescorefonts_Mac);
+//       Q_INIT_RESOURCE(musescorefonts_Mac);
+      Q_INIT_RESOURCE(musescorefonts_MScore);
+      Q_INIT_RESOURCE(musescorefonts_Gootville);
+      Q_INIT_RESOURCE(musescorefonts_Bravura);
+      Q_INIT_RESOURCE(musescorefonts_MuseJazz);
+      Q_INIT_RESOURCE(musescorefonts_Campania);
+      Q_INIT_RESOURCE(musescorefonts_FreeSerif);
+      Q_INIT_RESOURCE(musescorefonts_Free);
       Q_INIT_RESOURCE(shortcut_Mac);
 #else
       Q_INIT_RESOURCE(musescore);
