diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0565408c3..8a2e5afad 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -62,6 +62,9 @@ if (MSVC)
    set (MINGW false)
 endif (MSVC)
 
+include(FeatureSummary)
+include(FindPkgConfig)
+
 # We need this early, before FindQt5
 option(BUILD_WEBENGINE "Built in webengine support" ON)
 
@@ -135,7 +138,7 @@ option(BUILD_LAME    "Enable MP3 export" ON)                   # Requires libmp3
 option(DOWNLOAD_SOUNDFONT "Download the latest soundfont version as part of the build process" ON)
 
 # licence incompatibility, must never be distributed
-set(OCR OFF)
+# set(OCR OFF)
 
 SET(JACK_LONGNAME "JACK (Jack Audio Connection Kit)")
 SET(JACK_MIN_VERSION "0.98.0")
@@ -153,8 +156,8 @@ option(BUILD_CRASH_REPORTER "Build with crash reporter" OFF)
 set(CRASH_REPORT_URL "http://127.0.0.1:1127/post" CACHE STRING "URL where to send crash reports (valid if BUILD_CRASH_REPORTER is set to ON)")
 
 if (APPLE)
-      set (CMAKE_CXX_COMPILER   clang++)
-      set (CMAKE_CXX_COMPILER_ID "Clang")
+      #set (CMAKE_CXX_COMPILER   clang++)
+      #set (CMAKE_CXX_COMPILER_ID "Clang")
       # Currently, just 'ppc', 'i386' or 'ppc i386' are useful architectures,
       # because the Flash-Player is not yet available as a 64-bit version.
       # Flash is required for displaying the videos of MuseScore-Connect.
@@ -166,7 +169,7 @@ if (APPLE)
       # of MacOSX.
       #set(CMAKE_OSX_SYSROOT /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.9.sdk)
 
-      set(CMAKE_OSX_DEPLOYMENT_TARGET 10.12)      # Min version required
+      #set(CMAKE_OSX_DEPLOYMENT_TARGET 10.12)      # Min version required
       set(HAS_AUDIOFILE TRUE)            # Requires libsndfile
       set(MAC_APPCAST_URL "")
 
@@ -372,7 +375,7 @@ endif(BUILD_FOR_WINSTORE)
 ##
 
 if (USE_SYSTEM_FREETYPE)
-      if (APPLE)
+      if (NOPE)
             PKGCONFIG (freetype2 2.5.2 FREETYPE_INCLUDE_DIRS FREETYPE_LIBDIR FREETYPE_LIBRARIES FREETYPE_CPP)
             if (FREETYPE_INCLUDE_DIRS)
                   STRING(REGEX REPLACE  "\"" "" FREETYPE_INCLUDE_DIRS ${FREETYPE_INCLUDE_DIRS})
@@ -381,10 +384,14 @@ if (USE_SYSTEM_FREETYPE)
             else (FREETYPE_INCLUDE_DIRS)
                   message(FATAL_ERROR "freetype >= 2.5.2 is required\n")
             endif (FREETYPE_INCLUDE_DIRS)
-      else (APPLE)
+      else (NOPE)
             find_package(Freetype REQUIRED)
-      endif (APPLE)
+      endif (NOPE)
 endif (USE_SYSTEM_FREETYPE)
+if (USE_SYSTEM_POPPLER)
+    pkg_check_modules(PKG_POPPLER poppler REQUIRED)
+    pkg_check_modules(PKG_POPPLER_QT5 poppler-qt5 REQUIRED)
+endif ()
 
 ##
 ## ALSA >= 1.0.0
@@ -832,8 +839,8 @@ if (OMR)
    include_directories(${PROJECT_SOURCE_DIR}/thirdparty/poppler)
    include_directories(${PROJECT_SOURCE_DIR}/thirdparty/poppler/poppler)
  else (NOT USE_SYSTEM_POPPLER)
-   include_directories(/usr/include/poppler)
-   include_directories(/usr/include/poppler/qt5)
+   include_directories(${PKG_POPPLER_INCLUDEDIR}/poppler)
+   include_directories(${PKG_POPPLER_QT5_INCLUDEDIR}/poppler/qt5)
  endif (NOT USE_SYSTEM_POPPLER)
 endif (OMR)
 
@@ -914,6 +921,8 @@ if (MSVC)
    set(VS_STARTUP_PROJECT mscore)
 endif (MSVC)
 
+feature_summary(WHAT ALL   FATAL_ON_MISSING_REQUIRED_PACKAGES)
+
 ## TEMP: Display all variables!
 ### message(STATUS "===========================================================")
 ### message(STATUS "VARIABLES:")
diff --git a/mscore/CMakeLists.txt b/mscore/CMakeLists.txt
index e22d05df4..637d8a785 100644
--- a/mscore/CMakeLists.txt
+++ b/mscore/CMakeLists.txt
@@ -743,7 +743,7 @@ else (MINGW)
            if (APPLE)
                set(PORTMIDI_LIB portmidi)
            else (APPLE)
-               set(PORTMIDI_LIB -lportmidi -lporttime) # Remove -lporttime on RPM-based systems where PortTime is part of PortMidi.
+               set(PORTMIDI_LIB -lportmidi) # Remove -lporttime on RPM-based systems where PortTime is part of PortMidi.
            endif (APPLE)
            target_link_libraries(mscore ${PORTMIDI_LIB})
        endif (USE_PORTMIDI)
