diff --git a/mscore/scoreview.cpp b/mscore/scoreview.cpp
index a3d7ea148b3dd6e452094fb01db6d2df1c3c9c01..cbd8bbdeb16234f87d83015f7136d567c069ebd1 100644
--- a/mscore/scoreview.cpp
+++ b/mscore/scoreview.cpp
@@ -2534,7 +2534,7 @@ void ScoreView::cmd(const char* s)
                   cv->updateGrips();
                   }},
       #ifdef OMR
-            {{"show-omr"}, [](ScoreView* cv, const QByteArray&) {
+            {{"show-omr"}, [this](ScoreView* cv, const QByteArray&) {
                   if (cv->score()->masterScore()->omr())
                         showOmr(!_score->masterScore()->showOmr());
                   }},
