diff --git a/mscore/qml/palettes/Palette.qml b/mscore/qml/palettes/Palette.qml
index 3d1a134f3644d..9210fc812534e 100644
--- a/mscore/qml/palettes/Palette.qml
+++ b/mscore/qml/palettes/Palette.qml
@@ -767,7 +767,7 @@ GridView {
                 // force not hiding palette cell if it is being dragged to a score
                 enabled: paletteCell.paletteDrag
                 target: mscore
-                onElementDraggedToScoreView: paletteCell.paletteDrag = false
+                function onElementDraggedToScoreView() { paletteCell.paletteDrag = false; }
             }
         } // end ItemDelegate
     } // end DelegateModel
diff --git a/mscore/qml/palettes/PaletteTree.qml b/mscore/qml/palettes/PaletteTree.qml
index 1140aad4e56bb..c7506e1110035 100644
--- a/mscore/qml/palettes/PaletteTree.qml
+++ b/mscore/qml/palettes/PaletteTree.qml
@@ -771,7 +771,7 @@ ListView {
 
     Connections {
         target: palettesWidget
-        onHasFocusChanged: {
+        function onHasFocusChanged () {
             if (!palettesWidget.hasFocus) {
                 paletteSelectionModel.clearSelection();
                 expandedPopupIndex = null;
diff --git a/mscore/qml/palettes/PalettesWidgetHeader.qml b/mscore/qml/palettes/PalettesWidgetHeader.qml
index 9a0fcb3ddbca7..f7f1664d9958a 100644
--- a/mscore/qml/palettes/PalettesWidgetHeader.qml
+++ b/mscore/qml/palettes/PalettesWidgetHeader.qml
@@ -224,7 +224,7 @@ Item {
 
     Connections {
         target: palettesWidget
-        onHasFocusChanged: {
+        function onHasFocusChanged () {
             if (!palettesWidget.hasFocus && !palettesListPopup.inMenuAction)
                 palettesListPopup.visible = false;
         }
