diff --git a/mscore/musescore.cpp b/mscore/musescore.cpp
index 38d2a2a62d67157b0a55cec2d60dbcbee4bc8f6f..3aac6f690abadd3b2bc013504459c97215c2717f 100644
--- a/mscore/musescore.cpp
+++ b/mscore/musescore.cpp
@@ -6937,14 +6937,12 @@ void MuseScore::updateWindowTitle(Score* score)
             scoreTitle = score->title();
       else
             scoreTitle = score->masterScore()->title() + "-" + score->title();
+      setWindowTitle("MuseScore " VERSION " : " + scoreTitle + " [*]");
 #ifdef Q_OS_MAC
-      setWindowTitle(scoreTitle);
       if (score->masterScore()->created())
             setWindowFilePath(QString());
       else
             setWindowFilePath(score->masterScore()->fileInfo()->absoluteFilePath());
-#else
-      setWindowTitle(MUSESCORE_NAME_VERSION ": " + scoreTitle + "[*]");
 #endif
       }
 
