diff --git a/mscore/importmidi/importmidi.cpp b/mscore/importmidi/importmidi.cpp
index a5d8894686c32f633dffd65a9b9109d1164c0b81..8df55b450dea88d0df9070a763c780d1e46329a8 100644
--- a/mscore/importmidi/importmidi.cpp
+++ b/mscore/importmidi/importmidi.cpp
@@ -66,6 +66,11 @@
 #include <set>
 
 
+#ifndef QT_DEBUG
+#undef Q_ASSERT_X
+#define Q_ASSERT_X(cond,where,what) /**/
+#endif
+
 namespace Ms {
 
 extern void updateNoteLines(Segment*, int track);
diff --git a/mscore/importmidi/importmidi_beat.cpp b/mscore/importmidi/importmidi_beat.cpp
index 6c356e42525dd635f22cb3c3b0765e66fa1cccc9..6676cc8c1e2531ab06fd9763bd594b0ded8b7514 100644
--- a/mscore/importmidi/importmidi_beat.cpp
+++ b/mscore/importmidi/importmidi_beat.cpp
@@ -13,6 +13,10 @@
 
 #include <functional>
 
+#ifndef QT_DEBUG
+#undef Q_ASSERT_X
+#define Q_ASSERT_X(cond,where,what) /**/
+#endif
 
 namespace Ms {
 namespace MidiBeat {
diff --git a/mscore/importmidi/importmidi_chord.cpp b/mscore/importmidi/importmidi_chord.cpp
index 2aa70aabeccf0a5ed671154605bb96ead2c4efd7..da9043b036be146552b3cdceab080a1e08a0546f 100644
--- a/mscore/importmidi/importmidi_chord.cpp
+++ b/mscore/importmidi/importmidi_chord.cpp
@@ -11,6 +11,11 @@
 #include <set>
 
 
+#ifndef QT_DEBUG
+#undef Q_ASSERT_X
+#define Q_ASSERT_X(cond,where,what) /**/
+#endif
+
 namespace Ms {
 namespace MChord {
 
diff --git a/mscore/importmidi/importmidi_chordname.cpp b/mscore/importmidi/importmidi_chordname.cpp
index b0e5e6c1fd055c9d42d5532ed30c314240319d8e..cf55044ce6c994b0b2d44f0f4e170a7698f93250 100644
--- a/mscore/importmidi/importmidi_chordname.cpp
+++ b/mscore/importmidi/importmidi_chordname.cpp
@@ -12,6 +12,11 @@
 
 // From XF Format Specifications V 2.01 (January 13, 1999, YAMAHA CORPORATION)
 
+#ifndef QT_DEBUG
+#undef Q_ASSERT_X
+#define Q_ASSERT_X(cond,where,what) /**/
+#endif
+
 namespace Ms {
 namespace MidiChordName {
 
diff --git a/mscore/importmidi/importmidi_clef.cpp b/mscore/importmidi/importmidi_clef.cpp
index 2552386496e437192681ae611c14d39844f3791e..860f36328d1bb8204c7306528c6d3578d621017e 100644
--- a/mscore/importmidi/importmidi_clef.cpp
+++ b/mscore/importmidi/importmidi_clef.cpp
@@ -32,6 +32,11 @@
 #include <set>
 
 
+#ifndef QT_DEBUG
+#undef Q_ASSERT_X
+#define Q_ASSERT_X(cond,where,what) /**/
+#endif
+
 namespace Ms {
 
 namespace MidiClef {
diff --git a/mscore/importmidi/importmidi_delegate.cpp b/mscore/importmidi/importmidi_delegate.cpp
index 7295df9c7639b8ca1c6fd032aaf6b79c863126e4..cf2d20fc6bbe3f10bdfaa80bd944d33d9ee17164 100644
--- a/mscore/importmidi/importmidi_delegate.cpp
+++ b/mscore/importmidi/importmidi_delegate.cpp
@@ -1,5 +1,10 @@
 #include "importmidi_delegate.h"
 
+#ifndef QT_DEBUG
+#undef Q_ASSERT_X
+#define Q_ASSERT_X(cond,where,what) /**/
+#endif
+
 namespace Ms {
 
 TimeSigEditor::TimeSigEditor(const QStringList &values, QWidget *parent)
diff --git a/mscore/importmidi/importmidi_drum.cpp b/mscore/importmidi/importmidi_drum.cpp
index 0f404516cc7c7930076c5569aec7b6e5d087acd6..c6800c929db5714ad76336737fb43a6ce9f6b037 100644
--- a/mscore/importmidi/importmidi_drum.cpp
+++ b/mscore/importmidi/importmidi_drum.cpp
@@ -11,6 +11,11 @@
 #include <set>
 
 
+#ifndef QT_DEBUG
+#undef Q_ASSERT_X
+#define Q_ASSERT_X(cond,where,what) /**/
+#endif
+
 namespace Ms {
 
 namespace MidiDrum {
diff --git a/mscore/importmidi/importmidi_fraction.cpp b/mscore/importmidi/importmidi_fraction.cpp
index 73aaef0ee096a09a56da578ee23ba512cc435139..697d36da3eb1adb1fbbb6184db9bc3300ffe8a11 100644
--- a/mscore/importmidi/importmidi_fraction.cpp
+++ b/mscore/importmidi/importmidi_fraction.cpp
@@ -4,6 +4,11 @@
 #include <limits>
 
 
+#ifndef QT_DEBUG
+#undef Q_ASSERT_X
+#define Q_ASSERT_X(cond,where,what) /**/
+#endif
+
 namespace Ms {
 
 #ifdef QT_DEBUG
diff --git a/mscore/importmidi/importmidi_inner.cpp b/mscore/importmidi/importmidi_inner.cpp
index 743f8195b88822559b78278df0a83b8b6df80a6d..4d85773ff03fa68c1733ad2c73bc93cd56dfa0d9 100644
--- a/mscore/importmidi/importmidi_inner.cpp
+++ b/mscore/importmidi/importmidi_inner.cpp
@@ -6,6 +6,11 @@
 #include "midi/midifile.h"
 
 
+#ifndef QT_DEBUG
+#undef Q_ASSERT_X
+#define Q_ASSERT_X(cond,where,what) /**/
+#endif
+
 namespace Ms {
 
 MTrack::MTrack()
diff --git a/mscore/importmidi/importmidi_instrument.cpp b/mscore/importmidi/importmidi_instrument.cpp
index 3483962d8dd2c3c47c62a8271b4e086a461f7219..aa0a0065476b62454d16264d50e47168302ef092 100644
--- a/mscore/importmidi/importmidi_instrument.cpp
+++ b/mscore/importmidi/importmidi_instrument.cpp
@@ -12,6 +12,11 @@
 #include <set>
 
 
+#ifndef QT_DEBUG
+#undef Q_ASSERT_X
+#define Q_ASSERT_X(cond,where,what) /**/
+#endif
+
 namespace Ms {
 
 extern QList<InstrumentGroup*> instrumentGroups;
diff --git a/mscore/importmidi/importmidi_key.cpp b/mscore/importmidi/importmidi_key.cpp
index 642dda23cb378aa9378d69b4b18d7490cef9165a..dee1ac70826c7e0bb8a744a616865198d39e45fa 100644
--- a/mscore/importmidi/importmidi_key.cpp
+++ b/mscore/importmidi/importmidi_key.cpp
@@ -15,6 +15,11 @@
 // "Inferring Score Level Musical Information From Low-Level Musical Data", 2004
 // by Jürgen Kilian
 
+#ifndef QT_DEBUG
+#undef Q_ASSERT_X
+#define Q_ASSERT_X(cond,where,what) /**/
+#endif
+
 namespace Ms {
 namespace MidiKey {
 
diff --git a/mscore/importmidi/importmidi_lrhand.cpp b/mscore/importmidi/importmidi_lrhand.cpp
index aba9dab6d98ce4f18ba052e6ec1f7da15a49d7ff..0a98d2737547b1b1424a204efeb76d8a92528590 100644
--- a/mscore/importmidi/importmidi_lrhand.cpp
+++ b/mscore/importmidi/importmidi_lrhand.cpp
@@ -6,6 +6,11 @@
 #include "mscore/preferences.h"
 
 
+#ifndef QT_DEBUG
+#undef Q_ASSERT_X
+#define Q_ASSERT_X(cond,where,what) /**/
+#endif
+
 namespace Ms {
 
 namespace LRHand {
diff --git a/mscore/importmidi/importmidi_lyrics.cpp b/mscore/importmidi/importmidi_lyrics.cpp
index f6f95471d2071a75c209493b9e11b29f66f82d2e..bfb037945c42ec502a913ee68a8d3bcc2e19d6ba 100644
--- a/mscore/importmidi/importmidi_lyrics.cpp
+++ b/mscore/importmidi/importmidi_lyrics.cpp
@@ -15,6 +15,11 @@
 #include <set>
 
 
+#ifndef QT_DEBUG
+#undef Q_ASSERT_X
+#define Q_ASSERT_X(cond,where,what) /**/
+#endif
+
 namespace Ms {
 
 namespace MidiLyrics {
diff --git a/mscore/importmidi/importmidi_meter.cpp b/mscore/importmidi/importmidi_meter.cpp
index 9c7bd9869af884cb8a520720093b795a69f9b347..ca53589e58dadadfab21501fbc5445487296b00b 100644
--- a/mscore/importmidi/importmidi_meter.cpp
+++ b/mscore/importmidi/importmidi_meter.cpp
@@ -7,6 +7,11 @@
 #include "importmidi_inner.h"
 
 
+#ifndef QT_DEBUG
+#undef Q_ASSERT_X
+#define Q_ASSERT_X(cond,where,what) /**/
+#endif
+
 namespace Ms {
 namespace Meter {
 
diff --git a/mscore/importmidi/importmidi_model.cpp b/mscore/importmidi/importmidi_model.cpp
index af462e443321da538355b1098d554cee722c44cc..a6ccc4962291d11b51f1dd793d5a41b4d7cb4b8e 100644
--- a/mscore/importmidi/importmidi_model.cpp
+++ b/mscore/importmidi/importmidi_model.cpp
@@ -5,6 +5,11 @@
 #include "libmscore/instrtemplate.h"
 
 
+#ifndef QT_DEBUG
+#undef Q_ASSERT_X
+#define Q_ASSERT_X(cond,where,what) /**/
+#endif
+
 namespace Ms {
 
 class TracksModel::Column
diff --git a/mscore/importmidi/importmidi_operations.cpp b/mscore/importmidi/importmidi_operations.cpp
index 8a812a0817f030a0b846a45b0596e83debd4142e..b87ecc8004ca26c6307b57e023a4a0b7a9cbe3ef 100644
--- a/mscore/importmidi/importmidi_operations.cpp
+++ b/mscore/importmidi/importmidi_operations.cpp
@@ -1,6 +1,11 @@
 #include "importmidi_operations.h"
 
 
+#ifndef QT_DEBUG
+#undef Q_ASSERT_X
+#define Q_ASSERT_X(cond,where,what) /**/
+#endif
+
 namespace Ms {
 
 MidiOperations::Data midiImportOperations;
diff --git a/mscore/importmidi/importmidi_panel.cpp b/mscore/importmidi/importmidi_panel.cpp
index b503db01a07731da63a68d74c3f37f41c2a6990e..0260be01fbf3007fe2ef2c8a1cbd297566a07671 100644
--- a/mscore/importmidi/importmidi_panel.cpp
+++ b/mscore/importmidi/importmidi_panel.cpp
@@ -10,6 +10,11 @@
 #include "libmscore/score.h"
 
 
+#ifndef QT_DEBUG
+#undef Q_ASSERT_X
+#define Q_ASSERT_X(cond,where,what) /**/
+#endif
+
 namespace Ms {
 
 ImportMidiPanel::ImportMidiPanel(QWidget *parent)
diff --git a/mscore/importmidi/importmidi_quant.cpp b/mscore/importmidi/importmidi_quant.cpp
index d82f0da0c51a86b3639d5117e5fced827169922d..6dff8ca76d0a8692c7efdb734e7438b8a2a699ec 100644
--- a/mscore/importmidi/importmidi_quant.cpp
+++ b/mscore/importmidi/importmidi_quant.cpp
@@ -15,6 +15,10 @@
 #include <set>
 #include <deque>
 
+#ifndef QT_DEBUG
+#undef Q_ASSERT_X
+#define Q_ASSERT_X(cond,where,what) /**/
+#endif
 
 namespace Ms {
 
diff --git a/mscore/importmidi/importmidi_simplify.cpp b/mscore/importmidi/importmidi_simplify.cpp
index 735d82f92a023f835b0034eb396f01e02861dc4b..7df4003408e49c2f1d17abee003be3ebcfe2144d 100644
--- a/mscore/importmidi/importmidi_simplify.cpp
+++ b/mscore/importmidi/importmidi_simplify.cpp
@@ -13,6 +13,11 @@
 #include "midi/midifile.h"
 
 
+#ifndef QT_DEBUG
+#undef Q_ASSERT_X
+#define Q_ASSERT_X(cond,where,what) /**/
+#endif
+
 namespace Ms {
 namespace Simplify {
 
diff --git a/mscore/importmidi/importmidi_swing.cpp b/mscore/importmidi/importmidi_swing.cpp
index beed6ccf79659a66e2331b29a7b384172c120444..c97b2940e52ed897add96a1b1d4c4be8878e12b4 100644
--- a/mscore/importmidi/importmidi_swing.cpp
+++ b/mscore/importmidi/importmidi_swing.cpp
@@ -10,6 +10,11 @@
 #include "importmidi_fraction.h"
 
 
+#ifndef QT_DEBUG
+#undef Q_ASSERT_X
+#define Q_ASSERT_X(cond,where,what) /**/
+#endif
+
 namespace Ms {
 namespace Swing {
 
diff --git a/mscore/importmidi/importmidi_tempo.cpp b/mscore/importmidi/importmidi_tempo.cpp
index a9fc0c5646571125cf55171647d92a424779c2e3..c9174cc1ba42a4a57f834766b9b67d17105f66bf 100644
--- a/mscore/importmidi/importmidi_tempo.cpp
+++ b/mscore/importmidi/importmidi_tempo.cpp
@@ -9,6 +9,11 @@
 #include "importmidi_operations.h"
 
 
+#ifndef QT_DEBUG
+#undef Q_ASSERT_X
+#define Q_ASSERT_X(cond,where,what) /**/
+#endif
+
 namespace Ms {
 namespace MidiTempo {
 
diff --git a/mscore/importmidi/importmidi_tie.cpp b/mscore/importmidi/importmidi_tie.cpp
index 6ccbdff4772de8375a52f7670d6365d4bf12d5cf..b4eee64f035289b0bab2fa52ae7d069386238cc1 100644
--- a/mscore/importmidi/importmidi_tie.cpp
+++ b/mscore/importmidi/importmidi_tie.cpp
@@ -24,6 +24,11 @@
 #endif
 
 
+#ifndef QT_DEBUG
+#undef Q_ASSERT_X
+#define Q_ASSERT_X(cond,where,what) /**/
+#endif
+
 namespace Ms {
 namespace MidiTie {
 
diff --git a/mscore/importmidi/importmidi_tuplet.cpp b/mscore/importmidi/importmidi_tuplet.cpp
index 65e098c114e20ad03086ed0bd5e719b7cdce5c16..0cbcc7938864886df1e03cf5360186aa0fe1713f 100644
--- a/mscore/importmidi/importmidi_tuplet.cpp
+++ b/mscore/importmidi/importmidi_tuplet.cpp
@@ -12,6 +12,13 @@
 #include <set>
 
 
+#ifndef QT_DEBUG
+#undef Q_ASSERT_X
+#define Q_ASSERT_X(cond,where,what) /**/
+#undef Q_ASSERT
+#define Q_ASSERT(cond) /**/
+#endif
+
 namespace Ms {
 namespace MidiTuplet {
 
diff --git a/mscore/importmidi/importmidi_tuplet_detect.cpp b/mscore/importmidi/importmidi_tuplet_detect.cpp
index 8a18edae544a2be4ffc507e7b622559e1582994f..16ea980f1bbe48e4200aea86cbb7a211ab4a6038 100644
--- a/mscore/importmidi/importmidi_tuplet_detect.cpp
+++ b/mscore/importmidi/importmidi_tuplet_detect.cpp
@@ -10,6 +10,11 @@
 #include <set>
 
 
+#ifndef QT_DEBUG
+#undef Q_ASSERT_X
+#define Q_ASSERT_X(cond,where,what) /**/
+#endif
+
 namespace Ms {
 namespace MidiTuplet {
 
diff --git a/mscore/importmidi/importmidi_tuplet_filter.cpp b/mscore/importmidi/importmidi_tuplet_filter.cpp
index e19aa869b8e36ad9bbbe4ac720cdd5c3a522564b..75b7804ece6e71b9195c402a00eb11b388205c22 100644
--- a/mscore/importmidi/importmidi_tuplet_filter.cpp
+++ b/mscore/importmidi/importmidi_tuplet_filter.cpp
@@ -9,6 +9,11 @@
 #include <set>
 
 
+#ifndef QT_DEBUG
+#undef Q_ASSERT_X
+#define Q_ASSERT_X(cond,where,what) /**/
+#endif
+
 namespace Ms {
 namespace MidiTuplet {
 
diff --git a/mscore/importmidi/importmidi_tuplet_tonotes.cpp b/mscore/importmidi/importmidi_tuplet_tonotes.cpp
index 94c932b23838cf0a898161e7d0ce3271d1ef8f62..cfcd6feff6fcf6ce03856684dcb7bda36bc3196c 100644
--- a/mscore/importmidi/importmidi_tuplet_tonotes.cpp
+++ b/mscore/importmidi/importmidi_tuplet_tonotes.cpp
@@ -12,6 +12,11 @@
 #include "libmscore/chordrest.h"
 
 
+#ifndef QT_DEBUG
+#undef Q_ASSERT_X
+#define Q_ASSERT_X(cond,where,what) /**/
+#endif
+
 namespace Ms {
 namespace MidiTuplet {
 
diff --git a/mscore/importmidi/importmidi_tuplet_voice.cpp b/mscore/importmidi/importmidi_tuplet_voice.cpp
index afb3752d8ae8ead1256181b46446163b795e8508..fe27ea3ecea96ef41b8142d2e9757cd4afaeb388 100644
--- a/mscore/importmidi/importmidi_tuplet_voice.cpp
+++ b/mscore/importmidi/importmidi_tuplet_voice.cpp
@@ -10,6 +10,10 @@
 
 #include <set>
 
+#ifndef QT_DEBUG
+#undef Q_ASSERT_X
+#define Q_ASSERT_X(cond,where,what) /**/
+#endif
 
 namespace Ms {
 namespace MidiTuplet {
diff --git a/mscore/importmidi/importmidi_view.cpp b/mscore/importmidi/importmidi_view.cpp
index 128c1114eb9a012522c6a68dc0e9b288f9d76010..897f7a9832657cc1523ca4fcd7e4f3d850d46fb8 100644
--- a/mscore/importmidi/importmidi_view.cpp
+++ b/mscore/importmidi/importmidi_view.cpp
@@ -1,6 +1,11 @@
 #include "importmidi_view.h"
 
 
+#ifndef QT_DEBUG
+#undef Q_ASSERT_X
+#define Q_ASSERT_X(cond,where,what) /**/
+#endif
+
 class SizedHHeaderView : public QHeaderView
       {
    public:
diff --git a/mscore/importmidi/importmidi_voice.cpp b/mscore/importmidi/importmidi_voice.cpp
index dc0cd71e87673d98255a1635011298d9ebb153d5..3fdcf04456e763777b920216d7f4b717f79f1a85 100644
--- a/mscore/importmidi/importmidi_voice.cpp
+++ b/mscore/importmidi/importmidi_voice.cpp
@@ -11,6 +11,11 @@
 #include "libmscore/durationtype.h"
 
 
+#ifndef QT_DEBUG
+#undef Q_ASSERT_X
+#define Q_ASSERT_X(cond,where,what) /**/
+#endif
+
 namespace Ms {
 namespace MidiVoice {
 
