diff --git a/mscore/musescore.cpp b/mscore/musescore.cpp
index 38d2a2a62d67157b0a55cec2d60dbcbee4bc8f6f..064b3775cf83de29e425e5d03687c62c1a436304 100644
--- a/mscore/musescore.cpp
+++ b/mscore/musescore.cpp
@@ -1398,10 +1398,10 @@ MuseScore::MuseScore()
       }
       menuFile->addSeparator();
       menuFile->addAction(getAction("print"));
-#ifndef Q_OS_MAC // On Mac, the Quit option is already in the "MuseScore" menu
+// #ifndef Q_OS_MAC // On Mac, the Quit option is already in the "MuseScore" menu
       menuFile->addSeparator();
       menuFile->addAction(getAction("quit"));
-#endif
+// #endif
 
       //---------------------
       //    Menu Edit
