diff --git a/mscore/network/loginmanager.cpp b/mscore/network/loginmanager.cpp
index 36220d136..7e639c032 100644
--- a/mscore/network/loginmanager.cpp
+++ b/mscore/network/loginmanager.cpp
@@ -977,8 +977,10 @@ QNetworkRequest ApiRequest::buildRequest() const
       r.setRawHeader("Accept", "application/json");
       const ApiInfo& apiInfo = ApiInfo::instance();
       r.setHeader(QNetworkRequest::UserAgentHeader, apiInfo.userAgent);
+#ifdef USE_WEBENGINE
       r.setRawHeader(apiInfo.clientIdHeader, apiInfo.clientId);
       r.setRawHeader(apiInfo.apiKeyHeader, apiInfo.apiKey);
+#endif
 
       return r;
       }
