diff --git a/mscore/cloud/loginmanager.cpp b/mscore/cloud/loginmanager.cpp
index 9f36f897..bf8c126b 100644
--- a/mscore/cloud/loginmanager.cpp
+++ b/mscore/cloud/loginmanager.cpp
@@ -935,9 +935,10 @@ QNetworkRequest ApiRequest::buildRequest() const
       r.setRawHeader("Accept", "application/json");
       const ApiInfo& apiInfo = ApiInfo::instance();
       r.setHeader(QNetworkRequest::UserAgentHeader, apiInfo.userAgent);
+#ifdef USE_WEBENGINE
       r.setRawHeader(apiInfo.clientIdHeader, apiInfo.clientId);
       r.setRawHeader(apiInfo.apiKeyHeader, apiInfo.apiKey);
-
+#endif
       return r;
       }
 
