diff --git a/mscore/scoreview.cpp b/mscore/scoreview.cpp
index 0990539b4f6adf93884dd060e1010321c3ec538c..92aeea2fcfdee923f76eefee14ae4717078f42bd 100644
--- a/mscore/scoreview.cpp
+++ b/mscore/scoreview.cpp
@@ -2746,7 +2746,7 @@ void ScoreView::cmd(const char* s)
                   cv->updateGrips();
                   }},
       #ifdef OMR
-            {{"show-omr"}, [](ScoreView* cv, const QByteArray&) {
+            {{"show-omr"}, [this](ScoreView* cv, const QByteArray&) {
                   if (cv->score()->masterScore()->omr())
                         cv->showOmr(!cv->score()->masterScore()->showOmr());
                   }},
