diff --git a/mscore/cloud/loginmanager.cpp b/mscore/cloud/loginmanager.cpp
index 4abf7e3e4edf421e5db903147c7557d872b2d76d..c855606d523b25b0d2a5fe15929ff29caa95e3e0 100644
--- a/mscore/cloud/loginmanager.cpp
+++ b/mscore/cloud/loginmanager.cpp
@@ -858,9 +858,10 @@ QNetworkRequest ApiRequest::buildRequest() const
       r.setRawHeader("Accept", "application/json");
       const ApiInfo& apiInfo = ApiInfo::instance();
       r.setHeader(QNetworkRequest::UserAgentHeader, apiInfo.userAgent);
+#ifdef USE_WEBENGINE
       r.setRawHeader(apiInfo.clientIdHeader, apiInfo.clientId);
       r.setRawHeader(apiInfo.apiKeyHeader, apiInfo.apiKey);
-
+#endif
       return r;
       }
 
