diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7e8b10f12cb430adf3948d44e2ba6b4d083feda1..b732f4ec0657a5c8b9b0dc4d76e18ee9cff1cbda 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -62,6 +62,9 @@ if (MSVC)
    set (MINGW false)
 endif (MSVC)
 
+include(FeatureSummary)
+include(FindPkgConfig)
+
 # We need this early, before FindQt5
 option(BUILD_WEBENGINE "Built in webengine support" ON)
 
@@ -180,8 +183,8 @@ if (BUILD_CRASH_REPORTER)
 endif(BUILD_CRASH_REPORTER)
 
 if (APPLE)
-      set (CMAKE_CXX_COMPILER   clang++)
-      set (CMAKE_CXX_COMPILER_ID "Clang")
+#       set (CMAKE_CXX_COMPILER   clang++)
+#       set (CMAKE_CXX_COMPILER_ID "Clang")
       set(CMAKE_OSX_ARCHITECTURES x86_64)
 
       set(CMAKE_OSX_DEPLOYMENT_TARGET 10.10)
@@ -251,6 +254,7 @@ endif (${CMAKE_CXX_COMPILER_ID} MATCHES "GNU")
 if (NOT MSVC)
    set(CMAKE_CXX_FLAGS_DEBUG   "-g")
    set(CMAKE_CXX_FLAGS_RELEASE "-O2 -DNDEBUG -DQT_NO_DEBUG")
+   add_definitions("-DQT_NO_DEBUG")
 endif (NOT MSVC)
 
 if (APPLE)
@@ -371,7 +375,7 @@ endif(BUILD_FOR_WINSTORE)
 ##
 
 if (USE_SYSTEM_FREETYPE)
-      if (APPLE)
+      if (NOPE)
             PKGCONFIG1 (freetype2 2.5.2 FREETYPE_INCLUDE_DIRS FREETYPE_LIBDIR FREETYPE_LIBRARIES FREETYPE_CPP)
             if (FREETYPE_INCLUDE_DIRS)
                   STRING(REGEX REPLACE  "\"" "" FREETYPE_INCLUDE_DIRS ${FREETYPE_INCLUDE_DIRS})
@@ -380,10 +384,14 @@ if (USE_SYSTEM_FREETYPE)
             else (FREETYPE_INCLUDE_DIRS)
                   message(FATAL_ERROR "freetype >= 2.5.2 is required\n")
             endif (FREETYPE_INCLUDE_DIRS)
-      else (APPLE)
+      else (NOPE)
             find_package(Freetype REQUIRED)
-      endif (APPLE)
+      endif (NOPE)
 endif (USE_SYSTEM_FREETYPE)
+if (USE_SYSTEM_POPPLER)
+    pkg_check_modules(PKG_POPPLER poppler REQUIRED)
+    pkg_check_modules(PKG_POPPLER_QT5 poppler-qt5 REQUIRED)
+endif ()
 
 ##
 ## ALSA >= 1.0.0
@@ -777,8 +785,8 @@ if (OMR)
    include_directories(${PROJECT_SOURCE_DIR}/thirdparty/poppler)
    include_directories(${PROJECT_SOURCE_DIR}/thirdparty/poppler/poppler)
  else (NOT USE_SYSTEM_POPPLER)
-   include_directories(/usr/include/poppler)
-   include_directories(/usr/include/poppler/qt5)
+   include_directories(${PKG_POPPLER_INCLUDEDIR}/poppler)
+   include_directories(${PKG_POPPLER_QT5_INCLUDEDIR}/poppler/qt5)
  endif (NOT USE_SYSTEM_POPPLER)
 endif (OMR)
 
@@ -937,6 +945,8 @@ if (MSVC)
 
 endif (MSVC)
 
+feature_summary(WHAT ALL   FATAL_ON_MISSING_REQUIRED_PACKAGES)
+
 ## TEMP: Display all variables!
 ### message(STATUS "===========================================================")
 ### message(STATUS "VARIABLES:")
