diff --git a/mscore/musescore.cpp b/mscore/musescore.cpp
index d2f2cbac..dea7c46a 100644
--- a/mscore/musescore.cpp
+++ b/mscore/musescore.cpp
@@ -16,6 +16,7 @@
 #include <QStyleFactory>
 #include <QStandardPaths>
 #include <QDir>
+#include <QSGRendererInterface>
 
 #include "config.h"
 
@@ -7520,6 +7532,12 @@ MuseScoreApplication* MuseScoreApplication::initApplication(int& argc, char** ar
       QCoreApplication::setOrganizationDomain("musescore.org");
       QCoreApplication::setApplicationVersion(MuseScore::fullVersion());
 
+#if defined(__linux__)
+      qWarning() << "QQW sceneGraphBackend was:" << QQuickWindow::sceneGraphBackend();
+      QQuickWindow::setSceneGraphBackend(QSGRendererInterface::Software);
+      qWarning() << "QQW sceneGraphBackend now:" << QQuickWindow::sceneGraphBackend();
+#endif
+
 #ifdef BUILD_CRASH_REPORTER
       {
       static_assert(sizeof(CRASHREPORTER_EXECUTABLE) > 1,
