diff --git a/build/MacOSXBundleInfo.plist.in b/build/MacOSXBundleInfo.plist.in
index 5dce1d118a9b494a8dcb251963b54744a926efa8..6ab5e09c5326d327f6e907b8e7bd596ee1b9dd5a 100644
--- a/build/MacOSXBundleInfo.plist.in
+++ b/build/MacOSXBundleInfo.plist.in
@@ -447,7 +447,7 @@
 	<key>LSRequiresCarbon</key>
 	<true/>
 	<key>LSMinimumSystemVersion</key>
-	<string>10.10.0</string>
+	<string>10.7.0</string>
 	<key>NSHumanReadableCopyright</key>
 	<string>${MACOSX_BUNDLE_COPYRIGHT}</string>
 	<key>LSApplicationCategoryType</key>
diff --git a/mscore/macos/cocoabridge.mm b/mscore/macos/cocoabridge.mm
index 25d629efe9ca43049b9d2950caa877c98f0e706d..2d8be5fc3724860ade0240d19c0a9b22a113bd8a 100644
--- a/mscore/macos/cocoabridge.mm
+++ b/mscore/macos/cocoabridge.mm
@@ -51,8 +51,10 @@ bool CocoaBridge::isSystemDarkModeSupported()
 
 void CocoaBridge::setWindowAppearanceIsDark(bool flag)
       {
+#if __MAC_OS_X_VERSION_MAX_ALLOWED >= 101400
       if (@available(macOS 10.14, *))
             [NSApp setAppearance:[NSAppearance appearanceNamed:flag ? NSAppearanceNameDarkAqua : NSAppearanceNameAqua]];
+#endif
       }
 
 void CocoaBridge::setAllowsAutomaticWindowTabbing(bool flag)
