diff --git a/avsomr/CMakeLists.txt b/avsomr/CMakeLists.txt
index d4d20eb3..4a20318e 100644
--- a/avsomr/CMakeLists.txt
+++ b/avsomr/CMakeLists.txt
@@ -72,13 +72,13 @@ if (NOT MSVC)
    set_target_properties (
       avsomr
       PROPERTIES
-         COMPILE_FLAGS "${PCH_INCLUDE} -g -Wall -Wextra -Winvalid-pch"
+         COMPILE_FLAGS "${PCH_INCLUDE} -g -Wall -Wextra -Winvalid-pch -DAVSOMR"
       )
 else (NOT MSVC)
    set_target_properties (
       avsomr
       PROPERTIES
-         COMPILE_FLAGS "${PCH_INCLUDE}"
+         COMPILE_FLAGS "${PCH_INCLUDE} -DAVSOMR"
       )
 endif (NOT MSVC)   
 
diff --git a/avsomr/msmrwriter.cpp b/avsomr/msmrwriter.cpp
index 070e1752..3d9e305a 100644
--- a/avsomr/msmrwriter.cpp
+++ b/avsomr/msmrwriter.cpp
@@ -47,7 +47,7 @@ bool MsmrWriter::saveMsmrFile(Ms::MasterScore* score, QIODevice* file, const QFi
 
       QByteArray mscz;
       QBuffer b(&mscz);
-      bool ok = score->saveCompressedFile(&b, info, false, false);
+      bool ok = score->saveCompressedFile(&b, info.absolutePath(), false, false);
       if (!ok) {
             LOGE() << "failed save mscz file";
             return false;
