diff --git a/mscore/musescore.cpp b/mscore/musescore.cpp
index 4f8d9f5a4a70e271a9b5a47a5a2e5d6c32c039e1..2a998f80b4ec2c24fcf8509e4589ed50c5397262 100644
--- a/mscore/musescore.cpp
+++ b/mscore/musescore.cpp
@@ -1393,10 +1393,10 @@ MuseScore::MuseScore()
             menuFile->addAction(getAction("media"));
       menuFile->addSeparator();
       menuFile->addAction(getAction("print"));
-#ifndef Q_OS_MAC
+// #ifndef Q_OS_MAC // Mac Qt will move the Quit action in which case the separator isn't show either
       menuFile->addSeparator();
       menuFile->addAction(getAction("quit"));
-#endif
+// #endif
 
       //---------------------
       //    Menu Edit
