diff --git a/mscore/macos/cocoabridge.mm b/mscore/macos/cocoabridge.mm
index 25d629ef..866f0739 100644
--- a/mscore/macos/cocoabridge.mm
+++ b/mscore/macos/cocoabridge.mm
@@ -24,16 +24,20 @@ id<NSObject> darkModeObserverToken;
 
 void CocoaBridge::observeDarkModeSwitches(std::function<void()> f)
       {
+#if QT_MACOS_PLATFORM_SDK_EQUAL_OR_ABOVE(__MAC_10_14)
       if (@available(macOS 10.14, *))
             darkModeObserverToken = [[NSDistributedNotificationCenter defaultCenter]
                                      addObserverForName:@"AppleInterfaceThemeChangedNotification" object:nil
                                      queue:nil usingBlock:^(NSNotification*) {   f();   }];
+#endif
       }
 
 void CocoaBridge::removeObservers()
       {
+#if QT_MACOS_PLATFORM_SDK_EQUAL_OR_ABOVE(__MAC_10_14)
       if (@available(macOS 10.14, *))
             [[NSDistributedNotificationCenter defaultCenter] removeObserver:darkModeObserverToken];
+#endif
       }
 
 bool CocoaBridge::isSystemDarkTheme()
@@ -44,15 +48,19 @@ bool CocoaBridge::isSystemDarkTheme()
 
 bool CocoaBridge::isSystemDarkModeSupported()
       {
+#if QT_MACOS_PLATFORM_SDK_EQUAL_OR_ABOVE(__MAC_10_14)
       if (@available(macOS 10.14, *))
             return true;
+#endif
       return false;
       }
 
 void CocoaBridge::setWindowAppearanceIsDark(bool flag)
       {
+#if __MAC_OS_X_VERSION_MAX_ALLOWED >= 101400
       if (@available(macOS 10.14, *))
             [NSApp setAppearance:[NSAppearance appearanceNamed:flag ? NSAppearanceNameDarkAqua : NSAppearanceNameAqua]];
+#endif
       }
 
 void CocoaBridge::setAllowsAutomaticWindowTabbing(bool flag)
