diff --git a/mscore/pianoroll/pianoview.cpp b/mscore/pianoroll/pianoview.cpp
index e358c5c0b8c7b5822a0859a214e289b7c62a58bf..1d37b80bfee613b83b839f81389f40bbd4a92909 100644
--- a/mscore/pianoroll/pianoview.cpp
+++ b/mscore/pianoroll/pianoview.cpp
@@ -1658,7 +1660,7 @@ void PianoView::updateNotes()
       scene()->clear();
       clearNoteData();
 
-      int staffIdx = _staff->idx();
+      int staffIdx = _staff ? _staff->idx() : -1;
       if (staffIdx == -1)
             return;
 
