diff --git a/src/plugins/platformthemes/gtk2/gtk2.pro b/src/plugins/platformthemes/gtk2/gtk2.pro
index 7ee872e..855fc72 100644
--- a/src/plugins/platformthemes/gtk2/gtk2.pro
+++ b/src/plugins/platformthemes/gtk2/gtk2.pro
@@ -1,6 +1,6 @@
 TARGET = qgtk2
 
-QT += core-private gui-private
+QT += dbus core-private gui-private
 greaterThan(QT_MAJOR_VERSION, 5)|greaterThan(QT_MINOR_VERSION, 7): \
     QT += theme_support-private
 else: \
