--- core/orig.bytearraymodel_p.cpp	2017-06-16 14:15:07.394003240 +0200
+++ core/bytearraymodel_p.cpp	2017-06-16 14:15:24.925003373 +0200
@@ -339,12 +339,14 @@
     else if( fillLength > lengthToEnd )
         fillLength = addSize( fillLength, offset, false );
 
-    memset( &mData[offset], fillByte, fillLength );
-    mModified = true;
+    if (fillLength) {
+        memset( &mData[offset], fillByte, fillLength );
+        mModified = true;
 
-    emit p->contentsChanged( ArrayChangeMetricsList::oneReplacement(offset,fillLength,fillLength) );
-    if( ! wasModifiedBefore )
-        emit p->modifiedChanged( true );
+        emit p->contentsChanged( ArrayChangeMetricsList::oneReplacement(offset,fillLength,fillLength) );
+        if( ! wasModifiedBefore )
+            emit p->modifiedChanged( true );
+    }
     return fillLength;
 }
 
