--- program/orig.program.cpp	2015-12-09 11:13:38.000000000 +0100
+++ program/program.cpp	2015-12-27 23:30:26.000000000 +0100
@@ -90,7 +90,7 @@
 
     OktetaAboutData aboutData;
     KAboutData::setApplicationData( aboutData );
-    QApplication::setWindowIcon(QIcon::fromTheme(QStringLiteral("okteta")));
+    QApplication::setWindowIcon(QIcon::fromTheme(QStringLiteral("okteta"), mApp.windowIcon()));
 
     KDBusService programDBusService;
 
