Description: load testability from environment variable
Author: Thomi Richards <thomi.richards@canonical.com>
Forwarded: https://bugreports.qt-project.org/browse/QTBUG-32974

--- qtbase/src/gui/kernel/qguiapplication.cpp
+++ qtbase/src/gui/kernel/qguiapplication.cpp
@@ -1256,6 +1256,9 @@
         argc = j;
     }
 
+    if (qgetenv("QT_LOAD_TESTABILITY").toInt() > 0)
+        loadTestability = true;
+
 #if defined(QT_DEBUG) && defined(Q_OS_LINUX)
     if (!doGrabUnderDebugger && !QGuiApplicationPrivate::noGrab && runningUnderDebugger()) {
         QGuiApplicationPrivate::noGrab = true;
