diff --git a/configure.ac b/configure.ac
index 5b18fd34..0b181800 100644
--- a/configure.ac
+++ b/configure.ac
@@ -312,7 +312,7 @@ if test "x$enable_cxx" = "x1" ; then
     JE_CXXFLAGS_ADD([-g3])
 
     SAVED_LIBS="${LIBS}"
-    JE_APPEND_VS(LIBS, -lstdc++)
+    JE_APPEND_VS(LIBS, -lc++)
     JE_COMPILABLE([libstdc++ linkage], [
 #include <stdlib.h>
 ], [[
@@ -2526,8 +2526,7 @@ fi
 dnl ============================================================================
 dnl Enable background threads if possible.
 
-if test "x${have_pthread}" = "x1" -a "x${je_cv_os_unfair_lock}" != "xyes" -a \
-       "x${abi}" != "xmacho" ; then
+if test "x${have_pthread}" = "x1" -a "x${je_cv_os_unfair_lock}" != "xyes" ; then
   AC_DEFINE([JEMALLOC_BACKGROUND_THREAD], [ ], [ ])
 fi
 
