diff --git a/configure.ac b/configure.ac
index 5b18fd34..7fdc0abb 100644
--- a/configure.ac
+++ b/configure.ac
@@ -499,7 +499,7 @@ typedef unsigned __int32 uint32_t;
 	fclose(f);
 	return 0;
 ]])],
-                   [je_cv_lg_vaddr=`cat conftest.out`],
+                   [je_cv_lg_vaddr=`cat conftest.out`;if test "x${je_cv_lg_vaddr}" = "x0" ; then je_cv_lg_vaddr=48 ;fi],
                    [je_cv_lg_vaddr=error],
                    [je_cv_lg_vaddr=57]))
       if test "x${je_cv_lg_vaddr}" != "x" ; then
