diff --git a/data/konversationui.rc b/data/konversationui.rc
index 171d32a..4fb8091 100644
--- a/data/konversationui.rc
+++ b/data/konversationui.rc
@@ -66,7 +66,7 @@
 
   <Menu name="tabContextMenu">
     <Action name="tab_notifications" />
-    <Action name="tab_encoding" />
+<!--     <Action name="tab_encoding" /> -->
     <Separator />
     <ActionList name="server_actions" />
     <Separator />

diff --git a/src/viewer/viewcontainer.cpp b/src/viewer/viewcontainer.cpp
index aafa5cc..c9ea59d 100644
--- a/src/viewer/viewcontainer.cpp
+++ b/src/viewer/viewcontainer.cpp
@@ -2194,6 +2194,7 @@ void ViewContainer::showViewContextMenu(QWidget* tab, const QPoint& pos)
 
     if (!menu) return;
 
+    menu->setTitle("Tab Context Menu");
     KToggleAction* autoJoinAction = qobject_cast<KToggleAction*>(actionCollection()->action("tab_autojoin"));
     KToggleAction* autoConnectAction = qobject_cast<KToggleAction*>(actionCollection()->action("tab_autoconnect"));
     QAction* rejoinAction = actionCollection()->action("rejoin_channel");
@@ -2209,8 +2210,10 @@ void ViewContainer::showViewContextMenu(QWidget* tab, const QPoint& pos)
 
     if (viewType == ChatWindow::Channel)
     {
+#ifndef Q_OS_OSX
         QAction* action = actionCollection()->action("tab_encoding");
         menu->insertAction(action, autoJoinAction);
+#endif
 
         Channel *channel = static_cast<Channel*>(view);
         if (channel->rejoinable() && rejoinAction)
@@ -2220,16 +2223,22 @@ void ViewContainer::showViewContextMenu(QWidget* tab, const QPoint& pos)
         }
     }
 
+#ifndef Q_OS_OSX
     if (viewType == ChatWindow::Konsole)
     {
         QAction* action = actionCollection()->action("tab_encoding");
         menu->insertAction(action, renameAct);
     }
+#endif
 
     if (viewType == ChatWindow::Status)
     {
+#ifndef Q_OS_OSX
         QAction* action = actionCollection()->action("tab_encoding");
         menu->insertAction(action, autoConnectAction);
+#else
+        QAction* action;
+#endif
 
         QList<QAction *> serverActions;
 
