--- mesonbuild/minstall.py	2021-04-16 18:38:49.817468340 +0200
+++ mesonbuild/minstall.py	2021-04-16 18:41:17.320604166 +0200
@@ -653,8 +653,11 @@
             if file_copied:
                 self.did_install_something = True
                 try:
-                    self.fix_rpath(outname, t.rpath_dirs_to_remove, install_rpath, final_path,
-                                         install_name_mappings, verbose=False)
+                    if install_rpath:
+                        self.fix_rpath(outname, t.rpath_dirs_to_remove, install_rpath, final_path,
+                                             install_name_mappings, verbose=False)
+                    else:
+                        print("RPATH changes at install time disabled")
                 except SystemExit as e:
                     if isinstance(e.code, int) and e.code == 0:
                         pass
