diff -urN meson-0.51.1-orig/mesonbuild/compilers/clike.py meson-0.51.1/mesonbuild/compilers/clike.py
--- meson-0.51.1-orig/mesonbuild/compilers/clike.py	2019-08-12 01:17:05.000000000 -0400
+++ meson-0.51.1/mesonbuild/compilers/clike.py	2019-08-12 01:19:02.000000000 -0400
@@ -991,7 +991,7 @@
         for f in files:
             if not f.is_file():
                 continue
-            archs = mesonlib.darwin_get_object_archs(f)
+            archs = mesonlib.darwin_get_object_archs(str(f))
             if archs and env.machines.host.cpu_family in archs:
                 return f
             else:
