diff --git a/src/modules/platforms/mac/MacPlatformIntegration.mm b/src/modules/platforms/mac/MacPlatformIntegration.mm
index bbb37ab7839f991d307121bc7877a1436eabf4cf..c2bd07eefa71526d0bf9d39c5ba90fb2fc3ce3a7 100644
--- a/src/modules/platforms/mac/MacPlatformIntegration.mm
+++ b/src/modules/platforms/mac/MacPlatformIntegration.mm
@@ -249,7 +250,7 @@ void MacPlatformIntegration::timerEvent(QTimerEvent *event)
 
 				m_notifications.remove(allNotifications.at(i));
 			}
-			else if (m_notifications[allNotifications.at(i)]->getCreationTime().secsTo(QDateTime::currentDateTime()) >= visibilityDuration)
+			else if (m_notifications[allNotifications.at(i)]->getMessage().creationTime.secsTo(QDateTime::currentDateTime()) >= visibilityDuration)
 			{
 				[[NSUserNotificationCenter defaultUserNotificationCenter] removeDeliveredNotification:existingNotifications[allNotifications.at(i)]];
 
