diff --git a/CMakeLists.txt b/CMakeLists.txt
index ae4843c865fe470aac356c01a7712342a24a596e..d05599d2c35839574f8e69596bd13202af3578f8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -91,6 +91,10 @@ set_package_properties(Qt5WebKit PROPERTIES URL "https://www.qt.io/" DESCRIPTION
 set_package_properties(Qt5WebKitWidgets PROPERTIES URL "https://www.qt.io/" DESCRIPTION "QtWebKit based backend (widgets)" TYPE OPTIONAL)
 set_package_properties(Hunspell PROPERTIES URL "https://hunspell.github.io/" DESCRIPTION "Generic spell checking support" TYPE OPTIONAL)
 
+if(ENABLE_SPELLCHECK)
+    find_package(KF5Sonnet)
+endif()
+
 set(otter_src
 	src/main.cpp
 	src/core/ActionExecutor.cpp
@@ -399,7 +403,7 @@ if (Qt5WebKitWidgets_FOUND AND ENABLE_QTWEBKIT)
 		src/modules/backends/web/qtwebkit/3rdparty/qtftp/qurlinfo.cpp
 	)
 
-	if (HUNSPELL_FOUND AND ENABLE_SPELLCHECK)
+	if ((HUNSPELL_FOUND  OR KF5Sonnet_FOUND) AND ENABLE_SPELLCHECK)
 		set(otter_src
 			${otter_src}
 			src/modules/backends/web/qtwebkit/QtWebKitSpellChecker.cpp
@@ -455,20 +459,22 @@ if (HUNSPELL_FOUND AND ENABLE_SPELLCHECK)
 	add_definitions(-DOTTER_ENABLE_SPELLCHECK)
 	include_directories(${HUNSPELL_INCLUDE_DIR})
 
-	set(otter_src
-		${otter_src}
-		3rdparty/sonnet/src/core/client.cpp
-		3rdparty/sonnet/src/core/loader.cpp
-		3rdparty/sonnet/src/core/speller.cpp
-		3rdparty/sonnet/src/core/spellerplugin.cpp
-		3rdparty/sonnet/src/core/settings.cpp
-		3rdparty/sonnet/src/core/textbreaks.cpp
-		3rdparty/sonnet/src/core/tokenizer.cpp
-		3rdparty/sonnet/src/ui/highlighter.cpp
-		3rdparty/sonnet/src/plugins/hunspell/hunspellclient.cpp
-		3rdparty/sonnet/src/plugins/hunspell/hunspelldebug.cpp
-		3rdparty/sonnet/src/plugins/hunspell/hunspelldict.cpp
-	)
+    if (NOT KF5Sonnet_FOUND)
+        set(otter_src
+            ${otter_src}
+            3rdparty/sonnet/src/core/client.cpp
+            3rdparty/sonnet/src/core/loader.cpp
+            3rdparty/sonnet/src/core/speller.cpp
+            3rdparty/sonnet/src/core/spellerplugin.cpp
+            3rdparty/sonnet/src/core/settings.cpp
+            3rdparty/sonnet/src/core/textbreaks.cpp
+            3rdparty/sonnet/src/core/tokenizer.cpp
+            3rdparty/sonnet/src/ui/highlighter.cpp
+            3rdparty/sonnet/src/plugins/hunspell/hunspellclient.cpp
+            3rdparty/sonnet/src/plugins/hunspell/hunspelldebug.cpp
+            3rdparty/sonnet/src/plugins/hunspell/hunspelldict.cpp
+        )
+    endif()
 endif ()
 
 if (WIN32)
@@ -546,9 +552,13 @@ if (Qt5WebKitWidgets_FOUND AND ENABLE_QTWEBKIT)
 	target_link_libraries(otter-browser Qt5::WebKit Qt5::WebKitWidgets)
 endif ()
 
-if (HUNSPELL_FOUND AND ENABLE_SPELLCHECK)
-	target_link_libraries(otter-browser ${HUNSPELL_LIBRARIES})
-endif ()
+if(ENABLE_SPELLCHECK)
+    if (KF5Sonnet_FOUND)
+        target_link_libraries(otter-browser KF5::SonnetCore KF5::SonnetUi)
+    elseif (HUNSPELL_FOUND)
+        target_link_libraries(otter-browser ${HUNSPELL_LIBRARIES})
+    endif ()
+endif()
 
 if (WIN32)
 	target_link_libraries(otter-browser Qt5::WinExtras ole32 shell32 advapi32 user32)
