diff --git a/CMakeLists.txt b/CMakeLists.txt
index 54b20c08091bb73837ad54ea100f292541b277ab..de24302326ebef7373f471d1026183e10d353435 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -82,7 +82,7 @@ option(ENABLE_QTWEBKIT "Enable QtWebKit backend (requires Qt 5.4)" ON)
 option(ENABLE_CRASHREPORTS "Enable built-in crash reporting (only for official builds)" OFF)
 
 find_package(Qt5 5.6.0 REQUIRED COMPONENTS Core DBus Gui Multimedia Network PrintSupport Qml Svg Widgets XmlPatterns)
-find_package(Qt5WebEngineWidgets 5.9.0 QUIET)
+find_package(Qt5WebEngineWidgets 5.8.0 QUIET)
 find_package(Qt5WebKitWidgets 5.212.0 QUIET)
 find_package(Hunspell 1.5.0 QUIET)
 

diff --git a/src/modules/backends/web/qtwebengine/QtWebEngineWebWidget.cpp b/src/modules/backends/web/qtwebengine/QtWebEngineWebWidget.cpp
index 4ac50a19c4feb78945a8366f1c5e9b3c0e3beaff..06f0674b5301ff696fba6c3da608e52629f4f1e8 100644
--- a/src/modules/backends/web/qtwebengine/QtWebEngineWebWidget.cpp
+++ b/src/modules/backends/web/qtwebengine/QtWebEngineWebWidget.cpp
@@ -56,6 +56,9 @@
 #include <QtWidgets/QAction>
 #include <QtWidgets/QToolTip>
 #include <QtWidgets/QVBoxLayout>
+#if QTWEBENGINECORE_VERSION < 0x050900
+#include <QtCore/QFile>
+#endif
 
 namespace Otter
 {
@@ -188,10 +191,21 @@ void QtWebEngineWebWidget::search(const QString &query, const QString &searchEng
 
 		if (method == QNetworkAccessManager::PostOperation)
 		{
+#if QTWEBENGINECORE_VERSION < 0x050900
+			QFile file(QLatin1String(":/modules/backends/web/qtwebengine/resources/sendPost.js"));
+
+			if (file.open(QIODevice::ReadOnly))
+			{
+				m_page->runJavaScript(QString(file.readAll()).arg(request.url().toString()).arg(QString(body)));
+
+				file.close();
+			}
+#else
 			QWebEngineHttpRequest httpRequest(request.url(), QWebEngineHttpRequest::Post);
 			httpRequest.setPostData(body);
 
 			m_page->load(httpRequest);
+#endif
 		}
 		else
 		{

commit 38178b5bd4ab8ab22ed8c109b7cdd48748f2b220
Author: Michał Dutkiewicz <emdeck@gmail.com>
Date:   Wed Jul 12 13:27:38 2017 +0200

    [Mac] Switch away from deprecated API

diff --git b/src/modules/platforms/mac/MacPlatformIntegration.mm a/src/modules/platforms/mac/MacPlatformIntegration.mm
index 059adc9..38a8b4a 100644
--- b/src/modules/platforms/mac/MacPlatformIntegration.mm
+++ a/src/modules/platforms/mac/MacPlatformIntegration.mm
@@ -78,7 +79,7 @@ static MacPlatformIntegrationDockIconView *getSharedDockIconView = nil;
 
 	NSRect boundary = [self bounds];
 
-	[[NSApp applicationIconImage] drawInRect:boundary fromRect:NSZeroRect operation:NSCompositingOperationCopy fraction:1.0];
+	[[NSApp applicationIconImage] drawInRect:boundary fromRect:NSZeroRect operation:NSCompositeCopy fraction:1.0];
 
 	if (m_value < 0)
 	{
commit a036898704f2e7e43fddd105ff8e1ec9dcc71590
Author: Michał Dutkiewicz <emdeck@gmail.com>
Date:   Tue Aug 8 19:42:35 2017 +0200

    [Mac] Drop extra checks since 10.10 is now the oldest supported version

diff --git b/src/modules/platforms/mac/MacPlatformIntegration.mm a/src/modules/platforms/mac/MacPlatformIntegration.mm
index cbd61ec..059adc9 100644
--- b/src/modules/platforms/mac/MacPlatformIntegration.mm
+++ a/src/modules/platforms/mac/MacPlatformIntegration.mm
@@ -31,6 +31,7 @@
 #include <QtCore/QDateTime>
 #include <QtCore/QFileInfo>
 #include <QtCore/QMimeData>
+#include <QtCore/QSysInfo>
 #include <QtCore/QTemporaryFile>
 #include <QtCore/QXmlStreamWriter>
 #include <QtGui/QDesktopServices>
@@ -356,6 +357,11 @@ void MacPlatformIntegration::updateTransfersProgress()
 
 void MacPlatformIntegration::showNotification(Notification *notification)
 {
+	if (QSysInfo::MacintoshVersion < QSysInfo::MV_10_8)
+	{
+		return;
+	}
+
 	QString title;
 
 	switch (notification->getLevel())
@@ -442,7 +448,7 @@ QVector<ApplicationInformation> MacPlatformIntegration::getApplicationsForMimeTy
 
 bool MacPlatformIntegration::canShowNotifications() const
 {
-	return true;
+	return (QSysInfo::MacintoshVersion >= QSysInfo::MV_10_8);
 }
 
 }
