diff --git a/src/modules/backends/web/qtwebengine/QtWebEngineWebWidget.cpp b/src/modules/backends/web/qtwebengine/QtWebEngineWebWidget.cpp
index 4ac50a19c4feb78945a8366f1c5e9b3c0e3beaff..9a3cdc09448d3cc9866d51912d979bf16d47b317 100644
--- a/src/modules/backends/web/qtwebengine/QtWebEngineWebWidget.cpp
+++ b/src/modules/backends/web/qtwebengine/QtWebEngineWebWidget.cpp
@@ -188,10 +188,21 @@ void QtWebEngineWebWidget::search(const QString &query, const QString &searchEng
 
 		if (method == QNetworkAccessManager::PostOperation)
 		{
+#if QTWEBENGINECORE_VERSION < 0x050700
+-			QFile file(QLatin1String(":/modules/backends/web/qtwebengine/resources/sendPost.js"));
+-
+-			if (file.open(QIODevice::ReadOnly))
+-			{
+-				m_page->runJavaScript(QString(file.readAll()).arg(request.url().toString()).arg(QString(body)));
+-
+-				file.close();
+-			}
+#else
 			QWebEngineHttpRequest httpRequest(request.url(), QWebEngineHttpRequest::Post);
 			httpRequest.setPostData(body);
 
 			m_page->load(httpRequest);
+#endif
 		}
 		else
 		{
