diff --git a/CMakeLists.txt b/CMakeLists.txt
index 188a84638f8b9204613d3cdf98564ba04df3c4d1..525b4633f3a3d255a65915a4b4b92bf43bc36477 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -82,6 +82,7 @@ find_package(Qt5 5.4.0 REQUIRED COMPONENTS Core DBus Gui Multimedia Network Prin
 find_package(Qt5WebEngineWidgets 5.9.0 QUIET)
 find_package(Qt5WebKitWidgets 5.4.0 QUIET)
 find_package(Hunspell 1.3.0 QUIET)
+find_package(KF5Sonnet)
 
 set_package_properties(Qt5WebEngineCore PROPERTIES URL "https://www.qt.io/" DESCRIPTION "QtWebEngine based backend (core)" TYPE OPTIONAL)
 set_package_properties(Qt5WebEngineWidgets PROPERTIES URL "https://www.qt.io/" DESCRIPTION "QtWebEngine based backend (widgets)" TYPE OPTIONAL)
@@ -381,7 +382,7 @@ if (Qt5WebKitWidgets_FOUND AND ENABLE_QTWEBKIT)
 		src/modules/backends/web/qtwebkit/3rdparty/qtftp/qurlinfo.cpp
 	)
 
-	if (HUNSPELL_FOUND)
+	if (HUNSPELL_FOUND OR KF5Sonnet_FOUND)
 		set(otter_src
 			${otter_src}
 			src/modules/backends/web/qtwebkit/qwebkitplatformplugin.h
@@ -439,20 +440,22 @@ if (HUNSPELL_FOUND)
 	add_definitions(-DOTTER_ENABLE_SPELLCHECK -DQT_STATICPLUGIN)
 	include_directories(${HUNSPELL_INCLUDE_DIR})
 
-	set(otter_src
-		${otter_src}
-		3rdparty/sonnet/src/core/client.cpp
-		3rdparty/sonnet/src/core/loader.cpp
-		3rdparty/sonnet/src/core/speller.cpp
-		3rdparty/sonnet/src/core/spellerplugin.cpp
-		3rdparty/sonnet/src/core/settings.cpp
-		3rdparty/sonnet/src/core/textbreaks.cpp
-		3rdparty/sonnet/src/core/tokenizer.cpp
-		3rdparty/sonnet/src/ui/highlighter.cpp
-		3rdparty/sonnet/src/plugins/hunspell/hunspellclient.cpp
-		3rdparty/sonnet/src/plugins/hunspell/hunspelldebug.cpp
-		3rdparty/sonnet/src/plugins/hunspell/hunspelldict.cpp
-	)
+	if (NOT KF5Sonnet_FOUND)
+		set(otter_src
+			${otter_src}
+			3rdparty/sonnet/src/core/client.cpp
+			3rdparty/sonnet/src/core/loader.cpp
+			3rdparty/sonnet/src/core/speller.cpp
+			3rdparty/sonnet/src/core/spellerplugin.cpp
+			3rdparty/sonnet/src/core/settings.cpp
+			3rdparty/sonnet/src/core/textbreaks.cpp
+			3rdparty/sonnet/src/core/tokenizer.cpp
+			3rdparty/sonnet/src/ui/highlighter.cpp
+			3rdparty/sonnet/src/plugins/hunspell/hunspellclient.cpp
+			3rdparty/sonnet/src/plugins/hunspell/hunspelldebug.cpp
+			3rdparty/sonnet/src/plugins/hunspell/hunspelldict.cpp
+		)
+	endif (NOT KF5Sonnet_FOUND)
 endif (HUNSPELL_FOUND)
 
 if (WIN32)
@@ -524,9 +527,11 @@ if (Qt5WebKitWidgets_FOUND AND ENABLE_QTWEBKIT)
 	target_link_libraries(otter-browser Qt5::WebKit Qt5::WebKitWidgets)
 endif (Qt5WebKitWidgets_FOUND AND ENABLE_QTWEBKIT)
 
-if (HUNSPELL_FOUND)
+if (KF5Sonnet_FOUND)
+	target_link_libraries(otter-browser KF5::SonnetCore KF5::SonnetUi)
+elseif (HUNSPELL_FOUND)
 	target_link_libraries(otter-browser ${HUNSPELL_LIBRARIES})
-endif (HUNSPELL_FOUND)
+endif (KF5Sonnet_FOUND)
 
 if (WIN32)
 	target_link_libraries(otter-browser Qt5::WinExtras ole32 shell32 advapi32 user32)
