diff --git a/src/modules/platforms/mac/MacPlatformIntegration.mm b/src/modules/platforms/mac/MacPlatformIntegration.mm
index bbb37ab7839f991d307121bc7877a1436eabf4cf..04661544ab6fa9b2e27ab910e5f50c62f8430fc6 100644
--- a/src/modules/platforms/mac/MacPlatformIntegration.mm
+++ b/src/modules/platforms/mac/MacPlatformIntegration.mm
@@ -31,6 +31,7 @@
 #include <QtCore/QDateTime>
 #include <QtCore/QFileInfo>
 #include <QtCore/QMimeData>
+#include <QtCore/QSysInfo>
 #include <QtCore/QTemporaryFile>
 #include <QtCore/QXmlStreamWriter>
 #include <QtGui/QDesktopServices>
@@ -78,7 +79,7 @@ static MacPlatformIntegrationDockIconView *getSharedDockIconView = nil;
 
 	NSRect boundary = [self bounds];
 
-	[[NSApp applicationIconImage] drawInRect:boundary fromRect:NSZeroRect operation:NSCompositingOperationCopy fraction:1.0];
+	[[NSApp applicationIconImage] drawInRect:boundary fromRect:NSZeroRect operation:NSCompositeCopy fraction:1.0];
 
 	if (m_value < 0)
 	{
@@ -362,6 +363,11 @@ void MacPlatformIntegration::updateTransfersProgress()
 
 void MacPlatformIntegration::showNotification(Notification *notification)
 {
+	if (QSysInfo::MacintoshVersion < QSysInfo::MV_10_8)
+	{
+		return;
+	}
+
 	const quint64 identifier(++m_notificationIdentifier);
 
 	m_notifications[identifier] = notification;
@@ -432,7 +438,7 @@ QVector<ApplicationInformation> MacPlatformIntegration::getApplicationsForMimeTy
 
 bool MacPlatformIntegration::canShowNotifications() const
 {
-	return true;
+	return (QSysInfo::MacintoshVersion >= QSysInfo::MV_10_8);
 }
 
 }
