diff --git kmail/kmsystemtray.cpp kmail/kmsystemtray.cpp
index a960a79..f3c3fc2 100644
--- kmail/kmsystemtray.cpp
+++ kmail/kmsystemtray.cpp
@@ -105,7 +105,7 @@ KMSystemTray::KMSystemTray(QObject *parent)
 bool KMSystemTray::buildPopupMenu()
 {
     KMMainWidget * mainWidget = kmkernel->getKMMainWidget();
-    if ( !mainWidget ) {
+    if ( !mainWidget || kmkernel->shuttingDown() ) {
         return false;
     }
 
