diff --git akregator/src/actionmanagerimpl.cpp akregator/src/actionmanagerimpl.cpp
index 98c05a1..2c44d1c 100644
--- akregator/src/actionmanagerimpl.cpp
+++ akregator/src/actionmanagerimpl.cpp
@@ -163,7 +163,11 @@ void ActionManagerImpl::setTrayIcon(TrayIcon* trayIcon)
         return;
     else d->trayIcon = trayIcon;
 
+#ifdef Q_OS_MAC
+    KMenu* traypop = trayIcon->contextMenu();
+#else
     QMenu* traypop = trayIcon->contextMenu();
+#endif
 
     if (actionCollection()->action("feed_fetch_all"))
         traypop->addAction(actionCollection()->action("feed_fetch_all"));
diff --git kaddressbook/kaddressbookpart.cpp kaddressbook/kaddressbookpart.cpp
index 0d6b689..e9ce934 100644
--- kaddressbook/kaddressbookpart.cpp
+++ kaddressbook/kaddressbookpart.cpp
@@ -67,6 +67,9 @@ void KAddressBookPart::updateQuickSearchText()
 void KAddressBookPart::initAction()
 {
     KAction *action = new KAction( KIcon( QLatin1String("configure") ), i18n( "&Configure KAddressBook..." ), this );
+#ifdef Q_OS_MAC
+    action->setMenuRole(QAction::PreferencesRole);
+#endif
     actionCollection()->addAction( QLatin1String("kaddressbook_configure"), action );
     connect( action, SIGNAL(triggered(bool)), mMainWidget,
              SLOT(configure()) );
diff --git kjots/kjotspart.cpp kjots/kjotspart.cpp
index 1eaa902..8b2dc88 100644
--- kjots/kjotspart.cpp
+++ kjots/kjotspart.cpp
@@ -80,6 +80,9 @@ KJotsPart::~KJotsPart()
 void KJotsPart::initAction()
 {
   KAction *action = new KAction( KIcon( QLatin1String("configure") ), i18n( "&Configure KJots..." ), this );
+#ifdef Q_OS_MAC
+  action->setMenuRole(QAction::PreferencesRole);
+#endif
   actionCollection()->addAction( QLatin1String("kjots_configure"), action );
   connect( action, SIGNAL(triggered(bool)), mComponent,
            SLOT(configure()) );
diff --git kmail/kmsystemtray.cpp kmail/kmsystemtray.cpp
index a960a79..fd8a206 100644
--- kmail/kmsystemtray.cpp
+++ kmail/kmsystemtray.cpp
@@ -115,7 +115,18 @@ bool KMSystemTray::buildPopupMenu()
 
     contextMenu()->clear();
 
+#ifndef Q_OS_MAC
     contextMenu()->addTitle(qApp->windowIcon(), i18n("KMail"));
+#else
+    { // emulate addTitle by adding an inactive menu item.
+        KAction *act = new KAction( i18n("KMail"), this );
+        act->setEnabled(false);
+        act->setIcon(qApp->windowIcon());
+        act->setIconVisibleInMenu(true);
+        contextMenu()->addAction(act);
+        contextMenu()->addSeparator();
+    }
+#endif
     QAction * action;
     if ( ( action = mainWidget->action(QLatin1String("check_mail")) ) )
         contextMenu()->addAction( action );
diff --git kmail/messageactions.cpp kmail/messageactions.cpp
index 0ae3ec2..bec2e02 100644
--- kmail/messageactions.cpp
+++ kmail/messageactions.cpp
@@ -357,16 +357,16 @@ void MessageActions::updateMailingListActions( const Akonadi::Item& messageItem
         mMailingListActionMenu->menu()->clear();
         qDeleteAll(mMailListActionList);
         mMailListActionList.clear();
-#ifndef Q_WS_MACX
+// #ifndef Q_WS_MACX
         if ( !listId.isEmpty() )
             mMailingListActionMenu->menu()->addTitle( listId );
-#else
-//      // emulate addTitle by adding an inactive menu item.
-        KAction *act = new KAction( listId, this );
-        act->setEnabled(false);
-        mMailingListActionMenu->addAction( act );
-        mMailingListActionMenu->addSeparator();
-#endif
+// #else
+// //      // emulate addTitle by adding an inactive menu item.
+//         KAction *act = new KAction( listId, this );
+//         act->setEnabled(false);
+//         mMailingListActionMenu->addAction( act );
+//         mMailingListActionMenu->addSeparator();
+// #endif
 
         if ( mailList.features() & MessageCore::MailingList::ArchivedAt )
             // IDEA: this may be something you want to copy - "Copy in submenu"?
diff --git korganizer/actionmanager.cpp korganizer/actionmanager.cpp
index 6e8e610..2fbe7c6 100644
--- korganizer/actionmanager.cpp
+++ korganizer/actionmanager.cpp
@@ -708,6 +708,9 @@ void ActionManager::initActions()
 
   if ( mIsPart ) {
     action = new KAction( KIcon( QLatin1String("configure") ), i18n( "&Configure KOrganizer..." ), this );
+#ifdef Q_OS_MAC
+    action->setMenuRole(QAction::PreferencesRole);
+#endif
     mACollection->addAction( QLatin1String("korganizer_configure"), action );
     connect( action, SIGNAL(triggered(bool)), mCalendarView,
              SLOT(edit_options()) );
diff --git messagelist/core/widgetbase.cpp messagelist/core/widgetbase.cpp
index a3c4ea9..2041132 100644
--- messagelist/core/widgetbase.cpp
+++ messagelist/core/widgetbase.cpp
@@ -381,18 +381,18 @@ void Widget::themeMenuAboutToShow()
 
 static inline void addMenuTitle(KMenu *menu, const char *titleText, QObject *parent, bool separatorAbove=false)
 {
-#ifndef Q_WS_MACX
+// #ifndef Q_WS_MACX
     menu->addTitle( i18n( titleText ) );
-#else
-    //     // emulate addTitle by adding an inactive menu item.
-    KAction *act = new KAction( i18n(titleText), parent );
-    act->setEnabled(false);
-    if( separatorAbove ){
-        menu->addSeparator();
-    }
-    menu->addAction( act );
-    menu->addSeparator();
-#endif
+// #else
+//     //     // emulate addTitle by adding an inactive menu item.
+//     KAction *act = new KAction( i18n(titleText), parent );
+//     act->setEnabled(false);
+//     if( separatorAbove ){
+//         menu->addSeparator();
+//     }
+//     menu->addAction( act );
+//     menu->addSeparator();
+// #endif
 }
 
 void Widget::themeMenuAboutToShow(KMenu *menu)
