--- kleopatra/CMakeLists.txt	2014-11-13 23:13:07.000000000 +0100
+++ kleopatra/CMakeLists.txt	2015-08-19 23:32:12.000000000 +0200
@@ -71,7 +71,7 @@
 
 add_subdirectory(kgpgconf)
 add_subdirectory(kwatchgnupg)
-add_subdirectory(tests)
+#add_subdirectory(tests)
 
 if(WIN32)
   set(_kleopatra_extra_uiserver_SRCS uiserver/uiserver_win.cpp)
--- messagecomposer//CMakeLists.txt	2015-06-10 13:51:56.000000000 +0200
+++ messagecomposer//CMakeLists.txt	2015-08-19 23:34:49.000000000 +0200
@@ -17,7 +17,7 @@
 add_definitions( -DQT_NO_CAST_FROM_ASCII )
 add_definitions( -DQT_NO_CAST_TO_ASCII )
 
-add_subdirectory( tests )
+#add_subdirectory( tests )
 include_directories(${Grantlee_INCLUDE_DIRS})
 
 
