diff --git kalarm/CMakeLists.txt kalarm/CMakeLists.txt
index 717357d..7665171 100644
--- kalarm/CMakeLists.txt
+++ kalarm/CMakeLists.txt
@@ -198,7 +198,7 @@ endif()
 kde4_add_kcfg_files(kalarm_bin_SRCS GENERATE_MOC kalarmconfig.kcfgc)
 
 #if (UNIX)
-kde4_add_app_icon(kalarm_bin_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/appicons/ox*-app-kalarm.png")
+kde4_add_app_icon(kalarm_bin_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/appicons/hi*-app-kalarm.png")
 kde4_add_executable(kalarm_bin ${kalarm_bin_SRCS})
 if(KALARM_USE_AKONADI)
     add_dependencies(kalarm_bin kalarm_resource_xml kalarmdir_resource_xml)
diff --git kalarm/autostart/CMakeLists.txt kalarm/autostart/CMakeLists.txt
index 2c562d3..ae20add 100644
--- kalarm/autostart/CMakeLists.txt
+++ kalarm/autostart/CMakeLists.txt
@@ -6,7 +6,7 @@ set(kalarmautostart_SRCS
 	autostart.cpp 
 )
 
-kde4_add_app_icon(kalarmautostart_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/../appicons/ox*-app-kalarm.png")
+kde4_add_app_icon(kalarmautostart_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/../appicons/hi*-app-kalarm.png")
 
 kde4_add_executable(kalarmautostart ${kalarmautostart_SRCS})
 
diff --git ktnef/CMakeLists.txt ktnef/CMakeLists.txt
index aeacc5c..231062a 100644
--- ktnef/CMakeLists.txt
+++ ktnef/CMakeLists.txt
@@ -18,7 +18,7 @@ set(ktnef_SRCS
   qwmf.cpp
 )
 
-kde4_add_app_icon(ktnef_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/pics/hi*-app-ktnef.png")
+kde4_add_app_icon(ktnef_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/pics/*-app-ktnef.png")
 
 kde4_add_ui_files(ktnef_SRCS ui/attachpropertywidgetbase.ui)
 
diff --git korgac/CMakeLists.txt korgac/CMakeLists.txt
index 6ba0331..2f72668 100644
--- korgac/CMakeLists.txt
+++ korgac/CMakeLists.txt
@@ -53,7 +53,7 @@ if(RUNTIME_PLUGINS_STATIC)
   add_definitions(-DSERIALIZER_PLUGIN_STATIC)
 endif()
 
-kde4_add_app_icon(korgac_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/pixmap/hi*-apps-korganizer.png")
+kde4_add_app_icon(korgac_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/pixmaps/hi*-apps-korgac.png")
 
 if(KORGAC_AKONADI_AGENT)
   kde4_add_plugin(korgac ${korgac_SRCS})
