diff --git kmail/CMakeLists.txt kmail/CMakeLists.txt
index 621180c..781c8aa 100644
--- kmail/CMakeLists.txt
+++ kmail/CMakeLists.txt
@@ -376,6 +376,12 @@ if (KDEPIM_BUILD_DESKTOP)
     kdepim
     ${QT_QTSCRIPT_LIBRARY}
   )
+  if (APPLE)
+    set_target_properties(kmail PROPERTIES MACOSX_BUNDLE_INFO_PLIST ${CMAKE_SOURCE_DIR}/Info2.plist.template)
+    set_target_properties(kmail PROPERTIES MACOSX_BUNDLE_GUI_IDENTIFIER "org.kde.kmail4")
+    set_target_properties(kmail PROPERTIES MACOSX_BUNDLE_BUNDLE_NAME "KMail 4")
+  endif (APPLE)
+
 
 
   add_subdirectory(mailmerge/tests/)
diff --git kontact/src/CMakeLists.txt kontact/src/CMakeLists.txt
index 1a9902a..9b2b723 100644
--- kontact/src/CMakeLists.txt
+++ kontact/src/CMakeLists.txt
@@ -58,6 +58,12 @@ target_link_libraries(kontact_bin
   kontactprivate
 )
 
+if (APPLE)
+    set_target_properties(kontact_bin PROPERTIES MACOSX_BUNDLE_INFO_PLIST ${CMAKE_SOURCE_DIR}/Info2.plist.template)
+    set_target_properties(kontact_bin PROPERTIES MACOSX_BUNDLE_GUI_IDENTIFIER "org.kde.kontact4")
+    set_target_properties(kontact_bin PROPERTIES MACOSX_BUNDLE_BUNDLE_NAME "Kontact 4")
+endif (APPLE)
+
 install(TARGETS kontact_bin ${INSTALL_TARGETS_DEFAULT_ARGS})
 
 ########### next target ###############
diff --git Info2.plist.template Info2.plist.template
new file mode 100644
index 0000000..db30210
--- /dev/null
+++ Info2.plist.template
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple Computer//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+    <key>CFBundleDevelopmentRegion</key>
+    <string>English</string>
+    <key>CFBundleExecutable</key>
+    <string>${MACOSX_BUNDLE_EXECUTABLE_NAME}</string>
+    <key>CFBundleGetInfoString</key>
+    <string>${MACOSX_BUNDLE_INFO_STRING}</string>
+    <key>CFBundleIconFile</key>
+    <string>${MACOSX_BUNDLE_ICON_FILE}</string>
+    <key>CFBundleIdentifier</key>
+    <string>${MACOSX_BUNDLE_GUI_IDENTIFIER}</string>
+    <key>CFBundleInfoDictionaryVersion</key>
+    <string>6.0</string>
+    <key>CFBundleLongVersionString</key>
+    <string>${MACOSX_BUNDLE_LONG_VERSION_STRING}</string>
+    <key>CFBundleName</key>
+    <string>${MACOSX_BUNDLE_BUNDLE_NAME}</string>
+    <key>CFBundlePackageType</key>
+    <string>APPL</string>
+    <key>CFBundleShortVersionString</key>
+    <string>${MACOSX_BUNDLE_SHORT_VERSION_STRING}</string>
+    <key>CFBundleVersion</key>
+    <string>${MACOSX_BUNDLE_BUNDLE_VERSION}</string>
+    <key>CSResourcesFileMapped</key>
+    <true/>
+    <key>LSRequiresCarbon</key>
+    <true/>
+    <key>NSHumanReadableCopyright</key>
+    <string>${MACOSX_BUNDLE_COPYRIGHT}</string>
+</dict>
+</plist>
