diff --git src/assistant/assistant/helpviewer.h src/assistant/assistant/helpviewer.h
index 57dc1d2..9d7f88e 100644
--- src/assistant/assistant/helpviewer.h
+++ src/assistant/assistant/helpviewer.h
@@ -130,7 +130,9 @@ protected:
     void wheelEvent(QWheelEvent *event) override;
     void mousePressEvent(QMouseEvent *event) override;
     void mouseReleaseEvent(QMouseEvent *event) override;
+#if !defined(BROWSER_QTWEBKIT)
     void resizeEvent(QResizeEvent *e) override;
+#endif
 
 private slots:
     void actionChanged();
