diff --git b/clang-tools-extra/clangd/xpc/test-client/orig.ClangdXPCTestClient.cpp a/clang-tools-extra/clangd/xpc/test-client/ClangdXPCTestClient.cpp
index 908544f..c1cbade 100644
--- b/clang-tools-extra/clangd/xpc/test-client/orig.ClangdXPCTestClient.cpp
+++ a/clang-tools-extra/clangd/xpc/test-client/ClangdXPCTestClient.cpp
@@ -48,6 +48,9 @@ static void dumpXPCObject(xpc_object_t Object, llvm::raw_ostream &OS) {
 int main(int argc, char *argv[]) {
   // Open the ClangdXPC dylib in the framework.
   std::string LibPath = getLibraryPath();
+#ifndef RTLD_FIRST
+#define RTLD_FIRST 0
+#endif
   void *dlHandle = dlopen(LibPath.c_str(), RTLD_LOCAL | RTLD_FIRST);
   if (!dlHandle) {
     llvm::errs() << "Failed to load framework from \'" << LibPath << "\'\n";
