diff --git b/compiler-rt/orig.CMakeLists.txt a/compiler-rt/CMakeLists.txt
index b44ad2c..a210695 100644
--- b/compiler-rt/orig.CMakeLists.txt
+++ a/compiler-rt/CMakeLists.txt
@@ -181,7 +181,7 @@ pythonize_bool(SANITIZER_CAN_USE_CXXABI)
 
 macro(handle_default_cxx_lib var)
   # Specifying -stdlib= in CMAKE_CXX_FLAGS overrides the defaults.
-  if (CMAKE_CXX_FLAGS MATCHES "-stdlib=([a-zA-Z+]*)")
+  if (CMAKE_CXX_FLAGS MATCHES "-stdlib=([a-zA-Z+_]*)")
     set(${var}_LIBNAME "${CMAKE_MATCH_1}")
     set(${var}_SYSTEM 1)
   elseif (${var} STREQUAL "default")
@@ -453,8 +453,9 @@ append_list_if(COMPILER_RT_HAS_NOSTDINCXX_FLAG -nostdinc++ SANITIZER_COMMON_CFLA
 append_list_if(COMPILER_RT_HAS_NOSTDLIBXX_FLAG -nostdlib++ SANITIZER_COMMON_LINK_FLAGS)
 
 # Remove -stdlib= which is unused when passing -nostdinc++...
-string(REGEX MATCHALL "-stdlib=[a-zA-Z+]*" stdlib_flag "${CMAKE_CXX_FLAGS}")
-string(REGEX REPLACE "-stdlib=[a-zA-Z+]*" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
+string(REGEX MATCHALL "-stdlib=[a-zA-Z+_]*" stdlib_flag "${CMAKE_CXX_FLAGS}")
+string(REGEX REPLACE "-stdlib=[a-zA-Z+_]*" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
+message(WARNING "stdlib_flag=${stdlib_flag} pruned from CMAKE_CXX_FLAGS->\"${CMAKE_CXX_FLAGS}\"")
 
 # ...we need it to build some runtimes and tests so readd it where appropriate.
 list(APPEND COMPILER_RT_COMMON_CFLAGS ${stdlib_flag})
diff --git b/libcxxabi/orig.CMakeLists.txt a/libcxxabi/CMakeLists.txt
index c49be65..898d416 100644
--- b/libcxxabi/orig.CMakeLists.txt
+++ a/libcxxabi/CMakeLists.txt
@@ -272,6 +272,8 @@ if (LIBCXXABI_HAS_NOSTDINCXX_FLAG)
   # See: https://gitlab.kitware.com/cmake/cmake/issues/19227
   set(CMAKE_CXX_IMPLICIT_INCLUDE_DIRECTORIES "")
   # Remove -stdlib flags to prevent them from causing an unused flag warning.
+  string(REPLACE "-stdlib=macports-libstdc++" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
+  string(REPLACE "-stdlib=libstdc++_macports" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
   string(REPLACE "--stdlib=libc++" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
   string(REPLACE "--stdlib=libstdc++" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
   string(REPLACE "-stdlib=libc++" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
