diff --git a/src/main.cpp b/src/main.cpp
index 2e4c31f8df32f33b6f6ec93024d54affda0a4efd..dbf00438447dc320ec8694a5c3deae4d43535a58 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -149,7 +149,7 @@ int main(int argc, char* argv[])
     const QCommandLineOption keepRunningOption(QStringLiteral("keeprunning"), i18nc("@info:shell","Keep the program running and generate "
                                                     "the backtrace at startup"));
     const QCommandLineOption threadOption(QStringLiteral("thread"), i18nc("@info:shell","The <thread id> of the failing thread"), QStringLiteral("threadid"));
-    const QCommandLineOption dialogOption(QStringLiteral("dialog"), i18nc("@info:shell","Do not show a notification but launch the debug dialog directly"));
+//     const QCommandLineOption dialogOption(QStringLiteral("dialog"), i18nc("@info:shell","Do not show a notification but launch the debug dialog directly"));
 
     parser.addOptions({
         signalOption,
@@ -165,7 +165,7 @@ int main(int argc, char* argv[])
         restartedOption,
         keepRunningOption,
         threadOption,
-        dialogOption
+//         dialogOption
     });
 
     // Add all unknown options but make sure to print a warning.
@@ -201,7 +201,7 @@ int main(int argc, char* argv[])
     DrKonqi::setRestarted(parser.isSet(restartedOption));
     DrKonqi::setKeepRunning(parser.isSet(keepRunningOption));
     DrKonqi::setThread(parser.value(threadOption).toInt());
-    auto forceDialog = parser.isSet(dialogOption);
+    auto forceDialog = true /*parser.isSet(dialogOption)*/;
 
 #if HAVE_X11
     const QString startupId = parser.value(startupIdOption);
