diff --git a/CMakeLists.txt b/CMakeLists.txt
index 364f90acd69abb6ae0976621b803f49f0c40a62c..d94310e1f11cf70fed1dd6b266da2506b2dcaba4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -4,8 +4,8 @@ project(drkonqi)
 set(PROJECT_VERSION "5.14.80")
 set(PROJECT_VERSION_MAJOR 5)
 
-set(QT_MIN_VERSION "5.11.0")
-set(KF5_MIN_VERSION "5.50.0")
+set(QT_MIN_VERSION "5.9.0")
+set(KF5_MIN_VERSION "5.47.0")
 
 find_package(ECM ${KF5_MIN_VERSION} REQUIRED NO_MODULE)
 

commit bb1e983c2547ccab05726a8097b82599df3e3f94
Author: Laurent Montel <montel@kde.org>
Date:   Thu Oct 25 08:07:14 2018 +0200

    We depend against 5.11

diff --git b/src/bugzillaintegration/bugzillalib.cpp a/src/bugzillaintegration/bugzillalib.cpp
index 7d7b576..770ffac 100644
--- b/src/bugzillaintegration/bugzillalib.cpp
+++ a/src/bugzillaintegration/bugzillalib.cpp
@@ -66,7 +66,11 @@ BugzillaManager::BugzillaManager(const QString &bugTrackerUrl, QObject *parent)
     m_xmlRpcClient->setUserAgent(QStringLiteral("DrKonqi"));
     // Allow constructors for ReportInterface and assistant dialogs to finish.
     // We do not want them to be racing the remote Bugzilla database.
+#if QT_VERSION >= QT_VERSION_CHECK(5, 10, 0)
     QMetaObject::invokeMethod (this, &BugzillaManager::lookupVersion, Qt::QueuedConnection);
+#else
+    QMetaObject::invokeMethod (this, "lookupVersion", Qt::QueuedConnection);
+#endif
 }
 
 // BEGIN Checks of Bugzilla software versions.
