diff --git a/src/bugzillaintegration/bugzillalib.cpp b/src/bugzillaintegration/bugzillalib.cpp
index 77aebd2a19dab923948f867f5d04ee1f2bc0c5c3..6a1da8026d15852709a1b31ca75c52c92ee859a1 100644
--- a/src/bugzillaintegration/bugzillalib.cpp
+++ b/src/bugzillaintegration/bugzillalib.cpp
@@ -37,6 +38,7 @@ static const char showBugUrl[] = "show_bug.cgi?id=%1";
 // static) and is slightly synchronizing across threads WRT the filter hash.
 struct QMessageFilterContainer {
     QMessageFilterContainer();
+    ~QMessageFilterContainer();
     void insert(const QString &needle, const QString &replace);
     void clear();
 
@@ -62,6 +64,13 @@ QMessageFilterContainer::QMessageFilterContainer()
 });
 }
 
+QMessageFilterContainer::~QMessageFilterContainer()
+{
+    // restore the default message handler when we're about to quit, in order
+    // to prevent crashing if Qt logs a message during the global destruction phase.
+    qInstallMessageHandler(nullptr);
+}
+
 void QMessageFilterContainer::insert(const QString &needle, const QString &replace)
 {
     if (needle.isEmpty()) {
