diff --git a/CMakeLists.txt b/CMakeLists.txt
index 364f90acd69abb6ae0976621b803f49f0c40a62c..d94310e1f11cf70fed1dd6b266da2506b2dcaba4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -4,8 +4,8 @@ project(drkonqi)
 set(PROJECT_VERSION "5.14.80")
 set(PROJECT_VERSION_MAJOR 5)
 
-set(QT_MIN_VERSION "5.11.0")
-set(KF5_MIN_VERSION "5.50.0")
+set(QT_MIN_VERSION "5.9.0")
+set(KF5_MIN_VERSION "5.47.0")
 
 find_package(ECM ${KF5_MIN_VERSION} REQUIRED NO_MODULE)
 

