diff --git CMakeLists.txt CMakeLists.txt
index ef38ad776863b55863c03af511f5e5791b0cfcc9..2648a62e3089b6a49efabb5ecca62d5d95c65d92 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -29,7 +29,9 @@ include(FeatureSummary)
 find_package(Qt5 ${QT_MIN_VERSION} CONFIG REQUIRED COMPONENTS Core Widgets Test DBus Concurrent)
 
 find_package(KF5 ${KF5_MIN_VERSION} REQUIRED COMPONENTS I18n CoreAddons Service ConfigWidgets JobWidgets KIO Crash Completion XmlRpcClient WidgetsAddons Wallet Notifications IdleTime)
-
+if(APPLE)
+    find_package(KF5 ${KF5_MIN_VERSION} REQUIRED COMPONENTS WindowSystem)
+endif()
 find_package(Qt5X11Extras ${QT_MIN_VERSION} CONFIG)
 set_package_properties(Qt5X11Extras PROPERTIES TYPE RECOMMENDED PURPOSE "Recommended for better integration on X11.")
 
diff --git src/CMakeLists.txt src/CMakeLists.txt
index 30c6e587a4d7772817a6bae93449c5b5724e5f54..8bc7e9f11ce88b3ea65a486130fa379a1e61d8c3 100644
--- src/CMakeLists.txt
+++ src/CMakeLists.txt
@@ -107,6 +107,11 @@ if (${Qt5X11Extras_FOUND})
         Qt5::X11Extras
     )
 endif()
+if (APPLE)
+    target_link_libraries(drkonqi
+        KF5::WindowSystem
+    )
+endif()
 
 install(TARGETS drkonqi DESTINATION ${KDE_INSTALL_LIBEXECDIR})
 install(PROGRAMS org.kde.drkonqi.desktop DESTINATION ${XDG_APPS_INSTALL_DIR})
diff --git src/aboutbugreportingdialog.cpp src/aboutbugreportingdialog.cpp
index 625f512b02420d0918fa9b96b071406e27092228..baea8ab8c9762d2feb5ad19a2abfe9d303da6aed 100644
--- src/aboutbugreportingdialog.cpp
+++ src/aboutbugreportingdialog.cpp
@@ -36,7 +36,7 @@ AboutBugReportingDialog::AboutBugReportingDialog(QWidget * parent):
 {
     setAttribute(Qt::WA_DeleteOnClose, true);
 
-    setWindowIcon(QIcon::fromTheme(QStringLiteral("help-hint")));
+    setWindowIcon(QIcon::fromTheme(QStringLiteral("help-hint"), windowIcon()));
     setWindowTitle(i18nc("@title title of the dialog", "About Bug Reporting - Help"));
 
     QVBoxLayout* layout = new QVBoxLayout(this);
diff --git src/bugzillaintegration/reportassistantdialog.cpp src/bugzillaintegration/reportassistantdialog.cpp
index 8ef91b111608b8fc345ea0a80596ec03851a3e8e..de5174b8b7574ec4fe22924b2c7343163ef0271f 100644
--- src/bugzillaintegration/reportassistantdialog.cpp
+++ src/bugzillaintegration/reportassistantdialog.cpp
@@ -50,7 +50,7 @@ ReportAssistantDialog::ReportAssistantDialog(QWidget * parent) :
 
     //Set window properties
     setWindowTitle(i18nc("@title:window","Crash Reporting Assistant"));
-    setWindowIcon(QIcon::fromTheme(QStringLiteral("tools-report-bug")));
+    setWindowIcon(QIcon::fromTheme(QStringLiteral("tools-report-bug"), windowIcon()));
 
     connect(this, &ReportAssistantDialog::currentPageChanged, this, &ReportAssistantDialog::currentPageChanged_slot);
     connect(button(QDialogButtonBox::Help), &QPushButton::clicked, this, &ReportAssistantDialog::showHelp);
diff --git src/drkonqidialog.cpp src/drkonqidialog.cpp
index 2290d661713c87c0ceaa8704ecc4a984b5cc1bc6..fda7a93d528312414558e0b8b5caf2d6ea50e735 100644
--- src/drkonqidialog.cpp
+++ src/drkonqidialog.cpp
@@ -53,7 +53,7 @@ DrKonqiDialog::DrKonqiDialog(QWidget * parent) :
 
     //Setting dialog title and icon
     setWindowTitle(DrKonqi::crashedApplication()->name());
-    setWindowIcon(QIcon::fromTheme(QStringLiteral("tools-report-bug")));
+    setWindowIcon(QIcon::fromTheme(QStringLiteral("tools-report-bug"), windowIcon()));
 
     QVBoxLayout* l = new QVBoxLayout(this);
     m_tabWidget = new QTabWidget(this);
diff --git src/main.cpp src/main.cpp
index 2426c1a5f761e607d8cf5e0286c71bd6b7772f8c..cd1853a1b033785ad09e27f4b230df2d6077f888 100644
--- src/main.cpp
+++ src/main.cpp
@@ -36,6 +36,9 @@
 #if HAVE_X11
 #include <QX11Info>
 #endif
+#ifdef Q_OS_MACOS
+#include <KWindowSystem>
+#endif
 
 #include "drkonqi.h"
 #include "drkonqidialog.h"
@@ -79,7 +82,7 @@ int main(int argc, char* argv[])
                          QStringLiteral("gkiagia@users.sourceforge.net"));
     aboutData.addAuthor(i18nc("@info:credit","A. L. Spehr"), QString(),
                          QStringLiteral("spehr@kde.org"));
-    qa.setWindowIcon(QIcon::fromTheme(QStringLiteral("tools-report-bug")));
+    qa.setWindowIcon(QIcon::fromTheme(QStringLiteral("tools-report-bug"), qa.windowIcon()));
     qa.setDesktopFileName(QStringLiteral("org.kde.drkonqi"));
 
     QCommandLineParser parser;
@@ -150,6 +153,9 @@ int main(int argc, char* argv[])
         DrKonqiDialog *w = new DrKonqiDialog();
         QObject::connect(w, &DrKonqiDialog::rejected, &qa, &QApplication::quit);
         w->show();
+#ifdef Q_OS_MACOS
+        KWindowSystem::forceActiveWindow(w->winId());
+#endif
     };
 
     bool restarted = parser.isSet(restartedOption);
