diff --git a/karbon/CMakeLists.txt b/karbon/CMakeLists.txt
index ba775ad..ee332fd 100644
--- a/karbon/CMakeLists.txt
+++ b/karbon/CMakeLists.txt
@@ -51,7 +51,7 @@ file(GLOB karbon_icons "${CMAKE_CURRENT_SOURCE_DIR}/pics/action/*-actions-*.png"
 ecm_install_icons(ICONS ${karbon_icons} DESTINATION ${DATA_INSTALL_DIR}/karbon/icons)
 
 file(GLOB karbon_app_icons "${CMAKE_CURRENT_SOURCE_DIR}/pics/app/*-apps-calligrakarbon.png")
-ecm_add_app_icon(karbon_KDEINIT_SRCS ICONS ${karbon_app_icons})
+ecm_add_app_icon(karbon_KDEINIT_SRCS ICONS ${CMAKE_CURRENT_SOURCE_DIR}/pics/app/sc-apps-calligrakarbon.svgz ${karbon_app_icons})
 ecm_install_icons(ICONS
     ${karbon_app_icons}
     "${CMAKE_CURRENT_SOURCE_DIR}/pics/app/sc-apps-calligrakarbon.svgz"
@@ -64,7 +64,6 @@ if (APPLE)
    set_target_properties(karbon PROPERTIES MACOSX_BUNDLE_INFO_PLIST ${CMAKE_CURRENT_SOURCE_DIR}/Info.plist.template)
    set_target_properties(karbon PROPERTIES MACOSX_BUNDLE_GUI_IDENTIFIER "org.calligra.karbon")
    set_target_properties(karbon PROPERTIES MACOSX_BUNDLE_BUNDLE_NAME "Karbon")
-   install( FILES ${CMAKE_CURRENT_BINARY_DIR}/karbon_KDEINIT_SRCS.icns DESTINATION ${BUNDLE_INSTALL_DIR}/karbon.app/Contents/Resources)
 endif()
 
 target_link_libraries(kdeinit_karbon karbonui)
diff --git a/words/app/CMakeLists.txt b/words/app/CMakeLists.txt
index 8cf0157..770134e 100644
--- a/words/app/CMakeLists.txt
+++ b/words/app/CMakeLists.txt
@@ -8,7 +8,7 @@ include_directories(
 set(words_KDEINIT_SRCS main.cpp )
 
 file(GLOB ICONS_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/../pics/*-apps-calligrawords.png")
-ecm_add_app_icon(words_KDEINIT_SRCS ICONS ${ICONS_SRCS})
+ecm_add_app_icon(words_KDEINIT_SRCS ICONS ${CMAKE_CURRENT_SOURCE_DIR}/../pics/sc-apps-calligrawords.svgz ${ICONS_SRCS})
 
 kf5_add_kdeinit_executable( calligrawords ${words_KDEINIT_SRCS})
 
@@ -16,7 +16,6 @@ if (APPLE)
    set_target_properties(calligrawords PROPERTIES MACOSX_BUNDLE_INFO_PLIST ${CMAKE_CURRENT_SOURCE_DIR}/Info.plist.template)
    set_target_properties(calligrawords PROPERTIES MACOSX_BUNDLE_GUI_IDENTIFIER "org.kde.calligrawords")
    set_target_properties(calligrawords PROPERTIES MACOSX_BUNDLE_BUNDLE_NAME "Words")
-   install( FILES ${CMAKE_CURRENT_BINARY_DIR}/words_KDEINIT_SRCS.icns DESTINATION ${BUNDLE_INSTALL_DIR}/calligrawords.app/Contents/Resources)
 endif ()
 
 target_link_libraries(kdeinit_calligrawords   komain )
diff --git a/braindump/src/CMakeLists.txt b/braindump/src/CMakeLists.txt
index 95e0772..3bae7d5 100644
--- a/braindump/src/CMakeLists.txt
+++ b/braindump/src/CMakeLists.txt
@@ -43,7 +43,7 @@ ki18n_wrap_ui(braindump_SRCS
 )
 
 file(GLOB ICONS_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/../data/icons/*-apps-braindump.png")
-ecm_add_app_icon( braindump_SRCS ICONS ${ICONS_SRCS})
+ecm_add_app_icon( braindump_SRCS ICONS ${CMAKE_CURRENT_SOURCE_DIR}/../data/icons/sc-apps-braindump.svgz ${ICONS_SRCS})
 
 add_executable(braindump ${braindump_SRCS})
 
diff --git a/flow/part/CMakeLists.txt b/flow/part/CMakeLists.txt
index f5d0893..8fe8f62 100644
--- a/flow/part/CMakeLists.txt
+++ b/flow/part/CMakeLists.txt
@@ -36,7 +36,7 @@ install(TARGETS flowpart DESTINATION ${PLUGIN_INSTALL_DIR}/calligra/parts)
 set(flow_KDEINIT_SRCS main.cpp )
 
 file(GLOB ICONS_SRCS "../pics/hi*-app-calligraflow.png")
-ecm_add_app_icon(flow_KDEINIT_SRCS ICONS ${ICONS_SRCS})
+ecm_add_app_icon(flow_KDEINIT_SRCS ICONS ../pics/hisc-app-calligraflow.svgz ${ICONS_SRCS})
 
 kf5_add_kdeinit_executable(calligraflow ${flow_KDEINIT_SRCS})
 
@@ -55,6 +55,3 @@ install( FILES org.kde.calligraflow.appdata.xml DESTINATION ${KDE_INSTALL_METAIN
 ### GUI files ###
 install( FILES flow.rc DESTINATION ${KXMLGUI_INSTALL_DIR}/flow)
 install( FILES flowrc DESTINATION ${CONFIG_INSTALL_DIR} )
-if(APPLE)
-   install( FILES ${CMAKE_CURRENT_BINARY_DIR}/flow_KDEINIT_SRCS.icns DESTINATION ${BUNDLE_INSTALL_DIR}/calligraflow.app/Contents/Resources)
-endif()
diff --git a/gemini/CMakeLists.txt b/gemini/CMakeLists.txt
index c8713d6..f9b2712 100644
--- a/gemini/CMakeLists.txt
+++ b/gemini/CMakeLists.txt
@@ -58,7 +58,7 @@ if (WIN32)
     set(calligragemini_SRCS ${calligragemini_SRCS} calligrageminiwin.rc)
 else ()
     file(GLOB ICONS_SRCS "pics/*-apps-calligragemini.png")
-    ecm_add_app_icon(calligragemini_SRCS ICONS ${ICONS_SRCS})
+    ecm_add_app_icon(calligragemini_SRCS ICONS pics/sc-apps-calligragemini.svgz ${ICONS_SRCS})
 endif ()
 
 # Thumbnail helper application
@@ -93,9 +93,6 @@ install(FILES     calligragemini.qml                   DESTINATION ${DATA_INSTAL
 #install(FILES     calligragemini.rc                    DESTINATION ${DATA_INSTALL_DIR}/calligragemini)
 install(DIRECTORY qml                                       DESTINATION ${DATA_INSTALL_DIR}/calligragemini)
 install(DIRECTORY themes                                    DESTINATION ${DATA_INSTALL_DIR}/calligragemini)
-if(APPLE)
-   install( FILES ${CMAKE_CURRENT_BINARY_DIR}/calligragemini_SRCS.icns DESTINATION ${BUNDLE_INSTALL_DIR}/calligragemini.app/Contents/Resources)
-endif()
 
 add_subdirectory(pics)
 add_subdirectory(cloud)
diff --git a/plan/CMakeLists.txt b/plan/CMakeLists.txt
index 25427da..b4136b0 100644
--- a/plan/CMakeLists.txt
+++ b/plan/CMakeLists.txt
@@ -132,7 +132,7 @@ endif()
 set(calligraplan_KDEINIT_SRCS main.cpp )
 
 file(GLOB ICONS_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/pics/*-apps-calligraplan.png")
-ecm_add_app_icon(calligraplan_KDEINIT_SRCS ICONS ${ICONS_SRCS})
+ecm_add_app_icon(calligraplan_KDEINIT_SRCS ICONS ${CMAKE_CURRENT_SOURCE_DIR}/pics/sc-apps-calligraplan.svgz ${ICONS_SRCS})
 
 kf5_add_kdeinit_executable( calligraplan ${calligraplan_KDEINIT_SRCS})
 
@@ -140,7 +140,6 @@ if (APPLE)
    set_target_properties(calligraplan PROPERTIES MACOSX_BUNDLE_INFO_PLIST ${CMAKE_CURRENT_SOURCE_DIR}/Info.plist.template)
    set_target_properties(calligraplan PROPERTIES MACOSX_BUNDLE_GUI_IDENTIFIER "org.calligra.plan")
    set_target_properties(calligraplan PROPERTIES MACOSX_BUNDLE_BUNDLE_NAME "Plan")
-   install( FILES ${CMAKE_CURRENT_BINARY_DIR}/calligraplan_KDEINIT_SRCS.icns DESTINATION ${BUNDLE_INSTALL_DIR}/calligraplan.app/Contents/Resources)
 endif ()
 
 target_link_libraries(kdeinit_calligraplan komain)
diff --git a/sheets/CMakeLists.txt b/sheets/CMakeLists.txt
index 0256de5..186ab40 100644
--- a/sheets/CMakeLists.txt
+++ b/sheets/CMakeLists.txt
@@ -460,7 +460,7 @@ if (SHOULD_BUILD_APP_SHEETS)
 set (calligrasheets_KDEINIT_SRCS part/Main.cpp )
 
 file(GLOB ICONS_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/data/pics/*-apps-calligrasheets.png")
-ecm_add_app_icon(calligrasheets_KDEINIT_SRCS ICONS ${ICONS_SRCS})
+ecm_add_app_icon(calligrasheets_KDEINIT_SRCS ICONS ${CMAKE_CURRENT_SOURCE_DIR}/data/pics/sc-apps-calligrasheets.svgz ${ICONS_SRCS})
 
 kf5_add_kdeinit_executable( calligrasheets ${calligrasheets_KDEINIT_SRCS})
 
@@ -468,7 +468,6 @@ if (APPLE)
 set_target_properties(calligrasheets PROPERTIES MACOSX_BUNDLE_INFO_PLIST ${CMAKE_CURRENT_SOURCE_DIR}/Info.plist.template)
 set_target_properties(calligrasheets PROPERTIES MACOSX_BUNDLE_GUI_IDENTIFIER "org.calligra.sheets")
 set_target_properties(calligrasheets PROPERTIES MACOSX_BUNDLE_BUNDLE_NAME "Calligra Sheets 2")
-install( FILES ${CMAKE_CURRENT_BINARY_DIR}/calligrasheets_KDEINIT_SRCS.icns DESTINATION ${BUNDLE_INSTALL_DIR}/calligrasheets.app/Contents/Resources)
 endif ()
 
 target_link_libraries(kdeinit_calligrasheets komain KF5::KDELibs4Support)
diff --git a/stage/app/CMakeLists.txt b/stage/app/CMakeLists.txt
index b5b4464..bb74ef2 100644
--- a/stage/app/CMakeLists.txt
+++ b/stage/app/CMakeLists.txt
@@ -10,7 +10,7 @@ include_directories(
 set(stage_KDEINIT_SRCS main.cpp )
 
 file(GLOB ICONS_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/../pics/*-apps-calligrastage.png")
-ecm_add_app_icon(stage_KDEINIT_SRCS ICONS ${ICONS_SRCS})
+ecm_add_app_icon(stage_KDEINIT_SRCS ICONS ${CMAKE_CURRENT_SOURCE_DIR}/../pics/*-apps-calligrastage.svgz ${ICONS_SRCS})
 
 kf5_add_kdeinit_executable(calligrastage ${stage_KDEINIT_SRCS})
 
@@ -18,7 +18,6 @@ if (APPLE)
    set_target_properties(calligrastage PROPERTIES MACOSX_BUNDLE_INFO_PLIST ${CMAKE_CURRENT_SOURCE_DIR}/Info.plist.template)
    set_target_properties(calligrastage PROPERTIES MACOSX_BUNDLE_GUI_IDENTIFIER "org.kde.calligrastage")
    set_target_properties(calligrastage PROPERTIES MACOSX_BUNDLE_BUNDLE_NAME "Calligra Stage 2")
-   install( FILES ${CMAKE_CURRENT_BINARY_DIR}/stage_KDEINIT_SRCS.icns DESTINATION ${BUNDLE_INSTALL_DIR}/calligrastage.app/Contents/Resources)
 endif ()
 
 target_link_libraries(kdeinit_calligrastage komain)
diff --git a/plan/workpackage/CMakeLists.txt b/plan/workpackage/CMakeLists.txt
index ec50d6d..42aa83f 100644
--- a/plan/workpackage/CMakeLists.txt
+++ b/plan/workpackage/CMakeLists.txt
@@ -80,9 +80,10 @@ install(TARGETS calligraplanworkpart DESTINATION ${PLUGIN_INSTALL_DIR})
 ########### Calligra PlanWork executable ###############
 
 set(calligraplanwork_KDEINIT_SRCS main.cpp commandlineparser.cpp)
+file(GLOB calligraplanwork_icons ${CMAKE_CURRENT_SOURCE_DIR}/pics/*-apps-calligraplanwork.png)
+ecm_add_app_icon(calligraplanwork_KDEINIT_SRCS ICONS ${CMAKE_CURRENT_SOURCE_DIR}/pics/sc-apps-calligraplanwork.svgz ${calligraplanwork_icons})
 
 kf5_add_kdeinit_executable(calligraplanwork ${calligraplanwork_KDEINIT_SRCS})
-
 if (APPLE)
    set_target_properties(calligraplanwork PROPERTIES MACOSX_BUNDLE_INFO_PLIST ${CMAKE_CURRENT_SOURCE_DIR}/Info.plist.template)
    set_target_properties(calligraplanwork PROPERTIES MACOSX_BUNDLE_GUI_IDENTIFIER "org.calligra.plan.work")
diff --git a/libs/main/KoApplication.cpp b/libs/main/KoApplication.cpp
index 7b23f8d..01786f1 100644
--- a/libs/main/KoApplication.cpp
+++ b/libs/main/KoApplication.cpp
@@ -139,7 +139,7 @@ KoApplication::KoApplication(const QByteArray &nativeMimeType,
 
     KAboutData::setApplicationData( aboutData );
 
-    setWindowIcon(QIcon::fromTheme(windowIconName));
+    setWindowIcon(QIcon::fromTheme(windowIconName, windowIcon()));
 
     KoApplication::KoApp = this;
 
diff --git a/plan/workpackage/commandlineparser.cpp b/plan/workpackage/commandlineparser.cpp
index e7bf9a1..432f99a 100644
--- a/plan/workpackage/commandlineparser.cpp
+++ b/plan/workpackage/commandlineparser.cpp
@@ -40,7 +40,7 @@ CommandLineParser::CommandLineParser()
 {
     KAboutData *aboutData = KPlatoWork::newAboutData();
     KAboutData::setApplicationData( *aboutData );
-    qApp->setWindowIcon(QIcon::fromTheme(QStringLiteral("calligraplanwork")));
+    qApp->setWindowIcon(QIcon::fromTheme(QStringLiteral("calligraplanwork"), qApp->windowIcon()));
 
     aboutData->setupCommandLine(&m_commandLineParser);
     m_commandLineParser.addHelpOption();
