diff --git src/kapidox/argparserutils.py src/kapidox/argparserutils.py
index 7d29c57..0115656 100644
--- src/kapidox/argparserutils.py
+++ src/kapidox/argparserutils.py
@@ -74,6 +74,8 @@ def add_output_group(parser):
             help='Generate man page documentation.')
     group.add_argument('--qhp', action='store_true',
             help='Generate Qt Compressed Help documentation.')
+    group.add_argument('--chm', action='store_true', default=False,
+            help='Generate Compressed HTML Help (chm) documentation.')
     group.add_argument('--searchengine', action='store_true',
             help="Enable Doxygen's search engine feature.")
     group.add_argument('--api-searchbox', action='store_true',
@@ -105,6 +107,8 @@ def add_paths_group(parser):
             help='(Path to) the doxygen executable.')
     group.add_argument('--qhelpgenerator', default='qhelpgenerator',
             help='(Path to) the qhelpgenerator executable.')
+    group.add_argument('--chmcompiler', default='',
+            help='(Path to) the HTML help (chm) compiler executable.')
     return group
 
 
diff --git src/kapidox/generator.py src/kapidox/generator.py
index ff8fcac..babf13f 100644
--- src/kapidox/generator.py
+++ src/kapidox/generator.py
@@ -106,9 +106,11 @@ class Context(object):
         'qhp',
         'searchengine',
         'api_searchbox',
+        'chm',
         # Binaries
         'doxygen',
         'qhelpgenerator',
+        'chmcompiler',
     )
 
     def __init__(self, args, **kwargs):
@@ -121,9 +123,11 @@ class Context(object):
         self.qhp = args.qhp
         self.searchengine = args.searchengine
         self.api_searchbox = args.api_searchbox
+        self.chm = args.chm
         # Binaries
         self.doxygen = args.doxygen
         self.qhelpgenerator = args.qhelpgenerator
+        self.chmcompiler = args.chmcompiler
 
         for key in self.__slots__:
             if not hasattr(self, key):
@@ -622,6 +626,13 @@ def generate_apidocs(ctx, tmp_dir, doxyfile_entries=None, keep_temp_dirs=False):
                 GENERATE_MAN=ctx.man_pages,
                 GENERATE_QHP=ctx.qhp)
                 #, SEARCHENGINE=ctx.searchengine)
+        if ctx.chmcompiler != '' and ctx.chm:
+            writer.write_entries(
+                GENERATE_HTMLHELP="YES",
+                GENERATE_CHI="NO",
+                HHC_LOCATION=ctx.chmcompiler,
+                CHM_FILE=ctx.fancyname + ".chm",
+                SEARCHENGINE="NO")
 
         if doxyfile_entries:
             writer.write_entries(**doxyfile_entries)
@@ -812,25 +823,28 @@ def indexer(lib):
     """
 
     doclist = []
-    tree = xmlET.parse(lib.outputdir + '/searchdata.xml')
-    for doc_child in tree.getroot():
-        field = {}
-        for child in doc_child:
-            if child.attrib['name'] == "type":
-                if child.text == 'source':
-                    field = None
-                    break; # We go to next <doc>
-                field['type'] = child.text
-            elif child.attrib['name'] == "name":
-                field['name'] = child.text
-            elif child.attrib['name'] == "url":
-                field['url'] = child.text
-            elif child.attrib['name'] == "keywords":
-                field['keyword'] = child.text
-            elif child.attrib['name'] == "text":
-                field['text'] = "" if child.text is None else child.text
-        if field is not None:
-            doclist.append(field)
+    try:
+        tree = xmlET.parse(lib.outputdir + '/searchdata.xml')
+        for doc_child in tree.getroot():
+            field = {}
+            for child in doc_child:
+                if child.attrib['name'] == "type":
+                    if child.text == 'source':
+                        field = None
+                        break; # We go to next <doc>
+                    field['type'] = child.text
+                elif child.attrib['name'] == "name":
+                    field['name'] = child.text
+                elif child.attrib['name'] == "url":
+                    field['url'] = child.text
+                elif child.attrib['name'] == "keywords":
+                    field['keyword'] = child.text
+                elif child.attrib['name'] == "text":
+                    field['text'] = "" if child.text is None else child.text
+            if field is not None:
+                doclist.append(field)
+    except:
+        pass
 
     indexdic = {
         'name': lib.name,
