diff --git src/CMakeLists.txt src/CMakeLists.txt
index 9ee4491..76a8b7c 100644
--- src/CMakeLists.txt
+++ src/CMakeLists.txt
@@ -526,6 +526,8 @@ endif()
 
 # Build everything except main and inkview.c in a shared library.
 add_library(inkscape_base SHARED ${inkscape_SRC} ${sp_SRC})
+set_target_properties(inkscape_base PROPERTIES
+    INSTALL_NAME_DIR ${CMAKE_INSTALL_PREFIX}/lib/inkscape)
 
 # make executables for inkscape and inkview
 add_executable(inkscape ${main_SRC})
