diff --git CMakeLists.txt CMakeLists.txt
index cddbfbe..9a2fa73 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -42,7 +42,10 @@ set(CMAKE_INCLUDE_CURRENT_DIR TRUE)
 
 if(APPLE)
     SET(CMAKE_MACOSX_RPATH TRUE)
-    SET(CMAKE_INSTALL_RPATH "@loader_path/../lib/inkscape")
+    # amend the rpath
+    SET(CMAKE_INSTALL_RPATH "@loader_path/../lib/inkscape;${CMAKE_INSTALL_RPATH}")
+    # make sure shared libraries will store full paths in their ID
+    SET(CMAKE_BUILD_WITH_INSTALL_NAME_DIR TRUE)
 else()
     SET(CMAKE_INSTALL_RPATH "$ORIGIN/../lib/inkscape")
 endif()
diff --git src/CMakeLists.txt src/CMakeLists.txt
index 9ee4491..c020deb 100644
--- src/CMakeLists.txt
+++ src/CMakeLists.txt
@@ -526,6 +526,9 @@ endif()
 
 # Build everything except main and inkview.c in a shared library.
 add_library(inkscape_base SHARED ${inkscape_SRC} ${sp_SRC})
+# use the "magic" path specifier for this library's ID (Mac specific)
+set_target_properties(inkscape_base PROPERTIES
+    INSTALL_NAME_DIR @rpath)
 
 # make executables for inkscape and inkview
 add_executable(inkscape ${main_SRC})
