diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4e735a1..3bdfac2 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -38,8 +38,10 @@ CMAKE_MINIMUM_REQUIRED( VERSION 2.6.0 )
 SET( CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/CMake" )
 SET( CMAKE_AUTOMOC ON )
 
-# try Qt5 first, and prefer that if found
-find_package(Qt5Core QUIET)
+if(NOT WITH_QT4)
+    # try Qt5 first, and prefer that if found
+    find_package(Qt5Core QUIET)
+endif()
 IF (Qt5Core_FOUND)
     find_package(Qt5Widgets REQUIRED)
     find_package(Qt5Xml REQUIRED)
