diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4e735a1..82dbdce 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -61,11 +61,11 @@ IF (Qt5Core_FOUND)
         SET(QT_PLATFORM_SPECIFIC_LIBRARIES)
     ENDIF()
 
-    IF(APPLE)
+#     IF(APPLE)
         find_package(Qt5MacExtras REQUIRED)
         include_directories(${Qt5MacExtras_INCLUDE_DIRS})
         set(QT_PLATFORM_SPECIFIC_LIBRARIES ${Qt5MacExtras_LIBRARIES})
-    ENDIF()
+#     ENDIF()
 
     IF(NOT APPLE)
         find_package(Qt5DBus)
@@ -124,7 +124,7 @@ ELSE()
 
     SET(CHARM_MAC_HIGHRES_SUPPORT_ENABLED OFF)
 
-    IF(UNIX AND NOT APPLE)
+    IF(UNIX) # AND NOT APPLE)
         find_package(Qt4 COMPONENTS QtDBus)
         IF (QT_QTDBUS_FOUND)
             set(HAVE_DBUS ON)
