diff --git a/src/macports1.0/macports.tcl b/src/macports1.0/macports.tcl
index 209f10834..54576a861 100644
--- a/src/macports1.0/macports.tcl
+++ b/src/macports1.0/macports.tcl
@@ -210,8 +210,8 @@ proc macports::ch_logging {mport} {
 
     set ::debuglogname $logname
 
-    # Append to the file if it already exists
-    set ::debuglog [open $::debuglogname a]
+    # Truncate to the file if it already exists
+    set ::debuglog [open $::debuglogname w]
     puts $::debuglog version:1
 
     ui_debug "Starting logging for $portname @$portinfo(version)_$portinfo(revision)$portinfo(canonical_active_variants)"
