commit 803ecf0dd53ea51cd469c58aea163ae471dab96e
Author: René Bertin <rjvbertin@gmail.com>
Date:   Fri Aug 9 16:24:50 2024 +0200

    patch-nowarn-unsigned-source.diff

diff --git a/src/macports1.0/macports.tcl b/src/macports1.0/macports.tcl
index e45f42312..3d76688b6 100644
--- a/src/macports1.0/macports.tcl
+++ b/src/macports1.0/macports.tcl
@@ -1264,13 +1264,13 @@ proc mportinit {{up_ui_options {}} {up_options {}} {up_variations {}}} {
                         set sources_default [concat [list $url] $flags]
                     }
                 }
-                if {[string match rsync://*rsync.macports.org/release/ports/ $url]} {
-                    ui_warn "MacPorts is configured to use an unsigned source for the ports tree.\
-Please edit sources.conf and change '$url' to '[string range $url 0 end-14]macports/release/tarballs/ports.tar'."
-                } elseif {[string match rsync://rsync.macports.org/release/* $url]} {
-                    ui_warn "MacPorts is configured to use an older rsync URL for the ports tree.\
-Please edit sources.conf and change '$url' to '[string range $url 0 26]macports/release/tarballs/ports.tar'."
-                }
+#                 if {[string match rsync://*rsync.macports.org/release/ports/ $url]} {
+#                     ui_warn "MacPorts is configured to use an unsigned source for the ports tree.\
+# Please edit sources.conf and change '$url' to '[string range $url 0 end-14]macports/release/tarballs/ports.tar'."
+#                 } elseif {[string match rsync://rsync.macports.org/release/* $url]} {
+#                     ui_warn "MacPorts is configured to use an older rsync URL for the ports tree.\
+# Please edit sources.conf and change '$url' to '[string range $url 0 26]macports/release/tarballs/ports.tar'."
+#                 }
                 switch -- [macports::getprotocol $url] {
                     rsync -
                     https -
