commit f54bb00021d940912cca2fe527e2e367f40f4920
Author: René Bertin <rjvbertin@gmail.com>
Date:   Thu May 30 01:06:44 2024 +0200

    patch-conditional-ccachedir-creation.diff

diff --git a/src/port1.0/portconfigure.tcl b/src/port1.0/portconfigure.tcl
index d1e20b513..c2071f2f8 100644
--- a/src/port1.0/portconfigure.tcl
+++ b/src/port1.0/portconfigure.tcl
@@ -410,15 +410,17 @@ proc portconfigure::configure_start {args} {
     if {${configure.ccache}} {
         global ccache_dir ccache_size macportsuser
         # Create ccache directory with correct permissions with root privileges
-        elevateToRoot "configure ccache"
-        if {[catch {
+        if {![file exists ${ccache_dir}] || [file type ${ccache_dir}] ne "directory"} {
+            try {
+                elevateToRoot "Set up ccache directory `${ccache_dir}`"
                 file mkdir ${ccache_dir}
                 file attributes ${ccache_dir} -owner ${macportsuser} -permissions 0755
-            } result]} {
-            ui_warn "ccache_dir ${ccache_dir} could not be created; disabling ccache: $result"
-            set configure.ccache no
+            } catch {{*} eCode eMessage} {
+                ui_warn "ccache_dir ${ccache_dir} could not be created; disabling ccache: $eMessage"
+                set configure.ccache no
+            }
+            dropPrivileges
         }
-        dropPrivileges
 
         # Initialize ccache directory with the given maximum size
         if {${configure.ccache}} {
