diff --git a/src/port/port.tcl b/src/port/port.tcl
index bb70731dddbdac1a629a4df800bdad767ae3ff12..ec70c2d91d4cca5a2e2124057057de41a096d4e2 100755
--- a/src/port/port.tcl
+++ b/src/port/port.tcl
@@ -3782,7 +3782,7 @@ proc action_portcmds { action portlist opts } {
                 work {
                     # output the path to the port's work directory
                     set workpath [macports::getportworkpath_from_portdir $portdir $portname]
-                    if {[file exists $workpath]} {
+                    if {[file exists $workpath] || [macports::global_option_isset ports_force]} {
                         puts $workpath
                     }
                 }
