diff --git a/src/port1.0/portutil.tcl b/src/port1.0/portutil.tcl
index a1488c51..826c628c 100644
--- a/src/port1.0/portutil.tcl
+++ b/src/port1.0/portutil.tcl
@@ -94,6 +94,29 @@ proc handle_option {option args} {
     }
 }
 
+##
+# Handle option-contains
+#
+# @param option name of the option
+# @param args arguments, can contain one of lsearch's options
+proc handle_option-contains {option args} {
+    global $option user_options option_procs
+
+    if {![info exists user_options($option)] && [info exists $option]} {
+        set arg0 [lindex $args 0]
+        if {[string index $arg0 0] eq "-"} {
+            set searchopt $arg0
+            set args [lrange $args 1 end]
+        } else {
+            set searchopt "-exact"
+        }
+        if {[lsearch $searchopt [set $option] $args] >= 0} {
+            return yes
+        }
+    }
+    return no
+}
+
 ##
 # Handle option-append
 #
@@ -207,6 +230,7 @@ proc handle_option-replace {option args} {
 proc options {args} {
     foreach option $args {
         interp alias {} $option {} handle_option $option
+        interp alias {} $option-contains {} handle_option-contains $option
         interp alias {} $option-append {} handle_option-append $option
         interp alias {} $option-prepend {} handle_option-prepend $option
         interp alias {} $option-delete {} handle_option-delete $option
