diff --git a/src/port/port.tcl b/src/port/port.tcl
index 2c5b59f93..3a280c350 100755
--- a/src/port/port.tcl
+++ b/src/port/port.tcl
@@ -2539,9 +2539,16 @@ proc action_activate { action portlist opts } {
     }
     foreachport $portlist {
         set composite_version [composite_version $portversion [array get variations]]
+        array set actoptions [array get options]
+        set force 0
+        if ([info exists actoptions(ports_force)]) {
+            if ([string is true -strict $actoptions(ports_force)]) {
+                set force 1
+            }
+        }
         if {![info exists options(ports_activate_no-exec)]
             && ![catch {set ilist [registry::installed $portname $composite_version]}]
-            && [llength $ilist] == 1} {
+            && ([llength $ilist] == 1 && !$force)} {
 
             set i [lindex $ilist 0]
             set regref [registry::entry open $portname [lindex $i 1] [lindex $i 2] [lindex $i 3] [lindex $i 5]]
diff --git a/src/registry2.0/portimage.tcl b/src/registry2.0/portimage.tcl
index 70b79aae8..315944194 100644
--- a/src/registry2.0/portimage.tcl
+++ b/src/registry2.0/portimage.tcl
@@ -97,6 +97,7 @@ proc activate {name {version ""} {revision ""} {variants 0} {optionslist ""}} {
         set registry_open yes
     }
     set todeactivate [list]
+    set actaction "Activating"
 
     registry::read {
 
@@ -110,9 +111,14 @@ proc activate {name {version ""} {revision ""} {variants 0} {optionslist ""}} {
         set location [$requested location]
 
         if {[$requested state] eq "installed"} {
-            ui_info "${name} @${specifier} is already active."
-            #registry::entry close $requested
-            return
+            if {$force} {
+                set actaction "Reactivating"
+                lappend todeactivate $requested
+            } else {
+                ui_info "${name} @${specifier} is already active."
+                #registry::entry close $requested
+                return
+            }
         }
 
         # this shouldn't be possible
@@ -143,7 +149,7 @@ proc activate {name {version ""} {revision ""} {variants 0} {optionslist ""}} {
             }
         }
 
-        ui_msg "$UI_PREFIX [format [msgcat::mc "Activating %s @%s"] $name $specifier]"
+        ui_msg "$UI_PREFIX [format [msgcat::mc "$actaction %s @%s"] $name $specifier]"
 
         _activate_contents $requested $rename_list
     } finally {
