commit 625406bca86dc9eeed0a9ec3f45b18e4eeddb9ab
Author: René Bertin <rjvbertin@gmail.com>
Date:   Fri Aug 9 15:25:10 2024 +0200

    patch-port-environment-option.diff

diff --git a/src/port/port.tcl b/src/port/port.tcl
index bb70731dd..4d42da609 100755
--- a/src/port/port.tcl
+++ b/src/port/port.tcl
@@ -2648,6 +2648,12 @@ proc action_version { action portlist opts } {
 }
 
 
+proc action_environment { action portlist opts } {
+    set status [macports::environment]
+    return $status
+}
+
+
 proc action_platform { action portlist opts } {
     global macports::os_platform macports::os_major macports::os_arch
     if {![macports::ui_isset ports_quiet]} {
@@ -4083,6 +4089,7 @@ set action_array [dict create \
     diagnose    [list action_diagnose       [ACTION_ARGS_NONE]] \
     \
     version     [list action_version        [ACTION_ARGS_NONE]] \
+    environment [list action_environment    [ACTION_ARGS_NONE]] \
     platform    [list action_platform       [ACTION_ARGS_NONE]] \
     \
     uninstall   [list action_uninstall      [ACTION_ARGS_PORTS]] \
@@ -4340,6 +4347,9 @@ proc parse_options { action ui_options_name global_options_name } {
             set opts [string range $arg 1 end]
             foreach c [split $opts {}] {
                 switch -- $c {
+                    e {
+                        set ui_options(ports_env) yes
+                    }
                     v {
                         set ui_options(ports_verbose) yes
                     }
