diff --git src/macports1.0/macports.tcl src/macports1.0/macports.tcl
index a4ce074..51b0571 100644
--- src/macports1.0/macports.tcl
+++ src/macports1.0/macports.tcl
@@ -158,8 +211,8 @@ proc macports::ch_logging {mport} {
 
     set ::debuglogname $logname
 
-    # Append to the file if it already exists
-    set ::debuglog [open $::debuglogname a]
+    # Truncate to the file if it already exists
+    set ::debuglog [open $::debuglogname w]
     puts $::debuglog version:1
 }
 proc macports::push_log {mport} {

