commit 64021e044152893977d50cd09b6b6f4f5c957c6d
Author: René Bertin <rjvbertin@gmail.com>
Date:   Thu May 30 01:06:42 2024 +0200

    patch-allow-reactivation.diff
    CHECKME!!

diff --git a/src/port/port.tcl b/src/port/port.tcl
index d0d46e1b6..758177662 100755
--- a/src/port/port.tcl
+++ b/src/port/port.tcl
@@ -2208,8 +2208,14 @@ proc action_activate { action portlist opts } {
     }
     foreachport $portlist {
         set composite_version [composite_version $portversion $variations]
+        if {[dict exists $options ports_force]} {
+            set force 1
+        } else {
+            set force 0
+        }
         if {![dict exists $options ports_activate_no-exec]
-            && ![catch {registry_installed $portname $composite_version} regref]} {
+            && ![catch {registry_installed $portname $composite_version} regref]
+            && !$force} {
 
             if {[$regref installtype] eq "image" && [registry::run_target $regref activate $options]} {
                 continue
diff --git a/src/registry2.0/portimage.tcl b/src/registry2.0/portimage.tcl
index 95478fb2c..b84a2561f 100644
--- a/src/registry2.0/portimage.tcl
+++ b/src/registry2.0/portimage.tcl
@@ -91,6 +91,7 @@ proc activate {name {version ""} {revision ""} {variants 0} {options ""}} {
         set rename_list [dict get $options portactivate_rename_files]
     }
     set todeactivate [list]
+    set actaction "Activating"
 
     registry::read {
 
@@ -104,9 +105,14 @@ proc activate {name {version ""} {revision ""} {variants 0} {options ""}} {
         set location [$requested location]
 
         if {[$requested state] eq "installed"} {
-            ui_info "${name} @${specifier} is already active."
-            #registry::entry close $requested
-            return
+            if {$force} {
+                set actaction "Reactivating"
+                lappend todeactivate $requested
+            } else {
+                ui_info "${name} @${specifier} is already active."
+                #registry::entry close $requested
+                return
+            }
         }
 
         # this shouldn't be possible
@@ -137,7 +143,7 @@ proc activate {name {version ""} {revision ""} {variants 0} {options ""}} {
             }
         }
 
-        ui_msg "$UI_PREFIX [format [msgcat::mc "Activating %s @%s"] $name $specifier]"
+        ui_msg "$UI_PREFIX [format [msgcat::mc "$actaction %s @%s"] $name $specifier]"
 
         _activate_contents $requested $rename_list
     } finally {
