diff --git a/src/macports1.0/macports.tcl b/src/macports1.0/macports.tcl
index 2faa45e2a..dadda9b2c 100644
--- a/src/macports1.0/macports.tcl
+++ b/src/macports1.0/macports.tcl
@@ -686,12 +686,15 @@ proc mportinit {{up_ui_options {}} {up_options {}} {up_variations {}}} {
         macports::ui_init $priority
     }
 
+    global tcl_platform
     package require Pextlib 1.0
     package require registry 1.0
     package require registry2 2.0
-    package require machista 1.0
+    if {[string tolower $tcl_platform(os)] eq "darwin"} {
+        package require machista 1.0
+    }
 
-    global auto_path env tcl_platform \
+    global auto_path env \
         macports::autoconf::macports_conf_path \
         macports::macports_user_dir \
         macports::bootstrap_options \
@@ -751,9 +754,12 @@ proc mportinit {{up_ui_options {}} {up_options {}} {up_variations {}}} {
 
     # set up platform info variables
     set os_arch $tcl_platform(machine)
+    set os_platform [string tolower $tcl_platform(os)]
     # Set os_arch to match `uname -p`
     if {$os_arch eq "Power Macintosh"} {
         set os_arch "powerpc"
+    } elseif {${os_platform} eq "linux" && ($os_arch eq {i586} || $os_arch eq {i686})} {
+        set os_arch "i386"
     } elseif {$os_arch eq "i586" || $os_arch eq "i686" || $os_arch eq "x86_64"} {
         set os_arch "i386"
     } elseif {$os_arch eq "arm64"} {
@@ -761,8 +767,13 @@ proc mportinit {{up_ui_options {}} {up_options {}} {up_variations {}}} {
     }
     set os_version $tcl_platform(osVersion)
     set os_major [lindex [split $os_version .] 0]
+    if {${os_platform} eq "linux" && (${os_major} > 3) } {
+        # RJVB : on Linux, 2.x kernels may need special treatment, but booting
+        # a 3.x or 4.x kernel (or later, presumably) doesn't make enough difference
+        # for the userland to justify upgrading all ports.
+        set os_major 3
+    }
     set os_minor [lindex [split $os_version .] 1]
-    set os_platform [string tolower $tcl_platform(os)]
     # Remove trailing "Endian"
     set os_endian [string range $tcl_platform(byteOrder) 0 end-6]
     set os_subplatform {}
@@ -4044,9 +4146,18 @@ proc macports::_upgrade {portname dspec variationslist optionslist {depscachenam
     }
     set requestedflag [$regref requested]
     set os_platform_installed [$regref os_platform]
-    set os_major_installed [$regref os_major]
-    set cxx_stdlib_installed [$regref cxx_stdlib]
-    set cxx_stdlib_overridden [$regref cxx_stdlib_overridden]
+    if {$macports::os_platform eq "darwin"} {
+        set os_major_installed [$regref os_major]
+        set cxx_stdlib_installed [$regref cxx_stdlib]
+        set cxx_stdlib_overridden [$regref cxx_stdlib_overridden]
+    } else {
+        # esp. on Linux there's no point in looking at the kernel version
+        global macports::os_major
+        set os_major_installed ${macports::os_major}
+        # nor at cxx_ stdlib stuff
+        set cxx_stdlib_installed []
+        set cxx_stdlib_overridden []
+    }
 
     # Before we do
     # dependencies, we need to figure out the final variants,
@@ -5009,6 +5026,7 @@ proc macports::revupgrade_update_cxx_stdlib {fancy_output {revupgrade_progress "
 # @return 1 if ports were rebuilt and this function should be called again,
 #         0 otherwise.
 proc macports::revupgrade_scanandrebuild {broken_port_counts_name opts} {
+    global macports::os_platform
     upvar $broken_port_counts_name broken_port_counts
     array set options $opts
 
@@ -5026,7 +5044,7 @@ proc macports::revupgrade_scanandrebuild {broken_port_counts_name opts} {
     set broken_files {}
     set binaries [registry::file search active 1 binary 1]
     set binary_count [llength $binaries]
-    if {$binary_count > 0} {
+    if {$binary_count > 0 && $macports::os_platform eq "darwin"} {
         ui_msg "$macports::ui_prefix Scanning binaries for linking errors"
         set handle [machista::create_handle]
         if {$handle eq "NULL"} {
