--- src/port1.0/orig.portutil.tcl	2015-10-10 19:45:01.612205963 +0200
+++ src/port1.0/portutil.tcl	2015-10-10 19:45:22.887205409 +0200
@@ -856,10 +856,14 @@
 # debugging purposes.
 proc environment_array_to_string {environment_array} {
     upvar 1 ${environment_array} env_array
-    foreach {key value} [array get env_array] {
-        lappend env_list $key='$value'
+    if {[array size env_array] > 0} {
+        foreach {key value} [array get env_array] {
+            lappend env_list $key='$value'
+        }
+        return "\n[join [lsort $env_list] "\n"]"
+    } else {
+        return ""
     }
-    return "\n[join [lsort $env_list] "\n"]"
 }
 
 ########### Distname utility functions ###########
