diff --git a/vendor/critcl-3.3.1/build.tcl b/vendor/critcl-3.3.1/build.tcl
index 61955b93c..4444e344b 100755
--- a/vendor/critcl-3.3.1/build.tcl
+++ b/vendor/critcl-3.3.1/build.tcl
@@ -241,7 +241,9 @@ proc process-install-options {} {
 		set argv [lrange $argv 2 end]
 		set key  [string range $o 2 end]
 		global paths
-		set paths($key) [norm $path]
+		#set paths($key) [norm $path]
+		# RJVB : DO NOT NORMALISE the requested prefix!!!
+		set paths($key) $path
 	    }
 	    -- break
 	    default {
