diff --git a/src/macports1.0/macports.tcl b/src/macports1.0/macports.tcl
index 97da311201800f16edfed4cc30d6b234dea84155..59f35b4332724ef10ef375a85cf15229f72ffb90 100644
--- a/src/macports1.0/macports.tcl
+++ b/src/macports1.0/macports.tcl
@@ -798,10 +819,10 @@ proc mportinit {{up_ui_options {}} {up_options {}} {up_variations {}}} {
                         set sources_default [concat [list $url] $flags]
                     }
                 }
-                if {[string match rsync://*rsync.macports.org/release/ports/ $url]} {
-                    ui_warn "MacPorts is configured to use an unsigned source for the ports tree.\
-Please edit sources.conf and change '$url' to '[string range $url 0 end-6]tarballs/ports.tar'."
-                }
+#                 if {[string match rsync://*rsync.macports.org/release/ports/ $url]} {
+#                     ui_warn "MacPorts is configured to use an unsigned source for the ports tree.\
+# Please edit sources.conf and change '$url' to '[string range $url 0 end-6]tarballs/ports.tar'."
+#                 }
                 lappend sources [concat [list $url] $flags]
             } else {
                 ui_warn "$sources_conf specifies invalid source '$line', ignored."
