diff --git a/doc/macports.conf.in b/doc/macports.conf.in
index e5ad2f0b6..7fb27501e 100644
--- a/doc/macports.conf.in
+++ b/doc/macports.conf.in
@@ -1,6 +1,15 @@
 # MacPorts system-wide configuration file.
 # Commented-out values are defaults unless otherwise noted.
 
+# Set normalise_paths if you want your prefix to be a symlink,
+# e.g. to a directory on a different volume. This reverts the fix for
+# https://trac.macports.org/ticket/69327
+# and can thus cause the symptoms mentioned in that ticket.
+# NB : this has to be set BEFORE defining prefix or any other path
+# defined in here you want to prevent from being normalised; the
+# option can be turned off and on multiple times.
+# normalise_paths   	yes
+
 # Directory under which MacPorts should install ports. This must be
 # where MacPorts itself is installed.
 prefix              	@prefix_expanded@
diff --git a/src/macports1.0/macports.tcl b/src/macports1.0/macports.tcl
index 644e0004d..ca4f02cf1 100644
--- a/src/macports1.0/macports.tcl
+++ b/src/macports1.0/macports.tcl
@@ -63,7 +63,7 @@ namespace eval macports {
         revupgrade_autorun revupgrade_mode revupgrade_check_id_loadcmds \
         host_blacklist preferred_hosts sandbox_enable sandbox_network delete_la_files cxx_stdlib \
         default_compilers pkg_post_unarchive_deletions ui_interactive \
-        use_shorter_workpath] {
+        use_shorter_workpath normalise_paths] {
             dict set bootstrap_options $opt {}
     }
     # Config file options that are a filesystem path and should be fully resolved
@@ -89,7 +89,7 @@ namespace eval macports {
         os_platform os_subplatform macos_version macos_version_major macosx_version macosx_sdk_version \
         macosx_deployment_target packagemaker_path default_compilers sandbox_enable sandbox_network \
         delete_la_files cxx_stdlib pkg_post_unarchive_deletions \
-        use_shorter_workpath {*}$user_options]
+        use_shorter_workpath normalise_paths {*}$user_options]
 
     # Options set in the portfile interpreter but only in system_options
     variable portinterp_private_options [list clonebin_path]
@@ -1063,7 +1063,8 @@ proc mportinit {{up_ui_options {}} {up_options {}} {up_variations {}}} {
         macports::ui_options \
         macports::global_options \
         macports::global_variations \
-        macports::use_shorter_workpath
+        macports::use_shorter_workpath \
+        macports::normalise_paths
 
     # Disable unknown(n)'s behavior of running unknown commands in the system
     # shell
@@ -1211,7 +1212,7 @@ proc mportinit {{up_ui_options {}} {up_options {}} {up_variations {}}} {
                 if {[dict exists $bootstrap_options $option]} {
                     global macports::$option
                     set val [string trim $val]
-                    if {[dict exists $bootstrap_options $option is_path]} {
+                    if {(![info exists macports::normalise_paths] || ${macports::normalise_paths}) && [dict exists $bootstrap_options $option is_path]} {
                         if {[catch {set $option [realpath $val]}]} {
                             set $option [file normalize $val]
                         }
@@ -1769,6 +1770,9 @@ match macports.conf.default."
     if {![info exists use_shorter_workpath]} {
         set use_shorter_workpath no
     }
+    if {![info exists normalise_paths]} {
+        set normalise_paths no
+    }
 
     # make tools we run operate in UTF-8 mode
     set env(LANG) en_US.UTF-8
diff --git a/src/port1.0/portmain.tcl b/src/port1.0/portmain.tcl
index 6021648a4..78cb1a3e6 100644
--- a/src/port1.0/portmain.tcl
+++ b/src/port1.0/portmain.tcl
@@ -61,7 +61,7 @@ options prefix name version revision epoch categories maintainers \
         compiler.support_environment_paths \
         compiler.support_environment_sdkroot \
         add_users use_xcode source_date_epoch \
-        use_shorter_workpath
+        use_shorter_workpath normalise_paths
 
 proc portmain::check_option_integer {option action args} {
     if {$action eq "set" && ![string is wideinteger -strict $args]} {
