diff --git a/src/macports1.0/macports.tcl b/src/macports1.0/macports.tcl
index 97da311201800f16edfed4cc30d6b234dea84155..33daef33b40a60c544bb133dc1e0a7fe7fb10720 100644
--- a/src/macports1.0/macports.tcl
+++ b/src/macports1.0/macports.tcl
@@ -4680,34 +4738,39 @@ proc macports::revupgrade_update_binary {fancy_output {revupgrade_progress ""}}
 # @param revupgrade_progress
 #        Progress display callback name
 proc macports::revupgrade_update_cxx_stdlib {fancy_output {revupgrade_progress ""}} {
-    set maybe_cxx_ports [registry::entry search state installed cxx_stdlib -null]
-    set maybe_cxx_len [llength $maybe_cxx_ports]
-    if {$maybe_cxx_len > 0} {
-        ui_msg "$macports::ui_prefix Updating database of C++ stdlib usage"
-        set i 1
-        if {$fancy_output} {
-            $revupgrade_progress start
-        }
-        foreach maybe_port $maybe_cxx_ports {
-            registry::write {
-                if {$fancy_output} {
-                    $revupgrade_progress update $i $maybe_cxx_len
-                }
-                incr i
-                set binary_files {}
-                foreach filehandle [registry::file search id [$maybe_port id] binary 1] {
-                    lappend binary_files [$filehandle actual_path]
-                }
-                $maybe_port cxx_stdlib [get_actual_cxx_stdlib $binary_files]
-                if {[catch {$maybe_port cxx_stdlib_overridden}]} {
-                    # can't tell after the fact, assume not overridden
-                    $maybe_port cxx_stdlib_overridden 0
+    global macports::os_platform
+    if {$macports::os_platform eq "darwin"} {
+        set maybe_cxx_ports [registry::entry search state installed cxx_stdlib -null]
+        set maybe_cxx_len [llength $maybe_cxx_ports]
+        if {$maybe_cxx_len > 0} {
+            ui_msg "$macports::ui_prefix Updating database of C++ stdlib usage"
+            set i 1
+            if {$fancy_output} {
+                $revupgrade_progress start
+            }
+            foreach maybe_port $maybe_cxx_ports {
+                registry::write {
+                    if {$fancy_output} {
+                        $revupgrade_progress update $i $maybe_cxx_len
+                    }
+                    incr i
+                    set binary_files {}
+                    foreach filehandle [registry::file search id [$maybe_port id] binary 1] {
+                        lappend binary_files [$filehandle actual_path]
+                    }
+                    $maybe_port cxx_stdlib [get_actual_cxx_stdlib $binary_files]
+                    if {[catch {$maybe_port cxx_stdlib_overridden}]} {
+                        # can't tell after the fact, assume not overridden
+                        $maybe_port cxx_stdlib_overridden 0
+                    }
                 }
             }
+            if {$fancy_output} {
+                $revupgrade_progress finish
+            }
         }
-        if {$fancy_output} {
-            $revupgrade_progress finish
-        }
+    } else {
+        ui_debug "Skipping C++ stdlib checks on $macports::os_platform"
     }
 }
 
@@ -4983,9 +5042,11 @@ proc macports::revupgrade_scanandrebuild {broken_port_counts_name opts} {
         }
         set broken_cxx_ports [registry::entry search state installed cxx_stdlib_overridden 0 cxx_stdlib $wrong_stdlib]
         foreach cxx_port $broken_cxx_ports {
-            ui_info "[$cxx_port name] is using $wrong_stdlib (this installation is configured to use ${macports::cxx_stdlib})"
+            ui_warn "[$cxx_port name] is using $wrong_stdlib (this installation is configured to use ${macports::cxx_stdlib})"
         }
-        set broken_ports [lsort -unique [concat $broken_ports $broken_cxx_ports]]
+#         set broken_ports [lsort -unique [concat $broken_ports $broken_cxx_ports]]
+        # just using a different cxx_stdlib isn't necessarily an error if it works!
+        set broken_ports [lsort -unique [concat $broken_ports]]
 
         if {[llength $broken_ports] == 0} {
             ui_msg "$macports::ui_prefix No broken ports found."
