Index: src/macports1.0/macports.tcl
===================================================================
--- src/macports1.0/macports.tcl	(revision 140947)
+++ src/macports1.0/macports.tcl	(working copy)
@@ -1950,6 +2012,7 @@
             if {![macports::ui_isset ports_debug]} {
                 ui_msg {}
             }
+            ui_msg "Can't install $portinfo(name) because conflicting ports are active: $conflictlist"
             return -code error "Can't install $portinfo(name) because conflicting ports are active: $conflictlist"
         }
     }
