--- src/macports1.0/orig.macports.tcl	2015-10-10 12:44:21.527987995 +0200
+++ src/macports1.0/macports.tcl	2015-10-10 12:44:34.062987669 +0200
@@ -670,12 +670,16 @@
 
     # set up platform info variables
     set os_arch $tcl_platform(machine)
+    set os_platform [string tolower $tcl_platform(os)]
     if {$os_arch eq "Power Macintosh"} {set os_arch "powerpc"}
-    if {$os_arch eq "i586" || $os_arch eq "i686" || $os_arch eq "x86_64"} {set os_arch "i386"}
+    if {${os_platform} eq "linux"} {
+        if {$os_arch eq {i586} || $os_arch eq {i686}} {set os_arch "i386"}
+    } else {
+        if {$os_arch eq "i586" || $os_arch eq "i686" || $os_arch eq "x86_64"} {set os_arch "i386"}
+    }
     set os_version $tcl_platform(osVersion)
     set os_major [lindex [split $os_version .] 0]
     set os_minor [lindex [split $os_version .] 1]
-    set os_platform [string tolower $tcl_platform(os)]
     # Remove trailing "Endian"
     set os_endian [string range $tcl_platform(byteOrder) 0 end-6]
     set macosx_version {}
