diff --git a/aclocal.m4 b/aclocal.m4
index cc92d20c3..c5a9769f4 100644
--- a/aclocal.m4
+++ b/aclocal.m4
@@ -865,7 +865,7 @@ AC_DEFUN([MP_PROG_DAEMONDO],[
 #	None.
 #
 # Requires:
-#   curl-config or user parameters to define the flags.
+#   curl-config (or curl-config.host) or user parameters to define the flags.
 #
 # Results:
 #   defines some help strings.
@@ -877,7 +877,7 @@ AC_DEFUN([MP_LIBCURL_FLAGS],[
 		   [  curlprefix=$withval ])
 
 	if test "x$curlprefix" = "x"; then
-		AC_PATH_PROG([CURL_CONFIG], [curl-config])
+		AC_PATH_PROGS([CURL_CONFIG], [curl-config curl-config.host])
 	else
 		AC_PATH_PROG([CURL_CONFIG], [curl-config], , [$curlprefix/bin])
 	fi
@@ -888,9 +888,14 @@ AC_DEFUN([MP_LIBCURL_FLAGS],[
 
 	CFLAGS_LIBCURL=$($CURL_CONFIG --cflags)
 	if test "x$curlprefix" = "x"; then
-		# System curl-config emits absurd output for --libs
-		# See rdar://7244457
-		LDFLAGS_LIBCURL="-lcurl"
+		if test "$CURL_CONFIG" = "/usr/bin/curl-config.host"; then
+			# non-standard  curl-config; presume user-provided and trustworthy
+			LDFLAGS_LIBCURL=$($CURL_CONFIG --libs)
+		else
+			# System curl-config emits absurd output for --libs
+			# See rdar://7244457
+			LDFLAGS_LIBCURL="-lcurl"
+		fi
 	else
 		# Due to a bug in dist, --arch flags are improperly supplied by curl-config.
 		# Get rid of them.
