--- opt/local/libexec/macports/lib/port1.0/portconfigure.tcl	2015-08-10 18:23:02.613950000 +0200
+++ opt/local/libexec/macports/lib/port1.0/portconfigure.tcl	2015-11-06 18:33:22.370185538 +0100
@@ -196,7 +196,11 @@
 default configure.cflags        {${configure.optflags}}
 default configure.objcflags     {${configure.optflags}}
 default configure.cppflags      {-I${prefix}/include}
-default configure.ldflags       {"-L${prefix}/lib -Wl,-headerpad_max_install_names"}
+if {${os.platform} eq "darwin"} {
+	default configure.ldflags  {"-L${prefix}/lib -Wl,-headerpad_max_install_names"}
+} else {
+	default configure.ldflags  {"-L${prefix}/lib"}
+}
 default configure.libs          {}
 default configure.fflags        {${configure.optflags}}
 default configure.f90flags      {${configure.optflags}}
