Index: src/port/port.tcl
===================================================================
--- src/port/port.tcl	(revision 140947)
+++ src/port/port.tcl	(working copy)
@@ -2814,6 +2814,12 @@
 }
 
 
+proc action_environment { action portlist opts } {
+    set status [macports::environment]
+    return $status
+}
+
+
 proc action_platform { action portlist opts } {
     if {![macports::ui_isset ports_quiet]} {
         puts -nonewline "Platform: "
@@ -4253,6 +4259,7 @@
     diagnose    [list action_diagnose       [ACTION_ARGS_NONE]] \
     \
     version     [list action_version        [ACTION_ARGS_NONE]] \
+    environment [list action_environment    [ACTION_ARGS_NONE]] \
     platform    [list action_platform       [ACTION_ARGS_NONE]] \
     \
     uninstall   [list action_uninstall      [ACTION_ARGS_PORTS]] \
@@ -4511,6 +4518,9 @@
             set opts [string range $arg 1 end]
             foreach c [split $opts {}] {
                 switch -- $c {
+                    e {
+                        set ui_options(ports_env) yes
+                    }
                     v {
                         set ui_options(ports_verbose) yes
                     }
