commit 388ae909694f64949e93ff5d93a2ed5b26f06fb4
Author: René Bertin <rjvbertin@gmail.com>
Date:   Wed May 29 22:30:58 2024 +0200

    patch-port-force-edit.diff

diff --git a/src/port/port.tcl b/src/port/port.tcl
index 273e4f030..0a7e7785b 100755
--- a/src/port/port.tcl
+++ b/src/port/port.tcl
@@ -3689,7 +3689,7 @@ proc action_portcmds { action portlist opts } {
                     set editor_var "ports_${action}_editor"
                     if {[dict exists $opts $editor_var]} {
                         set editor [join [dict get $opts $editor_var]]
-                    } else {
+                    } elseif {![macports::global_option_isset ports_force]} {
                         foreach ed { MP_EDITOR VISUAL EDITOR } {
                             if {[info exists env($ed)]} {
                                 set editor $env($ed)
@@ -3699,7 +3699,14 @@ proc action_portcmds { action portlist opts } {
                     }
 
                     # Use a reasonable canned default if no editor specified or set in env
-                    if { $editor eq "" } { set editor "/usr/bin/vi" }
+                    # or the user used the -f option to bypass any custom settings.
+                    if { $editor eq "" } {
+                        if {[file exists ${macports::prefix}/bin/vim]} {
+                            set editor "${macports::prefix}/bin/vim"
+                        } else {
+                            set editor "/usr/bin/vi"
+                        }
+                    }
 
                     # Invoke the editor
                     if {[catch {exec -ignorestderr >@stdout <@stdin {*}$editor $portfile} result]} {
