--- /opt/local/var/lnxports/build/_opt_local_site-ports_sysutils_MacPorts/MacPorts-devel/work/base/src/port1.0/portfetch.tcl	2014-11-20 16:26:29.063309000 +0100
+++ portfetch.tcl	2016-10-29 19:19:59.351068604 +0200
@@ -553,7 +553,19 @@
                     file delete -force "${distpath}/${distfile}.TMP"
                     throw
                 } catch {{*} eCode eMessage} {
-                    ui_debug [msgcat::mc "Fetching distfile failed: %s" $eMessage]
+                    ui_warn [msgcat::mc "Fetching distfile failed: %s" $eMessage]
+				ui_debug "complete command:"
+                    ui_debug "curl fetch {*}$fetch_options $file_url \"${distpath}/${distfile}.TMP\""
+                    if {${eMessage} eq "gnutls_handshake() failed: Handshake failed"} {
+                        if {![catch {system "curl -L $file_url -o \"${distpath}/${distfile}.TMP\""} err]} {
+                            ui_debug [msgcat::mc "Fetching distfile succeeded via curl"]
+                            file rename -force "${distpath}/${distfile}.TMP" "${distpath}/${distfile}"
+                            set fetched 1
+                            break
+                        } else {
+                            ui_warn [msgcat::mc "Fetching distfile failed also via curl: %s" $err]
+                        }
+                    }
                     set lastError $eMessage
                 }
             }
