diff --git a/src/port1.0/portconfigure.tcl b/src/port1.0/portconfigure.tcl
index 8591c211119dc72260e03ed538d4a56dd7469e6e..1305b62617a8e5bb5ba7a2ba75258e6e4f94f399 100644
--- a/src/port1.0/portconfigure.tcl
+++ b/src/port1.0/portconfigure.tcl
@@ -115,10 +117,10 @@ proc portconfigure::stdlib_trace {opt action args} {
 }
 # helper function to set configure.cxx_stdlib
 proc portconfigure::configure_get_cxx_stdlib {} {
-    global cxx_stdlib compiler.cxx_standard
+    global cxx_stdlib compiler.cxx_standard os.platform
     if {${compiler.cxx_standard} eq ""} {
         return ""
-    } elseif {${cxx_stdlib} eq "libstdc++" && ${compiler.cxx_standard} >= 2011} {
+    } elseif {${os.platform} eq "darwin" && ${cxx_stdlib} eq "libstdc++" && ${compiler.cxx_standard} >= 2011} {
         return macports-libstdc++
     } else {
         return ${cxx_stdlib}
