

Index: src/port1.0/portmain.tcl
===================================================================
--- src/port1.0/portmain.tcl	(revision 140947)
+++ src/port1.0/portmain.tcl	(working copy)
@@ -156,21 +156,24 @@
 # or a shared directory owned by the group so use ~/.macports
 default altprefix {[file join $user_home .macports]}
 if { $euid != 0 && (([info exists workpath] && [file exists $workpath] && ![file writable $workpath]) || ([info exists portdbpath] && ![file writable [file join $portdbpath build]])) } {
-
-    # set global variable indicating to other functions to use ~/.macports as well
-    set usealtworkpath yes
-
-    default worksymlink {[file normalize [file join ${altprefix}${portpath} work]]}
-    default distpath {[file normalize [file join ${altprefix}${portdbpath} distfiles ${dist_subdir}]]}
-    set portbuildpath "${altprefix}${portbuildpath}"
-
-    ui_debug "Going to use alternate build prefix: $altprefix"
-    ui_debug "workpath = $workpath"
-} else {
+    ui_debug "Stock MacPorts would now use a work directory under ${altprefix} but this version trusts you have write access under ${prefix}/var/macports/build"
+}
+# if { $euid != 0 && (([info exists workpath] && [file exists $workpath] && ![file writable $workpath]) || ([info exists portdbpath] && ![file writable [file join $portdbpath build]])) } {
+#
+#     # set global variable indicating to other functions to use ~/.macports as well
+#     set usealtworkpath yes
+#
+#     default worksymlink {[file normalize [file join ${altprefix}${portpath} work]]}
+#     default distpath {[file normalize [file join ${altprefix}${portdbpath} distfiles ${dist_subdir}]]}
+#     set portbuildpath "${altprefix}${portbuildpath}"
+#
+#     ui_debug "Going to use alternate build prefix: $altprefix"
+#     ui_debug "workpath = $workpath"
+# } else {
     set usealtworkpath no
     default worksymlink {[file normalize [file join $portpath work]]}
     default distpath {[file normalize [file join $portdbpath distfiles ${dist_subdir}]]}
-}
+# }
 
 # end gsoc08-privileges
 
