diff --git a/src/cregistry/registry.c b/src/cregistry/registry.c
index 2b24642f4..a27e4d75a 100644
--- a/src/cregistry/registry.c
+++ b/src/cregistry/registry.c
@@ -551,6 +551,7 @@ int reg_vacuum(char *db_path) {
        least 1% of the total db size. */
     if (result && freesize >= 1000000 && dbsize > 0 && ((double)freesize / (double)dbsize) >= 0.01) {
         result = 0;
+        fprintf(stderr, "Vacuuming the registry ...") ; fflush(stderr);
         if (sqlite3_prepare_v2(db, "VACUUM", -1, &stmt, NULL) == SQLITE_OK) {
             sqlite3_step(stmt);
             r = sqlite3_reset(stmt);
@@ -560,6 +561,9 @@ int reg_vacuum(char *db_path) {
         }
         if (stmt) {
             sqlite3_finalize(stmt);
+             fprintf(stderr, result ? " done\n" : " failed\n");
+        } else {
+             fprintf(stderr, " failed\n");
         }
     }
 
