commit bd44338b3a39dd3901aa60fe36772cabf0b1c423
Author: René Bertin <rjvbertin@gmail.com>
Date:   Thu May 30 13:31:12 2024 +0200

    patch-parallel-xz.diff

diff --git a/src/port1.0/portinstall.tcl b/src/port1.0/portinstall.tcl
index 33ee3509e..46d9dcbd0 100644
--- a/src/port1.0/portinstall.tcl
+++ b/src/port1.0/portinstall.tcl
@@ -74,7 +74,8 @@ proc portinstall::create_archive {location archive.type} {
     global workpath destpath portpath subport version revision portvariants \
            epoch configure.cxx_stdlib cxx_stdlib PortInfo \
            archive.env archive.cmd archive.pre_args archive.args \
-           archive.post_args archive.dir depends_lib depends_run
+           archive.post_args archive.dir depends_lib depends_run \
+           build.jobs prefix
     set archive.env {}
     set archive.cmd {}
     set archive.pre_args {}
@@ -126,6 +127,7 @@ proc portinstall::create_archive {location archive.type} {
                 set archive.cmd "$tar"
                 set archive.pre_args {-cvf}
                 if {[regexp {z2?$} ${archive.type}]} {
+                    set extraargs ""
                     if {[regexp {bz2?$} ${archive.type}]} {
                         if {![catch {binaryInPath lbzip2}]} {
                             set gzip "lbzip2"
@@ -139,8 +141,13 @@ proc portinstall::create_archive {location archive.type} {
                         set gzip "lzma"
                         set level ""
                     } elseif {[regexp {xz$} ${archive.type}]} {
-                        set gzip "xz"
+                        if {![catch {binaryInPath xz}]} {
+                            set gzip "xz"
+                        } else {
+                            set gzip "$prefix/sbin/xz"
+                        }
                         set level 6
+                        set extraargs "--threads=${build.jobs}"
                     } else {
                         set gzip "gzip"
                         set level 9
@@ -151,9 +158,9 @@ proc portinstall::create_archive {location archive.type} {
                         set hint ""
                     }
                     if {[catch {set gzip [findBinary $gzip $hint]} errmsg] == 0} {
-                        ui_debug "Using $gzip"
+                        ui_debug "Using $gzip -c$level $extraargs"
                         set archive.args {- .}
-                        set archive.post_args "| $gzip -c$level > [shellescape ${location}]"
+                        set archive.post_args "| $gzip -c$level $extraargs > [shellescape ${location}]"
                     } else {
                         ui_debug $errmsg
                         return -code error "No '$gzip' was found on this system!"
