diff --git a/src/port1.0/portutil.tcl b/src/port1.0/portutil.tcl
index 17b1cd192..10748ef52 100644
--- a/src/port1.0/portutil.tcl
+++ b/src/port1.0/portutil.tcl
@@ -1736,10 +1778,14 @@ proc portutil::create_workpath {} {
     }
     file mkdir $portbuildpath
     # Create and link build dir if link or link target are missing or wrong type.
-    if {[catch {file type $subbuildpath} ftype] || $ftype eq "directory" || ![file isdirectory $subbuildpath]} {
+    # RJVB : add a global variable to enable the mechanism, settable via macports.conf of in the Portfile
+    if {([catch {file type $subbuildpath} ftype] || $ftype eq "directory" || ![file isdirectory $subbuildpath])
+        && 0
+    } {
         # This doesn't need to be unguessable, just unique (and short). Four
         # random characters is enough for ~1.7M build dirs.
-        set shortpath [mkdtemp [file dirname $portbuildpath]/[string range $subport 0 3]XXXX]
+        # RJVB : but Linux wants 6 random chars at least.
+        set shortpath [mkdtemp [file dirname $portbuildpath]/[string range $subport 0 3]XXXXXX]
         if {$ftype eq "directory"} {
             delete $shortpath
             file rename $subbuildpath $shortpath
