diff --git a/src/registry2.0/registry_util.tcl b/src/registry2.0/registry_util.tcl
index 35c9d5178..ecf19a827 100644
--- a/src/registry2.0/registry_util.tcl
+++ b/src/registry2.0/registry_util.tcl
@@ -126,7 +126,7 @@ proc run_target {port target options} {
         return 0
     }
 
-    if {![catch {set mport [mportopen_installed [$port name] [$port version] [$port revision] [$port variants] $options]}]} {
+    if {![catch {set mport [mportopen_installed [$port name] [$port version] [$port revision] [$port variants] $options]} err]} {
         if {[catch {set result [mportexec $mport $target]} result] || $result != 0} {
             ui_debug $::errorInfo
             catch {mportclose $mport}
@@ -158,7 +158,7 @@ proc run_target {port target options} {
         }
     } else {
         ui_debug $::errorInfo
-        ui_warn "Failed to open Portfile from registry for $portspec"
+        ui_warn "Failed to open Portfile from registry for $portspec ($err); registry=[$port portfile]"
     }
     return 0
 }
