diff --git src/pextlib1.0/xinstall.c src/pextlib1.0/xinstall.c
index f28ec50e..0705e0b2 100644
--- src/pextlib1.0/xinstall.c
+++ src/pextlib1.0/xinstall.c
@@ -1074,7 +1074,14 @@ install_dir(Tcl_Interp *interp, char *path)
 	if ((gid != (gid_t)-1 || uid != (uid_t)-1) && chown(path, uid, gid))
 		{ /* Don't bother to warn */ };
 	if (chmod(path, mode))
-		{ /* Don't bother to warn */ };
+		{
+			char errmsg[255];
+
+			snprintf(errmsg, sizeof errmsg, "%s: chmod(%s,0%o), %s",
+				 funcname, path, mode, strerror(errno));
+			Tcl_SetResult(interp, errmsg, TCL_VOLATILE);
+			return TCL_ERROR;
+		}
 	return TCL_OK;
 }
 
