--- src/port1.0/portutil.tcl	2015-10-10 21:12:10.000000000 +0200
+++ src/port1.0/portutil.tcl	2015-10-28 16:26:26.000000000 +0100
@@ -93,6 +93,7 @@
 
     if {![info exists user_options($option)]} {
         set $option $args
+        options_export $option
     }
 }
 
@@ -110,6 +111,7 @@
         } else {
             set $option $args
         }
+        options_export $option
     }
 }
 
@@ -127,6 +129,7 @@
         } else {
             set $option $args
         }
+        options_export $option
     }
 }
 
@@ -144,6 +147,7 @@
             set temp [ldelete $temp $val]
         }
         set $option $temp
+        options_export $option
     }
 }
 
@@ -161,6 +165,7 @@
             set temp [strsed $temp $val]
         }
         set $option $temp
+        options_export $option
     }
 }
 
@@ -190,6 +195,7 @@
                 continue
             }
             set $option [lreplace [set $option] $index $index $new]
+            options_export $option
         }
     }
 }
--- src/port/port.tcl	2015-10-10 21:12:10.000000000 +0200
+++ src/port/port.tcl	2015-10-28 16:34:59.000000000 +0100
@@ -1984,6 +1984,7 @@
 
 proc action_info { action portlist opts } {
     global global_variations
+    global global_options
     set status 0
     if {[require_portlist portlist]} {
         return 1
@@ -2198,6 +2199,20 @@
                 set inf "$portinfo(name) @"
                 append inf [composite_version $portinfo(version) $portinfo(active_variants)]
                 set ropt "fullname"
+            } elseif {$opt eq "var"} {
+                if {[info exists global_options(ports_info_var)]} {
+                    set opt ${global_options(ports_info_var)}
+                    set ropt $opt
+                    if {[info exists portinfo($opt)]} {
+                        set inf "$portinfo($opt)"
+                    } else {
+                        ui_error "variable \"$opt\" cannot be queried via info --var"
+                        return 1
+                    }
+                } else {
+                    ui_error "info --var requires a variable name"
+                    return 1
+                }
             } else {
                 # Map from friendly name
                 set ropt [map_friendly_field_names $opt]
@@ -4394,7 +4409,7 @@
                  depends description epoch fullname heading homepage index license
                  line long_description
                  maintainer maintainers name patchfiles platform platforms portdir
-                 pretty replaced_by revision subports variant variants version}
+                 pretty replaced_by revision subports variant variants version {var 1}}
     contents    {size {units 1}}
     deps        {index no-build}
     rdeps       {index no-build full}
