diff --git a/vendor/tclx8.4/Makefile.in b/vendor/tclx8.4/Makefile.in
index b812bbe711f7e0a68b96b932837f09f323b4bc0a..1b94c3c3fdebeb0bd9969ab844731c479bf299ae 100644
--- a/vendor/tclx8.4/Makefile.in
+++ b/vendor/tclx8.4/Makefile.in
@@ -123,7 +123,7 @@ OBJEXT		= @OBJEXT@
 RANLIB		= @RANLIB@
 RANLIB_STUB	= @RANLIB_STUB@
 SHLIB_CFLAGS	= @SHLIB_CFLAGS@
-SHLIB_LD	= @SHLIB_LD@
+SHLIB_LD	= @SHLIB_LD@ $(LDFLAGS_DEFAULT)
 SHLIB_LD_LIBS	= @SHLIB_LD_LIBS@
 STLIB_LD	= @STLIB_LD@
 TCL_DEFS	= @TCL_DEFS@
diff --git a/src/cregistry/Makefile.in b/src/cregistry/Makefile.in
index 408f94b7c24d7975670d96254a1a83c87554ca3b..2ffeec7e4b2a2a280f5aafb561756bf4556f410d 100644
--- a/src/cregistry/Makefile.in
+++ b/src/cregistry/Makefile.in
@@ -4,7 +4,6 @@ VPATH  = @srcdir@
 
 OBJS = registry.o entry.o sql.o vercomp.o util.o file.o portgroup.o
 STLIB_NAME = cregistry.a
-RANLIB = ranlib
 
 SQLEXT_NAME = macports.sqlext
 SQLEXT_OBJS = sqlext.o vercomp.o
@@ -23,8 +22,8 @@ all:: ${STLIB_NAME} ${SQLEXT_NAME}
 	${CC} -c -DUSE_TCL_STUBS ${CFLAGS} ${CPPFLAGS} ${SHLIB_CFLAGS} $< -o $@
 
 ${STLIB_NAME}:: ${OBJS}
-	${STLIB_LD} ${STLIB_NAME} ${OBJS}
-	${RANLIB} ${STLIB_NAME}
+	@STLIB_LD@ ${STLIB_NAME} ${OBJS}
+	@RANLIB@ ${STLIB_NAME}
 
 ${SQLEXT_NAME}: ${SQLEXT_OBJS}
 	${SHLIB_LD} ${SQLEXT_OBJS} -o $@
diff --git a/src/macports1.0/Makefile.in b/src/macports1.0/Makefile.in
index 4af97003e7e9d74458435158e3eb1f766510fd6c..a3349f48cbe6f40c912237b89a63e218ba168e9c 100644
--- a/src/macports1.0/Makefile.in
+++ b/src/macports1.0/Makefile.in
@@ -2,6 +2,9 @@ srcdir = @srcdir@
 VPATH  = @srcdir@
 
 include ../../Mk/macports.autoconf.mk
+# for some reason, Tcl's pkgIndex.tcl generation crashes in the MacPorts.so
+# init procedure when built with LTO (on Linux at least).
+CFLAGS += -fno-lto
 
 SRCS=		macports.tcl macports_dlist.tcl macports_util.tcl \
 		macports_autoconf.tcl diagnose.tcl reclaim.tcl selfupdate.tcl
