diff --git src/registry2.0/portimage.tcl src/registry2.0/portimage.tcl
index b576719c03c932fa9476ded1810c318e84e416f1..7d7a5afaed1beeef6aea9ffb21ae57fe2ae7eebb 100644
--- src/registry2.0/portimage.tcl
+++ src/registry2.0/portimage.tcl
@@ -93,6 +93,7 @@ proc activate {name {version ""} {revision ""} {variants 0} {optionslist ""}} {
         set registry_open yes
     }
     set todeactivate [list]
+    set actaction "Activating"
 
     registry::read {
 
@@ -121,7 +122,12 @@ proc activate {name {version ""} {revision ""} {variants 0} {optionslist ""}} {
             return -code error "Image error: Can't find image file $location"
         }
         if {[$requested state] eq "installed"} {
-            return -code error "Image error: ${name} @${specifier} is already active."
+            if {$force} {
+                set actaction "Reactivating"
+                lappend todeactivate $requested
+            } else {
+                return -code error "Image error: ${name} @${specifier} is already active."
+            }
         }
     }
     foreach a $todeactivate {
@@ -130,7 +136,7 @@ proc activate {name {version ""} {revision ""} {variants 0} {optionslist ""}} {
         }
     }
 
-    ui_msg "$UI_PREFIX [format [msgcat::mc "Activating %s @%s"] $name $specifier]"
+    ui_msg "$UI_PREFIX [format [msgcat::mc "$actaction %s @%s"] $name $specifier]"
 
     _activate_contents $requested
 }

diff --git src/port/port.tcl src/port/port.tcl
index 3320fba6917010c351e7316ae91560c3b7ec77d9..2f99da7c6c528fb448304af289ba9b31e81323e9 100644
--- src/port/port.tcl
+++ src/port/port.tcl
@@ -2485,9 +2506,11 @@ proc action_activate { action portlist opts } {
     }
     foreachport $portlist {
         set composite_version [composite_version $portversion [array get variations]]
+        array set actoptions [array get options]
+        set force [expr [info exists actoptions(ports_force)] && [string is true -strict $actoptions(ports_force)]]
         if {![info exists options(ports_activate_no-exec)]
             && ![catch {set ilist [registry::installed $portname $composite_version]}]
-            && [llength $ilist] == 1} {
+            && ([llength $ilist] == 1 && !$force)} {
 
             set i [lindex $ilist 0]
             set regref [registry::entry open $portname [lindex $i 1] [lindex $i 2] [lindex $i 3] [lindex $i 5]]

