diff --git a/src/port1.0/portconfigure.tcl b/src/port1.0/portconfigure.tcl
index 8591c211119dc72260e03ed538d4a56dd7469e6e..70672e393cd46eb422406b78643966428e566554 100644
--- a/src/port1.0/portconfigure.tcl
+++ b/src/port1.0/portconfigure.tcl
@@ -308,15 +315,17 @@ proc portconfigure::configure_start {args} {
     global configure.ccache ccache_dir ccache_size macportsuser
     if {${configure.ccache}} {
         # Create ccache directory with correct permissions with root privileges
-        elevateToRoot "configure ccache"
-        if {[catch {
-                file mkdir ${ccache_dir}
-                file attributes ${ccache_dir} -owner ${macportsuser} -permissions 0755
-            } result]} {
-            ui_warn "ccache_dir ${ccache_dir} could not be created; disabling ccache: $result"
-            set configure.ccache no
+        if {![file exists ${ccache_dir}] || [file type ${ccache_dir}] ne "directory"} {
+            elevateToRoot "configure ccache"
+            if {[catch {
+                    file mkdir ${ccache_dir}
+                    file attributes ${ccache_dir} -owner ${macportsuser} -permissions 0755
+                } result]} {
+                ui_warn "ccache_dir ${ccache_dir} could not be created; disabling ccache: $result"
+                set configure.ccache no
+            }
+            dropPrivileges
         }
-        dropPrivileges
 
         # Initialize ccache directory with the given maximum size
         if {${configure.ccache}} {
