

Index: MacPorts-svn/src/pextlib1.0/Pextlib.c
===================================================================
--- src/pextlib1.0/Pextlib.c	(revision 140947)
+++ src/pextlib1.0/Pextlib.c	(working copy)
@@ -68,6 +68,8 @@
 #ifdef __MACH__
 #include <mach-o/loader.h>
 #include <mach-o/fat.h>
+#elif defined(linux)
+#include <elf.h>
 #endif
 
 #include <tcl.h>
@@ -483,7 +485,6 @@
     return TCL_OK;
 }
 
-#ifdef __MACH__
 /**
  * Tcl function to determine whether a file given by path is binary (in terms of being Mach-O)
  * Defined on Mac-Systems only, because the necessary headers are only available there.
@@ -534,6 +535,7 @@
         fclose(file);
         return TCL_ERROR;
     }
+#ifdef __MACH__
     if (magic == MH_MAGIC || magic == MH_MAGIC_64) {
         fclose(file);
         /* this is a mach-o file */
@@ -573,12 +575,23 @@
         Tcl_SetObjResult(interp, Tcl_NewBooleanObj(false));
         return TCL_OK;
     }
+#elif defined(linux)
+    {
+        char mstring[4];
+        memcpy( &mstring[0], htonl(magic), 4 );
+        if (strncmp( mstring, ELFMAG, 4 ) == 0) {
+            fclose(file);
+            /* this is an ELF object */
+            Tcl_SetObjResult(interp, Tcl_NewBooleanObj(true));
+            return TCL_OK;
+        }
+    }
+#endif
     fclose(file);
 
     Tcl_SetObjResult(interp, Tcl_NewBooleanObj(false));
     return TCL_OK;
 }
-#endif
 
 /* Check if the configured DNS server(s) incorrectly return a result for
    a nonexistent hostname. Returns true if broken, false if OK. */
@@ -641,9 +654,7 @@
 	Tcl_CreateObjCommand(interp, "unsetenv", UnsetEnvCmd, NULL, NULL);
 	Tcl_CreateObjCommand(interp, "lchown", lchownCmd, NULL, NULL);
 	Tcl_CreateObjCommand(interp, "realpath", RealpathCmd, NULL, NULL);
-#ifdef __MACH__
     Tcl_CreateObjCommand(interp, "fileIsBinary", fileIsBinaryCmd, NULL, NULL);
-#endif
 
     Tcl_CreateObjCommand(interp, "readline", ReadlineCmd, NULL, NULL);
     Tcl_CreateObjCommand(interp, "rl_history", RLHistoryCmd, NULL, NULL);
