diff --git src/port/port.tcl src/port/port.tcl
index 5312e5a..cd543f7 100644
--- src/port/port.tcl
+++ src/port/port.tcl
@@ -4433,6 +4463,8 @@ proc parse_options { action ui_options_name global_options_name } {
     upvar $global_options_name global_options
     global cmdname cmd_opts_array
 
+    # RJVB
+    set global_options(ports_ignore_different) yes
     while {[moreargs]} {
         set arg [lookahead]
 

