diff --git a/Makefile b/Makefile
index 54049cd..8f59582 100644
--- a/Makefile
+++ b/Makefile
@@ -23,8 +23,10 @@
 # Change this if you want to install somewhere else. In particular
 # you may wish to remove the middle "local/" part of the path.
 
-INSTALL_PLUGINS_DIR	=	/usr/local/lib/ladspa/
-INSTALL_LRDF_DIR	=	/usr/local/share/ladspa/rdf/
+DESTDIR             =
+PREFIX              =    /usr/local
+INSTALL_PLUGINS_DIR	=	${DESTDIR}${PREFIX}/lib/ladspa/
+INSTALL_LRDF_DIR	=	${DESTDIR}${PREFIX}/share/ladspa/rdf/
 
 # NO EDITING below this line is required
 # if all you want to do is install and use the plugins.
@@ -37,7 +39,8 @@ INSTALL_LRDF_DIR	=	/usr/local/share/ladspa/rdf/
 OS := $(shell uname -s)
 
 CC      = gcc
-CFLAGS  = -I. -O3 -Wall -fomit-frame-pointer -funroll-loops -ffast-math -c -fPIC -DPIC
+OPTFLAGS = -O3
+CFLAGS  = -I. $(OPTFLAGS) -Wall -fomit-frame-pointer -funroll-loops -ffast-math -c -fPIC -DPIC
 ifeq ($(OS),Darwin)
 LDFLAGS = -nostartfiles -shared -Wl,-install_name,symbolic -lc -lm
 else
@@ -63,10 +66,10 @@ tap_dynamics_st.o: tap_dynamics_presets.h
 
 install: all
 	-mkdir -p          $(INSTALL_PLUGINS_DIR)
-	cp *.so            $(INSTALL_PLUGINS_DIR)
+	cp -pv *.so        $(INSTALL_PLUGINS_DIR)
 	-mkdir -p          $(INSTALL_LRDF_DIR)
-	cp tap-plugins.rdf $(INSTALL_LRDF_DIR)
-	cp tap_reverb.rdf  $(INSTALL_LRDF_DIR)
+	cp -pv tap-plugins.rdf $(INSTALL_LRDF_DIR)
+	cp -pv tap_reverb.rdf  $(INSTALL_LRDF_DIR)
 
 clean:
 	-rm -f *.so *.o *~
