diff --git a/libs/ui/KisApplication.cpp b/libs/ui/KisApplication.cpp
index 3078548..d766132 100644
--- a/libs/ui/KisApplication.cpp
+++ b/libs/ui/KisApplication.cpp
@@ -714,6 +714,8 @@ KisApplication::~KisApplication()
     if (!isRunning()) {
         KisResourceCacheDb::deleteTemporaryResources();
     }
+    // RJVB
+    KisIconUtils::clearIconCache();
 }
 
 void KisApplication::setSplashScreen(QWidget *splashScreen)
diff --git a/libs/widgetutils/kis_icon_utils.cpp b/libs/widgetutils/kis_icon_utils.cpp
index a3b87cb..1d5e2f7 100644
--- a/libs/widgetutils/kis_icon_utils.cpp
+++ b/libs/widgetutils/kis_icon_utils.cpp
@@ -164,6 +164,8 @@ void updateIconCommon(QObject *object)
 
 void clearIconCache() {
         s_cache.clear();
+	   // RJVB
+	   s_icons.clear();
 }
 
 void updateIcon(QAbstractButton *button)
