diff --git a/krita/main.cc b/krita/main.cc
index 3016772..4d90632 100644
--- a/krita/main.cc
+++ b/krita/main.cc
@@ -230,7 +230,11 @@ extern "C" MAIN_EXPORT int MAIN_FN(int argc, char **argv)
 {
     // The global initialization of the random generator
     qsrand(time(0));
+#if !defined(Q_OS_OSX) || defined(APPLE_STANDALONE_BUNDLE)
     bool runningInKDE = !qgetenv("KDE_FULL_SESSION").isEmpty();
+#else
+    bool runningInKDE = true;
+#endif
 
 #if defined HAVE_X11
     qputenv("QT_QPA_PLATFORM", "xcb");
@@ -385,8 +389,10 @@ extern "C" MAIN_EXPORT int MAIN_FN(int argc, char **argv)
             qputenv("GST_PLUGIN_SCANNER", appimageMountDir + QFile::encodeName("/usr/lib/gstreamer-1.0/gst-plugin-scanner"));
         }
     }
-#else
+#elif !defined(Q_OS_OSX) || defined(APPLE_STANDALONE_BUNDLE)
     qputenv("XDG_DATA_DIRS", QFile::encodeName(QDir(root + "share").absolutePath()));
+#else
+    qputenv("XDG_DATA_DIRS", QFile::encodeName(QStandardPaths::standardLocations(QStandardPaths::GenericDataLocation).join(QLatin1Char(':'))));
 #endif
 
     dbgKrita << "Setting XDG_DATA_DIRS" << qgetenv("XDG_DATA_DIRS");
diff --git a/libs/resources/KoResourcePaths.cpp b/libs/resources/KoResourcePaths.cpp
index b555d9d..b6e3010 100644
--- a/libs/resources/KoResourcePaths.cpp
+++ b/libs/resources/KoResourcePaths.cpp
@@ -14,6 +14,7 @@
 #include <QFileInfo>
 #include <QDebug>
 #include <QCoreApplication>
+#include <QLibraryInfo>
 #include <QMutex>
 #include "kis_debug.h"
 #include "ksharedconfig.h"
@@ -77,7 +78,7 @@ static const Qt::CaseSensitivity cs = Qt::CaseSensitive;
 #endif
 
 QString getInstallationPrefix() {
-#ifdef Q_OS_MACOS
+#ifdef APPLE_STANDALONE_BUNDLE
     QString appPath = qApp->applicationDirPath();
 
     dbgResources << "1" << appPath;
@@ -106,7 +107,8 @@ QString getInstallationPrefix() {
     // any files other than libraries
     return QStandardPaths::writableLocation(QStandardPaths::AppDataLocation) + "/";
 #else
-    return qApp->applicationDirPath() + "/../";
+//     return qApp->applicationDirPath() + "/../";
+    return QLibraryInfo::location(QLibraryInfo::PrefixPath);
 #endif
 }
 
@@ -388,7 +390,7 @@ QStringList KoResourcePaths::findDirsInternal(const QString &type)
                 QStandardPaths::locateAll(d->mapTypeToQStandardPaths(type), alias + '/', QStandardPaths::LocateDirectory);
         appendResources(&dirs, aliasDirs, true);
 
-#ifdef Q_OS_MACOS
+#ifdef APPLE_STANDALONE_BUNDLE
         dbgResources << "MAC:" << getApplicationRoot();
         QStringList bundlePaths;
         bundlePaths << getApplicationRoot() + "/share/krita/" + alias;
diff --git a/libs/ui/thememanager.cpp b/libs/ui/thememanager.cpp
index 067b476..0e9b30d 100644
--- a/libs/ui/thememanager.cpp
+++ b/libs/ui/thememanager.cpp
@@ -179,7 +179,7 @@ void ThemeManager::slotChangePalette()
     qApp->setProperty("KDE_COLOR_SCHEME_PATH", filename);
     qApp->setPalette(palette);
 
-#ifdef Q_OS_MACOS
+#ifdef APPLE_STANDALONE_BUNDLE
     if (theme == "Krita bright" || theme.isEmpty()) {
         qApp->setStyle("Macintosh");
         qApp->style()->polish(qApp);
