diff --git a/krita/main.cc b/krita/main.cc
index 3d68822..ea378c7 100644
--- a/krita/main.cc
+++ b/krita/main.cc
@@ -31,6 +31,7 @@
 #include <QLoggingCategory>
 #include <QLocale>
 #include <QSettings>
+#include <QStandardPaths>
 
 #include <KisApplication.h>
 #include <KoConfig.h>
@@ -71,7 +72,11 @@ extern "C" int main(int argc, char **argv)
     qputenv("QT_XCB_NO_EVENT_COMPRESSION", "1");
 #endif
 
+#if !defined(Q_OS_OSX) || defined(APPLE_STANDALONE_BUNDLE)
     bool runningInKDE = !qgetenv("KDE_FULL_SESSION").isEmpty();
+#else
+    bool runningInKDE = true;
+#endif
 
     /**
      * Disable debug output by default. (krita.input enables tablet debugging.)
@@ -107,8 +112,10 @@ extern "C" int main(int argc, char **argv)
 
 #ifdef Q_OS_LINUX
     qputenv("XDG_DATA_DIRS", QFile::encodeName(KoResourcePaths::getApplicationRoot() + "share") + ":" + qgetenv("XDG_DATA_DIRS"));
-#else
+#elif !defined(Q_OS_OSX) || defined(APPLE_STANDALONE_BUNDLE)
     qputenv("XDG_DATA_DIRS", QFile::encodeName(KoResourcePaths::getApplicationRoot() + "share"));
+#else
+    qputenv("XDG_DATA_DIRS", QFile::encodeName(QStandardPaths::standardLocations(QStandardPaths::GenericDataLocation).join(QLatin1Char(':'))));
 #endif
 
     qDebug() << "Setting XDG_DATA_DIRS" << qgetenv("XDG_DATA_DIRS");
diff --git a/libs/ui/thememanager.cpp b/libs/ui/thememanager.cpp
index 2917171..7430ff3 100644
--- a/libs/ui/thememanager.cpp
+++ b/libs/ui/thememanager.cpp
@@ -188,7 +188,7 @@ void ThemeManager::slotChangePalette()
     //qDebug() << ">>>>>>>>>>>>>>>>>> going to set palette on app" << theme;
     qApp->setPalette(palette);
 
-#ifdef Q_OS_MAC
+#ifdef APPLE_STANDALONE_BUNDLE
     if (theme == "Krita bright" || theme.isEmpty()) {
         qApp->setStyle("Macintosh");
         qApp->style()->polish(qApp);
diff --git a/libs/widgetutils/KoResourcePaths.cpp b/libs/widgetutils/KoResourcePaths.cpp
index 7b5b5e9..dde3311 100644
--- a/libs/widgetutils/KoResourcePaths.cpp
+++ b/libs/widgetutils/KoResourcePaths.cpp
@@ -26,6 +26,7 @@
 #include <QFileInfo>
 #include <QDebug>
 #include <QApplication>
+#include <QLibraryInfo>
 #include <QMutex>
 #include "kis_debug.h"
 
@@ -87,7 +88,7 @@ static const Qt::CaseSensitivity cs = Qt::CaseSensitive;
 #endif
 
 QString getInstallationPrefix() {
-#ifdef Q_OS_MAC
+#ifdef APPLE_STANDALONE_BUNDLE
      QString appPath = qApp->applicationDirPath();
 
      debugWidgetUtils << "1" << appPath;
@@ -116,7 +117,8 @@ QString getInstallationPrefix() {
      debugWidgetUtils << ">>>>>>>>>>>" << bundlePath;
      return bundlePath;
  #else
-     return qApp->applicationDirPath() + "/../";
+//      return qApp->applicationDirPath() + "/../";
+     return QLibraryInfo::location(QLibraryInfo::PrefixPath);
  #endif
 }
 
@@ -343,7 +345,7 @@ QStringList KoResourcePaths::findDirsInternal(const QString &type)
             QStandardPaths::locateAll(d->mapTypeToQStandardPaths(type), alias + '/', QStandardPaths::LocateDirectory);
         appendResources(&dirs, aliasDirs, true);
 
-#ifdef Q_OS_MAC
+#ifdef APPLE_STANDALONE_BUNDLE
         debugWidgetUtils << "MAC:" << getApplicationRoot();
         QStringList bundlePaths;
         bundlePaths << getApplicationRoot() + "/share/krita/" + alias;
