--- a/plugins/impex/raw/3rdparty/libkdcraw/src/kdcraw_p.h	2016-09-27 12:30:38.000000000 +0200
+++ b/plugins/impex/raw/3rdparty/libkdcraw/src/kdcraw_p.h	2017-05-10 01:23:08.000000000 +0200
@@ -31,7 +31,7 @@
 #include <QByteArray>
 
 // Pragma directives to reduce warnings from LibRaw header files.
-#if not defined(__APPLE__) && defined(__GNUC__)
+#if !defined(__APPLE__) && defined(__GNUC__)
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wdeprecated-declarations"
 #endif
@@ -46,7 +46,7 @@
 #include <libraw.h>
 
 // Restore warnings
-#if not defined(__APPLE__) && defined(__GNUC__)
+#if !defined(__APPLE__) && defined(__GNUC__)
 #pragma GCC diagnostic pop
 #endif
 
