--- libs/widgetutils/KoResourcePaths.cpp.orig	2016-09-27 12:30:38.000000000 +0200
+++ libs/widgetutils/KoResourcePaths.cpp	2016-11-05 22:55:25.000000000 +0100
@@ -87,7 +87,7 @@
 #endif
 
 QString getInstallationPrefix() {
-#ifdef Q_OS_MAC
+#ifdef APPLE_STANDALONE_BUNDLE
      QString appPath = qApp->applicationDirPath();
 
      debugWidgetUtils << "1" << appPath;
@@ -343,7 +343,7 @@
             QStandardPaths::locateAll(d->mapTypeToQStandardPaths(type), alias + '/', QStandardPaths::LocateDirectory);
         appendResources(&dirs, aliasDirs, true);
 
-#ifdef Q_OS_MAC
+#ifdef APPLE_STANDALONE_BUNDLE
         debugWidgetUtils << "MAC:" << getApplicationRoot();
         QStringList bundlePaths;
         bundlePaths << getApplicationRoot() + "/share/krita/" + alias;
--- libs/ui/thememanager.cpp.orig	2016-09-27 12:30:38.000000000 +0200
+++ libs/ui/thememanager.cpp	2016-11-05 23:01:24.000000000 +0100
@@ -188,7 +188,7 @@
     //qDebug() << ">>>>>>>>>>>>>>>>>> going to set palette on app" << theme;
     qApp->setPalette(palette);
 
-#ifdef Q_OS_MAC
+#ifdef APPLE_STANDALONE_BUNDLE
     if (theme == "Krita bright" || theme.isEmpty()) {
         qApp->setStyle("Macintosh");
         qApp->style()->polish(qApp);
