diff --git a/CMakeLists.txt b/CMakeLists.txt
index ef8be26..61e0d3f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -271,9 +271,7 @@ macro_ensure_out_of_source_build("Compiling Krita inside the source directory is
 
 # Note: OPTIONAL_COMPONENTS does not seem to be reliable
 # (as of ECM 5.15.0, CMake 3.2)
-if (NOT WIN32 AND NOT APPLE)
-
-    find_package(Qt5 ${MIN_QT_VERSION} REQUIRED X11Extras)
+if (NOT WIN32)
 
     find_package(Qt5DBus ${MIN_QT_VERSION} QUIET)
     set(HAVE_DBUS ${Qt5DBus_FOUND})
@@ -287,20 +285,24 @@ if (NOT WIN32 AND NOT APPLE)
     macro_bool_to_01(KF5Crash_FOUND HAVE_KCRASH)
     macro_log_feature(${KF5Crash_FOUND} "kcrash" "KDE's Crash Handler" "http://api.kde.org/frameworks-api/frameworks5-apidocs/kcrash/html/index.html" FALSE "" "Optionally used to provide crash reporting on Linux")
 
-    find_package(X11)
-    if(X11_FOUND)
-        find_package(Qt5 ${MIN_QT_VERSION} REQUIRED NO_MODULE COMPONENTS X11Extras)
-        set(HAVE_X11 TRUE)
-        add_definitions(-DHAVE_X11)
-    else()
-        set(HAVE_X11 FALSE)
-    endif()
+    if (NOT APPLE)
+        find_package(Qt5 ${MIN_QT_VERSION} REQUIRED X11Extras)
 
-    find_package(XCB COMPONENTS XCB ATOM)
-    if(XCB_FOUND)
-        set(HAVE_XCB TRUE)
-    else()
-        set(HAVE_XCB FALSE)
+        find_package(X11)
+        if(X11_FOUND)
+            find_package(Qt5 ${MIN_QT_VERSION} REQUIRED NO_MODULE COMPONENTS X11Extras)
+            set(HAVE_X11 TRUE)
+            add_definitions(-DHAVE_X11)
+        else()
+            set(HAVE_X11 FALSE)
+        endif()
+
+        find_package(XCB COMPONENTS XCB ATOM)
+        if(XCB_FOUND)
+            set(HAVE_XCB TRUE)
+        else()
+            set(HAVE_XCB FALSE)
+        endif()
     endif()
 else()
     set(HAVE_DBUS FALSE)
