diff --git a/libs/ui/input/kis_extended_modifiers_mapper_osx.mm b/libs/ui/input/kis_extended_modifiers_mapper_osx.mm
index b959f5a..7a2fc6c 100644
--- a/libs/ui/input/kis_extended_modifiers_mapper_osx.mm
+++ b/libs/ui/input/kis_extended_modifiers_mapper_osx.mm
@@ -9,10 +9,33 @@
 #import <QDebug>
 #import <QFlag>
 #import <qmutex.h>
+#import <Availability.h>
+#ifndef __MAC_NA
+#define __MAC_NA 9999
+#endif
 #import <AppKit/AppKit.h>
 
 #include "krita_container_utils.h"
 
+#if !QT_MACOS_PLATFORM_SDK_EQUAL_OR_ABOVE(__MAC_10_12)
+#define NSEventTypeKeyDown NSKeyDown
+#define NSEventTypeKeyUp NSKeyUp
+#define NSEventTypeFlagsChanged NSFlagsChanged
+#define NSEventTypeMouseExited NSMouseExited
+#define NSEventMaskFlagsChanged NSFlagsChangedMask
+#define NSEventMaskKeyDown NSKeyDownMask
+#define NSEventMaskKeyUp NSKeyUpMask
+#define NSEventMaskLeftMouseUp NSLeftMouseUpMask
+#define NSEventMaskLeftMouseDown NSLeftMouseDownMask
+#define NSEventMaskMouseEntered NSMouseEnteredMask
+#define NSEventMaskMouseExited NSMouseExitedMask
+typedef NSUInteger NSEventModifierFlags;
+#define NSEventModifierFlagShift NSShiftKeyMask
+#define NSEventModifierFlagCommand NSCommandKeyMask
+#define NSEventModifierFlagControl NSControlKeyMask
+#define NSEventModifierFlagOption NSAlternateKeyMask
+#endif
+
 class ExtendedModifiers
 {
 public:
diff --git a/krita/Info.plist.template b/krita/Info.plist.template
index 6e3f09d..d1511c3 100644
--- a/krita/Info.plist.template
+++ b/krita/Info.plist.template
@@ -37,7 +37,7 @@
     <key>LSRequiresCarbon</key>
     <true/>
     <key>LSMinimumSystemVersion</key>
-    <string>10.13</string>
+    <string>10.9</string>
     <key>NSSupportsAutomaticGraphicsSwitching</key>
     <true/>
     <key>NSHumanReadableCopyright</key>
