diff --git a/libs/ui/widgets/kis_color_space_selector.cc b/libs/ui/widgets/kis_color_space_selector.cc
index a63e7ae..167ce8a 100644
--- a/libs/ui/widgets/kis_color_space_selector.cc
+++ b/libs/ui/widgets/kis_color_space_selector.cc
@@ -197,6 +197,9 @@ void KisColorSpaceSelector::setCurrentProfile(const QString& name)
 
 void KisColorSpaceSelector::setCurrentColorSpace(const KoColorSpace* colorSpace)
 {
+  if (!colorSpace) {
+      return;
+  }
   setCurrentColorModel(colorSpace->colorModelId());
   setCurrentColorDepth(colorSpace->colorDepthId());
   setCurrentProfile(colorSpace->profile()->name());

