--- app/orig.CMakeLists.txt	2014-08-16 21:12:15.000000000 +0200
+++ app/CMakeLists.txt	2014-08-16 21:35:13.000000000 +0200
@@ -12,7 +12,7 @@
     add_definitions(-DWITH_WELCOMEPAGE)
 	qt4_add_resources(kdevelop_SRCS welcomepage/welcomepage.qrc)
 endif(KDECLARATIVE_FOUND)
-kde4_add_app_icon(kdevelop_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/../pics/hi*-app-kdevelop.png")
+#kde4_add_app_icon(kdevelop_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/../pics/hi*-app-kdevelop.png")
 
 kde4_add_executable(kdevelop ${kdevelop_SRCS} )
 
@@ -37,7 +37,7 @@
 
 install(FILES kdevelop! DESTINATION bin PERMISSIONS OWNER_EXECUTE OWNER_WRITE OWNER_READ GROUP_EXECUTE GROUP_READ WORLD_EXECUTE WORLD_READ)
 
-kde4_add_app_icon(kdevelop_bin_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/../pics/hi*-app-kdevelop.png")
+#kde4_add_app_icon(kdevelop_bin_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/../pics/hi*-app-kdevelop.png")
 
 if (Q_WS_MAC)
 	#sets MacOSX bundle properties for all targets in this cmake file
