diff --git kstyle/demo/oxygendemodialog.cpp kstyle/demo/oxygendemodialog.cpp
index 4a742e3..5005d5f 100644
--- kstyle/demo/oxygendemodialog.cpp
+++ kstyle/demo/oxygendemodialog.cpp
@@ -75,7 +75,9 @@ namespace Oxygen
         buttonBox->addButton( _enableCheckBox, QDialogButtonBox::ResetRole );
 
         _rightToLeftCheckBox = new QCheckBox( i18n( "Right to left layout" ) );
+        _rightToLeftCheckBox->setChecked( qApp->isRightToLeft() );
         connect( _rightToLeftCheckBox, SIGNAL(toggled(bool)), SLOT(toggleRightToLeft(bool)) );
+        connect( qApp, SIGNAL(layoutDirectionChanged(Qt::LayoutDirection)), SLOT(layoutDirectionChanged(Qt::LayoutDirection)) );
         buttonBox->addButton( _rightToLeftCheckBox, QDialogButtonBox::ResetRole );
 
         // connections
@@ -218,6 +220,10 @@ namespace Oxygen
     { qApp->setLayoutDirection( value ? Qt::RightToLeft:Qt::LeftToRight ); }
 
     //_______________________________________________________________
+    void DemoDialog::layoutDirectionChanged( Qt::LayoutDirection direction )
+    { _rightToLeftCheckBox->setChecked( direction == Qt::RightToLeft ); }
+
+    //_______________________________________________________________
     void DemoDialog::closeEvent( QCloseEvent* event )
     {
         emit abortSimulations();
diff --git kstyle/demo/oxygendemodialog.h kstyle/demo/oxygendemodialog.h
index b296ec7..9f0015d 100644
--- kstyle/demo/oxygendemodialog.h
+++ kstyle/demo/oxygendemodialog.h
@@ -75,6 +75,7 @@ namespace Oxygen
 
         //! toggle RightToLeft
         virtual void toggleRightToLeft( bool );
+        virtual void layoutDirectionChanged( Qt::LayoutDirection );
 
         protected:
 
diff --git kstyle/demo/oxygenmdidemowidget.cpp kstyle/demo/oxygenmdidemowidget.cpp
index 3e5422e..bfd1fb9 100644
--- kstyle/demo/oxygenmdidemowidget.cpp
+++ kstyle/demo/oxygenmdidemowidget.cpp
@@ -40,6 +40,8 @@ namespace Oxygen
     {
         setLayout( new QVBoxLayout() );
         QMenuBar* menuBar = new QMenuBar( this );
+        // let the menubar appear in the MDI window on all platforms
+        menuBar->setNativeMenuBar( false );
         layout()->addWidget( menuBar );
 
         QWidget* widget = new QWidget( this );
@@ -47,12 +49,28 @@ namespace Oxygen
         ui.setupUi( widget );
 
         QMenu* menu = menuBar->addMenu( i18n( "Layout" ) );
-        connect( menu->addAction( i18n( "Tile" ) ), SIGNAL(triggered()), this, SLOT(setLayoutTiled()) );
-        connect( menu->addAction( i18n( "Cascade" ) ), SIGNAL(triggered()), this, SLOT(setLayoutCascade()) );
-        connect( menu->addAction( i18n( "Tabs" ) ), SIGNAL(triggered()), this, SLOT(setLayoutTabbed()) );
+        QActionGroup *aGroup = new QActionGroup( menu );
+        QAction *action = menu->addAction( i18n( "Tile" ) );
+        action->setCheckable( true );
+        aGroup->addAction( action );
+        connect( action, SIGNAL(triggered()), this, SLOT(setLayoutTiled()) );
+        action = menu->addAction( i18n( "Cascade" ) );
+        action->setCheckable( true );
+        aGroup->addAction( action );
+        connect( action, SIGNAL(triggered()), this, SLOT(setLayoutCascade()) );
+        action = menu->addAction( i18n( "Tabs" ) );
+        action->setCheckable( true );
+        aGroup->addAction( action );
+        connect( action, SIGNAL(triggered()), this, SLOT(setLayoutTabbed()) );
+
+        menu->addSeparator();
+
+        _rightToLeftLayout = menu->addAction( i18n( "Right to left layout" ) );
+        _rightToLeftLayout->setCheckable( true );
+        connect( _rightToLeftLayout, SIGNAL(triggered()), this, SLOT(toggleDirection()) );
+        connect( qApp, SIGNAL(layoutDirectionChanged(Qt::LayoutDirection)), SLOT(layoutDirectionChanged(Qt::LayoutDirection)) );
 
         menu = menuBar->addMenu( i18n( "Tools" ) );
-        QAction* action;
         connect( action = menu->addAction( QIcon::fromTheme( QStringLiteral( "arrow-right" ) ), i18n( "Select Next Window" ) ), SIGNAL(triggered()), ui.mdiArea, SLOT(activateNextSubWindow()) );
         action->setShortcut( Qt::CTRL + Qt::Key_Tab );
         addAction( action );
@@ -82,6 +100,14 @@ namespace Oxygen
     { ui.mdiArea->setViewMode( QMdiArea::TabbedView ); }
 
     //______________________________________________________________
+    void MdiDemoWidget::toggleDirection( void )
+    { qApp->setLayoutDirection( qApp->isLeftToRight() ? Qt::RightToLeft : Qt::LeftToRight ); }
+
+    //_______________________________________________________________
+    void MdiDemoWidget::layoutDirectionChanged( Qt::LayoutDirection direction )
+    { _rightToLeftLayout->setChecked( direction == Qt::RightToLeft ); }
+
+    //______________________________________________________________
     void MdiDemoWidget::benchmark( void )
     {
         if( !isVisible() ) return;
diff --git kstyle/demo/oxygenmdidemowidget.h kstyle/demo/oxygenmdidemowidget.h
index eafbeee..7147f49 100644
--- kstyle/demo/oxygenmdidemowidget.h
+++ kstyle/demo/oxygenmdidemowidget.h
@@ -54,12 +54,15 @@ namespace Oxygen
         void setLayoutTiled( void );
         void setLayoutCascade( void );
         void setLayoutTabbed( void );
+        void toggleDirection( void );
+        void layoutDirectionChanged( Qt::LayoutDirection );
 
         void benchmark( void );
 
         private:
 
         Ui_MdiDemoWidget ui;
+	   QAction *_rightToLeftLayout;
 
     };
 
