--- kstyle/config/orig.main.cpp	2015-12-15 11:28:21.000000000 +0100
+++ kstyle/config/main.cpp	2015-12-21 22:52:12.000000000 +0100
@@ -38,7 +38,7 @@
 
     QApplication app( argc, argv );
     app.setApplicationName( i18n( "Oxygen Settings" ) );
-    app.setWindowIcon( QIcon::fromTheme( QStringLiteral( "oxygen-settings" ) ) );
+    app.setWindowIcon( QIcon::fromTheme( QStringLiteral( "oxygen-settings" ), app.windowIcon() ) );
 
     KCMultiDialog dialog;
     dialog.setWindowTitle( i18n( "Oxygen Settings" ) );
--- kstyle/demo/orig.main.cpp	2015-12-15 11:28:21.000000000 +0100
+++ kstyle/demo/main.cpp	2015-12-21 22:52:20.000000000 +0100
@@ -56,7 +56,7 @@
         #endif
 
         app.setApplicationName( i18n( "Oxygen Demo" ) );
-        app.setWindowIcon( QIcon::fromTheme( QStringLiteral( "oxygen" ) ) );
+        app.setWindowIcon( QIcon::fromTheme( QStringLiteral( "oxygen" ), app.windowIcon() ) );
         DemoDialog dialog;
         dialog.show();
         bool result = app.exec();
--- kstyle/config/orig.CMakeLists.txt	2015-12-15 11:28:21.000000000 +0100
+++ kstyle/config/CMakeLists.txt	2015-12-21 22:51:45.000000000 +0100
@@ -56,6 +56,8 @@
 ########### next target ###############
 if(NOT OXYGEN_USE_KDE4)
 
+  include(ECMAddAppIcon)
+
   find_package(KF5KCMUtils CONFIG)
   set_package_properties(KF5KCMUtils PROPERTIES
       DESCRIPTION "KF5 KCM utils Framework"
@@ -66,6 +68,8 @@
   if( KF5KCMUtils_FOUND )
 
       set(oxygen_settings_SOURCES main.cpp)
+	 file(GLOB ICONS_SRCS "${CMAKE_CURRENT_BINARY_DIR}/icons/*oxygen.png")
+	 ecm_add_app_icon(oxygen_settings_SOURCES ICONS ${ICONS_SRCS})
       add_executable(oxygen-settings5 ${oxygen_settings_SOURCES} )
       target_link_libraries(oxygen-settings5 Qt5::Core Qt5::Gui Qt5::Widgets )
       target_link_libraries(oxygen-settings5 KF5::I18n KF5::KCMUtils)
