--- qt-everywhere-opensource-src-5.5.0/qtbase/src/corelib/io/orig.qstandardpaths_unix.cpp	2015-06-29 22:05:01.000000000 +0200
+++ qt-everywhere-opensource-src-5.5.0/qtbase/src/corelib/io/qstandardpaths_unix.cpp	2015-11-05 19:48:04.774516076 +0100
@@ -255,7 +255,7 @@
     // http://standards.freedesktop.org/basedir-spec/latest/
     QString xdgDataDirsEnv = QFile::decodeName(qgetenv("XDG_DATA_DIRS"));
     if (xdgDataDirsEnv.isEmpty()) {
-        dirs.append(QString::fromLatin1("/usr/local/share"));
+        dirs.append(QString::fromLatin1("@PREFIX@/share"));
         dirs.append(QString::fromLatin1("/usr/share"));
     } else {
         dirs = xdgDataDirsEnv.split(QLatin1Char(':'), QString::SkipEmptyParts);
@@ -287,6 +287,7 @@
     // http://standards.freedesktop.org/basedir-spec/latest/
     const QString xdgConfigDirs = QFile::decodeName(qgetenv("XDG_CONFIG_DIRS"));
     if (xdgConfigDirs.isEmpty())
+        dirs.append(QString::fromLatin1("@PREFIX@/etc/xdg"));
         dirs.append(QString::fromLatin1("/etc/xdg"));
     else
         dirs = xdgConfigDirs.split(QLatin1Char(':'));
