--- a/qtbase/src/corelib/io/io.pri
+++ b/qtbase/src/corelib/io/io.pri
@@ -32,7 +32,9 @@
         io/qresource_p.h \
         io/qresource_iterator_p.h \
         io/qsavefile.h \
+        io/qextstandardpaths.h \
         io/qstandardpaths.h \
+        io/qstandardpaths_p.h \
         io/qstorageinfo.h \
         io/qstorageinfo_p.h \
         io/qurl.h \
--- a/qtbase/src/corelib/io/qstandardpaths.cpp
+++ b/qtbase/src/corelib/io/qstandardpaths.cpp
@@ -31,8 +31,10 @@
 **
 ****************************************************************************/
 
+#undef QT_USE_EXTSTANDARDPATHS
 #include "qstandardpaths.h"
 
+#include <qglobal.h>
 #include <qdir.h>
 #include <qfileinfo.h>
 
@@ -399,6 +401,85 @@
     return QFileInfo(path).isFile();
 }
 
+#if defined(Q_OS_OSX)
+//#include <QDebug>
+/*!
+    This function enables or disables XDG locations on platforms where Qt can be configured
+    to use this kind of locations instead of locations following OS guidelines.
+    Currently this concerns only Mac OS X //and Cygwin?//
+ */
+void QStandardPaths::setXDGLocationsEnabled(bool xdgMode)
+{
+    usingXDGLocations = xdgMode;
+    isSetXDGLocations = true;
+    //qDebug() << Q_FUNC_INFO << "Setting usingXDGLocations=" << xdgMode;
+}
+
+/*!
+    This function returns true when XDG locations are used, unless QT_STANDARDPATHS_CANNOT_SWITCH
+    is set, in which case switching is disabled and the function always returns false.
+ */
+bool QStandardPaths::isXDGLocationsEnabled()
+{
+    if (qEnvironmentVariableIsSet("QT_STANDARDPATHS_CANNOT_SWITCH"))
+        return false;
+    else
+        return usingXDGLocations;
+}
+
+QString QStandardPaths::locate(StandardLocation type, bool xdgMode, const QString &fileName, LocateOptions options)
+{
+    const QStringList &dirs = standardLocations(type, xdgMode);
+    for (QStringList::const_iterator dir = dirs.constBegin(); dir != dirs.constEnd(); ++dir) {
+        const QString path = *dir + QLatin1Char('/') + fileName;
+        if (existsAsSpecified(path, options))
+            return path;
+    }
+    return QString();
+}
+
+QStringList QStandardPaths::locateAll(StandardLocation type, bool xdgMode, const QString &fileName, LocateOptions options)
+{
+    const QStringList &dirs = standardLocations(type, xdgMode);
+    QStringList result;
+    for (QStringList::const_iterator dir = dirs.constBegin(); dir != dirs.constEnd(); ++dir) {
+        const QString path = *dir + QLatin1Char('/') + fileName;
+        if (existsAsSpecified(path, options))
+            result.append(path);
+    }
+    //qDebug() << Q_FUNC_INFO << "search list=" << dirs;
+    return result;
+}
+
+/*!
+   Tries to find a file or directory called \a fileName in the standard locations
+   for \a type.
+
+   The full path to the first file or directory (depending on \a options) found is returned.
+   If no such file or directory can be found, an empty string is returned.
+ */
+QString QStandardPaths::locate(StandardLocation type, const QString &fileName, LocateOptions options)
+{
+    return QStandardPaths::locate(type, isXDGLocationsEnabled(), fileName, options);
+}
+
+/*!
+   Tries to find all files or directories called \a fileName in the standard locations
+   for \a type.
+
+   The \a options flag allows to specify whether to look for files or directories.
+
+   Returns the list of all the files that were found.
+ */
+QStringList QStandardPaths::locateAll(StandardLocation type, const QString &fileName, LocateOptions options)
+{
+    QStringList ret = QStandardPaths::locateAll(type, isXDGLocationsEnabled(), fileName, options);
+    //qDebug() << Q_FUNC_INFO << "type=" << type << "name=" << fileName << "found" << ret;
+    return ret;
+}
+
+#else // !Q_OS_OSX
+
 /*!
    Tries to find a file or directory called \a fileName in the standard locations
    for \a type.
@@ -436,6 +517,7 @@
     }
     return result;
 }
+#endif // Q_OS_OSX
 
 #ifdef Q_OS_WIN
 static QStringList executableExtensions()
--- a/qtbase/src/corelib/io/qstandardpaths.h
+++ b/qtbase/src/corelib/io/qstandardpaths.h
@@ -41,6 +41,8 @@ QT_BEGIN_NAMESPACE
 
 #ifndef QT_NO_STANDARDPATHS
 
+class Q_CORE_EXPORT QStandardPathsConfiguration;
+
 class Q_CORE_EXPORT QStandardPaths
 {
 public:
@@ -68,6 +70,10 @@ public:
         AppLocalDataLocation = DataLocation
     };
 
+#if defined(Q_OS_OSX)
+    static bool isXDGLocationsEnabled();
+#endif // Q_OS_OSX
+
     static QString writableLocation(StandardLocation type);
     static QStringList standardLocations(StandardLocation type);
 
@@ -91,14 +97,97 @@ public:
     static void setTestModeEnabled(bool testMode);
     static bool isTestModeEnabled();
 
+#if defined(Q_OS_OSX)
+protected:
+    static void setXDGLocationsEnabled(bool xdgMode);
+
+    static QString writableLocation(StandardLocation type, bool xdgMode);
+    static QStringList standardLocations(StandardLocation type, bool xdgMode);
+
+    static QString locate(StandardLocation type, bool xdgMode, const QString &fileName, LocateOptions options = LocateFile);
+    static QStringList locateAll(StandardLocation type, bool xdgMode, const QString &fileName, LocateOptions options = LocateFile);
+#endif // Q_OS_OSX
+
 private:
     // prevent construction
     QStandardPaths();
     ~QStandardPaths();
+#if defined(Q_OS_OSX)
+    static bool usingXDGLocations, isSetXDGLocations;
+
+    friend class QStandardPathsConfiguration;
+    friend class QExtStandardPaths;
+#endif // Q_OS_OSX
+};
+
+/* **** Extended QStandardPaths **** */
+
+#ifdef QT_EXTSTANDARDPATHS_XDG_DEFAULT
+#if QT_EXTSTANDARDPATHS_XDG_DEFAULT == runtime
+#define QSPDEFAULTXDGMODE  QStandardPaths::isXDGLocationsEnabled()
+#else
+#define QSPDEFAULTXDGMODE  QT_EXTSTANDARDPATHS_XDG_DEFAULT
+#endif // "runtime"
+#else
+#define QSPDEFAULTXDGMODE  false
+#endif
+
+/*!
+    \class QExtStandardPaths
+    \inmodule QtCore
+    \brief The QExtStandardPaths class provides configurable methods for accessing standard paths.
+
+    This class inherits and elaborates on \class QStandardPaths, providing access to the support for
+    native vs. XDG-compliant standard locations that QStandardPaths has on certain platforms (currently
+    only Mac OS X).
+    When the QT_USE_EXTSTANDARDPATHS macro is defined, this class will replace QStandardPaths, and
+    in that case the QT_EXTSTANDARDPATHS_XDG_DEFAULT macro will define the behaviour of code that does
+    not use QExtStandardPaths explicitly itself. When undefined or QT_EXTSTANDARDPATHS_XDG_DEFAULT=false,
+    QExtStandardPaths will use native locations, even if QStandardPaths::isXDGLocationsEnabled() returns true.
+    When QT_EXTSTANDARDPATHS_XDG_DEFAULT=runtime, behaviour is controlled at runtime through
+    QStandardPaths::setXDGLocationsEnabled() and QStandardPaths::isXDGLocationsEnabled().
+    In all other cases QStandardPaths will use XDG-compliant locations.
+*/
+class Q_CORE_EXPORT QExtStandardPaths : public QStandardPaths
+{
+public:
+#if defined(Q_OS_OSX)
+    static bool isXDGLocationsEnabled()
+    {
+        return QSPDEFAULTXDGMODE;
+    }
+    static QString writableLocation(StandardLocation type, bool xdgMode=QSPDEFAULTXDGMODE)
+    {
+        return QStandardPaths::writableLocation(type, xdgMode);
+    }
+    static QStringList standardLocations(StandardLocation type, bool xdgMode=QSPDEFAULTXDGMODE)
+    {
+        return QStandardPaths::standardLocations(type, xdgMode);
+    }
+
+    static QString locate(StandardLocation type, const QString &fileName, LocateOptions options = LocateFile, bool xdgMode=QSPDEFAULTXDGMODE)
+    {
+        return QStandardPaths::locate(type, xdgMode, fileName, options);
+    }
+    static QStringList locateAll(StandardLocation type, const QString &fileName, LocateOptions options = LocateFile, bool xdgMode=QSPDEFAULTXDGMODE)
+    {
+        return QStandardPaths::locateAll(type, xdgMode, fileName, options);
+    }
+#endif
+
+private:
+    // prevent construction
+    QExtStandardPaths();
+    ~QExtStandardPaths();
 };
 
 #endif // QT_NO_STANDARDPATHS
 
 QT_END_NAMESPACE
 
+#ifdef QT_USE_EXTSTANDARDPATHS
+#define QStandardPaths  QExtStandardPaths
+#endif // QT_USE_EXTSTANDARDPATHS
+
+
 #endif // QSTANDARDPATHS_H
--- a/qtbase/src/corelib/io/qstandardpaths_mac.mm
+++ b/qtbase/src/corelib/io/qstandardpaths_mac.mm
@@ -31,13 +31,16 @@
 **
 ****************************************************************************/
 
+#undef QT_USE_EXTSTANDARDPATHS
 #include "qstandardpaths.h"
 #include <qdir.h>
 #include <qurl.h>
 #include <private/qcore_mac_p.h>
+#include <private/qfilesystemengine_p.h>
 
 #ifndef QT_BOOTSTRAPPED
 #include <qcoreapplication.h>
+#include <qlibraryinfo.h>
 #endif
 
 #include <CoreFoundation/CoreFoundation.h>
@@ -136,8 +139,155 @@
     return path;
 }
 
-QString QStandardPaths::writableLocation(StandardLocation type)
+bool QStandardPaths::usingXDGLocations = false;
+bool QStandardPaths::isSetXDGLocations = false;
+
+static void normaliseDirs(QStringList &dirs)
+{
+    // Normalise paths, skip relative paths
+    QMutableListIterator<QString> it(dirs);
+    while (it.hasNext()) {
+        const QString dir = it.next();
+        if (!dir.startsWith(QLatin1Char('/')))
+            it.remove();
+        else
+            it.setValue(QDir::cleanPath(dir));
+    }
+
+    // Remove duplicates from the list, there's no use for duplicated
+    // paths in XDG_CONFIG_DIRS - if it's not found in the given
+    // directory the first time, it won't be there the second time.
+    // Plus duplicate paths causes problems for example for mimetypes,
+    // where duplicate paths here lead to duplicated mime types returned
+    // for a file, eg "text/plain,text/plain" instead of "text/plain"
+    dirs.removeDuplicates();
+}
+
+static QStringList xdgCacheDirs()
+{
+    QStringList dirs;
+    // http://standards.freedesktop.org/basedir-spec/latest/
+    QString xdgConfigDirsEnv = QFile::decodeName(qgetenv("XDG_CACHE_HOME"));
+    if (xdgConfigDirsEnv.isEmpty()) {
+#ifndef QT_BOOTSTRAPPED
+        dirs.append(QDir::homePath() + QString::fromLatin1("/.cache"));
+#endif
+    } else {
+        dirs = xdgConfigDirsEnv.split(QLatin1Char(':'), QString::SkipEmptyParts);
+
+        normaliseDirs(dirs);
+    }
+    return dirs;
+}
+
+static QStringList xdgConfigDirs()
+{
+    QStringList dirs;
+    // http://standards.freedesktop.org/basedir-spec/latest/
+    QString xdgConfigDirsEnv = QFile::decodeName(qgetenv("XDG_CONFIG_DIRS"));
+    if (xdgConfigDirsEnv.isEmpty()) {
+#ifndef QT_BOOTSTRAPPED
+        dirs.append(QLibraryInfo::location(QLibraryInfo::PrefixPath) + QString::fromLatin1("/etc/xdg"));
+#endif
+    } else {
+        dirs = xdgConfigDirsEnv.split(QLatin1Char(':'), QString::SkipEmptyParts);
+
+        normaliseDirs(dirs);
+    }
+    return dirs;
+}
+
+static QStringList xdgDataDirs()
+{
+    QStringList dirs;
+    // http://standards.freedesktop.org/basedir-spec/latest/
+    QString xdgDataDirsEnv = QFile::decodeName(qgetenv("XDG_DATA_DIRS"));
+    if (xdgDataDirsEnv.isEmpty()) {
+#ifndef QT_BOOTSTRAPPED
+        dirs.append(QLibraryInfo::location(QLibraryInfo::PrefixPath) + QString::fromLatin1("/share"));
+#endif
+    } else {
+        dirs = xdgDataDirsEnv.split(QLatin1Char(':'), QString::SkipEmptyParts);
+
+        normaliseDirs(dirs);
+    }
+    return dirs;
+}
+
+static QString xdgRuntimeDir()
 {
+    const uid_t myUid = geteuid();
+    // http://standards.freedesktop.org/basedir-spec/latest/
+    QString xdgRTDir = QFile::decodeName(qgetenv("XDG_RUNTIME_DIR"));
+    if (xdgRTDir.isEmpty()) {
+        const QString userName = QFileSystemEngine::resolveUserName(myUid);
+//         xdgRTDir = QDir::tempPath() + QLatin1String("/runtime-") + userName;
+        // NSTemporaryDirectory() returns the default $TMPDIR value, regardless of its current setting,
+        // which is more in line with XDG_RUNTIME_DIR requirements.
+        xdgRTDir = QString::fromNSString(NSTemporaryDirectory()) + QLatin1String("runtime-") + userName;
+        QDir dir(xdgRTDir);
+        if (!dir.exists()) {
+            if (!QDir().mkdir(xdgRTDir)) {
+                qWarning("QStandardPaths: error creating runtime directory %s: %s",
+                         qPrintable(xdgRTDir), qPrintable(qt_error_string(errno)));
+                return QString();
+            }
+        }
+    } else {
+        qWarning("QStandardPaths: XDG_RUNTIME_DIR is set, using '%s'", qPrintable(xdgRTDir));
+    }
+    // "The directory MUST be owned by the user"
+    QFileInfo fileInfo(xdgRTDir);
+    if (fileInfo.ownerId() != myUid) {
+        qWarning("QStandardPaths: wrong ownership on runtime directory %s, %d instead of %d", qPrintable(xdgRTDir),
+                 fileInfo.ownerId(), myUid);
+        return QString();
+    }
+    // "and he MUST be the only one having read and write access to it. Its Unix access mode MUST be 0700."
+    // since the current user is the owner, set both xxxUser and xxxOwner
+    QFile file(xdgRTDir);
+    const QFile::Permissions wantedPerms = QFile::ReadUser | QFile::WriteUser | QFile::ExeUser
+                                           | QFile::ReadOwner | QFile::WriteOwner | QFile::ExeOwner;
+    if (file.permissions() != wantedPerms && !file.setPermissions(wantedPerms)) {
+        qWarning("QStandardPaths: could not set correct permissions on runtime directory %s: %s",
+                 qPrintable(xdgRTDir), qPrintable(file.errorString()));
+        return QString();
+    }
+    return xdgRTDir;
+}
+
+QString QStandardPaths::writableLocation(StandardLocation type, bool xdgMode)
+{
+    if (xdgMode) {
+        const QString prefix = (isTestModeEnabled())? QDir::homePath() + QLatin1String("/.qttest") : QDir::homePath();
+        QString path;
+        switch (type) {
+        case GenericDataLocation:
+        case AppDataLocation:
+        case AppLocalDataLocation:
+            path = prefix + QLatin1String("/.local/share");
+            if (type != GenericDataLocation)
+                appendOrganizationAndApp(path);
+            return path;
+        case GenericCacheLocation:
+        case CacheLocation:
+            path = prefix + QLatin1String("/.cache");
+            if (type == CacheLocation)
+                appendOrganizationAndApp(path);
+            return path;
+        case GenericConfigLocation:
+        case ConfigLocation:
+            return prefix + QLatin1String("/.config");
+        case ApplicationsLocation:
+            path = writableLocation(GenericDataLocation, xdgMode) + QLatin1String("/applications");
+            return path;
+        case RuntimeLocation:
+            return xdgRuntimeDir();
+        default:
+            break;
+        }
+    }
+
     if (isTestModeEnabled()) {
         const QString qttestDir = QDir::homePath() + QLatin1String("/.qttest");
         QString path;
@@ -162,6 +312,9 @@
             if (type == AppConfigLocation)
                 appendOrganizationAndApp(path);
             return path;
+        case ApplicationsLocation:
+            path = qttestDir + QLatin1String("/Applications");
+            return path;
         default:
             break;
         }
@@ -178,16 +331,45 @@
     case GenericCacheLocation:
     case CacheLocation:
     case RuntimeLocation:
+    case FontsLocation:
+        // the font location that is writable for all users is ~/Library/Fonts
         return macLocation(type, kUserDomain);
     default:
         return macLocation(type, kOnAppropriateDisk);
     }
 }
 
-QStringList QStandardPaths::standardLocations(StandardLocation type)
+QString QStandardPaths::writableLocation(StandardLocation type)
+{
+    return QStandardPaths::writableLocation(type, isXDGLocationsEnabled());
+}
+
+QStringList QStandardPaths::standardLocations(StandardLocation type, bool xdgMode)
 {
     QStringList dirs;
 
+    if (xdgMode) {
+        switch (type) {
+        case GenericDataLocation:
+            dirs.append(xdgDataDirs());
+            break;
+        case GenericConfigLocation:
+        case ConfigLocation:
+            dirs.append(xdgConfigDirs());
+            break;
+        case GenericCacheLocation:
+        case CacheLocation:
+            dirs.append(xdgCacheDirs());
+            break;
+        case ApplicationsLocation:
+            QStringList xdgDirs = xdgDataDirs();
+            for (int i = 0; i < xdgDirs.count(); ++i)
+                xdgDirs[i].append(QLatin1String("/applications"));
+            dirs.append(xdgDirs);
+            break;
+        }
+    }
+
     if (type == GenericDataLocation || type == AppDataLocation || type == AppLocalDataLocation || type == GenericCacheLocation || type == CacheLocation) {
         const QString path = macLocation(type, kOnAppropriateDisk);
         if (!path.isEmpty())
@@ -195,6 +377,14 @@
     }
 
     if (type == AppDataLocation || type == AppLocalDataLocation) {
+        if (xdgMode) {
+            QStringList xdgDirs = xdgDataDirs();
+            for (int i = 0; i < xdgDirs.count(); ++i) {
+                appendOrganizationAndApp(xdgDirs[i]);
+            }
+            dirs.append(xdgDirs);
+        }
+
         CFBundleRef mainBundle = CFBundleGetMainBundle();
         if (mainBundle) {
             CFURLRef bundleUrl = CFBundleCopyBundleURL(mainBundle);
@@ -218,11 +408,22 @@
                 dirs.append(bundlePath + resourcesPath);
         }
     }
-    const QString localDir = writableLocation(type);
+    if (type == FontsLocation) {
+        // /Library/Fonts
+        dirs.append(macLocation(type,kLocalDomain));
+        // /System/Library/Fonts
+        dirs.append(macLocation(type,kSystemDomain));
+    }
+    const QString localDir = writableLocation(type, xdgMode);
     dirs.prepend(localDir);
     return dirs;
 }
 
+QStringList QStandardPaths::standardLocations(StandardLocation type)
+{
+    return QStandardPaths::standardLocations(type, isXDGLocationsEnabled());
+}
+
 #ifndef QT_BOOTSTRAPPED
 QString QStandardPaths::displayName(StandardLocation type)
 {
--- /dev/null
+++ b/qtbase/src/corelib/io/qstandardpaths_p.h
@@ -0,0 +1,71 @@
+/****************************************************************************
+**
+** Copyright (C) 2014 Digia Plc and/or its subsidiary(-ies).
+** Contact: http://www.qt-project.org/legal
+**
+** This file is part of the QtCore module of the Qt Toolkit.
+**
+** $QT_BEGIN_LICENSE:LGPL21$
+** Commercial License Usage
+** Licensees holding valid commercial Qt licenses may use this file in
+** accordance with the commercial license agreement provided with the
+** Software or, alternatively, in accordance with the terms contained in
+** a written agreement between you and Digia. For licensing terms and
+** conditions see http://qt.digia.com/licensing. For further information
+** use the contact form at http://qt.digia.com/contact-us.
+**
+** GNU Lesser General Public License Usage
+** Alternatively, this file may be used under the terms of the GNU Lesser
+** General Public License version 2.1 or version 3 as published by the Free
+** Software Foundation and appearing in the file LICENSE.LGPLv21 and
+** LICENSE.LGPLv3 included in the packaging of this file. Please review the
+** following information to ensure the GNU Lesser General Public License
+** requirements will be met: https://www.gnu.org/licenses/lgpl.html and
+** http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html.
+**
+** In addition, as a special exception, Digia gives you certain additional
+** rights. These rights are described in the Digia Qt LGPL Exception
+** version 1.1, included in the file LGPL_EXCEPTION.txt in this package.
+**
+** $QT_END_LICENSE$
+**
+****************************************************************************/
+
+#ifndef QSTANDARDPATHS_P_H
+#define QSTANDARDPATHS_P_H
+
+#include <QtCore/QtGlobal>
+
+#if defined(Q_OS_OSX)
+
+#include <QtCore/qstandardpaths.h>
+
+QT_BEGIN_NAMESPACE
+
+
+#ifndef QT_NO_STANDARDPATHS
+
+class Q_CORE_EXPORT QStandardPathsConfiguration
+{
+public:
+    QStandardPathsConfiguration(const bool useXDGLocations)
+        : usingXDGLocations(useXDGLocations)
+    {
+        QStandardPaths::setXDGLocationsEnabled(useXDGLocations);
+    }
+    inline bool usesXDGLocations() const
+    {
+        return usingXDGLocations;
+    }
+protected:
+    const bool usingXDGLocations;
+};
+
+bool Q_CORE_EXPORT setXDGLocationsEnabled(bool useXDGLocations);
+
+#endif // QT_NO_STANDARDPATHS
+
+QT_END_NAMESPACE
+
+#endif // Q_OS_OSX
+#endif // QSTANDARDPATHS_P_H
--- /dev/null
+++ b/qtbase/src/corelib/io/qextstandardpaths.h
@@ -0,0 +1,40 @@
+/****************************************************************************
+**
+** Copyright (C) 2016 René J.V. Bertin
+** Contact: http://www.qt.io/licensing/
+**
+** This file is part of the QtCore module of the Qt Toolkit.
+**
+** $QT_BEGIN_LICENSE:LGPL21$
+** Commercial License Usage
+** Licensees holding valid commercial Qt licenses may use this file in
+** accordance with the commercial license agreement provided with the
+** Software or, alternatively, in accordance with the terms contained in
+** a written agreement between you and The Qt Company. For licensing terms
+** and conditions see http://www.qt.io/terms-conditions. For further
+** information use the contact form at http://www.qt.io/contact-us.
+**
+** GNU Lesser General Public License Usage
+** Alternatively, this file may be used under the terms of the GNU Lesser
+** General Public License version 2.1 or version 3 as published by the Free
+** Software Foundation and appearing in the file LICENSE.LGPLv21 and
+** LICENSE.LGPLv3 included in the packaging of this file. Please review the
+** following information to ensure the GNU Lesser General Public License
+** requirements will be met: https://www.gnu.org/licenses/lgpl.html and
+** http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html.
+**
+** As a special exception, The Qt Company gives you certain additional
+** rights. These rights are described in The Qt Company LGPL Exception
+** version 1.1, included in the file LGPL_EXCEPTION.txt in this package.
+**
+** $QT_END_LICENSE$
+**
+****************************************************************************/
+
+#ifndef QEXTSTANDARDPATHS_H
+#define QEXTSTANDARDPATHS_H
+
+#define QT_USE_EXTSTANDARDPATHS
+#include <QtCore/qstandardpaths.h>
+
+#endif // QEXTSTANDARDPATHS_H
--- /dev/null
+++ b/qtbase/include/QtCore/qstandardpaths_p.h
@@ -0,0 +1 @@
+#include "../../src/corelib/io/qstandardpaths_p.h"
--- /dev/null
+++ b/qtbase/include/QtCore/QExtStandardPaths
@@ -0,0 +1 @@
+#include "qextstandardpaths.h"
--- /dev/null
+++ b/qtbase/mkspecs/features/mac/data/qstandardpaths_use_xdg.cpp
@@ -0,0 +1,55 @@
+/****************************************************************************
+**
+** Copyright (C) 2015 RJV Bertin
+** More appropriate legalese here.
+**
+** This file is part of the QtCore module of the Qt Toolkit.
+**
+** $QT_BEGIN_LICENSE:LGPL21$
+** Commercial License Usage
+** Licensees holding valid commercial Qt licenses may use this file in
+** accordance with the commercial license agreement provided with the
+** Software or, alternatively, in accordance with the terms contained in
+** a written agreement between you and Digia. For licensing terms and
+** conditions see http://qt.digia.com/licensing. For further information
+** use the contact form at http://qt.digia.com/contact-us.
+**
+** GNU Lesser General Public License Usage
+** Alternatively, this file may be used under the terms of the GNU Lesser
+** General Public License version 2.1 or version 3 as published by the Free
+** Software Foundation and appearing in the file LICENSE.LGPLv21 and
+** LICENSE.LGPLv3 included in the packaging of this file. Please review the
+** following information to ensure the GNU Lesser General Public License
+** requirements will be met: https://www.gnu.org/licenses/lgpl.html and
+** http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html.
+**
+** In addition, as a special exception, Digia gives you certain additional
+** rights. These rights are described in the Digia Qt LGPL Exception
+** version 1.1, included in the file LGPL_EXCEPTION.txt in this package.
+**
+** $QT_END_LICENSE$
+**
+****************************************************************************/
+
+#include <QtCore/QtGlobal>
+#if defined(Q_OS_OSX)
+
+#include <qstandardpaths_p.h>
+
+#ifndef QT_NO_STANDARDPATHS
+
+const QString qspUsesXDGLocationsTag(QLatin1String("$Id: @(#) " __FILE__ " - " __DATE__ " $"));
+
+// defining a static instance works for now, but see
+// https://wiki.qt.io/Coding_Conventions#Compiler.2FPlatform_specific_issues
+static QStandardPathsConfiguration qspUsesXDGLocations(true);
+
+bool setXDGLocationsEnabled(bool useXDGLocations)
+{
+    QStandardPathsConfiguration dummy(useXDGLocations);
+    // this should force even clever linkers to preserve qspUsesXDGLocations
+    return qspUsesXDGLocations.usesXDGLocations() != useXDGLocations;
+}
+
+#endif //Q_NO_STANDARDPATHS
+#endif //Q_OS_OSX
\ No newline at end of file
--- /dev/null
+++ b/qtbase/mkspecs/features/mac/qsp_xdg.prf
@@ -0,0 +1,2 @@
+qsp_use_xdg: DEFINES += QSP_USE_XDG
+SOURCES += $$PWD/data/qstandardpaths_use_xdg.cpp
\ No newline at end of file
--- /dev/null
+++ b/qtbase/src/qsp_xdg/qsp_xdg.pro
@@ -0,0 +1,9 @@
+TARGET     = QtQspXDG
+QT = core
+
+QMAKE_LIBS += $$QMAKE_LIBS_QSP_XDG
+
+HEADERS =
+SOURCES = qstandardpaths_use_xdg.cpp
+
+load(qt_module)
\ No newline at end of file
--- /dev/null
+++ b/qtbase/src/qsp_xdg/qstandardpaths_use_xdg.cpp
@@ -0,0 +1,53 @@
+/****************************************************************************
+**
+** Copyright (C) 2015 RJV Bertin
+** More appropriate legalese here.
+**
+** This file is part of the QtCore module of the Qt Toolkit.
+**
+** $QT_BEGIN_LICENSE:LGPL21$
+** Commercial License Usage
+** Licensees holding valid commercial Qt licenses may use this file in
+** accordance with the commercial license agreement provided with the
+** Software or, alternatively, in accordance with the terms contained in
+** a written agreement between you and Digia. For licensing terms and
+** conditions see http://qt.digia.com/licensing. For further information
+** use the contact form at http://qt.digia.com/contact-us.
+**
+** GNU Lesser General Public License Usage
+** Alternatively, this file may be used under the terms of the GNU Lesser
+** General Public License version 2.1 or version 3 as published by the Free
+** Software Foundation and appearing in the file LICENSE.LGPLv21 and
+** LICENSE.LGPLv3 included in the packaging of this file. Please review the
+** following information to ensure the GNU Lesser General Public License
+** requirements will be met: https://www.gnu.org/licenses/lgpl.html and
+** http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html.
+**
+** In addition, as a special exception, Digia gives you certain additional
+** rights. These rights are described in the Digia Qt LGPL Exception
+** version 1.1, included in the file LGPL_EXCEPTION.txt in this package.
+**
+** $QT_END_LICENSE$
+**
+****************************************************************************/
+
+#include <QtCore/QtGlobal>
+#if defined(Q_OS_OSX)
+
+#include <qstandardpaths_p.h>
+
+#ifndef QT_NO_STANDARDPATHS
+
+const QString qspUsesXDGLocationsTag(QLatin1String("$Id: @(#) " __FILE__ " - " __DATE__ " $"));
+
+static QStandardPathsConfiguration qspUsesXDGLocations(true);
+
+bool setXDGLocationsEnabled(bool useXDGLocations)
+{
+    QStandardPathsConfiguration dummy(useXDGLocations);
+    // this should force even clever linkers to preserve qspUsesXDGLocations
+    return qspUsesXDGLocations.usesXDGLocations() != useXDGLocations;
+}
+
+#endif //Q_NO_STANDARDPATHS
+#endif //Q_OS_OSX
\ No newline at end of file
--- a/qtbase/src/src.pro
+++ b/qtbase/src/src.pro
@@ -122,6 +122,10 @@
 src_printsupport.target = sub-printsupport
 src_printsupport.depends = src_corelib src_gui src_widgets src_tools_uic
 
+src_qsp_xdg.subdir = $$PWD/qsp_xdg
+src_qsp_xdg.target = sub-qsp_xdg
+src_qsp_xdg.depends = src_corelib
+
 src_plugins.subdir = $$PWD/plugins
 src_plugins.target = sub-plugins
 src_plugins.depends = src_sql src_xml src_network
@@ -182,6 +186,8 @@
 }
 SUBDIRS += src_plugins
 
+mac : SUBDIRS += src_qsp_xdg
+
 nacl: SUBDIRS -= src_network src_testlib
 
 android:!android-no-sdk: SUBDIRS += src_android
