--- a/qtbase/src/gui/kernel/qguiapplication.cpp
+++ b/qtbase/src/gui/kernel/qguiapplication.cpp
@@ -1270,6 +1270,10 @@
         argc = j;
     }
 
+    // https://bugreports.qt-project.org/browse/QTBUG-32974
+    if (qgetenv("QT_LOAD_TESTABILITY").toInt() > 0)
+        loadTestability = true;
+
     // Load environment exported generic plugins
     QByteArray envPlugins = qgetenv("QT_QPA_GENERIC_PLUGINS");
     if (!envPlugins.isEmpty()) {
