--- qt5kdweb-mp9-work/qtwebengine-opensource-src-5.7.0/src/3rdparty/chromium/base/mac/sdk_forward_declarations.h.orig	2016-05-26 15:53:47.000000000 +0200
+++ qt5kdweb-mp9-work/qtwebengine-opensource-src-5.7.0/src/3rdparty/chromium/base/mac/sdk_forward_declarations.h	2016-11-14 16:29:27.827664713 +0100
@@ -15,6 +15,7 @@
 #import <CoreWLAN/CoreWLAN.h>
 #import <ImageCaptureCore/ImageCaptureCore.h>
 #import <IOBluetooth/IOBluetooth.h>
+#import <CoreBluetooth/CoreBluetooth.h>
 #include <stdint.h>
 
 #include "base/base_export.h"
