--- a/qtbase/configure	2015-09-25 21:20:38.000000000 +0200
+++ b/qtbase/configure	2015-09-25 21:44:16.150436412 +0200
@@ -3562,6 +3562,11 @@
 [ "$OPT_VERBOSE" = "yes" ] && tty=/dev/stderr
 eval "`LC_ALL=C $TEST_COMPILER $SYSROOT_FLAG $TEST_COMPILER_CXXFLAGS -xc++ -E -v - < /dev/null 2>&1 > /dev/null | $AWK "$awkprog" | tee $tty`"
 unset tty
+# remove unnecessary and unwanted paths (for MacPorts we don't want /usr/local/include either)
+DEFAULT_INCDIRS=`echo ${DEFAULT_INCDIRS} | sed -e 's|/usr/include/[xi][0-9][^-]*-linux-gnu[^ ]*||g' -e 's|/usr/lib/gcc/[xi][0-9][^-]*-linux-gnu/[^ ]*||g' -e 's|/usr/include/c++[^ ]*||g' -e 's|/usr/local/include||g'`
+DEFAULT_LIBDIRS=`echo ${DEFAULT_LIBDIRS} | sed -e 's|/usr/lib/[xi][0-9][^-]*-linux-gnu||g' -e 's|/lib/[xi][0-9][^-]*-linux-gnu||g' -e 's|/usr/lib/gcc/[xi][0-9][^-]*-linux-gnu/[0-9][^ ]*||g'`
+echo "Pruned DEFAULT_INCDIRS=$DEFAULT_INCDIRS"
+echo "Pruned DEFAULT_LIBDIRS=$DEFAULT_LIBDIRS"
 
 #setup the build parts
 if [ -z "$CFG_BUILD_PARTS" ]; then
