--- a/qtbase/src/gui/text/qtexthtmlparser.cpp
+++ b/qtbase/src/gui/text/qtexthtmlparser.cpp
@@ -1926,6 +1926,30 @@
         decl.d->propertyId = QCss::FontFamily;
         QVector<QCss::Value> values;
         val.type = QCss::Value::String;
+        val.variant = QFontDatabase::systemFont(QFontDatabase::FixedFont).family();
+        values << val;
+        val.type = QCss::Value::TermOperatorComma;
+        val.variant = QVariant();
+        values << val;
+        val.type = QCss::Value::String;
+        val.variant = QLatin1String("Monaco");
+        values << val;
+        val.type = QCss::Value::TermOperatorComma;
+        val.variant = QVariant();
+        values << val;
+        val.type = QCss::Value::String;
+        val.variant = QLatin1String("Lucida Console");
+        values << val;
+        val.type = QCss::Value::TermOperatorComma;
+        val.variant = QVariant();
+        values << val;
+        val.type = QCss::Value::String;
+        val.variant = QLatin1String("Consolas");
+        values << val;
+        val.type = QCss::Value::TermOperatorComma;
+        val.variant = QVariant();
+        values << val;
+        val.type = QCss::Value::String;
         val.variant = QLatin1String("Courier New");
         values << val;
         val.type = QCss::Value::TermOperatorComma;
