--- a/qtbase/src/platformsupport/themes/genericunix/qgenericunixthemes.cpp
+++ b/qtbase/src/platformsupport/themes/genericunix/qgenericunixthemes.cpp
@@ -678,7 +678,11 @@
 #endif
     if (name == QLatin1String(QGnomeTheme::name))
         return new QGnomeTheme;
+#ifdef Q_OS_OSX
+    return NULL;
+#else
     return new QGenericUnixTheme;
+#endif
 }
 
 QStringList QGenericUnixTheme::themeNames()
--- a/qtbase/src/platformsupport/themes/themes.pri
+++ b/qtbase/src/platformsupport/themes/themes.pri
@@ -1,3 +1,3 @@
-unix:!mac {
+unix {
     include($$PWD/genericunix/genericunix.pri)
 }
--- a/qtbase/src/plugins/platforms/cocoa/qcocoaintegration.mm
+++ b/qtbase/src/plugins/platforms/cocoa/qcocoaintegration.mm
@@ -53,6 +53,8 @@
 
 #include <IOKit/graphics/IOGraphicsLib.h>
 
+#include <private/qgenericunixthemes_p.h>
+
 static void initResources()
 {
     Q_INIT_RESOURCE(qcocoaresources);
@@ -494,9 +496,20 @@
 
 QPlatformTheme *QCocoaIntegration::createPlatformTheme(const QString &name) const
 {
-    if (name == QLatin1String(QCocoaTheme::name))
+    qDebug() << "QCocoaIntegration::createPlatformTheme(" << name << ")";
+    if (name == QLatin1String(QCocoaTheme::name)) {
+        //qDebug() << "returning the QCocoaTheme";
         return new QCocoaTheme;
-    return QPlatformIntegration::createPlatformTheme(name);
+    }
+    QPlatformTheme *unixTheme = QGenericUnixTheme::createUnixTheme(name);
+    if (unixTheme) {
+        qDebug() << "returning" << unixTheme;
+        return unixTheme;
+    }
+    else {
+        qDebug() << "returning QPlatformIntegration::createPlatformTheme(" << name << ")";
+        return QPlatformIntegration::createPlatformTheme(name);
+    }
 }
 
 QCocoaServices *QCocoaIntegration::services() const
