--- a/qtbase/src/corelib/kernel/qobject.cpp
+++ b/qtbase/src/corelib/kernel/qobject.cpp
@@ -3916,7 +3916,7 @@
 
 static void dumpRecursive(int level, QObject *object)
 {
-#if defined(QT_DEBUG)
+// #if defined(QT_DEBUG)
     if (object) {
         QByteArray buf;
         buf.fill(' ', level / 2 * 8);
@@ -3945,10 +3945,10 @@
                 dumpRecursive(level+1, children.at(i));
         }
     }
-#else
-    Q_UNUSED(level)
-        Q_UNUSED(object)
-#endif
+// #else
+//     Q_UNUSED(level)
+//         Q_UNUSED(object)
+// #endif
 }
 
 /*!
@@ -3979,7 +3979,7 @@
 
 void QObject::dumpObjectInfo()
 {
-#if defined(QT_DEBUG)
+// #if defined(QT_DEBUG)
     qDebug("OBJECT %s::%s", metaObject()->className(),
            objectName().isEmpty() ? "unnamed" : objectName().toLocal8Bit().data());
 
@@ -4039,7 +4039,7 @@
     } else {
         qDebug("        <None>");
     }
-#endif
+// #endif
 }
 
 #ifndef QT_NO_USERDATA
