--- a/qttools/src/qtpaths/qtpaths.cpp
+++ b/qttools/src/qtpaths/qtpaths.cpp
@@ -206,8 +208,13 @@
                                  QCoreApplication::translate("qtpaths", "Location of Qt plugins."));
     parser.addOption(plugindir);
 
+    QCommandLineOption qspalt(QStringList() << QStringLiteral("altmode") << QStringLiteral("alt-mode"),
+                                 QCoreApplication::translate("qtpaths", "Activate ALT mode."));
+    parser.addOption(qspalt);
+
     parser.process(app);
 
+    QStandardPaths::setALTLocationsEnabled(parser.isSet(qspalt));
     QStandardPaths::enableTestMode(parser.isSet(testmode));
 
     QStringList results;
