diff --git a/qtbase/src/plugins/platforms/cocoa/qcocoatheme.mm b/qtbase/src/plugins/platforms/cocoa/qcocoatheme.mm
index 11749e1..37d15ab 100644
--- a/qtbase/src/plugins/platforms/cocoa/qcocoatheme.mm
+++ b/qtbase/src/plugins/platforms/cocoa/qcocoatheme.mm
@@ -56,6 +56,8 @@
 #include <qpa/qplatformintegration.h>
 #include <qpa/qplatformnativeinterface.h>
 
+#include <QtCore/qstandardpaths.h>
+
 QT_BEGIN_NAMESPACE
 
 const char *QCocoaTheme::name = "cocoa";
@@ -272,6 +274,29 @@ QPixmap QCocoaTheme::fileIconPixmap(const QFileInfo &fileInfo, const QSizeF &siz
     return pixmap;
 }
 
+// adapted from QGenericUnixTheme::xdgIconThemePaths()
+static QStringList xdgIconThemePaths()
+{
+    QStringList paths;
+    // Add home directory first in search path
+    const QFileInfo homeIconDir(QDir::homePath() + QStringLiteral("/.icons"));
+    if (homeIconDir.isDir())
+        paths.prepend(homeIconDir.absoluteFilePath());
+
+    QStringList xdgDirs = QStandardPaths::standardLocations(QStandardPaths::GenericDataLocation);
+    if (!xdgDirs.isEmpty()) {
+        foreach (const QString &xdgDir, xdgDirs) {
+            const QFileInfo xdgIconsDir(xdgDir + QStringLiteral("/icons"));
+            if (xdgIconsDir.isDir())
+                paths.append(xdgIconsDir.absoluteFilePath());
+            const QFileInfo pixmapsIconsDir(xdgDir + QStringLiteral("/pixmaps"));
+            if (pixmapsIconsDir.isDir())
+                paths.append(pixmapsIconsDir.absoluteFilePath());
+        }
+    }
+    return paths;
+}
+
 QVariant QCocoaTheme::themeHint(ThemeHint hint) const
 {
     switch (hint) {
@@ -295,6 +320,20 @@ QVariant QCocoaTheme::themeHint(ThemeHint hint) const
     }
     case QPlatformTheme::PasswordMaskCharacter:
         return QVariant(QChar(kBulletUnicode));
+    case QPlatformTheme::SystemIconThemeName:
+        if (QStandardPaths::isXDGLocationsEnabled()) {
+            return QVariant(QString(QStringLiteral("Ciment")));
+        }
+    case QPlatformTheme::SystemIconFallbackThemeName:
+        if (QStandardPaths::isXDGLocationsEnabled()) {
+            return QVariant(QString(QStringLiteral("hicolor")));
+        }
+    case QPlatformTheme::IconThemeSearchPaths:
+        if (QStandardPaths::isXDGLocationsEnabled()) {
+            return xdgIconThemePaths();
+        }
+    case QPlatformTheme::DialogButtonBoxButtonsHaveIcons:
+        return false;
     default:
         break;
     }
