--- a/qtbase/src//plugins/platforms/cocoa/qcocoawindow.mm.orig	2016-05-25 17:46:17.000000000 +0200
+++ b/qtbase/src//plugins/platforms/cocoa/qcocoawindow.mm	2016-10-07 18:17:10.000000000 +0200
@@ -1801,7 +1803,7 @@
     NSUInteger screenIndex = [[NSScreen screens] indexOfObject:m_nsWindow.screen];
     if (screenIndex != NSNotFound) {
         QCocoaScreen *cocoaScreen = QCocoaIntegration::instance()->screenAtIndex(screenIndex);
-        if (cocoaScreen)
+        if (cocoaScreen && !window()->parent())
             window()->setScreen(cocoaScreen->screen());
     }
 
