diff --git a/qtbase/src/plugins/platforms/cocoa/qnsview.mm b/qtbase/src/plugins/platforms/cocoa/qnsview.mm
index 0d80333..f8449a2 100644
--- a/qtbase/src/plugins/platforms/cocoa/qnsview.mm
+++ b/qtbase/src/plugins/platforms/cocoa/qnsview.mm
@@ -655,11 +655,11 @@ QT_WARNING_POP
 {
     if (m_isMenuView)
         return NO;
-    if (m_platformWindow->shouldRefuseKeyWindowAndFirstResponder())
+    if (m_platformWindow && m_platformWindow->shouldRefuseKeyWindowAndFirstResponder())
         return NO;
     if (m_window && (m_window->flags() & Qt::WindowTransparentForInput) )
         return NO;
-    if ((m_window->flags() & Qt::ToolTip) == Qt::ToolTip)
+    if (m_window && (m_window->flags() & Qt::ToolTip) == Qt::ToolTip)
         return NO;
     return YES;
 }
