--- a/qtbase/src/gui/opengl/orig.qopengl.h	2014-09-11 12:48:05.000000000 +0200
+++ b/qtbase/src/gui/opengl/qopengl.h	2015-03-25 20:54:53.000000000 +0100
@@ -98,7 +98,7 @@
 typedef GLfloat GLdouble;
 # endif
 #else // non-ES2 platforms
-# if defined(Q_OS_MAC)
+# if defined(Q_OS_MAC_NOTX11)
 #  include <OpenGL/gl.h>
 #  if MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_X_VERSION_10_7
 #   define GL_DO_NOT_WARN_IF_MULTI_GL_VERSION_HEADERS_INCLUDED
--- a/qtbase/mkspecs/common/orig.mac.conf	2014-09-11 12:48:07.000000000 +0200
+++ b/qtbase/mkspecs/common/mac.conf	2015-03-25 20:53:23.000000000 +0100
@@ -11,9 +11,7 @@
 QMAKE_LIBDIR            =
 
 # sdk.prf will prefix the proper SDK sysroot
-QMAKE_INCDIR_OPENGL     = \
-    /System/Library/Frameworks/OpenGL.framework/Headers \
-    /System/Library/Frameworks/AGL.framework/Headers/
+QMAKE_INCDIR_OPENGL     = 
 
 QMAKE_FIX_RPATH         = install_name_tool -id
 
@@ -21,7 +19,8 @@
 QMAKE_LFLAGS_GCSECTIONS = -Wl,-dead_strip
 
 QMAKE_LIBS_DYNLOAD      =
-QMAKE_LIBS_OPENGL       = -framework OpenGL -framework AGL
+QMAKE_LIBS_OPENGL       = -lGL
+QMAKE_LIBS_X11          = -lXext -lX11
 QMAKE_LIBS_THREAD       =
 
 QMAKE_AR                = ar cq
