diff --git a/qtbase/src/corelib/kernel/qcoreapplication.cpp b/qtbase/src/corelib/kernel/qcoreapplication.cpp
index f5b1520..0089da8 100644
--- a/qtbase/src/corelib/kernel/qcoreapplication.cpp
+++ b/qtbase/src/corelib/kernel/qcoreapplication.cpp
@@ -324,7 +324,12 @@ Qt::HANDLE qt_application_thread_id = 0;
 #endif // QT_NO_QOBJECT
 
 QCoreApplication *QCoreApplication::self = 0;
+#if defined(Q_OS_MAC)
+uint QCoreApplicationPrivate::attribs = (1 << Qt::AA_SynthesizeMouseForUnhandledTouchEvents)
+                                    | (1 << Qt::AA_DontShowIconsInMenus);
+#else
 uint QCoreApplicationPrivate::attribs = (1 << Qt::AA_SynthesizeMouseForUnhandledTouchEvents);
+#endif
 
 struct QCoreApplicationData {
     QCoreApplicationData() Q_DECL_NOTHROW {
