diff --git a/qtbase/src/widgets/widgets/qpushbutton.cpp b/qtbase/src/widgets/widgets/qpushbutton.cpp
index a7bb9fa..08f6a6a 100644
--- a/qtbase/src/widgets/widgets/qpushbutton.cpp
+++ b/qtbase/src/widgets/widgets/qpushbutton.cpp
@@ -332,8 +332,13 @@ void QPushButton::initStyleOption(QStyleOptionButton *option) const
     if (!d->flat && !d->down)
         option->state |= QStyle::State_Raised;
     option->text = d->text;
-    option->icon = d->icon;
-    option->iconSize = iconSize();
+    if (style()->styleHint(QStyle::SH_DialogButtonBox_ButtonsHaveIcons)) {
+	    option->icon = d->icon;
+	    option->iconSize = iconSize();
+    } else {
+	    option->icon = QIcon();
+	    option->iconSize = QSize(0,0);
+    }
 }
 
 void QPushButton::setAutoDefault(bool enable)
@@ -403,7 +408,7 @@ QSize QPushButton::sizeHint() const
     bool showButtonBoxIcons = qobject_cast<QDialogButtonBox*>(parentWidget())
                           && style()->styleHint(QStyle::SH_DialogButtonBox_ButtonsHaveIcons);
 
-    if (!icon().isNull() || showButtonBoxIcons) {
+    if (!icon().isNull() && showButtonBoxIcons) {
         int ih = opt.iconSize.height();
         int iw = opt.iconSize.width() + 4;
         w += iw;
