diff --git a/qtbase/src/widgets/widgets/qpushbutton.cpp b/qtbase/src/widgets/widgets/qpushbutton.cpp
index a7bb9fa..69dfd7c 100644
--- a/qtbase/src/widgets/widgets/qpushbutton.cpp
+++ b/qtbase/src/widgets/widgets/qpushbutton.cpp
@@ -403,7 +403,7 @@ QSize QPushButton::sizeHint() const
     bool showButtonBoxIcons = qobject_cast<QDialogButtonBox*>(parentWidget())
                           && style()->styleHint(QStyle::SH_DialogButtonBox_ButtonsHaveIcons);
 
-    if (!icon().isNull() || showButtonBoxIcons) {
+    if (!icon().isNull() && showButtonBoxIcons) {
         int ih = opt.iconSize.height();
         int iw = opt.iconSize.width() + 4;
         w += iw;
