diff --git a/qtbase/configure b/qtbase/configure
index 4ce83be..1d43d83 100755
--- a/qtbase/configure
+++ b/qtbase/configure
@@ -122,7 +122,7 @@ shellQuoteLines()
 {
     # The call of the outer echo makes the shell word-split the output of
     # the nested pipe, thus effectively converting newlines to spaces.
-    echo `echo "$1" | sed 's,^[^ ]* .*$,"&",'`
+    echo `echo $@ | sed 's,^[^ ]* .*$,&,'`
 }
 
 makeabs()
@@ -3659,6 +3659,11 @@ unset tty
 [ "$OPT_VERBOSE" = "yes" ] && tty=/dev/stderr
 eval "`LC_ALL=C $TEST_COMPILER $SYSROOT_FLAG $TEST_COMPILER_CXXFLAGS -xc++ -E -v - < /dev/null 2>&1 > /dev/null | $AWK "$awkprog" | tee $tty`"
 unset tty
+# remove unnecessary and unwanted paths (for MacPorts we don't want /usr/local/include either)
+DEFAULT_INCDIRS=`echo ${DEFAULT_INCDIRS} | sed -e 's|/usr/include/[xi][0-9][^-]*-linux-gnu[^ ]*||g' -e 's|/usr/lib/gcc/[xi][0-9][^-]*-linux-gnu/[^ ]*||g' -e 's|/usr/include/c++[^ ]*||g' -e 's|/usr/local/include||g'`
+DEFAULT_LIBDIRS=`echo ${DEFAULT_LIBDIRS} | sed -e 's|/usr/lib/[xi][0-9][^-]*-linux-gnu||g' -e 's|/lib/[xi][0-9][^-]*-linux-gnu||g' -e 's|/usr/lib/gcc/[xi][0-9][^-]*-linux-gnu/[0-9][^ ]*||g'`
+echo "Pruned DEFAULT_INCDIRS=$DEFAULT_INCDIRS"
+echo "Pruned DEFAULT_LIBDIRS=$DEFAULT_LIBDIRS"
 
 echo "Done running configuration tests."
 
