diff --git orig.meson.build meson.build
index 4e56544..97b651b 100644
--- orig.meson.build
+++ meson.build
@@ -249,7 +249,7 @@ libxkbcommon = library(
     link_args: libxkbcommon_link_args,
     link_depends: libxkbcommon_link_deps,
     gnu_symbol_visibility: 'hidden',
-    version: '0.0.0',
+    version: '1.0.0',
     install: true,
     include_directories: include_directories('src', 'include'),
 )
@@ -317,7 +317,7 @@ You can disable X11 support with -Denable-x11=false.''')
         link_args: libxkbcommon_x11_link_args,
         link_depends: libxkbcommon_x11_link_deps,
         gnu_symbol_visibility: 'hidden',
-        version: '0.0.0',
+        version: '1.0.0',
         install: true,
         include_directories: include_directories('src', 'include'),
         link_with: libxkbcommon,
