diff --git a/qtbase/src/plugins/platforms/offscreen/qoffscreenintegration.cpp b/qtbase/src/plugins/platforms/offscreen/qoffscreenintegration.cpp
index 0c3995001920ce14f34721a4a1cf7ae3a8f5eaaf..9fbb266f63bfeef4fa7f3420d917d1e81b267ee2 100644
--- a/qtbase/src/plugins/platforms/offscreen/qoffscreenintegration.cpp
+++ b/qtbase/src/plugins/platforms/offscreen/qoffscreenintegration.cpp
@@ -45,6 +45,7 @@
 #include <QtEventDispatcherSupport/private/qgenericunixeventdispatcher_p.h>
 #if defined(Q_OS_MAC)
 #include <qpa/qplatformfontdatabase.h>
+#include <QtFontDatabaseSupport/private/qcoretextfontdatabase_p.h>
 #else
 #include <QtFontDatabaseSupport/private/qgenericunixfontdatabase_p.h>
 #endif
@@ -64,6 +65,8 @@
 
 QT_BEGIN_NAMESPACE
 
+class QCoreTextFontEngine;
+
 template <typename BaseEventDispatcher>
 class QOffscreenEventDispatcher : public BaseEventDispatcher
 {
@@ -98,7 +101,7 @@ QOffscreenIntegration::QOffscreenIntegration()
 {
 #if defined(Q_OS_UNIX)
 #if defined(Q_OS_MAC)
-    m_fontDatabase.reset(new QPlatformFontDatabase());
+    m_fontDatabase.reset(new QCoreTextFontDatabaseEngineFactory<QCoreTextFontEngine>);
 #else
     m_fontDatabase.reset(new QGenericUnixFontDatabase());
 #endif
