diff --git test/testutil/helper.c.orig test/testutil/helper.c
index b1acd5b..0fd8035 100644
--- test/testutil/helper.c.orig
+++ test/testutil/helper.c
@@ -19,7 +19,7 @@
 #define timezone _timezone
 #endif
 
-#if defined(__FreeBSD__) || defined(__wasi__) || (defined(__APPLE__) && !defined(OPENSSL_NO_APPLE_CRYPTO_RANDOM) && !(defined(__MAC_OS_X_VERSION_MIN_REQUIRED) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 1050))
+#if defined(__FreeBSD__) || defined(__wasi__) || (defined(__APPLE__) && !defined(OPENSSL_NO_APPLE_CRYPTO_RANDOM))
 #define USE_TIMEGM
 #endif
 
