diff --git orig.Configure Configure
index cca1ac8..a5fd98e 100755
--- orig.Configure
+++ Configure
@@ -1628,8 +1628,7 @@ if (!$disabled{makedepend}) {
         # If the makedepcmd attribute is defined, copy it.  If not, the
         # build files will have to fend for themselves.
         $config{makedepcmd} = $target{makedepcmd} if $target{makedepcmd};
-    } elsif (($predefined_C{__GNUC__} // -1) >= 3
-             && !($predefined_C{__APPLE_CC__} && !$predefined_C{__clang__})) {
+    } elsif (($predefined_C{__GNUC__} // -1) >= 3) {
         # We know that GNU C version 3 and up as well as all clang
         # versions support dependency generation, but Xcode did not
         # handle $cc -M before clang support (but claims __GNUC__ = 3)
