diff --git CMakeLists.txt CMakeLists.txt
index ecd8e3d3..01924431 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -10,7 +10,7 @@ feature_summary(WHAT REQUIRED_PACKAGES_NOT_FOUND FATAL_ON_MISSING_REQUIRED_PACKA
 
 set(CMAKE_MODULE_PATH ${ECM_MODULE_PATH})
 
-set(REQUIRED_QT_VERSION 5.14.0)
+set(REQUIRED_QT_VERSION 5.9.8)
 
 find_package(Qt5 ${REQUIRED_QT_VERSION} CONFIG REQUIRED Xml)
 option(KCONFIG_USE_GUI "Build components using Qt5Gui" ON)
diff --git autotests/kconfigtest.cpp autotests/kconfigtest.cpp
index 20414e45..6cf64fe8 100644
--- autotests/kconfigtest.cpp
+++ autotests/kconfigtest.cpp
@@ -1751,6 +1751,7 @@ void KConfigTest::testQByteArrayUtf8()
 
 void KConfigTest::testQStringUtf8_data()
 {
+#if QT_VERSION >= QT_VERSION_CHECK(5, 11, 0)
     QTest::addColumn<QByteArray>("data");
     QTest::newRow("1") << QByteArray("Téléchargements\tTéléchargements");
     QTest::newRow("2") << QByteArray("$¢ह€𐍈\t$¢ह€𐍈");
@@ -1768,10 +1769,12 @@ void KConfigTest::testQStringUtf8_data()
     QTest::newRow("9") << QByteArray("\xe0\xa0\x80\t\xe0\xa0\x80");
     QTest::newRow("10") << QByteArray("\xf0\x90\x80\x80\t\xf0\x90\x80\x80");
     QTest::newRow("11") << QByteArray("\xf4\x8f\xbf\xbf\t\xf4\x8f\xbf\xbf");
+#endif
 }
 
 void KConfigTest::testQStringUtf8()
 {
+#if QT_VERSION >= QT_VERSION_CHECK(5, 11, 0)
     QFETCH(QByteArray, data);
     const QList<QByteArray> d = data.split('\t');
     const QByteArray value = d[0];
@@ -1796,6 +1799,7 @@ void KConfigTest::testQStringUtf8()
     KConfig config2(file.fileName(), KConfig::SimpleConfig);
     KConfigGroup general2(&config2, "General");
     QCOMPARE(value, general2.readEntry("key", QByteArray()));
+#endif
 }
 
 void KConfigTest::testNewlines()
diff --git src/core/kconfig.cpp src/core/kconfig.cpp
index 52fad902..165e23b1 100644
--- src/core/kconfig.cpp
+++ src/core/kconfig.cpp
@@ -36,6 +36,11 @@
 #include <QDBusMetaType>
 #endif
 
+#if QT_VERSION < QT_VERSION_CHECK(5, 14, 0)
+#   define storeRelaxed(val)   store(val)
+#   define loadRelaxed()       load()
+#endif
+
 bool KConfigPrivate::mappingsRegistered = false;
 
 // For caching purposes
diff --git src/kconf_update/kconf_update.cpp src/kconf_update/kconf_update.cpp
index 99f6fd5c..51ba98a5 100644
--- src/kconf_update/kconf_update.cpp
+++ src/kconf_update/kconf_update.cpp
@@ -28,6 +28,10 @@
 #include "kconf_update_debug.h"
 #include "kconfigutils.h"
 
+#if QT_VERSION < QT_VERSION_CHECK(5, 10, 0)
+#   define birthTime()  created()
+#endif
+
 // Convenience wrapper around qCDebug to prefix the output with metadata of
 // the file.
 #define qCDebugFile(CATEGORY) \
diff --git src/kconfig_compiler/KConfigXmlParser.cpp src/kconfig_compiler/KConfigXmlParser.cpp
index e1010564..c07f2969 100644
--- src/kconfig_compiler/KConfigXmlParser.cpp
+++ src/kconfig_compiler/KConfigXmlParser.cpp
@@ -21,6 +21,14 @@
 #include <QStringList>
 #include <QTextStream>
 #include <iostream>
+
+#if QT_VERSION >= QT_VERSION_CHECK(5, 15, 0)
+#   define ANCHOREDPATTERN(expression) QRegularExpression::anchoredPattern(expression)
+#else
+// from the current Qt sources:
+#   define ANCHOREDPATTERN(expression) (QString() + QLatin1String("\\A(?:") + expression + QLatin1String(")\\z"))
+#endif
+
 //TODO: Move preprocessDefault to Header / CPP implementation.
 // it makes no sense for a parser to process those values and generate code.
 
@@ -64,7 +72,7 @@ static void preProcessDefault(QString &defaultValue, const QString &name,
         defaultValue = QLatin1String("default") + name;
 
     } else if (type == QLatin1String("Color") && !defaultValue.isEmpty()) {
-        const QRegularExpression colorRe(QRegularExpression::anchoredPattern(
+        const QRegularExpression colorRe(ANCHOREDPATTERN(
                                          QStringLiteral("\\d+,\\s*\\d+,\\s*\\d+(,\\s*\\d+)?")));
 
         if (colorRe.match(defaultValue).hasMatch()) {
@@ -72,7 +80,6 @@ static void preProcessDefault(QString &defaultValue, const QString &name,
         } else {
             defaultValue = QLatin1String("QColor( \"") + defaultValue + QLatin1String("\" )");
         }
-
     } else if (type == QLatin1String("Enum")) {
         QList<CfgEntry::Choice>::ConstIterator it;
         for (it = choices.choices.constBegin(); it != choices.choices.constEnd(); ++it) {
@@ -435,7 +442,7 @@ CfgEntry *KConfigXmlParser::parseEntry(const QString &group, const QString &pare
 KConfigXmlParser::KConfigXmlParser(const KConfigParameters &cfg, const QString &inputFileName)
     : cfg(cfg), mInputFileName(inputFileName)
 {
-    mValidNameRegexp.setPattern(QRegularExpression::anchoredPattern(QStringLiteral("[a-zA-Z_][a-zA-Z0-9_]*")));
+    mValidNameRegexp.setPattern(ANCHOREDPATTERN(QStringLiteral("[a-zA-Z_][a-zA-Z0-9_]*")));
 }
 
 void KConfigXmlParser::start()
