diff --git a/CMakeLists.txt b/CMakeLists.txt
index 336a68d..d8c1473 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,6 +2,7 @@ project(lzfse C)
 cmake_minimum_required(VERSION 2.8.6)
 
 include(CheckCCompilerFlag)
+include(FeatureSummary)
 
 # If LZFSE is being bundled in another project, we don't want to
 # install anything.  However, we want to let people override this, so
@@ -135,3 +136,5 @@ if(NOT LZFSE_DISABLE_TESTS)
         -P ${CMAKE_CURRENT_SOURCE_DIR}/tests/round-trip.cmake)
   endforeach()
 endif()
+
+feature_summary(WHAT ALL   FATAL_ON_MISSING_REQUIRED_PACKAGES)
