diff --git Source/MediaInfo/orig.ExternalCommandHelpers.cpp Source/MediaInfo/ExternalCommandHelpers.cpp
index 411eceb..c7293a2 100644
--- Source/MediaInfo/orig.ExternalCommandHelpers.cpp
+++ Source/MediaInfo/ExternalCommandHelpers.cpp
@@ -40,7 +40,7 @@ const ZtringList ffmpeg_PossibleNames= // Hack, ZtringList needs to handle C++11
     #ifdef WINDOWS
     __T("ffmpeg-ma.exe;ffmpeg.exe")
     #else
-    __T("ffmpeg-ma;ffmpeg")
+    __T("ffmpeg-ma;@FFMPEG@")
     #endif //WINDOWS
 };
 
