diff --git Source/MediaInfo/Export/orig.Export_Graph.cpp Source/MediaInfo/Export/Export_Graph.cpp
index c10aaa1..35eb6dc 100644
--- Source/MediaInfo/Export/orig.Export_Graph.cpp
+++ Source/MediaInfo/Export/Export_Graph.cpp
@@ -130,7 +130,12 @@ Ztring Dot2Svg(const Ztring& Dot)
         return ToReturn;
 
     int Renderers_Size=0;
+// see https://forum.graphviz.org/t/api-for-function-gvpluginlist-when-did-it-change/3135/2
+#ifndef DTTREEMATCH
     char** Renderers=gvPluginList(Context, "render", &Renderers_Size);
+#else
+    char** Renderers=gvPluginList(Context, "render", &Renderers_Size, "");
+#endif
     for (int Pos=0; Pos<Renderers_Size; Pos++) {
         if (!strcmp(Renderers[Pos], "cairo"))
             Cairo=true;
