diff --git src/dolphinmainwindow.cpp src/dolphinmainwindow.cpp
index d4f2b06..c69202d 100644
--- src/dolphinmainwindow.cpp
+++ src/dolphinmainwindow.cpp
@@ -84,6 +84,15 @@ namespace {
     const int CurrentDolphinVersion = 200;
 };
 
+static inline bool canHideMenuBar()
+{
+#ifdef Q_OS_OSX
+    return !QGuiApplication::platformName().contains(QLatin1String("cocoa"));
+#else
+    return true;
+#endif
+}
+
 DolphinMainWindow::DolphinMainWindow() :
     KXmlGuiWindow(0),
     m_newFileMenu(0),
@@ -158,11 +167,13 @@ DolphinMainWindow::DolphinMainWindow() :
         resize(750, 500);
     }
 
-    const bool showMenu = !menuBar()->isHidden();
-    QAction* showMenuBarAction = actionCollection()->action(KStandardAction::name(KStandardAction::ShowMenubar));
-    showMenuBarAction->setChecked(showMenu);  // workaround for bug #171080
-    if (!showMenu) {
-        createControlButton();
+    if (canHideMenuBar()) {
+        const bool showMenu = !menuBar()->isHidden();
+        QAction* showMenuBarAction = actionCollection()->action(KStandardAction::name(KStandardAction::ShowMenubar));
+        showMenuBarAction->setChecked(showMenu);  // workaround for bug #171080
+        if (!showMenu) {
+            createControlButton();
+        }
     }
 }
 
@@ -666,12 +677,14 @@ void DolphinMainWindow::compareFiles()
 
 void DolphinMainWindow::toggleShowMenuBar()
 {
-    const bool visible = menuBar()->isVisible();
-    menuBar()->setVisible(!visible);
-    if (visible) {
-        createControlButton();
-    } else {
-        deleteControlButton();
+    if (canHideMenuBar()) {
+        const bool visible = menuBar()->isVisible();
+        menuBar()->setVisible(!visible);
+        if (visible) {
+            createControlButton();
+        } else {
+            deleteControlButton();
+        }
     }
 }
 
@@ -879,7 +892,7 @@ void DolphinMainWindow::updateControlMenu()
 
 void DolphinMainWindow::updateToolBar()
 {
-    if (!menuBar()->isVisible()) {
+    if (canHideMenuBar() && !menuBar()->isVisible()) {
         createControlButton();
     }
 }
