--- orig.boostcpp.jam	2020-06-21 21:41:48.000000000 +0200
+++ boostcpp.jam	2020-06-21 21:43:44.000000000 +0200
@@ -148,7 +148,7 @@
             <base> <toolset> <threading> <runtime> <arch-and-model> ;
     case 1.69 :
         .format-name-args =
-            <base> <toolset> <threading> <runtime> <arch-and-model> ;
+            <base> <threading> <runtime> ;
     }
 }
 else if $(layout) = tagged
@@ -163,7 +163,7 @@
             <base> <threading> <runtime> ;
     case 1.69 :
         .format-name-args =
-            <base> <threading> <runtime> <arch-and-model> ;
+            <base> <threading> <runtime> ;
     }
 }
 else if $(layout) = system
