--- tools/build/src/tools/clang-darwin.jam.orig	2019-08-14 05:03:50.000000000 -0700
+++ tools/build/src/tools/clang-darwin.jam	2019-08-28 08:47:22.000000000 -0700
@@ -49,7 +49,7 @@
 #   compile and link options allow you to specify addition command line options for each version
 rule init ( version ? :  command * : options * )
 {
-    command = [ common.get-invocation-command clang-darwin : clang++ 
+    command = [ common.get-invocation-command clang-darwin : __MACPORTS_CXX__
         : $(command) : /usr/bin /usr/local/bin ] ;
 
     # Determine the version
--- tools/build/src/tools/orig.clang-linux.jam	2019-08-14 14:03:50.000000000 +0200
+++ tools/build/src/tools/clang-linux.jam	2020-06-21 11:48:17.285202564 +0200
@@ -45,7 +45,7 @@
 }
                        
 rule init ( version ? :  command * : options * ) {
-  command = [ common.get-invocation-command clang-linux : clang++ 
+  command = [ common.get-invocation-command clang-linux : __MACPORTS_CXX__
     : $(command) ] ;
                 
   # Determine the version
--- tools/build/src/tools/orig.gcc.jam	2019-08-14 14:03:50.000000000 +0200
+++ tools/build/src/tools/gcc.jam	2020-06-21 11:49:16.446444333 +0200
@@ -154,7 +154,7 @@
     local tool-command = ;
     if $(command)
     {
-        tool-command = [ common.get-invocation-command-nodefault gcc : g++ :
+        tool-command = [ common.get-invocation-command-nodefault __MACPORTS_CC__ : __MACPORTS_CXX__ :
             $(command) ] ;
         if ! $(tool-command)
         {
--- tools/build/src/tools/orig.gcc.py	2019-08-14 14:03:50.000000000 +0200
+++ tools/build/src/tools/gcc.py	2020-06-21 11:51:08.558797994 +0200
@@ -101,7 +101,7 @@
 
     # Information about the gcc command...
     #   The command.
-    command = to_seq(common.get_invocation_command('gcc', 'g++', command))
+    command = to_seq(common.get_invocation_command('__MACPORTS_CC__', '__MACPORTS_CXX__', command))
     #   The root directory of the tool install.
     root = feature.get_values('<root>', options)
     root = root[0] if root else ''
