diff --git Source/WebCore/page/Crypto.cpp Source/WebCore/page/Crypto.cpp
index 1d8e3863..29d7bbc1 100644
--- Source/WebCore/page/Crypto.cpp
+++ Source/WebCore/page/Crypto.cpp
@@ -37,7 +37,7 @@
 #include <wtf/CryptographicallyRandomNumber.h>
 #include <wtf/UUID.h>
 
-#if OS(DARWIN)
+#if OS(DARWIN) && (__MAC_OS_X_VERSION_MIN_REQUIRED > 1090)
 #include <CommonCrypto/CommonCryptor.h>
 #include <CommonCrypto/CommonRandom.h>
 #endif
@@ -58,7 +58,7 @@ ExceptionOr<void> Crypto::getRandomValues(ArrayBufferView& array)
         return Exception { ExceptionCode::TypeMismatchError };
     if (array.byteLength() > 65536)
         return Exception { ExceptionCode::QuotaExceededError };
-#if OS(DARWIN)
+#if OS(DARWIN) && (__MAC_OS_X_VERSION_MIN_REQUIRED > 1090)
     auto rc = CCRandomGenerateBytes(array.baseAddress(), array.byteLength());
     RELEASE_ASSERT(rc == kCCSuccess);
 #else
