diff --git Source/cmake/OptionsCommon.cmake.orig Source/cmake/OptionsCommon.cmake
index d88f27e..fef14df 100644
--- Source/cmake/OptionsCommon.cmake.orig
+++ Source/cmake/OptionsCommon.cmake
@@ -1,3 +1,5 @@
+include(CheckLinkerFlag)
+
 set(CMAKE_CXX_STANDARD 20)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS OFF)
@@ -194,6 +196,14 @@ option(GCC_OFFLINEASM_SOURCE_MAP
 
 option(USE_APPLE_ICU "Use Apple's internal ICU" ${APPLE})
 
+# Pass --reduce-memory-overheads to the bfd linker in order to save memory
+check_linker_flag(CXX "-Wl,--reduce-memory-overheads" HAS_REDUCE_MEMOH)
+if (HAS_REDUCE_MEMOH)
+    message(STATUS "Instructing linker to reduce memory overheads")
+    set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,--reduce-memory-overheads")
+    set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,--reduce-memory-overheads")
+endif ()
+
 # Enable the usage of OpenMP.
 #  - At this moment, OpenMP is only used as an alternative implementation
 #    to native threads for the parallelization of the SVG filters.
