diff --git Source/JavaScriptCore/API/orig.WebKitAvailability.h Source/JavaScriptCore/API/WebKitAvailability.h
index 11e4087..4ef0b7c 100644
--- Source/JavaScriptCore/API/orig.WebKitAvailability.h
+++ Source/JavaScriptCore/API/WebKitAvailability.h
@@ -28,6 +28,11 @@
 
 #if defined(__APPLE__)
 
+#if defined(BUILDING_GTK__)
+// prevent clash with old QuickDraw symbols in /usr/include/MacTypes.h
+#include <JavaScriptCore/MacTypes.h>
+#endif
+
 #include <AvailabilityMacros.h>
 #include <CoreFoundation/CoreFoundation.h>
 
