diff --git Source/cmake/orig.WebKitCompilerFlags.cmake Source/cmake/WebKitCompilerFlags.cmake
index 074fdf5..acfb9d4 100644
--- Source/cmake/orig.WebKitCompilerFlags.cmake
+++ Source/cmake/WebKitCompilerFlags.cmake
@@ -198,10 +198,10 @@ if (COMPILER_IS_GCC_OR_CLANG AND NOT MSVC)
 endif ()
 
 # Ninja tricks compilers into turning off color support.
-if (CMAKE_GENERATOR STREQUAL "Ninja")
-    WEBKIT_PREPEND_GLOBAL_COMPILER_FLAGS(-fcolor-diagnostics
-                                         -fdiagnostics-color=always)
-endif ()
+# if (CMAKE_GENERATOR STREQUAL "Ninja")
+#     WEBKIT_PREPEND_GLOBAL_COMPILER_FLAGS(-fcolor-diagnostics
+#                                          -fdiagnostics-color=always)
+# endif ()
 
 
 string(TOLOWER ${CMAKE_HOST_SYSTEM_PROCESSOR} LOWERCASE_CMAKE_HOST_SYSTEM_PROCESSOR)
