diff --git Source/WebKit/orig.PlatformGTK.cmake Source/WebKit/PlatformGTK.cmake
index e229687..631a982 100644
--- Source/WebKit/orig.PlatformGTK.cmake
+++ Source/WebKit/PlatformGTK.cmake
@@ -244,6 +244,7 @@ list(INSERT WebKit_INCLUDE_DIRECTORIES 0
 )
 
 list(APPEND WebKit_PRIVATE_INCLUDE_DIRECTORIES
+    "${JavaScriptCoreGLib_DERIVED_SOURCES_DIR}"
     "${JavaScriptCoreGLib_DERIVED_SOURCES_DIR}/jsc"
 )
 
diff --git Tools/TestWebKitAPI/orig.PlatformGTK.cmake Tools/TestWebKitAPI/PlatformGTK.cmake
index 87c714b..72bd474 100644
--- Tools/TestWebKitAPI/orig.PlatformGTK.cmake
+++ Tools/TestWebKitAPI/PlatformGTK.cmake
@@ -98,6 +98,7 @@ set(TestJSC_SYSTEM_INCLUDE_DIRECTORIES
 set(TestJSC_PRIVATE_INCLUDE_DIRECTORIES
     ${CMAKE_BINARY_DIR}
     ${TESTWEBKITAPI_DIR}
+    "${JavaScriptCoreGLib_DERIVED_SOURCES_DIR}"
     "${JavaScriptCoreGLib_DERIVED_SOURCES_DIR}/jsc"
 )
 
