diff --git Source/WTF/wtf/orig.CurrentTime.cpp Source/WTF/wtf/CurrentTime.cpp
index c042a0f..e7c6362 100644
--- Source/WTF/wtf/orig.CurrentTime.cpp
+++ Source/WTF/wtf/CurrentTime.cpp
@@ -288,7 +288,12 @@ MonotonicTime MonotonicTime::now()
 ApproximateTime ApproximateTime::now()
 {
 #if OS(DARWIN)
+#if MAC_OS_X_VERSION_MAX_ALLOWED > 101000
     return fromMachApproximateTime(mach_approximate_time());
+#else
+    // anything better we can do??
+    return fromMachApproximateTime(mach_absolute_time());
+#endif
 #elif OS(LINUX)
     struct timespec ts { };
     clock_gettime(CLOCK_MONOTONIC_COARSE, &ts);
