diff --git Source/WTF/wtf/Assertions.h Source/WTF/wtf/Assertions.h
index 6f9c7ed8..4481c83e 100644
--- Source/WTF/wtf/Assertions.h
+++ Source/WTF/wtf/Assertions.h
@@ -47,7 +47,7 @@
 #include <stdlib.h>
 #include <wtf/ExportMacros.h>
 
-#if OS(DARWIN)
+#if PLATFORM(COCOA)
 #include <wtf/spi/darwin/AbortWithReasonSPI.h>
 #endif
 
@@ -840,7 +840,7 @@ inline void compilerFenceForCrash()
 
 #endif /* __cplusplus */
 
-#if OS(DARWIN)
+#if PLATFORM(COCOA)
 #define CRASH_WITH_EXTRA_SECURITY_IMPLICATION_AND_INFO(abortReason, abortMsg, ...) do { \
         if (g_wtfConfig.useSpecialAbortForExtraSecurityImplications) \
             abort_with_reason(OS_REASON_WEBKIT, abortReason, abortMsg, OS_REASON_FLAG_SECURITY_SENSITIVE); \
diff --git Source/WTF/wtf/PlatformJSCOnly.cmake Source/WTF/wtf/PlatformJSCOnly.cmake
index bc0aada1..b243c6af 100644
--- Source/WTF/wtf/PlatformJSCOnly.cmake
+++ Source/WTF/wtf/PlatformJSCOnly.cmake
@@ -94,10 +94,12 @@ elseif (APPLE)
         ${WTF_DERIVED_SOURCES_DIR}/mach_excServer.c
         ${WTF_DERIVED_SOURCES_DIR}/mach_excUser.c
     )
-    list(APPEND WTF_PUBLIC_HEADERS
-        spi/darwin/AbortWithReasonSPI.h
-        spi/darwin/ProcessMemoryFootprint.h
-    )
+    if ("${PORT}" STREQUAL "Mac")
+	    list(APPEND WTF_PUBLIC_HEADERS
+		   spi/darwin/AbortWithReasonSPI.h
+		   spi/darwin/ProcessMemoryFootprint.h
+	    )
+	endif()
 elseif (CMAKE_SYSTEM_NAME MATCHES "Linux")
     list(APPEND WTF_SOURCES
         linux/CurrentProcessMemoryStatus.cpp
