diff --git Source/WTF/wtf/orig.Gigacage.h Source/WTF/wtf/Gigacage.h
index 1d4f068..2bc9d2b 100644
--- Source/WTF/wtf/orig.Gigacage.h
+++ Source/WTF/wtf/Gigacage.h
@@ -31,6 +31,11 @@
 #if USE(SYSTEM_MALLOC)
 #define GIGACAGE_ENABLED 0
 
+#if OS(DARWIN)
+#include <wtf/MathExtras.h>
+#include <mach/mach_vm.h>
+#endif
+
 namespace Gigacage {
 
 constexpr bool hasCapacityToUseLargeGigacage = OS_CONSTANT(EFFECTIVE_ADDRESS_WIDTH) > 36;
