diff --git Source/WTF/wtf/posix/orig.CPUTimePOSIX.cpp Source/WTF/wtf/posix/CPUTimePOSIX.cpp
index fd16889..a4d23ff 100644
--- Source/WTF/wtf/posix/orig.CPUTimePOSIX.cpp
+++ Source/WTF/wtf/posix/CPUTimePOSIX.cpp
@@ -53,7 +53,7 @@ std::optional<CPUTime> CPUTime::get()
     mach_msg_type_number_t threadInfoCount = TASK_THREAD_TIMES_INFO_COUNT;
     kern_return_t result = task_info(mach_task_self(), TASK_THREAD_TIMES_INFO, reinterpret_cast<task_info_t>(&threadInfoData), &threadInfoCount);
     if (result != KERN_SUCCESS)
-        return nullopt;
+        return std::nullopt;
 
     int64_t userTime = timeValueToMicroseconds(threadInfoData.user_time);
     int64_t systemTime = timeValueToMicroseconds(threadInfoData.system_time);
@@ -63,7 +63,7 @@ std::optional<CPUTime> CPUTime::get()
     mach_msg_type_number_t taskInfoCount = TASK_BASIC_INFO_COUNT;
     result = task_info(mach_task_self(), TASK_BASIC_INFO, reinterpret_cast<task_info_t>(&taskInfoData), &taskInfoCount);
     if (result != KERN_SUCCESS)
-        return nullopt;
+        return std::nullopt;
 
     userTime += timeValueToMicroseconds(taskInfoData.user_time);
     systemTime += timeValueToMicroseconds(taskInfoData.system_time);
