diff --git a/dolphin/src/CMakeLists.txt b/dolphin/src/CMakeLists.txt
index 28768e2..4f6ee6d 100644
--- a/dolphin/src/CMakeLists.txt
+++ b/dolphin/src/CMakeLists.txt
@@ -368,3 +368,4 @@ install( FILES settings/services/servicemenu.knsrc DESTINATION ${CONFIG_INSTALL_
 #install( PROGRAMS settings/services/servicemenuinstallation DESTINATION ${BIN_INSTALL_DIR} )
 #install( PROGRAMS settings/services/servicemenudeinstallation DESTINATION ${BIN_INSTALL_DIR} )
 
+kde4_install_icons( ${ICON_INSTALL_DIR}  )
diff --git a/konqueror/CMakeLists.txt b/konqueror/CMakeLists.txt
index 18045cc..c40c4ad 100644
--- a/konqueror/CMakeLists.txt
+++ b/konqueror/CMakeLists.txt
@@ -33,3 +33,5 @@ install(FILES profile_kde_devel.desktop      DESTINATION ${DATA_INSTALL_DIR}/kon
 if(NOT WIN32)
    install(FILES profile_midnightcommander.desktop DESTINATION ${DATA_INSTALL_DIR}/konqueror/profiles/ RENAME midnightcommander)
 endif(NOT WIN32)
+
+kde4_install_icons( ${ICON_INSTALL_DIR}  )
diff --git a/konqueror/src/CMakeLists.txt b/konqueror/src/CMakeLists.txt
index b3ef9f9..ddbd925 100644
--- a/konqueror/src/CMakeLists.txt
+++ b/konqueror/src/CMakeLists.txt
@@ -92,6 +92,7 @@ qt4_add_dbus_adaptor(konqueror_KDEINIT_SRCS org.kde.Konqueror.SessionManager.xml
 qt4_add_dbus_interface(konqueror_KDEINIT_SRCS org.kde.Konqueror.SessionManager.xml konqsessionmanager_interface)
 
 kde4_add_app_icon(konqueror_KDEINIT_SRCS "${KDE4_ICON_INSTALL_DIR}/oxygen/*/apps/konqueror.png")
+kde4_install_icons( ${ICON_INSTALL_DIR}  )
 kde4_add_kdeinit_executable(konqueror ${konqueror_KDEINIT_SRCS} konqmain.cpp)
 # libkfile is only needed because of KNewFileMenu
 target_link_libraries(kdeinit_konqueror konquerorprivate konq ${KDE4_KPARTS_LIBS} ${KDE4_KCMUTILS_LIBS} ${KDE4_KFILE_LIBS})
