--- orig.configure	2012-06-28 16:14:25.000000000 +0200
+++ configure	2016-11-26 13:33:38.817378090 +0100
@@ -20306,30 +20306,13 @@
 
 
 # Check whether --enable-spectrogram or --disable-spectrogram was given.
-if test "${enable_spectrogram+set}" = set; then
-  enableval="$enable_spectrogram"
-   if test "$enableval" = "yes"; then
-      if test "x$with_x" != xyes; then
-	{ { echo "$as_me:$LINENO: error: spectrogram: --with-x option must be specified" >&5
-echo "$as_me: error: spectrogram: --with-x option must be specified" >&2;}
-   { (exit 1); exit 1; }; }
-      fi
-
 cat >>confdefs.h <<\_ACEOF
 #define SUPPORT_SOUNDSPEC 1
 _ACEOF
 
-    fi
-fi;
-
-
-if test "x$enable_spectrogram" = "xyes"; then
   ENABLE_SOUND_SPEC_TRUE=
   ENABLE_SOUND_SPEC_FALSE='#'
-else
-  ENABLE_SOUND_SPEC_TRUE='#'
-  ENABLE_SOUND_SPEC_FALSE=
-fi
+LIBS="-lX11  $LIBS"
 
 
 # Check whether --enable-spline or --disable-spline was given.
