--- src/kpassworddialog.ui.orig	2017-09-03 10:16:09.000000000 +0200
+++ src/kpassworddialog.ui	2017-10-01 10:59:30.597185589 +0200
@@ -2,7 +2,16 @@
 <ui version="4.0">
  <class>KPasswordDialog</class>
  <widget class="QDialog" name="KPasswordDialog">
+  <property name="sizePolicy">
+   <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
+    <horstretch>0</horstretch>
+    <verstretch>0</verstretch>
+   </sizepolicy>
+  </property>
   <layout class="QVBoxLayout" name="vBoxLayout">
+   <property name="sizeConstraint">
+    <enum>QLayout::SetFixedSize</enum>
+   </property>
    <item>
     <layout class="QHBoxLayout" name="hboxLayout">
      <item>
@@ -35,7 +44,7 @@
       <enum>Qt::Vertical</enum>
      </property>
      <property name="sizeType">
-      <enum>QSizePolicy::Fixed</enum>
+      <enum>QSizePolicy::Ignored</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
