diff --git a/daemon/corewlan.m b/daemon/corewlan.m
index 7bce5307aa2749a98c1ad05ca5423902a23028ba..a244c6d535fe887664c55e4f7b8eb7e40d4a6b20 100644
--- a/daemon/corewlan.m
+++ b/daemon/corewlan.m
@@ -27,7 +27,11 @@
 static struct wlan_state {
 	CWWiFiClient *client;
 	CWInterface *iface;
-	NSSet<CWChannel *> *supportedChannels;
+#if defined(MAC_OS_X_VERSION_10_10) && MAC_OS_X_VERSION_MIN_REQUIRED >= MAC_OS_X_VERSION_10_10
+    NSSet<CWChannel *> *supportedChannels;
+#else
+    NSSet *supportedChannels;
+#endif
 } state;
 
 int corewlan_init() {
