diff --git a/CMakeLists.txt b/CMakeLists.txt
index 562e20b..61e4ed2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -122,7 +122,10 @@ if(UMASK_EXEC)
 endif()
 
 if(NOT CMAKE_BUILD_TYPE)
-  set(CMAKE_BUILD_TYPE "Release")
+  set(CMAKE_BUILD_TYPE "MacPorts")
+  message(STATUS "Build type set to ${CMAKE_BUILD_TYPE}")
+else()
+  message(STATUS "Build type is ${CMAKE_BUILD_TYPE}")
 endif()
 
 if(NOT BUILD_SHARED_LIBS)
@@ -184,8 +187,8 @@ if(CMAKE_COMPILER_IS_GNUCC)
     endif()
   endif()
 
-  if(CMAKE_BUILD_TYPE STREQUAL "Release")
-    add_definitions(-DNDEBUG)
+  if(CMAKE_BUILD_TYPE STREQUAL "MacPorts")
+    #add_definitions(-DNDEBUG)
     check_c_compiler_flag(-Wno-unused-result Wno-unused-result)
     if(Wno-unused-result)
       set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-unused-result")
@@ -237,8 +240,8 @@ else()
 endif()
 
 if(${CMAKE_C_COMPILER_ID} STREQUAL "Clang")
-  if(CMAKE_BUILD_TYPE STREQUAL "Release")
-    add_definitions(-DNDEBUG)
+  if(CMAKE_BUILD_TYPE STREQUAL "MacPorts")
+    #add_definitions(-DNDEBUG)
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-unused-parameter")
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-unused-macros -Wno-padded")
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-c11-extensions -Wno-gnu")
@@ -431,6 +434,7 @@ if(wayland-client_FOUND)
       ${GIO_LIBRARY}
       ${Gmodule_LIBRARIES}
       ${GMODULE_LIBRARY}
+      -L${wayland-client_LIBRARY_DIRS}
       ${wayland-client_LIBRARIES})
   include_directories(SYSTEM ${wayland-client_INCLUDE_DIRS})
   message(STATUS "Wayland library found.")
@@ -599,14 +603,14 @@ configure_file(${CMAKE_CURRENT_SOURCE_DIR}/buildflags.h.in
                ${CMAKE_CURRENT_BINARY_DIR}/generated/buildflags.h)
 include_directories(${CMAKE_CURRENT_BINARY_DIR}/generated/)
 
-# Configure RPATH
-set(CMAKE_SKIP_BUILD_RPATH FALSE)
-set(CMAKE_BUILD_WITH_INSTALL_RPATH FALSE)
-set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
-set(CMAKE_BUILD_RPATH_USE_ORIGIN TRUE)
-if(NOT FREEBSD)
-  set(CMAKE_INSTALL_RPATH "\$ORIGIN/../${CMAKE_INSTALL_LIBDIR}:\$ORIGIN/..")
-endif()
+# # Configure RPATH
+# set(CMAKE_SKIP_BUILD_RPATH FALSE)
+# set(CMAKE_BUILD_WITH_INSTALL_RPATH FALSE)
+# set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
+# set(CMAKE_BUILD_RPATH_USE_ORIGIN TRUE)
+# if(NOT FREEBSD)
+#   set(CMAKE_INSTALL_RPATH "\$ORIGIN/../${CMAKE_INSTALL_LIBDIR}:\$ORIGIN/..")
+# endif()
 
 if(GTK3_FOUND)
   # pkg_check_modules(gio REQUIRED gio-2.0) pkg_check_modules(gio-unix REQUIRED
