diff --git plugins/patchreview/patchreview.cpp plugins/patchreview/patchreview.cpp
index 27fc276..1201946 100644
--- plugins/patchreview/patchreview.cpp
+++ plugins/patchreview/patchreview.cpp
@@ -384,6 +384,7 @@ void PatchReviewPlugin::closeReview()
                 ICore::self()->uiController()->switchToArea( QStringLiteral("code"), KDevelop::IUiController::ThisWindow );
         }
     }
+    m_active =false;
 }
 
 void PatchReviewPlugin::cancelReview() {
@@ -403,6 +404,7 @@ void PatchReviewPlugin::startReview( IPatchSource* patch, IPatchReview::ReviewMo
     Q_UNUSED( mode );
     emit startingNewReview();
     setPatch( patch );
+    m_active = true;
     QMetaObject::invokeMethod( this, "updateReview", Qt::QueuedConnection );
 }
 
@@ -522,7 +524,8 @@ void PatchReviewPlugin::setPatch( IPatchSource* patch ) {
 
 PatchReviewPlugin::PatchReviewPlugin( QObject *parent, const QVariantList & )
     : KDevelop::IPlugin( QStringLiteral("kdevpatchreview"), parent ),
-    m_patch( nullptr ), m_factory( new PatchReviewToolViewFactory( this ) )
+    m_patch( nullptr ), m_factory( new PatchReviewToolViewFactory( this ) ),
+    m_active( false)
 {
     qRegisterMetaType<const Diff2::DiffModel*>( "const Diff2::DiffModel*" );
 
@@ -552,7 +555,9 @@ PatchReviewPlugin::PatchReviewPlugin( QObject *parent, const QVariantList & )
 }
 
 void PatchReviewPlugin::documentClosed( IDocument* doc ) {
-    removeHighlighting( doc->url() );
+    if (m_active) {
+        removeHighlighting( doc->url() );
+    }
 }
 
 void PatchReviewPlugin::documentSaved( IDocument* doc ) {
@@ -562,12 +567,12 @@ void PatchReviewPlugin::documentSaved( IDocument* doc ) {
     // Also, don't automatically update local patch sources, because
     // they may correspond to static files which don't match any more
     // after an edit was done.
-    if( m_patch && doc->url() != m_patch->file() && !dynamic_cast<LocalPatchSource*>(m_patch.data()) )
+    if( m_active && m_patch && doc->url() != m_patch->file() && !dynamic_cast<LocalPatchSource*>(m_patch.data()) )
         forceUpdate();
 }
 
 void PatchReviewPlugin::textDocumentCreated( IDocument* doc ) {
-    if (m_patch) {
+    if (m_active && m_patch) {
         addHighlighting( doc->url(), doc );
     }
 }
diff --git plugins/patchreview/patchreview.h plugins/patchreview/patchreview.h
index 428994d..101ea91 100644
--- plugins/patchreview/patchreview.h
+++ plugins/patchreview/patchreview.h
@@ -134,6 +134,7 @@ private:
     uint m_depth = 0; // depth of the patch represented by m_modelList
     typedef QMap< QUrl, QPointer< PatchHighlighter > > HighlightMap;
     HighlightMap m_highlighters;
+    bool m_active;
 
     friend class PatchReviewToolView; // to access slot exporterSelected();
 };
