diff --git src/CMakeLists.txt src/CMakeLists.txt
index 18777ac4..2f258f68 100644
--- src/CMakeLists.txt
+++ src/CMakeLists.txt
@@ -236,7 +236,9 @@ PUBLIC
 )
 # optional dependencies
 if(Poppler_Qt5_FOUND)
-	target_link_libraries(kdeinit_kile PUBLIC Poppler::Qt5)
+     set_target_properties(kdeinit_kile PROPERTIES CXX_STANDARD 17)
+	target_link_libraries(kdeinit_kile PUBLIC ${Poppler_Qt5_LIBRARY})
+     target_include_directories(kdeinit_kile PRIVATE ${Poppler_Core_INCLUDE_DIR} ${Poppler_Qt5_INCLUDE_DIR})
 endif()
 
 if(Okular5_FOUND)
