diff --git gtk2/common/common.h gtk2/common/common.h
index ebbdbffa2dabb62a4e5e6c6f1066b41078b9050f..e550cb603a19447222a1a98e268ef21eaa0b6b47 100644
--- gtk2/common/common.h
+++ gtk2/common/common.h
@@ -703,9 +703,10 @@ typedef struct {
                      noMenuBgndOpacityApps,
                      noBgndImageApps;
     Strings          noMenuStripeApps;
-    // for completeness:
+    // for completeness, unused:
     bool             onlyTicksInMenu;
     bool             buttonStyleMenuSections;
+    int              menuCloseDelay;
 } Options;
 
 void qtcSetupGradient(Gradient *grad, EGradientBorder border, int numStops, ...);
diff --git gtk2/common/config_file.cpp gtk2/common/config_file.cpp
index 46fce5fb4bc05f10dc8d7462b5932ea3c6c9bab0..a343de665c664fd1167c448be10d409218f2493f 100644
--- gtk2/common/config_file.cpp
+++ gtk2/common/config_file.cpp
@@ -1241,6 +1241,7 @@ bool qtcReadConfig(const char *file, Options *opts, Options *defOpts)
             CFG_READ_ROUND(round);
             CFG_READ_INT(highlightFactor);
             CFG_READ_INT(menuDelay);
+            CFG_READ_INT(menuCloseDelay);
             CFG_READ_INT(sliderWidth);
             CFG_READ_INT(tabBgnd);
             CFG_READ_TB_BORDER(toolbarBorders);
@@ -1613,6 +1614,7 @@ void qtcDefaultSettings(Options *opts)
     opts->splitterHighlight=DEFAULT_SPLITTER_HIGHLIGHT_FACTOR;
     opts->crSize=CR_LARGE_SIZE;
     opts->menuDelay=DEFAULT_MENU_DELAY;
+    opts->menuCloseDelay=-1; // unused
     opts->sliderWidth=DEFAULT_SLIDER_WIDTH;
     opts->selectionAppearance=APPEARANCE_HARSH_GRADIENT;
     opts->fadeLines=true;
diff --git qt4/common/common.h qt4/common/common.h
index 25b16275f6cc9ea643bfd5779f28a548b55d51c8..1768a92e98ec73d77bcbfd694fd5a32b0027f7da 100644
--- qt4/common/common.h
+++ qt4/common/common.h
@@ -155,6 +155,10 @@ enum {
 #define MIN_MENU_DELAY       1
 #define MAX_MENU_DELAY     500
 
+#define DEFAULT_MENU_CLOSE_DELAY 300
+#define MIN_MENU_CLOSE_DELAY       0
+#define MAX_MENU_CLOSE_DELAY    5000
+
 #define DEFAULT_SLIDER_WIDTH  15
 #define MIN_SLIDER_WIDTH_ROUND 7
 #define MIN_SLIDER_WIDTH_THIN_GROOVE 9
@@ -723,6 +727,7 @@ typedef struct {
                      highlightFactor,
                      lighterPopupMenuBgnd,
                      menuDelay,
+                     menuCloseDelay,
                      sliderWidth,
                      tabBgnd,
                      colorSelTab,
diff --git qt4/common/config_file.cpp qt4/common/config_file.cpp
index 1982cda56ba76a6a6b6030b09b2ee3f9b4d6f218..09d43d8122d43e20e60319b3b0e4f26c3b6ffd71 100644
--- qt4/common/config_file.cpp
+++ qt4/common/config_file.cpp
@@ -933,6 +933,9 @@ void qtcCheckConfig(Options *opts)
     else if(opts->menuDelay<MIN_MENU_DELAY || opts->menuDelay>MAX_MENU_DELAY)
         opts->menuDelay=DEFAULT_MENU_DELAY;
 
+    if(opts->menuCloseDelay<MIN_MENU_CLOSE_DELAY || opts->menuCloseDelay>MAX_MENU_CLOSE_DELAY)
+        opts->menuCloseDelay=DEFAULT_MENU_CLOSE_DELAY;
+
     if(0==opts->sliderWidth%2)
         opts->sliderWidth++;
 
@@ -1166,6 +1169,7 @@ bool qtcReadConfig(const QString &file, Options *opts, Options *defOpts, bool ch
             CFG_READ_ROUND(round);
             CFG_READ_INT(highlightFactor);
             CFG_READ_INT(menuDelay);
+            CFG_READ_INT(menuCloseDelay);
             CFG_READ_INT(sliderWidth);
             CFG_READ_INT(tabBgnd);
             CFG_READ_TB_BORDER(toolbarBorders);
@@ -1461,6 +1465,7 @@ void qtcDefaultSettings(Options *opts)
     opts->splitterHighlight=DEFAULT_SPLITTER_HIGHLIGHT_FACTOR;
     opts->crSize=CR_LARGE_SIZE;
     opts->menuDelay=DEFAULT_MENU_DELAY;
+    opts->menuCloseDelay=DEFAULT_MENU_CLOSE_DELAY;
     opts->sliderWidth=DEFAULT_SLIDER_WIDTH;
     opts->selectionAppearance=APPEARANCE_HARSH_GRADIENT;
     opts->fadeLines=true;
@@ -2192,6 +2197,7 @@ bool qtcWriteConfig(KConfig *cfg, const Options &opts, const Options &def, bool
         CFG_WRITE_ENTRY(round);
         CFG_WRITE_ENTRY_NUM(highlightFactor);
         CFG_WRITE_ENTRY_NUM(menuDelay);
+        CFG_WRITE_ENTRY_NUM(menuCloseDelay);
         CFG_WRITE_ENTRY_NUM(sliderWidth);
         CFG_WRITE_ENTRY(toolbarBorders);
         CFG_WRITE_APPEARANCE_ENTRY(appearance, APP_ALLOW_BASIC);
diff --git qt4/config/qtcurveconfig.cpp qt4/config/qtcurveconfig.cpp
index f04348439801e4cecb88bbee18b6de8151929cf9..d6f108ee84da772a7f668b6fc3d1dee1533ebe8d 100644
--- qt4/config/qtcurveconfig.cpp
+++ qt4/config/qtcurveconfig.cpp
@@ -948,6 +948,9 @@ QtCurveConfig::QtCurveConfig(QWidget *parent)
     menuDelay->setRange(MIN_MENU_DELAY, MAX_MENU_DELAY);
     menuDelay->setValue(DEFAULT_MENU_DELAY);
 
+    menuCloseDelay->setRange(MIN_MENU_CLOSE_DELAY, MAX_MENU_CLOSE_DELAY);
+    menuCloseDelay->setValue(DEFAULT_MENU_CLOSE_DELAY);
+
     gbFactor->setRange(MIN_GB_FACTOR, MAX_GB_FACTOR);
     gbFactor->setValue(DEF_GB_FACTOR);
 
@@ -985,6 +988,7 @@ QtCurveConfig::QtCurveConfig(QWidget *parent)
     connect(lighterPopupMenuBgnd, SIGNAL(valueChanged(int)), SLOT(updateChanged()));
     connect(tabBgnd, SIGNAL(valueChanged(int)), SLOT(updateChanged()));
     connect(menuDelay, SIGNAL(valueChanged(int)), SLOT(updateChanged()));
+    connect(menuCloseDelay, SIGNAL(valueChanged(int)), SLOT(updateChanged()));
     connect(sliderWidth, SIGNAL(valueChanged(int)), SLOT(sliderWidthChanged()));
     connect(menuStripe, SIGNAL(currentIndexChanged(int)), SLOT(menuStripeChanged()));
     connect(customMenuStripeColor, SIGNAL(changed(const QColor &)), SLOT(updateChanged()));
@@ -2980,6 +2984,7 @@ void QtCurveConfig::setOptions(Options &opts)
     opts.lighterPopupMenuBgnd=lighterPopupMenuBgnd->value();
     opts.tabBgnd=tabBgnd->value();
     opts.menuDelay=menuDelay->value();
+    opts.menuCloseDelay=menuCloseDelay->value();
     opts.sliderWidth=sliderWidth->value();
     opts.menuStripe=(EShade)menuStripe->currentIndex();
     opts.customMenuStripeColor=customMenuStripeColor->color();
@@ -3222,6 +3227,7 @@ void QtCurveConfig::setWidgetOptions(const Options &opts)
     lighterPopupMenuBgnd->setValue(opts.lighterPopupMenuBgnd);
     tabBgnd->setValue(opts.tabBgnd);
     menuDelay->setValue(opts.menuDelay);
+    menuCloseDelay->setValue(opts.menuCloseDelay);
     sliderWidth->setValue(opts.sliderWidth);
     menuStripe->setCurrentIndex(opts.menuStripe);
     customMenuStripeColor->setColor(opts.customMenuStripeColor);
@@ -3660,6 +3666,7 @@ bool QtCurveConfig::settingsChanged(const Options &opts)
          lighterPopupMenuBgnd->value()!=opts.lighterPopupMenuBgnd ||
          tabBgnd->value()!=opts.tabBgnd ||
          menuDelay->value()!=opts.menuDelay ||
+         menuCloseDelay->value()!=opts.menuCloseDelay ||
          sliderWidth->value()!=opts.sliderWidth ||
          menuStripe->currentIndex()!=opts.menuStripe ||
          menuStripeAppearance->currentIndex()!=opts.menuStripeAppearance ||
diff --git qt4/config/qtcurveconfigbase.ui qt4/config/qtcurveconfigbase.ui
index 658d4a3d9380eef425921278890f9dd0086ea55a..11666e2fee8c1b325abda82b0278fbac0bd3546f 100644
--- qt4/config/qtcurveconfigbase.ui
+++ qt4/config/qtcurveconfigbase.ui
@@ -3265,21 +3265,38 @@
          </property>
         </widget>
        </item>
+        <item row="11" column="1">
+        <widget class="KIntNumInput" name="menuCloseDelay">
+         <property name="toolTip">
+          <string>&lt;p&gt;Defines the delay before a sub-menu closes. Applicable on Qt 5.5 or later</string>
+         </property>
+         <property name="suffix">
+          <string>ms</string>
+         </property>
+        </widget>
+       </item>
        <item row="11" column="0">
+        <widget class="QLabel" name="label_103">
+         <property name="text">
+          <string>Sub-menu close delay:</string>
+         </property>
+        </widget>
+       </item>
+       <item row="12" column="0">
         <widget class="QLabel" name="label_44">
          <property name="text">
           <string>Stripe:</string>
          </property>
         </widget>
        </item>
-       <item row="11" column="1">
+       <item row="12" column="1">
         <widget class="QComboBox" name="menuStripe">
          <property name="toolTip">
           <string>&lt;p&gt;Defines the coloration of the menu-stripe that will be drawn on the left of popup menus - but only if this option is not set to 'none'.</string>
          </property>
         </widget>
        </item>
-       <item row="12" column="1">
+       <item row="13" column="1">
         <widget class="KColorButton" name="customMenuStripeColor">
          <property name="enabled">
           <bool>false</bool>
@@ -3295,7 +3312,7 @@
          </property>
         </widget>
        </item>
-       <item row="13" column="1">
+       <item row="14" column="1">
         <widget class="QComboBox" name="menuStripeAppearance">
          <property name="enabled">
           <bool>false</bool>
@@ -3305,14 +3322,14 @@
          </property>
         </widget>
        </item>
-       <item row="14" column="0">
+       <item row="15" column="0">
         <widget class="QLabel" name="label_61">
          <property name="text">
           <string>Border popupmenus:</string>
          </property>
         </widget>
        </item>
-       <item row="14" column="1">
+       <item row="15" column="1">
         <widget class="QCheckBox" name="popupBorder">
          <property name="toolTip">
           <string>&lt;p&gt;If enabled, the whole popup menu will have a border, otherwise no border will be drawn. This option should be enabled unless you have also enabled shadows in your window manager.</string>
@@ -3322,14 +3339,14 @@
          </property>
         </widget>
        </item>
-       <item row="15" column="0">
+       <item row="16" column="0">
         <widget class="QLabel" name="label_97">
          <property name="text">
           <string>Use menubar colour settings:</string>
          </property>
         </widget>
        </item>
-       <item row="15" column="1">
+       <item row="16" column="1">
         <widget class="QCheckBox" name="shadePopupMenu">
          <property name="text">
           <string>Enabled</string>
diff --git qt5/common/common.h qt5/common/common.h
index 5fbbfac09a666389ce6645aebeda158e170eaeaf..9cae0dd5341e8eb82722b61410b6a33d6710a2d2 100644
--- qt5/common/common.h
+++ qt5/common/common.h
@@ -154,6 +154,10 @@ enum {
 #define MIN_MENU_DELAY       1
 #define MAX_MENU_DELAY     500
 
+#define DEFAULT_MENU_CLOSE_DELAY 300
+#define MIN_MENU_CLOSE_DELAY       0
+#define MAX_MENU_CLOSE_DELAY    5000
+
 #define DEFAULT_SLIDER_WIDTH  15
 #define MIN_SLIDER_WIDTH_ROUND 7
 #define MIN_SLIDER_WIDTH_THIN_GROOVE 9
@@ -717,6 +721,7 @@ typedef struct {
                      highlightFactor,
                      lighterPopupMenuBgnd,
                      menuDelay,
+                     menuCloseDelay,
                      sliderWidth,
                      tabBgnd,
                      colorSelTab,
diff --git qt5/common/config_file.cpp qt5/common/config_file.cpp
index 4ab1e095607352e366e1f996aba505647ea63b13..59fafcdc4ca482da22dcb963d374a40030e965e2 100644
--- qt5/common/config_file.cpp
+++ qt5/common/config_file.cpp
@@ -942,6 +942,9 @@ void qtcCheckConfig(Options *opts)
     else if(opts->menuDelay<MIN_MENU_DELAY || opts->menuDelay>MAX_MENU_DELAY)
         opts->menuDelay=DEFAULT_MENU_DELAY;
 
+    if(opts->menuCloseDelay<MIN_MENU_CLOSE_DELAY || opts->menuCloseDelay>MAX_MENU_CLOSE_DELAY)
+        opts->menuCloseDelay=DEFAULT_MENU_CLOSE_DELAY;
+
     if(0==opts->sliderWidth%2)
         opts->sliderWidth++;
 
@@ -1173,6 +1176,7 @@ bool qtcReadConfig(const QString &file, Options *opts, Options *defOpts, bool ch
             CFG_READ_ROUND(round);
             CFG_READ_INT(highlightFactor);
             CFG_READ_INT(menuDelay);
+            CFG_READ_INT(menuCloseDelay);
             CFG_READ_INT(sliderWidth);
             CFG_READ_INT(tabBgnd);
             CFG_READ_TB_BORDER(toolbarBorders);
@@ -1476,6 +1480,7 @@ void qtcDefaultSettings(Options *opts)
     opts->splitterHighlight=DEFAULT_SPLITTER_HIGHLIGHT_FACTOR;
     opts->crSize=CR_LARGE_SIZE;
     opts->menuDelay=DEFAULT_MENU_DELAY;
+    opts->menuCloseDelay=DEFAULT_MENU_CLOSE_DELAY;
     opts->sliderWidth=DEFAULT_SLIDER_WIDTH;
     opts->selectionAppearance=APPEARANCE_HARSH_GRADIENT;
     opts->fadeLines=true;
@@ -2204,6 +2209,7 @@ bool qtcWriteConfig(KConfig *cfg, const Options &opts, const Options &def, bool
         CFG_WRITE_ENTRY(round);
         CFG_WRITE_ENTRY_NUM(highlightFactor);
         CFG_WRITE_ENTRY_NUM(menuDelay);
+        CFG_WRITE_ENTRY_NUM(menuCloseDelay);
         CFG_WRITE_ENTRY_NUM(sliderWidth);
         CFG_WRITE_ENTRY(toolbarBorders);
         CFG_WRITE_APPEARANCE_ENTRY(appearance, APP_ALLOW_BASIC);
diff --git qt5/config/qtcurveconfig.cpp qt5/config/qtcurveconfig.cpp
index f04e371af9114d6071cf699d4643114c8fb0bee4..e4af7354efff76a3e2261f622c332095539a2d3b 100644
--- qt5/config/qtcurveconfig.cpp
+++ qt5/config/qtcurveconfig.cpp
@@ -957,6 +957,9 @@ QtCurveConfig::QtCurveConfig(QWidget *parent)
     menuDelay->setRange(MIN_MENU_DELAY, MAX_MENU_DELAY);
     menuDelay->setValue(DEFAULT_MENU_DELAY);
 
+    menuCloseDelay->setRange(MIN_MENU_CLOSE_DELAY, MAX_MENU_CLOSE_DELAY);
+    menuCloseDelay->setValue(DEFAULT_MENU_CLOSE_DELAY);
+
     gbFactor->setRange(MIN_GB_FACTOR, MAX_GB_FACTOR);
     gbFactor->setValue(DEF_GB_FACTOR);
 
@@ -995,7 +998,7 @@ QtCurveConfig::QtCurveConfig(QWidget *parent)
                                                 CImagePropertiesDialog::POS|
                                                 CImagePropertiesDialog::SCALE);
 
-    for (auto *w: {lighterPopupMenuBgnd, tabBgnd, menuDelay, crHighlight,
+    for (auto *w: {lighterPopupMenuBgnd, tabBgnd, menuDelay, menuCloseDelay, crHighlight,
                 expanderHighlight, colorSelTab, highlightFactor, bgndOpacity,
                 dlgOpacity, menuBgndOpacity, splitterHighlight, gbFactor}) {
         connect(qtcSlot(w, valueChanged, (int)), qtcSlot(this, updateChanged));
@@ -2944,6 +2947,7 @@ void QtCurveConfig::setOptions(Options &opts)
     opts.lighterPopupMenuBgnd=lighterPopupMenuBgnd->value();
     opts.tabBgnd=tabBgnd->value();
     opts.menuDelay=menuDelay->value();
+    opts.menuCloseDelay=menuCloseDelay->value();
     opts.sliderWidth=sliderWidth->value();
     opts.menuStripe=(EShade)menuStripe->currentIndex();
     opts.customMenuStripeColor=customMenuStripeColor->color();
@@ -3188,6 +3192,7 @@ void QtCurveConfig::setWidgetOptions(const Options &opts)
     lighterPopupMenuBgnd->setValue(opts.lighterPopupMenuBgnd);
     tabBgnd->setValue(opts.tabBgnd);
     menuDelay->setValue(opts.menuDelay);
+    menuCloseDelay->setValue(opts.menuCloseDelay);
     sliderWidth->setValue(opts.sliderWidth);
     menuStripe->setCurrentIndex(opts.menuStripe);
     customMenuStripeColor->setColor(opts.customMenuStripeColor);
diff --git qt5/config/qtcurveconfigbase.ui qt5/config/qtcurveconfigbase.ui
index cff9f72c6ebc69367cfe7f3d8d534a670f91fb87..3c03918b2c152546204f51de983cffc4152c465d 100644
--- qt5/config/qtcurveconfigbase.ui
+++ qt5/config/qtcurveconfigbase.ui
@@ -3312,7 +3312,7 @@
        <item row="10" column="0">
         <widget class="QLabel" name="label_45">
          <property name="text">
-          <string>Sub-menu delay:</string>
+          <string>Sub-menu open delay:</string>
          </property>
         </widget>
        </item>
@@ -3329,21 +3329,21 @@
          </property>
         </widget>
        </item>
-       <item row="11" column="0">
+       <item row="12" column="0">
         <widget class="QLabel" name="label_44">
          <property name="text">
           <string>Stripe:</string>
          </property>
         </widget>
        </item>
-       <item row="11" column="1">
+       <item row="12" column="1">
         <widget class="QComboBox" name="menuStripe">
          <property name="toolTip">
           <string>&lt;p&gt;Defines the coloration of the menu-stripe that will be drawn on the left of popup menus - but only if this option is not set to 'none'.</string>
          </property>
         </widget>
        </item>
-       <item row="12" column="1">
+       <item row="13" column="1">
         <widget class="KColorButton" name="customMenuStripeColor">
          <property name="enabled">
           <bool>false</bool>
@@ -3359,7 +3359,7 @@
          </property>
         </widget>
        </item>
-       <item row="13" column="1">
+       <item row="14" column="1">
         <widget class="QComboBox" name="menuStripeAppearance">
          <property name="enabled">
           <bool>false</bool>
@@ -3369,14 +3369,14 @@
          </property>
         </widget>
        </item>
-       <item row="14" column="0">
+       <item row="15" column="0">
         <widget class="QLabel" name="label_61">
          <property name="text">
           <string>Border popupmenus:</string>
          </property>
         </widget>
        </item>
-       <item row="14" column="1">
+       <item row="15" column="1">
         <widget class="QCheckBox" name="popupBorder">
          <property name="toolTip">
           <string>&lt;p&gt;If enabled, the whole popup menu will have a border, otherwise no border will be drawn. This option should be enabled unless you have also enabled shadows in your window manager.</string>
@@ -3386,14 +3386,14 @@
          </property>
         </widget>
        </item>
-       <item row="15" column="0">
+       <item row="16" column="0">
         <widget class="QLabel" name="label_97">
          <property name="text">
           <string>Use menubar colour settings:</string>
          </property>
         </widget>
        </item>
-       <item row="15" column="1">
+       <item row="16" column="1">
         <widget class="QCheckBox" name="shadePopupMenu">
          <property name="text">
           <string>Enabled</string>
@@ -3413,6 +3413,23 @@
          </property>
         </widget>
        </item>
+       <item row="11" column="1">
+        <widget class="QSpinBox" name="menuCloseDelay">
+         <property name="toolTip">
+          <string>Defines the delay before a sub-menu closes. Applicable on Qt5.5 or later</string>
+         </property>
+         <property name="suffix">
+          <string>ms</string>
+         </property>
+        </widget>
+       </item>
+       <item row="11" column="0">
+        <widget class="QLabel" name="label_72">
+         <property name="text">
+          <string>Sub-menu close delay:</string>
+         </property>
+        </widget>
+       </item>
       </layout>
      </widget>
      <widget class="QWidget" name="page_22">
diff --git qt5/style/qtcurve_api.cpp qt5/style/qtcurve_api.cpp
index 73236ddd79bc54e0dddb4b83809275ddb81d34c8..54efc6461dd1a9bfbc6ca0215a99ed985d769652 100644
--- qt5/style/qtcurve_api.cpp
+++ qt5/style/qtcurve_api.cpp
@@ -1812,6 +1812,12 @@ Style::styleHint(StyleHint hint, const QStyleOption *option,
     }
     case SH_Menu_SubMenuPopupDelay:
         return opts.menuDelay;
+#if QT_VERSION >= QT_VERSION_CHECK(5,5,0)
+    case SH_Menu_SloppySubMenus:
+        return opts.menuCloseDelay > 0;
+    case SH_Menu_SubMenuSloppyCloseTimeout:
+        return opts.menuCloseDelay;
+#endif
     case SH_ToolButton_PopupDelay:
         return 250;
     case SH_ComboBox_PopupFrameStyle:
