diff --git qt5/style/qtcurve_plugin.cpp qt5/style/qtcurve_plugin.cpp
index 481fffca..182563b9 100644
--- qt5/style/qtcurve_plugin.cpp
+++ qt5/style/qtcurve_plugin.cpp
@@ -156,12 +156,14 @@ StylePlugin::~StylePlugin()
     qtcInfo("Deleting QtCurve plugin (%p)\n", this);
     if (!m_styleInstances.isEmpty()) {
         qtcWarn("there remain(s) %d Style instance(s)\n", m_styleInstances.count());
-        foreach (Style *that, m_styleInstances) {
-            // don't let ~Style() touch m_styleInstances from here.
-            that->m_plugin = nullptr;
-            // each instance should already have disconnected from the D-Bus
-            // and disconnected from receiving select signals.
-            delete that;
+        if (qApp->applicationName() != "rkward") {
+            foreach (Style *that, m_styleInstances) {
+                // don't let ~Style() touch m_styleInstances from here.
+                that->m_plugin = nullptr;
+                // each instance should already have disconnected from the D-Bus
+                // and disconnected from receiving select signals.
+                delete that;
+            }
         }
         m_styleInstances.clear();
     }
