diff --git qt5/style/qtcurve_plugin.cpp qt5/style/qtcurve_plugin.cpp
index ce363ac25deff50de852dc00583e14b04c8adb36..20152ad0a9745618cb3aca7bbe535aa36068c554 100644
--- qt5/style/qtcurve_plugin.cpp
+++ qt5/style/qtcurve_plugin.cpp
@@ -129,6 +129,11 @@ StylePlugin::create(const QString &key)
     if (key.toLower() == "qtcurve") {
         qtc = new Style;
         qtc->m_plugin = this;
+        foreach (Style *that, m_styleInstances) {
+            qtcWarn("severing DBus connections and signals from Style instance %p\n", that);
+            that->disconnectDBus();
+            that->disconnect();
+        }
         m_styleInstances << qtc;
     } else {
         qtc = nullptr;
