diff --git qt5/style/qtcurve_api.cpp qt5/style/qtcurve_api.cpp
index 642022a35c655a10a47b4ff5747ada32c7cbe55e..ce7e41aa7cdc7ded1976c572518cf6204e5b1f11 100644
--- qt5/style/qtcurve_api.cpp
+++ qt5/style/qtcurve_api.cpp
@@ -3043,6 +3043,27 @@ Style::drawControl(ControlElement element, const QStyleOption *option,
                                       opts.menuStripeAppearance, WIDGET_OTHER);
 
                 if (!menuItem->text.isEmpty()) {
+                    int textAlignment;
+                    QFont font(menuItem->font);
+                    QRect textRect;
+#if 1
+                    font.setBold(true);
+                    font.setUnderline(true);
+                    font.setLetterSpacing(QFont::PercentageSpacing, 95);
+                    painter->setFont(font);
+                    QRect miRect(menuItem->rect.left() + 5 +
+                                 (!reverse && doStripe ? stripeWidth : 0),
+                                 menuItem->rect.center().y() - painter->fontMetrics().ascent() / 2 - 5,
+                                 menuItem->rect.width() -
+                                 (9 + (doStripe ? stripeWidth : 0)), 1);
+                    QColor lineCol = use[MENU_SEP_SHADE];
+                    if (miRect.y() >= rx.height() / 2) {
+                        drawFadedLine(painter, miRect, lineCol, reverse,
+                                  !reverse, true);
+                    }
+                    textAlignment = Qt::AlignLeft | Qt::AlignBottom;
+                    textRect = r.adjusted(5, 5, -5, -5);
+#else
                     QStyleOption opt;
                     opt.rect = r.adjusted(2, 2, -3, -2);
                     opt.state = State_Raised | State_Enabled | State_Horizontal;
@@ -3050,12 +3070,13 @@ Style::drawControl(ControlElement element, const QStyleOption *option,
                                    getFill(&opt, use), use, true,
                                    WIDGET_NO_ETCH_BTN);
 
-                    QFont font(menuItem->font);
-
                     font.setBold(true);
                     painter->setFont(font);
-                    drawItemTextWithRole(painter, r,
-                                         Qt::AlignHCenter | Qt::AlignVCenter,
+                    textAlignment = Qt::AlignHCenter | Qt::AlignVCenter;
+                    textRect = r;
+#endif
+                    drawItemTextWithRole(painter, textRect,
+                                         textAlignment,
                                          palette, state & State_Enabled,
                                          menuItem->text, QPalette::Text);
                 } else {
