diff --git qt5/common/common.h qt5/common/common.h
index bb103fd..a23be01 100644
--- qt5/common/common.h
+++ qt5/common/common.h
@@ -44,6 +44,7 @@
 #include <map>
 #include <set>
 #include <QString>
+#include <QFont>
 #include <QPixmap>
 #include <QSet>
 
@@ -878,6 +879,11 @@ typedef struct {
                      useQtFileDialogApps,
                      windowDragWhiteList,
                      windowDragBlackList;
+    bool             onlyTicksInMenu;
+    // these are runtime constants for the user-option "onlyTicksInMenu":
+    QFont            tickFont;
+    QString          menuTick;
+    int              fontTickWidth;
 } Options;
 
 void qtcSetupGradient(Gradient *grad, EGradientBorder border, int numStops, ...);
diff --git qt5/style/qtcurve.h qt5/style/qtcurve.h
index 6836b9d..d9395bc 100644
--- qt5/style/qtcurve.h
+++ qt5/style/qtcurve.h
@@ -30,6 +30,7 @@
 #include <QSet>
 #include <QCache>
 #include <QColor>
+#include <QFont>
 #include <QStyleOption>
 #include <QtGlobal>
 #include <QCommonStyle>
diff --git qt5/style/qtcurve.cpp qt5/style/qtcurve.cpp
index b2a7738..c4c1d13 100644
--- qt5/style/qtcurve.cpp
+++ qt5/style/qtcurve.cpp
@@ -647,6 +647,9 @@ void Style::init(bool initial)
                              opts.dlgOpacity != 100 ||
                              opts.menuBgndOpacity != 100);
 
+    opts.fontTickWidth=-1;
+    opts.menuTick=QString(QChar(0x2713));
+
 #ifdef QTC_QT5_ENABLE_KDE
     // We need to set the decoration colours for the preview now...
     if (m_isPreview) {
diff --git qt5/common/config_file.cpp qt5/common/config_file.cpp
index 362381a..98c40fc 100644
--- qt5/common/config_file.cpp
+++ qt5/common/config_file.cpp
@@ -1275,6 +1275,7 @@ bool qtcReadConfig(const QString &file, Options *opts, Options *defOpts, bool ch
             CFG_READ_BOOL(thinSbarGroove);
             CFG_READ_BOOL(colorSliderMouseOver);
             CFG_READ_BOOL(menuIcons);
+            CFG_READ_BOOL(onlyTicksInMenu);
             CFG_READ_BOOL(forceAlternateLvCols);
             CFG_READ_BOOL(invertBotTab);
             CFG_READ_INT_BOOL(menubarHiding, HIDE_KEYBOARD);
@@ -1624,6 +1625,12 @@ void qtcDefaultSettings(Options *opts)
     opts->titlebarAppearance=APPEARANCE_CUSTOM1;
     opts->inactiveTitlebarAppearance=APPEARANCE_CUSTOM1;
     opts->titlebarButtonAppearance=APPEARANCE_GRADIENT;
+#ifdef Q_OS_OSX
+    opts->onlyTicksInMenu=true;
+#else
+    opts->onlyTicksInMenu=false;
+#endif
+
     /* Read system config file... */
     {
     static const char * systemFilename=nullptr;
@@ -2292,6 +2299,7 @@ bool qtcWriteConfig(KConfig *cfg, const Options &opts, const Options &def, bool
         CFG_WRITE_ENTRY(thinSbarGroove);
         CFG_WRITE_ENTRY(colorSliderMouseOver);
         CFG_WRITE_ENTRY(menuIcons);
+        CFG_WRITE_ENTRY(onlyTicksInMenu);
         CFG_WRITE_ENTRY(forceAlternateLvCols);
         CFG_WRITE_ENTRY_NUM(square);
         CFG_WRITE_ENTRY(invertBotTab);
diff --git qt5/style/qtcurve_api.cpp qt5/style/qtcurve_api.cpp
index b8535da..b747587 100644
--- qt5/style/qtcurve_api.cpp
+++ qt5/style/qtcurve_api.cpp
@@ -47,6 +47,7 @@
 #include <QCheckBox>
 #include <QRadioButton>
 #include <QTextEdit>
+#include <QFontMetrics>
 #include <QDial>
 #include <QLabel>
 #include <QStackedLayout>
@@ -3018,6 +3019,21 @@ Style::drawControl(ControlElement element, const QStyleOption *option,
             bool checked = menuItem->checked;
             bool enabled = state & State_Enabled;
 
+            if (checkable && opts.onlyTicksInMenu) {
+                if (opts.fontTickWidth <= 0) {
+                    opts.tickFont = QFont(widget->font());
+                    opts.fontTickWidth = QFontMetrics(opts.tickFont).width(opts.menuTick);
+                    qWarning() << "menuFont:" << opts.tickFont.toString() << "tickMark:" << opts.menuTick
+                        << "width=" << opts.fontTickWidth << "/" << QFontMetrics(opts.tickFont).boundingRect(opts.menuTick).width();
+                }
+                // check if the font tick is wider than the current margin
+                // and adjust if necessary
+                if (opts.fontTickWidth > checkcol) {
+                    int dx = opts.fontTickWidth - checkcol;
+                    r.adjust(-dx, 0, dx, 0);
+                    checkcol += dx;
+                }
+            }
             if (opts.menuStripe && !comboMenu)
                 drawBevelGradient(menuStripeCol(), painter,
                                   QRect(reverse ? r.right()-stripeWidth : r.x(),
@@ -3046,7 +3062,7 @@ Style::drawControl(ControlElement element, const QStyleOption *option,
                 checkRect = visualRect(menuItem->direction, menuItem->rect,
                                        checkRect);
                 if (checkable) {
-                    if ((menuItem->checkType &
+                    if (!opts.onlyTicksInMenu && (menuItem->checkType &
                          QStyleOptionMenuItem::Exclusive) &&
                         menuItem->icon.isNull()) {
                         QStyleOptionButton button;
@@ -3065,7 +3081,7 @@ Style::drawControl(ControlElement element, const QStyleOption *option,
                             if (checked)
                                 button.state |= State_On;
                             button.palette = palette;
-                            drawPrimitive(PE_IndicatorCheckBox, &button,
+                            drawPrimitive(PE_IndicatorMenuCheckMark, &button,
                                           painter, widget);
                         } else if (checked) {
                             int iconSize = qMax(menuItem->maxIconWidth, 20);
@@ -3078,9 +3094,7 @@ Style::drawControl(ControlElement element, const QStyleOption *option,
                             sunkenRect = visualRect(menuItem->direction,
                                                     menuItem->rect, sunkenRect);
                             opt.state = menuItem->state;
-                            opt.state |= State_Raised | State_Horizontal;
-                            if (checked)
-                                opt.state |= State_On;
+                            opt.state |= State_Raised | State_Horizontal | State_On;
                             drawLightBevel(painter, sunkenRect, &opt, widget,
                                            ROUNDED_ALL,
                                            getFill(&opt, m_buttonCols),
diff --git qt5/style/qtcurve_primitive.cpp qt5/style/qtcurve_primitive.cpp
index a8a2bed..856218a 100644
--- qt5/style/qtcurve_primitive.cpp
+++ qt5/style/qtcurve_primitive.cpp
@@ -1658,6 +1658,8 @@ Style::drawPrimitiveIndicatorCheckBox(PrimitiveElement element,
                                  State_Selected));
     int crSize = opts.crSize + (doEtch ? 2 : 0);
     QRect rect(r.x(), r.y() + (view ? -1 : 0), crSize, crSize);
+    // render checkable menu items with only a tick if so requested
+    bool onlyTicksInMenu = menu && opts.onlyTicksInMenu;
 
     // For OO.o 3.2 need to fill widget background!
     if (isOO) {
@@ -1675,7 +1677,8 @@ Style::drawPrimitiveIndicatorCheckBox(PrimitiveElement element,
         if (r == QRect(0, 0, 15, 15)) {
             rect.adjust(-1, -1, -1, -1);
         }
-    } else {
+    } else if (!onlyTicksInMenu) {
+        // don't do this when rendering a regular menu so we get just a tick
         if (isOO && r == QRect(0, 0, opts.crSize, opts.crSize)) {
             rect.adjust(0, -1, 0, -1);
         }
@@ -1764,10 +1767,44 @@ Style::drawPrimitiveIndicatorCheckBox(PrimitiveElement element,
         }
     }
     if (state & State_On || selectedOOMenu) {
-        QPixmap *pix = getPixmap(checkRadioCol(option), PIX_CHECK, 1.0);
+        if (onlyTicksInMenu) {
+            // only tickmarks (= without the box) in menu; adjust its horizontal position
+            // the normal, hardcoded margin is 20 (pixels). If that's too small, compensate,
+            // using a hand-tuned ad-hoc recipe tested over point size range 8 - 32 pt
+            if (opts.fontTickWidth > 20){
+                int dx = (opts.fontTickWidth - 20) + 3;
+                rect.adjust(dx, -dx, 2 * dx, dx);
+            } else if (opts.fontTickWidth > 10){
+                int dx = (opts.fontTickWidth - 10) / 2;
+                rect.adjust(dx, -dx, dx, dx);
+            } else {
+                rect.adjust(6, 0, 0, 0);
+            }
+            // get the font for adjusting; widget will be the menu holding the menuitem
+            // but that uses the same font as the menuitem.
+            QFont font(widget->font());
+#ifndef Q_OS_OSX
+            font.setBold(true);
+#else
+            // not that there is much variation in the CheckMark symbol on OS X, but
+            // we pick one from a font that's always supposed to be there and that's
+            // a good choice functionally speaking.
+            font.setFamily(QStringLiteral("Apple Symbols"));
+#endif
+            // adjust the size so the tickmark looks just about right
+            font.setPointSizeF(font.pointSizeF() * 1.3);
+            painter->save();
+            painter->setFont(font);
+            // render the tickmark using the Unicode "Check Mark" symbol (✓)
+            drawItemTextWithRole(painter, rect, Qt::AlignHCenter|Qt::AlignVCenter, palette, true,
+                             opts.menuTick, QPalette::Text);
+            painter->restore();
+        } else {
+            QPixmap *pix = getPixmap(checkRadioCol(option), PIX_CHECK, 1.0);
 
-        painter->drawPixmap(rect.center().x() - pix->width() / 2,
-                            rect.center().y() - pix->height() / 2, *pix);
+            painter->drawPixmap(rect.center().x() - pix->width() / 2,
+                                rect.center().y() - pix->height() / 2, *pix);
+        }
     } else if (state & State_NoChange) {
         // tri-state
         int x(rect.center().x()), y(rect.center().y());
diff --git qt4/common/common.h qt4/common/common.h
index 313db33..e32e76c 100644
--- qt4/common/common.h
+++ qt4/common/common.h
@@ -886,6 +886,7 @@ typedef struct {
                      useQtFileDialogApps,
                      windowDragWhiteList,
                      windowDragBlackList;
+    bool             onlyTicksInMenu;
 } Options;
 
 void qtcSetupGradient(Gradient *grad, EGradientBorder border, int numStops, ...);
diff --git qt4/common/config_file.cpp qt4/common/config_file.cpp
index c58ad1a..acad221 100644
--- qt4/common/config_file.cpp
+++ qt4/common/config_file.cpp
@@ -1277,6 +1277,7 @@ bool qtcReadConfig(const QString &file, Options *opts, Options *defOpts, bool ch
             CFG_READ_BOOL(thinSbarGroove);
             CFG_READ_BOOL(colorSliderMouseOver);
             CFG_READ_BOOL(menuIcons);
+            CFG_READ_BOOL(onlyTicksInMenu);
             CFG_READ_BOOL(forceAlternateLvCols);
             CFG_READ_BOOL(invertBotTab);
             CFG_READ_INT_BOOL(menubarHiding, HIDE_KEYBOARD);
@@ -1567,6 +1568,11 @@ void qtcDefaultSettings(Options *opts)
     opts->thinSbarGroove=true;
     opts->colorSliderMouseOver=false;
     opts->menuIcons=true;
+#ifdef Q_OS_OSX
+    opts->onlyTicksInMenu=true;
+#else
+    opts->onlyTicksInMenu=false;
+#endif
     opts->forceAlternateLvCols=false;
     opts->invertBotTab=true;
     opts->menubarHiding=HIDE_NONE;
@@ -2295,6 +2301,7 @@ bool qtcWriteConfig(KConfig *cfg, const Options &opts, const Options &def, bool
         CFG_WRITE_ENTRY(thinSbarGroove);
         CFG_WRITE_ENTRY(colorSliderMouseOver);
         CFG_WRITE_ENTRY(menuIcons);
+        CFG_WRITE_ENTRY(onlyTicksInMenu);
         CFG_WRITE_ENTRY(forceAlternateLvCols);
         CFG_WRITE_ENTRY_NUM(square);
         CFG_WRITE_ENTRY(invertBotTab);
diff --git qt4/style/qtcurve.cpp qt4/style/qtcurve.cpp
index 951ec1a..c9fa418 100644
--- qt4/style/qtcurve.cpp
+++ qt4/style/qtcurve.cpp
@@ -4315,6 +4315,8 @@ void Style::drawPrimitive(PrimitiveElement element, const QStyleOption *option,
                                 ((State_Sunken|State_Enabled)==state || (State_Sunken|State_Enabled|State_Selected)==state));
             int   crSize(opts.crSize+(doEtch ? 2 : 0));
             QRect rect(r.x(), r.y()+(view ? -1 : 0), crSize, crSize);
+            // render checkable menu items with only a tick; should become an option
+            bool onlyTicksInMenu = menu && opts.onlyTicksInMenu;
 
             painter->save();
 
@@ -4332,7 +4334,7 @@ void Style::drawPrimitive(PrimitiveElement element, const QStyleOption *option,
                 if(r==QRect(0, 0, 15, 15))
                     rect.adjust(-1, -1, -1, -1);
             }
-            else
+            else if (!onlyTicksInMenu)
             {
                 if(isOO && r==QRect(0, 0, opts.crSize, opts.crSize))
                     rect.adjust(0, -1, 0, -1);
@@ -4414,9 +4416,34 @@ void Style::drawPrimitive(PrimitiveElement element, const QStyleOption *option,
 
             if(state&State_On || selectedOOMenu)
             {
-                QPixmap *pix(getPixmap(checkRadioCol(option), PIX_CHECK, 1.0));
-
-                painter->drawPixmap(rect.center().x()-(pix->width()/2), rect.center().y()-(pix->height()/2), *pix);
+                if (onlyTicksInMenu) {
+                    // only tickmarks (= without the box) in menu; adjust its horizontal position
+                    rect.adjust(6, 0, 0, 0);
+                    // get the font for adjusting; widget will be the menu holding the menuitem
+                    // but that uses the same font as the menuitem.
+                    QFont font(widget->font());
+#ifndef Q_OS_OSX
+                    font.setBold(true);
+#else
+                    // not that there is much variation in the CheckMark symbol on OS X, but
+                    // we pick one from a font that's always supposed to be there and that's
+                    // a good choice functionally speaking.
+                    font.setFamily(QStringLiteral("Apple Symbols"));
+#endif
+                    // adjust the size so the tickmark looks just about right
+                    font.setPointSizeF(font.pointSizeF() * 1.3);
+                    painter->save();
+                    painter->setFont(font);
+                    // render the tickmark using the Unicode "Check Mark" symbol (✓)
+                    drawItemTextWithRole(painter, rect, Qt::AlignHCenter|Qt::AlignVCenter, palette, true,
+                                     QString(QChar(0x2713)), QPalette::Text);
+                    painter->restore();
+                }
+                else
+                {
+                        QPixmap *pix(getPixmap(checkRadioCol(option), PIX_CHECK, 1.0));
+                        painter->drawPixmap(rect.center().x()-(pix->width()/2), rect.center().y()-(pix->height()/2), *pix);
+                }
             }
             else if (state&State_NoChange)    // tri-state
             {
@@ -6146,7 +6173,7 @@ void Style::drawControl(ControlElement element, const QStyleOption *option, QPai
                     checkRect = visualRect(menuItem->direction, menuItem->rect, checkRect);
                     if (checkable)
                     {
-                        if ((menuItem->checkType & QStyleOptionMenuItem::Exclusive) && menuItem->icon.isNull())
+                        if (!opts.onlyTicksInMenu && (menuItem->checkType & QStyleOptionMenuItem::Exclusive) && menuItem->icon.isNull())
                         {
                             QStyleOptionButton button;
                             button.rect = checkRect;
@@ -6166,7 +6193,7 @@ void Style::drawControl(ControlElement element, const QStyleOption *option, QPai
                                 if (checked)
                                     button.state |= State_On;
                                 button.palette = palette;
-                                drawPrimitive(PE_IndicatorCheckBox, &button, painter, widget);
+                                drawPrimitive(PE_IndicatorMenuCheckMark, &button, painter, widget);
                             }
                             else if (checked)
                             {
@@ -6177,9 +6204,7 @@ void Style::drawControl(ControlElement element, const QStyleOption *option, QPai
 
                                 sunkenRect = visualRect(menuItem->direction, menuItem->rect, sunkenRect);
                                 opt.state = menuItem->state;
-                                opt.state|=State_Raised|State_Horizontal;
-                                if (checked)
-                                    opt.state |= State_On;
+                                opt.state|=State_Raised|State_Horizontal|State_On;
                                 drawLightBevel(painter, sunkenRect, &opt, widget, ROUNDED_ALL, getFill(&opt, m_buttonCols), m_buttonCols);
                             }
                         }
diff --git gtk2/common/common.h gtk2/common/common.h
index cb0ec87..84cbc43 100644
--- gtk2/common/common.h
+++ gtk2/common/common.h
@@ -703,6 +703,8 @@ typedef struct {
                      noMenuBgndOpacityApps,
                      noBgndImageApps;
     Strings          noMenuStripeApps;
+    // for completeness:
+    bool             onlyTicksInMenu;
 } Options;
 
 void qtcSetupGradient(Gradient *grad, EGradientBorder border, int numStops, ...);
diff --git gtk2/common/config_file.cpp gtk2/common/config_file.cpp
index 96936e2..ce83b07 100644
--- gtk2/common/config_file.cpp
+++ gtk2/common/config_file.cpp
@@ -1354,6 +1354,7 @@ bool qtcReadConfig(const char *file, Options *opts, Options *defOpts)
             CFG_READ_BOOL(thinSbarGroove);
             CFG_READ_BOOL(colorSliderMouseOver);
             CFG_READ_BOOL(menuIcons);
+            CFG_READ_BOOL(onlyTicksInMenu);
             CFG_READ_BOOL(forceAlternateLvCols);
             CFG_READ_BOOL(invertBotTab);
             CFG_READ_INT_BOOL(menubarHiding, HIDE_KEYBOARD);
@@ -1708,6 +1709,11 @@ void qtcDefaultSettings(Options *opts)
     opts->thinSbarGroove=true;
     opts->colorSliderMouseOver=false;
     opts->menuIcons=true;
+#ifdef Q_OS_OSX
+    opts->onlyTicksInMenu=true;
+#else
+    opts->onlyTicksInMenu=false;
+#endif
     opts->forceAlternateLvCols=false;
     opts->invertBotTab=true;
     opts->menubarHiding=HIDE_NONE;
diff --git qt5/config/qtcurveconfig.cpp qt5/config/qtcurveconfig.cpp
index 304e3bf..91b6471 100644
--- qt5/config/qtcurveconfig.cpp
+++ qt5/config/qtcurveconfig.cpp
@@ -65,6 +65,7 @@
 #include <QTemporaryDir>
 #include <QTemporaryFile>
 #include <QStatusBar>
+#include <QActionGroup>
 
 // KDE
 #include <klocalizedstring.h>
@@ -243,7 +244,7 @@ CStylePreview::CStylePreview(QWidget *parent)
                                  i18n("(C) Craig Drummond, 2003-2011 & "
                                       "Yichao Yu, 2013-2015")))
 {
-    setWindowIcon(QIcon::fromTheme("preferences-desktop-theme"));
+    setWindowIcon(QIcon::fromTheme("preferences-desktop-theme", qApp->windowIcon()));
 
     QWidget *main = new QWidget(this);
     setObjectName("QtCurvePreview");
@@ -256,6 +257,13 @@ CStylePreview::CStylePreview(QWidget *parent)
     statusBar()->setSizeGripEnabled(true);
     toolBar()->setToolButtonStyle(Qt::ToolButtonTextBesideIcon);
     setCaption(i18n("Preview Window"));
+    // implement the exclusive nature of the items supposed to be mutually exclusive
+    // This can still be done in .ui files but would have to be maintained by hand
+    // as Qt's Designer no longer supports QActionGroups.
+    QActionGroup *aGroup = new QActionGroup(menu2SubMenu);
+    aGroup->addAction(exclusiveItem1);
+    aGroup->addAction(exclusiveItem2);
+    aGroup->addAction(exclusiveItem3);
 }
 
 CStylePreview::~CStylePreview()
@@ -1024,7 +1032,7 @@ QtCurveConfig::QtCurveConfig(QWidget *parent)
                 dwtTextAsPerTitleBar, dwtEffectAsPerTitleBar, dwtRoundTopOnly,
                 smallRadio, gtkComboMenus, mapKdeIcons, colorMenubarMouseOver,
                 useHighlightForMenu, gbLabel_bold, gbLabel_centred, fadeLines,
-                menuIcons, stdBtnSizes, forceAlternateLvCols, boldProgress,
+                menuIcons, onlyTicksInMenu, stdBtnSizes, forceAlternateLvCols, boldProgress,
                 coloredTbarMo, borderSelection, squareEntry, squareLvSelection,
                 squareScrollViews, squareFrame, squareTabFrame, squareSlider,
                 squareScrollbarSlider, squareWindows, squareTooltips,
@@ -3061,6 +3069,7 @@ void QtCurveConfig::setOptions(Options &opts)
     opts.gbLabel=getGroupBoxLabelFlags();
     opts.fadeLines=fadeLines->isChecked();
     opts.menuIcons=menuIcons->isChecked();
+    opts.onlyTicksInMenu=onlyTicksInMenu->isChecked();
     opts.stdBtnSizes=stdBtnSizes->isChecked();
     opts.boldProgress=boldProgress->isChecked();
     opts.coloredTbarMo=coloredTbarMo->isChecked();
@@ -3320,6 +3329,7 @@ void QtCurveConfig::setWidgetOptions(const Options &opts)
                                             : GBV_STANDARD);
     fadeLines->setChecked(opts.fadeLines);
     menuIcons->setChecked(opts.menuIcons);
+    onlyTicksInMenu->setChecked(opts.onlyTicksInMenu);
     stdBtnSizes->setChecked(opts.stdBtnSizes);
     boldProgress->setChecked(opts.boldProgress);
     boldProgress_false->setChecked(!opts.boldProgress);
@@ -3709,6 +3719,7 @@ bool QtCurveConfig::settingsChanged(const Options &opts)
          getGroupBoxLabelFlags()!=opts.gbLabel ||
          fadeLines->isChecked()!=opts.fadeLines ||
          menuIcons->isChecked()!=opts.menuIcons ||
+         onlyTicksInMenu->isChecked()!=opts.onlyTicksInMenu ||
          stdBtnSizes->isChecked()!=opts.stdBtnSizes ||
          boldProgress->isChecked()!=opts.boldProgress ||
          coloredTbarMo->isChecked()!=opts.coloredTbarMo ||
diff --git qt5/config/qtcurveconfigbase.ui qt5/config/qtcurveconfigbase.ui
index 81d693b..35044bc 100644
--- qt5/config/qtcurveconfigbase.ui
+++ qt5/config/qtcurveconfigbase.ui
@@ -174,7 +174,16 @@
           <item>
            <widget class="QWidget" name="previewWidget" native="true">
             <layout class="QGridLayout" name="gridLayout_5">
-             <property name="margin">
+             <property name="leftMargin">
+              <number>0</number>
+             </property>
+             <property name="topMargin">
+              <number>0</number>
+             </property>
+             <property name="rightMargin">
+              <number>0</number>
+             </property>
+             <property name="bottomMargin">
               <number>0</number>
              </property>
              <item row="0" column="0" colspan="2">
@@ -1597,7 +1606,16 @@
           <property name="spacing">
            <number>0</number>
           </property>
-          <property name="margin">
+          <property name="leftMargin">
+           <number>0</number>
+          </property>
+          <property name="topMargin">
+           <number>0</number>
+          </property>
+          <property name="rightMargin">
+           <number>0</number>
+          </property>
+          <property name="bottomMargin">
            <number>0</number>
           </property>
           <item>
@@ -1657,7 +1675,16 @@
           <property name="spacing">
            <number>0</number>
           </property>
-          <property name="margin">
+          <property name="leftMargin">
+           <number>0</number>
+          </property>
+          <property name="topMargin">
+           <number>0</number>
+          </property>
+          <property name="rightMargin">
+           <number>0</number>
+          </property>
+          <property name="bottomMargin">
            <number>0</number>
           </property>
           <item>
@@ -1714,7 +1741,16 @@
           <property name="spacing">
            <number>0</number>
           </property>
-          <property name="margin">
+          <property name="leftMargin">
+           <number>0</number>
+          </property>
+          <property name="topMargin">
+           <number>0</number>
+          </property>
+          <property name="rightMargin">
+           <number>0</number>
+          </property>
+          <property name="bottomMargin">
            <number>0</number>
           </property>
           <item>
@@ -1768,7 +1804,16 @@
           <property name="spacing">
            <number>0</number>
           </property>
-          <property name="margin">
+          <property name="leftMargin">
+           <number>0</number>
+          </property>
+          <property name="topMargin">
+           <number>0</number>
+          </property>
+          <property name="rightMargin">
+           <number>0</number>
+          </property>
+          <property name="bottomMargin">
            <number>0</number>
           </property>
           <item>
@@ -1819,7 +1864,16 @@
           <property name="spacing">
            <number>0</number>
           </property>
-          <property name="margin">
+          <property name="leftMargin">
+           <number>0</number>
+          </property>
+          <property name="topMargin">
+           <number>0</number>
+          </property>
+          <property name="rightMargin">
+           <number>0</number>
+          </property>
+          <property name="bottomMargin">
            <number>0</number>
           </property>
           <item>
@@ -1870,7 +1924,16 @@
           <property name="spacing">
            <number>0</number>
           </property>
-          <property name="margin">
+          <property name="leftMargin">
+           <number>0</number>
+          </property>
+          <property name="topMargin">
+           <number>0</number>
+          </property>
+          <property name="rightMargin">
+           <number>0</number>
+          </property>
+          <property name="bottomMargin">
            <number>0</number>
           </property>
           <item>
@@ -1961,7 +2024,16 @@
           <property name="spacing">
            <number>0</number>
           </property>
-          <property name="margin">
+          <property name="leftMargin">
+           <number>0</number>
+          </property>
+          <property name="topMargin">
+           <number>0</number>
+          </property>
+          <property name="rightMargin">
+           <number>0</number>
+          </property>
+          <property name="bottomMargin">
            <number>0</number>
           </property>
           <item>
@@ -2018,7 +2090,16 @@
           <property name="spacing">
            <number>0</number>
           </property>
-          <property name="margin">
+          <property name="leftMargin">
+           <number>0</number>
+          </property>
+          <property name="topMargin">
+           <number>0</number>
+          </property>
+          <property name="rightMargin">
+           <number>0</number>
+          </property>
+          <property name="bottomMargin">
            <number>0</number>
           </property>
           <item>
@@ -3143,14 +3224,24 @@
          </property>
         </widget>
        </item>
-       <item row="4" column="0">
+       <item row="4" column="1">
+        <widget class="QCheckBox" name="onlyTicksInMenu">
+         <property name="toolTip">
+          <string>&lt;p&gt;If enabled, checked menu items get only a check mark instead of a checkbox or radiobutton.</string>
+         </property>
+         <property name="text">
+          <string>Check marks only</string>
+         </property>
+        </widget>
+       </item>
+       <item row="5" column="0">
         <widget class="QLabel" name="label_26">
          <property name="text">
           <string>Shade background by:</string>
          </property>
         </widget>
        </item>
-       <item row="4" column="1">
+       <item row="5" column="1">
         <widget class="QSpinBox" name="lighterPopupMenuBgnd">
          <property name="toolTip">
           <string>&lt;p&gt;Defines the amount by which the popup menu background will be lightened (or darkened).</string>
@@ -3160,14 +3251,14 @@
          </property>
         </widget>
        </item>
-       <item row="5" column="0">
+       <item row="6" column="0">
         <widget class="QLabel" name="label_75">
          <property name="text">
           <string>Background appearance:</string>
          </property>
         </widget>
        </item>
-       <item row="5" column="1">
+       <item row="6" column="1">
         <layout class="QHBoxLayout" name="horizontalLayout_6">
          <item>
           <widget class="QComboBox" name="menuBgndAppearance">
@@ -3185,7 +3276,7 @@
          </item>
         </layout>
        </item>
-       <item row="6" column="1">
+       <item row="7" column="1">
         <widget class="QComboBox" name="menuBgndGrad">
          <property name="toolTip">
           <string>&lt;p&gt;Defines background gradient orientation.</string>
@@ -3199,7 +3290,7 @@
          </property>
         </widget>
        </item>
-       <item row="7" column="1">
+       <item row="8" column="1">
         <layout class="QHBoxLayout" name="horizontalLayout_7">
          <item>
           <widget class="QComboBox" name="menuBgndImage">
@@ -3218,14 +3309,14 @@
          </item>
         </layout>
        </item>
-       <item row="8" column="0">
+       <item row="9" column="0">
         <widget class="QLabel" name="label_45">
          <property name="text">
           <string>Sub-menu delay:</string>
          </property>
         </widget>
        </item>
-       <item row="8" column="1">
+       <item row="9" column="1">
         <widget class="QSpinBox" name="menuDelay">
          <property name="toolTip">
           <string>&lt;p&gt;Defines the delay before a sub-menu appears.</string>
@@ -3238,21 +3329,21 @@
          </property>
         </widget>
        </item>
-       <item row="9" column="0">
+       <item row="10" column="0">
         <widget class="QLabel" name="label_44">
          <property name="text">
           <string>Stripe:</string>
          </property>
         </widget>
        </item>
-       <item row="9" column="1">
+       <item row="10" column="1">
         <widget class="QComboBox" name="menuStripe">
          <property name="toolTip">
           <string>&lt;p&gt;Defines the coloration of the menu-stripe that will be drawn on the left of popup menus - but only if this option is not set to 'none'.</string>
          </property>
         </widget>
        </item>
-       <item row="10" column="1">
+       <item row="11" column="1">
         <widget class="KColorButton" name="customMenuStripeColor">
          <property name="enabled">
           <bool>false</bool>
@@ -3268,7 +3359,7 @@
          </property>
         </widget>
        </item>
-       <item row="11" column="1">
+       <item row="12" column="1">
         <widget class="QComboBox" name="menuStripeAppearance">
          <property name="enabled">
           <bool>false</bool>
@@ -3278,14 +3369,14 @@
          </property>
         </widget>
        </item>
-       <item row="12" column="0">
+       <item row="13" column="0">
         <widget class="QLabel" name="label_61">
          <property name="text">
           <string>Border popupmenus:</string>
          </property>
         </widget>
        </item>
-       <item row="12" column="1">
+       <item row="13" column="1">
         <widget class="QCheckBox" name="popupBorder">
          <property name="toolTip">
           <string>&lt;p&gt;If enabled, the whole popup menu will have a border, otherwise no border will be drawn. This option should be enabled unless you have also enabled shadows in your window manager.</string>
@@ -3295,14 +3386,14 @@
          </property>
         </widget>
        </item>
-       <item row="13" column="0">
+       <item row="14" column="0">
         <widget class="QLabel" name="label_97">
          <property name="text">
           <string>Use menubar colour settings:</string>
          </property>
         </widget>
        </item>
-       <item row="13" column="1">
+       <item row="14" column="1">
         <widget class="QCheckBox" name="shadePopupMenu">
          <property name="text">
           <string>Enabled</string>
@@ -4029,12 +4120,12 @@
                   <property name="toolTip">
                    <string>&lt;p&gt;Set the position of the gradient value. 0% means the top of horizontal gradients, and the left for vertical gradients.</string>
                   </property>
-                  <property name="maximum">
-                   <number>100</number>
-                  </property>
                   <property name="suffix">
                    <string>%</string>
                   </property>
+                  <property name="maximum">
+                   <number>100</number>
+                  </property>
                  </widget>
                 </item>
                 <item row="1" column="1">
@@ -4048,12 +4139,12 @@
                   <property name="toolTip">
                    <string>&lt;p&gt;Sets the strength of the gradient at the defined position. 110% would lighten a color by 10%, 90% would darken it by 10%, and 100% would be the original color.</string>
                   </property>
-                  <property name="maximum">
-                   <number>200</number>
-                  </property>
                   <property name="suffix">
                    <string>%</string>
                   </property>
+                  <property name="maximum">
+                   <number>200</number>
+                  </property>
                  </widget>
                 </item>
                 <item row="2" column="1">
@@ -4067,12 +4158,12 @@
                   <property name="toolTip">
                    <string>&lt;p&gt;Sets the alpha value of the gradient at the defined position.</string>
                   </property>
-                  <property name="maximum">
-                   <number>100</number>
-                  </property>
                   <property name="suffix">
                    <string>%</string>
                   </property>
+                  <property name="maximum">
+                   <number>100</number>
+                  </property>
                  </widget>
                 </item>
                </layout>
@@ -4667,12 +4758,12 @@
    <slot>setEnabled(bool)</slot>
    <hints>
     <hint type="sourcelabel">
-     <x>352</x>
-     <y>398</y>
+     <x>367</x>
+     <y>57</y>
     </hint>
     <hint type="destinationlabel">
-     <x>355</x>
-     <y>315</y>
+     <x>367</x>
+     <y>57</y>
     </hint>
    </hints>
   </connection>
@@ -4683,7 +4774,7 @@
    <slot>exportKDE3()</slot>
    <hints>
     <hint type="sourcelabel">
-     <x>266</x>
+     <x>367</x>
      <y>57</y>
     </hint>
     <hint type="destinationlabel">
@@ -4699,8 +4790,8 @@
    <slot>exportQt()</slot>
    <hints>
     <hint type="sourcelabel">
-     <x>434</x>
-     <y>89</y>
+     <x>367</x>
+     <y>57</y>
     </hint>
     <hint type="destinationlabel">
      <x>532</x>
@@ -4715,12 +4806,12 @@
    <slot>setEnabled(bool)</slot>
    <hints>
     <hint type="sourcelabel">
-     <x>440</x>
-     <y>264</y>
+     <x>367</x>
+     <y>57</y>
     </hint>
     <hint type="destinationlabel">
-     <x>440</x>
-     <y>290</y>
+     <x>367</x>
+     <y>57</y>
     </hint>
    </hints>
   </connection>
@@ -4731,12 +4822,12 @@
    <slot>setEnabled(bool)</slot>
    <hints>
     <hint type="sourcelabel">
-     <x>473</x>
-     <y>261</y>
+     <x>367</x>
+     <y>57</y>
     </hint>
     <hint type="destinationlabel">
-     <x>425</x>
-     <y>322</y>
+     <x>367</x>
+     <y>57</y>
     </hint>
    </hints>
   </connection>
diff --git qt5/config/stylepreview.ui qt5/config/stylepreview.ui
index ec36ee3..545af54 100644
--- qt5/config/stylepreview.ui
+++ qt5/config/stylepreview.ui
@@ -28,6 +28,42 @@
       </attribute>
       <layout class="QVBoxLayout">
        <item>
+        <widget class="QMenuBar" name="menuBarPreview">
+         <property name="nativeMenuBar">
+          <bool>false</bool>
+         </property>
+         <widget class="QMenu" name="menuMenu_1">
+          <property name="title">
+           <string>&amp;Menu 1</string>
+          </property>
+          <addaction name="separator"/>
+          <addaction name="actionRegular_item_1"/>
+          <addaction name="separator"/>
+          <addaction name="actionRegular_item_2"/>
+         </widget>
+         <widget class="QMenu" name="menuMenu_2">
+          <property name="title">
+           <string>Menu &amp;2</string>
+          </property>
+          <widget class="QMenu" name="menu2SubMenu">
+           <property name="title">
+            <string>&amp;Submenu</string>
+           </property>
+           <addaction name="actionRegular_submenu_item"/>
+          </widget>
+          <addaction name="actionCheckable_item"/>
+          <addaction name="separator"/>
+          <addaction name="menu2SubMenu"/>
+          <addaction name="separator"/>
+          <addaction name="exclusiveItem1"/>
+          <addaction name="exclusiveItem2"/>
+          <addaction name="exclusiveItem3"/>
+         </widget>
+         <addaction name="menuMenu_1"/>
+         <addaction name="menuMenu_2"/>
+        </widget>
+       </item>
+       <item>
         <layout class="QHBoxLayout">
          <item>
           <layout class="QVBoxLayout" name="verticalLayout">
@@ -40,7 +76,7 @@
               <item>
                <widget class="QRadioButton" name="RadioButton1">
                 <property name="text">
-                 <string>Radio button</string>
+                 <string>&amp;Radio button</string>
                 </property>
                 <property name="checked">
                  <bool>true</bool>
@@ -50,7 +86,7 @@
               <item>
                <widget class="QRadioButton" name="RadioButton2">
                 <property name="text">
-                 <string>Radio button</string>
+                 <string>Radio &amp;button</string>
                 </property>
                </widget>
               </item>
@@ -236,7 +272,7 @@
            </sizepolicy>
           </property>
           <property name="windowTitle">
-           <string>Dock Widget</string>
+           <string>&amp;Dock Widget</string>
           </property>
           <widget class="QWidget" name="dockWidgetContents">
            <layout class="QVBoxLayout" name="verticalLayout_2">
@@ -269,12 +305,6 @@
               <attribute name="headerMinimumSectionSize">
                <number>32</number>
               </attribute>
-              <attribute name="headerDefaultSectionSize">
-               <number>80</number>
-              </attribute>
-              <attribute name="headerMinimumSectionSize">
-               <number>32</number>
-              </attribute>
               <column>
                <property name="text">
                 <string>Column 1</string>
@@ -511,6 +541,74 @@ Come to wash my sins away...</string>
     </widget>
    </item>
   </layout>
+  <action name="actionRegular_item_1">
+   <property name="text">
+    <string>&amp;Regular item 1</string>
+   </property>
+   <property name="menuRole">
+    <enum>QAction::NoRole</enum>
+   </property>
+  </action>
+  <action name="actionRegular_item_2">
+   <property name="text">
+    <string>Regular &amp;item 2</string>
+   </property>
+   <property name="menuRole">
+    <enum>QAction::NoRole</enum>
+   </property>
+  </action>
+  <action name="actionCheckable_item">
+   <property name="checkable">
+    <bool>true</bool>
+   </property>
+   <property name="text">
+    <string>&amp;Checkable item</string>
+   </property>
+   <property name="menuRole">
+    <enum>QAction::NoRole</enum>
+   </property>
+  </action>
+  <action name="exclusiveItem1">
+   <property name="checkable">
+    <bool>true</bool>
+   </property>
+   <property name="text">
+    <string>&amp;Exclusive item 1</string>
+   </property>
+   <property name="menuRole">
+    <enum>QAction::NoRole</enum>
+   </property>
+  </action>
+  <action name="exclusiveItem2">
+   <property name="checkable">
+    <bool>true</bool>
+   </property>
+   <property name="checked">
+    <bool>true</bool>
+   </property>
+   <property name="text">
+    <string>Exclusive &amp;item 2</string>
+   </property>
+   <property name="menuRole">
+    <enum>QAction::NoRole</enum>
+   </property>
+  </action>
+  <action name="exclusiveItem3">
+   <property name="checkable">
+    <bool>true</bool>
+   </property>
+   <property name="text">
+    <string>Exclusive item &amp;3</string>
+   </property>
+   <property name="menuRole">
+    <enum>QAction::NoRole</enum>
+   </property>
+  </action>
+  <action name="actionRegular_submenu_item">
+   <property name="text">
+    <string>regular submenu item</string>
+   </property>
+  </action>
  </widget>
  <resources/>
  <connections>
diff --git qt4/config/qtcurveconfig.cpp qt4/config/qtcurveconfig.cpp
index aef4c26..e5378b1 100644
--- qt4/config/qtcurveconfig.cpp
+++ qt4/config/qtcurveconfig.cpp
@@ -38,6 +38,7 @@
 #include <QFrame>
 #include <QTabWidget>
 #include <QFileInfo>
+#include <QActionGroup>
 #include <QBoxLayout>
 #include <QGridLayout>
 #include <QTreeWidget>
@@ -253,6 +254,13 @@ CStylePreview::CStylePreview(QWidget *parent)
     statusBar()->setSizeGripEnabled(true);
     toolBar()->setToolButtonStyle(Qt::ToolButtonTextBesideIcon);
     setCaption(i18n("Preview Window"));
+    // implement the exclusive nature of the items supposed to be mutually exclusive
+    // This can still be done in .ui files but would have to be maintained by hand
+    // as Qt's Designer no longer supports QActionGroups.
+    QActionGroup *aGroup = new QActionGroup(menu2SubMenu);
+    aGroup->addAction(exclusiveItem1);
+    aGroup->addAction(exclusiveItem2);
+    aGroup->addAction(exclusiveItem3);
 }
 
 CStylePreview::~CStylePreview()
@@ -1125,6 +1133,7 @@ QtCurveConfig::QtCurveConfig(QWidget *parent)
     connect(gbLabel_centred, SIGNAL(toggled(bool)), SLOT(updateChanged()));
     connect(fadeLines, SIGNAL(toggled(bool)), SLOT(updateChanged()));
     connect(menuIcons, SIGNAL(toggled(bool)), SLOT(updateChanged()));
+    connect(onlyTicksInMenu, SIGNAL(toggled(bool)), SLOT(updateChanged()));
     connect(stdBtnSizes, SIGNAL(toggled(bool)), SLOT(updateChanged()));
     connect(forceAlternateLvCols, SIGNAL(toggled(bool)), SLOT(updateChanged()));
     connect(titlebarAlignment, SIGNAL(currentIndexChanged(int)), SLOT(updateChanged()));
@@ -3085,6 +3094,7 @@ void QtCurveConfig::setOptions(Options &opts)
     opts.gbLabel=getGroupBoxLabelFlags();
     opts.fadeLines=fadeLines->isChecked();
     opts.menuIcons=menuIcons->isChecked();
+    opts.onlyTicksInMenu=onlyTicksInMenu->isChecked();
     opts.stdBtnSizes=stdBtnSizes->isChecked();
     opts.boldProgress=boldProgress->isChecked();
     opts.coloredTbarMo=coloredTbarMo->isChecked();
@@ -3344,6 +3354,7 @@ void QtCurveConfig::setWidgetOptions(const Options &opts)
                                             : GBV_STANDARD);
     fadeLines->setChecked(opts.fadeLines);
     menuIcons->setChecked(opts.menuIcons);
+    onlyTicksInMenu->setChecked(opts.onlyTicksInMenu);
     stdBtnSizes->setChecked(opts.stdBtnSizes);
     boldProgress->setChecked(opts.boldProgress);
     boldProgress_false->setChecked(!opts.boldProgress);
diff --git qt4/config/qtcurveconfigbase.ui qt4/config/qtcurveconfigbase.ui
index 5247266..6ef3295 100644
--- qt4/config/qtcurveconfigbase.ui
+++ qt4/config/qtcurveconfigbase.ui
@@ -3160,14 +3160,24 @@
          </property>
         </widget>
        </item>
-       <item row="4" column="0">
+       <item row="4" column="1">
+        <widget class="QCheckBox" name="onlyTicksInMenu">
+         <property name="toolTip">
+          <string>&lt;p&gt;If enabled, checked menu items get only a check mark instead of a checkbox or radiobutton.</string>
+         </property>
+         <property name="text">
+          <string>Check marks only</string>
+         </property>
+        </widget>
+       </item>
+       <item row="5" column="0">
         <widget class="QLabel" name="label_26">
          <property name="text">
           <string>Shade background by:</string>
          </property>
         </widget>
        </item>
-       <item row="4" column="1">
+       <item row="5" column="1">
         <widget class="KIntNumInput" name="lighterPopupMenuBgnd">
          <property name="toolTip">
           <string>&lt;p&gt;Defines the amount by which the popup menu background will be lightened (or darkened).</string>
@@ -3177,14 +3187,14 @@
          </property>
         </widget>
        </item>
-       <item row="5" column="0">
+       <item row="6" column="0">
         <widget class="QLabel" name="label_75">
          <property name="text">
           <string>Background appearance:</string>
          </property>
         </widget>
        </item>
-       <item row="5" column="1">
+       <item row="6" column="1">
         <layout class="QHBoxLayout" name="horizontalLayout_6">
          <item>
           <widget class="QComboBox" name="menuBgndAppearance">
@@ -3202,21 +3212,21 @@
          </item>
         </layout>
        </item>
-       <item row="6" column="1">
+       <item row="7" column="1">
         <widget class="QComboBox" name="menuBgndGrad">
          <property name="toolTip">
           <string>&lt;p&gt;Defines background gradient orientation.</string>
          </property>
         </widget>
        </item>
-       <item row="7" column="0">
+       <item row="8" column="0">
         <widget class="QLabel" name="label_xx">
          <property name="text">
           <string>Background image:</string>
          </property>
         </widget>
        </item>
-       <item row="7" column="1">
+       <item row="8" column="1">
         <layout class="QHBoxLayout" name="horizontalLayout_7">
          <item>
           <widget class="QComboBox" name="menuBgndImage">
@@ -3235,14 +3245,14 @@
          </item>
         </layout>
        </item>
-       <item row="8" column="0">
+       <item row="9" column="0">
         <widget class="QLabel" name="label_45">
          <property name="text">
           <string>Sub-menu delay:</string>
          </property>
         </widget>
        </item>
-       <item row="8" column="1">
+       <item row="9" column="1">
         <widget class="KIntNumInput" name="menuDelay">
          <property name="toolTip">
           <string>&lt;p&gt;Defines the delay before a sub-menu appears.</string>
@@ -3255,21 +3265,21 @@
          </property>
         </widget>
        </item>
-       <item row="9" column="0">
+       <item row="10" column="0">
         <widget class="QLabel" name="label_44">
          <property name="text">
           <string>Stripe:</string>
          </property>
         </widget>
        </item>
-       <item row="9" column="1">
+       <item row="10" column="1">
         <widget class="QComboBox" name="menuStripe">
          <property name="toolTip">
           <string>&lt;p&gt;Defines the coloration of the menu-stripe that will be drawn on the left of popup menus - but only if this option is not set to 'none'.</string>
          </property>
         </widget>
        </item>
-       <item row="10" column="1">
+       <item row="11" column="1">
         <widget class="KColorButton" name="customMenuStripeColor">
          <property name="enabled">
           <bool>false</bool>
@@ -3285,7 +3295,7 @@
          </property>
         </widget>
        </item>
-       <item row="11" column="1">
+       <item row="12" column="1">
         <widget class="QComboBox" name="menuStripeAppearance">
          <property name="enabled">
           <bool>false</bool>
@@ -3295,14 +3305,14 @@
          </property>
         </widget>
        </item>
-       <item row="12" column="0">
+       <item row="13" column="0">
         <widget class="QLabel" name="label_61">
          <property name="text">
           <string>Border popupmenus:</string>
          </property>
         </widget>
        </item>
-       <item row="12" column="1">
+       <item row="13" column="1">
         <widget class="QCheckBox" name="popupBorder">
          <property name="toolTip">
           <string>&lt;p&gt;If enabled, the whole popup menu will have a border, otherwise no border will be drawn. This option should be enabled unless you have also enabled shadows in your window manager.</string>
@@ -3312,14 +3322,14 @@
          </property>
         </widget>
        </item>
-       <item row="13" column="0">
+       <item row="14" column="0">
         <widget class="QLabel" name="label_97">
          <property name="text">
           <string>Use menubar colour settings:</string>
          </property>
         </widget>
        </item>
-       <item row="13" column="1">
+       <item row="14" column="1">
         <widget class="QCheckBox" name="shadePopupMenu">
          <property name="text">
           <string>Enabled</string>
diff --git qt4/config/stylepreview.ui qt4/config/stylepreview.ui
index 29b1ef8..4c6ba8a 100644
--- qt4/config/stylepreview.ui
+++ qt4/config/stylepreview.ui
@@ -28,6 +28,42 @@
       </attribute>
       <layout class="QVBoxLayout">
        <item>
+        <widget class="QMenuBar" name="menuBarPreview">
+         <property name="nativeMenuBar">
+          <bool>false</bool>
+         </property>
+         <widget class="QMenu" name="menuMenu_1">
+          <property name="title">
+           <string>&amp;Menu 1</string>
+          </property>
+          <addaction name="separator"/>
+          <addaction name="actionRegular_item_1"/>
+          <addaction name="separator"/>
+          <addaction name="actionRegular_item_2"/>
+         </widget>
+         <widget class="QMenu" name="menuMenu_2">
+          <property name="title">
+           <string>Menu &amp;2</string>
+          </property>
+          <widget class="QMenu" name="menu2SubMenu">
+           <property name="title">
+            <string>&amp;Submenu</string>
+           </property>
+           <addaction name="actionRegular_submenu_item"/>
+          </widget>
+          <addaction name="actionCheckable_item"/>
+          <addaction name="separator"/>
+          <addaction name="menu2SubMenu"/>
+          <addaction name="separator"/>
+          <addaction name="exclusiveItem1"/>
+          <addaction name="exclusiveItem2"/>
+          <addaction name="exclusiveItem3"/>
+         </widget>
+         <addaction name="menuMenu_1"/>
+         <addaction name="menuMenu_2"/>
+        </widget>
+       </item>
+       <item>
         <layout class="QHBoxLayout">
          <item>
           <layout class="QVBoxLayout" name="verticalLayout">
@@ -269,12 +305,6 @@
               <attribute name="headerMinimumSectionSize">
                <number>32</number>
               </attribute>
-              <attribute name="headerDefaultSectionSize">
-               <number>80</number>
-              </attribute>
-              <attribute name="headerMinimumSectionSize">
-               <number>32</number>
-              </attribute>
               <column>
                <property name="text">
                 <string>Column 1</string>
@@ -511,6 +541,74 @@ Come to wash my sins away...</string>
     </widget>
    </item>
   </layout>
+  <action name="actionRegular_item_1">
+   <property name="text">
+    <string>&amp;Regular item 1</string>
+   </property>
+   <property name="menuRole">
+    <enum>QAction::NoRole</enum>
+   </property>
+  </action>
+  <action name="actionRegular_item_2">
+   <property name="text">
+    <string>Regular &amp;item 2</string>
+   </property>
+   <property name="menuRole">
+    <enum>QAction::NoRole</enum>
+   </property>
+  </action>
+  <action name="actionCheckable_item">
+   <property name="checkable">
+    <bool>true</bool>
+   </property>
+   <property name="text">
+    <string>&amp;Checkable item</string>
+   </property>
+   <property name="menuRole">
+    <enum>QAction::NoRole</enum>
+   </property>
+  </action>
+  <action name="exclusiveItem1">
+   <property name="checkable">
+    <bool>true</bool>
+   </property>
+   <property name="text">
+    <string>&amp;Exclusive item 1</string>
+   </property>
+   <property name="menuRole">
+    <enum>QAction::NoRole</enum>
+   </property>
+  </action>
+  <action name="exclusiveItem2">
+   <property name="checkable">
+    <bool>true</bool>
+   </property>
+   <property name="checked">
+    <bool>true</bool>
+   </property>
+   <property name="text">
+    <string>Exclusive &amp;item 2</string>
+   </property>
+   <property name="menuRole">
+    <enum>QAction::NoRole</enum>
+   </property>
+  </action>
+  <action name="exclusiveItem3">
+   <property name="checkable">
+    <bool>true</bool>
+   </property>
+   <property name="text">
+    <string>Exclusive item &amp;3</string>
+   </property>
+   <property name="menuRole">
+    <enum>QAction::NoRole</enum>
+   </property>
+  </action>
+  <action name="actionRegular_submenu_item">
+   <property name="text">
+    <string>regular submenu item</string>
+   </property>
+  </action>
  </widget>
  <customwidgets>
   <customwidget>
