diff --git qt4/kwin/qtcurveclient.cpp qt4/kwin/qtcurveclient.cpp
index 80855cf..31307c6 100644
--- qt4/kwin/qtcurveclient.cpp
+++ qt4/kwin/qtcurveclient.cpp
@@ -1138,6 +1138,7 @@ void QtCurveClient::informAppOfActiveChange()
 {
     if (Handler()->wStyle()->pixelMetric(
             (QStyle::PixelMetric)QtC_ShadeMenubarOnlyWhenActive, nullptr, nullptr)) {
+#ifdef X11_SEND_QTCURVE_ACTIVE_WINDOW
         union {
             char _buff[32];
             xcb_client_message_event_t ev;
@@ -1149,6 +1150,7 @@ void QtCurveClient::informAppOfActiveChange()
         xev->type = qtc_x11_qtc_active_window;
         xev->data.data32[0] = isActive() ? 1 : 0;
         qtcX11SendEvent(false, windowId(), XCB_EVENT_MASK_NO_EVENT, xev);
+#endif
         qtcX11Flush();
     }
 }
diff --git qt5/kwin/qtcurveclient.cpp qt5/kwin/qtcurveclient.cpp
index d784bd4..966664c 100644
--- qt5/kwin/qtcurveclient.cpp
+++ qt5/kwin/qtcurveclient.cpp
@@ -1139,6 +1139,7 @@ void QtCurveClient::informAppOfActiveChange()
 {
     if (Handler()->wStyle()->pixelMetric(
             (QStyle::PixelMetric)QtC_ShadeMenubarOnlyWhenActive, nullptr, nullptr)) {
+#ifdef X11_SEND_QTCURVE_ACTIVE_WINDOW
         union {
             char _buff[32];
             xcb_client_message_event_t ev;
@@ -1150,6 +1151,7 @@ void QtCurveClient::informAppOfActiveChange()
         xev->type = qtc_x11_qtc_active_window;
         xev->data.data32[0] = isActive() ? 1 : 0;
         qtcX11SendEvent(false, windowId(), XCB_EVENT_MASK_NO_EVENT, xev);
+#endif
         qtcX11Flush();
     }
 }
