diff --git qt4/common/common.h qt4/common/common.h
index e32e76c..6f1945f 100644
--- qt4/common/common.h
+++ qt4/common/common.h
@@ -885,7 +885,8 @@ typedef struct {
                      statusbarApps,
                      useQtFileDialogApps,
                      windowDragWhiteList,
-                     windowDragBlackList;
+                     windowDragBlackList,
+                     nonnativeMenubarApps;
     bool             onlyTicksInMenu;
 } Options;
 
diff --git qt4/common/config_file.cpp qt4/common/config_file.cpp
index ef684f7..2efc647 100644
--- qt4/common/config_file.cpp
+++ qt4/common/config_file.cpp
@@ -1348,6 +1348,7 @@ bool qtcReadConfig(const QString &file, Options *opts, Options *defOpts, bool ch
             CFG_READ_STRING_LIST(menubarApps);
             CFG_READ_STRING_LIST(statusbarApps);
             CFG_READ_STRING_LIST(useQtFileDialogApps);
+            CFG_READ_STRING_LIST(nonnativeMenubarApps);
             CFG_READ_STRING_LIST(windowDragWhiteList);
             CFG_READ_STRING_LIST(windowDragBlackList);
             readDoubleList(cfg, "customShades", opts->customShades, QTC_NUM_STD_SHADES);
@@ -2371,6 +2372,7 @@ bool qtcWriteConfig(KConfig *cfg, const Options &opts, const Options &def, bool
         CFG_WRITE_STRING_LIST_ENTRY(menubarApps);
         CFG_WRITE_STRING_LIST_ENTRY(statusbarApps);
         CFG_WRITE_STRING_LIST_ENTRY(useQtFileDialogApps);
+        CFG_WRITE_STRING_LIST_ENTRY(nonnativeMenubarApps);
 
         for(int i=APPEARANCE_CUSTOM1; i<(APPEARANCE_CUSTOM1+NUM_CUSTOM_GRAD); ++i)
         {
diff --git qt4/config/qtcurveconfig.cpp qt4/config/qtcurveconfig.cpp
index e5378b1..740150f 100644
--- qt4/config/qtcurveconfig.cpp
+++ qt4/config/qtcurveconfig.cpp
@@ -1209,6 +1209,7 @@ QtCurveConfig::QtCurveConfig(QWidget *parent)
     connect(menubarApps, SIGNAL(editingFinished()), SLOT(updateChanged()));
     connect(statusbarApps, SIGNAL(editingFinished()), SLOT(updateChanged()));
     connect(noMenuStripeApps, SIGNAL(editingFinished()), SLOT(updateChanged()));
+    connect(nonnativeMenubarApps, SIGNAL(editingFinished()), SLOT(updateChanged()));
 
     menubarBlend->setIcon(KIcon("configure"));
     connect(menubarBlend, SIGNAL(clicked(bool)), SLOT(menubarTitlebarBlend()));
@@ -3175,6 +3176,7 @@ void QtCurveConfig::setOptions(Options &opts)
     opts.menubarApps=toSet(menubarApps->text());
     opts.statusbarApps=toSet(statusbarApps->text());
     opts.noMenuStripeApps=toSet(noMenuStripeApps->text());
+    opts.nonnativeMenubarApps=toSet(nonnativeMenubarApps->text());
 
     if(IMG_FILE==opts.bgndImage.type)
     {
@@ -3524,6 +3526,7 @@ void QtCurveConfig::setWidgetOptions(const Options &opts)
     noMenuBgndOpacityApps->setText(toString(opts.noMenuBgndOpacityApps));
     noBgndImageApps->setText(toString(opts.noBgndImageApps));
     useQtFileDialogApps->setText(toString(opts.useQtFileDialogApps));
+    nonnativeMenubarApps->setText(toString(opts.nonnativeMenubarApps));
     menubarApps->setText(toString(opts.menubarApps));
     statusbarApps->setText(toString(opts.statusbarApps));
     noMenuStripeApps->setText(toString(opts.noMenuStripeApps));
@@ -3830,6 +3833,7 @@ bool QtCurveConfig::settingsChanged(const Options &opts)
          toSet(menubarApps->text())!=opts.menubarApps ||
          toSet(statusbarApps->text())!=opts.statusbarApps ||
          toSet(noMenuStripeApps->text())!=opts.noMenuStripeApps ||
+         toSet(nonnativeMenubarApps->text())!=opts.nonnativeMenubarApps ||
 
          diffShades(opts) ||
 
diff --git qt4/config/qtcurveconfigbase.ui qt4/config/qtcurveconfigbase.ui
index 6ef3295..53d8bb6 100644
--- qt4/config/qtcurveconfigbase.ui
+++ qt4/config/qtcurveconfigbase.ui
@@ -3850,6 +3850,20 @@
            </property>
           </widget>
          </item>
+         <item row="8" column="0">
+          <widget class="QLabel" name="label_89a">
+           <property name="text">
+            <string>Force non-native menubar:</string>
+           </property>
+          </widget>
+         </item>
+         <item row="8" column="1">
+          <widget class="KLineEdit" name="nonnativeMenubarApps">
+           <property name="toolTip">
+            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Comma separated list of names of KDE/Qt applications where QtCurve needs to instruct Qt not to use the native menubar.&lt;/p&gt;&lt;p&gt;This feature has effect only on Mac OS X/macOS and is intended for use with KF5 and Qt5 applications.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
+           </property>
+          </widget>
+         </item>
         </layout>
        </item>
        <item>
diff --git qt4/style/qtcurve.cpp qt4/style/qtcurve.cpp
index 208d115..0b5092e 100644
--- qt4/style/qtcurve.cpp
+++ qt4/style/qtcurve.cpp
@@ -963,6 +963,11 @@ void Style::init(bool initial)
 #endif
 
         if (initial) {
+#ifdef Q_OS_MAC
+            if (opts.nonnativeMenubarApps.contains("kde") || opts.nonnativeMenubarApps.contains(appName)) {
+                QCoreApplication::setAttribute(Qt::AA_DontUseNativeMenuBar);
+            }
+#endif
             QDBusConnection::sessionBus().connect(
                 QString(), "/KGlobalSettings", "org.kde.KGlobalSettings",
                 "notifyChange", this, SLOT(kdeGlobalSettingsChange(int, int)));
@@ -4429,7 +4434,7 @@ void Style::drawPrimitive(PrimitiveElement element, const QStyleOption *option,
                     // get the font for adjusting; widget will be the menu holding the menuitem
                     // but that uses the same font as the menuitem.
                     QFont font(widget->font());
-#ifndef Q_OS_OSX
+#ifndef Q_OS_MAC
                     font.setBold(true);
 #else
                     // not that there is much variation in the CheckMark symbol on OS X, but
diff --git qt5/common/common.h qt5/common/common.h
index 045c13a..e991810 100644
--- qt5/common/common.h
+++ qt5/common/common.h
@@ -878,7 +878,8 @@ typedef struct {
                      statusbarApps,
                      useQtFileDialogApps,
                      windowDragWhiteList,
-                     windowDragBlackList;
+                     windowDragBlackList,
+                     nonnativeMenubarApps;
     bool             onlyTicksInMenu;
     // these are runtime constants for the user-option "onlyTicksInMenu":
     QFont            tickFont;
diff --git qt5/common/config_file.cpp qt5/common/config_file.cpp
index 13ba077..7630f4e 100644
--- qt5/common/config_file.cpp
+++ qt5/common/config_file.cpp
@@ -1343,6 +1343,7 @@ bool qtcReadConfig(const QString &file, Options *opts, Options *defOpts, bool ch
             CFG_READ_STRING_LIST(menubarApps);
             CFG_READ_STRING_LIST(statusbarApps);
             CFG_READ_STRING_LIST(useQtFileDialogApps);
+            CFG_READ_STRING_LIST(nonnativeMenubarApps);
             CFG_READ_STRING_LIST(windowDragWhiteList);
             CFG_READ_STRING_LIST(windowDragBlackList);
             readDoubleList(cfg, "customShades", opts->customShades, QTC_NUM_STD_SHADES);
@@ -2368,6 +2369,7 @@ bool qtcWriteConfig(KConfig *cfg, const Options &opts, const Options &def, bool
         CFG_WRITE_STRING_LIST_ENTRY(menubarApps);
         CFG_WRITE_STRING_LIST_ENTRY(statusbarApps);
         CFG_WRITE_STRING_LIST_ENTRY(useQtFileDialogApps);
+        CFG_WRITE_STRING_LIST_ENTRY(nonnativeMenubarApps);
 
         for(int i=APPEARANCE_CUSTOM1; i<(APPEARANCE_CUSTOM1+NUM_CUSTOM_GRAD); ++i)
         {
diff --git qt5/config/qtcurveconfig.cpp qt5/config/qtcurveconfig.cpp
index 91b6471..df8ce89 100644
--- qt5/config/qtcurveconfig.cpp
+++ qt5/config/qtcurveconfig.cpp
@@ -1077,7 +1077,7 @@ QtCurveConfig::QtCurveConfig(QWidget *parent)
 
     for (auto *w: {noBgndGradientApps, noBgndOpacityApps,
                 noMenuBgndOpacityApps, noBgndImageApps, useQtFileDialogApps,
-                menubarApps, statusbarApps, noMenuStripeApps}) {
+                menubarApps, statusbarApps, noMenuStripeApps, nonnativeMenubarApps}) {
         connect(qtcSlot(w, editingFinished), qtcSlot(this, updateChanged));
     }
 
@@ -3150,6 +3150,7 @@ void QtCurveConfig::setOptions(Options &opts)
     opts.menubarApps=toSet(menubarApps->text());
     opts.statusbarApps=toSet(statusbarApps->text());
     opts.noMenuStripeApps=toSet(noMenuStripeApps->text());
+    opts.nonnativeMenubarApps=toSet(nonnativeMenubarApps->text());
 
     if(IMG_FILE==opts.bgndImage.type)
     {
@@ -3501,6 +3502,7 @@ void QtCurveConfig::setWidgetOptions(const Options &opts)
     menubarApps->setText(toString(opts.menubarApps));
     statusbarApps->setText(toString(opts.statusbarApps));
     noMenuStripeApps->setText(toString(opts.noMenuStripeApps));
+    nonnativeMenubarApps->setText(toString(opts.nonnativeMenubarApps));
 
     bgndImageDlg->set(getThemeFile(opts.bgndImage.pixmap.file), opts.bgndImage.width, opts.bgndImage.height,
                       opts.bgndImage.pos, opts.bgndImage.onBorder);
@@ -3804,6 +3806,7 @@ bool QtCurveConfig::settingsChanged(const Options &opts)
          toSet(menubarApps->text())!=opts.menubarApps ||
          toSet(statusbarApps->text())!=opts.statusbarApps ||
          toSet(noMenuStripeApps->text())!=opts.noMenuStripeApps ||
+         toSet(nonnativeMenubarApps->text())!=opts.nonnativeMenubarApps ||
 
          diffShades(opts) ||
 
diff --git qt5/config/qtcurveconfigbase.ui qt5/config/qtcurveconfigbase.ui
index 35044bc..863e818 100644
--- qt5/config/qtcurveconfigbase.ui
+++ qt5/config/qtcurveconfigbase.ui
@@ -721,7 +721,7 @@
           <string>&lt;p&gt;Controls whether a group boxe's label should be centred, or left/right aligned.&lt;/p&gt;</string>
          </property>
          <property name="text">
-          <string>Centred (KDE4 only)</string>
+          <string>Centred (KDE only)</string>
          </property>
         </widget>
        </item>
@@ -793,7 +793,7 @@
        <item row="4" column="1">
         <widget class="QCheckBox" name="gtkComboMenus">
          <property name="toolTip">
-          <string>&lt;p&gt;In Gtk, the list that is attached to a combo-box appears in the same style as a popup-menu. KDE4 has the ability to mimic this look.&lt;/p&gt;</string>
+          <string>&lt;p&gt;In Gtk, the list that is attached to a combo-box appears in the same style as a popup-menu. KDE has the ability to mimic this look.&lt;/p&gt;</string>
          </property>
          <property name="text">
           <string>Gtk-style</string>
@@ -2450,7 +2450,7 @@
        <item row="10" column="1">
         <widget class="QCheckBox" name="titlebarButtons_useHover">
          <property name="toolTip">
-          <string>&lt;p&gt;If enabled, symbols will use the KDE4 hover color when moused-over.</string>
+          <string>&lt;p&gt;If enabled, symbols will use the KDE hover color when moused-over.</string>
          </property>
          <property name="text">
           <string>Use hover color for mouse-over of symbols</string>
@@ -3732,10 +3732,10 @@
         <widget class="QCheckBox" name="stdBtnSizes">
          <property name="toolTip">
           <string>&lt;p&gt;Auto-default buttons are buttons that can become 'default' buttons - i.e. will react the user pressing 'return'.&lt;/p&gt;
-&lt;p&gt;In Gtk2 applications, most 'auto-default' buttons have a taller appearance than standard buttons. QtCurve attempts to mimic this for KDE4 applications - this may be disabled by enabling this option.&lt;/p&gt;</string>
+&lt;p&gt;In Gtk2 applications, most 'auto-default' buttons have a taller appearance than standard buttons. QtCurve attempts to mimic this for KDE applications - this may be disabled by enabling this option.&lt;/p&gt;</string>
          </property>
          <property name="text">
-          <string>Use standard size for 'auto-default' buttons (KDE4)</string>
+          <string>Use standard size for 'auto-default' buttons (KDE)</string>
          </property>
         </widget>
        </item>
@@ -3882,7 +3882,7 @@
          <item row="5" column="1">
           <widget class="KLineEdit" name="useQtFileDialogApps">
            <property name="toolTip">
-            <string>&lt;p&gt;Comma separated list of names of Qt4 applications which should not use the KDE4 file dialog.</string>
+            <string>&lt;p&gt;Comma separated list of names of Qt4 applications which should not use the KDE file dialog.</string>
            </property>
           </widget>
          </item>
@@ -3896,7 +3896,7 @@
          <item row="6" column="1">
           <widget class="KLineEdit" name="menubarApps">
            <property name="toolTip">
-            <string>&lt;p&gt;Comma separated list of names of KDE4 applications where QtCurve needs to manually save whether the menubar is hidden or not.&lt;/p&gt;&lt;p&gt;Use &lt;i&gt;kde&lt;/i&gt; to set for all KDE applications. The menubar state of Gtk2 applications is always manually saved.&lt;/p&gt;</string>
+            <string>&lt;p&gt;Comma separated list of names of KDE applications where QtCurve needs to manually save whether the menubar is hidden or not.&lt;/p&gt;&lt;p&gt;Use &lt;i&gt;kde&lt;/i&gt; to set for all KDE applications. The menubar state of Gtk2 applications is always manually saved.&lt;/p&gt;</string>
            </property>
           </widget>
          </item>
@@ -3910,7 +3910,21 @@
          <item row="7" column="1">
           <widget class="KLineEdit" name="statusbarApps">
            <property name="toolTip">
-            <string>&lt;p&gt;Comma separated list of names of KDE4 applications where QtCurve needs to manually save whether the statusbar is hidden or not.&lt;/p&gt;&lt;p&gt;Use &lt;i&gt;kde&lt;/i&gt; to set for all KDE applications. The statusbar state of Gtk2 applications is always manually saved.&lt;/p&gt;</string>
+            <string>&lt;p&gt;Comma separated list of names of KDE applications where QtCurve needs to manually save whether the statusbar is hidden or not.&lt;/p&gt;&lt;p&gt;Use &lt;i&gt;kde&lt;/i&gt; to set for all KDE applications. The statusbar state of Gtk2 applications is always manually saved.&lt;/p&gt;</string>
+           </property>
+          </widget>
+         </item>
+         <item row="8" column="0">
+          <widget class="QLabel" name="label_89a">
+           <property name="text">
+            <string>Force non-native menubar:</string>
+           </property>
+          </widget>
+         </item>
+         <item row="8" column="1">
+          <widget class="KLineEdit" name="nonnativeMenubarApps">
+           <property name="toolTip">
+            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Comma separated list of names of KDE/Qt applications where QtCurve needs to instruct Qt not to use the native menubar.&lt;/p&gt;&lt;p&gt;Try this on Mac for applications where certain menu items do not behave as expected when they are part of the menus of the Mac-style toplevel menubar, but behave correctly for instance when they are part of a popup/context menu.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
           </widget>
          </item>
diff --git qt5/style/qtcurve.cpp qt5/style/qtcurve.cpp
index ce15754..b45d893 100644
--- qt5/style/qtcurve.cpp
+++ qt5/style/qtcurve.cpp
@@ -374,18 +374,6 @@ void Style::init(bool initial)
     if(!initial)
         freeColors();
 
-#ifdef QTC_QT5_ENABLE_KDE
-    if (initial) {
-        QString name(QApplication::applicationName());
-
-        if(name.isEmpty())
-            name = qAppName();
-
-        if(name.isEmpty())
-            name = "QtApp";
-    }
-#endif
-
     if (m_isPreview) {
         if (m_isPreview != PREVIEW_WINDOW) {
             opts.bgndOpacity = opts.dlgOpacity = opts.menuBgndOpacity = 100;
@@ -394,6 +382,11 @@ void Style::init(bool initial)
         qtcReadConfig(QString(), &opts);
 
         if (initial) {
+#ifdef Q_OS_OSX
+            if (opts.nonnativeMenubarApps.contains("kde") || opts.nonnativeMenubarApps.contains(appName)) {
+                QCoreApplication::setAttribute(Qt::AA_DontUseNativeMenuBar);
+            }
+#endif
             connectDBus();
         }
     }
