diff --git a/qt5/common/config_file.cpp b/qt5/common/config_file.cpp
index 7630f4e..770371f 100644
--- a/qt5/common/config_file.cpp
+++ b/qt5/common/config_file.cpp
@@ -1306,14 +1306,14 @@ bool qtcReadConfig(const QString &file, Options *opts, Options *defOpts, bool ch
             CFG_READ_ALIGN(titlebarAlignment);
             CFG_READ_EFFECT(titlebarEffect);
             CFG_READ_BOOL(centerTabText);
-#if defined CONFIG_DIALOG
+// #if defined CONFIG_DIALOG
             CFG_READ_INT(expanderHighlight);
             CFG_READ_BOOL(mapKdeIcons);
-#endif
+// #endif
             CFG_READ_BOOL(gtkButtonOrder);
-#if defined CONFIG_DIALOG
+// #if defined CONFIG_DIALOG
             CFG_READ_BOOL(reorderGtkButtons);
-#endif
+// #endif
             CFG_READ_APPEARANCE(titlebarAppearance, APP_ALLOW_NONE);
             CFG_READ_APPEARANCE(inactiveTitlebarAppearance, APP_ALLOW_NONE);
 
@@ -1336,10 +1336,10 @@ bool qtcReadConfig(const QString &file, Options *opts, Options *defOpts, bool ch
             CFG_READ_STRING_LIST(noBgndOpacityApps);
             CFG_READ_STRING_LIST(noMenuBgndOpacityApps);
             CFG_READ_STRING_LIST(noBgndImageApps);
-#ifdef CONFIG_DIALOG
+// #ifdef CONFIG_DIALOG
             if (opts->version < qtcMakeVersion(1, 7, 2))
                 opts->noMenuBgndOpacityApps << "gtk";
-#endif
+// #endif
             CFG_READ_STRING_LIST(menubarApps);
             CFG_READ_STRING_LIST(statusbarApps);
             CFG_READ_STRING_LIST(useQtFileDialogApps);
@@ -1477,9 +1475,9 @@ void qtcDefaultSettings(Options *opts)
     opts->round=ROUND_EXTRA;
     opts->gtkButtonOrder=false;
     opts->dwtAppearance=APPEARANCE_CUSTOM1;
-#if defined CONFIG_DIALOG
+// #if defined CONFIG_DIALOG
     opts->reorderGtkButtons=false;
-#endif
+// #endif
     opts->bgndImage.type=IMG_NONE;
     opts->bgndImage.width=opts->bgndImage.height=0;
     opts->bgndImage.onBorder=false;
@@ -1619,10 +1617,10 @@ void qtcDefaultSettings(Options *opts)
                             << "vmplayer";
     opts->noMenuStripeApps << "gtk" << "soffice.bin";
 
-#if defined CONFIG_DIALOG
+// #if defined CONFIG_DIALOG
     opts->mapKdeIcons=true;
     opts->expanderHighlight=DEFAULT_EXPANDER_HIGHLIGHT_FACTOR;
-#endif
+// #endif
     opts->titlebarAppearance=APPEARANCE_CUSTOM1;
     opts->inactiveTitlebarAppearance=APPEARANCE_CUSTOM1;
     opts->titlebarButtonAppearance=APPEARANCE_GRADIENT;
@@ -2354,9 +2352,9 @@ bool qtcWriteConfig(KConfig *cfg, const Options &opts, const Options &def, bool
         CFG_WRITE_ENTRY(gtkComboMenus);
         CFG_WRITE_ENTRY(doubleGtkComboArrow);
         CFG_WRITE_ENTRY(gtkButtonOrder);
-#if defined CONFIG_DIALOG
+// #if defined CONFIG_DIALOG
         CFG_WRITE_ENTRY(reorderGtkButtons);
-#endif
+// #endif
         CFG_WRITE_ENTRY(mapKdeIcons);
         CFG_WRITE_ENTRY(shading);
         CFG_WRITE_ENTRY(titlebarAlignment);
