diff --git qt5/style/qtcurve.cpp qt5/style/qtcurve.cpp
index ca9965a2..3587563c 100644
--- qt5/style/qtcurve.cpp
+++ qt5/style/qtcurve.cpp
@@ -95,6 +95,16 @@ public:
         , m_dbusConnected(false)
     {}
     ~DBusHelper()
+    {
+        disconnect();
+    }
+    void init()
+    {
+        if(!m_dBus) {
+            m_dBus = new QDBusInterface("org.kde.kwin", "/QtCurve", "org.kde.QtCurve");
+        }
+    }
+    void disconnect()
     {
         if (m_dBus) {
             m_dBus->disconnect();
@@ -363,6 +373,7 @@ Style::Style() :
     m_blurHelper(new BlurHelper(this)),
     m_shortcutHandler(new ShortcutHandler(this))
 {
+    qtcInfo("New style instance %p\n", this);
     const char *env = getenv(QTCURVE_PREVIEW_CONFIG);
 #ifdef QTC_QT5_ENABLE_KDE
     m_configFile = KSharedConfig::openConfig();
@@ -413,18 +424,7 @@ void Style::init(bool initial)
             // prepare the cleanup handler
             if (QCoreApplication::instance()) {
                 std::call_once(m_dBusHelper->m_aboutToQuitInit, [this] {
-                    connect(QCoreApplication::instance(), &QCoreApplication::aboutToQuit, this, [this] () {
-                            // disconnect from the session DBus. We're no longer interested in the
-                            // information it might send when the app we're serving is shutting down.
-                            disconnectDBus();
-                            // Stop listening to select signals. We shouldn't stop emitting signals
-                            // (like QObject::destroyed) but we can reduce the likelihood that pending
-                            // signals will be sent to us post-mortem.
-#ifdef QTC_QT5_ENABLE_KDE
-                            disconnect(KWindowSystem::self(), &KWindowSystem::compositingChanged,
-                                       this, &Style::compositingToggled);
-#endif
-                        } );
+                    connect(QCoreApplication::instance(), &QCoreApplication::aboutToQuit, this, &Style::aboutToQuit);
                 } );
             }
         }
@@ -700,7 +700,25 @@ void Style::connectDBus()
 {
     if (m_dBusHelper->m_dbusConnected)
         return;
-    auto bus = QDBusConnection::sessionBus();
+    static const QStringList excludeKnown = {QStringLiteral("rkward"), QStringLiteral("mscore"), QStringLiteral("adg")};
+    QStringList userExcludes = {};
+    if (qEnvironmentVariableIsSet("QTCURVE_NO_DBUS_APPS")) {
+        userExcludes = QString::fromUtf8(qgetenv("QTCURVE_NO_DBUS_APPS")).split(':');
+        if (qEnvironmentVariableIsSet("QTCURVE_DEBUG") && !userExcludes.isEmpty()) {
+            qDebug() << Q_FUNC_INFO << this << "user list of apps not to have a DBus connection:" << userExcludes;
+        }
+    }
+    if (excludeKnown.contains(appName) || userExcludes.contains(appName)) {
+        // these have a way of juggling styles that interferes with our cleanup and can cause a hang just before exit;
+        // avoid that by not connecting to any DBus signals. The only thing they lose is runtime reflection of
+        // theming changes; a small price to pay.
+        m_dBusHelper->m_dbusConnected = false;
+        if (qEnvironmentVariableIsSet("QTCURVE_DEBUG")) {
+            qWarning() << Q_FUNC_INFO << this << "no DBus connection for" << appName;
+        }
+        return;
+    }
+    QDBusConnection bus(QDBusConnection::sessionBus());
     if (bus.isConnected()) {
         m_dBusHelper->m_dbusConnected = true;
         bus.connect(QString(), "/KGlobalSettings", "org.kde.KGlobalSettings",
@@ -732,12 +750,12 @@ void Style::disconnectDBus()
 {
     if (!m_dBusHelper->m_dbusConnected)
         return;
-    auto bus = QDBusConnection::sessionBus();
+    QDBusConnection bus(QDBusConnection::sessionBus());
+    m_dBusHelper->m_dbusConnected = false;
     if (!bus.isConnected())
         return;
-    m_dBusHelper->m_dbusConnected = false;
     if (getenv("QTCURVE_DEBUG")) {
-        qWarning() << Q_FUNC_INFO << this << "Disconnecting from" << bus.name() << "/" << bus.baseService();
+        qWarning() << Q_FUNC_INFO << this << appName << "disconnecting from" << bus.name() << "/" << bus.baseService();
         dumpObjectInfo();
     }
     bus.disconnect(QString(), "/KGlobalSettings", "org.kde.KGlobalSettings",
@@ -764,6 +782,33 @@ void Style::disconnectDBus()
     }
 }
 
+void Style::aboutToQuit()
+{
+    // be prepared for the global takedown. In particular, try to avoid having
+    // to deal with events and signals when it is no longer safe to do so, in
+    // an attempt to address the crashing on exit in certain applications.
+    if (getenv("QTCURVE_DEBUG")) {
+        qWarning() << Q_FUNC_INFO << this << appName << "is about to quit";
+    }
+    // disconnect from the session DBus. We're no longer interested in the
+    // information it might send when the app we're serving is shutting down.
+    disconnectDBus();
+    m_dBusHelper->disconnect();
+    // Stop listening to select signals. We shouldn't stop emitting signals
+    // (like QObject::destroyed) but we can reduce the likelihood that pending
+    // signals will be sent to us post-mortem.
+#ifdef QTC_QT5_ENABLE_KDE
+    disconnect(KWindowSystem::self(), &KWindowSystem::compositingChanged,
+               this, &Style::compositingToggled);
+#endif
+    if (QGuiApplication::platformName() != QLatin1String("cocoa")) {
+        // let's process all pending events now, rather than allow them to come
+        // in later when they can no longer be handled safely. Don't do this
+        // when running under Apple Cocoa (MuseScore can crash because of it).
+        QCoreApplication::processEvents();
+    }
+}
+
 Style::~Style()
 {
     qtcInfo("Deleting style instance %p\n", this);
@@ -4481,11 +4526,11 @@ void Style::emitMenuSize(QWidget *w, unsigned short size, bool force)
         if (oldSize != size) {
             w->setProperty(constMenuSizeProperty, size);
             qtcX11SetMenubarSize(wid, size);
-            if(!m_dBusHelper->m_dBus)
-                m_dBusHelper->m_dBus = new QDBusInterface("org.kde.kwin", "/QtCurve",
-                                             "org.kde.QtCurve");
+#ifndef Q_OS_MACOS
+            m_dBusHelper->init();
             m_dBusHelper->m_dBus->call(QDBus::NoBlock, "menuBarSize",
                           (unsigned int)wid, (int)size);
+#endif
         }
     }
 }
@@ -4493,12 +4538,12 @@ void Style::emitMenuSize(QWidget *w, unsigned short size, bool force)
 void Style::emitStatusBarState(QStatusBar *sb)
 {
     if (opts.statusbarHiding & HIDE_KWIN) {
-        if (!m_dBusHelper->m_dBus)
-            m_dBusHelper->m_dBus = new QDBusInterface("org.kde.kwin", "/QtCurve",
-                                        "org.kde.QtCurve");
+#ifndef Q_OS_MACOS
+        m_dBusHelper->init();
         m_dBusHelper->m_dBus->call(QDBus::NoBlock, "statusBarState",
                      (unsigned int)qtcGetWid(sb->window()),
                      sb->isVisible());
+#endif
     }
 }
 }
diff --git qt5/style/qtcurve.h qt5/style/qtcurve.h
index 00072d4e..8272604f 100644
--- qt5/style/qtcurve.h
+++ qt5/style/qtcurve.h
@@ -187,6 +187,7 @@ public:
 private:
     void init(bool initial);
     void connectDBus();
+    void disconnectDBus();
     void freeColor(QSet<QColor*> &freedColors, QColor **cols);
     void freeColors();
     void polishFormLayout(QFormLayout *layout);
@@ -361,7 +362,7 @@ private:
     int getFrameRound(const QWidget *widget) const;
 
 private slots:
-    void disconnectDBus();
+    void aboutToQuit();
     void kdeGlobalSettingsChange(int type, int);
     void borderSizesChanged();
     void toggleMenuBar(unsigned int xid);
diff --git qt5/style/qtcurve_plugin.cpp qt5/style/qtcurve_plugin.cpp
index 664bc486..29d982be 100644
--- qt5/style/qtcurve_plugin.cpp
+++ qt5/style/qtcurve_plugin.cpp
@@ -41,6 +41,9 @@
 #endif
 #include <QDebug>
 
+#include <sys/time.h>
+#include <signal.h>
+
 #include <qtcurve-utils/log.h>
 
 namespace QtCurve {
@@ -151,6 +154,13 @@ void StylePlugin::unregisterCallback()
     }
 }
 
+static char *appName = nullptr;
+static void handleALRM(int sig)
+{
+    fprintf(stderr, "%s: final shutdown taking longer than 20 seconds: bailing (consider adding the app name to QTCURVE_NO_DBUS_APPS!)\n", appName);
+    raise(SIGHUP);
+}
+
 StylePlugin::~StylePlugin()
 {
     qtcInfo("Deleting QtCurve plugin (%p)\n", this);
@@ -165,7 +175,17 @@ StylePlugin::~StylePlugin()
                 qtcDebug("Taking back ownership of instance %p from parent %p\n", that, that->parent());
                 that->setParent(nullptr);
             }
+            // let's give us 20 seconds to exit completely
+            static struct itimerval rtt, ortt;
+            rtt.it_value.tv_sec= (unsigned long) 20, rtt.it_value.tv_usec= (unsigned long) 0;
+            rtt.it_interval.tv_sec= 0, rtt.it_interval.tv_usec= 0;
+            const auto old_handler = signal(SIGALRM, handleALRM);
+            setitimer(ITIMER_REAL, &rtt, &ortt);
+            appName = const_cast<char*>(qPrintable(qAppName()));
             delete that;
+            qtcDebug("Style instance %p deleted, revoking bail-out timer\n", that);
+            setitimer(ITIMER_REAL, &ortt, nullptr);
+            signal(SIGALRM, old_handler);
         }
         m_styleInstances.clear();
     }
