diff --git qt5/common/config_file.cpp qt5/common/config_file.cpp
index b402b89..9b21643 100644
--- qt5/common/config_file.cpp
+++ qt5/common/config_file.cpp
@@ -650,7 +650,7 @@ static bool readBoolEntry(QtCConfig &cfg, const QString &key, bool def)
 {
     const QString &val(readStringEntry(cfg, key));
 
-    return val.isEmpty() ? def : (val=="true" ? true : false);
+    return val.isEmpty() ? def : (val==QLatin1String("true") ? true : false);
 }
 
 static void readDoubleList(QtCConfig &cfg, const char *key, double *list, int count)
diff --git qt5/common/kf5_utils.h qt5/common/kf5_utils.h
index 7fc391c..4249f7a 100644
--- qt5/common/kf5_utils.h
+++ qt5/common/kf5_utils.h
@@ -41,6 +41,7 @@ namespace QtCurve {
 QDialogButtonBox *createDialogButtonBox(QDialog *dialog);
 
 class InputDialog : public QDialog {
+    Q_OBJECT
 public:
     explicit InputDialog(QWidget *parent=nullptr, Qt::WindowFlags=0);
 
@@ -75,11 +76,11 @@ loadKIcon(const QString &name)
 
 // TODO probably merge with utils/dirs
 static inline QString
-saveLocation(QStandardPaths::StandardLocation type, const QString &suffix)
+saveLocation(QStandardPaths::StandardLocation type, const char *suffix)
 {
     QString path = QStandardPaths::writableLocation(type);
     QTC_RET_IF_FAIL(!path.isEmpty(), path);
-    path += '/' + suffix;
+    path += '/' + QString::fromUtf8(suffix);
     QDir().mkpath(path);
     return path;
 }
diff --git qt5/config/imagepropertiesdialog.h qt5/config/imagepropertiesdialog.h
index 02cf6d3..b8d0864 100644
--- qt5/config/imagepropertiesdialog.h
+++ qt5/config/imagepropertiesdialog.h
@@ -32,6 +32,7 @@
 #include "ui_imageproperties.h"
 
 class CImagePropertiesDialog : public QDialog, public Ui::ImageProperties {
+    Q_OBJECT
 public:
     enum {
         BASIC = 0x00,
diff --git qt5/config/qtcurveconfig.cpp qt5/config/qtcurveconfig.cpp
index cce4883..1afdfbf 100644
--- qt5/config/qtcurveconfig.cpp
+++ qt5/config/qtcurveconfig.cpp
@@ -120,20 +120,22 @@ static inline QString getFileName(const QString &f)
 
 static QString getThemeFile(const QString &file)
 {
+    QLatin1String doubleSlash("//");
+    QLatin1String slash("/");
     if (file.startsWith(THEME_IMAGE_PREFIX BGND_FILE)) {
         QString f(QtCurve::getConfDir() + file);
 
         if (QFile::exists(f)) {
-            return f.replace("//", "/");
+            return f.replace(doubleSlash, slash);
         }
     }
-    if (!file.startsWith("/")) {
+    if (!file.startsWith(slash)) {
         QString f(QtCurve::qtcSaveDir() + file);
         if (QFile::exists(f)) {
-            return f.replace("//", "/");
+            return f.replace(doubleSlash, slash);
         }
     }
-    return QString(file).replace("//", "/");
+    return QString(file).replace(doubleSlash, slash);
 }
 
 static void removeFile(const QString &f)
@@ -285,6 +287,7 @@ QSize CStylePreview::sizeHint() const
 
 class CWorkspace : public QMdiArea
 {
+    Q_OBJECT
     public:
 
     CWorkspace(QWidget *parent) : QMdiArea(parent)
@@ -305,6 +308,7 @@ class CWorkspace : public QMdiArea
 };
 
 class CharSelectDialog : public QDialog {
+    Q_OBJECT
 public:
     CharSelectDialog(QWidget *parent, int v)
         : QDialog(parent)
diff --git qt5/style/qtcurve.cpp qt5/style/qtcurve.cpp
index 029992e..538d130 100644
--- qt5/style/qtcurve.cpp
+++ qt5/style/qtcurve.cpp
@@ -318,7 +318,7 @@ static void parseWindowLine(const QString &line, QList<int> &data)
 }
 #endif
 
-Style::Style() :
+Style::Style(QObject *parent) :
     m_popupMenuCols(0L),
     m_sliderCols(0L),
     m_defBtnCols(0L),
diff --git qt5/style/qtcurve.h qt5/style/qtcurve.h
index f4d5b0a..09ffc52 100644
--- qt5/style/qtcurve.h
+++ qt5/style/qtcurve.h
@@ -112,8 +112,8 @@ public:
         ICN_UNSHADE
     };
 
-    Style();
-    ~Style();
+    explicit Style(QObject *parent=0);
+    virtual ~Style();
 
     void polish(QApplication *app) override;
     void polish(QPalette &palette) override;
