diff --git qt5/style/qtcurve.cpp qt5/style/qtcurve.cpp
index 21a879aa80b39b6a970f6c97c68f2b163f648654..681d87dbc70019494f1b5111a71d60561dc6e15e 100644
--- qt5/style/qtcurve.cpp
+++ qt5/style/qtcurve.cpp
@@ -420,6 +420,10 @@ void Style::init(bool initial)
             if (QCoreApplication::instance()) {
                 std::call_once(m_dBusHelper->m_aboutToQuitInit, [this] {
                     connect(QCoreApplication::instance(), &QCoreApplication::aboutToQuit, this, [this] () {
+                            if (getenv("QTCURVE_DEBUG")) {
+                                qWarning() << Q_FUNC_INFO << this << "About to Quit";
+                                this->dumpObjectInfo();
+                            }
                             // disconnect from the session DBus. We're no longer interested in the
                             // information it might send when the app we're serving is shutting down.
                             disconnectDBus();
@@ -704,9 +708,10 @@ void Style::init(bool initial)
 
 void Style::connectDBus()
 {
+#ifndef Q_OS_MACOS
     if (m_dBusHelper->m_dbusConnected)
         return;
-    auto bus = QDBusConnection::sessionBus();
+    QDBusConnection bus(QDBusConnection::sessionBus());
     if (bus.isConnected()) {
         m_dBusHelper->m_dbusConnected = true;
         bus.connect(QString(), "/KGlobalSettings", "org.kde.KGlobalSettings",
@@ -733,13 +738,15 @@ void Style::connectDBus()
             }
         }
     }
+#endif
 }
 
 void Style::disconnectDBus()
 {
+#ifndef Q_OS_MACOS
     if (!m_dBusHelper->m_dbusConnected)
         return;
-    auto bus = QDBusConnection::sessionBus();
+    QDBusConnection bus(QDBusConnection::sessionBus());
     if (!bus.isConnected())
         return;
     m_dBusHelper->m_dbusConnected = false;
@@ -770,6 +777,7 @@ void Style::disconnectDBus()
                            this, SLOT(toggleStatusBar(unsigned int)));
         }
     }
+#endif
 }
 
 Style::~Style()
@@ -4507,11 +4515,13 @@ void Style::emitMenuSize(QWidget *w, unsigned short size, bool force)
         if (oldSize != size) {
             w->setProperty(constMenuSizeProperty, size);
             qtcX11SetMenubarSize(wid, size);
+#ifndef Q_OS_MACOS
             if(!m_dBusHelper->m_dBus)
                 m_dBusHelper->m_dBus = new QDBusInterface("org.kde.kwin", "/QtCurve",
                                              "org.kde.QtCurve");
             m_dBusHelper->m_dBus->call(QDBus::NoBlock, "menuBarSize",
                           (unsigned int)wid, (int)size);
+#endif
         }
     }
 }
@@ -4519,12 +4529,14 @@ void Style::emitMenuSize(QWidget *w, unsigned short size, bool force)
 void Style::emitStatusBarState(QStatusBar *sb)
 {
     if (opts.statusbarHiding & HIDE_KWIN) {
+#ifndef Q_OS_MACOS
         if (!m_dBusHelper->m_dBus)
             m_dBusHelper->m_dBus = new QDBusInterface("org.kde.kwin", "/QtCurve",
                                         "org.kde.QtCurve");
         m_dBusHelper->m_dBus->call(QDBus::NoBlock, "statusBarState",
                      (unsigned int)qtcGetWid(sb->window()),
                      sb->isVisible());
+#endif
     }
 }
 }
diff --git qt5/style/qtcurve_plugin.cpp qt5/style/qtcurve_plugin.cpp
index 481fffca..182563b9 100644
--- qt5/style/qtcurve_plugin.cpp
+++ qt5/style/qtcurve_plugin.cpp
@@ -156,12 +156,15 @@ StylePlugin::~StylePlugin()
     qtcInfo("Deleting QtCurve plugin (%p)\n", this);
     if (!m_styleInstances.isEmpty()) {
         qtcWarn("there remain(s) %d Style instance(s)\n", m_styleInstances.count());
-        foreach (Style *that, m_styleInstances) {
-            // don't let ~Style() touch m_styleInstances from here.
-            that->m_plugin = nullptr;
-            // each instance should already have disconnected from the D-Bus
-            // and disconnected from receiving select signals.
-            delete that;
+        if (qApp->applicationName() != "rkward" && qApp->applicationName() != "Affinic Debugger"
+             && qApp->applicationName() != "mscore") {
+            foreach (Style *that, m_styleInstances) {
+                // don't let ~Style() touch m_styleInstances from here.
+                that->m_plugin = nullptr;
+                // each instance should already have disconnected from the D-Bus
+                // and disconnected from receiving select signals.
+                delete that;
+            }
         }
         m_styleInstances.clear();
     }
