diff --git qt5/style/qtcurve.cpp qt5/style/qtcurve.cpp
index b45d893..42e0f35 100644
--- qt5/style/qtcurve.cpp
+++ qt5/style/qtcurve.cpp
@@ -736,8 +738,10 @@ void Style::freeColor(QSet<QColor *> &freedColors, QColor **cols)
 
 void Style::freeColors()
 {
-    if(0!=m_progressBarAnimateTimer)
+    if(0!=m_progressBarAnimateTimer) {
         killTimer(m_progressBarAnimateTimer);
+        m_progressBarAnimateTimer = 0;
+    }
 
     QSet<QColor*> freedColors;
 
diff --git qt5/style/qtcurve_api.cpp qt5/style/qtcurve_api.cpp
index 497986a..f1090bd 100644
--- qt5/style/qtcurve_api.cpp
+++ qt5/style/qtcurve_api.cpp
@@ -1364,7 +1364,7 @@ bool Style::eventFilter(QObject *object, QEvent *event)
         // So we have to check on object.
         if (object && !m_progressBars.isEmpty()) {
             m_progressBars.remove(reinterpret_cast<QProgressBar*>(object));
-            if (m_progressBars.isEmpty()) {
+            if (m_progressBars.isEmpty() && m_progressBarAnimateTimer) {
                 killTimer(m_progressBarAnimateTimer);
                 m_progressBarAnimateTimer = 0;
             }
diff --git qt4/style/qtcurve.cpp qt4/style/qtcurve.cpp
index 582bcf7..5447e07 100644
--- qt4/style/qtcurve.cpp
+++ qt4/style/qtcurve.cpp
@@ -1285,8 +1285,10 @@ void Style::freeColor(QSet<QColor*> &freedColors, QColor **cols)
 
 void Style::freeColors()
 {
-    if(0!=m_progressBarAnimateTimer)
+    if(0!=m_progressBarAnimateTimer) {
         killTimer(m_progressBarAnimateTimer);
+        m_progressBarAnimateTimer = 0;
+    }
 
     QSet<QColor*> freedColors;
 
@@ -2755,7 +2757,7 @@ bool Style::eventFilter(QObject *object, QEvent *event)
             if(object && !m_progressBars.isEmpty())
             {
                 m_progressBars.remove(reinterpret_cast<QProgressBar*>(object));
-                if (m_progressBars.isEmpty())
+                if (m_progressBars.isEmpty() && m_progressBarAnimateTimer)
                 {
                     killTimer(m_progressBarAnimateTimer);
                     m_progressBarAnimateTimer = 0;
