diff --git qt5/style/qtcurve_utils.cpp qt5/style/qtcurve_utils.cpp
index 608876b..057d4f1 100644
--- qt5/style/qtcurve_utils.cpp
+++ qt5/style/qtcurve_utils.cpp
@@ -238,16 +238,8 @@ updateMenuBarEvent(QMouseEvent *event, QMenuBar *menu)
         }
     };
 
-    struct HackedMenu: public QMenuBar {
-        void
-        send(QMouseEvent *ev)
-        {
-            event(ev);
-        }
-    };
-
     if (((HackEvent*)event)->adjust()) {
-        ((HackedMenu*)menu)->send(event);
+        static_cast<QObject*>(menu)->event(event);
         return true;
     }
     return false;
diff --git qt4/style/qtcurve.cpp qt4/style/qtcurve.cpp
index 315a507..c675cc3 100644
--- qt4/style/qtcurve.cpp
+++ qt4/style/qtcurve.cpp
@@ -2348,16 +2348,8 @@ static bool updateMenuBarEvent(QMouseEvent *event, QMenuBar *menu)
         }
     };
 
-    struct HackedMenu : public QMenuBar {
-        void
-        send(QMouseEvent *ev)
-        {
-            event(ev);
-        }
-    };
-
     if (((HackEvent*)event)->adjust()) {
-        ((HackedMenu*)menu)->send(event);
+        static_cast<QObject*>(menu)->event(event);
         return true;
     }
     return false;
