--- script/lib/orig.install-application.js	2018-06-21 18:28:06.000000000 +0200
+++ script/lib/install-application.js	2018-06-26 17:45:40.819381910 +0200
@@ -27,7 +27,7 @@
  * https://standards.freedesktop.org/basedir-spec/basedir-spec-latest.html#variables
  */
 function findBaseIconThemeDirPath () {
-  const defaultBaseIconThemeDir = '/usr/share/icons/hicolor'
+  const defaultBaseIconThemeDir = '@@PREFIX@@/share/icons/hicolor'
   const dataDirsString = process.env.XDG_DATA_DIRS
   if (dataDirsString) {
     const dataDirs = dataDirsString.split(path.delimiter)
