diff --git a/lib/libzfs/libzfs_util.c b/lib/libzfs/libzfs_util.c
index a1e8db8..4834fde 100644
--- a/lib/libzfs/libzfs_util.c
+++ b/lib/libzfs/libzfs_util.c
@@ -741,7 +741,7 @@ libzfs_run_process(const char *path, char *argv[], int flags)
 static int
 libzfs_load_module(const char *module)
 {
-	char *argv[4] = {"/sbin/kextload", NULL, (char *)0};
+	char *argv[6] = {"/sbin/kextload", "-r", NULL, NULL, (char *)0};
 	char *modpath = NULL;
 	struct stat s1, s2;
 	int ret;
@@ -785,9 +785,10 @@ libzfs_load_module(const char *module)
 
 	if( ret == 0 ){
 		// we have a candidate
-		argv[1] = modpath;
+		argv[2] = KERNEL_MODPREFIX;
+		argv[3] = modpath;
 		if (!libzfs_module_loaded(module)){
-			// fprintf( stderr, gettext("Executing `%s %s`\n"), argv[0], argv[1] );
+			// fprintf( stderr, gettext("Executing `%s %s %s %s`\n"), argv[0], argv[1], argv[2], argv[3] );
 			ret = libzfs_run_process("/sbin/kextload", argv, 0);
 			if (ret == 0) {
 				int try = 0;
@@ -804,7 +805,7 @@ libzfs_load_module(const char *module)
 }
 
 #define	MODLOAD_CMD \
-	"/sbin/kextload " KERNEL_MODPREFIX "/zfs.kext"
+	"/sbin/kextload -r " KERNEL_MODPREFIX " " KERNEL_MODPREFIX "/zfs.kext"
 
 libzfs_handle_t *
 libzfs_init(void)
