diff --git a/Makefile.am b/Makefile.am
index 4b4b8fd..611ea97 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -74,18 +74,18 @@ pkg-utils: @DEFAULT_PACKAGE@-utils
 
 if CONFIG_USER
 install-data-hook::
-	@echo
-	for plistfile in $(launchddaemon_DATA); do \
-		bn=`basename "$$plistfile"`; \
-		echo "Attempting to unload $$bn ..."; \
-		$(DESTDIR)/bin/launchctl unload -F \
-		$(DESTDIR)$(launchddaemondir)/"$$bn" \
-		    || echo "Ignoring errors"; \
-		echo "Attempting to load $$bn ..."; \
-		$(DESTDIR)/bin/launchctl load -wF \
-		$(DESTDIR)$(launchddaemondir)/"$$bn" \
-		    || echo "Ignoring errors"; \
-	done
+#	@echo
+#	for plistfile in $(launchddaemon_DATA); do \
+#		bn=`basename "$$plistfile"`; \
+#		echo "Attempting to unload $$bn ..."; \
+#		$(DESTDIR)/bin/launchctl unload -F \
+#		$(DESTDIR)$(launchddaemondir)/"$$bn" \
+#		    || echo "Ignoring errors"; \
+#		echo "Attempting to load $$bn ..."; \
+#		$(DESTDIR)/bin/launchctl load -wF \
+#		$(DESTDIR)$(launchddaemondir)/"$$bn" \
+#		    || echo "Ignoring errors"; \
+#	done
 	echo "Install done."
 
 uninstall-hook::
