diff --git plugins/qmakemanager/qmakemanager.cpp plugins/qmakemanager/qmakemanager.cpp
index e9dd77987ed263d925135c627ffcd281cba87d9a..f725eae125524a2d8187a18084bbca1a8d13fa49 100644
--- plugins/qmakemanager/qmakemanager.cpp
+++ plugins/qmakemanager/qmakemanager.cpp
@@ -245,9 +245,16 @@ ProjectFolderItem* QMakeProjectManager::buildFolderItem(IProject* project, const
             // new project
             QMakeFolderItem* root = dynamic_cast<QMakeFolderItem*>(project->projectItem());
             Q_ASSERT(root);
-            qmscope->setMkSpecs(root->projectFiles().first()->mkSpecs());
-            if (root->projectFiles().first()->qmakeCache()) {
-                qmscope->setQMakeCache(root->projectFiles().first()->qmakeCache());
+            if (root && !root->projectFiles().isEmpty()) {
+              qmscope->setMkSpecs(root->projectFiles().first()->mkSpecs());
+              if (root->projectFiles().first()->qmakeCache()) {
+                  qmscope->setQMakeCache(root->projectFiles().first()->qmakeCache());
+              }
+            } else {
+              qCWarning(KDEV_QMAKE) << "Project" << project->name() << "does not have a valid root projectItem or projectFiles"
+                << project->projectItem() << (root ? root->projectFiles() : QList<QMakeProjectFile*>());
+              delete qmscope;
+              return nullptr;
             }
         }
 
