diff --git plugins/cmake/cmakeutils.cpp plugins/cmake/cmakeutils.cpp
index b2fd618f03..133bf9eb1f 100644
--- plugins/cmake/cmakeutils.cpp
+++ plugins/cmake/cmakeutils.cpp
@@ -314,7 +314,10 @@ QString currentBuildType( KDevelop::IProject* project, int builddir )
 
 QString findExecutable()
 {
-    auto cmake = QStandardPaths::findExecutable(QStringLiteral("cmake"));
+    auto cmake = QStandardPaths::findExecutable(QStringLiteral("cmake"), QStringList(QStringLiteral("@PREFIX@/bin")));
+    if (cmake.isEmpty()) {
+        cmake = QStandardPaths::findExecutable(QStringLiteral("cmake"));
+    }
 #ifdef Q_OS_WIN
     if (cmake.isEmpty())
         cmake = QStandardPaths::findExecutable("cmake",{
@@ -332,7 +335,7 @@ KDevelop::Path currentInstallDir( KDevelop::IProject* project, int builddir )
 #ifdef Q_OS_WIN
         QStringLiteral("C:\\Program Files");
 #else
-        QStringLiteral("/usr/local");
+        QStringLiteral("@PREFIX@");
 #endif
     return KDevelop::Path(readBuildDirParameter( project, Config::Specific::cmakeInstallDirKey, defaultInstallDir, builddir ));
 }
