diff --git plugins/customscript/customscript_plugin.cpp plugins/customscript/customscript_plugin.cpp
index bd15aa4968..8a1d96e311 100644
--- plugins/customscript/customscript_plugin.cpp
+++ plugins/customscript/customscript_plugin.cpp
@@ -244,12 +244,12 @@ KDevelop::SourceFormatterStyle CustomScriptPlugin::predefinedStyle(const QString
         result.setCaption(i18n("Gnu Indent: Original Berkeley indent style"));
         result.setContent(QStringLiteral("indent -orig"));
         result.setUsePreview(true);
-    } else if (name == QLatin1String("kdev_format_source")) {
-        result.setCaption(QStringLiteral("KDevelop: kdev_format_source"));
-        result.setContent(QStringLiteral("kdev_format_source $FILE $TMPFILE"));
+    } else if (name == QLatin1String("kdev_format_source5")) {
+        result.setCaption(QStringLiteral("KDevelop: kdev_format_source5"));
+        result.setContent(QStringLiteral("kdev_format_source5 $FILE $TMPFILE"));
         result.setUsePreview(false);
         result.setDescription(i18n("Description:<br />"
-                                   "<b>kdev_format_source</b> is a script bundled with KDevelop "
+                                   "<b>kdev_format_source5</b> is a script bundled with KDevelop "
                                    "which allows using fine-grained formatting rules by placing "
                                    "meta-files called <b>format_sources</b> into the file-system.<br /><br />"
                                    "Each line of the <b>format_sources</b> files defines a list of wildcards "
@@ -276,7 +276,7 @@ KDevelop::SourceFormatterStyle CustomScriptPlugin::predefinedStyle(const QString
 QList<KDevelop::SourceFormatterStyle> CustomScriptPlugin::predefinedStyles()
 {
     QList<KDevelop::SourceFormatterStyle> styles = stylesFromLanguagePlugins();
-    styles << predefinedStyle(QStringLiteral("kdev_format_source"));
+    styles << predefinedStyle(QStringLiteral("kdev_format_source5"));
     styles << predefinedStyle(QStringLiteral("GNU_indent_GNU"));
     styles << predefinedStyle(QStringLiteral("GNU_indent_KR"));
     styles << predefinedStyle(QStringLiteral("GNU_indent_orig"));
