diff --git a/kdevplatform/project/CMakeLists.txt b/kdevplatform/project/CMakeLists.txt
index c61e8de32e80c29a648766fe88c62b78f5561c36..8fdf6df3a6a18ad2867be8a0d06dd631d6884a5b 100644
--- a/kdevplatform/project/CMakeLists.txt
+++ b/kdevplatform/project/CMakeLists.txt
@@ -13,6 +13,7 @@ set(KDevPlatformProject_LIB_SRCS
     debug.cpp
     projectproxymodel.cpp
     abstractfilemanagerplugin.cpp
+    projectwatcher.cpp
     filemanagerlistjob.cpp
     projectfiltermanager.cpp
     interfaces/iprojectbuilder.cpp
diff --git a/kdevplatform/project/abstractfilemanagerplugin.cpp b/kdevplatform/project/abstractfilemanagerplugin.cpp
index 0f43c7ec55b9e2ea1120b2f6d86c314b5a3dd044..a8edd5f7ad2182b59c807bdeb8b46801054d9963 100644
--- a/kdevplatform/project/abstractfilemanagerplugin.cpp
+++ b/kdevplatform/project/abstractfilemanagerplugin.cpp
@@ -41,6 +41,7 @@
 #include <serialization/indexedstring.h>
 
 #include "projectfiltermanager.h"
+#include "projectwatcher.h"
 #include "debug.h"
 
 #define ifDebug(x)
@@ -103,7 +104,7 @@ public:
 
     void removeFolder(ProjectFolderItem* folder);
 
-    QHash<IProject*, KDirWatch*> m_watchers;
+    QHash<IProject*, ProjectWatcher*> m_watchers;
     QHash<IProject*, QList<FileManagerListJob*> > m_projectJobs;
     QVector<QString> m_stoppedFolders;
     ProjectFilterManager m_filters;
@@ -271,7 +272,7 @@ void AbstractFileManagerPluginPrivate::created(const QString& path_)
     const IndexedString indexedPath(path.pathOrUrl());
     const IndexedString indexedParent(path.parent().pathOrUrl());
 
-    QHashIterator<IProject*, KDirWatch*> it(m_watchers);
+    QHashIterator<IProject*, ProjectWatcher*> it(m_watchers);
     while (it.hasNext()) {
         const auto p = it.next().key();
         if ( !p->projectItem()->model() ) {
@@ -334,7 +335,7 @@ void AbstractFileManagerPluginPrivate::deleted(const QString& path_)
     const Path path(QUrl::fromLocalFile(path_));
     const IndexedString indexed(path.pathOrUrl());
 
-    QHashIterator<IProject*, KDirWatch*> it(m_watchers);
+    QHashIterator<IProject*, ProjectWatcher*> it(m_watchers);
     while (it.hasNext()) {
         const auto p = it.next().key();
         if (path == p->path()) {
@@ -491,7 +492,7 @@ ProjectFolderItem *AbstractFileManagerPlugin::import( IProject *project )
 
     ///TODO: check if this works for remote files when something gets changed through another KDE app
     if ( project->path().isLocalFile() ) {
-        auto watcher = new KDirWatch( project );
+        auto watcher = new ProjectWatcher(project, &d->m_filters);
 
         // set up the signal handling
         connect(watcher, &KDirWatch::created,
diff --git a/kdevplatform/project/abstractfilemanagerplugin.h b/kdevplatform/project/abstractfilemanagerplugin.h
index c99e7e0c2316bf24df76d303fdb599a8095193da..0bb9e9a3c899505fd1e06235b31f619407a94b26 100644
--- a/kdevplatform/project/abstractfilemanagerplugin.h
+++ b/kdevplatform/project/abstractfilemanagerplugin.h
@@ -34,6 +34,7 @@ class KDirWatch;
 namespace KDevelop {
 
 class AbstractFileManagerPluginPrivate;
+class AbstractFileManagerPluginImportBenchmark;
 
 /**
  * This class can be used as a common base for file managers.
@@ -118,6 +119,8 @@ Q_SIGNALS:
 private:
     const QScopedPointer<class AbstractFileManagerPluginPrivate> d;
     friend class AbstractFileManagerPluginPrivate;
+public:
+    friend class AbstractFileManagerPluginImportBenchmark;
 };
 
 }
diff --git a/kdevplatform/project/projectwatcher.cpp b/kdevplatform/project/projectwatcher.cpp
new file mode 100644
index 0000000000000000000000000000000000000000..429cefb303db6f51d7afb03056e940e06e270ce5
--- /dev/null
+++ b/kdevplatform/project/projectwatcher.cpp
@@ -0,0 +1,57 @@
+/***************************************************************************
+ *   This file is part of KDevelop                                         *
+ *   Copyright 2017 René Bertin <rjvbertin@gmail.com>                      *
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU Library General Public License as       *
+ *   published by the Free Software Foundation; either version 2 of the    *
+ *   License, or (at your option) any later version.                       *
+ *                                                                         *
+ *   This program is distributed in the hope that it will be useful,       *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU General Public License for more details.                          *
+ *                                                                         *
+ *   You should have received a copy of the GNU Library General Public     *
+ *   License along with this program; if not, write to the                 *
+ *   Free Software Foundation, Inc.,                                       *
+ *   51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.         *
+ ***************************************************************************/
+
+#include "projectwatcher.h"
+#include "iproject.h"
+#include "projectfiltermanager.h"
+#include "path.h"
+
+#include <KDirWatch>
+
+using namespace KDevelop;
+
+KDevelop::ProjectWatcher::ProjectWatcher(IProject* project, ProjectFilterManager* filter)
+    : KDirWatch(project)
+    , m_project(project)
+    , m_filter(filter)
+    , m_watchedCount(0)
+{}
+
+void KDevelop::ProjectWatcher::addDir(const QString& path, WatchModes watchModes)
+{
+    if (/*m_filter->isValid(Path(path), true, m_project) &&*/ !contains(path)) {
+        KDirWatch::addDir(path, watchModes);
+        m_watchedCount += 1;
+    }
+}
+
+void KDevelop::ProjectWatcher::removeDir(const QString& path)
+{
+    if (contains(path)) {
+        KDirWatch::removeDir(path);
+        m_watchedCount -= 1;
+    }
+}
+
+int KDevelop::ProjectWatcher::size() const
+{
+    return m_watchedCount;
+}
+
diff --git a/kdevplatform/project/projectwatcher.h b/kdevplatform/project/projectwatcher.h
new file mode 100644
index 0000000000000000000000000000000000000000..74d4d4b9359d9dcad5bfb5080d0f4dece425a9ae
--- /dev/null
+++ b/kdevplatform/project/projectwatcher.h
@@ -0,0 +1,62 @@
+/***************************************************************************
+ *   This file is part of KDevelop                                         *
+ *   Copyright 2017 René Bertin <rjvbertin@gmail.com>                      *
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU Library General Public License as       *
+ *   published by the Free Software Foundation; either version 2 of the    *
+ *   License, or (at your option) any later version.                       *
+ *                                                                         *
+ *   This program is distributed in the hope that it will be useful,       *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU General Public License for more details.                          *
+ *                                                                         *
+ *   You should have received a copy of the GNU Library General Public     *
+ *   License along with this program; if not, write to the                 *
+ *   Free Software Foundation, Inc.,                                       *
+ *   51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.         *
+ ***************************************************************************/
+
+#ifndef KDEVPLATFORM_PROJECTWATCHER_H
+#define KDEVPLATFORM_PROJECTWATCHER_H
+
+#include "projectexport.h"
+
+#include <KDirWatch>
+
+namespace KDevelop {
+
+class IProject;
+class ProjectFilterManager;
+
+class KDEVPLATFORMPROJECT_EXPORT ProjectWatcher : public KDirWatch
+{
+    Q_OBJECT
+public:
+    /**
+     * Create a dirwatcher for @p project based on KDirWatch and
+     * enhanced with support for the project @p filter.
+     */
+    explicit ProjectWatcher(IProject* project, ProjectFilterManager* filter);
+
+    /**
+     * Add directory @p path to the project dirwatcher if it is not
+     * rejected by the project filter.
+     */
+    void addDir(const QString& path, WatchModes watchModes = WatchDirOnly);
+    void removeDir(const QString& path);
+
+    /**
+     * return the current number of directories being watched.
+     */
+    int size() const;
+
+private:
+    IProject* m_project;
+    ProjectFilterManager* m_filter;
+    int m_watchedCount;
+};
+
+}
+#endif //KDEVPLATFORM_PROJECTWATCHER_H
diff --git a/kdevplatform/project/tests/CMakeLists.txt b/kdevplatform/project/tests/CMakeLists.txt
index 74e5baada32fcc570f410e50c26bfc83a5066830..5a8d80cfe1a933bb839114cf7059152fc235adbd 100644
--- a/kdevplatform/project/tests/CMakeLists.txt
+++ b/kdevplatform/project/tests/CMakeLists.txt
@@ -21,3 +21,12 @@ target_link_libraries(abstractfilemanagerpluginimporttest
     KDev::Tests
     Qt5::QuickWidgets
 )
+
+add_executable(abstractfilemanagerpluginimportbenchmark
+    abstractfilemanagerpluginimportbenchmark.cpp
+)
+ecm_mark_nongui_executable(abstractfilemanagerpluginimportbenchmark)
+target_link_libraries(abstractfilemanagerpluginimportbenchmark
+    KDev::Project
+    KDev::Tests
+)
diff --git a/kdevplatform/project/tests/abstractfilemanagerpluginimportbenchmark.cpp b/kdevplatform/project/tests/abstractfilemanagerpluginimportbenchmark.cpp
new file mode 100644
index 0000000000000000000000000000000000000000..ec78f1805b2bb1ceb97661ef5ef6f2510caaf3ea
--- /dev/null
+++ b/kdevplatform/project/tests/abstractfilemanagerpluginimportbenchmark.cpp
@@ -0,0 +1,205 @@
+/* This file is part of KDevelop
+    Copyright 2017 René J.V. Bertin <rjvbertin@gmail.com>
+
+    This library is free software; you can redistribute it and/or
+    modify it under the terms of the GNU Library General Public
+    License as published by the Free Software Foundation; either
+    version 2 of the License, or (at your option) any later version.
+
+    This library is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+    Library General Public License for more details.
+
+    You should have received a copy of the GNU Library General Public License
+    along with this library; see the file COPYING.LIB.  If not, write to
+    the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+    Boston, MA 02110-1301, USA.
+*/
+
+#include <interfaces/iplugin.h>
+#include <interfaces/icore.h>
+#include <interfaces/iplugincontroller.h>
+
+#include <project/abstractfilemanagerplugin.h>
+#include <project/projectmodel.h>
+#include <project/projectwatcher.h>
+
+#include <shell/projectcontroller.h>
+
+#include <tests/autotestshell.h>
+#include <tests/testcore.h>
+#include <tests/testproject.h>
+#include <tests/testplugincontroller.h>
+
+#include <util/path.h>
+
+#include <KJob>
+#include <KDirWatch>
+
+#include <QApplication>
+#include <QList>
+#include <QFileInfo>
+#include <QElapsedTimer>
+#include <QMap>
+#include <QDebug>
+#include <QTextStream>
+
+using namespace KDevelop;
+
+namespace KDevelop {
+
+// wrap the ProjectController to make its addProject() method public
+class ProjectControllerWrapper : public ProjectController
+{
+    Q_OBJECT
+public:
+    ProjectControllerWrapper(Core* core)
+        : ProjectController(core)
+    {}
+
+    using ProjectController::addProject;
+};
+
+class AbstractFileManagerPluginImportBenchmark : public QObject
+{
+    Q_OBJECT
+public:
+    AbstractFileManagerPluginImportBenchmark(AbstractFileManagerPlugin* manager, const QString& path,
+                                             TestCore* core)
+        : QObject(core)
+        , m_out(stdout)
+        , m_core(core)
+    {
+        m_manager = manager;
+        m_project = new TestProject(Path(path));
+    }
+
+    void start()
+    {
+        m_projectNumber = s_numBenchmarksRunning++;
+        m_out << "Starting import of project " << m_project->path().toLocalFile() << endl;
+        ProjectControllerWrapper *projectController = qobject_cast<ProjectControllerWrapper*>(m_core->projectController());
+        projectController->addProject(m_project);
+        m_timer.start();
+        auto root = m_manager->import(m_project);
+        int elapsed = m_timer.elapsed();
+        m_out << "\tcreating dirwatcher took "
+            << elapsed / 1000.0 << " seconds" << endl;
+        auto import = m_manager->createImportJob(root);
+        connect(import, &KJob::finished,
+            this, &AbstractFileManagerPluginImportBenchmark::projectImportDone);
+        m_timer.restart();
+        import->start();
+    }
+
+    AbstractFileManagerPlugin* m_manager;
+    TestProject* m_project;
+    QElapsedTimer m_timer;
+    int m_projectNumber;
+    QTextStream m_out;
+    TestCore* m_core;
+
+    static int s_numBenchmarksRunning;
+
+Q_SIGNALS:
+    void finished();
+
+private Q_SLOTS:
+    void projectImportDone(KJob* job)
+    {
+        Q_UNUSED(job);
+        int elapsed = m_timer.elapsed();
+        // The number of items in the dirwatcher can be obtained as follows,
+        // provided dirs are added one by one and not with a recursive call
+        // to ProjectWatcher::addDir():
+        ProjectWatcher* watcher = qobject_cast<ProjectWatcher*>(m_manager->projectWatcher(m_project));
+        int watched = watcher ? watcher->size() : -1;
+        m_out << "importing " << m_project->fileSet().size()
+            << " items into project #" << m_projectNumber
+            << " with " << watched << " call(s) to KDirWatch::addDir()"
+            << " took " << elapsed / 1000.0 << " seconds" << endl;
+
+        s_numBenchmarksRunning -= 1;
+        if (s_numBenchmarksRunning <= 0) {
+            emit finished();
+        }
+    }
+
+};
+
+int AbstractFileManagerPluginImportBenchmark::s_numBenchmarksRunning = 0;
+}
+
+int main(int argc, char** argv)
+{
+    if (argc < 2) {
+        qWarning() << "Usage:" << argv[0] << "projectDir1 [...projectDirN]";
+        return 1;
+    }
+    QApplication app(argc, argv);
+    QTextStream qout(stdout);
+    // measure the total test time, this provides an indication
+    // of overhead and how well multiple projects are imported in parallel
+    // (= how different is the total time from the import time of the largest
+    // project). When testing a single project the difference between this
+    // value and total runtime will provide an estimate of the time required
+    // to destroy the dirwatcher.
+    QElapsedTimer runTimer;
+
+    AutoTestShell::init({"no plugins"});
+    auto core = TestCore::initialize();
+    // load/activate the "Project Filter" plugin (it won't be available to us without this step):
+    core->pluginController()->allPluginsForExtension(QStringLiteral("org.kdevelop.IProjectFilter"));
+    auto projectController = new ProjectControllerWrapper(core);
+    delete core->projectController();
+    core->setProjectController(projectController);
+    auto manager = new AbstractFileManagerPlugin({}, core);
+
+    const char *kdwMethod[] = {"FAM", "Inotify", "Stat", "QFSWatch"};
+    qout << "KDirWatch backend: " << kdwMethod[KDirWatch().internalMethod()] << endl;
+
+    QList<AbstractFileManagerPluginImportBenchmark*> benchmarks;
+
+    for (int i = 1 ; i < argc ; ++i) {
+        const QString path = QString::fromUtf8(argv[i]);
+        if (QFileInfo(path).isDir()) {
+            const auto benchmark = new AbstractFileManagerPluginImportBenchmark(manager, path, core);
+            benchmarks << benchmark;
+            QObject::connect(benchmark, &AbstractFileManagerPluginImportBenchmark::finished,
+                             &app, [&runTimer, &qout] {
+                                qout << "Done in " << runTimer.elapsed() / 1000.0
+                                    << " seconds total\n";
+                                QCoreApplication::instance()->quit();
+                             });
+            if (qEnvironmentVariableIsSet("BENCHMARK_ORIGINAL_DIRWATCHER")) {
+                // benchmark the creation and deletion of the original dirwatcher:
+                KDirWatch *watcher = new KDirWatch(benchmark->m_project);
+                qout << "Benchmarking KDirWatch for all of " << argv[i];
+                benchmark->m_timer.start();
+                watcher->addDir(benchmark->m_project->path().toLocalFile(), KDirWatch::WatchSubDirs | KDirWatch:: WatchFiles );
+                int elapsed = benchmark->m_timer.elapsed();
+                qout << "\tfeeding the watcher: " << elapsed / 1000.0 << " seconds\n";
+                benchmark->m_timer.restart();
+                delete watcher;
+                elapsed = benchmark->m_timer.elapsed();
+                qout << "\tdeleting the watcher: " << elapsed / 1000.0 << " seconds\n";
+                qout.flush();
+            }
+        }
+    }
+
+    if (benchmarks.isEmpty()) {
+        qWarning() << "no projects to import (arguments must be directories)";
+        return 1;
+    }
+
+    runTimer.start();
+    for (auto benchmark : benchmarks) {
+        benchmark->start();
+    }
+
+    return app.exec();
+}
+
+#include "abstractfilemanagerpluginimportbenchmark.moc"
