--- orig.QMobipocketConfig.cmake	2014-09-30 11:11:47.000000000 +0200
+++ QMobipocketConfig.cmake	2016-04-10 14:59:13.000000000 +0200
@@ -2,6 +2,7 @@
 get_filename_component( _currentDir  ${_currentDir} PATH)
 get_filename_component( _currentDir  ${_currentDir} PATH)
 get_filename_component( _currentDir  ${_currentDir} PATH)
+get_filename_component( _currentDir  ${_currentDir} PATH)
 
 # find the full paths to the library and the includes:
 find_path(QMOBIPOCKET_INCLUDE_DIR qmobipocket/mobipocket.h
