diff --git a/CMakeLists.txt b/CMakeLists.txt
index ea2e644..ca8ae43 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,26 +1,62 @@
 cmake_minimum_required(VERSION 2.8.9)
 project(mobipocket)
 
-find_package(KDE4 4.6.0 REQUIRED)
-include(KDE4Defaults)
-include(MacroLibrary)
-include(MacroOptionalAddSubdirectory)
+option(TARGET_QT5 "Build only libQMobiPocket against Qt5" FALSE)
+
+if(TARGET_QT5)
+    find_package(ECM 1.7.0 REQUIRED NO_MODULE)
+
+    set(CMAKE_MODULE_PATH ${ECM_MODULE_PATH} ${ECM_KDE_MODULE_DIR})
+
+    set(REQUIRED_QT_VERSION "5.2")
+    find_package(Qt5 ${REQUIRED_QT_VERSION} CONFIG REQUIRED Core Gui)
+    include(KDEInstallDirs)
+    include(KDEFrameworkCompilerSettings NO_POLICY_SCOPE)
+    include(KDECMakeSettings)
+
+    include(FeatureSummary)
+    add_definitions(-DTARGET_QT5)
+    find_package(Strigi)
+    set_package_properties(Strigi PROPERTIES DESCRIPTION "Provide a strigi analyser for Mobipocket documents"
+            URL "http://strigi.sourceforge.net"
+            TYPE OPTIONAL
+        )
+else()
+    find_package(KDE4 4.6.0 REQUIRED)
+    include(KDE4Defaults)
+    include(MacroLibrary)
+    include(MacroOptionalAddSubdirectory)
+
+    find_package(Strigi)
+    macro_log_feature(STRIGI_FOUND "strigi libraries" "Provide a strigi analyzer for Mobipocket documents" "" FALSE "" "")
+
+    add_definitions(${QT_DEFINITIONS} ${KDE4_DEFINITIONS})
+    include_directories(${KDE4_INCLUDES})
 
-find_package(Strigi)
-macro_log_feature(STRIGI_FOUND "strigi libraries" "Provide a strigi analyzer for Mobipocket documents" "" FALSE "" "")
+endif()
 
-add_definitions(${QT_DEFINITIONS} ${KDE4_DEFINITIONS})
-include_directories(${KDE4_INCLUDES})
 
 add_subdirectory(lib)
 
-macro_optional_add_subdirectory(thumbnailers)
-if(STRIGI_FOUND)
-  macro_optional_add_subdirectory(strigi)
+if(TARGET_QT5)
+    if(STRIGI_STREAMS_LIBRARY)
+        add_subdirectory(strigi)
+    endif()
+    install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/QMobipocket5Config.cmake
+        DESTINATION ${LIB_INSTALL_DIR}/cmake/QMobipocket5 )
+elseif()
+    macro_optional_add_subdirectory(thumbnailers)
+    if(STRIGI_FOUND)
+        macro_optional_add_subdirectory(strigi)
+    endif()
+    install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/QMobipocketConfig.cmake
+        DESTINATION ${LIB_INSTALL_DIR}/cmake/QMobipocket )
 endif()
 
-install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/QMobipocketConfig.cmake
-        DESTINATION ${LIB_INSTALL_DIR}/cmake/QMobipocket )
+if(TARGET_QT5)
+    feature_summary(WHAT ALL   FATAL_ON_MISSING_REQUIRED_PACKAGES)
+else()
+    macro_display_feature_log()
+endif()
 
-macro_display_feature_log()
 
diff --git a/QMobipocket5Config.cmake b/QMobipocket5Config.cmake
new file mode 100644
index 0000000..c48b387
--- /dev/null
+++ b/QMobipocket5Config.cmake
@@ -0,0 +1,20 @@
+get_filename_component( _currentDir  ${CMAKE_CURRENT_LIST_FILE} PATH)
+get_filename_component( _currentDir  ${_currentDir} PATH)
+get_filename_component( _currentDir  ${_currentDir} PATH)
+get_filename_component( _currentDir  ${_currentDir} PATH)
+
+# find the full paths to the library and the includes:
+find_path(QMOBIPOCKET5_INCLUDE_DIR qmobipocket/mobipocket.h
+          HINTS ${_currentDir}/include
+          NO_DEFAULT_PATH)
+
+find_library(QMOBIPOCKET5_LIBRARY qmobipocket5 
+             HINTS ${_currentDir}/lib
+             NO_DEFAULT_PATH)
+
+set(QMOBIPOCKET5_LIBRARIES ${QMOBIPOCKET5_LIBRARY})
+
+if(QMOBIPOCKET5_INCLUDE_DIR AND QMOBIPOCKET5_LIBRARY)
+    set(QMOBIPOCKET5_FOUND TRUE)
+endif()
+
diff --git a/QMobipocketConfig.cmake b/QMobipocketConfig.cmake
index d615afc..234d8db 100644
--- a/QMobipocketConfig.cmake
+++ b/QMobipocketConfig.cmake
@@ -8,7 +8,7 @@ find_path(QMOBIPOCKET_INCLUDE_DIR qmobipocket/mobipocket.h
           HINTS ${_currentDir}/include
           NO_DEFAULT_PATH)
 
-find_library(QMOBIPOCKET_LIBRARY qmobipocket 
+find_library(QMOBIPOCKET_LIBRARY qmobipocket5 
              HINTS ${_currentDir}/lib
              NO_DEFAULT_PATH)
 
diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
index 90fc229..e3ee397 100644
--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -5,19 +5,38 @@ set (QMOBIPOCKET_SRCS
     qfilestream.cpp
 )
 
-kde4_add_library(qmobipocket SHARED ${QMOBIPOCKET_SRCS})
+if(TARGET_QT5)
+    add_library(qmobipocket5 SHARED ${QMOBIPOCKET_SRCS})
 
-target_link_libraries (qmobipocket
-    ${QT_QTCORE_LIBRARY}
-    ${QT_QTGUI_LIBRARY}
-)
+    target_link_libraries (qmobipocket5
+        Qt5::Core
+        Qt5::Gui
+    )
 
-install(FILES
-    mobipocket.h
-    qmobipocket_export.h
-    qfilestream.h
-    DESTINATION ${INCLUDE_INSTALL_DIR}/qmobipocket/ COMPONENT Devel
-)
+    install(FILES
+        mobipocket.h
+        qmobipocket_export.h
+        qfilestream.h
+        DESTINATION ${INCLUDE_INSTALL_DIR}/qmobipocket/ COMPONENT Devel
+    )
+
+    install(TARGETS qmobipocket5 ${INSTALL_TARGETS_DEFAULT_ARGS})
+    set_target_properties(qmobipocket5 PROPERTIES VERSION 1.0.0 SOVERSION 1 EXPORT_NAME qmobipocket )
+else()
+    add_library(qmobipocket SHARED ${QMOBIPOCKET_SRCS})
+
+    target_link_libraries (qmobipocket
+        ${QT_QTCORE_LIBRARY}
+        ${QT_QTGUI_LIBRARY}
+    )
+
+    install(FILES
+        mobipocket.h
+        qmobipocket_export.h
+        qfilestream.h
+        DESTINATION ${INCLUDE_INSTALL_DIR}/qmobipocket/ COMPONENT Devel
+    )
 
-install(TARGETS qmobipocket ${INSTALL_TARGETS_DEFAULT_ARGS})
-set_target_properties(qmobipocket PROPERTIES VERSION 1.0.0 SOVERSION 1 )
+    install(TARGETS qmobipocket ${INSTALL_TARGETS_DEFAULT_ARGS})
+    set_target_properties(qmobipocket PROPERTIES VERSION 1.0.0 SOVERSION 1 )
+endif()
diff --git a/lib/mobipocket.cpp b/lib/mobipocket.cpp
index 5b731b4..602aefd 100644
--- a/lib/mobipocket.cpp
+++ b/lib/mobipocket.cpp
@@ -15,7 +15,11 @@
 #include <QtCore/QBuffer>
 #include <QtCore/QTextCodec>
 #include <QtGui/QImageReader>
+#ifdef TARGET_QT5
+#include <QtCore/QDebug>
+#else
 #include <kdebug.h>
+#endif
 
 namespace Mobipocket {
 
@@ -133,11 +137,19 @@ struct DocumentPrivate
 
 void DocumentPrivate::parseHtmlHead(const QString& data)
 {
+#ifdef TARGET_QT5
+    static QRegExp title(QStringLiteral("<dc:title.*>(.*)</dc:title>"), Qt::CaseInsensitive);
+    static QRegExp author(QStringLiteral("<dc:creator.*>(.*)</dc:creator>"), Qt::CaseInsensitive);
+    static QRegExp copyright(QStringLiteral("<dc:rights.*>(.*)</dc:rights>"), Qt::CaseInsensitive);
+    static QRegExp subject(QStringLiteral("<dc:subject.*>(.*)</dc:subject>"), Qt::CaseInsensitive);
+    static QRegExp description(QStringLiteral("<dc:description.*>(.*)</dc:description>"), Qt::CaseInsensitive);
+#else
     static QRegExp title("<dc:title.*>(.*)</dc:title>", Qt::CaseInsensitive);
     static QRegExp author("<dc:creator.*>(.*)</dc:creator>", Qt::CaseInsensitive);
     static QRegExp copyright("<dc:rights.*>(.*)</dc:rights>", Qt::CaseInsensitive);
     static QRegExp subject("<dc:subject.*>(.*)</dc:subject>", Qt::CaseInsensitive);
     static QRegExp description("<dc:description.*>(.*)</dc:description>", Qt::CaseInsensitive);
+#endif
     title.setMinimal(true);
     author.setMinimal(true);
     copyright.setMinimal(true);
diff --git a/lib/qmobipocket_export.h b/lib/qmobipocket_export.h
index 923134a..5d2e46b 100644
--- a/lib/qmobipocket_export.h
+++ b/lib/qmobipocket_export.h
@@ -18,8 +18,14 @@
 #ifndef _QMOBIPOCKET_EXPORT_H
 #define _QMOBIPOCKET_EXPORT_H
 
+#ifdef TARGET_QT5
+#include <QtCore/QtGlobal>
+#define KDE_EXPORT  Q_DECL_EXPORT
+#define KDE_IMPORT  Q_DECL_IMPORT
+#else
 /* needed for KDE_EXPORT and KDE_IMPORT macros */
 #include <kdemacros.h>
+#endif
 
 #ifndef QMOBIPOCKET_EXPORT
 # if defined(KDELIBS_STATIC_LIBS)
diff --git a/strigi/CMakeLists.txt b/strigi/CMakeLists.txt
index 895284f..a02675f 100644
--- a/strigi/CMakeLists.txt
+++ b/strigi/CMakeLists.txt
@@ -9,13 +9,27 @@ set(mobianalyzer_SRCS
   mobiendanalyzer.cpp
 )
 
-kde4_add_library(mobi MODULE ${mobianalyzer_SRCS})
+if(TARGET_QT5)
+    add_library(mobi5 MODULE ${mobianalyzer_SRCS})
 
-target_link_libraries(mobi ${STRIGI_STREAMS_LIBRARY} ${STRIGI_STREAMANALYZER_LIBRARY} ${QT_QTCORE_LIBRARY} ${QT_QTGUI_LIBRARY} qmobipocket)
+    target_link_libraries(mobi5 ${STRIGI_STREAMS_LIBRARY} ${STRIGI_STREAMANALYZER_LIBRARY}
+        Qt5::Core Qt5::Gui qmobipocket5)
 
-# install the module in the right directory so it is picked up
-set_target_properties(mobi PROPERTIES PREFIX ${STRIGI_LINE_ANALYZER_PREFIX})
+    # install the module in the right directory so it is picked up
+    set_target_properties(mobi5 PROPERTIES PREFIX strigila_)
 
-########### install files ###############
+    ########### install files ###############
 
-install(TARGETS mobi LIBRARY DESTINATION ${LIB_INSTALL_DIR}/strigi)
+    install(TARGETS mobi5 LIBRARY DESTINATION ${LIB_INSTALL_DIR}/strigi)
+else()
+    kde4_add_library(mobi MODULE ${mobianalyzer_SRCS})
+
+    target_link_libraries(mobi ${STRIGI_STREAMS_LIBRARY} ${STRIGI_STREAMANALYZER_LIBRARY} ${QT_QTCORE_LIBRARY} ${QT_QTGUI_LIBRARY} qmobipocket)
+
+    # install the module in the right directory so it is picked up
+    set_target_properties(mobi PROPERTIES PREFIX ${STRIGI_LINE_ANALYZER_PREFIX})
+
+    ########### install files ###############
+
+    install(TARGETS mobi LIBRARY DESTINATION ${LIB_INSTALL_DIR}/strigi)
+endif()
