diff --git a/kdevplatform/shell/runtimecontroller.cpp b/kdevplatform/shell/runtimecontroller.cpp
index f840240614..929530f2a9 100644
--- a/kdevplatform/shell/runtimecontroller.cpp
+++ b/kdevplatform/shell/runtimecontroller.cpp
@@ -38,15 +38,19 @@ public:
 
     void startProcess(KProcess *process) const override {
         connect(process, &QProcess::errorOccurred,
-            this, [](QProcess::ProcessError error) {
-            qCWarning(SHELL) << "process finished with error:" << error;
+            this, [process](QProcess::ProcessError error) {
+            qCWarning(SHELL) << "process `"
+                << process->program().join(QLatin1Char(' '))
+                << "` finished with error:" << error;
         });
         process->start();
     }
     void startProcess(QProcess *process) const override {
         connect(process, &QProcess::errorOccurred,
-            this, [](QProcess::ProcessError error) {
-            qCWarning(SHELL) << "process finished with error:" << error;
+            this, [process](QProcess::ProcessError error) {
+            qCWarning(SHELL) << "process `"
+                << process->program() << process->arguments().join(QLatin1Char(' '))
+                << "` finished with error:" << error;
         });
         process->start();
     }
