diff --git CMakeLists.txt CMakeLists.txt
index f318fd2e7c..14122efe95 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -1,5 +1,9 @@
 cmake_minimum_required(VERSION 3.0)
-project(KDevelop VERSION 5.6.2)
+if (APPLE)
+	project(KDevelop VERSION 5.6.2 LANGUAGES C CXX OBJCXX)
+else()
+	project(KDevelop VERSION 5.6.2)
+endif()
 
 # KDevelop SOVERSION
 # E.g. for KDevelop 5.2.0 => SOVERSION 52 (we only promise ABI compatibility between patch version updates)
diff --git plugins/welcomepage/CMakeLists.txt plugins/welcomepage/CMakeLists.txt
index 9849403094..4a12ef0724 100644
--- plugins/welcomepage/CMakeLists.txt
+++ plugins/welcomepage/CMakeLists.txt
@@ -29,7 +29,7 @@ target_link_libraries(kdevwelcomepage
     KF5::Declarative
 )
 # see https://bugs.launchpad.net/ubuntu/+source/gcc-5/+bug/1568899
-if (UNIX AND CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+if (UNIX AND NOT APPLE AND CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
     message(WARNING "Applying workaround for https://bugs.launchpad.net/ubuntu/+source/gcc-5/+bug/1568899")
     target_link_libraries(kdevwelcomepage gcc_s gcc)
 endif()
