diff --git kdevplatform/documentation/documentationview.cpp kdevplatform/documentation/documentationview.cpp
index d1ce72222508d6ecd99bf695353fb8512c0714e0..2b9a61bde789e7feed5a415003f4ee3598248190 100644
--- kdevplatform/documentation/documentationview.cpp
+++ kdevplatform/documentation/documentationview.cpp
@@ -19,6 +19,7 @@
 
 #include "documentationview.h"
 
+#include <QEvent>
 #include <QWidgetAction>
 #include <QAction>
 #include <QIcon>
@@ -37,6 +38,7 @@
 #include <interfaces/idocumentationproviderprovider.h>
 #include <interfaces/idocumentationcontroller.h>
 #include <interfaces/iplugincontroller.h>
+#include <sublime/idealdockwidget.h>
 #include "documentationfindwidget.h"
 #include "standarddocumentationview.h"
 #include "debug.h"
@@ -64,11 +72,44 @@ DocumentationView::DocumentationView(QWidget* parent, ProvidersModel* model)
 
     mCurrent = mHistory.end();
 
+    if (ICore::self()->shuttingDown()) {
+        KMessageBox::ButtonCode ret = KMessageBox::warningYesNo(this,
+                i18n("A documentation toolview (DocumentationView) is being created during shutdown.\n"
+                    "Do you want to report this (will cause a crash)?"));
+        if (ret == KMessageBox::Yes) {
+            qFatal("DocumentationView created during shutdown");
+        } else {
+            return;
+        }
+    }
+
     setFocusProxy(mIdentifiers);
 
     QMetaObject::invokeMethod(this, "initialize", Qt::QueuedConnection);
+
+    floatStandaloneWindows();
+}
+
+bool DocumentationView::event(QEvent* e)
+{
+    if (e->type() == QEvent::ParentChange) {
+        // we'll have to make the new IdealDockWidget parent
+        // behave the way we'd like it to behave.
+        floatStandaloneWindows();
+    }
+    return QWidget::event(e);
 }
 
+void DocumentationView::floatStandaloneWindows()
+{
+    Sublime::IdealDockWidget* dockWidget = dynamic_cast<Sublime::IdealDockWidget*>(parent());
+    if (dockWidget) {
+        dockWidget->setFloating(false);
+        dockWidget->setFloatsAsStandalone(true);
+    }
+}
+
+
 QList<QAction*> DocumentationView::contextMenuActions() const
 {
     // TODO: also show providers
diff --git kdevplatform/documentation/documentationview.h kdevplatform/documentation/documentationview.h
index 53e8ace5df8598e5ce17b14bad006eb9242ea1e6..4b63816ccbf86653eb15dde2b897cc05c0a44f0c 100644
--- kdevplatform/documentation/documentationview.h
+++ kdevplatform/documentation/documentationview.h
@@ -58,11 +58,15 @@ public Q_SLOTS:
     void changedProvider(int);
     void showHome();
 
+protected:
+    bool event(QEvent* e) override;
+
 private:
     void mousePressEvent(QMouseEvent* event) override;
     void setupActions();
     void updateView();
     void returnPressed();
+    void floatStandaloneWindows();
 
     QAction* mForward;
     QAction* mBack;
