diff --git kdevplatform/vcs/vcspluginhelper.cpp kdevplatform/vcs/vcspluginhelper.cpp
index 1df425220d..08e383aa8e 100644
--- kdevplatform/vcs/vcspluginhelper.cpp
+++ kdevplatform/vcs/vcspluginhelper.cpp
@@ -487,6 +487,7 @@ void VcsPluginHelper::annotationContextMenuAboutToShow( KTextEditor::View* view,
     connect(historyAction, &QAction::triggered, this, [this, rev]() {
         history(rev);
     });
+    menu->setTearOffEnabled(true);
 }
 
 void VcsPluginHelper::handleAnnotationBorderVisibilityChanged(View* view, bool visible)
diff --git plugins/filemanager/filemanager.cpp plugins/filemanager/filemanager.cpp
index f566586b19..19c75e8255 100644
--- plugins/filemanager/filemanager.cpp
+++ plugins/filemanager/filemanager.cpp
@@ -115,6 +115,7 @@ void FileManager::fillContextMenu(const KFileItem& item, QMenu* menu)
     KDevelop::ContextMenuExtension::populateMenu(tmpMenu, extensions);
     contextActions.append(tmpMenu->actions());
     delete tmpMenu;
+    menu->setTearOffEnabled(true);
 }
 
 void FileManager::openFile(const KFileItem& file)
diff --git plugins/projectmanagerview/projectbuildsetwidget.cpp plugins/projectmanagerview/projectbuildsetwidget.cpp
index 4849e69f9f..1a04dc30b6 100644
--- plugins/projectmanagerview/projectbuildsetwidget.cpp
+++ plugins/projectmanagerview/projectbuildsetwidget.cpp
@@ -134,6 +134,7 @@ void ProjectBuildSetWidget::showContextMenu( const QPoint& p )
 
 
     QMenu m(this);
+    m.setTearOffEnabled(true);
     m.setTitle(i18nc("@title:menu", "Build Set"));
     m.addAction(QIcon::fromTheme(QStringLiteral("list-remove")), i18nc("@action:inmenu", "Remove from Build Set"), this, SLOT(removeItems()) );
 
