diff --git plugins/lldb/formatters/qt.py plugins/lldb/formatters/qt.py
index ece10a1476..764627a5a3 100644
--- plugins/lldb/formatters/qt.py
+++ plugins/lldb/formatters/qt.py
@@ -23,7 +23,10 @@ from __future__ import print_function
 import time
 import datetime as dt
 import string
-from urlparse import urlsplit, urlunsplit
+try:
+    from urlparse import urlsplit, urlunsplit
+except:
+    from urllib.parse import urlsplit, urlunsplit
 
 import locale
 import lldb
