diff --git plugins/cmake/cmakemanager.cpp plugins/cmake/cmakemanager.cpp
index 9836dc6d8d98ccc02fed03f30df6db925e5c8114..b40786518d92171fe6b2d4e9d78efe8d773326be 100644
--- plugins/cmake/cmakemanager.cpp
+++ plugins/cmake/cmakemanager.cpp
@@ -158,6 +158,7 @@ public:
             addSubjob(importJob);
             ExecuteCompositeJob::start();
         } else {
+            qCWarning(CMAKE) << CMake::currentCMakeExecutable(project) << "doesn't support cmake-file-api mode, trying server mode to import" << project->name();
             tryCMakeServer();
         }
     }
@@ -178,7 +179,7 @@ private:
         Q_ASSERT(code > 0);
         Q_ASSERT(!server->isServerAvailable());
 
-        qCDebug(CMAKE) << "CMake does not provide server mode, using compile_commands.json to import" << project->name();
+        qCWarning(CMAKE) << CMake::currentCMakeExecutable(project) << "doesn't provide server mode, using compile_commands.json to import" << project->name();
 
         // parse the JSON file
         auto* job = new CMakeImportJsonJob(project, this);
