diff --git a/cmake/modules/FindClang.cmake b/cmake/modules/FindClang.cmake
index 2f2764cfa2..e0254119a1 100644
--- a/cmake/modules/FindClang.cmake
+++ b/cmake/modules/FindClang.cmake
@@ -101,6 +101,8 @@ if(CLANG_FOUND)
   string(REPLACE "svn" "" CLANG_VERSION_CLEAN "${CLANG_VERSION}")
   # dito for git
   string(REPLACE "git" "" CLANG_VERSION_CLEAN "${CLANG_VERSION}")
+  string(REPLACE "." ";" CLANG_VERSION_LIST "${CLANG_VERSION_CLEAN}")
+  list(GET CLANG_VERSION_LIST 0 CLANG_BRANCH)
 
   find_path(CLANG_BUILTIN_DIR
             # cpuid.h because it is defined in ClangSupport constructor as valid clang builtin dir indicator
@@ -111,10 +113,17 @@ if(CLANG_FOUND)
                           "../../../clang/${CLANG_VERSION}/include"
                           "clang/${CLANG_VERSION_CLEAN}/include"
                           "../../../clang/${CLANG_VERSION_CLEAN}/include"
+                          "clang/${CLANG_BRANCH}/include"
             NO_DEFAULT_PATH
   )
 
   if (NOT CLANG_BUILTIN_DIR)
+	 message(WARNING "After looking in \"${CLANG_LIBRARY_DIRS}\" and \"${CLANG_INCLUDE_DIRS}\" with suffixes\n"
+                          "clang/${CLANG_VERSION}/include ;"
+                          "../../../clang/${CLANG_VERSION}/include ;"
+                          "clang/${CLANG_VERSION_CLEAN}/include ;"
+                          "../../../clang/${CLANG_VERSION_CLEAN}/include ;"
+                          "clang/${CLANG_BRANCH}/include")
       message(FATAL_ERROR "Could not find Clang builtin directory")
   endif()
   get_filename_component(CLANG_BUILTIN_DIR ${CLANG_BUILTIN_DIR} ABSOLUTE)
@@ -124,22 +133,25 @@ if(CLANG_FOUND)
     COMMAND ${LLVM_CONFIG_EXECUTABLE} --src-root
     OUTPUT_VARIABLE _llvmSourceRoot
     OUTPUT_STRIP_TRAILING_WHITESPACE
+    ERROR_QUIET
   )
-  string(FIND "${LLVM_INCLUDE_DIRS}" "${_llvmSourceRoot}" _llvmIsInstalled)
-  if (NOT _llvmIsInstalled)
-    message(STATUS "Detected that llvm-config comes from a build-tree, adding more include directories for Clang")
-    list(APPEND CLANG_INCLUDE_DIRS
-         "${LLVM_INSTALL_PREFIX}/tools/clang/include" # build dir
-    )
-
-    # check whether the source is from llvm-project.git (currently recommended way to clone the LLVM projects)
-    # contains all LLVM projects in the top-level directory
-    get_filename_component(_llvmProjectClangIncludeDir ${_llvmSourceRoot}/../clang/include REALPATH)
-    if (EXISTS ${_llvmProjectClangIncludeDir})
-        message(STATUS "  Note: llvm-project.git structure detected, using different include path pointing into source dir")
-        list(APPEND CLANG_INCLUDE_DIRS "${_llvmProjectClangIncludeDir}") # source dir
-    else()
-        list(APPEND CLANG_INCLUDE_DIRS "${_llvmSourceRoot}/tools/clang/include") # source dir
+  if (_llvmSourceRoot)
+    string(FIND "${LLVM_INCLUDE_DIRS}" "${_llvmSourceRoot}" _llvmIsInstalled)
+    if (NOT _llvmIsInstalled)
+      message(STATUS "Detected that llvm-config comes from a build-tree, adding more include directories for Clang")
+      list(APPEND CLANG_INCLUDE_DIRS
+           "${LLVM_INSTALL_PREFIX}/tools/clang/include" # build dir
+      )
+
+      # check whether the source is from llvm-project.git (currently recommended way to clone the LLVM projects)
+      # contains all LLVM projects in the top-level directory
+      get_filename_component(_llvmProjectClangIncludeDir ${_llvmSourceRoot}/../clang/include REALPATH)
+      if (EXISTS ${_llvmProjectClangIncludeDir})
+          message(STATUS "  Note: llvm-project.git structure detected, using different include path pointing into source dir")
+          list(APPEND CLANG_INCLUDE_DIRS "${_llvmProjectClangIncludeDir}") # source dir
+      else()
+          list(APPEND CLANG_INCLUDE_DIRS "${_llvmSourceRoot}/tools/clang/include") # source dir
+      endif()
     endif()
   endif()
 
diff --git a/cmake/modules/FindLLVM.cmake b/cmake/modules/FindLLVM.cmake
index 3799046069..4aadae15d3 100644
--- a/cmake/modules/FindLLVM.cmake
+++ b/cmake/modules/FindLLVM.cmake
@@ -132,16 +132,16 @@ if (LLVM_FOUND)
     set(LLVM_SHARED_MODE OFF)
   endif()
 
-  # potentially add include dir from binary dir for non-installed LLVM
-  execute_process(
-    COMMAND ${LLVM_CONFIG_EXECUTABLE} --src-root
-    OUTPUT_VARIABLE _llvmSourceRoot
-    OUTPUT_STRIP_TRAILING_WHITESPACE
-  )
-  string(FIND "${LLVM_INCLUDE_DIRS}" "${_llvmSourceRoot}" _llvmIsInstalled)
-  if (NOT _llvmIsInstalled)
-    list(APPEND LLVM_INCLUDE_DIRS "${LLVM_INSTALL_PREFIX}/include")
-  endif()
+#   # potentially add include dir from binary dir for non-installed LLVM
+#   execute_process(
+#     COMMAND ${LLVM_CONFIG_EXECUTABLE} --src-root
+#     OUTPUT_VARIABLE _llvmSourceRoot
+#     OUTPUT_STRIP_TRAILING_WHITESPACE
+#   )
+#   string(FIND "${LLVM_INCLUDE_DIRS}" "${_llvmSourceRoot}" _llvmIsInstalled)
+#   if (NOT _llvmIsInstalled)
+#     list(APPEND LLVM_INCLUDE_DIRS "${LLVM_INSTALL_PREFIX}/include")
+#   endif()
 endif()
 
 if (LLVM_FIND_REQUIRED AND NOT LLVM_FOUND)
