diff --git cmake/modules/FindClangTidy.cmake cmake/modules/FindClangTidy.cmake
index e2ed5fd65d..6abfa87999 100644
--- cmake/modules/FindClangTidy.cmake
+++ cmake/modules/FindClangTidy.cmake
@@ -14,7 +14,16 @@
 # See the License for more information.
 #=============================================================================
 
-find_program(ClangTidy_EXECUTABLE NAMES clang-tidy)
+find_program(ClangTidy_EXECUTABLE NAMES clang-tidy
+	clang-tidy-17
+	clang-tidy-16
+	clang-tidy-15
+	clang-tidy-14
+	clang-tidy-13
+	clang-tidy-mp-12
+	clang-tidy-mp-9.0
+	clang-tidy-mp-8.0
+	clang-tidy-mp-5.0)
 
 include(FindPackageHandleStandardArgs)
 find_package_handle_standard_args(ClangTidy DEFAULT_MSG ClangTidy_EXECUTABLE)
