diff --git a/vcs/widgets/vcsdiffpatchsources.cpp b/vcs/widgets/vcsdiffpatchsources.cpp
index 42cf1d8..cc4ba6a 100644
--- a/vcs/widgets/vcsdiffpatchsources.cpp
+++ b/vcs/widgets/vcsdiffpatchsources.cpp
@@ -20,6 +20,7 @@
 
 #include <QHBoxLayout>
 #include <QVBoxLayout>
+#include <QSizePolicy>
 #include <QDir>
 #include <QFontDatabase>
 #include <QLabel>
@@ -44,6 +45,25 @@
 
 using namespace KDevelop;
 
+class VCSCommitMessageEditor : public KTextEdit {
+    Q_OBJECT
+public:
+    VCSCommitMessageEditor()
+        : m_minWidth(KTextEdit::minimumSizeHint().width())
+    {}
+    void setMinWidth(int w)
+    {
+        m_minWidth = w;
+        setSizePolicy(QSizePolicy::Preferred, QSizePolicy::Preferred);
+    }
+    QSize minimumSizeHint() const override
+    {
+        return QSize(m_minWidth, KTextEdit::minimumSizeHint().height());
+    }
+protected:
+    int m_minWidth;
+};
+
 VCSCommitDiffPatchSource::VCSCommitDiffPatchSource(VCSDiffUpdater* updater)
     : VCSDiffPatchSource(updater), m_vcs(updater->vcs())
 {
@@ -51,15 +71,23 @@ VCSCommitDiffPatchSource::VCSCommitDiffPatchSource(VCSDiffUpdater* updater)
     m_commitMessageWidget = new QWidget;
     QVBoxLayout* layout = new QVBoxLayout(m_commitMessageWidget.data());
 
-    m_commitMessageEdit = new KTextEdit;
-    m_commitMessageEdit.data()->setFont(QFontDatabase::systemFont(QFontDatabase::FixedFont));
-    m_commitMessageEdit.data()->setLineWrapMode(QTextEdit::NoWrap);
-    m_vcs->setupCommitMessageEditor(updater->url(), m_commitMessageEdit.data());
+    VCSCommitMessageEditor *editor = new VCSCommitMessageEditor;
+    m_commitMessageEdit = editor;
+    editor->setFont(QFontDatabase::systemFont(QFontDatabase::FixedFont));
+    editor->setLineWrapMode(QTextEdit::NoWrap);
+    // set the message editor to be 72 characters wide.
+    // Given the widget margins that requires 74 actual characters.
+    editor->setMinWidth(editor->fontMetrics().width(QString(74, 'm')));
+    m_vcs->setupCommitMessageEditor(updater->url(), editor);
 
     QHBoxLayout* titleLayout = new QHBoxLayout;
-    titleLayout->addWidget(new QLabel(i18n("Commit Message:")));
+    QLabel *label = new QLabel(i18n("Commit Message:"));
+    // KTextEdit toolTips aren't always reliable, so if one was added to the message editor
+    // during set-up, we copy to the label's tooltip.
+    label->setToolTip(editor->toolTip());
+    titleLayout->addWidget(label);
 
-    m_oldMessages = new KComboBox(m_commitMessageWidget.data());
+    m_oldMessages = new KComboBox(editor);
 
     m_oldMessages->addItem(i18n("Old Messages"));
     foreach(const QString& message, oldMessages())
@@ -71,7 +99,7 @@ VCSCommitDiffPatchSource::VCSCommitDiffPatchSource(VCSDiffUpdater* updater)
     titleLayout->addWidget(m_oldMessages);
 
     layout->addLayout(titleLayout);
-    layout->addWidget(m_commitMessageEdit.data());
+    layout->addWidget(editor);
     connect(this, &VCSCommitDiffPatchSource::reviewCancelled, this, &VCSCommitDiffPatchSource::addMessageToHistory);
     connect(this, &VCSCommitDiffPatchSource::reviewFinished, this, &VCSCommitDiffPatchSource::addMessageToHistory);
 }
@@ -159,6 +187,7 @@ VCSDiffPatchSource::~VCSDiffPatchSource()
 {
     QFile::remove(m_file.toLocalFile());
     delete m_updater;
+    qDebug() << "~VCSDiffPatchSource()" << this;
 }
 
 QUrl VCSDiffPatchSource::baseDir() const {
@@ -208,11 +237,16 @@ void VCSDiffPatchSource::updateFromDiff(VcsDiff vcsdiff)
 void VCSDiffPatchSource::update() {
     if(!m_updater)
         return;
+    m_updater->setContextLines(m_contextLines == 0? INT_MAX : m_contextLines);
     updateFromDiff(m_updater->update());
 }
 
 VCSCommitDiffPatchSource::~VCSCommitDiffPatchSource() {
     delete m_commitMessageWidget.data();
+//     if (m_commitMessageWidget.data()) {
+//         m_commitMessageWidget.data()->deleteLater();
+//     }
+    qDebug() << "~VCSCommitDiffPatchSource()" << this;
 }
 
 bool VCSCommitDiffPatchSource::canSelectFiles() const {
@@ -296,6 +330,7 @@ bool showVcsDiff(IPatchSource* vcsDiff)
 
 VcsDiff VCSStandardDiffUpdater::update() const
 {
+    m_vcs->setDiffContextLines(m_contextLines);
     QScopedPointer<VcsJob> diffJob(m_vcs->diff(m_url,
                                    KDevelop::VcsRevision::createSpecialRevision(KDevelop::VcsRevision::Base),
                                    KDevelop::VcsRevision::createSpecialRevision(KDevelop::VcsRevision::Working)));
@@ -317,3 +352,4 @@ VCSStandardDiffUpdater::~VCSStandardDiffUpdater() {
 VCSDiffUpdater::~VCSDiffUpdater() {
 }
 
+#include "vcsdiffpatchsources.moc"
diff --git a/plugins/git/gitmessagehighlighter.cpp b/plugins/git/gitmessagehighlighter.cpp
index da7660d..93eda58 100644
--- a/plugins/git/gitmessagehighlighter.cpp
+++ b/plugins/git/gitmessagehighlighter.cpp
@@ -35,15 +35,27 @@ static const int lineLenLimit = 72;
 void GitMessageHighlighter::applyErrorFormat(GitMessageHighlighter* text, bool warning, const QString& tooltip, int startPos, int endPos)
 {
     QTextCharFormat format;
-    format.setFontUnderline(true);
-    format.setUnderlineStyle(QTextCharFormat::SpellCheckUnderline);
-    format.setUnderlineColor(warning ? Qt::yellow : Qt::red);
+    format.setFontLetterSpacing(125);
+    if (!warning) {
+        format.setFontItalic(true);
+    } else {
+        // make the 1st character stand out too
+        // (letterspacing is applied *after* the startPos character)
+        if (startPos > 0) {
+            startPos -= 1;
+        }
+    }
     format.setToolTip(tooltip);
     text->setFormat(startPos, endPos, format);
 }
 
 GitMessageHighlighter::GitMessageHighlighter(QTextEdit* parent): Sonnet::Highlighter(parent)
 {
+    if (parent) {
+        parent->setToolTip(i18n("Suggested commit format: a summary line of about %1 characters maximum,\n"
+            "an empty line, and optional details with a max. line length of %2 characters.",
+            summarySoftLimit, lineLenLimit));
+    }
 }
 
 GitMessageHighlighter::~GitMessageHighlighter()
@@ -77,7 +89,7 @@ void GitMessageHighlighter::highlightBlock(const QString& text)
                     applyErrorFormat(this,
                         lineLength <= summaryHardLimit,
                         i18n("Try to keep summary length below %1 characters.", summarySoftLimit),
-                        startPos, endPos
+                        startPos+summarySoftLimit, endPos
                     );
                 } else {
                     for(int i=startPos; i<endPos; i++) {
