diff --git a/core/smb4kmounter.cpp b/core/smb4kmounter.cpp
index 67ee6ae..3bfa48f 100644
--- a/core/smb4kmounter.cpp
+++ b/core/smb4kmounter.cpp
@@ -1766,7 +1766,7 @@ bool Smb4KMounter::fillMountActionArgs(Smb4KShare *, QVariantMap&)
 {
   qWarning() << "Smb4KMounter::fillMountActionArgs() is not implemented!";
   qWarning() << "Mounting under this operating system is not supported...";
-  return false
+  return false;
 }
 #endif
 
