diff --git a/core/smb4kscanner_p.cpp b/core/smb4kscanner_p.cpp
index 031c4a4..5e7d589 100644
--- a/core/smb4kscanner_p.cpp
+++ b/core/smb4kscanner_p.cpp
@@ -224,6 +224,12 @@ void Smb4KLookupDomainsJob::startProcess1()
 
 void Smb4KLookupDomainsJob::startProcess2(const QStringList& ipAddresses)
 {
+  if (ipAddresses.isEmpty()) {
+    emitResult();
+    emit finished();
+    return;
+  }
+
   //
   // Find shell commands
   //

