diff --git a/smb4k/smb4kmainwindow.cpp b/smb4k/smb4kmainwindow.cpp
index b2b42ce3abaf7ec184d53a6065dedc5f2f5e41d8..4771077ca9104188fb18fe68f50fdbd753ea66f7 100644
--- a/smb4k/smb4kmainwindow.cpp
+++ b/smb4k/smb4kmainwindow.cpp
@@ -335,7 +335,9 @@ void Smb4KMainWindow::setupMenuBar()
     if (QString::compare("bookmarks", action->objectName()) == 0)
     {
       Smb4KBookmarkMenu *menu = static_cast<Smb4KBookmarkMenu *>(actionCollection()->action("bookmarks_menu"));
-      action->setMenu(menu->menu());
+      if (menu) {
+        action->setMenu(menu->menu());
+      }
       break;
     }
     else
@@ -569,7 +571,10 @@ void Smb4KMainWindow::setupDynamicActionList(QDockWidget* dock)
     // disable the bookmark action.
     //
     disconnect(this, SLOT(slotEnableBookmarkAction()));
-    actionCollection()->action("bookmark_action")->setEnabled(false);
+    QAction *bookmark_action2 = actionCollection()->action("bookmark_action");
+    if (bookmark_action2) {
+      bookmark_action2->setEnabled(false);
+    }
 
     // 
     // Get also the bookmark menu and disable the bookmark action
@@ -611,7 +616,11 @@ void Smb4KMainWindow::setupDynamicActionList(QDockWidget* dock)
     
     for (QAction *action : dockActionCollection->actions())
     {
-      if (action->objectName() == "bookmark_action")
+      if (!action)
+      {
+        continue;
+      }
+      else if (action->objectName() == "bookmark_action")
       {
         if (bookmarkMenu)
         {
@@ -1240,8 +1249,11 @@ void Smb4KMainWindow::slotEnableBookmarkAction()
         if (action)
         {
           // Bookmark action of the main window
-          actionCollection()->action("bookmark_action")->setEnabled(action->isEnabled());
-          
+          QAction *bookmark_action2 = actionCollection()->action("bookmark_action");
+          if (bookmark_action2) {
+            bookmark_action2->setEnabled(action->isEnabled());
+          }
+
           // Bookmark action of the bookmark menu
           Smb4KBookmarkMenu *bookmarkMenu = findChild<Smb4KBookmarkMenu *>();
           
@@ -1275,7 +1287,10 @@ void Smb4KMainWindow::slotEnableBookmarkAction()
         if (action)
         {
           // Bookmark action of the main window
-          actionCollection()->action("bookmark_action")->setEnabled(action->isEnabled());
+          QAction *bookmark_action2 = actionCollection()->action("bookmark_action");
+          if (bookmark_action2) {
+            bookmark_action2->setEnabled(action->isEnabled());
+          }
           
           // Bookmark action of the bookmark menu
           Smb4KBookmarkMenu *bookmarkMenu = findChild<Smb4KBookmarkMenu *>();
@@ -1310,8 +1325,11 @@ void Smb4KMainWindow::slotEnableBookmarkAction()
         if (action)
         {
           // Bookmark action of the main window
-          actionCollection()->action("bookmark_action")->setEnabled(action->isEnabled());
-          
+          QAction *bookmark_action2 = actionCollection()->action("bookmark_action");
+          if (bookmark_action2) {
+            bookmark_action2->setEnabled(action->isEnabled());
+          }
+
           // Bookmark action of the bookmark menu
           Smb4KBookmarkMenu *bookmarkMenu = findChild<Smb4KBookmarkMenu *>();
           
