diff --git a/CMakeLists.txt b/CMakeLists.txt
index dad230c..0969fcf 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -17,6 +17,7 @@ include(KDECompilerSettings)
 include(KDECMakeSettings)
 include(FeatureSummary)
 include(ECMInstallIcons)
+include(ECMAddAppIcon)
 
 # Qt5 modules
 find_package(Qt5Core 5.4.0 REQUIRED)
diff --git a/smb4k/CMakeLists.txt b/smb4k/CMakeLists.txt
index 0e833d2..14daa33 100644
--- a/smb4k/CMakeLists.txt
+++ b/smb4k/CMakeLists.txt
@@ -26,6 +26,10 @@ set(smb4k_SRCS
   smb4kmainwindow.cpp
   smb4ksystemtray.cpp)
 
+# Sets the icon on Windows and OSX
+file(GLOB ICONS_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/icons/oxygen/*-apps-smb4k.png")
+ecm_add_app_icon(smb4k_SRCS ICONS ${ICONS_SRCS})
+
 add_executable(smb4k ${smb4k_SRCS})
 
 target_link_libraries(smb4k
@@ -38,7 +42,7 @@ target_link_libraries(smb4k
   KF5::ConfigWidgets
   smb4kcore)
 
-install(TARGETS smb4k DESTINATION ${KDE_INSTALL_BINDIR})
+install(TARGETS smb4k DESTINATION ${KDE_INSTALL_TARGETS_DEFAULT_ARGS})
 
 ########### install files ###############
 
