diff --git a/CMakeLists.txt b/CMakeLists.txt
index dad230c..0969fcf 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -17,6 +17,7 @@ include(KDECompilerSettings)
 include(KDECMakeSettings)
 include(FeatureSummary)
 include(ECMInstallIcons)
+include(ECMAddAppIcon)
 
 # Qt5 modules
 find_package(Qt5Core 5.4.0 REQUIRED)
diff --git a/smb4k/CMakeLists.txt b/smb4k/CMakeLists.txt
index 2735210..aa401aa 100644
--- a/smb4k/CMakeLists.txt
+++ b/smb4k/CMakeLists.txt
@@ -119,6 +119,10 @@ set(smb4k_SRCS
   smb4kmainwindow.cpp
   smb4ksystemtray.cpp)
 
+# Sets the icon on Windows and OSX
+file(GLOB ICONS_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/icons/oxygen/*-apps-smb4k.png")
+ecm_add_app_icon(smb4k_SRCS ICONS ${ICONS_SRCS})
+
 add_executable(smb4k ${smb4k_SRCS})
 
 target_link_libraries(smb4k
@@ -137,7 +141,7 @@ install(TARGETS smb4knetworkbrowser DESTINATION ${KDE_INSTALL_PLUGINDIR})
 install(TARGETS smb4knetworksearch DESTINATION ${KDE_INSTALL_PLUGINDIR})
 install(TARGETS smb4ksharesview DESTINATION ${KDE_INSTALL_PLUGINDIR})
 install(TARGETS smb4kconfigdialog DESTINATION ${KDE_INSTALL_PLUGINDIR})
-install(TARGETS smb4k DESTINATION ${KDE_INSTALL_BINDIR})
+install(TARGETS smb4k DESTINATION ${KDE_INSTALL_TARGETS_DEFAULT_ARGS})
 
 install(PROGRAMS org.kde.smb4k.desktop DESTINATION ${XDG_APPS_INSTALL_DIR})
 
