diff --git a/smb4k/smb4kmainwindow.cpp b/smb4k/smb4kmainwindow.cpp
index 603506f..4f42d2a 100644
--- a/smb4k/smb4kmainwindow.cpp
+++ b/smb4k/smb4kmainwindow.cpp
@@ -1097,10 +1097,15 @@ void Smb4KMainWindow::slotActivePartChanged(KParts::Part *part)
   for (int i = 0; i < m_active_part->actionCollection()->actions().size(); ++i)
   {
     QAction *action = m_active_part->actionCollection()->action(i);
+    if (!action) {
+      continue;
+    }
 
     if (QString::compare(action->objectName(), "bookmark_action") == 0)
     {
-      actionCollection()->action("bookmark_action")->setEnabled(action->isEnabled());
+      if (actionCollection()->action("bookmark_action")) {
+        actionCollection()->action("bookmark_action")->setEnabled(action->isEnabled());
+      }
       connect(action, SIGNAL(changed()), this, SLOT(slotEnableBookmarkAction()));
       continue;
     }

