diff --git a/CMakeLists.txt b/CMakeLists.txt
index dad230c..8232495 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -48,6 +49,10 @@ find_package(KF5Parts 5.9.0 REQUIRED)
 find_package(KF5ConfigWidgets 5.9.0 REQUIRED)
 find_package(KF5Notifications 5.9.0 REQUIRED)
 
+if(APPLE)
+    add_definitions(-DQ_OS_FREEBSD)
+endif()
+
 # Make sure that Smb4K builds when several custom targets
 # with the same name exist (happens in the po directory).
 if (EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/po")
diff --git a/core/CMakeLists.txt b/core/CMakeLists.txt
index 2c24452..70e685b 100644
--- a/core/CMakeLists.txt
+++ b/core/CMakeLists.txt
@@ -51,6 +51,8 @@ if (${CMAKE_HOST_SYSTEM_NAME} MATCHES "Linux")
   kconfig_add_kcfg_files(smb4kcore_LIB_SRCS smb4ksettings.kcfgc smb4kmountsettings_linux.kcfgc)
 elseif(${CMAKE_HOST_SYSTEM_NAME} MATCHES "FreeBSD")
   kconfig_add_kcfg_files(smb4kcore_LIB_SRCS smb4ksettings.kcfgc smb4kmountsettings_freebsd.kcfgc)
+elseif(APPLE)
+  kconfig_add_kcfg_files(smb4kcore_LIB_SRCS smb4ksettings.kcfgc smb4kmountsettings_freebsd.kcfgc)
 elseif(${CMAKE_HOST_SYSTEM_NAME} MATCHES "NetBSD")
   kconfig_add_kcfg_files(smb4kcore_LIB_SRCS smb4ksettings.kcfgc smb4kmountsettings_freebsd.kcfgc)
 else ()
