diff --git b/AquaProxy/Modern_Root_Certificates.pkg/orig.postinstall a/AquaProxy/Modern_Root_Certificates.pkg/postinstall
index d3b84ef..6ffd151 100755
--- b/AquaProxy/Modern_Root_Certificates.pkg/orig.postinstall
+++ a/AquaProxy/Modern_Root_Certificates.pkg/postinstall
@@ -1,17 +1,25 @@
 #!/bin/bash
 
 CERTIFICATES=(
-	"./ISRG Root X1.cer"
-	"./USERTrust RSA Certification Authority.crt"
-	"./DigiCert Global Root G2.cer"
-	"./DigiCert Global Root G3.cer"
-	"./COMODO ECC Certification Authority.crt"
+	"@PREFIX@/etc/openssl/ISRG Root X1.cer"
+	"@PREFIX@/etc/openssl/USERTrust RSA Certification Authority.crt"
+	"@PREFIX@/etc/openssl/DigiCert Global Root G2.cer"
+	"@PREFIX@/etc/openssl/DigiCert Global Root G3.cer"
+	"@PREFIX@/etc/openssl/COMODO ECC Certification Authority.crt"
 )
 
 KEYCHAINS=$(security list-keychains | tr -d '"' | tr -d ' ')
 
 for cert in "${CERTIFICATES[@]}"
 do
+	if [ "${DESTROOT}" != "" ] ;then
+		cert="${DESTROOT}/${cert}"
+	fi
+	if [ ! -r "${cert}" ] ;then
+		echo "Fatal error: certificate \"${cert}\" does not exist"
+		exit 1
+	fi
+
 	cert_name="$(basename "$cert" | sed 's/\.[^.]*$//')"
 	
 	cert_already_installed=false
@@ -26,7 +34,10 @@ do
 	
 	if [ "$cert_already_installed" = false ]
 	then
+		echo "Installing certicate \"${cert}\""
 		security -v add-trusted-cert -d -r trustRoot -k /Library/Keychains/System.keychain "$cert"
+	else
+		echo "Skipping already installed certicate \"${cert}\""
 	fi
 done
 
@@ -37,6 +48,5 @@ then
 		pid=$(echo $pid_uid | cut -d, -f1)
 		uid=$(echo $pid_uid | cut -d, -f2)
 		launchctl bsexec "$pid" chroot -u "$uid" / launchctl stop Wowfunhappy.AquaProxy.HTTP || true
-		launchctl bsexec "$pid" chroot -u "$uid" / launchctl start Wowfunhappy.AquaProxy.HTTP
 	done
-fi
\ No newline at end of file
+fi
diff --git b/AquaProxy/Aqua_Proxy.pkg/Scripts/orig.postinstall a/AquaProxy/Aqua_Proxy.pkg/Scripts/postinstall
index a82a780..134f4e0 100755
--- b/AquaProxy/Aqua_Proxy.pkg/Scripts/orig.postinstall
+++ a/AquaProxy/Aqua_Proxy.pkg/Scripts/postinstall
@@ -1,11 +1,17 @@
 #!/bin/bash
 
+if [ "${1}" = "--create-certificates" ] ;then
 # Clean up any old "Aqua Proxy" certificates from previous installations.
-sudo security delete-certificate -c "Aqua Proxy" /Library/Keychains/System.keychain >/dev/null 2>&1
+	set -e
+
+	cd ${DESTROOT}@PREFIX@/Library/AquaProxy
+	openssl req -x509 -newkey rsa:4096 -subj '/CN=Aqua Proxy' -nodes -days 999999 -keyout AquaProxy-key.pem -out AquaProxy-cert.pem
 
-cd /Library/AquaProxy
-openssl req -x509 -newkey rsa:4096 -subj '/CN=Aqua Proxy' -nodes -days 999999 -keyout AquaProxy-key.pem -out AquaProxy-cert.pem
-security -v add-trusted-cert -d -r trustRoot -k /Library/Keychains/System.keychain /Library/AquaProxy/AquaProxy-cert.pem
+	exit 0
+fi
+
+sudo security delete-certificate -c "Aqua Proxy" /Library/Keychains/System.keychain >/dev/null 2>&1
+security -v add-trusted-cert -d -r trustRoot -k /Library/Keychains/System.keychain @PREFIX@/Library/AquaProxy/AquaProxy-cert.pem
 
 sleep 10 # Make sure certificates are installed. Particularly necessary on Snow Leopard.
 
@@ -15,15 +21,15 @@ for pid_uid in $(ps -axo pid,uid,args | grep -i "[l]oginwindow.app" | awk '{prin
 	if (( $(echo "${OSTYPE:6} > 13" | bc -l) ))
 	then
 		#Running OS X 10.10 or above
-		launchctl bootstrap gui/$uid /Library/LaunchAgents/Wowfunhappy.AquaProxy.HTTP.plist
-		launchctl bootstrap gui/$uid /Library/LaunchAgents/Wowfunhappy.AquaProxy.IMAP.plist
-		launchctl bootstrap gui/$uid /Library/LaunchAgents/Wowfunhappy.AquaProxy.SyncProxiesWithShell.plist
-		launchctl bootstrap gui/$uid /Library/LaunchAgents/Wowfunhappy.AquaProxy.Restarter.plist
+		launchctl bootstrap gui/$uid @PREFIX@/Library/LaunchAgents/Wowfunhappy.AquaProxy.HTTP.plist
+		launchctl bootstrap gui/$uid @PREFIX@/Library/LaunchAgents/Wowfunhappy.AquaProxy.IMAP.plist
+		launchctl bootstrap gui/$uid @PREFIX@/Library/LaunchAgents/Wowfunhappy.AquaProxy.SyncProxiesWithShell.plist
+		launchctl bootstrap gui/$uid @PREFIX@/Library/LaunchAgents/Wowfunhappy.AquaProxy.Restarter.plist
 	else
 		#Running OS X 10.9 or below
-		launchctl bsexec "$pid" chroot -u "$uid" / launchctl load /Library/LaunchAgents/Wowfunhappy.AquaProxy.HTTP.plist
-		launchctl bsexec "$pid" chroot -u "$uid" / launchctl load /Library/LaunchAgents/Wowfunhappy.AquaProxy.IMAP.plist
-		launchctl bsexec "$pid" chroot -u "$uid" / launchctl load /Library/LaunchAgents/Wowfunhappy.AquaProxy.SyncProxiesWithShell.plist
-		launchctl bsexec "$pid" chroot -u "$uid" / launchctl load /Library/LaunchAgents/Wowfunhappy.AquaProxy.Restarter.plist
+		launchctl bsexec "$pid" chroot -u "$uid" / launchctl load @PREFIX@/Library/LaunchAgents/Wowfunhappy.AquaProxy.HTTP.plist
+		launchctl bsexec "$pid" chroot -u "$uid" / launchctl load @PREFIX@/Library/LaunchAgents/Wowfunhappy.AquaProxy.IMAP.plist
+		launchctl bsexec "$pid" chroot -u "$uid" / launchctl load @PREFIX@/Library/LaunchAgents/Wowfunhappy.AquaProxy.SyncProxiesWithShell.plist
+		launchctl bsexec "$pid" chroot -u "$uid" / launchctl load @PREFIX@/Library/LaunchAgents/Wowfunhappy.AquaProxy.Restarter.plist
 	fi
-done
\ No newline at end of file
+done
