commit 4edfeaf75db07b006330c464202a7b85d3c0b6dd
Author: René Bertin <rjvbertin@gmail.com>
Date:   Thu May 30 01:06:43 2024 +0200

    patch-headerpaths,pad.diff

diff --git a/src/port1.0/portconfigure.tcl b/src/port1.0/portconfigure.tcl
index 846f83ab7..d1e20b513 100644
--- a/src/port1.0/portconfigure.tcl
+++ b/src/port1.0/portconfigure.tcl
@@ -266,7 +266,7 @@ proc portconfigure::configure_get_cppflags {} {
     if {${compiler.limit_flags}} {
         return {}
     } else {
-        return -I${prefix}/include
+        return -isystem${prefix}/include
     }
 }
 default configure.ldflags       {[portconfigure::configure_get_ldflags]}
@@ -275,7 +275,11 @@ proc portconfigure::configure_get_ldflags {} {
     if {${compiler.limit_flags}} {
         return -Wl,-headerpad_max_install_names
     } else {
-        return "-L${prefix}/lib -Wl,-headerpad_max_install_names"
+        if {[option os.platform] eq "darwin"} {
+            return "-L${prefix}/lib -Wl,-headerpad_max_install_names"
+        } else {
+            return "-L${prefix}/lib -Wl,--enable-new-dtags -Wl,-rpath,${prefix}/lib"
+        }
     }
 }
 default configure.libs          {}
