commit 4c6c675c39ef1762e06064a790269e495a34353c
Author: René Bertin <rjvbertin@gmail.com>
Date:   Wed May 29 22:04:52 2024 +0200

    patch-silent-command-exec.diff

diff --git a/src/port1.0/portutil.tcl b/src/port1.0/portutil.tcl
index ff4f41ee0..da93be12d 100644
--- a/src/port1.0/portutil.tcl
+++ b/src/port1.0/portutil.tcl
@@ -464,7 +464,7 @@ proc command_exec {args} {
     array set env [array get ${varprefix}.env_array]
     # Call the command.
     set fullcmdstring "$command_prefix $cmdstring $command_suffix"
-    ui_info "Executing: $fullcmdstring"
+    ui_debug "Executing: $fullcmdstring"
     set code [catch {system {*}$notty {*}$callback {*}$nice $fullcmdstring} result]
     # Save variables in order to re-throw the same error code.
     set errcode $::errorCode
