commit e096dc2832a435f75232d12d6e28a892d1036def
Author: René Bertin <rjvbertin@gmail.com>
Date:   Wed May 29 22:24:30 2024 +0200

    patch-ui_info-in-info.diff

diff --git a/src/macports1.0/macports.tcl b/src/macports1.0/macports.tcl
index c2bbbdad8..11fe656fa 100644
--- a/src/macports1.0/macports.tcl
+++ b/src/macports1.0/macports.tcl
@@ -397,6 +397,13 @@ proc ui_message {priority prefix args} {
        }
     } 
 
+    # adopt msg priority if the current phase equals the current priority
+    # this can only happen for ui_info during `port info` (currently).
+    if {[info exists current_phase]
+        && ${current_phase} eq ${priority}} {
+        set priority "msg"
+    }
+
     foreach chan $channels($priority) {
         if {[lindex $args 0] eq "-nonewline"} {
             puts -nonewline $chan $prefix[lindex $args 1]
diff --git a/src/port/port.tcl b/src/port/port.tcl
index fcaa582d6..273e4f030 100755
--- a/src/port/port.tcl
+++ b/src/port/port.tcl
@@ -1677,6 +1677,7 @@ proc action_info { action portlist opts } {
         return 1
     }
 
+    set_phase "info"
     set separator ""
     global global_variations global_options
     set gvariations [dict create {*}[array get global_variations]]
