commit f9d70bb6e096ed52d8eba378b3b5c5851b641acb
Author: René Bertin <rjvbertin@gmail.com>
Date:   Thu May 30 01:09:51 2024 +0200

    patch-nowarn-unsigned-source.diff

diff --git a/src/macports1.0/macports.tcl b/src/macports1.0/macports.tcl
index 937d86e02..04272b517 100644
--- a/src/macports1.0/macports.tcl
+++ b/src/macports1.0/macports.tcl
@@ -1246,10 +1246,10 @@ proc mportinit {{up_ui_options {}} {up_options {}} {up_variations {}}} {
                         set sources_default [concat [list $url] $flags]
                     }
                 }
-                if {[string match rsync://*rsync.macports.org/release/ports/ $url]} {
-                    ui_warn "MacPorts is configured to use an unsigned source for the ports tree.\
-Please edit sources.conf and change '$url' to '[string range $url 0 end-6]tarballs/ports.tar'."
-                }
+#                 if {[string match rsync://*rsync.macports.org/release/ports/ $url]} {
+#                     ui_warn "MacPorts is configured to use an unsigned source for the ports tree.\
+# Please edit sources.conf and change '$url' to '[string range $url 0 end-6]tarballs/ports.tar'."
+#                 }
                 switch -- [macports::getprotocol $url] {
                     rsync -
                     https -
