commit 7bad56cd8fff2415fb121cfd7c9a50369a11c507
Author: René Bertin <rjvbertin@gmail.com>
Date:   Tue May 28 19:38:45 2024 +0200

    patch-port-environment-option.diff

diff --git a/src/port/port.tcl b/src/port/port.tcl
index eb1afc7f6..458adbca9 100755
--- a/src/port/port.tcl
+++ b/src/port/port.tcl
@@ -2581,6 +2581,12 @@ proc action_version { action portlist opts } {
 }
 
 
+proc action_environment { action portlist opts } {
+    set status [macports::environment]
+    return $status
+}
+
+
 proc action_platform { action portlist opts } {
     global macports::os_platform macports::os_major macports::os_arch
     if {![macports::ui_isset ports_quiet]} {
@@ -4009,6 +4015,7 @@ set action_array [dict create \
     diagnose    [list action_diagnose       [ACTION_ARGS_NONE]] \
     \
     version     [list action_version        [ACTION_ARGS_NONE]] \
+    environment [list action_environment    [ACTION_ARGS_NONE]] \
     platform    [list action_platform       [ACTION_ARGS_NONE]] \
     \
     uninstall   [list action_uninstall      [ACTION_ARGS_PORTS]] \
@@ -4266,6 +4273,9 @@ proc parse_options { action ui_options_name global_options_name } {
             set opts [string range $arg 1 end]
             foreach c [split $opts {}] {
                 switch -- $c {
+                    e {
+                        set ui_options(ports_env) yes
+                    }
                     v {
                         set ui_options(ports_verbose) yes
                     }
