commit a730d172d357b14da2621b38f3b3cdb531593934
Author: René Bertin <rjvbertin@gmail.com>
Date:   Thu May 30 13:26:27 2024 +0200

    patch-macports-stdlibc-only-on-darwin.diff

diff --git a/src/port1.0/portconfigure.tcl b/src/port1.0/portconfigure.tcl
index c2071f2f8..2b80b896e 100644
--- a/src/port1.0/portconfigure.tcl
+++ b/src/port1.0/portconfigure.tcl
@@ -144,10 +144,10 @@ proc portconfigure::stdlib_trace {opt action args} {
 }
 # helper function to set configure.cxx_stdlib
 proc portconfigure::configure_get_cxx_stdlib {} {
-    global cxx_stdlib compiler.cxx_standard
+    global cxx_stdlib compiler.cxx_standard os.platform
     if {${compiler.cxx_standard} eq ""} {
         return ""
-    } elseif {${cxx_stdlib} eq "libstdc++" && ${compiler.cxx_standard} >= 2011} {
+    } elseif {${os.platform} eq "darwin" && ${cxx_stdlib} eq "libstdc++" && ${compiler.cxx_standard} >= 2011} {
         return macports-libstdc++
     } else {
         return ${cxx_stdlib}
