commit a03e6dbcde3926e37ea0cc252f5f4e03905ea260
Author: René Bertin <rjvbertin@gmail.com>
Date:   Wed May 29 21:54:01 2024 +0200

    patch-no-update-nagging.diff

diff --git a/src/macports1.0/macports.tcl b/src/macports1.0/macports.tcl
index e339ca03c..bbdc1a71f 100644
--- a/src/macports1.0/macports.tcl
+++ b/src/macports1.0/macports.tcl
@@ -1851,9 +1851,9 @@ match macports.conf.default."
         set default_source_url [lindex $sources_default 0]
         if {[macports::getprotocol $default_source_url] eq "file" || [macports::getprotocol $default_source_url] eq "rsync"} {
             set default_portindex [macports::getindex $default_source_url]
-            if {[file exists $default_portindex] && [clock seconds] - [file mtime $default_portindex] > 1209600} {
-                ui_warn "port definitions are more than two weeks old, consider updating them by running 'port selfupdate'."
-            }
+#             if {[file exists $default_portindex] && [clock seconds] - [file mtime $default_portindex] > 1209600} {
+#                 ui_warn "port definitions are more than two weeks old, consider updating them by running 'port selfupdate'."
+#             }
         }
     }
 
