commit 531ad2bb9e441f5e672e06817b387046e467976c
Author: René Bertin <rjvbertin@gmail.com>
Date:   Wed May 29 13:31:54 2024 +0200

    patch-portprovides-support-symlinked-prefix.diff

diff --git a/src/port/port.tcl b/src/port/port.tcl
index 458adbca9..b5bb55f3d 100755
--- a/src/port/port.tcl
+++ b/src/port/port.tcl
@@ -2156,6 +2156,11 @@ proc action_notes { action portlist opts } {
 }
 
 
+proc macports::normalize { filename } {
+    set prefmap [list [file dirname [file normalize "${macports::prefix}/foo"]] ${macports::prefix}]
+    return [string map ${prefmap} [file normalize $filename]]
+}
+
 proc action_provides { action portlist opts } {
     # In this case, portname is going to be used for the filename... since
     # that is the first argument we expect... perhaps there is a better way
@@ -2165,7 +2170,7 @@ proc action_provides { action portlist opts } {
         return 1
     }
     foreach filename $portlist {
-        set file [file normalize $filename]
+        set file [macports::normalize $filename]
         if {[file exists $file] || ![catch {file type $file}]} {
             if {![file isdirectory $file] || [file type $file] eq "link"} {
                 set port [registry::file_registered $file]
