commit 971bae73c4771fc018efc33bd8f0714e67a30d73
Author: René Bertin <rjvbertin@gmail.com>
Date:   Wed May 29 22:04:51 2024 +0200

    patch-truncate-logfile.diff

diff --git a/src/macports1.0/macports.tcl b/src/macports1.0/macports.tcl
index bbdc1a71f..c2bbbdad8 100644
--- a/src/macports1.0/macports.tcl
+++ b/src/macports1.0/macports.tcl
@@ -283,8 +283,8 @@ proc macports::ch_logging {mport} {
     file mkdir $logdir
     set debuglogname [file join $logdir main.log]
 
-    # Append to the file if it already exists
-    set debuglog [open $debuglogname a]
+    # Truncate the file if it already exists
+    set debuglog [open $debuglogname w]
     puts $debuglog version:1
 
     ui_debug "Starting logging for $portname @[dict get $portinfo version]_[dict get $portinfo revision][dict get $portinfo canonical_active_variants]"
