commit 672302360e4a1b02a6e2cc7c95f0c48f3f6b9245
Author: René Bertin <rjvbertin@gmail.com>
Date:   Wed May 29 22:00:02 2024 +0200

    port-o-option-default.diff

diff --git a/src/port/port.tcl b/src/port/port.tcl
old mode 100644
new mode 100755
index ae04f161e..fcaa582d6
--- a/src/port/port.tcl
+++ b/src/port/port.tcl
@@ -4235,6 +4235,8 @@ proc parse_options { action ui_options_name global_options_name } {
 
     set options_order(${action}) {}
 
+    # RJVB
+    set global_options(ports_ignore_different) yes
     while {[moreargs]} {
         set arg [lookahead]
 
