diff --git qtbase/src/corelib/io/io.pri qtbase/src/corelib/io/io.pri
--- qtbase/src/corelib/io/io.pri.orig     2015-03-20 21:16:37.000000000 +0100
+++ qtbase/src/corelib/io/io.pri        2015-03-20 21:15:27.000000000 +0100
@@ -32,6 +32,7 @@
         io/qresource_iterator_p.h \
         io/qsavefile.h \
         io/qstandardpaths.h \
+        io/qstandardpaths_p.h \
         io/qurl.h \
         io/qurl_p.h \
         io/qurlquery.h \
diff --git qtbase/src/corelib/io/qstandardpaths.cpp qtbase/src/corelib/io/qstandardpaths.cpp
index b41aee5..4e3d3ce 100644
--- qtbase/src/corelib/io/qstandardpaths.cpp
+++ qtbase/src/corelib/io/qstandardpaths.cpp
@@ -351,6 +351,32 @@ static bool existsAsSpecified(const QString &path, QStandardPaths::LocateOptions
     return QFileInfo(path).isFile();
 }
 
+#if defined(Q_OS_OSX)
+/*!
+    This function enables or disables XDG locations on platforms where Qt can be configured
+    to use this kind of locations instead of locations following OS guidelines. This is a
+    feature that shouldn't be changed back and forth, so setXDGLocationsEnabled() has effect
+    only the first time it is called.
+    Currently this concerns only Mac OS X //and Cygwin?//
+ */
+void QStandardPaths::setXDGLocationsEnabled(bool xdgMode)
+{
+    static bool xdgSet = false;
+    if (!xdgSet) {
+        usingXDGLocations = xdgMode;
+        xdgSet = true;
+    }
+}
+
+/*!
+    This function returns true when XDG locations are used.
+ */
+bool QStandardPaths::isXDGLocationsEnabled()
+{
+    return usingXDGLocations;
+}
+#endif // Q_OS_OSX
+
 /*!
    Tries to find a file or directory called \a fileName in the standard locations
    for \a type.
diff --git qtbase/src/corelib/io/qstandardpaths.h qtbase/src/corelib/io/qstandardpaths.h
index e305450..0540aac 100644
--- qtbase/src/corelib/io/qstandardpaths.h
+++ qtbase/src/corelib/io/qstandardpaths.h
@@ -49,6 +49,8 @@ QT_BEGIN_NAMESPACE
 
 #ifndef QT_NO_STANDARDPATHS
 
+class Q_CORE_EXPORT QStandardPathsConfiguration;
+
 class Q_CORE_EXPORT QStandardPaths
 {
 public:
@@ -73,6 +73,10 @@ public:
         AppLocalDataLocation = DataLocation
     };
 
+#if defined(Q_OS_OSX)
+    static bool isXDGLocationsEnabled();
+#endif // Q_OS_OSX
+
     static QString writableLocation(StandardLocation type);
     static QStringList standardLocations(StandardLocation type);
 
@@ -96,10 +112,20 @@ public:
     static void setTestModeEnabled(bool testMode);
     static bool isTestModeEnabled();
 
+#if defined(Q_OS_OSX)
+protected:
+    static void setXDGLocationsEnabled(bool xdgMode);
+#endif // Q_OS_OSX
+
 private:
     // prevent construction
     QStandardPaths();
     ~QStandardPaths();
+#if defined(Q_OS_OSX)
+    static bool usingXDGLocations;
+
+    friend class QStandardPathsConfiguration;
+#endif // Q_OS_OSX
 };
 
 #endif // QT_NO_STANDARDPATHS
diff --git qtbase/src/corelib/io/qstandardpaths_p.h qtbase/src/corelib/io/qstandardpaths_p.h
new file mode 100644
index 0000000..18e7731
--- /dev/null
+++ qtbase/src/corelib/io/qstandardpaths_p.h
@@ -0,0 +1,62 @@
+/****************************************************************************
+**
+** Copyright (C) 2014 Digia Plc and/or its subsidiary(-ies).
+** Contact: http://www.qt-project.org/legal
+**
+** This file is part of the QtCore module of the Qt Toolkit.
+**
+** $QT_BEGIN_LICENSE:LGPL21$
+** Commercial License Usage
+** Licensees holding valid commercial Qt licenses may use this file in
+** accordance with the commercial license agreement provided with the
+** Software or, alternatively, in accordance with the terms contained in
+** a written agreement between you and Digia. For licensing terms and
+** conditions see http://qt.digia.com/licensing. For further information
+** use the contact form at http://qt.digia.com/contact-us.
+**
+** GNU Lesser General Public License Usage
+** Alternatively, this file may be used under the terms of the GNU Lesser
+** General Public License version 2.1 or version 3 as published by the Free
+** Software Foundation and appearing in the file LICENSE.LGPLv21 and
+** LICENSE.LGPLv3 included in the packaging of this file. Please review the
+** following information to ensure the GNU Lesser General Public License
+** requirements will be met: https://www.gnu.org/licenses/lgpl.html and
+** http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html.
+**
+** In addition, as a special exception, Digia gives you certain additional
+** rights. These rights are described in the Digia Qt LGPL Exception
+** version 1.1, included in the file LGPL_EXCEPTION.txt in this package.
+**
+** $QT_END_LICENSE$
+**
+****************************************************************************/
+
+#ifndef QSTANDARDPATHS_P_H
+#define QSTANDARDPATHS_P_H
+
+#include <QtCore/QtGlobal>
+
+#if defined(Q_OS_OSX)
+
+#include <QtCore/qstandardpaths.h>
+
+QT_BEGIN_NAMESPACE
+
+
+#ifndef QT_NO_STANDARDPATHS
+
+class Q_CORE_EXPORT QStandardPathsConfiguration
+{
+public:
+    QStandardPathsConfiguration(bool useXDGLocations)
+    {
+        QStandardPaths::setXDGLocationsEnabled(useXDGLocations);
+    }
+};
+
+#endif // QT_NO_STANDARDPATHS
+
+QT_END_NAMESPACE
+
+#endif // Q_OS_OSX
+#endif // QSTANDARDPATHS_P_H
diff --git qtbase/src/corelib/io/qstandardpaths_mac.cpp qtbase/src/corelib/io/qstandardpaths_mac.cpp
index aff9112..39d2f81 100644
--- qtbase/src/corelib/io/qstandardpaths_mac.cpp
+++ qtbase/src/corelib/io/qstandardpaths_mac.cpp
@@ -45,6 +45,7 @@
 
 #ifndef QT_BOOTSTRAPPED
 #include <qcoreapplication.h>
+#include <qlibraryinfo.h>
 #endif
 
 #include <CoreFoundation/CoreFoundation.h>
@@ -133,8 +134,109 @@ static QString macLocation(QStandardPaths::StandardLocation type, short domain)
     return path;
 }
 
+bool QStandardPaths::usingXDGLocations = false;
+
+static void normaliseDirs(QStringList &dirs)
+{
+    // Normalise paths, skip relative paths
+    QMutableListIterator<QString> it(dirs);
+    while (it.hasNext()) {
+        const QString dir = it.next();
+        if (!dir.startsWith(QLatin1Char('/')))
+            it.remove();
+        else
+            it.setValue(QDir::cleanPath(dir));
+    }
+
+    // Remove duplicates from the list, there's no use for duplicated
+    // paths in XDG_CONFIG_DIRS - if it's not found in the given
+    // directory the first time, it won't be there the second time.
+    // Plus duplicate paths causes problems for example for mimetypes,
+    // where duplicate paths here lead to duplicated mime types returned
+    // for a file, eg "text/plain,text/plain" instead of "text/plain"
+    dirs.removeDuplicates();
+}
+
+static QStringList xdgCacheDirs()
+{
+    QStringList dirs;
+    // http://standards.freedesktop.org/basedir-spec/latest/
+    QString xdgConfigDirsEnv = QFile::decodeName(qgetenv("XDG_CACHE_HOME"));
+    if (xdgConfigDirsEnv.isEmpty()) {
+#ifndef QT_BOOTSTRAPPED
+        dirs.append(QDir::homePath() + QString::fromLatin1("/.cache"));
+#endif
+    } else {
+        dirs = xdgConfigDirsEnv.split(QLatin1Char(':'), QString::SkipEmptyParts);
+
+        normaliseDirs(dirs);
+    }
+    return dirs;
+}
+
+static QStringList xdgConfigDirs()
+{
+    QStringList dirs;
+    // http://standards.freedesktop.org/basedir-spec/latest/
+    QString xdgConfigDirsEnv = QFile::decodeName(qgetenv("XDG_CONFIG_DIRS"));
+    if (xdgConfigDirsEnv.isEmpty()) {
+#ifndef QT_BOOTSTRAPPED
+        dirs.append(QLibraryInfo::location(QLibraryInfo::PrefixPath) + QString::fromLatin1("/etc/xdg"));
+#endif
+    } else {
+        dirs = xdgConfigDirsEnv.split(QLatin1Char(':'), QString::SkipEmptyParts);
+
+        normaliseDirs(dirs);
+    }
+    return dirs;
+}
+
+static QStringList xdgDataDirs()
+{
+    QStringList dirs;
+    // http://standards.freedesktop.org/basedir-spec/latest/
+    QString xdgDataDirsEnv = QFile::decodeName(qgetenv("XDG_DATA_DIRS"));
+    if (xdgDataDirsEnv.isEmpty()) {
+#ifndef QT_BOOTSTRAPPED
+        dirs.append(QLibraryInfo::location(QLibraryInfo::PrefixPath) + QString::fromLatin1("/share"));
+#endif
+    } else {
+        dirs = xdgDataDirsEnv.split(QLatin1Char(':'), QString::SkipEmptyParts);
+
+        normaliseDirs(dirs);
+    }
+    return dirs;
+}
+
 QString QStandardPaths::writableLocation(StandardLocation type)
 {
+    if (usingXDGLocations) {
+        const QString prefix = (isTestModeEnabled())? QDir::homePath() + QLatin1String("/.qttest") : QDir::homePath();
+        QString path;
+        switch (type) {
+        case GenericDataLocation:
+        case DataLocation:
+            path = prefix + QLatin1String("/.local/share");
+            if (type != GenericDataLocation)
+                appendOrganizationAndApp(path);
+            return path;
+        case GenericCacheLocation:
+        case CacheLocation:
+            path = prefix + QLatin1String("/.cache");
+            if (type == CacheLocation)
+                appendOrganizationAndApp(path);
+            return path;
+        case GenericConfigLocation:
+        case ConfigLocation:
+            return prefix + QLatin1String("/.config");
+        case ApplicationsLocation:
+            path = writableLocation(GenericDataLocation) + QLatin1String("/applications");
+            return path;
+        default:
+            break;
+        }
+    }
+
     if (isTestModeEnabled()) {
         const QString qttestDir = QDir::homePath() + QLatin1String("/.qttest");
         QString path;
@@ -154,6 +256,9 @@ QString QStandardPaths::writableLocation(StandardLocation type)
         case GenericConfigLocation:
         case ConfigLocation:
             return qttestDir + QLatin1String("/Preferences");
+        case ApplicationsLocation:
+            path = qttestDir + QLatin1String("/Applications");
+            return path;
         default:
             break;
         }
@@ -179,6 +284,28 @@ QStringList QStandardPaths::standardLocations(StandardLocation type)
 {
     QStringList dirs;
 
+    if (usingXDGLocations ) {
+        switch (type) {
+        case GenericDataLocation:
+            dirs.append(xdgDataDirs());
+            break;
+        case GenericConfigLocation:
+        case ConfigLocation:
+            dirs.append(xdgConfigDirs());
+            break;
+        case GenericCacheLocation:
+        case CacheLocation:
+            dirs.append(xdgCacheDirs());
+            break;
+        case ApplicationsLocation:
+            QStringList xdgDirs = xdgDataDirs();
+            for (int i = 0; i < xdgDirs.count(); ++i)
+                xdgDirs[i].append(QLatin1String("/applications"));
+            dirs.append(xdgDirs);
+            break;
+        }
+    }
+
     if (type == GenericDataLocation || type == DataLocation || type == GenericCacheLocation || type == CacheLocation) {
         const QString path = macLocation(type, kOnAppropriateDisk);
         if (!path.isEmpty())
@@ -186,6 +313,14 @@ QStringList QStandardPaths::standardLocations(StandardLocation type)
     }
 
     if (type == DataLocation) {
+        if (usingXDGLocations) {
+            QStringList xdgDirs = xdgDataDirs();
+            for (int i = 0; i < xdgDirs.count(); ++i) {
+                appendOrganizationAndApp(xdgDirs[i]);
+            }
+            dirs.append(xdgDirs);
+        }
+
         CFBundleRef mainBundle = CFBundleGetMainBundle();
         if (mainBundle) {
             CFURLRef bundleUrl = CFBundleCopyBundleURL(mainBundle);
