--- qtbase/orig.configure	2015-03-27 15:34:55.000000000 +0100
+++ qtbase/configure	2015-03-27 15:35:09.000000000 +0100
@@ -292,14 +292,14 @@
 
     echo "$1" | while read line; do
         case "$line" in
-            QMAKE_CC=*|QMAKE_CXX=*|QMAKE_FIX_RPATH=*|QMAKE_AR=*|QMAKE_RANLIB=*|QMAKE_LINK=*|QMAKE_LINK_SHLIB=*)
-                # Prefix tool with toolchain path
-                var=$(echo "$line" | cut -d '=' -f 1)
-                val=$(echo "$line" | cut -d '=' -f 2-)
-                sdk_val=$(/usr/bin/xcrun -sdk $sdk -find $(echo $val | cut -d ' ' -f 1))
-                val=$(echo $sdk_val $(echo $val | cut -s -d ' ' -f 2-))
-                echo "$var=$val"
-            ;;
+#            QMAKE_CC=*|QMAKE_CXX=*|QMAKE_FIX_RPATH=*|QMAKE_AR=*|QMAKE_RANLIB=*|QMAKE_LINK=*|QMAKE_LINK_SHLIB=*)
+#                # Prefix tool with toolchain path
+#                var=$(echo "$line" | cut -d '=' -f 1)
+#                val=$(echo "$line" | cut -d '=' -f 2-)
+#                sdk_val=$(/usr/bin/xcrun -sdk $sdk -find $(echo $val | cut -d ' ' -f 1))
+#                val=$(echo $sdk_val $(echo $val | cut -s -d ' ' -f 2-))
+#                echo "$var=$val"
+#            ;;
             QMAKE_CFLAGS=*|QMAKE_CXXFLAGS=*|QMAKE_OBJECTIVE_CFLAGS=*)
                 echo "$line -isysroot $sysroot $version_min_flag"
             ;;
