diff --git src/pextlib1.0/sip_copy_proc.c src/pextlib1.0/sip_copy_proc.c
index e191fcd..e79e2d4 100644
--- src/pextlib1.0/sip_copy_proc.c
+++ src/pextlib1.0/sip_copy_proc.c
@@ -271,6 +271,11 @@ static copy_needed_return_t copy_needed(const char *path, char *const argv[],
 }
 
 static char *lazy_copy(const char *path, struct stat *in_st) {
+#ifndef SF_RESTRICTED
+    (void) path;
+    (void) in_st;
+    return NULL;
+#else
     char *retval = NULL;
     uid_t euid = geteuid();
     int outfd = -1;
@@ -422,6 +427,7 @@ lazy_copy_out:
         free(target_path);
     }
     return retval;
+#endif // SF_RESTRICTED
 }
 
 /**

