diff --git src/port1.0/portconfigure.tcl src/port1.0/portconfigure.tcl
index a62a423634fa6690a68d3168b74a4a061a924182..d34d8f2dc53b4b0218453f8fc6f33f8d8bc23fa9 100644
--- src/port1.0/portconfigure.tcl
+++ src/port1.0/portconfigure.tcl
@@ -424,5 +426,5 @@ proc portconfigure::configure_get_universal_ldflags {} {
 # internal proc to determine if the compiler supports -arch
 proc portconfigure::arch_flag_supported {compiler} {
-    return [regexp {^gcc-4|llvm|apple|clang} $compiler]
+    return (${os.platform} eq "darwin") && [regexp {^gcc-4|llvm|apple|clang} $compiler]
 }
 
