diff --git src/port1.0/portconfigure.tcl src/port1.0/portconfigure.tcl
index a62a423634fa6690a68d3168b74a4a061a924182..ca00daa4af9db550228ab9f2084314fa643f9e95 100644
--- src/port1.0/portconfigure.tcl
+++ src/port1.0/portconfigure.tcl
@@ -424,5 +426,10 @@ proc portconfigure::configure_get_universal_ldflags {} {
 # internal proc to determine if the compiler supports -arch
 proc portconfigure::arch_flag_supported {compiler} {
-    return [regexp {^gcc-4|llvm|apple|clang} $compiler]
+    global os.platform
+    if {${os.platform} eq "darwin"} {
+        return [regexp {^gcc-4|llvm|apple|clang} $compiler]
+    } else {
+        return 0
+    }
 }
 
