diff --git src/pextlib1.0/xinstall.c src/pextlib1.0/xinstall.c
index f28ec50e..d12b531d 100644
--- src/pextlib1.0/xinstall.c
+++ src/pextlib1.0/xinstall.c
@@ -109,6 +109,8 @@ extern int copyfile(const char *from, const char *to, void *state,
 
 #ifndef HAVE_SETMODE
 #include "setmode.h"
+#else
+#include <bsd/unistd.h>
 #endif
 
 /* Bootstrap aid - this doesn't exist in most older releases */

diff --git a/src/pextlib1.0/Pextlib.c b/src/pextlib1.0/Pextlib.c
index 03e6732a..b9042aeb 100644
--- a/src/pextlib1.0/Pextlib.c
+++ b/src/pextlib1.0/Pextlib.c
@@ -105,6 +107,8 @@ extern char **environ;
 
 #ifndef HAVE_SETMODE
 #include "setmode.h"
+#else
+#include <bsd/unistd.h>
 #endif
 
 __attribute__((format(printf, 3, 0)))
