diff --git modules/access/avcapture.m modules/access/avcapture.m
index 38777226cfe56211aa0bd1efc9623ac7ca4cf650..d31f2168885b9b2d68d7dc562ea7a4849edbfb6e 100644
--- modules/access/avcapture.m
+++ modules/access/avcapture.m
@@ -43,14 +43,6 @@
 #import <AVFoundation/AVFoundation.h>
 #import <CoreMedia/CoreMedia.h>
 
-#ifndef MAC_OS_X_VERSION_10_14
-@interface AVCaptureDevice (AVCaptureDeviceAuthorizationSince10_14)
-
-+ (void)requestAccessForMediaType:(AVMediaType)mediaType completionHandler:(void (^)(BOOL granted))handler API_AVAILABLE(macos(10.14), ios(7.0));
-
-@end
-#endif
-
 /*****************************************************************************
 * Local prototypes
 *****************************************************************************/
@@ -334,6 +326,11 @@ static int Open(vlc_object_t *p_this)
             goto error;
         }
 
+// only compile this block if the requestAccessForMediaType method is supported;
+// the code inside `if (@available(foo)) {}` is only *executed* when foo is true but
+// it is always parsed at compile time; @available is not an #ifdef that protects against
+// undefined types or functions.
+#ifdef MAC_OS_X_VERSION_10_14
         if (@available(macOS 10.14, *)) {
             msg_Dbg(p_demux, "Check user consent for access to the video device");
 
@@ -350,6 +347,7 @@ static int Open(vlc_object_t *p_this)
                 goto error;
             }
         }
+#endif
 
         input = [AVCaptureDeviceInput deviceInputWithDevice:(__bridge AVCaptureDevice *)p_sys->device error:&o_returnedError];
 
diff --git modules/access/avaudiocapture.m modules/access/avaudiocapture.m
index e20d8aadf382ee493f2cdfe0cf91a7d9bf951b51..4d6587314c390055836bfe5e9f375aaa08fd3874 100644
--- modules/access/avaudiocapture.m
+++ modules/access/avaudiocapture.m
@@ -45,14 +45,6 @@
 #import <CoreMedia/CoreMedia.h>
 
 
-#ifndef MAC_OS_X_VERSION_10_14
-@interface AVCaptureDevice (AVCaptureDeviceAuthorizationSince10_14)
-
-+ (void)requestAccessForMediaType:(AVMediaType)mediaType completionHandler:(void (^)(BOOL granted))handler API_AVAILABLE(macos(10.14), ios(7.0));
-
-@end
-#endif
-
 /*****************************************************************************
  * Struct
  *****************************************************************************/
@@ -221,6 +213,11 @@ static int Open(vlc_object_t *p_this)
             return VLC_EGENERIC;
         }
 
+// only compile this block if the requestAccessForMediaType method is supported;
+// the code inside `if (@available(foo)) {}` is only *executed* when foo is true but
+// it is always parsed at compile time; @available is not an #ifdef that protects against
+// undefined types or functions.
+#ifdef MAC_OS_X_VERSION_10_14
         if (@available(macOS 10.14, *)) {
             msg_Dbg(p_demux, "Check user consent for access to the audio device");
 
@@ -240,6 +237,7 @@ static int Open(vlc_object_t *p_this)
                 return VLC_EGENERIC;
             }
         }
+#endif
 
         NSError *error = nil;
         AVCaptureDeviceInput *input = [AVCaptureDeviceInput deviceInputWithDevice:device error:&error];
