diff --git a/app/CMakeLists.txt b/app/CMakeLists.txt
index 6d191ce..32e5680 100644
--- a/app/CMakeLists.txt
+++ b/app/CMakeLists.txt
@@ -354,10 +354,6 @@ add_library(digikamgui_src
 ######################### digiKam GUI objects ############################
 
 
-# this is only required by Win & OSX
-file(GLOB ICONS_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/../data/icons/apps/*-apps-digikam.png")
-ecm_add_app_icon(digikam_SRCS ICONS ${ICONS_SRCS})
-
 set(DIGIKAM_OBJECTS
             $<TARGET_OBJECTS:digikamdatabasemain_src>
             $<TARGET_OBJECTS:digikamfacesengine_src>
@@ -493,6 +489,10 @@ set(digikam_SRCS
     main/main.cpp
 )
 
+# this is only required by Win & OSX
+file(GLOB ICONS_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/../data/icons/apps/*-apps-digikam.png")
+ecm_add_app_icon(digikam_SRCS ICONS ${ICONS_SRCS})
+
 add_executable(digikam ${digikam_SRCS})
 
 add_dependencies(digikam digikam-gitversion)
