diff --git a/core/libs/video/player/mediaplayerview.cpp b/core/libs/video/player/mediaplayerview.cpp
index 0e9b96446034360a54071efc4514d18030b39c58..796569ed764fc75a6727c6f4fd4c5f40c1787260 100644
--- a/core/libs/video/player/mediaplayerview.cpp
+++ b/core/libs/video/player/mediaplayerview.cpp
@@ -354,7 +354,7 @@ void MediaPlayerView::slotPlayerStateChanged(QtAV::AVPlayer::State state)
     {
         int rotate = 0;
 
-#if QTAV_VERSION > QTAV_VERSION_CHK(1, 12, 0)
+#if QTAV_VERSION >= QTAV_VERSION_CHK(1, 12, 0)
 
         // fix wrong rotation from QtAV git/master
 
diff --git a/core/libs/video/player/slidevideo.cpp b/core/libs/video/player/slidevideo.cpp
index 62cbfa5f425cdbb627c809e8b2352bb7357b57e6..f773eb8042e057a85e796c928c109bb1620a298c 100644
--- a/core/libs/video/player/slidevideo.cpp
+++ b/core/libs/video/player/slidevideo.cpp
@@ -263,7 +263,7 @@ void SlideVideo::slotPlayerStateChanged(QtAV::AVPlayer::State state)
     {
         int rotate = 0;
 
-#if QTAV_VERSION > QTAV_VERSION_CHK(1, 12, 0)
+#if QTAV_VERSION >= QTAV_VERSION_CHK(1, 12, 0)
 
         // fix wrong rotation from QtAV git/master
 
