diff --git a/core/libs/settings/applicationsettings.cpp b/core/libs/settings/applicationsettings.cpp
index 71efa00..7da637c 100644
--- a/core/libs/settings/applicationsettings.cpp
+++ b/core/libs/settings/applicationsettings.cpp
@@ -236,11 +236,11 @@ void ApplicationSettings::readSettings()
     d->scrollItemToCenter               = group.readEntry(d->configScrollItemToCenterEntry,                          false);
     d->stringComparisonType             = (StringComparisonType) group.readEntry(d->configStringComparisonTypeEntry, (int) Natural);
 
-#ifdef Q_OS_LINUX
+// #ifdef Q_OS_LINUX
     setApplicationStyle(group.readEntry(d->configApplicationStyleEntry, qApp->style()->objectName()));
-#else
-    setApplicationStyle(QLatin1String("Fusion"));
-#endif
+// #else
+//     setApplicationStyle(QLatin1String("Fusion"));
+// #endif
 
     d->iconTheme                        = group.readEntry(d->configIconThemeEntry,                                   QString());
     d->scanAtStart                      = group.readEntry(d->configScanAtStartEntry,                                 true);
diff --git a/core/showfoto/main/showfotosettings.cpp b/core/showfoto/main/showfotosettings.cpp
index 0a46d61..565ee90 100644
--- a/core/showfoto/main/showfotosettings.cpp
+++ b/core/showfoto/main/showfotosettings.cpp
@@ -243,11 +243,11 @@ void ShowfotoSettings::init()
     d->showPhotoWB             = false;
     d->showPhotoDate           = true;
     d->showPhotoMode           = true;
-#ifdef Q_OS_LINUX
+// #ifdef Q_OS_LINUX
     d->applicationStyle        = qApp->style()->objectName();
-#else
-    d->applicationStyle        = QLatin1String("Fusion");
-#endif
+// #else
+//     d->applicationStyle        = QLatin1String("Fusion");
+// #endif
     d->iconTheme               = QString();
 }
 
@@ -259,9 +259,9 @@ void ShowfotoSettings::readSettings()
     d->deleteItem2Trash        = group.readEntry(d->configDeleteItem2Trash, true);
     d->theme                   = group.readEntry(d->configCurrentTheme, Digikam::ThemeManager::instance()->defaultThemeName());
     d->rightSideBarStyle       = group.readEntry(d->configRightSideBarStyle, 0);
-#ifdef Q_OS_LINUX
+// #ifdef Q_OS_LINUX
     d->applicationStyle        = group.readEntry(d->configApplicationStyle, qApp->style()->objectName());
-#endif
+// #endif
     d->iconTheme               = group.readEntry(d->configIconTheme, QString());
     d->itemCenter              = group.readEntry(d->configItemCenter, false);
     d->showSplash              = group.readEntry(d->configShowSplash, true);
diff --git a/core/showfoto/setup/showfotosetupmisc.cpp b/core/showfoto/setup/showfotosetupmisc.cpp
index fb190c2..effd13f 100644
--- a/core/showfoto/setup/showfotosetupmisc.cpp
+++ b/core/showfoto/setup/showfotosetupmisc.cpp
@@ -134,10 +134,10 @@ SetupMisc::SetupMisc(QWidget* const parent)
         d->applicationStyle->addItem(styleList.at(i));
     }
 
-#ifndef Q_OS_LINUX
-    // See Bug #365262
-    appStyleHbox->setVisible(false);
-#endif
+// #ifndef Q_OS_LINUX
+//     // See Bug #365262
+//     appStyleHbox->setVisible(false);
+// #endif
 
     DHBox* const iconThemeHbox = new DHBox(panel);
     d->iconThemeLabel          = new QLabel(i18n("Icon theme (changes after restart):"), iconThemeHbox);
@@ -226,9 +226,9 @@ void SetupMisc::readSettings()
     d->sidebarType->setCurrentIndex(d->settings->getRightSideBarStyle());
     d->sortOrderComboBox->setCurrentIndex(d->settings->getSortRole());
     d->sortReverse->setChecked(d->settings->getReverseSort());
-#ifdef Q_OS_LINUX
+// #ifdef Q_OS_LINUX
     d->applicationStyle->setCurrentIndex(d->applicationStyle->findText(d->settings->getApplicationStyle(), Qt::MatchFixedString));
-#endif
+// #endif
     d->iconTheme->setCurrentIndex(d->iconTheme->findData(d->settings->getIconTheme()));
 }
 
@@ -241,9 +241,9 @@ void SetupMisc::applySettings()
     d->settings->setRightSideBarStyle(d->sidebarType->currentIndex());
     d->settings->setSortRole(d->sortOrderComboBox->currentIndex());
     d->settings->setReverseSort(d->sortReverse->isChecked());
-#ifdef Q_OS_LINUX
+// #ifdef Q_OS_LINUX
     d->settings->setApplicationStyle(d->applicationStyle->currentText());
-#endif
+// #endif
     d->settings->setIconTheme(d->iconTheme->currentData().toString());
     d->settings->syncConfig();
 }
diff --git a/core/utilities/setup/setupmisc.cpp b/core/utilities/setup/setupmisc.cpp
index a183688..e9225b9 100644
--- a/core/utilities/setup/setupmisc.cpp
+++ b/core/utilities/setup/setupmisc.cpp
@@ -153,10 +153,10 @@ SetupMisc::SetupMisc(QWidget* const parent)
         d->applicationStyle->addItem(styleList.at(i));
     }
 
-#ifndef Q_OS_LINUX
-    // See Bug #365262
-    appStyleHbox->setVisible(false);
-#endif
+// #ifndef Q_OS_LINUX
+//     // See Bug #365262
+//     appStyleHbox->setVisible(false);
+// #endif
 
     // --------------------------------------------------------
 
@@ -229,9 +229,9 @@ void SetupMisc::applySettings()
     settings->setSidebarTitleStyle(d->sidebarType->currentIndex() == 0 ? DMultiTabBar::ActiveIconText : DMultiTabBar::AllIconsText);
     settings->setStringComparisonType((ApplicationSettings::StringComparisonType)d->stringComparisonType->itemData(d->stringComparisonType->currentIndex()).toInt());
 
-#ifdef Q_OS_LINUX
+// #ifdef Q_OS_LINUX
     settings->setApplicationStyle(d->applicationStyle->currentText());
-#endif
+// #endif
 
     settings->setIconTheme(d->iconTheme->currentData().toString());
     settings->saveSettings();
@@ -251,9 +251,9 @@ void SetupMisc::readSettings()
     d->sidebarType->setCurrentIndex(settings->getSidebarTitleStyle() == DMultiTabBar::ActiveIconText ? 0 : 1);
     d->stringComparisonType->setCurrentIndex(settings->getStringComparisonType());
 
-#ifdef Q_OS_LINUX
+// #ifdef Q_OS_LINUX
     d->applicationStyle->setCurrentIndex(d->applicationStyle->findText(settings->getApplicationStyle(), Qt::MatchFixedString));
-#endif
+// #endif
 
     d->iconTheme->setCurrentIndex(d->iconTheme->findData(settings->getIconTheme()));
 }
