--- digikam-5.9.0/core/libs/database/coredb/orig.coredbaccess.cpp	2018-03-19 10:41:58.000000000 +0100
+++ digikam-5.9.0/core/libs/database/coredb/coredbaccess.cpp	2018-03-25 16:48:08.000000000 +0200
@@ -110,6 +110,12 @@
 
 CoreDbAccess::CoreDbAccess()
 {
+    if (!d) {
+        qCCritical(DIGIKAM_COREDB_LOG) << Q_FUNC_INFO
+            << "called with a NULL d-ptr; this shouldn't happen. Trying to proceed with eyes, toes and fingers crossed";
+        DbEngineParameters dummy;
+        setParameters(dummy, DatabaseSlave);
+    }
     // You will want to call setParameters before constructing CoreDbAccess
     Q_ASSERT(d);
 
