--- digikam-6.2.0/core/cmake/modules/orig.FindImageMagick.cmake	2019-08-26 16:35:17.000000000 +0200
+++ digikam-6.2.0/core/cmake/modules/FindImageMagick.cmake	2019-08-26 16:36:42.000000000 +0200
@@ -89,7 +89,7 @@
                   ${ImageMagick_INCLUDE_DIRS}
                   "[HKEY_LOCAL_MACHINE\\SOFTWARE\\ImageMagick\\Current;BinPath]/include"
               PATH_SUFFIXES
-                  ImageMagick ImageMagick-6 ImageMagick-7
+                  ImageMagick ImageMagick-7 ImageMagick-6
               DOC "Path to the ImageMagick arch-independent include dir."
               NO_DEFAULT_PATH
     )
@@ -103,7 +103,7 @@
                   ${ImageMagick_INCLUDE_DIRS}
                   "[HKEY_LOCAL_MACHINE\\SOFTWARE\\ImageMagick\\Current;BinPath]/include"
               PATH_SUFFIXES
-                  ImageMagick ImageMagick-6 ImageMagick-7
+                  ImageMagick ImageMagick-7 ImageMagick-6
               DOC "Path to the ImageMagick arch-specific include dir."
               NO_DEFAULT_PATH
     )
@@ -235,12 +235,12 @@
 
         FIND_IMAGEMAGICK_API(Magick++ Magick++.h
                              Magick++ CORE_RL_Magick++_
-                             Magick++-6 Magick++-7
+                             Magick++-7 Magick++-6
                              Magick++-Q8 Magick++-Q16 Magick++-Q16HDRI Magick++-Q8HDRI
-                             Magick++-6.Q64 Magick++-6.Q32 Magick++-6.Q64HDRI Magick++-6.Q32HDRI
-                             Magick++-6.Q16 Magick++-6.Q8 Magick++-6.Q16HDRI Magick++-6.Q8HDRI
                              Magick++-7.Q64 Magick++-7.Q32 Magick++-7.Q64HDRI Magick++-7.Q32HDRI
                              Magick++-7.Q16 Magick++-7.Q8 Magick++-7.Q16HDRI Magick++-7.Q8HDRI
+                             Magick++-6.Q64 Magick++-6.Q32 Magick++-6.Q64HDRI Magick++-6.Q32HDRI
+                             Magick++-6.Q16 Magick++-6.Q8 Magick++-6.Q16HDRI Magick++-6.Q8HDRI
         )
 
         list(APPEND ImageMagick_REQUIRED_VARS ImageMagick_Magick++_LIBRARY)
@@ -249,12 +249,12 @@
 
         FIND_IMAGEMAGICK_API(MagickWand "wand/MagickWand.h;MagickWand/MagickWand.h"
                              Wand MagickWand CORE_RL_wand_ CORE_RL_MagickWand_
-                             MagickWand-6 MagickWand-7
+                             MagickWand-7 MagickWand-6
                              MagickWand-Q16 MagickWand-Q8 MagickWand-Q16HDRI MagickWand-Q8HDRI
-                             MagickWand-6.Q64 MagickWand-6.Q32 MagickWand-6.Q64HDRI MagickWand-6.Q32HDRI
-                             MagickWand-6.Q16 MagickWand-6.Q8 MagickWand-6.Q16HDRI MagickWand-6.Q8HDRI
                              MagickWand-7.Q64 MagickWand-7.Q32 MagickWand-7.Q64HDRI MagickWand-7.Q32HDRI
                              MagickWand-7.Q16 MagickWand-7.Q8 MagickWand-7.Q16HDRI MagickWand-7.Q8HDRI
+                             MagickWand-6.Q64 MagickWand-6.Q32 MagickWand-6.Q64HDRI MagickWand-6.Q32HDRI
+                             MagickWand-6.Q16 MagickWand-6.Q8 MagickWand-6.Q16HDRI MagickWand-6.Q8HDRI
         )
 
         list(APPEND ImageMagick_REQUIRED_VARS ImageMagick_MagickWand_LIBRARY)
@@ -263,12 +263,12 @@
 
         FIND_IMAGEMAGICK_API(MagickCore "magick/MagickCore.h;MagickCore/MagickCore.h"
                              Magick MagickCore CORE_RL_magick_ CORE_RL_MagickCore_
-                             MagickCore-6 MagickCore-7
+                             MagickCore-7 MagickCore-6
                              MagickCore-Q16 MagickCore-Q8 MagickCore-Q16HDRI MagickCore-Q8HDRI
-                             MagickCore-6.Q64 MagickCore-6.Q32 MagickCore-6.Q64HDRI MagickCore-6.Q32HDRI
-                             MagickCore-6.Q16 MagickCore-6.Q8 MagickCore-6.Q16HDRI MagickCore-6.Q8HDRI
                              MagickCore-7.Q64 MagickCore-7.Q32 MagickCore-7.Q64HDRI MagickCore-7.Q32HDRI
                              MagickCore-7.Q16 MagickCore-7.Q8 MagickCore-7.Q16HDRI MagickCore-7.Q8HDRI
+                             MagickCore-6.Q64 MagickCore-6.Q32 MagickCore-6.Q64HDRI MagickCore-6.Q32HDRI
+                             MagickCore-6.Q16 MagickCore-6.Q8 MagickCore-6.Q16HDRI MagickCore-6.Q8HDRI
         )
 
         list(APPEND ImageMagick_REQUIRED_VARS ImageMagick_MagickCore_LIBRARY)
--- digikam-6.2.0/core/tests/dimg/orig.magickloader.cpp	2019-08-26 18:16:30.000000000 +0200
+++ digikam-6.2.0/core/tests/dimg/magickloader.cpp	2019-08-26 18:21:29.000000000 +0200
@@ -139,7 +139,11 @@
         if (inf->encoder) mode.append(QLatin1Char('W'));
         else              mode.append(QLatin1Char('-'));
 
+#if MagickLibVersion >= 0x700
+        QString module = QLatin1String(inf->magick_module);
+#else
         QString module = QLatin1String(inf->module);
+#endif
         QString mime   = QMimeDatabase().mimeTypeForFile(QFileInfo(QString::fromLatin1("foo.%1").arg(module))).name();
 
         if (module != QLatin1String("DNG")  &&
