diff --git a/core/libs/widgets/mainview/dxmlguiwindow.cpp b/core/libs/widgets/mainview/dxmlguiwindow.cpp
index 557b2c9358fe1377814ce61767ed0a588e8ea839..b869f4c332aa4b8c0cde295d730114b57f06a146 100644
--- a/core/libs/widgets/mainview/dxmlguiwindow.cpp
+++ b/core/libs/widgets/mainview/dxmlguiwindow.cpp
@@ -347,7 +347,7 @@ void DXmlGuiWindow::createSidebarActions()
 void DXmlGuiWindow::createSettingsActions()
 {
     d->showMenuBarAction   = KStandardAction::showMenubar(this, SLOT(slotShowMenuBar()), actionCollection());
-#ifdef Q_OS_OSX
+#ifdef Q_OS_OSX_STOCK
     // Under MacOS the menu bar visibility is managed by desktop.
     d->showMenuBarAction->setVisible(false);
 #endif
