--- digikam-7.1.0/core/cmake/modules/FindQtAV.cmake.orig	2020-09-24 13:30:52.856817367 +0200
+++ digikam-7.1.0/core/cmake/modules/FindQtAV.cmake	2020-09-24 13:33:49.747264799 +0200
@@ -23,7 +23,7 @@
 
 Message(STATUS "QtAV search path: ${_qt5_install_prefix}")
 
-if (NOT APPLE)
+if (NOT APPLE_QTAV_FRAMEWORKS)
 
     find_path(QTAV_CORE_INCLUDE_DIR
               NAMES QtAV.h
@@ -101,7 +101,7 @@
 
 if(QtAV_FOUND)
 
-    if(NOT APPLE)
+    if(NOT APPLE_QTAV_FRAMEWORKS)
 
         if(EXISTS ${QTAV_CORE_INCLUDE_DIR}/version.h)
 
