diff --git a/core/cmake/modules/FindQtAV.cmake b/core/cmake/modules/FindQtAV.cmake
index 91b7add7a05e14458be5a4bfc4114c5c85f8d8e7..3cfa5e1a36272f3b38867d8310ae9cbea98d6e99 100644
--- a/core/cmake/modules/FindQtAV.cmake
+++ b/core/cmake/modules/FindQtAV.cmake
@@ -22,7 +22,7 @@ find_package(Qt5
 
 Message(STATUS "QtAV search path: ${_qt5_install_prefix}")
 
-if (NOT APPLE)
+if (NOT APPLE_QTAV_FRAMEWORKS)
     find_path(QTAV_CORE_INCLUDE_DIR
               NAMES QtAV.h
               HINTS ${Qt5Core_INCLUDE_DIRS}
@@ -95,7 +95,7 @@ find_package_handle_standard_args(QtAV REQUIRED_VARS QTAV_LIBRARIES QTAV_INCLUDE
 
 if(QtAV_FOUND)
 
-    if (NOT APPLE)
+    if (NOT APPLE_QTAV_FRAMEWORKS)
         file(READ ${QTAV_CORE_INCLUDE_DIR}/version.h QTAV_VERSION_CONTENT)
     else()
         file(READ ${QTAV_CORE_INCLUDE_DIR}/QtAV/version.h QTAV_VERSION_CONTENT)
