--- digikam-5.4.0/core/libs/dmetadata/orig.metaengine_p.cpp	2017-01-08 23:08:44.000000000 +0100
+++ digikam-5.4.0/core/libs/dmetadata/metaengine_p.cpp	2017-05-14 00:16:26.000000000 +0200
@@ -49,7 +49,7 @@
 #include "digikam_debug.h"
 
 // Pragma directives to reduce warnings from Exiv2.
-#if not defined(__APPLE__) && defined(__GNUC__)
+#if !defined(__APPLE__) && defined(__GNUC__)
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wdeprecated-declarations"
 #endif
@@ -723,7 +723,7 @@
 }  // namespace Digikam
 
 // Restore warnings
-#if not defined(__APPLE__) && defined(__GNUC__)
+#if !defined(__APPLE__) && defined(__GNUC__)
 #pragma GCC diagnostic pop
 #endif
 
--- digikam-5.4.0/core/app/utils/orig.libopencv.h.cmake.in	2017-01-08 23:08:44.000000000 +0100
+++ digikam-5.4.0/core/app/utils/libopencv.h.cmake.in	2017-05-14 00:22:08.000000000 +0200
@@ -27,7 +27,7 @@
 #define LIB_OPEN_CV_H
 
 // Pragma directives to reduce warnings from OpenCV header files.
-#if not defined(__APPLE__) && defined(__GNUC__)
+#if !defined(__APPLE__) && defined(__GNUC__)
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wnon-virtual-dtor"
 #pragma GCC diagnostic ignored "-Woverloaded-virtual"
@@ -82,7 +82,7 @@
 #endif
 
 // Restore warnings
-#if not defined(__APPLE__) && defined(__GNUC__)
+#if !defined(__APPLE__) && defined(__GNUC__)
 #pragma GCC diagnostic pop
 #endif
 
--- digikam-5.4.0/core/libs/database/imagehistory/orig.imagehistorygraph_boost.h	2017-01-08 23:08:44.000000000 +0100
+++ digikam-5.4.0/core/libs/database/imagehistory/imagehistorygraph_boost.h	2017-05-14 00:23:01.000000000 +0200
@@ -28,7 +28,7 @@
 #include "digikam_config.h"
 
 // Pragma directives to reduce warnings from Boost header files.
-#if not defined(__APPLE__) && defined(__GNUC__)
+#if !defined(__APPLE__) && defined(__GNUC__)
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wunused-local-typedefs"
 #endif
@@ -1554,7 +1554,7 @@
 } // namespace Digikam
 
 // Restore warnings
-#if not defined(__APPLE__) && defined(__GNUC__)
+#if !defined(__APPLE__) && defined(__GNUC__)
 #pragma GCC diagnostic pop
 #endif
 
--- digikam-5.4.0/core/libs/dimg/filters/greycstoration/orig.greycstorationfilter.cpp	2017-01-08 23:08:44.000000000 +0100
+++ digikam-5.4.0/core/libs/dimg/filters/greycstoration/greycstorationfilter.cpp	2017-05-14 00:24:19.000000000 +0200
@@ -52,7 +52,7 @@
 #define GREYSTORATION_USING_GFACT 1
 
 // Pragma directives to reduce warnings from CImg header files.
-#if not defined(__APPLE__) && defined(__GNUC__)
+#if !defined(__APPLE__) && defined(__GNUC__)
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wunused-local-typedefs"
 #pragma GCC diagnostic ignored "-Wunused-but-set-variable"
@@ -71,7 +71,7 @@
 #include "cimg/CImg.h"
 
 // Restore warnings
-#if not defined(__APPLE__) && defined(__GNUC__)
+#if !defined(__APPLE__) && defined(__GNUC__)
 #pragma GCC diagnostic pop
 #endif
 
--- digikam-5.4.0/core/libs/dimg/filters/orig.randomnumbergenerator.cpp	2017-01-08 23:08:44.000000000 +0100
+++ digikam-5.4.0/core/libs/dimg/filters/randomnumbergenerator.cpp	2017-05-14 00:25:01.000000000 +0200
@@ -26,7 +26,7 @@
 // Boost includes
 
 // Pragma directives to reduce warnings from Boost header files.
-#if not defined(__APPLE__) && defined(__GNUC__)
+#if !defined(__APPLE__) && defined(__GNUC__)
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wundef"
 #endif
@@ -44,7 +44,7 @@
 #include <boost/random/variate_generator.hpp>
 
 // Restore warnings
-#if not defined(__APPLE__) && defined(__GNUC__)
+#if !defined(__APPLE__) && defined(__GNUC__)
 #pragma GCC diagnostic pop
 #endif
 
--- digikam-5.4.0/core/libs/dimg/filters/sharp/orig.matrix.cpp	2017-01-08 23:08:44.000000000 +0100
+++ digikam-5.4.0/core/libs/dimg/filters/sharp/matrix.cpp	2017-05-14 00:25:54.000000000 +0200
@@ -43,7 +43,7 @@
 // Eigen includes
 
 // Pragma directives to reduce warnings from Eigen header files.
-#if not defined(__APPLE__) && defined(__GNUC__)
+#if !defined(__APPLE__) && defined(__GNUC__)
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wunused-local-typedefs"
 #pragma GCC diagnostic ignored "-Wdeprecated-declarations"
@@ -53,7 +53,7 @@
 #include <Eigen/LU>
 
 // Restore warnings
-#if not defined(__APPLE__) && defined(__GNUC__)
+#if !defined(__APPLE__) && defined(__GNUC__)
 #pragma GCC diagnostic pop
 #endif
 
--- digikam-5.4.0/core/libs/jpegutils/orig.jpegutils.cpp	2017-01-08 23:08:44.000000000 +0100
+++ digikam-5.4.0/core/libs/jpegutils/jpegutils.cpp	2017-05-14 00:26:32.000000000 +0200
@@ -53,7 +53,7 @@
 }
 
 // Pragma directives to reduce warnings from libjpeg transupp header file.
-#if not defined(__APPLE__) && defined(__GNUC__)
+#if !defined(__APPLE__) && defined(__GNUC__)
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wunused-parameter"
 #endif
@@ -69,7 +69,7 @@
 }
 
 // Restore warnings
-#if not defined(__APPLE__) && defined(__GNUC__)
+#if !defined(__APPLE__) && defined(__GNUC__)
 #pragma GCC diagnostic pop
 #endif
 
--- digikam-5.4.0/core/libs/rawengine/orig.drawdecoder_p.h	2017-01-08 23:08:44.000000000 +0100
+++ digikam-5.4.0/core/libs/rawengine/drawdecoder_p.h	2017-05-14 00:27:03.000000000 +0200
@@ -31,7 +31,7 @@
 #include <QByteArray>
 
 // Pragma directives to reduce warnings from LibRaw header files.
-#if not defined(__APPLE__) && defined(__GNUC__)
+#if !defined(__APPLE__) && defined(__GNUC__)
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wdeprecated-declarations"
 #endif
@@ -46,7 +46,7 @@
 #include <libraw.h>
 
 // Restore warnings
-#if not defined(__APPLE__) && defined(__GNUC__)
+#if !defined(__APPLE__) && defined(__GNUC__)
 #pragma GCC diagnostic pop
 #endif
 
