--- digikam-6.2.0/core/utilities/setup/orig.setupmisc.cpp	2019-07-31 18:12:57.000000000 +0200
+++ digikam-6.2.0/core/utilities/setup/setupmisc.cpp	2019-08-25 14:06:33.000000000 +0200
@@ -248,6 +248,7 @@
 
     QMap<QString, QString> iconThemes;
     QMap<QString, QString> themeWhiteList;
+    themeWhiteList.insert(QLatin1String("ciment"),          i18nc("icon theme", "Ciment"));
     themeWhiteList.insert(QLatin1String("adwaita"),         i18nc("icon theme", "Adwaita"));
     themeWhiteList.insert(QLatin1String("breeze"),          i18nc("icon theme", "Breeze"));
     themeWhiteList.insert(QLatin1String("breeze-dark"),     i18nc("icon theme", "Breeze Dark"));
