--- digikam-7.2.0/core/libs/metadataengine/engine/orig.metaengine_exif.cpp	2021-03-22 08:40:57.000000000 +0100
+++ digikam-7.2.0/core/libs/metadataengine/engine/metaengine_exif.cpp	2023-09-08 18:26:45.369451383 +0200
@@ -743,8 +743,10 @@
     }
     catch (Exiv2::AnyError& e)
     {
-        d->printExiv2ExceptionError(QString::fromLatin1("Cannot find Exif key '%1' into image using Exiv2 ")
-                                    .arg(QLatin1String(exifTagName)), e);
+//         d->printExiv2ExceptionError(QString::fromLatin1("Cannot find Exif key '%1' into image using Exiv2 ")
+//                                     .arg(QLatin1String(exifTagName)), e);
+        d->printExiv2MessageHandler(e.code(), qPrintable(QString::fromLatin1("Cannot find Exif key '%1' in image using Exiv2 ")
+                                    .arg(QLatin1String(exifTagName))));
     }
     catch (...)
     {
@@ -776,8 +778,10 @@
     }
     catch (Exiv2::AnyError& e)
     {
-        d->printExiv2ExceptionError(QString::fromLatin1("Cannot find Exif key '%1' into image using Exiv2 ")
-                                    .arg(QLatin1String(exifTagName)), e);
+//         d->printExiv2ExceptionError(QString::fromLatin1("Cannot find Exif key '%1' into image using Exiv2 ")
+//                                     .arg(QLatin1String(exifTagName)), e);
+        d->printExiv2MessageHandler(e.code(), qPrintable(QString::fromLatin1("Cannot find Exif key '%1' in image using Exiv2 ")
+                                    .arg(QLatin1String(exifTagName))));
     }
     catch (...)
     {
@@ -886,8 +890,10 @@
     }
     catch (Exiv2::AnyError& e)
     {
-        d->printExiv2ExceptionError(QString::fromLatin1("Cannot find Exif key '%1' in the image using Exiv2 ")
-                                    .arg(QLatin1String(exifTagName)), e);
+//         d->printExiv2ExceptionError(QString::fromLatin1("Cannot find Exif key '%1' in the image using Exiv2 ")
+//                                     .arg(QLatin1String(exifTagName)), e);
+        d->printExiv2MessageHandler(e.code(), qPrintable(QString::fromLatin1("Cannot find Exif key '%1' in the image using Exiv2 ")
+                                    .arg(QLatin1String(exifTagName))));
     }
     catch (...)
     {
@@ -935,8 +941,10 @@
     }
     catch (Exiv2::AnyError& e)
     {
-        d->printExiv2ExceptionError(QString::fromLatin1("Cannot find Exif key '%1' into image using Exiv2 ")
-                                    .arg(QLatin1String(exifTagName)), e);
+//         d->printExiv2ExceptionError(QString::fromLatin1("Cannot find Exif key '%1' into image using Exiv2 ")
+//                                     .arg(QLatin1String(exifTagName)), e);
+        d->printExiv2MessageHandler(e.code(), qPrintable(QString::fromLatin1("Cannot find Exif key '%1' in image using Exiv2 ")
+                                    .arg(QLatin1String(exifTagName))));
     }
     catch (...)
     {
