diff --git a/core/CMakeLists.txt b/core/CMakeLists.txt
index 9a37acf1eede7971f551a46a52a5e156f6b1785f..8ff6fc0dd1be5c0f129f4ec9d3b4b96b52731a38 100644
--- a/core/CMakeLists.txt
+++ b/core/CMakeLists.txt
@@ -259,7 +259,7 @@ set_package_properties("Exiv2"     PROPERTIES
 # -- check Media player --------------------------------------------------------
 
 find_package(FFmpeg COMPONENTS AVCODEC AVFILTER AVFORMAT AVUTIL SWSCALE)
-find_package(QtAV)
+# find_package(QtAV)
 
 if(ENABLE_MEDIAPLAYER)
 
