--- digikam-6.2.0/core/libs/dplugins/webservices/o2/src/orig.o0baseauth.cpp	2019-09-02 12:53:52.000000000 +0200
+++ digikam-6.2.0/core/libs/dplugins/webservices/o2/src/o0baseauth.cpp	2019-09-02 13:14:49.501215673 +0200
@@ -107,7 +107,9 @@
 }
 
 int O0BaseAuth::localPort() {
-    return localPort_;
+    bool ok = false;
+    int portOverride = qEnvironmentVariableIntValue("DK_WEBSERVICE_AUTH_PORT", &ok);
+    return Q_UNLIKELY(ok && portOverride > 0) ? portOverride : localPort_;
 }
 
 void O0BaseAuth::setLocalPort(int value) {
--- digikam-6.2.0/core/libs/dplugins/webservices/o2/src/orig.o2.cpp	2019-09-02 15:09:24.000000000 +0200
+++ digikam-6.2.0/core/libs/dplugins/webservices/o2/src/o2.cpp	2019-09-02 15:09:28.181215673 +0200
@@ -182,7 +182,7 @@
         } else {
             // Start listening to authentication replies
             if (!replyServer_->isListening()) {
-                if (replyServer_->listen(QHostAddress::Any, localPort_)) {
+                if (replyServer_->listen(QHostAddress::Any, localPort())) {
                     qDebug() << "O2::link: Reply server listening on port" << localPort();
                 } else {
                     qWarning() << "O2::link: Reply server failed to start listening on port" << localPort();
