diff --git a/core/libs/widgets/mainview/dxmlguiwindow.cpp b/core/libs/widgets/mainview/dxmlguiwindow.cpp
index 02dd549fe4d85e3fbef9f380b6f64fde29456ea9..8f557845bca204b4e60f7f7cff24c7d6db03a793 100644
--- a/core/libs/widgets/mainview/dxmlguiwindow.cpp
+++ b/core/libs/widgets/mainview/dxmlguiwindow.cpp
@@ -354,7 +354,7 @@ void DXmlGuiWindow::createSidebarActions()
 void DXmlGuiWindow::createSettingsActions()
 {
     d->showMenuBarAction   = KStandardAction::showMenubar(this, SLOT(slotShowMenuBar()), actionCollection());
-#ifdef Q_OS_MACOS
+#ifdef Q_OS_MACOS_STOCK
     // Under MacOS the menu bar visibility is managed by desktop.
 
     d->showMenuBarAction->setVisible(false);
