--- digikam-5.7.0/core/libs/widgets/mainview/orig.dxmlguiwindow.cpp	2017-09-05 10:51:22.000000000 +0200
+++ digikam-5.7.0/core/libs/widgets/mainview/dxmlguiwindow.cpp	2018-01-05 23:57:32.000000000 +0100
@@ -315,7 +315,7 @@
 void DXmlGuiWindow::createSettingsActions()
 {
     d->showMenuBarAction   = KStandardAction::showMenubar(this, SLOT(slotShowMenuBar()), actionCollection());
-#ifdef Q_OS_OSX
+#ifdef Q_OS_OSX_STOCK
     // Under MacOS the menu bar visibility is managed by desktop.
     d->showMenuBarAction->setVisible(false);
 #endif
