--- digikam-5.9.0/core/libs/threadimageio/orig.videothumbnailer.h	2018-03-19 10:41:59.000000000 +0100
+++ digikam-5.9.0/core/libs/threadimageio/videothumbnailer.h	2018-03-25 19:22:10.000000000 +0200
@@ -67,7 +67,7 @@
 
 private Q_SLOTS:
 
-    void slotFrameError();
+    void slotFrameError(const QString errMsg=QString());
     void slotFrameExtracted(const QtAV::VideoFrame& frame);
 
 private:
--- digikam-5.9.0/core/libs/threadimageio/orig.videothumbnailer.cpp	2018-03-25 19:17:34.000000000 +0200
+++ digikam-5.9.0/core/libs/threadimageio/videothumbnailer.cpp	2018-03-25 19:22:17.000000000 +0200
@@ -142,7 +142,7 @@
     connect(d->extractor, SIGNAL(frameExtracted(QtAV::VideoFrame)),
             this, SLOT(slotFrameExtracted(QtAV::VideoFrame)));
 
-#if QTAV_VERSION > QTAV_VERSION_CHK(1, 12, 0)
+#if QTAV_VERSION >= QTAV_VERSION_CHK(1, 12, 0)
     connect(d->extractor, SIGNAL(error(QString)),
             this, SLOT(slotFrameError()));
 #else
@@ -242,14 +242,18 @@
     d->extractor->extract();
 }
 
-void VideoThumbnailer::slotFrameError()
+void VideoThumbnailer::slotFrameError(const QString errMsg)
 {
     if (d->audioFile)
     {
-        qCDebug(DIGIKAM_GENERAL_LOG) << "Audio file has no embedded image for " << d->name;
+        if (!errMsg.isEmpty())
+            qCWarning(DIGIKAM_GENERAL_LOG) << "Audio file has no embedded image for " << d->name << ":" << errMsg;
+        else
+            qCDebug(DIGIKAM_GENERAL_LOG) << "Audio file has no embedded image for " << d->name;
         emit signalThumbnailFailed(d->file);
         return;
-    }
+    } else if (!errMsg.isEmpty())
+        qCWarning(DIGIKAM_GENERAL_LOG) << d->file << "has no embedded image for " << d->name << ":" << errMsg;
 
     tryExtractVideoFrame();
 }
