diff --git a/core/libs/threadimageio/fileio/loadingcache.cpp b/core/libs/threadimageio/fileio/loadingcache.cpp
index 17fed673bac1c63bc14d65aa8fb9016be138b912..e8d782bf8dab9850ae0bd8f2c802cdb6c51e6c5d 100644
--- a/core/libs/threadimageio/fileio/loadingcache.cpp
+++ b/core/libs/threadimageio/fileio/loadingcache.cpp
@@ -26,6 +26,7 @@
 // Qt includes
 
 #include <QCoreApplication>
+#include <QGuiApplication>
 #include <QEvent>
 #include <QCache>
 #include <QMap>
@@ -304,13 +304,15 @@ void LoadingCache::removeThumbnails()
 
 void LoadingCache::setThumbnailCacheSize(int numberOfQImages, int numberOfQPixmaps)
 {
-    d->thumbnailImageCache.setMaxCost(numberOfQImages *
+    if (QGuiApplication::primaryScreen()) {
+	    d->thumbnailImageCache.setMaxCost(numberOfQImages *
                                       ThumbnailSize::maxThumbsSize() *
                                       ThumbnailSize::maxThumbsSize() * 4);
 
-    d->thumbnailPixmapCache.setMaxCost(numberOfQPixmaps *
+	    d->thumbnailPixmapCache.setMaxCost(numberOfQPixmaps *
                                        ThumbnailSize::maxThumbsSize() *
                                        ThumbnailSize::maxThumbsSize() * QPixmap::defaultDepth() / 8);
+	}
 }
 
 void LoadingCache::setFileWatch(LoadingCacheFileWatch* const watch)
