--- digikam-5.8.0/core/libs/threadimageio/orig.videothumbnailer.cpp	2018-01-09 15:13:47.000000000 +0100
+++ digikam-5.8.0/core/libs/threadimageio/videothumbnailer.cpp	2018-03-25 14:49:36.659696977 +0200
@@ -142,7 +142,7 @@
     connect(d->extractor, SIGNAL(frameExtracted(QtAV::VideoFrame)),
             this, SLOT(slotFrameExtracted(QtAV::VideoFrame)));
 
-#if QTAV_VERSION > QTAV_VERSION_CHK(1, 12, 0)
+#if QTAV_VERSION >= QTAV_VERSION_CHK(1, 12, 0)
     connect(d->extractor, SIGNAL(error(QString)),
             this, SLOT(slotFrameError()));
 #else
