diff --git a/core/libs/dimg/filters/icc/iccprofile.cpp b/core/libs/dimg/filters/icc/iccprofile.cpp
index 4789f4c..259cef5 100644
--- a/core/libs/dimg/filters/icc/iccprofile.cpp
+++ b/core/libs/dimg/filters/icc/iccprofile.cpp
@@ -561,9 +561,9 @@ QStringList IccProfile::defaultSearchPaths()
         dataDirs << QLatin1String("/usr/share");
     }
 
-    if (!dataDirs.contains(QLatin1String("/usr/local/share")))
+    if (!dataDirs.contains(QLatin1String("@PREFIX@/share")))
     {
-        dataDirs << QLatin1String("/usr/local/share");
+        dataDirs << QLatin1String("@PREFIX@/share");
     }
 
     foreach (const QString& dataDir, dataDirs)
diff --git a/core/utilities/setup/setupicc.cpp b/core/utilities/setup/setupicc.cpp
index 0ac1581..cd35238 100644
--- a/core/utilities/setup/setupicc.cpp
+++ b/core/utilities/setup/setupicc.cpp
@@ -936,6 +936,7 @@ void SetupICC::slotShowDefaultSearchPaths()
     QString text = i18n("On Linux, the default search paths include "
                         "<ul>"
                         "<li>/usr/share/color/icc</li>"
+                        "<li>@PREFIX@/share/color/icc</li>"
                         "<li>/usr/local/share/color/icc</li>"
                         "<li>~/.local/share/color/icc/</li>"
                         "<li>~/.local/share/icc/</li>"
