diff --git a/core/libs/widgets/mainview/dxmlguiwindow.cpp b/core/libs/widgets/mainview/dxmlguiwindow.cpp
index 53e3b2a..e13296e 100644
--- a/core/libs/widgets/mainview/dxmlguiwindow.cpp
+++ b/core/libs/widgets/mainview/dxmlguiwindow.cpp
@@ -364,7 +364,7 @@ void DXmlGuiWindow::createSettingsActions()
 {
     d->showMenuBarAction   = KStandardAction::showMenubar(this, SLOT(slotShowMenuBar()), actionCollection());
 
-#ifdef Q_OS_MACOS
+#ifdef Q_OS_MACOS_STOCK
 
     // Under MacOS the menu bar visibility is managed by desktop.
 
