diff --git a/core/libs/threadimageio/fileio/loadingcache.cpp b/core/libs/threadimageio/fileio/loadingcache.cpp
index 418ea951bd541cf33d00ae332fb3bb3aaa6d7d06..84c6fd880db7b4c5909406ebd3ad562221508076 100644
--- a/core/libs/threadimageio/fileio/loadingcache.cpp
+++ b/core/libs/threadimageio/fileio/loadingcache.cpp
@@ -25,6 +25,7 @@
 
 // Qt includes
 
+#include <QGuiApplication>
 #include <QCache>
 #include <QMap>
 
@@ -330,13 +331,15 @@ void LoadingCache::removeThumbnails()
 
 void LoadingCache::setThumbnailCacheSize(int numberOfQImages, int numberOfQPixmaps)
 {
-    d->thumbnailImageCache.setMaxCost(numberOfQImages *
+    if (QGuiApplication::primaryScreen()) {
+	    d->thumbnailImageCache.setMaxCost(numberOfQImages *
                                       ThumbnailSize::maxThumbsSize() *
                                       ThumbnailSize::maxThumbsSize() * 4);
 
-    d->thumbnailPixmapCache.setMaxCost(numberOfQPixmaps *
+	    d->thumbnailPixmapCache.setMaxCost(numberOfQPixmaps *
                                        ThumbnailSize::maxThumbsSize() *
                                        ThumbnailSize::maxThumbsSize() * QPixmap::defaultDepth() / 8);
+	}
 }
 
 void LoadingCache::setFileWatch(LoadingCacheFileWatch* const watch)
