--- digikam-5.9.0/core/libs/threadimageio/orig.loadingcache.cpp	2018-03-19 10:41:59.000000000 +0100
+++ digikam-5.9.0/core/libs/threadimageio/loadingcache.cpp	2018-03-25 17:02:46.000000000 +0200
@@ -26,6 +26,7 @@
 // Qt includes
 
 #include <QCoreApplication>
+#include <QGuiApplication>
 #include <QEvent>
 #include <QCache>
 #include <QHash>
@@ -303,8 +304,10 @@
 
 void LoadingCache::setThumbnailCacheSize(int numberOfQImages, int numberOfQPixmaps)
 {
-    d->thumbnailImageCache.setMaxCost(numberOfQImages * ThumbnailSize::maxThumbsSize() * ThumbnailSize::maxThumbsSize() * 4);
-    d->thumbnailPixmapCache.setMaxCost(numberOfQPixmaps * ThumbnailSize::maxThumbsSize() * ThumbnailSize::maxThumbsSize() * QPixmap::defaultDepth() / 8);
+    if (QGuiApplication::primaryScreen()) {
+        d->thumbnailImageCache.setMaxCost(numberOfQImages * ThumbnailSize::maxThumbsSize() * ThumbnailSize::maxThumbsSize() * 4);
+        d->thumbnailPixmapCache.setMaxCost(numberOfQPixmaps * ThumbnailSize::maxThumbsSize() * ThumbnailSize::maxThumbsSize() * QPixmap::defaultDepth() / 8);
+    }
 }
 
 void LoadingCache::setFileWatch(LoadingCacheFileWatch* watch)
