--- digikam-5.4.0/orig.CMakeLists.txt	2017-01-08 23:08:44.000000000 +0100
+++ digikam-5.4.0/CMakeLists.txt	2017-05-14 02:26:22.000000000 +0200
@@ -31,6 +31,8 @@
 include(GenerateExportHeader)
 include(FeatureSummary)
 
+kde_enable_named_operators() ##string(REPLACE "-fno-operator-names" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
+
 # COMPILATION OPTIONS MANAGEMENT ###########################################################################################################
 
 # doc and translations options
