diff --git a/core/libs/dplugins/webservices/o2/src/o2.cpp b/core/libs/dplugins/webservices/o2/src/o2.cpp
index 7a3096a0ecccd9e276002f881ad3bcd6922a14d3..51c609fa588bd5c4d958578be00fa738020e49a5 100644
--- a/core/libs/dplugins/webservices/o2/src/o2.cpp
+++ b/core/libs/dplugins/webservices/o2/src/o2.cpp
@@ -185,7 +185,7 @@ void O2::link() {
         } else {
             // Start listening to authentication replies
             if (!replyServer()->isListening()) {
-                if (replyServer()->listen(QHostAddress::Any, localPort_)) {
+                if (replyServer()->listen(QHostAddress::Any, localPort())) {
                     qDebug() << "O2::link: Reply server listening on port" << localPort();
                 } else {
                     qWarning() << "O2::link: Reply server failed to start listening on port" << localPort();
