diff --git a/core/app/utils/digikam_globals.cpp b/core/app/utils/digikam_globals.cpp
index c4c254e0439cd7b8234cf995d19829353880b4fa..e49f378cc8a8fb41545365f18a8221e7f9d7d084 100644
--- a/core/app/utils/digikam_globals.cpp
+++ b/core/app/utils/digikam_globals.cpp
@@ -36,6 +36,7 @@
 #include <QStandardPaths>
 #include <QLibrary>
 #include <QSysInfo>
+#include <QDir>
 
 // KDE includes
 
@@ -311,7 +312,13 @@ QString toolButtonStyleSheet()
 
 QString macOSBundlePrefix()
 {
-    return QString::fromUtf8("/Applications/digiKam.org/digikam.app/Contents/");
+    // /path/to/digikam.app/Contents/MacOS/digikam
+    auto appdir = QDir(QCoreApplication::applicationFilePath());
+    // /path/to/digikam.app/Contents/MacOS
+    appdir.cdUp();
+    // /path/to/digikam.app/Contents
+    appdir.cdUp();
+    return appdir.canonicalPath() + QString::fromUtf8("/");
 }
 
 } // namespace Digikam
diff --git a/core/libs/database/server/databaseserver.cpp b/core/libs/database/server/databaseserver.cpp
index 6ee0ebd90e46918acaa7b93ce5041c0492cb23d0..70afa94ffd349268ca5b4f0a71caef93450a69f3 100644
--- a/core/libs/database/server/databaseserver.cpp
+++ b/core/libs/database/server/databaseserver.cpp
@@ -480,12 +480,13 @@ DatabaseServerError DatabaseServer::createMysqlFiles() const
 
 #endif
 
-#ifdef Q_OS_MACOS
-
-        mysqlInitCmdArgs << QDir::toNativeSeparators(QString::fromLatin1("--basedir=%1")
-            .arg(macOSBundlePrefix()));
-
-#endif
+// NOT FOR MACPORTS!
+// #ifdef Q_OS_MACOS
+// 
+//         mysqlInitCmdArgs << QDir::toNativeSeparators(QString::fromLatin1("--basedir=%1")
+//             .arg(macOSBundlePrefix()));
+// 
+// #endif
 
         mysqlInitCmdArgs << QDir::toNativeSeparators(QString::fromLatin1("--datadir=%1").arg(d->dataDir));
 
@@ -518,12 +519,13 @@ DatabaseServerError DatabaseServer::startMysqlServer()
     mysqldCmdArgs << QDir::toNativeSeparators(QString::fromLatin1("--defaults-file=%1").arg(d->actualConfig))
                   << QDir::toNativeSeparators(QString::fromLatin1("--datadir=%1").arg(d->dataDir));
 
-#ifdef Q_OS_MACOS
-
-    mysqldCmdArgs << QDir::toNativeSeparators(QString::fromLatin1("--basedir=%1")
-        .arg(macOSBundlePrefix()));
-
-#endif
+// NOT FOR MACPORTS!
+// #ifdef Q_OS_MACOS
+// 
+//     mysqldCmdArgs << QDir::toNativeSeparators(QString::fromLatin1("--basedir=%1")
+//         .arg(macOSBundlePrefix()));
+// 
+// #endif
 
 #ifdef Q_OS_WIN
 
diff --git a/core/libs/dimg/filters/icc/iccprofile.cpp b/core/libs/dimg/filters/icc/iccprofile.cpp
index 4789f4c..259cef5 100644
--- a/core/libs/dimg/filters/icc/iccprofile.cpp
+++ b/core/libs/dimg/filters/icc/iccprofile.cpp
@@ -561,9 +561,9 @@ QStringList IccProfile::defaultSearchPaths()
         dataDirs << QLatin1String("/usr/share");
     }
 
-    if (!dataDirs.contains(QLatin1String("/usr/local/share")))
+    if (!dataDirs.contains(QLatin1String("@PREFIX@/share")))
     {
-        dataDirs << QLatin1String("/usr/local/share");
+        dataDirs << QLatin1String("@PREFIX@/share");
     }
 
     foreach (const QString& dataDir, dataDirs)
diff --git a/core/utilities/setup/setupicc.cpp b/core/utilities/setup/setupicc.cpp
index 0ac1581..cd35238 100644
--- a/core/utilities/setup/setupicc.cpp
+++ b/core/utilities/setup/setupicc.cpp
@@ -936,6 +936,7 @@ void SetupICC::slotShowDefaultSearchPaths()
     QString text = i18n("On Linux, the default search paths include "
                         "<ul>"
                         "<li>/usr/share/color/icc</li>"
+                        "<li>@PREFIX@/share/color/icc</li>"
                         "<li>/usr/local/share/color/icc</li>"
                         "<li>~/.local/share/color/icc/</li>"
                         "<li>~/.local/share/icc/</li>"
