--- a/core/libs/settings/applicationsettings.cpp	2018-02-05 18:04:32.669053537 +0100
+++ b/core/libs/settings/applicationsettings.cpp	2018-02-05 18:08:20.169967734 +0100
@@ -236,7 +236,7 @@
     d->showTrashDeleteDialog             = group.readEntry(d->configShowTrashDeleteDialogEntry,                       true);
     d->showPermanentDeleteDialog         = group.readEntry(d->configShowPermanentDeleteDialogEntry,                   true);
     d->sidebarApplyDirectly              = group.readEntry(d->configApplySidebarChangesDirectlyEntry,                 false);
-    d->useNativeFileDialog               = group.readEntry(d->configUseNativeFileDialogEntry,                         false);
+    d->useNativeFileDialog               = group.readEntry(d->configUseNativeFileDialogEntry,                         true);
     d->drawFramesToGrouped               = group.readEntry(d->configDrawFramesToGroupedEntry,                         true);
     d->scrollItemToCenter                = group.readEntry(d->configScrollItemToCenterEntry,                          false);
     d->showOnlyPersonTagsInPeopleSidebar = group.readEntry(d->configShowOnlyPersonTagsInPeopleSidebarEntry,           true);
