diff --git a/core/libs/video/player/mediaplayerview.cpp b/core/libs/video/player/mediaplayerview.cpp
index e347398444d3571683437cef712fbf2bdf8a1895..05b923ac55924bd7a4899dab3486d470a361b51f 100644
--- a/core/libs/video/player/mediaplayerview.cpp
+++ b/core/libs/video/player/mediaplayerview.cpp
@@ -343,7 +343,7 @@ void MediaPlayerView::slotPlayerStateChanged(QtAV::AVPlayer::State state)
     if (state == QtAV::AVPlayer::PlayingState)
     {
         int rotate = 0;
-#if QTAV_VERSION > QTAV_VERSION_CHK(1, 12, 0)
+#if QTAV_VERSION >= QTAV_VERSION_CHK(1, 12, 0)
         // fix wrong rotation from QtAV git/master
         rotate     = d->player->statistics().video_only.rotate;
 #endif
diff --git a/core/libs/video/player/slidevideo.cpp b/core/libs/video/player/slidevideo.cpp
index 395f755f34c46aa1f0564e5ade6ec84c7056d53f..e3390163046d3853bc0068807f196fd4ecd2aa70 100644
--- a/core/libs/video/player/slidevideo.cpp
+++ b/core/libs/video/player/slidevideo.cpp
@@ -249,7 +249,7 @@ void SlideVideo::slotPlayerStateChanged(QtAV::AVPlayer::State state)
     if (state == QtAV::AVPlayer::PlayingState)
     {
         int rotate = 0;
-#if QTAV_VERSION > QTAV_VERSION_CHK(1, 12, 0)
+#if QTAV_VERSION >= QTAV_VERSION_CHK(1, 12, 0)
         // fix wrong rotation from QtAV git/master
         rotate     = d->player->statistics().video_only.rotate;
 #endif
