diff --git a/core/libs/settings/applicationsettings.cpp b/core/libs/settings/applicationsettings.cpp
index d80e836c21d69e4fb8f72c4244e86d2fad023074..8d8d62a8d1bef45a2c1b61070e9b9a467cbf11f5 100644
--- a/core/libs/settings/applicationsettings.cpp
+++ b/core/libs/settings/applicationsettings.cpp
@@ -241,7 +241,7 @@ void ApplicationSettings::readSettings()
 #ifdef Q_OS_OSX
     d->useNativeFileDialog               = group.readEntry(d->configUseNativeFileDialogEntry,                         true);
 #else
-    d->useNativeFileDialog               = group.readEntry(d->configUseNativeFileDialogEntry,                         false);
+    d->useNativeFileDialog               = group.readEntry(d->configUseNativeFileDialogEntry,                         true);
 #endif
 
     d->drawFramesToGrouped               = group.readEntry(d->configDrawFramesToGroupedEntry,                         true);
