diff --git src/sycoca/kbuildsycoca.cpp src/sycoca/kbuildsycoca.cpp
index 5e63907..7767b8e 100644
--- src/sycoca/kbuildsycoca.cpp
+++ src/sycoca/kbuildsycoca.cpp
@@ -491,7 +491,9 @@ bool KBuildSycoca::recreate(bool incremental)
         // These directories may have been created with 0700 permission
         // better delete them if they are empty
         QString appsDir = QStandardPaths::writableLocation(QStandardPaths::ApplicationsLocation);
-        QDir().remove(appsDir);
+        if (appsDir != QStringLiteral("/Applications") && QDir(appsDir).count() == 0) {
+            QDir().remove(appsDir);
+        }
         // was doing the same with servicetypes, but I don't think any of these gets created-by-mistake anymore.
     }
     if (d->m_sycocaStrategy == KSycocaPrivate::StrategyMemFile) {
