--- src/sycoca/orig.vfolder_menu.cpp	2015-11-08 13:35:05.000000000 +0100
+++ src/sycoca/vfolder_menu.cpp	2015-11-24 00:54:37.000000000 +0100
@@ -741,7 +741,10 @@
     QString result;
 
     QString xdgMenuPrefix = QString::fromLocal8Bit(qgetenv("XDG_MENU_PREFIX"));
-    if (!xdgMenuPrefix.isEmpty()) {
+    // enforce kf5 prefix unless XDG_MENU_PREFIX is set (based on LaunchPad ticket #297195)
+    if (xdgMenuPrefix.isEmpty()) {
+        xdgMenuPrefix = QStringLiteral("@APPMENUPREFIX@");
+
         QFileInfo fileInfo(fileName);
 
         QString fileNameOnly = fileInfo.fileName();
