--- src/sycoca/orig.kbuildservicefactory.cpp	2016-05-07 17:23:20.000000000 +0200
+++ src/sycoca/kbuildservicefactory.cpp	2016-05-30 22:41:05.000000000 +0200
@@ -352,7 +352,7 @@
                 entry->setServiceOffersOffset(offersOffset);
                 offersOffset += offerEntrySize * numOffers;
             } else {
-                qWarning() << "Not found:" << stName;
+//                 qWarning() << "Not found:" << stName;
             }
         }
     }
@@ -384,7 +384,7 @@
             }
         }
         if (offset == -1) {
-            qDebug() << "Didn't find servicetype or mimetype" << stName;
+//             qDebug() << "Didn't find servicetype or mimetype" << stName;
             continue;
         }
 
