diff --git src/sycoca/kmimeassociations.cpp src/sycoca/kmimeassociations.cpp
index 25ce3fe..c376fb2 100644
--- src/sycoca/kmimeassociations.cpp
+++ src/sycoca/kmimeassociations.cpp
@@ -59,8 +59,12 @@ void KMimeAssociations::parseAllMimeAppsList()
     }
     mimeappsFileNames.append(QStringLiteral("mimeapps.list"));
     // list the dirs in the order of the spec (XDG_CONFIG_HOME, XDG_CONFIG_DIRS, XDG_DATA_HOME, XDG_DATA_DIRS)
-    const QStringList mimeappsDirs = QStandardPaths::standardLocations(QStandardPaths::GenericConfigLocation)
+    QStringList mimeappsDirs = QStandardPaths::standardLocations(QStandardPaths::GenericConfigLocation)
                                     + QStandardPaths::standardLocations(QStandardPaths::ApplicationsLocation);
+    mimeappsDirs.removeDuplicates();
+#ifdef Q_OS_MACOS
+    mimeappsDirs.removeAll(QLatin1String("/Applications"));
+#endif
     QStringList mimeappsFiles;
     // collect existing files
     foreach (const QString &dir, mimeappsDirs) {
diff --git src/sycoca/vfolder_menu.cpp src/sycoca/vfolder_menu.cpp
index 5acbf8a..56173c0 100644
--- src/sycoca/vfolder_menu.cpp
+++ src/sycoca/vfolder_menu.cpp
@@ -793,6 +793,10 @@ void
 VFolderMenu::initDirs()
 {
     m_defaultAppDirs = QStandardPaths::standardLocations(QStandardPaths::ApplicationsLocation);
+#ifdef Q_OS_MACOS
+    m_defaultAppDirs.removeDuplicates();
+    m_defaultAppDirs.removeAll(QStringLiteral("/Applications"));
+#endif
     m_defaultDirectoryDirs = QStandardPaths::locateAll(QStandardPaths::GenericDataLocation, QStringLiteral("desktop-directories"), QStandardPaths::LocateDirectory);
 }
 
