diff --git orig.qjsc.c qjsc.c
index f9e1928..e133769 100644
--- orig.qjsc.c
+++ qjsc.c
@@ -427,13 +439,27 @@ static int output_executable(const char *out_filename, const char *cfilename,
     bn_suffix = "";
 
     arg = argv;
-    *arg++ = CONFIG_CC;
+    char *qjscc = getenv("QJSCC");
+    if (qjscc) {
+        qjscc = strdup(qjscc);
+    }
+    // check a 2nd time, now if strdup() succeeded. If not, fall back on the default silently
+    if (qjscc) {
+        *arg++ = qjscc = strdup(qjscc);
+    } else {
+        *arg++ = CONFIG_CC;
+    }
     *arg++ = "-O2";
 #ifdef CONFIG_LTO
     if (use_lto) {
         *arg++ = "-flto";
         lto_suffix = ".lto";
     }
+#endif
+#ifdef PGO_GEN
+    if (PGO_GEN && strlen(PGO_GEN) > 0) {
+        *arg++ = PGO_GEN;
+    }
 #endif
     /* XXX: use the executable path to find the includes files and
        libraries */
@@ -462,6 +488,9 @@ static int output_executable(const char *out_filename, const char *cfilename,
 
     ret = exec_cmd((char **)argv);
     unlink(cfilename);
+    if (qjscc) {
+	    free(qjscc);
+    }
     return ret;
 }
 #else
