diff --git orig.Makefile Makefile
index 3b1c745..751bea2 100644
--- orig.Makefile
+++ Makefile
@@ -160,7 +160,8 @@ endif
 CFLAGS+=$(DEFINES)
 CFLAGS_DEBUG=$(CFLAGS) -O0
 CFLAGS_SMALL=$(CFLAGS) -Os
-CFLAGS_OPT=$(CFLAGS) -O2
+CFLAGS_OPT=$(CFLAGS)
+CFLAGS_LTO=-flto
 CFLAGS_NOLTO:=$(CFLAGS_OPT)
 ifdef CONFIG_COSMO
 LDFLAGS+=-s # better to strip by default
@@ -168,9 +169,9 @@ else
 LDFLAGS+=-g
 endif
 ifdef CONFIG_LTO
-CFLAGS_SMALL+=-flto
-CFLAGS_OPT+=-flto
-LDFLAGS+=-flto
+CFLAGS_SMALL+=$(CFLAGS_LTO)
+CFLAGS_OPT+=$(CFLAGS_LTO)
+LDFLAGS+=$(CFLAGS_LTO)
 endif
 ifdef CONFIG_PROFILE
 CFLAGS+=-p
