diff --git orig.qjsc.c qjsc.c
index 96637c8..452c782 100644
--- orig.qjsc.c
+++ qjsc.c
@@ -427,7 +427,16 @@ static int output_executable(const char *out_filename, const char *cfilename,
     bn_suffix = "";
 
     arg = argv;
-    *arg++ = CONFIG_CC;
+    char *qjscc = getenv("QJSCC");
+    if (qjscc) {
+        qjscc = strdup(qjscc);
+    }
+    // check a 2nd time, now if strdup() succeeded. If not, fall back on the default silently
+    if (qjscc) {
+        *arg++ = qjscc = strdup(qjscc);
+    } else {
+        *arg++ = CONFIG_CC;
+    }
     *arg++ = "-O2";
 #ifdef CONFIG_LTO
     if (use_lto) {
@@ -462,6 +471,9 @@ static int output_executable(const char *out_filename, const char *cfilename,
 
     ret = exec_cmd((char **)argv);
     unlink(cfilename);
+    if (qjscc) {
+	    free(qjscc);
+    }
     return ret;
 }
 #else
