--- cairo-1.0.gir.orig	2022-07-11 14:41:00.000000000 +0200
+++ cairo-1.0.gir	2022-07-11 14:52:04.986996795 +0200
@@ -6,7 +6,7 @@
   <package name="cairo-gobject"/>
   <c:include name="cairo-gobject.h"/>
   <namespace name="cairo" version="1.0"
-	     shared-library="libcairo-gobject.so.2"
+	     shared-library="@PREFIX@/lib/libcairo-gobject.so.2"
 	     c:identifier-prefixes="cairo"
 	     c:symbol-prefixes="cairo">
     <record name="Context" c:type="cairo_t" foreign="1"
--- GIRepository-2.0.gir.orig	2022-07-11 14:43:49.495503238 +0200
+++ GIRepository-2.0.gir	2022-07-11 14:52:10.837132434 +0200
@@ -11,7 +11,7 @@
   <c:include name="girepository.h"/>
   <namespace name="GIRepository"
              version="2.0"
-             shared-library="libgirepository-1.0.so.1"
+             shared-library="@PREFIX@/lib/libgirepository-1.0.so.1"
              c:identifier-prefixes="GI"
              c:symbol-prefixes="g,gi">
     <alias name="ArgInfo" c:type="GIArgInfo">
