diff --git qtbase/src/corelib/kernel/qobject.cpp qtbase/src/corelib/kernel/qobject.cpp
index e588808..f0e2d7c 100644
--- qtbase/src/corelib/kernel/qobject.cpp
+++ qtbase/src/corelib/kernel/qobject.cpp
@@ -1605,7 +1605,7 @@ int QObject::startTimer(int interval, Qt::TimerType timerType)
     Q_D(QObject);
 
     if (interval < 0) {
-        qWarning("QObject::startTimer: Timers cannot have negative intervals");
+        qWarning() << "QObject::startTimer: Timer" << this << "cannot have a negative interval";
         return 0;
     }
 
