diff --git src/kdeinit/CMakeLists.txt src/kdeinit/CMakeLists.txt
index f94db71..58f4c98 100644
--- src/kdeinit/CMakeLists.txt
+++ src/kdeinit/CMakeLists.txt
@@ -46,7 +46,7 @@ if (NOT WIN32)
   add_executable(kdeinit5_shutdown ../wrapper.cpp)
   ecm_mark_nongui_executable(kdeinit5_shutdown)
 
-  target_link_libraries(kdeinit5_shutdown  ${KINIT_SOCKET_LIBRARY} Qt5::Core)
+  target_link_libraries(kdeinit5_shutdown  PUBLIC ${KINIT_SOCKET_LIBRARY} Qt5::Core PRIVATE Qt5::QspXDG)
 
   install(TARGETS kdeinit5_shutdown ${KF5_INSTALL_TARGETS_DEFAULT_ARGS} )
 
@@ -54,7 +54,7 @@ if (NOT WIN32)
   add_executable(kdeinit5_wrapper ../wrapper.cpp)
   ecm_mark_nongui_executable(kdeinit5_wrapper)
 
-  target_link_libraries(kdeinit5_wrapper  ${KINIT_SOCKET_LIBRARY} Qt5::Core)
+  target_link_libraries(kdeinit5_wrapper  PUBLIC ${KINIT_SOCKET_LIBRARY} Qt5::Core PRIVATE Qt5::QspXDG)
 
   if (NOT X11_FOUND)
     target_compile_definitions(kdeinit5_wrapper PRIVATE -DNO_DISPLAY)
diff --git src/kshell/CMakeLists.txt src/kshell/CMakeLists.txt
index 5c2e63d..e43e84a 100644
--- src/kshell/CMakeLists.txt
+++ src/kshell/CMakeLists.txt
@@ -1,6 +1,6 @@
 add_executable(kshell5 shell.cpp)
 ecm_mark_nongui_executable(kshell5)
 
-target_link_libraries(kshell5 ${KINIT_SOCKET_LIBRARY} Qt5::Core)
+target_link_libraries(kshell5 PUBLIC ${KINIT_SOCKET_LIBRARY} Qt5::Core PRIVATE Qt5::QspXDG)
 
 install(TARGETS kshell5 ${KF5_INSTALL_TARGETS_DEFAULT_ARGS} )
diff --git src/kwrapper/CMakeLists.txt src/kwrapper/CMakeLists.txt
index 79cb70e..fc20aa1 100644
--- src/kwrapper/CMakeLists.txt
+++ src/kwrapper/CMakeLists.txt
@@ -7,5 +7,5 @@ endif ()
 add_executable(kwrapper5 ${kwrapper_SRCS})
 ecm_mark_nongui_executable(kwrapper5)
 
-target_link_libraries(kwrapper5 ${KINIT_SOCKET_LIBRARY} Qt5::Core)
+target_link_libraries(kwrapper5 PUBLIC ${KINIT_SOCKET_LIBRARY} Qt5::Core PRIVATE Qt5::QspXDG)
 install(TARGETS kwrapper5 ${KF5_INSTALL_TARGETS_DEFAULT_ARGS} )
