--- src/ioslaves/http/kcookiejar/orig.kcookiewin.cpp	2015-12-20 19:16:02.000000000 +0100
+++ src/ioslaves/http/kcookiejar/kcookiewin.cpp	2015-12-20 19:16:17.000000000 +0100
@@ -64,7 +64,7 @@
     setModal(true);
     setObjectName(QStringLiteral("cookiealert"));
     setWindowTitle(i18n("Cookie Alert"));
-    setWindowIcon(QIcon::fromTheme(QStringLiteral("preferences-web-browser-cookies")));
+    setWindowIcon(QIcon::fromTheme(QStringLiteral("preferences-web-browser-cookies"), windowIcon()));
     // all cookies in the list should have the same window at this time, so let's take the first
     if (cookieList.first().windowIds().count() > 0) {
         KWindowSystem::setMainWindow(this, cookieList.first().windowIds().first());
--- src/kpasswdserver/orig.kpasswdserver.cpp	2015-12-06 22:30:08.000000000 +0100
+++ src/kpasswdserver/kpasswdserver.cpp	2015-12-20 19:42:35.000000000 +0100
@@ -571,7 +571,7 @@
             connect(dlg, SIGNAL(finished(int)), this, SLOT(retryDialogDone(int)));
             connect(this, SIGNAL(destroyed(QObject*)), dlg, SLOT(deleteLater()));
             dlg->setWindowTitle(i18n("Retry Authentication"));
-            dlg->setWindowIcon(QIcon::fromTheme(QStringLiteral("dialog-password")));
+            dlg->setWindowIcon(QIcon::fromTheme(QStringLiteral("dialog-password"), dlg->windowIcon()));
             dlg->setObjectName(QStringLiteral("warningOKCancel"));
             QDialogButtonBox* buttonBox = new QDialogButtonBox(QDialogButtonBox::Yes|QDialogButtonBox::Cancel);
             buttonBox->button(QDialogButtonBox::Yes)->setText(i18nc("@action:button filter-continue", "Retry"));
