diff --git src/core/CMakeLists.txt src/core/CMakeLists.txt
index 3d8d4cb..8f29eaa 100644
--- src/core/CMakeLists.txt
+++ src/core/CMakeLists.txt
@@ -1,5 +1,6 @@
 
 find_package(Qt5Core ${REQUIRED_QT_VERSION} REQUIRED NO_MODULE)
+find_package(Qt5QspXDG ${REQUIRED_QT_VERSION} REQUIRED NO_MODULE)
 
 set(libkconfigcore_SRCS
    kconfig.cpp
@@ -26,7 +27,7 @@ target_compile_definitions(KF5ConfigCore
 
 target_include_directories(KF5ConfigCore INTERFACE "$<INSTALL_INTERFACE:${KDE_INSTALL_INCLUDEDIR_KF5}/KConfigCore>")
 
-target_link_libraries(KF5ConfigCore PUBLIC Qt5::Core)
+target_link_libraries(KF5ConfigCore PUBLIC Qt5::Core PRIVATE Qt5::QspXDG)
 if(WIN32)
     target_link_libraries(KF5ConfigCore PRIVATE ${KDEWIN_LIBRARIES})
 endif()
diff --git src/kconfig_compiler/CMakeLists.txt src/kconfig_compiler/CMakeLists.txt
index 004a649..747b53e 100644
--- src/kconfig_compiler/CMakeLists.txt
+++ src/kconfig_compiler/CMakeLists.txt
@@ -17,8 +17,9 @@ else()
 endif()
 
 find_package(Qt5Xml ${REQUIRED_QT_VERSION} REQUIRED NO_MODULE)
+find_package(Qt5QspXDG ${REQUIRED_QT_VERSION} REQUIRED NO_MODULE)
 
-target_link_libraries(kconfig_compiler Qt5::Xml)
+target_link_libraries(kconfig_compiler PUBLIC Qt5::Xml PRIVATE Qt5::QspXDG)
 
 ecm_mark_nongui_executable(kconfig_compiler)
 
