diff --git src/plasma/CMakeLists.txt src/plasma/CMakeLists.txt
index 73f308c..17324e8 100644
--- src/plasma/CMakeLists.txt
+++ src/plasma/CMakeLists.txt
@@ -115,6 +115,7 @@ PRIVATE
     Qt5::Sql
     Qt5::Svg
     Qt5::DBus
+    Qt5::QspXDG
     KF5::Archive
     KF5::GuiAddons #kimagecache
     KF5::I18n
diff --git src/plasmaquick/CMakeLists.txt src/plasmaquick/CMakeLists.txt
index 1ddc611..c6e0baf 100644
--- src/plasmaquick/CMakeLists.txt
+++ src/plasmaquick/CMakeLists.txt
@@ -51,6 +51,7 @@ target_link_libraries(KF5PlasmaQuick
         KF5::XmlGui
         KF5::Declarative
         KF5::QuickAddons
+        Qt5::QspXDG
 )
 
 if(HAVE_X11)
diff --git src/declarativeimports/calendar/CMakeLists.txt src/declarativeimports/calendar/CMakeLists.txt
index 40ead91..298952c 100644
--- src/declarativeimports/calendar/orig.CMakeLists.txt	2015-12-15 22:19:34.000000000 +0100
+++ src/declarativeimports/calendar/CMakeLists.txt	2015-12-15 22:22:06.000000000 +0100
@@ -13,13 +13,17 @@
 
 add_library(calendarplugin SHARED ${calendar_SRCS})
 
+# calendarplugin doesn't depend directly on another KF5 fw, so add the activator
 target_link_libraries(calendarplugin
+PUBLIC
     Qt5::Core
     Qt5::Quick
     Qt5::Qml
     Qt5::Gui
     KF5::I18n
     KF5::CalendarEvents
+PRIVATE
+    Qt5::QspXDG
 )
 
 install(TARGETS calendarplugin DESTINATION ${KDE_INSTALL_QMLDIR}/org/kde/plasma/calendar)
