--- src/orig.kmessagebox.cpp	2015-12-08 17:32:31.000000000 +0100
+++ src/kmessagebox.cpp	2015-12-08 17:37:43.000000000 +0100
@@ -126,6 +126,7 @@
 // This method has been copied from KWindowSystem to avoid depending on it
 static void setMainWindow(QWidget *subWidget, WId mainWindowId)
 {
+#ifndef Q_OS_OSX
     // Set the WA_NativeWindow attribute to force the creation of the QWindow.
     // Without this QWidget::windowHandle() returns 0.
     subWidget->setAttribute(Qt::WA_NativeWindow, true);
@@ -136,6 +137,7 @@
     // mainWindow is not the child of any object, so make sure it gets deleted at some point
     QObject::connect(subWidget, &QObject::destroyed, mainWindow, &QObject::deleteLater);
     subWindow->setTransientParent(mainWindow);
+#endif
 }
 
 /**
