--- src/orig.CMakeLists.txt	2015-11-04 15:56:57.000000000 +0100
+++ src/CMakeLists.txt	2015-11-04 15:59:12.000000000 +0100
@@ -50,7 +50,7 @@
 )
 add_library(ktranscript MODULE ${ktranscript_LIB_SRCS})
 generate_export_header(ktranscript BASE_NAME KTranscript)
-target_link_libraries(ktranscript PRIVATE Qt5::Script Qt5::Core)
+target_link_libraries(ktranscript PRIVATE Qt5::Script Qt5::Core Qt5::QspXDG)
 
 install(TARGETS ktranscript DESTINATION ${KDE_INSTALL_PLUGINDIR}/kf5)
 
