diff --git src/klauncher/klauncher.cpp src/klauncher/klauncher.cpp
index 8b3d343..24421d2 100644
--- src/klauncher/klauncher.cpp
+++ src/klauncher/klauncher.cpp
@@ -551,6 +551,56 @@ static void appendLong(QByteArray &ba, long l)
     memcpy(ba.data() + sz, &l, sizeof(long));
 }
 
+bool QTimeoutProcess::startDetached(const QString &exe, const QStringList &args,
+                          const QString &cwd, qint64 *pid, int timeoutMs)
+{
+    bool ret = false;
+#ifdef Q_OS_OSX
+    QProcess *proc = new QTimeoutProcess;
+    connect(proc, SIGNAL(finished(int,QProcess::ExitStatus)), proc, SLOT(slotFinished(int,QProcess::ExitStatus)));
+    if (!cwd.isEmpty()) {
+        proc->setWorkingDirectory(cwd);
+    }
+    proc->start(exe, args);
+    if (proc->waitForStarted()) {
+        ret = true;
+        if (pid) {
+            *pid = proc->pid();
+        }
+        if (timeoutMs > 0) {
+            proc->startTimer(timeoutMs);
+        }
+    }
+#endif
+    return ret;
+}
+
+QTimeoutProcess::~QTimeoutProcess()
+{
+#ifdef Q_OS_OSX
+    if (state() == QProcess::Running) {
+        // prevent the parent dtor from yelling because the process still runs
+        kill();
+        waitForFinished(500);
+    }
+#endif
+}
+
+void QTimeoutProcess::timerEvent(QTimerEvent *)
+{
+#ifdef Q_OS_OSX
+    terminate();
+    deleteLater();
+#endif
+}
+
+void QTimeoutProcess::slotFinished(int exitCode, QProcess::ExitStatus exitStatus)
+{
+#ifdef Q_OS_OSX
+    deleteLater();
+#endif
+}
+
 void
 KLauncher::requestStart(KLaunchRequest *request)
 {
@@ -589,6 +639,27 @@ KLauncher::requestStart(KLaunchRequest *request)
 #endif
         QByteArray data((char *)&request->pid, sizeof(int));
         processRequestReturn(LAUNCHER_OK, data);
+#ifdef Q_OS_OSX
+        // we now know that 'process' started successfully, so we can start a background
+        // process that will wait a preconfigured time for the <executable> to launch
+        // and activate it (= bring to foreground). The only way to do this in combination
+        // with QProcess is to use AppleScript
+        QStringList osacommand;
+        osacommand << QStringLiteral("-e")
+            << QStringLiteral(
+                "delay 0.5\n"
+                "tell application \"System Events\"\n"
+                " repeat\n"
+                "  if (exists application process \"%1\") then\n"
+                "   tell application \"%1\" to activate\n"
+                "   exit repeat\n"
+                "  end if\n"
+                "  delay 0.2\n"
+                " end repeat\n"
+                "end tell").arg(QFileInfo(executable).baseName());
+        bool osaOK = QTimeoutProcess::startDetached(QStringLiteral("/usr/bin/osascript"), osacommand, QString(), Q_NULLPTR, 2000);
+        qCDebug(KLAUNCHER) << "osascript" << osacommand << ":" << osaOK;
+#endif
     }
     return;
 
diff --git src/klauncher/klauncher.h src/klauncher/klauncher.h
index e155f72..f0f3279 100644
--- src/klauncher/klauncher.h
+++ src/klauncher/klauncher.h
@@ -261,4 +261,20 @@ protected Q_SLOTS:
     void slotGotOutput();
     void slotFinished(int exitCode, QProcess::ExitStatus exitStatus);
 };
+
+// For Q_OS_OSX:
+class QTimeoutProcess : public QProcess
+{
+    Q_OBJECT
+public:
+    static bool startDetached(const QString &exe, const QStringList &args,
+                              const QString &cwd = QString(), qint64 *pid = 0,
+                              int timeoutMs = 0);
+    virtual ~QTimeoutProcess();
+protected:
+    void timerEvent(QTimerEvent *) Q_DECL_OVERRIDE;
+protected Q_SLOTS:
+    void slotFinished(int exitCode, QProcess::ExitStatus exitStatus);
+};
+
 #endif
