--- src/orig2CMakeLists.txt	2015-11-04 15:16:43.000000000 +0100
+++ src/CMakeLists.txt	2015-11-14 12:56:55.000000000 +0100
@@ -57,6 +57,10 @@
     # KAUTH_POLICY_GEN_SRCS has been generated from ConfigureChecks.cmake
     add_executable(kauth-policy-gen ${KAUTH_POLICY_GEN_SRCS})
     add_executable(KF5::kauth-policy-gen ALIAS kauth-policy-gen)
+    if (APPLE)
+        include(ECMMarkNonGuiExecutable)
+        ecm_mark_nongui_executable(kauth-policy-gen)
+    endif ()
 
     # KAUTH_POLICY_GEN_LIBRARIES has been generated from ConfigureChecks.cmake
     set(KAUTH_POLICY_GEN_LIBRARIES ${KAUTH_POLICY_GEN_LIBRARIES} Qt5::QspXDG)
