--- src/service/orig.Application.cpp	2015-12-06 22:15:31.000000000 +0100
+++ src/service/Application.cpp	2015-12-25 22:34:24.000000000 +0100
@@ -59,6 +59,9 @@
 #include "Debug.h"
 #include "common/dbus/common.h"
 
+#ifdef Q_OS_OSX
+#include <CoreFoundation/CoreFoundation.h>
+#endif
 
 namespace {
     QList<QThread *> s_moduleThreads;
@@ -307,6 +310,21 @@
 
 int main(int argc, char **argv)
 {
+#ifdef Q_OS_OSX
+    CFBundleRef mainBundle = CFBundleGetMainBundle();
+    if (mainBundle) {
+        // get the application's Info Dictionary. For app bundles this would live in the bundle's Info.plist,
+        // for regular executables it is obtained in another way.
+        CFMutableDictionaryRef infoDict = (CFMutableDictionaryRef) CFBundleGetInfoDictionary(mainBundle);
+        if (infoDict) {
+            // Add or set the "LSUIElement" key with/to value "1". This can simply be a CFString.
+            CFDictionarySetValue(infoDict, CFSTR("LSUIElement"), CFSTR("1"));
+            // That's it. We're now considered as an "agent" by the window server, and thus will have
+            // neither menubar nor presence in the Dock or App Switcher.
+        }
+    }
+#endif
+
     Application application(argc, argv);
     application.setApplicationName(QStringLiteral("ActivityManager"));
     application.setOrganizationDomain(QStringLiteral("kde.org"));
Exit 1
--- src/service/orig.CMakeLists.txt	2015-12-06 22:15:31.000000000 +0100
+++ src/service/CMakeLists.txt	2015-12-25 22:36:12.000000000 +0100
@@ -77,6 +77,9 @@
    KF5::Service
    KF5::WindowSystem
    )
+if (APPLE)
+   target_link_libraries (kactivitymanagerd "-framework CoreFoundation")
+endif ()
 
 ########### install application ###############
 
