diff --git src/kdeclarative/CMakeLists.txt src/kdeclarative/CMakeLists.txt
index ce12546..b93d9e4 100644
--- src/kdeclarative/CMakeLists.txt
+++ src/kdeclarative/CMakeLists.txt
@@ -22,6 +22,7 @@ target_link_libraries(KF5Declarative
     KF5::ConfigCore # KCoreConfigSkeleton, in ConfigPropertyMap
     Qt5::Qml
     KF5::Package
+    Qt5::QspXDG
   PRIVATE
     Qt5::Quick  #QQuickImageProvider
     KF5::I18n       #i18n bindings
diff --git src/kpackagelauncherqml/CMakeLists.txt src/kpackagelauncherqml/CMakeLists.txt
index 7744b77..b254ec0 100644
--- src/kpackagelauncherqml/CMakeLists.txt
+++ src/kpackagelauncherqml/CMakeLists.txt
@@ -12,6 +12,7 @@ target_link_libraries(kpackagelauncherqml
  Qt5::Gui
  Qt5::Quick
  Qt5::Widgets #needs a qapplication for the Desktop QtControlsStyle
+ Qt5::QspXDG
  KF5::Declarative
  KF5::I18n
  KF5::Package
diff --git src/qmlcontrols/draganddrop/CMakeLists.txt src/qmlcontrols/draganddrop/CMakeLists.txt
index e8127e4..598f54b 100644
--- src/qmlcontrols/draganddrop/CMakeLists.txt
+++ src/qmlcontrols/draganddrop/CMakeLists.txt
@@ -15,6 +15,7 @@ target_link_libraries(draganddropplugin
         Qt5::Quick
         Qt5::Qml
         Qt5::Gui
+	   Qt5::QspXDG
 )
 
 install(TARGETS draganddropplugin DESTINATION ${KDE_INSTALL_QMLDIR}/org/kde/draganddrop)
diff --git src/qmlcontrols/kcoreaddons/CMakeLists.txt src/qmlcontrols/kcoreaddons/CMakeLists.txt
index 3f77f2d..da7ca86 100644
--- src/qmlcontrols/kcoreaddons/CMakeLists.txt
+++ src/qmlcontrols/kcoreaddons/CMakeLists.txt
@@ -11,6 +11,7 @@ target_link_libraries(kcoreaddonsplugin
         Qt5::Core
         Qt5::Quick
         Qt5::Qml
+	   Qt5::QspXDG
         KF5::CoreAddons
 )
 
diff --git src/qmlcontrols/kioplugin/CMakeLists.txt src/qmlcontrols/kioplugin/CMakeLists.txt
index 7b258e0..2bfbcb0 100644
--- src/qmlcontrols/kioplugin/CMakeLists.txt
+++ src/qmlcontrols/kioplugin/CMakeLists.txt
@@ -9,6 +9,7 @@ add_library(kio SHARED ${kioplugin_SRCS})
 target_link_libraries(kio
         Qt5::Core
         Qt5::Qml
+	   Qt5::QspXDG
         KF5::KIOWidgets
 )
 
diff --git src/qmlcontrols/kquickcontrols/private/CMakeLists.txt src/qmlcontrols/kquickcontrols/private/CMakeLists.txt
index da355c1..bf69ea4 100644
--- src/qmlcontrols/kquickcontrols/private/CMakeLists.txt
+++ src/qmlcontrols/kquickcontrols/private/CMakeLists.txt
@@ -12,6 +12,7 @@ target_link_libraries(kquickcontrolsprivateplugin
         Qt5::Core
         Qt5::Quick
         Qt5::Qml
+	   Qt5::QspXDG
         KF5::I18n
         KF5::ConfigGui
         KF5::WidgetsAddons
diff --git src/qmlcontrols/kquickcontrolsaddons/CMakeLists.txt src/qmlcontrols/kquickcontrolsaddons/CMakeLists.txt
index be2d534..d48ad8a 100644
--- src/qmlcontrols/kquickcontrolsaddons/CMakeLists.txt
+++ src/qmlcontrols/kquickcontrolsaddons/CMakeLists.txt
@@ -27,6 +27,7 @@ target_link_libraries(kquickcontrolsaddonsplugin
         Qt5::Quick
         Qt5::Qml
         Qt5::Gui
+	   Qt5::QspXDG
         KF5::IconThemes
         KF5::QuickAddons
         ${KQUICKCONTROLSADDONS_EXTRA_LIBS})
diff --git src/qmlcontrols/kwindowsystemplugin/CMakeLists.txt src/qmlcontrols/kwindowsystemplugin/CMakeLists.txt
index ce0ea74..5f7d4c7 100644
--- src/qmlcontrols/kwindowsystemplugin/CMakeLists.txt
+++ src/qmlcontrols/kwindowsystemplugin/CMakeLists.txt
@@ -9,6 +9,7 @@ add_library(kwindowsystem SHARED ${kwindowsystemplugin_SRCS})
 target_link_libraries(kwindowsystem
         Qt5::Core
         Qt5::Qml
+	   Qt5::QspXDG
         KF5::WindowSystem
 )
 
diff --git src/quickaddons/CMakeLists.txt src/quickaddons/CMakeLists.txt
index 8dc431f..6554963 100644
--- src/quickaddons/CMakeLists.txt
+++ src/quickaddons/CMakeLists.txt
@@ -13,6 +13,7 @@ target_link_libraries(KF5QuickAddons
   PUBLIC
     Qt5::Qml
     KF5::CoreAddons
+    Qt5::QspXDG
   PRIVATE
     Qt5::Quick  #QQuickImageProvider
     KF5::ConfigCore # KCoreConfigSkeleton, in ConfigPropertyMap
