--- src/orig.notifybyaudio.cpp	2015-12-13 15:37:23.000000000 +0100
+++ src/notifybyaudio.cpp	2015-12-13 15:46:09.000000000 +0100
@@ -22,6 +22,7 @@
 #include "notifybyaudio.h"
 #include "debug_p.h"
 
+#include <QApplication>
 #include <QDateTime>
 #include <QFile>
 #include <QTextStream>
@@ -54,8 +55,9 @@
     }
     QString soundFilename = config->readEntry(QStringLiteral("Sound"));
     if (soundFilename.isEmpty()) {
-        qCWarning(LOG_KNOTIFICATIONS) << "Audio notification requested, but no sound file provided in notifyrc file, aborting audio notification";
-
+        qCWarning(LOG_KNOTIFICATIONS) << "Audio notification requested because" << notification->text()
+            << ", but no sound file provided in file" << config->configfile->name() << ", aborting audio notification";
+        QApplication::beep();
         finish(notification);
         return;
     }
@@ -65,8 +67,10 @@
         soundURL = QUrl::fromLocalFile(QStandardPaths::locate(QStandardPaths::GenericDataLocation, QStringLiteral("sounds/") + soundFilename));
 
         if (soundURL.isEmpty()) {
-            qCWarning(LOG_KNOTIFICATIONS) << "Audio notification requested, but sound file from notifyrc file was not found, aborting audio notification";
-
+            qCWarning(LOG_KNOTIFICATIONS) << "Audio notification requested" << notification->text()
+                << ", but sound file" << soundURL.fileName() << "from file" << config->configfile->name()
+                << "was not found, aborting audio notification";
+            QApplication::beep();
             finish(notification);
             return;
         }
