diff --git src/kdebugdialog/CMakeLists.txt src/kdebugdialog/CMakeLists.txt
index 0a1ac78..848e061 100644
--- src/kdebugdialog/CMakeLists.txt
+++ src/kdebugdialog/CMakeLists.txt
@@ -1,3 +1,4 @@
+include(ECMAddAppIcon)
 set(kdebugdialog_SRCS 
    main.cpp 
    kabstractdebugdialog.cpp 
@@ -6,6 +7,10 @@ set(kdebugdialog_SRCS
 
 ki18n_wrap_ui(kdebugdialog_SRCS kdebugdialog.ui)
 
+# Sets the icon on Windows and OSX
+file(GLOB ICONS_SRCS "${CMAKE_CURRENT_BINARY_DIR}/icons/*tools-report-bug.png")
+ecm_add_app_icon(kdebugdialog_SRCS ICONS ${ICONS_SRCS})
+
 add_executable(kdebugdialog5 ${kdebugdialog_SRCS})
 
 target_link_libraries(kdebugdialog5 Qt5::Widgets KF5::ItemViews KF5::KDELibs4Support)
