diff --git src/codecs/CMakeLists.txt src/codecs/CMakeLists.txt
index 9df7796..8c34fcc 100644
--- src/codecs/CMakeLists.txt
+++ src/codecs/CMakeLists.txt
@@ -11,5 +11,6 @@ add_library(KF5::BalooCodecs ALIAS KF5BalooCodecs)
 
 target_link_libraries(KF5BalooCodecs
     Qt5::Core
+    Qt5::QspXDG
     KF5::CoreAddons
 )
diff --git src/engine/CMakeLists.txt src/engine/CMakeLists.txt
index 96d63f4..519bb8b 100644
--- src/engine/CMakeLists.txt
+++ src/engine/CMakeLists.txt
@@ -31,6 +31,7 @@ add_library(KF5::BalooEngine ALIAS KF5BalooEngine)
 target_link_libraries(KF5BalooEngine
     PUBLIC
     Qt5::Core
+    Qt5::QspXDG
     KF5::CoreAddons
     ${LMDB_LIBRARIES}
     PRIVATE
diff --git src/file/CMakeLists.txt src/file/CMakeLists.txt
index 767e7dc..f15a263 100644
--- src/file/CMakeLists.txt
+++ src/file/CMakeLists.txt
@@ -45,6 +45,7 @@ set(file_static_lib_SRCS
 add_library(baloofilecommon STATIC ${file_static_lib_SRCS})
 target_link_libraries(baloofilecommon
     Qt5::DBus
+    Qt5::QspXDG
     KF5::I18n
     KF5::Solid
     KF5::FileMetaData
diff --git src/file/extractor/CMakeLists.txt src/file/extractor/CMakeLists.txt
index 6b58ce1..aaf487b 100644
--- src/file/extractor/CMakeLists.txt
+++ src/file/extractor/CMakeLists.txt
@@ -22,6 +22,7 @@ target_compile_definitions(baloo_file_extractor PRIVATE -DPROJECT_VERSION="${PRO
 target_link_libraries(baloo_file_extractor
   Qt5::DBus
   Qt5::Widgets
+  Qt5::QspXDG
   KF5::FileMetaData
   KF5::I18n
   KF5::ConfigCore
diff --git src/kioslaves/kded/CMakeLists.txt src/kioslaves/kded/CMakeLists.txt
index 804c944..d639d07 100644
--- src/kioslaves/kded/CMakeLists.txt
+++ src/kioslaves/kded/CMakeLists.txt
@@ -9,6 +9,7 @@ target_link_libraries(baloosearchmodule
     KF5::CoreAddons
     KF5::DBusAddons
     KF5::KIOCore
+    Qt5::QspXDG
 )
 
 install(TARGETS baloosearchmodule DESTINATION ${PLUGIN_INSTALL_DIR}/kf5/kded)
diff --git src/kioslaves/search/CMakeLists.txt src/kioslaves/search/CMakeLists.txt
index 4d21ff7..d614ce7 100644
--- src/kioslaves/search/CMakeLists.txt
+++ src/kioslaves/search/CMakeLists.txt
@@ -6,6 +6,7 @@ target_link_libraries(kio_baloosearch
   KF5::KIOCore
   KF5::I18n
   KF5::Baloo
+  Qt5::QspXDG
 )
 
 set_target_properties(kio_baloosearch PROPERTIES OUTPUT_NAME "baloosearch")
diff --git src/kioslaves/tags/CMakeLists.txt src/kioslaves/tags/CMakeLists.txt
index 17ed226..668b683 100644
--- src/kioslaves/tags/CMakeLists.txt
+++ src/kioslaves/tags/CMakeLists.txt
@@ -7,6 +7,7 @@ target_link_libraries(tags
   KF5::FileMetaData
   KF5::I18n
   KF5::Baloo
+  Qt5::QspXDG
 )
 
 install(TARGETS tags DESTINATION ${PLUGIN_INSTALL_DIR}/kf5/kio)
diff --git src/kioslaves/timeline/CMakeLists.txt src/kioslaves/timeline/CMakeLists.txt
index c1df10f..0d2d1b6 100644
--- src/kioslaves/timeline/CMakeLists.txt
+++ src/kioslaves/timeline/CMakeLists.txt
@@ -11,6 +11,7 @@ target_link_libraries(timeline
   KF5::KIOWidgets
   KF5::I18n
   KF5::Baloo
+  Qt5::QspXDG
 )
 
 install(TARGETS timeline DESTINATION ${PLUGIN_INSTALL_DIR}/kf5/kio)
diff --git src/lib/CMakeLists.txt src/lib/CMakeLists.txt
index 87f5b7c..a52f7f7 100644
--- src/lib/CMakeLists.txt
+++ src/lib/CMakeLists.txt
@@ -35,6 +35,7 @@ add_library(KF5::Baloo ALIAS KF5Baloo)
 target_link_libraries(KF5Baloo
     PUBLIC
     Qt5::Core
+    Qt5::QspXDG
     KF5::CoreAddons
     KF5::FileMetaData
 
diff --git src/qml/CMakeLists.txt src/qml/CMakeLists.txt
index 9798496..f05bb91 100644
--- src/qml/CMakeLists.txt
+++ src/qml/CMakeLists.txt
@@ -4,7 +4,7 @@ set(balooplugin_SRCS
     )
 
 add_library(balooplugin SHARED ${balooplugin_SRCS})
-target_link_libraries(balooplugin Qt5::Core Qt5::Qml KF5::Baloo)
+target_link_libraries(balooplugin Qt5::Core Qt5::Qml Qt5::QspXDG KF5::Baloo)
 install(TARGETS balooplugin DESTINATION ${QML_INSTALL_DIR}/org/kde/baloo/)
 
 install(FILES qmldir DESTINATION ${QML_INSTALL_DIR}/org/kde/baloo)
diff --git src/qml/experimental/CMakeLists.txt src/qml/experimental/CMakeLists.txt
index 2cea7f8..7e49f1d 100644
--- src/qml/experimental/CMakeLists.txt
+++ src/qml/experimental/CMakeLists.txt
@@ -21,6 +21,7 @@ target_link_libraries(baloomonitorplugin
     Qt5::Core
     Qt5::Qml
     Qt5::DBus
+    Qt5::QspXDG
     KF5::I18n
     KF5::CoreAddons
     PRIVATE
diff --git src/tools/balooctl/CMakeLists.txt src/tools/balooctl/CMakeLists.txt
index 8c51aed..091d1f2 100644
--- src/tools/balooctl/CMakeLists.txt
+++ src/tools/balooctl/CMakeLists.txt
@@ -24,6 +24,7 @@ target_compile_definitions(balooctl PRIVATE -DPROJECT_VERSION="${PROJECT_VERSION
 
 target_link_libraries(balooctl
     Qt5::DBus
+    Qt5::QspXDG
     KF5::CoreAddons
     KF5::ConfigCore
     KF5::I18n
diff --git src/tools/baloosearch/CMakeLists.txt src/tools/baloosearch/CMakeLists.txt
index 7fedc22..ba2c880 100644
--- src/tools/baloosearch/CMakeLists.txt
+++ src/tools/baloosearch/CMakeLists.txt
@@ -9,6 +9,7 @@ target_link_libraries(baloosearch
   KF5::I18n
   KF5::CoreAddons
   KF5::Baloo
+  Qt5::QspXDG
 )
 
 install(TARGETS baloosearch DESTINATION ${BIN_INSTALL_DIR})
diff --git src/tools/balooshow/CMakeLists.txt src/tools/balooshow/CMakeLists.txt
index c0b65b8..61d7421 100644
--- src/tools/balooshow/CMakeLists.txt
+++ src/tools/balooshow/CMakeLists.txt
@@ -11,6 +11,7 @@ target_link_libraries(balooshow
     KF5::FileMetaData
     KF5::CoreAddons
     KF5::I18n
+    Qt5::QspXDG
 )
 
 install(TARGETS balooshow DESTINATION ${BIN_INSTALL_DIR})
