diff --git src/CMakeLists.txt src/CMakeLists.txt
index e37e7d5..4b70389 100644
--- src/CMakeLists.txt
+++ src/CMakeLists.txt
@@ -237,6 +237,7 @@ else()
 endif ()
 
 if (APPLE)
+    add_definitions(-DBUNDLE_INSTALL_DIR="\\"${BUNDLE_INSTALL_DIR}\\"")
     set(libkdelibs4support_SRCS ${libkdelibs4support_SRCS}
                             kdecore/klocale_mac.cpp
                             )
diff --git src/kdecore/kstandarddirs.cpp src/kdecore/kstandarddirs.cpp
index 4161425..3f3a36a 100644
--- src/kdecore/kstandarddirs.cpp
+++ src/kdecore/kstandarddirs.cpp
@@ -1335,6 +1335,10 @@ QStringList KStandardDirs::systemPaths(const QString &pstr)
         exePaths << KShell::tildeExpand(tokens[ i ]);
     }
 
+#ifdef Q_OS_OSX
+    exePaths << QString::fromLatin1(BUNDLE_INSTALL_DIR);
+#endif
+
     return exePaths;
 }
 
