--- src/orig.notifybypopup.cpp	2015-11-08 13:30:47.000000000 +0100
+++ src/notifybypopup.cpp	2015-11-27 23:03:30.000000000 +0100
@@ -183,7 +183,11 @@
 #ifdef Q_WS_WIN
         startfdo = true;
 #else
-        if (qgetenv("KDE_FULL_SESSION").isEmpty()) {
+#ifndef Q_OS_OSX
+	   // ignore KDE_FULL_SESSION on OS X, where it might be set only to enable full theming
+        if (qgetenv("KDE_FULL_SESSION").isEmpty())
+#endif
+	   {
             QDBusMessage message = QDBusMessage::createMethodCall(QStringLiteral("org.freedesktop.DBus"),
                                                                   QStringLiteral("/org/freedesktop/DBus"),
                                                                   QStringLiteral("org.freedesktop.DBus"),
