diff --git src/api/KWallet/CMakeLists.txt src/api/KWallet/CMakeLists.txt
index e2a5631..64b426e 100644
--- src/api/KWallet/CMakeLists.txt
+++ src/api/KWallet/CMakeLists.txt
@@ -70,6 +70,7 @@ target_link_libraries(KF5Wallet PUBLIC  Qt5::Gui
                                         Qt5::Widgets
                                 PRIVATE KF5::WindowSystem # KWindowSystem::allowExternalProcessWindowActivation
                                         KF5::ConfigCore   # used to store the wallet to be used
+                                        Qt5::QspXDG
 )
 
 if(MAC_USE_OSXKEYCHAIN)
diff --git src/runtime/kwalletd/backend/CMakeLists.txt src/runtime/kwalletd/backend/CMakeLists.txt
index 20bd4d9..f9833d8 100644
--- src/runtime/kwalletd/backend/CMakeLists.txt
+++ src/runtime/kwalletd/backend/CMakeLists.txt
@@ -47,14 +47,14 @@ generate_export_header(kwalletbackend5)
 
 ecm_setup_version(${KF5_VERSION} VARIABLE_PREFIX KWALLETBACKEND SOVERSION 5)
 
-target_link_libraries(kwalletbackend5 Qt5::Widgets KF5::WidgetsAddons KF5::CoreAddons KF5::Notifications KF5::I18n ${LIBGCRYPT_LIBRARIES})
+target_link_libraries(kwalletbackend5 PUBLIC Qt5::Widgets KF5::WidgetsAddons KF5::CoreAddons KF5::Notifications KF5::I18n ${LIBGCRYPT_LIBRARIES} PRIVATE Qt5::QspXDG)
 if(KF5Gpgmepp_FOUND)
-target_link_libraries(kwalletbackend5 KF5::Gpgmepp)
+target_link_libraries(kwalletbackend5 PUBLIC KF5::Gpgmepp)
 endif(KF5Gpgmepp_FOUND)
 
 # link with advapi32 on windows
 if(WIN32 AND NOT WINCE)
-   target_link_libraries(kwalletbackend5 advapi32)
+   target_link_libraries(kwalletbackend5 PUBLIC advapi32)
 endif(WIN32 AND NOT WINCE)
 
 set_target_properties(kwalletbackend5 PROPERTIES VERSION ${KWALLETBACKEND_VERSION_STRING} SOVERSION ${KWALLETBACKEND_SOVERSION} )
