--- orig.CMakeLists.txt	2015-11-25 22:57:46.000000000 +0100
+++ CMakeLists.txt	2015-11-25 22:58:09.000000000 +0100
@@ -17,6 +17,7 @@
 
 set(REQUIRED_QT_VERSION 5.3.0)
 find_package(Qt5Core ${REQUIRED_QT_VERSION} REQUIRED NO_MODULE)
+find_package(Qt5QspXDG ${REQUIRED_QT_VERSION} REQUIRED NO_MODULE)
 find_package(Qt5Designer ${REQUIRED_QT_VERSION} NO_MODULE)
 set_package_properties(Qt5Designer PROPERTIES
    PURPOSE "Required to build the Qt Designer plugins"
