diff --git a/src/core/CMakeLists.txt b/src/core/CMakeLists.txt
index 3d8d4cb..ed60620 100644
--- a/src/core/CMakeLists.txt
+++ b/src/core/CMakeLists.txt
@@ -1,5 +1,6 @@
 
 find_package(Qt5Core ${REQUIRED_QT_VERSION} REQUIRED NO_MODULE)
+find_package(Qt5QspXDG ${REQUIRED_QT_VERSION} REQUIRED NO_MODULE)
 
 set(libkconfigcore_SRCS
    kconfig.cpp
diff --git a/src/kconfig_compiler/CMakeLists.txt b/src/kconfig_compiler/CMakeLists.txt
index 004a649..e16e78e 100644
--- a/src/kconfig_compiler/CMakeLists.txt
+++ b/src/kconfig_compiler/CMakeLists.txt
@@ -17,6 +17,7 @@ else()
 endif()
 
 find_package(Qt5Xml ${REQUIRED_QT_VERSION} REQUIRED NO_MODULE)
+find_package(Qt5QspXDG ${REQUIRED_QT_VERSION} REQUIRED NO_MODULE)
 
 target_link_libraries(kconfig_compiler Qt5::Xml)
 
