diff --git src/solid/CMakeLists.txt src/solid/CMakeLists.txt
index f2b43b2..64910a0 100644
--- src/solid/CMakeLists.txt
+++ src/solid/CMakeLists.txt
@@ -16,7 +16,7 @@ add_library(KF5::Solid ALIAS KF5Solid)
 # Apps must include <Solid/File> or <solid/file.h>
 target_include_directories(KF5Solid INTERFACE "$<INSTALL_INTERFACE:${KDE_INSTALL_INCLUDEDIR_KF5}/Solid>")
 
-target_link_libraries(KF5Solid PUBLIC Qt5::Core
+target_link_libraries(KF5Solid PUBLIC Qt5::Core PRIVATE Qt5::QspXDG
                                PRIVATE Qt5::DBus
                                        Qt5::Xml
                                        Qt5::Widgets
@@ -89,7 +89,7 @@ install(TARGETS KF5Solid EXPORT KF5SolidTargets ${KF5_INSTALL_TARGETS_DEFAULT_AR
 add_library(KF5Solid_static STATIC ${solid_LIB_SRCS})
 set_target_properties(KF5Solid_static PROPERTIES COMPILE_FLAGS -DSOLID_STATIC_DEFINE=1)
 
-target_link_libraries(KF5Solid_static PUBLIC Qt5::Core)
+target_link_libraries(KF5Solid_static PUBLIC Qt5::Core PRIVATE Qt5::QspXDG)
 target_link_libraries(KF5Solid_static PRIVATE Qt5::DBus Qt5::Xml Qt5::Widgets ${solid_OPTIONAL_LIBS})
 target_include_directories(KF5Solid_static PUBLIC "$<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR};${CMAKE_CURRENT_SOURCE_DIR}/..;${CMAKE_CURRENT_BINARY_DIR}/..>")
 
