--- src/ioslaves/mailto/orig.CMakeLists.txt	2015-10-05 10:11:41.000000000 +0200
+++ src/ioslaves/mailto/CMakeLists.txt	2015-11-14 22:39:20.000000000 +0100
@@ -3,7 +3,7 @@
 ecm_mark_nongui_executable(kmailservice5)
 
 target_link_libraries(kmailservice5
-   Qt5::Gui
+   PUBLIC Qt5::Gui PRIVATE Qt5::QspXDG
 )
 
 install(TARGETS kmailservice5 ${KF5_INSTALL_TARGETS_DEFAULT_ARGS})
Exit 1
--- src/ioslaves/trash/tests/orig.CMakeLists.txt	2015-10-05 10:11:41.000000000 +0200
+++ src/ioslaves/trash/tests/CMakeLists.txt	2015-11-14 22:40:16.000000000 +0100
@@ -27,4 +27,4 @@
 add_executable(lockingtest lockingtest.cpp ../kinterprocesslock.cpp)
 ecm_mark_nongui_executable(lockingtest)
 
-target_link_libraries(lockingtest Qt5::Core Qt5::DBus)
+target_link_libraries(lockingtest PUBLIC Qt5::Core Qt5::DBus PRIVATE Qt5::QspXDG)
