diff --git src/service/Application.cpp src/service/Application.cpp
index 682606a..abb8027 100644
--- src/service/Application.cpp
+++ src/service/Application.cpp
@@ -307,6 +307,9 @@ namespace  {
 
 int main(int argc, char **argv)
 {
+    // make sure we stay in the background on OS X, and don't show up in Dock or App Switcher
+    putenv(qstrdup("QT_MAC_DISABLE_FOREGROUND_APPLICATION_TRANSFORM=1"));
+
     Application application(argc, argv);
     application.setApplicationName(QStringLiteral("ActivityManager"));
     application.setOrganizationDomain(QStringLiteral("kde.org"));
diff --git src/service/CMakeLists.txt src/service/CMakeLists.txt
index 8b56f7c..0083838 100644
--- src/service/CMakeLists.txt
+++ src/service/CMakeLists.txt
@@ -65,6 +65,7 @@ qt5_add_dbus_adaptor (
    )
 
 add_executable (kactivitymanagerd ${kactivitymanager_SRCS})
+ecm_mark_nongui_executable (kactivitymanagerd)
 
 target_link_libraries (kactivitymanagerd
 PUBLIC
diff --git CMakeLists.txt CMakeLists.txt
--- orig.CMakeLists.txt	2015-12-15 22:16:26.000000000 +0100
+++ CMakeLists.txt	2015-12-16 14:34:18.000000000 +0100
@@ -45,6 +45,7 @@
 include (KDECompilerSettings)
 include (GenerateExportHeader)
 include (ECMGenerateHeaders)
+include (ECMMarkNonGuiExecutable)
 
 # Qt
 set (CMAKE_AUTOMOC ON)
