diff --git src/kdeui/kpushbutton.cpp src/kdeui/kpushbutton.cpp
index 98534fa..2781c75 100644
--- src/kdeui/kpushbutton.cpp
+++ src/kdeui/kpushbutton.cpp
@@ -253,7 +253,10 @@ void KPushButton::paintEvent(QPaintEvent *)
     if (d->delayedMenu) {
         option.features |= QStyleOptionButton::HasMenu;
     }
-
+    if (!style()->styleHint(QStyle::SH_DialogButtonBox_ButtonsHaveIcons, 0, this)) {
+        option.icon = QIcon();
+        option.iconSize = QSize(0,0);
+    }
     p.drawControl(QStyle::CE_PushButton, option);
 }
 
