diff --git src/kiod/CMakeLists.txt src/kiod/CMakeLists.txt
index 51d7f23..2f9696d 100644
--- src/kiod/CMakeLists.txt
+++ src/kiod/CMakeLists.txt
@@ -1,6 +1,11 @@
 set(kiod_SRCS kiod_main.cpp)
 
 add_executable(kiod5 ${kiod_SRCS})
+if (APPLE)
+    include(ECMMarkNonGuiExecutable)
+    # Mark it as non-gui so we won't create an app bundle on Mac OS X
+    ecm_mark_nongui_executable(kiod5)
+endif ()
 
 target_link_libraries(kiod5
    KF5::KIOCore     # ksslcertificatemanager
