diff --git src/core/kconfig.cpp src/core/kconfig.cpp
index 07fa6f5..37975ff 100644
--- src/core/kconfig.cpp
+++ src/core/kconfig.cpp
@@ -78,7 +78,7 @@ KConfigPrivate::KConfigPrivate(KConfig::OpenFlags flags,
 #ifdef Q_OS_WIN
             QFile::decodeName(qgetenv("WINDIR") + "/kde5rc");
 #else
-            QStringLiteral("/etc/kde5rc");
+            QStringLiteral("@PREFIX@/etc/kde5rc");
 #endif
         if (!QFileInfo(etc_kderc).isReadable()) {
             etc_kderc.clear();
