diff --git src/kdeui/kapplication.cpp src/kdeui/kapplication.cpp
index 6ffaf6e..8484974 100644
--- src/kdeui/kapplication.cpp
+++ src/kdeui/kapplication.cpp
@@ -745,7 +745,7 @@ void KApplicationPrivate::parseCommandLine()
     if (args && args->isSet("icon")) {
         q->setWindowIcon(QIcon::fromTheme(args->getOption("icon")));
     } else {
-        q->setWindowIcon(QIcon::fromTheme(componentData.aboutData()->programIconName()));
+        q->setWindowIcon(QIcon::fromTheme(componentData.aboutData()->programIconName(), q->windowIcon()));
     }
 
     if (!args) {
