--- src/ioslaves/http/kcookiejar/orig.kcookiewin.cpp	2015-12-20 19:16:02.000000000 +0100
+++ src/ioslaves/http/kcookiejar/kcookiewin.cpp	2015-12-20 19:16:17.000000000 +0100
@@ -64,7 +64,7 @@
     setModal(true);
     setObjectName(QStringLiteral("cookiealert"));
     setWindowTitle(i18n("Cookie Alert"));
-    setWindowIcon(QIcon::fromTheme(QStringLiteral("preferences-web-browser-cookies")));
+    setWindowIcon(QIcon::fromTheme(QStringLiteral("preferences-web-browser-cookies"), windowIcon()));
     // all cookies in the list should have the same window at this time, so let's take the first
     if (cookieList.first().windowIds().count() > 0) {
         KWindowSystem::setMainWindow(this, cookieList.first().windowIds().first());
