diff --git CMakeLists.txt CMakeLists.txt
index 4b9a5ff..edcda15 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -29,6 +29,7 @@ find_package(KF5Service ${KF5_DEP_VERSION} REQUIRED)
 
 include(ECMPackageConfigHelpers)
 include(ECMSetupVersion)
+include(ECMMarkNonGuiExecutable)
 
 ecm_setup_version(
   ${KF5_VERSION}
diff --git src/CMakeLists.txt src/CMakeLists.txt
index 5e95df8..449e757 100644
--- src/CMakeLists.txt
+++ src/CMakeLists.txt
@@ -3,12 +3,7 @@ set(kded_KDEINIT_SRCS kded.cpp kdedadaptor.cpp
 )
 
 kf5_add_kdeinit_executable(kded5 ${kded_KDEINIT_SRCS})
-
-if (APPLE)
-    set_target_properties(kded5 PROPERTIES MACOSX_BUNDLE_INFO_PLIST ${CMAKE_CURRENT_SOURCE_DIR}/Info.plist.template)
-    set_target_properties(kded5 PROPERTIES MACOSX_BUNDLE_GUI_IDENTIFIER "org.kded.kded5")
-    set_target_properties(kded5 PROPERTIES MACOSX_BUNDLE_BUNDLE_NAME "KDE Daemon")
-endif ()
+ecm_mark_nongui_executable(kded5)
 
 target_link_libraries(kdeinit_kded5
     Qt5::Widgets # QApplication
diff --git src/kded.cpp src/kded.cpp
index 6929d7d..e40b1d5 100644
--- src/kded.cpp
+++ src/kded.cpp
@@ -678,6 +678,8 @@ extern "C" Q_DECL_EXPORT int kdemain(int argc, char *argv[])
 {
     //options.add("check", qi18n("Check Sycoca database only once"));
 
+    // make sure we stay in the background on OS X, and don't show up in Dock or App Switcher
+    putenv(qstrdup("QT_MAC_DISABLE_FOREGROUND_APPLICATION_TRANSFORM=1"));
     // WABA: Make sure not to enable session management.
     putenv(qstrdup("SESSION_MANAGER="));
 
