--- autotests/orig.CMakeLists.txt	2015-11-10 18:23:52.000000000 +0100
+++ autotests/CMakeLists.txt	2015-11-10 18:24:02.000000000 +0100
@@ -21,6 +21,7 @@
                endif()
                add_test(plasma-${_testname} ${_testname})
                ecm_mark_as_test(${_testname})
+			ecm_mark_nongui_executable(${_testname})
        ENDFOREACH(_testname)
 ENDMACRO(PLASMA_UNIT_TESTS)
 
--- src/kpackagetool/orig.CMakeLists.txt	2015-10-05 10:18:13.000000000 +0200
+++ src/kpackagetool/CMakeLists.txt	2015-11-10 18:33:15.000000000 +0100
@@ -1,8 +1,11 @@
+include(ECMMarkNonGuiExecutable)
+
 add_executable(kpackagetool5
     main.cpp
     kpackagetool.cpp
     ../kpackage/private/packagejobthread.cpp
 )
+ecm_mark_nongui_executable(kpackagetool5)
 
 target_link_libraries(kpackagetool5 KF5::Archive KF5::Package KF5::I18n KF5::ConfigCore KF5::CoreAddons)
 
