diff --git src/api/KWallet/CMakeLists.txt src/api/KWallet/CMakeLists.txt
index e2a5631..ff5d141 100644
--- src/api/KWallet/CMakeLists.txt
+++ src/api/KWallet/CMakeLists.txt
@@ -75,8 +75,6 @@ target_link_libraries(KF5Wallet PUBLIC  Qt5::Gui
 if(MAC_USE_OSXKEYCHAIN)
    target_link_libraries(KF5Wallet PRIVATE ${CARBON_LIBRARY}
                                               ${SECURITY_LIBRARY})
-elseif(APPLE)
-    target_link_libraries(KF5Wallet PRIVATE ${CARBON_LIBRARY})
 else()
     target_link_libraries(KF5Wallet PRIVATE Qt5::DBus)
     install(FILES ${kwallet_xml} DESTINATION ${KDE_INSTALL_DBUSINTERFACEDIR} RENAME kf5_org.kde.KWallet.xml)
diff --git src/runtime/kwalletd/CMakeLists.txt src/runtime/kwalletd/CMakeLists.txt
index ba9e7ba..a21ee8e 100644
--- src/runtime/kwalletd/CMakeLists.txt
+++ src/runtime/kwalletd/CMakeLists.txt
@@ -113,6 +113,9 @@ target_link_libraries(kwalletd5
 if (KF5Gpgmepp_FOUND)
     target_link_libraries(kwalletd5 KF5::Gpgmepp )
     kde_target_enable_exceptions(kwalletd5 PRIVATE)
+if (APPLE)
+    target_link_libraries(kwalletd5 "-lgpgme" )
+endif ()
 endif(KF5Gpgmepp_FOUND)
 
 install(TARGETS kwalletd5  ${KF5_INSTALL_TARGETS_DEFAULT_ARGS})
