diff --git a/qtbase/mkspecs/common/macx.conf b/qtbase/mkspecs/common/macx.conf
index 69d731b..bcadbc2 100644
--- a/qtbase/mkspecs/common/macx.conf
+++ b/qtbase/mkspecs/common/macx.conf
@@ -5,4 +5,9 @@
 QMAKE_PLATFORM         += macos osx macx
 QMAKE_MAC_SDK           = macosx
 
+# X11 is not inexistant on OS X; provide the library requirements but leave it to the
+# user to make them accessible during the Qt build.
+QMAKE_LIBS_X11          = -lXext -lX11
+QMAKE_LIBS_OPENVG       = -lOpenVG
+
 include(mac.conf)
diff --git a/qtbase/src/platformsupport/linuxaccessibility/linuxaccessibility.pro b/qtbase/src/platformsupport/linuxaccessibility/linuxaccessibility.pro
index 6d68909..4d9f620 100644
--- a/qtbase/src/platformsupport/linuxaccessibility/linuxaccessibility.pro
+++ b/qtbase/src/platformsupport/linuxaccessibility/linuxaccessibility.pro
@@ -8,6 +8,9 @@ DEFINES += QT_NO_CAST_FROM_ASCII
 PRECOMPILED_HEADER = ../../corelib/global/qt_pch.h
 
 include(../../3rdparty/atspi2/atspi2.pri)
+darwin: {
+    INCLUDEPATH += $$PWD/../../gui/kernel
+}
 
 HEADERS += \
     application_p.h \
