diff --git a/qtbase/src/plugins/platforms/cocoa/qnsview.mm b/qtbase/src/plugins/platforms/cocoa/qnsview.mm
index 14f82e1..e74b251 100644
--- a/qtbase/src/plugins/platforms/cocoa/qnsview.mm
+++ b/qtbase/src/plugins/platforms/cocoa/qnsview.mm
@@ -37,6 +37,8 @@
 **
 ****************************************************************************/
 
+# macstrop:///aqua/qt5-kde-devel/files/qt580/patch-silence-qnsview-warnings.diff NOOP
+
 #include <QtCore/qglobal.h>
 
 #include "qnsview.h"
