--- qt-everywhere-opensource-src-5.8.0/qtdeclarative/src/qml/jsruntime/qv4engine.cpp.orig	2017-10-05 16:58:49.000000000 +0200
+++ qt-everywhere-opensource-src-5.8.0/qtdeclarative/src/qml/jsruntime/qv4engine.cpp	2017-10-31 13:21:14.000000000 +0100
@@ -166,7 +166,7 @@
         bool jitDisabled = true;
 
 #ifdef V4_ENABLE_JIT
-        static const bool forceMoth = !qEnvironmentVariableIsEmpty("QV4_FORCE_INTERPRETER") ||
+        const bool forceMoth = !qEnvironmentVariableIsEmpty("QV4_FORCE_INTERPRETER") ||
                                       !OSAllocator::canAllocateExecutableMemory();
         if (forceMoth) {
             factory = new Moth::ISelFactory;
# @@ -850,8 +850,9 @@
#      ExecutionContext *c = currentContext;
#      while (c) {
#          CallContext *callCtx = c->asCallContext();
# -        if (callCtx && callCtx->d()->v4Function) {
# -            base.setUrl(callCtx->d()->v4Function->sourceFile());
# +        if (callCtx && callCtx->d()->function) {
# +            if (callCtx->d()->function->function)
# +                base.setUrl(callCtx->d()->function->function->sourceFile());
#              break;
#          }
#          c = parentContext(c);
