# https://github.com/annulen/webkit/issues/778

diff --git a/Source/WebKit/qt/WebCoreSupport/QWebPageAdapter.cpp b/Source/WebKit/qt/WebCoreSupport/QWebPageAdapter.cpp
index 00dc0c48841..1c14fcd07ae 100644
--- a/Source/WebKit/qt/WebCoreSupport/QWebPageAdapter.cpp
+++ b/Source/WebKit/qt/WebCoreSupport/QWebPageAdapter.cpp
@@ -206,7 +206,11 @@ static void openNewWindow(const QUrl& url, Frame* frame)
 }
 
 // FIXME: Find a better place
-Ref<UserContentController> s_userContentProvider = UserContentController::create();
+static UserContentController& userContentProvider()
+{
+    static NeverDestroyed<Ref<UserContentController>> s_userContentProvider(UserContentController::create());
+    return s_userContentProvider.get();
+}
 
 QWebPageAdapter::QWebPageAdapter()
     : settings(0)
@@ -242,7 +246,7 @@ void QWebPageAdapter::initializeWebCorePage()
     pageConfiguration.databaseProvider = &WebDatabaseProvider::singleton();
     pageConfiguration.storageNamespaceProvider = WebStorageNamespaceProvider::create(
         QWebSettings::globalSettings()->localStoragePath());
-    pageConfiguration.userContentController = &s_userContentProvider.get();
+    pageConfiguration.userContentController = &userContentProvider();
     pageConfiguration.visitedLinkStore = &VisitedLinkStoreQt::singleton();
     page = new Page(pageConfiguration);

