--- qtwebkit-5.212.0-alpha2/Source/cmake/orig.OptionsQt.cmake	2017-06-16 01:36:48.000000000 +0200
+++ qtwebkit-5.212.0-alpha2/Source/cmake/OptionsQt.cmake	2017-11-11 22:21:23.000000000 +0100
@@ -386,15 +386,14 @@
     set(ZLIB_FOUND 1)
 endif ()
 
-if (NOT APPLE)
-    find_package(ICU REQUIRED)
-else ()
-    set(ICU_INCLUDE_DIRS
-        "${WEBCORE_DIR}/icu"
-        "${JAVASCRIPTCORE_DIR}/icu"
-        "${WTF_DIR}/icu"
-    )
-    set(ICU_LIBRARIES libicucore.dylib)
+find_package(ICU REQUIRED)
+if (APPLE)
+#     set(ICU_INCLUDE_DIRS
+#         "${WEBCORE_DIR}/icu"
+#         "${JAVASCRIPTCORE_DIR}/icu"
+#         "${WTF_DIR}/icu"
+#     )
+#     set(ICU_LIBRARIES libicucore.dylib)
     find_library(COREFOUNDATION_LIBRARY CoreFoundation)
     if (QT_STATIC_BUILD)
         find_library(CARBON_LIBRARY Carbon)
