diff --git a/hw/xquartz/xpr/xprScreen.c b/hw/xquartz/xpr/xprScreen.c
index 30f2218b6e118d13c661e4446b41bcdaddb2d2f3..f9376b3c6e64837b9774c002b617e4e697bc2a6a 100644
--- a/hw/xquartz/xpr/xprScreen.c
+++ b/hw/xquartz/xpr/xprScreen.c
@@ -292,10 +292,12 @@ xprDisplayInit(void)
     /* With PseudoramiX, the X server only sees one screen; only PseudoramiX
        itself knows about all of the screens. */
 
-    if (noPseudoramiXExtension)
+    if (noPseudoramiXExtension) {
         darwinScreensFound = displayCount;
-    else
+    } else {
+        PseudoramiXExtensionInit();
         darwinScreensFound = 1;
+    }
 
     if (xp_init(XP_BACKGROUND_EVENTS | XP_NO_DEFERRED_UPDATES) != Success)
         FatalError("Could not initialize the Xplugin library.");
diff --git a/mi/miinitext.c b/mi/miinitext.c
index 8cb26d0c4..2e4aba534 100644
--- a/mi/miinitext.c
+++ b/mi/miinitext.c
@@ -100,9 +100,6 @@ SOFTWARE.
 #include "misc.h"
 #include "extension.h"
 #include "extinit.h"
-#ifdef INXQUARTZ
-#include "nonsdk_extinit.h"
-#endif
 #include "micmap.h"
 #include "os.h"
 #include "globals.h"
@@ -129,12 +126,6 @@ static const ExtensionModule staticExtensions[] = {
 #endif
 #ifdef PANORAMIX
     {PanoramiXExtensionInit, "XINERAMA", &noPanoramiXExtension},
-#endif
-#ifdef INXQUARTZ
-    /* PseudoramiXExtensionInit must be done before RRExtensionInit, or
-     * XQuartz will render windows offscreen.
-     */
-    {PseudoramiXExtensionInit, "PseudoramiX", &noPseudoramiXExtension},
 #endif
     /* must be before Render to layer DisplayCursor correctly */
     {XFixesExtensionInit, "XFIXES", &noXFixesExtension},
