diff --git a/configure.ac b/configure.ac
index 7694d80d9b5ebe225ed3fcbe90ef2c5b3a403a1d..5fa350eae0084629878e7b6da774fffbbf3c5233 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1542,6 +1542,8 @@ MI_EXT_LIB='$(top_builddir)/mi/libmiext.la'
 MI_INC='-I$(top_srcdir)/mi'
 FB_LIB='$(top_builddir)/fb/libfb.la'
 FB_INC='-I$(top_srcdir)/fb'
+MIEXT_ROOTLESS_INC='-I$(top_srcdir)/miext/rootless'
+MIEXT_ROOTLESS_LIB='$(top_builddir)/miext/rootless/librootless.la'
 MIEXT_SHADOW_INC='-I$(top_srcdir)/miext/shadow'
 MIEXT_SHADOW_LIB='$(top_builddir)/miext/shadow/libshadow.la'
 MIEXT_SYNC_INC='-I$(top_srcdir)/miext/sync'
@@ -2143,6 +2145,7 @@ if test "x$XWIN" = xyes; then
 		PKG_CHECK_MODULES(WINDOWSWM, $WINDOWSWMPROTO)
 		XWINMODULES_CFLAGS="$XWINMODULES_CFLAGS $WINDOWSWM_CFLAGS"
 		AC_DEFINE(ROOTLESS,1,[Build Rootless code])
+        ROOTLESS=yes
 	fi
 
 	if test "x$WINDOWSDRI" = xauto; then
@@ -2209,6 +2212,7 @@ dnl Darwin / OS X DDX
 if test "x$XQUARTZ" = xyes; then
 	AC_DEFINE(XQUARTZ,1,[Have Quartz])
 	AC_DEFINE(ROOTLESS,1,[Build Rootless code])
+    ROOTLESS=yes
 
 	XQUARTZ_LIBS="$FB_LIB $FIXES_LIB $XEXT_LIB $DBE_LIB $RECORD_LIB $RANDR_LIB $RENDER_LIB $DAMAGE_LIB $MIEXT_SYNC_LIB $MIEXT_DAMAGE_LIB $MIEXT_SHADOW_LIB $XI_LIB $XKB_LIB $XKB_STUB_LIB $XPSTUBS_LIB $PRESENT_LIB"
 	AC_SUBST([XQUARTZ_LIBS])
@@ -2338,7 +2342,11 @@ if test "$KDRIVE" = yes; then
     
     KDRIVE_CFLAGS="$XSERVER_CFLAGS"
 
-    KDRIVE_PURE_LIBS="$FB_LIB $MI_LIB $FIXES_LIB $XEXT_LIB $DBE_LIB $RECORD_LIB $GLX_LIBS $RANDR_LIB $RENDER_LIB $DAMAGE_LIB $DRI3_LIB $PRESENT_LIB $MIEXT_SYNC_LIB $MIEXT_DAMAGE_LIB $MIEXT_SHADOW_LIB $XI_LIB $XKB_LIB $XKB_STUB_LIB $COMPOSITE_LIB $OS_LIB"
+    if test "x$ROOTLESS" = xyes; then
+        KDRIVE_PURE_LIBS="$FB_LIB $MI_LIB $FIXES_LIB $XEXT_LIB $DBE_LIB $RECORD_LIB $GLX_LIBS $RANDR_LIB $RENDER_LIB $DAMAGE_LIB $DRI3_LIB $PRESENT_LIB $MIEXT_SYNC_LIB $MIEXT_DAMAGE_LIB $MIEXT_ROOTLESS_LIB $MIEXT_SHADOW_LIB $XI_LIB $XKB_LIB $XKB_STUB_LIB $COMPOSITE_LIB $OS_LIB"
+    else
+        KDRIVE_PURE_LIBS="$FB_LIB $MI_LIB $FIXES_LIB $XEXT_LIB $DBE_LIB $RECORD_LIB $GLX_LIBS $RANDR_LIB $RENDER_LIB $DAMAGE_LIB $DRI3_LIB $PRESENT_LIB $MIEXT_SYNC_LIB $MIEXT_DAMAGE_LIB $MIEXT_SHADOW_LIB $XI_LIB $XKB_LIB $XKB_STUB_LIB $COMPOSITE_LIB $OS_LIB"
+    fi
     KDRIVE_LIB='$(top_builddir)/hw/kdrive/src/libkdrive.la'
     KDRIVE_MAIN_LIB="$MAIN_LIB"
     KDRIVE_LOCAL_LIBS="$DIX_LIB $KDRIVE_LIB"
diff --git a/hw/kdrive/ephyr/Makefile.am b/hw/kdrive/ephyr/Makefile.am
index d12559b39407058183ea69fd53729c81f3be5d43..6f72cd5e47e8f54702af9746566cc224b3d4e61a 100644
--- a/hw/kdrive/ephyr/Makefile.am
+++ b/hw/kdrive/ephyr/Makefile.am
@@ -76,6 +76,10 @@ Xephyr_LDADD = 						\
 	@KDRIVE_LIBS@					\
 	@XEPHYR_LIBS@
 
+if XQUARTZ
+Xephyr_LDADD += -lXplugin
+endif
+
 Xephyr_DEPENDENCIES = @KDRIVE_LOCAL_LIBS@ $(XEPHYR_GLAMOR_LIB)
 
 Xephyr_LDFLAGS = $(LD_EXPORT_SYMBOLS_FLAG)
diff --git a/hw/kdrive/ephyr/ephyr.c b/hw/kdrive/ephyr/ephyr.c
index 4f1f19860f77569e4da51cf248362ead5e02f713..4f651fd9d9d5427a27f4557f130151b3fb20dc78 100644
--- a/hw/kdrive/ephyr/ephyr.c
+++ b/hw/kdrive/ephyr/ephyr.c
@@ -61,6 +61,13 @@ Bool EphyrWantGrayScale = 0;
 Bool EphyrWantResize = 0;
 Bool EphyrWantNoHostGrab = 0;
 
+#ifdef __APPLE__
+// used in librootless
+int darwinMainScreenX = 0;
+int darwinMainScreenY = 0;
+Bool no_configure_window = FALSE;
+#endif
+
 Bool
 ephyrInitialize(KdCardInfo * card, EphyrPriv * priv)
 {
