diff --git a/CMakeLists.txt b/CMakeLists.txt
index a4e36f54..849c27f9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -4,8 +4,8 @@ project(drkonqi)
 set(PROJECT_VERSION "5.16.80")
 set(PROJECT_VERSION_MAJOR 5)
 
-set(QT_MIN_VERSION "5.12.0")
-set(KF5_MIN_VERSION "5.58.0")
+set(QT_MIN_VERSION "5.9.0")
+set(KF5_MIN_VERSION "5.52.0")
 
 find_package(ECM ${KF5_MIN_VERSION} REQUIRED NO_MODULE)
 
diff --git a/src/bugzillaintegration/bugzillalib.cpp b/src/bugzillaintegration/bugzillalib.cpp
index e4479463..46b03b6a 100644
--- a/src/bugzillaintegration/bugzillalib.cpp
+++ b/src/bugzillaintegration/bugzillalib.cpp
@@ -107,7 +107,11 @@ BugzillaManager::BugzillaManager(const QString &bugTrackerUrl, QObject *parent)
     // the constructors.
     // I am not sure why this is so weirdly done TBH. Might deserve some looking
     // into.
+#if QT_VERSION >= QT_VERSION_CHECK(5, 10, 0)
     QMetaObject::invokeMethod(this, &BugzillaManager::lookupVersion, Qt::QueuedConnection);
+#else
+    QMetaObject::invokeMethod (this, "lookupVersion", Qt::QueuedConnection);
+#endif
 }
 
 void BugzillaManager::lookupVersion()
diff --git a/src/systeminformation.cpp b/src/systeminformation.cpp
index 1a7c8b11..2df1731a 100644
--- a/src/systeminformation.cpp
+++ b/src/systeminformation.cpp
@@ -28,6 +28,7 @@
 # include <sys/utsname.h>
 #endif
 
+#include <QFile>
 
 #include <KProcess>
 #include "drkonqi_debug.h"
@@ -36,7 +37,9 @@
 #include <KSharedConfig>
 #include <kcoreaddons_version.h>
 #include <KCoreAddons>
+#if KCOREADDONS_VERSION >= QT_VERSION_CHECK(5, 58, 0)
 #include <KOSRelease>
+#endif
 #include <QStandardPaths>
 
 static const QString OS_UNSPECIFIED = QStringLiteral("unspecified");
@@ -81,7 +84,11 @@ static QString fetchBasicOperatingSystem()
 SystemInformation::Config::Config()
     : basicOperatingSystem(fetchBasicOperatingSystem())
     , lsbReleasePath(QStandardPaths::findExecutable(QStringLiteral("lsb_release")))
+#if KCOREADDONS_VERSION >= QT_VERSION_CHECK(5, 58, 0)
     , osReleasePath(/* Use KOSRelease default */)
+#else
+    , osReleasePath(QStringLiteral("/etc/os-release"))
+#endif
 {
 }
 
@@ -236,8 +243,36 @@ QString SystemInformation::fetchOSDetailInformation() const
 
 QString SystemInformation::fetchOSReleaseInformation()
 {
+#if KCOREADDONS_VERSION >= QT_VERSION_CHECK(5, 58, 0)
     KOSRelease os(m_infoConfig.osReleasePath);
     return m_distributionPrettyName = os.prettyName();
+#else
+    QFile data(m_infoConfig.osReleasePath);
+    if (!data.open(QIODevice::ReadOnly | QIODevice::Text)) {
+        return QString();
+    }
+
+    QMap<QString,QString> distroInfos;
+
+    QTextStream in(&data);
+    while (!in.atEnd()) {
+        const QString line = in.readLine();
+
+        // its format is one simple NAME=VALUE per line
+        // don't use QString.split() here since its value might contain '=''
+        const int index = line.indexOf(QLatin1Char('='));
+        if ( index != -1 ) {
+            const QString key = line.left(index);
+            const QString value = line.mid(index+1);
+            distroInfos.insert(key, value);
+        }
+    }
+
+    // the PRETTY_NAME entry should be the most appropriate one,
+    // but I could be wrong.
+    const QString prettyName = distroInfos.value(QStringLiteral("PRETTY_NAME"), QStringLiteral("Linux"));
+    return m_distributionPrettyName = prettyName;
+#endif
 }
 
 QString SystemInformation::operatingSystem() const
