diff --git a/CMakeLists.txt b/CMakeLists.txt
index 364f90acd69abb6ae0976621b803f49f0c40a62c..276128127f217e78ab2c07c3fa034e6f20d92c6d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -23,7 +23,9 @@ include(FeatureSummary)
 find_package(Qt5 ${QT_MIN_VERSION} CONFIG REQUIRED COMPONENTS Core Widgets Test DBus Concurrent)
 
 find_package(KF5 ${KF5_MIN_VERSION} REQUIRED COMPONENTS I18n CoreAddons Service ConfigWidgets JobWidgets KIO Crash Completion XmlRpcClient WidgetsAddons Wallet Notifications IdleTime)
-
+if(APPLE)
+    find_package(KF5 ${KF5_MIN_VERSION} REQUIRED COMPONENTS WindowSystem)
+endif()
 find_package(Qt5X11Extras ${QT_MIN_VERSION} CONFIG)
 set_package_properties(Qt5X11Extras PROPERTIES TYPE RECOMMENDED PURPOSE "Recommended for better integration on X11.")
 
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 0f4cc51a9f3782ef673147f34f91e0723a90bbed..46b86d721bbe75f24fb667bc0699b5dc04196866 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -111,6 +111,11 @@ if (${Qt5X11Extras_FOUND})
         Qt5::X11Extras
     )
 endif()
+if (APPLE)
+    target_link_libraries(drkonqi
+        KF5::WindowSystem
+    )
+endif()
 
 if (WIN32)
     target_link_libraries(drkonqi kdewin)
diff --git a/src/aboutbugreportingdialog.cpp b/src/aboutbugreportingdialog.cpp
index 51335bcc07f5246b786a629bc2d4a1310e95fd2e..56d2059681f5aee8c9597f2067bff5832b9b7ab4 100644
--- a/src/aboutbugreportingdialog.cpp
+++ b/src/aboutbugreportingdialog.cpp
@@ -36,7 +36,7 @@ AboutBugReportingDialog::AboutBugReportingDialog(QWidget * parent):
 {
     setAttribute(Qt::WA_DeleteOnClose, true);
 
-    setWindowIcon(QIcon::fromTheme(QStringLiteral("help-hint")));
+    setWindowIcon(QIcon::fromTheme(QStringLiteral("help-hint"), windowIcon()));
     setWindowTitle(i18nc("@title title of the dialog", "About Bug Reporting - Help"));
 
     QVBoxLayout* layout = new QVBoxLayout(this);
diff --git a/src/bugzillaintegration/reportassistantdialog.cpp b/src/bugzillaintegration/reportassistantdialog.cpp
index 8ef91b111608b8fc345ea0a80596ec03851a3e8e..de5174b8b7574ec4fe22924b2c7343163ef0271f 100644
--- a/src/bugzillaintegration/reportassistantdialog.cpp
+++ b/src/bugzillaintegration/reportassistantdialog.cpp
@@ -50,7 +50,7 @@ ReportAssistantDialog::ReportAssistantDialog(QWidget * parent) :
 
     //Set window properties
     setWindowTitle(i18nc("@title:window","Crash Reporting Assistant"));
-    setWindowIcon(QIcon::fromTheme(QStringLiteral("tools-report-bug")));
+    setWindowIcon(QIcon::fromTheme(QStringLiteral("tools-report-bug"), windowIcon()));
 
     connect(this, &ReportAssistantDialog::currentPageChanged, this, &ReportAssistantDialog::currentPageChanged_slot);
     connect(button(QDialogButtonBox::Help), &QPushButton::clicked, this, &ReportAssistantDialog::showHelp);
diff --git a/src/drkonqidialog.cpp b/src/drkonqidialog.cpp
index bfda9a22107a50f0950b0f216936819f6589bc77..01e03c5d055be330e273329a442112123673066e 100644
--- a/src/drkonqidialog.cpp
+++ b/src/drkonqidialog.cpp
@@ -55,7 +55,7 @@ DrKonqiDialog::DrKonqiDialog(QWidget * parent) :
 
     //Setting dialog title and icon
     setWindowTitle(DrKonqi::crashedApplication()->name());
-    setWindowIcon(QIcon::fromTheme(QStringLiteral("tools-report-bug")));
+    setWindowIcon(QIcon::fromTheme(QStringLiteral("tools-report-bug"), windowIcon()));
 
     QVBoxLayout* l = new QVBoxLayout(this);
     m_tabWidget = new QTabWidget(this);
diff --git a/src/main.cpp b/src/main.cpp
index 77197d414ff0ccffd40947d580e8efda6c0919c4..7c87ec5dcea49e690d8f5c669765ed37c5894b80 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -38,6 +38,9 @@
 #if HAVE_X11
 #include <QX11Info>
 #endif
+#ifdef Q_OS_MACOS
+#include <KWindowSystem>
+#endif
 
 #include "drkonqi.h"
 #include "drkonqidialog.h"
@@ -81,7 +84,7 @@ int main(int argc, char* argv[])
                          QStringLiteral("gkiagia@users.sourceforge.net"));
     aboutData.addAuthor(i18nc("@info:credit","A. L. Spehr"), QString(),
                          QStringLiteral("spehr@kde.org"));
-    qa.setWindowIcon(QIcon::fromTheme(QStringLiteral("tools-report-bug")));
+    qa.setWindowIcon(QIcon::fromTheme(QStringLiteral("tools-report-bug"), qa.windowIcon()));
     qa.setDesktopFileName(QStringLiteral("org.kde.drkonqi"));
 
     QCommandLineParser parser;
@@ -152,6 +155,9 @@ int main(int argc, char* argv[])
         DrKonqiDialog *w = new DrKonqiDialog();
         QObject::connect(w, &DrKonqiDialog::rejected, &qa, &QApplication::quit);
         w->show();
+#ifdef Q_OS_MACOS
+        KWindowSystem::forceActiveWindow(w->winId());
+#endif
     };
 
     bool restarted = parser.isSet(restartedOption);
