diff --git orig.FFT.cpp FFT.cpp
index 984901f..c419ea6 100644
--- src/common/orig.FFT.cpp
+++ src/common/FFT.cpp
@@ -21,23 +21,23 @@
     you must obtain a valid commercial licence before doing so.
 */
 
-#include "FFT.h"
-#include "Thread.h"
-#include "Profiler.h"
-#include "Allocators.h"
-#include "VectorOps.h"
-#include "VectorOpsComplex.h"
-
 // Define USE_FFTW_WISDOM if you are defining HAVE_FFTW3 and you want
 // to use FFTW_MEASURE mode with persistent wisdom files. This will
 // make things much slower on first use if no suitable wisdom has been
 // saved, but may be faster during subsequent use.
-//#define USE_FFTW_WISDOM 1
+#define USE_FFTW_WISDOM 1
 
 // Define FFT_MEASUREMENT to include timing measurement code callable
 // via the static method FFT::tune(). Must be defined when the header
 // is included as well.
-//#define FFT_MEASUREMENT 1
+#define FFT_MEASUREMENT 1
+
+#include "FFT.h"
+#include "Thread.h"
+#include "Profiler.h"
+#include "Allocators.h"
+#include "VectorOps.h"
+#include "VectorOpsComplex.h"
 
 #ifdef FFT_MEASUREMENT
 #define FFT_MEASUREMENT_RETURN_RESULT_TEXT 1
