--- attica-0.4.2/lib/orig.knowledgebaseentry.h	2013-05-01 00:45:43.000000000 +0200
+++ attica-0.4.2/lib/knowledgebaseentry.h	2015-10-08 21:59:49.000000000 +0200
@@ -29,6 +29,7 @@
 #include <QtCore/QSharedDataPointer>
 
 #include <QtCore/QUrl>
+#include <QMap>
 
 
 namespace Attica
--- attica-0.4.2/lib/orig.knowledgebaseentry.cpp	2013-05-01 00:45:43.000000000 +0200
+++ attica-0.4.2/lib/knowledgebaseentry.cpp	2015-10-08 21:59:33.000000000 +0200
@@ -23,8 +23,6 @@
 
 #include "knowledgebaseentry.h"
 
-#include <QMap>
-
 using namespace Attica;
 
 class KnowledgeBaseEntry::Private : public QSharedData {
