--- attica-0.4.2/orig.CMakeLists.txt	2013-05-01 00:45:43.000000000 +0200
+++ attica-0.4.2/CMakeLists.txt	2015-10-08 21:47:22.000000000 +0200
@@ -1,6 +1,8 @@
 project(attica)
 cmake_minimum_required(VERSION 2.8)
 
+set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules" )
+
 option(QT4_BUILD "Force building with Qt4 even if Qt5 is found")
 if (NOT QT4_BUILD)
   find_package(Qt5Core QUIET)
