diff --git a/src/konqmainwindow.cpp b/src/konqmainwindow.cpp
index 0351dcd593cbe9ec7c01d52315e1c3c33c3c4dc8..a90a71c179742c73bfc3a11a3d7b5cd093049f47 100644
--- a/src/konqmainwindow.cpp
+++ b/src/konqmainwindow.cpp
@@ -137,7 +137,9 @@
 #include <kxmlguifactory.h>
 #include <sonnet/configdialog.h>
 #include <kwindowsystem.h>
+#if KONQ_HAVE_X11
 #include <netwm.h>
+#endif
 
 #include <kauthorized.h>
 #include <QDBusConnection>
