diff --git a/plugins/domtreeviewer/CMakeLists.txt b/plugins/domtreeviewer/CMakeLists.txt
index 48cba192f17ac6dde05169c3532d2fa05923f7d0..29f764d8b7e6fa847311a80aecd8d63e1a16addd 100644
--- a/plugins/domtreeviewer/CMakeLists.txt
+++ b/plugins/domtreeviewer/CMakeLists.txt
@@ -15,7 +15,7 @@ ki18n_wrap_ui(domtreeviewerplugin_PART_SRCS domtreeviewbase.ui texteditwidget.ui
 
 add_library(domtreeviewerplugin MODULE ${domtreeviewerplugin_PART_SRCS})
 
-
+kde_target_enable_exceptions(domtreeviewerplugin PRIVATE)
 
 target_link_libraries(domtreeviewerplugin KF5::KHtml KF5::KDELibs4Support)
 
diff --git a/plugins/webarchiver/CMakeLists.txt b/plugins/webarchiver/CMakeLists.txt
index 6baebefbb6f68d437dc67c9afe4014985c12eb94..0eee470a7d5fbb3aee46fb428ebe7a6bf4ca743b 100644
--- a/plugins/webarchiver/CMakeLists.txt
+++ b/plugins/webarchiver/CMakeLists.txt
@@ -10,6 +10,8 @@ ki18n_wrap_ui(webarchiverplugin_PART_SRCS archiveviewbase.ui )
 
 add_library(webarchiverplugin MODULE ${webarchiverplugin_PART_SRCS})
 
+kde_target_enable_exceptions(webarchiverplugin PRIVATE)
+
 target_link_libraries(webarchiverplugin KF5::KHtml KF5::KDELibs4Support KF5::KIOCore KF5::Archive)
 install(TARGETS webarchiverplugin  DESTINATION ${KDE_INSTALL_PLUGINDIR} )
 
