diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 9d73d4bc60b0e008469312ff346069cf58f7659e..3603d8d4a3edfa219f3348e4bc2447ebbf61a7a6 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -95,7 +95,7 @@ qt5_add_dbus_adaptor(konqueror_KDEINIT_SRCS org.kde.Konqueror.SessionManager.xml
 qt5_add_dbus_interface(konqueror_KDEINIT_SRCS org.kde.Konqueror.SessionManager.xml konqsessionmanager_interface)
 
 file(GLOB ICONS_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/../pics/*-apps-konqueror.png")
-ecm_add_app_icon(konqueror_KDEINIT_SRCS ICONS ${ICONS_SRCS})
+ecm_add_app_icon(konqueror_KDEINIT_SRCS ICONS ${CMAKE_CURRENT_SOURCE_DIR}/../pics/hisc-app-konqueror.svgz ${ICONS_SRCS})
 kf5_add_kdeinit_executable(konqueror ${konqueror_KDEINIT_SRCS} konqmain.cpp)
 target_link_libraries(kdeinit_konqueror
    konquerorprivate
