diff --git a/settings/konqhtml/pluginopts.cpp b/settings/konqhtml/pluginopts.cpp
index d7f4750..46790b0 100644
--- a/settings/konqhtml/pluginopts.cpp
+++ b/settings/konqhtml/pluginopts.cpp
@@ -420,6 +420,10 @@ void KPluginOptions::dirLoad(KSharedConfig::Ptr config, bool useDefault)
     } else { //keep sync with kdebase/apps/nsplugins
         paths.append(QStringLiteral("$HOME/.mozilla/plugins"));
         paths.append(QStringLiteral("$HOME/.netscape/plugins"));
+#ifdef Q_OS_OSX
+        paths.append(QStringLiteral("$HOME/Library/Internet Plug-Ins"));
+        paths.append(QStringLiteral("/Library/Internet Plug-Ins"));
+#endif
         paths.append(QStringLiteral("/usr/lib/firefox/plugins"));
         paths.append(QStringLiteral("/usr/lib64/browser-plugins"));
         paths.append(QStringLiteral("/usr/lib/browser-plugins"));
