diff --git a/CMakeLists.txt b/CMakeLists.txt
index 77cf38a..da5d36d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -15,6 +15,7 @@ include(KDEInstallDirs)
 include(KDECMakeSettings)
 include(KDECompilerSettings NO_POLICY_SCOPE)
 include(ECMInstallIcons)
+include(ECMMarkNonGuiExecutable)
 include(ECMSetupVersion)
 
 set(KONQUEROR_LIB_VERSION "5.0.97")
diff --git a/plugins/fsview/CMakeLists.txt b/plugins/fsview/CMakeLists.txt
index 31ca04c..24c6dcb 100644
--- a/plugins/fsview/CMakeLists.txt
+++ b/plugins/fsview/CMakeLists.txt
@@ -16,6 +16,7 @@ set(libfsview_SRCS
 set(fsview_SRCS main.cpp ${libfsview_SRCS} )
 
 add_executable(fsview ${fsview_SRCS})
+ecm_mark_nongui_executable(fsview)
 
 target_link_libraries(fsview KF5::KIOWidgets KF5::KDELibs4Support)
 
diff --git a/client/CMakeLists.txt b/client/CMakeLists.txt
index 745b0f3..5f49c48 100644
--- a/client/CMakeLists.txt
+++ b/client/CMakeLists.txt
@@ -24,6 +25,7 @@ else (WIN32)
     endif(X11_FOUND)
     install(TARGETS kdeinit_kfmclient  ${KDE_INSTALL_TARGETS_DEFAULT_ARGS} )
 endif (WIN32)
+ecm_mark_nongui_executable(kfmclient)
 
 install(TARGETS kfmclient ${KDE_INSTALL_TARGETS_DEFAULT_ARGS})
 
