--- a/src/orig.konqmain.cpp	2016-12-06 05:33:26.000000000 +0100
+++ b/src/konqmain.cpp	2016-12-28 21:44:56.000000000 +0100
@@ -58,6 +58,9 @@
 
 extern "C" Q_DECL_EXPORT int kdemain(int argc, char **argv)
 {
+//#ifdef Q_OS_MACOS
+    QCoreApplication::setAttribute(Qt::AA_ShareOpenGLContexts);
+//#endif
     KonquerorApplication app(argc, argv);    
     KLocalizedString::setApplicationDomain("konqueror");
 
