--- src/orig.kstandardaction.cpp	2015-11-08 13:13:44.000000000 +0100
+++ src/kstandardaction.cpp	2015-12-07 23:02:00.000000000 +0100
@@ -171,6 +171,10 @@
             break;
         }
 
+        // Set the text before setting the MenuRole, as on OS X setText will do some heuristic role guessing.
+        // This ensures user menu items get the intended role out of the list below.
+        pAction->setText(sLabel);
+
         switch (id) {
         case Quit:
             pAction->setMenuRole(QAction::QuitRole);
@@ -189,7 +193,6 @@
             break;
         }
 
-        pAction->setText(sLabel);
         if (pInfo->psToolTip) {
             pAction->setToolTip(i18n(pInfo->psToolTip));
         }
