--- a/README
+++ b/README
@@ -1,4 +1,6 @@
-                        PulseAudio Volume Control 0.99.2
+
+
+                         PulseAudio Volume Control 2.0
 
    Copyright 2006-2009 Lennart Poettering, 2009-2011 Colin Guthrie
    <mzcnihpbageby (at) 0pointer (dot) de>
@@ -26,83 +28,91 @@
 
 News
 
+   Fri Mar 8 2013:
+
+   [10]Version 2.0 released.
+
+   Tue Sep 27 2011:
+
+   [11]Version 1.0 released.
+
    Thu Sep 15 2011:
 
-   [10]Version 0.99.2 (v1 RC2) released; digitial receiver UI.
+   [12]Version 0.99.2 (v1 RC2) released; digitial receiver UI.
 
    Tue Aug 2 2011:
 
-   [11]Version 0.99.1 (v1 RC1) released; gtk3 support; source output
+   [13]Version 0.99.1 (v1 RC1) released; gtk3 support; source output
    volume support; key command handling; bugfixes.
 
    Wed Oct 14 2009:
 
-   [12]Version 0.9.10 released; minor fixes; translation updates.
+   [14]Version 0.9.10 released; minor fixes; translation updates.
 
    Thu Sep 10 2009:
 
-   [13]Version 0.9.9 released; allow configuring device ports; various
+   [15]Version 0.9.9 released; allow configuring device ports; various
    updates.
 
    Mon Apr 13 2009:
 
-   [14]Version 0.9.8 released; allow configuring card profiles; various
+   [16]Version 0.9.8 released; allow configuring card profiles; various
    updates.
 
    Tue Sep 9 2008:
 
-   [15]Version 0.9.7 released; show volume meter for each stream and
+   [17]Version 0.9.7 released; show volume meter for each stream and
    device.
 
    Fri Mar 28 2008:
 
-   [16]Version 0.9.6 released; draw radio buttons instead of check boxes
+   [18]Version 0.9.6 released; draw radio buttons instead of check boxes
    when moving streams; fix crasher when connecting to a PA server that
    has no default sink/source configured.
 
    Tue Oct 30 2007:
 
-   [17]Version 0.9.5 released; rework UI; add context menu option to kill
+   [19]Version 0.9.5 released; rework UI; add context menu option to kill
    a playback stream; add context menu option to select the default
    device; use prettier channel names; filter non-application streams by
    default; support muting streams
 
    Sat Aug 26 2006:
 
-   [18]Version 0.9.4 released; add the ability to move a stream to another
+   [20]Version 0.9.4 released; add the ability to move a stream to another
    device while it is playing; limit the list of sinks/sources shown by
    their type; other cleanups
 
    Mon Jul 24 2006:
 
-   [19]Version 0.9.3 released; show client names only if they are set
+   [21]Version 0.9.3 released; show client names only if they are set
 
    Sat Jul 8 2006:
 
-   [20]Version 0.9.2 released; update for [21]PulseAudio 0.9.2
+   [22]Version 0.9.2 released; update for [23]PulseAudio 0.9.2
 
    Fri Jun 2 2006:
 
-   [22]Version 0.9.1 released; add a .desktop file; update icons
+   [24]Version 0.9.1 released; add a .desktop file; update icons
    everywhere
 
    Sat May 27 2006:
 
-   [23]Version 0.9.0 released; update for Polypaudio 0.9.0; show client
+   [25]Version 0.9.0 released; update for Polypaudio 0.9.0; show client
    name of playback streams; make volume slider more responsive
 
    Fri Apr 21 2006:
 
-   [24]Version 0.8 released; initial release
+   [26]Version 0.8 released; initial release
 
 Overview
 
    PulseAudio Volume Control (pavucontrol) is a simple GTK based volume
-   control tool ("mixer") for the [25]PulseAudio sound server. In contrast
+   control tool ("mixer") for the [27]PulseAudio sound server. In contrast
    to classic mixer tools this one allows you to control both the volume
    of hardware devices and of each playback stream separately.
 
-   Everybody loves [26]screenshots.
+   Everybody loves [28]screenshots.
 
 Current Status
 
@@ -121,9 +131,9 @@
    Unix versions) since it uses GNU autoconf for source code
    configuration.
 
-   pavucontrol requires [27]gtkmm and glademmm installed.
+   pavucontrol requires [29]gtkmm and glademmm installed.
 
-   Obviously pavucontrol requires an installation of [28]PulseAudio
+   Obviously pavucontrol requires an installation of [30]PulseAudio
    (version 0.9.16).
 
 Installation
@@ -140,12 +150,12 @@
 Download
 
    The newest release is always available from
-   [29]http://freedesktop.org/software/pulseaudio/pavucontrol/
+   [31]http://freedesktop.org/software/pulseaudio/pavucontrol/
 
-   The current release is [30]0.99.2
+   The current release is [32]2.0
 
-   Get pavucontrol's development sources from the [31]GIT [32]repository
-   ([33]gitweb):
+   Get pavucontrol's development sources from the [33]GIT [34]repository
+   ([35]gitweb):
 git clone git://git.0pointer.de/pavucontrol
      __________________________________________________________________
 
@@ -155,36 +165,38 @@
 
 References
 
-   1. README#license
-   2. README#news
-   3. README#overview
-   4. README#status
-   5. README#documentation
-   6. README#requirements
-   7. README#installation
-   8. README#acks
-   9. README#download
-  10. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.99.2.tar.gz
-  11. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.99.1.tar.gz
-  12. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.9.10.tar.gz
-  13. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.9.9.tar.gz
-  14. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.9.8.tar.gz
-  15. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.9.7.tar.gz
-  16. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.9.6.tar.gz
-  17. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.9.5.tar.gz
-  18. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.9.4.tar.gz
-  19. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.9.3.tar.gz
-  20. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.9.2.tar.gz
-  21. http://pulseaudio.org/
-  22. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.9.1.tar.gz
-  23. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.9.0.tar.gz
-  24. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.8.tar.gz
-  25. http://0pointer.de/lennart/projects/pulseaudio/
-  26. http://freedesktop.org/software/pulseaudio/pavucontrol//screenshot.png
-  27. http://gtkmm.sf.net/
-  28. http://pulseaudio.org/
-  29. http://freedesktop.org/software/pulseaudio/pavucontrol/
-  30. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.99.2.tar.gz
-  31. http://git-scm.org/
-  32. git://git.0pointer.de/pavucontrol
-  33. http://git.0pointer.de/?p=pavucontrol.git
+   1. README.html#license
+   2. README.html#news
+   3. README.html#overview
+   4. README.html#status
+   5. README.html#documentation
+   6. README.html#requirements
+   7. README.html#installation
+   8. README.html#acks
+   9. README.html#download
+  10. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-2.0.tar.xz
+  11. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-1.0.tar.xz
+  12. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.99.2.tar.gz
+  13. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.99.1.tar.gz
+  14. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.9.10.tar.gz
+  15. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.9.9.tar.gz
+  16. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.9.8.tar.gz
+  17. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.9.7.tar.gz
+  18. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.9.6.tar.gz
+  19. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.9.5.tar.gz
+  20. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.9.4.tar.gz
+  21. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.9.3.tar.gz
+  22. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.9.2.tar.gz
+  23. http://pulseaudio.org/
+  24. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.9.1.tar.gz
+  25. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.9.0.tar.gz
+  26. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.8.tar.gz
+  27. http://0pointer.de/lennart/projects/pulseaudio/
+  28. http://freedesktop.org/software/pulseaudio/pavucontrol//screenshot.png
+  29. http://gtkmm.sf.net/
+  30. http://pulseaudio.org/
+  31. http://freedesktop.org/software/pulseaudio/pavucontrol/
+  32. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-2.0.tar.gz
+  33. http://git-scm.org/
+  34. git://git.0pointer.de/pavucontrol
+  35. http://git.0pointer.de/?p=pavucontrol.git
--- a/doc/README
+++ b/doc/README
@@ -1,4 +1,5 @@
-                        PulseAudio Volume Control 0.99.2
+
+                         PulseAudio Volume Control 2.0
 
    Copyright 2006-2009 Lennart Poettering, 2009-2011 Colin Guthrie
    <mzcnihpbageby (at) 0pointer (dot) de>
@@ -26,83 +27,91 @@
 
 News
 
+   Fri Mar 8 2013:
+
+   [10]Version 2.0 released.
+
+   Tue Sep 27 2011:
+
+   [11]Version 1.0 released.
+
    Thu Sep 15 2011:
 
-   [10]Version 0.99.2 (v1 RC2) released; digitial receiver UI.
+   [12]Version 0.99.2 (v1 RC2) released; digitial receiver UI.
 
    Tue Aug 2 2011:
 
-   [11]Version 0.99.1 (v1 RC1) released; gtk3 support; source output
+   [13]Version 0.99.1 (v1 RC1) released; gtk3 support; source output
    volume support; key command handling; bugfixes.
 
    Wed Oct 14 2009:
 
-   [12]Version 0.9.10 released; minor fixes; translation updates.
+   [14]Version 0.9.10 released; minor fixes; translation updates.
 
    Thu Sep 10 2009:
 
-   [13]Version 0.9.9 released; allow configuring device ports; various
+   [15]Version 0.9.9 released; allow configuring device ports; various
    updates.
 
    Mon Apr 13 2009:
 
-   [14]Version 0.9.8 released; allow configuring card profiles; various
+   [16]Version 0.9.8 released; allow configuring card profiles; various
    updates.
 
    Tue Sep 9 2008:
 
-   [15]Version 0.9.7 released; show volume meter for each stream and
+   [17]Version 0.9.7 released; show volume meter for each stream and
    device.
 
    Fri Mar 28 2008:
 
-   [16]Version 0.9.6 released; draw radio buttons instead of check boxes
+   [18]Version 0.9.6 released; draw radio buttons instead of check boxes
    when moving streams; fix crasher when connecting to a PA server that
    has no default sink/source configured.
 
    Tue Oct 30 2007:
 
-   [17]Version 0.9.5 released; rework UI; add context menu option to kill
+   [19]Version 0.9.5 released; rework UI; add context menu option to kill
    a playback stream; add context menu option to select the default
    device; use prettier channel names; filter non-application streams by
    default; support muting streams
 
    Sat Aug 26 2006:
 
-   [18]Version 0.9.4 released; add the ability to move a stream to another
+   [20]Version 0.9.4 released; add the ability to move a stream to another
    device while it is playing; limit the list of sinks/sources shown by
    their type; other cleanups
 
    Mon Jul 24 2006:
 
-   [19]Version 0.9.3 released; show client names only if they are set
+   [21]Version 0.9.3 released; show client names only if they are set
 
    Sat Jul 8 2006:
 
-   [20]Version 0.9.2 released; update for [21]PulseAudio 0.9.2
+   [22]Version 0.9.2 released; update for [23]PulseAudio 0.9.2
 
    Fri Jun 2 2006:
 
-   [22]Version 0.9.1 released; add a .desktop file; update icons
+   [24]Version 0.9.1 released; add a .desktop file; update icons
    everywhere
 
    Sat May 27 2006:
 
-   [23]Version 0.9.0 released; update for Polypaudio 0.9.0; show client
+   [25]Version 0.9.0 released; update for Polypaudio 0.9.0; show client
    name of playback streams; make volume slider more responsive
 
    Fri Apr 21 2006:
 
-   [24]Version 0.8 released; initial release
+   [26]Version 0.8 released; initial release
 
 Overview
 
    PulseAudio Volume Control (pavucontrol) is a simple GTK based volume
-   control tool ("mixer") for the [25]PulseAudio sound server. In contrast
+   control tool ("mixer") for the [27]PulseAudio sound server. In contrast
    to classic mixer tools this one allows you to control both the volume
    of hardware devices and of each playback stream separately.
 
-   Everybody loves [26]screenshots.
+   Everybody loves [28]screenshots.
 
 Current Status
 
@@ -121,9 +130,9 @@
    Unix versions) since it uses GNU autoconf for source code
    configuration.
 
-   pavucontrol requires [27]gtkmm and glademmm installed.
+   pavucontrol requires [29]gtkmm and glademmm installed.
 
-   Obviously pavucontrol requires an installation of [28]PulseAudio
+   Obviously pavucontrol requires an installation of [30]PulseAudio
    (version 0.9.16).
 
 Installation
@@ -140,12 +149,12 @@
 Download
 
    The newest release is always available from
-   [29]http://freedesktop.org/software/pulseaudio/pavucontrol/
+   [31]http://freedesktop.org/software/pulseaudio/pavucontrol/
 
-   The current release is [30]0.99.2
+   The current release is [32]2.0
 
-   Get pavucontrol's development sources from the [31]GIT [32]repository
-   ([33]gitweb):
+   Get pavucontrol's development sources from the [33]GIT [34]repository
+   ([35]gitweb):
 git clone git://git.0pointer.de/pavucontrol
      __________________________________________________________________
 
@@ -155,36 +164,38 @@
 
 References
 
-   1. README#license
-   2. README#news
-   3. README#overview
-   4. README#status
-   5. README#documentation
-   6. README#requirements
-   7. README#installation
-   8. README#acks
-   9. README#download
-  10. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.99.2.tar.gz
-  11. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.99.1.tar.gz
-  12. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.9.10.tar.gz
-  13. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.9.9.tar.gz
-  14. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.9.8.tar.gz
-  15. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.9.7.tar.gz
-  16. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.9.6.tar.gz
-  17. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.9.5.tar.gz
-  18. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.9.4.tar.gz
-  19. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.9.3.tar.gz
-  20. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.9.2.tar.gz
-  21. http://pulseaudio.org/
-  22. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.9.1.tar.gz
-  23. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.9.0.tar.gz
-  24. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.8.tar.gz
-  25. http://0pointer.de/lennart/projects/pulseaudio/
-  26. http://freedesktop.org/software/pulseaudio/pavucontrol//screenshot.png
-  27. http://gtkmm.sf.net/
-  28. http://pulseaudio.org/
-  29. http://freedesktop.org/software/pulseaudio/pavucontrol/
-  30. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.99.2.tar.gz
-  31. http://git-scm.org/
-  32. git://git.0pointer.de/pavucontrol
-  33. http://git.0pointer.de/?p=pavucontrol.git
+   1. README.html#license
+   2. README.html#news
+   3. README.html#overview
+   4. README.html#status
+   5. README.html#documentation
+   6. README.html#requirements
+   7. README.html#installation
+   8. README.html#acks
+   9. README.html#download
+  10. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-2.0.tar.xz
+  11. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-1.0.tar.xz
+  12. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.99.2.tar.gz
+  13. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.99.1.tar.gz
+  14. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.9.10.tar.gz
+  15. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.9.9.tar.gz
+  16. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.9.8.tar.gz
+  17. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.9.7.tar.gz
+  18. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.9.6.tar.gz
+  19. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.9.5.tar.gz
+  20. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.9.4.tar.gz
+  21. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.9.3.tar.gz
+  22. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.9.2.tar.gz
+  23. http://pulseaudio.org/
+  24. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.9.1.tar.gz
+  25. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.9.0.tar.gz
+  26. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-0.8.tar.gz
+  27. http://0pointer.de/lennart/projects/pulseaudio/
+  28. http://freedesktop.org/software/pulseaudio/pavucontrol//screenshot.png
+  29. http://gtkmm.sf.net/
+  30. http://pulseaudio.org/
+  31. http://freedesktop.org/software/pulseaudio/pavucontrol/
+  32. http://freedesktop.org/software/pulseaudio/pavucontrol/pavucontrol-2.0.tar.gz
+  33. http://git-scm.org/
+  34. git://git.0pointer.de/pavucontrol
+  35. http://git.0pointer.de/?p=pavucontrol.git
