diff --git release/makemacapp.in release/makemacapp.in
index 43441b8b..d787e958 100644
--- release/makemacapp.in
+++ release/makemacapp.in
@@ -34,8 +34,7 @@ if [ -f $PACKAGE_NAME.dmg ]; then
 fi
 
 umask 022
-TMPDIR=`mktemp -d /tmp/$PACKAGE_NAME-build.XXXXXX`
-APPROOT="$TMPDIR/dmg/TigerVNC.app"
+APPROOT="./TigerVNC Viewer.app"
 mkdir -p "$APPROOT/Contents/MacOS"
 mkdir -p "$APPROOT/Contents/Resources"
 
@@ -49,6 +48,8 @@ for lang in `cat "$SRCDIR/po/LINGUAS"`; do
 		"$APPROOT/Contents/Resources/locale/$lang/LC_MESSAGES/tigervnc.mo"
 done
 
+exit
+
 install -m 644 $SRCDIR/LICENCE.TXT $TMPDIR/dmg/
 install -m 644 $SRCDIR/README.rst $TMPDIR/dmg/
 
