diff --git CMakeLists.txt CMakeLists.txt
index ba6b3203..be766834 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -137,6 +137,7 @@ find_package(Pixman REQUIRED)
 
 # Check for gettext
 option(ENABLE_NLS "Enable translation of program messages" ON)
+option(USE_GETTEXT_FROM_LIBINTL "Force the use of gettext functions from libintl instead of libc" OFF)
 if(ENABLE_NLS)
   # Tools
   find_package(Gettext)
@@ -153,7 +154,7 @@ if(ENABLE_NLS)
       set(CMAKE_REQUIRED_LIBRARIES ${ICONV_LIBRARIES})
       set(CMAKE_REQUIRED_FLAGS -fno-builtin-dgettext)
       check_function_exists(dgettext LIBC_HAS_DGETTEXT)
-      if(LIBC_HAS_DGETTEXT)
+      if(LIBC_HAS_DGETTEXT AND NOT USE_GETTEXT_FROM_LIBINTL)
         set(GETTEXT_FOUND TRUE)
       else()
         find_library(LIBINTL_LIBRARY NAMES intl libintl)
@@ -301,17 +302,17 @@ if(UNIX AND NOT APPLE)
 endif()
 
 # Check for SELinux library
-if(UNIX AND NOT APPLE)
-  check_include_files(selinux/selinux.h HAVE_SELINUX_H)
-  if(HAVE_SELINUX_H)
-    set(CMAKE_REQUIRED_LIBRARIES -lselinux)
-    set(CMAKE_REQUIRED_LIBRARIES)
-    set(SELINUX_LIBS selinux)
-    add_definitions("-DHAVE_SELINUX")
-  else()
-    message(WARNING "Could not find SELinux development files")
-  endif()
-endif()
+#if(UNIX AND NOT APPLE)
+#  check_include_files(selinux/selinux.h HAVE_SELINUX_H)
+#  if(HAVE_SELINUX_H)
+#    set(CMAKE_REQUIRED_LIBRARIES -lselinux)
+#    set(CMAKE_REQUIRED_LIBRARIES)
+#    set(SELINUX_LIBS selinux)
+#    add_definitions("-DHAVE_SELINUX")
+#  else()
+#    message(WARNING "Could not find SELinux development files")
+#  endif()
+#endif()
 
 # check for systemd support (socket activation)
 if(UNIX AND NOT APPLE)
