diff --git kstyle/CMakeLists.txt kstyle/CMakeLists.txt
index 7d526d9..b72d624 100644
--- kstyle/CMakeLists.txt
+++ kstyle/CMakeLists.txt
@@ -23,10 +23,12 @@ if(BREEZE_USE_KDE4)
   add_definitions(${QT_DEFINITIONS} ${KDE4_DEFINITIONS})
   include_directories(${KDE4_INCLUDES})
 
+  if(NOT APPLE AND NOT WIN32)
   ### XCB
   find_package(PkgConfig REQUIRED)
   pkg_check_modules(XCB xcb x11-xcb)
   set(BREEZE_HAVE_X11 ${XCB_FOUND})
+  endif(NOT APPLE AND NOT WIN32)
 
 else()
 
diff --git CMakeLists.txt CMakeLists.txt
index e2ec52a..c206a90 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -6,6 +6,8 @@ include(FeatureSummary)
 
 if(USE_KDE4)
   add_subdirectory(kstyle)
+  add_subdirectory(icons)
+  add_subdirectory(icons-dark)
 else()
   find_package(ECM 0.0.9 REQUIRED NO_MODULE)
   set(CMAKE_MODULE_PATH ${ECM_MODULE_PATH} ${ECM_KDE_MODULE_DIR})
