--- CMakeLists.txt.orig	2017-01-13 15:05:40.000000000 +0100
+++ CMakeLists.txt	2017-03-23 21:53:53.421226130 +0100
@@ -673,7 +673,7 @@ if(NOT CMake_TEST_EXTERNAL_CMAKE)
 
   # The CMake executables usually do not need any rpath to run in the build or
   # install tree.
-  set(CMAKE_SKIP_RPATH ON CACHE INTERNAL "CMake does not need RPATHs.")
+  set(CMAKE_SKIP_RPATH OFF CACHE INTERNAL "CMake does not need RPATHs.")
 
   # Load install destinations.
   include(Source/CMakeInstallDestinations.cmake)
