--- ./Modules/Platform/Darwin-Intel.cmake.orig	2014-06-09 17:46:52.000000000 +0200
+++ ./Modules/Platform/Darwin-Intel.cmake	2014-07-03 12:26:08.000000000 +0200
@@ -21,7 +21,8 @@
 macro(__darwin_compiler_intel lang)
   set(CMAKE_${lang}_VERBOSE_FLAG "-v -Wl,-v") # also tell linker to print verbose output
   set(CMAKE_SHARED_LIBRARY_CREATE_${lang}_FLAGS "-dynamiclib -Wl,-headerpad_max_install_names")
-  set(CMAKE_SHARED_MODULE_CREATE_${lang}_FLAGS "-bundle -Wl,-headerpad_max_install_names")
+  #set(CMAKE_SHARED_MODULE_CREATE_${lang}_FLAGS "-bundle -Wl,-headerpad_max_install_names")
+  set(CMAKE_SHARED_MODULE_CREATE_${lang}_FLAGS "-dynamiclib -Wl,-headerpad_max_install_names")
 
   if(NOT CMAKE_${lang}_COMPILER_VERSION VERSION_LESS 12.0)
     set(CMAKE_${lang}_COMPILE_OPTIONS_VISIBILITY "-fvisibility=")
--- ./Modules/Platform/Darwin-XL-C.cmake.orig	2014-06-09 17:46:52.000000000 +0200
+++ ./Modules/Platform/Darwin-XL-C.cmake	2014-07-03 12:26:15.000000000 +0200
@@ -1,5 +1,6 @@
 set(CMAKE_SHARED_LIBRARY_CREATE_C_FLAGS "-qmkshrobj")
-set(CMAKE_SHARED_MODULE_CREATE_C_FLAGS "-bundle")
+#set(CMAKE_SHARED_MODULE_CREATE_C_FLAGS "-bundle")
+set(CMAKE_SHARED_MODULE_CREATE_C_FLAGS "-dynamiclib")
 
 # Enable shared library versioning.
 set(CMAKE_SHARED_LIBRARY_SONAME_C_FLAG "-Wl,-install_name")
--- ./Modules/Platform/Darwin-XL-CXX.cmake.orig	2014-06-09 17:46:52.000000000 +0200
+++ ./Modules/Platform/Darwin-XL-CXX.cmake	2014-07-03 12:26:22.000000000 +0200
@@ -1,5 +1,6 @@
 set(CMAKE_SHARED_LIBRARY_CREATE_CXX_FLAGS "-qmkshrobj")
-set(CMAKE_SHARED_MODULE_CREATE_CXX_FLAGS "-bundle")
+#set(CMAKE_SHARED_MODULE_CREATE_CXX_FLAGS "-bundle")
+set(CMAKE_SHARED_MODULE_CREATE_CXX_FLAGS "-dynamiclib")
 
 # Enable shared library versioning.
 set(CMAKE_SHARED_LIBRARY_SONAME_CXX_FLAG "-Wl,-install_name")
--- ./Modules/Platform/Darwin.cmake.orig	2014-06-09 17:46:52.000000000 +0200
+++ ./Modules/Platform/Darwin.cmake	2014-07-03 12:26:31.000000000 +0200
@@ -50,7 +50,8 @@
 
 set(CMAKE_PLATFORM_HAS_INSTALLNAME 1)
 set(CMAKE_SHARED_LIBRARY_CREATE_C_FLAGS "-dynamiclib -Wl,-headerpad_max_install_names")
-set(CMAKE_SHARED_MODULE_CREATE_C_FLAGS "-bundle -Wl,-headerpad_max_install_names")
+#set(CMAKE_SHARED_MODULE_CREATE_C_FLAGS "-bundle -Wl,-headerpad_max_install_names")
+set(CMAKE_SHARED_MODULE_CREATE_C_FLAGS "-dynamiclib -Wl,-headerpad_max_install_names")
 set(CMAKE_SHARED_MODULE_LOADER_C_FLAG "-Wl,-bundle_loader,")
 set(CMAKE_SHARED_MODULE_LOADER_CXX_FLAG "-Wl,-bundle_loader,")
 set(CMAKE_FIND_LIBRARY_SUFFIXES ".dylib" ".so" ".a")
--- Modules/Platform/orig.Darwin-GNU.cmake	2015-02-10 21:59:20.000000000 +0100
+++ Modules/Platform/Darwin-GNU.cmake	2015-02-10 22:03:10.000000000 +0100
@@ -22,7 +22,8 @@
   set(CMAKE_${lang}_VERBOSE_FLAG "-v -Wl,-v") # also tell linker to print verbose output
   # GNU does not have -shared on OS X
   set(CMAKE_SHARED_LIBRARY_CREATE_${lang}_FLAGS "-dynamiclib -Wl,-headerpad_max_install_names")
-  set(CMAKE_SHARED_MODULE_CREATE_${lang}_FLAGS "-bundle -Wl,-headerpad_max_install_names")
+  #set(CMAKE_SHARED_MODULE_CREATE_${lang}_FLAGS "-bundle -Wl,-headerpad_max_install_names")
+  set(CMAKE_SHARED_MODULE_CREATE_${lang}_FLAGS "-dynamiclib -Wl,-headerpad_max_install_names")
 
   if(NOT CMAKE_${lang}_COMPILER_VERSION VERSION_LESS 4.3)
     set(CMAKE_${lang}_SYSTEM_FRAMEWORK_SEARCH_FLAG "-iframework ")
--- Modules/Platform/orig.Darwin-Clang.cmake	2015-02-10 21:59:20.000000000 +0100
+++ Modules/Platform/Darwin-Clang.cmake	2015-02-10 22:04:52.000000000 +0100
@@ -21,7 +21,8 @@
 macro(__darwin_compiler_clang lang)
   set(CMAKE_${lang}_VERBOSE_FLAG "-v -Wl,-v") # also tell linker to print verbose output
   set(CMAKE_SHARED_LIBRARY_CREATE_${lang}_FLAGS "-dynamiclib -Wl,-headerpad_max_install_names")
-  set(CMAKE_SHARED_MODULE_CREATE_${lang}_FLAGS "-bundle -Wl,-headerpad_max_install_names")
+  #set(CMAKE_SHARED_MODULE_CREATE_${lang}_FLAGS "-bundle -Wl,-headerpad_max_install_names")
+  set(CMAKE_SHARED_MODULE_CREATE_${lang}_FLAGS "-dynamiclib -Wl,-headerpad_max_install_names")
   set(CMAKE_${lang}_SYSROOT_FLAG "-isysroot")
   set(CMAKE_${lang}_OSX_DEPLOYMENT_TARGET_FLAG "-mmacosx-version-min=")
   if(NOT CMAKE_${lang}_COMPILER_VERSION VERSION_LESS 3.1)
