diff --git Source/QtDialog/CMakeLists.txt Source/QtDialog/CMakeLists.txt
index 03c2fb4..a73a37f 100644
--- Source/QtDialog/CMakeLists.txt
+++ Source/QtDialog/CMakeLists.txt
@@ -14,8 +14,8 @@ project(QtDialog)
 if(POLICY CMP0020)
   cmake_policy(SET CMP0020 NEW) # Drop when CMake >= 2.8.11 required
 endif()
-find_package(Qt5Widgets QUIET)
-if (Qt5Widgets_FOUND)
+#find_package(Qt5Widgets QUIET)
+if (FALSE)
   include_directories(${Qt5Widgets_INCLUDE_DIRS})
   add_definitions(${Qt5Widgets_DEFINITONS})
   macro(qt4_wrap_ui)
