diff --git a/Makefile.am b/Makefile.am
index 5dcd759..cdc87a3 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -48,7 +48,7 @@ libOpenCL_la_CFLAGS= $(NO_DEPRECATED_DECLARATIONS_FLAGS) \
 
 libOpenCL_la_SOURCES = ocl_icd_loader.c ocl_icd_loader.h ocl_icd_debug.h
 nodist_libOpenCL_la_SOURCES = ocl_icd_loader_gen.c ocl_icd.h
-libOpenCL_la_LDFLAGS = -version-info 1:0:0 $(PTHREAD_CFLAGS)
+libOpenCL_la_LDFLAGS = -version-info 0:0:0 $(PTHREAD_CFLAGS)
 libOpenCL_la_LIBADD = $(PTHREAD_LIBS)
 if USE_MAP
 libOpenCL_la_LDFLAGS += -Wl,--version-script,ocl_icd_loader_gen.map
