diff --git kdevplatform/shell/mainwindow_p.cpp kdevplatform/shell/mainwindow_p.cpp
index b50c444..958fc16 100644
--- kdevplatform/shell/mainwindow_p.cpp
+++ kdevplatform/shell/mainwindow_p.cpp
@@ -233,6 +233,12 @@ void MainWindowPrivate::setupActions()
     action->setIcon( KIcon("preferences-other") );
     action->setText( i18n("Configure Editor..."));
     action->setWhatsThis( i18nc("@info:whatsthis", "Configure various aspects of this editor.") );
+#ifdef Q_OS_MAC
+    // prevent this configure menuitem to be stuck in the Application menu as the Preferences item. The KAction constructor
+    // cannot do it because this action instance is allocated as a QAction.
+    kDebug() << "### setupActions(): setting QAction::NoRole from" << action->menuRole() << "for menuAction with text" << action->text();
+    action->setMenuRole(QAction::NoRole);
+#endif
 
     action =  KStandardAction::configureNotifications(this, SLOT(configureNotifications()), actionCollection());
     action->setText( i18n("Configure Notifications...") );
