diff --git shell/CMakeLists.txt shell/CMakeLists.txt
index 19a9560..0453839 100644
--- shell/CMakeLists.txt
+++ shell/CMakeLists.txt
@@ -141,6 +141,10 @@ LINK_PRIVATE
     KDev::Util
     KDev::Documentation
 )
+if(APPLE)
+    find_package(Qt5MacExtras ${REQUIRED_QT_VERSION} REQUIRED NO_MODULE)
+    target_link_libraries(KDevPlatformShell PRIVATE Qt5::MacExtras)
+endif()
 
 install(FILES
     mainwindow.h
diff --git shell/runcontroller.cpp shell/runcontroller.cpp
index 1a411da..d72b681 100644
--- shell/runcontroller.cpp
+++ shell/runcontroller.cpp
@@ -24,6 +24,9 @@ Boston, MA 02110-1301, USA.
 #include <QDBusConnection>
 #include <QPalette>
 #include <QSignalMapper>
+#ifdef Q_OS_OSX
+#include <QtMacExtras/QtMac>
+#endif
 
 #include <KAboutData>
 #include <KActionCollection>
@@ -666,6 +669,16 @@ void KDevelop::RunController::checkState()
         d->unityLauncher->setProgress(0);
     }
 
+#ifdef Q_OS_OSX
+    if (running) {
+        int percentDone = jobCount > 0 ? qRound(double(totalProgress) / double(jobCount)) : 0;
+        QtMac::setBadgeLabelText(QStringLiteral("%1%").arg(percentDone));
+    } else {
+        // remove the label
+        QtMac::setBadgeLabelText(QString());
+    }
+#endif
+
     if ( ( d->state != Running ? false : true ) == running ) {
         d->state = running ? Running : Idle;
         emit runStateChanged(d->state);
