diff --git interfaces/iproblem.h interfaces/iproblem.h
index 5e736ac..1e1863b 100644
--- interfaces/iproblem.h
+++ interfaces/iproblem.h
@@ -32,7 +32,7 @@ namespace KDevelop
 class IAssistant;
 
 /// Interface for the Problem classes
-class IProblem : public QSharedData
+class KDEVPLATFORMLANGUAGE_EXPORT IProblem : public QSharedData
 {
 public:
     typedef QExplicitlySharedDataPointer<IProblem> Ptr;
