diff --git sublime/container.cpp sublime/container.cpp
index b04f6c3..45dbc68 100644
--- sublime/container.cpp
+++ sublime/container.cpp
@@ -27,6 +27,7 @@
 #include <QMouseEvent>
 #include <QPointer>
 #include <QStackedWidget>
+#include <QStyleFactory>
 #include <QStyleOptionTabBarBase>
 #include <QStylePainter>
 #include <QTabBar>
@@ -54,6 +55,19 @@ public:
     ContainerTabBar(Container* container)
         : QTabBar(container), m_container(container)
     {
+        if (QApplication::style()->objectName() == QLatin1String("macintosh")) {
+            static QPointer<QStyle> qTabBarStyle = QStyleFactory::create("fusion");
+            if (qTabBarStyle) {
+                setStyle(qTabBarStyle);
+            }
+        }
+        // configure the QTabBar style so it behaves as appropriately as possible,
+        // even if we end up using the native Macintosh style because the user's
+        // Qt doesn't have the Fusion style installed.
+        setDocumentMode(true);
+        setUsesScrollButtons(true);
+        setElideMode(Qt::ElideNone);
+
         installEventFilter(this);
     }
 
