diff --git plugins/git/gitplugin.cpp plugins/git/gitplugin.cpp
index 5dca6ec..067fa1e 100644
--- plugins/git/gitplugin.cpp
+++ plugins/git/gitplugin.cpp
@@ -1190,7 +1190,7 @@ void GitPlugin::parseGitStatusOutput(DVcsJob* job)
 
         VcsStatusInfo status;
         status.setUrl(QUrl::fromLocalFile(dotGit.absoluteFilePath(curr.toString())));
-        qDebug() << "lalala" << line << state;
+//         qDebug() << "lalala" << line << state;
         status.setState(messageToState(state));
         processedFiles.append(status.url());
 
