diff --git a/sublime/container.cpp b/sublime/container.cpp
index 43e280ff2d1a75efa60f72a419628ab3e2805fe4..972f36953f3112a70ac3643b24bc89a85090baa3 100644
--- a/sublime/container.cpp
+++ b/sublime/container.cpp
@@ -495,7 +500,9 @@ void Container::setCurrentWidget(QWidget* w)
     //this function is called from MainWindow::activateView()
     //which does the activation without any additional signals
     {
+#ifndef Q_OS_MACOS
         QSignalBlocker blocker(d->tabBar);
+#endif
         d->tabBar->setCurrentIndex(d->stack->indexOf(w));
     }
     if (View* view = viewForWidget(w))

