diff --git kdevplatform/project/helper.cpp kdevplatform/project/helper.cpp
index 6df9f90..0fedba2 100644
--- kdevplatform/project/helper.cpp
+++ kdevplatform/project/helper.cpp
@@ -23,6 +23,8 @@
 #include <QApplication>
 #include <QDir>
 #include <QFileInfo>
+#include <QStandardPaths>
+#include <QProcess>
 
 #include <KIO/CopyJob>
 #include <KIO/DeleteJob>
@@ -216,6 +218,29 @@ bool KDevelop::copyPath(const KDevelop::IProject* project, const KDevelop::Path&
 Path KDevelop::proposedBuildFolder(const Path& sourceFolder)
 {
     Path proposedBuildFolder;
+    const QString customDirScript = QStandardPaths::findExecutable("kdevelop-custom-build-dir");
+    if (!customDirScript.isEmpty()) {
+        QProcess customDirHelper;
+        customDirHelper.start(customDirScript, QStringList() << sourceFolder.path());
+        if (customDirHelper.waitForFinished()) {
+            const QString customDir = customDirHelper.readAllStandardOutput().trimmed();
+            if (!customDir.isEmpty() && QDir(customDir).exists()) {
+                return Path(customDir);
+            } else {
+                qWarning() << "Ignoring inexistent" << customDir << "returned by"
+                    << customDirScript << "for source folder" << sourceFolder.path()
+                    << "helper error:" << customDirHelper.readAllStandardError();
+            }
+        } else {
+            qWarning() << "Custom build dir script" << customDirScript << "found but failed for source folder"
+                << sourceFolder.path() << "output=" << customDirHelper.readAll();
+        }
+    }
+    // check if the project is already configured for an out-of-source build:
+    const QString oosBuildPath = sourceFolder.parent().path() + QStringLiteral("/build");
+    if (QDir(oosBuildPath).exists()) {
+        return proposedBuildFolder;
+    }
     if (sourceFolder.path().contains(QStringLiteral("/src/"))) {
         const QString srcBuildPath = sourceFolder.path().replace(QStringLiteral("/src/"), QStringLiteral("/build/"));
         Q_ASSERT(!srcBuildPath.isEmpty());
