diff --git project/interfaces/ibuildsystemmanager.h project/interfaces/ibuildsystemmanager.h
index ee10c2f..fe15dcd 100644
--- project/interfaces/ibuildsystemmanager.h
+++ project/interfaces/ibuildsystemmanager.h
@@ -64,6 +64,11 @@ public:
     virtual Path::List includeDirectories(ProjectBaseItem*) const = 0;
 
     /**
+     * Provide a list of include directories.
+     */
+    virtual Path::List frameworkDirectories(ProjectBaseItem*) const = 0;
+
+    /**
      * Provide a list of preprocessor defines for the project item
      */
     virtual QHash<QString,QString> defines(ProjectBaseItem*) const = 0;
