diff --git shell/documentationcontroller.cpp shell/documentationcontroller.cpp
index 9f7f923..e770ae1 100644
--- shell/documentationcontroller.cpp
+++ shell/documentationcontroller.cpp
@@ -100,7 +100,8 @@ public:
         return new DocumentationView(parent, m_providersModel.data());
     }
 
-    Qt::DockWidgetArea defaultPosition() override { return Qt::RightDockWidgetArea; }
+//     Qt::DockWidgetArea defaultPosition() override { return Qt::RightDockWidgetArea; }
+    Qt::DockWidgetArea defaultPosition() override { return Qt::NoDockWidgetArea; }
     QString id() const override { return QStringLiteral("org.kdevelop.DocumentationView"); }
 
 private:
