diff --git outputview/outputfilteringstrategies.cpp outputview/outputfilteringstrategies.cpp
index f82420f..afa275c 100644
--- outputview/outputfilteringstrategies.cpp
+++ outputview/outputfilteringstrategies.cpp
@@ -142,7 +142,7 @@ FilteredItem CompilerFilterStrategy::actionInLine(const QString& line)
     // A list of filters for possible compiler, linker, and make actions
     static const ActionFormat ACTION_FILTERS[] = {
         ActionFormat( ki18nc("compiling a file: %1 file %2 compiler", "compiling %1 (%2)"), 1, 2,
-                      QStringLiteral("(?:^|[^=])\\b(gcc|CC|cc|distcc|c\\+\\+|g\\+\\+|clang(?:\\+\\+)|mpicc|icc|icpc)\\s+.*-c.*[/ '\\\\]+(\\w+\\.(?:cpp|CPP|c|C|cxx|CXX|cs|java|hpf|f|F|f90|F90|f95|F95))")),
+                      QStringLiteral("(?:^|[^=])\\b(gcc|CC|cc|distcc|c\\+\\+|g\\+\\+|clang(?:\\+\\+)|mpicc|icc|icpc)\\s+.*-c.*[/ '\\\\]+(\\w+\\.(?:cpp|CPP|c|C|cxx|CXX|cs|m|M|mm|MM|java|hpf|f|F|f90|F90|f95|F95))")),
         //moc and uic
         ActionFormat( ki18nc("generating a file: %1 file %2 generator tool", "generating %1 (%2)"), 1, 2,
                       QStringLiteral("/(moc|uic)\\b.*\\s-o\\s([^\\s;]+)")),
