diff --git vcs/vcspluginhelper.cpp vcs/vcspluginhelper.cpp
index 26ab57c..62453a3 100644
--- vcs/vcspluginhelper.cpp
+++ vcs/vcspluginhelper.cpp
@@ -184,7 +184,9 @@ void VcsPluginHelper::disposeEventually(KTextEditor::Document *)
 
 void VcsPluginHelper::setupFromContext(Context* context)
 {
-    d->ctxUrls = context->urls();
+    if (context) {
+        d->ctxUrls = context->urls();
+    }
 }
 
 QList<QUrl> VcsPluginHelper::contextUrlList() const
diff --git vcs/widgets/vcsdiffpatchsources.cpp vcs/widgets/vcsdiffpatchsources.cpp
index 195b9b6..2a41375 100644
--- vcs/widgets/vcsdiffpatchsources.cpp
+++ vcs/widgets/vcsdiffpatchsources.cpp
@@ -158,6 +158,7 @@ VCSDiffPatchSource::~VCSDiffPatchSource()
 {
     QFile::remove(m_file.toLocalFile());
     delete m_updater;
+    qDebug() << "~VCSDiffPatchSource()" << this;
 }
 
 QUrl VCSDiffPatchSource::baseDir() const {
@@ -212,6 +241,10 @@ void VCSDiffPatchSource::update() {
 
 VCSCommitDiffPatchSource::~VCSCommitDiffPatchSource() {
     delete m_commitMessageWidget.data();
+//     if (m_commitMessageWidget.data()) {
+//         m_commitMessageWidget.data()->deleteLater();
+//     }
+    qDebug() << "~VCSCommitDiffPatchSource()" << this;
 }
 
 bool VCSCommitDiffPatchSource::canSelectFiles() const {
