diff --git plugins/patchreview/patchreviewtoolview.cpp plugins/patchreview/patchreviewtoolview.cpp
index c088414..f59367f 100644
--- plugins/patchreview/patchreviewtoolview.cpp
+++ plugins/patchreview/patchreviewtoolview.cpp
@@ -135,7 +135,8 @@ void PatchReviewToolView::patchChanged() {
         m_exportMenu->model()->setInputData(QJsonObject {
             { QStringLiteral("urls"), QJsonArray { p->file().toString() } },
             { QStringLiteral("mimeType"), { QStringLiteral("text/x-patch") } },
-            { QStringLiteral("localBaseDir"), { p->baseDir().toString() } }
+            { QStringLiteral("localBaseDir"), { p->baseDir().toString() } },
+            { QStringLiteral("updateComment"), { QStringLiteral("patch updated through KDevelop's patchreview plugin") } }
         });
     }
 #endif
@@ -226,6 +227,7 @@ void PatchReviewToolView::showEditDialog() {
 #ifdef WITH_PURPOSE
     m_exportMenu = new Purpose::Menu(this);
     connect(m_exportMenu, &Purpose::Menu::finished, this, [](const QJsonObject &output, int error, const QString &message) {
+        qWarning() << Q_FUNC_INFO << "Purpose finished with output" << output << "error=" << error;
         if (error==0) {
             KMessageBox::information(nullptr, i18n("<qt>You can find the new request at:<br /><a href='%1'>%1</a> </qt>", output["url"].toString()),
                                     QString(), QString(), KMessageBox::AllowLink);

