diff --git sublime/container.cpp sublime/container.cpp
index b04f6c3..a104607 100644
--- sublime/container.cpp
+++ sublime/container.cpp
@@ -27,6 +27,7 @@
 #include <QMouseEvent>
 #include <QPointer>
 #include <QStackedWidget>
+#include <QStyleFactory>
 #include <QStyleOptionTabBarBase>
 #include <QStylePainter>
 #include <QTabBar>
@@ -42,10 +43,14 @@
 #include "document.h"
 #include <ksqueezedtextlabel.h>
 
+#include <QDebug>
+
 namespace Sublime {
 
 // struct ContainerPrivate
 
+static QStyle *qTabBarStyle = nullptr;
+
 class ContainerTabBar : public QTabBar
 {
     Q_OBJECT
@@ -54,6 +59,15 @@ public:
     ContainerTabBar(Container* container)
         : QTabBar(container), m_container(container)
     {
+        if (QApplication::style()->objectName() == QLatin1String("macintosh")) {
+            if (!qTabBarStyle) {
+                qTabBarStyle = QStyleFactory::create("fusion");
+            }
+            if (qTabBarStyle) {
+                qWarning() << "Setting style" << qTabBarStyle;
+                setStyle(qTabBarStyle);
+            }
+        }
         installEventFilter(this);
     }
 
