diff --git language/duchain/topducontextdynamicdata.cpp language/duchain/topducontextdynamicdata.cpp
index c56fac0..ec7cd23 100644
--- language/duchain/topducontextdynamicdata.cpp
+++ language/duchain/topducontextdynamicdata.cpp
@@ -407,7 +407,7 @@ Item TopDUContextDynamicData::DUChainItemStorage<Item>::getItemForIndex(uint ind
                                     "Potentially, the context has been deleted without deleting its children.");
     item->rebuildDynamicData(parent, index);
   } else {
-    qCWarning(LANGUAGE) << "invalid item for index" << index << offsets.size() << offsets.value(realIndex).dataOffset;
+    qCDebug(LANGUAGE) << "invalid item for index" << index << offsets.size() << offsets.value(realIndex).dataOffset;
   }
 
   return item;
