diff --git a/vcs/widgets/vcsdiffpatchsources.cpp b/vcs/widgets/vcsdiffpatchsources.cpp
index 42cf1d8..cbba87a 100644
--- a/vcs/widgets/vcsdiffpatchsources.cpp
+++ b/vcs/widgets/vcsdiffpatchsources.cpp
@@ -53,6 +53,10 @@ VCSCommitDiffPatchSource::VCSCommitDiffPatchSource(VCSDiffUpdater* updater)
     m_commitMessageEdit = new KTextEdit;
     m_commitMessageEdit.data()->setFont(QFontDatabase::systemFont(QFontDatabase::FixedFont));
     m_commitMessageEdit.data()->setLineWrapMode(QTextEdit::NoWrap);
+    // set the message editor to be 72 characters wide.
+    // Given the widget margins that requires 74 actual characters.
+    int w72 = m_commitMessageEdit.data()->fontMetrics().width(QString(74, 'm'));
+    m_commitMessageEdit.data()->setMinimumSize(QSize(w72,0));
     m_vcs->setupCommitMessageEditor(updater->url(), m_commitMessageEdit.data());
 
     QHBoxLayout* titleLayout = new QHBoxLayout;
@@ -211,7 +215,9 @@ void VCSDiffPatchSource::update() {
 }
 
 VCSCommitDiffPatchSource::~VCSCommitDiffPatchSource() {
-    delete m_commitMessageWidget.data();
+    if (m_commitMessageWidget.data()) {
+        m_commitMessageWidget.data()->deleteLater();
+    }
 }
 
 bool VCSCommitDiffPatchSource::canSelectFiles() const {
diff --git a/plugins/git/gitmessagehighlighter.cpp b/plugins/git/gitmessagehighlighter.cpp
index da7660d..a4cd2f8 100644
--- a/plugins/git/gitmessagehighlighter.cpp
+++ b/plugins/git/gitmessagehighlighter.cpp
@@ -38,6 +38,11 @@ void GitMessageHighlighter::applyErrorFormat(GitMessageHighlighter* text, bool w
     format.setFontUnderline(true);
     format.setUnderlineStyle(QTextCharFormat::SpellCheckUnderline);
     format.setUnderlineColor(warning ? Qt::yellow : Qt::red);
+    if (warning) {
+        format.setFontItalic(true);
+    } else {
+        format.setFontOverline(true);
+    }
     format.setToolTip(tooltip);
     text->setFormat(startPos, endPos, format);
 }
