diff --git plugins/patchreview/patchreview.cpp plugins/patchreview/patchreview.cpp
index 0050b2f..234348a 100644
--- plugins/patchreview/patchreview.cpp
+++ plugins/patchreview/patchreview.cpp
@@ -333,6 +333,9 @@ void PatchReviewPlugin::clearPatch( QObject* _patch ) {
 void PatchReviewPlugin::closeReview()
 {
     if( m_patch ) {
+        if( m_highlighters.size() ) {
+            qDebug() << "### Closing patchreview" << m_patch << "with" << m_highlighters.size() << "open documents";
+        }
         removeHighlighting();
         m_modelList.reset( 0 );
 
diff --git vcs/vcspluginhelper.cpp vcs/vcspluginhelper.cpp
index 26ab57c..62453a3 100644
--- vcs/vcspluginhelper.cpp
+++ vcs/vcspluginhelper.cpp
@@ -184,7 +184,9 @@ void VcsPluginHelper::disposeEventually(KTextEditor::Document *)
 
 void VcsPluginHelper::setupFromContext(Context* context)
 {
-    d->ctxUrls = context->urls();
+    if (context) {
+        d->ctxUrls = context->urls();
+    }
 }
 
 QList<QUrl> VcsPluginHelper::contextUrlList() const
diff --git vcs/widgets/vcsdiffpatchsources.cpp vcs/widgets/vcsdiffpatchsources.cpp
index 195b9b6..2a41375 100644
--- vcs/widgets/vcsdiffpatchsources.cpp
+++ vcs/widgets/vcsdiffpatchsources.cpp
@@ -158,6 +158,7 @@ VCSDiffPatchSource::~VCSDiffPatchSource()
 {
     QFile::remove(m_file.toLocalFile());
     delete m_updater;
+    qDebug() << "~VCSDiffPatchSource()" << this;
 }
 
 QUrl VCSDiffPatchSource::baseDir() const {
@@ -211,7 +212,11 @@ void VCSDiffPatchSource::update() {
 }
 
 VCSCommitDiffPatchSource::~VCSCommitDiffPatchSource() {
-    delete m_commitMessageWidget.data();
+//     delete m_commitMessageWidget.data();
+    if (m_commitMessageEdit.data()) {
+        m_commitMessageWidget.data()->deleteLater();
+    }
+    qDebug() << "~VCSCommitDiffPatchSource()" << this;
 }
 
 bool VCSCommitDiffPatchSource::canSelectFiles() const {
