diff --git plugins/patchreview/patchreview.cpp plugins/patchreview/patchreview.cpp
index 18b63db..04f3a25 100644
--- plugins/patchreview/patchreview.cpp
+++ plugins/patchreview/patchreview.cpp
@@ -304,7 +304,14 @@ PatchReviewPlugin::~PatchReviewPlugin() {
     removeHighlighting();
     // Tweak to work around a crash on OS X; see https://bugs.kde.org/show_bug.cgi?id=338829
     // and http://qt-project.org/forums/viewthread/38406/#162801
-    m_patch->deleteLater();
+    if( m_patch ) {
+        disconnect( m_patch, SIGNAL( patchChanged() ), this, SLOT( notifyPatchChanged() ) );
+        if ( qobject_cast<LocalPatchSource*>( m_patch ) ) {
+            // make sure we don't leak this
+            // TODO: what about other patch sources?
+            m_patch->deleteLater();
+        }
+    }
 }
 
 void PatchReviewPlugin::clearPatch( QObject* _patch ) {
@@ -476,7 +483,7 @@ void PatchReviewPlugin::setPatch( IPatchSource* patch ) {
         if ( qobject_cast<LocalPatchSource*>( m_patch ) ) {
             // make sure we don't leak this
             // TODO: what about other patch sources?
-            delete m_patch;
+            m_patch->deleteLater();
         }
     }
     m_patch = patch;
