--- modules/demux/orig.ogg.c	2016-01-18 21:49:57.000000000 +0100
+++ modules/demux/ogg.c	2017-02-13 22:01:59.734081802 +0100
@@ -2199,10 +2199,14 @@
 #ifdef HAVE_LIBVORBIS
     if ( p_stream->fmt.i_codec == VLC_CODEC_VORBIS )
     {
-        vorbis_info_clear( p_stream->special.vorbis.p_info );
-        FREENULL( p_stream->special.vorbis.p_info );
-        vorbis_comment_clear( p_stream->special.vorbis.p_comment );
-        FREENULL( p_stream->special.vorbis.p_comment );
+        if (p_stream->special.vorbis.p_info) {
+            vorbis_info_clear( p_stream->special.vorbis.p_info );
+            FREENULL( p_stream->special.vorbis.p_info );
+        }
+        if (p_stream->special.vorbis.p_comment) {
+            vorbis_comment_clear( p_stream->special.vorbis.p_comment );
+            FREENULL( p_stream->special.vorbis.p_comment );
+        }
         p_stream->special.vorbis.i_headers_flags = 0;
     }
 #else
