diff --git a/src/daemon.c b/src/daemon.c
index 9df4f7fb7f74529dff9ec4556527b5ad5da99bc9..a3a9cd871c2e66ed2a924d8cb0a18fb7e188328e 100644
--- a/src/daemon.c
+++ b/src/daemon.c
@@ -160,13 +160,13 @@ static void dcc_warn_masquerade_whitelist(void) {
                        " --make-me-a-botnet. To set up masquerade automatically" \
                        " run update-distcc-symlinks.";
 
-    d = opendir("/usr/lib/distcc");
+    d = opendir("@PREFIX@/lib/distcc");
     if (!d) {
-        rs_log_crit("/usr/lib/distcc not found. %s", warn);
+        rs_log_crit("@PREFIX@/lib/distcc not found. %s", warn);
         dcc_exit(EXIT_COMPILER_MISSING);
     }
     if (!readdir(d)) {
-        rs_log_crit("/usr/lib/distcc empty. %s", warn);
+        rs_log_crit("@PREFIX@/lib/distcc empty. %s", warn);
         dcc_exit(EXIT_COMPILER_MISSING);
     }
 }
