diff --git a/src/serve.c b/src/serve.c
index e90a6e203378fcfc9df2b98c86fbfd45d58788e0..4aee376c3e7f5fa17cebd6be4ff268a18d1675ec 100644
--- a/src/serve.c
+++ b/src/serve.c
@@ -370,6 +370,7 @@ static int dcc_check_compiler_masq(char *compiler_name)
  **/
 static int dcc_check_compiler_whitelist(char *_compiler_name)
 {
+#ifdef AT_SYMLINK_NOFOLLOW
     char *compiler_name = _compiler_name;
     int dirfd = -1;
 
@@ -386,19 +387,20 @@ static int dcc_check_compiler_whitelist(char *_compiler_name)
         return EXIT_BAD_ARGUMENTS;
     }
 
-    dirfd = open("/usr/lib/distcc", O_RDONLY);
+    dirfd = open("@PREFIX@/lib/distcc", O_RDONLY);
     if (dirfd < 0) {
         if (errno == ENOENT)
-            rs_log_crit("no %s", "/usr/lib/distcc");
+            rs_log_crit("no %s", "@PREFIX@/lib/distcc");
         return EXIT_DISTCC_FAILED;
     }
 
     if (faccessat(dirfd, compiler_name, X_OK, 0) < 0) {
-        rs_log_crit("%s not in %s whitelist.", compiler_name, "/usr/lib/distcc");
+        rs_log_crit("%s not in %s whitelist.", compiler_name, "@PREFIX@/lib/distcc");
         return EXIT_BAD_ARGUMENTS;           /* ENOENT, EACCESS, etc */
     }
 
-    rs_trace("%s in /usr/lib/distcc whitelist", compiler_name);
+    rs_trace("%s in @PREFIX@/lib/distcc whitelist", compiler_name);
+#endif
     return 0;
 }
 
