diff --git ext/qt/gstqsgtexture.cc ext/qt/orig.gstqsgtexture.cc
index ad1425e..2b314e0 100644
--- ext/qt/orig.gstqsgtexture.cc
+++ ext/qt/gstqsgtexture.cc
@@ -35,7 +35,7 @@ GST_DEBUG_CATEGORY_STATIC (GST_CAT_DEFAULT);
 
 GstQSGTexture::GstQSGTexture ()
 {
-  static volatile gsize _debug;
+  static gsize _debug;
 
   initializeOpenGLFunctions();
 
diff --git ext/qt/gstqtglutility.cc ext/qt/orig.gstqtglutility.cc
index 5e81388..20e8ead 100644
--- ext/qt/orig.gstqtglutility.cc
+++ ext/qt/gstqtglutility.cc
@@ -58,7 +58,7 @@ gst_qt_get_gl_display ()
 {
   GstGLDisplay *display = NULL;
   QGuiApplication *app = static_cast<QGuiApplication *> (QCoreApplication::instance ());
-  static volatile gsize _debug;
+  static gsize _debug;
 
   g_assert (app != NULL);
 
diff --git ext/qt/qtitem.cc ext/qt/orig.qtitem.cc
index 479017c..65b883e 100644
--- ext/qt/orig.qtitem.cc
+++ ext/qt/qtitem.cc
@@ -102,7 +102,7 @@ void InitializeSceneGraph::run()
 
 QtGLVideoItem::QtGLVideoItem()
 {
-  static volatile gsize _debug;
+  static gsize _debug;
 
   if (g_once_init_enter (&_debug)) {
     GST_DEBUG_CATEGORY_INIT (GST_CAT_DEFAULT, "qtglwidget", 0, "Qt GL Widget");
diff --git ext/qt/qtwindow.cc ext/qt/orig.qtwindow.cc
index c6db392..62d26b6 100644
--- ext/qt/orig.qtwindow.cc
+++ ext/qt/qtwindow.cc
@@ -103,7 +103,7 @@ QtGLWindow::QtGLWindow ( QWindow * parent, QQuickWindow *src ) :
   QQuickWindow( parent ), source (src)
 {
   QGuiApplication *app = static_cast<QGuiApplication *> (QCoreApplication::instance ());
-  static volatile gsize _debug;
+  static gsize _debug;
 
   g_assert (app != NULL);
 
@@ -152,7 +152,7 @@ QtGLWindow::beforeRendering()
 
   g_mutex_lock (&this->priv->lock);
 
-  static volatile gsize once = 0;
+  static gsize once = 0;
   if (g_once_init_enter(&once)) {
     this->priv->start = QDateTime::currentDateTime().toMSecsSinceEpoch();
     g_once_init_leave(&once,1);
