--- orig.CMakeLists.txt	2019-02-07 22:39:50.243336916 +0100
+++ CMakeLists.txt	2019-02-07 22:40:02.753336916 +0100
@@ -48,7 +48,7 @@
     QUIET
 )
 
-ecm_find_qmlmodule(org.kde.kcm 1.0)
+# ecm_find_qmlmodule(org.kde.kcm 1.0)
 
 find_package(KF5Kirigami2 2.1 REQUIRED)
 
@@ -58,7 +58,7 @@
 add_subdirectory(app)
 add_subdirectory(categories)
 add_subdirectory(icons)
-add_subdirectory(sidebar)
+# add_subdirectory(sidebar)
 add_subdirectory(doc)
 
 if(KF5KHtml_FOUND)
