--- app/orig.main.cpp	2018-09-04 12:25:47.000000000 +0200
+++ app/main.cpp	2019-02-07 22:07:38.963336916 +0100
@@ -59,7 +59,7 @@
     KAboutData::setApplicationData(aboutData);
 
     application.setAttribute(Qt::AA_UseHighDpiPixmaps, true);
-    application.setWindowIcon(QIcon::fromTheme(QStringLiteral("preferences-system")));
+    application.setWindowIcon(QIcon::fromTheme(QStringLiteral("preferences-system"), application.windowIcon()));
     QCommandLineParser parser;
     parser.addHelpOption();
     parser.addVersionOption();
