--- app/orig.main.cpp	2015-12-15 11:36:20.000000000 +0100
+++ app/main.cpp	2015-12-21 10:19:52.451275562 +0100
@@ -41,7 +41,7 @@
     KAboutData::setApplicationData(aboutData);
 
     application.setAttribute(Qt::AA_UseHighDpiPixmaps, true);
-    application.setWindowIcon(QIcon::fromTheme("preferences-system"));
+    application.setWindowIcon(QIcon::fromTheme("preferences-system", application.windowIcon()));
     QCommandLineParser parser;
     parser.addHelpOption();
     parser.addVersionOption();
--- app/orig.SettingsBase.cpp	2015-12-15 11:36:20.000000000 +0100
+++ app/SettingsBase.cpp	2015-12-21 10:19:30.575274650 +0100
@@ -53,7 +53,7 @@
     // Prepare the view area
     stackedWidget = new QStackedWidget( this );
     setWindowTitle(i18n("System Settings"));
-    setWindowIcon(QIcon::fromTheme("preferences-system"));
+    setWindowIcon(QIcon::fromTheme("preferences-system", windowIcon()));
     setCentralWidget(stackedWidget);
     setWindowFlags( windowFlags() | Qt::WindowContextHelpButtonHint );
     // Initialise search
