diff --git a/cmd/os/macos/mount_zfs/Makefile.am b/cmd/os/macos/mount_zfs/Makefile.am
index 6c2dd63e5..01618e7d2 100644
--- a/cmd/os/macos/mount_zfs/Makefile.am
+++ b/cmd/os/macos/mount_zfs/Makefile.am
@@ -2,7 +2,7 @@
 # Linux steals the "mount_zfs" name first, and since it's one
 # giant Makefile you can no longer have two tools with the same name.
 #
-bin_PROGRAMS += macos_mount_zfs
+sbin_PROGRAMS += macos_mount_zfs
 CPPCHECKTARGETS += macos_mount_zfs
 
 AM_CPPFLAGS += -I$(top_srcdir)/lib/libzfs
