diff --git a/include/os/macos/spl/sys/debug.h b/include/os/macos/spl/sys/debug.h
index 2ed7ccba0..4c0120159 100644
--- a/include/os/macos/spl/sys/debug.h
+++ b/include/os/macos/spl/sys/debug.h
@@ -263,7 +263,7 @@ __attribute__((noinline)) int assfail(const char *str, const char *file,
 #endif /* MACOS_ASSERT_SHOULD_PANIC */
 #endif /* NDEBUG */
 
-#define	zfs_fallthrough		__attribute__((__fallthrough__))
+#define	zfs_fallthrough		[[fallthrough]]
 
 #ifdef  __cplusplus
 }
diff --git a/lib/libspl/include/sys/feature_tests.h b/lib/libspl/include/sys/feature_tests.h
index 396052f12..72460e72f 100644
--- a/lib/libspl/include/sys/feature_tests.h
+++ b/lib/libspl/include/sys/feature_tests.h
@@ -31,7 +31,7 @@
 
 #if !defined(zfs_fallthrough) && !defined(_LIBCPP_VERSION)
 #if defined(HAVE_IMPLICIT_FALLTHROUGH)
-#define	zfs_fallthrough		__attribute__((__fallthrough__))
+#define	zfs_fallthrough		[[fallthrough]]
 #else
 #define	zfs_fallthrough		((void)0)
 #endif
