--- orig.kcalc_button.cpp	2016-05-24 17:17:45.000000000 +0200
+++ kcalc_button.cpp	2016-05-24 17:20:50.000000000 +0200
@@ -41,6 +41,7 @@
 
 	// use preferred size policy for vertical
 	setSizePolicy(QSizePolicy::Minimum, QSizePolicy::Preferred);
+	setAttribute(Qt::WA_LayoutUsesWidgetRect);
 }
 
 //------------------------------------------------------------------------------
@@ -54,6 +55,7 @@
 
 	// use preferred size policy for vertical
 	setSizePolicy(QSizePolicy::Minimum, QSizePolicy::Preferred);
+	setAttribute(Qt::WA_LayoutUsesWidgetRect);
 }
 
 //------------------------------------------------------------------------------
