diff --git plugins/cppcheck/parameters.h plugins/cppcheck/parameters.h
index 672dbb373b..0867100de0 100644
--- plugins/cppcheck/parameters.h
+++ plugins/cppcheck/parameters.h
@@ -35,7 +35,7 @@ namespace defaults
 {
 
 // global settings
-static const QString executablePath = QStringLiteral("/usr/bin/cppcheck");
+static const QString executablePath = QStringLiteral("@PREFIX@/bin/cppcheck");
 static const bool hideOutputView = true;
 static const bool showXmlOutput = false;
 

