diff --git plugins/CMakeLists.txt plugins/CMakeLists.txt
index b1e6cff6813e16a8a3360ce8c2a6847b6f2cb20c..f8c12c1deabc79623ad7119e7028c5edc14b75ef 100644
--- plugins/CMakeLists.txt
+++ plugins/CMakeLists.txt
@@ -86,14 +86,16 @@ ecm_optional_add_subdirectory(cvs)
 ecm_optional_add_subdirectory(git)
 ecm_optional_add_subdirectory(perforce)
 
-find_package(SubversionLibrary)
-set_package_properties(SubversionLibrary PROPERTIES
-    PURPOSE "Support for Subversion integration"
-    URL "http://subversion.tigris.org"
-    TYPE OPTIONAL)
+if (NOT IGNORE_SUBVERSION)
+    find_package(SubversionLibrary)
+    set_package_properties(SubversionLibrary PROPERTIES
+        PURPOSE "Support for Subversion integration"
+        URL "http://subversion.tigris.org"
+        TYPE OPTIONAL)
 
-if(SubversionLibrary_FOUND)
-    ecm_optional_add_subdirectory(subversion)
+    if(SubversionLibrary_FOUND)
+        ecm_optional_add_subdirectory(subversion)
+    endif()
 endif()
 
 add_subdirectory(vcschangesview)
