diff --git plugins/clang/tests/test_clangutils.cpp plugins/clang/tests/test_clangutils.cpp
index 723b9dff0a68d57774bf28406a3c05cdf28a3eaa..3ff839ce54ef731b71c0df59d2a76ca1882a05d4 100644
--- plugins/clang/tests/test_clangutils.cpp
+++ plugins/clang/tests/test_clangutils.cpp
@@ -271,6 +271,8 @@ void TestClangUtils::testGetRawContents_data()
 void TestClangUtils::testRangeForIncludePathSpec()
 {
     QCOMPARE(ClangUtils::rangeForIncludePathSpec("#include <vector>"), KTextEditor::Range(0, 10, 0, 16));
+    QCOMPARE(ClangUtils::rangeForIncludePathSpec("# include <vector>"), KTextEditor::Range(0, 11, 0, 17));
+    QCOMPARE(ClangUtils::rangeForIncludePathSpec("#\t include <vector>"), KTextEditor::Range(0, 12, 0, 18));
     QCOMPARE(ClangUtils::rangeForIncludePathSpec("#include <foo\\\".h>"), KTextEditor::Range(0, 10, 0, 17));
     QCOMPARE(ClangUtils::rangeForIncludePathSpec("#include \"foo\\\".h\""), KTextEditor::Range(0, 10, 0, 17));
     QCOMPARE(ClangUtils::rangeForIncludePathSpec("#include \"foo<>.h\""), KTextEditor::Range(0, 10, 0, 17));
diff --git plugins/clang/util/clangutils.cpp plugins/clang/util/clangutils.cpp
index 121d1d341a9e8cb62ed194f2854d737a9058a3ed..43223bfea10931690b624aa87d369d61948e6d10 100644
--- plugins/clang/util/clangutils.cpp
+++ plugins/clang/util/clangutils.cpp
@@ -78,7 +78,7 @@ QVector<UnsavedFile> ClangUtils::unsavedFiles()
 
 KTextEditor::Range ClangUtils::rangeForIncludePathSpec(const QString& line, const KTextEditor::Range& originalRange)
 {
-    static const QRegularExpression pattern(QStringLiteral("^\\s*(#include|#import)"));
+    static const QRegularExpression pattern(QStringLiteral("^\\s*(#\\s*include|#\\s*import)"));
     if (!line.contains(pattern)) {
         return KTextEditor::Range::invalid();
     }
