diff --git kdevplatform/documentation/CMakeLists.txt kdevplatform/documentation/CMakeLists.txt
index 65b9414..c29287b 100644
--- kdevplatform/documentation/CMakeLists.txt
+++ kdevplatform/documentation/CMakeLists.txt
@@ -1,6 +1,8 @@
 add_definitions(-DTRANSLATION_DOMAIN=\"kdevplatform\")
 
-find_package(Qt5WebEngineWidgets CONFIG)
+if(NOT USE_QTWEBKIT)
+    find_package(Qt5WebEngineWidgets CONFIG)
+endif()
 if(TARGET Qt5::WebEngineWidgets)
     set_package_properties(Qt5WebEngineWidgets PROPERTIES
         PURPOSE "QtWebEngine, for integrated documentation"
