diff --git debuggers/gdb/CMakeLists.txt debuggers/gdb/CMakeLists.txt
index 2b59866..e78c354 100644
--- debuggers/gdb/CMakeLists.txt
+++ debuggers/gdb/CMakeLists.txt
@@ -13,11 +13,13 @@ function(add_debuggable_executable target)
     set_target_properties(${target} PROPERTIES COMPILE_FLAGS "${CMAKE_CXX_FLAGS_DEBUG} ${_flags}")
 endfunction()
 
-find_package(KF5SysGuard)
-if(KF5SysGuard_FOUND)
-    add_definitions(
-        -DKDEV_ENABLE_GDB_ATTACH_DIALOG
-    )
+if(NOT APPLE)
+    find_package(KF5SysGuard)
+    if(KF5SysGuard_FOUND)
+        add_definitions(
+            -DKDEV_ENABLE_GDB_ATTACH_DIALOG
+        )
+    endif()
 endif()
 
 if (CMAKE_VERSION VERSION_GREATER "2.9" OR NOT CMAKE_GENERATOR MATCHES "Ninja")
