--- thumbnail/CMakeLists.txt.orig	2017-06-16 10:08:37.921890824 +0200
+++ thumbnail/CMakeLists.txt	2017-06-16 10:10:02.696891468 +0200
@@ -94,7 +94,7 @@
 
 ############ next target ###############
 
-find_package(Qt5WebEngineWidgets 5.7.0 QUIET)
+#find_package(Qt5WebEngineWidgets 5.7.0 QUIET)
 set_package_properties(Qt5WebEngineWidgets PROPERTIES
                        TYPE OPTIONAL
                        PURPOSE "Provides support for HTML previews in the thumbnail ioslave"
