--- grantlee-0.5.1/orig.CMakeLists.txt	2014-09-22 13:01:25.000000000 +0200
+++ grantlee-0.5.1/CMakeLists.txt	2015-12-18 20:45:29.000000000 +0100
@@ -96,7 +96,9 @@
 set( LIB_INSTALL_DIR lib${LIB_SUFFIX} )
 set( PLUGIN_INSTALL_DIR lib${LIB_SUFFIX}/grantlee/${Grantlee_MAJOR_MINOR_VERSION_STRING} )
 set( BIN_INSTALL_DIR bin )
-set( INCLUDE_INSTALL_DIR include )
+if( NOT INCLUDE_INSTALL_DIR )
+  set( INCLUDE_INSTALL_DIR include )
+endif()
 set( DATA_INSTALL_DIR share/apps )
 
 # set up RPATH/install_name_dir
