--- src/orig.kpassworddialog.cpp	2015-12-06 22:41:24.000000000 +0100
+++ src/kpassworddialog.cpp	2015-12-20 21:28:04.000000000 +0100
@@ -69,7 +69,7 @@
     : QDialog(parent), d(new KPasswordDialogPrivate(this))
 {
     setWindowTitle(tr("Password"));
-    setWindowIcon(QIcon::fromTheme(QStringLiteral("dialog-password")));
+    setWindowIcon(QIcon::fromTheme(QStringLiteral("dialog-password"), windowIcon()));
     d->m_flags = flags;
     d->init();
 }
