diff --git a/lprof.qrc b/lprof.qrc
index 1f62b55..8b8fe5e 100755
--- a/lprof.qrc
+++ b/lprof.qrc
@@ -9,9 +9,5 @@
         <file>src/images/Norman_Koren-small.png</file>
         <file>src/images/logo.png</file>
         <file alias="pcheckerbackgr.png">src/images/pcheckerbackgr.png</file>
-        <file alias="lprof_de.qm">translations/lprof_de.qm</file>
-        <file alias="lprof_fr.qm">translations/lprof_fr.qm</file>
-        <file alias="lprof_no.qm">translations/lprof_no.qm</file>
-        <file alias="lprof_ru.qm">translations/lprof_ru.qm</file>
     </qresource>
     </RCC> 
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index b8784b5..07dad20 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -7,6 +7,34 @@ ADD_SUBDIRECTORY(3rdparty)
 # Built ICCtoIT8 executable
 ADD_SUBDIRECTORY(ICCtoIT8)
 
+# the lprof code would REALLY benefit from a shared subdir and collected headers.
+# For now, contaminate the build tree with the header search path.
+SET(lprof_inc_dirs)
+FOREACH(lprof_inc_dir checkerqt gammaqt IDqt lprofqt moncalcontrolqt moncalqt monqt parmsqt reference_inst_qt libqtlcmswidgets liblprof videoLUTutils lcms/include)
+    SET(lprof_inc_dirs "${lprof_inc_dirs};${CMAKE_SOURCE_DIR}/src/${lprof_inc_dir}")
+ENDFOREACH(lprof_inc_dir)
+
+# Normally these are set for us. Since we're overriding the normal include path we've got to add them ourselves, but it's
+# probably a good idea to be explicit about what we use anyway.
+SET(lprof_qt_incs "${QT_INCLUDE_DIR};${QT_QT_INCLUDE_DIR};${QT_QTCORE_INCLUDE_DIR};${QT_QTGUI_INCLUDE_DIR};${QT_QTSVG_INCLUDE_DIR};${QT_QT3SUPPORT_INCLUDE_DIR};${QT_QTASSISTANT_INCLUDE_DIR}")
+
+
+IF(NOT LIBVIGRAIMPEX_FOUND)
+    # We'll have to use our local copy
+    SET(impex_lib impexlocal)
+    SET(impex_include "${CMAKE_CURRENT_SOURCE_DIR}/3rdparty/vigra/include")
+    SET(lprof_compile_flags "${lprof_compile_flags} -DVIGRA_STATIC_LIB")
+ELSE(NOT LIBVIGRAIMPEX_FOUND)
+    SET(impex_lib "${LIBVIGRAIMPEX_LIBRARIES}")
+    SET(impex_include "${LIBVIGRAIMPEX_INCLUDE_DIR}")
+    # Static/nonstatic flag should've been added by find script
+ENDIF(NOT LIBVIGRAIMPEX_FOUND)
+
+SET_DIRECTORY_PROPERTIES(
+    PROPERTIES
+    INCLUDE_DIRECTORIES "${lprof_inc_dirs};${lprof_qt_incs};${ARGYLL_INCLUDE_DIRS};${impex_include};${LIBUSB_INCLUDE_DIR};${X11_INCLUDE_DIR};${ZLIB_INCLUDE_DIR};${JPEG_INCLUDE_DIR};${TIFF_INCLUDE_DIR};${CMAKE_CURRENT_BINARY_DIR};${WINDDK_INCLUDE_DIR}"
+)
+
 # Determine how to build videoLUTutils
 IF(WIN32)
     SET(VIDEOLUT_SOURCES "videoLUTutils/LUTutil-NT.cpp")
@@ -120,34 +148,6 @@ IF(MSVC)
     SET(lprof_link_flags "/NODEFAULTLIB:LIBCMTD;LIBCMT")
 ENDIF(MSVC)
 
-# the lprof code would REALLY benefit from a shared subdir and collected headers.
-# For now, contaminate the build tree with the header search path.
-SET(lprof_inc_dirs)
-FOREACH(lprof_inc_dir checkerqt gammaqt IDqt lprofqt moncalcontrolqt moncalqt monqt parmsqt reference_inst_qt libqtlcmswidgets liblprof videoLUTutils lcms/include)
-    SET(lprof_inc_dirs "${lprof_inc_dirs};${CMAKE_SOURCE_DIR}/src/${lprof_inc_dir}")
-ENDFOREACH(lprof_inc_dir)
-
-# Normally these are set for us. Since we're overriding the normal include path we've got to add them ourselves, but it's
-# probably a good idea to be explicit about what we use anyway.
-SET(lprof_qt_incs "${QT_INCLUDE_DIR};${QT_QT_INCLUDE_DIR};${QT_QTCORE_INCLUDE_DIR};${QT_QTGUI_INCLUDE_DIR};${QT_QTSVG_INCLUDE_DIR};${QT_QT3SUPPORT_INCLUDE_DIR};${QT_QTASSISTANT_INCLUDE_DIR}")
-
-
-IF(NOT LIBVIGRAIMPEX_FOUND)
-    # We'll have to use our local copy
-    SET(impex_lib impexlocal)
-    SET(impex_include "${CMAKE_CURRENT_SOURCE_DIR}/3rdparty/vigra/include")
-    SET(lprof_compile_flags "${lprof_compile_flags} -DVIGRA_STATIC_LIB")
-ELSE(NOT LIBVIGRAIMPEX_FOUND)
-    SET(impex_lib "${LIBVIGRAIMPEX_LIBRARIES}")
-    SET(impex_include "${LIBVIGRAIMPEX_INCLUDE_DIR}")
-    # Static/nonstatic flag should've been added by find script
-ENDIF(NOT LIBVIGRAIMPEX_FOUND)
-
-SET_DIRECTORY_PROPERTIES(
-    PROPERTIES
-    INCLUDE_DIRECTORIES "${lprof_inc_dirs};${lprof_qt_incs};${ARGYLL_INCLUDE_DIRS};${impex_include};${LIBUSB_INCLUDE_DIR};${X11_INCLUDE_DIR};${ZLIB_INCLUDE_DIR};${JPEG_INCLUDE_DIR};${TIFF_INCLUDE_DIR};${CMAKE_CURRENT_BINARY_DIR};${WINDDK_INCLUDE_DIR}"
-)
-
 TARGET_LINK_LIBRARIES(lprof liblprof lcmslocal argyll ${X11_VIDMODE_LIBRARIES} ${X11_LIBRARIES} ${impex_lib} ${LIBUSB_LIBRARIES} ${QT_LIBRARIES} ${JPEG_LIBRARIES} ${TIFF_LIBRARIES} ${PNG_LIBRARIES} ${ZLIB_LIBRARIES} "-framework AppKit" ${COMMON_PLATFORM_LIBS} ${WINDDK_LIBRARIES})
 
 SET_TARGET_PROPERTIES(lprof
diff --git a/src/ICCtoIT8/CMakeLists.txt b/src/ICCtoIT8/CMakeLists.txt
index 9eb10e3..359f4f6 100644
--- a/src/ICCtoIT8/CMakeLists.txt
+++ b/src/ICCtoIT8/CMakeLists.txt
@@ -1,3 +1,8 @@
+SET_DIRECTORY_PROPERTIES(
+    PROPERTIES
+    INCLUDE_DIRECTORIES "${CMAKE_SOURCE_DIR}/src/lcms/include;${CMAKE_SOURCE_DIR}/src/lcms/include/libqtlcmswidgets;${CMAKE_SOURCE_DIR}/src/liblprof;${ARGYLL_INCLUDE_DIRS}"
+)
+
 SET(ICC2IT8_SOURCES
     getopt.c
     icc2it8.c
@@ -7,11 +12,6 @@ ADD_EXECUTABLE(icc2it8
     ${ICC2IT8_SOURCES}
 )
 
-SET_DIRECTORY_PROPERTIES(
-    PROPERTIES
-    INCLUDE_DIRECTORIES "${CMAKE_SOURCE_DIR}/src/lcms/include;${CMAKE_SOURCE_DIR}/src/liblprof;${ARGYLL_INCLUDE_DIRS}"
-)
-
 TARGET_LINK_LIBRARIES(icc2it8 liblprof lcmslocal ${COMMON_PLATFORM_LIBS})
 
 INSTALL(
diff --git a/src/videoLUTutils/LUTutil-OSX.cpp b/src/videoLUTutils/LUTutil-OSX.cpp
index 7b300ab..d1ed6f1 100644
--- a/src/videoLUTutils/LUTutil-OSX.cpp
+++ b/src/videoLUTutils/LUTutil-OSX.cpp
@@ -137,12 +137,12 @@ static ramp_type interpol(ramp_type_private ramp[], int i, int size, int rampsiz
 
 void gamma_ramp::getGammaRamp()
 {
-    CGTableCount count;
+    uint32_t count;
     CGDisplayErr error_code;
     // CGDirectDisplayID dspID = CGMainDisplayID();
 
     error_code = CGGetDisplayTransferByTable (display,
-                                              (CGTableCount) size,
+                                              (uint32_t) size,
                                               &ramp_buffer[0],
                                               &ramp_buffer[size],
                                               &ramp_buffer[size*2],
@@ -170,7 +170,7 @@ void gamma_ramp::setGammaRamp()
         }
 
     error_code = CGSetDisplayTransferByTable (display,
-                                              (CGTableCount) size,
+                                              (uint32_t) size,
                                               &ramp_buffer[0],
                                               &ramp_buffer[size],
                                               &ramp_buffer[size*2]);
diff --git a/src/videoLUTutils/LUTutil.h b/src/videoLUTutils/LUTutil.h
index 4fac31a..5ccbb41 100644
--- a/src/videoLUTutils/LUTutil.h
+++ b/src/videoLUTutils/LUTutil.h
@@ -48,6 +48,7 @@ typedef double ramp_type;
    #include <Carbon/Carbon.h>
    #include <CoreServices/CoreServices.h>
    #include <IOKit/graphics/IOGraphicsLib.h>
+   #include <CoreGraphics/CoreGraphics.h>
    typedef CGGammaValue ramp_type_private;
    typedef CGDirectDisplayID displayHandle;
 #elif defined (UNIX)
