diff --git b/src/orig.toolchain.rs a/src/toolchain.rs
index e65d9f4..79d7589 100644
--- b/src/orig.toolchain.rs
+++ a/src/toolchain.rs
@@ -257,6 +257,9 @@ impl<'a> Toolchain<'a> {
     /// Infallible function that describes the version of rustc in an installed distribution
     #[tracing::instrument(level = "trace")]
     pub fn rustc_version(&self) -> String {
+	  if env::var("RUSTUP_DONT_CALL_RUSTC").is_ok() {
+		 String::from("(RUSTUP_DONT_CALL_RUSTC set so not trying to read rustc version)")
+	  } else {
         match self.create_command("rustc") {
             Ok(mut cmd) => {
                 cmd.arg("--version");
@@ -300,6 +303,7 @@ impl<'a> Toolchain<'a> {
             }
             Err(_) => String::from("(rustc does not exist)"),
         }
+	  }
     }
 
     pub(crate) fn command(&self, binary: &str) -> anyhow::Result<Command> {
