diff --git Source/GUI/Qt/orig.mainwindow.cpp Source/GUI/Qt/mainwindow.cpp
index 1da1ec5..df59375 100644
--- Source/GUI/Qt/orig.mainwindow.cpp
+++ Source/GUI/Qt/mainwindow.cpp
@@ -542,9 +542,15 @@ void MainWindow::refreshDisplay() {
     QFont font;
     if (!settings->value("monoFont", "").toString().isEmpty()) {
         font.fromString(settings->value("monoFont", "").toString());
+#if QT_VERSION >= QT_VERSION_CHECK(5, 13, 0)
         font.setFamilies(QStringList({ font.family(), "Cascadia Mono", "Courier New", "Malgun Gothic", "MS Gothic", "NSimSun", "Mono" }));
+#endif
     } else {
+#if QT_VERSION >= QT_VERSION_CHECK(5, 13, 0)
         font.setFamilies(QStringList({ "Cascadia Mono", "Courier New", "Malgun Gothic", "MS Gothic", "NSimSun", "Mono" }));
+#else
+        font.setFamily("Cascadia Mono");
+#endif
     }
     font.setStyleHint(QFont::TypeWriter);
 
diff --git Source/GUI/Qt/orig.prefs.cpp Source/GUI/Qt/prefs.cpp
index 77393f5..3efbda2 100644
--- Source/GUI/Qt/orig.prefs.cpp
+++ Source/GUI/Qt/prefs.cpp
@@ -70,7 +70,11 @@ Preferences::Preferences(QSettings* settings, Core* C, QWidget *parent) :
     } else {
         QStringList preferredMonoFonts = { "Cascadia Mono", "Mono" };
         QFont font;
+#if QT_VERSION >= QT_VERSION_CHECK(5, 13, 0)
         font.setFamilies(preferredMonoFonts);
+#else
+        font.setFamily(preferredMonoFonts.at(0));
+#endif
         font.setStyleHint(QFont::TypeWriter);
         ui->comboBox_font->setCurrentFont(font);
         ui->comboBox_fontSize->setCurrentText(QString::number(font.pointSize()));
