--- modules/highgui/src/window_QT.cpp.orig	2016-12-23 13:54:44.000000000 +0100
+++ modules/highgui/src/window_QT.cpp	2017-01-28 12:41:25.000000000 +0100
@@ -103,7 +103,7 @@
     float       dx;//spacing letter in Qt (0 default) in pixel
     int         line_type;//<- pointSize in Qt
     */
-    CvFont f = {nameFont,color,style,NULL,NULL,NULL,0,0,0,weight,spacing,pointSize};
+    CvFont f = {nameFont,color,style,NULL,NULL,NULL,0,0,0,weight,static_cast<float>(spacing),pointSize};
     return f;
 }
 
