diff --git CMakeLists.txt CMakeLists.txt
index d4c39bea70640c6939c3def4530d97464cd3197a..52344581d82d07d380f866c02433921eaed200c9 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -48,7 +48,7 @@ find_package(KF5DBusAddons ${KF5_DEP_VERSION} REQUIRED)
 find_package(KF5WindowSystem ${KF5_DEP_VERSION} REQUIRED)
 
 # no X11 stuff on mac
-if (NOT APPLE)
+if (NOT APPLE OR APPLE_FORCE_X11)
     find_package(XCB MODULE COMPONENTS XCB KEYSYMS XTEST XKB)
     set_package_properties(XCB PROPERTIES DESCRIPTION "X protocol C-language Binding"
                        URL "http://xcb.freedesktop.org"
