diff --git src/runtime/plugins/CMakeLists.txt src/runtime/plugins/CMakeLists.txt
index 29e45d4d09f67a1509cac28f015e9a52f35e46b0..d01c1bd93466703c2de45ae66ecb9b468c6f72ec 100644
--- src/runtime/plugins/CMakeLists.txt
+++ src/runtime/plugins/CMakeLists.txt
@@ -1,4 +1,4 @@
-if (${XCB_XCB_FOUND} AND ${XCB_KEYSYMS_FOUND} AND ${XCB_XKB_FOUND})
+if (XCB_XCB_FOUND AND XCB_KEYSYMS_FOUND AND XCB_XKB_FOUND)
     add_subdirectory(xcb)
 endif()
 # if (APPLE)

