diff --git a/src/plugins/drumsticksoundbackend/drumsticksoundbackend.cpp b/src/plugins/drumsticksoundbackend/drumsticksoundbackend.cpp
index 72d30a6e..f5486cb 100644
--- a/src/plugins/drumsticksoundbackend/drumsticksoundbackend.cpp
+++ b/src/plugins/drumsticksoundbackend/drumsticksoundbackend.cpp
@@ -291,7 +291,7 @@ void DrumstickSoundBackend::startTimidity()
     if (!availableOutputPorts().contains(QStringLiteral("TiMidity:0"))) {
 //        qCDebug(MINUET) << "Starting TiMidity++ at" << MinuetSettings::timidityLocation().remove(QStringLiteral("file://"));
 //        m_timidityProcess.setProgram(MinuetSettings::timidityLocation().remove(QStringLiteral("file://")), QStringList() << MinuetSettings::timidityParameters());
-        m_timidityProcess.setProgram("/usr/bin/timidity", QStringList() << "-iA");
+        m_timidityProcess.setProgram("@PREFIX@/bin/timidity", QStringList());
         m_timidityProcess.start();
         if (!m_timidityProcess.waitForStarted(-1)) {
             error = m_timidityProcess.errorString();
diff --git a/src/shell/wizard.cpp b/src/shell/wizard.cpp
index 038e0e6..a361431 100644
--- a/src/shell/wizard.cpp
+++ b/src/shell/wizard.cpp
@@ -68,8 +68,8 @@ void Wizard::adjustSettings()
 {
     if (!m_timidityPath.isEmpty()) {
         MinuetSettings::setTimidityLocation(m_timidityPath);
-        if (MinuetSettings::timidityParameters().isEmpty())
-            MinuetSettings::setTimidityParameters(QStringLiteral("-iA"));
+//         if (MinuetSettings::timidityParameters().isEmpty())
+//             MinuetSettings::setTimidityParameters(QStringLiteral("-iA"));
         if (MinuetSettings::midiOutputPort().isEmpty())
             MinuetSettings::setMidiOutputPort(QStringLiteral("TiMidity:0"));
     }
@@ -91,7 +91,7 @@ void Wizard::checkSystem()
 
     item = addTreeWidgetItem(i18n("TiMidity++ configuration file"), i18n("Required for setting up TiMidity++"));
 
-    QDirIterator it(QStringLiteral("/etc"), QDirIterator::Subdirectories);
+    QDirIterator it(QStringLiteral("@PREFIX@/share/timidity"), QDirIterator::Subdirectories);
     QString timidityConfig;
     while (it.hasNext()) {
         QString file = it.next();
