--- gcc/orig.incpath.c	2017-02-20 12:38:16.000000000 +0100
+++ gcc/incpath.c	2017-02-20 12:38:50.000000000 +0100
@@ -129,6 +129,9 @@
   int relocated = cpp_relocated ();
   size_t len;
 
+  if (cxx_stdinc) {
+      add_path (xstrdup ("@PREFIX@/libexec/llvm-3.9/include/c++/v1"), SYSTEM, true, false);
+  }
   if (iprefix && (len = cpp_GCC_INCLUDE_DIR_len) != 0)
     {
       /* Look for directories that start with the standard prefix.
--- gcc/cp/orig.g++spec.c	2016-01-04 15:30:50.000000000 +0100
+++ gcc/cp/g++spec.c	2017-05-10 20:22:44.000000000 +0200
@@ -46,13 +46,17 @@
 #endif
 
 #ifndef LIBSTDCXX
-#define LIBSTDCXX "stdc++"
+#define LIBSTDCXX "c++"
+#define LIBSTDCXXABI "c++abi"
+#define LIBSUPCXX "supc++"
 #endif
-#ifndef LIBSTDCXX_PROFILE
+#undef LIBSTDCXX_PROFILE
 #define LIBSTDCXX_PROFILE LIBSTDCXX
-#endif
+
 #ifndef LIBSTDCXX_STATIC
-#define LIBSTDCXX_STATIC NULL
+// this is a hack, probably to be dropped on systems that have a new enough libc++
+// which has `operator delete(void*, unsigned long)`
+#define LIBSTDCXX_STATIC "stdc++"
 #endif
 
 void
@@ -347,6 +351,13 @@
 		       CL_DRIVER, &new_decoded_options[j]);
       added_libraries++;
       j++;
+      // add -lc++abi and -lsupc++
+      generate_option (OPT_l, LIBSTDCXXABI, 1,
+		       CL_DRIVER, &new_decoded_options[j]);
+      generate_option (OPT_l, LIBSUPCXX, 1,
+		       CL_DRIVER, &new_decoded_options[j]);
+      added_libraries += 2;
+      j += 2;
       /* Add target-dependent static library, if necessary.  */
       if ((static_link || library > 1) && LIBSTDCXX_STATIC != NULL)
 	{
