diff --git a/src/meson.build b/src/meson.build
index 42450ef..98f2698 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -185,6 +185,7 @@ if get_option('libraries')
 		# To avoid an unnecessary SONAME bump, wayland 1.x.y produces
 		# libwayland-server.so.0.x.y.
 		version: '.'.join(['0', wayland_version[1], wayland_version[2]]),
+		darwin_versions: ['2.0.0', '0.@0@.@1@'.format(wayland_version[1], wayland_version[2])],
 		dependencies: [
 			epoll_dep,
 			ffi_dep,
@@ -235,6 +236,7 @@ if get_option('libraries')
 		# To avoid an unnecessary SONAME bump, wayland 1.x.y produces
 		# libwayland-client.so.0.x.y.
 		version: '.'.join(['0', wayland_version[1], wayland_version[2]]),
+		darwin_versions: ['4.0.0', '0.@0@.@1@'.format(wayland_version[1], wayland_version[2])],
 		dependencies: [
 			epoll_dep,
 			ffi_dep,
