diff --git a/src/connection.c b/src/connection.c
index af79450..c78ae5e 100644
--- a/src/connection.c
+++ b/src/connection.c
@@ -1023,10 +1026,11 @@ wl_closure_invoke(struct wl_closure *closure, uint32_t flags,
 
 	implementation = target->implementation;
 	if (!implementation[opcode]) {
-		wl_abort("listener function for opcode %u of %s is NULL\n",
+		wl_log("listener function for opcode %u of %s is NULL\n",
 			 opcode, target->interface->name);
+	} else {
+		ffi_call(&cif, implementation[opcode], NULL, ffi_args);
 	}
-	ffi_call(&cif, implementation[opcode], NULL, ffi_args);
 
 	wl_closure_clear_fds(closure);
 }
