diff --git b/tests/orig.meson.build a/tests/meson.build
index 66337e7..5072470 100644
--- b/tests/orig.meson.build
+++ a/tests/meson.build
@@ -100,7 +100,7 @@ foreach protocol_file : protocol_files
 			code
 		],
 		link_args: [
-			'-Wl,--unresolved-symbols=ignore-all',
+			'-Wl,-flat_namespace', '-Wl,-undefined,warning',
 		],
 		dependencies: libwayland,
 		c_args: [
@@ -130,7 +130,7 @@ foreach protocol_file : protocol_files
 				client_header,
 				server_header,
 			],
-			link_args: [ '-Wl,--unresolved-symbols=ignore-all' ],
+			link_args: [ '-Wl,-flat_namespace', '-Wl,-undefined,warning' ],
 			dependencies: libwayland,
 			cpp_args: [
 				'-Wall',
