--- porg/orig.logger.cc	2014-06-19 18:58:42.000000000 +0200
+++ porg/logger.cc	2015-01-28 20:02:30.000000000 +0100
@@ -131,13 +131,22 @@
 		for (uint i(0); i < Opt::args().size(); ++i)
 			command += Opt::args()[i] + " ";
 		
+#ifdef __APPLE__
+		set_env("DYLD_FORCE_FLAT_NAMESPACE", "1");
+		set_env("DYLD_INSERT_LIBRARIES", libporg);
+#else
 		set_env("LD_PRELOAD", libporg);
+#endif
 		set_env("PORG_TMPFILE", tmpfile);
 		if (Out::debug())
 			set_env("PORG_DEBUG", "yes");
 
 		Out::dbg_title("settings");
+#ifdef __APPLE__
+		Out::dbg("DYLD_FORCE_FLAT_NAMESPACE=1 DYLD_INSERT_LIBRARIES=" + libporg); 
+#else
 		Out::dbg("LD_PRELOAD: " + libporg); 
+#endif
 		Out::dbg("INCLUDE:    " + Opt::include()); 
 		Out::dbg("EXCLUDE:    " + Opt::exclude()); 
 		Out::dbg("command:    " + command);
@@ -195,7 +204,11 @@
 //
 static string search_libporg()
 {
+#ifdef __APPLE__
+	string libpath(LIBDIR "/libporg-log.dylib");
+#else
 	string libpath(LIBDIR "/libporg-log.so");
+#endif
 	struct stat s;
 	
 	if (!stat(libpath.c_str(), &s))
@@ -204,7 +217,11 @@
 	glob_t g;
 	memset(&g, 0, sizeof(g));
 	
+#ifdef __APPLE__
+	if (!glob(LIBDIR "/libporg-log.[0-9]*.dylib", 0, 0, &g) && g.gl_pathc)
+#else
 	if (!glob(LIBDIR "/libporg-log.so.[0-9]*", 0, 0, &g) && g.gl_pathc)
+#endif
 		libpath = g.gl_pathv[0];
 	
 	globfree(&g);
--- scripts/orig.paco2porg	2014-06-19 22:04:42.000000000 +0200
+++ scripts/paco2porg	2015-01-28 20:11:14.000000000 +0100
@@ -49,7 +49,7 @@
 	logdir=
 	for dir in /etc /usr/local/etc /usr/etc /opt/etc; do
 		if [ -f $dir/${p}rc ]; then
-			logdir=$(sed -n '/^LOGDIR=/ s///p' $dir/${p}rc 2>/dev/null)
+			logdir=$(ssed -n '/^LOGDIR=/ s///p' $dir/${p}rc 2>/dev/null)
 			break;
 		fi
 	done
@@ -62,7 +62,7 @@
 	case $1 in
 		-v|--verbose) 	opt_verb=-v;;
 		-f|--force)		opt_force=-f;;
-		-V|--version)	porg --version | sed "s/^porg/$me/"; exit 0;;
+		-V|--version)	porg --version | ssed "s/^porg/$me/"; exit 0;;
 		--paco-logdir=*)
 			paco_dir=${1#*=}
 			[ "$paco_dir" ] || die "Option '${1%=*}' requires an argument"
@@ -117,24 +117,24 @@
 
 	echo '#!'$porg_version'' > $porg_log
 	
-	sed -n 's/^#d:/#t:/p' $paco_log >> $porg_log
+	ssed -n 's/^#d:/#t:/p' $paco_log >> $porg_log
 
-	echo "#a:$(sed -n '/^#:Author: */  s///p' $paco_log)" >> $porg_log
-	echo "#S:$(sed -n '/^#:Summary: */ s///p' $paco_log)" >> $porg_log
-	echo "#u:$(sed -n '/^#:URL: */     s///p' $paco_log)" >> $porg_log
-	echo "#l:$(sed -n '/^#:License: */ s///p' $paco_log)" >> $porg_log
+	echo "#a:$(ssed -n '/^#:Author: */  s///p' $paco_log)" >> $porg_log
+	echo "#S:$(ssed -n '/^#:Summary: */ s///p' $paco_log)" >> $porg_log
+	echo "#u:$(ssed -n '/^#:URL: */     s///p' $paco_log)" >> $porg_log
+	echo "#l:$(ssed -n '/^#:License: */ s///p' $paco_log)" >> $porg_log
 
 	(grep '^#c:' $paco_log || echo '#c:') >> $porg_log
 	(grep '^#i:' $paco_log || echo '#i:') >> $porg_log
 
 	desc='^#:Description$'
-	sed -n "/^#:/ { /$desc/,\$ { /$desc/! s/#:/#d:/p } }" $paco_log \
-	| sed '1 { /^#d:$/d }; $ { /^#d:$/d }' >> $porg_log
+	ssed -n "/^#:/ { /$desc/,\$ { /$desc/! s/#:/#d:/p } }" $paco_log \
+	| ssed '1 { /^#d:$/d }; $ { /^#d:$/d }' >> $porg_log
 	echo '#d:' >> $porg_log
 
 	# 2) Logged files
 
-	for buf in $(sed -n '/^-\?\// s//\//p' $paco_log); do
+	for buf in $(ssed -n '/^-\?\// s//\//p' $paco_log); do
 		
 		path=${buf%%|*}
 		size=0
