--- doc/orig.Makefile.in	2016-05-17 21:50:52.000000000 +0200
+++ doc/Makefile.in	2017-12-28 13:06:46.304112753 +0100
@@ -331,16 +331,16 @@
 @ENABLE_GROP_TRUE@	grop.desktop
 
 @ENABLE_GROP_TRUE@grop_appsdir = \
-@ENABLE_GROP_TRUE@	$(DESTDIR)$(datadir)/applications
+@ENABLE_GROP_TRUE@	$(datadir)/applications
 
 @ENABLE_GROP_TRUE@grop_icons_DATA = \
 @ENABLE_GROP_TRUE@	grop.png
 
 @ENABLE_GROP_TRUE@grop_iconsdir = \
-@ENABLE_GROP_TRUE@	$(DESTDIR)$(datadir)/pixmaps
+@ENABLE_GROP_TRUE@	$(datadir)/pixmaps
 
 @ENABLE_GROP_FALSE@porg_icon_path = \
-@ENABLE_GROP_FALSE@	$(DESTDIR)$(pkgdatadir)/porg.png
+@ENABLE_GROP_FALSE@	$(pkgdatadir)/porg.png
 
 @ENABLE_GROP_TRUE@porg_icon_path = \
 @ENABLE_GROP_TRUE@	$(grop_iconsdir)/grop.png
--- doc/orig.Makefile.am	2015-04-14 09:13:15.000000000 +0200
+++ doc/Makefile.am	2017-12-28 13:06:24.151297244 +0100
@@ -40,13 +40,13 @@
 	grop.desktop
 
 grop_appsdir = \
-	$(DESTDIR)$(datadir)/applications
+	$(datadir)/applications
 	
 grop_icons_DATA = \
 	grop.png
 
 grop_iconsdir = \
-	$(DESTDIR)$(datadir)/pixmaps
+	$(datadir)/pixmaps
 
 logme_files += \
 	$(grop_appsdir)/grop.desktop \
@@ -58,7 +58,7 @@
 else
 
 porg_icon_path = \
-	$(DESTDIR)$(pkgdatadir)/porg.png
+	$(pkgdatadir)/porg.png
 
 endif
 
