From 9dce698faa8c4d274389b933717aea475910ca4b Mon Sep 17 00:00:00 2001
From: Aaron Madlon-Kay <aaron@madlon-kay.com>
Date: Wed, 13 Nov 2019 16:34:46 +0900
Subject: [PATCH] Changes for MacPorts

---
 CMakeLists.txt             | 18 +++++++++++-------
 cmake/FindQScintilla.cmake |  2 +-
 2 files changed, 12 insertions(+), 8 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 04e8a828..ee00b942 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -76,6 +76,8 @@
     set(CMAKE_BUILD_TYPE "Release")
 endif()
 
+include(FeatureSummary)
+
 if(MSVC)
     if(CMAKE_CL_64)
         # Paths for 64-bit windows builds
@@ -428,7 +430,7 @@
 
 #enable version check for macOS
 if(APPLE)
-    add_definitions(-DCHECKNEWVERSION)
+    #add_definitions(-DCHECKNEWVERSION)
 endif()
 
 # SQLCipher option
@@ -659,3 +661,5 @@
 endif()
 set(CPACK_GENERATOR ${CPACK_DEFAULT_GEN} CACHE STRING "CPack pkg type(s) to generate")
 include(CPack)
+
+feature_summary(WHAT ALL INCLUDE_QUIET_PACKAGES FATAL_ON_MISSING_REQUIRED_PACKAGES)
# diff --git a/cmake/FindQScintilla.cmake b/cmake/FindQScintilla.cmake
# index f469637c..dd46e49b 100644
# --- a/cmake/FindQScintilla.cmake
# +++ b/cmake/FindQScintilla.cmake
# @@ -80,7 +80,7 @@ endif ()
#  
#  
#  find_library ( QSCINTILLA_LIBRARY
# -  NAMES qscintilla qscintilla2 libqscintilla2
# +  NAMES qscintilla qscintilla2 libqscintilla2 qscintilla2_qt5 libqscintilla2_qt5
#    HINTS ${QT_LIBRARY_DIR}
#  )
#  
# -- 
# 2.24.0
# 
