diff --git CMakeLists.txt CMakeLists.txt
index a306ad9..d21da41 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -13,6 +13,8 @@ endif()
 
 add_definitions(-std=c++11)
 
+include(FeatureSummary)
+
 if(WIN32 AND MSVC)
 	project("DB Browser for SQLite")
 	if(CMAKE_CL_64)
@@ -399,7 +401,7 @@ if(ENABLE_TESTING)
 	add_subdirectory(src/tests)
 endif()
 
-if(UNIX AND NOT APPLE)
+if(UNIX)
 	install(FILES src/icons/${PROJECT_NAME}.png
 		DESTINATION share/icons/hicolor/256x256/apps/)
 	
@@ -408,7 +410,7 @@ if(UNIX AND NOT APPLE)
 
 	install(FILES distri/${PROJECT_NAME}.desktop.appdata.xml
 		DESTINATION share/appdata/)
-endif(UNIX AND NOT APPLE)
+endif(UNIX)
 
 if(WIN32 AND MSVC)
 	install(TARGETS ${PROJECT_NAME}
@@ -514,3 +516,5 @@ else(WIN32 AND NOT UNIX)
 endif(WIN32 AND NOT UNIX)
 set(CPACK_PACKAGE_EXECUTABLES "DB Browser for SQLite" "DB Browser for SQLite")
 include(CPack)
+
+feature_summary(WHAT ALL INCLUDE_QUIET_PACKAGES FATAL_ON_MISSING_REQUIRED_PACKAGES)
diff --git cmake/FindQScintilla.cmake cmake/FindQScintilla.cmake
index f469637..b4e14bd 100644
--- cmake/FindQScintilla.cmake
+++ cmake/FindQScintilla.cmake
@@ -80,7 +80,7 @@ endif ()
 
 
 find_library ( QSCINTILLA_LIBRARY
-  NAMES qscintilla qscintilla2 libqscintilla2
+  NAMES qscintilla qscintilla2 libqscintilla2 libqscintilla2_qt5
   HINTS ${QT_LIBRARY_DIR}
 )
 
