diff --git libusb/os/orig.darwin_usb.c libusb/os/darwin_usb.c
index a06e1d7..32f272d 100644
--- libusb/os/orig.darwin_usb.c
+++ libusb/os/darwin_usb.c
@@ -1130,6 +1130,10 @@ static enum libusb_error darwin_cache_device_descriptor (struct libusb_context *
   (*device)->GetDeviceProduct (device, &idProduct);
   (*device)->GetDeviceVendor (device, &idVendor);
 
+  // RJVB: ignore my Behringer UMC204HD which reboots
+  if (idVendor == 0x1397 && idProduct == 0x0508) {
+      return darwin_to_libusb(kIOReturnNotPermitted);
+  }
   /* According to Apple's documentation the device must be open for DeviceRequest but we may not be able to open some
    * devices and Apple's USB Prober doesn't bother to open the device before issuing a descriptor request.  Still,
    * to follow the spec as closely as possible, try opening the device */
