diff --git org.kde.desktop/Button.qml org.kde.desktop/Button.qml
index ae82523..096ab31 100644
--- org.kde.desktop/Button.qml
+++ org.kde.desktop/Button.qml
@@ -58,7 +58,8 @@ T.Button {
         hasFocus: controlRoot.activeFocus
         activeControl: controlRoot.isDefault ? "default" : "f"
         properties: {
-            "icon": controlRoot.icon ? (controlRoot.icon.name || controlRoot.icon.source) : "",
+            //"icon": controlRoot.icon ? (controlRoot.icon.name || controlRoot.icon.source) : "",
+            "icon": controlRoot.icon ? (controlRoot.icon.source) : "",
             "iconColor": controlRoot.icon && controlRoot.icon.color.a > 0? controlRoot.icon.color : Kirigami.Theme.textColor,
             "iconWidth": controlRoot.icon && controlRoot.icon.width ? controlRoot.icon.width : 0,
             "iconHeight": controlRoot.icon && controlRoot.icon.height ? controlRoot.icon.height : 0,
diff --git org.kde.desktop/MenuItem.qml org.kde.desktop/MenuItem.qml
index e3d4d93..65899d7 100644
--- org.kde.desktop/MenuItem.qml
+++ org.kde.desktop/MenuItem.qml
@@ -66,8 +66,10 @@ T.MenuItem {
         }
         Kirigami.Icon {
             Layout.alignment: Qt.AlignVCenter
-            visible: (controlRoot.ListView.view && controlRoot.ListView.view.hasIcons) || (controlRoot.icon != undefined && (controlRoot.icon.name.length > 0 || controlRoot.icon.source.length > 0))
-            source: controlRoot.icon ? (controlRoot.icon.name || controlRoot.icon.source) : ""
+            //visible: (controlRoot.ListView.view && controlRoot.ListView.view.hasIcons) || (controlRoot.icon != undefined && (controlRoot.icon.name.length > 0 || controlRoot.icon.source.length > 0))
+            //source: controlRoot.icon ? (controlRoot.icon.name || controlRoot.icon.source) : ""
+            visible: (controlRoot.ListView.view && controlRoot.ListView.view.hasIcons) || (controlRoot.icon != undefined && (controlRoot.icon.source.length > 0))
+            source: controlRoot.icon ? (controlRoot.icon.source) : ""
             color: controlRoot.icon ? controlRoot.icon.color : "transparent"
             //hovered is for retrocompatibility
             selected: (controlRoot.highlighted || controlRoot.hovered)
diff --git org.kde.desktop/TabButton.qml org.kde.desktop/TabButton.qml
index 5cc932e..c21515b 100644
--- org.kde.desktop/TabButton.qml
+++ org.kde.desktop/TabButton.qml
@@ -88,7 +88,8 @@ T.TabButton {
             "orientation": orientation,
             "tabpos": tabpos,
             "selectedpos": selectedpos,
-            "icon": control.icon ? (control.icon.name || control.icon.source) : "",
+            //"icon": control.icon ? (control.icon.name || control.icon.source) : "",
+            "icon": control.icon ? control.icon.source : "",
             "iconColor": controlRoot.icon && controlRoot.icon.color.a > 0? controlRoot.icon.color : Kirigami.Theme.textColor
         }
 
diff --git org.kde.desktop/ToolButton.qml org.kde.desktop/ToolButton.qml
index bb5db78..fcdcc12 100644
--- org.kde.desktop/ToolButton.qml
+++ org.kde.desktop/ToolButton.qml
@@ -60,7 +60,8 @@ T.ToolButton {
         hasFocus: false
         activeControl: controlRoot.isDefault ? "default" : "f"
         properties: {
-            "icon": controlRoot.icon ? (controlRoot.icon.name || controlRoot.icon.source) : "",
+            //"icon": controlRoot.icon ? (controlRoot.icon.name || controlRoot.icon.source) : "",
+            "icon": controlRoot.icon ? (controlRoot.icon.source) : "",
             "iconColor": controlRoot.icon && controlRoot.icon.color.a > 0? controlRoot.icon.color : Kirigami.Theme.textColor,
             "toolButtonStyle": controlRoot.display == T.ToolButton.IconOnly
                                 ? Qt.ToolButtonIconOnly :
diff --git org.kde.desktop/private/MobileTextActionsToolBar.qml org.kde.desktop/private/MobileTextActionsToolBar.qml
index e1e0043..cdda7e2 100644
--- org.kde.desktop/private/MobileTextActionsToolBar.qml
+++ org.kde.desktop/private/MobileTextActionsToolBar.qml
@@ -57,7 +57,7 @@ Popup {
     contentItem: RowLayout {
         ToolButton {
             focusPolicy: Qt.NoFocus
-            icon.name: "edit-cut"
+            //icon.name: "edit-cut"
             visible: controlRoot.selectedText.length > 0
             onClicked: {
                 controlRoot.cut();
@@ -65,7 +65,7 @@ Popup {
         }
         ToolButton {
             focusPolicy: Qt.NoFocus
-            icon.name: "edit-copy"
+            //icon.name: "edit-copy"
             visible: controlRoot.selectedText.length > 0
             onClicked: {
                 controlRoot.copy();
@@ -73,7 +73,7 @@ Popup {
         }
         ToolButton {
             focusPolicy: Qt.NoFocus
-            icon.name: "edit-paste"
+            //icon.name: "edit-paste"
             onClicked: {
                 controlRoot.paste();
             }
diff --git tests/buttonIcons.qml tests/buttonIcons.qml
index f5a9e91..2ceb58c 100644
--- tests/buttonIcons.qml
+++ tests/buttonIcons.qml
@@ -36,12 +36,12 @@ Kirigami.ApplicationWindow {
             Controls.MenuItem {
                 checkable: true
                 text: "Item1"
-                icon.name: "go-next"
+                //icon.name: "go-next"
                 icon.color: "red"
             }
             Controls.MenuItem {
                 text: "Item2"
-                icon.name: "folder-video"
+                //icon.name: "folder-video"
             }
             Controls.MenuSeparator {
             }
@@ -54,12 +54,12 @@ Kirigami.ApplicationWindow {
                 Controls.MenuItem {
                     checkable: true
                     text: "Item1"
-                    icon.name: "go-next"
+                    //icon.name: "go-next"
                     icon.color: "red"
                 }
                 Controls.MenuItem {
                     text: "Item2"
-                    icon.name: "folder-video"
+                    //icon.name: "folder-video"
                 }
                 Controls.MenuItem {
                     text: "Item3"
@@ -71,11 +71,11 @@ Kirigami.ApplicationWindow {
 
             Controls.MenuItem {
                 text: "Item1"
-                icon.name: "go-next"
+                //icon.name: "go-next"
             }
             Controls.MenuItem {
                 text: "Item2"
-                icon.name: "go-next"
+                //icon.name: "go-next"
             }
         }
     }
@@ -101,12 +101,12 @@ Kirigami.ApplicationWindow {
                 
                 Kirigami.Theme.inherit: true
                 text: "text"
-                icon.name: "go-previous"
+                //icon.name: "go-previous"
             }
             Controls.Button {
                 id: coloredIconButton
                 text: "text"
-                icon.name: "go-previous"
+                //icon.name: "go-previous"
             }
             RowLayout {
                 Controls.Label {
@@ -143,7 +143,7 @@ Kirigami.ApplicationWindow {
             }
             Controls.ToolButton {
                 text: "text"
-                icon.name: "go-previous"
+                //icon.name: "go-previous"
             }
         }
     }
