--- plugins/cmake/orig.cmakemanager.cpp	2018-12-06 10:53:25.000000000 +0100
+++ plugins/cmake/cmakemanager.cpp	2019-02-07 10:27:05.053336916 +0100
@@ -172,6 +172,7 @@
 
         server->deleteLater();
         server = nullptr;
+	   qCWarning(CMAKE) << "CMake server not available, using compile_commands.json to import" << project->name();
 
         // parse the JSON file
         CMakeImportJsonJob* job = new CMakeImportJsonJob(project, this);
