diff --git kdevplatform/documentation/documentationview.cpp kdevplatform/documentation/documentationview.cpp
index 2c6cccf8bd742e69022be8ee002ea49246c93f03..8154a7c9bb6bce240093136e2d8f5c2dd462f2a8 100644
--- kdevplatform/documentation/documentationview.cpp
+++ kdevplatform/documentation/documentationview.cpp
@@ -30,6 +30,7 @@
 #include <QShortcut>
 
 #include <KLocalizedString>
+#include <KMessageBox>
 
 #include <interfaces/icore.h>
 #include <interfaces/idocumentationprovider.h>
@@ -44,6 +45,10 @@ using namespace KDevelop;
 DocumentationView::DocumentationView(QWidget* parent, ProvidersModel* model)
     : QWidget(parent), mProvidersModel(model)
 {
+    if (ICore::self()->shuttingDown()) {
+        qCWarning(DOCUMENTATION) << "DocumentationView created during shutdown";
+    }
+
     setWindowIcon(QIcon::fromTheme(QStringLiteral("documentation"), windowIcon()));
     setWindowTitle(i18n("Documentation"));
 
@@ -62,6 +67,17 @@ DocumentationView::DocumentationView(QWidget* parent, ProvidersModel* model)
 
     mCurrent = mHistory.end();
 
+    if (ICore::self()->shuttingDown()) {
+        KMessageBox::ButtonCode ret = KMessageBox::warningYesNo(this,
+                i18n("A documentation toolview (DocumentationView) is being created during shutdown.\n"
+                    "Do you want to report this (will cause a crash)?"));
+        if (ret == KMessageBox::Yes) {
+            qFatal("DocumentationView created during shutdown");
+        } else {
+            return;
+        }
+    }
+
     setFocusProxy(mIdentifiers);
 
     QMetaObject::invokeMethod(this, "initialize", Qt::QueuedConnection);
@@ -280,7 +296,9 @@ void DocumentationView::updateView()
     if (mCurrent != mHistory.end()) {
         w = (*mCurrent)->documentationWidget(mFindDoc, this);
         Q_ASSERT(w);
-        QWidget::setTabOrder(mIdentifiers, w);
+        if (mIdentifiers->window() == w->window()) {
+            QWidget::setTabOrder(mIdentifiers, w);
+        }
     } else {
         // placeholder widget at location of doc view
         w = new QWidget(this);
