diff --git tests/units/akonadi/CMakeLists.txt tests/units/akonadi/CMakeLists.txt
index 15e887f..324dc78 100644
--- tests/units/akonadi/CMakeLists.txt
+++ tests/units/akonadi/CMakeLists.txt
@@ -5,7 +5,6 @@ zanshin_auto_tests(
   akonadiartifactqueriestest
   akonadicontextqueriestest
   akonadicontextrepositorytest
-  akonadidatasourcequeriestest
   akonadidatasourcerepositorytest
   akonadinotequeriestest
   akonadinoterepositorytest
diff --git tests/units/domain/CMakeLists.txt tests/units/domain/CMakeLists.txt
index 3b4d23c..8924279 100644
--- tests/units/domain/CMakeLists.txt
+++ tests/units/domain/CMakeLists.txt
@@ -2,11 +2,9 @@ zanshin_auto_tests(
   artifacttest
   contexttest
   datasourcetest
-  livequerytest
   mockitotest
   notetest
   projecttest
-  queryresulttest
   tagtest
   tasktest
 )
diff --git tests/units/presentation/CMakeLists.txt tests/units/presentation/CMakeLists.txt
index 3ce5a70..e669e15 100644
--- tests/units/presentation/CMakeLists.txt
+++ tests/units/presentation/CMakeLists.txt
@@ -1,19 +1,9 @@
 zanshin_auto_tests(
-  applicationmodeltest
   artifacteditormodeltest
   artifactfilterproxymodeltest
-  availablepagesmodeltest
   availablepagessortfilterproxymodeltest
-  availablesourcesmodeltest
   errorhandlertest
   errorhandlingmodelbasetest
-  inboxpagemodeltest
   metatypestest
   pagemodeltest
-  projectpagemodeltest
-  querytreemodeltest
-  tagpagemodeltest
-  tasklistmodeltest
-  contextpagemodeltest
-  workdaypagemodeltest
 )
diff --git tests/units/widgets/CMakeLists.txt tests/units/widgets/CMakeLists.txt
index ab740a3..3d4dd1c 100644
--- tests/units/widgets/CMakeLists.txt
+++ tests/units/widgets/CMakeLists.txt
@@ -1,5 +1,4 @@
 zanshin_auto_tests(
-  applicationcomponentstest
   availablepagesviewtest
   availablesourcesviewtest
   datasourcecomboboxtest
