diff --git src/akonadi/akonadiserializer.cpp src/akonadi/akonadiserializer.cpp
index 5116fa5..000dc9d 100644
--- src/akonadi/akonadiserializer.cpp
+++ src/akonadi/akonadiserializer.cpp
@@ -348,6 +348,30 @@ void Serializer::clearItem(Akonadi::Item *item)
         item->clearTag(tag);
 }
 
+#ifdef __APPLE__
+// see http://en.cppreference.com/w/cpp/algorithm/accumulate
+namespace std {
+    template<class InputIt, class T>
+    T accumulate(InputIt first, InputIt last, T init)
+    {
+        for (; first != last; ++first) {
+            init = init + *first;
+        }
+        return init;
+    }
+
+    template<class InputIt, class T, class BinaryOperation>
+    T accumulate(InputIt first, InputIt last, T init, 
+                 BinaryOperation op)
+    {
+        for (; first != last; ++first) {
+            init = op(init, *first);
+        }
+        return init;
+    }
+}
+#endif
+
 Akonadi::Item::List Serializer::filterDescendantItems(const Akonadi::Item::List &potentialChildren, const Akonadi::Item &ancestorItem)
 {
     if (potentialChildren.isEmpty())
