--- src/app/kontact_plugin.cpp
+++ src/app/kontact_plugin.cpp
@@ -25,16 +25,16 @@
 
 #include <KDE/KontactInterface/Core>
 
-EXPORT_KONTACT_PLUGIN(Plugin, zanshin)
+EXPORT_KONTACT_PLUGIN(ZanshinPlugin, zanshin)
 
-Plugin::Plugin(KontactInterface::Core *core, const QVariantList&)
+ZanshinPlugin::ZanshinPlugin(KontactInterface::Core *core, const QVariantList&)
     : KontactInterface::Plugin(core, core, "zanshin")
 {
     qputenv("ZANSHIN_KONTACT_PLUGIN", "1");
     setComponentData(KontactPluginFactory::componentData());
 }
 
-KParts::ReadOnlyPart *Plugin::createPart()
+KParts::ReadOnlyPart *ZanshinPlugin::createPart()
 {
     return loadPart();
 }
--- src/app/kontact_plugin.h
+++ src/app/kontact_plugin.h
@@ -26,12 +26,12 @@
 
 #include <KDE/KontactInterface/Plugin>
 
-class Plugin : public KontactInterface::Plugin
+class ZanshinPlugin : public KontactInterface::Plugin
 {
     Q_OBJECT
 
 public:
-    Plugin(KontactInterface::Core *core, const QVariantList &);
+    ZanshinPlugin(KontactInterface::Core *core, const QVariantList &);
 
     int weight() const { return 449; }
 
