--- orig.CMakeLists.txt	2016-07-18 16:28:24.000000000 +0200
+++ CMakeLists.txt	2016-11-23 18:39:02.105072690 +0100
@@ -14,11 +14,11 @@
 endif()
 
 string(TOLOWER "${CMAKE_BUILD_TYPE}" cmake_build_type_tolower)
-if(    NOT cmake_build_type_tolower STREQUAL "debug"
-   AND NOT cmake_build_type_tolower STREQUAL "release"
-   AND NOT cmake_build_type_tolower STREQUAL "relwithdebinfo")
-  message(FATAL_ERROR "Unknown build type \"${CMAKE_BUILD_TYPE}\". Allowed values are Debug, Release, RelWithDebInfo (case-insensitive).")
-endif()
+#if(    NOT cmake_build_type_tolower STREQUAL "debug"
+#   AND NOT cmake_build_type_tolower STREQUAL "release"
+#   AND NOT cmake_build_type_tolower STREQUAL "relwithdebinfo")
+#  message(FATAL_ERROR "Unknown build type \"${CMAKE_BUILD_TYPE}\". Allowed values are Debug, Release, RelWithDebInfo (case-insensitive).")
+#endif()
 
 
 #############################################################################
