diff --git a/src/registry2.0/portimage.tcl b/src/registry2.0/portimage.tcl
index 1da893500727eed0e0583cd02784827a7fbe8ec5..0dc4f1df05f96ebda2488d4ffc31acfbc4a04c3b 100644
--- a/src/registry2.0/portimage.tcl
+++ b/src/registry2.0/portimage.tcl
@@ -586,7 +592,8 @@ proc extract_archive_to_imagedir {location} {
         } else {
             set cmdstring "${unarchive.pipe_cmd} ( ${unarchive.cmd} ${unarchive.pre_args} ${unarchive.args} )"
         }
-        system -callback portimage::_extract_progress $cmdstring
+#         system -callback portimage::_extract_progress $cmdstring
+        system $cmdstring
     } on error {_ eOptions} {
         ::file delete -force $extractdir
         throw [dict get $eOptions -errorcode] [dict get $eOptions -errorinfo]
@@ -626,13 +633,14 @@ proc _extract_progress {event} {
     }
 }
 
-proc _progress {args} {
-    if {[macports::ui_isset ports_verbose]} {
-        return
-    }
-
-    ui_progress_generic {*}${args}
-}
+proc _progress {args} {}
+# {
+#     if {[macports::ui_isset ports_verbose]} {
+#         return
+#     }
+#
+#     ui_progress_generic {*}${args}
+# }
 
 proc _get_port_conflicts {port} {
     global registry::tdbc_connection
