diff --git a/src/port1.0/portconfigure.tcl b/src/port1.0/portconfigure.tcl
index 8591c211..89b60907 100644
--- a/src/port1.0/portconfigure.tcl
+++ b/src/port1.0/portconfigure.tcl
@@ -473,7 +473,12 @@ proc portconfigure::configure_get_universal_ldflags {} {
 
 # internal proc to determine if the compiler supports -arch
 proc portconfigure::arch_flag_supported {compiler} {
-    return [regexp {^gcc-4|llvm|apple|clang} $compiler]
+    global os.platform
+    if {${os.platform} eq "darwin"} {
+        return [regexp {^gcc-4|llvm|apple|clang} $compiler]
+    } else {
+        return 0
+    }
 }
 
 proc portconfigure::compiler_port_name {compiler} {
