--- src/doc/orig.documentvalidator.cpp	2017-04-13 08:08:16.000000000 +0200
+++ src/doc/documentvalidator.cpp	2017-04-30 01:39:25.000000000 +0200
@@ -105,7 +105,7 @@
             }
 #ifdef Q_OS_MAC
             setlocale(LC_NUMERIC_MASK, requestedLocale.toUtf8().constData());
-#elifdef Q_OS_WIN
+#elif defined(Q_OS_WIN)
             std::locale::global(std::locale(requestedLocale.toUtf8().constData()));
 #else
             setlocale(LC_NUMERIC, requestedLocale.toUtf8().constData());
