diff --git src/CMakeLists.txt src/CMakeLists.txt
index 74bd06d..d052500 100644
--- src/CMakeLists.txt
+++ src/CMakeLists.txt
@@ -227,7 +227,7 @@
 
 # Sets the icon on Windows and OSX
 file(GLOB ICONS_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/../data/icons/*-apps-kdenlive.png")
-ecm_add_app_icon(kdenlive_SRCS ICONS ${ICONS_SRCS})
+ecm_add_app_icon(kdenlive_SRCS ICONS ${CMAKE_CURRENT_SOURCE_DIR}/../data/icons/sc-apps-kdenlive.svgz ${ICONS_SRCS})
 
 qt5_add_dbus_adaptor(kdenlive_SRCS
     org.kdenlive.MainWindow.xml
