diff --git src/mainwindow.cpp src/mainwindow.cpp
index 8f20c3a..aa7f966 100644
--- src/mainwindow.cpp
+++ src/mainwindow.cpp
@@ -124,6 +124,15 @@ QMap <QString,QImage> MainWindow::m_lumacache;
     return a.first < b.first;
 }*/
 
+// determine the the default KDE style as defined BY THE USER
+// (as opposed to whatever style KDE considers default)
+static QString defaultStyle(const char *fallback=Q_NULLPTR)
+{
+    KSharedConfigPtr kdeGlobals = KSharedConfig::openConfig(QStringLiteral("kdeglobals"), KConfig::NoGlobals);
+    KConfigGroup cg(kdeGlobals, "KDE");
+    return cg.readEntry("widgetStyle", fallback);
+}
+
 MainWindow::MainWindow(const QString &MltPath, const QUrl &Url, const QString & clipsToLoad, QWidget *parent) :
     KXmlGuiWindow(parent),
     m_timelineArea(NULL),
@@ -183,12 +192,12 @@ MainWindow::MainWindow(const QString &MltPath, const QUrl &Url, const QString &
     connect(ThemeManager::instance(), SIGNAL(signalThemeChanged(const QString &)), this, SLOT(slotThemeChanged(const QString &)), Qt::DirectConnection);
 
     QString desktopStyle = QApplication::style()->objectName();
-    if (!desktopStyle.isEmpty() && QString::compare(desktopStyle, QLatin1String("breeze"), Qt::CaseInsensitive) != 0 && QString::compare(desktopStyle, QLatin1String("oxygen"), Qt::CaseInsensitive) != 0 && KdenliveSettings::widgetstyle().isEmpty()) {
-        if (availableStyles.contains(QLatin1String("breeze"), Qt::CaseInsensitive)) {
-            // Auto switch to Breeze theme
-            KdenliveSettings::setWidgetstyle(QStringLiteral("Breeze"));
-        }
-    }
+//     if (!desktopStyle.isEmpty() && QString::compare(desktopStyle, QLatin1String("breeze"), Qt::CaseInsensitive) != 0 && QString::compare(desktopStyle, QLatin1String("oxygen"), Qt::CaseInsensitive) != 0 && KdenliveSettings::widgetstyle().isEmpty()) {
+//         if (availableStyles.contains(QLatin1String("breeze"), Qt::CaseInsensitive)) {
+//             // Auto switch to Breeze theme
+//             KdenliveSettings::setWidgetstyle(QStringLiteral("Breeze"));
+//         }
+//     }
 
     if (!KdenliveSettings::widgetstyle().isEmpty() && QString::compare(desktopStyle, KdenliveSettings::widgetstyle(), Qt::CaseInsensitive) != 0) {
         // User wants a custom widget style, init
@@ -3401,7 +3410,11 @@ void MainWindow::slotChangeStyle(QAction *a)
 
 void MainWindow::doChangeStyle()
 {
-    QApplication::setStyle(QStyleFactory::create(KdenliveSettings::widgetstyle()));
+    QString newStyle = KdenliveSettings::widgetstyle();
+    if (newStyle.isEmpty() || newStyle == QStringLiteral("Default")) {
+        newStyle = defaultStyle("Breeze");
+    }
+    QApplication::setStyle(QStyleFactory::create(newStyle));
     // Changing widget style resets color theme, so update color theme again
     ThemeManager::instance()->slotChangePalette();
 }
