diff --git CMakeLists.txt CMakeLists.txt
index dfed818..1c8d17f 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -30,6 +30,8 @@ include(KDEInstallDirs)
 include(KDECMakeSettings)
 include(KDECompilerSettings)
 include(ECMOptionalAddSubdirectory)
+include(ECMMarkNonGuiExecutable)
+include(ECMAddAppIcon)
 
 add_definitions(-DQT_NO_URL_CAST_FROM_STRING)
 add_definitions(-DTRANSLATION_DOMAIN=\"kdenlive\")
diff --git renderer/CMakeLists.txt renderer/CMakeLists.txt
index cf4ef53..eff5d02 100644
--- renderer/CMakeLists.txt
+++ renderer/CMakeLists.txt
@@ -7,6 +7,7 @@ set(kdenlive_render_SRCS
 )
 
 add_executable(kdenlive_render ${kdenlive_render_SRCS})
+ecm_mark_nongui_executable(kdenlive_render)
 
 include_directories(
   
diff --git src/CMakeLists.txt src/CMakeLists.txt
index 74bd06d..d052500 100644
--- src/CMakeLists.txt
+++ src/CMakeLists.txt
@@ -214,6 +214,10 @@ if(BUILD_JogShuttle)
   )
 endif(BUILD_JogShuttle)
 
+# Sets the icon on Windows and OSX
+file(GLOB ICONS_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/../data/icons/*-apps-kdenlive.png")
+ecm_add_app_icon(kdenlive_SRCS ICONS ${ICONS_SRCS})
+
 qt5_add_dbus_adaptor(kdenlive_SRCS
   org.kdenlive.MainWindow.xml
   mainwindow.h
