diff --git plugins/cmake/cmakemanager.cpp plugins/cmake/cmakemanager.cpp
index 5c6a11fb7a9b8ca8af3699464f382fde4e91151f..59aed82648c2a23fc3026dde079c377ecbf2542d 100644
--- plugins/cmake/cmakemanager.cpp
+++ plugins/cmake/cmakemanager.cpp
@@ -170,8 +170,7 @@ private:
         Q_ASSERT(code > 0);
         Q_ASSERT(!server->isServerAvailable());
 
-        server->deleteLater();
-        server = nullptr;
+        qCWarning(CMAKE) << "CMake server not available, using compile_commands.json to import" << project->name();
 
         // parse the JSON file
         CMakeImportJsonJob* job = new CMakeImportJsonJob(project, this);
