diff --git plugins/cmake/cmakemanager.cpp plugins/cmake/cmakemanager.cpp
index 0207dc3274..e63d144309 100644
--- plugins/cmake/cmakemanager.cpp
+++ plugins/cmake/cmakemanager.cpp
@@ -170,7 +170,7 @@ private:
         Q_ASSERT(code > 0);
         Q_ASSERT(!server->isServerAvailable());
 
-        qCDebug(CMAKE) << "CMake does not provide server mode, using compile_commands.json to import" << project->name();
+        qCWarning(CMAKE) << "CMake does not provide server mode, using compile_commands.json to import" << project->name();
 
         // parse the JSON file
         CMakeImportJsonJob* job = new CMakeImportJsonJob(project, this);
