diff --git plugins/CMakeLists.txt plugins/CMakeLists.txt
index 978285d8e8..da1f9e8519 100644
--- plugins/CMakeLists.txt
+++ plugins/CMakeLists.txt
@@ -123,6 +127,7 @@ add_subdirectory(sourceformatter)
 add_subdirectory(standardoutputview)
 add_subdirectory(switchtobuddy)
 add_subdirectory(testview)
+ecm_optional_add_subdirectory(meson)
 ecm_optional_add_subdirectory(classbrowser)
 ecm_optional_add_subdirectory(executeplasmoid)
 ecm_optional_add_subdirectory(ghprovider)
