diff --git plugins/cmake/cmakemanager.cpp plugins/cmake/cmakemanager.cpp
index 2a4746fdb3bfb9324e2dc216d9ff9f8aa96a4942..7ff7155a554ddf41ec739b9b1321fb48af4edb54 100644
--- plugins/cmake/cmakemanager.cpp
+++ plugins/cmake/cmakemanager.cpp
@@ -170,7 +170,7 @@ private:
         Q_ASSERT(code > 0);
         Q_ASSERT(!server->isServerAvailable());
 
-        server.clear();
+        qCWarning(CMAKE) << "CMake server not available, using compile_commands.json to import" << project->name();
 
         // parse the JSON file
         CMakeImportJsonJob* job = new CMakeImportJsonJob(project, this);
