diff --git src/ephy-history-dialog.c src/ephy-history-dialog.c
index 7f9e9f6db..f1c0eccde 100644
--- src/ephy-history-dialog.c
+++ src/ephy-history-dialog.c
@@ -82,7 +82,7 @@ static EphyHistoryURL *
 get_url_from_row (GtkListBoxRow *row)
 {
   return ephy_history_url_new (hdy_action_row_get_subtitle (HDY_ACTION_ROW (row)),
-                               hdy_action_row_get_title (HDY_ACTION_ROW (row)),
+                               hdy_preferences_row_get_title (HDY_PREFERENCES_ROW (row)),
                                0,
                                0,
                                0);
@@ -248,7 +248,7 @@ create_row (EphyHistoryDialog *self,
 
   /* Row */
   row = hdy_action_row_new ();
-  hdy_action_row_set_title (HDY_ACTION_ROW (row), url->title);
+  hdy_preferences_row_set_title (HDY_PREFERENCES_ROW (row), url->title);
   hdy_action_row_set_subtitle (HDY_ACTION_ROW (row), url->url);
   gtk_list_box_row_set_activatable (GTK_LIST_BOX_ROW (row), TRUE);
   gtk_widget_set_tooltip_text (row, url->url);
diff --git src/preferences/cookies-dialog.c src/preferences/cookies-dialog.c
index 776050b09..0cbe4268c 100644
--- src/preferences/cookies-dialog.c
+++ src/preferences/cookies-dialog.c
@@ -135,7 +135,7 @@ cookie_add (EphyCookiesDialog *self,
 
   /* Row */
   row = hdy_action_row_new ();
-  hdy_action_row_set_title (HDY_ACTION_ROW (row), domain);
+  hdy_preferences_row_set_title (HDY_PREFERENCES_ROW (row), domain);
 
   button = gtk_button_new_from_icon_name ("user-trash-symbolic", GTK_ICON_SIZE_BUTTON);
   gtk_widget_set_valign (button, GTK_ALIGN_CENTER);
@@ -207,7 +207,7 @@ filter_func (GtkListBoxRow *row,
   gboolean result = TRUE;
 
   if (search_text)
-    result = !!strstr (hdy_action_row_get_title (HDY_ACTION_ROW (row)), search_text);
+    result = !!strstr (hdy_preferences_row_get_title (HDY_PREFERENCES_ROW (row)), search_text);
 
   if (result)
     ephy_data_dialog_set_has_search_results (EPHY_DATA_DIALOG (self), TRUE);
diff --git src/preferences/passwords-dialog.c src/preferences/passwords-dialog.c
index b07ed095a..91605c5fa 100644
--- src/preferences/passwords-dialog.c
+++ src/preferences/passwords-dialog.c
@@ -282,7 +282,7 @@ populate_model_cb (GList    *records,
 
     row = hdy_expander_row_new ();
     g_object_set_data (G_OBJECT (row), "record", record);
-    hdy_expander_row_set_title (HDY_EXPANDER_ROW (row), ephy_password_record_get_origin (record));
+    hdy_preferences_row_set_title (HDY_PREFERENCES_ROW (row), ephy_password_record_get_origin (record));
     hdy_expander_row_set_subtitle (HDY_EXPANDER_ROW (row), ephy_password_record_get_username (record));
     hdy_expander_row_set_show_enable_switch (HDY_EXPANDER_ROW (row), FALSE);
 
@@ -294,7 +294,7 @@ populate_model_cb (GList    *records,
 
     /* Username */
     sub_row = hdy_action_row_new ();
-    hdy_action_row_set_title (HDY_ACTION_ROW (sub_row), _("Username"));
+    hdy_preferences_row_set_title (HDY_PREFERENCES_ROW (sub_row), _("Username"));
     gtk_container_add (GTK_CONTAINER (row), sub_row);
 
     entry = gtk_entry_new ();
@@ -323,7 +323,7 @@ populate_model_cb (GList    *records,
 
     /* Password */
     sub_row = hdy_action_row_new ();
-    hdy_action_row_set_title (HDY_ACTION_ROW (sub_row), _("Password"));
+    hdy_preferences_row_set_title (HDY_PREFERENCES_ROW (sub_row), _("Password"));
     gtk_container_add (GTK_CONTAINER (row), sub_row);
 
     entry = gtk_entry_new ();
diff --git src/preferences/prefs-general-page.c src/preferences/prefs-general-page.c
index 7542cecd7..8a505a2bc 100644
--- src/preferences/prefs-general-page.c
+++ src/preferences/prefs-general-page.c
@@ -478,7 +478,7 @@ language_editor_add (PrefsGeneralPage *general_page,
   }
 
   row = hdy_action_row_new ();
-  hdy_action_row_set_title (HDY_ACTION_ROW (row), desc);
+  hdy_preferences_row_set_title (HDY_PREFERENCES_ROW (row), desc);
   g_object_set_data (G_OBJECT (row), "code", g_strdup (code));
   gtk_style_context_add_class (gtk_widget_get_style_context (GTK_WIDGET (row)), "row");
 
diff --git src/preferences/prefs-sync-page.c src/preferences/prefs-sync-page.c
index 3b5e43c26..3df267bfd 100644
--- src/preferences/prefs-sync-page.c
+++ src/preferences/prefs-sync-page.c
@@ -161,7 +161,7 @@ sync_secrets_store_finished_cb (EphySyncService *service,
   g_assert (EPHY_IS_PREFS_SYNC_PAGE (sync_page));
 
   if (!error) {
-    hdy_action_row_set_title (HDY_ACTION_ROW (sync_page->sync_firefox_account_row),
+    hdy_preferences_row_set_title (HDY_PREFERENCES_ROW (sync_page->sync_firefox_account_row),
                               ephy_sync_utils_get_sync_user ());
     gtk_widget_hide (sync_page->sync_page_box);
     gtk_widget_show (sync_page->sync_firefox_account_box);
@@ -718,7 +718,7 @@ prefs_sync_page_setup (PrefsSyncPage *sync_page)
     gtk_widget_hide (sync_page->sync_options_box);
   } else {
     sync_set_last_sync_time (sync_page);
-    hdy_action_row_set_title (HDY_ACTION_ROW (sync_page->sync_firefox_account_row), user);
+    hdy_preferences_row_set_title (HDY_PREFERENCES_ROW (sync_page->sync_firefox_account_row), user);
     gtk_widget_hide (sync_page->sync_page_box);
   }
 
