diff --git lib/ephy-sync-utils.c lib/ephy-sync-utils.c
index 75c08fc6377dcfaa720155a08367962f2b052dca..e11add0a24905deb9f04abb876b7e18dab34daca 100644
--- lib/ephy-sync-utils.c
+++ lib/ephy-sync-utils.c
@@ -31,7 +31,7 @@
 #include <string.h>
 #include <webkit2/webkit2.h>
 #if defined(__linux__)
-#include <sys/random.h>
+// #include <sys/random.h>
 #elif defined(__FreeBSD__) || defined(__OpenBSD__)
 #include <unistd.h>
 #endif
@@ -167,6 +167,18 @@ ephy_sync_utils_base64_urlsafe_decode (const char *text,
   return out;
 }
 
+static gssize getrandom(guint8 *out, gsize num_bytes, int ignored)
+{
+    gssize ret = 0;
+    FILE *fp;
+    if ((fp = fopen ("/dev/urandom", "r"))) {
+        errno = 0;
+        ret = (gssize) fread (out, sizeof (guint8), num_bytes, fp);
+        fclose (fp);
+    }
+    return ret;
+}
+
 /*
  * This is mainly required by Nettle's RSA support.
  * From Nettle's documentation: random_ctx and random is a randomness generator.
