diff --git qtbase/src/corelib/kernel/qobject.cpp qtbase/src/corelib/kernel/qobject.cpp
index 84cc2d5..7a0fddb 100644
--- qtbase/src/corelib/kernel/qobject.cpp
+++ qtbase/src/corelib/kernel/qobject.cpp
@@ -3883,7 +3883,7 @@ QList<QByteArray> QObject::dynamicPropertyNames() const
 
 static void dumpRecursive(int level, QObject *object)
 {
-#if defined(QT_DEBUG)
+// #if defined(QT_DEBUG)
     if (object) {
         QByteArray buf;
         buf.fill(' ', level / 2 * 8);
@@ -3912,10 +3912,10 @@ static void dumpRecursive(int level, QObject *object)
                 dumpRecursive(level+1, children.at(i));
         }
     }
-#else
-    Q_UNUSED(level)
-        Q_UNUSED(object)
-#endif
+// #else
+//     Q_UNUSED(level)
+//         Q_UNUSED(object)
+// #endif
 }
 
 /*!
@@ -3946,7 +3946,7 @@ void QObject::dumpObjectTree()
 
 void QObject::dumpObjectInfo()
 {
-#if defined(QT_DEBUG)
+// #if defined(QT_DEBUG)
     qDebug("OBJECT %s::%s", metaObject()->className(),
            objectName().isEmpty() ? "unnamed" : objectName().toLocal8Bit().data());
 
@@ -4010,7 +4010,7 @@ void QObject::dumpObjectInfo()
     } else {
         qDebug("        <None>");
     }
-#endif
+// #endif
 }
 
 #ifndef QT_NO_USERDATA
