diff --git src/dialogs/filetypeconfigwidget.ui src/dialogs/filetypeconfigwidget.ui
index 2c1aa022e0e0f222c53d557c0020ccb49f0ba538..869eba6a268d2624d01f605378a5fdab43a508c6 100644
--- src/dialogs/filetypeconfigwidget.ui
+++ src/dialogs/filetypeconfigwidget.ui
@@ -206,6 +206,9 @@
         <property name="whatsThis">
          <string>Sets priority for this file type. If more than one file type selects the same file, the one with the highest priority will be used.</string>
         </property>
+        <property name="minimum">
+         <number>-99</number>
+        </property>
        </widget>
       </item>
      </layout>
diff --git src/mode/katemodeconfigpage.cpp src/mode/katemodeconfigpage.cpp
index d0eeac540025044de547a4797aeef75b1b78284e..d477e0caad365a015908c0b7be0fc1ec6cbe082d 100644
--- src/mode/katemodeconfigpage.cpp
+++ src/mode/katemodeconfigpage.cpp
@@ -41,6 +41,7 @@
 #include <QLayout>
 #include <QPushButton>
 #include <QToolButton>
+#include <QSpinBox>
 //END Includes
 
 ModeConfigPage::ModeConfigPage(QWidget *parent)
@@ -87,6 +88,9 @@ ModeConfigPage::ModeConfigPage(QWidget *parent)
     connect(ui->cmbHl, SIGNAL(activated(int)), this, SLOT(slotChanged()));
     connect(ui->cmbIndenter, SIGNAL(activated(int)), this, SLOT(slotChanged()));
 
+    // make the context help a bit easier to access
+    ui->sbPriority->setToolTip(ui->sbPriority->whatsThis());
+
     layout->addWidget(newWidget);
     setLayout(layout);
 }
