diff --git src/view/kateview.cpp src/view/kateview.cpp
index ac6b86a04728290bd09d24b59cd9f632f48af746..570e6085b490faeb30046543c6061cc301ab5ca8 100644
--- src/view/kateview.cpp
+++ src/view/kateview.cpp
@@ -3050,7 +3050,11 @@ void KTextEditor::ViewPrivate::aboutToShowContextMenu()
     QMenu *menu = qobject_cast<QMenu *>(sender());
 
     if (menu) {
-        emit contextMenuAboutToShow(this, menu);
+        if (mainWindow()->activeView() == this) {
+            emit contextMenuAboutToShow(this, menu);
+        } else {
+            qCDebug(LOG_KTE()) << Q_FUNC_INFO << "not emitting contextMenuAboutToShow for background view" << this;
+        }
     }
 }
 
