diff --git src/utils/kateconfig.cpp src/utils/kateconfig.cpp
index 1043eff4b3f4843fc9c9de5ae63bf1618161b6bd..5455de3ff9302ebb220b7c30c1ccc30fc49bbda3 100644
--- src/utils/kateconfig.cpp
+++ src/utils/kateconfig.cpp
@@ -426,7 +426,7 @@ void KateDocumentConfig::readConfig(const KConfigGroup &config)
      *          do it, see also the KTextEdit class
      */
     if (isGlobal()) {
-        const QSettings settings(QStringLiteral("KDE"), QStringLiteral("Sonnet"));
+        const QSettings settings(QSettings::IniFormat, QSettings::UserScope, QStringLiteral("KDE"), QStringLiteral("Sonnet"));
         setOnTheFlySpellCheck(settings.value(QStringLiteral("checkerEnabledByDefault"), false).toBool());
     }
 
