--- src/data/katepart5ui.rc.orig	2017-06-03 12:20:31.000000000 +0200
+++ src/data/katepart5ui.rc	2017-06-13 18:44:34.000000000 +0200
@@ -94,8 +94,6 @@
   <Menu name="tools" noMerge="1"><text>&amp;Tools</text>
     <Action name="tools_toggle_write_lock" group="tools_operations" />
     <Separator group="tools_operations" />
-    <Action name="tools_mode" group="tools_operations" />
-    <Action name="tools_highlighting" group="tools_operations" />
     <Action name="tools_indentation" group="tools_operations" />
     <Action name="set_encoding" group="tools_operations" />
     <Action name="set_eol" group="tools_operations" />
@@ -146,6 +144,9 @@
   <Separator group="popup_operations" />
   <Action name="edit_select_all" group="popup_operations" />
   <Action name="edit_deselect" group="popup_operations" />
+  <Separator group="popup_operations"/>
+   <Action name="tools_mode" group="popup_operations"/>
+   <Action name="tools_highlighting" group="popup_operations"/>
   <Separator group="popup_operations" />
   <Action name="spelling_suggestions" group="popup_operations" />
   <Separator group="popup_operations" />
@@ -162,4 +163,37 @@
   <Action name="edit_redo" group="edit_operations" />
 </ToolBar>
 
+<ActionProperties scheme="Default">
+  <Action shortcut="Ctrl+G" name="edit_find_next"/>
+  <Action shortcut="Ctrl+L" name="go_goto_line"/>
+  <Action shortcut="" name="Next Editing Line"/>
+  <Action shortcut="" name="Previous Editing Line"/>
+  <Action shortcut="Ctrl+Shift+G" name="edit_find_prev"/>
+  <Action shortcut="Ctrl+E" name="edit_find_selected"/>
+  <Action shortcut="Ctrl+Shift+E" name="edit_find_selected_backwards"/>
+  <Action shortcut="Ctrl+Up" name="beginning_of_document"/>
+  <Action shortcut="Ctrl+Left" name="beginning_of_line"/>
+  <Action shortcut="Ctrl+Down" name="end_of_document"/>
+  <Action shortcut="" name="scroll_line_down"/>
+  <Action shortcut="" name="scroll_line_up"/>
+  <Action shortcut="Alt+Shift+Left" name="select_word_left"/>
+  <Action shortcut="Alt+Shift+Right" name="select_word_right"/>
+  <Action shortcut="Alt+Left" name="word_left"/>
+  <Action shortcut="Alt+Right" name="word_right"/>
+  <Action shortcut="Ctrl+Right" name="end_of_line"/>
+  <Action shortcut="" name="tools_scripts_jumpIndentDown"/>
+  <Action shortcut="" name="tools_scripts_jumpIndentUp"/>
+  <Action shortcut="Ctrl+]" name="tools_indent"/>
+  <Action shortcut="Ctrl+[" name="tools_unindent"/>
+  <Action shortcut="Ctrl+Shift+Left" name="select_beginning_of_line"/>
+  <Action shortcut="Ctrl+Shift+Right" name="select_end_of_line"/>
+  <Action shortcut="Ctrl+Shift+Up" name="select_beginning_of_document"/>
+  <Action shortcut="Ctrl+Shift+Down" name="select_end_of_document"/>
+  <Action shortcut="" name="tools_scripts_moveLinesDown"/>
+  <Action shortcut="" name="tools_scripts_moveLinesUp"/>
+  <Action shortcut="Meta+Ctrl+-" name="folding_toplevel"/>
+  <Action shortcut="" name="delete_word_left"/>
+  <Action shortcut="" name="bookmarks_toggle"/>
+</ActionProperties>
+
 </gui>
