--- src/data/katepart5ui.rc.orig	2017-06-03 12:20:31.000000000 +0200
+++ src/data/katepart5ui.rc	2017-06-13 18:44:34.000000000 +0200
@@ -94,8 +94,6 @@
   <Menu name="tools" noMerge="1"><text>&amp;Tools</text>
     <Action name="tools_toggle_write_lock" group="tools_operations" />
     <Separator group="tools_operations" />
-    <Action name="tools_mode" group="tools_operations" />
-    <Action name="tools_highlighting" group="tools_operations" />
     <Action name="tools_indentation" group="tools_operations" />
     <Action name="set_encoding" group="tools_operations" />
     <Action name="set_eol" group="tools_operations" />
@@ -146,6 +144,9 @@
   <Separator group="popup_operations" />
   <Action name="edit_select_all" group="popup_operations" />
   <Action name="edit_deselect" group="popup_operations" />
+  <Separator group="popup_operations"/>
+   <Action name="tools_mode" group="popup_operations"/>
+   <Action name="tools_highlighting" group="popup_operations"/>
   <Separator group="popup_operations" />
   <Action name="spelling_suggestions" group="popup_operations" />
   <Separator group="popup_operations" />
