diff --git src/render/katerenderer.cpp src/render/katerenderer.cpp
index 17efac0..1e3c81f 100644
--- src/render/katerenderer.cpp
+++ src/render/katerenderer.cpp
@@ -941,7 +941,7 @@ void KateRenderer::updateFontHeight()
     // use height of font + round down, ensure we have at least one pixel
     // we round down to avoid artifacts: line height too large vs. qt background rendering of text attributes
     const qreal height = config()->fontMetrics().height();
-    m_fontHeight = qMax(1, qFloor(height));
+    m_fontHeight = qMax(1, qCeil(height));
 }
 
 qreal KateRenderer::spaceWidth() const
