diff --git src/document/katedocument.cpp src/document/katedocument.cpp
index b6095a4..5c1d457 100644
--- src/document/katedocument.cpp
+++ src/document/katedocument.cpp
@@ -4875,11 +4875,11 @@ QString KTextEditor::DocumentPrivate::reasonedMOHString() const
 
     switch (m_modOnHdReason) {
     case OnDiskModified:
-        return i18n("The file '%1' was modified by another program.", str);
-        break;
     case OnDiskCreated:
-        return i18n("The file '%1' was created by another program.", str);
+        return i18n("The file '%1' was modified by another program.", str);
         break;
+// 	   return i18n("The file '%1' was created by another program.", str);
+//         break;
     case OnDiskDeleted:
         return i18n("The file '%1' was deleted by another program.", str);
         break;
