diff --git src/view/kateviewinternal.cpp src/view/kateviewinternal.cpp
index 35545cc..02d5e96 100644
--- src/view/kateviewinternal.cpp
+++ src/view/kateviewinternal.cpp
@@ -2690,7 +2690,12 @@ void KateViewInternal::mouseDoubleClickEvent(QMouseEvent *e)
 
         // Move cursor to end (or beginning) of selected word
         if (m_view->selection()) {
+#if defined(Q_OS_OSX)
+            // avoid "Data set on unsupported clipboard mode" warnings.
+            //QApplication::clipboard()->setText(m_view->selectionText());
+#else
             QApplication::clipboard()->setText(m_view->selectionText(), QClipboard::Selection);
+#endif
         }
 
         moveCursorToSelectionEdge();
