diff --git src/file/CMakeLists.txt src/file/CMakeLists.txt
index 49a34dd..262e45f 100644
--- src/file/CMakeLists.txt
+++ src/file/CMakeLists.txt
@@ -59,6 +59,12 @@ target_link_libraries(baloo_file
     balooxapian
 )
 
+if (APPLE)
+    set_target_properties(baloo_file PROPERTIES MACOSX_BUNDLE_INFO_PLIST ${CMAKE_CURRENT_SOURCE_DIR}/Info.plist.template)
+    set_target_properties(baloo_file PROPERTIES MACOSX_BUNDLE_GUI_IDENTIFIER "org.kde.baloo.file")
+    set_target_properties(baloo_file PROPERTIES MACOSX_BUNDLE_BUNDLE_NAME "KDE Baloo File Daemon service")
+endif (APPLE)
+
 install(TARGETS baloo_file ${INSTALL_TARGETS_DEFAULT_ARGS})
 install(FILES baloo_file.desktop DESTINATION ${AUTOSTART_INSTALL_DIR})
 install(FILES org.kde.baloo.file.indexer.xml DESTINATION ${DBUS_INTERFACES_INSTALL_DIR})
diff --git src/file/Info.plist.template src/file/Info.plist.template
new file mode 100644
index 0000000..c39ddb9
--- /dev/null
+++ src/file/Info.plist.template
@@ -0,0 +1,36 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple Computer//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+    <key>CFBundleDevelopmentRegion</key>
+    <string>English</string>
+    <key>CFBundleExecutable</key>
+    <string>${MACOSX_BUNDLE_EXECUTABLE_NAME}</string>
+    <key>CFBundleGetInfoString</key>
+    <string>${MACOSX_BUNDLE_INFO_STRING}</string>
+    <key>CFBundleIconFile</key>
+    <string>${MACOSX_BUNDLE_ICON_FILE}</string>
+    <key>CFBundleIdentifier</key>
+    <string>${MACOSX_BUNDLE_GUI_IDENTIFIER}</string>
+    <key>CFBundleInfoDictionaryVersion</key>
+    <string>6.0</string>
+    <key>CFBundleLongVersionString</key>
+    <string>${MACOSX_BUNDLE_LONG_VERSION_STRING}</string>
+    <key>CFBundleName</key>
+    <string>${MACOSX_BUNDLE_BUNDLE_NAME}</string>
+    <key>CFBundlePackageType</key>
+    <string>APPL</string>
+    <key>CFBundleShortVersionString</key>
+    <string>${MACOSX_BUNDLE_SHORT_VERSION_STRING}</string>
+    <key>CFBundleVersion</key>
+    <string>${MACOSX_BUNDLE_BUNDLE_VERSION}</string>
+    <key>CSResourcesFileMapped</key>
+    <true/>
+    <key>LSRequiresCarbon</key>
+    <true/>
+    <key>LSUIElement</key>
+    <string>1</string>
+    <key>NSHumanReadableCopyright</key>
+    <string>${MACOSX_BUNDLE_COPYRIGHT}</string>
+</dict>
+</plist>
