--- qtlocation/src/plugins/position/geoclue/orig.qgeopositioninfosource_geocluemaster.cpp	2014-09-11 12:48:31.000000000 +0200
+++ qtlocation/src/plugins/position/geoclue/qgeopositioninfosource_geocluemaster.cpp	2015-03-23 00:11:22.000000000 +0100
@@ -289,7 +289,7 @@
         const QVariant v = i.value();
 
         GValue *value = new GValue;
-        *value = G_VALUE_INIT;
+        memset(value, 0, sizeof(GValue));
 
         switch (v.userType()) {
         case QMetaType::QString:
