diff --git a/qtbase/src/platformsupport/fontdatabases/fontdatabases.pro b/qtbase/src/platformsupport/fontdatabases/fontdatabases.pro
index 9376c3b..4d5052e 100644
--- a/qtbase/src/platformsupport/fontdatabases/fontdatabases.pro
+++ b/qtbase/src/platformsupport/fontdatabases/fontdatabases.pro
@@ -9,7 +9,7 @@ PRECOMPILED_HEADER = ../../corelib/global/qt_pch.h
 
 darwin:!if(watchos:CONFIG(simulator, simulator|device)) {
     include($$PWD/mac/coretext.pri)
-} else {
+}
     qtConfig(freetype) {
         include($$PWD/basic/basic.pri)
     }
@@ -28,6 +28,6 @@ darwin:!if(watchos:CONFIG(simulator, simulator|device)) {
     winrt {
         include($$PWD/winrt/winrt.pri)
     }
-}
+
 
 load(qt_module)
