diff --git a/qtbase/src/platformsupport/fontdatabases/mac/qfontengine_coretext.mm b/qtbase/src/platformsupport/fontdatabases/mac/qfontengine_coretext.mm
index 7a06d5f1c93a0920ba57b8e9b84c5e1d528d5bf9..ff1eea6b1ed6831ed1cb5342c4b0fe4cc16cfeb1 100644
--- a/qtbase/src/platformsupport/fontdatabases/mac/qfontengine_coretext.mm
+++ b/qtbase/src/platformsupport/fontdatabases/mac/qfontengine_coretext.mm
@@ -183,9 +189,11 @@ QCoreTextFontEngine::QCoreTextFontEngine(CTFontRef font, const QFontDef &def)
     fontDef = def;
     transform = qt_transform_from_fontdef(fontDef);
     ctfont = font;
-    CFRetain(ctfont);
-    cgFont = CTFontCopyGraphicsFont(font, NULL);
-    init();
+    if (font) {
+        CFRetain(ctfont);
+        cgFont = CTFontCopyGraphicsFont(font, NULL);
+        init();
+    }
 }
 
 QCoreTextFontEngine::QCoreTextFontEngine(CGFontRef font, const QFontDef &def)
@@ -195,15 +203,17 @@ QCoreTextFontEngine::QCoreTextFontEngine(CGFontRef font, const QFontDef &def)
     transform = qt_transform_from_fontdef(fontDef);
     cgFont = font;
     // Keep reference count balanced
-    CFRetain(cgFont);
-    ctfont = CTFontCreateWithGraphicsFont(font, fontDef.pixelSize, &transform, NULL);
-    init();
+    if (font) {
+        CFRetain(cgFont);
+        ctfont = CTFontCreateWithGraphicsFont(font, fontDef.pixelSize, &transform, NULL);
+        init();
+    }
 }
 
 QCoreTextFontEngine::~QCoreTextFontEngine()
 {
-    CFRelease(cgFont);
-    CFRelease(ctfont);
+    if (cgFont) CFRelease(cgFont);
+    if (ctfont) CFRelease(ctfont);
 }
 
 void QCoreTextFontEngine::init()
