diff --git src/lib/rng/system_rng/orig.system_rng.cpp src/lib/rng/system_rng/system_rng.cpp
index 7993173..826924a 100644
--- src/lib/rng/system_rng/orig.system_rng.cpp
+++ src/lib/rng/system_rng/system_rng.cpp
@@ -30,7 +30,9 @@
    #include <stdlib.h>
 #elif defined(BOTAN_TARGET_OS_HAS_GETRANDOM)
    #include <errno.h>
+   #if defined(__GLIBC__) && __GLIBC__ == 2 && __GLIBC_MINOR__ >= 25
    #include <sys/random.h>
+   #endif
    #include <sys/syscall.h>
    #include <unistd.h>
 #elif defined(BOTAN_TARGET_OS_HAS_DEV_RANDOM)
@@ -219,7 +221,13 @@ class System_RNG_Impl final : public RandomNumberGenerator {
          size_t len = output.size();
          while(len > 0) {
    #if defined(__GLIBC__) && __GLIBC__ == 2 && __GLIBC_MINOR__ < 25
+     #ifdef SYS_getrandom
             const ssize_t got = ::syscall(SYS_getrandom, buf, len, flags);
+     #elif defined(__NR_getrandom)
+            const ssize_t got = ::syscall(__NR_getrandom, buf, len, flags);
+     #else
+	  #error "No getrandom syscall!"
+     #endif
    #else
             const ssize_t got = ::getrandom(buf, len, flags);
    #endif
