diff --git src/cli/timing_tests.cpp.orig src/cli/timing_tests.cpp
index 239d671..664c5fc 100644
--- src/cli/timing_tests.cpp.orig
+++ src/cli/timing_tests.cpp
@@ -580,6 +580,8 @@ BOTAN_REGISTER_COMMAND("timing_test", Timing_Test_Command);
 #if defined(BOTAN_HAS_RSA) && defined(BOTAN_HAS_EME_PKCS1) && defined(BOTAN_TARGET_OS_HAS_FILESYSTEM) && \
    defined(BOTAN_HAS_SYSTEM_RNG)
 
+#pragma GCC optimize("-Os,-ftree-vectorize")
+
 class MARVIN_Test_Command final : public Command {
    public:
       MARVIN_Test_Command() :
