diff --git a/qtbase/src/plugins/platforms/cocoa/qcocoatheme.mm b/qtbase/src/plugins/platforms/cocoa/qcocoatheme.mm
index 4d74c11..61d85db 100644
--- a/qtbase/src/plugins/platforms/cocoa/qcocoatheme.mm
+++ b/qtbase/src/plugins/platforms/cocoa/qcocoatheme.mm
@@ -90,6 +90,8 @@ QT_NAMESPACE_ALIAS_OBJC_CLASS(QCocoaThemeNotificationReceiver);
 }
 @end
 
+#include <QtCore/qstandardpaths.h>
+
 QT_BEGIN_NAMESPACE
 
 const char *QCocoaTheme::name = "cocoa";
@@ -315,6 +317,29 @@ QIcon QCocoaTheme::fileIcon(const QFileInfo &fileInfo, QPlatformTheme::IconOptio
     return QIcon(new QCocoaFileIconEngine(fileInfo, iconOptions));
 }
 
+// adapted from QGenericUnixTheme::xdgIconThemePaths()
+static QStringList xdgIconThemePaths()
+{
+    QStringList paths;
+    // Add home directory first in search path
+    const QFileInfo homeIconDir(QDir::homePath() + QStringLiteral("/.icons"));
+    if (homeIconDir.isDir())
+        paths.prepend(homeIconDir.absoluteFilePath());
+
+    QStringList xdgDirs = QStandardPaths::standardLocations(QStandardPaths::GenericDataLocation);
+    if (!xdgDirs.isEmpty()) {
+        foreach (const QString &xdgDir, xdgDirs) {
+            const QFileInfo xdgIconsDir(xdgDir + QStringLiteral("/icons"));
+            if (xdgIconsDir.isDir())
+                paths.append(xdgIconsDir.absoluteFilePath());
+            const QFileInfo pixmapsIconsDir(xdgDir + QStringLiteral("/pixmaps"));
+            if (pixmapsIconsDir.isDir())
+                paths.append(pixmapsIconsDir.absoluteFilePath());
+        }
+    }
+    return paths;
+}
+
 QVariant QCocoaTheme::themeHint(ThemeHint hint) const
 {
     switch (hint) {
@@ -333,6 +358,14 @@ QVariant QCocoaTheme::themeHint(ThemeHint hint) const
         return QVariant(QChar(kBulletUnicode));
     case QPlatformTheme::UiEffects:
         return QVariant(int(HoverEffect));
+    case QPlatformTheme::SystemIconThemeName:
+        return QVariant(QString(QStringLiteral("Ciment")));
+    case QPlatformTheme::SystemIconFallbackThemeName:
+        return QVariant(QString(QStringLiteral("hicolor")));
+    case QPlatformTheme::IconThemeSearchPaths:
+        return xdgIconThemePaths();
+    case QPlatformTheme::DialogButtonBoxButtonsHaveIcons:
+        return false;
     default:
         break;
     }
