diff --git a/qtbase/src/platformsupport/fontdatabases/fontdatabases.pro b/qtbase/src/platformsupport/fontdatabases/fontdatabases.pro
index 9376c3b..4d5052e 100644
--- a/qtbase/src/platformsupport/fontdatabases/fontdatabases.pro
+++ b/qtbase/src/platformsupport/fontdatabases/fontdatabases.pro
@@ -9,7 +9,7 @@ PRECOMPILED_HEADER = ../../corelib/global/qt_pch.h
 
 darwin:!if(watchos:CONFIG(simulator, simulator|device)) {
     include($$PWD/mac/coretext.pri)
-} else {
+}
     qtConfig(freetype) {
         include($$PWD/basic/basic.pri)
     }
@@ -28,6 +28,6 @@ darwin:!if(watchos:CONFIG(simulator, simulator|device)) {
     winrt {
         include($$PWD/winrt/winrt.pri)
     }
-}
+
 
 load(qt_module)
diff --git a/qtbase/src/gui/configure.json b/qtbase/src/gui/configure.json
index 1f50116..7ebda8d 100644
--- a/qtbase/src/gui/configure.json
+++ b/qtbase/src/gui/configure.json
@@ -471,7 +471,7 @@
         },
         "fontconfig": {
             "label": "Fontconfig",
-            "condition": "!config.win32 && !config.darwin && features.system-freetype && libs.fontconfig",
+            "condition": "!config.win32 && features.system-freetype && libs.fontconfig",
             "output": [ "privateFeature", "feature" ]
         },
         "gbm": {
