diff --git a/qtbase/src/widgets/widgets/qtabbar.cpp b/qtbase/src/widgets/widgets/qtabbar.cpp
index 6e34198..94a8a97 100644
--- a/qtbase/src/widgets/widgets/qtabbar.cpp
+++ b/qtbase/src/widgets/widgets/qtabbar.cpp
@@ -178,7 +178,8 @@ void QTabBarPrivate::initBasicStyleOption(QStyleOptionTab *option, int tabIndex)
         option->palette.setColor(q->foregroundRole(), tab.textColor);
 #ifdef Q_OS_MACOS
     else if (isCurrent && !documentMode
-             && (QSysInfo::MacintoshVersion < QSysInfo::MV_10_10 || q->isActiveWindow())) {
+             && (QSysInfo::MacintoshVersion < QSysInfo::MV_10_10 || q->isActiveWindow())
+             && q->style()->objectName() == QStringLiteral("macintosh")) {
         option->palette.setColor(QPalette::WindowText, Qt::white);
     }
 #endif
