diff --git a/qtbase/src/widgets/widgets/qmenu_mac.mm b/qtbase/src/widgets/widgets/qmenu_mac.mm
index f9f3ad0..a579152 100644
--- a/qtbase/src/widgets/widgets/qmenu_mac.mm
+++ b/qtbase/src/widgets/widgets/qmenu_mac.mm
@@ -104,7 +104,9 @@ void QMenu::setAsDockMenu()
     QPlatformNativeInterface::NativeResourceForIntegrationFunction function = resolvePlatformFunction("setdockmenu");
     if (function) {
         typedef void (*SetDockMenuFunction)(QPlatformMenu *platformMenu);
-        reinterpret_cast<SetDockMenuFunction>(function)(d->createPlatformMenu());
+        QPlatformMenu *pMenu = d->createPlatformMenu();
+        if (pMenu)
+            reinterpret_cast<SetDockMenuFunction>(function)(pMenu);
     }
 }
 
