diff --git a/src/effects/VST/VSTEffect.cpp b/src/effects/VST/VSTEffect.cpp
index 3cd95bd1c339ea1a744849562f387161d8be3ea2..b48eb3d838f1bf6981e5aad3bb860b3eec00c165 100644
--- a/src/effects/VST/VSTEffect.cpp
+++ b/src/effects/VST/VSTEffect.cpp
@@ -1110,14 +1110,23 @@ void VSTEffect::BundleDeleter::operator() (void* p) const
 void VSTEffect::ResourceDeleter::operator() (void *p) const
 {
    if (mpHandle) {
-      int resource = (int)p;
-      CFBundleCloseBundleResourceMap(mpHandle->get(), resource);
+      CFBundleRefNum *resource = reinterpret_cast<CFBundleRefNum*>(p);
+      if (resource)
+      {
+          CFBundleCloseBundleResourceMap(mpHandle->get(), *resource);
+          // resource is ours to free
+          free(resource);
+      }
    }
 }
 #endif
 
 VSTEffect::VSTEffect(const wxString & path, VSTEffect *master)
 :  mPath(path),
+#ifdef __WXMAC__
+   // mResource must be initialised explicitly when building against libstdc++ instead of libc++
+   mResource(ResourceHandle{nullptr, nullptr}),
+#endif
    mMaster(master)
 {
    mHost = NULL;
@@ -1994,7 +2003,7 @@ bool VSTEffect::Load()
    // Start clean
    mBundleRef.reset();
 
-   mResource = ResourceHandle{};
+   mResource = ResourceHandle{nullptr, nullptr};
 
    // Convert the path to a CFSTring
    wxCFStringRef path(realPath);
@@ -2064,11 +2073,12 @@ bool VSTEffect::Load()
    mBundleRef = std::move(bundleRef);
 
    // Open the resource map ... some plugins (like GRM Tools) need this.
-   mResource = ResourceHandle {
-      reinterpret_cast<char*>(
-         CFBundleOpenBundleResourceMap(mBundleRef.get())),
-      ResourceDeleter{&mBundleRef}
-   };
+   // CFBundleOpenBundleResourceMap() returns a CFBundleRefNum which is
+   // an int, so we use a cache variable to avoid having to upcast to a
+   // pointer (upcasting is easy, downcasting back to an int less so).
+   CFBundleRefNum *resource = new CFBundleRefNum;
+   *resource = CFBundleOpenBundleResourceMap(mBundleRef.get());
+   mResource = ResourceHandle {resource, ResourceDeleter{&mBundleRef}};
 
 #elif defined(__WXMSW__)
 
@@ -2270,7 +2280,7 @@ void VSTEffect::Unload()
    if (mModule)
    {
 #if defined(__WXMAC__)
-      mResource = ResourceHandle{};
+      mResource = ResourceHandle{nullptr, nullptr};
       mBundleRef.reset();
 #endif
 
diff --git a/src/effects/VST/VSTEffect.h b/src/effects/VST/VSTEffect.h
index ccf4e12b7ae0279c5900b8517c18a1a7e65aef51..4cad1618c87403a1cf4fa17023bcffe448c2f27f 100644
--- a/src/effects/VST/VSTEffect.h
+++ b/src/effects/VST/VSTEffect.h
@@ -308,7 +308,7 @@ private:
       void operator() (void*) const;
    };
    using ResourceHandle = std::unique_ptr<
-      char, ResourceDeleter
+      CFBundleRefNum, ResourceDeleter
    >;
    ResourceHandle mResource;
 #endif
