diff --git a/src/MemoryX.h b/src/MemoryX.h
index 56e99171a9eb87fe574291d0ec9d086efb0c1fbb..621afa3001ba354f36e80edb86c72a45dca426cb 100644
--- a/src/MemoryX.h
+++ b/src/MemoryX.h
@@ -11,7 +13,7 @@
 // std:: containers knowing about rvalue references
 #undef __AUDACITY_OLD_STD__
 
-#if defined(__MAC_OS_X_VERSION_MIN_REQUIRED) && __MAC_OS_X_VERSION_MIN_REQUIRED <= __MAC_10_6
+#if 0 //defined(__MAC_OS_X_VERSION_MIN_REQUIRED) && __MAC_OS_X_VERSION_MIN_REQUIRED <= __MAC_10_6
 
 #define __AUDACITY_OLD_STD__
 
@@ -1177,7 +1179,7 @@ namespace std
    namespace tr1
    {
 #endif
-      template<typename T> struct hash;
+//       template<typename T> struct hash;
       template<> struct hash< wxString > {
          size_t operator () (const wxString &str) const // noexcept
          {
