--- var/macports/sources/rsync.macports.org/release/ports/kde/kdelibs4/Portfile	2014-06-24 18:30:27.000000000 +0200
+++ site-ports/kde/kdelibs4/Portfile	2014-07-07 00:44:15.000000000 +0200
@@ -6,7 +6,7 @@
 
 name                kdelibs4
 version             4.12.5
-revision            2
+revision            3
 categories          kde kde4
 maintainers         nicos
 license             LGPL-2+ GPL-2+ BSD
@@ -61,7 +61,8 @@
                     patch-cmake-modules-FindQt4.cmake.diff \
                     patch-cmake-modules-KDE4Macros.cmake.diff \
                     patch-removeFindFlex.diff \
-                    patch-kapplications-raster.diff
+                    patch-kapplications-raster.diff \
+				avoid-kwindowinfo-destructor.patch
 
 patch.pre_args      -p1
 
