diff --git a/src/dolphinmainwindow.cpp b/src/dolphinmainwindow.cpp
index 37f6bb8e461f43a6e686e23e74e105f8af5f9f9c..d97047d3054dba494dd68ea543c5613a2071bbbf 100644
--- a/src/dolphinmainwindow.cpp
+++ b/src/dolphinmainwindow.cpp
@@ -100,6 +100,15 @@ namespace {
     const int MaxActivateTabShortcuts = 9;
 }
 
+static inline bool canHideMenuBar()
+{
+#ifdef Q_OS_OSX
+    return !QGuiApplication::platformName().contains(QLatin1String("cocoa"));
+#else
+    return true;
+#endif
+}
+
 DolphinMainWindow::DolphinMainWindow() :
     KXmlGuiWindow(nullptr),
     m_newFileMenu(nullptr),
@@ -190,11 +199,13 @@ DolphinMainWindow::DolphinMainWindow() :
         resize(750, 500);
     }
 
-    const bool showMenu = !menuBar()->isHidden();
-    QAction* showMenuBarAction = actionCollection()->action(KStandardAction::name(KStandardAction::ShowMenubar));
-    showMenuBarAction->setChecked(showMenu);  // workaround for bug #171080
-    if (!showMenu) {
-        createControlButton();
+    if (canHideMenuBar()) {
+        const bool showMenu = !menuBar()->isHidden();
+        QAction* showMenuBarAction = actionCollection()->action(KStandardAction::name(KStandardAction::ShowMenubar));
+        showMenuBarAction->setChecked(showMenu);  // workaround for bug #171080
+        if (!showMenu) {
+            createControlButton();
+        }
     }
 
     // enable middle-click on back/forward/up to open in a new tab
@@ -935,12 +950,14 @@ void DolphinMainWindow::compareFiles()
 
 void DolphinMainWindow::toggleShowMenuBar()
 {
-    const bool visible = menuBar()->isVisible();
-    menuBar()->setVisible(!visible);
-    if (visible) {
-        createControlButton();
-    } else {
-        deleteControlButton();
+    if (canHideMenuBar()) {
+        const bool visible = menuBar()->isVisible();
+        menuBar()->setVisible(!visible);
+        if (visible) {
+            createControlButton();
+        } else {
+            deleteControlButton();
+        }
     }
 }
 
@@ -1190,7 +1207,7 @@ void DolphinMainWindow::updateControlMenu()
 
 void DolphinMainWindow::updateToolBar()
 {
-    if (!menuBar()->isVisible()) {
+    if (canHideMenuBar() && !menuBar()->isVisible()) {
         createControlButton();
     }
 }
