diff --git a/src/panels/folders/folderspanel.cpp b/src/panels/folders/folderspanel.cpp
index 2e7e4dc28c1ee8d88a511044efec48bed940af37..323b422cfc3cc417e492ae338fb329658eb6a1e1 100644
--- a/src/panels/folders/folderspanel.cpp
+++ b/src/panels/folders/folderspanel.cpp
@@ -294,12 +294,13 @@ void FoldersPanel::slotLoadingCompleted()
 
 void FoldersPanel::startFadeInAnimation()
 {
-    QPropertyAnimation* anim = new QPropertyAnimation(m_controller->view(), "opacity", this);
-    anim->setStartValue(0);
-    anim->setEndValue(1);
-    anim->setEasingCurve(QEasingCurve::InOutQuad);
-    anim->start(QAbstractAnimation::DeleteWhenStopped);
-    anim->setDuration(200);
+//     QPropertyAnimation* anim = new QPropertyAnimation(m_controller->view(), "opacity", this);
+//     anim->setStartValue(0);
+//     anim->setEndValue(1);
+//     anim->setEasingCurve(QEasingCurve::InOutQuad);
+//     anim->start(QAbstractAnimation::DeleteWhenStopped);
+//     anim->setDuration(200);
+    m_controller->view()->setOpacity(1);
 }
 
 void FoldersPanel::loadTree(const QUrl& url, FoldersPanel::NavigationBehaviour navigationBehaviour)
diff --git a/src/kitemviews/kitemlistwidget.cpp b/src/kitemviews/kitemlistwidget.cpp
index 49a13f68fa096ef549a069d6d30147602f673bb9..f159ae04ac798aaec8270923138ec5daa532c5d9 100644
--- a/src/kitemviews/kitemlistwidget.cpp
+++ b/src/kitemviews/kitemlistwidget.cpp
@@ -245,27 +245,29 @@ void KItemListWidget::setHovered(bool hovered)
 
     m_hovered = hovered;
 
-    if (!m_hoverAnimation) {
-        m_hoverAnimation = new QPropertyAnimation(this, "hoverOpacity", this);
-        const int duration = style()->styleHint(QStyle::SH_Widget_Animate) ? 200 : 1;
-        m_hoverAnimation->setDuration(duration);
-        connect(m_hoverAnimation, &QPropertyAnimation::finished, this, &KItemListWidget::slotHoverAnimationFinished);
-    }
-    m_hoverAnimation->stop();
+//     if (!m_hoverAnimation) {
+//         m_hoverAnimation = new QPropertyAnimation(this, "hoverOpacity", this);
+//         const int duration = style()->styleHint(QStyle::SH_Widget_Animate) ? 200 : 1;
+//         m_hoverAnimation->setDuration(duration);
+//         connect(m_hoverAnimation, &QPropertyAnimation::finished, this, &KItemListWidget::slotHoverAnimationFinished);
+//     }
+//     m_hoverAnimation->stop();
 
     if (hovered) {
         const qreal startValue = qMax(hoverOpacity(), qreal(0.1));
-        m_hoverAnimation->setStartValue(startValue);
-        m_hoverAnimation->setEndValue(1.0);
+//         m_hoverAnimation->setStartValue(startValue);
+//         m_hoverAnimation->setEndValue(1.0);
+        setHoverOpacity(1.0);
         if (m_enabledSelectionToggle && !(QApplication::mouseButtons() & Qt::LeftButton)) {
             initializeSelectionToggle();
         }
     } else {
-        m_hoverAnimation->setStartValue(hoverOpacity());
-        m_hoverAnimation->setEndValue(0.0);
+//         m_hoverAnimation->setStartValue(hoverOpacity());
+//         m_hoverAnimation->setEndValue(0.0);
+        setHoverOpacity(0.0);
     }
 
-    m_hoverAnimation->start();
+//     m_hoverAnimation->start();
 
     hoveredChanged(hovered);
     update();
diff --git a/src/kitemviews/private/kitemlistsmoothscroller.cpp b/src/kitemviews/private/kitemlistsmoothscroller.cpp
index cdb7f255b978763ead545ee9901871d6cc191270..8cc9198c0d740f28f08359925d6912551160add0 100644
--- a/src/kitemviews/private/kitemlistsmoothscroller.cpp
+++ b/src/kitemviews/private/kitemlistsmoothscroller.cpp
@@ -36,10 +36,10 @@ KItemListSmoothScroller::KItemListSmoothScroller(QScrollBar* scrollBar,
     m_animation = new QPropertyAnimation(this);
 #if QT_VERSION >= QT_VERSION_CHECK(5, 10, 0)
     const int animationDuration = m_scrollBar->style()->styleHint(QStyle::SH_Widget_Animation_Duration, nullptr, m_scrollBar);
-    const bool animationEnabled = (animationDuration > 0);
+    const bool animationEnabled = false /*(animationDuration > 0)*/;
  #else
     const int animationDuration = 100;
-    const bool animationEnabled = m_scrollBar->style()->styleHint(QStyle::SH_Widget_Animate, nullptr, m_scrollBar);
+    const bool animationEnabled = false /*m_scrollBar->style()->styleHint(QStyle::SH_Widget_Animate, nullptr, m_scrollBar)*/;
 #endif
     m_animation->setDuration(animationEnabled ? animationDuration : 1);
     connect(m_animation, &QPropertyAnimation::stateChanged,
diff --git a/src/kitemviews/private/kitemlistviewanimation.cpp b/src/kitemviews/private/kitemlistviewanimation.cpp
index ccc387529b994e29bd85d0163cf042c09c216753..2b720c9400415716ec186d7a86d830f8f25ee39f 100644
--- a/src/kitemviews/private/kitemlistviewanimation.cpp
+++ b/src/kitemviews/private/kitemlistviewanimation.cpp
@@ -114,7 +114,8 @@ void KItemListViewAnimation::start(QGraphicsWidget* widget, AnimationType type,
     stop(widget, type);
 
     QPropertyAnimation* propertyAnim = nullptr;
-    const int animationDuration = widget->style()->styleHint(QStyle::SH_Widget_Animate) ? 200 : 1;
+    const bool animationEnabled = widget->style()->styleHint(QStyle::SH_Widget_Animate);
+    const int animationDuration = 200;
 
     switch (type) {
     case MovingAnimation: {
@@ -123,27 +124,39 @@ void KItemListViewAnimation::start(QGraphicsWidget* widget, AnimationType type,
             return;
         }
 
-        propertyAnim = new QPropertyAnimation(widget, "pos");
-        propertyAnim->setDuration(animationDuration);
-        propertyAnim->setEndValue(newPos);
+        if (animationEnabled) {
+            propertyAnim = new QPropertyAnimation(widget, "pos");
+            propertyAnim->setDuration(animationDuration);
+            propertyAnim->setEndValue(newPos);
+        } else {
+            widget->setPos(newPos);
+        }
         break;
     }
 
     case CreateAnimation: {
-        propertyAnim = new QPropertyAnimation(widget, "opacity");
-        propertyAnim->setEasingCurve(QEasingCurve::InQuart);
-        propertyAnim->setDuration(animationDuration);
-        propertyAnim->setStartValue(0.0);
-        propertyAnim->setEndValue(1.0);
+        if (animationEnabled) {
+            propertyAnim = new QPropertyAnimation(widget, "opacity");
+            propertyAnim->setEasingCurve(QEasingCurve::InQuart);
+            propertyAnim->setDuration(animationDuration);
+            propertyAnim->setStartValue(0.0);
+            propertyAnim->setEndValue(1.0);
+        } else {
+            widget->setOpacity(1.0);
+        }
         break;
     }
 
     case DeleteAnimation: {
-        propertyAnim = new QPropertyAnimation(widget, "opacity");
-        propertyAnim->setEasingCurve(QEasingCurve::OutQuart);
-        propertyAnim->setDuration(animationDuration);
-        propertyAnim->setStartValue(1.0);
-        propertyAnim->setEndValue(0.0);
+        if (animationEnabled) {
+            propertyAnim = new QPropertyAnimation(widget, "opacity");
+            propertyAnim->setEasingCurve(QEasingCurve::OutQuart);
+            propertyAnim->setDuration(animationDuration);
+            propertyAnim->setStartValue(1.0);
+            propertyAnim->setEndValue(0.0);
+        } else {
+            widget->setOpacity(0.0);
+        }
         break;
     }
 
@@ -153,9 +166,13 @@ void KItemListViewAnimation::start(QGraphicsWidget* widget, AnimationType type,
             return;
         }
 
-        propertyAnim = new QPropertyAnimation(widget, "size");
-        propertyAnim->setDuration(animationDuration);
-        propertyAnim->setEndValue(newSize);
+        if (animationEnabled) {
+            propertyAnim = new QPropertyAnimation(widget, "size");
+            propertyAnim->setDuration(animationDuration);
+            propertyAnim->setEndValue(newSize);
+        } else {
+            widget->resize(newSize);
+        }
         break;
     }
 
@@ -163,11 +180,13 @@ void KItemListViewAnimation::start(QGraphicsWidget* widget, AnimationType type,
         break;
     }
 
-    Q_ASSERT(propertyAnim);
-    connect(propertyAnim, &QPropertyAnimation::finished, this, &KItemListViewAnimation::slotFinished);
-    m_animation[type].insert(widget, propertyAnim);
+//     Q_ASSERT(propertyAnim);
+    if (propertyAnim) {
+        connect(propertyAnim, &QPropertyAnimation::finished, this, &KItemListViewAnimation::slotFinished);
+        m_animation[type].insert(widget, propertyAnim);
 
-    propertyAnim->start();
+        propertyAnim->start();
+    }
 }
 
 void KItemListViewAnimation::stop(QGraphicsWidget* widget, AnimationType type)
