diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index fc7b33cbd4a78bd4881e213341c5cea4ff92c724..415cb54fc5936c6041f2a8449b08c44b5c630119 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -304,6 +304,22 @@ ecm_add_app_icon(dolphin_SRCS ICONS ${ICONS_SRCS})
 
 kf5_add_kdeinit_executable(dolphin ${dolphin_SRCS})
 
+if (APPLE)
+    # own plist template
+    set_target_properties (dolphin PROPERTIES MACOSX_BUNDLE_INFO_PLIST ${CMAKE_CURRENT_SOURCE_DIR}/MacOSXBundleInfo.plist.in)
+    # the MacOSX bundle display name property (CFBundleDisplayName) is not currently supported by cmake,
+    # so has to be set for all targets in this cmake file
+    set(MACOSX_BUNDLE_DISPLAY_NAME Okular5)
+    set_target_properties(dolphin PROPERTIES MACOSX_BUNDLE_GUI_IDENTIFIER "org.kde.Dolphin")
+    set_target_properties(dolphin PROPERTIES MACOSX_BUNDLE_BUNDLE_NAME "Dolphin5")
+    set_target_properties(dolphin PROPERTIES MACOSX_BUNDLE_DISPLAY_NAME "Dolphin5")
+    set_target_properties(dolphin PROPERTIES MACOSX_BUNDLE_INFO_STRING "Dolphin, KDE's file manager focusing on usability")
+    set_target_properties(dolphin PROPERTIES MACOSX_BUNDLE_LONG_VERSION_STRING "Dolphin ${KDE_APPLICATIONS_VERSION}")
+    set_target_properties(dolphin PROPERTIES MACOSX_BUNDLE_SHORT_VERSION_STRING "${KDE_APPLICATIONS_VERSION_MAJOR}.${KDE_APPLICATIONS_VERSION_MINOR}")
+    set_target_properties(dolphin PROPERTIES MACOSX_BUNDLE_BUNDLE_VERSION "${KDE_APPLICATIONS_VERSION}")
+    set_target_properties(dolphin PROPERTIES MACOSX_BUNDLE_COPYRIGHT "2002-2020 The Dolphin Authors")
+endif (APPLE)
+
 
 target_link_libraries(kdeinit_dolphin PUBLIC
     dolphinprivate
diff --git a/src/MacOSXBundleInfo.plist.in b/src/MacOSXBundleInfo.plist.in
new file mode 100644
index 0000000000000000000000000000000000000000..2f043cf2568a327fd9b382dee6374541e7c39452
--- /dev/null
+++ b/src/MacOSXBundleInfo.plist.in
@@ -0,0 +1,73 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple Computer//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+        <key>NSPrincipalClass</key>
+        <string>NSApplication</string>
+        <key>NSHighResolutionCapable</key>
+        <string>True</string>
+        <key>CFBundleDevelopmentRegion</key>
+        <string>English</string>
+        <key>CFBundleExecutable</key>
+        <string>${MACOSX_BUNDLE_EXECUTABLE_NAME}</string>
+        <key>CFBundleGetInfoString</key>
+        <string>${MACOSX_BUNDLE_INFO_STRING}</string>
+        <key>CFBundleIconFile</key>
+        <string>${MACOSX_BUNDLE_ICON_FILE}</string>
+        <key>CFBundleIdentifier</key>
+        <string>${MACOSX_BUNDLE_GUI_IDENTIFIER}</string>
+        <key>CFBundleInfoDictionaryVersion</key>
+        <string>6.0</string>
+        <key>CFBundleLongVersionString</key>
+        <string>${MACOSX_BUNDLE_LONG_VERSION_STRING}</string>
+        <key>CFBundleName</key>
+        <string>${MACOSX_BUNDLE_BUNDLE_NAME}</string>
+        <key>CFBundlePackageType</key>
+        <string>APPL</string>
+        <key>CFBundleShortVersionString</key>
+        <string>${MACOSX_BUNDLE_SHORT_VERSION_STRING}</string>
+        <key>CFBundleSignature</key>
+        <string>????</string>
+        <key>CFBundleVersion</key>
+        <string>${MACOSX_BUNDLE_BUNDLE_VERSION}</string>
+        <key>CSResourcesFileMapped</key>
+        <true/>
+        <key>LSRequiresCarbon</key>
+        <true/>
+        <key>NSHumanReadableCopyright</key>
+        <string>${MACOSX_BUNDLE_COPYRIGHT}</string>
+        <key>CFBundleDocumentTypes</key>
+        <array>
+            <dict>
+                <key>CFBundleTypeName</key>
+                <string>Folder</string>
+                <key>CFBundleTypeOSTypes</key>
+                <array>
+                    <string>fold</string>
+                </array>
+                <key>CFBundleTypeRole</key>
+                <string>Editor</string>
+            </dict>
+            <dict>
+                <key>CFBundleTypeName</key>
+                <string>Volume</string>
+                <key>CFBundleTypeOSTypes</key>
+                <array>
+                    <string>disk</string>
+                </array>
+                <key>CFBundleTypeRole</key>
+                <string>Editor</string>
+            </dict>
+            <dict>
+                <key>CFBundleTypeExtensions</key>
+                <array>
+                    <string>*</string>
+                </array>
+                <key>CFBundleTypeName</key>
+                <string>NSStringPboardType</string>
+                <key>CFBundleTypeRole</key>
+                <string>Viewer</string>
+            </dict>
+        </array>
+</dict>
+</plist>
diff --git a/src/dolphinmainwindow.cpp b/src/dolphinmainwindow.cpp
index 3ac3ab1c113b7b6e27486c538755759df917349d..1d9456bdbd014f41e9acf189abf6f01b2f7e7654 100644
--- a/src/dolphinmainwindow.cpp
+++ b/src/dolphinmainwindow.cpp
@@ -931,7 +931,11 @@ void DolphinMainWindow::compareFiles()
     QUrl urlA = items.at(0).url();
     QUrl urlB = items.at(1).url();
 
+#ifdef Q_OS_MACOS
+    QString command(QStringLiteral("open -a kompare --args -c \""));
+#else
     QString command(QStringLiteral("kompare -c \""));
+#endif
     command.append(urlA.toDisplayString(QUrl::PreferLocalFile));
     command.append("\" \"");
     command.append(urlB.toDisplayString(QUrl::PreferLocalFile));
diff --git a/src/dolphinpart.cpp b/src/dolphinpart.cpp
index 0c41b2becaeecab1fac93cde86fc143978818ac1..8e4810792d3017550a4ba8feaa98fc33da350d19 100644
--- a/src/dolphinpart.cpp
+++ b/src/dolphinpart.cpp
@@ -547,7 +547,11 @@ void DolphinPart::slotFindFile()
     if (!(actions.isEmpty())) {
         actions.first()->trigger();
     } else {
+#ifdef Q_OS_MACOS
+        KIO::CommandLauncherJob *job = new KIO::CommandLauncherJob(QStringLiteral("open -a kfind --args"), {url().toString()}, this);
+#else
         KIO::CommandLauncherJob *job = new KIO::CommandLauncherJob(QStringLiteral("kfind"), {url().toString()}, this);
+#endif
         job->setDesktopName(QStringLiteral("org.kde.kfind"));
         job->setUiDelegate(new KDialogJobUiDelegate(KJobUiDelegate::AutoHandlingEnabled, widget()));
         job->start();
diff --git a/src/dolphintabwidget.cpp b/src/dolphintabwidget.cpp
index fba6fe084992dc50a7b20f8ce21d984b2afc4bc3..9614529a7df500bcf132645f08476044775e7ae5 100644
--- a/src/dolphintabwidget.cpp
+++ b/src/dolphintabwidget.cpp
@@ -334,7 +334,7 @@ void DolphinTabWidget::detachTab(int index)
     }
     args << QStringLiteral("--new-window");
 
-    KIO::CommandLauncherJob *job = new KIO::CommandLauncherJob("dolphin", args, this);
+    KIO::CommandLauncherJob *job = new KIO::CommandLauncherJob(QCoreApplication::applicationFilePath(), args, this);
     job->setDesktopName(QStringLiteral("org.kde.dolphin"));
     job->start();
 
diff --git a/src/global.cpp b/src/global.cpp
index 32a2d4ebb04bf6a57a7059e0c41e4af89824bcf7..7e93e66d08b4d35c93953b52893f15981e4fabb9 100644
--- a/src/global.cpp
+++ b/src/global.cpp
@@ -27,6 +27,7 @@
 #include <KIO/ApplicationLauncherJob>
 #include <KService>
 #include <KWindowSystem>
+#include <KShell>
 
 #include <QApplication>
 #include <QIcon>
@@ -53,7 +54,7 @@ QUrl Dolphin::homeUrl()
 
 void Dolphin::openNewWindow(const QList<QUrl> &urls, QWidget *window, const OpenNewWindowFlags &flags)
 {
-    QString command = QStringLiteral("dolphin --new-window");
+    QString command = KShell::quoteArgs((QStringLiteral("%1 --new-window").arg(QCoreApplication::applicationFilePath()));
 
     if (flags.testFlag(OpenNewWindowFlag::Select)) {
         command.append(QLatin1String(" --select"));
diff --git a/src/main.cpp b/src/main.cpp
index 802e64d2579352b70a32752da732ffdb2d379fb4..0c39c6133d312b37c8dde8da77e8570b7d30f6eb 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -45,6 +45,59 @@
 #endif
 #include <iostream>
 
+#ifdef Q_OS_MACOS
+class OpenFileEventHandler : public QObject
+{
+    Q_OBJECT
+public:
+    OpenFileEventHandler(QApplication *parent)
+        : QObject(parent)
+    {
+        parent->installEventFilter(this);
+    }
+
+    bool eventFilter(QObject *obj, QEvent *event) override
+    {
+        if (event->type() == QEvent::FileOpen) {
+            QFileOpenEvent *openEvent = static_cast<QFileOpenEvent*>(event);
+            qCDebug(DolphinDebug) << "File open event:" << openEvent->url();
+            m_urls.append(Dolphin::validateUris(QStringList(openEvent->file())));
+            if (m_mainWindow && !m_urls.isEmpty()) {
+                if (GeneralSettings::openExternallyCalledFolderInNewTab()) {
+                    const auto serviceName = QStringLiteral("org.kde.dolphin-%1").arg(QCoreApplication::applicationPid());
+                    if (!Dolphin::attachToExistingInstance(m_urls, false, false, serviceName)) {
+                        qCWarning(DolphinDebug) << "Failed to open" << m_urls;
+                    }
+                } else {
+                    Dolphin::openNewWindow(m_urls, m_mainWindow);
+                }
+                m_urls.clear();
+            }
+            return true;
+        }
+        return QObject::eventFilter(obj, event);
+    }
+
+    QList<QUrl> popUrls()
+    {
+        const auto ret = m_urls;
+        m_urls.clear();
+        return ret;
+    }
+
+
+    void setMainWindow(DolphinMainWindow *w)
+    {
+        m_mainWindow = w;
+    }
+
+private:
+    DolphinMainWindow* m_mainWindow = nullptr;
+    QList<QUrl> m_urls;
+};
+
+#endif
+
 extern "C" Q_DECL_EXPORT int kdemain(int argc, char **argv)
 {
 #ifndef Q_OS_WIN
@@ -70,6 +123,10 @@ extern "C" Q_DECL_EXPORT int kdemain(int argc, char **argv)
     app.setWindowIcon(QIcon::fromTheme(QStringLiteral("system-file-manager"), app.windowIcon()));
 
     KCrash::initialize();
+#ifdef Q_OS_MACOS
+    // start processing Mac OS FileOpenEvents
+    const auto openEventHandler = new OpenFileEventHandler(&app);
+#endif
 
     Kdelibs4ConfigMigrator migrate(QStringLiteral("dolphin"));
     migrate.setConfigFiles(QStringList() << QStringLiteral("dolphinrc"));
@@ -140,17 +197,31 @@ extern "C" Q_DECL_EXPORT int kdemain(int argc, char **argv)
     const bool openFiles = parser.isSet(QStringLiteral("select"));
     const QStringList args = parser.positionalArguments();
     QList<QUrl> urls = Dolphin::validateUris(args);
-    // We later mutate urls, so we need to store if it was empty originally
-    const bool startedWithURLs = !urls.isEmpty();
-
 
     if (parser.isSet(QStringLiteral("daemon"))) {
         KDBusService dolphinDBusService;
         DBusInterface interface;
         interface.setAsDaemon();
+#ifdef Q_OS_MACOS
+        // we probably shouldn't be accepting requests via Apple's LaunchServices
+        app.removeEventFilter(openEventHandler);
+#endif
         return app.exec();
     }
 
+#ifdef Q_OS_MACOS
+    // Get the file open events that have been queued; a priori those are only
+    // for the documents or folders with which the user launched us. This has to
+    // be done in "jit" fashion so we don't drop any events. Any such events \
+    // coming in from now until we call setMainWindow will get lost.
+    // Since D11382 we also need to get the startup urls before setting startedWithURLs.
+    QCoreApplication::sendPostedEvents(&app, QEvent::FileOpen);
+    QCoreApplication::processEvents();
+    urls.append(openEventHandler->popUrls());
+#endif
+    // We later mutate urls, so we need to store if it was empty originally
+    const bool startedWithURLs = !urls.isEmpty();
+
     if (!parser.isSet(QStringLiteral("new-window"))) {
         if (Dolphin::attachToExistingInstance(urls, openFiles, splitView)) {
             // Successfully attached to existing instance of Dolphin
@@ -158,6 +229,14 @@ extern "C" Q_DECL_EXPORT int kdemain(int argc, char **argv)
         }
     }
 
+    // Open the main window now, so we can activate normal file open event
+    // handling (on Mac) as soon as possible.
+    DolphinMainWindow* mainWindow = new DolphinMainWindow();
+
+#ifdef Q_OS_MACOS
+    openEventHandler->setMainWindow(mainWindow);
+#endif
+
     if (!startedWithURLs) {
         // We need at least one URL to open Dolphin
         urls.append(Dolphin::homeUrl());
@@ -168,8 +247,6 @@ extern "C" Q_DECL_EXPORT int kdemain(int argc, char **argv)
         urls.append(urls.last());
     }
 
-    DolphinMainWindow* mainWindow = new DolphinMainWindow();
-
     if (openFiles) {
         mainWindow->openFiles(urls, splitView);
     } else {
@@ -205,3 +282,5 @@ extern "C" Q_DECL_EXPORT int kdemain(int argc, char **argv)
 
     return app.exec(); // krazy:exclude=crash;
 }
+
+#include "main.moc"
diff --git a/src/settings/services/servicemenuinstaller/CMakeLists.txt b/src/settings/services/servicemenuinstaller/CMakeLists.txt
index 9888999368267e519553bc125e1d4c3adec2e440..398edf0315444f6cd2985169455d21c5a28f4b21 100644
--- a/src/settings/services/servicemenuinstaller/CMakeLists.txt
+++ b/src/settings/services/servicemenuinstaller/CMakeLists.txt
@@ -1,7 +1,9 @@
+include(ECMMarkNonGuiExecutable)
 remove_definitions(-DTRANSLATION_DOMAIN=\"dolphin\")
 add_definitions(-DTRANSLATION_DOMAIN=\"dolphin_servicemenuinstaller\")
 
 add_executable(servicemenuinstaller servicemenuinstaller.cpp)
+ecm_mark_nongui_executable(servicemenuinstaller)
 target_link_libraries(servicemenuinstaller PRIVATE
     Qt5::Core
     Qt5::Gui
