diff --git a/src/panels/folders/folderspanel.cpp b/src/panels/folders/folderspanel.cpp
index 2e7e4dc28c1ee8d88a511044efec48bed940af37..323b422cfc3cc417e492ae338fb329658eb6a1e1 100644
--- a/src/panels/folders/folderspanel.cpp
+++ b/src/panels/folders/folderspanel.cpp
@@ -294,12 +294,13 @@ void FoldersPanel::slotLoadingCompleted()
 
 void FoldersPanel::startFadeInAnimation()
 {
-    QPropertyAnimation* anim = new QPropertyAnimation(m_controller->view(), "opacity", this);
-    anim->setStartValue(0);
-    anim->setEndValue(1);
-    anim->setEasingCurve(QEasingCurve::InOutQuad);
-    anim->start(QAbstractAnimation::DeleteWhenStopped);
-    anim->setDuration(200);
+//     QPropertyAnimation* anim = new QPropertyAnimation(m_controller->view(), "opacity", this);
+//     anim->setStartValue(0);
+//     anim->setEndValue(1);
+//     anim->setEasingCurve(QEasingCurve::InOutQuad);
+//     anim->start(QAbstractAnimation::DeleteWhenStopped);
+//     anim->setDuration(200);
+    m_controller->view()->setOpacity(1);
 }
 
 void FoldersPanel::loadTree(const QUrl& url, FoldersPanel::NavigationBehaviour navigationBehaviour)
