commit 958b54fc53bfe85330b3d6ed13bae5415be5a961
Author: René Bertin <rjvbertin@gmail.com>
Date:   Thu May 30 13:59:49 2024 +0200

    patch-no-update-nagging.diff

diff --git a/src/macports1.0/macports.tcl b/src/macports1.0/macports.tcl
index dbf055d1e..e552fbf24 100644
--- a/src/macports1.0/macports.tcl
+++ b/src/macports1.0/macports.tcl
@@ -1446,9 +1446,9 @@ match macports.conf.default."
         set default_source_url [lindex $sources_default 0]
         if {[macports::getprotocol $default_source_url] eq "file" || [macports::getprotocol $default_source_url] eq "rsync"} {
             set default_portindex [macports::getindex $default_source_url]
-            if {[file exists $default_portindex] && [clock seconds] - [file mtime $default_portindex] > 1209600} {
-                ui_warn "port definitions are more than two weeks old, consider updating them by running 'port selfupdate'."
-            }
+#             if {[file exists $default_portindex] && [clock seconds] - [file mtime $default_portindex] > 1209600} {
+#                 ui_warn "port definitions are more than two weeks old, consider updating them by running 'port selfupdate'."
+#             }
         }
     }
 
