commit 66f2a4d0ac8ba6e64416ab5fb6dfa179af7df7d1
Author: René Bertin <rjvbertin@gmail.com>
Date:   Thu May 30 13:59:40 2024 +0200

    patch-ignore-empty-env_array.diff

diff --git a/src/port1.0/portutil.tcl b/src/port1.0/portutil.tcl
index 80b334042..b212f642d 100644
--- a/src/port1.0/portutil.tcl
+++ b/src/port1.0/portutil.tcl
@@ -898,10 +898,14 @@ proc append_to_environment_value {command key args} {
 # debugging purposes.
 proc environment_array_to_string {environment_array} {
     upvar 1 ${environment_array} env_array
-    foreach {key value} [array get env_array] {
-        lappend env_list $key='$value'
+    if {[array size env_array] > 0} {
+        foreach {key value} [array get env_array] {
+            lappend env_list $key='$value'
+        }
+        return "\n[join [lsort $env_list] "\n"]"
+    } else {
+        return ""
     }
-    return "\n[join [lsort $env_list] "\n"]"
 }
 
 ########### Distname utility functions ###########
