commit bf8c04ddc5fa85de4ae93f0396426a135dfb9b63
Author: René Bertin <rjvbertin@gmail.com>
Date:   Thu May 30 14:00:01 2024 +0200

    patch-parallel-xz.diff

diff --git a/src/port1.0/portinstall.tcl b/src/port1.0/portinstall.tcl
index 70bd4eebf..0f12f0175 100644
--- a/src/port1.0/portinstall.tcl
+++ b/src/port1.0/portinstall.tcl
@@ -82,7 +82,8 @@ proc portinstall::create_archive {location archive.type} {
            portinstall::file_is_binary portinstall::cxx_stdlib_overridden \
            cxx_stdlib PortInfo installPlist \
            archive.env archive.cmd archive.pre_args archive.args \
-           archive.post_args archive.dir depends_lib depends_run
+           archive.post_args archive.dir depends_lib depends_run \
+           build.jobs prefix
     set archive.env {}
     set archive.cmd {}
     set archive.pre_args {}
@@ -122,6 +123,7 @@ proc portinstall::create_archive {location archive.type} {
                 set archive.cmd "$tar"
                 set archive.pre_args {-cvf}
                 if {[regexp {z2?$} ${archive.type}]} {
+                    set extraargs ""
                     if {[regexp {bz2?$} ${archive.type}]} {
                         if {![catch {binaryInPath lbzip2}]} {
                             set gzip "lbzip2"
@@ -135,8 +137,13 @@ proc portinstall::create_archive {location archive.type} {
                         set gzip "lzma"
                         set level ""
                     } elseif {[regexp {xz$} ${archive.type}]} {
-                        set gzip "xz"
+                        if {![catch {binaryInPath xz}]} {
+                            set gzip "xz"
+                        } else {
+                            set gzip "$prefix/sbin/xz"
+                        }
                         set level 6
+                        set extraargs "--threads=${build.jobs}"
                     } else {
                         set gzip "gzip"
                         set level 9
@@ -147,9 +154,9 @@ proc portinstall::create_archive {location archive.type} {
                         set hint ""
                     }
                     if {[catch {set gzip [findBinary $gzip $hint]} errmsg] == 0} {
-                        ui_debug "Using $gzip"
+                        ui_debug "Using $gzip -c$level $extraargs"
                         set archive.args {- .}
-                        set archive.post_args "| $gzip -c$level > ${location}"
+                        set archive.post_args "| $gzip -c$level $extraargs > ${location}"
                     } else {
                         ui_debug $errmsg
                         return -code error "No '$gzip' was found on this system!"
