commit 80df2b51c8b2cb263a748e273308ec3620ee12ae
Author: René Bertin <rjvbertin@gmail.com>
Date:   Thu May 30 14:00:02 2024 +0200

    patch-support-lto.diff

diff --git a/src/cregistry/Makefile.in b/src/cregistry/Makefile.in
index 390c38d25..7d1986674 100644
--- a/src/cregistry/Makefile.in
+++ b/src/cregistry/Makefile.in
@@ -4,7 +4,6 @@ VPATH  = @srcdir@
 
 OBJS = registry.o entry.o sql.o vercomp.o util.o file.o portgroup.o
 STLIB_NAME = cregistry.a
-RANLIB = ranlib
 
 SQLEXT_NAME = macports.sqlext
 SQLEXT_OBJS = sqlext.o vercomp.o
@@ -23,8 +22,8 @@ all:: ${STLIB_NAME}
 	${CC} -c -DUSE_TCL_STUBS ${CFLAGS} ${CPPFLAGS} ${SHLIB_CFLAGS} $< -o $@
 
 ${STLIB_NAME}:: ${OBJS}
-	${STLIB_LD} ${STLIB_NAME} ${OBJS}
-	${RANLIB} ${STLIB_NAME}
+	@STLIB_LD@ ${STLIB_NAME} ${OBJS}
+	@RANLIB@ ${STLIB_NAME}
 
 ${SQLEXT_NAME}: ${SQLEXT_OBJS}
 	${SHLIB_LD} ${SQLEXT_OBJS} -o $@
diff --git a/src/darwintracelib1.0/Makefile.in b/src/darwintracelib1.0/Makefile.in
index be77bdf89..28d5c7d6c 100644
--- a/src/darwintracelib1.0/Makefile.in
+++ b/src/darwintracelib1.0/Makefile.in
@@ -40,6 +40,9 @@ INSTALLDIR = $(TCL_PACKAGE_PATH)/darwintrace1.0
 # case of darwintrace we still need them, though.
 CFLAGS_PEDANTIC =
 CFLAGS += -fPIC $(UNIVERSAL_ARCHFLAGS)
+# we cannot rely on having a recent enough 'lipo' command that knows LLVM bitcode
+# so disable LTO here
+CFLAGS += -fno-lto
 SHLIB_LDFLAGS += $(UNIVERSAL_ARCHFLAGS)
 ifeq (darwin,@OS_PLATFORM@)
 SHLIB_LDFLAGS += -install_name $(INSTALLDIR)/$(SHLIB_NAME)
diff --git a/src/macports1.0/Makefile.in b/src/macports1.0/Makefile.in
index 4af97003e..a3349f48c 100644
--- a/src/macports1.0/Makefile.in
+++ b/src/macports1.0/Makefile.in
@@ -2,6 +2,9 @@ srcdir = @srcdir@
 VPATH  = @srcdir@
 
 include ../../Mk/macports.autoconf.mk
+# for some reason, Tcl's pkgIndex.tcl generation crashes in the MacPorts.so
+# init procedure when built with LTO (on Linux at least).
+CFLAGS += -fno-lto
 
 SRCS=		macports.tcl macports_dlist.tcl macports_util.tcl \
 		macports_autoconf.tcl diagnose.tcl reclaim.tcl selfupdate.tcl
diff --git a/vendor/tclx8.4/Makefile.in b/vendor/tclx8.4/Makefile.in
index b812bbe71..9c57ca5a8 100644
--- a/vendor/tclx8.4/Makefile.in
+++ b/vendor/tclx8.4/Makefile.in
@@ -123,7 +123,7 @@ OBJEXT		= @OBJEXT@
 RANLIB		= @RANLIB@
 RANLIB_STUB	= @RANLIB_STUB@
 SHLIB_CFLAGS	= @SHLIB_CFLAGS@
-SHLIB_LD	= @SHLIB_LD@
+SHLIB_LD	= @SHLIB_LD@ $(LDFLAGS_DEFAULT)
 SHLIB_LD_LIBS	= @SHLIB_LD_LIBS@
 STLIB_LD	= @STLIB_LD@
 TCL_DEFS	= @TCL_DEFS@
@@ -168,7 +168,8 @@ CONFIG_CLEAN_FILES = Makefile
 CPPFLAGS	= @CPPFLAGS@
 LIBS		= @PKG_LIBS@ @LIBS@
 AR		= @AR@
-CFLAGS		= @CFLAGS@
+# disable building with link-time optimisation
+CFLAGS		= @CFLAGS@ -fno-lto
 COMPILE		= $(CC) $(DEFS) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
 
 #========================================================================
