diff --git a/src/port/port.tcl b/src/port/port.tcl
index a62a1400f..7a7e84c98 100755
--- a/src/port/port.tcl
+++ b/src/port/port.tcl
@@ -2181,6 +2181,20 @@ proc action_info { action portlist opts } {
                 set inf "$portinfo(name) @"
                 append inf [composite_version $portinfo(version) $portinfo(active_variants)]
                 set ropt "fullname"
+            } elseif {$opt eq "var"} {
+                if {[info exists global_options(ports_info_var)]} {
+                    set opt ${global_options(ports_info_var)}
+                    set ropt $opt
+                    if {[info exists portinfo($opt)]} {
+                        set inf "$portinfo($opt)"
+                    } else {
+                        ui_error "variable \"$opt\" cannot be queried via info --var"
+                        return 1
+                    }
+                } else {
+                    ui_error "info --var requires a variable name"
+                    return 1
+                }
             } else {
                 # Map from friendly name
                 set ropt [map_friendly_field_names $opt]
@@ -4462,7 +4476,7 @@ array set cmd_opts_array {
                  depends description epoch fullname heading homepage index license
                  line long_description
                  maintainer maintainers name patchfiles platform platforms portdir
-                 pretty replaced_by revision subports variant variants version}
+                 pretty replaced_by revision subports variant variants version {var 1}}
     contents    {size {units 1}}
     deps        {index no-build}
     rdeps       {index no-build full}
diff --git a/src/port1.0/portutil.tcl b/src/port1.0/portutil.tcl
index b212f642d..893028e80 100644
--- a/src/port1.0/portutil.tcl
+++ b/src/port1.0/portutil.tcl
@@ -90,6 +90,7 @@ proc handle_option {option args} {
 
     if {![info exists user_options($option)]} {
         set $option $args
+        options_export $option
     }
 }
 
@@ -107,6 +108,7 @@ proc handle_option-append {option args} {
         } else {
             set $option $args
         }
+        options_export $option
     }
 }
 
@@ -124,6 +126,7 @@ proc handle_option-prepend {option args} {
         } else {
             set $option $args
         }
+        options_export $option
     }
 }
 
@@ -141,6 +144,7 @@ proc handle_option-delete {option args} {
             set temp [ldelete $temp $val]
         }
         set $option $temp
+        options_export $option
     }
 }
 
@@ -158,6 +162,7 @@ proc handle_option-strsed {option args} {
             set temp [strsed $temp $val]
         }
         set $option $temp
+        options_export $option
     }
 }
 
@@ -187,6 +192,7 @@ proc handle_option-replace {option args} {
                 continue
             }
             set $option [lreplace [set $option] $index $index $new]
+            options_export $option
         }
     }
 }
