commit e7508fbf91b660a6e7c5bb53535e97dc2a0bcfc3
Author: René Bertin <rjvbertin@gmail.com>
Date:   Thu May 30 13:59:38 2024 +0200

    patch-no-CCPRINT.diff

diff --git a/src/port1.0/portmain.tcl b/src/port1.0/portmain.tcl
index 23bb4ef6c..5544c8d51 100644
--- a/src/port1.0/portmain.tcl
+++ b/src/port1.0/portmain.tcl
@@ -156,7 +156,7 @@ default universal_possible {[expr {${os.universal_supported} && [llength ${confi
 
 default compiler.cpath {${prefix}/include}
 default compiler.library_path {${prefix}/lib}
-default compiler.log_verbose_output yes
+default compiler.log_verbose_output no
 default compiler.limit_flags no
 default compiler.support_environment_paths no
 default compiler.support_environment_sdkroot no
