commit 493913de2fcb31c469d204ff0d6e64fa7a63f22e
Author: René Bertin <rjvbertin@gmail.com>
Date:   Thu May 30 13:59:42 2024 +0200

    patch-portprovides-support-symlinked-prefix.diff

diff --git a/src/port/port.tcl b/src/port/port.tcl
index a62a1400f..f3467e5a0 100755
--- a/src/port/port.tcl
+++ b/src/port/port.tcl
@@ -2472,6 +2472,11 @@ proc action_notes { action portlist opts } {
 }
 
 
+proc macports::normalize { filename } {
+    set prefmap [list [file dirname [file normalize "${macports::prefix}/foo"]] ${macports::prefix}]
+    return [string map ${prefmap} [file normalize $filename]]
+}
+
 proc action_provides { action portlist opts } {
     # In this case, portname is going to be used for the filename... since
     # that is the first argument we expect... perhaps there is a better way
@@ -2481,7 +2486,7 @@ proc action_provides { action portlist opts } {
         return 1
     }
     foreach filename $portlist {
-        set file [file normalize $filename]
+        set file [macports::normalize $filename]
         if {[file exists $file] || ![catch {file type $file}]} {
             if {![file isdirectory $file] || [file type $file] eq "link"} {
                 set port [registry::file_registered $file]
