commit 579d832faaaf99a1b632772d6ceeeeb0bdc0c6d0
Author: René Bertin <rjvbertin@gmail.com>
Date:   Thu May 30 13:59:59 2024 +0200

    patch-nowarn-unsigned-source.diff

diff --git a/src/macports1.0/macports.tcl b/src/macports1.0/macports.tcl
index 857843079..115d9a72d 100644
--- a/src/macports1.0/macports.tcl
+++ b/src/macports1.0/macports.tcl
@@ -918,10 +918,10 @@ proc mportinit {{up_ui_options {}} {up_options {}} {up_variations {}}} {
                         set sources_default [concat [list $url] $flags]
                     }
                 }
-                if {[string match rsync://*rsync.macports.org/release/ports/ $url]} {
-                    ui_warn "MacPorts is configured to use an unsigned source for the ports tree.\
-Please edit sources.conf and change '$url' to '[string range $url 0 end-6]tarballs/ports.tar'."
-                }
+#                 if {[string match rsync://*rsync.macports.org/release/ports/ $url]} {
+#                     ui_warn "MacPorts is configured to use an unsigned source for the ports tree.\
+# Please edit sources.conf and change '$url' to '[string range $url 0 end-6]tarballs/ports.tar'."
+#                 }
                 lappend sources [concat [list $url] $flags]
                 if {[info exists macports::ui_options(ports_env)]} {
                     ui_msg "$macports::ui_prefix $sources_conf: [lindex $sources end]"
