commit 66b974e20685c2027f7f1399fb29d5e9dba26345
Author: René Bertin <rjvbertin@gmail.com>
Date:   Thu May 30 13:59:53 2024 +0200

    patch-ui_info-in-info.diff

diff --git a/src/macports1.0/macports.tcl b/src/macports1.0/macports.tcl
index 70340bc04..15b72f49d 100644
--- a/src/macports1.0/macports.tcl
+++ b/src/macports1.0/macports.tcl
@@ -322,6 +322,13 @@ proc ui_message {priority prefix args} {
        }
     } 
 
+    # adopt msg priority if the current phase equals the current priority
+    # this can only happen for ui_info during `port info` (currently).
+    if {[info exists macports::current_phase]
+        && ${macports::current_phase} eq ${priority}} {
+        set priority "msg"
+    }
+
     foreach chan $macports::channels($priority) {
         if {[lindex $args 0] eq "-nonewline"} {
             puts -nonewline $chan $prefix[lindex $args 1]
diff --git a/src/port/port.tcl b/src/port/port.tcl
index c2abc557a..3b11c7a28 100755
--- a/src/port/port.tcl
+++ b/src/port/port.tcl
@@ -1944,6 +1944,7 @@ proc action_info { action portlist opts } {
         return 1
     }
 
+    set_phase "info"
     set separator ""
     foreachport $portlist {
         set index_only 0
