commit a20cc071c9d73ee39accc8af80327203892f1b06
Author: René Bertin <rjvbertin@gmail.com>
Date:   Thu May 30 14:00:03 2024 +0200

    patch-export-logfilename.diff

diff --git a/src/macports1.0/macports.tcl b/src/macports1.0/macports.tcl
index df2553e4a..adfe702d5 100644
--- a/src/macports1.0/macports.tcl
+++ b/src/macports1.0/macports.tcl
@@ -1607,6 +1607,9 @@ proc macports::worker_init {workername portpath porturl portbuildpath options va
     $workername alias _portnameactive _portnameactive
     $workername alias get_actual_cxx_stdlib macports::get_actual_cxx_stdlib
     $workername alias shellescape macports::shellescape
+    ## RJVB
+    $workername alias get_logfile macports::get_logfile
+    $workername alias flush_logfile macports::flush_logfile
 
     # New Registry/Receipts stuff
     $workername alias registry_new registry::new_entry
@@ -5845,6 +5848,24 @@ proc macports::shellescape {arg} {
     return [regsub -all -- {[^A-Za-z0-9.:@%/+=_-]} $arg {\\&}]
 }
 
+## RJVB
+# return the logfile name, or an empty string
+proc macports::get_logfile {} {
+	if {[info exists ::debuglogname]} {
+		return $::debuglogname
+	} else {
+		return ""
+	}
+}
+proc macports::flush_logfile {} {
+	if {[info exists ::debuglog]} {
+		flush $::debuglog
+		return yes
+	} else {
+		return no
+	}
+}
+
 ##
 # Given a list of maintainers as recorded in a Portfile, return a list of lists
 # in [key value ...] format describing all maintainers. Valid keys are 'email'
