commit 81a8f222d5622b61b405a5b9478bfd5e42d049b5
Author: René Bertin <rjvbertin@gmail.com>
Date:   Thu May 30 14:00:01 2024 +0200

    patch-linux-no-default-compdeps.diff

diff --git a/src/port1.0/portconfigure.tcl b/src/port1.0/portconfigure.tcl
index ec96bcc48..1d5433521 100644
--- a/src/port1.0/portconfigure.tcl
+++ b/src/port1.0/portconfigure.tcl
@@ -778,8 +778,10 @@ proc portconfigure::configure_get_default_compiler {} {
 
     if {[option compiler.whitelist] ne ""} {
         set search_list [option compiler.whitelist]
+	   ui_debug "Picking compiler from whitelist ${search_list}"
     } else {
         set search_list [option compiler.fallback]
+	   ui_debug "Picking compiler from fallbacks ${search_list}"
     }
     foreach compiler $search_list {
         set allowed yes
@@ -1256,10 +1258,11 @@ proc compiler.command_line_tools_version {compiler} {
 }
 # internal function to choose compiler fallback list based on platform
 proc portconfigure::get_compiler_fallback {} {
-    global default_compilers xcodeversion
+    global default_compilers xcodeversion os.platform
 
     # Check our override
     if {[info exists default_compilers]} {
+	   ui_debug "Taking fallback compilers from default_compilers=${default_compilers}"
         return $default_compilers
     }
 
@@ -1269,6 +1272,7 @@ proc portconfigure::get_compiler_fallback {} {
     } else {
         set available_apple_compilers [portconfigure::get_apple_compilers_xcode_version]
     }
+    ui_debug "available_apple_compilers=${available_apple_compilers}"
     set system_compilers ""
     foreach c ${available_apple_compilers} {
         set vmin [portconfigure::get_min_command_line $c]
@@ -1302,7 +1306,11 @@ proc portconfigure::get_compiler_fallback {} {
             }
         }
     }
-    set compilers ""
+    if {${os.platform} eq "darwin"} {
+        set compilers ""
+    } else {
+        set compilers "cc"
+    }
     lappend compilers {*}${system_compilers}
     # when building for PowerPC architectures, prefer GCC to Clang
     if {[option configure.build_arch] eq "ppc" || [option configure.build_arch] eq "ppc64"} {
@@ -1581,7 +1589,15 @@ proc portconfigure::add_automatic_compiler_dependencies {} {
 port::register_callback portconfigure::add_automatic_compiler_dependencies
 # and an option to turn it off if required
 options configure.compiler.add_deps
-default configure.compiler.add_deps yes
+default configure.compiler.add_deps {[portconfigure::configure_get_add_deps]}
+proc portconfigure::configure_get_add_deps {} {
+    if {[option os.platform] eq "darwin"} {
+        return yes
+    } else {
+        return no
+    }
+}
+
 # helper function to add dependencies for a given compiler
 proc portconfigure::add_compiler_port_dependencies {compiler} {
     global os.major porturl
