commit 7e1f34a09ee12555c1111c41bb21da182c1866e5
Author: René Bertin <rjvbertin@gmail.com>
Date:   Thu May 30 13:59:52 2024 +0200

    patch-silent-command-exec.diff

diff --git a/src/port1.0/portutil.tcl b/src/port1.0/portutil.tcl
index 14f8a90be..b58d0f720 100644
--- a/src/port1.0/portutil.tcl
+++ b/src/port1.0/portutil.tcl
@@ -477,7 +477,7 @@ proc command_exec {args} {
     array set env [array get ${varprefix}.env_array]
     # Call the command.
     set fullcmdstring "$command_prefix $cmdstring $command_suffix"
-    ui_info "Executing: $fullcmdstring"
+    ui_debug "Executing: $fullcmdstring"
     set code [catch {system {*}$notty {*}$callback {*}$nice $fullcmdstring} result]
     # Save variables in order to re-throw the same error code.
     set errcode $::errorCode
