commit c2e4698bfd5f8a515a548dae3d6a6cb57a28aaa9
Author: René Bertin <rjvbertin@gmail.com>
Date:   Thu May 30 13:59:51 2024 +0200

    patch-truncate-logfile.diff

diff --git a/src/macports1.0/macports.tcl b/src/macports1.0/macports.tcl
index e552fbf24..70340bc04 100644
--- a/src/macports1.0/macports.tcl
+++ b/src/macports1.0/macports.tcl
@@ -210,8 +210,8 @@ proc macports::ch_logging {mport} {
 
     set ::debuglogname $logname
 
-    # Append to the file if it already exists
-    set ::debuglog [open $::debuglogname a]
+    # Truncate to the file if it already exists
+    set ::debuglog [open $::debuglogname w]
     puts $::debuglog version:1
 
     ui_debug "Starting logging for $portname @$portinfo(version)_$portinfo(revision)$portinfo(canonical_active_variants)"
