commit 62ca74ed62022353f75c810cbae453333eb13895
Author: René Bertin <rjvbertin@gmail.com>
Date:   Thu May 30 13:59:51 2024 +0200

    port-o-option-default.diff

diff --git a/src/port/port.tcl b/src/port/port.tcl
old mode 100644
new mode 100755
index ea9730270..c2abc557a
--- a/src/port/port.tcl
+++ b/src/port/port.tcl
@@ -4554,6 +4554,8 @@ proc parse_options { action ui_options_name global_options_name } {
 
     set options_order(${action}) {}
 
+    # RJVB
+    set global_options(ports_ignore_different) yes
     while {[moreargs]} {
         set arg [lookahead]
 
