--- qtbase//mkspecs/features/mac/orig.sdk.prf	2014-09-11 12:48:07.000000000 +0200
+++ qtbase//mkspecs/features/mac/sdk.prf	2015-03-23 22:13:52.000000000 +0100
@@ -27,7 +27,7 @@
 QMAKESPEC_NAME = $$basename(QMAKESPEC)
 
 # Resolve SDK version of various tools
-for(tool, $$list(QMAKE_CC QMAKE_CXX QMAKE_FIX_RPATH QMAKE_AR QMAKE_RANLIB QMAKE_LINK QMAKE_LINK_SHLIB)) {
+for(tool, $$list(QMAKE_FIX_RPATH QMAKE_AR QMAKE_RANLIB QMAKE_LINK QMAKE_LINK_SHLIB)) {
     tool_variable = QMAKE_MAC_SDK.$${QMAKESPEC_NAME}.$${QMAKE_MAC_SDK}.$${tool}
     !isEmpty($$tool_variable) {
         $$tool = $$eval($$tool_variable)
