diff --git a/src/test/mtest.cpp b/src/test/mtest.cpp
index eccd6f8..9636181 100644
--- a/src/test/mtest.cpp
+++ b/src/test/mtest.cpp
@@ -18,6 +18,8 @@
 #include <sys/resource.h>
 #endif
 
+#if defined(__APPLE__)
+#include <malloc/malloc.h>
 #if !defined(__APPLE__) && !defined(__FreeBSD__)
 #include <malloc.h>
 #endif
diff --git a/src/test/testoverflow.cpp b/src/test/testoverflow.cpp
index 40ed799..db7bfa5 100644
--- a/src/test/testoverflow.cpp
+++ b/src/test/testoverflow.cpp
@@ -2,6 +2,8 @@
 
 #ifndef __APPLE__
 #include <malloc.h>
+#else
+#include <malloc/malloc.h>
 #endif
 
 
