--- pdf/orig.pdf_fmap.c	2022-04-04 15:46:22.000000000 +0200
+++ pdf/pdf_fmap.c	2023-04-02 19:34:16.000000000 +0200
@@ -673,6 +673,10 @@
                 continue;
 
             sf = sfopen(result, "r", ctx->memory);
+            if (!sf ) {
+                dmprintf1(ctx->memory, "Couldn't open font file \"%s\"\n", result);
+                continue;
+            }
             code = sgets(sf, magic, 4, &nread);
             if (code < 0 || nread < 4) {
                 sfclose(sf);
--- pcl/pl/orig.pllfont.c	2022-04-04 15:46:22.000000000 +0200
+++ pcl/pl/pllfont.c	2023-04-02 19:26:06.000000000 +0200
@@ -719,12 +719,14 @@
                           "TrueType font %s in file %s not found in table\n",
                           buffer, path);
                 in = sfopen(path, "r", mem);
-                code =
-                    get_name_from_tt_file(in, mem, buffer, WINDOWSNAME);
-                sfclose(in);
-                dmprintf1(mem, "Windows name %s\n", buffer);
-                if (code < 0)
-                    return code;
+                if (in) {
+                    code =
+                        get_name_from_tt_file(in, mem, buffer, WINDOWSNAME);
+                    sfclose(in);
+                    dmprintf1(mem, "Windows name %s\n", buffer);
+                    if (code < 0)
+                        return code;
+                }
             }
 #endif
         }
