commit 75ef81a1094465c36c091aa02e3108fb1a884821
Author: Albert Astals Cid <aacid@kde.org>
Date:   Thu Jul 4 22:26:53 2019 +0200

    QImage::byteCount -> QImage::sizeInByes

diff --git autotests/pictest.cpp autotests/pictest.cpp
index dcd8211..fc8113c 100644
--- autotests/pictest.cpp
+++ autotests/pictest.cpp
@@ -210,11 +210,11 @@ private Q_SLOTS:
             QFile picDumpFile(fileNameBase + QStringLiteral("-expected.data"));
             QVERIFY2(picDumpFile.open(QIODevice::WriteOnly), qPrintable(picDumpFile.errorString()));
             picDumpFile.write(reinterpret_cast<const char *>(inputImage.bits()),
-                              inputImage.sizeInBytes());
+                              inputImage.byteCount());
             QFile pngDumpFile(fileNameBase + QStringLiteral("-actual.data"));
             QVERIFY2(pngDumpFile.open(QIODevice::WriteOnly), qPrintable(pngDumpFile.errorString()));
             pngDumpFile.write(reinterpret_cast<const char *>(expImage.bits()),
-                              expImage.sizeInBytes());
+                              expImage.byteCount());
             QString msg = QStringLiteral("Read image (")
                         + picDumpFile.fileName()
                         + QStringLiteral(") differed from expected image (")
diff --git autotests/readtest.cpp autotests/readtest.cpp
index ae00fc3..bbda287 100644
--- autotests/readtest.cpp
+++ autotests/readtest.cpp
@@ -35,8 +35,8 @@ static void writeImageData(const char *name, const QString &filename, const QIma
 {
     QFile file(filename);
     if (file.open(QIODevice::WriteOnly)) {
-        qint64 written = file.write(reinterpret_cast<const char *>(image.bits()), image.sizeInBytes());
-        if (written == image.sizeInBytes()) {
+        qint64 written = file.write(reinterpret_cast<const char *>(image.bits()), image.byteCount());
+        if (written == image.byteCount()) {
             QTextStream(stdout) << "       " << name
                                 << " written to " << filename << "\n";
         } else {
diff --git tests/imagedump.cpp tests/imagedump.cpp
index fec1870..1ecba11 100644
--- tests/imagedump.cpp
+++ tests/imagedump.cpp
@@ -116,8 +116,8 @@ int main(int argc, char **argv)
         }
         img = img.convertToFormat(qformat);
     }
-    qint64 written = output.write(reinterpret_cast<const char *>(img.bits()), img.sizeInBytes());
-    if (written != img.sizeInBytes()) {
+    qint64 written = output.write(reinterpret_cast<const char *>(img.bits()), img.byteCount());
+    if (written != img.byteCount()) {
         QTextStream(stderr) << "Could not write image data to " << files.at(1)
                             << ":" << output.errorString() << "\n";
         return 5;
