diff --git src/unix/stackwalk.cpp src/unix/stackwalk.cpp
index eccb910..8040474 100644
--- src/unix/stackwalk.cpp
+++ src/unix/stackwalk.cpp
@@ -249,7 +249,7 @@ int wxStackWalker::InitFrames(wxStackFrame *arr, size_t n, void **addresses, cha
     // build the command line for executing addr2line or atos under OS X using
     // char* directly to avoid the conversions from Unicode
 #ifdef __WXOSX__
-    int len = snprintf(g_buf, BUFSIZE, "atos -p %d", (int)getpid());
+    int len = snprintf(g_buf, BUFSIZE, "xcrun atos -p %d", (int)getpid());
 #else
     int len = snprintf(g_buf, BUFSIZE, "addr2line -C -f -e \"%s\"", (const char*) exepath.mb_str());
 #endif
