--- src/lib/utils/orig.os_utils.cpp	2019-03-30 09:39:06.000000000 +0100
+++ src/lib/utils/os_utils.cpp	2019-06-11 16:53:33.000000000 +0200
@@ -350,7 +350,7 @@
 
       ptr = ::mmap(nullptr, 2*page_size,
                    PROT_READ | PROT_WRITE,
-                   MAP_ANONYMOUS | MAP_PRIVATE | MAP_NOCORE,
+                   MAP_ANON | MAP_PRIVATE | MAP_NOCORE,
                    /*fd=*/-1, /*offset=*/0);
 
       if(ptr == MAP_FAILED)
