--- src/orig.kicontheme.cpp	2017-06-20 17:27:42.000000000 +0200
+++ src/kicontheme.cpp	2017-06-20 17:32:10.000000000 +0200
@@ -58,6 +58,7 @@
                 QIcon::setThemeName(iconThemeName); // Qt looks under :/icons automatically
                 // Tell KIconTheme about the theme, in case KIconLoader is used directly
                 *_themeOverride() = iconThemeName;
+                qWarning() << Q_FUNC_INFO << "Using theme" << iconThemeName << "from" << iconThemeRcc;
             } else {
                 qWarning() << "No index.theme found in" << iconThemeRcc;
                 QResource::unregisterResource(iconThemeRcc, iconSubdir);
@@ -535,8 +536,13 @@
 
     QString theme;
     // Check application specific config for a theme setting.
-    KConfigGroup app_cg(KSharedConfig::openConfig(QString(), KConfig::NoGlobals), "Icons");
+    KSharedConfigPtr kconf = KSharedConfig::openConfig(QString(), KConfig::NoGlobals);
+    KConfigGroup app_cg(kconf, "Icons");
     theme = app_cg.readEntry("Theme", QString());
+    if (!theme.isEmpty()) {
+        qDebug() << Q_FUNC_INFO << "Using theme" << theme << "according to the Icons section in"
+            << QStandardPaths::locate(kconf->locationType(), kconf->name());
+    }
     if (theme.isEmpty() || theme == QLatin1String("hicolor")) {
         // No theme, try to use Qt's. A Platform plugin might have set
         // a good theme there.
