--- qtbase/src/gui/kernel/orig.qguiapplication.cpp	2015-03-03 04:26:47.000000000 -0800
+++ qtbase/src/gui/kernel/qguiapplication.cpp	2015-03-03 04:30:24.000000000 -0800
@@ -1258,6 +1258,10 @@
         argc = j;
     }
 
+    // https://bugreports.qt-project.org/browse/QTBUG-32974
+    if (qgetenv("QT_LOAD_TESTABILITY").toInt() > 0)
+        loadTestability = true;
+
     // Load environment exported generic plugins
     foreach (const QByteArray &plugin, qgetenv("QT_QPA_GENERIC_PLUGINS").split(','))
         pluginList << plugin;
