--- qtbase/tests/auto/other/qaccessibilitymac/tst_qaccessibilitymac_helpers.mm
+++ qtbase/tests/auto/other/qaccessibilitymac/tst_qaccessibilitymac_helpers.mm
@@ -116,7 +116,7 @@
 
 - (AXUIElementRef) ref { return reference; }
 - (void) print {
-    NSLog(@"Accessible Object role: '%@', description: '%@', value: '%@', rect: '%@'", self.role, self.description, self.value, NSStringFromRect(self.rect));
+    NSLog(@"Accessible Object role: '%@', description: '%@', value: '%@', rect: '%@'", self.role, self.description, self.value, NSStringFromRect(NSRectFromCGRect(self.rect)));
     NSLog(@"    Children: %ld", [[self childList] count]);
 }
 
