diff --git orig.configure configure
index baef4e2..86bb59f 100755
--- orig.configure
+++ configure
@@ -32042,24 +32042,6 @@ if test ${ACCESS_UNALIGNED} = no; then
 	CPPFLAGS="$CPPFLAGS -DNO_UNALIGNED_ACCESS"
 fi
 
-if test x$host_darwin = xyes; then
-	{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for ranlib that supports -no_warning_for_no_symbols option" >&5
-$as_echo_n "checking for ranlib that supports -no_warning_for_no_symbols option... " >&6; }
-	if $RANLIB -no_warning_for_no_symbols 2>&1 | grep -q "unknown option"; then :
-  { $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
-$as_echo "no" >&6; }
-else
-
-			# avoid AR calling ranlib, libtool calls it anyway. suppress no symbols warning.
-			AR_FLAGS="Scru"
-			RANLIB="$RANLIB -no_warning_for_no_symbols"
-			{ $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
-$as_echo "yes" >&6; }
-
-
-fi
-fi
-
 case "x$libgc" in
 	xincluded)
 		# Pass CPPFLAGS to libgc configure
diff --git orig.configure.ac configure.ac
index 2ab3a14..7df4860 100644
--- orig.configure.ac
+++ configure.ac
@@ -5635,20 +5635,6 @@ if test ${ACCESS_UNALIGNED} = no; then
 	CPPFLAGS="$CPPFLAGS -DNO_UNALIGNED_ACCESS"
 fi
 
-if test x$host_darwin = xyes; then
-	AC_MSG_CHECKING([for ranlib that supports -no_warning_for_no_symbols option])
-	AS_IF(
-		[$RANLIB -no_warning_for_no_symbols 2>&1 | grep -q "unknown option"],
-		[AC_MSG_RESULT([no])],
-		[
-			# avoid AR calling ranlib, libtool calls it anyway. suppress no symbols warning.
-			AR_FLAGS="Scru"
-			RANLIB="$RANLIB -no_warning_for_no_symbols"
-			AC_MSG_RESULT([yes])
-		]
-	)
-fi
-
 case "x$libgc" in
 	xincluded)
 		# Pass CPPFLAGS to libgc configure
