--- orig.CMakeLists.txt	2016-10-02 11:58:37.000000000 +0200
+++ CMakeLists.txt	2016-12-10 12:55:39.000000000 +0100
@@ -18,6 +18,8 @@
 
 include(ECMQtDeclareLoggingCategory)
 
+include(ECMMarkNonGuiExecutable)
+
 include(KDEInstallDirs)
 include(KDEFrameworkCompilerSettings NO_POLICY_SCOPE)
 include(KDECMakeSettings)
--- src/console/orig.CMakeLists.txt	2016-10-02 11:58:37.000000000 +0200
+++ src/console/CMakeLists.txt	2016-12-10 12:56:26.000000000 +0100
@@ -1,5 +1,6 @@
 #This binary is used for any non-GUI Kross scripts, e.g. Calligra Sheets has some
 add_executable(kf5kross main.cpp)
+ecm_mark_nongui_executable(kf5kross)
 target_link_libraries(kf5kross
    KF5::CoreAddons
    KF5::KrossCore
--- autotests/orig.CMakeLists.txt	2016-10-02 11:58:37.000000000 +0200
+++ autotests/CMakeLists.txt	2016-12-10 12:56:12.000000000 +0100
@@ -5,6 +5,7 @@
 set(krossqtstest_SRCS main.cpp)
 add_executable(krossqtstest qts.cpp)
 ecm_mark_as_test(krossqtstest)
+ecm_mark_nongui_executable(krossqtstest)
 target_link_libraries(krossqtstest
    KF5::KrossCore
    Qt5::Script
@@ -17,6 +18,7 @@
 
 add_executable(metafunctiontest metafunctiontest.cpp)
 ecm_mark_as_test(metafunctiontest)
+ecm_mark_nongui_executable(metafunctiontest)
 target_link_libraries(metafunctiontest
    Qt5::Test
    KF5::KrossCore
