--- misc/kde4breeze/src/orig.CMakeLists.txt	2016-03-01 14:25:55.000000000 +0100
+++ misc/kde4breeze/src/CMakeLists.txt	2016-03-09 18:21:34.000000000 +0100
@@ -7,6 +7,7 @@
 
 # Set the name of the application
 add_executable(kde4breeze ${kde4breeze_SRCS})
+ecm_mark_nongui_executable(kde4breeze)
 
 # Select which libraries we need to link to
 target_link_libraries(kde4breeze Qt5::Widgets KF5::CoreAddons KF5::ConfigCore)
