diff --git CMakeLists.txt CMakeLists.txt
index f54bfc82..0806d960 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -8,8 +8,8 @@ include(WriteBasicConfigVersionFile)
 include(FeatureSummary)
 
 ################# Qt/KDE #################
-set(QT_MIN_VERSION "5.15.0")
-set(KF5_MIN_VERSION "5.78")
+set(QT_MIN_VERSION "5.9.8")
+set(KF5_MIN_VERSION "5.60")
 
 
 find_package(ECM ${KF5_MIN_VERSION} REQUIRED NO_MODULE)
diff --git kdecoration/CMakeLists.txt kdecoration/CMakeLists.txt
index 66498bd3..1b950c0b 100644
--- kdecoration/CMakeLists.txt
+++ kdecoration/CMakeLists.txt
@@ -72,9 +72,9 @@ add_library(oxygendecoration MODULE
 
 target_link_libraries(oxygendecoration
     PUBLIC
-        Qt::Core
-        Qt::Gui
-        Qt::DBus
+        Qt5::Core
+        Qt5::Gui
+        Qt5::DBus
     PRIVATE
         oxygenstyle5
         oxygenstyleconfig5
@@ -89,7 +89,7 @@ target_link_libraries(oxygendecoration
 if(OXYGEN_HAVE_X11)
   target_link_libraries(oxygendecoration
     PUBLIC
-      Qt::X11Extras
+      Qt5::X11Extras
       XCB::XCB)
 endif()
 
diff --git kstyle/CMakeLists.txt kstyle/CMakeLists.txt
index b0680f14..e372f9f2 100644
--- kstyle/CMakeLists.txt
+++ kstyle/CMakeLists.txt
@@ -69,13 +69,13 @@ set(oxygen_PART_SRCS
 kconfig_add_kcfg_files(oxygen_PART_SRCS oxygenstyleconfigdata.kcfgc)
 add_library(oxygen MODULE ${oxygen_PART_SRCS})
 
-target_link_libraries(oxygen Qt::Core Qt::Gui Qt::Widgets Qt::DBus Qt::Quick)
+target_link_libraries(oxygen Qt5::Core Qt5::Gui Qt5::Widgets Qt5::DBus Qt5::Quick)
 target_link_libraries(oxygen KF5::GuiAddons KF5::Style KF5::WindowSystem)
 target_link_libraries(oxygen oxygenstyle5)
 
 if(OXYGEN_HAVE_X11)
     target_link_libraries(oxygen XCB::XCB)
-    target_link_libraries(oxygen Qt::X11Extras)
+    target_link_libraries(oxygen Qt5::X11Extras)
 endif()
 
 if(OXYGEN_HAVE_KWAYLAND)
diff --git kstyle/config/CMakeLists.txt kstyle/config/CMakeLists.txt
index 2e788eb3..8618a3c1 100644
--- kstyle/config/CMakeLists.txt
+++ kstyle/config/CMakeLists.txt
@@ -27,7 +27,7 @@ add_library(kstyle_oxygen_config MODULE
     ${kstyle_oxygen_config_PART_FORMS_HEADERS}
 )
 
-target_link_libraries(kstyle_oxygen_config Qt::Core Qt::Gui Qt::Widgets Qt::DBus)
+target_link_libraries(kstyle_oxygen_config Qt5::Core Qt5::Gui Qt5::Widgets Qt5::DBus)
 target_link_libraries(kstyle_oxygen_config KF5::Completion KF5::I18n)
 target_link_libraries(kstyle_oxygen_config oxygenstyleconfig5)
 
@@ -45,7 +45,7 @@ set_package_properties(KF5KCMUtils PROPERTIES
 if( KF5KCMUtils_FOUND )
     set(oxygen_settings_SOURCES main.cpp)
     add_executable(oxygen-settings5 ${oxygen_settings_SOURCES} )
-    target_link_libraries(oxygen-settings5 Qt::Core Qt::Gui Qt::Widgets )
+    target_link_libraries(oxygen-settings5 Qt5::Core Qt5::Gui Qt5::Widgets )
     target_link_libraries(oxygen-settings5 KF5::I18n KF5::KCMUtils)
 
     install(TARGETS oxygen-settings5 ${KDE_INSTALL_TARGETS_DEFAULT_ARGS})
diff --git kstyle/demo/CMakeLists.txt kstyle/demo/CMakeLists.txt
index cad32dfe..64cee42f 100644
--- kstyle/demo/CMakeLists.txt
+++ kstyle/demo/CMakeLists.txt
@@ -42,7 +42,7 @@ add_executable(oxygen-demo5
     ${oxygen_demo_FORMS_HEADERS}
 )
 
-target_link_libraries(oxygen-demo5 Qt::Core Qt::Gui Qt::Widgets)
+target_link_libraries(oxygen-demo5 Qt5::Core Qt5::Gui Qt5::Widgets)
 target_link_libraries(oxygen-demo5 KF5::Completion KF5::ConfigGui KF5::WidgetsAddons KF5::I18n)
 if(KF5ConfigWidgets_FOUND)
     target_link_libraries(oxygen-demo5 KF5::ConfigWidgets)
diff --git liboxygen/CMakeLists.txt liboxygen/CMakeLists.txt
index 5534b73b..6105cecb 100644
--- liboxygen/CMakeLists.txt
+++ liboxygen/CMakeLists.txt
@@ -24,9 +24,9 @@ generate_export_header(oxygenstyle5 BASE_NAME oxygen EXPORT_FILE_NAME oxygen_exp
 #     interface, since other Oxygen targets link to this library
 target_link_libraries(oxygenstyle5
     PUBLIC
-        Qt::Core
-        Qt::Gui
-        Qt::Widgets
+        Qt5::Core
+        Qt5::Gui
+        Qt5::Widgets
         KF5::ConfigCore
         KF5::ConfigWidgets
         KF5::WindowSystem
@@ -37,7 +37,7 @@ target_link_libraries(oxygenstyle5
 target_include_directories(oxygenstyle5 PUBLIC "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>")
 
 if(OXYGEN_HAVE_X11)
-    target_link_libraries(oxygenstyle5 PRIVATE XCB::XCB Qt::X11Extras)
+    target_link_libraries(oxygenstyle5 PRIVATE XCB::XCB Qt5::X11Extras)
 endif()
 
 set_target_properties(oxygenstyle5 PROPERTIES VERSION ${PROJECT_VERSION} SOVERSION ${PROJECT_VERSION_MAJOR})
@@ -81,9 +81,9 @@ generate_export_header(oxygenstyleconfig5 BASE_NAME oxygen_config EXPORT_FILE_NA
 target_link_libraries(oxygenstyleconfig5
     PUBLIC
         oxygenstyle5
-        Qt::Core
-        Qt::Gui
-        Qt::Widgets
+        Qt5::Core
+        Qt5::Gui
+        Qt5::Widgets
     PRIVATE
         KF5::I18n
 )
diff --git kstyle/demo/oxygenschemechooser.cpp kstyle/demo/oxygenschemechooser.cpp
index f7ccc0e49b300534dcc8e1a59ee6e6b55f74ac7c..fe422f45883104d1fb54b61eec0e577062d6e1d8 100644
--- kstyle/demo/oxygenschemechooser.cpp
+++ kstyle/demo/oxygenschemechooser.cpp
@@ -41,7 +41,16 @@ ColorSchemeChooser::ColorSchemeChooser( QWidget* parent )
 {
     auto manager = new KColorSchemeManager( parent );
 
-    auto selectionMenu = manager->createSchemeSelectionMenu(this);
+    QString scheme;
+    if (menu()) {
+        const auto action = menu()->activeAction();
+        scheme = KLocalizedString::removeAcceleratorMarker( action->text() );
+    } else {
+        KSharedConfigPtr config = KSharedConfig::openConfig( QLatin1String( "kdeglobals" ) );
+        KConfigGroup group( config, "General" );
+        scheme = group.readEntry( "ColorScheme", QStringLiteral( "Breeze" ) );
+    }
+    auto selectionMenu = manager->createSchemeSelectionMenu(scheme, this);
 
     setMenu( selectionMenu->menu() );
 
