diff --git CMakeLists.txt CMakeLists.txt
index c4c4a69c..a7a8d4db 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -8,8 +8,8 @@ include(WriteBasicConfigVersionFile)
 include(FeatureSummary)
 
 ################# Qt/KDE #################
-set(QT_MIN_VERSION "5.15.0")
-set(KF5_MIN_VERSION "5.74")
+set(QT_MIN_VERSION "5.9.8")
+set(KF5_MIN_VERSION "5.60")
 
 
 find_package(ECM ${KF5_MIN_VERSION} REQUIRED NO_MODULE)
@@ -23,7 +23,9 @@ find_package(KF5 ${KF5_MIN_VERSION} REQUIRED COMPONENTS
     WidgetsAddons
     Service
     Completion
-    FrameworkIntegration
+    FrameworkIntegration)
+# we do need a more recent KWindowSystem because of the KWindowShadow class
+find_package(KF5 5.74.0 REQUIRED COMPONENTS
     WindowSystem)
 
 find_package(XCB COMPONENTS XCB)
diff --git kstyle/demo/oxygenschemechooser.cpp kstyle/demo/oxygenschemechooser.cpp
index acdaec2d..75fdd206 100644
--- kstyle/demo/oxygenschemechooser.cpp
+++ kstyle/demo/oxygenschemechooser.cpp
@@ -42,12 +42,48 @@ ColorSchemeChooser::ColorSchemeChooser( QWidget* parent )
 {
     auto manager = new KColorSchemeManager( parent );
 
+#if KIO_VERSION >= QT_VERSION_CHECK(5, 67, 0)
     auto selectionMenu = manager->createSchemeSelectionMenu(this);
+#else
+    const auto scheme( currentSchemeName() );
 
-    setMenu( selectionMenu->menu() );
+    auto selectionMenu = manager->createSchemeSelectionMenu( scheme, this );
+#endif
+
+    manager->activateScheme(manager->indexForScheme(scheme));
 
+    setMenu( selectionMenu->menu() );
+#if KIO_VERSION < QT_VERSION_CHECK(5, 67, 0)
+    menu()->setIcon( QIcon::fromTheme( QStringLiteral( "preferences-desktop-color" ) ) );
+    menu()->setTitle( i18n( "&Color Theme" ) );
+#endif
     setIcon( menu()->icon() );
     setText( menu()->title() );
 }
 
+#if KIO_VERSION < QT_VERSION_CHECK(5, 67, 0)
+QString ColorSchemeChooser::currentDesktopDefaultScheme() const
+{
+    KSharedConfigPtr config = KSharedConfig::openConfig( QLatin1String( "kdeglobals" ) );
+    KConfigGroup group( config, "General" );
+    return group.readEntry( "ColorScheme", QStringLiteral( "Breeze" ) );
+}
+
+QString ColorSchemeChooser::currentSchemeName() const
+{
+    if ( !menu() )
+    {
+        return currentDesktopDefaultScheme();
+    }
+
+    QAction* const action = menu()->activeAction();
+
+    if ( action )
+    {
+        return KLocalizedString::removeAcceleratorMarker( action->text() );
+    }
+    return currentDesktopDefaultScheme();
+}
+#endif
+
 }  // namespace Oxygen
diff --git kstyle/demo/oxygenschemechooser.h kstyle/demo/oxygenschemechooser.h
index 9d9ab32e..3e1f5489 100644
--- kstyle/demo/oxygenschemechooser.h
+++ kstyle/demo/oxygenschemechooser.h
@@ -39,6 +39,13 @@ class ColorSchemeChooser : public QPushButton
     Q_OBJECT
     public:
         explicit ColorSchemeChooser( QWidget* parent );
+
+#if KIO_VERSION < QT_VERSION_CHECK(5, 67, 0)
+        QString currentSchemeName() const;
+
+    private:
+        QString currentDesktopDefaultScheme() const;
+#endif
 };
 
 }  // namespace Oxygen
