diff --git kstyle/config/CMakeLists.txt kstyle/config/CMakeLists.txt
index a619e9f5285d09b9ef097d6a5e24614cce27b4c8..6993003d42ccbb0400a42fa40daef3f66d5672d2 100644
--- kstyle/config/CMakeLists.txt
+++ kstyle/config/CMakeLists.txt
@@ -35,6 +35,8 @@ install(TARGETS kstyle_oxygen_config DESTINATION ${QT_PLUGIN_INSTALL_DIR})
 install(FILES oxygenstyleconfig.desktop DESTINATION  ${SERVICES_INSTALL_DIR})
 
 ########### next target ###############
+include(ECMAddAppIcon)
+
 find_package(KF5KCMUtils CONFIG)
 set_package_properties(KF5KCMUtils PROPERTIES
     DESCRIPTION "KF5 KCM utils Framework"
@@ -44,6 +46,8 @@ set_package_properties(KF5KCMUtils PROPERTIES
 
 if( KF5KCMUtils_FOUND )
     set(oxygen_settings_SOURCES main.cpp)
+    file(GLOB ICONS_SRCS "${CMAKE_CURRENT_BINARY_DIR}/icons/*oxygen.png")
+    ecm_add_app_icon(oxygen_settings_SOURCES ICONS ${ICONS_SRCS})
     add_executable(oxygen-settings5 ${oxygen_settings_SOURCES} )
     target_link_libraries(oxygen-settings5 Qt5::Core Qt5::Gui Qt5::Widgets )
     target_link_libraries(oxygen-settings5 KF5::I18n KF5::KCMUtils)
