--- kstyle/config/orig.main.cpp	2015-12-15 11:28:21.000000000 +0100
+++ kstyle/config/main.cpp	2015-12-21 22:52:12.000000000 +0100
@@ -38,7 +38,7 @@
 
     QApplication app( argc, argv );
     app.setApplicationName( i18n( "Oxygen Settings" ) );
-    app.setWindowIcon( QIcon::fromTheme( QStringLiteral( "oxygen-settings" ) ) );
+    app.setWindowIcon( QIcon::fromTheme( QStringLiteral( "oxygen-settings" ), app.windowIcon() ) );
 
     KCMultiDialog dialog;
     dialog.setWindowTitle( i18n( "Oxygen Settings" ) );
--- kstyle/demo/orig.main.cpp	2015-12-15 11:28:21.000000000 +0100
+++ kstyle/demo/main.cpp	2015-12-21 22:52:20.000000000 +0100
@@ -56,7 +56,7 @@
         #endif
 
         app.setApplicationName( i18n( "Oxygen Demo" ) );
-        app.setWindowIcon( QIcon::fromTheme( QStringLiteral( "oxygen" ) ) );
+        app.setWindowIcon( QIcon::fromTheme( QStringLiteral( "oxygen" ), app.windowIcon() ) );
         DemoDialog dialog;
         dialog.show();
         bool result = app.exec();
diff --git kstyle/config/CMakeLists.txt kstyle/config/CMakeLists.txt
index 2e788eb3c5aa49342369d01d3be2fa60d54df08e..c486e49fa72b04df10cd9c2cfb779b558c40796a 100644
--- kstyle/config/CMakeLists.txt
+++ kstyle/config/CMakeLists.txt
@@ -35,6 +35,7 @@ install(TARGETS kstyle_oxygen_config DESTINATION ${KDE_INSTALL_QTPLUGINDIR})
 install(FILES oxygenstyleconfig.desktop DESTINATION  ${KDE_INSTALL_KSERVICES5DIR})
 
 ########### next target ###############
+include(ECMAddAppIcon)
 find_package(KF5KCMUtils CONFIG)
 set_package_properties(KF5KCMUtils PROPERTIES
     DESCRIPTION "KF5 KCM utils Framework"
@@ -44,6 +45,8 @@ set_package_properties(KF5KCMUtils PROPERTIES
 
 if( KF5KCMUtils_FOUND )
     set(oxygen_settings_SOURCES main.cpp)
+    file(GLOB ICONS_SRCS "${CMAKE_CURRENT_BINARY_DIR}/icons/*oxygen.png")
+    ecm_add_app_icon(oxygen_settings_SOURCES ICONS ${ICONS_SRCS})
     add_executable(oxygen-settings5 ${oxygen_settings_SOURCES} )
     target_link_libraries(oxygen-settings5 Qt::Core Qt::Gui Qt::Widgets )
     target_link_libraries(oxygen-settings5 KF5::I18n KF5::KCMUtils)
