diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4449bc2df3b399d4fa68563b7706611fd6e5f6ab..b12a0e4d124571ba34fd245de31ebd0460e58bda 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -21,6 +21,12 @@ option(BUILD_WITH_QT4 "Build qtkeychain with Qt4 no matter if Qt5 was found" OFF
 option(BUILD_TEST_APPLICATION "Build test application" ON)
 option(BUILD_TRANSLATIONS "Build translations" ON)
 option(QTKEYCHAIN_STATIC "Build static library" OFF)
+if(UNIX AND NOT APPLE AND NOT ANDROID)
+    option(USE_LIBSECRET "Build with GNOME libsecret support" ON)
+else()
+    # on Mac this will use the KF5 Wallet backend.
+    option(USE_LIBSECRET "Build with GNOME libsecret support" OFF)
+endif()
 
 if(CMAKE_SYSTEM_NAME STREQUAL Android)
     set(ANDROID 1)
@@ -39,10 +45,13 @@ if( NOT BUILD_WITH_QT4 )
     find_package(Qt5Core QUIET)
 endif()
 
+if(USE_LIBSECRET)
+    add_definitions(-DUSE_LIBSECRET)
+endif()
 if (Qt5Core_FOUND AND NOT BUILD_WITH_QT4)
   set(QTKEYCHAIN_VERSION_INFIX 5)
 
-  if(UNIX AND NOT APPLE AND NOT ANDROID)
+  if(USE_LIBSECRET)
     find_package(Qt5DBus REQUIRED)
     include_directories(${Qt5DBus_INCLUDE_DIRS})
     set(QTDBUS_LIBRARIES ${Qt5DBus_LIBRARIES})
@@ -87,7 +96,7 @@ if (Qt5Core_FOUND AND NOT BUILD_WITH_QT4)
   endif()
 else()
   set(QTKEYCHAIN_VERSION_INFIX "")
-  if(UNIX AND NOT APPLE)
+  if(USE_LIBSECRET)
     find_package(Qt4 COMPONENTS QtCore QtDBus REQUIRED)
     set(QTDBUS_LIBRARIES ${QT_QTDBUS_LIBRARY})
     macro(qt_add_dbus_interface)
@@ -121,7 +130,7 @@ set(qtkeychain_SOURCES
 
 add_definitions( -Wall )
 
-if(WIN32)
+if(WIN32 AND NOT USE_LIBSECRET)
     list(APPEND qtkeychain_SOURCES keychain_win.cpp)
     if (NOT USE_CREDENTIAL_STORE)
         list(APPEND qtkeychain_LIBRARIES crypt32)
@@ -133,7 +142,7 @@ if(WIN32)
     endif()
 endif()
 
-if(APPLE)
+if(APPLE AND NOT USE_LIBSECRET)
     if(IOS)
         list(APPEND qtkeychain_SOURCES keychain_ios.cpp)
     else()
@@ -147,7 +156,7 @@ if(APPLE)
     list(APPEND qtkeychain_LIBRARIES ${SECURITY_LIBRARY})
 endif()
 
-if(UNIX AND NOT APPLE AND NOT ANDROID)
+if(USE_LIBSECRET)
     option(LIBSECRET_SUPPORT "Build with libsecret support" ON)
 
     if(LIBSECRET_SUPPORT)
@@ -248,7 +257,7 @@ ecm_setup_version("${QTKEYCHAIN_VERSION}" VARIABLE_PREFIX SNORE
   PACKAGE_VERSION_FILE "${CMAKE_CURRENT_BINARY_DIR}/Qt${QTKEYCHAIN_VERSION_INFIX}KeychainConfigVersion.cmake"
   SOVERSION ${QTKEYCHAIN_VERSION})
 
-if(UNIX AND NOT APPLE AND NOT ANDROID)
+if(USE_LIBSECRET)
     set(PRI_EXTRA_DEPS "dbus")
 endif()
 ecm_generate_pri_file(BASE_NAME Qt${QTKEYCHAIN_VERSION_INFIX}Keychain
diff --git a/keychain_p.h b/keychain_p.h
index ab7f0be5cf1e7a7f46cc8da97af0b5de0bb3c923..39da324025a493d018d1bacefc0ae9a41238e415 100644
--- a/keychain_p.h
+++ b/keychain_p.h
@@ -15,7 +15,7 @@
 #include <QSettings>
 #include <QQueue>
 
-#if defined(Q_OS_UNIX) && !defined(Q_OS_DARWIN) && !defined(Q_OS_ANDROID)
+#if defined(USE_LIBSECRET)
 
 #include <QDBusPendingCallWatcher>
 
@@ -49,7 +49,7 @@ public:
     Mode mode;
     QByteArray data;
 
-#if defined(Q_OS_UNIX) && !defined(Q_OS_DARWIN) && !defined(Q_OS_ANDROID)
+#if defined(USE_LIBSECRET)
     org::kde::KWallet* iface;
     int walletHandle;
 
@@ -93,7 +93,7 @@ public:
     explicit ReadPasswordJobPrivate( const QString &service_, ReadPasswordJob* qq );
     void scheduledStart();
 
-#if defined(Q_OS_UNIX) && !defined(Q_OS_DARWIN) && !defined(Q_OS_ANDROID)
+#if defined(USE_LIBSECRET)
     void fallbackOnError(const QDBusError& err);
 
 private Q_SLOTS:
@@ -116,7 +116,7 @@ public:
     explicit WritePasswordJobPrivate( const QString &service_, WritePasswordJob* qq );
     void scheduledStart();
 
-#if defined(Q_OS_UNIX) && !defined(Q_OS_DARWIN) && !defined(Q_OS_ANDROID)
+#if defined(USE_LIBSECRET)
     void fallbackOnError(const QDBusError& err);
 #endif
 
@@ -130,7 +130,7 @@ public:
 
     void scheduledStart();
 
-#if defined(Q_OS_UNIX) && !defined(Q_OS_DARWIN) && !defined(Q_OS_ANDROID)
+#if defined(USE_LIBSECRET)
     void fallbackOnError(const QDBusError& err);
 #endif
 
diff --git a/keychain_unix.cpp b/keychain_unix.cpp
index 30b26c3fd18da4d0d9f448be7a5333743a361d52..a8eac2cd424013b06c7bfef4fc22cbf22f97168a 100644
--- a/keychain_unix.cpp
+++ b/keychain_unix.cpp
@@ -47,6 +47,11 @@ static DesktopEnvironment getKdeVersion() {
 }
 
 static DesktopEnvironment detectDesktopEnvironment() {
+#if defined(Q_OS_MACOS) && defined(USE_LIBSECRET)
+    // we've been built to run in/with KF5 frameworks installed;
+    // pretend the DE is Plasma5
+    return DesktopEnv_Plasma5;
+#endif
     QByteArray xdgCurrentDesktop = qgetenv("XDG_CURRENT_DESKTOP");
     if ( xdgCurrentDesktop == "GNOME" ) {
         return DesktopEnv_Gnome;
