diff --git a/waypipe-c/handlers.c b/waypipe-c/handlers.c
index 9f1749d..4e40f12 100644
--- a/waypipe-c/handlers.c
+++ b/waypipe-c/handlers.c
@@ -353,11 +353,14 @@ struct wp_object *create_wp_object(uint32_t id, const struct wp_interface *type)
 	new_obj->is_zombie = false;
 
 	if (type == &intf_zwp_linux_buffer_params_v1) {
+#ifndef __APPLE__
 		struct obj_zwp_linux_dmabuf_params *params =
 				(struct obj_zwp_linux_dmabuf_params *)new_obj;
+		// no dmabuf (requires libgbm) and GCC raises an array-bounds error here
 		for (int i = 0; i < MAX_DMABUF_PLANES; i++) {
 			params->add[i].fd = -1;
 		}
+#endif
 	} else if (type == &intf_wl_surface) {
 		((struct obj_wl_surface *)new_obj)->scale = 1;
 	}
