diff --git a/waypipe-c/test/diff_roundtrip.c b/waypipe-c/test/diff_roundtrip.c
index fd4bbc8..ebf810f 100644
--- a/waypipe-c/test/diff_roundtrip.c
+++ b/waypipe-c/test/diff_roundtrip.c
@@ -187,6 +187,16 @@ static bool run_subtest(int i, const struct subtest test, char *diff,
 	return all_success;
 }
 
+static void *aligned_alloc(size_t alignment, size_t size)
+{
+	void *memptr = NULL;
+	if (posix_memalign(&memptr, alignment, size) == 0 ){
+		return memptr;
+	} else {
+		return NULL;
+	}
+}
+
 log_handler_func_t log_funcs[2] = {test_log_handler, test_log_handler};
 int main(int argc, char **argv)
 {
