diff --git a/waypipe-c/mainloop.c b/waypipe-c/mainloop.c
index 454e37c..a711cad 100644
--- a/waypipe-c/mainloop.c
+++ b/waypipe-c/mainloop.c
@@ -36,6 +36,9 @@
 #include <sys/uio.h>
 #include <unistd.h>
 
+#undef __DARWIN_ALIGN32
+#define __DARWIN_ALIGN32(len) (((len) + sizeof (uint32_t) - 1) & (size_t) ~(sizeof (uint32_t) - 1))
+
 // The maximum number of fds libwayland can recvmsg at once
 #define MAX_LIBWAY_FDS 28
 static ssize_t iovec_read(
