diff --git a/waypipe-c/mainloop.c b/waypipe-c/mainloop.c
index 454e37c..27f7a95 100644
--- a/waypipe-c/mainloop.c
+++ b/waypipe-c/mainloop.c
@@ -41,7 +41,8 @@
 static ssize_t iovec_read(
 		int conn, char *buf, size_t buflen, struct int_window *fds)
 {
-	char cmsgdata[(CMSG_LEN(MAX_LIBWAY_FDS * sizeof(int32_t)))] = {0};
+	char cmsgdata[(CMSG_LEN(MAX_LIBWAY_FDS * sizeof(int32_t)))];
+	memset(cmsgdata, 0, sizeof(cmsgdata));
 	struct iovec the_iovec;
 	the_iovec.iov_len = buflen;
 	the_iovec.iov_base = buf;
