diff --git a/meson.build b/meson.build
index b934d0b..b11a826 100644
--- a/meson.build
+++ b/meson.build
@@ -26,7 +26,7 @@ project(
   ).stdout(),
   license : 'MIT',
   meson_version : '>= 0.45',
-  default_options : ['buildtype=debugoptimized', 'b_ndebug=if-release', 'c_std=c99', 'cpp_std=c++11']
+  default_options : ['buildtype=debugoptimized', 'b_ndebug=if-release', 'c_std11', 'cpp_std=c++11']
 )
 
 cc = meson.get_compiler('c')
@@ -122,7 +122,8 @@ with_dri_r200 = _drivers.contains('r200')
 with_dri_nouveau = _drivers.contains('nouveau')
 with_dri_swrast = _drivers.contains('swrast')
 
-with_dri = _drivers.length() != 0 and _drivers != ['']
+#with_dri = _drivers.length() != 0 and _drivers != ['']
+with_dri = true
 
 _drivers = get_option('gallium-drivers')
 if _drivers.contains('auto')
@@ -326,7 +327,7 @@ elif _egl == 'true'
     error('No platforms specified, consider -Dplatforms=drm,x11,surfaceless at least')
   elif not ['disabled', 'dri'].contains(with_glx)
     error('EGL requires dri, but a GLX is being built without dri')
-  elif ['darwin', 'windows'].contains(host_machine.system())
+  elif ['windows'].contains(host_machine.system())
     error('EGL is not available on Windows or MacOS')
   endif
   with_egl = true
@@ -381,7 +382,8 @@ if with_vulkan_icd_dir == ''
   with_vulkan_icd_dir = join_paths(get_option('datadir'), 'vulkan/icd.d')
 endif
 
-with_dri2 = (with_dri or with_any_vk) and with_dri_platform == 'drm'
+#with_dri2 = (with_dri or with_any_vk) and with_dri_platform == 'drm'
+with_dri2 = true
 _dri3 = get_option('dri3')
 if _dri3 == 'auto'
   with_dri3 = system_has_kms_drm and with_dri2
@@ -728,16 +730,16 @@ if with_platform_haiku
 endif
 
 prog_python = import('python3').find_python()
-has_mako = run_command(
-  prog_python, '-c',
-  '''
-from distutils.version import StrictVersion
-import mako
-assert StrictVersion(mako.__version__) > StrictVersion("0.8.0")
-  ''')
-if has_mako.returncode() != 0
-  error('Python (3.x) mako module >= 0.8.0 required to build mesa.')
-endif
+# has_mako = run_command(
+#  prog_python, '-c',
+#  '''
+# from distutils.version import StrictVersion
+# import mako
+# assert StrictVersion(mako.__version__) > StrictVersion("0.8.0")
+#   ''')
+# if has_mako.returncode() != 0
+#   error('Python (3.x) mako module >= 0.8.0 required to build mesa.')
+# endif
 
 if cc.get_id() == 'gcc' and cc.version().version_compare('< 4.4.6')
   error('When using GCC, version 4.4.6 or later is required.')
@@ -1085,11 +1087,11 @@ elif with_dri_i965 and get_option('shader-cache')
 endif
 
 # Determine whether or not the rt library is needed for time functions
-if cc.has_function('clock_gettime')
+# if cc.has_function('clock_gettime')
   dep_clock = null_dep
-else
-  dep_clock = cc.find_library('rt')
-endif
+# else
+#   dep_clock = cc.find_library('rt')
+# endif
 
 # TODO: some of these may be conditional
 dep_zlib = dependency('zlib', version : '>= 1.2.3')
@@ -1169,7 +1171,8 @@ endforeach
 with_gallium_drisw_kms = false
 dep_libdrm = dependency(
   'libdrm', version : '>=' + _drm_ver,
-  required : with_dri2 or with_dri3
+  #required : with_dri2 or with_dri3
+  required : with_dri3
 )
 if dep_libdrm.found()
   pre_args += '-DHAVE_LIBDRM'
diff --git a/src/gallium/state_trackers/dri/Makefile.sources b/src/gallium/state_trackers/dri/Makefile.sources
index a610293..5ed1120 100644
--- a/src/gallium/state_trackers/dri/Makefile.sources
+++ b/src/gallium/state_trackers/dri/Makefile.sources
@@ -10,8 +10,8 @@ common_SOURCES := \
 	dri_screen.c \
 	dri_screen.h
 
-dri2_SOURCES := \
-	dri2.c
+#dri2_SOURCES := \
+#	dri2.c
 
 drisw_SOURCES := \
 	drisw.c
diff --git a/src/gallium/state_trackers/dri/meson.build b/src/gallium/state_trackers/dri/meson.build
index 4bb4115..adf28f5 100644
--- a/src/gallium/state_trackers/dri/meson.build
+++ b/src/gallium/state_trackers/dri/meson.build
@@ -37,9 +37,9 @@ if with_dri
   files_libdri += files('drisw.c')
 endif
 
-if with_dri2
-  files_libdri += files('dri2.c')
-endif
+#if with_dri2
+#  files_libdri += files('dri2.c')
+#endif
 
 libdri_c_args = []
 if with_gallium_softpipe
diff --git a/src/mesa/drivers/dri/swrast/meson.build b/src/mesa/drivers/dri/swrast/meson.build
index 5ceb1e3..5f8a59a 100644
--- a/src/mesa/drivers/dri/swrast/meson.build
+++ b/src/mesa/drivers/dri/swrast/meson.build
@@ -23,7 +23,7 @@ libswrast_dri = static_library(
   files('swrast.c', 'swrast_priv.h'),
   c_args : [c_vis_args],
   include_directories : [inc_common, inc_dri_common],
-  dependencies : dep_libdrm,
+  #dependencies : dep_libdrm,
 )
 
 dri_drivers += libswrast_dri
