diff --git a/meson.build b/meson.build
index b934d0bfc762a6fcb8d78c8405cc53ccc2193f68..28e99c19fe8ec8b34f2a7bc3dc5c62178cabcfca 100644
--- a/meson.build
+++ b/meson.build
@@ -326,7 +326,7 @@ elif _egl == 'true'
     error('No platforms specified, consider -Dplatforms=drm,x11,surfaceless at least')
   elif not ['disabled', 'dri'].contains(with_glx)
     error('EGL requires dri, but a GLX is being built without dri')
-  elif ['darwin', 'windows'].contains(host_machine.system())
+  elif ['windows'].contains(host_machine.system())
     error('EGL is not available on Windows or MacOS')
   endif
   with_egl = true
@@ -728,16 +728,16 @@ if with_platform_haiku
 endif
 
 prog_python = import('python3').find_python()
-has_mako = run_command(
-  prog_python, '-c',
-  '''
-from distutils.version import StrictVersion
-import mako
-assert StrictVersion(mako.__version__) > StrictVersion("0.8.0")
-  ''')
-if has_mako.returncode() != 0
-  error('Python (3.x) mako module >= 0.8.0 required to build mesa.')
-endif
+# has_mako = run_command(
+#  prog_python, '-c',
+#  '''
+# from distutils.version import StrictVersion
+# import mako
+# assert StrictVersion(mako.__version__) > StrictVersion("0.8.0")
+#   ''')
+# if has_mako.returncode() != 0
+#   error('Python (3.x) mako module >= 0.8.0 required to build mesa.')
+# endif
 
 if cc.get_id() == 'gcc' and cc.version().version_compare('< 4.4.6')
   error('When using GCC, version 4.4.6 or later is required.')
@@ -1085,11 +1085,11 @@ elif with_dri_i965 and get_option('shader-cache')
 endif
 
 # Determine whether or not the rt library is needed for time functions
-if cc.has_function('clock_gettime')
+# if cc.has_function('clock_gettime')
   dep_clock = null_dep
-else
-  dep_clock = cc.find_library('rt')
-endif
+# else
+#   dep_clock = cc.find_library('rt')
+# endif
 
 # TODO: some of these may be conditional
 dep_zlib = dependency('zlib', version : '>= 1.2.3')
