diff --git a/include/GL/mesa_glinterop.h b/include/GL/mesa_glinterop.h
index 45fda93ca2495608a8d0340643755604f1042bc1..140a8baed4da194efe03ebced754758b4eafaf5a 100644
--- a/include/GL/mesa_glinterop.h
+++ b/include/GL/mesa_glinterop.h
@@ -58,7 +58,7 @@ extern "C" {
 #endif
 
 /* Forward declarations to avoid inclusion of GL/glx.h */
-#ifndef GLX_H
+#if !defined(GLX_H) && !defined(APPLE_GLX_CONTEXT_H)
 struct _XDisplay;
 struct __GLXcontextRec;
 #endif
