diff --git a/meson.build b/meson.build
index b934d0b..1c2069c 100644
--- a/meson.build
+++ b/meson.build
@@ -26,7 +26,7 @@ project(
   ).stdout(),
   license : 'MIT',
   meson_version : '>= 0.45',
-  default_options : ['buildtype=debugoptimized', 'b_ndebug=if-release', 'c_std=c99', 'cpp_std=c++11']
+  default_options : ['buildtype=debugoptimized', 'b_ndebug=if-release', 'c_std11', 'cpp_std=c++11']
 )
 
 cc = meson.get_compiler('c')