diff --git a/meson.build b/meson.build
index b934d0b..cbe2fcb 100644
--- a/meson.build
+++ b/meson.build
@@ -343,7 +343,7 @@ if with_egl and not (with_platform_drm or with_platform_surfaceless)
   endif
 endif
 
-pre_args += '-DGLX_USE_TLS'
+# pre_args += '-DGLX_USE_TLS'
 if with_glx != 'disabled'
   if not (with_platform_x11 and with_any_opengl)
     error('Cannot build GLX support without X11 platform support and at least one OpenGL API')
diff --git a/src/meson.build b/src/meson.build
index 3b91c6a..bb10f31 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -111,6 +111,6 @@ if with_glx != 'disabled'
     libraries : _gl,
     libraries_private : gl_priv_libs,
     requires_private : gl_priv_reqs,
-    variables : ['glx_tls=yes'],
+    variables : ['glx_tls=no'],
   )
 endif
