diff --git a/include/c11/threads_posix.h b/include/c11/threads_posix.h
index 45cb6075e6e08a083bd1faba07d87aff206ea58b..fd1c0150d1c791937b81ef2c4cbb91081b84d638 100644
--- a/include/c11/threads_posix.h
+++ b/include/c11/threads_posix.h
@@ -48,6 +48,12 @@ Configuration macro:
 #endif
 
 
+#ifdef __MACH__
+#include <mach/mach.h>
+#include <mach/mach_time.h>
+#include <sys/sysctl.h>
+#endif
+
 #include <pthread.h>
 
 /*---------------------------- macros ----------------------------*/
@@ -388,7 +394,19 @@ timespec_get(struct timespec *ts, int base)
 {
     if (!ts) return 0;
     if (base == TIME_UTC) {
+#ifndef __MACH__
         clock_gettime(CLOCK_REALTIME, ts);
+#else
+        static int calibrated = 0;
+        static mach_timebase_info_data_t sTimebaseInfo;
+        if (!calibrated) {
+            mach_timebase_info(&sTimebaseInfo);
+            calibrated = 1;
+        }
+        unsigned long long now = mach_absolute_time() * sTimebaseInfo.numer / sTimebaseInfo.denom;
+        ts->tv_sec = now / 1000000000;
+        ts->tv_nsec = now % 1000000000;
+#endif
         return base;
     }
     return 0;
diff --git a/src/mesa/main/macros.h b/src/mesa/main/macros.h
index 03a228b4474b3ef35aa72e804ca378f68f32da3c..220d5fa71ed031293d72e811f4999b63a00b003d 100644
--- a/src/mesa/main/macros.h
+++ b/src/mesa/main/macros.h
@@ -677,6 +677,9 @@ minify(unsigned value, unsigned levels)
  *
  * \sa ROUND_DOWN_TO()
  */
+#ifdef ALIGN
+#undef ALIGN
+#endif
 static inline uintptr_t
 ALIGN(uintptr_t value, int32_t alignment)
 {
diff --git a/src/compiler/nir/nir.h b/src/compiler/nir/nir.h
index a0ae9a4430e86f958fdf92ceebe76539e0512856..367f4280091af80d4895551a5789472d2292e76c 100644
--- a/src/compiler/nir/nir.h
+++ b/src/compiler/nir/nir.h
@@ -31,7 +31,6 @@
 #include "util/hash_table.h"
 #include "compiler/glsl/list.h"
 #include "GL/gl.h" /* GLenum */
-#include "util/list.h"
 #include "util/ralloc.h"
 #include "util/set.h"
 #include "util/bitset.h"
@@ -47,6 +46,10 @@
 
 #include "nir_opcodes.h"
 
+// LIST_ENTRY set by sys/queue.h on Mac
+#undef LIST_ENTRY
+#include "util/list.h"
+
 #if defined(_WIN32) && !defined(snprintf)
 #define snprintf _snprintf
 #endif
