--- src/orig.CMakeLists.txt	2019-02-15 17:17:28.000000000 +0100
+++ src/CMakeLists.txt	2019-03-12 15:07:59.000000000 +0100
@@ -60,7 +60,9 @@
                                       COMPILE_FLAGS "-DBUILD_QTUI"
                                       RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR})
   target_link_libraries(quasselclient mod_qtui mod_uisupport mod_client mod_common ${COMMON_LIBRARIES} ${CLIENT_LIBRARIES} ${QUASSEL_SSL_LIBRARIES} ${QT_QTMAIN_LIBRARY})
-  install(TARGETS quasselclient RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
+  install(TARGETS quasselclient
+  	RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+	BUNDLE DESTINATION ${BUNDLE_INSTALL_DIR})
 endif()
 
 if(WANT_MONO)
@@ -71,7 +73,9 @@
                                 COMPILE_FLAGS "-DBUILD_MONO"
                                 RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR})
   target_link_libraries(quassel mod_qtui mod_uisupport mod_client mod_core mod_common ${COMMON_LIBRARIES} ${CLIENT_LIBRARIES} ${QUASSEL_SSL_LIBRARIES} ${QT_QTMAIN_LIBRARY})
-  install(TARGETS quassel RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
+  install(TARGETS quassel
+  	RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+	BUNDLE DESTINATION ${BUNDLE_INSTALL_DIR})
 endif()
 
 # Build bundles for MacOSX
