--- src/common/orig.quassel.cpp	2019-03-12 17:33:58.000000000 +0100
+++ src/common/quassel.cpp	2019-03-12 17:48:53.000000000 +0100
@@ -450,7 +450,7 @@
 #elif defined Q_OS_MAC
     dataDirNames << QDir::homePath() + "/Library/Application Support/Quassel/"
                  << QCoreApplication::applicationDirPath();
-#else
+#endif
     // Linux et al
 
     // XDG_DATA_HOME is the location for users to override system-installed files, usually in .local/share
@@ -463,7 +463,7 @@
     // Now whatever is configured through XDG_DATA_DIRS
     QString xdgDataDirs = QFile::decodeName(qgetenv("XDG_DATA_DIRS"));
     if (xdgDataDirs.isEmpty())
-        dataDirNames << "/usr/local/share" << "/usr/share";
+        dataDirNames << "@PREFIX@/share" << "/usr/share";
     else
         dataDirNames << xdgDataDirs.split(':', QString::SkipEmptyParts);
 
@@ -474,8 +474,6 @@
     for (int i = 0; i < dataDirNames.count(); i++)
         dataDirNames[i] = QDir::cleanPath(dataDirNames.at(i)) + "/quassel/";
 
-#endif
-
     // Add resource path and workdir just in case.
     // Workdir should have precedence
     dataDirNames.prepend(QCoreApplication::applicationDirPath() + "/data/");
--- src/uisupport/orig.icon.cpp	2019-03-12 18:35:55.000000000 +0100
+++ src/uisupport/icon.cpp	2019-03-12 18:35:59.000000000 +0100
@@ -32,7 +32,7 @@
 {
     static std::set<QString> warnedAbout;
     if (warnedAbout.insert(iconName).second) {
-        qWarning() << "Missing icon:" << iconName << qPrintable(extra);
+        qWarning() << "Missing icon:" << iconName << "in theme" << QIcon::themeName() << qPrintable(extra);
     }
 }
 
