diff --git a/src/plugins/phabricator/phabricatorjobs.cpp b/src/plugins/phabricator/phabricatorjobs.cpp
index ad3b94d72e6b7de9345d04afa723dbcb392282a6..e2afdb399d485931d7e2dacfc2a9defe4b9e8dcd 100644
--- a/src/plugins/phabricator/phabricatorjobs.cpp
+++ b/src/plugins/phabricator/phabricatorjobs.cpp
@@ -215,6 +215,8 @@ bool Phabricator::DiffRevList::buildArcCommand(const QString& workDir, const QSt
         m_arcCmd.setProgram(arc);
         m_arcCmd.setArguments(args);
         m_arcCmd.setWorkingDirectory(workDir);
+        disconnect(&m_arcCmd, static_cast<void(QProcess::*)(int, QProcess::ExitStatus)>(&QProcess::finished),
+                nullptr, nullptr);
         connect(&m_arcCmd, static_cast<void(QProcess::*)(int, QProcess::ExitStatus)>(&QProcess::finished),
                 this, &DiffRevList::done);
         setPercent(33);
