diff --git meson.build orig.meson.build
index badf091..ba3d01d 100644
--- meson.build
+++ orig.meson.build
@@ -370,7 +370,6 @@ foreach f : check_functions
 endforeach
 
 if cc.has_header_symbol('sys/syscall.h', 'SYS_memfd_create') \
-  or cc.has_header_symbol('sys/syscall.h', '__NR_memfd_create') \
   or cc.has_function('memfd_create')
   cdata.set('HAVE_MEMFD', 1)
 endif
diff --git src/pulsecore/memfd-wrappers.h src/pulsecore/orig.memfd-wrappers.h
index ccc92ba..4708751 100644
--- src/pulsecore/memfd-wrappers.h
+++ src/pulsecore/orig.memfd-wrappers.h
@@ -35,11 +35,7 @@
  */
 
 static inline int memfd_create(const char *name, unsigned int flags) {
-#ifdef SYS_memfd_create
     return syscall(SYS_memfd_create, name, flags);
-#else
-    return syscall(__NR_memfd_create, name, flags);
-#endif
 }
 
 /* memfd_create(2) flags */
