--- src/orig.default.pa	2015-01-23 02:45:48.000000000 +0100
+++ src/default.pa	2015-01-23 14:23:05.000000000 +0100
@@ -113,12 +117,12 @@
 
 ### If autoexit on idle is enabled we want to make sure we only quit
 ### when no local session needs us anymore.
-.ifexists module-console-kit.so
-load-module module-console-kit
-.endif
-.ifexists module-systemd-login.so
-load-module module-systemd-login
-.endif
+#.ifexists module-console-kit.so
+#load-module module-console-kit
+#.endif
+#.ifexists module-systemd-login.so
+#load-module module-systemd-login
+#.endif
 
 ### Enable positioned event sounds
 load-module module-position-event-sounds
