diff --git src/pulse/orig.meson.build src/pulse/meson.build
index c132630..e2c3341 100644
--- src/pulse/orig.meson.build
+++ src/pulse/meson.build
@@ -80,6 +80,16 @@ else
   versioning_link_args = []
 endif
 
+libpulse_compat_version = '@0@.0.0'.format(
+  libpulse_version_info[0],
+)
+
+libpulse_current_version = '@0@.@1@.@2@'.format(
+  libpulse_version_info[0],
+  libpulse_version_info[1],
+  libpulse_version_info[2],
+)
+
 libpulse = shared_library('pulse',
   libpulse_sources,
   libpulse_headers,
@@ -92,6 +102,7 @@ libpulse = shared_library('pulse',
   dependencies : [libm_dep, thread_dep, libpulsecommon_dep, dbus_dep, dl_dep, iconv_dep, libintl_dep, platform_dep, platform_socket_dep, libatomic_ops_dep],
   implicit_include_directories : false,
   vs_module_defs : 'libpulse.def',
+  darwin_versions : [libpulse_compat_version, libpulse_current_version],
 )
 
 libpulse_dep = declare_dependency(link_with: libpulse)
@@ -101,6 +112,16 @@ install_headers(
   subdir : 'pulse'
 )
 
+libpulse_simple_compat_version = '@0@.0.0'.format(
+  libpulse_simple_version_info[0] + 1,
+)
+
+libpulse_simple_current_version = '@0@.@1@.@2@'.format(
+  libpulse_simple_version_info[0] + 1,
+  libpulse_simple_version_info[1],
+  libpulse_simple_version_info[2],
+)
+
 libpulse_simple = shared_library('pulse-simple',
   'simple.c',
   'simple.h',
@@ -111,11 +132,21 @@ libpulse_simple = shared_library('pulse-simple',
   dependencies : [libpulse_dep, libpulsecommon_dep],
   install : true,
   install_rpath : privlibdir,
+  darwin_versions : [libpulse_simple_compat_version, libpulse_simple_current_version],
 )
 
 libpulse_simple_dep = declare_dependency(link_with: libpulse_simple)
 
 if glib_dep.found()
+  libpulse_mainloop_glib_compat_version = '@0@.0.0'.format(
+    libpulse_mainloop_glib_version_info[0] + 1,
+  )
+  libpulse_mainloop_glib_current_version = '@0@.@1@.@2@'.format(
+    libpulse_mainloop_glib_version_info[0] + 1,
+    libpulse_mainloop_glib_version_info[1],
+    libpulse_mainloop_glib_version_info[2],
+  )
+
   libpulse_mainloop_glib = shared_library('pulse-mainloop-glib',
     'glib-mainloop.c',
     'glib-mainloop.h',
@@ -126,6 +157,7 @@ if glib_dep.found()
     dependencies : [libpulse_dep, libpulsecommon_dep, glib_dep],
     install : true,
     install_rpath : privlibdir,
+    darwin_versions : [libpulse_mainloop_glib_compat_version, libpulse_mainloop_glib_current_version],
   )
 
   libpulse_mainloop_glib_dep = declare_dependency(link_with: libpulse_mainloop_glib)
