diff --git src/modules/orig.meson.build src/modules/meson.build
index 3636ce0..67535bc 100644
--- src/modules/orig.meson.build
+++ src/modules/meson.build
@@ -331,5 +331,6 @@ foreach m : all_modules
     link_args : [nodelete_link_args, no_undefined_args],
     link_with : extra_libs,
     name_prefix : '',
+    name_suffix : 'so',
     implicit_include_directories : false)
 endforeach
diff --git orig.meson.build meson.build
index ea897f1..d2427b7 100644
--- orig.meson.build
+++ meson.build
@@ -153,7 +153,7 @@ cdata.set_quoted('PA_BUILDDIR', meson.current_build_dir())
 if host_machine.system() == 'windows'
   cdata.set_quoted('PA_SOEXT', '.dll')
 elif host_machine.system() == 'darwin'
-  cdata.set_quoted('PA_SOEXT', '.dylib')
+  cdata.set_quoted('PA_SOEXT', '.so')
 else
   cdata.set_quoted('PA_SOEXT', '.so')
 endif
