--- src/daemon/default.pa.in.orig2	2015-02-03 14:09:14.000000000 +0100
+++ src/daemon/default.pa.in	2015-02-03 14:13:21.000000000 +0100
@@ -153,12 +153,12 @@
 
 ### If autoexit on idle is enabled we want to make sure we only quit
 ### when no local session needs us anymore.
-.ifexists module-console-kit@PA_SOEXT@
-load-module module-console-kit
-.endif
-.ifexists module-systemd-login@PA_SOEXT@
-load-module module-systemd-login
-.endif
+#.ifexists module-console-kit@PA_SOEXT@
+#load-module module-console-kit
+#.endif
+#.ifexists module-systemd-login@PA_SOEXT@
+#load-module module-systemd-login
+#.endif
 
 ### Enable positioned event sounds
 load-module module-position-event-sounds
