Add entry for Objective-C++

Upstream-Status: Submitted [https://bugs.freedesktop.org/show_bug.cgi?id=98823]

diff --git data/freedesktop.org.xml.in data/freedesktop.org.xml.in
index 66c44fa..183edbc 100644
--- data/freedesktop.org.xml.in
+++ data/freedesktop.org.xml.in
@@ -6122,6 +6122,15 @@ command to generate the output files.
     </magic>
     <glob pattern="*.m"/>
   </mime-type>
+  <mime-type type="text/x-objc++src">
+    <_comment>Objective-C++ source code</_comment>
+    <sub-class-of type="text/x-c++src"/>
+    <sub-class-of type="text/x-objcsrc"/>
+    <magic priority="30">
+      <match type="string" value="#import" offset="0"/>
+    </magic>
+    <glob pattern="*.mm"/>
+  </mime-type>
   <mime-type type="text/x-ocaml">
     <_comment>OCaml source code</_comment>
     <glob pattern="*.ml"/>

+    <comment xml:lang="uk">вихідний код мовою Objective-C++</comment>
+    <comment xml:lang="vi">Mã nguồn Objective-C++</comment>
+    <comment xml:lang="zh_CN">Objective-C++ 源代码</comment>
+    <comment xml:lang="zh_TW">Objective-C++ 原始碼</comment>
+    <sub-class-of type="text/x-c++src"/>
+    <sub-class-of type="text/x-objcsrc"/>
+    <magic priority="30">
+      <match value="#import" type="string" offset="0"/>
+    </magic>
+    <glob pattern="*.mm"/>
+  </mime-type>
   <mime-type type="text/x-ocaml">
     <comment>OCaml source code</comment>
     <comment xml:lang="ar">شفرة مصدر OCaml</comment>
