diff --git rsvg_internals/src/filters/component_transfer.rs.orig rsvg_internals/src/filters/component_transfer.rs
index 235435f..6845eac 100644
--- rsvg_internals/src/filters/component_transfer.rs.orig
+++ rsvg_internals/src/filters/component_transfer.rs
@@ -261,7 +261,7 @@ macro_rules! func_or_default {
                 }
             }
             _ => &$func_default,
-        };
+        }
     };
 }
 
