--- old.configure	2017-10-25 16:13:17.000000000 +0200
+++ configure	2017-10-25 15:35:12.000000000 +0200
@@ -2653,9 +2653,9 @@
 dxva2_extralibs="-luser32"
 vda_framework_deps="VideoDecodeAcceleration_VDADecoder_h blocks_extension"
 vda_framework_extralibs="-framework VideoDecodeAcceleration"
-vda_deps="vda_framework pthreads"
-vda_extralibs="-framework CoreFoundation -framework QuartzCore"
-videotoolbox_hwaccel_deps="videotoolbox pthreads"
+vda_deps=""
+vda_extralibs="-framework CoreFoundation -framework QuartzCore -framework CoreVideo"
+videotoolbox_hwaccel_deps="videotoolbox"
 videotoolbox_hwaccel_extralibs="-framework QuartzCore"
 xvmc_deps="X11_extensions_XvMClib_h"
 
--- libavcodec/old.libavcodec.v	2017-10-25 16:19:43.000000000 +0200
+++ libavcodec/libavcodec.v	2017-10-25 16:18:07.000000000 +0200
@@ -1,9 +1,10 @@
 LIBAVCODEC_MAJOR {
     global:
         av*;
+	   ff*;
         #deprecated, remove after next bump
         audio_resample;
         audio_resample_close;
     local:
         *;
 };
