diff --git a/build_test_mono.sh b/build_test_mono.sh
index 5b437c2..1113276 100755
--- a/build_test_mono.sh
+++ b/build_test_mono.sh
@@ -7,5 +7,4 @@ then
     outbits=24
 fi
 
-cargo build
-./target/debug/dsd2dxd -c 1 -f $1 -b $2 -e $3 --level=$4 < $5 | ffplay -f s${outbits}le -ar 352.8k -ch_layout mono -i -
\ No newline at end of file
+${DSD2DXD} -c 1 -f $1 -b $2 -e $3 --level=$4 < $5 | ${FFPLAY} -autoexit -hide_banner -nodisp -f s${outbits}le -ar 352.8k -ch_layout mono -i -
diff --git a/build_test_stereo.sh b/build_test_stereo.sh
index e07dad0..43ccfeb 100755
--- a/build_test_stereo.sh
+++ b/build_test_stereo.sh
@@ -7,5 +7,4 @@ then
     outbits=24
 fi
 
-cargo build
-./target/debug/dsd2dxd -f $1 -b $2 -e $3 --level=$4 -v < $5 | ffplay -f s${outbits}le -ar 352.8k -ch_layout stereo -i -
\ No newline at end of file
+${DSD2DXD} -f $1 -b $2 -e $3 --level=$4 -v < $5 | ${FFPLAY} -autoexit -hide_banner -nodisp -f s${outbits}le -ar 352.8k -ch_layout stereo -i -
diff --git a/build_test_stereo_flt.sh b/build_test_stereo_flt.sh
index 72435d8..db47326 100755
--- a/build_test_stereo_flt.sh
+++ b/build_test_stereo_flt.sh
@@ -1,4 +1,3 @@
 #!/usr/bin/env bash
 
-cargo build
-./target/debug/dsd2dxd -f $1 -b 32 -d F -e $2 --level=$3 < $4 | ffplay -f f32le -ar 352.8k -ch_layout stereo -i -
\ No newline at end of file
+${DSD2DXD} -f $1 -b 32 -d F -e $2 --level=$3 < $4 | ${FFPLAY} -autoexit -hide_banner -nodisp -f f32le -ar 352.8k -ch_layout stereo -i -
diff --git a/run_all_tests.sh b/run_all_tests.sh
index 0d99c64..9425c78 100755
--- a/run_all_tests.sh
+++ b/run_all_tests.sh
@@ -1,5 +1,10 @@
 #!/usr/bin/env bash
 
+printenv
+
+set -x
+
+mkdir -p out
 rm -rf out/*
 
 ./test_all_44k_mults.sh
@@ -9,4 +14,4 @@ rm -rf out/*
 ./build_test_stereo.sh P 16 L -4 test/1kHz_stereo_p.dsd
 ./build_test_stereo_flt.sh P L 0 test/1kHz_stereo_p.dsd
 
-dsd2dxd -R -a -o w -f p -e l -r 88200 -p out . - < test/1kHz_stereo_p.dsd
\ No newline at end of file
+${DSD2DXD} -R -a -o w -f p -e l -r 88200 -p out . - < test/1kHz_stereo_p.dsd
diff --git a/test_all_44k_mults.sh b/test_all_44k_mults.sh
index fb4eedf..98098a0 100755
--- a/test_all_44k_mults.sh
+++ b/test_all_44k_mults.sh
@@ -1,23 +1,23 @@
 #!/usr/bin/env bash
 
-cargo install --path . && dsd2dxd -r 88200   -v test/1kHz_stereo_p.dsf | ffplay -f s24le -ar 88200  -ch_layout stereo -i -
-dsd2dxd -r 176400  test/1kHz_stereo_p.dsf    | ffplay -f s24le -ar 176400 -ch_layout stereo -i -
-dsd2dxd -r 352800  test/1kHz_stereo_p.dsf    | ffplay -f s24le -ar 352800 -ch_layout stereo -i -
-dsd2dxd -r 88200   test/1kHz_stereo_128.dsf  | ffplay -f s24le -ar 88200  -ch_layout stereo -i -
-dsd2dxd -r 176400  test/1kHz_stereo_128.dsf  | ffplay -f s24le -ar 176400 -ch_layout stereo -i -
-dsd2dxd -r 352800  test/1kHz_stereo_128.dsf  | ffplay -f s24le -ar 352800 -ch_layout stereo -i -
+${DSD2DXD} -r 88200   -v test/1kHz_stereo_p.dsf | ${FFPLAY} -autoexit -hide_banner -nodisp -f s24le -ar 88200  -ch_layout stereo -i -
+${DSD2DXD} -r 176400  test/1kHz_stereo_p.dsf    | ${FFPLAY} -autoexit -hide_banner -nodisp -f s24le -ar 176400 -ch_layout stereo -i -
+${DSD2DXD} -r 352800  test/1kHz_stereo_p.dsf    | ${FFPLAY} -autoexit -hide_banner -nodisp -f s24le -ar 352800 -ch_layout stereo -i -
+${DSD2DXD} -r 88200   test/1kHz_stereo_128.dsf  | ${FFPLAY} -autoexit -hide_banner -nodisp -f s24le -ar 88200  -ch_layout stereo -i -
+${DSD2DXD} -r 176400  test/1kHz_stereo_128.dsf  | ${FFPLAY} -autoexit -hide_banner -nodisp -f s24le -ar 176400 -ch_layout stereo -i -
+${DSD2DXD} -r 352800  test/1kHz_stereo_128.dsf  | ${FFPLAY} -autoexit -hide_banner -nodisp -f s24le -ar 352800 -ch_layout stereo -i -
 
-dsd2dxd -r 88200   test/1kHz_stereo_256.dsf  | ffplay -f s24le -ar 88200   -ch_layout stereo -i -
-dsd2dxd -r 176400  test/1kHz_stereo_256.dsf  | ffplay -f s24le -ar 176400 -ch_layout stereo -i -
-dsd2dxd -r 352800  test/1kHz_stereo_256.dsf  | ffplay -f s24le -ar 352800 -ch_layout stereo -i -
+${DSD2DXD} -r 88200   test/1kHz_stereo_256.dsf  | ${FFPLAY} -autoexit -hide_banner -nodisp -f s24le -ar 88200   -ch_layout stereo -i -
+${DSD2DXD} -r 176400  test/1kHz_stereo_256.dsf  | ${FFPLAY} -autoexit -hide_banner -nodisp -f s24le -ar 176400 -ch_layout stereo -i -
+${DSD2DXD} -r 352800  test/1kHz_stereo_256.dsf  | ${FFPLAY} -autoexit -hide_banner -nodisp -f s24le -ar 352800 -ch_layout stereo -i -
 
-dsd2dxd -r 88200   test/1kHz_stereo_p.dsf    > /dev/null
-dsd2dxd -r 176400  test/1kHz_stereo_p.dsf    > /dev/null 
-dsd2dxd -r 352800  test/1kHz_stereo_p.dsf    > /dev/null
-dsd2dxd -r 88200   test/1kHz_stereo_128.dsf  > /dev/null
-dsd2dxd -r 176400  test/1kHz_stereo_128.dsf  > /dev/null
-dsd2dxd -r 352800  test/1kHz_stereo_128.dsf  > /dev/null
+${DSD2DXD} -r 88200   test/1kHz_stereo_p.dsf    > /dev/null
+${DSD2DXD} -r 176400  test/1kHz_stereo_p.dsf    > /dev/null 
+${DSD2DXD} -r 352800  test/1kHz_stereo_p.dsf    > /dev/null
+${DSD2DXD} -r 88200   test/1kHz_stereo_128.dsf  > /dev/null
+${DSD2DXD} -r 176400  test/1kHz_stereo_128.dsf  > /dev/null
+${DSD2DXD} -r 352800  test/1kHz_stereo_128.dsf  > /dev/null
 
-dsd2dxd -r 88200  test/1kHz_stereo_256.dsf  > /dev/null
-dsd2dxd -r 176400  test/1kHz_stereo_256.dsf  > /dev/null
-dsd2dxd -r 352800  test/1kHz_stereo_256.dsf  > /dev/null
\ No newline at end of file
+${DSD2DXD} -r 88200  test/1kHz_stereo_256.dsf  > /dev/null
+${DSD2DXD} -r 176400  test/1kHz_stereo_256.dsf  > /dev/null
+${DSD2DXD} -r 352800  test/1kHz_stereo_256.dsf  > /dev/null
diff --git a/test_all_48k_mults.sh b/test_all_48k_mults.sh
index e085a3a..1ea153c 100755
--- a/test_all_48k_mults.sh
+++ b/test_all_48k_mults.sh
@@ -1,21 +1,21 @@
 #!/usr/bin/env bash
 
-cargo install --path . && dsd2dxd -r 96000  test/1kHz_stereo_p.dsf | ffplay -f s24le -ar 96k -ch_layout stereo -i -
-dsd2dxd -r 192000  test/1kHz_stereo_p.dsf   | ffplay -f s24le -ar 192k -ch_layout stereo -i -
-dsd2dxd -r 384000  test/1kHz_stereo_p.dsf   | ffplay -f s24le -ar 384k -ch_layout stereo -i -
-dsd2dxd -r 96000   test/1kHz_stereo_128.dsf  | ffplay -f s24le -ar 96k -ch_layout stereo -i -
-dsd2dxd -r 192000  test/1kHz_stereo_128.dsf | ffplay -f s24le -ar 192k -ch_layout stereo -i -
-dsd2dxd -r 384000  test/1kHz_stereo_128.dsf | ffplay -f s24le -ar 384k -ch_layout stereo -i -
-dsd2dxd -r 96000   test/1kHz_stereo_256.dsf | ffplay -f s24le -ar 96k -ch_layout stereo -i -
-dsd2dxd -r 192000  test/1kHz_stereo_256.dsf | ffplay -f s24le -ar 192k -ch_layout stereo -i -
-dsd2dxd -r 384000  test/1kHz_stereo_256.dsf | ffplay -f s24le -ar 384k -ch_layout stereo -i -
+${DSD2DXD} -r 96000  test/1kHz_stereo_p.dsf | ${FFPLAY} -autoexit -hide_banner -nodisp -f s24le -ar 96k -ch_layout stereo -i -
+${DSD2DXD} -r 192000  test/1kHz_stereo_p.dsf   | ${FFPLAY} -autoexit -hide_banner -nodisp -f s24le -ar 192k -ch_layout stereo -i -
+${DSD2DXD} -r 384000  test/1kHz_stereo_p.dsf   | ${FFPLAY} -autoexit -hide_banner -nodisp -f s24le -ar 384k -ch_layout stereo -i -
+${DSD2DXD} -r 96000   test/1kHz_stereo_128.dsf  | ${FFPLAY} -autoexit -hide_banner -nodisp -f s24le -ar 96k -ch_layout stereo -i -
+${DSD2DXD} -r 192000  test/1kHz_stereo_128.dsf | ${FFPLAY} -autoexit -hide_banner -nodisp -f s24le -ar 192k -ch_layout stereo -i -
+${DSD2DXD} -r 384000  test/1kHz_stereo_128.dsf | ${FFPLAY} -autoexit -hide_banner -nodisp -f s24le -ar 384k -ch_layout stereo -i -
+${DSD2DXD} -r 96000   test/1kHz_stereo_256.dsf | ${FFPLAY} -autoexit -hide_banner -nodisp -f s24le -ar 96k -ch_layout stereo -i -
+${DSD2DXD} -r 192000  test/1kHz_stereo_256.dsf | ${FFPLAY} -autoexit -hide_banner -nodisp -f s24le -ar 192k -ch_layout stereo -i -
+${DSD2DXD} -r 384000  test/1kHz_stereo_256.dsf | ${FFPLAY} -autoexit -hide_banner -nodisp -f s24le -ar 384k -ch_layout stereo -i -
 
-dsd2dxd -r 96000   -v test/1kHz_stereo_p.dsf    > /dev/null
-dsd2dxd -r 192000  -v test/1kHz_stereo_p.dsf   > /dev/null
-dsd2dxd -r 384000  -v test/1kHz_stereo_p.dsf   > /dev/null
-dsd2dxd -r 96000   -v test/1kHz_stereo_128.dsf  > /dev/null
-dsd2dxd -r 192000  -v test/1kHz_stereo_128.dsf > /dev/null
-dsd2dxd -r 384000  -v test/1kHz_stereo_128.dsf > /dev/null
-dsd2dxd -r 96000   -v test/1kHz_stereo_256.dsf > /dev/null
-dsd2dxd -r 192000  -v test/1kHz_stereo_256.dsf > /dev/null
-dsd2dxd -r 384000  -v test/1kHz_stereo_256.dsf > /dev/null
\ No newline at end of file
+${DSD2DXD} -r 96000   -v test/1kHz_stereo_p.dsf    > /dev/null
+${DSD2DXD} -r 192000  -v test/1kHz_stereo_p.dsf   > /dev/null
+${DSD2DXD} -r 384000  -v test/1kHz_stereo_p.dsf   > /dev/null
+${DSD2DXD} -r 96000   -v test/1kHz_stereo_128.dsf  > /dev/null
+${DSD2DXD} -r 192000  -v test/1kHz_stereo_128.dsf > /dev/null
+${DSD2DXD} -r 384000  -v test/1kHz_stereo_128.dsf > /dev/null
+${DSD2DXD} -r 96000   -v test/1kHz_stereo_256.dsf > /dev/null
+${DSD2DXD} -r 192000  -v test/1kHz_stereo_256.dsf > /dev/null
+${DSD2DXD} -r 384000  -v test/1kHz_stereo_256.dsf > /dev/null
