--- src/utils/orig.kateconfig.cpp	2015-11-08 13:35:41.000000000 +0100
+++ src/utils/kateconfig.cpp	2015-12-04 21:45:49.000000000 +0100
@@ -1148,7 +1148,7 @@
     if (isGlobal()) {
         // WARNING: this is slightly hackish, but it's currently the only way to
         //          do it, see also the KTextEdit class
-        QSettings settings(QStringLiteral("KDE"), QStringLiteral("Sonnet"));
+        QSettings settings(QSettings::IniFormat, QSettings::UserScope, QStringLiteral("KDE"), QStringLiteral("Sonnet"));
         return settings.value(QStringLiteral("checkerEnabledByDefault"), false).toBool();
         //KConfigGroup configGroup(KSharedConfig::openConfig(), "Spelling");
         //return configGroup.readEntry("checkerEnabledByDefault", false);
