--- CMakeLists.txt.orig	2014-07-08 20:34:01.000000000 +0200
+++ CMakeLists.txt	2014-07-08 20:55:42.000000000 +0200
@@ -337,9 +337,9 @@
         macro_optional_add_subdirectory(knode)
       endif()
 
-      if(Q_WS_X11)
+      #if(Q_WS_X11)
         macro_optional_add_subdirectory(ktimetracker)
-      endif()
+      #endif()
 
     endif()
     macro_optional_add_subdirectory(kontact) # must be the last one.
--- ./ktimetracker/ktimetrackerutility.cpp.orig	2014-04-20 08:50:28.000000000 +0200
+++ ./ktimetracker/ktimetrackerutility.cpp	2014-07-08 23:24:39.000000000 +0200
@@ -26,8 +26,10 @@
 
 #include <math.h>
 #include <stdlib.h>
-#include <X11/Xlib.h>
-#include <fixx11h.h>
+#ifdef Q_WS_X11
+#	include <X11/Xlib.h>
+#	include <fixx11h.h>
+#endif
 
 QString getFocusWindow()
 {
