--- kmail/kmmainwidget.cpp
+++ kmail/kmmainwidget.cpp
@@ -3917,6 +3917,9 @@
 //-----------------------------------------------------------------------------
 void KMMainWidget::updateFolderMenu()
 {
+    if (kmkernel->shuttingDown()) {
+        return;
+    }
     if (!CommonKernel->outboxCollectionFolder().isValid()) {
         QTimer::singleShot(1000,this,SLOT(updateFolderMenu()));
         return;
