diff --git knode/kncollectionview.cpp knode/kncollectionview.cpp
index f8bce14..04ab82b 100644
--- knode/kncollectionview.cpp
+++ knode/kncollectionview.cpp
@@ -172,9 +172,14 @@ void KNCollectionView::removeGroup( KNGroup::Ptr g )
   if (!g->listItem())
     return;
 
-  g->listItem()->setHidden( true ); // work around bug 248256
-  delete g->listItem();
+  // RJVB
+  KNCollectionViewItem *item = g->listItem();
+  item->setDisabled(true);
+  item->setHidden( true ); // work around bug 248256
   g->setListItem(0);
+  updateGroup(g);
+  qApp->processEvents();
+  delete item;
 }
 
 
