--- /opt/local/var/macports/sources/rsync.macports.org/release/ports/kde/kdepim4/Portfile	2014-05-29 11:00:23.000000000 +0200
+++ /opt/local/site-ports/kde/kdepim4/Portfile	2014-07-08 23:50:10.000000000 +0200
@@ -7,7 +7,7 @@
 
 name                kdepim4
 version             4.12.5
-revision            1
+revision            2
 categories          kde kde4
 maintainers         intevation.de:bjoern.ricks nicos
 license             GPL-2+ LGPL-2+
@@ -37,13 +37,15 @@
 
 #QFileSystemWatcher patch: Inspired from https://git.reviewboard.kde.org/r/117993/
 patchfiles-append       patch-clang-headers.diff \
-                        patch-QFileSystemWatcher.diff 
+                        patch-QFileSystemWatcher.diff \
+                        enable-ktimetracker.patch
 
 if {![variant_isset docs]} {
     depends_lib-delete  port:dblatex
     patchfiles-append   patch-CMakeLists.diff
 }
 
+
 #Blacklist gcc42 and llvm-gcc-42
 compiler.blacklist  gcc-4.2 apple-gcc-4.2 llvm-gcc-4.2 macports-llvm-gcc-4.2
 #Also blacklist clang before Xcode 4.2
