diff --git a/src/libs/core/core.pri b/src/libs/core/core.pri
index 0f7c62e..5718e5f 100644
--- a/src/libs/core/core.pri
+++ b/src/libs/core/core.pri
@@ -2,7 +2,7 @@ ZEAL_LIB_NAME = Core
 
 QT += network
 
-unix:!macx {
+unix: {
     CONFIG += link_pkgconfig
     PKGCONFIG += libarchive
 }
diff --git a/src/libs/util/util.pri b/src/libs/util/util.pri
index f9928f9..06bf7dc 100644
--- a/src/libs/util/util.pri
+++ b/src/libs/util/util.pri
@@ -1,6 +1,6 @@
 ZEAL_LIB_NAME = Util
 
-unix:!macx {
+unix: {
     CONFIG += link_pkgconfig
     PKGCONFIG += sqlite3
 }
