diff --git a/CMakeLists.txt b/CMakeLists.txt
index 01d050f..f147a88 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -4,9 +4,11 @@ cmake_minimum_required(VERSION 3.5.1)
 set(CMAKE_ERROR_DEPRECATED TRUE)
 set(CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake")
 
-project(Zeal VERSION 0.6.1)
+project(Zeal VERSION @VERSION@)
 set(RELEASE_DATE 2018-09-28)
 
+include(FeatureSummary)
+
 # Project information.
 if(APPLE)
     set(PROJECT_OUTPUT_NAME "Zeal")
@@ -23,3 +25,5 @@ set(QT_MINIMUM_VERSION 5.9.5)
 
 add_subdirectory(assets)
 add_subdirectory(src)
+
+feature_summary(WHAT ALL   FATAL_ON_MISSING_REQUIRED_PACKAGES)
