diff --git a/src/libs/ui/docsetsdialog.cpp b/src/libs/ui/docsetsdialog.cpp
index 1434cf8..f7d8a15 100644
--- a/src/libs/ui/docsetsdialog.cpp
+++ b/src/libs/ui/docsetsdialog.cpp
@@ -403,6 +403,7 @@ void DocsetsDialog::downloadProgress(qint64 received, qint64 total)
 
     if (reply->property(DownloadTypeProperty).toInt() == DownloadDocset) {
         const QString docsetName = reply->property(DocsetNameProperty).toString();
+        ui->combinedProgressBar->setToolTip(docsetName);
 
         QTemporaryFile *tmpFile = m_tmpFiles[docsetName];
         if (!tmpFile) {
@@ -453,8 +454,8 @@ void DocsetsDialog::extractionCompleted(const QString &filePath)
         listItem->setData(ProgressItemDelegate::ShowProgressRole, false);
     }
 
-    resetProgress();
     delete m_tmpFiles.take(docsetName);
+    resetProgress();
 }
 
 void DocsetsDialog::extractionError(const QString &filePath, const QString &errorString)
@@ -477,6 +478,9 @@ void DocsetsDialog::extractionProgress(const QString &filePath, qint64 extracted
     QListWidgetItem *listItem = findDocsetListItem(docsetName);
     if (listItem)
         listItem->setData(ProgressItemDelegate::ValueRole, percent(extracted, total));
+    ui->combinedProgressBar->show();
+    ui->combinedProgressBar->setValue(percent(extracted, total));
+    ui->combinedProgressBar->setToolTip(QString::asprintf("%s [%gMb]", qPrintable(docsetName), total / (1024 * 1024.0)));
 }
 
 void DocsetsDialog::loadDocsetList()
@@ -617,6 +621,9 @@ void DocsetsDialog::enableControls()
 {
     // Available docsets
     ui->refreshButton->setEnabled(true);
+    if (m_tmpFiles.isEmpty()) {
+        ui->buttonBox->button(QDialogButtonBox::Close)->setEnabled(true);
+    }
 
     if (m_isStorageReadOnly) {
         return;
@@ -650,6 +657,8 @@ void DocsetsDialog::disableControls()
 
     // Available docsets
     ui->refreshButton->setEnabled(false);
+    // Don't let the user close the dialog (and get the "please don't" popup as a reaction)
+    ui->buttonBox->button(QDialogButtonBox::Close)->setEnabled(false);
 }
 
 QListWidgetItem *DocsetsDialog::findDocsetListItem(const QString &name) const
