diff --git a/assets/freedesktop/CMakeLists.txt b/assets/freedesktop/CMakeLists.txt
index 0e566ef..bc10937 100644
--- a/assets/freedesktop/CMakeLists.txt
+++ b/assets/freedesktop/CMakeLists.txt
@@ -1,4 +1,4 @@
-if(UNIX AND NOT APPLE)
+if(UNIX)
     find_package(ECM 1.0.0 REQUIRED NO_MODULE)
     set(CMAKE_MODULE_PATH ${ECM_MODULE_PATH})
     include(ECMInstallIcons)
diff --git a/src/app/CMakeLists.txt b/src/app/CMakeLists.txt
index cd04a08..519016c 100644
--- a/src/app/CMakeLists.txt
+++ b/src/app/CMakeLists.txt
@@ -42,7 +42,8 @@ if(APPLE)
         MACOSX_BUNDLE_COPYRIGHT ${PROJECT_COPYRIGHT}
         RESOURCE "resources/zeal.icns"
     )
-elseif(UNIX)
+endif()
+if(UNIX)
     find_package(ECM REQUIRED NO_MODULE)
     set(CMAKE_MODULE_PATH ${ECM_KDE_MODULE_DIR})
     include(KDEInstallDirs)
