diff --git a/src/app/main.cpp b/src/app/main.cpp
index 31feb7e..a7a3680 100644
--- a/src/app/main.cpp
+++ b/src/app/main.cpp
@@ -77,6 +77,9 @@ CommandLineParameters parseCommandLine(const QStringList &arguments)
     parser.addOption({{QStringLiteral("f"), QStringLiteral("force")},
                       QObject::tr("Force the application run.")});
 
+    parser.addOptions({
+        {QStringLiteral("no-sandbox"), QObject::tr("Ignored, for forward compatibility only.\n")}});
+
 #ifdef Q_OS_WIN32
     parser.addOption(QCommandLineOption({QStringLiteral("register")},
                                         QObject::tr("Register protocol handlers")));
