diff --git a/assets/assets.pro b/assets/assets.pro
index bb39f61..a934597 100644
--- a/assets/assets.pro
+++ b/assets/assets.pro
@@ -2,7 +2,7 @@ include($$ZEAL_COMMON_PRI)
 
 TEMPLATE = aux
 
-unix:!macx {
+unix: {
     TARGET = zeal
     appicons16.files=freedesktop/appicons/16/*
     appicons24.files=freedesktop/appicons/24/*
diff --git a/src/app/app.pro b/src/app/app.pro
index 71350ff..3d27217 100644
--- a/src/app/app.pro
+++ b/src/app/app.pro
@@ -27,6 +27,9 @@ win32 {
 macx {
     TARGET = Zeal
     ICON = resources/zeal.icns
+    target.path = $$PREFIX/bin
+
+    INSTALLS += target
 }
 
 # FIXME: Hardcoded link line & cyclic dependencies.
