diff --git kdevplatform/shell/openprojectdialog.cpp kdevplatform/shell/openprojectdialog.cpp
index e621edadc98c4aa492d65260f5f4ed2d072d1e1d..c9f4ad7c719ccb3912b949fb61aefafe3d385688 100644
--- kdevplatform/shell/openprojectdialog.cpp
+++ kdevplatform/shell/openprojectdialog.cpp
@@ -15,6 +15,7 @@
 #include <QPushButton>
 #include <QFileInfo>
 #include <QFileDialog>
+#include <QRegularExpression>
 
 #include <KColorScheme>
 #include <KIO/StatJob>
@@ -308,7 +313,20 @@ void OpenProjectDialog::openPageAccepted()
 
 void OpenProjectDialog::validateProjectName( const QString& name )
 {
-    m_projectName = name;
+    if (name != m_projectName) {
+        m_projectName = name;
+        QUrl url(m_url.adjusted(QUrl::StripTrailingSlash | QUrl::RemoveFilename));
+        // construct a version of the project name that's safe for use as a filename:
+        // TODO: do an additional replace of QDir::separator() with "@"?
+        QString safeName = m_projectName;
+        safeName.replace(QRegularExpression(QStringLiteral("[\\\\/]")), QStringLiteral("@"));
+        safeName = safeName.replace(QChar(':'), QChar('='));
+        safeName = safeName.replace(QRegExp("\\s"), QStringLiteral("_"));
+        safeName += '.' + ShellExtension::getInstance()->projectFileExtension();
+        m_url.setPath(url.path() + QLatin1Char('/') + safeName);
+        m_urlIsDirectory = false;
+        qCDebug(SHELL) << "project name:" << m_projectName << "file name:" << safeName;
+    }
     validateProjectInfo();
 }
 
diff --git kdevplatform/shell/projectcontroller.cpp kdevplatform/shell/projectcontroller.cpp
index d687c269d22d363652b599099b46999fd22b85cb..f6ca4412c788e91273e1c737716d50397e69b08c 100644
--- kdevplatform/shell/projectcontroller.cpp
+++ kdevplatform/shell/projectcontroller.cpp
@@ -414,6 +426,9 @@ bool equalProjectFile( const QString& configPath, OpenProjectDialog* dlg )
     KSharedConfigPtr cfg = KSharedConfig::openConfig( configPath, KConfig::SimpleConfig );
     KConfigGroup grp = cfg->group( "Project" );
     QString defaultName = dlg->projectFileUrl().adjusted(QUrl::RemoveFilename | QUrl::StripTrailingSlash).fileName();
+    qCDebug(SHELL) << "configPath=" << configPath << "defaultName=" << defaultName
+        << "projName=" << dlg->projectName() << "projectMan=" << dlg->projectManager()
+        << "grp.Name=" << grp.readEntry( "Name", QString() ) << "grp.Manager=" << grp.readEntry( "Manager", QString() );
     return (grp.readEntry( "Name", QString() ) == dlg->projectName() || dlg->projectName() == defaultName) &&
            grp.readEntry( "Manager", QString() ) == dlg->projectManager();
 }
@@ -429,7 +444,8 @@ QUrl ProjectDialogProvider::askProjectConfigLocation(bool fetch, const QUrl& sta
     }
 
     QUrl projectFileUrl = dlg->projectFileUrl();
-    qCDebug(SHELL) << "selected project:" << projectFileUrl << dlg->projectName() << dlg->projectManager();
+    qCDebug(SHELL) << "selected project:" << projectFileUrl << "selectedUrl=" << dlg->selectedUrl()
+        << "projectName=" << dlg->projectName() << "projectManager=" << dlg->projectManager();
     if ( dlg->projectManager() == QLatin1String("<built-in>") ) {
         return projectFileUrl;
     }
@@ -438,10 +454,17 @@ QUrl ProjectDialogProvider::askProjectConfigLocation(bool fetch, const QUrl& sta
     bool writeProjectConfigToFile = true;
     if( projectFileExists( projectFileUrl ) )
     {
-        // check whether config is equal
-        bool shouldAsk = true;
-        if( projectFileUrl == dlg->selectedUrl() )
+        // check whether we should question the user about overriding an existing project file or not.
+        // We don't need to do that when the file we're importing (dlg->selectedUrl) is already an
+        // existing .kdev4 project file (we just verified that it exists):
+        bool isKDevProject = QFileInfo(dlg->selectedUrl().url()).suffix() == QStringLiteral("kdev4");
+        bool shouldAsk = !isKDevProject;
+        if( !isKDevProject && projectFileUrl == dlg->selectedUrl() )
         {
+            // We're importing a project from another type of project file, post the
+            // override dialog if there's a discrepancy between the project file URL
+            // and the information stored in the dialog and the project settings.
+            qCWarning(SHELL) << "Importing a foreign project type:" << projectFileUrl.url();
             if( projectFileUrl.isLocalFile() )
             {
                 shouldAsk = !equalProjectFile( projectFileUrl.toLocalFile(), dlg );
