diff --git plugins/grepview/grepdialog.cpp plugins/grepview/grepdialog.cpp
index 3b1d7e52af071eea610206b1fb26d0fc8832b1c4..16cf6bc2bd0b3ea42e64cd7c0bf7b6781a6ef101 100644
--- plugins/grepview/grepdialog.cpp
+++ plugins/grepview/grepdialog.cpp
@@ -46,7 +46,7 @@
 #include "grepoutputview.h"
 #include "grepfindthread.h"
 #include "greputil.h"
-
+#include "debug.h"
 
 using namespace KDevelop;
 
@@ -422,6 +422,7 @@ bool GrepDialog::checkProjectsOpened()
     }
 
     // do the grep jobs one by one
+    qCWarning(PLUGIN_GREPVIEW) << "All projects loaded, repeating search history";
     connect(m_plugin, &GrepViewPlugin::grepJobFinished, this, &GrepDialog::nextHistory);
     QTimer::singleShot(0, this, [=]() {nextHistory(true);});
 
diff --git plugins/grepview/grepoutputview.cpp plugins/grepview/grepoutputview.cpp
index a7cae38b2ba8bbca1bbccc8c9a587c4f9c94f475..7aa9ff20c8dab297bda1eb8316d183395a2b4e50 100644
--- plugins/grepview/grepoutputview.cpp
+++ plugins/grepview/grepoutputview.cpp
@@ -17,6 +17,7 @@
 #include "grepdialog.h"
 #include "greputil.h"
 #include "grepjob.h"
+#include "debug.h"
 
 #include <interfaces/icore.h>
 #include <interfaces/isession.h>
@@ -152,10 +153,6 @@ GrepOutputView::GrepOutputView(QWidget* parent, GrepViewPlugin* plugin)
         m_settingsHistory << settings;
     }
 
-    // rerun the grep jobs with settings from the history
-    GrepDialog* dlg = new GrepDialog(m_plugin, this, false);
-    dlg->historySearch(m_settingsHistory);
-
     updateCheckable();
 }
 
@@ -339,6 +336,7 @@ void GrepOutputView::refresh()
             m_settingsHistory.takeAt(m_settingsHistory.count() - 1 - index)
         });
 
+        qCWarning(PLUGIN_GREPVIEW) << "Creating new GrepDialog for search #" << index;
         GrepDialog* dlg = new GrepDialog(m_plugin, this, false);
         dlg->historySearch(refresh_history);
     }
