diff --git a/kdevplatform/sublime/idealdockwidget.cpp b/kdevplatform/sublime/idealdockwidget.cpp
index 4f63ec150791ed562f8a291bddfbb3e67a68ce60..590cf2b54f4c15fb9a2aaa488e9a0823c5fabc78 100644
--- a/kdevplatform/sublime/idealdockwidget.cpp
+++ b/kdevplatform/sublime/idealdockwidget.cpp
@@ -37,8 +37,25 @@
 #include <QToolBar>
 #include <QVBoxLayout>
 
+#include <QDebug>
+#include <QApplication>
+
 using namespace Sublime;
 
+static void unfloat(IdealDockWidget* that, bool floating)
+{
+    if (floating && that && that->isVisible()) {
+        qWarning() << "Unfloating dock widget" << that;
+        that->setTitleBarWidget(0);
+        // remove unwanted flags
+        Qt::WindowFlags flags = that->windowFlags() & ~(Qt::Tool|Qt::FramelessWindowHint|Qt::X11BypassWindowManagerHint);
+        // set wanted flags
+        that->setWindowFlags(flags | Qt::Window);
+        that->updateGeometry();
+        that->setVisible(true);
+    }
+}
+
 IdealDockWidget::IdealDockWidget(IdealController *controller, Sublime::MainWindow *parent)
     : QDockWidget(parent),
       m_area(nullptr),
@@ -61,6 +78,15 @@ IdealDockWidget::IdealDockWidget(IdealController *controller, Sublime::MainWindo
     setFeatures(QDockWidget::DockWidgetClosable | QDockWidget::DockWidgetMovable | QDockWidget::DockWidgetFloatable);
     // do not allow to move docks to the top dock area (no buttonbar there in our current UI)
     setAllowedAreas(Qt::LeftDockWidgetArea | Qt::RightDockWidgetArea | Qt::BottomDockWidgetArea);
+
+    connect(this, &QDockWidget::topLevelChanged, this, [this] (bool floating) {
+        unfloat(this, floating); } );
+    // somehow this connect never triggers??
+    connect(qApp, &QApplication::aboutToQuit, this, [this] () {
+        /*if (isFloating())*/ {
+            qWarning() << "Re-docking" << this << "at exit";
+            setFloating(false);
+        } } );
 }
 
 IdealDockWidget::~IdealDockWidget()
