diff --git kdevplatform/project/CMakeLists.txt kdevplatform/project/CMakeLists.txt
index c61e8de32e80c29a648766fe88c62b78f5561c36..d8b14d82407b1ec5a886d2fbd60dd77bd8bceb33 100644
--- kdevplatform/project/CMakeLists.txt
+++ kdevplatform/project/CMakeLists.txt
@@ -15,6 +15,7 @@ set(KDevPlatformProject_LIB_SRCS
     abstractfilemanagerplugin.cpp
     filemanagerlistjob.cpp
     projectfiltermanager.cpp
+    projectwatcher.cpp
     interfaces/iprojectbuilder.cpp
     interfaces/iprojectfilemanager.cpp
     interfaces/ibuildsystemmanager.cpp
diff --git kdevplatform/project/abstractfilemanagerplugin.cpp kdevplatform/project/abstractfilemanagerplugin.cpp
index 09960eabfed72fae6454b5d555f7f52a3ee23ecf..e74d4a9690ecb5d3c9fd62a4ceecfec616344220 100644
--- kdevplatform/project/abstractfilemanagerplugin.cpp
+++ kdevplatform/project/abstractfilemanagerplugin.cpp
@@ -1,3 +1,4 @@
+#define TIME_IMPORT_JOB
 /***************************************************************************
  *   This file is part of KDevelop                                         *
  *   Copyright 2010-2012 Milian Wolff <mail@milianw.de>                    *
@@ -27,6 +28,9 @@
 #include <QHashIterator>
 #include <QFileInfo>
 #include <QApplication>
+#ifdef TIME_IMPORT_JOB
+#include <QElapsedTimer>
+#endif
 
 #include <KMessageBox>
 #include <KLocalizedString>
@@ -38,6 +42,7 @@
 #include <serialization/indexedstring.h>
 
 #include "projectfiltermanager.h"
+#include "projectwatcher.h"
 #include "debug.h"
 
 #define ifDebug(x)
@@ -86,7 +91,7 @@ public:
                      const KIO::UDSEntryList& entries);
 
     void deleted(const QString &path);
-    void created(const QString &path);
+    void dirty(const QString &path);
 
     void projectClosing(IProject* project);
     void jobFinished(KJob* job);
@@ -100,7 +105,7 @@ public:
 
     void removeFolder(ProjectFolderItem* folder);
 
-    QHash<IProject*, KDirWatch*> m_watchers;
+    QHash<IProject*, ProjectWatcher*> m_watchers;
     QHash<IProject*, QList<FileManagerListJob*> > m_projectJobs;
     QVector<QString> m_stoppedFolders;
     ProjectFilterManager m_filters;
@@ -117,12 +122,26 @@ void AbstractFileManagerPluginPrivate::projectClosing(IProject* project)
         }
         m_projectJobs.remove(project);
     }
+#ifdef TIME_IMPORT_JOB
+    QElapsedTimer timer;
+    int watched;
+    if (m_watchers.contains(project)) {
+        watched = m_watchers[project]->size();
+        timer.start();
+    }
+#endif
     delete m_watchers.take(project);
+#ifdef TIME_IMPORT_JOB
+    if (timer.isValid()) {
+        qCInfo(FILEMANAGER) << "Deleting" << watched << "entry dir watcher took" << timer.elapsed() / 1000.0 << "seconds for project" << project->name();
+    }
+#endif
     m_filters.remove(project);
 }
 
 KIO::Job* AbstractFileManagerPluginPrivate::eventuallyReadFolder(ProjectFolderItem* item)
 {
+    ProjectWatcher* watcher = m_watchers.value( item->project(), nullptr );
     FileManagerListJob* listJob = new FileManagerListJob( item );
     m_projectJobs[ item->project() ] << listJob;
     qCDebug(FILEMANAGER) << "adding job" << listJob << item << item->path() << "for project" << item->project();
@@ -247,9 +262,9 @@ void AbstractFileManagerPluginPrivate::addJobItems(FileManagerListJob* job,
     }
 }
 
-void AbstractFileManagerPluginPrivate::created(const QString& path_)
+void AbstractFileManagerPluginPrivate::dirty(const QString& path_)
 {
-    qCDebug(FILEMANAGER) << "created:" << path_;
+    qCDebug(FILEMANAGER) << "dirty:" << path_;
     QFileInfo info(path_);
 
     ///FIXME: share memory with parent
@@ -257,7 +272,7 @@ void AbstractFileManagerPluginPrivate::created(const QString& path_)
     const IndexedString indexedPath(path.pathOrUrl());
     const IndexedString indexedParent(path.parent().pathOrUrl());
 
-    QHashIterator<IProject*, KDirWatch*> it(m_watchers);
+    QHashIterator<IProject*, ProjectWatcher*> it(m_watchers);
     while (it.hasNext()) {
         const auto p = it.next().key();
         if ( !p->projectItem()->model() ) {
@@ -285,21 +300,6 @@ void AbstractFileManagerPluginPrivate::created(const QString& path_)
             // also gets triggered for kate's backup files
             continue;
         }
-        foreach ( ProjectFolderItem* parentItem, p->foldersForPath(indexedParent) ) {
-            if ( info.isDir() ) {
-                ProjectFolderItem* folder = q->createFolderItem( p, path, parentItem );
-                if (folder) {
-                    emit q->folderAdded( folder );
-                    auto job = eventuallyReadFolder( folder );
-                    job->start();
-                }
-            } else {
-                ProjectFileItem* file = q->createFileItem( p, path, parentItem );
-                if (file) {
-                    emit q->fileAdded( file );
-                }
-            }
-        }
     }
 }
 
@@ -320,7 +320,7 @@ void AbstractFileManagerPluginPrivate::deleted(const QString& path_)
     const Path path(QUrl::fromLocalFile(path_));
     const IndexedString indexed(path.pathOrUrl());
 
-    QHashIterator<IProject*, KDirWatch*> it(m_watchers);
+    QHashIterator<IProject*, ProjectWatcher*> it(m_watchers);
     while (it.hasNext()) {
         const auto p = it.next().key();
         if (path == p->path()) {
@@ -436,6 +436,9 @@ void AbstractFileManagerPluginPrivate::removeFolder(ProjectFolderItem* folder)
             job->removeSubDir(folder);
         }
     }
+    ProjectWatcher* watcher = m_watchers.value(folder->project(), nullptr);
+    Q_ASSERT(watcher);
+    watcher->removeDir(folder->path().toLocalFile());
     folder->parent()->removeRow( folder->row() );
 }
 
@@ -477,14 +480,14 @@ ProjectFolderItem *AbstractFileManagerPlugin::import( IProject *project )
 
     ///TODO: check if this works for remote files when something gets changed through another KDE app
     if ( project->path().isLocalFile() ) {
-        d->m_watchers[project] = new KDirWatch( project );
+        auto watcher = new ProjectWatcher(project);
 
-        connect(d->m_watchers[project], &KDirWatch::created,
-                this, [&] (const QString& path_) { d->created(path_); });
-        connect(d->m_watchers[project], &KDirWatch::deleted,
+        // set up the signal handling; feeding the dirwatcher is handled by FileManagerListJob.
+        connect(watcher, &KDirWatch::dirty,
+                this, [&] (const QString& path_) { d->dirty(path_); });
+        connect(watcher, &KDirWatch::deleted,
                 this, [&] (const QString& path_) { d->deleted(path_); });
-
-        d->m_watchers[project]->addDir(project->path().toLocalFile(), KDirWatch::WatchSubDirs | KDirWatch:: WatchFiles );
+        d->m_watchers[project] = watcher;
     }
 
     d->m_filters.add(project);
@@ -661,6 +664,18 @@ KDirWatch* AbstractFileManagerPlugin::projectWatcher( IProject* project ) const
     return d->m_watchers.value( project, nullptr );
 }
 
+int AbstractFileManagerPlugin::watchedItems( IProject* project ) const
+{
+    auto watcher = d->m_watchers.value( project, nullptr );
+    Q_ASSERT(watcher);
+    return watcher->size();
+}
+
+void AbstractFileManagerPlugin::projectClosing( IProject *project )
+{
+    d->projectClosing(project);
+}
+
 //END Plugin
 
 #include "moc_abstractfilemanagerplugin.cpp"
diff --git kdevplatform/project/abstractfilemanagerplugin.h kdevplatform/project/abstractfilemanagerplugin.h
index c99e7e0c2316bf24df76d303fdb599a8095193da..395fc47e1767670bf674750d0aba54ee21c68bb0 100644
--- kdevplatform/project/abstractfilemanagerplugin.h
+++ kdevplatform/project/abstractfilemanagerplugin.h
@@ -34,6 +34,7 @@ class KDirWatch;
 namespace KDevelop {
 
 class AbstractFileManagerPluginPrivate;
+class ProjectController;
 
 /**
  * This class can be used as a common base for file managers.
@@ -103,6 +104,16 @@ protected:
      */
     KDirWatch* projectWatcher( IProject* project ) const;
 
+    /**
+     * @return the number of watched directories for the given @p project.
+     */
+    int watchedItems( IProject* project ) const;
+
+    /**
+     * tell the plugin that the given @p project is going to be closed.
+     */
+    void projectClosing( IProject* project );
+
 Q_SIGNALS:
     void reloadedFileItem(KDevelop::ProjectFileItem* file);
     void reloadedFolderItem(KDevelop::ProjectFolderItem* folder);
@@ -118,6 +129,8 @@ Q_SIGNALS:
 private:
     const QScopedPointer<class AbstractFileManagerPluginPrivate> d;
     friend class AbstractFileManagerPluginPrivate;
+public:
+    friend class ProjectController;
 };
 
 }
diff --git kdevplatform/project/filemanagerlistjob.cpp kdevplatform/project/filemanagerlistjob.cpp
index 06e1d69c70ba762cc476086794a79324871401f9..782996bf592cafeed3517af7d97017950d74302d 100644
--- kdevplatform/project/filemanagerlistjob.cpp
+++ kdevplatform/project/filemanagerlistjob.cpp
@@ -18,6 +18,7 @@
 */
 
 #include "filemanagerlistjob.h"
+#include "projectwatcher.h"
 
 #include <interfaces/iproject.h>
 #include <project/projectmodel.h>
@@ -28,10 +29,12 @@
 #include <QtConcurrentRun>
 #include <QDir>
 
+#include <KDirWatch>
+
 using namespace KDevelop;
 
-FileManagerListJob::FileManagerListJob(ProjectFolderItem* item)
-    : KIO::Job(), m_item(item), m_aborted(false)
+FileManagerListJob::FileManagerListJob(ProjectFolderItem* item, ProjectWatcher* watcher)
+    : KIO::Job(), m_item(item), m_aborted(false), m_watcher(watcher)
 {
     qRegisterMetaType<KIO::UDSEntryList>("KIO::UDSEntryList");
     qRegisterMetaType<KIO::Job*>();
@@ -95,6 +98,7 @@ void FileManagerListJob::startNextJob()
             if (m_aborted) {
                 return;
             }
+            m_watcher->addDir(path.toLocalFile());
             KIO::UDSEntryList results;
             std::transform(entries.begin(), entries.end(), std::back_inserter(results), [] (const QFileInfo& info) -> KIO::UDSEntry {
                 KIO::UDSEntry entry;
diff --git kdevplatform/project/filemanagerlistjob.h kdevplatform/project/filemanagerlistjob.h
index 0d32c992c47ebc7d084a019badb64988f0be9ec0..92b101de4e6a44bb1962274d4bc39064bf914aa8 100644
--- kdevplatform/project/filemanagerlistjob.h
+++ kdevplatform/project/filemanagerlistjob.h
@@ -33,13 +33,14 @@
 namespace KDevelop
 {
     class ProjectFolderItem;
+    class ProjectWatcher;
 
 class FileManagerListJob : public KIO::Job
 {
     Q_OBJECT
 
 public:
-    explicit FileManagerListJob(ProjectFolderItem* item);
+    explicit FileManagerListJob(ProjectFolderItem* item, ProjectWatcher* watcher);
     ProjectFolderItem* item() const;
 
     void addSubDir(ProjectFolderItem* item);
@@ -73,6 +74,7 @@ private:
     QElapsedTimer m_subTimer;
     qint64 m_subWaited = 0;
 #endif
+    ProjectWatcher* m_watcher;
 };
 
 }
diff --git kdevplatform/project/projectwatcher.cpp kdevplatform/project/projectwatcher.cpp
new file mode 100644
index 0000000000000000000000000000000000000000..531b5141e014fc6d52db5426c01cdc433f738afa
--- /dev/null
+++ kdevplatform/project/projectwatcher.cpp
@@ -0,0 +1,58 @@
+/***************************************************************************
+ *   This file is part of KDevelop                                         *
+ *   Copyright 2017 René Bertin <rjvbertin@gmail.com>                      *
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU Library General Public License as       *
+ *   published by the Free Software Foundation; either version 2 of the    *
+ *   License, or (at your option) any later version.                       *
+ *                                                                         *
+ *   This program is distributed in the hope that it will be useful,       *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU General Public License for more details.                          *
+ *                                                                         *
+ *   You should have received a copy of the GNU Library General Public     *
+ *   License along with this program; if not, write to the                 *
+ *   Free Software Foundation, Inc.,                                       *
+ *   51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.         *
+ ***************************************************************************/
+
+#include "projectwatcher.h"
+
+#include <QMutexLocker>
+
+#include <KDirWatch>
+
+using namespace KDevelop;
+
+QMutex KDevelop::ProjectWatcher::s_mutex;
+
+KDevelop::ProjectWatcher::ProjectWatcher(QObject* parent)
+    : KDirWatch(parent)
+    , m_watchedCount(0)
+{}
+
+void KDevelop::ProjectWatcher::addDir(const QString& path, WatchModes watchModes)
+{
+    QMutexLocker lock(&s_mutex);
+    if (!contains(path)) {
+        KDirWatch::addDir(path, watchModes);
+        m_watchedCount += 1;
+    }
+}
+
+void KDevelop::ProjectWatcher::removeDir(const QString& path)
+{
+    QMutexLocker lock(&s_mutex);
+    if (contains(path)) {
+        KDirWatch::removeDir(path);
+        m_watchedCount -= 1;
+    }
+}
+
+int KDevelop::ProjectWatcher::size() const
+{
+    return m_watchedCount;
+}
+
diff --git kdevplatform/project/projectwatcher.h kdevplatform/project/projectwatcher.h
new file mode 100644
index 0000000000000000000000000000000000000000..483c99b47eb4170abfedb3a9256a38ed76c6d30e
--- /dev/null
+++ kdevplatform/project/projectwatcher.h
@@ -0,0 +1,52 @@
+/***************************************************************************
+ *   This file is part of KDevelop                                         *
+ *   Copyright 2017 René Bertin <rjvbertin@gmail.com>                      *
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU Library General Public License as       *
+ *   published by the Free Software Foundation; either version 2 of the    *
+ *   License, or (at your option) any later version.                       *
+ *                                                                         *
+ *   This program is distributed in the hope that it will be useful,       *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU General Public License for more details.                          *
+ *                                                                         *
+ *   You should have received a copy of the GNU Library General Public     *
+ *   License along with this program; if not, write to the                 *
+ *   Free Software Foundation, Inc.,                                       *
+ *   51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.         *
+ ***************************************************************************/
+
+#ifndef KDEVPLATFORM_PROJECTWATCHER_H
+#define KDEVPLATFORM_PROJECTWATCHER_H
+
+#include <KDirWatch>
+#include <QMutex>
+
+namespace KDevelop {
+
+class ProjectWatcher : public KDirWatch
+{
+public:
+    explicit ProjectWatcher(QObject* parent = nullptr);
+
+    void addDir(const QString& path, WatchModes watchModes = WatchDirOnly);
+    void removeDir(const QString& path);
+
+    int size() const;
+
+private:
+    // The QFileSystemWatcher backend doesn't like to be called
+    // too often/concurrently; prevent concurrent calls regardless
+    // of what KDirWatch instance they come from (=> static QMutex).
+    // Concurrent entry adding/removing can happen in trees with lots
+    // of directories with very few elements.
+    // (On Mac, this improves speed and prevent lock-ups; on Linux
+    // it will prevent occasional double-free issues.)
+    static QMutex s_mutex;
+    int m_watchedCount;
+};
+
+}
+#endif //KDEVPLATFORM_PROJECTWATCHER_H
diff --git kdevplatform/project/tests/CMakeLists.txt kdevplatform/project/tests/CMakeLists.txt
index 74e5baada32fcc570f410e50c26bfc83a5066830..5a8d80cfe1a933bb839114cf7059152fc235adbd 100644
--- kdevplatform/project/tests/CMakeLists.txt
+++ kdevplatform/project/tests/CMakeLists.txt
@@ -21,3 +21,12 @@ target_link_libraries(abstractfilemanagerpluginimporttest
     KDev::Tests
     Qt5::QuickWidgets
 )
+
+add_executable(abstractfilemanagerpluginimportbenchmark
+    abstractfilemanagerpluginimportbenchmark.cpp
+)
+ecm_mark_nongui_executable(abstractfilemanagerpluginimportbenchmark)
+target_link_libraries(abstractfilemanagerpluginimportbenchmark
+    KDev::Project
+    KDev::Tests
+)
diff --git kdevplatform/project/tests/abstractfilemanagerpluginimportbenchmark.cpp kdevplatform/project/tests/abstractfilemanagerpluginimportbenchmark.cpp
new file mode 100644
index 0000000000000000000000000000000000000000..09c5562e6330936a0eeacc09964439bc359af5b4
--- /dev/null
+++ kdevplatform/project/tests/abstractfilemanagerpluginimportbenchmark.cpp
@@ -0,0 +1,177 @@
+/* This file is part of KDevelop
+    Copyright 2017 Ren? J.V. Bertin <rjvbertin@gmail.com>
+
+    This library is free software; you can redistribute it and/or
+    modify it under the terms of the GNU Library General Public
+    License as published by the Free Software Foundation; either
+    version 2 of the License, or (at your option) any later version.
+
+    This library is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+    Library General Public License for more details.
+
+    You should have received a copy of the GNU Library General Public License
+    along with this library; see the file COPYING.LIB.  If not, write to
+    the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+    Boston, MA 02110-1301, USA.
+*/
+
+#include <project/abstractfilemanagerplugin.h>
+#include <project/projectmodel.h>
+
+#include <tests/autotestshell.h>
+#include <tests/testcore.h>
+#include <tests/testproject.h>
+
+#include <util/path.h>
+
+#include <KJob>
+#include <KDirWatch>
+
+#include <QCoreApplication>
+#include <QList>
+#include <QFileInfo>
+#include <QElapsedTimer>
+#include <QMap>
+#include <QDebug>
+
+using namespace KDevelop;
+
+class TestFileManagerPlugin : public AbstractFileManagerPlugin
+{
+    Q_OBJECT
+public:
+    TestFileManagerPlugin(QObject* parent = nullptr)
+        : AbstractFileManagerPlugin({}, parent)
+    {}
+
+    using AbstractFileManagerPlugin::watchedItems;
+
+    using AbstractFileManagerPlugin::projectClosing;
+};
+
+class AFMPBenchmark : public QObject {
+    Q_OBJECT
+public:
+    AFMPBenchmark(TestFileManagerPlugin* manager, const QString& path, QObject* parent)
+        : QObject(parent)
+    {
+        if (QFileInfo(path).isDir()) {
+            m_manager = manager;
+            m_project = new TestProject(Path(path));
+        } else {
+            qWarning() << path << "is not a directory, ignoring";
+            m_project = nullptr;
+        }
+    }
+
+    void start()
+    {
+        if (!m_project) {
+            return;
+        }
+        m_projectNumber = s_count++;
+        qInfo() << "Starting import of project #" << m_projectNumber << "at" << m_project->path();
+        m_timer.start();
+        auto root = m_manager->import(m_project);
+        int elapsed = m_timer.elapsed();
+        qInfo() << "\tcreating dirwatcher took"
+            << elapsed / 1000.0 << "seconds";
+        auto import = m_manager->createImportJob(root);
+        QObject::connect(import, &KJob::finished, this, &AFMPBenchmark::projectImportDone);
+        m_timer.restart();
+        import->start();
+    }
+
+    void projectClosing()
+    {
+        m_timer.restart();
+        m_manager->projectClosing(m_project);
+        int elapsed = m_timer.elapsed();
+        qInfo() << "\tclosing project" << m_projectNumber << ":" << elapsed / 1000.0 << "seconds";
+    }
+
+    TestFileManagerPlugin* m_manager;
+    TestProject* m_project;
+    QElapsedTimer m_timer;
+    int m_projectNumber;
+
+    static int s_count;
+
+Q_SIGNALS:
+    void finished();
+
+private Q_SLOTS:
+    void projectImportDone(KJob* job)
+    {
+        Q_UNUSED(job);
+        int elapsed = m_timer.elapsed();
+        qInfo() << "imported project" << m_projectNumber
+            << "with" << m_project->fileSet().size()
+            << "files (watched:" << m_manager->watchedItems(m_project) << "):"
+            << elapsed / 1000.0 << "seconds";
+
+        s_count -= 1;
+        if (s_count <= 0) {
+            qInfo() << "Done.";
+            emit finished();
+        }
+    }
+
+};
+
+int AFMPBenchmark::s_count = 0;
+
+int main(int argc, char** argv)
+{
+    if (argc < 2) {
+        qWarning() << "Usage:" << argv[0] << "projectDir1 [...projectDirN]";
+        return 1;
+    }
+    QCoreApplication app(argc, argv);
+
+    AutoTestShell::init();
+    auto core = TestCore::initialize(Core::NoUi);
+    auto manager = new TestFileManagerPlugin(core);
+
+    const char *kdwMethod[] = {"FAM", "Inotify", "Stat", "QFSWatch"};
+    qInfo() << "KDirWatch backend:" << kdwMethod[KDirWatch().internalMethod()];
+
+    QList<AFMPBenchmark*> benchmarks;
+
+    for (int i = 1 ; i < argc ; ++i) {
+        const auto benchmark = new AFMPBenchmark(manager, QString::fromUtf8(argv[i]), core);
+        benchmarks << benchmark;
+        QObject::connect(benchmark, &AFMPBenchmark::finished,
+                     &app, [&benchmarks] {
+                        for (auto benchmark : benchmarks) {
+                            benchmark->projectClosing();
+                        }
+                        QCoreApplication::instance()->quit();
+                     });
+        if (qEnvironmentVariableIsSet("BENCHMARK_ORIGINAL_DIRWATCHER")) {
+            // benchmark the creation and deletion of the original dirwatcher:
+            KDirWatch *watcher = new KDirWatch(benchmark->m_project);
+            qInfo() << "Benchmarking KDirWatch for all of" << argv[i];
+            benchmark->m_timer.start();
+            watcher->addDir(benchmark->m_project->path().toLocalFile(), KDirWatch::WatchSubDirs | KDirWatch:: WatchFiles );
+            int elapsed = benchmark->m_timer.elapsed();
+            qInfo() << "\tfeeding the watcher:" << elapsed / 1000.0 << "seconds";
+            benchmark->m_timer.restart();
+            delete watcher;
+            elapsed = benchmark->m_timer.elapsed();
+            qInfo() << "\tdeleting the watcher:" << elapsed / 1000.0 << "seconds";
+        }
+    }
+    for (auto benchmark : benchmarks) {
+        benchmark->start();
+    }
+
+    if (benchmarks.first()->s_count > 0) {
+        return app.exec();
+    }
+    return 1;
+}
+
+#include "abstractfilemanagerpluginimportbenchmark.moc"
