diff --git plugins/outlineview/outlinewidget.cpp plugins/outlineview/outlinewidget.cpp
index 6358b4a254bdea070dec89569fb5d9f96b2e435f..9ca049c90bf723ee19eea402e86c4e6f55fc59c3 100644
--- plugins/outlineview/outlinewidget.cpp
+++ plugins/outlineview/outlinewidget.cpp
@@ -57,6 +57,8 @@ OutlineWidget::OutlineWidget(QWidget* parent, OutlineViewPlugin* plugin)
     // sort action
     m_sortAlphabeticallyAction = new QAction(QIcon::fromTheme(QStringLiteral("view-sort-ascending")), QString(), this);
     m_sortAlphabeticallyAction->setToolTip(i18n("Sort alphabetically"));
+    // reuse the tooltip for the name under which the action will show up in the context menu:
+    m_sortAlphabeticallyAction->setText(m_sortAlphabeticallyAction->toolTip());
     m_sortAlphabeticallyAction->setCheckable(true);
     connect(m_sortAlphabeticallyAction, &QAction::triggered, this, [this](bool sort) {
         // calling sort with -1 will restore the original order
@@ -70,6 +72,11 @@ OutlineWidget::OutlineWidget(QWidget* parent, OutlineViewPlugin* plugin)
     connect(m_tree, &QTreeView::activated, this, &OutlineWidget::activated);
     m_filter->setPlaceholderText(i18n("Filter..."));
     auto filterAction = new QWidgetAction(this);
+    // give the action a sensible name and function in the context menu (where it'll appear anyway)
+    filterAction->setText(i18n("Clear filter"));
+    connect(filterAction, &QAction::triggered, this, [this](bool) {
+        m_filter->setText(QString());
+    });
     filterAction->setDefaultWidget(m_filter);
     addAction(filterAction);
 
