diff --git plugins/manpage/manpageplugin.cpp plugins/manpage/manpageplugin.cpp
index 0690b5add0..fea9b5b2ac 100644
--- plugins/manpage/manpageplugin.cpp
+++ plugins/manpage/manpageplugin.cpp
@@ -91,7 +91,8 @@ IDocumentation::Ptr ManPagePlugin::documentationForDeclaration( Declaration* dec
 
     // Don't show man-page documentation for files that are not in /usr/include, because then we
     // most probably will be confusing the global function-name with a local one
-    if (!dec->topContext()->url().str().startsWith(QLatin1String("/usr/")))
+    if (!dec->topContext()->url().str().startsWith(QLatin1String("/usr/"))
+            && !dec->topContext()->url().str().startsWith(QLatin1String("@PREFIX@")))
         return {};
 
     ///@todo Do more verification to make sure that we're showing the correct documentation for the declaration
