diff --git plugins/clang/clangparsejob.cpp plugins/clang/clangparsejob.cpp
index 200772dc2c..adaca61e80 100644
--- plugins/clang/clangparsejob.cpp
+++ plugins/clang/clangparsejob.cpp
@@ -366,8 +366,9 @@ void ClangParseJob::run(ThreadWeaver::JobPointer /*self*/, ThreadWeaver::Thread*
                 // otherwise no editor component is open for this document and we can dispose
                 // the TU to save memory
                 // share the session data with all contexts that are pinned to this TU
-                DUChainWriteLocker lock;
-                context->setAst(IAstContainer::Ptr(session.data()));
+//                 DUChainWriteLocker lock;
+//                 context->setAst(IAstContainer::Ptr(session.data()));
+                // qWarning() << Q_FUNC_INFO << "Not caching session for" << context->url();
             }
             languageSupport()->codeHighlighting()->highlightDUChain(context);
         }
diff --git plugins/clang/duchain/parsesession.cpp plugins/clang/duchain/parsesession.cpp
index 361639808b..29105de086 100644
--- plugins/clang/duchain/parsesession.cpp
+++ plugins/clang/duchain/parsesession.cpp
@@ -44,6 +44,9 @@
 #include <QMimeType>
 
 #include <algorithm>
+#ifdef Q_OS_UNIX
+#include <unistd.h>
+#endif
 
 using namespace KDevelop;
 
@@ -309,10 +312,16 @@ ParseSessionData::ParseSessionData(const QVector<UnsavedFile>& unsavedFiles, Cla
     } else {
         qCWarning(KDEV_CLANG) << "Failed to parse translation unit:" << tuUrl;
     }
+#ifdef Q_OS_UNIX
+    // it should be safe now to unlink the file behind Qt's back so that
+    // it is guaranteed to be deleted when KDevelop exists, cleanly or not.
+    unlink(m_definesFile.fileName().toUtf8().constData());
+#endif
 }
 
 ParseSessionData::~ParseSessionData()
 {
+    // qWarning() << Q_FUNC_INFO << "tr.unit" << m_unit << "m_definesFile=" << m_definesFile.fileName() << "m_file=" << clang_getCString(clang_getFileName(m_file));
     clang_disposeTranslationUnit(m_unit);
 }
 
@@ -327,6 +336,7 @@ QByteArray ParseSessionData::writeDefinesFile(const QMap<QString, QString>& defi
     for (auto it = defines.begin(); it != defines.end(); ++it) {
         definesStream << QStringLiteral("#define ") << it.key() << ' ' << it.value() << '\n';
     }
+    m_definesFile.close();
 
     return m_definesFile.fileName().toUtf8();
 }
