diff --git a/plugins/cmake/cmakecommitchangesjob.cpp b/plugins/cmake/cmakecommitchangesjob.cpp
index 3218a96..b9b151a 100644
--- a/plugins/cmake/cmakecommitchangesjob.cpp
+++ b/plugins/cmake/cmakecommitchangesjob.cpp
@@ -162,7 +162,7 @@ Path::List CMakeCommitChangesJob::addProjectData(const CMakeProjectData& data)
     QString dir = m_path.toLocalFile();
     if(data.vm.value("CMAKE_INCLUDE_CURRENT_DIR")==QStringList("ON")) {
         m_directories += dir;
-        m_directories += CMakeParserUtils::binaryPath(dir, m_project->path().toLocalFile(), CMake::currentBuildDir(m_project).toLocalFile(QUrl::RemoveTrailingSlash));
+        m_directories += CMakeParserUtils::binaryPath(dir, m_project->path().toLocalFile(), CMake::currentCanonicalBuildDir(m_project).toLocalFile(QUrl::RemoveTrailingSlash));
     }
     m_directories += resolvePaths(m_path, data.properties[DirectoryProperty][dir]["INCLUDE_DIRECTORIES"]);
     m_directories.removeAll(QString());
diff --git a/plugins/cmake/cmakeimportjsonjob.cpp b/plugins/cmake/cmakeimportjsonjob.cpp
index 004ae18..dbfbed9 100644
--- a/plugins/cmake/cmakeimportjsonjob.cpp
+++ b/plugins/cmake/cmakeimportjsonjob.cpp
@@ -94,7 +94,7 @@ CMakeFilesCompilationData importCommands(const Path& commandsFile)
         ret.includes = kTransform<Path::List>(result.paths, convert);
         ret.frameworkDirectories = kTransform<Path::List>(result.frameworkDirectories, convert);
         ret.defines = result.defines;
-        const Path path(rt->pathInHost(Path(entry[KEY_FILE].toString())));
+        const Path path(rt->pathInHost(Path((QFileInfo(entry[KEY_FILE].toString()).canonicalFilePath()))));
         qCDebug(CMAKE) << "entering..." << path << entry[KEY_FILE];
         data.files[path] = ret;
     }
@@ -142,7 +142,7 @@ void CMakeImportJsonJob::start()
         return;
     }
 
-    const Path currentBuildDir = CMake::currentBuildDir(m_project);
+    const Path currentBuildDir = CMake::currentCanonicalBuildDir(m_project);
     Q_ASSERT (!currentBuildDir.isEmpty());
 
     const Path targetsFilePath = CMake::targetDirectoriesFile(m_project);
diff --git a/plugins/cmake/cmakemanager.cpp b/plugins/cmake/cmakemanager.cpp
index 5c6a11f..b004936 100644
--- a/plugins/cmake/cmakemanager.cpp
+++ b/plugins/cmake/cmakemanager.cpp
@@ -128,7 +128,7 @@ Path CMakeManager::buildDirectory(KDevelop::ProjectBaseItem *item) const
 //     if(fi)
 //         ret.addPath(fi->buildDir());
 //     return ret;
-    return Path(CMake::currentBuildDir(item->project()));
+    return Path(CMake::currentCanonicalBuildDir(item->project()));
 }
 
 KDevelop::ProjectFolderItem* CMakeManager::import( KDevelop::IProject *project )
diff --git a/plugins/cmake/cmakemodelitems.cpp b/plugins/cmake/cmakemodelitems.cpp
index 1d98bce..ff64669 100644
--- a/plugins/cmake/cmakemodelitems.cpp
+++ b/plugins/cmake/cmakemodelitems.cpp
@@ -32,7 +32,7 @@ QUrl CMakeTargetItem::builtUrl() const
     if (!m_builtUrl.isEmpty())
         return m_builtUrl.toUrl();
 
-    const KDevelop::Path buildDir = CMake::currentBuildDir(project());
+    const KDevelop::Path buildDir = CMake::currentCanonicalBuildDir(project());
     if (buildDir.isEmpty())
         return QUrl();
 
diff --git a/plugins/cmake/cmakeserverimportjob.cpp b/plugins/cmake/cmakeserverimportjob.cpp
index f93ded0..acb7d82 100644
--- a/plugins/cmake/cmakeserverimportjob.cpp
+++ b/plugins/cmake/cmakeserverimportjob.cpp
@@ -163,7 +163,8 @@ void CMakeServerImportJob::doStart()
 {
     connect(m_server.data(), &CMakeServer::response, this, &CMakeServerImportJob::processResponse);
 
-    m_server->handshake(m_project->path(), CMake::currentBuildDir(m_project));
+    // maybe use the canonical project path too?
+    m_server->handshake(m_project->path(), CMake::currentCanonicalBuildDir(m_project));
 }
 
 void CMakeServerImportJob::processResponse(const QJsonObject& response)
diff --git a/plugins/cmake/cmakeutils.cpp b/plugins/cmake/cmakeutils.cpp
index 9d6b631..3e5fdf8 100644
--- a/plugins/cmake/cmakeutils.cpp
+++ b/plugins/cmake/cmakeutils.cpp
@@ -165,7 +165,7 @@ namespace CMake
 
 KDevelop::Path::List resolveSystemDirs(KDevelop::IProject* project, const QStringList& dirs)
 {
-    const KDevelop::Path buildDir(CMake::currentBuildDir(project));
+    const KDevelop::Path buildDir(CMake::currentCanonicalBuildDir(project));
     const KDevelop::Path installDir(CMake::currentInstallDir(project));
 
     KDevelop::Path::List newList;
@@ -200,7 +200,7 @@ KDevelop::Path::List resolveSystemDirs(KDevelop::IProject* project, const QStrin
 bool checkForNeedingConfigure( KDevelop::IProject* project )
 {
     const QString currentRuntime = ICore::self()->runtimeController()->currentRuntime()->name();
-    const KDevelop::Path builddir = currentBuildDir(project);
+    const KDevelop::Path builddir = currentCanonicalBuildDir(project);
     const bool isValid = (buildDirRuntime(project, -1) == currentRuntime || buildDirRuntime(project, -1).isEmpty()) && builddir.isValid();
 
     if( !isValid )
@@ -208,7 +208,7 @@ bool checkForNeedingConfigure( KDevelop::IProject* project )
         CMakeBuildDirChooser bd;
 
         bd.setProject( project );
-        const auto builddirs = CMake::allBuildDirs(project);
+        const auto builddirs = CMake::allCanonicalBuildDirs(project);
         bd.setAlreadyUsed( builddirs );
         bd.setShowAvailableBuildDirs(!builddirs.isEmpty());
         bd.setCMakeExecutable(currentCMakeExecutable(project));
@@ -298,6 +298,12 @@ KDevelop::Path currentBuildDir( KDevelop::IProject* project, int builddir )
     return KDevelop::Path(readBuildDirParameter( project, Config::Specific::buildDirPathKey, QString(), builddir ));
 }
 
+KDevelop::Path currentCanonicalBuildDir( KDevelop::IProject* project, int builddir )
+{
+    const QString buildDir(readBuildDirParameter( project, Config::Specific::buildDirPathKey, QString(), builddir ));
+    return KDevelop::Path(QFileInfo(buildDir).canonicalFilePath());
+}
+
 KDevelop::Path commandsFile(KDevelop::IProject* project)
 {
     auto currentBuildDir = CMake::currentBuildDir(project);
@@ -629,6 +635,17 @@ QStringList allBuildDirs(KDevelop::IProject* project)
     return result;
 }
 
+QStringList allCanonicalBuildDirs(KDevelop::IProject* project)
+{
+    QStringList result;
+    int bdCount = buildDirCount(project);
+    for (int i = 0; i < bdCount; ++i) {
+        QString buildDir = buildDirGroup( project, i ).readEntry( Config::Specific::buildDirPathKey );
+        result += QFileInfo(buildDir).canonicalFilePath();
+    }
+    return result;
+}
+
 QString executeProcess(const QString& execName, const QStringList& args)
 {
     Q_ASSERT(!execName.isEmpty());
diff --git a/plugins/cmake/cmakeutils.h b/plugins/cmake/cmakeutils.h
index 3642b94..06a1ce1 100644
--- a/plugins/cmake/cmakeutils.h
+++ b/plugins/cmake/cmakeutils.h
@@ -89,6 +89,12 @@ namespace CMake
     KDEVCMAKECOMMON_EXPORT KDevelop::Path currentBuildDir( KDevelop::IProject* project, int builddir = -1 );
 
     /**
+     * @returns the current canonical builddir for the given project or an empty url if none
+     * has been set by the user. See QFileInfo::canonicalFilePath()
+     */
+    KDEVCMAKECOMMON_EXPORT KDevelop::Path currentCanonicalBuildDir( KDevelop::IProject* project, int builddir = -1 );
+
+    /**
      * @returns the path to the 'compile_commands.json' file in the current builddir for the given project
      * or an empty url if none has been set by the user.
      */
@@ -191,6 +197,11 @@ namespace CMake
     KDEVCMAKECOMMON_EXPORT QStringList allBuildDirs(KDevelop::IProject* project);
 
     /**
+     * Retrieves the canonical form of the configured build directories for @p project.
+     */
+    KDEVCMAKECOMMON_EXPORT QStringList allCanonicalBuildDirs(KDevelop::IProject* project);
+
+    /**
      * Attempts to migrate the CMake configuration to per-builddir format.
      * Silently returns if the migration has already been performed.
      */
diff --git a/plugins/cmake/settings/cmakepreferences.cpp b/plugins/cmake/settings/cmakepreferences.cpp
index f2bcb81..9fb8209 100644
--- a/plugins/cmake/settings/cmakepreferences.cpp
+++ b/plugins/cmake/settings/cmakepreferences.cpp
@@ -336,7 +336,7 @@ void CMakePreferences::removeBuildDir()
     if(curr < 0)
         return;
 
-    Path removedPath = CMake::currentBuildDir( m_project );
+    Path removedPath = CMake::currentCanonicalBuildDir( m_project );
     QString removed = removedPath.toLocalFile();
     if(QDir(removed).exists())
     {
