diff --git kdevplatform/vcs/vcspluginhelper.cpp kdevplatform/vcs/vcspluginhelper.cpp
index 4c01c353e0a53c3ce70a5ea2ec5ae8aa97b4b08a..3a93a0f20733a28419073c7e22a2d3340507bdc9 100644
--- kdevplatform/vcs/vcspluginhelper.cpp
+++ kdevplatform/vcs/vcspluginhelper.cpp
@@ -185,7 +185,9 @@ void VcsPluginHelper::disposeEventually(KTextEditor::Document *)
 
 void VcsPluginHelper::setupFromContext(Context* context)
 {
-    d->ctxUrls = context->urls();
+    if (context) {
+        d->ctxUrls = context->urls();
+    }
 }
 
 QList<QUrl> VcsPluginHelper::contextUrlList() const
diff --git kdevplatform/vcs/widgets/vcsdiffpatchsources.cpp kdevplatform/vcs/widgets/vcsdiffpatchsources.cpp
index 80a4212025dcfd82d11ae1399ee58c671cccb1fa..0b9b09377f5dfde56ddc02864eeb0d8b1901d500 100644
--- kdevplatform/vcs/widgets/vcsdiffpatchsources.cpp
+++ kdevplatform/vcs/widgets/vcsdiffpatchsources.cpp
@@ -160,6 +160,7 @@ VCSDiffPatchSource::~VCSDiffPatchSource()
 {
     QFile::remove(m_file.toLocalFile());
     delete m_updater;
+    qDebug() << "~VCSDiffPatchSource()" << this;
 }
 
 QUrl VCSDiffPatchSource::baseDir() const {
@@ -214,6 +215,10 @@ void VCSDiffPatchSource::update() {
 
 VCSCommitDiffPatchSource::~VCSCommitDiffPatchSource() {
     delete m_commitMessageWidget.data();
+//     if (m_commitMessageWidget.data()) {
+//         m_commitMessageWidget.data()->deleteLater();
+//     }
+    qDebug() << "~VCSCommitDiffPatchSource()" << this;
 }
 
 bool VCSCommitDiffPatchSource::canSelectFiles() const {
diff --git plugins/patchreview/patchreview.cpp plugins/patchreview/patchreview.cpp
index 57e1d9dbb2cf90fa9d0ed2d40b7041b15a00a3d6..9d65b6fadddd6e4475c9afb2c0ea600107eb30f9 100644
--- plugins/patchreview/patchreview.cpp
+++ plugins/patchreview/patchreview.cpp
@@ -598,6 +622,9 @@ KDevelop::ContextMenuExtension PatchReviewPlugin::contextMenuExtension(KDevelop:
         urls << econtext->url();
     }
 
+#ifndef Q_OS_MACOS
+    // don't on Mac for now, using the git/show_diffs or git/commit actions after using this 
+    // cause a crash.
     if (urls.size() == 1) {
         QAction* reviewAction = new QAction( QIcon::fromTheme(QStringLiteral("text-x-patch")),
                                              i18n("Review Patch"), parent);
@@ -607,6 +634,7 @@ KDevelop::ContextMenuExtension PatchReviewPlugin::contextMenuExtension(KDevelop:
         cm.addAction( KDevelop::ContextMenuExtension::VcsGroup, reviewAction );
         return cm;
     }
+#endif
 
     return KDevelop::IPlugin::contextMenuExtension(context, parent);
 }
