diff --git kdevplatform/interfaces/idocumentation.h kdevplatform/interfaces/idocumentation.h
index 3add12403520ca1d65ebca7bb4df3be93d9b4ead..c210c4c37a5838ecbb116acee777301b527daf3d 100644
--- kdevplatform/interfaces/idocumentation.h
+++ kdevplatform/interfaces/idocumentation.h
@@ -61,6 +61,13 @@ public:
 
     virtual IDocumentationProvider* provider() const = 0;
 
+    /** Documentation can override this method if it supports using an external browser.
+     * The method should trigger the external browser and return true in that case,
+     * or otherwise return false. When using an external browser the documentation
+     * toolview is not sollicited at all (i.e. it will not open nor be added to a toolbar).
+     */
+    virtual bool viewInExternalBrowser() { return false; }
+
 Q_SIGNALS:
     void descriptionChanged();
 };
diff --git kdevplatform/shell/documentationcontroller.cpp kdevplatform/shell/documentationcontroller.cpp
index 242e90911a200aa4b27aa7d82587ad83dfc17266..7d19098367cea5bc1729bc3c08e70e587cda8638 100644
--- kdevplatform/shell/documentationcontroller.cpp
+++ kdevplatform/shell/documentationcontroller.cpp
@@ -222,18 +222,20 @@ QList< IDocumentationProvider* > DocumentationController::documentationProviders
 
 void KDevelop::DocumentationController::showDocumentation(const IDocumentation::Ptr& doc)
 {
-    QWidget* w = ICore::self()->uiController()->findToolView(i18n("Documentation"), m_factory, KDevelop::IUiController::CreateAndRaise);
-    if(!w) {
-        qCWarning(SHELL) << "Could not add documentation toolview";
-        return;
-    }
+    if (!doc->viewInExternalBrowser()) {
+        QWidget* w = ICore::self()->uiController()->findToolView(i18n("Documentation"), m_factory, KDevelop::IUiController::CreateAndRaise);
+        if(!w) {
+            qCWarning(SHELL) << "Could not add documentation toolview";
+            return;
+        }
 
-    DocumentationView* view = dynamic_cast<DocumentationView*>(w);
-    if( !view ) {
-        qCWarning(SHELL) << "Could not cast toolview" << w << "to DocumentationView class!";
-        return;
+        DocumentationView* view = dynamic_cast<DocumentationView*>(w);
+        if( !view ) {
+            qCWarning(SHELL) << "Could not cast toolview" << w << "to DocumentationView class!";
+            return;
+        }
+        view->showDocumentation(doc);
     }
-    view->showDocumentation(doc);
 }
 
 void DocumentationController::changedDocumentationProviders()
diff --git plugins/qthelp/CMakeLists.txt plugins/qthelp/CMakeLists.txt
index 5a8e6e2d1d3677fd590983d2144c10004fffa184..42f111ac52fd98b5e29d75e861f1a9b95d9c0d20 100644
--- plugins/qthelp/CMakeLists.txt
+++ plugins/qthelp/CMakeLists.txt
@@ -10,12 +10,18 @@ set(kdevqthelp_SRCS
     qthelpproviderabstract.cpp
     qthelpprovider.cpp
     qthelpdocumentation.cpp
+    qthelpexternalassistant.cpp
     qthelpqtdoc.cpp
     qthelp_config_shared.cpp
     qthelpconfig.cpp # Configuration module for QtHelp plugin
     qthelpnetwork.cpp
     ${kdevqthelp_LOG_SRCS}
 )
+if(APPLE)
+    set(kdevqthelp_SRCS ${kdevqthelp_SRCS}
+        qthelpexternalassistant_mac.mm
+    )
+endif()
 
 ki18n_wrap_ui(kdevqthelp_SRCS
     qthelpconfig.ui
@@ -27,6 +33,9 @@ kdevplatform_add_plugin(kdevqthelp JSON kdevqthelp.json SOURCES ${kdevqthelp_SRC
 target_link_libraries(kdevqthelp
     KF5::KCMUtils KF5::I18n KF5::KIOWidgets KF5::TextEditor KF5::IconThemes Qt5::Help KF5::NewStuff
     KDev::Language KDev::Documentation KDev::Interfaces)
+if(APPLE)
+    target_link_libraries(kdevqthelp "-framework AppKit")
+endif()
 
 if(BUILD_TESTING)
     add_subdirectory(tests)
diff --git plugins/qthelp/qthelp_config_shared.cpp plugins/qthelp/qthelp_config_shared.cpp
index 16fa12105e24446255b81efe41fa4195ff02aafe..ed35c72b476e21a10bacf50585ca583d66592ec0 100644
--- plugins/qthelp/qthelp_config_shared.cpp
+++ plugins/qthelp/qthelp_config_shared.cpp
@@ -1,6 +1,7 @@
 /*  This file is part of KDevelop
 
     Copyright 2010 Milian Wolff <mail@milianw.de>
+    Copyright 2017 René J.V. Bertin <rjvbertin@gmail.com>
 
     This library is free software; you can redistribute it and/or
     modify it under the terms of the GNU Library General Public
@@ -25,7 +26,8 @@
 
 void qtHelpReadConfig(QStringList& iconList, QStringList& nameList,
                       QStringList& pathList, QStringList& ghnsList,
-                      QString& searchDir, bool& loadQtDoc)
+                      QString& searchDir, bool& loadQtDoc,
+                      ExternalViewerSettings& extViewer)
 {
     KConfigGroup cg(KSharedConfig::openConfig(), "QtHelp Documentation");
     iconList = cg.readEntry("iconList", QStringList());
@@ -34,11 +36,14 @@ void qtHelpReadConfig(QStringList& iconList, QStringList& nameList,
     ghnsList = cg.readEntry("ghnsList", QStringList());
     searchDir = cg.readEntry("searchDir", QString());
     loadQtDoc = cg.readEntry("loadQtDocs", true);
+    extViewer.useExtViewer = cg.readEntry("useExternalViewer", false);
+    extViewer.extViewerExecutable = cg.readEntry("externalViewerExecutable", QString());
 }
 
 void qtHelpWriteConfig(const QStringList& iconList, const QStringList& nameList,
                        const QStringList& pathList, const QStringList& ghnsList,
-                       const QString& searchDir, const bool loadQtDoc)
+                       const QString& searchDir, const bool loadQtDoc,
+                       const ExternalViewerSettings& extViewer)
 {
     KConfigGroup cg(KSharedConfig::openConfig(), "QtHelp Documentation");
     cg.writeEntry("iconList", iconList);
@@ -47,4 +52,6 @@ void qtHelpWriteConfig(const QStringList& iconList, const QStringList& nameList,
     cg.writeEntry("ghnsList", ghnsList);
     cg.writeEntry("searchDir", searchDir);
     cg.writeEntry("loadQtDocs", loadQtDoc);
+    cg.writeEntry("useExternalViewer", extViewer.useExtViewer);
+    cg.writeEntry("externalViewerExecutable", extViewer.extViewerExecutable);
 }
diff --git plugins/qthelp/qthelp_config_shared.h plugins/qthelp/qthelp_config_shared.h
index dc3cd684ba92d77c2a014c344ea9f85d86c13b29..1f8c4fc7623c6a14aaa86297280c2e844b9c4ae2 100644
--- plugins/qthelp/qthelp_config_shared.h
+++ plugins/qthelp/qthelp_config_shared.h
@@ -23,18 +23,27 @@
 
 #include <QStringList>
 
+typedef struct ExternalViewerSettings {
+    bool useExtViewer;
+    QString extViewerExecutable;
+} ExternalViewerSettings;
+
 void qtHelpReadConfig(QStringList& iconList,
                       QStringList& nameList,
                       QStringList& pathList,
                       QStringList& ghnsList,
                       QString& searchDir,
-                      bool& loadQtDoc);
+                      bool& loadQtDoc,
+                      ExternalViewerSettings& extViewer
+                     );
 
 void qtHelpWriteConfig(const QStringList& iconList,
                        const QStringList& nameList,
                        const QStringList& pathList,
                        const QStringList& ghnsList,
                        const QString& searchDir,
-                       const bool loadQtDoc);
+                       const bool loadQtDoc,
+                       const ExternalViewerSettings& extViewer
+                      );
 
 #endif // QTHELP_CONFIG_SHARED_H
diff --git plugins/qthelp/qthelpconfig.cpp plugins/qthelp/qthelpconfig.cpp
index 10bc17df5ce145f932fae1539943b75e62b41616..a0b37be8474c565573a1595503d3644fe4f0a6ce 100644
--- plugins/qthelp/qthelpconfig.cpp
+++ plugins/qthelp/qthelpconfig.cpp
@@ -36,6 +36,7 @@
 #include "qthelp_config_shared.h"
 #include "debug.h"
 #include "qthelpplugin.h"
+#include "qthelpexternalassistant.h"
 
 enum Column
 {
@@ -119,6 +120,27 @@ QtHelpConfig::QtHelpConfig(QtHelpPlugin* plugin, QWidget *parent)
     m_configWidget->qchSearchDir->setMode(KFile::Directory);
     connect(m_configWidget->qchSearchDir, &KUrlRequester::textChanged,
             this, &QtHelpConfig::changed);
+    connect(m_configWidget->externalViewerCheckBox, &QCheckBox::toggled,
+            this, static_cast<void(QtHelpConfig::*)()>(&QtHelpConfig::changed));
+    m_configWidget->externalViewerCheckBox->setToolTip(i18n("Use Qt's Assistant as an external viewer,\n"
+        "instead of the embedded viewer."));
+    m_configWidget->externalViewerExecutable->setToolTip(
+        i18n("The path to the Assistant copy to use as an external viewer,\n"
+            "can also be a script calling the Assistant or any other\n"
+            "application that understands the same remote control protocol.\n"
+            "KDevelop will try to find the application when this is not set.\n"
+            "It is your responsibility to ensure that the viewer is configured\n"
+            "to use a qhc help collection matching KDevelop's collection!"));
+    QString currentExtViewer;
+    if (KDevelop::QtHelpExternalAssistant::hasExternalViewer(&currentExtViewer)) {
+        m_configWidget->externalViewerExecutable->setPlaceholderText(
+            i18n("path to Qt Assistant (%1)", currentExtViewer));
+    } else {
+        m_configWidget->externalViewerExecutable->setPlaceholderText(
+            i18n("path to Qt Assistant"));
+    }
+    connect(m_configWidget->externalViewerExecutable, &KUrlRequester::textChanged,
+            this, &QtHelpConfig::changed);
 
     // Set availability information for QtHelp
     m_configWidget->messageAvailabilityQtDocs->setCloseButtonVisible(false);
@@ -128,6 +150,8 @@ QtHelpConfig::QtHelpConfig(QtHelpPlugin* plugin, QWidget *parent)
         m_configWidget->messageAvailabilityQtDocs->setText(
             i18n("The command \"qmake -query\" could not provide a path to a QtHelp file (QCH)."));
         m_configWidget->loadQtDocsCheckBox->setVisible(false);
+        m_configWidget->externalViewerCheckBox->setVisible(false);
+        m_configWidget->externalViewerExecutable->setVisible(false);
     }
     reset();
 }
@@ -154,8 +178,10 @@ void QtHelpConfig::apply()
     }
     QString searchDir = m_configWidget->qchSearchDir->text();
     bool loadQtDoc = m_configWidget->loadQtDocsCheckBox->isChecked();
+    ExternalViewerSettings extViewer = { m_configWidget->externalViewerCheckBox->isChecked(),
+        m_configWidget->externalViewerExecutable->text() };
 
-    qtHelpWriteConfig(iconList, nameList, pathList, ghnsList, searchDir, loadQtDoc);
+    qtHelpWriteConfig(iconList, nameList, pathList, ghnsList, searchDir, loadQtDoc, extViewer);
     static_cast<QtHelpPlugin*>(plugin())->readConfig();
 }
 
@@ -166,7 +192,8 @@ void QtHelpConfig::reset()
     QStringList iconList, nameList, pathList, ghnsList;
     QString searchDir;
     bool loadQtDoc;
-    qtHelpReadConfig(iconList, nameList, pathList, ghnsList, searchDir, loadQtDoc);
+    ExternalViewerSettings extViewer;
+    qtHelpReadConfig(iconList, nameList, pathList, ghnsList, searchDir, loadQtDoc, extViewer);
 
     const int size = qMin(qMin(iconList.size(), nameList.size()), pathList.size());
     for(int i = 0; i < size; ++i) {
@@ -175,6 +202,8 @@ void QtHelpConfig::reset()
     }
     m_configWidget->qchSearchDir->setText(searchDir);
     m_configWidget->loadQtDocsCheckBox->setChecked(loadQtDoc);
+    m_configWidget->externalViewerCheckBox->setChecked(extViewer.useExtViewer);
+    m_configWidget->externalViewerExecutable->setText(extViewer.extViewerExecutable);
 
     emit changed();
 }
diff --git plugins/qthelp/qthelpconfig.ui plugins/qthelp/qthelpconfig.ui
index bc04f8075468a618c08448d818c70734d67c3252..b3e3cfed64382f57b1b2325ddde3d248ada3d479 100644
--- plugins/qthelp/qthelpconfig.ui
+++ plugins/qthelp/qthelpconfig.ui
@@ -14,7 +14,16 @@
    <locale language="English" country="UnitedStates"/>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout_2">
-   <property name="margin">
+   <property name="leftMargin">
+    <number>0</number>
+   </property>
+   <property name="topMargin">
+    <number>0</number>
+   </property>
+   <property name="rightMargin">
+    <number>0</number>
+   </property>
+   <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
@@ -48,14 +57,14 @@
        </widget>
       </item>
       <item row="1" column="1">
-       <widget class="KUrlRequester" name="qchSearchDir" native="true"/>
+       <widget class="KUrlRequester" name="qchSearchDir"/>
       </item>
       <item row="2" column="0" colspan="2">
-       <widget class="KMessageWidget" name="messageAvailabilityQtDocs" native="true">
+       <widget class="KMessageWidget" name="messageAvailabilityQtDocs">
         <property name="enabled">
          <bool>true</bool>
         </property>
-        <property name="text" stdset="0">
+        <property name="text">
          <string/>
         </property>
        </widget>
@@ -165,6 +174,48 @@
     </widget>
    </item>
    <item>
+    <widget class="QGroupBox" name="boxAutoLoad2">
+     <layout class="QFormLayout" name="formLayout2">
+      <property name="fieldGrowthPolicy">
+       <enum>QFormLayout::ExpandingFieldsGrow</enum>
+      </property>
+      <property name="formAlignment">
+       <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
+      </property>
+      <property name="topMargin">
+       <number>3</number>
+      </property>
+      <property name="bottomMargin">
+       <number>3</number>
+      </property>
+      <item row="2" column="1">
+       <widget class="KUrlRequester" name="externalViewerExecutable">
+        <property name="mode">
+         <set>KFile::ExistingOnly|KFile::File|KFile::LocalOnly</set>
+        </property>
+       </widget>
+      </item>
+      <item row="2" column="0">
+       <widget class="QCheckBox" name="externalViewerCheckBox">
+        <property name="text">
+         <string/>
+        </property>
+       </widget>
+      </item>
+      <item row="1" column="0">
+       <widget class="QLabel" name="labelExternalViewer">
+        <property name="text">
+         <string>Use &amp;external viewer:</string>
+        </property>
+        <property name="buddy">
+         <cstring>externalViewerCheckBox</cstring>
+        </property>
+       </widget>
+      </item>
+     </layout>
+    </widget>
+   </item>
+   <item>
     <spacer name="verticalSpacer">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
@@ -181,16 +232,16 @@
  </widget>
  <customwidgets>
   <customwidget>
-   <class>KMessageWidget</class>
-   <extends>QFrame</extends>
-   <header location="global">kmessagewidget.h</header>
-  </customwidget>
-  <customwidget>
    <class>KUrlRequester</class>
    <extends>QWidget</extends>
-   <header location="global">kurlrequester.h</header>
+   <header>kurlrequester.h</header>
    <container>1</container>
   </customwidget>
+  <customwidget>
+   <class>KMessageWidget</class>
+   <extends>QFrame</extends>
+   <header>kmessagewidget.h</header>
+  </customwidget>
  </customwidgets>
  <resources/>
  <connections/>
diff --git plugins/qthelp/qthelpdocumentation.cpp plugins/qthelp/qthelpdocumentation.cpp
index 19613339d01531b1f7716a9a8e0c2f212bdc856c..10a38194cc3c5aeea703c10cce444ea58ad3191b 100644
--- plugins/qthelp/qthelpdocumentation.cpp
+++ plugins/qthelp/qthelpdocumentation.cpp
@@ -37,6 +37,7 @@
 #include <documentation/standarddocumentationview.h>
 #include "qthelpnetwork.h"
 #include "qthelpproviderabstract.h"
+#include "qthelpexternalassistant.h"
 
 using namespace KDevelop;
 
@@ -300,6 +301,14 @@ IDocumentationProvider* QtHelpDocumentation::provider() const
     return m_provider;
 }
 
+bool QtHelpDocumentation::viewInExternalBrowser()
+{
+    if (QtHelpExternalAssistant::useExternalViewer()) {
+        return QtHelpExternalAssistant::openUrl(m_current.value());
+    }
+    return false;
+}
+
 QtHelpAlternativeLink::QtHelpAlternativeLink(const QString& name, const QtHelpDocumentation* doc, QObject* parent)
     : QAction(name, parent), mDoc(doc), mName(name)
 {
diff --git plugins/qthelp/qthelpdocumentation.h plugins/qthelp/qthelpdocumentation.h
index e649bf79b864b20fe3afa58fcc4db54ef36f338e..ee6a377486ca281dc7fc8c942061cb0bf3ff2640 100644
--- plugins/qthelp/qthelpdocumentation.h
+++ plugins/qthelp/qthelpdocumentation.h
@@ -54,6 +54,8 @@ class QtHelpDocumentation : public KDevelop::IDocumentation
 
         static QtHelpProviderAbstract* s_provider;
 
+        bool viewInExternalBrowser() override;
+
     public Q_SLOTS:
         void viewContextMenuRequested(const QPoint& pos);
 
diff --git plugins/qthelp/qthelpexternalassistant.cpp plugins/qthelp/qthelpexternalassistant.cpp
new file mode 100644
index 0000000000000000000000000000000000000000..b2ee8044b1d6b22b2fbace238872e813f19c0c8a
--- /dev/null
+++ plugins/qthelp/qthelpexternalassistant.cpp
@@ -0,0 +1,162 @@
+/*  This file is part of KDevelop
+    Copyright 2017 René J.V. Bertin <rjvbertin@gmail.com>
+
+    This library is free software; you can redistribute it and/or
+    modify it under the terms of the GNU Library General Public
+    License as published by the Free Software Foundation; either
+    version 2 of the License, or (at your option) any later version.
+
+    This library is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+    Library General Public License for more details.
+
+    You should have received a copy of the GNU Library General Public License
+    along with this library; see the file COPYING.LIB.  If not, write to
+    the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+    Boston, MA 02110-1301, USA.
+*/
+
+#include "qthelpexternalassistant.h"
+
+#include <QString>
+#include <QUrl>
+#include <QStandardPaths>
+#include <QProcess>
+#include <QApplication>
+#include <QFileInfo>
+
+#include <interfaces/icore.h>
+
+#include "debug.h"
+
+using namespace KDevelop;
+
+QtHelpExternalAssistant* QtHelpExternalAssistant::s_instance = nullptr;
+QProcess* QtHelpExternalAssistant::s_externalViewerProcess = nullptr;
+bool QtHelpExternalAssistant::s_useExternalViewer = false;
+QString QtHelpExternalAssistant::s_externalViewerExecutable;
+
+QtHelpExternalAssistant::QtHelpExternalAssistant(QObject *parent)
+    : QObject(parent)
+{
+}
+
+QtHelpExternalAssistant::~QtHelpExternalAssistant()
+{
+    if (s_externalViewerProcess) {
+        // stop monitoring the process. Prevents potential warnings
+        // (QCoreApplication::postEvent: Unexpected null receiver)
+        s_externalViewerProcess->disconnect();
+#ifdef Q_OS_UNIX
+        qint64 pid = s_externalViewerProcess->processId();
+        if (pid > 0) {
+            QProcess* hup = new QProcess();
+            hup->startDetached(QString::fromLatin1("kill -1 %1").arg(pid));
+            hup->waitForFinished(50);
+            hup->deleteLater();
+        } else
+#endif
+            s_externalViewerProcess->terminate();
+        // give the process a bit of time to react, just to prevent
+        // "QProcess destroyed while process still running" warnings.
+        s_externalViewerProcess->waitForFinished(50);
+        s_externalViewerProcess->deleteLater();
+        s_externalViewerProcess = nullptr;
+    }
+    if (this == s_instance) {
+        s_instance = nullptr;
+    }
+}
+
+void QtHelpExternalAssistant::externalViewerExit(int exitCode, QProcess::ExitStatus exitStatus)
+{
+    if (this == s_instance) {
+        qCDebug(QTHELP) << "externalViewer" << this << "has exited with code"
+            << exitCode << "and status" << exitStatus;
+        deleteLater();
+        s_externalViewerProcess = nullptr;
+    }
+}
+
+void QtHelpExternalAssistant::setUseExternalViewer(bool extViewer)
+{
+    s_useExternalViewer = extViewer;
+}
+
+void QtHelpExternalAssistant::setExternalViewerExecutable(QString& path)
+{
+    s_externalViewerExecutable = path;
+}
+
+#ifndef Q_OS_MACOS
+bool QtHelpExternalAssistant::hasExternalViewer(QString* path)
+{
+    bool ret = false;
+    if (!s_externalViewerExecutable.isEmpty()) {
+        QFileInfo info(s_externalViewerExecutable);
+        if (info.isExecutable()) {
+            ret = true;
+        } else {
+            qCWarning(QTHELP) << "External viewer" << s_externalViewerExecutable << "is not executable!";
+        }
+        // always return the configured executable path
+        if (path) {
+            *path = s_externalViewerExecutable;
+        }
+    } else if (path) {
+        // look for the application on the current path. The result is not cached so the
+        // lookup is performed only when we can return the result.
+        const QString assistant = QStandardPaths::findExecutable(QLatin1String("assistant"));
+        if (!assistant.isEmpty()) {
+            ret = true;
+            *path = assistant;
+            qCDebug(QTHELP) << "External viewer found at" << assistant;
+        } else {
+            qCWarning(QTHELP) << "External viewer not found!";
+        }
+    }
+    return ret;
+}
+#endif
+
+QtHelpExternalAssistant* QtHelpExternalAssistant::instance()
+{
+    QString path;
+
+    if (ICore::self()->shuttingDown()) {
+        // don't create a new instance if we're shutting down.
+        return s_instance;
+    }
+
+    if (!s_externalViewerProcess && hasExternalViewer(&path)) {
+        if (!s_instance) {
+            s_instance = new QtHelpExternalAssistant(qApp);
+        }
+        s_externalViewerProcess = new QProcess();
+        connect(s_externalViewerProcess, static_cast<void(QProcess::*)(int, QProcess::ExitStatus)>(&QProcess::finished),
+                s_instance, &QtHelpExternalAssistant::externalViewerExit);
+        QStringList args = {"-enableRemoteControl"};
+        s_externalViewerProcess->start(path, args, QIODevice::WriteOnly|QIODevice::Append);
+        if (!s_externalViewerProcess->waitForStarted()) {
+            s_externalViewerProcess->deleteLater();
+            s_externalViewerProcess = nullptr;
+        }
+    }
+    return s_instance;
+}
+
+bool QtHelpExternalAssistant::openUrl(const QUrl& url)
+{
+    if (instance()) {
+        const QByteArray command = QByteArrayLiteral("setSource ")
+            + url.toString().toUtf8() + QByteArrayLiteral("\n");
+        if (s_externalViewerProcess->write(command) < 0
+            || s_externalViewerProcess->write("show contents\n") < 0) {
+            return false;
+        }
+        s_externalViewerProcess->write("syncContents\n");
+        return true;
+    }
+    return false;
+}
diff --git plugins/qthelp/qthelpexternalassistant.h plugins/qthelp/qthelpexternalassistant.h
new file mode 100644
index 0000000000000000000000000000000000000000..af32f42249592064ac507e8f3ba2aba83623d15e
--- /dev/null
+++ plugins/qthelp/qthelpexternalassistant.h
@@ -0,0 +1,92 @@
+/*  This file is part of KDevelop
+    Copyright 2017 René J.V. Bertin <rjvbertin@gmail.com>
+
+    This library is free software; you can redistribute it and/or
+    modify it under the terms of the GNU Library General Public
+    License as published by the Free Software Foundation; either
+    version 2 of the License, or (at your option) any later version.
+
+    This library is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+    Library General Public License for more details.
+
+    You should have received a copy of the GNU Library General Public License
+    along with this library; see the file COPYING.LIB.  If not, write to
+    the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+    Boston, MA 02110-1301, USA.
+*/
+
+#ifndef QTHELPEXTERNALASSISTANT_H
+#define QTHELPEXTERNALASSISTANT_H
+
+#include <QObject>
+#include <QProcess>
+
+#include "debug.h"
+
+class QUrl;
+
+namespace KDevelop{
+
+/**
+ * @brief Provides support for using Qt's Assistant (or compatible applications)
+ * as an external viewer for QtHelp documentation. A single instance will be
+ * launched when required (one external browser per KDevelop session), and
+ * controlled via the Assistant's remote control protocol. The application will
+ * be terminated when KDevelop exits.
+ */
+class QtHelpExternalAssistant : public QObject
+{
+    Q_OBJECT
+public:
+    QtHelpExternalAssistant(QObject* parent);
+    ~QtHelpExternalAssistant();
+
+    /**
+     * @brief returns the current external viewer instance,
+     * creating and initialising it if necessary.
+     */
+    static QtHelpExternalAssistant* instance();
+
+    /**
+     * @brief open @param url in the external viewer,
+     * launching it if required. No attempt is made to raise
+     * or unhide the viewer, beyond what the viewer does itself.
+     * @returns true when the command is transmitted successfully.
+     */
+    static bool openUrl(const QUrl& url);
+
+    /**
+     * @brief enable or disable use of the external viewer
+     */
+    static void setUseExternalViewer(bool extViewer);
+    static bool useExternalViewer() { return s_useExternalViewer; }
+
+    /**
+     * @brief sets the @param path to the external viewer executable.
+     */
+    static void setExternalViewerExecutable(QString& path);
+    /**
+     * @brief checks if a valid external viewer executable has been
+     * configured; @returns true if this is the case and links can thus
+     * be opened in this viewer instead of in the embedded browser.
+     * 
+     * @param path can be used to obtain the configured path or the
+     * path to the Assistant copy found automatically.
+     */
+    static bool hasExternalViewer(QString* path = nullptr);
+
+private Q_SLOTS:
+    void externalViewerExit(int exitCode, QProcess::ExitStatus exitStatus);
+
+private:
+    static QtHelpExternalAssistant* s_instance;
+    static QProcess* s_externalViewerProcess;
+    static bool s_useExternalViewer;
+    static QString s_externalViewerExecutable;
+};
+
+}
+
+#endif //QTHELPEXTERNALASSISTANT_H
diff --git plugins/qthelp/qthelpexternalassistant_mac.mm plugins/qthelp/qthelpexternalassistant_mac.mm
new file mode 100644
index 0000000000000000000000000000000000000000..c787425d7278169965d56098575901e26d14fd3a
--- /dev/null
+++ plugins/qthelp/qthelpexternalassistant_mac.mm
@@ -0,0 +1,92 @@
+/*  This file is part of KDevelop
+    Copyright 2017 René J.V. Bertin <rjvbertin@gmail.com>
+
+    This library is free software; you can redistribute it and/or
+    modify it under the terms of the GNU Library General Public
+    License as published by the Free Software Foundation; either
+    version 2 of the License, or (at your option) any later version.
+
+    This library is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+    Library General Public License for more details.
+
+    You should have received a copy of the GNU Library General Public License
+    along with this library; see the file COPYING.LIB.  If not, write to
+    the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+    Boston, MA 02110-1301, USA.
+*/
+
+#include "qthelpexternalassistant.h"
+
+#include <QString>
+#include <QUrl>
+#include <QStandardPaths>
+#include <QFileInfo>
+
+#include <AppKit/AppKit.h>
+
+#include "debug.h"
+
+using namespace KDevelop;
+
+bool QtHelpExternalAssistant::hasExternalViewer(QString* path)
+{
+    bool ret = false;
+    QString execPath;
+    QFileInfo info;
+
+    if (!s_externalViewerExecutable.isEmpty()) {
+        execPath = s_externalViewerExecutable;
+        info = QFileInfo(execPath);
+        if (info.isBundle()) {
+            NSString* bundleExec = [[NSBundle bundleWithPath:s_externalViewerExecutable.toNSString()] executablePath];
+            if (bundleExec) {
+                execPath = QString::fromNSString(bundleExec);
+                info = QFileInfo(execPath);
+                qCDebug(QTHELP) << "Found bundle exec" << execPath << "for external viewer" << s_externalViewerExecutable;
+            } else {
+                qCWarning(QTHELP) << "Could not find the bundle exec for external viewer" << s_externalViewerExecutable;
+            }
+        }
+        if (info.isExecutable()) {
+            ret = true;
+        } else {
+            qCWarning(QTHELP) << "External viewer" << execPath << "is not executable!";
+        }
+        // always return the configured executable path
+        if (path) {
+            *path = execPath;
+        }
+    } else if (path) {
+        // look for the application on the current path. The result is not cached so the
+        // lookup is performed only when we can return the result.
+        // First try to find a regular executable, which the user could have created
+        // to invoke the right Assistant.app
+        const QString assistant = QStandardPaths::findExecutable(QLatin1String("assistant"));
+        if (!assistant.isEmpty()) {
+            ret = true;
+            *path = assistant;
+            qCDebug(QTHELP) << "External viewer found at" << assistant;
+        } else {
+            // try to locate a Qt5 Assistant app bundle
+            // There is no way to predict which we'll get beyond "probably the same as last time".
+            NSString *assistantAppBundle = [[NSWorkspace sharedWorkspace]
+                absolutePathForAppBundleWithIdentifier:@"org.qt-project.assistant"];
+            if (!assistantAppBundle) {
+                // fall back to finding whatever Assistant.app application might be available.
+                assistantAppBundle = [[NSWorkspace sharedWorkspace] fullPathForApplication:@"Assistant.app"];
+            }
+            NSString* assistantExec = assistantAppBundle ? [[NSBundle bundleWithPath:assistantAppBundle] executablePath] : nullptr;
+            if (assistantExec) {
+                execPath = QString::fromNSString(assistantExec);
+                ret = true;
+                *path = execPath;
+                qCDebug(QTHELP) << "Found bundle exec" << execPath << "for Assistant.app" << QString::fromNSString(assistantAppBundle);
+            } else {
+                qCWarning(QTHELP) << "External viewer not found!";
+            }
+        }
+    }
+    return ret;
+}
diff --git plugins/qthelp/qthelpplugin.cpp plugins/qthelp/qthelpplugin.cpp
index 02103b6a23ff520c813fd9eaa2047652cbc10481..613ac35e8d07b2f07d36ec1cbb4acb9ca9d657dd 100644
--- plugins/qthelp/qthelpplugin.cpp
+++ plugins/qthelp/qthelpplugin.cpp
@@ -26,6 +26,7 @@
 #include <QDirIterator>
 #include "qthelpprovider.h"
 #include "qthelpqtdoc.h"
+#include "qthelpexternalassistant.h"
 #include "qthelp_config_shared.h"
 #include "debug.h"
 #include "qthelpconfig.h"
@@ -39,6 +40,8 @@ QtHelpPlugin::QtHelpPlugin(QObject* parent, const QVariantList& args)
     , m_qtHelpProviders()
     , m_qtDoc(new QtHelpQtDoc(this, QVariantList()))
     , m_loadSystemQtDoc(false)
+    , m_useExternalViewer(false)
+    , m_externalViewerExecutable(QString())
 {
     Q_UNUSED(args);
     s_plugin = this;
@@ -55,11 +58,16 @@ void QtHelpPlugin::readConfig()
 {
     QStringList iconList, nameList, pathList, ghnsList;
     QString searchDir;
-    qtHelpReadConfig(iconList, nameList, pathList, ghnsList, searchDir, m_loadSystemQtDoc);
+    ExternalViewerSettings extViewer;
+    qtHelpReadConfig(iconList, nameList, pathList, ghnsList, searchDir, m_loadSystemQtDoc, extViewer);
 
     searchHelpDirectory(pathList, nameList, iconList, searchDir);
     loadQtHelpProvider(pathList, nameList, iconList);
     loadQtDocumentation(m_loadSystemQtDoc);
+    m_useExternalViewer = extViewer.useExtViewer;
+    m_externalViewerExecutable = extViewer.extViewerExecutable;
+    KDevelop::QtHelpExternalAssistant::setUseExternalViewer(m_useExternalViewer);
+    KDevelop::QtHelpExternalAssistant::setExternalViewerExecutable(m_externalViewerExecutable);
 
     emit changedProvidersList();
 }
@@ -159,6 +167,11 @@ bool QtHelpPlugin::isQtHelpQtDocLoaded() const
     return m_loadSystemQtDoc;
 }
 
+bool QtHelpPlugin::useExternalViewer() const
+{
+    return m_useExternalViewer;
+}
+
 bool QtHelpPlugin::isQtHelpAvailable() const
 {
     return !m_qtDoc->qchFiles().isEmpty();
diff --git plugins/qthelp/qthelpplugin.h plugins/qthelp/qthelpplugin.h
index 9c76abd40629a60649cdbe5ecb7ff7705d6a49f6..b1efa1d4068374274d3b226b89e088afcd777ef5 100644
--- plugins/qthelp/qthelpplugin.h
+++ plugins/qthelp/qthelpplugin.h
@@ -42,6 +42,8 @@ class QtHelpPlugin : public KDevelop::IPlugin, public KDevelop::IDocumentationPr
         QList<QtHelpProvider*> qtHelpProviderLoaded();
         bool isQtHelpQtDocLoaded() const;
         bool isQtHelpAvailable() const;
+        bool useExternalViewer() const;
+        QString externalViewerExecutable() const;
 
         int configPages() const override;
         KDevelop::ConfigPage* configPage(int number, QWidget* parent) override;
@@ -59,6 +61,8 @@ class QtHelpPlugin : public KDevelop::IPlugin, public KDevelop::IDocumentationPr
         QList<QtHelpProvider*> m_qtHelpProviders;
         QtHelpQtDoc* m_qtDoc;
         bool m_loadSystemQtDoc;
+        bool m_useExternalViewer;
+        QString m_externalViewerExecutable;
 };
 
 #endif // QTHELPPLUGIN_H
diff --git plugins/qthelp/tests/CMakeLists.txt plugins/qthelp/tests/CMakeLists.txt
index 06c751e8a2c254ba175e827fe9036fe635105d4c..bce551d62e5de047b66dcc41652b665258ef758d 100644
--- plugins/qthelp/tests/CMakeLists.txt
+++ plugins/qthelp/tests/CMakeLists.txt
@@ -8,12 +8,18 @@ set(test_qthelpplugin_SRCS
     ../qthelpproviderabstract.cpp
     ../qthelpprovider.cpp
     ../qthelpdocumentation.cpp
+    ../qthelpexternalassistant.cpp
     ../qthelpqtdoc.cpp
     ../qthelp_config_shared.cpp
     ../qthelpconfig.cpp
     ../qthelpnetwork.cpp
     ${kdevqthelp_LOG_SRCS}
 )
+if(APPLE)
+    set(test_qthelpplugin_SRCS ${test_qthelpplugin_SRCS}
+        ../qthelpexternalassistant_mac.mm
+    )
+endif()
 ki18n_wrap_ui(test_qthelpplugin_SRCS
    ../qthelpconfig.ui
    ../qthelpconfigeditdialog.ui
@@ -24,3 +30,6 @@ ecm_add_test(${test_qthelpplugin_SRCS}
     TEST_NAME test_qthelpplugin
     LINK_LIBRARIES Qt5::Test KF5::NewStuff KF5::KIOWidgets KF5::TextEditor KF5::IconThemes Qt5::Help KDev::Tests KDev::Documentation
 )
+if(APPLE)
+    target_link_libraries(test_qthelpplugin "-framework AppKit")
+endif()
diff --git plugins/qthelp/tests/test_qthelpplugin.cpp plugins/qthelp/tests/test_qthelpplugin.cpp
index bb47683da08cf1b6ac5d13dd5d5b1fe83b9eebee..23951a08c46b82f7a7169c5b1b107b6604ff00ce 100644
--- plugins/qthelp/tests/test_qthelpplugin.cpp
+++ plugins/qthelp/tests/test_qthelpplugin.cpp
@@ -58,7 +58,8 @@ void TestQtHelpPlugin::init()
 {
     m_plugin = new QtHelpPlugin(m_testCore, QVariantList());
     // write default config and read it
-    qtHelpWriteConfig(QStringList(), QStringList(), QStringList(), QStringList(), QString(), true);
+    ExternalViewerSettings extView = {false, QString()};
+    qtHelpWriteConfig(QStringList(), QStringList(), QStringList(), QStringList(), QString(), true, extView);
     m_plugin->readConfig();
 }
 
@@ -82,7 +83,8 @@ void TestQtHelpPlugin::testDefaultValue()
 
 void TestQtHelpPlugin::testUnsetQtHelpDoc()
 {
-    qtHelpWriteConfig(QStringList(), QStringList(), QStringList(), QStringList(), QString(), false);
+    ExternalViewerSettings extView = {false, QString()};
+    qtHelpWriteConfig(QStringList(), QStringList(), QStringList(), QStringList(), QString(), false, extView);
     m_plugin->readConfig();
 
     QCOMPARE(m_plugin->providers().size(), 0);
@@ -95,7 +97,8 @@ void TestQtHelpPlugin::testAddOneValidProvider()
     name << QStringLiteral("file1");
     icon << QStringLiteral("myIcon");
     ghns << QStringLiteral("0");
-    qtHelpWriteConfig(icon, name, path, ghns, QString(), true);
+    ExternalViewerSettings extView = {false, QString()};
+    qtHelpWriteConfig(icon, name, path, ghns, QString(), true, extView);
     m_plugin->readConfig();
 
     QCOMPARE(m_plugin->qtHelpProviderLoaded().size(), 1);
@@ -111,7 +114,8 @@ void TestQtHelpPlugin::testAddTwoDifferentValidProvider()
     name << QStringLiteral("file1") << QStringLiteral("file2");
     icon << QStringLiteral("myIcon") << QStringLiteral("myIcon");
     ghns << QStringLiteral("0") << QStringLiteral("0");
-    qtHelpWriteConfig(icon, name, path, ghns, QString(), true);
+    ExternalViewerSettings extView = {false, QString()};
+    qtHelpWriteConfig(icon, name, path, ghns, QString(), true, extView);
     m_plugin->readConfig();
 
     QCOMPARE(m_plugin->qtHelpProviderLoaded().size(), 2);
@@ -132,7 +136,8 @@ void TestQtHelpPlugin::testAddInvalidProvider()
     name << QStringLiteral("file1");
     icon << QStringLiteral("myIcon");
     ghns << QStringLiteral("0");
-    qtHelpWriteConfig(icon, name, path, ghns, QString(), true);
+    ExternalViewerSettings extView = {false, QString()};
+    qtHelpWriteConfig(icon, name, path, ghns, QString(), true, extView);
     m_plugin->readConfig();
 
     QCOMPARE(m_plugin->qtHelpProviderLoaded().size(), 0);
@@ -145,7 +150,8 @@ void TestQtHelpPlugin::testAddTwiceSameProvider()
     name << QStringLiteral("file1") << QStringLiteral("file2");
     icon << QStringLiteral("myIcon") << QStringLiteral("myIcon");
     ghns << QStringLiteral("0") << QStringLiteral("0");
-    qtHelpWriteConfig(icon, name, path, ghns, QString(), true);
+    ExternalViewerSettings extView = {false, QString()};
+    qtHelpWriteConfig(icon, name, path, ghns, QString(), true, extView);
     m_plugin->readConfig();
 
     QCOMPARE(m_plugin->qtHelpProviderLoaded().size(), 1);
@@ -158,7 +164,8 @@ void TestQtHelpPlugin::testRemoveOneProvider()
     name << QStringLiteral("file1") << QStringLiteral("file2");
     icon << QStringLiteral("myIcon") << QStringLiteral("myIcon");
     ghns << QStringLiteral("0") << QStringLiteral("0");
-    qtHelpWriteConfig(icon, name, path, ghns, QString(), true);
+    ExternalViewerSettings extView = {false, QString()};
+    qtHelpWriteConfig(icon, name, path, ghns, QString(), true, extView);
     m_plugin->readConfig();
 
     QCOMPARE(m_plugin->qtHelpProviderLoaded().size(), 2);
@@ -168,7 +175,7 @@ void TestQtHelpPlugin::testRemoveOneProvider()
     name.removeAt(1);
     icon.removeAt(1);
     ghns.removeAt(1);
-    qtHelpWriteConfig(icon, name, path, ghns, QString(), true);
+    qtHelpWriteConfig(icon, name, path, ghns, QString(), true, extView);
     m_plugin->readConfig();
 
     QCOMPARE(m_plugin->qtHelpProviderLoaded().size(), 1);
