diff --git kdevplatform/shell/openprojectdialog.cpp kdevplatform/shell/openprojectdialog.cpp
index e621edadc98c4aa492d65260f5f4ed2d072d1e1d..c9f4ad7c719ccb3912b949fb61aefafe3d385688 100644
--- kdevplatform/shell/openprojectdialog.cpp
+++ kdevplatform/shell/openprojectdialog.cpp
@@ -304,7 +310,19 @@ void OpenProjectDialog::openPageAccepted()
 
 void OpenProjectDialog::validateProjectName( const QString& name )
 {
-    m_projectName = name;
+    if (name != m_projectName) {
+        m_projectName = name;
+        QUrl url(m_url.adjusted(QUrl::StripTrailingSlash | QUrl::RemoveFilename));
+        // construct a version of the project name that's safe for use as a filename:
+        // TODO: do an additional replace of QDir::separator() with "@"?
+        QString safeName = m_projectName.replace(QRegExp("[\\\\/]"), QStringLiteral("@"));
+        safeName = safeName.replace(QChar(':'), QChar('='));
+        safeName = safeName.replace(QRegExp("\\s"), QStringLiteral("_"));
+        safeName += '.' + ShellExtension::getInstance()->projectFileExtension();
+        m_url.setPath(url.path() + '/' + safeName);
+        m_urlIsDirectory = false;
+        qCDebug(SHELL) << "project name:" << m_projectName << "file name:" << safeName;
+    }
     validateProjectInfo();
 }
 
diff --git kdevplatform/shell/projectcontroller.cpp kdevplatform/shell/projectcontroller.cpp
index d687c269d22d363652b599099b46999fd22b85cb..f6ca4412c788e91273e1c737716d50397e69b08c 100644
--- kdevplatform/shell/projectcontroller.cpp
+++ kdevplatform/shell/projectcontroller.cpp
@@ -429,7 +446,8 @@ QUrl ProjectDialogProvider::askProjectConfigLocation(bool fetch, const QUrl& sta
     }
 
     QUrl projectFileUrl = dlg->projectFileUrl();
-    qCDebug(SHELL) << "selected project:" << projectFileUrl << dlg->projectName() << dlg->projectManager();
+    qCDebug(SHELL) << "selected project:" << projectFileUrl << "selectedUrl=" << dlg->selectedUrl()
+        << "projectName=" << dlg->projectName() << "projectManager=" << dlg->projectManager();
     if ( dlg->projectManager() == QLatin1String("<built-in>") ) {
         return projectFileUrl;
     }
@@ -439,8 +457,9 @@ QUrl ProjectDialogProvider::askProjectConfigLocation(bool fetch, const QUrl& sta
     if( projectFileExists( projectFileUrl ) )
     {
         // check whether config is equal
-        bool shouldAsk = true;
-        if( projectFileUrl == dlg->selectedUrl() )
+        bool isKDevProject = QFileInfo(projectFileUrl.url()).completeSuffix() == QStringLiteral("kdev4");
+        bool shouldAsk = !isKDevProject;
+        if( !isKDevProject && projectFileUrl == dlg->selectedUrl() )
         {
             if( projectFileUrl.isLocalFile() )
             {
