diff --git CMakeLists.txt CMakeLists.txt
index 39e4297799..1c8cb193bb 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -1,5 +1,10 @@
 cmake_minimum_required(VERSION 3.0)
 project(KDevelop VERSION 5.1.80)
+execute_process(
+    COMMAND generate-patchversion.sh ${KDevelop_VERSION_MAJOR} ${KDevelop_VERSION_MINOR} ${KDevelop_VERSION_PATCH}
+    WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+    OUTPUT_VARIABLE KDevelop_VERSION_PATCH
+    OUTPUT_STRIP_TRAILING_WHITESPACE)
 
 # KDevelop SOVERSION
 # E.g. for KDevelop 5.2.0 => SOVERSION 52 (we only promise ABI compatibility between patch version updates)
diff --git generate-patchversion.sh generate-patchversion.sh
new file mode 100755
index 0000000000..1f815d68fb
--- /dev/null
+++ generate-patchversion.sh
@@ -0,0 +1,11 @@
+#!/bin/sh
+# usage: generate-patchversion.sh MAJOR MINOR PATCH
+
+if [ -d .git ] ;then
+    # obtain the tag-based full version string for the state currently checked out
+    GITVERSION="`git describe --tags`"
+    # output the amended patch version string
+    echo "${GITVERSION}" | sed -e "s/v${1}\.${2}\.//g"
+else
+    echo $3
+fi
