diff --git plugins/CMakeLists.txt plugins/CMakeLists.txt
index e94d5e235e..adaac866b5 100644
--- plugins/CMakeLists.txt
+++ plugins/CMakeLists.txt
@@ -9,7 +9,9 @@ if (KDEVELOP_BUILD_IDE)
     # TODO: Make compile under Windows
     add_subdirectory(debuggercommon)
     add_subdirectory(lldb)
-    add_subdirectory(gdb)
+    if(NOT APPLE)
+        add_subdirectory(gdb)
+    endif()
 endif()
 # END: Debuggers
 
