diff --git kdevplatform/documentation/documentationview.cpp kdevplatform/documentation/documentationview.cpp
index 2c6cccf8bd742e69022be8ee002ea49246c93f03..09035d027737a882e0d493256a4a4cabe4886e46 100644
--- kdevplatform/documentation/documentationview.cpp
+++ kdevplatform/documentation/documentationview.cpp
@@ -39,11 +39,18 @@
 #include "documentationfindwidget.h"
 #include "debug.h"
 
+#include <QDockWidget>
+
 using namespace KDevelop;
 
 DocumentationView::DocumentationView(QWidget* parent, ProvidersModel* model)
     : QWidget(parent), mProvidersModel(model)
 {
+    if (ICore::self()->shuttingDown()) {
+        qCWarning(DOCUMENTATION) << "DocumentationView created during shutdown; ignoring";
+        return;
+    }
+
     setWindowIcon(QIcon::fromTheme(QStringLiteral("documentation"), windowIcon()));
     setWindowTitle(i18n("Documentation"));
 
@@ -65,6 +72,20 @@ DocumentationView::DocumentationView(QWidget* parent, ProvidersModel* model)
     setFocusProxy(mIdentifiers);
 
     QMetaObject::invokeMethod(this, "initialize", Qt::QueuedConnection);
+    QDockWidget* dockWidget = dynamic_cast<QDockWidget*>(parent);
+    if (dockWidget) {
+        dockWidget->setFloating(false);
+        QMetaObject::invokeMethod(dockWidget, "closeRequested", Qt::QueuedConnection);
+        if (dockWidget->widget()) {
+            qWarning() << "closing" << dockWidget << "widget" << dockWidget->widget();
+            dockWidget->widget()->close();
+        }
+        connect(ICore::self(), &ICore::aboutToShutdown, dockWidget, [dockWidget] () {
+            /*if (isFloating())*/ {
+                qWarning() << "Re-docking documentation" << dockWidget << "at exit";
+                dockWidget->setFloating(false);
+            } } );
+    }
 }
 
 QList<QAction*> DocumentationView::contextMenuActions() const
@@ -280,7 +301,9 @@ void DocumentationView::updateView()
     if (mCurrent != mHistory.end()) {
         w = (*mCurrent)->documentationWidget(mFindDoc, this);
         Q_ASSERT(w);
-        QWidget::setTabOrder(mIdentifiers, w);
+        if (mIdentifiers->window() == w->window()) {
+            QWidget::setTabOrder(mIdentifiers, w);
+        }
     } else {
         // placeholder widget at location of doc view
         w = new QWidget(this);
