diff --git app/kdevelopui.rc app/kdevelopui.rc
index fd6021d0e2..b7b355148c 100644
--- app/kdevelopui.rc
+++ app/kdevelopui.rc
@@ -172,6 +172,8 @@
     <DefineGroup append="show_toolbar_merge" name="show_toolbar_merge"/>
     <DefineGroup append="show_merge" name="show_merge" />
     <DefineGroup append="configure_merge" name="configure_merge" />
+    <Separator/>
+    <Action name="widgetstyle_submenu"/>
   </Menu>
 
   <Menu name="help">
