diff --git kdevplatform/documentation/documentationview.cpp kdevplatform/documentation/documentationview.cpp
index 2c6cccf8bd742e69022be8ee002ea49246c93f03..7476b4e1344dab046e08d8e4573d5d3d50f69fcd 100644
--- kdevplatform/documentation/documentationview.cpp
+++ kdevplatform/documentation/documentationview.cpp
@@ -39,6 +39,8 @@
 #include "documentationfindwidget.h"
 #include "debug.h"
 
+#include <sublime/idealdockwidget.h>
+
 using namespace KDevelop;
 
 DocumentationView::DocumentationView(QWidget* parent, ProvidersModel* model)
@@ -70,6 +72,11 @@ DocumentationView::DocumentationView(QWidget* parent, ProvidersModel* model)
     setFocusProxy(mIdentifiers);
 
     QMetaObject::invokeMethod(this, "initialize", Qt::QueuedConnection);
+    Sublime::IdealDockWidget* dockWidget = dynamic_cast<Sublime::IdealDockWidget*>(parent);
+    if (dockWidget) {
+        dockWidget->setFloating(false);
+        dockWidget->setFloatsAsStandalone(true);
+    }
 }
 
 QList<QAction*> DocumentationView::contextMenuActions() const
