diff --git a/nixnote.cpp b/nixnote.cpp
index 777edfd141cf7abea9ac94a754ca3b684020c9a8..0f9749ab399583b155b566cbe2190fb92a0ac077 100644
--- a/nixnote.cpp
+++ b/nixnote.cpp
@@ -280,7 +280,10 @@ void NixNote::setupGui() {
     // Setup the GUI
     //this->setStyleSheet("background-color: white;");
     //statusBar();    setWindowTitle(tr("NixNote 2"));
-    setWindowIcon(QIcon(global.getIconResource(":windowIcon")));
+    const auto wIcon = QIcon(global.getIconResource(":windowIcon"));
+    if (!wIcon.isNull()) {
+        setWindowIcon(wIcon);
+    }
 
     //QLOG_TRACE() << "Setting up menu bar";
     searchText = new LineEdit();
@@ -2945,7 +2948,10 @@ void NixNote::openPreferences() {
     if (prefs.okButtonPressed) {
         setSyncTimer();
         bool showTrayIcon = global.showTrayIcon();
-        setWindowIcon(global.getIconResource(":windowIcon"));
+        const auto wIcon = QIcon(global.getIconResource(":windowIcon"));
+        if (!wIcon.isNull()) {
+            setWindowIcon(wIcon);
+        }
         trayIcon->setIcon(global.getIconResource(":trayIcon"));
         if (!showTrayIcon) {
             //trayIconBehavior();
@@ -3613,7 +3619,10 @@ void NixNote::reloadIcons() {
         global.loadTheme(global.resourceList,global.colorList,newThemeName);
     }
 
-    setWindowIcon(QIcon(global.getIconResource(":windowIcon")));
+    const auto wIcon = QIcon(global.getIconResource(":windowIcon"));
+    if (!wIcon.isNull()) {
+        setWindowIcon(wIcon);
+    }
     leftArrowButton->setIcon(global.getIconResource(":leftArrowIcon"));
     rightArrowButton->setIcon(global.getIconResource(":rightArrowIcon"));
     homeButton->setIcon(global.getIconResource(":homeIcon"));
