diff --git a/nixnote.cpp b/nixnote.cpp
index 777edfd141cf7abea9ac94a754ca3b684020c9a8..1974ed9ba57d5ac32666325a0303e3004b286fe4 100644
--- a/nixnote.cpp
+++ b/nixnote.cpp
@@ -1278,10 +1281,12 @@ void NixNote::closeEvent(QCloseEvent *event) {
     syncRunner.keepRunning = false;
     syncThread.quit();
 
-    if (trayIcon->isVisible())
-        trayIcon->hide();
-    if (trayIcon != NULL)
+    if (trayIcon != NULL) {
+        if (trayIcon->isVisible())
+            trayIcon->hide();
         delete trayIcon;
+        trayIcon = NULL;
+    }
 
     QMainWindow::closeEvent(event);
     QLOG_DEBUG() << "Quitting";
