diff --git a/NixNote2.pro b/NixNote2.pro
index 2b548d3565427952ab4f5615259212eea14a68a0..ca2cbca6070ec7fb274ef047297f04802710c702 100644
--- a/NixNote2.pro
+++ b/NixNote2.pro
@@ -7,13 +7,15 @@
 greaterThan(QT_MAJOR_VERSION, 4) {
     QT       += core gui widgets printsupport webkit webkitwidgets sql network xml dbus qml
     DEFINES += QT_DISABLE_DEPRECATED_BEFORE=0
-    unix:INCLUDEPATH += /usr/include/poppler/qt5
+    unix {
+        CONFIG += link_pkgconfig
+        PKGCONFIG += poppler-qt5 libcurl
+    }
 #    unix:INCLUDEPATH += /usr/include/tidy
     win32:INCLUDEPATH +="$$PWD/winlib/includes/poppler/qt5"
     win32:INCLUDEPATH+= "$$PWD/winlib/includes"
     win32:LIBS += -L"$$PWD/winlib" -lpoppler-qt5
-    unix:LIBS +=    -lcurl \
-               -lpthread -L/usr/lib -lpoppler-qt5 -g -rdynamic
+    unix:!mac:LIBS += -lpthread -g -rdynamic
     win32:LIBS += -L"$$PWD/winlib" -lpoppler-qt5 -ltidy
     win32:RC_ICONS += "$$PWD/images/windowIcon.ico"
 }
@@ -38,7 +40,7 @@ CONFIG(debug, debug|release) {
         MOC_DIR = build/debug
 }
 
-CONFIG += debug_and_release
+# CONFIG += debug_and_release
 
 TRANSLATIONS = \
     translations/nixnote2_cs_CZ.ts \
@@ -452,15 +454,15 @@ HEADERS  += nixnote.h \
 
 
 
-unix:QMAKE_CXXFLAGS +=-g -O2 -fstack-protector --param=ssp-buffer-size=4 -Wformat -Werror=format-security
-unix:QMAKE_LFLAGS += -Wl,-Bsymbolic-functions -Wl,-z,relro
+# unix:QMAKE_CXXFLAGS +=-g -O2 -fstack-protector --param=ssp-buffer-size=4 -Wformat -Werror=format-security
+# unix:QMAKE_LFLAGS += -Wl,-Bsymbolic-functions -Wl,-z,relro
 
 win32:QMAKE_CXXFLAGS +=-g -O2 --param=ssp-buffer-size=4 -Wformat -Werror=format-security
 win32:QMAKE_LFLAGS += -Wl,-Bsymbolic-functions
 win32:DEFINES += SMTP_BUILD
 
 isEmpty(PREFIX) {
-    PREFIX = /usr/local
+    PREFIX = @PREFIX@
 }
 
 binary.path = $$PREFIX/bin/
@@ -482,7 +484,7 @@ qss.path = $$PREFIX/share/nixnote2/qss
 qss.files = qss/*
 
 pixmap.path = $$PREFIX/share/pixmaps/
-pixmap.extra = cp images/windowIcon.png images/nixnote2.png
+pixmap.extra = cp $$PWD/images/windowIcon.png $$PWD/images/nixnote2.png
 pixmap.files = images/nixnote2.png
 
 help.path = $$PREFIX/share/nixnote2/help
diff --git a/plugins/hunspell/Hunspell.pro b/plugins/hunspell/Hunspell.pro
index ab5f7f05298313f3a2002e0551b98552db90f341..1df81af7de22ce6b0f15dc21d951c0ba754985bf 100644
--- a/plugins/hunspell/Hunspell.pro
+++ b/plugins/hunspell/Hunspell.pro
@@ -1,5 +1,5 @@
-TARGET = ../hunspellplugin
-CONFIG += plugin
+TARGET = hunspellplugin
+CONFIG += plugin link_pkgconfig
 TEMPLATE = lib
 
 SOURCES += hunspellplugin.cpp \
@@ -10,6 +10,6 @@ HEADERS += hunspellplugin.h \
            hunspellinterface.h \
            ../../utilities/spellchecker.h
 
-LIBS +=    -lhunspell
+PKGCONFIG += hunspell
 
-QMAKE_LFLAGS += -Wl,-soname,libhunspellplugin.so.1
+unix:QMAKE_LFLAGS += -Wl,-soname,libhunspellplugin.so.1
