diff --git a/dialog/screencapture.cpp b/dialog/screencapture.cpp
index c3100af78af5f756bdcc84515e84a779c156e0f3..84d28ecba4594cbcc2cae5824698ad8aefb895cf 100644
--- a/dialog/screencapture.cpp
+++ b/dialog/screencapture.cpp
@@ -24,6 +24,9 @@
 
 #include <QDesktopWidget>
 #include <QApplication>
+#if QT_VERSION >= QT_VERSION_CHECK(5, 0, 0)
+#include <QScreen>
+#endif
 
 ScreenCapture::ScreenCapture(QWidget *parent) :
     QDialog(parent)
@@ -36,7 +39,14 @@ ScreenCapture::ScreenCapture(QWidget *parent) :
     sizeDesktop = QApplication::desktop()->size();
     resize(sizeDesktop);
 
+#if QT_VERSION >= QT_VERSION_CHECK(5, 0, 0)
+    // QPixmap::grabWindow is deprecated in Qt5,
+    // do a grab of the desktop on the screen we are currently on
+    QScreen *primaryScreen = QGuiApplication::primaryScreen();
+    desktopPixmapBkg = primaryScreen->grabWindow(QApplication::desktop()->winId());
+#else
     desktopPixmapBkg = QPixmap::grabWindow(QApplication::desktop()->winId());
+#endif
     desktopPixmapClr = desktopPixmapBkg;
 
     move(0, 0);
