diff --git a/gui/nmainmenubar.cpp b/gui/nmainmenubar.cpp
index 2cf559f1cccc29d631840d42e713b18a7ef6f4c5..cfb5ffb8adf8b5a2e8baa4025016d3bdfdcc63ae 100644
--- a/gui/nmainmenubar.cpp
+++ b/gui/nmainmenubar.cpp
@@ -258,6 +258,7 @@ void NMainMenuBar::setupEditMenu() {
     editMenu->addMenu(themeMenu);
 
     preferencesAction = new QAction(tr("Preferences"), this);
+    preferencesAction->setMenuRole(QAction::PreferencesRole);
     setupShortcut(preferencesAction, QString("Edit_Preferences"));
     editMenu->addAction(preferencesAction);
     connect(preferencesAction, SIGNAL(triggered()), parent, SLOT(openPreferences()));
@@ -579,10 +580,12 @@ void NMainMenuBar::setupHelpMenu() {
 
     aboutAction = new QAction(tr("&About"), this);
     aboutAction->setToolTip(tr("About"));
+    aboutAction->setMenuRole(QAction::AboutRole);
     connect(aboutAction, SIGNAL(triggered()), parent, SLOT(openAbout()));
     helpMenu->addAction(aboutAction);
 
     aboutQtAction = new QAction(tr("About &Qt"), this);
+    aboutQtAction->setMenuRole(QAction::AboutQtRole);
     aboutQtAction->setToolTip(tr("About"));
     connect(aboutQtAction, SIGNAL(triggered()), parent, SLOT(openQtAbout()));
     helpMenu->addAction(aboutQtAction);
