diff --git a/utilities/spellchecker.h b/utilities/spellchecker.h
index 08a166cfd546d4d14ded22122e7ee9451ddcd9cf..e0a3ddace3ed1adea457695932733d5e24f68fa2 100644
--- a/utilities/spellchecker.h
+++ b/utilities/spellchecker.h
@@ -27,7 +27,7 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 
 // Windows Check
 #ifndef _WIN32
-#include <hunspell/hunspell.hxx>
+class Hunspell;
 
 class SpellChecker : public QObject
 {
diff --git a/utilities/spellchecker.cpp b/utilities/spellchecker.cpp
index 66ebf677221014d80a07b3d464d66af794104557..df817e44d465d58a9067f77a3457cc7040dfeb2e 100644
--- a/utilities/spellchecker.cpp
+++ b/utilities/spellchecker.cpp
@@ -24,6 +24,7 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 #include <QLocale>
 #include <QTextStream>
 
+#include <hunspell.hxx>
 
 SpellChecker::SpellChecker(QObject *parent) :
     QObject(parent)
diff --git a/gui/nbrowserwindow.cpp b/gui/nbrowserwindow.cpp
index 8d3c2fa4777104670326dcb3fccb21e5ccedb338..6178c5efe1c3b186a2a76ecea194e54678706230 100644
--- a/gui/nbrowserwindow.cpp
+++ b/gui/nbrowserwindow.cpp
@@ -3801,10 +3800,11 @@ void NBrowserWindow::loadPlugins() {
         QDir pluginsDir(dirList[i]);
         QStringList filter;
         filter.append("libhunspellplugin.so");
+        filter.append("libhunspellplugin.dylib");
         foreach (QString fileName, pluginsDir.entryList(filter)) {
             QPluginLoader pluginLoader(pluginsDir.absoluteFilePath(fileName));
             QObject *plugin = pluginLoader.instance();
-            if (fileName == "libhunspellplugin.so") {
+            if (fileName == "libhunspellplugin.so" || fileName == "libhunspellplugin.dylib") {
                 if (plugin) {
                     hunspellInterface = qobject_cast<HunspellInterface *>(plugin);
                     if (hunspellInterface) {
diff --git a/nixnote.cpp b/nixnote.cpp
index 777edfd141cf7abea9ac94a754ca3b684020c9a8..e4e9aabaf5b1e4def7c26533acff6a3ef6787645 100644
--- a/nixnote.cpp
+++ b/nixnote.cpp
@@ -3860,10 +3870,12 @@ void NixNote::loadPlugins() {
         QStringList filter;
         filter.append("libwebcamplugin.so");
         filter.append("libhunspellplugin.so");
+        filter.append("libwebcamplugin.dylib");
+        filter.append("libhunspellplugin.dylib");
         foreach (QString fileName, pluginsDir.entryList(filter)) {
             QPluginLoader pluginLoader(pluginsDir.absoluteFilePath(fileName));
             QObject *plugin = pluginLoader.instance();
-            if (fileName == "libwebcamplugin.so") {
+            if (fileName == "libwebcamplugin.so" || fileName == "libwebcamplugin.dylib") {
                 if (plugin) {
                     webcamInterface = qobject_cast<WebCamInterface *>(plugin);
                     if (webcamInterface) {
@@ -3876,7 +3888,7 @@ void NixNote::loadPlugins() {
 
             // The Hunspell plugin isn't actually used here. We just use this as a
             // check to be sure that the menu should be available.
-            if (fileName == "libhunspellplugin.so") {
+            if (fileName == "libhunspellplugin.so" || fileName == "libhunspellplugin.dylib") {
                 if (plugin) {
                     HunspellInterface *hunspellInterface;
                     hunspellInterface = qobject_cast<HunspellInterface *>(plugin);
