diff --git a/gui/nbrowserwindow.cpp b/gui/nbrowserwindow.cpp
index 8d3c2fa4777104670326dcb3fccb21e5ccedb338..74c6d7b9a78439fa6a9a95c7dbd829826d09cde6 100644
--- a/gui/nbrowserwindow.cpp
+++ b/gui/nbrowserwindow.cpp
@@ -3791,9 +3791,12 @@ void NBrowserWindow::loadPlugins() {
     QStringList dirList;
     dirList.append(global.fileManager.getProgramDirPath(""));
     dirList.append(global.fileManager.getProgramDirPath("")+"/plugins");
-    dirList.append("/usr/lib/nixnote2/");
-    dirList.append("/usr/local/lib/nixnote2/");
-    dirList.append("/usr/local/lib");
+#if defined(Q_OS_MACOS) && defined(USE_QSP)
+    // support installing additional plugins in the standard locations where they might be found
+    dirList.append(QStandardPaths::locate(QStandardPaths::AppDataLocation, "plugins", QStandardPaths::LocateDirectory));
+#endif
+    dirList.append("@PREFIX@/lib/nixnote2/");
+    dirList.append("@PREFIX@/lib");
     dirList.append("/usr/lib");
 
     for (int i=0; i<dirList.size(); i++) {
diff --git a/nixnote.cpp b/nixnote.cpp
index 777edfd141cf7abea9ac94a754ca3b684020c9a8..850ddacedffdab15752229b43bee3be465cd4102 100644
--- a/nixnote.cpp
+++ b/nixnote.cpp
@@ -3849,9 +3864,12 @@ void NixNote::loadPlugins() {
     QStringList dirList;
     dirList.append(global.fileManager.getProgramDirPath(""));
     dirList.append(global.fileManager.getProgramDirPath("")+"/plugins");
-    dirList.append("/usr/lib/nixnote2/");
-    dirList.append("/usr/local/lib/nixnote2/");
-    dirList.append("/usr/local/lib");
+#if defined(Q_OS_MACOS) && defined(USE_QSP)
+    // support installing additional plugins in the standard locations where they might be found
+    dirList.append(QStandardPaths::locate(QStandardPaths::AppDataLocation, "plugins", QStandardPaths::LocateDirectory));
+#endif
+    dirList.append("@PREFIX@/lib/nixnote2/");
+    dirList.append("@PREFIX@/lib");
     dirList.append("/usr/lib");
 
     // Start loading plugins
diff --git a/settings/filemanager.cpp b/settings/filemanager.cpp
index 93b8f4b79ca164e21a833b38a1528aa66d6358a3..3e7110c6f6173740b9195303d59f3568deb43e67 100644
--- a/settings/filemanager.cpp
+++ b/settings/filemanager.cpp
@@ -22,6 +22,9 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 #include <iostream>
 #include <cstdlib>
 
+#ifdef USE_QSP
+#include <QLibraryInfo>
+#endif
 
 //*******************************************
 //* This class is used to find the location
@@ -34,19 +37,44 @@ FileManager::FileManager() {
 }
 
 void FileManager::setup(QString homeDirPath, QString programDirPath, int id) {
-
+#ifdef Q_OS_MACOS
+    QString appDirPath = QCoreApplication::applicationDirPath();
+    if (appDirPath.endsWith(".app/Contents/MacOS")) {
+        // get rid of the MacOS component
+        appDirPath.chop(5);
+        programDirPath = appDirPath + "/Resources/";
+    }
+#endif
     homeDir.setPath(toPlatformPathSeparator(homeDirPath));
     programDir.setPath(toPlatformPathSeparator(programDirPath));
     createDirOrCheckWriteable(homeDir);
     this->homeDirPath = slashTerminatePath(homeDir.path());
     this->programDirPath = slashTerminatePath(programDir.path());
 
+#ifdef USE_QSP
+#ifdef Q_OS_MACOS
+    // get the resources from the app bundle
+    dataDirPath = programDirPath;
+#else
+    dataDirPath = QLibraryInfo::location(QLibraryInfo::PrefixPath) + "/share/nixnote2/";
+#endif
+    const QString userDataDir(QStandardPaths::writableLocation(QStandardPaths::AppDataLocation)+"/");
+    const QString configDir(QStandardPaths::writableLocation(QStandardPaths::AppConfigLocation) + "/");
+#else
+    // preserve the legacy behaviour
+    dataDirPat = programDirPath;
+    const QString userDataDir(homeDirPath);
+    const QString configDir(getHomeDirPath(""));
+#endif
+
+    dataDir.setPath(dataDirPath);
+
     // Read only files that everyone uses
-    imagesDir.setPath(programDirPath+"images");
+    imagesDir.setPath(dataDirPath+"images");
     checkExistingReadableDir(imagesDir);
     imagesDirPath = slashTerminatePath(imagesDir.path());
 
-    javaDir.setPath(programDirPath+"java");
+    javaDir.setPath(dataDirPath+"java");
     checkExistingReadableDir(javaDir);
     javaDirPath = slashTerminatePath(javaDir.path());
 
@@ -54,25 +82,25 @@ void FileManager::setup(QString homeDirPath, QString programDirPath, int id) {
 //    checkExistingReadableDir(spellDir);
 //    spellDirPath = slashTerminatePath(spellDir.path());
 
-    spellDirUser.setPath(homeDirPath+"spell");
+    spellDirUser.setPath(userDataDir+"spell");
     createDirOrCheckWriteable(spellDirUser);
     spellDirPathUser = slashTerminatePath(spellDirUser.path());
 
     //xmlDir.setPath(programDirPath+"xml");
     //checkExistingReadableDir(xmlDir);
 
-    translateDir.setPath(programDirPath+"translations");
+    translateDir.setPath(dataDirPath+"translations");
     checkExistingReadableDir(translateDir);
     translateDirPath= slashTerminatePath(translateDir.path());
 
-    qssDir.setPath(programDirPath+"qss");
+    qssDir.setPath(dataDirPath+"qss");
     checkExistingReadableDir(qssDir);
     qssDirPath = slashTerminatePath(qssDir.path());
 
 
     // Read/write directories that only we use
 
-    QString settingsFile = getHomeDirPath("") + "nixnote.conf";
+    QString settingsFile = configDir + "nixnote.conf";
     QSettings globalSettings(settingsFile, QSettings::IniFormat);
 
     if (id <=0) {
@@ -82,19 +110,23 @@ void FileManager::setup(QString homeDirPath, QString programDirPath, int id) {
         id = accountId;
     }
 
-    qssDirUser.setPath(homeDirPath+"qss");
+    qssDirUser.setPath(userDataDir+"qss");
     createDirOrCheckWriteable(qssDirUser);
     qssDirPathUser = slashTerminatePath(qssDirUser.path());
 
-    logsDir.setPath(homeDirPath+"logs-" +QString::number(id));
+    logsDir.setPath(userDataDir+"logs-" +QString::number(id));
     createDirOrCheckWriteable(logsDir);
     logsDirPath = slashTerminatePath(logsDir.path());
 
-    tmpDir.setPath(homeDirPath+"tmp-" +QString::number(id));
+    tmpDir.setPath(userDataDir+"tmp-" +QString::number(id));
     createDirOrCheckWriteable(tmpDir);
     tmpDirPath = slashTerminatePath(tmpDir.path());
 
+#ifdef USE_QSP
+    dbDir.setPath(userDataDir+"db-" +QString::number(id));
+#else
     dbDir.setPath(homeDirPath+"db-" +QString::number(id));
+#endif
     createDirOrCheckWriteable(dbDir);
     dbDirPath = slashTerminatePath(dbDir.path());
 
@@ -155,7 +187,7 @@ void FileManager::deleteTopLevelFiles(QDir dir, bool exitOnFail) {
 void FileManager::createDirOrCheckWriteable(QDir dir) {
 
     if (!dir.exists()) {
-         if (!dir.mkdir(dir.path())) {
+         if (!dir.mkpath(dir.path())) {
              QLOG_FATAL() << "Failed to create directory '" << dir.path() << "'.  Aborting program.";
              exit(16);
          }
@@ -203,11 +235,19 @@ void FileManager::checkExistingWriteableDir(QDir dir) {
 
 
 QDir FileManager::getProgramDirFile(QString relativePath) {
+#ifdef USE_QSP
+    return QDir(dataDir.dirName() + toPlatformPathSeparator(relativePath));
+#else
     return QDir(programDir.dirName() + toPlatformPathSeparator(relativePath));
+#endif
 }
 
 QString FileManager::getProgramDirPath(QString relativePath) {
+#ifdef USE_QSP
+    return dataDirPath + toPlatformPathSeparator(relativePath);
+#else
     return programDirPath + toPlatformPathSeparator(relativePath);
+#endif
 }
 QDir FileManager::getHomeDirFile(QString relativePath) {
     return QDir(homeDir.dirName() + toPlatformPathSeparator(relativePath));
diff --git a/settings/filemanager.h b/settings/filemanager.h
index c80d7aeee32e5c80bbaed344fb0c3e0481faf5a5..27965a3b1cd0ab14289226390d09183eadbabda5 100644
--- a/settings/filemanager.h
+++ b/settings/filemanager.h
@@ -27,6 +27,16 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 #include <QDir>
 #include <exception>
 
+// Under Qt5, use QStandardPaths except when told not to, or on MS Windows
+// (the latter choice may be unwise but I cannot test).
+#if QT_VERSION >= QT_VERSION_CHECK(5, 0, 0) && !defined(DONT_USE_QSTANDARDPATHS) && !defined(_WIN32)
+#define USE_QSP
+#endif
+
+#ifdef USE_QSP
+#include <QStandardPaths>
+#endif
+
 //************************************************
 //* This class is used to retrieve the
 //* location of certain directories.
@@ -45,6 +55,9 @@ private:
     QString  homeDirPath;
     QDir homeDir;
 
+    QString  dataDirPath;
+    QDir dataDir;
+
     QString dbDirPath;
     QDir dbDir;
 
diff --git a/global.cpp b/global.cpp
index 78a75ecdab9160e00b5a594032f3604e3becb914..d803dc9b5307ff53fb798a58da3f1affcd397c72 100644
--- a/global.cpp
+++ b/global.cpp
@@ -128,7 +128,12 @@ void Global::setup(StartupConfig startupConfig, bool guiAvailable) {
     this->guiAvailable = guiAvailable;
     fileManager.setup(startupConfig.homeDirPath, startupConfig.programDirPath, startupConfig.accountId);
     shortcutKeys = new ShortcutKeys();
+#ifdef USE_QSP
+    const QString configDir = QStandardPaths::writableLocation(QStandardPaths::AppConfigLocation) + "/";
+    QString settingsFile = configDir + "nixnote.conf";
+#else
     QString settingsFile = fileManager.getHomeDirPath("") + "nixnote.conf";
+#endif
 
     globalSettings = new QSettings(settingsFile, QSettings::IniFormat);
     int accountId = startupConfig.accountId;
@@ -150,7 +155,11 @@ void Global::setup(StartupConfig startupConfig, bool guiAvailable) {
     sharedMemory = new CrossMemoryMapper(key);
 
 
+#ifdef USE_QSP
+    settingsFile = configDir + "nixnote-"+QString::number(accountId)+".conf";
+#else
     settingsFile = fileManager.getHomeDirPath("") + "nixnote-"+QString::number(accountId)+".conf";
+#endif
 
     settings = new QSettings(settingsFile, QSettings::IniFormat);
 
@@ -269,12 +278,18 @@ void Global::setup(StartupConfig startupConfig, bool guiAvailable) {
 // This is because we want to find other paths (like images).  This
 // allows for users to run it out of a non-path location.
 QString Global::getProgramDirPath() {
+#ifdef USE_QSP
+    // FileManager::getProgramDirPath() already returns exactly what
+    // we want to return ourselves.
+    return fileManager.getProgramDirPath("");
+#else
     QString path = QCoreApplication::applicationDirPath();
     if (path.endsWith("/bin")) {
         path.chop(3);
         return path+"share/nixnote2";
     }
     return path;
+#endif
 }
 
 
@@ -934,7 +949,11 @@ void Global::loadTheme(QHash<QString, QString> &resourceList, QHash<QString,QStr
     this->loadThemeFile(resourceList,colorList, systemTheme, theme);
 
 #ifndef _WIN32
+#ifdef USE_QSP
+    QFile userTheme(QStandardPaths::locate(QStandardPaths::AppConfigLocation, "theme.ini"));
+#else
     QFile userTheme(fileManager.getHomeDirPath("theme.ini"));
+#endif
 #else
     QFile userTheme(fileManager.getHomeDirPath("theme.ini").replace("\\","/"));
 #endif
@@ -1003,7 +1022,11 @@ QStringList Global::getThemeNames() {
 #endif
     this->getThemeNamesFromFile(systemTheme, values);
 #ifndef _WIN32
+#ifdef USE_QSP
+    QFile userTheme(QStandardPaths::locate(QStandardPaths::AppConfigLocation, "theme.ini"));
+#else
     QFile userTheme(fileManager.getHomeDirPath("theme.ini"));
+#endif
 #else
     QFile userTheme(fileManager.getHomeDirPath("theme.ini").replace("\\","/"));
 #endif
diff --git a/settings/accountsmanager.cpp b/settings/accountsmanager.cpp
index b837285af746bd6a413660cb23667762cfb361a8..c315c2bec08b056757a0123cedfe4afcb46c9200 100644
--- a/settings/accountsmanager.cpp
+++ b/settings/accountsmanager.cpp
@@ -30,7 +30,11 @@ AccountsManager::AccountsManager(int id, QObject *parent) :
     QObject(parent)
 {
     currentId = id;
+#ifdef USE_QSP
+    configFile = QStandardPaths::writableLocation(QStandardPaths::AppConfigLocation) + "/accounts.conf";
+#else
     configFile = global.fileManager.getHomeDirPath("")+"accounts.conf";
+#endif
     if (!QFile(configFile).exists()) {
         QFile xmlFile(configFile);
         xmlFile.open(QIODevice::WriteOnly | QIODevice::Text);
diff --git a/settings/startupconfig.cpp b/settings/startupconfig.cpp
index 16a284d607d5ff976ff56cf407a6d95dda1bb939..3ede8073c145cf6ea327caded7b9b6da1c480ed8 100644
--- a/settings/startupconfig.cpp
+++ b/settings/startupconfig.cpp
@@ -30,7 +30,11 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 
 StartupConfig::StartupConfig()
 {
+#ifdef USE_QSP
+    homeDirPath = QStandardPaths::writableLocation(QStandardPaths::AppDataLocation) + "/";
+#else
     homeDirPath = QDir().homePath() + QString("/.nixnote/");
+#endif
     this->forceNoStartMinimized = false;
     this->startupNewNote = false;
     this->sqlExec = false;
diff --git a/dialog/preferences/appearancepreferences.cpp b/dialog/preferences/appearancepreferences.cpp
index 562a36cc549da45c26961abf044ad3eea4b385c0..4f05591f10f8cbf71cc72c33b4b9bcd92093c13d 100644
--- a/dialog/preferences/appearancepreferences.cpp
+++ b/dialog/preferences/appearancepreferences.cpp
@@ -318,12 +318,16 @@ void AppearancePreferences::saveValues() {
 
 
     // See if the user has overridden the window icon
-
+#ifndef __APPLE__
         //Copy the nixnote2.desktop so we can override the app icon
         // Ideally, we could use QSettings since it is ini format, but
         // it puts [Desktop Entry] as [Desktop%20Enry], which screws
         // things up.
+#ifdef USE_QSP
+        QString systemFile = QLibraryInfo::location(QLibraryInfo::PrefixPath) + "/share/applications/nixnote2.desktop";
+#else
         QString systemFile = "/usr/share/applications/nixnote2.desktop";
+#endif
         QFile systemIni(systemFile);
         QStringList desktopData;
 
@@ -341,16 +345,18 @@ void AppearancePreferences::saveValues() {
         systemIni.close();
 
         // Now, write it back out
-        QString userFile =  QDir::homePath()+"/.local/share/applications/nixnote2.desktop";
-        QFile userIni(userFile);
-        if (userIni.open(QIODevice::WriteOnly)) {
-            QTextStream data(&userIni);
-            for (int i=0; i<desktopData.size(); i++) {
-                data << desktopData[i] << "\n";
+        if (!desktopData.isEmpty()) {
+            QString userFile =  QDir::homePath()+"/.local/share/applications/nixnote2.desktop";
+            QFile userIni(userFile);
+            if (userIni.open(QIODevice::WriteOnly)) {
+                QTextStream data(&userIni);
+                for (int i=0; i<desktopData.size(); i++) {
+                    data << desktopData[i] << "\n";
+                }
             }
+            userIni.close();
         }
-        userIni.close();
-
+#endif
 //    }
 
     // Setup if the user wants to start NixNote the next time they login.
@@ -366,9 +372,13 @@ void AppearancePreferences::saveValues() {
     if (autoStart->isChecked()) {
 #ifdef _WIN32
         QFile::link(QCoreApplication::applicationFilePath(), QDesktopServices::storageLocation(QDesktopServices::ApplicationsLocation) + QDir::separator() + "Startup" + QDir::separator() + fileInfo.completeBaseName() + ".lnk");
-#else
+#elif !defined(__APPLE__)
         //Copy the nixnote2.desktop to the ~/.config/autostart directory
+#ifdef USE_QSP
+        QString systemFile = QLibraryInfo::location(QLibraryInfo::PrefixPath) + "/share/applications/nixnote2.desktop";
+#else
         QString systemFile = "/usr/share/applications/nixnote2.desktop";
+#endif
         QFile systemIni(systemFile);
         QStringList desktopData;
 
