diff --git a/gui/shortcutkeys.cpp b/gui/shortcutkeys.cpp
index 761fdedd664458f330edfca06952164b218089af..3e19c63487660f366207791ab890512bda31b547 100644
--- a/gui/shortcutkeys.cpp
+++ b/gui/shortcutkeys.cpp
@@ -243,7 +243,7 @@ void ShortcutKeys::loadCustomKeys(QString fileName) {
         }
         file.close();
     } else {
-        QLOG_DEBUG() << "Unable to open file for reading or file does not exist.";
+        qDebug() << Q_FUNC_INFO << "Unable to open" << fileName << "for reading or file does not exist.";
     }
 }
 
diff --git a/exits/exitmanager.cpp b/exits/exitmanager.cpp
index 156a4cac3ab4d6b754cf2824d3b2e04cf36cd6c7..c1a453670c9fab66cbc2b70c9fac938b8d766d4f 100644
--- a/exits/exitmanager.cpp
+++ b/exits/exitmanager.cpp
@@ -54,18 +54,20 @@ void ExitManager::loadExitPoint(QString name, int goodVersion) {
     if (fileName =="")  {
         QLOG_ERROR() << tr("Script filename is blank. Disabling exit ") << name;
         enabled = false;
+        return;
     }
 
     if (version != goodVersion) {
         enabled = false;
-        QLOG_ERROR() << tr("Unknown script version. Disabling exit.") << name;
+        QLOG_ERROR() << tr("Unknown script version. Disabling exit.") << version << "," << name;
     }
 
     QString script;
     QFile f(fileName);
     if (!f.exists()) {
-        QLOG_ERROR() << tr("Script file doesn't exist or cannot be read. Disabling exit ") << name;
+        QLOG_ERROR() << tr("Script file doesn't exist or cannot be read. Disabling exit ") << fileName;
         enabled = false;
+        return;
     } else {
         f.open(QIODevice::ReadOnly);
         script = f.readAll();
