diff --git a/src/codeview.cpp b/src/codeview.cpp
index 5ebfa9b..a6fe64f 100644
--- a/src/codeview.cpp
+++ b/src/codeview.cpp
@@ -22,7 +22,10 @@
 #include "core.h"
 
 
- 
+#if QT_VERSION < QT_VERSION_CHECK(5, 11, 0)
+#define horizontalAdvance(t) width((t))
+#endif
+
 
 
 CodeView::CodeView()
diff --git a/src/consolewidget.cpp b/src/consolewidget.cpp
index 3ce529d..44f2615 100644
--- a/src/consolewidget.cpp
+++ b/src/consolewidget.cpp
@@ -27,6 +27,10 @@
 #define ASCII_BACKSPACE     '\b'
 #define ANSI_CSI           "\033["
 
+#if QT_VERSION < QT_VERSION_CHECK(5, 11, 0)
+#define horizontalAdvance(t) width((t))
+#endif
+
 static QColor red(255,0,0);
 
 ConsoleWidget::ConsoleWidget(QWidget *parent)
diff --git a/src/memorywidget.cpp b/src/memorywidget.cpp
index f0e77b0..01f5cf1 100644
--- a/src/memorywidget.cpp
+++ b/src/memorywidget.cpp
@@ -18,6 +18,10 @@
 #include "log.h"
 #include "util.h"
 
+#if QT_VERSION < QT_VERSION_CHECK(5, 11, 0)
+#define horizontalAdvance(t) width((t))
+#endif
+
 static const int PAD_ADDR_LEFT = 10; //!< Pad length left to the address field
 static const int PAD_ADDR_RIGHT = 10; //!< Pad length right to the address field.
 static const int PAD_HEX_LEFT = 10;   //!< Pad length left to the hex field.
diff --git a/src/tagscanner.cpp b/src/tagscanner.cpp
index 6cd40b2..f4e27dc 100644
--- a/src/tagscanner.cpp
+++ b/src/tagscanner.cpp
@@ -224,7 +224,11 @@ int TagScanner::scan(QString filepath, QList<Tag> *taglist)
     etagsCmd += filepath;
     QString name = g_ctagsCmd;
     QStringList argList;
+#if QT_VERSION >= QT_VERSION_CHECK(5, 14, 0)
     argList = etagsCmd.split(' ',  Qt::SkipEmptyParts);
+#else
+    argList = etagsCmd.split(' ',  QString::SkipEmptyParts);
+#endif
 
     QByteArray stdoutContent;
     QByteArray stderrContent;
@@ -238,7 +242,11 @@ int TagScanner::scan(QString filepath, QList<Tag> *taglist)
     QString all = stderrContent;
     if(!all.isEmpty())
     {
+#if QT_VERSION >= QT_VERSION_CHECK(5, 14, 0)
         QStringList outputList = all.split('\n', Qt::SkipEmptyParts);
+#else
+        QStringList outputList = all.split('\n', QString::SkipEmptyParts);
+#endif
         for(int r = 0;r < outputList.size();r++)
         {
             QString text = outputList[r];
diff --git a/src/variableinfowindow.cpp b/src/variableinfowindow.cpp
index 806f3f9..6671235 100644
--- a/src/variableinfowindow.cpp
+++ b/src/variableinfowindow.cpp
@@ -5,6 +5,9 @@
 
 #include "core.h"
 
+#if QT_VERSION < QT_VERSION_CHECK(5, 11, 0)
+#define horizontalAdvance(t) width((t))
+#endif
 
 VariableInfoWindow::VariableInfoWindow(QFont *font)
     : QWidget()
diff --git a/src/watchvarctl.cpp b/src/watchvarctl.cpp
index adaaf68..fae80aa 100644
--- a/src/watchvarctl.cpp
+++ b/src/watchvarctl.cpp
@@ -802,7 +802,11 @@ void WatchVarCtl::deleteSelected()
     }
 
     // Loop through the items
+#if QT_VERSION >= QT_VERSION_CHECK(5, 14, 0)
     QSet<QTreeWidgetItem *> itemSet(items.begin(), items.end());
+#else
+    QSet<QTreeWidgetItem *> itemSet(items.toSet());
+#endif
     QSet<QTreeWidgetItem *>::const_iterator setItr = itemSet.constBegin();
     for (;setItr != itemSet.constEnd();++setItr)
     {
