diff --git a/app_templates/django_project/manage.py b/app_templates/django_project/manage.py
index 3e4eedc9ff5862c99e999b544a5b34c4e9af877a..fe5aa28391e8e8eaf4f102b67ffa0abfdaffeb89 100644
--- a/app_templates/django_project/manage.py
+++ b/app_templates/django_project/manage.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3.5
 from django.core.management import execute_manager
 import imp
 try:
diff --git a/codecompletion/context.cpp b/codecompletion/context.cpp
index 66a7a08cbf5e2ffc32b1eab67d547da744434d7a..2b0942211f02414cc1500c095bbc8af6b67f6d20 100644
--- a/codecompletion/context.cpp
+++ b/codecompletion/context.cpp
@@ -126,7 +126,7 @@ PythonCodeCompletionContext::ItemList PythonCodeCompletionContext::shebangItems(
     if ( m_position.line == 0 && ( m_text.startsWith('#') || m_text.isEmpty() ) ) {
         QString i18ndescr = i18n("insert Shebang line");
         shebangGroup << CompletionTreeItemPointer(new KeywordItem(KDevelop::CodeCompletionContext::Ptr(this),
-                                                    "#!/usr/bin/env python\n", i18ndescr, f));
+                                                    "#!/usr/bin/env python3.5\n", i18ndescr, f));
         shebangGroup << CompletionTreeItemPointer(new KeywordItem(KDevelop::CodeCompletionContext::Ptr(this),
                                                     "#!/usr/bin/env python3\n", i18ndescr, f));
     }
diff --git a/codestyle.py b/codestyle.py
index 843f8c1872cff3466960f1274e47249091d398cc..3f75a82187035cd23cbfaf2a987c05c11ea04d10 100755
--- a/codestyle.py
+++ b/codestyle.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#!/usr/bin/env python3.5
 
 from contextlib import redirect_stdout
 from io import StringIO
diff --git a/documentation_src/introspection/introspect.py b/documentation_src/introspection/introspect.py
index 49e9025a0617787182a02050060a9c71fb1e10c3..89aabb69b884c443198ae955c8a211ccf5d345e6 100644
--- a/documentation_src/introspection/introspect.py
+++ b/documentation_src/introspection/introspect.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3.5
 # -*- Coding:utf-8 -*-
 
 # Copyright 2013 by Sven Brauch <svenbrauch@googlemail.com>
diff --git a/documentation_src/numpy/generate_numpy_doc.py b/documentation_src/numpy/generate_numpy_doc.py
index d861e1fab13faf78fb5ac58c23db565509dece17..52963ce0baab8013b8a2a096bbdee51790f750d8 100644
--- a/documentation_src/numpy/generate_numpy_doc.py
+++ b/documentation_src/numpy/generate_numpy_doc.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3.5
 
 # Copyright 2014 Sven Brauch <svenbrauch@gmail.com>
 # License: GPL v2+
diff --git a/documentation_src/pyqt/sip_to_xml5.py b/documentation_src/pyqt/sip_to_xml5.py
index 210064d3b3603ff265d1460a25f220879e350a04..ae7350dc919fade64a1aaa00013e5ee45ea519b0 100644
--- a/documentation_src/pyqt/sip_to_xml5.py
+++ b/documentation_src/pyqt/sip_to_xml5.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3.5
 # This file is part of KDevelop
 # Copyright 2011 by Victor Varvariuc <victor.varvariuc@gmail.com>, Sven Brauch <svenbrauch@googlemail.com>
 
diff --git a/parser/conversionGenerator.py b/parser/conversionGenerator.py
index 85b1ad05a2495c6d676ced4d81b61e658e3c5ec8..b2d670a1604f5da032393d10ca0c20e58fa328dc 100644
--- a/parser/conversionGenerator.py
+++ b/parser/conversionGenerator.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3.5
 
 # Copyright 2014 by Sven Brauch
 # License: GPL v2+
diff --git a/pythonpythonparser.py b/pythonpythonparser.py
index c76ed5832c11939445fac6b573ff4d3f2dff0456..f44647e32ac04f5a703e0ca6c85fc34cff5691d8 100755
--- a/pythonpythonparser.py
+++ b/pythonpythonparser.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#!/usr/bin/env python3.5
 # -*- coding: utf-8 -*-
 
 #
diff --git a/runtest.py b/runtest.py
index e99399195a8f6365881e5c4eb75375c341156478..0f9fe1de3f1cf9fda9c00ad3f081c9e0492398ea 100755
--- a/runtest.py
+++ b/runtest.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#!/usr/bin/env python3.5
 # -*- coding: utf-8 -*-
 
 # Copyright 2014 by Sven Brauch
