diff --git a/app_templates/django_project/manage.py b/app_templates/django_project/manage.py
index 3e4eedc..fe5aa28 100644
--- a/app_templates/django_project/manage.py
+++ b/app_templates/django_project/manage.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3.5
 from django.core.management import execute_manager
 import imp
 try:
diff --git a/codecompletion/context.cpp b/codecompletion/context.cpp
index 39db76d..81b180f 100644
--- a/codecompletion/context.cpp
+++ b/codecompletion/context.cpp
@@ -123,7 +123,7 @@ PythonCodeCompletionContext::ItemList PythonCodeCompletionContext::shebangItems(
     if ( m_position.line == 0 && ( m_text.startsWith('#') || m_text.isEmpty() ) ) {
         QString i18ndescr = i18n("insert Shebang line");
         shebangGroup << CompletionTreeItemPointer(new KeywordItem(KDevelop::CodeCompletionContext::Ptr(this),
-                                                    "#!/usr/bin/env python\n", i18ndescr, f));
+                                                    "#!/usr/bin/env python3.5\n", i18ndescr, f));
         shebangGroup << CompletionTreeItemPointer(new KeywordItem(KDevelop::CodeCompletionContext::Ptr(this),
                                                     "#!/usr/bin/env python3\n", i18ndescr, f));
     }
diff --git a/documentation_src/introspection/introspect.py b/documentation_src/introspection/introspect.py
index 49e9025..89aabb6 100644
--- a/documentation_src/introspection/introspect.py
+++ b/documentation_src/introspection/introspect.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3.5
 # -*- Coding:utf-8 -*-
 
 # Copyright 2013 by Sven Brauch <svenbrauch@googlemail.com>
diff --git a/documentation_src/numpy/generate_numpy_doc.py b/documentation_src/numpy/generate_numpy_doc.py
index d861e1f..52963ce 100644
--- a/documentation_src/numpy/generate_numpy_doc.py
+++ b/documentation_src/numpy/generate_numpy_doc.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3.5
 
 # Copyright 2014 Sven Brauch <svenbrauch@gmail.com>
 # License: GPL v2+
diff --git a/documentation_src/pyqt/sip_to_xml5.py b/documentation_src/pyqt/sip_to_xml5.py
index 210064d..ae7350d 100644
--- a/documentation_src/pyqt/sip_to_xml5.py
+++ b/documentation_src/pyqt/sip_to_xml5.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3.5
 # This file is part of KDevelop
 # Copyright 2011 by Victor Varvariuc <victor.varvariuc@gmail.com>, Sven Brauch <svenbrauch@googlemail.com>
 
diff --git a/parser/conversionGenerator.py b/parser/conversionGenerator.py
index 385dbcb..3dc2a3a 100644
--- a/parser/conversionGenerator.py
+++ b/parser/conversionGenerator.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3.5
 
 # Copyright 2014 by Sven Brauch
 # License: GPL v2+
diff --git a/pythonpythonparser.py b/pythonpythonparser.py
index c76ed58..f44647e 100755
--- a/pythonpythonparser.py
+++ b/pythonpythonparser.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#!/usr/bin/env python3.5
 # -*- coding: utf-8 -*-
 
 #
diff --git a/runtest.py b/runtest.py
index e993991..0f9fe1d 100755
--- a/runtest.py
+++ b/runtest.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#!/usr/bin/env python3.5
 # -*- coding: utf-8 -*-
 
 # Copyright 2014 by Sven Brauch
