diff --git CMakeLists.txt CMakeLists.txt
index 74e1d5c..1bb21c3 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -46,7 +46,7 @@ ecm_setup_version(PROJECT VARIABLE_PREFIX KWINDOWSYSTEM
                         SOVERSION 5)
 
 # Dependencies
-set(REQUIRED_QT_VERSION 5.14.0)
+set(REQUIRED_QT_VERSION 5.9.8)
 find_package(Qt5 ${REQUIRED_QT_VERSION} CONFIG REQUIRED Gui Widgets)
 
 if (APPLE)
@@ -80,7 +80,7 @@ if (IS_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/po")
 endif()
 
 add_definitions(-DQT_NO_FOREACH)
-add_definitions(-DQT_DISABLE_DEPRECATED_BEFORE=0x050e00)
+# add_definitions(-DQT_DISABLE_DEPRECATED_BEFORE=0x050e00)
 
 add_subdirectory(src)
 if (BUILD_TESTING)
diff --git src/platforms/xcb/kwindowshadow.cpp src/platforms/xcb/kwindowshadow.cpp
index 65c248f..50900d2 100644
--- src/platforms/xcb/kwindowshadow.cpp
+++ src/platforms/xcb/kwindowshadow.cpp
@@ -25,9 +25,13 @@ bool KWindowShadowTilePrivateX11::create()
     xcb_create_pixmap(connection, depth, pixmap, rootWindow, width, height);
     xcb_create_gc(connection, gc, pixmap, 0, nullptr);
 
+#if QT_VERSION >= QT_VERSION_CHECK(5, 10, 0)
     xcb_put_image(connection, XCB_IMAGE_FORMAT_Z_PIXMAP, pixmap, gc, width, height,
                   0, 0, 0, depth, image.sizeInBytes(), image.constBits());
-
+#else
+    xcb_put_image(connection, XCB_IMAGE_FORMAT_Z_PIXMAP, pixmap, gc, width, height,
+                  0, 0, 0, depth, image.byteCount(), image.constBits());
+#endif
     return true;
 }
 
