diff --git src/kkeyserver.cpp src/kkeyserver.cpp
index c005456..39a6a84 100644
--- src/kkeyserver.cpp
+++ src/kkeyserver.cpp
@@ -106,6 +106,17 @@ uint stringUserToMod(const QString &mod)
 
 bool isShiftAsModifierAllowed(int keyQt)
 {
+#ifdef Q_OS_MACOS
+    static enum { unset=0, cocoa=1, xcb=2 } currentPlatform;
+    if (currentPlatform == unset) {
+        const QString platformName = QGuiApplication::platformName();
+        if (platformName == QLatin1String("cocoa")) {
+            currentPlatform = cocoa;
+        } else if (platformName == QLatin1String("xcb")) {
+            currentPlatform = xcb;
+        }
+    }
+#endif
     // remove any modifiers
     keyQt &= ~Qt::KeyboardModifierMask;
 
@@ -285,7 +296,17 @@ bool isShiftAsModifierAllowed(int keyQt)
         return true;
 
     default:
+#ifdef Q_OS_MACOS
+        switch (currentPlatform) {
+            default:
+            case cocoa:
+                return true;
+            case xcb:
+                return false;
+        }
+#else
         return false;
+#endif
     }
 }
 
