diff --git src/kwindowsystem.cpp src/kwindowsystem.cpp
index 4dfcce8..35eb7b5 100644
--- src/kwindowsystem.cpp
+++ src/kwindowsystem.cpp
@@ -462,6 +462,11 @@ WId KWindowSystem::transientFor(WId win)
 
 void KWindowSystem::setMainWindow(QWidget *subWidget, WId mainWindowId)
 {
+#ifdef Q_OS_MACOS
+    if (!QWidget::find(mainWindowId)) {
+        return;
+    }
+#endif
     // Set the WA_NativeWindow attribute to force the creation of the QWindow.
     // Without this QWidget::windowHandle() returns 0.
     subWidget->setAttribute(Qt::WA_NativeWindow, true);

