diff --git CMakeLists.txt CMakeLists.txt
index 9d70049..1c88e65 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -37,7 +37,8 @@ if(OXYGEN_USE_KDE4)
     PURPOSE "Required to oxygen window decoration for kwin"
  )
 
-  ### XCB
+if(NOT APPLE AND NOT WIN32)
+   ### XCB
   pkg_check_modules(XCB xcb x11-xcb)
   set(OXYGEN_HAVE_X11 ${XCB_FOUND})
 
@@ -45,7 +46,11 @@ if(OXYGEN_USE_KDE4)
   add_subdirectory(kstyle)
   if(KDE4Workspace_FOUND)
      add_subdirectory(kwin-decoration)
-  endif()
+  endif(KDE4Workspace_FOUND)
+else()
+  add_subdirectory(liboxygen)
+  add_subdirectory(kstyle)
+endif(NOT APPLE AND NOT WIN32)
 
 else()
 
