--- qtbase/mkspecs/common/orig.clang.conf	2015-03-27 15:43:52.000000000 +0100
+++ qtbase/mkspecs/common/clang.conf	2015-03-27 15:44:23.000000000 +0100
@@ -4,8 +4,8 @@
 
 QMAKE_COMPILER          = gcc clang llvm   # clang pretends to be gcc
 
-QMAKE_CC                = clang
-QMAKE_CXX               = clang++
+QMAKE_CC                = macportsclang
+QMAKE_CXX               = macportsclang++
 
 QMAKE_LINK              = $$QMAKE_CXX
 QMAKE_LINK_SHLIB        = $$QMAKE_CXX
