diff --git a/zconf.h b/zconf.h
index 0876e834e99aeab70a1ecc52b589c3529984b6d2..320460619c5bc46cca3e9efb126b762bf673f2a2 100644
--- a/zconf.h
+++ b/zconf.h
@@ -408,11 +408,11 @@ typedef uLong FAR uLongf;
    typedef unsigned long z_crc_t;
 #endif
 
-#if 1    /* was set to #if 1 by ./configure */
+#ifdef HAVE_UNISTD_H    /* may be set to #if 1 by ./configure */
 #  define Z_HAVE_UNISTD_H
 #endif
 
-#if 1    /* was set to #if 1 by ./configure */
+#ifdef HAVE_STDARG_H    /* may be set to #if 1 by ./configure */
 #  define Z_HAVE_STDARG_H
 #endif
 
