diff --git a/CMakeLists.txt b/CMakeLists.txt
index 33ee139..8092aa1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -29,6 +29,7 @@ option(SKIP_CPUID_CHECK "Assume CPU supports fast CRC" OFF)
 option(BUILD_GENERIC_CODE "Avoid architecture-specific code paths" OFF)
 option(FORCE_CHUNK_COPY "Force chunk-copy optimization" OFF)
 option(FORCE_UNALIGNED_READ_64LE "Force unaligned 64-bit read optimizaton" OFF)
+option(DISABLE_PCLMUL "Don't check for nor use PCLMUL" OFF)
 
 if(SKIP_CPUID_CHECK)
     add_definitions(-DSKIP_CPUID_CHECK)
@@ -95,10 +96,15 @@ if(UNIX OR MINGW)
                 add_definitions(-DHAS_SSE42)
             endif()
 
-            check_c_compiler_flag(-mpclmul HAS_PCLMUL)
-            if(HAS_PCLMUL)
-                set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mpclmul")
-                add_definitions(-DHAS_PCLMUL)
+		  if (NOT DISABLE_PCLMUL)
+                check_c_compiler_flag(-mpclmul HAS_PCLMUL)
+                if(HAS_PCLMUL)
+                    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mpclmul")
+                    add_definitions(-DHAS_PCLMUL)
+                endif()
+            else()
+                set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mno-pclmul")
+                add_definitions(-UHAS_PCLMUL)
             endif()
         endif()
     endif()
