diff --git a/CMakeLists.txt b/CMakeLists.txt
index 33ee139..018bc52 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -240,6 +240,17 @@ if(BUILD_SHARED_LIBS)
         # Creates zlib1.dll when building shared library version
         set_target_properties(zlib PROPERTIES SUFFIX "1.dll")
     endif()
+
+    add_library(zlib_static STATIC ${ZLIB_SRCS} ${ZLIB_PUBLIC_HDRS} ${ZLIB_PRIVATE_HDRS})
+    if(UNIX OR MINGW)
+        set_target_properties(zlib_static PROPERTIES OUTPUT_NAME z)
+    endif()
+    if(NOT SKIP_INSTALL_LIBRARIES AND NOT SKIP_INSTALL_ALL)
+        install(TARGETS zlib_static
+            RUNTIME DESTINATION bin
+            ARCHIVE DESTINATION lib
+            LIBRARY DESTINATION lib)
+    endif()
 else()
     add_library(zlib STATIC ${ZLIB_SRCS} ${ZLIB_PUBLIC_HDRS} ${ZLIB_PRIVATE_HDRS})
     if(UNIX OR MINGW)
