diff --git a/libcxxabi/CMakeLists.txt b/libcxxabi/CMakeLists.txt
index 6de2b5a..c49be65 100644
--- a/libcxxabi/CMakeLists.txt
+++ b/libcxxabi/CMakeLists.txt
@@ -358,11 +358,11 @@ if (LIBCXXABI_ENABLE_ASSERTIONS)
   endif()
   # On Release builds cmake automatically defines NDEBUG, so we
   # explicitly undefine it:
-  if (uppercase_CMAKE_BUILD_TYPE STREQUAL "RELEASE")
+  if (uppercase_CMAKE_BUILD_TYPE STREQUAL "RELEASE" OR uppercase_CMAKE_BUILD_TYPE STREQUAL "MACPORTS")
     list(APPEND LIBCXXABI_COMPILE_FLAGS -UNDEBUG)
   endif()
 else()
-  if (NOT uppercase_CMAKE_BUILD_TYPE STREQUAL "RELEASE")
+  if (NOT uppercase_CMAKE_BUILD_TYPE STREQUAL "RELEASE" AND NOT uppercase_CMAKE_BUILD_TYPE STREQUAL "MACPORTS")
     list(APPEND LIBCXXABI_COMPILE_FLAGS -DNDEBUG)
   endif()
 endif()
diff --git b/openmp/runtime/CMakeLists.txt.orig a/openmp/runtime/CMakeLists.txt
index 43299dd..e1e8677 100644
--- b/openmp/runtime/CMakeLists.txt.orig
+++ a/openmp/runtime/CMakeLists.txt
@@ -195,7 +195,7 @@ set(RELEASE_BUILD FALSE)
 set(DEBUG_BUILD FALSE)
 set(RELWITHDEBINFO_BUILD FALSE)
 set(MINSIZEREL_BUILD FALSE)
-if("${uppercase_CMAKE_BUILD_TYPE}" STREQUAL "RELEASE")
+if("${uppercase_CMAKE_BUILD_TYPE}" STREQUAL "RELEASE" OR "${uppercase_CMAKE_BUILD_TYPE}" STREQUAL "MACPORTS")
   set(RELEASE_BUILD TRUE)
 elseif("${uppercase_CMAKE_BUILD_TYPE}" STREQUAL "DEBUG")
   set(DEBUG_BUILD TRUE)

diff --git b/llvm/CMakeLists.txt.orig a/llvm/CMakeLists.txt
index 79de9eb..563cb38 100644
--- b/llvm/CMakeLists.txt.orig
+++ a/llvm/CMakeLists.txt
@@ -381,7 +381,7 @@ string(TOUPPER "${CMAKE_BUILD_TYPE}" uppercase_CMAKE_BUILD_TYPE)
 
 if (CMAKE_BUILD_TYPE AND
     NOT uppercase_CMAKE_BUILD_TYPE MATCHES "^(DEBUG|RELEASE|RELWITHDEBINFO|MINSIZEREL)$")
-  message(WARNING "Unknown value for CMAKE_BUILD_TYPE: ${CMAKE_BUILD_TYPE}")
+  message(STATUS "Custom value for CMAKE_BUILD_TYPE: ${CMAKE_BUILD_TYPE}")
 endif()
 
 # LLVM_INSTALL_PACKAGE_DIR needs to be declared prior to adding the tools
