diff --git b/libcxx/include/orig.optional a/libcxx/include/optional
index e7e5193..4694a43 100644
--- b/libcxx/include/orig.optional
+++ a/libcxx/include/optional
@@ -298,7 +298,15 @@ struct __optional_destruct_base<_Tp, false>
     bool __engaged_;
 
     _LIBCPP_INLINE_VISIBILITY
-    _LIBCPP_CONSTEXPR_SINCE_CXX20 ~__optional_destruct_base()
+#if (!defined(__apple_build_version__) && __clang_major__ > 9) || !defined(__clang__)
+    // Suppose that clang_major is an actual LLVM version number (and not some Apple fancy)
+    // and thus determine whether it supports const_expr dtors (clang 10 and up).
+    // __apple_build_version__ is hopefully defined by all AppleClang builds. If not, an
+    // error will be raised about not supporting const_expr dtors and the user will know
+    // that a newer compiler has to be used.
+    _LIBCPP_CONSTEXPR_SINCE_CXX20
+#endif
+    ~__optional_destruct_base()
     {
         if (__engaged_)
             __val_.~value_type();
