diff --git b/llvm/tools/llvm-symbolizer/origCMakeLists.txt a/llvm/tools/llvm-symbolizer/CMakeLists.txt
index a6bc047..f07dcc3 100644
--- b/llvm/tools/llvm-symbolizer/origCMakeLists.txt
+++ a/llvm/tools/llvm-symbolizer/CMakeLists.txt
@@ -3,6 +3,10 @@
 # This means that we need LLVM libraries to be compiled for these
 # targets as well. Currently, there is no support for such a build strategy.
 
+# we want to build and install this tool
+set(LLVM_BUILD_TOOLS ON)
+set(LLVM_INSTALL_BINUTILS_SYMLINKS ON)
+
 set(LLVM_TARGET_DEFINITIONS Opts.td)
 tablegen(LLVM Opts.inc -gen-opt-parser-defs)
 add_public_tablegen_target(SymbolizerOptsTableGen)
