diff --git src/kioslave/kioslave.cpp src/kioslave/kioslave.cpp
index be5b34c..3fb996d 100644
--- src/kioslave/kioslave.cpp
+++ src/kioslave/kioslave.cpp
@@ -79,7 +79,7 @@ int main(int argc, char **argv)
         return 1;
     }
 
-    qDebug() << "trying to load" << libname << "from" << libpath;
+//     qDebug() << "trying to load" << libname << "from" << libpath;
     QLibrary lib(libpath);
     if (!lib.load()) {
         fprintf(stderr, "could not open %s: %s\n", qPrintable(libname),
