--- src/orig.kwidgetjobtracker.cpp	2015-12-06 22:31:39.000000000 +0100
+++ src/kwidgetjobtracker.cpp	2015-12-20 19:47:17.000000000 +0100
@@ -465,7 +465,7 @@
 
 void KWidgetJobTracker::Private::ProgressWidget::init()
 {
-    setWindowIcon(QIcon::fromTheme(QStringLiteral("document-save")));
+    setWindowIcon(QIcon::fromTheme(QStringLiteral("document-save"), windowIcon()));
 
     QVBoxLayout *topLayout = new QVBoxLayout(this);
 
