--- src/widgets/orig.CMakeLists.txt	2015-11-08 13:32:49.000000000 +0100
+++ src/widgets/CMakeLists.txt	2015-12-02 23:01:24.000000000 +0100
@@ -27,6 +27,7 @@
                         KF5::PeopleBackend
                         KF5::CoreAddons
                         KF5::Service
+                        Qt5::QspXDG
 )
 
 set_target_properties (KF5PeopleWidgets PROPERTIES VERSION ${KPEOPLE_VERSION_STRING} SOVERSION ${KPEOPLE_SOVERSION} EXPORT_NAME PeopleWidgets)
--- src/backends/orig.CMakeLists.txt	2015-11-08 13:32:49.000000000 +0100
+++ src/backends/CMakeLists.txt	2015-12-02 23:00:38.000000000 +0100
@@ -11,6 +11,8 @@
 target_link_libraries (KF5PeopleBackend
                     PUBLIC
                         Qt5::Widgets #needed for QAction, otherwise QtGui
+                    PRIVATE
+                        Qt5::QspXDG
 )
 
 target_include_directories (KF5PeopleBackend INTERFACE "$<INSTALL_INTERFACE:${KF5_INCLUDE_INSTALL_DIR}/KPeople>")
--- src/orig.CMakeLists.txt	2015-12-02 22:57:28.000000000 +0100
+++ src/CMakeLists.txt	2015-12-02 23:00:00.000000000 +0100
@@ -31,6 +31,7 @@
                         KF5::CoreAddons
                         KF5::PeopleBackend
                         KF5::Service
+                        Qt5::QspXDG
 )
 
 set_target_properties (KF5People PROPERTIES VERSION ${KPEOPLE_VERSION_STRING} SOVERSION ${KPEOPLE_SOVERSION} EXPORT_NAME People)
