--- orig.CMakeLists.txt	2015-11-05 17:12:28.000000000 +0100
+++ CMakeLists.txt	2015-11-05 17:20:52.000000000 +0100
@@ -30,6 +30,7 @@
 
 # Dependencies
 set(REQUIRED_QT_VERSION 5.3.0)
+find_package(Qt5QspXDG ${REQUIRED_QT_VERSION} REQUIRED NO_MODULE)
 find_package(Qt5Widgets ${REQUIRED_QT_VERSION} REQUIRED NO_MODULE)
 find_package(Qt5Svg ${REQUIRED_QT_VERSION} REQUIRED NO_MODULE)
 find_package(Qt5DBus ${REQUIRED_QT_VERSION} REQUIRED NO_MODULE)
Exit 1
--- src/tools/ksvg2icns/orig.CMakeLists.txt	2015-10-05 10:08:51.000000000 +0200
+++ src/tools/ksvg2icns/CMakeLists.txt	2015-11-05 17:21:05.000000000 +0100
@@ -1,5 +1,5 @@
 add_executable(ksvg2icns ksvg2icns.cpp)
 ecm_mark_nongui_executable(ksvg2icns)
-target_link_libraries(ksvg2icns Qt5::Gui Qt5::Svg)
+target_link_libraries(ksvg2icns PUBLIC Qt5::Gui Qt5::Svg PRIVATE Qt5::QspXDG)
 
 install(TARGETS ksvg2icns ${KF5_INSTALL_TARGETS_DEFAULT_ARGS})
