--- src/kapidox/data/orig.Doxyfile.global	2015-10-19 22:22:40.000000000 +0200
+++ src/kapidox/data/Doxyfile.global	2015-12-01 20:46:28.000000000 +0100
@@ -90,6 +90,7 @@
                          *.dox \
                          *.c++ \
                          *.cxx \
+                         *.mm \
                          *.h++ \
                          *.hh \
                          *.qml \
