--- src/kjs/api/orig.CMakeLists.txt	2015-12-06 15:08:04.000000000 +0100
+++ src/kjs/api/CMakeLists.txt	2015-12-21 14:37:42.000000000 +0100
@@ -12,7 +12,8 @@
 
 target_link_libraries(KF5JSApi
     PUBLIC Qt5::Core
-    PRIVATE KF5::JS)
+    PRIVATE KF5::JS
+            Qt5::QspXDG)
 
 set_target_properties(KF5JSApi PROPERTIES VERSION ${KJS_VERSION_STRING}
                                           SOVERSION ${KJS_SOVERSION}
Exit 1
--- src/kjs/orig.CMakeLists.txt	2015-12-06 15:08:04.000000000 +0100
+++ src/kjs/CMakeLists.txt	2015-12-21 14:37:02.000000000 +0100
@@ -191,6 +191,9 @@
 if(PCRE_FOUND)
    target_link_libraries(KF5JS PRIVATE ${PCRE_LIBRARIES})
 endif()
+
+target_link_libraries(KF5JS PRIVATE Qt5::QspXDG)
+
 target_compile_definitions(KF5JS PRIVATE MAKE_KJS_LIB=1)
 
 set_target_properties(KF5JS PROPERTIES VERSION ${KJS_VERSION_STRING}
Exit 1
--- orig.CMakeLists.txt	2015-12-21 14:24:13.000000000 +0100
+++ CMakeLists.txt	2015-12-21 14:35:18.000000000 +0100
@@ -30,6 +30,7 @@
 
 set(REQUIRED_QT_VERSION 5.3.0)
 find_package(Qt5Core ${REQUIRED_QT_VERSION} REQUIRED NO_MODULE)
+find_package(Qt5QspXDG ${REQUIRED_QT_VERSION} REQUIRED NO_MODULE)
 
 find_package(Perl)
 set_package_properties(Perl PROPERTIES URL "http://www.perl.org" TYPE REQUIRED PURPOSE "Needed for building kdelibs")
