--- src/kdeinit/orig.ktoolinvocation_x11.cpp	2015-11-08 13:35:05.000000000 +0100
+++ src/kdeinit/ktoolinvocation_x11.cpp	2015-12-07 22:24:51.000000000 +0100
@@ -45,6 +45,10 @@
 #include <QUrl>
 #include <QUrlQuery>
 
+#ifdef Q_OS_OSX
+#include <QtGui/QDesktopServices>
+#endif
+
 static QStringList splitEmailAddressList(const QString &aStr)
 {
     // This is a copy of KPIM::splitEmailAddrList().
@@ -259,6 +263,11 @@
         return;
     }
 
+#ifdef Q_OS_OSX
+    Q_UNUSED(startup_id);
+    QDesktopServices::openUrl(QUrl(url));
+#else
+
     QStringList args;
     args << url;
     QString error;
@@ -345,6 +354,7 @@
                           i18n("Could not launch the browser:\n\n%1", error),
                           i18n("Could not launch Browser"));
     }
+#endif // Q_OS_OSX
 }
 
 void KToolInvocation::invokeTerminal(const QString &command,
Exit 1
--- src/orig.CMakeLists.txt	2015-12-04 21:13:14.000000000 +0100
+++ src/CMakeLists.txt	2015-12-07 22:29:06.000000000 +0100
@@ -86,6 +86,10 @@
     KF5::DBusAddons   # KDEInitInterface
     Qt5::Xml          # (for vfolder menu) QDomDocument
 )
+if(APPLE)
+    find_package(Qt5 ${REQUIRED_QT_VERSION} CONFIG REQUIRED Gui)
+    target_link_libraries(KF5Service PRIVATE Qt5::Gui)
+endif()
 
 set_target_properties(KF5Service PROPERTIES VERSION ${KSERVICE_VERSION_STRING}
                                             SOVERSION ${KSERVICE_SOVERSION}
