diff --git src/backends/polkit-1/Polkit1Backend.cpp src/backends/polkit-1/Polkit1Backend.cpp
index 170e646..b010299 100644
--- src/backends/polkit-1/Polkit1Backend.cpp
+++ src/backends/polkit-1/Polkit1Backend.cpp
@@ -92,7 +92,13 @@ void Polkit1Backend::preAuthAction(const QString &action, QWidget *parent)
     }
 
     // Are we running our KDE auth agent?
-    if (QDBusConnection::sessionBus().interface()->isServiceRegistered(QLatin1String("org.kde.polkit-kde-authentication-agent-1"))) {
+    bool agentRegistered = QDBusConnection::sessionBus().interface()->isServiceRegistered(QLatin1String("org.kde.polkit-kde-authentication-agent-1"));
+    bool legacyAgentRegistered = false;
+    if (!agentRegistered) {
+        // try the agent as it registered on Plasma4 desktops:
+        legacyAgentRegistered = QDBusConnection::sessionBus().interface()->isServiceRegistered(QLatin1String("org.kde.Polkit1AuthAgent"));
+    }
+    if (agentRegistered || legacyAgentRegistered) {
         // Check if we actually are entitled to use GUI capabilities
         if (qApp == nullptr || !qobject_cast<QApplication *>(qApp)) {
             qCDebug(KAUTH) << "Not streaming parent as we are on a TTY application";
@@ -103,7 +109,9 @@ void Polkit1Backend::preAuthAction(const QString &action, QWidget *parent)
 
         // Send it over the bus to our agent
         QDBusMessage methodCall =
-            QDBusMessage::createMethodCall(QLatin1String("org.kde.polkit-kde-authentication-agent-1"), QLatin1String("/org/kde/Polkit1AuthAgent"), QLatin1String("org.kde.Polkit1AuthAgent"),
+            QDBusMessage::createMethodCall(
+                QLatin1String(agentRegistered ? "org.kde.polkit-kde-authentication-agent-1" : "org.kde.Polkit1AuthAgent")
+                , QLatin1String("/org/kde/Polkit1AuthAgent"), QLatin1String("org.kde.Polkit1AuthAgent"),
                                            QLatin1String("setWIdForAction"));
 
         methodCall << action;
@@ -116,7 +124,7 @@ void Polkit1Backend::preAuthAction(const QString &action, QWidget *parent)
             qCWarning(KAUTH) << "ERROR while streaming the parent!!" << call.error();
         }
     } else {
-        qCDebug(KAUTH) << "KDE polkit agent appears too old or not registered on the bus";
+        qCWarning(KAUTH) << "KDE polkit agent appears too old or not registered on the bus";
     }
 }
 
