--- src/plugins/coreplugin/actionmanager/orig.actionmanager.cpp	2015-06-22 17:07:17.000000000 +0200
+++ src/plugins/coreplugin/actionmanager/actionmanager.cpp	2016-06-03 10:56:11.060858830 +0200
@@ -222,6 +222,10 @@
     if (it !=  d->m_idContainerMap.constEnd())
         return it.value();
 
+    if (!QGuiApplication::platformName().contains(QLatin1String("cocoa"))) {
+        QCoreApplication::setAttribute(Qt::AA_DontUseNativeMenuBar);
+    }
+    QCoreApplication::setAttribute(Qt::AA_DontShowIconsInMenus);
     QMenuBar *mb = new QMenuBar; // No parent (System menu bar on Mac OS X)
     mb->setObjectName(id.toString());
 
