diff --git a/src/launcher/apps-common.c b/src/launcher/apps-common.c
index bf5dd98..b23e70d 100644
--- a/src/launcher/apps-common.c
+++ b/src/launcher/apps-common.c
@@ -299,7 +299,7 @@ const GSList *get_apps_locations()
 
     apps_locations = load_locations_from_env(apps_locations, "XDG_DATA_DIRS", "applications", NULL);
 
-    apps_locations = g_slist_append(apps_locations, g_strdup("/usr/local/share/applications"));
+    apps_locations = g_slist_append(apps_locations, g_strdup("@PREFIX@/share/applications"));
     apps_locations = g_slist_append(apps_locations, g_strdup("/usr/share/applications"));
     apps_locations = g_slist_append(apps_locations, g_strdup("/opt/share/applications"));
 
diff --git a/src/launcher/icon-theme-common.c b/src/launcher/icon-theme-common.c
index 5a0ffc6..ddb2981 100644
--- a/src/launcher/icon-theme-common.c
+++ b/src/launcher/icon-theme-common.c
@@ -60,8 +60,8 @@ const GSList *get_icon_locations()
 
     icon_locations = load_locations_from_env(icon_locations, "XDG_DATA_DIRS", ".icons", ".pixmaps", NULL);
 
-    icon_locations = g_slist_append(icon_locations, g_strdup("/usr/local/share/icons"));
-    icon_locations = g_slist_append(icon_locations, g_strdup("/usr/local/share/pixmaps"));
+    icon_locations = g_slist_append(icon_locations, g_strdup("@PREFIX@/share/icons"));
+    icon_locations = g_slist_append(icon_locations, g_strdup("@PREFIX@/share/pixmaps"));
     icon_locations = g_slist_append(icon_locations, g_strdup("/usr/share/icons"));
     icon_locations = g_slist_append(icon_locations, g_strdup("/usr/share/pixmaps"));
     icon_locations = g_slist_append(icon_locations, g_strdup("/opt/share/icons"));
diff --git a/themes/horizontal-dark-opaque.tint2rc b/themes/horizontal-dark-opaque.tint2rc
index 56e33fa..0e9d05c 100644
--- a/themes/horizontal-dark-opaque.tint2rc
+++ b/themes/horizontal-dark-opaque.tint2rc
@@ -160,7 +160,7 @@ launcher_icon_theme_override = 0
 startup_notifications = 1
 launcher_tooltip = 1
 launcher_item_app = /usr/share/applications/tint2conf.desktop
-launcher_item_app = /usr/local/share/applications/tint2conf.desktop
+launcher_item_app = @PREFIX@/share/applications/tint2conf.desktop
 launcher_item_app = /usr/share/applications/firefox.desktop
 launcher_item_app = /usr/share/applications/iceweasel.desktop
 launcher_item_app = /usr/share/applications/chromium-browser.desktop
diff --git a/themes/horizontal-dark-transparent.tint2rc b/themes/horizontal-dark-transparent.tint2rc
index 4085d0a..bf94cf1 100644
--- a/themes/horizontal-dark-transparent.tint2rc
+++ b/themes/horizontal-dark-transparent.tint2rc
@@ -160,7 +160,7 @@ launcher_tooltip = 1
 launcher_item_app = /usr/share/applications/firefox.desktop
 launcher_item_app = /usr/share/applications/iceweasel.desktop
 launcher_item_app = /usr/share/applications/google-chrome.desktop
-launcher_item_app = /usr/local/share/applications/tint2conf.desktop
+launcher_item_app = @PREFIX@/share/applications/tint2conf.desktop
 
 #-------------------------------------
 # Clock
diff --git a/themes/horizontal-light-transparent.tint2rc b/themes/horizontal-light-transparent.tint2rc
index 429be06..5f0d436 100644
--- a/themes/horizontal-light-transparent.tint2rc
+++ b/themes/horizontal-light-transparent.tint2rc
@@ -160,7 +160,7 @@ launcher_icon_theme_override = 0
 startup_notifications = 1
 launcher_tooltip = 1
 launcher_item_app = /usr/share/applications/tint2conf.desktop
-launcher_item_app = /usr/local/share/applications/tint2conf.desktop
+launcher_item_app = @PREFIX@/share/applications/tint2conf.desktop
 launcher_item_app = /usr/share/applications/firefox.desktop
 launcher_item_app = /usr/share/applications/iceweasel.desktop
 launcher_item_app = /usr/share/applications/chromium-browser.desktop
diff --git a/themes/tint2rc b/themes/tint2rc
index b9dc345..9a57ab4 100644
--- a/themes/tint2rc
+++ b/themes/tint2rc
@@ -137,7 +137,7 @@ launcher_icon_theme_override = 0
 startup_notifications = 1
 launcher_tooltip = 1
 launcher_item_app = /usr/share/applications/tint2conf.desktop
-launcher_item_app = /usr/local/share/applications/tint2conf.desktop
+launcher_item_app = @PREFIX@/share/applications/tint2conf.desktop
 launcher_item_app = /usr/share/applications/firefox.desktop
 launcher_item_app = /usr/share/applications/iceweasel.desktop
 launcher_item_app = /usr/share/applications/chromium-browser.desktop
diff --git a/themes/vertical-dark-opaque.tint2rc b/themes/vertical-dark-opaque.tint2rc
index 5f36d2a..1d28805 100644
--- a/themes/vertical-dark-opaque.tint2rc
+++ b/themes/vertical-dark-opaque.tint2rc
@@ -160,7 +160,7 @@ launcher_tooltip = 1
 launcher_item_app = /usr/share/applications/firefox.desktop
 launcher_item_app = /usr/share/applications/iceweasel.desktop
 launcher_item_app = /usr/share/applications/google-chrome.desktop
-launcher_item_app = /usr/local/share/applications/tint2conf.desktop
+launcher_item_app = @PREFIX@/share/applications/tint2conf.desktop
 
 #-------------------------------------
 # Clock
diff --git a/themes/vertical-dark-transparent.tint2rc b/themes/vertical-dark-transparent.tint2rc
index 4c309bf..c26771f 100644
--- a/themes/vertical-dark-transparent.tint2rc
+++ b/themes/vertical-dark-transparent.tint2rc
@@ -160,7 +160,7 @@ launcher_tooltip = 1
 launcher_item_app = /usr/share/applications/firefox.desktop
 launcher_item_app = /usr/share/applications/iceweasel.desktop
 launcher_item_app = /usr/share/applications/google-chrome.desktop
-launcher_item_app = /usr/local/share/applications/tint2conf.desktop
+launcher_item_app = @PREFIX@/share/applications/tint2conf.desktop
 
 #-------------------------------------
 # Clock
diff --git a/themes/vertical-light-opaque.tint2rc b/themes/vertical-light-opaque.tint2rc
index 9980b98..4a0b387 100644
--- a/themes/vertical-light-opaque.tint2rc
+++ b/themes/vertical-light-opaque.tint2rc
@@ -170,7 +170,7 @@ launcher_tooltip = 1
 launcher_item_app = /usr/share/applications/firefox.desktop
 launcher_item_app = /usr/share/applications/iceweasel.desktop
 launcher_item_app = /usr/share/applications/google-chrome.desktop
-launcher_item_app = /usr/local/share/applications/tint2conf.desktop
+launcher_item_app = @PREFIX@/share/applications/tint2conf.desktop
 
 #-------------------------------------
 # Clock
diff --git a/themes/vertical-light-transparent.tint2rc b/themes/vertical-light-transparent.tint2rc
index 0fbe6a9..e754c63 100644
--- a/themes/vertical-light-transparent.tint2rc
+++ b/themes/vertical-light-transparent.tint2rc
@@ -160,7 +160,7 @@ launcher_icon_theme_override = 0
 startup_notifications = 1
 launcher_tooltip = 1
 launcher_item_app = /usr/share/applications/tint2conf.desktop
-launcher_item_app = /usr/local/share/applications/tint2conf.desktop
+launcher_item_app = @PREFIX@/share/applications/tint2conf.desktop
 launcher_item_app = /usr/share/applications/firefox.desktop
 launcher_item_app = /usr/share/applications/iceweasel.desktop
 launcher_item_app = /usr/share/applications/chromium-browser.desktop
