diff --git kate/main.cpp kate/main.cpp
index 1313b4cd586a60351576474a217c522aabc26628..89f1df0a4949fb15309a1ba4ef5c8d7ccfcb69da 100644
--- kate/main.cpp
+++ kate/main.cpp
@@ -79,6 +79,11 @@ int main(int argc, char **argv)
      */
     QCoreApplication::setAttribute(Qt::AA_UseHighDpiPixmaps, true);
     QCoreApplication::setAttribute(Qt::AA_EnableHighDpiScaling, true);
+#ifdef Q_OS_OSX
+    if (!QGuiApplication::platformName().contains(QLatin1String("cocoa"))) {
+        QCoreApplication::setAttribute(Qt::AA_DontUseNativeMenuBar);
+    }
+#endif
 
     /**
      * Create application first
diff --git kwrite/main.cpp kwrite/main.cpp
index 51ea46ac465269c71d84f65164d5c6cfb393bad7..ef4db7ab93fe32da77715585feebc5b81567bf82 100644
--- kwrite/main.cpp
+++ kwrite/main.cpp
@@ -69,6 +69,11 @@ extern "C" Q_DECL_EXPORT int main(int argc, char **argv)
      */
     QCoreApplication::setAttribute(Qt::AA_UseHighDpiPixmaps, true);
     QCoreApplication::setAttribute(Qt::AA_EnableHighDpiScaling, true);
+#ifdef Q_OS_OSX
+    if (!QGuiApplication::platformName().contains(QLatin1String("cocoa"))) {
+        QCoreApplication::setAttribute(Qt::AA_DontUseNativeMenuBar);
+    }
+#endif
 
     /**
      * Create application first
