diff --git addons/kate-ctags/CMakeLists.txt addons/kate-ctags/CMakeLists.txt
index f85359cd64e0100f85adda6b31f6331be2e5bee9..6ea0a67f3d5ede67ac14daea60e3d1bd5234703c 100644
--- addons/kate-ctags/CMakeLists.txt
+++ addons/kate-ctags/CMakeLists.txt
@@ -1,5 +1,7 @@
 add_definitions(-DTRANSLATION_DOMAIN=\"kate-ctags-plugin\")
 
+include(ECMQtDeclareLoggingCategory)
+
 include_directories( ${CMAKE_CURRENT_BINARY_DIR} )
 
 set(ctagsplugin_SRC
@@ -10,6 +12,12 @@ set(ctagsplugin_SRC
     kate_ctags_plugin.cpp
 )
 
+ecm_qt_declare_logging_category(ctagsplugin_SRC
+    HEADER kate_ctags_debug.h
+    IDENTIFIER KTECTAGS
+    CATEGORY_NAME "katectagsplugin"
+)
+
 ki18n_wrap_ui(ctagsplugin_SRC kate_ctags.ui CTagsGlobalConfig.ui)
 
 # resource for ui file and stuff
diff --git addons/kate-ctags/kate_ctags_view.cpp addons/kate-ctags/kate_ctags_view.cpp
index e7e8ec4220e28258d7a60b818031d1b3c37400cc..b39bdd62be292e00752c26e94f564c55a076de62 100644
--- addons/kate-ctags/kate_ctags_view.cpp
+++ addons/kate-ctags/kate_ctags_view.cpp
@@ -20,6 +20,7 @@
  */
 
 #include "kate_ctags_view.h"
+#include "kate_ctags_debug.h"
 
 #include <QFileInfo>
 #include <QFileDialog>
@@ -270,7 +271,7 @@ void KateCTagsView::gotoTagForTypes(const QString &word, const QStringList &type
     Tags::TagList list = Tags::getMatches(m_ctagsUi.tagsFile->text(), word, false, types);
     if (list.size() == 0) list = Tags::getMatches(m_commonDB, word, false, types);
 
-    //qDebug() << "found" << list.count() << word << types;
+    //qCDebug(KTECTAGS) << "found" << list.count() << word << types;
     setNewLookupText(word);
 
     if ( list.count() < 1) {
@@ -348,7 +349,7 @@ QString KateCTagsView::currentWord( )
 {
     KTextEditor::View *kv = m_mWin->activeView();
     if (!kv) {
-        qDebug() << "no KTextEditor::View" << endl;
+        qCDebug(KTECTAGS) << "no KTextEditor::View" << endl;
         return QString();
     }
 
@@ -357,7 +358,7 @@ QString KateCTagsView::currentWord( )
     }
 
     if (!kv->cursorPosition().isValid()) {
-        qDebug() << "cursor not valid!" << endl;
+        qCDebug(KTECTAGS) << "cursor not valid!" << endl;
         return QString();
     }
 
@@ -382,7 +383,7 @@ QString KateCTagsView::currentWord( )
         endPos++;
     }
     if  (startPos == endPos) {
-        qDebug() << "no word found!" << endl;
+        qCDebug(KTECTAGS) << "no word found!" << endl;
         return QString();
     }
 
@@ -396,7 +397,7 @@ QString KateCTagsView::currentWord( )
         linestr.remove(0, 1);
     }
 
-    //qDebug() << linestr;
+    //qCDebug(KTECTAGS) << linestr;
     return linestr;
 }
 
@@ -440,7 +441,7 @@ void KateCTagsView::jumpToTag(const QString &file, const QString &pattern, const
 
     // open/activate the new file
     QFileInfo fInfo(file);
-    //qDebug() << pattern << file << fInfo.absoluteFilePath();
+    //qCDebug(KTECTAGS) << pattern << file << fInfo.absoluteFilePath();
     m_mWin->openUrl(QUrl::fromLocalFile(fInfo.absoluteFilePath()));
 
     // any view active?
