diff --git addons/kate-ctags/kate_ctags_plugin.cpp addons/kate-ctags/kate_ctags_plugin.cpp
index 7eb87dadab79ae3108361e52e75030252ab408c2..384b99beb5859c47b6527fc28a6362c527f39126 100644
--- addons/kate-ctags/kate_ctags_plugin.cpp
+++ addons/kate-ctags/kate_ctags_plugin.cpp
@@ -219,7 +219,7 @@ void KateCTagsConfigPage::updateGlobalDB()
         targets += target + QLatin1Char(' ');
     }
 
-    QString file = QStandardPaths::writableLocation(QStandardPaths::DataLocation) + QLatin1String("/katectags");
+    QString file = QStandardPaths::writableLocation(QStandardPaths::GenericDataLocation) + QLatin1String("/kate/katectags");
     QDir().mkpath(file);
     file += QLatin1String("/common_db");
 
diff --git addons/kate-ctags/kate_ctags_view.cpp addons/kate-ctags/kate_ctags_view.cpp
index 1728bcd700c1b6b0971fe7ecb6bbb0ee9c4aab74..b91fef20a962d59679130d4a3fab96554ef560af 100644
--- addons/kate-ctags/kate_ctags_view.cpp
+++ addons/kate-ctags/kate_ctags_view.cpp
@@ -20,6 +20,7 @@
  */
 
 #include "kate_ctags_view.h"
+#include "kate_ctags_plugin.h"
 
 #include <QFileInfo>
 #include <QFileDialog>
@@ -48,6 +49,7 @@ KateCTagsView::KateCTagsView(KTextEditor::Plugin *plugin, KTextEditor::MainWindo
                                         QIcon::fromTheme(QStringLiteral("application-x-ms-dos-executable")),
                                         i18n("CTags"));
     m_mWin = mainWin;
+    QWidget *ctagsWidget = new QWidget(m_toolView);
 
     QAction *back = actionCollection()->addAction(QLatin1String("ctags_return_step"));
     back->setText(i18n("Jump back one step"));
@@ -65,6 +67,16 @@ KateCTagsView::KateCTagsView(KTextEditor::Plugin *plugin, KTextEditor::MainWindo
     lookup->setText(i18n("Lookup Current Text"));
     connect(lookup, &QAction::triggered, this, &KateCTagsView::lookupTag);
 
+    QAction *updateDB = actionCollection()->addAction(QLatin1String("ctags_update_global_db"));
+    updateDB->setText(i18n("Update the Database"));
+    connect(updateDB, &QAction::triggered, this, [this, plugin, mainWin] (bool) {
+            KateCTagsPlugin *p = dynamic_cast<KateCTagsPlugin*>(plugin);
+            QDialog *confWin = new QDialog(mainWin->window());
+            auto confPage = p->configPage(0, confWin);
+            confPage->show();
+            confWin->exec();
+        });
+
     // popup menu
     m_menu = new KActionMenu(i18n("CTags"), this);
     actionCollection()->addAction(QLatin1String("popup_ctags"), m_menu);
@@ -75,7 +87,6 @@ KateCTagsView::KateCTagsView(KTextEditor::Plugin *plugin, KTextEditor::MainWindo
 
     connect(m_menu->menu(), &QMenu::aboutToShow, this, &KateCTagsView::aboutToShow);
 
-    QWidget *ctagsWidget = new QWidget(m_toolView);
     m_ctagsUi.setupUi(ctagsWidget);
     m_ctagsUi.cmdEdit->setText(DEFAULT_CTAGS_CMD);
 
@@ -117,7 +128,7 @@ KateCTagsView::KateCTagsView(KTextEditor::Plugin *plugin, KTextEditor::MainWindo
 
     m_mWin->guiFactory()->addClient(this);
 
-    m_commonDB = QStandardPaths::writableLocation(QStandardPaths::DataLocation) + QLatin1String("/katectags/common_db");
+    m_commonDB = QStandardPaths::writableLocation(QStandardPaths::GenericDataLocation) + QLatin1String("/kate/katectags/common_db");
 }
 
 
@@ -493,7 +504,7 @@ void KateCTagsView::updateSessionDB()
       targets += target + QLatin1Char(' ');
     }
 
-    QString pluginFolder = QStandardPaths::writableLocation(QStandardPaths::DataLocation) + QLatin1String("/katectags");
+    QString pluginFolder = QStandardPaths::writableLocation(QStandardPaths::GenericDataLocation) + QLatin1String("/kate/katectags");
     QDir().mkpath(pluginFolder);
 
     if (m_ctagsUi.tagsFile->text().isEmpty()) {
diff --git addons/kate-ctags/katectagsplugin.desktop addons/kate-ctags/katectagsplugin.desktop
index e47ab2d7c63735364da336841c048121c7a39a22..7ac95dbde84230a14756aa64371ae414e4effeff 100644
--- addons/kate-ctags/katectagsplugin.desktop
+++ addons/kate-ctags/katectagsplugin.desktop
@@ -1,6 +1,7 @@
 [Desktop Entry]
 Type=Service
-ServiceTypes=KTextEditor/Plugin
+ServiceTypes=KTextEditor/Plugin,KDevelop/Plugin
+X-KDE-ServiceTypes=KTextEditor/Plugin,KDevelop/Plugin
 X-KDE-Library=katectagsplugin
 Name=CTags
 Name[ar]=وسوم سي
diff --git addons/kate-ctags/ui.rc addons/kate-ctags/ui.rc
index 6f73ae5076b2e8788d220745c2a3ccbe11566520..deb99ff7d08f9c5b2ace384a5ba5ede987a77317 100644
--- addons/kate-ctags/ui.rc
+++ addons/kate-ctags/ui.rc
@@ -9,6 +9,8 @@
       <Action name="ctags_lookup_current_as_declaration"/>
       <Action name="ctags_lookup_current_as_definition"/>
       <Action name="ctags_jump_to_next_matcht"/>
+      <Separator/>
+      <Action name="ctags_update_global_db"/>
     </Menu>
   </MenuBar>
   <Menu name="ktexteditor_popup" noMerge="1">
