diff --git addons/lspclient/lspclientpluginview.cpp addons/lspclient/lspclientpluginview.cpp
index b3ac71b1e..430a5ce74 100644
--- addons/lspclient/lspclientpluginview.cpp
+++ addons/lspclient/lspclientpluginview.cpp
@@ -231,7 +231,7 @@ public:
 
         connect(m_mainWindow, &KTextEditor::MainWindow::viewChanged, this, &self_type::updateState);
         connect(m_mainWindow, &KTextEditor::MainWindow::unhandledShortcutOverride, this, &self_type::handleEsc);
-        connect(m_serverManager.get(), &LSPClientServerManager::serverChanged, this, &self_type::updateState);
+        connect(m_serverManager.data(), &LSPClientServerManager::serverChanged, this, &self_type::updateState);
 
         m_findDef = actionCollection()->addAction(QStringLiteral("lspclient_find_definition"), this, &self_type::goToDefinition);
         m_findDef->setText(i18n("Go to Definition"));
@@ -298,7 +298,7 @@ public:
                                                  KTextEditor::MainWindow::Bottom,
                                                  QIcon::fromTheme(QStringLiteral("application-x-ms-dos-executable")),
                                                  i18n("LSP Client")));
-        m_tabWidget = new QTabWidget(m_toolView.get());
+        m_tabWidget = new QTabWidget(m_toolView.data());
         m_tabWidget->setFocusPolicy(Qt::NoFocus);
         m_tabWidget->setTabsClosable(true);
         KAcceleratorManager::setNoAccel(m_tabWidget);
@@ -364,7 +364,7 @@ public:
         KTextEditor::View *activeView = m_mainWindow->activeView();
         auto server = m_serverManager->findServer(activeView);
         if (server)
-            m_serverManager->restart(server.get());
+            m_serverManager->restart(server.data());
     }
 
     void restartAll()
@@ -710,7 +710,7 @@ public:
 
         // activate the resulting tab
         m_tabWidget->setCurrentIndex(index);
-        m_mainWindow->showToolView(m_toolView.get());
+        m_mainWindow->showToolView(m_toolView.data());
     }
 
     void showMessage(const QString & text, KTextEditor::Message::MessageType level)
@@ -734,7 +734,7 @@ public:
             if (!m_ranges.empty()) {
                 clearAllLocationMarks();
             } else if (m_toolView->isVisible()) {
-                m_mainWindow->hideToolView(m_toolView.get());
+                m_mainWindow->hideToolView(m_toolView.data());
             }
         }
     }
@@ -889,7 +889,7 @@ public:
             if (line == mark.line && m_diagnosticsTree) {
                 m_diagnosticsTree->scrollToItem(item);
                 m_tabWidget->setCurrentWidget(m_diagnosticsTree);
-                m_mainWindow->showToolView(m_toolView.get());
+                m_mainWindow->showToolView(m_toolView.data());
                 handled = true;
                 break;
             }
@@ -954,7 +954,7 @@ public:
             hoverEnabled = caps.hoverProvider;
             highlightEnabled = caps.documentHighlightProvider;
 
-            connect(server.get(), &LSPClientServer::publishDiagnostics,
+            connect(server.data(), &LSPClientServer::publishDiagnostics,
                 this, &self_type::onDiagnostics, Qt::UniqueConnection);
         }
 
@@ -977,7 +977,7 @@ public:
         m_completion->setServer(server);
         if (m_complDocOn)
             m_completion->setSelectedDocumentation(m_complDocOn->isChecked());
-        updateCompletion(activeView, server.get());
+        updateCompletion(activeView, server.data());
 
         // update marks if applicable
         if (m_markTree && activeView)
@@ -1005,7 +1005,7 @@ public:
 
         if (!registered && server && server->capabilities().completionProvider.provider) {
             qCInfo(LSPCLIENT) << "registering cci";
-            cci->registerCompletionModel(m_completion.get());
+            cci->registerCompletionModel(m_completion.data());
             m_completionViews.insert(view);
 
             connect(view, &KTextEditor::View::destroyed, this,
@@ -1015,7 +1015,7 @@ public:
 
         if (registered && !server) {
             qCInfo(LSPCLIENT) << "unregistering cci";
-            cci->unregisterCompletionModel(m_completion.get());
+            cci->unregisterCompletionModel(m_completion.data());
             m_completionViews.remove(view);
         }
     }
diff --git addons/lspclient/lspclientservermanager.cpp addons/lspclient/lspclientservermanager.cpp
index 69c178cc6..0e6a152ac 100644
--- addons/lspclient/lspclientservermanager.cpp
+++ addons/lspclient/lspclientservermanager.cpp
@@ -164,7 +164,7 @@ public:
         int count = 0;
         for (const auto & el: m_servers) {
             for (const auto & s: el) {
-                disconnect(s.get(), nullptr, this, nullptr);
+                disconnect(s.data(), nullptr, this, nullptr);
                 if (s->state() != LSPClientServer::State::None) {
                     auto handler = [&q, &count, s] () {
                         if (s->state() != LSPClientServer::State::None) {
@@ -173,7 +173,7 @@ public:
                             }
                         }
                     };
-                    connect(s.get(), &LSPClientServer::stateChanged, this, handler);
+                    connect(s.data(), &LSPClientServer::stateChanged, this, handler);
                     ++count;
                     s->stop(-1, -1);
                 }
@@ -222,7 +222,7 @@ public:
         // find entry for server(s) and move out
         for (auto & m : m_servers) {
             for (auto it = m.begin() ; it != m.end(); ) {
-                if (!server || it->get() == server) {
+                if (!server || it->data() == server) {
                     servers.push_back(*it);
                     it = m.erase(it);
                 } else {
@@ -253,7 +253,7 @@ private:
         // close docs
         for (const auto & server : servers) {
             // controlling server here, so disable usual state tracking response
-            disconnect(server.get(), nullptr, this, nullptr);
+            disconnect(server.data(), nullptr, this, nullptr);
             for (auto it = m_docs.begin(); it != m_docs.end(); ) {
                 auto &item = it.value();
                 if (item.server == server) {
@@ -397,7 +397,7 @@ private:
                 init = applyCustomInit(init, cmdline, root, projectView);
                 server.reset(new LSPClientServer(cmdline, root, init));
                 m_servers[root][mode] = server;
-                connect(server.get(), &LSPClientServer::stateChanged,
+                connect(server.data(), &LSPClientServer::stateChanged,
                     this, &self_type::onStateChanged, Qt::UniqueConnection);
                 if (!server->start()) {
                     showMessage(i18n("Failed to start server: %1", cmdline.join(QLatin1Char(' '))),
diff --git addons/lspclient/lspclientsymbolview.cpp addons/lspclient/lspclientsymbolview.cpp
index 56adcbba7..8b95093de 100644
--- addons/lspclient/lspclientsymbolview.cpp
+++ addons/lspclient/lspclientsymbolview.cpp
@@ -85,7 +85,7 @@ public:
                                                       QIcon::fromTheme(QStringLiteral("code-context")),
                                                       i18n("LSP Client Symbol Outline")));
 
-        QWidget *container = new QWidget(m_toolview.get());
+        QWidget *container = new QWidget(m_toolview.data());
         QHBoxLayout *layout = new QHBoxLayout(container);
 
         m_symbols = new QTreeWidget();
@@ -132,7 +132,7 @@ public:
         connect(&m_currentItemTimer, &QTimer::timeout, this, &self_type::updateCurrentTreeItem);
 
         connect(m_mainWindow, &KTextEditor::MainWindow::viewChanged, this, &self_type::viewChanged);
-        connect(m_serverManager.get(), &LSPClientServerManager::serverChanged, this, &self_type::refresh);
+        connect(m_serverManager.data(), &LSPClientServerManager::serverChanged, this, &self_type::refresh);
 
         // initial trigger
         configUpdated();
