diff --git addons/kate-ctags/kate_ctags_view.cpp addons/kate-ctags/kate_ctags_view.cpp
index f8a5573ac6b941ec598112da94513dc4e2c5e380..c195b9364fa65e92844953e246fcbd9efd735cb9 100644
--- addons/kate-ctags/kate_ctags_view.cpp
+++ addons/kate-ctags/kate_ctags_view.cpp
@@ -98,7 +98,7 @@ KateCTagsView::KateCTagsView(KTextEditor::Plugin *plugin, KTextEditor::MainWindo
 
     connect(m_menu->menu(), &QMenu::aboutToShow, this, &KateCTagsView::aboutToShow);
 
-    QWidget *ctagsWidget = new QWidget(m_toolView);
+    QWidget *ctagsWidget = new QWidget(m_toolView.data());
     m_ctagsUi.setupUi(ctagsWidget);
     m_ctagsUi.cmdEdit->setText(DEFAULT_CTAGS_CMD);
 
@@ -136,6 +136,7 @@ KateCTagsView::KateCTagsView(KTextEditor::Plugin *plugin, KTextEditor::MainWindo
 
     connect(m_mWin, &KTextEditor::MainWindow::unhandledShortcutOverride, this, &KateCTagsView::handleEsc);
 
+    m_toolView->layout()->addWidget(ctagsWidget);
     m_toolView->installEventFilter(this);
 
     m_mWin->guiFactory()->addClient(this);
@@ -151,7 +152,9 @@ KateCTagsView::~KateCTagsView()
         m_mWin->guiFactory()->removeClient( this );
     }
 
-    delete m_toolView;
+    if (m_toolView) {
+        delete m_toolView;
+    }
 }
 
 /******************************************************************/
diff --git addons/kate-ctags/kate_ctags_view.h addons/kate-ctags/kate_ctags_view.h
index 752cef220c1134ee424baf8c9ef01e5827b7b855..4200f57bf4d986a6ae3f993a40d0ab77233042ba 100644
--- addons/kate-ctags/kate_ctags_view.h
+++ addons/kate-ctags/kate_ctags_view.h
@@ -99,7 +99,7 @@ private:
     
 
     QPointer<KTextEditor::MainWindow> m_mWin;
-    QWidget               *m_toolView;
+    QPointer<QWidget>      m_toolView;
     Ui::kateCtags          m_ctagsUi;
 
     QPointer<KActionMenu>  m_menu;
