diff --git addons/kate-ctags/kate_ctags_plugin.cpp addons/kate-ctags/kate_ctags_plugin.cpp
index 7eb87dadab79ae3108361e52e75030252ab408c2..bb2f5c817b504a744dc790cf80afd22d5da49c6a 100644
--- addons/kate-ctags/kate_ctags_plugin.cpp
+++ addons/kate-ctags/kate_ctags_plugin.cpp
@@ -129,6 +129,7 @@ void KateCTagsConfigPage::apply()
         nr = QStringLiteral("%1").arg(i,3);
         config.writeEntry(QStringLiteral("GlobalTarget_")+nr, m_confUi.targetList->item(i)->text());
     }
+    qWarning() << "Config file:" << config.config()->name();
     config.sync();
 }
 
diff --git addons/kate-ctags/kate_ctags_view.cpp addons/kate-ctags/kate_ctags_view.cpp
index b39bdd62be292e00752c26e94f564c55a076de62..e213391fdf3ce23ad33bb8efcfe9071a8d3986e4 100644
--- addons/kate-ctags/kate_ctags_view.cpp
+++ addons/kate-ctags/kate_ctags_view.cpp
@@ -20,6 +20,7 @@
  */
 
 #include "kate_ctags_view.h"
+#include "kate_ctags_plugin.h"
 #include "kate_ctags_debug.h"
 
 #include <QFileInfo>
@@ -66,6 +67,28 @@ KateCTagsView::KateCTagsView(KTextEditor::Plugin *plugin, KTextEditor::MainWindo
     lookup->setText(i18n("Lookup Current Text"));
     connect(lookup, &QAction::triggered, this, &KateCTagsView::lookupTag);
 
+    QAction *updateDB = actionCollection()->addAction(QLatin1String("ctags_update_global_db"));
+    updateDB->setText(i18n("Configure ..."));
+    connect(updateDB, &QAction::triggered, this, [plugin, mainWin] (bool) {
+            KateCTagsPlugin *p = dynamic_cast<KateCTagsPlugin*>(plugin);
+            QDialog *confWin = new QDialog(mainWin->window());
+            auto confPage = p->configPage(0, nullptr);
+            auto controls = new QDialogButtonBox(QDialogButtonBox::Ok|QDialogButtonBox::Cancel, Qt::Horizontal, nullptr);
+            connect(confWin, &QDialog::accepted, confPage, &KTextEditor::ConfigPage::apply);
+            connect(controls, &QDialogButtonBox::accepted, confWin, &QDialog::accept);
+            connect(controls, &QDialogButtonBox::rejected, confWin, &QDialog::reject);
+            auto layout = new QVBoxLayout;
+            layout->addWidget(confPage);
+            layout->addWidget(controls);
+            confWin->setLayout(layout);
+            confWin->show();
+            confWin->exec();
+            confWin->deleteLater();
+            confPage->deleteLater();
+            controls->deleteLater();
+            layout->deleteLater();
+    });
+
     // popup menu
     m_menu = new KActionMenu(i18n("CTags"), this);
     actionCollection()->addAction(QLatin1String("popup_ctags"), m_menu);
@@ -125,7 +148,9 @@ KateCTagsView::KateCTagsView(KTextEditor::Plugin *plugin, KTextEditor::MainWindo
 /******************************************************************/
 KateCTagsView::~KateCTagsView()
 {
-    m_mWin->guiFactory()->removeClient( this );
+    if (m_mWin && m_mWin->guiFactory()) {
+        m_mWin->guiFactory()->removeClient( this );
+    }
 
     delete m_toolView;
 }
diff --git addons/kate-ctags/kate_ctags_view.h addons/kate-ctags/kate_ctags_view.h
index 75e563f4d939f96c60b93f35a8b0a74d2a533c05..752cef220c1134ee424baf8c9ef01e5827b7b855 100644
--- addons/kate-ctags/kate_ctags_view.h
+++ addons/kate-ctags/kate_ctags_view.h
@@ -98,7 +98,7 @@ private:
     void jumpToTag(const QString &file, const QString &pattern, const QString &word);
     
 
-    KTextEditor::MainWindow *m_mWin;
+    QPointer<KTextEditor::MainWindow> m_mWin;
     QWidget               *m_toolView;
     Ui::kateCtags          m_ctagsUi;
 
diff --git addons/kate-ctags/katectagsplugin.desktop addons/kate-ctags/katectagsplugin.desktop
index ef39656542a31c164e43d2f811e397dc72844b6b..660b14bb5f8a40916ed42375bf0785e75fd14ec2 100644
--- addons/kate-ctags/katectagsplugin.desktop
+++ addons/kate-ctags/katectagsplugin.desktop
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Type=Service
-ServiceTypes=KTextEditor/Plugin
+ServiceTypes=KTextEditor/Plugin,KDevelop/Plugin
 X-KDE-Library=katectagsplugin
 Name=CTags
 Name[ar]=وسوم سي
diff --git addons/kate-ctags/ui.rc addons/kate-ctags/ui.rc
index 6f73ae5076b2e8788d220745c2a3ccbe11566520..41ee008ccd21bf2ca176191f4bf54f5755f084b2 100644
--- addons/kate-ctags/ui.rc
+++ addons/kate-ctags/ui.rc
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!DOCTYPE kpartgui>
-<gui name="katectags-plugin" library="katectagsplugin" version="3" translationDomain="kate-ctags-plugin">
+<gui name="katectags-plugin" library="katectagsplugin" version="4" translationDomain="kate-ctags-plugin">
   <MenuBar>
     <Menu name="CTags Menubar">
       <text>CTags</text>
@@ -9,6 +9,8 @@
       <Action name="ctags_lookup_current_as_declaration"/>
       <Action name="ctags_lookup_current_as_definition"/>
       <Action name="ctags_jump_to_next_matcht"/>
+      <Separator/>
+      <Action name="ctags_update_global_db"/>
     </Menu>
   </MenuBar>
   <Menu name="ktexteditor_popup" noMerge="1">
