--- kate/orig.katepluginmanager.cpp	2016-01-03 00:00:57.000000000 +0100
+++ kate/katepluginmanager.cpp	2016-01-03 00:03:03.000000000 +0100
@@ -48,8 +48,8 @@
 
 KatePluginManager::~KatePluginManager()
 {
-    // than unload the plugins
-    unloadAllPlugins();
+    // don't unload the plugins because that's prone to crashing
+//     unloadAllPlugins();
 }
 
 void KatePluginManager::setupPluginList()
