diff --git kate/katepluginmanager.cpp kate/katepluginmanager.cpp
index 5daef486f41375f89fe6d83b8ff008d9b3bc04c6..0c3bc91e571a9518e95676eccb462a2b2e45d598 100644
--- kate/katepluginmanager.cpp
+++ kate/katepluginmanager.cpp
@@ -55,7 +55,8 @@ KatePluginManager::KatePluginManager(QObject *parent)
 
 KatePluginManager::~KatePluginManager()
 {
-    unloadAllPlugins();
+    // don't unload the plugins because that's prone to crashing
+//     unloadAllPlugins();
 }
 
 void KatePluginManager::setupPluginList()
