--- umbrello/orig.CMakeLists.txt	2017-06-05 21:12:48.000000000 +0200
+++ umbrello/CMakeLists.txt	2017-06-16 22:03:40.362672455 +0200
@@ -572,6 +572,7 @@
     )
     ecm_add_app_icon(umbrello_SRCS
         ICONS
+        ${CMAKE_CURRENT_SOURCE_DIR}/pics/global/sc-apps-umbrello.svgz
         ${CMAKE_CURRENT_SOURCE_DIR}/pics/global/16-apps-umbrello.png
         ${CMAKE_CURRENT_SOURCE_DIR}/pics/global/22-apps-umbrello.png
         ${CMAKE_CURRENT_SOURCE_DIR}/pics/global/32-apps-umbrello.png
