--- unittests/orig.CMakeLists.txt	2016-10-07 01:13:32.000000000 +0200
+++ unittests/CMakeLists.txt	2016-10-30 10:01:00.369962632 +0100
@@ -113,7 +113,7 @@
 ecm_add_executable(TEST_umlroledialog ${TEST_umlroledialog_SRCS})
 target_link_libraries(TEST_umlroledialog ${LIBS})
 
-find_package(LLVM CONFIG)
+# find_package(LLVM CONFIG)
 if(LLVM_FOUND)
     message(STATUS "Found LLVM ${LLVM_PACKAGE_VERSION}")
     message(STATUS "Using LLVMConfig.cmake in: ${LLVM_DIR}")
