--- umbrello/orig.CMakeLists.txt	2016-02-03 21:31:35.000000000 +0100
+++ umbrello/CMakeLists.txt	2016-02-03 21:31:49.000000000 +0100
@@ -534,14 +534,10 @@
         ${LIBXSLT_LIBRARIES}
         ${LIBXML2_LIBRARIES}
     )
+    file(GLOB ICONS_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/pics/global/*-apps-umbrello.png")
     ecm_add_app_icon(umbrello_SRCS
         ICONS
-        ${CMAKE_CURRENT_SOURCE_DIR}/pics/global/hi16-apps-umbrello.png
-        ${CMAKE_CURRENT_SOURCE_DIR}/pics/global/hi22-apps-umbrello.png
-        ${CMAKE_CURRENT_SOURCE_DIR}/pics/global/hi32-apps-umbrello.png
-        ${CMAKE_CURRENT_SOURCE_DIR}/pics/global/hi48-apps-umbrello.png
-        ${CMAKE_CURRENT_SOURCE_DIR}/pics/global/hi64-apps-umbrello.png
-        ${CMAKE_CURRENT_SOURCE_DIR}/pics/global/hi128-apps-umbrello.png
+        ${ICONS_SRCS}
     )
     add_executable(umbrello ${umbrello_SRCS})
     target_link_libraries(umbrello libumbrello)
