--- unittests/CMakeLists.txt.orig	2016-12-06 21:07:36.000000000 +0100
+++ unittests/CMakeLists.txt	2016-12-24 20:14:07.000000000 +0100
@@ -115,8 +115,8 @@
 ecm_add_executable(testumlroledialog ${testumlroledialog_SRCS})
 target_link_libraries(testumlroledialog ${LIBS})
 
-find_package(LLVM CONFIG)
-find_package(CLANG QUIET)
+# find_package(LLVM CONFIG)
+# find_package(CLANG QUIET)
 if(LLVM_FOUND)
     message(STATUS "Found LLVM ${LLVM_PACKAGE_VERSION}")
     message(STATUS "Using LLVMConfig.cmake in: ${LLVM_DIR}")
