--- umbrello/codeimport/orig.cppimport.cpp	2016-01-07 00:18:00.000000000 +0100
+++ umbrello/codeimport/cppimport.cpp	2016-02-03 20:49:49.000000000 +0100
@@ -115,6 +115,7 @@
     ms_driver->addIncludePath(QLatin1String("/usr/include"));
     ms_driver->addIncludePath(QLatin1String("/usr/include/c++"));
     ms_driver->addIncludePath(QLatin1String("/usr/include/g++"));
+    ms_driver->addIncludePath(QLatin1String("@PREFIX@/include"));
     ms_driver->addIncludePath(QLatin1String("/usr/local/include"));
     const QStringList incPathList = Import_Utils::includePathList();
     if (incPathList.count()) {
--- umbrello/docgenerators/orig.xmi2docbook.sh	2016-02-03 21:17:29.000000000 +0100
+++ umbrello/docgenerators/xmi2docbook.sh	2016-02-03 21:17:42.000000000 +0100
@@ -1,9 +1,9 @@
 #! /bin/bash
 
 echo Converting from XMI to docbook...
-java -cp /usr/share/java/xalan-j2-2.6.0.jar org.apache.xalan.xslt.Process -xml -in $1.xmi -xsl /home/gael/Logiciels/kde3.5-svn/kdesdk/umbrello.withdocgen/umbrello/docgenerators/xmi2docbook.xsl -out $1.docbook 
+java -cp @PREFIX@/share/java/xalan.jar org.apache.xalan.xslt.Process -xml -in $1.xmi -xsl @PREFIX@/share/umbrello/xmi2docbook.xsl -out $1.docbook 
 
 echo Converting from docbook to XHTML...
-java -cp /usr/share/java/xalan-j2-2.6.0.jar org.apache.xalan.xslt.Process -in $1.docbook -xsl /home/gael/Logiciels/kde3.5-svn/kdesdk/umbrello.withdocgen/umbrello/docgenerators/docbook2xhtml.xsl -out $1.html -html
+java -cp @PREFIX@/share/java/xalan.jar org.apache.xalan.xslt.Process -in $1.docbook -xsl @PREFIX@/share/umbrello/docbook2xhtml.xsl -out $1.html -html
 
 echo done.
