--- unittests/CMakeLists.txt.orig	2017-06-16 22:04:20.184672757 +0200
+++ unittests/CMakeLists.txt	2017-06-16 22:04:58.505673048 +0200
@@ -118,8 +118,8 @@
 ecm_add_executable(testcrashhandler testcrashhandler.cpp)
 target_link_libraries(testcrashhandler ${LIBS})
 
-find_package(LLVM CONFIG)
-find_package(CLANG QUIET)
+# find_package(LLVM CONFIG)
+# find_package(CLANG QUIET)
 if(LLVM_FOUND AND CLANG_FOUND)
     message(STATUS "Found LLVM ${LLVM_PACKAGE_VERSION}")
     message(STATUS "Using LLVMConfig.cmake in: ${LLVM_DIR}")
