diff --git b/libpkgconf/orig.stdinc.h a/libpkgconf/stdinc.h
index b38870b..764fb31 100644
--- b/libpkgconf/orig.stdinc.h
+++ a/libpkgconf/stdinc.h
@@ -26,6 +26,9 @@
 #include <sys/types.h>
 #include <stdint.h>
 
+#ifdef __APPLE__
+# include <errno.h>
+#endif
 #ifdef _WIN32
 # define WIN32_LEAN_AND_MEAN
 # include <windows.h>
diff --git b/libpkgconf/orig.bsdstubs.h a/libpkgconf/bsdstubs.h
index 4e52d41..2db4bed 100644
--- b/libpkgconf/orig.bsdstubs.h
+++ a/libpkgconf/bsdstubs.h
@@ -16,6 +16,9 @@
 #ifndef LIBPKGCONF_BSDSTUBS_H
 #define LIBPKGCONF_BSDSTUBS_H
 
+#ifdef __APPLE__
+# include <errno.h>
+#endif
 #include <libpkgconf/libpkgconf-api.h>
 
 #ifdef __cplusplus
