diff --git a/libktcore/plugin/pluginmanager.cpp b/libktcore/plugin/pluginmanager.cpp
index 1294dc9001120377ee4ce8daff59ae795d61be0d..72eaa94bbffa700569775623cd093c868d5dd7bd 100644
--- a/libktcore/plugin/pluginmanager.cpp
+++ b/libktcore/plugin/pluginmanager.cpp
@@ -59,7 +59,7 @@ namespace kt
             //simple workaround for the situation i have in debian --Nick
             QStringList paths = QCoreApplication::libraryPaths();
             if (paths.isEmpty())
-                paths << QLatin1String("/usr/lib/x86_64-linux-gnu/plugins");
+                paths << QLatin1String("@QTPLUGINS@");
 
             QCoreApplication::addLibraryPath(paths.first().remove(QLatin1String("qt5/")));
             pluginsMetaData = KPluginLoader::findPlugins(QStringLiteral("ktorrent"));
