diff --git a/ktorrent/main.cpp b/ktorrent/main.cpp
index d296bb3d..4c91a7e3 100644
--- a/ktorrent/main.cpp
+++ b/ktorrent/main.cpp
@@ -111,7 +111,7 @@ int main(int argc, char** argv)
     KLocalizedString::setApplicationDomain("ktorrent");
 
     QApplication app(argc, argv);
-    app.setWindowIcon(QIcon::fromTheme(QStringLiteral("ktorrent")));
+    app.setWindowIcon(QIcon::fromTheme(QStringLiteral("ktorrent"), app.windowIcon()));
     KCrash::initialize();
 
     QCommandLineParser parser;
