diff --git a/ktupnptest/CMakeLists.txt b/ktupnptest/CMakeLists.txt
index 8d0f59a6d90c4c7d1f8c52a0badb8bf36baf18d9..08a7bd9a15d7922a7d4d2837e59840305000d439 100644
--- a/ktupnptest/CMakeLists.txt
+++ b/ktupnptest/CMakeLists.txt
@@ -2,6 +2,7 @@ set(ktupnptest_SRC main.cpp upnptestwidget.cpp)
 
 ki18n_wrap_ui(ktupnptest_SRC upnptestwidget.ui)
 add_executable(ktupnptest ${ktupnptest_SRC})
+ecm_mark_nongui_executable(ktupnptest)
 
 target_link_libraries(
     ktupnptest
