diff --git a/ktorrent/CMakeLists.txt b/ktorrent/CMakeLists.txt
index e8c29af3fc9cd347bc4e25ca443b96dad408348c..d07574a98f5ab7ec4c7b684c1fdc98ce2f6e825e 100644
--- a/ktorrent/CMakeLists.txt
+++ b/ktorrent/CMakeLists.txt
@@ -112,6 +112,24 @@ target_link_libraries(ktorrent_app
     KF5::XmlGui
 )
 
+# we provide our own Info.plist containing the information on what documents we can open.
+if(APPLE)
+    # own plist template
+    set_target_properties (ktorrent_app PROPERTIES MACOSX_BUNDLE_INFO_PLIST ${CMAKE_CURRENT_SOURCE_DIR}/MacOSXBundleInfo.plist.in)
+
+    # the MacOSX bundle display name property (CFBundleDisplayName) is not currently supported by cmake,
+    # so has to be set for all targets in this cmake file
+    set(MACOSX_BUNDLE_DISPLAY_NAME KTorrent)
+    set_target_properties(ktorrent_app PROPERTIES MACOSX_BUNDLE_GUI_IDENTIFIER "org.kde.ktorrent")
+    set_target_properties(ktorrent_app PROPERTIES MACOSX_BUNDLE_BUNDLE_NAME "${MACOSX_BUNDLE_DISPLAY_NAME}")
+    set_target_properties(ktorrent_app PROPERTIES MACOSX_BUNDLE_DISPLAY_NAME "${MACOSX_BUNDLE_DISPLAY_NAME}")
+    set_target_properties(ktorrent_app PROPERTIES MACOSX_BUNDLE_INFO_STRING "${MACOSX_BUNDLE_DISPLAY_NAME} - A BitTorrent program by KDE")
+    set_target_properties(ktorrent_app PROPERTIES MACOSX_BUNDLE_LONG_VERSION_STRING "${MACOSX_BUNDLE_DISPLAY_NAME} ${KDE_APPLICATIONS_VERSION}")
+    set_target_properties(ktorrent_app PROPERTIES MACOSX_BUNDLE_SHORT_VERSION_STRING "${KDE_APPLICATIONS_VERSION}")
+    set_target_properties(ktorrent_app PROPERTIES MACOSX_BUNDLE_BUNDLE_VERSION "${KDE_APPLICATIONS_VERSION}")
+    set_target_properties(ktorrent_app PROPERTIES MACOSX_BUNDLE_COPYRIGHT "(C) 2005 - 2011 Joris Guisson and Ivan Vasic")
+endif()
+
 install(TARGETS ktorrent_app ${INSTALL_TARGETS_DEFAULT_ARGS})
 install(PROGRAMS org.kde.ktorrent.desktop  DESTINATION  ${XDG_APPS_INSTALL_DIR} )
 install(FILES ktorrentui.rc DESTINATION ${KXMLGUI_INSTALL_DIR}/ktorrent )
diff --git a/ktorrent/MacOSXBundleInfo.plist.in b/ktorrent/MacOSXBundleInfo.plist.in
new file mode 100644
index 0000000000000000000000000000000000000000..76ac3abcf81f6544ea11a84ca8dd61f9d2de3067
--- /dev/null
+++ b/ktorrent/MacOSXBundleInfo.plist.in
@@ -0,0 +1,100 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple Computer//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+        <key>NSPrincipalClass</key>
+        <string>NSApplication</string>
+        <key>NSHighResolutionCapable</key>
+        <string>True</string>
+        <key>CFBundleDevelopmentRegion</key>
+        <string>English</string>
+        <key>CFBundleExecutable</key>
+        <string>${MACOSX_BUNDLE_EXECUTABLE_NAME}</string>
+        <key>CFBundleGetInfoString</key>
+        <string>${MACOSX_BUNDLE_INFO_STRING}</string>
+        <key>CFBundleIconFile</key>
+        <string>${MACOSX_BUNDLE_ICON_FILE}</string>
+        <key>CFBundleIdentifier</key>
+        <string>${MACOSX_BUNDLE_GUI_IDENTIFIER}</string>
+        <key>CFBundleInfoDictionaryVersion</key>
+        <string>6.0</string>
+        <key>CFBundleLongVersionString</key>
+        <string>${MACOSX_BUNDLE_LONG_VERSION_STRING}</string>
+        <key>CFBundleName</key>
+        <string>${MACOSX_BUNDLE_BUNDLE_NAME}</string>
+        <key>CFBundlePackageType</key>
+        <string>APPL</string>
+        <key>CFBundleShortVersionString</key>
+        <string>${MACOSX_BUNDLE_SHORT_VERSION_STRING}</string>
+        <key>CFBundleSignature</key>
+        <string>KTOR</string>
+        <key>CFBundleVersion</key>
+        <string>${MACOSX_BUNDLE_BUNDLE_VERSION}</string>
+        <key>CSResourcesFileMapped</key>
+        <true/>
+        <key>LSRequiresCarbon</key>
+        <true/>
+        <key>NSHumanReadableCopyright</key>
+        <string>${MACOSX_BUNDLE_COPYRIGHT}</string>
+        <key>LSMultipleInstancesProhibited</key>
+        <true/>
+        <key>CFBundleDocumentTypes</key>
+        <array>
+            <dict>
+                <key>CFBundleTypeExtensions</key>
+                <array>
+                        <string>torrent</string>
+                </array>
+                <key>CFBundleTypeName</key>
+                <string>BitTorrent Document</string>
+                <key>CFBundleTypeRole</key>
+                <string>Viewer</string>
+                <key>LSHandlerRank</key>
+                <string>Owner</string>
+                <key>LSItemContentTypes</key>
+                <array>
+                    <string>org.bittorrent.torrent</string>
+                </array>
+            </dict>
+        </array>
+        <key>CFBundleURLTypes</key>
+        <array>
+            <dict>
+                <key>CFBundleURLName</key>
+                <string>BitTorrent Magnet URL</string>
+                <key>CFBundleURLSchemes</key>
+                <array>
+                    <string>magnet</string>
+                </array>
+            </dict>
+        </array>
+        <key>UTExportedTypeDeclarations</key>
+        <array>
+            <dict>
+                <key>UTTypeConformsTo</key>
+                <array>
+                    <string>public.data</string>
+                    <string>public.item</string>
+                    <string>com.bittorrent.torrent</string>
+                </array>
+                <key>UTTypeDescription</key>
+                <string>BitTorrent Document</string>
+                <key>UTTypeIdentifier</key>
+                <string>org.bittorrent.torrent</string>
+                <key>UTTypeReferenceURL</key>
+                <string>http://www.bittorrent.org/beps/bep_0000.html</string>
+                <key>UTTypeTagSpecification</key>
+                <dict>
+                    <key>com.apple.ostype</key>
+                    <string>TORR</string>
+                    <key>public.filename-extension</key>
+                    <array>
+                        <string>torrent</string>
+                    </array>
+                    <key>public.mime-type</key>
+                    <string>application/x-bittorrent</string>
+                </dict>
+            </dict>
+        </array>
+</dict>
+</plist>
diff --git a/ktorrent/main.cpp b/ktorrent/main.cpp
index d296bb3d1bd03f7e778dc11c52dba91a82639716..35006892460f00555b01b526193770e46fa8fff8 100644
--- a/ktorrent/main.cpp
+++ b/ktorrent/main.cpp
@@ -35,6 +35,7 @@
 #include <QCommandLineParser>
 #include <QDir>
 #include <QFile>
+#include <QFileOpenEvent>
 
 #include <KAboutData>
 #include <KCrash>
@@ -90,6 +91,32 @@ bool GrabPIDLock()
 }
 #endif
 
+class OpenFileEventHandler : public QObject
+{
+    Q_OBJECT
+public:
+    OpenFileEventHandler(QApplication *parent, kt::GUI *w)
+        : QObject(parent)
+        , m_widget(w)
+    {
+        parent->installEventFilter(this);
+    }
+
+    bool eventFilter(QObject *obj, QEvent *event) override
+    {
+        if (event->type() == QEvent::FileOpen) {
+            QFileOpenEvent *openEvent = static_cast<QFileOpenEvent*>(event);
+            qDebug() << "File open event:" << openEvent->url() << "for widget" << m_widget;
+            m_widget->loadSilently(openEvent->url());
+            return true;
+        }
+        return QObject::eventFilter(obj, event);
+    }
+private:
+    kt::GUI *m_widget;
+};
+
+
 
 int main(int argc, char** argv)
 {
@@ -236,6 +263,8 @@ int main(int argc, char** argv)
         QObject::connect(&dbusService, &KDBusService::activateRequested, handleCmdLine);
         handleCmdLine(QStringList(), QString());
 
+        const OpenFileEventHandler fileOpenHandler(&app, &widget);
+
         app.setQuitOnLastWindowClosed(false);
         app.exec();
     }
@@ -257,3 +286,4 @@ int main(int argc, char** argv)
 
 
 
+#include "main.moc"
