diff --git include/QtCore/../../src/corelib/global/orig.qendian.h include/QtCore/../../src/corelib/global/qendian.h
index 615f523..1a188b4 100644
--- include/QtCore/../../src/corelib/global/orig.qendian.h
+++ include/QtCore/../../src/corelib/global/qendian.h
@@ -48,6 +48,9 @@
 #include <stdlib.h>
 #include <string.h>
 
+// RJVB
+#include <limits>
+
 #ifdef min // MSVC
 #undef min
 #undef max
diff --git include/QtCore/../../src/corelib/tools/orig.qbytearraymatcher.h include/QtCore/../../src/corelib/tools/qbytearraymatcher.h
index dafaea9..6318b61 100644
--- include/QtCore/../../src/corelib/tools/orig.qbytearraymatcher.h
+++ include/QtCore/../../src/corelib/tools/qbytearraymatcher.h
@@ -42,6 +42,9 @@
 
 #include <QtCore/qbytearray.h>
 
+//RJVB
+#include <limits>
+
 QT_BEGIN_NAMESPACE
 
 
