diff --git examples/vulkan/hellovulkancubes/renderer.cpp examples/vulkan/hellovulkancubes/renderer.cpp
index 597c6fc4..5ada79ce 100644
--- examples/vulkan/hellovulkancubes/renderer.cpp
+++ examples/vulkan/hellovulkancubes/renderer.cpp
@@ -99,10 +99,10 @@ void Renderer::preInitResources()
 {
     const QVector<int> sampleCounts = m_window->supportedSampleCounts();
     if (DBG)
-        qWarning() << "Supported sample counts:" << sampleCounts;
+        qDebug() << "Supported sample counts:" << sampleCounts;
     if (sampleCounts.contains(4)) {
         if (DBG)
-            qWarning("Requesting 4x MSAA");
+            qDebug("Requesting 4x MSAA");
         m_window->setSampleCount(4);
     }
 }
@@ -110,7 +110,7 @@ void Renderer::preInitResources()
 void Renderer::initResources()
 {
     if (DBG)
-        qWarning("Renderer init");
+        qDebug("Renderer init");
 
     m_animating = true;
     m_framePending = false;
@@ -533,7 +533,7 @@ void Renderer::releaseSwapChainResources()
 void Renderer::releaseResources()
 {
     if (DBG)
-        qWarning("Renderer release");
+        qDebug("Renderer release");
 
     m_pipelinesFuture.waitForFinished();
 
@@ -771,7 +771,7 @@ void Renderer::ensureInstanceBuffer()
         VkMemoryRequirements memReq;
         m_devFuncs->vkGetBufferMemoryRequirements(dev, m_instBuf, &memReq);
         if (DBG)
-            qWarning("Allocating %u bytes for instance data", uint32_t(memReq.size));
+            qDebug("Allocating %u bytes for instance data", uint32_t(memReq.size));
 
         VkMemoryAllocateInfo memAllocInfo = {
             VK_STRUCTURE_TYPE_MEMORY_ALLOCATE_INFO,
@@ -790,7 +790,7 @@ void Renderer::ensureInstanceBuffer()
 
     if (m_instCount != m_preparedInstCount) {
         if (DBG)
-            qWarning("Preparing instances %d..%d", m_preparedInstCount, m_instCount - 1);
+            qDebug("Preparing instances %d..%d", m_preparedInstCount, m_instCount - 1);
         char *p = m_instData.data();
         p += m_preparedInstCount * PER_INSTANCE_DATA_SIZE;
         auto gen = [](float a, float b) { return float((qrand() % int(b - a + 1)) + a); };
@@ -807,13 +807,10 @@ void Renderer::ensureInstanceBuffer()
     }
 
     quint8 *p;
-    VkResult err = m_devFuncs->vkMapMemory(dev, m_instBufMem, 0, m_instData.size(), 0,
+    VkResult err = m_devFuncs->vkMapMemory(dev, m_instBufMem, 0, m_instCount * PER_INSTANCE_DATA_SIZE, 0,
                                            reinterpret_cast<void **>(&p));
     if (err != VK_SUCCESS)
         qFatal("Failed to map memory: %d", err);
-//     if (DBG) {
-//         qWarning("Copying m_instData[%d] into buffer size %lu", m_instData.size(), m_instCount * PER_INSTANCE_DATA_SIZE);
-//     }
     memcpy(p, m_instData.constData(), m_instData.size());
     m_devFuncs->vkUnmapMemory(dev, m_instBufMem);
 }
