diff --git a/src/core/config/common.pri b/src/core/config/common.pri
index c8c3d55f..ddb4ca4b 100644
--- a/src/core/config/common.pri
+++ b/src/core/config/common.pri
@@ -60,13 +60,6 @@ precompile_header {
     gn_args += enable_precompiled_headers=false
 }
 
-CONFIG(release, debug|release):!isDeveloperBuild() {
-    gn_args += is_official_build=true
-} else {
-    gn_args += is_official_build=false
-    !isDeveloperBuild(): gn_args += is_unsafe_developer_build=false
-}
-
 CONFIG(release, debug|release) {
     gn_args += is_debug=false
     force_debug_info {
