diff --git src/corelib/global/qendian.h src/corelib/global/qendian.h
index 615f5238..6067836d 100644
--- src/corelib/global/qendian.h
+++ src/corelib/global/qendian.h
@@ -47,6 +47,8 @@
 // include stdlib.h and hope that it defines __GLIBC__ for glibc-based systems
 #include <stdlib.h>
 #include <string.h>
+// for newer G++
+#include <limits>
 
 #ifdef min // MSVC
 #undef min
diff --git src/corelib/tools/qbytearraymatcher.h src/corelib/tools/qbytearraymatcher.h
index dafaea9c..b1793525 100644
--- src/corelib/tools/qbytearraymatcher.h
+++ src/corelib/tools/qbytearraymatcher.h
@@ -40,6 +40,8 @@
 #ifndef QBYTEARRAYMATCHER_H
 #define QBYTEARRAYMATCHER_H
 
+#include <limits>
+
 #include <QtCore/qbytearray.h>
 
 QT_BEGIN_NAMESPACE
