--- src/konfigurator/konfigurator.cpp.orig	2016-12-06 21:00:21.000000000 +0100
+++ src/konfigurator/konfigurator.cpp	2016-12-23 22:00:51.000000000 +0100
@@ -60,7 +60,7 @@
                      QStringLiteral(),
                      QStringLiteral("staikos@kde.org"));
     setAboutData(about);
-    setNeedsAuthorization(true);
+    setNeedsAuthorization(false);
 
     QVBoxLayout *vbox = new QVBoxLayout(this);
     //PORT QT5 vbox->setSpacing(KDialog::spacingHint());
@@ -267,21 +267,22 @@
     QVariantMap args;
     KAuth::Action action = authAction();
     if (!action.isValid()) {
-        qDebug() << "There's no authAction, not saving settings";
-        return;
-    }
-    action.setArguments(args);
+//         qDebug() << "There's no authAction, not saving settings";
+//         return;
+    } else {
+        action.setArguments(args);
 
-    KAuth::ExecuteJob *j = action.execute();
+        KAuth::ExecuteJob *j = action.execute();
 
-    if (!j->exec()) {
-        if (j->error() == KAuth::ActionReply::AuthorizationDeniedError) {
-            KMessageBox::error(this, i18n("Permission denied."), i18n("KDE Wallet Control Module"));
-        } else {
-            KMessageBox::error(this, i18n("Error while authenticating action:\n%1", j->errorString()), i18n("KDE Wallet Control Module"));
+        if (!j->exec()) {
+            if (j->error() == KAuth::ActionReply::AuthorizationDeniedError) {
+                KMessageBox::error(this, i18n("Permission denied."), i18n("KDE Wallet Control Module"));
+            } else {
+                KMessageBox::error(this, i18n("Error while authenticating action:\n%1", j->errorString()), i18n("KDE Wallet Control Module"));
+            }
+            load();
+            return;
         }
-        load();
-        return;
     }
 
     KConfigGroup config(_cfg, "Wallet");
