--- src/manager/orig.main.cpp	2015-12-09 13:02:04.000000000 +0100
+++ src/manager/main.cpp	2015-12-19 22:40:01.000000000 +0100
@@ -40,7 +40,7 @@
      */
     a.setAttribute(Qt::AA_UseHighDpiPixmaps, true);
 
-    a.setWindowIcon(QIcon::fromTheme(QStringLiteral("kwalletmanager")));
+    a.setWindowIcon(QIcon::fromTheme(QStringLiteral("kwalletmanager"), a.windowIcon()));
 
     KAboutData aboutData(QStringLiteral("kwalletmanager5"),
                  i18n("Wallet Manager"),
