diff --git src/manager/kwalletmanager.cpp src/manager/kwalletmanager.cpp
index 64a90dd..0214bfe 100644
--- src/manager/kwalletmanager.cpp
+++ src/manager/kwalletmanager.cpp
@@ -165,6 +165,8 @@ void KWalletManager::configUI() {
     connect(act, &QAction::triggered, this, &KWalletManager::setupWallet);
     if (_tray) {
         _tray->contextMenu()->addAction(act);
+        _tray->setStandardActionsEnabled(true);
+        connect(_tray->action(QLatin1String("minimizeRestore")), SIGNAL(triggered()), SLOT(show()));
     }
     act = actionCollection()->addAction(QLatin1String("close_all_wallets"));
     act->setText(i18n("Close &All Wallets"));
