diff --git src/konfigurator/CMakeLists.txt src/konfigurator/CMakeLists.txt
index 87030fe..83eb53b 100644
--- src/konfigurator/CMakeLists.txt
+++ src/konfigurator/CMakeLists.txt
@@ -1,4 +1,6 @@
 ########### next target ###############
+include(ECMMarkNonGuiExecutable)
+
 add_definitions(-DTRANSLATION_DOMAIN=\"kcmkwallet\")
 
 set(kcm_kwallet_PART_SRCS konfigurator.cpp )
@@ -24,6 +26,7 @@ install(TARGETS kcm_kwallet5  DESTINATION ${PLUGIN_INSTALL_DIR})
 set(kcm_kwallet_helper_SRCS savehelper.cpp)
 
 add_executable(kcm_kwallet_helper5 ${kcm_kwallet_helper_SRCS})
+ecm_mark_nongui_executable(kcm_kwallet_helper5)
 target_link_libraries(kcm_kwallet_helper5
     Qt5::Core
     KF5::Auth
