--- orig.configure	2014-12-09 00:30:18.000000000 +0100
+++ configure	2014-12-11 21:38:33.000000000 +0100
@@ -7881,15 +7881,15 @@
     QT_CONFIG="$QT_CONFIG concurrent"
 fi
 
-if [ "$CFG_XMLPATTERNS" = "yes" -a "$CFG_EXCEPTIONS" = "no" ]; then
-    echo "QtXmlPatterns was requested, but it can't be built due to exceptions being disabled."
-    exit 1
-fi
-if [ "$CFG_XMLPATTERNS" = "auto" -a "$CFG_EXCEPTIONS" != "no" ]; then
+#if [ "$CFG_XMLPATTERNS" = "yes" -a "$CFG_EXCEPTIONS" = "no" ]; then
+#    echo "QtXmlPatterns was requested, but it can't be built due to exceptions being disabled."
+#    exit 1
+#fi
+#if [ "$CFG_XMLPATTERNS" = "auto" -a "$CFG_EXCEPTIONS" != "no" ]; then
     CFG_XMLPATTERNS="$canBuildQtXmlPatterns"
-elif [ "$CFG_EXCEPTIONS" = "no" ]; then
-    CFG_XMLPATTERNS="no"
-fi
+#elif [ "$CFG_EXCEPTIONS" = "no" ]; then
+#    CFG_XMLPATTERNS="no"
+#fi
 if [ "$CFG_XMLPATTERNS" = "yes" ]; then
     QT_CONFIG="$QT_CONFIG xmlpatterns"
 else
--- src/xmlpatterns/utils/orig.qpatternistlocale_p.h	2014-04-10 20:37:12.000000000 +0200
+++ src/xmlpatterns/utils/qpatternistlocale_p.h	2014-12-11 23:58:30.000000000 +0100
@@ -274,7 +274,8 @@
 #endif
 
 #ifdef QT_NO_EXCEPTIONS
-#error "Patternist uses exceptions and cannot be built without."
+// #error "Patternist uses exceptions and cannot be built without."
+#warning "Patternist uses exceptions and will ignore the fact QtCore is built without"
 #endif
 
 QT_END_NAMESPACE
--- src/xmlpatterns/orig.xmlpatterns.pro	2014-12-16 13:58:27.000000000 +0100
+++ src/xmlpatterns/xmlpatterns.pro	2014-12-16 13:59:46.000000000 +0100
@@ -2,8 +2,13 @@
 QPRO_PWD = $$PWD
 QT = core \
     network
+DEFINES -= QT_NO_EXCEPTIONS
 DEFINES += QT_BUILD_XMLPATTERNS_LIB \
-    QT_NO_USING_NAMESPACE
+    QT_NO_USING_NAMESPACE QT_EXCEPTIONS
+QMAKE_CXXFLAGS_EXCEPTIONS_OFF -= -fno-exceptions
+QMAKE_CFLAGS -= -fno-exceptions
+QMAKE_CXXFLAGS -= -fno-exceptions
+QMAKE_LFLAGS -= -fno-exceptions
 win32-msvc*|win32-icc:QMAKE_LFLAGS += /BASE:0x61000000
 unix|win32-g++*:QMAKE_PKGCONFIG_REQUIRES = QtCore \
     QtNetwork
--- src/corelib/orig.corelib.pro	2014-12-16 13:46:18.000000000 +0100
+++ src/corelib/corelib.pro	2014-12-16 13:47:05.000000000 +0100
@@ -1,7 +1,12 @@
 TARGET	   = QtCore
 QPRO_PWD   = $$PWD
 QT         =
-DEFINES   += QT_BUILD_CORE_LIB QT_NO_USING_NAMESPACE
+DEFINES   -= QT_NO_EXCEPTIONS
+DEFINES   += QT_BUILD_CORE_LIB QT_NO_USING_NAMESPACE QT_EXCEPTIONS
+QMAKE_CXXFLAGS_EXCEPTIONS_OFF -= -fno-exceptions
+QMAKE_CFLAGS -= -fno-exceptions
+QMAKE_CXXFLAGS -= -fno-exceptions
+QMAKE_LFLAGS -= -fno-exceptions
 win32-msvc*|win32-icc:QMAKE_LFLAGS += /BASE:0x67000000
 irix-cc*:QMAKE_CXXFLAGS += -no_prelink -ptused
 
