--- src/xmlpatterns/orig.xmlpatterns.pro	2014-12-16 13:58:27.000000000 +0100
+++ src/xmlpatterns/xmlpatterns.pro	2014-12-16 13:59:46.000000000 +0100
@@ -2,8 +2,13 @@
 QPRO_PWD = $$PWD
 QT = core \
     network
+DEFINES -= QT_NO_EXCEPTIONS
 DEFINES += QT_BUILD_XMLPATTERNS_LIB \
-    QT_NO_USING_NAMESPACE
+    QT_NO_USING_NAMESPACE QT_EXCEPTIONS
+QMAKE_CXXFLAGS_EXCEPTIONS_OFF -= -fno-exceptions
+QMAKE_CFLAGS -= -fno-exceptions
+QMAKE_CXXFLAGS -= -fno-exceptions
+QMAKE_LFLAGS -= -fno-exceptions
 win32-msvc*|win32-icc:QMAKE_LFLAGS += /BASE:0x61000000
 unix|win32-g++*:QMAKE_PKGCONFIG_REQUIRES = QtCore \
     QtNetwork
