--- src/gui/kernel/orig.qwidget_mac.mm	2014-10-03 15:06:54.000000000 +0200
+++ src/gui/kernel/qwidget_mac.mm	2014-10-03 15:08:07.000000000 +0200
@@ -4619,7 +4619,7 @@
 
         setWSGeometry(false, oldRect);
 
-        if (isResize && QApplicationPrivate::graphicsSystem())
+        if (isResize && q->parentWidget() && QApplicationPrivate::graphicsSystem())
             invalidateBuffer_resizeHelper(oldp, olds);
     }
 
