--- src/pim/agent/agent.cpp
+++ src/pim/agent/agent.cpp
@@ -82,6 +82,7 @@
     changeRecorder()->itemFetchScope().setAncestorRetrieval(Akonadi::ItemFetchScope::Parent);
     changeRecorder()->itemFetchScope().setFetchRemoteIdentification(false);
     changeRecorder()->itemFetchScope().setFetchModificationTime(false);
+    changeRecorder()->itemFetchScope().fetchFullPayload(true);
     changeRecorder()->setChangeRecordingEnabled(false);
 
     new BalooIndexerAdaptor(this);

