diff --git modules/services_discovery/bonjour.m modules/services_discovery/bonjour.m
index f9dd45c..a2f4e69 100644
--- modules/services_discovery/bonjour.m
+++ modules/services_discovery/bonjour.m
@@ -82,7 +82,7 @@ NSString *const VLCBonjourProtocolServiceName = @"VLCBonjourProtocolServiceName"
 
 struct services_discovery_sys_t
 {
-    CFTypeRef _Nullable discoveryController;
+    CFTypeRef discoveryController;
 };
 
 @implementation VLCNetServiceDiscoveryController
