diff --git modules/keystore/keychain.m modules/keystore/keychain.m
index 9b9be73..53bee80 100644
--- modules/keystore/keychain.m
+++ modules/keystore/keychain.m
@@ -446,7 +446,7 @@ static unsigned int Remove(vlc_keystore *p_keystore,
         matchCount = matches.count;
 
         for (NSUInteger x = 0; x < matchCount; x++) {
-            status = SecKeychainItemDelete((__bridge SecKeychainItemRef _Nonnull)(matches[x]));
+            status = SecKeychainItemDelete((__bridge SecKeychainItemRef)(matches[x]));
             if (status != noErr) {
                 msg_Err(p_keystore, "Deletion error %i (%s)", status , [ErrorForStatus(status) UTF8String]);
                 failed = YES;

diff --git modules/gui/macosx/Windows.m modules/gui/macosx/Windows.m
index 3303f16..2154188 100644
--- modules/gui/macosx/Windows.m
+++ modules/gui/macosx/Windows.m
@@ -118,7 +118,7 @@
     NSInvocation *invoc = [[NSInvocation alloc] init];
     [invoc setSelector:@selector(orderOut:)];
     [invoc setTarget: self];
-    [invoc setArgument:(__bridge void * __nonnull)sender atIndex:2];
+    [invoc setArgument:(__bridge void *)sender atIndex:2];
     [self orderOut: sender animate: animate callback: invoc];
 }
 
