diff --git modules/gui/macosx/InputManager.m modules/gui/macosx/InputManager.m
index de4252f..04c51c7 100644
--- modules/gui/macosx/InputManager.m
+++ modules/gui/macosx/InputManager.m
@@ -180,7 +180,9 @@ static int InputEvent(vlc_object_t *p_this, const char *psz_var,
 
     var_DelCallback(pl_Get(getIntf()), "input-current", InputThreadChanged, (__bridge void *)self);
 
+#if  !__has_feature(objc_arc)
     dispatch_release(informInputChangedQueue);
+#endif
 }
 
 - (void)inputThreadChanged
