diff --git modules/services_discovery/bonjour.m modules/services_discovery/bonjour.m
index f9dd45c..a2f4e69 100644
--- modules/services_discovery/bonjour.m
+++ modules/services_discovery/bonjour.m
@@ -82,7 +82,7 @@ NSString *const VLCBonjourProtocolServiceName = @"VLCBonjourProtocolServiceName"
 
 struct services_discovery_sys_t
 {
-    CFTypeRef _Nullable discoveryController;
+    CFTypeRef discoveryController;
 };
 
 @implementation VLCNetServiceDiscoveryController
diff --git modules/gui/macosx/misc.m modules/gui/macosx/misc.m
index e272ec7..f432f74 100644
--- modules/gui/macosx/misc.m
+++ modules/gui/macosx/misc.m
@@ -142,17 +142,17 @@ static NSMapTable *VLCAdditions_userInfo = NULL;
 
 - (void)dealloc
 {
-    NSMapRemove(VLCAdditions_userInfo, (__bridge const void * __nullable)(self));
+    NSMapRemove(VLCAdditions_userInfo, (__bridge const void *)(self));
 }
 
 - (void)setUserInfo: (void *)userInfo
 {
-    NSMapInsert(VLCAdditions_userInfo, (__bridge const void * __nullable)(self), (void*)userInfo);
+    NSMapInsert(VLCAdditions_userInfo, (__bridge const void *)(self), (void*)userInfo);
 }
 
 - (void *)userInfo
 {
-    return NSMapGet(VLCAdditions_userInfo, (__bridge const void * __nullable)(self));
+    return NSMapGet(VLCAdditions_userInfo, (__bridge const void *)(self));
 }
 @end
 
diff --git modules/gui/macosx/VLCPlaylist.m modules/gui/macosx/VLCPlaylist.m
index e34f80f..85b88d7 100644
--- modules/gui/macosx/VLCPlaylist.m
+++ modules/gui/macosx/VLCPlaylist.m
@@ -170,7 +170,7 @@
     [self initStrings];
 }
 
-- (void)setOutlineView:(VLCPlaylistView * __nullable)outlineView
+- (void)setOutlineView:(VLCPlaylistView *)outlineView
 {
     _outlineView = outlineView;
     [_outlineView setDelegate:self];
@@ -191,7 +191,7 @@
     [self reloadStyles];
 }
 
-- (void)setPlaylistHeaderView:(NSTableHeaderView * __nullable)playlistHeaderView
+- (void)setPlaylistHeaderView:(NSTableHeaderView *)playlistHeaderView
 {
     VLCMainMenu *mainMenu = [[VLCMain sharedInstance] mainMenu];
     _playlistHeaderView = playlistHeaderView;
diff --git modules/gui/macosx/SPMediaKeyTap.m modules/gui/macosx/SPMediaKeyTap.m
index d05275b..c63f4be 100644
--- modules/gui/macosx/SPMediaKeyTap.m
+++ modules/gui/macosx/SPMediaKeyTap.m
@@ -81,7 +81,7 @@ static CGEventRef tapEventCallback(CGEventTapProxy proxy, CGEventType type, CGEv
                                   kCGEventTapOptionDefault,
                                   CGEventMaskBit(NX_SYSDEFINED),
                                   tapEventCallback,
-                                  (__bridge void * __nullable)(self));
+                                  (__bridge void *)(self));
     assert(_eventPort != NULL);
 
     _eventPortSource = CFMachPortCreateRunLoopSource(kCFAllocatorSystemDefault, _eventPort, 0);
