diff --git Makefile.am Makefile.am
index 8cfed50..52632f7 100644
--- Makefile.am
+++ b/Makefile.am
@@ -133,7 +133,6 @@ EXTRA_DIST += \
 	extras/package/macosx/README.MacOSX.rtf \
 	extras/package/macosx/Resources/dsa_pub.pem \
 	extras/package/macosx/Resources/English.lproj/About.xib \
-	extras/package/macosx/Resources/English.lproj/AddonManager.xib \
 	extras/package/macosx/Resources/English.lproj/AudioEffects.xib \
 	extras/package/macosx/Resources/English.lproj/Bookmarks.xib \
 	extras/package/macosx/Resources/English.lproj/BWQuincyMain.xib \
