diff --git modules/gui/macosx/coredialogs.m modules/gui/macosx/coredialogs.m
index a782088..24f18cf 100644
--- modules/gui/macosx/coredialogs.m
+++ modules/gui/macosx/coredialogs.m
@@ -304,7 +304,7 @@ static void updateProgressCallback(vlc_dialog_id *p_id,
 - (void)updateDisplayedProgressDialog:(NSArray *)dialogData
 
 {
-    if (!progressIndicator.indeterminate) {
+    if (!progressIndicator.isIndeterminate) {
         progressIndicator.doubleValue = [dialogData[1] doubleValue];
         progressDescriptionLabel.stringValue = dialogData[2];
     }
