diff --git modules/video_filter/Makefile.am modules/video_filter/Makefile.am
index b2ff9cd..9bd6bf6 100644
--- modules/video_filter/Makefile.am
+++ modules/video_filter/Makefile.am
@@ -183,9 +183,9 @@ endif
 libblend_plugin_la_SOURCES = video_filter/blend.cpp
 video_filter_LTLIBRARIES += libblend_plugin.la
 
-libopencv_example_plugin_la_SOURCES = video_filter/opencv_example.cpp video_filter/filter_event_info.h
-libopencv_example_plugin_la_CPPFLAGS = $(AM_CPPFLAGS) $(OPENCV_CFLAGS)
-libopencv_example_plugin_la_LIBADD = $(OPENCV_LIBS)
-libopencv_example_plugin_la_LDFLAGS = $(AM_LDFLAGS) -rpath '$(video_filterdir)'
-video_filter_LTLIBRARIES += $(LTLIBopencv_example)
-EXTRA_LTLIBRARIES += libopencv_example_plugin.la
+# libopencv_example_plugin_la_SOURCES = video_filter/opencv_example.cpp video_filter/filter_event_info.h
+# libopencv_example_plugin_la_CPPFLAGS = $(AM_CPPFLAGS) $(OPENCV_CFLAGS)
+# libopencv_example_plugin_la_LIBADD = $(OPENCV_LIBS)
+# libopencv_example_plugin_la_LDFLAGS = $(AM_LDFLAGS) -rpath '$(video_filterdir)'
+# video_filter_LTLIBRARIES += $(LTLIBopencv_example)
+# EXTRA_LTLIBRARIES += libopencv_example_plugin.la
