diff --git configure.ac configure.ac
index a085ac3..f0b2c0a 100644
--- configure.ac
+++ configure.ac
@@ -69,6 +69,7 @@ AM_PROG_CC_C_O
 AC_PROG_CXX
 AC_PROG_CPP
 AC_PROG_OBJC
+AC_PROG_OBJCXX
 _AM_DEPENDENCIES([OBJC])
 AC_PROG_EGREP
 AC_PROG_MAKE_SET
diff --git modules/gui/qt/Makefile.am modules/gui/qt/Makefile.am
index b48e4ee..289fccc 100644
--- modules/gui/qt/Makefile.am
+++ modules/gui/qt/Makefile.am
@@ -12,7 +12,7 @@ guidir = $(pluginsdir)/gui
 # For each ressource (icon, png, ...)
 #   - Add it to DEPS_res
 
-SUFFIXES += .ui .h .hpp .moc.cpp
+SUFFIXES += .ui .h .hpp .moc.cpp .mm
 
 libqt_plugin_la_CXXFLAGS = $(AM_CXXFLAGS) $(QT_CFLAGS) $(CXXFLAGS_qt)
 libqt_plugin_la_LIBADD = $(QT_LIBS) $(LIBS_qt) $(LIBM)
@@ -26,7 +26,9 @@ libqt_plugin_la_LIBADD += $(X_LIBS) $(X_PRE_LIB) -lX11
 endif
 endif
 if HAVE_DARWIN
+libqt_plugin_la_OBJCXXFLAGS = $(AM_CXXFLAGS) $(QT_CFLAGS) $(CXXFLAGS_qt)
 libqt_plugin_la_LDFLAGS += -Wl,-framework,Cocoa
+libqt_plugin_la_LIBTOOLFLAGS = --tag=CXX
 endif
 
 gui_LTLIBRARIES =  $(LTLIBqt)
@@ -118,6 +120,9 @@ libqt_plugin_la_SOURCES = \
 if HAVE_WIN32
 libqt_plugin_la_SOURCES += main_interface_win32.cpp
 endif
+if HAVE_DARWIN
+libqt_plugin_la_SOURCES += util/searchlineedit_mac.mm
+endif
 
 # Meta-object compilation
 
