diff --git a/qtbase/src/network/bearer/qnetworkconfigmanager_p.cpp b/qtbase/src/network/bearer/qnetworkconfigmanager_p.cpp
index a903ecda5fe7c8fa177f4e011efd409eb990d371..e4e5094a4de13f57cdc99c8168debfbf7e5a42e3 100644
--- a/qtbase/src/network/bearer/qnetworkconfigmanager_p.cpp
+++ b/qtbase/src/network/bearer/qnetworkconfigmanager_p.cpp
@@ -60,10 +60,11 @@ QT_BEGIN_NAMESPACE
 QNetworkConfigurationManagerPrivate::QNetworkConfigurationManagerPrivate()
     : QObject(), pollTimer(0), mutex(QMutex::Recursive),
       loader(QBearerEngineFactoryInterface_iid, QLatin1String("/bearer")),
-      forcedPolling(0), firstUpdate(true)
+      forcedPolling(0), firstUpdate(true), quitting(false)
 {
     qRegisterMetaType<QNetworkConfiguration>();
     qRegisterMetaType<QNetworkConfigurationPrivatePointer>();
+    connect(qApp, &QCoreApplication::aboutToQuit, this, &QNetworkConfigurationManagerPrivate::aboutToQuit);
 }
 
 void QNetworkConfigurationManagerPrivate::initialize()
@@ -352,6 +353,10 @@ void QNetworkConfigurationManagerPrivate::configurationChanged(QNetworkConfigura
 
 void QNetworkConfigurationManagerPrivate::updateConfigurations()
 {
+    if (quitting) {
+        return;
+    }
+
     typedef QMultiMap<int, QString> PluginKeyMap;
     typedef PluginKeyMap::const_iterator PluginKeyMapConstIterator;
     QMutexLocker locker(&mutex);
@@ -456,6 +461,10 @@ QList<QBearerEngine *> QNetworkConfigurationManagerPrivate::engines() const
 
 void QNetworkConfigurationManagerPrivate::startPolling()
 {
+    if (quitting) {
+        qWarning() << Q_FUNC_INFO << "ignored because the application is quitting";
+        return;
+    }
     QMutexLocker locker(&mutex);
     if (!pollTimer) {
         pollTimer = new QTimer(this);
@@ -509,6 +518,11 @@ void QNetworkConfigurationManagerPrivate::disablePolling()
     --forcedPolling;
 }
 
+void QNetworkConfigurationManagerPrivate::aboutToQuit()
+{
+    quitting = true;
+}
+
 QT_END_NAMESPACE
 
 #endif // QT_NO_BEARERMANAGEMENT
diff --git a/qtbase/src/network/bearer/qnetworkconfigmanager_p.h b/qtbase/src/network/bearer/qnetworkconfigmanager_p.h
index 380e25c22f7e5f3a138d346badd248d82e529fc2..0a4b4fd9d56d5b9113da78153519682fe75cf374 100644
--- a/qtbase/src/network/bearer/qnetworkconfigmanager_p.h
+++ b/qtbase/src/network/bearer/qnetworkconfigmanager_p.h
@@ -107,6 +107,7 @@ private Q_SLOTS:
     void configurationAdded(QNetworkConfigurationPrivatePointer ptr);
     void configurationRemoved(QNetworkConfigurationPrivatePointer ptr);
     void configurationChanged(QNetworkConfigurationPrivatePointer ptr);
+    void aboutToQuit();
 
     void pollEngines();
 
@@ -130,6 +131,7 @@ private:
     bool updating;
 
     bool firstUpdate;
+    bool quitting;
 };
 
 Q_NETWORK_EXPORT QNetworkConfigurationManagerPrivate *qNetworkConfigurationManagerPrivate();
