diff --git a/src/app/main.cpp b/src/app/main.cpp
index 9ff0993..570c644 100644
--- a/src/app/main.cpp
+++ b/src/app/main.cpp
@@ -33,6 +33,7 @@ struct CommandLineParameters
 {
     bool forceMinimized;
     bool preventActivation;
+    bool noQWESandBoxing;
 
 #ifdef Q_OS_WINDOWS
     bool attachConsole;
@@ -69,6 +70,11 @@ CommandLineParameters parseCommandLine(const QStringList &arguments)
         {QStringLiteral("minimized"), QObject::tr("Start minimized regardless of settings.")}
     });
 
+    parser.addOptions({
+        {QStringLiteral("no-sandbox"), QObject::tr("Do not use sandboxing in the QtWebEngine backend (only).\n"
+            "Can be needed when started via the zealsearch plugin.")}
+    });
+
 #ifdef Q_OS_WINDOWS
     parser.addOptions({
         {QStringLiteral("attach-console"), QObject::tr("Attach console for logging.")},
@@ -82,6 +88,7 @@ CommandLineParameters parseCommandLine(const QStringList &arguments)
 
     CommandLineParameters clParams;
     clParams.forceMinimized = parser.isSet(QStringLiteral("minimized"));
+    clParams.noQWESandBoxing = parser.isSet(QStringLiteral("no-sandbox"));
     clParams.preventActivation = false;
 
 #ifdef Q_OS_WINDOWS
@@ -192,6 +199,11 @@ int main(int argc, char *argv[])
     QScopedPointer<QApplication> qapp(new QApplication(argc, argv));
 
     const CommandLineParameters clParams = parseCommandLine(qapp->arguments());
+#ifndef USE_WEBKIT
+    if (clParams.noQWESandBoxing) {
+        qputenv("QTWEBENGINE_DISABLE_SANDBOX", "1");
+    }
+#endif
 
 #ifdef Q_OS_WINDOWS
     const static QHash<QString, QString> protocols = {
