diff --git a/CMakeLists.txt b/CMakeLists.txt
index 749c306..f6aae98 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -23,10 +23,14 @@ endif()
 set(PROJECT_COMPANY_NAME "Oleg Shparber")
 set(PROJECT_COPYRIGHT "© 2013-2025 Oleg Shparber and other contributors")
 
+include(FeatureSummary)
+
 # Find available major Qt version. It will be stored in QT_VERSION_MAJOR.
 if(NOT ZEAL_USE_QT5)
     find_package(QT NAMES Qt6 COMPONENTS Core)
     set(QT_MINIMUM_VERSION 6.2.0)
+else()
+    option(USE_WEBKIT "Use the faster, leaner, legacy QtWebKit instead of QtWebEngine" OFF)
 endif()
 
 if(NOT QT_FOUND)
@@ -72,3 +76,5 @@ add_custom_target(zeal_version
 
 add_subdirectory(assets)
 add_subdirectory(src)
+
+feature_summary(WHAT ALL   FATAL_ON_MISSING_REQUIRED_PACKAGES)
diff --git a/src/app/CMakeLists.txt b/src/app/CMakeLists.txt
index d2521b0..5f5c0c5 100644
--- a/src/app/CMakeLists.txt
+++ b/src/app/CMakeLists.txt
@@ -1,6 +1,12 @@
-find_package(Qt${QT_VERSION_MAJOR} COMPONENTS Concurrent WebEngineWidgets Widgets REQUIRED)
-if (Qt${QT_VERSION_MAJOR}Widgets_VERSION VERSION_LESS QT_MINIMUM_VERSION)
-    message(FATAL_ERROR "Qt version >= ${QT_MINIMUM_VERSION} is required.")
+if (USE_WEBKIT)
+    add_definitions(-DUSE_WEBKIT)
+    find_package(Qt${QT_VERSION_MAJOR} COMPONENTS Concurrent WebKitWidgets Widgets REQUIRED)
+    message(STATUS "Using QtWebKitWidgets (backported!)")
+else()
+    find_package(Qt${QT_VERSION_MAJOR} COMPONENTS Concurrent WebEngineWidgets Widgets REQUIRED)
+    if (Qt${QT_VERSION_MAJOR}Widgets_VERSION VERSION_LESS QT_MINIMUM_VERSION)
+        message(WARNING "Qt version >= ${QT_MINIMUM_VERSION} might be required.")
+    endif()
 endif()
 
 # Define output binary name.
diff --git a/src/app/resources/browser/assets/css/darkmode.css b/src/app/resources/browser/assets/css/darkmode.css
new file mode 100644
index 0000000..97a64f0
--- /dev/null
+++ b/src/app/resources/browser/assets/css/darkmode.css
@@ -0,0 +1,11 @@
+/* Dark mode (see #466). */
+/* TODO: Check if we need -webkit-filter for old Qt WebKit. */
+html {
+    -webkit-filter: invert() hue-rotate(180deg) contrast(70%);
+    filter: invert() hue-rotate(180deg) contrast(70%);
+}
+
+html img {
+    -webkit-filter: invert() hue-rotate(180deg) contrast(120%);
+    filter: invert() hue-rotate(180deg) contrast(120%);
+}
diff --git a/src/app/resources/browser/assets/css/highlight-webkit.css b/src/app/resources/browser/assets/css/highlight-webkit.css
new file mode 100644
index 0000000..78a7b1c
--- /dev/null
+++ b/src/app/resources/browser/assets/css/highlight-webkit.css
@@ -0,0 +1,10 @@
+/* Highlight on navigation to an anchor. */
+@-webkit-keyframes targetNavigatedAnimation {
+    from { background: #ffffff; }
+    50% { background: #ffff00; }
+    to { background: #ffffff; }
+}
+
+*:target {
+    -webkit-animation: targetNavigatedAnimation .5s linear;
+}
diff --git a/src/app/zeal.qrc b/src/app/zeal.qrc
index a2c60f3..b885c25 100644
--- a/src/app/zeal.qrc
+++ b/src/app/zeal.qrc
@@ -7,6 +7,8 @@
         <file alias="assets/css/fa-solid.min.css">resources/browser/assets/css/fa-solid.min.css</file>
         <file alias="assets/css/fontawesome.min.css">resources/browser/assets/css/fontawesome.min.css</file>
         <file alias="assets/css/highlight.css">resources/browser/assets/css/highlight.css</file>
+        <file alias="assets/css/highlight-webkit.css">resources/browser/assets/css/highlight-webkit.css</file>
+        <file alias="assets/css/darkmode.css">resources/browser/assets/css/darkmode.css</file>
         <file alias="assets/css/welcome.min.css">resources/browser/assets/css/welcome.min.css</file>
         <file alias="assets/webfonts/fa-brands-400.woff2">resources/browser/assets/webfonts/fa-brands-400.woff2</file>
         <file alias="assets/webfonts/fa-solid-900.woff2">resources/browser/assets/webfonts/fa-solid-900.woff2</file>
diff --git a/src/libs/browser/CMakeLists.txt b/src/libs/browser/CMakeLists.txt
index 5f242bc..04796c2 100644
--- a/src/libs/browser/CMakeLists.txt
+++ b/src/libs/browser/CMakeLists.txt
@@ -1,12 +1,29 @@
-add_library(Browser STATIC
-    searchtoolbar.cpp
-    settings.cpp
-    urlrequestinterceptor.cpp
-    webbridge.cpp
-    webcontrol.cpp
-    webpage.cpp
-    webview.cpp
-)
+if(USE_WEBKIT)
+    add_definitions(-DUSE_WEBKIT)
+    add_library(Browser STATIC
+        webcontrol.cpp
+        searchtoolbar.cpp
+        webbridge.cpp
+        webview.cpp
+        settings.cpp
+    )
 
-find_package(Qt${QT_VERSION_MAJOR} COMPONENTS WebChannel WebEngineWidgets REQUIRED)
-target_link_libraries(Browser PRIVATE Qt${QT_VERSION_MAJOR}::WebChannel Qt${QT_VERSION_MAJOR}::WebEngineWidgets)
+    target_link_libraries(Browser)
+
+    find_package(Qt${QT_VERSION_MAJOR} COMPONENTS WebKitWidgets REQUIRED)
+    target_link_libraries(Browser PRIVATE Qt${QT_VERSION_MAJOR}::WebKitWidgets)
+    message(STATUS "Using QtWebKitWidgets (backported!)")
+else()
+    add_library(Browser STATIC
+        searchtoolbar.cpp
+        settings.cpp
+        urlrequestinterceptor.cpp
+        webbridge.cpp
+        webcontrol.cpp
+        webpage.cpp
+        webview.cpp
+    )
+
+    find_package(Qt${QT_VERSION_MAJOR} COMPONENTS WebChannel WebEngineWidgets REQUIRED)
+    target_link_libraries(Browser PRIVATE Qt${QT_VERSION_MAJOR}::WebChannel Qt${QT_VERSION_MAJOR}::WebEngineWidgets)
+endif()
diff --git a/src/libs/browser/searchtoolbar.cpp b/src/libs/browser/searchtoolbar.cpp
index 5581857..01ca63c 100644
--- a/src/libs/browser/searchtoolbar.cpp
+++ b/src/libs/browser/searchtoolbar.cpp
@@ -10,12 +10,21 @@
 #include <QLineEdit>
 #include <QStyle>
 #include <QToolButton>
+#ifdef USE_WEBKIT
+#include <QWebPage>
+#include <QWebView>
+using QWEBPAGE = QWebPage;
+using QWEBVIEW = QWebView;
+#else
 #include <QWebEnginePage>
 #include <QWebEngineView>
+using QWEBPAGE = QWebEnginePage;
+using QWEBVIEW = QWebEngineView;
+#endif
 
 using namespace Zeal::Browser;
 
-SearchToolBar::SearchToolBar(QWebEngineView *webView, QWidget *parent)
+SearchToolBar::SearchToolBar(QWEBVIEW *webView, QWidget *parent)
     : QWidget(parent)
     , m_webView(webView)
 {
@@ -118,6 +127,7 @@ bool SearchToolBar::eventFilter(QObject *object, QEvent *event)
         case Qt::Key_Up:
         case Qt::Key_PageDown:
         case Qt::Key_PageUp:
+            QCoreApplication::sendEvent(m_webView, event);
             QCoreApplication::sendEvent(m_webView->focusProxy(), event);
             return true;
         default:
@@ -125,6 +135,7 @@ bool SearchToolBar::eventFilter(QObject *object, QEvent *event)
         }
     }
 
+
     return QWidget::eventFilter(object, event);
 }
 
@@ -155,8 +166,12 @@ void SearchToolBar::findNext()
         return;
     }
 
+#ifdef USE_WEBKIT
+    QWebPage::FindFlags ff = QWebPage::FindWrapsAroundDocument;
+#else
     QWebEnginePage::FindFlags ff;
-    ff.setFlag(QWebEnginePage::FindCaseSensitively, m_matchCaseButton->isChecked());
+#endif
+    ff.setFlag(QWEBPAGE::FindCaseSensitively, m_matchCaseButton->isChecked());
     m_webView->findText(m_lineEdit->text(), ff);
 }
 
@@ -166,15 +181,23 @@ void SearchToolBar::findPrevious()
         return;
     }
 
+#ifdef USE_WEBKIT
+    QWebPage::FindFlags ff = QWebPage::FindWrapsAroundDocument;
+#else
     QWebEnginePage::FindFlags ff;
-    ff.setFlag(QWebEnginePage::FindCaseSensitively, m_matchCaseButton->isChecked());
-    ff.setFlag(QWebEnginePage::FindBackward);
+#endif
+    ff.setFlag(QWEBPAGE::FindCaseSensitively, m_matchCaseButton->isChecked());
+    ff.setFlag(QWEBPAGE::FindBackward);
     m_webView->findText(m_lineEdit->text(), ff);
 }
 
 void SearchToolBar::hideHighlight()
 {
+#ifdef USE_WEBKIT
+    m_webView->findText(QString(), QWebPage::HighlightAllOccurrences);
+#else
     m_webView->findText(QString());
+#endif
 }
 
 void SearchToolBar::updateHighlight()
@@ -182,8 +205,12 @@ void SearchToolBar::updateHighlight()
     hideHighlight();
 
     if (m_highlightAllButton->isChecked()) {
+#ifdef USE_WEBKIT
+        QWebPage::FindFlags ff = QWebPage::HighlightAllOccurrences;
+#else
         QWebEnginePage::FindFlags ff;
-        ff.setFlag(QWebEnginePage::FindCaseSensitively, m_matchCaseButton->isChecked());
+#endif
+        ff.setFlag(QWEBPAGE::FindCaseSensitively, m_matchCaseButton->isChecked());
         m_webView->findText(m_lineEdit->text(), ff);
     }
 }
diff --git a/src/libs/browser/searchtoolbar.h b/src/libs/browser/searchtoolbar.h
index 8562f8d..9b7fee5 100644
--- a/src/libs/browser/searchtoolbar.h
+++ b/src/libs/browser/searchtoolbar.h
@@ -8,7 +8,12 @@
 
 class QLineEdit;
 class QToolButton;
+#ifdef USE_WEBKIT
+class QWebView;
+#else
 class QWebEngineView;
+using QWEBVIEW = QWebEngineView;
+#endif
 
 namespace Zeal {
 namespace Browser {
@@ -18,7 +23,11 @@ class SearchToolBar final : public QWidget
     Q_OBJECT
     Q_DISABLE_COPY_MOVE(SearchToolBar)
 public:
+#ifdef USE_WEBKIT
+    explicit SearchToolBar(QWebView *webView, QWidget *parent = nullptr);
+#else
     explicit SearchToolBar(QWebEngineView *webView, QWidget *parent = nullptr);
+#endif
 
     void setText(const QString &text);
     void activate();
@@ -44,7 +53,11 @@ private:
     QToolButton *m_highlightAllButton = nullptr;
     QToolButton *m_matchCaseButton = nullptr;
 
+#ifdef USE_WEBKIT
+    QWebView *m_webView = nullptr;
+#else
     QWebEngineView *m_webView = nullptr;
+#endif
 };
 
 } // namespace Browser
diff --git a/src/libs/browser/settings.cpp b/src/libs/browser/settings.cpp
index ee58955..6c85959 100644
--- a/src/libs/browser/settings.cpp
+++ b/src/libs/browser/settings.cpp
@@ -3,19 +3,33 @@
 
 #include "settings.h"
 
+#ifndef USE_WEBKIT
 #include "urlrequestinterceptor.h"
+#endif
 
 #include <core/settings.h>
 
 #include <QFileInfo>
 #include <QLoggingCategory>
+#ifdef USE_WEBKIT
+#include <QWebSettings>
+#include <QUrl>
+using QWEBSETTINGS = QWebSettings;
+#else
 #include <QWebEngineProfile>
 #include <QWebEngineSettings>
+using QWEBSETTINGS = QWebEngineSettings;
 #include <QWebEngineScript>
 #include <QWebEngineScriptCollection>
+#endif
 
 namespace {
+#ifdef USE_WEBKIT
+constexpr char DarkModeCssPath[] = ":/browser/assets/css/darkmode.css";
+constexpr char HighlightOnNavigateCssPath[] = ":/browser/assets/css/highlight-webkit.css";
+#else
 constexpr char HighlightOnNavigateCssPath[] = ":/browser/assets/css/highlight.css";
+#endif
 }
 
 using namespace Zeal;
@@ -32,15 +46,21 @@ Settings::Settings(Core::Settings *appSettings, QObject *parent)
     Q_ASSERT(!m_webProfile);
 
 #if QT_VERSION < QT_VERSION_CHECK(6, 0, 0)
+#ifdef USE_WEBKIT
+    m_webProfile = nullptr;
+#else
     // Create a new off-the-record profile.
     m_webProfile = new QWebEngineProfile(this);
+#endif //USE_WEBKIT
 #else
     // Default profile for Qt 6 is off-the-record.
     m_webProfile = QWebEngineProfile::defaultProfile();
 #endif
 
+#ifndef USE_WEBKIT
     // Setup URL interceptor.
     m_webProfile->setRequestInterceptor(new UrlRequestInterceptor(this));
+#endif
 
     // Listen to settings changes.
     connect(m_appSettings, &Core::Settings::updated, this, &Settings::applySettings);
@@ -49,18 +69,27 @@ Settings::Settings(Core::Settings *appSettings, QObject *parent)
 
 void Settings::applySettings()
 {
-    m_webProfile->settings()->setAttribute(QWebEngineSettings::ScrollAnimatorEnabled,
+#ifdef USE_WEBKIT
+    if (m_appSettings->isDarkModeEnabled()) {
+        setCustomStyleSheet(QStringLiteral("_zeal_darkmodestylesheet"), DarkModeCssPath);
+    }
+
+    QWEBSETTINGS::globalSettings()->setAttribute(QWEBSETTINGS::ScrollAnimatorEnabled,
+                                                 m_appSettings->isSmoothScrollingEnabled);
+#else
+    m_webProfile->settings()->setAttribute(QWEBSETTINGS::ScrollAnimatorEnabled,
                                            m_appSettings->isSmoothScrollingEnabled);
 
     // Qt 6.7+ does not require restart to enable dark mode.
 #if QT_VERSION >= QT_VERSION_CHECK(6, 7, 0)
-    m_webProfile->settings()->setAttribute(QWebEngineSettings::ForceDarkMode,
+    m_webProfile->settings()->setAttribute(QWEBSETTINGS::ForceDarkMode,
                                            m_appSettings->isDarkModeEnabled());
 #endif
 
     // Apply custom CSS.
     // TODO: Apply to all open pages.
     m_webProfile->scripts()->clear(); // Remove all scripts first.
+#endif
 
     if (m_appSettings->isHighlightOnNavigateEnabled) {
         setCustomStyleSheet(QStringLiteral("_zeal_highlightstylesheet"), HighlightOnNavigateCssPath);
@@ -73,8 +102,13 @@ void Settings::applySettings()
 
 QWebEngineProfile *Settings::defaultProfile()
 {
+#ifdef USE_WEBKIT
+    qCWarning(log) << Q_FUNC_INFO << "shouldn't be called when using the QtWebKit backend!!";
+    return nullptr;
+#else
     Q_ASSERT(m_webProfile);
     return m_webProfile;
+#endif
 }
 
 void Settings::setCustomStyleSheet(const QString &name, const QString &path)
@@ -86,6 +120,13 @@ void Settings::setCustomStyleSheet(const QString &name, const QString &path)
         return;
     }
 
+#ifdef USE_WEBKIT
+    QByteArray ba = QByteArrayLiteral("body { background-color: white; }");
+    ba += QString::fromUtf8(file.readAll()).toUtf8();
+    file.close();
+    const QString cssUrl = QLatin1String("data:text/css;charset=utf-8;base64,") + ba.toBase64();
+    QWEBSETTINGS::globalSettings()->setUserStyleSheetUrl(QUrl(cssUrl));
+#else
     QString stylesheet = QString::fromUtf8(file.readAll());
     file.close();
 
@@ -111,4 +152,5 @@ void Settings::setCustomStyleSheet(const QString &name, const QString &path)
     script.setRunsOnSubFrames(true);
     script.setWorldId(QWebEngineScript::ApplicationWorld);
     m_webProfile->scripts()->insert(script);
+#endif
 }
diff --git a/src/libs/browser/settings.h b/src/libs/browser/settings.h
index 9d665d7..75ffccc 100644
--- a/src/libs/browser/settings.h
+++ b/src/libs/browser/settings.h
@@ -6,7 +6,15 @@
 
 #include <QObject>
 
+#ifdef USE_WEBKIT
+class QWebEngineProfile : public QObject
+{
+public:
+    Q_OBJECT
+};
+#else
 class QWebEngineProfile;
+#endif
 
 namespace Zeal {
 
@@ -24,7 +32,6 @@ public:
     explicit Settings(Core::Settings *appSettings, QObject *parent = nullptr);
 
     static QWebEngineProfile *defaultProfile();
-
 private slots:
     void applySettings();
 
diff --git a/src/libs/browser/webcontrol.cpp b/src/libs/browser/webcontrol.cpp
index acf00b6..f02c148 100644
--- a/src/libs/browser/webcontrol.cpp
+++ b/src/libs/browser/webcontrol.cpp
@@ -12,10 +12,17 @@
 #include <QDataStream>
 #include <QKeyEvent>
 #include <QVBoxLayout>
+#ifdef USE_WEBKIT
+#include <QStyle>
+#include <QWebFrame>
+#include <QWebHistory>
+#include <QWebPage>
+#else
 #include <QWebChannel>
 #include <QWebEngineHistory>
 #include <QWebEnginePage>
 #include <QWebEngineSettings>
+#endif
 
 using namespace Zeal::Browser;
 
@@ -29,15 +36,28 @@ WebControl::WebControl(QWidget *parent)
     m_webView = new WebView();
     setFocusProxy(m_webView);
 
-    connect(m_webView->page(), &QWebEnginePage::linkHovered, this, [this](const QString &link) {
+#ifdef USE_WEBKIT
+    connect(m_webView->page(), &QWebPage::linkHovered, this, [this](const QString &link)
+#else
+    connect(m_webView->page(), &QWebEnginePage::linkHovered, this, [this](const QString &link)
+#endif
+    {
         if (Core::NetworkAccessManager::isLocalUrl(QUrl(link))) {
             return;
         }
 
+        setToolTip(link);
+#ifdef USE_WEBKIT
         m_webView->setToolTip(link);
+#endif
     });
+#ifdef USE_WEBKIT
+    connect(m_webView, &QWebView::titleChanged, this, &WebControl::titleChanged);
+    connect(m_webView, &QWebView::urlChanged, this, &WebControl::urlChanged);
+#else
     connect(m_webView, &QWebEngineView::titleChanged, this, &WebControl::titleChanged);
     connect(m_webView, &QWebEngineView::urlChanged, this, &WebControl::urlChanged);
+#endif
 
     layout->addWidget(m_webView);
 
@@ -76,22 +96,34 @@ void WebControl::resetZoom()
 
 void WebControl::setJavaScriptEnabled(bool enabled)
 {
+#ifdef USE_WEBKIT
+    m_webView->page()->settings()->setAttribute(QWebSettings::JavascriptEnabled, enabled);
+#else
     m_webView->page()->settings()->setAttribute(QWebEngineSettings::JavascriptEnabled, enabled);
+#endif
 }
 
 void WebControl::setWebBridgeObject(const QString &name, QObject *object)
 {
+#ifdef USE_WEBKIT
+    connect(m_webView->page()->mainFrame(), &QWebFrame::javaScriptWindowObjectCleared,
+            this, [=]() {
+        m_webView->page()->mainFrame()->addToJavaScriptWindowObject(name, object);
+    });
+#else
     QWebEnginePage *page = m_webView->page();
 
     QWebChannel *channel = new QWebChannel(page);
     channel->registerObject(name, object);
 
     page->setWebChannel(channel);
+#endif
 }
 
 void WebControl::load(const QUrl &url)
 {
     m_webView->load(url);
+    m_webView->setFocus();
 }
 
 void WebControl::activateSearchBar()
@@ -141,7 +173,11 @@ QUrl WebControl::url() const
     return m_webView->url();
 }
 
+#ifdef USE_WEBKIT
+QWebHistory *WebControl::history() const
+#else
 QWebEngineHistory *WebControl::history() const
+#endif
 {
     return m_webView->history();
 }
diff --git a/src/libs/browser/webcontrol.h b/src/libs/browser/webcontrol.h
index 877c0a1..10948c6 100644
--- a/src/libs/browser/webcontrol.h
+++ b/src/libs/browser/webcontrol.h
@@ -7,7 +7,11 @@
 
 #include <QWidget>
 
+#ifdef USE_WEBKIT
+class QWebHistory;
+#else
 class QWebEngineHistory;
+#endif
 
 namespace Zeal {
 namespace Browser {
@@ -30,7 +34,11 @@ public:
     QString title() const;
     QUrl url() const;
 
+#ifdef USE_WEBKIT
+    QWebHistory *history() const;
+#else
     QWebEngineHistory *history() const;
+#endif
     void restoreHistory(const QByteArray &array);
     QByteArray saveHistory() const;
 
diff --git a/src/libs/browser/webview.cpp b/src/libs/browser/webview.cpp
index f13fc73..00fc77e 100644
--- a/src/libs/browser/webview.cpp
+++ b/src/libs/browser/webview.cpp
@@ -5,7 +5,9 @@
 #include "webview.h"
 
 #include "webcontrol.h"
+#ifndef USE_WEBKIT
 #include "webpage.h"
+#endif
 
 #include <core/application.h>
 #include <core/settings.h>
@@ -18,29 +20,39 @@
 #include <QMenu>
 #include <QMessageBox>
 #include <QPushButton>
+#ifndef USE_WEBKIT
 #include <QVector>
 #include <QWebEngineProfile>
 #include <QWebEngineSettings>
+#endif
 #include <QWheelEvent>
 
+#ifndef USE_WEBKIT
 #if QT_VERSION < QT_VERSION_CHECK(6, 0, 0)
 #include <QWebEngineContextMenuData>
 #else
 #include <QWebEngineContextMenuRequest>
 #endif
+#endif //USE_WEBKIT
 
 using namespace Zeal::Browser;
 
 WebView::WebView(QWidget *parent)
-    : QWebEngineView(parent)
+    : QWEBVIEW(parent)
 {
+#ifdef USE_WEBKIT
+    page()->setNetworkAccessManager(Core::Application::instance()->networkManager());
+#else
     setPage(new WebPage(this));
+#endif
     setZoomLevel(defaultZoomLevel());
 
+#ifndef USE_WEBKIT
     // Enable plugins for PDF support.
     settings()->setAttribute(QWebEngineSettings::PluginsEnabled, true);
 
     settings()->setAttribute(QWebEngineSettings::FocusOnNavigationEnabled, false);
+#endif
 
     QApplication::instance()->installEventFilter(this);
 }
@@ -97,7 +109,7 @@ void WebView::resetZoom()
     setZoomLevel(defaultZoomLevel());
 }
 
-QWebEngineView *WebView::createWindow(QWebEnginePage::WebWindowType type)
+QWEBVIEW *WebView::createWindow(QWEBPAGE::WebWindowType type)
 {
     Q_UNUSED(type)
     return Core::Application::instance()->mainWindow()->createTab()->webControl()->m_webView;
@@ -106,16 +118,25 @@ QWebEngineView *WebView::createWindow(QWebEnginePage::WebWindowType type)
 void WebView::contextMenuEvent(QContextMenuEvent *event)
 {
 #if QT_VERSION < QT_VERSION_CHECK(6, 0, 0)
+#ifdef USE_WEBKIT
+    const QWebHitTestResult hitTestResult = hitTestContent(event->pos());
+    // Return standard menu for input fields.
+    if (hitTestResult.isContentEditable()) {
+        QWebView::contextMenuEvent(event);
+        return;
+    }
+#else
     const QWebEngineContextMenuData& contextData = page()->contextMenuData();
 
     if (!contextData.isValid()) {
-        QWebEngineView::contextMenuEvent(event);
+        QWEBVIEW::contextMenuEvent(event);
         return;
     }
+#endif //USE_WEBKIT
 #else
     QWebEngineContextMenuRequest *contextMenuRequest = lastContextMenuRequest();
     if (contextMenuRequest == nullptr) {
-        QWebEngineView::contextMenuEvent(event);
+        QWEBVIEW::contextMenuEvent(event);
         return;
     }
 #endif
@@ -129,7 +150,11 @@ void WebView::contextMenuEvent(QContextMenuEvent *event)
     m_contextMenu = new QMenu(this);
 
 #if QT_VERSION < QT_VERSION_CHECK(6, 0, 0)
+#ifdef USE_WEBKIT
+    const QUrl linkUrl = hitTestResult.linkUrl();
+#else
     QUrl linkUrl = contextData.linkUrl();
+#endif
 #else
     QUrl linkUrl = contextMenuRequest->linkUrl();
 #endif
@@ -139,7 +164,7 @@ void WebView::contextMenuEvent(QContextMenuEvent *event)
 
         if (scheme != QLatin1String("javascript")) {
             m_contextMenu->addAction(tr("Open Link in New Tab"), this, [this]() {
-                triggerPageAction(QWebEnginePage::WebAction::OpenLinkInNewWindow);
+                triggerPageAction(QWEBPAGE::WebAction::OpenLinkInNewWindow);
             });
         }
 
@@ -150,23 +175,26 @@ void WebView::contextMenuEvent(QContextMenuEvent *event)
                 });
             }
 
-            m_contextMenu->addAction(pageAction(QWebEnginePage::CopyLinkToClipboard));
+            m_contextMenu->addAction(pageAction(QWEBPAGE::CopyLinkToClipboard));
         }
     }
 
-
+#ifdef USE_WEBKIT
+    if (hitTestResult.isContentSelected())
+#else // USE_WEBKIT
 #if QT_VERSION < QT_VERSION_CHECK(6, 0, 0)
     const QString selectedText = contextData.selectedText();
 #else
     const QString selectedText = contextMenuRequest->selectedText();
 #endif
-
-    if (!selectedText.isEmpty()) {
+    if (!selectedText.isEmpty())
+#endif // USE_WEBKIT
+    {
         if (!m_contextMenu->isEmpty()) {
             m_contextMenu->addSeparator();
         }
 
-        m_contextMenu->addAction(pageAction(QWebEnginePage::Copy));
+        m_contextMenu->addAction(pageAction(QWEBPAGE::Copy));
     }
 
     if (!linkUrl.isValid() && url().scheme() != QLatin1String("qrc")) {
@@ -174,8 +202,8 @@ void WebView::contextMenuEvent(QContextMenuEvent *event)
             m_contextMenu->addSeparator();
         }
 
-        m_contextMenu->addAction(pageAction(QWebEnginePage::Back));
-        m_contextMenu->addAction(pageAction(QWebEnginePage::Forward));
+        m_contextMenu->addAction(pageAction(QWEBPAGE::Back));
+        m_contextMenu->addAction(pageAction(QWEBPAGE::Forward));
         m_contextMenu->addSeparator();
 
         m_contextMenu->addAction(tr("Open Page in Desktop Browser"), this, [this]() {
@@ -261,5 +289,23 @@ bool WebView::eventFilter(QObject *watched, QEvent *event)
         }
     }
 
-    return QWebEngineView::eventFilter(watched, event);
+    return QWEBVIEW::eventFilter(watched, event);
+}
+
+#ifdef USE_WEBKIT
+QWebHitTestResult WebView::hitTestContent(const QPoint &pos) const
+{
+    return page()->mainFrame()->hitTestContent(pos);
+}
+
+bool WebView::isExternalUrl(const QUrl &url)
+{
+    static const QStringList localSchemes = {
+        QStringLiteral("file"),
+        QStringLiteral("qrc"),
+    };
+
+    const QString scheme = url.scheme();
+    return !scheme.isEmpty() && !localSchemes.contains(scheme);
 }
+#endif
diff --git a/src/libs/browser/webview.h b/src/libs/browser/webview.h
index eb6137d..7d3e6a6 100644
--- a/src/libs/browser/webview.h
+++ b/src/libs/browser/webview.h
@@ -5,12 +5,21 @@
 #ifndef ZEAL_BROWSER_WEBVIEW_H
 #define ZEAL_BROWSER_WEBVIEW_H
 
+#ifdef USE_WEBKIT
+#include <QWebFrame>
+#include <QWebView>
+using QWEBVIEW = QWebView;
+using QWEBPAGE = QWebPage;
+#else
 #include <QWebEngineView>
+using QWEBVIEW = QWebEngineView;
+using QWEBPAGE = QWebEnginePage;
+#endif
 
 namespace Zeal {
 namespace Browser {
 
-class WebView final : public QWebEngineView
+class WebView final : public QWEBVIEW
 {
     Q_OBJECT
     Q_DISABLE_COPY_MOVE(WebView)
@@ -34,10 +43,14 @@ signals:
     void zoomLevelChanged();
 
 protected:
-    QWebEngineView *createWindow(QWebEnginePage::WebWindowType type) override;
+    QWEBVIEW *createWindow(QWEBPAGE::WebWindowType type) override;
     void contextMenuEvent(QContextMenuEvent *event) override;
 
 private:
+#ifdef USE_WEBKIT
+    QWebHitTestResult hitTestContent(const QPoint &pos) const;
+    static bool isExternalUrl(const QUrl &url);
+#endif
     bool handleMouseReleaseEvent(QMouseEvent *event);
     bool handleWheelEvent(QWheelEvent *event);
 
diff --git a/src/libs/core/CMakeLists.txt b/src/libs/core/CMakeLists.txt
index 0cc20e1..8d4d831 100644
--- a/src/libs/core/CMakeLists.txt
+++ b/src/libs/core/CMakeLists.txt
@@ -10,15 +10,27 @@ add_library(Core STATIC
 
 target_link_libraries(Core PRIVATE Registry Ui)
 
-find_package(Qt${QT_VERSION_MAJOR} COMPONENTS Network WebEngineCore Widgets REQUIRED)
-target_link_libraries(Core PRIVATE Qt${QT_VERSION_MAJOR}::Network Qt${QT_VERSION_MAJOR}::WebEngineCore Qt${QT_VERSION_MAJOR}::Widgets)
+if (USE_WEBKIT)
+     add_definitions(-DUSE_WEBKIT)
+     find_package(Qt${QT_VERSION_MAJOR} COMPONENTS Network WebKit Widgets REQUIRED)
+     target_link_libraries(Core PRIVATE Qt${QT_VERSION_MAJOR}::Network Qt${QT_VERSION_MAJOR}::WebKit Qt${QT_VERSION_MAJOR}::Widgets)
+     message(STATUS "Using QtWebKit (backported!)")
+else()
+	find_package(Qt${QT_VERSION_MAJOR} COMPONENTS Network WebEngineCore Widgets REQUIRED)
+	target_link_libraries(Core PRIVATE Qt${QT_VERSION_MAJOR}::Network Qt${QT_VERSION_MAJOR}::WebEngineCore Qt${QT_VERSION_MAJOR}::Widgets)
+endif()
 
 if(QT_VERSION_MAJOR EQUAL 6)
     find_package(Qt6 COMPONENTS WebEngineCore REQUIRED)
     target_link_libraries(Core PRIVATE Qt6::WebEngineCore)
 else()
-    find_package(Qt5 COMPONENTS WebEngineWidgets REQUIRED)
-    target_link_libraries(Core PRIVATE Qt5::WebEngineWidgets)
+    if (USE_WEBKIT)
+        find_package(Qt5 COMPONENTS WebKitWidgets REQUIRED)
+        target_link_libraries(Core PRIVATE Qt5::WebKitWidgets)
+    else()
+        find_package(Qt5 COMPONENTS WebEngineWidgets REQUIRED)
+        target_link_libraries(Core PRIVATE Qt5::WebEngineWidgets)
+    endif()
 endif()
 
 find_package(LibArchive QUIET)
diff --git a/src/libs/core/settings.cpp b/src/libs/core/settings.cpp
index 4ed53bb..6c199a4 100644
--- a/src/libs/core/settings.cpp
+++ b/src/libs/core/settings.cpp
@@ -12,8 +12,14 @@
 #include <QStandardPaths>
 #include <QUrl>
 #include <QUuid>
+#ifdef USE_WEBKIT
+#include <QWebSettings>
+using QWEBSETTINGS = QWebSettings;
+#else
 #include <QWebEngineProfile>
 #include <QWebEngineSettings>
+using QWEBSETTINGS = QWebEngineSettings;
+#endif
 
 #if QT_VERSION >= QT_VERSION_CHECK(6, 5, 0)
 #include <QStyleHints>
@@ -45,6 +51,12 @@ Settings::Settings(QObject *parent)
 #else
     qRegisterMetaType<ContentAppearance>("ContentAppearance");
     qRegisterMetaType<ExternalLinkPolicy>("ExternalLinkPolicy");
+#ifdef USE_WEBKIT
+    // Enable local storage due to https://github.com/zealdocs/zeal/issues/872.
+    QWEBSETTINGS *webSettings = QWebSettings::globalSettings();
+    webSettings->setLocalStoragePath(Application::cacheLocation() + QLatin1String("/localStorage"));
+    webSettings->setAttribute(QWebSettings::LocalStorageEnabled, true);
+#endif
 #endif
 
     load();
@@ -124,19 +136,23 @@ void Settings::load()
     }
 #endif
 
+#ifdef USE_WEBKIT
+    QWEBSETTINGS *webSettings = QWebSettings::globalSettings();
+#else
     // Fonts
-    QWebEngineSettings *webSettings = QWebEngineProfile::defaultProfile()->settings();
+    QWEBSETTINGS *webSettings = QWebEngineProfile::defaultProfile()->settings();
+#endif
     serifFontFamily = settings->value(QStringLiteral("serif_font_family"),
-                                      webSettings->fontFamily(QWebEngineSettings::SerifFont)).toString();
+                                      webSettings->fontFamily(QWEBSETTINGS::SerifFont)).toString();
     sansSerifFontFamily = settings->value(QStringLiteral("sans_serif_font_family"),
-                                          webSettings->fontFamily(QWebEngineSettings::SansSerifFont)).toString();
+                                          webSettings->fontFamily(QWEBSETTINGS::SansSerifFont)).toString();
     fixedFontFamily = settings->value(QStringLiteral("fixed_font_family"),
-                                      webSettings->fontFamily(QWebEngineSettings::FixedFont)).toString();
+                                      webSettings->fontFamily(QWEBSETTINGS::FixedFont)).toString();
 
-    static const QMap<QString, QWebEngineSettings::FontFamily> fontFamilies = {
-        {QStringLiteral("sans-serif"), QWebEngineSettings::SansSerifFont},
-        {QStringLiteral("serif"), QWebEngineSettings::SerifFont},
-        {QStringLiteral("monospace"), QWebEngineSettings::FixedFont}
+    static const QMap<QString, QWEBSETTINGS::FontFamily> fontFamilies = {
+        {QStringLiteral("sans-serif"), QWEBSETTINGS::SansSerifFont},
+        {QStringLiteral("serif"), QWEBSETTINGS::SerifFont},
+        {QStringLiteral("monospace"), QWEBSETTINGS::FixedFont}
     };
 
     defaultFontFamily = settings->value(QStringLiteral("default_font_family"),
@@ -147,23 +163,23 @@ void Settings::load()
         defaultFontFamily = QStringLiteral("serif");
     }
 
-    webSettings->setFontFamily(QWebEngineSettings::SansSerifFont, sansSerifFontFamily);
-    webSettings->setFontFamily(QWebEngineSettings::SerifFont, serifFontFamily);
-    webSettings->setFontFamily(QWebEngineSettings::FixedFont, fixedFontFamily);
+    webSettings->setFontFamily(QWEBSETTINGS::SansSerifFont, sansSerifFontFamily);
+    webSettings->setFontFamily(QWEBSETTINGS::SerifFont, serifFontFamily);
+    webSettings->setFontFamily(QWEBSETTINGS::FixedFont, fixedFontFamily);
 
     const QString defaultFontFamilyResolved = webSettings->fontFamily(fontFamilies.value(defaultFontFamily));
-    webSettings->setFontFamily(QWebEngineSettings::StandardFont, defaultFontFamilyResolved);
+    webSettings->setFontFamily(QWEBSETTINGS::StandardFont, defaultFontFamilyResolved);
 
     defaultFontSize = settings->value(QStringLiteral("default_font_size"),
-                                      webSettings->fontSize(QWebEngineSettings::DefaultFontSize)).toInt();
+                                      webSettings->fontSize(QWEBSETTINGS::DefaultFontSize)).toInt();
     defaultFixedFontSize = settings->value(QStringLiteral("default_fixed_font_size"),
-                                           webSettings->fontSize(QWebEngineSettings::DefaultFixedFontSize)).toInt();
+                                           webSettings->fontSize(QWEBSETTINGS::DefaultFixedFontSize)).toInt();
     minimumFontSize = settings->value(QStringLiteral("minimum_font_size"),
-                                      webSettings->fontSize(QWebEngineSettings::MinimumFontSize)).toInt();
+                                      webSettings->fontSize(QWEBSETTINGS::MinimumFontSize)).toInt();
 
-    webSettings->setFontSize(QWebEngineSettings::DefaultFontSize, defaultFontSize);
-    webSettings->setFontSize(QWebEngineSettings::DefaultFixedFontSize, defaultFixedFontSize);
-    webSettings->setFontSize(QWebEngineSettings::MinimumFontSize, minimumFontSize);
+    webSettings->setFontSize(QWEBSETTINGS::DefaultFontSize, defaultFontSize);
+    webSettings->setFontSize(QWEBSETTINGS::DefaultFixedFontSize, defaultFixedFontSize);
+    webSettings->setFontSize(QWEBSETTINGS::MinimumFontSize, minimumFontSize);
 
     isHighlightOnNavigateEnabled = settings->value(QStringLiteral("highlight_on_navigate"), true).toBool();
     customCssFile = settings->value(QStringLiteral("custom_css_file")).toString();
@@ -340,6 +356,7 @@ void Settings::migrate(QSettings *settings) const
         }
     }
 
+
     //
     // Pre 0.3
     //
diff --git a/src/libs/ui/CMakeLists.txt b/src/libs/ui/CMakeLists.txt
index 3015ea2..ed81312 100644
--- a/src/libs/ui/CMakeLists.txt
+++ b/src/libs/ui/CMakeLists.txt
@@ -22,5 +22,12 @@ add_library(Ui STATIC
 
 target_link_libraries(Ui PRIVATE Browser Sidebar QxtGlobalShortcut Widgets Registry Util)
 
-find_package(Qt${QT_VERSION_MAJOR} COMPONENTS WebEngineWidgets REQUIRED)
-target_link_libraries(Ui PRIVATE Qt${QT_VERSION_MAJOR}::WebEngineWidgets)
+if (USE_WEBKIT)
+    add_definitions(-DUSE_WEBKIT)
+    find_package(Qt${QT_VERSION_MAJOR} COMPONENTS WebKitWidgets REQUIRED)
+    target_link_libraries(Ui PRIVATE Qt${QT_VERSION_MAJOR}::WebKitWidgets)
+    message(STATUS "Using QtWebKitWidgets (backported!)")
+else()
+    find_package(Qt${QT_VERSION_MAJOR} COMPONENTS WebEngineWidgets REQUIRED)
+    target_link_libraries(Ui PRIVATE Qt${QT_VERSION_MAJOR}::WebEngineWidgets)
+endif()
diff --git a/src/libs/ui/browsertab.cpp b/src/libs/ui/browsertab.cpp
index b6611bf..1a33f88 100644
--- a/src/libs/ui/browsertab.cpp
+++ b/src/libs/ui/browsertab.cpp
@@ -20,7 +20,15 @@
 #include <QStyle>
 #include <QToolButton>
 #include <QVBoxLayout>
+#ifdef USE_WEBKIT
+#include <QWebHistory>
+using QWEBHISTORY = QWebHistory;
+using QWEBHISTORYITEM = QWebHistoryItem;
+#else
 #include <QWebEngineHistory>
+using QWEBHISTORY = QWebEngineHistory;
+using QWEBHISTORYITEM = QWebEngineHistoryItem;
+#endif
 
 using namespace Zeal;
 using namespace Zeal::WidgetUi;
@@ -64,11 +72,11 @@ BrowserTab::BrowserTab(QWidget *parent)
     auto backMenu = new QMenu(m_backButton);
     connect(backMenu, &QMenu::aboutToShow, this, [this, backMenu]() {
         backMenu->clear();
-        QWebEngineHistory *history = m_webControl->history();
-        QList<QWebEngineHistoryItem> items = history->backItems(10);
+        QWEBHISTORY *history = m_webControl->history();
+        QList<QWEBHISTORYITEM> items = history->backItems(10);
         for (auto it = items.crbegin(); it != items.crend(); ++it) {
             const QIcon icon = docsetIcon(it->url());
-            const QWebEngineHistoryItem item = *it;
+            const QWEBHISTORYITEM item = *it;
             backMenu->addAction(icon, it->title(), this, [=](bool) { history->goToItem(item); });
         }
     });
@@ -86,9 +94,9 @@ BrowserTab::BrowserTab(QWidget *parent)
     auto forwardMenu = new QMenu(m_forwardButton);
     connect(forwardMenu, &QMenu::aboutToShow, this, [this, forwardMenu]() {
         forwardMenu->clear();
-        QWebEngineHistory *history = m_webControl->history();
+        QWEBHISTORY *history = m_webControl->history();
         const auto forwardItems = history->forwardItems(10);
-        for (const QWebEngineHistoryItem &item : forwardItems) {
+        for (const QWEBHISTORYITEM &item : forwardItems) {
             const QIcon icon = docsetIcon(item.url());
             forwardMenu->addAction(icon, item.title(), this, [=](bool) { history->goToItem(item); });
         }
@@ -135,7 +143,6 @@ BrowserTab::BrowserTab(QWidget *parent)
 
         // TODO: Add custom 'Page has been removed' page.
         navigateToStartPage();
-
         // TODO: Cleanup history.
     });
 }
diff --git a/src/libs/ui/settingsdialog.cpp b/src/libs/ui/settingsdialog.cpp
index e6d4221..397348c 100644
--- a/src/libs/ui/settingsdialog.cpp
+++ b/src/libs/ui/settingsdialog.cpp
@@ -13,8 +13,14 @@
 
 #include <QDir>
 #include <QFileDialog>
+#ifdef USE_WEBKIT
+#include <QWebSettings>
+using QWEBSETTINGS = QWebSettings;
+#else
 #include <QWebEngineProfile>
 #include <QWebEngineSettings>
+using QWEBSETTINGS = QWebEngineSettings;
+#endif
 
 using namespace Zeal;
 using namespace Zeal::WidgetUi;
@@ -24,9 +30,9 @@ namespace {
 constexpr int AvailableFontSizes[] = {9, 10, 11, 12, 13, 14, 15, 16, 17, 18,
                                       20, 22, 24, 26, 28, 30, 32, 34, 36,
                                       40, 44, 48, 56, 64, 72};
-constexpr QWebEngineSettings::FontFamily BasicFontFamilies[] = {QWebEngineSettings::SerifFont,
-                                                                QWebEngineSettings::SansSerifFont,
-                                                                QWebEngineSettings::FixedFont};
+constexpr QWEBSETTINGS::FontFamily BasicFontFamilies[] = {QWEBSETTINGS::SerifFont,
+                                                                QWEBSETTINGS::SansSerifFont,
+                                                                QWEBSETTINGS::FixedFont};
 } // namespace
 
 SettingsDialog::SettingsDialog(QWidget *parent)
@@ -66,51 +72,55 @@ SettingsDialog::SettingsDialog(QWidget *parent)
         ui->globalHotKeyGroupBox->setToolTip(tr("Global shortcuts are not supported on the current platform."));
     }
 
-    QWebEngineSettings *webSettings = Browser::Settings::defaultProfile()->settings();
+#ifdef USE_WEBKIT
+    QWEBSETTINGS *webSettings = QWebSettings::globalSettings();
+#else
+    QWEBSETTINGS *webSettings = Browser::Settings::defaultProfile()->settings();
+#endif
 
     // Avoid casting in each connect.
     auto currentIndexChangedSignal
             = static_cast<void (QComboBox::*)(int)>(&QComboBox::currentIndexChanged);
 
-    auto syncStandardFont = [this, webSettings](QWebEngineSettings::FontFamily fontFamily,
+    auto syncStandardFont = [this, webSettings](QWEBSETTINGS::FontFamily fontFamily,
             const QFont &font) {
         const int index = ui->defaultFontComboBox->currentIndex();
         if (BasicFontFamilies[index] == fontFamily) {
-            webSettings->setFontFamily(QWebEngineSettings::StandardFont, font.family());
+            webSettings->setFontFamily(QWEBSETTINGS::StandardFont, font.family());
         }
     };
 
     connect(ui->defaultFontComboBox, currentIndexChangedSignal,
             this, [webSettings](int index) {
         const QString fontFamily = webSettings->fontFamily(BasicFontFamilies[index]);
-        webSettings->setFontFamily(QWebEngineSettings::StandardFont, fontFamily);
+        webSettings->setFontFamily(QWEBSETTINGS::StandardFont, fontFamily);
     });
 
     connect(ui->serifFontComboBox, &QFontComboBox::currentFontChanged,
             this, [webSettings, syncStandardFont](const QFont &font) {
-        webSettings->setFontFamily(QWebEngineSettings::SerifFont, font.family());
-        syncStandardFont(QWebEngineSettings::SerifFont, font);
+        webSettings->setFontFamily(QWEBSETTINGS::SerifFont, font.family());
+        syncStandardFont(QWEBSETTINGS::SerifFont, font);
     });
     connect(ui->sansSerifFontComboBox, &QFontComboBox::currentFontChanged,
             this, [webSettings, syncStandardFont](const QFont &font) {
-        webSettings->setFontFamily(QWebEngineSettings::SansSerifFont, font.family());
-        syncStandardFont(QWebEngineSettings::SansSerifFont, font);
+        webSettings->setFontFamily(QWEBSETTINGS::SansSerifFont, font.family());
+        syncStandardFont(QWEBSETTINGS::SansSerifFont, font);
     });
     connect(ui->fixedFontComboBox, &QFontComboBox::currentFontChanged,
             this, [webSettings, syncStandardFont](const QFont &font) {
-        webSettings->setFontFamily(QWebEngineSettings::FixedFont, font.family());
-        syncStandardFont(QWebEngineSettings::FixedFont, font);
+        webSettings->setFontFamily(QWEBSETTINGS::FixedFont, font.family());
+        syncStandardFont(QWEBSETTINGS::FixedFont, font);
     });
 
     connect(ui->fontSizeComboBox, currentIndexChangedSignal, this, [webSettings](int index) {
-        webSettings->setFontSize(QWebEngineSettings::DefaultFontSize, AvailableFontSizes[index]);
+        webSettings->setFontSize(QWEBSETTINGS::DefaultFontSize, AvailableFontSizes[index]);
     });
     connect(ui->fixedFontSizeComboBox, currentIndexChangedSignal, this, [webSettings](int index) {
-        webSettings->setFontSize(QWebEngineSettings::DefaultFixedFontSize, AvailableFontSizes[index]);
+        webSettings->setFontSize(QWEBSETTINGS::DefaultFixedFontSize, AvailableFontSizes[index]);
     });
     connect(ui->minFontSizeComboBox, currentIndexChangedSignal, this, [webSettings](int index) {
         const int fontSize = index == 0 ? 0 : AvailableFontSizes[index-1];
-        webSettings->setFontSize(QWebEngineSettings::MinimumFontSize, fontSize);
+        webSettings->setFontSize(QWEBSETTINGS::MinimumFontSize, fontSize);
     });
 
     loadSettings();
@@ -267,7 +277,7 @@ void SettingsDialog::saveSettings()
     settings->isFuzzySearchEnabled = ui->fuzzySearchCheckBox->isChecked();
 
     // Content Tab
-#if QT_VERSION < QT_VERSION_CHECK(6, 7, 0)
+#if QT_VERSION < QT_VERSION_CHECK(6, 7, 0) && !defined(USE_WEBKIT)
     // Applying dark mode requires restart.
     ui->appearanceLabel->setText(tr("Appearance (requires restart):"));
 #endif
