diff --git a/assets/CMakeLists.txt b/assets/CMakeLists.txt
index 24790bb..78fcd10 100644
--- a/assets/CMakeLists.txt
+++ b/assets/CMakeLists.txt
@@ -1,4 +1,4 @@
-if(UNIX AND NOT APPLE)
+if(UNIX)
     find_package(ECM REQUIRED NO_MODULE)
     set(CMAKE_MODULE_PATH ${ECM_KDE_MODULE_DIR})
     include(KDEInstallDirs)
diff --git a/src/app/CMakeLists.txt b/src/app/CMakeLists.txt
index 8a4f49e..d1cef7f 100644
--- a/src/app/CMakeLists.txt
+++ b/src/app/CMakeLists.txt
@@ -36,7 +36,8 @@ if(APPLE)
         MACOSX_BUNDLE_COPYRIGHT ${PROJECT_COPYRIGHT}
         RESOURCE "resources/zeal.icns"
     )
-elseif(UNIX)
+endif()
+if(UNIX)
     find_package(ECM REQUIRED NO_MODULE)
     set(CMAKE_MODULE_PATH ${ECM_KDE_MODULE_DIR})
     include(KDEInstallDirs)
