diff --git a/src/libs/browser/webview.cpp b/src/libs/browser/webview.cpp
index f13fc73..5411e66 100644
--- a/src/libs/browser/webview.cpp
+++ b/src/libs/browser/webview.cpp
@@ -62,7 +74,8 @@ void WebView::setZoomLevel(int level)
     m_zoomLevel = level;
 
     // Scale the webview relative to the DPI of the screen.
-    const double dpiZoomFactor = logicalDpiY() / 96.0;
+    const auto lDpIY = logicalDpiY();
+    const double dpiZoomFactor = lDpIY > 96 ? lDpIY / 86.0 : 1.0;
 
     setZoomFactor(availableZoomLevels().at(level) / 100.0 * dpiZoomFactor);
     emit zoomLevelChanged();
