diff --git a/CMakeLists.txt b/CMakeLists.txt
index 20dd221..749c306 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -7,7 +7,7 @@ set(CMAKE_ERROR_DEPRECATED TRUE)
 set(CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake")
 
 project(Zeal
-    VERSION 0.7.3
+    VERSION @VERSION@
     DESCRIPTION "A simple documentation browser."
     HOMEPAGE_URL "https://zealdocs.org"
     LANGUAGES CXX
@@ -23,6 +23,8 @@ endif()
 set(PROJECT_COMPANY_NAME "Oleg Shparber")
 set(PROJECT_COPYRIGHT "© 2013-2025 Oleg Shparber and other contributors")
 
+include(FeatureSummary)
+
 # Find available major Qt version. It will be stored in QT_VERSION_MAJOR.
 if(NOT ZEAL_USE_QT5)
     find_package(QT NAMES Qt6 COMPONENTS Core)
@@ -66,9 +66,9 @@ add_custom_target(zeal_version
     VERBATIM
 )
 
-if(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.24.0")
-    set(CMAKE_COMPILE_WARNING_AS_ERROR ON)
-endif()
+# if(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.24.0")
+#     set(CMAKE_COMPILE_WARNING_AS_ERROR ON)
+# endif()
 
 add_subdirectory(assets)
 add_subdirectory(src)
@@ -72,3 +74,5 @@ add_custom_target(zeal_version
 
 add_subdirectory(assets)
 add_subdirectory(src)
+
+feature_summary(WHAT ALL   FATAL_ON_MISSING_REQUIRED_PACKAGES)
