diff --git a/src/libs/core/application.cpp b/src/libs/core/application.cpp
index d86f59bc0e55937cb05c89d9ec8b9cb066418bb4..cfe0159559471fb993ee9a744f6a8b1575a13737 100644
--- a/src/libs/core/application.cpp
+++ b/src/libs/core/application.cpp
@@ -221,9 +221,9 @@ void Application::checkForUpdates(bool quiet)
         const auto latestVersion
                 = QVersionNumber::fromString(versionInfo[QLatin1String("version")].toString());
         if (latestVersion > version()) {
-            emit updateCheckDone(latestVersion.toString());
+            emit updateCheckDone(latestVersion.toString(), quiet);
         } else if (!quiet) {
-            emit updateCheckDone();
+            emit updateCheckDone(QString(), quiet);
         }
     });
 }
diff --git a/src/libs/core/application.h b/src/libs/core/application.h
index 0b4006c74e83c36278435076d628aad8ad607701..8a8ce56a967d6a036013f0c792e6837856252379 100644
--- a/src/libs/core/application.h
+++ b/src/libs/core/application.h
@@ -72,7 +72,7 @@ signals:
     void extractionCompleted(const QString &filePath);
     void extractionError(const QString &filePath, const QString &errorString);
     void extractionProgress(const QString &filePath, qint64 extracted, qint64 total);
-    void updateCheckDone(const QString &version = QString());
+    void updateCheckDone(const QString &version = QString(), bool quiet = false);
     void updateCheckError(const QString &message);
 
 private slots:
diff --git a/src/libs/ui/mainwindow.cpp b/src/libs/ui/mainwindow.cpp
index c4791ddb1c8446a8be46d36787d3e368f48d1e0a..5cd333120bc9ac54d2e89b6eeca551c3379f90f9 100644
--- a/src/libs/ui/mainwindow.cpp
+++ b/src/libs/ui/mainwindow.cpp
@@ -54,6 +54,7 @@
 #include <QTabBar>
 #include <QWebHistory>
 #include <QWebSettings>
+#include <QDebug>
 
 using namespace Zeal;
 using namespace Zeal::WidgetUi;
@@ -164,24 +165,28 @@ MainWindow::MainWindow(Core::Application *app, QWidget *parent) :
         QMessageBox::warning(this, QStringLiteral("Zeal"), message);
     });
 
-    connect(m_application, &Core::Application::updateCheckDone, this, [this](const QString &version) {
+    connect(m_application, &Core::Application::updateCheckDone, this, [this](const QString &version, bool quiet) {
         if (version.isEmpty()) {
             QMessageBox::information(this, QStringLiteral("Zeal"),
-                                     tr("You are using the latest version."));
+                                     tr("You are using the latest version (%1).").arg(QCoreApplication::applicationVersion()));
             return;
         }
 
         // TODO: Remove this ugly workaround for #637.
         qApp->setQuitOnLastWindowClosed(false);
-        const int ret
+        if (quiet) {
+            qInfo() << tr("Zeal %1 is available.").arg(version);
+        } else {
+            const int ret
                 = QMessageBox::information(this, QStringLiteral("Zeal"),
                                            tr("Zeal <b>%1</b> is available. Open download page?").arg(version),
                                            QMessageBox::Yes | QMessageBox::Default,
                                            QMessageBox::No | QMessageBox::Escape);
-        qApp->setQuitOnLastWindowClosed(true);
+            qApp->setQuitOnLastWindowClosed(true);
 
-        if (ret == QMessageBox::Yes) {
-            QDesktopServices::openUrl(QUrl(QStringLiteral("https://zealdocs.org/download.html")));
+            if (ret == QMessageBox::Yes) {
+                QDesktopServices::openUrl(QUrl(QStringLiteral("https://zealdocs.org/download.html")));
+            }
         }
     });
 
