diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8a9cf13..20dd221 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -31,7 +31,7 @@ endif()
 
 if(NOT QT_FOUND)
     find_package(QT NAMES Qt5 REQUIRED COMPONENTS Core)
-    set(QT_MINIMUM_VERSION 5.15.2)
+    set(QT_MINIMUM_VERSION 5.9.8)
 endif()
 
 message(NOTICE "Detected Qt version: ${QT_VERSION}")
diff --git a/src/libs/browser/settings.cpp b/src/libs/browser/settings.cpp
index 16b7a6e..ee58955 100644
--- a/src/libs/browser/settings.cpp
+++ b/src/libs/browser/settings.cpp
@@ -40,7 +40,7 @@ Settings::Settings(Core::Settings *appSettings, QObject *parent)
 #endif
 
     // Setup URL interceptor.
-    m_webProfile->setUrlRequestInterceptor(new UrlRequestInterceptor(this));
+    m_webProfile->setRequestInterceptor(new UrlRequestInterceptor(this));
 
     // Listen to settings changes.
     connect(m_appSettings, &Core::Settings::updated, this, &Settings::applySettings);
diff --git a/src/libs/ui/docsetlistitemdelegate.cpp b/src/libs/ui/docsetlistitemdelegate.cpp
index 3051cd8..a8c92a7 100644
--- a/src/libs/ui/docsetlistitemdelegate.cpp
+++ b/src/libs/ui/docsetlistitemdelegate.cpp
@@ -48,8 +48,12 @@ void DocsetListItemDelegate::paint(QPainter *painter,
     const QFontMetrics fontMetrics(font);
     const int margin = 4; // Random small number
 
-    QRect textRect = option.rect.adjusted(-margin, 0, -margin, 0);
+    QRect textRect = option.rect;
+#if QT_VERSION >= QT_VERSION_CHECK(5, 11, 0)
     textRect.setLeft(textRect.right() - fontMetrics.horizontalAdvance(text) - 2);
+#else
+    textRect.setLeft(textRect.right() - fontMetrics.width(text) - 2);
+#endif
     textRect = QStyle::visualRect(option.direction, option.rect, textRect);
     // Constant LeftToRight because we don't need to flip it any further.
     // Vertically align the text in the middle to match QCommonStyle behavior.
diff --git a/src/libs/ui/searchitemdelegate.cpp b/src/libs/ui/searchitemdelegate.cpp
index 1995ad2..0b022ec 100644
--- a/src/libs/ui/searchitemdelegate.cpp
+++ b/src/libs/ui/searchitemdelegate.cpp
@@ -122,8 +122,14 @@ void SearchItemDelegate::paint(QPainter *painter, const QStyleOptionViewItem &op
             if (matchIndex == -1 || matchIndex >= elidedText.length() - 1)
                 break;
 
+#if QT_VERSION >= QT_VERSION_CHECK(5, 11, 0)
             QRect highlightRect = textRect.adjusted(fm.horizontalAdvance(elidedText.left(matchIndex)), 2, 0, -2);
             highlightRect.setWidth(fm.horizontalAdvance(elidedText.mid(matchIndex, m_highlight.length())));
+#else
+            QRect highlightRect
+                    = textRect.adjusted(fm.width(elidedText.left(matchIndex)), 2, 0, -2);
+            highlightRect.setWidth(fm.width(elidedText.mid(matchIndex, m_highlight.length())));
+#endif
 
             QPainterPath path;
             path.addRoundedRect(highlightRect, 2, 2);
@@ -191,7 +197,11 @@ QSize SearchItemDelegate::sizeHint(const QStyleOptionViewItem &option,
         size.rwidth() = (decorationWidth + margin) * roles.size() + margin;
     }
 
+#if QT_VERSION >= QT_VERSION_CHECK(5, 11, 0)
     size.rwidth() += opt.fontMetrics.horizontalAdvance(index.data().toString()) + margin * 2;
+#else
+    size.rwidth() += opt.fontMetrics.width(index.data().toString()) + margin * 2;
+#endif
 
     return size;
 }
diff --git a/src/libs/ui/widgets/searchedit.cpp b/src/libs/ui/widgets/searchedit.cpp
index c32216c..da8a937 100644
--- a/src/libs/ui/widgets/searchedit.cpp
+++ b/src/libs/ui/widgets/searchedit.cpp
@@ -54,7 +54,12 @@ void SearchEdit::selectQuery()
     const int pos = hasSelectedText() ? selectionStart() : cursorPosition();
     const int queryPos = queryStart();
     const int textSize = text().size();
+#if QT_VERSION >= QT_VERSION_CHECK(5, 10, 0)
     if (pos >= queryPos && selectionEnd() < textSize) {
+#else
+    const int selectionEnd = hasSelectedText() ? pos + selectedText().size() : -1;
+    if (pos >= queryPos && selectionEnd < textSize) {
+#endif
         setSelection(queryPos, textSize);
         return;
     }
@@ -111,7 +116,7 @@ void SearchEdit::focusInEvent(QFocusEvent *event)
 {
     QLineEdit::focusInEvent(event);
 
-    // Do not change the default behavior when focused with mouse.
+    // Do not change the default behaviour when focused with mouse.
     if (event->reason() == Qt::MouseFocusReason)
         return;
 
@@ -124,13 +129,13 @@ void SearchEdit::showCompletions(const QString &newValue)
         return;
 
     const int frameWidth = style()->pixelMetric(QStyle::PM_DefaultFrameWidth);
-    const int textWidth = fontMetrics().horizontalAdvance(newValue);
+    const int textWidth = fontMetrics().width(newValue);
 
     if (m_prefixCompleter)
         m_prefixCompleter->setCompletionPrefix(text());
 
     const QString completed = currentCompletion(newValue).mid(newValue.size());
-    const QSize labelSize(fontMetrics().horizontalAdvance(completed), size().height());
+    const QSize labelSize(fontMetrics().width(completed), size().height());
     const int shiftX = static_cast<int>(window()->devicePixelRatioF() * (frameWidth + 2)) + textWidth;
 
     m_completionLabel->setMinimumSize(labelSize);
