--- examples/tools/orig.tools.pro	2015-04-07 11:26:12.000000000 +0200
+++ examples/tools/tools.pro	2015-04-14 21:56:01.000000000 +0200
@@ -7,8 +7,6 @@
                 i18n \
                 inputpanel \
                 contiguouscache \
-                plugandpaintplugins \
-                plugandpaint \
                 regexp \
                 settingseditor \
                 styleplugin \
