--- demos/orig.demos.pro	2015-04-14 21:58:20.000000000 +0200
+++ demos/demos.pro	2015-04-14 21:58:34.000000000 +0200
@@ -53,12 +53,12 @@
 
 !contains(QT_EDITION, Console):!cross_compile:!embedded:!wince*:SUBDIRS += demos_arthurplugin
 
-!cross_compile:{
-contains(QT_BUILD_PARTS, tools):{
-!wince*:SUBDIRS += demos_sqlbrowser demos_qtdemo
-wince*:SUBDIRS += demos_sqlbrowser
-}
-}
+#!cross_compile:{
+#contains(QT_BUILD_PARTS, tools):{
+SUBDIRS += demos_sqlbrowser demos_qtdemo
+#wince*:SUBDIRS += demos_sqlbrowser
+#}
+#}
 contains(QT_CONFIG, phonon):!static:SUBDIRS += demos_mediaplayer
 contains(QT_CONFIG, webkit):contains(QT_CONFIG, svg):!symbian:SUBDIRS += demos_browser
 contains(QT_CONFIG, declarative):SUBDIRS += demos_declarative demos_helper
--- demos/qtdemo/orig.qtdemo.pro	2015-04-14 21:58:50.000000000 +0200
+++ demos/qtdemo/qtdemo.pro	2015-04-14 21:59:10.000000000 +0200
@@ -74,7 +74,7 @@
 symbian: include($$QT_SOURCE_TREE/demos/symbianpkgrules.pri)
 
 # install
-target.path = $$[QT_INSTALL_BINS]
+target.path = $$[QT_INSTALL_DEMOS]
 sources.files = $$SOURCES $$HEADERS $$FORMS $$RESOURCES qtdemo.pro images xml *.ico *.icns *.rc *.plist
 sources.path = $$[QT_INSTALL_DEMOS]/qtdemo
 
