diff --git src/3rdparty/webkit/Source/WebKit.pro src/3rdparty/webkit/Source/WebKit.pro
index 9be0f4a..531e933 100644
--- src/3rdparty/webkit/Source/WebKit.pro
+++ src/3rdparty/webkit/Source/WebKit.pro
@@ -40,3 +40,6 @@ symbian {
 }
 
 include(WebKit/qt/docs/docs.pri)
+
+QMAKE_CXXFLAGS -= -flto
+QMAKE_LFLAGS -= -flto
diff --git src/3rdparty/webkit/Source/WebKit/qt/QtWebKit.pro src/3rdparty/webkit/Source/WebKit/qt/QtWebKit.pro
index 32798f7..8880846 100644
--- src/3rdparty/webkit/Source/WebKit/qt/QtWebKit.pro
+++ src/3rdparty/webkit/Source/WebKit/qt/QtWebKit.pro
@@ -355,3 +355,6 @@ symbian {
         }
     }
 }
+
+QMAKE_CXXFLAGS -= -flto
+QMAKE_LFLAGS -= -flto
diff --git src/script/script.pro src/script/script.pro
index d947276..849ab6e 100644
--- src/script/script.pro
+++ src/script/script.pro
@@ -107,3 +107,6 @@ integrity {
 
 # WebKit doesn't compile in C++0x mode
 *-g++*:QMAKE_CXXFLAGS -= -std=c++0x -std=gnu++0x
+
+QMAKE_CXXFLAGS -= -flto
+QMAKE_LFLAGS -= -flto
