diff --git src/gui/text/qfontengine_coretext.mm src/gui/text/qfontengine_coretext.mm
index 204d685..adb2074 100644
--- src/gui/text/qfontengine_coretext.mm
+++ src/gui/text/qfontengine_coretext.mm
@@ -886,7 +969,7 @@ void QCoreTextFontEngine::getUnscaledGlyph(glyph_t glyph, QPainterPath *path, gl
 
 QFixed QCoreTextFontEngine::emSquareSize() const
 {
-    return QFixed::QFixed(int(CTFontGetUnitsPerEm(ctfont)));
+    return QFixed(int(CTFontGetUnitsPerEm(ctfont)));
 }
 
 QFontEngine *QCoreTextFontEngine::cloneWithSize(qreal pixelSize) const

