diff --git src/corelib/tools/qsimd_p.h src/corelib/tools/qsimd_p.h
index 82a70a2..5c6fa2e 100644
--- src/corelib/tools/qsimd_p.h
+++ src/corelib/tools/qsimd_p.h
@@ -87,6 +87,12 @@ QT_BEGIN_HEADER
 #include <tmmintrin.h>
 #endif
 
+// Building with LTO and clang: exclude SSE4 intrinsics
+#undef QT_HAVE_SSE4_1
+#undef QT_HAVE_SSE4_2
+#undef __SSE4_1__
+#undef __SSE4_2__
+
 // SSE4.1 intrinsics
 #if defined(QT_HAVE_SSE4_1) && (defined(__SSE4_1__) || defined(Q_CC_MSVC))
 #include <smmintrin.h>
diff --git src/corelib/tools/qstring.cpp src/corelib/tools/qstring.cpp
index 3b046ed..c416862 100644
--- src/corelib/tools/qstring.cpp
+++ src/corelib/tools/qstring.cpp
@@ -3580,6 +3580,12 @@ bool QString::endsWith(const QChar &c, Qt::CaseSensitivity cs) const
     Use toLocal8Bit() instead.
 */
 
+// Building with LTO and clang: exclude SSE4 intrinsics
+#undef QT_HAVE_SSE4_1
+#undef QT_HAVE_SSE4_2
+#undef __SSE4_1__
+#undef __SSE4_2__
+
 #if defined(QT_ALWAYS_HAVE_SSE2)
 static inline __m128i mergeQuestionMarks(__m128i chunk)
 {
