--- QtCore.framework/Headers/orig.qconfig.h	2014-12-16 21:48:09.000000000 +0100
+++ QtCore.framework/Headers/qconfig.h	2015-01-01 11:36:10.000000000 +0100
@@ -74,12 +74,6 @@
 # define QT_NO_EGL
 #endif
 
-#if defined(QT_NO_EXCEPTIONS) && defined(QT_EXCEPTIONS)
-# undef QT_NO_EXCEPTIONS
-#elif !defined(QT_NO_EXCEPTIONS) && !defined(QT_EXCEPTIONS)
-# define QT_NO_EXCEPTIONS
-#endif
-
 #if defined(QT_NO_FREETYPE) && defined(QT_FREETYPE)
 # undef QT_NO_FREETYPE
 #elif !defined(QT_NO_FREETYPE) && !defined(QT_FREETYPE)
