--- src/gui/widgets/orig.qmenu_mac.mm	2014-04-10 20:37:12.000000000 +0200
+++ src/gui/widgets/qmenu_mac.mm	2014-09-16 16:25:00.000000000 +0200
@@ -638,7 +638,7 @@
 
 static NSMenuItem *createNSMenuItem(const QString &title)
 {
-    NSMenuItem *item = [[NSMenuItem alloc] 
+    NSMenuItem *item = [[NSMenuItem alloc]
                          initWithTitle:qt_mac_QStringToNSString(title)
                          action:@selector(qtDispatcherToQAction:) keyEquivalent:@""];
     [item setTarget:nil];
@@ -852,32 +852,7 @@
 #endif
         break;
     case QAction::TextHeuristicRole: {
-        QString aboutString = QMenuBar::tr("About").toLower();
-        if (t.startsWith(aboutString) || t.endsWith(aboutString)) {
-            if (t.indexOf(QRegExp(QString::fromLatin1("qt$"), Qt::CaseInsensitive)) == -1) {
-#ifndef QT_MAC_USE_COCOA
-                ret = kHICommandAbout;
-#else
-                ret = [loader aboutMenuItem];
-#endif
-            } else {
-#ifndef QT_MAC_USE_COCOA
-                ret = kHICommandAboutQt;
-#else
-                ret = [loader aboutQtMenuItem];
-#endif
-            }
-        } else if (t.startsWith(QMenuBar::tr("Config").toLower())
-                   || t.startsWith(QMenuBar::tr("Preference").toLower())
-                   || t.startsWith(QMenuBar::tr("Options").toLower())
-                   || t.startsWith(QMenuBar::tr("Setting").toLower())
-                   || t.startsWith(QMenuBar::tr("Setup").toLower())) {
-#ifndef QT_MAC_USE_COCOA
-            ret = kHICommandPreferences;
-#else
-            ret = [loader preferencesMenuItem];
-#endif
-        } else if (t.startsWith(QMenuBar::tr("Quit").toLower())
+	   if (t.startsWith(QMenuBar::tr("Quit").toLower())
                    || t.startsWith(QMenuBar::tr("Exit").toLower())) {
 #ifndef QT_MAC_USE_COCOA
             ret = kHICommandQuit;
