diff --git cmake/orig.FindRust.cmake cmake/FindRust.cmake
index 1a91db3..cda0281 100644
--- cmake/orig.FindRust.cmake
+++ cmake/FindRust.cmake
@@ -447,7 +447,7 @@ set(RUSTFLAGS "")
 
 if(NOT CMAKE_BUILD_TYPE)
     set(CARGO_BUILD_TYPE "debug")
-elseif(${CMAKE_BUILD_TYPE} STREQUAL "Release" OR ${CMAKE_BUILD_TYPE} STREQUAL "MinSizeRel")
+elseif(${CMAKE_BUILD_TYPE} STREQUAL "Release" OR ${CMAKE_BUILD_TYPE} STREQUAL "MinSizeRel" OR ${CMAKE_BUILD_TYPE} STREQUAL "MacPorts")
     set(CARGO_BUILD_TYPE "release")
     list(APPEND CARGO_ARGS "--release")
 elseif(${CMAKE_BUILD_TYPE} STREQUAL "RelWithDebInfo")
