diff --git CMakeLists.txt CMakeLists.txt
index cc1022f..543f72e 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -28,7 +28,7 @@ check_include_file("unistd.h" HAVE_UNISTD_H)
 include(CheckLibraryExists)
 check_library_exists(crc32c crc32c_value "" HAVE_CRC32C)
 check_library_exists(snappy snappy_compress "" HAVE_SNAPPY)
-check_library_exists(tcmalloc malloc "" HAVE_TCMALLOC)
+# check_library_exists(tcmalloc malloc "" HAVE_TCMALLOC)
 
 include(CheckSymbolExists)
 check_symbol_exists(fdatasync "unistd.h" HAVE_FDATASYNC)
