diff --git a/qtbase/src/gui/kernel/qguiapplication.cpp b/qtbase/src/gui/kernel/qguiapplication.cpp
index 7beab72..1252af7 100644
--- a/qtbase/src/gui/kernel/qguiapplication.cpp
+++ b/qtbase/src/gui/kernel/qguiapplication.cpp
@@ -1397,6 +1397,10 @@ void QGuiApplicationPrivate::init()
         argc = j;
     }
 
+    // https://bugreports.qt-project.org/browse/QTBUG-32974
+    if (qgetenv("QT_LOAD_TESTABILITY").toInt() > 0)
+        loadTestability = true;
+
     // Load environment exported generic plugins
     QByteArray envPlugins = qgetenv("QT_QPA_GENERIC_PLUGINS");
     if (!envPlugins.isEmpty())
