diff --git misc/orig.Makefile misc/Makefile
index d5012b9..e132787 100644
--- misc/orig.Makefile
+++ misc/Makefile
@@ -45,7 +45,7 @@ CFILES2 = lipo.c size.c strings.c nm.c checksyms.c inout.c \
 	 indr.c strip.c segedit.c cmpdylib.c \
 	 pagestuff.c redo_prebinding.c seg_addr_table.c \
 	 check_dylib.c seg_hack.c install_name_tool.c \
-	 codesign_allocate.c ctf_insert.c bitcode_strip.c
+	 codesign_allocate.c ctf_insert.c bitcode_strip.c vtool.c
 INSTALL_FILES = $(CFILES1) $(CFILES2) Makefile notes Info.plist
 
 OS_PROGS = lipo.NEW strings.NEW
@@ -54,7 +54,7 @@ PROGS = $(OS_PROGS) size.NEW nm.NEW libtool.NEW checksyms.NEW \
 	indr.NEW strip.NEW nmedit.NEW segedit.NEW cmpdylib.NEW \
 	pagestuff.NEW redo_prebinding.NEW seg_addr_table.NEW check_dylib.NEW \
 	seg_hack.NEW install_name_tool.NEW codesign_allocate.NEW \
-	ctf_insert.NEW bitcode_strip.NEW
+	ctf_insert.NEW bitcode_strip.NEW vtool.NEW
 
 all:	$(OFILE_DIR) $(SYMROOT) $(PROGS) lib_ofiles
 
@@ -251,6 +251,14 @@ bitcode_strip.NEW: bitcode_strip.o
 		$(OFILE_DIR)/bitcode_strip.private.o
 	$(DSYMUTIL) $(SYMROOT)/bitcode_strip.NEW
 
+vtool.NEW: vtool.o
+	$(CC) $(RC_CFLAGS) -nostdlib -r \
+		-o $(OBJROOT)/vtool.private.o \
+		$(OFILE_DIR)/vtool.o $(LIBSTUFF)
+	$(CC) $(RC_CFLAGS) $(SDK) -o $(SYMROOT)/vtool.NEW \
+		$(OFILE_DIR)/vtool.private.o
+	$(DSYMUTIL) $(SYMROOT)/vtool.NEW
+
 
 clean: shlib_clean
 	-rm -f \
@@ -359,6 +367,8 @@ dt_install:
 	install -c -s -m 555 $(SYMROOT)/nmedit.NEW $(DSTROOT)$(USRBINDIR)/nmedit
 	install -c -s -m 555 $(SYMROOT)/install_name_tool.NEW \
 			  $(DSTROOT)$(USRBINDIR)/install_name_tool
+	install -c -s -m 555 $(SYMROOT)/vtool.NEW \
+			  $(DSTROOT)$(USRBINDIR)/vtool
 	install -c -s -m 555 $(SYMROOT)/codesign_allocate.NEW \
 			  $(DSTROOT)$(USRBINDIR)/codesign_allocate
 	(cd $(DSTROOT)$(USRBINDIR); rm -f codesign_allocate-p; \
diff --git man/orig.Makefile man/Makefile
index d3b3aca..f3122f2 100644
--- man/orig.Makefile
+++ man/Makefile
@@ -10,7 +10,7 @@ COMMON_MAN1 = as.1 nm.1 otool.1 ranlib.1 segedit.1 size.1 \
 	      strip.1 libtool.1 cmpdylib.1 pagestuff.1 \
 	      redo_prebinding.1 nmedit.1 install_name_tool.1 \
 	      codesign_allocate.1 ctf_insert.1 bitcode_strip.1 \
-	      llvm-otool.1 nm-classic.1 size-classic.1 otool-classic.1
+	      llvm-otool.1 nm-classic.1 size-classic.1 otool-classic.1 vtool.1
 DEAD = gprof.1
 
 DYLD_MAN3 = dyld.3 NSModule.3 NSObjectFileImage.3 dyld_debug.3
