diff --git as/orig.driver.c as/driver.c
index c3adbb5..b3d0e2a 100644
--- as/orig.driver.c
+++ as/driver.c
@@ -33,7 +33,7 @@ char **envp)
 
     int i=0, j=0;
     uint32_t count=0, verbose=0, run_clang=0;
-    char *p=NULL, c=NULL, *arch_name=NULL, *as=NULL, *as_local=NULL;
+    char *p=NULL, c=0, *arch_name=NULL, *as=NULL, *as_local=NULL;
     char **new_argv;
     const char *CLANG = "clang";
     char *prefix=NULL, buf[MAXPATHLEN], resolved_name[PATH_MAX];
