diff --git orig.CMakeLists.txt CMakeLists.txt
index 1f2e023..f38e66c 100644
--- orig.CMakeLists.txt
+++ CMakeLists.txt
@@ -65,9 +65,9 @@ if(UNIX)
 	option(ENABLE_REPRODUCIBLE_BUILDS "Enable reproducible builds"                 OFF)
 endif()
 
-if(APPLE)
+# if(APPLE)
 	option(USE_ICONV           "Link with and use iconv library"                    ON)
-endif()
+# endif()
 
 if(MSVC)
 	# This option must match the settings used in your program, in particular if you
diff --git cmake/orig.FindIconv.cmake cmake/FindIconv.cmake
index 9e6ded9..b7b44bf 100644
--- cmake/orig.FindIconv.cmake
+++ cmake/FindIconv.cmake
@@ -12,7 +12,8 @@ if(ICONV_INCLUDE_DIR AND ICONV_LIBRARIES)
 endif()
 
 find_path(ICONV_INCLUDE_DIR iconv.h)
-check_function_exists(iconv_open libc_has_iconv)
+# for the use of an external libiconv:
+# check_function_exists(iconv_open libc_has_iconv)
 find_library(iconv_lib NAMES iconv libiconv libiconv-2 c)
 
 if(ICONV_INCLUDE_DIR AND libc_has_iconv)
