diff --git bus/Makefile.am bus/Makefile.am
index 1595432..1f406b6 100644
--- bus/Makefile.am
+++ bus/Makefile.am
@@ -49,6 +49,7 @@ EXTRA_DIST =					\
 	legacy-config/session.conf.in		\
 	legacy-config/system.conf.in		\
 	org.freedesktop.dbus-session.plist.in	\
+	org.freedesktop.dbus-system.plist.in	\
 	example-system-enable-stats.conf.in	\
 	example-session-disable-stats.conf.in	\
 	$(NULL)
@@ -74,6 +75,8 @@ dist_examples_DATA = \
 if DBUS_ENABLE_LAUNCHD
 agentdir=$(LAUNCHD_AGENT_DIR)
 agent_DATA=org.freedesktop.dbus-session.plist
+daemondir=$(LAUNCHD_DAEMON_DIR)
+daemon_DATA=org.freedesktop.dbus-system.plist
 endif
 
 if DBUS_BUS_ENABLE_KQUEUE
diff --git configure.ac configure.ac
index 8772abb..8a032ea 100644
--- configure.ac
+++ configure.ac
@@ -122,6 +122,7 @@ LT_LANG([Windows Resource])
 # Set some internal variables depending on the platform for later use.
 dbus_win=no
 dbus_cygwin=no
+dbus_darwin=no
 dbus_unix=no
 case "${host}" in
     *-mingw32ce*)
@@ -135,6 +136,10 @@ case "${host}" in
         dbus_cygwin=yes
         dbus_unix=yes
         ;;
+    *darwin*)
+        dbus_darwin=yes
+        dbus_unix=yes
+        ;;
     *)
         dbus_unix=yes
        ;;
@@ -176,7 +181,7 @@ fi
 
 # For best security, assume that all non-Windows platforms can do
 # credentials-passing.
-AS_IF([test "$dbus_win" = yes],
+AS_IF([test "$dbus_win" = yes || test "$dbus_darwin" = yes],
     [DBUS_SESSION_CONF_MAYBE_AUTH_EXTERNAL="<!--<auth>EXTERNAL</auth>-->"],
     [DBUS_SESSION_CONF_MAYBE_AUTH_EXTERNAL="<auth>EXTERNAL</auth>"])
 AC_SUBST([DBUS_SESSION_CONF_MAYBE_AUTH_EXTERNAL])
@@ -225,7 +230,8 @@ AC_ARG_WITH(system-pid-file, AS_HELP_STRING([--with-system-pid-file=[pidfile]],[
 AC_ARG_WITH(system-socket, AS_HELP_STRING([--with-system-socket=[filename]],[UNIX domain socket for systemwide daemon]))
 AC_ARG_WITH(console-auth-dir, AS_HELP_STRING([--with-console-auth-dir=[dirname]],[directory to check for console ownerhip]))
 AC_ARG_WITH(console-owner-file, AS_HELP_STRING([--with-console-owner-file=[filename]],[file whose owner determines current console owner]))
-AC_ARG_WITH(launchd-agent-dir, AS_HELP_STRING([--with-launchd-agent-dir=[dirname]],[directory to put the launchd agent (default: /Library/LaunchAgents)]))
+AC_ARG_WITH(launchd-agent-dir, AS_HELP_STRING([--with-launchd-agent-dir=[dirname]],[directory to put the launchd agent (session bus) (default: /Library/LaunchAgents)]))
+AC_ARG_WITH(launchd-daemon-dir, AS_HELP_STRING([--with-launchd-daemon-dir=[dirname]],[directory to put the launchd daemon (system bus) (default: /Library/LaunchDaemons)]))
 AC_ARG_WITH(dbus_user, AS_HELP_STRING([--with-dbus-user=<user>],[User for running the DBUS daemon (messagebus)]))
 AC_ARG_WITH([test_user],
   [AS_HELP_STRING([--with-test-user=<user>],
@@ -966,6 +972,15 @@ fi
 
 AC_SUBST(LAUNCHD_AGENT_DIR)
 
+#### Directory to place launchd daemon file
+if test "x$with_launchd_daemon_dir" = "x"; then
+   LAUNCHD_DAEMON_DIR="/Library/LaunchDaemons"
+else
+   LAUNCHD_DAEMON_DIR="$with_launchd_daemon_dir"
+fi
+
+AC_SUBST(LAUNCHD_DAEMON_DIR)
+
 dnl console owner file
 if test x$enable_console_owner_file = xno ; then
     have_console_owner_file=no;
@@ -1734,6 +1749,7 @@ bus/legacy-config/session.conf
 bus/example-system-enable-stats.conf
 bus/example-session-disable-stats.conf
 bus/org.freedesktop.dbus-session.plist
+bus/org.freedesktop.dbus-system.plist
 bus/dbus.service
 bus/dbus.socket
 bus/systemd-user/dbus.service
@@ -1829,6 +1845,7 @@ echo "
 "
 if test x$have_launchd = xyes; then
         echo "        launchd agent dir:        ${LAUNCHD_AGENT_DIR}"
+        echo "        launchd daemon dir:       ${LAUNCHD_DAEMON_DIR}"
 fi
 echo
 
diff --git dbus/dbus-bus.c dbus/dbus-bus.c
index 35fc135..bbcb5ad 100644
--- dbus/dbus-bus.c
+++ dbus/dbus-bus.c
@@ -519,6 +519,10 @@ out:
     _DBUS_ASSERT_ERROR_IS_SET (error);
 
   _DBUS_UNLOCK (bus);
+#ifdef DBUS_ENABLE_LAUNCHD
+  _dbus_verbose ("internal_bus_get(type=%d,private=%d) : found and registered connection to %s",
+         type, private, address);
+#endif
   return connection;
 }
 
diff --git dbus/dbus-sysdeps-unix.c dbus/dbus-sysdeps-unix.c
index 6b32df6..b5f2ca6 100644
--- dbus/dbus-sysdeps-unix.c
+++ dbus/dbus-sysdeps-unix.c
@@ -4329,28 +4329,45 @@ _dbus_lookup_launchd_socket (DBusString *socket_path,
 
   _DBUS_ASSERT_ERROR_IS_CLEAR (error);
 
+  const char *fromEnv = getenv (launchd_env_var);
+  dbus_bool_t hasEnvVar = (fromEnv && *fromEnv);
+
   if (_dbus_check_setuid ())
     {
-      dbus_set_error_const (error, DBUS_ERROR_NOT_SUPPORTED,
-                            "Unable to find launchd socket when setuid");
-      return FALSE;
-    }
+      if (!hasEnvVar)
+        {
+          dbus_set_error_const (error, DBUS_ERROR_NOT_SUPPORTED,
+                                "Unable to find launchd socket when setuid");
+          return FALSE;
+        }
 
-  i = 0;
-  argv[i] = "launchctl";
-  ++i;
-  argv[i] = "getenv";
-  ++i;
-  argv[i] = (char*)launchd_env_var;
-  ++i;
-  argv[i] = NULL;
-  ++i;
+      i = 0;
+      argv[i] = "launchctl";
+      ++i;
+      argv[i] = "getenv";
+      ++i;
+      argv[i] = (char*)launchd_env_var;
+      ++i;
+      argv[i] = NULL;
+      ++i;
 
-  _dbus_assert (i == _DBUS_N_ELEMENTS (argv));
+      _dbus_assert (i == _DBUS_N_ELEMENTS (argv));
+
+      if (!_read_subprocess_line_argv(argv[0], TRUE, argv, socket_path, error) && !hasEnvVar)
+        {
+          return FALSE;
+        }
+      else
+        {
+          // prefer the value obtained from launchctl
+          hasEnvVar = FALSE;
+        }
+    }
 
-  if (!_read_subprocess_line_argv(argv[0], TRUE, argv, socket_path, error))
+  if (hasEnvVar)
     {
-      return FALSE;
+      _dbus_string_append (socket_path, fromEnv);
+      return TRUE;
     }
 
   /* no error, but no result either */
@@ -4376,13 +4393,6 @@ _dbus_lookup_session_address_launchd (DBusString *address, DBusError  *error)
   dbus_bool_t valid_socket;
   DBusString socket_path;
 
-  if (_dbus_check_setuid ())
-    {
-      dbus_set_error_const (error, DBUS_ERROR_NOT_SUPPORTED,
-                            "Unable to find launchd socket when setuid");
-      return FALSE;
-    }
-
   if (!_dbus_string_init (&socket_path))
     {
       _DBUS_SET_OOM (error);
