--- data/orig.CMakeLists.txt	2015-10-05 10:29:56.000000000 +0200
+++ data/CMakeLists.txt	2015-11-07 22:32:10.000000000 +0100
@@ -2,7 +2,7 @@
 
 add_executable(parsetrigrams parsetrigrams.cpp)
 
-TARGET_LINK_LIBRARIES(parsetrigrams PUBLIC Qt5::Core)
+TARGET_LINK_LIBRARIES(parsetrigrams PUBLIC Qt5::Core PRIVATE Qt5::QspXDG)
 #INSTALL(TARGETS parsetrigrams DESTINATION ${CMAKE_CURRENT_BINARY_DIR})
 
 macro(create_trigrams_map _target_name _in_DIR _out_FILE)
Exit 1
--- src/core/orig.CMakeLists.txt	2015-10-05 10:29:56.000000000 +0200
+++ src/core/CMakeLists.txt	2015-11-07 22:32:56.000000000 +0100
@@ -30,7 +30,7 @@
   REQUIRED_HEADERS SonnetCore_HEADERS
 )
 
-target_link_libraries(KF5SonnetCore PUBLIC Qt5::Core)
+target_link_libraries(KF5SonnetCore PUBLIC Qt5::Core PRIVATE Qt5::QspXDG)
 
 set_target_properties(KF5SonnetCore PROPERTIES VERSION ${SONNET_VERSION_STRING}
                                                SOVERSION ${SONNET_SOVERSION}
