diff --git src/kdeinit/CMakeLists.txt src/kdeinit/CMakeLists.txt
index f94db71..58f4c98 100644
--- src/kdeinit/CMakeLists.txt
+++ src/kdeinit/CMakeLists.txt
@@ -21,12 +21,15 @@
 endif ()
 
 target_link_libraries(kdeinit5 ${kdeinit_LIBS} ${KINIT_SOCKET_LIBRARY}
+PUBLIC
     Qt5::Gui #QFont::initialize
     KF5::WindowSystem # KStartupInfo
     KF5::Crash # KCrash::loadedByKdeinit
     KF5::I18n #i18n() used on some error messages
     KF5::ConfigCore
     Qt5::DBus
+PRIVATE
+    Qt5::QspXDG
 )
 
 target_compile_definitions(kdeinit5 PRIVATE
@@ -36,7 +39,7 @@
 )
 
 if (X11_FOUND)
-  target_link_libraries(kdeinit5 ${X11_X11_LIB})
+  target_link_libraries(kdeinit5 PUBLIC ${X11_X11_LIB})
 endif()
 
 install(TARGETS kdeinit5 ${KF5_INSTALL_TARGETS_DEFAULT_ARGS} )
@@ -46,7 +46,7 @@ if (NOT WIN32)
   add_executable(kdeinit5_shutdown ../wrapper.cpp)
   ecm_mark_nongui_executable(kdeinit5_shutdown)
 
-  target_link_libraries(kdeinit5_shutdown  ${KINIT_SOCKET_LIBRARY} Qt5::Core)
+  target_link_libraries(kdeinit5_shutdown  PUBLIC ${KINIT_SOCKET_LIBRARY} Qt5::Core PRIVATE Qt5::QspXDG)
 
   install(TARGETS kdeinit5_shutdown ${KF5_INSTALL_TARGETS_DEFAULT_ARGS} )
 
@@ -54,7 +54,7 @@ if (NOT WIN32)
   add_executable(kdeinit5_wrapper ../wrapper.cpp)
   ecm_mark_nongui_executable(kdeinit5_wrapper)
 
-  target_link_libraries(kdeinit5_wrapper  ${KINIT_SOCKET_LIBRARY} Qt5::Core)
+  target_link_libraries(kdeinit5_wrapper  PUBLIC ${KINIT_SOCKET_LIBRARY} Qt5::Core PRIVATE Qt5::QspXDG)
 
   if (NOT X11_FOUND)
     target_compile_definitions(kdeinit5_wrapper PRIVATE -DNO_DISPLAY)
diff --git src/kshell/CMakeLists.txt src/kshell/CMakeLists.txt
index 5c2e63d..e43e84a 100644
--- src/kshell/CMakeLists.txt
+++ src/kshell/CMakeLists.txt
@@ -1,6 +1,6 @@
 add_executable(kshell5 shell.cpp)
 ecm_mark_nongui_executable(kshell5)
 
-target_link_libraries(kshell5 ${KINIT_SOCKET_LIBRARY} Qt5::Core)
+target_link_libraries(kshell5 PUBLIC ${KINIT_SOCKET_LIBRARY} Qt5::Core PRIVATE Qt5::QspXDG)
 
 install(TARGETS kshell5 ${KF5_INSTALL_TARGETS_DEFAULT_ARGS} )
diff --git src/kwrapper/CMakeLists.txt src/kwrapper/CMakeLists.txt
index 79cb70e..fc20aa1 100644
--- src/kwrapper/CMakeLists.txt
+++ src/kwrapper/CMakeLists.txt
@@ -7,5 +7,5 @@ endif ()
 add_executable(kwrapper5 ${kwrapper_SRCS})
 ecm_mark_nongui_executable(kwrapper5)
 
-target_link_libraries(kwrapper5 ${KINIT_SOCKET_LIBRARY} Qt5::Core)
+target_link_libraries(kwrapper5 PUBLIC ${KINIT_SOCKET_LIBRARY} Qt5::Core PRIVATE Qt5::QspXDG)
 install(TARGETS kwrapper5 ${KF5_INSTALL_TARGETS_DEFAULT_ARGS} )
diff --git src/klauncher/CMakeLists.txt src/klauncher/CMakeLists.txt
index 79cb70e..fc20aa1 100644
--- src/klauncher/CMakeLists.txt
+++ src/klauncher/CMakeLists.txt
@@ -16,16 +16,19 @@
 kf5_add_kdeinit_executable(klauncher NOGUI ${klauncher_SRCS})
 set_target_properties(kdeinit_klauncher PROPERTIES DEFINE_SYMBOL MAKE_KLAUNCHER)
 target_link_libraries(kdeinit_klauncher
+PUBLIC
    KF5::KIOCore # KIO::ConnectionServer
    KF5::KIOWidgets # KRun
    KF5::Crash  # sig_handler
    KF5::WindowSystem # KStartupInfo + fixx11
    KF5::I18n   # i18n()
    Qt5::DBus
+PRIVATE
+   Qt5::QspXDG
 )
 
 if (HAVE_XCB)
-    target_link_libraries(kdeinit_klauncher XCB::XCB)
+    target_link_libraries(kdeinit_klauncher PUBLIC XCB::XCB)
 endif()
 
 install(TARGETS kdeinit_klauncher ${KF5_INSTALL_TARGETS_DEFAULT_ARGS})
