diff --git a/src/private/xdgbasedirs.cpp b/src/private/xdgbasedirs.cpp
index 9a933a67e06505a9bedb01851394bd918833d198..19b0f0cf0372fe32014f3dceb069aec25a6d43f9 100644
--- a/src/private/xdgbasedirs.cpp
+++ b/src/private/xdgbasedirs.cpp
@@ -226,7 +226,7 @@ QStringList XdgBaseDirs::systemPathList(const char *resource)
                 dataDirs = instance()->systemPathList("XDG_DATA_DIRS", defaultPathList.toLocal8Bit().constData());
             }
 #else
-            QStringList dataDirs = instance()->systemPathList("XDG_DATA_DIRS", "/usr/local/share:/usr/share");
+            QStringList dataDirs = instance()->systemPathList("XDG_DATA_DIRS", "@PREFIX@/share:/usr/share");
 #endif
 
 #ifdef Q_OS_WIN
@@ -326,7 +326,7 @@ QString XdgBaseDirs::findResourceFile(const char *resource, const QString &relPa
 QString XdgBaseDirs::findExecutableFile(const QString &relPath, const QStringList &searchPath)
 {
     if (instance()->mExecutableDirs.isEmpty()) {
-        QStringList executableDirs = instance()->systemPathList("PATH", "/usr/local/bin:/usr/bin");
+        QStringList executableDirs = instance()->systemPathList("PATH", "@PREFIX@/bin:/usr/bin");
 
         const QString prefixExecutableDir = QStringLiteral(AKONADIPREFIX "/bin");
         if (!executableDirs.contains(prefixExecutableDir)) {
@@ -399,7 +399,7 @@ QString XdgBaseDirs::findExecutableFile(const QString &relPath, const QStringLis
 QStringList XdgBaseDirs::findPluginDirs()
 {
     if (instance()->mPluginDirs.isEmpty()) {
-        QStringList pluginDirs = instance()->systemPathList("QT_PLUGIN_PATH", AKONADILIB ":" AKONADILIB "/qt5/plugins/:" AKONADILIB "/kf5/:" AKONADILIB "/kf5/plugins/:/usr/lib/qt5/plugins/");
+        QStringList pluginDirs = instance()->systemPathList("QT_PLUGIN_PATH", AKONADILIB ":" AKONADILIB "/qt5/plugins/:" AKONADILIB "/kf5/:" AKONADILIB "/kf5/plugins/:@PREFIX@/share/qt5/plugins/");
         if (QCoreApplication::instance() != nullptr) {
             Q_FOREACH (const QString &libraryPath, QCoreApplication::instance()->libraryPaths()) {
                 if (!pluginDirs.contains(libraryPath)) {
diff --git a/src/server/storage/dbconfigmysql.cpp b/src/server/storage/dbconfigmysql.cpp
index 8ba80da6b16a4140e6a81359f62d5bd55becb457..c4fb44c8253befcffb53bd9b344a3440d7d287b6 100644
--- a/src/server/storage/dbconfigmysql.cpp
+++ b/src/server/storage/dbconfigmysql.cpp
@@ -83,7 +83,10 @@ bool DbConfigMysql::init(QSettings &settings)
                                          << QStringLiteral("/usr/local/libexec")
                                          << QStringLiteral("/usr/libexec")
                                          << QStringLiteral("/opt/mysql/libexec")
-                                         << QStringLiteral("/opt/local/lib/mysql5/bin")
+                                         << QStringLiteral("@PREFIX@/lib/mariadb/bin")
+                                         << QStringLiteral("@PREFIX@/lib/mysql55/bin")
+                                         << QStringLiteral("@PREFIX@/lib/mysql56/bin")
+                                         << QStringLiteral("@PREFIX@/lib/mysql57/bin")
                                          << QStringLiteral("/opt/mysql/sbin");
     if (defaultServerPath.isEmpty()) {
         defaultServerPath = XdgBaseDirs::findExecutableFile(QStringLiteral("mysqld"), mysqldSearchPath);
diff --git a/src/server/storage/dbconfigpostgresql.cpp b/src/server/storage/dbconfigpostgresql.cpp
index 72801f0429fc6aae012850cc0b31c831fd2cc6f5..8bcc1897f0cd9321d1917c51844d06894e2ee42d 100644
--- a/src/server/storage/dbconfigpostgresql.cpp
+++ b/src/server/storage/dbconfigpostgresql.cpp
@@ -79,11 +79,11 @@ bool DbConfigPostgresql::init(QSettings &settings)
         }
 #endif
         postgresSearchPath << QStringLiteral("/usr/sbin")
-                           << QStringLiteral("/usr/local/sbin");
+                           << QStringLiteral("@PREFIX@/sbin");
         // Locale all versions in /usr/lib/postgresql (i.e. /usr/lib/postgresql/X.Y) in reversed
         // sorted order, so we search from the newest one to the oldest.
         QStringList postgresVersionedSearchPaths;
-        QDir versionedDir(QStringLiteral("/usr/lib/postgresql"));
+        QDir versionedDir(QStringLiteral("@PREFIX@/lib/postgresql"));
         if (versionedDir.exists()) {
             const auto versionedDirs = versionedDir.entryInfoList(QDir::Dirs | QDir::NoDotAndDotDot, QDir::Name | QDir::Reversed);
             for (const auto &path : versionedDirs) {
@@ -94,6 +94,13 @@ bool DbConfigPostgresql::init(QSettings &settings)
                     postgresVersionedSearchPaths.append(path.absoluteFilePath() + QStringLiteral("/bin"));
                 }
             }
+            QStringList mpVersions = {QStringLiteral("96"), QStringLiteral("95"), QStringLiteral("94"), QStringLiteral("93")};
+            for (const auto &path : mpVersions) {
+                QString mpDir = QStringLiteral("@PREFIX@/lib/postgresql") + path;
+                if (QDir(mpDir).exists()) {
+                    postgresVersionedSearchPaths.append(mpDir + QStringLiteral("/bin"));
+                }
+            }
         }
         postgresSearchPath.append(postgresVersionedSearchPaths);
 
