diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9cf7977440eadbfe50bd3aaed1ea25f1b83c9fae..5005cef47a48cac78d8053c80312ec2bf09992d6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -27,2 +27,3 @@ include(ECMCoverageOption)
 include(KDEPackageAppTemplates)
+include(ECMMarkNonGuiExecutable)
 
diff --git a/KF5AkonadiMacros.cmake b/KF5AkonadiMacros.cmake
index 4ba50a1e834240cf66c9d72294d1cfab8f24e2d4..5e311fbf536633aa7630cc15fea4b37148723874 100644
--- a/KF5AkonadiMacros.cmake
+++ b/KF5AkonadiMacros.cmake
@@ -24,2 +24,3 @@ function(add_akonadi_isolated_test)
         ecm_mark_as_test(${_name})
+        ecm_mark_nongui_executable(${_name})
         target_link_libraries(${_name}
diff --git a/autotests/akonadicontrol/CMakeLists.txt b/autotests/akonadicontrol/CMakeLists.txt
index b933a90cc4accdb9f1edad87eebec81995798eaa..e28c045bc3c7810f250d21ced1c314dac2947718 100644
--- a/autotests/akonadicontrol/CMakeLists.txt
+++ b/autotests/akonadicontrol/CMakeLists.txt
@@ -12,2 +12,3 @@ macro(add_unit_test _source)
     add_executable(${_name} ${_test})
+    ecm_mark_nongui_executable(${_name})
     add_test(AkonadiControl-${_name} ${_name})
diff --git a/autotests/libs/CMakeLists.txt b/autotests/libs/CMakeLists.txt
index 3caa354e7d0ff4dcd49054cb4493e5a409f1142f..4e22f3b4324f0a4cb7a6ad993793818c34cf89f7 100644
--- a/autotests/libs/CMakeLists.txt
+++ b/autotests/libs/CMakeLists.txt
@@ -30,2 +30,3 @@ macro(add_akonadi_test _source)
   add_executable( ${_name} ${_test} )
+  ecm_mark_nongui_executable(${_name})
   add_test(NAME ${_name} COMMAND ${_name} )
@@ -45,2 +46,3 @@ macro(add_akonadi_test_widgets _source)
   add_executable( ${_name} ${_test} )
+  ecm_mark_nongui_executable(${_name})
   add_test(NAME ${_name} COMMAND ${_name} )
@@ -88,2 +90,3 @@ add_executable(akonadi-firstrun
 )
+ecm_mark_nongui_executable(akonadi-firstrun)
 target_link_libraries( akonadi-firstrun  Qt5::Test Qt5::Core KF5::AkonadiCore KF5::DBusAddons KF5::ConfigCore Qt5::Widgets)
diff --git a/autotests/libs/testresource/CMakeLists.txt b/autotests/libs/testresource/CMakeLists.txt
index 6ac1f5f4c054c5102465514e1a46c14beb02aa35..f8fe61faca1fce6a6281dddc72c310ecdb03ace1 100644
--- a/autotests/libs/testresource/CMakeLists.txt
+++ b/autotests/libs/testresource/CMakeLists.txt
@@ -44,2 +44,3 @@ qt5_add_dbus_adaptor(knutresource_SRCS
 add_executable(akonadi_knut_resource ${knutresource_SRCS})
+ecm_mark_nongui_executable(akonadi_knut_resource)
 
diff --git a/autotests/libs/testrunner/CMakeLists.txt b/autotests/libs/testrunner/CMakeLists.txt
index f8589fc9f751c32c0e0854075c8190577830e552..4e5bbb5431017dbe278d6978a70b0efd47c5240d 100644
--- a/autotests/libs/testrunner/CMakeLists.txt
+++ b/autotests/libs/testrunner/CMakeLists.txt
@@ -11,2 +11,3 @@ set(akonaditest_SRCS
 add_executable(akonaditest ${akonaditest_SRCS})
+ecm_mark_nongui_executable(akonaditest)
 
diff --git a/autotests/private/CMakeLists.txt b/autotests/private/CMakeLists.txt
index 946d86296aa756a641f17569038bcae4d3a3949b..93e28e5bfa8f21908fee03c760f2d9f08c9a0258 100644
--- a/autotests/private/CMakeLists.txt
+++ b/autotests/private/CMakeLists.txt
@@ -9,2 +9,3 @@ macro(add_unit_test _source)
     add_executable(${_name} ${_source})
+    ecm_mark_nongui_executable(${_name})
     add_test(NAME AkonadiPrivate-${_name} COMMAND ${_name})
diff --git a/autotests/server/CMakeLists.txt b/autotests/server/CMakeLists.txt
index 6ac1a4d12b30adad58c1443a9b622a53d0917d39..1fd9d44e16824c61f901b197b789a6cfbf828c86 100644
--- a/autotests/server/CMakeLists.txt
+++ b/autotests/server/CMakeLists.txt
@@ -53,2 +53,3 @@ macro(add_server_test _source)
     add_executable(${_name} ${_test})
+    ecm_mark_nongui_executable(${_name})
     add_test(NAME AkonadiServer-${_name} COMMAND ${_name})
diff --git a/src/agentserver/CMakeLists.txt b/src/agentserver/CMakeLists.txt
index 6ad6564df6395f88ee3ca1cd45c0494ccd9b66d0..0da77cb4b0d4bb81fc4391310053875a7c03640b 100644
--- a/src/agentserver/CMakeLists.txt
+++ b/src/agentserver/CMakeLists.txt
@@ -11,2 +11,3 @@ ecm_qt_declare_logging_category(akonadi_agent_server_srcs HEADER akonadiagentser
 add_executable(akonadi_agent_server ${akonadi_agent_server_srcs})
+ecm_mark_nongui_executable(akonadi_agent_server)
 
@@ -28,2 +29,3 @@ set(akonadi_agent_launcher_SRCS
 add_executable(akonadi_agent_launcher MACOSX_BUNDLE ${akonadi_agent_launcher_SRCS})
+ecm_mark_nongui_executable(akonadi_agent_launcher)
 
@@ -52,2 +54,2 @@ endif()
 install(TARGETS akonadi_agent_server
-        ${KF5_INSTALL_TARGETS_DEFAULT_ARGS}})
+        ${KF5_INSTALL_TARGETS_DEFAULT_ARGS})
diff --git a/src/akonadicontrol/CMakeLists.txt b/src/akonadicontrol/CMakeLists.txt
index a62ea936e5c52a4c5d10520a4e0f9143352947da..1c1ea8fa162cb8bce66bbd384f8f00cd6e9b3b4c 100644
--- a/src/akonadicontrol/CMakeLists.txt
+++ b/src/akonadicontrol/CMakeLists.txt
@@ -33,2 +33,3 @@ qt5_add_dbus_interface(control_SRCS ${Akonadi_SOURCE_DIR}/src/interfaces/org.fre
 add_executable(akonadi_control ${control_SRCS})
+ecm_mark_nongui_executable(akonadi_control)
 set_target_properties(akonadi_control PROPERTIES OUTPUT_NAME akonadi_control)
diff --git a/src/akonadictl/CMakeLists.txt b/src/akonadictl/CMakeLists.txt
index c163e4e78aafe145966994050804fff7806ae932..e02d2c3117879171588e4db91b5b3bd478c60a0e 100644
--- a/src/akonadictl/CMakeLists.txt
+++ b/src/akonadictl/CMakeLists.txt
@@ -17,2 +17,3 @@ qt5_add_dbus_interfaces(akonadictl_SRCS
 add_executable(akonadictl ${akonadictl_SRCS})
+ecm_mark_nongui_executable(akonadictl)
 set_target_properties(akonadictl PROPERTIES OUTPUT_NAME akonadictl)
@@ -27,3 +28,3 @@ target_link_libraries(akonadictl
 install(TARGETS akonadictl
-        ${KF5_INSTALL_TARGETS_DEFAULT_ARGS}}
+        ${KF5_INSTALL_TARGETS_DEFAULT_ARGS}
 )
diff --git a/src/asapcat/CMakeLists.txt b/src/asapcat/CMakeLists.txt
index 5fbd00bfd6956b2f3233b54f1ad647232dc378a0..8dfa706d6db80401dbb8e9e7d5a81417870ada3e 100644
--- a/src/asapcat/CMakeLists.txt
+++ b/src/asapcat/CMakeLists.txt
@@ -6,2 +6,3 @@ set(asapcat_srcs
 add_executable(asapcat ${asapcat_srcs})
+ecm_mark_nongui_executable(asapcat)
 
diff --git a/src/private/protocolgen/CMakeLists.txt b/src/private/protocolgen/CMakeLists.txt
index 5c894d6cb0d39b97143efd322c95530bf3aa3031..6d584f72e48ffbfed57bce19f1e2f7a788bb9bf6 100644
--- a/src/private/protocolgen/CMakeLists.txt
+++ b/src/private/protocolgen/CMakeLists.txt
@@ -12,2 +12,3 @@ set(protocolgen_SRCS
 add_executable(protocolgen ${protocolgen_SRCS})
+ecm_mark_nongui_executable(protocolgen)
 target_link_libraries(protocolgen
diff --git a/src/rds/CMakeLists.txt b/src/rds/CMakeLists.txt
index 6fc1bdd24aca75baa186757223c72a07a47a8640..a0e40738b0a201f68b8a7de4a4e6602e9dfbf42e 100644
--- a/src/rds/CMakeLists.txt
+++ b/src/rds/CMakeLists.txt
@@ -9,2 +9,3 @@ set(akonadi_rds_srcs
 add_executable(akonadi_rds ${akonadi_rds_srcs})
+ecm_mark_nongui_executable(akonadi_rds)
 
diff --git a/src/selftest/CMakeLists.txt b/src/selftest/CMakeLists.txt
index c13ef91c7506910f034a7d9906296ecc593d8322..3f64674076ecc713751c0a5226648572141c4818 100644
--- a/src/selftest/CMakeLists.txt
+++ b/src/selftest/CMakeLists.txt
@@ -5,2 +5,3 @@ set(selftest_SRCS
 add_executable(akonadiselftest ${selftest_SRCS})
+ecm_mark_nongui_executable(akonadiselftest)
 
diff --git a/src/server/CMakeLists.txt b/src/server/CMakeLists.txt
index 2ba7c11dd1c2dd8c983892d2083dd54c7b3503cc..0a6986d97fc1505debb0029ee4e327fceb41d235 100644
--- a/src/server/CMakeLists.txt
+++ b/src/server/CMakeLists.txt
@@ -170,2 +170,3 @@ target_link_libraries(libakonadiserver
 add_executable(akonadiserver ${akonadiserver_SRCS})
+ecm_mark_nongui_executable(akonadiserver)
 set_target_properties(akonadiserver PROPERTIES OUTPUT_NAME akonadiserver)
diff --git a/src/xml/CMakeLists.txt b/src/xml/CMakeLists.txt
index 71e6ef49c3da66fd0ea089a742eb79900e655cd5..b51db790b873fa6c7b1f7e6aa987a8d02a9712a5 100644
--- a/src/xml/CMakeLists.txt
+++ b/src/xml/CMakeLists.txt
@@ -34,2 +34,3 @@ ecm_generate_headers(AkonadiXml_HEADERS
 add_executable(akonadi2xml akonadi2xml.cpp)
+ecm_mark_nongui_executable(akonadi2xml)
 
diff --git a/src/xml/autotests/CMakeLists.txt b/src/xml/autotests/CMakeLists.txt
index d11c03bd5712042bf99aa1be4c5f0312aace6fff..198a1971ccdde93530f407ebd026a8e1c32c0d01 100644
--- a/src/xml/autotests/CMakeLists.txt
+++ b/src/xml/autotests/CMakeLists.txt
@@ -9,2 +9,3 @@ macro(add_libakonadixml_test _source)
   add_executable( ${_name} ${_test} )
+  ecm_mark_nongui_executable(${_name})
   add_test(NAME ${_name} COMMAND ${_name} )
# diff --git a/templates/akonadiresource/src/CMakeLists.txt b/templates/akonadiresource/src/CMakeLists.txt
# index 4e582b287660b583c4de761bf19d6b59adbd2976..cc5c30eb13fe9ef47f06875b367ce08f5c7836a2 100644
# --- a/templates/akonadiresource/src/CMakeLists.txt
# +++ b/templates/akonadiresource/src/CMakeLists.txt
# @@ -26,2 +26,3 @@ qt5_add_dbus_adaptor(%{APPNAMELC}resource_SRCS
#  add_executable(akonadi_%{APPNAMELC}_resource ${%{APPNAMELC}resource_SRCS})
# +ecm_mark_nongui_executable(akonadi_%{APPNAMELC}_resource)
#  
diff --git a/tests/libs/CMakeLists.txt b/tests/libs/CMakeLists.txt
index f35f80d33a1c862536c4d37355b815b95c75783e..e4f7b73053e966cafcfec5555a9548e2586e652d 100644
--- a/tests/libs/CMakeLists.txt
+++ b/tests/libs/CMakeLists.txt
@@ -28,2 +28,3 @@ macro(add_akonadi_demo _source)
   add_executable(${_name} ${_test})
+  ecm_mark_nongui_executable(${_name})
   target_link_libraries(${_name}
diff --git a/tests/libs/etm_test_app/CMakeLists.txt b/tests/libs/etm_test_app/CMakeLists.txt
index dc16b580130bc449c2934c4ce56c802745b34024..eff9ddf1f06a5e23969e7b03dfb01a536cceb382 100644
--- a/tests/libs/etm_test_app/CMakeLists.txt
+++ b/tests/libs/etm_test_app/CMakeLists.txt
@@ -15,2 +15,3 @@ set(etmtestapp_SRCS
 add_executable(akonadi_etm_test_app ${etmtestapp_SRCS})
+ecm_mark_nongui_executable(akonadi_etm_test_app)
 
