diff --git a/CMakeLists.txt b/CMakeLists.txt
index 02e5e5d6e69a9a0d8928fc9fac3b26d1314b423a..71a33b30b8bea61f759f95085cc19549ab0f92f5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -30,7 +30,8 @@ set(KMIMELIB_VERSION "5.6.3")
 set(AKONADI_VERSION "5.6.3")
 
 set(QT_REQUIRED_VERSION "5.7.0")
-find_package(Qt5 ${QT_REQUIRED_VERSION} CONFIG REQUIRED Widgets Test WebEngine WebEngineWidgets)
+# find_package(Qt5 ${QT_REQUIRED_VERSION} CONFIG REQUIRED Widgets Test WebEngine WebEngineWidgets)
+find_package(Qt5 ${QT_REQUIRED_VERSION} CONFIG REQUIRED Widgets Test)
 ########### Find packages ###########
 find_package(KF5KIO ${KF5_VERSION} CONFIG REQUIRED)
 find_package(KF5I18n ${KF5_VERSION} CONFIG REQUIRED)
diff --git a/src/contact-editor/CMakeLists.txt b/src/contact-editor/CMakeLists.txt
index 7ab0b0951d6b47c5a7540e205b1152bfa38bc12c..42242ba871d9431d84e3df34f5a4387c0c1bab3c 100644
--- a/src/contact-editor/CMakeLists.txt
+++ b/src/contact-editor/CMakeLists.txt
@@ -89,11 +89,11 @@ set(contacteditor_editor_generalinfo_SRCS
     ${contacteditor_editor_generalinfo_mail_SRCS}
     )
 
-set(contacteditor_addresslocation_web_editor_SRCS
-    editor/addresseditor/webengine/addresseslocationengineviewer.cpp
-    editor/addresseditor/webengine/addresseslocationenginepage.cpp
-    )
-
+# set(contacteditor_addresslocation_web_editor_SRCS
+#     editor/addresseditor/webengine/addresseslocationengineviewer.cpp
+#     editor/addresseditor/webengine/addresseslocationenginepage.cpp
+#     )
+# 
 set(contacteditor_addresslocation_editor_SRCS
     ${contacteditor_addresslocation_web_editor_SRCS}
     editor/addresseditor/addresslocationwidget.cpp
@@ -176,8 +176,8 @@ target_link_libraries(KF5ContactEditor
     KF5::TextWidgets
     KF5::WidgetsAddons
     Grantlee5::Templates
-    Qt5::WebEngine
-    Qt5::WebEngineWidgets
+#     Qt5::WebEngine
+#     Qt5::WebEngineWidgets
     )
 
 set_target_properties(KF5ContactEditor PROPERTIES
@@ -231,8 +231,9 @@ install( FILES
 
 
 if (BUILD_TESTING)
-    add_subdirectory(editor/addresseditor/autotests/)
-    add_subdirectory(editor/addresseditor/tests)
+    include(ECMMarkAsTest)
+#     add_subdirectory(editor/addresseditor/autotests/)
+#     add_subdirectory(editor/addresseditor/tests)
     add_subdirectory(editor/generalinfoeditor/autotests/)
     add_subdirectory(editor/generalinfoeditor/tests)
     add_subdirectory(editor/customfieldeditor/autotests/)
diff --git a/src/contact-editor/editor/addresseditor/addresseslocationwidget.cpp b/src/contact-editor/editor/addresseditor/addresseslocationwidget.cpp
index d9a911d8db7aa3e4aa1989ae63b6de9a4be9531a..70c1dc6d5736d42bcefc51aef37165e00b84aa51 100644
--- a/src/contact-editor/editor/addresseditor/addresseslocationwidget.cpp
+++ b/src/contact-editor/editor/addresseditor/addresseslocationwidget.cpp
@@ -21,7 +21,7 @@
 */
 
 #include "addresseslocationwidget.h"
-#include "webengine/addresseslocationengineviewer.h"
+// #include "webengine/addresseslocationengineviewer.h"
 #include "addresslocationwidget.h"
 using namespace ContactEditor;
 
@@ -32,13 +32,13 @@ AddressesLocationWidget::AddressesLocationWidget(QWidget *parent)
     mAddressLocationWidget = new AddressLocationWidget(this);
     mAddressLocationWidget->setObjectName(QStringLiteral("addresslocationwidget"));
     addWidget(mAddressLocationWidget);
-    mAddressesLocationViewer = new AddressesLocationEngineViewer(this);
-    mAddressesLocationViewer->setObjectName(QStringLiteral("addresseslocationviewer"));
-    addWidget(mAddressesLocationViewer);
-    connect(mAddressesLocationViewer, &AddressesLocationEngineViewer::modifyAddress, mAddressLocationWidget, &AddressLocationWidget::slotModifyAddress);
-    connect(mAddressLocationWidget, &AddressLocationWidget::addNewAddress, mAddressesLocationViewer, &AddressesLocationEngineViewer::addAddress);
-    connect(mAddressLocationWidget, &AddressLocationWidget::updateAddressCanceled, mAddressesLocationViewer, &AddressesLocationEngineViewer::updateAddressCanceled);
-    connect(mAddressLocationWidget, &AddressLocationWidget::updateAddress, mAddressesLocationViewer, &AddressesLocationEngineViewer::replaceAddress);
+//     mAddressesLocationViewer = new AddressesLocationEngineViewer(this);
+//     mAddressesLocationViewer->setObjectName(QStringLiteral("addresseslocationviewer"));
+//     addWidget(mAddressesLocationViewer);
+//     connect(mAddressesLocationViewer, &AddressesLocationEngineViewer::modifyAddress, mAddressLocationWidget, &AddressLocationWidget::slotModifyAddress);
+//     connect(mAddressLocationWidget, &AddressLocationWidget::addNewAddress, mAddressesLocationViewer, &AddressesLocationEngineViewer::addAddress);
+//     connect(mAddressLocationWidget, &AddressLocationWidget::updateAddressCanceled, mAddressesLocationViewer, &AddressesLocationEngineViewer::updateAddressCanceled);
+//     connect(mAddressLocationWidget, &AddressLocationWidget::updateAddress, mAddressesLocationViewer, &AddressesLocationEngineViewer::replaceAddress);
     setSizes(QList<int>() << 400 << 300);
 }
 
@@ -48,7 +48,7 @@ AddressesLocationWidget::~AddressesLocationWidget()
 
 void AddressesLocationWidget::loadContact(const KContacts::Addressee &contact)
 {
-    mAddressesLocationViewer->setAddresses(contact.addresses());
+//     mAddressesLocationViewer->setAddresses(contact.addresses());
 }
 
 void AddressesLocationWidget::storeContact(KContacts::Addressee &contact) const
@@ -60,7 +60,7 @@ void AddressesLocationWidget::storeContact(KContacts::Addressee &contact) const
         contact.removeAddress(oldAddresses.at(i));
     }
 
-    const KContacts::Address::List addressList = mAddressesLocationViewer->addresses();
+    const KContacts::Address::List addressList /*= mAddressesLocationViewer->addresses()*/;
     // insert the new ones
     const int numberOfAddress(addressList.count());
     for (int i = 0; i < numberOfAddress; ++i) {
@@ -73,6 +73,6 @@ void AddressesLocationWidget::storeContact(KContacts::Addressee &contact) const
 
 void AddressesLocationWidget::setReadOnly(bool readOnly)
 {
-    mAddressesLocationViewer->setReadOnly(readOnly);
+//     mAddressesLocationViewer->setReadOnly(readOnly);
     mAddressLocationWidget->setReadOnly(readOnly);
 }
