diff --git outputview/outputfilteringstrategies.cpp outputview/outputfilteringstrategies.cpp
index f82420f..afa275c 100644
--- outputview/outputfilteringstrategies.cpp
+++ outputview/outputfilteringstrategies.cpp
@@ -158,7 +158,7 @@ const QVector<ErrorFormat> ERROR_FILTERS {
 // A list of filters for possible compiler, linker, and make actions
 const QVector<ActionFormat> ACTION_FILTERS {
     ActionFormat( I18N_NOOP2_NOSTRIP("", "compiling"), 1, 2, "(?:^|[^=])\\b(gcc|CC|cc|distcc|c\\+\\+|"
-                     "g\\+\\+|clang|clang\\+\\+|mpicc|icc|icpc)\\s+.*-c.*[/ '\\\\]+(\\w+\\.(?:cpp|CPP|c|C|cxx|CXX|cs|"
+                     "g\\+\\+|clang|clang\\+\\+|mpicc|icc|icpc)\\s+.*-c.*[/ '\\\\]+(\\w+\\.(?:cpp|CPP|c|C|cxx|CXX|cs|m|M|mm|MM"
                      "java|hpf|f|F|f90|F90|f95|F95))"),
     //moc and uic
     ActionFormat( I18N_NOOP2_NOSTRIP("", "generating"), 1, 2, "/(moc|uic)\\b.*\\s-o\\s([^\\s;]+)"),
