diff --git util/environmentselectionwidget.cpp util/environmentselectionwidget.cpp
index 1e2cdf8..c2537a0 100644
--- util/environmentselectionwidget.cpp
+++ util/environmentselectionwidget.cpp
@@ -100,6 +100,11 @@ EnvironmentGroupList EnvironmentSelectionWidget::environment() const
     return d->model->environment();
 }
 
+KComboBox *EnvironmentSelectionWidget::comboBox() const
+{
+    return d->comboBox;
+}
+
 }
 
 #include "environmentselectionwidget.moc"
diff --git util/environmentselectionwidget.h util/environmentselectionwidget.h
index bd05642..169644d 100644
--- util/environmentselectionwidget.h
+++ util/environmentselectionwidget.h
@@ -70,6 +70,8 @@ public:
      */
     EnvironmentGroupList environment() const;
 
+   KComboBox *comboBox() const;
+
 public slots:
     /**
      * Makes the widget re-read its environment group list.
