--- MacBuntu-OS/orig.index.theme	2016-10-28 21:23:06.000000000 +0200
+++ MacBuntu-OS/index.theme	2017-02-13 13:15:27.000000000 +0100
@@ -1,8 +1,8 @@
 [Icon Theme]
-Name=MacBuntu-OS
+Name=osx
 Comment=MacBuntu-OS icons. (Oct 2016) For personal use/fun only! Do not use for bad or commercial purposes. (DO NOT MAKE Knock off devices or systems.)
 
-Inherits=Adwaita,hicolor
+Inherits=gnome,hicolor
 
 Example=directory-x-normal
 
@@ -228,7 +228,7 @@
 Context=Places
 MinSize=8
 MaxSize=512
-Type=fixed
+Type=Scalable
 
 [status/16]
 Size=16
