diff --git ksimond/src/CMakeLists.txt ksimond/src/CMakeLists.txt
index 5ea2968cf77d929942301e550a33a7516339e049..de1c105977b320e563af00434c5f17da2b1e4229 100644
--- ksimond/src/CMakeLists.txt
+++ ksimond/src/CMakeLists.txt
@@ -6,7 +6,7 @@ set(ksimond_SRCS
 kde4_add_kcfg_files(ksimond_SRCS ../config/ksimondconfiguration.kcfgc)
 
 kde4_add_app_icon(ksimond_SRCS "../icons/hi*-app-ksimond.png")
-kde4_add_executable(ksimond ${ksimond_SRCS})
+kde4_add_executable(ksimond NOGUI ${ksimond_SRCS})
 
 target_link_libraries(ksimond ${KDE4_KDEUI_LIBS} ${KDE4_KUTILS_LIBS}
   simonuicomponents simoninfo)
diff --git simond/src/CMakeLists.txt simond/src/CMakeLists.txt
index c1ef11c61c671cc72046cc79e19baa78ad5825c0..edf35c4667add27345d1cd40e35ef21eb95ac746 100644
--- simond/src/CMakeLists.txt
+++ simond/src/CMakeLists.txt
@@ -15,7 +15,7 @@ if(${BackendType} STREQUAL both)
 endif()
 
 kde4_add_app_icon(simond_SRCS "../icons/hi*-app-simond.png")
-kde4_add_executable(simond ${simond_SRCS})
+kde4_add_executable(simond NOGUI ${simond_SRCS})
 
 target_link_libraries(simond ${KDE4_KDECORE_LIBS} ${QT_QTNETWORK_LIBRARY} simonrecognizer
   ${QT_QTCORE_LIBRARY} ${QT_QTXML_LIBRARY} simonscenarios simonmodelcompilation simonddatabaseaccess
diff --git sscd/src/CMakeLists.txt sscd/src/CMakeLists.txt
index 561f1674eec92f59096bf5ac175f2c0dbe45162b..9dd420e912ba6eedb31b8a0d380271283c4620f8 100644
--- sscd/src/CMakeLists.txt
+++ sscd/src/CMakeLists.txt
@@ -9,7 +9,7 @@ set(sscd_SRCS
 
 kde4_add_app_icon(sscd_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/../icons/hi*-app-sscd.png")
 
-kde4_add_executable(sscd ${sscd_SRCS})
+kde4_add_executable(sscd NOGUI ${sscd_SRCS})
 
 target_link_libraries(sscd ${QT_QTCORE_LIBRARY} ${QT_QTSQL_LIBRARY}
   ${QT_QTNETWORK_LIBRARY} sscobjects)

