--- a/src/MixerBoard.cpp
+++ b/src/MixerBoard.cpp
@@ -763,7 +763,7 @@
 {
    wxPaintDC dc(this);
 
-   #ifdef __WXMAC__
+   #if defined(__WXMAC__) || defined(__APPLE__)
       // Fill with correct color, not scroller background. Done automatically on Windows.
       AColor::Medium(&dc, false);
       dc.DrawRectangle(this->GetClientRect());
--- a/src/widgets/Meter.cpp
+++ b/src/widgets/Meter.cpp
@@ -354,7 +354,7 @@
 void Meter::OnPaint(wxPaintEvent & WXUNUSED(event))
 {
    wxPaintDC dc(this);
-  #ifdef __WXMAC__
+  #if defined(__WXMAC__) || defined(__APPLE__)
    // Mac OS X automatically double-buffers the screen for you,
    // so our bitmap is unneccessary
    HandlePaint(dc);
@@ -1028,7 +1028,7 @@
    wxClientDC dc(this);
    int i;
 
-  #ifdef __WXMAC__
+  #if defined(__WXMAC__) || defined(__APPLE__)
    // Mac OS X automatically double-buffers the screen for you,
    // so our bitmap is unneccessary
    for(i=0; i<mNumBars; i++)
