--- a/lib-src/FileDialog/configure.ac
+++ b/lib-src/FileDialog/configure.ac
@@ -42,7 +42,7 @@
             debug_preference="")
 
 AC_ARG_WITH(wx-version,
-            [AS_HELP_STRING([--with-wx-version],[override default wxWidgets version [2.6,2.8]])],
+            [AS_HELP_STRING([--with-wx-version],[override default wxWidgets version [2.8, 3.0]])],
             wx_preference="--version=$withval",
             wx_preference="")
 
@@ -67,27 +67,48 @@
 
 PKG_CHECK_MODULES(GTK, gtk+-2.0, have_gtk="yes", have_gtk="no")
 
-dnl OS-specific configuration
+dnl Get wx version
+wx_version=`${WX_CONFIG} $wxconfigargs --version`
+
+dnl OS- and wx-version-specific configuration
 
 AC_CANONICAL_HOST
 
 IMPLEMENTATION="generic"
-case "${host_os}" in
-   darwin*)
-      dnl Mac OS X configuration
-      IMPLEMENTATION="mac"
+case "${wx_version}" in
+   2.8.*)
+      case "${host_os}" in
+#         darwin*)
+#            dnl Mac OS X configuration
+#            IMPLEMENTATION="mac"
+#         ;;
+
+         cygwin*)
+            dnl Windows/CygWin configuration
+            IMPLEMENTATION="win"
+         ;;
+
+         *)
+            if test "$have_gtk" = "yes"; then
+               AC_DEFINE(HAVE_GTK, 1, [Define to 1 if GTK+ is available.])
+               IMPLEMENTATION="gtk"
+            fi
+         ;;
+      esac
    ;;
 
-   cygwin*)
-      dnl Windows/CygWin configuration
-      IMPLEMENTATION="win"
+   3.0.*)
+      dnl wx3.0 version for all platforms
+      IMPLEMENTATION="wx30"
    ;;
 
    *)
-      if test "$have_gtk" = "yes"; then
-         AC_DEFINE(HAVE_GTK, 1, [Define to 1 if GTK+ is available.])
-         IMPLEMENTATION="gtk"
-      fi
+      wx_list=`${WX_CONFIG} --list`
+      AC_MSG_ERROR([Unable to locate a suitable configuration of wxWidgets v2.8.x
+      or v3.0.x. The currently available configurations are listed below. If
+      necessary, either install the package for your distribution or download a
+      suitable version of wxWidgets from http://wxwidgets.org.
+      ${wx_list}])
    ;;
 esac
 
@@ -96,6 +117,7 @@
 AM_CONDITIONAL([GTK], test "$IMPLEMENTATION" = "gtk")
 AM_CONDITIONAL([MAC], test "$IMPLEMENTATION" = "mac")
 AM_CONDITIONAL([WINDOWS], test "$IMPLEMENTATION" = "win")
+AM_CONDITIONAL([WX30], test "$IMPLEMENTATION" = "wx30")
 
 echo "Implementation to use: $IMPLEMENTATION"
 
