diff --git a/src/toolbars/ToolManager.cpp b/src/toolbars/ToolManager.cpp
index 011193e..b125ad1 100644
--- a/src/toolbars/ToolManager.cpp
+++ b/src/toolbars/ToolManager.cpp
@@ -64,6 +64,8 @@
 #include "../AColor.h"
 #include "../AllThemeResources.h"
 #include "../ImageManipulation.h"
+// hack to deactivate a bit in wx/confbase.h on which clang chokes:
+#define __VMS
 #include "../Prefs.h"
 #include "../Project.h"
 #include "../Theme.h"
