diff --git a/src/effects/VST/VSTEffect.cpp b/src/effects/VST/VSTEffect.cpp
index 9b4c6aa..8077131 100644
--- a/src/effects/VST/VSTEffect.cpp
+++ b/src/effects/VST/VSTEffect.cpp
@@ -43,7 +43,7 @@
 
 #if defined(__WXMAC__)
 #include <dlfcn.h>
-#include <wx/mac/private.h>
+#include <wx/osx/private.h>
 #elif defined(__WXMSW__)
 #include <wx/dynlib.h>
 #else
diff --git a/src/toolbars/ToolManager.cpp b/src/toolbars/ToolManager.cpp
index b125ad1..d6bbfc0 100644
--- a/src/toolbars/ToolManager.cpp
+++ b/src/toolbars/ToolManager.cpp
@@ -48,7 +48,7 @@
 #include <wx/popupwin.h>
 
 #if defined(__WXMAC__)
-#include <wx/mac/uma.h>
+#include <wx/osx/uma.h>
 #endif
 
 #include "ToolManager.h"
diff --git a/src/widgets/Meter.cpp b/src/widgets/Meter.cpp
index a2d0bb0..20ba1d9 100644
--- a/src/widgets/Meter.cpp
+++ b/src/widgets/Meter.cpp
@@ -53,7 +53,7 @@
 #include <wx/msgdlg.h>
 
 #if defined(__WXMAC__)
-#include <wx/mac/uma.h>
+#include <wx/osx/uma.h>
 #endif
 
 #include <math.h>
