diff --git a/lib-src/lib-widget-extra/NonGuiThread.cpp b/lib-src/lib-widget-extra/NonGuiThread.cpp
index 9b15de5..1b69eee 100644
--- a/lib-src/lib-widget-extra/NonGuiThread.cpp
+++ b/lib-src/lib-widget-extra/NonGuiThread.cpp
@@ -16,6 +16,9 @@ take place in the background without killing the GUI.
 *//**********************************************************************/
 
 #include <wx/wx.h>
+
+// horrible kludge to prevent evtloopsrc.h from attempting to include wx/osx/evtloopsrc.h ...
+#undef __DARWIN__
 #include <wx/apptrait.h>
 #include "NonGuiThread.h"
 
diff --git a/src/Project.cpp b/src/Project.cpp
index f5cc69a..a5fda02 100644
--- a/src/Project.cpp
+++ b/src/Project.cpp
@@ -54,6 +54,9 @@ scroll information.  It also has some status flags.
 #include <stdio.h>
 #include <iostream>
 #include <wx/wxprec.h>
+
+// horrible kludge to prevent evtloopsrc.h from attempting to include wx/osx/evtloopsrc.h ...
+#undef __DARWIN__
 #include <wx/apptrait.h>
 
 #include <wx/defs.h>
