diff --git a/src/MixerBoard.cpp b/src/MixerBoard.cpp
index 8d8b650..153c81a 100644
--- a/src/MixerBoard.cpp
+++ b/src/MixerBoard.cpp
@@ -331,8 +331,8 @@ MixerTrackCluster::MixerTrackCluster(wxWindow* parent,
    #endif // wxUSE_TOOLTIPS
 
    #ifdef __WXMAC__
-      wxSizeEvent dummyEvent;
-      this->OnSize(dummyEvent);
+//       wxSizeEvent dummyEvent;
+//       this->OnSize(dummyEvent);
       UpdateGain();
    #endif
 }
