--- boost/container/orig.slist.hpp	2015-07-21 23:40:52.000000000 +0200
+++ boost/container/slist.hpp	2017-05-15 18:03:27.000000000 +0200
@@ -1661,7 +1661,7 @@
 
 #ifndef BOOST_CONTAINER_DOXYGEN_INVOKED
 
-#if defined(__clang__) && defined(_LIBCPP_VERSION)
+#if defined(_LIBCPP_VERSION)
    #define BOOST_CONTAINER_CLANG_INLINE_STD_NS
    #pragma GCC diagnostic push
    #pragma GCC diagnostic ignored "-Wc++11-extensions"
--- boost/asio/detail/orig.config.hpp	2017-05-15 18:06:26.000000000 +0200
+++ boost/asio/detail/config.hpp	2017-05-15 18:06:45.000000000 +0200
@@ -71,7 +71,7 @@
 #endif // defined(BOOST_ASIO_MSVC)
 
 // Clang / libc++ detection.
-#if defined(__clang__)
+// #if defined(__clang__)
 # if (__cplusplus >= 201103)
 #  if __has_include(<__config>)
 #   include <__config>
@@ -80,7 +80,7 @@
 #   endif // defined(_LIBCPP_VERSION)
 #  endif // __has_include(<__config>)
 # endif // (__cplusplus >= 201103)
-#endif // defined(__clang__)
+// #endif // defined(__clang__)
 
 // Support move construction and assignment on compilers known to allow it.
 #if !defined(BOOST_ASIO_HAS_MOVE)
--- boost/move/detail/orig.iterator_traits.hpp	2017-05-15 17:54:42.000000000 +0200
+++ boost/move/detail/iterator_traits.hpp	2017-05-15 18:18:09.000000000 +0200
@@ -24,10 +24,12 @@
 
 #include <cstddef>
 
-#if defined(__clang__) && defined(_LIBCPP_VERSION)
+#if defined(_LIBCPP_VERSION)
    #define BOOST_MOVE_CLANG_INLINE_STD_NS
-   #pragma GCC diagnostic push
-   #pragma GCC diagnostic ignored "-Wc++11-extensions"
+   #if defined(__clang__)
+      #pragma GCC diagnostic push
+      #pragma GCC diagnostic ignored "-Wc++11-extensions"
+   #endif
    #define BOOST_MOVE_STD_NS_BEG _LIBCPP_BEGIN_NAMESPACE_STD
    #define BOOST_MOVE_STD_NS_END _LIBCPP_END_NAMESPACE_STD
 #else
--- libs/program_options/src/orig.parsers.cpp	2017-05-15 18:37:34.000000000 +0200
+++ libs/program_options/src/parsers.cpp	2017-05-15 18:38:07.000000000 +0200
@@ -6,6 +6,10 @@
 
 #include <boost/config.hpp>
 
+#if defined(_LIBCPP_VERSION) && !defined(__clang__)
+#include <set>
+#endif
+
 #define BOOST_PROGRAM_OPTIONS_SOURCE
 #include <boost/program_options/config.hpp>
 #include <boost/program_options/parsers.hpp>
