diff --git src/libaudcore/mainloop.cc src/libaudcore/mainloop.cc
index 397106f79f73b2bfc9967a891f5adfa674fd2204..906f0ed52a7c55c0d6057830e6a33f41525dddea 100644
--- src/libaudcore/mainloop.cc
+++ src/libaudcore/mainloop.cc
@@ -224,13 +224,27 @@ public:
     }
 
     void stop ()
-        { deleteLater (); }
+    {
+        if (timerId)
+        {
+            killTimer(timerId);
+            timerId = 0;
+        }
+        deleteLater ();
+    }
 
 protected:
     void customEvent (QEvent *)
-        { startTimer (params.interval_ms); }
-    void timerEvent (QTimerEvent *)
-        { run (); }
+        { timerId = startTimer (params.interval_ms); }
+    void timerEvent (QTimerEvent * event)
+    {
+        if (event->timerId() == timerId)
+        {
+            run ();
+        }
+    }
+private:
+    int timerId = 0;
 };
 
 #endif // USE_QT
