diff --git src/libaudqt/menu-qt.cc src/libaudqt/menu-qt.cc
index 8f405b54764f7d37cd089f4181df20d1625de06e..35eeec35c64bf1f1631ecac01b2a7dfb212c1a87 100644
--- src/libaudqt/menu-qt.cc
+++ src/libaudqt/menu-qt.cc
@@ -117,8 +117,12 @@ EXPORT QMenu * menu_build (ArrayRef<MenuItem> menu_items, const char * domain, Q
 
 EXPORT QMenuBar * menubar_build (ArrayRef<MenuItem> menu_items, const char * domain, QWidget * parent)
 {
+#ifdef Q_OS_MACOS
+    QMenuBar * m = new QMenuBar (nullptr);
+#else
     QMenuBar * m = new QMenuBar (parent);
     m->setContextMenuPolicy (Qt::PreventContextMenu);
+#endif
 
     for (auto & it : menu_items)
         m->addAction (new MenuAction (it, domain, parent));
