diff --git src/libaudqt/log-inspector.cc src/libaudqt/log-inspector.cc
index bcaefd5f2ce9b9e659fda92edd2a12ad9c85ba34..594945caf3aaa141c371744d6a2f3fd64c7bf309 100644
--- src/libaudqt/log-inspector.cc
+++ src/libaudqt/log-inspector.cc
@@ -29,6 +29,7 @@
 #include <QPushButton>
 #include <QTreeView>
 #include <QWidget>
+#include <QApplication>
 
 #include <libaudcore/audstrings.h>
 #include <libaudcore/hook.h>
@@ -203,6 +204,9 @@ LogEntryInspector::LogEntryInspector (QWidget * parent) :
     view->setUniformRowHeights (true);
     view->scrollToBottom ();
 
+    // set a platform-appropriate font
+    view->setFont(QApplication::font("QListView"));
+
     m_level_combobox.addItem (_("Debug"), audlog::Debug);
     m_level_combobox.addItem (_("Info"), audlog::Info);
     m_level_combobox.addItem (_("Warning"), audlog::Warning);
diff --git src/libaudqt/info-widget.cc src/libaudqt/info-widget.cc
index ef9b218a80f6513feecdf3f54774b70436b4109b..ec2f69bb27be5dd15bfedd6ab4d59d3a6b75b149 100644
--- src/libaudqt/info-widget.cc
+++ src/libaudqt/info-widget.cc
@@ -22,6 +22,7 @@
 #include "libaudqt.h"
 
 #include <QHeaderView>
+#include <QApplication>
 
 #include <libaudcore/i18n.h>
 #include <libaudcore/probe.h>
@@ -93,6 +94,7 @@ EXPORT InfoWidget::InfoWidget (QWidget * parent) :
     QTreeView (parent),
     m_model (new InfoModel (this))
 {
+    setFont(QApplication::font("QSmallFont"));
     setModel (m_model);
     header ()->hide ();
     setIndentation (0);
diff --git src/libaudqt/queue-manager-qt.cc src/libaudqt/queue-manager-qt.cc
index 11301b31aed77a8bbe6a41d7fa9d84452cce5510..72d0b10a43d8d6ce27ea0d189557dc1349a67c1b 100644
--- src/libaudqt/queue-manager-qt.cc
+++ src/libaudqt/queue-manager-qt.cc
@@ -27,6 +27,7 @@
 #include <QItemSelectionModel>
 #include <QTreeView>
 #include <QVBoxLayout>
+#include <QApplication>
 
 #include <libaudcore/audstrings.h>
 #include <libaudcore/playlist.h>
@@ -174,6 +175,7 @@ QueueManagerDialog::QueueManagerDialog (QWidget * parent) :
     m_treeview.setModel (& m_model);
     m_treeview.setSelectionMode (QAbstractItemView::ExtendedSelection);
     m_treeview.setHeaderHidden (true);
+    m_treeview.setFont(QApplication::font("QSmallFont"));
 
     update ();
 
