diff --git CMakeLists.txt CMakeLists.txt
index 1997ed3..c661688 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -3,6 +3,7 @@ project (ClangLazy)
 
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_SOURCE_DIR}/cmake)
 find_package(Clang 3.6 MODULE REQUIRED)
+# message(STATUS "Found Clang version ${LLVM_VERSION}")
 
 set( CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin )
 set( CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib )
@@ -17,7 +18,7 @@ elseif(MSVC)
     # disable trigger-happy warnings from Clang/LLVM headers
     set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /wd4267 /wd4244 /wd4291 /wd4800")
 endif()
-set (CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} -Wl,-flat_namespace -Wl,-undefined -Wl,suppress")
+# set (CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} -Wl,-flat_namespace -Wl,-undefined -Wl,suppress")
 if(NOT WIN32)
     set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC")
 endif()
@@ -39,6 +40,8 @@ LIST(REMOVE_ITEM LLVM_LIBS_LIST "-lLLVMSupport")    # Remove element
 macro(add_clang_plugin name)
     set (srcs ${ARGN})
 
+    add_definitions(${LLVM_CFLAGS})
+
     include_directories(${CLANG_INCLUDE_DIRS} ${CMAKE_SOURCE_DIR})
     link_directories("${LLVM_INSTALL_PREFIX}/lib" ${LLVM_LIBRARY_DIRS})
 
@@ -56,6 +59,11 @@ macro(add_clang_plugin name)
     foreach (llvm_lib ${LLVM_LIBS_LIST})
         target_link_libraries( ${name} ${llvm_lib} )
     endforeach()
+    if(${LLVM_VERSION} VERSION_GREATER "3.7.9")
+        target_link_libraries( ${name} "-lLLVM" )
+    else()
+        target_link_libraries( ${name} "-lLLVM-${LLVM_VERSION}" )
+    endif()
 
     foreach (user_lib ${USER_LIBS})
         target_link_libraries( ${name} ${user_lib} )
@@ -139,7 +147,7 @@ set_target_properties(ClangLazy PROPERTIES
 
 include("GNUInstallDirs")
 install(TARGETS ClangLazy     RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
-                              LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+                              LIBRARY DESTINATION @LLVM_INSTALL_PREFIX@/lib
                               ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 if(NOT WIN32)
     install(FILES ${CMAKE_SOURCE_DIR}/clazy DESTINATION bin PERMISSIONS OWNER_WRITE OWNER_EXECUTE OWNER_READ GROUP_EXECUTE GROUP_READ WORLD_READ WORLD_EXECUTE)
