diff --git a/CMakeLists.txt b/CMakeLists.txt
index 43410cc..2ca8d65 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -125,9 +131,10 @@ link_directories("${LLVM_INSTALL_PREFIX}/lib" ${LLVM_LIBRARY_DIRS})
 add_library(clazylib SHARED ${CLAZY_LIB_SRC})
 link_to_llvm(clazylib)
 
-if(APPLE)
-  target_link_libraries(clazylib LLVM)
-endif()
+# Not linking to the shared libLLVM on Mac
+# if(APPLE)
+#   target_link_libraries(clazylib LLVM)
+# endif()
 
 generate_export_header(clazylib)
 
