diff --git autotests/kconfigtest.cpp autotests/kconfigtest.cpp
index 410b5b8..c5af197 100644
--- autotests/kconfigtest.cpp
+++ autotests/kconfigtest.cpp
@@ -1756,6 +1756,7 @@ void KConfigTest::testQByteArrayUtf8()
     QCOMPARE(bytes, general2.readEntry("Utf8", QByteArray()));
 }
 
+#if QT_VERSION >= QT_VERSION_CHECK(5, 11, 0)
 void KConfigTest::testQStringUtf8_data()
 {
     QTest::addColumn<QByteArray>("data");
@@ -1804,6 +1805,7 @@ void KConfigTest::testQStringUtf8()
     KConfigGroup general2(&config2, "General");
     QCOMPARE(value, general2.readEntry("key", QByteArray()));
 }
+#endif
 
 void KConfigTest::testNewlines()
 {
