diff --git autotests/kconfigtest.cpp autotests/kconfigtest.cpp
index 410b5b8..c5af197 100644
--- autotests/kconfigtest.cpp
+++ autotests/kconfigtest.cpp
@@ -1758,6 +1758,7 @@ void KConfigTest::testQByteArrayUtf8()
 
 void KConfigTest::testQStringUtf8_data()
 {
+#if QT_VERSION >= QT_VERSION_CHECK(5, 11, 0)
     QTest::addColumn<QByteArray>("data");
     QTest::newRow("1") << QByteArray("Téléchargements\tTéléchargements");
     QTest::newRow("2") << QByteArray("$¢ह€𐍈\t$¢ह€𐍈");
@@ -1775,10 +1776,12 @@ void KConfigTest::testQStringUtf8_data()
     QTest::newRow("9") << QByteArray("\xe0\xa0\x80\t\xe0\xa0\x80");
     QTest::newRow("10") << QByteArray("\xf0\x90\x80\x80\t\xf0\x90\x80\x80");
     QTest::newRow("11") << QByteArray("\xf4\x8f\xbf\xbf\t\xf4\x8f\xbf\xbf");
+#endif
 }
 
 void KConfigTest::testQStringUtf8()
 {
+#if QT_VERSION >= QT_VERSION_CHECK(5, 11, 0)
     QFETCH(QByteArray, data);
     const QList<QByteArray> d = data.split('\t');
     const QByteArray value = d[0];
@@ -1803,6 +1806,7 @@ void KConfigTest::testQStringUtf8()
     KConfig config2(file.fileName(), KConfig::SimpleConfig);
     KConfigGroup general2(&config2, "General");
     QCOMPARE(value, general2.readEntry("key", QByteArray()));
+#endif
 }
 
 void KConfigTest::testNewlines()
