diff --git a/qtbase/src/platformsupport/themes/genericunix/qgenericunixthemes_p.h b/qtbase/src/platformsupport/themes/genericunix/qgenericunixthemes_p.h
index 0bcd346..a417d19 100644
--- a/qtbase/src/platformsupport/themes/genericunix/qgenericunixthemes_p.h
+++ b/qtbase/src/platformsupport/themes/genericunix/qgenericunixthemes_p.h
@@ -79,7 +79,7 @@
     QVariant themeHint(ThemeHint hint) const Q_DECL_OVERRIDE;
 
     static QStringList xdgIconThemePaths();
-#if !defined(QT_NO_DBUS) && !defined(QT_NO_SYSTEMTRAYICON) && !defined(Q_OS_OSX)
+#if !defined(QT_NO_DBUS) && !defined(QT_NO_SYSTEMTRAYICON)
     QPlatformSystemTrayIcon *createPlatformSystemTrayIcon() const Q_DECL_OVERRIDE;
 #endif
 
@@ -101,7 +101,7 @@
     const QPalette *palette(Palette type = SystemPalette) const Q_DECL_OVERRIDE;
 
     const QFont *font(Font type) const Q_DECL_OVERRIDE;
-#if !defined(QT_NO_DBUS) && !defined(QT_NO_SYSTEMTRAYICON) && !defined(Q_OS_OSX)
+#if !defined(QT_NO_DBUS) && !defined(QT_NO_SYSTEMTRAYICON)
     QPlatformSystemTrayIcon *createPlatformSystemTrayIcon() const Q_DECL_OVERRIDE;
 #endif
 
@@ -121,7 +121,7 @@
     QString standardButtonText(int button) const Q_DECL_OVERRIDE;
 
     virtual QString gtkFontName() const;
-#if !defined(QT_NO_DBUS) && !defined(QT_NO_SYSTEMTRAYICON) && !defined(Q_OS_OSX)
+#if !defined(QT_NO_DBUS) && !defined(QT_NO_SYSTEMTRAYICON)
     QPlatformSystemTrayIcon *createPlatformSystemTrayIcon() const Q_DECL_OVERRIDE;
 #endif
 
diff --git a/qtbase/src/platformsupport/themes/genericunix/qgenericunixthemes.cpp b/qtbase/src/platformsupport/themes/genericunix/qgenericunixthemes.cpp
index 0bcd346..a417d19 100644
--- a/qtbase/src/platformsupport/themes/genericunix/qgenericunixthemes.cpp
+++ b/qtbase/src/platformsupport/themes/genericunix/qgenericunixthemes.cpp
@@ -56,12 +56,6 @@
 #include <qpa/qplatformservices.h>
 #include <qpa/qplatformdialoghelper.h>
 
-#ifdef Q_OS_OSX
-// platformsupport plugins shouldn't use DBus on OS X
-// and systray functionality doesn't depend on it
-#define QT_NO_DBUS
-#endif
-
 #if !defined(QT_NO_DBUS) && !defined(QT_NO_SYSTEMTRAYICON)
 #include "QtPlatformSupport/private/qdbustrayicon_p.h"
 #include "QtPlatformSupport/private/qdbusplatformmenu_p.h"
