diff --git a/qtbase/src/platformsupport/fontdatabases/mac/qcoretextfontdatabase.mm b/qtbase/src/platformsupport/fontdatabases/mac/qcoretextfontdatabase.mm
index 78a0c4d..13107b6 100644
--- a/qtbase/src/platformsupport/fontdatabases/mac/qcoretextfontdatabase.mm
+++ b/qtbase/src/platformsupport/fontdatabases/mac/qcoretextfontdatabase.mm
@@ -221,6 +221,7 @@ void QCoreTextFontDatabase::populateFontDatabase()
         QPlatformFontDatabase::registerFontFamily(familyName);
 
 #if defined(Q_OS_OSX)
+        QMacAutoReleasePool pool;
         QString localizedFamilyName = QString::fromNSString([[NSFontManager sharedFontManager] localizedNameForFamily:(NSString*)familyNameRef face:nil]);
         if (familyName != localizedFamilyName)
             QPlatformFontDatabase::registerAliasToFontFamily(familyName, localizedFamilyName);
diff --git a/qtbase/src/plugins/platforms/cocoa/qcocoanativeinterface.mm b/qtbase/src/plugins/platforms/cocoa/qcocoanativeinterface.mm
index 9722303..b0eacc2 100644
--- a/qtbase/src/plugins/platforms/cocoa/qcocoanativeinterface.mm
+++ b/qtbase/src/plugins/platforms/cocoa/qcocoanativeinterface.mm
@@ -260,6 +260,7 @@ void QCocoaNativeInterface::setDockMenu(QPlatformMenu *platformMenu)
 
 void *QCocoaNativeInterface::qMenuToNSMenu(QPlatformMenu *platformMenu)
 {
+    QMacAutoReleasePool pool;
     QCocoaMenu *cocoaPlatformMenu = static_cast<QCocoaMenu *>(platformMenu);
     NSMenu *menu = cocoaPlatformMenu->nsMenu();
     return reinterpret_cast<void *>(menu);
diff --git a/qtbase/src/plugins/platforms/cocoa/qcocoawindow.mm b/qtbase/src/plugins/platforms/cocoa/qcocoawindow.mm
index 602da0a..a8f7cd8 100644
--- a/qtbase/src/plugins/platforms/cocoa/qcocoawindow.mm
+++ b/qtbase/src/plugins/platforms/cocoa/qcocoawindow.mm
@@ -1727,6 +1727,8 @@ void QCocoaWindow::applyContentBorderThickness(NSWindow *window)
     if (!window)
         return;
 
+    QMacAutoReleasePool pool;
+
     if (!m_drawContentBorderGradient) {
         [window setStyleMask:[window styleMask] & ~NSTexturedBackgroundWindowMask];
         [[[window contentView] superview] setNeedsDisplay:YES];
