diff --git a/qtbase/src/widgets/widgets/qmenu_mac.mm b/qtbase/src/widgets/widgets/qmenu_mac.mm
index 05d01a7..0b58609 100644
--- a/qtbase/src/widgets/widgets/qmenu_mac.mm
+++ b/qtbase/src/widgets/widgets/qmenu_mac.mm
@@ -95,7 +95,9 @@ void QMenu::setAsDockMenu()
     QPlatformNativeInterface::NativeResourceForIntegrationFunction function = resolvePlatformFunction("setdockmenu");
     if (function) {
         typedef void (*SetDockMenuFunction)(QPlatformMenu *platformMenu);
-        reinterpret_cast<SetDockMenuFunction>(function)(platformMenu());
+        QPlatformMenu *pMenu = platformMenu();
+        if (pMenu)
+            reinterpret_cast<SetDockMenuFunction>(function)(pMenu);
     }
 }
 
