diff --git a/qttools/src/assistant/assistant/assistant.pro b/qttools/src/assistant/assistant/assistant.pro
index ef5c595..42eb660 100644
--- a/qttools/src/assistant/assistant/assistant.pro
+++ b/qttools/src/assistant/assistant/assistant.pro
@@ -1,11 +1,11 @@
-qtHaveModule(webkitwidgets):!contains(QT_CONFIG, static) {
-    BROWSER = qtwebkit
-} else {
+# qtHaveModule(webkitwidgets):!contains(QT_CONFIG, static) {
+#     BROWSER = qtwebkit
+# } else {
     BROWSER = qtextbrowser
-}
+# }
 
 QT += widgets network help sql help
-qtHaveModule(printsupport): QT += printsupport
+QT += printsupport
 PROJECTNAME = Assistant
 
 include(../../shared/fontpanel/fontpanel.pri)
@@ -115,4 +115,4 @@ mac {
 
 contains(SQLPLUGINS, sqlite):QTPLUGIN += qsqlite
 
-load(qt_app)
+# load(qt_app)
