diff --git a/qtbase/mkspecs/common/mac.conf b/qtbase/mkspecs/common/mac.conf
index 5208379..4591a3b 100644
--- a/qtbase/mkspecs/common/mac.conf
+++ b/qtbase/mkspecs/common/mac.conf
@@ -17,9 +17,7 @@ QMAKE_EXTENSION_SHLIB   = dylib
 QMAKE_LIBDIR            =
 
 # sdk.prf will prefix the proper SDK sysroot
-QMAKE_INCDIR_OPENGL     = \
-    /System/Library/Frameworks/OpenGL.framework/Headers \
-    /System/Library/Frameworks/AGL.framework/Headers/
+QMAKE_INCDIR_OPENGL     = 
 
 QMAKE_FIX_RPATH         = install_name_tool -id
 
@@ -30,7 +28,8 @@ QMAKE_LFLAGS_REL_RPATH  =
 QMAKE_REL_RPATH_BASE    = @loader_path
 
 QMAKE_LIBS_DYNLOAD      =
-QMAKE_LIBS_OPENGL       = -framework OpenGL -framework AGL
+QMAKE_LIBS_OPENGL       = -lGL
+QMAKE_LIBS_X11          = -lXext -lX11
 QMAKE_LIBS_THREAD       =
 
 QMAKE_ACTOOL            = actool
diff --git a/qtbase/src/gui/opengl/qopengl.h b/qtbase/src/gui/opengl/qopengl.h
index cd44ddf..6e133c5 100644
--- a/qtbase/src/gui/opengl/qopengl.h
+++ b/qtbase/src/gui/opengl/qopengl.h
@@ -126,7 +126,7 @@ typedef char GLchar;
 #  include <QtGui/qopengles2ext.h>
 # endif // Q_OS_MAC
 #else // non-ES2 platforms
-# if defined(Q_OS_MAC)
+# if defined(Q_OS_MAC_NOTX11)
 #  include <OpenGL/gl.h>
 #  define GL_DO_NOT_WARN_IF_MULTI_GL_VERSION_HEADERS_INCLUDED
 #  include <OpenGL/gl3.h>
