diff --git a/panel-plugin/notification-plugin-dialogs.c b/panel-plugin/notification-plugin-dialogs.c
index 9161bf9..db0786d 100644
--- a/panel-plugin/notification-plugin-dialogs.c
+++ b/panel-plugin/notification-plugin-dialogs.c
@@ -97,7 +97,11 @@ notification_plugin_configure (XfcePanelPlugin      *plugin,
   gtk_container_add_with_properties (GTK_CONTAINER (gtk_dialog_get_content_area (GTK_DIALOG (dialog))),
 						                         grid, "expand", TRUE, "fill", TRUE, NULL);
   label = gtk_label_new (_("Number of notifications to show"));
-  gtk_label_set_xalign (GTK_LABEL (label), 0.0);
+#if GTK_CHECK_VERSION (3, 16, 0)
+  gtk_label_set_xalign (GTK_LABEL (label), 0);
+#else
+  gtk_widget_set_halign (label, GTK_ALIGN_START);
+#endif
   gtk_grid_attach (GTK_GRID (grid), GTK_WIDGET (label), 0, 0, 1, 1);
   spin = gtk_spin_button_new (adjustment, 1.0, 0);
   gtk_grid_attach (GTK_GRID (grid), GTK_WIDGET (spin), 1, 0, 1, 1);
@@ -105,7 +109,11 @@ notification_plugin_configure (XfcePanelPlugin      *plugin,
                           G_OBJECT (spin), "value");
 
   label = gtk_label_new (_("Only show notifications from today"));
-  gtk_label_set_xalign (GTK_LABEL (label), 0.0);
+#if GTK_CHECK_VERSION (3, 16, 0)
+  gtk_label_set_xalign (GTK_LABEL (label), 0);
+#else
+  gtk_widget_set_halign (label, GTK_ALIGN_START);
+#endif
   gtk_grid_attach (GTK_GRID (grid), GTK_WIDGET (label), 0, 1, 1, 1);
   check = gtk_switch_new ();
   gtk_widget_set_halign (check, GTK_ALIGN_END);
diff --git a/panel-plugin/notification-plugin-log.c b/panel-plugin/notification-plugin-log.c
index cc1fa4f..b0e5bf9 100644
--- a/panel-plugin/notification-plugin-log.c
+++ b/panel-plugin/notification-plugin-log.c
@@ -130,7 +130,11 @@ notification_plugin_menu_populate (NotificationPlugin *notification_plugin)
   box = gtk_box_new (GTK_ORIENTATION_HORIZONTAL, 6);
   label = gtk_label_new (NULL);
   gtk_label_set_markup_with_mnemonic (GTK_LABEL (label), _("<b>_Do not disturb</b>"));
-  gtk_label_set_xalign (GTK_LABEL (label), 0.0);
+#if GTK_CHECK_VERSION (3, 16, 0)
+  gtk_label_set_xalign (GTK_LABEL (label), 0);
+#else
+  gtk_widget_set_halign (label, GTK_ALIGN_START);
+#endif
   notification_plugin->do_not_disturb_switch = gtk_switch_new ();
   gtk_box_pack_start (GTK_BOX (box), label, TRUE, TRUE, 0);
   gtk_box_pack_start (GTK_BOX (box), notification_plugin->do_not_disturb_switch, FALSE, FALSE, 0);
@@ -222,7 +226,11 @@ G_GNUC_END_IGNORE_DEPRECATIONS
       g_free (tmp);
       summary = gtk_label_new (NULL);
       gtk_label_set_markup (GTK_LABEL (summary), markup);
+#if GTK_CHECK_VERSION (3, 16, 0)
       gtk_label_set_xalign (GTK_LABEL (summary), 0);
+#else
+      gtk_widget_set_halign (summary, GTK_ALIGN_START);
+#endif
       gtk_label_set_ellipsize (GTK_LABEL (summary), PANGO_ELLIPSIZE_END);
       gtk_label_set_max_width_chars (GTK_LABEL (summary), 40);
       g_free (markup);
@@ -238,7 +246,11 @@ G_GNUC_END_IGNORE_DEPRECATIONS
         g_free (tmp1);
       }
       g_free (tmp);
+#if GTK_CHECK_VERSION (3, 16, 0)
       gtk_label_set_xalign (GTK_LABEL (body), 0);
+#else
+      gtk_widget_set_halign (body, GTK_ALIGN_START);
+#endif
       gtk_label_set_ellipsize (GTK_LABEL (body), PANGO_ELLIPSIZE_END);
       gtk_label_set_max_width_chars (GTK_LABEL (body), 40);
 
