diff --git src/interface/resources/orig.MacInfo.plist.in src/interface/resources/MacInfo.plist.in
index 4b2dbf4..2c5820a 100644
--- src/interface/resources/orig.MacInfo.plist.in
+++ src/interface/resources/MacInfo.plist.in
@@ -40,8 +40,6 @@
 	<array>
 		<string>MacOSX</string>
 	</array>
-	<key>LSMinimumSystemVersion</key>
-	<string>10.13.2</string>
 	<key>LSApplicationCategoryType</key>
 	<string>public.app-category.utilities</string>
 	<key>NSNetworkVolumesUsageDescription</key>
