diff --git a/plugins/bazaar/bazaarplugin.cpp b/plugins/bazaar/bazaarplugin.cpp
index e4463a5ff71b3a59d17ed86aaaa73941ee9b0d09..d1e3403c33005a0903918dbb3a296148e2ac32c1 100644
--- a/plugins/bazaar/bazaarplugin.cpp
+++ b/plugins/bazaar/bazaarplugin.cpp
@@ -122,9 +122,11 @@ VcsJob* BazaarPlugin::createWorkingCopy(const VcsLocation& sourceRepository, con
     return job;
 }
 
-VcsJob* BazaarPlugin::diff(const QUrl& fileOrDirectory, const VcsRevision& srcRevision, const VcsRevision& dstRevision, VcsDiff::Type, IBasicVersionControl::RecursionMode recursion)
+VcsJob* BazaarPlugin::diff(const QUrl& fileOrDirectory, const VcsRevision& srcRevision, const VcsRevision& dstRevision, int contextLines, VcsDiff::Type, IBasicVersionControl::RecursionMode recursion)
 {
     Q_UNUSED(recursion);
+    // TODO support contextLines (less urgent because Phabricator doesn't support Bazaar)
+    Q_UNUSED(contextLines);
     VcsJob* job = new DiffJob(BazaarUtils::workingCopy(fileOrDirectory), BazaarUtils::getRevisionSpecRange(srcRevision, dstRevision), fileOrDirectory, this);
     return job;
 }
diff --git a/plugins/bazaar/bazaarplugin.h b/plugins/bazaar/bazaarplugin.h
index c5fee4615efb1ce26933599b09adeb5cf446745d..945e2923e9f7145a6382843be0a2c5f3f5b35881 100644
--- a/plugins/bazaar/bazaarplugin.h
+++ b/plugins/bazaar/bazaarplugin.h
@@ -50,7 +50,7 @@ public:
     KDevelop::VcsJob* copy(const QUrl& localLocationSrc, const QUrl& localLocationDstn) override;
     KDevelop::VcsImportMetadataWidget* createImportMetadataWidget(QWidget* parent) override;
     KDevelop::VcsJob* createWorkingCopy(const KDevelop::VcsLocation& sourceRepository, const QUrl& destinationDirectory, RecursionMode recursion=Recursive) override;
-    KDevelop::VcsJob* diff(const QUrl& fileOrDirectory, const KDevelop::VcsRevision& srcRevision, const KDevelop::VcsRevision& dstRevision, KDevelop::VcsDiff::Type, RecursionMode recursion=Recursive) override;
+    KDevelop::VcsJob* diff(const QUrl& fileOrDirectory, const KDevelop::VcsRevision& srcRevision, const KDevelop::VcsRevision& dstRevision, int contextLines=-1, KDevelop::VcsDiff::Type=KDevelop::VcsDiff::DiffUnified, RecursionMode recursion=Recursive) override;
     KDevelop::VcsJob* init(const QUrl& localRepositoryRoot) override;
     bool isVersionControlled(const QUrl& localLocation) override;
     KDevelop::VcsJob* log(const QUrl& localLocation, const KDevelop::VcsRevision& rev, long unsigned int limit) override;
diff --git a/plugins/cvs/cvsplugin.cpp b/plugins/cvs/cvsplugin.cpp
index b37980b20487150f3763422edc6f744b78536f35..64c6ce67d11aa7e2cd72567cac2eb283d314dfb7 100644
--- a/plugins/cvs/cvsplugin.cpp
+++ b/plugins/cvs/cvsplugin.cpp
@@ -393,9 +393,10 @@ KDevelop::VcsJob * CvsPlugin::commit(const QString & message, const QList<QUrl>
     return job;
 }
 
-KDevelop::VcsJob * CvsPlugin::diff(const QUrl & fileOrDirectory, const KDevelop::VcsRevision & srcRevision, const KDevelop::VcsRevision & dstRevision, KDevelop::VcsDiff::Type, KDevelop::IBasicVersionControl::RecursionMode)
+KDevelop::VcsJob * CvsPlugin::diff(const QUrl & fileOrDirectory, const KDevelop::VcsRevision & srcRevision, const KDevelop::VcsRevision & dstRevision, int contextLines, KDevelop::VcsDiff::Type, KDevelop::IBasicVersionControl::RecursionMode)
 {
-    CvsJob* job = d->m_proxy->diff(fileOrDirectory, srcRevision, dstRevision, QStringLiteral("-uN")/*always unified*/);
+    QString options = contextLines > 0? QStringLiteral("-uN -context=%1").arg(contextLines) : QStringLiteral("-uN")/*always unified*/;
+    CvsJob* job = d->m_proxy->diff(fileOrDirectory, srcRevision, dstRevision, options);
     return job;
 }
 
diff --git a/plugins/cvs/cvsplugin.h b/plugins/cvs/cvsplugin.h
index fa5efe6f9fbeaac0e327c5a29226f56f97e9dbbc..910006e82227767e1905cd073a930ab19f5cb931 100644
--- a/plugins/cvs/cvsplugin.h
+++ b/plugins/cvs/cvsplugin.h
@@ -75,7 +75,8 @@ public:
     KDevelop::VcsJob* diff(const QUrl& fileOrDirectory,
                                    const KDevelop::VcsRevision& srcRevision,
                                    const KDevelop::VcsRevision& dstRevision,
-                                   KDevelop::VcsDiff::Type,
+                                   int contextLines=-1,
+                                   KDevelop::VcsDiff::Type=KDevelop::VcsDiff::DiffUnified,
                                    KDevelop::IBasicVersionControl::RecursionMode = KDevelop::IBasicVersionControl::Recursive) override;
     KDevelop::VcsJob* log(const QUrl& localLocation,
                                   const KDevelop::VcsRevision& rev,
diff --git a/plugins/git/gitplugin.cpp b/plugins/git/gitplugin.cpp
index 35e2ec88bc16a35904022790269a47b33b2ad1ab..2cf882cb51558cbecb4aca560567bd08a5bda0ab 100644
--- a/plugins/git/gitplugin.cpp
+++ b/plugins/git/gitplugin.cpp
@@ -371,18 +371,21 @@ KDevelop::VcsJob* GitPlugin::status(const QList<QUrl>& localLocations, KDevelop:
 }
 
 VcsJob* GitPlugin::diff(const QUrl& fileOrDirectory, const KDevelop::VcsRevision& srcRevision, const KDevelop::VcsRevision& dstRevision,
-                        VcsDiff::Type /*type*/, IBasicVersionControl::RecursionMode recursion)
+                        int contextLines, VcsDiff::Type /*type*/, IBasicVersionControl::RecursionMode recursion)
 {
     //TODO: control different types
 
     DVcsJob* job = new GitJob(dotGitDirectory(fileOrDirectory), this, KDevelop::OutputJob::Silent);
     job->setType(VcsJob::Diff);
-    *job << "git" << "diff" << "--no-color" << "--no-ext-diff";
+    *job << "git" << "diff" << "--no-color" << "--no-ext-diff" << "--full-index";
     if (!usePrefix()) {
         // KDE's ReviewBoard now requires p1 patchfiles, so `git diff --no-prefix` to generate p0 patches
         // has become optional.
         *job << "--no-prefix";
     }
+    if (contextLines > 0) {
+        *job << QStringLiteral("-U%1").arg(contextLines);
+    }
     if (dstRevision.revisionType() == VcsRevision::Special &&
          dstRevision.specialType() == VcsRevision::Working) {
         if (srcRevision.revisionType() == VcsRevision::Special &&
diff --git a/plugins/git/gitplugin.h b/plugins/git/gitplugin.h
index e8b3191a7722e13b2a19367c52d5960d0b3453c0..0fc644288670b9990ad3eca8a62cdb43676f370e 100644
--- a/plugins/git/gitplugin.h
+++ b/plugins/git/gitplugin.h
@@ -106,7 +106,7 @@ public:
                              KDevelop::IBasicVersionControl::RecursionMode recursion = KDevelop::IBasicVersionControl::Recursive) override;
 
     KDevelop::VcsJob* diff(const QUrl& fileOrDirectory, const KDevelop::VcsRevision& srcRevision, const KDevelop::VcsRevision& dstRevision,
-                                   KDevelop::VcsDiff::Type, RecursionMode recursion) override;
+                                   int contextLines=-1, KDevelop::VcsDiff::Type=KDevelop::VcsDiff::DiffUnified, RecursionMode recursion=IBasicVersionControl::Recursive) override;
 
     KDevelop::VcsJob* log( const QUrl& localLocation, const KDevelop::VcsRevision& rev, unsigned long limit) override;
     KDevelop::VcsJob* log(const QUrl& localLocation, const KDevelop::VcsRevision& rev, const KDevelop::VcsRevision& limit) override;
diff --git a/plugins/git/tests/test_git.cpp b/plugins/git/tests/test_git.cpp
index 01202aedf0f72f57402d4994175d7f3427140670..799ac491bf3cb5e7b7b18f8910d41a19ddec338b 100644
--- a/plugins/git/tests/test_git.cpp
+++ b/plugins/git/tests/test_git.cpp
@@ -574,7 +574,7 @@ void GitInitTest::testDiff()
 
     VcsRevision srcrev = VcsRevision::createSpecialRevision(VcsRevision::Base);
     VcsRevision dstrev = VcsRevision::createSpecialRevision(VcsRevision::Working);
-    VcsJob* j = m_plugin->diff(QUrl::fromLocalFile(gitTest_BaseDir()), srcrev, dstrev, VcsDiff::DiffUnified, IBasicVersionControl::Recursive);
+    VcsJob* j = m_plugin->diff(QUrl::fromLocalFile(gitTest_BaseDir()), srcrev, dstrev, -1, VcsDiff::DiffUnified, IBasicVersionControl::Recursive);
     VERIFYJOB(j);
 
     KDevelop::VcsDiff d = j->fetchResults().value<KDevelop::VcsDiff>();
diff --git a/plugins/patchreview/patchreview.cpp b/plugins/patchreview/patchreview.cpp
index 47b6fa255ac06e6221f09cb3f9c45d1149f15b15..b15ba2095333427b96c6658cc8b37b5813b84317 100644
--- a/plugins/patchreview/patchreview.cpp
+++ b/plugins/patchreview/patchreview.cpp
@@ -223,6 +223,14 @@ void PatchReviewPlugin::forceUpdate() {
     }
 }
 
+void PatchReviewPlugin::forceUpdateWithContext(int contextLines) {
+    if( m_patch ) {
+        m_patch->update(contextLines);
+
+        notifyPatchChanged();
+    }
+}
+
 void PatchReviewPlugin::updateKompareModel() {
     if ( !m_patch ) {
         ///TODO: this method should be cleaned up, it can be called by the timer and
diff --git a/plugins/patchreview/patchreview.h b/plugins/patchreview/patchreview.h
index 92090e124a495377a44dcbc6ac37fe9c740f0ea8..e6b81327a7c8e9a490098a750862609480520fb4 100644
--- a/plugins/patchreview/patchreview.h
+++ b/plugins/patchreview/patchreview.h
@@ -100,6 +100,7 @@ public Q_SLOTS :
     void highlightPatch();
     void updateKompareModel();
     void forceUpdate();
+    void forceUpdateWithContext(int contextLines);
     void areaChanged(Sublime::Area* area);
     void executeFileReviewAction();
 
diff --git a/plugins/patchreview/patchreview.ui b/plugins/patchreview/patchreview.ui
index c706c0cca2b6387ccbd9dbd132d1b0eae1870df8..37c1aa3fc3dd69e6328e3e92d60b264db555f610 100644
--- a/plugins/patchreview/patchreview.ui
+++ b/plugins/patchreview/patchreview.ui
@@ -72,6 +72,28 @@
         </spacer>
        </item>
        <item>
+        <widget class="QSpinBox" name="diffContext">
+         <property name="toolTip">
+          <string>lines of context for the unified diff</string>
+         </property>
+         <property name="specialValueText">
+          <string>max</string>
+         </property>
+         <property name="suffix">
+          <string extracomment="lines of context text"/>
+         </property>
+         <property name="minimum">
+          <number>0</number>
+         </property>
+         <property name="maximum">
+          <number>999</number>
+         </property>
+         <property name="value">
+          <number>3</number>
+         </property>
+        </widget>
+       </item>
+       <item>
         <widget class="QPushButton" name="updateButton">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
diff --git a/plugins/patchreview/patchreviewtoolview.cpp b/plugins/patchreview/patchreviewtoolview.cpp
index c088414acf20dec576355524e8df0ef93d48686e..b9439c5c9ee166f4cc32c76250771dc0fdf8cd7a 100644
--- a/plugins/patchreview/patchreviewtoolview.cpp
+++ b/plugins/patchreview/patchreviewtoolview.cpp
@@ -126,6 +126,11 @@ void PatchReviewToolView::startingNewReview()
 }
 
 void PatchReviewToolView::patchChanged() {
+    if(m_resetCheckedUrls) {
+        // set the default number of context lines
+        m_editPatch.diffContext->setValue(3);
+    }
+
     fillEditFromPatch();
     kompareModelChanged();
 
@@ -251,6 +256,8 @@ void PatchReviewToolView::showEditDialog() {
     //connect( this, SIGNAL(finished(int)), this, SLOT(slotEditDialogFinished(int)) );
 
     connect( m_editPatch.updateButton, &QPushButton::clicked, m_plugin, &PatchReviewPlugin::forceUpdate );
+    connect( m_editPatch.diffContext, static_cast<void(QSpinBox::*)(int)>(&QSpinBox::valueChanged),
+             m_plugin, &PatchReviewPlugin::forceUpdateWithContext );
 
     connect( m_editPatch.testsButton, &QPushButton::clicked, this, &PatchReviewToolView::runTests );
 
diff --git a/plugins/perforce/perforceplugin.cpp b/plugins/perforce/perforceplugin.cpp
index fb16f9daa60940810114138c3e0ccf640cea03ed..58b27ed7a7d4c233b4cb5ac124a42d4a218fc6d6 100644
--- a/plugins/perforce/perforceplugin.cpp
+++ b/plugins/perforce/perforceplugin.cpp
@@ -309,7 +309,7 @@ KDevelop::VcsJob* PerforcePlugin::commit(const QString& message, const QList<QUr
     return job;
 }
 
-KDevelop::VcsJob* PerforcePlugin::diff(const QUrl& fileOrDirectory, const KDevelop::VcsRevision& srcRevision, const KDevelop::VcsRevision& dstRevision, KDevelop::VcsDiff::Type , KDevelop::IBasicVersionControl::RecursionMode /*recursion*/)
+KDevelop::VcsJob* PerforcePlugin::diff(const QUrl& fileOrDirectory, const KDevelop::VcsRevision& srcRevision, const KDevelop::VcsRevision& dstRevision, int contextLines, KDevelop::VcsDiff::Type , KDevelop::IBasicVersionControl::RecursionMode /*recursion*/)
 {
     QFileInfo curFile(fileOrDirectory.toLocalFile());
     QString depotSrcFileName = getRepositoryName(curFile);
@@ -328,7 +328,11 @@ KDevelop::VcsJob* PerforcePlugin::diff(const QUrl& fileOrDirectory, const KDevel
     case VcsRevision::Special:
         switch (dstRevision.revisionValue().value<VcsRevision::RevisionSpecialType>()) {
         case VcsRevision::Working:
-            *job << m_perforceExecutable << "diff" << "-du" << depotSrcFileName;
+            if (contextLines > 0) {
+                *job << m_perforceExecutable << "diff" << "-dU" << QString::number(contextLines) << depotSrcFileName;
+            } else {
+                *job << m_perforceExecutable << "diff" << "-du" << depotSrcFileName;
+            }
             break;
         case VcsRevision::Start:
         case VcsRevision::UserSpecialType:
diff --git a/plugins/perforce/perforceplugin.h b/plugins/perforce/perforceplugin.h
index dcb2560ec7b3859793fdc8f465292fd28e2b0fa4..5e6c1c12b30e490897f6b6d51b910186dac99769 100644
--- a/plugins/perforce/perforceplugin.h
+++ b/plugins/perforce/perforceplugin.h
@@ -88,6 +88,7 @@ public:
     KDevelop::VcsJob* diff(const QUrl& fileOrDirectory,
                            const KDevelop::VcsRevision& srcRevision,
                            const KDevelop::VcsRevision& dstRevision,
+                           int contextLines=-1,
                            KDevelop::VcsDiff::Type = KDevelop::VcsDiff::DiffUnified,
                            KDevelop::IBasicVersionControl::RecursionMode recursion = KDevelop::IBasicVersionControl::Recursive) override;
 
diff --git a/plugins/subversion/kdevsvnplugin.cpp b/plugins/subversion/kdevsvnplugin.cpp
index cb70196f6e927d0e311080906f7cb832f56611c4..7a28a4f95f51ee2e5e4b4d896889569c24f17aa6 100644
--- a/plugins/subversion/kdevsvnplugin.cpp
+++ b/plugins/subversion/kdevsvnplugin.cpp
@@ -231,20 +231,24 @@ KDevelop::VcsJob* KDevSvnPlugin::commit(const QString& message, const QList<QUrl
 KDevelop::VcsJob* KDevSvnPlugin::diff(const QUrl &fileOrDirectory,
                                       const KDevelop::VcsRevision& srcRevision,
                                       const KDevelop::VcsRevision& dstRevision,
+                                      int contextLines,
                                       KDevelop::VcsDiff::Type diffType,
                                       KDevelop::IBasicVersionControl::RecursionMode recurse)
 {
     KDevelop::VcsLocation loc(fileOrDirectory);
-    return diff2(loc, loc, srcRevision, dstRevision, diffType, recurse);
+    return diff2(loc, loc, srcRevision, dstRevision, contextLines, diffType, recurse);
 }
 
 KDevelop::VcsJob* KDevSvnPlugin::diff2(const KDevelop::VcsLocation& src,
                                        const KDevelop::VcsLocation& dst,
                                        const KDevelop::VcsRevision& srcRevision,
                                        const KDevelop::VcsRevision& dstRevision,
+                                       int contextLines,
                                        KDevelop::VcsDiff::Type diffType,
                                        KDevelop::IBasicVersionControl::RecursionMode recurse)
 {
+    // TODO: support custom context lines
+    Q_UNUSED(contextLines);
     SvnDiffJob* job = new SvnDiffJob(this);
     job->setSource(src);
     job->setDestination(dst);
diff --git a/plugins/subversion/kdevsvnplugin.h b/plugins/subversion/kdevsvnplugin.h
index 87ed0e8b9e923720cb93fc3c917bd6e5b0744f43..e4a57585d9e0e6ac4433d049f249225620ea2cfb 100644
--- a/plugins/subversion/kdevsvnplugin.h
+++ b/plugins/subversion/kdevsvnplugin.h
@@ -84,6 +84,7 @@ public:
     KDevelop::VcsJob* diff(const QUrl &fileOrDirectory,
                            const KDevelop::VcsRevision& srcRevision,
                            const KDevelop::VcsRevision& dstRevision,
+                           int contextLines=-1,
                            KDevelop::VcsDiff::Type = KDevelop::VcsDiff::DiffUnified,
                            KDevelop::IBasicVersionControl::RecursionMode recursion
                            = KDevelop::IBasicVersionControl::Recursive) override;
@@ -97,6 +98,7 @@ public:
                             const KDevelop::VcsLocation& localOrRepoLocationDst,
                             const KDevelop::VcsRevision& srcRevision,
                             const KDevelop::VcsRevision& dstRevision,
+                            int contextLines,
                             KDevelop::VcsDiff::Type = KDevelop::VcsDiff::DiffDontCare,
                             KDevelop::IBasicVersionControl::RecursionMode = KDevelop::IBasicVersionControl::Recursive);
 
@@ -154,6 +156,7 @@ private:
     QAction* copy_action;
     QAction* move_action;
     ThreadWeaver::Queue* m_jobQueue;
+    int m_contextLines=3;
 };
 #endif
 
diff --git a/vcs/interfaces/ibasicversioncontrol.h b/vcs/interfaces/ibasicversioncontrol.h
index f438c54bf0d7fb90663bb083bcde0c5860c7f739..51d8069ca5065ea8e3db3f4ef10e683d799be3f4 100644
--- a/vcs/interfaces/ibasicversioncontrol.h
+++ b/vcs/interfaces/ibasicversioncontrol.h
@@ -194,6 +194,7 @@ public:
     virtual VcsJob* diff( const QUrl& fileOrDirectory,
                           const VcsRevision& srcRevision,
                           const VcsRevision& dstRevision,
+                          int contextLines=-1,
                           VcsDiff::Type = VcsDiff::DiffUnified,
                           IBasicVersionControl::RecursionMode recursion
                                        = IBasicVersionControl::Recursive ) = 0;
diff --git a/vcs/interfaces/ipatchsource.h b/vcs/interfaces/ipatchsource.h
index 6e2939d2790a1309251aa73eb446276ab2a05805..117722f3a137786297b3f83b53f96f0ac42cf4b6 100644
--- a/vcs/interfaces/ipatchsource.h
+++ b/vcs/interfaces/ipatchsource.h
@@ -48,6 +48,12 @@ class KDEVPLATFORMVCS_EXPORT IPatchSource : public QObject
         ///should re-compare the files or whatever needs to be done
         ///If the patch has changed, patchChanged needs to be emitted
         virtual void update() = 0;
+        ///Explicit updating of the patch with a specific number of context lines.
+        ///The default implementation ignores the argument and calls update()
+        virtual void update(int contextLines) {
+            Q_UNUSED(contextLines);
+            update();
+        }
 
         ///Name of the patch file
         virtual QUrl file() const = 0;
diff --git a/vcs/widgets/vcsdiffpatchsources.cpp b/vcs/widgets/vcsdiffpatchsources.cpp
index 42cf1d844922c2076ea2127798ed02e9bf9c639a..0dbc5bc6b056e01a59a2fedd3a11ca399cbfc248 100644
--- a/vcs/widgets/vcsdiffpatchsources.cpp
+++ b/vcs/widgets/vcsdiffpatchsources.cpp
@@ -210,6 +210,14 @@ void VCSDiffPatchSource::update() {
     updateFromDiff(m_updater->update());
 }
 
+void VCSDiffPatchSource::update(int contextLines)
+{
+    if (m_updater) {
+        m_updater->setContextLines(contextLines > 0? contextLines : INT_MAX);
+        updateFromDiff(m_updater->update());
+    }
+}
+
 VCSCommitDiffPatchSource::~VCSCommitDiffPatchSource() {
     delete m_commitMessageWidget.data();
 }
@@ -297,7 +305,8 @@ VcsDiff VCSStandardDiffUpdater::update() const
 {
     QScopedPointer<VcsJob> diffJob(m_vcs->diff(m_url,
                                    KDevelop::VcsRevision::createSpecialRevision(KDevelop::VcsRevision::Base),
-                                   KDevelop::VcsRevision::createSpecialRevision(KDevelop::VcsRevision::Working)));
+                                   KDevelop::VcsRevision::createSpecialRevision(KDevelop::VcsRevision::Working),
+                                   m_contextLines));
     const bool success = diffJob ? diffJob->exec() : false;
     if (!success) {
         KMessageBox::error(nullptr, i18n("Could not create a patch for the current version."));
diff --git a/vcs/widgets/vcsdiffpatchsources.h b/vcs/widgets/vcsdiffpatchsources.h
index 796eb9eb0abc3d317f08354372893431a1693e82..b531a61e5f1f3590c54e104b96a01b7bdad93866 100644
--- a/vcs/widgets/vcsdiffpatchsources.h
+++ b/vcs/widgets/vcsdiffpatchsources.h
@@ -45,6 +45,13 @@ public:
     virtual KDevelop::VcsDiff update() const = 0;
     virtual KDevelop::IBasicVersionControl* vcs() const = 0;
     virtual QUrl url() const = 0;
+    void setContextLines(int n)
+    {
+        m_contextLines = n;
+    }
+    int contextLines() { return m_contextLines; }
+protected:
+    int m_contextLines=-1;
 };
 
 class KDEVPLATFORMVCS_EXPORT VCSStandardDiffUpdater : public VCSDiffUpdater {
@@ -77,6 +84,7 @@ class KDEVPLATFORMVCS_EXPORT VCSDiffPatchSource : public KDevelop::IPatchSource
     uint depth() const override ;
 
     void update() override ;
+    void update(int contextLines) override;
 
     bool isAlreadyApplied() const override { return true; }
 
