diff --git kdevplatform/plugins/filemanager/filemanager.cpp kdevplatform/plugins/filemanager/filemanager.cpp
index 5199903a75c7d6ec95d38ac5f9a2cec60a3404f9..74b709f843f74700a4a0b3849dcd824f23d21e86 100644
--- kdevplatform/plugins/filemanager/filemanager.cpp
+++ kdevplatform/plugins/filemanager/filemanager.cpp
@@ -118,6 +118,7 @@ void FileManager::fillContextMenu(KFileItem item, QMenu* menu)
     KDevelop::ContextMenuExtension::populateMenu(tmpMenu, extensions);
     contextActions.append(tmpMenu->actions());
     delete tmpMenu;
+    menu->setTearOffEnabled(true);
 }
 
 void FileManager::openFile(const KFileItem& file)
diff --git kdevplatform/shell/textdocument.cpp kdevplatform/shell/textdocument.cpp
index 5315c0a10cfdcf74cebdd1a4bd3f64474be716b2..a9f4b18d4b71dad90b33b04c213ccd5595636bb2 100644
--- kdevplatform/shell/textdocument.cpp
+++ kdevplatform/shell/textdocument.cpp
@@ -745,6 +745,7 @@ void KDevelop::TextDocument::populateContextMenu( KTextEditor::View* v, QMenu* m
     foreach ( QAction* action, d->m_addedContextMenu->actions() ) {
         menu->addAction(action);
     }
+    menu->setTearOffEnabled(true);
 }
 
 void KDevelop::TextDocument::repositoryCheckFinished(bool canRecreate) {
diff --git kdevplatform/vcs/vcspluginhelper.cpp kdevplatform/vcs/vcspluginhelper.cpp
index 485c8b8474fde00b7c54edda628995ba438317ce..31f0bb3c1da8dc9e2b2de4aae7fcb08ee4255362 100644
--- kdevplatform/vcs/vcspluginhelper.cpp
+++ kdevplatform/vcs/vcspluginhelper.cpp
@@ -433,6 +435,7 @@ void VcsPluginHelper::annotationContextMenuAboutToShow( KTextEditor::View* view,
     connect(action, &QAction::triggered, this, [this, rev]() {
         history(rev);
     });
+    menu->setTearOffEnabled(true);
 }
 
 void VcsPluginHelper::update()
diff --git kdevplatform/plugins/projectmanagerview/projectbuildsetwidget.cpp kdevplatform/plugins/projectmanagerview/projectbuildsetwidget.cpp
index bde3f7fa2667aa74e2f5071428560343bff6bc41..fea9f59db5a494d3c985ca8c0b5acc189a64522b 100644
--- kdevplatform/plugins/projectmanagerview/projectbuildsetwidget.cpp
+++ kdevplatform/plugins/projectmanagerview/projectbuildsetwidget.cpp
@@ -146,6 +146,7 @@ void ProjectBuildSetWidget::showContextMenu( const QPoint& p )
 
 
     QMenu m(this);
+    m.setTearOffEnabled(true);
     m.setTitle( i18n("Build Set") );
     m.addAction( QIcon::fromTheme(QStringLiteral("list-remove")), i18n( "Remove From Build Set" ), this, SLOT(removeItems()) );
 
