diff --git shell/mainwindow.cpp shell/mainwindow.cpp
index 6428f9ab634157edcb61895f7379067cd02fa537..df655fcd58a095ffb40b96f4eb8b7dd85ca76de8 100644
--- shell/mainwindow.cpp
+++ shell/mainwindow.cpp
@@ -383,7 +383,10 @@ void MainWindow::updateCaption()
         Sublime::Document* doc = area()->activeView()->document();
         Sublime::UrlDocument* urlDoc = dynamic_cast<Sublime::UrlDocument*>(doc);
         if(urlDoc)
+        {
+            setWindowFilePath(urlDoc->url().toLocalFile());
             title += Core::self()->projectController()->prettyFileName(urlDoc->url(), KDevelop::IProjectController::FormatPlain);
+        }
         else
             title += doc->title();
 
@@ -393,7 +396,9 @@ void MainWindow::updateCaption()
 
         title += QLatin1String(" ]");
     }
-
+    else {
+        setWindowFilePath({});
+    }
     setCaption(title);
 }
 
diff --git sublime/container.cpp sublime/container.cpp
index b9d2059c7084a44a8fc1f25a01173fe065110314..23f85fc136121182c787f747e4d1c8009660151f 100644
--- sublime/container.cpp
+++ sublime/container.cpp
@@ -284,6 +284,9 @@ Container::Container(QWidget *parent)
     d->documentListButton = new QToolButton(this);
     d->documentListButton->setIcon(QIcon::fromTheme(QStringLiteral("format-list-unordered")));
     d->documentListButton->setMenu(d->documentListMenu);
+#ifdef Q_OS_MACOS
+    d->documentListMenu->setAsDockMenu();
+#endif
     d->documentListButton->setPopupMode(QToolButton::InstantPopup);
     d->documentListButton->setAutoRaise(true);
     d->documentListButton->setToolTip(i18n("Show sorted list of opened documents"));
