diff --git sublime/container.cpp sublime/container.cpp
index d991bb2be85e336ff5ea140dc8fd463822948cd6..b82d840609e611eaabbc4a01ee395d1ea8626683 100644
--- sublime/container.cpp
+++ sublime/container.cpp
@@ -151,6 +151,11 @@ struct ContainerPrivate {
     QToolButton* documentListButton;
     QMenu* documentListMenu;
     QHash<View*, QAction*> documentListActionForView;
+    bool m_isCocoa;
+
+    ContainerPrivate()
+        : m_isCocoa(QGuiApplication::platformName() == QStringLiteral("cocoa"))
+    {}
 
     /**
      * Updates the context menu which is shown when
@@ -209,7 +214,7 @@ struct ContainerPrivate {
     void setAsDockMenu()
     {
 #ifdef Q_OS_MACOS
-        if (documentListMenu != currentDockMenu) {
+        if (m_isCocoa && documentListMenu != currentDockMenu) {
             documentListMenu->setAsDockMenu();
             currentDockMenu = documentListMenu;
         }
@@ -219,7 +224,7 @@ struct ContainerPrivate {
     ~ContainerPrivate()
     {
 #ifdef Q_OS_MACOS
-        if (documentListMenu == currentDockMenu) {
+        if (m_isCocoa && documentListMenu == currentDockMenu) {
             QMenu().setAsDockMenu();
             currentDockMenu = nullptr;
         }

