diff --git plugins/projectmanagerview/projectmanagerview.cpp plugins/projectmanagerview/projectmanagerview.cpp
index db9072b2cad3f54659d85801a5cef57ecb4c0dbe..74b0625b3d1d31ff1fbdf8a369e95ecd7857bf3f 100644
--- plugins/projectmanagerview/projectmanagerview.cpp
+++ plugins/projectmanagerview/projectmanagerview.cpp
@@ -140,6 +140,7 @@ ProjectManagerView::ProjectManagerView( ProjectManagerViewPlugin* plugin, QWidge
     // Need to set this to get horizontal scrollbar. Also needs to be done after
     // the setModel call
     m_ui->projectTreeView->header()->setSectionResizeMode( QHeaderView::ResizeToContents );
+    m_ui->projectTreeView->header()->setStretchLastSection(false);
 }
 
 bool ProjectManagerView::eventFilter(QObject* obj, QEvent* event)
diff --git plugins/projectmanagerview/projectmanagerview.ui plugins/projectmanagerview/projectmanagerview.ui
index 3b2a78a8e9cfd611e7a937dcd21d66950576c778..9d8214ac22ae817090e5ff2e1868688c96c58eb5 100644
--- plugins/projectmanagerview/projectmanagerview.ui
+++ plugins/projectmanagerview/projectmanagerview.ui
@@ -17,7 +17,16 @@
    <string>Project Manager</string>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout">
-   <property name="margin">
+   <property name="leftMargin">
+    <number>0</number>
+   </property>
+   <property name="topMargin">
+    <number>0</number>
+   </property>
+   <property name="rightMargin">
+    <number>0</number>
+   </property>
+   <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
@@ -38,6 +47,12 @@
          <property name="whatsThis">
           <string>Project Overview</string>
          </property>
+         <property name="sizeAdjustPolicy">
+          <enum>QAbstractScrollArea::AdjustToContents</enum>
+         </property>
+         <property name="textElideMode">
+          <enum>Qt::ElideNone</enum>
+         </property>
          <attribute name="headerVisible">
           <bool>false</bool>
          </attribute>
diff --git plugins/projectmanagerview/projectmodelitemdelegate.cpp plugins/projectmanagerview/projectmodelitemdelegate.cpp
index 3f8c387e50acd620e64440b413bab212ed100593..fbacbcb03bf4a41506ab98e2cd2b275088665b6e 100644
--- plugins/projectmanagerview/projectmodelitemdelegate.cpp
+++ plugins/projectmanagerview/projectmodelitemdelegate.cpp
@@ -113,7 +113,7 @@ void ProjectModelItemDelegate::paint(QPainter* painter, const QStyleOptionViewIt
 void ProjectModelItemDelegate::drawBranchName(QPainter* painter, const QStyleOptionViewItem& option,
                                               const QRect& rect, const QString& branchName) const
 {
-    QString text = option.fontMetrics.elidedText(branchName, Qt::ElideRight, rect.width());
+    QString text = option.fontMetrics.elidedText(branchName, Qt::ElideNone, rect.width());
 
     bool selected = option.state & QStyle::State_Selected;
     QPalette::ColorGroup colorGroup = selected ? QPalette::Active : QPalette::Disabled;
@@ -156,5 +156,5 @@ void ProjectModelItemDelegate::drawDisplay(QPainter* painter, const QStyleOption
     }
 
     QFontMetrics fm(painter->fontMetrics());
-    painter->drawText(rect, fm.elidedText(text, Qt::ElideRight, rect.width()));
+    painter->drawText(rect, fm.elidedText(text, Qt::ElideNone, rect.width()));
 }
diff --git plugins/projectmanagerview/projecttreeview.cpp plugins/projectmanagerview/projecttreeview.cpp
index b9d15aefa4c8ad290f844a9f5e978e88a268f8c7..be6c0cce1cc7916cbef0803478198818ead8e57e 100644
--- plugins/projectmanagerview/projecttreeview.cpp
+++ plugins/projectmanagerview/projecttreeview.cpp
@@ -130,6 +130,8 @@ ProjectTreeView::ProjectTreeView( QWidget *parent )
     setAutoExpandDelay(300);
     setItemDelegate(new ProjectModelItemDelegate(this));
 
+    setTextElideMode(Qt::ElideNone);
+
     connect( this, &ProjectTreeView::customContextMenuRequested, this, &ProjectTreeView::popupContextMenu );
     connect( this, &ProjectTreeView::activated, this, &ProjectTreeView::slotActivated );
 
@@ -524,6 +526,12 @@ void ProjectTreeView::keyPressEvent(QKeyEvent* event)
         QTreeView::keyPressEvent(event);
 }
 
+void ProjectTreeView::resizeEvent(QResizeEvent *event)
+{
+    header()->setMinimumSectionSize(viewport()->width());
+    QTreeView::resizeEvent(event);
+}
+
 void ProjectTreeView::drawBranches(QPainter* painter, const QRect& rect, const QModelIndex& index) const
 {
     if (WidgetColorizer::colorizeByProject()) {
diff --git plugins/projectmanagerview/projecttreeview.h plugins/projectmanagerview/projecttreeview.h
index ae42ffb3822680f27cf2dd2f96db35d16742ad3d..dcdfe283cf6e93aecb0bc916c6971b06ace81b68 100644
--- plugins/projectmanagerview/projecttreeview.h
+++ plugins/projectmanagerview/projecttreeview.h
@@ -69,6 +69,7 @@ class ProjectTreeView: public QTreeView
 
     protected:
         void keyPressEvent(QKeyEvent *event) override;
+        void resizeEvent(QResizeEvent *event) override;
         void dropEvent(QDropEvent* event) override;
         void drawBranches(QPainter* painter, const QRect& rect, const QModelIndex& index) const override;
 
