diff --git language/duchain/types/typeregister.cpp language/duchain/types/typeregister.cpp
index 9de46e5ce564c7bb3798a321d179bbc311afe35b..5a29a1823dcb09096e1522c7f6d80fd609ffaaf4 100644
--- language/duchain/types/typeregister.cpp
+++ language/duchain/types/typeregister.cpp
@@ -61,7 +61,8 @@ bool TypeSystem::isFactoryLoaded(const AbstractTypeData& data) const
 bool TypeSystem::ensureFactoryLoaded(const AbstractTypeData& data) const
 {
   if (!m_factories.contains(data.typeClassId)) {
-    qCWarning(LANGUAGE) << "Factory for this type not loaded:" << data.typeClassId;
+    qCWarning(LANGUAGE) << "Factory for this type not loaded:" << data.typeClassId
+        << "classSize=" << data.classSize() << "itemSize=" << data.itemSize();
     Q_ASSERT(false);
     return false;
   }
@@ -82,7 +83,7 @@ TypeSystem& TypeSystem::self() {
 
 void TypeSystem::registerTypeClassInternal(AbstractTypeFactory* repo, uint dataClassSize, uint identity)
 {
-  qCDebug(LANGUAGE) << "Registering type class" << identity;
+//   qCDebug(LANGUAGE) << "Registering type class" << identity;
   Q_ASSERT(repo);
   Q_ASSERT(!m_factories.contains(identity));
   m_factories.insert(identity, repo);
