diff --git shell/textdocument.cpp shell/textdocument.cpp
index 4c6f6e4..e6ccf0c 100644
--- shell/textdocument.cpp
+++ shell/textdocument.cpp
@@ -265,7 +265,7 @@ bool TextDocument::isTextDocument() const
     {
         /// @todo Somehow it can happen that d->document is zero, which makes
         /// code relying on "isTextDocument() == (bool)textDocument()" crash
-        qWarning() << "Broken text-document: " << url();
+//         qWarning() << "Broken text-document: " << url();
         return false;
     }
 
