diff --git shell/textdocument.cpp shell/textdocument.cpp
index 4c6f6e4..e6ccf0c 100644
--- shell/textdocument.cpp
+++ shell/textdocument.cpp
@@ -265,7 +265,7 @@ bool TextDocument::isTextDocument() const
     {
         /// @todo Somehow it can happen that d->document is zero, which makes
         /// code relying on "isTextDocument() == (bool)textDocument()" crash
-        qWarning() << "Broken text-document: " << url();
+//         qWarning() << "Broken text-document: " << url();
         return false;
     }
 
diff --git a/shell/filteredproblemstore.cpp b/shell/filteredproblemstore.cpp
index 40962a6..e4b4a03 100644
--- a/shell/filteredproblemstore.cpp
+++ b/shell/filteredproblemstore.cpp
@@ -24,6 +24,8 @@
 
 #include <KLocalizedString>
 
+#include <QDebug>
+
 using namespace KDevelop;
 
 namespace
@@ -33,10 +35,14 @@ namespace
 void addDiagnostics(ProblemStoreNode *node, const QVector<IProblem::Ptr> &diagnostics)
 {
     foreach (const IProblem::Ptr &ptr, diagnostics) {
-        ProblemNode *child = new ProblemNode(node, ptr);
-        node->addChild(child);
+        if (ptr.data()) {
+            ProblemNode *child = new ProblemNode(node, ptr);
+            node->addChild(child);
 
-        addDiagnostics(child, ptr->diagnostics());
+            addDiagnostics(child, ptr->diagnostics());
+        } else {
+            qWarning() << Q_FUNC_INFO << "caught NULL ptr in" << diagnostics;
+        }
     }
 }
 
