diff --git source/orig.CMakeLists.txt source/CMakeLists.txt
index 84d43d2..87cb6e5 100755
--- source/orig.CMakeLists.txt
+++ source/CMakeLists.txt
@@ -249,10 +249,8 @@ if(GCC)
         if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS "4.8")
             message(FATAL_ERROR "gcc version above 4.8 required to support hdr10plus")
         endif()
-        add_definitions(-std=gnu++11)
-    else()
-        add_definitions(-std=gnu++98)
     endif()
+    add_definitions(-std=gnu++11)
     if(ENABLE_PIC)
          add_definitions(-fPIC)
     endif(ENABLE_PIC)
