diff --git autotests/test_settings.cpp autotests/test_settings.cpp
index f2bc80c..5c2b3c3 100644
--- autotests/test_settings.cpp
+++ autotests/test_settings.cpp
@@ -36,7 +36,7 @@ using namespace Sonnet;
 
 void SonnetSettingsTest::testRestoreDoesNotSave()
 {
-    QSettings settings(QStringLiteral("KDE"), QStringLiteral("Sonnet"));
+    QSettings settings(QSettings::IniFormat, QSettings::UserScope, QStringLiteral("KDE"), QStringLiteral("Sonnet"));
     QString fileName = settings.fileName();
 
     QDateTime startTime = QFileInfo(fileName).lastModified();
@@ -71,7 +71,7 @@ void SonnetSettingsTest::testRestoreDoesNotSave()
 
 void SonnetSettingsTest::testSpellerAPIChangeSaves()
 {
-    QSettings settings(QStringLiteral("KDE"), QStringLiteral("Sonnet"));
+    QSettings settings(QSettings::IniFormat, QSettings::UserScope, QStringLiteral("KDE"), QStringLiteral("Sonnet"));
     Speller speller;
 
     // Default Language
diff --git src/core/settings.cpp src/core/settings.cpp
index 8614e35..15a4dad 100644
--- src/core/settings.cpp
+++ src/core/settings.cpp
@@ -229,7 +229,7 @@ int Settings::disableWordErrorCount() const
 
 void Settings::save()
 {
-    QSettings settings(QStringLiteral("KDE"), QStringLiteral("Sonnet"));
+    QSettings settings(QSettings::IniFormat, QSettings::UserScope, QStringLiteral("KDE"), QStringLiteral("Sonnet"));
     settings.setValue(QStringLiteral("defaultClient"), d->defaultClient);
     settings.setValue(QStringLiteral("defaultLanguage"), d->defaultLanguage);
     settings.setValue(QStringLiteral("checkUppercase"), d->checkUppercase);
@@ -271,7 +271,7 @@ static QStringList kdeWords()
 
 void Settings::restore()
 {
-    QSettings settings(QStringLiteral("KDE"), QStringLiteral("Sonnet"));
+    QSettings settings(QSettings::IniFormat, QSettings::UserScope, QStringLiteral("KDE"), QStringLiteral("Sonnet"));
     d->defaultClient = settings.value(QStringLiteral("defaultClient"), QString()).toString();
     d->defaultLanguage = settings.value(QStringLiteral("defaultLanguage"), QLocale::system().bcp47Name()).toString();
 
diff --git src/plugins/hspell/hspelldict.cpp src/plugins/hspell/hspelldict.cpp
index ccd8acb..9f1b945 100644
--- src/plugins/hspell/hspelldict.cpp
+++ src/plugins/hspell/hspelldict.cpp
@@ -42,7 +42,7 @@ HSpellDict::HSpellDict(const QString &lang)
         initialized = true;
     }
 
-    QSettings settings(QStringLiteral("KDE"), QStringLiteral("SonnetHSpellPlugin"));
+    QSettings settings(QSettings::IniFormat, QSettings::UserScope, QStringLiteral("KDE"), QStringLiteral("SonnetHSpellPlugin"));
     m_personalWords = QSet<QString>::fromList(settings.value(QStringLiteral("PersonalWords"), QStringList()).toStringList());
     QVariantHash replacementMap = settings.value(QStringLiteral("Replacements"), QVariant()).toHash();
     Q_FOREACH(const QString &key, replacementMap.keys()) {
@@ -132,7 +132,7 @@ bool HSpellDict::addToSession(const QString &word)
 
 void HSpellDict::storePersonalWords()
 {
-    QSettings settings(QStringLiteral("KDE"), QStringLiteral("SonnetHSpellPlugin"));
+    QSettings settings(QSettings::IniFormat, QSettings::UserScope, QStringLiteral("KDE"), QStringLiteral("SonnetHSpellPlugin"));
     settings.setValue(QStringLiteral("PersonalWords"), QVariant(m_personalWords.toList()));
     QVariantHash variantHash;
     Q_FOREACH(const QString &key, m_replacements.keys()) {
