diff --git a/CMakeLists.txt b/CMakeLists.txt
index d3bbe54..3fbd8b2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,7 +6,7 @@ set(LIBKDAV_VERSION "5.4.40")
 
 configure_file(libkdav-version.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/libkdav-version.h @ONLY)
 
-set(KF5_VERSION "5.31.0")
+set(KF5_VERSION "5.29.0")
 find_package(ECM ${KF5_VERSION} REQUIRED NO_MODULE)
 set(CMAKE_MODULE_PATH ${ECM_MODULE_PATH})
 
