--- orig.configure	2020-01-17 14:44:55.000000000 +0100
+++ configure	2020-06-04 14:04:06.776980693 +0200
@@ -24586,8 +24586,9 @@
   unknown*)
     { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: Could not determine float format." >&5
 $as_echo "$as_me: WARNING: Could not determine float format." >&2;}
-    { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: Conversions to and from \"double\" may be slow." >&5
-$as_echo "$as_me: WARNING: Conversions to and from \"double\" may be slow." >&2;}
+    { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: assuming IEEE little endian (Intel/x86)." >&5
+$as_echo "$as_me: WARNING: assuming IEEE little endian (Intel/x86)" >&2;}
+    $as_echo "#define HAVE_DOUBLE_IEEE_LITTLE_ENDIAN 1" >>confdefs.h
     ;;
   *)
     { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: oops, unrecognised float format: $gmp_cv_c_double_format" >&5
