diff --git a/configure b/configure
index c20027d7a319b877e802f94b70d0d663b0a7f96b..5794d152b2b40c051e9c0441fad47c268c73ac75 100755
--- a/configure
+++ b/configure
@@ -7209,7 +6381,7 @@ else
     # This has to be set to something or mtree won't create any directories.
     # It doesn't matter what it's set to for non-root though, as it can't
     # change the ownership anyway.
-    MTREE_GNAME="admin"
+    MTREE_GNAME="adm"
 fi
 
 
diff --git a/configure.ac b/configure.ac
index ab77f3746..3b8893a7f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -162,7 +162,7 @@ AC_PATH_PROG(MAN, [man])
 AC_PATH_PROG(MDFIND, [mdfind], [])
 AC_PATH_PROG(MDLS, [mdls], [])
 AC_PATH_PROG(MKBOM, [mkbom], [])
-AC_PATH_PROGS(MTREE, [mtree fmtree], [], [$PATH:/usr/sbin])
+AC_PATH_PROGS(MTREE, [mtree fmtree], [], [$PATH:/usr/sbin:/usr/local/bin])
 AC_PATH_PROG(OPEN, [open], [])
 AC_PATH_PROG(OPENSSL, [openssl], [])
 AC_PATH_PROG(PATCH, [patch], [])
@@ -181,6 +181,9 @@ AC_PATH_PROG(XZ, [xz], [])
 AC_PATH_PROG(ZIP, [zip])
 
 if test "x$MTREE" = "x"; then
+    AC_MSG_RESULT([The mtree or fmtree command is required. Use --with-mtree=/path/to/mtree \
+to indicate a non-standard location. Or install it, on Debian/Ubuntu via package freebsd-buildutils \
+or else from e.g. https://github.com/archiecobbs/mtree-port?tab=readme-ov-file])
     AC_MSG_ERROR([mtree not found])
 fi
 
@@ -236,7 +239,7 @@ else
     # This has to be set to something or mtree won't create any directories.
     # It doesn't matter what it's set to for non-root though, as it can't
     # change the ownership anyway.
-    MTREE_GNAME="admin"
+    MTREE_GNAME="adm"
 fi
 AC_SUBST(MTREE_GNAME)
 

