commit d81132d5b79479f11dda807f7414bfedf0a31501
Author: René Bertin <rjvbertin@gmail.com>
Date:   Wed May 29 13:37:42 2024 +0200

    lamarck/patch-aclocalm4-libcurl.diff

diff --git a/aclocal.m4 b/aclocal.m4
index cc92d20..c27f168 100644
--- a/aclocal.m4
+++ b/aclocal.m4
@@ -887,15 +887,7 @@ AC_DEFUN([MP_LIBCURL_FLAGS],[
 	fi
 
 	CFLAGS_LIBCURL=$($CURL_CONFIG --cflags)
-	if test "x$curlprefix" = "x"; then
-		# System curl-config emits absurd output for --libs
-		# See rdar://7244457
-		LDFLAGS_LIBCURL="-lcurl"
-	else
-		# Due to a bug in dist, --arch flags are improperly supplied by curl-config.
-		# Get rid of them.
-		LDFLAGS_LIBCURL=$($CURL_CONFIG --libs | [sed 's/-arch [A-Za-z0-9_]* //g'])
-	fi
+	LDFLAGS_LIBCURL=$($CURL_CONFIG --libs)
 
 	AC_SUBST(CFLAGS_LIBCURL)
 	AC_SUBST(LDFLAGS_LIBCURL)
