--- parsers.py.orig	2015-03-11 00:59:40.000000000 +0100
+++ parsers.py	2015-03-12 21:28:50.000000000 +0100
@@ -478,7 +478,7 @@
                     itm = VideoItem(cat, it)
                     vids[it['ID']] = itm
 
-        logger.info('category {0} has {1} videos'.format(cat, len(vids)))
+        logger.info(u'category {0} has {1} videos'.format(cat, len(vids)))
         self.alw.videos[cat] = vids
         self.loadingCategoryFinished.emit(CATEGORIES[self.lang][cat])
 
--- ui_main.py.orig	2015-03-11 00:59:40.000000000 +0100
+++ ui_main.py	2015-03-12 21:32:50.000000000 +0100
@@ -708,7 +708,7 @@
 
     def enable_button(self, btn):
         for k in self.buttons.keys():
-            if btn in self.buttons[k]:
+            if unicode(btn) in unicode(self.buttons[k]):
                 k.setEnabled(True)
 
     def set_page_button_style(self):
@@ -718,7 +718,7 @@
 
     def set_toggled_button(self, cat):
         for k in self.buttons.keys():
-            if cat in self.buttons[k]:
+            if unicode(cat) in unicode(self.buttons[k]):
                 k.set_current(True)
             else:
                 k.set_current(False)
--- artePlus.py.orig	2015-03-11 00:59:40.000000000 +0100
+++ artePlus.py	2015-03-14 00:08:43.000000000 +0100
@@ -658,7 +658,10 @@
             except:
                 pass
 
-        summary = self.videos[idx].summary.decode('utf-8', 'replace')
+        try:
+            summary = self.videos[idx].summary.decode('utf-8', 'replace')
+        except:
+            summary = self.videos[idx].summary
         font = QtGui.QFont("sans", 10)
         c_form = QtGui.QTextCharFormat()
         c_form.setFont(font)
