--- parsers.py.orig	2015-03-11 00:59:40.000000000 +0100
+++ parsers.py	2015-03-12 21:28:50.000000000 +0100
@@ -478,7 +478,7 @@
                     itm = VideoItem(cat, it)
                     vids[it['ID']] = itm
 
-        logger.info('category {0} has {1} videos'.format(cat, len(vids)))
+        logger.info(u'category {0} has {1} videos'.format(cat, len(vids)))
         self.alw.videos[cat] = vids
         self.loadingCategoryFinished.emit(CATEGORIES[self.lang][cat])
 
--- ui_main.py.orig	2015-03-11 00:59:40.000000000 +0100
+++ ui_main.py	2015-03-12 21:32:50.000000000 +0100
@@ -708,7 +708,7 @@
 
     def enable_button(self, btn):
         for k in self.buttons.keys():
-            if btn in self.buttons[k]:
+            if unicode(btn) in unicode(self.buttons[k]):
                 k.setEnabled(True)
 
     def set_page_button_style(self):
@@ -718,7 +718,7 @@
 
     def set_toggled_button(self, cat):
         for k in self.buttons.keys():
-            if cat in self.buttons[k]:
+            if unicode(cat) in unicode(self.buttons[k]):
                 k.set_current(True)
             else:
                 k.set_current(False)
--- artePlus.py.orig	2015-03-11 00:59:40.000000000 +0100
+++ artePlus.py	2015-03-14 00:08:43.000000000 +0100
@@ -658,7 +658,10 @@
             except:
                 pass
 
-        summary = self.videos[idx].summary.decode('utf-8', 'replace')
+        try:
+            summary = self.videos[idx].summary.decode('utf-8', 'replace')
+        except:
+            summary = self.videos[idx].summary
         font = QtGui.QFont("sans", 10)
         c_form = QtGui.QTextCharFormat()
         c_form.setFont(font)
diff --git arteLiveExtra.py arteLiveExtra.py
index 831e1fe..f982017 100644
--- arteLiveExtra.py
+++ arteLiveExtra.py
@@ -32,6 +32,7 @@ from threading import enumerate as enumerate_
 from PyQt4 import QtCore, QtGui
 
 import data
+import codecs
 
 from parsers import ExtraParser, VideoItem
 from loaders import LiveLoader, RateCounter
@@ -624,8 +625,8 @@ class ArteLiveExtra(object):
         p += u"\n\n"
 
         try:
-            with open(f, "a") as objf:
-                objf.write(p.encode('utf8', 'ignore'))
+            with codecs.open(f, "a", "utf-8") as objf:
+                objf.write(p)
         except Exception, why:
             self.main.show_warning(4, why)
 
diff --git arteLiveWeb.py arteLiveWeb.py
index eaf8eb9..8e3f119 100644
--- arteLiveWeb.py
+++ arteLiveWeb.py
@@ -17,6 +17,7 @@ import shutil
 import sys
 import pickle
 import logging
+import codecs
 logger = logging.getLogger(__name__)
 
 try:
@@ -680,8 +681,8 @@ class ArteLiveWeb(object):
         p += u"\n\n"
 
         try:
-            with open(f, "a") as objf:
-                objf.write(p.encode('utf8', 'ignore'))
+            with codecs.open(f, "a", "utf-8") as objf:
+                objf.write(p)
         except Exception, why:
             self.main.show_warning(4, why)
 
diff --git artePlus.py artePlus.py
index fe675d6..c028c47 100644
--- artePlus.py
+++ artePlus.py
@@ -20,6 +20,7 @@ import urllib2
 import re
 import gettext
 import logging
+import codecs
 logger = logging.getLogger(__name__)
 
 try:
@@ -377,8 +378,8 @@ class ArtePlus(object):
             f = self.fname[:-4]
 
         try:
-            with open(f, "a") as objf:
-                objf.write(movie.summary.decode('utf-8', 'replace'))
+            with codecs.open(f, "a", "utf-8") as objf:
+                objf.write(movie.summary)
         except Exception, why:
             self.main.show_warning(4, why)
 
