=== modified file 'arteconcert.py'
--- old/arteconcert.py	2018-09-21 05:44:22 +0000
+++ new/arteconcert.py	2018-12-18 09:09:34 +0000
@@ -1006,7 +1006,7 @@
     def __init__(self):
         self.ui = None
         self.cfg = {'lang': 'fr'}
-        ws = os.path.expanduser("~") + "/qarte-3/wspace/livePreviews"
+        ws = os.path.expanduser("~") + "/.Qarte/livePreviews"
         self.workspace = {'live_tmp': ws}
 
         self.arteconcert = ArteConcert(self)

=== modified file 'loadingscheduler.py'
--- old/loadingscheduler.py	2018-02-22 09:10:04 +0000
+++ new/loadingscheduler.py	2018-12-18 09:10:32 +0000
@@ -291,7 +291,7 @@
 class Core:
     def __init__(self):
         self.ui = None
-        ws = os.path.expanduser("~") + "/qarte-3/wspace"
+        ws = os.path.expanduser("~") + "/.Qarte"
         self.workspace = {'user': ws}
         self.cfg = {'lang': 'fr',
                     'crontab_cmd': 'export DISPLAY=:0 & LC_CTYPE="<lang>" Lang="<lang>"'\

