=== modified file 'core.py'
--- core.py	2017-02-09 08:23:48 +0000
+++ core.py	2017-02-27 12:45:43 +0000
@@ -48,6 +48,14 @@
 if LANG not in ['fr', 'de']:
     LANG = 'fr'
 
+try:
+    # ensure we're using UTF-8 encoding
+    locale.setlocale(locale.LC_ALL, (LOC, "utf-8"))
+except:
+    import traceback
+    (type,value,tb) = sys.exc_info()
+    lgg.warn("Error setting locale: %s" %(value))
+    traceback.print_tb(tb)
 
 class Core:
     """Define the main class of Qarte.

