=== modified file 'arteconcert.py'
--- arteconcert.py.old	2017-02-17 09:56:39 +0000
+++ arteconcert.py	2017-02-22 15:16:34 +0000
@@ -556,7 +556,10 @@
         """
         self.category = cat
         self.order = item['id']
-        self.title = item['title']
+        try:
+            self.title = item['title']
+        except:
+            self.title = ""
         self.date = item['date']
         self.expires = item['expires']
         self.duration = item['duration']

