=== modified file 'core.py'
--- old/core.py	2018-07-30 10:52:52 +0000
+++ new/core.py	2018-12-18 09:19:09 +0000
@@ -52,6 +52,14 @@
 if LANG not in ['fr', 'de']:
     LANG = 'fr'
 
+try:
+    # ensure we're using UTF-8 encoding
+    locale.setlocale(locale.LC_ALL, (LOC, "utf-8"))
+except:
+    import traceback
+    (type,value,tb) = sys.exc_info()
+    lgg.warn("Error setting locale: %s" %(value))
+    traceback.print_tb(tb)
 
 class Core:
     """Define the main class of Qarte.

