=== modified file 'artetv.py'
--- old/artetv.py	2018-08-16 07:04:44 +0000
+++ new/artetv.py	2018-12-18 15:16:01 +0000
@@ -32,7 +32,7 @@
 from PyQt5.QtGui import QFont, QTextCharFormat
 from PyQt5.QtCore import QObject, QCoreApplication, pyqtSignal, QLocale, QSize
 
-from data import ARTE_TV
+from data import ARTE_TV, ARTE_TV_BASE_URL
 from tvdownloadingconfig import TvDownloadingConfig
 from loader import Loader
 from gui.warning import WarningBox
@@ -87,7 +87,7 @@
         """
         self.load_summaries_index()
         links = self.get_site_pages()
-        self.parser = Parser(ARTE_TV.replace('fr', self.lang))
+        self.parser = Parser(ARTE_TV.replace('fr', self.lang), self.core)
         Thread(target=self.run_parser).start()
 
     def run_parser(self):
@@ -99,7 +99,7 @@
         videos = self.parser.videos
         self.videos = [TVItem(item) for item in videos]
         self.indexes = self.parser.indexes
-        for i in range(10):
+        for i in range(self.core.cfg['day_range']):
             vids, idxs = self.parser.get_next_page()
             self.sort_videos(vids, idxs)
 
@@ -420,6 +420,10 @@
         """Returns the list of the arte TV Guide web pages.
 
         """
+        #atv = [ARTE_TV_BASE_URL + "?sort=recent&country=FR"]
+        #days = self.core.cfg['day_range']
+        #for i in range(1,days):
+            #atv.append(ARTE_TV_BASE_URL + "?date=j-%s&country=FR" %i)
         if self.lang == 'fr':
             return ARTE_TV
 
@@ -857,11 +861,13 @@
             lgg.info("Thumbnail saved")
 
 class Parser:
-    def __init__(self, url):
+    def __init__(self, url, core):
         self.url = url
+        self.core = core
         self.videos = []
         self.indexes = []
-        diff = timedelta(days=10)
+        # RJVB: this would be the new "day_range" ?
+        diff = timedelta(days=self.core.cfg['day_range'])
         last = date.today() - diff
         #now = last.strftime("%y-%m-%d")
         now = last.strftime("%Y%m%d")
@@ -885,6 +891,10 @@
                         if video:
                             self.videos.insert(0, video)
                             self.indexes.insert(0, video['id'])
+                else:
+                    lgg.warning(page + " returned no content")
+            else:
+                lgg.warning(page + " returned no json content")
 
 
     def get_day_before(self, start):

=== modified file 'core.py'
--- old/core.py	2018-07-30 10:52:52 +0000
+++ new/core.py	2018-12-18 09:23:13 +0000
@@ -110,8 +110,11 @@
 
         cfg['version'] = VERSION
         cfg['live_quality'] = 'HTTP_EQ_1'
+        from data import DEFAULT_PREFS as defaults
         if not 'add_date' in cfg:
-            cfg['add_date'] = False
+            cfg['add_date'] = defaults['add_date']
+        if not 'day_range' in cfg:
+            cfg['day_range'] = defaults['day_range']
         return cfg
 
     def create_config(self):

=== modified file 'data.py'
--- old/data.py	2018-09-21 06:18:57 +0000
+++ new/data.py	2018-12-18 09:32:52 +0000
@@ -22,6 +22,7 @@
                 'load_pages': [True, False],
                 'videos_folder': None, 
                 'music_folder': None,
+                'day_range': 14,
                 'add_date': False,
                 'pitch_plus': True,
                 'pitch_plus_unique': 1, 
@@ -47,6 +48,8 @@
                 'filter_keys': []}
 
 ARTE_TV = "https://www.arte.tv/fr/guide/DAY/"
+# obsolete:
+ARTE_TV_BASE_URL = "http://www.arte.tv/guide/fr/plus7/toutes-les-videos"
 
 DAYS_FR = ['Dimanche', 'Lundi', 'Mardi', 'Mercredi', 'Jeudi',
                     'Vendredi', 'Samedi', 'Tous les jours']

=== modified file 'settings.py'
--- old/settings.py	2018-05-16 06:10:46 +0000
+++ new/settings.py	2018-12-18 09:23:13 +0000
@@ -20,7 +20,7 @@
 from PyQt5.QtWidgets import (QDialog, QGridLayout, QVBoxLayout, QHBoxLayout,
                              QLabel, QComboBox, QCheckBox, QSpacerItem,
                              QSizePolicy, QApplication, QLineEdit, QPushButton,
-                             QFrame, QFileDialog, )
+                             QFrame, QFileDialog, QSpinBox, )
 
 from data import VERSION_STR
 
@@ -72,6 +72,16 @@
         self.warning_lbl.hide()
         hl5.addWidget(self.warning_lbl)
         vl.addLayout(hl5)
+        hl16 = QHBoxLayout()
+        self.days_lbl = QLabel(self)
+        self.days_lbl.setText(_("Period to scan for videos (days):"))
+        self.days_spin = QSpinBox(self)
+        self.days_spin.setMinimum(0)
+        sp6 = QSpacerItem(40, 13, QSizePolicy.Expanding, QSizePolicy.Minimum)
+        hl16.addWidget(self.days_lbl)
+        hl16.addWidget(self.days_spin)
+        hl16.addItem(sp6)
+        vl.addLayout(hl16)
         hl15 = QHBoxLayout()
         self.add_date_chb = QCheckBox(self)
         self.add_date_chb.setText(_("Add the date to the file name"))
@@ -169,6 +179,7 @@
             self.summary_chb.setEnabled(True)
         if cfg['pitch_live']:
             self.summary1_chb.setEnabled(True)
+        self.days_spin.setValue(cfg['day_range'])
 
         self.folder_led.editingFinished.connect(self.on_videos_fld_changed)
         self.folder1_led.editingFinished.connect(self.on_music_fld_changed)
@@ -255,6 +266,7 @@
         self.cfg['pitch_live'] = self.summary1_chb.isChecked()
         self.cfg['pitch_live_unique'] = self.summary1_cmb.currentIndex()
         self.cfg['copythumb_live'] = self.copy_thumbs1_chb.isChecked()
+        self.cfg['day_range'] = self.days_spin.value()
         self.accept()
 
 

