=== modified file 'gui/basket.py'
--- old/gui/basket.py	2018-02-22 09:10:04 +0000
+++ new/gui/basket.py	2018-12-18 13:50:17 +0000
@@ -172,8 +172,11 @@
 
     def update_buttons(self):
         for r in range(self.rowCount()):
-            self.cellWidget(r, 2).row_ = r
-            self.cellWidget(r, 3).row_ = r
+            try:
+                self.cellWidget(r, 2).row_ = r
+                self.cellWidget(r, 3).row_ = r
+            except:
+                pass
 
     def config_download(self, idx):
         self.driver.configure_downloading(self.lst_movies[idx])


