=== modified file 'arteconcert.py'
--- arteconcert.py	2016-06-12 13:17:40 +0000
+++ arteconcert.py	2016-07-05 21:44:03 +0000
@@ -690,7 +690,7 @@
     def __init__(self):
         self.ui = None
         self.cfg = {'lang': 'fr'}
-        self.workspace = {'live_tmp': '/home/vincent/qarte-3/wspace/livePreviews'}
+        self.workspace = {'live_tmp': os.path.join(os.path.expanduser("~"), ".Qarte/livePreviews")}
 
         self.arteconcert = ArteConcert(self)
         self.arteconcert.config_parser()

=== modified file 'core.py'
--- core.py	2016-07-03 07:14:30 +0000
+++ core.py	2016-07-05 21:43:10 +0000
@@ -175,11 +175,11 @@
         dict(paths)
         """
         lgg.info('Set workspace')
-        #user = os.path.expanduser("~")
-        user = '/home/vincent/qarte-3/wspace'
+        user = os.path.expanduser("~")
+        #user = '/home/vincent/qarte-3/wspace'
         paths = {}
-        #paths['user'] = os.path.join(user, ".Qarte")
-        paths['user'] = user
+        paths['user'] = os.path.join(user, ".Qarte")
+        #paths['user'] = user
         paths['config'] = os.path.join(paths['user'], "user_config")
         paths['live_tmp'] = os.path.join(paths['user'], "livePreviews")
         paths['tv_tmp'] = os.path.join(paths['user'], "plusPreviews")

@@ -261,7 +261,7 @@
         lgg.info('Reload arte Live Web')
         # Language may be changed
         self.artelive.lang = self.cfg['lang']
-        self.arteliv.config_parser()
+        self.artelive.config_parser()
 
     def display_about_dialog(self):
         lgg.info('Call About dialog box')

=== modified file 'loadingscheduler.py'
--- loadingscheduler.py	2016-06-12 13:17:40 +0000
+++ loadingscheduler.py	2016-07-05 21:44:34 +0000
@@ -288,7 +288,8 @@
 class Core:
     def __init__(self):
         self.ui = None
-        self.workspace = {'user': '/home/vincent/qarte-3/wspace'}
+        #self.workspace = {'user': '/home/vincent/qarte-3/wspace'}
+        self.workspace = {'user': os.path.join(os.path.expanduser("~"), ".Qarte")}
         self.cfg = {'lang': 'fr',
                     'crontab_cmd': 'export DISPLAY=:0 & LC_CTYPE="<lang>" Lang="<lang>"'\
                                    ' <root> -a <ID> >> "<log>" 2>&1'}

