=== modified file 'arteconcert.py'
--- arteconcert.py	2016-10-07 14:09:33 +0000
+++ arteconcert.py	2016-11-09 18:12:42 +0000
@@ -13,6 +13,7 @@
 import os
 import sys
 import shutil
+import codecs
 import time
 import pickle
 import json
@@ -85,7 +86,7 @@
         cnt = self.get_update()
         if cnt:
             lgg.info('Concert list updated')
-            with open(fpath, 'w') as outf:
+            with codecs.open(fpath, 'w', "utf-8") as outf:
                 outf.write(cnt)
 
     def get_update(self):
@@ -103,7 +104,7 @@
         fpath = self.core.workspace['concerts_%s' % self.lang]
         if os.path.isfile(fpath):
             try:
-                with open(fpath, 'r') as objfile:
+                with codecs.open(fpath, 'r', "utf-8") as objfile:
                     cnt = objfile.read()
                     self.concerts =  json.loads(cnt)
             except Exception as why:
@@ -470,7 +471,7 @@
         summary = self.get_description(video)
         txt = "\n* %s\n  %s\n" %(title, summary)
         try:
-            with open(path, 'a') as outf:
+            with codecs.open(path, 'a', "utf-8") as outf:
                 outf.write(txt)
                 lgg.info("Summary saved")
         except Exception as why:

=== modified file 'artetv.py'
--- artetv.py	2016-10-26 07:38:00 +0000
+++ artetv.py	2016-11-09 18:57:46 +0000
@@ -13,6 +13,7 @@
 import os
 import sys
 import shutil
+import codecs
 import time
 import pickle
 import json
@@ -370,7 +371,7 @@
         self.summaries = {}
         path = "%s_%s" %(self.core.workspace['tv_summaries'], self.lang)
         if os.path.isfile(path):
-            with open(path, 'r') as inf:
+            with codecs.open(path, 'r', "utf-8") as inf:
                 try:
                     self.summaries = json.loads(inf.read())
                 except Exception as why:
@@ -392,7 +393,7 @@
         txt = ";".join(indexes)
         path = "%s_%s" %(self.core.workspace['tv_index'], self.lang)
         try:
-            with open(path, 'w') as outf:
+            with codecs.open(path, 'w', "utf-8") as outf:
                 outf.write(txt)
         except Exception as why:
             lgg.warning("Can't write indexes list")
@@ -410,7 +411,7 @@
     def read_video_indexes(self):
         path = "%s_%s" %(self.core.workspace['tv_index'], self.lang)
         if os.path.isfile(path):
-            with open(path, 'r') as inf:
+            with codecs.open(path, 'r', "utf-8") as inf:
                 return inf.readline().split(';')
 
         return ["0"]
@@ -789,7 +790,7 @@
         summary = self.get_description(video)
         txt = "\n* %s\n  %s\n" %(title, summary)
         try:
-            with open(path, 'a') as outf:
+            with codecs.open(path, 'a', "utf-8") as outf:
                 outf.write(txt)
                 lgg.info("Summary saved")
         except Exception as why:

