=== modified file 'arteconcert.py'
--- old/arteconcert.py	2018-09-21 05:44:22 +0000
+++ new/arteconcert.py	2018-12-18 09:22:34 +0000
@@ -916,7 +916,10 @@
         """
         self.category = cat
         self.order = item["programId"]
-        self.title = item["title"]
+        try:
+            self.title = item["title"]
+        except:
+            self.title = ""
         self.date = ""
         self.expires = ""
         self.desc = item["description"]

