--- src/libaudcore/orig.mainloop.cc	2018-01-14 14:55:25.000000000 +0100
+++ src/libaudcore/mainloop.cc	2018-01-14 15:33:57.000000000 +0100
@@ -184,13 +184,27 @@
     }
 
     void stop ()
-        { deleteLater (); }
+    {
+        if (timerId)
+        {
+            killTimer(timerId);
+            timerId = 0;
+        }
+        deleteLater ();
+    }
 
 protected:
     void customEvent (QEvent * event)
         { startTimer (params.interval_ms); }
     void timerEvent (QTimerEvent * event)
-        { run (); }
+    {
+        if (event->timerId() == timerId)
+        {
+            run ();
+        }
+    }
+private:
+    int timerId = 0;
 };
 
 #endif // USE_QT
