--- src/libaudcore/orig.playback.cc	2018-01-14 15:58:09.000000000 +0100
+++ src/libaudcore/playback.cc	2018-01-14 16:11:57.000000000 +0100
@@ -52,6 +52,10 @@
 #include "plugins-internal.h"
 #include "runtime.h"
 
+#ifdef USE_QT
+#include <QThread>
+#endif // USE_QT
+
 struct PlaybackState {
     bool playing = false;
     bool thread_running = false;
@@ -425,6 +429,17 @@
     return nullptr;
 }
 
+#ifdef USE_QT
+class PlaybackThread : public QThread
+{
+    void run() override
+    {
+        playback_thread(nullptr);
+        deleteLater();
+    }
+};
+#endif // USE_QT
+
 // main thread: starts playback of a new song
 void playback_play (int seek_time, bool pause)
 {
@@ -444,9 +459,15 @@
         pthread_cond_broadcast (& cond);
     else
     {
+#ifndef USE_QT
         pthread_t thread;
         pthread_create (& thread, nullptr, playback_thread, nullptr);
         pthread_detach (thread);
+#else // USE_QT
+        PlaybackThread *thread = new PlaybackThread();
+        thread->setObjectName(QStringLiteral("Playback Thread"));
+        thread->start();
+#endif // USE_QT
         pb_state.thread_running = true;
     }
 
