--- src/libaudqt/orig.util.cc	2016-09-22 03:08:54.000000000 +0200
+++ src/libaudqt/util.cc	2018-01-13 19:19:55.000000000 +0100
@@ -44,7 +44,7 @@
     static char * dummy_argv[] = {app_name, nullptr};
 
     qapp = new QApplication (dummy_argc, dummy_argv);
-    atexit ([] () { delete qapp; });
+//     atexit ([] () { delete qapp; });
 
     qapp->setAttribute (Qt::AA_UseHighDpiPixmaps);
 #if QT_VERSION >= QT_VERSION_CHECK(5, 3, 0)
