--- dbus/orig.dbus-sysdeps-unix.c	2016-09-15 22:15:03.000000000 +0200
+++ dbus/dbus-sysdeps-unix.c	2016-09-19 00:12:14.000000000 +0200
@@ -3836,6 +3836,13 @@
 
   _DBUS_ASSERT_ERROR_IS_CLEAR (error);
 
+  const char *c = getenv (launchd_env_var);
+  if (c && *c)
+    {
+      _dbus_string_append (socket_path, c);
+      return TRUE;
+    }
+
   if (_dbus_check_setuid ())
     {
       dbus_set_error_const (error, DBUS_ERROR_NOT_SUPPORTED,
@@ -3883,12 +3890,12 @@
   dbus_bool_t valid_socket;
   DBusString socket_path;
 
-  if (_dbus_check_setuid ())
-    {
-      dbus_set_error_const (error, DBUS_ERROR_NOT_SUPPORTED,
-                            "Unable to find launchd socket when setuid");
-      return FALSE;
-    }
+//   if (_dbus_check_setuid ())
+//     {
+//       dbus_set_error_const (error, DBUS_ERROR_NOT_SUPPORTED,
+//                             "Unable to find launchd socket when setuid");
+//       return FALSE;
+//     }
 
   if (!_dbus_string_init (&socket_path))
     {
--- bus/orig.activation-helper.c	2016-09-19 00:05:00.000000000 +0200
+++ bus/activation-helper.c	2016-09-18 22:54:28.000000000 +0200
@@ -159,6 +159,9 @@
   /* Ensure the bus is set to system */
   dbus_setenv ("DBUS_STARTER_ADDRESS", DBUS_SYSTEM_BUS_DEFAULT_ADDRESS);
   dbus_setenv ("DBUS_STARTER_BUS_TYPE", "system");
+#ifdef __APPLE__
+  dbus_setenv ("DBUS_LAUNCHD_SESSION_BUS_SOCKET", DBUS_SYSTEM_BUS_DEFAULT_ADDRESS);
+#endif
 #endif
 
   return TRUE;
