diff --git tools/dbus-launch.c tools/dbus-launch.c
index 74e396919868741672ef36e4083e3a9d54e6d80f..c158e180d24bd133688419cb2740a3a70f3c0e4f 100644
--- tools/dbus-launch.c
+++ tools/dbus-launch.c
@@ -559,7 +559,7 @@ kill_bus_when_session_ends (void)
 
       if (got_fatal_signal)
         {
-          verbose ("Got fatal signal %d, killing dbus-daemon\n",
+          verbose ("Got fatal signal %d, killing dbus-daemon@SUFFIX@\n",
                    got_fatal_signal);
           kill_bus_and_exit (0);
         }
@@ -1078,7 +1078,7 @@ main (int argc, char **argv)
 
       if (address != NULL)
 	{
-	  verbose ("dbus-daemon is already running. Returning existing parameters.\n");
+	  verbose ("dbus-daemon@SUFFIX@ is already running. Returning existing parameters.\n");
 	  pass_info (runprog, address, pid, wid, c_shell_syntax,
 			   bourne_shell_syntax, binary_syntax, argc, argv, remaining_args);
 	  exit (0);
@@ -1272,8 +1272,8 @@ main (int argc, char **argv)
       }
  #endif /* DBUS_ENABLE_EMBEDDED_TESTS */
 
-      execl (DBUS_DAEMONDIR"/dbus-daemon",
-             DBUS_DAEMONDIR"/dbus-daemon",
+      execl (DBUS_DAEMONDIR"/dbus-daemon@SUFFIX@",
+             DBUS_DAEMONDIR"/dbus-daemon@SUFFIX@",
              close_stderr ? "--syslog-only" : "--syslog",
              "--fork",
              "--print-pid", write_pid_fd_as_string,
@@ -1284,7 +1284,7 @@ main (int argc, char **argv)
 
       fprintf (stderr,
                "Failed to execute message bus daemon %s: %s.  Will try again without full path.\n",
-               DBUS_DAEMONDIR"/dbus-daemon", strerror (errno));
+               DBUS_DAEMONDIR"/dbus-daemon@SUFFIX@", strerror (errno));
       
       /*
        * If it failed, try running without full PATH.  Note this is needed
@@ -1292,8 +1292,8 @@ main (int argc, char **argv)
        * file and the dbus-daemon will not be in the install location during
        * build time.
        */
-      execlp ("dbus-daemon",
-              "dbus-daemon",
+      execlp ("dbus-daemon@SUFFIX@",
+              "dbus-daemon@SUFFIX@",
               close_stderr ? "--syslog-only" : "--syslog",
               "--fork",
               "--print-pid", write_pid_fd_as_string,
@@ -1412,7 +1412,7 @@ main (int argc, char **argv)
                   if (x11_get_address (&address, &bus_pid, &wid)
                        && address != NULL)
                     {
-                      verbose ("dbus-daemon is already running. Returning existing parameters.\n");
+                      verbose ("dbus-daemon@SUFFIX@ is already running. Returning existing parameters.\n");
                       /* Kill the old bus */
                       kill_bus();
                       pass_info (runprog, address, bus_pid, wid,
