--- dbus/orig.dbus-sysdeps-unix.c	2016-09-15 22:15:03.000000000 +0200
+++ dbus/dbus-sysdeps-unix.c	2016-09-20 18:17:11.000000000 +0200
@@ -3836,28 +3836,45 @@
 
   _DBUS_ASSERT_ERROR_IS_CLEAR (error);
 
+  const char *fromEnv = getenv (launchd_env_var);
+  dbus_bool_t hasEnvVar = (fromEnv && *fromEnv);
+
   if (_dbus_check_setuid ())
     {
-      dbus_set_error_const (error, DBUS_ERROR_NOT_SUPPORTED,
-                            "Unable to find launchd socket when setuid");
-      return FALSE;
+      if (!hasEnvVar)
+        {
+          dbus_set_error_const (error, DBUS_ERROR_NOT_SUPPORTED,
+                                "Unable to find launchd socket when setuid");
+          return FALSE;
+        }
+
+      i = 0;
+      argv[i] = "launchctl";
+      ++i;
+      argv[i] = "getenv";
+      ++i;
+      argv[i] = (char*)launchd_env_var;
+      ++i;
+      argv[i] = NULL;
+      ++i;
+
+      _dbus_assert (i == _DBUS_N_ELEMENTS (argv));
+
+      if (!_read_subprocess_line_argv(argv[0], TRUE, argv, socket_path, error) && !hasEnvVar)
+        {
+          return FALSE;
+        }
+      else
+        {
+          // prefer the value obtained from launchctl
+          hasEnvVar = FALSE;
+        }
     }
 
-  i = 0;
-  argv[i] = "launchctl";
-  ++i;
-  argv[i] = "getenv";
-  ++i;
-  argv[i] = (char*)launchd_env_var;
-  ++i;
-  argv[i] = NULL;
-  ++i;
-
-  _dbus_assert (i == _DBUS_N_ELEMENTS (argv));
-
-  if (!_read_subprocess_line_argv(argv[0], TRUE, argv, socket_path, error))
+  if (hasEnvVar)
     {
-      return FALSE;
+      _dbus_string_append (socket_path, fromEnv);
+      return TRUE;
     }
 
   /* no error, but no result either */
@@ -3883,13 +3900,6 @@
   dbus_bool_t valid_socket;
   DBusString socket_path;
 
-  if (_dbus_check_setuid ())
-    {
-      dbus_set_error_const (error, DBUS_ERROR_NOT_SUPPORTED,
-                            "Unable to find launchd socket when setuid");
-      return FALSE;
-    }
-
   if (!_dbus_string_init (&socket_path))
     {
       _DBUS_SET_OOM (error);
--- dbus/orig.dbus-bus.c	2015-09-30 16:48:40.000000000 +0200
+++ dbus/dbus-bus.c	2016-09-20 18:31:56.000000000 +0200
@@ -434,7 +434,13 @@
       return NULL;
     }
 
+#ifdef DBUS_ENABLE_LAUNCHD
+      // on Mac we will typically not have all 3 bus addresses; as long
+      // as we have the requested one we should be fine.
+  if (!init_connections_unlocked () && bus_connection_addresses[type] == NULL)
+#else
   if (!init_connections_unlocked ())
+#endif
     {
       _DBUS_SET_OOM (error);
       goto out;
@@ -518,6 +524,10 @@
     _DBUS_ASSERT_ERROR_IS_SET (error);
 
   _DBUS_UNLOCK (bus);
+#ifdef DBUS_ENABLE_LAUNCHD
+  _dbus_verbose ("internal_bus_get(type=%d,private=%d) : found and registered connection to %s",
+         type, private, address);
+#endif
   return connection;
 }
 
