--- dbus/orig.dbus-sysdeps-unix.c	2016-09-15 22:15:03.000000000 +0200
+++ dbus/dbus-sysdeps-unix.c	2016-09-17 20:38:57.000000000 +0200
@@ -3836,12 +3836,12 @@
 
   _DBUS_ASSERT_ERROR_IS_CLEAR (error);
 
-  if (_dbus_check_setuid ())
-    {
-      dbus_set_error_const (error, DBUS_ERROR_NOT_SUPPORTED,
-                            "Unable to find launchd socket when setuid");
-      return FALSE;
-    }
+//   if (_dbus_check_setuid ())
+//     {
+//       dbus_set_error_const (error, DBUS_ERROR_NOT_SUPPORTED,
+//                             "Unable to find launchd socket when setuid");
+//       return FALSE;
+//     }
 
   i = 0;
   argv[i] = "launchctl";
@@ -3857,7 +3857,15 @@
 
   if (!_read_subprocess_line_argv(argv[0], TRUE, argv, socket_path, error))
     {
-      return FALSE;
+	 const char *c = _dbus_getenv (launchd_env_var);
+      if (*c)
+        {
+          _dbus_string_init_const (&socket_path, c);
+        }
+      else
+        {
+          return FALSE;
+        }
     }
 
   /* no error, but no result either */
@@ -3883,12 +3891,12 @@
   dbus_bool_t valid_socket;
   DBusString socket_path;
 
-  if (_dbus_check_setuid ())
-    {
-      dbus_set_error_const (error, DBUS_ERROR_NOT_SUPPORTED,
-                            "Unable to find launchd socket when setuid");
-      return FALSE;
-    }
+//   if (_dbus_check_setuid ())
+//     {
+//       dbus_set_error_const (error, DBUS_ERROR_NOT_SUPPORTED,
+//                             "Unable to find launchd socket when setuid");
+//       return FALSE;
+//     }
 
   if (!_dbus_string_init (&socket_path))
     {
