diff --git a/CMakeLists.txt b/CMakeLists.txt
index 63138cd581b562d7b3a83b3cffdd18ce2aea2ea1..5dfdabb5f8ed4af5d6d7769029ec90c2ec1b293a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -375,7 +375,9 @@ if (KDEExperimentalPurpose_FOUND)
     target_link_libraries(okularpart KDEExperimental::PurposeWidgets)
 endif()
 
-set_target_properties(okularpart PROPERTIES PREFIX "")
+set_target_properties(okularpart PROPERTIES
+    PREFIX ""
+    INSTALL_NAME_DIR ${KDE_INSTALL_PLUGINDIR})
 
 if (Qt5TextToSpeech_FOUND)
    target_link_libraries(okularpart Qt5::TextToSpeech)
diff --git a/mobile/components/CMakeLists.txt b/mobile/components/CMakeLists.txt
index f1af2602a5aac56d6003417ba836a2c54b161935..270000004005950afb9c5716cec6fef4c2d61ec1 100644
--- a/mobile/components/CMakeLists.txt
+++ b/mobile/components/CMakeLists.txt
@@ -33,6 +33,9 @@ target_link_libraries(okularplugin
         okularpart
         )
 
+set_target_properties(okularplugin PROPERTIES
+    INSTALL_NAME_DIR ${KDE_INSTALL_QMLDIR}/org/kde/okular})
+
 install(TARGETS okularplugin DESTINATION ${KDE_INSTALL_QMLDIR}/org/kde/okular)
 install(FILES qmldir DESTINATION ${KDE_INSTALL_QMLDIR}/org/kde/okular)
 install(FILES DocumentView.qml DESTINATION ${KDE_INSTALL_QMLDIR}/org/kde/okular)
