diff --git a/ui/pageview.cpp b/ui/pageview.cpp
index 598468ea6..e616a38e2 100644
--- a/ui/pageview.cpp
+++ b/ui/pageview.cpp
@@ -1592,19 +1592,21 @@ QVariant PageView::capability( ViewCapability capability ) const
         case ZoomModality:
             return d->zoomMode;
         case Continuous:
-            return d->aViewContinuous->isChecked();
+            return d->aViewContinuous ? d->aViewContinuous->isChecked() : QVariant();
         case ViewModeModality:
         {
-            for (int i=0; i < d->aViewMode->menu()->actions().size(); ++i)
-            {
-                const QAction* action = d->aViewMode->menu()->actions().at(i);
-                if ( action->isChecked() )
-                    return action->data();
+            if (d->aViewMode) {
+                for (int i=0; i < d->aViewMode->menu()->actions().size(); ++i)
+                {
+                    const QAction* action = d->aViewMode->menu()->actions().at(i);
+                    if ( action && action->isChecked() )
+                        return action->data();
+                }
             }
             return QVariant();
         }
         case TrimMargins:
-            return d->aTrimMargins->isChecked();
+            return d->aTrimMargins ? d->aTrimMargins->isChecked() : QVariant();
     }
     return QVariant();
 }
