diff --git a/CMakeLists.txt b/CMakeLists.txt
index b5c7fb3..e7bff79 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -22,16 +22,18 @@ include(CheckTypeSize)
 find_package(CopperSpice REQUIRED)
 
 # file locations for installing
-if (CMAKE_SYSTEM_NAME MATCHES "Darwin")
+if (CMAKE_SYSTEM_NAME MATCHES "Darwin" AND (NOT CMAKE_BUILD_TYPE MATCHES "MacPorts"))
    include(GNUInstallDirs)
 
    # indicates where libraries are located relative to the executable
    set(CMAKE_INSTALL_RPATH "@executable_path/../Resources")
 
-elseif(CMAKE_SYSTEM_NAME MATCHES "(Linux|OpenBSD|FreeBSD|NetBSD|DragonFly)")
+elseif(CMAKE_SYSTEM_NAME MATCHES "(Linux|OpenBSD|FreeBSD|NetBSD|DragonFly)" OR CMAKE_BUILD_TYPE MATCHES "MacPorts")
    include(GNUInstallDirs)
 
-   set(CMAKE_INSTALL_RPATH "\$ORIGIN")
+   if (NOT CMAKE_BUILD_TYPE MATCHES "MacPorts")
+       set(CMAKE_INSTALL_RPATH "\$ORIGIN")
+   endif()
 
 elseif(MSVC)
    # use defaults
@@ -81,14 +83,14 @@ else()
 endif()
 
 
-# webkit temporarily removed, pending overflow issue resolution in CsWebKit
-add_definitions(-DQT_NO_WEBKIT)
+# # webkit temporarily removed, pending overflow issue resolution in CsWebKit
+# add_definitions(-DQT_NO_WEBKIT)
 
-# if("WebKit" IN_LIST COPPERSPICE_COMPONENTS)
-#   # webkit found and will be linked with this project
-# else()
-#   add_definitions(-DQT_NO_WEBKIT)
-# endif()
+if("WebKit" IN_LIST COPPERSPICE_COMPONENTS)
+  # webkit found and will be linked with this project
+else()
+  add_definitions(-DQT_NO_WEBKIT)
+endif()
 
 
 if(APPLE)
@@ -101,7 +103,9 @@ else()
     set(CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} -Wl,--no-undefined")
 endif()
 
-string(TIMESTAMP BUILD_DATE "%m/%d/%Y")
+if (NOT BUILD_DATE)
+    string(TIMESTAMP BUILD_DATE "%m/%d/%Y")
+endif()
 add_definitions(-DBUILD_DATE="${BUILD_DATE}")
 
 configure_file(
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 50d978d..8bbf38a 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -168,71 +168,86 @@ elseif(CMAKE_SYSTEM_NAME MATCHES "Windows")
 endif()
 
 if (CMAKE_SYSTEM_NAME MATCHES "Darwin")
-   install(TARGETS KitchenSink DESTINATION ../MacOS)
+   set(DEST "../MacOS")
 else()
-   install(TARGETS KitchenSink DESTINATION .)
+   if (CMAKE_BUILD_TYPE MATCHES "MacPorts")
+       set(DEST "${CMAKE_INSTALL_BINDIR}/KitchenSink")
+   else()
+       set(DEST ".")
+   endif()
 endif()
+install(TARGETS KitchenSink DESTINATION ${DEST})
 
-install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/../resources/script.js       DESTINATION ./resources)
-install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/../resources/sampleMenu.xml  DESTINATION ./resources)
+install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/../resources/script.js       DESTINATION ${DEST}/resources)
+install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/../resources/sampleMenu.xml  DESTINATION ${DEST}/resources)
 
-# installs the CopperSpice libraries
-cs_copy_library(CsCore)
-cs_copy_library(CsGui)
-cs_copy_library(CsNetwork)
-cs_copy_library(CsOpenGL)
-cs_copy_library(CsScript)
-cs_copy_library(CsSql)
-cs_copy_library(CsSvg)
-cs_copy_library(CsXml)
-cs_copy_library(CsXmlPatterns)
+if (NOT CMAKE_BUILD_TYPE MATCHES "MacPorts")
+    # installs the CopperSpice libraries
+    cs_copy_library(CsCore)
+    cs_copy_library(CsGui)
+    cs_copy_library(CsNetwork)
+    cs_copy_library(CsOpenGL)
+    cs_copy_library(CsScript)
+    cs_copy_library(CsSql)
+    cs_copy_library(CsSvg)
+    cs_copy_library(CsXml)
+    cs_copy_library(CsXmlPatterns)
 
-if("Multimedia" IN_LIST COPPERSPICE_COMPONENTS)
-   cs_copy_library(CsMultimedia)
-endif()
+    if("Multimedia" IN_LIST COPPERSPICE_COMPONENTS)
+       cs_copy_library(CsMultimedia)
+    endif()
 
-if("WebKit" IN_LIST COPPERSPICE_COMPONENTS)
-   # webkit temporarily removed, pending overflow issue resolution in CsWebKit
-   # cs_copy_library(CsWebKit)
-endif()
+    if("WebKit" IN_LIST COPPERSPICE_COMPONENTS)
+       # webkit temporarily removed, pending overflow issue resolution in CsWebKit
+       # cs_copy_library(CsWebKit)
+    endif()
 
-if (CMAKE_SYSTEM_NAME MATCHES "Darwin")
+    if (CMAKE_SYSTEM_NAME MATCHES "Darwin")
 
-   set(CPACK_DMG_VOLUME_NAME KitchenSink)
-   set(CPACK_DMG_FORMAT UDBZ)
+       set(CPACK_DMG_VOLUME_NAME KitchenSink)
+       set(CPACK_DMG_FORMAT UDBZ)
 
-   # installs the platform Gui plugin
-   cs_copy_plugins(CsGui ../plugins)
+       # installs the platform Gui plugin
+       cs_copy_plugins(CsGui ../plugins)
 
-   # two files
-   if("Multimedia" IN_LIST COPPERSPICE_COMPONENTS)
-      cs_copy_plugins(CsMultimedia ../plugins)
-   endif()
+       # two files
+       if("Multimedia" IN_LIST COPPERSPICE_COMPONENTS)
+          cs_copy_plugins(CsMultimedia ${DEST}/../plugins)
+       endif()
 
-   install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/../resources/ks.icns DESTINATION ./resources)
+       install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/../resources/ks.icns DESTINATION ${DEST}/resources)
 
-elseif(CMAKE_SYSTEM_NAME MATCHES "(Linux|OpenBSD|FreeBSD|NetBSD|DragonFly)")
+    elseif(CMAKE_SYSTEM_NAME MATCHES "(Linux|OpenBSD|FreeBSD|NetBSD|DragonFly)")
 
-   # installs the platform Gui plugin
-   cs_copy_plugins(CsGui)
+       # installs the platform Gui plugin
+       cs_copy_plugins(CsGui)
 
-   # three files
-   if("Multimedia" IN_LIST COPPERSPICE_COMPONENTS)
-      cs_copy_plugins(CsMultimedia)
-   endif()
+       # three files
+       if("Multimedia" IN_LIST COPPERSPICE_COMPONENTS)
+          cs_copy_plugins(CsMultimedia)
+       endif()
 
-   install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/../resources/ks.png DESTINATION ./resources)
+       install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/../resources/ks.png DESTINATION ${DEST}/resources)
 
-elseif(CMAKE_SYSTEM_NAME MATCHES "Windows")
+    elseif(CMAKE_SYSTEM_NAME MATCHES "Windows")
 
-   # installs the platform Gui plugin
-   cs_copy_plugins(CsGui)
+       # installs the platform Gui plugin
+       cs_copy_plugins(CsGui)
 
-   # two files
-   if("Multimedia" IN_LIST COPPERSPICE_COMPONENTS)
-      cs_copy_plugins(CsMultimedia)
-   endif()
+       # two files
+       if("Multimedia" IN_LIST COPPERSPICE_COMPONENTS)
+          cs_copy_plugins(CsMultimedia)
+       endif()
 
-   install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/../resources/ks.ico DESTINATION ./resources)
+       install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/../resources/ks.ico DESTINATION ${DEST}/resources)
 
+    endif()
+else()
+    if (CMAKE_SYSTEM_NAME MATCHES "Darwin")
+       set(CPACK_DMG_VOLUME_NAME KitchenSink)
+       set(CPACK_DMG_FORMAT UDBZ)
+    endif()
+    install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/../resources/ks.icns DESTINATION ${DEST}/resources)
+    install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/../resources/ks.png DESTINATION ${DEST}/resources)
+    install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/../resources/ks.ico DESTINATION ${DEST}/resources)
 endif()
