diff --git src/qtui/playlist.cc src/qtui/playlist.cc
index 7c4c02e..c3cea74 100644
--- src/qtui/playlist.cc
+++ src/qtui/playlist.cc
@@ -60,6 +60,7 @@ PlaylistWidget::PlaylistWidget (QWidget * parent, Playlist playlist) :

     updateSettings ();
     header->updateColumns ();
+    setSortingEnabled (true);

     /* get initial selection and focus from core */
     inUpdate = true;
diff --git src/qtui/playlist_header.cc src/qtui/playlist_header.cc
index e835338..aba31c7 100644
--- src/qtui/playlist_header.cc
+++ src/qtui/playlist_header.cc
@@ -133,6 +133,8 @@ PlaylistHeader::PlaylistHeader (PlaylistWidget * playlist) :

     setSectionsMovable (true);
     setStretchLastSection (true);
+    setSectionsClickable (true);
+    setSortIndicatorShown (true);

     connect (this, & QHeaderView::sectionResized, this, & PlaylistHeader::sectionResized);
     connect (this, & QHeaderView::sectionMoved, this, & PlaylistHeader::sectionMoved);
diff --git src/qtui/playlist_model.cc src/qtui/playlist_model.cc
index a78fff2e7403d83214ac8cba45322954879cdaba..944c7b22707d75a2475e4fc43ed5526930c5f818 100644
--- src/qtui/playlist_model.cc
+++ src/qtui/playlist_model.cc
@@ -117,6 +117,7 @@ QVariant PlaylistModel::data (const QModelIndex &index, int role) const
     switch (role)
     {
     case Qt::DisplayRole:
+    case Qt::ToolTipRole:
         if (s_fields[col] != Tuple::Invalid)
         {
             tuple = m_playlist.entry_tuple (index.row (), Playlist::NoWait);
