diff --git src/statusicon-qt/statusicon.cc src/statusicon-qt/statusicon.cc
index 2643c4e426fde4b7323e16612302064aecc0eaaa..64399abed91572acc00d44879a6f4c551a8002ec 100644
--- src/statusicon-qt/statusicon.cc
+++ src/statusicon-qt/statusicon.cc
@@ -89,6 +89,9 @@ const audqt::MenuItem StatusIcon::items[] =
     audqt::MenuSep (),
     audqt::MenuCommand ({N_("_Open Files ..."), "document-open"}, StatusIcon::open_files),
     audqt::MenuCommand ({N_("Se_ttings ..."), "preferences-system"}, audqt::prefswin_show),
+#ifdef Q_OS_MACOS
+    audqt::MenuCommand ({N_("_Hide"), "hide-application"}, StatusIcon::toggle_aud_ui),
+#endif
     audqt::MenuCommand ({N_("_Quit"), "application-exit"}, aud_quit),
 };
 
@@ -146,7 +149,9 @@ void StatusIcon::activate(QSystemTrayIcon::ActivationReason reason)
     switch (reason)
     {
         case QSystemTrayIcon::Trigger:
+#ifndef Q_OS_MACOS
             toggle_aud_ui ();
+#endif
             break;
 
         case QSystemTrayIcon::MiddleClick:
