diff --git src/playlist-manager-qt/playlist-manager-qt.cc src/playlist-manager-qt/playlist-manager-qt.cc
index 3dc39c413ca18b3d4295537ff640f68a89048097..6066119f99a353aec23ce1bfb3222ad6082409fd 100644
--- src/playlist-manager-qt/playlist-manager-qt.cc
+++ src/playlist-manager-qt/playlist-manager-qt.cc
@@ -324,7 +324,7 @@ static QToolButton * new_tool_button (const char * text, const char * icon)
     auto button = new QToolButton;
     button->setIcon (QIcon::fromTheme (icon));
     button->setText (audqt::translate_str (text));
-    button->setToolButtonStyle (Qt::ToolButtonTextBesideIcon);
+    button->setToolButtonStyle (Qt::ToolButtonFollowStyle);
     return button;
 }
 
