diff --git src/lyricwiki-qt/Makefile src/lyricwiki-qt/Makefile
index a17bc66d7497d8f37bdccaf4ec3f3af9d6389be0..3dfdf174538148a76a3653f024275e648e02c9ce 100644
--- src/lyricwiki-qt/Makefile
+++ src/lyricwiki-qt/Makefile
@@ -11,4 +11,4 @@ LD = ${CXX}
 
 CFLAGS += ${PLUGIN_CFLAGS}
 CPPFLAGS += ${PLUGIN_CPPFLAGS} ${QT_CFLAGS} ${GLIB_CFLAGS} ${XML_CFLAGS} -I../..
-LIBS += ${QT_LIBS} ${GLIB_LIBS}  ${XML_LIBS}
+LIBS += ${QT_LIBS} ${GLIB_LIBS}  ${XML_LIBS} -laudqt
diff --git src/lyricwiki-qt/lyricwiki.cc src/lyricwiki-qt/lyricwiki.cc
index d325347b328822b25b18137a02deab9cc59e1fd8..e942036aac99e79dafd1ae38e476a59e9e172916 100644
--- src/lyricwiki-qt/lyricwiki.cc
+++ src/lyricwiki-qt/lyricwiki.cc
@@ -356,6 +356,7 @@ static void update_lyrics_window (const char * title, const char * artist, const
     cursor.insertHtml ("<br><br>");
     cursor.insertText (lyrics);
 
+    doc.setDefaultFont (audqt::get_font_for_class ("QTipLabel"));
     textedit->setDocument (doc.clone ());
 }
 
diff --git src/playlist-manager-qt/playlist-manager-qt.cc src/playlist-manager-qt/playlist-manager-qt.cc
index 10dd1a84cea1feeb6fc01be0875b1cc598f37dcc..e5399e8cf818b2be08cde43751af6d5f5acfb247 100644
--- src/playlist-manager-qt/playlist-manager-qt.cc
+++ src/playlist-manager-qt/playlist-manager-qt.cc
@@ -20,7 +20,6 @@
 #include <QAbstractListModel>
 #include <QBoxLayout>
 #include <QFont>
-#include <QGuiApplication>
 #include <QHeaderView>
 #include <QIcon>
 #include <QMouseEvent>
@@ -64,7 +63,7 @@ public:
     PlaylistsModel () :
         m_rows (Playlist::n_playlists ()),
         m_playing (Playlist::playing_playlist ().index ()),
-        m_bold (QGuiApplication::font ())
+        m_bold (audqt::get_font_for_class ("QListView"))
     {
         m_bold.setBold (true);
     }
@@ -244,6 +243,7 @@ PlaylistsView::PlaylistsView ()
     setDragDropMode (InternalMove);
     setFrameShape (QFrame::NoFrame);
     setIndentation (0);
+    setFont (audqt::get_font_for_class ("QListView"));
 }
 
 void PlaylistsView::currentChanged (const QModelIndex & current, const QModelIndex & previous)
diff --git src/qtui/playlist.cc src/qtui/playlist.cc
index c3cea747b0b944f45adc95852daf6ba241c02c6f..4c24c5b417d7360d91b5149b4ef474794e8a049f 100644
--- src/qtui/playlist.cc
+++ src/qtui/playlist.cc
@@ -27,6 +27,7 @@
 #include <libaudcore/hook.h>
 #include <libaudcore/playlist.h>
 #include <libaudcore/runtime.h>
+#include <libaudqt/libaudqt.h>
 
 #include "playlist.h"
 #include "playlist_header.h"
@@ -58,6 +59,8 @@ PlaylistWidget::PlaylistWidget (QWidget * parent, Playlist playlist) :
     setSelectionMode (ExtendedSelection);
     setDragDropMode (DragDrop);
 
+    setFont (audqt::get_font_for_class ("QListView"));
+
     updateSettings ();
     header->updateColumns ();
     setSortingEnabled (true);
diff --git src/qtui/playlist_header.cc src/qtui/playlist_header.cc
index aba31c755531fe7bf5d439257dcdb99fa78dca0f..6618cd977691f9677f1300f58cc8510c9a6c35f0 100644
--- src/qtui/playlist_header.cc
+++ src/qtui/playlist_header.cc
@@ -136,6 +136,8 @@ PlaylistHeader::PlaylistHeader (PlaylistWidget * playlist) :
     setSectionsClickable (true);
     setSortIndicatorShown (true);
 
+    setFont (audqt::get_font_for_class ("QSmallFont"));
+
     connect (this, & QHeaderView::sectionResized, this, & PlaylistHeader::sectionResized);
     connect (this, & QHeaderView::sectionMoved, this, & PlaylistHeader::sectionMoved);
 }
diff --git src/qtui/status_bar.cc src/qtui/status_bar.cc
index a9b35fa281e1301c52ea7aa575480e238b19c66f..5bff3f8e46232e4b9d428dfc083316304a06dedd 100644
--- src/qtui/status_bar.cc
+++ src/qtui/status_bar.cc
@@ -25,6 +25,7 @@
 #include <libaudcore/mainloop.h>
 #include <libaudcore/playlist.h>
 #include <libaudcore/tinylock.h>
+#include <libaudqt/libaudqt.h>
 
 #include <QLabel>
 
@@ -52,6 +53,10 @@ StatusBar::StatusBar (QWidget * parent) :
     update_length ();
 
     setStyleSheet (normal_css);
+    QFont font (audqt::get_font_for_class ("QTipLabel"));
+    setFont (font);
+    codec_label->setFont (font);
+    length_label->setFont (font);
 
     audlog::subscribe (log_handler, audlog::Warning);
 
diff --git src/search-tool-qt/search-tool-qt.cc src/search-tool-qt/search-tool-qt.cc
index c9e3118fc9236b004ca5650557230e11f85861ed..b653802d169a02380ab83e9364ff6041722ae470 100644
--- src/search-tool-qt/search-tool-qt.cc
+++ src/search-tool-qt/search-tool-qt.cc
@@ -192,7 +192,7 @@ void ResultsModel::update ()
 
 QVariant ResultsModel::data (const QModelIndex & index, int role) const
 {
-    if (role == Qt::DisplayRole)
+    if (role == Qt::DisplayRole || role == Qt::ToolTipRole)
         return QString ((const char *) create_item_label (index.row ()));
     else
         return QVariant ();
@@ -741,9 +741,11 @@ QMimeData * ResultsModel::mimeData (const QModelIndexList & indexes) const
 void * SearchToolQt::get_qt_widget ()
 {
     s_search_entry = new QLineEdit;
-    s_search_entry->setContentsMargins (audqt::margins.TwoPt);
+    QMargins OnePt = audqt::margins.TwoPt / 2;
+    s_search_entry->setContentsMargins (OnePt);
     s_search_entry->setClearButtonEnabled (true);
     s_search_entry->setPlaceholderText (_("Search library"));
+    s_search_entry->setFont (audqt::get_font_for_class ("QListBox"));
 
     s_help_label = new QLabel (_("To import your music library into Audacious, "
      "choose a folder and then click the \"refresh\" icon."));
@@ -762,14 +764,26 @@ void * SearchToolQt::get_qt_widget ()
     s_results_list->setModel (& s_model);
     s_results_list->setSelectionMode (QTreeView::ExtendedSelection);
     s_results_list->setDragDropMode (QTreeView::DragOnly);
+    s_results_list->setFont (audqt::get_font_for_class ("QListView"));
 
     s_stats_label = new QLabel;
     s_stats_label->setAlignment (Qt::AlignCenter);
     s_stats_label->setContentsMargins (audqt::margins.TwoPt);
+    s_stats_label->setFont (audqt::get_font_for_class ("QSmallFont"));
 
     auto chooser = new QLineEdit;
+    chooser->setFont (audqt::get_font_for_class ("QLabel"));
 
     auto button = new QPushButton (QIcon::fromTheme ("view-refresh"), QString ());
+    if (button->icon().isNull())
+    {
+        QFont font (button->font());
+        font.setPixelSize(22);
+        font.setBold(true);
+        button->setFont (font);
+        button->setText (QChar (0x21BA));
+        button->setFixedSize (24, 24);
+    }
     button->setFlat (true);
     button->setFocusPolicy (Qt::NoFocus);
 
