diff --git src/qtui/main_window.cc src/qtui/main_window.cc
index 4a16a657f25e45a2ef758d2e3db2dd015b324cea..8bae4829e24867cf3224955e2aa190655005c8f9 100644
--- src/qtui/main_window.cc
+++ src/qtui/main_window.cc
@@ -132,7 +141,9 @@ MainWindow::MainWindow () :
     };
 
     addToolBar (Qt::TopToolBarArea, new ToolBar (this, items));
-    setUnifiedTitleAndToolBarOnMac (true);
+    // this worked more or less in Qt4 but is not recommended in Qt5, not
+    // if you want to do OpenGL or have dockable widgets
+//     setUnifiedTitleAndToolBarOnMac (true);
 
     if (m_search_tool)
         aud_plugin_add_watch (m_search_tool, plugin_watcher, this);
