diff --git a/CMakeLists.txt b/CMakeLists.txt
index 261eaaa..8d916d0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -29,14 +29,14 @@ include(ResourceMacros)
 include(TestLargeFiles)
 
 # location for install or package
-if(CMAKE_SYSTEM_NAME MATCHES "Darwin")
+if(CMAKE_SYSTEM_NAME MATCHES "Darwin" AND (NOT CMAKE_BUILD_TYPE MATCHES "MacPorts"))
    include(GNUInstallDirs)
 
    # rpath
    file(RELATIVE_PATH CS_BIN_RPATH "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_BINDIR}" "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}")
    set(CMAKE_INSTALL_RPATH "@executable_path/${CS_BIN_RPATH}")
 
-elseif(CMAKE_SYSTEM_NAME MATCHES "(Linux|OpenBSD|FreeBSD|NetBSD|DragonFly)")
+elseif(CMAKE_SYSTEM_NAME MATCHES "(Linux|OpenBSD|FreeBSD|NetBSD|DragonFly)" OR CMAKE_BUILD_TYPE MATCHES "MacPorts")
    include(GNUInstallDirs)
 
    # rpath
@@ -149,7 +149,9 @@ else()
 
 endif()
 
-string(TIMESTAMP BUILD_DATE "%Y-%m-%d")
+if (NOT BUILD_DATE)
+   string(TIMESTAMP BUILD_DATE "%Y-%m-%d")
+endif()
 add_definitions(-DQT_SHARED)
 
 check_include_file(dlfcn.h       HAVE_DLFCN_H)
@@ -345,7 +347,7 @@ if(CMAKE_SYSTEM_NAME MATCHES "(Linux|OpenBSD|FreeBSD|NetBSD|DragonFly)")
       TYPE REQUIRED
    )
 
-   if(NOT CMAKE_SYSTEM_NAME MATCHES "Linux")
+   if(NOT CMAKE_SYSTEM_NAME MATCHES "Linux" OR CMAKE_BUILD_TYPE MATCHES "MacPorts")
       find_package(Iconv)
       set_package_properties(Iconv PROPERTIES
          PURPOSE "Iconv support"
@@ -446,6 +448,7 @@ if(NOT Cups_FOUND)
 endif()
 
 if(NOT GLIB2_FOUND)
+    message(WARNING "GLib not found, setting QT_NO_GLIB")
    add_definitions(-DQT_NO_GLIB)
 endif()
 
@@ -536,7 +539,7 @@ if(WITH_GUI)
       set(BUILD_PLATFORMS_WINDOWS_PLUGIN TRUE)
    endif()
 
-   if(CMAKE_SYSTEM_NAME MATCHES "(Linux|OpenBSD|FreeBSD|NetBSD|DragonFly)")
+   if(CMAKE_SYSTEM_NAME MATCHES "(Linux|OpenBSD|FreeBSD|NetBSD|DragonFly)" OR CMAKE_BUILD_TYPE MATCHES "MacPorts")
       set(BUILD_PLATFORMS_XCB_PLUGIN TRUE)
    endif()
 
@@ -556,7 +559,7 @@ configure_file(
    @ONLY
 )
 
-if (CMAKE_SYSTEM_NAME MATCHES "Darwin")
+if (CMAKE_SYSTEM_NAME MATCHES "Darwin" AND (NOT CMAKE_BUILD_TYPE MATCHES "MacPorts"))
    set(PKG_PREFIX "CopperSpice.framework/Resources")
 
 elseif(CMAKE_SYSTEM_NAME MATCHES "Windows")
diff --git a/cmake/modules/FindMySQL.cmake b/cmake/modules/FindMySQL.cmake
index 35393fa..a6914b1 100644
--- a/cmake/modules/FindMySQL.cmake
+++ b/cmake/modules/FindMySQL.cmake
@@ -35,17 +35,28 @@ if (NOT WIN32)
 
 endif()
 
-if (_MySQL_use_pkgconfig)
-   pkg_check_modules(_libmariadb "libmariadb" QUIET IMPORTED_TARGET)
+if(CMAKE_SYSTEM_NAME MATCHES "Linux")
+   set(MySQL_INCLUDE_DIRS "/usr/include/mysql")
+   set(MySQL_LIBRARIES "/usr/lib/x86_64-linux-gnu/libmysqld.so")
+   if (NOT TARGET MySQL::MySQL)
+      add_library(MySQL::MySQL INTERFACE IMPORTED)
+
+      target_link_libraries(MySQL::MySQL
+         INTERFACE  "-lmysqld"
+      )
+   endif()
+   set(MySQL_FOUND 1)
+elseif (_MySQL_use_pkgconfig)
+   pkg_check_modules(_libmariadb "libmariadb"  IMPORTED_TARGET)
    unset(_mysql_target)
 
    if (_libmariadb_FOUND)
       set(_mysql_target "_libmariadb")
    else ()
-      pkg_check_modules(_mariadb "mariadb" QUIET IMPORTED_TARGET)
+      pkg_check_modules(_mariadb "mariadb"  IMPORTED_TARGET)
 
       if (NOT _mariadb_FOUND)
-         pkg_check_modules(_mysql "mysql" QUIET IMPORTED_TARGET)
+         pkg_check_modules(_mysql "mysql"  IMPORTED_TARGET)
 
          if (_mysql_FOUND)
             set(_mysql_target "_mysql")
diff --git a/src/core/CMakeLists.txt b/src/core/CMakeLists.txt
index 1520f25..1db17b7 100644
--- a/src/core/CMakeLists.txt
+++ b/src/core/CMakeLists.txt
@@ -116,6 +116,12 @@ elseif(CMAKE_SYSTEM_NAME MATCHES "Linux")
       dl
       ${CMAKE_THREAD_LIBS_INIT}
    )
+   if (CMAKE_BUILD_TYPE MATCHES "MacPorts")
+       target_link_libraries(CsCore
+          PRIVATE
+          iconv
+       )
+   endif()
    include_directories(${CMAKE_SOURCE_DIR}/src/3rdparty/forkfd)
 
 elseif(CMAKE_SYSTEM_NAME MATCHES "(OpenBSD|FreeBSD|NetBSD|DragonFly)")
