--- keditfiletype/orig.keditfiletype.cpp	2015-12-15 11:19:17.000000000 +0100
+++ keditfiletype/keditfiletype.cpp	2015-12-21 10:15:46.064265293 +0100
@@ -128,7 +128,7 @@
 int main(int argc, char ** argv)
 {
     QApplication app(argc, argv);
-    QApplication::setWindowIcon(QIcon::fromTheme("preferences-desktop-filetype-association"));
+    QApplication::setWindowIcon(QIcon::fromTheme("preferences-desktop-filetype-association", app.windowIcon()));
 
     KAboutData aboutData( "keditfiletype", i18n("File Type Editor"), PROJECT_VERSION,
        i18n("KDE file type editor - simplified version for editing a single file type"),
--- kdesu/orig.kdesu.cpp	2015-12-15 11:19:17.000000000 +0100
+++ kdesu/kdesu.cpp	2015-12-21 10:16:14.032266458 +0100
@@ -91,7 +91,7 @@
             "jansen@kde.org", "http://www.stack.nl/~geertj/");
     aboutData.addAuthor(i18n("Pietro Iglio"), i18n("Original author"),
             "iglio@fub.it");
-    app.setWindowIcon(QIcon::fromTheme("dialog-password"));
+    app.setWindowIcon(QIcon::fromTheme("dialog-password", app.windowIcon()));
 
     KAboutData::setApplicationData(aboutData);
 
--- kcmshell/orig.main.cpp	2015-12-23 01:13:22.000000000 +0100
+++ kcmshell/main.cpp	2015-12-23 01:13:42.000000000 +0100
@@ -283,10 +283,18 @@
     }
 
     if (parser.isSet("icon")) {
-        dlg->setWindowIcon(QIcon::fromTheme(parser.value("icon")));
+        const QIcon icon = QIcon::fromTheme(parser.value("icon"));
+        if (!icon.isNull()) {
+            dlg->setWindowIcon(icon);
+            app.setWindowIcon(icon);
+        }
     } else if (!parser.isSet("icon") && !modules.isEmpty()) {
         const QString iconName = KCModuleInfo(modules.first()).icon();
-        dlg->setWindowIcon( QIcon::fromTheme(iconName) );
+        const QIcon icon = QIcon::fromTheme(iconName);
+        if (!icon.isNull()) {
+            dlg->setWindowIcon(icon);
+            app.setWindowIcon(icon);
+        }
     }
     dlg->exec();
     delete dlg;
