diff --git src/plugins/texteditor/texteditor.cpp src/plugins/texteditor/texteditor.cpp
index 5ac69be40cefccc38ce67d59216e4257904ff5b9..fa09c52568a45cb1088fdd53afdeeeccf2e68000 100644
--- src/plugins/texteditor/texteditor.cpp
+++ src/plugins/texteditor/texteditor.cpp
@@ -373,13 +373,13 @@ public:
         }
     }
 
-private:
     void abortHandlers()
     {
         for (BaseHoverHandler *handler : m_handlers)
             handler->abort();
     }
 
+private:
     TextEditorWidget *m_widget;
     const QList<BaseHoverHandler *> &m_handlers;
 
@@ -829,6 +829,10 @@ TextEditorWidgetPrivate::TextEditorWidgetPrivate(TextEditorWidget *parent)
     m_fileEncodingLabelAction = m_toolBar->addWidget(m_fileEncodingLabel);
 
     m_extraSelections.reserve(NExtraSelectionKinds);
+    connect(qApp, &QCoreApplication::aboutToQuit, this, [&]() {
+        m_hoverHandlerRunner.abortHandlers();
+        m_hoverHandlers.clear();
+    });
 }
 
 TextEditorWidgetPrivate::~TextEditorWidgetPrivate()
