diff --git qtcreator.pri qtcreator.pri
index 407a18cbae..a0542a5a69 100644
--- qtcreator.pri
+++ qtcreator.pri
@@ -14,7 +14,8 @@ isEmpty(IDE_CASED_ID):               IDE_CASED_ID = QtCreator
 
 isEmpty(PRODUCT_BUNDLE_IDENTIFIER): PRODUCT_BUNDLE_IDENTIFIER = org.qt-project.$$IDE_ID
 
-CONFIG += c++14
+CONFIG -= c++14
+CONFIG += c++17
 
 defineReplace(qtLibraryTargetName) {
    unset(LIBRARY_NAME)
diff --git src/plugins/plugins.pro src/plugins/plugins.pro
index 9efcf77bf1..0ba96ca554 100644
--- src/plugins/plugins.pro
+++ src/plugins/plugins.pro
@@ -68,11 +68,11 @@ qtHaveModule(serialport) {
     warning("SerialTerminal plugin has been disabled since the Qt SerialPort module is not available.")
 }
 
-qtHaveModule(quick) {
-    SUBDIRS += qmlprofiler
-} else {
-    warning("QmlProfiler plugin has been disabled since the Qt Quick module is not available.")
-}
+# qtHaveModule(quick) {
+#     SUBDIRS += qmlprofiler
+# } else {
+#     warning("QmlProfiler plugin has been disabled since the Qt Quick module is not available.")
+# }
 
 qtHaveModule(help) {
     SUBDIRS += help
diff --git src/shared/clang/clang_installation.pri src/shared/clang/clang_installation.pri
index 4629aedaab..60a61519c3 100644
--- src/shared/clang/clang_installation.pri
+++ src/shared/clang/clang_installation.pri
@@ -238,10 +238,11 @@ isEmpty(LLVM_VERSION) {
 
     # Remove unwanted flags. It is a workaround for linking.
     # It is not intended for cross compiler linking.
-    LLVM_CXXFLAGS *= $$system($$llvm_config --cxxflags, lines)
+    LLVM_CXXFLAGS *= $$system($$llvm_config --cxxflags|sed -e 's/-std=[gnuc]*++[0-9yzx][0-9yzx]//g', lines)
     LLVM_CXXFLAGS ~= s,-fno-exceptions,
     LLVM_CXXFLAGS ~= s,-std=c++11,
     LLVM_CXXFLAGS ~= s,-std=c++0x,
+    LLVM_CXXFLAGS ~= s,-std=c++14,
     LLVM_CXXFLAGS ~= s,-O\S*,
     LLVM_CXXFLAGS ~= s,/O\S*,
     LLVM_CXXFLAGS ~= s,/W4,
diff --git tests/manual/clang-format-for-qtc/clang-format-for-qtc.pro tests/manual/clang-format-for-qtc/clang-format-for-qtc.pro
index 370c349d4a..fcb48da265 100644
--- tests/manual/clang-format-for-qtc/clang-format-for-qtc.pro
+++ tests/manual/clang-format-for-qtc/clang-format-for-qtc.pro
@@ -1,4 +1,4 @@
 QT -= gui
-CONFIG += c++14 console
+CONFIG += c++17 console
 CONFIG -= app_bundle
 SOURCES += test.cpp
diff --git tests/unit/unittest/unittest.pro tests/unit/unittest/unittest.pro
index 634a0a0e2b..75465faab2 100644
--- tests/unit/unittest/unittest.pro
+++ tests/unit/unittest/unittest.pro
@@ -1,7 +1,7 @@
 INCLUDEPATH += ../mockup
 
 QT += core network testlib widgets
-CONFIG += console c++14 testcase object_parallel_to_source
+CONFIG += console c++17 testcase object_parallel_to_source
 CONFIG -= app_bundle shared
 
 include(gmock_dependency.pri)
