diff --git src/tools/clangpchmanagerbackend/source/collectusedmacrosandsourcespreprocessorcallbacks.h src/tools/clangpchmanagerbackend/source/collectusedmacrosandsourcespreprocessorcallbacks.h
index 90108bd76b..d3de37173b 100644
--- src/tools/clangpchmanagerbackend/source/collectusedmacrosandsourcespreprocessorcallbacks.h
+++ src/tools/clangpchmanagerbackend/source/collectusedmacrosandsourcespreprocessorcallbacks.h
@@ -93,7 +93,8 @@ public:
                                    id,
                                    fileEntry->getSize(),
                                    fileEntry->getModificationTime(),
-                                   fileEntry->isInPCH());
+                                   // FileEntry::isInPCH() always returned false (by this point)
+                                   false);
         }
     }
 
