diff --git src/plugins/texteditor/textdocument.cpp src/plugins/texteditor/textdocument.cpp
index 2bfb51439a..6709233dd5 100644
--- src/plugins/texteditor/textdocument.cpp
+++ src/plugins/texteditor/textdocument.cpp
@@ -197,6 +197,11 @@ QTextCursor TextDocumentPrivate::indentOrUnindent(const QTextCursor &textCursor,
             }
             cursor.setPosition(start);
             cursor.setPosition(end, QTextCursor::KeepAnchor);
+        } else {
+            if (offset) {
+                *offset = tabSettings.columnAt(cursor.block().text(), cursor.positionInBlock())
+                    - column;
+            }
         }
     }
 
diff --git src/plugins/texteditor/texteditor.cpp src/plugins/texteditor/texteditor.cpp
index 5ac69be40c..2408ac48cd 100644
--- src/plugins/texteditor/texteditor.cpp
+++ src/plugins/texteditor/texteditor.cpp
@@ -1860,25 +1864,50 @@ void TextEditorWidget::sortSelectedLines()
 
 void TextEditorWidget::indent()
 {
-    int offset = 0;
+    int offset = 0, curPos = -1;
+    QTextCursor tc = textCursor();
+    if (!d->m_inBlockSelectionMode && !tc.hasSelection()) {
+        // current "stock" behaviour when no text has been selected is that
+        // (un)indent behaves just like the Tab key when the cursor is not at
+        // the start of a line text. So move it.
+        curPos = tc.position();
+        // ensure we can emulate hitting `Home` and send the cursor to the start of text, not start of line!
+        moveCursor(QTextCursor::EndOfLine);
+        d->handleHomeKey(false);
+    }
+    // our `tc` won't have the updated position, for that we need to use textCursor()!
     doSetTextCursor(textDocument()->indent(textCursor(), d->m_inBlockSelectionMode,
                                            d->m_blockSelection.positionColumn, &offset),
                     d->m_inBlockSelectionMode);
     if (d->m_inBlockSelectionMode) {
         d->m_blockSelection.anchorColumn += offset;
         d->m_blockSelection.positionColumn += offset;
+    } else if (curPos != -1) {
+        // restore the entry cursor position using `tc`, NOT textCursor()!
+        tc.setPosition(curPos + offset);
+        setTextCursor(tc);
     }
 }
 
 void TextEditorWidget::unindent()
 {
-    int offset = 0;
+    int offset = 0, curPos = -1;
+    QTextCursor tc = textCursor();
+    if (!d->m_inBlockSelectionMode && !tc.hasSelection()) {
+        // see under indent()!
+        curPos = tc.position();
+        moveCursor(QTextCursor::EndOfLine);
+        d->handleHomeKey(false);
+    }
     doSetTextCursor(textDocument()->unindent(textCursor(), d->m_inBlockSelectionMode,
                                              d->m_blockSelection.positionColumn, &offset),
                     d->m_inBlockSelectionMode);
     if (d->m_inBlockSelectionMode) {
         d->m_blockSelection.anchorColumn += offset;
         d->m_blockSelection.positionColumn += offset;
+    } else if (curPos != -1) {
+        tc.setPosition(curPos + offset);
+        setTextCursor(tc);
     }
 }
 
