diff --git share/qtcreator/themes/default.creatortheme share/qtcreator/themes/default.creatortheme
index 874d74c900..3546821f29 100644
--- share/qtcreator/themes/default.creatortheme
+++ share/qtcreator/themes/default.creatortheme
@@ -41,6 +41,7 @@ DoubleTabWidget2ndTabInactiveTextColor=ff000000
 EditorPlaceholderColor=ffe0dcd8
 FancyToolBarSeparatorColor=60ffffff
 FancyTabBarBackgroundColor=ffff0000
+FancyTabBarSelectedBackgroundColor=ff151515
 FancyTabWidgetDisabledSelectedTextColor=textDisabled
 FancyTabWidgetDisabledUnselectedTextColor=textDisabled
 FancyTabWidgetEnabledSelectedTextColor=ff3c3c3c
