diff --git src/plugins/help/helpplugin.cpp src/plugins/help/helpplugin.cpp
index 20570594835e6f7a73f100cf7c08c896355d96d1..fccf2a1df763da47976e4da51a401e9ff6c3f05d 100644
--- src/plugins/help/helpplugin.cpp
+++ src/plugins/help/helpplugin.cpp
@@ -98,6 +98,10 @@
 
 #include <QHelpEngine>
 
+#ifdef QTC_MAC_NATIVE_HELPVIEWER
+#include <QGuiApplication>
+#endif
+
 #include <functional>
 
 using namespace Help::Internal;
@@ -375,12 +379,14 @@ HelpViewer *HelpPlugin::createHelpViewer(qreal zoom)
     factories.append(qMakePair(QByteArray("textbrowser"), []() { return new TextBrowserHelpViewer(); }));
 
 #ifdef QTC_MAC_NATIVE_HELPVIEWER
+    if (QGuiApplication::platformName().contains(QLatin1String("cocoa"))) {
     // default setting
 #ifdef QTC_MAC_NATIVE_HELPVIEWER_DEFAULT
-     factories.prepend(qMakePair(QByteArray("native"), []() { return new MacWebKitHelpViewer(); }));
+         factories.prepend(qMakePair(QByteArray("native"), []() { return new MacWebKitHelpViewer(); }));
 #else
-     factories.append(qMakePair(QByteArray("native"), []() { return new MacWebKitHelpViewer(); }));
+         factories.append(qMakePair(QByteArray("native"), []() { return new MacWebKitHelpViewer(); }));
 #endif
+    }
 #endif
 
     HelpViewer *viewer = nullptr;
