diff --git src/libs/utils/touchbar/touchbar.pri src/libs/utils/touchbar/touchbar.pri
index 8d60ee423b4d96069e544fbd873f8d300198a780..37968fbf8528be3ee8161f9dc74b10ab0d4168cf 100644
--- src/libs/utils/touchbar/touchbar.pri
+++ src/libs/utils/touchbar/touchbar.pri
@@ -1,6 +1,6 @@
 HEADERS += $$PWD/touchbar.h
 
-macos {
+macosNew {
     HEADERS += \
         $$PWD/touchbar_mac_p.h \
         $$PWD/touchbar_appdelegate_mac_p.h
diff --git src/libs/utils/utils.qbs src/libs/utils/utils.qbs
index 2798f3ab1653fd28b407862b174568ffa3035cb3..0f896b153e2581b6caedee74d2d8a6461ad30c0b 100644
--- src/libs/utils/utils.qbs
+++ src/libs/utils/utils.qbs
@@ -370,7 +370,7 @@ Project {
             files: "touchbar.h"
             Group {
                 name: "TouchBar implementation"
-                condition: qbs.targetOS.contains("macos")
+                condition: qbs.targetOS.contains("macosNew")
                 files: [
                     "touchbar_appdelegate_mac_p.h",
                     "touchbar_mac_p.h",
@@ -380,7 +380,7 @@ Project {
             }
             Group {
                 name: "TouchBar stub"
-                condition: !qbs.targetOS.contains("macos")
+                condition: !qbs.targetOS.contains("macosNew")
                 files: "touchbar.cpp"
             }
         }
