diff --git a/core/smb4kcustomoptionsmanager_p.cpp b/core/smb4kcustomoptionsmanager_p.cpp
index 17ea961..364280c 100644
--- a/core/smb4kcustomoptionsmanager_p.cpp
+++ b/core/smb4kcustomoptionsmanager_p.cpp
@@ -54,6 +54,7 @@
 #include <KCoreAddons/KUser>
 #include <KConfigGui/KWindowConfig>
 #include <KIconThemes/KIconLoader>
+#include <KDELibs4Support/ktabwidget.h>
 
 
 Smb4KCustomOptionsDialog::Smb4KCustomOptionsDialog(Smb4KCustomOptions *options, QWidget *parent)
