--- orig.CMakeLists.txt	2016-10-29 14:57:12.000000000 +0200
+++ CMakeLists.txt	2016-10-29 15:14:51.512392248 +0200
@@ -47,11 +47,11 @@
     set(BUILD_TESTING OFF CACHE BOOL "Build the testing tree.")
 endif()
 
-find_package(KDevPlatform 4.90.90)
-set_package_properties(KDevPlatform PROPERTIES
-    PURPOSE "Required for plasmate"
-    TYPE OPTIONAL
-)
+# find_package(KDevPlatform 4.90.90)
+# set_package_properties(KDevPlatform PROPERTIES
+#    PURPOSE "Required for plasmate"
+#    TYPE OPTIONAL
+#)
 
 find_package(KF5Kirigami 1.0)
 set_package_properties(KF5Kirigami PROPERTIES
@@ -98,9 +98,9 @@
 add_subdirectory(lookandfeelexplorer)
 #add_subdirectory(remote-widgets-browser)
 #add_subdirectory(wallpaperviewer)
-if (KDevPlatform_FOUND AND Qt5WebKit_FOUND AND Qt5WebKitWidgets_FOUND)
-    add_subdirectory(plasmate)
-endif()
+#if (KDevPlatform_FOUND AND Qt5WebKit_FOUND AND Qt5WebKitWidgets_FOUND)
+#    add_subdirectory(plasmate)
+#endif()
 
 feature_summary(WHAT ALL FATAL_ON_MISSING_REQUIRED_PACKAGES)
 
