--- plasmate/plugins/savesystemview/git.cpp.orig	2017-03-04 19:03:14.000000000 +0100
+++ plasmate/plugins/savesystemview/git.cpp	2017-03-04 19:03:21.000000000 +0100
@@ -108,7 +108,7 @@
     m_branchesModel->initialize(m_branching, m_repositoryPath);
 
     KDevelop::VcsRevision rev = KDevelop::VcsRevision::createSpecialRevision(KDevelop::VcsRevision::Base);
-    KDevelop::VcsEventModel *vcsEventModel = new KDevelop::VcsEventModel(basicVersionControl, rev, m_repositoryPath, this);
+    KDevelop::VcsEventLogModel *vcsEventModel = new KDevelop::VcsEventLogModel(basicVersionControl, rev, m_repositoryPath, this);
     m_commitsModel->setSourceModel(vcsEventModel);
 
     return true;
