diff --git app/kdevideextension.cpp app/kdevideextension.cpp
index 91778ea..910f974 100644
--- app/kdevideextension.cpp
+++ app/kdevideextension.cpp
@@ -40,7 +40,7 @@ QString KDevIDEExtension::xmlFile()
 
 QString KDevIDEExtension::binaryPath()
 {
-    return QApplication::applicationDirPath() + "/kdevelop";
+    return QApplication::applicationDirPath() + "/kdevelop5";
 }
 
 KDevelop::AreaParams KDevIDEExtension::defaultArea()
