diff --git app/main.cpp app/main.cpp
index 1129e65..f1273d0 100644
--- app/main.cpp
+++ app/main.cpp
@@ -498,6 +593,11 @@ int main( int argc, char *argv[] )
     }
 
     KDevelopApplication app(argc, argv);
+#ifdef Q_OS_MACOS
+    if (!QGuiApplication::platformName().contains(QLatin1String("cocoa"))) {
+        QCoreApplication::setAttribute(Qt::AA_DontUseNativeMenuBar);
+    }
+#endif
 
     KCrash::initialize();
 
