diff --git formatters/customscript/customscript_plugin.cpp formatters/customscript/customscript_plugin.cpp
index 168581d082..bf68642c2a 100644
--- formatters/customscript/customscript_plugin.cpp
+++ formatters/customscript/customscript_plugin.cpp
@@ -253,12 +253,12 @@ KDevelop::SourceFormatterStyle CustomScriptPlugin::predefinedStyle(const QString
 		result.setCaption(i18n("Gnu Indent: Original Berkeley indent style"));
 		result.setContent("indent -orig");
 		result.setUsePreview(true);
-	} else if(name == "kdev_format_source") {
-		result.setCaption("KDevelop: kdev_format_source");
-		result.setContent("kdev_format_source $FILE $TMPFILE");
+	} else if(name == "kdev_format_source" || name == "kdev_format_source5") {
+		result.setCaption("KDevelop: kdev_format_source5");
+		result.setContent("kdev_format_source5 $FILE $TMPFILE");
 		result.setUsePreview(false);
 		result.setDescription(i18n( "Description:<br />"
-									"<b>kdev_format_source</b> is a script bundled with KDevelop "
+									"<b>kdev_format_source5</b> is a script bundled with KDevelop "
 			                        "which allows using fine-grained formatting rules by placing "
 									"meta-files called <b>format_sources</b> into the file-system.<br /><br />"
 									"Each line of the <b>format_sources</b> files defines a list of wildcards "
@@ -286,6 +286,7 @@ QList<KDevelop::SourceFormatterStyle> CustomScriptPlugin::predefinedStyles()
 {
 	QList<KDevelop::SourceFormatterStyle> styles = stylesFromLanguagePlugins();
 	styles << predefinedStyle("kdev_format_source");
+	styles << predefinedStyle("kdev_format_source5");
 	styles << predefinedStyle("GNU_indent_GNU");
 	styles << predefinedStyle("GNU_indent_KR");
 	styles << predefinedStyle("GNU_indent_orig");
