diff --git languages/clang/duchain/clangproblem.cpp languages/clang/duchain/clangproblem.cpp
index 7afbb50..a60a069 100644
--- languages/clang/duchain/clangproblem.cpp
+++ languages/clang/duchain/clangproblem.cpp
@@ -31,6 +31,8 @@
 
 #include <KLocalizedString>
 
+#include <typeinfo>
+
 using namespace KDevelop;
 
 namespace {
@@ -183,11 +185,22 @@ ClangFixits ClangProblem::allFixits() const
 {
     ClangFixits result;
     result << m_fixits;
+    int ok = 0, err = 0;
 
     for (const IProblem::Ptr& diagnostic : diagnostics()) {
         const Ptr problem(dynamic_cast<ClangProblem*>(diagnostic.data()));
-        Q_ASSERT(problem);
-        result << problem->allFixits();
+        if (problem) {
+            ok += 1;
+            result << problem->allFixits();
+        } else {
+            const auto ptr = diagnostic.data();
+            qWarning() << "\tdynamic cast to ClangProblem failed of" << ptr
+                << "type" << typeid(*ptr).name() << ((ClangProblem*)ptr)->toString();
+            err += 1;
+        }
+    }
+    if (ok || err) {
+        qWarning() << Q_FUNC_INFO << "ClangProblems added:" << ok << "; failed:" << err;
     }
     return result;
 }
