--- src/orig.notifybyaudio.cpp	2016-05-29 17:42:26.000000000 +0200
+++ src/notifybyaudio.cpp	2016-05-29 18:01:01.000000000 +0200
@@ -50,18 +50,25 @@
 
 void NotifyByAudio::notify(KNotification *notification, KNotifyConfig *config)
 {
-    if (!m_audioOutput) {
-        m_audioOutput = new Phonon::AudioOutput(Phonon::NotificationCategory, this);
-    }
     QString soundFilename = config->readEntry(QStringLiteral("Sound"));
     if (soundFilename.isEmpty()) {
         qCWarning(LOG_KNOTIFICATIONS) << "Audio notification requested because" << notification->text()
+            << "/" << config->eventid
             << ", but no sound file provided in file" << config->configfile->name() << ", aborting audio notification";
         QApplication::beep();
         finish(notification);
         return;
     }
 
+    if (soundFilename.compare(QStringLiteral("beep"), Qt::CaseInsensitive) == 0) {
+        QApplication::beep();
+        finish(notification);
+        return;
+    }
+
+    if (!m_audioOutput) {
+        m_audioOutput = new Phonon::AudioOutput(Phonon::NotificationCategory, this);
+    }
     QUrl soundURL = QUrl(soundFilename); // this CTOR accepts both absolute paths (/usr/share/sounds/blabla.ogg and blabla.ogg) w/o screwing the scheme
     if (soundURL.isRelative() && !soundURL.toString().startsWith('/')) { // QUrl considers url.scheme.isEmpty() == url.isRelative()
         soundURL = QUrl::fromLocalFile(QStandardPaths::locate(QStandardPaths::GenericDataLocation, QStringLiteral("sounds/") + soundFilename));
