diff --git src/kstatusnotifieritem.cpp src/kstatusnotifieritem.cpp
index ebd4818..d5d2a63 100644
--- src/kstatusnotifieritem.cpp
+++ src/kstatusnotifieritem.cpp
@@ -1004,13 +1004,20 @@ void KStatusNotifierItemPrivate::syncLegacySystemTrayIcon()
             }
         }
     } else {
+#ifdef Q_OS_MACOS
+        if (!iconName.isNull()) {
+            QIcon theIcon = QIcon::fromTheme(iconName);
+            systemTrayIcon->setIconWithMask(theIcon, status==KStatusNotifierItem::Passive);
+        } else {
+            systemTrayIcon->setIconWithMask(icon, status==KStatusNotifierItem::Passive);
+        }
+        QtMac::setBadgeLabelText(QString());
+#else
         if (!iconName.isNull()) {
             systemTrayIcon->setIcon(QIcon::fromTheme(iconName));
         } else {
             systemTrayIcon->setIcon(icon);
         }
-#ifdef Q_OS_MACOS
-        QtMac::setBadgeLabelText(QString());
 #endif
     }
 
diff --git src/kstatusnotifieritemprivate_p.h src/kstatusnotifieritemprivate_p.h
index 87e909d..5a0c3fb 100644
--- src/kstatusnotifieritemprivate_p.h
+++ src/kstatusnotifieritemprivate_p.h
@@ -82,6 +82,14 @@ public:
         connect(movie, SIGNAL(frameChanged(int)), this, SLOT(slotNewFrame()));
     }
 
+    void setIconWithMask(QIcon &icon, bool isMask)
+    {
+#if QT_VERSION >= QT_VERSION_CHECK(5,6,0)
+        icon.setIsMask(isMask);
+#endif
+        QSystemTrayIcon::setIcon(icon);
+    }
+
 Q_SIGNALS:
     void wheel(int);
 
