--- orig.CMakeLists.txt	2019-09-12 10:29:50.000000000 +0200
+++ CMakeLists.txt	2019-09-23 22:07:32.000000000 +0200
@@ -54,7 +54,7 @@
 include(KDEFrameworkCompilerSettings NO_POLICY_SCOPE)
 include(KDECMakeSettings)
 
-if (NOT APPLE AND NOT WIN32)
+if ((NOT APPLE OR APPLE_FORCE_X11) AND NOT WIN32)
     find_package(X11)
 endif()
 
--- src/orig.kpassivepopup.cpp	2019-07-07 20:36:24.000000000 +0200
+++ src/kpassivepopup.cpp	2019-09-23 22:15:40.000000000 +0200
@@ -40,6 +40,7 @@
 
 #if HAVE_X11
 #include <QX11Info>
+#define KWINDOWSYSTEM_HAVE_X11 1
 #include <netwm.h>
 #endif
 
