--- qtwebengine/src/3rdparty/chromium/device/BUILD.gn.orig	2016-05-26 14:58:54.000000000 +0200
+++ qtwebengine/src/3rdparty/chromium/device/BUILD.gn	2016-10-31 14:40:37.000000000 +0100
@@ -20,28 +20,6 @@
   sources = [
     "battery/battery_status_manager_win_unittest.cc",
     "battery/battery_status_service_unittest.cc",
-    "bluetooth/bluetooth_adapter_mac_unittest.mm",
-    "bluetooth/bluetooth_adapter_profile_chromeos_unittest.cc",
-    "bluetooth/bluetooth_adapter_unittest.cc",
-    "bluetooth/bluetooth_adapter_win_unittest.cc",
-    "bluetooth/bluetooth_advertisement_chromeos_unittest.cc",
-    "bluetooth/bluetooth_audio_sink_chromeos_unittest.cc",
-    "bluetooth/bluetooth_chromeos_unittest.cc",
-    "bluetooth/bluetooth_device_unittest.cc",
-    "bluetooth/bluetooth_device_win_unittest.cc",
-    "bluetooth/bluetooth_discovery_filter_unittest.cc",
-    "bluetooth/bluetooth_gatt_chromeos_unittest.cc",
-    "bluetooth/bluetooth_low_energy_win_unittest.cc",
-    "bluetooth/bluetooth_service_record_win_unittest.cc",
-    "bluetooth/bluetooth_socket_chromeos_unittest.cc",
-    "bluetooth/bluetooth_task_manager_win_unittest.cc",
-    "bluetooth/bluetooth_uuid_unittest.cc",
-    "bluetooth/test/bluetooth_test.cc",
-    "bluetooth/test/bluetooth_test.h",
-    "bluetooth/test/bluetooth_test_android.cc",
-    "bluetooth/test/bluetooth_test_android.h",
-    "bluetooth/test/test_bluetooth_adapter_observer.cc",
-    "bluetooth/test/test_bluetooth_adapter_observer.h",
     "nfc/nfc_chromeos_unittest.cc",
     "nfc/nfc_ndef_record_unittest.cc",
     "test/run_all_unittests.cc",
@@ -51,7 +29,6 @@
     "//base/test:test_support",
     "//device/battery",
     "//device/battery:mojo_bindings",
-    "//device/bluetooth",
     "//device/nfc",
     "//mojo/environment:chromium",
     "//net",
@@ -139,15 +116,15 @@
   }
 
   if (is_mac) {
-    libs = [ "IOBluetooth.framework" ]
+    # libs = [ "IOBluetooth.framework" ]
 
     # In the OSX 10.10 SDK, CoreBluetooth became a top level framework.
     # Previously, it was nested in IOBluetooth. In order for Chrome to run on
     # OSes older than OSX 10.10, the top level CoreBluetooth framework must be
     # weakly linked.
-    if (mac_sdk_version == "10.10") {
-      ldflags = [ "-weak_framework CoreBluetooth" ]
-    }
+    # if (mac_sdk_version == "10.10") {
+    #   ldflags = [ "-weak_framework CoreBluetooth" ]
+    # }
   }
 }
 
