diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7da8cfb..0652d60 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -34,6 +34,7 @@ include(KDEInstallDirs)
 include(KDECompilerSettings NO_POLICY_SCOPE)
 include(KDECMakeSettings)
 include(FeatureSummary)
+include(ECMAddAppIcon)
 
 # use sane compile flags
 add_definitions(
@@ -121,6 +122,9 @@ ki18n_wrap_ui(krdc_SRCS
     config/general.ui
 )
 
+file(GLOB krdc_ICON "${CMAKE_CURRENT_BINARY_DIR}/icons/*.png")
+ecm_add_app_icon(krdc_SRCS ICONS ${krdc_ICON})
+
 ecm_qt_declare_logging_category(krdc_SRCS HEADER krdc_debug.h IDENTIFIER KRDC CATEGORY_NAME KRDC)
 
 add_executable(krdc ${krdc_SRCS})
diff --git a/main.cpp b/main.cpp
index 4ec1f4e..82e3d7b 100644
--- a/main.cpp
+++ b/main.cpp
@@ -101,7 +101,7 @@ int main(int argc, char **argv)
     app.setApplicationDisplayName(aboutData.displayName());
     app.setOrganizationDomain(aboutData.organizationDomain());
     app.setApplicationVersion(aboutData.version());
-    app.setWindowIcon(QIcon::fromTheme(appName));
+    app.setWindowIcon(QIcon::fromTheme(appName, app.windowIcon()));
 
     QCommandLineParser parser;
     parser.addVersionOption();
