diff --git a/CMakeLists.txt b/CMakeLists.txt
index 236f1fbab171b7bad653033fe1ed0192cb2079d9..f8b8f26218c66158d5fafb1fa5d84979f2daf56d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -19,6 +19,7 @@ include(KDECompilerSettings NO_POLICY_SCOPE)
 include(ECMSetupVersion)
 include(ECMQtDeclareLoggingCategory)
 include(FeatureSummary)
+include(ECMAddAppIcon)
 
 find_package(KF5 ${KF5_MIN_VERSION} REQUIRED COMPONENTS
     Config
@@ -144,6 +145,9 @@ ki18n_wrap_ui(krdc_SRCS
     config/general.ui
 )
 
+file(GLOB krdc_ICON "${CMAKE_CURRENT_BINARY_DIR}/icons/*.png")
+ecm_add_app_icon(krdc_SRCS ICONS ${krdc_ICON})
+
 ecm_qt_declare_logging_category(krdc_SRCS HEADER krdc_debug.h IDENTIFIER KRDC CATEGORY_NAME KRDC)
 
 add_executable(krdc ${krdc_SRCS})
diff --git a/main.cpp b/main.cpp
index 3adbc64d093d7e8ec971aec72d659521a2dfc016..c4a60e67c1df6481d203516d7405431adb7e3da1 100644
--- a/main.cpp
+++ b/main.cpp
@@ -97,7 +97,7 @@ int main(int argc, char **argv)
     aboutData.setOrganizationDomain("kde.org");
 
     KAboutData::setApplicationData(aboutData);
-    app.setWindowIcon(QIcon::fromTheme(appName));
+    app.setWindowIcon(QIcon::fromTheme(appName, app.windowIcon()));
 
     QCommandLineParser parser;
     aboutData.setupCommandLine(&parser);
