diff --git a/vnc/vncclientthread.cpp b/vnc/vncclientthread.cpp
index e4e5032..4879f32 100644
--- a/vnc/vncclientthread.cpp
+++ b/vnc/vncclientthread.cpp
@@ -607,11 +607,13 @@ void VncClientThread::clientSetKeepalive()
         return;
     }
 
+#ifndef __APPLE__
     optval = m_keepalive.intervalSeconds;
     if (setsockopt(cl->sock, IPPROTO_TCP, TCP_KEEPIDLE, &optval, optlen) < 0) {
         qCritical(KRDC) << "setsockopt(TCP_KEEPIDLE)" << strerror(errno);
         return;
     }
+#endif
 
     optval = m_keepalive.intervalSeconds;
     if (setsockopt(cl->sock, IPPROTO_TCP, TCP_KEEPINTVL, &optval, optlen) < 0) {
