Subject: Add SOVERSION field to built shared library.
Origin: https://github.com/telegramdesktop/libtgvoip/commit/f8ba1605d0f62d6382ac4ebfa8e0e13c1795ccd7
From: Vitaly Zaitsev <vitaly@easycoding.org>
Acked-By: Nicholas Guriev <guriev-ns@ya.ru>
Date: Thu, 9 Jan 2020 19:14:36 +0100
Last-Update: Mon, 15 Jun 2020 08:16:54 +0300

--- a/Makefile.am
+++ b/Makefile.am
@@ -748,6 +748,7 @@ else
 CFLAGS += -DTGVOIP_NO_DSP
 endif
 
+libtgvoip_la_LDFLAGS = -version-number @LIBTGVOIP_MAJOR_VERSION@:@LIBTGVOIP_MINOR_VERSION@:@LIBTGVOIP_PATCH_VERSION@
 libtgvoip_la_SOURCES = $(SRC) $(TGVOIP_HDRS)
 tgvoipincludedir = $(includedir)/tgvoip
 nobase_tgvoipinclude_HEADERS = $(TGVOIP_HDRS)
--- a/configure.ac
+++ b/configure.ac
@@ -1,8 +1,14 @@
 #                                               -*- Autoconf -*-
 # Process this file with autoconf to produce a configure script.
 
+# Define version information
+m4_define([libtgvoip_major_version], 2)
+m4_define([libtgvoip_minor_version], 4)
+m4_define([libtgvoip_patch_version], 4)
+m4_define([libtgvoip_version],[libtgvoip_major_version.libtgvoip_minor_version.libtgvoip_patch_version])
+
 AC_PREREQ([2.69])
-AC_INIT([libtgvoip], [2.4.4], [https://github.com/grishka/libtgvoip/issues])
+AC_INIT([libtgvoip], [libtgvoip_version], [https://github.com/grishka/libtgvoip/issues])
 AC_CONFIG_SRCDIR([config.h.in])
 AC_CONFIG_HEADERS([config.h])
 AM_INIT_AUTOMAKE([subdir-objects])
@@ -24,6 +30,10 @@ AC_CHECK_LIB([pthread], [pthread_create]
 
 AC_CANONICAL_HOST
 
+AC_SUBST(LIBTGVOIP_MAJOR_VERSION, libtgvoip_major_version)
+AC_SUBST(LIBTGVOIP_MINOR_VERSION, libtgvoip_minor_version)
+AC_SUBST(LIBTGVOIP_PATCH_VERSION, libtgvoip_patch_version)
+
 AS_CASE([$host_cpu],
 	[i?86], [cpu_x86=yes],
 	[x86_64], [cpu_x86=yes],
