diff --git orig.CMakeLists.txt CMakeLists.txt
index 2a23c9b..0040e44 100644
--- orig.CMakeLists.txt
+++ CMakeLists.txt
@@ -145,8 +145,8 @@ set(CAIRO_VERSION "1.16.0")
 set(GLIB_REQUIRED "2.64")
 set(GTK_REQUIRED "3.24")
 set(GDK_PIXBUF_REQUIRED "2.40")
-set(FREETYPE_VERSION "2.10")
-set(FONTCONFIG_VERSION "2.13")
+set(FREETYPE_VERSION "2.8")
+set(FONTCONFIG_VERSION "2.12")
 
 find_package(Freetype ${FREETYPE_VERSION} REQUIRED)
 if(WITH_FONTCONFIGURATION_FONTCONFIG)
@@ -208,7 +208,7 @@ else()
   message(FATAL_ERROR "Invalid ENABLE_DCTDECODER value.")
 endif()
 
-set(QT5_VERSION "5.12") # Update QT_DISABLE_DEPRECATED_BEFORE in qt5/CMakeLists.txt when changing this
+set(QT5_VERSION "5.8") # Update QT_DISABLE_DEPRECATED_BEFORE in qt5/CMakeLists.txt when changing this
 find_soft_mandatory_package(ENABLE_QT5 Qt5Core ${QT5_VERSION})
 find_soft_mandatory_package(ENABLE_QT5 Qt5Gui ${QT5_VERSION})
 find_soft_mandatory_package(ENABLE_QT5 Qt5Xml ${QT5_VERSION})
diff --git qt5/orig.CMakeLists.txt qt5/CMakeLists.txt
index 80b0585..ce0026f 100644
--- qt5/orig.CMakeLists.txt
+++ qt5/CMakeLists.txt
@@ -5,7 +5,7 @@ if(ENABLE_QT_STRICT_ITERATORS)
   add_definitions(-DQT_STRICT_ITERATORS)
 endif()
 
-add_definitions(-DQT_DISABLE_DEPRECATED_BEFORE=0x050C00)
+add_definitions(-DQT_DISABLE_DEPRECATED_BEFORE=0x050800)
 add_definitions(-DQT_NO_DEPRECATED_WARNINGS)
 
 add_subdirectory(src)
