diff --git src/widgets/JobDialog.qml src/widgets/JobDialog.qml
index c32f54e6263cf0374594f49e4b82cafe0f10dd31..c53d21a6145e203dcf02f91c6ffa2e229e234e74 100644
--- src/widgets/JobDialog.qml
+++ src/widgets/JobDialog.qml
@@ -25,7 +25,7 @@ ApplicationWindow
     id: window
 
     flags: Qt.Dialog
-    title: configuration.pluginName
+//     title: configuration.pluginName
 
     width: Math.max(screen.width/6, 200)
     height: Math.max(screen.height/4, 300)
@@ -86,7 +86,7 @@ ApplicationWindow
                 }
                 Button {
                     text: i18n("Send")
-                    icon.name: "document-send"
+//                     icon.name: "document-send"
                     enabled: window.configuration && window.configuration.isReady
                     onClicked: {
                         view.pop();
@@ -95,7 +95,7 @@ ApplicationWindow
                 }
                 Button {
                     text: i18n("Cancel")
-                    icon.name: "dialog-cancel"
+//                     icon.name: "dialog-cancel"
                     onClicked: {
                         window.q.finished(null, 1 /* KIO::ERR_USER_CANCELED */, i18n("Configuration cancelled"));
                         window.visible = false;
