--- lib/orig.CMakeLists.txt	2013-05-01 00:45:43.000000000 +0200
+++ lib/CMakeLists.txt	2015-12-06 22:18:00.000000000 +0100
@@ -160,7 +160,7 @@
     publisherfield.h
     topic.h
     ${CMAKE_CURRENT_BINARY_DIR}/version.h
-    DESTINATION include/attica COMPONENT Devel
+    DESTINATION include@IPREFIX@/attica COMPONENT Devel
 )
 
 # Install package config file
Exit 1
--- lib/cmake/orig.libattica.pc.cmake	2013-05-01 00:45:43.000000000 +0200
+++ lib/cmake/libattica.pc.cmake	2015-12-06 22:18:35.000000000 +0100
@@ -1,12 +1,12 @@
 prefix=${CMAKE_INSTALL_PREFIX}
 exec_prefix=${CMAKE_INSTALL_PREFIX}/bin
 libdir=${LIB_DESTINATION}
-includedir=${CMAKE_INSTALL_PREFIX}/include
+includedir=${CMAKE_INSTALL_PREFIX}/include@IPREFIX@
 
 Name: libattica
 Description: Qt library to access Open Collaboration Services
 #Requires:
 Version: ${CMAKE_LIBATTICA_VERSION_MAJOR}.${CMAKE_LIBATTICA_VERSION_MINOR}.${CMAKE_LIBATTICA_VERSION_PATCH}
 Libs: -L${LIB_DESTINATION} -lattica
-Cflags: -I${CMAKE_INSTALL_PREFIX}/include
+Cflags: -I${CMAKE_INSTALL_PREFIX}/include@IPREFIX@
 
