diff --git src/plugins/phabricator/quick/difflistmodel.cpp src/plugins/phabricator/quick/difflistmodel.cpp
index d0f2beb49a85e7d7ffeb11f5ffb1f7227be27258..2d255a2c30993101e5478b52b61865017dfda386 100644
--- src/plugins/phabricator/quick/difflistmodel.cpp
+++ src/plugins/phabricator/quick/difflistmodel.cpp
@@ -35,15 +35,15 @@ DiffListModel::DiffListModel(QObject* parent)
 
 void DiffListModel::refresh()
 {
-    beginResetModel();
-    m_values.clear();
-    endResetModel();
-
     if (m_tempDir) {
         qCritical() << "DiffListModel::refresh() called while still active!";
         return;
     }
 
+    beginResetModel();
+    m_values.clear();
+    endResetModel();
+
     // our CWD should be the directory from which the application was launched, which
     // may or may not be a git, mercurial or svn working copy, so we create a temporary
     // directory in which we initialise a git repository. This may be an empty repo.
@@ -93,17 +93,12 @@ void DiffListModel::receivedDiffRevs(KJob* job)
 
     const auto diffRevList = dynamic_cast<Phabricator::DiffRevList*>(job);
     const auto revs = diffRevList->reviews();
-    QVector<Value> tmpValues;
-    foreach (const auto review, revs) {
-        auto status = diffRevList->statusMap()[review.second];
-        tmpValues += Value { review.second, review.first, status };
-    }
-    std::sort(tmpValues.begin(), tmpValues.end());
 
     beginResetModel();
     m_values.clear();
-    foreach (const auto value, tmpValues) {
-        m_values += value;
+    for (const auto review : revs) {
+        auto status = diffRevList->statusMap()[review.second];
+        m_values += Value { review.second, review.first, status };
     }
     endResetModel();
 
diff --git src/plugins/phabricator/quick/difflistmodel.h src/plugins/phabricator/quick/difflistmodel.h
index 751059f56518bdf6221dc90a492b54c43ae39c1a..7c6d6bc0e26007e8a7eef592d34bc73a3348aa3a 100644
--- src/plugins/phabricator/quick/difflistmodel.h
+++ src/plugins/phabricator/quick/difflistmodel.h
@@ -54,10 +54,6 @@ class DiffListModel : public QAbstractListModel
             QVariant summary;
             QVariant id;
             QVariant status;
-            inline bool operator<(const DiffListModel::Value &b) const
-            {
-                return summary.toString().localeAwareCompare(b.summary.toString());
-            }
 #ifndef QT_NO_DEBUG_STREAM
             operator QString() const
             {
