diff --git src/quick/RunningJob.qml src/quick/RunningJob.qml
index 9352755..88884ec 100644
--- src/quick/RunningJob.qml
+++ src/quick/RunningJob.qml
@@ -43,7 +43,7 @@ ColumnLayout {
     }
     ProgressBar {
         //FIXME: this is not really working yet, as QML doesn't understand ulong
-        value: root.job.percent
+        value: root.job.percent * 1.0
         to: 100
         Layout.fillWidth: true
     }
