diff --git src/plugins/CMakeLists.txt src/plugins/CMakeLists.txt
index c75e407..0771091 100644
--- src/plugins/CMakeLists.txt
+++ src/plugins/CMakeLists.txt
@@ -22,9 +22,10 @@ set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake" ${CMAKE_MODULE_PATH})
 include(QMLModules)
 find_qmlmodule(Ubuntu.OnlineAccounts 0.1)
 find_qmlmodule(org.kde.kdeconnect 1.0)
-find_qmlmodule(org.kde.kquickcontrolsaddons 2.0)
+#find_qmlmodule(org.kde.kquickcontrolsaddons 2.0)
+find_package(KF5Declarative)
 
-find_package(KAccounts)
+# find_package(KAccounts)
 find_package(KF5KIO ${KF5_DEP_VERSION} REQUIRED)
 
 set_package_properties(KAccounts PROPERTIES
