diff --git src/plugins/cmake/FindQMLModule.cmake.in src/plugins/cmake/FindQMLModule.cmake.in
index a983054..a18446b 100644
--- src/plugins/cmake/FindQMLModule.cmake.in
+++ src/plugins/cmake/FindQMLModule.cmake.in
@@ -27,7 +27,7 @@

 include(ECMFindModuleHelpersStub)

-execute_process(COMMAND qmlplugindump "@MODULE_NAME@" "@VERSION@" ERROR_VARIABLE ERRORS_OUTPUT OUTPUT_VARIABLE DISREGARD_VARIABLE RESULT_VARIABLE ExitCode)
+execute_process(COMMAND @QTBINDIR@/qmlplugindump "@MODULE_NAME@" "@VERSION@" ERROR_VARIABLE ERRORS_OUTPUT OUTPUT_VARIABLE DISREGARD_VARIABLE RESULT_VARIABLE ExitCode)

 if(ExitCode EQUAL 0)
     set(@GENMODULE@_FOUND TRUE)
