diff -r b193168bad98 src/xxdiff.pro
--- src/xxdiff.pro	Wed Apr 15 14:02:15 2015 -0400
+++ src/xxdiff.pro	Thu May 14 17:25:17 2015 +0200
@@ -40,9 +40,8 @@
 
 
 # To enable KDE support, uncomment the following lines:
-#QMAKE_CFLAGS += -DXX_KDE -I/usr/include/kde
-#QMAKE_CXXFLAGS += -DXX_KDE -I/usr/include/kde
-#QMAKE_LIBS += -lkdeui -lkio -lqt-mt
+DEFINES += XX_KDE
+QMAKE_LIBS += -lkdeui -lkio
 
 
 # Enable debugging modes within xxdiff.
