--- src/plugins/bearer/corewlan/corewlan.pro.orig	2012-03-14 10:01:12.000000000 -0400
+++ src/plugins/bearer/corewlan/corewlan.pro	2012-04-24 09:23:47.000000000 -0400
@@ -7,7 +7,7 @@ LIBS += -framework Foundation -framework SystemConfiguration
 contains(QT_CONFIG, corewlan) {
     LIBS += -framework CoreWLAN
 
-    isEmpty(QMAKE_MAC_SDK)|contains(QMAKE_MAC_SDK, ".*OSX10\.[6789]\.sdk/?$") {
+    isEmpty(QMAKE_MAC_SDK)|contains(QMAKE_MAC_SDK, "@DEVELOPER_SDK@/?$") {
          LIBS += -framework Security
     }
 }
