--- src/gui/util/orig.qsystemtrayicon_mac.mm	2014-04-10 20:37:12.000000000 +0200
+++ src/gui/util/qsystemtrayicon_mac.mm	2014-09-15 19:48:07.000000000 +0200
@@ -341,7 +341,7 @@
 -(void)mousePressed:(NSEvent *)mouseEvent button:(Qt::MouseButton)mouseButton
 {
     down = YES;
-    int clickCount = [mouseEvent clickCount];  
+    int clickCount = [mouseEvent clickCount];
     [self setNeedsDisplay:YES];
 
 #ifndef QT_MAC_USE_COCOA
@@ -532,7 +532,8 @@
             [item setState:action->isChecked() ? NSOnState : NSOffState];
             [item setToolTip:(NSString*)QCFString::toCFStringRef(action->toolTip())];
             const QIcon icon = action->icon();
-            if(!icon.isNull()) {
+		  // RJVB 20140915: don't forget to check against isIconVisibleInMenu()!
+            if(!icon.isNull() && action->isIconVisibleInMenu()) {
 #ifndef QT_MAC_USE_COCOA
                 const short scale = GetMBarHeight();
 #else
