--- src/corelib/io/orig.qfilesystemwatcher.cpp	2014-04-10 20:37:11.000000000 +0200
+++ src/corelib/io/qfilesystemwatcher.cpp	2014-12-11 13:33:27.000000000 +0100
@@ -544,9 +544,9 @@
     if(engine)
         p = engine->addPaths(p, &d->files, &d->directories);
 
-    if (!p.isEmpty())
-        qWarning("QFileSystemWatcher: failed to add paths: %s",
-                 qPrintable(p.join(QLatin1String(", "))));
+//     if (!p.isEmpty())
+//         qWarning("QFileSystemWatcher: failed to add paths: %s",
+//                  qPrintable(p.join(QLatin1String(", "))));
 }
 
 /*!
