--- orig.kcalc.cpp	2016-05-05 23:47:22.000000000 +0200
+++ kcalc.cpp	2016-05-16 22:26:33.000000000 +0200
@@ -2302,7 +2302,7 @@
 	aboutData.addAuthor(i18n("David Johnson"), QString(), "david@usermode.org");
 
 	KAboutData::setApplicationData(aboutData);
-	app.setWindowIcon(QIcon::fromTheme(QLatin1String("accessories-calculator")));
+	app.setWindowIcon(QIcon::fromTheme(QLatin1String("accessories-calculator"), app.windowIcon()));
 
 
 	QCommandLineParser parser;
