--- orig.CMakeLists.txt	2016-05-05 23:47:22.000000000 +0200
+++ CMakeLists.txt	2016-05-17 01:34:13.000000000 +0200
@@ -101,6 +101,9 @@
 kconfig_add_kcfg_files(kcalc_KDEINIT_SRCS kcalc_settings.kcfgc )
 
 #kde4_add_app_icon(kcalc_KDEINIT_SRCS "${KDE4_INSTALL_DIR}/share/icons/oxygen/*/apps/accessories-calculator.png")
+# Sets the icon on Windows and OSX
+file(GLOB ICONS_SRCS "${CMAKE_CURRENT_BINARY_DIR}/icons/*.png")
+ecm_add_app_icon(kcalc_KDEINIT_SRCS ICONS ${ICONS_SRCS})
 
 kf5_add_kdeinit_executable( kcalc ${kcalc_KDEINIT_SRCS})
 
