diff --git sublime/container.cpp sublime/container.cpp
index 43e280ff2d1a75efa60f72a419628ab3e2805fe4..3801509efed2d84c6e48cca386fb15e43bec8e34 100644
--- sublime/container.cpp
+++ sublime/container.cpp
@@ -490,14 +490,12 @@ QWidget* Container::currentWidget() const
 
 void Container::setCurrentWidget(QWidget* w)
 {
-    d->stack->setCurrentWidget(w);
-    //prevent from emitting activateView() signal on tabbar active tab change
-    //this function is called from MainWindow::activateView()
-    //which does the activation without any additional signals
-    {
-        QSignalBlocker blocker(d->tabBar);
-        d->tabBar->setCurrentIndex(d->stack->indexOf(w));
+
+    if (d->stack->currentWidget() == w) {
+        return;
     }
+    d->stack->setCurrentWidget(w);
+    d->tabBar->setCurrentIndex(d->stack->indexOf(w));
     if (View* view = viewForWidget(w))
     {
         statusChanged(view);
diff --git sublime/mainwindow.cpp sublime/mainwindow.cpp
index 38c7d724505fd51aa07693266042690882ebd02c..5d59b68210c1975ea96d60e52c616095cfc0d822 100644
--- sublime/mainwindow.cpp
+++ sublime/mainwindow.cpp
@@ -190,7 +190,7 @@ View *MainWindow::activeToolView() const
 
 void MainWindow::activateView(Sublime::View* view, bool focus)
 {
-    if (!d->viewContainers.contains(view))
+    if (!d->viewContainers.contains(view) || d->activeView == view)
         return;
 
     d->viewContainers[view]->setCurrentWidget(view->widget());
