diff --git shell/openprojectdialog.cpp shell/openprojectdialog.cpp
index 9ccca43..34eefad 100644
--- shell/openprojectdialog.cpp
+++ shell/openprojectdialog.cpp
@@ -75,7 +75,7 @@ OpenProjectDialog::OpenProjectDialog( bool fetch, const QUrl& startUrl, QWidget*
 {
     resize(QSize(700, 500));
 
-    const bool useKdeFileDialog = qEnvironmentVariableIsSet("KDE_FULL_SESSION");
+    const bool useKdeFileDialog = true; /*qEnvironmentVariableIsSet("KDE_FULL_SESSION")*/
     QStringList filters, allEntry;
     QString filterFormat = useKdeFileDialog
                          ? QStringLiteral("%1|%2 (%1)")
diff --git shell/documentcontroller.cpp shell/documentcontroller.cpp
index 0176c2e..5b44837 100644
--- shell/documentcontroller.cpp
+++ shell/documentcontroller.cpp
@@ -138,7 +138,7 @@ struct DocumentControllerPrivate
         auto parent = Core::self()->uiControllerInternal()->defaultMainWindow();
 
         // use special dialogs in a KDE session, native dialogs elsewhere
-        if (qEnvironmentVariableIsSet("KDE_FULL_SESSION")) {
+        /*if (qEnvironmentVariableIsSet("KDE_FULL_SESSION"))*/ {
             const auto result = KEncodingFileDialog::getOpenUrlsAndEncoding(QString(), dir,
                 filter, parent, caption);
             return {result.URLs, result.encoding};
