diff --git a/sublime/container.cpp b/sublime/container.cpp
index b04f6c3..c0c8cb7 100644
--- a/sublime/container.cpp
+++ b/sublime/container.cpp
@@ -42,6 +42,8 @@
 #include "document.h"
 #include <ksqueezedtextlabel.h>
 
+#include <QDebug>
+
 namespace Sublime {
 
 // struct ContainerPrivate
@@ -54,6 +56,9 @@ public:
     ContainerTabBar(Container* container)
         : QTabBar(container), m_container(container)
     {
+        setDocumentMode(true);
+        setUsesScrollButtons(true);
+        setElideMode(Qt::ElideNone);
         installEventFilter(this);
     }
 
