--- samba-4.8.9/source3/lib/orig.system_smbd.c	2018-01-14 21:41:58.000000000 +0100
+++ samba-4.8.9/source3/lib/system_smbd.c	2019-07-27 21:31:26.000000000 +0200
@@ -205,7 +205,11 @@
 			 gid_t primary_gid,
 			 gid_t **ret_groups, uint32_t *p_ngroups)
 {
+#if defined(DARWINOS)
+	int max_grp = 128;
+#else
 	int max_grp = MIN(128, groups_max());
+#endif
 	gid_t stack_groups[max_grp];
 	uint32_t ngrp;
 	gid_t *temp_groups = stack_groups;
