--- samba-4.8.9/orig.wscript_build	2019-07-27 01:39:25.000000000 +0200
+++ samba-4.8.9/wscript_build	2019-07-27 02:53:25.000000000 +0200
@@ -28,15 +28,15 @@
 
 # bld.ENABLE_MAGIC_ORDERING()
 
-bld.env.ABS_TOP_SRCDIR = bld.srcnode.abspath() + '/docs-xml'
-bld.env.ABS_TOP_BUILDDIR = bld.srcnode.abspath() + '/bin/default/docs-xml'
-bld.CONFIGURE_FILE('docs-xml/build/catalog.xml',
-                   ABS_TOP_BUILDDIR = bld.env.ABS_TOP_BUILDDIR,
-                   ABS_TOP_SRCDIR=bld.env.ABS_TOP_SRCDIR)
-bld.env.DOC_VERSION=version.STRING
-bld.CONFIGURE_FILE('docs-xml/build/DTD/samba.build.version',
-                   DOC_VERSION=bld.env.DOC_VERSION)
-bld.RECURSE('docs-xml')
+# bld.env.ABS_TOP_SRCDIR = bld.srcnode.abspath() + '/docs-xml'
+# bld.env.ABS_TOP_BUILDDIR = bld.srcnode.abspath() + '/bin/default/docs-xml'
+# bld.CONFIGURE_FILE('docs-xml/build/catalog.xml',
+#                    ABS_TOP_BUILDDIR = bld.env.ABS_TOP_BUILDDIR,
+#                    ABS_TOP_SRCDIR=bld.env.ABS_TOP_SRCDIR)
+# bld.env.DOC_VERSION=version.STRING
+# bld.CONFIGURE_FILE('docs-xml/build/DTD/samba.build.version',
+#                    DOC_VERSION=bld.env.DOC_VERSION)
+# bld.RECURSE('docs-xml')
 
 bld.RECURSE('lib/replace')
 bld.RECURSE('lib/socket')
