--- samba-4.8.9/lib/replace/system/orig.filesys.h	2018-01-14 21:41:58.000000000 +0100
+++ samba-4.8.9/lib/replace/system/filesys.h	2019-07-27 00:35:07.000000000 +0200
@@ -115,11 +115,13 @@
 #define VXFS_QUOTA
 #endif
 
+#ifndef __APPLE__
 #if HAVE_SYS_ATTRIBUTES_H
 #include <sys/attributes.h>
 #elif HAVE_ATTR_ATTRIBUTES_H
 #include <attr/attributes.h>
 #endif
+#endif
 
 /* mutually exclusive (SuSE 8.2) */
 #if HAVE_ATTR_XATTR_H
--- samba-4.8.9/source3/include/orig.includes.h	2018-01-25 20:55:34.000000000 +0100
+++ samba-4.8.9/source3/include/includes.h	2019-07-27 03:00:43.000000000 +0200
@@ -76,9 +76,11 @@
 #undef HAVE_LDAP
 #endif
 
+#ifndef __APPLE__
 #if HAVE_SYS_ATTRIBUTES_H
 #include <sys/attributes.h>
 #endif
+#endif
 
 #ifndef ENOATTR
 #if defined(ENODATA)
