diff --git src/extension/internal/pdfinput/pdf-input.cpp src/extension/internal/pdfinput/pdf-input.cpp
index 9621e306..6ffef0fb 100644
--- src/extension/internal/pdfinput/pdf-input.cpp
+++ src/extension/internal/pdfinput/pdf-input.cpp
@@ -18,17 +18,17 @@
 #include "pdf-input.h"
 
 #ifdef HAVE_POPPLER
-#include <poppler/goo/GooString.h>
-#include <poppler/ErrorCodes.h>
-#include <poppler/GlobalParams.h>
-#include <poppler/PDFDoc.h>
-#include <poppler/Page.h>
-#include <poppler/Catalog.h>
+#include <goo/GooString.h>
+#include <ErrorCodes.h>
+#include <GlobalParams.h>
+#include <PDFDoc.h>
+#include <Page.h>
+#include <Catalog.h>
 
 #ifdef HAVE_POPPLER_CAIRO
-#include <poppler/glib/poppler.h>
-#include <poppler/glib/poppler-document.h>
-#include <poppler/glib/poppler-page.h>
+#include <glib/poppler.h>
+#include <glib/poppler-document.h>
+#include <glib/poppler-page.h>
 #endif
 
 #include <gtkmm/alignment.h>
diff --git src/extension/internal/pdfinput/pdf-parser.cpp src/extension/internal/pdfinput/pdf-parser.cpp
index 2981d6ee..2413b1e1 100644
--- src/extension/internal/pdfinput/pdf-parser.cpp
+++ src/extension/internal/pdfinput/pdf-parser.cpp
@@ -41,7 +41,7 @@ extern "C" {
 #include "goo/GooString.h"
 #include "GlobalParams.h"
 #include "CharTypes.h"
-#include "Object.h"
+#include "@PREFIX@/libexec/poppler073/include/poppler/Object.h"
 #include "Array.h"
 #include "Dict.h"
 #include "Stream.h"
diff --git src/extension/internal/pdfinput/pdf-parser.h src/extension/internal/pdfinput/pdf-parser.h
index 0aa4c61b..637c310a 100644
--- src/extension/internal/pdfinput/pdf-parser.h
+++ src/extension/internal/pdfinput/pdf-parser.h
@@ -27,7 +27,7 @@ namespace Inkscape {
 using Inkscape::Extension::Internal::SvgBuilder;
 
 #include "glib/poppler-features.h"
-#include "Object.h"
+#include "@PREFIX@/libexec/poppler073/include/poppler/Object.h"
 
 class GooString;
 class XRef;
