--- a/configure	2019-09-10 12:18:06.513619282 +0200
+++ b/configure	2019-09-10 13:10:08.913619282 +0200
@@ -13077,7 +13077,7 @@
 int
 main ()
 {
- SSLeay_add_all_algorithms();
+ OpenSSL_add_all_algorithms();
   ;
   return 0;
 }
@@ -13101,7 +13101,7 @@
 int
 main ()
 {
- SSLeay_add_all_algorithms();
+ OpenSSL_add_all_algorithms();
   ;
   return 0;
 }
--- a/configure.ac	2018-10-17 02:01:20.000000000 +0200
+++ b/configure.ac	2019-09-10 13:10:20.643619282 +0200
@@ -2673,7 +2673,7 @@
 	AC_MSG_CHECKING([if programs using OpenSSL functions will link])
 	AC_LINK_IFELSE(
 		[AC_LANG_PROGRAM([[ #include <openssl/evp.h> ]],
-		[[ SSLeay_add_all_algorithms(); ]])],
+		[[ OpenSSL_add_all_algorithms(); ]])],
 		[
 			AC_MSG_RESULT([yes])
 		],
@@ -2684,7 +2684,7 @@
 			AC_MSG_CHECKING([if programs using OpenSSL need -ldl])
 			AC_LINK_IFELSE(
 				[AC_LANG_PROGRAM([[ #include <openssl/evp.h> ]],
-				[[ SSLeay_add_all_algorithms(); ]])],
+				[[ OpenSSL_add_all_algorithms(); ]])],
 				[
 					AC_MSG_RESULT([yes])
 				],
