--- orig.CMakeLists.txt	2016-04-30 17:37:11.000000000 +0200
+++ CMakeLists.txt	2016-04-30 17:37:24.000000000 +0200
@@ -18,6 +18,7 @@
     I18n
     Wallet
     WidgetsAddons
+    WindowSystem
 )
 
 find_package(KF5DocTools)
@@ -37,6 +38,7 @@
     KF5::I18n
     KF5::Wallet
     KF5::WidgetsAddons
+    KF5::WindowSystem
 )
 
 install(TARGETS ksshaskpass DESTINATION ${INSTALL_TARGETS_DEFAULT_ARGS})
--- src/orig.main.cpp	2016-04-30 17:35:47.000000000 +0200
+++ src/main.cpp	2016-04-30 17:36:06.000000000 +0200
@@ -24,6 +24,7 @@
 #include <KPasswordDialog>
 #include <KAboutData>
 #include <KLocalizedString>
+#include <KWindowSystem>
 
 #include <QApplication>
 #include <QCommandLineParser>
@@ -104,6 +105,7 @@
 
         kpd->setPrompt(dialog);
         kpd->setWindowTitle(i18n("Ksshaskpass"));
+        KWindowSystem::forceActiveWindow(kpd->winId(), 0);
         // We don't want to dump core when the password dialog is shown, because it could contain the entered password.
         // KPasswordDialog::disableCoreDumps() seems to be gone in KDE 4 -- do it manually
         struct rlimit rlim;
