diff --git a/./libclamav/c++/orig.detect.cpp b/./libclamav/c++/detect.cpp
index 4fb815c..95ed1cc 100644
--- a/./libclamav/c++/orig.detect.cpp
+++ b/./libclamav/c++/detect.cpp
@@ -21,6 +21,9 @@
  *  MA 02110-1301, USA.
  */
 
+#define __STDC_LIMIT_MACROS
+#define __STDC_CONSTANT_MACROS
+
 #include "llvm/ADT/Triple.h"
 #include "llvm/Config/llvm-config.h"
 #include "llvm/Support/raw_ostream.h"
diff --git a/./libclamav/c++/orig.bytecode2llvm.cpp b/./libclamav/c++/bytecode2llvm.cpp
index 7690029..bfc60ec 100644
--- a/./libclamav/c++/orig.bytecode2llvm.cpp
+++ b/./libclamav/c++/bytecode2llvm.cpp
@@ -21,6 +21,9 @@
  *  MA 02110-1301, USA.
  */
 
+#define __STDC_LIMIT_MACROS
+#define __STDC_CONSTANT_MACROS
+
 #define DEBUG_TYPE "clamavjit"
 #include <pthread.h>
 #ifndef _WIN32
diff --git a/./libclamav/c++/orig.PointerTracking.cpp b/./libclamav/c++/PointerTracking.cpp
index 8e9d96d..ed42f09 100644
--- a/./libclamav/c++/orig.PointerTracking.cpp
+++ b/./libclamav/c++/PointerTracking.cpp
@@ -15,6 +15,9 @@
  * */
 #ifndef _WIN32
 
+#define __STDC_LIMIT_MACROS
+#define __STDC_CONSTANT_MACROS
+
 #include "llvm/Analysis/ConstantFolding.h"
 #include "llvm/Analysis/LoopInfo.h"
 #include "llvm/Analysis/MemoryBuiltins.h"
diff --git a/./libclamav/c++/orig.ClamBCRTChecks.cpp b/./libclamav/c++/ClamBCRTChecks.cpp
index 6dc86fb..d62e8f3 100644
--- a/./libclamav/c++/orig.ClamBCRTChecks.cpp
+++ b/./libclamav/c++/ClamBCRTChecks.cpp
@@ -21,6 +21,9 @@
  *  MA 02110-1301, USA.
  */
 
+#define __STDC_LIMIT_MACROS
+#define __STDC_CONSTANT_MACROS
+
 #define DEBUG_TYPE "clambc-rtcheck"
 #include "ClamBCModule.h"
 #include "ClamBCDiagnostics.h"
