diff --git b/orig.CMakeLists.txt a/CMakeLists.txt
index 069bbf9..1afbb19 100644
--- b/orig.CMakeLists.txt
+++ a/CMakeLists.txt
@@ -93,7 +93,7 @@ set_package_properties(loginctl PROPERTIES
     )
 set(HAVE_LOGINCTL ${loginctl_FOUND})
 
-if ( NOT HAVE_LOGINCTL )
+#if ( NOT HAVE_LOGINCTL )
     find_package(ConsoleKit)
     set_package_properties(ConsoleKit PROPERTIES
         URL "https://github.com/ConsoleKit2/ConsoleKit2"
@@ -101,7 +101,7 @@ if ( NOT HAVE_LOGINCTL )
         TYPE RECOMMENDED
         )
     set(HAVE_CONSOLEKIT ${ConsoleKit_FOUND})
-endif ()
+#endif ()
 
 if (HAVE_LOGINCTL OR HAVE_CONSOLEKIT)
     set(HAVE_UNLOCK_CAPABILITY TRUE)
