diff --git b/orig.CMakeLists.txt a/CMakeLists.txt
index 069bbf9..c83a168 100644
--- b/orig.CMakeLists.txt
+++ a/CMakeLists.txt
@@ -113,6 +113,14 @@ add_feature_info("Unlock broken screenlocker"
 
 option(PAM_REQUIRED "Require building with PAM" ON)
 
+option(XSCREENSAVER_SHIM "Use XScreenSaver for the actual locking" OFF)
+add_feature_info("Rely on XScreenSaver for the actual screenlocking under X11"
+                 XSCREENSAVER_SHIM
+                 "the user will need to ensure s/he has xscreensaver running!")
+if (XSCREENSAVER_SHIM)
+	add_definitions(-DXSCREENSAVER_SHIM)
+endif()
+
 find_package(Seccomp)
 set_package_properties(Seccomp PROPERTIES
     TYPE
diff --git b/autotests/orig.keyboardgrabber.cpp a/autotests/keyboardgrabber.cpp
index 8b85517..950b330 100644
--- b/autotests/orig.keyboardgrabber.cpp
+++ a/autotests/keyboardgrabber.cpp
@@ -54,12 +54,16 @@ int main(int argc, char **argv)
     xcb_connection_t *c = xcb_connect(nullptr, &screen);
     Q_ASSERT(c);
 
+#ifndef XSCREENSAVER_SHIM
     xcb_grab_keyboard(c, 1, rootWindow(c, screen), XCB_CURRENT_TIME, XCB_GRAB_MODE_ASYNC, XCB_GRAB_MODE_ASYNC);
+#endif
     xcb_flush(c);
 
     const int exitCode = app.exec();
 
+#ifndef XSCREENSAVER_SHIM
     xcb_ungrab_keyboard(c, XCB_CURRENT_TIME);
+#endif
     xcb_flush(c);
     xcb_disconnect(c);
 
diff --git b/orig.ksldapp.cpp a/ksldapp.cpp
index 5fe28ea..ac6f626 100644
--- b/orig.ksldapp.cpp
+++ a/ksldapp.cpp
@@ -199,7 +199,7 @@ void KSldApp::initialize()
                 // there is at least one process blocking the auto lock of screen locker
                 return;
             }
-            if (m_lockGrace) {  // short-circuit if grace time is zero
+            if (m_lockGrace > 0) {  // short-circuit if grace time is zero
                 m_inGraceTime = true;
             } else if (m_lockGrace == -1) {
                 m_inGraceTime = true;  // if no timeout configured, grace time lasts forever
@@ -421,6 +421,7 @@ bool KSldApp::establishGrab()
     }
     XSync(QX11Info::display(), False);
     XServerGrabber serverGrabber;
+#ifndef XSCREENSAVER_SHIM
     if (!grabKeyboard()) {
         return false;
     }
@@ -430,8 +431,9 @@ bool KSldApp::establishGrab()
         XFlush(QX11Info::display());
         return false;
     }
+#endif
 
-#ifdef X11_Xinput_FOUND
+#if defined(X11_Xinput_FOUND) && !defined(XSCREENSAVER_SHIM)
     if (m_hasXInput2) {
         // get all devices
         Display *dpy = QX11Info::display();
@@ -484,14 +486,19 @@ bool KSldApp::establishGrab()
 
 static bool grabKeyboard()
 {
+#ifndef XSCREENSAVER_SHIM
     int rv = XGrabKeyboard( QX11Info::display(), QX11Info::appRootWindow(),
         True, GrabModeAsync, GrabModeAsync, CurrentTime );
 
     return (rv == GrabSuccess);
+#else
+    return true;
+#endif
 }
 
 static bool grabMouse()
 {
+#ifndef XSCREENSAVER_SHIM
 #define GRABEVENTS ButtonPressMask | ButtonReleaseMask | PointerMotionMask | \
                    EnterWindowMask | LeaveWindowMask
     int rv = XGrabPointer( QX11Info::display(), QX11Info::appRootWindow(),
@@ -500,6 +507,9 @@ static bool grabMouse()
 #undef GRABEVENTS
 
     return (rv == GrabSuccess);
+#else
+    return true;
+#endif
 }
 
 void KSldApp::doUnlock()
@@ -507,10 +517,12 @@ void KSldApp::doUnlock()
     qDebug() << "Grab Released";
     if (m_isX11) {
         xcb_connection_t *c = QX11Info::connection();
+#ifndef XSCREENSAVER_SHIM
         xcb_ungrab_keyboard(c, XCB_CURRENT_TIME);
         xcb_ungrab_pointer(c, XCB_CURRENT_TIME);
+#endif
         xcb_flush(c);
-#ifdef X11_Xinput_FOUND
+#if defined(X11_Xinput_FOUND) && !defined(XSCREENSAVER_SHIM)
         if (m_hasXInput2) {
             // get all devices
             Display *dpy = QX11Info::display();
