diff --git app/main.cpp app/main.cpp
index e1f07f8462..168875f3dd 100644
--- app/main.cpp
+++ app/main.cpp
@@ -356,6 +453,11 @@ int main( int argc, char *argv[] )
 
     KDevelopApplication app(argc, argv);
     KLocalizedString::setApplicationDomain("kdevelop");
+#ifdef Q_OS_MACOS
+    if (!QGuiApplication::platformName().contains(QLatin1String("cocoa"))) {
+        QCoreApplication::setAttribute(Qt::AA_DontUseNativeMenuBar);
+    }
+#endif
 
     static const char description[] = I18N_NOOP( "The KDevelop Integrated Development Environment" );
     KAboutData aboutData( QStringLiteral("kdevelop"), i18n( "KDevelop" ), QByteArray(VERSION), i18n(description), KAboutLicense::GPL,
