diff --git kdevplatform/shell/core.cpp kdevplatform/shell/core.cpp
index c1fd662f89cf826baea3247bf90c5348aead7c00..39879331859d33ab5a13aa20f2a5209774143770 100644
--- kdevplatform/shell/core.cpp
+++ kdevplatform/shell/core.cpp
@@ -349,9 +414,25 @@ Core *KDevelop::Core::self()
     return m_self;
 }
 
+#include <sys/resource.h>
+#ifdef Q_OS_LINUX
+#include <linux/fs.h>
+#define OPEN_MAX INR_OPEN_MAX
+#elif defined(Q_OS_MACOS)
+#include <sys/syslimits.h>
+#endif
+
 Core::Core(QObject *parent)
     : ICore(parent)
 {
+#ifdef OPEN_MAX
+    // set the maximum number of files
+    struct rlimit rlim;
+    getrlimit(RLIMIT_NOFILE, &rlim);
+    rlim.rlim_cur = qMin(rlim_t(OPEN_MAX), rlim.rlim_max);
+    setrlimit(RLIMIT_NOFILE, &rlim);
+#endif
+
     d = new CorePrivate(this);
 
     connect(QCoreApplication::instance(), &QCoreApplication::aboutToQuit, this, &Core::shutdown);
