diff --git kdevplatform/documentation/documentationview.cpp kdevplatform/documentation/documentationview.cpp
index 18e58c34be8a11c901f00976c0616d256bbfb226..fb54a345fe54dc901573ac276accf8af21e68932 100644
--- kdevplatform/documentation/documentationview.cpp
+++ kdevplatform/documentation/documentationview.cpp
@@ -28,6 +28,7 @@
 #include <QAbstractItemView>
 #include <QLineEdit>
 #include <QShortcut>
+#include <QEvent>
 
 #include <KLocalizedString>
 
@@ -39,6 +40,8 @@
 #include "documentationfindwidget.h"
 #include "debug.h"
 
+#include <sublime/idealdockwidget.h>
+
 using namespace KDevelop;
 
 DocumentationView::DocumentationView(QWidget* parent, ProvidersModel* model)
@@ -70,8 +73,30 @@ DocumentationView::DocumentationView(QWidget* parent, ProvidersModel* model)
     setFocusProxy(mIdentifiers);
 
     QMetaObject::invokeMethod(this, "initialize", Qt::QueuedConnection);
+
+    floatStandaloneWindows();
+}
+
+bool DocumentationView::event(QEvent* e)
+{
+    if (e->type() == QEvent::ParentChange) {
+        // we'll have to make the new IdealDockWidget parent
+        // behave the way we'd like it to behave.
+        floatStandaloneWindows();
+    }
+    return QWidget::event(e);
 }
 
+void DocumentationView::floatStandaloneWindows()
+{
+    Sublime::IdealDockWidget* dockWidget = dynamic_cast<Sublime::IdealDockWidget*>(parent());
+    if (dockWidget) {
+        dockWidget->setFloating(false);
+        dockWidget->setFloatsAsStandalone(true);
+    }
+}
+
+
 QList<QAction*> DocumentationView::contextMenuActions() const
 {
     // TODO: also show providers
diff --git kdevplatform/documentation/documentationview.h kdevplatform/documentation/documentationview.h
index 8d69bb483cf5d01fb19aad196612523b1ecb3f30..608bdefa004f6cee04ac68b359140b8563203cc8 100644
--- kdevplatform/documentation/documentationview.h
+++ kdevplatform/documentation/documentationview.h
@@ -58,10 +58,14 @@ public Q_SLOTS:
     void changedProvider(int);
     void showHome();
 
+protected:
+    bool event(QEvent* e) override;
+
 private:
     void setupActions();
     void updateView();
     void returnPressed();
+    void floatStandaloneWindows();
 
     QAction* mForward;
     QAction* mBack;
