diff --git kdevplatform/shell/CMakeLists.txt kdevplatform/shell/CMakeLists.txt
index 5a9c8c3ebe87566e77871c5cd3be7b066be296c5..e3af7617a52f8e073619233f57982c3da78b1938 100644
--- kdevplatform/shell/CMakeLists.txt
+++ kdevplatform/shell/CMakeLists.txt
@@ -158,6 +158,10 @@ PRIVATE
 if(APPLE)
     target_link_libraries(KDevPlatformShell PRIVATE "-framework AppKit")
 endif()
+if(UNIX)
+    target_link_libraries(KDevPlatformShell PRIVATE "-pthreads")
+endif()
+
 
 install(FILES
     mainwindow.h
diff --git plugins/astyle/CMakeLists.txt plugins/astyle/CMakeLists.txt
index cc863bde2d99809a05f3cb837c2087ac985f2383..5b61a500fdfba343e29e8c80e38ff85968a9ba18 100644
--- plugins/astyle/CMakeLists.txt
+++ plugins/astyle/CMakeLists.txt
@@ -37,6 +37,12 @@ target_link_libraries(kdevastyle
     KDev::Interfaces
     KDev::Util)
 
+if(UNIX AND NOT APPLE)
+    if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+        target_link_libraries(kdevastyle "-pthread")
+    endif()
+endif()
+
 if(NOT LibAStyle_FOUND)
     add_subdirectory(3rdparty/libastyle)
 endif()
diff --git plugins/debuggercommon/CMakeLists.txt plugins/debuggercommon/CMakeLists.txt
index 6530ae786d45e1e641af1a3cdbc7a06347cb53d1..6ec148aad6389480214d16b1db2e6e7eb67c7296 100644
--- plugins/debuggercommon/CMakeLists.txt
+++ plugins/debuggercommon/CMakeLists.txt
@@ -86,6 +86,12 @@ if(KF5SysGuard_FOUND)
     )
 endif()
 
+if(UNIX AND NOT APPLE)
+    if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+        target_link_libraries(kdevdebuggercommon PRIVATE "-pthread")
+    endif()
+endif()
+
 kde_target_enable_exceptions(kdevdebuggercommon PUBLIC)
 
 if(BUILD_TESTING)
diff --git plugins/subversion/CMakeLists.txt plugins/subversion/CMakeLists.txt
index 2bbd51ae2d22d935c2625318af69b139fcec8e38..c5d3851e387615dabcc4d2533ba7a65c5af50d30 100644
--- plugins/subversion/CMakeLists.txt
+++ plugins/subversion/CMakeLists.txt
@@ -66,4 +66,10 @@ target_link_libraries(kdevsubversion
     kdevsvncpp
 )
 
+if(UNIX AND NOT APPLE)
+    if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+        target_link_libraries(kdevsubversion "-pthread")
+    endif()
+endif()
+
 install(PROGRAMS org.kde.kdevelop_svn.desktop DESTINATION ${KDE_INSTALL_APPDIR})
