diff --git kdevplatform/language/codecompletion/codecompletionmodel.cpp kdevplatform/language/codecompletion/codecompletionmodel.cpp
index ed2d54ae45718522c32c5cc4c756b6da681cdd5d..cdae1e235514da9d1e07863d2cba60ca6bf78586 100644
--- kdevplatform/language/codecompletion/codecompletionmodel.cpp
+++ kdevplatform/language/codecompletion/codecompletionmodel.cpp
@@ -1,3 +1,4 @@
+
 /*
  * KDevelop Generic Code Completion Support
  *
@@ -104,7 +105,7 @@ void CodeCompletionModel::setForceWaitForModel(bool wait)
 
 CodeCompletionModel::CodeCompletionModel(QObject* parent)
     : KTextEditor::CodeCompletionModel(parent)
-    , m_forceWaitForModel(false)
+    , m_forceWaitForModel(true)
     , m_fullCompletion(true)
     , m_mutex(new QMutex)
     , m_thread(nullptr)
