diff --git plugins/welcomepage/qml/Develop.qml plugins/welcomepage/qml/Develop.qml
index 75fb438331..3369ed5260 100644
--- plugins/welcomepage/qml/Develop.qml
+++ plugins/welcomepage/qml/Develop.qml
@@ -20,7 +20,7 @@
  */
 
 import QtQuick 2.7
-import QtQuick.Layouts 1.6
+import QtQuick.Layouts 1.2
 import QtQuick.Controls 2.0
 import QtQuick.Controls 1.4 as QQC1
 
diff --git plugins/welcomepage/qml/Link.qml plugins/welcomepage/qml/Link.qml
index 37b67b29cf..4eecf26385 100644
--- plugins/welcomepage/qml/Link.qml
+++ plugins/welcomepage/qml/Link.qml
@@ -19,8 +19,8 @@
  */
 
 import QtQuick 2.7
-import QtQuick.Layouts 1.6
-import QtQuick.Controls 2.0
+import QtQuick.Layouts 1.3
+import QtQuick.Controls 1.0
 
 Label {
     id: root
diff --git plugins/welcomepage/qml/NewsFeed.qml plugins/welcomepage/qml/NewsFeed.qml
index 33f867dc81..d870dd9426 100644
--- plugins/welcomepage/qml/NewsFeed.qml
+++ plugins/welcomepage/qml/NewsFeed.qml
@@ -20,8 +20,8 @@
 
 import QtQuick 2.7
 import QtQuick.Controls 2.0
-import QtQuick.Layouts 1.6
-import QtQuick.XmlListModel 2.7
+import QtQuick.Layouts 1.3
+import QtQuick.XmlListModel 2.0
 
 import org.kde.kdevplatform 1.0
 
diff --git plugins/welcomepage/qml/StandardBackground.qml plugins/welcomepage/qml/StandardBackground.qml
index 30eebca17c..a54771b950 100644
--- plugins/welcomepage/qml/StandardBackground.qml
+++ plugins/welcomepage/qml/StandardBackground.qml
@@ -20,7 +20,7 @@
 
 import QtQuick 2.7
 import QtQuick.Controls 2.0
-import QtQuick.Layouts 1.6
+import QtQuick.Layouts 1.2
 
 GroupBox
 {
diff --git plugins/welcomepage/qml/StandardPage.qml plugins/welcomepage/qml/StandardPage.qml
index d13460a102..6ee1afdf62 100644
--- plugins/welcomepage/qml/StandardPage.qml
+++ plugins/welcomepage/qml/StandardPage.qml
@@ -19,7 +19,7 @@
  */
 
 import QtQuick 2.7
-import QtQuick.Layouts 1.6
+import QtQuick.Layouts 1.3
 
 Rectangle {
     id: root
diff --git plugins/welcomepage/qml/area_code.qml plugins/welcomepage/qml/area_code.qml
index 81a56a8c76..02c4011977 100644
--- plugins/welcomepage/qml/area_code.qml
+++ plugins/welcomepage/qml/area_code.qml
@@ -20,7 +20,7 @@
 
 import QtQuick 2.7
 import QtQuick.Controls 2.0
-import QtQuick.Layouts 1.6
+import QtQuick.Layouts 1.3
 
 StandardBackground {
     id: root
