diff --git app/plasma/CMakeLists.txt app/plasma/CMakeLists.txt
index 45593eae23289d03dee04737802a89ca90f53237..2c18d854fb827eea1f0bd637c38cfb61fc868688 100644
--- app/plasma/CMakeLists.txt
+++ app/plasma/CMakeLists.txt
@@ -1,11 +1,11 @@
 add_subdirectory(dataengine)
 add_subdirectory(applet)
 
-find_package(KF5Runner CONFIG)
-set_package_properties(KF5Runner PROPERTIES
-    PURPOSE "Framework for Plasma runners. For enabling the KDevelop runner"
-    URL "https://www.kde.org"
-    TYPE OPTIONAL)
-if(KF5Runner_FOUND)
-    add_subdirectory(runner)
-endif()
+# find_package(KF5Runner CONFIG)
+# set_package_properties(KF5Runner PROPERTIES
+#     PURPOSE "Framework for Plasma runners. For enabling the KDevelop runner"
+#     URL "https://www.kde.org"
+#     TYPE OPTIONAL)
+# if(KF5Runner_FOUND)
+#     add_subdirectory(runner)
+# endif()
