--- src/util/urlhandler.cpp.orig	2019-07-15 17:02:19.000000000 +0200
+++ src/util/urlhandler.cpp	2019-07-15 17:05:21.000000000 +0200
@@ -42,7 +42,10 @@
             u.setPath(QCoreApplication::applicationName());
         }
 
-        const QString helpcenter = QStandardPaths::findExecutable(QStringLiteral("khelpcenter"));
+        QString helpcenter = QStandardPaths::findExecutable(QStringLiteral("khelpcentre5"));
+        if (helpcenter.isEmpty()) {
+            helpcenter = QStandardPaths::findExecutable(QStringLiteral("khelpcenter"));
+        }
         if (helpcenter.isEmpty()) {
             if (QCoreApplication::organizationDomain() == QLatin1String("kde.org")) {
                 //if khelpcenter is not installed and it's a KDE application, use docs.kde.org
