diff --git CMakeLists.txt CMakeLists.txt
index 2e67faa..1fc33b9 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -17,7 +17,7 @@ include(KDECMakeSettings)
 set(REQUIRED_QT_VERSION 5.11.0)
 find_package(Qt5Gui ${REQUIRED_QT_VERSION} REQUIRED NO_MODULE)
 
-if (NOT APPLE AND NOT WIN32)
+if ((NOT APPLE OR APPLE_FORCE_X11) AND NOT WIN32)
     find_package(X11 MODULE)
     find_package(XCB MODULE COMPONENTS XCB)
 endif()
diff --git src/CMakeLists.txt src/CMakeLists.txt
index 811a5fb..4d9c7e1 100644
--- src/CMakeLists.txt
+++ src/CMakeLists.txt
@@ -33,7 +33,7 @@ target_compile_definitions(KF5GuiAddons INTERFACE "$<INSTALL_INTERFACE:KGUIADDON
 target_link_libraries(KF5GuiAddons PUBLIC Qt5::Gui)
 
 set(WITH_XCB)
-if (NOT APPLE AND X11_FOUND AND X11_Xkb_FOUND AND XCB_XCB_FOUND)
+if ((NOT APPLE OR APPLE_FORCE_X11) AND X11_FOUND AND X11_Xkb_FOUND AND XCB_XCB_FOUND)
     find_package(Qt5X11Extras ${REQUIRED_QT_VERSION} NO_MODULE)
     if (Qt5X11Extras_FOUND)
         add_library(kmodifierkey_xcb MODULE util/kmodifierkeyinfoprovider_xcb.cpp)
