diff --git a/CMakeLists.txt b/CMakeLists.txt
index 18ecfc0..9a72b0f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -16,6 +16,7 @@ find_package(KDevPlatform ${KDEVPLATFORM_VERSION} REQUIRED)
 
 
 # find the system python 3 interpreter, only used for determining search paths.
+set(PYTHON_EXECUTABLE "/opt/local/bin/python3.4")
 find_package(PythonInterp 3.0 REQUIRED)
 configure_file( "${kdevpython_SOURCE_DIR}/kdevpythonversion.h.cmake" "${kdevpython_BINARY_DIR}/kdevpythonversion.h" @ONLY )
 
