diff --git a/CMakeLists.txt b/CMakeLists.txt
index f7a1739..582411a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -17,6 +17,7 @@ find_package(KDevPlatform ${KDEVPLATFORM_VERSION} REQUIRED)
 
 
 # find the system python 3.4 interpreter, only used for determining search paths.
+set(PYTHON_EXECUTABLE "@PREFIX@/bin/python3.4")
 find_package(Python 3.4)
 configure_file( "${kdevpython_SOURCE_DIR}/kdevpythonversion.h.cmake" "${kdevpython_BINARY_DIR}/kdevpythonversion.h" @ONLY )
 
