diff --git CMakeLists.txt CMakeLists.txt
index dfaac55..1e87291 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -18,6 +18,7 @@ include(KDECMakeSettings)
 include(KDEFrameworkCompilerSettings)
 include(ECMInstallIcons)
 include(ECMAddTests)
+include(ECMMarkNonGuiExecutable)
 
 set(KF5_VERSION "5.16.0") # handled by release scripts
 set(KF5_DEP_VERSION "5.16.0") # handled by release scripts
diff --git src/file/CMakeLists.txt src/file/CMakeLists.txt
index f15a263..96635d4 100644
--- src/file/CMakeLists.txt
+++ src/file/CMakeLists.txt
@@ -59,6 +59,7 @@ set(file_SRCS
 )
 
 add_executable(baloo_file ${file_SRCS})
+ecm_mark_nongui_executable(baloo_file)
 target_compile_definitions(baloo_file PRIVATE -DPROJECT_VERSION="${PROJECT_VERSION}")
 
 target_link_libraries(baloo_file
diff --git src/file/extractor/CMakeLists.txt src/file/extractor/CMakeLists.txt
index aaf487b..0dc230d 100644
--- src/file/extractor/CMakeLists.txt
+++ src/file/extractor/CMakeLists.txt
@@ -17,6 +17,7 @@ set(EXTRACTOR_SRCS
 )
 
 add_executable(baloo_file_extractor ${EXTRACTOR_SRCS})
+ecm_mark_nongui_executable(baloo_file_extractor)
 target_compile_definitions(baloo_file_extractor PRIVATE -DPROJECT_VERSION="${PROJECT_VERSION}")
 
 target_link_libraries(baloo_file_extractor
diff --git src/tools/balooctl/CMakeLists.txt src/tools/balooctl/CMakeLists.txt
index 091d1f2..ef0eb45 100644
--- src/tools/balooctl/CMakeLists.txt
+++ src/tools/balooctl/CMakeLists.txt
@@ -19,6 +19,7 @@ set(DBUS_INTERFACES
 set_source_files_properties(${DBUS_INTERFACES} PROPERTIES GENERATED 1)
 
 add_executable(balooctl ${SRCS} ${DBUS_INTERFACES})
+ecm_mark_nongui_executable(balooctl)
 add_dependencies(balooctl BalooDBusInterfaces)
 target_compile_definitions(balooctl PRIVATE -DPROJECT_VERSION="${PROJECT_VERSION}")
 
diff --git src/tools/baloosearch/CMakeLists.txt src/tools/baloosearch/CMakeLists.txt
index ba2c880..9b241c2 100644
--- src/tools/baloosearch/CMakeLists.txt
+++ src/tools/baloosearch/CMakeLists.txt
@@ -3,6 +3,7 @@ add_definitions(-DTRANSLATION_DOMAIN=\"baloosearch\")
 set(BALOO_SEARCH_SRCS main.cpp)
 
 add_executable(baloosearch ${BALOO_SEARCH_SRCS})
+ecm_mark_nongui_executable(baloosearch)
 target_compile_definitions(baloosearch PRIVATE -DPROJECT_VERSION="${PROJECT_VERSION}")
 
 target_link_libraries(baloosearch
diff --git src/tools/balooshow/CMakeLists.txt src/tools/balooshow/CMakeLists.txt
index 61d7421..73b44b8 100644
--- src/tools/balooshow/CMakeLists.txt
+++ src/tools/balooshow/CMakeLists.txt
@@ -3,6 +3,7 @@ add_definitions(-DTRANSLATION_DOMAIN=\"balooshow\")
 set(SRCS main.cpp)
 
 add_executable(balooshow ${SRCS})
+ecm_mark_nongui_executable(balooshow)
 target_compile_definitions(balooshow PRIVATE -DPROJECT_VERSION="${PROJECT_VERSION}")
 
 target_link_libraries(balooshow
