diff --git a/projects/libcxx/include/orig.__config b/projects/libcxx/include/__config
index f15d2d0..36ae671 100644
--- a/projects/libcxx/include/orig.__config
+++ b/projects/libcxx/include/__config
@@ -21,6 +21,9 @@
 #pragma GCC system_header
 #endif
 
+// we are not going to use Apple Availability testing in these headers
+#define _LIBCPP_DISABLE_AVAILABILITY
+
 #ifdef __cplusplus
 
 #ifdef __GNUC__
