diff --git b/projects/libcxx/include/orig.__config a/projects/libcxx/include/__config
index 1ecced9..280a117 100644
--- b/projects/libcxx/include/orig.__config
+++ a/projects/libcxx/include/__config
@@ -20,6 +20,12 @@
 #pragma GCC system_header
 #endif
 
+// we are not going to use Apple Availability testing in these headers
+#define _LIBCPP_DISABLE_AVAILABILITY
+// make per TU ABI insulation enabled by default; it can be overridden by
+// -D_LIBCPP_HIDE_FROM_ABI_PER_TU=0
+#define _LIBCPP_HIDE_FROM_ABI_PER_TU_BY_DEFAULT
+
 #ifdef __cplusplus
 
 #ifdef __GNUC__
