diff --git a/blitz/convolve.cpp b/blitz/convolve.cpp
index cd3d19e09a204e91c38babf23e5a22092e9a975e..ec1415da8128e26b05e97a51b8701656133821c8 100644
--- a/blitz/convolve.cpp
+++ b/blitz/convolve.cpp
@@ -940,7 +940,7 @@ float* BlitzPrivate::getBlurKernel(int &kernel_width, float sigma)
     long i;
 
     if(sigma == 0.0)
-        return(false);
+        return(NULL);
     if(kernel_width == 0)
         kernel_width = 3;
 
