diff --git a/blitz/CMakeLists.txt b/blitz/CMakeLists.txt
index 985a157156fccd53de011ad1c05a653c1ff13125..d01b883db8f5cbd06c4c448f3080609bbabeabee 100644
--- a/blitz/CMakeLists.txt
+++ b/blitz/CMakeLists.txt
@@ -30,7 +30,7 @@ set(blitz_LIB_SRCS blitz.cpp colors.cpp convolve.cpp histogram.cpp
                    blitzcpu.cpp scale.cpp scalefilter.cpp gradient.cpp)
 
 if(HAVE_MMX)
-    if(NOT APPLE AND NOT WIN32)
+    if(NOT WIN32)
         set( blitz_LIB_SRCS ${blitz_LIB_SRCS} asm_scale.S )
         add_custom_command(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/asm_scale.o
             COMMAND gcc -c ${CMAKE_CURRENT_SOURCE_DIR}/asm_scale.S
@@ -47,31 +47,31 @@ if(COMMAND cmake_policy)
       cmake_policy(SET CMP0028 NEW)
 endif(COMMAND cmake_policy) 
 
-add_library(qimageblitz SHARED ${blitz_LIB_SRCS} ${blitz_LIB_EXTRA_SRCS})
+add_library(q5imageblitz SHARED ${blitz_LIB_SRCS} ${blitz_LIB_EXTRA_SRCS})
 
-target_link_libraries(qimageblitz ${Qt5Gui_LIBRARIES}) 
+target_link_libraries(q5imageblitz ${Qt5Gui_LIBRARIES}) 
 if (NOT CMAKE_COMPILER_IS_GNUCXX AND CMAKE_SYSTEM_NAME MATCHES SunOS)
-    target_link_libraries(qimageblitz -lm -lsunmath)
+    target_link_libraries(q5imageblitz -lm -lsunmath)
 endif (NOT CMAKE_COMPILER_IS_GNUCXX AND CMAKE_SYSTEM_NAME MATCHES SunOS)
 
-set_target_properties(qimageblitz PROPERTIES
+set_target_properties(q5imageblitz PROPERTIES
                             VERSION ${BLITZ_LIB_MAJOR_VERSION}.${BLITZ_LIB_MINOR_VERSION}.${BLITZ_LIB_PATCH_VERSION}
                             SOVERSION ${BLITZ_LIB_MAJOR_VERSION}
                             DEFINE_SYMBOL MAKE_QIMAGEBLITZ_LIB
 )
-install( TARGETS qimageblitz  
+install( TARGETS q5imageblitz  
     RUNTIME DESTINATION ${BIN_INSTALL_DIR} 
     LIBRARY DESTINATION ${LIB_INSTALL_DIR} 
     ARCHIVE DESTINATION ${LIB_INSTALL_DIR} 
 )
 
-install( FILES qimageblitz.h qimageblitz_export.h blitzcpu.h DESTINATION include/qimageblitz COMPONENT Devel )
+install( FILES qimageblitz.h qimageblitz_export.h blitzcpu.h DESTINATION include/q5imageblitz COMPONENT Devel )
 
 if(NOT WIN32)
-  configure_file("qimageblitz.pc.cmake" "${CMAKE_CURRENT_BINARY_DIR}/qimageblitz.pc"
+  configure_file("qimageblitz.pc.cmake" "${CMAKE_CURRENT_BINARY_DIR}/q5imageblitz.pc"
   @ONLY) 
 
-  install(FILES ${CMAKE_CURRENT_BINARY_DIR}/qimageblitz.pc DESTINATION 
+  install(FILES ${CMAKE_CURRENT_BINARY_DIR}/q5imageblitz.pc DESTINATION 
   ${LIB_INSTALL_DIR}/pkgconfig)  
 endif(NOT WIN32)
 
diff --git a/blitz/qimageblitz.pc.cmake b/blitz/qimageblitz.pc.cmake
index 96162ddcb3f686f46c7ea1bbd18fea23a2f8ad11..986b93aa936a25f20ba3d8e6f43ed58fff23cac4 100644
--- a/blitz/qimageblitz.pc.cmake
+++ b/blitz/qimageblitz.pc.cmake
@@ -1,11 +1,11 @@
 prefix=@CMAKE_INSTALL_PREFIX@
 exec_prefix=@CMAKE_INSTALL_PREFIX@
 libdir=@LIB_INSTALL_DIR@
-includedir=@INCLUDE_INSTALL_DIR@
+includedir=@CMAKE_INSTALL_PREFIX@/include/q5imageblitz
 
 Name: Blitz
 Description: Blitz image effect and filter library
 Version: @BLITZ_LIB_VERSION_STRING@
 Requires: QtGui
-Libs: -L${libdir} -lqimageblitz
+Libs: -L${libdir} -lq5imageblitz
 Cflags: -I${includedir}
diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index 005dbf83efe369cb7c272002f6a14b12c93bb8d2..6fe70d83a71b393e9c534af99107e4e3e4c5b83c 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -1,4 +1,4 @@
-project(blitztest)
+project(blitztest_qt5)
 
 if (WIN32)
  set(CMAKE_DEBUG_POSTFIX "d")
@@ -28,9 +28,9 @@ if(COMMAND cmake_policy)
   cmake_policy(SET CMP0028 NEW)
 endif(COMMAND cmake_policy) 
 
-add_executable(blitztest ${GUI_SOURCE_FILES} ${GUI_HEADER_FILES_HPP})
+add_executable(blitztest_qt5 ${GUI_SOURCE_FILES} ${GUI_HEADER_FILES_HPP})
 
-target_link_libraries(blitztest ${Qt5Widgets_LIBRARIES} qimageblitz)
+target_link_libraries(blitztest_qt5 ${Qt5Widgets_LIBRARIES} q5imageblitz)
 
-install(TARGETS blitztest DESTINATION ${BIN_INSTALL_DIR})
+install(TARGETS blitztest_qt5 DESTINATION ${BIN_INSTALL_DIR})
 
