diff --git kdevplatform/shell/core_p.h kdevplatform/shell/core_p.h
index d5ed3525cc9114642a2e18301bb98f60c0e0e98f..cd963d809ec0e7b2e7d71c392908a905d17f1dfe 100644
--- kdevplatform/shell/core_p.h
+++ kdevplatform/shell/core_p.h
@@ -26,6 +26,7 @@
 #include <KAboutData>
 
 #include <QPointer>
+class QDir;
 
 namespace KDevelop
 {
@@ -74,6 +79,7 @@ public:
     bool m_cleanedUp;
     bool m_shuttingDown;
     Core::Setup m_mode;
+    QDir* m_tmpDir = nullptr;
 };
 
 }
diff --git kdevplatform/shell/core.cpp kdevplatform/shell/core.cpp
index c1fd662f89cf826baea3247bf90c5348aead7c00..ede4f7f3ac5f18640a8d2e701bea16dd889a68e3 100644
--- kdevplatform/shell/core.cpp
+++ kdevplatform/shell/core.cpp
@@ -130,6 +171,26 @@ bool CorePrivate::initialize(Core::Setup mode, const QString& session )
     m_mode=mode;
 
     qCDebug(SHELL) << "Creating controllers";
+#if defined(Q_OS_UNIX)
+    auto tmpLocation = QStandardPaths::writableLocation(QStandardPaths::TempLocation);
+    const QString tmpName(QStringLiteral("/kdevelop-tmp-%1-").arg(getuid()));
+    const auto pos = tmpLocation.lastIndexOf(tmpName);
+    if (pos >= 0) {
+        // we must have been started by another KDevelop session,
+        // restore the default TempLocation
+        tmpLocation = tmpLocation.left(pos);
+    }
+    tmpLocation += tmpName + session;
+    m_tmpDir = new QDir(tmpLocation);
+    if (m_tmpDir->exists()) {
+        qCDebug(SHELL) << "Removing existing session temp dir" << tmpLocation;
+        m_tmpDir->removeRecursively();
+    }
+    m_tmpDir->mkpath(tmpLocation);
+    if (m_tmpDir->exists()) {
+        qputenv("TMPDIR", tmpLocation.toLatin1());
+    }
+#endif
 
     if( !sessionController )
     {
@@ -313,6 +396,11 @@ CorePrivate::~CorePrivate()
     delete testController.data();
     delete runtimeController.data();
 
+    if (m_tmpDir->exists()) {
+        m_tmpDir->removeRecursively();
+        delete m_tmpDir;
+    }
+
     selectionController.clear();
     projectController.clear();
     languageController.clear();
