diff --git plugins/clang/duchain/clanghelpers.cpp plugins/clang/duchain/clanghelpers.cpp
index 77fde1548c00ced469098695d2111e3d3c2e7947..ee314980ac6b12a6734f9dbce4c7e060aa95cedf 100644
--- plugins/clang/duchain/clanghelpers.cpp
+++ plugins/clang/duchain/clanghelpers.cpp
@@ -386,6 +386,20 @@ QString ClangHelpers::clangBuiltinIncludePath()
         if (QFileInfo(dir).isDir()) {
             return dir;
         }
+#elif defined(Q_OS_UNIX)
+        // a clang minor version upgrade since we were last built can
+        // cause problems if the "clang/$fullversion/include" path component
+        // changed. Try to generate the correct builtin_dir for the current
+        // major.minor.patchlevel version: drop the last 2 components then
+        // add them with the updated full version string.
+        auto hardcodedDir = QDir(QStringLiteral(KDEV_CLANG_BUILTIN_DIR));
+        hardcodedDir.cd(QStringLiteral("../.."));
+        dir = QDir::cleanPath(QStringLiteral("%1/%2/include")
+            .arg(hardcodedDir.path(), clangVersion()));
+        if (QFileInfo(dir).isDir()) {
+            clangDebug() << "Using builtin dir:" << dir;
+            return dir;
+        }
 #endif
 
         clangDebug() << "Using builtin dir:" << KDEV_CLANG_BUILTIN_DIR;
