--- orig.CMakeLists.txt	2018-01-02 21:49:46.000000000 +0100
+++ CMakeLists.txt	2018-01-02 21:52:06.000000000 +0100
@@ -219,7 +219,4 @@
 
 install(FILES org.kde.kdevelop.appdata.xml DESTINATION ${KDE_INSTALL_METAINFODIR})
 
-# Make it possible to use the po files fetched by the fetch-translations step
-ki18n_install(po)
-
 feature_summary(WHAT ALL INCLUDE_QUIET_PACKAGES FATAL_ON_MISSING_REQUIRED_PACKAGES)
