diff --git kdevplatform/shell/core_p.h kdevplatform/shell/core_p.h
index d5ed3525cc9114642a2e18301bb98f60c0e0e98f..cd963d809ec0e7b2e7d71c392908a905d17f1dfe 100644
--- kdevplatform/shell/core_p.h
+++ kdevplatform/shell/core_p.h
@@ -74,6 +78,7 @@ public:
     bool m_cleanedUp;
     bool m_shuttingDown;
     Core::Setup m_mode;
+    QString m_tmpLocation;
 };
 
 }
diff --git kdevplatform/shell/core.cpp kdevplatform/shell/core.cpp
index c1fd662f89cf826baea3247bf90c5348aead7c00..ede4f7f3ac5f18640a8d2e701bea16dd889a68e3 100644
--- kdevplatform/shell/core.cpp
+++ kdevplatform/shell/core.cpp
@@ -130,6 +171,19 @@ bool CorePrivate::initialize(Core::Setup mode, const QString& session )
     m_mode=mode;
 
     qCDebug(SHELL) << "Creating controllers";
+#if defined(Q_OS_UNIX)
+    m_tmpLocation = QStandardPaths::writableLocation(QStandardPaths::TempLocation);
+    if (!m_tmpLocation.isEmpty()) {
+        m_tmpLocation += QStringLiteral("/kdevelop-tmp-session-") + session;
+        QDir tmpDir(m_tmpLocation);
+        if (tmpDir.exists()) {
+            qCWarning(SHELL) << "Removing existing session temp dir" << m_tmpLocation;
+            tmpDir.removeRecursively();
+        }
+        tmpDir.mkpath(m_tmpLocation);
+        qputenv("TMPDIR", m_tmpLocation.toLatin1());
+    }
+#endif
 
     if( !sessionController )
     {
@@ -313,6 +387,13 @@ CorePrivate::~CorePrivate()
     delete testController.data();
     delete runtimeController.data();
 
+    if (!m_tmpLocation.isEmpty()) {
+        QDir tmpDir(m_tmpLocation);
+        if (tmpDir.exists()) {
+            tmpDir.removeRecursively();
+        }
+    }
+
     selectionController.clear();
     projectController.clear();
     languageController.clear();
