diff --git kdevplatform/serialization/itemrepositoryregistry.cpp kdevplatform/serialization/itemrepositoryregistry.cpp
index 5b8aade63a960af24bee8191df99fa4fa27bf471..a92ff4fec4c58bad620cd4710b28ad5af3b6f58e 100644
--- kdevplatform/serialization/itemrepositoryregistry.cpp
+++ kdevplatform/serialization/itemrepositoryregistry.cpp
@@ -27,6 +27,7 @@
 #include <KLocalizedString>
 
 #include <util/shellutils.h>
+#include <interfaces/icore.h>
 
 #include "abstractitemrepository.h"
 #include "debug.h"
@@ -203,9 +204,13 @@ void ItemRepositoryRegistry::registerRepository(AbstractItemRepository* reposito
   d->m_repositories.insert(repository, manager);
   if(!d->m_path.isEmpty()) {
     if(!repository->open(d->m_path)) {
-      d->deleteDataDirectory(d->m_path);
-      qCritical() << "failed to open a repository";
-      abort();
+      qCritical() << "failed to open repository" << repository->repositoryName() << "at" << d->m_path;
+      if (!ICore::self()->shuttingDown()) {
+        d->deleteDataDirectory(d->m_path, true);
+//         abort();
+        // let's just try again and see what happens...
+        repository->open(d->m_path);
+      }
     }
   }
 }
@@ -287,7 +292,7 @@ bool ItemRepositoryRegistryPrivate::open(const QString& path)
   foreach(AbstractItemRepository* repository, m_repositories.keys()) {
     if(!repository->open(path)) {
       deleteDataDirectory(path);
-      qCritical() << "failed to open a repository";
+      qCritical() << "failed to open repository" << repository->repositoryName() << "at" << path;
       abort();
     }
   }
