diff --git kdevplatform/language/codecompletion/codecompletionmodel.cpp kdevplatform/language/codecompletion/codecompletionmodel.cpp
index 2e3520699fd792969e60a320ed6f0c13e0323234..19a63faf8653f15105e2c1e989a6cd4d1f1eceda 100644
--- kdevplatform/language/codecompletion/codecompletionmodel.cpp
+++ kdevplatform/language/codecompletion/codecompletionmodel.cpp
@@ -96,7 +96,7 @@ void CodeCompletionModel::setForceWaitForModel(bool wait)
 
 CodeCompletionModel::CodeCompletionModel( QObject * parent )
   : KTextEditor::CodeCompletionModel(parent)
-  , m_forceWaitForModel(false)
+  , m_forceWaitForModel(true)
   , m_fullCompletion(true)
   , m_mutex(new QMutex)
   , m_thread(nullptr)
