commit efb70121bf0e0d0b20a63a9e8cac139a312e749a
Author: Friedrich W. H. Kossebau <kossebau@kde.org>
Date:   Wed Jul 10 23:46:33 2019 +0200

    shell: overwrite katectagsplugin to be disabled by default

diff --git kdevplatform/shell/plugincontroller.cpp kdevplatform/shell/plugincontroller.cpp
index 05b85280a0..58f523cc90 100644
--- kdevplatform/shell/plugincontroller.cpp
+++ kdevplatform/shell/plugincontroller.cpp
@@ -317,13 +317,6 @@ PluginController::PluginController(Core *core)
     d->plugins.reserve(d->plugins.size() + ktePlugins.size());
     for (const auto& info : ktePlugins) {
         auto data = info.rawData();
-        // temporary workaround for Kate's ctags plugin being enabled by default
-        // see https://mail.kde.org/pipermail/kwrite-devel/2019-July/004821.html
-        if (info.pluginId() == QLatin1String("katectagsplugin")) {
-            auto kpluginData = data[KEY_KPlugin()].toObject();
-            kpluginData[KEY_EnabledByDefault()] = false;
-            data[KEY_KPlugin()] = kpluginData;
-        }
         // add some KDevelop specific JSON data
         data[KEY_Category()] = KEY_Global();
         data[KEY_Mode()] = KEY_Gui();
