diff --git plugins/astyle/CMakeLists.txt plugins/astyle/CMakeLists.txt
index 8018af4fb9c463054867dc821738b9d3be1248d9..5d0eaccbfab1543684c4d781cef81252971e8ac2 100644
--- plugins/astyle/CMakeLists.txt
+++ plugins/astyle/CMakeLists.txt
@@ -30,6 +30,12 @@ target_link_libraries(kdevastyle
     KDev::Interfaces
     KDev::Util)
 
+if(UNIX AND NOT APPLE)
+    if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+        target_link_libraries(kdevastyle "-pthread")
+    endif()
+endif()
+
 add_subdirectory(3rdparty/libastyle)
 if(BUILD_TESTING)
     add_subdirectory(tests)
diff --git plugins/subversion/CMakeLists.txt plugins/subversion/CMakeLists.txt
index a7699df8758cba408715aaa376f41c2eed2667b7..a26c252bf4cfc59b58a9270aaffb06563cf0a619 100644
--- plugins/subversion/CMakeLists.txt
+++ plugins/subversion/CMakeLists.txt
@@ -66,4 +66,10 @@ target_link_libraries(kdevsubversion
     kdevsvncpp
 )
 
+if(UNIX AND NOT APPLE)
+    if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+        target_link_libraries(kdevsubversion "-pthread")
+    endif()
+endif()
+
 install(PROGRAMS org.kde.kdevelop_svn.desktop DESTINATION ${KDE_INSTALL_APPDIR})
diff --git plugins/debuggercommon/CMakeLists.txt plugins/debuggercommon/CMakeLists.txt
index f6b395f5921db4c81c5cc4dc19f15f6455ec4bc5..a25633e7c45d82599553a904088f440a13dfe99c 100644
--- plugins/debuggercommon/CMakeLists.txt
+++ plugins/debuggercommon/CMakeLists.txt
@@ -85,6 +85,12 @@ if(KF5SysGuard_FOUND)
     )
 endif()
 
+if(UNIX AND NOT APPLE)
+    if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+        target_link_libraries(kdevdebuggercommon PRIVATE "-pthread")
+    endif()
+endif()
+
 kde_target_enable_exceptions(kdevdebuggercommon PUBLIC)
 
 if(BUILD_TESTING)
