diff --git kdevplatform/shell/ktexteditorpluginintegration.cpp kdevplatform/shell/ktexteditorpluginintegration.cpp
index 80bc89afcf7c24a3f8d6877397e34c7ba392b969..967c55fbe7bf87340197c12914d2d56ac6c1985d 100644
--- kdevplatform/shell/ktexteditorpluginintegration.cpp
+++ kdevplatform/shell/ktexteditorpluginintegration.cpp
@@ -351,6 +351,20 @@ void MainWindow::addWidgetToViewBar(KTextEditor::View *view, QWidget *widget)
     m_mainWindow->viewBarContainer()->addViewBar(widget);
 }
 
+KTextEditor::View *MainWindow::openUrl(const QUrl &url, const QString &encoding)
+{
+    return activateView(KTextEditor::Editor::instance()->application()->openUrl(url, encoding));
+}
+
+bool MainWindow::showToolView(QWidget *widget)
+{
+    if (widget->parentWidget()) {
+        Core::self()->uiController()->raiseToolView(widget->parentWidget());
+        return true;
+    }
+    return false;
+}
+
 KTextEditor::MainWindow *MainWindow::interface() const
 {
     return m_interface;
diff --git kdevplatform/shell/ktexteditorpluginintegration.h kdevplatform/shell/ktexteditorpluginintegration.h
index 972d26d9bb2b4619e64b10cfce8cbe6a9fecd78b..5225bfc326df1f8ec42586c407125d3e160736bd 100644
--- kdevplatform/shell/ktexteditorpluginintegration.h
+++ kdevplatform/shell/ktexteditorpluginintegration.h
@@ -89,6 +89,9 @@ public Q_SLOTS:
     void hideViewBar(KTextEditor::View *view);
     void addWidgetToViewBar(KTextEditor::View *view, QWidget *widget);
 
+    KTextEditor::View *openUrl(const QUrl &url, const QString &encoding = QString());
+    bool showToolView(QWidget *widget);
+
 public:
     KTextEditor::MainWindow *interface() const;
 
