diff --git a/qtbase/src/corelib/kernel/qobject.cpp b/qtbase/src/corelib/kernel/qobject.cpp
index 901815c..fbe5af5 100644
--- a/qtbase/src/corelib/kernel/qobject.cpp
+++ b/qtbase/src/corelib/kernel/qobject.cpp
@@ -1615,7 +1615,7 @@ int QObject::startTimer(int interval, Qt::TimerType timerType)
     Q_D(QObject);
 
     if (Q_UNLIKELY(interval < 0)) {
-        qWarning("QObject::startTimer: Timers cannot have negative intervals");
+        qWarning() << "QObject::startTimer: Timer" << this << "cannot have a negative interval";
         return 0;
     }
     if (Q_UNLIKELY(!d->threadData->eventDispatcher.load())) {
