diff --git a/qtbase/src/plugins/platforms/xcb/gl_integrations/xcb_glx/qxcbglxwindow.cpp b/qtbase/src/plugins/platforms/xcb/gl_integrations/xcb_glx/qxcbglxwindow.cpp
index 8df8b28f..d7311a3e 100644
--- a/qtbase/src/plugins/platforms/xcb/gl_integrations/xcb_glx/qxcbglxwindow.cpp
+++ b/qtbase/src/plugins/platforms/xcb/gl_integrations/xcb_glx/qxcbglxwindow.cpp
@@ -60,7 +60,7 @@ const xcb_visualtype_t *QXcbGlxWindow::createVisual()
         return Q_NULLPTR;
     XVisualInfo *visualInfo = qglx_findVisualInfo(DISPLAY_FROM_XCB(scr), scr->screenNumber(), &m_format);
     if (!visualInfo) {
-        qWarning() << "No XVisualInfo for format" << m_format;
+        qDebug() << "No XVisualInfo for format" << m_format;
         return Q_NULLPTR;
     }
     const xcb_visualtype_t *xcb_visualtype = scr->visualForId(visualInfo->visualid);
diff --git a/qtbase/src/plugins/platforms/xcb/qxcbwindow.cpp b/qtbase/src/plugins/platforms/xcb/qxcbwindow.cpp
index 37d6336a..f0f51dd0 100644
--- a/qtbase/src/plugins/platforms/xcb/qxcbwindow.cpp
+++ b/qtbase/src/plugins/platforms/xcb/qxcbwindow.cpp
@@ -424,7 +424,7 @@ void QXcbWindow::create()
         visual = createVisual();
 
     if (!visual) {
-        qWarning() << "Falling back to using screens root_visual.";
+        qDebug() << "Falling back to using screens root_visual.";
         visual = platformScreen->visualForId(platformScreen->screen()->root_visual);
     }
 
