--- qt-everywhere-opensource-src-5.9.4/qtbase/src/corelib/tools/orig.qlocale.cpp	2018-02-21 18:22:20.585058527 +0100
+++ qt-everywhere-opensource-src-5.9.4/qtbase/src/corelib/tools/qlocale.cpp	2018-02-21 18:22:38.935058124 +0100
@@ -2172,11 +2172,14 @@
 
 #ifndef QT_NO_SYSTEMLOCALE
     if (d->m_data == systemData()) {
-        QVariant res = systemLocale()->query(type == LongFormat
+        const auto sysLoc = systemLocale();
+        if (sysLoc) {
+            QVariant res = sysLoc->query(type == LongFormat
                                              ? QSystemLocale::MonthNameLong : QSystemLocale::MonthNameShort,
                                              month);
-        if (!res.isNull())
-            return res.toString();
+            if (!res.isNull())
+                return res.toString();
+        }
     }
 #endif
 
