diff --git a/qtbase/src/widgets/styles/qmacstyle_mac.mm b/qtbase/src/widgets/styles/qmacstyle_mac.mm
index 2d4d47e..7f2a7d9 100644
--- a/qtbase/src/widgets/styles/qmacstyle_mac.mm
+++ b/qtbase/src/widgets/styles/qmacstyle_mac.mm
@@ -1798,6 +1798,7 @@ void QMacStylePrivate::setAutoDefaultButton(QObject *button) const
 
 QMacStylePrivate::QMacStylePrivate()
     : mouseDown(false), backingStoreNSView(nil)
+    , isCocoa(QGuiApplication::platformName() == QLatin1String("cocoa"))
 {
     defaultButtonStart = CFAbsoluteTimeGetCurrent();
     memset(&buttonState, 0, sizeof(ButtonState));
@@ -2017,7 +2018,7 @@ void QMacStylePrivate::drawNSViewInRect(QCocoaWidget widget, NSView *view, const
 
 void QMacStylePrivate::resolveCurrentNSView(QWindow *window)
 {
-    backingStoreNSView = window ? (NSView *)window->winId() : nil;
+    backingStoreNSView = (window && isCocoa) ? (NSView *)window->winId() : nil;
 }
 
 void QMacStylePrivate::drawColorlessButton(const HIRect &macRect, HIThemeButtonDrawInfo *bdi,
@@ -6060,7 +6061,7 @@ void QMacStyle::drawComplexControl(ComplexControl cc, const QStyleOptionComplex
                     else if (opt->styleObject)
                         window = opt->styleObject->property("_q_styleObjectWindow").value<QWindow *>();
 
-                    NSView *view = window ? (NSView *)window->winId() : nil;
+                    NSView *view = (window && d->isCocoa) ? (NSView *)window->winId() : nil;
                     bool isKey = false;
                     if (view)
                         isKey = [view.window isKeyWindow];
diff --git a/qtbase/src/widgets/styles/qmacstyle_mac_p_p.h b/qtbase/src/widgets/styles/qmacstyle_mac_p_p.h
index 228abf9..490c926 100644
--- a/qtbase/src/widgets/styles/qmacstyle_mac_p_p.h
+++ b/qtbase/src/widgets/styles/qmacstyle_mac_p_p.h
@@ -285,6 +285,7 @@ public:
     void *indicatorBranchButtonCell;
     NSView *backingStoreNSView;
     QHash<QCocoaWidget, NSView *> cocoaControls;
+    const bool isCocoa;
 };
 
 QT_END_NAMESPACE
