diff --git a/qtbase/mkspecs/features/create_cmake.prf b/qtbase/mkspecs/features/create_cmake.prf
index bb5083c..7497875 100644
--- a/qtbase/mkspecs/features/create_cmake.prf
+++ b/qtbase/mkspecs/features/create_cmake.prf
@@ -30,7 +30,7 @@ CMAKE_MODULE_NAME = $$cmakeModuleName($${MODULE})
 
 split_incpath {
     CMAKE_ADD_SOURCE_INCLUDE_DIRS = true
-    CMAKE_NO_PRIVATE_INCLUDES = true # Don't add private includes in the build dir which don't exist
+    #CMAKE_NO_PRIVATE_INCLUDES = true # Don't add private includes in the build dir which don't exist
     CMAKE_SOURCE_INCLUDES = \
         $$cmakeTargetPaths($$QT_MODULE_INCLUDE_BASE $$QT_MODULE_INCLUDE_BASE/Qt$${CMAKE_MODULE_NAME})
     CMAKE_SOURCE_PRIVATE_INCLUDES = \
@@ -53,7 +53,7 @@ contains(CMAKE_INCLUDE_DIR, "^\\.\\./.*") {
     CMAKE_INCLUDE_DIR_IS_ABSOLUTE = True
 }
 
-!exists($$first(QT.$${MODULE}_private.includes)): CMAKE_NO_PRIVATE_INCLUDES = true
+#!exists($$first(QT.$${MODULE}_private.includes)): CMAKE_NO_PRIVATE_INCLUDES = true
 
 CMAKE_LIB_DIR = $$cmakeRelativePath($$[QT_INSTALL_LIBS], $$[QT_INSTALL_PREFIX])
 contains(CMAKE_LIB_DIR,"^\\.\\./.*") {
