--- qt-everywhere-opensource-src-5.9.5/qtwebengine/examples/webenginewidgets/html2pdf/orig.html2pdf.cpp	2018-04-10 16:50:39.000000000 +0200
+++ qt-everywhere-opensource-src-5.9.5/qtwebengine/examples/webenginewidgets/html2pdf/html2pdf.cpp	2018-05-16 17:49:03.000000000 +0200
@@ -83,8 +83,8 @@
 {
     connect(m_page.data(), &QWebEnginePage::loadFinished,
             this, &Html2PdfConverter::loadFinished);
-    connect(m_page.data(), &QWebEnginePage::pdfPrintingFinished,
-            this, &Html2PdfConverter::pdfPrintingFinished);
+//     connect(m_page.data(), &QWebEnginePage::pdfPrintingFinished,
+//             this, &Html2PdfConverter::pdfPrintingFinished);
 }
 
 int Html2PdfConverter::run()
--- qt-everywhere-opensource-src-5.9.5/qtwebengine/examples/webenginewidgets/simplebrowser/orig.downloadwidget.cpp	2018-04-10 16:50:39.000000000 +0200
+++ qt-everywhere-opensource-src-5.9.5/qtwebengine/examples/webenginewidgets/simplebrowser/downloadwidget.cpp	2018-05-16 17:53:05.000000000 +0200
@@ -141,8 +141,8 @@
         m_progressBar->setValue(0);
         m_progressBar->setDisabled(true);
         m_progressBar->setFormat(
-            tr("interrupted: %1")
-            .arg(m_download->interruptReasonString()));
+            tr("interrupted")
+            );
         break;
     }
 
