--- qt-everywhere-opensource-src-5.9.4/qtwebengine/src/core/config/orig.common.pri	2018-01-15 13:00:11.000000000 +0100
+++ qt-everywhere-opensource-src-5.9.4/qtwebengine/src/core/config/common.pri	2018-02-14 22:45:19.673690472 +0100
@@ -46,13 +46,13 @@
 use?(proprietary_codecs): gn_args += proprietary_codecs=true ffmpeg_branding=\"Chrome\"
 
 CONFIG(release, debug|release) {
-    force_debug_info {
-        # Level 1 is not enough to generate all Chromium debug symbols on Windows
-        msvc: gn_args += symbol_level=2
-        else: gn_args += symbol_level=1
-    } else {
+#    force_debug_info {
+#        # Level 1 is not enough to generate all Chromium debug symbols on Windows
+#        msvc: gn_args += symbol_level=2
+#        else: gn_args += symbol_level=1
+#    } else {
         gn_args += symbol_level=0
-    }
+#    }
 }
 
 !webcore_debug: gn_args += remove_webcore_debug_symbols=true
