diff --git a/qtbase/src/plugins/bearer/bearer.pro b/qtbase/src/plugins/bearer/bearer.pro
index b362722b2828abd7673e07dd1dbda1575a2cb9e9..0f3b7465b00959d7778aa03240738db4bebe0a82 100644
--- a/qtbase/src/plugins/bearer/bearer.pro
+++ b/qtbase/src/plugins/bearer/bearer.pro
#@@ -9,7 +9,7 @@ QT_FOR_CONFIG += network-private
# #win32:SUBDIRS += nla
# win32:SUBDIRS += generic
# win32:!winrt: SUBDIRS += nativewifi
#-darwin:qtConfig(corewlan): SUBDIRS += corewlan
#+# darwin:qtConfig(corewlan): SUBDIRS += corewlan
# mac:SUBDIRS += generic
# android:SUBDIRS += android
# 
#
@@ -8,4 +8,6 @@ QT_FOR_CONFIG += network-private
 
 android:SUBDIRS += android
 
+mac:SUBDIRS += generic
+
 isEmpty(SUBDIRS):SUBDIRS = generic
