--- qt-everywhere-opensource-src-5.9.8/qtdeclarative/src/qml/jit/orig.qv4assembler.cpp	2019-04-02 12:10:14.000000000 +0200
+++ qt-everywhere-opensource-src-5.9.8/qtdeclarative/src/qml/jit/qv4assembler.cpp	2020-07-17 18:01:57.181862321 +0200
@@ -54,6 +54,13 @@
 #include <QBuffer>
 #include <QCoreApplication>
 
+#if defined(Q_OS_LINUX)
+#include <linux/limits.h>
+#ifndef PATH_MAX
+#define PATH_MAX MAXPATHLEN
+#endif
+#endif
+
 #if ENABLE(ASSEMBLER)
 
 #if USE(UDIS86)
