diff --git a/qtbase/configure b/qtbase/configure
index 08b49a8..150b32e 100755
--- a/qtbase/configure
+++ b/qtbase/configure
@@ -305,9 +305,9 @@ UNAME_RELEASE=`(uname -r) 2>/dev/null` || UNAME_RELEASE=unknown
 UNAME_SYSTEM=`(uname -s) 2>/dev/null`  || UNAME_SYSTEM=unknown
 
 BUILD_ON_MAC=no
-if [ -d /System/Library/Frameworks/Carbon.framework ]; then
-    BUILD_ON_MAC=yes
-fi
+# if [ -d /System/Library/Frameworks/Carbon.framework ]; then
+#     BUILD_ON_MAC=yes
+# fi
 if [ "$OSTYPE" = "msys" ]; then
     relpath=`(cd "$relpath"; pwd -W)`
     outpath=`pwd -W`
