diff --git a/qtbase/src/corelib/Qt5CoreConfigExtrasMkspecDir.cmake.in b/qtbase/src/corelib/Qt5CoreConfigExtrasMkspecDir.cmake.in
index c357237d0e..d5467879a0 100644
--- a/qtbase/src/corelib/Qt5CoreConfigExtrasMkspecDir.cmake.in
+++ b/qtbase/src/corelib/Qt5CoreConfigExtrasMkspecDir.cmake.in
@@ -1,6 +1,10 @@
 
+if(NOT CMAKE_MKSPEC)
+	set(CMAKE_MKSPEC "$${CMAKE_MKSPEC}")
+endif()
+
 !!IF isEmpty(CMAKE_HOST_DATA_DIR_IS_ABSOLUTE)
-set(_qt5_corelib_extra_includes \"${_qt5Core_install_prefix}/$${CMAKE_HOST_DATA_DIR}/mkspecs/$${CMAKE_MKSPEC}\")
+set(_qt5_corelib_extra_includes \"${_qt5Core_install_prefix}/$${CMAKE_HOST_DATA_DIR}/mkspecs/${CMAKE_MKSPEC}\")
 !!ELSE
-set(_qt5_corelib_extra_includes \"$${CMAKE_HOST_DATA_DIR}mkspecs/$${CMAKE_MKSPEC}\")
+set(_qt5_corelib_extra_includes \"$${CMAKE_HOST_DATA_DIR}mkspecs/${CMAKE_MKSPEC}\")
 !!ENDIF
diff --git a/qtbase/src/corelib/Qt5CoreConfigExtrasMkspecDirForInstall.cmake.in b/qtbase/src/corelib/Qt5CoreConfigExtrasMkspecDirForInstall.cmake.in
index 706304cf34..b2fff54289 100644
--- a/qtbase/src/corelib/Qt5CoreConfigExtrasMkspecDirForInstall.cmake.in
+++ b/qtbase/src/corelib/Qt5CoreConfigExtrasMkspecDirForInstall.cmake.in
@@ -1,4 +1,8 @@
 
+if(NOT CMAKE_MKSPEC)
+	set(CMAKE_MKSPEC "$${CMAKE_MKSPEC}")
+endif()
+
 !!IF isEmpty(CMAKE_INSTALL_DATA_DIR_IS_ABSOLUTE)
 set(_qt5_corelib_extra_includes \"${_qt5Core_install_prefix}/$${CMAKE_INSTALL_DATA_DIR}/mkspecs/$${CMAKE_MKSPEC}\")
 !!ELSE
