diff --git a/qtbase/mkspecs/common/macx.conf b/qtbase/mkspecs/common/macx.conf
index 4be0eb3..755a886 100644
--- a/qtbase/mkspecs/common/macx.conf
+++ b/qtbase/mkspecs/common/macx.conf
@@ -11,4 +11,9 @@ device.dir_affix = $${device.sdk}
 device.CONFIG = $${device.sdk}
 device.deployment_identifier = $${device.sdk}
 
+# X11 is not inexistant on OS X; provide the library requirements but leave it to the
+# user to make them accessible during the Qt build.
+QMAKE_LIBS_X11          = -lXext -lX11
+QMAKE_LIBS_OPENVG       = -lOpenVG
+
 include(mac.conf)
diff --git a/qtbase/src/platformsupport/linuxaccessibility/linuxaccessibility.pro b/qtbase/src/platformsupport/linuxaccessibility/linuxaccessibility.pro
index d6689c0..820a944 100644
--- a/qtbase/src/platformsupport/linuxaccessibility/linuxaccessibility.pro
+++ b/qtbase/src/platformsupport/linuxaccessibility/linuxaccessibility.pro
@@ -15,6 +15,10 @@ QDBUSXML2CPP_INTERFACE_HEADER_FLAGS = -i struct_marshallers_p.h
 
 QMAKE_USE += atspi/nolink
 
+darwin: {
+    INCLUDEPATH += $$PWD/../../gui/kernel
+}
+
 HEADERS += \
     application_p.h \
     bridge_p.h \
--- a/qtbase/src/gui/configure.json	2018-02-12 20:03:10.000000000 +0100
+++ b/qtbase/src/gui/configure.json	2018-02-12 22:14:10.000000000 +0100
@@ -1193,7 +1193,6 @@
         "xcb": {
             "label": "XCB",
             "section": "Platform plugins",
-            "autoDetect": "!config.darwin",
             "enable": "input.xcb == 'system' || input.xcb == 'qt'",
             "condition": "libs.xcb",
             "output": [ "privateFeature" ]
@@ -1202,7 +1201,6 @@
             "label": "Using system-provided XCB libraries",
             "enable": "input.xcb == 'system'",
             "disable": "input.xcb == 'qt'",
-            "autoDetect": "!config.darwin",
             "condition": "features.xcb && libs.xcb_syslibs",
             "output": [ "privateFeature" ]
         },
