--- a/qtbase/src/widgets/kernel/orig.qapplication.cpp	2024-08-02 15:35:42.141123144 +0200
+++ b/qtbase/src/widgets/kernel/qapplication.cpp	2024-08-02 15:37:29.873768602 +0200
@@ -1731,7 +1731,7 @@
 
     for (QWidgetList::ConstIterator it = all.constBegin(), cend = all.constEnd(); it != cend; ++it) {
         QWidget *w = *it;
-        if (w->isWindow() && w->windowType() != Qt::Desktop)
+        if (w && w->isWindow() && w->windowType() != Qt::Desktop)
             list.append(w);
     }
     return list;
