diff --git a/qtbase/src/plugins/sqldrivers/qsqldriverbase.pri b/qtbase/src/plugins/sqldrivers/qsqldriverbase.pri
index 4b78fa9..0a22fa1 100644
--- a/qtbase/src/plugins/sqldrivers/qsqldriverbase.pri
+++ b/qtbase/src/plugins/sqldrivers/qsqldriverbase.pri
@@ -1,7 +1,7 @@
 QT  = core core-private sql-private
 
 # For QMAKE_USE in the parent projects.
-include($$shadowed($$PWD)/qtsqldrivers-config.pri)
+!sqldrivers_standalone: include($$shadowed($$PWD)/qtsqldrivers-config.pri)
 
 PLUGIN_TYPE = sqldrivers
 load(qt_plugin)
