diff --git a/qtbase/tests/auto/corelib/plugin/qpluginloader/machtest/machtest.pro b/qtbase/tests/auto/corelib/plugin/qpluginloader/machtest/machtest.pro
index 7f7caa7..59326a6 100644
--- a/qtbase/tests/auto/corelib/plugin/qpluginloader/machtest/machtest.pro
+++ b/qtbase/tests/auto/corelib/plugin/qpluginloader/machtest/machtest.pro
@@ -9,13 +9,13 @@ load(qt)
 i386_d.target = good.i386.dylib
 i386_d.depends = EXPORT_VALID_ARCHS=i386
 i386.target = good.i386.dylib
-i386.commands = $(CXX) $(CXXFLAGS) -shared -o $@ -I$(INCPATH) $<
+i386.commands = $(CXX) $(filter-out -arch i386 x86_64,$(CXXFLAGS)) -shared -o $@ -I$(INCPATH) $<
 i386.depends += $$PWD/../fakeplugin.cpp
 
 x86_64_d.target = good.x86_64.dylib
 x86_64_d.depends = EXPORT_VALID_ARCHS=x86_64
 x86_64.target = good.x86_64.dylib
-x86_64.commands = $(CXX) $(CXXFLAGS) -shared -o $@ -I$(INCPATH) $<
+x86_64.commands = $(CXX) $(filter-out -arch i386 x86_64,$(CXXFLAGS)) -shared -o $@ -I$(INCPATH) $<
 x86_64.depends += $$PWD/../fakeplugin.cpp
 
 # Current Mac OS X toolchains have no compiler for PPC anymore
