diff --git a/qtbase/mkspecs/features/toolchain.prf b/qtbase/mkspecs/features/toolchain.prf
index ba41598..b592829 100644
--- a/qtbase/mkspecs/features/toolchain.prf
+++ b/qtbase/mkspecs/features/toolchain.prf
@@ -38,7 +38,7 @@ isEmpty($${target_prefix}.INCDIRS) {
         else: \
             cmd_prefix = "LC_ALL=C"
 
-        cxx_flags = $$QMAKE_CXXFLAGS
+        !darwin: cxx_flags = $$QMAKE_CXXFLAGS
 
         # Manually inject the sysroot for Apple Platforms because its resolution
         # normally does not happen until default_post.prf. This is especially
@@ -47,6 +47,9 @@ isEmpty($${target_prefix}.INCDIRS) {
         # UIKit simulator platforms will see the device SDK's sysroot in
         # QMAKE_DEFAULT_*DIRS, because they're handled in a single build pass.
         darwin {
+            opts = $$split(QMAKE_CXXFLAGS, " ")
+            for (opt, opts): \
+                !equals(opt, "-g"): cxx_flags += $$opt
             uikit:macx-xcode: \
                 cxx_flags += -isysroot $$sdk_path_device.value
             else: \
@@ -149,7 +152,7 @@ isEmpty($${target_prefix}.INCDIRS) {
     }
 
     unix {
-        isEmpty(QMAKE_DEFAULT_INCDIRS): QMAKE_DEFAULT_INCDIRS = /usr/include /usr/local/include
+        isEmpty(QMAKE_DEFAULT_INCDIRS): QMAKE_DEFAULT_INCDIRS = /usr/include /opt/local/include
         isEmpty(QMAKE_DEFAULT_LIBDIRS): QMAKE_DEFAULT_LIBDIRS = /lib /usr/lib
     }
 
