diff --git a/libexec/qt5/Library/Frameworks/QtWebEngineWidgets.framework/Versions/5/Headers/qwebenginepage.h b/libexec/qt5/Library/Frameworks/QtWebEngineWidgets.framework/Versions/5/Headers/qwebenginepage.h
index e85f9b3..ae99881 100644
--- a/libexec/qt5/Library/Frameworks/QtWebEngineWidgets.framework/Versions/5/Headers/qwebenginepage.h
+++ b/libexec/qt5/Library/Frameworks/QtWebEngineWidgets.framework/Versions/5/Headers/qwebenginepage.h
@@ -130,6 +130,7 @@ public:
         ViewSource,
         WebActionCount
     };
+    Q_ENUM(WebAction)
 
     enum FindFlag {
         FindBackward = 1,
@@ -143,12 +144,14 @@ public:
         WebDialog,
         WebBrowserBackgroundTab
     };
+    Q_ENUM(WebWindowType)
 
     enum PermissionPolicy {
         PermissionUnknown,
         PermissionGrantedByUser,
         PermissionDeniedByUser
     };
+    Q_ENUM(PermissionPolicy)
 
     // must match WebContentsAdapterClient::NavigationType
     enum NavigationType {
@@ -159,6 +162,7 @@ public:
         NavigationTypeReload,
         NavigationTypeOther
     };
+    Q_ENUM(NavigationType)
 
     enum Feature {
 #ifndef Q_QDOC
@@ -170,6 +174,7 @@ public:
         MediaAudioVideoCapture,
         MouseLock
     };
+    Q_ENUM(Feature)
 
     // Ex-QWebFrame enum
 
@@ -177,6 +182,7 @@ public:
         FileSelectOpen,
         FileSelectOpenMultiple,
     };
+    Q_ENUM(FileSelectionMode)
 
     // must match WebContentsAdapterClient::JavaScriptConsoleMessageLevel
     enum JavaScriptConsoleMessageLevel {
@@ -184,6 +190,7 @@ public:
         WarningMessageLevel,
         ErrorMessageLevel
     };
+    Q_ENUM(JavaScriptConsoleMessageLevel)
 
     // must match WebContentsAdapterClient::RenderProcessTerminationStatus
     enum RenderProcessTerminationStatus {
@@ -192,6 +199,7 @@ public:
         CrashedTerminationStatus,
         KilledTerminationStatus
     };
+    Q_ENUM(RenderProcessTerminationStatus)
 
     explicit QWebEnginePage(QObject *parent = Q_NULLPTR);
     QWebEnginePage(QWebEngineProfile *profile, QObject *parent = Q_NULLPTR);
