diff --git a/qtmultimedia/src/gsttools/gsttools.pro b/qtmultimedia/src/gsttools/gsttools.pro
index edbf603e30aa3619d361fac5d9cc8352f63aa1d6..45ea6fe8d6bba78340d855967b41abbc24036e34 100644
--- a/qtmultimedia/src/gsttools/gsttools.pro
+++ b/qtmultimedia/src/gsttools/gsttools.pro
@@ -103,5 +103,8 @@ HEADERS += $$PRIVATE_HEADERS
 
 DESTDIR = $$QT.multimedia.libs
 target.path = $$[QT_INSTALL_LIBS]
+darwin {
+    QMAKE_SONAME_PREFIX = $$target.path
+}
 
 INSTALLS += target
diff --git a/qtmultimedia/src/plugins/gstreamer/common.pri b/qtmultimedia/src/plugins/gstreamer/common.pri
index cbe87be4f9ee0333534b0527b5027c4a265d9c60..8bdba281433914bd275caceda7944caf69936461 100644
--- a/qtmultimedia/src/plugins/gstreamer/common.pri
+++ b/qtmultimedia/src/plugins/gstreamer/common.pri
@@ -5,7 +5,7 @@ qtHaveModule(widgets) {
     DEFINES += HAVE_WIDGETS
 }
 
-LIBS += -lqgsttools_p
+LIBS += -L$$OUT_PWD/../../../../lib -lqgsttools_p
 
 QMAKE_USE += gstreamer
 
diff --git a/qtmultimedia/src/plugins/plugins.pro b/qtmultimedia/src/plugins/plugins.pro
index 98a1bf2429ada348f83425c6ab9dcd6fdb115be0..0d9bcfb1c5b2245640941180e61a3586f58833b5 100644
--- a/qtmultimedia/src/plugins/plugins.pro
+++ b/qtmultimedia/src/plugins/plugins.pro
@@ -43,7 +43,7 @@ winrt {
     SUBDIRS += winrt
 }
 
-unix:!mac:!android {
+unix:!android {
     qtConfig(gstreamer) {
        SUBDIRS += gstreamer
     } else {
