diff --git a/qtbase/src/widgets/widgets/qtabbar.cpp b/qtbase/src/widgets/widgets/qtabbar.cpp
index adaafc7..0b8b0e3 100644
--- a/qtbase/src/widgets/widgets/qtabbar.cpp
+++ b/qtbase/src/widgets/widgets/qtabbar.cpp
@@ -180,6 +180,13 @@ void QTabBarPrivate::initBasicStyleOption(QStyleOptionTab *option, int tabIndex)
 
     if (tab.textColor.isValid())
         option->palette.setColor(q->foregroundRole(), tab.textColor);
+#ifdef Q_OS_MACOS
+    else if (isCurrent && !documentMode
+             && (QSysInfo::MacintoshVersion < QSysInfo::MV_10_10 || q->isActiveWindow())
+             && q->style()->objectName() == QStringLiteral("macintosh")) {
+        option->palette.setColor(QPalette::WindowText, Qt::white);
+    }
+#endif
     option->icon = tab.icon;
     option->iconSize = q->iconSize();  // Will get the default value then.
 
