diff --git a/qtbase/src/gui/text/qfontengine.cpp b/qtbase/src/gui/text/qfontengine.cpp
index 29c48da7be84e92150aab4e084ca41e1199d9dd0..9694910a67cdf5bbd9e3a166ec19312a2d982027 100644
--- a/qtbase/src/gui/text/qfontengine.cpp
+++ b/qtbase/src/gui/text/qfontengine.cpp
@@ -368,9 +368,13 @@ bool QFontEngine::supportsScript(QChar::Script script) const
     if (qt_useHarfbuzzNG()) {
 #if defined(Q_OS_DARWIN)
         // in AAT fonts, 'gsub' table is effectively replaced by 'mort'/'morx' table
+        // RJVB: assuming this only applies to using CoreText fonts, I've had crashes
+        // in Freetype (under QFontEngineFT::getSfntTableData()) due to the test below.
         uint len;
-        if (getSfntTableData(MAKE_TAG('m','o','r','t'), 0, &len) || getSfntTableData(MAKE_TAG('m','o','r','x'), 0, &len))
-            return true;
+        if (type() != QFontEngine::Freetype) {
+            if (getSfntTableData(MAKE_TAG('m','o','r','t'), 0, &len) || getSfntTableData(MAKE_TAG('m','o','r','x'), 0, &len))
+                return true;
+        }
 #endif
 
         bool ret = false;
