diff --git a/qtbase/src/gui/configure.json b/qtbase/src/gui/configure.json
index c265888..e2eb8bf 100644
--- a/qtbase/src/gui/configure.json
+++ b/qtbase/src/gui/configure.json
@@ -983,7 +983,7 @@
         "vnc": {
             "label": "VNC",
             "section": "Platform plugins",
-            "condition": "config.unix && !config.android && !config.darwin && features.regularexpression",
+            "condition": "config.unix && !config.android && features.regularexpression",
             "output": [ "privateFeature" ]
         },
         "mirclient": {
diff --git a/qtbase/src/plugins/platforms/vnc/qvncscreen.cpp b/qtbase/src/plugins/platforms/vnc/qvncscreen.cpp
index 67d33de..43e51f3 100644
--- a/qtbase/src/plugins/platforms/vnc/qvncscreen.cpp
+++ b/qtbase/src/plugins/platforms/vnc/qvncscreen.cpp
@@ -46,7 +46,6 @@
 #include <QtGui/QScreen>
 #include <QtCore/QRegularExpression>
 
-
 QT_BEGIN_NAMESPACE
 
 
@@ -69,22 +68,33 @@ bool QVncScreen::initialize()
     QRegularExpression sizeRx(QLatin1String("size=(\\d+)x(\\d+)"));
     QRegularExpression mmSizeRx(QLatin1String("mmsize=(?<width>(\\d*\\.)?\\d+)x(?<height>(\\d*\\.)?\\d+)"));
     QRegularExpression depthRx(QLatin1String("depth=(\\d+)"));
+    QRegularExpression densityRx(QLatin1String("density=(\\d+)"));
 
     mGeometry = QRect(0, 0, 1024, 768);
     mFormat = QImage::Format_ARGB32_Premultiplied;
     mDepth = 32;
-    mPhysicalSize = QSizeF(mGeometry.width()/96.*25.4, mGeometry.height()/96.*25.4);
+    double density = 96.0;
+    mPhysicalSize = QSizeF(0.0, 0.0);
 
     for (const QString &arg : mArgs) {
         QRegularExpressionMatch match;
         if (arg.contains(mmSizeRx, &match)) {
             mPhysicalSize = QSizeF(match.captured("width").toDouble(), match.captured("height").toDouble());
-        } else if (arg.contains(sizeRx, &match)) {
-            mGeometry.setSize(QSize(match.captured(1).toInt(), match.captured(2).toInt()));
-        } else if (arg.contains(depthRx, &match)) {
-            mDepth = match.captured(1).toInt();
+        } else {
+            if (arg.contains(densityRx, &match)) {
+                density = match.captured(1).toDouble();
+            }
+            if (arg.contains(sizeRx, &match)) {
+                mGeometry.setSize(QSize(match.captured(1).toInt(), match.captured(2).toInt()));
+            }
+            if (arg.contains(depthRx, &match)) {
+                mDepth = match.captured(1).toInt();
+            }
         }
     }
+    if (mPhysicalSize.isEmpty()) {
+        mPhysicalSize = QSizeF(mGeometry.width()/density*25.4, mGeometry.height()/density*25.4);
+    }
 
     QFbScreen::initializeCompositor();
 
@@ -139,8 +149,7 @@ void QVncScreen::enableClientCursor(QVncClient *client)
 void QVncScreen::disableClientCursor(QVncClient *client)
 {
 #if QT_CONFIG(cursor)
-    uint clientCount = clientCursor->removeClient(client);
-    if (clientCount == 0) {
+    if (clientCursor && clientCursor->removeClient(client) == 0) {
         delete clientCursor;
         clientCursor = nullptr;
     }
