diff --git a/qtimageformats/src/plugins/imageformats/jp2/qjp2handler.cpp b/qtimageformats/src/plugins/imageformats/jp2/qjp2handler.cpp
index 17373fd..0f8e2f8 100644
--- a/qtimageformats/src/plugins/imageformats/jp2/qjp2handler.cpp
+++ b/qtimageformats/src/plugins/imageformats/jp2/qjp2handler.cpp
@@ -46,6 +46,8 @@
 
 #include <jasper/jasper.h>
 
+#include <math.h>
+
 QT_BEGIN_NAMESPACE
 
 class QJp2HandlerPrivate
