diff --git a/qtbase/mkspecs/features/qt_common.prf b/qtbase/mkspecs/features/qt_common.prf
index b8e397cda3893bd015b6c1bf3271c4896912267a..60287245e92afb94ca619d1dc8d81961a84d8ad6 100644
--- a/qtbase/mkspecs/features/qt_common.prf
+++ b/qtbase/mkspecs/features/qt_common.prf
@@ -56,7 +56,8 @@ host_build:cross_compile: return()
 # -Wvla: use of variable-length arrays (an extension to C++)
 clang {
     clang_ver = $${QT_CLANG_MAJOR_VERSION}.$${QT_CLANG_MINOR_VERSION}
-    versionAtLeast(clang_ver, 3.5): \
+    # Clang 3.5 introduced -Wdate-time but Apple's clang on OS X 10.9 doesn't know it yet
+    versionAtLeast(clang_ver, 3.6): \
         QMAKE_CXXFLAGS_WARN_ON += -Wdate-time
 } else: gcc:!intel_icc {
     QMAKE_CXXFLAGS_WARN_ON += -Wvla
