diff --git src/settings/services/servicemenuinstaller/CMakeLists.txt src/settings/services/servicemenuinstaller/CMakeLists.txt
index b5591cad7..c5423a76a 100644
--- src/settings/services/servicemenuinstaller/CMakeLists.txt
+++ src/settings/services/servicemenuinstaller/CMakeLists.txt
@@ -1,7 +1,9 @@
+include(ECMMarkNonGuiExecutable)
 remove_definitions(-DTRANSLATION_DOMAIN=\"dolphin\")
 add_definitions(-DTRANSLATION_DOMAIN=\"dolphin_servicemenuinstaller\")
 
 add_executable(servicemenuinstaller servicemenuinstaller.cpp)
+ecm_mark_nongui_executable(servicemenuinstaller)
 target_link_libraries(servicemenuinstaller PRIVATE
     Qt5::Core
     KF5::I18n
