diff --git src/dolphinmainwindow.cpp src/dolphinmainwindow.cpp
index c9ed9c5cd5045b2d4f58bd8f7e171159d62f03da..13f5ab090a8bae7d8c179c4ef28958f4c72cd694 100644
--- src/dolphinmainwindow.cpp
+++ src/dolphinmainwindow.cpp
@@ -91,6 +91,15 @@ namespace {
     const int CurrentDolphinVersion = 200;
 }
 
+static inline bool canHideMenuBar()
+{
+#ifdef Q_OS_OSX
+    return !QGuiApplication::platformName().contains(QLatin1String("cocoa"));
+#else
+    return true;
+#endif
+}
+
 DolphinMainWindow::DolphinMainWindow() :
     KXmlGuiWindow(nullptr, Qt::WindowContextHelpButtonHint),
     m_newFileMenu(nullptr),
@@ -173,11 +182,13 @@ DolphinMainWindow::DolphinMainWindow() :
         resize(750, 500);
     }
 
-    const bool showMenu = !menuBar()->isHidden();
-    QAction* showMenuBarAction = actionCollection()->action(KStandardAction::name(KStandardAction::ShowMenubar));
-    showMenuBarAction->setChecked(showMenu);  // workaround for bug #171080
-    if (!showMenu) {
-        createControlButton();
+    if (canHideMenuBar()) {
+        const bool showMenu = !menuBar()->isHidden();
+        QAction* showMenuBarAction = actionCollection()->action(KStandardAction::name(KStandardAction::ShowMenubar));
+        showMenuBarAction->setChecked(showMenu);  // workaround for bug #171080
+        if (!showMenu) {
+            createControlButton();
+        }
     }
 
     // enable middle-click on back/forward/up to open in a new tab
@@ -830,12 +841,14 @@ void DolphinMainWindow::compareFiles()
 
 void DolphinMainWindow::toggleShowMenuBar()
 {
-    const bool visible = menuBar()->isVisible();
-    menuBar()->setVisible(!visible);
-    if (visible) {
-        createControlButton();
-    } else {
-        deleteControlButton();
+    if (canHideMenuBar()) {
+        const bool visible = menuBar()->isVisible();
+        menuBar()->setVisible(!visible);
+        if (visible) {
+            createControlButton();
+        } else {
+            deleteControlButton();
+        }
     }
 }
 
@@ -1047,7 +1060,7 @@ void DolphinMainWindow::updateControlMenu()
 
 void DolphinMainWindow::updateToolBar()
 {
-    if (!menuBar()->isVisible()) {
+    if (canHideMenuBar() && !menuBar()->isVisible()) {
         createControlButton();
     }
 }
