diff --git a/qtbase/src/corelib/io/qstandardpaths_unix.cpp b/qtbase/src/corelib/io/qstandardpaths_unix.cpp
index 78f7548..58b6fec 100644
--- a/qtbase/src/corelib/io/qstandardpaths_unix.cpp
+++ b/qtbase/src/corelib/io/qstandardpaths_unix.cpp
@@ -269,7 +269,7 @@ static QStringList xdgDataDirs()
     // http://standards.freedesktop.org/basedir-spec/latest/
     QString xdgDataDirsEnv = QFile::decodeName(qgetenv("XDG_DATA_DIRS"));
     if (xdgDataDirsEnv.isEmpty()) {
-        dirs.append(QString::fromLatin1("/usr/local/share"));
+        dirs.append(QString::fromLatin1("@PREFIX@/share"));
         dirs.append(QString::fromLatin1("/usr/share"));
     } else {
         dirs = xdgDataDirsEnv.split(QLatin1Char(':'), QString::SkipEmptyParts);
@@ -300,10 +300,12 @@ static QStringList xdgConfigDirs()
     QStringList dirs;
     // http://standards.freedesktop.org/basedir-spec/latest/
     const QString xdgConfigDirs = QFile::decodeName(qgetenv("XDG_CONFIG_DIRS"));
-    if (xdgConfigDirs.isEmpty())
+    if (xdgConfigDirs.isEmpty()) {
+        dirs.append(QString::fromLatin1("@PREFIX@/etc/xdg"));
         dirs.append(QString::fromLatin1("/etc/xdg"));
-    else
+    } else {
         dirs = xdgConfigDirs.split(QLatin1Char(':'));
+    }
     return dirs;
 }
 
