diff --git a/qtbase/tests/auto/testlib/selftests/benchlibcallgrind/tst_benchlibcallgrind.cpp b/qtbase/tests/auto/testlib/selftests/benchlibcallgrind/tst_benchlibcallgrind.cpp
index 2a4102e..d185358 100644
--- a/qtbase/tests/auto/testlib/selftests/benchlibcallgrind/tst_benchlibcallgrind.cpp
+++ b/qtbase/tests/auto/testlib/selftests/benchlibcallgrind/tst_benchlibcallgrind.cpp
@@ -73,9 +73,9 @@ void tst_BenchlibCallgrind::twoHundredMillionInstructions()
     QBENCHMARK {
         __asm__ __volatile__(
             "mov $100000000,%%eax   \n"
-            "LOOPTOP:               \n"
+            "1:                     \n"
             "dec %%eax              \n"
-            "jnz LOOPTOP            \n"
+            "jnz 1b                 \n"
             : /* no output */
             : /* no input */
             : /* clobber */ "eax"
