--- src/orig.main.cpp	2017-08-02 13:25:36.000000000 +0200
+++ src/main.cpp	2017-08-02 15:06:25.000000000 +0200
@@ -76,12 +76,19 @@
 #endif
 
     if (!libq4wine.load()){
-        qDebug()<<libq4wine.errorString();
-        libq4wine.load();
+        qWarning()<<libq4wine.errorString();
+        if (!libq4wine.load()) {
+            qCritical() << "Second attempt to load libq4wine-core failed too:" << libq4wine.errorString();
+            return -1;
+        }
     }
 
     // Getting corelib calss pointer
     CoreLibClassPointer = (CoreLibPrototype *) libq4wine.resolve("createCoreLib");
+    if (!CoreLibClassPointer) {
+        qCritical() << "Failed to resolve createCoreLib():" << libq4wine.errorString();
+        return -1;
+    }
     CoreLib.reset((corelib *)CoreLibClassPointer(true));
 
     if (!CoreLib.get()){
Exit 1
--- src//orig.CMakeLists.txt	2017-08-01 23:24:55.000000000 +0200
+++ src//CMakeLists.txt	2017-08-02 15:11:30.000000000 +0200
@@ -214,7 +214,7 @@
 file(APPEND ${CMAKE_SOURCE_DIR}/config.h "#define APP_PREF \"${CMAKE_INSTALL_PREFIX}\"\n" )
 
 IF( ${CMAKE_SYSTEM_NAME} MATCHES Darwin )
-    file(APPEND ${CMAKE_SOURCE_DIR}/config.h "#define _CORELIB_PATH_ \"q4wine.app/Contents/Frameworks/libq4wine-core\"\n" )
+    file(APPEND ${CMAKE_SOURCE_DIR}/config.h "#define _CORELIB_PATH_ QString(\"%1/../Frameworks/libq4wine-core.dylib\").arg(QCoreApplication::applicationDirPath())\n" )
 ELSE( ${CMAKE_SYSTEM_NAME} MATCHES Darwin )
     file(APPEND ${CMAKE_SOURCE_DIR}/config.h "#define _CORELIB_PATH_ \"libq4wine-core\"\n" )
 ENDIF( ${CMAKE_SYSTEM_NAME} MATCHES Darwin )
--- src/q4wine-lib/orig.q4wine-lib.cpp	2017-08-02 17:48:37.000000000 +0200
+++ src/q4wine-lib/q4wine-lib.cpp	2017-08-02 17:50:36.000000000 +0200
@@ -980,7 +980,7 @@
         bool corelib::runWineBinary(const ExecObject execObj, QString prefix_name, bool detach){
 #ifdef RELEASE
     #ifdef _OS_DARWIN_
-            QString binary = QString("%1/%2.app/Contents/MacOS/q4wine-helper").arg(QDir::currentPath(), APP_SHORT_NAME);
+            QString binary = QString("%1/q4wine-helper").arg(QCoreApplication::applicationDirPath());
     #else
             QString binary = QString("%1/bin/q4wine-helper").arg(QString::fromUtf8(APP_PREF));
     #endif
@@ -1059,9 +1059,9 @@
             args.append("--wrkdir");
             args.append(wrkdir);
 
-#ifdef DEBUG
+// #ifdef DEBUG
             qDebug()<<"[ii] corelib::runWineBinary: "<<binary<<args<<" at: "<<wrkdir;
-#endif
+// #endif
 
             if (detach){
                 QProcess proc(0);
