--- src/q4wine-gui/orig.appsettings.cpp	2017-05-02 11:55:05.000000000 +0200
+++ src/q4wine-gui/appsettings.cpp	2017-08-02 12:25:43.000000000 +0200
@@ -794,14 +794,18 @@
 }
 
 void AppSettings::cmdDesktopMenu_Regen_Click(){
+#ifndef _OS_DARWIN_
     system_menu sys_menu;
     sys_menu.create_dir_info();
     sys_menu.generateSystemMenu();
+#endif
 }
 
 void AppSettings::cmdDesktopMenu_Remove_Click(){
+#ifndef _OS_DARWIN_
     system_menu sys_menu;
     sys_menu.wipeSystemMenu();
+#endif
 }
 
 void AppSettings::cmdVersionManager_Click(){
