--- src/q4wine-gui/orig.process.cpp	2017-05-02 11:55:04.000000000 +0200
+++ src/q4wine-gui/process.cpp	2017-08-02 12:19:05.000000000 +0200
@@ -17,6 +17,9 @@
  ***************************************************************************/
 
 #include "process.h"
+#ifdef Q_OS_MACOS
+#include <locale.h>
+#endif
 
 Process::Process (QStringList args, QString exec, QString dir, QString info, QString caption, bool showErr, QStringList env, QWidget * parent, Qt::WindowFlags f)
 {
