diff --git kstyle/demo/CMakeLists.txt kstyle/demo/CMakeLists.txt
index 6aebd9bba2c06cfbcd34853495c8ee095db01d67..6b6a1bbc4dd42cad7f96468828fd07999773e8bc 100644
--- kstyle/demo/CMakeLists.txt
+++ kstyle/demo/CMakeLists.txt
@@ -46,6 +46,12 @@ if(OXYGEN_USE_KDE4)
 
 else()
 
+  find_package(KF5ConfigWidgets CONFIG)
+  if(KF5ConfigWidgets_FOUND)
+    set(oxygen_demo_SOURCES ${oxygen_demo_SOURCES} oxygenschemechooser.cpp)
+    add_definitions(-DHAVE_SCHEME_CHOOSER)
+  endif()
+
   ki18n_wrap_ui(oxygen_demo_FORMS_HEADERS ${oxygen_demo_FORMS})
 
   add_executable(oxygen-demo5
@@ -55,6 +61,9 @@ else()
 
   target_link_libraries(oxygen-demo5 Qt5::Core Qt5::Gui Qt5::Widgets)
   target_link_libraries(oxygen-demo5 KF5::Completion KF5::ConfigGui KF5::WidgetsAddons KF5::I18n)
+  if(KF5ConfigWidgets_FOUND)
+    target_link_libraries(oxygen-demo5 KF5::ConfigWidgets)
+  endif()
 
   ########### install files ###############
   install(TARGETS oxygen-demo5 ${INSTALL_TARGETS_DEFAULT_ARGS})
diff --git kstyle/demo/oxygendemodialog.cpp kstyle/demo/oxygendemodialog.cpp
index 4a742e3068a06bfc5fbd7b7c973ea7e5fbfcaaec..436799a0b7a34957702e375d91b16603c34b760d 100644
--- kstyle/demo/oxygendemodialog.cpp
+++ kstyle/demo/oxygendemodialog.cpp
@@ -34,6 +34,9 @@
 #include "oxygensliderdemowidget.h"
 #include "oxygentabdemowidget.h"
 #include "config-liboxygen.h"
+#ifdef HAVE_SCHEME_CHOOSER
+#include "oxygenschemechooser.h"
+#endif
 
 #include <QTextStream>
 #include <QShortcut>
@@ -78,6 +81,12 @@ namespace Oxygen
         connect( _rightToLeftCheckBox, SIGNAL(toggled(bool)), SLOT(toggleRightToLeft(bool)) );
         buttonBox->addButton( _rightToLeftCheckBox, QDialogButtonBox::ResetRole );
 
+#ifdef HAVE_SCHEME_CHOOSER
+        ColorSchemeChooser *colChooser = new ColorSchemeChooser( this );
+        colChooser->setFlat( true );
+        buttonBox->addButton( colChooser, QDialogButtonBox::ResetRole );
+#endif
+
         // connections
         connect( pageWidget, SIGNAL(currentPageChanged(KPageWidgetItem*,KPageWidgetItem*)), SLOT(updateWindowTitle(KPageWidgetItem*)) );
         connect( pageWidget, SIGNAL(currentPageChanged(KPageWidgetItem*,KPageWidgetItem*)), SLOT(updateEnableState(KPageWidgetItem*)) );
diff --git kstyle/demo/oxygenschemechooser.cpp kstyle/demo/oxygenschemechooser.cpp
new file mode 100644
index 0000000000000000000000000000000000000000..1d5e2f131029bc50c3037606f3e18cdb20c9f33f
--- /dev/null
+++ kstyle/demo/oxygenschemechooser.cpp
@@ -0,0 +1,81 @@
+/*************************************************************************************
+ * This file was part of KDevPlatform and now of Oxygen                              *
+ * Copyright 2016 Zhigalin Alexander <alexander@zhigalin.tk>                         *
+ * Copyright 2017 René J.V. Bertin <rjvbertin@gmail.com>                             *
+ *                                                                                   *
+ * This library is free software; you can redistribute it and/or                     *
+ * modify it under the terms of the GNU Lesser General Public                        *
+ * License as published by the Free Software Foundation; either                      *
+ * version 2.1 of the License, or (at your option) version 3, or any                 *
+ * later version accepted by the membership of KDE e.V. (or its                      *
+ * successor approved by the membership of KDE e.V.), which shall                    *
+ * act as a proxy defined in Section 6 of version 3 of the license.                  *
+ *                                                                                   *
+ * This library is distributed in the hope that it will be useful,                   *
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of                    *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU                 *
+ * Lesser General Public License for more details.                                   *
+ *                                                                                   *
+ * You should have received a copy of the GNU Lesser General Public                  *
+ * License along with this library.  If not, see <http://www.gnu.org/licenses/>.     *
+ *************************************************************************************/
+
+#include "oxygenschemechooser.h"
+
+#include <QStringList>
+#include <QActionGroup>
+#include <QAction>
+#include <QMenu>
+#include <QModelIndex>
+
+#include <KActionMenu>
+#include <KColorSchemeManager>
+#include <KLocalizedString>
+#include <KSharedConfig>
+#include <KConfigGroup>
+
+namespace Oxygen
+{
+
+ColorSchemeChooser::ColorSchemeChooser( QWidget* parent )
+    : QPushButton( parent )
+{
+    auto manager = new KColorSchemeManager( parent );
+
+    const auto scheme( currentSchemeName() );
+
+    auto selectionMenu = manager->createSchemeSelectionMenu( scheme, this );
+
+    manager->activateScheme(manager->indexForScheme(scheme));
+
+    setMenu( selectionMenu->menu() );
+    menu()->setIcon( QIcon::fromTheme( QStringLiteral( "preferences-desktop-color" ) ) );
+    menu()->setTitle( i18n( "&Color Theme" ) );
+    setIcon( menu()->icon() );
+    setText( menu()->title() );
+}
+
+QString ColorSchemeChooser::currentDesktopDefaultScheme() const
+{
+    KSharedConfigPtr config = KSharedConfig::openConfig( QLatin1String( "kdeglobals" ) );
+    KConfigGroup group( config, "General" );
+    return group.readEntry( "ColorScheme", QStringLiteral( "Breeze" ) );
+}
+
+QString ColorSchemeChooser::currentSchemeName() const
+{
+    if ( !menu() )
+    {
+        return currentDesktopDefaultScheme();
+    }
+
+    QAction* const action = menu()->activeAction();
+
+    if ( action )
+    {
+        return KLocalizedString::removeAcceleratorMarker( action->text() );
+    }
+    return currentDesktopDefaultScheme();
+}
+
+}  // namespace Oxygen
diff --git kstyle/demo/oxygenschemechooser.h kstyle/demo/oxygenschemechooser.h
new file mode 100644
index 0000000000000000000000000000000000000000..3176e37e12f8fcb7d30eb242e4aebda962a27457
--- /dev/null
+++ kstyle/demo/oxygenschemechooser.h
@@ -0,0 +1,51 @@
+/*************************************************************************************
+ * This file was part of KDevPlatform and now of Oxygen                              *
+ * Copyright 2016 Zhigalin Alexander <alexander@zhigalin.tk>                         *
+ *                                                                                   *
+ * This library is free software; you can redistribute it and/or                     *
+ * modify it under the terms of the GNU Lesser General Public                        *
+ * License as published by the Free Software Foundation; either                      *
+ * version 2.1 of the License, or (at your option) version 3, or any                 *
+ * later version accepted by the membership of KDE e.V. (or its                      *
+ * successor approved by the membership of KDE e.V.), which shall                    *
+ * act as a proxy defined in Section 6 of version 3 of the license.                  *
+ *                                                                                   *
+ * This library is distributed in the hope that it will be useful,                   *
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of                    *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU                 *
+ * Lesser General Public License for more details.                                   *
+ *                                                                                   *
+ * You should have received a copy of the GNU Lesser General Public                  *
+ * License along with this library.  If not, see <http://www.gnu.org/licenses/>.     *
+ *************************************************************************************/
+
+#ifndef COLORSCHEMECHOOSER_H
+#define COLORSCHEMECHOOSER_H
+
+#include <QObject>
+#include <QAction>
+#include <QPushButton>
+#include <QString>
+
+namespace Oxygen
+{
+/**
+ * Provides a menu that will offer to change the color scheme
+ *
+ * Furthermore, it will save the selection in the user configuration.
+ */
+class ColorSchemeChooser : public QPushButton
+{
+    Q_OBJECT
+    public:
+        explicit ColorSchemeChooser( QWidget* parent );
+
+        QString currentSchemeName() const;
+
+    private:
+        QString currentDesktopDefaultScheme() const;
+};
+
+}  // namespace Oxygen
+
+#endif // COLORSCHEMECHOOSER_H
