diff --git src/kdeinit/kinit.cpp src/kdeinit/kinit.cpp
index 02ab7945489b24da9f77644fb9bb6359f8ae932e..97daced07c31f0f7673074236664dcbb37c88f3b 100644
--- src/kdeinit/kinit.cpp
+++ src/kdeinit/kinit.cpp
@@ -169,2 +169,11 @@ static void cleanup_fds()
     }
+#ifdef Q_OS_MACOS
+    for (int fd = 3; fd < maxfd; ++fd) {
+        // set file to be closed after a successful exec*()
+        // This seems to be safe even with guarded files which
+        // would raise EXC_GUARDED when we close them.
+        // (https://bugs.kde.org/show_bug.cgi?id=405227)
+        fcntl(fd, F_SETFD, FD_CLOEXEC);
+    }
+#else
     for (int fd = 3; fd < maxfd; ++fd) {
@@ -175,2 +184,3 @@ static void cleanup_fds()
     }
+#endif
 }
