diff --git a/scintilla/gtk/ScintillaGTK.cxx b/scintilla/gtk/ScintillaGTK.cxx
index d2780a9..8aefeff 100644
--- a/scintilla/gtk/ScintillaGTK.cxx
+++ b/scintilla/gtk/ScintillaGTK.cxx
@@ -1886,7 +1886,8 @@ gint ScintillaGTK::PressThis(GdkEventButton *event) {
 
 		gtk_widget_grab_focus(PWidget(wMain));
 		if (event->button == 1) {
-#if PLAT_GTK_MACOSX
+#if PLAT_GTK_MACOSX_QUARTZ
+#warn "using Quartz/Cocoa keybindings"
 			const bool meta = ctrl;
 			// GDK reports the Command modifier key as GDK_MOD2_MASK for button events,
 			// not GDK_META_MASK like in key events.
@@ -1914,7 +1915,8 @@ gint ScintillaGTK::PressThis(GdkEventButton *event) {
 				gdk_window_get_origin(PWindow(wMain), &ox, &oy);
 				ContextMenu(Point(pt.x + ox, pt.y + oy));
 			} else {
-#if PLAT_GTK_MACOSX
+#if PLAT_GTK_MACOSX_QUARTZ
+#warn "using Quartz/Cocoa keybindings"
 				const bool meta = ctrl;
 				// GDK reports the Command modifier key as GDK_MOD2_MASK for button events,
 				// not GDK_META_MASK like in key events.
@@ -2286,9 +2288,10 @@ gboolean ScintillaGTK::KeyThis(GdkEventKey *event) {
 			key = KeyTranslate(key);
 
 		bool consumed = false;
-#if !(PLAT_GTK_MACOSX)
+#if !(PLAT_GTK_MACOSX_QUARTZ)
 		const bool meta = false;
 #else
+#warn "using Quartz/Cocoa keybindings"
 		const bool meta = ctrl;
 		ctrl = (event->state & GDK_META_MASK) != 0;
 #endif
diff --git a/scintilla/src/KeyMap.cxx b/scintilla/src/KeyMap.cxx
index da4f242..c0681a1 100644
--- a/scintilla/src/KeyMap.cxx
+++ b/scintilla/src/KeyMap.cxx
@@ -51,7 +51,8 @@ const std::map<KeyModifiers, Message> &KeyMap::GetKeyMap() const noexcept {
 	return kmap;
 }
 
-#if PLAT_GTK_MACOSX
+#if PLAT_GTK_MACOSX_QUARTZ
+#warn "using Quartz/Cocoa keybindings"
 #define OS_X_KEYS 1
 #else
 #define OS_X_KEYS 0
diff --git a/scintilla/src/Platform.h b/scintilla/src/Platform.h
index b0d1288..c4e78e6 100644
--- a/scintilla/src/Platform.h
+++ b/scintilla/src/Platform.h
@@ -18,6 +18,7 @@
 #define PLAT_GTK 0
 #define PLAT_GTK_WIN32 0
 #define PLAT_GTK_MACOSX 0
+#define PLAT_GTK_MACOSX_QUARTZ 0
 #define PLAT_MACOSX 0
 #define PLAT_WIN 0
 #define PLAT_WX  0
diff --git a/src/keybindings.h b/src/keybindings.h
index 2f2dcf9..7fe10d0 100644
--- a/src/keybindings.h
+++ b/src/keybindings.h
@@ -30,7 +30,7 @@ G_BEGIN_DECLS
  * presence, use together with keybindings_get_modifiers() which adds
  * @c GEANY_PRIMARY_MOD_MASK when needed.
  * @since 1.25. */
-#ifdef __APPLE__
+#ifdef __APPLE__QUARTZ__
 #define GEANY_PRIMARY_MOD_MASK GDK_META_MASK
 #else
 #define GEANY_PRIMARY_MOD_MASK GDK_CONTROL_MASK
