--- qtwebengine/src/core/orig.core_gyp_generator.pro	2015-01-19 19:20:46.000000000 +0100
+++ qtwebengine/src/core/core_gyp_generator.pro	2015-01-19 19:21:20.000000000 +0100
@@ -30,7 +30,8 @@
 RESOURCES += devtools.qrc
 
 # something fishy with qmake in 5.2 ?
-INCLUDEPATH += $$[QT_INSTALL_HEADERS] $$PWD
+#INCLUDEPATH += $$[QT_INSTALL_HEADERS] $$PWD
+INCLUDEPATH += $$PWD
 
 SOURCES = \
         access_token_store_qt.cpp \
