--- qtbase/tests/auto/testlib/selftests/benchlibcallgrind/tst_benchlibcallgrind.cpp
+++ qtbase/tests/auto/testlib/selftests/benchlibcallgrind/tst_benchlibcallgrind.cpp
@@ -73,9 +73,9 @@
     QBENCHMARK {
         __asm__ __volatile__(
             "mov $100000000,%%eax   \n"
-            "LOOPTOP:               \n"
+            "1:                     \n"
             "dec %%eax              \n"
-            "jnz LOOPTOP            \n"
+            "jnz 1b                 \n"
             : /* no output */
             : /* no input */
             : /* clobber */ "eax"
