diff --git qtbase/src/platformsupport/themes/genericunix/qgenericunixthemes.cpp qtbase/src/platformsupport/themes/genericunix/qgenericunixthemes.cpp
index e66fdeb..7a1797f 100644
--- qtbase/src/platformsupport/themes/genericunix/qgenericunixthemes.cpp
+++ qtbase/src/platformsupport/themes/genericunix/qgenericunixthemes.cpp
@@ -616,7 +616,11 @@ QPlatformTheme *QGenericUnixTheme::createUnixTheme(const QString &name)
 #endif
     if (name == QLatin1String(QGnomeTheme::name))
         return new QGnomeTheme;
+#ifdef Q_OS_OSX
+    return NULL;
+#else
     return new QGenericUnixTheme;
+#endif
 }
 
 QStringList QGenericUnixTheme::themeNames()
diff --git qtbase/src/platformsupport/themes/themes.pri qtbase/src/platformsupport/themes/themes.pri
index adee852..797ac77 100644
--- qtbase/src/platformsupport/themes/themes.pri
+++ qtbase/src/platformsupport/themes/themes.pri
@@ -1,3 +1,3 @@
-unix:!mac {
+unix {
     include($$PWD/genericunix/genericunix.pri)
 }
diff --git qtbase/src/plugins/platforms/cocoa/qcocoaintegration.mm qtbase/src/plugins/platforms/cocoa/qcocoaintegration.mm
index 2ef722d..371e488 100644
--- qtbase/src/plugins/platforms/cocoa/qcocoaintegration.mm
+++ qtbase/src/plugins/platforms/cocoa/qcocoaintegration.mm
@@ -61,6 +61,8 @@
 
 #include <IOKit/graphics/IOGraphicsLib.h>
 
+#include <private/qgenericunixthemes_p.h>
+
 static void initResources()
 {
     Q_INIT_RESOURCE(qcocoaresources);
@@ -501,9 +503,20 @@ QStringList QCocoaIntegration::themeNames() const
 
 QPlatformTheme *QCocoaIntegration::createPlatformTheme(const QString &name) const
 {
-    if (name == QLatin1String(QCocoaTheme::name))
+    qDebug() << "QCocoaIntegration::createPlatformTheme(" << name << ")";
+    if (name == QLatin1String(QCocoaTheme::name)) {
+        qDebug() << "returning the QCocoaTheme";
         return new QCocoaTheme;
-    return QPlatformIntegration::createPlatformTheme(name);
+    }
+    QPlatformTheme *unixTheme = QGenericUnixTheme::createUnixTheme(name);
+    if (unixTheme) {
+        qDebug() << "returning" << unixTheme;
+        return unixTheme;
+    }
+    else {
+        qDebug() << "returning QPlatformIntegration::createPlatformTheme(" << name << ")";
+        return QPlatformIntegration::createPlatformTheme(name);
+    }
 }
 
 QCocoaServices *QCocoaIntegration::services() const
