diff --git a/src/BuildInfo.h b/src/BuildInfo.h
index c44e59be40733999e14499b0d54a80e19b0d7ad3..f4581152482d80b02749951e69451ad8eb0b025f 100644
--- a/src/BuildInfo.h
+++ b/src/BuildInfo.h
@@ -19,6 +19,8 @@
 
 // RevisionIdent contains the GIT_DESCRIBE and REV_LONG defines from git commit information
 #include "RevisionIdent.h"
+// GitDesc contains the MacPorts port version
+#include "GitDesc.h"
 
 // This define replaces the original that modified the macro in wxwidgets
 #define CUSTOM_wxMAKE_VERSION_DOT_STRING_T(x, y, z) wxSTRINGIZE_T(x) wxT(".") wxSTRINGIZE_T(y) wxT(".") wxSTRINGIZE_T(z) wxT("(Tenacity)")
@@ -102,7 +104,8 @@ public:
             static wxString NoRevisionText = XO("No revision identifier was provided").Translation();
 
             #ifdef REV_LONG
-                return wxString("[[https://github.com/tenacityteam/tenacity/commit/") + REV_LONG + "|" + wxString(GIT_DESCRIBE) + "]]";
+                return wxString("[[https://github.com/tenacityteam/tenacity/commit/") + REV_LONG + "|" + wxString(GIT_DESCRIBE) + "]]"
+                    + wxT("(") + wxString(GIT_DESC) + wxT(")");
             #else
                 return NoRevisionText;
             #endif
