diff --gitkstyle/demo/oxygenframedemowidget.cppkstyle/demo/oxygenframedemowidget.cpp
index 2df781ec..cdf459e9 100644
---kstyle/demo/oxygenframedemowidget.cpp
+++kstyle/demo/oxygenframedemowidget.cpp
@@ -26,16 +26,28 @@
 
 #include "oxygenframedemowidget.h"
 
+#include <qglobal.h>
 #include <QButtonGroup>
 
 #include <KComboBox>
+#include <KMessageWidget>
+#if QT_VERSION >= QT_VERSION_CHECK(5, 0, 0)
+#    include <kwidgetsaddons_version.h>
+#else
+#    define KWIDGETSADDONS_VERSION QT_VERSION
+#    define QByteArrayLiteral /**/
+#endif
 
 namespace Oxygen
 {
 
     //_____________________________________________________________
     FrameDemoWidget::FrameDemoWidget( QWidget* parent ):
-        DemoWidget( parent )
+        DemoWidget( parent ),
+        posMsg(nullptr),
+        infoMsg(nullptr),
+        warnMsg(nullptr),
+        errMsg(nullptr)
     {
 
         ui.setupUi( this );
@@ -51,6 +63,60 @@ namespace Oxygen
         connect( ui.directionComboBox, SIGNAL(currentIndexChanged(int)), SLOT(updateLayoutDirection(int)) );
         connect( ui.flatGroupBoxCheckBox, SIGNAL(toggled(bool)), SLOT(toggleFlatGroupBox(bool)) );
 
+        addMessages();
+#if KWIDGETSADDONS_VERSION < QT_VERSION_CHECK(5, 48, 0)
+        qApp->installEventFilter(this);
+#endif
+    }
+
+    void FrameDemoWidget::addMessages()
+    {
+        delete posMsg;
+        delete infoMsg;
+        delete warnMsg;
+        delete errMsg;
+
+        posMsg = new KMessageWidget(QStringLiteral("A positive message"), ui.msgFrame);
+        posMsg->setMessageType(KMessageWidget::Positive);
+        posMsg->setWordWrap(true);
+        posMsg->setIcon(QIcon::fromTheme("dialog-positive"));
+        ui.verticalLayout_5->addWidget(posMsg);
+
+        infoMsg = new KMessageWidget(QStringLiteral("An information message"), ui.msgFrame);
+        infoMsg->setMessageType(KMessageWidget::Information);
+        infoMsg->setWordWrap(true);
+        infoMsg->setIcon(QIcon::fromTheme("dialog-information"));
+        ui.verticalLayout_5->addWidget(infoMsg);
+
+        warnMsg = new KMessageWidget(QStringLiteral("A warning message"), ui.msgFrame);
+        warnMsg->setMessageType(KMessageWidget::Warning);
+        warnMsg->setWordWrap(true);
+        warnMsg->setIcon(QIcon::fromTheme("dialog-warning"));
+        ui.verticalLayout_5->addWidget(warnMsg);
+
+        errMsg = new KMessageWidget(QStringLiteral("An error message"), ui.msgFrame);
+        errMsg->setMessageType(KMessageWidget::Error);
+        errMsg->setWordWrap(true);
+        errMsg->setIcon(QIcon::fromTheme("dialog-error"));
+        ui.verticalLayout_5->addWidget(errMsg);
+
+        ui.verticalLayout_5->addStretch();
+    }
+
+    bool FrameDemoWidget::eventFilter( QObject *obj, QEvent *event )
+    {
+#if KWIDGETSADDONS_VERSION < QT_VERSION_CHECK(5, 48, 0)
+        if (event->type() == QEvent::DynamicPropertyChange && obj == qApp) {
+            QDynamicPropertyChangeEvent *e = dynamic_cast<QDynamicPropertyChangeEvent*>(event);
+            if (e->propertyName() == QByteArrayLiteral("KDE_COLOR_SCHEME_PATH")) {
+                addMessages();
+            }
+        }
+#else
+        Q_UNUSED(obj);
+        Q_UNUSED(event);
+#endif
+        return false;
     }
 
     //_____________________________________________________________
diff --git kstyle/demo/oxygenframedemowidget.h kstyle/demo/oxygenframedemowidget.h
index a9459c3b71a0070f1f96037b8959df5097359556..aceae3a79795fd1ce7a1a45a80d64cb2de068fde 100644
--- kstyle/demo/oxygenframedemowidget.h
+++ kstyle/demo/oxygenframedemowidget.h
@@ -34,6 +34,8 @@
 #include "oxygendemowidget.h"
 #include "ui_oxygenframedemowidget.h"
 
+class KMessageWidget;
+
 namespace Oxygen
 {
     class FrameDemoWidget: public DemoWidget
@@ -69,6 +71,10 @@ namespace Oxygen
         //! layout direction
         void updateLayoutDirection( int );
 
+        void addMessages();
+
+        bool eventFilter( QObject *obj, QEvent *event );
+
         public Q_SLOTS:
 
         // benchmarking
@@ -77,6 +83,7 @@ namespace Oxygen
         private:
 
         Ui_FrameDemoWidget ui;
+        KMessageWidget *posMsg, *infoMsg, *warnMsg, *errMsg;
 
     };
 
diff --git kstyle/demo/ui/oxygenframedemowidget.ui kstyle/demo/ui/oxygenframedemowidget.ui
index eda67a7cc08a3671321fce675e7931bc61680ce1..76a6327e2a9abf4a8af2d3499062d42e37402047 100644
--- kstyle/demo/ui/oxygenframedemowidget.ui
+++ kstyle/demo/ui/oxygenframedemowidget.ui
@@ -101,7 +101,7 @@
         <item>
          <widget class="QRadioButton" name="raisedFrameRadioButton">
           <property name="text">
-           <string>Raised</string>
+           <string>&amp;Raised</string>
           </property>
           <property name="checked">
            <bool>true</bool>
@@ -114,7 +114,7 @@
            <bool>true</bool>
           </property>
           <property name="text">
-           <string comment="Flat frame. No frame is actually drawn.">Flat</string>
+           <string comment="Flat frame. No frame is actually drawn.">&amp;Flat</string>
           </property>
          </widget>
         </item>
@@ -185,6 +185,19 @@
      </item>
     </layout>
    </item>
+   <item>
+    <layout class="QHBoxLayout" name="msgLayout" stretch="1">
+     <item>
+      <widget class="QFrame" name="msgFrame">
+       <property name="frameShape">
+        <enum>QFrame::Panel</enum>
+       </property>
+       <layout class="QVBoxLayout" name="verticalLayout_5">
+       </layout>
+      </widget>
+     </item>
+    </layout>
+   </item>
   </layout>
  </widget>
  <customwidgets>
