diff --git kstyle/demo/oxygenframedemowidget.cpp kstyle/demo/oxygenframedemowidget.cpp
index 2df781ecf8d28c5270ac59d1719d3dab8039b71a..5b9217a3154fe98eff74f78d4469d0f5fffd293e 100644
--- kstyle/demo/oxygenframedemowidget.cpp
+++ kstyle/demo/oxygenframedemowidget.cpp
@@ -29,13 +29,18 @@
 #include <QButtonGroup>
 
 #include <KComboBox>
+#include <KMessageWidget>
 
 namespace Oxygen
 {
 
     //_____________________________________________________________
     FrameDemoWidget::FrameDemoWidget( QWidget* parent ):
-        DemoWidget( parent )
+        DemoWidget( parent ),
+        posMsg(nullptr),
+        infoMsg(nullptr),
+        warnMsg(nullptr),
+        errMsg(nullptr)
     {
 
         ui.setupUi( this );
@@ -51,6 +56,44 @@ namespace Oxygen
         connect( ui.directionComboBox, SIGNAL(currentIndexChanged(int)), SLOT(updateLayoutDirection(int)) );
         connect( ui.flatGroupBoxCheckBox, SIGNAL(toggled(bool)), SLOT(toggleFlatGroupBox(bool)) );
 
+        qApp->installEventFilter(this);
+        addMessages();
+    }
+
+    void FrameDemoWidget::addMessages()
+    {
+        delete posMsg;
+        delete infoMsg;
+        delete warnMsg;
+        delete errMsg;
+
+        posMsg = new KMessageWidget(QStringLiteral("A positive message"), ui.tab);
+        posMsg->setMessageType(KMessageWidget::Positive);
+        posMsg->setWordWrap(true);
+        ui.verticalLayout_4->addWidget(posMsg);
+        infoMsg = new KMessageWidget(QStringLiteral("An information message"), ui.tab);
+        infoMsg->setMessageType(KMessageWidget::Information);
+        infoMsg->setWordWrap(true);
+        ui.verticalLayout_4->addWidget(infoMsg);
+        warnMsg = new KMessageWidget(QStringLiteral("A warning message"), ui.tab);
+        warnMsg->setMessageType(KMessageWidget::Warning);
+        warnMsg->setWordWrap(true);
+        ui.verticalLayout_4->addWidget(warnMsg);
+        errMsg = new KMessageWidget(QStringLiteral("An error message"), ui.tab);
+        errMsg->setMessageType(KMessageWidget::Error);
+        errMsg->setWordWrap(true);
+        ui.verticalLayout_4->addWidget(errMsg);
+    }
+
+    bool FrameDemoWidget::eventFilter( QObject *obj, QEvent *event )
+    {
+        if (event->type() == QEvent::DynamicPropertyChange && obj == qApp) {
+            QDynamicPropertyChangeEvent *e = dynamic_cast<QDynamicPropertyChangeEvent*>(event);
+            if (e->propertyName() == QByteArrayLiteral("KDE_COLOR_SCHEME_PATH")) {
+                addMessages();
+            }
+        }
+        return false;
     }
 
     //_____________________________________________________________
diff --git kstyle/demo/oxygenframedemowidget.h kstyle/demo/oxygenframedemowidget.h
index a9459c3b71a0070f1f96037b8959df5097359556..aceae3a79795fd1ce7a1a45a80d64cb2de068fde 100644
--- kstyle/demo/oxygenframedemowidget.h
+++ kstyle/demo/oxygenframedemowidget.h
@@ -34,6 +34,8 @@
 #include "oxygendemowidget.h"
 #include "ui_oxygenframedemowidget.h"
 
+class KMessageWidget;
+
 namespace Oxygen
 {
     class FrameDemoWidget: public DemoWidget
@@ -69,6 +71,10 @@ namespace Oxygen
         //! layout direction
         void updateLayoutDirection( int );
 
+        void addMessages();
+
+        bool eventFilter( QObject *obj, QEvent *event );
+
         public Q_SLOTS:
 
         // benchmarking
@@ -77,6 +83,7 @@ namespace Oxygen
         private:
 
         Ui_FrameDemoWidget ui;
+        KMessageWidget *posMsg, *infoMsg, *warnMsg, *errMsg;
 
     };
 
