diff --git kstyle/demo/main.cpp kstyle/demo/main.cpp
index b75731fa1856011318ee7cdabdf1d3cf8bcc5f83..5f0bebb86d7e516d825be94c9cfc65b0d497c4a5 100644
--- kstyle/demo/main.cpp
+++ kstyle/demo/main.cpp
@@ -28,7 +28,9 @@
 #include "../oxygen.h"
 #include "config-liboxygen.h"
 
-#include <QApplication>
+#include <KCmdLineArgs>
+#include <KApplication>
+#include <KAboutData>
 
 #if QT_VERSION >= 0x050200
 #include <QCommandLineOption>
@@ -44,7 +46,19 @@ namespace Oxygen
 
     int run(int argc, char *argv[])
     {
-        QApplication app( argc, argv );
+        KAboutData aboutData(
+            "oxygen-demo",
+            "kstyle_config",
+            ki18n( "Oxygen Demo" ),
+            "@VERSION@",
+            ki18n( "Oxygen style demonstration" ),
+            KAboutData::License_GPL_V2,
+            ki18n( "(c) 2010, Hugo Pereira Da Costa" ));
+
+        aboutData.addAuthor( ki18n( "Hugo Pereira Da Costa" ),KLocalizedString(), "hugo.pereira@free.fr" );
+
+        KCmdLineArgs::init( argc, argv, &aboutData );
+        KApplication app;
 
         #if QT_VERSION >= 0x050200
         QCommandLineParser commandLine;

