diff --git kstyle/demo/oxygendemodialog.cpp kstyle/demo/oxygendemodialog.cpp
index 4a742e3068a06bfc5fbd7b7c973ea7e5fbfcaaec..5005d5f146f8fbe9053eb4158f6d93e46f0fd200 100644
--- kstyle/demo/oxygendemodialog.cpp
+++ kstyle/demo/oxygendemodialog.cpp
@@ -75,7 +75,9 @@ namespace Oxygen
         buttonBox->addButton( _enableCheckBox, QDialogButtonBox::ResetRole );
 
         _rightToLeftCheckBox = new QCheckBox( i18n( "Right to left layout" ) );
+        _rightToLeftCheckBox->setChecked( qApp->isRightToLeft() );
         connect( _rightToLeftCheckBox, SIGNAL(toggled(bool)), SLOT(toggleRightToLeft(bool)) );
+        connect( qApp, SIGNAL(layoutDirectionChanged(Qt::LayoutDirection)), SLOT(layoutDirectionChanged(Qt::LayoutDirection)) );
         buttonBox->addButton( _rightToLeftCheckBox, QDialogButtonBox::ResetRole );
 
         // connections
@@ -218,6 +220,10 @@ namespace Oxygen
     { qApp->setLayoutDirection( value ? Qt::RightToLeft:Qt::LeftToRight ); }
 
     //_______________________________________________________________
+    void DemoDialog::layoutDirectionChanged( Qt::LayoutDirection direction )
+    { _rightToLeftCheckBox->setChecked( direction == Qt::RightToLeft ); }
+
+    //_______________________________________________________________
     void DemoDialog::closeEvent( QCloseEvent* event )
     {
         emit abortSimulations();
diff --git kstyle/demo/oxygendemodialog.h kstyle/demo/oxygendemodialog.h
index b296ec7170bac18552ad3729564394d591e5175d..9f0015dfe73503ec1b2447253ee72d8c70a010b9 100644
--- kstyle/demo/oxygendemodialog.h
+++ kstyle/demo/oxygendemodialog.h
@@ -75,6 +75,7 @@ namespace Oxygen
 
         //! toggle RightToLeft
         virtual void toggleRightToLeft( bool );
+        virtual void layoutDirectionChanged( Qt::LayoutDirection );
 
         protected:
 
diff --git kstyle/demo/oxygenmdidemowidget.cpp kstyle/demo/oxygenmdidemowidget.cpp
index 3e5422eb1ead7c0a701c500b4131bd8b52463f60..d89f8fe9b5d9c5903c91921cd158869d42ca797f 100644
--- kstyle/demo/oxygenmdidemowidget.cpp
+++ kstyle/demo/oxygenmdidemowidget.cpp
@@ -40,6 +40,8 @@ namespace Oxygen
     {
         setLayout( new QVBoxLayout() );
         QMenuBar* menuBar = new QMenuBar( this );
+        // let the menubar appear in the MDI window on all platforms
+        menuBar->setNativeMenuBar( false );
         layout()->addWidget( menuBar );
 
         QWidget* widget = new QWidget( this );
@@ -47,12 +49,27 @@ namespace Oxygen
         ui.setupUi( widget );
 
         QMenu* menu = menuBar->addMenu( i18n( "Layout" ) );
-        connect( menu->addAction( i18n( "Tile" ) ), SIGNAL(triggered()), this, SLOT(setLayoutTiled()) );
-        connect( menu->addAction( i18n( "Cascade" ) ), SIGNAL(triggered()), this, SLOT(setLayoutCascade()) );
-        connect( menu->addAction( i18n( "Tabs" ) ), SIGNAL(triggered()), this, SLOT(setLayoutTabbed()) );
+        QActionGroup *aGroup = new QActionGroup( menu );
+        menu->addSection( i18n( "Exclusive actions" ) );
+        QAction *action = menu->addAction( i18n( "Tile" ) );
+        action->setCheckable( true );
+        aGroup->addAction( action );
+        connect( action, SIGNAL(triggered()), this, SLOT(setLayoutTiled()) );
+        action = menu->addAction( i18n( "Cascade" ) );
+        action->setCheckable( true );
+        aGroup->addAction( action );
+        connect( action, SIGNAL(triggered()), this, SLOT(setLayoutCascade()) );
+        action = menu->addAction( i18n( "Tabs" ) );
+        action->setCheckable( true );
+        aGroup->addAction( action );
+        connect( action, SIGNAL(triggered()), this, SLOT(setLayoutTabbed()) );
+
+        menu->addSeparator();
+
+        action = menu->addAction( i18n( "<- Check here" ) );
+        action->setCheckable( true );
 
         menu = menuBar->addMenu( i18n( "Tools" ) );
-        QAction* action;
         connect( action = menu->addAction( QIcon::fromTheme( QStringLiteral( "arrow-right" ) ), i18n( "Select Next Window" ) ), SIGNAL(triggered()), ui.mdiArea, SLOT(activateNextSubWindow()) );
         action->setShortcut( Qt::CTRL + Qt::Key_Tab );
         addAction( action );
