diff --git orig.CMakeLists.txt CMakeLists.txt
index b13f78c..3d4fcb5 100755
--- orig.CMakeLists.txt
+++ CMakeLists.txt
@@ -15,19 +15,18 @@ endif(WIN32)
 
 add_definitions(-DFAST_MATH -DCVD_FASTLOG)
 
-if(NOT MSVC)
-  set(CMAKE_C_FLAGS "-O3 -Wall -fomit-frame-pointer -pipe")
-endif(NOT MSVC)
 
 add_subdirectory(libmpcdec)
 add_subdirectory(libmpcpsy)
 add_subdirectory(libmpcenc)
 add_subdirectory(libwavformat)
-add_subdirectory(mpcgain)
+# the mpcgain tool needs the replay_gain library
+# add_subdirectory(mpcgain)
 add_subdirectory(mpcdec)
 add_subdirectory(mpcenc)
 add_subdirectory(mpc2sv8)
 add_subdirectory(mpccut)
-add_subdirectory(mpcchap)
+# the mpcchap utility needs libcuefile:
+# add_subdirectory(mpcchap)
 add_subdirectory(wavcmp)
 add_subdirectory(include)
diff --git libmpcdec/orig.CMakeLists.txt libmpcdec/CMakeLists.txt
index 0bc5b62..c6b715e 100755
--- libmpcdec/orig.CMakeLists.txt
+++ libmpcdec/CMakeLists.txt
@@ -1,7 +1,14 @@
 include_directories(${libmpc_SOURCE_DIR}/include)
 if(SHARED)
   add_library(mpcdec SHARED huffman mpc_decoder mpc_reader streaminfo mpc_bits_reader mpc_demux requant synth_filter ${libmpc_SOURCE_DIR}/common/crc32)
+  # The ABI hasn't changed, so set the shlib version info such that dependents don't require rebuilding/relinking
+  set_target_properties(mpcdec PROPERTIES
+  	VERSION "5.3.0"
+	MACHO_CURRENT_VERSION 6.3.0
+	MACHO_COMPATIBILITY_VERSION 6
+	SOVERSION 5)
 else(SHARED)
   add_library(mpcdec_static STATIC huffman mpc_decoder mpc_reader streaminfo mpc_bits_reader mpc_demux requant synth_filter ${libmpc_SOURCE_DIR}/common/crc32)
 endif(SHARED)
 
+install(TARGETS mpcdec DESTINATION ${CMAKE_INSTALL_PREFIX}/lib)
