diff --git mesonbuild/orig.minstall.py mesonbuild/minstall.py
index 1531be4..3058732 100644
--- mesonbuild/orig.minstall.py
+++ mesonbuild/minstall.py
@@ -721,8 +721,11 @@ class Installer:
             if file_copied:
                 self.did_install_something = True
                 try:
-                    self.fix_rpath(outname, t.rpath_dirs_to_remove, install_rpath, final_path,
-                                   install_name_mappings, verbose=False)
+                    if install_rpath:
+                        self.fix_rpath(outname, t.rpath_dirs_to_remove, install_rpath, final_path,
+                                             install_name_mappings, verbose=False)
+                    else:
+                        print("RPATH changes at install time disabled")
                 except SystemExit as e:
                     if isinstance(e.code, int) and e.code == 0:
                         pass
