diff --git src/CMakeLists.txt src/CMakeLists.txt
index 4a423b0f8271f6315509c35f232a3dd3bfc15037..357f51724fb51ba936321720903f23b1d87df9b6 100644
--- src/CMakeLists.txt
+++ src/CMakeLists.txt
@@ -95,7 +95,10 @@ PUBLIC
   Qt5::Core
   Qt5::Network
 )
-target_include_directories(KF5Attica INTERFACE "$<INSTALL_INTERFACE:${KDE_INSTALL_INCLUDEDIR_KF5}/Attica>")
+target_include_directories(KF5Attica
+INTERFACE
+  "$<INSTALL_INTERFACE:${KDE_INSTALL_INCLUDEDIR_KF5}/Attica;${KDE_INSTALL_INCLUDEDIR_KF5}/Attica/fwd>"
+)
 
 
 ecm_generate_headers(Attica_CamelCase_HEADERS
@@ -152,7 +155,9 @@ set(Attica_HEADERS
     platformdependent_v2.h
 )
 
-install(FILES ${Attica_CamelCase_HEADERS} DESTINATION ${KDE_INSTALL_INCLUDEDIR_KF5}/Attica/Attica COMPONENT Devel)
+install(FILES ${Attica_CamelCase_HEADERS}
+    DESTINATION ${KDE_INSTALL_INCLUDEDIR_KF5}/Attica/fwd/Attica
+    COMPONENT Devel)
 
 install(TARGETS KF5Attica
         EXPORT KF5AtticaTargets
