--- src/orig.CMakeLists.txt	2016-03-30 20:55:08.000000000 +0200
+++ src/CMakeLists.txt	2016-05-03 13:39:09.000000000 +0200
@@ -151,11 +151,16 @@
   PREFIX Attica
   REQUIRED_HEADERS Attica_HEADERS
 )
-install(FILES ${Attica_CamelCase_HEADERS} DESTINATION ${KDE_INSTALL_INCLUDEDIR_KF5}/Attica/Attica COMPONENT Devel)
+# install these headers to a directory that cannot clash with the installation directory for the regular
+# headers (= on case-insensitive filesystems) and that still allows existing code to do #include <Attica/Foo>
+# by adding a search path to the compiler.
+install(FILES ${Attica_CamelCase_HEADERS} DESTINATION ${KDE_INSTALL_INCLUDEDIR_KF5}/Attica/c++/Attica COMPONENT Devel)
 
+# add Attica/c++ as an additional search path, so that camelCase headers can be found as usual.
 install(TARGETS KF5Attica
         EXPORT KF5AtticaTargets
-        ${KF5_INSTALL_TARGETS_DEFAULT_ARGS})
+        ${KF5_INSTALL_TARGETS_DEFAULT_ARGS}
+        INCLUDES ${KDE_INSTALL_INCLUDEDIR_KF5}/Attica/c++)
 
 install(FILES
     atticabasejob.h # TODO: rename to basejob.h, add atticabasejob.h forwarding to basejob.h for compat
@@ -171,7 +176,10 @@
 )
 
 include(ECMGeneratePriFile)
-ecm_generate_pri_file(BASE_NAME Attica LIB_NAME KF5Attica DEPS "core" FILENAME_VAR PRI_FILENAME INCLUDE_INSTALL_DIR ${KDE_INSTALL_INCLUDEDIR_KF5}/Attica)
+# we need to construct an absolute path for the additional include directory ourselves,
+# ecm_generate_pri_file expects only a single directory through INCLUDE_INSTALL_DIR.
+ecm_generate_pri_file(BASE_NAME Attica LIB_NAME KF5Attica DEPS "core" FILENAME_VAR PRI_FILENAME
+        INCLUDE_INSTALL_DIR "${KDE_INSTALL_INCLUDEDIR_KF5}/Attica ${CMAKE_INSTALL_PREFIX}/${KDE_INSTALL_INCLUDEDIR_KF5}/Attica/c++")
 install(FILES ${PRI_FILENAME}
         DESTINATION ${ECM_MKSPECS_INSTALL_DIR})
 
--- src/cmake/orig.libKF5Attica.pc.cmake	2016-03-04 23:06:41.000000000 +0100
+++ src/cmake/libKF5Attica.pc.cmake	2016-05-03 13:07:03.000000000 +0200
@@ -1,11 +1,11 @@
 prefix=${CMAKE_INSTALL_PREFIX}
 exec_prefix=${CMAKE_INSTALL_PREFIX}/bin
 libdir=${LIB_INSTALL_DIR}
-includedir=${INCLUDE_INSTALL_DIR}/KF5Attica
+includedir=${INCLUDE_INSTALL_DIR}/KF5/Attica:${INCLUDE_INSTALL_DIR}/KF5/Attica/c++
 
 Name: lib${ATTICA_LIB_SONAME}
 Description: Qt library to access Open Collaboration Services
 #Requires:
 Version: ${CMAKE_LIBATTICA_VERSION_MAJOR}.${CMAKE_LIBATTICA_VERSION_MINOR}.${CMAKE_LIBATTICA_VERSION_PATCH}
 Libs: -L${LIB_INSTALL_DIR} -l${ATTICA_LIB_SONAME}
-Cflags: -I${INCLUDE_INSTALL_DIR}/KF5Attica
+Cflags: -I${INCLUDE_INSTALL_DIR}/KF5/Attica -I${INCLUDE_INSTALL_DIR}/KF5/Attica/c++
