diff --git a/examples/maildirresource/CMakeLists.txt b/examples/maildirresource/CMakeLists.txt
index e4d113c..7758cec 100644
--- a/examples/maildirresource/CMakeLists.txt
+++ b/examples/maildirresource/CMakeLists.txt
@@ -7,7 +7,7 @@ find_package(KF5 COMPONENTS REQUIRED Mime)
 
 add_library(${PROJECT_NAME} SHARED facade.cpp maildirresource.cpp domainadaptor.cpp)
 qt5_use_modules(${PROJECT_NAME} Core Network)
-target_link_libraries(${PROJECT_NAME} sink maildir KF5::Mime)
+target_link_libraries(${PROJECT_NAME} sink sink-maildir KF5::Mime)
 
 install(TARGETS ${PROJECT_NAME} LIBRARY DESTINATION ${SINK_RESOURCE_PLUGINS_PATH})
 
diff --git a/examples/maildirresource/libmaildir/CMakeLists.txt b/examples/maildirresource/libmaildir/CMakeLists.txt
index e7803f5..871e793 100644
--- a/examples/maildirresource/libmaildir/CMakeLists.txt
+++ b/examples/maildirresource/libmaildir/CMakeLists.txt
@@ -2,7 +2,7 @@
 
 set(maildir_LIB_SRCS keycache.cpp maildir.cpp)
 
-add_library(maildir ${LIBRARY_TYPE} ${maildir_LIB_SRCS})
-qt5_use_modules(maildir Core Network)
-# set_target_properties(maildir PROPERTIES VERSION ${GENERIC_LIB_VERSION} SOVERSION ${GENERIC_LIB_SOVERSION} )
-install(TARGETS maildir ${INSTALL_TARGETS_DEFAULT_ARGS})
+add_library(sink-maildir ${LIBRARY_TYPE} ${maildir_LIB_SRCS})
+qt5_use_modules(sink-maildir Core Network)
+# set_target_properties(sink-maildir PROPERTIES VERSION ${GENERIC_LIB_VERSION} SOVERSION ${GENERIC_LIB_SOVERSION} )
+install(TARGETS sink-maildir ${INSTALL_TARGETS_DEFAULT_ARGS})
