diff --git a/cmake/modules/FindLibgit2.cmake b/cmake/modules/FindLibgit2.cmake
index f33db4a..61de4ba 100644
--- a/cmake/modules/FindLibgit2.cmake
+++ b/cmake/modules/FindLibgit2.cmake
@@ -28,6 +28,7 @@ FIND_LIBRARY(LIBGIT2_LIBRARIES NAMES git2
 
 #Should be replaced by a version check
 include(CheckFunctionExists)
+set(CMAKE_REQUIRED_LIBRARIES ${LIBGIT2_LIBRARIES})
 CHECK_FUNCTION_EXISTS(git_buf_free HAVE_BUF_FREE)
 
 INCLUDE(FindPackageHandleStandardArgs)
