diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index ea97ad5..ef20a0b 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -407,12 +407,12 @@ if(HAVE_PORTAUDIO)
   list(APPEND EXTRA_DEFS -DQTAV_HAVE_PORTAUDIO=1)
   list(APPEND EXTRA_LIBS portaudio)
 endif()
-check_library_exists(pulse pa_threaded_mainloop_new "" HAVE_PULSE)
-if(HAVE_PULSE)
-  list(APPEND SOURCES output/audio/AudioOutputPulse.cpp)
-  list(APPEND EXTRA_DEFS -DQTAV_HAVE_PULSEAUDIO=1)
-  list(APPEND EXTRA_LIBS pulse)
-endif()
+# check_library_exists(pulse pa_threaded_mainloop_new "" HAVE_PULSE)
+# if(HAVE_PULSE)
+#   list(APPEND SOURCES output/audio/AudioOutputPulse.cpp)
+#   list(APPEND EXTRA_DEFS -DQTAV_HAVE_PULSEAUDIO=1)
+#   list(APPEND EXTRA_LIBS pulse)
+# endif()
 check_library_exists(OpenSLES slCreateEngine "" HAVE_OPENSL)
 if(HAVE_OPENSL)
   list(APPEND SOURCES output/audio/AudioOutputOpenSL.cpp)
