diff --git a/CMakeLists.txt b/CMakeLists.txt
index ef7ade0..da57c7d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -20,6 +20,8 @@ option(BUILD_PLAYERS "Build players" ON)
 option(BUILD_TESTS "Build tests" ON)
 option(BUILD_QT5OPENGL "Build with Qt5 OpenGL module" ON)
 
+include(FeatureSummary)
+
 list(APPEND CMAKE_FIND_ROOT_PATH ${QTDIR})
 
 find_package(Qt5Core REQUIRED)
@@ -68,7 +70,7 @@ if(NOT CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
   set(QTAV_INSTALL_HEADERS ${CMAKE_INSTALL_PREFIX}/include)
   set(QTAV_INSTALL_LIBS ${CMAKE_INSTALL_PREFIX}/lib)
   set(QTAV_INSTALL_BINS ${CMAKE_INSTALL_PREFIX}/bin)
-  set(QTAV_INSTALL_QML ${CMAKE_INSTALL_PREFIX}/qml)
+  set(QTAV_INSTALL_QML ${QT_INSTALL_QML})
 else()
   set(CMAKE_INSTALL_PREFIX ${QT_INSTALL_PREFIX} CACHE PATH "default install path" FORCE)
   set(QTAV_INSTALL_HEADERS ${QT_INSTALL_HEADERS})
@@ -123,4 +125,4 @@ if(BUILD_EXAMPLES OR BUILD_PLAYERS)
   add_subdirectory(examples)
 endif()
 
-
+feature_summary(WHAT ALL INCLUDE_QUIET_PACKAGES FATAL_ON_MISSING_REQUIRED_PACKAGES)
diff --git a/examples/CMakeLists.txt b/examples/CMakeLists.txt
index 32b3201..a304dd8 100644
--- a/examples/CMakeLists.txt
+++ b/examples/CMakeLists.txt
@@ -19,7 +19,7 @@ set(COMMON_HEADERS
 if(NOT IOS)
   list(APPEND COMMON_SRC common/ScreenSaver.cpp)
   if(APPLE)
-    list(APPEND COMMON_LIBS "-framework CoreServices")
+    list(APPEND COMMON_LIBS "-framework CoreServices -framework IOKit")
   endif()
 endif()
 
diff --git a/examples/player/MainWindow.cpp b/examples/player/MainWindow.cpp
index fb7cda4..f862790 100644
--- a/examples/player/MainWindow.cpp
+++ b/examples/player/MainWindow.cpp
@@ -112,11 +112,13 @@ MainWindow::MainWindow(QWidget *parent) :
   , m_preview(0)
   , m_shader(NULL)
 {
-    #if defined(Q_OS_MACX) && QT_VERSION >= QT_VERSION_CHECK(5, 0, 0)
-        QApplication::setStyle(QStyleFactory::create("Fusion"));
-    #endif
+//     #if defined(Q_OS_MACX) && QT_VERSION >= QT_VERSION_CHECK(5, 0, 0)
+//         QApplication::setStyle(QStyleFactory::create("Fusion"));
+//     #endif
 
+#ifndef Q_OS_MACOS
     setWindowIcon(QIcon(QString::fromLatin1(":/QtAV.svg")));
+#endif
     mpOSD = new OSDFilter(this);
     mpSubtitle = new SubtitleFilter(this);
     mpChannelAction = 0;
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index ea97ad5..e922ce7 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -486,7 +486,7 @@ target_link_libraries(${MODULE}
 )
 set_target_properties(${MODULE} PROPERTIES
   MACOSX_RPATH ON
-  FRAMEWORK ON
+#   FRAMEWORK ON
     #  MACOSX_FRAMEWORK_INFO_PLIST qtav-Info.plist
     #PUBLIC_HEADER ${SDK_HEADERS}
     #PRIVATE_HEADER ${SDK_PRIVATE_HEADERS}
diff --git a/widgets/CMakeLists.txt b/widgets/CMakeLists.txt
index 0084f8e..926182d 100644
--- a/widgets/CMakeLists.txt
+++ b/widgets/CMakeLists.txt
@@ -56,7 +56,7 @@ if(HAVE_QT5OPENGL)
 endif()
 set_target_properties(${MODULE} PROPERTIES
   MACOSX_RPATH ON
-  FRAMEWORK ON
+#   FRAMEWORK ON
   VERSION ${PROJECT_VERSION}
   SOVERSION ${SO_VERSION}
   OUTPUT_NAME ${MODULE}
