diff --git a/lib-src/FileDialog/mac/FileDialogPrivate.mm b/lib-src/FileDialog/mac/FileDialogPrivate.mm
index 508d8b8002dc18b17c8700f9eb7061c4178e0e6b..fca75e17da856475cb602d8d5040bcffce2fa7d5 100644
--- a/lib-src/FileDialog/mac/FileDialogPrivate.mm
+++ b/lib-src/FileDialog/mac/FileDialogPrivate.mm
@@ -47,7 +47,7 @@
 // implementation
 // ============================================================================
 
-@interface OSPanelDelegate : NSObject wxOSX_10_6_AND_LATER(<NSOpenSavePanelDelegate>)
+@interface OSPanelDelegate : NSObject <NSOpenSavePanelDelegate>
 {
     FileDialog* _dialog;
 }
