--- a/src/effects/VST/VSTControlOSX.mm
+++ b/src/effects/VST/VSTControlOSX.mm
@@ -99,9 +99,13 @@
    {
       CreateCarbon();
    }
+   if (!mHIView)
+   {
+      return false;
+   }
 #endif
 
-   if (!mView && !mHIView)
+   if (!mView)
    {
       return false;
    }
@@ -143,7 +147,7 @@ void VSTControl::CreateCocoa()
    // Get the final bounds of the effect GUI
    mLink->callDispatcher(effEditGetRect, 0, 0, &rect, 0.0);
 
-   NSRect frame = { 0, 0, rect->right - rect->left, rect->bottom - rect->top };
+   NSRect frame = { {0, 0}, {(CGFloat)(rect->right - rect->left), (CGFloat)(rect->bottom - rect->top)} };
 
    [mView setFrame:frame];
 
