diff --git a/src/MemoryX.h b/src/MemoryX.h
index a68eecf83fad5d9776940f75717ec5599f9f4179..f8e49de5717814a744ecdfea71dc5f2cfbd1d0b9 100644
--- a/src/MemoryX.h
+++ b/src/MemoryX.h
@@ -787,7 +787,7 @@ make_value_transform_iterator(const Iterator &iterator, Function function)
 // For using std::unordered_map on wxString
 namespace std
 {
-   template<typename T> struct hash;
+//    template<typename T> struct hash;
    template<> struct hash< wxString > {
       size_t operator () (const wxString &str) const // noexcept
       {
