--- a/src/widgets/NumericTextCtrl.cpp
+++ b/src/widgets/NumericTextCtrl.cpp
@@ -1184,8 +1184,8 @@
 {
    mAllowInvalidValue = false;
 
-   mDigitBoxW = 10;
-   mDigitBoxH = 16;
+   mDigitBoxW = 8;
+   mDigitBoxH = 14;
 
    mReadOnly = false;
    mMenuEnabled = true;
diff --git a/src/widgets/Ruler.cpp b/src/widgets/Ruler.cpp
index 33eb1b4..aebb050 100644
--- a/src/widgets/Ruler.cpp
+++ b/src/widgets/Ruler.cpp
@@ -992,10 +992,10 @@ void Ruler::Update(const TimeTrack* timetrack)// Envelope *speedEnv, long minSpe
       int desiredPixelHeight;
 
 
-      static const int MinPixelHeight = 10; // 8;
+      static const int MinPixelHeight = 7; // 8;
       static const int MaxPixelHeight =
 #ifdef __WXMAC__
-            10
+            9
 #else
             12
 #endif
