commit ae431bc5bd7c7967e72735779ad2796d43f54ad5
Author: James Crook <james.k.crook@gmail.com>
Date:   Fri Apr 5 17:04:54 2019 +0100

    Build LAME on macOS

diff --git b/src/export/ExportMP3.h a/src/export/ExportMP3.h
index 97e87f29e..343788e7b 100644
--- b/src/export/ExportMP3.h
+++ a/src/export/ExportMP3.h
@@ -20,7 +20,7 @@
 #define MODE_ABR           2
 #define MODE_CBR           3
 
-#if defined(__WXMSW__) || defined(__WXMAC__)
+#if defined(__WXMSW__)
 #define MP3_EXPORT_BUILT_IN 1
 #endif
 
