--- qt4/src/orig.CMakeLists.txt	2017-11-12 19:14:32.000000000 +0100
+++ qt4/src/CMakeLists.txt	2018-10-24 20:40:09.000000000 +0200
@@ -30,7 +30,8 @@
 )
 qt4_automoc(${poppler_qt4_SRCS})
 add_library(poppler-qt4 SHARED ${poppler_qt4_SRCS})
-set_target_properties(poppler-qt4 PROPERTIES VERSION 4.11.0 SOVERSION 4)
+# force the same compatibility version as the previous version installed (v0.57.0)
+set_target_properties(poppler-qt4 PROPERTIES VERSION 4.11.0 SOVERSION 16)
 if(MINGW)
     get_target_property(POPPLER_QT4_SOVERSION poppler-qt4 SOVERSION)
     set_target_properties(poppler-qt4 PROPERTIES SUFFIX "-${POPPLER_QT4_SOVERSION}${CMAKE_SHARED_LIBRARY_SUFFIX}")
