commit a308c5d616bf3c9b28d0e52f3952277a49b23969
Author: R.J.V. Bertin <rjvbertin@gmail.com>
Date:   Thu Jan 25 00:08:26 2018 +0100

    Make KDESU_USE_SUDO_DEFAULT mode build again

diff --git src/CMakeLists.txt src/CMakeLists.txt
index d9542c9..02fe3d1 100644
--- src/CMakeLists.txt
+++ src/CMakeLists.txt
@@ -51,7 +51,7 @@ target_link_libraries(KF5Su
 )
 
 if(KDESU_USE_SUDO_DEFAULT)
-   set_source_files_properties(su.cpp PROPERTIES 
+   set_source_files_properties(suprocess.cpp PROPERTIES 
                                COMPILE_FLAGS -DKDESU_USE_SUDO_DEFAULT)
 endif()
 
