diff --git src/xxdiff.pro src/xxdiff.pro
index c8227de..5f2cfca 100644
--- src/xxdiff.pro
+++ src/xxdiff.pro
@@ -29,8 +29,9 @@
 TEMPLATE = app
 CONFIG -= debug
 CONFIG += qt warn_on thread
+QT += widgets help
 
-DESTDIR=../bin
+DESTDIR=./bin
 TARGET = xxdiff
 
 
@@ -110,12 +111,17 @@ irix-n32:QMAKE_CFLAGS_RELEASE += -OPT:Olimit=4000
 ## macx-g++:QMAKE_CXXFLAGS += -D__GNU_LIBRARY__
 ## macx-g++:QMAKE_CXXFLAGS -= -fno-exceptions
 
-
 #----------------------------------------
 # Max OS X (macx-g++ for command line build)
 
 macx {
-   # Icon used to the application bundle
+   QMAKE_CFLAGS -= -O2
+   QMAKE_CXXFLAGS -= -O2
+   QMAKE_CFLAGS += -O3 -march=native -g -flto
+   QMAKE_CXXFLAGS += -O3 -march=native -g -flto -stdlib=libc++
+   QMAKE_LFLAGS += -L/opt/local/lib -Wl,-headerpad_max_install_names -O3 -march=native -g -flto -stdlib=libc++
+
+   # Icon used for the application bundle
    ICON = xxdiff.icns
 
    # Special targets to quickly deploy a standalone mac package (just
@@ -139,8 +145,8 @@ macx {
    deploy.depends = $$dmg.target
 
    QMAKE_EXTRA_TARGETS += macdeployqt dmg deploy
-   QMAKE_CXXFLAGS -= -O2
-   QMAKE_CXXFLAGS += -mdynamic-no-pic -O3 -ftracer -msse2 -msse3 -mssse3 -ftree-vectorize
+   QMAKE_CXXFLAGS += -mdynamic-no-pic
+   INCLUDEPATH += -I/opt/local/include/qt5
 }
 
 #----------------------------------------
@@ -186,6 +192,7 @@ HEADERS = \
 	main.h \
 	overview.h \
 	resParser.h \
+	resParser_lex.h \
 	resources.h \
 	resources.inline.h \
 	accelUtil.h \
