diff -r b193168bad98 src/app.cpp
--- src/app.cpp	Wed Apr 15 14:02:15 2015 -0400
+++ src/app.cpp	Thu May 14 23:35:22 2015 +0200
@@ -243,14 +242,14 @@
    // Read in the resources and create resources object.
    _resources = buildResources();
 
-// We do not force the style anymore.
-// #ifndef XX_KDE
-//    // By default, if not specified, force SGI style.
-//    if ( !_cmdline._forceStyle ) {
-//       _style = QStyleFactory::create( _resources->getStyleKey() );
-//       setStyle( _style );
-//    }
-// #endif
+#ifndef XX_KDE
+   if ( !_cmdline._forceStyle ) {
+      _style = QStyleFactory::create( _resources->getStyleKey() );
+      if (_style) {
+         setStyle( _style );
+      }
+   }
+#endif
 
 #ifndef XX_KDE
    if ( _cmdline._forceFont == false ) {
diff -r b193168bad98 src/resources.cpp
--- src/resources.cpp	Wed Apr 15 14:02:15 2015 -0400
+++ src/resources.cpp	Thu May 14 23:35:22 2015 +0200
@@ -98,9 +98,11 @@
    _preferredGeometry = _defaultGeometry;
    _styleKey =  // Default style.
 #ifdef Q_OS_MAC
-      "Macintosh (aqua)";
+      "Macintosh";
+#elif defined(Q_OS_WIN)
+      "Windows";
 #else
-      "Cleanlooks";
+      "";
 #endif
    _maximize = false;
 
