From 9d1a323fb307669e5d9585c797a46a3b2ebcf98d Mon Sep 17 00:00:00 2001
From: Vitaly Sverchinsky <sverchinsky@gmail.com>
Date: Fri, 4 Feb 2022 12:46:16 +0300
Subject: [PATCH] disable validator

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3439db2..6371a65 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -99,7 +99,7 @@

 # Add hosting examples, it includes the validator (must be done before any plug-ins to support running the validator after building)
 set(SDK_IDE_HOSTING_EXAMPLES_FOLDER FOLDER "HostingExamples")
-add_subdirectory(public.sdk/samples/vst-hosting)
+#add_subdirectory(public.sdk/samples/vst-hosting)

 # Add utilities
 set(SDK_IDE_UTILITIES_FOLDER FOLDER "Utilities")

