--- a/public.sdk/source/vst/moduleinfo/moduleinfoparser.cpp	2023-05-16 21:44:14.418845480 +0300
+++ b/public.sdk/source/vst/moduleinfo/moduleinfoparser.cpp	2023-05-16 21:44:43.868939740 +0300
@@ -39,6 +39,7 @@
 #include "jsoncxx.h"
 #include "pluginterfaces/base/ipluginbase.h"
 #include <stdexcept>
+#include <limits>

 //------------------------------------------------------------------------
 namespace Steinberg::ModuleInfoLib {
