diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2b63d52..18fbbc6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -70,7 +70,6 @@ include_directories(${ROOT} ${SDK_ROOT})
 #-------------------------------------------------------------------------------
 setupCoreAudioSupport()
 setupAaxSupport()
-setupJackSupport()
 
 #-------------------------------------------------------------------------------
 # Projects
