diff --git a/qttools/src/assistant/assistant.pro b/qttools/src/assistant/assistant.pro
index 34590d6..7d64e96 100644
--- a/qttools/src/assistant/assistant.pro
+++ b/qttools/src/assistant/assistant.pro
@@ -2,19 +2,16 @@ TEMPLATE = subdirs
 
 SUBDIRS += clucene \
            help \
-           assistant \
            qhelpgenerator \
            qcollectiongenerator \
            qhelpconverter
 
 help.depends = clucene
-assistant.depends = help
 qhelpgenerator.depends = help
 qcollectiongenerator.depends = help
 qhelpconverter.depends = help
 
 qtNomakeTools( \
-    assistant \
     qhelpgenerator \
     qcollectiongenerator \
     qhelpconverter \
