diff --git a/qtbase/mkspecs/common/clang.conf b/qtbase/mkspecs/common/clang.conf
index e003b94..576d1dd 100644
--- a/qtbase/mkspecs/common/clang.conf
+++ b/qtbase/mkspecs/common/clang.conf
@@ -39,6 +39,6 @@ QMAKE_LFLAGS_CXX1Z      =
 QMAKE_LFLAGS_LTCG       = $$QMAKE_CFLAGS_LTCG
 
 # Wrapper tools that understand .o/.a files with LLVM bytecode instead of machine code
-QMAKE_AR_LTCG           = llvm-ar cqs
-QMAKE_NM_LTCG           = llvm-nm -P
-QMAKE_RANLIB_LTCG       = true      # No need to run, since llvm-ar has "s"
+QMAKE_AR_LTCG           = @LLVM_AR@ cqs
+QMAKE_NM_LTCG           = @LLVM_NM@ -P
+QMAKE_RANLIB_LTCG       = @LLVM_RANLIB@
