diff --git a/qtbase/src/platformsupport/fontdatabases/fontdatabases.pri b/qtbase/src/platformsupport/fontdatabases/fontdatabases.pri
index 12b06d6..b786900 100644
--- a/qtbase/src/platformsupport/fontdatabases/fontdatabases.pri
+++ b/qtbase/src/platformsupport/fontdatabases/fontdatabases.pri
@@ -1,4 +1,4 @@
-!win32|contains(QT_CONFIG, freetype):!mac {
+!win32|contains(QT_CONFIG, freetype) {
     include($$PWD/basic/basic.pri)
 }
 
@@ -12,5 +12,8 @@ unix:!mac {
 
 mac {
     include($$PWD/mac/coretext.pri)
+    contains(QT_CONFIG,fontconfig) {
+        include($$PWD/fontconfig/fontconfig.pri)
+    }
 }
 
