diff --git a/qtbase/src/gui/image/qpixmap.cpp b/qtbase/src/gui/image/qpixmap.cpp
index 0aa05a0..170e86e 100644
--- a/qtbase/src/gui/image/qpixmap.cpp
+++ b/qtbase/src/gui/image/qpixmap.cpp
@@ -1148,7 +1148,7 @@ bool QPixmap::convertFromImage(const QImage &image, Qt::ImageConversionFlags fla
 QPixmap QPixmap::scaled(const QSize& s, Qt::AspectRatioMode aspectMode, Qt::TransformationMode mode) const
 {
     if (isNull()) {
-        qWarning("QPixmap::scaled: Pixmap is a null pixmap");
+//         qWarning("QPixmap::scaled: Pixmap is a null pixmap");
         return QPixmap();
     }
     if (s.isEmpty())
@@ -1184,7 +1184,7 @@ QPixmap QPixmap::scaled(const QSize& s, Qt::AspectRatioMode aspectMode, Qt::Tran
 QPixmap QPixmap::scaledToWidth(int w, Qt::TransformationMode mode) const
 {
     if (isNull()) {
-        qWarning("QPixmap::scaleWidth: Pixmap is a null pixmap");
+//         qWarning("QPixmap::scaleWidth: Pixmap is a null pixmap");
         return copy();
     }
     if (w <= 0)
@@ -1212,7 +1212,7 @@ QPixmap QPixmap::scaledToWidth(int w, Qt::TransformationMode mode) const
 QPixmap QPixmap::scaledToHeight(int h, Qt::TransformationMode mode) const
 {
     if (isNull()) {
-        qWarning("QPixmap::scaleHeight: Pixmap is a null pixmap");
+//         qWarning("QPixmap::scaleHeight: Pixmap is a null pixmap");
         return copy();
     }
     if (h <= 0)
