diff --git a/qttools/src/qtpaths/qtpaths.cpp b/qttools/src/qtpaths/qtpaths.cpp
index 37bdcf4..01e2d17 100644
--- a/qttools/src/qtpaths/qtpaths.cpp
+++ b/qttools/src/qtpaths/qtpaths.cpp
@@ -78,29 +78,32 @@ Q_NORETURN static void error(const QString &message)
 }
 
 
-/*
- * NOTE: that DataLocation and CacheLocation are missing as
- * they don't really make sense for a utility like this because
- * they include the application name.
- */
 static const struct StringEnum {
     const char *stringvalue;
     QStandardPaths::StandardLocation enumvalue;
 } lookupTableData[] = {
-    { "ApplicationsLocation", QStandardPaths::ApplicationsLocation },
-    { "DesktopLocation", QStandardPaths::DesktopLocation },
-    { "DocumentsLocation", QStandardPaths::DocumentsLocation },
-    { "FontsLocation", QStandardPaths::FontsLocation },
-    { "MusicLocation", QStandardPaths::MusicLocation },
-    { "MoviesLocation", QStandardPaths::MoviesLocation },
-    { "PicturesLocation", QStandardPaths::PicturesLocation },
-    { "HomeLocation", QStandardPaths::HomeLocation },
-    { "GenericCacheLocation", QStandardPaths::GenericCacheLocation },
-    { "GenericDataLocation", QStandardPaths::GenericDataLocation },
-    { "RuntimeLocation", QStandardPaths::RuntimeLocation },
-    { "ConfigLocation", QStandardPaths::ConfigLocation },
-    { "GenericConfigLocation", QStandardPaths::GenericConfigLocation },
-    { "DownloadLocation", QStandardPaths::DownloadLocation }
+    { "DesktopLocation", QStandardPaths::DesktopLocation, },
+    { "DocumentsLocation", QStandardPaths::DocumentsLocation, },
+    { "FontsLocation", QStandardPaths::FontsLocation, },
+    { "ApplicationsLocation", QStandardPaths::ApplicationsLocation, },
+    { "MusicLocation", QStandardPaths::MusicLocation, },
+    { "MoviesLocation", QStandardPaths::MoviesLocation, },
+    { "PicturesLocation", QStandardPaths::PicturesLocation, },
+    { "TempLocation", QStandardPaths::TempLocation, },
+    { "HomeLocation", QStandardPaths::HomeLocation, },
+    { "DataLocation", QStandardPaths::DataLocation, },
+    { "CacheLocation", QStandardPaths::CacheLocation, },
+    { "GenericDataLocation", QStandardPaths::GenericDataLocation, },
+    { "RuntimeLocation", QStandardPaths::RuntimeLocation, },
+    { "ConfigLocation", QStandardPaths::ConfigLocation, },
+    { "DownloadLocation", QStandardPaths::DownloadLocation, },
+    { "GenericCacheLocation", QStandardPaths::GenericCacheLocation, },
+    { "GenericConfigLocation", QStandardPaths::GenericConfigLocation, },
+#if defined(Q_OS_MACOS)
+    { "AppDataLocation", QStandardPaths::AppDataLocation, },
+    { "AppConfigLocation", QStandardPaths::AppConfigLocation, },
+    { "AppLocalDataLocation", QStandardPaths::AppLocalDataLocation }
+#endif
 };
 
 /**
