diff --git gcc/collect2.c gcc/collect2.c
index ddbd2be8bf8945d026341ef1d660d591cced36eb..32c1e4d1c92760c90840e69e6d9c58bd5ebb6e02 100644
--- gcc/collect2.c
+++ gcc/collect2.c
@@ -3004,7 +3004,11 @@ process_args (int *argcp, char **argv) {
   int argc = *argcp;
   for (i=0; i<argc; ++i)
     {
-      if (strcmp (argv[i], "-dsym") == 0)
+      if (strcmp (argv[i], "-v") == 0)
+	{
+	  verbose = true;
+	}
+      else if (strcmp (argv[i], "-dsym") == 0)
 	{
 	  flag_dsym = true;
 	  /* Remove the flag, as we handle all processing for it.  */
@@ -3036,10 +3040,13 @@ do_dsymutil (const char *output_file) {
   char **real_argv = XCNEWVEC (char *, 3);
   const char ** argv = CONST_CAST2 (const char **, char **,
 				    real_argv);
+  int arg = 0;
 
-  argv[0] = dsymutil;
-  argv[1] = output_file;
-  argv[2] = (char *) 0;
+  argv[arg++] = dsymutil;
+  if (verbose)
+    argv[arg++] = "--verbose";
+  argv[arg++] = output_file;
+  argv[arg++] = (char *) 0;
 
   pex = collect_execute (dsymutil, real_argv, NULL, NULL,
 			 PEX_LAST | PEX_SEARCH, false);
