diff --git a/coders/orig.dng.c b/coders/dng.c
index 9a8cc62..f8d6af5 100644
--- a/coders/orig.dng.c
+++ b/coders/dng.c
@@ -178,8 +178,10 @@ static void SetDNGProperties(Image *image,const libraw_data_t *raw_info,
   if (*raw_info->shootinginfo.BodySerial != '\0')
     (void) SetImageProperty(image,"dng:serial.number",
       raw_info->shootinginfo.BodySerial,exception);
+#if LIBRAW_MINOR_VERSION < 20
   (void) FormatLocaleString(property,MagickPathExtent,"%0.2f",
     raw_info->other.FlashEC);
+#endif
   (void) SetImageProperty(image,"dng:flash.exposure.compensation",property,
     exception);
   (void) FormatLocaleString(property,MagickPathExtent,"1/%0.1f",
diff --git a/coders/orig.svg.c b/coders/svg.c
index 39f5a1d..c330fad 100644
--- a/coders/orig.svg.c
+++ b/coders/svg.c
@@ -3906,7 +3906,7 @@ static MagickBooleanType TraceSVGImage(Image *image,ExceptionInfo *exception)
 {
 #if defined(MAGICKCORE_AUTOTRACE_DELEGATE)
   {
-    at_bitmap_type
+    at_bitmap
       *trace;
 
     at_fitting_opts_type
@@ -3937,6 +3937,19 @@ static MagickBooleanType TraceSVGImage(Image *image,ExceptionInfo *exception)
     /*
       Trace image and write as SVG.
     */
+    if (!at_input_init()) {
+      (void) LogMagickEvent(TraceEvent,GetMagickModule(),"%s","Autotrace `at_input_init()` failed");
+      return MagickFalse;
+    }
+    if (!at_output_init()) {
+      (void) LogMagickEvent(TraceEvent,GetMagickModule(),"%s","Autotrace `at_output_init()` failed");
+      return MagickFalse;
+    }
+    at_module_init();
+    at_spline_writer *writer = at_output_get_handler_by_suffix((char *) "svg");
+    if (!writer) {
+      (void) LogMagickEvent(TraceEvent,GetMagickModule(),"%s","Autotrace has no SVG output module built in");
+    }
     fitting_options=at_fitting_opts_new();
     output_options=at_output_opts_new();
     (void) SetImageGray(image,exception);
@@ -3964,7 +3977,7 @@ static MagickBooleanType TraceSVGImage(Image *image,ExceptionInfo *exception)
     }
     splines=at_splines_new_full(trace,fitting_options,NULL,NULL,NULL,NULL,NULL,
       NULL);
-    at_splines_write(at_output_get_handler_by_suffix((char *) "svg"),
+    at_splines_write(writer,
       GetBlobFileHandle(image),image->filename,output_options,splines,NULL,
       NULL);
     /*
