diff --git qmakefilevisitor.cpp qmakefilevisitor.cpp
index a90fd57..028221d 100644
--- qmakefilevisitor.cpp
+++ qmakefilevisitor.cpp
@@ -87,7 +87,7 @@ void QMakeFileVisitor::visitFunctionCall( QMake::FunctionCallAST* node )
         if( node->args.isEmpty() )
             return;
         QStringList arguments = getValueList( node->args );
-        Q_ASSERT(!arguments.isEmpty());
+//         Q_ASSERT(!arguments.isEmpty());
 
         ifDebug(kDebug(9024) << "found include" << node->identifier->value << arguments;)
         QString argument = arguments.join("").trimmed();
@@ -95,7 +95,7 @@ void QMakeFileVisitor::visitFunctionCall( QMake::FunctionCallAST* node )
         {
             argument = QFileInfo( m_baseFile->absoluteDir() + '/' + argument ).canonicalFilePath();
         }
-        if (argument.isEmpty()) {
+        if (arguments.isEmpty() || argument.isEmpty()) {
             kWarning() << "empty include file detected in line" << node->startLine;
             if( node->identifier->value.startsWith('!') ) {
                 visitNode( node->body );
