--- src/kcompletionbox.cpp.orig	2018-06-17 12:47:18.632373292 +0200
+++ src/kcompletionbox.cpp	2018-06-17 12:50:11.522373292 +0200
@@ -71,7 +71,7 @@
     if (qGuiApp->platformName() == QLatin1String("wayland"))
         q->setWindowFlags(Qt::ToolTip | Qt::FramelessWindowHint | Qt::BypassWindowManagerHint);
     else
-        q->setWindowFlags(Qt::Window | Qt::FramelessWindowHint | Qt::BypassWindowManagerHint);
+        q->setWindowFlags(Qt::Dialog | Qt::FramelessWindowHint | Qt::BypassWindowManagerHint);
     q->setUniformItemSizes(true);
 
     q->setLineWidth(1);
