Index: src/kcompletionbox.cpp
===================================================================
--- src/kcompletionbox.cpp
+++ src/kcompletionbox.cpp
@@ -65,7 +65,7 @@
     //also, q->setAttribute(Qt::WA_X11NetWmWindowTypeCombo); is broken in Qt xcb
     q->setProperty("_q_xcb_wm_window_type", 0x001000);
     q->setAttribute(Qt::WA_ShowWithoutActivating);
-    q->setWindowFlags(Qt::Window | Qt::FramelessWindowHint | Qt::BypassWindowManagerHint); // calls setVisible, so must be done after initializations
+    q->setWindowFlags(Qt::Dialog | Qt::FramelessWindowHint | Qt::BypassWindowManagerHint); // calls setVisible, so must be done after initializations
     q->setUniformItemSizes(true);
 
     q->setLineWidth(1);
