diff --git a/qtbase/src/platformsupport/fontdatabases/mac/qcoretextfontdatabase.mm b/qtbase/src/platformsupport/fontdatabases/mac/qcoretextfontdatabase.mm
index 0e16570..ee951f0 100644
--- a/qtbase/src/platformsupport/fontdatabases/mac/qcoretextfontdatabase.mm
+++ b/qtbase/src/platformsupport/fontdatabases/mac/qcoretextfontdatabase.mm
@@ -247,7 +247,7 @@ void QCoreTextFontDatabase::populateFamily(const QString &familyName)
     // A single family might match several different fonts with different styles eg.
     QCFType<CFArrayRef> matchingFonts = (CFArrayRef) CTFontDescriptorCreateMatchingFontDescriptors(nameOnlyDescriptor, 0);
     if (!matchingFonts) {
-        qWarning() << "QCoreTextFontDatabase: Found no matching fonts for family" << familyName;
+        qDebug() << "QCoreTextFontDatabase: Found no matching fonts for family" << familyName;
         return;
     }
 
@@ -508,8 +508,10 @@ static QString familyNameFromPostScriptName(NSString *psName)
     QCFType<CTFontDescriptorRef> fontDescriptor = (CTFontDescriptorRef) CTFontDescriptorCreateWithNameAndSize((CFStringRef)psName, 12.0);
     QCFString familyName = (CFStringRef) CTFontDescriptorCopyAttribute(fontDescriptor, kCTFontFamilyNameAttribute);
     QString name = QString::fromCFString(familyName);
+#ifdef QT_DEBUG
     if (name.isEmpty())
         qWarning() << "QCoreTextFontDatabase: Failed to resolve family name for PostScript name " << QString::fromCFString((CFStringRef)psName);
+#endif
 
     return name;
 }
