diff --git a/qtbase/src/printsupport/dialogs/qprintdialog.qrc b/qtbase/src/printsupport/dialogs/qprintdialog.qrc
index 5a579baa55840ee65363dc50aad054bc073ec2a6..017c90c16a6c8576e9442a8db9439edcbb92c98c 100644
--- a/qtbase/src/printsupport/dialogs/qprintdialog.qrc
+++ b/qtbase/src/printsupport/dialogs/qprintdialog.qrc
@@ -20,6 +20,8 @@
 <file>images/page-setup-32.png</file>
 <file>images/print-24.png</file>
 <file>images/print-32.png</file>
+<file>images/save-as-pdf-24.png</file>
+<file>images/save-as-pdf-32.png</file>
 <file>images/view-page-multi-24.png</file>
 <file>images/view-page-multi-32.png</file>
 <file>images/view-page-one-24.png</file>
diff --git a/qtbase/src/printsupport/dialogs/qprintpreviewdialog.cpp b/qtbase/src/printsupport/dialogs/qprintpreviewdialog.cpp
index dbcbaf3ce5c5cc51e35fbc6cea2de6420622520a..0b1270176a098a7da61937c92b5f269ebd23a587 100644
--- a/qtbase/src/printsupport/dialogs/qprintpreviewdialog.cpp
+++ b/qtbase/src/printsupport/dialogs/qprintpreviewdialog.cpp
@@ -163,6 +163,7 @@ public:
     void _q_setMode(QAction *action);
     void _q_pageNumEdited();
     void _q_print();
+    void _q_saveAsPdf();
     void _q_pageSetup();
     void _q_previewChanged();
     void _q_zoomFactorChanged();
@@ -214,6 +215,7 @@ public:
 
     QActionGroup *printerGroup;
     QAction *printAction;
+    QAction *pdfAction;
     QAction *pageSetupAction;
 
     QPointer<QObject> receiverToDisconnectOnClose;
@@ -309,6 +311,7 @@ void QPrintPreviewDialogPrivate::init(QPrinter *_printer)
     toolbar->addAction(overviewModeAction);
     toolbar->addSeparator();
     toolbar->addAction(pageSetupAction);
+    toolbar->addAction(pdfAction);
     toolbar->addAction(printAction);
 
     // Cannot use the actions' triggered signal here, since it doesn't autorepeat
@@ -424,10 +427,13 @@ void QPrintPreviewDialogPrivate::setupActions()
     // Print
     printerGroup = new QActionGroup(q);
     printAction = printerGroup->addAction(QCoreApplication::translate("QPrintPreviewDialog", "Print"));
+    pdfAction = printerGroup->addAction(QCoreApplication::translate("QPrintPreviewDialog", "Export to PDF"));
     pageSetupAction = printerGroup->addAction(QCoreApplication::translate("QPrintPreviewDialog", "Page setup"));
     qt_setupActionIcon(printAction, QLatin1String("print"));
+    qt_setupActionIcon(pdfAction, QLatin1String("save-as-pdf"));
     qt_setupActionIcon(pageSetupAction, QLatin1String("page-setup"));
     QObject::connect(printAction, SIGNAL(triggered(bool)), q, SLOT(_q_print()));
+    QObject::connect(pdfAction, SIGNAL(triggered(bool)), q, SLOT(_q_saveAsPdf()));
     QObject::connect(pageSetupAction, SIGNAL(triggered(bool)), q, SLOT(_q_pageSetup()));
 
     // Initial state:
@@ -571,18 +577,7 @@ void QPrintPreviewDialogPrivate::_q_print()
 
 #if defined(Q_OS_WIN) || defined(Q_OS_MAC)
     if (printer->outputFormat() != QPrinter::NativeFormat) {
-        QString title = QCoreApplication::translate("QPrintPreviewDialog", "Export to PDF");
-        QString suffix = QLatin1String(".pdf");
-        QString fileName = QFileDialog::getSaveFileName(q, title, printer->outputFileName(),
-                                                        QLatin1Char('*') + suffix);
-        if (!fileName.isEmpty()) {
-            if (QFileInfo(fileName).suffix().isEmpty())
-                fileName.append(suffix);
-            printer->setOutputFileName(fileName);
-        }
-        if (!printer->outputFileName().isEmpty())
-            preview->print();
-        q->accept();
+        _q_saveAsPdf();
         return;
     }
 #endif
@@ -595,6 +590,25 @@ void QPrintPreviewDialogPrivate::_q_print()
     }
 }
 
+void QPrintPreviewDialogPrivate::_q_saveAsPdf()
+{
+    Q_Q(QPrintPreviewDialog);
+
+    QString title = QCoreApplication::translate("QPrintPreviewDialog", "Export to PDF");
+    QString suffix = QLatin1String(".pdf");
+    QString fileName = QFileDialog::getSaveFileName(q, title, printer->outputFileName(),
+                                                    QLatin1Char('*') + suffix);
+    if (!fileName.isEmpty()) {
+        if (QFileInfo(fileName).suffix().isEmpty())
+            fileName.append(suffix);
+        printer->setOutputFileName(fileName);
+    }
+    if (!printer->outputFileName().isEmpty())
+        preview->print();
+    q->accept();
+    return;
+}
+
 void QPrintPreviewDialogPrivate::_q_pageSetup()
 {
     Q_Q(QPrintPreviewDialog);
diff --git a/qtbase/src/printsupport/dialogs/qprintpreviewdialog.h b/qtbase/src/printsupport/dialogs/qprintpreviewdialog.h
index 3b8f8a1171138c1bd5b07199db92feb60e8ef036..5d24e66ba1dbaa67dce15b0fcc8b2c66e70dbfb3 100644
--- a/qtbase/src/printsupport/dialogs/qprintpreviewdialog.h
+++ b/qtbase/src/printsupport/dialogs/qprintpreviewdialog.h
@@ -82,6 +82,7 @@ private:
     Q_PRIVATE_SLOT(d_func(), void _q_setMode(QAction *action))
     Q_PRIVATE_SLOT(d_func(), void _q_pageNumEdited())
     Q_PRIVATE_SLOT(d_func(), void _q_print())
+    Q_PRIVATE_SLOT(d_func(), void _q_saveAsPdf())
     Q_PRIVATE_SLOT(d_func(), void _q_pageSetup())
     Q_PRIVATE_SLOT(d_func(), void _q_previewChanged())
     Q_PRIVATE_SLOT(d_func(), void _q_zoomFactorChanged())
