diff --git a/qtmultimedia/src/plugins/avfoundation/camera/avfcamerautility.mm b/qtmultimedia/src/plugins/avfoundation/camera/avfcamerautility.mm
index 872075e..a653c39 100644
--- a/qtmultimedia/src/plugins/avfoundation/camera/avfcamerautility.mm
+++ b/qtmultimedia/src/plugins/avfoundation/camera/avfcamerautility.mm
@@ -49,6 +49,13 @@
 #include <algorithm>
 #include <limits>
 
+// see https://trac.macports.org/ticket/52922
+// see https://trac.macports.org/ticket/53949
+// from the 10.9 SDK
+@interface NSArray (MacPortsCompatibilityFirstObject)
+- (id)firstObject NS_AVAILABLE(10_6, 4_0);
+@end
+
 QT_BEGIN_NAMESPACE
 
 AVFPSRange qt_connection_framerates(AVCaptureConnection *videoConnection)
