diff --git a/qtbase/mkspecs/common/macx.conf b/qtbase/mkspecs/common/macx.conf
index 4be0eb3..755a886 100644
--- a/qtbase/mkspecs/common/macx.conf
+++ b/qtbase/mkspecs/common/macx.conf
@@ -11,4 +11,9 @@ device.dir_affix = $${device.sdk}
 device.CONFIG = $${device.sdk}
 device.deployment_identifier = $${device.sdk}
 
+# X11 is not inexistant on OS X; provide the library requirements but leave it to the
+# user to make them accessible during the Qt build.
+QMAKE_LIBS_X11          = -lXext -lX11
+QMAKE_LIBS_OPENVG       = -lOpenVG
+
 include(mac.conf)
diff --git a/qtbase/src/platformsupport/linuxaccessibility/linuxaccessibility.pro b/qtbase/src/platformsupport/linuxaccessibility/linuxaccessibility.pro
index d6689c0..820a944 100644
--- a/qtbase/src/platformsupport/linuxaccessibility/linuxaccessibility.pro
+++ b/qtbase/src/platformsupport/linuxaccessibility/linuxaccessibility.pro
@@ -15,6 +15,10 @@ QDBUSXML2CPP_INTERFACE_HEADER_FLAGS = -i struct_marshallers_p.h
 
 QMAKE_USE += atspi/nolink
 
+darwin: {
+    INCLUDEPATH += $$PWD/../../gui/kernel
+}
+
 HEADERS += \
     application_p.h \
     bridge_p.h \
