diff --git a/qtbase/src/corelib/kernel/qobject.cpp b/qtbase/src/corelib/kernel/qobject.cpp
index 3f50716..901815c 100644
--- a/qtbase/src/corelib/kernel/qobject.cpp
+++ b/qtbase/src/corelib/kernel/qobject.cpp
@@ -3989,7 +3989,7 @@ static void dumpRecursive(int level, const QObject *object)
             }
         }
 #endif
-        qDebug("%s%s::%s %s", (const char*)buf, object->metaObject()->className(), name.toLocal8Bit().data(),
+        qInfo("%s%s::%s %s", (const char*)buf, object->metaObject()->className(), name.toLocal8Bit().data(),
                flags.toLatin1().data());
         QObjectList children = object->children();
         if (!children.isEmpty()) {
@@ -4052,37 +4052,37 @@ void QObject::dumpObjectInfo()
 
 void QObject::dumpObjectInfo() const
 {
-    qDebug("OBJECT %s::%s", metaObject()->className(),
+    qInfo("OBJECT %s::%s", metaObject()->className(),
            objectName().isEmpty() ? "unnamed" : objectName().toLocal8Bit().data());
 
     Q_D(const QObject);
     QMutexLocker locker(signalSlotLock(this));
 
     // first, look for connections where this object is the sender
-    qDebug("  SIGNALS OUT");
+    qInfo("  SIGNALS OUT");
 
     if (d->connectionLists) {
         for (int signal_index = 0; signal_index < d->connectionLists->count(); ++signal_index) {
             const QMetaMethod signal = QMetaObjectPrivate::signal(metaObject(), signal_index);
-            qDebug("        signal: %s", signal.methodSignature().constData());
+            qInfo("        signal: %s", signal.methodSignature().constData());
 
             // receivers
             const QObjectPrivate::Connection *c =
                 d->connectionLists->at(signal_index).first;
             while (c) {
                 if (!c->receiver) {
-                    qDebug("          <Disconnected receiver>");
+                    qInfo("          <Disconnected receiver>");
                     c = c->nextConnectionList;
                     continue;
                 }
                 if (c->isSlotObject) {
-                    qDebug("          <functor or function pointer>");
+                    qInfo("          <functor or function pointer>");
                     c = c->nextConnectionList;
                     continue;
                 }
                 const QMetaObject *receiverMetaObject = c->receiver->metaObject();
                 const QMetaMethod method = receiverMetaObject->method(c->method());
-                qDebug("          --> %s::%s %s",
+                qInfo("          --> %s::%s %s",
                        receiverMetaObject->className(),
                        c->receiver->objectName().isEmpty() ? "unnamed" : qPrintable(c->receiver->objectName()),
                        method.methodSignature().constData());
@@ -4090,11 +4090,11 @@ void QObject::dumpObjectInfo() const
             }
         }
     } else {
-        qDebug( "        <None>" );
+        qInfo( "        <None>" );
     }
 
     // now look for connections where this object is the receiver
-    qDebug("  SIGNALS IN");
+    qInfo("  SIGNALS IN");
 
     if (d->senders) {
         for (QObjectPrivate::Connection *s = d->senders; s; s = s->next) {
@@ -4103,13 +4103,13 @@ void QObject::dumpObjectInfo() const
                 const QMetaMethod slot = metaObject()->method(s->method());
                 slotName = slot.methodSignature();
             }
-            qDebug("          <-- %s::%s %s",
+            qInfo("          <-- %s::%s %s",
                    s->sender->metaObject()->className(),
                    s->sender->objectName().isEmpty() ? "unnamed" : qPrintable(s->sender->objectName()),
                    slotName.constData());
         }
     } else {
-        qDebug("        <None>");
+        qInfo("        <None>");
     }
 }
 
