--- qt5kdweb-mp9-work/qtwebengine/src/3rdparty/chromium/mojo/public/tools/bindings/generators/mojom_js_generator.py.orig	2017-01-03 10:28:53.000000000 +0100
+++ qt5kdweb-mp9-work/qtwebengine/src/3rdparty/chromium/mojo/public/tools/bindings/generators/mojom_js_generator.py	2017-01-03 10:28:53.000000000 +0100
@@ -393,8 +393,6 @@
     return self.GetParameters()
 
   def GenerateFiles(self, args):
-    if self.variant:
-      raise Exception("Variants not supported in JavaScript bindings.")
 
     self.Write(self.GenerateAMDModule(),
         self.MatchMojomFilePath("%s.js" % self.module.name))
--- qt-everywhere-opensource-src-5.8.0/qtwebengine/src/3rdparty/chromium/mojo/public/tools/bindings/generators/mojom_java_generator.py.orig	2017-01-03 10:28:53.000000000 +0100
+++ qt-everywhere-opensource-src-5.8.0/qtwebengine/src/3rdparty/chromium/mojo/public/tools/bindings/generators/mojom_java_generator.py	2017-01-03 10:28:53.000000000 +0100
@@ -518,8 +518,6 @@
 
   def GenerateFiles(self, unparsed_args):
     # TODO(rockot): Support variant output for Java.
-    if self.variant:
-      raise Exception("Variants not supported in Java bindings.")
 
     parser = argparse.ArgumentParser()
     parser.add_argument('--java_output_directory', dest='java_output_directory')
