diff --git a/qtbase/src/platformsupport/services/genericunix/qgenericunixservices.cpp b/qtbase/src/platformsupport/services/genericunix/qgenericunixservices.cpp
index a24ab82..4c843bb 100644
--- a/qtbase/src/platformsupport/services/genericunix/qgenericunixservices.cpp
+++ b/qtbase/src/platformsupport/services/genericunix/qgenericunixservices.cpp
@@ -72,7 +72,12 @@ static inline QByteArray detectDesktopEnvironment()
     if (desktopSession == "xfce")
         return QByteArrayLiteral("XFCE");
 
+#ifdef Q_OS_MACOS
+    // default to the KDE DE
+    return QByteArrayLiteral("KDE");
+#else
     return QByteArrayLiteral("UNKNOWN");
+#endif
 }
 
 static inline bool checkExecutable(const QString &candidate, QString *result)
@@ -88,6 +93,10 @@ static inline bool detectWebBrowser(const QByteArray &desktop,
     const char *browsers[] = {"google-chrome", "firefox", "mozilla", "opera"};
 
     browser->clear();
+#ifdef Q_OS_MACOS
+    if (checkExecutable(QStringLiteral("open"), browser))
+        return true;
+#endif
     if (checkExecutable(QStringLiteral("xdg-open"), browser))
         return true;
 
diff --git a/qtbase/src/platformsupport/services/services.pro b/qtbase/src/platformsupport/services/services.pro
index 91957a0..0cbf5b1 100644
--- a/qtbase/src/platformsupport/services/services.pro
+++ b/qtbase/src/platformsupport/services/services.pro
@@ -7,7 +7,7 @@ CONFIG += static internal_module
 DEFINES += QT_NO_CAST_FROM_ASCII
 PRECOMPILED_HEADER = ../../corelib/global/qt_pch.h
 
-unix:!darwin: \
+unix: \
     include($$PWD/genericunix/genericunix.pri)
 
 load(qt_module)
diff --git a/qtbase/src/platformsupport/platformsupport.pro b/qtbase/src/platformsupport/platformsupport.pro
index 09e2922..857c5f4 100644
--- a/qtbase/src/platformsupport/platformsupport.pro
+++ b/qtbase/src/platformsupport/platformsupport.pro
@@ -15,7 +15,7 @@ qtConfig(evdev)|qtConfig(tslib)|qtConfig(libinput) {
     input.depends += devicediscovery
 }
 
-unix:!darwin: \
+unix: \
     SUBDIRS += services
 
 qtConfig(opengl): \
