--- qt5kdwe-lnx-work/qtwebengine/orig.qtwebengine.pro	2016-06-07 06:20:06.000000000 +0200
+++ qt5kdwe-lnx-work/qtwebengine/qtwebengine.pro	2016-10-25 09:42:18.348598948 +0200
@@ -6,7 +6,6 @@
     tools/scripts/* \
     tools/qmake/config.tests/khr/* \
     tools/qmake/config.tests/libcap/* \
-    tools/qmake/config.tests/libvpx/* \
     tools/qmake/config.tests/snappy/* \
     tools/qmake/config.tests/srtp/* \
     tools/qmake/mkspecs/features/*
