diff --git a/qtbase/mkspecs/features/qml_plugin.prf b/qtbase/mkspecs/features/qml_plugin.prf
index d49f4c4..673dc5e 100644
--- a/qtbase/mkspecs/features/qml_plugin.prf
+++ b/qtbase/mkspecs/features/qml_plugin.prf
@@ -22,7 +22,7 @@ if(win32|mac):!macx-xcode {
 CONFIG += relative_qt_rpath  # Qt's QML plugins should be relocatable
 
 # Qt libraries should only use Application Extension safe APIs
-darwin:!no_app_extension_api_only: CONFIG += app_extension_api_only
+# darwin:!no_app_extension_api_only: CONFIG += app_extension_api_only
 
 !no_cxx_module:isEmpty(CXX_MODULE) {
     CXX_MODULE = $$TARGET
diff --git a/qtbase/mkspecs/features/qt_module.prf b/qtbase/mkspecs/features/qt_module.prf
index 9d4ebfa..f696ee7 100644
--- a/qtbase/mkspecs/features/qt_module.prf
+++ b/qtbase/mkspecs/features/qt_module.prf
@@ -51,7 +51,7 @@ qtConfig(framework) {
 CONFIG += relative_qt_rpath  # Qt libraries should be relocatable
 
 # Qt libraries should only use Application Extension safe APIs
-darwin:!no_app_extension_api_only: CONFIG += app_extension_api_only
+# darwin:!no_app_extension_api_only: CONFIG += app_extension_api_only
 
 ucmodule = $$upper($$MODULE)
 
diff --git a/qtbase/mkspecs/features/qt_plugin.prf b/qtbase/mkspecs/features/qt_plugin.prf
index 62e1b69..dc7f8ee 100644
--- a/qtbase/mkspecs/features/qt_plugin.prf
+++ b/qtbase/mkspecs/features/qt_plugin.prf
@@ -33,7 +33,7 @@ tool_plugin {
 CONFIG += relative_qt_rpath  # Qt's plugins should be relocatable
 
 # Qt libraries should only use Application Extension safe APIs
-darwin:!no_app_extension_api_only: CONFIG += app_extension_api_only
+# darwin:!no_app_extension_api_only: CONFIG += app_extension_api_only
 
 CONFIG(static, static|shared)|prefix_build {
     isEmpty(MODULE): MODULE = $$basename(TARGET)
