diff --git a/qtbase/src/plugins/platforms/minimal/qminimalintegration.cpp b/qtbase/src/plugins/platforms/minimal/qminimalintegration.cpp
index 0df622a..c2a5ebd 100644
--- a/qtbase/src/plugins/platforms/minimal/qminimalintegration.cpp
+++ b/qtbase/src/plugins/platforms/minimal/qminimalintegration.cpp
@@ -129,7 +129,7 @@ public:
 QPlatformFontDatabase *QMinimalIntegration::fontDatabase() const
 {
     if (!m_fontDatabase && (m_options & EnableFonts)) {
-#if QT_CONFIG(fontconfig)
+#if 0 //QT_CONFIG(fontconfig)
         m_fontDatabase = new QGenericUnixFontDatabase;
 #elif defined(Q_OS_WINRT)
         m_fontDatabase = new QWinRTFontDatabase;
