--- qtwebengine-opensource-src-5.8.0/src/core/orig.gyp_run.pro	2018-02-13 14:49:20.000000000 +0100
+++ qtwebengine-opensource-src-5.8.0/src/core/gyp_run.pro	2018-02-13 14:57:31.000000000 +0100
@@ -26,10 +26,10 @@
 
 linux:qtConfig(separate_debug_info): GYP_CONFIG += linux_dump_symbols=1
 
-force_debug_info {
-    win32: GYP_CONFIG += win_release_extra_cflags=-Zi
-    else: GYP_CONFIG += release_extra_cflags=-g
-}
+#force_debug_info {
+#    win32: GYP_CONFIG += win_release_extra_cflags=-Zi
+#    else: GYP_CONFIG += release_extra_cflags=-g
+#}
 
 !warnings_are_errors: GYP_CONFIG += disable_fatal_linker_warnings=1
 
