diff --git a/src/lib/platform/OSXEventQueueBuffer.cpp b/src/lib/platform/OSXEventQueueBuffer.cpp
index 8e18afc36de81455415015d166141d8a2bf6e02a..4a298201bf346c7d408a96946902fe937ca0ae29 100644
--- a/src/lib/platform/OSXEventQueueBuffer.cpp
+++ b/src/lib/platform/OSXEventQueueBuffer.cpp
@@ -109,12 +109,16 @@ OSXEventQueueBuffer::addEvent(UInt32 dataID)
 
     if (error == noErr) {
     
-        assert(m_carbonEventQueue != NULL);
-        
-        error = PostEventToQueue(
-            m_carbonEventQueue,
-            event,
-            kEventPriorityStandard);
+//         assert(m_carbonEventQueue != NULL);
+
+        // don't be a lazy fool, do a real check
+        // (it will likely be true on shutdown...)
+        if (m_carbonEventQueue) {
+            error = PostEventToQueue(
+                m_carbonEventQueue,
+                event,
+                kEventPriorityStandard);
+        }
         
         ReleaseEvent(event);
     }
