--- src/orig.dispatch_common.h	2017-03-02 19:26:58.000000000 +0100
+++ src/dispatch_common.h	2017-05-26 20:23:25.000000000 +0200
@@ -28,7 +28,7 @@
 #define PLATFORM_HAS_GLX ENABLE_GLX
 #define PLATFORM_HAS_WGL 1
 #elif defined(__APPLE__)
-#define PLATFORM_HAS_EGL 0
+#define PLATFORM_HAS_EGL 1
 #define PLATFORM_HAS_GLX ENABLE_GLX
 #define PLATFORM_HAS_WGL 0
 #elif defined(ANDROID)
--- orig.configure.ac	2017-05-26 20:26:02.000000000 +0200
+++ configure.ac	2017-05-26 20:26:08.000000000 +0200
@@ -108,7 +108,7 @@
   ],
 
   [darwin*], [
-    build_egl=no
+    build_egl=yes
     build_wgl=no
     build_apple=yes
     has_znow=no
