diff --git a/src/assistant/assistant.pro b/src/assistant/assistant.pro
index 799582a404da0cb88cd8961b9e84578e97c2efd7..7e75b38d7b551cf426e627dbadf62116eac029a1 100644
--- a/src/assistant/assistant.pro
+++ b/src/assistant/assistant.pro
@@ -2,19 +2,19 @@ TEMPLATE = subdirs
 
 SUBDIRS += \
            help \
-           assistant \
-           qhelpgenerator \
-           qcollectiongenerator \
-           qhelpconverter
+           assistant
+#            qhelpgenerator \
+#            qcollectiongenerator \
+#            qhelpconverter
 
 assistant.depends = help
-qhelpgenerator.depends = help
-qcollectiongenerator.depends = help
-qhelpconverter.depends = help
-
-qtNomakeTools( \
-    assistant \
-    qhelpgenerator \
-    qcollectiongenerator \
-    qhelpconverter \
-)
+# qhelpgenerator.depends = help
+# qcollectiongenerator.depends = help
+# qhelpconverter.depends = help
+# 
+# qtNomakeTools( \
+#     assistant \
+#     qhelpgenerator \
+#     qcollectiongenerator \
+#     qhelpconverter \
+# )
diff --git a/src/assistant/assistant/assistant.pro b/src/assistant/assistant/assistant.pro
index 82f55092c14cf40ed4e8de2e5a29626abf65332d..e0f060aea5adb10a47672e21f1df9ec9416bfd62 100644
--- a/src/assistant/assistant/assistant.pro
+++ b/src/assistant/assistant/assistant.pro
@@ -1,12 +1,14 @@
-qtHaveModule(webkitwidgets):!contains(QT_CONFIG, static) {
+# qtHaveModule(webkitwidgets):!contains(QT_CONFIG, static) {
     BROWSER = qtwebkit
-} else {
-    BROWSER = qtextbrowser
-}
+# } else {
+#     BROWSER = qtextbrowser
+# }
+
+QT += widgets network sql
+QT += printsupport
+LIBS += -L../help -lQtHelpStatic
 
-QT += widgets network help sql help
-qtHaveModule(printsupport): QT += printsupport
-PROJECTNAME = Assistant
+PROJECTNAME = QtAssistant
 
 include(../../shared/fontpanel/fontpanel.pri)
 
@@ -115,8 +117,8 @@ win32 {
 
 mac {
     ICON = assistant.icns
-    TARGET = Assistant
+    TARGET = QtAssistant
     QMAKE_INFO_PLIST = Info_mac.plist
 }
 
-load(qt_app)
+# load(qt_app)
diff --git a/src/assistant/assistant/centralwidget.cpp b/src/assistant/assistant/centralwidget.cpp
index 43a8b9a3d9eaabd1c85d62b2786eb450d07595df..928ca80019bc7149fc7457ed911d7148d43dd7b7 100644
--- a/src/assistant/assistant/centralwidget.cpp
+++ b/src/assistant/assistant/centralwidget.cpp
@@ -26,6 +26,9 @@
 **
 ****************************************************************************/
 
+#include "../help/qhelpsearchengine.h"
+#include "../help/qhelpsearchquerywidget.h"
+
 #include "centralwidget.h"
 
 #include "findwidget.h"
diff --git a/src/assistant/assistant/helpenginewrapper.cpp b/src/assistant/assistant/helpenginewrapper.cpp
index 20e6ff7ffb3f7a320af022c9f8fe597aee97a700..22f942818c42abaa9b1d3fff432b950e2151bbaf 100644
--- a/src/assistant/assistant/helpenginewrapper.cpp
+++ b/src/assistant/assistant/helpenginewrapper.cpp
@@ -27,6 +27,9 @@
 ****************************************************************************/
 #include "tracer.h"
 
+#include "../help/qhelpsearchengine.h"
+#include "../help/qhelpsearchquerywidget.h"
+
 #include "helpenginewrapper.h"
 #include "../shared/collectionconfiguration.h"
 #include "../help/qhelpengine_p.h"
diff --git a/src/assistant/assistant/remotecontrol.cpp b/src/assistant/assistant/remotecontrol.cpp
index 1c16ac104b20337473f608d794dd7c33937c5e9f..7f5d499c4f88c49cd3e037154a50e37c44e4e9bb 100644
--- a/src/assistant/assistant/remotecontrol.cpp
+++ b/src/assistant/assistant/remotecontrol.cpp
@@ -27,6 +27,9 @@
 ****************************************************************************/
 #include "remotecontrol.h"
 
+#include "../help/qhelpsearchengine.h"
+#include "../help/qhelpsearchquerywidget.h"
+
 #include "centralwidget.h"
 #include "helpenginewrapper.h"
 #include "mainwindow.h"
diff --git a/src/assistant/assistant/searchwidget.cpp b/src/assistant/assistant/searchwidget.cpp
index 136403d6f8307dda44bb3cd3e452cc056226500f..970ecc8f405c77edee830f0f9d4819dd31279aba 100644
--- a/src/assistant/assistant/searchwidget.cpp
+++ b/src/assistant/assistant/searchwidget.cpp
@@ -27,6 +27,9 @@
 ****************************************************************************/
 #include "tracer.h"
 
+#include "../help/qhelpsearchengine.h"
+#include "../help/qhelpsearchquerywidget.h"
+
 #include "mainwindow.h"
 #include "searchwidget.h"
 
diff --git a/src/assistant/help/help.pro b/src/assistant/help/help.pro
index cd8aa64b7cc79c7c8b0c1281d7fc7d22e82940bf..d552c4e50d1a68c3a278d25461ea5bcdb8b9c0ab 100644
--- a/src/assistant/help/help.pro
+++ b/src/assistant/help/help.pro
@@ -1,4 +1,4 @@
-TARGET = QtHelp
+TARGET = QtHelpStatic
 
 QT = core-private gui widgets sql
 QT_PRIVATE = network
@@ -45,4 +45,6 @@ HEADERS += qhelpenginecore.h \
     qhelpsearchindexreader_default_p.h \
     qhelpsearchindexreader_p.h
 
-load(qt_module)
+# load(qt_module)
+TEMPLATE = lib
+CONFIG += staticlib
diff --git a/src/assistant/help/qhelpsearchresultwidget.cpp b/src/assistant/help/qhelpsearchresultwidget.cpp
index e0ca3cc087eb53072188a7dbfe16bb49f81bace6..dda639b6fb626166e1f79e0ba71b30e40ee9cfd1 100644
--- a/src/assistant/help/qhelpsearchresultwidget.cpp
+++ b/src/assistant/help/qhelpsearchresultwidget.cpp
@@ -37,6 +37,8 @@
 **
 ****************************************************************************/
 
+#include "qhelpsearchengine.h"
+
 #include "qhelpsearchresultwidget.h"
 
 #include <QtCore/QList>
