--- src/app/orig.CMakeLists.txt	2016-11-07 00:19:50.000000000 +0100
+++ src/app/CMakeLists.txt	2016-11-27 20:26:26.000000000 +0100
@@ -1,3 +1,5 @@
+include(ECMAddAppIcon)
+
 set(videoplayer_common_SRCS
     theStream.cpp
     actions.cpp
@@ -27,6 +29,9 @@
 
 ki18n_wrap_ui(videoplayer_app_SRCS videoSettingsWidget.ui loadView.ui audioView2.ui)
 
+file(GLOB ICONS_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/../../misc/*-apps-dragonplayer.png")
+ecm_add_app_icon(videoplayer_app_SRCS ICONS ${ICONS_SRCS})
+
 add_executable(dragon ${videoplayer_app_SRCS} )
 
 target_link_libraries(dragon
@@ -47,6 +52,18 @@
     ${PHONON_LIBRARY}
     )
 
+if(APPLE)
+    set_target_properties(dragon PROPERTIES
+        MACOSX_BUNDLE_BUNDLE_NAME       "Dragon"
+        MACOSX_BUNDLE_GUI_IDENTIFIER    "org.kde.dragon"
+        MACOSX_BUNDLE_DISPLAY_NAME      "Dragon Player"
+        MACOSX_BUNDLE_INFO_STRING       "Dragon, a video player that has a usability focus"
+        MACOSX_BUNDLE_COPYRIGHT         "Copyright 2006, Max Howell; Copyright 2007, Ian Monroe"
+        MACOSX_BUNDLE_LONG_VERSION_STRING "Dragon ${KDE_APPLICATIONS_VERSION}"
+        MACOSX_BUNDLE_SHORT_VERSION_STRING "${KDE_APPLICATIONS_VERSION}"
+        MACOSX_BUNDLE_BUNDLE_VERSION "${KDE_APPLICATIONS_VERSION}")
+endif()
+
 install(TARGETS dragon ${INSTALL_TARGETS_DEFAULT_ARGS})
 
 set(videoplayer_part_SRCS
