--- a/Makefile.am.orig	2015-10-18 15:49:52.000000000 +0200
+++ b/Makefile.am	2016-04-27 18:17:24.000000000 +0200
@@ -74,18 +74,18 @@
 
 if CONFIG_USER
 install-data-hook::
-	@echo
-	for plistfile in $(launchddaemon_DATA); do \
-		label=`defaults read "$$PWD"/"$$plistfile" Label`; \
-		echo "Attempting to remove $$label ..."; \
-		$(DESTDIR)/bin/launchctl remove "$$label" \
-		    || echo "Ignoring errors"; \
-		bn=`basename "$$plistfile"`; \
-		echo "Attempting to load $$bn ..."; \
-		$(DESTDIR)/bin/launchctl load -w \
-		$(DESTDIR)$(launchddaemondir)/"$$bn" \
-		    || echo "Ignoring errors"; \
-	done
+#	@echo
+#	for plistfile in $(launchddaemon_DATA); do \
+#		label=`defaults read "$$PWD"/"$$plistfile" Label`; \
+#		echo "Attempting to remove $$label ..."; \
+#		$(DESTDIR)/bin/launchctl remove "$$label" \
+#		    || echo "Ignoring errors"; \
+#		bn=`basename "$$plistfile"`; \
+#		echo "Attempting to load $$bn ..."; \
+#		$(DESTDIR)/bin/launchctl load -w \
+#		$(DESTDIR)$(launchddaemondir)/"$$bn" \
+#		    || echo "Ignoring errors"; \
+#	done
 	echo "Install done."
 
 uninstall-hook::
