diff --git a/module/zfs/zfs_osx.cpp b/module/zfs/zfs_osx.cpp
index a4b324e42..0a767e8af 100644
--- a/module/zfs/zfs_osx.cpp
+++ b/module/zfs/zfs_osx.cpp
@@ -341,8 +341,19 @@ net_lundman_zfs_zvol::start (IOService *provider)
       }
     }
 
+
+	/*
+	 * We need to call spl_vfs_start() here if we are NOT in boot/mountroot
+	 * situation.
+	 * If we are booting, then it is set in vfs_root. We
+	 * could do better checks for boot time maybe (can provider be
+	 * checked for IOkit match?)  zfs_boot_init() also sets it if
+	 * it detects it is not boot time, but kextload.
+	 */
 #ifdef ZFS_BOOT
 	zfs_boot_init(this);
+#else
+	spl_vfs_start();
 #endif
 
     return res;
