diff --git a/module/spl/KernelExports/zfs.exports b/module/spl/KernelExports/zfs.exports
index 4cd7d517b86e529bcb10eb0383d2ef7d242dd008..ec65a98393b8c500afb120a1e1620ba9029dfbb6 100644
--- a/spl-4d1859e3060bc1991d10d9215a3d1845e7caeaf8/module/spl/KernelExports/zfs.exports
+++ b/spl-4d1859e3060bc1991d10d9215a3d1845e7caeaf8/module/spl/KernelExports/zfs.exports
@@ -29,8 +29,6 @@ _kauth_cred_getgroups
 _rootvnode
 _decmpfs_decompress_file
 _decmpfs_file_is_compressed
-_decmpfs_cnode_alloc
-_decmpfs_cnode_free
 _decmpfs_cnode_init
 _decmpfs_cnode_destroy
 _cpuid_info
