diff --git a/lib/libspl/include/sys/byteorder.h b/lib/libspl/include/sys/byteorder.h
index 65ee187..e48c997 100644
--- a/lib/libspl/include/sys/byteorder.h
+++ b/lib/libspl/include/sys/byteorder.h
@@ -90,10 +90,13 @@ extern	in_port_t ntohs(in_port_t);
 #endif	/* !defined(_XPG4_2) || defined(__EXTENSIONS__) || defined(_XPG5) */
 #endif
 
-/* 10.8 is lacking in htonll */
+/* 10.8 and 10.9 are lacking in htonll and ntohll */
 #if !defined(htonll)
 #define htonll(x)       __DARWIN_OSSwapInt64(x)
 #endif
+#if !defined(ntohll)
+#define ntohll(x)       __DARWIN_OSSwapInt64(x)
+#endif
 
 
 #if !defined(_XPG4_2) || defined(__EXTENSIONS__)
diff --git a/lib/libzfs/libzfs_mount.c b/lib/libzfs/libzfs_mount.c
index 418dea3..e3b62a8 100644
--- a/lib/libzfs/libzfs_mount.c
+++ b/lib/libzfs/libzfs_mount.c
@@ -256,7 +256,7 @@ dir_is_empty_stat(const char *dirname)
 /*
  * "openat" came to OS X Version 10.10.
  */
-#if (MAC_OS_X_VERSION_MIN_REQUIRED >= MAC_OS_X_VERSION_10_10)
+#if defined(MAC_OS_X_VERSION_10_10) && (MAC_OS_X_VERSION_MIN_REQUIRED >= MAC_OS_X_VERSION_10_10)
 static boolean_t
 dir_is_empty_readdir(const char *dirname)
 {
