diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9af06181f89eafc493065304485e408518543d77..b9724b0b2f041093a4315ab576dae73aa6278121 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,6 +6,8 @@ include(SetKexiVersionInfo)
 
 project(Kexi VERSION ${KEXI_VERSION})
 
+option(SHOULD_BUILD_DOC "Build the documentation" OFF)
+
 include(ECMInstallIcons)
 include(ECMAddAppIcon)
 include(ECMSetupVersion)
@@ -91,6 +93,9 @@ macro_bool_to_01(KF5Crash_FOUND HAVE_KCRASH)
 macro_log_feature(${KF5Crash_FOUND} "KCrash" "KDE's Crash Handler"
                   "https://api.kde.org/frameworks/kcrash/html" FALSE ""
                   "Optionally used to provide crash reporting on Linux")
+if(SHOULD_BUILD_DOC)
+    find_package(KF5 ${KF5_DEP_VERSION} REQUIRED COMPONENTS DocTools)
+endif()
 
 set(REQUIRED_QT_VERSION 5.4.0)
 find_package(Qt5 ${REQUIRED_QT_VERSION} REQUIRED COMPONENTS Core Gui Widgets Xml Network PrintSupport Test)
diff --git a/doc/en/kexi/CMakeLists.txt b/doc/en/kexi/CMakeLists.txt
index 17d3b1ee475f9b80109aaf3c829f767c1b81e6df..764a5e44d484268a48170d376d5ee87f3a5b3df9 100644
--- a/doc/en/kexi/CMakeLists.txt
+++ b/doc/en/kexi/CMakeLists.txt
@@ -1,4 +1,4 @@
 ########### install files ###############
 #
 #
-kdoctools_create_handbook(index.docbook INSTALL_DESTINATION ${HTML_INSTALL_DIR}/en)
+kdoctools_create_handbook(index.docbook INSTALL_DESTINATION ${HTML_INSTALL_DIR}/en SUBDIR kexi)
