--- kexi-3.0.0/CMakeLists.txt.orig	2016-12-26 15:47:01.000000000 +0100
+++ kexi-3.0.0/CMakeLists.txt	2016-12-26 16:23:44.000000000 +0100
@@ -94,7 +94,8 @@
 
 set(REQUIRED_QT_VERSION 5.4.0)
 find_package(Qt5 ${REQUIRED_QT_VERSION} REQUIRED COMPONENTS Core Gui Widgets Xml Network PrintSupport Test)
-find_package(Qt5 ${REQUIRED_QT_VERSION} COMPONENTS UiTools WebKit WebKitWidgets)
+find_package(Qt5 ${REQUIRED_QT_VERSION} COMPONENTS UiTools)
+#QWK find_package(Qt5 ${REQUIRED_QT_VERSION} COMPONENTS WebKit WebKitWidgets)
 
 # use sane compile flags
 add_definitions(
@@ -162,7 +163,7 @@
                                   "http://community.kde.org/KProperty" FALSE "" "Required by Kexi")
 
 include(CheckIfQtGuiCanBeExecuted)
-include(CheckGlobalBreezeIcons)
+# include(CheckGlobalBreezeIcons)
 
  ###########################
 ############################
@@ -174,7 +175,7 @@
 ## Test for marble
 ##
 set(MARBLE_MIN_VERSION "0.19.2")
-find_package(KexiMarble)
+# find_package(KexiMarble)
 if(NOT MARBLE_FOUND)
     set(MARBLE_INCLUDE_DIR "")
 else()
