diff --git a/src/kexiutils/completer/KexiCompleter.cpp b/src/kexiutils/completer/KexiCompleter.cpp
index 503ce0df3b2cdf929d7a9c7854e4ec5db4e32af3..8c616706cf8092a5c0d0f86cd77839887ac5303a 100644
--- a/src/kexiutils/completer/KexiCompleter.cpp
+++ b/src/kexiutils/completer/KexiCompleter.cpp
@@ -1193,7 +1193,7 @@ void KexiCompleter::setPopup(QAbstractItemView *popup)
         delete d->popup;
     if (popup->model() != d->proxy)
         popup->setModel(d->proxy);
-#if defined(Q_OS_MAC) && !defined(QT_MAC_USE_COCOA)
+#if defined(Q_OS_MACOS) && !defined(QT_MAC_USE_COCOA)
      popup->show();
 #else
      popup->hide();
diff --git a/src/main/KexiBugReportDialog.cpp b/src/main/KexiBugReportDialog.cpp
index cf8a9404b86dbee6e31f0629ea8e8826850be6f7..47dcf71401cefd5dee85f3f7f8f47bf921e4f977 100644
--- a/src/main/KexiBugReportDialog.cpp
+++ b/src/main/KexiBugReportDialog.cpp
@@ -213,7 +213,7 @@ void KexiBugReportDialog::collectData()
 #elif defined(Q_OS_SOLARIS)
     m_op_sys = "Solaris";
     m_rep_platform = "Solaris Packages";
-#elif defined(Q_OS_MAC)
+#elif defined(Q_OS_MACOS)
     m_op_sys = "OS X";
     m_rep_platform = "Mac OS X Disk Images";
 #elif defined(Q_OS_WIN)
diff --git a/src/main/KexiMainWindow.cpp b/src/main/KexiMainWindow.cpp
index 81b9e8bb8f9eb05feb268f0d682a03ad9bd5af4c..cf2d002ffd65b38c0bdbba2b9a6bf11ca2a0f812 100644
--- a/src/main/KexiMainWindow.cpp
+++ b/src/main/KexiMainWindow.cpp
@@ -324,7 +324,7 @@ static bool setupIconTheme(KLocalizedString *errorMessage, KLocalizedString *det
 //! @todo 3.1 replace with KexiStyle
 bool setupApplication()
 {
-#if defined Q_OS_WIN || defined Q_OS_MAC
+#if defined Q_OS_WIN || defined Q_OS_MACOS
     // Only this style matches current Kexi theme and can be supported/tested
     const char *name = "breeze";
     QScopedPointer<QStyle> style(QStyleFactory::create(name));
diff --git a/src/main/KexiMenuWidget.cpp b/src/main/KexiMenuWidget.cpp
index 089b73aaa042b27aa99977da9d602a28cfa4ecdc..247ff808cac012e61d872be5206cca72732a89b3 100644
--- a/src/main/KexiMenuWidget.cpp
+++ b/src/main/KexiMenuWidget.cpp
@@ -2600,7 +2600,7 @@ void KexiMenuWidget::keyPressEvent(QKeyEvent *e)
         else if (key == Qt::Key_Right)
             key = Qt::Key_Left;
     }
-#ifndef Q_OS_MAC
+#ifndef Q_OS_MACOS
     if (key == Qt::Key_Tab) //means down
         key = Qt::Key_Down;
     if (key == Qt::Key_Backtab) //means up
diff --git a/src/main/KexiUserFeedbackAgent.cpp b/src/main/KexiUserFeedbackAgent.cpp
index d11d923e21b416c60887a2a562cbe5b62a049b3a..19fd6604ddfdfd1af1ffc516f283f925dcedcaad 100644
--- a/src/main/KexiUserFeedbackAgent.cpp
+++ b/src/main/KexiUserFeedbackAgent.cpp
@@ -203,7 +203,7 @@ void KexiUserFeedbackAgent::Private::updateData()
         }
         p.close();
     }
-#elif defined(Q_OS_MAC)
+#elif defined(Q_OS_MACOS)
     ADD("os", "mac", SystemInfoArea);
 #elif defined(Q_OS_WIN)
     ADD("os", "windows", SystemInfoArea);
diff --git a/src/main/startup/KexiWelcomeStatusBar.cpp b/src/main/startup/KexiWelcomeStatusBar.cpp
index 0fddee6a4196fcb8dfce6540f03b0ba8a41af562..9c5fbaff0e0bcdd5bbf716ef6482f84b60d68b8a 100644
--- a/src/main/startup/KexiWelcomeStatusBar.cpp
+++ b/src/main/startup/KexiWelcomeStatusBar.cpp
@@ -36,7 +36,7 @@
 #include <KSharedConfig>
 #include <KLocalizedString>
 #include <KMessageBox>
-#if defined(Q_OS_WIN) || defined(Q_OS_MAC)
+#if defined(Q_OS_WIN) || defined(Q_OS_MACOS)
 #  include <QNetworkAccessManager>
 #  include <QNetworkRequest>
 #  include <QNetworkReply>
@@ -200,9 +200,9 @@ public Q_SLOTS:
     }
 
 private Q_SLOTS:
-#ifdef USE_KIO_COPY
     void filesCopyFinished(KJob* job)
     {
+#ifdef USE_KIO_COPY
         if (job->error()) {
             //! @todo error...
             qDebug() << "ERROR:" << job->errorString();
@@ -213,13 +213,14 @@ private Q_SLOTS:
         Q_UNUSED(copyJob)
         //qDebug() << "DONE" << copyJob->destUrl();
         (void)copyFilesToDestinationDir();
+#endif
     }
-#else
 
-#define DOWNLOAD_BUFFER_SIZE 1024 * 50
 
     void fileDownloadFinished(QNetworkReply* reply)
     {
+#ifndef USE_KIO_COPY
+#define DOWNLOAD_BUFFER_SIZE 1024 * 50
         const bool ok = copyFile(reply);
         reply->deleteLater();
         if (!ok) {
@@ -230,10 +231,12 @@ private Q_SLOTS:
         }
         m_sourceFilesToDownload.removeFirst();
         downloadNextFile();
+#endif
     }
 
     bool copyFile(QNetworkReply* reply)
     {
+#ifndef USE_KIO_COPY
         if (reply->error() != QNetworkReply::NoError) {
             return false;
         }
@@ -257,9 +260,11 @@ private Q_SLOTS:
             return false;
         }
         return true;
+#endif
     }
 
 private:
+#ifndef USE_KIO_COPY
     void downloadNextFile() {
         if (m_sourceFilesToDownload.isEmpty()) {
             // success
