--- kexi-3.0.0/doc/CMakeLists.txt.orig	2016-12-26 18:46:37.000000000 +0100
+++ kexi-3.0.0/doc/CMakeLists.txt	2016-12-26 18:47:17.000000000 +0100
@@ -1,8 +1,10 @@
-add_subdirectory(de)
+find_package(KF5 ${KF5_DEP_VERSION} REQUIRED COMPONENTS DocTools)
+
+# add_subdirectory(de)
 add_subdirectory(en)
-add_subdirectory(it)
-add_subdirectory(nl)
-add_subdirectory(pt_BR)
-add_subdirectory(pt)
-add_subdirectory(sv)
-add_subdirectory(uk)
+# add_subdirectory(it)
+# add_subdirectory(nl)
+# add_subdirectory(pt_BR)
+# add_subdirectory(pt)
+# add_subdirectory(sv)
+# add_subdirectory(uk)
--- kexi-3.0.0/doc/en/kexi/CMakeLists.txt.orig	2016-09-19 16:41:39.000000000 +0200
+++ kexi-3.0.0/doc/en/kexi/CMakeLists.txt	2016-12-26 18:47:42.000000000 +0100
@@ -1,4 +1,4 @@
 ########### install files ###############
 #
 #
-kdoctools_create_handbook(index.docbook INSTALL_DESTINATION ${HTML_INSTALL_DIR}/en)
+kdoctools_create_handbook(index.docbook INSTALL_DESTINATION ${HTML_INSTALL_DIR}/en SUBDIR kexi)
