diff --git a/konqueror/webenginepart/src/webenginepart_ext.cpp b/konqueror/webenginepart/src/webenginepart_ext.cpp
index 5796766..0662d79 100644
--- a/konqueror/webenginepart/src/webenginepart_ext.cpp
+++ b/konqueror/webenginepart/src/webenginepart_ext.cpp
@@ -18,6 +18,7 @@
  *
  */
 
+#include <QtWebEngine/QtWebEngineVersion>
 #include "webenginepart_ext.h"
 
 #include "webenginepart.h"
@@ -97,16 +98,20 @@ WebView* WebEngineBrowserExtension::view()
 
 int WebEngineBrowserExtension::xOffset()
 {
+#if QTWEBENGINE_VERSION >= QT_VERSION_CHECK(5, 7, 0)
     if (view())
         return view()->page()->scrollPosition().x();
+#endif
 
     return KParts::BrowserExtension::xOffset();
 }
 
 int WebEngineBrowserExtension::yOffset()
 {
+#if QTWEBENGINE_VERSION >= QT_VERSION_CHECK(5, 7, 0)
    if (view())
         return view()->page()->scrollPosition().y();
+#endif
 
     return KParts::BrowserExtension::yOffset();
 }
@@ -356,6 +361,7 @@ void WebEngineBrowserExtension::slotSaveImageAs()
 
 void WebEngineBrowserExtension::slotSendImage()
 {
+#if QTWEBENGINE_VERSION >= QT_VERSION_CHECK(5, 7, 0)
     if (!view())
         return;
 
@@ -366,10 +372,12 @@ void WebEngineBrowserExtension::slotSendImage()
                                   QString(), //body
                                   QString(),
                                   urls); // attachments
+#endif
 }
 
 void WebEngineBrowserExtension::slotCopyImageURL()
 {
+#if QTWEBENGINE_VERSION >= QT_VERSION_CHECK(5, 7, 0)
     if (!view())
         return;
 
@@ -386,11 +394,13 @@ void WebEngineBrowserExtension::slotCopyImageURL()
     mimeData = new QMimeData;
     mimeData->setUrls(safeURLList);
     QApplication::clipboard()->setMimeData(mimeData, QClipboard::Selection);
+#endif
 }
 
 
 void WebEngineBrowserExtension::slotCopyImage()
 {
+#if QTWEBENGINE_VERSION >= QT_VERSION_CHECK(5, 7, 0)
     if (!view())
         return;
 
@@ -410,16 +420,20 @@ void WebEngineBrowserExtension::slotCopyImage()
 //    mimeData->setImageData(view()->contextMenuResult().pixmap());
     mimeData->setUrls(safeURLList);
     QApplication::clipboard()->setMimeData(mimeData, QClipboard::Selection);
+#endif
 }
 
 void WebEngineBrowserExtension::slotViewImage()
 {
+#if QTWEBENGINE_VERSION >= QT_VERSION_CHECK(5, 7, 0)
     if (view())
         emit createNewWindow(view()->contextMenuResult().mediaUrl());
+#endif
 }
 
 void WebEngineBrowserExtension::slotBlockImage()
 {
+#if QTWEBENGINE_VERSION >= QT_VERSION_CHECK(5, 7, 0)
     if (!view())
         return;
 
@@ -432,6 +446,7 @@ void WebEngineBrowserExtension::slotBlockImage()
         WebEngineSettings::self()->addAdFilter(url);
         reparseConfiguration();
     }
+#endif
 }
 
 void WebEngineBrowserExtension::slotBlockHost()
@@ -453,21 +468,25 @@ void WebEngineBrowserExtension::slotCopyLinkURL()
 
 void WebEngineBrowserExtension::slotCopyLinkText()
 {
+#if QTWEBENGINE_VERSION >= QT_VERSION_CHECK(5, 7, 0)
     if (view()) {
         QMimeData* data = new QMimeData;
         data->setText(view()->contextMenuResult().linkText());
         QApplication::clipboard()->setMimeData(data, QClipboard::Clipboard);
     }
+#endif
 }
 
 void WebEngineBrowserExtension::slotCopyEmailAddress()
 {
+#if QTWEBENGINE_VERSION >= QT_VERSION_CHECK(5, 7, 0)
     if (view()) {
         QMimeData* data = new QMimeData;
         const QUrl url(view()->contextMenuResult().linkUrl());
         data->setText(url.path());
         QApplication::clipboard()->setMimeData(data, QClipboard::Clipboard);
     }
+#endif
 }
 
 void WebEngineBrowserExtension::slotSaveLinkAs()
@@ -497,6 +516,7 @@ void WebEngineBrowserExtension::slotViewDocumentSource()
     }
 }
 
+#if QTWEBENGINE_VERSION >= QT_VERSION_CHECK(5, 7, 0)
 static bool isMultimediaElement(QWebEngineContextMenuData::MediaType mediaType)
 {
     switch(mediaType)
@@ -508,9 +528,11 @@ static bool isMultimediaElement(QWebEngineContextMenuData::MediaType mediaType)
             return false;
     }
 }
+#endif
 
 void WebEngineBrowserExtension::slotLoopMedia()
 {
+#if QTWEBENGINE_VERSION >= QT_VERSION_CHECK(5, 7, 0)
     if (!view())
         return;
 
@@ -518,10 +540,12 @@ void WebEngineBrowserExtension::slotLoopMedia()
     if (!isMultimediaElement( data.mediaType()))
         return;
     view()->page()->triggerAction(QWebEnginePage::ToggleMediaLoop);
+#endif
 }
 
 void WebEngineBrowserExtension::slotMuteMedia()
 {
+#if QTWEBENGINE_VERSION >= QT_VERSION_CHECK(5, 7, 0)
     if (!view())
         return;
 
@@ -529,10 +553,12 @@ void WebEngineBrowserExtension::slotMuteMedia()
     if (!isMultimediaElement( data.mediaType()))
         return;
     view()->page()->triggerAction(QWebEnginePage::ToggleMediaMute);
+#endif
 }
 
 void WebEngineBrowserExtension::slotPlayMedia()
 {
+#if QTWEBENGINE_VERSION >= QT_VERSION_CHECK(5, 7, 0)
     if (!view())
         return;
 
@@ -540,10 +566,12 @@ void WebEngineBrowserExtension::slotPlayMedia()
     if (!isMultimediaElement( data.mediaType()))
         return;
     view()->page()->triggerAction(QWebEnginePage::ToggleMediaPlayPause);
+#endif
 }
 
 void WebEngineBrowserExtension::slotShowMediaControls()
 {
+#if QTWEBENGINE_VERSION >= QT_VERSION_CHECK(5, 7, 0)
     if (!view())
         return;
 
@@ -551,6 +579,7 @@ void WebEngineBrowserExtension::slotShowMediaControls()
     if (!isMultimediaElement( data.mediaType()))
         return;
     view()->page()->triggerAction(QWebEnginePage::ToggleMediaControls);
+#endif
 }
 
 #if 0
@@ -570,6 +599,7 @@ static QUrl mediaUrlFrom(QWebElement& element)
 
 void WebEngineBrowserExtension::slotSaveMedia()
 {
+#if QTWEBENGINE_VERSION >= QT_VERSION_CHECK(5, 7, 0)
     if (!view())
         return;
 
@@ -577,10 +607,12 @@ void WebEngineBrowserExtension::slotSaveMedia()
     if (!isMultimediaElement( data.mediaType()))
         return;
     emit saveUrl(data.mediaUrl());
+#endif
 }
 
 void WebEngineBrowserExtension::slotCopyMedia()
 {
+#if QTWEBENGINE_VERSION >= QT_VERSION_CHECK(5, 7, 0)
     if (!view())
         return;
     QWebEngineContextMenuData data =  view()->contextMenuResult();
@@ -603,6 +635,7 @@ void WebEngineBrowserExtension::slotCopyMedia()
     mimeData = new QMimeData;
     mimeData->setUrls(safeURLList);
     QApplication::clipboard()->setMimeData(mimeData, QClipboard::Selection);
+#endif
 }
 
 void WebEngineBrowserExtension::slotTextDirectionChanged()
@@ -784,6 +817,7 @@ void WebEngineBrowserExtension::slotOpenSelection()
 
 void WebEngineBrowserExtension::slotLinkInTop()
 {
+#if QTWEBENGINE_VERSION >= QT_VERSION_CHECK(5, 7, 0)
     if (!view())
         return;
 
@@ -796,6 +830,7 @@ void WebEngineBrowserExtension::slotLinkInTop()
     const QUrl url(view()->contextMenuResult().linkUrl());
 
     emit openUrlRequest(url, uargs, bargs);
+#endif
 }
 
 ////
diff --git a/konqueror/webenginepart/src/webpage.cpp b/konqueror/webenginepart/src/webpage.cpp
index 1579581..10804d9 100644
--- a/konqueror/webenginepart/src/webpage.cpp
+++ b/konqueror/webenginepart/src/webpage.cpp
@@ -738,10 +738,12 @@ bool NewWindowPage::acceptNavigationRequest(const QUrl &url, NavigationType type
             case WebBrowserTab:
                 bargs.setNewTab(true);
                 break;
+#if QTWEBENGINE_VERSION >= QT_VERSION_CHECK(5, 7, 0)
             case WebBrowserBackgroundTab:
                 bargs.setNewTab(true);
                 wargs.setLowerWindow(true);
                 break;
+#endif
         }
 
         KParts::ReadOnlyPart* newWindowPart =0;
diff --git a/konqueror/webenginepart/src/webview.cpp b/konqueror/webenginepart/src/webview.cpp
index 4e829d1..4ff49ae 100644
--- a/konqueror/webenginepart/src/webview.cpp
+++ b/konqueror/webenginepart/src/webview.cpp
@@ -71,6 +71,10 @@ WebView::WebView(WebEnginePart* part, QWidget* parent)
     
     if (WebEngineSettings::self()->zoomToDPI())
         setZoomFactor(logicalDpiY() / 96.0f);
+
+#ifndef HAVE_WEBENGINECONTEXTMENUDATA
+    m_result = 0;
+#endif
 }
 
 WebView::~WebView()
@@ -131,6 +135,7 @@ static void extractMimeTypeFor(const QUrl& url, QString& mimeType)
 
 void WebView::contextMenuEvent(QContextMenuEvent* e)
 {
+#ifdef HAVE_WEBENGINECONTEXTMENUDATA
     m_result = page()->contextMenuData();
 
     // Clear the previous collection entries first...
@@ -187,6 +192,7 @@ void WebView::contextMenuEvent(QContextMenuEvent* e)
         return;
     }
 
+#endif
     QWebEngineView::contextMenuEvent(e);
 }
 
@@ -330,6 +336,7 @@ void WebView::partActionPopupMenu(KParts::BrowserExtension::ActionGroupMap& part
 {
     QList<QAction*> partActions;
 
+#ifdef HAVE_WEBENGINECONTEXTMENUDATA
     if (m_result.mediaUrl().isValid()) {
         QAction *action;
         action = new QAction(i18n("Save Image As..."), this);
@@ -376,6 +383,7 @@ void WebView::partActionPopupMenu(KParts::BrowserExtension::ActionGroupMap& part
             }
         }
     }
+#endif
 
     {
         QAction *separatorAction = new QAction(m_actionCollection);
@@ -418,33 +426,43 @@ void WebView::selectActionPopupMenu(KParts::BrowserExtension::ActionGroupMap& se
 
 void WebView::linkActionPopupMenu(KParts::BrowserExtension::ActionGroupMap& linkGroupMap)
 {
+#ifdef HAVE_WEBENGINECONTEXTMENUDATA
     Q_ASSERT(!m_result.linkUrl().isEmpty());
 
     const QUrl url(m_result.linkUrl());
+#else
+    const QUrl url;
+#endif
 
     QList<QAction*> linkActions;
 
     QAction* action;
 
+#ifdef HAVE_WEBENGINECONTEXTMENUDATA
     if (!m_result.selectedText().isEmpty()) {
         action = m_actionCollection->addAction(KStandardAction::Copy, QL1S("copy"),  m_part->browserExtension(), SLOT(copy()));
         action->setText(i18n("&Copy Text"));
         action->setEnabled(m_part->browserExtension()->isActionEnabled("copy"));
         linkActions.append(action);
     }
+#endif
 
     if (url.scheme() == "mailto") {
+#ifdef HAVE_WEBENGINECONTEXTMENUDATA
         action = new QAction(i18n("&Copy Email Address"), this);
         m_actionCollection->addAction(QL1S("copylinklocation"), action);
         connect(action, SIGNAL(triggered(bool)), m_part->browserExtension(), SLOT(slotCopyEmailAddress()));
         linkActions.append(action);
+#endif
     } else {
+#ifdef HAVE_WEBENGINECONTEXTMENUDATA
         if (!m_result.linkText().isEmpty()) {
             action = new QAction(QIcon::fromTheme("edit-copy"), i18n("Copy Link &Text"), this);
             m_actionCollection->addAction(QL1S("copylinktext"), action);
             connect(action, SIGNAL(triggered(bool)), m_part->browserExtension(), SLOT(slotCopyLinkText()));
             linkActions.append(action);
         }
+#endif
 
         action = new QAction(i18n("Copy Link &URL"), this);
         m_actionCollection->addAction(QL1S("copylinkurl"), action);
@@ -462,6 +480,7 @@ void WebView::linkActionPopupMenu(KParts::BrowserExtension::ActionGroupMap& link
 
 void WebView::multimediaActionPopupMenu(KParts::BrowserExtension::ActionGroupMap& mmGroupMap)
 {
+#ifdef HAVE_WEBENGINECONTEXTMENUDATA
     QList<QAction*> multimediaActions;
 
     const bool isVideoElement = m_result.mediaType() == QWebEngineContextMenuData::MediaTypeVideo;
@@ -514,6 +533,7 @@ void WebView::multimediaActionPopupMenu(KParts::BrowserExtension::ActionGroupMap
     multimediaActions.append(action);
 
     mmGroupMap.insert("partactions", multimediaActions);
+#endif
 }
 
 void WebView::slotStopAutoScroll()
diff --git a/konqueror/webenginepart/src/webview.h b/konqueror/webenginepart/src/webview.h
index dc461d4..ad38c8e 100644
--- a/konqueror/webenginepart/src/webview.h
+++ b/konqueror/webenginepart/src/webview.h
@@ -27,7 +27,13 @@
 #include <KParts/BrowserExtension>
 
 #include <QWebEngineView>
+#include <QtWebEngine/QtWebEngineVersion>
+#if QTWEBENGINE_VERSION >= QT_VERSION_CHECK(5, 7, 0)
+#define HAVE_WEBENGINECONTEXTMENUDATA
 #include <QWebEngineContextMenuData>
+#else
+typedef void* QWebEngineContextMenuData;
+#endif
 
 class QUrl;
 class WebEnginePart;
