diff --git a/konqueror/src/CMakeLists.txt b/konqueror/src/CMakeLists.txt
index 632983f..f8daf04 100644
--- a/konqueror/src/CMakeLists.txt
+++ b/konqueror/src/CMakeLists.txt
@@ -103,7 +103,7 @@ qt5_add_dbus_interface(konqueror_KDEINIT_SRCS org.kde.Konqueror.UndoManager.xml
 qt5_add_dbus_adaptor(konqueror_KDEINIT_SRCS org.kde.Konqueror.SessionManager.xml konqsessionmanager.h KonqSessionManager konqsessionmanageradaptor KonqSessionManagerAdaptor)
 qt5_add_dbus_interface(konqueror_KDEINIT_SRCS org.kde.Konqueror.SessionManager.xml konqsessionmanager_interface)
 
-kde4_add_app_icon(konqueror_KDEINIT_SRCS "${KDE4_ICON_INSTALL_DIR}/oxygen/*/apps/konqueror.png")
+kde4_add_app_icon(konqueror_KDEINIT_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/../pics/*-apps-konqueror.png")
 kf5_add_kdeinit_executable(konqueror ${konqueror_KDEINIT_SRCS} konqmain.cpp)
 target_link_libraries(kdeinit_konqueror
    konquerorprivate
diff --git a/konqueror/client/CMakeLists.txt b/konqueror/client/CMakeLists.txt
index 745b0f3..166b0f1 100644
--- a/konqueror/client/CMakeLists.txt
+++ b/konqueror/client/CMakeLists.txt
@@ -10,7 +10,8 @@ qt5_add_dbus_interface( kfmclient_SRCS ../src/org.kde.Konqueror.MainWindow.xml k
 
 add_definitions(-DQT_NO_CAST_ASCII)
 
-kde4_add_app_icon(kfmclient_SRCS "${KDE4_ICON_INSTALL_DIR}/oxygen/*/apps/system-file-manager.png")
+kde4_add_app_icon(kfmclient_SRCS "${CMAKE_INSTALL_PREFIX}/share/icons/oxygen/base/*/apps/system-file-manager.png"
+    "${CMAKE_INSTALL_PREFIX}/share/icons/oxygen/*/apps/system-file-manager.png")
 
 if (WIN32)
     add_definitions(-Dkdemain=main)
