diff --git a/nsplugins/plugin_paths.cpp b/nsplugins/plugin_paths.cpp
index 40e79b0..dab7266 100644
--- a/nsplugins/plugin_paths.cpp
+++ b/nsplugins/plugin_paths.cpp
@@ -40,6 +40,10 @@ QStringList getSearchPaths()
         // keep sync with kdebase/apps/konqueror/settings/konqhtml
         paths.append("$HOME/.mozilla/plugins");
         paths.append("$HOME/.netscape/plugins");
+#ifdef Q_OS_OSX
+	   paths.append("$HOME/Library/Internet Plug-Ins");
+	   paths.append("/Library/Internet Plug-Ins");
+#endif
         paths.append("/usr/lib/firefox/plugins");
         paths.append("/usr/lib64/browser-plugins");
         paths.append("/usr/lib/browser-plugins");
diff --git a/konqueror/settings/konqhtml/pluginopts.cpp b/konqueror/settings/konqhtml/pluginopts.cpp
index e4130bc..6c1425e 100644
--- a/konqueror/settings/konqhtml/pluginopts.cpp
+++ b/konqueror/settings/konqhtml/pluginopts.cpp
@@ -419,6 +419,10 @@ void KPluginOptions::dirLoad(KSharedConfig::Ptr config, bool useDefault)
     } else { //keep sync with kdebase/apps/nsplugins
         paths.append("$HOME/.mozilla/plugins");
         paths.append("$HOME/.netscape/plugins");
+#ifdef Q_OS_OSX
+	   paths.append("$HOME/Library/Internet Plug-Ins");
+	   paths.append("/Library/Internet Plug-Ins");
+#endif
         paths.append("/usr/lib/firefox/plugins");
         paths.append("/usr/lib64/browser-plugins");
         paths.append("/usr/lib/browser-plugins");
