diff --git a/keditbookmarks/CMakeLists.txt b/keditbookmarks/CMakeLists.txt
index c0d6c18..f88e826 100644
--- a/keditbookmarks/CMakeLists.txt
+++ b/keditbookmarks/CMakeLists.txt
@@ -79,7 +79,7 @@ kconfig_add_kcfg_files(keditbookmarks_SRCS settings.kcfgc )
 
 add_executable( keditbookmarks ${keditbookmarks_SRCS})
 
-target_link_libraries(keditbookmarks kbookmarkmodel_private
+target_link_libraries(keditbookmarks kbookmarkmodel_private5
    KF5::Parts
    KF5::Bookmarks
    KF5::WindowSystem
diff --git a/keditbookmarks/kbookmarkmodel/CMakeLists.txt b/keditbookmarks/kbookmarkmodel/CMakeLists.txt
index d013328..0f3d5d6 100644
--- a/keditbookmarks/kbookmarkmodel/CMakeLists.txt
+++ b/keditbookmarks/kbookmarkmodel/CMakeLists.txt
@@ -8,8 +8,8 @@ set(kbookmarkmodel_SRCS
   view.cpp
 )
 
-add_library(kbookmarkmodel_private SHARED ${kbookmarkmodel_SRCS})
-target_link_libraries(kbookmarkmodel_private
+add_library(kbookmarkmodel_private5 SHARED ${kbookmarkmodel_SRCS})
+target_link_libraries(kbookmarkmodel_private5
 PUBLIC
    Qt5::Widgets
    KF5::XmlGui # KActionCollection
@@ -19,12 +19,12 @@ PRIVATE
 )
 
 include(GenerateExportHeader)
-generate_export_header(kbookmarkmodel_private BASE_NAME kbookmarkmodel EXPORT_FILE_NAME kbookmarkmodel_export.h)
+generate_export_header(kbookmarkmodel_private5 BASE_NAME kbookmarkmodel EXPORT_FILE_NAME kbookmarkmodel_export.h)
 
-set_target_properties(kbookmarkmodel_private PROPERTIES
+set_target_properties(kbookmarkmodel_private5 PROPERTIES
      VERSION ${KEDITBOOKMARKS_VERSION}
      SOVERSION "6"
 )
 
 
-install(TARGETS kbookmarkmodel_private ${KDE_INSTALL_TARGETS_DEFAULT_ARGS})
+install(TARGETS kbookmarkmodel_private5 ${KDE_INSTALL_TARGETS_DEFAULT_ARGS})
diff --git a/keditbookmarks/kbookmarkmodel/tests/CMakeLists.txt b/keditbookmarks/kbookmarkmodel/tests/CMakeLists.txt
index 6175eaa..3977e25 100644
--- a/keditbookmarks/kbookmarkmodel/tests/CMakeLists.txt
+++ b/keditbookmarks/kbookmarkmodel/tests/CMakeLists.txt
@@ -11,7 +11,7 @@ set(kbookmarkmodeltest_SRCS kbookmarkmodeltest.cpp)
 add_executable(kbookmarkmodeltest ${kbookmarkmodeltest_SRCS})
 add_test(kbookmarkmodeltest kbookmarkmodeltest)
 ecm_mark_as_test(kbookmarkmodeltest)
-target_link_libraries(kbookmarkmodeltest kbookmarkmodel_private
+target_link_libraries(kbookmarkmodeltest kbookmarkmodel_private5
    KF5::Bookmarks
    Qt5::Test
 )
diff --git a/konqueror/sidebar/CMakeLists.txt b/konqueror/sidebar/CMakeLists.txt
index c021667..91d4172 100644
--- a/konqueror/sidebar/CMakeLists.txt
+++ b/konqueror/sidebar/CMakeLists.txt
@@ -12,15 +12,15 @@ add_subdirectory( places_module )
 add_subdirectory( default_entries )
 add_subdirectory( test )
 
-########### konqsidebarplugin lib: contains the base class for plugins ###############
+########### konqsidebarplugin5 lib: contains the base class for plugins ###############
 
 set(konqsidebarplugin_LIB_SRCS konqsidebarplugin.cpp )
 
-add_library(konqsidebarplugin SHARED ${konqsidebarplugin_LIB_SRCS})
+add_library(konqsidebarplugin5 SHARED ${konqsidebarplugin_LIB_SRCS})
 
-target_link_libraries(konqsidebarplugin KF5::Parts KF5::KDELibs4Support KF5::KIOFileWidgets)
-set_target_properties(konqsidebarplugin PROPERTIES VERSION ${KONQUEROR_LIB_VERSION} SOVERSION "5" )
-install(TARGETS konqsidebarplugin ${KDE_INSTALL_TARGETS_DEFAULT_ARGS} )
+target_link_libraries(konqsidebarplugin5 KF5::Parts KF5::KDELibs4Support KF5::KIOFileWidgets)
+set_target_properties(konqsidebarplugin5 PROPERTIES VERSION ${KONQUEROR_LIB_VERSION} SOVERSION "5" )
+install(TARGETS konqsidebarplugin5 ${KDE_INSTALL_TARGETS_DEFAULT_ARGS} )
 
 ########### konq_sidebar plugin: contains the KParts used by konqueror ###############
 
@@ -33,7 +33,7 @@ set(konq_sidebar_PART_SRCS
 
 add_library(konq_sidebar MODULE ${konq_sidebar_PART_SRCS})
 
-target_link_libraries(konq_sidebar KF5::Parts KF5::Konq konqsidebarplugin KF5::KCMUtils KF5::KDELibs4Support )
+target_link_libraries(konq_sidebar KF5::Parts KF5::Konq konqsidebarplugin5 KF5::KCMUtils KF5::KDELibs4Support )
 
 install(TARGETS konq_sidebar DESTINATION ${KDE_INSTALL_PLUGINDIR} )
 
diff --git a/konqueror/sidebar/history_module/CMakeLists.txt b/konqueror/sidebar/history_module/CMakeLists.txt
index 9d627a2..ca6f757 100644
--- a/konqueror/sidebar/history_module/CMakeLists.txt
+++ b/konqueror/sidebar/history_module/CMakeLists.txt
@@ -6,7 +6,7 @@ include_directories(
 ########### the sidebar module ###############
 
 add_library(konqsidebar_history MODULE history_module.cpp)
-target_link_libraries(konqsidebar_history konqsidebarplugin konquerorprivate KF5::Konq KF5::Parts)
+target_link_libraries(konqsidebar_history konqsidebarplugin5 konquerorprivate KF5::Konq KF5::Parts)
 install(TARGETS konqsidebar_history DESTINATION ${KDE_INSTALL_PLUGINDIR})
 install(FILES konqsidebar_history.desktop DESTINATION ${KDE_INSTALL_DATADIR}/konqsidebartng/plugins)
 
diff --git a/konqueror/sidebar/places_module/CMakeLists.txt b/konqueror/sidebar/places_module/CMakeLists.txt
index 14d7472..c32d5c5 100644
--- a/konqueror/sidebar/places_module/CMakeLists.txt
+++ b/konqueror/sidebar/places_module/CMakeLists.txt
@@ -6,7 +6,7 @@ set(konqsidebar_places_SRCS
 
 add_library(konqsidebar_places MODULE ${konqsidebar_places_SRCS})
 
-target_link_libraries(konqsidebar_places  KF5::Parts konqsidebarplugin)
+target_link_libraries(konqsidebar_places  KF5::Parts konqsidebarplugin5)
 
 install(TARGETS konqsidebar_places DESTINATION ${KDE_INSTALL_PLUGINDIR})
 
diff --git a/konqueror/sidebar/test/CMakeLists.txt b/konqueror/sidebar/test/CMakeLists.txt
index 38b0f36..36117a6 100644
--- a/konqueror/sidebar/test/CMakeLists.txt
+++ b/konqueror/sidebar/test/CMakeLists.txt
@@ -6,7 +6,7 @@ include(ECMAddTests)
 
 set(konq_sidebartest_PART_SRCS konqsidebartest.cpp )
 add_library(konq_sidebartest MODULE ${konq_sidebartest_PART_SRCS})
-target_link_libraries(konq_sidebartest KF5::Parts konqsidebarplugin )
+target_link_libraries(konq_sidebartest KF5::Parts konqsidebarplugin5 )
 #install(TARGETS konq_sidebartest  DESTINATION ${KDE_INSTALL_PLUGINDIR} )
 #install( FILES test.desktop  DESTINATION  ${KDE_INSTALL_DATADIR}/konqsidebartng/entries )
 
diff --git a/konqueror/sidebar/trees/CMakeLists.txt b/konqueror/sidebar/trees/CMakeLists.txt
index 12baabc..15232b8 100644
--- a/konqueror/sidebar/trees/CMakeLists.txt
+++ b/konqueror/sidebar/trees/CMakeLists.txt
@@ -17,7 +17,7 @@ set(konqsidebar_tree_PART_SRCS konqsidebar_oldtreemodule.cpp ${libkonq_sidebar_t
 
 add_library(konqsidebar_tree MODULE ${konqsidebar_tree_PART_SRCS})
 
-target_link_libraries(konqsidebar_tree ${KDE4_KDE3SUPPORT_LIBS} KF5::Parts  KF5::Konq konqsidebarplugin )
+target_link_libraries(konqsidebar_tree ${KDE4_KDE3SUPPORT_LIBS} KF5::Parts  KF5::Konq konqsidebarplugin5 )
 
 install(TARGETS konqsidebar_tree DESTINATION ${KDE_INSTALL_PLUGINDIR} )
 
diff --git a/konqueror/sidebar/trees/bookmark_module/CMakeLists.txt b/konqueror/sidebar/trees/bookmark_module/CMakeLists.txt
index 740f651..814be36 100644
--- a/konqueror/sidebar/trees/bookmark_module/CMakeLists.txt
+++ b/konqueror/sidebar/trees/bookmark_module/CMakeLists.txt
@@ -8,7 +8,7 @@ set(konq_sidebartree_bookmarks_SRCS bookmark_module.cpp bookmark_item.cpp ${libk
 
 add_library(konq_sidebartree_bookmarks MODULE ${konq_sidebartree_bookmarks_SRCS})
 
-target_link_libraries(konq_sidebartree_bookmarks   KF5::Parts konqsidebarplugin konq ${KDE4_KDE3SUPPORT_LIBS})
+target_link_libraries(konq_sidebartree_bookmarks   KF5::Parts konqsidebarplugin5 konq ${KDE4_KDE3SUPPORT_LIBS})
 
 install(TARGETS konq_sidebartree_bookmarks  DESTINATION ${KDE_INSTALL_PLUGINDIR} )
 
diff --git a/konqueror/sidebar/trees/dirtree_module/CMakeLists.txt b/konqueror/sidebar/trees/dirtree_module/CMakeLists.txt
index f473b5b..c1be918 100644
--- a/konqueror/sidebar/trees/dirtree_module/CMakeLists.txt
+++ b/konqueror/sidebar/trees/dirtree_module/CMakeLists.txt
@@ -10,7 +10,7 @@ set(konq_sidebartree_dirtree_SRCS dirtree_module.cpp dirtree_item.cpp ${libkonq_
 
 add_library(konq_sidebartree_dirtree MODULE ${konq_sidebartree_dirtree_SRCS})
 
-target_link_libraries(konq_sidebartree_dirtree  ${KDE4_KDE3SUPPORT_LIBS} KF5::Parts konq konqsidebarplugin  )
+target_link_libraries(konq_sidebartree_dirtree  ${KDE4_KDE3SUPPORT_LIBS} KF5::Parts konq konqsidebarplugin5  )
 
 install(TARGETS konq_sidebartree_dirtree  DESTINATION ${KDE_INSTALL_PLUGINDIR} )
 
diff --git a/konqueror/sidebar/web_module/CMakeLists.txt b/konqueror/sidebar/web_module/CMakeLists.txt
index 414109c..68d2320 100644
--- a/konqueror/sidebar/web_module/CMakeLists.txt
+++ b/konqueror/sidebar/web_module/CMakeLists.txt
@@ -6,7 +6,7 @@ add_library(konqsidebar_web MODULE ${konqsidebar_web_PART_SRCS})
 
 target_link_libraries(konqsidebar_web
    KF5::KHtml
-   konqsidebarplugin
+   konqsidebarplugin5
    KF5::Konq
    KF5::KIOGui)
 
