diff --git a/konqueror/src/konqmainwindow.cpp b/konqueror/src/konqmainwindow.cpp
index 5f53e0e..12b0a6a 100644
--- a/konqueror/src/konqmainwindow.cpp
+++ b/konqueror/src/konqmainwindow.cpp
@@ -142,7 +142,9 @@
 #include <kacceleratormanager.h>
 #include <kuser.h>
 #include <kxmlguifactory.h>
+#if defined(KONQ_HAVE_X11) && KONQ_HAVE_X11
 #include <netwm.h>
+#endif
 #include <sonnet/configdialog.h>
 
 #include <kauthorized.h>

