diff --git orig.meson.build meson.build
index 771c79c..81f52a6 100644
--- orig.meson.build
+++ meson.build
@@ -28,7 +28,7 @@ glib_req           = '>= @0@.@1@.@2@'.format(glib_major_req, glib_minor_req, gli
 pango_req          = '>= 1.41.0'
 fribidi_req        = '>= 0.19.7'
 atk_req            = '>= 2.35.1'
-at_spi2_atk_req    = '>= 2.15.1'
+at_spi2_atk_req    = '>= 2.14.1'
 cairo_req          = '>= 1.14.0'
 gdk_pixbuf_req     = '>= 2.30.0'
 introspection_req  = '>= 1.39.0'
