diff --git gdk/wayland/orig.gdkdevice-wayland.c gdk/wayland/gdkdevice-wayland.c
index 9dc86a3..514b5c3 100644
--- gdk/wayland/orig.gdkdevice-wayland.c
+++ gdk/wayland/gdkdevice-wayland.c
@@ -44,6 +44,12 @@
 #include <dev/evdev/input.h>
 #elif defined(HAVE_LINUX_INPUT_H)
 #include <linux/input.h>
+#else
+#define BTN_LEFT		0x110
+#define BTN_RIGHT		0x111
+#define BTN_MIDDLE		0x112
+#define BTN_STYLUS		0x14b
+#define BTN_STYLUS2		0x14c
 #endif
 
 #define BUTTON_BASE (BTN_LEFT - 1) /* Used to translate to 1-indexed buttons */
