diff --git gdk/wayland/orig.gdkdisplay-wayland.c gdk/wayland/gdkdisplay-wayland.c
index 42a6fb8..2c0a725 100644
--- gdk/wayland/orig.gdkdisplay-wayland.c
+++ gdk/wayland/gdkdisplay-wayland.c
@@ -597,12 +597,12 @@ _gdk_wayland_display_open (const gchar *display_name)
 
   GDK_NOTE (MISC, g_message ("opening display %s", display_name ? display_name : ""));
 
-  /* If this variable is unset then wayland initialisation will surely
-   * fail, logging a fatal error in the process.  Save ourselves from
-   * that.
-   */
-  if (g_getenv ("XDG_RUNTIME_DIR") == NULL)
-    return NULL;
+//   /* If this variable is unset then wayland initialisation will surely
+//    * fail, logging a fatal error in the process.  Save ourselves from
+//    * that.
+//    */
+//   if (g_getenv ("XDG_RUNTIME_DIR") == NULL)
+//     return NULL;
 
   wl_log_set_handler_client (log_handler);
 
