diff --git gtk/orig.gtkfilechoosernative.c gtk/gtkfilechoosernative.c
index 76f7cb6..e7759cc 100644
--- gtk/orig.gtkfilechoosernative.c
+++ gtk/gtkfilechoosernative.c
@@ -735,10 +735,11 @@ gtk_file_chooser_native_get_files (GtkFileChooser *chooser)
 
   switch (self->mode)
     {
+    case MODE_QUARTZ:
+      if (GDK_IS_QUARTZ_DISPLAY(gdk_display_get_default()))
     case MODE_PORTAL:
     case MODE_WIN32:
-    case MODE_QUARTZ:
-      return g_slist_copy_deep (self->custom_files, (GCopyFunc)g_object_ref, NULL);
+        return g_slist_copy_deep (self->custom_files, (GCopyFunc)g_object_ref, NULL);
 
     case MODE_FALLBACK:
     default:
@@ -761,6 +762,7 @@ gtk_file_chooser_native_show (GtkNativeDialog *native)
 #if defined (GDK_WINDOWING_QUARTZ) && \
   MAC_OS_X_VERSION_MAX_ALLOWED >= 1060
     if (gdk_quartz_osx_version() >= GDK_OSX_SNOW_LEOPARD &&
+        GDK_IS_QUARTZ_DISPLAY(gdk_display_get_default()) &&
         gtk_file_chooser_native_quartz_show (self))
     self->mode = MODE_QUARTZ;
 #endif
@@ -791,7 +793,8 @@ gtk_file_chooser_native_hide (GtkNativeDialog *native)
     case MODE_QUARTZ:
 #if defined (GDK_WINDOWING_QUARTZ) && \
   MAC_OS_X_VERSION_MAX_ALLOWED >= 1060
-      if (gdk_quartz_osx_version() >= GDK_OSX_SNOW_LEOPARD)
+      if (gdk_quartz_osx_version() >= GDK_OSX_SNOW_LEOPARD &&
+        GDK_IS_QUARTZ_DISPLAY(gdk_display_get_default()))
         gtk_file_chooser_native_quartz_hide (self);
 #endif
       break;
