diff --git orig.Makefile Makefile
index af9701c..f9aaf79 100644
--- orig.Makefile
+++ Makefile
@@ -64,9 +64,10 @@ endif
 $(info Build target is $(TARGET))
 
 # Default/Linux settings....
+prefix=__PREFIX__
 STRIP?=strip
 #CXXFLAGS+=-O2 -Wall -D_FILE_OFFSET_BITS=64 -D USE_UTF16
-CXXFLAGS+=-O2 -Wall -D_FILE_OFFSET_BITS=64
+CXXFLAGS+=-Wall -D_FILE_OFFSET_BITS=64 -I$(prefix)/include
 LDFLAGS+=
 LDLIBS+=-luuid #-licuio -licuuc
 FATBINFLAGS=
