diff --git a/src/xdg-desktop-portal-kde.cpp b/src/xdg-desktop-portal-kde.cpp
index 273fb86..9059e3c 100644
--- a/src/xdg-desktop-portal-kde.cpp
+++ b/src/xdg-desktop-portal-kde.cpp
@@ -28,6 +32,18 @@ Q_LOGGING_CATEGORY(XdgDesktopPortalKde, "xdp-kde")
 
 int main(int argc, char *argv[])
 {
+#ifdef __APPLE__
+    if (!qEnvironmentVariableIsSet("DISPLAY") || qEnvironmentVariableIsEmpty("DISPLAY")) {
+        QProcess proc;
+        QStringList args = {QStringLiteral("getenv"), QStringLiteral("DISPLAY")};
+        proc.start(QStringLiteral("launchctl"), args);
+        if (proc.waitForFinished()) {
+            qputenv("DISPLAY", proc.readAllStandardOutput().simplified());
+            qCWarning(XdgDesktopPortalKde) << "Set DISPLAY = " << qgetenv("DISPLAY");
+        }
+    }
+#endif
+
     QCoreApplication::setAttribute(Qt::AA_UseHighDpiPixmaps);
     QApplication a(argc, argv);
     a.setApplicationName(QStringLiteral("xdg-desktop-portal-kde"));
