diff --git a/src/xdg-desktop-portal-kde.cpp b/src/xdg-desktop-portal-kde.cpp
index 273fb86..3ab6682 100644
--- a/src/xdg-desktop-portal-kde.cpp
+++ b/src/xdg-desktop-portal-kde.cpp
@@ -21,6 +21,7 @@
 #include <QApplication>
 #include <QDBusConnection>
 #include <QLoggingCategory>
+#include <QDir>
 
 #include "desktopportal.h"
 
@@ -47,5 +48,9 @@ int main(int argc, char *argv[])
         return 1;
     }
 
+    if (QDir::currentPath() == QStringLiteral("/")) {
+        QDir::setCurrent(QDir::homePath());
+    }
+
     return a.exec();
 }
