diff --git a/src/notification.cpp b/src/notification.cpp
index 9b8606e..017c534 100644
--- a/src/notification.cpp
+++ b/src/notification.cpp
@@ -166,11 +166,12 @@ void NotificationPortal::RemoveNotification(const QString &app_id,
     qCDebug(XdgDesktopPortalKdeNotification) << "    app_id: " << app_id;
     qCDebug(XdgDesktopPortalKdeNotification) << "    id: " << id;
 
-    KNotification *notify = m_notifications.take(QStringLiteral("%1:%2").arg(app_id, id));
-    if (notify) {
-        notify->close();
-        notify->deleteLater();
-    }
+//     KNotification *notify = m_notifications.take(QStringLiteral("%1:%2").arg(app_id, id));
+//     if (notify) {
+//         notify->close();
+//         notify->deleteLater();
+//     }
+    m_notifications.remove(QStringLiteral("%1:%2").arg(app_id, id));
 }
 
 void NotificationPortal::notificationClosed()
