diff --git a/src/screenshotdialog.cpp b/src/screenshotdialog.cpp
index 7edb027..4d9ea7e 100644
--- a/src/screenshotdialog.cpp
+++ b/src/screenshotdialog.cpp
@@ -35,6 +35,7 @@
 
 #include <fcntl.h>
 #include <unistd.h>
+#include <errno.h>
 
 Q_LOGGING_CATEGORY(XdgDesktopPortalKdeScreenshotDialog, "xdp-kde-screenshot-dialog")
 
@@ -99,6 +100,29 @@ ScreenshotDialog::~ScreenshotDialog()
 {
     delete m_dialog;
 }
+
+#ifndef linux
+int pipe2(int *fd, int flags)
+{
+	errno = 0;
+	if (pipe(fd) < 0) {
+		return -1;
+	}
+	for (int i = 0 ; i < 2 ; ++i) {
+		errno = 0;
+		int cflags = fcntl(fd[i], F_GETFD, 0);
+		if (cflags < 0) {
+			return cflags;
+		}
+		errno = 0;
+		if (fcntl(fd[i], F_SETFD, cflags | flags) < 0) {
+			return -1;
+		}
+	}
+	return 0;
+}
+#endif
+
 void ScreenshotDialog::takeScreenshot()
 {
     int pipeFds[2];
