diff --git a/include/os/macos/spl/sys/mod_os.h b/include/os/macos/spl/sys/mod_os.h
index 8ed35f7993e378ceba1a26be681a5d714e609e15..c826b4800b8cc03535e3f518a5d018164fbe45bc 100644
--- a/include/os/macos/spl/sys/mod_os.h
+++ b/include/os/macos/spl/sys/mod_os.h
@@ -89,6 +89,14 @@ extern "C" {
     SYSCTL_LONG(parent, nbr, name, access, ptr, descr)
 #define	ZSYSCTL_ULONG(parent, nbr, name, access, ptr, val, descr) \
     SYSCTL_ULONG(parent, nbr, name, access, ptr, descr)
+#ifndef SYSCTL_ULONG
+/* Oid for a unsigned long.  The pointer must be non NULL. */
+#define SYSCTL_ULONG(parent, nbr, name, access, ptr, descr) \
+     SYSCTL_OID(parent, nbr, name, CTLTYPE_INT|access, \
+             ptr, 0, sysctl_handle_long, "LU", descr); \
+     typedef char _sysctl_##parent##_##name##_size_check[(__builtin_constant_p(ptr) || sizeof(*(ptr)) == sizeof(unsigned long)) ? 0 : -1]
+#endif
+
 /*
  * Appears to be no default for 64bit values in Linux, if
  * ZOL adds it using STANDARD_PARAM_DEF let us guess
