--- llvm-3.8.1.src/tools/lldb/packages/Python/lldbsuite/test/orig.lldbtest.py	2016-02-02 17:50:06.000000000 +0100
+++ llvm-3.8.1.src/tools/lldb/packages/Python/lldbsuite/test/lldbtest.py	2016-09-01 17:03:30.000000000 +0200
@@ -2131,7 +2131,7 @@
 
     def signBinary(self, binary_path):
         if sys.platform.startswith("darwin"):
-            codesign_cmd = "codesign --force --sign lldb_codesign %s" % (binary_path)
+            codesign_cmd = "echo Skipping codesign --force --sign lldb_codesign %s" % (binary_path)
             call(codesign_cmd, shell=True)
 
     def findBuiltClang(self):
--- llvm-3.9.0.src/tools/lldb/tools/debugserver/source/MacOSX/CMakeLists.txt.orig	2016-05-26 10:38:10.000000000 +0200
+++ llvm-3.9.0.src/tools/lldb/tools/debugserver/source/MacOSX/CMakeLists.txt	2016-09-07 22:07:21.000000000 +0200
@@ -59,7 +59,7 @@
 
 target_link_libraries(debugserver ${DEBUGSERVER_USED_LIBS})
 
-set(LLDB_CODESIGN_IDENTITY "lldb_codesign"
+set(LLDB_CODESIGN_IDENTITY ""
   CACHE STRING "Identity used for code signing. Set to empty string to skip the signing step.")
 if (NOT ("${LLDB_CODESIGN_IDENTITY}" STREQUAL ""))
   execute_process(
