diff --git plugins/grepview/grepfindthread.cpp plugins/grepview/grepfindthread.cpp
index 9388d5c..c0b5fe2 100644
--- plugins/grepview/grepfindthread.cpp
+++ plugins/grepview/grepfindthread.cpp
@@ -47,6 +47,8 @@ static QList<QUrl> thread_getProjectFiles(const QUrl& dir, int depth, const QStr
     if(!project)
         return res;
 
+    QString dirName = QFileInfo(QDir(dir.toLocalFile()).canonicalPath()).absoluteDir().path();
+
     const QSet<IndexedString> fileSet = project->fileSet();
     for (const IndexedString& item : fileSet) {
         if(abort)
@@ -69,7 +71,7 @@ static QList<QUrl> thread_getProjectFiles(const QUrl& dir, int depth, const QStr
                     continue;
             }
         }
-        if( QDir::match(include, url.fileName()) && !WildcardHelpers::match(exlude, url.toLocalFile()) )
+        if( QDir::match(include, url.fileName()) && !WildcardHelpers::match(exlude, url.toLocalFile().remove(dirName)) )
             res << url;
     }
 
@@ -80,6 +82,7 @@ static QList<QUrl> thread_findFiles(const QDir& dir, int depth, const QStringLis
                                    const QStringList& exclude, volatile bool &abort)
 {
     QFileInfoList infos = dir.entryInfoList(include, QDir::NoDotAndDotDot|QDir::Files|QDir::Readable|QDir::Hidden);
+    QString dirName = QFileInfo(dir.canonicalPath()).absoluteDir().path();
 
     if(!QFileInfo(dir.path()).isDir())
         infos << QFileInfo(dir.path());
@@ -87,7 +90,7 @@ static QList<QUrl> thread_findFiles(const QDir& dir, int depth, const QStringLis
     QList<QUrl> dirFiles;
     for (const QFileInfo& currFile : qAsConst(infos)) {
         QString currName = currFile.canonicalFilePath();
-        if(!WildcardHelpers::match(exclude, currName))
+        if(!WildcardHelpers::match(exclude, currName.remove(dirName)))
             dirFiles << QUrl::fromLocalFile(currName);
     }
     if(depth != 0)
