diff --git a/configure b/configure
index e3b032b..a042ae1 100755
--- a/configure
+++ b/configure
@@ -746,22 +746,8 @@ fi
 # Check for AMD64 hardware support.
 if [ x$TGT_ARCH = "xx86_64" -a $(uname -m) = "x86_64" ] ; then
  
-  # Check for PCLMUL support
-cat > $test.c << EOF
-void foo(void) {
-#ifndef __PCLMUL__
-    #error no pclmul
-#endif
-}
-EOF
-
-  if try $CC -c $CFLAGS $test.c ; then
-    CFLAGS="-DHAS_PCLMUL $CFLAGS"
-    SFLAGS="-DHAS_PCLMUL $SFLAGS"
-    echo "Checking for PCLMUL support ... Yes" | tee -a configure.log
-  else
-    echo "Checking for PCLMUL support ... No" | tee -a configure.log
-  fi
+    CFLAGS="-UHAS_PCLMUL $CFLAGS"
+    SFLAGS="-UHAS_PCLMUL $SFLAGS"
 
   # Check for SSE4.2 support. Not all compiler set __SSE4_2__ 
 cat > $test.c << EOF
