diff --git a/CMakeLists.txt b/CMakeLists.txt
index 098f888..a361bfd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -4,7 +4,7 @@ cmake_minimum_required(VERSION 3.1.0 FATAL_ERROR)
 # The cmake_minimum_required(VERSION) command implicitly invokes the cmake_policy(VERSION)
 # command to specify that the current project code is written for the given range of CMake
 # versions.
-project(pavucontrol-qt)
+project(pavucontrol-qt VERSION @VERSION@)
 
 option(UPDATE_TRANSLATIONS "Update source translation translations/*.ts files" OFF)
 
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 7dd356c..e8057ae 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -71,6 +71,17 @@ set_property(
      PROPERTY COMPILE_DEFINITIONS
      PAVUCONTROL_QT_DATA_DIR="${CMAKE_INSTALL_PREFIX}/share/${PROJECT_NAME}"
 )
+set_target_properties(pavucontrol-qt PROPERTIES
+    MACOSX_BUNDLE TRUE
+    MACOSX_BUNDLE_BUNDLE_NAME ${PROJECT_NAME}
+    MACOSX_BUNDLE_GUI_IDENTIFIER "org.PulseAudio.pavucontrol"
+    MACOSX_BUNDLE_BUNDLE_VERSION ${PROJECT_VERSION}
+    MACOSX_BUNDLE_LONG_VERSION_STRING ${PROJECT_VERSION}
+    MACOSX_BUNDLE_SHORT_VERSION_STRING "${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}"
+    MACOSX_BUNDLE_ICON_FILE "multimedia-volume-control.icns"
+    MACOSX_BUNDLE_COPYRIGHT "LGPL-2.1"
+    RESOURCE "resources/multimedia-volume-control.icns"
+)
 
 target_link_libraries(pavucontrol-qt
     Qt5::Widgets
@@ -82,6 +93,7 @@ install(TARGETS
     pavucontrol-qt
     RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
     COMPONENT Runtime
+    BUNDLE DESTINATION "${BUNDLE_INSTALL_DIR}"
 )
 
 install(FILES
