diff --git a/CMakeLists.txt b/CMakeLists.txt
index 098f8886167150b622e8cfbe37b501b39a2e870b..88f058f361113229e92c9153f8e73159450b821e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -16,11 +16,12 @@ set(CMAKE_AUTOMOC ON)
 set(CMAKE_AUTOUIC ON)
 
 include(GNUInstallDirs)
+include(FeatureSummary)
 
 # Minimum Versions
 set(GLIB_MINIMUM_VERSION "2.50.0")
 set(LXQTBT_MINIMUM_VERSION "0.13.0")
-set(QT_MINIMUM_VERSION "5.15.0")
+set(QT_MINIMUM_VERSION "5.6.3")
 
 find_package(Qt5Widgets ${QT_MINIMUM_VERSION} REQUIRED)
 find_package(Qt5LinguistTools ${QT_MINIMUM_VERSION} REQUIRED)
@@ -46,3 +47,5 @@ pkg_check_modules(
 )
 
 add_subdirectory(src)
+
+feature_summary(WHAT ALL   FATAL_ON_MISSING_REQUIRED_PACKAGES)
