diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 7dd356c..7f08155 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -35,6 +35,13 @@ set(pavucontrol-qt_SRCS
     elidinglabel.cc
 )
 
+if (APPLE)
+	set(pavucontrol-qt_SRCS ${pavucontrol-qt_SRCS}
+		resources/multimedia-volume-control.icns
+	)
+	set_source_files_properties(resources/multimedia-volume-control.icns PROPERTIES MACOSX_PACKAGE_LOCATION Resources)
+endif()
+
 set(pavucontrol-qt_UI
     cardwidget.ui
     devicewidget.ui
@@ -72,6 +86,18 @@ set_property(
      PAVUCONTROL_QT_DATA_DIR="${CMAKE_INSTALL_PREFIX}/share/${PROJECT_NAME}"
 )
 
+set_target_properties(pavucontrol-qt PROPERTIES
+    MACOSX_BUNDLE TRUE
+    MACOSX_BUNDLE_BUNDLE_NAME ${PROJECT_NAME}
+    MACOSX_BUNDLE_GUI_IDENTIFIER "org.PulseAudio.pavucontrol"
+    MACOSX_BUNDLE_BUNDLE_VERSION ${PROJECT_VERSION}
+    MACOSX_BUNDLE_LONG_VERSION_STRING ${PROJECT_VERSION}
+    MACOSX_BUNDLE_SHORT_VERSION_STRING "${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}"
+    MACOSX_BUNDLE_ICON_FILE "multimedia-volume-control.icns"
+    MACOSX_BUNDLE_COPYRIGHT "LGPL-2.1"
+    RESOURCE "resources/multimedia-volume-control.icns"
+)
+
 target_link_libraries(pavucontrol-qt
     Qt5::Widgets
     ${PULSE_LDFLAGS}
@@ -81,6 +107,7 @@ target_link_libraries(pavucontrol-qt
 install(TARGETS
     pavucontrol-qt
     RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
+    BUNDLE DESTINATION "${BUNDLE_INSTALL_DIR}"
     COMPONENT Runtime
 )
 
