diff --git a/src/pavucontrol.cc b/src/pavucontrol.cc
index 53873c6..29e951a 100644
--- a/src/pavucontrol.cc
+++ b/src/pavucontrol.cc
@@ -50,6 +50,7 @@
 #include <QCommandLineParser>
 #include <QCommandLineOption>
 #include <QString>
+#include <QDebug>
 
 static pa_context* context = nullptr;
 static pa_mainloop_api* api = nullptr;
@@ -581,7 +582,7 @@ void context_state_callback(pa_context *c, void *userdata) {
             context = nullptr;
 
             if (reconnect_timeout > 0) {
-                g_debug("%s", QObject::tr("Connection failed, attempting reconnect").toUtf8().constData());
+                qWarning() << QObject::tr("Connection failed, attempting reconnect").toUtf8().constData();
                 g_timeout_add_seconds(reconnect_timeout, connect_to_pulse, w);
             }
             return;
@@ -617,7 +618,12 @@ gboolean connect_to_pulse(gpointer userdata) {
     pa_context_set_state_callback(context, context_state_callback, w);
 
     w->setConnectingMessage();
-    if (pa_context_connect(context, nullptr, PA_CONTEXT_NOFAIL, nullptr) < 0) {
+#ifdef __APPLE__
+    pa_context_flags_t connectFlags = PA_CONTEXT_NOAUTOSPAWN;
+#else
+    pa_context_flags_t connectFlags = pa_context_flags_t(PA_CONTEXT_NOFAIL|PA_CONTEXT_NOAUTOSPAWN);
+#endif
+    if (pa_context_connect(context, nullptr, connectFlags, nullptr) < 0) {
         if (pa_context_errno(context) == PA_ERR_INVALID) {
             w->setConnectingMessage(QObject::tr("Connection to PulseAudio failed. Automatic retry in 5s.<br><br>"
                 "In this case this is likely because PULSE_SERVER in the Environment/X11 Root Window Properties"
@@ -637,6 +643,19 @@ gboolean connect_to_pulse(gpointer userdata) {
                 g_timeout_add_seconds(reconnect_timeout, connect_to_pulse, userdata);
             }
         }
+    } else {
+        // pump the GLib context until we've connected, before entering the Qt main loop.
+        // Not necessary under X11, but the order of events must be different enough under
+        // Apple's Cocoa QPA that the connection fails (events get lost maybe?)
+        auto gContext = g_main_context_default();
+        for (;;)
+        {
+            pa_context_state_t state = pa_context_get_state(context);
+//             qWarning() << Q_FUNC_INFO << "state=" << state;
+            if (state == PA_CONTEXT_READY || !PA_CONTEXT_IS_GOOD(state))
+                break;
+            g_main_context_iteration(gContext, true);
+        }
     }
 
     return false;
