diff --git a/src/devicewidget.ui b/src/devicewidget.ui
index 3b5f781..5b8ca1e 100644
--- a/src/devicewidget.ui
+++ b/src/devicewidget.ui
@@ -244,6 +244,12 @@
            <property name="suffix">
             <string> ms</string>
            </property>
+           <property name="minimum">
+            <number>-2000</number>
+           </property>
+           <property name="maximum">
+            <number>5000</number>
+           </property>
           </widget>
          </item>
         </layout>
