diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2f1d23d..2af7d37 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -15,6 +15,7 @@ set(CMAKE_MODULE_PATH ${ECM_MODULE_PATH} ${ECM_KDE_MODULE_DIR})
 include(ECMAddTests)
 include(ECMPackageConfigHelpers)
 include(ECMSetupVersion)
+include(ECMMarkNonGuiExecutable)
 
 include(KDEInstallDirs)
 include(KDECMakeSettings)
diff --git a/kdev-pg/CMakeLists.txt b/kdev-pg/CMakeLists.txt
index 5490835..665480e 100644
--- a/kdev-pg/CMakeLists.txt
+++ b/kdev-pg/CMakeLists.txt
@@ -93,9 +93,10 @@ ENDIF(FLEX_EXECUTABLE)
 
 
 # tell CMake to generate the executable
-ADD_EXECUTABLE(kdev-pg-qt ${parser_srcs} ${kdevpg_srcs} ${kdevpg_rcc_srcs})
-TARGET_LINK_LIBRARIES( kdev-pg-qt Qt5::Core)
-INSTALL(TARGETS kdev-pg-qt DESTINATION bin )
+ADD_EXECUTABLE(kdev-pg-qt ${parser_srcs} ${kdevpg_srcs} ${kdevpg_rcc_srcs})
+ECM_MARK_NONGUI_EXECUTABLE(kdev-pg-qt)
+TARGET_LINK_LIBRARIES( kdev-pg-qt Qt5::Core)
+INSTALL(TARGETS kdev-pg-qt DESTINATION bin )
 
 # in order to generate only the parser, call "make parser"
 ADD_CUSTOM_TARGET(parser echo "Generating the parser"
