--- src/orig.CMakeLists.txt	2016-11-03 22:49:10.000000000 +0100
+++ src/CMakeLists.txt	2016-12-10 13:29:14.000000000 +0100
@@ -1,5 +1,7 @@
 project(lokalize)
 
+include(ECMAddAppIcon)
+
 if(NOT WIN32)
     find_package(HUNSPELL REQUIRED)
 else(NOT WIN32)
@@ -154,7 +156,8 @@
 
 
 
-#kde4_add_app_icon(lokalize_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/../icons/hi*-app-lokalize.png")
+file(GLOB ICONS_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/../icons/global/*-apps-lokalize.png")
+ecm_add_app_icon(lokalize_SRCS ICONS ${ICONS_SRCS})
 
 add_executable(lokalize ${lokalize_SRCS})
 
