--- src/nokde-stubs/orig.prefs.cpp	2016-11-03 22:49:10.000000000 +0100
+++ src/nokde-stubs/prefs.cpp	2016-12-10 13:12:22.000000000 +0100
@@ -63,6 +63,7 @@
  , mWordCompletionLength(3)
  , mSuggCount(10)
 {
+    QSettings::setDefaultFormat(QSettings::IniFormat);
     QSettings s;
     mAuthorName = s.value(QStringLiteral("Author/Name"), QString()).toString();
     if (mAuthorName.isEmpty()) {mAuthorName = fullUserName(); if (mAuthorName.length()) mAuthorName[0]=mAuthorName.at(0).toUpper();}
