diff --git a/libcxx/src/CMakeLists.txt b/libcxx/src/CMakeLists.txt
index 9ff2a62..0132517 100644
--- a/libcxx/src/CMakeLists.txt
+++ b/libcxx/src/CMakeLists.txt
@@ -334,6 +334,7 @@ set(LIBCXX_EXPERIMENTAL_SOURCES
   )
 
 add_library(cxx_experimental STATIC ${LIBCXX_EXPERIMENTAL_SOURCES})
+target_compile_options(cxx_experimental PRIVATE "-fno-lto")
 target_link_libraries(cxx_experimental PUBLIC cxx-headers)
 if (LIBCXX_ENABLE_SHARED)
   target_link_libraries(cxx_experimental PRIVATE cxx_shared)
