diff --git a/libcxxabi/src/CMakeLists.txt b/libcxxabi/src/CMakeLists.txt
index 54002baf..206e1c1e 100644
--- a/libcxxabi/src/CMakeLists.txt
+++ b/libcxxabi/src/CMakeLists.txt
@@ -224,9 +224,12 @@ if (LIBCXXABI_ENABLE_SHARED)
 
     export_symbols("${CMAKE_CURRENT_SOURCE_DIR}/../lib/itanium-base.exp")
 
-    if (LIBCXXABI_ENABLE_NEW_DELETE_DEFINITIONS)
-      reexport_symbols("${CMAKE_CURRENT_SOURCE_DIR}/../lib/new-delete.exp")
-    endif()
+## ld64 from port:ld64-latest @450.3 refuses to reexport symbols that are not in a dylib
+## but instead in an object module that's being linked in. Pedantic but not incorrect,
+## the question is why new-delete.exp is being used on any system...
+#     if (LIBCXXABI_ENABLE_NEW_DELETE_DEFINITIONS)
+#       reexport_symbols("${CMAKE_CURRENT_SOURCE_DIR}/../lib/new-delete.exp")
+#     endif()
 
     if (LIBCXXABI_ENABLE_EXCEPTIONS)
       reexport_symbols("${CMAKE_CURRENT_SOURCE_DIR}/../lib/exceptions.exp")
