diff --git a/libcxx/lib/libc++unexp.exp b/libcxx/lib/libc++unexp.exp
index 9507fc57..5b025d0a 100644
--- a/libcxx/lib/libc++unexp.exp
+++ b/libcxx/lib/libc++unexp.exp
@@ -17,3 +17,42 @@ __ZTSNSt3__1[0-9][0-9]*__*
 *__hidden_allocator*
 # anything using __sso_allocator
 *__sso_allocator*
+# RJVB
+# don't export symbols we obtain from libMacportsLegacySupport.a modules
+# UNLESS they mask (fix) symbols from a system library
+# NB: not all modules are pulled in for an obvious reason!
+# atcalls.o (for openat(), unlinkat())
+_faccessat
+_fchmodat
+_fchownat
+_getattrlistat
+_linkat
+_mkdirat
+_openat
+_openat$NOCANCEL
+_readlinkat
+_renameat
+_symlinkat
+_unlinkat
+# fdopendir.o (for fdopendir)
+_fdopendir*
+# pthread_chdir.o
+___mpls_best_fchdir
+_pthread_*chdir_np
+# time.o (for clock_gettime())
+_clock_get*
+_clock_set*
+_timespec_get
+_mach_approximate_time
+_mach_continuous_approximate_time
+_mach_continuous_time
+# setattrlistat.o
+_setattrlistat
+# statxx.o
+_fstatat*
+# sysconf.o
+_macports_legacy_sysconf
+_sysconf
+# utimensat.o
+_futimens
+_utimensat
