diff --git a/libcxx/include/__config b/libcxx/include/__config
index 1081538..716e37e 100644
--- a/libcxx/include/__config
+++ b/libcxx/include/__config
@@ -33,6 +33,12 @@
 #  define _LIBCPP_COMPILER_GCC
 #endif
 
+// we are not going to use Apple Availability testing in these headers
+#define _LIBCPP_DISABLE_AVAILABILITY
+// make per TU ABI insulation enabled by default; it can be overridden by
+// -D_LIBCPP_HIDE_FROM_ABI_PER_TU=0
+#define _LIBCPP_HIDE_FROM_ABI_PER_TU_BY_DEFAULT
+
 #ifdef __cplusplus
 
 // _LIBCPP_VERSION represents the version of libc++, which matches the version of LLVM.
