diff --git tools/gyp/pylib/gyp/orig.xcode_emulation.py tools/gyp/pylib/gyp/xcode_emulation.py
index 192a523..504ba61 100644
--- tools/gyp/pylib/gyp/orig.xcode_emulation.py
+++ tools/gyp/pylib/gyp/xcode_emulation.py
@@ -605,7 +605,7 @@ class XcodeSettings:
 
         self._Appendf(cflags, "GCC_OPTIMIZATION_LEVEL", "-O%s", default="s")
 
-        if self._Test("GCC_GENERATE_DEBUGGING_SYMBOLS", "YES", default="YES"):
+        if self._Test("GCC_GENERATE_DEBUGGING_SYMBOLS", "YES", default="NO"):
             dbg_format = self._Settings().get("DEBUG_INFORMATION_FORMAT", "dwarf")
             if dbg_format == "dwarf":
                 cflags.append("-gdwarf-2")
diff --git deps/npm/node_modules/node-gyp/gyp/pylib/gyp/orig.xcode_emulation.py deps/npm/node_modules/node-gyp/gyp/pylib/gyp/xcode_emulation.py
index 85a63df..1ef43dc 100644
--- deps/npm/node_modules/node-gyp/gyp/pylib/gyp/orig.xcode_emulation.py
+++ deps/npm/node_modules/node-gyp/gyp/pylib/gyp/xcode_emulation.py
@@ -606,7 +606,7 @@ class XcodeSettings:
 
         self._Appendf(cflags, "GCC_OPTIMIZATION_LEVEL", "-O%s", default="s")
 
-        if self._Test("GCC_GENERATE_DEBUGGING_SYMBOLS", "YES", default="YES"):
+        if self._Test("GCC_GENERATE_DEBUGGING_SYMBOLS", "YES", default="NO"):
             dbg_format = self._Settings().get("DEBUG_INFORMATION_FORMAT", "dwarf")
             if dbg_format == "dwarf":
                 cflags.append("-gdwarf-2")
