diff --git configure.orig configure
index 56720e8..d4aaf4c 100755
--- configure.orig
+++ configure
@@ -4,13 +4,7 @@
 # Note that the mix of single and double quotes is intentional,
 # as is the fact that the ] goes on a new line.
 _=[ 'exec' '/bin/sh' '-c' '''
-command -v python3.13 >/dev/null && exec python3.13 "$0" "$@"
-command -v python3.12 >/dev/null && exec python3.12 "$0" "$@"
-command -v python3.11 >/dev/null && exec python3.11 "$0" "$@"
-command -v python3.10 >/dev/null && exec python3.10 "$0" "$@"
-command -v python3.9 >/dev/null && exec python3.9 "$0" "$@"
-command -v python3.8 >/dev/null && exec python3.8 "$0" "$@"
-command -v python3 >/dev/null && exec python3 "$0" "$@"
+command -v ${PYTHON_EXE} >/dev/null && exec ${PYTHON_EXE} "$0" "$@"
 exec python "$0" "$@"
 ''' "$0" "$@"
 ]
