--- llvm-3.8.1.src/tools/lldb/packages/Python/lldbsuite/test/orig.lldbtest.py	2016-02-02 17:50:06.000000000 +0100
+++ llvm-3.8.1.src/tools/lldb/packages/Python/lldbsuite/test/lldbtest.py	2016-09-01 17:03:30.000000000 +0200
@@ -2131,7 +2131,7 @@
 
     def signBinary(self, binary_path):
         if sys.platform.startswith("darwin"):
-            codesign_cmd = "codesign --force --sign lldb_codesign %s" % (binary_path)
+            codesign_cmd = "echo Skipping codesign --force --sign lldb_codesign %s" % (binary_path)
             call(codesign_cmd, shell=True)
 
     def findBuiltClang(self):
--- llvm-3.8.1.src/tools/lldb/tools/debugserver/source/MacOSX/orig.CMakeLists.txt	2015-10-23 00:48:52.000000000 +0200
+++ llvm-3.8.1.src/tools/lldb/tools/debugserver/source/MacOSX/CMakeLists.txt	2016-09-01 17:03:19.000000000 +0200
@@ -60,29 +60,29 @@
 target_link_libraries(debugserver ${DEBUGSERVER_USED_LIBS})
 
 # Sign the debugserver binary
-set (CODESIGN_IDENTITY lldb_codesign)
-execute_process(
-  COMMAND xcrun -f codesign_allocate
-  OUTPUT_STRIP_TRAILING_WHITESPACE
-  OUTPUT_VARIABLE CODESIGN_ALLOCATE
-  )
-# Older cmake versions don't support "-E env".
-if (${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION} LESS 3.2)
-  add_custom_command(TARGET debugserver
-    POST_BUILD
-    # Note: --entitlements option removed, as it causes errors when debugging.
-    # was: COMMAND CODESIGN_ALLOCATE=${CODESIGN_ALLOCATE} codesign --entitlements ${CMAKE_CURRENT_SOURCE_DIR}/../debugserver-entitlements.plist --force --sign ${CODESIGN_IDENTITY} debugserver
-    COMMAND CODESIGN_ALLOCATE=${CODESIGN_ALLOCATE} codesign --force --sign ${CODESIGN_IDENTITY} debugserver
-    WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/bin
-  )
-else()
-  add_custom_command(TARGET debugserver
-    POST_BUILD
-    # Note: --entitlements option removed (see comment above).
-    COMMAND ${CMAKE_COMMAND} -E env CODESIGN_ALLOCATE=${CODESIGN_ALLOCATE} codesign --force --sign ${CODESIGN_IDENTITY} debugserver
-    WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/bin
-  )
-endif()
+#set (CODESIGN_IDENTITY lldb_codesign)
+#execute_process(
+#  COMMAND xcrun -f codesign_allocate
+#  OUTPUT_STRIP_TRAILING_WHITESPACE
+#  OUTPUT_VARIABLE CODESIGN_ALLOCATE
+#  )
+## Older cmake versions don't support "-E env".
+#if (${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION} LESS 3.2)
+#  add_custom_command(TARGET debugserver
+#    POST_BUILD
+#    # Note: --entitlements option removed, as it causes errors when debugging.
+#    # was: COMMAND CODESIGN_ALLOCATE=${CODESIGN_ALLOCATE} codesign --entitlements ${CMAKE_CURRENT_SOURCE_DIR}/../debugserver-entitlements.plist --force --sign ${CODESIGN_IDENTITY} debugserver
+#    COMMAND CODESIGN_ALLOCATE=${CODESIGN_ALLOCATE} codesign --force --sign ${CODESIGN_IDENTITY} debugserver
+#    WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/bin
+#  )
+#else()
+#  add_custom_command(TARGET debugserver
+#    POST_BUILD
+#    # Note: --entitlements option removed (see comment above).
+#    COMMAND ${CMAKE_COMMAND} -E env CODESIGN_ALLOCATE=${CODESIGN_ALLOCATE} codesign --force --sign ${CODESIGN_IDENTITY} debugserver
+#    WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/bin
+#  )
+#endif()
 
 install(TARGETS debugserver
   RUNTIME DESTINATION bin
--- llvm-3.8.1.src/tools/lldb/tools/debugserver/source/MacOSX/orig.Makefile	2014-01-18 09:05:32.000000000 +0100
+++ llvm-3.8.1.src/tools/lldb/tools/debugserver/source/MacOSX/Makefile	2016-09-01 16:58:34.000000000 +0200
@@ -12,7 +12,7 @@
 
 TOOLNAME = debugserver
 
-CODESIGN_TOOLS := 1
+CODESIGN_TOOLS := 0
 
 TOOL_CODESIGN_IDENTITY := lldb_codesign
 
@@ -51,4 +51,4 @@
 endif
 
 $(GENERATED_MACH_SOURCES):
-	mig -I$(PROJ_OBJ_DIR)/../.. $(PROJ_SRC_DIR)/dbgnub-mig.defs
\ No newline at end of file
+	mig -I$(PROJ_OBJ_DIR)/../.. $(PROJ_SRC_DIR)/dbgnub-mig.defs
