diff --git app/CMakeLists.txt app/CMakeLists.txt
index 5909ac1..ca665b3 100644
--- app/CMakeLists.txt
+++ app/CMakeLists.txt
@@ -12,7 +12,9 @@ find_package(SharedMimeInfo REQUIRED)
 set(kdevelop_SRCS  main.cpp kdevideextension.cpp splash.cpp)
 # kde4_add_app_icon(kdevelop_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/../pics/hi*-app-kdevelop.png")
 if(APPLE)
-    #kde4_add_app_icon(kdevelop_SRCS "${KDE4_ICON_INSTALL_DIR}/oxygen/*/apps/kdevelop.png")
+    include(ECMAddAppIcon)
+    file(GLOB ICONS_SRCS "${CMAKE_CURRENT_BINARY_DIR}/icons/*.png")
+    ecm_add_app_icon(kdevelop_SRCS ICONS ${ICONS_SRCS})
 else()
     #kde4_add_app_icon(kdevelop_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/../pics/hi*-app-kdevelop.png")
 endif()
