diff --git src/controls/templates/SwipeListItem.qml src/controls/templates/SwipeListItem.qml
index 955126d..f5f8cc0 100644
--- src/controls/templates/SwipeListItem.qml
+++ src/controls/templates/SwipeListItem.qml
@@ -259,8 +259,6 @@ T2.ItemDelegate {
                 }
                 delegate: Controls.ToolButton {
                     anchors.verticalCenter: parent.verticalCenter
-                    icon.name: modelData.iconName !== "" ? modelData.iconName : undefined
-                    icon.source: modelData.iconSource !== "" ? modelData.iconSource : undefined
                     enabled: (modelData && modelData.enabled !== undefined) ? modelData.enabled : true;
                     visible: (modelData && modelData.visible !== undefined) ? modelData.visible : true;
                     onVisibleChanged: {
