diff --git src/controls/private/ContextDrawerActionItem.qml src/controls/private/ContextDrawerActionItem.qml
index daed514..65c3b37 100644
--- src/controls/private/ContextDrawerActionItem.qml
+++ src/controls/private/ContextDrawerActionItem.qml
@@ -64,7 +64,7 @@ BasicListItem {
     Icon {
         isMask: true
         Layout.alignment: Qt.AlignVCenter
-        Layout.rightMargin: !Settings.isMobile && mainFlickable.contentHeight > mainFlickable.height ? Units.gridUnit : 0
+        Layout.rightMargin: !Settings.isMobile
         Layout.preferredHeight: Units.iconSizes.small/2
         selected: listItem.checked || listItem.pressed
         Layout.preferredWidth: Layout.preferredHeight
