diff --git a/kcmkwin/kwindesktop/CMakeLists.txt b/kcmkwin/kwindesktop/CMakeLists.txt
index 67a0a9d383..2e36e5339b 100644
--- a/kcmkwin/kwindesktop/CMakeLists.txt
+++ b/kcmkwin/kwindesktop/CMakeLists.txt
@@ -35,4 +35,4 @@ kcoreaddons_desktop_to_json(kcm_kwin_virtualdesktops "kcm_kwin_virtualdesktops.d
 install(FILES virtualdesktopssettings.kcfg DESTINATION ${KDE_INSTALL_KCFGDIR})
 install(TARGETS kcm_kwin_virtualdesktops DESTINATION ${KDE_INSTALL_PLUGINDIR}/kcms)
 install(FILES kcm_kwin_virtualdesktops.desktop  DESTINATION  ${KDE_INSTALL_KSERVICES5DIR})
-# kpackage_install_package(package kcm_kwin_virtualdesktops kcms)
+kpackage_install_package(package kcm_kwin_virtualdesktops kcms)
diff --git a/kcmkwin/kwindesktop/package/contents/ui/main.qml b/kcmkwin/kwindesktop/package/contents/ui/main.qml
index 770bc22d68..15bb330e30 100644
--- a/kcmkwin/kwindesktop/package/contents/ui/main.qml
+++ b/kcmkwin/kwindesktop/package/contents/ui/main.qml
@@ -18,11 +18,11 @@
 */
 
 import QtQuick 2.5
-import QtQuick.Controls 2.5 as QQC2
+import QtQuick.Controls 2.2 as QQC2
 import QtQuick.Layouts 1.1
 
 import org.kde.kcm 1.2
-import org.kde.kirigami 2.10 as Kirigami
+import org.kde.kirigami 2.7 as Kirigami
 import org.kde.plasma.core 2.1 as PlasmaCore
 
 ScrollViewKCM {
@@ -126,9 +126,20 @@ ScrollViewKCM {
         model: kcm.desktopsModel.ready ? kcm.desktopsModel : null
 
         section.property: "DesktopRow"
-        section.delegate: Kirigami.ListSectionHeader {
+        section.delegate: Kirigami.AbstractListItem {
             width: desktopsList.width
-            label: i18n("Row %1", section)
+
+            backgroundColor: Kirigami.Theme.backgroundColor
+
+            hoverEnabled: false
+            supportsMouseEvents: false
+
+            Kirigami.Theme.inherit: false
+            Kirigami.Theme.colorSet: Kirigami.Theme.Window
+
+            QQC2.Label {
+                text: i18n("Row %1", section)
+            }
         }
 
         delegate: Kirigami.DelegateRecycler {
@@ -142,7 +153,6 @@ ScrollViewKCM {
         RowLayout {
             QQC2.Button {
                 text: i18nc("@action:button", "Add")
-                icon.name: "list-add"
 
                 onClicked: kcm.desktopsModel.createDesktop(i18n("New Desktop"))
             }
@@ -206,7 +216,6 @@ ScrollViewKCM {
                 QQC2.Button {
                     enabled: animationEnabled.checked && kcm.animationsModel.currentConfigurable
 
-                    icon.name: "configure"
 
                     onClicked: kcm.configureAnimation()
                 }
@@ -214,7 +223,6 @@ ScrollViewKCM {
                 QQC2.Button {
                     enabled: animationEnabled.checked
 
-                    icon.name: "dialog-information"
 
                     onClicked: kcm.showAboutAnimation()
                 }
