diff --git a/CMakeLists.txt b/CMakeLists.txt
index f9a40ee43a209818f970992eb7cea97a79f45c7c..d25d4de098d0dc6c95f7aa6c083b8000fc243002 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -280,12 +280,12 @@ set_package_properties(Xwayland PROPERTIES
     PURPOSE "Needed for running kwin_wayland"
 )
 
-find_package(Libcap)
-set_package_properties(Libcap PROPERTIES
-    TYPE OPTIONAL
-    PURPOSE "Needed for running kwin_wayland with real-time scheduling policy"
-)
-set(HAVE_LIBCAP ${Libcap_FOUND})
+# find_package(Libcap)
+# set_package_properties(Libcap PROPERTIES
+#     TYPE OPTIONAL
+#     PURPOSE "Needed for running kwin_wayland with real-time scheduling policy"
+# )
+# set(HAVE_LIBCAP ${Libcap_FOUND})
 
 find_package(hwdata)
 set_package_properties(hwdata PROPERTIES
@@ -710,20 +710,20 @@ set(kwin_WAYLAND_SRCS
 
 add_executable(kwin_wayland ${kwin_WAYLAND_SRCS} ${kwin_XWAYLAND_SRCS})
 target_link_libraries(kwin_wayland kwin KF5::Crash)
-if (HAVE_LIBCAP)
-    target_link_libraries(kwin_wayland ${Libcap_LIBRARIES})
-endif()
+# if (HAVE_LIBCAP)
+#     target_link_libraries(kwin_wayland ${Libcap_LIBRARIES})
+# endif()
 
 install(TARGETS kwin_wayland ${INSTALL_TARGETS_DEFAULT_ARGS})
-if (HAVE_LIBCAP)
-    install(
-    CODE "execute_process(
-            COMMAND
-                ${SETCAP_EXECUTABLE}
-                CAP_SYS_NICE=+ep
-                \$ENV{DESTDIR}${CMAKE_INSTALL_FULL_BINDIR}/kwin_wayland)"
-    )
-endif()
+# if (HAVE_LIBCAP)
+#     install(
+#     CODE "execute_process(
+#             COMMAND
+#                 ${SETCAP_EXECUTABLE}
+#                 CAP_SYS_NICE=+ep
+#                 \$ENV{DESTDIR}${CMAKE_INSTALL_FULL_BINDIR}/kwin_wayland)"
+#     )
+# endif()
 
 add_subdirectory(platformsupport)
 add_subdirectory(plugins)
