diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1963844..a46c5c4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,8 +2,8 @@ cmake_minimum_required(VERSION 3.5)
 set(PROJECT_VERSION "5.20.0")
 set(PROJECT_VERSION_MAJOR 5)
 
-set(QT_MIN_VERSION "5.15.0")
-set(KF5_MIN_VERSION "5.74")
+set(QT_MIN_VERSION "5.9.8")
+set(KF5_MIN_VERSION "5.60")
 set(WAYLAND_MIN_VERSION "1.15")
 
 project(KWaylandServer VERSION ${PROJECT_VERSION})
@@ -41,7 +41,7 @@ ecm_setup_version(PROJECT VARIABLE_PREFIX KWAYLANDSERVER
 # Dependencies
 find_package(Qt5 ${QT_MIN_VERSION} CONFIG REQUIRED Concurrent Gui WaylandClient)
 find_package(KF5Wayland ${KF5_MIN_VERSION} CONFIG REQUIRED)
-find_package(PlasmaWaylandProtocols CONFIG REQUIRED)
+# find_package(PlasmaWaylandProtocols CONFIG REQUIRED)
 
 find_package(QtWaylandScanner REQUIRED)
 
@@ -66,7 +66,7 @@ include_directories(${CMAKE_CURRENT_BINARY_DIR})
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=gnu90")
 
 add_definitions(-DQT_NO_FOREACH)
-add_definitions(-DQT_DISABLE_DEPRECATED_BEFORE=0x050d00)
+# add_definitions(-DQT_DISABLE_DEPRECATED_BEFORE=0x050d00)
 # Subdirectories
 if (IS_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/po")
     ecm_install_po_files_as_qm(po)
diff --git a/autotests/server/CMakeLists.txt b/autotests/server/CMakeLists.txt
index a179f18..bf08bae 100644
--- a/autotests/server/CMakeLists.txt
+++ b/autotests/server/CMakeLists.txt
@@ -79,14 +79,14 @@ ecm_mark_as_test(testViewporterInterface)
 ########################################################
 # Test ScreencastV1Interface
 ########################################################
-ecm_add_qtwayland_client_protocol(SCREENCAST_SRCS
-    PROTOCOL PROTOCOL ${PLASMA_WAYLAND_PROTOCOLS_DIR}/screencast.xml
-    BASENAME zkde-screencast-unstable-v1
-)
-add_executable(testScreencastV1Interface test_screencast.cpp ${SCREENCAST_SRCS})
-target_link_libraries(testScreencastV1Interface Qt5::Test Plasma::KWaylandServer Wayland::Client KF5::WaylandClient)
-add_test(NAME kwayland-testScreencastV1Interface COMMAND testScreencastV1Interface)
-ecm_mark_as_test(testScreencastV1Interface)
+# ecm_add_qtwayland_client_protocol(SCREENCAST_SRCS
+#     PROTOCOL PROTOCOL ${PLASMA_WAYLAND_PROTOCOLS_DIR}/screencast.xml
+#     BASENAME zkde-screencast-unstable-v1
+# )
+# add_executable(testScreencastV1Interface test_screencast.cpp ${SCREENCAST_SRCS})
+# target_link_libraries(testScreencastV1Interface Qt5::Test Plasma::KWaylandServer Wayland::Client KF5::WaylandClient)
+# add_test(NAME kwayland-testScreencastV1Interface COMMAND testScreencastV1Interface)
+# ecm_mark_as_test(testScreencastV1Interface)
 
 ########################################################
 # Test InputMethod Interface
diff --git a/src/server/CMakeLists.txt b/src/server/CMakeLists.txt
index cb45ea9..293e393 100644
--- a/src/server/CMakeLists.txt
+++ b/src/server/CMakeLists.txt
@@ -1,11 +1,11 @@
 set(SERVER_LIB_SRCS
     abstract_data_source.cpp
-    appmenu_interface.cpp
-    blur_interface.cpp
+#     appmenu_interface.cpp
+#     blur_interface.cpp
     buffer_interface.cpp
     clientconnection.cpp
     compositor_interface.cpp
-    contrast_interface.cpp
+#     contrast_interface.cpp
     datacontroldevice_v1_interface.cpp
     datacontroldevicemanager_v1_interface.cpp
     datacontroloffer_v1_interface.cpp
@@ -15,27 +15,27 @@ set(SERVER_LIB_SRCS
     dataoffer_interface.cpp
     datasource_interface.cpp
     display.cpp
-    dpms_interface.cpp
-    eglstream_controller_interface.cpp
-    fakeinput_interface.cpp
+#     dpms_interface.cpp
+#     eglstream_controller_interface.cpp
+#     fakeinput_interface.cpp
     filtered_display.cpp
     global.cpp
-    idle_interface.cpp
+#     idle_interface.cpp
     idleinhibit_v1_interface
-    inputmethod_v1_interface.cpp
-    keyboard_interface.cpp
+#     inputmethod_v1_interface.cpp
+#     keyboard_interface.cpp
     keyboard_shortcuts_inhibit_v1_interface.cpp
-    keystate_interface.cpp
+#     keystate_interface.cpp
     layershell_v1_interface.cpp
     linuxdmabuf_v1_interface.cpp
     output_interface.cpp
     outputchangeset.cpp
-    outputconfiguration_interface.cpp
-    outputdevice_interface.cpp
-    outputmanagement_interface.cpp
-    plasmashell_interface.cpp
-    plasmavirtualdesktop_interface.cpp
-    plasmawindowmanagement_interface.cpp
+#     outputconfiguration_interface.cpp
+#     outputdevice_interface.cpp
+#     outputmanagement_interface.cpp
+#     plasmashell_interface.cpp
+#     plasmavirtualdesktop_interface.cpp
+#     plasmawindowmanagement_interface.cpp
     pointer_interface.cpp
     pointerconstraints_interface.cpp
     pointerconstraints_interface_v1.cpp
@@ -49,25 +49,25 @@ set(SERVER_LIB_SRCS
     relativepointer_interface.cpp
     relativepointer_interface_v1.cpp
     resource.cpp
-    screencast_v1_interface.cpp
+#     screencast_v1_interface.cpp
     seat_interface.cpp
-    server_decoration_interface.cpp
-    server_decoration_palette_interface.cpp
-    shadow_interface.cpp
-    slide_interface.cpp
+#     server_decoration_interface.cpp
+#     server_decoration_palette_interface.cpp
+#     shadow_interface.cpp
+#     slide_interface.cpp
     subcompositor_interface.cpp
     surface_interface.cpp
     surfacerole.cpp
     tablet_interface.cpp
-    textinput_interface.cpp
-    textinput_interface_v0.cpp
-    textinput_interface_v2.cpp
+#     textinput_interface.cpp
+#     textinput_interface_v0.cpp
+#     textinput_interface_v2.cpp
     touch_interface.cpp
     viewporter_interface.cpp
     xdgdecoration_v1_interface.cpp
     xdgforeign_v2_interface.cpp
     xdgoutput_v1_interface.cpp
-    xdgshell_interface.cpp
+#     xdgshell_interface.cpp
 )
 
 ecm_qt_declare_logging_category(SERVER_LIB_SRCS
@@ -84,84 +84,84 @@ ecm_add_qtwayland_server_protocol(SERVER_LIB_SRCS
     BASENAME wayland
 )
 
-ecm_add_wayland_server_protocol(SERVER_LIB_SRCS
-    PROTOCOL ${PLASMA_WAYLAND_PROTOCOLS_DIR}/output-management.xml
-    BASENAME output-management
-)
-
-ecm_add_wayland_server_protocol(SERVER_LIB_SRCS
-    PROTOCOL ${PLASMA_WAYLAND_PROTOCOLS_DIR}/outputdevice.xml
-    BASENAME org_kde_kwin_outputdevice
-)
-
-ecm_add_wayland_server_protocol(SERVER_LIB_SRCS
-    PROTOCOL ${PLASMA_WAYLAND_PROTOCOLS_DIR}/plasma-shell.xml
-    BASENAME plasma-shell
-)
-
-ecm_add_qtwayland_server_protocol(SERVER_LIB_SRCS
-    PROTOCOL ${PLASMA_WAYLAND_PROTOCOLS_DIR}/plasma-virtual-desktop.xml
-    BASENAME org-kde-plasma-virtual-desktop
-)
-
-ecm_add_wayland_server_protocol(SERVER_LIB_SRCS
-    PROTOCOL ${PLASMA_WAYLAND_PROTOCOLS_DIR}/plasma-window-management.xml
-    BASENAME plasma-window-management
-)
-
-ecm_add_wayland_server_protocol(SERVER_LIB_SRCS
-    PROTOCOL ${PLASMA_WAYLAND_PROTOCOLS_DIR}/surface-extension.xml
-    BASENAME qt-surface-extension
-)
-
-ecm_add_qtwayland_server_protocol(SERVER_LIB_SRCS
-    PROTOCOL ${PLASMA_WAYLAND_PROTOCOLS_DIR}/idle.xml
-    BASENAME idle
-)
-
-ecm_add_wayland_server_protocol(SERVER_LIB_SRCS
-    PROTOCOL ${PLASMA_WAYLAND_PROTOCOLS_DIR}/fake-input.xml
-    BASENAME fake-input
-)
-
-ecm_add_qtwayland_server_protocol(SERVER_LIB_SRCS
-    PROTOCOL ${PLASMA_WAYLAND_PROTOCOLS_DIR}/shadow.xml
-    BASENAME shadow
-)
-ecm_add_qtwayland_server_protocol(SERVER_LIB_SRCS
-    PROTOCOL ${PLASMA_WAYLAND_PROTOCOLS_DIR}/dpms.xml
-    BASENAME dpms
-)
-
-ecm_add_qtwayland_server_protocol(SERVER_LIB_SRCS
-    PROTOCOL ${PLASMA_WAYLAND_PROTOCOLS_DIR}/blur.xml
-    BASENAME blur
-)
-
-ecm_add_qtwayland_server_protocol(SERVER_LIB_SRCS
-    PROTOCOL ${PLASMA_WAYLAND_PROTOCOLS_DIR}/contrast.xml
-    BASENAME contrast
-)
+# ecm_add_wayland_server_protocol(SERVER_LIB_SRCS
+#     PROTOCOL ${PLASMA_WAYLAND_PROTOCOLS_DIR}/output-management.xml
+#     BASENAME output-management
+# )
+# 
+# ecm_add_wayland_server_protocol(SERVER_LIB_SRCS
+#     PROTOCOL ${PLASMA_WAYLAND_PROTOCOLS_DIR}/outputdevice.xml
+#     BASENAME org_kde_kwin_outputdevice
+# )
+# 
+# ecm_add_wayland_server_protocol(SERVER_LIB_SRCS
+#     PROTOCOL ${PLASMA_WAYLAND_PROTOCOLS_DIR}/plasma-shell.xml
+#     BASENAME plasma-shell
+# )
+# 
+# ecm_add_qtwayland_server_protocol(SERVER_LIB_SRCS
+#     PROTOCOL ${PLASMA_WAYLAND_PROTOCOLS_DIR}/plasma-virtual-desktop.xml
+#     BASENAME org-kde-plasma-virtual-desktop
+# )
+# 
+# ecm_add_wayland_server_protocol(SERVER_LIB_SRCS
+#     PROTOCOL ${PLASMA_WAYLAND_PROTOCOLS_DIR}/plasma-window-management.xml
+#     BASENAME plasma-window-management
+# )
+# 
+# ecm_add_wayland_server_protocol(SERVER_LIB_SRCS
+#     PROTOCOL ${PLASMA_WAYLAND_PROTOCOLS_DIR}/surface-extension.xml
+#     BASENAME qt-surface-extension
+# )
+# 
+# ecm_add_qtwayland_server_protocol(SERVER_LIB_SRCS
+#     PROTOCOL ${PLASMA_WAYLAND_PROTOCOLS_DIR}/idle.xml
+#     BASENAME idle
+# )
+# 
+# ecm_add_wayland_server_protocol(SERVER_LIB_SRCS
+#     PROTOCOL ${PLASMA_WAYLAND_PROTOCOLS_DIR}/fake-input.xml
+#     BASENAME fake-input
+# )
+# 
+# ecm_add_qtwayland_server_protocol(SERVER_LIB_SRCS
+#     PROTOCOL ${PLASMA_WAYLAND_PROTOCOLS_DIR}/shadow.xml
+#     BASENAME shadow
+# )
+# ecm_add_qtwayland_server_protocol(SERVER_LIB_SRCS
+#     PROTOCOL ${PLASMA_WAYLAND_PROTOCOLS_DIR}/dpms.xml
+#     BASENAME dpms
+# )
+# 
+# ecm_add_qtwayland_server_protocol(SERVER_LIB_SRCS
+#     PROTOCOL ${PLASMA_WAYLAND_PROTOCOLS_DIR}/blur.xml
+#     BASENAME blur
+# )
+# 
+# ecm_add_qtwayland_server_protocol(SERVER_LIB_SRCS
+#     PROTOCOL ${PLASMA_WAYLAND_PROTOCOLS_DIR}/contrast.xml
+#     BASENAME contrast
+# )
 
 ecm_add_wayland_server_protocol(SERVER_LIB_SRCS
     PROTOCOL ${WaylandProtocols_DATADIR}/unstable/relative-pointer/relative-pointer-unstable-v1.xml
     BASENAME relativepointer-unstable-v1
 )
 
-ecm_add_wayland_server_protocol(SERVER_LIB_SRCS
-    PROTOCOL ${PLASMA_WAYLAND_PROTOCOLS_DIR}/slide.xml
-    BASENAME slide
-)
-
-ecm_add_qtwayland_server_protocol(SERVER_LIB_SRCS
-    PROTOCOL ${PLASMA_WAYLAND_PROTOCOLS_DIR}/server-decoration.xml
-    BASENAME server-decoration
-)
-
-ecm_add_wayland_server_protocol(SERVER_LIB_SRCS
-    PROTOCOL ${PLASMA_WAYLAND_PROTOCOLS_DIR}/text-input.xml
-    BASENAME text
-)
+# ecm_add_wayland_server_protocol(SERVER_LIB_SRCS
+#     PROTOCOL ${PLASMA_WAYLAND_PROTOCOLS_DIR}/slide.xml
+#     BASENAME slide
+# )
+# 
+# ecm_add_qtwayland_server_protocol(SERVER_LIB_SRCS
+#     PROTOCOL ${PLASMA_WAYLAND_PROTOCOLS_DIR}/server-decoration.xml
+#     BASENAME server-decoration
+# )
+# 
+# ecm_add_wayland_server_protocol(SERVER_LIB_SRCS
+#     PROTOCOL ${PLASMA_WAYLAND_PROTOCOLS_DIR}/text-input.xml
+#     BASENAME text
+# )
 
 
 ecm_add_qtwayland_server_protocol(SERVER_LIB_SRCS
@@ -169,10 +169,10 @@ ecm_add_qtwayland_server_protocol(SERVER_LIB_SRCS
     BASENAME text-input-unstable-v1
 )
 
-ecm_add_wayland_server_protocol(SERVER_LIB_SRCS
-    PROTOCOL ${PLASMA_WAYLAND_PROTOCOLS_DIR}/text-input-unstable-v2.xml
-    BASENAME text-input-unstable-v2
-)
+# ecm_add_wayland_server_protocol(SERVER_LIB_SRCS
+#     PROTOCOL ${PLASMA_WAYLAND_PROTOCOLS_DIR}/text-input-unstable-v2.xml
+#     BASENAME text-input-unstable-v2
+# )
 
 ecm_add_wayland_server_protocol(SERVER_LIB_SRCS
     PROTOCOL ${WaylandProtocols_DATADIR}/unstable/pointer-gestures/pointer-gestures-unstable-v1.xml
@@ -194,15 +194,15 @@ ecm_add_qtwayland_server_protocol(SERVER_LIB_SRCS
     BASENAME idle-inhibit-unstable-v1
 )
 
-ecm_add_qtwayland_server_protocol(SERVER_LIB_SRCS
-    PROTOCOL ${PLASMA_WAYLAND_PROTOCOLS_DIR}/appmenu.xml
-    BASENAME appmenu
-)
-
-ecm_add_qtwayland_server_protocol(SERVER_LIB_SRCS
-    PROTOCOL ${PLASMA_WAYLAND_PROTOCOLS_DIR}/server-decoration-palette.xml
-    BASENAME server-decoration-palette
-)
+# ecm_add_qtwayland_server_protocol(SERVER_LIB_SRCS
+#     PROTOCOL ${PLASMA_WAYLAND_PROTOCOLS_DIR}/appmenu.xml
+#     BASENAME appmenu
+# )
+# 
+# ecm_add_qtwayland_server_protocol(SERVER_LIB_SRCS
+#     PROTOCOL ${PLASMA_WAYLAND_PROTOCOLS_DIR}/server-decoration-palette.xml
+#     BASENAME server-decoration-palette
+# )
 
 ecm_add_qtwayland_server_protocol(SERVER_LIB_SRCS
     PROTOCOL ${WaylandProtocols_DATADIR}/unstable/xdg-output/xdg-output-unstable-v1.xml
@@ -219,15 +219,15 @@ ecm_add_qtwayland_server_protocol(SERVER_LIB_SRCS
     BASENAME xdg-decoration-unstable-v1
 )
 
-ecm_add_qtwayland_server_protocol(SERVER_LIB_SRCS
-    PROTOCOL ${PLASMA_WAYLAND_PROTOCOLS_DIR}/wayland-eglstream-controller.xml
-    BASENAME wl-eglstream-controller
-)
-
-ecm_add_wayland_server_protocol(SERVER_LIB_SRCS
-    PROTOCOL ${PLASMA_WAYLAND_PROTOCOLS_DIR}/keystate.xml
-    BASENAME keystate
-)
+# ecm_add_qtwayland_server_protocol(SERVER_LIB_SRCS
+#     PROTOCOL ${PLASMA_WAYLAND_PROTOCOLS_DIR}/wayland-eglstream-controller.xml
+#     BASENAME wl-eglstream-controller
+# )
+# 
+# ecm_add_wayland_server_protocol(SERVER_LIB_SRCS
+#     PROTOCOL ${PLASMA_WAYLAND_PROTOCOLS_DIR}/keystate.xml
+#     BASENAME keystate
+# )
 
 ecm_add_wayland_server_protocol(SERVER_LIB_SRCS
     PROTOCOL ${WaylandProtocols_DATADIR}/unstable/linux-dmabuf/linux-dmabuf-unstable-v1.xml
@@ -264,10 +264,10 @@ ecm_add_qtwayland_server_protocol(SERVER_LIB_SRCS
     BASENAME wp-primary-selection-unstable-v1
 )
 
-ecm_add_qtwayland_server_protocol(SERVER_LIB_SRCS
-    PROTOCOL PROTOCOL ${PLASMA_WAYLAND_PROTOCOLS_DIR}/screencast.xml
-    BASENAME zkde-screencast-unstable-v1
-)
+# ecm_add_qtwayland_server_protocol(SERVER_LIB_SRCS
+#     PROTOCOL PROTOCOL ${PLASMA_WAYLAND_PROTOCOLS_DIR}/screencast.xml
+#     BASENAME zkde-screencast-unstable-v1
+# )
 
 ecm_add_qtwayland_server_protocol(SERVER_LIB_SRCS
     PROTOCOL ${WaylandProtocols_DATADIR}/unstable/input-method/input-method-unstable-v1.xml
