--- src/orig.kcrash.cpp	2015-11-05 18:12:52.000000000 +0100
+++ src/kcrash.cpp	2015-11-05 18:13:05.000000000 +0100
@@ -343,7 +343,7 @@
     return s_crashHandler;
 }
 
-#ifndef Q_OS_WIN
+#if !defined(Q_OS_WIN) && !defined(Q_OS_OSX)
 static void
 closeAllFDs()
 {
@@ -384,7 +384,7 @@
         crashRecursionCounter++;
     }
 
-#if !defined(Q_OS_WIN)
+#if !defined(Q_OS_WIN) && !defined(Q_OS_OSX)
     if (!(s_flags & KeepFDs)) {
         closeAllFDs();
     }
@@ -714,7 +714,9 @@
         if (setgid(getgid()) < 0 || setuid(getuid()) < 0) {
             _exit(253);    // This cannot happen. Theoretically.
         }
+#ifndef Q_OS_OSX
         closeAllFDs(); // We are in the child now. Close FDs unconditionally.
+#endif
         setenv("KCRASH_AUTO_RESTARTED", "1", 1);
         execvp(argv[0], const_cast< char ** >(argv));
         fprintf(stderr, "KCrash failed to exec(), errno = %d\n", errno);
