diff --git b/orig.CMakeLists.txt a/CMakeLists.txt
index e0a43c3..f15602b 100644
--- b/orig.CMakeLists.txt
+++ a/CMakeLists.txt
@@ -21,7 +21,8 @@ else()
   message("Building for target host application: " ${GMIC_QT_HOST})
 endif()
 
-set (GMIC_PATH "../gmic/src" CACHE STRING "Define the path to the gmic headers")
+option(GMIC_DIR "GMic source tree" "gmic")
+set (GMIC_PATH "../${GMIC_DIR}/src" CACHE STRING "Define the path to the gmic headers")
 message("G'MIC path: " ${GMIC_PATH})
 
 #
