diff --git b/gmic-qt/src/Host/KritaPlugin/orig.host.cpp a/gmic-qt/src/Host/KritaPlugin/host.cpp
index 7f441e8..c171261 100644
--- b/gmic-qt/src/Host/KritaPlugin/orig.host.cpp
+++ a/gmic-qt/src/Host/KritaPlugin/host.cpp
@@ -25,7 +25,9 @@
 #include <QFileInfo>
 #include <QSharedMemory>
 #include <QStandardPaths>
+#if QT_VERSION >= QT_VERSION_CHECK(5, 10, 0)
 #include <QStringLiteral>
+#endif
 #include <QUuid>
 #include <QVector>
 #include <algorithm>
diff --git a/gmic-qt/CMakeLists.txt b/gmic-qt/CMakeLists.txt
index 6c76f54..291273c 100644
--- a/gmic-qt/CMakeLists.txt
+++ b/gmic-qt/CMakeLists.txt
@@ -283,10 +287,10 @@ if (QT_NO_CREATE_VERSIONLESS_FUNCTIONS)
   )
 else()
   set(gmic_qt_LIBRARIES
-          Qt::Core
-          Qt::Widgets
-          Qt::Gui
-          Qt::Network
+          Qt5::Core
+          Qt5::Widgets
+          Qt5::Gui
+          Qt5::Network
   )
 endif()
 
diff --git a/gmic-qt/src/Widgets/ProgressInfoWidget.cpp b/gmic-qt/src/Widgets/ProgressInfoWidget.cpp
index bb34721..37098ab 100644
--- a/gmic-qt/src/Widgets/ProgressInfoWidget.cpp
+++ b/gmic-qt/src/Widgets/ProgressInfoWidget.cpp
@@ -128,7 +128,11 @@ void ProgressInfoWidget::startFilterThreadAnimationAndShow()
   QString largestText(tr("[Processing 88:00:00.888]"));
 #endif
   QFontMetrics fm(ui->label->font());
+#if QT_VERSION >= QT_VERSION_CHECK(5, 11, 0)
   ui->label->setMinimumWidth(fm.horizontalAdvance(largestText));
+#else
+  ui->label->setMinimumWidth(fm.width(largestText));
+#endif
 
   _canceled = false;
   _mode = Mode::GmicProcessing;
