diff --git a/gmic-qt/CMakeLists.txt b/gmic-qt/CMakeLists.txt
index 6c76f54..9c6e422 100644
--- a/gmic-qt/CMakeLists.txt
+++ b/gmic-qt/CMakeLists.txt
@@ -229,20 +229,24 @@ if(ENABLE_CURL)
 endif()
 
 #
-# Test for OpenMP
+# Test for OpenMP - we know the port ensures that LLVM's libomp is available
 #
-find_package(OpenMP 2.0)
+# find_package(OpenMP 2.0)
+find_library(OMP_LIB omp REQUIRED)
 set_package_properties(OpenMP PROPERTIES
     DESCRIPTION "A low-level parallel execution library"
-    URL "http://openmp.org/wp/"
+    URL "http://openmp.llvm.org"
     TYPE OPTIONAL
     PURPOSE "Optionally used by gmic-qt")
 
-if (OpenMP_FOUND)
-    message(STATUS "G'Mic: using OpenMP ${OpenMP_CXX_VERSION}")
-    link_libraries(OpenMP::OpenMP_CXX)
+    message(STATUS "G'Mic: using OpenMP from libomp")
+    if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fopenmp")
+        add_definitions(-fopenmp)
+    else()
+        list(APPEND EXTRA_LIBRARIES ${OMP_LIB})
+    endif()
     add_definitions(-Dcimg_use_openmp)
-endif()
 
 #
 # LTO option
