diff --git a/kded/configeditor.cpp b/kded/configeditor.cpp
index 4ce5cfe..bccae2c 100644
--- a/kded/configeditor.cpp
+++ b/kded/configeditor.cpp
@@ -27,6 +27,21 @@
 
 #include "configeditor.h"
 
+#if QT_VERSION < QT_VERSION_CHECK(5, 10, 0)
+static QString qEnvironmentVariable(const char *varName, const QString &defaultValue)
+{
+    QByteArray value = qgetenv(varName);
+    if (value.isNull())
+        return defaultValue;
+// duplicated in qfile.h (QFile::decodeName)
+#if defined(Q_OS_DARWIN)
+    return QString::fromUtf8(value).normalized(QString::NormalizationForm_C);
+#else // other Unix
+    return QString::fromLocal8Bit(value);
+#endif
+}
+#endif
+
 void ConfigEditor::setGtk3ConfigValueGSettings(const QString &paramName, const QVariant &paramValue, const QString &category)
 {
     g_autoptr(GSettings) gsettings = g_settings_new(category.toUtf8().constData());
diff --git a/kded/gtkconfig.json b/kded/gtkconfig.json
index 931edcf..8777b92 100644
--- a/kded/gtkconfig.json
+++ b/kded/gtkconfig.json
@@ -66,6 +66,7 @@
         ],
         "Version": "0.1"
     },
+    "X-KDE-DBus-ModuleName": "gtkconfig",
     "X-KDE-Kded-autoload": true,
     "X-KDE-Kded-load-on-demand": false,
     "X-KDE-Kded-phase": 1
