diff --git a/kmplot/CMakeLists.txt b/kmplot/CMakeLists.txt
index 5996285..f45a932 100644
--- a/kmplot/CMakeLists.txt
+++ b/kmplot/CMakeLists.txt
@@ -74,7 +74,7 @@ qt5_add_dbus_adaptor(kmplot_SRCS org.kde.kmplot.KmPlot.xml kmplot.h KmPlot)
 kconfig_add_kcfg_files(kmplot_SRCS settings.kcfgc )
 
 file(GLOB ICONS_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/../icons/*-apps-kmplot.png")
-ecm_add_app_icon(kmplot_SRCS ICONS ${ICONS_SRCS})
+ecm_add_app_icon(kmplot_SRCS ICONS "${CMAKE_CURRENT_SOURCE_DIR}/../icons/sc-apps-kmplot.svgz ${ICONS_SRCS}")
 
 add_executable(kmplot ${kmplot_SRCS})
 
