diff --git a/src/modules/FFmpeg/FFDecVAAPI.cpp b/src/modules/FFmpeg/FFDecVAAPI.cpp
index a41bbdba304e779cc05e624b5adfdb76b9b5cd16..f8a02e1f34942aacf4daa74c475ec4c1dfd083dc 100644
--- a/src/modules/FFmpeg/FFDecVAAPI.cpp
+++ b/src/modules/FFmpeg/FFDecVAAPI.cpp
@@ -170,13 +170,16 @@ void FFDecVAAPI::downloadVideoFrame(Frame &decoded)
 bool FFDecVAAPI::open(StreamInfo &streamInfo)
 {
     const AVPixelFormat pix_fmt = streamInfo.pixelFormat();
+#ifdef USE_VULKAN
     if (pix_fmt == AV_PIX_FMT_YUV420P10 && QMPlay2Core.isVulkanRenderer())
     {
         auto vkInstance = static_pointer_cast<QmVk::Instance>(QMPlay2Core.gpuInstance());
         if (!vkInstance->supportedPixelFormats().contains(pix_fmt))
             return false;
     }
-    else if (pix_fmt != AV_PIX_FMT_YUV420P && pix_fmt != AV_PIX_FMT_YUVJ420P)
+    else
+#endif
+    if (pix_fmt != AV_PIX_FMT_YUV420P && pix_fmt != AV_PIX_FMT_YUVJ420P)
     {
         return false;
     }
