diff --git a/src/modules/AudioFilters/AVAudioFilter.cpp b/src/modules/AudioFilters/AVAudioFilter.cpp
index d3353492..8581c6b6 100644
--- a/src/modules/AudioFilters/AVAudioFilter.cpp
+++ b/src/modules/AudioFilters/AVAudioFilter.cpp
@@ -36,20 +36,20 @@ QStringList AVAudioFilter::getAvailableFilters()
         if (!filter)
             break;
 
+        const auto name = QString(filter->name);
+
         if (filter->flags & AVFILTER_FLAG_METADATA_ONLY)
             continue;
 
-        if (!filter->inputs)
+        if (!filter->inputs && name != "ladspa")
             continue;
 
-        if (avfilter_pad_get_type(filter->inputs, 0) != AVMEDIA_TYPE_AUDIO)
+        if (filter->inputs && avfilter_pad_get_type(filter->inputs, 0) != AVMEDIA_TYPE_AUDIO)
             continue;
 
         if (filter->outputs && avfilter_pad_get_type(filter->outputs, 0) != AVMEDIA_TYPE_AUDIO)
             continue;
 
-        const auto name = QString(filter->name);
-
         if (name.startsWith("anull") || name.startsWith("abuffer") || name == "join")
             continue;
 
