diff --git a/src/gui/VideoThr.cpp b/src/gui/VideoThr.cpp
index 58cbf8fb..59004164 100644
--- a/src/gui/VideoThr.cpp
+++ b/src/gui/VideoThr.cpp
@@ -825,7 +825,11 @@ void VideoThr::screenshot(Frame videoFrame)
     QImage img;
 
 #ifdef USE_OPENGL
+#ifdef Q_OS_MACOS
+    if (auto hwGLInterop = static_pointer_cast<OpenGLHWInterop>(getHWDecContext()))
+#else
     if (auto hwGLInterop = dynamic_pointer_cast<OpenGLHWInterop>(getHWDecContext()))
+#endif
     {
         videoFrame = hwGLInterop->getCpuFrame(videoFrame);
     }
diff --git a/src/qmplay2/opengl/OpenGLWriter.cpp b/src/qmplay2/opengl/OpenGLWriter.cpp
index 1f7d2b10..84ebbc54 100644
--- a/src/qmplay2/opengl/OpenGLWriter.cpp
+++ b/src/qmplay2/opengl/OpenGLWriter.cpp
@@ -219,7 +219,11 @@ bool OpenGLWriter::open()
 
 bool OpenGLWriter::setHWDecContext(const shared_ptr<HWDecContext> &hwDecContext)
 {
+#ifdef Q_OS_MACOS
+    auto hwInterop = static_pointer_cast<OpenGLHWInterop>(hwDecContext);
+#else
     auto hwInterop = dynamic_pointer_cast<OpenGLHWInterop>(hwDecContext);
+#endif
     if (hwDecContext && !hwInterop)
         return false;
 
