diff --git a/src/modules/FFmpeg/FFDecVAAPI.cpp b/src/modules/FFmpeg/FFDecVAAPI.cpp
index e54aac03726f80de8fc4e1382ba00cd39143085b..bb6a03fc29a7127b2733211ea339467f189327b9 100644
--- a/src/modules/FFmpeg/FFDecVAAPI.cpp
+++ b/src/modules/FFmpeg/FFDecVAAPI.cpp
@@ -59,7 +59,7 @@ public:
 
     bool init(quint32 *textures) override
     {
-        static bool isEGL = (qgetenv("QT_XCB_GL_INTEGRATION").compare("xcb_egl", Qt::CaseInsensitive) == 0);
+        static bool isEGL = (QString(qgetenv("QT_XCB_GL_INTEGRATION")).compare("xcb_egl", Qt::CaseInsensitive) == 0);
         if (isEGL)
             return false; // Not supported
         return (vaCreateSurfaceGLX(m_vaapi->VADisp, GL_TEXTURE_2D, *textures, &m_glSurface) == VA_STATUS_SUCCESS);
diff --git a/src/modules/FFmpeg/VAAPI.cpp b/src/modules/FFmpeg/VAAPI.cpp
index 6bd077a5ef64eddbc01f00502ad7553ba26d1cdc..7a4ba5dd3ecc861aa62853c82291ca0edea1613d 100644
--- a/src/modules/FFmpeg/VAAPI.cpp
+++ b/src/modules/FFmpeg/VAAPI.cpp
@@ -175,7 +175,9 @@ void VAAPI::maybeInitVPP(int surfaceW, int surfaceH)
                                 VAProcFilterDeinterlacing,
                                 vpp_deint_type,
                                 0,
+#if VA_MAJOR_VERSION >= 1
                                 {}
+#endif
                             };
                             if (vaCreateBuffer(VADisp, context_vpp, VAProcFilterParameterBufferType, sizeof deint_params, 1, &deint_params, &m_vppDeintBuff) != VA_STATUS_SUCCESS)
                                 m_vppDeintBuff = VA_INVALID_ID;
