--- gio/orig.gsettingsschema.c	2018-08-17 02:03:20.000000000 +0200
+++ gio/gsettingsschema.c	2018-11-14 11:59:28.906849574 +0100
@@ -921,6 +921,9 @@
 void
 g_settings_schema_unref (GSettingsSchema *schema)
 {
+  if (!schema) {
+    return;
+  }
   if (g_atomic_int_dec_and_test (&schema->ref_count))
     {
       if (schema->extends)
@@ -965,8 +968,10 @@
     if ((value = gvdb_table_get_raw_value (s->table, key)))
       break;
 
-  if G_UNLIKELY (value == NULL || !g_variant_is_of_type (value, G_VARIANT_TYPE_TUPLE))
-    g_error ("Settings schema '%s' does not contain a key named '%s'", schema->id, key);
+  if G_UNLIKELY (value == NULL || !g_variant_is_of_type (value, G_VARIANT_TYPE_TUPLE)) {
+    g_warning ("Settings schema '%s' does not contain a key named '%s'", schema->id, key);
+    return NULL;
+  }
 
   iter = g_variant_iter_new (value);
   g_variant_unref (value);
@@ -1246,6 +1251,9 @@
   memset (key, 0, sizeof *key);
 
   iter = g_settings_schema_get_value (schema, name);
+  if (G_UNLIKELY (iter == NULL)) {
+    return;
+  }
 
   key->schema = g_settings_schema_ref (schema);
   key->default_value = g_variant_iter_next_value (iter);
