diff --git a/CMakeLists.txt b/CMakeLists.txt
index 21d6932..3a93eeb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -211,7 +211,9 @@ install(FILES ${README_HIDDENLEVEL_FILES} DESTINATION ${DOC_INSTALL_DIR}/hiddenl
 install(FILES README.md COPYING-LGPL2.txt checks.json DESTINATION ${DOC_INSTALL_DIR})
 
 # Build docs
-set(MAN_INSTALL_DIR "man/man1")
+if (NOT MAN_INSTALL_DIR)
+    set(MAN_INSTALL_DIR "man/man1")
+endif()
 add_subdirectory(docs)
 
 if(CLAZY_BUILD_UTILS_LIB)
