diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9ca83df..cef7187 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -62,6 +62,12 @@ macro(link_to_llvm name)
   foreach(llvm_lib ${LLVM_LIBS_LIST})
     target_link_libraries(${name} ${llvm_lib})
   endforeach()
+  if(${LLVM_VERSION} VERSION_GREATER "3.7.9")
+      target_link_libraries( ${name} "-lLLVM" )
+  else()
+      target_link_libraries( ${name} "-lLLVM-${LLVM_VERSION}" )
+  endif()
+
 
   foreach(user_lib ${USER_LIBS})
     target_link_libraries(${name} ${user_lib})
@@ -132,8 +138,8 @@ include("GNUInstallDirs")
 
 install(TARGETS clazylib EXPORT LibClaryExport
   RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
-  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
-  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  LIBRARY DESTINATION @LLVM_INSTALL_PREFIX@/lib
+  ARCHIVE DESTINATION @LLVM_INSTALL_PREFIX@/lib
 )
 
 set(SYMBOL_FILE Lazy.exports)
@@ -210,8 +216,8 @@ set_target_properties(ClangLazy PROPERTIES
 
 install(TARGETS ClangLazy
   RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
-  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
-  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  LIBRARY DESTINATION @LLVM_INSTALL_PREFIX@/lib
+  ARCHIVE DESTINATION @LLVM_INSTALL_PREFIX@/lib
 )
 
 set(SHARE_INSTALL_DIR ${CMAKE_INSTALL_DATAROOTDIR} CACHE STRING "Share directory name")
