diff --git a/clazy.cmake b/clazy.cmake
index 9166d8a..f77671e 100644
--- a/clazy.cmake
+++ b/clazy.cmake
@@ -64,4 +64,4 @@ then
   exit
 fi
 
-${CLANGXX:-clang++} -Qunused-arguments -Xclang -load -Xclang ClangLazy@CMAKE_SHARED_LIBRARY_SUFFIX@ -Xclang -add-plugin -Xclang clang-lazy $@
+xCLANGPPx -Qunused-arguments -Xclang -load -Xclang @CMAKE_INSTALL_LIBDIR@/ClangLazy@CMAKE_SHARED_LIBRARY_SUFFIX@ -Xclang -add-plugin -Xclang clang-lazy $@
