diff --git CMakeLists.txt CMakeLists.txt
index fec9e28..7659baf 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -30,7 +30,7 @@ ecm_setup_version(PROJECT VARIABLE_PREFIX KWAYLAND
                         SOVERSION 5)
 
 # Dependencies
-set(REQUIRED_QT_VERSION 5.12.0)
+set(REQUIRED_QT_VERSION 5.9.8)
 find_package(Qt5 ${REQUIRED_QT_VERSION} CONFIG REQUIRED Concurrent Gui WaylandClient)
 
 find_package(QtWaylandScanner REQUIRED)
