diff --git CMakeLists.txt CMakeLists.txt
index 2ec561c02dfbb2a92bb41fbbfb1db615e2d104ac..a16aa85b1a91bcabea344b52e734b215b2eba786 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -40,8 +40,10 @@ set_package_properties(Wayland PROPERTIES
 
 find_package(WaylandScanner)
 
-find_package(EGL)
-set_package_properties(EGL PROPERTIES TYPE REQUIRED)
+if (NOT APPLE)
+    find_package(EGL)
+    set_package_properties(EGL PROPERTIES TYPE REQUIRED)
+endif()
 
 include(KDEInstallDirs)
 include(KDEFrameworkCompilerSettings NO_POLICY_SCOPE)
diff --git src/server/CMakeLists.txt src/server/CMakeLists.txt
index 7566ef30f74d10f8305d036110888e2d7a0ff715..e8934dee5e5d64d33d1df08f17f641877290c412 100644
--- src/server/CMakeLists.txt
+++ src/server/CMakeLists.txt
@@ -278,9 +278,14 @@ target_link_libraries(KF5WaylandServer
     PUBLIC Qt5::Gui
     PRIVATE
         Wayland::Server
-        EGL::EGL
         Qt5::Concurrent
 )
+if (NOT APPLE)
+    target_link_libraries(KF5WaylandServer
+        PRIVATE
+            EGL::EGL
+    )
+endif()
 
 set_target_properties(KF5WaylandServer PROPERTIES VERSION   ${KWAYLAND_VERSION_STRING}
                                                  SOVERSION ${KWAYLAND_SOVERSION}
diff --git src/tools/CMakeLists.txt src/tools/CMakeLists.txt
index 791b8b30de672c0ca16563db7a499f70bdec3b3f..0baa91b167489300e422ed2da2de8ee6aa2af7d4 100644
--- src/tools/CMakeLists.txt
+++ src/tools/CMakeLists.txt
@@ -1,4 +1,4 @@
-add_subdirectory(testserver)
+# add_subdirectory(testserver)
 
 include(ECMMarkAsTest)
 
