diff --git CMakeLists.txt CMakeLists.txt
index f0a0cd7..ed7ca41 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -281,7 +281,7 @@ else(DEFAULT_PYTHON_VERSION)
 endif(DEFAULT_PYTHON_VERSION)
 
 add_subdirectory(tools)
-#add_subdirectory(docs)
+add_subdirectory(docs)
 add_subdirectory(examples)
 # Due to version specific ABI tagging in Python 3 so files, we can build for
 # all Python 3 versions without a problem.
