diff --git sip/solid/orig.predicate.sip sip/solid/predicate.sip
index b95c64f..20866c5 100644
--- sip/solid/orig.predicate.sip
+++ sip/solid/predicate.sip
@@ -112,7 +112,7 @@ Py_BEGIN_ALLOW_THREADS
 #else
         PyObject *obj = PyInt_FromLong ((long) value);
 #endif
-        if (obj == NULL || PyList_SET_ITEM (l, i, obj) < 0)
+        if (obj == NULL || PyList_SetItem (l, i, obj) < 0)
         {
             Py_DECREF(l);
 
diff --git sip/solid/orig.powermanagement.sip sip/solid/powermanagement.sip
index 43aeb31..e6c15fd 100644
--- sip/solid/orig.powermanagement.sip
+++ sip/solid/powermanagement.sip
@@ -83,7 +83,7 @@ bool                    stopSuppressingScreenPowerManagement (int cookie);
 #else
         PyObject *obj = PyInt_FromLong ((long) value);
 #endif
-        if (obj == NULL || PyList_SET_ITEM (l, i, obj) < 0)
+        if (obj == NULL || PyList_SetItem (l, i, obj) < 0)
         {
             Py_DECREF(l);
 
diff --git sip/phonon/orig.abstractaudiodataoutput.sip sip/phonon/abstractaudiodataoutput.sip
index afde08b..101e9b4 100644
--- sip/phonon/orig.abstractaudiodataoutput.sip
+++ sip/phonon/abstractaudiodataoutput.sip
@@ -72,7 +72,7 @@ protected:
     foreach (Phonon::Experimental::AudioFormat value, set)
     {
         PyObject *obj = PyInt_FromLong ((long) value);
-        if (obj == NULL || PyList_SET_ITEM (l, i, obj) < 0)
+        if (obj == NULL || PyList_SetItem (l, i, obj) < 0)
         {
             Py_DECREF(l);
 
@@ -118,4 +118,4 @@ protected:
     return sipGetState(sipTransferObj);
 %End
 };
-*/
\ No newline at end of file
+*/
diff --git sip/phonon/orig.abstractvideodataoutput.sip sip/phonon/abstractvideodataoutput.sip
index fdc9a75..6a035c3 100644
--- sip/phonon/orig.abstractvideodataoutput.sip
+++ sip/phonon/abstractvideodataoutput.sip
@@ -72,7 +72,7 @@ protected:
     foreach (Phonon::Experimental::VideoFrame2::Format value, set)
     {
         PyObject *obj = PyInt_FromLong ((long) value);
-        if (obj == NULL || PyList_SET_ITEM (l, i, obj) < 0)
+        if (obj == NULL || PyList_SetItem (l, i, obj) < 0)
         {
             Py_DECREF(l);
 
