diff --git sip/kdecore/orig.typedefs.sip sip/kdecore/typedefs.sip
index fb2e3b9..bb1b81b 100644
--- sip/kdecore/orig.typedefs.sip
+++ sip/kdecore/typedefs.sip
@@ -23,7 +23,7 @@ typedef uint mode_t;
 
 typedef long time_t;
 
-typedef ulong size_t;
+//typedef ulong size_t;
 
 typedef int ssize_t;
 
diff --git sip/kdecore/orig.kprotocolinfo.sip sip/kdecore/kprotocolinfo.sip
index f2b59d8..f922d3b 100644
--- sip/kdecore/orig.kprotocolinfo.sip
+++ sip/kdecore/kprotocolinfo.sip
@@ -151,5 +151,5 @@ public:
 
 
 
-QDataStream&            operator >> (QDataStream& s, KProtocolInfo::ExtraField& field);
-QDataStream&            operator << (QDataStream& s, const KProtocolInfo::ExtraField& field);
+//ig QDataStream&            operator >> (QDataStream& s, KProtocolInfo::ExtraField& field);
+//ig QDataStream&            operator << (QDataStream& s, const KProtocolInfo::ExtraField& field);
diff --git sip/kdecore/orig.kserviceaction.sip sip/kdecore/kserviceaction.sip
index fdda115..73df36b 100644
--- sip/kdecore/orig.kserviceaction.sip
+++ sip/kdecore/kserviceaction.sip
@@ -44,5 +44,5 @@ public:
 // KServiceAction
 
 
-QDataStream&            operator >> (QDataStream& str, KServiceAction& act);
-QDataStream&            operator << (QDataStream& str, const KServiceAction& act);
+//ig QDataStream&            operator >> (QDataStream& str, KServiceAction& act);
+//ig QDataStream&            operator << (QDataStream& str, const KServiceAction& act);
diff --git sip/kdecore/orig.klocalizeddate.sip sip/kdecore/klocalizeddate.sip
index 7f8ee2d..5176f18 100644
--- sip/kdecore/orig.klocalizeddate.sip
+++ sip/kdecore/klocalizeddate.sip
@@ -99,6 +99,6 @@ public:
     bool                    operator >= (const QDate& other) const;
     QString                 formatDate (KLocale::DateTimeComponent component, KLocale::DateTimeComponentFormat format = KLocale::DefaultComponentFormat, KLocale::WeekNumberSystem weekNumberSystem = KLocale::DefaultWeekNumber) const;
 };
-QDataStream             operator << (QDataStream& out, const KLocalizedDate& date);
-QDataStream             operator >> (QDataStream& in, KLocalizedDate& date);
+//ig QDataStream             operator << (QDataStream& out, const KLocalizedDate& date);
+//ig QDataStream             operator >> (QDataStream& in, KLocalizedDate& date);
 //ig QDebug                  operator << (QDebug debug, const KLocalizedDate& date);
