Index: pykde4-4.13.97/CMakeLists.txt
===================================================================
--- CMakeLists.txt.orig
+++ CMakeLists.txt
@@ -13,6 +13,9 @@ include(KDE4Defaults)
 include(MacroLibrary)
 include(PythonMacros)
 include(FeatureSummary)
+find_package(PkgConfig REQUIRED)
+# this doesn't work ...
+pkg_check_modules(PYTHON xpython${_CURRENT_VERSION})
 
 option(PYKDEUIC4_ALTINSTALL "Enable parallel-installation of the PyKDE4 tools" FALSE)
 
