--- src/qtui/orig.main_window.cc	2018-01-14 14:32:05.000000000 +0100
+++ src/qtui/main_window.cc	2018-01-15 00:30:58.000000000 +0100
@@ -115,7 +124,9 @@
 
     addToolBar (Qt::TopToolBarArea, new ToolBar (this, items));
 
-    setUnifiedTitleAndToolBarOnMac (true);
+    // this worked more or less in Qt4 but is not recommended in Qt5, not
+    // if you want to do OpenGL or have dockable widgets
+//     setUnifiedTitleAndToolBarOnMac (true);
 
     updateToggles ();
 
