--- src/qtui/orig.tool_bar.cc	2016-09-22 03:14:58.000000000 +0200
+++ src/qtui/tool_bar.cc	2018-01-14 14:38:56.000000000 +0100
@@ -21,6 +21,7 @@
 
 #include <QAction>
 #include <QIcon>
+#include <QDebug>
 
 #include <libaudcore/runtime.h>
 #include <libaudqt/libaudqt.h>
@@ -32,9 +33,10 @@
     setMovable (false);
     setObjectName ("MainToolBar");
 
-#if defined(Q_OS_WIN32) || defined(Q_OS_MAC)
-    setIconSize (QSize (22, 22));
-#endif
+    // use the native toolbar icon size (or what/however the user configured).
+// #if defined(Q_OS_WIN32) || defined(Q_OS_MAC)
+//     setIconSize (QSize (22, 22));
+// #endif
 
     for (const ToolBarItem & item : items)
     {
