diff --git src/lib/core/CMakeLists.txt src/lib/core/CMakeLists.txt
index f7636a8..56c8c17 100644
--- src/lib/core/CMakeLists.txt
+++ src/lib/core/CMakeLists.txt
@@ -85,6 +85,7 @@ target_link_libraries (
    PRIVATE
       Qt5::DBus
       KF5::DBusAddons
+      Qt5::QspXDG
    )
 
 target_include_directories (
diff --git src/service/CMakeLists.txt src/service/CMakeLists.txt
index 5c8ecc5..8b56f7c 100644
--- src/service/CMakeLists.txt
+++ src/service/CMakeLists.txt
@@ -22,7 +22,7 @@
 
 add_library(kactivitymanagerd_plugin SHARED Plugin.cpp Module.cpp Event.cpp Debug.cpp)
 generate_export_header(kactivitymanagerd_plugin)
-target_link_libraries(kactivitymanagerd_plugin PUBLIC Qt5::Core Qt5::DBus KF5::CoreAddons KF5::ConfigCore)
+target_link_libraries(kactivitymanagerd_plugin PUBLIC Qt5::Core Qt5::DBus KF5::CoreAddons KF5::ConfigCore PRIVATE Qt5::QspXDG)
 
 add_subdirectory (plugins)
 
@@ -60,6 +60,7 @@
 add_executable (kactivitymanagerd ${kactivitymanager_SRCS})
 
 target_link_libraries (kactivitymanagerd
+PUBLIC
    Qt5::Core
    Qt5::DBus
    Qt5::Gui
@@ -69,6 +70,8 @@
    KF5::I18n
    KF5::WindowSystem
    kactivitymanagerd_plugin
+PRIVATE
+   PRIVATE Qt5::QspXDG
    )
 
 ########### install application ###############
