diff --git a/src/libalkimia.doxygen.in b/src/libalkimia.doxygen.in
index a502acb..60d9094 100644
--- a/src/libalkimia.doxygen.in
+++ b/src/libalkimia.doxygen.in
@@ -884,7 +884,7 @@ TOC_EXPAND             = NO
 # Qt's qhelpgenerator to generate a Qt Compressed Help (.qch) of the generated 
 # HTML documentation.
 
-GENERATE_QHP           = NO
+GENERATE_QHP           = yes
 
 # If the QHG_LOCATION tag is specified, the QCH_FILE tag can 
 # be used to specify the file name of the resulting .qch file. 
@@ -896,7 +896,7 @@ QCH_FILE               =
 # Qt Help Project output. For more information please see 
 # http://doc.trolltech.com/qthelpproject.html#namespace
 
-QHP_NAMESPACE          = 
+QHP_NAMESPACE          = org.kde.kmymoney
 
 # The QHP_VIRTUAL_FOLDER tag specifies the namespace to use when generating 
 # Qt Help Project output. For more information please see 
@@ -926,7 +926,7 @@ QHP_SECT_FILTER_ATTRS  =
 # If non-empty doxygen will try to run qhelpgenerator on the generated 
 # .qhp file.
 
-QHG_LOCATION           = 
+QHG_LOCATION           = @@QTBINSDIR@@/qhelpgenerator
 
 # The DISABLE_INDEX tag can be used to turn on/off the condensed index at 
 # top of each HTML page. The value NO (the default) enables the index and 
