diff --git src/script/data/indentation/cstyle.js src/script/data/indentation/cstyle.js
index 88fa611778ee114c85f4b9c4a58649ffdb8b98d3..500037f592c20e707c64413809e6f7641c16d494 100644
--- src/script/data/indentation/cstyle.js
+++ src/script/data/indentation/cstyle.js
@@ -514,6 +514,8 @@ function tryCondition(line)
                 if (currentString.search(/^\s*(if\b|[}]?\s*else|do\b|while\b|for)[^{]*$/) != -1)
                     indentation = firstPosVirtual;
                 break;
+            } else if (currentLine == 0 || lineDelimiter == 0) {
+                return indentation;
             }
         }
     }
diff --git src/utils/kateglobal.cpp src/utils/kateglobal.cpp
index b5d56e3b51f2fcec74c42e7db98f17840ad9693f..f1b0c5582e410858a17026c41831979bbb24e174 100644
--- src/utils/kateglobal.cpp
+++ src/utils/kateglobal.cpp
@@ -99,6 +99,15 @@ KTextEditor::EditorPrivate::EditorPrivate(QPointer<KTextEditor::EditorPrivate> &
     // remember this
     staticInstance = this;
 
+#if QT_VERSION < QT_VERSION_CHECK(5, 9, 1)
+    // disable the QML JIT compiler as a protection against an unknown bug
+    // in Qt's V4 engine which can provoke a crash in certain of our scripts.
+    // See https://bugreports.qt.io/browse/QTBUG-63045
+    // and https://bugs.kde.org/show_bug.cgi?id=385413
+    qputenv("QV4_FORCE_INTERPRETER", QByteArrayLiteral("1"));
+    qCDebug(LOG_KTE) << "QV4_FORCE_INTERPRETER set to 1";
+#endif
+
     // init libgit2, we require at least 0.22 which has this function!
 #if LIBGIT2_FOUND
     git_libgit2_init();
