diff --git src/view/kateview.cpp src/view/kateview.cpp
index fb865a4..48e19a6 100644
--- src/view/kateview.cpp
+++ src/view/kateview.cpp
@@ -2953,10 +2953,8 @@ QMenu *KTextEditor::ViewPrivate::contextMenu() const
                 if (w->objectName() == QLatin1String("ktexteditor_popup")) {
                     //perhaps optimize this block
                     QMenu *menu = (QMenu *)w;
-                    disconnect(menu, SIGNAL(aboutToShow()), this, SLOT(aboutToShowContextMenu()));
-                    disconnect(menu, SIGNAL(aboutToHide()), this, SLOT(aboutToHideContextMenu()));
-                    connect(menu, SIGNAL(aboutToShow()), this, SLOT(aboutToShowContextMenu()));
-                    connect(menu, SIGNAL(aboutToHide()), this, SLOT(aboutToHideContextMenu()));
+                    KTextEditor::ViewPrivate *self = const_cast<KTextEditor::ViewPrivate *>(this);
+                    self->setContextMenu(menu);
                     return menu;
                 }
             }
@@ -2997,6 +2995,8 @@ void KTextEditor::ViewPrivate::aboutToShowContextMenu()
 
     if (menu) {
         emit contextMenuAboutToShow(this, menu);
+    } else {
+        qWarning() << Q_FUNC_INFO << "cannot cast" << sender() << "to QMenu";
     }
 }
 

