diff --git libavcodec/videotoolboxenc.c libavcodec/orig.videotoolboxenc.c
index 58dd9f1..d1ba083 100644
--- libavcodec/videotoolboxenc.c
+++ libavcodec/orig.videotoolboxenc.c
@@ -1176,6 +1176,7 @@ static int vtenc_create_encoder(AVCodecContext   *avctx,
             return AVERROR_EXTERNAL;
         }
     }
+#endif
 
     // Dump the init encoder
     {
@@ -1204,7 +1205,6 @@ static int vtenc_create_encoder(AVCodecContext   *avctx,
         if (encoderID != NULL)
             CFRelease(encoderID);
     }
-#endif
 
     if (avctx->flags & AV_CODEC_FLAG_QSCALE && !vtenc_qscale_enabled()) {
         av_log(avctx, AV_LOG_ERROR, "Error: -q:v qscale not available for encoder. Use -b:v bitrate instead.\n");
