diff --git libavcodec/orig.librsvgdec.c libavcodec/librsvgdec.c
index c328fbc..756c26d 100644
--- libavcodec/orig.librsvgdec.c
+++ libavcodec/librsvgdec.c
@@ -90,8 +90,10 @@ static int librsvg_decode_frame(AVCodecContext *avctx, AVFrame *frame,
         goto end;
 
     avctx->pix_fmt = AV_PIX_FMT_RGB32;
+#if LIBRSVG_MAJOR_VERSION > 2 || LIBRSVG_MAJOR_VERSION == 2 && LIBRSVG_MINOR_VERSION >= 52
     viewport.width = dimensions.width;
     viewport.height = dimensions.height;
+#endif
 
     ret = ff_get_buffer(avctx, frame, 0);
     if (ret < 0)
