--- cmake/modules/orig.LXQtCompilerSettings.cmake	2018-05-21 19:06:51.000000000 +0200
+++ cmake/modules/LXQtCompilerSettings.cmake	2019-01-18 23:26:34.000000000 +0100
@@ -130,7 +130,7 @@
 # Linker flags
 # Do not allow undefined symbols
 #-----------------------------------------------------------------------------
-if (CMAKE_COMPILER_IS_GNUCXX OR LXQT_COMPILER_IS_CLANGCXX)
+if ((CMAKE_COMPILER_IS_GNUCXX OR LXQT_COMPILER_IS_CLANGCXX) AND NOT APPLE)
     # -Bsymbolic-functions: replace dynamic symbols used internally in
     #                       shared libs with direct addresses.
     set(SYMBOLIC_FLAGS
