diff --git a/cmake/modules/LXQtCompilerSettings.cmake b/cmake/modules/LXQtCompilerSettings.cmake
index acba891..c2fa2ae 100644
--- a/cmake/modules/LXQtCompilerSettings.cmake
+++ b/cmake/modules/LXQtCompilerSettings.cmake
@@ -156,7 +156,7 @@ endif()
 #-----------------------------------------------------------------------------
 if (CMAKE_COMPILER_IS_GNUCXX OR LXQT_COMPILER_IS_CLANGCXX)
     # Assume Xcode Clang is always used with Apple ld64
-    if (${CMAKE_CXX_COMPILER_ID} STREQUAL AppleClang)
+    if (APPLE)
         set(NO_UNDEFINED_FLAGS "-Wl,-undefined,error")
         # -Bsymbolic* make sense for ELF only
         set(SYMBOLIC_FLAGS "")
