diff --git a/CMakeLists.txt b/CMakeLists.txt
index ff3af7c..e2eaa69 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -12,7 +12,7 @@ set(LXQT_BUILD_TOOLS_MAJOR_VERSION 0)
 set(LXQT_BUILD_TOOLS_MINOR_VERSION 13)
 set(LXQT_BUILD_TOOLS_PATCH_VERSION 0)
 set(LXQT_BUILD_TOOLS_VERSION ${LXQT_BUILD_TOOLS_MAJOR_VERSION}.${LXQT_BUILD_TOOLS_MINOR_VERSION}.${LXQT_BUILD_TOOLS_PATCH_VERSION})
-set(LXQT_MIN_LINGUIST_VERSION "5.15.0")
+set(LXQT_MIN_LINGUIST_VERSION "5.9.8")
 
 include(CMakePackageConfigHelpers)
 include(GNUInstallDirs) # Standard directories for installation
