diff --git src/plugins/coreplugin/coreplugin.pro src/plugins/coreplugin/coreplugin.pro
index 239b0dc..04655af 100644
--- src/plugins/coreplugin/coreplugin.pro
+++ src/plugins/coreplugin/coreplugin.pro
@@ -263,7 +263,7 @@ win32 {
 else:macx {
     OBJECTIVE_SOURCES += \
         progressmanager/progressmanager_mac.mm
-    LIBS += -framework AppKit
+    LIBS += -framework AppKit -framework CoreServices
 }
 else:unix {
     SOURCES += progressmanager/progressmanager_x11.cpp
