--- src/plugins/coreplugin/actionmanager/orig.actionmanager.cpp	2015-09-24 16:35:49.963708197 +0200
+++ src/plugins/coreplugin/actionmanager/actionmanager.cpp	2015-09-24 16:36:06.845707758 +0200
@@ -225,6 +225,7 @@
     if (!QGuiApplication::platformName().contains(QLatin1String("cocoa"))) {
         QCoreApplication::setAttribute(Qt::AA_DontUseNativeMenuBar);
     }
+    QCoreApplication::setAttribute(Qt::AA_DontShowIconsInMenus);
     QMenuBar *mb = new QMenuBar; // No parent (System menu bar on Mac OS X)
     mb->setObjectName(id.toString());
 
