diff --git src/plugins/coreplugin/windowsupport.cpp src/plugins/coreplugin/windowsupport.cpp
--- src/plugins/coreplugin/windowsupport.cpp
+++ src/plugins/coreplugin/windowsupport.cpp
@@ -43,6 +43,9 @@
 #include <QEvent>
 #include <QMenu>
 #include <QWidget>
+#ifdef Q_OS_OSX
+#include <QGuiApplication>
+#endif
 
 namespace Core {
 namespace Internal {
@@ -153,7 +156,7 @@ void WindowSupport::updateFullScreenAction()
 void WindowList::addWindow(QWidget *window)
 {
 #ifdef Q_OS_OSX
-    if (!m_dockMenu) {
+    if (!m_dockMenu && QGuiApplication::platformName().contains(QLatin1String("cocoa"))) {
         m_dockMenu = new QMenu;
         m_dockMenu->setAsDockMenu();
     }
