diff --git a/init-repository b/init-repository
index b38a07e..7cf923e 100755
--- a/init-repository
+++ b/init-repository
@@ -204,7 +204,6 @@ my %PROTOCOLS = (
 );
 
 my %GERRIT_REPOS = map { $_ => "qt/$_" } qw(
-    qt3d
     qt5
     qtactiveqt
     qtandroidextras
@@ -212,28 +211,19 @@ my %GERRIT_REPOS = map { $_ => "qt/$_" } qw(
     qtconnectivity
     qtdeclarative
     qtdoc
-    qtdocgallery
-    qtenginio
-    qtfeedback
     qtgraphicaleffects
     qtimageformats
-    qtjsondb
     qtlocation
     qtmacextras
     qtmultimedia
-    qtpim
-    qtqa
     qtquick1
     qtquickcontrols
-    qtrepotools
     qtscript
     qtsensors
     qtserialport
     qtsvg
-    qtsystems
     qttools
     qttranslations
-    qtwayland
     qtwebkit
     qtwebkit-examples
     qtwebsockets
@@ -249,25 +239,22 @@ my @DEFAULT_REPOS = qw(
     qtconnectivity
     qtdeclarative
     qtdoc
-    qtenginio
     qtgraphicaleffects
     qtimageformats
+    qtlocation
     qtmacextras
     qtmultimedia
-    qtqa
     qtquick1
     qtquickcontrols
-    qtlocation
-    qtrepotools
     qtscript
     qtsensors
     qtserialport
     qtsvg
     qttools
     qttranslations
-    qtwebsockets
     qtwebkit
     qtwebkit-examples
+    qtwebsockets
     qtwinextras
     qtx11extras
     qtxmlpatterns
@@ -548,38 +535,38 @@ sub git_clone_one_submodule
 
     my $do_clone = (! -d "$submodule/.git");
     if ($do_clone) {
-        $self->exe('git', 'clone', @reference_args, ($mirror ? $mirror : $url), $submodule);
-    }
-
-    chdir($submodule) or confess "chdir $submodule: $OS_ERROR";
-
-    $self->exe('git', 'config', 'remote.origin.url', $url);
-    if ($mirror) {
-        $self->exe('git', 'config', 'remote.mirror.url', $mirror);
-        $self->exe('git', 'config', 'remote.mirror.fetch', '+refs/heads/*:refs/remotes/mirror/*');
-    }
-
-    if (!$do_clone) {
-        $self->exe('git', 'fetch', ($mirror ? $mirror : $url));
-    }
-
-    my $template = getcwd()."/../.commit-template";
-    if (-e $template) {
-        $self->exe('git', 'config', 'commit.template', $template);
-    }
-
-    $self->git_add_remotes($submodule);
-
-    if ($self->{'detach-alternates'}) {
-        $self->exe('git', 'repack', '-a');
-
-        my $alternates_path = '.git/objects/info/alternates';
-        if (-e $alternates_path) {
-            unlink($alternates_path) || confess "unlink $alternates_path: $OS_ERROR";
-        }
+        $self->exe('git', 'clone', '-b', 'v5.3.2', '--depth=1', ($mirror ? $mirror : $url), $submodule);
     }
 
-    chdir("..") or confess "cd ..: $OS_ERROR";
+#     chdir($submodule) or confess "chdir $submodule: $OS_ERROR";
+# 
+#     $self->exe('git', 'config', 'remote.origin.url', $url);
+#     if ($mirror) {
+#         $self->exe('git', 'config', 'remote.mirror.url', $mirror);
+#         $self->exe('git', 'config', 'remote.mirror.fetch', '+refs/heads/*:refs/remotes/mirror/*');
+#     }
+# 
+#     if (!$do_clone) {
+#         $self->exe('git', 'fetch', ($mirror ? $mirror : $url));
+#     }
+# 
+#     my $template = getcwd()."/../.commit-template";
+#     if (-e $template) {
+#         $self->exe('git', 'config', 'commit.template', $template);
+#     }
+# 
+#     $self->git_add_remotes($submodule);
+# 
+#     if ($self->{'detach-alternates'}) {
+#         $self->exe('git', 'repack', '-a');
+# 
+#         my $alternates_path = '.git/objects/info/alternates';
+#         if (-e $alternates_path) {
+#             unlink($alternates_path) || confess "unlink $alternates_path: $OS_ERROR";
+#         }
+#     }
+# 
+#     chdir("..") or confess "cd ..: $OS_ERROR";
 
     return;
 }
