diff --git a/ocl_icd_wrapper.c b/ocl_icd_wrapper.c
index 7cb5b55..ff9b70e 100644
--- a/ocl_icd_wrapper.c
+++ b/ocl_icd_wrapper.c
@@ -71,6 +71,14 @@ clGetExtensionFunctionAddress(const char *funcname) CL_API_SUFFIX__VERSION_1_2
   }
 }
 
+CL_API_ENTRY cl_int CL_API_CALL
+_clGetPlatformIDs_(cl_uint            num_entries,
+                    cl_platform_id    *platforms,
+                    cl_uint           *num_platforms) CL_API_SUFFIX__VERSION_1_0
+{
+  return clIcdGetPlatformIDsKHR(num_entries, platforms, num_platforms);
+}
+
 CL_API_ENTRY cl_int CL_API_CALL
 _clGetPlatformInfo_(cl_platform_id    platform,
                     cl_platform_info  param_name,
@@ -98,13 +106,48 @@ _clGetPlatformInfo_(cl_platform_id    platform,
   }
   else
   {
-    return clGetPlatformInfo(
+//     fprintf(stderr, "ocd_icd_wrapper::_clGetPlatformInfo(%p->%p,0x%x,%lu,%p,%p)\n",
+//     	 platform, platform->platform, param_name, param_value_size, param_value, param_value_size_ret);
+    cl_int ret = clGetPlatformInfo(
       platform->platform,
       param_name,
       param_value_size,
       param_value,
       param_value_size_ret
     );
+#ifdef __APPLE__
+    if (ret == CL_SUCCESS && param_name == CL_PLATFORM_EXTENSIONS)
+    {
+      static int len, ourLen = 0;
+      const char *missingExt = " cl_khr_icd";
+      if (param_value_size == 0 && !param_value && param_value_size_ret)
+      {
+        if (!ourLen) {
+          len = strlen(missingExt);
+        }
+        ourLen = (*param_value_size_ret += len);
+//         fprintf(stderr, "\tbumped the return value by %d to %zu\n", len, *param_value_size_ret);
+      }
+      if (param_value_size && param_value_size < ourLen)
+      {
+        return CL_INVALID_VALUE;
+      }
+      if (param_value)
+      {
+        char *exts = (char*) param_value;
+        if (!strstr(exts, missingExt))
+        {
+          strcat(exts, missingExt);
+//           fprintf(stderr, "\tappended%s to the extensions: \"%s\"\n", missingExt, exts);
+        }
+        if (param_value_size_ret)
+        {
+          *param_value_size_ret = strlen(param_value);
+        }
+      }
+    }
+#endif
+    return ret;
   }
 }
 
@@ -3376,7 +3479,7 @@ KHRicdVendorDispatch* createDispatchTable()
 #define DISPATCH_TABLE_ENTRY(fn) table->fn = _##fn##_;
 
   // OpenCL 1.0
-  //DISPATCH_TABLE_ENTRY(clGetPlatformIDs);
+  DISPATCH_TABLE_ENTRY(clGetPlatformIDs);
   DISPATCH_TABLE_ENTRY(clGetPlatformInfo);
   DISPATCH_TABLE_ENTRY(clGetDeviceIDs);
   DISPATCH_TABLE_ENTRY(clGetDeviceInfo);
