diff --git a/icd_dispatch.h b/icd_dispatch.h
index 57211eb..a556e5f 100644
--- a/icd_dispatch.h
+++ b/icd_dispatch.h
@@ -629,14 +629,14 @@ typedef CL_API_ENTRY cl_int (CL_API_CALL *KHRpfn_clEnqueueBarrierWithWaitList)(
 
 typedef CL_API_ENTRY void * (CL_API_CALL *KHRpfn_clGetExtensionFunctionAddressForPlatform)(
     cl_platform_id platform,
-    const char *   function_name) CL_API_SUFFIX__VERSION_1_2;
+    const char *   function_name);
 
 // Deprecated APIs
 typedef CL_API_ENTRY cl_int (CL_API_CALL *KHRpfn_clSetCommandQueueProperty)(
     cl_command_queue              command_queue,
     cl_command_queue_properties   properties,
     cl_bool                       enable,
-    cl_command_queue_properties * old_properties) CL_EXT_SUFFIX__VERSION_1_0_DEPRECATED;
+    cl_command_queue_properties * old_properties); // CL_EXT_SUFFIX__VERSION_1_0_DEPRECATED;
 
 typedef CL_API_ENTRY cl_mem (CL_API_CALL *KHRpfn_clCreateImage2D)(
     cl_context              context,
@@ -646,7 +646,7 @@ typedef CL_API_ENTRY cl_mem (CL_API_CALL *KHRpfn_clCreateImage2D)(
     size_t                  image_height,
     size_t                  image_row_pitch,
     void *                  host_ptr,
-    cl_int *                errcode_ret) CL_EXT_SUFFIX__VERSION_1_1_DEPRECATED;
+    cl_int *                errcode_ret); // CL_EXT_SUFFIX__VERSION_1_1_DEPRECATED;
 
 typedef CL_API_ENTRY cl_mem (CL_API_CALL *KHRpfn_clCreateImage3D)(
     cl_context              context,
@@ -658,22 +658,22 @@ typedef CL_API_ENTRY cl_mem (CL_API_CALL *KHRpfn_clCreateImage3D)(
     size_t                  image_row_pitch,
     size_t                  image_slice_pitch,
     void *                  host_ptr,
-    cl_int *                errcode_ret) CL_EXT_SUFFIX__VERSION_1_1_DEPRECATED;
+    cl_int *                errcode_ret); // CL_EXT_SUFFIX__VERSION_1_1_DEPRECATED;
 
-typedef CL_API_ENTRY cl_int (CL_API_CALL *KHRpfn_clUnloadCompiler)(void) CL_EXT_SUFFIX__VERSION_1_1_DEPRECATED;
+typedef CL_API_ENTRY cl_int (CL_API_CALL *KHRpfn_clUnloadCompiler)(void); // CL_EXT_SUFFIX__VERSION_1_1_DEPRECATED;
 
 typedef CL_API_ENTRY cl_int (CL_API_CALL *KHRpfn_clEnqueueMarker)(
     cl_command_queue    command_queue,
-    cl_event *          event) CL_EXT_SUFFIX__VERSION_1_1_DEPRECATED;
+    cl_event *          event);// CL_EXT_SUFFIX__VERSION_1_1_DEPRECATED;
 
 typedef CL_API_ENTRY cl_int (CL_API_CALL *KHRpfn_clEnqueueWaitForEvents)(
     cl_command_queue command_queue,
     cl_uint          num_events,
-    const cl_event * event_list) CL_EXT_SUFFIX__VERSION_1_1_DEPRECATED;
+    const cl_event * event_list); // CL_EXT_SUFFIX__VERSION_1_1_DEPRECATED;
 
-typedef CL_API_ENTRY cl_int (CL_API_CALL *KHRpfn_clEnqueueBarrier)(cl_command_queue command_queue) CL_EXT_SUFFIX__VERSION_1_1_DEPRECATED;
+typedef CL_API_ENTRY cl_int (CL_API_CALL *KHRpfn_clEnqueueBarrier)(cl_command_queue command_queue); // CL_EXT_SUFFIX__VERSION_1_1_DEPRECATED;
 
-typedef CL_API_ENTRY void * (CL_API_CALL *KHRpfn_clGetExtensionFunctionAddress)(const char *function_name) CL_EXT_SUFFIX__VERSION_1_1_DEPRECATED;
+typedef CL_API_ENTRY void * (CL_API_CALL *KHRpfn_clGetExtensionFunctionAddress)(const char *function_name); // CL_EXT_SUFFIX__VERSION_1_1_DEPRECATED;
 
 // GL and other APIs
 typedef CL_API_ENTRY cl_mem (CL_API_CALL *KHRpfn_clCreateFromGLBuffer)(
diff --git a/ocl_icd_wrapper.c b/ocl_icd_wrapper.c
index 7cb5b55..906a8b4 100644
--- a/ocl_icd_wrapper.c
+++ b/ocl_icd_wrapper.c
@@ -59,7 +59,7 @@ clIcdGetPlatformIDsKHR(cl_uint num_entries,
 }
 
 CL_API_ENTRY void* CL_API_CALL
-clGetExtensionFunctionAddress(const char *funcname) CL_API_SUFFIX__VERSION_1_2
+clGetExtensionFunctionAddress(const char *funcname) 
 {
   if (strcmp(funcname, "clIcdGetPlatformIDsKHR") == 0)
   {
@@ -142,6 +142,8 @@ _clGetDeviceIDs_(cl_platform_id   platform,
       devices[i]->platform = platform;
       devices[i]->device = _devices[i];
     }
+  }
+  if (_devices) {
     free(_devices);
   }
 
@@ -211,7 +213,7 @@ _clReleaseDevice_(cl_device_id device) CL_API_SUFFIX__VERSION_1_2
 }
 
 // Utility function to create a list of real devices
-cl_device_id* createDeviceList(cl_uint num, const cl_device_id *list)
+static cl_device_id* createDeviceList(cl_uint num, const cl_device_id *list)
 {
   cl_device_id *devices = NULL;
   if (list && num)
@@ -397,7 +399,9 @@ _clCreateContextFromType_(const cl_context_properties * properties,
       context->devices[i]->device = _devices[i];
       context->devices[i]->platform = m_platform;
     }
-    free(_devices);
+    if (_devices) {
+      free(_devices);
+    }
   }
 
   if (errcode_ret)
@@ -525,7 +529,7 @@ CL_API_ENTRY cl_int CL_API_CALL
 _clSetCommandQueueProperty_(cl_command_queue               command_queue ,
                             cl_command_queue_properties    properties ,
                             cl_bool                        enable ,
-                            cl_command_queue_properties *  old_properties) //CL_EXT_SUFFIX__VERSION_1_0_DEPRECATED
+                            cl_command_queue_properties *  old_properties) CL_EXT_SUFFIX__VERSION_1_0_DEPRECATED
 {
   return CL_INVALID_OPERATION;
 }
@@ -721,7 +725,7 @@ _clCreateImage2D_(cl_context              context ,
                   size_t                  image_height ,
                   size_t                  image_row_pitch ,
                   void *                  host_ptr ,
-                  cl_int *                errcode_ret) CL_API_SUFFIX__VERSION_1_0
+                  cl_int *                errcode_ret) CL_EXT_SUFFIX__VERSION_1_1_DEPRECATED
 {
   cl_image_desc desc =
   {
@@ -756,7 +760,7 @@ _clCreateImage3D_(cl_context              context,
                   size_t                  image_row_pitch ,
                   size_t                  image_slice_pitch ,
                   void *                  host_ptr ,
-                  cl_int *                errcode_ret) CL_API_SUFFIX__VERSION_1_0
+                  cl_int *                errcode_ret) CL_EXT_SUFFIX__VERSION_1_1_DEPRECATED
 {
   cl_image_desc desc =
   {
@@ -1142,7 +1146,9 @@ _clBuildProgram_(cl_program            program ,
     pfn_notify,
     user_data
   );
-  free(buildOptions);
+  if (buildOptions) {
+    free(buildOptions);
+  }
 
   if (_devices)
   {
@@ -1152,7 +1158,7 @@ _clBuildProgram_(cl_program            program ,
 }
 
 CL_API_ENTRY cl_int CL_API_CALL
-_clUnloadCompiler_(void) CL_API_SUFFIX__VERSION_1_0
+_clUnloadCompiler_(void) CL_EXT_SUFFIX__VERSION_1_1_DEPRECATED
 {
   return clUnloadCompiler();
 }
@@ -1200,7 +1206,9 @@ _clCompileProgram_(cl_program            program ,
     pfn_notify,
     user_data
   );
-  free(buildOptions);
+  if (buildOptions) {
+    free(buildOptions);
+  }
 
   if (_devices)
   {
@@ -1498,6 +1506,9 @@ _clSetKernelArg_(cl_kernel     kernel ,
     value
   );
 
+  if (type) {
+    free(type);
+  }
   return err;
 }
 
@@ -1819,6 +1830,8 @@ _clEnqueueReadBuffer_(cl_command_queue     command_queue ,
     (*event)->event = *_event;
     (*event)->context = command_queue->context;
     (*event)->queue = command_queue;
+  }
+  if (_event) {
     free(_event);
   }
   if (_wait_list)
@@ -1882,6 +1895,8 @@ _clEnqueueReadBufferRect_(cl_command_queue     command_queue ,
     (*event)->event = *_event;
     (*event)->context = command_queue->context;
     (*event)->queue = command_queue;
+  }
+  if (_event) {
     free(_event);
   }
   if (_wait_list)
@@ -1935,6 +1950,8 @@ _clEnqueueWriteBuffer_(cl_command_queue    command_queue ,
     (*event)->event = *_event;
     (*event)->context = command_queue->context;
     (*event)->queue = command_queue;
+  }
+  if (_event) {
     free(_event);
   }
   if (_wait_list)
@@ -1998,6 +2015,8 @@ _clEnqueueWriteBufferRect_(cl_command_queue     command_queue ,
     (*event)->event = *_event;
     (*event)->context = command_queue->context;
     (*event)->queue = command_queue;
+  }
+  if (_event) {
     free(_event);
   }
   if (_wait_list)
@@ -2051,6 +2070,8 @@ _clEnqueueCopyBuffer_(cl_command_queue     command_queue ,
     (*event)->event = *_event;
     (*event)->context = command_queue->context;
     (*event)->queue = command_queue;
+  }
+  if (_event) {
     free(_event);
   }
   if (_wait_list)
@@ -2112,6 +2133,8 @@ _clEnqueueCopyBufferRect_(cl_command_queue     command_queue ,
     (*event)->event = *_event;
     (*event)->context = command_queue->context;
     (*event)->queue = command_queue;
+  }
+  if (_event) {
     free(_event);
   }
   if (_wait_list)
@@ -2165,6 +2188,8 @@ _clEnqueueFillBuffer_(cl_command_queue    command_queue ,
     (*event)->event = *_event;
     (*event)->context = command_queue->context;
     (*event)->queue = command_queue;
+  }
+  if (_event) {
     free(_event);
   }
   if (_wait_list)
@@ -2216,6 +2241,8 @@ _clEnqueueFillImage_(cl_command_queue    command_queue ,
     (*event)->event = *_event;
     (*event)->context = command_queue->context;
     (*event)->queue = command_queue;
+  }
+  if (_event) {
     free(_event);
   }
   if (_wait_list)
@@ -2273,6 +2300,8 @@ _clEnqueueReadImage_(cl_command_queue      command_queue ,
     (*event)->event = *_event;
     (*event)->context = command_queue->context;
     (*event)->queue = command_queue;
+  }
+  if (_event) {
     free(_event);
   }
   if (_wait_list)
@@ -2330,6 +2359,8 @@ _clEnqueueWriteImage_(cl_command_queue     command_queue ,
     (*event)->event = *_event;
     (*event)->context = command_queue->context;
     (*event)->queue = command_queue;
+  }
+  if (_event) {
     free(_event);
   }
   if (_wait_list)
@@ -2383,6 +2414,8 @@ _clEnqueueCopyImage_(cl_command_queue      command_queue ,
     (*event)->event = *_event;
     (*event)->context = command_queue->context;
     (*event)->queue = command_queue;
+  }
+  if (_event) {
     free(_event);
   }
   if (_wait_list)
@@ -2436,6 +2469,8 @@ _clEnqueueCopyImageToBuffer_(cl_command_queue  command_queue ,
     (*event)->event = *_event;
     (*event)->context = command_queue->context;
     (*event)->queue = command_queue;
+  }
+  if (_event) {
     free(_event);
   }
   if (_wait_list)
@@ -2489,6 +2524,8 @@ _clEnqueueCopyBufferToImage_(cl_command_queue  command_queue ,
     (*event)->event = *_event;
     (*event)->context = command_queue->context;
     (*event)->queue = command_queue;
+  }
+  if (_event) {
     free(_event);
   }
   if (_wait_list)
@@ -2509,7 +2546,7 @@ _clEnqueueMapBuffer_(cl_command_queue  command_queue ,
                      cl_uint           num_events_in_wait_list ,
                      const cl_event *  event_wait_list ,
                      cl_event *        event ,
-                     cl_int *          errcode_ret) CL_API_SUFFIX__VERSION_1_0
+                     cl_int *          errcode_ret) CL_API_SUFFIX__VERSION_1_0 //CL_EXT_SUFFIX__VERSION_1_1_DEPRECATED
 {
   // Initialize event arguments
   cl_event *_wait_list = createEventList(
@@ -2545,6 +2582,8 @@ _clEnqueueMapBuffer_(cl_command_queue  command_queue ,
     (*event)->event = *_event;
     (*event)->context = command_queue->context;
     (*event)->queue = command_queue;
+  }
+  if (_event) {
     free(_event);
   }
   if (_wait_list)
@@ -2608,6 +2647,8 @@ _clEnqueueMapImage_(cl_command_queue   command_queue ,
     (*event)->event = *_event;
     (*event)->context = command_queue->context;
     (*event)->queue = command_queue;
+  }
+  if (_event) {
     free(_event);
   }
   if (_wait_list)
@@ -2658,6 +2699,8 @@ _clEnqueueUnmapMemObject_(cl_command_queue  command_queue ,
     (*event)->event = *_event;
     (*event)->context = command_queue->context;
     (*event)->queue = command_queue;
+  }
+  if (_event) {
     free(_event);
   }
   if (_wait_list)
@@ -2722,6 +2765,8 @@ _clEnqueueMigrateMemObjects_(cl_command_queue        command_queue ,
     (*event)->event = *_event;
     (*event)->context = command_queue->context;
     (*event)->queue = command_queue;
+  }
+  if (_event) {
     free(_event);
   }
   if (_wait_list)
@@ -2775,6 +2820,8 @@ _clEnqueueNDRangeKernel_(cl_command_queue  command_queue ,
     (*event)->event = *_event;
     (*event)->context = command_queue->context;
     (*event)->queue = command_queue;
+  }
+  if (_event) {
     free(_event);
   }
   if (_wait_list)
@@ -2820,6 +2867,8 @@ _clEnqueueTask_(cl_command_queue   command_queue ,
     (*event)->event = *_event;
     (*event)->context = command_queue->context;
     (*event)->queue = command_queue;
+  }
+  if (_event) {
     free(_event);
   }
   if (_wait_list)
@@ -2886,6 +2935,8 @@ _clEnqueueMarkerWithWaitList_(cl_command_queue  command_queue ,
     (*event)->event = *_event;
     (*event)->context = command_queue->context;
     (*event)->queue = command_queue;
+  }
+  if (_event) {
     free(_event);
   }
   if (_wait_list)
@@ -2929,6 +2980,8 @@ _clEnqueueBarrierWithWaitList_(cl_command_queue  command_queue ,
     (*event)->event = *_event;
     (*event)->context = command_queue->context;
     (*event)->queue = command_queue;
+  }
+  if (_event) {
     free(_event);
   }
   if (_wait_list)
@@ -2965,7 +3018,7 @@ _clEnqueueMarker_(cl_command_queue     command_queue ,
 CL_API_ENTRY cl_int CL_API_CALL
 _clEnqueueWaitForEvents_(cl_command_queue  command_queue ,
                          cl_uint           num_events ,
-                         const cl_event *  event_list) CL_API_SUFFIX__VERSION_1_0
+                         const cl_event *  event_list) CL_EXT_SUFFIX__VERSION_1_1_DEPRECATED
 {
   cl_event *_events = createEventList(num_events, event_list);
   cl_int err = clEnqueueWaitForEvents(
@@ -2978,13 +3031,13 @@ _clEnqueueWaitForEvents_(cl_command_queue  command_queue ,
 }
 
 CL_API_ENTRY cl_int CL_API_CALL
-_clEnqueueBarrier_(cl_command_queue  command_queue) CL_API_SUFFIX__VERSION_1_0
+_clEnqueueBarrier_(cl_command_queue  command_queue) CL_EXT_SUFFIX__VERSION_1_1_DEPRECATED
 {
   return clEnqueueBarrier(command_queue->queue);
 }
 
 CL_API_ENTRY void* CL_API_CALL
-_clGetExtensionFunctionAddress_(const char *funcname) CL_API_SUFFIX__VERSION_1_2
+_clGetExtensionFunctionAddress_(const char *funcname) CL_EXT_SUFFIX__VERSION_1_1_DEPRECATED
 {
   return NULL;
 }
@@ -3244,6 +3297,8 @@ _clEnqueueAcquireGLObjects_(cl_command_queue       command_queue,
     (*event)->event = *_event;
     (*event)->context = command_queue->context;
     (*event)->queue = command_queue;
+  }
+  if (_event) {
     free(_event);
   }
   if (_wait_list)
@@ -3299,6 +3354,8 @@ _clEnqueueReleaseGLObjects_(cl_command_queue       command_queue,
     (*event)->event = *_event;
     (*event)->context = command_queue->context;
     (*event)->queue = command_queue;
+  }
+  if (_event) {
     free(_event);
   }
   if (_wait_list)
@@ -3366,7 +3423,10 @@ _clCreateEventFromGLsyncKHR_(cl_context            context ,
 KHRicdVendorDispatch* createDispatchTable()
 {
   // Allocate table
-  KHRicdVendorDispatch *table = malloc(sizeof(KHRicdVendorDispatch));
+  static KHRicdVendorDispatch *table = NULL;
+  if (!table) {
+      table = malloc(sizeof(KHRicdVendorDispatch));
+  }
   if (!table)
   {
     return NULL;
