diff -U 3 -H -d -p -r -N -x '*.o' -x '*.orig' -- src/manager/kwalletmanager.cpp
--- src/manager/kwalletmanager.cpp	2014-10-09 09:07:50.000000000 +0200
+++ src/manager/kwalletmanager.cpp	2015-11-03 18:01:31.000000000 +0100
@@ -143,6 +143,10 @@ KWalletManager::KWalletManager(QWidget *
 	connect(act, SIGNAL(triggered()), SLOT(closeAllWallets()));
 	if (_tray) {
 		_tray->contextMenu()->addAction( act );
+        _tray->setStandardActionsEnabled(true);
+#ifdef Q_OS_MAC
+        connect(_tray->actionCollection()->action("minimizeRestore"), SIGNAL(triggered()), SLOT(show()));
+#endif
 	}
 	KStandardAction::quit(this, SLOT(shuttingDown()), actionCollection());
 	KStandardAction::keyBindings(guiFactory(), SLOT(configureShortcuts()),
