diff --git kdevplatform/interfaces/contextmenuextension.cpp kdevplatform/interfaces/contextmenuextension.cpp
index 8dc9a1135..427dc465f 100644
--- kdevplatform/interfaces/contextmenuextension.cpp
+++ kdevplatform/interfaces/contextmenuextension.cpp
@@ -24,6 +24,7 @@
 
 #include <QMenu>
 #include <QAction>
+#include <QDebug>
 
 #include <KLocalizedString>
 
@@ -92,6 +93,21 @@ void ContextMenuExtension::addAction( const QString& group, QAction* action )
     }
 }
 
+static void addContextActions(QList<QAction*> &target, const ContextMenuExtension &extension, const QString &group)
+{
+    const QList<QAction*> source = extension.actions( group );
+    foreach( QAction* act, source )
+    {
+        if( act )
+        {
+            target << act;
+        } else
+        {
+            qDebug() << "ignoring NULL QAction in" << group;
+        }
+    }
+}
+
 void ContextMenuExtension::populateMenu(QMenu* menu, const QList<ContextMenuExtension>& extensions)
 {
     QList<QAction*> buildActions;
@@ -102,35 +118,12 @@ void ContextMenuExtension::populateMenu(QMenu* menu, const QList<ContextMenuExte
     QList<QAction*> analyzeActions;
     foreach( const ContextMenuExtension &ext, extensions )
     {
-        foreach( QAction* act, ext.actions( ContextMenuExtension::BuildGroup ) )
-        {
-            buildActions << act;
-        }
-
-        foreach( QAction* act, ext.actions( ContextMenuExtension::VcsGroup ) )
-        {
-            vcsActions << act;
-        }
-
-        foreach( QAction* act, ext.actions( ContextMenuExtension::AnalyzeGroup ) )
-        {
-            analyzeActions << act;
-        }
-
-        foreach( QAction* act, ext.actions( ContextMenuExtension::ExtensionGroup ) )
-        {
-            extActions << act;
-        }
-
-        foreach( QAction* act, ext.actions( ContextMenuExtension::RefactorGroup ) )
-        {
-            refactorActions << act;
-        }
-
-        foreach( QAction* act, ext.actions( ContextMenuExtension::DebugGroup ) )
-        {
-            debugActions << act;
-        }
+        addContextActions( buildActions, ext, ContextMenuExtension::BuildGroup );
+        addContextActions( vcsActions, ext, ContextMenuExtension::VcsGroup );
+        addContextActions( analyzeActions, ext, ContextMenuExtension::AnalyzeGroup );
+        addContextActions( extActions, ext, ContextMenuExtension::ExtensionGroup );
+        addContextActions( refactorActions, ext, ContextMenuExtension::RefactorGroup );
+        addContextActions( debugActions, ext, ContextMenuExtension::DebugGroup );
     }
     
     if(!buildActions.isEmpty())
@@ -153,17 +146,16 @@ void ContextMenuExtension::populateMenu(QMenu* menu, const QList<ContextMenuExte
             menu->addAction( act );
         }
     }
-    
-    QMenu* debugmenu = menu;
-    if( debugActions.count() > 1 )
-    {
-        debugmenu = menu->addMenu( i18n("Debug") );
-    }
-    foreach( QAction* act, debugActions )
+
+    if( !debugActions.isEmpty() )
     {
-        debugmenu->addAction( act );
+        QMenu* debugmenu = menu->addMenu( i18n("Debug") );
+        foreach( QAction* act, debugActions )
+        {
+            debugmenu->addAction( act );
+        }
+        menu->addSeparator();
     }
-    menu->addSeparator();
 
     QMenu* refactormenu = menu;
     if( refactorActions.count() > 1 )
