diff --git kdevplatform/shell/sessioncontroller.cpp kdevplatform/shell/sessioncontroller.cpp
index ae4e355..f19e59c 100644
--- kdevplatform/shell/sessioncontroller.cpp
+++ kdevplatform/shell/sessioncontroller.cpp
@@ -320,7 +320,7 @@ public:
 public slots:
     void documentSavedOrClosed( KDevelop::IDocument* document )
     {
-        if(currentRecoveryFiles.contains(document->url()))
+        if( document && currentRecoveryFiles.contains(document->url()))
         {
             kDebug() << "deleting recovery-info for" << document->url();
             foreach(const QString& recoveryFileName, currentRecoveryFiles[document->url()])
