diff --git kdevplatform/debugger/breakpoint/breakpointwidget.cpp kdevplatform/debugger/breakpoint/breakpointwidget.cpp
index e86c188..45b0364 100644
--- kdevplatform/debugger/breakpoint/breakpointwidget.cpp
+++ kdevplatform/debugger/breakpoint/breakpointwidget.cpp
@@ -57,7 +57,7 @@ BreakpointWidget::BreakpointWidget(IDebugController *controller, QWidget *parent
                                           "breakpoint item allows you to change "
                                           "the breakpoint and will take you "
                                           "to the source in the editor window."));
-    setWindowIcon( QIcon::fromTheme( QStringLiteral( "media-playback-pause") ) );
+    setWindowIcon( QIcon::fromTheme( QStringLiteral( "media-playback-pause"), windowIcon() ) );
 
     m_breakpointsView = new QTableView(this);
     m_breakpointsView->setSelectionBehavior(QAbstractItemView::SelectRows);
diff --git kdevplatform/debugger/framestack/framestackwidget.cpp kdevplatform/debugger/framestack/framestackwidget.cpp
index 04bf8da..296ab99 100644
--- kdevplatform/debugger/framestack/framestackwidget.cpp
+++ kdevplatform/debugger/framestack/framestackwidget.cpp
@@ -67,7 +67,7 @@ FramestackWidget::FramestackWidget(IDebugController* controller, QWidget* parent
                       "program. By clicking on an item you "
                       "can see the values in any of the "
                       "previous calling functions."));
-    setWindowIcon(QIcon::fromTheme(QStringLiteral("view-list-text")));
+    setWindowIcon(QIcon::fromTheme(QStringLiteral("view-list-text"), windowIcon()));
     m_threadsWidget = new QWidget(this);
     m_threads = new QListView(m_threadsWidget);
     m_frames = new QTreeView(this);
diff --git kdevplatform/debugger/variable/variablewidget.cpp kdevplatform/debugger/variable/variablewidget.cpp
index 0bd5ab8..2cb3b71 100644
--- kdevplatform/debugger/variable/variablewidget.cpp
+++ kdevplatform/debugger/variable/variablewidget.cpp
@@ -74,7 +74,7 @@ VariableWidget::VariableWidget(IDebugController* controller, QWidget *parent)
 : QWidget(parent), variablesRoot_(controller->variableCollection()->root())
 {
   //setWindowIcon(QIcon::fromTheme("math_brace"));
-    setWindowIcon(QIcon::fromTheme(QStringLiteral("debugger")));
+    setWindowIcon(QIcon::fromTheme(QStringLiteral("debugger"), windowIcon()));
     setWindowTitle(i18n("Debugger Variables"));
 
     m_proxy = new VariableSortProxyModel;
diff --git kdevplatform/documentation/documentationview.cpp kdevplatform/documentation/documentationview.cpp
index 9d184a0..e1aa672 100644
--- kdevplatform/documentation/documentationview.cpp
+++ kdevplatform/documentation/documentationview.cpp
@@ -48,7 +48,7 @@ using namespace KDevelop;
 DocumentationView::DocumentationView(QWidget* parent, ProvidersModel* model)
     : QWidget(parent), mProvidersModel(model)
 {
-    setWindowIcon(QIcon::fromTheme(QStringLiteral("documentation")));
+    setWindowIcon(QIcon::fromTheme(QStringLiteral("documentation"), windowIcon()));
     setWindowTitle(i18n("Documentation"));
 
     setLayout(new QVBoxLayout(this));
diff --git kdevplatform/plugins/classbrowser/classwidget.cpp kdevplatform/plugins/classbrowser/classwidget.cpp
index 4a8db5c..a471fa0 100644
--- kdevplatform/plugins/classbrowser/classwidget.cpp
+++ kdevplatform/plugins/classbrowser/classwidget.cpp
@@ -46,7 +46,7 @@ ClassWidget::ClassWidget(QWidget* parent, ClassBrowserPlugin* plugin)
 {
   setObjectName("Class Browser Tree");
   setWindowTitle(i18n("Classes"));
-  setWindowIcon(QIcon::fromTheme("code-class"));
+  setWindowIcon(QIcon::fromTheme("code-class", windowIcon()));
 
   // Set tree in the plugin
   m_plugin->setActiveClassTree(m_tree);
diff --git kdevplatform/plugins/contextbrowser/contextbrowserview.cpp kdevplatform/plugins/contextbrowser/contextbrowserview.cpp
index d1dba72..57c966f 100644
--- kdevplatform/plugins/contextbrowser/contextbrowserview.cpp
+++ kdevplatform/plugins/contextbrowser/contextbrowserview.cpp
@@ -106,7 +106,7 @@ void ContextBrowserView::updateLockIcon(bool checked) {
 }
 
 ContextBrowserView::ContextBrowserView( ContextBrowserPlugin* plugin, QWidget* parent ) : QWidget(parent), m_plugin(plugin), m_navigationWidget(new QTextBrowser()), m_autoLocked(false) {
-    setWindowIcon( QIcon::fromTheme("code-context") );
+    setWindowIcon( QIcon::fromTheme("code-context", windowIcon()) );
 
     m_allowLockedUpdate = false;
 
diff --git kdevplatform/plugins/documentview/kdevdocumentview.cpp kdevplatform/plugins/documentview/kdevdocumentview.cpp
index 9ee5916..1569149 100644
--- kdevplatform/plugins/documentview/kdevdocumentview.cpp
+++ kdevplatform/plugins/documentview/kdevdocumentview.cpp
@@ -73,7 +73,7 @@ KDevDocumentView::KDevDocumentView( KDevDocumentViewPlugin *plugin, QWidget *par
 
     setObjectName( i18n( "Documents" ) );
 
-    setWindowIcon( QIcon::fromTheme( QStringLiteral( "document-multiple" ) ) );
+    setWindowIcon( QIcon::fromTheme( QStringLiteral( "document-multiple" ), windowIcon() ) );
     setWindowTitle( i18n( "Documents" ) );
 
     setFocusPolicy( Qt::NoFocus );
diff --git kdevplatform/plugins/externalscript/externalscriptview.cpp kdevplatform/plugins/externalscript/externalscriptview.cpp
index a7d2e7a..ee717a6 100644
--- kdevplatform/plugins/externalscript/externalscriptview.cpp
+++ kdevplatform/plugins/externalscript/externalscriptview.cpp
@@ -39,7 +39,7 @@ ExternalScriptView::ExternalScriptView( ExternalScriptPlugin* plugin, QWidget* p
   Ui::ExternalScriptViewBase::setupUi( this );
 
   setWindowTitle( i18n( "External Scripts" ) );
-  setWindowIcon( QIcon::fromTheme("dialog-scripts") );
+  setWindowIcon( QIcon::fromTheme("dialog-scripts", windowIcon()) );
 
   m_model = new QSortFilterProxyModel( this );
   m_model->setSourceModel( m_plugin->model() );
diff --git kdevplatform/plugins/filemanager/filemanager.cpp kdevplatform/plugins/filemanager/filemanager.cpp
index 4f332da..433f1be 100644
--- kdevplatform/plugins/filemanager/filemanager.cpp
+++ kdevplatform/plugins/filemanager/filemanager.cpp
@@ -56,7 +56,7 @@ FileManager::FileManager(KDevFileManagerPlugin *plugin, QWidget* parent)
     : QWidget(parent), m_plugin(plugin)
 {
     setObjectName(QStringLiteral("FileManager"));
-    setWindowIcon(QIcon::fromTheme(QStringLiteral("folder-sync")));
+    setWindowIcon(QIcon::fromTheme(QStringLiteral("folder-sync"), windowIcon()));
     setWindowTitle(i18n("File System"));
 
     KConfigGroup cg = KDevelop::ICore::self()->activeSession()->config()->group( "Filesystem" );
diff --git kdevplatform/plugins/grepview/grepoutputview.cpp kdevplatform/plugins/grepview/grepoutputview.cpp
index ab991ce..cc25669 100644
--- kdevplatform/plugins/grepview/grepoutputview.cpp
+++ kdevplatform/plugins/grepview/grepoutputview.cpp
@@ -70,7 +70,7 @@ GrepOutputView::GrepOutputView(QWidget* parent, GrepViewPlugin* plugin)
     Ui::GrepOutputView::setupUi(this);
 
     setWindowTitle(i18nc("@title:window", "Find/Replace Output View"));
-    setWindowIcon(QIcon::fromTheme("edit-find"));
+    setWindowIcon(QIcon::fromTheme("edit-find", windowIcon()));
 
     m_prev = new QAction(QIcon::fromTheme("go-previous"), i18n("&Previous Item"), this);
     m_prev->setEnabled(false);
diff --git kdevplatform/plugins/konsole/kdevkonsoleview.cpp kdevplatform/plugins/konsole/kdevkonsoleview.cpp
index 50ad5ca..4559d89 100644
--- kdevplatform/plugins/konsole/kdevkonsoleview.cpp
+++ kdevplatform/plugins/konsole/kdevkonsoleview.cpp
@@ -97,7 +97,7 @@ KDevKonsoleView::KDevKonsoleView( KDevKonsoleViewPlugin *plugin, QWidget* parent
     d->konsolepart = 0;
     setObjectName( i18n( "Konsole" ) );
 
-    setWindowIcon( QIcon::fromTheme( QStringLiteral( "utilities-terminal" ) ) );
+    setWindowIcon( QIcon::fromTheme( QStringLiteral( "utilities-terminal" ), windowIcon() ) );
     setWindowTitle( i18n( "Konsole" ) );
 
     d->m_vbox = new QVBoxLayout( this );
diff --git kdevplatform/plugins/outlineview/outlinewidget.cpp kdevplatform/plugins/outlineview/outlinewidget.cpp
index 8a49cd6..2bc82c4 100644
--- kdevplatform/plugins/outlineview/outlinewidget.cpp
+++ kdevplatform/plugins/outlineview/outlinewidget.cpp
@@ -48,7 +48,7 @@ OutlineWidget::OutlineWidget(QWidget* parent, OutlineViewPlugin* plugin)
     setObjectName("Outline View");
     setWindowTitle(i18n("Outline"));
     setWhatsThis(i18n("Outline View"));
-    setWindowIcon(QIcon::fromTheme("code-class")); //TODO: better icon?
+    setWindowIcon(QIcon::fromTheme("code-class", windowIcon())); //TODO: better icon?
 
     m_proxy->setSourceModel(m_model);
     m_proxy->setFilterCaseSensitivity(Qt::CaseInsensitive);
diff --git kdevplatform/plugins/problemreporter/problemsview.cpp kdevplatform/plugins/problemreporter/problemsview.cpp
index e4fd505..659ff6d 100644
--- kdevplatform/plugins/problemreporter/problemsview.cpp
+++ kdevplatform/plugins/problemreporter/problemsview.cpp
@@ -94,7 +94,7 @@ ProblemsView::ProblemsView(QWidget* parent)
     : QMainWindow(parent)
 {
     setWindowTitle(i18n("Problems"));
-    setWindowIcon(QIcon::fromTheme("script-error"));
+    setWindowIcon(QIcon::fromTheme("script-error", windowIcon()));
 
     m_toolBar = new QToolBar(this);
     m_toolBar->setMovable(false);
diff --git kdevplatform/plugins/projectmanagerview/projectmanagerview.cpp kdevplatform/plugins/projectmanagerview/projectmanagerview.cpp
index fd8bc02..0a8f701 100644
--- kdevplatform/plugins/projectmanagerview/projectmanagerview.cpp
+++ kdevplatform/plugins/projectmanagerview/projectmanagerview.cpp
@@ -78,7 +78,7 @@ ProjectManagerView::ProjectManagerView( ProjectManagerViewPlugin* plugin, QWidge
 
     m_ui->projectTreeView->installEventFilter(this);
 
-    setWindowIcon( QIcon::fromTheme( "project-development" ) );
+    setWindowIcon( QIcon::fromTheme( "project-development", windowIcon() ) );
 
     KConfigGroup pmviewConfig(ICore::self()->activeSession()->config(), sessionConfigGroup);
     if (pmviewConfig.hasKey(splitterStateConfigKey)) {
diff --git kdevplatform/plugins/testview/testview.cpp kdevplatform/plugins/testview/testview.cpp
index 12a9a8b..ef8a3ef 100644
--- kdevplatform/plugins/testview/testview.cpp
+++ kdevplatform/plugins/testview/testview.cpp
@@ -69,7 +69,7 @@ TestView::TestView(TestViewPlugin* plugin, QWidget* parent)
 , m_tree(new QTreeView(this))
 , m_filter(new KRecursiveFilterProxyModel(this))
 {
-    setWindowIcon(QIcon::fromTheme("preflight-verifier"));
+    setWindowIcon(QIcon::fromTheme("preflight-verifier", windowIcon()));
     setWindowTitle(i18n("Unit Tests"));
 
     QVBoxLayout* layout = new QVBoxLayout(this);
diff --git kdevplatform/plugins/vcschangesview/vcschangesview.cpp kdevplatform/plugins/vcschangesview/vcschangesview.cpp
index acf770f..7f6aa2c 100644
--- kdevplatform/plugins/vcschangesview/vcschangesview.cpp
+++ kdevplatform/plugins/vcschangesview/vcschangesview.cpp
@@ -51,7 +51,7 @@ VcsChangesView::VcsChangesView(VcsProjectIntegrationPlugin* plugin, QWidget* par
     setContextMenuPolicy(Qt::CustomContextMenu);
     setTextElideMode(Qt::ElideLeft);
     setWordWrap(true);
-    setWindowIcon(QIcon::fromTheme(QStringLiteral("exchange-positions")));
+    setWindowIcon(QIcon::fromTheme(QStringLiteral("exchange-positions"), windowIcon()));
     
     connect(this, &VcsChangesView::customContextMenuRequested, this, &VcsChangesView::popupContextMenu);
     
