diff --git kdevplatform/CMakeLists.txt kdevplatform/CMakeLists.txt
index 5a030b1..19194b3 100644
--- kdevplatform/CMakeLists.txt
+++ kdevplatform/CMakeLists.txt
@@ -61,7 +61,8 @@ endif()
 if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
     # TODO: kf5 remove
     # in Qt 4, clang support is lacking...
-    add_definitions("-DQ_COMPILER_INITIALIZER_LISTS=1")
+    # handled in Qt 4.8.7
+    # add_definitions("-DQ_COMPILER_INITIALIZER_LISTS=1")
 endif()
 
 configure_file( ${CMAKE_CURRENT_SOURCE_DIR}/config-kdevplatform.h.cmake
