diff --git kdevplatform/plugins/grepview/grepfindthread.cpp kdevplatform/plugins/grepview/grepfindthread.cpp
index 9388d5c..c0b5fe2 100644
--- kdevplatform/plugins/grepview/grepfindthread.cpp
+++ kdevplatform/plugins/grepview/grepfindthread.cpp
@@ -44,6 +44,8 @@ static QList<QUrl> thread_getProjectFiles(const QUrl dir, int depth, const QStri
     if(!project)
         return res;
 
+    QString dirName = QFileInfo(QDir(dir.toLocalFile()).canonicalPath()).absoluteDir().path();
+
     const QSet<IndexedString> fileSet = project->fileSet();
     foreach( const IndexedString &item, fileSet )
     {
@@ -67,7 +69,7 @@ static QList<QUrl> thread_getProjectFiles(const QUrl dir, int depth, const QStri
                     continue;
             }
         }
-        if( QDir::match(include, url.fileName()) && !QDir::match(exlude, url.toLocalFile()) )
+        if( QDir::match(include, url.fileName()) && !QDir::match(exlude, url.toLocalFile().remove(dirName)) )
             res << url;
     }
 
@@ -78,6 +80,7 @@ static QList<QUrl> thread_findFiles(const QDir& dir, int depth, const QStringLis
                                    const QStringList& exclude, volatile bool &abort)
 {
     QFileInfoList infos = dir.entryInfoList(include, QDir::NoDotAndDotDot|QDir::Files|QDir::Readable);
+    QString dirName = QFileInfo(dir.canonicalPath()).absoluteDir().path();
 
     if(!QFileInfo(dir.path()).isDir())
         infos << QFileInfo(dir.path());
@@ -86,7 +89,7 @@ static QList<QUrl> thread_findFiles(const QDir& dir, int depth, const QStringLis
     foreach(const QFileInfo &currFile, infos)
     {
         QString currName = currFile.canonicalFilePath();
-        if(!QDir::match(exclude, currName))
+        if(!QDir::match(exclude, QString(currName).remove(dirName)))
             dirFiles << QUrl::fromLocalFile(currName);
     }
     if(depth != 0)
