diff --git lang/qt/tests/t-support.h lang/qt/tests/t-support.h
index 59d9a4d..12dc113 100644
--- lang/qt/tests/t-support.h
+++ lang/qt/tests/t-support.h
@@ -42,7 +42,7 @@ public:
     char *getPassphrase(const char *useridHint, const char *description,
                         bool previousWasBad, bool &canceled) Q_DECL_OVERRIDE
     {
-        return strdup("abc");
+        return qstrdup("abc");
     }
 };
 } // namespace GpgME
