--- src/orig.build_theme.sh	2019-09-03 15:41:17.000000000 +0200
+++ src/build_theme.sh	2019-09-23 18:12:50.238334000 +0200
@@ -42,7 +42,7 @@
   THEME_BUILD_DIR="$(mktemp -d)"
   create_folders "${THEME_BUILD_DIR}"
   cp -R gtk2/* "${THEME_BUILD_DIR}/gtk-2.0/"
-  python3 render_assets.py -c "$1" -a "${THEME_BUILD_DIR}/assets" \
+  @PREFIX@/bin/python3.7 render_assets.py -c "$1" -a "${THEME_BUILD_DIR}/assets" \
     -g "${THEME_BUILD_DIR}/gtk-2.0" -G "${THEME_BUILD_DIR}" -b "$4"
   build_sass gtk318/gtk.scss "${THEME_BUILD_DIR}/gtk-3.18/gtk.css" "${THEME_BUILD_DIR}"
   build_sass gtk320/gtk.scss "${THEME_BUILD_DIR}/gtk-3.20/gtk.css" "${THEME_BUILD_DIR}"
