--- cmake/orig.FindGTKEngine.cmake	2019-09-23 18:08:18.608334000 +0200
+++ cmake/FindGTKEngine.cmake	2019-09-23 18:08:38.658334000 +0200
@@ -14,7 +14,9 @@
 foreach(_ver ${_versions})
   find_library(GTKPIXMAP_PLUGIN NAMES pixmap
     PATHS
-    ${KDE_INSTALL_FULL_LIBDIR}/gtk-2.0/${_ver}.0/engines)
+    ${KDE_INSTALL_FULL_LIBDIR}/gtk-2.0/${_ver}.0/engines
+    /usr/lib//gtk-2.0/${_ver}.0/engines
+    /usr/lib/x86_64-linux-gnu/gtk-2.0/${_ver}.0/engines)
 endforeach()
 
 if(NOT ${GTKPIXMAP_PLUGIN} STREQUAL "GTKPIXMAP_PLUGIN-NOTFOUND")
