diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5375e3b..d7f8b3b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -189,12 +188,12 @@ generate_export_header(okularcore BASE_NAME okularcore EXPORT_FILE_NAME "${CMAKE
 
 # Special handling for linking okularcore on OSX/Apple
 IF(APPLE)
-    SET(OKULAR_IOKIT "-framework IOKit" CACHE STRING "Apple IOKit framework")
+    SET(OKULAR_APPLE_FRAMEWORKS "-framework IOKit -framework CoreFoundation -framework CoreGraphics" CACHE STRING "required Apple frameworks")
 ENDIF(APPLE)
 
 target_link_libraries(okularcore
 PRIVATE
-    ${OKULAR_IOKIT}
+    ${OKULAR_APPLE_FRAMEWORKS}
     KF5::Archive
     KF5::JS
     KF5::JSApi
