diff --git a/core/document.cpp b/core/document.cpp
index f31345d5c..cc482f63f 100644
--- a/core/document.cpp
+++ b/core/document.cpp
@@ -4951,7 +4951,7 @@ void DocumentPrivate::requestDone(PixmapRequest *req)
             }
 
         DocumentObserver *observer = req->observer();
-        if (m_observers.contains(observer)) {
+        if (observer && m_observers.contains(observer)) {
             // [MEM] 1.2 append memory allocation descriptor to the FIFO
             qulonglong memoryBytes = 0;
             const TilesManager *tm = req->d->tilesManager();
