diff --git a/core/texteditors_p.h b/core/texteditors_p.h
index b3a068e..3a688b9 100644
--- a/core/texteditors_p.h
+++ b/core/texteditors_p.h
@@ -22,7 +22,7 @@ static inline QHash< int, QString > buildEditorsMap()
 {
     QHash< int, QString > editors;
     editors.insert( SettingsCore::EnumExternalEditor::Kate,
-        QStringLiteral( "kate --line %l --column %c" ) );
+        QStringLiteral( "kate5 --line %l --column %c" ) );
     editors.insert( SettingsCore::EnumExternalEditor::Kile,
         QStringLiteral( "kile --line %l" ) );
     editors.insert( SettingsCore::EnumExternalEditor::Scite,
