diff --git a/CMakeLists.txt b/CMakeLists.txt
--- okular-16.08.0/orig.CMakeLists.txt	2016-07-21 23:46:24.000000000 +0200
+++ okular-16.08.0/CMakeLists.txt	2016-09-06 14:51:47.240204221 +0200
@@ -145,10 +145,10 @@
 
 # Special handling for linking okularcore on OSX/Apple
 IF(APPLE)
-    SET(OKULAR_IOKIT "-framework IOKit" CACHE STRING "Apple IOKit framework")
+    SET(OKULAR_APPLE_FRAMEWORKS "-framework IOKit -framework CoreFoundation -framework CoreGraphics" CACHE STRING "required Apple frameworks")
 ENDIF(APPLE)
 
-target_link_libraries(okularcore  ${OKULAR_IOKIT} ${KDE4_KIO_LIBS} ${KDE4_PHONON_LIBRARY} ${KDE4_KJSAPI_LIBRARY} ${MATH_LIB} ${KDE4_THREADWEAVER_LIBRARY} ${ZLIB_LIBRARIES} )
+target_link_libraries(okularcore  ${OKULAR_APPLE_FRAMEWORKS} ${KDE4_KIO_LIBS} ${KDE4_PHONON_LIBRARY} ${KDE4_KJSAPI_LIBRARY} ${MATH_LIB} ${KDE4_THREADWEAVER_LIBRARY} ${ZLIB_LIBRARIES} )
 
 if(LibKScreen_FOUND)
     target_link_libraries(okularcore ${LibKScreen_LIBRARY})
