diff --git a/mobile/components/CMakeLists.txt b/mobile/components/CMakeLists.txt
index 9da375f..f3111d0 100644
--- a/mobile/components/CMakeLists.txt
+++ b/mobile/components/CMakeLists.txt
@@ -29,7 +29,9 @@ target_link_libraries(okularplugin
         Qt5::Xml
         Qt5::Svg
         okularcore
-        okularpart
+        KF5::Bookmarks
+        KF5::I18n
+        KF5::IconThemes
         )
 
 install(TARGETS okularplugin DESTINATION ${KDE_INSTALL_QMLDIR}/org/kde/okular)
