diff --git a/conf/preferencesdialog.cpp b/conf/preferencesdialog.cpp
index d03a67816..eadff5a49 100644
--- a/conf/preferencesdialog.cpp
+++ b/conf/preferencesdialog.cpp
@@ -37,7 +37,7 @@ PreferencesDialog::PreferencesDialog(QWidget *parent, KConfigSkeleton *skeleton,
     m_debug = new DlgDebug(this);
 #endif
 
-    addPage(m_general, i18n("General"), QStringLiteral("okular"), i18n("General Options"));
+    addPage( m_general, i18n("General"), QStringLiteral("okular5"), i18n("General Options") );
     addPage(m_accessibility, i18n("Accessibility"), QStringLiteral("preferences-desktop-accessibility"), i18n("Accessibility Reading Aids"));
     addPage(m_performance, i18n("Performance"), QStringLiteral("preferences-system-performance"), i18n("Performance Tuning"));
     if (embedMode == Okular::ViewerWidgetMode) {
@@ -53,7 +53,7 @@ PreferencesDialog::PreferencesDialog(QWidget *parent, KConfigSkeleton *skeleton,
 #ifdef OKULAR_DEBUG_CONFIGPAGE
     addPage(m_debug, "Debug", "system-run", "Debug options");
 #endif
-    setHelp(QStringLiteral("configure"), QStringLiteral("okular"));
+    setHelp(QStringLiteral("configure"), QStringLiteral("okular5"));
 }
 
 void PreferencesDialog::switchToAnnotationsPage()
