diff --git a/core/texteditors_p.h b/core/texteditors_p.h
--- okular-16.08.0/core/orig.texteditors_p.h	2016-07-21 23:46:24.000000000 +0200
+++ okular-16.08.0/core/texteditors_p.h	2016-09-06 14:58:51.825194878 +0200
@@ -22,7 +22,7 @@
 {
     QHash< int, QString > editors;
     editors.insert( SettingsCore::EnumExternalEditor::Kate,
-        QString::fromLatin1( "kate --line %l --column %c" ) );
+        QString::fromLatin1( "kate5 --line %l --column %c" ) );
     editors.insert( SettingsCore::EnumExternalEditor::Kile,
         QString::fromLatin1( "kile --line %l" ) );
     editors.insert( SettingsCore::EnumExternalEditor::Scite,
