diff --git a/core/texteditors_p.h b/core/texteditors_p.h
index 3fa8149a8..ed9696cbc 100644
--- a/core/texteditors_p.h
+++ b/core/texteditors_p.h
@@ -20,8 +20,8 @@ namespace Okular
 static inline QHash<int, QString> buildEditorsMap()
 {
     QHash<int, QString> editors;
-    editors.insert(SettingsCore::EnumExternalEditor::Kate, QStringLiteral("kate --line %l --column %c"));
-    editors.insert(SettingsCore::EnumExternalEditor::Kile, QStringLiteral("kile --line %l"));
+    editors.insert(SettingsCore::EnumExternalEditor::Kate, QStringLiteral("kate5 --line %l --column %c"));
+    editors.insert(SettingsCore::EnumExternalEditor::Kile, QStringLiteral("kile5 --line %l"));
     editors.insert(SettingsCore::EnumExternalEditor::Scite, QStringLiteral("scite %f \"-goto:%l,%c\""));
     editors.insert(SettingsCore::EnumExternalEditor::Emacsclient, QStringLiteral("emacsclient -a emacs --no-wait +%l %f"));
     editors.insert(SettingsCore::EnumExternalEditor::Lyxclient, QStringLiteral("lyxclient -g %f %l"));
