diff --git a/CMakeLists.txt b/CMakeLists.txt
index d76f67e0c..80fb7d204 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -216,7 +216,7 @@ generate_export_header(okularcore BASE_NAME okularcore EXPORT_FILE_NAME "${CMAKE
 
 # Special handling for linking okularcore on OSX/Apple
 IF(APPLE)
-    SET(OKULAR_IOKIT "-framework IOKit" CACHE STRING "Apple IOKit framework")
+    SET(OKULAR_APPLE_FRAMEWORKS "-framework IOKit -framework CoreFoundation -framework CoreGraphics" CACHE STRING "required Apple frameworks")
 ENDIF(APPLE)
 
 # Extra library needed by imported synctex code on Windows
@@ -226,7 +226,7 @@ endif(WIN32)
 
 target_link_libraries(okularcore
 PRIVATE
-    ${OKULAR_IOKIT}
+    ${OKULAR_APPLE_FRAMEWORKS}
     ${SHLWAPI}
     KF5::Archive
     KF5::JS
