diff --git a/core/document.cpp b/core/document.cpp
index cacb0a54f5daa88dff09b8efed6ccf2590740fb2..e2a4801385df8816f05d70121a6cf6666ef5d3d4 100644
--- a/core/document.cpp
+++ b/core/document.cpp
@@ -5386,7 +5386,7 @@ void DocumentPrivate::requestDone( PixmapRequest * req )
             }
 
         DocumentObserver *observer = req->observer();
-        if ( m_observers.contains(observer) )
+        if ( observer && m_observers.contains(observer) )
         {
             // [MEM] 1.2 append memory allocation descriptor to the FIFO
             qulonglong memoryBytes = 0;
diff --git a/mobile/app/package/contents/ui/MainView.qml b/mobile/app/package/contents/ui/MainView.qml
index 751a68846adf75c4242b8266888a87c3a0c44561..ec5cf8ffe116002a0528f4884866f1bcc1ee7110 100644
--- a/mobile/app/package/contents/ui/MainView.qml
+++ b/mobile/app/package/contents/ui/MainView.qml
@@ -18,7 +18,7 @@
  */
 
 import QtQuick 2.1
-import QtQuick.Controls 2.3 as QQC2
+import QtQuick.Controls 2.2 as QQC2
 import org.kde.okular 2.0 as Okular
 import org.kde.kirigami 2.0 as Kirigami
 
diff --git a/mobile/app/package/contents/ui/OkularDrawer.qml b/mobile/app/package/contents/ui/OkularDrawer.qml
index 7bb8ade697387739806ca680953b65ed63f7a025..c90ffbd283df0e019b86f6b244f56f2dbab60ba9 100644
--- a/mobile/app/package/contents/ui/OkularDrawer.qml
+++ b/mobile/app/package/contents/ui/OkularDrawer.qml
@@ -18,7 +18,7 @@
  */
 
 import QtQuick 2.1
-import QtQuick.Controls 2.5 as QQC2
+import QtQuick.Controls 2.2 as QQC2
 import org.kde.kirigami 2.0 as Kirigami
 import org.kde.okular 2.0 as Okular
 
@@ -69,7 +69,7 @@ Kirigami.OverlayDrawer {
                     QQC2.ToolButton {
                         id: thumbnailsButton
                         text: tabsToolbar.width > Kirigami.Units.gridUnit * 30 ? i18n("Thumbnails") : ""
-                        icon.name: "view-preview"
+                        Component.onCompleted: { if (icon) { icon.name = "view-preview" } }
                         checkable: true
                         flat: false
                         onCheckedChanged: {
@@ -83,7 +83,7 @@ Kirigami.OverlayDrawer {
                         id: tocButton
                         enabled: documentItem.tableOfContents.count > 0
                         text: tabsToolbar.width > Kirigami.Units.gridUnit * 30 ? i18n("Table of contents") : ""
-                        icon.name: "view-table-of-contents-ltr"
+                        Component.onCompleted: { if (icon) { icon.name = "view-table-of-contents-ltr" } }
                         checkable: true
                         flat: false
                         onCheckedChanged: {
@@ -97,7 +97,7 @@ Kirigami.OverlayDrawer {
                         id: bookmarksButton
                         enabled: documentItem.bookmarkedPages.length > 0
                         text: tabsToolbar.width > Kirigami.Units.gridUnit * 30 ? i18n("Bookmarks") : ""
-                        icon.name: "bookmarks-organize"
+                        Component.onCompleted: { if (icon) { icon.name = "bookmarks-organize" } }
                         checkable: true
                         flat: false
                         onCheckedChanged: {
diff --git a/mobile/app/package/contents/ui/ThumbnailsBase.qml b/mobile/app/package/contents/ui/ThumbnailsBase.qml
index 4244c10c9075e934947a5542a91180931a854b4d..7349573188df5193023afb2bd307be460fad3645 100644
--- a/mobile/app/package/contents/ui/ThumbnailsBase.qml
+++ b/mobile/app/package/contents/ui/ThumbnailsBase.qml
@@ -18,7 +18,7 @@
  */
 
 import QtQuick 2.1
-import QtQuick.Controls 2.3 as QQC2
+import QtQuick.Controls 2.2 as QQC2
 import QtGraphicalEffects 1.0
 import org.kde.okular 2.0 as Okular
 import org.kde.kirigami 2.0 as Kirigami
diff --git a/mobile/app/package/contents/ui/main.qml b/mobile/app/package/contents/ui/main.qml
index b2e06b9c0e5365a1aa5a73f05193878d61e3d502..c84398d28ea338075a04ee72fa76e7ab328a3502 100644
--- a/mobile/app/package/contents/ui/main.qml
+++ b/mobile/app/package/contents/ui/main.qml
@@ -18,7 +18,7 @@
  */
 
 import QtQuick 2.7
-import QtQuick.Dialogs 1.3 as QQD
+import QtQuick.Dialogs 1.2 as QQD
 import org.kde.okular 2.0 as Okular
 import org.kde.kirigami 2.0 as Kirigami
 import org.kde.okular.app 2.0
diff --git a/mobile/components/pageitem.cpp b/mobile/components/pageitem.cpp
index 75930bfe9bae9a86ac7d429081bc15091107c12e..8910dc4564a18b6abe4a7c81254a74cd628469c4 100644
--- a/mobile/components/pageitem.cpp
+++ b/mobile/components/pageitem.cpp
@@ -358,6 +358,10 @@ void PageItem::requestPixmap()
 
 void PageItem::paint()
 {
+    if (!m_documentItem || !m_page || !window() || width() <= 0 || height() < 0) {
+        return;
+    }
+
     Observer *observer = m_isThumbnail ? m_documentItem.data()->thumbnailObserver() : m_documentItem.data()->pageviewObserver();
     const int flags = PagePainter::Accessibility | PagePainter::Highlights | PagePainter::Annotations;
 

