--- src/libbluray/bdj/orig.bdj.c	2017-03-02 14:33:49.000000000 +0100
+++ src/libbluray/bdj/bdj.c	2017-08-22 13:42:09.288127060 +0200
@@ -426,6 +426,8 @@
     // pre-defined search paths for libbluray.jar
     static const char * const jar_paths[] = {
 #ifndef _WIN32
+        "@PREFIX@/share/java/" BDJ_JARFILE,
+        "@PREFIX@/share/libbluray/lib/" BDJ_JARFILE,
         "/usr/share/java/" BDJ_JARFILE,
         "/usr/share/libbluray/lib/" BDJ_JARFILE,
 #endif
