diff --git a/src/main.cpp b/src/main.cpp
index 8013529..a9101fa 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -210,7 +210,7 @@ int pcm_track_helper(
     fprintf(stderr, "\33[2K\r");
     fprintf(stderr, "%3.1f%%\t", dec->getPositionAsPercent());
     if (ok) {
-        fprintf(stderr, "Conversion completed sucessfully.\n");
+        fprintf(stderr, "Conversion completed successfully.\n");
     } else {
         fprintf(stderr, "\nError during conversion.\n");
         fprintf(stderr, "encoding: %s\n", ok ? "succeeded" : "FAILED");
