--- specs/orig.macx-g++	2018-07-05 11:54:58.000000000 +0200
+++ specs/macx-g++	2018-12-15 14:33:59.000000000 +0100
@@ -11,7 +11,7 @@
 QMAKE_INCREMENTAL_STYLE = sublibs
 QMAKE_COMPILER_DEFINES  += __APPLE__ __GNUC__
 
-QMAKE_CC		= cc
+QMAKE_CC		= /opt/local/bin/clang-mp-5.0
 QMAKE_LEX		= flex
 QMAKE_LEXFLAGS		=
 QMAKE_YACC		= yacc
@@ -20,11 +20,9 @@
 QMAKE_YACC_HEADER       = $base.tab.h
 QMAKE_YACC_SOURCE       = $base.tab.c
 QMAKE_RESOURCE		= /Developer/Tools/Rez
-QMAKE_CFLAGS		= -pipe
 QMAKE_CFLAGS_DEPS	= -M
 QMAKE_CFLAGS_WARN_ON	= -Wall -W
 QMAKE_CFLAGS_WARN_OFF	= -w
-QMAKE_CFLAGS_RELEASE	= -Os
 QMAKE_CFLAGS_DEBUG	= -g
 QMAKE_CFLAGS_SHLIB	= -fPIC
 QMAKE_EXTENSION_SHLIB	= dylib
@@ -33,7 +31,7 @@
 QMAKE_CFLAGS_PPC        = -arch ppc
 QMAKE_CFLAGS_X86        = -arch i386
 
-QMAKE_CXX		= c++
+QMAKE_CXX		= /opt/local/bin/clang++-mp-5.0
 QMAKE_CXXFLAGS		= $$QMAKE_CFLAGS
 QMAKE_CXXFLAGS_DEPS	= $$QMAKE_CFLAGS_DEPS
 QMAKE_CXXFLAGS_WARN_ON	= $$QMAKE_CFLAGS_WARN_ON
@@ -49,6 +47,7 @@
 QMAKE_LIBDIR		=
 QMAKE_INCDIR_QT		= $$[QT_INSTALL_HEADERS]
 QMAKE_LIBDIR_QT		= $$[QT_INSTALL_LIBS]
+QMAKE_FRAMEWORKDIR_QT	= $$[QT_INSTALL_FRAMEWORKS]
 QMAKE_INCDIR_OPENGL	= /System/Library/Frameworks/OpenGL.framework/Headers \
 	/System/Library/Frameworks/AGL.framework/Headers/
 
