diff --git a/applications/mail/CMakeLists.txt b/applications/mail/CMakeLists.txt
index 7c6fe755cf361cfa425390a5a15959666c9b6c8e..6c806c5afdb06613e52b9ad1ba69992e5d423bf9 100644
--- a/applications/mail/CMakeLists.txt
+++ b/applications/mail/CMakeLists.txt
@@ -5,6 +5,12 @@ cmake_policy(SET CMP0063 NEW)
 
 include(CPack)
 include(FeatureSummary)
+
+if(APPLE OR WIN32)
+    find_package (ECM 5.29.0 REQUIRED NO_MODULE)
+    set(CMAKE_MODULE_PATH ${KDevelop_SOURCE_DIR}/cmake/modules ${ECM_MODULE_PATH})
+    include(ECMAddAppIcon)
+endif()
 find_package(PkgConfig)
 
 ################# set KDE specific information #################
@@ -24,11 +30,19 @@ set (QT_MIN_VERSION "5.6.0")
 find_package(KF5 REQUIRED COMPONENTS Package)
 
 # install executable
-add_executable(${PROJECT_NAME} main.cpp)
+set(kube_mail_SRCS main.cpp)
+
+if(APPLE OR WIN32)
+    # Sets the icon on Windows and OSX
+    file(GLOB ICONS_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/../../icons/*-apps-kube_icon.*")
+    ecm_add_app_icon(kube_mail_SRCS ICONS ${ICONS_SRCS})
+endif()
+
+add_executable(${PROJECT_NAME} ${kube_mail_SRCS})
 qt5_use_modules(${PROJECT_NAME} Gui Quick Widgets WebEngine)
 target_link_libraries(${PROJECT_NAME} KF5::Package)
 
-install(TARGETS ${PROJECT_NAME} DESTINATION bin)
+install(TARGETS ${PROJECT_NAME} DESTINATION ${KDE_INSTALL_TARGETS_DEFAULT_ARGS})
 
 install(PROGRAMS org.kde.kube.desktop DESTINATION ${KDE_INSTALL_APPDIR} )
 install(FILES org.kde.kube.appdata.xml DESTINATION ${KDE_INSTALL_DATAROOTDIR}/appdata/)

