diff --git a/framework/domain/mimetreeparser/CMakeLists.txt b/framework/domain/mimetreeparser/CMakeLists.txt
index 64da26568cd4fdd8b1e1a51aa1836f6457eecdf3..8dfdd1a8e6ce89e19b3f047579352f95cac8ed69 100644
--- a/framework/domain/mimetreeparser/CMakeLists.txt
+++ b/framework/domain/mimetreeparser/CMakeLists.txt
@@ -7,7 +7,7 @@ set(mimetreeparser_SRCS
 add_library(mimetreeparser SHARED ${mimetreeparser_SRCS})
 
 qt5_use_modules(mimetreeparser Core Gui)
-target_link_libraries(mimetreeparser KF5::Mime KF5::MimeTreeParser)
+target_link_libraries(mimetreeparser KF5::Mime KF5::MimeTreeParser Gpgmepp QGpgme)
 
 install(TARGETS mimetreeparser
    DESTINATION ${LIB_INSTALL_DIR})
