diff --git a/CMakeLists.txt b/CMakeLists.txt
index bd3e297..ab43f01 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -22,13 +22,15 @@ include(CheckTypeSize)
 find_package(CopperSpice REQUIRED)
 
 # file locations for installing
-if (CMAKE_SYSTEM_NAME MATCHES "Darwin")
+if (CMAKE_SYSTEM_NAME MATCHES "Darwin" AND (NOT CMAKE_BUILD_TYPE MATCHES "MacPorts"))
    include(GNUInstallDirs)
    set(CMAKE_INSTALL_RPATH "@executable_path")
 
-elseif(CMAKE_SYSTEM_NAME MATCHES "(Linux|OpenBSD|FreeBSD|NetBSD|DragonFly)")
+elseif(CMAKE_SYSTEM_NAME MATCHES "(Linux|OpenBSD|FreeBSD|NetBSD|DragonFly)" OR CMAKE_BUILD_TYPE MATCHES "MacPorts")
    include(GNUInstallDirs)
-   set(CMAKE_INSTALL_RPATH "\$ORIGIN")
+   if (NOT CMAKE_BUILD_TYPE MATCHES "MacPorts")
+       set(CMAKE_INSTALL_RPATH "\$ORIGIN")
+   endif()
 
 elseif(MSVC)
    # use defaults
@@ -77,7 +79,9 @@ else()
     set(CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} -Wl,--no-undefined")
 endif()
 
-string(TIMESTAMP BUILD_DATE "%m/%d/%Y")
+if (NOT BUILD_DATE)
+    string(TIMESTAMP BUILD_DATE "%m/%d/%Y")
+endif()
 add_definitions(-DBUILD_DATE="${BUILD_DATE}")
 
 configure_file(
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 4af7209..489b5a1 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -20,10 +20,14 @@ target_link_libraries(PepperMill
    CopperSpice::CsCore
 )
 
-install(TARGETS PepperMill DESTINATION .)
-
-# installs the CopperSpice libraries
-cs_copy_library(CsCore)
+if (CMAKE_BUILD_TYPE MATCHES "MacPorts")
+    message(WARNING "PepperMill goes to ${CMAKE_INSTALL_BINDIR}")
+    install(TARGETS PepperMill DESTINATION ${CMAKE_INSTALL_BINDIR})
+else()
+    install(TARGETS PepperMill DESTINATION .)
+    # installs the CopperSpice libraries
+    cs_copy_library(CsCore)
+endif()
 
 if (CMAKE_SYSTEM_NAME MATCHES "Darwin")
    set(CPACK_DMG_VOLUME_NAME PepperMill)
