diff --git src/doc/documentvalidator.cpp src/doc/documentvalidator.cpp
index fa75716..898f497 100644
--- src/doc/documentvalidator.cpp
+++ src/doc/documentvalidator.cpp
@@ -85,7 +85,7 @@ bool DocumentValidator::validate(const double currentVersion)
         // Make sure Qt locale and C++ locale have the same numeric separator, might not be the case
         // With some locales since C++ and Qt use a different database for locales
         char *separator = localeconv()->decimal_point;
-	if (newLocale.isEmpty()) {
+        if (newLocale.isEmpty() && mlt.attribute("LC_NUMERIC") != QStringLiteral("C")) {
             // Requested locale not available, ask for install
             KMessageBox::sorry(QApplication::activeWindow(), i18n("The document was created in \"%1\" locale, which is not installed on your system. Please install that language pack. Until then, Kdenlive might not be able to correctly open the document.", mlt.attribute("LC_NUMERIC")));
         }
