diff --git a/library/rt-backends/macsynth/macsynth.cpp b/library/rt-backends/macsynth/macsynth.cpp
index c80968b..50074a7 100644
--- a/library/rt-backends/macsynth/macsynth.cpp
+++ b/library/rt-backends/macsynth/macsynth.cpp
@@ -22,7 +22,7 @@
 #include "macsynth.h"
 
 #include <CoreServices/CoreServices.h>
-#include <AudioUnit/AUAudioUnit.h>
+#include <AudioUnit/AudioUnit.h>
 #include <AudioToolbox/AudioToolbox.h>
 
 #define PRETTY_NAME "DLS Synth"

