diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0ac4e19..db5fc10 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -128,7 +128,7 @@ if(BUILD_TESTING)
   add_subdirectory(tests)
 endif()
 
-if(${CMAKE_SYSTEM} MATCHES "Linux")
+#if(${CMAKE_SYSTEM} MATCHES "Linux")
     find_package(Doxygen)
     if(DOXYGEN_FOUND)
         configure_file(
@@ -140,11 +140,11 @@ if(${CMAKE_SYSTEM} MATCHES "Linux")
             WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR})
     endif()
     add_subdirectory(doc)
-    configure_file(drumstick-alsa.pc.in drumstick-alsa.pc IMMEDIATE @ONLY)
+#    configure_file(drumstick-alsa.pc.in drumstick-alsa.pc IMMEDIATE @ONLY)
     configure_file(drumstick.spec.in drumstick.spec IMMEDIATE @ONLY)
-    install(FILES
-	${CMAKE_CURRENT_BINARY_DIR}/drumstick-alsa.pc
-	DESTINATION lib${LIB_SUFFIX}/pkgconfig )
+#    install(FILES
+#	${CMAKE_CURRENT_BINARY_DIR}/drumstick-alsa.pc
+#	DESTINATION lib${LIB_SUFFIX}/pkgconfig )
     # XML mime types
     set( SHARED_MIME_INFO_MINIMUM_VERSION "0.30" )
     set( XDG_MIME_INSTALL_DIR "${CMAKE_INSTALL_DATAROOTDIR}/mime/packages" )
@@ -153,7 +153,7 @@ if(${CMAKE_SYSTEM} MATCHES "Linux")
         install( FILES drumstick.xml DESTINATION ${XDG_MIME_INSTALL_DIR} )
         update_xdg_mimetypes( ${XDG_MIME_INSTALL_DIR} )
     endif()
-endif()
+#endif()
 
 configure_file(drumstick-file.pc.in drumstick-file.pc IMMEDIATE @ONLY)
 install(FILES
diff --git a/library/file/CMakeLists.txt b/library/file/CMakeLists.txt
index 3b4d866..f68843d 100644
--- a/library/file/CMakeLists.txt
+++ b/library/file/CMakeLists.txt
@@ -58,7 +58,6 @@ if(STATIC_DRUMSTICK)
         STATIC_LIB "libdrumstick-file")
 else()
     set_target_properties(drumstick-file PROPERTIES
-        FRAMEWORK TRUE
         VERSION ${VERSION_MAJOR}.${VERSION_MINOR}.${VERSION_PATCH}
         SOVERSION ${VERSION_MAJOR})
 
@@ -68,7 +67,6 @@ install(TARGETS drumstick-file
 	RUNTIME DESTINATION bin
 	ARCHIVE DESTINATION lib${LIB_SUFFIX}
 	LIBRARY DESTINATION lib${LIB_SUFFIX}
-	FRAMEWORK DESTINATION lib${LIB_SUFFIX}
 )
 
 install(FILES ${drumstick-file_HEADERS}
diff --git a/library/file/file.pro b/library/file/file.pro
index 4874447..2907a8c 100644
--- a/library/file/file.pro
+++ b/library/file/file.pro
@@ -24,14 +24,14 @@ static {
     DEFINES += DRUMSTICK_STATIC
 }
 
-macx:!static {
-    TARGET = drumstick-file
-    CONFIG += lib_bundle
-    FRAMEWORK_HEADERS.version = Versions
-    FRAMEWORK_HEADERS.files = $$HEADERS
-    FRAMEWORK_HEADERS.path = Headers/drumstick
-    QMAKE_BUNDLE_DATA += FRAMEWORK_HEADERS
-    #QMAKE_LFLAGS_SONAME = -Wl,-install_name,@executable_path/../Frameworks/
-    QMAKE_SONAME_PREFIX = @rpath
-    QMAKE_TARGET_BUNDLE_PREFIX = net.sourceforge
-}
+#macx:!static {
+#    TARGET = drumstick-file
+#    CONFIG += lib_bundle
+#    FRAMEWORK_HEADERS.version = Versions
+#    FRAMEWORK_HEADERS.files = $$HEADERS
+#    FRAMEWORK_HEADERS.path = Headers/drumstick
+#    QMAKE_BUNDLE_DATA += FRAMEWORK_HEADERS
+#    #QMAKE_LFLAGS_SONAME = -Wl,-install_name,@executable_path/../Frameworks/
+#    QMAKE_SONAME_PREFIX = @rpath
+#    QMAKE_TARGET_BUNDLE_PREFIX = net.sourceforge
+#}
diff --git a/library/rt-backends/rt-backends.pro b/library/rt-backends/rt-backends.pro
index 67776bd..4dbc0e9 100644
--- a/library/rt-backends/rt-backends.pro
+++ b/library/rt-backends/rt-backends.pro
@@ -17,15 +17,15 @@ unix:!macx {
 
 macx {
     SUBDIRS += mac-in mac-out macsynth
-    exists(/Library/Frameworks/FluidSynth.framework/Headers/*) {
-        SUBDIRS += synth
-    }
+#    exists(/Library/Frameworks/FluidSynth.framework/Headers/*) {
+#        SUBDIRS += synth
+#    }
 }
-else {
+# else {
     packagesExist(fluidsynth) {
         SUBDIRS += synth
     }
-}
+# }
 
 win32 {
     SUBDIRS += win-in win-out
diff --git a/library/rt-backends/synth/synth.pro b/library/rt-backends/synth/synth.pro
index f112089..ce96ec1 100644
--- a/library/rt-backends/synth/synth.pro
+++ b/library/rt-backends/synth/synth.pro
@@ -14,16 +14,16 @@ HEADERS += ../../include/rtmidiinput.h \
 
 SOURCES += synthoutput.cpp synthengine.cpp
 
-macx {
-    INCLUDEPATH += /Library/Frameworks/FluidSynth.framework/Headers
-    QMAKE_LFLAGS += -F/Library/Frameworks
-    LIBS += -framework FluidSynth
-} else {
+#macx {
+#    INCLUDEPATH += /Library/Frameworks/FluidSynth.framework/Headers
+#    QMAKE_LFLAGS += -F/Library/Frameworks
+#    LIBS += -framework FluidSynth
+#} else {
     CONFIG += link_pkgconfig
     packagesExist(fluidsynth) {
         PKGCONFIG += fluidsynth
     }
-}
+#}
 
 win32 {
     TARGET_EXT = .dll
diff --git a/library/rt/CMakeLists.txt b/library/rt/CMakeLists.txt
index f4ceeb2..fbc70d1 100644
--- a/library/rt/CMakeLists.txt
+++ b/library/rt/CMakeLists.txt
@@ -112,7 +112,6 @@ if(STATIC_DRUMSTICK)
 else() # STATIC_DRUMSTICK
 
     set_target_properties(drumstick-rt PROPERTIES
-        FRAMEWORK TRUE
         VERSION ${VERSION_MAJOR}.${VERSION_MINOR}.${VERSION_PATCH}
         SOVERSION ${VERSION_MAJOR})
 endif()
@@ -121,7 +120,6 @@ install(TARGETS drumstick-rt
 	RUNTIME DESTINATION bin
 	ARCHIVE DESTINATION lib${LIB_SUFFIX}
 	LIBRARY DESTINATION lib${LIB_SUFFIX}
-	FRAMEWORK DESTINATION lib${LIB_SUFFIX}
 )
 
 install(FILES ${drumstick-rt_HEADERS}
diff --git a/library/rt/rt.pro b/library/rt/rt.pro
index f3c397d..952ec0c 100644
--- a/library/rt/rt.pro
+++ b/library/rt/rt.pro
@@ -22,17 +22,17 @@ win32 {
     TARGET_EXT = .dll
 }
 
-macx:!static {
-    TARGET = drumstick-rt
-    CONFIG += lib_bundle
-    FRAMEWORK_HEADERS.version = Versions
-    FRAMEWORK_HEADERS.files = $$HEADERS
-    FRAMEWORK_HEADERS.path = Headers/drumstick
-    QMAKE_BUNDLE_DATA += FRAMEWORK_HEADERS
-    #QMAKE_LFLAGS_SONAME = -Wl,-install_name,@executable_path/../Frameworks/
-    QMAKE_SONAME_PREFIX = @rpath
-    QMAKE_TARGET_BUNDLE_PREFIX = net.sourceforge
-}
+#macx:!static {
+#    TARGET = drumstick-rt
+#    CONFIG += lib_bundle
+#    FRAMEWORK_HEADERS.version = Versions
+#    FRAMEWORK_HEADERS.files = $$HEADERS
+#    FRAMEWORK_HEADERS.path = Headers/drumstick
+#    QMAKE_BUNDLE_DATA += FRAMEWORK_HEADERS
+#    #QMAKE_LFLAGS_SONAME = -Wl,-install_name,@executable_path/../Frameworks/
+#    QMAKE_SONAME_PREFIX = @rpath
+#    QMAKE_TARGET_BUNDLE_PREFIX = net.sourceforge
+#}
 
 static {
     LIBS += -L$$OUT_PWD/../../build/lib/drumstick
@@ -91,11 +91,11 @@ static {
     packagesExist(fluidsynth) {
         DEFINES += SYNTH_BACKEND
         LIBS += -l$$qtLibraryTarget(drumstick-rt-synth)
-        macx {
-            LIBS += -framework FluidSynth
-        } else {
+#        macx {
+#            LIBS += -framework FluidSynth
+#        } else {
             CONFIG += link_pkgconfig
             PKGCONFIG += fluidsynth
-        }
+#        }
     }
 }
diff --git a/tests/fileTest/fileTest.pro b/tests/fileTest/fileTest.pro
index 0ff4c08..443c38d 100644
--- a/tests/fileTest/fileTest.pro
+++ b/tests/fileTest/fileTest.pro
@@ -8,11 +8,11 @@ SOURCES += filetest.cpp
 DEFINES += SRCDIR=\\\"$$PWD/\\\"
 INCLUDEPATH += . ../../library/include
 DESTDIR = ../../build/bin
-macx:!static {
-    QMAKE_LFLAGS += -F$$OUT_PWD/../../build/lib -L$$OUT_PWD/../../build/lib
-    LIBS += -framework drumstick-file
-} else {
+#macx:!static {
+#    QMAKE_LFLAGS += -F$$OUT_PWD/../../build/lib -L$$OUT_PWD/../../build/lib
+#    LIBS += -framework drumstick-file
+#} else {
     LIBS += -L$$OUT_PWD/../../build/lib \
             -l$$qtLibraryTarget(drumstick-file)
-}
+#}
 
diff --git a/tests/rtTest/rtTest.pro b/tests/rtTest/rtTest.pro
index 141c660..8f1efd1 100644
--- a/tests/rtTest/rtTest.pro
+++ b/tests/rtTest/rtTest.pro
@@ -8,10 +8,10 @@ SOURCES += rttest.cpp
 DEFINES += SRCDIR=\\\"$$PWD/\\\"
 INCLUDEPATH += . ../../library/include
 DESTDIR = ../../build/bin
-macx:!static {
-    QMAKE_LFLAGS += -F$$OUT_PWD/../../build/lib -L$$OUT_PWD/../../build/lib
-    LIBS += -framework drumstick-rt
-} else {
+#macx:!static {
+#    QMAKE_LFLAGS += -F$$OUT_PWD/../../build/lib -L$$OUT_PWD/../../build/lib
+#    LIBS += -framework drumstick-rt
+#} else {
     LIBS += -L$$OUT_PWD/../../build/lib \
             -l$$qtLibraryTarget(drumstick-rt)
-}
+#}
diff --git a/utils/dumpove/dumpove.pro b/utils/dumpove/dumpove.pro
index 6127162..609a331 100644
--- a/utils/dumpove/dumpove.pro
+++ b/utils/dumpove/dumpove.pro
@@ -8,12 +8,12 @@ include (../../global.pri)
 HEADERS += dumpove.h
 SOURCES += dumpove.cpp
 
-macx:!static {
-    QMAKE_LFLAGS += -F$$OUT_PWD/../../build/lib -L$$OUT_PWD/../../build/lib
-    LIBS += -framework drumstick-file
-    LIBS += -l$$qtLibraryTarget(drumstick-common)
-} else {
+#macx:!static {
+#    QMAKE_LFLAGS += -F$$OUT_PWD/../../build/lib -L$$OUT_PWD/../../build/lib
+#    LIBS += -framework drumstick-file
+#    LIBS += -l$$qtLibraryTarget(drumstick-common)
+#} else {
     LIBS = -L$$OUT_PWD/../../build/lib \
         -l$$qtLibraryTarget(drumstick-common) \
         -l$$qtLibraryTarget(drumstick-file)
-}
+#}
diff --git a/utils/dumpsmf/dumpsmf.pro b/utils/dumpsmf/dumpsmf.pro
index bdbaab7..036c502 100644
--- a/utils/dumpsmf/dumpsmf.pro
+++ b/utils/dumpsmf/dumpsmf.pro
@@ -8,12 +8,12 @@ include (../../global.pri)
 HEADERS += dumpsmf.h
 SOURCES += dumpsmf.cpp
 
-macx:!static {
-    QMAKE_LFLAGS += -F$$OUT_PWD/../../build/lib -L$$OUT_PWD/../../build/lib
-    LIBS += -framework drumstick-file
-    LIBS += -l$$qtLibraryTarget(drumstick-common)
-} else {
+#macx:!static {
+#    QMAKE_LFLAGS += -F$$OUT_PWD/../../build/lib -L$$OUT_PWD/../../build/lib
+#    LIBS += -framework drumstick-file
+#    LIBS += -l$$qtLibraryTarget(drumstick-common)
+#} else {
     LIBS = -L$$OUT_PWD/../../build/lib \
         -l$$qtLibraryTarget(drumstick-common) \
         -l$$qtLibraryTarget(drumstick-file)
-}
+#}
diff --git a/utils/dumpwrk/dumpwrk.pro b/utils/dumpwrk/dumpwrk.pro
index 49e2fd3..2ac7280 100644
--- a/utils/dumpwrk/dumpwrk.pro
+++ b/utils/dumpwrk/dumpwrk.pro
@@ -8,12 +8,12 @@ include (../../global.pri)
 HEADERS += dumpwrk.h
 SOURCES += dumpwrk.cpp
 
-macx:!static {
-    QMAKE_LFLAGS += -F$$OUT_PWD/../../build/lib -L$$OUT_PWD/../../build/lib
-    LIBS += -framework drumstick-file
-    LIBS += -l$$qtLibraryTarget(drumstick-common)
-} else {
+#macx:!static {
+#    QMAKE_LFLAGS += -F$$OUT_PWD/../../build/lib -L$$OUT_PWD/../../build/lib
+#    LIBS += -framework drumstick-file
+#    LIBS += -l$$qtLibraryTarget(drumstick-common)
+#} else {
     LIBS = -L$$OUT_PWD/../../build/lib \
         -l$$qtLibraryTarget(drumstick-common) \
         -l$$qtLibraryTarget(drumstick-file)
-}
+#}
diff --git a/utils/vpiano/vpiano.pro b/utils/vpiano/vpiano.pro
index 7d35eb2..4168d57 100644
--- a/utils/vpiano/vpiano.pro
+++ b/utils/vpiano/vpiano.pro
@@ -25,15 +25,16 @@ RESOURCES += pianokeybd.qrc
 
 # libs
 macx:!static {
-    QMAKE_LFLAGS += -F$$OUT_PWD/../../build/lib -L$$OUT_PWD/../../build/lib
-    LIBS += -framework drumstick-rt
-    LIBS += -l$$qtLibraryTarget(drumstick-common)
+#    QMAKE_LFLAGS += -F$$OUT_PWD/../../build/lib -L$$OUT_PWD/../../build/lib
+#    LIBS += -framework drumstick-rt
+#    LIBS += -l$$qtLibraryTarget(drumstick-common)
     ICON = ../../icons/drumstick.icns
-} else {
+}
+#else {
     LIBS += -L$$OUT_PWD/../../build/lib/
     LIBS += -l$$qtLibraryTarget(drumstick-common) \
             -l$$qtLibraryTarget(drumstick-rt)
-}
+#}
 
 static {
     LIBS += -L$$OUT_PWD/../../build/lib/drumstick/
@@ -43,13 +44,13 @@ static {
     packagesExist(fluidsynth) {
         DEFINES += SYNTH_BACKEND
         LIBS += -l$$qtLibraryTarget(drumstick-rt-synth)
-        macx {
-            QMAKE_LFLAGS += -F/Library/Frameworks
-            LIBS += -framework FluidSynth
-        } else {
+#        macx {
+#            QMAKE_LFLAGS += -F/Library/Frameworks
+#            LIBS += -framework FluidSynth
+#        } else {
             CONFIG += link_pkgconfig
             PKGCONFIG += fluidsynth
-        }
+#        }
     }
 
     linux* {
