diff --git Magick++/bin/Magick++-config.in Magick++/bin/Magick++-config.in
index 67aa72d..61e3032 100755
--- Magick++/bin/Magick++-config.in
+++ Magick++/bin/Magick++-config.in
@@ -19,6 +19,8 @@ if test $# -eq 0; then
       exit 1
 fi
 
+export PKG_CONFIG_PATH="@exec_prefix@/lib/pkgconfig:${PKG_CONFIG_PATH}"
+
 while test $# -gt 0; do
   case "$1" in
     -*=*) optarg=`echo "$1" | sed 's/[-_a-zA-Z0-9]*=//'` ;;
diff --git magick/Magick-config.in magick/Magick-config.in
index e42a020..09bbd54 100644
--- magick/Magick-config.in
+++ magick/Magick-config.in
@@ -16,6 +16,8 @@ if test $# -eq 0; then
       exit 1
 fi
 
+export PKG_CONFIG_PATH="@exec_prefix@/lib/pkgconfig:${PKG_CONFIG_PATH}"
+
 while test $# -gt 0; do
   case "$1" in
     -*=*) optarg=`echo "$1" | sed 's/[-_a-zA-Z0-9]*=//'` ;;
diff --git magick/MagickCore-config.in magick/MagickCore-config.in
index 2e74c3b..736375d 100644
--- magick/MagickCore-config.in
+++ magick/MagickCore-config.in
@@ -16,6 +16,8 @@ if test $# -eq 0; then
       exit 1
 fi
 
+export PKG_CONFIG_PATH="@exec_prefix@/lib/pkgconfig:${PKG_CONFIG_PATH}"
+
 while test $# -gt 0; do
   case "$1" in
     -*=*) optarg=`echo "$1" | sed 's/[-_a-zA-Z0-9]*=//'` ;;
diff --git wand/MagickWand-config.in wand/MagickWand-config.in
index b2e1ca8..952ad5a 100644
--- wand/MagickWand-config.in
+++ wand/MagickWand-config.in
@@ -16,6 +16,8 @@ if test $# -eq 0; then
       exit 1
 fi
 
+export PKG_CONFIG_PATH="@exec_prefix@/lib/pkgconfig:${PKG_CONFIG_PATH}"
+
 while test $# -gt 0; do
   case "$1" in
     -*=*) optarg=`echo "$1" | sed 's/[-_a-zA-Z0-9]*=//'` ;;
diff --git wand/Wand-config.in wand/Wand-config.in
index 024700f..fd4a45e 100644
--- wand/Wand-config.in
+++ wand/Wand-config.in
@@ -16,6 +16,8 @@ if test $# -eq 0; then
       exit 1
 fi
 
+export PKG_CONFIG_PATH="@exec_prefix@/lib/pkgconfig:${PKG_CONFIG_PATH}"
+
 while test $# -gt 0; do
   case "$1" in
     -*=*) optarg=`echo "$1" | sed 's/[-_a-zA-Z0-9]*=//'` ;;
