--- a/qtbase/src/widgets/kernel/orig.qaction.cpp	2023-07-12 18:19:43.926846935 +0200
+++ b/qtbase/src/widgets/kernel/qaction.cpp	2023-07-12 18:20:01.007251962 +0200
@@ -565,12 +565,12 @@
     Q_D(QAction);
     for (int i = d->widgets.size()-1; i >= 0; --i) {
         QWidget *w = d->widgets.at(i);
-        w->removeAction(this);
+        if (w) w->removeAction(this);
     }
 #if QT_CONFIG(graphicsview)
     for (int i = d->graphicsWidgets.size()-1; i >= 0; --i) {
         QGraphicsWidget *w = d->graphicsWidgets.at(i);
-        w->removeAction(this);
+        if (w) w->removeAction(this);
     }
 #endif
     if (d->group)
