diff --git a/qtbase/mkspecs/features/link_ltcg.prf b/qtbase/mkspecs/features/link_ltcg.prf
index 4da061e..90f17e9 100644
--- a/qtbase/mkspecs/features/link_ltcg.prf
+++ b/qtbase/mkspecs/features/link_ltcg.prf
@@ -2,9 +2,9 @@ gcc {
     # on Windows, MinGW's support for -flto=N is broken
     intel_icc {
         QMAKE_LFLAGS_LTCG ~= s/-ipo/-ipo=$$QMAKE_HOST.cpu_count/
-    } else: !clang:!equals(QMAKE_HOST.os, Windows):greaterThan(QMAKE_HOST.cpu_count, 1) {
-        # Override LTO number of jobs
-        QMAKE_LFLAGS_LTCG ~= s/^-flto$/-flto=$$QMAKE_HOST.cpu_count/
+    } else: !clang:!equals(QMAKE_HOST.os, Windows) {
+        # Override LTO number of jobs, but let make determine how many via its job server
+        QMAKE_LFLAGS_LTCG ~= s/^-flto$/-flto=auto/
     }
 
     # When doing link-time code generation, we need to pass the compiler
