diff --git a/qtbase/src/corelib/kernel/qeventdispatcher_glib.cpp b/qtbase/src/corelib/kernel/qeventdispatcher_glib.cpp
index 8cefa7a..6450c47 100644
--- a/qtbase/src/corelib/kernel/qeventdispatcher_glib.cpp
+++ b/qtbase/src/corelib/kernel/qeventdispatcher_glib.cpp
@@ -260,8 +260,9 @@ static gboolean postEventSourcePrepare(GSource *s, gint *timeout)
     *timeout = canWait ? -1 : 0;
 
     GPostEventSource *source = reinterpret_cast<GPostEventSource *>(s);
-    return (!canWait
-            || (source->serialNumber.load() != source->lastSerialNumber));
+    // RJVB: wakeUpCalled backported from 5.12
+    source->d->wakeUpCalled = source->serialNumber.load() != source->lastSerialNumber;
+    return !canWait || source->d->wakeUpCalled;
 }
 
 static gboolean postEventSourceCheck(GSource *source)
diff --git a/qtbase/src/corelib/kernel/qeventdispatcher_glib_p.h b/qtbase/src/corelib/kernel/qeventdispatcher_glib_p.h
index 8e9925d..4494b81 100644
--- a/qtbase/src/corelib/kernel/qeventdispatcher_glib_p.h
+++ b/qtbase/src/corelib/kernel/qeventdispatcher_glib_p.h
@@ -108,10 +108,15 @@ public:
     GSocketNotifierSource *socketNotifierSource;
     GTimerSource *timerSource;
     GIdleTimerSource *idleTimerSource;
+    // RJVB: backported from 5.12
+    bool wakeUpCalled = true;
 
     void runTimersOnceWithNormalPriority();
 };
 
+// RJVB
+#define QEVENTDISPATCHERGLIBPRIVATE_HAS_WAKEUP_CALLED
+
 QT_END_NAMESPACE
 
 #endif // QEVENTDISPATCHER_GLIB_P_H
