diff --git a/qtserialbus/src/plugins/canbus/socketcan/socketcanbackend.cpp b/qtserialbus/src/plugins/canbus/socketcan/socketcanbackend.cpp
index 2ec4379..92a89ab 100644
--- a/qtserialbus/src/plugins/canbus/socketcan/socketcanbackend.cpp
+++ b/qtserialbus/src/plugins/canbus/socketcan/socketcanbackend.cpp
@@ -49,6 +49,7 @@
 #include <net/if.h>
 #include <sys/ioctl.h>
 #include <sys/time.h>
+#include <linux/sockios.h>
 
 #ifndef CANFD_MTU
 // CAN FD support was added by Linux kernel 3.6
