--- a/qtwebengine-opensource-src-5.8.0/src/3rdparty/chromium/net/cert/x509_certificate_mac.cc	2019-12-09 12:16:46.000000000 +0100
+++ b/qtwebengine-opensource-src-5.8.0/src/3rdparty/chromium/net/cert/x509_certificate_mac.cc	2019-12-09 12:17:04.000000000 +0100
@@ -76,7 +76,7 @@
 void GetCertDateForOID(const x509_util::CSSMCachedCertificate& cached_cert,
                        const CSSM_OID* oid,
                        Time* result) {
-  *result = Time::Time();
+  *result = Time();
 
   x509_util::CSSMFieldValue field;
   OSStatus status = cached_cert.GetField(oid, &field);
