diff --git a/qtbase/src/plugins/platforms/cocoa/qcocoamenu.mm b/qtbase/src/plugins/platforms/cocoa/qcocoamenu.mm
index e177a24..899f523 100644
--- a/qtbase/src/plugins/platforms/cocoa/qcocoamenu.mm
+++ b/qtbase/src/plugins/platforms/cocoa/qcocoamenu.mm
@@ -343,6 +343,19 @@ void QCocoaMenu::insertNative(QCocoaMenuItem *item, QCocoaMenuItem *beforeItem)
     if (item->isMerged())
         return;
 
+    NSMenu *m = [item->nsItem() menu];
+    if (m) {
+        QString mTitle = QCFString::toQString([m title]);
+        if (beforeItem) {
+            qWarning() << Q_FUNC_INFO << "Menu item" << item->text() << "is already in menu" << mTitle
+                       << "after item" << beforeItem->text()
+                       << ", remove it from the other menu first before inserting into" << QCFString::toQString([nsMenu() title]);
+        } else {
+            qWarning() << Q_FUNC_INFO << "Menu item" << item->text() << "is already in menu" << mTitle
+                       << ", remove it from the other menu first before inserting into" << QCFString::toQString([nsMenu() title]);
+        }
+        return;
+    }
     // if the item we're inserting before is merged, skip along until
     // we find a non-merged real item to insert ahead of.
     while (beforeItem && beforeItem->isMerged()) {
@@ -395,7 +408,8 @@ void QCocoaMenu::removeMenuItem(QPlatformMenuItem *menuItem)
     m_menuItems.removeOne(cocoaItem);
     if (!cocoaItem->isMerged()) {
         if (m_nativeMenu != [cocoaItem->nsItem() menu]) {
-            qWarning("Item to remove does not belong to this menu");
+            qWarning() << Q_FUNC_INFO << "Item" << cocoaItem->text() << "to remove does not belong to this menu"
+                << QCFString::toQString([nsMenu() title]);
             return;
         }
         [m_nativeMenu removeItem: cocoaItem->nsItem()];
