--- autotests/CMakeLists.txt.orig	2017-06-05 23:19:31.000000000 +0200
+++ autotests/CMakeLists.txt	2017-06-13 18:39:35.000000000 +0200
@@ -5,6 +5,7 @@
 
 include(ECMMarkAsTest)
 include(ECMAddTests)
+include(ECMMarkNonGuiExecutable)
 
 find_package(KF5CoreAddons REQUIRED)
 find_package(KF5XmlGui REQUIRED)
@@ -21,6 +22,7 @@
 MACRO(PLASMA_UNIT_TESTS)
        FOREACH(_testname ${ARGN})
                add_executable(${_testname} ${_testname}.cpp)
+               ecm_mark_nongui_executable(${_testname})
                target_include_directories(${_testname} PRIVATE "$<BUILD_INTERFACE:$<TARGET_PROPERTY:KF5PlasmaQuick,INCLUDE_DIRECTORIES>>;")
                target_link_libraries(${_testname} Qt5::Qml Qt5::Test
                                      KF5::Plasma KF5::PlasmaQuick
@@ -54,6 +56,7 @@
 )
 
 add_executable(storagetest storagetest.cpp ../src/plasma/private/storage.cpp ../src/plasma/private/storagethread.cpp ../src/plasma/debug_p.cpp)
+ecm_mark_nongui_executable(storagetest)
 target_link_libraries(storagetest Qt5::Gui Qt5::Test Qt5::Sql KF5::KIOCore KF5::Plasma KF5::CoreAddons)
 
 if(HAVE_X11)
@@ -73,6 +76,7 @@
     ../src/declarativeimports/core/datamodel.cpp
     ../src/declarativeimports/core/datasource.cpp
     )
+ecm_mark_nongui_executable(sortfiltermodeltest)
 
 target_link_libraries(sortfiltermodeltest
     KF5::Plasma
