diff --git src/TerminalDisplay.cpp src/TerminalDisplay.cpp
index 8c241fc..ae28626 100644
--- src/TerminalDisplay.cpp
+++ src/TerminalDisplay.cpp
@@ -2867,7 +2867,9 @@ void TerminalDisplay::copyToX11Selection()
     mimeData->setText(text);
     mimeData->setHtml(html);
 
-    QApplication::clipboard()->setMimeData(mimeData, QClipboard::Selection);
+    if (QApplication::clipboard()->supportsSelection()) {
+        QApplication::clipboard()->setMimeData(mimeData, QClipboard::Selection);
+    }
 
     if (_autoCopySelectedText)
         QApplication::clipboard()->setMimeData(mimeData, QClipboard::Clipboard);
@@ -2898,8 +2900,10 @@ void TerminalDisplay::pasteFromClipboard(bool appendEnter)
 
 void TerminalDisplay::pasteFromX11Selection(bool appendEnter)
 {
-    QString text = QApplication::clipboard()->text(QClipboard::Selection);
-    doPaste(text, appendEnter);
+    if (QApplication::clipboard()->supportsSelection()) {
+        QString text = QApplication::clipboard()->text(QClipboard::Selection);
+        doPaste(text, appendEnter);
+    }
 }
 
 /* ------------------------------------------------------------------------- */
