# commit 17cf652a9cd2f2c2c35b20c42cfcbfad8b83cb43
diff --git src/main.cpp src/main.cpp
index c7d797c..2b70aff 100644
--- src/main.cpp
+++ src/main.cpp
@@ -158,15 +158,14 @@
     // of Konsole is started.
     QObject::connect(&dbusService, &KDBusService::activateRequested, &konsoleApp, &Application::slotActivateRequested);
 
-    if (!konsoleApp.newInstance()) {
+    if (app->isSessionRestored()) {
+        restoreSession(konsoleApp);
+    } else if (!konsoleApp.newInstance()) {
         // An argument that printed something and mean we should just quit was passed.
         delete app;
         return 0;
     }
 
-    if (app->isSessionRestored())
-        restoreSession(konsoleApp);
-
     // Since we've allocated the QApplication on the heap for the KDBusService workaround,
     // we need to delete it manually before returning from main().
     int ret = app->exec();
