diff --git generators/CMakeLists.txt generators/CMakeLists.txt
index f8edfe9..11c37f5 100644
--- generators/CMakeLists.txt
+++ generators/CMakeLists.txt
@@ -76,8 +76,8 @@ set_package_properties("libepub" PROPERTIES
         TYPE RECOMMENDED
         PURPOSE "Support for EPub documents in Okular.")
 
-find_package(QMobipocket)
-set_package_properties("libqmobipocket" PROPERTIES
+find_package(QMobipocket5)
+set_package_properties("libqmobipocket5" PROPERTIES
         DESCRIPTION  "A library for reading Mobipocket documents"
         URL "https://projects.kde.org/projects/kde/kdegraphics/kdegraphics-mobipocket"
         TYPE RECOMMENDED
