diff --git a/generators/CMakeLists.txt b/generators/CMakeLists.txt
index 3a223ab..5cd3c4b 100644
--- a/generators/CMakeLists.txt
+++ b/generators/CMakeLists.txt
@@ -76,12 +76,12 @@ set_package_properties("libepub" PROPERTIES
         TYPE RECOMMENDED
         PURPOSE "Support for EPub documents in Okular.")
 
-find_package(QMobipocket)
-set_package_properties("libqmobipocket" PROPERTIES
-        DESCRIPTION  "A library for reading Mobipocket documents"
-        URL "https://projects.kde.org/projects/kde/kdegraphics/kdegraphics-mobipocket"
-        TYPE RECOMMENDED
-        PURPOSE "Support for Mobipocket documents in Okular.")
+#find_package(QMobipocket)
+#set_package_properties("libqmobipocket" PROPERTIES
+#        DESCRIPTION  "A library for reading Mobipocket documents"
+#        URL "https://projects.kde.org/projects/kde/kdegraphics/kdegraphics-mobipocket"
+#        TYPE RECOMMENDED
+#        PURPOSE "Support for Mobipocket documents in Okular.")
 
 # let's enable the generators properly configured
 
