diff --git a/src/qcocoa-qpa/qnsview.mm b/src/qcocoa-qpa/qnsview.mm
index 67ce005d815f68f31507722374edeb47cd08ff7c..ff26e4ea45f422ddc9e3d9a80baf8adbf904791a 100644
--- a/src/qcocoa-qpa/qnsview.mm
+++ b/src/qcocoa-qpa/qnsview.mm
@@ -841,7 +841,7 @@ static bool _q_dontOverrideCtrlLMB = false;
         // Check if the click is outside all popups.
         bool inside = false;
         QPointF qtScreenPoint = qt_mac_flipPoint([self screenMousePoint:theEvent]);
-        for (QList<QCocoaWindow *>::const_iterator it = popups->begin(); it != popups->end(); ++it) {
+        for (QList<QCocoaWindow *>::const_iterator it = popups->cbegin(); it != popups->cend(); ++it) {
             if ((*it)->geometry().contains(qtScreenPoint.toPoint())) {
                 inside = true;
                 break;
