diff --git Source/bmalloc/libpas/src/libpas/orig.pas_probabilistic_guard_malloc_allocator.c Source/bmalloc/libpas/src/libpas/pas_probabilistic_guard_malloc_allocator.c
index d4e79ba..f70c3be 100644
--- Source/bmalloc/libpas/src/libpas/orig.pas_probabilistic_guard_malloc_allocator.c
+++ Source/bmalloc/libpas/src/libpas/pas_probabilistic_guard_malloc_allocator.c
@@ -114,6 +114,10 @@ pas_allocation_result pas_probabilistic_guard_malloc_allocate(pas_large_heap* la
      * ensure physical addresses are released
      * TODO: investigate using MADV_FREE_REUSABLE instead
      */
+#if defined(linux) && !defined(MADV_FREE)
+#warning "Defining MADV_FREE (8)"
+#define MADV_FREE 8
+#endif
     int madvise_res = madvise((void *) upper_guard_page, page_size, MADV_FREE);
     PAS_ASSERT(!madvise_res);
 
