diff --git Source/WebCore/PAL/pal/crypto/gcrypt/orig.Initialization.h Source/WebCore/PAL/pal/crypto/gcrypt/Initialization.h
index ef9a09d..fab6577 100644
--- Source/WebCore/PAL/pal/crypto/gcrypt/orig.Initialization.h
+++ Source/WebCore/PAL/pal/crypto/gcrypt/Initialization.h
@@ -37,9 +37,16 @@ static inline void initialize()
     // returned version string.
     gcry_check_version(nullptr);
 
+    static char done = 0;
+    if (done) {
+	    return;
+    }
+
     // Pre-allocate 16kB of secure memory and finish the initialization.
-    gcry_control(GCRYCTL_INIT_SECMEM, 16384, nullptr);
-    gcry_control(GCRYCTL_INITIALIZATION_FINISHED, nullptr);
+    if (gcry_control(GCRYCTL_INIT_SECMEM, 16384, nullptr) == 0 &&
+	    gcry_control(GCRYCTL_INITIALIZATION_FINISHED, nullptr) == 0) {
+	    done = 1;
+    }
 }
 
 } // namespace PAL
