diff --git Source/WTF/wtf/orig.CompactPointerTuple.h Source/WTF/wtf/CompactPointerTuple.h
index 0659138..f18beb4 100644
--- Source/WTF/wtf/orig.CompactPointerTuple.h
+++ Source/WTF/wtf/CompactPointerTuple.h
@@ -33,7 +33,7 @@
 #include <wtf/StdLibExtras.h>
 
 #if OS(DARWIN)
-#include <mach/vm_param.h>
+#include <mach/mach_vm.h>
 #endif
 
 namespace WTF {
diff --git Source/WTF/wtf/orig.Packed.h Source/WTF/wtf/Packed.h
index 274b963..3c769bd 100644
--- Source/WTF/wtf/orig.Packed.h
+++ Source/WTF/wtf/Packed.h
@@ -34,7 +34,7 @@
 #include <wtf/UnalignedAccess.h>
 
 #if OS(DARWIN)
-#include <mach/vm_param.h>
+#include <mach/mach_vm.h>
 #endif
 
 namespace WTF {
diff --git Source/JavaScriptCore/bytecode/orig.CodeOrigin.h Source/JavaScriptCore/bytecode/CodeOrigin.h
index 3497138..128cf9e 100644
--- Source/JavaScriptCore/bytecode/orig.CodeOrigin.h
+++ Source/JavaScriptCore/bytecode/CodeOrigin.h
@@ -35,7 +35,7 @@
 #include <wtf/Vector.h>
 
 #if OS(DARWIN)
-#include <mach/vm_param.h>
+#include <mach/mach_vm.h>
 #endif
 
 namespace JSC {
