--- Source/WebKit/PlatformGTK.cmake.orig	2023-04-12 08:51:20.000000000 -0400
+++ Source/WebKit/PlatformGTK.cmake	2023-05-16 14:52:50.000000000 -0400
@@ -470,7 +470,7 @@
 # Commands for building the built-in injected bundle.
 add_library(webkit${WEBKITGTK_API_INFIX}gtkinjectedbundle MODULE "${WEBKIT_DIR}/WebProcess/InjectedBundle/API/glib/WebKitInjectedBundleMain.cpp")
 ADD_WEBKIT_PREFIX_HEADER(webkit${WEBKITGTK_API_INFIX}gtkinjectedbundle)
-target_link_libraries(webkit${WEBKITGTK_API_INFIX}gtkinjectedbundle WebKit)
+target_link_libraries(webkit${WEBKITGTK_API_INFIX}gtkinjectedbundle WebKit WebCore)
 
 target_include_directories(webkit${WEBKITGTK_API_INFIX}gtkinjectedbundle PRIVATE
     $<TARGET_PROPERTY:WebKit,INCLUDE_DIRECTORIES>
