--- Source/WebCore/platform/graphics/gstreamer/mse/WebKitMediaSourceGStreamer.cpp.orig	2021-09-01 10:16:50
+++ Source/WebCore/platform/graphics/gstreamer/mse/WebKitMediaSourceGStreamer.cpp	2024-12-12 16:26:08
@@ -711,7 +711,7 @@
         double rate;
         GstFormat format;
         GstSeekType startType;
-        int64_t start;
+        gint64 start;
         gst_event_parse_seek(event, &rate, &format, nullptr, &startType, &start, nullptr, nullptr);
         if (format != GST_FORMAT_TIME || startType != GST_SEEK_TYPE_SET) {
             GST_ERROR_OBJECT(element, "Rejecting unsupported seek event: %" GST_PTR_FORMAT, event);
