--- Source/WebKit/UIProcess/Launcher/glib/ProcessLauncherGLib.cpp.orig	2023-05-29 08:59:44
+++ Source/WebKit/UIProcess/Launcher/glib/ProcessLauncherGLib.cpp	2024-12-23 15:22:35
@@ -49,6 +49,8 @@
 
 #if !USE(SYSTEM_MALLOC) && OS(LINUX)
 #include <bmalloc/valgrind.h>
+elif !USE(SYSTEM_MALLOC) && OS(DARWIN)
+#include <bmalloc/valgrind.h>
 #endif
 
 namespace WebKit {
@@ -92,9 +94,12 @@
 static bool isSandboxEnabled(const ProcessLauncher::LaunchOptions& launchOptions)
 {
 #if !USE(SYSTEM_MALLOC)
+// why isnt this defined here????
+#ifdef RUNNING_ON_VALGRIND
     if (RUNNING_ON_VALGRIND)
         return false;
 #endif
+#endif
 
     if (const char* sandboxEnv = g_getenv("WEBKIT_DISABLE_SANDBOX_THIS_IS_DANGEROUS")) {
         if (!strcmp(sandboxEnv, "1"))
