--- Source/WebCore/Headers.cmake.orig	2021-09-01 10:16:50
+++ Source/WebCore/Headers.cmake	2024-12-12 16:59:03
@@ -1397,6 +1397,8 @@
     platform/graphics/transforms/TransformOperations.h
     platform/graphics/transforms/TransformationMatrix.h
     platform/graphics/transforms/TranslateTransformOperation.h
+
+    platform/graphics/texmap/coordinated/CoordinatedGraphicsLayer.h
 
     platform/mediacapabilities/AudioConfiguration.h
     platform/mediacapabilities/ColorGamut.h
