diff --git Source/WebCore/platform/graphics/opengl/orig.GraphicsContextGLOpenGLCommon.cpp Source/WebCore/platform/graphics/opengl/GraphicsContextGLOpenGLCommon.cpp
index 1e035a7..a106a5e 100644
--- Source/WebCore/platform/graphics/opengl/orig.GraphicsContextGLOpenGLCommon.cpp
+++ Source/WebCore/platform/graphics/opengl/GraphicsContextGLOpenGLCommon.cpp
@@ -2029,11 +2029,15 @@ void GraphicsContextGLOpenGL::vertexAttribDivisor(GCGLuint index, GCGLuint divis
     getExtensions().vertexAttribDivisor(index, divisor);
 }
 
-#if HAVE(OPENGL_4) && ENABLE(WEBGL2)
+#if (defined(__APPLE__) || HAVE(OPENGL_4)) && ENABLE(WEBGL2)
 void GraphicsContextGLOpenGL::primitiveRestartIndex(GCGLuint index)
 {
     makeContextCurrent();
+#ifdef __APPLE__
+    // we're a shim; Apple doesn't have OpenGL 3
+#else
     ::glPrimitiveRestartIndex(index);
+#endif
 }
 #endif
 
