diff --git Source/cmake/orig.FindGI.cmake Source/cmake/FindGI.cmake
index fdc56b2..9b64ca7 100644
--- Source/cmake/orig.FindGI.cmake
+++ Source/cmake/FindGI.cmake
@@ -342,7 +342,7 @@ function(GI_INTROSPECT namespace nsversion header)
         VERBATIM
         COMMAND_EXPAND_LISTS
         COMMAND ${CMAKE_COMMAND} -E env "CC=${CMAKE_C_COMPILER}" "CFLAGS=${CMAKE_C_FLAGS}"
-            "${GI_SCANNER_EXE}" --quiet --warn-all --warn-error --no-libtool
+            "${GI_SCANNER_EXE}" --quiet --warn-all --no-libtool
             "--output=${gir_path}"
             "--library=$<TARGET_FILE_BASE_NAME:${opt_TARGET}>"
             "--library-path=$<TARGET_FILE_DIR:${opt_TARGET}>"
