diff --git Source/ThirdParty/ANGLE/src/libANGLE/orig.Display.cpp Source/ThirdParty/ANGLE/src/libANGLE/Display.cpp
index 1278657..d1ba9f4 100644
--- Source/ThirdParty/ANGLE/src/libANGLE/orig.Display.cpp
+++ Source/ThirdParty/ANGLE/src/libANGLE/Display.cpp
@@ -395,12 +395,14 @@ rx::DisplayImpl *CreateDisplayFromAttribs(EGLAttrib displayType,
                 break;
             }
 #        endif
+#if !defined(ANGLE_PLATFORM_MACOS)
             if (deviceType == EGL_PLATFORM_ANGLE_DEVICE_TYPE_EGL_ANGLE)
             {
                 impl = new rx::DisplayEGL(state);
                 break;
             }
             else
+#endif
             {
 #        if defined(ANGLE_USE_X11)
                 if (platformType == EGL_PLATFORM_X11_EXT)
@@ -408,7 +410,9 @@ rx::DisplayImpl *CreateDisplayFromAttribs(EGLAttrib displayType,
                     impl = new rx::DisplayGLX(state);
                     break;
                 }
+			 else
 #        endif
+                    impl = nullptr;
             }
 #    elif defined(ANGLE_PLATFORM_ANDROID)
             impl = new rx::DisplayAndroid(state);
diff --git Source/WebCore/platform/graphics/texmap/orig.TextureMapperPlatformLayerDmabuf.cpp Source/WebCore/platform/graphics/texmap/TextureMapperPlatformLayerDmabuf.cpp
index e35a654..37ec6ce 100644
--- Source/WebCore/platform/graphics/texmap/orig.TextureMapperPlatformLayerDmabuf.cpp
+++ Source/WebCore/platform/graphics/texmap/TextureMapperPlatformLayerDmabuf.cpp
@@ -27,7 +27,7 @@
 #include "config.h"
 #include "TextureMapperPlatformLayerDmabuf.h"
 
-#if USE(ANGLE) && USE(NICOSIA)
+#if USE(ANGLE) && USE(NICOSIA) && USE(EGL)
 
 #include "GLContextEGL.h"
 
diff --git Source/WebCore/platform/graphics/texmap/orig.TextureMapperPlatformLayerDmabuf.h Source/WebCore/platform/graphics/texmap/TextureMapperPlatformLayerDmabuf.h
index 253a20f..21e4acd 100644
--- Source/WebCore/platform/graphics/texmap/orig.TextureMapperPlatformLayerDmabuf.h
+++ Source/WebCore/platform/graphics/texmap/TextureMapperPlatformLayerDmabuf.h
@@ -26,7 +26,7 @@
 
 #pragma once
 
-#if USE(ANGLE) && USE(NICOSIA)
+#if USE(ANGLE) && USE(NICOSIA) && USE(EGL)
 
 #include "TextureMapperPlatformLayerBuffer.h"
 
