diff --git orig.CMakeLists.txt CMakeLists.txt
index a79000b..9f8f6d5 100644
--- orig.CMakeLists.txt
+++ CMakeLists.txt
@@ -11,6 +11,7 @@
 cmake_minimum_required(VERSION 3.10)
 
 project(WebKit)
+include(FeatureSummary)
 
 if (NOT CMAKE_BUILD_TYPE)
     message(WARNING "No CMAKE_BUILD_TYPE value specified, defaulting to RelWithDebInfo.")
@@ -202,3 +203,4 @@ endif ()
 # Print the features list last, for maximum visibility.
 # -----------------------------------------------------------------------------
 PRINT_WEBKIT_OPTIONS()
+feature_summary(WHAT ALL   FATAL_ON_MISSING_REQUIRED_PACKAGES)
