diff --git Source/WebKit/Platform/IPC/orig.Connection.cpp Source/WebKit/Platform/IPC/Connection.cpp
index 3ff418a..b8077db 100644
--- Source/WebKit/Platform/IPC/orig.Connection.cpp
+++ Source/WebKit/Platform/IPC/Connection.cpp
@@ -668,7 +668,7 @@ std::unique_ptr<Decoder> Connection::waitForSyncReply(uint64_t syncRequestID, St
         timedOut = !SyncMessageState::singleton().wait(absoluteTime);
     }
 
-#if OS(DARWIN)
+#if !USE(UNIX_DOMAIN_SOCKETS) && OS(DARWIN)
     RELEASE_LOG_ERROR(IPC, "Connection::waitForSyncReply: Timed-out while waiting for reply for %{public}s::%{public}s from process %d, id = %" PRIu64, messageReceiverName.toString().data(), messageName.toString().data(), remoteProcessID(), syncRequestID);
 #else
     RELEASE_LOG_ERROR(IPC, "Connection::waitForSyncReply: Timed-out while waiting for reply for %s::%s, id = %" PRIu64, messageReceiverName.toString().data(), messageName.toString().data(), syncRequestID);
