diff --git Source/JavaScriptCore/runtime/orig.DateInstance.cpp Source/JavaScriptCore/runtime/DateInstance.cpp
index 915854b..76d77b4 100644
--- Source/JavaScriptCore/runtime/orig.DateInstance.cpp
+++ Source/JavaScriptCore/runtime/DateInstance.cpp
@@ -75,6 +75,10 @@ const GregorianDateTime* DateInstance::calculateGregorianDateTimeUTC(VM& vm) con
     if (!m_data)
         m_data = vm.dateInstanceCache.add(milli);
 
+    if (!m_data) {
+	    // this can happen when finishCreation() was called with time <= 0
+	    return nullptr;
+    }
     if (m_data->m_gregorianDateTimeUTCCachedForMS != milli) {
         msToGregorianDateTime(vm, milli, WTF::UTCTime, m_data->m_cachedGregorianDateTimeUTC);
         m_data->m_gregorianDateTimeUTCCachedForMS = milli;
