--- Source/WebKit/Platform/IPC/Attachment.h.orig	2023-02-20 09:22:20
+++ Source/WebKit/Platform/IPC/Attachment.h	2024-12-23 12:35:57
@@ -26,7 +26,9 @@
 
 #pragma once
 
-#if OS(DARWIN)
+// PlatformGTK activates UNIX_DOMAIN_SOCKETS
+// - which exist on DARWIN - so going with that
+#if PLATFORM(COCOA)
 #include <wtf/MachSendRight.h>
 #endif
 
@@ -38,12 +40,15 @@
 
 // IPC::Attachment is a type representing objects that cannot be transferred as data,
 // rather they are transferred via operating system cross-process communication primitives.
-#if OS(DARWIN)
+#if PLATFORM(COCOA)
 using Attachment = MachSendRight;
 #elif OS(WINDOWS)
 using Attachment = int; // Windows does not need attachments at the moment.
 #elif USE(UNIX_DOMAIN_SOCKETS)
 using Attachment = UnixFileDescriptor;
+#elif OS(DARWIN)
+// not sure if need this
+using Attachment = UnixFileDescriptor;
 #else
 #error Unsupported platform
 #endif
