diff --git Source/WebKit/Platform/IPC/unix/orig.ConnectionUnix.cpp Source/WebKit/Platform/IPC/unix/ConnectionUnix.cpp
index 77af9b2..45226fb 100644
--- Source/WebKit/Platform/IPC/unix/orig.ConnectionUnix.cpp
+++ Source/WebKit/Platform/IPC/unix/ConnectionUnix.cpp
@@ -511,7 +511,7 @@ bool Connection::sendOutputMessage(UnixMessage& outputMessage)
         }
 
         if (m_isConnected)
-            WTFLogAlways("Error sending IPC message: %s", strerror(errno));
+            WTFLogAlways("Error sending IPC message of length %d: %s", message.msg_iovlen, strerror(errno));
         return false;
     }
     return true;
