diff --git Source/WebKit/Platform/IPC/unix/ConnectionUnix.cpp.orig Source/WebKit/Platform/IPC/unix/ConnectionUnix.cpp
index e01364f..6489537 100644
--- Source/WebKit/Platform/IPC/unix/ConnectionUnix.cpp.orig
+++ Source/WebKit/Platform/IPC/unix/ConnectionUnix.cpp
@@ -535,7 +535,7 @@ bool Connection::sendOutputMessage(UnixMessage& outputMessage)
         }
 
         if (m_isConnected)
-            WTFLogAlways("Error sending IPC message: %s", safeStrerror(errno).data());
+            WTFLogAlways("Error sending IPC message of length %d: %s", message.msg_iovlen, safeStrerror(errno).data());
         return false;
     }
 
