diff --git a/tests/Rebase.cpp b/tests/Rebase.cpp
index 83c7491..161e1e4 100644
--- a/tests/Rebase.cpp
+++ b/tests/Rebase.cpp
@@ -49,15 +49,15 @@ private slots:
 
 private:
     QSharedPointer<Repository> repo;
-    Signature sig;
+    Signature *sig;
 
     OId commitIndexToRef(const QString &refSpec);
     void writeToIndex(const QString &path, const QString &text);
 };
 
 TestRebase::TestRebase()
-    : TestBase(),
-      sig("unknown", "unknown")
+    : TestBase()
+    , sig(nullptr)
 {
 }
 
@@ -65,6 +65,7 @@ void TestRebase::init()
 {
     TestBase::init();
     initTestRepo();
+    sig = new Signature("unknown", "unknown");
     repo = QSharedPointer<Repository>(new Repository);
     repo->open(testdir + "/.git");
 }
@@ -73,6 +74,7 @@ void TestRebase::cleanup()
 {
     repo.clear();
     TestBase::cleanup();
+    delete sig;
 }
 
 OId TestRebase::commitIndexToRef(const QString &refSpec)
@@ -80,7 +82,7 @@ OId TestRebase::commitIndexToRef(const QString &refSpec)
     Tree index = repo->lookupTree(repo->index().createTree());
     QList<Commit> parents;
     parents.append(repo->lookupCommit(repo->head().target()));
-    return repo->createCommit(index, parents, sig, sig, "commit", refSpec);
+    return repo->createCommit(index, parents, *sig, *sig, "commit", refSpec);
 }
 
 void TestRebase::writeToIndex(const QString &path, const QString &text)
@@ -114,8 +116,8 @@ void TestRebase::TestRebasingMasterOntoAnotherBranchProducesCorrectTopology()
     Reference refUpstream = repo->lookupRef("refs/remotes/origin/master");
     Rebase rebase = repo->rebase(refMaster, refUpstream, refOnto, RebaseOptions(CheckoutOptions(CheckoutOptions::Safe, CheckoutOptions::RecreateMissing)));
     rebase.next();
-    OId oidRebasedMaster = rebase.commit(sig, sig, QString());
-    rebase.finish(sig);
+    OId oidRebasedMaster = rebase.commit(*sig, *sig, QString());
+    rebase.finish(*sig);
 
     // check results
     RevWalk walk(*repo);
