diff --git src/freac.cpp src/freac.cpp
index 236afbb..b9edad6 100644
--- src/freac.cpp
+++ src/freac.cpp
@@ -128,9 +128,9 @@ freac::freac::freac() : Application(appName)
 	 */
 	Utilities::SetProcessPriority();
 
-	/* Set default comment unless already set.
-	 */
-	config->SetStringValue(Config::CategoryTagsID, Config::TagsDefaultCommentID, config->GetStringValue(Config::CategoryTagsID, Config::TagsDefaultCommentID, String(appLongName).Append(" <").Append(website).Append(">")));
+// 	/* Set default comment unless already set.
+// 	 */
+// 	config->SetStringValue(Config::CategoryTagsID, Config::TagsDefaultCommentID, config->GetStringValue(Config::CategoryTagsID, Config::TagsDefaultCommentID, String(appLongName).Append(" <").Append(website).Append(">")));
 
 	/* Connect version information to BoCA callbacks.
 	 */
