diff --git lib/error.c.orig lib/error.c
index 78335b9..7f842de 100644
--- lib/error.c.orig
+++ lib/error.c
@@ -333,9 +333,9 @@ gss_display_status (OM_uint32 * minor_status,
 	_gss_mech_api_t mech;
 
 	mech = _gss_find_mech (mech_type);
-	return mech->display_status (minor_status, status_value, status_type,
+	return mech ? mech->display_status (minor_status, status_value, status_type,
 				     mech_type, message_context,
-				     status_string);
+				     status_string) : GSS_S_BAD_STATUS;
       }
       break;
 
