diff --git libavcodec/orig.videotoolboxenc.c libavcodec/videotoolboxenc.c
index d1ba083..58dd9f1 100644
--- libavcodec/orig.videotoolboxenc.c
+++ libavcodec/videotoolboxenc.c
@@ -1176,7 +1176,6 @@ static int vtenc_create_encoder(AVCodecContext   *avctx,
             return AVERROR_EXTERNAL;
         }
     }
-#endif
 
     // Dump the init encoder
     {
@@ -1205,6 +1204,7 @@ static int vtenc_create_encoder(AVCodecContext   *avctx,
         if (encoderID != NULL)
             CFRelease(encoderID);
     }
+#endif
 
     if (avctx->flags & AV_CODEC_FLAG_QSCALE && !vtenc_qscale_enabled()) {
         av_log(avctx, AV_LOG_ERROR, "Error: -q:v qscale not available for encoder. Use -b:v bitrate instead.\n");
