diff --git src/core/Entry.cpp src/core/Entry.cpp
index 4ae6b091..b8e5102a 100644
--- src/core/Entry.cpp
+++ src/core/Entry.cpp
@@ -1011,7 +1011,8 @@ QString Entry::resolveMultiplePlaceholdersRecursive(const QString& str, int maxD
     static QRegularExpression placeholderRegEx("(\\{[^\\}]+?\\})", QRegularExpression::CaseInsensitiveOption);
 
     if (maxDepth <= 0) {
-        qWarning("Maximum depth of replacement has been reached. Entry uuid: %s", uuid().toString().toLatin1().data());
+        qWarning("Maximum depth of multiple placeholder replacement of \"%s\" has been reached. Entry uuid: %s",
+            qPrintable(str), uuid().toString().toLatin1().data());
         return str;
     }
 
@@ -1033,7 +1034,8 @@ QString Entry::resolveMultiplePlaceholdersRecursive(const QString& str, int maxD
 QString Entry::resolvePlaceholderRecursive(const QString& placeholder, int maxDepth) const
 {
     if (maxDepth <= 0) {
-        qWarning("Maximum depth of replacement has been reached. Entry uuid: %s", uuid().toString().toLatin1().data());
+        qWarning("Maximum depth of placeholder replacement of \"%s\" has been reached. Entry uuid: %s",
+            qPrintable(placeholder), uuid().toString().toLatin1().data());
         return placeholder;
     }
 
@@ -1174,7 +1176,8 @@ QString Entry::resolveDateTimePlaceholder(Entry::PlaceholderType placeholderType
 QString Entry::resolveReferencePlaceholderRecursive(const QString& placeholder, int maxDepth) const
 {
     if (maxDepth <= 0) {
-        qWarning("Maximum depth of replacement has been reached. Entry uuid: %s", uuid().toString().toLatin1().data());
+        qWarning("Maximum depth of reference placeholder resolution of \"%s\"  has been reached. Entry uuid: %s",
+            qPrintable(placeholder), uuid().toString().toLatin1().data());
         return placeholder;
     }
 
