diff --git cmake/productsets/creativepractical.cmake cmake/productsets/creativepractical.cmake
new file mode 100644
--- /dev/null
+++ cmake/productsets/creativepractical.cmake
@@ -0,0 +1,15 @@
+#defines the set of products commonly wanted by Pragmatic Creationists ;)
+
+calligra_define_productset(CREATIVEPRACTICAL "Creative & Practical suite"
+    OPTIONAL
+        # apps
+        KARBON
+        KRITA STAGE
+	   BRAINDUMP
+	   KEXI
+        # extras
+        APP_CONVERTER
+        FILEMANAGER
+        # docs
+        DOC
+)
