diff --git krita/plugins/tools/defaulttools/tests/CMakeLists.txt b/krita/plugins/tools/defaulttools/tests/CMakeLists.txt
index 229f51e..29054ac 100644
--- krita/plugins/tools/defaulttools/tests/CMakeLists.txt
+++ krita/plugins/tools/defaulttools/tests/CMakeLists.txt
@@ -5,6 +5,6 @@ macro_add_unittest_definitions()
 
 ########### next target ###############
 
-set(move_stroke_test_SRCS move_stroke_test.cpp ${CMAKE_SOURCE_DIR}/krita/sdk/tests/stroke_testing_utils.cpp)
-kde4_add_unit_test(MoveStrokeTest TESTNAME krita-ui-MoveStrokeTest  ${move_stroke_test_SRCS})
-target_link_libraries(MoveStrokeTest  ${KDE4_KDEUI_LIBS}  kritadefaulttools kritaui kritaimage ${QT_QTTEST_LIBRARY})
+# set(move_stroke_test_SRCS move_stroke_test.cpp ${CMAKE_SOURCE_DIR}/krita/sdk/tests/stroke_testing_utils.cpp)
+# kde4_add_unit_test(MoveStrokeTest TESTNAME krita-ui-MoveStrokeTest  ${move_stroke_test_SRCS})
+# target_link_libraries(MoveStrokeTest  ${KDE4_KDEUI_LIBS}  kritadefaulttools kritaui kritaimage ${QT_QTTEST_LIBRARY})
diff --git krita/plugins/tools/tool_transform2/tests/CMakeLists.txt b/krita/plugins/tools/tool_transform2/tests/CMakeLists.txt
index 9b04aec..a96c8b8 100644
--- krita/plugins/tools/tool_transform2/tests/CMakeLists.txt
+++ krita/plugins/tools/tool_transform2/tests/CMakeLists.txt
@@ -5,6 +5,6 @@ macro_add_unittest_definitions()
 
 ########### next target ###############
 
-set(test_save_load_transform_args_SRCS test_save_load_transform_args.cpp )
-kde4_add_unit_test(TestSaveLoadTransformArgs TESTNAME krita-ui-TestSaveLoadTransformArgs  ${test_save_load_transform_args_SRCS})
-target_link_libraries(TestSaveLoadTransformArgs  ${KDE4_KDEUI_LIBS}  kritatooltransform kritaui kritaimage ${QT_QTTEST_LIBRARY})
+# set(test_save_load_transform_args_SRCS test_save_load_transform_args.cpp )
+# kde4_add_unit_test(TestSaveLoadTransformArgs TESTNAME krita-ui-TestSaveLoadTransformArgs  ${test_save_load_transform_args_SRCS})
+# target_link_libraries(TestSaveLoadTransformArgs  ${KDE4_KDEUI_LIBS}  kritatooltransform kritaui kritaimage ${QT_QTTEST_LIBRARY})
diff --git krita/ui/tests/CMakeLists.txt b/krita/ui/tests/CMakeLists.txt
index 98620ad..0f770f8 100644
--- krita/ui/tests/CMakeLists.txt
+++ krita/ui/tests/CMakeLists.txt
@@ -100,9 +100,9 @@ target_link_libraries(FreehandStrokeTest  ${KDE4_KDEUI_LIBS}  kritaui kritaimage
 
 ########### next target ###############
 
-set(fill_processing_visitor_test_SRCS fill_processing_visitor_test.cpp ${CMAKE_SOURCE_DIR}/krita/sdk/tests/stroke_testing_utils.cpp)
-kde4_add_unit_test(FillProcessingVisitorTest TESTNAME krita-ui-FillProcessingVisitorTest  ${fill_processing_visitor_test_SRCS})
-target_link_libraries(FillProcessingVisitorTest  ${KDE4_KDEUI_LIBS}  kritadefaulttools kritaui kritaimage ${QT_QTTEST_LIBRARY})
+# set(fill_processing_visitor_test_SRCS fill_processing_visitor_test.cpp ${CMAKE_SOURCE_DIR}/krita/sdk/tests/stroke_testing_utils.cpp)
+# kde4_add_unit_test(FillProcessingVisitorTest TESTNAME krita-ui-FillProcessingVisitorTest  ${fill_processing_visitor_test_SRCS})
+# target_link_libraries(FillProcessingVisitorTest  ${KDE4_KDEUI_LIBS}  kritadefaulttools kritaui kritaimage ${QT_QTTEST_LIBRARY})
 
 
 ########### next target ###############
