diff --git flow/part/FlowView.cpp flow/part/FlowView.cpp
index 5ce755f..eb6a655 100644
--- flow/part/FlowView.cpp
+++ flow/part/FlowView.cpp
@@ -58,6 +58,7 @@ void FlowView::initializeGUI()
 void FlowView::initializeActions()
 {
     actionCollection()->action("configure")->setText(i18n("Configure Flow..."));
+    actionCollection()->action("configure")->setMenuRole(QAction::PreferencesRole);
 }
 
 void FlowView::updateGui()
