diff --git gemini/main.cpp gemini/main.cpp
index 701ec21..43ad8da 100644
--- gemini/main.cpp
+++ gemini/main.cpp
@@ -144,11 +144,13 @@ int main( int argc, char** argv )
     QIcon::setThemeName("oxygen");
 #endif
 
+#ifndef Q_OS_MAC
     if (qgetenv("KDE_FULL_SESSION").isEmpty()) {
         // There are two themes that work for Krita, oxygen and plastique. Try to set plastique first, then oxygen
         qobject_cast<QApplication*>(QApplication::instance())->setStyle("Plastique");
         qobject_cast<QApplication*>(QApplication::instance())->setStyle("Oxygen");
     }
+#endif
 
     // then create the pixmap from an xpm: we cannot get the
     // location of our datadir before we've started our components,
diff --git krita/gemini/main.cpp krita/gemini/main.cpp
index 19dc5c8..13e1b4a 100644
--- krita/gemini/main.cpp
+++ krita/gemini/main.cpp
@@ -139,11 +139,13 @@ int main( int argc, char** argv )
     app.setEventFilter(&KisTabletSupportX11::eventFilter);
 #endif
 	
+#ifndef Q_OS_MACX
 	if (qgetenv("KDE_FULL_SESSION").isEmpty()) {
         // There are two themes that work for Krita, oxygen and plastique. Try to set plastique first, then oxygen
         qobject_cast<QApplication*>(QApplication::instance())->setStyle("Plastique");
 		qobject_cast<QApplication*>(QApplication::instance())->setStyle("Oxygen");
     }
+#endif
 
     // Prepare to show window fullscreen if required
     bool showFullscreen = false;
diff --git krita/ui/KisApplication.cpp krita/ui/KisApplication.cpp
index 9e8dae1..80fae75 100644
--- krita/ui/KisApplication.cpp
+++ krita/ui/KisApplication.cpp
@@ -147,14 +147,14 @@ KisApplication::KisApplication(const QString &key)
     KoGlobal::initialize();
 
 #ifdef Q_OS_MACX
-    if ( QSysInfo::MacintoshVersion > QSysInfo::MV_10_8 )
-    {
-        // fix Mac OS X 10.9 (mavericks) font issue
-        // https://bugreports.qt-project.org/browse/QTBUG-32789
-        QFont::insertSubstitution(".Lucida Grande UI", "Lucida Grande");
-    }
+//     if ( QSysInfo::MacintoshVersion > QSysInfo::MV_10_8 )
+//     {
+//         // fix Mac OS X 10.9 (mavericks) font issue
+//         // https://bugreports.qt-project.org/browse/QTBUG-32789
+//         QFont::insertSubstitution(".Lucida Grande UI", "Lucida Grande");
+//     }
     setAttribute(Qt::AA_DontShowIconsInMenus, true);
-#endif
+#else
 
 
     if (applicationName() == "krita" && qgetenv("KDE_FULL_SESSION").isEmpty()) {
@@ -163,7 +163,7 @@ KisApplication::KisApplication(const QString &key)
         setStyle("Breeze");
         setStyle("Oxygen");
     }
-
+#endif
 }
 
 #if defined(Q_OS_WIN) && defined(ENV32BIT)
diff --git krita/ui/thememanager.cpp krita/ui/thememanager.cpp
index fd525a4..74289e5 100644
--- krita/ui/thememanager.cpp
+++ krita/ui/thememanager.cpp
@@ -186,7 +186,7 @@ void ThemeManager::slotChangePalette()
 #endif
     } else {
 #ifdef __APPLE__
-        qApp->setStyle("Plastique");
+//         qApp->setStyle("Plastique");
         qApp->style()->polish(kapp);
 #endif
     }
diff --git libs/main/KoApplication.cpp libs/main/KoApplication.cpp
index de1c97f..aeb6566 100644
--- libs/main/KoApplication.cpp
+++ libs/main/KoApplication.cpp
@@ -146,21 +146,22 @@ KoApplication::KoApplication(const QByteArray &nativeMimeType)
 #endif
 
 #ifdef Q_OS_MACX
-    if ( QSysInfo::MacintoshVersion > QSysInfo::MV_10_8 )
-    {
-        // fix Mac OS X 10.9 (mavericks) font issue
-        // https://bugreports.qt-project.org/browse/QTBUG-32789
-        QFont::insertSubstitution(".Lucida Grande UI", "Lucida Grande");
-    }
+//     if ( QSysInfo::MacintoshVersion > QSysInfo::MV_10_8 )
+//     {
+//         // fix Mac OS X 10.9 (mavericks) font issue
+//         // https://bugreports.qt-project.org/browse/QTBUG-32789
+//         QFont::insertSubstitution(".Lucida Grande UI", "Lucida Grande");
+//     }
 
     setAttribute(Qt::AA_DontShowIconsInMenus, true);
-#endif
+#else
 
     if ((applicationName() == "krita" || applicationName() == "calligragemini") && qgetenv("KDE_FULL_SESSION").isEmpty()) {
         // There are two themes that work for Krita, oxygen and plastique. Try to set plastique first, then oxygen
         setStyle("Plastique");
         setStyle("Oxygen");
     }
+#endif
 }
 
 // This gets called before entering KApplication::KApplication
