--- libs/pigment/CMakeLists.txt
+++ libs/pigment/CMakeLists.txt	2024-08-31 00:54:24.000000000 +0800
@@ -26,7 +26,7 @@
     ko_compile_for_all_implementations_no_scalar(__per_arch_factory_objs compositeops/KoOptimizedCompositeOpFactoryPerArch.cpp)
 
     message("Following objects are generated from the per-arch lib")
-    message(${__per_arch_factory_objs})
+    message("${__per_arch_factory_objs}")
 endif()
 
 add_subdirectory(tests)
