diff --git krita/image/kis_image_config.cpp krita/image/kis_image_config.cpp
index 846100e..7f751dd 100644
--- krita/image/kis_image_config.cpp
+++ krita/image/kis_image_config.cpp
@@ -28,6 +28,9 @@
 #include <QThread>
 #include <QApplication>
 
+#ifdef Q_OS_MAC
+#include <errno.h>
+#endif
 
 KisImageConfig::KisImageConfig()
     : m_config(KGlobal::config()->group(""))
@@ -267,16 +270,21 @@ int KisImageConfig::totalRAM()
 #   if defined ENV32BIT
     totalMemory = qMin(totalMemory, 2000);
 #   endif
-#elif defined Q_OS_MAC64
+#elif defined Q_OS_MAC
     int mib[2] = { CTL_HW, HW_MEMSIZE };
     u_int namelen = sizeof(mib) / sizeof(mib[0]);
     uint64_t size;
     size_t len = sizeof(size);
 
-    if (sysctl(mib, namelen, &size, &len, NULL, 0) > 0) {
-        totalMemory = size;
+    errno = 0;
+    if (sysctl(mib, namelen, &size, &len, NULL, 0) >= 0) {
+        totalMemory = size >> 20;
+        kDebug() << "sysctl(\"hw.memsize\") returned size=" << size << " =>" << totalMemory << "MiB";
         error = 0;
     }
+    else {
+        kDebug() << "sysctl(\"hw.memsize\") raised error" << strerror(errno);
+    }
 #endif
 
     if(error) {
