diff --git flow/part/CMakeLists.txt flow/part/CMakeLists.txt
index 58882f1..90ac5be 100644
--- flow/part/CMakeLists.txt
+++ flow/part/CMakeLists.txt
@@ -54,3 +54,6 @@ install( FILES flow.appdata.xml DESTINATION ${SHARE_INSTALL_PREFIX}/appdata/)
 ### GUI files ###
 install( FILES flow.rc DESTINATION ${DATA_INSTALL_DIR}/flow)
 install( FILES flowrc DESTINATION ${CONFIG_INSTALL_DIR} )
+if(APPLE)
+   install( FILES ${CMAKE_CURRENT_BINARY_DIR}/flow_KDEINIT_SRCS.icns DESTINATION ${BUNDLE_INSTALL_DIR}/calligraflow.app/Contents/Resources)
+endif()
diff --git gemini/CMakeLists.txt gemini/CMakeLists.txt
index 85123fa..2344778 100644
--- gemini/CMakeLists.txt
+++ gemini/CMakeLists.txt
@@ -56,7 +56,7 @@ set(thumbnailhelper_SRCS
 if (WIN32)
     set(calligragemini_SRCS ${calligragemini_SRCS} calligrageminiwin.rc)
 else ()
-    kde4_add_app_icon(calligragemini_SRCS "${KDE4_ICON_DIR}/oxygen/*/apps/calligragemini.png")
+    kde4_add_app_icon(calligragemini_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/pics/hi*-app-calligragemini.png")
 endif ()
 
 # Thumbnail helper application
@@ -94,6 +94,9 @@ install(FILES     calligragemini.qml                   DESTINATION ${DATA_INSTAL
 #install(FILES     calligragemini.rc                    DESTINATION ${DATA_INSTALL_DIR}/calligragemini)
 install(DIRECTORY qml                                       DESTINATION ${DATA_INSTALL_DIR}/calligragemini)
 install(DIRECTORY themes                                    DESTINATION ${DATA_INSTALL_DIR}/calligragemini)
+if(APPLE)
+   install( FILES ${CMAKE_CURRENT_BINARY_DIR}/calligragemini_SRCS.icns DESTINATION ${BUNDLE_INSTALL_DIR}/calligragemini.app/Contents/Resources)
+endif()
 
 add_subdirectory(pics)
 add_subdirectory(cloud)
diff --git karbon/CMakeLists.txt karbon/CMakeLists.txt
index b574779..28d78ff 100644
--- karbon/CMakeLists.txt
+++ karbon/CMakeLists.txt
@@ -50,10 +50,11 @@ set(karbon_KDEINIT_SRCS main.cpp )
 kde4_add_app_icon( karbon_KDEINIT_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/pics/app/hi*-app-calligrakarbon.png" )
 kde4_add_kdeinit_executable( karbon ${karbon_KDEINIT_SRCS})
 
-if (Q_WS_MAC)
+if (APPLE)
    set_target_properties(karbon PROPERTIES MACOSX_BUNDLE_INFO_PLIST ${CMAKE_CURRENT_SOURCE_DIR}/Info.plist.template)
    set_target_properties(karbon PROPERTIES MACOSX_BUNDLE_GUI_IDENTIFIER "org.calligra.karbon")
    set_target_properties(karbon PROPERTIES MACOSX_BUNDLE_BUNDLE_NAME "Karbon 2")
+   install( FILES ${CMAKE_CURRENT_BINARY_DIR}/karbon_KDEINIT_SRCS.icns DESTINATION ${BUNDLE_INSTALL_DIR}/karbon.app/Contents/Resources)
 endif ()
 
 target_link_libraries(kdeinit_karbon karbonui)
diff --git plan/CMakeLists.txt plan/CMakeLists.txt
index ad39f57..2e0eab0 100644
--- plan/CMakeLists.txt
+++ plan/CMakeLists.txt
@@ -126,10 +126,11 @@ kde4_add_app_icon(calligraplan_KDEINIT_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/pics/hi
 
 kde4_add_kdeinit_executable( calligraplan ${calligraplan_KDEINIT_SRCS})
 
-if (Q_WS_MAC)
+if (APPLE)
    set_target_properties(calligraplan PROPERTIES MACOSX_BUNDLE_INFO_PLIST ${CMAKE_CURRENT_SOURCE_DIR}/Info.plist.template)
    set_target_properties(calligraplan PROPERTIES MACOSX_BUNDLE_GUI_IDENTIFIER "org.calligra.plan")
    set_target_properties(calligraplan PROPERTIES MACOSX_BUNDLE_BUNDLE_NAME "Plan")
+   install( FILES ${CMAKE_CURRENT_BINARY_DIR}/calligraplan_KDEINIT_SRCS.icns DESTINATION ${BUNDLE_INSTALL_DIR}/calligraplan.app/Contents/Resources)
 endif ()
 
 target_link_libraries(kdeinit_calligraplan komain)
diff --git sheets/CMakeLists.txt sheets/CMakeLists.txt
index b0cc134..4b775f1 100644
--- sheets/CMakeLists.txt
+++ sheets/CMakeLists.txt
@@ -414,10 +414,11 @@ kde4_add_app_icon(calligrasheets_KDEINIT_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/data/
 
 kde4_add_kdeinit_executable( calligrasheets ${calligrasheets_KDEINIT_SRCS})
 
-if (Q_WS_MAC)
+if (APPLE)
 set_target_properties(calligrasheets PROPERTIES MACOSX_BUNDLE_INFO_PLIST ${CMAKE_CURRENT_SOURCE_DIR}/Info.plist.template)
 set_target_properties(calligrasheets PROPERTIES MACOSX_BUNDLE_GUI_IDENTIFIER "org.calligra.sheets")
 set_target_properties(calligrasheets PROPERTIES MACOSX_BUNDLE_BUNDLE_NAME "Calligra Sheets 2")
+install( FILES ${CMAKE_CURRENT_BINARY_DIR}/calligrasheets_KDEINIT_SRCS.icns DESTINATION ${BUNDLE_INSTALL_DIR}/calligrasheets.app/Contents/Resources)
 endif ()
 
 target_link_libraries(kdeinit_calligrasheets komain )
diff --git stage/app/CMakeLists.txt stage/app/CMakeLists.txt
index 079bece..5bad1e3 100644
--- stage/app/CMakeLists.txt
+++ stage/app/CMakeLists.txt
@@ -12,10 +12,11 @@ kde4_add_app_icon(stage_KDEINIT_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/../pics/hi*-ap
 
 kde4_add_kdeinit_executable(calligrastage ${stage_KDEINIT_SRCS})
 
-if (Q_WS_MAC)
+if (APPLE)
    set_target_properties(calligrastage PROPERTIES MACOSX_BUNDLE_INFO_PLIST ${CMAKE_CURRENT_SOURCE_DIR}/Info.plist.template)
    set_target_properties(calligrastage PROPERTIES MACOSX_BUNDLE_GUI_IDENTIFIER "org.calligra.stage")
    set_target_properties(calligrastage PROPERTIES MACOSX_BUNDLE_BUNDLE_NAME "Calligra Stage 2")
+   install( FILES ${CMAKE_CURRENT_BINARY_DIR}/stage_KDEINIT_SRCS.icns DESTINATION ${BUNDLE_INSTALL_DIR}/calligrastage.app/Contents/Resources)
 endif ()
 
 target_link_libraries(kdeinit_calligrastage komain )
diff --git words/app/CMakeLists.txt words/app/CMakeLists.txt
index 1e73971..070d289 100644
--- words/app/CMakeLists.txt
+++ words/app/CMakeLists.txt
@@ -10,10 +10,11 @@ kde4_add_app_icon(words_KDEINIT_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/../pics/hi*-ap
 
 kde4_add_kdeinit_executable( calligrawords ${words_KDEINIT_SRCS})
 
-if (Q_WS_MAC)
+if (APPLE)
    set_target_properties(calligrawords PROPERTIES MACOSX_BUNDLE_INFO_PLIST ${CMAKE_CURRENT_SOURCE_DIR}/Info.plist.template)
    set_target_properties(calligrawords PROPERTIES MACOSX_BUNDLE_GUI_IDENTIFIER "org.calligra.words")
    set_target_properties(calligrawords PROPERTIES MACOSX_BUNDLE_BUNDLE_NAME "Words")
+   install( FILES ${CMAKE_CURRENT_BINARY_DIR}/words_KDEINIT_SRCS.icns DESTINATION ${BUNDLE_INSTALL_DIR}/calligrawords.app/Contents/Resources)
 endif ()
 
 target_link_libraries(kdeinit_calligrawords  ${KDE4_KDEUI_LIBS} komain )
diff --git words/part/CMakeLists.txt words/part/CMakeLists.txt
index 9143176..098e2f9 100644
--- words/part/CMakeLists.txt
+++ words/part/CMakeLists.txt
@@ -188,14 +188,17 @@ install(TARGETS authorpart DESTINATION ${PLUGIN_INSTALL_DIR})
 
 set(author_KDEINIT_SRCS author/main.cpp )
 
-#kde4_add_app_icon(author_KDEINIT_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/../pics/author/hi*-app-calligraauthor.png")
+if(APPLE)
+   kde4_add_app_icon(author_KDEINIT_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/../pics/hi*-app-calligraauthor.png")
+endif()
 
 kde4_add_kdeinit_executable( calligraauthor ${author_KDEINIT_SRCS})
 
-if (Q_WS_MAC)
+if (APPLE)
    set_target_properties(calligraauthor PROPERTIES MACOSX_BUNDLE_INFO_PLIST ${CMAKE_CURRENT_SOURCE_DIR}/../app/Info.plist.template)
    set_target_properties(calligraauthor PROPERTIES MACOSX_BUNDLE_GUI_IDENTIFIER "org.calligra.author")
    set_target_properties(calligraauthor PROPERTIES MACOSX_BUNDLE_BUNDLE_NAME "Author")
+   install( FILES ${CMAKE_CURRENT_BINARY_DIR}/author_KDEINIT_SRCS.icns DESTINATION ${BUNDLE_INSTALL_DIR}/calligraauthor.app/Contents/Resources)
 endif ()
 
 target_link_libraries(kdeinit_calligraauthor  ${KDE4_KDEUI_LIBS} komain )
