diff --git orig.ftpfs.c ftpfs.c
index 0346354..a24ba25 100644
--- orig.ftpfs.c
+++ ftpfs.c
@@ -38,6 +38,9 @@
 
 struct ftpfs ftpfs;
 static char error_buf[CURL_ERROR_SIZE];
+char* get_curl_error_buf() {
+  return error_buf;
+}
 
 struct buffer {
   uint8_t* p;
diff --git orig.path_utils.c path_utils.c
index db3d7e4..eb08eaf 100644
--- orig.path_utils.c
+++ path_utils.c
@@ -14,6 +14,18 @@
 #include <stdlib.h>
 #include <glib.h>
 
+char* get_encoded_path(const char* path) {
+  const char* encoded = curl_easy_escape(ftpfs.connection, path, 0);
+  if (!encoded) {
+    extern char* get_curl_error_buf();
+    fprintf(stderr, "curl failed to encode \"%s\" (%s)\n", path, get_curl_error_buf());
+    encoded = path;
+  } else {
+    DEBUG(2, "Using encoded path \"%s\" instead of \"%s\"\n", encoded, path);
+  }
+  return encoded;
+}
+
 char* get_file_name(const char* path) {
   const char* filename = strrchr(path, '/');
   if (filename == NULL) filename = path;
@@ -23,6 +35,12 @@ char* get_file_name(const char* path) {
   if (ftpfs.codepage) {
     convert_charsets(ftpfs.iocharset, ftpfs.codepage, &ret);
   }
+//   const char* encoded = get_encoded_path(ret);
+//   if (encoded != ret) {
+//     free(ret);
+//     ret = strdup(encoded);
+//     curl_free(encoded);
+//   }
   
   return ret;
 }
@@ -30,17 +48,28 @@ char* get_file_name(const char* path) {
 char* get_full_path(const char* path) {
   char* ret;
   char* converted_path = NULL;
+  int free_path = 0;
   
   ++path;
 
   if (ftpfs.codepage && strlen(path)) {
     converted_path = strdup(path);
     convert_charsets(ftpfs.iocharset, ftpfs.codepage, &converted_path);
-    path = converted_path;
+    path = get_encoded_path(converted_path);
+    free_path = 1;
+  } else {
+    const char *encoded = get_encoded_path(path);
+    if (encoded != path) {
+	 path = encoded;
+	 free_path = 1;
+    }
   }
 
   ret = g_strdup_printf("%s%s", ftpfs.host, path);
 
+  if (free_path) {
+    curl_free(path);
+  }
   free(converted_path);
 
   return ret;
@@ -49,17 +78,28 @@ char* get_full_path(const char* path) {
 char* get_fulldir_path(const char* path) {
   char* ret;
   char* converted_path = NULL;
+  int free_path = 0;
 
   ++path;
 
   if (ftpfs.codepage && strlen(path)) {
     converted_path = strdup(path);
     convert_charsets(ftpfs.iocharset, ftpfs.codepage, &converted_path);
-    path = converted_path;
+    path = get_encoded_path(converted_path);
+    free_path = 1;
+  } else {
+    const char *encoded = get_encoded_path(path);
+    if (encoded != path) {
+	 path = encoded;
+	 free_path = 1;
+    }
   }
 
   ret = g_strdup_printf("%s%s%s", ftpfs.host, path, strlen(path) ? "/" : "");
 
+  if (free_path) {
+    curl_free(path);
+  }
   free(converted_path);
 
   return ret;
@@ -69,6 +109,7 @@ char* get_dir_path(const char* path) {
   char* ret;
   char* converted_path = NULL;
   const char *lastdir;
+  int free_path = 0;
 
   ++path;
   
@@ -78,8 +119,16 @@ char* get_dir_path(const char* path) {
   if (ftpfs.codepage && (lastdir - path > 0)) {
     converted_path = g_strndup(path, lastdir - path);
     convert_charsets(ftpfs.iocharset, ftpfs.codepage, &converted_path);
-    path = converted_path;
+    path = get_encoded_path(converted_path);
     lastdir = path + strlen(path);
+    free_path = 1;
+  } else {
+    const char *encoded = get_encoded_path(path);
+    if (encoded != path) {
+	 path = encoded;
+      lastdir = path + strlen(path);
+	 free_path = 1;
+    }
   }
 
   ret = g_strdup_printf("%s%.*s%s",
@@ -88,6 +137,9 @@ char* get_dir_path(const char* path) {
                         path,
                         lastdir - path ? "/" : "");
 
+  if (free_path) {
+    curl_free(path);
+  }
   free(converted_path);
 
   return ret;
