diff --git a/kdeui/util/kcrash.cpp b/kdeui/util/kcrash.cpp
index 45eb46b..9bb9f44 100644
--- a/kdeui/util/kcrash.cpp
+++ b/kdeui/util/kcrash.cpp
@@ -313,6 +313,10 @@ KCrash::defaultCrashHandler (int sig)
         crashRecursionCounter++;
     }
 
+#ifndef Q_OS_MAC
+    // On Apple OS X, closing all FDs now will cause a second (SIGILL) crash,
+    // ending with "Unable to start Dr. Konqi". This is because the libdispatch
+    // library, which can manage multi-threading, has some FDs of its own.
 #if !defined(Q_OS_WIN)
     if (!(s_flags & KeepFDs))
         closeAllFDs();
@@ -321,6 +325,7 @@ KCrash::defaultCrashHandler (int sig)
         close(ConnectionNumber(QX11Info::display()));
 # endif
 #endif
+#endif
 
     if (crashRecursionCounter < 3)
     {
