diff --git a/src/app/exercisecontroller.cpp b/src/app/exercisecontroller.cpp
index e3d8ea9..7aea057 100644
--- a/src/app/exercisecontroller.cpp
+++ b/src/app/exercisecontroller.cpp
@@ -120,7 +120,7 @@ bool ExerciseController::mergeJsonFiles(const QString directoryName, QJsonObject
 #if defined(Q_OS_ANDROID)
     QStringList jsonDirs;
     jsonDirs += "/data/data/org.kde.minuet/qt-reserved-files/share/minuet/" + directoryName;
-#elif defined(Q_OS_LINUX)
+#elif defined(Q_OS_LINUX) || defined(Q_OS_MACOS)
     QStringList jsonDirs = QStandardPaths::locateAll(QStandardPaths::AppDataLocation, directoryName, QStandardPaths::LocateDirectory);
 #elif defined(Q_OS_WIN)
     QStringList jsonDirs = QStandardPaths::locateAll(QStandardPaths::AppDataLocation, QStringLiteral("minuet/") + directoryName, QStandardPaths::LocateDirectory);
