diff --git a/src/plugins/fluidsynthsoundcontroller/fluidsynthsoundcontroller.cpp b/src/plugins/fluidsynthsoundcontroller/fluidsynthsoundcontroller.cpp
index 8be38b8..13794be 100644
--- a/src/plugins/fluidsynthsoundcontroller/fluidsynthsoundcontroller.cpp
+++ b/src/plugins/fluidsynthsoundcontroller/fluidsynthsoundcontroller.cpp
@@ -46,12 +46,13 @@ FluidSynthSoundController::FluidSynthSoundController(QObject *parent)
 
     fluid_synth_cc(m_synth, 1, 100, 0);
 
-#ifdef Q_OS_LINUX
-    int fluid_res = fluid_synth_sfload(m_synth, QStandardPaths::locate(QStandardPaths::AppDataLocation, QStringLiteral("soundfonts/GeneralUser-v1.47.sf2")).toLatin1(), 1);
-#endif
-#ifdef Q_OS_WIN
-    int fluid_res = fluid_synth_sfload(m_synth, QStandardPaths::locate(QStandardPaths::AppDataLocation, QStringLiteral("minuet/soundfonts/GeneralUser-v1.47.sf2")).toLatin1(), 1);
-#endif
+// #ifdef Q_OS_LINUX
+//     int fluid_res = fluid_synth_sfload(m_synth, QStandardPaths::locate(QStandardPaths::AppDataLocation, QStringLiteral("soundfonts/GeneralUser-v1.47.sf2")).toLatin1(), 1);
+// #endif
+// #ifdef Q_OS_WIN
+//     int fluid_res = fluid_synth_sfload(m_synth, QStandardPaths::locate(QStandardPaths::AppDataLocation, QStringLiteral("minuet/soundfonts/GeneralUser-v1.47.sf2")).toLatin1(), 1);
+// #endif
+    int fluid_res = fluid_synth_sfload(m_synth, "@PREFIX@/share/sounds/sf2/GeneralUser_GS_v1.471.sf2", 1);
     if (fluid_res == FLUID_FAILED)
         qDebug() << "Error when loading soundfont!";
 

