diff --git a/src/app/main.cpp b/src/app/main.cpp
index 4ecd622..9cc7fa1 100644
--- a/src/app/main.cpp
+++ b/src/app/main.cpp
@@ -51,7 +51,7 @@ int main(int argc, char **argv)
 
     aboutData.addAuthor(i18n("Sandro S. Andrade"),i18n("Developer"), QStringLiteral("sandroandrade@kde.org"));
     aboutData.addAuthor(i18n("Alessandro Longo"),i18n("Minuet Icon Designer"), QStringLiteral("alessandro.longo@kdemail.net"));
-    application.setWindowIcon(QIcon::fromTheme(QStringLiteral("minuet")));
+    application.setWindowIcon(QIcon::fromTheme(QStringLiteral("minuet"), application.windowIcon()));
     QCommandLineParser parser;
     parser.addHelpOption();
     parser.addVersionOption();
