diff --git a/src/app/CMakeLists.txt b/src/app/CMakeLists.txt
index 52f5f2b..87e8441 100644
--- a/src/app/CMakeLists.txt
+++ b/src/app/CMakeLists.txt
@@ -36,7 +36,7 @@ IF(NOT ${CMAKE_SYSTEM_NAME} MATCHES "Android")
         ${CMAKE_CURRENT_SOURCE_DIR}/icons/22-actions-minuet-rhythms.svg
     )
 
-    ecm_add_app_icon(minuet_SRCS ICONS ${minuet_ICONS_PNG})
+    ecm_add_app_icon(minuet_SRCS ICONS ${CMAKE_CURRENT_SOURCE_DIR}/icons/sc-apps-minuet.svgz)
     ecm_install_icons(ICONS ${minuet_ICONS_PNG} ${minuet_ICONS_SVG} DESTINATION ${ICON_INSTALL_DIR} THEME hicolor)
 ENDIF(NOT ${CMAKE_SYSTEM_NAME} MATCHES "Android")
 
@@ -69,5 +69,5 @@ ENDIF(NOT ${CMAKE_SYSTEM_NAME} MATCHES "Android")
 IF(NOT ${CMAKE_SYSTEM_NAME} MATCHES "Android")
     install(TARGETS minuet ${INSTALL_TARGETS_DEFAULT_ARGS})
     install(PROGRAMS org.kde.minuet.desktop DESTINATION ${XDG_APPS_INSTALL_DIR})
-    install(TARGETS minuet RUNTIME DESTINATION bin)
+#     install(TARGETS minuet RUNTIME DESTINATION bin)
 ENDIF(NOT ${CMAKE_SYSTEM_NAME} MATCHES "Android")
diff --git a/src/app/main.cpp b/src/app/main.cpp
index 57ea415..06f4699 100644
--- a/src/app/main.cpp
+++ b/src/app/main.cpp
@@ -54,7 +54,9 @@ int main(int argc, char *argv[])
     aboutData.addAuthor("Alessandro Longo", i18n("Minuet Icon Designer"), QStringLiteral("alessandro.longo@kdemail.net"));
 #endif
 
+#ifndef Q_OS_MACOS
     application.setWindowIcon(QIcon(QStringLiteral(":/minuet.png")));
+#endif
 
     QCommandLineParser parser;
     parser.addHelpOption();
