diff --git a/src/app/qml/Main.qml b/src/app/qml/Main.qml
index 2b95787..8fc1d08 100644
--- a/src/app/qml/Main.qml
+++ b/src/app/qml/Main.qml
@@ -29,7 +29,7 @@ import QtQuick.Controls.Material 2.0
 ApplicationWindow {
     id: applicationWindow
     visible: true
-    width: Screen.desktopAvailableWidth; height: Screen.desktopAvailableHeight
+    width: Screen.width; height: Screen.height
 
     property string titleText: "Minuet"
 

