diff --git a/src/plugins/CMakeLists.txt b/src/plugins/CMakeLists.txt
index e0d9918..dbd4b6d 100644
--- a/src/plugins/CMakeLists.txt
+++ b/src/plugins/CMakeLists.txt
@@ -5,12 +5,12 @@ else()
     message(FATAL_ERROR "Program pkg-config not found")
 endif()
 
-pkg_check_modules(ALSA REQUIRED alsa>=1.0.0)
+pkg_check_modules(ALSA alsa>=1.0.0)
 if(ALSA_FOUND)
     link_directories(${ALSA_LIBDIR})
     include_directories(${ALSA_INCLUDEDIR})
 else()
-    message(FATAL_ERROR "Please install ALSA library and headers (libasound2)")
+    message(WARNING "Please install ALSA library and headers (libasound2)")
 endif()
 
 pkg_check_modules(DRUMSTICK_ALSA drumstick-alsa>=1.0.1)
