--- src/kcmultidialog.cpp.orig	2018-01-24 15:45:54.065268836 +0100
+++ src/kcmultidialog.cpp	2018-01-24 15:48:15.989306651 +0100
@@ -377,7 +377,7 @@
     const QUrl docUrl = QUrl(QStringLiteral("help:/")).resolved(QUrl(docPath)); // same code as in KHelpClient::invokeHelp
     const QString docUrlScheme = docUrl.scheme();
     if (docUrlScheme == QLatin1String("help") || docUrlScheme == QLatin1String("man") || docUrlScheme == QLatin1String("info")) {
-        QProcess::startDetached(QStringLiteral("khelpcenter"), QStringList() << docUrl.toString());
+        QProcess::startDetached(QStringLiteral("khelpcentre5"), QStringList() << docUrl.toString());
     } else {
         QDesktopServices::openUrl(docUrl);
     }
