diff --git a/build.rs b/build.rs
index 7892f5e..509ad2d 100644
--- a/build.rs
+++ b/build.rs
@@ -13,9 +13,9 @@ enum BuildError {
 fn main() -> Result<(), BuildError> {
     let bindings_rs = "src/lib.rs";
     let mut autocxx = autocxx_build::Builder::new(bindings_rs, &[PathBuf::from("third-party/cdm")])
-        .extra_clang_args(&["-std=c++20"])
+        .extra_clang_args(&["-std=c++14"])
         .build()?;
-    autocxx.std("c++20").compile("cdm-api");
+    autocxx.std("c++14").compile("cdm-api");
     println!("cargo:rerun-if-changed={}", bindings_rs);
 
     let no_paths: [&str; 0] = [];
