--- src/lib/orig.CMakeLists.txt	2015-11-08 13:14:11.000000000 +0100
+++ src/lib/CMakeLists.txt	2015-11-27 00:04:55.000000000 +0100
@@ -105,6 +105,7 @@
   PUBLIC
     Qt5::Core
   PRIVATE
+    Qt5::QspXDG
     ${kcoreaddons_OPTIONAL_LIBS}
 )
 if(NOT WIN32)
--- src/desktoptojson/orig.CMakeLists.txt	2016-02-06 14:39:47.000000000 +0100
+++ src/desktoptojson/CMakeLists.txt	2016-02-15 14:03:21.000000000 +0100
@@ -9,7 +9,7 @@
 # Mark it as non-gui so we won't create an app bundle on Mac OS X
 ecm_mark_nongui_executable(desktoptojson)
 
-target_link_libraries(desktoptojson Qt5::Core)
+target_link_libraries(desktoptojson PUBLIC Qt5::Core PRIVATE Qt5::QspXDG)
 target_compile_definitions(desktoptojson PRIVATE BUILDING_DESKTOPTOJSON_TOOL=1)
 
 install(TARGETS desktoptojson EXPORT KF5CoreAddonsToolingTargets ${KF5_INSTALL_TARGETS_DEFAULT_ARGS})
