diff --git a/lib/Basic/LaneBasedExecutionQueue.cpp b/lib/Basic/LaneBasedExecutionQueue.cpp
index 6908c136..c594e4e1 100644
--- a/lib/Basic/LaneBasedExecutionQueue.cpp
+++ b/lib/Basic/LaneBasedExecutionQueue.cpp
@@ -31,7 +31,11 @@
 #include <signal.h>
 
 #ifdef __APPLE__
-#include <pthread/spawn.h>
+#if __has_include(<pthread/spawn.h>)
+#	include <pthread/spawn.h>
+#else
+#	include <pthread.h>
+#endif
 #endif
 
 using namespace llbuild;
diff --git a/lib/Basic/Subprocess.cpp b/lib/Basic/Subprocess.cpp
index 4ff47783..2e9f86a5 100644
--- a/lib/Basic/Subprocess.cpp
+++ b/lib/Basic/Subprocess.cpp
@@ -47,7 +47,11 @@
 #endif
 
 #ifdef __APPLE__
-#include <pthread/spawn.h>
+#if __has_include(<pthread/spawn.h>)
+#	include <pthread/spawn.h>
+#else
+#	include <pthread.h>
+#endif
 #include "TargetConditionals.h"
 #endif
 
@@ -367,9 +371,9 @@ static int posix_spawn_file_actions_addchdir_polyfill(posix_spawn_file_actions_t
   return ENOSYS;
 #elif defined(__APPLE__) && defined(__MAC_OS_X_VERSION_MIN_REQUIRED) && __MAC_OS_X_VERSION_MIN_REQUIRED < 101500
   // Conditionally available on macOS if building with a deployment target older than 10.15
-  if (__builtin_available(macOS 10.15, *)) {
-    return posix_spawn_file_actions_addchdir_np(file_actions, path);
-  }
+//   if (__builtin_available(macOS 10.15, *)) {
+//     return posix_spawn_file_actions_addchdir_np(file_actions, path);
+//   }
   return ENOSYS;
 #elif defined(__GLIBC__) || defined(__APPLE__) || defined(__FreeBSD__) || defined(__ANDROID__) || defined(__musl__)
   // Pre-standard posix_spawn_file_actions_addchdir_np version available in:
@@ -397,7 +401,7 @@ namespace {
   static std::atomic<QualityOfService> defaultQualityOfService{
     QualityOfService::Normal };
 
-#if defined(__APPLE__)
+#if defined(__APPLE__) && defined(QOS_CLASS_USER_INTERACTIVE)
   qos_class_t _getDarwinQOSClass(QualityOfService level) {
     switch (level) {
       case QualityOfService::Normal:
@@ -427,7 +431,7 @@ void llbuild::basic::setDefaultQualityOfService(QualityOfService level) {
 }
 
 void llbuild::basic::setCurrentThreadQualityOfService(QualityOfService level) {
-#if defined(__APPLE__)
+#if defined(__APPLE__) && defined(QOS_CLASS_USER_INTERACTIVE)
   pthread_set_qos_class_self_np(
       _getDarwinQOSClass(level), 0);
 #endif
@@ -1044,7 +1048,7 @@ void llbuild::basic::spawnProcess(
 #endif
 
   // On Darwin, set the QoS of launched processes to one of the current thread.
-#ifdef __APPLE__
+#if defined(__APPLE__) && defined(QOS_CLASS_USER_INTERACTIVE)
   posix_spawnattr_set_qos_class_np(&attributes, qos_class_self());
 #endif
 
diff --git a/lib/llvm/Support/Unix/Signals.inc b/lib/llvm/Support/Unix/Signals.inc
index eea3e61f..03e9203f 100644
--- a/lib/llvm/Support/Unix/Signals.inc
+++ b/lib/llvm/Support/Unix/Signals.inc
@@ -567,7 +567,7 @@ void llvm::sys::PrintStackTraceOnErrorSignal(StringRef Argv0,
 
   AddSignalHandler(PrintStackTraceSignalHandler, nullptr);
 
-#if defined(__APPLE__) && TARGET_OS_OSX && ENABLE_CRASH_OVERRIDES
+#if defined(__APPLE__) && ENABLE_CRASH_OVERRIDES
   // Environment variable to disable any kind of crash dialog.
   if (DisableCrashReporting || getenv("LLVM_DISABLE_CRASH_REPORT")) {
     mach_port_t self = mach_task_self();
diff --git a/utils/adjust-times/adjust-times.cpp b/utils/adjust-times/adjust-times.cpp
index 727d165a..57876b08 100644
--- a/utils/adjust-times/adjust-times.cpp
+++ b/utils/adjust-times/adjust-times.cpp
@@ -35,21 +35,21 @@
 /// or to the current time.
 static int file_time_set_fixed(const char* filename, struct timespec time_to_set) {
 #if defined __APPLE__
-  if (__builtin_available(macOS 10.13, iOS 11.0, watchOS 4.0, tvOS 11.0, *)) {
+//   if (__builtin_available(macOS 10.13, iOS 11.0, watchOS 4.0, tvOS 11.0, *)) {
     struct timespec times[2] = { time_to_set, time_to_set };
     return utimensat(AT_FDCWD, filename, times, 0);
-  } else {
-    // If requested to reset atime/mtime to "now", use the shortcut
-    // instead of computing and using the current time.
-    if (time_to_set.tv_nsec == UTIME_NOW) {
-      return utimes(filename, NULL);
-    }
-    struct timeval tv;
-    tv.tv_sec = time_to_set.tv_sec;
-    tv.tv_usec = (__darwin_suseconds_t)(time_to_set.tv_nsec / 1000);
-    struct timeval times[2] = { tv, tv };
-    return utimes(filename, times);
-  }
+//   } else {
+//     // If requested to reset atime/mtime to "now", use the shortcut
+//     // instead of computing and using the current time.
+//     if (time_to_set.tv_nsec == UTIME_NOW) {
+//       return utimes(filename, NULL);
+//     }
+//     struct timeval tv;
+//     tv.tv_sec = time_to_set.tv_sec;
+//     tv.tv_usec = (__darwin_suseconds_t)(time_to_set.tv_nsec / 1000);
+//     struct timeval times[2] = { tv, tv };
+//     return utimes(filename, times);
+//   }
 #elif !defined(_WIN32)
   // POSIX.1-2008 platforms (Linux, FreeBSD, OpenBSD, etc.) provide utimensat.
   struct timespec times[2] = { time_to_set, time_to_set };
