diff --git a/products/libllbuild/CMakeLists.txt b/products/libllbuild/CMakeLists.txt
index c52be0b3..23b420b6 100644
--- a/products/libllbuild/CMakeLists.txt
+++ b/products/libllbuild/CMakeLists.txt
@@ -60,14 +60,15 @@ add_custom_target(install-llbuild
                           -DCMAKE_INSTALL_COMPONENT=llbuild
                           -P "${CMAKE_BINARY_DIR}/cmake_install.cmake")
 
-if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
   # On OS X, add a target to create the llbuild.framework product.
   add_library(llbuild-framework SHARED ${SOURCES})
-  set_target_properties(llbuild-framework PROPERTIES FRAMEWORK 1)
-  set_target_properties(llbuild-framework PROPERTIES MACOSX_RPATH "@rpath")
+  if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+    set_target_properties(llbuild-framework PROPERTIES FRAMEWORK 1)
+    set_target_properties(llbuild-framework PROPERTIES MACOSX_RPATH "@rpath")
+    set_target_properties(llbuild-framework PROPERTIES LINK_FLAGS " -compatibility_version 1 -current_version 1")
+  endif()
   set_target_properties(llbuild-framework PROPERTIES OUTPUT_NAME llbuild)
   set_target_properties(llbuild-framework PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${CMAKE_LIBRARY_OUTPUT_DIRECTORY})
-  set_target_properties(llbuild-framework PROPERTIES LINK_FLAGS " -compatibility_version 1 -current_version 1")
   target_link_libraries(llbuild-framework PRIVATE
     llbuildBuildSystem
     llbuildCore
@@ -77,33 +78,40 @@ if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
     SQLite3::SQLite3
     curses)
 
-  # Manually set up the remaining framework structure.
-  set(LLBUILD_FW_INPUTS)
+  if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+    # Manually set up the remaining framework structure.
+    set(LLBUILD_FW_INPUTS)
+
+    # Copy the public headers.
+    file(GLOB headers include/llbuild/*.h)
+    foreach(header ${headers})
+      get_filename_component(name ${header} NAME)
+      set(dst ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/llbuild.framework/Versions/A/Headers/${name})
+      list(APPEND LLBUILD_FW_INPUTS ${dst})
+      add_custom_command(OUTPUT ${dst} COMMAND ${CMAKE_COMMAND} -E make_directory ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/llbuild.framework/Versions/A/Headers/ COMMAND ${CMAKE_COMMAND} -E copy ${header} ${dst} DEPENDS ${header})
+    endforeach()
+
+    # Copy the framework module map.
+    set(src ${CMAKE_CURRENT_SOURCE_DIR}/../llbuild-framework/llbuild-module.modulemap)
+    set(dst ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/llbuild.framework/Versions/A/Modules/module.modulemap)
+    list(APPEND LLBUILD_FW_INPUTS ${dst})
+    add_custom_command(OUTPUT ${dst} COMMAND ${CMAKE_COMMAND} -E make_directory ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/llbuild.framework/Versions/A/Modules/ COMMAND ${CMAKE_COMMAND} -E copy ${src} ${dst} DEPENDS ${src})
 
-  # Copy the public headers.
-  file(GLOB headers include/llbuild/*.h)
-  foreach(header ${headers})
-    get_filename_component(name ${header} NAME)
-    set(dst ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/llbuild.framework/Versions/A/Headers/${name})
+    # Create the appropriate symlinks.
+    set(dst ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/llbuild.framework/Headers)
+    list(APPEND LLBUILD_FW_INPUTS ${dst})
+    add_custom_command(OUTPUT ${dst} COMMAND ${CMAKE_COMMAND} -E create_symlink Versions/Current/Headers ${dst})
+    set(dst ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/llbuild.framework/Modules)
     list(APPEND LLBUILD_FW_INPUTS ${dst})
-    add_custom_command(OUTPUT ${dst} COMMAND ${CMAKE_COMMAND} -E make_directory ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/llbuild.framework/Versions/A/Headers/ COMMAND ${CMAKE_COMMAND} -E copy ${header} ${dst} DEPENDS ${header})
-  endforeach()
-
-  # Copy the framework module map.
-  set(src ${CMAKE_CURRENT_SOURCE_DIR}/../llbuild-framework/llbuild-module.modulemap)
-  set(dst ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/llbuild.framework/Versions/A/Modules/module.modulemap)
-  list(APPEND LLBUILD_FW_INPUTS ${dst})
-  add_custom_command(OUTPUT ${dst} COMMAND ${CMAKE_COMMAND} -E make_directory ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/llbuild.framework/Versions/A/Modules/ COMMAND ${CMAKE_COMMAND} -E copy ${src} ${dst} DEPENDS ${src})
-
-  # Create the appropriate symlinks.
-  set(dst ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/llbuild.framework/Headers)
-  list(APPEND LLBUILD_FW_INPUTS ${dst})
-  add_custom_command(OUTPUT ${dst} COMMAND ${CMAKE_COMMAND} -E create_symlink Versions/Current/Headers ${dst})
-  set(dst ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/llbuild.framework/Modules)
-  list(APPEND LLBUILD_FW_INPUTS ${dst})
-  add_custom_command(OUTPUT ${dst} COMMAND ${CMAKE_COMMAND} -E create_symlink Versions/Current/Modules ${dst})
+    add_custom_command(OUTPUT ${dst} COMMAND ${CMAKE_COMMAND} -E create_symlink Versions/Current/Modules ${dst})
+  endif()
 
   # Add a custom target for the setup tasks.
   add_custom_target(llbuild-framework-setup DEPENDS ${LLBUILD_FW_INPUTS})
   add_dependencies(llbuild-framework llbuild-framework-setup)
-endif()
+	install(TARGETS llbuild-framework
+	  ARCHIVE DESTINATION lib${LLBUILD_LIBDIR_SUFFIX}
+	  LIBRARY DESTINATION lib${LLBUILD_LIBDIR_SUFFIX}
+	  FRAMEWORK DESTINATION lib${LLBUILD_LIBDIR_SUFFIX}
+	  RUNTIME DESTINATION bin
+	  COMPONENT llbuild)
