diff --git a/lib/Basic/Subprocess.cpp b/lib/Basic/Subprocess.cpp
index 4ff4778..04d3eb2 100644
--- a/lib/Basic/Subprocess.cpp
+++ b/lib/Basic/Subprocess.cpp
@@ -367,9 +367,9 @@ static int posix_spawn_file_actions_addchdir_polyfill(posix_spawn_file_actions_t
   return ENOSYS;
 #elif defined(__APPLE__) && defined(__MAC_OS_X_VERSION_MIN_REQUIRED) && __MAC_OS_X_VERSION_MIN_REQUIRED < 101500
   // Conditionally available on macOS if building with a deployment target older than 10.15
-  if (__builtin_available(macOS 10.15, *)) {
-    return posix_spawn_file_actions_addchdir_np(file_actions, path);
-  }
+//   if (__builtin_available(macOS 10.15, *)) {
+//     return posix_spawn_file_actions_addchdir_np(file_actions, path);
+//   }
   return ENOSYS;
 #elif defined(__GLIBC__) || defined(__APPLE__) || defined(__FreeBSD__) || defined(__ANDROID__) || defined(__musl__)
   // Pre-standard posix_spawn_file_actions_addchdir_np version available in:
diff --git a/lib/llvm/Support/Unix/Signals.inc b/lib/llvm/Support/Unix/Signals.inc
index eea3e61..03e9203 100644
--- a/lib/llvm/Support/Unix/Signals.inc
+++ b/lib/llvm/Support/Unix/Signals.inc
@@ -567,7 +567,7 @@ void llvm::sys::PrintStackTraceOnErrorSignal(StringRef Argv0,
 
   AddSignalHandler(PrintStackTraceSignalHandler, nullptr);
 
-#if defined(__APPLE__) && TARGET_OS_OSX && ENABLE_CRASH_OVERRIDES
+#if defined(__APPLE__) && ENABLE_CRASH_OVERRIDES
   // Environment variable to disable any kind of crash dialog.
   if (DisableCrashReporting || getenv("LLVM_DISABLE_CRASH_REPORT")) {
     mach_port_t self = mach_task_self();
diff --git a/utils/adjust-times/adjust-times.cpp b/utils/adjust-times/adjust-times.cpp
index 727d165..57876b0 100644
--- a/utils/adjust-times/adjust-times.cpp
+++ b/utils/adjust-times/adjust-times.cpp
@@ -35,21 +35,21 @@
 /// or to the current time.
 static int file_time_set_fixed(const char* filename, struct timespec time_to_set) {
 #if defined __APPLE__
-  if (__builtin_available(macOS 10.13, iOS 11.0, watchOS 4.0, tvOS 11.0, *)) {
+//   if (__builtin_available(macOS 10.13, iOS 11.0, watchOS 4.0, tvOS 11.0, *)) {
     struct timespec times[2] = { time_to_set, time_to_set };
     return utimensat(AT_FDCWD, filename, times, 0);
-  } else {
-    // If requested to reset atime/mtime to "now", use the shortcut
-    // instead of computing and using the current time.
-    if (time_to_set.tv_nsec == UTIME_NOW) {
-      return utimes(filename, NULL);
-    }
-    struct timeval tv;
-    tv.tv_sec = time_to_set.tv_sec;
-    tv.tv_usec = (__darwin_suseconds_t)(time_to_set.tv_nsec / 1000);
-    struct timeval times[2] = { tv, tv };
-    return utimes(filename, times);
-  }
+//   } else {
+//     // If requested to reset atime/mtime to "now", use the shortcut
+//     // instead of computing and using the current time.
+//     if (time_to_set.tv_nsec == UTIME_NOW) {
+//       return utimes(filename, NULL);
+//     }
+//     struct timeval tv;
+//     tv.tv_sec = time_to_set.tv_sec;
+//     tv.tv_usec = (__darwin_suseconds_t)(time_to_set.tv_nsec / 1000);
+//     struct timeval times[2] = { tv, tv };
+//     return utimes(filename, times);
+//   }
 #elif !defined(_WIN32)
   // POSIX.1-2008 platforms (Linux, FreeBSD, OpenBSD, etc.) provide utimensat.
   struct timespec times[2] = { time_to_set, time_to_set };
