diff --git a/products/libllbuild/CMakeLists.txt b/products/libllbuild/CMakeLists.txt
index c52be0b3..0fd44993 100644
--- a/products/libllbuild/CMakeLists.txt
+++ b/products/libllbuild/CMakeLists.txt
@@ -106,4 +106,10 @@ if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
   # Add a custom target for the setup tasks.
   add_custom_target(llbuild-framework-setup DEPENDS ${LLBUILD_FW_INPUTS})
   add_dependencies(llbuild-framework llbuild-framework-setup)
+	install(TARGETS llbuild-framework
+	  ARCHIVE DESTINATION lib${LLBUILD_LIBDIR_SUFFIX}
+	  LIBRARY DESTINATION lib${LLBUILD_LIBDIR_SUFFIX}
+	  FRAMEWORK DESTINATION lib${LLBUILD_LIBDIR_SUFFIX}
+	  RUNTIME DESTINATION bin
+	  COMPONENT llbuild)
 endif()
