diff --git a/skroogeconvert/CMakeLists.txt b/skroogeconvert/CMakeLists.txt
index 28b4901..952f0b1 100644
--- a/skroogeconvert/CMakeLists.txt
+++ b/skroogeconvert/CMakeLists.txt
@@ -18,12 +18,15 @@ MESSAGE( STATUS "..:: CMAKE SKROOGECONVERT ::..")
 
 PROJECT(SKROOGECONVERT)
 
+INCLUDE(ECMMarkNonGuiExecutable)
+
 LINK_DIRECTORIES (${LIBRARY_OUTPUT_PATH})
 
 SET(skroogeconvert_SRCS
    main.cpp
  )
 ADD_EXECUTABLE(skroogeconvert ${skroogeconvert_SRCS})
+ECM_MARK_NONGUI_EXECUTABLE(skroogeconvert)
 
 TARGET_LINK_LIBRARIES(skroogeconvert Qt5::Widgets skgbasemodeler skgbankmodeler)
 
