diff --git a/skrooge/main.cpp b/skrooge/main.cpp
index c49fec7..ced76f5 100644
--- a/skrooge/main.cpp
+++ b/skrooge/main.cpp
@@ -79,7 +79,7 @@ int main(int argc, char** argv)
     qInstallMessageHandler(SKGMessageOutput);
 
     QApplication app(argc, argv);
-    app.setWindowIcon(SKGServices::fromTheme(QStringLiteral("skrooge")));
+    app.setWindowIcon(SKGServices::fromTheme(QStringLiteral("skrooge"), app.windowIcon()));
 
     // Migration kf4 => kf5
     Kdelibs4ConfigMigrator migrate(QStringLiteral("skrooge"));
