diff --git a/skgbasemodeler/skgdocument.cpp b/skgbasemodeler/skgdocument.cpp
index 62e6d08..8730199 100644
--- a/skgbasemodeler/skgdocument.cpp
+++ b/skgbasemodeler/skgdocument.cpp
@@ -1433,10 +1433,10 @@ SKGError SKGDocument::load(const QString& iName, const QString& iPassword, bool
     QProcess sqlite3Process;
     QString mode;
 #ifdef SKGCIPHER
-    sqlite3Process.start(QStringLiteral("sqlcipher"), QStringList() << QStringLiteral("-version"));
+    sqlite3Process.start(QStringLiteral("@@PREFIX@@/sqlcipher"), QStringList() << QStringLiteral("-version"));
     mode = QStringLiteral("SQLCipher");
 #else
-    sqlite3Process.start(QStringLiteral("sqlite3"), QStringList() << QStringLiteral("-version"));
+    sqlite3Process.start(QStringLiteral("@@PREFIX@@/sqlite3"), QStringList() << QStringLiteral("-version"));
     mode = QStringLiteral("SQLite");
 #endif
     if (sqlite3Process.waitForFinished()) {
