diff --git a/skrooge/CMakeLists.txt b/skrooge/CMakeLists.txt
index 03bc220..1bd3e94 100644
--- a/skrooge/CMakeLists.txt
+++ b/skrooge/CMakeLists.txt
@@ -28,7 +28,11 @@ SET(skrooge_SRCS
    main.cpp
  )
 INCLUDE(ECMAddAppIcon)
-ECM_ADD_APP_ICON(skrooge_SRCS ICONS "${CMAKE_CURRENT_SOURCE_DIR}/icons_hicolor/128-apps-skrooge.png")
+file(GLOB ICONS_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/icons_hicolor/[0-9][0-9]-apps-skrooge.png")
+ECM_ADD_APP_ICON(skrooge_SRCS ICONS
+    "${CMAKE_CURRENT_SOURCE_DIR}/icons_hicolor/256-apps-skrooge.png"
+    "${CMAKE_CURRENT_SOURCE_DIR}/icons_hicolor/128-apps-skrooge.png"
+    ${ICONS_SRCS})
 
 ADD_EXECUTABLE(skrooge ${skrooge_SRCS})
 
diff --git a/skrooge/main.cpp b/skrooge/main.cpp
index c49fec7..cf88f4d 100644
--- a/skrooge/main.cpp
+++ b/skrooge/main.cpp
@@ -79,7 +79,10 @@ int main(int argc, char** argv)
     qInstallMessageHandler(SKGMessageOutput);
 
     QApplication app(argc, argv);
-    app.setWindowIcon(SKGServices::fromTheme(QStringLiteral("skrooge")));
+    QIcon appIcon = SKGServices::fromTheme(QStringLiteral("skrooge"));
+    if (!appIcon.isNull()) {
+        app.setWindowIcon(appIcon);
+    }
 
     // Migration kf4 => kf5
     Kdelibs4ConfigMigrator migrate(QStringLiteral("skrooge"));
