diff --git CMakeLists.txt CMakeLists.txt
index 3973d9a..b16360a 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -45,6 +45,7 @@ include(ECMInstallIcons)
 include(ECMAddTests)
 include(ECMAddQch)
 include(ECMQtDeclareLoggingCategory)
+include(ECMMarkNonGuiExecutable)
 
 option(BUILD_QCH "Build API documentation in QCH format (for e.g. Qt Assistant, Qt Creator & KDevelop)" OFF)
 add_feature_info(QCH ${BUILD_QCH} "API documentation in QCH format (for e.g. Qt Assistant, Qt Creator & KDevelop)")
diff --git src/file/CMakeLists.txt src/file/CMakeLists.txt
index de4dc85..ef80b78 100644
--- src/file/CMakeLists.txt
+++ src/file/CMakeLists.txt
@@ -61,6 +61,7 @@ set(file_SRCS
 )
 
 add_executable(baloo_file ${file_SRCS})
+ecm_mark_nongui_executable(baloo_file)
 target_compile_definitions(baloo_file PRIVATE -DPROJECT_VERSION="${PROJECT_VERSION}")
 
 target_link_libraries(baloo_file
diff --git src/file/extractor/CMakeLists.txt src/file/extractor/CMakeLists.txt
index c9c9e4f..421f2bf 100644
--- src/file/extractor/CMakeLists.txt
+++ src/file/extractor/CMakeLists.txt
@@ -18,6 +18,7 @@ set(EXTRACTOR_SRCS
 ecm_qt_declare_logging_category(EXTRACTOR_SRCS HEADER baloodebug.h IDENTIFIER BALOO CATEGORY_NAME org.kde.baloo)
 
 add_executable(baloo_file_extractor ${EXTRACTOR_SRCS})
+ecm_mark_nongui_executable(baloo_file_extractor)
 target_compile_definitions(baloo_file_extractor PRIVATE -DPROJECT_VERSION="${PROJECT_VERSION}")
 
 target_link_libraries(baloo_file_extractor
diff --git src/tools/balooctl/CMakeLists.txt src/tools/balooctl/CMakeLists.txt
index d7be238..f13c446 100644
--- src/tools/balooctl/CMakeLists.txt
+++ src/tools/balooctl/CMakeLists.txt
@@ -22,6 +22,7 @@ set(DBUS_INTERFACES
 SET_SOURCE_FILES_PROPERTIES(${DBUS_INTERFACES} PROPERTIES GENERATED 1 SKIP_AUTOMOC 1)
 
 add_executable(balooctl ${SRCS} ${DBUS_INTERFACES})
+ecm_mark_nongui_executable(balooctl)
 add_dependencies(balooctl BalooDBusInterfaces)
 target_compile_definitions(balooctl PRIVATE -DPROJECT_VERSION="${PROJECT_VERSION}")
 
diff --git src/tools/baloosearch/CMakeLists.txt src/tools/baloosearch/CMakeLists.txt
index 1030a98..a216e3f 100644
--- src/tools/baloosearch/CMakeLists.txt
+++ src/tools/baloosearch/CMakeLists.txt
@@ -3,6 +3,7 @@ add_definitions(-DTRANSLATION_DOMAIN=\"baloosearch5\")
 set(BALOO_SEARCH_SRCS main.cpp)
 
 add_executable(baloosearch ${BALOO_SEARCH_SRCS})
+ecm_mark_nongui_executable(baloosearch)
 target_compile_definitions(baloosearch PRIVATE -DPROJECT_VERSION="${PROJECT_VERSION}")
 
 target_link_libraries(baloosearch
diff --git src/tools/balooshow/CMakeLists.txt src/tools/balooshow/CMakeLists.txt
index f01f093..1c3ff00 100644
--- src/tools/balooshow/CMakeLists.txt
+++ src/tools/balooshow/CMakeLists.txt
@@ -3,6 +3,7 @@ add_definitions(-DTRANSLATION_DOMAIN=\"balooshow5\")
 set(SRCS main.cpp)
 
 add_executable(balooshow ${SRCS})
+ecm_mark_nongui_executable(balooshow)
 target_compile_definitions(balooshow PRIVATE -DPROJECT_VERSION="${PROJECT_VERSION}")
 
 target_link_libraries(balooshow
