commit a5561623bb8c13ab3f720e1e50ecfca789b30db9
Author: R.J.V. Bertin <rjvbertin@gmail.com>
Date:   Thu Mar 8 14:38:03 2018 +0100

    use preferred Mac platform token

diff --git a/src/core/loader.cpp b/src/core/loader.cpp
index 9dbc3ae..9870a16 100644
--- a/src/core/loader.cpp
+++ b/src/core/loader.cpp
@@ -33,7 +33,7 @@
 
 #ifdef SONNET_STATIC
 #include "../plugins/hunspell/hunspellclient.h"
-#ifdef Q_OS_MAC
+#ifdef Q_OS_MACOS
 #include "../plugins/nsspellchecker/nsspellcheckerclient.h"
 #endif
 #endif
@@ -289,7 +289,7 @@ void Loader::loadPlugins()
         qCWarning(SONNET_LOG_CORE) << "Sonnet: No speller backends available!";
     }
 #else
-#ifdef Q_OS_MAC
+#ifdef Q_OS_MACOS
     loadPlugin(QString());
 #endif
     loadPlugin(QStringLiteral("Hunspell"));
@@ -317,7 +317,7 @@ void Loader::loadPlugin(const QString &pluginPath)
     if (pluginPath == QLatin1String("Hunspell")) {
         client = new HunspellClient(this);
     }
-#ifdef Q_OS_MAC
+#ifdef Q_OS_MACOS
     else {
         client = new NSSpellCheckerClient(this);
     }
diff --git a/src/plugins/hunspell/hunspellclient.cpp b/src/plugins/hunspell/hunspellclient.cpp
index da83f6d..637e9ad 100644
--- a/src/plugins/hunspell/hunspellclient.cpp
+++ b/src/plugins/hunspell/hunspellclient.cpp
@@ -37,8 +37,8 @@ HunspellClient::HunspellClient(QObject *parent)
     QStringList dirList;
     const QString AFF_MASK = QStringLiteral("*.aff");
 
-#if defined(Q_OS_MAC) || defined(Q_OS_WIN)
-#ifdef Q_OS_MAC
+#if defined(Q_OS_MACOS) || defined(Q_OS_WIN)
+#ifdef Q_OS_MACOS
     QDir lodir(QStringLiteral("/Applications/LibreOffice.app/Contents/Resources/extensions"));
 #endif
 #ifdef Q_OS_WIN
diff --git a/src/plugins/hunspell/hunspelldict.h b/src/plugins/hunspell/hunspelldict.h
index 5cf0000..a24f71c 100644
--- a/src/plugins/hunspell/hunspelldict.h
+++ b/src/plugins/hunspell/hunspelldict.h
@@ -25,7 +25,7 @@
 #include "hunspell.hxx"
 
 #ifndef HUNSPELL_MAIN_DICT_PATH
-#ifdef Q_OS_MAC
+#ifdef Q_OS_MACOS
 #define HUNSPELL_MAIN_DICT_PATH "/System/Library/Spelling"
 #else
 #define HUNSPELL_MAIN_DICT_PATH "/usr/share/hunspell"
