diff --git channels/rdpsnd/client/mac/CMakeLists.txt channels/rdpsnd/client/mac/CMakeLists.txt
index 8b846561a..807e526ad 100644
--- channels/rdpsnd/client/mac/CMakeLists.txt
+++ channels/rdpsnd/client/mac/CMakeLists.txt
@@ -19,14 +19,20 @@
 
 define_channel_client_subsystem("rdpsnd" "mac" "")
 
-find_library(COCOA_LIBRARY Cocoa REQUIRED)
 FIND_LIBRARY(CORE_FOUNDATION CoreFoundation)
 FIND_LIBRARY(CORE_AUDIO CoreAudio REQUIRED)
 FIND_LIBRARY(AUDIO_TOOL AudioToolbox REQUIRED)
-FIND_LIBRARY(AV_FOUNDATION AVFoundation REQUIRED)
-
-set(${MODULE_PREFIX}_SRCS
-	rdpsnd_mac.m)
+if (CMAKE_HOST_SYSTEM_VERSION VERSION_GREATER_EQUAL 14)
+    find_library(COCOA_LIBRARY Cocoa REQUIRED)
+    FIND_LIBRARY(AV_FOUNDATION AVFoundation REQUIRED)
+
+    set(${MODULE_PREFIX}_SRCS
+        rdpsnd_mac.m)
+else()
+    message(WARNING "Darwin version ${CMAKE_HOST_SYSTEM_VERSION}, using legacy RDPSND implementation")
+    set(${MODULE_PREFIX}_SRCS
+        rdpsnd_mac.c)
+endif()
 
 include_directories(..)
 include_directories(${MACAUDIO_INCLUDE_DIRS})
@@ -37,10 +43,13 @@ add_channel_client_subsystem_library(${MODULE_PREFIX} ${MODULE_NAME} ${CHANNEL_N
 
 set(${MODULE_PREFIX}_LIBS ${${MODULE_PREFIX}_LIBS} 
     ${AUDIO_TOOL}
-    ${AV_FOUNDATION}
     ${CORE_AUDIO}
-    ${COCOA_LIBRARY}
     ${CORE_FOUNDATION})
+if (CMAKE_HOST_SYSTEM_VERSION VERSION_GREATER_EQUAL 14)
+    set(${MODULE_PREFIX}_LIBS ${${MODULE_PREFIX}_LIBS} 
+        ${AV_FOUNDATION}
+        ${COCOA_LIBRARY})
+endif()
 
 set(${MODULE_PREFIX}_LIBS ${${MODULE_PREFIX}_LIBS} freerdp winpr)
 
