diff --git b/gmic-qt/orig.CMakeLists.txt a/gmic-qt/CMakeLists.txt
index 27021d7..b3524e6 100644
--- b/gmic-qt/orig.CMakeLists.txt
+++ a/gmic-qt/CMakeLists.txt
@@ -380,8 +380,6 @@ elseif (CMAKE_BUILD_TYPE STREQUAL "RelWithDebInfo")
       endif (NOT ENABLE_SYSTEM_GMIC)
       set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_CXX_FLAGS_RELWITHDEBINFO} -O2")
     endif()
-else()
-    message(FATAL_ERROR "Build type not recognized (${CMAKE_BUILD_TYPE})")
 endif()
 
 if (NOT ENABLE_SYSTEM_GMIC)
