diff --git b/gmic-qt/src/Host/KritaPlugin/orig.host.cpp a/gmic-qt/src/Host/KritaPlugin/host.cpp
index 7f441e8..c171261 100644
--- b/gmic-qt/src/Host/KritaPlugin/orig.host.cpp
+++ a/gmic-qt/src/Host/KritaPlugin/host.cpp
@@ -25,7 +25,9 @@
 #include <QFileInfo>
 #include <QSharedMemory>
 #include <QStandardPaths>
+#if QT_VERSION >= QT_VERSION_CHECK(5, 10, 0)
 #include <QStringLiteral>
+#endif
 #include <QUuid>
 #include <QVector>
 #include <algorithm>
