diff --git kdevplatform/shell/textdocument.cpp kdevplatform/shell/textdocument.cpp
index 35e0f1d..b6a457b 100644
--- kdevplatform/shell/textdocument.cpp
+++ kdevplatform/shell/textdocument.cpp
@@ -724,6 +724,7 @@ void KDevelop::TextDocument::populateContextMenu( KTextEditor::View* v, QMenu* m
     foreach ( QAction* action, d->m_addedContextMenu->actions() ) {
         menu->addAction(action);
     }
+    menu->setTearOffEnabled(true);
 }
 
 void KDevelop::TextDocument::repositoryCheckFinished(bool canRecreate) {
diff --git kdevplatform/vcs/vcspluginhelper.cpp kdevplatform/vcs/vcspluginhelper.cpp
index 3a93a0f..f8786f3 100644
--- kdevplatform/vcs/vcspluginhelper.cpp
+++ kdevplatform/vcs/vcspluginhelper.cpp
@@ -440,6 +440,7 @@ void VcsPluginHelper::annotationContextMenuAboutToShow( KTextEditor::View* view,
     connect(historyAction, &QAction::triggered, this, [this, rev]() {
         history(rev);
     });
+    menu->setTearOffEnabled(true);
 }
 
 void VcsPluginHelper::handleAnnotationBorderVisibilityChanged(View* view, bool visible)
diff --git plugins/filemanager/filemanager.cpp plugins/filemanager/filemanager.cpp
index d948231..031810f 100644
--- plugins/filemanager/filemanager.cpp
+++ plugins/filemanager/filemanager.cpp
@@ -115,6 +115,7 @@ void FileManager::fillContextMenu(const KFileItem& item, QMenu* menu)
     KDevelop::ContextMenuExtension::populateMenu(tmpMenu, extensions);
     contextActions.append(tmpMenu->actions());
     delete tmpMenu;
+    menu->setTearOffEnabled(true);
 }
 
 void FileManager::openFile(const KFileItem& file)
diff --git plugins/projectmanagerview/projectbuildsetwidget.cpp plugins/projectmanagerview/projectbuildsetwidget.cpp
index e91bf06..5df4400 100644
--- plugins/projectmanagerview/projectbuildsetwidget.cpp
+++ plugins/projectmanagerview/projectbuildsetwidget.cpp
@@ -141,6 +141,7 @@ void ProjectBuildSetWidget::showContextMenu( const QPoint& p )
 
 
     QMenu m(this);
+    m.setTearOffEnabled(true);
     m.setTitle( i18n("Build Set") );
     m.addAction( QIcon::fromTheme(QStringLiteral("list-remove")), i18n( "Remove From Build Set" ), this, SLOT(removeItems()) );
 
