diff --git kdevplatform/sublime/container.cpp kdevplatform/sublime/container.cpp
index 49f98d6..7b27fed 100644
--- kdevplatform/sublime/container.cpp
+++ kdevplatform/sublime/container.cpp
@@ -155,6 +155,11 @@ public:
     QToolButton* documentListButton;
     QMenu* documentListMenu;
     QHash<View*, QAction*> documentListActionForView;
+    bool m_isCocoa;
+
+    ContainerPrivate()
+        : m_isCocoa(QGuiApplication::platformName() == QStringLiteral("cocoa"))
+    {}
 
     /**
      * Updates the context menu which is shown when
@@ -213,7 +218,7 @@ public:
     void setAsDockMenu()
     {
 #ifdef Q_OS_MACOS
-        if (documentListMenu != currentDockMenu) {
+        if (m_isCocoa && documentListMenu != currentDockMenu) {
             documentListMenu->setAsDockMenu();
             currentDockMenu = documentListMenu;
         }
@@ -223,7 +228,7 @@ public:
     ~ContainerPrivate()
     {
 #ifdef Q_OS_MACOS
-        if (documentListMenu == currentDockMenu) {
+        if (m_isCocoa && documentListMenu == currentDockMenu) {
             QMenu().setAsDockMenu();
             currentDockMenu = nullptr;
         }
