diff --git CMakeLists.txt CMakeLists.txt
index d60aa8ccd07bbcad1727d0d9fd7648ecda352681..bc690f7a62b87ef9bcf3f39b9671137273f6e240 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -11,6 +11,12 @@ set(KDEV_PLUGIN_VERSION 30)
 option(KDEVELOP_BUILD_IDE "Build the IDE" ON)
 option(KDEVELOP_BUILD_CLANG_PARSER "Build the Clang-based C/C++ parser (formerly kdev-clang)" ON)
 
+if(APPLE OR WIN32)
+    option(KDEVELOP_NATIVE_DIALOGS "Use native file dialogs instead of KDE dialogs" ON)
+else()
+    option(KDEVELOP_NATIVE_DIALOGS "Use native file dialogs instead of KDE dialogs" OFF)
+endif()
+
 # we need some parts of the ECM CMake helpers
 find_package (ECM 5.14.0 REQUIRED NO_MODULE)
 set(CMAKE_MODULE_PATH ${KDevelop_SOURCE_DIR}/cmake/modules ${ECM_MODULE_PATH})
@@ -98,6 +115,10 @@ add_definitions(
     -DQT_USE_QSTRINGBUILDER
 )
 
+if(KDEVELOP_NATIVE_DIALOGS)
+    add_definitions(-DKDEV_USE_NATIVE_DIALOGS)
+endif()
+
 function(add_compile_flag_if_supported _flag)
     unset(_have_flag CACHE)
     string(REGEX REPLACE "[-=]" "_" _varname ${_flag})
diff --git kdevplatform/shell/documentcontroller.cpp kdevplatform/shell/documentcontroller.cpp
index f10010354087bddc45caa6cc66125c8d810275e3..83df1dc0f18cc558e6735f5b41203bb1f2353fdb 100644
--- kdevplatform/shell/documentcontroller.cpp
+++ kdevplatform/shell/documentcontroller.cpp
@@ -131,12 +131,11 @@ public:
         const auto filter = i18n("*|Text File\n");
         auto parent = Core::self()->uiControllerInternal()->defaultMainWindow();
 
-        // use special dialogs in a KDE session, native dialogs elsewhere
-        if (qEnvironmentVariableIsSet("KDE_FULL_SESSION")) {
-            const auto result = KEncodingFileDialog::getOpenUrlsAndEncoding(QString(), dir,
-                filter, parent, caption);
-            return {result.URLs, result.encoding};
-        }
+#ifndef KDEV_USE_NATIVE_DIALOGS
+        const auto result = KEncodingFileDialog::getOpenUrlsAndEncoding(QString(), dir,
+            filter, parent, caption);
+        return {result.URLs, result.encoding};
+#endif
 
         // note: can't just filter on text files using the native dialog, just display all files
         // see https://phabricator.kde.org/D622#11679
diff --git kdevplatform/shell/openprojectdialog.cpp kdevplatform/shell/openprojectdialog.cpp
index e621edadc98c4aa492d65260f5f4ed2d072d1e1d..f9f746961a7d8b9cda3653ccb00fbe0cf367c073 100644
--- kdevplatform/shell/openprojectdialog.cpp
+++ kdevplatform/shell/openprojectdialog.cpp
@@ -87,7 +87,11 @@ OpenProjectDialog::OpenProjectDialog(bool fetch, const QUrl& startUrl,
         delete helpButton;
     }
 
-    const bool useKdeFileDialog = qEnvironmentVariableIsSet("KDE_FULL_SESSION");
+#ifndef KDEV_USE_NATIVE_DIALOGS
+    const bool useKdeFileDialog = true;
+#else
+    const bool useKdeFileDialog = false;
+#endif
     QStringList filters, allEntry;
     QString filterFormat = useKdeFileDialog
                          ? QStringLiteral("%1|%2 (%1)")
