diff --git a/app/main.cpp b/app/main.cpp
index 806fcab..468e4c6 100644
--- a/app/main.cpp
+++ b/app/main.cpp
@@ -131,7 +131,7 @@ int main(int argc, char *argv[])
     app.setApplicationName(aboutData.data()->componentName());
     app.setApplicationDisplayName(aboutData.data()->displayName());
     app.setOrganizationDomain(aboutData.data()->organizationDomain());
-    QApplication::setWindowIcon(QIcon::fromTheme(QStringLiteral("gwenview")));
+    QApplication::setWindowIcon(QIcon::fromTheme(QStringLiteral("gwenview"), app.windowIcon()));
 
     QCommandLineParser parser;
     aboutData.data()->setupCommandLine(&parser);
diff --git a/importer/importdialog.cpp b/importer/importdialog.cpp
index 2a4d468..f04fb62 100644
--- a/importer/importdialog.cpp
+++ b/importer/importdialog.cpp
@@ -209,7 +209,10 @@ ImportDialog::ImportDialog()
 
     d->mCentralWidget->setCurrentWidget(d->mThumbnailPage);
 
-    setWindowIcon(KIcon("gwenview"));
+    KIcon wIcon("gwenview");
+    if (!wIcon.isNull()) {
+        setWindowIcon(KIcon("gwenview"));
+    }
     setAutoSaveSettings();
 }
 
