diff --git a/lib/gwenviewconfig.kcfg b/lib/gwenviewconfig.kcfg
index bf640240..77c94c75 100644
--- a/lib/gwenviewconfig.kcfg
+++ b/lib/gwenviewconfig.kcfg
@@ -163,7 +163,7 @@
                 <choice name="DocumentView::SoftwareAnimation"/>
                 <choice name="DocumentView::GLAnimation"/>
             </choices>
-            <default>DocumentView::SoftwareAnimation</default>
+            <default>DocumentView::NoAnimation</default>
         </entry>
 
         <entry name="ZoomMode" type="Enum">
