diff --git knotify/CMakeLists.txt knotify/CMakeLists.txt
index 00e7c34..1bfe423 100644
--- knotify/CMakeLists.txt
+++ knotify/CMakeLists.txt
@@ -28,7 +28,7 @@ knotify_export.h
 
 qt4_add_dbus_interfaces(knotify_SRCS ${KDE4_DBUS_INTERFACES_DIR}/org.kde.KSpeech.xml)
 
-kde4_add_app_icon(knotify_SRCS "${CMAKE_SOURCE_DIR}/pics/oxygen/*/apps/preferences-desktop-notification.png")
+kde4_add_app_icon(knotify_SRCS "${KDE4_ICON_INSTALL_DIR}/oxygen/*/apps/preferences-desktop-notification.png")
 
 set (knotify_OUTPUT_NAME knotify4)
 kde4_add_executable( knotify ${knotify_SRCS})
diff --git knotify/main.cpp knotify/main.cpp
index b6266f6..d4583eb 100644
--- knotify/main.cpp
+++ knotify/main.cpp
@@ -32,9 +32,27 @@
 
 #include "knotify.h"
 
+#ifdef Q_OS_MAC
+#include <CoreFoundation/CoreFoundation.h>
+#endif
 
 int main(int argc, char **argv)
 {
+#ifdef Q_OS_MAC
+    CFBundleRef mainBundle = CFBundleGetMainBundle();
+    if (mainBundle) {
+        // get the application's Info Dictionary. For app bundles this would live in the bundle's Info.plist,
+        // for regular executables it is obtained in another way.
+        CFMutableDictionaryRef infoDict = (CFMutableDictionaryRef) CFBundleGetInfoDictionary(mainBundle);
+        if (infoDict) {
+            // Add or set the "LSUIElement" key with/to value "1". This can simply be a CFString.
+            CFDictionarySetValue(infoDict, CFSTR("LSUIElement"), CFSTR("1"));
+            // That's it. We're now considered as an "agent" by the window server, and thus will have
+            // neither menubar nor presence in the Dock or App Switcher.
+        }
+    }
+#endif
+
     KAboutData aboutdata("knotify", "knotify4", ki18n("KNotify"),
                          KDE_VERSION_STRING, ki18n("KDE Notification Daemon"),
                          KAboutData::License_GPL, ki18n("(C) 1997-2008, KDE Developers"));
diff --git knotify/org.kde.knotify.service.cmake knotify/org.kde.knotify.service.cmake
index 4e80235..e09638a 100644
--- knotify/org.kde.knotify.service.cmake
+++ knotify/org.kde.knotify.service.cmake
@@ -1,3 +1,3 @@
 [D-BUS Service]
 Name=org.kde.knotify
-Exec=@CMAKE_INSTALL_PREFIX@/bin/knotify4
+Exec=@BUNDLE_INSTALL_DIR@/knotify4.app/Contents/MacOS/knotify4
