diff --git gcc/orig.collect2.cc gcc/collect2.cc
index d81c7f2..46c2833 100644
--- gcc/orig.collect2.cc
+++ gcc/collect2.cc
@@ -785,7 +785,7 @@ main (int argc, char **argv)
       PLUGIN_LD_SUFFIX,
       "ld.gold",
       "ld.bfd",
-      "ld.lld",
+      "ld64.lld",
       "ld.mold"
     };
   static const char *const real_ld_suffix = "real-ld";
@@ -1077,6 +1077,9 @@ main (int argc, char **argv)
       linker_name = concat (DEFAULT_LINKER,
 			    &ld_suffixes[selected_linker][2],
 			    NULL);
+	 if (verbose) {
+		 fprintf(stderr, "### Checking if requested linker \"%s\" exists\n", linker_name);
+	 }
       if (access (linker_name, X_OK) == 0)
 	ld_file_name = linker_name;
     }
