diff --git a/module/os/macos/spl/spl-thread.c b/module/os/macos/spl/spl-thread.c
index c870330879c21d0b4a3cd4971cbf9022575321c7..d582324ce35b88df7ecafd44f35241c02ed2b207 100644
--- a/module/os/macos/spl/spl-thread.c
+++ b/module/os/macos/spl/spl-thread.c
@@ -209,7 +209,8 @@ set_thread_throughput_named(thread_t thread,
 	    THREAD_THROUGHPUT_QOS_POLICY,
 	    (thread_policy_t)&qosp,
 	    THREAD_THROUGHPUT_QOS_POLICY_COUNT);
-	if (qoskret != KERN_SUCCESS) {
+	// warn if failure and not an invalid argument (which could be OS version related)
+	if (qoskret != KERN_SUCCESS && qoskret != KERN_INVALID_ARGUMENT) {
 		printf("SPL: %s:%d: WARNING failed to set"
 		    " thread throughput policy retval: %d "
 		    " (THREAD_THROUGHPUT_QOS_POLICY %x), %s\n",
@@ -244,7 +245,8 @@ set_thread_latency_named(thread_t thread,
 	    THREAD_LATENCY_QOS_POLICY,
 	    (thread_policy_t)&qosp,
 	    THREAD_LATENCY_QOS_POLICY_COUNT);
-	if (qoskret != KERN_SUCCESS) {
+	// warn if failure and not an invalid argument (which could be OS version related)
+	if (qoskret != KERN_SUCCESS && qoskret != KERN_INVALID_ARGUMENT) {
 		printf("SPL: %s:%d: WARNING failed to set"
 		    " thread latency policy retval: %d "
 		    " (THREAD_LATENCY_QOS_POLICY %x), %s",
