diff --git a/lib/libzfs/os/macos/libzfs_util_os.c b/lib/libzfs/os/macos/libzfs_util_os.c
index a0b43466639c081f94d5b9658240f3c961b5e100..11bf6c4046bab2fcd0d02f0f569f9fc2b737c9b0 100644
--- a/lib/libzfs/os/macos/libzfs_util_os.c
+++ b/lib/libzfs/os/macos/libzfs_util_os.c
@@ -103,7 +103,7 @@ libzfs_module_loaded(const char *module)
 static int
 libzfs_load_module_impl(const char *module)
 {
-	char *argv[4] = {"/sbin/kextload", (char *)module, (char *)0};
+	char *argv[4] = {"/sbin/kextload", "-r", KERNEL_MODPREFIX, (char *)module, (char *)0};
 	char *load_str, *timeout_str;
 	long timeout = 10; /* seconds */
 	long busy_timeout = 10; /* milliseconds */
