diff --git include/vlc_fixups.h include/vlc_fixups.h
index b448d83..638b864 100644
--- include/vlc_fixups.h
+++ include/vlc_fixups.h
@@ -239,8 +239,8 @@ static inline locale_t newlocale(int mask, const char * locale, locale_t base)
 }
 #endif
 
-#if !defined (HAVE_STATIC_ASSERT)
-# define _Static_assert(x, s) ((void) sizeof (struct { unsigned:-!(x); }))
+#if !defined (HAVE_STATIC_ASSERT) && !defined (static_assert)
+# define _Static_assert(x, s) ((void) sizeof (struct { unsigned a[(x) ? 1 : -1];}))
 # define static_assert _Static_assert
 #endif
 
--- config.h.in.orig    2016-03-22 15:34:33.000000000 +0000
+++ config.h.in 2016-03-22 15:34:57.000000000 +0000
@@ -793,6 +793,7 @@
 /* Define to `int' if <stddef.h> does not define. */
 #undef ssize_t

+#define static_assert _Static_assert
 #include <vlc_fixups.h>



