diff --git a/mac/Wrapper.c b/mac/Wrapper.c
index 62dcd9bf6..8f2c1a2d7 100644
--- a/mac/Wrapper.c
+++ b/mac/Wrapper.c
@@ -43,8 +43,14 @@ int main(int argc, char *argv[])
    {
       strcpy(++slash, audacity);
    }
+   // change argv[0] so that `ps` will show the actually running binary
+   argv[0] = path;
 
-   unsetenv("DYLD_LIBRARY_PATH");
+   if (!getenv("AUDACITY_PRESERVE_LIBRARY_PATH")) {
+      unsetenv("DYLD_LIBRARY_PATH");
+   }
 
    execve(path, argv, environ);
+   perror(path);
+   exit(-1);
 }
