--- src/orig.kpassworddialog.ui	2017-06-03 12:22:27.000000000 +0200
+++ src/kpassworddialog.ui	2017-06-15 16:40:59.649747774 +0200
@@ -6,11 +6,20 @@
    <rect>
     <x>0</x>
     <y>0</y>
-    <width>400</width>
-    <height>227</height>
+    <width>412</width>
+    <height>255</height>
    </rect>
   </property>
+  <property name="sizePolicy">
+   <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
+    <horstretch>0</horstretch>
+    <verstretch>0</verstretch>
+   </sizepolicy>
+  </property>
   <layout class="QVBoxLayout" name="vBoxLayout">
+   <property name="sizeConstraint">
+    <enum>QLayout::SetFixedSize</enum>
+   </property>
    <item>
     <layout class="QHBoxLayout" name="hboxLayout">
      <item>
@@ -43,7 +52,7 @@
       <enum>Qt::Vertical</enum>
      </property>
      <property name="sizeType">
-      <enum>QSizePolicy::Fixed</enum>
+      <enum>QSizePolicy::Ignored</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
@@ -59,14 +68,14 @@
    <item>
     <widget class="QRadioButton" name="anonymousRadioButton">
      <property name="text">
-      <string>No password, use anonymous (or guest) login</string>
+      <string>&amp;No password, use anonymous (or guest) login</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QRadioButton" name="usePasswordButton">
      <property name="text">
-      <string>Use this password:</string>
+      <string>&amp;Use this password:</string>
      </property>
      <property name="checked">
       <bool>true</bool>
@@ -76,7 +85,16 @@
    <item>
     <widget class="QWidget" name="credentialsGroup" native="true">
      <layout class="QFormLayout" name="formLayout">
-      <property name="margin">
+      <property name="leftMargin">
+       <number>1</number>
+      </property>
+      <property name="topMargin">
+       <number>1</number>
+      </property>
+      <property name="rightMargin">
+       <number>1</number>
+      </property>
+      <property name="bottomMargin">
        <number>1</number>
       </property>
       <item row="1" column="0">
@@ -90,7 +108,7 @@
        <widget class="QLineEdit" name="userEdit">
         <property name="clearButtonEnabled">
          <bool>true</bool>
-       </property>
+        </property>
        </widget>
       </item>
       <item row="3" column="0">
@@ -126,6 +144,9 @@
       </item>
       <item row="5" column="1">
        <widget class="QCheckBox" name="keepCheckBox">
+        <property name="enabled">
+         <bool>true</bool>
+        </property>
         <property name="text">
          <string>Remember password</string>
         </property>
@@ -174,7 +195,7 @@
      <x>252</x>
      <y>192</y>
     </hint>
-  </hints>
+   </hints>
   </connection>
   <connection>
    <sender>usePasswordButton</sender>
