--- qtbase/mkspecs/common/macx.conf
+++ qtbase/mkspecs/common/macx.conf
@@ -5,4 +5,9 @@
 QMAKE_PLATFORM         += osx macx
 QMAKE_MAC_SDK           = macosx
 
+# X11 is not inexistant on OS X; provide the library requirements but leave it to the
+# user to make them accessible during the Qt build.
+QMAKE_LIBS_X11          = -lXext -lX11
+QMAKE_LIBS_OPENVG       = -lOpenVG
+
 include(mac.conf)
# NB 5.6.1 : this hunk is now in patch-enable-fontconfig.pri !
--- qtbase/src/platformsupport/fontdatabases/fontdatabases.pri
+++ qtbase/src/platformsupport/fontdatabases/fontdatabases.pri
@@ -1,8 +1,8 @@
-!win32|contains(QT_CONFIG, freetype):!mac {
+!win32|contains(QT_CONFIG, freetype) {
     include($$PWD/basic/basic.pri)
 }
 
-unix:!mac {
+unix {
     CONFIG += qpa/genericunixfontdatabase
     include($$PWD/genericunix/genericunix.pri)
     contains(QT_CONFIG,fontconfig) {
--- qtbase/src/plugins/platforms/offscreen/offscreen.pro
+++ qtbase/src/plugins/platforms/offscreen/offscreen.pro
@@ -18,7 +18,7 @@
 
 OTHER_FILES += offscreen.json
 
-contains(QT_CONFIG, xlib):contains(QT_CONFIG, opengl):!contains(QT_CONFIG, opengles2) {
+contains(QT_CONFIG, xlib):contains(QT_CONFIG, opengl):!contains(QT_CONFIG, opengles2):!mac {
     SOURCES += qoffscreenintegration_x11.cpp
     HEADERS += qoffscreenintegration_x11.h
     system(echo "Using X11 offscreen integration with GLX")
--- qtbase/src/plugins/platforms/xcb/xcb-plugin.pro
+++ qtbase/src/plugins/platforms/xcb/xcb-plugin.pro
@@ -56,6 +56,9 @@
                        LIBXI_PATCH=$$QMAKE_LIBXI_VERSION_PATCH
         }
     }
+    mac {
+	    LIBS += -lGL
+    }
 }
 
 # to support custom cursors with depth > 1
diff --git qtbase/src/platformsupport/linuxaccessibility/linuxaccessibility.pri qtbase/src/platformsupport/linuxaccessibility/linuxaccessibility.pri
index 1b65fb1..0d9b56d 100644
--- qtbase/src/platformsupport/linuxaccessibility/linuxaccessibility.pri
+++ qtbase/src/platformsupport/linuxaccessibility/linuxaccessibility.pri
@@ -4,6 +4,9 @@ contains(QT_CONFIG, accessibility-atspi-bridge) {
     include(../../3rdparty/atspi2/atspi2.pri)
 
     INCLUDEPATH += $$PWD
+    darwin: {
+        INCLUDEPATH += $$PWD/../../gui/kernel
+    }
 
     HEADERS += \
         $$PWD/application_p.h \
--- qtbase/src/platformsupport/themes/orig.themes.pri	2014-09-11 12:48:01.000000000 +0200
+++ qtbase/src/platformsupport/themes/themes.pri	2017-08-07 13:31:25.000000000 +0200
@@ -1,3 +1,3 @@
-unix:!mac {
+unix {
     include($$PWD/genericunix/genericunix.pri)
 }
