--- qtbase/src/gui/opengl/orig.qopenglext.h	2014-09-11 12:48:04.000000000 +0200
+++ qtbase/src/gui/opengl/qopenglext.h	2017-08-06 22:41:15.000000000 +0200
@@ -6221,8 +6221,12 @@
 #ifndef GL_ARB_shader_objects
 /* GL types for program/shader text and shader object handles */
 typedef char GLcharARB;
+#ifdef __APPLE__
+typedef void *GLhandleARB;
+#else
 typedef unsigned int GLhandleARB;
 #endif
+#endif
 
 /* GL type for "half" precision (s10e5) float data in host memory */
 #ifndef GL_ARB_half_float_pixel
