diff --git a/src/video/videowidget.cpp b/src/video/videowidget.cpp
index 45330c6..6b28c34 100644
--- a/src/video/videowidget.cpp
+++ b/src/video/videowidget.cpp
@@ -115,9 +115,15 @@ private:
         Q_ASSERT(!m_frame.isNull()); // ctor may construct null if allocation fails
         m_frame.fill(0);
         pitches[0] = m_frame.bytesPerLine();
+#if QT_VERSION >= QT_VERSION_CHECK(5, 10, 0)
         lines[0] = m_frame.sizeInBytes() / m_frame.bytesPerLine();
 
         return  m_frame.sizeInBytes();
+#else
+        lines[0] = m_frame.byteCount() / m_frame.bytesPerLine();
+
+        return  m_frame.byteCount();
+#endif
     }
 
     virtual void formatCleanUpCallback()
