diff --git a/src/mediaobject.h b/src/mediaobject.h
index 8033113..3fd5013 100644
--- a/src/mediaobject.h
+++ b/src/mediaobject.h
@@ -194,10 +194,12 @@ signals:
 
     void availableChaptersChanged(int);
     void availableTitlesChanged(int);
+    void availableAnglesChanged(int);
 
     void chapterChanged(int chapterNumber);
     void titleChanged(int titleNumber);
     void durationChanged(qint64 newDuration);
+    void angleChanged(int titleNumber);
 
     /**
      * New widget size computed by VLC.
