diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index a10cd83..90b61d1 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -20,8 +20,8 @@ endfunction()
 ##################################
 
 if (LibHeif_FOUND)
-    kimageformats_add_plugin("kimg_heif${QT_MAJOR_VERSION}" SOURCES "heif.cpp")
-    target_link_libraries("kimg_heif${QT_MAJOR_VERSION}" PkgConfig::LibHeif)
+    kimageformats_add_plugin("kimg_heif" SOURCES "heif.cpp")
+    target_link_libraries("kimg_heif" PkgConfig::LibHeif)
 
     if (QT_MAJOR_VERSION STREQUAL "5")
         install(FILES heif.desktop DESTINATION ${KDE_INSTALL_KSERVICES5DIR}/qimageioplugins/)
