diff --git a/qttools/src/qtpaths/qtpaths.cpp b/qttools/src/qtpaths/qtpaths.cpp
index 320ab96..2f2f3a2 100644
--- a/qttools/src/qtpaths/qtpaths.cpp
+++ b/qttools/src/qtpaths/qtpaths.cpp
@@ -46,6 +46,8 @@
 
 #include <stdio.h>
 
+#include <qstandardpaths_p.h>
+
 QT_USE_NAMESPACE
 
 /**
@@ -202,8 +213,13 @@ int main(int argc, char **argv)
                                  QCoreApplication::translate("qtpaths", "Location of Qt plugins."));
     parser.addOption(plugindir);
 
+    QCommandLineOption qspxdg(QStringList() << QStringLiteral("xdgmode") << QStringLiteral("xdg-mode"),
+                                 QCoreApplication::translate("qtpaths", "Activate XDG mode."));
+    parser.addOption(qspxdg);
+
     parser.process(app);
 
+    QStandardPathsConfiguration qspUsesXDGLocations(parser.isSet(qspxdg));
     QStandardPaths::enableTestMode(parser.isSet(testmode));
 
     QStringList results;
