--- a/lib-src/libnyquist/nyquist/ffts/src/fftlib.c
+++ b/lib-src/libnyquist/nyquist/ffts/src/fftlib.c
@@ -2109,8 +2109,8 @@
 }
 }
 
-void ifftrecurs(float *ioptr, long M, float *Utbl, long Ustride, long NDiffU, long StageCnt);
-void ifftrecurs(float *ioptr, long M, float *Utbl, long Ustride, long NDiffU, long StageCnt){
+static inline void ifftrecurs(float *ioptr, long M, float *Utbl, long Ustride, long NDiffU, long StageCnt);
+static inline void ifftrecurs(float *ioptr, long M, float *Utbl, long Ustride, long NDiffU, long StageCnt){
 /* recursive bfstages calls to maximize on chip cache efficiency */
 long i1;
 if (M <= MCACHE)
