diff --git a/src/pextlib1.0/Pextlib.c b/src/pextlib1.0/Pextlib.c
index 1a5420c2..c656e867 100644
--- a/src/pextlib1.0/Pextlib.c
+++ b/src/pextlib1.0/Pextlib.c
@@ -44,6 +44,11 @@
 /* required for vasprintf(3) on Linux */
 #define _GNU_SOURCE
 #endif
+#ifdef __FreeBSD__
+// for u_int etc.
+#define __BSD_VISIBLE 1
+#endif
+
 /* required for clearenv(3)/setenv(3)/unsetenv(3) on macOS */
 #define _DARWIN_C_SOURCE

diff --git a/src/pextlib1.0/adv-flock.c b/src/pextlib1.0/adv-flock.c
index b59a6cc7..ce6388f7 100644
--- a/src/pextlib1.0/adv-flock.c
+++ b/src/pextlib1.0/adv-flock.c
@@ -38,6 +38,10 @@
   hides flock on macOS */
 #define _XOPEN_SOURCE 500L
 #endif
+#ifdef __FreeBSD__
+// for u_int etc.
+#define __BSD_VISIBLE 1
+#endif

 #if HAVE_SYS_FILE_H
 #include <sys/file.h>
