diff --git a/configure b/configure
index c20027d7a319b877e802f94b70d0d663b0a7f96b..5794d152b2b40c051e9c0441fad47c268c73ac75 100755
--- a/configure
+++ b/configure
@@ -7209,7 +6381,7 @@ else
     # This has to be set to something or mtree won't create any directories.
     # It doesn't matter what it's set to for non-root though, as it can't
     # change the ownership anyway.
-    MTREE_GNAME="admin"
+    MTREE_GNAME="adm"
 fi
 
 
diff --git a/configure.ac b/configure.ac
index ab77f374608ada3361882732e50d2c10c91b62f2..76fd981d285aaa139abfb1ffae7a54c528427f36 100644
--- a/configure.ac
+++ b/configure.ac
@@ -236,7 +239,7 @@ else
     # This has to be set to something or mtree won't create any directories.
     # It doesn't matter what it's set to for non-root though, as it can't
     # change the ownership anyway.
-    MTREE_GNAME="admin"
+    MTREE_GNAME="adm"
 fi
 AC_SUBST(MTREE_GNAME)
 

