diff --git a/src/macports1.0/macports.tcl b/src/macports1.0/macports.tcl
index de40dbfe9..68b98aacc 100644
--- a/src/macports1.0/macports.tcl
+++ b/src/macports1.0/macports.tcl
@@ -6581,7 +6860,11 @@ proc macports::get_compiler_version {compiler developer_dir} {
 
     switch -- [file tail ${compiler}] {
         clang {
-            set re {clang(?:_.*)?-([0-9.]+)}
+            if {${macports::os_platform} eq "darwin"} {
+                set re {clang(?:_.*)?-([0-9.]+)}
+            } else {
+                set re {clang version ([0-9.]+)}
+            }
         }
         llvm-gcc-4.2 {
             set re {LLVM build ([0-9.]+)}
@@ -6590,6 +6873,9 @@ proc macports::get_compiler_version {compiler developer_dir} {
         gcc-4.0 {
             set re {build ([0-9.]+)}
         }
+        gcc {
+            set re {gcc version ([0-9.]+)}
+        }
         default {
             return -code error "don't know how to determine build number of compiler \"${compiler}\""
         }
