diff --git a/src/port1.0/portconfigure.tcl b/src/port1.0/portconfigure.tcl
index fc2a7fb58f75bbb87fd6d8847b4ed6e5c20493ad..03991b1ae3a367b7f88f315d291b39e4ebe1d653 100644
--- a/src/port1.0/portconfigure.tcl
+++ b/src/port1.0/portconfigure.tcl
@@ -599,15 +599,20 @@ proc portconfigure::configure_get_universal_archflags {} {
 
 # internal proc to determine if the compiler supports -arch
 proc portconfigure::arch_flag_supported {compiler {multiple_arch_flags no}} {
-    if {${multiple_arch_flags}} {
-        return [regexp {^gcc-4|llvm|apple|clang} ${compiler}]
-    } else {
-        # GCC prior to 4.7 does not accept -arch flag
-        if {[regexp {^macports(?:-[^-]+)?-gcc-4\.[0-6]} ${compiler}]} {
-            return no
+    global os.platform
+    if {${os.platform} eq "darwin"} {
+        if {${multiple_arch_flags}} {
+            return [regexp {^gcc-4|llvm|apple|clang} ${compiler}]
         } else {
-            return yes
+            # GCC prior to 4.7 does not accept -arch flag
+            if {[regexp {^macports(?:-[^-]+)?-gcc-4\.[0-6]} ${compiler}]} {
+                return no
+            } else {
+                return yes
+            }
         }
+    } else {
+        return no
     }
 }
 
