commit d0e37c73ec0cd992aee1e06ee89d201ab544f0b1
Author: René Bertin <rjvbertin@gmail.com>
Date:   Wed May 29 18:38:52 2024 +0200

    lamarck/patch-no-arch-flag.diff

diff --git a/src/port1.0/portconfigure.tcl b/src/port1.0/portconfigure.tcl
index b596453a9..8ced6421a 100644
--- a/src/port1.0/portconfigure.tcl
+++ b/src/port1.0/portconfigure.tcl
@@ -728,6 +728,10 @@ proc portconfigure::configure_get_universal_archflags {} {
 
 # internal proc to determine if the compiler supports -arch
 proc portconfigure::arch_flag_supported {compiler {multiple_arch_flags no}} {
+    global os.platform
+    if {${os.platform} ne "darwin"} {
+        return no
+    }
     if {${multiple_arch_flags}} {
         return [regexp {^gcc-4|llvm|apple|clang} ${compiler}]
     } else {
