diff --git a/configure b/configure
index 375b0946..7f9421ef 100755
--- a/configure
+++ b/configure
@@ -7366,15 +7366,15 @@ fi
    fi

    CFLAGS_LIBCURL=$($CURL_CONFIG --cflags)
-   if test "x$curlprefix" = "x"; then
-       # System curl-config emits absurd output for --libs
-       # See rdar://7244457
-       LDFLAGS_LIBCURL="-lcurl"
-   else
+#  if test "x$curlprefix" = "x"; then
+#      # System curl-config emits absurd output for --libs
+#      # See rdar://7244457
+#      LDFLAGS_LIBCURL="-lcurl"
+#  else
        # Due to a bug in dist, --arch flags are improperly supplied by curl-config.
        # Get rid of them.
        LDFLAGS_LIBCURL=$($CURL_CONFIG --libs | sed 's/-arch [A-Za-z0-9_]* //g')
-   fi
+#  fi



