diff --git a/src/driver/slbt_host_params.c b/src/driver/slbt_host_params.c
index 5811e0269b519c68b3114c8a1b5827fd09d23885..9892e21fe1f828340b4970b1e0c909d01d82b8af 100644
--- a/src/driver/slbt_host_params.c
+++ b/src/driver/slbt_host_params.c
@@ -49,7 +49,7 @@ static const char*ldrpath_elf[] = {
 		exep,exes,impp,imps,                       \
 		ldenv,pic}
 
-SLBT_FLAVOR_SETTINGS(host_flavor_default,       \
+SLBT_FLAVOR_SETTINGS(host_flavor_unix,       \
 	"elf","-fPIC",                          \
 	"lib",".a","lib",".so",".so","",        \
 	"","","","",                            \
@@ -79,6 +79,12 @@ SLBT_FLAVOR_SETTINGS(host_flavor_darwin,        \
 	"","","","",                            \
 	"DYLD_LIBRARY_PATH");
 
+#ifdef __APPLE__
+#define host_flavor_default host_flavor_darwin
+#else
+#define host_flavor_default host_flavor_unix
+#endif
+
 
 static void slbt_get_host_quad(
 	char *	hostbuf,
