diff --git a/src/logic/linkcmd/slbt_linkcmd_dsolib.c b/src/logic/linkcmd/slbt_linkcmd_dsolib.c
index c395118..0e47cb6 100644
--- a/src/logic/linkcmd/slbt_linkcmd_dsolib.c
+++ b/src/logic/linkcmd/slbt_linkcmd_dsolib.c
@@ -103,9 +103,11 @@ int slbt_exec_link_create_library(
 				&depsmeta) < 0)
 			return SLBT_NESTED_ERROR(dctx);
 
+#ifndef __APPLE__
 	/* --no-undefined */
 	if (dctx->cctx->drvflags & SLBT_DRIVER_NO_UNDEFINED)
 		*ectx->noundef = "-Wl,--no-undefined";
+#endif
 
 	/* -soname */
 	dot   = strrchr(dctx->cctx->output,'.');
@@ -121,7 +123,11 @@ int slbt_exec_link_create_library(
 				"-Wl,%s",dctx->cctx->output) < 0)
 			return SLBT_BUFFER_ERROR(dctx);
 
+#ifdef __APPLE__
+		*ectx->soname  = "-Wl,-install_name";
+#else
 		*ectx->soname  = "-Wl,-soname";
+#endif
 		*ectx->lsoname = soname;
 
 	} else if (relfilename && dctx->cctx->verinfo.verinfo) {
@@ -135,7 +141,11 @@ int slbt_exec_link_create_library(
 					dctx->cctx->settings.osdfussix) < 0)
 			return SLBT_BUFFER_ERROR(dctx);
 
+#ifdef __APPLE__
+		*ectx->soname  = "-Wl,-install_name";
+#else
 		*ectx->soname  = "-Wl,-soname";
+#endif
 		*ectx->lsoname = soname;
 
 	} else if (relfilename) {
@@ -147,7 +157,11 @@ int slbt_exec_link_create_library(
 					dctx->cctx->settings.dsosuffix) < 0)
 			return SLBT_BUFFER_ERROR(dctx);
 
+#ifdef __APPLE__
+		*ectx->soname  = "-Wl,-install_name";
+#else
 		*ectx->soname  = "-Wl,-soname";
+#endif
 		*ectx->lsoname = soname;
 
 	} else if (dctx->cctx->drvflags & SLBT_DRIVER_AVOID_VERSION) {
@@ -158,7 +172,11 @@ int slbt_exec_link_create_library(
 					dctx->cctx->settings.dsosuffix) < 0)
 			return SLBT_BUFFER_ERROR(dctx);
 
+#ifdef __APPLE__
+		*ectx->soname  = "-Wl,-install_name";
+#else
 		*ectx->soname  = "-Wl,-soname";
+#endif
 		*ectx->lsoname = soname;
 
 	} else {
@@ -171,7 +189,11 @@ int slbt_exec_link_create_library(
 					dctx->cctx->settings.osdfussix) < 0)
 			return SLBT_BUFFER_ERROR(dctx);
 
+#ifdef __APPLE__
+		*ectx->soname  = "-Wl,-install_name";
+#else
 		*ectx->soname  = "-Wl,-soname";
+#endif
 		*ectx->lsoname = soname;
 	}
 
diff --git a/src/logic/linkcmd/slbt_linkcmd_executable.c b/src/logic/linkcmd/slbt_linkcmd_executable.c
index cbb6ffe..9182f19 100644
--- a/src/logic/linkcmd/slbt_linkcmd_executable.c
+++ b/src/logic/linkcmd/slbt_linkcmd_executable.c
@@ -114,9 +114,11 @@ int slbt_exec_link_create_executable(
 				&depsmeta) < 0)
 			return SLBT_NESTED_ERROR(dctx);
 
+#ifndef __APPLE__
 	/* --no-undefined */
 	if (dctx->cctx->drvflags & SLBT_DRIVER_NO_UNDEFINED)
 		*ectx->noundef = "-Wl,--no-undefined";
+#endif
 
 	/* executable wrapper: create */
 	if (slbt_snprintf(wrapper,sizeof(wrapper),
