diff --git a/src/driver/slbt_driver_ctx.c b/src/driver/slbt_driver_ctx.c
index ec468a2..f5659de 100644
--- a/src/driver/slbt_driver_ctx.c
+++ b/src/driver/slbt_driver_ctx.c
@@ -978,6 +978,8 @@ int slbt_get_driver_ctx(
 
 	/* shared and static objects: enable by default, disable by ~switch */
 	cctx.drvflags = flags | SLBT_DRIVER_SHARED | SLBT_DRIVER_STATIC;
+	// RJVB
+	cctx.drvflags |= SLBT_DRIVER_SILENT;
 
 	/* full annotation when annotation is on; */
 	if (!(cctx.drvflags & SLBT_DRIVER_ANNOTATE_NEVER))
diff --git a/src/internal/slibtool_lconf_impl.c b/src/internal/slibtool_lconf_impl.c
index 8271f59..453176a 100644
--- a/src/internal/slibtool_lconf_impl.c
+++ b/src/internal/slibtool_lconf_impl.c
@@ -125,15 +125,16 @@ static int slbt_lconf_trace_openat_plain(
 		serr[0] = 0;
 	}
 
-	slbt_dprintf(
-		fderr,
-		"%s: %s: openat(%s,%c%s%c,%s,%d) = %d%s.\n",
-		dctx->program,
-		"lconf",
-		scwd,
-		'"',path,'"',
-		(oflag == O_DIRECTORY) ? "O_DIRECTORY" : "O_RDONLY",
-		mode,ret,serr);
+	if (ret < 0)
+		slbt_dprintf(
+			fderr,
+			"%s: %s: openat(%s,%c%s%c,%s,%d) = %d%s.\n",
+			dctx->program,
+			"lconf",
+			scwd,
+			'"',path,'"',
+			(oflag == O_DIRECTORY) ? "O_DIRECTORY" : "O_RDONLY",
+			mode,ret,serr);
 
 	return ret;
 }
@@ -169,39 +170,40 @@ static int slbt_lconf_trace_openat_annotated(
 		serr[0] = 0;
 	}
 
-	slbt_dprintf(
-		fderr,
-		"%s%s%s%s: %s%s%s: openat(%s%s%s%s,%s%s%c%s%c%s,%s%s%s%s,%d) = %s%d%s%s%s%s%s.\n",
+	if (ret < 0)
+		slbt_dprintf(
+			fderr,
+			"%s%s%s%s: %s%s%s: openat(%s%s%s%s,%s%s%c%s%c%s,%s%s%s%s,%d) = %s%d%s%s%s%s%s.\n",
 
-		aclr_bold,aclr_magenta,
-		dctx->program,
-		aclr_reset,
+			aclr_bold,aclr_magenta,
+			dctx->program,
+			aclr_reset,
 
-		aclr_bold,
-		"lconf",
-		aclr_reset,
+			aclr_bold,
+			"lconf",
+			aclr_reset,
 
-		aclr_bold,aclr_blue,
-		scwd,
-		aclr_reset,
+			aclr_bold,aclr_blue,
+			scwd,
+			aclr_reset,
 
-		aclr_bold,aclr_green,
-		'"',path,'"',
-		aclr_reset,
+			aclr_bold,aclr_green,
+			'"',path,'"',
+			aclr_reset,
 
-		aclr_bold,aclr_blue,
-		(oflag == O_DIRECTORY) ? "O_DIRECTORY" : "O_RDONLY",
-		aclr_reset,
+			aclr_bold,aclr_blue,
+			(oflag == O_DIRECTORY) ? "O_DIRECTORY" : "O_RDONLY",
+			aclr_reset,
 
-		mode,
+			mode,
 
-		aclr_bold,
-		ret,
-		aclr_reset,
+			aclr_bold,
+			ret,
+			aclr_reset,
 
-		aclr_bold,aclr_red,
-		serr,
-		aclr_reset);
+			aclr_bold,aclr_red,
+			serr,
+			aclr_reset);
 
 	return ret;
 }
@@ -248,27 +250,28 @@ static int slbt_lconf_trace_fstat_plain(
 		serr[0] = 0;
 	}
 
-	slbt_dprintf(
-		fderr,
-		"%s: %s: %s(%s%s%s%s%s,...) = %d%s%s",
-		dctx->program,
-		"lconf",
-		path ? "fstatat" : "fstat",
-		scwd,
-		path ? "," : "",
-		path ? quot : "",
-		path ? path : "",
-		path ? quot : "",
-		ret,
-		serr,
-		ret ? ".\n" : "");
-
-	if (ret == 0)
+	if (ret) 
 		slbt_dprintf(
 			fderr,
-			" {.st_dev = %ld, .st_ino = %ld}.\n",
-			st->st_dev,
-			st->st_ino);
+			"%s: %s: %s(%s%s%s%s%s,...) = %d%s%s",
+			dctx->program,
+			"lconf",
+			path ? "fstatat" : "fstat",
+			scwd,
+			path ? "," : "",
+			path ? quot : "",
+			path ? path : "",
+			path ? quot : "",
+			ret,
+			serr,
+			ret ? ".\n" : "");
+
+// 	if (ret == 0)
+// 		slbt_dprintf(
+// 			fderr,
+// 			" {.st_dev = %ld, .st_ino = %ld}.\n",
+// 			st->st_dev,
+// 			st->st_ino);
 
 	return ret;
 }
@@ -304,57 +307,58 @@ static int slbt_lconf_trace_fstat_annotated(
 		serr[0] = 0;
 	}
 
-	slbt_dprintf(
-		fderr,
-		"%s%s%s%s: %s%s%s: %s(%s%s%s%s%s%s%s%s%s%s%s,...) = %s%d%s%s%s%s%s%s",
-
-		aclr_bold,aclr_magenta,
-		dctx->program,
-		aclr_reset,
-
-		aclr_bold,
-		"lconf",
-		aclr_reset,
-
-		path ? "fstatat" : "fstat",
-
-		aclr_bold,aclr_blue,
-		scwd,
-		aclr_reset,
-
-		aclr_bold,aclr_green,
-		path ? "," : "",
-		path ? quot : "",
-		path ? path : "",
-		path ? quot : "",
-		aclr_reset,
+	if (ret)
+		slbt_dprintf(
+			fderr,
+			"%s%s%s%s: %s%s%s: %s(%s%s%s%s%s%s%s%s%s%s%s,...) = %s%d%s%s%s%s%s%s",
 
-		aclr_bold,
-		ret,
-		aclr_reset,
+			aclr_bold,aclr_magenta,
+			dctx->program,
+			aclr_reset,
 
-		aclr_bold,aclr_red,
-		serr,
-		aclr_reset,
+			aclr_bold,
+			"lconf",
+			aclr_reset,
 
-		ret ? ".\n" : "");
+			path ? "fstatat" : "fstat",
 
-	if (ret == 0)
-		slbt_dprintf(
-			fderr,
-			" {%s%s.st_dev%s = %s%ld%s, %s%s.st_ino%s = %s%ld%s}.\n",
+			aclr_bold,aclr_blue,
+			scwd,
+			aclr_reset,
 
-			aclr_bold,aclr_yellow,aclr_reset,
+			aclr_bold,aclr_green,
+			path ? "," : "",
+			path ? quot : "",
+			path ? path : "",
+			path ? quot : "",
+			aclr_reset,
 
 			aclr_bold,
-			st->st_dev,
+			ret,
 			aclr_reset,
 
-			aclr_bold,aclr_yellow,aclr_reset,
+			aclr_bold,aclr_red,
+			serr,
+			aclr_reset,
 
-			aclr_bold,
-			st->st_ino,
-			aclr_reset);
+			ret ? ".\n" : "");
+
+// 	if (ret == 0)
+// 		slbt_dprintf(
+// 			fderr,
+// 			" {%s%s.st_dev%s = %s%ld%s, %s%s.st_ino%s = %s%ld%s}.\n",
+// 
+// 			aclr_bold,aclr_yellow,aclr_reset,
+// 
+// 			aclr_bold,
+// 			st->st_dev,
+// 			aclr_reset,
+// 
+// 			aclr_bold,aclr_yellow,aclr_reset,
+// 
+// 			aclr_bold,
+// 			st->st_ino,
+// 			aclr_reset);
 
 	return ret;
 }
diff --git a/src/output/slbt_output_exec.c b/src/output/slbt_output_exec.c
index 187c3ac..61ac22a 100644
--- a/src/output/slbt_output_exec.c
+++ b/src/output/slbt_output_exec.c
@@ -94,6 +94,8 @@ int slbt_output_exec(
 	const struct slbt_exec_ctx *	ectx,
 	const char *			step)
 {
+	return 0;
+#if 0
 	int fdout = slbt_driver_fdout(dctx);
 
 	if (dctx->cctx->drvflags & SLBT_DRIVER_ANNOTATE_NEVER)
@@ -107,6 +109,7 @@ int slbt_output_exec(
 
 	else
 		return slbt_output_exec_plain(dctx,ectx,step);
+#endif
 }
 
 int slbt_output_compile(
diff --git a/src/output/slbt_output_fdcwd.c b/src/output/slbt_output_fdcwd.c
index 650f288..5a3a01f 100644
--- a/src/output/slbt_output_fdcwd.c
+++ b/src/output/slbt_output_fdcwd.c
@@ -42,7 +42,7 @@ static int slbt_output_fdcwd_plain(const struct slbt_driver_ctx * dctx)
 		strerror_r(errno,path,sizeof(path));
 	}
 
-	if (slbt_dprintf(
+	if (ferror && slbt_dprintf(
 			fderr,
 			"%s: %s: {.fdcwd=%s, .realpath%s=%c%s%c}.\n",
 			dctx->program,
