diff --git a/src/driver/slbt_amain.c b/src/driver/slbt_amain.c
index 65d61de252666199974835193a50405bab6f283f..1ebed5547ce8f0d04d4d682682676d40dff1227b 100644
--- a/src/driver/slbt_amain.c
+++ b/src/driver/slbt_amain.c
@@ -33,6 +33,8 @@ static const char * const slbt_ver_plain[6] = {
 		"",""
 };
 
+bool slbt_use_colours = true;
+
 static ssize_t slbt_version(struct slbt_driver_ctx * dctx, int fdout)
 {
 	const struct slbt_source_version * verinfo;
@@ -40,7 +42,7 @@ static ssize_t slbt_version(struct slbt_driver_ctx * dctx, int fdout)
 	bool gitver;
 
 	verinfo = slbt_source_version();
-	verclr  = isatty(fdout) ? slbt_ver_color : slbt_ver_plain;
+	verclr  = slbt_use_colours && isatty(fdout) ? slbt_ver_color : slbt_ver_plain;
 	gitver  = strcmp(verinfo->commit,"unknown");
 
 	return slbt_dprintf(fdout,vermsg,
@@ -101,6 +106,10 @@ int slbt_main(char ** argv, char ** envp, const struct slbt_fd_ctx * fdctx)
 	harg = (!argv || !argv[0] || !argv[1] || argv[2])
 		? 0 : argv[1];
 
+	if (getenv("SLIBTOOL_MONOCHROME")) {
+		slbt_use_colours = false;
+	}
+
 	/* --version only? */
 	if (harg && (!strcmp(harg,"--version")
 				|| !strcmp(harg,"--help-all")
diff --git a/src/driver/slbt_driver_ctx.c b/src/driver/slbt_driver_ctx.c
index d1f1cde13ad3c52cc21ef39559a6d869a3df00a7..72fe6adb463b5e6a61b269e8df41215dbaf1bc27 100644
--- a/src/driver/slbt_driver_ctx.c
+++ b/src/driver/slbt_driver_ctx.c
@@ -128,10 +128,11 @@ static void slbt_output_raw_vector(int fderr, char ** argv, char ** envp)
 	char *		dot;
 	const char *	color;
 	bool		fcolor;
+	extern bool	slbt_use_colours;
 
 	(void)envp;
 
-	if ((fcolor = isatty(fderr)))
+	if ((fcolor = (slbt_use_colours && isatty(fderr))))
 		slbt_dprintf(fderr,"%s%s",aclr_bold,aclr_red);
 
 	slbt_dprintf(fderr,"\n\n\n%s",argv[0]);
diff --git a/src/internal/argv/argv.h b/src/internal/argv/argv.h
index 5dc5adb9d71e69096d335029d247fc46a213d003..b56a3ed6eb5ef99797c9d8b15f9f309d35dcd111 100644
--- a/src/internal/argv/argv.h
+++ b/src/internal/argv/argv.h
@@ -791,11 +791,12 @@ static void argv_usage(
 	const char			ccyan []  = "\x1b[36m";
 	const char *			color     = ccyan;
 	bool				fcolor;
+	extern bool 		slbt_use_colours;
 
 	fshort = mode ? !strcmp(mode,"short") : 0;
 	flong  = fshort ? 0 : mode && !strcmp(mode,"long");
 	fboth  = !fshort && !flong;
-	fcolor = isatty(fd);
+	fcolor = slbt_use_colours && isatty(fd);
 
 	if (fcolor)
 		argv_dprintf(fd,"%s%s",cbold,cgreen);
