diff --git a/src/driver/slbt_host_params.c b/src/driver/slbt_host_params.c
index b83fa8a..e370e01 100644
--- a/src/driver/slbt_host_params.c
+++ b/src/driver/slbt_host_params.c
@@ -160,6 +160,7 @@ int slbt_init_host_params(
 	char *		machinequad[4];
 	char *		arprobeargv[4];
 	char		archivename[] = "/tmp/slibtool.ar.probe.XXXXXXXXXXXXXXXX";
+	char *	fromEnv;
 
 	/* base */
 	if ((base = strrchr(cctx->cargv[0],'/')))
@@ -294,7 +295,10 @@ int slbt_init_host_params(
 	/* ar */
 	if (host->ar)
 		cfgmeta->ar = cfgmeta_ar ? cfgmeta_ar : cfgexplicit;
-	else {
+	else if ((fromEnv = getenv("SLIBTOOL_AR"))) {
+		host->ar = drvhost->ar = strdup(fromEnv);
+		cfgmeta->ar = cfgexplicit;
+	} else {
 		if (!(drvhost->ar = calloc(1,toollen)))
 			return -1;
 
@@ -371,7 +375,10 @@ int slbt_init_host_params(
 	/* ranlib */
 	if (host->ranlib)
 		cfgmeta->ranlib = cfgmeta_ranlib ? cfgmeta_ranlib : cfgexplicit;
-	else {
+	else if ((fromEnv = getenv("SLIBTOOL_RANLIB"))) {
+		host->ranlib = drvhost->ranlib = strdup(fromEnv);
+		cfgmeta->ranlib = cfgexplicit;
+	} else {
 		if (!(drvhost->ranlib = calloc(1,toollen)))
 			return -1;
 
