diff --git a/src/logic/slbt_exec_link.c b/src/logic/slbt_exec_link.c
index 0bfe7e0432bdba0405bdd137391688f9d5222db0..52390ceee86ca47c859cdf3a206c2d7c5d686960 100644
--- a/src/logic/slbt_exec_link.c
+++ b/src/logic/slbt_exec_link.c
@@ -1208,15 +1208,22 @@ static int slbt_exec_link_create_archive(
 			arfilename,true))
 		return SLBT_NESTED_ERROR(dctx);
 
-	/* ar spawn */
-	if ((slbt_spawn(ectx,true) < 0) || ectx->exitcode)
-		return SLBT_SPAWN_ERROR(dctx);
-
-	/* input objects associated with .la archives */
-	for (parg=ectx->cargv; *parg; parg++)
-		if (slbt_adjust_wrapper_argument(*parg,true))
-			if (slbt_archive_import(dctx,ectx,output,*parg))
-				return SLBT_NESTED_ERROR(dctx);
+    if (strstr(program, "llvm-ar")) {
+        ectx->argv    = ectx->altv;
+        /* ar spawn */
+        if ((slbt_spawn(ectx,true) < 0) || ectx->exitcode)
+            return SLBT_SPAWN_ERROR(dctx);
+    } else {
+        /* ar spawn */
+        if ((slbt_spawn(ectx,true) < 0) || ectx->exitcode)
+            return SLBT_SPAWN_ERROR(dctx);
+
+        /* input objects associated with .la archives */
+        for (parg=ectx->cargv; *parg; parg++)
+            if (slbt_adjust_wrapper_argument(*parg,true))
+                if (slbt_archive_import(dctx,ectx,output,*parg))
+                    return SLBT_NESTED_ERROR(dctx);
+    }
 
 	if (fprimary && (dctx->cctx->drvflags & SLBT_DRIVER_DISABLE_SHARED)) {
 		strcpy(arlink,output);
