diff --git a/src/logic/linkcmd/slbt_linkcmd_argv.c b/src/logic/linkcmd/slbt_linkcmd_argv.c
index 440be9e..133f62b 100644
--- a/src/logic/linkcmd/slbt_linkcmd_argv.c
+++ b/src/logic/linkcmd/slbt_linkcmd_argv.c
@@ -324,10 +324,18 @@ int slbt_exec_link_adjust_argument_vector(
 	for (; *carg; ) {
 		dpath = 0;
 
+#ifndef __APPLE__
 		if (!strcmp(*carg,"-Wl,--whole-archive"))
 			fwholearchive = true;
 		else if (!strcmp(*carg,"-Wl,--no-whole-archive"))
 			fwholearchive = false;
+#else
+		// -noall_load is the default
+		if (!strcmp(*carg,"-Wl,-all_load") || !strncmp(*carg, "-Wl,-force_load", 15))
+			fwholearchive = true;
+		else
+			fwholearchive = false;
+#endif
 
 
 
@@ -378,15 +386,32 @@ int slbt_exec_link_adjust_argument_vector(
 			*aarg++ = *carg++;
 
 		} else if (!(strcmp(dot,".a"))) {
+#ifndef __APPLE__
 			if (flibrary && !fwholearchive)
 				*aarg++ = "-Wl,--whole-archive";
+#else
+			bool needslib = true;
+			if (flibrary && !fwholearchive) {
+				*aarg = NULL;
+				asprintf(aarg, "-Wl,-force_load,%s", *carg);
+				aarg++;
+				needslib = false;
+			}
+#endif
 
 			dpath = lib;
 			sprintf(lib,"%s.slibtool.deps",*carg);
+#ifndef __APPLE__
 			*aarg++ = *carg++;
 
 			if (flibrary && !fwholearchive)
 				*aarg++ = "-Wl,--no-whole-archive";
+#else
+			if (needslib) {
+				*aarg++ = *carg;
+			}
+			carg++;
+#endif
 
 		} else if (strcmp(dot,dctx->cctx->settings.dsosuffix)) {
 			*aarg++ = *carg++;
