diff --git a/src/driver/slbt_amain.c b/src/driver/slbt_amain.c
index b6bbe40..4d96370 100644
--- a/src/driver/slbt_amain.c
+++ b/src/driver/slbt_amain.c
@@ -35,6 +35,8 @@ static const char * const slbt_ver_plain[6] = {
 		"",""
 };
 
+bool slbt_use_colours = true;
+
 static ssize_t slbt_version(struct slbt_driver_ctx * dctx, int fdout)
 {
 	const struct slbt_source_version * verinfo;
@@ -42,7 +44,7 @@ static ssize_t slbt_version(struct slbt_driver_ctx * dctx, int fdout)
 	bool gitver;
 
 	verinfo = slbt_source_version();
-	verclr  = isatty(fdout) ? slbt_ver_color : slbt_ver_plain;
+	verclr  = slbt_use_colours && isatty(fdout) ? slbt_ver_color : slbt_ver_plain;
 	gitver  = strcmp(verinfo->commit,"unknown");
 
 	if (dctx->cctx->drvflags & SLBT_DRIVER_ANNOTATE_NEVER)
@@ -165,6 +167,10 @@ int slbt_main(char ** argv, char ** envp, const struct slbt_fd_ctx * fdctx)
                           | SLBT_DRIVER_DEBUG
                           | SLBT_DRIVER_LEGABITS);
 
+	if (getenv("SLIBTOOL_MONOCHROME")) {
+		slbt_use_colours = false;
+	}
+
 	/* driver context */
 	if ((ret = slbt_get_driver_ctx(argv,envp,flags|noclr,fdctx,&dctx)))
 		return (ret == SLBT_USAGE)
diff --git a/src/driver/slbt_driver_ctx.c b/src/driver/slbt_driver_ctx.c
index ec468a2..1ff48fc 100644
--- a/src/driver/slbt_driver_ctx.c
+++ b/src/driver/slbt_driver_ctx.c
@@ -66,10 +66,11 @@ static void slbt_output_raw_vector(int fderr, char ** argv, char ** envp, bool f
 	char **		parg;
 	char *		dot;
 	const char *	color;
+	extern bool	slbt_use_colours;
 
 	(void)envp;
 
-	if (fcolor)
+	if (slbt_use_colours && fcolor)
 		slbt_dprintf(fderr,"%s%s",aclr_bold,aclr_red);
 
 	slbt_dprintf(fderr,"\n\n\n%s",argv[0]);
diff --git a/src/internal/argv/argv.h b/src/internal/argv/argv.h
index d17865b..89417f9 100644
--- a/src/internal/argv/argv.h
+++ b/src/internal/argv/argv.h
@@ -851,6 +851,7 @@ static void argv_usage_impl(
 	const char			cblue []  = "\x1b[34m";
 	const char			ccyan []  = "\x1b[36m";
 	const char *			color     = ccyan;
+	extern bool 		slbt_use_colours;
 
 	(void)argv_usage;
 	(void)argv_usage_plain;
@@ -858,6 +859,8 @@ static void argv_usage_impl(
 	fshort = mode ? !strcmp(mode,"short") : 0;
 	flong  = fshort ? 0 : mode && !strcmp(mode,"long");
 	fboth  = !fshort && !flong;
+	// RJVB
+	fcolor &= slbt_use_colours;
 
 	if (fcolor)
 		argv_dprintf(fd,"%s%s",cbold,cgreen);
