diff --git a/src/arbits/output/slbt_ar_output_members.c b/src/arbits/output/slbt_ar_output_members.c
index bb6686b..d5e4f72 100644
--- a/src/arbits/output/slbt_ar_output_members.c
+++ b/src/arbits/output/slbt_ar_output_members.c
@@ -6,6 +6,9 @@
 
 #include <time.h>
 #include <locale.h>
+#ifdef __APPLE__
+#	include <printf.h>
+#endif
 #include <inttypes.h>
 #include <slibtool/slibtool.h>
 #include <slibtool/slibtool_output.h>
