diff --git a/src/driver/slbt_driver_ctx.c b/src/driver/slbt_driver_ctx.c
index 22eb0bcb4fc56764895eb6f148da6eff066f4cd3..64dcb8e654fcccb86a684550767d520be7b79114 100644
--- a/src/driver/slbt_driver_ctx.c
+++ b/src/driver/slbt_driver_ctx.c
@@ -787,10 +787,15 @@ static int slbt_init_host_params(
 	toollen =  fnative ? 0 : strlen(host->host);
 	toollen += strlen("-utility-name");
 
+	char *fromEnv;
+
 	/* ar */
 	if (host->ar)
 		cfgmeta->ar = cfgexplicit;
-	else {
+	else if ((fromEnv = getenv("SLIBTOOL_AR"))) {
+		host->ar = drvhost->ar = strdup(fromEnv);
+		cfgmeta->ar = cfgexplicit;
+	} else {
 		if (!(drvhost->ar = calloc(1,toollen)))
 			return -1;
 
@@ -863,7 +868,10 @@ static int slbt_init_host_params(
 	/* ranlib */
 	if (host->ranlib)
 		cfgmeta->ranlib = cfgexplicit;
-	else {
+	else if ((fromEnv = getenv("SLIBTOOL_RANLIB"))) {
+		host->ranlib = drvhost->ranlib = strdup(fromEnv);
+		cfgmeta->ranlib = cfgexplicit;
+	} else {
 		if (!(drvhost->ranlib = calloc(1,toollen)))
 			return -1;
 
