diff --git a/Makefile.in b/Makefile.in
index cbec8441871ca7a9f351ed1f0c5220df706d0ae1..c6a3bf66bb7d939d0d9b1ee37b62b49a2172aee6 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -144,10 +144,10 @@ install-extras:
 install-app-extras:
 
 .cflags-host:
-	@printf '%s' "$(CFLAGS)"
+	@echo -n $(CFLAGS)
 
 .cflags-native:
-	@printf '%s' "$(NATIVE_CFLAGS)"
+	@echo -n $(NATIVE_CFLAGS)
 
 
 include ./ccenv/host.mk
