diff --git a/sofort/ccenv/ccenv.sh b/sofort/ccenv/ccenv.sh
index 04db31e872c642a29667a6a7da86e415c6993fae..91ba8ad0a66e3f58af0661dc200f46f65fd972f9 100644
--- a/sofort/ccenv/ccenv.sh
+++ b/sofort/ccenv/ccenv.sh
@@ -622,6 +622,15 @@ ccenv_set_cc_binfmt()
 		&& ccenv_use_objdump=yes
 	fi
 
+	# MACHO / file
+	if [ -x "/usr/bin/file" ] && [ -z "$ccenv_cc_binfmt" ]; then
+		/usr/bin/file -h  $ccenv_image 2>/dev/null       \
+			| grep -i "$ccenv_image: Mach-O"            \
+				> /dev/null                       \
+		&& ccenv_cc_binfmt='MACHO'                        \
+		&& ccenv_use_objdump=yes
+	fi
+
 	# PE / objdump (bfd)
 	if [ -n "$ccenv_objdump" ] && [ -z "$ccenv_cc_binfmt" ]; then
 		$ccenv_objdump -h  $ccenv_image 2>/dev/null       \
