diff --git a/src/logic/slbt_exec_link.c b/src/logic/slbt_exec_link.c
index 5f94038c1ca9b1751f12ee2dd791e9ebebca7f47..13b07f3103051a03c499a6f88b414094cd8ffd1c 100644
--- a/src/logic/slbt_exec_link.c
+++ b/src/logic/slbt_exec_link.c
@@ -1214,15 +1233,24 @@ static int slbt_exec_link_create_archive(
 			arfilename,true))
 		return SLBT_NESTED_ERROR(dctx);
 
-	/* ar spawn */
-	if ((slbt_spawn(ectx,true) < 0) || ectx->exitcode)
-		return SLBT_SPAWN_ERROR(dctx);
-
-	/* input objects associated with .la archives */
-	for (parg=ectx->cargv; *parg; parg++)
-		if (slbt_adjust_wrapper_argument(*parg,true))
-			if (slbt_archive_import(dctx,ectx,output,*parg))
-				return SLBT_NESTED_ERROR(dctx);
+	if (strstr(program, "llvm-ar")) {
+		ectx->argv	= ectx->altv;
+		if (!(dctx->cctx->drvflags & SLBT_DRIVER_SILENT))
+			slbt_output_link(dctx,ectx);
+		/* ar spawn */
+		if ((slbt_spawn(ectx,true) < 0) || ectx->exitcode)
+			return SLBT_SPAWN_ERROR(dctx);
+	} else {
+		/* ar spawn */
+		if ((slbt_spawn(ectx,true) < 0) || ectx->exitcode)
+			return SLBT_SPAWN_ERROR(dctx);
+
+		/* input objects associated with .la archives */
+		for (parg=ectx->cargv; *parg; parg++)
+			if (slbt_adjust_wrapper_argument(*parg,true))
+				if (slbt_archive_import(dctx,ectx,output,*parg))
+					return SLBT_NESTED_ERROR(dctx);
+	}
 
 	if (fprimary && (dctx->cctx->drvflags & SLBT_DRIVER_DISABLE_SHARED)) {
 		strcpy(arlink,output);
