diff --git a/src/driver/slbt_amain.c b/src/driver/slbt_amain.c
index 78a09733e7ef3912ceaf5e46f8cc56af2cdb5736..0ca3cd8d2f11b4c2d76bc98b8ce58eb158b460c1 100644
--- a/src/driver/slbt_amain.c
+++ b/src/driver/slbt_amain.c
@@ -35,6 +35,8 @@ static const char * const slbt_ver_plain[6] = {
 		"",""
 };
 
+bool slbt_use_colours = true;
+
 static ssize_t slbt_version(struct slbt_driver_ctx * dctx, int fdout)
 {
 	const struct slbt_source_version * verinfo;
@@ -42,7 +44,7 @@ static ssize_t slbt_version(struct slbt_driver_ctx * dctx, int fdout)
 	bool gitver;
 
 	verinfo = slbt_source_version();
-	verclr  = isatty(fdout) ? slbt_ver_color : slbt_ver_plain;
+	verclr  = slbt_use_colours && isatty(fdout) ? slbt_ver_color : slbt_ver_plain;
 	gitver  = strcmp(verinfo->commit,"unknown");
 
 	if (dctx->cctx->drvflags & SLBT_DRIVER_ANNOTATE_NEVER)
@@ -108,6 +113,10 @@ int slbt_main(char ** argv, char ** envp, const struct slbt_fd_ctx * fdctx)
 	harg = (!argv || !argv[0] || !argv[1] || argv[2])
 		? 0 : argv[1];
 
+	if (getenv("SLIBTOOL_MONOCHROME")) {
+		slbt_use_colours = false;
+	}
+
 	/* --version only? */
 	if (harg && (!strcmp(harg,"--version")
 				|| !strcmp(harg,"--help-all")
diff --git a/src/driver/slbt_driver_ctx.c b/src/driver/slbt_driver_ctx.c
index 13efa37515d0fa1c3b2ebf5a194a299aef9106b9..385dfa22e28fcfe607ec5887908c9a4141784a58 100644
--- a/src/driver/slbt_driver_ctx.c
+++ b/src/driver/slbt_driver_ctx.c
@@ -127,10 +127,11 @@ static void slbt_output_raw_vector(int fderr, char ** argv, char ** envp, bool f
 	char **		parg;
 	char *		dot;
 	const char *	color;
+	extern bool	slbt_use_colours;
 
 	(void)envp;
 
-	if (fcolor)
+	if (slbt_use_colours && fcolor)
 		slbt_dprintf(fderr,"%s%s",aclr_bold,aclr_red);
 
 	slbt_dprintf(fderr,"\n\n\n%s",argv[0]);
diff --git a/src/internal/argv/argv.h b/src/internal/argv/argv.h
index 5496ae8027d65d9b7570d954a712197f7a4de381..99705bfd70f47d039a9d39214d6648d282a7301b 100644
--- a/src/internal/argv/argv.h
+++ b/src/internal/argv/argv.h
@@ -796,6 +796,7 @@ static void argv_usage_impl(
 	const char			cblue []  = "\x1b[34m";
 	const char			ccyan []  = "\x1b[36m";
 	const char *			color     = ccyan;
+	extern bool 		slbt_use_colours;
 
 	(void)argv_usage;
 	(void)argv_usage_plain;
@@ -803,6 +804,8 @@ static void argv_usage_impl(
 	fshort = mode ? !strcmp(mode,"short") : 0;
 	flong  = fshort ? 0 : mode && !strcmp(mode,"long");
 	fboth  = !fshort && !flong;
+	// RJVB
+	fcolor &= slbt_use_colours;
 
 	if (fcolor)
 		argv_dprintf(fd,"%s%s",cbold,cgreen);
