diff --git a/src/arbits/slbt_archive_meta.c b/src/arbits/slbt_archive_meta.c
index 8c8f999..6e67b40 100644
--- a/src/arbits/slbt_archive_meta.c
+++ b/src/arbits/slbt_archive_meta.c
@@ -85,18 +85,23 @@ static int slbt_ar_read_decimal_64(const char * mark, int len, uint64_t * dec)
 	int       i;
 	uint64_t  res;
 
-	for (; len && (mark[len-1]==AR_DEC_PADDING); )
-		len--;
+// 	for (; len && (mark[len-1]==AR_DEC_PADDING); )
+// 		len--;
 
 	for (i=0,res=0; i<len; i++) {
 		if ((mark[i] >= '0') && (mark[i] <= '9')) {
 			res *= 10;
 			res += (mark[i] - '0');
 		} else {
-			return -1;
+			break;
 		}
 	}
 
+// 	fprintf(stderr, "mark=<%s>[%d], i=%d, mark[i]='%c' res=%lu\n", mark, len, i, mark[i], res);
+	if (mark[i] != AR_DEC_PADDING) {
+		return -1;
+	}
+
 	*dec = res;
 
 	return 0;
