diff --git a/src/driver/slbt_driver_ctx.c b/src/driver/slbt_driver_ctx.c
index a898205de7dc89fa3d796fc37cae3487ba43d992..f353028db26337a2f38364cad0d1b9730493eb27 100644
--- a/src/driver/slbt_driver_ctx.c
+++ b/src/driver/slbt_driver_ctx.c
@@ -751,10 +751,15 @@ static int slbt_init_host_params(
 	toollen =  fnative ? 0 : strlen(host->host);
 	toollen += strlen("-utility-name");
 
+    char *fromEnv;
+
 	/* ar */
 	if (host->ar)
 		cfgmeta->ar = cfgexplicit;
-	else {
+    else if ((fromEnv = getenv("SLIBTOOL_AR"))) {
+        host->ar = drvhost->ar = strdup(fromEnv);
+        cfgmeta->ar = cfgexplicit;
+    } else {
 		if (!(drvhost->ar = calloc(1,toollen)))
 			return -1;
 
@@ -827,7 +832,10 @@ static int slbt_init_host_params(
 	/* ranlib */
 	if (host->ranlib)
 		cfgmeta->ranlib = cfgexplicit;
-	else {
+    else if ((fromEnv = getenv("SLIBTOOL_RANLIB"))) {
+        host->ranlib = drvhost->ranlib = strdup(fromEnv);
+        cfgmeta->ranlib = cfgexplicit;
+    } else {
 		if (!(drvhost->ranlib = calloc(1,toollen)))
 			return -1;
 
