diff --git a/src/logic/slbt_exec_link.c b/src/logic/slbt_exec_link.c
index 0bfe7e0432bdba0405bdd137391688f9d5222db0..6a2e36f650540a07dd7f616f928007253fde9f15 100644
--- a/src/logic/slbt_exec_link.c
+++ b/src/logic/slbt_exec_link.c
@@ -373,10 +375,18 @@ static int slbt_exec_link_adjust_argument_vector(
 	for (; *carg; ) {
 		dpath = 0;
 
+#ifndef __APPLE__
 		if (!strcmp(*carg,"-Wl,--whole-archive"))
 			fwholearchive = true;
 		else if (!strcmp(*carg,"-Wl,--no-whole-archive"))
 			fwholearchive = false;
+#else
+		// -noall_load is the default
+		if (!strcmp(*carg,"-Wl,-all_load"))
+			fwholearchive = true;
+		else
+			fwholearchive = false;
+#endif
 
 
 
@@ -421,15 +431,24 @@ static int slbt_exec_link_adjust_argument_vector(
 			*aarg++ = *carg++;
 
 		} else if (!(strcmp(dot,".a"))) {
+#ifndef __APPLE__
 			if (flibrary && !fwholearchive)
 				*aarg++ = "-Wl,--whole-archive";
+#else
+			if (flibrary && !fwholearchive)
+				*aarg++ = "-Wl,-all_load";
+#endif
 
 			dpath = lib;
 			sprintf(lib,"%s.slibtool.deps",*carg);
 			*aarg++ = *carg++;
 
+#ifndef __APPLE__
 			if (flibrary && !fwholearchive)
 				*aarg++ = "-Wl,--no-whole-archive";
+#else
+			// -noall_load is obsolete and ignored with a warning
+#endif
 
 		} else if (strcmp(dot,dctx->cctx->settings.dsosuffix)) {
 			*aarg++ = *carg++;
