diff --git a/src/logic/slbt_exec_link.c b/src/logic/slbt_exec_link.c
index 84d3933..70a70bc 100644
--- a/src/logic/slbt_exec_link.c
+++ b/src/logic/slbt_exec_link.c
@@ -1423,9 +1423,11 @@ static int slbt_exec_link_create_library(
 				&depsmeta) < 0)
 			return SLBT_NESTED_ERROR(dctx);
 
+#ifndef __APPLE__
 	/* --no-undefined */
 	if (dctx->cctx->drvflags & SLBT_DRIVER_NO_UNDEFINED)
 		*ectx->noundef = "-Wl,--no-undefined";
+#endif
 
 	/* -soname */
 	dot   = strrchr(dctx->cctx->output,'.');
@@ -1442,7 +1444,11 @@ static int slbt_exec_link_create_library(
 				>= sizeof(soname))
 			return SLBT_BUFFER_ERROR(dctx);
 
+#ifdef __APPLE__
+		*ectx->soname  = "-Wl,-install_name";
+#else
 		*ectx->soname  = "-Wl,-soname";
+#endif
 		*ectx->lsoname = soname;
 
 	} else if (relfilename && dctx->cctx->verinfo.verinfo) {
@@ -1456,7 +1462,11 @@ static int slbt_exec_link_create_library(
 				>= sizeof(soname))
 			return SLBT_BUFFER_ERROR(dctx);
 
+#ifdef __APPLE__
+		*ectx->soname  = "-Wl,-install_name";
+#else
 		*ectx->soname  = "-Wl,-soname";
+#endif
 		*ectx->lsoname = soname;
 
 	} else if (relfilename) {
@@ -1468,7 +1478,11 @@ static int slbt_exec_link_create_library(
 				>= sizeof(soname))
 			return SLBT_BUFFER_ERROR(dctx);
 
+#ifdef __APPLE__
+		*ectx->soname  = "-Wl,-install_name";
+#else
 		*ectx->soname  = "-Wl,-soname";
+#endif
 		*ectx->lsoname = soname;
 
 	} else if (dctx->cctx->drvflags & SLBT_DRIVER_AVOID_VERSION) {
@@ -1479,7 +1493,11 @@ static int slbt_exec_link_create_library(
 				>= sizeof(soname))
 			return SLBT_BUFFER_ERROR(dctx);
 
+#ifdef __APPLE__
+		*ectx->soname  = "-Wl,-install_name";
+#else
 		*ectx->soname  = "-Wl,-soname";
+#endif
 		*ectx->lsoname = soname;
 
 	} else {
@@ -1492,7 +1510,11 @@ static int slbt_exec_link_create_library(
 				>= sizeof(soname))
 			return SLBT_BUFFER_ERROR(dctx);
 
+#ifdef __APPLE__
+		*ectx->soname  = "-Wl,-install_name";
+#else
 		*ectx->soname  = "-Wl,-soname";
+#endif
 		*ectx->lsoname = soname;
 	}
 
@@ -1642,9 +1664,11 @@ static int slbt_exec_link_create_executable(
 				&depsmeta) < 0)
 			return SLBT_NESTED_ERROR(dctx);
 
+#ifndef __APPLE__
 	/* --no-undefined */
 	if (dctx->cctx->drvflags & SLBT_DRIVER_NO_UNDEFINED)
 		*ectx->noundef = "-Wl,--no-undefined";
+#endif
 
 	/* executable wrapper: create */
 	if ((size_t)snprintf(wrapper,sizeof(wrapper),
