diff --git a/src/driver/slbt_amain.c b/src/driver/slbt_amain.c
index b6bbe40..09fa72a 100644
--- a/src/driver/slbt_amain.c
+++ b/src/driver/slbt_amain.c
@@ -74,8 +74,11 @@ static void slbt_perform_driver_actions(struct slbt_driver_ctx * dctx)
 	if (dctx->cctx->mode == SLBT_MODE_INSTALL)
 		slbt_exec_install(dctx,0);
 
-	if (dctx->cctx->mode == SLBT_MODE_LINK)
+	if (dctx->cctx->mode == SLBT_MODE_LINK) {
+		extern void slbt_ignore_ccache(struct slbt_driver_ctx *dctx);
+		slbt_ignore_ccache(dctx);
 		slbt_exec_link(dctx,0);
+	}
 
 	if (dctx->cctx->mode == SLBT_MODE_UNINSTALL)
 		slbt_exec_uninstall(dctx,0);
diff --git a/src/logic/slbt_exec_link.c b/src/logic/slbt_exec_link.c
index 84d3933..ce82439 100644
--- a/src/logic/slbt_exec_link.c
+++ b/src/logic/slbt_exec_link.c
@@ -12,6 +12,8 @@
 #include <fcntl.h>
 #include <errno.h>
 #include <sys/stat.h>
+#include <string.h>
+#include <libgen.h>
 
 #include <slibtool/slibtool.h>
 #include "slibtool_spawn_impl.h"
@@ -2220,3 +2222,21 @@ int slbt_exec_link(
 
 	return ret;
 }
+
+void slbt_ignore_ccache(struct slbt_driver_ctx *dctx)
+{
+	char *argv0 = strdup(dctx->cctx->cargv[0]);
+	char *compiler = basename(argv0);
+	if (strcmp(compiler, "ccache") == 0 && dctx->cctx->cargv[1]) {
+		// shift the entire argument array
+		char **args;
+		args = dctx->cctx->cargv;
+		while (args[1]) {
+			*args = args[1];
+			args++;
+		}
+		// terminate the array
+		*args = NULL;
+	}
+	free(argv0);
+}
