diff --git a/src/logic/slbt_exec_link.c b/src/logic/slbt_exec_link.c
index 84d3933..544f9cd 100644
--- a/src/logic/slbt_exec_link.c
+++ b/src/logic/slbt_exec_link.c
@@ -1354,21 +1354,35 @@ static int slbt_exec_link_create_archive(
 			arfilename,true))
 		return SLBT_NESTED_ERROR(dctx);
 
-	/* ar spawn */
-	if ((slbt_spawn(ectx,true) < 0) && (ectx->pid < 0)) {
-		return SLBT_SPAWN_ERROR(dctx);
+	if (strstr(program, "llvm-ar")) {
+		ectx->argv	= ectx->altv;
+		if (!(dctx->cctx->drvflags & SLBT_DRIVER_SILENT))
+			slbt_output_link(dctx,ectx);
+		/* ar spawn */
+		if ((slbt_spawn(ectx,true) < 0) && (ectx->pid < 0) {)
+			return SLBT_SPAWN_ERROR(dctx);
+		} else if (ectx->exitcode) {
+			return SLBT_CUSTOM_ERROR(
+				dctx,
+				SLBT_ERR_AR_ERROR);
+		}
+	} else {
+		/* ar spawn */
+		if ((slbt_spawn(ectx,true) < 0) && (ectx->pid < 0)) {
+			return SLBT_SPAWN_ERROR(dctx);
 
-	} else if (ectx->exitcode) {
-		return SLBT_CUSTOM_ERROR(
-			dctx,
-			SLBT_ERR_AR_ERROR);
-	}
+		} else if (ectx->exitcode) {
+			return SLBT_CUSTOM_ERROR(
+				dctx,
+				SLBT_ERR_AR_ERROR);
+		}
 
-	/* input objects associated with .la archives */
-	for (parg=ectx->cargv; *parg; parg++)
-		if (slbt_adjust_wrapper_argument(*parg,true))
-			if (slbt_archive_import(dctx,ectx,output,*parg))
-				return SLBT_NESTED_ERROR(dctx);
+		/* input objects associated with .la archives */
+		for (parg=ectx->cargv; *parg; parg++)
+			if (slbt_adjust_wrapper_argument(*parg,true))
+				if (slbt_archive_import(dctx,ectx,output,*parg))
+					return SLBT_NESTED_ERROR(dctx);
+	}
 
 	return 0;
 }
