diff --git a/src/driver/slbt_amain.c b/src/driver/slbt_amain.c
index 65d61de252666199974835193a50405bab6f283f..9ea81322bf98cf73636d893269aa182fd537ab1b 100644
--- a/src/driver/slbt_amain.c
+++ b/src/driver/slbt_amain.c
@@ -69,8 +69,11 @@ static void slbt_perform_driver_actions(struct slbt_driver_ctx * dctx)
 	if (dctx->cctx->mode == SLBT_MODE_INSTALL)
 		slbt_exec_install(dctx,0);
 
-	if (dctx->cctx->mode == SLBT_MODE_LINK)
+	if (dctx->cctx->mode == SLBT_MODE_LINK) {
+		extern void slbt_ignore_ccache(struct slbt_driver_ctx *dctx);
+		slbt_ignore_ccache(dctx);
 		slbt_exec_link(dctx,0);
+	}
 
 	if (dctx->cctx->mode == SLBT_MODE_UNINSTALL)
 		slbt_exec_uninstall(dctx,0);
diff --git a/src/logic/slbt_exec_link.c b/src/logic/slbt_exec_link.c
index 5f94038c1ca9b1751f12ee2dd791e9ebebca7f47..90801b07b5110c2c2e4d30d72045fc51c04a34e2 100644
--- a/src/logic/slbt_exec_link.c
+++ b/src/logic/slbt_exec_link.c
@@ -11,6 +11,8 @@
 #include <fcntl.h>
 #include <errno.h>
 #include <sys/stat.h>
+#include <string.h>
+#include <libgen.h>
 
 #include <slibtool/slibtool.h>
 #include "slibtool_spawn_impl.h"
@@ -1904,3 +1906,21 @@ int slbt_exec_link(
 
 	return ret;
 }
+
+void slbt_ignore_ccache(struct slbt_driver_ctx *dctx)
+{
+	char *argv0 = strdup(dctx->cctx->cargv[0]);
+	char *compiler = basename(argv0);
+	if (strcmp(compiler, "ccache") == 0 && dctx->cctx->cargv[1]) {
+		// shift the entire argument array
+		char **args;
+		args = dctx->cctx->cargv;
+		while (args[1]) {
+			*args = args[1];
+			args++;
+		}
+		// terminate the array
+		*args = NULL;
+	}
+	free(argv0);
+}
