diff --git Makefile-options Makefile-options
index e2c9298..426877b 100644
--- Makefile-options
+++ Makefile-options
@@ -214,14 +214,14 @@ else
 	endif
 endif
 
-# Decide whether to build using Wine
-ifneq ($(BUILD_OSX),True)
-ifneq ($(BUILD_WIN32),True)
-	ifeq ($(shell type winegcc 1> /dev/null 2> /dev/null; echo $$?),0)
-		USE_WINE = True
-	endif
-endif
-endif
+# # Decide whether to build using Wine
+# ifneq ($(BUILD_OSX),True)
+# ifneq ($(BUILD_WIN32),True)
+# 	ifeq ($(shell type winegcc 1> /dev/null 2> /dev/null; echo $$?),0)
+# 		USE_WINE = True
+# 	endif
+# endif
+# endif
 
 # Set release specific options
 ifeq ($(findstring release,$(config)),release)
