diff --git a/src/core/loader.cpp b/src/core/loader.cpp
index 2440cc9..a111988 100644
--- a/src/core/loader.cpp
+++ b/src/core/loader.cpp
@@ -329,6 +329,22 @@ void Loader::loadPlugin(const QString &pluginPath)
         plugin.unload(); // don't leave it in memory
         return;
     }
+    if (qEnvironmentVariableIsSet("SONNET_USE_HUNSPELL")) {
+        if (client->name() != QStringLiteral("Hunspell")) {
+            qCWarning(SONNET_LOG_CORE) << "Ignoring" << client->name();
+            plugin.unload();
+            return;
+        }
+    }
+#ifdef Q_OS_MACOS
+    if (qEnvironmentVariableIsSet("SONNET_USE_NSSPELLCHECKER")) {
+        if (client->name() != QStringLiteral("NSSpellChecker")) {
+            qCWarning(SONNET_LOG_CORE) << "Ignoring" << client->name();
+            plugin.unload();
+            return;
+        }
+    }
+#endif
 #else
     Client *client = nullptr;
     if (pluginPath == QLatin1String("Hunspell")) {
