diff --git a/autotests/test_settings.cpp b/autotests/test_settings.cpp
index 4757547..ccb9f2e 100644
--- a/autotests/test_settings.cpp
+++ b/autotests/test_settings.cpp
@@ -42,7 +42,7 @@ void SonnetSettingsTest::initTestCase()
 
 void SonnetSettingsTest::testRestoreDoesNotSave()
 {
-    QSettings settings(QStringLiteral("KDE"), QStringLiteral("Sonnet"));
+    QSettings settings(QSettings::IniFormat, QSettings::UserScope, QStringLiteral("KDE"), QStringLiteral("Sonnet"));
     QString fileName = settings.fileName();
 
     QDateTime startTime = QFileInfo(fileName).lastModified();
@@ -76,7 +76,7 @@ void SonnetSettingsTest::testRestoreDoesNotSave()
 
 void SonnetSettingsTest::testSpellerAPIChangeSaves()
 {
-    QSettings settings(QStringLiteral("KDE"), QStringLiteral("Sonnet"));
+    QSettings settings(QSettings::IniFormat, QSettings::UserScope, QStringLiteral("KDE"), QStringLiteral("Sonnet"));
     Speller speller;
 
     // Default Language
diff --git a/src/core/settings.cpp b/src/core/settings.cpp
index 4793cd1..4745fdb 100644
--- a/src/core/settings.cpp
+++ b/src/core/settings.cpp
@@ -228,7 +228,7 @@ int Settings::disableWordErrorCount() const
 
 void Settings::save()
 {
-    QSettings settings(QStringLiteral("KDE"), QStringLiteral("Sonnet"));
+    QSettings settings(QSettings::IniFormat, QSettings::UserScope, QStringLiteral("KDE"), QStringLiteral("Sonnet"));
     settings.setValue(QStringLiteral("defaultClient"), d->defaultClient);
     settings.setValue(QStringLiteral("defaultLanguage"), d->defaultLanguage);
     settings.setValue(QStringLiteral("checkUppercase"), d->checkUppercase);
@@ -268,7 +268,7 @@ static QStringList kdeWords()
 
 void Settings::restore()
 {
-    QSettings settings(QStringLiteral("KDE"), QStringLiteral("Sonnet"));
+    QSettings settings(QSettings::IniFormat, QSettings::UserScope, QStringLiteral("KDE"), QStringLiteral("Sonnet"));
     d->defaultClient = settings.value(QStringLiteral("defaultClient"), QString()).toString();
     d->defaultLanguage = settings.value(QStringLiteral("defaultLanguage"),
                                         QLocale::system().bcp47Name()).toString();
diff --git a/src/plugins/hspell/hspelldict.cpp b/src/plugins/hspell/hspelldict.cpp
index ea59c4d..170a57f 100644
--- a/src/plugins/hspell/hspelldict.cpp
+++ b/src/plugins/hspell/hspelldict.cpp
@@ -43,7 +43,7 @@ HSpellDict::HSpellDict(const QString &lang)
         initialized = true;
     }
 
-    QSettings settings(QStringLiteral("KDE"), QStringLiteral("SonnetHSpellPlugin"));
+    QSettings settings(QSettings::IniFormat, QSettings::UserScope, QStringLiteral("KDE"), QStringLiteral("SonnetHSpellPlugin"));
     m_personalWords
         = QSet<QString>::fromList(settings.value(QStringLiteral("PersonalWords"),
                                                  QStringList()).toStringList());
@@ -135,7 +135,7 @@ bool HSpellDict::addToSession(const QString &word)
 
 void HSpellDict::storePersonalWords()
 {
-    QSettings settings(QStringLiteral("KDE"), QStringLiteral("SonnetHSpellPlugin"));
+    QSettings settings(QSettings::IniFormat, QSettings::UserScope, QStringLiteral("KDE"), QStringLiteral("SonnetHSpellPlugin"));
     settings.setValue(QStringLiteral("PersonalWords"), QVariant(m_personalWords.toList()));
     QVariantHash variantHash;
     Q_FOREACH (const QString &key, m_replacements.keys()) {
