--- qtwebengine/orig.qtwebengine.pro	2014-12-05 08:25:19.000000000 -0800
+++ qtwebengine/qtwebengine.pro	2015-02-27 13:34:37.000000000 -0800
@@ -3,6 +3,5 @@
 # As long as we are a module separate from the rest of Qt, we want to unconditionally build examples.
 # Once part of Qt 5, this should be removed and we should respect the Qt wide configuration.
 QTWEBENGINE_BUILD_PARTS = $$QT_BUILD_PARTS
-QTWEBENGINE_BUILD_PARTS *= examples
 
 load(qt_parts)
--- qtwebengine/tools/qmake/mkspecs/features/default_pre.prf.orig	2015-03-04 16:53:53.000000000 +0100
+++ qtwebengine/tools/qmake/mkspecs/features/default_pre.prf	2015-03-04 16:56:37.000000000 +0100
@@ -11,6 +11,8 @@
 
 EOL = $$escape_expand(\\n)
 
+# Call the original default_pre.
+load(default_pre)
 load(functions)
 
 # Check platform support and run config tests early enough to bail
@@ -18,6 +20,3 @@
    load(configure)
    runConfigure()
 }
-
-# Call the original default_pre.
-load(default_pre)
