diff --git qtbase/src/widgets/widgets/qmenu.cpp qtbase/src/widgets/widgets/qmenu.cpp
index 5e5126e..f008dc3 100644
--- qtbase/src/widgets/widgets/qmenu.cpp
+++ qtbase/src/widgets/widgets/qmenu.cpp
@@ -1550,8 +1550,14 @@ QAction *QMenu::addSeparator()
 QAction *QMenu::addSection(const QString &text)
 {
     QAction *action = new QAction(text, this);
+#ifndef Q_OS_OSX
     action->setSeparator(true);
     addAction(action);
+#else
+    action->setEnabled(false);
+    addAction(action);
+    addSeparator();
+#endif
     return action;
 }
 
@@ -1574,8 +1580,14 @@ QAction *QMenu::addSection(const QString &text)
 QAction *QMenu::addSection(const QIcon &icon, const QString &text)
 {
     QAction *action = new QAction(icon, text, this);
+#ifndef Q_OS_OSX
     action->setSeparator(true);
     addAction(action);
+#else
+    action->setEnabled(false);
+    addAction(action);
+    addSeparator();
+#endif
     return action;
 }
 
