diff --git a/libs/resources/KisResourceStorage.cpp b/libs/resources/KisResourceStorage.cpp
index 298b85c..684a320 100644
--- a/libs/resources/KisResourceStorage.cpp
+++ b/libs/resources/KisResourceStorage.cpp
@@ -101,7 +101,7 @@ KisResourceStorage::KisResourceStorage(const QString &location)
             d->storagePlugin.reset(KisStoragePluginRegistry::instance()->m_storageFactoryMap[StorageType::AdobeStyleLibrary]->create(location));
             d->storageType = StorageType::AdobeStyleLibrary;
             d->valid = d->storagePlugin->isValid();
-    } else if (d->location == "memory" || !QUuid::fromString(d->location).isNull() || (!d->location.isEmpty() && !fi.exists())) {
+    } else if (d->location == "memory" || !QUuid(d->location).isNull() || (!d->location.isEmpty() && !fi.exists())) {
         d->storagePlugin.reset(KisStoragePluginRegistry::instance()->m_storageFactoryMap[StorageType::Memory]->create(location));
         d->name = location;
         d->storageType = StorageType::Memory;
diff --git a/libs/widgetutils/KoFileDialog.cpp b/libs/widgetutils/KoFileDialog.cpp
index 70438ba..747b889 100644
--- a/libs/widgetutils/KoFileDialog.cpp
+++ b/libs/widgetutils/KoFileDialog.cpp
@@ -162,7 +162,7 @@ void KoFileDialog::createFileDialog()
 #endif
 
     bool optionDontUseNative;
-    if (!qEnvironmentVariable("APPIMAGE").isEmpty()) {
+    if (qEnvironmentVariableIsSet("APPIMAGE") && !qEnvironmentVariableIsSet("APPIMAGE")) {
         // AppImages don't have access to platform plugins. BUG: 447805
         optionDontUseNative = false;
     } else {
diff --git a/libs/ui/dialogs/kis_dlg_preferences.cc b/libs/ui/dialogs/kis_dlg_preferences.cc
index 07439c2..49ca321 100644
--- a/libs/ui/dialogs/kis_dlg_preferences.cc
+++ b/libs/ui/dialogs/kis_dlg_preferences.cc
@@ -322,7 +322,7 @@ GeneralTab::GeneralTab(QWidget *_parent, const char *_name)
 #endif
     m_chkNativeFileDialog->setChecked(!group.readEntry("DontUseNativeFileDialog", dontUseNative));
 
-    if (!qEnvironmentVariable("APPIMAGE").isEmpty()) {
+    if (qEnvironmentVariableIsSet("APPIMAGE") && !qEnvironmentVariableIsEmpty("APPIMAGE")) {
         // AppImages don't have access to platform plugins. BUG: 447805
         // Setting the checkbox to false is 
         m_chkNativeFileDialog->setChecked(false);
diff --git a/libs/ui/dialogs/kis_dlg_layer_style.cpp b/libs/ui/dialogs/kis_dlg_layer_style.cpp
index 3efdff2..3ebfae1 100644
--- a/libs/ui/dialogs/kis_dlg_layer_style.cpp
+++ b/libs/ui/dialogs/kis_dlg_layer_style.cpp
@@ -728,8 +728,9 @@ void StylesSelector::addNewStyle(const QString &location, KisPSDLayerStyleSP sty
     ui.cmbStyleCollections->setCurrentText(location);
     loadStyles(ui.cmbStyleCollections->currentText());
     KIS_ASSERT(m_resourceModel->resourceForId(style->resourceId()));
+    const auto cRow = m_resourceModel->indexForResource(style).row();
     ui.listStyles->setCurrentIndex(m_locationsProxyModel->mapFromSource(
-        m_resourceModel->indexForResource(style).siblingAtColumn(ui.listStyles->modelColumn())));
+        m_resourceModel->indexForResource(style).sibling(cRow, ui.listStyles->modelColumn())));
 
     notifyExternalStyleChanged(style->name(), style->uuid());
 }
diff --git a/plugins/impex/webp/dlg_webp_import.cpp b/plugins/impex/webp/dlg_webp_import.cpp
index c951e4c..9111549 100644
--- a/plugins/impex/webp/dlg_webp_import.cpp
+++ b/plugins/impex/webp/dlg_webp_import.cpp
@@ -22,7 +22,7 @@ KisDlgWebPImport::KisDlgWebPImport()
 {
     enableButtonApply(false);
 
-    setMainWidget(m_rawWidget.get());
+    setMainWidget(m_rawWidget.data());
 
     m_rawWidget->hasAnimation->setAttribute(Qt::WA_TransparentForMouseEvents);
     m_rawWidget->hasTransparency->setAttribute(Qt::WA_TransparentForMouseEvents);
diff --git a/plugins/extensions/resourcemanager/dlg_bundle_manager.cpp b/plugins/extensions/resourcemanager/dlg_bundle_manager.cpp
index 6174172..1d97031 100644
--- a/plugins/extensions/resourcemanager/dlg_bundle_manager.cpp
+++ b/plugins/extensions/resourcemanager/dlg_bundle_manager.cpp
@@ -17,7 +17,9 @@
 #include <QMessageBox>
 #include <QInputDialog>
 #include <QItemSelectionModel>
+#if QT_VERSION >= QT_VERSION_CHECK(5, 10, 0)
 #include <QStringLiteral>
+#endif
 
 
 #include <kconfiggroup.h>
