diff --git a/libs/pigment/compositeops/KoVcMultiArchBuildSupport.h b/libs/pigment/compositeops/KoVcMultiArchBuildSupport.h
index 5976508..a61e93e 100644
--- a/libs/pigment/compositeops/KoVcMultiArchBuildSupport.h
+++ b/libs/pigment/compositeops/KoVcMultiArchBuildSupport.h
@@ -11,6 +11,8 @@
 
 #ifdef HAVE_VC
 
+#include <limits>
+
 #if defined(__clang__)
 
 #pragma GCC diagnostic ignored "-Wlocal-type-template-args"
