diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2f61a45..2029f30 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -730,7 +733,7 @@ set_package_properties(ZLIB PROPERTIES
     PURPOSE "Optionally used by the G'Mic and the PSD plugins")
 macro_bool_to_01(ZLIB_FOUND HAVE_ZLIB)
 
-find_package(OpenEXR CONFIG QUIET)
+find_package(OpenEXR 2.5 CONFIG QUIET)
 if(TARGET OpenEXR::OpenEXR)
     set(OPENEXR_LIBRARIES OpenEXR::OpenEXR)
 else()
diff --git a/plugins/impex/exr/CMakeLists.txt b/plugins/impex/exr/CMakeLists.txt
index 04a46ea..198abb3 100644
--- a/plugins/impex/exr/CMakeLists.txt
+++ b/plugins/impex/exr/CMakeLists.txt
@@ -1,3 +1,6 @@
+set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++14")
+
 add_subdirectory(tests)
 
 include_directories(SYSTEM ${OPENEXR_INCLUDE_DIR} )
