diff --git a/plugins/extensions/qmic/QMic.cpp b/plugins/extensions/qmic/QMic.cpp
index e205c77..2860019 100644
--- a/plugins/extensions/qmic/QMic.cpp
+++ b/plugins/extensions/qmic/QMic.cpp
@@ -55,10 +55,12 @@ void QMic::slotQMic(bool again)
         return;
     }
 
+    QString errorString = i18n("Krita cannot launch the gmic-qt plugin. No bundled library found.");
     for (const auto &loader : offers) {
         auto *factory = qobject_cast<KPluginFactory *>(loader->instance());
         if (!factory) {
             warnPlugins << "(GMic) This is not a Krita plugin: " << loader->fileName() << loader->errorString();
+		  errorString += QStringLiteral(" ") + loader->errorString();
 
             continue;
         }
@@ -77,7 +79,7 @@ void QMic::slotQMic(bool again)
     }
 
     if (!plugin) {
-        QMessageBox::warning(qApp->activeWindow(), i18nc("@title:window", "Krita"), i18n("Krita cannot launch the gmic-qt plugin. No bundled library found."));
+        QMessageBox::warning(qApp->activeWindow(), i18nc("@title:window", "Krita"), errorString);
         return;
     }
 
