diff --git a/libs/ui/KisApplication.cpp b/libs/ui/KisApplication.cpp
index cf10ac3..1b4de69 100644
--- a/libs/ui/KisApplication.cpp
+++ b/libs/ui/KisApplication.cpp
@@ -568,6 +568,8 @@ bool KisApplication::start(const KisApplicationArguments &args)
 KisApplication::~KisApplication()
 {
     delete d;
+    // RJVB
+    KisIconUtils::clearIconCache();
 }
 
 void KisApplication::setSplashScreen(QWidget *splashScreen)
diff --git a/libs/widgetutils/kis_icon_utils.cpp b/libs/widgetutils/kis_icon_utils.cpp
index fdec007..9e0c4be 100644
--- a/libs/widgetutils/kis_icon_utils.cpp
+++ b/libs/widgetutils/kis_icon_utils.cpp
@@ -190,6 +190,8 @@ void clearIconCache() {
     #if defined CACHE_ICONS
         s_cache.clear();
     #endif
+        // RJVB
+        s_icons.clear();
 }
 
 void updateIcon(QAbstractButton *button)
