--- src/orig.kcmultidialog.cpp	2016-03-05 19:47:44.000000000 +0100
+++ src/kcmultidialog.cpp	2016-05-07 22:09:34.000000000 +0200
@@ -377,7 +377,7 @@
 
     QUrl docUrl = QUrl(QStringLiteral("help:/")).resolved(QUrl(docPath)); // same code as in KHelpClient::invokeHelp
     if (docUrl.scheme() == QLatin1String("help") || docUrl.scheme() == QLatin1String("man") || docUrl.scheme() == QLatin1String("info")) {
-        QProcess::startDetached(QStringLiteral("khelpcenter"), QStringList() << docUrl.toString());
+        QProcess::startDetached(QStringLiteral("khelpcentre5"), QStringList() << docUrl.toString());
     } else {
         QDesktopServices::openUrl(docUrl);
     }
